<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-ee1857a059c0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_github_catalyst_lib_index_-eccae9-1932eeecf006.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-318896a86f35.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-9fad2423070b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - capoony/In3RPayne_PopGenomics
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:1381D70:1C3C722:6605D0E0" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDoxMzgxRDcwOjFDM0M3MjI6NjYwNUQwRTAiLCJ2aXNpdG9yX2lkIjoiNjQ1MjI0MTg1NjIzODgxNzgwNyIsInJlZ2lvbl9lZGdlIjoiaWFkIiwicmVnaW9uX3JlbmRlciI6ImlhZCJ9" data-pjax-transient="true" name="visitor-payload">
           <meta content="aa4b4a1530a4ab63f1d501920cb5442c81a57c42ad93d3eaac6ca6ab3417f9f8" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:543525703" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="Contribute to capoony/In3RPayne_PopGenomics development by creating an account on GitHub." name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/capoony/In3RPayne_PopGenomics" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/fc35b0e6cd2a3d20034b8381d37a4f81f3962e9dd22d6d5f79230b1f63d9216b/capoony/In3RPayne_PopGenomics" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - capoony/In3RPayne_PopGenomics" name="twitter:title">
                    <meta content="Contribute to capoony/In3RPayne_PopGenomics development by creating an account on GitHub." name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/fc35b0e6cd2a3d20034b8381d37a4f81f3962e9dd22d6d5f79230b1f63d9216b/capoony/In3RPayne_PopGenomics" property="og:image">
                      <meta content="Contribute to capoony/In3RPayne_PopGenomics development by creating an account on GitHub." property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - capoony/In3RPayne_PopGenomics" property="og:title"/>
                        <meta content="https://github.com/capoony/In3RPayne_PopGenomics" property="og:url"/>
                        <meta content="Contribute to capoony/In3RPayne_PopGenomics development by creating an account on GitHub." property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="def536b5d69b8afc2117ff3aa904c09699292b9d809266f1d19b952c3b02461c" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="5aaf0d46cfb23ac0376126aa517fe17c5c6293814f8cfb7fb33598c257a2a6b9" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/capoony/In3RPayne_PopGenomics git https://github.com/capoony/In3RPayne_PopGenomics.git" name="go-import"/>
                        <meta content="12515027" name="octolytics-dimension-user_id">
                         <meta content="capoony" name="octolytics-dimension-user_login">
                          <meta content="543525703" name="octolytics-dimension-repository_id">
                           <meta content="capoony/In3RPayne_PopGenomics" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="543525703" name="octolytics-dimension-repository_network_root_id">
                               <meta content="capoony/In3RPayne_PopGenomics" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/capoony/In3RPayne_PopGenomics" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-99519581d0f8.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/capoony/In3RPayne_PopGenomics","user_id":null}}' data-hydro-click-hmac="3e7dbf5d02bd80727090850ed930cbee07953104ffc5bd2fb877a257496b472f" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fcapoony%2FIn3RPayne_PopGenomics">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=bCildNq%2B1pSZCgvipceZAyCYFhHKzv4Pg%2FdGQ7pgzFQM7%2BH5uin6W9wobzh1Ilm1sI2IwrMWmMREZP1sI7cx%2BW%2F9C6BRzYhAKcypa9KWKSalIetEPpzC2fHckg14tjYjE1yELXAfMcZxb1RxDPl0tdkHvWdv7n7gy0bld3verSHnyG%2FgPyp0Ki%2FJ4SG6XTsvFX7VV%2FSXPM%2Fz2h%2B5ShbctMVh7prYMyTb5AhlqhWP5Jql5SU%2FvAKtAZw5Rxyrae8OS0Hc8%2Bw%2Fz5cjbBMGbOh9QJVbSt2ADqCpj7t7sfZLQY0Q8qtqEzn2qPkJNHWa3PcovwSFIBy%2FTgXpweXlzc3R45dm8QLTgQKD6pUKk3fp2xEs18qZA2ap0lWS%2FPQM23Sb73wXwXd7Qb%2Fu5PnYXcPuta6wReOUaYJsYWlVNQP76wcNKBvsI4uMhFZhd9QCFcf04O%2B2zXvKJZ%2BZYxLU84oDC9yOftFGb9dfIoNKNtmDjoBhlQEskd6sYcjyc%2BzS9wH2RgITqjrXmuw5Kvk9xrrnIUboX658PugKlc8il20K--pULXerUJ15IsN3bG--PvhviKp5eXsIKE6lqG0r1w%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="capoony" data-current-repository="capoony/In3RPayne_PopGenomics" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="sEoRZ9AevYbk0kX7MO9fYUSisUy-Ag8K4DclKRkYpqloaN1qZaUmCziy_IygH3rCI7olBLBWVVBB8xJU1kfr4A" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:capoony/In3RPayne_PopGenomics">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-579940d4-8b35-4cba-87ae-e56d249f2f2d" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-579940d4-8b35-4cba-87ae-e56d249f2f2d">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="Q3HCCS1Y4XXdEoztZgmW5BwiEw+T0GKJBqkUa7AwLsf9tdf/UlfHDpcT7Z237U/8rZ7INRTLIllP4PDuY05H9w=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="+8ir6h42qe/bcOsbq8hvQ55xtqDpAwYGjiv8/XhDaZJxTwW2HhwUOJfIxPCSHTGbWxlVruu7VPqpjYmmFtDCfw==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="qMzHRcbmiQcfVIuDrPGE8bHWhvy1fKKKTVUSDBOjgrTFAm8Z5BR6ZZW5+3kKo1L/ZY1BhzhCkACdeXCrbubA1w==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="EufCqKx6S7m/ZOgfCBN8sfmnoizBKhmfdEhy3wAzocfk+gfOUOl7x3rJRoa6/wiZT+cqBWm3ltRqFMnly08N1w==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/capoony/In3RPayne_PopGenomics","user_id":null}}' data-hydro-click-hmac="3e7dbf5d02bd80727090850ed930cbee07953104ffc5bd2fb877a257496b472f" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fcapoony%2FIn3RPayne_PopGenomics">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/capoony/In3RPayne_PopGenomics","user_id":null}}' data-hydro-click-hmac="3e7dbf5d02bd80727090850ed930cbee07953104ffc5bd2fb877a257496b472f" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=capoony%2FIn3RPayne_PopGenomics">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-b96036dd-9beb-4c37-a92c-8cf9295a83d0" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-51aa4d4d-2b94-4bb1-86a5-22348f7c9368" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-51aa4d4d-2b94-4bb1-86a5-22348f7c9368" id="tooltip-b96036dd-9beb-4c37-a92c-8cf9295a83d0" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/capoony/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/capoony" rel="author">
            capoony
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/capoony/In3RPayne_PopGenomics">
            In3RPayne_PopGenomics
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/capoony/In3RPayne_PopGenomics","user_id":null}}' data-hydro-click-hmac="b71c6b511e43e26e7b46e44539a812779133ba5dfc5fdef3c9b50ef9f74530aa" data-view-component="true" href="/login?return_to=%2Fcapoony%2FIn3RPayne_PopGenomics" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":543525703,"auth_type":"LOG_IN","originating_url":"https://github.com/capoony/In3RPayne_PopGenomics","user_id":null}}' data-hydro-click-hmac="1681c59e4a949d8bcd32d01eddce2184c4db834f865e52999db1a5b7ae517ab0" data-view-component="true" href="/login?return_to=%2Fcapoony%2FIn3RPayne_PopGenomics" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
             0
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":543525703,"auth_type":"LOG_IN","originating_url":"https://github.com/capoony/In3RPayne_PopGenomics","user_id":null}}' data-hydro-click-hmac="e9318306ef9cf2a8cc86073697748d3a62792672bc6857b390961e2f222147f9" data-view-component="true" href="/login?return_to=%2Fcapoony%2FIn3RPayne_PopGenomics" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
              0
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <h3 class="sr-only">
          License
         </h3>
         <div class="mb-2">
          <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="/capoony/In3RPayne_PopGenomics/blob/master/LICENSE">
           <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
            </path>
           </svg>
           MIT license
          </a>
         </div>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/capoony/In3RPayne_PopGenomics/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           stars
          </a>
          <a class="Link--secondary no-underline mr-3" href="/capoony/In3RPayne_PopGenomics/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/capoony/In3RPayne_PopGenomics/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/capoony/In3RPayne_PopGenomics/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/capoony/In3RPayne_PopGenomics/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":543525703,"auth_type":"LOG_IN","originating_url":"https://github.com/capoony/In3RPayne_PopGenomics","user_id":null}}' data-hydro-click-hmac="e9318306ef9cf2a8cc86073697748d3a62792672bc6857b390961e2f222147f9" data-view-component="true" href="/login?return_to=%2Fcapoony%2FIn3RPayne_PopGenomics" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/capoony/In3RPayne_PopGenomics","user_id":null}}' data-hydro-click-hmac="b71c6b511e43e26e7b46e44539a812779133ba5dfc5fdef3c9b50ef9f74530aa" data-view-component="true" href="/login?return_to=%2Fcapoony%2FIn3RPayne_PopGenomics" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /capoony/In3RPayne_PopGenomics" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/capoony/In3RPayne_PopGenomics" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /capoony/In3RPayne_PopGenomics/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /capoony/In3RPayne_PopGenomics/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /capoony/In3RPayne_PopGenomics/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /capoony/In3RPayne_PopGenomics/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /capoony/In3RPayne_PopGenomics/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/capoony/In3RPayne_PopGenomics/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /capoony/In3RPayne_PopGenomics/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-a3e765b7-8ba9-4d82-becf-fa262aca51b7-list" aria-haspopup="true" aria-labelledby="tooltip-5b5a4d44-98fd-46c8-87b0-f4f8b95a8d13" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-a3e765b7-8ba9-4d82-becf-fa262aca51b7-button" popovertarget="action-menu-a3e765b7-8ba9-4d82-becf-fa262aca51b7-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-a3e765b7-8ba9-4d82-becf-fa262aca51b7-button" id="tooltip-5b5a4d44-98fd-46c8-87b0-f4f8b95a8d13" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-a3e765b7-8ba9-4d82-becf-fa262aca51b7-button" anchor-offset="normal" data-view-component="true" id="action-menu-a3e765b7-8ba9-4d82-becf-fa262aca51b7-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-a3e765b7-8ba9-4d82-becf-fa262aca51b7-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-a3e765b7-8ba9-4d82-becf-fa262aca51b7-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/capoony/In3RPayne_PopGenomics" id="item-8a4077c6-511e-4c48-b6c5-70db41ff5589" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/issues" id="item-09d9b64f-988f-4e8b-b0b5-be013f551167" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/pulls" id="item-515b14e5-dd31-4e0a-bfea-0ebb83fa3989" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/actions" id="item-5527d4e9-2546-41f7-95b7-814a0f72d281" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/projects" id="item-cd33e3ff-a9fd-44fc-84fb-a738db17dd4c" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/security" id="item-7590263a-3cb3-49a0-9798-0b5574792c1a" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/pulse" id="item-b1fbb24f-2daa-4895-b860-1e515f8ae3a4" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         capoony/In3RPayne_PopGenomics
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/capoony/In3RPayne_PopGenomics/spoofed_commit_check/f55f5e2fc3f9a02f64dfbc12454f053bbf468b9b">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":543525703,"defaultBranch":"master","name":"In3RPayne_PopGenomics","ownerLogin":"capoony","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2022-09-30T09:31:26.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/12515027?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"master","listCacheKey":"v0:1676199120.508091","canEdit":false,"refType":"branch","currentOid":"f55f5e2fc3f9a02f64dfbc12454f053bbf468b9b"},"tree":{"items":[{"name":"data","path":"data","contentType":"directory"},{"name":"scripts","path":"scripts","contentType":"directory"},{"name":"shell","path":"shell","contentType":"directory"},{"name":".DS_Store","path":".DS_Store","contentType":"file"},{"name":"LICENSE","path":"LICENSE","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":6,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/capoony/In3RPayne_PopGenomics/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/capoony/In3RPayne_PopGenomics.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone capoony/In3RPayne_PopGenomics","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/capoony/In3RPayne_PopGenomics/archive/refs/heads/master.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=543525703"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"41","overviewFiles":[{"displayName":"README.md","repoName":"In3RPayne_PopGenomics","refName":"master","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eBioinformatic pipeline for population genomic inference of \u003cem\u003eIn(3R)Payne\u003c/em\u003e\u003c/h1\u003e\u003ca id=\"user-content-bioinformatic-pipeline-for-population-genomic-inference-of-in3rpayne\" class=\"anchor\" aria-label=\"Permalink: Bioinformatic pipeline for population genomic inference of In(3R)Payne\" href=\"#bioinformatic-pipeline-for-population-genomic-inference-of-in3rpayne\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003esee Material and Methods in \u003ca href=\"https://www.biorxiv.org/content/10.1101/2023.01.31.526462v1\" rel=\"nofollow\"\u003eKapun \u003cem\u003eet al.\u003c/em\u003e (2023)\u003c/a\u003e for more details. Most scripts were originally written in Python v.2. They were lifted over to Python v.3 using the \u003ca href=\"https://docs.python.org/3/library/2to3.html\" rel=\"nofollow\"\u003e\u003ccode\u003e2to3\u003c/code\u003e\u003c/a\u003e tool and storded in the /scripts/v3 folder. The original files can be found in the /scripts folder.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1) Map and generate phased sequencing data\u003c/h2\u003e\u003ca id=\"user-content-1-map-and-generate-phased-sequencing-data\" class=\"anchor\" aria-label=\"Permalink: 1) Map and generate phased sequencing data\" href=\"#1-map-and-generate-phased-sequencing-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1.1) USA\u003c/h3\u003e\u003ca id=\"user-content-11-usa\" class=\"anchor\" aria-label=\"Permalink: 1.1) USA\" href=\"#11-usa\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFor each of the newly sequenced library in the USA, test the quality with FASTQC, trim the raw reads with cutadapt, map the reads with bbmap, sort and deduplicate the BAM file with picard and realign around InDels with GATK\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"mkdir /data/mapping\n\n## loop through raw reads in folder /data and store BAM in /data/mapping\nfor file in /data/USA/*_R1.fq.gz\ndo\n\n    tmp=${file##*/}\n    i=${tmp%%_*}\n\n    sh shell/mapping.sh \\\n        /data/USA/${i}_R1.fq.gz\t\\\n        /data/USA/${i}_R2.fq.gz\t\\\t\n        ${i} \\\n        /data/USA/mapping/${i} \\\n        bbmap\n\ndone \"\u003e\u003cpre\u003emkdir /data/mapping\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# loop through raw reads in folder /data and store BAM in /data/mapping\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e /data/USA/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_R1.fq.gz\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\n    tmp=\u003cspan class=\"pl-smi\"\u003e${file\u003cspan class=\"pl-k\"\u003e##*/\u003c/span\u003e}\u003c/span\u003e\n    i=\u003cspan class=\"pl-smi\"\u003e${tmp\u003cspan class=\"pl-k\"\u003e%%\u003c/span\u003e_\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e}\u003c/span\u003e\n\n    sh shell/mapping.sh \\\n        /data/USA/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e_R1.fq.gz\t\\\n        /data/USA/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e_R2.fq.gz\t\u003cspan class=\"pl-cce\"\u003e\\\t\u003c/span\u003e\n        \u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \\\n        /data/USA/mapping/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \\\n        bbmap\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFollowing the approach in \u003ca href=\"https://onlinelibrary.wiley.com/doi/full/10.1111/mec.12594\" rel=\"nofollow\"\u003eKapun \u003cem\u003eet al.\u003c/em\u003e (2014)\u003c/a\u003e we bioinformatically obtained haploid genomes from hemiclones. This requires \u003ca href=\"https:/sourceforge.net/p/popoolation2/wiki/Main/\"\u003ePoPoolation2\u003c/a\u003e from (\u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3232374/\" rel=\"nofollow\"\u003eKofler et al. 2011\u003c/a\u003e)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"mkdir /data/USA\nmkdir /data/consensus\n\n## synchronize as MPILEUP based on hologenome from Kapun et al. (2020) only including 3L and 3R\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/USA_BAM.txt \\\n    -l /data/regions.bed.txt \u0026gt; /data/USA/USA.mpileup\n\n## use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format\njava -jar /scripts/popoolation2_1201/mpileup2sync.jar \\\n    -input /data/USA/USA.mpileup \\\n    -threads 20 \\\n    -output /data/USA/USA.sync\n\n## calculate max coverage threshold (see Kapun et al. 2014)\npython3 /scripts/v3/max_coverage.py \\\n    -input /data/USA/USA.sync \\\n    -max-coverage 0.05 \\\n    -output /data/USA/USA.cov\n\n## use GNU parallel to parallelize the phasing (see Kapun et al. 2014 for details)\nparallel -a /data/USA/USA.sync \\\n    -k \\\n    -pipepart \\\n    -cat python3 /scripts/v3/extract_consensus.py \\\n        -input {} \\\n        -min-coverage 10 \\\n        -min-count 10 \\\n        -max-coverage /data/USA/USA.cov \\\n        -output /data/consensus/USA \\\n        | gzip \u0026gt; /data/consensus/USA.consensus.gz\n\n ## convert consensus to VCF and retain site if \u0026gt; 50% of all samples with non-N's\npython3 /scripts/v3/cons2vcf.py \\\n    -input /data/consensus/USA.consensus.gz \\\n    -output /data/consensus/America \\\n    -ind 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58 \\\n    -N-cutoff 0.5 \\\n    -names Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81\"\u003e\u003cpre\u003emkdir /data/USA\nmkdir /data/consensus\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# synchronize as MPILEUP based on hologenome from Kapun et al. (2020) only including 3L and 3R\u003c/span\u003e\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/USA_BAM.txt \\\n    -l /data/regions.bed.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/USA/USA.mpileup\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format\u003c/span\u003e\njava -jar /scripts/popoolation2_1201/mpileup2sync.jar \\\n    -input /data/USA/USA.mpileup \\\n    -threads 20 \\\n    -output /data/USA/USA.sync\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# calculate max coverage threshold (see Kapun et al. 2014)\u003c/span\u003e\npython3 /scripts/v3/max_coverage.py \\\n    -input /data/USA/USA.sync \\\n    -max-coverage 0.05 \\\n    -output /data/USA/USA.cov\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# use GNU parallel to parallelize the phasing (see Kapun et al. 2014 for details)\u003c/span\u003e\nparallel -a /data/USA/USA.sync \\\n    -k \\\n    -pipepart \\\n    -cat python3 /scripts/v3/extract_consensus.py \\\n        -input {} \\\n        -min-coverage 10 \\\n        -min-count 10 \\\n        -max-coverage /data/USA/USA.cov \\\n        -output /data/consensus/USA \\\n        \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/USA.consensus.gz\n\n \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# convert consensus to VCF and retain site if \u0026gt; 50% of all samples with non-N's\u003c/span\u003e\npython3 /scripts/v3/cons2vcf.py \\\n    -input /data/consensus/USA.consensus.gz \\\n    -output /data/consensus/America \\\n    -ind 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58 \\\n    -N-cutoff 0.5 \\\n    -names Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1.2) Worldwide samples from DGN dataset (see Lack, \u003cem\u003eet al.\u003c/em\u003e \u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5100052/\" rel=\"nofollow\"\u003e2016\u003c/a\u003e)\u003c/h3\u003e\u003ca id=\"user-content-12-worldwide-samples-from-dgn-dataset-see-lack-et-al-2016\" class=\"anchor\" aria-label=\"Permalink: 1.2) Worldwide samples from DGN dataset (see Lack, et al.2016)\" href=\"#12-worldwide-samples-from-dgn-dataset-see-lack-et-al-2016\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nmkdir /data/DGNs\n\n## download raw data from SRA and map with same pipeline as above\nwhile IFS=',' read -r name SRA\ndo \n    sh /shell/obtain-n-map-africa.sh \\\n        /data/DGN/\n        ${name} \\\n        ${SRA}\ndone \u0026lt; /data/DGN_SRA.txt\n\n## now merge all strains to a big mpileup\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/DGN_BAM.txt \n    -l /data/regions.bed \\\n    | gzip \u0026gt; /data/DGN/DGN.mpileup.gz\n\n## convert major allele to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)\ngunzip -c /data/DGN/DGN.mpileup.gz \\\n    | parallel \\\n    -pipe  \\\n    -k \\\n    -cat python3 /scripts/v3/mpileup2cons.py \\\n        -c 10,200 \\\n        -m {} \\\n        -u 0.9 \\\n        -b 20 | gzip \u0026gt; /data/consensus/DGN.consensus.gz\n\"\u003e\u003cpre\u003emkdir /data/DGNs\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# download raw data from SRA and map with same pipeline as above\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e IFS=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e -r name SRA\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e \n    sh /shell/obtain-n-map-africa.sh \\\n        /data/DGN/\n        \u003cspan class=\"pl-smi\"\u003e${name}\u003c/span\u003e \\\n        \u003cspan class=\"pl-smi\"\u003e${SRA}\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e /data/DGN_SRA.txt\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# now merge all strains to a big mpileup\u003c/span\u003e\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/DGN_BAM.txt \n    -l /data/regions.bed \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/DGN/DGN.mpileup.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# convert major allele to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)\u003c/span\u003e\ngunzip -c /data/DGN/DGN.mpileup.gz \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e parallel \\\n    -pipe  \\\n    -k \\\n    -cat python3 /scripts/v3/mpileup2cons.py \\\n        -c 10,200 \\\n        -m {} \\\n        -u 0.9 \\\n        -b 20 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/DGN.consensus.gz\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1.3) Zambia from DGN dataset (see Lack, \u003cem\u003eet al.\u003c/em\u003e \u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5100052/\" rel=\"nofollow\"\u003e2016\u003c/a\u003e)\u003c/h3\u003e\u003ca id=\"user-content-13-zambia-from-dgn-dataset-see-lack-et-al-2016\" class=\"anchor\" aria-label=\"Permalink: 1.3) Zambia from DGN dataset (see Lack, et al.2016)\" href=\"#13-zambia-from-dgn-dataset-see-lack-et-al-2016\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nmkdir /data/Zambia\n\n## download raw data from SRA and map with same pipeline as above\nwhile IFS=',' read -r name SRA\ndo \n    sh /shell/obtain-n-map-africa.sh \\\n        /data/Zambia/\n        ${name} \\\n        ${SRA}\ndone \u0026lt; /data/Zambia_SRA.txt\n\n## now merge all Zambian strains to a big mpileup\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/Zambia_BAM.txt \n    -l /data/regions.bed \\\n    | gzip \u0026gt; /data/Zambia/Zambia.mpileup.gz\n\n## convert to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)\ngunzip -c /data/Zambia/Zambia.mpileup.gz \\\n    | parallel \\\n    -pipe  \\\n    -k \\\n    -cat python3 /scripts/v3/mpileup2cons.py \\\n        -c 10,200 \\\n        -m {} \\\n        -u 0.9 \\\n        -b 20 | gzip \u0026gt; /data/consensus/Zambia.consensus.gz\n\n## convert consensus to VCF and retain site if \u0026gt; 50% of all samples with non-N's\npython3 /scripts/v3/cons2vcf.py \\\n    -input /data/consensus/Zambia.consensus.gz \\\n    -output /data/consensus/Africa \\\n    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162 \\\n    -N-cutoff 0.5 \\\n    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388\"\u003e\u003cpre\u003emkdir /data/Zambia\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# download raw data from SRA and map with same pipeline as above\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e IFS=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e -r name SRA\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e \n    sh /shell/obtain-n-map-africa.sh \\\n        /data/Zambia/\n        \u003cspan class=\"pl-smi\"\u003e${name}\u003c/span\u003e \\\n        \u003cspan class=\"pl-smi\"\u003e${SRA}\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e /data/Zambia_SRA.txt\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# now merge all Zambian strains to a big mpileup\u003c/span\u003e\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/Zambia_BAM.txt \n    -l /data/regions.bed \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/Zambia/Zambia.mpileup.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# convert to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)\u003c/span\u003e\ngunzip -c /data/Zambia/Zambia.mpileup.gz \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e parallel \\\n    -pipe  \\\n    -k \\\n    -cat python3 /scripts/v3/mpileup2cons.py \\\n        -c 10,200 \\\n        -m {} \\\n        -u 0.9 \\\n        -b 20 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/Zambia.consensus.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# convert consensus to VCF and retain site if \u0026gt; 50% of all samples with non-N's\u003c/span\u003e\npython3 /scripts/v3/cons2vcf.py \\\n    -input /data/consensus/Zambia.consensus.gz \\\n    -output /data/consensus/Africa \\\n    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162 \\\n    -N-cutoff 0.5 \\\n    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1.4) Portugal (see Kapun, \u003cem\u003eet al.\u003c/em\u003e \u003ca href=\"https://onlinelibrary.wiley.com/doi/full/10.1111/mec.12594\" rel=\"nofollow\"\u003e2014\u003c/a\u003e and Franssen, \u003cem\u003eet al.\u003c/em\u003e \u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5100052/\" rel=\"nofollow\"\u003e2016\u003c/a\u003e)\u003c/h3\u003e\u003ca id=\"user-content-14-portugal-see-kapun-et-al-2014-and-franssen-et-al-2016\" class=\"anchor\" aria-label=\"Permalink: 1.4) Portugal (see Kapun, et al.2014 and Franssen, et al.2016)\" href=\"#14-portugal-see-kapun-et-al-2014-and-franssen-et-al-2016\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nmkdir /data/Portugal\n\n## get and map data from Franssen et al. 2014\nwhile IFS=',' read -r name FWD REV\ndo \n    sh /shell/obtain-n-map-portugal.sh \\\n        /data/Portugal /\n        ${name} \\\n        ${FWD} \\\n        ${REV}\ndone \u0026lt; /data/Portugal_SRA.txt\n\n## get and map data from Kapun et al. 2016\nwhile IFS=',' read -r name SRA\ndo \n    sh /shell/obtain-n-map-africa.sh \\\n        /data/Portugal /\n        ${name} \\\n        ${SRA}\ndone \u0026lt; /data/Portugal_SRA2.txt\n\n## now merge all Portugese strains to a big mpileup\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/Portugal_BAM.txt \n    -l /data/regions.bed \\\n    | gzip \u0026gt; /data/Portugal/Portugal.mpileup.gz\n\n## use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format\njava -jar /scripts/popoolation2_1201/mpileup2sync.jar \\\n    -input /data/Portugal/Portugal.mpileup \\\n    -threads 20 \\\n    -output /data/Portugal/Portugal.sync\n\n## calculate max coverage threshold (see Kapun et al. 2014)\npython3 /scripts/v3/max_coverage.py \\\n    -input /data/Portugal/Portugal.sync \\\n    -max-coverage 0.05 \\\n    -output /data/Portugal/Portugal.cov\n\n## use GNU parallel to parallelize the phasing (see Kapun et al. 2014 for details)\nparallel -a /data/Portugal/Portugal.sync \\\n    -k \\\n    -pipepart \\\n    -cat python3 /scripts/v3/extract_consensus.py \\\n        -input {} \\\n        -min-coverage 10 \\\n        -min-count 10 \\\n        -max-coverage /data/Portugal/Portugal.cov \\\n        -output /data/consensus/Portugal \\\n        | gzip \u0026gt; /data/consensus/Portugal.consensus.gz\n\n## convert consensus to VCF and retain site if \u0026gt; 50% of all samples with non-N's\npython3 /scripts/v3/cons2vcf.py \\\n    -input /data/consensus/Portugal.consensus.gz \\\n    -output /data/consensus/Portugal \\\n    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26 \\\n    -N-cutoff 0.5 \\\n    -names F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0\"\u003e\u003cpre\u003emkdir /data/Portugal\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# get and map data from Franssen et al. 2014\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e IFS=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e -r name FWD REV\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e \n    sh /shell/obtain-n-map-portugal.sh \\\n        /data/Portugal /\n        \u003cspan class=\"pl-smi\"\u003e${name}\u003c/span\u003e \\\n        \u003cspan class=\"pl-smi\"\u003e${FWD}\u003c/span\u003e \\\n        \u003cspan class=\"pl-smi\"\u003e${REV}\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e /data/Portugal_SRA.txt\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# get and map data from Kapun et al. 2016\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e IFS=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e -r name SRA\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e \n    sh /shell/obtain-n-map-africa.sh \\\n        /data/Portugal /\n        \u003cspan class=\"pl-smi\"\u003e${name}\u003c/span\u003e \\\n        \u003cspan class=\"pl-smi\"\u003e${SRA}\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e /data/Portugal_SRA2.txt\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# now merge all Portugese strains to a big mpileup\u003c/span\u003e\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/Portugal_BAM.txt \n    -l /data/regions.bed \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/Portugal/Portugal.mpileup.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format\u003c/span\u003e\njava -jar /scripts/popoolation2_1201/mpileup2sync.jar \\\n    -input /data/Portugal/Portugal.mpileup \\\n    -threads 20 \\\n    -output /data/Portugal/Portugal.sync\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# calculate max coverage threshold (see Kapun et al. 2014)\u003c/span\u003e\npython3 /scripts/v3/max_coverage.py \\\n    -input /data/Portugal/Portugal.sync \\\n    -max-coverage 0.05 \\\n    -output /data/Portugal/Portugal.cov\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# use GNU parallel to parallelize the phasing (see Kapun et al. 2014 for details)\u003c/span\u003e\nparallel -a /data/Portugal/Portugal.sync \\\n    -k \\\n    -pipepart \\\n    -cat python3 /scripts/v3/extract_consensus.py \\\n        -input {} \\\n        -min-coverage 10 \\\n        -min-count 10 \\\n        -max-coverage /data/Portugal/Portugal.cov \\\n        -output /data/consensus/Portugal \\\n        \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/Portugal.consensus.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# convert consensus to VCF and retain site if \u0026gt; 50% of all samples with non-N's\u003c/span\u003e\npython3 /scripts/v3/cons2vcf.py \\\n    -input /data/consensus/Portugal.consensus.gz \\\n    -output /data/consensus/Portugal \\\n    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26 \\\n    -N-cutoff 0.5 \\\n    -names F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1.5) Sweden (see Kapopoulou, \u003cem\u003eet al.\u003c/em\u003e \u003ca href=\"https://www.nature.com/articles/s41598-020-79720-1\" rel=\"nofollow\"\u003e2020\u003c/a\u003e)\u003c/h3\u003e\u003ca id=\"user-content-15-sweden-see-kapopoulou-et-al-2020\" class=\"anchor\" aria-label=\"Permalink: 1.5) Sweden (see Kapopoulou, et al.2020)\" href=\"#15-sweden-see-kapopoulou-et-al-2020\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUse \u003ca href=\"https:/www.ibm.com/aspera/connect/\"\u003eAspera\u003c/a\u003e and \u003ca href=\"https:/www.ncbi.nlm.nih.gov/sra/docs/sradownload/\"\u003esratoolkit\u003c/a\u003e to download and convert raw data\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nmkdir /data/Sweden\n\n## get and map data \nwhile IFS=$' \\t\\n'\nread -r A B C D E F SRA H name remainder\ndo\n\n## used the ASPERA app to download from SRA \nAspera\\ Connect.app/Contents/Resources/ascp \\\n    -i Aspera\\ Connect.app/Contents/Resources/asperaweb_id_dsa.openssh \\\n    -T anonftp@ftp.ncbi.nlm.nih.gov:/sra/sra-instant/reads/ByRun/sra/SRR/${SRA:0:6}/$SRA/$SRA.sra \\\n    /data/Sweden/\n\n## convert to FASTQ\nscripts/sratoolkit.2.9.2-mac64/bin/fastq-dump.2.5.2 \\\n    -gzip \\\n    -split-3 \\\n    -outdir /data/Sweden/ \\\n    -A $name \\\n    /data/Sweden/$SRA.sra\n\nrm /data/Sweden/$SRA.sra\n\nmkdir -p /data/Sweden/$name\n\n## trim with cutadapt (min base quality \u0026gt;18, minium trimmed readlength 75bp) and map with bbmap (standard parameters) against D. mel v. 6.04\nsh shell/mapping.sh \\\n    /data/Sweden/${name}_1.fastq.gz \\\n    /data/Sweden/${name}_2.fastq.gz \\\n    $name \\\n    /data/Sweden/$name \\\n    bbmap\n\ndone \u0026lt; /data/Sweden/Sweden_SRA.txt\n\n## now merge all Portugese strains to a big mpileup\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/Sweden_BAM.txt \n    -l /data/regions.bed \\\n    | gzip \u0026gt; /data/Sweden/Sweden.mpileup.gz\n\n## convert to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)\ngunzip -c /data/Sweden/Sweden.mpileup.gz \\\n    | parallel \\\n    -pipe  \\\n    -k \\\n    -cat python3 /scripts/v3/mpileup2cons.py \\\n        -c 10,200 \\\n        -m {} \\\n        -u 0.9 \\\n        -b 20 | gzip \u0026gt; /data/consensus/Sweden.consensus.gz\n\n## Now merge data from Portugal and Sweden\npython3 /scripts/merge_consensus.py \\\n    -consensus /data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz \\\n    | gzip \u0026gt; /data/consensus/Europe.cons.gz\n\n## convert consensus to VCF and retain site if \u0026gt; 50% of all samples with non-N's\npython3 /scripts/v3/cons2vcf.py \\\n    -input /data/consensus/Europe.cons.gz \\\n    -output /data/consensus/Europe \\\n    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40 \\\n    -N-cutoff 0.5 \\\n    -names F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94\"\u003e\u003cpre\u003emkdir /data/Sweden\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# get and map data \u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e IFS=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$'\u003c/span\u003e \u003cspan class=\"pl-cce\"\u003e\\t\\n\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e -r A B C D E F SRA H name remainder\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# used the ASPERA app to download from SRA \u003c/span\u003e\nAspera\u003cspan class=\"pl-cce\"\u003e\\ \u003c/span\u003eConnect.app/Contents/Resources/ascp \\\n    -i Aspera\u003cspan class=\"pl-cce\"\u003e\\ \u003c/span\u003eConnect.app/Contents/Resources/asperaweb_id_dsa.openssh \\\n    -T anonftp@ftp.ncbi.nlm.nih.gov:/sra/sra-instant/reads/ByRun/sra/SRR/\u003cspan class=\"pl-smi\"\u003e${SRA\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e0\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e6}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$SRA\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$SRA\u003c/span\u003e.sra \\\n    /data/Sweden/\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# convert to FASTQ\u003c/span\u003e\nscripts/sratoolkit.2.9.2-mac64/bin/fastq-dump.2.5.2 \\\n    -gzip \\\n    -split-3 \\\n    -outdir /data/Sweden/ \\\n    -A \u003cspan class=\"pl-smi\"\u003e$name\u003c/span\u003e \\\n    /data/Sweden/\u003cspan class=\"pl-smi\"\u003e$SRA\u003c/span\u003e.sra\n\nrm /data/Sweden/\u003cspan class=\"pl-smi\"\u003e$SRA\u003c/span\u003e.sra\n\nmkdir -p /data/Sweden/\u003cspan class=\"pl-smi\"\u003e$name\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# trim with cutadapt (min base quality \u0026gt;18, minium trimmed readlength 75bp) and map with bbmap (standard parameters) against D. mel v. 6.04\u003c/span\u003e\nsh shell/mapping.sh \\\n    /data/Sweden/\u003cspan class=\"pl-smi\"\u003e${name}\u003c/span\u003e_1.fastq.gz \\\n    /data/Sweden/\u003cspan class=\"pl-smi\"\u003e${name}\u003c/span\u003e_2.fastq.gz \\\n    \u003cspan class=\"pl-smi\"\u003e$name\u003c/span\u003e \\\n    /data/Sweden/\u003cspan class=\"pl-smi\"\u003e$name\u003c/span\u003e \\\n    bbmap\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e /data/Sweden/Sweden_SRA.txt\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# now merge all Portugese strains to a big mpileup\u003c/span\u003e\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/Sweden_BAM.txt \n    -l /data/regions.bed \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/Sweden/Sweden.mpileup.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# convert to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)\u003c/span\u003e\ngunzip -c /data/Sweden/Sweden.mpileup.gz \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e parallel \\\n    -pipe  \\\n    -k \\\n    -cat python3 /scripts/v3/mpileup2cons.py \\\n        -c 10,200 \\\n        -m {} \\\n        -u 0.9 \\\n        -b 20 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/Sweden.consensus.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Now merge data from Portugal and Sweden\u003c/span\u003e\npython3 /scripts/merge_consensus.py \\\n    -consensus /data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/Europe.cons.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# convert consensus to VCF and retain site if \u0026gt; 50% of all samples with non-N's\u003c/span\u003e\npython3 /scripts/v3/cons2vcf.py \\\n    -input /data/consensus/Europe.cons.gz \\\n    -output /data/consensus/Europe \\\n    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40 \\\n    -N-cutoff 0.5 \\\n    -names F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1.6) Australia\u003c/h3\u003e\u003ca id=\"user-content-16-australia\" class=\"anchor\" aria-label=\"Permalink: 1.6) Australia\" href=\"#16-australia\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe downloaded the raw sequencing data of \u003ca href=\"https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.13161\" rel=\"nofollow\"\u003eRane \u003cem\u003eet al.\u003c/em\u003e 2015\u003c/a\u003e from SRA (accession: \u003ca href=\"https:/www.ncbi.nlm.nih.gov/bioproject/PRJNA221876/\"\u003ePRJNA221876\u003c/a\u003e) and then classified the karyotpyes as follows:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nmkdir /data/Australia\n\n## loop through raw reads in folder /data and store BAM in /data/mapping\nfor file in /data/Australia/*_R1.fq.gz\ndo\n\n    tmp=${file##*/}\n    i=${tmp%%_*}\n\n    sh shell/mapping.sh \\\n        /data/Australia/${i}_1.fq.gz\t\\\n        /data/Australia/${i}_2.fq.gz\t\\\t\n        ${i} \\\n        /data/Australia/mapping/${i} \\\n        bbmap\n\ndone \n\n## synchronize as MPILEUP based on hologenome from Kapun et al. (2020) only including 3L and 3R\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/Australia_BAM.txt \\\n    -l /data/regions.bed.txt \u0026gt; /data/Australia/Australia.mpileup\n\n## use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format\njava -jar /scripts/popoolation2_1201/mpileup2sync.jar \\\n    -input /data/Australia/Australia.mpileup \\\n    -threads 20 \\\n    -output /data/Australia/Australia.sync\n\"\u003e\u003cpre\u003emkdir /data/Australia\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# loop through raw reads in folder /data and store BAM in /data/mapping\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e /data/Australia/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_R1.fq.gz\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\n    tmp=\u003cspan class=\"pl-smi\"\u003e${file\u003cspan class=\"pl-k\"\u003e##*/\u003c/span\u003e}\u003c/span\u003e\n    i=\u003cspan class=\"pl-smi\"\u003e${tmp\u003cspan class=\"pl-k\"\u003e%%\u003c/span\u003e_\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e}\u003c/span\u003e\n\n    sh shell/mapping.sh \\\n        /data/Australia/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e_1.fq.gz\t\\\n        /data/Australia/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e_2.fq.gz\t\u003cspan class=\"pl-cce\"\u003e\\\t\u003c/span\u003e\n        \u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \\\n        /data/Australia/mapping/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \\\n        bbmap\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# synchronize as MPILEUP based on hologenome from Kapun et al. (2020) only including 3L and 3R\u003c/span\u003e\nsamtools mpileup \\\n    -B \\\n    -f /data/Dmel_6.04_hologenome_v2.fasta \\\n    -b /data/Australia_BAM.txt \\\n    -l /data/regions.bed.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/Australia/Australia.mpileup\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format\u003c/span\u003e\njava -jar /scripts/popoolation2_1201/mpileup2sync.jar \\\n    -input /data/Australia/Australia.mpileup \\\n    -threads 20 \\\n    -output /data/Australia/Australia.sync\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThen we used the dataset of In(3R)Payne-specific marker SNPs \u003ccode\u003edata/fixed095.txt\u003c/code\u003e from \u003ca href=\"https://onlinelibrary.wiley.com/doi/full/10.1111/mec.12594\" rel=\"nofollow\"\u003eKapun \u003cem\u003eet al.\u003c/em\u003e 2014\u003c/a\u003e to test for the frequency of inversion-specific alleles in the individuals\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n## identify the average frequency of inversion-specific alleles in the individual libraries\npython3 /scripts/v3/Australia-karyos.py \\\n    -karyo /data/fixed095.txt \\\n    -input /data/Australia/Australia.sync \\\n    -names IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \\\n    \u0026gt; /data/Australia/Australia_karyos.txt\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# identify the average frequency of inversion-specific alleles in the individual libraries\u003c/span\u003e\npython3 /scripts/v3/Australia-karyos.py \\\n    -karyo /data/fixed095.txt \\\n    -input /data/Australia/Australia.sync \\\n    -names IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \\\n    \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/Australia/Australia_karyos.txt\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe only considered samples where the inversion-specific alleles at four SNPs showed an average frequency of \u0026gt;90% (considered \"inverted\") or \u0026lt;10% (considered \"standard\").\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThen, we further downloaded the imputed SNP data from from DataDryad \u003ca href=\"https:/datadryad.org/stash/dataset/doi:10.5061/dryad.5q0m8\"\u003edoi:10.5061/dryad.5q0m8.\u003c/a\u003e and used the karyotypic classification from above to calculate poppulation genetic statistics for each group separately.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2) Phylogenetic approach\u003c/h2\u003e\u003ca id=\"user-content-2-phylogenetic-approach\" class=\"anchor\" aria-label=\"Permalink: 2) Phylogenetic approach\" href=\"#2-phylogenetic-approach\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHere, we investigated the phylogenetic signal of worldwide samples based on SNPs located within \u003cem\u003eIn(3R)P\u003c/em\u003e or in distance to the inversion.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.1) Adjust Australian samples to match the other data\u003c/h3\u003e\u003ca id=\"user-content-21-adjust-australian-samples-to-match-the-other-data\" class=\"anchor\" aria-label=\"Permalink: 2.1) Adjust Australian samples to match the other data\" href=\"#21-adjust-australian-samples-to-match-the-other-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAt first, we lifted genomic coordinates of the data, originally mapped to the \u003cem\u003eD. melanogaster\u003c/em\u003e reference genome v. 5 to v.6 coordinates.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## convert coordinates\npython3 /scripts/v3/Flybase_version_converter.py \\\n    -input /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs.vcf \\\n    -output /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs_v6.vcf\n\npython3 /scripts/v3/Flybase_version_converter.py \\\n    -input /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs.vcf \\\n    -output /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs_v6.vcf\n\ncat /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs_v6.vcf \\\n/data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs_v6.vcf \\\n\u0026gt; /data/consensus/Australia.vcf\n\n## then convert from VCF to cons format\npython3 /scripts/v3/vcf2cons.py \\\n    /data/consensus/Australia.vcf \\\n    | gzip \u0026gt; /data/Australia/Australia.cons.gz\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# convert coordinates\u003c/span\u003e\npython3 /scripts/v3/Flybase_version_converter.py \\\n    -input /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs.vcf \\\n    -output /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs_v6.vcf\n\npython3 /scripts/v3/Flybase_version_converter.py \\\n    -input /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs.vcf \\\n    -output /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs_v6.vcf\n\ncat /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs_v6.vcf \\\n/data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs_v6.vcf \\\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/Australia.vcf\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# then convert from VCF to cons format\u003c/span\u003e\npython3 /scripts/v3/vcf2cons.py \\\n    /data/consensus/Australia.vcf \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/Australia/Australia.cons.gz\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.2) Merge all files in *.cons format\u003c/h3\u003e\u003ca id=\"user-content-22-merge-all-files-in-cons-format\" class=\"anchor\" aria-label=\"Permalink: 2.2) Merge all files in *.cons format\" href=\"#22-merge-all-files-in-cons-format\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow, we merge the cons files and either draw randomly selected SNPs inside \u003cem\u003eIn(3R)Payne\u003c/em\u003e or which are in distance to the Inversion boundaries\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## Merge and select Inv SNPs\npython3 /scripts/merge_consensus.py \\\n    - /data/Inv.bed \\\n    -consensus /data/consensus/Zambia.consensus.gz,/data/consensus/DGN.consensus.gz,/data/consensus/USA.consensus.gz,/data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz,/data/Australia/Australia.cons.gz \\\n    | gzip \u0026gt; /data/consensus/Inv.cons.gz\n\n## Merge and select Non-Inv SNPs\npython3 /scripts/merge_consensus.py \\\n    - /data/NoInv.bed \\\n    -consensus /data/consensus/Zambia.consensus.gz,/data/consensus/DGN.consensus.gz,/data/consensus/USA.consensus.gz,/data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz,/data/Australia/Australia.cons.gz \\\n    | gzip \u0026gt; /data/consensus/NoInv.cons.gz\n\n## convert merged cons to nexus format\npython3 /scripts/v3/cons2nexus.py \\\n    -input /data/consensus/Inv.cons.gz \\\n    -population 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,362,363,364,365,366,367,368,369,370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,418,419,420,421,422,423,424,425,426,427,428,429,430,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,452,453,454,455,456,457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483 \\\n    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388,FR180,FR217,FR229,FR361,CK1,CO10N,CO13N,CO14,CO15N,CO16,CO1,CO2,CO4N,CO8N,CO9N,EZ25,EZ2,EZ9N,GA125,GA129,GA132,GA141,GA145,GA160,GA185,GU10,GU6,GU7,GU9,KN133N,KN20N,KN35,KR39,KR42,KR4N,KR7,KT6,NG10N,NG1N,NG3N,NG9,SP221,TZ10,TZ14,TZ8,UM118,UM37,UM526,ZO65,ZS11,ZS37,ZS5,CK2,ED2,ED3,ED5N,ED6N,ED10N,EZ5N,GA130,GA191,GU2,KN6,KN34,KT1,NG6N,NG7,RC1,RC5,RG2,RG3,RG4N,RG5,RG6N,RG7,RG8,RG9,RG10,RG11N,RG13N,RG15,RG18N,RG19,RG21N,RG22,RG24,RG25,RG28,RG32N,RG33,RG34,RG35,RG36,RG37N,RG38N,RG39,SP80,SP173,SP188,SP235,SP241,SP254,UG5N,UG7,UG19,UG28N,ZI91a,ZI261a,ZI268,ZI468,ZL130,FR14,FR70,FR151,FR310,RAL-301,RAL-303,RAL-304,RAL-306,RAL-307,RAL-313,RAL-315,RAL-324,RAL-335,RAL-357,RAL-358,RAL-360,RAL-362,RAL-365,RAL-375,RAL-379,RAL-380,RAL-391,RAL-399,RAL-427,RAL-437,RAL-486,RAL-513,RAL-517,RAL-555,RAL-639,RAL-705,RAL-707,RAL-714,RAL-730,RAL-732,RAL-765,RAL-774,RAL-786,RAL-799,RAL-820,RAL-852,Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81,F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,168_hot_R2,150_hot_R2,143_hot_R2,136_hot_R1,129_hot_R1,117_hot_R1,106_hot_R1,100_cold_R1,96_cold_R1,91_cold_R1,89_cold_R2,80_cold_R2,53_cold_R2,52_cold_R2,21_cold_R3,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94,IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \\\n    \u0026gt; /data/consensus/Inv.nexus\n\npython3 /scripts/v3/cons2nexus.py \\\n    -input /data/consensus/NoInv.cons.gz \\\n    -population 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,362,363,364,365,366,367,368,369,370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,418,419,420,421,422,423,424,425,426,427,428,429,430,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,452,453,454,455,456,457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483 \\\n    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388,FR180,FR217,FR229,FR361,CK1,CO10N,CO13N,CO14,CO15N,CO16,CO1,CO2,CO4N,CO8N,CO9N,EZ25,EZ2,EZ9N,GA125,GA129,GA132,GA141,GA145,GA160,GA185,GU10,GU6,GU7,GU9,KN133N,KN20N,KN35,KR39,KR42,KR4N,KR7,KT6,NG10N,NG1N,NG3N,NG9,SP221,TZ10,TZ14,TZ8,UM118,UM37,UM526,ZO65,ZS11,ZS37,ZS5,CK2,ED2,ED3,ED5N,ED6N,ED10N,EZ5N,GA130,GA191,GU2,KN6,KN34,KT1,NG6N,NG7,RC1,RC5,RG2,RG3,RG4N,RG5,RG6N,RG7,RG8,RG9,RG10,RG11N,RG13N,RG15,RG18N,RG19,RG21N,RG22,RG24,RG25,RG28,RG32N,RG33,RG34,RG35,RG36,RG37N,RG38N,RG39,SP80,SP173,SP188,SP235,SP241,SP254,UG5N,UG7,UG19,UG28N,ZI91a,ZI261a,ZI268,ZI468,ZL130,FR14,FR70,FR151,FR310,RAL-301,RAL-303,RAL-304,RAL-306,RAL-307,RAL-313,RAL-315,RAL-324,RAL-335,RAL-357,RAL-358,RAL-360,RAL-362,RAL-365,RAL-375,RAL-379,RAL-380,RAL-391,RAL-399,RAL-427,RAL-437,RAL-486,RAL-513,RAL-517,RAL-555,RAL-639,RAL-705,RAL-707,RAL-714,RAL-730,RAL-732,RAL-765,RAL-774,RAL-786,RAL-799,RAL-820,RAL-852,Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81,F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,168_hot_R2,150_hot_R2,143_hot_R2,136_hot_R1,129_hot_R1,117_hot_R1,106_hot_R1,100_cold_R1,96_cold_R1,91_cold_R1,89_cold_R2,80_cold_R2,53_cold_R2,52_cold_R2,21_cold_R3,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94,IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \\\n    \u0026gt; /data/consensus/NoInv.nexus\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Merge and select Inv SNPs\u003c/span\u003e\npython3 /scripts/merge_consensus.py \\\n    - /data/Inv.bed \\\n    -consensus /data/consensus/Zambia.consensus.gz,/data/consensus/DGN.consensus.gz,/data/consensus/USA.consensus.gz,/data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz,/data/Australia/Australia.cons.gz \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/Inv.cons.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Merge and select Non-Inv SNPs\u003c/span\u003e\npython3 /scripts/merge_consensus.py \\\n    - /data/NoInv.bed \\\n    -consensus /data/consensus/Zambia.consensus.gz,/data/consensus/DGN.consensus.gz,/data/consensus/USA.consensus.gz,/data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz,/data/Australia/Australia.cons.gz \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/NoInv.cons.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# convert merged cons to nexus format\u003c/span\u003e\npython3 /scripts/v3/cons2nexus.py \\\n    -input /data/consensus/Inv.cons.gz \\\n    -population 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,362,363,364,365,366,367,368,369,370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,418,419,420,421,422,423,424,425,426,427,428,429,430,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,452,453,454,455,456,457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483 \\\n    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388,FR180,FR217,FR229,FR361,CK1,CO10N,CO13N,CO14,CO15N,CO16,CO1,CO2,CO4N,CO8N,CO9N,EZ25,EZ2,EZ9N,GA125,GA129,GA132,GA141,GA145,GA160,GA185,GU10,GU6,GU7,GU9,KN133N,KN20N,KN35,KR39,KR42,KR4N,KR7,KT6,NG10N,NG1N,NG3N,NG9,SP221,TZ10,TZ14,TZ8,UM118,UM37,UM526,ZO65,ZS11,ZS37,ZS5,CK2,ED2,ED3,ED5N,ED6N,ED10N,EZ5N,GA130,GA191,GU2,KN6,KN34,KT1,NG6N,NG7,RC1,RC5,RG2,RG3,RG4N,RG5,RG6N,RG7,RG8,RG9,RG10,RG11N,RG13N,RG15,RG18N,RG19,RG21N,RG22,RG24,RG25,RG28,RG32N,RG33,RG34,RG35,RG36,RG37N,RG38N,RG39,SP80,SP173,SP188,SP235,SP241,SP254,UG5N,UG7,UG19,UG28N,ZI91a,ZI261a,ZI268,ZI468,ZL130,FR14,FR70,FR151,FR310,RAL-301,RAL-303,RAL-304,RAL-306,RAL-307,RAL-313,RAL-315,RAL-324,RAL-335,RAL-357,RAL-358,RAL-360,RAL-362,RAL-365,RAL-375,RAL-379,RAL-380,RAL-391,RAL-399,RAL-427,RAL-437,RAL-486,RAL-513,RAL-517,RAL-555,RAL-639,RAL-705,RAL-707,RAL-714,RAL-730,RAL-732,RAL-765,RAL-774,RAL-786,RAL-799,RAL-820,RAL-852,Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81,F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,168_hot_R2,150_hot_R2,143_hot_R2,136_hot_R1,129_hot_R1,117_hot_R1,106_hot_R1,100_cold_R1,96_cold_R1,91_cold_R1,89_cold_R2,80_cold_R2,53_cold_R2,52_cold_R2,21_cold_R3,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94,IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \\\n    \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/Inv.nexus\n\npython3 /scripts/v3/cons2nexus.py \\\n    -input /data/consensus/NoInv.cons.gz \\\n    -population 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,362,363,364,365,366,367,368,369,370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,418,419,420,421,422,423,424,425,426,427,428,429,430,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,452,453,454,455,456,457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483 \\\n    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388,FR180,FR217,FR229,FR361,CK1,CO10N,CO13N,CO14,CO15N,CO16,CO1,CO2,CO4N,CO8N,CO9N,EZ25,EZ2,EZ9N,GA125,GA129,GA132,GA141,GA145,GA160,GA185,GU10,GU6,GU7,GU9,KN133N,KN20N,KN35,KR39,KR42,KR4N,KR7,KT6,NG10N,NG1N,NG3N,NG9,SP221,TZ10,TZ14,TZ8,UM118,UM37,UM526,ZO65,ZS11,ZS37,ZS5,CK2,ED2,ED3,ED5N,ED6N,ED10N,EZ5N,GA130,GA191,GU2,KN6,KN34,KT1,NG6N,NG7,RC1,RC5,RG2,RG3,RG4N,RG5,RG6N,RG7,RG8,RG9,RG10,RG11N,RG13N,RG15,RG18N,RG19,RG21N,RG22,RG24,RG25,RG28,RG32N,RG33,RG34,RG35,RG36,RG37N,RG38N,RG39,SP80,SP173,SP188,SP235,SP241,SP254,UG5N,UG7,UG19,UG28N,ZI91a,ZI261a,ZI268,ZI468,ZL130,FR14,FR70,FR151,FR310,RAL-301,RAL-303,RAL-304,RAL-306,RAL-307,RAL-313,RAL-315,RAL-324,RAL-335,RAL-357,RAL-358,RAL-360,RAL-362,RAL-365,RAL-375,RAL-379,RAL-380,RAL-391,RAL-399,RAL-427,RAL-437,RAL-486,RAL-513,RAL-517,RAL-555,RAL-639,RAL-705,RAL-707,RAL-714,RAL-730,RAL-732,RAL-765,RAL-774,RAL-786,RAL-799,RAL-820,RAL-852,Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81,F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,168_hot_R2,150_hot_R2,143_hot_R2,136_hot_R1,129_hot_R1,117_hot_R1,106_hot_R1,100_cold_R1,96_cold_R1,91_cold_R1,89_cold_R2,80_cold_R2,53_cold_R2,52_cold_R2,21_cold_R3,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94,IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \\\n    \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/NoInv.nexus\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUsing the NEXUS files, we plotted phylogenetic networks based on the Neighbor-Net inference method with Splits-Tree.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3) Population Genetics Analysis\u003c/h2\u003e\u003ca id=\"user-content-3-population-genetics-analysis\" class=\"anchor\" aria-label=\"Permalink: 3) Population Genetics Analysis\" href=\"#3-population-genetics-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.1) Genetic variation\u003c/h3\u003e\u003ca id=\"user-content-31-genetic-variation\" class=\"anchor\" aria-label=\"Permalink: 3.1) Genetic variation\" href=\"#31-genetic-variation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUsing \u003ca href=\"https:/vcftools.github.io/index.html\"\u003eVCFtools\u003c/a\u003e we first calculated nucleotide diveristy (\u003cem\u003e\u003c/em\u003e) and Tajima's \u003cem\u003eD\u003c/em\u003e as measures of genetic diversity in the different karyotypes\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n## make directory \nmkdir /data/PopGen\n\n## loop through continents/countries\nfor continent in Africa Europe America Australia\ndo\n    ## loop through karyotypes\n    for karyotype in Inv Std All\n    do\n        ## loop through regions\n        for file in data/${continent}_*_${karyotype}.samples\n        do      \n            ## isolate region name\n            IFS='_ ' read -r -A array \u0026lt;\u0026lt;\u0026lt; \u0026quot;$file\u0026quot;\n            region=${array[2]}\n\n            ## calculate \n            /scripts/vcftools_0.1.13/bin/vcftools \\\n                -keep data/${continent}_${region}_${karyotype}.samples \\\n                -window-pi 100000 \\\n                -window-pi-step 100000 \\\n                -gzvcf /data/consensus/${continent}.vcf.gz \\\n                -out /data/PopGen/${continent}_${region}_${karyotype}_100k.pi\n\n            ## calculate Tajima's D\n            /scripts/vcftools_0.1.13/bin/vcftools \\\n                -keep data/${continent}_${region}_${karyotype}.samples \\\n                -TajimaD 100000 \\\n                -gzvcf /data/consensus/${continent}.vcf.gz \\\n                -out /data/PopGen/${continent}_${region}_${karyotype}_100k\n        \n        done\n    done\n\n## then merge the results in a single large file for  and Tajima's D, plot averages by karyotype and Geographic origin and test for significant differences.\npython3 /scripts/v3/merge-div.py \\\n    -input /data/PopGen/Africa_Zambia_Inv_100k.pi,/data/PopGen/Africa_Zambia_Std_100k.pi,/data/PopGen/Africa_Zambia_All_100k.pi,/data/PopGen/Europe_Portugal_Inv_100k.pi,/data/PopGen/Europe_Portugal_Std_100k.pi,/data/PopGen/Europe_Portugal_All_100k.pi,/data/PopGen/Europe_Sweden_Std_100k.pi,/data/PopGen/America_Florida_Inv_100k.pi,/data/PopGen/America_Florida_Std_100k.pi,/data/PopGen/America_Florida_All_100k.pi,/data/PopGen/America_Maine_Std_100k.pi,/data/PopGen/Australia_Queensland_Inv_100k.pi,/data/PopGen/Australia_Queensland_Std_100k.pi,/data/PopGen/Australia_Queensland_All_100k.pi,/data/PopGen/Australia_Victoria_Std_100k.pi \\\n    -names Zambia-Inv,Zambia-Std,Zambia-All,Portugal-Inv,Portugal-Std,Portugal-All,Sweden-Std,Florida-Inv,Florida-Std,Florida-All,Maine-Std,Queensland-Inv,Queensland-Std,Queensland-All,Victoria-Std \\\n    \u0026gt; /data/PopGen/AllData.pi\n\nRscript /scripts/statNplot.R /data/PopGen/AllData.pi\n\npython3 /scripts/v3/merge-div.py \\\n    -input /data/PopGen/Africa_Zambia_Inv_100k.Tajima.D,/data/PopGen/Africa_Zambia_Std_100k.Tajima.D,/data/PopGen/Africa_Zambia_All_100k.Tajima.D,/data/PopGen/Europe_Portugal_Inv_100k.Tajima.D,/data/PopGen/Europe_Portugal_Std_100k.Tajima.D,/data/PopGen/Europe_Portugal_All_100k.Tajima.D,/data/PopGen/Europe_Sweden_Std_100k.Tajima.D,/data/PopGen/America_Florida_Inv_100k.Tajima.D,/data/PopGen/America_Florida_Std_100k.Tajima.D,/data/PopGen/America_Florida_All_100k.Tajima.D,/data/PopGen/America_Maine_Std_100k.Tajima.D,/data/PopGen/Australia_Queensland_Inv_100k.Tajima.D,/data/PopGen/Australia_Queensland_Std_100k.Tajima.D,/data/PopGen/Australia_Queensland_All_100k.Tajima.D,/data/PopGen/Australia_Victoria_Std_100k.Tajima.D \\\n    -names Zambia-Inv,Zambia-Std,Zambia-All,Portugal-Inv,Portugal-Std,Portugal-All,Sweden-Std,Florida-Inv,Florida-Std,Florida-All,Maine-Std,Queensland-Inv,Queensland-Std,Queensland-All,Victoria-Std \\\n    \u0026gt; /data/PopGen/AllData.Tajima.D\n\nRscript /scripts/statNplot.R /data/PopGen/AllData.Tajima.D\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# make directory \u003c/span\u003e\nmkdir /data/PopGen\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# loop through continents/countries\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003econtinent\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e Africa Europe America Australia\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# loop through karyotypes\u003c/span\u003e\n    \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ekaryotype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e Inv Std All\n    \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# loop through regions\u003c/span\u003e\n        \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e data/\u003cspan class=\"pl-smi\"\u003e${continent}\u003c/span\u003e_\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e${karyotype}\u003c/span\u003e.samples\n        \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e      \n            \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# isolate region name\u003c/span\u003e\n            IFS=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_ \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e -r -A array \u003cspan class=\"pl-k\"\u003e\u0026lt;\u0026lt;\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e$file\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n            region=\u003cspan class=\"pl-smi\"\u003e${array[2]}\u003c/span\u003e\n\n            \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# calculate \u003c/span\u003e\n            /scripts/vcftools_0.1.13/bin/vcftools \\\n                -keep data/\u003cspan class=\"pl-smi\"\u003e${continent}\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e${region}\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e${karyotype}\u003c/span\u003e.samples \\\n                -window-pi 100000 \\\n                -window-pi-step 100000 \\\n                -gzvcf /data/consensus/\u003cspan class=\"pl-smi\"\u003e${continent}\u003c/span\u003e.vcf.gz \\\n                -out /data/PopGen/\u003cspan class=\"pl-smi\"\u003e${continent}\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e${region}\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e${karyotype}\u003c/span\u003e_100k.pi\n\n            \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# calculate Tajima's D\u003c/span\u003e\n            /scripts/vcftools_0.1.13/bin/vcftools \\\n                -keep data/\u003cspan class=\"pl-smi\"\u003e${continent}\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e${region}\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e${karyotype}\u003c/span\u003e.samples \\\n                -TajimaD 100000 \\\n                -gzvcf /data/consensus/\u003cspan class=\"pl-smi\"\u003e${continent}\u003c/span\u003e.vcf.gz \\\n                -out /data/PopGen/\u003cspan class=\"pl-smi\"\u003e${continent}\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e${region}\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e${karyotype}\u003c/span\u003e_100k\n        \n        \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n    \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# then merge the results in a single large file for  and Tajima's D, plot averages by karyotype and Geographic origin and test for significant differences.\u003c/span\u003e\npython3 /scripts/v3/merge-div.py \\\n    -input /data/PopGen/Africa_Zambia_Inv_100k.pi,/data/PopGen/Africa_Zambia_Std_100k.pi,/data/PopGen/Africa_Zambia_All_100k.pi,/data/PopGen/Europe_Portugal_Inv_100k.pi,/data/PopGen/Europe_Portugal_Std_100k.pi,/data/PopGen/Europe_Portugal_All_100k.pi,/data/PopGen/Europe_Sweden_Std_100k.pi,/data/PopGen/America_Florida_Inv_100k.pi,/data/PopGen/America_Florida_Std_100k.pi,/data/PopGen/America_Florida_All_100k.pi,/data/PopGen/America_Maine_Std_100k.pi,/data/PopGen/Australia_Queensland_Inv_100k.pi,/data/PopGen/Australia_Queensland_Std_100k.pi,/data/PopGen/Australia_Queensland_All_100k.pi,/data/PopGen/Australia_Victoria_Std_100k.pi \\\n    -names Zambia-Inv,Zambia-Std,Zambia-All,Portugal-Inv,Portugal-Std,Portugal-All,Sweden-Std,Florida-Inv,Florida-Std,Florida-All,Maine-Std,Queensland-Inv,Queensland-Std,Queensland-All,Victoria-Std \\\n    \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/PopGen/AllData.pi\n\nRscript /scripts/statNplot.R /data/PopGen/AllData.pi\n\npython3 /scripts/v3/merge-div.py \\\n    -input /data/PopGen/Africa_Zambia_Inv_100k.Tajima.D,/data/PopGen/Africa_Zambia_Std_100k.Tajima.D,/data/PopGen/Africa_Zambia_All_100k.Tajima.D,/data/PopGen/Europe_Portugal_Inv_100k.Tajima.D,/data/PopGen/Europe_Portugal_Std_100k.Tajima.D,/data/PopGen/Europe_Portugal_All_100k.Tajima.D,/data/PopGen/Europe_Sweden_Std_100k.Tajima.D,/data/PopGen/America_Florida_Inv_100k.Tajima.D,/data/PopGen/America_Florida_Std_100k.Tajima.D,/data/PopGen/America_Florida_All_100k.Tajima.D,/data/PopGen/America_Maine_Std_100k.Tajima.D,/data/PopGen/Australia_Queensland_Inv_100k.Tajima.D,/data/PopGen/Australia_Queensland_Std_100k.Tajima.D,/data/PopGen/Australia_Queensland_All_100k.Tajima.D,/data/PopGen/Australia_Victoria_Std_100k.Tajima.D \\\n    -names Zambia-Inv,Zambia-Std,Zambia-All,Portugal-Inv,Portugal-Std,Portugal-All,Sweden-Std,Florida-Inv,Florida-Std,Florida-All,Maine-Std,Queensland-Inv,Queensland-Std,Queensland-All,Victoria-Std \\\n    \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/PopGen/AllData.Tajima.D\n\nRscript /scripts/statNplot.R /data/PopGen/AllData.Tajima.D\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIn addition, I plotted the values of \u003cem\u003e\u003c/em\u003e and Tajima's \u003cem\u003eD\u003c/em\u003e as lineplots in R\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"### plot  for Inverted and Standard Chromosomes\nlibrary(tidyverse)\n\nDATA=read.table(\u0026quot;/data/PopGen/AllData.pi\u0026quot;,header=T)\n\nDATA.3R=subset(DATA,DATA$C==\u0026quot;3R\u0026quot;)\nDATA.3R=subset(DATA,DATA$InvStatus!=\u0026quot;All\u0026quot;)\nDATA.3R=subset(DATA.3R,DATA.3R$Origin %in% c(\u0026quot;Florida\u0026quot;,\u0026quot;Portugal\u0026quot;,\u0026quot;Zambia\u0026quot;,\u0026quot;Queensland\u0026quot;))\nDATA.3R$Origin \u0026lt;- factor(DATA.3R$Origin,levels=c(\u0026quot;Zambia\u0026quot;,\u0026quot;Portugal\u0026quot;,\u0026quot;Florida\u0026quot;,\u0026quot;Queensland\u0026quot;))\n\nP=ggplot(DATA.3R,aes(\n    x=P,\n    y=pi,\n    col=InvStatus))+\n  geom_line(lwd=1.5)+\n  geom_rect(mapping = aes(\n        xmin=16432209,\n        xmax=24744010,\n        ymin=0,\n        ymax=(max(pi))),\n    color=\u0026quot;black\u0026quot;,\n    alpha=0,\n    lwd=0.5,\n    lty=2)+\n  facet_grid(Origin~ .)+\n  theme_classic()\n\nggsave(\u0026quot;/data/PopGen/AllData.pi.pdf\u0026quot;,\n    P,\n    width=12,\n    height=8)\n\n### plot Tajima's D for Inverted and Standard Chromosomes\nlibrary(tidyverse)\n\nDATA=read.table(\u0026quot;/data/PopGen/AllData.Tajima.D\u0026quot;,header=T)\n\nDATA.3R=subset(DATA,DATA$C==\u0026quot;3R\u0026quot;)\nDATA.3R=subset(DATA.3R,DATA.3R$Origin %in% c(\u0026quot;Florida\u0026quot;,\u0026quot;Portugal\u0026quot;,\u0026quot;Zambia\u0026quot;,\u0026quot;Queensland\u0026quot;))\nDATA.3R$Origin \u0026lt;- factor(DATA.3R$Origin,levels=c(\u0026quot;Zambia\u0026quot;,\u0026quot;Portugal\u0026quot;,\u0026quot;Florida\u0026quot;,\u0026quot;Queensland\u0026quot;))\n\nP=ggplot(DATA.3R,aes(\n    x=P,\n    y=pi,\n    col=InvStatus))+\n  geom_line(lwd=1.5)+\n  geom_rect(mapping = aes(\n        xmin=16432209,\n        xmax=24744010,\n        ymin=0,\n        ymax=(max(pi))),\n    color=\u0026quot;black\u0026quot;,\n    alpha=0,\n    lwd=0.5,\n    lty=2)+\n  facet_grid(Origin~ .)+\n  theme_classic()+\n  geom_hline(yintercept=0)\n\nggsave(\u0026quot;/data/PopGen/AllData.Tajima.D.pdf\u0026quot;,\n    P,\n    width=12,\n    height=8)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## plot  for Inverted and Standard Chromosomes\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003etidyverse\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003eDATA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/PopGen/AllData.pi\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003eDATA.3R\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eDATA\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eDATA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eC\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e3R\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eDATA.3R\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eDATA\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eDATA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eInvStatus\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAll\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eDATA.3R\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eDATA.3R\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eDATA.3R\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOrigin\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFlorida\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePortugal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZambia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eQueensland\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eDATA.3R\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOrigin\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eDATA.3R\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOrigin\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZambia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePortugal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFlorida\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eQueensland\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-v\"\u003eP\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eggplot(\u003cspan class=\"pl-smi\"\u003eDATA.3R\u003c/span\u003e,aes(\n    \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eP\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003epi\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eInvStatus\u003c/span\u003e))\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_line(\u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1.5\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_rect(\u003cspan class=\"pl-v\"\u003emapping\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e aes(\n        \u003cspan class=\"pl-v\"\u003exmin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16432209\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003exmax\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e24744010\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003eymin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003eymax\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(max(\u003cspan class=\"pl-c1\"\u003epi\u003c/span\u003e))),\n    \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  facet_grid(\u003cspan class=\"pl-smi\"\u003eOrigin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme_classic()\n\nggsave(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/PopGen/AllData.pi.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-smi\"\u003eP\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## plot Tajima's D for Inverted and Standard Chromosomes\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003etidyverse\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003eDATA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/PopGen/AllData.Tajima.D\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003eDATA.3R\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eDATA\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eDATA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eC\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e3R\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eDATA.3R\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eDATA.3R\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eDATA.3R\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOrigin\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFlorida\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePortugal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZambia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eQueensland\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eDATA.3R\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOrigin\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eDATA.3R\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOrigin\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZambia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePortugal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFlorida\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eQueensland\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-v\"\u003eP\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eggplot(\u003cspan class=\"pl-smi\"\u003eDATA.3R\u003c/span\u003e,aes(\n    \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eP\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003epi\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eInvStatus\u003c/span\u003e))\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_line(\u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1.5\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_rect(\u003cspan class=\"pl-v\"\u003emapping\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e aes(\n        \u003cspan class=\"pl-v\"\u003exmin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16432209\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003exmax\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e24744010\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003eymin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003eymax\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(max(\u003cspan class=\"pl-c1\"\u003epi\u003c/span\u003e))),\n    \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  facet_grid(\u003cspan class=\"pl-smi\"\u003eOrigin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme_classic()\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_hline(\u003cspan class=\"pl-v\"\u003eyintercept\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e)\n\nggsave(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/PopGen/AllData.Tajima.D.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-smi\"\u003eP\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.2) Genetic differentiation\u003c/h3\u003e\u003ca id=\"user-content-32-genetic-differentiation\" class=\"anchor\" aria-label=\"Permalink: 3.2) Genetic differentiation\" href=\"#32-genetic-differentiation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAgain using \u003ca href=\"https:/vcftools.github.io/index.html\"\u003eVCFtools\u003c/a\u003e we calculated pairwise FST among karyotypes in different geographic regions both for single SNPs and in 100kb windows. Then, we merged the FST values, plotted the averages with respect to karyotype and tested for significant differences with respect to the karyotype and origin. Moreover, we tested if window-wise FST is correlated within the inverted genomic region.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n## loop through samples\ncontinent=( \u0026quot;Africa\u0026quot; \u0026quot;Europe\u0026quot; \u0026quot;Europe\u0026quot; \u0026quot;Europe\u0026quot; \u0026quot;America\u0026quot; \u0026quot;America\u0026quot; \u0026quot;America\u0026quot; \u0026quot;Australia\u0026quot; \u0026quot;Australia\u0026quot; \u0026quot;Australia\u0026quot; )\nFST=( \u0026quot;ZI-ZS\u0026quot; \u0026quot;PI-PS\u0026quot; \u0026quot;PI-SS\u0026quot; \u0026quot;PS-SS\u0026quot; \u0026quot;FI-FS\u0026quot; \u0026quot;FI-MS\u0026quot; \u0026quot;FS-MS\u0026quot; \u0026quot;II-IS\u0026quot; \u0026quot;II-YS\u0026quot; \u0026quot;IS-YS\u0026quot; )\nPop1=( \u0026quot;Africa_Zambia_Inv\u0026quot; \u0026quot;Europe_Portugal_Inv\u0026quot; \u0026quot;Europe_Portugal_Inv\u0026quot; \u0026quot;Europe_Portugal_Std\u0026quot; \u0026quot;America_Florida_Inv\u0026quot; \u0026quot;America_Florida_Inv\u0026quot; \u0026quot;America_Florida_Std\u0026quot; \u0026quot;Australia_Queensland_Inv\u0026quot; \u0026quot;Australia_Queensland_Inv\u0026quot; \u0026quot;Australia_Queensland_Std\u0026quot; )\nPop2=( \u0026quot;Africa_Zambia_Std\u0026quot; \u0026quot;Europe_Portugal_Std\u0026quot; \u0026quot;Europe_Sweden_Std\u0026quot; \u0026quot;Europe_Sweden_Std\u0026quot; \u0026quot;America_Florida_Std\u0026quot; \u0026quot;America_Maine_Std\u0026quot; \u0026quot;America_Maine_Std\u0026quot; \u0026quot;Australia_Queensland_Std\u0026quot; \u0026quot;Australia_Victoria_Std\u0026quot; \u0026quot;Australia_Victoria_Std\u0026quot; )\n\nfor index in ${!FST[@]}\ndo\n    ## window-wise FST\n    /scripts/vcftools_0.1.13/cpp/vcftools \\\n        -gzvcf /data/consensus/${continent[index]}.vcf.gz \\\n        -weir-fst-pop /data/${Pop1[index]}.samples \\\n        -weir-fst-pop /data/${Pop2[index]}.samples \\\n        -fst-window-size 100000 \\\n        -fst-window-step 100000 \\\n        -out /data/PopGen/${FST[index]}\n    \n    ## SNP-wise FST\n    /scripts/vcftools_0.1.13/cpp/vcftools \\\n        -gzvcf /data/consensus/${continent[index]}.vcf.gz \\\n        -weir-fst-pop /data/${Pop1[index]}.samples \\\n        -weir-fst-pop /data/${Pop2[index]}.samples \\\n        -out /data/PopGen/${FST[index]}\ndone\n\n## now merge all FST values\npython3 /scripts/v3/merge-diff.py \\\n    -input /data/PopGen/ZI-ZS.windowed.weir.fst,/data/PopGen/PI-PS.windowed.weir.fst,/data/PopGen/PI-SS.windowed.weir.fst,/data/PopGen/PS-SS.windowed.weir.fst,/data/PopGen/FI-FS.windowed.weir.fst,/data/PopGen/FI-MS.windowed.weir.fst,/data/PopGen/FS-MS.windowed.weir.fst,/data/PopGen/II-IS.windowed.weir.fst,/data/PopGen/II-YS.windowed.weir.fst,/data/PopGen/IS-YS.windowed.weir.fst \\\n    -names ZI-ZS,PI-PS,PI-SS,PS-SS,FI-FS,FI-MS,FS-MS,II-IS,II-YS,IS-YS \\\n \u0026gt; /data/PopGen/AllData.fst\n\n## plot average FSTs per karyotype and geography, test for significant differences, and investigate if windowed FST within the inverted regions is correlated among geographic regions \nRscript /scripts/statNcompareFST.R /data/PopGen/AllData.fst\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# loop through samples\u003c/span\u003e\ncontinent=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\nFST=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZI-ZS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePI-PS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePI-SS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePS-SS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFI-FS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFI-MS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS-MS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eII-IS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eII-YS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIS-YS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\nPop1=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica_Zambia_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope_Portugal_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope_Portugal_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope_Portugal_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Florida_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Florida_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Florida_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia_Queensland_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia_Queensland_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia_Queensland_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\nPop2=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica_Zambia_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope_Portugal_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope_Sweden_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope_Sweden_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Florida_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Maine_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Maine_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia_Queensland_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia_Victoria_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia_Victoria_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eindex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003eFST[@]}\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# window-wise FST\u003c/span\u003e\n    /scripts/vcftools_0.1.13/cpp/vcftools \\\n        -gzvcf /data/consensus/\u003cspan class=\"pl-smi\"\u003e${continent[index]}\u003c/span\u003e.vcf.gz \\\n        -weir-fst-pop /data/\u003cspan class=\"pl-smi\"\u003e${Pop1[index]}\u003c/span\u003e.samples \\\n        -weir-fst-pop /data/\u003cspan class=\"pl-smi\"\u003e${Pop2[index]}\u003c/span\u003e.samples \\\n        -fst-window-size 100000 \\\n        -fst-window-step 100000 \\\n        -out /data/PopGen/\u003cspan class=\"pl-smi\"\u003e${FST[index]}\u003c/span\u003e\n    \n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# SNP-wise FST\u003c/span\u003e\n    /scripts/vcftools_0.1.13/cpp/vcftools \\\n        -gzvcf /data/consensus/\u003cspan class=\"pl-smi\"\u003e${continent[index]}\u003c/span\u003e.vcf.gz \\\n        -weir-fst-pop /data/\u003cspan class=\"pl-smi\"\u003e${Pop1[index]}\u003c/span\u003e.samples \\\n        -weir-fst-pop /data/\u003cspan class=\"pl-smi\"\u003e${Pop2[index]}\u003c/span\u003e.samples \\\n        -out /data/PopGen/\u003cspan class=\"pl-smi\"\u003e${FST[index]}\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# now merge all FST values\u003c/span\u003e\npython3 /scripts/v3/merge-diff.py \\\n    -input /data/PopGen/ZI-ZS.windowed.weir.fst,/data/PopGen/PI-PS.windowed.weir.fst,/data/PopGen/PI-SS.windowed.weir.fst,/data/PopGen/PS-SS.windowed.weir.fst,/data/PopGen/FI-FS.windowed.weir.fst,/data/PopGen/FI-MS.windowed.weir.fst,/data/PopGen/FS-MS.windowed.weir.fst,/data/PopGen/II-IS.windowed.weir.fst,/data/PopGen/II-YS.windowed.weir.fst,/data/PopGen/IS-YS.windowed.weir.fst \\\n    -names ZI-ZS,PI-PS,PI-SS,PS-SS,FI-FS,FI-MS,FS-MS,II-IS,II-YS,IS-YS \\\n \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/PopGen/AllData.fst\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# plot average FSTs per karyotype and geography, test for significant differences, and investigate if windowed FST within the inverted regions is correlated among geographic regions \u003c/span\u003e\nRscript /scripts/statNcompareFST.R /data/PopGen/AllData.fst\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNext, we also calculated SNP-wise FST and used \u003ca href=\"http:/pcingola.github.io/SnpEff/\"\u003eSnpEFF\u003c/a\u003e to annotate the VCF files.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## loop through continents/countries\nfor continent in Africa Europe America Australia\ndo   \n    ## annotate with SNPeff\n    java -Xmx4g -jar /scripts/snpEff/snpEff.jar \\\n        BDGP6.82 \\\n        /data/consensus/${continent}.vcf.gz \\\n        | gzip \u0026gt; /data/consensus/${continent}_ann.vcf.gz\ndone\n\n## Isolate positions with FST \u0026gt;= 0.9 and append annotation\ncontinent=( \u0026quot;Africa\u0026quot; \u0026quot;Europe\u0026quot; \u0026quot;Europe\u0026quot; \u0026quot;Europe\u0026quot; \u0026quot;America\u0026quot; \u0026quot;America\u0026quot; \u0026quot;America\u0026quot; \u0026quot;Australia\u0026quot; \u0026quot;Australia\u0026quot; \u0026quot;Australia\u0026quot; )\nFST=( \u0026quot;ZI-ZS\u0026quot; \u0026quot;PI-PS\u0026quot; \u0026quot;PI-SS\u0026quot; \u0026quot;PS-SS\u0026quot; \u0026quot;FI-FS\u0026quot; \u0026quot;FI-MS\u0026quot; \u0026quot;FS-MS\u0026quot; \u0026quot;II-IS\u0026quot; \u0026quot;II-YS\u0026quot; \u0026quot;IS-YS\u0026quot; )\n\nfor index in ${!FST[@]}\ndo  \n    ## append to full dataset\n    python3 /scripts/v3/AppendAnnotFromVCF.py \\\n        /data/consensus/${continent[index]}_ann.vcf.gz \\\n        /data/PopGen/${FST[index]}.weir.fst \\\n        | sort -k1,1 -k2,2 \\\n        | \u0026gt; /data/PopGen/${FST[index]}_ann.weir.fst\n    \n    ## additionally isolate candidates\n    awk '$NF\u0026gt;=0.9 \u0026amp;\u0026amp; $NF!=\u0026quot;nan\u0026quot;' /data/PopGen/${FST[index]}.weir.fst \\\n    | python3 /scripts/v3/AppendAnnotFromVCF.py \\\n        /data/consensus/${continent[index]}_ann.vcf.gz \\\n        - \\\n        | sort -k1,1 -k2,2 \\\n        | \u0026gt; /data/PopGen/${FST[index]}_cand_ann.weir.fst\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# loop through continents/countries\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003econtinent\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e Africa Europe America Australia\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e   \n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# annotate with SNPeff\u003c/span\u003e\n    java -Xmx4g -jar /scripts/snpEff/snpEff.jar \\\n        BDGP6.82 \\\n        /data/consensus/\u003cspan class=\"pl-smi\"\u003e${continent}\u003c/span\u003e.vcf.gz \\\n        \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/consensus/\u003cspan class=\"pl-smi\"\u003e${continent}\u003c/span\u003e_ann.vcf.gz\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Isolate positions with FST \u0026gt;= 0.9 and append annotation\u003c/span\u003e\ncontinent=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\nFST=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZI-ZS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePI-PS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePI-SS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePS-SS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFI-FS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFI-MS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS-MS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eII-IS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eII-YS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIS-YS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eindex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003eFST[@]}\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e  \n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# append to full dataset\u003c/span\u003e\n    python3 /scripts/v3/AppendAnnotFromVCF.py \\\n        /data/consensus/\u003cspan class=\"pl-smi\"\u003e${continent[index]}\u003c/span\u003e_ann.vcf.gz \\\n        /data/PopGen/\u003cspan class=\"pl-smi\"\u003e${FST[index]}\u003c/span\u003e.weir.fst \\\n        \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -k1,1 -k2,2 \\\n        \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/PopGen/\u003cspan class=\"pl-smi\"\u003e${FST[index]}\u003c/span\u003e_ann.weir.fst\n    \n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# additionally isolate candidates\u003c/span\u003e\n    awk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e$NF\u0026gt;=0.9 \u0026amp;\u0026amp; $NF!=\"nan\"\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e /data/PopGen/\u003cspan class=\"pl-smi\"\u003e${FST[index]}\u003c/span\u003e.weir.fst \\\n    \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e python3 /scripts/v3/AppendAnnotFromVCF.py \\\n        /data/consensus/\u003cspan class=\"pl-smi\"\u003e${continent[index]}\u003c/span\u003e_ann.vcf.gz \\\n        - \\\n        \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -k1,1 -k2,2 \\\n        \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/PopGen/\u003cspan class=\"pl-smi\"\u003e${FST[index]}\u003c/span\u003e_cand_ann.weir.fst\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow test in R if there is a significant overlap among candidate SNPs and Genes across continents and plot the results as barplots using the \u003ca href=\"https://cran.r-project.org/web/packages/SuperExactTest/index.html\" rel=\"nofollow\"\u003eSuperExactTest\u003c/a\u003e package\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## first SNP-wise comparison\n\nlibrary(SuperExactTest)\n\n## read background and idenitfy SNPs inside the inversion\nSEu.f=na.omit(read.table(gzfile('/data/PopGen/PI-PS_ann.weir.fst'),header=F,na.string='nan',sep='\\t', quote = ''))\nSEu.f.in3rp=subset(SEu.f,SEu.f[,1]=='3R' \u0026amp; SEu.f[,2]\u0026gt; 16432209 \u0026amp; SEu.f[,2] \u0026lt; 24744010)\nSAf.f=na.omit(read.table(gzfile('/data/PopGen/ZI-ZS_ann.weir.fst'),header=F,na.string='nan',sep='\\t', quote = ''))\nSAf.f.in3rp=subset(SAf.f,SAf.f[,1]=='3R' \u0026amp; SAf.f[,2]\u0026gt; 16432209 \u0026amp; SAf.f[,2] \u0026lt; 24744010)\nSAu.f=na.omit(read.table(gzfile('/data/PopGen/II-IS_ann.weir.fst'),header=F,na.string='nan',sep='\\t', quote = ''))\nSAu.f.in3rp=subset(SAu.f,SAu.f[,1]=='3R' \u0026amp; SAu.f[,2]\u0026gt; 16432209 \u0026amp; SAu.f[,2] \u0026lt; 24744010)\nSNA.f=na.omit(read.table(gzfile('/data/PopGen/FI-FS_ann.weir.fst'),header=F,na.string='nan',sep='\\t', quote = ''))\nSNA.f.in3rp=subset(SNA.f,SNA.f[,1]=='3R' \u0026amp; SNA.f[,2]\u0026gt; 16432209 \u0026amp; SNA.f[,2] \u0026lt; 24744010)\n\n## now for genes\nSAf.f.genes=unique(SAf.f$V5[sapply(SAf.f$V5,FUN=function(x){x!=''})])\nSEu.f.genes=unique(SEu.f$V5[sapply(SEu.f$V5,FUN=function(x){x!=''})])\nSAu.f.genes=unique(SAu.f$V5[sapply(SAu.f$V5,FUN=function(x){x!=''})])\nSNA.f.genes=unique(SNA.f$V5[sapply(SNA.f$V5,FUN=function(x){x!=''})])\n\nSAf.f.in3rp.genes=unique(SAf.f.in3rp$V5[sapply(SAf.f.in3rp$V5,FUN=function(x){x!=''})])\nSEu.f.in3rp.genes=unique(SEu.f.in3rp$V5[sapply(SEu.f.in3rp$V5,FUN=function(x){x!=''})])\nSAu.f.in3rp.genes=unique(SAu.f.in3rp$V5[sapply(SAu.f.in3rp$V5,FUN=function(x){x!=''})])\nSNA.f.in3rp.genes=unique(SNA.f.in3rp$V5[sapply(SNA.f.in3rp$V5,FUN=function(x){x!=''})])\n\n## now for SNPs\nSEu.f=unique(paste(SEu.f[,1],SEu.f[,2],sep='_'))\nSEu.f.in3rp=unique(paste(SEu.f.in3rp[,1],SEu.f.in3rp[,2],sep='_'))\nSAf.f=unique(paste(SAf.f[,1],SAf.f[,2],sep='_'))\nSAf.f.in3rp=unique(paste(SAf.f.in3rp[,1],SAf.f.in3rp[,2],sep='_'))\nSAu.f=unique(paste(SAu.f[,1],SAu.f[,2],sep='_'))\nSAu.f.in3rp=unique(paste(SAu.f.in3rp[,1],SAu.f.in3rp[,2],sep='_'))\nSNA.f=unique(paste(SNA.f[,1],SNA.f[,2],sep='_'))\nSNA.f.in3rp=unique(paste(SNA.f.in3rp[,1],SNA.f.in3rp[,2],sep='_'))\n\n# get overlapping SNPs \u0026amp; genes\nFullnoAu=Reduce(intersect,list(SEu.f,SAf.f,SNA.f))\nFullnoAu.in3p=Reduce(intersect,list(SEu.f.in3rp,SAf.f.in3rp,SNA.f.in3rp))\n\nFull=Reduce(intersect,list(SEu.f,SAf.f,SNA.f,SAu.f))\nFull.in3p=Reduce(intersect,list(SEu.f.in3rp,SAf.f.in3rp,SNA.f.in3rp,SAu.f.in3rp))\n\nFullnoAu.genes=Reduce(intersect,list(SAf.f.genes,SEu.f.genes,SNA.f.genes))\nFullnoAu.in3p.genes=Reduce(intersect,list(SEu.f.in3rp.genes,SAf.f.in3rp.genes,SNA.f.in3rp.genes))\n\nFull.genes=Reduce(intersect,list(SEu.f.genes,SAf.f.genes,SNA.f.genes,SAu.f.genes))\nFull.in3p.genes=Reduce(intersect,list(SEu.f.in3rp.genes,SAf.f.in3rp.genes,SNA.f.in3rp.genes,SAu.f.in3rp.genes))\n\ndir.create('/data/PopGen/candidates')\n\n# candidates\nSEu.c=read.table(paste('/data/PopGen/PI-PS_cand_ann.weir.fst',sep=''),sep='\\t', quote = '')\nSEu.c.in3rp=subset(SEu.c,SEu.c[,1]=='3R' \u0026amp; SEu.c[,2]\u0026gt; 16432209 \u0026amp; SEu.c[,2] \u0026lt; 24744010)\nSEu.c.genes=unique(SEu.c$V5[sapply(SEu.c$V5,FUN=function(x){x!=''})])\nSEu.c.in3rp.genes=unique(SEu.c.in3rp$V5[sapply(SEu.c.in3rp$V5,FUN=function(x){x!=''})])\nSEu.c=unique(paste(SEu.c[,1],SEu.c[,2],sep='_'))\nSEu.c.in3rp=unique(paste(SEu.c.in3rp[,1],SEu.c.in3rp[,2],sep='_'))\nSEu.c.ovNoAu.in3rp=SEu.c[SEu.c %in% FullnoAu.in3p]\nSEu.c.ov.in3rp=SEu.c[SEu.c %in% Full.in3p]\nSEu.c.ovNoAu=SEu.c[SEu.c %in% FullnoAu]\nSEu.c.ov=SEu.c[SEu.c %in% Full]\n\nSAf.c=read.table(paste('/data/PopGen/ZI-ZS_cand_ann.weir.fst',sep=''),sep='\\t', quote = '')\nSAf.c.in3rp=subset(SAf.c,SAf.c[,1]=='3R' \u0026amp; SAf.c[,2]\u0026gt; 16432209 \u0026amp; SAf.c[,2] \u0026lt; 24744010)\nSAf.c.genes=unique(SAf.c$V5[sapply(SAf.c$V5,FUN=function(x){x!=''})])\nSAf.c.in3rp.genes=unique(SAf.c.in3rp$V5[sapply(SAf.c.in3rp$V5,FUN=function(x){x!=''})])\nSAf.c=unique(paste(SAf.c[,1],SAf.c[,2],sep='_'))\nSAf.c.in3rp=paste(SAf.c.in3rp[,1],SAf.c.in3rp[,2],sep='_')\nSAf.c.ovNoAu.in3rp=SAf.c[SAf.c %in% FullnoAu.in3p]\nSAf.c.ov.in3rp=SAf.c[SAf.c %in% Full.in3p]\nSAf.c.ovNoAu=SAf.c[SAf.c %in% FullnoAu]\nSAf.c.ov=SAf.c[SAf.c %in% Full]\n\nSAu.c=read.table(paste('/data/PopGen/II-IS_cand_ann.weir.fst',sep=''),sep='\\t', quote = '')\nSAu.c.in3rp=subset(SAu.c,SAu.c[,1]=='3R' \u0026amp; SAu.c[,2]\u0026gt; 16432209 \u0026amp; SAu.c[,2] \u0026lt; 24744010)\nSAu.c.genes=unique(SAu.c$V5[sapply(SAu.c$V5,FUN=function(x){x!=''})])\nSAu.c.in3rp.genes=unique(SAu.c.in3rp$V5[sapply(SAu.c.in3rp$V5,FUN=function(x){x!=''})])\nSAu.c=paste(SAu.c[,1],SAu.c[,2],sep='_')\nSAu.c.in3rp=paste(SAu.c.in3rp[,1],SAu.c.in3rp[,2],sep='_')\nSAu.c.ov.in3rp=SAu.c[SAu.c %in% Full.in3p]\nSAu.c.ov=SAu.c[SAu.c %in% Full]\n\n\nSNA.c=read.table(paste('/data/PopGen/FI-FS_cand_ann.weir.fst',sep=''),sep='\\t', quote = '')\nSNA.c.in3rp=subset(SNA.c,SNA.c[,1]=='3R' \u0026amp; SNA.c[,2]\u0026gt; 16432209 \u0026amp; SNA.c[,2] \u0026lt; 24744010)\nSNA.c.genes=unique(SNA.c$V5[sapply(SNA.c$V5,FUN=function(x){x!=''})])\nSNA.c.in3rp.genes=unique(SNA.c.in3rp$V5[sapply(SNA.c.in3rp$V5,FUN=function(x){x!=''})])\nSNA.c=paste(SNA.c[,1],SNA.c[,2],sep='_')\nSNA.c.in3rp=paste(SNA.c.in3rp[,1],SNA.c.in3rp[,2],sep='_')\nSNA.c.ovNoAu.in3rp=SNA.c[SNA.c %in% FullnoAu.in3p]\nSNA.c.ov.in3rp=SNA.c[SNA.c %in% Full.in3p]\nSNA.c.ovNoAu=SNA.c[SNA.c %in% FullnoAu]\nSNA.c.ov=SNA.c[SNA.c %in% Full]\n\n\n## overlap in candidate SNPs from all continents except Australia and in all genomic regions\nsink(paste('/data/PopGen/candidates/All-noAu.txt',sep=''))\npdf(paste('/data/PopGen/candidates/All-noAu.pdf',sep=''),width=8,height=8)\nsets=list('Europe'=unique(SEu.c.ovNoAu),'NorthAmerica'=unique(SNA.c.ovNoAu),'Africa'=unique(SAf.c.ovNoAu))\nres=supertest(sets,n=length(FullnoAu))\nplot(res,Layout='landscape',degree=2:4,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=300)\ndev.off()\nprint(summary(res))\nsink()\nwrite.table(data.frame('NAME'=summary(res)[1],'GENES'=summary(res)[8]),file=paste('/data/PopGen/candidates/All-noAU.list',sep=''),row.names=F,quote = F,sep='\\t')\n\n## overlap in candidate SNPs from all continents except Australia and in In(3R)P only\nsink(paste('/data/PopGen/candidates/All-noAu-in3rpOnly.txt',sep=''))\npdf(paste('/data/PopGen/candidates/All-noAu-in3rpOnly.pdf',sep=''),width=8,height=8)\nsets.in3rp=list('Europe'=SEu.c.ovNoAu.in3rp,'NorthAmerica'=SNA.c.ovNoAu.in3rp,'Africa'=SAf.c.ovNoAu.in3rp)\nres.in3rp=supertest(sets.in3rp,n=length(FullnoAu.in3p))\nplot(res.in3rp,Layout='landscape',degree=2:4,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=300)\ndev.off()\nprint(summary(res.in3rp))\nsink()\nwrite.table(data.frame('NAME'=summary(res.in3rp)[1],'GENES'=summary(res.in3rp)[8]),file=paste('/data/PopGen/candidates/All-noAu-in3rpOnly.list',sep=''),row.names=F,quote = F,sep='\\t')\n\n## overlap in candidate SNPs from all continents in In(3R)P only\nsink(paste('/data/PopGen/candidates/All-in3rpOnly.txt',sep=''))\npdf(paste('/data/PopGen/candidates/All-in3rpOnly.pdf',sep=''),width=8,height=8)\nsets.in3rp.AU=list('Europe'=SEu.c.ov.in3rp,'NorthAmerica'=SNA.c.ov.in3rp,'Africa'=SAf.c.ov.in3rp,'Australia'=SAu.c.ov.in3rp)\nres.in3rp.AU=supertest(sets.in3rp.AU,n=length(Full.in3p))\nplot(res.in3rp.AU,Layout='landscape',degree=2:8,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=50)\ndev.off()\nprint(summary(res.in3rp.AU))\nsink()\nwrite.table(data.frame('NAME'=summary(res.in3rp.AU)[1],'GENES'=summary(res.in3rp.AU)[8]),file=paste('/data/PopGen/candidates/All-in3rpOnly.list',sep=''),row.names=F,quote = F,sep='\\t')\n\n## overlap in candidate SNPs from all continents and in all genomic regions\nsink(paste('/data/PopGen/candidates/All.txt',sep=''))\npdf(paste('/data/PopGen/candidates/All.pdf',sep=''),width=8,height=8)\nsets.AU=list('Europe'=SEu.c.ov,'NorthAmerica'=SNA.c.ov,'Africa'=SAf.c.ov,'Australia'=SAu.c.ov)\nres.AU=supertest(sets.AU,n=length(Full.in3p))\nplot(res.AU,Layout='landscape',degree=2:8,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=0,maxMinusLog10PValue=50)\ndev.off()\nprint(summary(res.AU))\nsink()\nwrite.table(data.frame('NAME'=summary(res.AU)[1],'GENES'=summary(res.AU)[8]),file=paste('/data/PopGen/candidates/All.list',sep=''),row.names=F,quote = F,sep='\\t')\n\n# now gene-based\n\n## get overlapping genes\nSAf.c.ovNoAu.in3rp.genes=SAf.c.genes[SAf.c.genes %in% FullnoAu.in3p.genes]\nSAf.c.ov.in3rp.genes=SAf.c.genes[SAf.c.genes %in% Full.in3p.genes]\nSAf.c.ovNoAu.genes=SAf.c.genes[SAf.c.genes %in% FullnoAu.genes]\nSAf.c.ov.genes=SAf.c.genes[SAf.c.genes %in% Full.genes]\n\nSEu.c.ovNoAu.in3rp.genes=SEu.c.genes[SEu.c.genes %in% FullnoAu.in3p.genes]\nSEu.c.ov.in3rp.genes=SEu.c.genes[SEu.c.genes %in% Full.in3p.genes]\nSEu.c.ovNoAu.genes=SEu.c.genes[SEu.c.genes %in% FullnoAu.genes]\nSEu.c.ov.genes=SEu.c.genes[SEu.c.genes %in% Full.genes]\n\nSNA.c.ovNoAu.in3rp.genes=SNA.c.genes[SNA.c.genes %in% FullnoAu.in3p.genes]\nSNA.c.ov.in3rp.genes=SNA.c.genes[SNA.c.genes %in% Full.in3p.genes]\nSNA.c.ovNoAu.genes=SNA.c.genes[SNA.c.genes %in% FullnoAu.genes]\nSNA.c.ov.genes=SNA.c.genes[SNA.c.genes %in% Full.genes]\n\nSAu.c.ov.in3rp.genes=SAu.c.genes[SAu.c.genes %in% Full.in3p.genes]\nSAu.c.ov.genes=SAu.c.genes[SAu.c.genes %in% Full.genes]\n\n## overlap in candidate genes from all continents except Australia and in all genomic regions\nsink(paste('/data/PopGen/candidates/All-noAu-genes.txt',sep=''))\npdf(paste('/data/PopGen/candidates/All-noAu-genes.pdf',sep=''),width=8,height=8)\nsets=list('Europe'=unique(SEu.c.ovNoAu.genes),'NorthAmerica'=unique(SNA.c.ovNoAu.genes),'Africa'=unique(SAf.c.ovNoAu.genes))\nres=supertest(sets,n=length(FullnoAu))\nplot(res,Layout='landscape',degree=2:4,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=300)\ndev.off()\nprint(summary(res))\nsink()\nwrite.table(data.frame('NAME'=summary(res)[1],'GENES'=summary(res)[8]),file=paste('/data/PopGen/candidates/All-nAU-genes.list',sep=''),row.names=F,quote = F,sep='\\t')\n\n## overlap in candidate genes from all continents except Australia in In(3R)P only\nsink(paste('/data/PopGen/candidates/All-noAu-in3rpOnly-genes.txt',sep=''))\npdf(paste('/data/PopGen/candidates/All-noAu-in3rpOnly-genes.pdf',sep=''),width=8,height=8)\nsets.in3rp=list('Europe'=SEu.c.ovNoAu.in3rp.genes,'NorthAmerica'=SNA.c.ovNoAu.in3rp.genes,'Africa'=SAf.c.ovNoAu.in3rp.genes)\nres.in3rp=supertest(sets.in3rp,n=length(FullnoAu.in3p))\nplot(res.in3rp,Layout='landscape',degree=2:4,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=300)\ndev.off()\nprint(summary(res.in3rp))\nsink()\nwrite.table(data.frame('NAME'=summary(res.in3rp)[1],'GENES'=summary(res.in3rp)[8]),file=paste('/data/PopGen/candidates/All-noAU-in3rpOnly-genes.list',sep=''),row.names=F,quote = F,sep='\\t')\n\n## overlap in candidate genes from all continents in In(3R)P only\nsink(paste('/data/PopGen/candidates/All-in3rpOnly-genes.txt',sep=''))\npdf(paste('/data/PopGen/candidates/All-in3rpOnly-genes.pdf',sep=''),width=8,height=8)\nsets.in3rp.AU=list('Europe'=SEu.c.ov.in3rp.genes,'NorthAmerica'=SNA.c.ov.in3rp.genes,'Africa'=SAf.c.ov.in3rp.genes,'Australia'=SAu.c.ov.in3rp.genes)\nres.in3rp.AU=supertest(sets.in3rp.AU,n=length(Full.in3p))\nplot(res.in3rp.AU,Layout='landscape',degree=2:8,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=200)\ndev.off()\nprint(summary(res.in3rp.AU))\nsink()\nwrite.table(data.frame('NAME'=summary(res.in3rp.AU)[1],'GENES'=summary(res.in3rp.AU)[8]),file=paste('/data/PopGen/candidates/All-in3rpOnly-genes.list',sep=''),row.names=F,quote = F,sep='\\t')\n\n## overlap in candidate genes from all continents and all genomic regions\nsink(paste('/data/PopGen/candidates/All-genes.txt',sep=''))\npdf(paste('/data/PopGen/candidates/All-genes.pdf',sep=''),width=8,height=8)\nsets.AU=list('Europe'=SEu.c.ov.genes,'NorthAmerica'=SNA.c.ov.genes,'Africa'=SAf.c.ov.genes,'Australia'=SAu.c.ov.genes)\nres.AU=supertest(sets.AU,n=length(Full.in3p))\nplot(res.AU,Layout='landscape',degree=2:8,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=0,maxMinusLog10PValue=200)\ndev.off()\nprint(summary(res.AU))\nsink()\nwrite.table(data.frame('NAME'=summary(res.AU)[1],'GENES'=summary(res.AU)[8]),file=paste('/data/PopGen/candidates/All-genes.list',sep=''),row.names=F,quote = F,sep='\\t')\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# first SNP-wise comparison\u003c/span\u003e\n\nlibrary(\u003cspan class=\"pl-smi\"\u003eSuperExactTest\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# read background and idenitfy SNPs inside the inversion\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eSEu.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ena.omit(read.table(gzfile(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/PI-PS_ann.weir.fst\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ena.string\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enan\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSEu.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eSEu.f\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSEu.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e3R\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSEu.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16432209\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSEu.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e24744010\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSAf.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ena.omit(read.table(gzfile(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/ZI-ZS_ann.weir.fst\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ena.string\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enan\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSAf.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eSAf.f\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAf.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e3R\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSAf.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16432209\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSAf.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e24744010\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSAu.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ena.omit(read.table(gzfile(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/II-IS_ann.weir.fst\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ena.string\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enan\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSAu.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eSAu.f\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAu.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e3R\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSAu.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16432209\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSAu.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e24744010\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSNA.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ena.omit(read.table(gzfile(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/FI-FS_ann.weir.fst\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ena.string\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enan\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSNA.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eSNA.f\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNA.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e3R\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSNA.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16432209\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSNA.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e24744010\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# now for genes\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eSAf.f.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSAf.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSAf.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSEu.f.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSEu.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSEu.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSAu.f.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSAu.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSAu.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSNA.f.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSNA.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSNA.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\n\u003cspan class=\"pl-v\"\u003eSAf.f.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSAf.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSAf.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSEu.f.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSEu.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSEu.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSAu.f.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSAu.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSAu.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSNA.f.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSNA.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSNA.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# now for SNPs\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eSEu.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(paste(\u003cspan class=\"pl-smi\"\u003eSEu.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSEu.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSEu.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(paste(\u003cspan class=\"pl-smi\"\u003eSEu.f.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSEu.f.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSAf.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(paste(\u003cspan class=\"pl-smi\"\u003eSAf.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSAf.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSAf.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(paste(\u003cspan class=\"pl-smi\"\u003eSAf.f.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSAf.f.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSAu.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(paste(\u003cspan class=\"pl-smi\"\u003eSAu.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSAu.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSAu.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(paste(\u003cspan class=\"pl-smi\"\u003eSAu.f.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSAu.f.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSNA.f\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(paste(\u003cspan class=\"pl-smi\"\u003eSNA.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSNA.f\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSNA.f.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(paste(\u003cspan class=\"pl-smi\"\u003eSNA.f.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSNA.f.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e get overlapping SNPs \u0026amp; genes\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eFullnoAu\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eReduce(\u003cspan class=\"pl-smi\"\u003eintersect\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eSEu.f\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAf.f\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNA.f\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eFullnoAu.in3p\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eReduce(\u003cspan class=\"pl-smi\"\u003eintersect\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eSEu.f.in3rp\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAf.f.in3rp\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNA.f.in3rp\u003c/span\u003e))\n\n\u003cspan class=\"pl-v\"\u003eFull\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eReduce(\u003cspan class=\"pl-smi\"\u003eintersect\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eSEu.f\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAf.f\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNA.f\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAu.f\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eFull.in3p\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eReduce(\u003cspan class=\"pl-smi\"\u003eintersect\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eSEu.f.in3rp\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAf.f.in3rp\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNA.f.in3rp\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAu.f.in3rp\u003c/span\u003e))\n\n\u003cspan class=\"pl-v\"\u003eFullnoAu.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eReduce(\u003cspan class=\"pl-smi\"\u003eintersect\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eSAf.f.genes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSEu.f.genes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNA.f.genes\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eFullnoAu.in3p.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eReduce(\u003cspan class=\"pl-smi\"\u003eintersect\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eSEu.f.in3rp.genes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAf.f.in3rp.genes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNA.f.in3rp.genes\u003c/span\u003e))\n\n\u003cspan class=\"pl-v\"\u003eFull.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eReduce(\u003cspan class=\"pl-smi\"\u003eintersect\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eSEu.f.genes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAf.f.genes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNA.f.genes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAu.f.genes\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eFull.in3p.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eReduce(\u003cspan class=\"pl-smi\"\u003eintersect\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eSEu.f.in3rp.genes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAf.f.in3rp.genes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNA.f.in3rp.genes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAu.f.in3rp.genes\u003c/span\u003e))\n\ndir.create(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e candidates\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eSEu.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/PI-PS_cand_ann.weir.fst\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSEu.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e3R\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16432209\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e24744010\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSEu.c.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSEu.c.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSEu.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSEu.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSEu.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(paste(\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSEu.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(paste(\u003cspan class=\"pl-smi\"\u003eSEu.c.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSEu.c.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSEu.c.ovNoAu.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu.in3p\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSEu.c.ov.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.in3p\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSEu.c.ovNoAu\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSEu.c.ov\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSEu.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull\u003c/span\u003e]\n\n\u003cspan class=\"pl-v\"\u003eSAf.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/ZI-ZS_cand_ann.weir.fst\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSAf.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e3R\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16432209\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e24744010\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSAf.c.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSAf.c.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSAf.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSAf.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSAf.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(paste(\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eSAf.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eSAf.c.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSAf.c.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSAf.c.ovNoAu.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu.in3p\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSAf.c.ov.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.in3p\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSAf.c.ovNoAu\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSAf.c.ov\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAf.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull\u003c/span\u003e]\n\n\u003cspan class=\"pl-v\"\u003eSAu.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/II-IS_cand_ann.weir.fst\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSAu.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e3R\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16432209\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e24744010\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSAu.c.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSAu.c.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSAu.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSAu.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSAu.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSAu.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eSAu.c.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSAu.c.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSAu.c.ov.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.in3p\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSAu.c.ov\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAu.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull\u003c/span\u003e]\n\n\n\u003cspan class=\"pl-v\"\u003eSNA.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/FI-FS_cand_ann.weir.fst\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSNA.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e3R\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16432209\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e24744010\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSNA.c.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSNA.c.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSNA.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e[sapply(\u003cspan class=\"pl-smi\"\u003eSNA.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV5\u003c/span\u003e,\u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e})])\n\u003cspan class=\"pl-v\"\u003eSNA.c\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSNA.c.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eSNA.c.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eSNA.c.in3rp\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSNA.c.ovNoAu.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu.in3p\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSNA.c.ov.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.in3p\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSNA.c.ovNoAu\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSNA.c.ov\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSNA.c\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull\u003c/span\u003e]\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# overlap in candidate SNPs from all continents except Australia and in all genomic regions\u003c/span\u003e\nsink(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-noAu.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\npdf(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-noAu.pdf\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSEu.c.ovNoAu\u003c/span\u003e),\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNorthAmerica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSNA.c.ovNoAu\u003c/span\u003e),\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSAf.c.ovNoAu\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esupertest(\u003cspan class=\"pl-smi\"\u003esets\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eFullnoAu\u003c/span\u003e))\nplot(\u003cspan class=\"pl-smi\"\u003eres\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eLayout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003elandscape\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edegree\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esize\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ex.pos\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eminMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emaxMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\ndev.off()\nprint(summary(\u003cspan class=\"pl-smi\"\u003eres\u003c/span\u003e))\nsink()\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNAME\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eGENES\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e]),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-noAU.list\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# overlap in candidate SNPs from all continents except Australia and in In(3R)P only\u003c/span\u003e\nsink(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-noAu-in3rpOnly.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\npdf(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-noAu-in3rpOnly.pdf\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esets.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c.ovNoAu.in3rp\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNorthAmerica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c.ovNoAu.in3rp\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c.ovNoAu.in3rp\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eres.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esupertest(\u003cspan class=\"pl-smi\"\u003esets.in3rp\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eFullnoAu.in3p\u003c/span\u003e))\nplot(\u003cspan class=\"pl-smi\"\u003eres.in3rp\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eLayout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003elandscape\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edegree\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esize\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ex.pos\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eminMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emaxMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\ndev.off()\nprint(summary(\u003cspan class=\"pl-smi\"\u003eres.in3rp\u003c/span\u003e))\nsink()\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNAME\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.in3rp\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eGENES\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.in3rp\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e]),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-noAu-in3rpOnly.list\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# overlap in candidate SNPs from all continents in In(3R)P only\u003c/span\u003e\nsink(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-in3rpOnly.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\npdf(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-in3rpOnly.pdf\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esets.in3rp.AU\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c.ov.in3rp\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNorthAmerica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c.ov.in3rp\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c.ov.in3rp\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAustralia\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAu.c.ov.in3rp\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eres.in3rp.AU\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esupertest(\u003cspan class=\"pl-smi\"\u003esets.in3rp.AU\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eFull.in3p\u003c/span\u003e))\nplot(\u003cspan class=\"pl-smi\"\u003eres.in3rp.AU\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eLayout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003elandscape\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edegree\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esize\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ex.pos\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eminMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emaxMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e)\ndev.off()\nprint(summary(\u003cspan class=\"pl-smi\"\u003eres.in3rp.AU\u003c/span\u003e))\nsink()\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNAME\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.in3rp.AU\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eGENES\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.in3rp.AU\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e]),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-in3rpOnly.list\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# overlap in candidate SNPs from all continents and in all genomic regions\u003c/span\u003e\nsink(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\npdf(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All.pdf\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esets.AU\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c.ov\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNorthAmerica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c.ov\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c.ov\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAustralia\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAu.c.ov\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eres.AU\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esupertest(\u003cspan class=\"pl-smi\"\u003esets.AU\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eFull.in3p\u003c/span\u003e))\nplot(\u003cspan class=\"pl-smi\"\u003eres.AU\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eLayout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003elandscape\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edegree\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esize\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ex.pos\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eminMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emaxMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e)\ndev.off()\nprint(summary(\u003cspan class=\"pl-smi\"\u003eres.AU\u003c/span\u003e))\nsink()\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNAME\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.AU\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eGENES\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.AU\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e]),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All.list\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now gene-based\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# get overlapping genes\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eSAf.c.ovNoAu.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAf.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu.in3p.genes\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSAf.c.ov.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAf.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.in3p.genes\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSAf.c.ovNoAu.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAf.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu.genes\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSAf.c.ov.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAf.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.genes\u003c/span\u003e]\n\n\u003cspan class=\"pl-v\"\u003eSEu.c.ovNoAu.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSEu.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu.in3p.genes\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSEu.c.ov.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSEu.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.in3p.genes\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSEu.c.ovNoAu.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSEu.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu.genes\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSEu.c.ov.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSEu.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.genes\u003c/span\u003e]\n\n\u003cspan class=\"pl-v\"\u003eSNA.c.ovNoAu.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSNA.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu.in3p.genes\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSNA.c.ov.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSNA.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.in3p.genes\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSNA.c.ovNoAu.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSNA.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullnoAu.genes\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSNA.c.ov.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSNA.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.genes\u003c/span\u003e]\n\n\u003cspan class=\"pl-v\"\u003eSAu.c.ov.in3rp.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAu.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAu.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.in3p.genes\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eSAu.c.ov.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAu.c.genes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eSAu.c.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFull.genes\u003c/span\u003e]\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# overlap in candidate genes from all continents except Australia and in all genomic regions\u003c/span\u003e\nsink(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-noAu-genes.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\npdf(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-noAu-genes.pdf\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSEu.c.ovNoAu.genes\u003c/span\u003e),\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNorthAmerica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSNA.c.ovNoAu.genes\u003c/span\u003e),\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunique(\u003cspan class=\"pl-smi\"\u003eSAf.c.ovNoAu.genes\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esupertest(\u003cspan class=\"pl-smi\"\u003esets\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eFullnoAu\u003c/span\u003e))\nplot(\u003cspan class=\"pl-smi\"\u003eres\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eLayout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003elandscape\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edegree\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esize\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ex.pos\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eminMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emaxMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\ndev.off()\nprint(summary(\u003cspan class=\"pl-smi\"\u003eres\u003c/span\u003e))\nsink()\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNAME\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eGENES\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e]),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-nAU-genes.list\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# overlap in candidate genes from all continents except Australia in In(3R)P only\u003c/span\u003e\nsink(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-noAu-in3rpOnly-genes.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\npdf(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-noAu-in3rpOnly-genes.pdf\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esets.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c.ovNoAu.in3rp.genes\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNorthAmerica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c.ovNoAu.in3rp.genes\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c.ovNoAu.in3rp.genes\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eres.in3rp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esupertest(\u003cspan class=\"pl-smi\"\u003esets.in3rp\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eFullnoAu.in3p\u003c/span\u003e))\nplot(\u003cspan class=\"pl-smi\"\u003eres.in3rp\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eLayout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003elandscape\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edegree\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esize\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ex.pos\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eminMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emaxMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\ndev.off()\nprint(summary(\u003cspan class=\"pl-smi\"\u003eres.in3rp\u003c/span\u003e))\nsink()\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNAME\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.in3rp\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eGENES\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.in3rp\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e]),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-noAU-in3rpOnly-genes.list\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# overlap in candidate genes from all continents in In(3R)P only\u003c/span\u003e\nsink(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-in3rpOnly-genes.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\npdf(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-in3rpOnly-genes.pdf\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esets.in3rp.AU\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c.ov.in3rp.genes\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNorthAmerica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c.ov.in3rp.genes\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c.ov.in3rp.genes\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAustralia\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAu.c.ov.in3rp.genes\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eres.in3rp.AU\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esupertest(\u003cspan class=\"pl-smi\"\u003esets.in3rp.AU\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eFull.in3p\u003c/span\u003e))\nplot(\u003cspan class=\"pl-smi\"\u003eres.in3rp.AU\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eLayout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003elandscape\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edegree\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esize\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ex.pos\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eminMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emaxMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e200\u003c/span\u003e)\ndev.off()\nprint(summary(\u003cspan class=\"pl-smi\"\u003eres.in3rp.AU\u003c/span\u003e))\nsink()\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNAME\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.in3rp.AU\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eGENES\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.in3rp.AU\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e]),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-in3rpOnly-genes.list\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# overlap in candidate genes from all continents and all genomic regions\u003c/span\u003e\nsink(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-genes.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\npdf(paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-genes.pdf\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esets.AU\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSEu.c.ov.genes\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNorthAmerica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNA.c.ov.genes\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAf.c.ov.genes\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAustralia\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSAu.c.ov.genes\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eres.AU\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esupertest(\u003cspan class=\"pl-smi\"\u003esets.AU\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eFull.in3p\u003c/span\u003e))\nplot(\u003cspan class=\"pl-smi\"\u003eres.AU\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eLayout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003elandscape\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edegree\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esize\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ex.pos\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eminMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emaxMinusLog10PValue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e200\u003c/span\u003e)\ndev.off()\nprint(summary(\u003cspan class=\"pl-smi\"\u003eres.AU\u003c/span\u003e))\nsink()\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNAME\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.AU\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eGENES\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esummary(\u003cspan class=\"pl-smi\"\u003eres.AU\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e]),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/PopGen/candidates/All-genes.list\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.3) African origin\u003c/h3\u003e\u003ca id=\"user-content-33-african-origin\" class=\"anchor\" aria-label=\"Permalink: 3.3) African origin\" href=\"#33-african-origin\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHere, we tested if inversion-specific candidate alleles in Florida are also found in the standard arrangement in flies for the ancestral African origin.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nmkdir /data/InvAlleles\n\ncontinent=( \u0026quot;Africa\u0026quot; \u0026quot;America\u0026quot; )\nregion=( \u0026quot;Zambia\u0026quot; \u0026quot;Florida\u0026quot;  )\nPop1=( \u0026quot;Africa_Zambia_Inv\u0026quot; \u0026quot;America_Florida_Inv\u0026quot; )\nPop2=( \u0026quot;Africa_Zambia_Std\u0026quot; \u0026quot;America_Florida_Std\u0026quot; )\nFST=( \u0026quot;ZI-ZS\u0026quot; \u0026quot;FI-FS\u0026quot; )\n\nfor index in ${!FST[@]}\ndo\n    ## Identify inversion-specific allele in candidates from specific region\n    python3 /scripts/InvAf_from_VCF.py \\\n        --input /data/consensus/${continent[index]}.vcf.gz \\\n        --inv /data/${Pop1[index]}.samples \\\n        --std /data/${Pop2[index]}.samples \\\n        | python3 /scripts/v3/OverlapSNPs.py \\\n        --source /data/PopGen/${FST[index]}_cand_ann.weir.fst \\\n        --target - \\\n        \u0026gt; /data/InvAlleles/${continent[index]}.alleles\n    \n    Sample=( \u0026quot;Africa_Zambia_Inv\u0026quot; \u0026quot;Africa_Zambia_Std\u0026quot; \u0026quot;Europe_Portugal_Inv\u0026quot; \u0026quot;Europe_Portugal_Std\u0026quot; \u0026quot;Europe_Sweden_Std\u0026quot; \u0026quot;America_Florida_Inv\u0026quot; \u0026quot;America_Florida_Std\u0026quot; \u0026quot;America_Maine_Std\u0026quot; \u0026quot;Australia_Queensland_Inv\u0026quot; \u0026quot;Australia_Queensland_Std\u0026quot; \u0026quot;Australia_Victoria_Std\u0026quot; )\n\n    for index2 in ${!Sample[@]}\n    do\n        ## get allele frequency of inversion-specific allele in subset of data\n        python3 /scripts/AF_by_alleleNind.py \\\n            --input /data/consensus/${continent[index]}.vcf.gz \\\n            --AFl /data/InvAlleles/${continent[index]}.alleles \\\n            --pops /data/${Sample[index2]}.samples \\\n            \u0026gt; /data/InvAlleles/${continent[index]}_${Sample[index2]}.af\n\n    done\ndone\"\u003e\u003cpre\u003emkdir /data/InvAlleles\n\ncontinent=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\nregion=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZambia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFlorida\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e  )\nPop1=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica_Zambia_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Florida_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\nPop2=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica_Zambia_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Florida_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\nFST=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZI-ZS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFI-FS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eindex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003eFST[@]}\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Identify inversion-specific allele in candidates from specific region\u003c/span\u003e\n    python3 /scripts/InvAf_from_VCF.py \\\n        --input /data/consensus/\u003cspan class=\"pl-smi\"\u003e${continent[index]}\u003c/span\u003e.vcf.gz \\\n        --inv /data/\u003cspan class=\"pl-smi\"\u003e${Pop1[index]}\u003c/span\u003e.samples \\\n        --std /data/\u003cspan class=\"pl-smi\"\u003e${Pop2[index]}\u003c/span\u003e.samples \\\n        \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e python3 /scripts/v3/OverlapSNPs.py \\\n        --source /data/PopGen/\u003cspan class=\"pl-smi\"\u003e${FST[index]}\u003c/span\u003e_cand_ann.weir.fst \\\n        --target - \\\n        \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/InvAlleles/\u003cspan class=\"pl-smi\"\u003e${continent[index]}\u003c/span\u003e.alleles\n    \n    Sample=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica_Zambia_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica_Zambia_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope_Portugal_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope_Portugal_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope_Sweden_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Florida_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Florida_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Maine_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia_Queensland_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia_Queensland_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia_Victoria_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\n\n    \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eindex2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003eSample[@]}\u003c/span\u003e\n    \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# get allele frequency of inversion-specific allele in subset of data\u003c/span\u003e\n        python3 /scripts/AF_by_alleleNind.py \\\n            --input /data/consensus/\u003cspan class=\"pl-smi\"\u003e${continent[index]}\u003c/span\u003e.vcf.gz \\\n            --AFl /data/InvAlleles/\u003cspan class=\"pl-smi\"\u003e${continent[index]}\u003c/span\u003e.alleles \\\n            --pops /data/\u003cspan class=\"pl-smi\"\u003e${Sample[index2]}\u003c/span\u003e.samples \\\n            \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/InvAlleles/\u003cspan class=\"pl-smi\"\u003e${continent[index]}\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e${Sample[index2]}\u003c/span\u003e.af\n\n    \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow plot with \u003cem\u003eR\u003c/em\u003e for inversion-specific alleles in North America\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nFlS=read.table(\u0026quot;/data/InvAlleles/America_America_Florida_S.af\u0026quot;,header=F)\nFlI=read.table(\u0026quot;/data/InvAlleles/America_America_Florida_I.af\u0026quot;,header=F)\nMaS=read.table(\u0026quot;/data/InvAlleles/America_America_Maine_S.af\u0026quot;,header=F)\nZaI=read.table(\u0026quot;/data/InvAlleles/America_Africa_Zambia_I.af\u0026quot;,header=F)\nZaS=read.table(\u0026quot;/data/InvAlleles/America_Africa_Zambia_S.af\u0026quot;,header=F)\nPoI=read.table(\u0026quot;/data/InvAlleles/America_Europe_Portugal_I.af\u0026quot;,header=F)\nPoS=read.table(\u0026quot;/data/InvAlleles/America_Europe_Portugal_S.af\u0026quot;,header=F)\nSwS=read.table(\u0026quot;/data/InvAlleles/America_Europe_Sweden_S.af\u0026quot;,header=F)\n\nINTER\u0026lt;-Reduce(intersect,list(FlS$V2,FlI$V2,MaS$V2,ZaI$V2,ZaS$V2,PoI$V2,PoS$V2,SwS$V2))\n\nNEW=list(\u0026quot;Florida-I\u0026quot;=FlI[,4][FlI$V2 %in% INTER],\u0026quot;Florida-S\u0026quot;=FlS[,4][FlS$V2 %in% INTER],\u0026quot;Maine-S\u0026quot;=MaS[,4][MaS$V2 %in% INTER],\u0026quot;Zambia-I\u0026quot;=ZaI[,4][ZaI$V2 %in% INTER],\u0026quot;Zambia-S\u0026quot;=ZaS[,4][ZaS$V2 %in% INTER],\u0026quot;Portugal-I\u0026quot;=PoI[,4][PoI$V2 %in% INTER],\u0026quot;Portugal-S\u0026quot;=PoS[,4][PoS$V2 %in% INTER],\u0026quot;Sweden-S\u0026quot;=SwS[,4][SwS$V2 %in% INTER])\n\n## plot AFs in all populations\npdf(\u0026quot;/data/InvAlleles/Inversion-specific-AF.pdf\u0026quot;,width=12,height=6)\nboxplot(NEW,ylab=\u0026quot;Allele Frequencies\u0026quot;)\ndev.off()\n\n## add position information and write to file\nNEW=data.frame(list(\u0026quot;POS\u0026quot;=FlI[,2][FlI$V2 %in% INTER],\u0026quot;Florida-I\u0026quot;=FlI[,4][FlI$V2 %in% INTER],\u0026quot;Florida-S\u0026quot;=FlS[,4][FlS$V2 %in% INTER],\u0026quot;Maine-S\u0026quot;=MaS[,4][MaS$V2 %in% INTER],\u0026quot;Zambia-I\u0026quot;=ZaI[,4][ZaI$V2 %in% INTER],\u0026quot;Zambia-S\u0026quot;=ZaS[,4][ZaS$V2 %in% INTER],\u0026quot;Portugal-I\u0026quot;=PoI[,4][PoI$V2 %in% INTER],\u0026quot;Portugal-S\u0026quot;=PoS[,4][PoS$V2 %in% INTER],\u0026quot;Sweden-S\u0026quot;=SwS[,4][SwS$V2 %in% INTER]))\n\nwrite.table(NEW,\u0026quot;/data/InvAlleles/Inversion-specific-AF.txt\u0026quot;,quote=F,row.names=F)\n\n## insolate positions with ABS(STD-INV)\u0026gt;0.5 in Africa\nNEW2=NEW[abs(NEW[,5]-NEW[,6])\u0026gt;0.5,]\nwrite.table(NEW2,\u0026quot;/data/InvAlleles/Inversion-specific-AF0.5.txt\u0026quot;,quote=F,row.names=F)\n\nH=hist(NEW2[,1],breaks=100,xlim=c(0,32000000))\n\n# plot genomic positions of these SNPS\npdf(\u0026quot;/data/InvAlleles/Inversion-specific-AF_dist.pdf\u0026quot;,width=12,height=6)\nplot(H$mids/1000000,H$counts,type=\u0026quot;l\u0026quot;,xlim=c(15,27),xlab=\u0026quot;Genomic Position (Mbp)\u0026quot;,ylab=\u0026quot;No. of SNPs\u0026quot;)\nrect(16.432209,0,24.744010,max(H$counts),border=\u0026quot;grey\u0026quot;,lty=2)\ndev.off()\n\n## test if significant allelic difference between STD and INV in Africa\nsink(\u0026quot;/data/InvAlleles/Inversion-specific-AF.stat\u0026quot;)\nwilcox.test(NEW[,5],NEW[,6])\nsink()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003eFlS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/America_America_Florida_S.af\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eFlI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/America_America_Florida_I.af\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eMaS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/America_America_Maine_S.af\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eZaI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/America_Africa_Zambia_I.af\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eZaS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/America_Africa_Zambia_S.af\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003ePoI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/America_Europe_Portugal_I.af\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003ePoS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/America_Europe_Portugal_S.af\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSwS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/America_Europe_Sweden_S.af\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eReduce(\u003cspan class=\"pl-smi\"\u003eintersect\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eFlS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eFlI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eMaS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eZaI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eZaS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ePoI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ePoS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSwS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e))\n\n\u003cspan class=\"pl-v\"\u003eNEW\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFlorida-I\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFlI\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eFlI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFlorida-S\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFlS\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eFlS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMaine-S\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMaS\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eMaS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZambia-I\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZaI\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eZaI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZambia-S\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZaS\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eZaS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePortugal-I\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePoI\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003ePoI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePortugal-S\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePoS\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003ePoS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSweden-S\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSwS\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eSwS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e])\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# plot AFs in all populations\u003c/span\u003e\npdf(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/Inversion-specific-AF.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\nboxplot(\u003cspan class=\"pl-smi\"\u003eNEW\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAllele Frequencies\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# add position information and write to file\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eNEW\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePOS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFlI\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eFlI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFlorida-I\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFlI\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eFlI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFlorida-S\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFlS\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eFlS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMaine-S\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMaS\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eMaS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZambia-I\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZaI\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eZaI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZambia-S\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZaS\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eZaS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePortugal-I\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePoI\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003ePoI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePortugal-S\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePoS\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003ePoS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSweden-S\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSwS\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e][\u003cspan class=\"pl-smi\"\u003eSwS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eINTER\u003c/span\u003e]))\n\nwrite.table(\u003cspan class=\"pl-smi\"\u003eNEW\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/Inversion-specific-AF.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# insolate positions with ABS(STD-INV)\u0026gt;0.5 in Africa\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eNEW2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eNEW\u003c/span\u003e[abs(\u003cspan class=\"pl-smi\"\u003eNEW\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eNEW\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e])\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e,]\nwrite.table(\u003cspan class=\"pl-smi\"\u003eNEW2\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/Inversion-specific-AF0.5.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003eH\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ehist(\u003cspan class=\"pl-smi\"\u003eNEW2\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003ebreaks\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e32000000\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e plot genomic positions of these SNPS\u003c/span\u003e\npdf(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/Inversion-specific-AF_dist.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\nplot(\u003cspan class=\"pl-smi\"\u003eH\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emids\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1000000\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eH\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003el\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e27\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGenomic Position (Mbp)\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNo. of SNPs\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nrect(\u003cspan class=\"pl-c1\"\u003e16.432209\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e24.744010\u003c/span\u003e,max(\u003cspan class=\"pl-smi\"\u003eH\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003eborder\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egrey\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\ndev.off()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# test if significant allelic difference between STD and INV in Africa\u003c/span\u003e\nsink(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/InvAlleles/Inversion-specific-AF.stat\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nwilcox.test(\u003cspan class=\"pl-smi\"\u003eNEW\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eNEW\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e])\nsink()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.4) Linkage Disequilibrium\u003c/h3\u003e\u003ca id=\"user-content-34-linkage-disequilibrium\" class=\"anchor\" aria-label=\"Permalink: 3.4) Linkage Disequilibrium\" href=\"#34-linkage-disequilibrium\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHere, we assessed how linkage disequilibrium decays within the genomic region spanned by \u003cem\u003eIn(3R)Payne\u003c/em\u003e in inverted and non-inverted haplotypes.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.4.1) SNP-wise LD with the inversion\u003c/h4\u003e\u003ca id=\"user-content-341-snp-wise-ld-with-the-inversion\" class=\"anchor\" aria-label=\"Permalink: 3.4.1) SNP-wise LD with the inversion\" href=\"#341-snp-wise-ld-with-the-inversion\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFirst, we calculated and plotted LD of each SNP with \u003cem\u003eIn(3R)Payne\u003c/em\u003e along the 3R chromosomal arm with the Inversion.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nmkdir /data/LD\n\n## Africa\npython3 /scripts/v3/LD_bar.py \\\n--input /data/Africa/Africa.cons.gz \\\n--output /data/LD/Zambia_Bar \\\n--ind ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388 \\\n--chromosome 3R \\\n--N-cutoff 0.1 \\\n--color blue,green,yellow,red,purple \\\n--min-allele 0.1 \\\n--Inv /data/Africa_Zambia_Inv.samples \\\n--Std /data/Africa_Zambia_Std.samples \\\n\n## Europe\npython3 /scripts/v3/LD_bar.py \\\n--input /data/Europe/Europe.cons.gz \\\n--output /data/LD/Portugal_Bar \\\n--ind F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94 \\\n--chromosome 3R \\\n--N-cutoff 0.1 \\\n--color blue,green,yellow,red,purple \\\n--min-allele 0.1 \\\n--Inv /data/Europe_Portugal_Inv.samples \\\n--Std /data/Europe_Portugal_Std.samples \\\n\n## America\npython3 /scripts/v3/LD_bar.py \\\n--input /data/USA/USA.cons.gz \\\n--output /data/LD/USA_Bar \\\n--ind Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81 \\\n--chromosome 3R \\\n--N-cutoff 0.1 \\\n--color blue,green,yellow,red,purple \\\n--min-allele 0.1 \\\n--Inv /data/America_Florida_Inv.samples \\\n--Std /data/America_Florida_Std.samples \\\n\n## Australia\npython3 /scripts/v3/LD_bar.py \\\n--input /data/Australia/Australia.cons.gz \\\n--output /data/LD/Australia_Bar \\\n--ind Ii1,Ii2,Ii3,Ii4,Ii5,Ii6,Ii7,Ii8,Ii9,Ii10,Ii11,Ii12,Ii13,Ii14,Ii15,Ii16,Ii17,Ii18,Ii19,Is1,Is2,Is3,Is4,Is5,Is6,Is7,Is8,Is9,Is10,Is11,Is12,Is13,Is14,Is15,Is16,Is17,Is18,YS1,YS2,YS3,YS4,YS5,YS6,YS7,YS8,YS9,YS10,YS11,YS12,YS13,YS14,YS15,YS16,YS17,YS18 \\\n--chromosome 3R \\\n--N-cutoff 0.1 \\\n--color blue,green,yellow,red,purple \\\n--min-allele 0.1 \\\n--Inv /data/Australia_Queensland_Inv.samples \\\n--Std /data/Australia_Queensland_Std.samples \\\n\"\u003e\u003cpre\u003emkdir /data/LD\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Africa\u003c/span\u003e\npython3 /scripts/v3/LD_bar.py \\\n--input /data/Africa/Africa.cons.gz \\\n--output /data/LD/Zambia_Bar \\\n--ind ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388 \\\n--chromosome 3R \\\n--N-cutoff 0.1 \\\n--color blue,green,yellow,red,purple \\\n--min-allele 0.1 \\\n--Inv /data/Africa_Zambia_Inv.samples \\\n--Std /data/Africa_Zambia_Std.samples \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Europe\u003c/span\u003e\npython3 /scripts/v3/LD_bar.py \\\n--input /data/Europe/Europe.cons.gz \\\n--output /data/LD/Portugal_Bar \\\n--ind F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94 \\\n--chromosome 3R \\\n--N-cutoff 0.1 \\\n--color blue,green,yellow,red,purple \\\n--min-allele 0.1 \\\n--Inv /data/Europe_Portugal_Inv.samples \\\n--Std /data/Europe_Portugal_Std.samples \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# America\u003c/span\u003e\npython3 /scripts/v3/LD_bar.py \\\n--input /data/USA/USA.cons.gz \\\n--output /data/LD/USA_Bar \\\n--ind Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81 \\\n--chromosome 3R \\\n--N-cutoff 0.1 \\\n--color blue,green,yellow,red,purple \\\n--min-allele 0.1 \\\n--Inv /data/America_Florida_Inv.samples \\\n--Std /data/America_Florida_Std.samples \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Australia\u003c/span\u003e\npython3 /scripts/v3/LD_bar.py \\\n--input /data/Australia/Australia.cons.gz \\\n--output /data/LD/Australia_Bar \\\n--ind Ii1,Ii2,Ii3,Ii4,Ii5,Ii6,Ii7,Ii8,Ii9,Ii10,Ii11,Ii12,Ii13,Ii14,Ii15,Ii16,Ii17,Ii18,Ii19,Is1,Is2,Is3,Is4,Is5,Is6,Is7,Is8,Is9,Is10,Is11,Is12,Is13,Is14,Is15,Is16,Is17,Is18,YS1,YS2,YS3,YS4,YS5,YS6,YS7,YS8,YS9,YS10,YS11,YS12,YS13,YS14,YS15,YS16,YS17,YS18 \\\n--chromosome 3R \\\n--N-cutoff 0.1 \\\n--color blue,green,yellow,red,purple \\\n--min-allele 0.1 \\\n--Inv /data/Australia_Queensland_Inv.samples \\\n--Std /data/Australia_Queensland_Std.samples \\\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.4.2) LD heatmaps\u003c/h4\u003e\u003ca id=\"user-content-342-ld-heatmaps\" class=\"anchor\" aria-label=\"Permalink: 3.4.2) LD heatmaps\" href=\"#342-ld-heatmaps\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNext, we calculated LD heatmaps within an among karyotypes focusing on a subset of 5000 SNPs from populations with mixed karyotypes in different continents. The numbers of the parameter \u003ccode\u003eind\u003c/code\u003e correspond to the (zero-based) position of the chosen individuals (from *.samples files) in the consensus block. See \u003ca href=\"https://onlinelibrary.wiley.com/doi/full/10.1111/mec.12594\" rel=\"nofollow\"\u003eKapun \u003cem\u003eet al.\u003c/em\u003e 2014\u003c/a\u003e for more details. Below, we are showing the code for the LD analysis within karyotypes, the \u003ccode\u003eind\u003c/code\u003e parameter needs to be adjusted (based on the *.sample files in data/) when repeating the analyses for LD within karyotypes.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n## Zambia - full\npython3 /scripts/LD_heatmap.py \\\n    --input /data/Africa/Africa.cons.gz \\\n    --output /data/LD/Zambia \\\n    --subsample 5000 \\\n    --ind 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,20,21,31,32,33,51,54,72,74,76,90,94,105,114,116,119,120,126,135,137,146,151,158 \\\n    --chromosome 3R \\\n    --N-cutoff 0.1 \\\n    --color purple,red,yellow,green,blue \\\n    --min-allele 0.3\n\n## Portugal - full \npython3 /scripts/LD_heatmap.py \\\n    --input /data/Europe/Europe.cons.gz \\\n    --output /data/LD/Portugal \\\n    --subsample 5000 \\\n    --ind 0,1,2,3,4,5,6,10,11,13,17,22,23,25 \\\n    --chromosome 3R \\\n    --N-cutoff 0.1 \\\n    --color purple,red,yellow,green,blue \\\n    --min-allele 0.3\n\n## Florida - full \npython3 /scripts/LD_heatmap.py \\\n    --input /data/USA/USA.cons.gz \\\n    --output /data/LD/Florida \\\n    --subsample 5000 \\\n    --ind 0,1,2,3,4,5,6,7,8,9,10,11,13,14,15,17,18,19,21,22,23,24,25,26,27,29,30,31,32,33,34,35,36,37,38 \\\n    --chromosome 3R \\\n    --N-cutoff 0.1 \\\n    --color purple,red,yellow,green,blue \\\n    --min-allele 0.3\n\n## Australia - full \npython3 /scripts/LD_heatmap.py \\\n    --input /data/Australia/Australia.cons.gz \\\n    --output /data/LD/Queensland \\\n    --subsample 5000 \\\n    --ind 0,1,3,6,9,10,11,15,16,17,25,32,35,36,12,13,14,19,21,24,26,27,29,31,33 \\\n    --chromosome 3R \\\n    --N-cutoff 0.1 \\\n    --color purple,red,yellow,green,blue \\\n    --min-allele 0.3\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Zambia - full\u003c/span\u003e\npython3 /scripts/LD_heatmap.py \\\n    --input /data/Africa/Africa.cons.gz \\\n    --output /data/LD/Zambia \\\n    --subsample 5000 \\\n    --ind 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,20,21,31,32,33,51,54,72,74,76,90,94,105,114,116,119,120,126,135,137,146,151,158 \\\n    --chromosome 3R \\\n    --N-cutoff 0.1 \\\n    --color purple,red,yellow,green,blue \\\n    --min-allele 0.3\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Portugal - full \u003c/span\u003e\npython3 /scripts/LD_heatmap.py \\\n    --input /data/Europe/Europe.cons.gz \\\n    --output /data/LD/Portugal \\\n    --subsample 5000 \\\n    --ind 0,1,2,3,4,5,6,10,11,13,17,22,23,25 \\\n    --chromosome 3R \\\n    --N-cutoff 0.1 \\\n    --color purple,red,yellow,green,blue \\\n    --min-allele 0.3\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Florida - full \u003c/span\u003e\npython3 /scripts/LD_heatmap.py \\\n    --input /data/USA/USA.cons.gz \\\n    --output /data/LD/Florida \\\n    --subsample 5000 \\\n    --ind 0,1,2,3,4,5,6,7,8,9,10,11,13,14,15,17,18,19,21,22,23,24,25,26,27,29,30,31,32,33,34,35,36,37,38 \\\n    --chromosome 3R \\\n    --N-cutoff 0.1 \\\n    --color purple,red,yellow,green,blue \\\n    --min-allele 0.3\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Australia - full \u003c/span\u003e\npython3 /scripts/LD_heatmap.py \\\n    --input /data/Australia/Australia.cons.gz \\\n    --output /data/LD/Queensland \\\n    --subsample 5000 \\\n    --ind 0,1,3,6,9,10,11,15,16,17,25,32,35,36,12,13,14,19,21,24,26,27,29,31,33 \\\n    --chromosome 3R \\\n    --N-cutoff 0.1 \\\n    --color purple,red,yellow,green,blue \\\n    --min-allele 0.3\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.4.3) Decay of LD\u003c/h4\u003e\u003ca id=\"user-content-343-decay-of-ld\" class=\"anchor\" aria-label=\"Permalink: 3.4.3) Decay of LD\" href=\"#343-decay-of-ld\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNext, we estimated differences in the decay of LD with respect to different karyotypes based on the \u003ccode\u003e*.dist.gz\u003c/code\u003e output files from the analyses above and tested for significant differences with non-linear mixed models.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n## loop through samples\ncontinent=( \u0026quot;Africa\u0026quot; \u0026quot;Europe\u0026quot; \u0026quot;America\u0026quot; \u0026quot;Australia\u0026quot; )\nregion=( \u0026quot;Zambia\u0026quot; \u0026quot;Portugal\u0026quot; \u0026quot;Florida\u0026quot; \u0026quot;Queensland\u0026quot; )\nPop1=( \u0026quot;Africa_Zambia_Inv\u0026quot; \u0026quot;Europe_Portugal_Inv\u0026quot;  \u0026quot;America_Florida_Inv\u0026quot;  \u0026quot;Australia_Queensland_Inv\u0026quot;  )\nPop2=( \u0026quot;Africa_Zambia_Std\u0026quot; \u0026quot;Europe_Portugal_Std\u0026quot; \u0026quot;America_Florida_Std\u0026quot; \u0026quot;Australia_Queensland_Std\u0026quot; )\n\nfor index in ${!Pop1[@]}\ndo  \n\n    ## plot the decay\n    python3 /scripts/v3/LD_decay.py \\\n        --vcf /data/consensus/${continent[index]}.vcf.gz \\\n        --samples /data/${Pop1[index]}.samples,/data/${Pop2[index]}.samples \\\n        --names I,S \\\n        --combsample 50000 \\\n        --SNPs 5000 \\\n        --maxdist 100000 \\\n        --threshold 0.1 \\\n        --N-threshold 0.5 \\\n        --region 3R:16432209-24744010 \\\n        --out /data/LD/${region[index]}_decay\n\n    ## test for significant differences\n    Rscript /scripts/NLS.R /data/LD/${region[index]}_decay\n\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# loop through samples\u003c/span\u003e\ncontinent=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\nregion=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZambia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePortugal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFlorida\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eQueensland\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\nPop1=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica_Zambia_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope_Portugal_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Florida_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia_Queensland_Inv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e  )\nPop2=( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAfrica_Zambia_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEurope_Portugal_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAmerica_Florida_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAustralia_Queensland_Std\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eindex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003ePop1[@]}\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e  \n\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# plot the decay\u003c/span\u003e\n    python3 /scripts/v3/LD_decay.py \\\n        --vcf /data/consensus/\u003cspan class=\"pl-smi\"\u003e${continent[index]}\u003c/span\u003e.vcf.gz \\\n        --samples /data/\u003cspan class=\"pl-smi\"\u003e${Pop1[index]}\u003c/span\u003e.samples,/data/\u003cspan class=\"pl-smi\"\u003e${Pop2[index]}\u003c/span\u003e.samples \\\n        --names I,S \\\n        --combsample 50000 \\\n        --SNPs 5000 \\\n        --maxdist 100000 \\\n        --threshold 0.1 \\\n        --N-threshold 0.5 \\\n        --region 3R:16432209-24744010 \\\n        --out /data/LD/\u003cspan class=\"pl-smi\"\u003e${region[index]}\u003c/span\u003e_decay\n\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# test for significant differences\u003c/span\u003e\n    Rscript /scripts/NLS.R /data/LD/\u003cspan class=\"pl-smi\"\u003e${region[index]}\u003c/span\u003e_decay\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4) Transcriptomic analysis\u003c/h3\u003e\u003ca id=\"user-content-4-transcriptomic-analysis\" class=\"anchor\" aria-label=\"Permalink: 4) Transcriptomic analysis\" href=\"#4-transcriptomic-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.1) Trim and map RNASeq data\u003c/h4\u003e\u003ca id=\"user-content-41-trim-and-map-rnaseq-data\" class=\"anchor\" aria-label=\"Permalink: 4.1) Trim and map RNASeq data\" href=\"#41-trim-and-map-rnaseq-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## loop through all libraries and trim\n\nmkdir -p /data/RNASeq/trimmed\nfor i in /data/RNASeq/*.fq.gz\ndo\n\n    ## use only a subset of the variable i as the name\n    name=${i##*/}\n    n=\u0026quot;$( cut -d '.' -f 1 \u0026lt;\u0026lt;\u0026lt; \u0026quot;$name\u0026quot; )\u0026quot;\n\n    ## trim with cutadapt\n    /scripts/cutadapt \\\n    -q 18 \\\n    --minimum-length 75 \\\n    -o /data/RNASeq/trimmed/$n.fq.gz \\\n    ${i} \n\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# loop through all libraries and trim\u003c/span\u003e\n\nmkdir -p /data/RNASeq/trimmed\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e /data/RNASeq/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fq.gz\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# use only a subset of the variable i as the name\u003c/span\u003e\n    name=\u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e##*/\u003c/span\u003e}\u003c/span\u003e\n    n=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f 1 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u0026lt;\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e$name\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# trim with cutadapt\u003c/span\u003e\n    /scripts/cutadapt \\\n    -q 18 \\\n    --minimum-length 75 \\\n    -o /data/RNASeq/trimmed/\u003cspan class=\"pl-smi\"\u003e$n\u003c/span\u003e.fq.gz \\\n    \u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eGet reference transcriptome\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"cd /data/RNASeq/\ncurl -O ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/dmel_r6.17_FB2017_04/fasta/dmel-all-CDS-r6.17.fasta.gz\ncurl -O ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/dmel_r6.17_FB2017_04/fasta/dmel-all-r6.17.gtf.gz\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /data/RNASeq/\ncurl -O ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/dmel_r6.17_FB2017_04/fasta/dmel-all-CDS-r6.17.fasta.gz\ncurl -O ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/dmel_r6.17_FB2017_04/fasta/dmel-all-r6.17.gtf.gz\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNext, we pseudomapped the trimmed reads with \u003ca href=\"https://github.com/pachterlab/kallisto\"\u003eKallisto\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## index the transcriptome\n/scripts/kallisto index \\\n    -i /data/RNASeq/dmel-all-CDS-r6.17.idx \\\n    /data/RNASeq/dmel-all-CDS-r6.17.fasta.gz\n\nmkdir /data/RNASeq/kallisto\n\n## loop through all libraries and map with kallisto\nfor i in /data/RNASeq/*.fq.gz\ndo\n\n    name=${i##*/}\n    n=\u0026quot;$( cut -d '.' -f 1 \u0026lt;\u0026lt;\u0026lt; \u0026quot;$name\u0026quot; )\u0026quot;\n\n    kallisto quant \\\n    -i /data/RNASeq/dmel-all-CDS-r6.17.idx \\\n    -o /data/RNASeq/kallisto/$n \\\n    --single -l 101 -s 10 -b 100 --rf-stranded -t 24 \\\n    /data/RNASeq/trimmed/$name \\\n\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# index the transcriptome\u003c/span\u003e\n/scripts/kallisto index \\\n    -i /data/RNASeq/dmel-all-CDS-r6.17.idx \\\n    /data/RNASeq/dmel-all-CDS-r6.17.fasta.gz\n\nmkdir /data/RNASeq/kallisto\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# loop through all libraries and map with kallisto\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e /data/RNASeq/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fq.gz\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\n    name=\u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e##*/\u003c/span\u003e}\u003c/span\u003e\n    n=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f 1 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u0026lt;\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e$name\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n    kallisto quant \\\n    -i /data/RNASeq/dmel-all-CDS-r6.17.idx \\\n    -o /data/RNASeq/kallisto/\u003cspan class=\"pl-smi\"\u003e$n\u003c/span\u003e \\\n    --single -l 101 -s 10 -b 100 --rf-stranded -t 24 \\\n    /data/RNASeq/trimmed/\u003cspan class=\"pl-smi\"\u003e$name\u003c/span\u003e \\\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAfter that, we merged the read counts at gene level across all datasets\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n## merge datasets and sum up Pseudocounts for gene-level estimates\nmkdir -p /data/RNASeq/count\n\npython3 /scripts/v3/mergeKallisto.py \\\n    /data/RNASeq/kallisto \\\n    /data/RNASeq/count/analyses \\\n    /data/RNASeq/dmel-all-r6.17.gtf.gz\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# merge datasets and sum up Pseudocounts for gene-level estimates\u003c/span\u003e\nmkdir -p /data/RNASeq/count\n\npython3 /scripts/v3/mergeKallisto.py \\\n    /data/RNASeq/kallisto \\\n    /data/RNASeq/count/analyses \\\n    /data/RNASeq/dmel-all-r6.17.gtf.gz\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.2) Test for differential expression\u003c/h4\u003e\u003ca id=\"user-content-42-test-for-differential-expression\" class=\"anchor\" aria-label=\"Permalink: 4.2) Test for differential expression\" href=\"#42-test-for-differential-expression\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThen, we used \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/edgeR.html\" rel=\"nofollow\"\u003eedgeR\u003c/a\u003e and \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/limma.html\" rel=\"nofollow\"\u003elimma\u003c/a\u003e to test for differential expression (DE) among different factors and test for enrichment of Gene Ontology (GO) terms.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"### Analysis of RNA-Seq data of inversion lines\n\n## install packages\n#install.packages(\u0026quot;readr\u0026quot;)\n#install.packages(\u0026quot;tximport\u0026quot;)\n#install.packages(\u0026quot;edgeR\u0026quot;)\n#install.packages(\u0026quot;RColorBrewer\u0026quot;)\n#install.packages(\u0026quot;limma\u0026quot;)\n#install.packages(\u0026quot;pheatmap\u0026quot;)\n\nI=\u0026quot;/data/RNASeq/count/analyses_counts.txt\u0026quot;\nO=\u0026quot;/data/RNASeq/\u0026quot;\n\n#Factors\n\nTreat=factor(c('FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS'))\nTreat=relevel(Treat,ref=\u0026quot;FS\u0026quot;)\nTemp=factor(c('18','18','18','18','18','18','18','18','18','25','25','25','25','25','25','25','25','25','18','18','18','18','18','18','18','18','18','25','25','25','25','25','25','25','25','25','18','18','18','18','18','18','18','18','18','25','25','25','25','25','25','25','25','25'))\nRep=factor(c(10,14,17,18,22,23,26,5,9,5,9,10,14,17,18,22,23,26,13,16,2,21,27,3,4,6,7,2,3,4,6,7,13,16,21,27,1,11,12,15,19,20,24,25,8,1,8,11,12,15,19,20,24,25))\n\n## 1a) read Data\nrawData\u0026lt;-read.table(I,header=T)\nrawCounts=rawData[,3:56]\nrownames(rawCounts)\u0026lt;-rawData[,1]\n\n## 2) create DGE object using edgeR\nlibrary(edgeR)\nFullDGE\u0026lt;-DGEList(rawCounts,group=names(rawCounts))\n\n## 3) add factors to object\nFullDGE$samples$treat \u0026lt;- Treat\nFullDGE$samples$temp \u0026lt;- Temp\n\n## Transform counts (Counts per million: CPM) and normalize the counts using the TMM method in edgeR\nFullDGE \u0026lt;- calcNormFactors(FullDGE, method = \u0026quot;TMM\u0026quot;)\ncpmFullDGE\u0026lt;-cpm(FullDGE,log=T)\n\nout=paste(O,\u0026quot;/countdata/CPM.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\nTAB=cpmFullDGE\nwrite.table(data.frame(\u0026quot;Gene\u0026quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)\n\n## remove genes covered by less than 2 read per million in at least nine samples\nkeep \u0026lt;- rowSums(cpmFullDGE \u0026gt;= 2) \u0026gt;=9\nFullDGE \u0026lt;- FullDGE[keep,] \ncpmTFullDGE \u0026lt;- cpmFullDGE[keep,] \n\n## Plot density histograms to quantify the effect of filtering out lowly expressed genes\ndir.create(paste(O,\u0026quot;/quality\u0026quot;,sep=\u0026quot;\u0026quot;))\npdf(paste(O,\u0026quot;/quality/CPM-hist.pdf\u0026quot;,sep=\u0026quot;\u0026quot;),width=15,height=5)\nnsamples \u0026lt;- ncol(FullDGE)\ncol \u0026lt;- topo.colors(nsamples)\npar(mfrow=c(1,2))\nplot(density(cpmFullDGE[,1]), col=col[1], lwd=2, ylim=c(0,0.25), las=2,\n     main=\u0026quot;\u0026quot;, xlab=\u0026quot;\u0026quot;)\nabline(v=0, lty=3)\n\nfor (i in 2:nsamples){\n  den \u0026lt;- density(cpmFullDGE[,i])\n  lines(den$x, den$y, col=col[i], lwd=2)\n} \nlegend(\u0026quot;topright\u0026quot;, legend=colnames(FullDGE), text.col=col, bty=\u0026quot;n\u0026quot;,ncol = 3,cex=0.5)\ntitle(main=\u0026quot;A. Raw data\u0026quot;, xlab=\u0026quot;Log-cpm\u0026quot;)\n\nplot(density(cpmTFullDGE[,1]), col=col[1], lwd=2, ylim=c(0,0.25), las=2,\n     main=\u0026quot;\u0026quot;, xlab=\u0026quot;\u0026quot;)\ntitle(main=\u0026quot;B. Filtered data\u0026quot;, xlab=\u0026quot;Log-cpm\u0026quot;) \nabline(v=0, lty=3)\n\nfor (i in 2:nsamples){\n  den \u0026lt;- density(cpmTFullDGE[,i])\n  lines(den$x, den$y, col=col[i], lwd=2)\n} \nlegend(\u0026quot;topright\u0026quot;, legend=colnames(FullDGE), text.col=col, bty=\u0026quot;n\u0026quot;,ncol = 3,cex=0.5)\ndev.off()\n\n## explore Data with MDS to identify clustering\nlibrary(RColorBrewer)\npdf(paste(O,\u0026quot;/quality/MDS.pdf\u0026quot;,sep=\u0026quot;\u0026quot;),width=14,height=12)\npar(mfrow=c(2,2))\n\ncol.Treat \u0026lt;- Treat\nlevels(col.Treat) \u0026lt;- brewer.pal(nlevels(col.Treat), \u0026quot;Set1\u0026quot;)\nplotMDS(cpmTFullDGE, col=as.character(col.Treat))\ntitle(main=\u0026quot;Karypop: Dim 1/2\u0026quot;)\n\ncol.Temp \u0026lt;- Temp\nlevels(col.Temp) \u0026lt;- brewer.pal(nlevels(col.Temp), \u0026quot;Set1\u0026quot;)\nplotMDS(cpmTFullDGE, col=as.character(col.Temp))\ntitle(main=\u0026quot;Temperature: Dim 1/2\u0026quot;)\n\nplotMDS(cpmTFullDGE, col=as.character(col.Treat), dim=c(3,4))\ntitle(main=\u0026quot;Karypop: Dim 3/4\u0026quot;)\n\nplotMDS(cpmTFullDGE, col=as.character(col.Temp), dim=c(3,4))\ntitle(main=\u0026quot;Temperature: Dim 3/4\u0026quot;)\n\ndev.off()\n\n### explore data with PCA to explore clustering\n\npdf(paste(O,\u0026quot;/quality/PCA.pdf\u0026quot;,sep=\u0026quot;\u0026quot;),width=12,height=8)\npar(mfrow=c(2,3))\nnewcpm=t(cpmTFullDGE)\nPCA=prcomp(newcpm)\ncolKP=rainbow(length(levels(Treat)))[Treat]\ncolTemp=rainbow(length(levels(Temp)))[as.factor(Temp)]\ncolPop=rainbow(length(levels(Rep)))[as.factor(Rep)]\n\nplot(PCA$x[,1],PCA$x[,2],col=colKP,main=\u0026quot;Karypop\u0026quot;,pch=16,cex=3)\n#text(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colKP)\n\nplot(PCA$x[,1],PCA$x[,2],col=colTemp,main=\u0026quot;Temp\u0026quot;,pch=16,cex=3)\n#text(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colTemp)\n\nplot(PCA$x[,1],PCA$x[,2],col=colPop,main=\u0026quot;Replicate\u0026quot;,pch=16,cex=3)\n#text(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colPop)\n\nplot(PCA$x[,3],PCA$x[,4],col=colKP,main=\u0026quot;Karypop\u0026quot;,pch=16,cex=3)\n#text(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colKP)\n\nplot(PCA$x[,3],PCA$x[,4],col=colTemp,main=\u0026quot;Temp\u0026quot;,pch=16,cex=3)\n#text(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colTemp)\n\nplot(PCA$x[,3],PCA$x[,4],col=colPop,main=\u0026quot;Replicate\u0026quot;,pch=16,cex=3)\n#text(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colPop)\n\ndev.off()\n\n### Limma analysis with singe grouping factor\n\nlibrary(limma)\n\nTreat=relevel(Treat,ref=\u0026quot;FS\u0026quot;)\ndir.create(paste(O,\u0026quot;/limma\u0026quot;,sep=\u0026quot;\u0026quot;))\nG=interaction(Treat,Temp)\n\ndesign.SP=model.matrix(~0+G)\ncolnames(design.SP)=levels(G)\n\n## Do VOOM\nvoomFullDGE.SP \u0026lt;- voom(FullDGE, design.SP, plot=F)\n\n# write transformed data to file\nout=paste(O,\u0026quot;/countdata/voomCounts.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\nTAB=voomFullDGE.SP$E\nwrite.table(data.frame(\u0026quot;Gene\u0026quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)\n\ncorfit.SP \u0026lt;- duplicateCorrelation(voomFullDGE.SP,design.SP,block=Rep)\nvfit.SP \u0026lt;- lmFit(voomFullDGE.SP, design.SP, block=Rep, correlation=corfit.SP$consensus)\n\n############ Karypop\n\ncont.matrix.SP.Treat \u0026lt;- makeContrasts(\n  \u0026quot;FS-FI\u0026quot;=(FS.18+FS.25)/2-(FI.18+FI.25)/2,\n  \u0026quot;FS-MS\u0026quot;=(FS.18+FS.25)/2-(MS.18+MS.25)/2,\n  \u0026quot;MS-FI\u0026quot;=(MS.18+MS.25)/2-(FI.18+FI.25)/2,\n  levels=design.SP)\n\n# now calculate the contrasts:\nvfit.Treat \u0026lt;- contrasts.fit(vfit.SP, cont.matrix.SP.Treat)\nefit.Treat \u0026lt;- eBayes(vfit.Treat)\n\n# now export the tables with the p-values and log2-fc for each of the contrasts\nfor (name in colnames(cont.matrix.SP.Treat)){\n  # print the name\n  print(name)\n  \n  # make an output file in outputfolder\n  out=paste(O,\u0026quot;/limma/\u0026quot;,name,\u0026quot;.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n  \n  # generate summary table with Benjamini Hochberg adjsuted p-values for all genes\n  TAB=topTable(efit.Treat,coef=name, adjust.method=\u0026quot;BH\u0026quot;, n=Inf)\n  \n  # export table to text file\n  write.table(data.frame(\u0026quot;Gene\u0026quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)\n}\n\n# another cool thing about eBayes is that it calculates an F-value testing if the contrasts are different from 0. In the above case we are comparing all levels for the factor \u0026quot;Treat\u0026quot; only. Ebays combines the t-statistics for all the contrasts into an overall test of significance for that gene. The resulting F-value will thus tell us if \u0026quot;Treat\u0026quot; is significantly differnet from 0. See limma manual page 61. Note that I append another column to the dataframe with the P-value adjusted a la Benjamini-Hochberg\nefit.F.Treat=data.frame(\u0026quot;F\u0026quot;=efit.Treat$F,\u0026quot;Pval\u0026quot;=efit.Treat$F.p.value,\u0026quot;Pval.BH\u0026quot;=p.adjust(efit.Treat$F.p.value, method=\u0026quot;BH\u0026quot;))\nrownames(efit.F.Treat)=rownames(efit.Treat)\n\n# export table as above for the contrasts\n\n# make an output file in outputfolder\nout=paste(O,\u0026quot;/limma/Treat-F-value.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n\n# export table to text file\nwrite.table(data.frame(\u0026quot;Gene\u0026quot;=rownames(efit.F.Treat),efit.F.Treat),file=out,row.names=F,quote = F)\n\n############ Temp\n\ncont.matrix.SP.Temp \u0026lt;- makeContrasts(\n  \u0026quot;T18-25\u0026quot;=(FS.18+FI.18+MS.18)/3-(FS.25+FI.25+MS.25)/3,\n  levels=design.SP)\n\n# now calculate the contrasts:\nvfit.Temp \u0026lt;- contrasts.fit(vfit.SP, cont.matrix.SP.Temp)\nefit.Temp \u0026lt;- eBayes(vfit.Temp)\n\n# now export the tables with the p-values and log2-fc for each of the contrasts\nfor (name in colnames(cont.matrix.SP.Temp)){\n  # print the name\n  print(name)\n  \n  # make an output file in outputfolder\n  out=paste(O,\u0026quot;/limma/\u0026quot;,name,\u0026quot;.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n  \n  # generate summary table with Benjamini Hochberg adjsuted p-values for all genes\n  TAB=topTable(efit.Temp,coef=name, adjust.method=\u0026quot;BH\u0026quot;, n=Inf)\n  \n  # export table to text file\n  write.table(data.frame(\u0026quot;Gene\u0026quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)\n}\n\nefit.F.Temp=data.frame(\u0026quot;F\u0026quot;=efit.Temp$F,\u0026quot;Pval\u0026quot;=efit.Temp$F.p.value,\u0026quot;Pval.BH\u0026quot;=p.adjust(efit.Temp$F.p.value, method=\u0026quot;BH\u0026quot;))\nrownames(efit.F.Temp)=rownames(efit.Temp)\n\n# export table as above for the contrasts\n\n# make an output file in outputfolder\nout=paste(O,\u0026quot;/limma/Temp-F-value.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n\n# export table to text file\nwrite.table(data.frame(\u0026quot;Gene\u0026quot;=rownames(efit.F.Temp),efit.F.Temp),file=out,row.names=F,quote = F)\n\ncont.matrix.SP.interactions \u0026lt;- makeContrasts(\n  \u0026quot;FS-FI_int\u0026quot;=(FS.18-FS.25)-(FI.18-FI.25),\n  \u0026quot;FI-MS_int\u0026quot;=(FI.18-FI.25)-(MS.18-MS.25),\n  \u0026quot;FS-MS_int\u0026quot;=(FS.18-FS.25)-(MS.18-MS.25),\n  levels=design.SP)\n\n# now calculate the contrasts:\nvfit.interactions \u0026lt;- contrasts.fit(vfit.SP, cont.matrix.SP.interactions)\nefit.interactions \u0026lt;- eBayes(vfit.interactions)\n\n# now export the tables with the p-values and log2-fc for each of the contrasts\nfor (name in colnames(cont.matrix.SP.interactions)){\n  # print the name\n  print(name)\n  \n  # make an output file in outputfolder\n  out=paste(O,\u0026quot;/limma/\u0026quot;,name,\u0026quot;.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n  \n  # generate summary table with Benjamini Hochberg adjsuted p-values for all genes\n  TAB=topTable(efit.interactions,coef=name, adjust.method=\u0026quot;BH\u0026quot;, n=Inf)\n  \n  # export table to text file\n  write.table(data.frame(\u0026quot;Gene\u0026quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)\n}\n\n######## Interactions\n\nefit.F.interactions=data.frame(\u0026quot;F\u0026quot;=efit.interactions$F,\u0026quot;Pval\u0026quot;=efit.interactions$F.p.value,\u0026quot;Pval.BH\u0026quot;=p.adjust(efit.interactions$F.p.value, method=\u0026quot;BH\u0026quot;))\nrownames(efit.F.interactions)=rownames(efit.interactions)\n\n# export table as above for the contrasts\n\n# make an output file in outputfolder\nout=paste(O,\u0026quot;/limma/interactions-F-value.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n\n# export table to text file\nwrite.table(data.frame(\u0026quot;Gene\u0026quot;=rownames(efit.F.interactions),efit.F.interactions),file=out,row.names=F,quote = F)\n\n############ Other comparisons\n\ncont.matrix.SP.other \u0026lt;- makeContrasts(\n  \u0026quot;FS-FI_18\u0026quot;=FS.18-FI.18,\n  \u0026quot;FS-MS_18\u0026quot;=FS.18-MS.18,\n  \u0026quot;MS-FI_18\u0026quot;=MS.18-FI.18,\n  \u0026quot;FS-FI_25\u0026quot;=FS.25-FI.25,\n  \u0026quot;FS-MS_25\u0026quot;=FS.25-MS.25,\n  \u0026quot;MS-FI_25\u0026quot;=MS.25-FI.25,\n  levels=design.SP)\n\n# now calculate the contrasts:\nvfit.other \u0026lt;- contrasts.fit(vfit.SP, cont.matrix.SP.other)\nefit.other \u0026lt;- eBayes(vfit.other)\n\n# now export the tables with the p-values and log2-fc for each of the contrasts\nfor (name in colnames(cont.matrix.SP.other)){\n  # print the name\n  print(name)\n  \n  # make an output file in outputfolder\n  out=paste(O,\u0026quot;/limma/\u0026quot;,name,\u0026quot;.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n  \n  # generate summary table with Benjamini Hochberg adjsuted p-values for all genes\n  TAB=topTable(efit.other,coef=name, adjust.method=\u0026quot;BH\u0026quot;, n=Inf)\n  \n  # export table to text file\n  write.table(data.frame(\u0026quot;Gene\u0026quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)\n}\n\n######## make heatmaps\n\ndir.create(paste(O,\u0026quot;/heatmaps\u0026quot;,sep=\u0026quot;\u0026quot;))\n\nlibrary(gplots)\n\nfor (name in colnames(efit.Temp)) {\n  col.pan \u0026lt;- colorpanel(100, \u0026quot;red\u0026quot;, \u0026quot;yellow\u0026quot;, \u0026quot;blue\u0026quot;)\n  To \u0026lt;- topTable(efit.Temp,coef=name, adjust.method=\u0026quot;BH\u0026quot;,n=Inf)\n  logCPM \u0026lt;- cpmFullDGE[rownames(To[1:50,]),]\n  logCPM \u0026lt;- t(scale(t(logCPM)))\n  pdf(paste(O,\u0026quot;/heatmaps/Heatmap-\u0026quot;,name,\u0026quot;.pdf\u0026quot;,sep=\u0026quot;\u0026quot;), width=10, height=10)\n  heatmap.2(logCPM, col=col.pan, Rowv=TRUE, scale=\u0026quot;none\u0026quot;,trace=\u0026quot;none\u0026quot;, dendrogram=\u0026quot;column\u0026quot;, cexRow=1, cexCol=1.4, density.info=\u0026quot;none\u0026quot;,margin=c(10,9), lhei=c(2,10), lwid=c(2,6),main = paste(name,': Heatmap of top 50 genes',sep=\u0026quot;\u0026quot;))\n  dev.off()\n}\n\nfor (name in colnames(efit.Treat)) {\n  col.pan \u0026lt;- colorpanel(100, \u0026quot;red\u0026quot;, \u0026quot;yellow\u0026quot;, \u0026quot;blue\u0026quot;)\n  To \u0026lt;- topTable(efit.Treat,coef=name, adjust.method=\u0026quot;BH\u0026quot;,n=Inf)\n  logCPM \u0026lt;- cpmFullDGE[rownames(To[1:50,]),]\n  logCPM \u0026lt;- t(scale(t(logCPM)))\n  pdf(paste(O,\u0026quot;/heatmaps/Heatmap-\u0026quot;,name,\u0026quot;.pdf\u0026quot;,sep=\u0026quot;\u0026quot;), width=10, height=10)\n  heatmap.2(logCPM, col=col.pan, Rowv=TRUE, scale=\u0026quot;none\u0026quot;,trace=\u0026quot;none\u0026quot;, dendrogram=\u0026quot;column\u0026quot;, cexRow=1, cexCol=1.4, density.info=\u0026quot;none\u0026quot;,margin=c(10,9), lhei=c(2,10), lwid=c(2,6),main = paste(name,': Heatmap of top 50 genes',sep=\u0026quot;\u0026quot;))\n  dev.off()\n}\n\nfor (name in colnames(efit.interactions)) {\n  col.pan \u0026lt;- colorpanel(100, \u0026quot;red\u0026quot;, \u0026quot;yellow\u0026quot;, \u0026quot;blue\u0026quot;)\n  To \u0026lt;- topTable(efit.interactions,coef=name, adjust.method=\u0026quot;BH\u0026quot;,n=Inf)\n  logCPM \u0026lt;- cpmFullDGE[rownames(To[1:50,]),]\n  logCPM \u0026lt;- t(scale(t(logCPM)))\n  pdf(paste(O,\u0026quot;/heatmaps/Heatmap-\u0026quot;,name,\u0026quot;.pdf\u0026quot;,sep=\u0026quot;\u0026quot;), width=10, height=10)\n  heatmap.2(logCPM, col=col.pan, Rowv=TRUE, scale=\u0026quot;none\u0026quot;,trace=\u0026quot;none\u0026quot;, dendrogram=\u0026quot;column\u0026quot;, cexRow=1, cexCol=1.4, density.info=\u0026quot;none\u0026quot;,margin=c(10,9), lhei=c(2,10), lwid=c(2,6),main = paste(name,': Heatmap of top 50 genes',sep=\u0026quot;\u0026quot;))\n  dev.off()\n}\n\nfor (name in colnames(efit.other)) {\n  col.pan \u0026lt;- colorpanel(100, \u0026quot;red\u0026quot;, \u0026quot;yellow\u0026quot;, \u0026quot;blue\u0026quot;)\n  To \u0026lt;- topTable(efit.other,coef=name, adjust.method=\u0026quot;BH\u0026quot;,n=Inf)\n  logCPM \u0026lt;- cpmFullDGE[rownames(To[1:50,]),]\n  logCPM \u0026lt;- t(scale(t(logCPM)))\n  pdf(paste(O,\u0026quot;/heatmaps/Heatmap-\u0026quot;,name,\u0026quot;.pdf\u0026quot;,sep=\u0026quot;\u0026quot;), width=10, height=10)\n  heatmap.2(logCPM, col=col.pan, Rowv=TRUE, scale=\u0026quot;none\u0026quot;,trace=\u0026quot;none\u0026quot;, dendrogram=\u0026quot;column\u0026quot;, cexRow=1, cexCol=1.4, density.info=\u0026quot;none\u0026quot;,margin=c(10,9), lhei=c(2,10), lwid=c(2,6),main = paste(name,': Heatmap of top 50 genes',sep=\u0026quot;\u0026quot;))\n  dev.off()\n}\n\n  \n## GO-term enrichment \nlibrary(goseq)\ndir.create(paste(O,\u0026quot;/GO\u0026quot;,sep=\u0026quot;\u0026quot;))\nfor (name in colnames(efit.Treat$lods)) {\n  print(name)\n  CS=topTable(efit.Treat,coef=name, adjust.method=\u0026quot;BH\u0026quot;, n=Inf)\n  genes=as.integer(CS$adj.P.Val\u0026lt;0.05)\n  names(genes)=row.names(CS)\n  pwf=nullp(genes,\u0026quot;dm6\u0026quot;,\u0026quot;ensGene\u0026quot;)\n  GO.wall=goseq(pwf,\u0026quot;dm6\u0026quot;,\u0026quot;ensGene\u0026quot;)\n  GO.wall=data.frame(GO.wall,\u0026quot;P.adjus_BH\u0026quot;=p.adjust(GO.wall$over_represented_pvalue,method=\u0026quot;BH\u0026quot;))\n  name=gsub(\u0026quot;:\u0026quot;,\u0026quot;-\u0026quot;,name)\n  out=paste(O,\u0026quot;/GO/GO-\u0026quot;,name,\u0026quot;.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n  write.table(GO.wall,file=out,row.names=F,quote = F)\n}\nfor (name in colnames(efit.Treat$lods)) {\ngoana(efit.Treat, coef = name, FDR = 0.05, trend = FALSE,species=\u0026quot;Dm\u0026quot;,convert=F)\n}\n\nrequire(\u0026quot;org.Dm.eg.db\u0026quot;, character.only = TRUE)\nGO2ALLEGS \u0026lt;- \u0026quot;org.Dm.egGO2ALLEGS\u0026quot;\nEG.GO \u0026lt;- AnnotationDbi::toTable(get(GO2ALLEGS))\nd \u0026lt;- duplicated(EG.GO[, c(\u0026quot;gene_id\u0026quot;, \u0026quot;go_id\u0026quot;, \u0026quot;Ontology\u0026quot;)])\nEG.GO \u0026lt;- EG.GO[!d, ]\nde.by.go \u0026lt;- split(EG.GO$gene_id, paste(EG.GO$go_id, EG.GO$Ontology, sep=\u0026quot;.\u0026quot;))\n\ndir.create(paste(O,\u0026quot;/topgo\u0026quot;,sep=\u0026quot;\u0026quot;))\nfor (name in colnames(efit.Treat$lods)) {\n  print(name)\n  CS=topTable(efit.Treat,coef=name, adjust.method=\u0026quot;BH\u0026quot;, n=Inf)\n  genes=as.integer(CS$adj.P.Val\u0026lt;0.05)\n  E=mapIds(org.Dm.eg.db,\n           keys=row.names(CS),\n           column=\u0026quot;ENTREZID\u0026quot;,\n           keytype=\u0026quot;FLYBASE\u0026quot;,\n           multiVals=\u0026quot;first\u0026quot;)\n  names(genes)=E\n  cand=subset(genes,genes==1)\n  GO=goana(names(cand),universe=names(genes),species=\u0026quot;Dm\u0026quot;,trend = \u0026quot;Length\u0026quot;)\n  \n  GO=data.frame(GO,\u0026quot;P.adjus_BH\u0026quot;=p.adjust(GO$P.DE,method=\u0026quot;BH\u0026quot;))\n  \n  DATAB \u0026lt;- lapply(de.by.go, FUN=function(x) { x[x %in% names(cand)] })\n  for (i in 1:nrow(GO)){\n    x=paste(rownames(GO)[i],GO$Ont[i],sep=\u0026quot;.\u0026quot;)\n    if (length(DATAB[[x]])!=0){\n      E=mapIds(org.Dm.eg.db,\n               keys=DATAB[[x]],\n               column=\u0026quot;FLYBASE\u0026quot;,\n               keytype=\u0026quot;ENTREZID\u0026quot;,\n               multiVals=\u0026quot;first\u0026quot;)\n      GO$gene[i]=paste(E,collapse=\u0026quot;,\u0026quot;)\n    }\n    else {\n      GO$gene[i]=\u0026quot;\u0026quot;  \n    }}\n  \n  name=gsub(\u0026quot;:\u0026quot;,\u0026quot;-\u0026quot;,name)\n  out=paste(O,\u0026quot;/topgo/GO-\u0026quot;,name,\u0026quot;.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n  write.table(GO[ order(GO[,6]),],file=out,row.names=T,quote = F,sep=\u0026quot;$\u0026quot;)\n}\n\nfor (name in colnames(efit.Temp$lods)) {\n  print(name)\n  CS=topTable(efit.Temp,coef=name, adjust.method=\u0026quot;BH\u0026quot;, n=Inf)\n  genes=as.integer(CS$adj.P.Val\u0026lt;0.05)\n  E=mapIds(org.Dm.eg.db,\n           keys=row.names(CS),\n           column=\u0026quot;ENTREZID\u0026quot;,\n           keytype=\u0026quot;FLYBASE\u0026quot;,\n           multiVals=\u0026quot;first\u0026quot;)\n  names(genes)=E\n  cand=subset(genes,genes==1)\n  GO=goana(names(cand),universe=names(genes),species=\u0026quot;Dm\u0026quot;,trend = \u0026quot;Length\u0026quot;)\n  \n  GO=data.frame(GO,\u0026quot;P.adjus_BH\u0026quot;=p.adjust(GO$P.DE,method=\u0026quot;BH\u0026quot;))\n  \n  DATAB \u0026lt;- lapply(de.by.go, FUN=function(x) { x[x %in% names(cand)] })\n  for (i in 1:nrow(GO)){\n    x=paste(rownames(GO)[i],GO$Ont[i],sep=\u0026quot;.\u0026quot;)\n    if (length(DATAB[[x]])!=0){\n      E=mapIds(org.Dm.eg.db,\n               keys=DATAB[[x]],\n               column=\u0026quot;FLYBASE\u0026quot;,\n               keytype=\u0026quot;ENTREZID\u0026quot;,\n               multiVals=\u0026quot;first\u0026quot;)\n      GO$gene[i]=paste(E,collapse=\u0026quot;,\u0026quot;)\n    }\n    else {\n      GO$gene[i]=\u0026quot;\u0026quot;  \n    }}\n  \n  name=gsub(\u0026quot;:\u0026quot;,\u0026quot;-\u0026quot;,name)\n  out=paste(O,\u0026quot;/topgo/GO-\u0026quot;,name,\u0026quot;.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n  write.table(GO[ order(GO[,6]),],file=out,row.names=T,quote = F,sep=\u0026quot;$\u0026quot;)\n}\n\n\nfor (name in colnames(efit.interactions$lods)) {\n  print(name)\n  CS=topTable(efit.interactions,coef=name, adjust.method=\u0026quot;BH\u0026quot;, n=Inf)\n  genes=as.integer(CS$adj.P.Val\u0026lt;0.05)\n  E=mapIds(org.Dm.eg.db,\n           keys=row.names(CS),\n           column=\u0026quot;ENTREZID\u0026quot;,\n           keytype=\u0026quot;FLYBASE\u0026quot;,\n           multiVals=\u0026quot;first\u0026quot;)\n  names(genes)=E\n  cand=subset(genes,genes==1)\n  GO=goana(names(cand),universe=names(genes),species=\u0026quot;Dm\u0026quot;,trend = \u0026quot;Length\u0026quot;)\n  \n  GO=data.frame(GO,\u0026quot;P.adjus_BH\u0026quot;=p.adjust(GO$P.DE,method=\u0026quot;BH\u0026quot;))\n  \n  DATAB \u0026lt;- lapply(de.by.go, FUN=function(x) { x[x %in% names(cand)] })\n  for (i in 1:nrow(GO)){\n    x=paste(rownames(GO)[i],GO$Ont[i],sep=\u0026quot;.\u0026quot;)\n    if (length(DATAB[[x]])!=0){\n      E=mapIds(org.Dm.eg.db,\n               keys=DATAB[[x]],\n               column=\u0026quot;FLYBASE\u0026quot;,\n               keytype=\u0026quot;ENTREZID\u0026quot;,\n               multiVals=\u0026quot;first\u0026quot;)\n      GO$gene[i]=paste(E,collapse=\u0026quot;,\u0026quot;)\n    }\n    else {\n      GO$gene[i]=\u0026quot;\u0026quot;  \n    }}\n  \n  name=gsub(\u0026quot;:\u0026quot;,\u0026quot;-\u0026quot;,name)\n  out=paste(O,\u0026quot;/topgo/GO-\u0026quot;,name,\u0026quot;.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n  write.table(GO[ order(GO[,6]),],file=out,row.names=T,quote = F,sep=\u0026quot;$\u0026quot;)\n}\n\n\n\nfor (name in colnames(efit.other$lods)) {\n  print(name)\n  CS=topTable(efit.other,coef=name, adjust.method=\u0026quot;BH\u0026quot;, n=Inf)\n  genes=as.integer(CS$adj.P.Val\u0026lt;0.05)\n  E=mapIds(org.Dm.eg.db,\n           keys=row.names(CS),\n           column=\u0026quot;ENTREZID\u0026quot;,\n           keytype=\u0026quot;FLYBASE\u0026quot;,\n           multiVals=\u0026quot;first\u0026quot;)\n  names(genes)=E\n  cand=subset(genes,genes==1)\n  GO=goana(names(cand),universe=names(genes),species=\u0026quot;Dm\u0026quot;,trend = \u0026quot;Length\u0026quot;)\n  \n  GO=data.frame(GO,\u0026quot;P.adjus_BH\u0026quot;=p.adjust(GO$P.DE,method=\u0026quot;BH\u0026quot;))\n  \n  DATAB \u0026lt;- lapply(de.by.go, FUN=function(x) { x[x %in% names(cand)] })\n  for (i in 1:nrow(GO)){\n    x=paste(rownames(GO)[i],GO$Ont[i],sep=\u0026quot;.\u0026quot;)\n    if (length(DATAB[[x]])!=0){\n      E=mapIds(org.Dm.eg.db,\n               keys=DATAB[[x]],\n               column=\u0026quot;FLYBASE\u0026quot;,\n               keytype=\u0026quot;ENTREZID\u0026quot;,\n               multiVals=\u0026quot;first\u0026quot;)\n      GO$gene[i]=paste(E,collapse=\u0026quot;,\u0026quot;)\n    }\n    else {\n      GO$gene[i]=\u0026quot;\u0026quot;  \n    }}\n  \n  name=gsub(\u0026quot;:\u0026quot;,\u0026quot;-\u0026quot;,name)\n  out=paste(O,\u0026quot;/topgo/GO-\u0026quot;,name,\u0026quot;.txt\u0026quot;,sep=\u0026quot;\u0026quot;)\n  write.table(GO[ order(GO[,6]),],file=out,row.names=T,quote = F,sep=\u0026quot;$\u0026quot;)\n}\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## Analysis of RNA-Seq data of inversion lines\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# install packages\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003einstall.packages(\"readr\")\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003einstall.packages(\"tximport\")\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003einstall.packages(\"edgeR\")\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003einstall.packages(\"RColorBrewer\")\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003einstall.packages(\"limma\")\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003einstall.packages(\"pheatmap\")\u003c/span\u003e\n\n\u003cspan class=\"pl-v\"\u003eI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/RNASeq/count/analyses_counts.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/RNASeq/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eFactors\u003c/span\u003e\n\n\u003cspan class=\"pl-v\"\u003eTreat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFI\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eMS\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eTreat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erelevel(\u003cspan class=\"pl-smi\"\u003eTreat\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eref\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eTemp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e25\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eRep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(c(\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e14\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e17\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e18\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e22\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e23\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e14\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e17\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e18\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e22\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e23\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e13\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e21\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e27\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e7\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e7\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e13\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e21\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e27\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e11\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e19\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e24\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e25\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e11\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e19\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e24\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e25\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# 1a) read Data\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003erawData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eread.table(\u003cspan class=\"pl-smi\"\u003eI\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003erawCounts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003erawData\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e56\u003c/span\u003e]\nrownames(\u003cspan class=\"pl-smi\"\u003erawCounts\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003erawData\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# 2) create DGE object using edgeR\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003eedgeR\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eDGEList(\u003cspan class=\"pl-smi\"\u003erawCounts\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003egroup\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003enames(\u003cspan class=\"pl-smi\"\u003erawCounts\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# 3) add factors to object\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etreat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTreat\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etemp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTemp\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Transform counts (Counts per million: CPM) and normalize the counts using the TMM method in edgeR\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e calcNormFactors(\u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTMM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ecpmFullDGE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003ecpm(\u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elog\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/countdata/CPM.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eTAB\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecpmFullDGE\u003c/span\u003e\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# remove genes covered by less than 2 read per million in at least nine samples\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ekeep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rowSums(\u003cspan class=\"pl-smi\"\u003ecpmFullDGE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ekeep\u003c/span\u003e,] \n\u003cspan class=\"pl-smi\"\u003ecpmTFullDGE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecpmFullDGE\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ekeep\u003c/span\u003e,] \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Plot density histograms to quantify the effect of filtering out lowly expressed genes\u003c/span\u003e\ndir.create(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/quality\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\npdf(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/quality/CPM-hist.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ensamples\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ncol(\u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ecol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e topo.colors(\u003cspan class=\"pl-smi\"\u003ensamples\u003c/span\u003e)\npar(\u003cspan class=\"pl-v\"\u003emfrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e))\nplot(density(\u003cspan class=\"pl-smi\"\u003ecpmFullDGE\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]), \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecol\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e], \u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.25\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elas\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\n     \u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-v\"\u003ev\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ensamples\u003c/span\u003e){\n  \u003cspan class=\"pl-smi\"\u003eden\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e density(\u003cspan class=\"pl-smi\"\u003ecpmFullDGE\u003c/span\u003e[,\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e])\n  lines(\u003cspan class=\"pl-smi\"\u003eden\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eden\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ey\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecol\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e], \u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n} \nlegend(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopright\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elegend\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ecolnames(\u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003etext.col\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecol\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ebty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003en\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003encol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e)\ntitle(\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eA. Raw data\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLog-cpm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nplot(density(\u003cspan class=\"pl-smi\"\u003ecpmTFullDGE\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]), \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecol\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e], \u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.25\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elas\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\n     \u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ntitle(\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eB. Filtered data\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLog-cpm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \nabline(\u003cspan class=\"pl-v\"\u003ev\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ensamples\u003c/span\u003e){\n  \u003cspan class=\"pl-smi\"\u003eden\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e density(\u003cspan class=\"pl-smi\"\u003ecpmTFullDGE\u003c/span\u003e[,\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e])\n  lines(\u003cspan class=\"pl-smi\"\u003eden\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eden\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ey\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecol\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e], \u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n} \nlegend(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopright\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elegend\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ecolnames(\u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003etext.col\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecol\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ebty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003en\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003encol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e)\ndev.off()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# explore Data with MDS to identify clustering\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003eRColorBrewer\u003c/span\u003e)\npdf(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/quality/MDS.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e14\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e)\npar(\u003cspan class=\"pl-v\"\u003emfrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003ecol.Treat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTreat\u003c/span\u003e\nlevels(\u003cspan class=\"pl-smi\"\u003ecol.Treat\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e brewer.pal(nlevels(\u003cspan class=\"pl-smi\"\u003ecol.Treat\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSet1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplotMDS(\u003cspan class=\"pl-smi\"\u003ecpmTFullDGE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eas.character(\u003cspan class=\"pl-smi\"\u003ecol.Treat\u003c/span\u003e))\ntitle(\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKarypop: Dim 1/2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003ecol.Temp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTemp\u003c/span\u003e\nlevels(\u003cspan class=\"pl-smi\"\u003ecol.Temp\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e brewer.pal(nlevels(\u003cspan class=\"pl-smi\"\u003ecol.Temp\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSet1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplotMDS(\u003cspan class=\"pl-smi\"\u003ecpmTFullDGE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eas.character(\u003cspan class=\"pl-smi\"\u003ecol.Temp\u003c/span\u003e))\ntitle(\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperature: Dim 1/2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nplotMDS(\u003cspan class=\"pl-smi\"\u003ecpmTFullDGE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eas.character(\u003cspan class=\"pl-smi\"\u003ecol.Treat\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003edim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e))\ntitle(\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKarypop: Dim 3/4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nplotMDS(\u003cspan class=\"pl-smi\"\u003ecpmTFullDGE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eas.character(\u003cspan class=\"pl-smi\"\u003ecol.Temp\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003edim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e))\ntitle(\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperature: Dim 3/4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\ndev.off()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## explore data with PCA to explore clustering\u003c/span\u003e\n\npdf(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/quality/PCA.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\npar(\u003cspan class=\"pl-v\"\u003emfrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003enewcpm\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003et(\u003cspan class=\"pl-smi\"\u003ecpmTFullDGE\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eprcomp(\u003cspan class=\"pl-smi\"\u003enewcpm\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003ecolKP\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erainbow(length(levels(\u003cspan class=\"pl-smi\"\u003eTreat\u003c/span\u003e)))[\u003cspan class=\"pl-smi\"\u003eTreat\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003ecolTemp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erainbow(length(levels(\u003cspan class=\"pl-smi\"\u003eTemp\u003c/span\u003e)))[as.factor(\u003cspan class=\"pl-smi\"\u003eTemp\u003c/span\u003e)]\n\u003cspan class=\"pl-v\"\u003ecolPop\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erainbow(length(levels(\u003cspan class=\"pl-smi\"\u003eRep\u003c/span\u003e)))[as.factor(\u003cspan class=\"pl-smi\"\u003eRep\u003c/span\u003e)]\n\nplot(\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecolKP\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKarypop\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etext(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colKP)\u003c/span\u003e\n\nplot(\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecolTemp\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etext(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colTemp)\u003c/span\u003e\n\nplot(\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecolPop\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eReplicate\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etext(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colPop)\u003c/span\u003e\n\nplot(\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecolKP\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKarypop\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etext(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colKP)\u003c/span\u003e\n\nplot(\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecolTemp\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etext(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colTemp)\u003c/span\u003e\n\nplot(\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003ePCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecolPop\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eReplicate\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etext(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colPop)\u003c/span\u003e\n\ndev.off()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## Limma analysis with singe grouping factor\u003c/span\u003e\n\nlibrary(\u003cspan class=\"pl-smi\"\u003elimma\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003eTreat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erelevel(\u003cspan class=\"pl-smi\"\u003eTreat\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eref\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndir.create(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/limma\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-v\"\u003eG\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003einteraction(\u003cspan class=\"pl-smi\"\u003eTreat\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eTemp\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003edesign.SP\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003emodel.matrix(\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eG\u003c/span\u003e)\ncolnames(\u003cspan class=\"pl-smi\"\u003edesign.SP\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elevels(\u003cspan class=\"pl-smi\"\u003eG\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Do VOOM\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003evoomFullDGE.SP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e voom(\u003cspan class=\"pl-smi\"\u003eFullDGE\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edesign.SP\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eplot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e write transformed data to file\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/countdata/voomCounts.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eTAB\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003evoomFullDGE.SP\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eE\u003c/span\u003e\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003ecorfit.SP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e duplicateCorrelation(\u003cspan class=\"pl-smi\"\u003evoomFullDGE.SP\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003edesign.SP\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eblock\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRep\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003evfit.SP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lmFit(\u003cspan class=\"pl-smi\"\u003evoomFullDGE.SP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edesign.SP\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eblock\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRep\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecorrelation\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecorfit.SP\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003econsensus\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e########### Karypop\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003econt.matrix.SP.Treat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e makeContrasts(\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS-FI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eFS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.25\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eFI.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFI.25\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS-MS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eFS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.25\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eMS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.25\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMS-FI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eMS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.25\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eFI.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFI.25\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\n  \u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edesign.SP\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now calculate the contrasts:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003evfit.Treat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e contrasts.fit(\u003cspan class=\"pl-smi\"\u003evfit.SP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003econt.matrix.SP.Treat\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e eBayes(\u003cspan class=\"pl-smi\"\u003evfit.Treat\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now export the tables with the p-values and log2-fc for each of the contrasts\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003econt.matrix.SP.Treat\u003c/span\u003e)){\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e print the name\u003c/span\u003e\n  print(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e make an output file in outputfolder\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/limma/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e generate summary table with Benjamini Hochberg adjsuted p-values for all genes\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003eTAB\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003etopTable(\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e export table to text file\u003c/span\u003e\n  write.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n}\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e another cool thing about eBayes is that it calculates an F-value testing if the contrasts are different from 0. In the above case we are comparing all levels for the factor \"Treat\" only. Ebays combines the t-statistics for all the contrasts into an overall test of significance for that gene. The resulting F-value will thus tell us if \"Treat\" is significantly differnet from 0. See limma manual page 61. Note that I append another column to the dataframe with the P-value adjusted a la Benjamini-Hochberg\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eefit.F.Treat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePval\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.p.value\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePval.BH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ep.adjust(\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.p.value\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nrownames(\u003cspan class=\"pl-smi\"\u003eefit.F.Treat\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e export table as above for the contrasts\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e make an output file in outputfolder\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/limma/Treat-F-value.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e export table to text file\u003c/span\u003e\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eefit.F.Treat\u003c/span\u003e),\u003cspan class=\"pl-smi\"\u003eefit.F.Treat\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e########### Temp\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003econt.matrix.SP.Temp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e makeContrasts(\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eT18-25\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eFS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFI.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.18\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eFS.25\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFI.25\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.25\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\n  \u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edesign.SP\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now calculate the contrasts:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003evfit.Temp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e contrasts.fit(\u003cspan class=\"pl-smi\"\u003evfit.SP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003econt.matrix.SP.Temp\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eefit.Temp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e eBayes(\u003cspan class=\"pl-smi\"\u003evfit.Temp\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now export the tables with the p-values and log2-fc for each of the contrasts\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003econt.matrix.SP.Temp\u003c/span\u003e)){\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e print the name\u003c/span\u003e\n  print(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e make an output file in outputfolder\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/limma/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e generate summary table with Benjamini Hochberg adjsuted p-values for all genes\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003eTAB\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003etopTable(\u003cspan class=\"pl-smi\"\u003eefit.Temp\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e export table to text file\u003c/span\u003e\n  write.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n}\n\n\u003cspan class=\"pl-v\"\u003eefit.F.Temp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eefit.Temp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePval\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eefit.Temp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.p.value\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePval.BH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ep.adjust(\u003cspan class=\"pl-smi\"\u003eefit.Temp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.p.value\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nrownames(\u003cspan class=\"pl-smi\"\u003eefit.F.Temp\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eefit.Temp\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e export table as above for the contrasts\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e make an output file in outputfolder\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/limma/Temp-F-value.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e export table to text file\u003c/span\u003e\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eefit.F.Temp\u003c/span\u003e),\u003cspan class=\"pl-smi\"\u003eefit.F.Temp\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003econt.matrix.SP.interactions\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e makeContrasts(\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS-FI_int\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eFS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.25\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eFI.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFI.25\u003c/span\u003e),\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFI-MS_int\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eFI.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFI.25\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eMS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.25\u003c/span\u003e),\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS-MS_int\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eFS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.25\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eMS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.25\u003c/span\u003e),\n  \u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edesign.SP\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now calculate the contrasts:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003evfit.interactions\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e contrasts.fit(\u003cspan class=\"pl-smi\"\u003evfit.SP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003econt.matrix.SP.interactions\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eefit.interactions\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e eBayes(\u003cspan class=\"pl-smi\"\u003evfit.interactions\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now export the tables with the p-values and log2-fc for each of the contrasts\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003econt.matrix.SP.interactions\u003c/span\u003e)){\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e print the name\u003c/span\u003e\n  print(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e make an output file in outputfolder\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/limma/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e generate summary table with Benjamini Hochberg adjsuted p-values for all genes\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003eTAB\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003etopTable(\u003cspan class=\"pl-smi\"\u003eefit.interactions\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e export table to text file\u003c/span\u003e\n  write.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n}\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e####### Interactions\u003c/span\u003e\n\n\u003cspan class=\"pl-v\"\u003eefit.F.interactions\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eefit.interactions\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePval\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eefit.interactions\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.p.value\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePval.BH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ep.adjust(\u003cspan class=\"pl-smi\"\u003eefit.interactions\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.p.value\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nrownames(\u003cspan class=\"pl-smi\"\u003eefit.F.interactions\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eefit.interactions\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e export table as above for the contrasts\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e make an output file in outputfolder\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/limma/interactions-F-value.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e export table to text file\u003c/span\u003e\nwrite.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eefit.F.interactions\u003c/span\u003e),\u003cspan class=\"pl-smi\"\u003eefit.F.interactions\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e########### Other comparisons\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003econt.matrix.SP.other\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e makeContrasts(\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS-FI_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFI.18\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS-MS_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.18\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMS-FI_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.18\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFI.18\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS-FI_25\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.25\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFI.25\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS-MS_25\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.25\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.25\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMS-FI_25\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.25\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFI.25\u003c/span\u003e,\n  \u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edesign.SP\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now calculate the contrasts:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003evfit.other\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e contrasts.fit(\u003cspan class=\"pl-smi\"\u003evfit.SP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003econt.matrix.SP.other\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eefit.other\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e eBayes(\u003cspan class=\"pl-smi\"\u003evfit.other\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now export the tables with the p-values and log2-fc for each of the contrasts\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003econt.matrix.SP.other\u003c/span\u003e)){\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e print the name\u003c/span\u003e\n  print(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e make an output file in outputfolder\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/limma/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e generate summary table with Benjamini Hochberg adjsuted p-values for all genes\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003eTAB\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003etopTable(\u003cspan class=\"pl-smi\"\u003eefit.other\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e export table to text file\u003c/span\u003e\n  write.table(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-smi\"\u003eTAB\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n}\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e####### make heatmaps\u003c/span\u003e\n\ndir.create(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/heatmaps\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\nlibrary(\u003cspan class=\"pl-smi\"\u003egplots\u003c/span\u003e)\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eefit.Temp\u003c/span\u003e)) {\n  \u003cspan class=\"pl-smi\"\u003ecol.pan\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colorpanel(\u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eyellow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \u003cspan class=\"pl-smi\"\u003eTo\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e topTable(\u003cspan class=\"pl-smi\"\u003eefit.Temp\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecpmFullDGE\u003c/span\u003e[rownames(\u003cspan class=\"pl-smi\"\u003eTo\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e,]),]\n  \u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e t(scale(t(\u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e)))\n  pdf(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/heatmaps/Heatmap-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e)\n  heatmap.2(\u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecol.pan\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eRowv\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etrace\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edendrogram\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecolumn\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexRow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexCol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1.4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edensity.info\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emargin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elhei\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elwid\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e: Heatmap of top 50 genes\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n  dev.off()\n}\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e)) {\n  \u003cspan class=\"pl-smi\"\u003ecol.pan\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colorpanel(\u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eyellow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \u003cspan class=\"pl-smi\"\u003eTo\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e topTable(\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecpmFullDGE\u003c/span\u003e[rownames(\u003cspan class=\"pl-smi\"\u003eTo\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e,]),]\n  \u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e t(scale(t(\u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e)))\n  pdf(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/heatmaps/Heatmap-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e)\n  heatmap.2(\u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecol.pan\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eRowv\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etrace\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edendrogram\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecolumn\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexRow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexCol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1.4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edensity.info\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emargin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elhei\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elwid\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e: Heatmap of top 50 genes\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n  dev.off()\n}\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eefit.interactions\u003c/span\u003e)) {\n  \u003cspan class=\"pl-smi\"\u003ecol.pan\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colorpanel(\u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eyellow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \u003cspan class=\"pl-smi\"\u003eTo\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e topTable(\u003cspan class=\"pl-smi\"\u003eefit.interactions\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecpmFullDGE\u003c/span\u003e[rownames(\u003cspan class=\"pl-smi\"\u003eTo\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e,]),]\n  \u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e t(scale(t(\u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e)))\n  pdf(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/heatmaps/Heatmap-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e)\n  heatmap.2(\u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecol.pan\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eRowv\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etrace\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edendrogram\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecolumn\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexRow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexCol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1.4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edensity.info\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emargin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elhei\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elwid\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e: Heatmap of top 50 genes\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n  dev.off()\n}\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eefit.other\u003c/span\u003e)) {\n  \u003cspan class=\"pl-smi\"\u003ecol.pan\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colorpanel(\u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eyellow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \u003cspan class=\"pl-smi\"\u003eTo\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e topTable(\u003cspan class=\"pl-smi\"\u003eefit.other\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecpmFullDGE\u003c/span\u003e[rownames(\u003cspan class=\"pl-smi\"\u003eTo\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e,]),]\n  \u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e t(scale(t(\u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e)))\n  pdf(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/heatmaps/Heatmap-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e)\n  heatmap.2(\u003cspan class=\"pl-smi\"\u003elogCPM\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecol.pan\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eRowv\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etrace\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edendrogram\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecolumn\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexRow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexCol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1.4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edensity.info\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emargin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elhei\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elwid\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e: Heatmap of top 50 genes\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n  dev.off()\n}\n\n  \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# GO-term enrichment \u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003egoseq\u003c/span\u003e)\ndir.create(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/GO\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003elods\u003c/span\u003e)) {\n  print(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eCS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003etopTable(\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003egenes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eas.integer(\u003cspan class=\"pl-smi\"\u003eCS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eadj.P.Val\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e)\n  names(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erow.names(\u003cspan class=\"pl-smi\"\u003eCS\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003epwf\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003enullp(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edm6\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eensGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eGO.wall\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003egoseq(\u003cspan class=\"pl-smi\"\u003epwf\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edm6\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eensGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eGO.wall\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eGO.wall\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eP.adjus_BH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ep.adjust(\u003cspan class=\"pl-smi\"\u003eGO.wall\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eover_represented_pvalue\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n  \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003egsub(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/GO/GO-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  write.table(\u003cspan class=\"pl-smi\"\u003eGO.wall\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n}\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003elods\u003c/span\u003e)) {\ngoana(\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eFDR\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etrend\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003especies\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003econvert\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n}\n\nrequire(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorg.Dm.eg.db\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003echaracter.only\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eGO2ALLEGS\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorg.Dm.egGO2ALLEGS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eEG.GO\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003eAnnotationDbi\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003etoTable(get(\u003cspan class=\"pl-smi\"\u003eGO2ALLEGS\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e duplicated(\u003cspan class=\"pl-smi\"\u003eEG.GO\u003c/span\u003e[, c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene_id\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ego_id\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOntology\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)])\n\u003cspan class=\"pl-smi\"\u003eEG.GO\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eEG.GO\u003c/span\u003e[\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e, ]\n\u003cspan class=\"pl-smi\"\u003ede.by.go\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e split(\u003cspan class=\"pl-smi\"\u003eEG.GO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene_id\u003c/span\u003e, paste(\u003cspan class=\"pl-smi\"\u003eEG.GO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ego_id\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eEG.GO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOntology\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\ndir.create(paste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/topgo\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003elods\u003c/span\u003e)) {\n  print(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eCS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003etopTable(\u003cspan class=\"pl-smi\"\u003eefit.Treat\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003egenes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eas.integer(\u003cspan class=\"pl-smi\"\u003eCS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eadj.P.Val\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003emapIds(\u003cspan class=\"pl-smi\"\u003eorg.Dm.eg.db\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003ekeys\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erow.names(\u003cspan class=\"pl-smi\"\u003eCS\u003c/span\u003e),\n           \u003cspan class=\"pl-v\"\u003ecolumn\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eENTREZID\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003ekeytype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFLYBASE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003emultiVals\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efirst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  names(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eE\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003ecand\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003egoana(names(\u003cspan class=\"pl-smi\"\u003ecand\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003euniverse\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003enames(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003especies\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etrend\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLength\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-v\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eP.adjus_BH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ep.adjust(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eP.DE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n  \n  \u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lapply(\u003cspan class=\"pl-smi\"\u003ede.by.go\u003c/span\u003e, \u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e) { \u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e names(\u003cspan class=\"pl-smi\"\u003ecand\u003c/span\u003e)] })\n  \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003enrow(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e)){\n    \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(rownames(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e)[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOnt\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e (length(\u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e]])\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e){\n      \u003cspan class=\"pl-v\"\u003eE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003emapIds(\u003cspan class=\"pl-smi\"\u003eorg.Dm.eg.db\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003ekeys\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e]],\n               \u003cspan class=\"pl-v\"\u003ecolumn\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFLYBASE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003ekeytype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eENTREZID\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003emultiVals\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efirst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n      \u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecollapse\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    }\n    \u003cspan class=\"pl-k\"\u003eelse\u003c/span\u003e {\n      \u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e  \n    }}\n  \n  \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003egsub(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/topgo/GO-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  write.table(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e[ order(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e]),],\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e$\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n}\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eefit.Temp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003elods\u003c/span\u003e)) {\n  print(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eCS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003etopTable(\u003cspan class=\"pl-smi\"\u003eefit.Temp\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003egenes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eas.integer(\u003cspan class=\"pl-smi\"\u003eCS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eadj.P.Val\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003emapIds(\u003cspan class=\"pl-smi\"\u003eorg.Dm.eg.db\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003ekeys\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erow.names(\u003cspan class=\"pl-smi\"\u003eCS\u003c/span\u003e),\n           \u003cspan class=\"pl-v\"\u003ecolumn\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eENTREZID\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003ekeytype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFLYBASE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003emultiVals\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efirst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  names(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eE\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003ecand\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003egoana(names(\u003cspan class=\"pl-smi\"\u003ecand\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003euniverse\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003enames(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003especies\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etrend\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLength\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-v\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eP.adjus_BH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ep.adjust(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eP.DE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n  \n  \u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lapply(\u003cspan class=\"pl-smi\"\u003ede.by.go\u003c/span\u003e, \u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e) { \u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e names(\u003cspan class=\"pl-smi\"\u003ecand\u003c/span\u003e)] })\n  \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003enrow(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e)){\n    \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(rownames(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e)[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOnt\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e (length(\u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e]])\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e){\n      \u003cspan class=\"pl-v\"\u003eE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003emapIds(\u003cspan class=\"pl-smi\"\u003eorg.Dm.eg.db\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003ekeys\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e]],\n               \u003cspan class=\"pl-v\"\u003ecolumn\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFLYBASE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003ekeytype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eENTREZID\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003emultiVals\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efirst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n      \u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecollapse\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    }\n    \u003cspan class=\"pl-k\"\u003eelse\u003c/span\u003e {\n      \u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e  \n    }}\n  \n  \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003egsub(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/topgo/GO-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  write.table(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e[ order(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e]),],\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e$\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n}\n\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eefit.interactions\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003elods\u003c/span\u003e)) {\n  print(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eCS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003etopTable(\u003cspan class=\"pl-smi\"\u003eefit.interactions\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003egenes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eas.integer(\u003cspan class=\"pl-smi\"\u003eCS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eadj.P.Val\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003emapIds(\u003cspan class=\"pl-smi\"\u003eorg.Dm.eg.db\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003ekeys\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erow.names(\u003cspan class=\"pl-smi\"\u003eCS\u003c/span\u003e),\n           \u003cspan class=\"pl-v\"\u003ecolumn\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eENTREZID\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003ekeytype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFLYBASE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003emultiVals\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efirst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  names(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eE\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003ecand\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003egoana(names(\u003cspan class=\"pl-smi\"\u003ecand\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003euniverse\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003enames(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003especies\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etrend\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLength\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-v\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eP.adjus_BH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ep.adjust(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eP.DE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n  \n  \u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lapply(\u003cspan class=\"pl-smi\"\u003ede.by.go\u003c/span\u003e, \u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e) { \u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e names(\u003cspan class=\"pl-smi\"\u003ecand\u003c/span\u003e)] })\n  \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003enrow(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e)){\n    \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(rownames(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e)[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOnt\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e (length(\u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e]])\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e){\n      \u003cspan class=\"pl-v\"\u003eE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003emapIds(\u003cspan class=\"pl-smi\"\u003eorg.Dm.eg.db\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003ekeys\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e]],\n               \u003cspan class=\"pl-v\"\u003ecolumn\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFLYBASE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003ekeytype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eENTREZID\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003emultiVals\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efirst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n      \u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecollapse\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    }\n    \u003cspan class=\"pl-k\"\u003eelse\u003c/span\u003e {\n      \u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e  \n    }}\n  \n  \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003egsub(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/topgo/GO-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  write.table(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e[ order(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e]),],\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e$\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n}\n\n\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eefit.other\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003elods\u003c/span\u003e)) {\n  print(\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eCS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003etopTable(\u003cspan class=\"pl-smi\"\u003eefit.other\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadjust.method\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003egenes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eas.integer(\u003cspan class=\"pl-smi\"\u003eCS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eadj.P.Val\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003emapIds(\u003cspan class=\"pl-smi\"\u003eorg.Dm.eg.db\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003ekeys\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003erow.names(\u003cspan class=\"pl-smi\"\u003eCS\u003c/span\u003e),\n           \u003cspan class=\"pl-v\"\u003ecolumn\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eENTREZID\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003ekeytype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFLYBASE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n           \u003cspan class=\"pl-v\"\u003emultiVals\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efirst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  names(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eE\u003c/span\u003e\n  \u003cspan class=\"pl-v\"\u003ecand\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003egoana(names(\u003cspan class=\"pl-smi\"\u003ecand\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003euniverse\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003enames(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003especies\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etrend\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLength\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \n  \u003cspan class=\"pl-v\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eP.adjus_BH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ep.adjust(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eP.DE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n  \n  \u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lapply(\u003cspan class=\"pl-smi\"\u003ede.by.go\u003c/span\u003e, \u003cspan class=\"pl-en\"\u003eFUN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e) { \u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e names(\u003cspan class=\"pl-smi\"\u003ecand\u003c/span\u003e)] })\n  \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003enrow(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e)){\n    \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(rownames(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e)[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e],\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOnt\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e],\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e (length(\u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e]])\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e){\n      \u003cspan class=\"pl-v\"\u003eE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003emapIds(\u003cspan class=\"pl-smi\"\u003eorg.Dm.eg.db\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003ekeys\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDATAB\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e]],\n               \u003cspan class=\"pl-v\"\u003ecolumn\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFLYBASE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003ekeytype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eENTREZID\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n               \u003cspan class=\"pl-v\"\u003emultiVals\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efirst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n      \u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecollapse\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    }\n    \u003cspan class=\"pl-k\"\u003eelse\u003c/span\u003e {\n      \u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e  \n    }}\n  \n  \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003egsub(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e)\n  \u003cspan class=\"pl-v\"\u003eout\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste(\u003cspan class=\"pl-smi\"\u003eO\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/topgo/GO-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ename\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  write.table(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e[ order(\u003cspan class=\"pl-smi\"\u003eGO\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e]),],\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eout\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e$\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n}\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThen, we added additional information, such as gene name, length and genomic coordinates to the DE datasets and isolated candidate genes based on p-value\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\ncd /data/RNASeq/limma\nmkdir -p /data/RNASeq/limma/FullInfo/candidates\n\nfor i in *.txt\ndo\n    ## add info about genes\n    python3 /scripts/v3/AddGeneInfo.py \\\n        /data/RNASeq/limma/${i} \\\n        /data/RNASeq/dmel-all-r6.17.gtf.gz \\\n        \u0026gt; /data/RNASeq/limma/FullInfo/${i}\n\n    ## isolate candidate genes with p\u0026lt;0.05\n    printf 'Gene\\tFBgn\\tlogFC\\tPval\\n' \u0026gt; /data/RNASeq/limma/FullInfo/candidates/${i}\n    awk '$12\u0026lt;0.05 {print $1\u0026quot;\\t\u0026quot;$2\u0026quot;\\t\u0026quot;$8\u0026quot;\\t\u0026quot;$12}' ${i} \u0026gt;\u0026gt; /data/RNASeq/limma/FullInfo/candidates/${i}\n\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /data/RNASeq/limma\nmkdir -p /data/RNASeq/limma/FullInfo/candidates\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.txt\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# add info about genes\u003c/span\u003e\n    python3 /scripts/v3/AddGeneInfo.py \\\n        /data/RNASeq/limma/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \\\n        /data/RNASeq/dmel-all-r6.17.gtf.gz \\\n        \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/RNASeq/limma/FullInfo/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\n\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# isolate candidate genes with p\u0026lt;0.05\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eGene\\tFBgn\\tlogFC\\tPval\\n\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/RNASeq/limma/FullInfo/candidates/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\n    awk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e$12\u0026lt;0.05 {print $1\"\\t\"$2\"\\t\"$8\"\\t\"$12}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e /data/RNASeq/limma/FullInfo/candidates/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.2) Test for positional overrepresentation\u003c/h4\u003e\u003ca id=\"user-content-42-test-for-positional-overrepresentation\" class=\"anchor\" aria-label=\"Permalink: 4.2) Test for positional overrepresentation\" href=\"#42-test-for-positional-overrepresentation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## test for overrepresentation of candidates with Fisher's Exact tests\nmkdir /data/RNASeq/overrepresentation\n\nprintf \u0026quot;Dataset\\tCand-Inv\\tCand-NonInv\\tNonCand-Inv\\tNonCand-NonInv\\tFET-pval\\tCand-Inv_Breakpoint\\tCand-Inv_Body\\tNonCand-Inv_Breakpoint\\tNonCand-Inv_Body\\tFET-pval-InsideInv\\n\u0026quot; \\\n\u0026gt; /data/RNASeq/overrepresentation/FET.txt\n\nfor i in /data/RNASeq/limma/FullInfo/*\ndo\n\n    python3 /scripts/FET.py --input ${i} \\\n    \u0026gt;\u0026gt; /data/RNASeq/overrepresentation/FET.txt\n\ndone\n\n## make manhattan plots\nmkdir -p /data/RNASeq/overrepresentation/manhattan/logFC\nmkdir -p /data/RNASeq/overrepresentation/manhattan/P\n\nfor i in /data/RNASeq/limma/FullInfo/*.txt\ndo\n\n    ## Ignore datasets based on F-statistics generated above.\n    if [[ ${i} != *\u0026quot;-F-\u0026quot;* ]]\n    then\n\n        ## make manhattan plots based on the -log10 P-values\n        python3 /scripts/plot-manhattan.py \\\n            --full ${i} \\\n            --CF -2 \\\n            --log \\\n            --out /data/RNASeq/overrepresentation/manhattan/P\n\n        ## make manhattan plots based on the log2-flod difference\n        python3 /TransInv/scripts/plot-manhattan.py \\\n            --full ${i} \\\n            --CF 7 \\\n            --out /data/RNASeq/overrepresentation/manhattan/logFC\n\n    fi\n\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# test for overrepresentation of candidates with Fisher's Exact tests\u003c/span\u003e\nmkdir /data/RNASeq/overrepresentation\n\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDataset\\tCand-Inv\\tCand-NonInv\\tNonCand-Inv\\tNonCand-NonInv\\tFET-pval\\tCand-Inv_Breakpoint\\tCand-Inv_Body\\tNonCand-Inv_Breakpoint\\tNonCand-Inv_Body\\tFET-pval-InsideInv\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/RNASeq/overrepresentation/FET.txt\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e /data/RNASeq/limma/FullInfo/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\n    python3 /scripts/FET.py --input \u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \\\n    \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e /data/RNASeq/overrepresentation/FET.txt\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# make manhattan plots\u003c/span\u003e\nmkdir -p /data/RNASeq/overrepresentation/manhattan/logFC\nmkdir -p /data/RNASeq/overrepresentation/manhattan/P\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e /data/RNASeq/limma/FullInfo/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.txt\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Ignore datasets based on F-statistics generated above.\u003c/span\u003e\n    \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e [[ \u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e-F-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e ]]\n    \u003cspan class=\"pl-k\"\u003ethen\u003c/span\u003e\n\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# make manhattan plots based on the -log10 P-values\u003c/span\u003e\n        python3 /scripts/plot-manhattan.py \\\n            --full \u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \\\n            --CF -2 \\\n            --log \\\n            --out /data/RNASeq/overrepresentation/manhattan/P\n\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# make manhattan plots based on the log2-flod difference\u003c/span\u003e\n        python3 /TransInv/scripts/plot-manhattan.py \\\n            --full \u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \\\n            --CF 7 \\\n            --out /data/RNASeq/overrepresentation/manhattan/logFC\n\n    \u003cspan class=\"pl-k\"\u003efi\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.3) Compare by temperarture\u003c/h4\u003e\u003ca id=\"user-content-43-compare-by-temperarture\" class=\"anchor\" aria-label=\"Permalink: 4.3) Compare by temperarture\" href=\"#43-compare-by-temperarture\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNext, we tested how the p-value distribution differed by temperature. We therefore manually compiled a table with all p-values from the different analyses (see data/)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\ndir.create('/data/RNASeq/CompareByTemp')\n\n## load pvalue data from limma\nData=read.table(\u0026quot;/data/CompareByTemp_Pval.txt\u0026quot;,header=T)\n\n## Compare p-value distributions of 18 and 25 temp \n\npdf(\u0026quot;/data/RNASeq/CompareByTemp/Pval-genewise-byTemp.pdf\u0026quot;,width=12,height=8)\npar(mfrow=(c(3,3)))\n\nplot(Data$FS.FI.25,Data$FS.FI.18,col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;FS vs. FI - Temp\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;18C\u0026quot;)\nplot(Data$FS.MS.25,Data$FS.MS.18,col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;FS vs. MS - Temp\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;18C\u0026quot;)\nplot(Data$MS.FI.25,Data$MS.FI.18,col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;MS vs. FI - Temp\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;18C\u0026quot;)\nplot(Data$FS.FI.25,Data$FS.FI,col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;FS vs. FI - 25 vs. Average\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\nplot(Data$FS.MS.25,Data$FS.MS,col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;MS vs. FI - 25 vs. Average\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\nplot(Data$MS.FI.25,Data$MS.FI,col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;MS vs. FI - 25 vs. Average\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\nplot(Data$FS.FI.18,Data$FS.FI,col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;FS vs. FI - 18 vs. Average\u0026quot;,xlab=\u0026quot;18C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\nplot(Data$FS.MS.18,Data$FS.MS,col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;MS vs. FI - 18 vs. Average\u0026quot;,xlab=\u0026quot;18C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\nplot(Data$MS.FI.18,Data$MS.FI,col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;MS vs. FI - 18 vs. Average\u0026quot;,xlab=\u0026quot;18C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\n\ndev.off()\n\npdf(\u0026quot;/data/RNASeq/CompareByTemp/Pval-genewise-byTemp-QQ.pdf\u0026quot;,width=12,height=8)\npar(mfrow=(c(3,3)))\n\nplot(sort(Data$FS.FI.25),sort(Data$FS.FI.18),col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;FS vs. FI - Temp\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;18C\u0026quot;)\nabline(0,1,col=\u0026quot;red\u0026quot;,lty=2,lwd=2)\nplot(sort(Data$FS.MS.25),sort(Data$FS.MS.18),col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;FS vs. MS - Temp\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;18C\u0026quot;)\nabline(0,1,col=\u0026quot;red\u0026quot;,lty=2,lwd=2)\nplot(sort(Data$MS.FI.25),sort(Data$MS.FI.18),col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;MS vs. FI - Temp\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;18C\u0026quot;)\nabline(0,1,col=\u0026quot;red\u0026quot;,lty=2,lwd=2)\nplot(sort(Data$FS.FI.25),sort(Data$FS.FI),col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;FS vs. FI - 25 vs. Average\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\nabline(0,1,col=\u0026quot;red\u0026quot;,lty=2,lwd=2)\nplot(sort(Data$FS.MS.25),sort(Data$FS.MS),col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;FS vs. MS - 25 vs. Average\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\nabline(0,1,col=\u0026quot;red\u0026quot;,lty=2,lwd=2)\nplot(sort(Data$MS.FI.25),sort(Data$MS.FI),col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;MS vs. FI - 25 vs. Average\u0026quot;,xlab=\u0026quot;25C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\nabline(0,1,col=\u0026quot;red\u0026quot;,lty=2,lwd=2)\nplot(sort(Data$FS.FI.18),sort(Data$FS.FI),col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;FS vs. FI - 18 vs. Average\u0026quot;,xlab=\u0026quot;18C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\nabline(0,1,col=\u0026quot;red\u0026quot;,lty=2,lwd=2)\nplot(sort(Data$FS.MS.18),sort(Data$FS.MS),col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;FS vs. MS - 18 vs. Average\u0026quot;,xlab=\u0026quot;18C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\nabline(0,1,col=\u0026quot;red\u0026quot;,lty=2,lwd=2)\nplot(sort(Data$MS.FI.18),sort(Data$MS.FI),col=rgb(0,0,0,0.2),pch=16,main = \u0026quot;MS vs. FI - 18 vs. Average\u0026quot;,xlab=\u0026quot;18C\u0026quot;, ylab=\u0026quot;average\u0026quot;)\nabline(0,1,col=\u0026quot;red\u0026quot;,lty=2,lwd=2)\n\ndev.off()\n\"\u003e\u003cpre\u003edir.create(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/data/RNASeq/CompareByTemp\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# load pvalue data from limma\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eread.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/CompareByTemp_Pval.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Compare p-value distributions of 18 and 25 temp \u003c/span\u003e\n\npdf(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/RNASeq/CompareByTemp/Pval-genewise-byTemp.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\npar(\u003cspan class=\"pl-v\"\u003emfrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(c(\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)))\n\nplot(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI.25\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI.18\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS vs. FI - Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplot(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS.25\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS.18\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS vs. MS - Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplot(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI.25\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI.18\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMS vs. FI - Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplot(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI.25\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS vs. FI - 25 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplot(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS.25\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMS vs. FI - 25 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplot(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI.25\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMS vs. FI - 25 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplot(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI.18\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS vs. FI - 18 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplot(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS.18\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMS vs. FI - 18 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplot(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI.18\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMS vs. FI - 18 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\ndev.off()\n\npdf(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/data/RNASeq/CompareByTemp/Pval-genewise-byTemp-QQ.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e)\npar(\u003cspan class=\"pl-v\"\u003emfrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(c(\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)))\n\nplot(sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI.25\u003c/span\u003e),sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI.18\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS vs. FI - Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\nplot(sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS.25\u003c/span\u003e),sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS.18\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS vs. MS - Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\nplot(sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI.25\u003c/span\u003e),sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI.18\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMS vs. FI - Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\nplot(sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI.25\u003c/span\u003e),sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS vs. FI - 25 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\nplot(sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS.25\u003c/span\u003e),sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS vs. MS - 25 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\nplot(sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI.25\u003c/span\u003e),sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMS vs. FI - 25 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e25C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\nplot(sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI.18\u003c/span\u003e),sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.FI\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS vs. FI - 18 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\nplot(sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS.18\u003c/span\u003e),sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFS.MS\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFS vs. MS - 18 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\nplot(sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI.18\u003c/span\u003e),sort(\u003cspan class=\"pl-smi\"\u003eData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMS.FI\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ergb(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003epch\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMS vs. FI - 18 vs. Average\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18C\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eaverage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.4) Overlap between Genomic and RNASeq candidates\u003c/h4\u003e\u003ca id=\"user-content-44-overlap-between-genomic-and-rnaseq-candidates\" class=\"anchor\" aria-label=\"Permalink: 4.4) Overlap between Genomic and RNASeq candidates\" href=\"#44-overlap-between-genomic-and-rnaseq-candidates\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFinally, we tested for overlap between genomic and RNASeq candidates based on rank-rank hypergeometric tests as implemented in the \u003ca href=\"https://github.com/RRHO2/RRHO2\"\u003eRRHO2\u003c/a\u003e package in \u003cem\u003eR\u003c/em\u003e. First perpare the two datasets\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nmkdir -p /data/candidates/RRHO\n\n## rank genes based on average FST in the dataset from Florida\npython3 /scripts/v3/RankGenesByStat.py \\\n    --input /data/PopGen/FI-FS_ann.weir.fst \\\n    --perc 0.25 \\\n    --gtf /data/RNASeq/dmel-all-r6.17.gtf.gz \\\n    \u0026gt; /data/candidates/RRHO/Genes.fst\n\n\n## loop through all RNASeq comparisons\nfor i in FS-FI FS-FI_18 FS-FI_25 FS-FI_int\ndo\n\n    mkdir /data/candidates/RRHO/RRHO-${i}\n\n    ## make input matrix for RRHO\n    python3 /scripts/v3/Input4RRHO.py \\\n        --inputs /data/candidates/RRHO/Genes.fst,/data/RNASeq/limma/FullInfo/${i}.txt \\\n        --genes 1,1 \\\n        --stats -2,-2 \\\n        --type FST,P \\\n        \u0026gt; /data/candidates/RRHO/Genes-${i}.rrho\n\n    ## execute R code for each dataset \n    printf \u0026quot;\u0026quot;\u0026quot;\n    library(RRHO2)\n    Rdata=read.table('/data/candidates/RRHO/Genes-${i}.rrho',header=T)\n\n    DNA=data.frame('gene'=Rdata\\$genes1,'val'=Rdata\\$rank1)\n    RNA=data.frame('gene'=Rdata\\$genes2,'val'=Rdata\\$rank2)\n    Test=RRHO2(DNA,\n        RNA,\n        plots = T,\n        labels=c('DNA','RNA'),\n        outputdir = '/data/candidates/RRHO/RRHO-${i}',\n        alternative = 'enrichment')\n\n    \u0026quot;\u0026quot;\u0026quot; \u0026gt; /data/candidates/RRHO/RRHO-${i}/test.r\n\n    Rscript /data/candidates/RRHO/RRHO-${i}/test.r\n\ndone\n\"\u003e\u003cpre\u003emkdir -p /data/candidates/RRHO\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# rank genes based on average FST in the dataset from Florida\u003c/span\u003e\npython3 /scripts/v3/RankGenesByStat.py \\\n    --input /data/PopGen/FI-FS_ann.weir.fst \\\n    --perc 0.25 \\\n    --gtf /data/RNASeq/dmel-all-r6.17.gtf.gz \\\n    \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/candidates/RRHO/Genes.fst\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# loop through all RNASeq comparisons\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e FS-FI FS-FI_18 FS-FI_25 FS-FI_int\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\n    mkdir /data/candidates/RRHO/RRHO-\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\n\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# make input matrix for RRHO\u003c/span\u003e\n    python3 /scripts/v3/Input4RRHO.py \\\n        --inputs /data/candidates/RRHO/Genes.fst,/data/RNASeq/limma/FullInfo/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e.txt \\\n        --genes 1,1 \\\n        --stats -2,-2 \\\n        --type FST,P \\\n        \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/candidates/RRHO/Genes-\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e.rrho\n\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# execute R code for each dataset \u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    library(RRHO2)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    Rdata=read.table('/data/candidates/RRHO/Genes-\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e.rrho',header=T)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    DNA=data.frame('gene'=Rdata\u003cspan class=\"pl-cce\"\u003e\\$\u003c/span\u003egenes1,'val'=Rdata\u003cspan class=\"pl-cce\"\u003e\\$\u003c/span\u003erank1)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    RNA=data.frame('gene'=Rdata\u003cspan class=\"pl-cce\"\u003e\\$\u003c/span\u003egenes2,'val'=Rdata\u003cspan class=\"pl-cce\"\u003e\\$\u003c/span\u003erank2)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    Test=RRHO2(DNA,\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        RNA,\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        plots = T,\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        labels=c('DNA','RNA'),\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        outputdir = '/data/candidates/RRHO/RRHO-\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e',\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        alternative = 'enrichment')\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e /data/candidates/RRHO/RRHO-\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e/test.r\n\n    Rscript /data/candidates/RRHO/RRHO-\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e/test.r\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eReferences\u003c/h2\u003e\u003ca id=\"user-content-references\" class=\"anchor\" aria-label=\"Permalink: References\" href=\"#references\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eKapopoulou, A., Kapun, M., Pieper, B., Pavlidis, P., Wilches, R., Duchen, P., et al. 2020. Demographic analyses of a new sample of haploid genomes from a Swedish population of \u003cem\u003eDrosophila melanogaster\u003c/em\u003e. Sci Rep 10: 22415. Nature Publishing Group.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eKapun, M., Schalkwyk, H. van, McAllister, B., Flatt, T. \u0026amp; Schltterer, C. 2014. Inference of chromosomal inversion dynamics from Pool-Seq data in natural and laboratory populations of \u003cem\u003eDrosophila melanogaster\u003c/em\u003e. Molecular Ecology 23: 18131827.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eKapun, M., Barrn, M.G., Staubach, F., Obbard, D.J., Wiberg, R.A.W., Vieira, J., et al. 2020. Genomic Analysis of European \u003cem\u003eDrosophila melanogaster\u003c/em\u003e Populations Reveals Longitudinal Structure, Continent-Wide Selection, and Previously Unknown DNA Viruses. Mol Biol Evol 37: 26612678. Oxford Academic.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eKapun, M., Mitchell, E.D., Kawecki, T.J., Schmidt, P. \u0026amp; Flatt, T. 2023. An Ancestral Balanced Inversion Polymorphism Confers Global Adaptation. bioRxiv.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eKofler, R., Pandey, R.V. \u0026amp; Schlotterer, C. 2011. PoPoolation2: identifying differentiation between populations using sequencing of pooled DNA samples (Pool-Seq). Bioinformatics 27: 34353436.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLack, J.B., Lange, J.D., Tang, A.D., B, C.-D., Russell \u0026amp; Pool, J.E. 2016. A Thousand Fly Genomes: An Expanded \u003cem\u003eDrosophila\u003c/em\u003e Genome Nexus. Mol Biol Evol 33: msw195-3313.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eRane, R.V., Rako, L., Kapun, M., Lee, S.F. \u0026amp; Hoffmann, A.A. 2015. Genomic evidence for role of inversion 3RP of \u003cem\u003eDrosophila melanogaster\u003c/em\u003e in facilitating climate change adaptation. Molecular Ecology 24: 24232432.\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"Bioinformatic pipeline for population genomic inference of In(3R)Payne","anchor":"bioinformatic-pipeline-for-population-genomic-inference-of-in3rpayne","htmlText":"Bioinformatic pipeline for population genomic inference of In(3R)Payne"},{"level":2,"text":"1) Map and generate phased sequencing data","anchor":"1-map-and-generate-phased-sequencing-data","htmlText":"1) Map and generate phased sequencing data"},{"level":3,"text":"1.1) USA","anchor":"11-usa","htmlText":"1.1) USA"},{"level":3,"text":"1.2) Worldwide samples from DGN dataset (see Lack, et al.2016)","anchor":"12-worldwide-samples-from-dgn-dataset-see-lack-et-al-2016","htmlText":"1.2) Worldwide samples from DGN dataset (see Lack, et al.2016)"},{"level":3,"text":"1.3) Zambia from DGN dataset (see Lack, et al.2016)","anchor":"13-zambia-from-dgn-dataset-see-lack-et-al-2016","htmlText":"1.3) Zambia from DGN dataset (see Lack, et al.2016)"},{"level":3,"text":"1.4) Portugal (see Kapun, et al.2014 and Franssen, et al.2016)","anchor":"14-portugal-see-kapun-et-al-2014-and-franssen-et-al-2016","htmlText":"1.4) Portugal (see Kapun, et al.2014 and Franssen, et al.2016)"},{"level":3,"text":"1.5) Sweden (see Kapopoulou, et al.2020)","anchor":"15-sweden-see-kapopoulou-et-al-2020","htmlText":"1.5) Sweden (see Kapopoulou, et al.2020)"},{"level":3,"text":"1.6) Australia","anchor":"16-australia","htmlText":"1.6) Australia"},{"level":2,"text":"2) Phylogenetic approach","anchor":"2-phylogenetic-approach","htmlText":"2) Phylogenetic approach"},{"level":3,"text":"2.1) Adjust Australian samples to match the other data","anchor":"21-adjust-australian-samples-to-match-the-other-data","htmlText":"2.1) Adjust Australian samples to match the other data"},{"level":3,"text":"2.2) Merge all files in *.cons format","anchor":"22-merge-all-files-in-cons-format","htmlText":"2.2) Merge all files in *.cons format"},{"level":2,"text":"3) Population Genetics Analysis","anchor":"3-population-genetics-analysis","htmlText":"3) Population Genetics Analysis"},{"level":3,"text":"3.1) Genetic variation","anchor":"31-genetic-variation","htmlText":"3.1) Genetic variation"},{"level":3,"text":"3.2) Genetic differentiation","anchor":"32-genetic-differentiation","htmlText":"3.2) Genetic differentiation"},{"level":3,"text":"3.3) African origin","anchor":"33-african-origin","htmlText":"3.3) African origin"},{"level":3,"text":"3.4) Linkage Disequilibrium","anchor":"34-linkage-disequilibrium","htmlText":"3.4) Linkage Disequilibrium"},{"level":4,"text":"3.4.1) SNP-wise LD with the inversion","anchor":"341-snp-wise-ld-with-the-inversion","htmlText":"3.4.1) SNP-wise LD with the inversion"},{"level":4,"text":"3.4.2) LD heatmaps","anchor":"342-ld-heatmaps","htmlText":"3.4.2) LD heatmaps"},{"level":4,"text":"3.4.3) Decay of LD","anchor":"343-decay-of-ld","htmlText":"3.4.3) Decay of LD"},{"level":3,"text":"4) Transcriptomic analysis","anchor":"4-transcriptomic-analysis","htmlText":"4) Transcriptomic analysis"},{"level":4,"text":"4.1) Trim and map RNASeq data","anchor":"41-trim-and-map-rnaseq-data","htmlText":"4.1) Trim and map RNASeq data"},{"level":4,"text":"4.2) Test for differential expression","anchor":"42-test-for-differential-expression","htmlText":"4.2) Test for differential expression"},{"level":4,"text":"4.2) Test for positional overrepresentation","anchor":"42-test-for-positional-overrepresentation","htmlText":"4.2) Test for positional overrepresentation"},{"level":4,"text":"4.3) Compare by temperarture","anchor":"43-compare-by-temperarture","htmlText":"4.3) Compare by temperarture"},{"level":4,"text":"4.4) Overlap between Genomic and RNASeq candidates","anchor":"44-overlap-between-genomic-and-rnaseq-candidates","htmlText":"4.4) Overlap between Genomic and RNASeq candidates"},{"level":2,"text":"References","anchor":"references","htmlText":"References"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fcapoony%2FIn3RPayne_PopGenomics"}},{"displayName":"LICENSE","repoName":"In3RPayne_PopGenomics","refName":"master","path":"LICENSE","preferredFileType":"license","tabName":"MIT","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fcapoony%2FIn3RPayne_PopGenomics"}}],"overviewFilesProcessingTime":243.24368199999998}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
.bhqztV{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.bhqztV span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.bhqztV:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.bhqztV:focus{outline:2px solid transparent;}/*!sc*/
.bhqztV:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.bhqztV:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.bhqztV::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:transparent;border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.bhqztV::after{background-color:transparent;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,bhqztV,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="master branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        master
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/capoony/In3RPayne_PopGenomics/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/capoony/In3RPayne_PopGenomics/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/capoony/In3RPayne_PopGenomics/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/capoony/In3RPayne_PopGenomics/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/capoony/In3RPayne_PopGenomics/commits/master/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           41 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/capoony/In3RPayne_PopGenomics/commits/master/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="data, (Directory)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/tree/master/data" title="data">
                          data
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="data, (Directory)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/tree/master/data" title="data">
                          data
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="scripts, (Directory)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/tree/master/scripts" title="scripts">
                          scripts
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="scripts, (Directory)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/tree/master/scripts" title="scripts">
                          scripts
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="shell, (Directory)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/tree/master/shell" title="shell">
                          shell
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="shell, (Directory)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/tree/master/shell" title="shell">
                          shell
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".DS_Store, (File)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/blob/master/.DS_Store" title=".DS_Store">
                          .DS_Store
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".DS_Store, (File)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/blob/master/.DS_Store" title=".DS_Store">
                          .DS_Store
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/blob/master/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/blob/master/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/capoony/In3RPayne_PopGenomics/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-law" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="MIT license">
                        MIT license
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Bioinformatic pipeline for population genomic inference of
                      <em>
                       In(3R)Payne
                      </em>
                     </h1>
                     <a aria-label="Permalink: Bioinformatic pipeline for population genomic inference of In(3R)Payne" class="anchor" href="#bioinformatic-pipeline-for-population-genomic-inference-of-in3rpayne" id="user-content-bioinformatic-pipeline-for-population-genomic-inference-of-in3rpayne">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     see Material and Methods in
                     <a href="https://www.biorxiv.org/content/10.1101/2023.01.31.526462v1" rel="nofollow">
                      Kapun
                      <em>
                       et al.
                      </em>
                      (2023)
                     </a>
                     for more details. Most scripts were originally written in Python v.2. They were lifted over to Python v.3 using the
                     <a href="https://docs.python.org/3/library/2to3.html" rel="nofollow">
                      <code>
                       2to3
                      </code>
                     </a>
                     tool and storded in the /scripts/v3 folder. The original files can be found in the /scripts folder.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      1) Map and generate phased sequencing data
                     </h2>
                     <a aria-label="Permalink: 1) Map and generate phased sequencing data" class="anchor" href="#1-map-and-generate-phased-sequencing-data" id="user-content-1-map-and-generate-phased-sequencing-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      1.1) USA
                     </h3>
                     <a aria-label="Permalink: 1.1) USA" class="anchor" href="#11-usa" id="user-content-11-usa">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     For each of the newly sequenced library in the USA, test the quality with FASTQC, trim the raw reads with cutadapt, map the reads with bbmap, sort and deduplicate the BAM file with picard and realign around InDels with GATK
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="mkdir /data/mapping

## loop through raw reads in folder /data and store BAM in /data/mapping
for file in /data/USA/*_R1.fq.gz
do

    tmp=${file##*/}
    i=${tmp%%_*}

    sh shell/mapping.sh \
        /data/USA/${i}_R1.fq.gz	\
        /data/USA/${i}_R2.fq.gz	\	
        ${i} \
        /data/USA/mapping/${i} \
        bbmap

done " dir="auto">
                     <pre>mkdir /data/mapping

<span class="pl-c"><span class="pl-c">#</span># loop through raw reads in folder /data and store BAM in /data/mapping</span>
<span class="pl-k">for</span> <span class="pl-smi">file</span> <span class="pl-k">in</span> /data/USA/<span class="pl-k">*</span>_R1.fq.gz
<span class="pl-k">do</span>

    tmp=<span class="pl-smi">${file<span class="pl-k">##*/</span>}</span>
    i=<span class="pl-smi">${tmp<span class="pl-k">%%</span>_<span class="pl-k">*</span>}</span>

    sh shell/mapping.sh \
        /data/USA/<span class="pl-smi">${i}</span>_R1.fq.gz	\
        /data/USA/<span class="pl-smi">${i}</span>_R2.fq.gz	<span class="pl-cce">\	</span>
        <span class="pl-smi">${i}</span> \
        /data/USA/mapping/<span class="pl-smi">${i}</span> \
        bbmap

<span class="pl-k">done</span> </pre>
                    </div>
                    <p dir="auto">
                     Following the approach in
                     <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/mec.12594" rel="nofollow">
                      Kapun
                      <em>
                       et al.
                      </em>
                      (2014)
                     </a>
                     we bioinformatically obtained haploid genomes from hemiclones. This requires
                     <a href="https:/sourceforge.net/p/popoolation2/wiki/Main/">
                      PoPoolation2
                     </a>
                     from (
                     <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3232374/" rel="nofollow">
                      Kofler et al. 2011
                     </a>
                     )
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="mkdir /data/USA
mkdir /data/consensus

## synchronize as MPILEUP based on hologenome from Kapun et al. (2020) only including 3L and 3R
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/USA_BAM.txt \
    -l /data/regions.bed.txt &gt; /data/USA/USA.mpileup

## use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format
java -jar /scripts/popoolation2_1201/mpileup2sync.jar \
    -input /data/USA/USA.mpileup \
    -threads 20 \
    -output /data/USA/USA.sync

## calculate max coverage threshold (see Kapun et al. 2014)
python3 /scripts/v3/max_coverage.py \
    -input /data/USA/USA.sync \
    -max-coverage 0.05 \
    -output /data/USA/USA.cov

## use GNU parallel to parallelize the phasing (see Kapun et al. 2014 for details)
parallel -a /data/USA/USA.sync \
    -k \
    -pipepart \
    -cat python3 /scripts/v3/extract_consensus.py \
        -input {} \
        -min-coverage 10 \
        -min-count 10 \
        -max-coverage /data/USA/USA.cov \
        -output /data/consensus/USA \
        | gzip &gt; /data/consensus/USA.consensus.gz

 ## convert consensus to VCF and retain site if &gt; 50% of all samples with non-N's
python3 /scripts/v3/cons2vcf.py \
    -input /data/consensus/USA.consensus.gz \
    -output /data/consensus/America \
    -ind 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58 \
    -N-cutoff 0.5 \
    -names Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81" dir="auto">
                     <pre>mkdir /data/USA
mkdir /data/consensus

<span class="pl-c"><span class="pl-c">#</span># synchronize as MPILEUP based on hologenome from Kapun et al. (2020) only including 3L and 3R</span>
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/USA_BAM.txt \
    -l /data/regions.bed.txt <span class="pl-k">&gt;</span> /data/USA/USA.mpileup

<span class="pl-c"><span class="pl-c">#</span># use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format</span>
java -jar /scripts/popoolation2_1201/mpileup2sync.jar \
    -input /data/USA/USA.mpileup \
    -threads 20 \
    -output /data/USA/USA.sync

<span class="pl-c"><span class="pl-c">#</span># calculate max coverage threshold (see Kapun et al. 2014)</span>
python3 /scripts/v3/max_coverage.py \
    -input /data/USA/USA.sync \
    -max-coverage 0.05 \
    -output /data/USA/USA.cov

<span class="pl-c"><span class="pl-c">#</span># use GNU parallel to parallelize the phasing (see Kapun et al. 2014 for details)</span>
parallel -a /data/USA/USA.sync \
    -k \
    -pipepart \
    -cat python3 /scripts/v3/extract_consensus.py \
        -input {} \
        -min-coverage 10 \
        -min-count 10 \
        -max-coverage /data/USA/USA.cov \
        -output /data/consensus/USA \
        <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/consensus/USA.consensus.gz

 <span class="pl-c"><span class="pl-c">#</span># convert consensus to VCF and retain site if &gt; 50% of all samples with non-N's</span>
python3 /scripts/v3/cons2vcf.py \
    -input /data/consensus/USA.consensus.gz \
    -output /data/consensus/America \
    -ind 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58 \
    -N-cutoff 0.5 \
    -names Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      1.2) Worldwide samples from DGN dataset (see Lack,
                      <em>
                       et al.
                      </em>
                      <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5100052/" rel="nofollow">
                       2016
                      </a>
                      )
                     </h3>
                     <a aria-label="Permalink: 1.2) Worldwide samples from DGN dataset (see Lack, et al.2016)" class="anchor" href="#12-worldwide-samples-from-dgn-dataset-see-lack-et-al-2016" id="user-content-12-worldwide-samples-from-dgn-dataset-see-lack-et-al-2016">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
mkdir /data/DGNs

## download raw data from SRA and map with same pipeline as above
while IFS=',' read -r name SRA
do 
    sh /shell/obtain-n-map-africa.sh \
        /data/DGN/
        ${name} \
        ${SRA}
done &lt; /data/DGN_SRA.txt

## now merge all strains to a big mpileup
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/DGN_BAM.txt 
    -l /data/regions.bed \
    | gzip &gt; /data/DGN/DGN.mpileup.gz

## convert major allele to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)
gunzip -c /data/DGN/DGN.mpileup.gz \
    | parallel \
    -pipe  \
    -k \
    -cat python3 /scripts/v3/mpileup2cons.py \
        -c 10,200 \
        -m {} \
        -u 0.9 \
        -b 20 | gzip &gt; /data/consensus/DGN.consensus.gz
" dir="auto">
                     <pre>mkdir /data/DGNs

<span class="pl-c"><span class="pl-c">#</span># download raw data from SRA and map with same pipeline as above</span>
<span class="pl-k">while</span> IFS=<span class="pl-s"><span class="pl-pds">'</span>,<span class="pl-pds">'</span></span> <span class="pl-c1">read</span> -r name SRA
<span class="pl-k">do</span> 
    sh /shell/obtain-n-map-africa.sh \
        /data/DGN/
        <span class="pl-smi">${name}</span> \
        <span class="pl-smi">${SRA}</span>
<span class="pl-k">done</span> <span class="pl-k">&lt;</span> /data/DGN_SRA.txt

<span class="pl-c"><span class="pl-c">#</span># now merge all strains to a big mpileup</span>
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/DGN_BAM.txt 
    -l /data/regions.bed \
    <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/DGN/DGN.mpileup.gz

<span class="pl-c"><span class="pl-c">#</span># convert major allele to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)</span>
gunzip -c /data/DGN/DGN.mpileup.gz \
    <span class="pl-k">|</span> parallel \
    -pipe  \
    -k \
    -cat python3 /scripts/v3/mpileup2cons.py \
        -c 10,200 \
        -m {} \
        -u 0.9 \
        -b 20 <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/consensus/DGN.consensus.gz
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      1.3) Zambia from DGN dataset (see Lack,
                      <em>
                       et al.
                      </em>
                      <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5100052/" rel="nofollow">
                       2016
                      </a>
                      )
                     </h3>
                     <a aria-label="Permalink: 1.3) Zambia from DGN dataset (see Lack, et al.2016)" class="anchor" href="#13-zambia-from-dgn-dataset-see-lack-et-al-2016" id="user-content-13-zambia-from-dgn-dataset-see-lack-et-al-2016">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
mkdir /data/Zambia

## download raw data from SRA and map with same pipeline as above
while IFS=',' read -r name SRA
do 
    sh /shell/obtain-n-map-africa.sh \
        /data/Zambia/
        ${name} \
        ${SRA}
done &lt; /data/Zambia_SRA.txt

## now merge all Zambian strains to a big mpileup
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/Zambia_BAM.txt 
    -l /data/regions.bed \
    | gzip &gt; /data/Zambia/Zambia.mpileup.gz

## convert to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)
gunzip -c /data/Zambia/Zambia.mpileup.gz \
    | parallel \
    -pipe  \
    -k \
    -cat python3 /scripts/v3/mpileup2cons.py \
        -c 10,200 \
        -m {} \
        -u 0.9 \
        -b 20 | gzip &gt; /data/consensus/Zambia.consensus.gz

## convert consensus to VCF and retain site if &gt; 50% of all samples with non-N's
python3 /scripts/v3/cons2vcf.py \
    -input /data/consensus/Zambia.consensus.gz \
    -output /data/consensus/Africa \
    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162 \
    -N-cutoff 0.5 \
    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388" dir="auto">
                     <pre>mkdir /data/Zambia

<span class="pl-c"><span class="pl-c">#</span># download raw data from SRA and map with same pipeline as above</span>
<span class="pl-k">while</span> IFS=<span class="pl-s"><span class="pl-pds">'</span>,<span class="pl-pds">'</span></span> <span class="pl-c1">read</span> -r name SRA
<span class="pl-k">do</span> 
    sh /shell/obtain-n-map-africa.sh \
        /data/Zambia/
        <span class="pl-smi">${name}</span> \
        <span class="pl-smi">${SRA}</span>
<span class="pl-k">done</span> <span class="pl-k">&lt;</span> /data/Zambia_SRA.txt

<span class="pl-c"><span class="pl-c">#</span># now merge all Zambian strains to a big mpileup</span>
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/Zambia_BAM.txt 
    -l /data/regions.bed \
    <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/Zambia/Zambia.mpileup.gz

<span class="pl-c"><span class="pl-c">#</span># convert to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)</span>
gunzip -c /data/Zambia/Zambia.mpileup.gz \
    <span class="pl-k">|</span> parallel \
    -pipe  \
    -k \
    -cat python3 /scripts/v3/mpileup2cons.py \
        -c 10,200 \
        -m {} \
        -u 0.9 \
        -b 20 <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/consensus/Zambia.consensus.gz

<span class="pl-c"><span class="pl-c">#</span># convert consensus to VCF and retain site if &gt; 50% of all samples with non-N's</span>
python3 /scripts/v3/cons2vcf.py \
    -input /data/consensus/Zambia.consensus.gz \
    -output /data/consensus/Africa \
    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162 \
    -N-cutoff 0.5 \
    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      1.4) Portugal (see Kapun,
                      <em>
                       et al.
                      </em>
                      <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/mec.12594" rel="nofollow">
                       2014
                      </a>
                      and Franssen,
                      <em>
                       et al.
                      </em>
                      <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5100052/" rel="nofollow">
                       2016
                      </a>
                      )
                     </h3>
                     <a aria-label="Permalink: 1.4) Portugal (see Kapun, et al.2014 and Franssen, et al.2016)" class="anchor" href="#14-portugal-see-kapun-et-al-2014-and-franssen-et-al-2016" id="user-content-14-portugal-see-kapun-et-al-2014-and-franssen-et-al-2016">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
mkdir /data/Portugal

## get and map data from Franssen et al. 2014
while IFS=',' read -r name FWD REV
do 
    sh /shell/obtain-n-map-portugal.sh \
        /data/Portugal /
        ${name} \
        ${FWD} \
        ${REV}
done &lt; /data/Portugal_SRA.txt

## get and map data from Kapun et al. 2016
while IFS=',' read -r name SRA
do 
    sh /shell/obtain-n-map-africa.sh \
        /data/Portugal /
        ${name} \
        ${SRA}
done &lt; /data/Portugal_SRA2.txt

## now merge all Portugese strains to a big mpileup
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/Portugal_BAM.txt 
    -l /data/regions.bed \
    | gzip &gt; /data/Portugal/Portugal.mpileup.gz

## use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format
java -jar /scripts/popoolation2_1201/mpileup2sync.jar \
    -input /data/Portugal/Portugal.mpileup \
    -threads 20 \
    -output /data/Portugal/Portugal.sync

## calculate max coverage threshold (see Kapun et al. 2014)
python3 /scripts/v3/max_coverage.py \
    -input /data/Portugal/Portugal.sync \
    -max-coverage 0.05 \
    -output /data/Portugal/Portugal.cov

## use GNU parallel to parallelize the phasing (see Kapun et al. 2014 for details)
parallel -a /data/Portugal/Portugal.sync \
    -k \
    -pipepart \
    -cat python3 /scripts/v3/extract_consensus.py \
        -input {} \
        -min-coverage 10 \
        -min-count 10 \
        -max-coverage /data/Portugal/Portugal.cov \
        -output /data/consensus/Portugal \
        | gzip &gt; /data/consensus/Portugal.consensus.gz

## convert consensus to VCF and retain site if &gt; 50% of all samples with non-N's
python3 /scripts/v3/cons2vcf.py \
    -input /data/consensus/Portugal.consensus.gz \
    -output /data/consensus/Portugal \
    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26 \
    -N-cutoff 0.5 \
    -names F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0" dir="auto">
                     <pre>mkdir /data/Portugal

<span class="pl-c"><span class="pl-c">#</span># get and map data from Franssen et al. 2014</span>
<span class="pl-k">while</span> IFS=<span class="pl-s"><span class="pl-pds">'</span>,<span class="pl-pds">'</span></span> <span class="pl-c1">read</span> -r name FWD REV
<span class="pl-k">do</span> 
    sh /shell/obtain-n-map-portugal.sh \
        /data/Portugal /
        <span class="pl-smi">${name}</span> \
        <span class="pl-smi">${FWD}</span> \
        <span class="pl-smi">${REV}</span>
<span class="pl-k">done</span> <span class="pl-k">&lt;</span> /data/Portugal_SRA.txt

<span class="pl-c"><span class="pl-c">#</span># get and map data from Kapun et al. 2016</span>
<span class="pl-k">while</span> IFS=<span class="pl-s"><span class="pl-pds">'</span>,<span class="pl-pds">'</span></span> <span class="pl-c1">read</span> -r name SRA
<span class="pl-k">do</span> 
    sh /shell/obtain-n-map-africa.sh \
        /data/Portugal /
        <span class="pl-smi">${name}</span> \
        <span class="pl-smi">${SRA}</span>
<span class="pl-k">done</span> <span class="pl-k">&lt;</span> /data/Portugal_SRA2.txt

<span class="pl-c"><span class="pl-c">#</span># now merge all Portugese strains to a big mpileup</span>
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/Portugal_BAM.txt 
    -l /data/regions.bed \
    <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/Portugal/Portugal.mpileup.gz

<span class="pl-c"><span class="pl-c">#</span># use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format</span>
java -jar /scripts/popoolation2_1201/mpileup2sync.jar \
    -input /data/Portugal/Portugal.mpileup \
    -threads 20 \
    -output /data/Portugal/Portugal.sync

<span class="pl-c"><span class="pl-c">#</span># calculate max coverage threshold (see Kapun et al. 2014)</span>
python3 /scripts/v3/max_coverage.py \
    -input /data/Portugal/Portugal.sync \
    -max-coverage 0.05 \
    -output /data/Portugal/Portugal.cov

<span class="pl-c"><span class="pl-c">#</span># use GNU parallel to parallelize the phasing (see Kapun et al. 2014 for details)</span>
parallel -a /data/Portugal/Portugal.sync \
    -k \
    -pipepart \
    -cat python3 /scripts/v3/extract_consensus.py \
        -input {} \
        -min-coverage 10 \
        -min-count 10 \
        -max-coverage /data/Portugal/Portugal.cov \
        -output /data/consensus/Portugal \
        <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/consensus/Portugal.consensus.gz

<span class="pl-c"><span class="pl-c">#</span># convert consensus to VCF and retain site if &gt; 50% of all samples with non-N's</span>
python3 /scripts/v3/cons2vcf.py \
    -input /data/consensus/Portugal.consensus.gz \
    -output /data/consensus/Portugal \
    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26 \
    -N-cutoff 0.5 \
    -names F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      1.5) Sweden (see Kapopoulou,
                      <em>
                       et al.
                      </em>
                      <a href="https://www.nature.com/articles/s41598-020-79720-1" rel="nofollow">
                       2020
                      </a>
                      )
                     </h3>
                     <a aria-label="Permalink: 1.5) Sweden (see Kapopoulou, et al.2020)" class="anchor" href="#15-sweden-see-kapopoulou-et-al-2020" id="user-content-15-sweden-see-kapopoulou-et-al-2020">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Use
                     <a href="https:/www.ibm.com/aspera/connect/">
                      Aspera
                     </a>
                     and
                     <a href="https:/www.ncbi.nlm.nih.gov/sra/docs/sradownload/">
                      sratoolkit
                     </a>
                     to download and convert raw data
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
mkdir /data/Sweden

## get and map data 
while IFS=$' \t\n'
read -r A B C D E F SRA H name remainder
do

## used the ASPERA app to download from SRA 
Aspera\ Connect.app/Contents/Resources/ascp \
    -i Aspera\ Connect.app/Contents/Resources/asperaweb_id_dsa.openssh \
    -T anonftp@ftp.ncbi.nlm.nih.gov:/sra/sra-instant/reads/ByRun/sra/SRR/${SRA:0:6}/$SRA/$SRA.sra \
    /data/Sweden/

## convert to FASTQ
scripts/sratoolkit.2.9.2-mac64/bin/fastq-dump.2.5.2 \
    -gzip \
    -split-3 \
    -outdir /data/Sweden/ \
    -A $name \
    /data/Sweden/$SRA.sra

rm /data/Sweden/$SRA.sra

mkdir -p /data/Sweden/$name

## trim with cutadapt (min base quality &gt;18, minium trimmed readlength 75bp) and map with bbmap (standard parameters) against D. mel v. 6.04
sh shell/mapping.sh \
    /data/Sweden/${name}_1.fastq.gz \
    /data/Sweden/${name}_2.fastq.gz \
    $name \
    /data/Sweden/$name \
    bbmap

done &lt; /data/Sweden/Sweden_SRA.txt

## now merge all Portugese strains to a big mpileup
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/Sweden_BAM.txt 
    -l /data/regions.bed \
    | gzip &gt; /data/Sweden/Sweden.mpileup.gz

## convert to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)
gunzip -c /data/Sweden/Sweden.mpileup.gz \
    | parallel \
    -pipe  \
    -k \
    -cat python3 /scripts/v3/mpileup2cons.py \
        -c 10,200 \
        -m {} \
        -u 0.9 \
        -b 20 | gzip &gt; /data/consensus/Sweden.consensus.gz

## Now merge data from Portugal and Sweden
python3 /scripts/merge_consensus.py \
    -consensus /data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz \
    | gzip &gt; /data/consensus/Europe.cons.gz

## convert consensus to VCF and retain site if &gt; 50% of all samples with non-N's
python3 /scripts/v3/cons2vcf.py \
    -input /data/consensus/Europe.cons.gz \
    -output /data/consensus/Europe \
    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40 \
    -N-cutoff 0.5 \
    -names F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94" dir="auto">
                     <pre>mkdir /data/Sweden

<span class="pl-c"><span class="pl-c">#</span># get and map data </span>
<span class="pl-k">while</span> IFS=<span class="pl-s"><span class="pl-pds">$'</span> <span class="pl-cce">\t\n</span><span class="pl-pds">'</span></span>
<span class="pl-c1">read</span> -r A B C D E F SRA H name remainder
<span class="pl-k">do</span>

<span class="pl-c"><span class="pl-c">#</span># used the ASPERA app to download from SRA </span>
Aspera<span class="pl-cce">\ </span>Connect.app/Contents/Resources/ascp \
    -i Aspera<span class="pl-cce">\ </span>Connect.app/Contents/Resources/asperaweb_id_dsa.openssh \
    -T anonftp@ftp.ncbi.nlm.nih.gov:/sra/sra-instant/reads/ByRun/sra/SRR/<span class="pl-smi">${SRA<span class="pl-k">:</span>0<span class="pl-k">:</span>6}</span>/<span class="pl-smi">$SRA</span>/<span class="pl-smi">$SRA</span>.sra \
    /data/Sweden/

<span class="pl-c"><span class="pl-c">#</span># convert to FASTQ</span>
scripts/sratoolkit.2.9.2-mac64/bin/fastq-dump.2.5.2 \
    -gzip \
    -split-3 \
    -outdir /data/Sweden/ \
    -A <span class="pl-smi">$name</span> \
    /data/Sweden/<span class="pl-smi">$SRA</span>.sra

rm /data/Sweden/<span class="pl-smi">$SRA</span>.sra

mkdir -p /data/Sweden/<span class="pl-smi">$name</span>

<span class="pl-c"><span class="pl-c">#</span># trim with cutadapt (min base quality &gt;18, minium trimmed readlength 75bp) and map with bbmap (standard parameters) against D. mel v. 6.04</span>
sh shell/mapping.sh \
    /data/Sweden/<span class="pl-smi">${name}</span>_1.fastq.gz \
    /data/Sweden/<span class="pl-smi">${name}</span>_2.fastq.gz \
    <span class="pl-smi">$name</span> \
    /data/Sweden/<span class="pl-smi">$name</span> \
    bbmap

<span class="pl-k">done</span> <span class="pl-k">&lt;</span> /data/Sweden/Sweden_SRA.txt

<span class="pl-c"><span class="pl-c">#</span># now merge all Portugese strains to a big mpileup</span>
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/Sweden_BAM.txt 
    -l /data/regions.bed \
    <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/Sweden/Sweden.mpileup.gz

<span class="pl-c"><span class="pl-c">#</span># convert to cons fileformat given that these libraries should be from haploid embryos (see Kapopoulou et al. 2020 for more details)</span>
gunzip -c /data/Sweden/Sweden.mpileup.gz \
    <span class="pl-k">|</span> parallel \
    -pipe  \
    -k \
    -cat python3 /scripts/v3/mpileup2cons.py \
        -c 10,200 \
        -m {} \
        -u 0.9 \
        -b 20 <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/consensus/Sweden.consensus.gz

<span class="pl-c"><span class="pl-c">#</span># Now merge data from Portugal and Sweden</span>
python3 /scripts/merge_consensus.py \
    -consensus /data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz \
    <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/consensus/Europe.cons.gz

<span class="pl-c"><span class="pl-c">#</span># convert consensus to VCF and retain site if &gt; 50% of all samples with non-N's</span>
python3 /scripts/v3/cons2vcf.py \
    -input /data/consensus/Europe.cons.gz \
    -output /data/consensus/Europe \
    -ind 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40 \
    -N-cutoff 0.5 \
    -names F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      1.6) Australia
                     </h3>
                     <a aria-label="Permalink: 1.6) Australia" class="anchor" href="#16-australia" id="user-content-16-australia">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We downloaded the raw sequencing data of
                     <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.13161" rel="nofollow">
                      Rane
                      <em>
                       et al.
                      </em>
                      2015
                     </a>
                     from SRA (accession:
                     <a href="https:/www.ncbi.nlm.nih.gov/bioproject/PRJNA221876/">
                      PRJNA221876
                     </a>
                     ) and then classified the karyotpyes as follows:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
mkdir /data/Australia

## loop through raw reads in folder /data and store BAM in /data/mapping
for file in /data/Australia/*_R1.fq.gz
do

    tmp=${file##*/}
    i=${tmp%%_*}

    sh shell/mapping.sh \
        /data/Australia/${i}_1.fq.gz	\
        /data/Australia/${i}_2.fq.gz	\	
        ${i} \
        /data/Australia/mapping/${i} \
        bbmap

done 

## synchronize as MPILEUP based on hologenome from Kapun et al. (2020) only including 3L and 3R
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/Australia_BAM.txt \
    -l /data/regions.bed.txt &gt; /data/Australia/Australia.mpileup

## use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format
java -jar /scripts/popoolation2_1201/mpileup2sync.jar \
    -input /data/Australia/Australia.mpileup \
    -threads 20 \
    -output /data/Australia/Australia.sync
" dir="auto">
                     <pre>mkdir /data/Australia

<span class="pl-c"><span class="pl-c">#</span># loop through raw reads in folder /data and store BAM in /data/mapping</span>
<span class="pl-k">for</span> <span class="pl-smi">file</span> <span class="pl-k">in</span> /data/Australia/<span class="pl-k">*</span>_R1.fq.gz
<span class="pl-k">do</span>

    tmp=<span class="pl-smi">${file<span class="pl-k">##*/</span>}</span>
    i=<span class="pl-smi">${tmp<span class="pl-k">%%</span>_<span class="pl-k">*</span>}</span>

    sh shell/mapping.sh \
        /data/Australia/<span class="pl-smi">${i}</span>_1.fq.gz	\
        /data/Australia/<span class="pl-smi">${i}</span>_2.fq.gz	<span class="pl-cce">\	</span>
        <span class="pl-smi">${i}</span> \
        /data/Australia/mapping/<span class="pl-smi">${i}</span> \
        bbmap

<span class="pl-k">done</span> 

<span class="pl-c"><span class="pl-c">#</span># synchronize as MPILEUP based on hologenome from Kapun et al. (2020) only including 3L and 3R</span>
samtools mpileup \
    -B \
    -f /data/Dmel_6.04_hologenome_v2.fasta \
    -b /data/Australia_BAM.txt \
    -l /data/regions.bed.txt <span class="pl-k">&gt;</span> /data/Australia/Australia.mpileup

<span class="pl-c"><span class="pl-c">#</span># use PoPoolation2 (Kofler et al. 2011) to covert MPILEUP to SYNC format</span>
java -jar /scripts/popoolation2_1201/mpileup2sync.jar \
    -input /data/Australia/Australia.mpileup \
    -threads 20 \
    -output /data/Australia/Australia.sync
</pre>
                    </div>
                    <p dir="auto">
                     Then we used the dataset of In(3R)Payne-specific marker SNPs
                     <code>
                      data/fixed095.txt
                     </code>
                     from
                     <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/mec.12594" rel="nofollow">
                      Kapun
                      <em>
                       et al.
                      </em>
                      2014
                     </a>
                     to test for the frequency of inversion-specific alleles in the individuals
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
## identify the average frequency of inversion-specific alleles in the individual libraries
python3 /scripts/v3/Australia-karyos.py \
    -karyo /data/fixed095.txt \
    -input /data/Australia/Australia.sync \
    -names IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \
    &gt; /data/Australia/Australia_karyos.txt
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># identify the average frequency of inversion-specific alleles in the individual libraries</span>
python3 /scripts/v3/Australia-karyos.py \
    -karyo /data/fixed095.txt \
    -input /data/Australia/Australia.sync \
    -names IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \
    <span class="pl-k">&gt;</span> /data/Australia/Australia_karyos.txt
</pre>
                    </div>
                    <p dir="auto">
                     We only considered samples where the inversion-specific alleles at four SNPs showed an average frequency of &gt;90% (considered "inverted") or &lt;10% (considered "standard").
                    </p>
                    <p dir="auto">
                     Then, we further downloaded the imputed SNP data from from DataDryad
                     <a href="https:/datadryad.org/stash/dataset/doi:10.5061/dryad.5q0m8">
                      doi:10.5061/dryad.5q0m8.
                     </a>
                     and used the karyotypic classification from above to calculate poppulation genetic statistics for each group separately.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      2) Phylogenetic approach
                     </h2>
                     <a aria-label="Permalink: 2) Phylogenetic approach" class="anchor" href="#2-phylogenetic-approach" id="user-content-2-phylogenetic-approach">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Here, we investigated the phylogenetic signal of worldwide samples based on SNPs located within
                     <em>
                      In(3R)P
                     </em>
                     or in distance to the inversion.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.1) Adjust Australian samples to match the other data
                     </h3>
                     <a aria-label="Permalink: 2.1) Adjust Australian samples to match the other data" class="anchor" href="#21-adjust-australian-samples-to-match-the-other-data" id="user-content-21-adjust-australian-samples-to-match-the-other-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     At first, we lifted genomic coordinates of the data, originally mapped to the
                     <em>
                      D. melanogaster
                     </em>
                     reference genome v. 5 to v.6 coordinates.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## convert coordinates
python3 /scripts/v3/Flybase_version_converter.py \
    -input /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs.vcf \
    -output /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs_v6.vcf

python3 /scripts/v3/Flybase_version_converter.py \
    -input /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs.vcf \
    -output /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs_v6.vcf

cat /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs_v6.vcf \
/data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs_v6.vcf \
&gt; /data/consensus/Australia.vcf

## then convert from VCF to cons format
python3 /scripts/v3/vcf2cons.py \
    /data/consensus/Australia.vcf \
    | gzip &gt; /data/Australia/Australia.cons.gz
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># convert coordinates</span>
python3 /scripts/v3/Flybase_version_converter.py \
    -input /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs.vcf \
    -output /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs_v6.vcf

python3 /scripts/v3/Flybase_version_converter.py \
    -input /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs.vcf \
    -output /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs_v6.vcf

cat /data/Australia/3L-3R_Imputed_Merged_Variant_calls/3L_IiIsYs_v6.vcf \
/data/Australia/3L-3R_Imputed_Merged_Variant_calls/3R_IiIsYs_v6.vcf \
<span class="pl-k">&gt;</span> /data/consensus/Australia.vcf

<span class="pl-c"><span class="pl-c">#</span># then convert from VCF to cons format</span>
python3 /scripts/v3/vcf2cons.py \
    /data/consensus/Australia.vcf \
    <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/Australia/Australia.cons.gz
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.2) Merge all files in *.cons format
                     </h3>
                     <a aria-label="Permalink: 2.2) Merge all files in *.cons format" class="anchor" href="#22-merge-all-files-in-cons-format" id="user-content-22-merge-all-files-in-cons-format">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Now, we merge the cons files and either draw randomly selected SNPs inside
                     <em>
                      In(3R)Payne
                     </em>
                     or which are in distance to the Inversion boundaries
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Merge and select Inv SNPs
python3 /scripts/merge_consensus.py \
    - /data/Inv.bed \
    -consensus /data/consensus/Zambia.consensus.gz,/data/consensus/DGN.consensus.gz,/data/consensus/USA.consensus.gz,/data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz,/data/Australia/Australia.cons.gz \
    | gzip &gt; /data/consensus/Inv.cons.gz

## Merge and select Non-Inv SNPs
python3 /scripts/merge_consensus.py \
    - /data/NoInv.bed \
    -consensus /data/consensus/Zambia.consensus.gz,/data/consensus/DGN.consensus.gz,/data/consensus/USA.consensus.gz,/data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz,/data/Australia/Australia.cons.gz \
    | gzip &gt; /data/consensus/NoInv.cons.gz

## convert merged cons to nexus format
python3 /scripts/v3/cons2nexus.py \
    -input /data/consensus/Inv.cons.gz \
    -population 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,362,363,364,365,366,367,368,369,370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,418,419,420,421,422,423,424,425,426,427,428,429,430,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,452,453,454,455,456,457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483 \
    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388,FR180,FR217,FR229,FR361,CK1,CO10N,CO13N,CO14,CO15N,CO16,CO1,CO2,CO4N,CO8N,CO9N,EZ25,EZ2,EZ9N,GA125,GA129,GA132,GA141,GA145,GA160,GA185,GU10,GU6,GU7,GU9,KN133N,KN20N,KN35,KR39,KR42,KR4N,KR7,KT6,NG10N,NG1N,NG3N,NG9,SP221,TZ10,TZ14,TZ8,UM118,UM37,UM526,ZO65,ZS11,ZS37,ZS5,CK2,ED2,ED3,ED5N,ED6N,ED10N,EZ5N,GA130,GA191,GU2,KN6,KN34,KT1,NG6N,NG7,RC1,RC5,RG2,RG3,RG4N,RG5,RG6N,RG7,RG8,RG9,RG10,RG11N,RG13N,RG15,RG18N,RG19,RG21N,RG22,RG24,RG25,RG28,RG32N,RG33,RG34,RG35,RG36,RG37N,RG38N,RG39,SP80,SP173,SP188,SP235,SP241,SP254,UG5N,UG7,UG19,UG28N,ZI91a,ZI261a,ZI268,ZI468,ZL130,FR14,FR70,FR151,FR310,RAL-301,RAL-303,RAL-304,RAL-306,RAL-307,RAL-313,RAL-315,RAL-324,RAL-335,RAL-357,RAL-358,RAL-360,RAL-362,RAL-365,RAL-375,RAL-379,RAL-380,RAL-391,RAL-399,RAL-427,RAL-437,RAL-486,RAL-513,RAL-517,RAL-555,RAL-639,RAL-705,RAL-707,RAL-714,RAL-730,RAL-732,RAL-765,RAL-774,RAL-786,RAL-799,RAL-820,RAL-852,Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81,F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,168_hot_R2,150_hot_R2,143_hot_R2,136_hot_R1,129_hot_R1,117_hot_R1,106_hot_R1,100_cold_R1,96_cold_R1,91_cold_R1,89_cold_R2,80_cold_R2,53_cold_R2,52_cold_R2,21_cold_R3,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94,IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \
    &gt; /data/consensus/Inv.nexus

python3 /scripts/v3/cons2nexus.py \
    -input /data/consensus/NoInv.cons.gz \
    -population 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,362,363,364,365,366,367,368,369,370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,418,419,420,421,422,423,424,425,426,427,428,429,430,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,452,453,454,455,456,457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483 \
    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388,FR180,FR217,FR229,FR361,CK1,CO10N,CO13N,CO14,CO15N,CO16,CO1,CO2,CO4N,CO8N,CO9N,EZ25,EZ2,EZ9N,GA125,GA129,GA132,GA141,GA145,GA160,GA185,GU10,GU6,GU7,GU9,KN133N,KN20N,KN35,KR39,KR42,KR4N,KR7,KT6,NG10N,NG1N,NG3N,NG9,SP221,TZ10,TZ14,TZ8,UM118,UM37,UM526,ZO65,ZS11,ZS37,ZS5,CK2,ED2,ED3,ED5N,ED6N,ED10N,EZ5N,GA130,GA191,GU2,KN6,KN34,KT1,NG6N,NG7,RC1,RC5,RG2,RG3,RG4N,RG5,RG6N,RG7,RG8,RG9,RG10,RG11N,RG13N,RG15,RG18N,RG19,RG21N,RG22,RG24,RG25,RG28,RG32N,RG33,RG34,RG35,RG36,RG37N,RG38N,RG39,SP80,SP173,SP188,SP235,SP241,SP254,UG5N,UG7,UG19,UG28N,ZI91a,ZI261a,ZI268,ZI468,ZL130,FR14,FR70,FR151,FR310,RAL-301,RAL-303,RAL-304,RAL-306,RAL-307,RAL-313,RAL-315,RAL-324,RAL-335,RAL-357,RAL-358,RAL-360,RAL-362,RAL-365,RAL-375,RAL-379,RAL-380,RAL-391,RAL-399,RAL-427,RAL-437,RAL-486,RAL-513,RAL-517,RAL-555,RAL-639,RAL-705,RAL-707,RAL-714,RAL-730,RAL-732,RAL-765,RAL-774,RAL-786,RAL-799,RAL-820,RAL-852,Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81,F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,168_hot_R2,150_hot_R2,143_hot_R2,136_hot_R1,129_hot_R1,117_hot_R1,106_hot_R1,100_cold_R1,96_cold_R1,91_cold_R1,89_cold_R2,80_cold_R2,53_cold_R2,52_cold_R2,21_cold_R3,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94,IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \
    &gt; /data/consensus/NoInv.nexus" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># Merge and select Inv SNPs</span>
python3 /scripts/merge_consensus.py \
    - /data/Inv.bed \
    -consensus /data/consensus/Zambia.consensus.gz,/data/consensus/DGN.consensus.gz,/data/consensus/USA.consensus.gz,/data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz,/data/Australia/Australia.cons.gz \
    <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/consensus/Inv.cons.gz

<span class="pl-c"><span class="pl-c">#</span># Merge and select Non-Inv SNPs</span>
python3 /scripts/merge_consensus.py \
    - /data/NoInv.bed \
    -consensus /data/consensus/Zambia.consensus.gz,/data/consensus/DGN.consensus.gz,/data/consensus/USA.consensus.gz,/data/consensus/Portugal.consensus.gz,/data/consensus/Sweden.consensus.gz,/data/Australia/Australia.cons.gz \
    <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/consensus/NoInv.cons.gz

<span class="pl-c"><span class="pl-c">#</span># convert merged cons to nexus format</span>
python3 /scripts/v3/cons2nexus.py \
    -input /data/consensus/Inv.cons.gz \
    -population 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,362,363,364,365,366,367,368,369,370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,418,419,420,421,422,423,424,425,426,427,428,429,430,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,452,453,454,455,456,457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483 \
    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388,FR180,FR217,FR229,FR361,CK1,CO10N,CO13N,CO14,CO15N,CO16,CO1,CO2,CO4N,CO8N,CO9N,EZ25,EZ2,EZ9N,GA125,GA129,GA132,GA141,GA145,GA160,GA185,GU10,GU6,GU7,GU9,KN133N,KN20N,KN35,KR39,KR42,KR4N,KR7,KT6,NG10N,NG1N,NG3N,NG9,SP221,TZ10,TZ14,TZ8,UM118,UM37,UM526,ZO65,ZS11,ZS37,ZS5,CK2,ED2,ED3,ED5N,ED6N,ED10N,EZ5N,GA130,GA191,GU2,KN6,KN34,KT1,NG6N,NG7,RC1,RC5,RG2,RG3,RG4N,RG5,RG6N,RG7,RG8,RG9,RG10,RG11N,RG13N,RG15,RG18N,RG19,RG21N,RG22,RG24,RG25,RG28,RG32N,RG33,RG34,RG35,RG36,RG37N,RG38N,RG39,SP80,SP173,SP188,SP235,SP241,SP254,UG5N,UG7,UG19,UG28N,ZI91a,ZI261a,ZI268,ZI468,ZL130,FR14,FR70,FR151,FR310,RAL-301,RAL-303,RAL-304,RAL-306,RAL-307,RAL-313,RAL-315,RAL-324,RAL-335,RAL-357,RAL-358,RAL-360,RAL-362,RAL-365,RAL-375,RAL-379,RAL-380,RAL-391,RAL-399,RAL-427,RAL-437,RAL-486,RAL-513,RAL-517,RAL-555,RAL-639,RAL-705,RAL-707,RAL-714,RAL-730,RAL-732,RAL-765,RAL-774,RAL-786,RAL-799,RAL-820,RAL-852,Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81,F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,168_hot_R2,150_hot_R2,143_hot_R2,136_hot_R1,129_hot_R1,117_hot_R1,106_hot_R1,100_cold_R1,96_cold_R1,91_cold_R1,89_cold_R2,80_cold_R2,53_cold_R2,52_cold_R2,21_cold_R3,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94,IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \
    <span class="pl-k">&gt;</span> /data/consensus/Inv.nexus

python3 /scripts/v3/cons2nexus.py \
    -input /data/consensus/NoInv.cons.gz \
    -population 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,362,363,364,365,366,367,368,369,370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,418,419,420,421,422,423,424,425,426,427,428,429,430,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,452,453,454,455,456,457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483 \
    -names ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388,FR180,FR217,FR229,FR361,CK1,CO10N,CO13N,CO14,CO15N,CO16,CO1,CO2,CO4N,CO8N,CO9N,EZ25,EZ2,EZ9N,GA125,GA129,GA132,GA141,GA145,GA160,GA185,GU10,GU6,GU7,GU9,KN133N,KN20N,KN35,KR39,KR42,KR4N,KR7,KT6,NG10N,NG1N,NG3N,NG9,SP221,TZ10,TZ14,TZ8,UM118,UM37,UM526,ZO65,ZS11,ZS37,ZS5,CK2,ED2,ED3,ED5N,ED6N,ED10N,EZ5N,GA130,GA191,GU2,KN6,KN34,KT1,NG6N,NG7,RC1,RC5,RG2,RG3,RG4N,RG5,RG6N,RG7,RG8,RG9,RG10,RG11N,RG13N,RG15,RG18N,RG19,RG21N,RG22,RG24,RG25,RG28,RG32N,RG33,RG34,RG35,RG36,RG37N,RG38N,RG39,SP80,SP173,SP188,SP235,SP241,SP254,UG5N,UG7,UG19,UG28N,ZI91a,ZI261a,ZI268,ZI468,ZL130,FR14,FR70,FR151,FR310,RAL-301,RAL-303,RAL-304,RAL-306,RAL-307,RAL-313,RAL-315,RAL-324,RAL-335,RAL-357,RAL-358,RAL-360,RAL-362,RAL-365,RAL-375,RAL-379,RAL-380,RAL-391,RAL-399,RAL-427,RAL-437,RAL-486,RAL-513,RAL-517,RAL-555,RAL-639,RAL-705,RAL-707,RAL-714,RAL-730,RAL-732,RAL-765,RAL-774,RAL-786,RAL-799,RAL-820,RAL-852,Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81,F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,168_hot_R2,150_hot_R2,143_hot_R2,136_hot_R1,129_hot_R1,117_hot_R1,106_hot_R1,100_cold_R1,96_cold_R1,91_cold_R1,89_cold_R2,80_cold_R2,53_cold_R2,52_cold_R2,21_cold_R3,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94,IiR-1,IiR-2,IiR-3,IiR-4,IiR-5,IiR-6,IiR-7,IiR-8,IiR-9,IiR-10,IiR-11,IiR-12,IiR-13,IiR-14,IiR-15,IiR-16,IiR-17,IiR-18,IiR-19,IsR-1,IsR-2,IsR-3,IsR-4,IsR-5,IsR-6,IsR-7,IsR-8,IsR-9,IsR-10,IsR-11,IsR-12,IsR-13,IsR-14,IsR-15,IsR-16,IsR-17,IsR-18,YSR-1,YSR-2,YSR-3,YSR-4,YSR-5,YSR-6,YSR-7,YSR-8,YSR-9,YSR-10,YSR-11,YSR-12,YSR-13,YSR-14,YSR-15,YSR-16,YSR-17,YSR-18 \
    <span class="pl-k">&gt;</span> /data/consensus/NoInv.nexus</pre>
                    </div>
                    <p dir="auto">
                     Using the NEXUS files, we plotted phylogenetic networks based on the Neighbor-Net inference method with Splits-Tree.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      3) Population Genetics Analysis
                     </h2>
                     <a aria-label="Permalink: 3) Population Genetics Analysis" class="anchor" href="#3-population-genetics-analysis" id="user-content-3-population-genetics-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      3.1) Genetic variation
                     </h3>
                     <a aria-label="Permalink: 3.1) Genetic variation" class="anchor" href="#31-genetic-variation" id="user-content-31-genetic-variation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Using
                     <a href="https:/vcftools.github.io/index.html">
                      VCFtools
                     </a>
                     we first calculated nucleotide diveristy (
                     <em>
                      
                     </em>
                     ) and Tajima's
                     <em>
                      D
                     </em>
                     as measures of genetic diversity in the different karyotypes
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
## make directory 
mkdir /data/PopGen

## loop through continents/countries
for continent in Africa Europe America Australia
do
    ## loop through karyotypes
    for karyotype in Inv Std All
    do
        ## loop through regions
        for file in data/${continent}_*_${karyotype}.samples
        do      
            ## isolate region name
            IFS='_ ' read -r -A array &lt;&lt;&lt; &quot;$file&quot;
            region=${array[2]}

            ## calculate 
            /scripts/vcftools_0.1.13/bin/vcftools \
                -keep data/${continent}_${region}_${karyotype}.samples \
                -window-pi 100000 \
                -window-pi-step 100000 \
                -gzvcf /data/consensus/${continent}.vcf.gz \
                -out /data/PopGen/${continent}_${region}_${karyotype}_100k.pi

            ## calculate Tajima's D
            /scripts/vcftools_0.1.13/bin/vcftools \
                -keep data/${continent}_${region}_${karyotype}.samples \
                -TajimaD 100000 \
                -gzvcf /data/consensus/${continent}.vcf.gz \
                -out /data/PopGen/${continent}_${region}_${karyotype}_100k
        
        done
    done

## then merge the results in a single large file for  and Tajima's D, plot averages by karyotype and Geographic origin and test for significant differences.
python3 /scripts/v3/merge-div.py \
    -input /data/PopGen/Africa_Zambia_Inv_100k.pi,/data/PopGen/Africa_Zambia_Std_100k.pi,/data/PopGen/Africa_Zambia_All_100k.pi,/data/PopGen/Europe_Portugal_Inv_100k.pi,/data/PopGen/Europe_Portugal_Std_100k.pi,/data/PopGen/Europe_Portugal_All_100k.pi,/data/PopGen/Europe_Sweden_Std_100k.pi,/data/PopGen/America_Florida_Inv_100k.pi,/data/PopGen/America_Florida_Std_100k.pi,/data/PopGen/America_Florida_All_100k.pi,/data/PopGen/America_Maine_Std_100k.pi,/data/PopGen/Australia_Queensland_Inv_100k.pi,/data/PopGen/Australia_Queensland_Std_100k.pi,/data/PopGen/Australia_Queensland_All_100k.pi,/data/PopGen/Australia_Victoria_Std_100k.pi \
    -names Zambia-Inv,Zambia-Std,Zambia-All,Portugal-Inv,Portugal-Std,Portugal-All,Sweden-Std,Florida-Inv,Florida-Std,Florida-All,Maine-Std,Queensland-Inv,Queensland-Std,Queensland-All,Victoria-Std \
    &gt; /data/PopGen/AllData.pi

Rscript /scripts/statNplot.R /data/PopGen/AllData.pi

python3 /scripts/v3/merge-div.py \
    -input /data/PopGen/Africa_Zambia_Inv_100k.Tajima.D,/data/PopGen/Africa_Zambia_Std_100k.Tajima.D,/data/PopGen/Africa_Zambia_All_100k.Tajima.D,/data/PopGen/Europe_Portugal_Inv_100k.Tajima.D,/data/PopGen/Europe_Portugal_Std_100k.Tajima.D,/data/PopGen/Europe_Portugal_All_100k.Tajima.D,/data/PopGen/Europe_Sweden_Std_100k.Tajima.D,/data/PopGen/America_Florida_Inv_100k.Tajima.D,/data/PopGen/America_Florida_Std_100k.Tajima.D,/data/PopGen/America_Florida_All_100k.Tajima.D,/data/PopGen/America_Maine_Std_100k.Tajima.D,/data/PopGen/Australia_Queensland_Inv_100k.Tajima.D,/data/PopGen/Australia_Queensland_Std_100k.Tajima.D,/data/PopGen/Australia_Queensland_All_100k.Tajima.D,/data/PopGen/Australia_Victoria_Std_100k.Tajima.D \
    -names Zambia-Inv,Zambia-Std,Zambia-All,Portugal-Inv,Portugal-Std,Portugal-All,Sweden-Std,Florida-Inv,Florida-Std,Florida-All,Maine-Std,Queensland-Inv,Queensland-Std,Queensland-All,Victoria-Std \
    &gt; /data/PopGen/AllData.Tajima.D

Rscript /scripts/statNplot.R /data/PopGen/AllData.Tajima.D" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># make directory </span>
mkdir /data/PopGen

<span class="pl-c"><span class="pl-c">#</span># loop through continents/countries</span>
<span class="pl-k">for</span> <span class="pl-smi">continent</span> <span class="pl-k">in</span> Africa Europe America Australia
<span class="pl-k">do</span>
    <span class="pl-c"><span class="pl-c">#</span># loop through karyotypes</span>
    <span class="pl-k">for</span> <span class="pl-smi">karyotype</span> <span class="pl-k">in</span> Inv Std All
    <span class="pl-k">do</span>
        <span class="pl-c"><span class="pl-c">#</span># loop through regions</span>
        <span class="pl-k">for</span> <span class="pl-smi">file</span> <span class="pl-k">in</span> data/<span class="pl-smi">${continent}</span>_<span class="pl-k">*</span>_<span class="pl-smi">${karyotype}</span>.samples
        <span class="pl-k">do</span>      
            <span class="pl-c"><span class="pl-c">#</span># isolate region name</span>
            IFS=<span class="pl-s"><span class="pl-pds">'</span>_ <span class="pl-pds">'</span></span> <span class="pl-c1">read</span> -r -A array <span class="pl-k">&lt;&lt;&lt;</span> <span class="pl-s"><span class="pl-pds">"</span>$file<span class="pl-pds">"</span></span>
            region=<span class="pl-smi">${array[2]}</span>

            <span class="pl-c"><span class="pl-c">#</span># calculate </span>
            /scripts/vcftools_0.1.13/bin/vcftools \
                -keep data/<span class="pl-smi">${continent}</span>_<span class="pl-smi">${region}</span>_<span class="pl-smi">${karyotype}</span>.samples \
                -window-pi 100000 \
                -window-pi-step 100000 \
                -gzvcf /data/consensus/<span class="pl-smi">${continent}</span>.vcf.gz \
                -out /data/PopGen/<span class="pl-smi">${continent}</span>_<span class="pl-smi">${region}</span>_<span class="pl-smi">${karyotype}</span>_100k.pi

            <span class="pl-c"><span class="pl-c">#</span># calculate Tajima's D</span>
            /scripts/vcftools_0.1.13/bin/vcftools \
                -keep data/<span class="pl-smi">${continent}</span>_<span class="pl-smi">${region}</span>_<span class="pl-smi">${karyotype}</span>.samples \
                -TajimaD 100000 \
                -gzvcf /data/consensus/<span class="pl-smi">${continent}</span>.vcf.gz \
                -out /data/PopGen/<span class="pl-smi">${continent}</span>_<span class="pl-smi">${region}</span>_<span class="pl-smi">${karyotype}</span>_100k
        
        <span class="pl-k">done</span>
    <span class="pl-k">done</span>

<span class="pl-c"><span class="pl-c">#</span># then merge the results in a single large file for  and Tajima's D, plot averages by karyotype and Geographic origin and test for significant differences.</span>
python3 /scripts/v3/merge-div.py \
    -input /data/PopGen/Africa_Zambia_Inv_100k.pi,/data/PopGen/Africa_Zambia_Std_100k.pi,/data/PopGen/Africa_Zambia_All_100k.pi,/data/PopGen/Europe_Portugal_Inv_100k.pi,/data/PopGen/Europe_Portugal_Std_100k.pi,/data/PopGen/Europe_Portugal_All_100k.pi,/data/PopGen/Europe_Sweden_Std_100k.pi,/data/PopGen/America_Florida_Inv_100k.pi,/data/PopGen/America_Florida_Std_100k.pi,/data/PopGen/America_Florida_All_100k.pi,/data/PopGen/America_Maine_Std_100k.pi,/data/PopGen/Australia_Queensland_Inv_100k.pi,/data/PopGen/Australia_Queensland_Std_100k.pi,/data/PopGen/Australia_Queensland_All_100k.pi,/data/PopGen/Australia_Victoria_Std_100k.pi \
    -names Zambia-Inv,Zambia-Std,Zambia-All,Portugal-Inv,Portugal-Std,Portugal-All,Sweden-Std,Florida-Inv,Florida-Std,Florida-All,Maine-Std,Queensland-Inv,Queensland-Std,Queensland-All,Victoria-Std \
    <span class="pl-k">&gt;</span> /data/PopGen/AllData.pi

Rscript /scripts/statNplot.R /data/PopGen/AllData.pi

python3 /scripts/v3/merge-div.py \
    -input /data/PopGen/Africa_Zambia_Inv_100k.Tajima.D,/data/PopGen/Africa_Zambia_Std_100k.Tajima.D,/data/PopGen/Africa_Zambia_All_100k.Tajima.D,/data/PopGen/Europe_Portugal_Inv_100k.Tajima.D,/data/PopGen/Europe_Portugal_Std_100k.Tajima.D,/data/PopGen/Europe_Portugal_All_100k.Tajima.D,/data/PopGen/Europe_Sweden_Std_100k.Tajima.D,/data/PopGen/America_Florida_Inv_100k.Tajima.D,/data/PopGen/America_Florida_Std_100k.Tajima.D,/data/PopGen/America_Florida_All_100k.Tajima.D,/data/PopGen/America_Maine_Std_100k.Tajima.D,/data/PopGen/Australia_Queensland_Inv_100k.Tajima.D,/data/PopGen/Australia_Queensland_Std_100k.Tajima.D,/data/PopGen/Australia_Queensland_All_100k.Tajima.D,/data/PopGen/Australia_Victoria_Std_100k.Tajima.D \
    -names Zambia-Inv,Zambia-Std,Zambia-All,Portugal-Inv,Portugal-Std,Portugal-All,Sweden-Std,Florida-Inv,Florida-Std,Florida-All,Maine-Std,Queensland-Inv,Queensland-Std,Queensland-All,Victoria-Std \
    <span class="pl-k">&gt;</span> /data/PopGen/AllData.Tajima.D

Rscript /scripts/statNplot.R /data/PopGen/AllData.Tajima.D</pre>
                    </div>
                    <p dir="auto">
                     In addition, I plotted the values of
                     <em>
                      
                     </em>
                     and Tajima's
                     <em>
                      D
                     </em>
                     as lineplots in R
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="### plot  for Inverted and Standard Chromosomes
library(tidyverse)

DATA=read.table(&quot;/data/PopGen/AllData.pi&quot;,header=T)

DATA.3R=subset(DATA,DATA$C==&quot;3R&quot;)
DATA.3R=subset(DATA,DATA$InvStatus!=&quot;All&quot;)
DATA.3R=subset(DATA.3R,DATA.3R$Origin %in% c(&quot;Florida&quot;,&quot;Portugal&quot;,&quot;Zambia&quot;,&quot;Queensland&quot;))
DATA.3R$Origin &lt;- factor(DATA.3R$Origin,levels=c(&quot;Zambia&quot;,&quot;Portugal&quot;,&quot;Florida&quot;,&quot;Queensland&quot;))

P=ggplot(DATA.3R,aes(
    x=P,
    y=pi,
    col=InvStatus))+
  geom_line(lwd=1.5)+
  geom_rect(mapping = aes(
        xmin=16432209,
        xmax=24744010,
        ymin=0,
        ymax=(max(pi))),
    color=&quot;black&quot;,
    alpha=0,
    lwd=0.5,
    lty=2)+
  facet_grid(Origin~ .)+
  theme_classic()

ggsave(&quot;/data/PopGen/AllData.pi.pdf&quot;,
    P,
    width=12,
    height=8)

### plot Tajima's D for Inverted and Standard Chromosomes
library(tidyverse)

DATA=read.table(&quot;/data/PopGen/AllData.Tajima.D&quot;,header=T)

DATA.3R=subset(DATA,DATA$C==&quot;3R&quot;)
DATA.3R=subset(DATA.3R,DATA.3R$Origin %in% c(&quot;Florida&quot;,&quot;Portugal&quot;,&quot;Zambia&quot;,&quot;Queensland&quot;))
DATA.3R$Origin &lt;- factor(DATA.3R$Origin,levels=c(&quot;Zambia&quot;,&quot;Portugal&quot;,&quot;Florida&quot;,&quot;Queensland&quot;))

P=ggplot(DATA.3R,aes(
    x=P,
    y=pi,
    col=InvStatus))+
  geom_line(lwd=1.5)+
  geom_rect(mapping = aes(
        xmin=16432209,
        xmax=24744010,
        ymin=0,
        ymax=(max(pi))),
    color=&quot;black&quot;,
    alpha=0,
    lwd=0.5,
    lty=2)+
  facet_grid(Origin~ .)+
  theme_classic()+
  geom_hline(yintercept=0)

ggsave(&quot;/data/PopGen/AllData.Tajima.D.pdf&quot;,
    P,
    width=12,
    height=8)" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>## plot  for Inverted and Standard Chromosomes</span>
library(<span class="pl-smi">tidyverse</span>)

<span class="pl-v">DATA</span><span class="pl-k">=</span>read.table(<span class="pl-s"><span class="pl-pds">"</span>/data/PopGen/AllData.pi<span class="pl-pds">"</span></span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">T</span>)

<span class="pl-v">DATA.3R</span><span class="pl-k">=</span>subset(<span class="pl-smi">DATA</span>,<span class="pl-smi">DATA</span><span class="pl-k">$</span><span class="pl-smi">C</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>3R<span class="pl-pds">"</span></span>)
<span class="pl-v">DATA.3R</span><span class="pl-k">=</span>subset(<span class="pl-smi">DATA</span>,<span class="pl-smi">DATA</span><span class="pl-k">$</span><span class="pl-smi">InvStatus</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">"</span>All<span class="pl-pds">"</span></span>)
<span class="pl-v">DATA.3R</span><span class="pl-k">=</span>subset(<span class="pl-smi">DATA.3R</span>,<span class="pl-smi">DATA.3R</span><span class="pl-k">$</span><span class="pl-smi">Origin</span> <span class="pl-k">%in%</span> c(<span class="pl-s"><span class="pl-pds">"</span>Florida<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Portugal<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Zambia<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Queensland<span class="pl-pds">"</span></span>))
<span class="pl-smi">DATA.3R</span><span class="pl-k">$</span><span class="pl-smi">Origin</span> <span class="pl-k">&lt;-</span> <span class="pl-k">factor</span>(<span class="pl-smi">DATA.3R</span><span class="pl-k">$</span><span class="pl-smi">Origin</span>,<span class="pl-v">levels</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>Zambia<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Portugal<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Florida<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Queensland<span class="pl-pds">"</span></span>))

<span class="pl-v">P</span><span class="pl-k">=</span>ggplot(<span class="pl-smi">DATA.3R</span>,aes(
    <span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">P</span>,
    <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-c1">pi</span>,
    <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">InvStatus</span>))<span class="pl-k">+</span>
  geom_line(<span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">1.5</span>)<span class="pl-k">+</span>
  geom_rect(<span class="pl-v">mapping</span> <span class="pl-k">=</span> aes(
        <span class="pl-v">xmin</span><span class="pl-k">=</span><span class="pl-c1">16432209</span>,
        <span class="pl-v">xmax</span><span class="pl-k">=</span><span class="pl-c1">24744010</span>,
        <span class="pl-v">ymin</span><span class="pl-k">=</span><span class="pl-c1">0</span>,
        <span class="pl-v">ymax</span><span class="pl-k">=</span>(max(<span class="pl-c1">pi</span>))),
    <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>,
    <span class="pl-v">alpha</span><span class="pl-k">=</span><span class="pl-c1">0</span>,
    <span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">0.5</span>,
    <span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>)<span class="pl-k">+</span>
  facet_grid(<span class="pl-smi">Origin</span><span class="pl-k">~</span> <span class="pl-smi">.</span>)<span class="pl-k">+</span>
  theme_classic()

ggsave(<span class="pl-s"><span class="pl-pds">"</span>/data/PopGen/AllData.pi.pdf<span class="pl-pds">"</span></span>,
    <span class="pl-smi">P</span>,
    <span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">12</span>,
    <span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)

<span class="pl-c"><span class="pl-c">#</span>## plot Tajima's D for Inverted and Standard Chromosomes</span>
library(<span class="pl-smi">tidyverse</span>)

<span class="pl-v">DATA</span><span class="pl-k">=</span>read.table(<span class="pl-s"><span class="pl-pds">"</span>/data/PopGen/AllData.Tajima.D<span class="pl-pds">"</span></span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">T</span>)

<span class="pl-v">DATA.3R</span><span class="pl-k">=</span>subset(<span class="pl-smi">DATA</span>,<span class="pl-smi">DATA</span><span class="pl-k">$</span><span class="pl-smi">C</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>3R<span class="pl-pds">"</span></span>)
<span class="pl-v">DATA.3R</span><span class="pl-k">=</span>subset(<span class="pl-smi">DATA.3R</span>,<span class="pl-smi">DATA.3R</span><span class="pl-k">$</span><span class="pl-smi">Origin</span> <span class="pl-k">%in%</span> c(<span class="pl-s"><span class="pl-pds">"</span>Florida<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Portugal<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Zambia<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Queensland<span class="pl-pds">"</span></span>))
<span class="pl-smi">DATA.3R</span><span class="pl-k">$</span><span class="pl-smi">Origin</span> <span class="pl-k">&lt;-</span> <span class="pl-k">factor</span>(<span class="pl-smi">DATA.3R</span><span class="pl-k">$</span><span class="pl-smi">Origin</span>,<span class="pl-v">levels</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>Zambia<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Portugal<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Florida<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Queensland<span class="pl-pds">"</span></span>))

<span class="pl-v">P</span><span class="pl-k">=</span>ggplot(<span class="pl-smi">DATA.3R</span>,aes(
    <span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">P</span>,
    <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-c1">pi</span>,
    <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">InvStatus</span>))<span class="pl-k">+</span>
  geom_line(<span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">1.5</span>)<span class="pl-k">+</span>
  geom_rect(<span class="pl-v">mapping</span> <span class="pl-k">=</span> aes(
        <span class="pl-v">xmin</span><span class="pl-k">=</span><span class="pl-c1">16432209</span>,
        <span class="pl-v">xmax</span><span class="pl-k">=</span><span class="pl-c1">24744010</span>,
        <span class="pl-v">ymin</span><span class="pl-k">=</span><span class="pl-c1">0</span>,
        <span class="pl-v">ymax</span><span class="pl-k">=</span>(max(<span class="pl-c1">pi</span>))),
    <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>,
    <span class="pl-v">alpha</span><span class="pl-k">=</span><span class="pl-c1">0</span>,
    <span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">0.5</span>,
    <span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>)<span class="pl-k">+</span>
  facet_grid(<span class="pl-smi">Origin</span><span class="pl-k">~</span> <span class="pl-smi">.</span>)<span class="pl-k">+</span>
  theme_classic()<span class="pl-k">+</span>
  geom_hline(<span class="pl-v">yintercept</span><span class="pl-k">=</span><span class="pl-c1">0</span>)

ggsave(<span class="pl-s"><span class="pl-pds">"</span>/data/PopGen/AllData.Tajima.D.pdf<span class="pl-pds">"</span></span>,
    <span class="pl-smi">P</span>,
    <span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">12</span>,
    <span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      3.2) Genetic differentiation
                     </h3>
                     <a aria-label="Permalink: 3.2) Genetic differentiation" class="anchor" href="#32-genetic-differentiation" id="user-content-32-genetic-differentiation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Again using
                     <a href="https:/vcftools.github.io/index.html">
                      VCFtools
                     </a>
                     we calculated pairwise FST among karyotypes in different geographic regions both for single SNPs and in 100kb windows. Then, we merged the FST values, plotted the averages with respect to karyotype and tested for significant differences with respect to the karyotype and origin. Moreover, we tested if window-wise FST is correlated within the inverted genomic region.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='
## loop through samples
continent=( "Africa" "Europe" "Europe" "Europe" "America" "America" "America" "Australia" "Australia" "Australia" )
FST=( "ZI-ZS" "PI-PS" "PI-SS" "PS-SS" "FI-FS" "FI-MS" "FS-MS" "II-IS" "II-YS" "IS-YS" )
Pop1=( "Africa_Zambia_Inv" "Europe_Portugal_Inv" "Europe_Portugal_Inv" "Europe_Portugal_Std" "America_Florida_Inv" "America_Florida_Inv" "America_Florida_Std" "Australia_Queensland_Inv" "Australia_Queensland_Inv" "Australia_Queensland_Std" )
Pop2=( "Africa_Zambia_Std" "Europe_Portugal_Std" "Europe_Sweden_Std" "Europe_Sweden_Std" "America_Florida_Std" "America_Maine_Std" "America_Maine_Std" "Australia_Queensland_Std" "Australia_Victoria_Std" "Australia_Victoria_Std" )

for index in ${!FST[@]}
do
    ## window-wise FST
    /scripts/vcftools_0.1.13/cpp/vcftools \
        -gzvcf /data/consensus/${continent[index]}.vcf.gz \
        -weir-fst-pop /data/${Pop1[index]}.samples \
        -weir-fst-pop /data/${Pop2[index]}.samples \
        -fst-window-size 100000 \
        -fst-window-step 100000 \
        -out /data/PopGen/${FST[index]}
    
    ## SNP-wise FST
    /scripts/vcftools_0.1.13/cpp/vcftools \
        -gzvcf /data/consensus/${continent[index]}.vcf.gz \
        -weir-fst-pop /data/${Pop1[index]}.samples \
        -weir-fst-pop /data/${Pop2[index]}.samples \
        -out /data/PopGen/${FST[index]}
done

## now merge all FST values
python3 /scripts/v3/merge-diff.py \
    -input /data/PopGen/ZI-ZS.windowed.weir.fst,/data/PopGen/PI-PS.windowed.weir.fst,/data/PopGen/PI-SS.windowed.weir.fst,/data/PopGen/PS-SS.windowed.weir.fst,/data/PopGen/FI-FS.windowed.weir.fst,/data/PopGen/FI-MS.windowed.weir.fst,/data/PopGen/FS-MS.windowed.weir.fst,/data/PopGen/II-IS.windowed.weir.fst,/data/PopGen/II-YS.windowed.weir.fst,/data/PopGen/IS-YS.windowed.weir.fst \
    -names ZI-ZS,PI-PS,PI-SS,PS-SS,FI-FS,FI-MS,FS-MS,II-IS,II-YS,IS-YS \
 &gt; /data/PopGen/AllData.fst

## plot average FSTs per karyotype and geography, test for significant differences, and investigate if windowed FST within the inverted regions is correlated among geographic regions 
Rscript /scripts/statNcompareFST.R /data/PopGen/AllData.fst
' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># loop through samples</span>
continent=( <span class="pl-s"><span class="pl-pds">"</span>Africa<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia<span class="pl-pds">"</span></span> )
FST=( <span class="pl-s"><span class="pl-pds">"</span>ZI-ZS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>PI-PS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>PI-SS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>PS-SS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>FI-FS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>FI-MS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>FS-MS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>II-IS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>II-YS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>IS-YS<span class="pl-pds">"</span></span> )
Pop1=( <span class="pl-s"><span class="pl-pds">"</span>Africa_Zambia_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe_Portugal_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe_Portugal_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe_Portugal_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Florida_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Florida_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Florida_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia_Queensland_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia_Queensland_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia_Queensland_Std<span class="pl-pds">"</span></span> )
Pop2=( <span class="pl-s"><span class="pl-pds">"</span>Africa_Zambia_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe_Portugal_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe_Sweden_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe_Sweden_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Florida_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Maine_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Maine_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia_Queensland_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia_Victoria_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia_Victoria_Std<span class="pl-pds">"</span></span> )

<span class="pl-k">for</span> <span class="pl-smi">index</span> <span class="pl-k">in</span> <span class="pl-smi">${<span class="pl-k">!</span>FST[@]}</span>
<span class="pl-k">do</span>
    <span class="pl-c"><span class="pl-c">#</span># window-wise FST</span>
    /scripts/vcftools_0.1.13/cpp/vcftools \
        -gzvcf /data/consensus/<span class="pl-smi">${continent[index]}</span>.vcf.gz \
        -weir-fst-pop /data/<span class="pl-smi">${Pop1[index]}</span>.samples \
        -weir-fst-pop /data/<span class="pl-smi">${Pop2[index]}</span>.samples \
        -fst-window-size 100000 \
        -fst-window-step 100000 \
        -out /data/PopGen/<span class="pl-smi">${FST[index]}</span>
    
    <span class="pl-c"><span class="pl-c">#</span># SNP-wise FST</span>
    /scripts/vcftools_0.1.13/cpp/vcftools \
        -gzvcf /data/consensus/<span class="pl-smi">${continent[index]}</span>.vcf.gz \
        -weir-fst-pop /data/<span class="pl-smi">${Pop1[index]}</span>.samples \
        -weir-fst-pop /data/<span class="pl-smi">${Pop2[index]}</span>.samples \
        -out /data/PopGen/<span class="pl-smi">${FST[index]}</span>
<span class="pl-k">done</span>

<span class="pl-c"><span class="pl-c">#</span># now merge all FST values</span>
python3 /scripts/v3/merge-diff.py \
    -input /data/PopGen/ZI-ZS.windowed.weir.fst,/data/PopGen/PI-PS.windowed.weir.fst,/data/PopGen/PI-SS.windowed.weir.fst,/data/PopGen/PS-SS.windowed.weir.fst,/data/PopGen/FI-FS.windowed.weir.fst,/data/PopGen/FI-MS.windowed.weir.fst,/data/PopGen/FS-MS.windowed.weir.fst,/data/PopGen/II-IS.windowed.weir.fst,/data/PopGen/II-YS.windowed.weir.fst,/data/PopGen/IS-YS.windowed.weir.fst \
    -names ZI-ZS,PI-PS,PI-SS,PS-SS,FI-FS,FI-MS,FS-MS,II-IS,II-YS,IS-YS \
 <span class="pl-k">&gt;</span> /data/PopGen/AllData.fst

<span class="pl-c"><span class="pl-c">#</span># plot average FSTs per karyotype and geography, test for significant differences, and investigate if windowed FST within the inverted regions is correlated among geographic regions </span>
Rscript /scripts/statNcompareFST.R /data/PopGen/AllData.fst
</pre>
                    </div>
                    <p dir="auto">
                     Next, we also calculated SNP-wise FST and used
                     <a href="http:/pcingola.github.io/SnpEff/">
                      SnpEFF
                     </a>
                     to annotate the VCF files.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## loop through continents/countries
for continent in Africa Europe America Australia
do   
    ## annotate with SNPeff
    java -Xmx4g -jar /scripts/snpEff/snpEff.jar \
        BDGP6.82 \
        /data/consensus/${continent}.vcf.gz \
        | gzip &gt; /data/consensus/${continent}_ann.vcf.gz
done

## Isolate positions with FST &gt;= 0.9 and append annotation
continent=( &quot;Africa&quot; &quot;Europe&quot; &quot;Europe&quot; &quot;Europe&quot; &quot;America&quot; &quot;America&quot; &quot;America&quot; &quot;Australia&quot; &quot;Australia&quot; &quot;Australia&quot; )
FST=( &quot;ZI-ZS&quot; &quot;PI-PS&quot; &quot;PI-SS&quot; &quot;PS-SS&quot; &quot;FI-FS&quot; &quot;FI-MS&quot; &quot;FS-MS&quot; &quot;II-IS&quot; &quot;II-YS&quot; &quot;IS-YS&quot; )

for index in ${!FST[@]}
do  
    ## append to full dataset
    python3 /scripts/v3/AppendAnnotFromVCF.py \
        /data/consensus/${continent[index]}_ann.vcf.gz \
        /data/PopGen/${FST[index]}.weir.fst \
        | sort -k1,1 -k2,2 \
        | &gt; /data/PopGen/${FST[index]}_ann.weir.fst
    
    ## additionally isolate candidates
    awk '$NF&gt;=0.9 &amp;&amp; $NF!=&quot;nan&quot;' /data/PopGen/${FST[index]}.weir.fst \
    | python3 /scripts/v3/AppendAnnotFromVCF.py \
        /data/consensus/${continent[index]}_ann.vcf.gz \
        - \
        | sort -k1,1 -k2,2 \
        | &gt; /data/PopGen/${FST[index]}_cand_ann.weir.fst
done" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># loop through continents/countries</span>
<span class="pl-k">for</span> <span class="pl-smi">continent</span> <span class="pl-k">in</span> Africa Europe America Australia
<span class="pl-k">do</span>   
    <span class="pl-c"><span class="pl-c">#</span># annotate with SNPeff</span>
    java -Xmx4g -jar /scripts/snpEff/snpEff.jar \
        BDGP6.82 \
        /data/consensus/<span class="pl-smi">${continent}</span>.vcf.gz \
        <span class="pl-k">|</span> gzip <span class="pl-k">&gt;</span> /data/consensus/<span class="pl-smi">${continent}</span>_ann.vcf.gz
<span class="pl-k">done</span>

<span class="pl-c"><span class="pl-c">#</span># Isolate positions with FST &gt;= 0.9 and append annotation</span>
continent=( <span class="pl-s"><span class="pl-pds">"</span>Africa<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia<span class="pl-pds">"</span></span> )
FST=( <span class="pl-s"><span class="pl-pds">"</span>ZI-ZS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>PI-PS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>PI-SS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>PS-SS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>FI-FS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>FI-MS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>FS-MS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>II-IS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>II-YS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>IS-YS<span class="pl-pds">"</span></span> )

<span class="pl-k">for</span> <span class="pl-smi">index</span> <span class="pl-k">in</span> <span class="pl-smi">${<span class="pl-k">!</span>FST[@]}</span>
<span class="pl-k">do</span>  
    <span class="pl-c"><span class="pl-c">#</span># append to full dataset</span>
    python3 /scripts/v3/AppendAnnotFromVCF.py \
        /data/consensus/<span class="pl-smi">${continent[index]}</span>_ann.vcf.gz \
        /data/PopGen/<span class="pl-smi">${FST[index]}</span>.weir.fst \
        <span class="pl-k">|</span> sort -k1,1 -k2,2 \
        <span class="pl-k">|</span> <span class="pl-k">&gt;</span> /data/PopGen/<span class="pl-smi">${FST[index]}</span>_ann.weir.fst
    
    <span class="pl-c"><span class="pl-c">#</span># additionally isolate candidates</span>
    awk <span class="pl-s"><span class="pl-pds">'</span>$NF&gt;=0.9 &amp;&amp; $NF!="nan"<span class="pl-pds">'</span></span> /data/PopGen/<span class="pl-smi">${FST[index]}</span>.weir.fst \
    <span class="pl-k">|</span> python3 /scripts/v3/AppendAnnotFromVCF.py \
        /data/consensus/<span class="pl-smi">${continent[index]}</span>_ann.vcf.gz \
        - \
        <span class="pl-k">|</span> sort -k1,1 -k2,2 \
        <span class="pl-k">|</span> <span class="pl-k">&gt;</span> /data/PopGen/<span class="pl-smi">${FST[index]}</span>_cand_ann.weir.fst
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Now test in R if there is a significant overlap among candidate SNPs and Genes across continents and plot the results as barplots using the
                     <a href="https://cran.r-project.org/web/packages/SuperExactTest/index.html" rel="nofollow">
                      SuperExactTest
                     </a>
                     package
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## first SNP-wise comparison

library(SuperExactTest)

## read background and idenitfy SNPs inside the inversion
SEu.f=na.omit(read.table(gzfile('/data/PopGen/PI-PS_ann.weir.fst'),header=F,na.string='nan',sep='\t', quote = ''))
SEu.f.in3rp=subset(SEu.f,SEu.f[,1]=='3R' &amp; SEu.f[,2]&gt; 16432209 &amp; SEu.f[,2] &lt; 24744010)
SAf.f=na.omit(read.table(gzfile('/data/PopGen/ZI-ZS_ann.weir.fst'),header=F,na.string='nan',sep='\t', quote = ''))
SAf.f.in3rp=subset(SAf.f,SAf.f[,1]=='3R' &amp; SAf.f[,2]&gt; 16432209 &amp; SAf.f[,2] &lt; 24744010)
SAu.f=na.omit(read.table(gzfile('/data/PopGen/II-IS_ann.weir.fst'),header=F,na.string='nan',sep='\t', quote = ''))
SAu.f.in3rp=subset(SAu.f,SAu.f[,1]=='3R' &amp; SAu.f[,2]&gt; 16432209 &amp; SAu.f[,2] &lt; 24744010)
SNA.f=na.omit(read.table(gzfile('/data/PopGen/FI-FS_ann.weir.fst'),header=F,na.string='nan',sep='\t', quote = ''))
SNA.f.in3rp=subset(SNA.f,SNA.f[,1]=='3R' &amp; SNA.f[,2]&gt; 16432209 &amp; SNA.f[,2] &lt; 24744010)

## now for genes
SAf.f.genes=unique(SAf.f$V5[sapply(SAf.f$V5,FUN=function(x){x!=''})])
SEu.f.genes=unique(SEu.f$V5[sapply(SEu.f$V5,FUN=function(x){x!=''})])
SAu.f.genes=unique(SAu.f$V5[sapply(SAu.f$V5,FUN=function(x){x!=''})])
SNA.f.genes=unique(SNA.f$V5[sapply(SNA.f$V5,FUN=function(x){x!=''})])

SAf.f.in3rp.genes=unique(SAf.f.in3rp$V5[sapply(SAf.f.in3rp$V5,FUN=function(x){x!=''})])
SEu.f.in3rp.genes=unique(SEu.f.in3rp$V5[sapply(SEu.f.in3rp$V5,FUN=function(x){x!=''})])
SAu.f.in3rp.genes=unique(SAu.f.in3rp$V5[sapply(SAu.f.in3rp$V5,FUN=function(x){x!=''})])
SNA.f.in3rp.genes=unique(SNA.f.in3rp$V5[sapply(SNA.f.in3rp$V5,FUN=function(x){x!=''})])

## now for SNPs
SEu.f=unique(paste(SEu.f[,1],SEu.f[,2],sep='_'))
SEu.f.in3rp=unique(paste(SEu.f.in3rp[,1],SEu.f.in3rp[,2],sep='_'))
SAf.f=unique(paste(SAf.f[,1],SAf.f[,2],sep='_'))
SAf.f.in3rp=unique(paste(SAf.f.in3rp[,1],SAf.f.in3rp[,2],sep='_'))
SAu.f=unique(paste(SAu.f[,1],SAu.f[,2],sep='_'))
SAu.f.in3rp=unique(paste(SAu.f.in3rp[,1],SAu.f.in3rp[,2],sep='_'))
SNA.f=unique(paste(SNA.f[,1],SNA.f[,2],sep='_'))
SNA.f.in3rp=unique(paste(SNA.f.in3rp[,1],SNA.f.in3rp[,2],sep='_'))

# get overlapping SNPs &amp; genes
FullnoAu=Reduce(intersect,list(SEu.f,SAf.f,SNA.f))
FullnoAu.in3p=Reduce(intersect,list(SEu.f.in3rp,SAf.f.in3rp,SNA.f.in3rp))

Full=Reduce(intersect,list(SEu.f,SAf.f,SNA.f,SAu.f))
Full.in3p=Reduce(intersect,list(SEu.f.in3rp,SAf.f.in3rp,SNA.f.in3rp,SAu.f.in3rp))

FullnoAu.genes=Reduce(intersect,list(SAf.f.genes,SEu.f.genes,SNA.f.genes))
FullnoAu.in3p.genes=Reduce(intersect,list(SEu.f.in3rp.genes,SAf.f.in3rp.genes,SNA.f.in3rp.genes))

Full.genes=Reduce(intersect,list(SEu.f.genes,SAf.f.genes,SNA.f.genes,SAu.f.genes))
Full.in3p.genes=Reduce(intersect,list(SEu.f.in3rp.genes,SAf.f.in3rp.genes,SNA.f.in3rp.genes,SAu.f.in3rp.genes))

dir.create('/data/PopGen/candidates')

# candidates
SEu.c=read.table(paste('/data/PopGen/PI-PS_cand_ann.weir.fst',sep=''),sep='\t', quote = '')
SEu.c.in3rp=subset(SEu.c,SEu.c[,1]=='3R' &amp; SEu.c[,2]&gt; 16432209 &amp; SEu.c[,2] &lt; 24744010)
SEu.c.genes=unique(SEu.c$V5[sapply(SEu.c$V5,FUN=function(x){x!=''})])
SEu.c.in3rp.genes=unique(SEu.c.in3rp$V5[sapply(SEu.c.in3rp$V5,FUN=function(x){x!=''})])
SEu.c=unique(paste(SEu.c[,1],SEu.c[,2],sep='_'))
SEu.c.in3rp=unique(paste(SEu.c.in3rp[,1],SEu.c.in3rp[,2],sep='_'))
SEu.c.ovNoAu.in3rp=SEu.c[SEu.c %in% FullnoAu.in3p]
SEu.c.ov.in3rp=SEu.c[SEu.c %in% Full.in3p]
SEu.c.ovNoAu=SEu.c[SEu.c %in% FullnoAu]
SEu.c.ov=SEu.c[SEu.c %in% Full]

SAf.c=read.table(paste('/data/PopGen/ZI-ZS_cand_ann.weir.fst',sep=''),sep='\t', quote = '')
SAf.c.in3rp=subset(SAf.c,SAf.c[,1]=='3R' &amp; SAf.c[,2]&gt; 16432209 &amp; SAf.c[,2] &lt; 24744010)
SAf.c.genes=unique(SAf.c$V5[sapply(SAf.c$V5,FUN=function(x){x!=''})])
SAf.c.in3rp.genes=unique(SAf.c.in3rp$V5[sapply(SAf.c.in3rp$V5,FUN=function(x){x!=''})])
SAf.c=unique(paste(SAf.c[,1],SAf.c[,2],sep='_'))
SAf.c.in3rp=paste(SAf.c.in3rp[,1],SAf.c.in3rp[,2],sep='_')
SAf.c.ovNoAu.in3rp=SAf.c[SAf.c %in% FullnoAu.in3p]
SAf.c.ov.in3rp=SAf.c[SAf.c %in% Full.in3p]
SAf.c.ovNoAu=SAf.c[SAf.c %in% FullnoAu]
SAf.c.ov=SAf.c[SAf.c %in% Full]

SAu.c=read.table(paste('/data/PopGen/II-IS_cand_ann.weir.fst',sep=''),sep='\t', quote = '')
SAu.c.in3rp=subset(SAu.c,SAu.c[,1]=='3R' &amp; SAu.c[,2]&gt; 16432209 &amp; SAu.c[,2] &lt; 24744010)
SAu.c.genes=unique(SAu.c$V5[sapply(SAu.c$V5,FUN=function(x){x!=''})])
SAu.c.in3rp.genes=unique(SAu.c.in3rp$V5[sapply(SAu.c.in3rp$V5,FUN=function(x){x!=''})])
SAu.c=paste(SAu.c[,1],SAu.c[,2],sep='_')
SAu.c.in3rp=paste(SAu.c.in3rp[,1],SAu.c.in3rp[,2],sep='_')
SAu.c.ov.in3rp=SAu.c[SAu.c %in% Full.in3p]
SAu.c.ov=SAu.c[SAu.c %in% Full]


SNA.c=read.table(paste('/data/PopGen/FI-FS_cand_ann.weir.fst',sep=''),sep='\t', quote = '')
SNA.c.in3rp=subset(SNA.c,SNA.c[,1]=='3R' &amp; SNA.c[,2]&gt; 16432209 &amp; SNA.c[,2] &lt; 24744010)
SNA.c.genes=unique(SNA.c$V5[sapply(SNA.c$V5,FUN=function(x){x!=''})])
SNA.c.in3rp.genes=unique(SNA.c.in3rp$V5[sapply(SNA.c.in3rp$V5,FUN=function(x){x!=''})])
SNA.c=paste(SNA.c[,1],SNA.c[,2],sep='_')
SNA.c.in3rp=paste(SNA.c.in3rp[,1],SNA.c.in3rp[,2],sep='_')
SNA.c.ovNoAu.in3rp=SNA.c[SNA.c %in% FullnoAu.in3p]
SNA.c.ov.in3rp=SNA.c[SNA.c %in% Full.in3p]
SNA.c.ovNoAu=SNA.c[SNA.c %in% FullnoAu]
SNA.c.ov=SNA.c[SNA.c %in% Full]


## overlap in candidate SNPs from all continents except Australia and in all genomic regions
sink(paste('/data/PopGen/candidates/All-noAu.txt',sep=''))
pdf(paste('/data/PopGen/candidates/All-noAu.pdf',sep=''),width=8,height=8)
sets=list('Europe'=unique(SEu.c.ovNoAu),'NorthAmerica'=unique(SNA.c.ovNoAu),'Africa'=unique(SAf.c.ovNoAu))
res=supertest(sets,n=length(FullnoAu))
plot(res,Layout='landscape',degree=2:4,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=300)
dev.off()
print(summary(res))
sink()
write.table(data.frame('NAME'=summary(res)[1],'GENES'=summary(res)[8]),file=paste('/data/PopGen/candidates/All-noAU.list',sep=''),row.names=F,quote = F,sep='\t')

## overlap in candidate SNPs from all continents except Australia and in In(3R)P only
sink(paste('/data/PopGen/candidates/All-noAu-in3rpOnly.txt',sep=''))
pdf(paste('/data/PopGen/candidates/All-noAu-in3rpOnly.pdf',sep=''),width=8,height=8)
sets.in3rp=list('Europe'=SEu.c.ovNoAu.in3rp,'NorthAmerica'=SNA.c.ovNoAu.in3rp,'Africa'=SAf.c.ovNoAu.in3rp)
res.in3rp=supertest(sets.in3rp,n=length(FullnoAu.in3p))
plot(res.in3rp,Layout='landscape',degree=2:4,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=300)
dev.off()
print(summary(res.in3rp))
sink()
write.table(data.frame('NAME'=summary(res.in3rp)[1],'GENES'=summary(res.in3rp)[8]),file=paste('/data/PopGen/candidates/All-noAu-in3rpOnly.list',sep=''),row.names=F,quote = F,sep='\t')

## overlap in candidate SNPs from all continents in In(3R)P only
sink(paste('/data/PopGen/candidates/All-in3rpOnly.txt',sep=''))
pdf(paste('/data/PopGen/candidates/All-in3rpOnly.pdf',sep=''),width=8,height=8)
sets.in3rp.AU=list('Europe'=SEu.c.ov.in3rp,'NorthAmerica'=SNA.c.ov.in3rp,'Africa'=SAf.c.ov.in3rp,'Australia'=SAu.c.ov.in3rp)
res.in3rp.AU=supertest(sets.in3rp.AU,n=length(Full.in3p))
plot(res.in3rp.AU,Layout='landscape',degree=2:8,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=50)
dev.off()
print(summary(res.in3rp.AU))
sink()
write.table(data.frame('NAME'=summary(res.in3rp.AU)[1],'GENES'=summary(res.in3rp.AU)[8]),file=paste('/data/PopGen/candidates/All-in3rpOnly.list',sep=''),row.names=F,quote = F,sep='\t')

## overlap in candidate SNPs from all continents and in all genomic regions
sink(paste('/data/PopGen/candidates/All.txt',sep=''))
pdf(paste('/data/PopGen/candidates/All.pdf',sep=''),width=8,height=8)
sets.AU=list('Europe'=SEu.c.ov,'NorthAmerica'=SNA.c.ov,'Africa'=SAf.c.ov,'Australia'=SAu.c.ov)
res.AU=supertest(sets.AU,n=length(Full.in3p))
plot(res.AU,Layout='landscape',degree=2:8,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=0,maxMinusLog10PValue=50)
dev.off()
print(summary(res.AU))
sink()
write.table(data.frame('NAME'=summary(res.AU)[1],'GENES'=summary(res.AU)[8]),file=paste('/data/PopGen/candidates/All.list',sep=''),row.names=F,quote = F,sep='\t')

# now gene-based

## get overlapping genes
SAf.c.ovNoAu.in3rp.genes=SAf.c.genes[SAf.c.genes %in% FullnoAu.in3p.genes]
SAf.c.ov.in3rp.genes=SAf.c.genes[SAf.c.genes %in% Full.in3p.genes]
SAf.c.ovNoAu.genes=SAf.c.genes[SAf.c.genes %in% FullnoAu.genes]
SAf.c.ov.genes=SAf.c.genes[SAf.c.genes %in% Full.genes]

SEu.c.ovNoAu.in3rp.genes=SEu.c.genes[SEu.c.genes %in% FullnoAu.in3p.genes]
SEu.c.ov.in3rp.genes=SEu.c.genes[SEu.c.genes %in% Full.in3p.genes]
SEu.c.ovNoAu.genes=SEu.c.genes[SEu.c.genes %in% FullnoAu.genes]
SEu.c.ov.genes=SEu.c.genes[SEu.c.genes %in% Full.genes]

SNA.c.ovNoAu.in3rp.genes=SNA.c.genes[SNA.c.genes %in% FullnoAu.in3p.genes]
SNA.c.ov.in3rp.genes=SNA.c.genes[SNA.c.genes %in% Full.in3p.genes]
SNA.c.ovNoAu.genes=SNA.c.genes[SNA.c.genes %in% FullnoAu.genes]
SNA.c.ov.genes=SNA.c.genes[SNA.c.genes %in% Full.genes]

SAu.c.ov.in3rp.genes=SAu.c.genes[SAu.c.genes %in% Full.in3p.genes]
SAu.c.ov.genes=SAu.c.genes[SAu.c.genes %in% Full.genes]

## overlap in candidate genes from all continents except Australia and in all genomic regions
sink(paste('/data/PopGen/candidates/All-noAu-genes.txt',sep=''))
pdf(paste('/data/PopGen/candidates/All-noAu-genes.pdf',sep=''),width=8,height=8)
sets=list('Europe'=unique(SEu.c.ovNoAu.genes),'NorthAmerica'=unique(SNA.c.ovNoAu.genes),'Africa'=unique(SAf.c.ovNoAu.genes))
res=supertest(sets,n=length(FullnoAu))
plot(res,Layout='landscape',degree=2:4,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=300)
dev.off()
print(summary(res))
sink()
write.table(data.frame('NAME'=summary(res)[1],'GENES'=summary(res)[8]),file=paste('/data/PopGen/candidates/All-nAU-genes.list',sep=''),row.names=F,quote = F,sep='\t')

## overlap in candidate genes from all continents except Australia in In(3R)P only
sink(paste('/data/PopGen/candidates/All-noAu-in3rpOnly-genes.txt',sep=''))
pdf(paste('/data/PopGen/candidates/All-noAu-in3rpOnly-genes.pdf',sep=''),width=8,height=8)
sets.in3rp=list('Europe'=SEu.c.ovNoAu.in3rp.genes,'NorthAmerica'=SNA.c.ovNoAu.in3rp.genes,'Africa'=SAf.c.ovNoAu.in3rp.genes)
res.in3rp=supertest(sets.in3rp,n=length(FullnoAu.in3p))
plot(res.in3rp,Layout='landscape',degree=2:4,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=300)
dev.off()
print(summary(res.in3rp))
sink()
write.table(data.frame('NAME'=summary(res.in3rp)[1],'GENES'=summary(res.in3rp)[8]),file=paste('/data/PopGen/candidates/All-noAU-in3rpOnly-genes.list',sep=''),row.names=F,quote = F,sep='\t')

## overlap in candidate genes from all continents in In(3R)P only
sink(paste('/data/PopGen/candidates/All-in3rpOnly-genes.txt',sep=''))
pdf(paste('/data/PopGen/candidates/All-in3rpOnly-genes.pdf',sep=''),width=8,height=8)
sets.in3rp.AU=list('Europe'=SEu.c.ov.in3rp.genes,'NorthAmerica'=SNA.c.ov.in3rp.genes,'Africa'=SAf.c.ov.in3rp.genes,'Australia'=SAu.c.ov.in3rp.genes)
res.in3rp.AU=supertest(sets.in3rp.AU,n=length(Full.in3p))
plot(res.in3rp.AU,Layout='landscape',degree=2:8,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=2,maxMinusLog10PValue=200)
dev.off()
print(summary(res.in3rp.AU))
sink()
write.table(data.frame('NAME'=summary(res.in3rp.AU)[1],'GENES'=summary(res.in3rp.AU)[8]),file=paste('/data/PopGen/candidates/All-in3rpOnly-genes.list',sep=''),row.names=F,quote = F,sep='\t')

## overlap in candidate genes from all continents and all genomic regions
sink(paste('/data/PopGen/candidates/All-genes.txt',sep=''))
pdf(paste('/data/PopGen/candidates/All-genes.pdf',sep=''),width=8,height=8)
sets.AU=list('Europe'=SEu.c.ov.genes,'NorthAmerica'=SNA.c.ov.genes,'Africa'=SAf.c.ov.genes,'Australia'=SAu.c.ov.genes)
res.AU=supertest(sets.AU,n=length(Full.in3p))
plot(res.AU,Layout='landscape',degree=2:8,sort.by = 'size',x.pos=c(0.1,0.9),minMinusLog10PValue=0,maxMinusLog10PValue=200)
dev.off()
print(summary(res.AU))
sink()
write.table(data.frame('NAME'=summary(res.AU)[1],'GENES'=summary(res.AU)[8]),file=paste('/data/PopGen/candidates/All-genes.list',sep=''),row.names=F,quote = F,sep='\t')" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># first SNP-wise comparison</span>

library(<span class="pl-smi">SuperExactTest</span>)

<span class="pl-c"><span class="pl-c">#</span># read background and idenitfy SNPs inside the inversion</span>
<span class="pl-v">SEu.f</span><span class="pl-k">=</span>na.omit(read.table(gzfile(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/PI-PS_ann.weir.fst<span class="pl-pds">'</span></span>),<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">na.string</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>nan<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
<span class="pl-v">SEu.f.in3rp</span><span class="pl-k">=</span>subset(<span class="pl-smi">SEu.f</span>,<span class="pl-smi">SEu.f</span>[,<span class="pl-c1">1</span>]<span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">'</span>3R<span class="pl-pds">'</span></span> <span class="pl-k">&amp;</span> <span class="pl-smi">SEu.f</span>[,<span class="pl-c1">2</span>]<span class="pl-k">&gt;</span> <span class="pl-c1">16432209</span> <span class="pl-k">&amp;</span> <span class="pl-smi">SEu.f</span>[,<span class="pl-c1">2</span>] <span class="pl-k">&lt;</span> <span class="pl-c1">24744010</span>)
<span class="pl-v">SAf.f</span><span class="pl-k">=</span>na.omit(read.table(gzfile(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/ZI-ZS_ann.weir.fst<span class="pl-pds">'</span></span>),<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">na.string</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>nan<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
<span class="pl-v">SAf.f.in3rp</span><span class="pl-k">=</span>subset(<span class="pl-smi">SAf.f</span>,<span class="pl-smi">SAf.f</span>[,<span class="pl-c1">1</span>]<span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">'</span>3R<span class="pl-pds">'</span></span> <span class="pl-k">&amp;</span> <span class="pl-smi">SAf.f</span>[,<span class="pl-c1">2</span>]<span class="pl-k">&gt;</span> <span class="pl-c1">16432209</span> <span class="pl-k">&amp;</span> <span class="pl-smi">SAf.f</span>[,<span class="pl-c1">2</span>] <span class="pl-k">&lt;</span> <span class="pl-c1">24744010</span>)
<span class="pl-v">SAu.f</span><span class="pl-k">=</span>na.omit(read.table(gzfile(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/II-IS_ann.weir.fst<span class="pl-pds">'</span></span>),<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">na.string</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>nan<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
<span class="pl-v">SAu.f.in3rp</span><span class="pl-k">=</span>subset(<span class="pl-smi">SAu.f</span>,<span class="pl-smi">SAu.f</span>[,<span class="pl-c1">1</span>]<span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">'</span>3R<span class="pl-pds">'</span></span> <span class="pl-k">&amp;</span> <span class="pl-smi">SAu.f</span>[,<span class="pl-c1">2</span>]<span class="pl-k">&gt;</span> <span class="pl-c1">16432209</span> <span class="pl-k">&amp;</span> <span class="pl-smi">SAu.f</span>[,<span class="pl-c1">2</span>] <span class="pl-k">&lt;</span> <span class="pl-c1">24744010</span>)
<span class="pl-v">SNA.f</span><span class="pl-k">=</span>na.omit(read.table(gzfile(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/FI-FS_ann.weir.fst<span class="pl-pds">'</span></span>),<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">na.string</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>nan<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
<span class="pl-v">SNA.f.in3rp</span><span class="pl-k">=</span>subset(<span class="pl-smi">SNA.f</span>,<span class="pl-smi">SNA.f</span>[,<span class="pl-c1">1</span>]<span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">'</span>3R<span class="pl-pds">'</span></span> <span class="pl-k">&amp;</span> <span class="pl-smi">SNA.f</span>[,<span class="pl-c1">2</span>]<span class="pl-k">&gt;</span> <span class="pl-c1">16432209</span> <span class="pl-k">&amp;</span> <span class="pl-smi">SNA.f</span>[,<span class="pl-c1">2</span>] <span class="pl-k">&lt;</span> <span class="pl-c1">24744010</span>)

<span class="pl-c"><span class="pl-c">#</span># now for genes</span>
<span class="pl-v">SAf.f.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SAf.f</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SAf.f</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SEu.f.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SEu.f</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SEu.f</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SAu.f.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SAu.f</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SAu.f</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SNA.f.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SNA.f</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SNA.f</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])

<span class="pl-v">SAf.f.in3rp.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SAf.f.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SAf.f.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SEu.f.in3rp.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SEu.f.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SEu.f.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SAu.f.in3rp.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SAu.f.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SAu.f.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SNA.f.in3rp.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SNA.f.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SNA.f.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])

<span class="pl-c"><span class="pl-c">#</span># now for SNPs</span>
<span class="pl-v">SEu.f</span><span class="pl-k">=</span>unique(paste(<span class="pl-smi">SEu.f</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SEu.f</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>))
<span class="pl-v">SEu.f.in3rp</span><span class="pl-k">=</span>unique(paste(<span class="pl-smi">SEu.f.in3rp</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SEu.f.in3rp</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>))
<span class="pl-v">SAf.f</span><span class="pl-k">=</span>unique(paste(<span class="pl-smi">SAf.f</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SAf.f</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>))
<span class="pl-v">SAf.f.in3rp</span><span class="pl-k">=</span>unique(paste(<span class="pl-smi">SAf.f.in3rp</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SAf.f.in3rp</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>))
<span class="pl-v">SAu.f</span><span class="pl-k">=</span>unique(paste(<span class="pl-smi">SAu.f</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SAu.f</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>))
<span class="pl-v">SAu.f.in3rp</span><span class="pl-k">=</span>unique(paste(<span class="pl-smi">SAu.f.in3rp</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SAu.f.in3rp</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>))
<span class="pl-v">SNA.f</span><span class="pl-k">=</span>unique(paste(<span class="pl-smi">SNA.f</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SNA.f</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>))
<span class="pl-v">SNA.f.in3rp</span><span class="pl-k">=</span>unique(paste(<span class="pl-smi">SNA.f.in3rp</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SNA.f.in3rp</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>))

<span class="pl-c"><span class="pl-c">#</span> get overlapping SNPs &amp; genes</span>
<span class="pl-v">FullnoAu</span><span class="pl-k">=</span>Reduce(<span class="pl-smi">intersect</span>,<span class="pl-k">list</span>(<span class="pl-smi">SEu.f</span>,<span class="pl-smi">SAf.f</span>,<span class="pl-smi">SNA.f</span>))
<span class="pl-v">FullnoAu.in3p</span><span class="pl-k">=</span>Reduce(<span class="pl-smi">intersect</span>,<span class="pl-k">list</span>(<span class="pl-smi">SEu.f.in3rp</span>,<span class="pl-smi">SAf.f.in3rp</span>,<span class="pl-smi">SNA.f.in3rp</span>))

<span class="pl-v">Full</span><span class="pl-k">=</span>Reduce(<span class="pl-smi">intersect</span>,<span class="pl-k">list</span>(<span class="pl-smi">SEu.f</span>,<span class="pl-smi">SAf.f</span>,<span class="pl-smi">SNA.f</span>,<span class="pl-smi">SAu.f</span>))
<span class="pl-v">Full.in3p</span><span class="pl-k">=</span>Reduce(<span class="pl-smi">intersect</span>,<span class="pl-k">list</span>(<span class="pl-smi">SEu.f.in3rp</span>,<span class="pl-smi">SAf.f.in3rp</span>,<span class="pl-smi">SNA.f.in3rp</span>,<span class="pl-smi">SAu.f.in3rp</span>))

<span class="pl-v">FullnoAu.genes</span><span class="pl-k">=</span>Reduce(<span class="pl-smi">intersect</span>,<span class="pl-k">list</span>(<span class="pl-smi">SAf.f.genes</span>,<span class="pl-smi">SEu.f.genes</span>,<span class="pl-smi">SNA.f.genes</span>))
<span class="pl-v">FullnoAu.in3p.genes</span><span class="pl-k">=</span>Reduce(<span class="pl-smi">intersect</span>,<span class="pl-k">list</span>(<span class="pl-smi">SEu.f.in3rp.genes</span>,<span class="pl-smi">SAf.f.in3rp.genes</span>,<span class="pl-smi">SNA.f.in3rp.genes</span>))

<span class="pl-v">Full.genes</span><span class="pl-k">=</span>Reduce(<span class="pl-smi">intersect</span>,<span class="pl-k">list</span>(<span class="pl-smi">SEu.f.genes</span>,<span class="pl-smi">SAf.f.genes</span>,<span class="pl-smi">SNA.f.genes</span>,<span class="pl-smi">SAu.f.genes</span>))
<span class="pl-v">Full.in3p.genes</span><span class="pl-k">=</span>Reduce(<span class="pl-smi">intersect</span>,<span class="pl-k">list</span>(<span class="pl-smi">SEu.f.in3rp.genes</span>,<span class="pl-smi">SAf.f.in3rp.genes</span>,<span class="pl-smi">SNA.f.in3rp.genes</span>,<span class="pl-smi">SAu.f.in3rp.genes</span>))

dir.create(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates<span class="pl-pds">'</span></span>)

<span class="pl-c"><span class="pl-c">#</span> candidates</span>
<span class="pl-v">SEu.c</span><span class="pl-k">=</span>read.table(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/PI-PS_cand_ann.weir.fst<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>)
<span class="pl-v">SEu.c.in3rp</span><span class="pl-k">=</span>subset(<span class="pl-smi">SEu.c</span>,<span class="pl-smi">SEu.c</span>[,<span class="pl-c1">1</span>]<span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">'</span>3R<span class="pl-pds">'</span></span> <span class="pl-k">&amp;</span> <span class="pl-smi">SEu.c</span>[,<span class="pl-c1">2</span>]<span class="pl-k">&gt;</span> <span class="pl-c1">16432209</span> <span class="pl-k">&amp;</span> <span class="pl-smi">SEu.c</span>[,<span class="pl-c1">2</span>] <span class="pl-k">&lt;</span> <span class="pl-c1">24744010</span>)
<span class="pl-v">SEu.c.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SEu.c</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SEu.c</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SEu.c.in3rp.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SEu.c.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SEu.c.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SEu.c</span><span class="pl-k">=</span>unique(paste(<span class="pl-smi">SEu.c</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SEu.c</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>))
<span class="pl-v">SEu.c.in3rp</span><span class="pl-k">=</span>unique(paste(<span class="pl-smi">SEu.c.in3rp</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SEu.c.in3rp</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>))
<span class="pl-v">SEu.c.ovNoAu.in3rp</span><span class="pl-k">=</span><span class="pl-smi">SEu.c</span>[<span class="pl-smi">SEu.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu.in3p</span>]
<span class="pl-v">SEu.c.ov.in3rp</span><span class="pl-k">=</span><span class="pl-smi">SEu.c</span>[<span class="pl-smi">SEu.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.in3p</span>]
<span class="pl-v">SEu.c.ovNoAu</span><span class="pl-k">=</span><span class="pl-smi">SEu.c</span>[<span class="pl-smi">SEu.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu</span>]
<span class="pl-v">SEu.c.ov</span><span class="pl-k">=</span><span class="pl-smi">SEu.c</span>[<span class="pl-smi">SEu.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full</span>]

<span class="pl-v">SAf.c</span><span class="pl-k">=</span>read.table(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/ZI-ZS_cand_ann.weir.fst<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>)
<span class="pl-v">SAf.c.in3rp</span><span class="pl-k">=</span>subset(<span class="pl-smi">SAf.c</span>,<span class="pl-smi">SAf.c</span>[,<span class="pl-c1">1</span>]<span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">'</span>3R<span class="pl-pds">'</span></span> <span class="pl-k">&amp;</span> <span class="pl-smi">SAf.c</span>[,<span class="pl-c1">2</span>]<span class="pl-k">&gt;</span> <span class="pl-c1">16432209</span> <span class="pl-k">&amp;</span> <span class="pl-smi">SAf.c</span>[,<span class="pl-c1">2</span>] <span class="pl-k">&lt;</span> <span class="pl-c1">24744010</span>)
<span class="pl-v">SAf.c.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SAf.c</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SAf.c</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SAf.c.in3rp.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SAf.c.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SAf.c.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SAf.c</span><span class="pl-k">=</span>unique(paste(<span class="pl-smi">SAf.c</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SAf.c</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>))
<span class="pl-v">SAf.c.in3rp</span><span class="pl-k">=</span>paste(<span class="pl-smi">SAf.c.in3rp</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SAf.c.in3rp</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>)
<span class="pl-v">SAf.c.ovNoAu.in3rp</span><span class="pl-k">=</span><span class="pl-smi">SAf.c</span>[<span class="pl-smi">SAf.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu.in3p</span>]
<span class="pl-v">SAf.c.ov.in3rp</span><span class="pl-k">=</span><span class="pl-smi">SAf.c</span>[<span class="pl-smi">SAf.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.in3p</span>]
<span class="pl-v">SAf.c.ovNoAu</span><span class="pl-k">=</span><span class="pl-smi">SAf.c</span>[<span class="pl-smi">SAf.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu</span>]
<span class="pl-v">SAf.c.ov</span><span class="pl-k">=</span><span class="pl-smi">SAf.c</span>[<span class="pl-smi">SAf.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full</span>]

<span class="pl-v">SAu.c</span><span class="pl-k">=</span>read.table(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/II-IS_cand_ann.weir.fst<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>)
<span class="pl-v">SAu.c.in3rp</span><span class="pl-k">=</span>subset(<span class="pl-smi">SAu.c</span>,<span class="pl-smi">SAu.c</span>[,<span class="pl-c1">1</span>]<span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">'</span>3R<span class="pl-pds">'</span></span> <span class="pl-k">&amp;</span> <span class="pl-smi">SAu.c</span>[,<span class="pl-c1">2</span>]<span class="pl-k">&gt;</span> <span class="pl-c1">16432209</span> <span class="pl-k">&amp;</span> <span class="pl-smi">SAu.c</span>[,<span class="pl-c1">2</span>] <span class="pl-k">&lt;</span> <span class="pl-c1">24744010</span>)
<span class="pl-v">SAu.c.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SAu.c</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SAu.c</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SAu.c.in3rp.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SAu.c.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SAu.c.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SAu.c</span><span class="pl-k">=</span>paste(<span class="pl-smi">SAu.c</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SAu.c</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>)
<span class="pl-v">SAu.c.in3rp</span><span class="pl-k">=</span>paste(<span class="pl-smi">SAu.c.in3rp</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SAu.c.in3rp</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>)
<span class="pl-v">SAu.c.ov.in3rp</span><span class="pl-k">=</span><span class="pl-smi">SAu.c</span>[<span class="pl-smi">SAu.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.in3p</span>]
<span class="pl-v">SAu.c.ov</span><span class="pl-k">=</span><span class="pl-smi">SAu.c</span>[<span class="pl-smi">SAu.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full</span>]


<span class="pl-v">SNA.c</span><span class="pl-k">=</span>read.table(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/FI-FS_cand_ann.weir.fst<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>)
<span class="pl-v">SNA.c.in3rp</span><span class="pl-k">=</span>subset(<span class="pl-smi">SNA.c</span>,<span class="pl-smi">SNA.c</span>[,<span class="pl-c1">1</span>]<span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">'</span>3R<span class="pl-pds">'</span></span> <span class="pl-k">&amp;</span> <span class="pl-smi">SNA.c</span>[,<span class="pl-c1">2</span>]<span class="pl-k">&gt;</span> <span class="pl-c1">16432209</span> <span class="pl-k">&amp;</span> <span class="pl-smi">SNA.c</span>[,<span class="pl-c1">2</span>] <span class="pl-k">&lt;</span> <span class="pl-c1">24744010</span>)
<span class="pl-v">SNA.c.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SNA.c</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SNA.c</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SNA.c.in3rp.genes</span><span class="pl-k">=</span>unique(<span class="pl-smi">SNA.c.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>[sapply(<span class="pl-smi">SNA.c.in3rp</span><span class="pl-k">$</span><span class="pl-smi">V5</span>,<span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>){<span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>})])
<span class="pl-v">SNA.c</span><span class="pl-k">=</span>paste(<span class="pl-smi">SNA.c</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SNA.c</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>)
<span class="pl-v">SNA.c.in3rp</span><span class="pl-k">=</span>paste(<span class="pl-smi">SNA.c.in3rp</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">SNA.c.in3rp</span>[,<span class="pl-c1">2</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>)
<span class="pl-v">SNA.c.ovNoAu.in3rp</span><span class="pl-k">=</span><span class="pl-smi">SNA.c</span>[<span class="pl-smi">SNA.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu.in3p</span>]
<span class="pl-v">SNA.c.ov.in3rp</span><span class="pl-k">=</span><span class="pl-smi">SNA.c</span>[<span class="pl-smi">SNA.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.in3p</span>]
<span class="pl-v">SNA.c.ovNoAu</span><span class="pl-k">=</span><span class="pl-smi">SNA.c</span>[<span class="pl-smi">SNA.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu</span>]
<span class="pl-v">SNA.c.ov</span><span class="pl-k">=</span><span class="pl-smi">SNA.c</span>[<span class="pl-smi">SNA.c</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full</span>]


<span class="pl-c"><span class="pl-c">#</span># overlap in candidate SNPs from all continents except Australia and in all genomic regions</span>
sink(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-noAu.txt<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
pdf(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-noAu.pdf<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">8</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)
<span class="pl-v">sets</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">'</span>Europe<span class="pl-pds">'</span></span><span class="pl-k">=</span>unique(<span class="pl-smi">SEu.c.ovNoAu</span>),<span class="pl-s"><span class="pl-pds">'</span>NorthAmerica<span class="pl-pds">'</span></span><span class="pl-k">=</span>unique(<span class="pl-smi">SNA.c.ovNoAu</span>),<span class="pl-s"><span class="pl-pds">'</span>Africa<span class="pl-pds">'</span></span><span class="pl-k">=</span>unique(<span class="pl-smi">SAf.c.ovNoAu</span>))
<span class="pl-v">res</span><span class="pl-k">=</span>supertest(<span class="pl-smi">sets</span>,<span class="pl-v">n</span><span class="pl-k">=</span>length(<span class="pl-smi">FullnoAu</span>))
plot(<span class="pl-smi">res</span>,<span class="pl-v">Layout</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>landscape<span class="pl-pds">'</span></span>,<span class="pl-v">degree</span><span class="pl-k">=</span><span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">4</span>,<span class="pl-v">sort.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>size<span class="pl-pds">'</span></span>,<span class="pl-v">x.pos</span><span class="pl-k">=</span>c(<span class="pl-c1">0.1</span>,<span class="pl-c1">0.9</span>),<span class="pl-v">minMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">maxMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">300</span>)
dev.off()
print(summary(<span class="pl-smi">res</span>))
sink()
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">'</span>NAME<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res</span>)[<span class="pl-c1">1</span>],<span class="pl-s"><span class="pl-pds">'</span>GENES<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res</span>)[<span class="pl-c1">8</span>]),<span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-noAU.list<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>)

<span class="pl-c"><span class="pl-c">#</span># overlap in candidate SNPs from all continents except Australia and in In(3R)P only</span>
sink(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-noAu-in3rpOnly.txt<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
pdf(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-noAu-in3rpOnly.pdf<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">8</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)
<span class="pl-v">sets.in3rp</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">'</span>Europe<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SEu.c.ovNoAu.in3rp</span>,<span class="pl-s"><span class="pl-pds">'</span>NorthAmerica<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SNA.c.ovNoAu.in3rp</span>,<span class="pl-s"><span class="pl-pds">'</span>Africa<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SAf.c.ovNoAu.in3rp</span>)
<span class="pl-v">res.in3rp</span><span class="pl-k">=</span>supertest(<span class="pl-smi">sets.in3rp</span>,<span class="pl-v">n</span><span class="pl-k">=</span>length(<span class="pl-smi">FullnoAu.in3p</span>))
plot(<span class="pl-smi">res.in3rp</span>,<span class="pl-v">Layout</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>landscape<span class="pl-pds">'</span></span>,<span class="pl-v">degree</span><span class="pl-k">=</span><span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">4</span>,<span class="pl-v">sort.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>size<span class="pl-pds">'</span></span>,<span class="pl-v">x.pos</span><span class="pl-k">=</span>c(<span class="pl-c1">0.1</span>,<span class="pl-c1">0.9</span>),<span class="pl-v">minMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">maxMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">300</span>)
dev.off()
print(summary(<span class="pl-smi">res.in3rp</span>))
sink()
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">'</span>NAME<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.in3rp</span>)[<span class="pl-c1">1</span>],<span class="pl-s"><span class="pl-pds">'</span>GENES<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.in3rp</span>)[<span class="pl-c1">8</span>]),<span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-noAu-in3rpOnly.list<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>)

<span class="pl-c"><span class="pl-c">#</span># overlap in candidate SNPs from all continents in In(3R)P only</span>
sink(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-in3rpOnly.txt<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
pdf(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-in3rpOnly.pdf<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">8</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)
<span class="pl-v">sets.in3rp.AU</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">'</span>Europe<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SEu.c.ov.in3rp</span>,<span class="pl-s"><span class="pl-pds">'</span>NorthAmerica<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SNA.c.ov.in3rp</span>,<span class="pl-s"><span class="pl-pds">'</span>Africa<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SAf.c.ov.in3rp</span>,<span class="pl-s"><span class="pl-pds">'</span>Australia<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SAu.c.ov.in3rp</span>)
<span class="pl-v">res.in3rp.AU</span><span class="pl-k">=</span>supertest(<span class="pl-smi">sets.in3rp.AU</span>,<span class="pl-v">n</span><span class="pl-k">=</span>length(<span class="pl-smi">Full.in3p</span>))
plot(<span class="pl-smi">res.in3rp.AU</span>,<span class="pl-v">Layout</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>landscape<span class="pl-pds">'</span></span>,<span class="pl-v">degree</span><span class="pl-k">=</span><span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">8</span>,<span class="pl-v">sort.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>size<span class="pl-pds">'</span></span>,<span class="pl-v">x.pos</span><span class="pl-k">=</span>c(<span class="pl-c1">0.1</span>,<span class="pl-c1">0.9</span>),<span class="pl-v">minMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">maxMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">50</span>)
dev.off()
print(summary(<span class="pl-smi">res.in3rp.AU</span>))
sink()
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">'</span>NAME<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.in3rp.AU</span>)[<span class="pl-c1">1</span>],<span class="pl-s"><span class="pl-pds">'</span>GENES<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.in3rp.AU</span>)[<span class="pl-c1">8</span>]),<span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-in3rpOnly.list<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>)

<span class="pl-c"><span class="pl-c">#</span># overlap in candidate SNPs from all continents and in all genomic regions</span>
sink(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All.txt<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
pdf(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All.pdf<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">8</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)
<span class="pl-v">sets.AU</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">'</span>Europe<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SEu.c.ov</span>,<span class="pl-s"><span class="pl-pds">'</span>NorthAmerica<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SNA.c.ov</span>,<span class="pl-s"><span class="pl-pds">'</span>Africa<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SAf.c.ov</span>,<span class="pl-s"><span class="pl-pds">'</span>Australia<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SAu.c.ov</span>)
<span class="pl-v">res.AU</span><span class="pl-k">=</span>supertest(<span class="pl-smi">sets.AU</span>,<span class="pl-v">n</span><span class="pl-k">=</span>length(<span class="pl-smi">Full.in3p</span>))
plot(<span class="pl-smi">res.AU</span>,<span class="pl-v">Layout</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>landscape<span class="pl-pds">'</span></span>,<span class="pl-v">degree</span><span class="pl-k">=</span><span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">8</span>,<span class="pl-v">sort.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>size<span class="pl-pds">'</span></span>,<span class="pl-v">x.pos</span><span class="pl-k">=</span>c(<span class="pl-c1">0.1</span>,<span class="pl-c1">0.9</span>),<span class="pl-v">minMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">0</span>,<span class="pl-v">maxMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">50</span>)
dev.off()
print(summary(<span class="pl-smi">res.AU</span>))
sink()
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">'</span>NAME<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.AU</span>)[<span class="pl-c1">1</span>],<span class="pl-s"><span class="pl-pds">'</span>GENES<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.AU</span>)[<span class="pl-c1">8</span>]),<span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All.list<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>)

<span class="pl-c"><span class="pl-c">#</span> now gene-based</span>

<span class="pl-c"><span class="pl-c">#</span># get overlapping genes</span>
<span class="pl-v">SAf.c.ovNoAu.in3rp.genes</span><span class="pl-k">=</span><span class="pl-smi">SAf.c.genes</span>[<span class="pl-smi">SAf.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu.in3p.genes</span>]
<span class="pl-v">SAf.c.ov.in3rp.genes</span><span class="pl-k">=</span><span class="pl-smi">SAf.c.genes</span>[<span class="pl-smi">SAf.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.in3p.genes</span>]
<span class="pl-v">SAf.c.ovNoAu.genes</span><span class="pl-k">=</span><span class="pl-smi">SAf.c.genes</span>[<span class="pl-smi">SAf.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu.genes</span>]
<span class="pl-v">SAf.c.ov.genes</span><span class="pl-k">=</span><span class="pl-smi">SAf.c.genes</span>[<span class="pl-smi">SAf.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.genes</span>]

<span class="pl-v">SEu.c.ovNoAu.in3rp.genes</span><span class="pl-k">=</span><span class="pl-smi">SEu.c.genes</span>[<span class="pl-smi">SEu.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu.in3p.genes</span>]
<span class="pl-v">SEu.c.ov.in3rp.genes</span><span class="pl-k">=</span><span class="pl-smi">SEu.c.genes</span>[<span class="pl-smi">SEu.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.in3p.genes</span>]
<span class="pl-v">SEu.c.ovNoAu.genes</span><span class="pl-k">=</span><span class="pl-smi">SEu.c.genes</span>[<span class="pl-smi">SEu.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu.genes</span>]
<span class="pl-v">SEu.c.ov.genes</span><span class="pl-k">=</span><span class="pl-smi">SEu.c.genes</span>[<span class="pl-smi">SEu.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.genes</span>]

<span class="pl-v">SNA.c.ovNoAu.in3rp.genes</span><span class="pl-k">=</span><span class="pl-smi">SNA.c.genes</span>[<span class="pl-smi">SNA.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu.in3p.genes</span>]
<span class="pl-v">SNA.c.ov.in3rp.genes</span><span class="pl-k">=</span><span class="pl-smi">SNA.c.genes</span>[<span class="pl-smi">SNA.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.in3p.genes</span>]
<span class="pl-v">SNA.c.ovNoAu.genes</span><span class="pl-k">=</span><span class="pl-smi">SNA.c.genes</span>[<span class="pl-smi">SNA.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">FullnoAu.genes</span>]
<span class="pl-v">SNA.c.ov.genes</span><span class="pl-k">=</span><span class="pl-smi">SNA.c.genes</span>[<span class="pl-smi">SNA.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.genes</span>]

<span class="pl-v">SAu.c.ov.in3rp.genes</span><span class="pl-k">=</span><span class="pl-smi">SAu.c.genes</span>[<span class="pl-smi">SAu.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.in3p.genes</span>]
<span class="pl-v">SAu.c.ov.genes</span><span class="pl-k">=</span><span class="pl-smi">SAu.c.genes</span>[<span class="pl-smi">SAu.c.genes</span> <span class="pl-k">%in%</span> <span class="pl-smi">Full.genes</span>]

<span class="pl-c"><span class="pl-c">#</span># overlap in candidate genes from all continents except Australia and in all genomic regions</span>
sink(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-noAu-genes.txt<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
pdf(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-noAu-genes.pdf<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">8</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)
<span class="pl-v">sets</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">'</span>Europe<span class="pl-pds">'</span></span><span class="pl-k">=</span>unique(<span class="pl-smi">SEu.c.ovNoAu.genes</span>),<span class="pl-s"><span class="pl-pds">'</span>NorthAmerica<span class="pl-pds">'</span></span><span class="pl-k">=</span>unique(<span class="pl-smi">SNA.c.ovNoAu.genes</span>),<span class="pl-s"><span class="pl-pds">'</span>Africa<span class="pl-pds">'</span></span><span class="pl-k">=</span>unique(<span class="pl-smi">SAf.c.ovNoAu.genes</span>))
<span class="pl-v">res</span><span class="pl-k">=</span>supertest(<span class="pl-smi">sets</span>,<span class="pl-v">n</span><span class="pl-k">=</span>length(<span class="pl-smi">FullnoAu</span>))
plot(<span class="pl-smi">res</span>,<span class="pl-v">Layout</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>landscape<span class="pl-pds">'</span></span>,<span class="pl-v">degree</span><span class="pl-k">=</span><span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">4</span>,<span class="pl-v">sort.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>size<span class="pl-pds">'</span></span>,<span class="pl-v">x.pos</span><span class="pl-k">=</span>c(<span class="pl-c1">0.1</span>,<span class="pl-c1">0.9</span>),<span class="pl-v">minMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">maxMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">300</span>)
dev.off()
print(summary(<span class="pl-smi">res</span>))
sink()
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">'</span>NAME<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res</span>)[<span class="pl-c1">1</span>],<span class="pl-s"><span class="pl-pds">'</span>GENES<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res</span>)[<span class="pl-c1">8</span>]),<span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-nAU-genes.list<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>)

<span class="pl-c"><span class="pl-c">#</span># overlap in candidate genes from all continents except Australia in In(3R)P only</span>
sink(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-noAu-in3rpOnly-genes.txt<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
pdf(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-noAu-in3rpOnly-genes.pdf<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">8</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)
<span class="pl-v">sets.in3rp</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">'</span>Europe<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SEu.c.ovNoAu.in3rp.genes</span>,<span class="pl-s"><span class="pl-pds">'</span>NorthAmerica<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SNA.c.ovNoAu.in3rp.genes</span>,<span class="pl-s"><span class="pl-pds">'</span>Africa<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SAf.c.ovNoAu.in3rp.genes</span>)
<span class="pl-v">res.in3rp</span><span class="pl-k">=</span>supertest(<span class="pl-smi">sets.in3rp</span>,<span class="pl-v">n</span><span class="pl-k">=</span>length(<span class="pl-smi">FullnoAu.in3p</span>))
plot(<span class="pl-smi">res.in3rp</span>,<span class="pl-v">Layout</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>landscape<span class="pl-pds">'</span></span>,<span class="pl-v">degree</span><span class="pl-k">=</span><span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">4</span>,<span class="pl-v">sort.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>size<span class="pl-pds">'</span></span>,<span class="pl-v">x.pos</span><span class="pl-k">=</span>c(<span class="pl-c1">0.1</span>,<span class="pl-c1">0.9</span>),<span class="pl-v">minMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">maxMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">300</span>)
dev.off()
print(summary(<span class="pl-smi">res.in3rp</span>))
sink()
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">'</span>NAME<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.in3rp</span>)[<span class="pl-c1">1</span>],<span class="pl-s"><span class="pl-pds">'</span>GENES<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.in3rp</span>)[<span class="pl-c1">8</span>]),<span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-noAU-in3rpOnly-genes.list<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>)

<span class="pl-c"><span class="pl-c">#</span># overlap in candidate genes from all continents in In(3R)P only</span>
sink(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-in3rpOnly-genes.txt<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
pdf(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-in3rpOnly-genes.pdf<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">8</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)
<span class="pl-v">sets.in3rp.AU</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">'</span>Europe<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SEu.c.ov.in3rp.genes</span>,<span class="pl-s"><span class="pl-pds">'</span>NorthAmerica<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SNA.c.ov.in3rp.genes</span>,<span class="pl-s"><span class="pl-pds">'</span>Africa<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SAf.c.ov.in3rp.genes</span>,<span class="pl-s"><span class="pl-pds">'</span>Australia<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SAu.c.ov.in3rp.genes</span>)
<span class="pl-v">res.in3rp.AU</span><span class="pl-k">=</span>supertest(<span class="pl-smi">sets.in3rp.AU</span>,<span class="pl-v">n</span><span class="pl-k">=</span>length(<span class="pl-smi">Full.in3p</span>))
plot(<span class="pl-smi">res.in3rp.AU</span>,<span class="pl-v">Layout</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>landscape<span class="pl-pds">'</span></span>,<span class="pl-v">degree</span><span class="pl-k">=</span><span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">8</span>,<span class="pl-v">sort.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>size<span class="pl-pds">'</span></span>,<span class="pl-v">x.pos</span><span class="pl-k">=</span>c(<span class="pl-c1">0.1</span>,<span class="pl-c1">0.9</span>),<span class="pl-v">minMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">maxMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">200</span>)
dev.off()
print(summary(<span class="pl-smi">res.in3rp.AU</span>))
sink()
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">'</span>NAME<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.in3rp.AU</span>)[<span class="pl-c1">1</span>],<span class="pl-s"><span class="pl-pds">'</span>GENES<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.in3rp.AU</span>)[<span class="pl-c1">8</span>]),<span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-in3rpOnly-genes.list<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>)

<span class="pl-c"><span class="pl-c">#</span># overlap in candidate genes from all continents and all genomic regions</span>
sink(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-genes.txt<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>))
pdf(paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-genes.pdf<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">8</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)
<span class="pl-v">sets.AU</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">'</span>Europe<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SEu.c.ov.genes</span>,<span class="pl-s"><span class="pl-pds">'</span>NorthAmerica<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SNA.c.ov.genes</span>,<span class="pl-s"><span class="pl-pds">'</span>Africa<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SAf.c.ov.genes</span>,<span class="pl-s"><span class="pl-pds">'</span>Australia<span class="pl-pds">'</span></span><span class="pl-k">=</span><span class="pl-smi">SAu.c.ov.genes</span>)
<span class="pl-v">res.AU</span><span class="pl-k">=</span>supertest(<span class="pl-smi">sets.AU</span>,<span class="pl-v">n</span><span class="pl-k">=</span>length(<span class="pl-smi">Full.in3p</span>))
plot(<span class="pl-smi">res.AU</span>,<span class="pl-v">Layout</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>landscape<span class="pl-pds">'</span></span>,<span class="pl-v">degree</span><span class="pl-k">=</span><span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">8</span>,<span class="pl-v">sort.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>size<span class="pl-pds">'</span></span>,<span class="pl-v">x.pos</span><span class="pl-k">=</span>c(<span class="pl-c1">0.1</span>,<span class="pl-c1">0.9</span>),<span class="pl-v">minMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">0</span>,<span class="pl-v">maxMinusLog10PValue</span><span class="pl-k">=</span><span class="pl-c1">200</span>)
dev.off()
print(summary(<span class="pl-smi">res.AU</span>))
sink()
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">'</span>NAME<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.AU</span>)[<span class="pl-c1">1</span>],<span class="pl-s"><span class="pl-pds">'</span>GENES<span class="pl-pds">'</span></span><span class="pl-k">=</span>summary(<span class="pl-smi">res.AU</span>)[<span class="pl-c1">8</span>]),<span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">'</span>/data/PopGen/candidates/All-genes.list<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>),<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-cce">\t</span><span class="pl-pds">'</span></span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      3.3) African origin
                     </h3>
                     <a aria-label="Permalink: 3.3) African origin" class="anchor" href="#33-african-origin" id="user-content-33-african-origin">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Here, we tested if inversion-specific candidate alleles in Florida are also found in the standard arrangement in flies for the ancestral African origin.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='
mkdir /data/InvAlleles

continent=( "Africa" "America" )
region=( "Zambia" "Florida"  )
Pop1=( "Africa_Zambia_Inv" "America_Florida_Inv" )
Pop2=( "Africa_Zambia_Std" "America_Florida_Std" )
FST=( "ZI-ZS" "FI-FS" )

for index in ${!FST[@]}
do
    ## Identify inversion-specific allele in candidates from specific region
    python3 /scripts/InvAf_from_VCF.py \
        --input /data/consensus/${continent[index]}.vcf.gz \
        --inv /data/${Pop1[index]}.samples \
        --std /data/${Pop2[index]}.samples \
        | python3 /scripts/v3/OverlapSNPs.py \
        --source /data/PopGen/${FST[index]}_cand_ann.weir.fst \
        --target - \
        &gt; /data/InvAlleles/${continent[index]}.alleles
    
    Sample=( "Africa_Zambia_Inv" "Africa_Zambia_Std" "Europe_Portugal_Inv" "Europe_Portugal_Std" "Europe_Sweden_Std" "America_Florida_Inv" "America_Florida_Std" "America_Maine_Std" "Australia_Queensland_Inv" "Australia_Queensland_Std" "Australia_Victoria_Std" )

    for index2 in ${!Sample[@]}
    do
        ## get allele frequency of inversion-specific allele in subset of data
        python3 /scripts/AF_by_alleleNind.py \
            --input /data/consensus/${continent[index]}.vcf.gz \
            --AFl /data/InvAlleles/${continent[index]}.alleles \
            --pops /data/${Sample[index2]}.samples \
            &gt; /data/InvAlleles/${continent[index]}_${Sample[index2]}.af

    done
done' dir="auto">
                     <pre>mkdir /data/InvAlleles

continent=( <span class="pl-s"><span class="pl-pds">"</span>Africa<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America<span class="pl-pds">"</span></span> )
region=( <span class="pl-s"><span class="pl-pds">"</span>Zambia<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Florida<span class="pl-pds">"</span></span>  )
Pop1=( <span class="pl-s"><span class="pl-pds">"</span>Africa_Zambia_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Florida_Inv<span class="pl-pds">"</span></span> )
Pop2=( <span class="pl-s"><span class="pl-pds">"</span>Africa_Zambia_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Florida_Std<span class="pl-pds">"</span></span> )
FST=( <span class="pl-s"><span class="pl-pds">"</span>ZI-ZS<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>FI-FS<span class="pl-pds">"</span></span> )

<span class="pl-k">for</span> <span class="pl-smi">index</span> <span class="pl-k">in</span> <span class="pl-smi">${<span class="pl-k">!</span>FST[@]}</span>
<span class="pl-k">do</span>
    <span class="pl-c"><span class="pl-c">#</span># Identify inversion-specific allele in candidates from specific region</span>
    python3 /scripts/InvAf_from_VCF.py \
        --input /data/consensus/<span class="pl-smi">${continent[index]}</span>.vcf.gz \
        --inv /data/<span class="pl-smi">${Pop1[index]}</span>.samples \
        --std /data/<span class="pl-smi">${Pop2[index]}</span>.samples \
        <span class="pl-k">|</span> python3 /scripts/v3/OverlapSNPs.py \
        --source /data/PopGen/<span class="pl-smi">${FST[index]}</span>_cand_ann.weir.fst \
        --target - \
        <span class="pl-k">&gt;</span> /data/InvAlleles/<span class="pl-smi">${continent[index]}</span>.alleles
    
    Sample=( <span class="pl-s"><span class="pl-pds">"</span>Africa_Zambia_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Africa_Zambia_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe_Portugal_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe_Portugal_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe_Sweden_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Florida_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Florida_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Maine_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia_Queensland_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia_Queensland_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia_Victoria_Std<span class="pl-pds">"</span></span> )

    <span class="pl-k">for</span> <span class="pl-smi">index2</span> <span class="pl-k">in</span> <span class="pl-smi">${<span class="pl-k">!</span>Sample[@]}</span>
    <span class="pl-k">do</span>
        <span class="pl-c"><span class="pl-c">#</span># get allele frequency of inversion-specific allele in subset of data</span>
        python3 /scripts/AF_by_alleleNind.py \
            --input /data/consensus/<span class="pl-smi">${continent[index]}</span>.vcf.gz \
            --AFl /data/InvAlleles/<span class="pl-smi">${continent[index]}</span>.alleles \
            --pops /data/<span class="pl-smi">${Sample[index2]}</span>.samples \
            <span class="pl-k">&gt;</span> /data/InvAlleles/<span class="pl-smi">${continent[index]}</span>_<span class="pl-smi">${Sample[index2]}</span>.af

    <span class="pl-k">done</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Now plot with
                     <em>
                      R
                     </em>
                     for inversion-specific alleles in North America
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='
FlS=read.table("/data/InvAlleles/America_America_Florida_S.af",header=F)
FlI=read.table("/data/InvAlleles/America_America_Florida_I.af",header=F)
MaS=read.table("/data/InvAlleles/America_America_Maine_S.af",header=F)
ZaI=read.table("/data/InvAlleles/America_Africa_Zambia_I.af",header=F)
ZaS=read.table("/data/InvAlleles/America_Africa_Zambia_S.af",header=F)
PoI=read.table("/data/InvAlleles/America_Europe_Portugal_I.af",header=F)
PoS=read.table("/data/InvAlleles/America_Europe_Portugal_S.af",header=F)
SwS=read.table("/data/InvAlleles/America_Europe_Sweden_S.af",header=F)

INTER&lt;-Reduce(intersect,list(FlS$V2,FlI$V2,MaS$V2,ZaI$V2,ZaS$V2,PoI$V2,PoS$V2,SwS$V2))

NEW=list("Florida-I"=FlI[,4][FlI$V2 %in% INTER],"Florida-S"=FlS[,4][FlS$V2 %in% INTER],"Maine-S"=MaS[,4][MaS$V2 %in% INTER],"Zambia-I"=ZaI[,4][ZaI$V2 %in% INTER],"Zambia-S"=ZaS[,4][ZaS$V2 %in% INTER],"Portugal-I"=PoI[,4][PoI$V2 %in% INTER],"Portugal-S"=PoS[,4][PoS$V2 %in% INTER],"Sweden-S"=SwS[,4][SwS$V2 %in% INTER])

## plot AFs in all populations
pdf("/data/InvAlleles/Inversion-specific-AF.pdf",width=12,height=6)
boxplot(NEW,ylab="Allele Frequencies")
dev.off()

## add position information and write to file
NEW=data.frame(list("POS"=FlI[,2][FlI$V2 %in% INTER],"Florida-I"=FlI[,4][FlI$V2 %in% INTER],"Florida-S"=FlS[,4][FlS$V2 %in% INTER],"Maine-S"=MaS[,4][MaS$V2 %in% INTER],"Zambia-I"=ZaI[,4][ZaI$V2 %in% INTER],"Zambia-S"=ZaS[,4][ZaS$V2 %in% INTER],"Portugal-I"=PoI[,4][PoI$V2 %in% INTER],"Portugal-S"=PoS[,4][PoS$V2 %in% INTER],"Sweden-S"=SwS[,4][SwS$V2 %in% INTER]))

write.table(NEW,"/data/InvAlleles/Inversion-specific-AF.txt",quote=F,row.names=F)

## insolate positions with ABS(STD-INV)&gt;0.5 in Africa
NEW2=NEW[abs(NEW[,5]-NEW[,6])&gt;0.5,]
write.table(NEW2,"/data/InvAlleles/Inversion-specific-AF0.5.txt",quote=F,row.names=F)

H=hist(NEW2[,1],breaks=100,xlim=c(0,32000000))

# plot genomic positions of these SNPS
pdf("/data/InvAlleles/Inversion-specific-AF_dist.pdf",width=12,height=6)
plot(H$mids/1000000,H$counts,type="l",xlim=c(15,27),xlab="Genomic Position (Mbp)",ylab="No. of SNPs")
rect(16.432209,0,24.744010,max(H$counts),border="grey",lty=2)
dev.off()

## test if significant allelic difference between STD and INV in Africa
sink("/data/InvAlleles/Inversion-specific-AF.stat")
wilcox.test(NEW[,5],NEW[,6])
sink()' dir="auto">
                     <pre><span class="pl-v">FlS</span><span class="pl-k">=</span>read.table(<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/America_America_Florida_S.af<span class="pl-pds">"</span></span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>)
<span class="pl-v">FlI</span><span class="pl-k">=</span>read.table(<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/America_America_Florida_I.af<span class="pl-pds">"</span></span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>)
<span class="pl-v">MaS</span><span class="pl-k">=</span>read.table(<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/America_America_Maine_S.af<span class="pl-pds">"</span></span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>)
<span class="pl-v">ZaI</span><span class="pl-k">=</span>read.table(<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/America_Africa_Zambia_I.af<span class="pl-pds">"</span></span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>)
<span class="pl-v">ZaS</span><span class="pl-k">=</span>read.table(<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/America_Africa_Zambia_S.af<span class="pl-pds">"</span></span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>)
<span class="pl-v">PoI</span><span class="pl-k">=</span>read.table(<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/America_Europe_Portugal_I.af<span class="pl-pds">"</span></span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>)
<span class="pl-v">PoS</span><span class="pl-k">=</span>read.table(<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/America_Europe_Portugal_S.af<span class="pl-pds">"</span></span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>)
<span class="pl-v">SwS</span><span class="pl-k">=</span>read.table(<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/America_Europe_Sweden_S.af<span class="pl-pds">"</span></span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">F</span>)

<span class="pl-smi">INTER</span><span class="pl-k">&lt;-</span>Reduce(<span class="pl-smi">intersect</span>,<span class="pl-k">list</span>(<span class="pl-smi">FlS</span><span class="pl-k">$</span><span class="pl-smi">V2</span>,<span class="pl-smi">FlI</span><span class="pl-k">$</span><span class="pl-smi">V2</span>,<span class="pl-smi">MaS</span><span class="pl-k">$</span><span class="pl-smi">V2</span>,<span class="pl-smi">ZaI</span><span class="pl-k">$</span><span class="pl-smi">V2</span>,<span class="pl-smi">ZaS</span><span class="pl-k">$</span><span class="pl-smi">V2</span>,<span class="pl-smi">PoI</span><span class="pl-k">$</span><span class="pl-smi">V2</span>,<span class="pl-smi">PoS</span><span class="pl-k">$</span><span class="pl-smi">V2</span>,<span class="pl-smi">SwS</span><span class="pl-k">$</span><span class="pl-smi">V2</span>))

<span class="pl-v">NEW</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">"</span>Florida-I<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">FlI</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">FlI</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Florida-S<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">FlS</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">FlS</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Maine-S<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">MaS</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">MaS</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Zambia-I<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">ZaI</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">ZaI</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Zambia-S<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">ZaS</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">ZaS</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Portugal-I<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">PoI</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">PoI</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Portugal-S<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">PoS</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">PoS</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Sweden-S<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">SwS</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">SwS</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>])

<span class="pl-c"><span class="pl-c">#</span># plot AFs in all populations</span>
pdf(<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/Inversion-specific-AF.pdf<span class="pl-pds">"</span></span>,<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">12</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">6</span>)
boxplot(<span class="pl-smi">NEW</span>,<span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Allele Frequencies<span class="pl-pds">"</span></span>)
dev.off()

<span class="pl-c"><span class="pl-c">#</span># add position information and write to file</span>
<span class="pl-v">NEW</span><span class="pl-k">=</span><span class="pl-k">data.frame</span>(<span class="pl-k">list</span>(<span class="pl-s"><span class="pl-pds">"</span>POS<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">FlI</span>[,<span class="pl-c1">2</span>][<span class="pl-smi">FlI</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Florida-I<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">FlI</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">FlI</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Florida-S<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">FlS</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">FlS</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Maine-S<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">MaS</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">MaS</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Zambia-I<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">ZaI</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">ZaI</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Zambia-S<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">ZaS</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">ZaS</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Portugal-I<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">PoI</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">PoI</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Portugal-S<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">PoS</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">PoS</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>],<span class="pl-s"><span class="pl-pds">"</span>Sweden-S<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">SwS</span>[,<span class="pl-c1">4</span>][<span class="pl-smi">SwS</span><span class="pl-k">$</span><span class="pl-smi">V2</span> <span class="pl-k">%in%</span> <span class="pl-smi">INTER</span>]))

write.table(<span class="pl-smi">NEW</span>,<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/Inversion-specific-AF.txt<span class="pl-pds">"</span></span>,<span class="pl-v">quote</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>)

<span class="pl-c"><span class="pl-c">#</span># insolate positions with ABS(STD-INV)&gt;0.5 in Africa</span>
<span class="pl-v">NEW2</span><span class="pl-k">=</span><span class="pl-smi">NEW</span>[abs(<span class="pl-smi">NEW</span>[,<span class="pl-c1">5</span>]<span class="pl-k">-</span><span class="pl-smi">NEW</span>[,<span class="pl-c1">6</span>])<span class="pl-k">&gt;</span><span class="pl-c1">0.5</span>,]
write.table(<span class="pl-smi">NEW2</span>,<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/Inversion-specific-AF0.5.txt<span class="pl-pds">"</span></span>,<span class="pl-v">quote</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>)

<span class="pl-v">H</span><span class="pl-k">=</span>hist(<span class="pl-smi">NEW2</span>[,<span class="pl-c1">1</span>],<span class="pl-v">breaks</span><span class="pl-k">=</span><span class="pl-c1">100</span>,<span class="pl-v">xlim</span><span class="pl-k">=</span>c(<span class="pl-c1">0</span>,<span class="pl-c1">32000000</span>))

<span class="pl-c"><span class="pl-c">#</span> plot genomic positions of these SNPS</span>
pdf(<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/Inversion-specific-AF_dist.pdf<span class="pl-pds">"</span></span>,<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">12</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">6</span>)
plot(<span class="pl-smi">H</span><span class="pl-k">$</span><span class="pl-smi">mids</span><span class="pl-k">/</span><span class="pl-c1">1000000</span>,<span class="pl-smi">H</span><span class="pl-k">$</span><span class="pl-smi">counts</span>,<span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>l<span class="pl-pds">"</span></span>,<span class="pl-v">xlim</span><span class="pl-k">=</span>c(<span class="pl-c1">15</span>,<span class="pl-c1">27</span>),<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Genomic Position (Mbp)<span class="pl-pds">"</span></span>,<span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>No. of SNPs<span class="pl-pds">"</span></span>)
rect(<span class="pl-c1">16.432209</span>,<span class="pl-c1">0</span>,<span class="pl-c1">24.744010</span>,max(<span class="pl-smi">H</span><span class="pl-k">$</span><span class="pl-smi">counts</span>),<span class="pl-v">border</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>grey<span class="pl-pds">"</span></span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
dev.off()

<span class="pl-c"><span class="pl-c">#</span># test if significant allelic difference between STD and INV in Africa</span>
sink(<span class="pl-s"><span class="pl-pds">"</span>/data/InvAlleles/Inversion-specific-AF.stat<span class="pl-pds">"</span></span>)
wilcox.test(<span class="pl-smi">NEW</span>[,<span class="pl-c1">5</span>],<span class="pl-smi">NEW</span>[,<span class="pl-c1">6</span>])
sink()</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      3.4) Linkage Disequilibrium
                     </h3>
                     <a aria-label="Permalink: 3.4) Linkage Disequilibrium" class="anchor" href="#34-linkage-disequilibrium" id="user-content-34-linkage-disequilibrium">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Here, we assessed how linkage disequilibrium decays within the genomic region spanned by
                     <em>
                      In(3R)Payne
                     </em>
                     in inverted and non-inverted haplotypes.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      3.4.1) SNP-wise LD with the inversion
                     </h4>
                     <a aria-label="Permalink: 3.4.1) SNP-wise LD with the inversion" class="anchor" href="#341-snp-wise-ld-with-the-inversion" id="user-content-341-snp-wise-ld-with-the-inversion">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     First, we calculated and plotted LD of each SNP with
                     <em>
                      In(3R)Payne
                     </em>
                     along the 3R chromosomal arm with the Inversion.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
mkdir /data/LD

## Africa
python3 /scripts/v3/LD_bar.py \
--input /data/Africa/Africa.cons.gz \
--output /data/LD/Zambia_Bar \
--ind ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388 \
--chromosome 3R \
--N-cutoff 0.1 \
--color blue,green,yellow,red,purple \
--min-allele 0.1 \
--Inv /data/Africa_Zambia_Inv.samples \
--Std /data/Africa_Zambia_Std.samples \

## Europe
python3 /scripts/v3/LD_bar.py \
--input /data/Europe/Europe.cons.gz \
--output /data/LD/Portugal_Bar \
--ind F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94 \
--chromosome 3R \
--N-cutoff 0.1 \
--color blue,green,yellow,red,purple \
--min-allele 0.1 \
--Inv /data/Europe_Portugal_Inv.samples \
--Std /data/Europe_Portugal_Std.samples \

## America
python3 /scripts/v3/LD_bar.py \
--input /data/USA/USA.cons.gz \
--output /data/LD/USA_Bar \
--ind Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81 \
--chromosome 3R \
--N-cutoff 0.1 \
--color blue,green,yellow,red,purple \
--min-allele 0.1 \
--Inv /data/America_Florida_Inv.samples \
--Std /data/America_Florida_Std.samples \

## Australia
python3 /scripts/v3/LD_bar.py \
--input /data/Australia/Australia.cons.gz \
--output /data/LD/Australia_Bar \
--ind Ii1,Ii2,Ii3,Ii4,Ii5,Ii6,Ii7,Ii8,Ii9,Ii10,Ii11,Ii12,Ii13,Ii14,Ii15,Ii16,Ii17,Ii18,Ii19,Is1,Is2,Is3,Is4,Is5,Is6,Is7,Is8,Is9,Is10,Is11,Is12,Is13,Is14,Is15,Is16,Is17,Is18,YS1,YS2,YS3,YS4,YS5,YS6,YS7,YS8,YS9,YS10,YS11,YS12,YS13,YS14,YS15,YS16,YS17,YS18 \
--chromosome 3R \
--N-cutoff 0.1 \
--color blue,green,yellow,red,purple \
--min-allele 0.1 \
--Inv /data/Australia_Queensland_Inv.samples \
--Std /data/Australia_Queensland_Std.samples \
" dir="auto">
                     <pre>mkdir /data/LD

<span class="pl-c"><span class="pl-c">#</span># Africa</span>
python3 /scripts/v3/LD_bar.py \
--input /data/Africa/Africa.cons.gz \
--output /data/LD/Zambia_Bar \
--ind ZI81,ZI508,ZI505,ZI486,ZI448,ZI446,ZI405,ZI397N,ZI384,ZI362,ZI341,ZI319,ZI293,ZI292,ZI291,ZI264,ZI237,ZI233,ZI228,ZI226,ZI221,ZI220,ZI99,ZI91,ZI86,ZI85,ZI76,ZI68,ZI61,ZI530,ZI527,ZI524,ZI517,ZI514N,ZI50N,ZI504,ZI491,ZI490,ZI488,ZI477,ZI472,ZI471,ZI468,ZI467,ZI466,ZI460,ZI458,ZI457,ZI456,ZI455N,ZI453,ZI447,ZI445,ZI444,ZI433,ZI431,ZI421,ZI418N,ZI413,ZI402,ZI398,ZI395,ZI394N,ZI392,ZI386,ZI378,ZI377,ZI374,ZI373,ZI370,ZI365,ZI364,ZI359,ZI358,ZI357N,ZI348,ZI344,ZI342,ZI339,ZI336,ZI335,ZI332,ZI33,ZI329,ZI324,ZI321,ZI320,ZI31N,ZI316,ZI314,ZI313,ZI311N,ZI303,ZI295,ZI286,ZI281,ZI28,ZI279,ZI276,ZI271,ZI27,ZI269,ZI268,ZI265,ZI263,ZI261,ZI26,ZI255,ZI254N,ZI253,ZI252,ZI251N,ZI250,ZI240,ZI239,ZI235,ZI232,ZI231,ZI225,ZI219,ZI216N,ZI213,ZI212,ZI211,ZI210,ZI207,ZI206,ZI200,ZI198,ZI196,ZI194,ZI193,ZI191,ZI185,ZI184,ZI182,ZI181,ZI179,ZI178,ZI176,ZI173,ZI172,ZI170,ZI167,ZI165,ZI164,ZI161,ZI157,ZI152,ZI138,ZI136,ZI134N,ZI129,ZI126,ZI118N,ZI117,ZI114N,ZI104,ZI10,ZI56,ZI420,ZI388 \
--chromosome 3R \
--N-cutoff 0.1 \
--color blue,green,yellow,red,purple \
--min-allele 0.1 \
--Inv /data/Africa_Zambia_Inv.samples \
--Std /data/Africa_Zambia_Std.samples \

<span class="pl-c"><span class="pl-c">#</span># Europe</span>
python3 /scripts/v3/LD_bar.py \
--input /data/Europe/Europe.cons.gz \
--output /data/LD/Portugal_Bar \
--ind F0-8-1,F0-11-1,F0-16-1,F0-21-1,F0-24-1,F0-25-1,hot-168-1,cold-21-0,cold-89-0,cold-91-0,F0-2-0,F0-3-0,F0-4-0,F0-5-0,F0-6-0,F0-7-0,F0-9-0,F0-10-0,F0-12-0,F0-13-0,F0-14-0,F0-15-0,F0-17-0,F0-18-0,F0-27-0,F0-29-0,SU02n,SU05n,SU07n,SU08,SU21n,SU25n,SU26n,SU29,SU37n,SU58n,SU75n,SU81n,SU93n,SU94 \
--chromosome 3R \
--N-cutoff 0.1 \
--color blue,green,yellow,red,purple \
--min-allele 0.1 \
--Inv /data/Europe_Portugal_Inv.samples \
--Std /data/Europe_Portugal_Std.samples \

<span class="pl-c"><span class="pl-c">#</span># America</span>
python3 /scripts/v3/LD_bar.py \
--input /data/USA/USA.cons.gz \
--output /data/LD/USA_Bar \
--ind Florida_S_1142,Florida_S_1145,Florida_S_1153,Florida_S_1155,Florida_S_1156,Florida_S_1157,Florida_S_1163,Florida_S_1164,Florida_S_1167,Florida_S_1218,Florida_S_1189,Florida_S_1170,Florida_S_1178,Florida_S_1203,Florida_S_1204,Florida_S_1158,Florida_S_1149,Florida_S_1174,Florida_S_1160,Florida_I_1153,Florida_I_1165,Florida_I_1169,Florida_I_1203,Florida_I_1218,Florida_I_1142,Florida_I_1146,Florida_I_1147,Florida_I_1149,Florida_I_1150,Florida_I_1178,Florida_I_1143,Florida_I_1156,Florida_I_1160,Florida_I_1161,Florida_I_1162,Florida_I_1164,Florida_I_1174,Florida_I_1152,Florida_I_1158,Maine_S_10-96,Maine_S_10-95,Maine_S_10-82,Maine_S_10-53,Maine_S_10-73,Maine_S_10-24,Maine_S_10-72,Maine_S_10-12,Maine_S_10-77,Maine_S_10-89,Maine_S_10-76,Maine_S_10-69,Maine_S_10-93,Maine_S_10-57,Maine_S_10-58,Maine_S_10-60,Maine_S_10-67,Maine_S_10-84,Maine_S_10-79,Maine_S_10-81 \
--chromosome 3R \
--N-cutoff 0.1 \
--color blue,green,yellow,red,purple \
--min-allele 0.1 \
--Inv /data/America_Florida_Inv.samples \
--Std /data/America_Florida_Std.samples \

<span class="pl-c"><span class="pl-c">#</span># Australia</span>
python3 /scripts/v3/LD_bar.py \
--input /data/Australia/Australia.cons.gz \
--output /data/LD/Australia_Bar \
--ind Ii1,Ii2,Ii3,Ii4,Ii5,Ii6,Ii7,Ii8,Ii9,Ii10,Ii11,Ii12,Ii13,Ii14,Ii15,Ii16,Ii17,Ii18,Ii19,Is1,Is2,Is3,Is4,Is5,Is6,Is7,Is8,Is9,Is10,Is11,Is12,Is13,Is14,Is15,Is16,Is17,Is18,YS1,YS2,YS3,YS4,YS5,YS6,YS7,YS8,YS9,YS10,YS11,YS12,YS13,YS14,YS15,YS16,YS17,YS18 \
--chromosome 3R \
--N-cutoff 0.1 \
--color blue,green,yellow,red,purple \
--min-allele 0.1 \
--Inv /data/Australia_Queensland_Inv.samples \
--Std /data/Australia_Queensland_Std.samples \
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      3.4.2) LD heatmaps
                     </h4>
                     <a aria-label="Permalink: 3.4.2) LD heatmaps" class="anchor" href="#342-ld-heatmaps" id="user-content-342-ld-heatmaps">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Next, we calculated LD heatmaps within an among karyotypes focusing on a subset of 5000 SNPs from populations with mixed karyotypes in different continents. The numbers of the parameter
                     <code>
                      ind
                     </code>
                     correspond to the (zero-based) position of the chosen individuals (from *.samples files) in the consensus block. See
                     <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/mec.12594" rel="nofollow">
                      Kapun
                      <em>
                       et al.
                      </em>
                      2014
                     </a>
                     for more details. Below, we are showing the code for the LD analysis within karyotypes, the
                     <code>
                      ind
                     </code>
                     parameter needs to be adjusted (based on the *.sample files in data/) when repeating the analyses for LD within karyotypes.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
## Zambia - full
python3 /scripts/LD_heatmap.py \
    --input /data/Africa/Africa.cons.gz \
    --output /data/LD/Zambia \
    --subsample 5000 \
    --ind 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,20,21,31,32,33,51,54,72,74,76,90,94,105,114,116,119,120,126,135,137,146,151,158 \
    --chromosome 3R \
    --N-cutoff 0.1 \
    --color purple,red,yellow,green,blue \
    --min-allele 0.3

## Portugal - full 
python3 /scripts/LD_heatmap.py \
    --input /data/Europe/Europe.cons.gz \
    --output /data/LD/Portugal \
    --subsample 5000 \
    --ind 0,1,2,3,4,5,6,10,11,13,17,22,23,25 \
    --chromosome 3R \
    --N-cutoff 0.1 \
    --color purple,red,yellow,green,blue \
    --min-allele 0.3

## Florida - full 
python3 /scripts/LD_heatmap.py \
    --input /data/USA/USA.cons.gz \
    --output /data/LD/Florida \
    --subsample 5000 \
    --ind 0,1,2,3,4,5,6,7,8,9,10,11,13,14,15,17,18,19,21,22,23,24,25,26,27,29,30,31,32,33,34,35,36,37,38 \
    --chromosome 3R \
    --N-cutoff 0.1 \
    --color purple,red,yellow,green,blue \
    --min-allele 0.3

## Australia - full 
python3 /scripts/LD_heatmap.py \
    --input /data/Australia/Australia.cons.gz \
    --output /data/LD/Queensland \
    --subsample 5000 \
    --ind 0,1,3,6,9,10,11,15,16,17,25,32,35,36,12,13,14,19,21,24,26,27,29,31,33 \
    --chromosome 3R \
    --N-cutoff 0.1 \
    --color purple,red,yellow,green,blue \
    --min-allele 0.3" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># Zambia - full</span>
python3 /scripts/LD_heatmap.py \
    --input /data/Africa/Africa.cons.gz \
    --output /data/LD/Zambia \
    --subsample 5000 \
    --ind 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,20,21,31,32,33,51,54,72,74,76,90,94,105,114,116,119,120,126,135,137,146,151,158 \
    --chromosome 3R \
    --N-cutoff 0.1 \
    --color purple,red,yellow,green,blue \
    --min-allele 0.3

<span class="pl-c"><span class="pl-c">#</span># Portugal - full </span>
python3 /scripts/LD_heatmap.py \
    --input /data/Europe/Europe.cons.gz \
    --output /data/LD/Portugal \
    --subsample 5000 \
    --ind 0,1,2,3,4,5,6,10,11,13,17,22,23,25 \
    --chromosome 3R \
    --N-cutoff 0.1 \
    --color purple,red,yellow,green,blue \
    --min-allele 0.3

<span class="pl-c"><span class="pl-c">#</span># Florida - full </span>
python3 /scripts/LD_heatmap.py \
    --input /data/USA/USA.cons.gz \
    --output /data/LD/Florida \
    --subsample 5000 \
    --ind 0,1,2,3,4,5,6,7,8,9,10,11,13,14,15,17,18,19,21,22,23,24,25,26,27,29,30,31,32,33,34,35,36,37,38 \
    --chromosome 3R \
    --N-cutoff 0.1 \
    --color purple,red,yellow,green,blue \
    --min-allele 0.3

<span class="pl-c"><span class="pl-c">#</span># Australia - full </span>
python3 /scripts/LD_heatmap.py \
    --input /data/Australia/Australia.cons.gz \
    --output /data/LD/Queensland \
    --subsample 5000 \
    --ind 0,1,3,6,9,10,11,15,16,17,25,32,35,36,12,13,14,19,21,24,26,27,29,31,33 \
    --chromosome 3R \
    --N-cutoff 0.1 \
    --color purple,red,yellow,green,blue \
    --min-allele 0.3</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      3.4.3) Decay of LD
                     </h4>
                     <a aria-label="Permalink: 3.4.3) Decay of LD" class="anchor" href="#343-decay-of-ld" id="user-content-343-decay-of-ld">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Next, we estimated differences in the decay of LD with respect to different karyotypes based on the
                     <code>
                      *.dist.gz
                     </code>
                     output files from the analyses above and tested for significant differences with non-linear mixed models.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='
## loop through samples
continent=( "Africa" "Europe" "America" "Australia" )
region=( "Zambia" "Portugal" "Florida" "Queensland" )
Pop1=( "Africa_Zambia_Inv" "Europe_Portugal_Inv"  "America_Florida_Inv"  "Australia_Queensland_Inv"  )
Pop2=( "Africa_Zambia_Std" "Europe_Portugal_Std" "America_Florida_Std" "Australia_Queensland_Std" )

for index in ${!Pop1[@]}
do  

    ## plot the decay
    python3 /scripts/v3/LD_decay.py \
        --vcf /data/consensus/${continent[index]}.vcf.gz \
        --samples /data/${Pop1[index]}.samples,/data/${Pop2[index]}.samples \
        --names I,S \
        --combsample 50000 \
        --SNPs 5000 \
        --maxdist 100000 \
        --threshold 0.1 \
        --N-threshold 0.5 \
        --region 3R:16432209-24744010 \
        --out /data/LD/${region[index]}_decay

    ## test for significant differences
    Rscript /scripts/NLS.R /data/LD/${region[index]}_decay

done' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># loop through samples</span>
continent=( <span class="pl-s"><span class="pl-pds">"</span>Africa<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia<span class="pl-pds">"</span></span> )
region=( <span class="pl-s"><span class="pl-pds">"</span>Zambia<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Portugal<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Florida<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Queensland<span class="pl-pds">"</span></span> )
Pop1=( <span class="pl-s"><span class="pl-pds">"</span>Africa_Zambia_Inv<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe_Portugal_Inv<span class="pl-pds">"</span></span>  <span class="pl-s"><span class="pl-pds">"</span>America_Florida_Inv<span class="pl-pds">"</span></span>  <span class="pl-s"><span class="pl-pds">"</span>Australia_Queensland_Inv<span class="pl-pds">"</span></span>  )
Pop2=( <span class="pl-s"><span class="pl-pds">"</span>Africa_Zambia_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Europe_Portugal_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>America_Florida_Std<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>Australia_Queensland_Std<span class="pl-pds">"</span></span> )

<span class="pl-k">for</span> <span class="pl-smi">index</span> <span class="pl-k">in</span> <span class="pl-smi">${<span class="pl-k">!</span>Pop1[@]}</span>
<span class="pl-k">do</span>  

    <span class="pl-c"><span class="pl-c">#</span># plot the decay</span>
    python3 /scripts/v3/LD_decay.py \
        --vcf /data/consensus/<span class="pl-smi">${continent[index]}</span>.vcf.gz \
        --samples /data/<span class="pl-smi">${Pop1[index]}</span>.samples,/data/<span class="pl-smi">${Pop2[index]}</span>.samples \
        --names I,S \
        --combsample 50000 \
        --SNPs 5000 \
        --maxdist 100000 \
        --threshold 0.1 \
        --N-threshold 0.5 \
        --region 3R:16432209-24744010 \
        --out /data/LD/<span class="pl-smi">${region[index]}</span>_decay

    <span class="pl-c"><span class="pl-c">#</span># test for significant differences</span>
    Rscript /scripts/NLS.R /data/LD/<span class="pl-smi">${region[index]}</span>_decay

<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4) Transcriptomic analysis
                     </h3>
                     <a aria-label="Permalink: 4) Transcriptomic analysis" class="anchor" href="#4-transcriptomic-analysis" id="user-content-4-transcriptomic-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      4.1) Trim and map RNASeq data
                     </h4>
                     <a aria-label="Permalink: 4.1) Trim and map RNASeq data" class="anchor" href="#41-trim-and-map-rnaseq-data" id="user-content-41-trim-and-map-rnaseq-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## loop through all libraries and trim

mkdir -p /data/RNASeq/trimmed
for i in /data/RNASeq/*.fq.gz
do

    ## use only a subset of the variable i as the name
    name=${i##*/}
    n=&quot;$( cut -d '.' -f 1 &lt;&lt;&lt; &quot;$name&quot; )&quot;

    ## trim with cutadapt
    /scripts/cutadapt \
    -q 18 \
    --minimum-length 75 \
    -o /data/RNASeq/trimmed/$n.fq.gz \
    ${i} 

done" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># loop through all libraries and trim</span>

mkdir -p /data/RNASeq/trimmed
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> /data/RNASeq/<span class="pl-k">*</span>.fq.gz
<span class="pl-k">do</span>

    <span class="pl-c"><span class="pl-c">#</span># use only a subset of the variable i as the name</span>
    name=<span class="pl-smi">${i<span class="pl-k">##*/</span>}</span>
    n=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-s"><span class="pl-pds">$(</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>.<span class="pl-pds">'</span></span> -f 1 <span class="pl-k">&lt;&lt;&lt;</span> <span class="pl-s"><span class="pl-pds">"</span>$name<span class="pl-pds">"</span></span> <span class="pl-pds">)</span></span><span class="pl-pds">"</span></span>

    <span class="pl-c"><span class="pl-c">#</span># trim with cutadapt</span>
    /scripts/cutadapt \
    -q 18 \
    --minimum-length 75 \
    -o /data/RNASeq/trimmed/<span class="pl-smi">$n</span>.fq.gz \
    <span class="pl-smi">${i}</span> 

<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Get reference transcriptome
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /data/RNASeq/
curl -O ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/dmel_r6.17_FB2017_04/fasta/dmel-all-CDS-r6.17.fasta.gz
curl -O ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/dmel_r6.17_FB2017_04/fasta/dmel-all-r6.17.gtf.gz" dir="auto">
                     <pre><span class="pl-c1">cd</span> /data/RNASeq/
curl -O ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/dmel_r6.17_FB2017_04/fasta/dmel-all-CDS-r6.17.fasta.gz
curl -O ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/dmel_r6.17_FB2017_04/fasta/dmel-all-r6.17.gtf.gz</pre>
                    </div>
                    <p dir="auto">
                     Next, we pseudomapped the trimmed reads with
                     <a href="https://github.com/pachterlab/kallisto">
                      Kallisto
                     </a>
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## index the transcriptome
/scripts/kallisto index \
    -i /data/RNASeq/dmel-all-CDS-r6.17.idx \
    /data/RNASeq/dmel-all-CDS-r6.17.fasta.gz

mkdir /data/RNASeq/kallisto

## loop through all libraries and map with kallisto
for i in /data/RNASeq/*.fq.gz
do

    name=${i##*/}
    n=&quot;$( cut -d '.' -f 1 &lt;&lt;&lt; &quot;$name&quot; )&quot;

    kallisto quant \
    -i /data/RNASeq/dmel-all-CDS-r6.17.idx \
    -o /data/RNASeq/kallisto/$n \
    --single -l 101 -s 10 -b 100 --rf-stranded -t 24 \
    /data/RNASeq/trimmed/$name \

done" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># index the transcriptome</span>
/scripts/kallisto index \
    -i /data/RNASeq/dmel-all-CDS-r6.17.idx \
    /data/RNASeq/dmel-all-CDS-r6.17.fasta.gz

mkdir /data/RNASeq/kallisto

<span class="pl-c"><span class="pl-c">#</span># loop through all libraries and map with kallisto</span>
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> /data/RNASeq/<span class="pl-k">*</span>.fq.gz
<span class="pl-k">do</span>

    name=<span class="pl-smi">${i<span class="pl-k">##*/</span>}</span>
    n=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-s"><span class="pl-pds">$(</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>.<span class="pl-pds">'</span></span> -f 1 <span class="pl-k">&lt;&lt;&lt;</span> <span class="pl-s"><span class="pl-pds">"</span>$name<span class="pl-pds">"</span></span> <span class="pl-pds">)</span></span><span class="pl-pds">"</span></span>

    kallisto quant \
    -i /data/RNASeq/dmel-all-CDS-r6.17.idx \
    -o /data/RNASeq/kallisto/<span class="pl-smi">$n</span> \
    --single -l 101 -s 10 -b 100 --rf-stranded -t 24 \
    /data/RNASeq/trimmed/<span class="pl-smi">$name</span> \

<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     After that, we merged the read counts at gene level across all datasets
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
## merge datasets and sum up Pseudocounts for gene-level estimates
mkdir -p /data/RNASeq/count

python3 /scripts/v3/mergeKallisto.py \
    /data/RNASeq/kallisto \
    /data/RNASeq/count/analyses \
    /data/RNASeq/dmel-all-r6.17.gtf.gz" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># merge datasets and sum up Pseudocounts for gene-level estimates</span>
mkdir -p /data/RNASeq/count

python3 /scripts/v3/mergeKallisto.py \
    /data/RNASeq/kallisto \
    /data/RNASeq/count/analyses \
    /data/RNASeq/dmel-all-r6.17.gtf.gz</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      4.2) Test for differential expression
                     </h4>
                     <a aria-label="Permalink: 4.2) Test for differential expression" class="anchor" href="#42-test-for-differential-expression" id="user-content-42-test-for-differential-expression">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Then, we used
                     <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html" rel="nofollow">
                      edgeR
                     </a>
                     and
                     <a href="https://bioconductor.org/packages/release/bioc/html/limma.html" rel="nofollow">
                      limma
                     </a>
                     to test for differential expression (DE) among different factors and test for enrichment of Gene Ontology (GO) terms.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="### Analysis of RNA-Seq data of inversion lines

## install packages
#install.packages(&quot;readr&quot;)
#install.packages(&quot;tximport&quot;)
#install.packages(&quot;edgeR&quot;)
#install.packages(&quot;RColorBrewer&quot;)
#install.packages(&quot;limma&quot;)
#install.packages(&quot;pheatmap&quot;)

I=&quot;/data/RNASeq/count/analyses_counts.txt&quot;
O=&quot;/data/RNASeq/&quot;

#Factors

Treat=factor(c('FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FI','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','FS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS','MS'))
Treat=relevel(Treat,ref=&quot;FS&quot;)
Temp=factor(c('18','18','18','18','18','18','18','18','18','25','25','25','25','25','25','25','25','25','18','18','18','18','18','18','18','18','18','25','25','25','25','25','25','25','25','25','18','18','18','18','18','18','18','18','18','25','25','25','25','25','25','25','25','25'))
Rep=factor(c(10,14,17,18,22,23,26,5,9,5,9,10,14,17,18,22,23,26,13,16,2,21,27,3,4,6,7,2,3,4,6,7,13,16,21,27,1,11,12,15,19,20,24,25,8,1,8,11,12,15,19,20,24,25))

## 1a) read Data
rawData&lt;-read.table(I,header=T)
rawCounts=rawData[,3:56]
rownames(rawCounts)&lt;-rawData[,1]

## 2) create DGE object using edgeR
library(edgeR)
FullDGE&lt;-DGEList(rawCounts,group=names(rawCounts))

## 3) add factors to object
FullDGE$samples$treat &lt;- Treat
FullDGE$samples$temp &lt;- Temp

## Transform counts (Counts per million: CPM) and normalize the counts using the TMM method in edgeR
FullDGE &lt;- calcNormFactors(FullDGE, method = &quot;TMM&quot;)
cpmFullDGE&lt;-cpm(FullDGE,log=T)

out=paste(O,&quot;/countdata/CPM.txt&quot;,sep=&quot;&quot;)
TAB=cpmFullDGE
write.table(data.frame(&quot;Gene&quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)

## remove genes covered by less than 2 read per million in at least nine samples
keep &lt;- rowSums(cpmFullDGE &gt;= 2) &gt;=9
FullDGE &lt;- FullDGE[keep,] 
cpmTFullDGE &lt;- cpmFullDGE[keep,] 

## Plot density histograms to quantify the effect of filtering out lowly expressed genes
dir.create(paste(O,&quot;/quality&quot;,sep=&quot;&quot;))
pdf(paste(O,&quot;/quality/CPM-hist.pdf&quot;,sep=&quot;&quot;),width=15,height=5)
nsamples &lt;- ncol(FullDGE)
col &lt;- topo.colors(nsamples)
par(mfrow=c(1,2))
plot(density(cpmFullDGE[,1]), col=col[1], lwd=2, ylim=c(0,0.25), las=2,
     main=&quot;&quot;, xlab=&quot;&quot;)
abline(v=0, lty=3)

for (i in 2:nsamples){
  den &lt;- density(cpmFullDGE[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
} 
legend(&quot;topright&quot;, legend=colnames(FullDGE), text.col=col, bty=&quot;n&quot;,ncol = 3,cex=0.5)
title(main=&quot;A. Raw data&quot;, xlab=&quot;Log-cpm&quot;)

plot(density(cpmTFullDGE[,1]), col=col[1], lwd=2, ylim=c(0,0.25), las=2,
     main=&quot;&quot;, xlab=&quot;&quot;)
title(main=&quot;B. Filtered data&quot;, xlab=&quot;Log-cpm&quot;) 
abline(v=0, lty=3)

for (i in 2:nsamples){
  den &lt;- density(cpmTFullDGE[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
} 
legend(&quot;topright&quot;, legend=colnames(FullDGE), text.col=col, bty=&quot;n&quot;,ncol = 3,cex=0.5)
dev.off()

## explore Data with MDS to identify clustering
library(RColorBrewer)
pdf(paste(O,&quot;/quality/MDS.pdf&quot;,sep=&quot;&quot;),width=14,height=12)
par(mfrow=c(2,2))

col.Treat &lt;- Treat
levels(col.Treat) &lt;- brewer.pal(nlevels(col.Treat), &quot;Set1&quot;)
plotMDS(cpmTFullDGE, col=as.character(col.Treat))
title(main=&quot;Karypop: Dim 1/2&quot;)

col.Temp &lt;- Temp
levels(col.Temp) &lt;- brewer.pal(nlevels(col.Temp), &quot;Set1&quot;)
plotMDS(cpmTFullDGE, col=as.character(col.Temp))
title(main=&quot;Temperature: Dim 1/2&quot;)

plotMDS(cpmTFullDGE, col=as.character(col.Treat), dim=c(3,4))
title(main=&quot;Karypop: Dim 3/4&quot;)

plotMDS(cpmTFullDGE, col=as.character(col.Temp), dim=c(3,4))
title(main=&quot;Temperature: Dim 3/4&quot;)

dev.off()

### explore data with PCA to explore clustering

pdf(paste(O,&quot;/quality/PCA.pdf&quot;,sep=&quot;&quot;),width=12,height=8)
par(mfrow=c(2,3))
newcpm=t(cpmTFullDGE)
PCA=prcomp(newcpm)
colKP=rainbow(length(levels(Treat)))[Treat]
colTemp=rainbow(length(levels(Temp)))[as.factor(Temp)]
colPop=rainbow(length(levels(Rep)))[as.factor(Rep)]

plot(PCA$x[,1],PCA$x[,2],col=colKP,main=&quot;Karypop&quot;,pch=16,cex=3)
#text(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colKP)

plot(PCA$x[,1],PCA$x[,2],col=colTemp,main=&quot;Temp&quot;,pch=16,cex=3)
#text(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colTemp)

plot(PCA$x[,1],PCA$x[,2],col=colPop,main=&quot;Replicate&quot;,pch=16,cex=3)
#text(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colPop)

plot(PCA$x[,3],PCA$x[,4],col=colKP,main=&quot;Karypop&quot;,pch=16,cex=3)
#text(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colKP)

plot(PCA$x[,3],PCA$x[,4],col=colTemp,main=&quot;Temp&quot;,pch=16,cex=3)
#text(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colTemp)

plot(PCA$x[,3],PCA$x[,4],col=colPop,main=&quot;Replicate&quot;,pch=16,cex=3)
#text(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colPop)

dev.off()

### Limma analysis with singe grouping factor

library(limma)

Treat=relevel(Treat,ref=&quot;FS&quot;)
dir.create(paste(O,&quot;/limma&quot;,sep=&quot;&quot;))
G=interaction(Treat,Temp)

design.SP=model.matrix(~0+G)
colnames(design.SP)=levels(G)

## Do VOOM
voomFullDGE.SP &lt;- voom(FullDGE, design.SP, plot=F)

# write transformed data to file
out=paste(O,&quot;/countdata/voomCounts.txt&quot;,sep=&quot;&quot;)
TAB=voomFullDGE.SP$E
write.table(data.frame(&quot;Gene&quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)

corfit.SP &lt;- duplicateCorrelation(voomFullDGE.SP,design.SP,block=Rep)
vfit.SP &lt;- lmFit(voomFullDGE.SP, design.SP, block=Rep, correlation=corfit.SP$consensus)

############ Karypop

cont.matrix.SP.Treat &lt;- makeContrasts(
  &quot;FS-FI&quot;=(FS.18+FS.25)/2-(FI.18+FI.25)/2,
  &quot;FS-MS&quot;=(FS.18+FS.25)/2-(MS.18+MS.25)/2,
  &quot;MS-FI&quot;=(MS.18+MS.25)/2-(FI.18+FI.25)/2,
  levels=design.SP)

# now calculate the contrasts:
vfit.Treat &lt;- contrasts.fit(vfit.SP, cont.matrix.SP.Treat)
efit.Treat &lt;- eBayes(vfit.Treat)

# now export the tables with the p-values and log2-fc for each of the contrasts
for (name in colnames(cont.matrix.SP.Treat)){
  # print the name
  print(name)
  
  # make an output file in outputfolder
  out=paste(O,&quot;/limma/&quot;,name,&quot;.txt&quot;,sep=&quot;&quot;)
  
  # generate summary table with Benjamini Hochberg adjsuted p-values for all genes
  TAB=topTable(efit.Treat,coef=name, adjust.method=&quot;BH&quot;, n=Inf)
  
  # export table to text file
  write.table(data.frame(&quot;Gene&quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)
}

# another cool thing about eBayes is that it calculates an F-value testing if the contrasts are different from 0. In the above case we are comparing all levels for the factor &quot;Treat&quot; only. Ebays combines the t-statistics for all the contrasts into an overall test of significance for that gene. The resulting F-value will thus tell us if &quot;Treat&quot; is significantly differnet from 0. See limma manual page 61. Note that I append another column to the dataframe with the P-value adjusted a la Benjamini-Hochberg
efit.F.Treat=data.frame(&quot;F&quot;=efit.Treat$F,&quot;Pval&quot;=efit.Treat$F.p.value,&quot;Pval.BH&quot;=p.adjust(efit.Treat$F.p.value, method=&quot;BH&quot;))
rownames(efit.F.Treat)=rownames(efit.Treat)

# export table as above for the contrasts

# make an output file in outputfolder
out=paste(O,&quot;/limma/Treat-F-value.txt&quot;,sep=&quot;&quot;)

# export table to text file
write.table(data.frame(&quot;Gene&quot;=rownames(efit.F.Treat),efit.F.Treat),file=out,row.names=F,quote = F)

############ Temp

cont.matrix.SP.Temp &lt;- makeContrasts(
  &quot;T18-25&quot;=(FS.18+FI.18+MS.18)/3-(FS.25+FI.25+MS.25)/3,
  levels=design.SP)

# now calculate the contrasts:
vfit.Temp &lt;- contrasts.fit(vfit.SP, cont.matrix.SP.Temp)
efit.Temp &lt;- eBayes(vfit.Temp)

# now export the tables with the p-values and log2-fc for each of the contrasts
for (name in colnames(cont.matrix.SP.Temp)){
  # print the name
  print(name)
  
  # make an output file in outputfolder
  out=paste(O,&quot;/limma/&quot;,name,&quot;.txt&quot;,sep=&quot;&quot;)
  
  # generate summary table with Benjamini Hochberg adjsuted p-values for all genes
  TAB=topTable(efit.Temp,coef=name, adjust.method=&quot;BH&quot;, n=Inf)
  
  # export table to text file
  write.table(data.frame(&quot;Gene&quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)
}

efit.F.Temp=data.frame(&quot;F&quot;=efit.Temp$F,&quot;Pval&quot;=efit.Temp$F.p.value,&quot;Pval.BH&quot;=p.adjust(efit.Temp$F.p.value, method=&quot;BH&quot;))
rownames(efit.F.Temp)=rownames(efit.Temp)

# export table as above for the contrasts

# make an output file in outputfolder
out=paste(O,&quot;/limma/Temp-F-value.txt&quot;,sep=&quot;&quot;)

# export table to text file
write.table(data.frame(&quot;Gene&quot;=rownames(efit.F.Temp),efit.F.Temp),file=out,row.names=F,quote = F)

cont.matrix.SP.interactions &lt;- makeContrasts(
  &quot;FS-FI_int&quot;=(FS.18-FS.25)-(FI.18-FI.25),
  &quot;FI-MS_int&quot;=(FI.18-FI.25)-(MS.18-MS.25),
  &quot;FS-MS_int&quot;=(FS.18-FS.25)-(MS.18-MS.25),
  levels=design.SP)

# now calculate the contrasts:
vfit.interactions &lt;- contrasts.fit(vfit.SP, cont.matrix.SP.interactions)
efit.interactions &lt;- eBayes(vfit.interactions)

# now export the tables with the p-values and log2-fc for each of the contrasts
for (name in colnames(cont.matrix.SP.interactions)){
  # print the name
  print(name)
  
  # make an output file in outputfolder
  out=paste(O,&quot;/limma/&quot;,name,&quot;.txt&quot;,sep=&quot;&quot;)
  
  # generate summary table with Benjamini Hochberg adjsuted p-values for all genes
  TAB=topTable(efit.interactions,coef=name, adjust.method=&quot;BH&quot;, n=Inf)
  
  # export table to text file
  write.table(data.frame(&quot;Gene&quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)
}

######## Interactions

efit.F.interactions=data.frame(&quot;F&quot;=efit.interactions$F,&quot;Pval&quot;=efit.interactions$F.p.value,&quot;Pval.BH&quot;=p.adjust(efit.interactions$F.p.value, method=&quot;BH&quot;))
rownames(efit.F.interactions)=rownames(efit.interactions)

# export table as above for the contrasts

# make an output file in outputfolder
out=paste(O,&quot;/limma/interactions-F-value.txt&quot;,sep=&quot;&quot;)

# export table to text file
write.table(data.frame(&quot;Gene&quot;=rownames(efit.F.interactions),efit.F.interactions),file=out,row.names=F,quote = F)

############ Other comparisons

cont.matrix.SP.other &lt;- makeContrasts(
  &quot;FS-FI_18&quot;=FS.18-FI.18,
  &quot;FS-MS_18&quot;=FS.18-MS.18,
  &quot;MS-FI_18&quot;=MS.18-FI.18,
  &quot;FS-FI_25&quot;=FS.25-FI.25,
  &quot;FS-MS_25&quot;=FS.25-MS.25,
  &quot;MS-FI_25&quot;=MS.25-FI.25,
  levels=design.SP)

# now calculate the contrasts:
vfit.other &lt;- contrasts.fit(vfit.SP, cont.matrix.SP.other)
efit.other &lt;- eBayes(vfit.other)

# now export the tables with the p-values and log2-fc for each of the contrasts
for (name in colnames(cont.matrix.SP.other)){
  # print the name
  print(name)
  
  # make an output file in outputfolder
  out=paste(O,&quot;/limma/&quot;,name,&quot;.txt&quot;,sep=&quot;&quot;)
  
  # generate summary table with Benjamini Hochberg adjsuted p-values for all genes
  TAB=topTable(efit.other,coef=name, adjust.method=&quot;BH&quot;, n=Inf)
  
  # export table to text file
  write.table(data.frame(&quot;Gene&quot;=rownames(TAB),TAB),file=out,row.names=F,quote = F)
}

######## make heatmaps

dir.create(paste(O,&quot;/heatmaps&quot;,sep=&quot;&quot;))

library(gplots)

for (name in colnames(efit.Temp)) {
  col.pan &lt;- colorpanel(100, &quot;red&quot;, &quot;yellow&quot;, &quot;blue&quot;)
  To &lt;- topTable(efit.Temp,coef=name, adjust.method=&quot;BH&quot;,n=Inf)
  logCPM &lt;- cpmFullDGE[rownames(To[1:50,]),]
  logCPM &lt;- t(scale(t(logCPM)))
  pdf(paste(O,&quot;/heatmaps/Heatmap-&quot;,name,&quot;.pdf&quot;,sep=&quot;&quot;), width=10, height=10)
  heatmap.2(logCPM, col=col.pan, Rowv=TRUE, scale=&quot;none&quot;,trace=&quot;none&quot;, dendrogram=&quot;column&quot;, cexRow=1, cexCol=1.4, density.info=&quot;none&quot;,margin=c(10,9), lhei=c(2,10), lwid=c(2,6),main = paste(name,': Heatmap of top 50 genes',sep=&quot;&quot;))
  dev.off()
}

for (name in colnames(efit.Treat)) {
  col.pan &lt;- colorpanel(100, &quot;red&quot;, &quot;yellow&quot;, &quot;blue&quot;)
  To &lt;- topTable(efit.Treat,coef=name, adjust.method=&quot;BH&quot;,n=Inf)
  logCPM &lt;- cpmFullDGE[rownames(To[1:50,]),]
  logCPM &lt;- t(scale(t(logCPM)))
  pdf(paste(O,&quot;/heatmaps/Heatmap-&quot;,name,&quot;.pdf&quot;,sep=&quot;&quot;), width=10, height=10)
  heatmap.2(logCPM, col=col.pan, Rowv=TRUE, scale=&quot;none&quot;,trace=&quot;none&quot;, dendrogram=&quot;column&quot;, cexRow=1, cexCol=1.4, density.info=&quot;none&quot;,margin=c(10,9), lhei=c(2,10), lwid=c(2,6),main = paste(name,': Heatmap of top 50 genes',sep=&quot;&quot;))
  dev.off()
}

for (name in colnames(efit.interactions)) {
  col.pan &lt;- colorpanel(100, &quot;red&quot;, &quot;yellow&quot;, &quot;blue&quot;)
  To &lt;- topTable(efit.interactions,coef=name, adjust.method=&quot;BH&quot;,n=Inf)
  logCPM &lt;- cpmFullDGE[rownames(To[1:50,]),]
  logCPM &lt;- t(scale(t(logCPM)))
  pdf(paste(O,&quot;/heatmaps/Heatmap-&quot;,name,&quot;.pdf&quot;,sep=&quot;&quot;), width=10, height=10)
  heatmap.2(logCPM, col=col.pan, Rowv=TRUE, scale=&quot;none&quot;,trace=&quot;none&quot;, dendrogram=&quot;column&quot;, cexRow=1, cexCol=1.4, density.info=&quot;none&quot;,margin=c(10,9), lhei=c(2,10), lwid=c(2,6),main = paste(name,': Heatmap of top 50 genes',sep=&quot;&quot;))
  dev.off()
}

for (name in colnames(efit.other)) {
  col.pan &lt;- colorpanel(100, &quot;red&quot;, &quot;yellow&quot;, &quot;blue&quot;)
  To &lt;- topTable(efit.other,coef=name, adjust.method=&quot;BH&quot;,n=Inf)
  logCPM &lt;- cpmFullDGE[rownames(To[1:50,]),]
  logCPM &lt;- t(scale(t(logCPM)))
  pdf(paste(O,&quot;/heatmaps/Heatmap-&quot;,name,&quot;.pdf&quot;,sep=&quot;&quot;), width=10, height=10)
  heatmap.2(logCPM, col=col.pan, Rowv=TRUE, scale=&quot;none&quot;,trace=&quot;none&quot;, dendrogram=&quot;column&quot;, cexRow=1, cexCol=1.4, density.info=&quot;none&quot;,margin=c(10,9), lhei=c(2,10), lwid=c(2,6),main = paste(name,': Heatmap of top 50 genes',sep=&quot;&quot;))
  dev.off()
}

  
## GO-term enrichment 
library(goseq)
dir.create(paste(O,&quot;/GO&quot;,sep=&quot;&quot;))
for (name in colnames(efit.Treat$lods)) {
  print(name)
  CS=topTable(efit.Treat,coef=name, adjust.method=&quot;BH&quot;, n=Inf)
  genes=as.integer(CS$adj.P.Val&lt;0.05)
  names(genes)=row.names(CS)
  pwf=nullp(genes,&quot;dm6&quot;,&quot;ensGene&quot;)
  GO.wall=goseq(pwf,&quot;dm6&quot;,&quot;ensGene&quot;)
  GO.wall=data.frame(GO.wall,&quot;P.adjus_BH&quot;=p.adjust(GO.wall$over_represented_pvalue,method=&quot;BH&quot;))
  name=gsub(&quot;:&quot;,&quot;-&quot;,name)
  out=paste(O,&quot;/GO/GO-&quot;,name,&quot;.txt&quot;,sep=&quot;&quot;)
  write.table(GO.wall,file=out,row.names=F,quote = F)
}
for (name in colnames(efit.Treat$lods)) {
goana(efit.Treat, coef = name, FDR = 0.05, trend = FALSE,species=&quot;Dm&quot;,convert=F)
}

require(&quot;org.Dm.eg.db&quot;, character.only = TRUE)
GO2ALLEGS &lt;- &quot;org.Dm.egGO2ALLEGS&quot;
EG.GO &lt;- AnnotationDbi::toTable(get(GO2ALLEGS))
d &lt;- duplicated(EG.GO[, c(&quot;gene_id&quot;, &quot;go_id&quot;, &quot;Ontology&quot;)])
EG.GO &lt;- EG.GO[!d, ]
de.by.go &lt;- split(EG.GO$gene_id, paste(EG.GO$go_id, EG.GO$Ontology, sep=&quot;.&quot;))

dir.create(paste(O,&quot;/topgo&quot;,sep=&quot;&quot;))
for (name in colnames(efit.Treat$lods)) {
  print(name)
  CS=topTable(efit.Treat,coef=name, adjust.method=&quot;BH&quot;, n=Inf)
  genes=as.integer(CS$adj.P.Val&lt;0.05)
  E=mapIds(org.Dm.eg.db,
           keys=row.names(CS),
           column=&quot;ENTREZID&quot;,
           keytype=&quot;FLYBASE&quot;,
           multiVals=&quot;first&quot;)
  names(genes)=E
  cand=subset(genes,genes==1)
  GO=goana(names(cand),universe=names(genes),species=&quot;Dm&quot;,trend = &quot;Length&quot;)
  
  GO=data.frame(GO,&quot;P.adjus_BH&quot;=p.adjust(GO$P.DE,method=&quot;BH&quot;))
  
  DATAB &lt;- lapply(de.by.go, FUN=function(x) { x[x %in% names(cand)] })
  for (i in 1:nrow(GO)){
    x=paste(rownames(GO)[i],GO$Ont[i],sep=&quot;.&quot;)
    if (length(DATAB[[x]])!=0){
      E=mapIds(org.Dm.eg.db,
               keys=DATAB[[x]],
               column=&quot;FLYBASE&quot;,
               keytype=&quot;ENTREZID&quot;,
               multiVals=&quot;first&quot;)
      GO$gene[i]=paste(E,collapse=&quot;,&quot;)
    }
    else {
      GO$gene[i]=&quot;&quot;  
    }}
  
  name=gsub(&quot;:&quot;,&quot;-&quot;,name)
  out=paste(O,&quot;/topgo/GO-&quot;,name,&quot;.txt&quot;,sep=&quot;&quot;)
  write.table(GO[ order(GO[,6]),],file=out,row.names=T,quote = F,sep=&quot;$&quot;)
}

for (name in colnames(efit.Temp$lods)) {
  print(name)
  CS=topTable(efit.Temp,coef=name, adjust.method=&quot;BH&quot;, n=Inf)
  genes=as.integer(CS$adj.P.Val&lt;0.05)
  E=mapIds(org.Dm.eg.db,
           keys=row.names(CS),
           column=&quot;ENTREZID&quot;,
           keytype=&quot;FLYBASE&quot;,
           multiVals=&quot;first&quot;)
  names(genes)=E
  cand=subset(genes,genes==1)
  GO=goana(names(cand),universe=names(genes),species=&quot;Dm&quot;,trend = &quot;Length&quot;)
  
  GO=data.frame(GO,&quot;P.adjus_BH&quot;=p.adjust(GO$P.DE,method=&quot;BH&quot;))
  
  DATAB &lt;- lapply(de.by.go, FUN=function(x) { x[x %in% names(cand)] })
  for (i in 1:nrow(GO)){
    x=paste(rownames(GO)[i],GO$Ont[i],sep=&quot;.&quot;)
    if (length(DATAB[[x]])!=0){
      E=mapIds(org.Dm.eg.db,
               keys=DATAB[[x]],
               column=&quot;FLYBASE&quot;,
               keytype=&quot;ENTREZID&quot;,
               multiVals=&quot;first&quot;)
      GO$gene[i]=paste(E,collapse=&quot;,&quot;)
    }
    else {
      GO$gene[i]=&quot;&quot;  
    }}
  
  name=gsub(&quot;:&quot;,&quot;-&quot;,name)
  out=paste(O,&quot;/topgo/GO-&quot;,name,&quot;.txt&quot;,sep=&quot;&quot;)
  write.table(GO[ order(GO[,6]),],file=out,row.names=T,quote = F,sep=&quot;$&quot;)
}


for (name in colnames(efit.interactions$lods)) {
  print(name)
  CS=topTable(efit.interactions,coef=name, adjust.method=&quot;BH&quot;, n=Inf)
  genes=as.integer(CS$adj.P.Val&lt;0.05)
  E=mapIds(org.Dm.eg.db,
           keys=row.names(CS),
           column=&quot;ENTREZID&quot;,
           keytype=&quot;FLYBASE&quot;,
           multiVals=&quot;first&quot;)
  names(genes)=E
  cand=subset(genes,genes==1)
  GO=goana(names(cand),universe=names(genes),species=&quot;Dm&quot;,trend = &quot;Length&quot;)
  
  GO=data.frame(GO,&quot;P.adjus_BH&quot;=p.adjust(GO$P.DE,method=&quot;BH&quot;))
  
  DATAB &lt;- lapply(de.by.go, FUN=function(x) { x[x %in% names(cand)] })
  for (i in 1:nrow(GO)){
    x=paste(rownames(GO)[i],GO$Ont[i],sep=&quot;.&quot;)
    if (length(DATAB[[x]])!=0){
      E=mapIds(org.Dm.eg.db,
               keys=DATAB[[x]],
               column=&quot;FLYBASE&quot;,
               keytype=&quot;ENTREZID&quot;,
               multiVals=&quot;first&quot;)
      GO$gene[i]=paste(E,collapse=&quot;,&quot;)
    }
    else {
      GO$gene[i]=&quot;&quot;  
    }}
  
  name=gsub(&quot;:&quot;,&quot;-&quot;,name)
  out=paste(O,&quot;/topgo/GO-&quot;,name,&quot;.txt&quot;,sep=&quot;&quot;)
  write.table(GO[ order(GO[,6]),],file=out,row.names=T,quote = F,sep=&quot;$&quot;)
}



for (name in colnames(efit.other$lods)) {
  print(name)
  CS=topTable(efit.other,coef=name, adjust.method=&quot;BH&quot;, n=Inf)
  genes=as.integer(CS$adj.P.Val&lt;0.05)
  E=mapIds(org.Dm.eg.db,
           keys=row.names(CS),
           column=&quot;ENTREZID&quot;,
           keytype=&quot;FLYBASE&quot;,
           multiVals=&quot;first&quot;)
  names(genes)=E
  cand=subset(genes,genes==1)
  GO=goana(names(cand),universe=names(genes),species=&quot;Dm&quot;,trend = &quot;Length&quot;)
  
  GO=data.frame(GO,&quot;P.adjus_BH&quot;=p.adjust(GO$P.DE,method=&quot;BH&quot;))
  
  DATAB &lt;- lapply(de.by.go, FUN=function(x) { x[x %in% names(cand)] })
  for (i in 1:nrow(GO)){
    x=paste(rownames(GO)[i],GO$Ont[i],sep=&quot;.&quot;)
    if (length(DATAB[[x]])!=0){
      E=mapIds(org.Dm.eg.db,
               keys=DATAB[[x]],
               column=&quot;FLYBASE&quot;,
               keytype=&quot;ENTREZID&quot;,
               multiVals=&quot;first&quot;)
      GO$gene[i]=paste(E,collapse=&quot;,&quot;)
    }
    else {
      GO$gene[i]=&quot;&quot;  
    }}
  
  name=gsub(&quot;:&quot;,&quot;-&quot;,name)
  out=paste(O,&quot;/topgo/GO-&quot;,name,&quot;.txt&quot;,sep=&quot;&quot;)
  write.table(GO[ order(GO[,6]),],file=out,row.names=T,quote = F,sep=&quot;$&quot;)
}
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>## Analysis of RNA-Seq data of inversion lines</span>

<span class="pl-c"><span class="pl-c">#</span># install packages</span>
<span class="pl-c"><span class="pl-c">#</span>install.packages("readr")</span>
<span class="pl-c"><span class="pl-c">#</span>install.packages("tximport")</span>
<span class="pl-c"><span class="pl-c">#</span>install.packages("edgeR")</span>
<span class="pl-c"><span class="pl-c">#</span>install.packages("RColorBrewer")</span>
<span class="pl-c"><span class="pl-c">#</span>install.packages("limma")</span>
<span class="pl-c"><span class="pl-c">#</span>install.packages("pheatmap")</span>

<span class="pl-v">I</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/data/RNASeq/count/analyses_counts.txt<span class="pl-pds">"</span></span>
<span class="pl-v">O</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/data/RNASeq/<span class="pl-pds">"</span></span>

<span class="pl-c"><span class="pl-c">#</span>Factors</span>

<span class="pl-v">Treat</span><span class="pl-k">=</span><span class="pl-k">factor</span>(c(<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FI<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>FS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>MS<span class="pl-pds">'</span></span>))
<span class="pl-v">Treat</span><span class="pl-k">=</span>relevel(<span class="pl-smi">Treat</span>,<span class="pl-v">ref</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>FS<span class="pl-pds">"</span></span>)
<span class="pl-v">Temp</span><span class="pl-k">=</span><span class="pl-k">factor</span>(c(<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>18<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>25<span class="pl-pds">'</span></span>))
<span class="pl-v">Rep</span><span class="pl-k">=</span><span class="pl-k">factor</span>(c(<span class="pl-c1">10</span>,<span class="pl-c1">14</span>,<span class="pl-c1">17</span>,<span class="pl-c1">18</span>,<span class="pl-c1">22</span>,<span class="pl-c1">23</span>,<span class="pl-c1">26</span>,<span class="pl-c1">5</span>,<span class="pl-c1">9</span>,<span class="pl-c1">5</span>,<span class="pl-c1">9</span>,<span class="pl-c1">10</span>,<span class="pl-c1">14</span>,<span class="pl-c1">17</span>,<span class="pl-c1">18</span>,<span class="pl-c1">22</span>,<span class="pl-c1">23</span>,<span class="pl-c1">26</span>,<span class="pl-c1">13</span>,<span class="pl-c1">16</span>,<span class="pl-c1">2</span>,<span class="pl-c1">21</span>,<span class="pl-c1">27</span>,<span class="pl-c1">3</span>,<span class="pl-c1">4</span>,<span class="pl-c1">6</span>,<span class="pl-c1">7</span>,<span class="pl-c1">2</span>,<span class="pl-c1">3</span>,<span class="pl-c1">4</span>,<span class="pl-c1">6</span>,<span class="pl-c1">7</span>,<span class="pl-c1">13</span>,<span class="pl-c1">16</span>,<span class="pl-c1">21</span>,<span class="pl-c1">27</span>,<span class="pl-c1">1</span>,<span class="pl-c1">11</span>,<span class="pl-c1">12</span>,<span class="pl-c1">15</span>,<span class="pl-c1">19</span>,<span class="pl-c1">20</span>,<span class="pl-c1">24</span>,<span class="pl-c1">25</span>,<span class="pl-c1">8</span>,<span class="pl-c1">1</span>,<span class="pl-c1">8</span>,<span class="pl-c1">11</span>,<span class="pl-c1">12</span>,<span class="pl-c1">15</span>,<span class="pl-c1">19</span>,<span class="pl-c1">20</span>,<span class="pl-c1">24</span>,<span class="pl-c1">25</span>))

<span class="pl-c"><span class="pl-c">#</span># 1a) read Data</span>
<span class="pl-smi">rawData</span><span class="pl-k">&lt;-</span>read.table(<span class="pl-smi">I</span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">T</span>)
<span class="pl-v">rawCounts</span><span class="pl-k">=</span><span class="pl-smi">rawData</span>[,<span class="pl-c1">3</span><span class="pl-k">:</span><span class="pl-c1">56</span>]
rownames(<span class="pl-smi">rawCounts</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">rawData</span>[,<span class="pl-c1">1</span>]

<span class="pl-c"><span class="pl-c">#</span># 2) create DGE object using edgeR</span>
library(<span class="pl-smi">edgeR</span>)
<span class="pl-smi">FullDGE</span><span class="pl-k">&lt;-</span>DGEList(<span class="pl-smi">rawCounts</span>,<span class="pl-v">group</span><span class="pl-k">=</span>names(<span class="pl-smi">rawCounts</span>))

<span class="pl-c"><span class="pl-c">#</span># 3) add factors to object</span>
<span class="pl-smi">FullDGE</span><span class="pl-k">$</span><span class="pl-smi">samples</span><span class="pl-k">$</span><span class="pl-smi">treat</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">Treat</span>
<span class="pl-smi">FullDGE</span><span class="pl-k">$</span><span class="pl-smi">samples</span><span class="pl-k">$</span><span class="pl-smi">temp</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">Temp</span>

<span class="pl-c"><span class="pl-c">#</span># Transform counts (Counts per million: CPM) and normalize the counts using the TMM method in edgeR</span>
<span class="pl-smi">FullDGE</span> <span class="pl-k">&lt;-</span> calcNormFactors(<span class="pl-smi">FullDGE</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>TMM<span class="pl-pds">"</span></span>)
<span class="pl-smi">cpmFullDGE</span><span class="pl-k">&lt;-</span>cpm(<span class="pl-smi">FullDGE</span>,<span class="pl-v">log</span><span class="pl-k">=</span><span class="pl-c1">T</span>)

<span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/countdata/CPM.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
<span class="pl-v">TAB</span><span class="pl-k">=</span><span class="pl-smi">cpmFullDGE</span>
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span><span class="pl-k">=</span>rownames(<span class="pl-smi">TAB</span>),<span class="pl-smi">TAB</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-c"><span class="pl-c">#</span># remove genes covered by less than 2 read per million in at least nine samples</span>
<span class="pl-smi">keep</span> <span class="pl-k">&lt;-</span> rowSums(<span class="pl-smi">cpmFullDGE</span> <span class="pl-k">&gt;</span><span class="pl-k">=</span> <span class="pl-c1">2</span>) <span class="pl-k">&gt;</span><span class="pl-k">=</span><span class="pl-c1">9</span>
<span class="pl-smi">FullDGE</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">FullDGE</span>[<span class="pl-smi">keep</span>,] 
<span class="pl-smi">cpmTFullDGE</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">cpmFullDGE</span>[<span class="pl-smi">keep</span>,] 

<span class="pl-c"><span class="pl-c">#</span># Plot density histograms to quantify the effect of filtering out lowly expressed genes</span>
dir.create(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/quality<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>))
pdf(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/quality/CPM-hist.pdf<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>),<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">15</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">5</span>)
<span class="pl-smi">nsamples</span> <span class="pl-k">&lt;-</span> ncol(<span class="pl-smi">FullDGE</span>)
<span class="pl-smi">col</span> <span class="pl-k">&lt;-</span> topo.colors(<span class="pl-smi">nsamples</span>)
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">2</span>))
plot(density(<span class="pl-smi">cpmFullDGE</span>[,<span class="pl-c1">1</span>]), <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">col</span>[<span class="pl-c1">1</span>], <span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>, <span class="pl-v">ylim</span><span class="pl-k">=</span>c(<span class="pl-c1">0</span>,<span class="pl-c1">0.25</span>), <span class="pl-v">las</span><span class="pl-k">=</span><span class="pl-c1">2</span>,
     <span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
abline(<span class="pl-v">v</span><span class="pl-k">=</span><span class="pl-c1">0</span>, <span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">3</span>)

<span class="pl-k">for</span> (<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-smi">nsamples</span>){
  <span class="pl-smi">den</span> <span class="pl-k">&lt;-</span> density(<span class="pl-smi">cpmFullDGE</span>[,<span class="pl-smi">i</span>])
  lines(<span class="pl-smi">den</span><span class="pl-k">$</span><span class="pl-smi">x</span>, <span class="pl-smi">den</span><span class="pl-k">$</span><span class="pl-smi">y</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">col</span>[<span class="pl-smi">i</span>], <span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
} 
legend(<span class="pl-s"><span class="pl-pds">"</span>topright<span class="pl-pds">"</span></span>, <span class="pl-v">legend</span><span class="pl-k">=</span>colnames(<span class="pl-smi">FullDGE</span>), <span class="pl-v">text.col</span><span class="pl-k">=</span><span class="pl-smi">col</span>, <span class="pl-v">bty</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>n<span class="pl-pds">"</span></span>,<span class="pl-v">ncol</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>,<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">0.5</span>)
title(<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>A. Raw data<span class="pl-pds">"</span></span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Log-cpm<span class="pl-pds">"</span></span>)

plot(density(<span class="pl-smi">cpmTFullDGE</span>[,<span class="pl-c1">1</span>]), <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">col</span>[<span class="pl-c1">1</span>], <span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>, <span class="pl-v">ylim</span><span class="pl-k">=</span>c(<span class="pl-c1">0</span>,<span class="pl-c1">0.25</span>), <span class="pl-v">las</span><span class="pl-k">=</span><span class="pl-c1">2</span>,
     <span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
title(<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>B. Filtered data<span class="pl-pds">"</span></span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Log-cpm<span class="pl-pds">"</span></span>) 
abline(<span class="pl-v">v</span><span class="pl-k">=</span><span class="pl-c1">0</span>, <span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">3</span>)

<span class="pl-k">for</span> (<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-smi">nsamples</span>){
  <span class="pl-smi">den</span> <span class="pl-k">&lt;-</span> density(<span class="pl-smi">cpmTFullDGE</span>[,<span class="pl-smi">i</span>])
  lines(<span class="pl-smi">den</span><span class="pl-k">$</span><span class="pl-smi">x</span>, <span class="pl-smi">den</span><span class="pl-k">$</span><span class="pl-smi">y</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">col</span>[<span class="pl-smi">i</span>], <span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
} 
legend(<span class="pl-s"><span class="pl-pds">"</span>topright<span class="pl-pds">"</span></span>, <span class="pl-v">legend</span><span class="pl-k">=</span>colnames(<span class="pl-smi">FullDGE</span>), <span class="pl-v">text.col</span><span class="pl-k">=</span><span class="pl-smi">col</span>, <span class="pl-v">bty</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>n<span class="pl-pds">"</span></span>,<span class="pl-v">ncol</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>,<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">0.5</span>)
dev.off()

<span class="pl-c"><span class="pl-c">#</span># explore Data with MDS to identify clustering</span>
library(<span class="pl-smi">RColorBrewer</span>)
pdf(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/quality/MDS.pdf<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>),<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">14</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">12</span>)
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">2</span>))

<span class="pl-smi">col.Treat</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">Treat</span>
levels(<span class="pl-smi">col.Treat</span>) <span class="pl-k">&lt;-</span> brewer.pal(nlevels(<span class="pl-smi">col.Treat</span>), <span class="pl-s"><span class="pl-pds">"</span>Set1<span class="pl-pds">"</span></span>)
plotMDS(<span class="pl-smi">cpmTFullDGE</span>, <span class="pl-v">col</span><span class="pl-k">=</span>as.character(<span class="pl-smi">col.Treat</span>))
title(<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Karypop: Dim 1/2<span class="pl-pds">"</span></span>)

<span class="pl-smi">col.Temp</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">Temp</span>
levels(<span class="pl-smi">col.Temp</span>) <span class="pl-k">&lt;-</span> brewer.pal(nlevels(<span class="pl-smi">col.Temp</span>), <span class="pl-s"><span class="pl-pds">"</span>Set1<span class="pl-pds">"</span></span>)
plotMDS(<span class="pl-smi">cpmTFullDGE</span>, <span class="pl-v">col</span><span class="pl-k">=</span>as.character(<span class="pl-smi">col.Temp</span>))
title(<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Temperature: Dim 1/2<span class="pl-pds">"</span></span>)

plotMDS(<span class="pl-smi">cpmTFullDGE</span>, <span class="pl-v">col</span><span class="pl-k">=</span>as.character(<span class="pl-smi">col.Treat</span>), <span class="pl-v">dim</span><span class="pl-k">=</span>c(<span class="pl-c1">3</span>,<span class="pl-c1">4</span>))
title(<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Karypop: Dim 3/4<span class="pl-pds">"</span></span>)

plotMDS(<span class="pl-smi">cpmTFullDGE</span>, <span class="pl-v">col</span><span class="pl-k">=</span>as.character(<span class="pl-smi">col.Temp</span>), <span class="pl-v">dim</span><span class="pl-k">=</span>c(<span class="pl-c1">3</span>,<span class="pl-c1">4</span>))
title(<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Temperature: Dim 3/4<span class="pl-pds">"</span></span>)

dev.off()

<span class="pl-c"><span class="pl-c">#</span>## explore data with PCA to explore clustering</span>

pdf(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/quality/PCA.pdf<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>),<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">12</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">3</span>))
<span class="pl-v">newcpm</span><span class="pl-k">=</span>t(<span class="pl-smi">cpmTFullDGE</span>)
<span class="pl-v">PCA</span><span class="pl-k">=</span>prcomp(<span class="pl-smi">newcpm</span>)
<span class="pl-v">colKP</span><span class="pl-k">=</span>rainbow(length(levels(<span class="pl-smi">Treat</span>)))[<span class="pl-smi">Treat</span>]
<span class="pl-v">colTemp</span><span class="pl-k">=</span>rainbow(length(levels(<span class="pl-smi">Temp</span>)))[as.factor(<span class="pl-smi">Temp</span>)]
<span class="pl-v">colPop</span><span class="pl-k">=</span>rainbow(length(levels(<span class="pl-smi">Rep</span>)))[as.factor(<span class="pl-smi">Rep</span>)]

plot(<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">2</span>],<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">colKP</span>,<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Karypop<span class="pl-pds">"</span></span>,<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">3</span>)
<span class="pl-c"><span class="pl-c">#</span>text(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colKP)</span>

plot(<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">2</span>],<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">colTemp</span>,<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Temp<span class="pl-pds">"</span></span>,<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">3</span>)
<span class="pl-c"><span class="pl-c">#</span>text(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colTemp)</span>

plot(<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">1</span>],<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">2</span>],<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">colPop</span>,<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Replicate<span class="pl-pds">"</span></span>,<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">3</span>)
<span class="pl-c"><span class="pl-c">#</span>text(PCA$x[,1],PCA$x[,2],labels=interaction(Treat,Temp,Rep),col=colPop)</span>

plot(<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">3</span>],<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">4</span>],<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">colKP</span>,<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Karypop<span class="pl-pds">"</span></span>,<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">3</span>)
<span class="pl-c"><span class="pl-c">#</span>text(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colKP)</span>

plot(<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">3</span>],<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">4</span>],<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">colTemp</span>,<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Temp<span class="pl-pds">"</span></span>,<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">3</span>)
<span class="pl-c"><span class="pl-c">#</span>text(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colTemp)</span>

plot(<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">3</span>],<span class="pl-smi">PCA</span><span class="pl-k">$</span><span class="pl-smi">x</span>[,<span class="pl-c1">4</span>],<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">colPop</span>,<span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Replicate<span class="pl-pds">"</span></span>,<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">3</span>)
<span class="pl-c"><span class="pl-c">#</span>text(PCA$x[,3],PCA$x[,4],labels=interaction(Treat,Temp,Rep),col=colPop)</span>

dev.off()

<span class="pl-c"><span class="pl-c">#</span>## Limma analysis with singe grouping factor</span>

library(<span class="pl-smi">limma</span>)

<span class="pl-v">Treat</span><span class="pl-k">=</span>relevel(<span class="pl-smi">Treat</span>,<span class="pl-v">ref</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>FS<span class="pl-pds">"</span></span>)
dir.create(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/limma<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>))
<span class="pl-v">G</span><span class="pl-k">=</span>interaction(<span class="pl-smi">Treat</span>,<span class="pl-smi">Temp</span>)

<span class="pl-v">design.SP</span><span class="pl-k">=</span>model.matrix(<span class="pl-k">~</span><span class="pl-c1">0</span><span class="pl-k">+</span><span class="pl-smi">G</span>)
colnames(<span class="pl-smi">design.SP</span>)<span class="pl-k">=</span>levels(<span class="pl-smi">G</span>)

<span class="pl-c"><span class="pl-c">#</span># Do VOOM</span>
<span class="pl-smi">voomFullDGE.SP</span> <span class="pl-k">&lt;-</span> voom(<span class="pl-smi">FullDGE</span>, <span class="pl-smi">design.SP</span>, <span class="pl-v">plot</span><span class="pl-k">=</span><span class="pl-c1">F</span>)

<span class="pl-c"><span class="pl-c">#</span> write transformed data to file</span>
<span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/countdata/voomCounts.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
<span class="pl-v">TAB</span><span class="pl-k">=</span><span class="pl-smi">voomFullDGE.SP</span><span class="pl-k">$</span><span class="pl-smi">E</span>
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span><span class="pl-k">=</span>rownames(<span class="pl-smi">TAB</span>),<span class="pl-smi">TAB</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">corfit.SP</span> <span class="pl-k">&lt;-</span> duplicateCorrelation(<span class="pl-smi">voomFullDGE.SP</span>,<span class="pl-smi">design.SP</span>,<span class="pl-v">block</span><span class="pl-k">=</span><span class="pl-smi">Rep</span>)
<span class="pl-smi">vfit.SP</span> <span class="pl-k">&lt;-</span> lmFit(<span class="pl-smi">voomFullDGE.SP</span>, <span class="pl-smi">design.SP</span>, <span class="pl-v">block</span><span class="pl-k">=</span><span class="pl-smi">Rep</span>, <span class="pl-v">correlation</span><span class="pl-k">=</span><span class="pl-smi">corfit.SP</span><span class="pl-k">$</span><span class="pl-smi">consensus</span>)

<span class="pl-c"><span class="pl-c">#</span>########### Karypop</span>

<span class="pl-smi">cont.matrix.SP.Treat</span> <span class="pl-k">&lt;-</span> makeContrasts(
  <span class="pl-s"><span class="pl-pds">"</span>FS-FI<span class="pl-pds">"</span></span><span class="pl-k">=</span>(<span class="pl-smi">FS.18</span><span class="pl-k">+</span><span class="pl-smi">FS.25</span>)<span class="pl-k">/</span><span class="pl-c1">2</span><span class="pl-k">-</span>(<span class="pl-smi">FI.18</span><span class="pl-k">+</span><span class="pl-smi">FI.25</span>)<span class="pl-k">/</span><span class="pl-c1">2</span>,
  <span class="pl-s"><span class="pl-pds">"</span>FS-MS<span class="pl-pds">"</span></span><span class="pl-k">=</span>(<span class="pl-smi">FS.18</span><span class="pl-k">+</span><span class="pl-smi">FS.25</span>)<span class="pl-k">/</span><span class="pl-c1">2</span><span class="pl-k">-</span>(<span class="pl-smi">MS.18</span><span class="pl-k">+</span><span class="pl-smi">MS.25</span>)<span class="pl-k">/</span><span class="pl-c1">2</span>,
  <span class="pl-s"><span class="pl-pds">"</span>MS-FI<span class="pl-pds">"</span></span><span class="pl-k">=</span>(<span class="pl-smi">MS.18</span><span class="pl-k">+</span><span class="pl-smi">MS.25</span>)<span class="pl-k">/</span><span class="pl-c1">2</span><span class="pl-k">-</span>(<span class="pl-smi">FI.18</span><span class="pl-k">+</span><span class="pl-smi">FI.25</span>)<span class="pl-k">/</span><span class="pl-c1">2</span>,
  <span class="pl-v">levels</span><span class="pl-k">=</span><span class="pl-smi">design.SP</span>)

<span class="pl-c"><span class="pl-c">#</span> now calculate the contrasts:</span>
<span class="pl-smi">vfit.Treat</span> <span class="pl-k">&lt;-</span> contrasts.fit(<span class="pl-smi">vfit.SP</span>, <span class="pl-smi">cont.matrix.SP.Treat</span>)
<span class="pl-smi">efit.Treat</span> <span class="pl-k">&lt;-</span> eBayes(<span class="pl-smi">vfit.Treat</span>)

<span class="pl-c"><span class="pl-c">#</span> now export the tables with the p-values and log2-fc for each of the contrasts</span>
<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">cont.matrix.SP.Treat</span>)){
  <span class="pl-c"><span class="pl-c">#</span> print the name</span>
  print(<span class="pl-smi">name</span>)
  
  <span class="pl-c"><span class="pl-c">#</span> make an output file in outputfolder</span>
  <span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/limma/<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
  
  <span class="pl-c"><span class="pl-c">#</span> generate summary table with Benjamini Hochberg adjsuted p-values for all genes</span>
  <span class="pl-v">TAB</span><span class="pl-k">=</span>topTable(<span class="pl-smi">efit.Treat</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>, <span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  
  <span class="pl-c"><span class="pl-c">#</span> export table to text file</span>
  write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span><span class="pl-k">=</span>rownames(<span class="pl-smi">TAB</span>),<span class="pl-smi">TAB</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)
}

<span class="pl-c"><span class="pl-c">#</span> another cool thing about eBayes is that it calculates an F-value testing if the contrasts are different from 0. In the above case we are comparing all levels for the factor "Treat" only. Ebays combines the t-statistics for all the contrasts into an overall test of significance for that gene. The resulting F-value will thus tell us if "Treat" is significantly differnet from 0. See limma manual page 61. Note that I append another column to the dataframe with the P-value adjusted a la Benjamini-Hochberg</span>
<span class="pl-v">efit.F.Treat</span><span class="pl-k">=</span><span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>F<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">efit.Treat</span><span class="pl-k">$</span><span class="pl-c1">F</span>,<span class="pl-s"><span class="pl-pds">"</span>Pval<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">efit.Treat</span><span class="pl-k">$</span><span class="pl-c1">F</span><span class="pl-smi">.p.value</span>,<span class="pl-s"><span class="pl-pds">"</span>Pval.BH<span class="pl-pds">"</span></span><span class="pl-k">=</span>p.adjust(<span class="pl-smi">efit.Treat</span><span class="pl-k">$</span><span class="pl-c1">F</span><span class="pl-smi">.p.value</span>, <span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>))
rownames(<span class="pl-smi">efit.F.Treat</span>)<span class="pl-k">=</span>rownames(<span class="pl-smi">efit.Treat</span>)

<span class="pl-c"><span class="pl-c">#</span> export table as above for the contrasts</span>

<span class="pl-c"><span class="pl-c">#</span> make an output file in outputfolder</span>
<span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/limma/Treat-F-value.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> export table to text file</span>
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span><span class="pl-k">=</span>rownames(<span class="pl-smi">efit.F.Treat</span>),<span class="pl-smi">efit.F.Treat</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-c"><span class="pl-c">#</span>########### Temp</span>

<span class="pl-smi">cont.matrix.SP.Temp</span> <span class="pl-k">&lt;-</span> makeContrasts(
  <span class="pl-s"><span class="pl-pds">"</span>T18-25<span class="pl-pds">"</span></span><span class="pl-k">=</span>(<span class="pl-smi">FS.18</span><span class="pl-k">+</span><span class="pl-smi">FI.18</span><span class="pl-k">+</span><span class="pl-smi">MS.18</span>)<span class="pl-k">/</span><span class="pl-c1">3</span><span class="pl-k">-</span>(<span class="pl-smi">FS.25</span><span class="pl-k">+</span><span class="pl-smi">FI.25</span><span class="pl-k">+</span><span class="pl-smi">MS.25</span>)<span class="pl-k">/</span><span class="pl-c1">3</span>,
  <span class="pl-v">levels</span><span class="pl-k">=</span><span class="pl-smi">design.SP</span>)

<span class="pl-c"><span class="pl-c">#</span> now calculate the contrasts:</span>
<span class="pl-smi">vfit.Temp</span> <span class="pl-k">&lt;-</span> contrasts.fit(<span class="pl-smi">vfit.SP</span>, <span class="pl-smi">cont.matrix.SP.Temp</span>)
<span class="pl-smi">efit.Temp</span> <span class="pl-k">&lt;-</span> eBayes(<span class="pl-smi">vfit.Temp</span>)

<span class="pl-c"><span class="pl-c">#</span> now export the tables with the p-values and log2-fc for each of the contrasts</span>
<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">cont.matrix.SP.Temp</span>)){
  <span class="pl-c"><span class="pl-c">#</span> print the name</span>
  print(<span class="pl-smi">name</span>)
  
  <span class="pl-c"><span class="pl-c">#</span> make an output file in outputfolder</span>
  <span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/limma/<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
  
  <span class="pl-c"><span class="pl-c">#</span> generate summary table with Benjamini Hochberg adjsuted p-values for all genes</span>
  <span class="pl-v">TAB</span><span class="pl-k">=</span>topTable(<span class="pl-smi">efit.Temp</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>, <span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  
  <span class="pl-c"><span class="pl-c">#</span> export table to text file</span>
  write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span><span class="pl-k">=</span>rownames(<span class="pl-smi">TAB</span>),<span class="pl-smi">TAB</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)
}

<span class="pl-v">efit.F.Temp</span><span class="pl-k">=</span><span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>F<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">efit.Temp</span><span class="pl-k">$</span><span class="pl-c1">F</span>,<span class="pl-s"><span class="pl-pds">"</span>Pval<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">efit.Temp</span><span class="pl-k">$</span><span class="pl-c1">F</span><span class="pl-smi">.p.value</span>,<span class="pl-s"><span class="pl-pds">"</span>Pval.BH<span class="pl-pds">"</span></span><span class="pl-k">=</span>p.adjust(<span class="pl-smi">efit.Temp</span><span class="pl-k">$</span><span class="pl-c1">F</span><span class="pl-smi">.p.value</span>, <span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>))
rownames(<span class="pl-smi">efit.F.Temp</span>)<span class="pl-k">=</span>rownames(<span class="pl-smi">efit.Temp</span>)

<span class="pl-c"><span class="pl-c">#</span> export table as above for the contrasts</span>

<span class="pl-c"><span class="pl-c">#</span> make an output file in outputfolder</span>
<span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/limma/Temp-F-value.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> export table to text file</span>
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span><span class="pl-k">=</span>rownames(<span class="pl-smi">efit.F.Temp</span>),<span class="pl-smi">efit.F.Temp</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">cont.matrix.SP.interactions</span> <span class="pl-k">&lt;-</span> makeContrasts(
  <span class="pl-s"><span class="pl-pds">"</span>FS-FI_int<span class="pl-pds">"</span></span><span class="pl-k">=</span>(<span class="pl-smi">FS.18</span><span class="pl-k">-</span><span class="pl-smi">FS.25</span>)<span class="pl-k">-</span>(<span class="pl-smi">FI.18</span><span class="pl-k">-</span><span class="pl-smi">FI.25</span>),
  <span class="pl-s"><span class="pl-pds">"</span>FI-MS_int<span class="pl-pds">"</span></span><span class="pl-k">=</span>(<span class="pl-smi">FI.18</span><span class="pl-k">-</span><span class="pl-smi">FI.25</span>)<span class="pl-k">-</span>(<span class="pl-smi">MS.18</span><span class="pl-k">-</span><span class="pl-smi">MS.25</span>),
  <span class="pl-s"><span class="pl-pds">"</span>FS-MS_int<span class="pl-pds">"</span></span><span class="pl-k">=</span>(<span class="pl-smi">FS.18</span><span class="pl-k">-</span><span class="pl-smi">FS.25</span>)<span class="pl-k">-</span>(<span class="pl-smi">MS.18</span><span class="pl-k">-</span><span class="pl-smi">MS.25</span>),
  <span class="pl-v">levels</span><span class="pl-k">=</span><span class="pl-smi">design.SP</span>)

<span class="pl-c"><span class="pl-c">#</span> now calculate the contrasts:</span>
<span class="pl-smi">vfit.interactions</span> <span class="pl-k">&lt;-</span> contrasts.fit(<span class="pl-smi">vfit.SP</span>, <span class="pl-smi">cont.matrix.SP.interactions</span>)
<span class="pl-smi">efit.interactions</span> <span class="pl-k">&lt;-</span> eBayes(<span class="pl-smi">vfit.interactions</span>)

<span class="pl-c"><span class="pl-c">#</span> now export the tables with the p-values and log2-fc for each of the contrasts</span>
<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">cont.matrix.SP.interactions</span>)){
  <span class="pl-c"><span class="pl-c">#</span> print the name</span>
  print(<span class="pl-smi">name</span>)
  
  <span class="pl-c"><span class="pl-c">#</span> make an output file in outputfolder</span>
  <span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/limma/<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
  
  <span class="pl-c"><span class="pl-c">#</span> generate summary table with Benjamini Hochberg adjsuted p-values for all genes</span>
  <span class="pl-v">TAB</span><span class="pl-k">=</span>topTable(<span class="pl-smi">efit.interactions</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>, <span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  
  <span class="pl-c"><span class="pl-c">#</span> export table to text file</span>
  write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span><span class="pl-k">=</span>rownames(<span class="pl-smi">TAB</span>),<span class="pl-smi">TAB</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)
}

<span class="pl-c"><span class="pl-c">#</span>####### Interactions</span>

<span class="pl-v">efit.F.interactions</span><span class="pl-k">=</span><span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>F<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">efit.interactions</span><span class="pl-k">$</span><span class="pl-c1">F</span>,<span class="pl-s"><span class="pl-pds">"</span>Pval<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">efit.interactions</span><span class="pl-k">$</span><span class="pl-c1">F</span><span class="pl-smi">.p.value</span>,<span class="pl-s"><span class="pl-pds">"</span>Pval.BH<span class="pl-pds">"</span></span><span class="pl-k">=</span>p.adjust(<span class="pl-smi">efit.interactions</span><span class="pl-k">$</span><span class="pl-c1">F</span><span class="pl-smi">.p.value</span>, <span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>))
rownames(<span class="pl-smi">efit.F.interactions</span>)<span class="pl-k">=</span>rownames(<span class="pl-smi">efit.interactions</span>)

<span class="pl-c"><span class="pl-c">#</span> export table as above for the contrasts</span>

<span class="pl-c"><span class="pl-c">#</span> make an output file in outputfolder</span>
<span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/limma/interactions-F-value.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> export table to text file</span>
write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span><span class="pl-k">=</span>rownames(<span class="pl-smi">efit.F.interactions</span>),<span class="pl-smi">efit.F.interactions</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-c"><span class="pl-c">#</span>########### Other comparisons</span>

<span class="pl-smi">cont.matrix.SP.other</span> <span class="pl-k">&lt;-</span> makeContrasts(
  <span class="pl-s"><span class="pl-pds">"</span>FS-FI_18<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">FS.18</span><span class="pl-k">-</span><span class="pl-smi">FI.18</span>,
  <span class="pl-s"><span class="pl-pds">"</span>FS-MS_18<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">FS.18</span><span class="pl-k">-</span><span class="pl-smi">MS.18</span>,
  <span class="pl-s"><span class="pl-pds">"</span>MS-FI_18<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">MS.18</span><span class="pl-k">-</span><span class="pl-smi">FI.18</span>,
  <span class="pl-s"><span class="pl-pds">"</span>FS-FI_25<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">FS.25</span><span class="pl-k">-</span><span class="pl-smi">FI.25</span>,
  <span class="pl-s"><span class="pl-pds">"</span>FS-MS_25<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">FS.25</span><span class="pl-k">-</span><span class="pl-smi">MS.25</span>,
  <span class="pl-s"><span class="pl-pds">"</span>MS-FI_25<span class="pl-pds">"</span></span><span class="pl-k">=</span><span class="pl-smi">MS.25</span><span class="pl-k">-</span><span class="pl-smi">FI.25</span>,
  <span class="pl-v">levels</span><span class="pl-k">=</span><span class="pl-smi">design.SP</span>)

<span class="pl-c"><span class="pl-c">#</span> now calculate the contrasts:</span>
<span class="pl-smi">vfit.other</span> <span class="pl-k">&lt;-</span> contrasts.fit(<span class="pl-smi">vfit.SP</span>, <span class="pl-smi">cont.matrix.SP.other</span>)
<span class="pl-smi">efit.other</span> <span class="pl-k">&lt;-</span> eBayes(<span class="pl-smi">vfit.other</span>)

<span class="pl-c"><span class="pl-c">#</span> now export the tables with the p-values and log2-fc for each of the contrasts</span>
<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">cont.matrix.SP.other</span>)){
  <span class="pl-c"><span class="pl-c">#</span> print the name</span>
  print(<span class="pl-smi">name</span>)
  
  <span class="pl-c"><span class="pl-c">#</span> make an output file in outputfolder</span>
  <span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/limma/<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
  
  <span class="pl-c"><span class="pl-c">#</span> generate summary table with Benjamini Hochberg adjsuted p-values for all genes</span>
  <span class="pl-v">TAB</span><span class="pl-k">=</span>topTable(<span class="pl-smi">efit.other</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>, <span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  
  <span class="pl-c"><span class="pl-c">#</span> export table to text file</span>
  write.table(<span class="pl-k">data.frame</span>(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span><span class="pl-k">=</span>rownames(<span class="pl-smi">TAB</span>),<span class="pl-smi">TAB</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)
}

<span class="pl-c"><span class="pl-c">#</span>####### make heatmaps</span>

dir.create(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/heatmaps<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>))

library(<span class="pl-smi">gplots</span>)

<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">efit.Temp</span>)) {
  <span class="pl-smi">col.pan</span> <span class="pl-k">&lt;-</span> colorpanel(<span class="pl-c1">100</span>, <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>yellow<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>)
  <span class="pl-smi">To</span> <span class="pl-k">&lt;-</span> topTable(<span class="pl-smi">efit.Temp</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>,<span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  <span class="pl-smi">logCPM</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">cpmFullDGE</span>[rownames(<span class="pl-smi">To</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">50</span>,]),]
  <span class="pl-smi">logCPM</span> <span class="pl-k">&lt;-</span> t(scale(t(<span class="pl-smi">logCPM</span>)))
  pdf(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/heatmaps/Heatmap-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.pdf<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>), <span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">10</span>, <span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">10</span>)
  heatmap.2(<span class="pl-smi">logCPM</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">col.pan</span>, <span class="pl-v">Rowv</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">scale</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>,<span class="pl-v">trace</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, <span class="pl-v">dendrogram</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>column<span class="pl-pds">"</span></span>, <span class="pl-v">cexRow</span><span class="pl-k">=</span><span class="pl-c1">1</span>, <span class="pl-v">cexCol</span><span class="pl-k">=</span><span class="pl-c1">1.4</span>, <span class="pl-v">density.info</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>,<span class="pl-v">margin</span><span class="pl-k">=</span>c(<span class="pl-c1">10</span>,<span class="pl-c1">9</span>), <span class="pl-v">lhei</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">10</span>), <span class="pl-v">lwid</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">6</span>),<span class="pl-v">main</span> <span class="pl-k">=</span> paste(<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">'</span>: Heatmap of top 50 genes<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>))
  dev.off()
}

<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">efit.Treat</span>)) {
  <span class="pl-smi">col.pan</span> <span class="pl-k">&lt;-</span> colorpanel(<span class="pl-c1">100</span>, <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>yellow<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>)
  <span class="pl-smi">To</span> <span class="pl-k">&lt;-</span> topTable(<span class="pl-smi">efit.Treat</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>,<span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  <span class="pl-smi">logCPM</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">cpmFullDGE</span>[rownames(<span class="pl-smi">To</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">50</span>,]),]
  <span class="pl-smi">logCPM</span> <span class="pl-k">&lt;-</span> t(scale(t(<span class="pl-smi">logCPM</span>)))
  pdf(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/heatmaps/Heatmap-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.pdf<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>), <span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">10</span>, <span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">10</span>)
  heatmap.2(<span class="pl-smi">logCPM</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">col.pan</span>, <span class="pl-v">Rowv</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">scale</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>,<span class="pl-v">trace</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, <span class="pl-v">dendrogram</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>column<span class="pl-pds">"</span></span>, <span class="pl-v">cexRow</span><span class="pl-k">=</span><span class="pl-c1">1</span>, <span class="pl-v">cexCol</span><span class="pl-k">=</span><span class="pl-c1">1.4</span>, <span class="pl-v">density.info</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>,<span class="pl-v">margin</span><span class="pl-k">=</span>c(<span class="pl-c1">10</span>,<span class="pl-c1">9</span>), <span class="pl-v">lhei</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">10</span>), <span class="pl-v">lwid</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">6</span>),<span class="pl-v">main</span> <span class="pl-k">=</span> paste(<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">'</span>: Heatmap of top 50 genes<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>))
  dev.off()
}

<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">efit.interactions</span>)) {
  <span class="pl-smi">col.pan</span> <span class="pl-k">&lt;-</span> colorpanel(<span class="pl-c1">100</span>, <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>yellow<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>)
  <span class="pl-smi">To</span> <span class="pl-k">&lt;-</span> topTable(<span class="pl-smi">efit.interactions</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>,<span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  <span class="pl-smi">logCPM</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">cpmFullDGE</span>[rownames(<span class="pl-smi">To</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">50</span>,]),]
  <span class="pl-smi">logCPM</span> <span class="pl-k">&lt;-</span> t(scale(t(<span class="pl-smi">logCPM</span>)))
  pdf(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/heatmaps/Heatmap-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.pdf<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>), <span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">10</span>, <span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">10</span>)
  heatmap.2(<span class="pl-smi">logCPM</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">col.pan</span>, <span class="pl-v">Rowv</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">scale</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>,<span class="pl-v">trace</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, <span class="pl-v">dendrogram</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>column<span class="pl-pds">"</span></span>, <span class="pl-v">cexRow</span><span class="pl-k">=</span><span class="pl-c1">1</span>, <span class="pl-v">cexCol</span><span class="pl-k">=</span><span class="pl-c1">1.4</span>, <span class="pl-v">density.info</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>,<span class="pl-v">margin</span><span class="pl-k">=</span>c(<span class="pl-c1">10</span>,<span class="pl-c1">9</span>), <span class="pl-v">lhei</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">10</span>), <span class="pl-v">lwid</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">6</span>),<span class="pl-v">main</span> <span class="pl-k">=</span> paste(<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">'</span>: Heatmap of top 50 genes<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>))
  dev.off()
}

<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">efit.other</span>)) {
  <span class="pl-smi">col.pan</span> <span class="pl-k">&lt;-</span> colorpanel(<span class="pl-c1">100</span>, <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>yellow<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>)
  <span class="pl-smi">To</span> <span class="pl-k">&lt;-</span> topTable(<span class="pl-smi">efit.other</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>,<span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  <span class="pl-smi">logCPM</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">cpmFullDGE</span>[rownames(<span class="pl-smi">To</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">50</span>,]),]
  <span class="pl-smi">logCPM</span> <span class="pl-k">&lt;-</span> t(scale(t(<span class="pl-smi">logCPM</span>)))
  pdf(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/heatmaps/Heatmap-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.pdf<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>), <span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">10</span>, <span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">10</span>)
  heatmap.2(<span class="pl-smi">logCPM</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">col.pan</span>, <span class="pl-v">Rowv</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">scale</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>,<span class="pl-v">trace</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, <span class="pl-v">dendrogram</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>column<span class="pl-pds">"</span></span>, <span class="pl-v">cexRow</span><span class="pl-k">=</span><span class="pl-c1">1</span>, <span class="pl-v">cexCol</span><span class="pl-k">=</span><span class="pl-c1">1.4</span>, <span class="pl-v">density.info</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>,<span class="pl-v">margin</span><span class="pl-k">=</span>c(<span class="pl-c1">10</span>,<span class="pl-c1">9</span>), <span class="pl-v">lhei</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">10</span>), <span class="pl-v">lwid</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">6</span>),<span class="pl-v">main</span> <span class="pl-k">=</span> paste(<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">'</span>: Heatmap of top 50 genes<span class="pl-pds">'</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>))
  dev.off()
}

  
<span class="pl-c"><span class="pl-c">#</span># GO-term enrichment </span>
library(<span class="pl-smi">goseq</span>)
dir.create(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/GO<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>))
<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">efit.Treat</span><span class="pl-k">$</span><span class="pl-smi">lods</span>)) {
  print(<span class="pl-smi">name</span>)
  <span class="pl-v">CS</span><span class="pl-k">=</span>topTable(<span class="pl-smi">efit.Treat</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>, <span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  <span class="pl-v">genes</span><span class="pl-k">=</span>as.integer(<span class="pl-smi">CS</span><span class="pl-k">$</span><span class="pl-smi">adj.P.Val</span><span class="pl-k">&lt;</span><span class="pl-c1">0.05</span>)
  names(<span class="pl-smi">genes</span>)<span class="pl-k">=</span>row.names(<span class="pl-smi">CS</span>)
  <span class="pl-v">pwf</span><span class="pl-k">=</span>nullp(<span class="pl-smi">genes</span>,<span class="pl-s"><span class="pl-pds">"</span>dm6<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>ensGene<span class="pl-pds">"</span></span>)
  <span class="pl-v">GO.wall</span><span class="pl-k">=</span>goseq(<span class="pl-smi">pwf</span>,<span class="pl-s"><span class="pl-pds">"</span>dm6<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>ensGene<span class="pl-pds">"</span></span>)
  <span class="pl-v">GO.wall</span><span class="pl-k">=</span><span class="pl-k">data.frame</span>(<span class="pl-smi">GO.wall</span>,<span class="pl-s"><span class="pl-pds">"</span>P.adjus_BH<span class="pl-pds">"</span></span><span class="pl-k">=</span>p.adjust(<span class="pl-smi">GO.wall</span><span class="pl-k">$</span><span class="pl-smi">over_represented_pvalue</span>,<span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>))
  <span class="pl-v">name</span><span class="pl-k">=</span>gsub(<span class="pl-s"><span class="pl-pds">"</span>:<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>)
  <span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/GO/GO-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
  write.table(<span class="pl-smi">GO.wall</span>,<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)
}
<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">efit.Treat</span><span class="pl-k">$</span><span class="pl-smi">lods</span>)) {
goana(<span class="pl-smi">efit.Treat</span>, <span class="pl-v">coef</span> <span class="pl-k">=</span> <span class="pl-smi">name</span>, <span class="pl-v">FDR</span> <span class="pl-k">=</span> <span class="pl-c1">0.05</span>, <span class="pl-v">trend</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>,<span class="pl-v">species</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Dm<span class="pl-pds">"</span></span>,<span class="pl-v">convert</span><span class="pl-k">=</span><span class="pl-c1">F</span>)
}

require(<span class="pl-s"><span class="pl-pds">"</span>org.Dm.eg.db<span class="pl-pds">"</span></span>, <span class="pl-v">character.only</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
<span class="pl-smi">GO2ALLEGS</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>org.Dm.egGO2ALLEGS<span class="pl-pds">"</span></span>
<span class="pl-smi">EG.GO</span> <span class="pl-k">&lt;-</span> <span class="pl-e">AnnotationDbi</span><span class="pl-k">::</span>toTable(get(<span class="pl-smi">GO2ALLEGS</span>))
<span class="pl-smi">d</span> <span class="pl-k">&lt;-</span> duplicated(<span class="pl-smi">EG.GO</span>[, c(<span class="pl-s"><span class="pl-pds">"</span>gene_id<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>go_id<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Ontology<span class="pl-pds">"</span></span>)])
<span class="pl-smi">EG.GO</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">EG.GO</span>[<span class="pl-k">!</span><span class="pl-smi">d</span>, ]
<span class="pl-smi">de.by.go</span> <span class="pl-k">&lt;-</span> split(<span class="pl-smi">EG.GO</span><span class="pl-k">$</span><span class="pl-smi">gene_id</span>, paste(<span class="pl-smi">EG.GO</span><span class="pl-k">$</span><span class="pl-smi">go_id</span>, <span class="pl-smi">EG.GO</span><span class="pl-k">$</span><span class="pl-smi">Ontology</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>))

dir.create(paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/topgo<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>))
<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">efit.Treat</span><span class="pl-k">$</span><span class="pl-smi">lods</span>)) {
  print(<span class="pl-smi">name</span>)
  <span class="pl-v">CS</span><span class="pl-k">=</span>topTable(<span class="pl-smi">efit.Treat</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>, <span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  <span class="pl-v">genes</span><span class="pl-k">=</span>as.integer(<span class="pl-smi">CS</span><span class="pl-k">$</span><span class="pl-smi">adj.P.Val</span><span class="pl-k">&lt;</span><span class="pl-c1">0.05</span>)
  <span class="pl-v">E</span><span class="pl-k">=</span>mapIds(<span class="pl-smi">org.Dm.eg.db</span>,
           <span class="pl-v">keys</span><span class="pl-k">=</span>row.names(<span class="pl-smi">CS</span>),
           <span class="pl-v">column</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>ENTREZID<span class="pl-pds">"</span></span>,
           <span class="pl-v">keytype</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>FLYBASE<span class="pl-pds">"</span></span>,
           <span class="pl-v">multiVals</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>first<span class="pl-pds">"</span></span>)
  names(<span class="pl-smi">genes</span>)<span class="pl-k">=</span><span class="pl-smi">E</span>
  <span class="pl-v">cand</span><span class="pl-k">=</span>subset(<span class="pl-smi">genes</span>,<span class="pl-smi">genes</span><span class="pl-k">==</span><span class="pl-c1">1</span>)
  <span class="pl-v">GO</span><span class="pl-k">=</span>goana(names(<span class="pl-smi">cand</span>),<span class="pl-v">universe</span><span class="pl-k">=</span>names(<span class="pl-smi">genes</span>),<span class="pl-v">species</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Dm<span class="pl-pds">"</span></span>,<span class="pl-v">trend</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Length<span class="pl-pds">"</span></span>)
  
  <span class="pl-v">GO</span><span class="pl-k">=</span><span class="pl-k">data.frame</span>(<span class="pl-smi">GO</span>,<span class="pl-s"><span class="pl-pds">"</span>P.adjus_BH<span class="pl-pds">"</span></span><span class="pl-k">=</span>p.adjust(<span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">P.DE</span>,<span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>))
  
  <span class="pl-smi">DATAB</span> <span class="pl-k">&lt;-</span> lapply(<span class="pl-smi">de.by.go</span>, <span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>) { <span class="pl-smi">x</span>[<span class="pl-smi">x</span> <span class="pl-k">%in%</span> names(<span class="pl-smi">cand</span>)] })
  <span class="pl-k">for</span> (<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>nrow(<span class="pl-smi">GO</span>)){
    <span class="pl-v">x</span><span class="pl-k">=</span>paste(rownames(<span class="pl-smi">GO</span>)[<span class="pl-smi">i</span>],<span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">Ont</span>[<span class="pl-smi">i</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>)
    <span class="pl-k">if</span> (length(<span class="pl-smi">DATAB</span>[[<span class="pl-smi">x</span>]])<span class="pl-k">!=</span><span class="pl-c1">0</span>){
      <span class="pl-v">E</span><span class="pl-k">=</span>mapIds(<span class="pl-smi">org.Dm.eg.db</span>,
               <span class="pl-v">keys</span><span class="pl-k">=</span><span class="pl-smi">DATAB</span>[[<span class="pl-smi">x</span>]],
               <span class="pl-v">column</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>FLYBASE<span class="pl-pds">"</span></span>,
               <span class="pl-v">keytype</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>ENTREZID<span class="pl-pds">"</span></span>,
               <span class="pl-v">multiVals</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>first<span class="pl-pds">"</span></span>)
      <span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">i</span>]<span class="pl-k">=</span>paste(<span class="pl-smi">E</span>,<span class="pl-v">collapse</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>)
    }
    <span class="pl-k">else</span> {
      <span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">i</span>]<span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>  
    }}
  
  <span class="pl-v">name</span><span class="pl-k">=</span>gsub(<span class="pl-s"><span class="pl-pds">"</span>:<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>)
  <span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/topgo/GO-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
  write.table(<span class="pl-smi">GO</span>[ order(<span class="pl-smi">GO</span>[,<span class="pl-c1">6</span>]),],<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">T</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>$<span class="pl-pds">"</span></span>)
}

<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">efit.Temp</span><span class="pl-k">$</span><span class="pl-smi">lods</span>)) {
  print(<span class="pl-smi">name</span>)
  <span class="pl-v">CS</span><span class="pl-k">=</span>topTable(<span class="pl-smi">efit.Temp</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>, <span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  <span class="pl-v">genes</span><span class="pl-k">=</span>as.integer(<span class="pl-smi">CS</span><span class="pl-k">$</span><span class="pl-smi">adj.P.Val</span><span class="pl-k">&lt;</span><span class="pl-c1">0.05</span>)
  <span class="pl-v">E</span><span class="pl-k">=</span>mapIds(<span class="pl-smi">org.Dm.eg.db</span>,
           <span class="pl-v">keys</span><span class="pl-k">=</span>row.names(<span class="pl-smi">CS</span>),
           <span class="pl-v">column</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>ENTREZID<span class="pl-pds">"</span></span>,
           <span class="pl-v">keytype</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>FLYBASE<span class="pl-pds">"</span></span>,
           <span class="pl-v">multiVals</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>first<span class="pl-pds">"</span></span>)
  names(<span class="pl-smi">genes</span>)<span class="pl-k">=</span><span class="pl-smi">E</span>
  <span class="pl-v">cand</span><span class="pl-k">=</span>subset(<span class="pl-smi">genes</span>,<span class="pl-smi">genes</span><span class="pl-k">==</span><span class="pl-c1">1</span>)
  <span class="pl-v">GO</span><span class="pl-k">=</span>goana(names(<span class="pl-smi">cand</span>),<span class="pl-v">universe</span><span class="pl-k">=</span>names(<span class="pl-smi">genes</span>),<span class="pl-v">species</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Dm<span class="pl-pds">"</span></span>,<span class="pl-v">trend</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Length<span class="pl-pds">"</span></span>)
  
  <span class="pl-v">GO</span><span class="pl-k">=</span><span class="pl-k">data.frame</span>(<span class="pl-smi">GO</span>,<span class="pl-s"><span class="pl-pds">"</span>P.adjus_BH<span class="pl-pds">"</span></span><span class="pl-k">=</span>p.adjust(<span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">P.DE</span>,<span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>))
  
  <span class="pl-smi">DATAB</span> <span class="pl-k">&lt;-</span> lapply(<span class="pl-smi">de.by.go</span>, <span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>) { <span class="pl-smi">x</span>[<span class="pl-smi">x</span> <span class="pl-k">%in%</span> names(<span class="pl-smi">cand</span>)] })
  <span class="pl-k">for</span> (<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>nrow(<span class="pl-smi">GO</span>)){
    <span class="pl-v">x</span><span class="pl-k">=</span>paste(rownames(<span class="pl-smi">GO</span>)[<span class="pl-smi">i</span>],<span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">Ont</span>[<span class="pl-smi">i</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>)
    <span class="pl-k">if</span> (length(<span class="pl-smi">DATAB</span>[[<span class="pl-smi">x</span>]])<span class="pl-k">!=</span><span class="pl-c1">0</span>){
      <span class="pl-v">E</span><span class="pl-k">=</span>mapIds(<span class="pl-smi">org.Dm.eg.db</span>,
               <span class="pl-v">keys</span><span class="pl-k">=</span><span class="pl-smi">DATAB</span>[[<span class="pl-smi">x</span>]],
               <span class="pl-v">column</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>FLYBASE<span class="pl-pds">"</span></span>,
               <span class="pl-v">keytype</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>ENTREZID<span class="pl-pds">"</span></span>,
               <span class="pl-v">multiVals</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>first<span class="pl-pds">"</span></span>)
      <span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">i</span>]<span class="pl-k">=</span>paste(<span class="pl-smi">E</span>,<span class="pl-v">collapse</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>)
    }
    <span class="pl-k">else</span> {
      <span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">i</span>]<span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>  
    }}
  
  <span class="pl-v">name</span><span class="pl-k">=</span>gsub(<span class="pl-s"><span class="pl-pds">"</span>:<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>)
  <span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/topgo/GO-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
  write.table(<span class="pl-smi">GO</span>[ order(<span class="pl-smi">GO</span>[,<span class="pl-c1">6</span>]),],<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">T</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>$<span class="pl-pds">"</span></span>)
}


<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">efit.interactions</span><span class="pl-k">$</span><span class="pl-smi">lods</span>)) {
  print(<span class="pl-smi">name</span>)
  <span class="pl-v">CS</span><span class="pl-k">=</span>topTable(<span class="pl-smi">efit.interactions</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>, <span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  <span class="pl-v">genes</span><span class="pl-k">=</span>as.integer(<span class="pl-smi">CS</span><span class="pl-k">$</span><span class="pl-smi">adj.P.Val</span><span class="pl-k">&lt;</span><span class="pl-c1">0.05</span>)
  <span class="pl-v">E</span><span class="pl-k">=</span>mapIds(<span class="pl-smi">org.Dm.eg.db</span>,
           <span class="pl-v">keys</span><span class="pl-k">=</span>row.names(<span class="pl-smi">CS</span>),
           <span class="pl-v">column</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>ENTREZID<span class="pl-pds">"</span></span>,
           <span class="pl-v">keytype</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>FLYBASE<span class="pl-pds">"</span></span>,
           <span class="pl-v">multiVals</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>first<span class="pl-pds">"</span></span>)
  names(<span class="pl-smi">genes</span>)<span class="pl-k">=</span><span class="pl-smi">E</span>
  <span class="pl-v">cand</span><span class="pl-k">=</span>subset(<span class="pl-smi">genes</span>,<span class="pl-smi">genes</span><span class="pl-k">==</span><span class="pl-c1">1</span>)
  <span class="pl-v">GO</span><span class="pl-k">=</span>goana(names(<span class="pl-smi">cand</span>),<span class="pl-v">universe</span><span class="pl-k">=</span>names(<span class="pl-smi">genes</span>),<span class="pl-v">species</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Dm<span class="pl-pds">"</span></span>,<span class="pl-v">trend</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Length<span class="pl-pds">"</span></span>)
  
  <span class="pl-v">GO</span><span class="pl-k">=</span><span class="pl-k">data.frame</span>(<span class="pl-smi">GO</span>,<span class="pl-s"><span class="pl-pds">"</span>P.adjus_BH<span class="pl-pds">"</span></span><span class="pl-k">=</span>p.adjust(<span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">P.DE</span>,<span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>))
  
  <span class="pl-smi">DATAB</span> <span class="pl-k">&lt;-</span> lapply(<span class="pl-smi">de.by.go</span>, <span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>) { <span class="pl-smi">x</span>[<span class="pl-smi">x</span> <span class="pl-k">%in%</span> names(<span class="pl-smi">cand</span>)] })
  <span class="pl-k">for</span> (<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>nrow(<span class="pl-smi">GO</span>)){
    <span class="pl-v">x</span><span class="pl-k">=</span>paste(rownames(<span class="pl-smi">GO</span>)[<span class="pl-smi">i</span>],<span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">Ont</span>[<span class="pl-smi">i</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>)
    <span class="pl-k">if</span> (length(<span class="pl-smi">DATAB</span>[[<span class="pl-smi">x</span>]])<span class="pl-k">!=</span><span class="pl-c1">0</span>){
      <span class="pl-v">E</span><span class="pl-k">=</span>mapIds(<span class="pl-smi">org.Dm.eg.db</span>,
               <span class="pl-v">keys</span><span class="pl-k">=</span><span class="pl-smi">DATAB</span>[[<span class="pl-smi">x</span>]],
               <span class="pl-v">column</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>FLYBASE<span class="pl-pds">"</span></span>,
               <span class="pl-v">keytype</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>ENTREZID<span class="pl-pds">"</span></span>,
               <span class="pl-v">multiVals</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>first<span class="pl-pds">"</span></span>)
      <span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">i</span>]<span class="pl-k">=</span>paste(<span class="pl-smi">E</span>,<span class="pl-v">collapse</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>)
    }
    <span class="pl-k">else</span> {
      <span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">i</span>]<span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>  
    }}
  
  <span class="pl-v">name</span><span class="pl-k">=</span>gsub(<span class="pl-s"><span class="pl-pds">"</span>:<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>)
  <span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/topgo/GO-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
  write.table(<span class="pl-smi">GO</span>[ order(<span class="pl-smi">GO</span>[,<span class="pl-c1">6</span>]),],<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">T</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>$<span class="pl-pds">"</span></span>)
}



<span class="pl-k">for</span> (<span class="pl-smi">name</span> <span class="pl-k">in</span> colnames(<span class="pl-smi">efit.other</span><span class="pl-k">$</span><span class="pl-smi">lods</span>)) {
  print(<span class="pl-smi">name</span>)
  <span class="pl-v">CS</span><span class="pl-k">=</span>topTable(<span class="pl-smi">efit.other</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-smi">name</span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>, <span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)
  <span class="pl-v">genes</span><span class="pl-k">=</span>as.integer(<span class="pl-smi">CS</span><span class="pl-k">$</span><span class="pl-smi">adj.P.Val</span><span class="pl-k">&lt;</span><span class="pl-c1">0.05</span>)
  <span class="pl-v">E</span><span class="pl-k">=</span>mapIds(<span class="pl-smi">org.Dm.eg.db</span>,
           <span class="pl-v">keys</span><span class="pl-k">=</span>row.names(<span class="pl-smi">CS</span>),
           <span class="pl-v">column</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>ENTREZID<span class="pl-pds">"</span></span>,
           <span class="pl-v">keytype</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>FLYBASE<span class="pl-pds">"</span></span>,
           <span class="pl-v">multiVals</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>first<span class="pl-pds">"</span></span>)
  names(<span class="pl-smi">genes</span>)<span class="pl-k">=</span><span class="pl-smi">E</span>
  <span class="pl-v">cand</span><span class="pl-k">=</span>subset(<span class="pl-smi">genes</span>,<span class="pl-smi">genes</span><span class="pl-k">==</span><span class="pl-c1">1</span>)
  <span class="pl-v">GO</span><span class="pl-k">=</span>goana(names(<span class="pl-smi">cand</span>),<span class="pl-v">universe</span><span class="pl-k">=</span>names(<span class="pl-smi">genes</span>),<span class="pl-v">species</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Dm<span class="pl-pds">"</span></span>,<span class="pl-v">trend</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Length<span class="pl-pds">"</span></span>)
  
  <span class="pl-v">GO</span><span class="pl-k">=</span><span class="pl-k">data.frame</span>(<span class="pl-smi">GO</span>,<span class="pl-s"><span class="pl-pds">"</span>P.adjus_BH<span class="pl-pds">"</span></span><span class="pl-k">=</span>p.adjust(<span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">P.DE</span>,<span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>))
  
  <span class="pl-smi">DATAB</span> <span class="pl-k">&lt;-</span> lapply(<span class="pl-smi">de.by.go</span>, <span class="pl-en">FUN</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>) { <span class="pl-smi">x</span>[<span class="pl-smi">x</span> <span class="pl-k">%in%</span> names(<span class="pl-smi">cand</span>)] })
  <span class="pl-k">for</span> (<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>nrow(<span class="pl-smi">GO</span>)){
    <span class="pl-v">x</span><span class="pl-k">=</span>paste(rownames(<span class="pl-smi">GO</span>)[<span class="pl-smi">i</span>],<span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">Ont</span>[<span class="pl-smi">i</span>],<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>)
    <span class="pl-k">if</span> (length(<span class="pl-smi">DATAB</span>[[<span class="pl-smi">x</span>]])<span class="pl-k">!=</span><span class="pl-c1">0</span>){
      <span class="pl-v">E</span><span class="pl-k">=</span>mapIds(<span class="pl-smi">org.Dm.eg.db</span>,
               <span class="pl-v">keys</span><span class="pl-k">=</span><span class="pl-smi">DATAB</span>[[<span class="pl-smi">x</span>]],
               <span class="pl-v">column</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>FLYBASE<span class="pl-pds">"</span></span>,
               <span class="pl-v">keytype</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>ENTREZID<span class="pl-pds">"</span></span>,
               <span class="pl-v">multiVals</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>first<span class="pl-pds">"</span></span>)
      <span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">i</span>]<span class="pl-k">=</span>paste(<span class="pl-smi">E</span>,<span class="pl-v">collapse</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>)
    }
    <span class="pl-k">else</span> {
      <span class="pl-smi">GO</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">i</span>]<span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>  
    }}
  
  <span class="pl-v">name</span><span class="pl-k">=</span>gsub(<span class="pl-s"><span class="pl-pds">"</span>:<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>)
  <span class="pl-v">out</span><span class="pl-k">=</span>paste(<span class="pl-smi">O</span>,<span class="pl-s"><span class="pl-pds">"</span>/topgo/GO-<span class="pl-pds">"</span></span>,<span class="pl-smi">name</span>,<span class="pl-s"><span class="pl-pds">"</span>.txt<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
  write.table(<span class="pl-smi">GO</span>[ order(<span class="pl-smi">GO</span>[,<span class="pl-c1">6</span>]),],<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-smi">out</span>,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">T</span>,<span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>$<span class="pl-pds">"</span></span>)
}
</pre>
                    </div>
                    <p dir="auto">
                     Then, we added additional information, such as gene name, length and genomic coordinates to the DE datasets and isolated candidate genes based on p-value
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
cd /data/RNASeq/limma
mkdir -p /data/RNASeq/limma/FullInfo/candidates

for i in *.txt
do
    ## add info about genes
    python3 /scripts/v3/AddGeneInfo.py \
        /data/RNASeq/limma/${i} \
        /data/RNASeq/dmel-all-r6.17.gtf.gz \
        &gt; /data/RNASeq/limma/FullInfo/${i}

    ## isolate candidate genes with p&lt;0.05
    printf 'Gene\tFBgn\tlogFC\tPval\n' &gt; /data/RNASeq/limma/FullInfo/candidates/${i}
    awk '$12&lt;0.05 {print $1&quot;\t&quot;$2&quot;\t&quot;$8&quot;\t&quot;$12}' ${i} &gt;&gt; /data/RNASeq/limma/FullInfo/candidates/${i}

done" dir="auto">
                     <pre><span class="pl-c1">cd</span> /data/RNASeq/limma
mkdir -p /data/RNASeq/limma/FullInfo/candidates

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>.txt
<span class="pl-k">do</span>
    <span class="pl-c"><span class="pl-c">#</span># add info about genes</span>
    python3 /scripts/v3/AddGeneInfo.py \
        /data/RNASeq/limma/<span class="pl-smi">${i}</span> \
        /data/RNASeq/dmel-all-r6.17.gtf.gz \
        <span class="pl-k">&gt;</span> /data/RNASeq/limma/FullInfo/<span class="pl-smi">${i}</span>

    <span class="pl-c"><span class="pl-c">#</span># isolate candidate genes with p&lt;0.05</span>
    <span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">'</span>Gene\tFBgn\tlogFC\tPval\n<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> /data/RNASeq/limma/FullInfo/candidates/<span class="pl-smi">${i}</span>
    awk <span class="pl-s"><span class="pl-pds">'</span>$12&lt;0.05 {print $1"\t"$2"\t"$8"\t"$12}<span class="pl-pds">'</span></span> <span class="pl-smi">${i}</span> <span class="pl-k">&gt;&gt;</span> /data/RNASeq/limma/FullInfo/candidates/<span class="pl-smi">${i}</span>

<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      4.2) Test for positional overrepresentation
                     </h4>
                     <a aria-label="Permalink: 4.2) Test for positional overrepresentation" class="anchor" href="#42-test-for-positional-overrepresentation" id="user-content-42-test-for-positional-overrepresentation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## test for overrepresentation of candidates with Fisher's Exact tests
mkdir /data/RNASeq/overrepresentation

printf &quot;Dataset\tCand-Inv\tCand-NonInv\tNonCand-Inv\tNonCand-NonInv\tFET-pval\tCand-Inv_Breakpoint\tCand-Inv_Body\tNonCand-Inv_Breakpoint\tNonCand-Inv_Body\tFET-pval-InsideInv\n&quot; \
&gt; /data/RNASeq/overrepresentation/FET.txt

for i in /data/RNASeq/limma/FullInfo/*
do

    python3 /scripts/FET.py --input ${i} \
    &gt;&gt; /data/RNASeq/overrepresentation/FET.txt

done

## make manhattan plots
mkdir -p /data/RNASeq/overrepresentation/manhattan/logFC
mkdir -p /data/RNASeq/overrepresentation/manhattan/P

for i in /data/RNASeq/limma/FullInfo/*.txt
do

    ## Ignore datasets based on F-statistics generated above.
    if [[ ${i} != *&quot;-F-&quot;* ]]
    then

        ## make manhattan plots based on the -log10 P-values
        python3 /scripts/plot-manhattan.py \
            --full ${i} \
            --CF -2 \
            --log \
            --out /data/RNASeq/overrepresentation/manhattan/P

        ## make manhattan plots based on the log2-flod difference
        python3 /TransInv/scripts/plot-manhattan.py \
            --full ${i} \
            --CF 7 \
            --out /data/RNASeq/overrepresentation/manhattan/logFC

    fi

done" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># test for overrepresentation of candidates with Fisher's Exact tests</span>
mkdir /data/RNASeq/overrepresentation

<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>Dataset\tCand-Inv\tCand-NonInv\tNonCand-Inv\tNonCand-NonInv\tFET-pval\tCand-Inv_Breakpoint\tCand-Inv_Body\tNonCand-Inv_Breakpoint\tNonCand-Inv_Body\tFET-pval-InsideInv\n<span class="pl-pds">"</span></span> \
<span class="pl-k">&gt;</span> /data/RNASeq/overrepresentation/FET.txt

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> /data/RNASeq/limma/FullInfo/<span class="pl-k">*</span>
<span class="pl-k">do</span>

    python3 /scripts/FET.py --input <span class="pl-smi">${i}</span> \
    <span class="pl-k">&gt;&gt;</span> /data/RNASeq/overrepresentation/FET.txt

<span class="pl-k">done</span>

<span class="pl-c"><span class="pl-c">#</span># make manhattan plots</span>
mkdir -p /data/RNASeq/overrepresentation/manhattan/logFC
mkdir -p /data/RNASeq/overrepresentation/manhattan/P

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> /data/RNASeq/limma/FullInfo/<span class="pl-k">*</span>.txt
<span class="pl-k">do</span>

    <span class="pl-c"><span class="pl-c">#</span># Ignore datasets based on F-statistics generated above.</span>
    <span class="pl-k">if</span> [[ <span class="pl-smi">${i}</span> <span class="pl-k">!=</span> <span class="pl-k">*</span><span class="pl-s"><span class="pl-pds">"</span>-F-<span class="pl-pds">"</span></span><span class="pl-k">*</span> ]]
    <span class="pl-k">then</span>

        <span class="pl-c"><span class="pl-c">#</span># make manhattan plots based on the -log10 P-values</span>
        python3 /scripts/plot-manhattan.py \
            --full <span class="pl-smi">${i}</span> \
            --CF -2 \
            --log \
            --out /data/RNASeq/overrepresentation/manhattan/P

        <span class="pl-c"><span class="pl-c">#</span># make manhattan plots based on the log2-flod difference</span>
        python3 /TransInv/scripts/plot-manhattan.py \
            --full <span class="pl-smi">${i}</span> \
            --CF 7 \
            --out /data/RNASeq/overrepresentation/manhattan/logFC

    <span class="pl-k">fi</span>

<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      4.3) Compare by temperarture
                     </h4>
                     <a aria-label="Permalink: 4.3) Compare by temperarture" class="anchor" href="#43-compare-by-temperarture" id="user-content-43-compare-by-temperarture">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Next, we tested how the p-value distribution differed by temperature. We therefore manually compiled a table with all p-values from the different analyses (see data/)
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
dir.create('/data/RNASeq/CompareByTemp')

## load pvalue data from limma
Data=read.table(&quot;/data/CompareByTemp_Pval.txt&quot;,header=T)

## Compare p-value distributions of 18 and 25 temp 

pdf(&quot;/data/RNASeq/CompareByTemp/Pval-genewise-byTemp.pdf&quot;,width=12,height=8)
par(mfrow=(c(3,3)))

plot(Data$FS.FI.25,Data$FS.FI.18,col=rgb(0,0,0,0.2),pch=16,main = &quot;FS vs. FI - Temp&quot;,xlab=&quot;25C&quot;, ylab=&quot;18C&quot;)
plot(Data$FS.MS.25,Data$FS.MS.18,col=rgb(0,0,0,0.2),pch=16,main = &quot;FS vs. MS - Temp&quot;,xlab=&quot;25C&quot;, ylab=&quot;18C&quot;)
plot(Data$MS.FI.25,Data$MS.FI.18,col=rgb(0,0,0,0.2),pch=16,main = &quot;MS vs. FI - Temp&quot;,xlab=&quot;25C&quot;, ylab=&quot;18C&quot;)
plot(Data$FS.FI.25,Data$FS.FI,col=rgb(0,0,0,0.2),pch=16,main = &quot;FS vs. FI - 25 vs. Average&quot;,xlab=&quot;25C&quot;, ylab=&quot;average&quot;)
plot(Data$FS.MS.25,Data$FS.MS,col=rgb(0,0,0,0.2),pch=16,main = &quot;MS vs. FI - 25 vs. Average&quot;,xlab=&quot;25C&quot;, ylab=&quot;average&quot;)
plot(Data$MS.FI.25,Data$MS.FI,col=rgb(0,0,0,0.2),pch=16,main = &quot;MS vs. FI - 25 vs. Average&quot;,xlab=&quot;25C&quot;, ylab=&quot;average&quot;)
plot(Data$FS.FI.18,Data$FS.FI,col=rgb(0,0,0,0.2),pch=16,main = &quot;FS vs. FI - 18 vs. Average&quot;,xlab=&quot;18C&quot;, ylab=&quot;average&quot;)
plot(Data$FS.MS.18,Data$FS.MS,col=rgb(0,0,0,0.2),pch=16,main = &quot;MS vs. FI - 18 vs. Average&quot;,xlab=&quot;18C&quot;, ylab=&quot;average&quot;)
plot(Data$MS.FI.18,Data$MS.FI,col=rgb(0,0,0,0.2),pch=16,main = &quot;MS vs. FI - 18 vs. Average&quot;,xlab=&quot;18C&quot;, ylab=&quot;average&quot;)

dev.off()

pdf(&quot;/data/RNASeq/CompareByTemp/Pval-genewise-byTemp-QQ.pdf&quot;,width=12,height=8)
par(mfrow=(c(3,3)))

plot(sort(Data$FS.FI.25),sort(Data$FS.FI.18),col=rgb(0,0,0,0.2),pch=16,main = &quot;FS vs. FI - Temp&quot;,xlab=&quot;25C&quot;, ylab=&quot;18C&quot;)
abline(0,1,col=&quot;red&quot;,lty=2,lwd=2)
plot(sort(Data$FS.MS.25),sort(Data$FS.MS.18),col=rgb(0,0,0,0.2),pch=16,main = &quot;FS vs. MS - Temp&quot;,xlab=&quot;25C&quot;, ylab=&quot;18C&quot;)
abline(0,1,col=&quot;red&quot;,lty=2,lwd=2)
plot(sort(Data$MS.FI.25),sort(Data$MS.FI.18),col=rgb(0,0,0,0.2),pch=16,main = &quot;MS vs. FI - Temp&quot;,xlab=&quot;25C&quot;, ylab=&quot;18C&quot;)
abline(0,1,col=&quot;red&quot;,lty=2,lwd=2)
plot(sort(Data$FS.FI.25),sort(Data$FS.FI),col=rgb(0,0,0,0.2),pch=16,main = &quot;FS vs. FI - 25 vs. Average&quot;,xlab=&quot;25C&quot;, ylab=&quot;average&quot;)
abline(0,1,col=&quot;red&quot;,lty=2,lwd=2)
plot(sort(Data$FS.MS.25),sort(Data$FS.MS),col=rgb(0,0,0,0.2),pch=16,main = &quot;FS vs. MS - 25 vs. Average&quot;,xlab=&quot;25C&quot;, ylab=&quot;average&quot;)
abline(0,1,col=&quot;red&quot;,lty=2,lwd=2)
plot(sort(Data$MS.FI.25),sort(Data$MS.FI),col=rgb(0,0,0,0.2),pch=16,main = &quot;MS vs. FI - 25 vs. Average&quot;,xlab=&quot;25C&quot;, ylab=&quot;average&quot;)
abline(0,1,col=&quot;red&quot;,lty=2,lwd=2)
plot(sort(Data$FS.FI.18),sort(Data$FS.FI),col=rgb(0,0,0,0.2),pch=16,main = &quot;FS vs. FI - 18 vs. Average&quot;,xlab=&quot;18C&quot;, ylab=&quot;average&quot;)
abline(0,1,col=&quot;red&quot;,lty=2,lwd=2)
plot(sort(Data$FS.MS.18),sort(Data$FS.MS),col=rgb(0,0,0,0.2),pch=16,main = &quot;FS vs. MS - 18 vs. Average&quot;,xlab=&quot;18C&quot;, ylab=&quot;average&quot;)
abline(0,1,col=&quot;red&quot;,lty=2,lwd=2)
plot(sort(Data$MS.FI.18),sort(Data$MS.FI),col=rgb(0,0,0,0.2),pch=16,main = &quot;MS vs. FI - 18 vs. Average&quot;,xlab=&quot;18C&quot;, ylab=&quot;average&quot;)
abline(0,1,col=&quot;red&quot;,lty=2,lwd=2)

dev.off()
" dir="auto">
                     <pre>dir.create(<span class="pl-s"><span class="pl-pds">'</span>/data/RNASeq/CompareByTemp<span class="pl-pds">'</span></span>)

<span class="pl-c"><span class="pl-c">#</span># load pvalue data from limma</span>
<span class="pl-v">Data</span><span class="pl-k">=</span>read.table(<span class="pl-s"><span class="pl-pds">"</span>/data/CompareByTemp_Pval.txt<span class="pl-pds">"</span></span>,<span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">T</span>)

<span class="pl-c"><span class="pl-c">#</span># Compare p-value distributions of 18 and 25 temp </span>

pdf(<span class="pl-s"><span class="pl-pds">"</span>/data/RNASeq/CompareByTemp/Pval-genewise-byTemp.pdf<span class="pl-pds">"</span></span>,<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">12</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>(c(<span class="pl-c1">3</span>,<span class="pl-c1">3</span>)))

plot(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI.25</span>,<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI.18</span>,<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>FS vs. FI - Temp<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS.25</span>,<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS.18</span>,<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>FS vs. MS - Temp<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI.25</span>,<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI.18</span>,<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>MS vs. FI - Temp<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI.25</span>,<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI</span>,<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>FS vs. FI - 25 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS.25</span>,<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS</span>,<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>MS vs. FI - 25 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI.25</span>,<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI</span>,<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>MS vs. FI - 25 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI.18</span>,<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI</span>,<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>FS vs. FI - 18 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS.18</span>,<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS</span>,<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>MS vs. FI - 18 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI.18</span>,<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI</span>,<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>MS vs. FI - 18 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)

dev.off()

pdf(<span class="pl-s"><span class="pl-pds">"</span>/data/RNASeq/CompareByTemp/Pval-genewise-byTemp-QQ.pdf<span class="pl-pds">"</span></span>,<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">12</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">8</span>)
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>(c(<span class="pl-c1">3</span>,<span class="pl-c1">3</span>)))

plot(sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI.25</span>),sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI.18</span>),<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>FS vs. FI - Temp<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>)
abline(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
plot(sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS.25</span>),sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS.18</span>),<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>FS vs. MS - Temp<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>)
abline(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
plot(sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI.25</span>),sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI.18</span>),<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>MS vs. FI - Temp<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>)
abline(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
plot(sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI.25</span>),sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI</span>),<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>FS vs. FI - 25 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)
abline(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
plot(sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS.25</span>),sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS</span>),<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>FS vs. MS - 25 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)
abline(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
plot(sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI.25</span>),sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI</span>),<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>MS vs. FI - 25 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>25C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)
abline(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
plot(sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI.18</span>),sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.FI</span>),<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>FS vs. FI - 18 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)
abline(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
plot(sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS.18</span>),sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">FS.MS</span>),<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>FS vs. MS - 18 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)
abline(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
plot(sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI.18</span>),sort(<span class="pl-smi">Data</span><span class="pl-k">$</span><span class="pl-smi">MS.FI</span>),<span class="pl-v">col</span><span class="pl-k">=</span>rgb(<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0</span>,<span class="pl-c1">0.2</span>),<span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>,<span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>MS vs. FI - 18 vs. Average<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>18C<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>average<span class="pl-pds">"</span></span>)
abline(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">lwd</span><span class="pl-k">=</span><span class="pl-c1">2</span>)

dev.off()
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      4.4) Overlap between Genomic and RNASeq candidates
                     </h4>
                     <a aria-label="Permalink: 4.4) Overlap between Genomic and RNASeq candidates" class="anchor" href="#44-overlap-between-genomic-and-rnaseq-candidates" id="user-content-44-overlap-between-genomic-and-rnaseq-candidates">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Finally, we tested for overlap between genomic and RNASeq candidates based on rank-rank hypergeometric tests as implemented in the
                     <a href="https://github.com/RRHO2/RRHO2">
                      RRHO2
                     </a>
                     package in
                     <em>
                      R
                     </em>
                     . First perpare the two datasets
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
mkdir -p /data/candidates/RRHO

## rank genes based on average FST in the dataset from Florida
python3 /scripts/v3/RankGenesByStat.py \
    --input /data/PopGen/FI-FS_ann.weir.fst \
    --perc 0.25 \
    --gtf /data/RNASeq/dmel-all-r6.17.gtf.gz \
    &gt; /data/candidates/RRHO/Genes.fst


## loop through all RNASeq comparisons
for i in FS-FI FS-FI_18 FS-FI_25 FS-FI_int
do

    mkdir /data/candidates/RRHO/RRHO-${i}

    ## make input matrix for RRHO
    python3 /scripts/v3/Input4RRHO.py \
        --inputs /data/candidates/RRHO/Genes.fst,/data/RNASeq/limma/FullInfo/${i}.txt \
        --genes 1,1 \
        --stats -2,-2 \
        --type FST,P \
        &gt; /data/candidates/RRHO/Genes-${i}.rrho

    ## execute R code for each dataset 
    printf &quot;&quot;&quot;
    library(RRHO2)
    Rdata=read.table('/data/candidates/RRHO/Genes-${i}.rrho',header=T)

    DNA=data.frame('gene'=Rdata\$genes1,'val'=Rdata\$rank1)
    RNA=data.frame('gene'=Rdata\$genes2,'val'=Rdata\$rank2)
    Test=RRHO2(DNA,
        RNA,
        plots = T,
        labels=c('DNA','RNA'),
        outputdir = '/data/candidates/RRHO/RRHO-${i}',
        alternative = 'enrichment')

    &quot;&quot;&quot; &gt; /data/candidates/RRHO/RRHO-${i}/test.r

    Rscript /data/candidates/RRHO/RRHO-${i}/test.r

done
" dir="auto">
                     <pre>mkdir -p /data/candidates/RRHO

<span class="pl-c"><span class="pl-c">#</span># rank genes based on average FST in the dataset from Florida</span>
python3 /scripts/v3/RankGenesByStat.py \
    --input /data/PopGen/FI-FS_ann.weir.fst \
    --perc 0.25 \
    --gtf /data/RNASeq/dmel-all-r6.17.gtf.gz \
    <span class="pl-k">&gt;</span> /data/candidates/RRHO/Genes.fst


<span class="pl-c"><span class="pl-c">#</span># loop through all RNASeq comparisons</span>
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> FS-FI FS-FI_18 FS-FI_25 FS-FI_int
<span class="pl-k">do</span>

    mkdir /data/candidates/RRHO/RRHO-<span class="pl-smi">${i}</span>

    <span class="pl-c"><span class="pl-c">#</span># make input matrix for RRHO</span>
    python3 /scripts/v3/Input4RRHO.py \
        --inputs /data/candidates/RRHO/Genes.fst,/data/RNASeq/limma/FullInfo/<span class="pl-smi">${i}</span>.txt \
        --genes 1,1 \
        --stats -2,-2 \
        --type FST,P \
        <span class="pl-k">&gt;</span> /data/candidates/RRHO/Genes-<span class="pl-smi">${i}</span>.rrho

    <span class="pl-c"><span class="pl-c">#</span># execute R code for each dataset </span>
    <span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span><span class="pl-pds">"</span></span>
<span class="pl-s">    library(RRHO2)</span>
<span class="pl-s">    Rdata=read.table('/data/candidates/RRHO/Genes-<span class="pl-smi">${i}</span>.rrho',header=T)</span>
<span class="pl-s"></span>
<span class="pl-s">    DNA=data.frame('gene'=Rdata<span class="pl-cce">\$</span>genes1,'val'=Rdata<span class="pl-cce">\$</span>rank1)</span>
<span class="pl-s">    RNA=data.frame('gene'=Rdata<span class="pl-cce">\$</span>genes2,'val'=Rdata<span class="pl-cce">\$</span>rank2)</span>
<span class="pl-s">    Test=RRHO2(DNA,</span>
<span class="pl-s">        RNA,</span>
<span class="pl-s">        plots = T,</span>
<span class="pl-s">        labels=c('DNA','RNA'),</span>
<span class="pl-s">        outputdir = '/data/candidates/RRHO/RRHO-<span class="pl-smi">${i}</span>',</span>
<span class="pl-s">        alternative = 'enrichment')</span>
<span class="pl-s"></span>
<span class="pl-s">    <span class="pl-pds">"</span><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> /data/candidates/RRHO/RRHO-<span class="pl-smi">${i}</span>/test.r

    Rscript /data/candidates/RRHO/RRHO-<span class="pl-smi">${i}</span>/test.r

<span class="pl-k">done</span>
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      References
                     </h2>
                     <a aria-label="Permalink: References" class="anchor" href="#references" id="user-content-references">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Kapopoulou, A., Kapun, M., Pieper, B., Pavlidis, P., Wilches, R., Duchen, P., et al. 2020. Demographic analyses of a new sample of haploid genomes from a Swedish population of
                     <em>
                      Drosophila melanogaster
                     </em>
                     . Sci Rep 10: 22415. Nature Publishing Group.
                    </p>
                    <p dir="auto">
                     Kapun, M., Schalkwyk, H. van, McAllister, B., Flatt, T. &amp; Schltterer, C. 2014. Inference of chromosomal inversion dynamics from Pool-Seq data in natural and laboratory populations of
                     <em>
                      Drosophila melanogaster
                     </em>
                     . Molecular Ecology 23: 18131827.
                    </p>
                    <p dir="auto">
                     Kapun, M., Barrn, M.G., Staubach, F., Obbard, D.J., Wiberg, R.A.W., Vieira, J., et al. 2020. Genomic Analysis of European
                     <em>
                      Drosophila melanogaster
                     </em>
                     Populations Reveals Longitudinal Structure, Continent-Wide Selection, and Previously Unknown DNA Viruses. Mol Biol Evol 37: 26612678. Oxford Academic.
                    </p>
                    <p dir="auto">
                     Kapun, M., Mitchell, E.D., Kawecki, T.J., Schmidt, P. &amp; Flatt, T. 2023. An Ancestral Balanced Inversion Polymorphism Confers Global Adaptation. bioRxiv.
                    </p>
                    <p dir="auto">
                     Kofler, R., Pandey, R.V. &amp; Schlotterer, C. 2011. PoPoolation2: identifying differentiation between populations using sequencing of pooled DNA samples (Pool-Seq). Bioinformatics 27: 34353436.
                    </p>
                    <p dir="auto">
                     Lack, J.B., Lange, J.D., Tang, A.D., B, C.-D., Russell &amp; Pool, J.E. 2016. A Thousand Fly Genomes: An Expanded
                     <em>
                      Drosophila
                     </em>
                     Genome Nexus. Mol Biol Evol 33: msw195-3313.
                    </p>
                    <p dir="auto">
                     Rane, R.V., Rako, L., Kapun, M., Lee, S.F. &amp; Hoffmann, A.A. 2015. Genomic evidence for role of inversion 3RP of
                     <em>
                      Drosophila melanogaster
                     </em>
                     in facilitating climate change adaptation. Molecular Ecology 24: 24232432.
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="fL0E31caD0gv01Vgs0QnG2ljZ+QHgCqOC6wA4QglQPVKmB48BRI/i0TKra5cQNhwUgFXomDf1X3Bn7s0Txl1oA==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <div class="f4 my-3 color-fg-muted text-italic">
                 No description, website, or topics provided.
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <h3 class="sr-only">
                 License
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="#MIT-1-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                   </path>
                  </svg>
                  MIT license
                 </a>
                </div>
                <include-fragment src="/capoony/In3RPayne_PopGenomics/hovercards/citation/sidebar_partial?tree_name=master">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fcapoony%2FIn3RPayne_PopGenomics&amp;report=capoony+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/capoony/In3RPayne_PopGenomics/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/capoony/packages?repo_name=In3RPayne_PopGenomics">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/capoony/In3RPayne_PopGenomics/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="Python 98.1" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#3572A5 !important;;width: 98.1%;">
                 </span>
                 <span aria-label="Other 1.9" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#ededed !important;;width: 1.9%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/capoony/In3RPayne_PopGenomics/search?l=python">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#3572A5;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Python
                  </span>
                  <span>
                   98.1%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <span class="d-inline-flex flex-items-center flex-nowrap text-small mr-3">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#ededed;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Other
                  </span>
                  <span>
                   1.9%
                  </span>
                 </span>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
        2024 GitHub,Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You cant perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
