<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","report_hydro_web_vitals","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-ca8ae01e33b4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-fb4b8d40f206.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - dozmorovlab/excluderanges: Genomic coordinates of problematic genomic regions as GRanges
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:B3BAC6:1087AD9:6605996D" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDpCM0JBQzY6MTA4N0FEOTo2NjA1OTk2RCIsInZpc2l0b3JfaWQiOiI2NDUyMjQxODU2MjM4ODE3ODA3IiwicmVnaW9uX2VkZ2UiOiJpYWQiLCJyZWdpb25fcmVuZGVyIjoiaWFkIn0=" data-pjax-transient="true" name="visitor-payload">
           <meta content="3c92afb3bad2e1c681bb5593c94a6dff16b876dc63bcbac0438da19d8d8bbc63" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:399130487" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="Genomic coordinates of problematic genomic regions as GRanges - dozmorovlab/excluderanges" name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/dozmorovlab/excluderanges" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/5de380e139f20ebe5aab6686028933bd5933437a59c386d148a0201a469696c1/dozmorovlab/excluderanges" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - dozmorovlab/excluderanges: Genomic coordinates of problematic genomic regions as GRanges" name="twitter:title">
                    <meta content="Genomic coordinates of problematic genomic regions as GRanges - dozmorovlab/excluderanges" name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/5de380e139f20ebe5aab6686028933bd5933437a59c386d148a0201a469696c1/dozmorovlab/excluderanges" property="og:image">
                      <meta content="Genomic coordinates of problematic genomic regions as GRanges - dozmorovlab/excluderanges" property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - dozmorovlab/excluderanges: Genomic coordinates of problematic genomic regions as GRanges" property="og:title"/>
                        <meta content="https://github.com/dozmorovlab/excluderanges" property="og:url"/>
                        <meta content="Genomic coordinates of problematic genomic regions as GRanges - dozmorovlab/excluderanges" property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="584826db69b6459b3c4c29172349cd9ed6729235b962a93107214b2469aa5b40" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="a2a6f0b866a9de2e69b1f990e0730e22d502d619ef6d55aae603cacc6e2f5285" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/dozmorovlab/excluderanges git https://github.com/dozmorovlab/excluderanges.git" name="go-import"/>
                        <meta content="26965156" name="octolytics-dimension-user_id">
                         <meta content="dozmorovlab" name="octolytics-dimension-user_login">
                          <meta content="399130487" name="octolytics-dimension-repository_id">
                           <meta content="dozmorovlab/excluderanges" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="399130487" name="octolytics-dimension-repository_network_root_id">
                               <meta content="dozmorovlab/excluderanges" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/dozmorovlab/excluderanges" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-11f6759e1cef.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/dozmorovlab/excluderanges","user_id":null}}' data-hydro-click-hmac="a374aa6cafcb14ae48f0e99be03b80dcd8c7409e9bc67440e13e3cf691ad7349" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fdozmorovlab%2Fexcluderanges">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=aJYwZOwmK7Kg3n01NjXXbQjjdqxqnb6uM1bUavYyCPmKe4uwhiPRkJvy8sZATW2d8eqXgSKkOcw1KpZxefPkvxPG%2FgnEjGwHOvbS9W1Gi5TJDQngi3jaSc%2BZfj5aHCFpv2aO24PJekpA0lOns0gFHOtY1CyXpoBqjVuzoTW61edc9qbURAvGxbLNygNgn6F8790ne833qLApidNIh0EJ1r1lxJf2LvWOzCitLoOCDXTYRyQrTu%2FDIYt80EordLrVJTBsMCmeUp9tH7tmqMc%2Fk93byeb7ClrRstgRGTCk2ECSBQhMFoPdfvQdwUpa4TEMAMFLoOBgPUlE2nV29%2FLrKAgELwUWckXFDiKVdq0oFb1jRauqfbzzVCoDum2ZOPLiYp82SI%2BE96c%2FjayjfZ2Rpa8QiIYwMah%2FlgD21wYha2A3M2XtSdAGYIDpZDJ2zwx5uCaUoQYLHq75U8KZFTT9TvCZBBcvq0hSV4o5%2B4mU3MYIFvG0Q51f3kf%2BLcAtIS9W0KSpTgGnZi9pm5lJVpZhjXUUQ4tiiho5%2B2I%3D--MvWu4akwYU3ycT5Q--SP%2F6fHHQcsBFR2d1k%2FLlnQ%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="dozmorovlab" data-current-owner="" data-current-repository="dozmorovlab/excluderanges" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="1ShPIFSPOmhg5OXyCeSOMXFrSfkcGvWyTFmhaj1JlduA9hGImLeXgZ0Szx4gDESa5beIEm6FgDqQF9Pfzd1XUw" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:dozmorovlab/excluderanges">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-3d986d68-62bf-4558-a048-bed2a9ab8466" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-3d986d68-62bf-4558-a048-bed2a9ab8466">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="IFI6HIYzckL9xyuFS1FxG1YaJWfE8Tsy5p+D980/Sy2eli/q+TxUObfGSvWatagD56b+XUPqe+Kv1mdyHkEiHQ=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="E5A6F/QXS78ziMBb2z9231NTJ2y0zZFmG/Z3LSR2fNOZF5RL9D32aH8w77Di6igHljvEYrZ1w5o8UAJ2SuXXPg==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="7oxmwUw1yakljBgkrgGSgrLixmGN40xThPHXAoot1IeDQs6dbsc6y69haN4IU0SMZrkBGgDdftlU3bWl92iW5A==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="EJ3HqcGdNBb2qqhl6rXq71yWQ6tAIw61cqyQ4WCwE3fmgALPPQ4EaDMHBvxYWZ7H6tbLgui+gf5s8Cvbq8y/Zw==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/dozmorovlab/excluderanges","user_id":null}}' data-hydro-click-hmac="a374aa6cafcb14ae48f0e99be03b80dcd8c7409e9bc67440e13e3cf691ad7349" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fdozmorovlab%2Fexcluderanges">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/dozmorovlab/excluderanges","user_id":null}}' data-hydro-click-hmac="a374aa6cafcb14ae48f0e99be03b80dcd8c7409e9bc67440e13e3cf691ad7349" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=dozmorovlab%2Fexcluderanges">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-54ea53ac-2a36-4b4a-8976-3166572a63eb" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-27a47765-be5e-4d64-a8ef-acb3c6b504ce" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-27a47765-be5e-4d64-a8ef-acb3c6b504ce" id="tooltip-54ea53ac-2a36-4b4a-8976-3166572a63eb" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="organization" data-hovercard-url="/orgs/dozmorovlab/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/dozmorovlab" rel="author">
            dozmorovlab
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/dozmorovlab/excluderanges">
            excluderanges
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/dozmorovlab/excluderanges","user_id":null}}' data-hydro-click-hmac="3822ebf388e351cb494c014ce1f52f9ff9f3fa8de8c5e2d1a5c2e63b5e20c2f3" data-view-component="true" href="/login?return_to=%2Fdozmorovlab%2Fexcluderanges" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":399130487,"auth_type":"LOG_IN","originating_url":"https://github.com/dozmorovlab/excluderanges","user_id":null}}' data-hydro-click-hmac="3a8029da534ca2900957f52847925d9e08f33a8259c4f1ec7a24cc4b38ed7198" data-view-component="true" href="/login?return_to=%2Fdozmorovlab%2Fexcluderanges" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="2">
             2
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":399130487,"auth_type":"LOG_IN","originating_url":"https://github.com/dozmorovlab/excluderanges","user_id":null}}' data-hydro-click-hmac="3f0663a950a48ca3d7e57bc2973af705480f6fae3a46689bad573fee46c102d6" data-view-component="true" href="/login?return_to=%2Fdozmorovlab%2Fexcluderanges" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="19 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="19">
              19
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <p class="f4 mb-3">
          Genomic coordinates of problematic genomic regions as GRanges
         </p>
         <div class="mb-2 d-flex flex-items-center Link--secondary">
          <svg aria-hidden="true" class="octicon octicon-link flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
           </path>
          </svg>
          <span class="flex-auto min-width-0 css-truncate css-truncate-target width-fit">
           <a class="text-bold" href="https://dozmorovlab.github.io/excluderanges/" rel="noopener noreferrer" role="link" target="_blank" title="https://dozmorovlab.github.io/excluderanges/">
            dozmorovlab.github.io/excluderanges/
           </a>
          </span>
         </div>
         <h3 class="sr-only">
          License
         </h3>
         <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default d-inline mb-2">
          <summary class="Link--muted">
           <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
            </path>
           </svg>
           Unknown, MIT licenses found
          </summary>
          <details-dialog aria-label="Licenses found" class="Box Box--overlay d-flex flex-column anim-fade-in fast">
           <div class="Box-header">
            <button aria-label="Close licenses dialog" class="Box-btn-octicon btn-octicon float-right" data-action="toggle" data-close-dialog="" type="button">
             <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
              </path>
             </svg>
            </button>
            <h3 class="Box-title">
             Licenses found
            </h3>
           </div>
           <a aria-label="Unknown license" class="Link--primary no-underline" href="/dozmorovlab/excluderanges/blob/master/./LICENSE">
            <div class="Box-row Box-row--hover-gray border-top rounded-0">
             <div class="text-bold">
              Unknown
             </div>
             <span class="f6 color-fg-muted">
              LICENSE
             </span>
            </div>
           </a>
           <a aria-label="MIT license" class="Link--primary no-underline" href="/dozmorovlab/excluderanges/blob/master/./LICENSE.md">
            <div class="Box-row Box-row--hover-gray border-top rounded-0">
             <div class="text-bold">
              MIT
             </div>
             <span class="f6 color-fg-muted">
              LICENSE.md
             </span>
            </div>
           </a>
          </details-dialog>
         </details>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/dozmorovlab/excluderanges/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            19
           </span>
           stars
          </a>
          <a class="Link--secondary no-underline mr-3" href="/dozmorovlab/excluderanges/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            2
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/dozmorovlab/excluderanges/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/dozmorovlab/excluderanges/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/dozmorovlab/excluderanges/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":399130487,"auth_type":"LOG_IN","originating_url":"https://github.com/dozmorovlab/excluderanges","user_id":null}}' data-hydro-click-hmac="3f0663a950a48ca3d7e57bc2973af705480f6fae3a46689bad573fee46c102d6" data-view-component="true" href="/login?return_to=%2Fdozmorovlab%2Fexcluderanges" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/dozmorovlab/excluderanges","user_id":null}}' data-hydro-click-hmac="3822ebf388e351cb494c014ce1f52f9ff9f3fa8de8c5e2d1a5c2e63b5e20c2f3" data-view-component="true" href="/login?return_to=%2Fdozmorovlab%2Fexcluderanges" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /dozmorovlab/excluderanges" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/dozmorovlab/excluderanges" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /dozmorovlab/excluderanges/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/dozmorovlab/excluderanges/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="issues-repo-tab-count" title="1">
            1
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /dozmorovlab/excluderanges/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/dozmorovlab/excluderanges/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /dozmorovlab/excluderanges/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/dozmorovlab/excluderanges/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /dozmorovlab/excluderanges/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/dozmorovlab/excluderanges/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /dozmorovlab/excluderanges/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/dozmorovlab/excluderanges/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/dozmorovlab/excluderanges/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /dozmorovlab/excluderanges/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/dozmorovlab/excluderanges/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-46a3198c-36a6-4c5e-9de1-090623513624-list" aria-haspopup="true" aria-labelledby="tooltip-b95dddcb-a532-4943-8ff9-f5c2d5ac2f6e" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-46a3198c-36a6-4c5e-9de1-090623513624-button" popovertarget="action-menu-46a3198c-36a6-4c5e-9de1-090623513624-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-46a3198c-36a6-4c5e-9de1-090623513624-button" id="tooltip-b95dddcb-a532-4943-8ff9-f5c2d5ac2f6e" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-46a3198c-36a6-4c5e-9de1-090623513624-button" anchor-offset="normal" data-view-component="true" id="action-menu-46a3198c-36a6-4c5e-9de1-090623513624-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-46a3198c-36a6-4c5e-9de1-090623513624-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-46a3198c-36a6-4c5e-9de1-090623513624-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/dozmorovlab/excluderanges" id="item-c51ff4df-3e18-4adc-b9a7-16d16488ca2f" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/dozmorovlab/excluderanges/issues" id="item-c81e5ed6-3a53-4c40-84ff-95e73accb23c" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/dozmorovlab/excluderanges/pulls" id="item-c7d6ef61-a653-40c2-9100-64cde6b07568" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/dozmorovlab/excluderanges/actions" id="item-291df095-1c70-4e35-9b3a-a69c89925c82" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/dozmorovlab/excluderanges/projects" id="item-7ae585c4-6a44-4fb8-88f9-be249f2f1ac6" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/dozmorovlab/excluderanges/security" id="item-e17f8a9f-f4ab-4e88-b195-e4d6c459b910" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/dozmorovlab/excluderanges/pulse" id="item-7e7e7964-ab74-464a-ade2-3dc044ebb39a" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         dozmorovlab/excluderanges
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/dozmorovlab/excluderanges/spoofed_commit_check/26de0ddc03f806ae211cf0defda4a7490dcf0121">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":399130487,"defaultBranch":"master","name":"excluderanges","ownerLogin":"dozmorovlab","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-08-23T14:14:51.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/26965156?v=4","public":true,"private":false,"isOrgOwned":true},"currentUser":null,"refInfo":{"name":"master","listCacheKey":"v0:1668947541.2632792","canEdit":false,"refType":"branch","currentOid":"26de0ddc03f806ae211cf0defda4a7490dcf0121"},"tree":{"items":[{"name":".github","path":".github","contentType":"directory"},{"name":"R","path":"R","contentType":"directory"},{"name":"inst","path":"inst","contentType":"directory"},{"name":"man","path":"man","contentType":"directory"},{"name":"vignettes","path":"vignettes","contentType":"directory"},{"name":"DESCRIPTION","path":"DESCRIPTION","contentType":"file"},{"name":"LICENSE","path":"LICENSE","contentType":"file"},{"name":"LICENSE.md","path":"LICENSE.md","contentType":"file"},{"name":"NAMESPACE","path":"NAMESPACE","contentType":"file"},{"name":"NEWS.md","path":"NEWS.md","contentType":"file"},{"name":"README.Rmd","path":"README.Rmd","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":12,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/dozmorovlab/excluderanges/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/dozmorovlab/excluderanges.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone dozmorovlab/excluderanges","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/dozmorovlab/excluderanges/archive/refs/heads/master.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=399130487"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"92","overviewFiles":[{"displayName":"README.md","repoName":"excluderanges","refName":"master","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#excluderanges-genomic-ranges-of-problematic-genomic-regions\"\u003eexcluderanges, genomic ranges of problematic genomic\nregions\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#installation-instructions\"\u003eInstallation instructions\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#use-excluderanges\"\u003eUse excluderanges\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#compare-the-number-of-excludable-regions\"\u003eCompare the number of excludable\nregions\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#compare-the-width-of-excludable-regions\"\u003eCompare the width of excludable\nregions\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#compare-overlaps-among-sets\"\u003eCompare overlaps among sets\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#metadata-analysis\"\u003eMetadata analysis\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bedbase-data-download\"\u003eBEDbase data download\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#mitochondrial-dna-sequences-numts\"\u003eMitochondrial DNA sequences,\nNUMTs\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#centromeres-telomeres-etc\"\u003eCentromeres, telomeres, etc.\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#gaps-in-mouse-genome\"\u003eGaps in mouse genome\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#cutrun-excludable-sets\"\u003eCUT\u0026amp;RUN excludable sets\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#summary-table\"\u003eSummary table\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#citation\"\u003eCitation\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#code-of-conduct\"\u003eCode of Conduct\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#references\"\u003eReferences\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eexcluderanges, genomic ranges of problematic genomic regions\u003c/h1\u003e\u003ca id=\"user-content-excluderanges-genomic-ranges-of-problematic-genomic-regions\" class=\"anchor\" aria-label=\"Permalink: excluderanges, genomic ranges of problematic genomic regions\" href=\"#excluderanges-genomic-ranges-of-problematic-genomic-regions\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOgata, Jonathan D, Wancen Mu, Eric S Davis, Bingjie Xue, J Chuck\nHarrell, Nathan C Sheffield, Douglas H Phanstiel, Michael I Love, and\nMikhail G Dozmorov. “Excluderanges: Exclusion Sets for T2T-CHM13,\nGRCm39, and Other Genome Assemblies.” \u003cem\u003eBioinformatics\u003c/em\u003e, April 3, 2023,\nPMID: \u003ca href=\"https://pubmed.ncbi.nlm.nih.gov/37067481/\" rel=\"nofollow\"\u003e37067481\u003c/a\u003e, DOI:\n\u003ca href=\"https://doi.org/10.1093/bioinformatics/btad198\" rel=\"nofollow\"\u003e10.1093/bioinformatics/btad198\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/dozmorovlab/excluderanges/blob/master/man/figures/logo.png\"\u003e\u003cimg id=\"user-content-excluderanges_logo_by_mike_love\" src=\"/dozmorovlab/excluderanges/raw/master/man/figures/logo.png\" align=\"right\" width=\"125\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCoordinates of problematic genomic regions that should be avoided when\nworking with genomic data (Ogata et al. 2023). For human, mouse, and\nselected model organisms. Please, open an\n\u003ca href=\"https://github.com/dozmorovlab/excluderanges/issues\"\u003eissue\u003c/a\u003e to suggest\nother resources. Package on Bioconductor:\n\u003cem\u003e\u003ca href=\"https://bioconductor.org/packages/3.18/excluderanges\" rel=\"nofollow\"\u003eexcluderanges\u003c/a\u003e\u003c/em\u003e.\u003c/p\u003e\n\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNew (2022-09-20)\u003c/strong\u003e - Exclusion sets for human\n\u003ca href=\"http://bedbase.org/#/bedsplash/8329d8c624880308ab51ba05149a737d\" rel=\"nofollow\"\u003eT2T-CHM13\u003c/a\u003e\nand mouse\n\u003ca href=\"http://bedbase.org/#/bedsplash/edc716833d4b5ee75c34a0692fc353d5\" rel=\"nofollow\"\u003eGRCm39/mm39\u003c/a\u003e\ngenome assemblies are available. Download all data from \u003ca href=\"https://drive.google.com/drive/folders/1sF9m8Y3eZouTZ3IEEywjs2kfHOWFBSJT?usp=sharing\" rel=\"nofollow\"\u003eGoogle\nDrive\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTL;DR\u003c/strong\u003e - For human hg38 genome assembly,\n\u003ca href=\"https://twitter.com/anshulkundaje/status/1263546023151992832?s=20\" rel=\"nofollow\"\u003eAnshul\u003c/a\u003e\nand we recommend \u003ca href=\"https://www.encodeproject.org/files/ENCFF356LFX/\" rel=\"nofollow\"\u003eENCFF356LFX exclusion list\nregions\u003c/a\u003e. Also\navailable as \u003ccode\u003ehg38.Kundaje.GRCh38_unified_Excludable\u003c/code\u003e excluderanges\nobject (AnnotationHub ID: AH107305) and\n\u003ca href=\"http://bedbase.org/#/bedsplash/1a561729234c2844303a051b16f66656\" rel=\"nofollow\"\u003eBEDbase.org\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eBED files of exclusion regions are available on the \u003ca href=\"https://www.encodeproject.org/search/?searchTerm=exclusion+list\" rel=\"nofollow\"\u003eENCODE\nproject\u003c/a\u003e\nwebsite and scattered across various websites, such as\n\u003ca href=\"https://github.com/Boyle-Lab/Blacklist/tree/master/lists\"\u003eBlacklist\u003c/a\u003e\n(Amemiya, Kundaje, and Boyle 2019),\n\u003ca href=\"https://github.com/ewimberley/peakPass/tree/main/excludedlists\"\u003ePeakpass\u003c/a\u003e\n(Wimberley and Heber 2019),\n\u003ca href=\"https://github.com/sklasfeld/GreenscreenProject/tree/main/data\"\u003eGreenscreen\u003c/a\u003e\n(Klasfeld and Wagner 2022). Human and mouse genome assemblies have the\nlargest number of exclusion sets generated by multiple labs. These\nexclusion sets frequently lack annotation and curation methods, creating\nuncertainty what to use. The purpose of this package is to provide a\nunified place for informed retrieval of exclusion regions.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNaming convention: \u003ccode\u003e\u0026lt;genome assembly\u0026gt;.\u0026lt;lab\u0026gt;.\u0026lt;original file name\u0026gt;\u003c/code\u003e, e.g.,\n\u003ccode\u003ehg19.Birney.wgEncodeDacMapabilityConsensusExcludable\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSee \u003ca href=\"/dozmorovlab/excluderanges/blob/master/inst/scripts/make-data.R\"\u003emake-data.R\u003c/a\u003e how we created excluderanges\nobjects.\u003c/p\u003e\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInstallation instructions\u003c/h2\u003e\u003ca id=\"user-content-installation-instructions\" class=\"anchor\" aria-label=\"Permalink: Installation instructions\" href=\"#installation-instructions\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://www.bioconductor.org/install/#install-R\" rel=\"nofollow\"\u003eInstall\u003c/a\u003e the latest\nrelease of R, then get the latest version of Bioconductor by starting R\nand entering the commands:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# if (!require(\u0026quot;BiocManager\u0026quot;, quietly = TRUE))\n#     install.packages(\u0026quot;BiocManager\u0026quot;)\n# BiocManager::install(version = \u0026quot;3.18\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e if (!require(\"BiocManager\", quietly = TRUE))\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e     install.packages(\"BiocManager\")\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e BiocManager::install(version = \"3.18\")\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThen, install additional packages using the following code:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# BiocManager::install(\u0026quot;AnnotationHub\u0026quot;, update = FALSE) \n# BiocManager::install(\u0026quot;GenomicRanges\u0026quot;, update = FALSE)\n# BiocManager::install(\u0026quot;plyranges\u0026quot;, update = FALSE)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e BiocManager::install(\"AnnotationHub\", update = FALSE) \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e BiocManager::install(\"GenomicRanges\", update = FALSE)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e BiocManager::install(\"plyranges\", update = FALSE)\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eUse excluderanges\u003c/h1\u003e\u003ca id=\"user-content-use-excluderanges\" class=\"anchor\" aria-label=\"Permalink: Use excluderanges\" href=\"#use-excluderanges\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eGet an overview of what’s available\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"suppressMessages(library(GenomicRanges))\nsuppressMessages(library(AnnotationHub))\nah \u0026lt;- AnnotationHub()\nquery_data \u0026lt;- subset(ah, preparerclass == \u0026quot;excluderanges\u0026quot;)\n# You can search for multiple terms\n# query_data \u0026lt;- query(ah, c(\u0026quot;excluderanges\u0026quot;, \u0026quot;Kundaje\u0026quot;, \u0026quot;hg38\u0026quot;))\nquery_data\n#\u0026gt; AnnotationHub with 82 records\n#\u0026gt; # snapshotDate(): 2023-10-20\n#\u0026gt; # $dataprovider: UCSC, GitHub, ENCODE, UCSChub, excluderanges, Stanford.edu,...\n#\u0026gt; # $species: Homo sapiens, Mus musculus, Drosophila melanogaster, Danio rerio...\n#\u0026gt; # $rdataclass: GRanges\n#\u0026gt; # additional mcols(): taxonomyid, genome, description,\n#\u0026gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,\n#\u0026gt; #   rdatapath, sourceurl, sourcetype \n#\u0026gt; # retrieve records with, e.g., 'object[[\u0026quot;AH107304\u0026quot;]]' \n#\u0026gt; \n#\u0026gt;              title                                 \n#\u0026gt;   AH107304 | T2T.excluderanges                     \n#\u0026gt;   AH107305 | hg38.Kundaje.GRCh38_unified_Excludable\n#\u0026gt;   AH107306 | hg38.Bernstein.Mint_Excludable_GRCh38 \n#\u0026gt;   AH107307 | hg38.Boyle.hg38-Excludable.v2         \n#\u0026gt;   AH107308 | hg38.Kundaje.GRCh38.Excludable        \n#\u0026gt;   ...        ...                                   \n#\u0026gt;   AH107381 | danRer10.UCSC.scaffold                \n#\u0026gt;   AH107382 | dm6.UCSC.other                        \n#\u0026gt;   AH107383 | dm3.UCSC.contig                       \n#\u0026gt;   AH107384 | dm3.UCSC.scaffold                     \n#\u0026gt;   AH107385 | TAIR10.UCSC.araTha1.gap\"\u003e\u003cpre\u003esuppressMessages(library(\u003cspan class=\"pl-smi\"\u003eGenomicRanges\u003c/span\u003e))\nsuppressMessages(library(\u003cspan class=\"pl-smi\"\u003eAnnotationHub\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eah\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e AnnotationHub()\n\u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e subset(\u003cspan class=\"pl-smi\"\u003eah\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epreparerclass\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eexcluderanges\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e You can search for multiple terms\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e query_data \u0026lt;- query(ah, c(\"excluderanges\", \"Kundaje\", \"hg38\"))\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; AnnotationHub with 82 records\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # snapshotDate(): 2023-10-20\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # $dataprovider: UCSC, GitHub, ENCODE, UCSChub, excluderanges, Stanford.edu,...\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # $species: Homo sapiens, Mus musculus, Drosophila melanogaster, Danio rerio...\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # $rdataclass: GRanges\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # additional mcols(): taxonomyid, genome, description,\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; #   rdatapath, sourceurl, sourcetype \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # retrieve records with, e.g., 'object[[\"AH107304\"]]' \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;              title                                 \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107304 | T2T.excluderanges                     \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107305 | hg38.Kundaje.GRCh38_unified_Excludable\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107306 | hg38.Bernstein.Mint_Excludable_GRCh38 \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107307 | hg38.Boyle.hg38-Excludable.v2         \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107308 | hg38.Kundaje.GRCh38.Excludable        \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   ...        ...                                   \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107381 | danRer10.UCSC.scaffold                \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107382 | dm6.UCSC.other                        \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107383 | dm3.UCSC.contig                       \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107384 | dm3.UCSC.scaffold                     \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107385 | TAIR10.UCSC.araTha1.gap\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003ehg38.Kundaje.GRCh38_unified_Excludable\u003c/code\u003e object recommended by Anshul\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"excludeGR.hg38.Kundaje.1 \u0026lt;- query_data[[\u0026quot;AH107305\u0026quot;]]\n#\u0026gt; loading from cache\n# Always a good idea to sort GRanges and keep standard chromosomes\nexcludeGR.hg38.Kundaje.1 \u0026lt;- excludeGR.hg38.Kundaje.1 %\u0026gt;% \n  sort() %\u0026gt;% keepStandardChromosomes(pruning.mode = \u0026quot;tidy\u0026quot;)\nexcludeGR.hg38.Kundaje.1\n#\u0026gt; GRanges object with 910 ranges and 0 metadata columns:\n#\u0026gt;         seqnames            ranges strand\n#\u0026gt;            \u0026lt;Rle\u0026gt;         \u0026lt;IRanges\u0026gt;  \u0026lt;Rle\u0026gt;\n#\u0026gt;     [1]     chr1     628903-635104      *\n#\u0026gt;     [2]     chr1   5850087-5850571      *\n#\u0026gt;     [3]     chr1   8909610-8910014      *\n#\u0026gt;     [4]     chr1   9574580-9574997      *\n#\u0026gt;     [5]     chr1 32043823-32044203      *\n#\u0026gt;     ...      ...               ...    ...\n#\u0026gt;   [906]     chrY 11290797-11334278      *\n#\u0026gt;   [907]     chrY 11493053-11592850      *\n#\u0026gt;   [908]     chrY 11671014-11671046      *\n#\u0026gt;   [909]     chrY 11721528-11749472      *\n#\u0026gt;   [910]     chrY 56694632-56889743      *\n#\u0026gt;   -------\n#\u0026gt;   seqinfo: 24 sequences from hg38 genome\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH107305\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; loading from cache\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Always a good idea to sort GRanges and keep standard chromosomes\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.1\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  sort() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% keepStandardChromosomes(\u003cspan class=\"pl-v\"\u003epruning.mode\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etidy\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; GRanges object with 910 ranges and 0 metadata columns:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;         seqnames            ranges strand\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;            \u0026lt;Rle\u0026gt;         \u0026lt;IRanges\u0026gt;  \u0026lt;Rle\u0026gt;\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [1]     chr1     628903-635104      *\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [2]     chr1   5850087-5850571      *\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [3]     chr1   8909610-8910014      *\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [4]     chr1   9574580-9574997      *\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [5]     chr1 32043823-32044203      *\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     ...      ...               ...    ...\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [906]     chrY 11290797-11334278      *\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [907]     chrY 11493053-11592850      *\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [908]     chrY 11671014-11671046      *\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [909]     chrY 11721528-11749472      *\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [910]     chrY 56694632-56889743      *\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   -------\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   seqinfo: 24 sequences from hg38 genome\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSave the data in a BED file, if needed.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Note that rtracklayer::import and rtracklayer::export perform unexplained\n# start coordinate conversion, likely related to 0- and 1-based coordinate\n# system. We recommend converting GRanges to a data frame and save tab-separated\nwrite.table(as.data.frame(excludeGR.hg38.Kundaje.1), \n            file = \u0026quot;hg38.Kundaje.GRCh38_unified_Excludable.bed\u0026quot;,\n            sep = \u0026quot;\\t\u0026quot;, row.names = FALSE, col.names = FALSE, quote = FALSE)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Note that rtracklayer::import and rtracklayer::export perform unexplained\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e start coordinate conversion, likely related to 0- and 1-based coordinate\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e system. We recommend converting GRanges to a data frame and save tab-separated\u003c/span\u003e\nwrite.table(as.data.frame(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.1\u003c/span\u003e), \n            \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehg38.Kundaje.GRCh38_unified_Excludable.bed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can load other excludable regions for the hg38 genome assembly and\ncompare them.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"query_data \u0026lt;- query(ah, c(\u0026quot;excluderanges\u0026quot;, \u0026quot;hg38\u0026quot;))\nquery_data\n#\u0026gt; AnnotationHub with 17 records\n#\u0026gt; # snapshotDate(): 2023-10-20\n#\u0026gt; # $dataprovider: UCSC, ENCODE, GitHub, UCSChub\n#\u0026gt; # $species: Homo sapiens\n#\u0026gt; # $rdataclass: GRanges\n#\u0026gt; # additional mcols(): taxonomyid, genome, description,\n#\u0026gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,\n#\u0026gt; #   rdatapath, sourceurl, sourcetype \n#\u0026gt; # retrieve records with, e.g., 'object[[\u0026quot;AH107305\u0026quot;]]' \n#\u0026gt; \n#\u0026gt;              title                                 \n#\u0026gt;   AH107305 | hg38.Kundaje.GRCh38_unified_Excludable\n#\u0026gt;   AH107306 | hg38.Bernstein.Mint_Excludable_GRCh38 \n#\u0026gt;   AH107307 | hg38.Boyle.hg38-Excludable.v2         \n#\u0026gt;   AH107308 | hg38.Kundaje.GRCh38.Excludable        \n#\u0026gt;   AH107309 | hg38.Lareau.hg38.full.Excludable      \n#\u0026gt;   ...        ...                                   \n#\u0026gt;   AH107355 | hg38.UCSC.telomere                    \n#\u0026gt;   AH107356 | hg38.UCSC.short_arm                   \n#\u0026gt;   AH107357 | hg38.UCSC.heterochromatin             \n#\u0026gt;   AH107358 | hg38.UCSC.contig                      \n#\u0026gt;   AH107359 | hg38.UCSC.scaffold\nexcludeGR.hg38.Bernstein \u0026lt;- query_data[[\u0026quot;AH107306\u0026quot;]]\n#\u0026gt; loading from cache\nexcludeGR.hg38.Boyle     \u0026lt;- query_data[[\u0026quot;AH107307\u0026quot;]]\n#\u0026gt; loading from cache\nexcludeGR.hg38.Kundaje.2 \u0026lt;- query_data[[\u0026quot;AH107308\u0026quot;]]\n#\u0026gt; loading from cache\nexcludeGR.hg38.Lareau    \u0026lt;- query_data[[\u0026quot;AH107309\u0026quot;]]\n#\u0026gt; loading from cache\nexcludeGR.hg38.Reddy     \u0026lt;- query_data[[\u0026quot;AH107310\u0026quot;]]\n#\u0026gt; loading from cache\nexcludeGR.hg38.Wimberley \u0026lt;- query_data[[\u0026quot;AH107311\u0026quot;]]\n#\u0026gt; loading from cache\nexcludeGR.hg38.Wold      \u0026lt;- query_data[[\u0026quot;AH107312\u0026quot;]]\n#\u0026gt; loading from cache\nexcludeGR.hg38.Yeo       \u0026lt;- query_data[[\u0026quot;AH107313\u0026quot;]]\n#\u0026gt; loading from cache\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e query(\u003cspan class=\"pl-smi\"\u003eah\u003c/span\u003e, c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eexcluderanges\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehg38\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; AnnotationHub with 17 records\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # snapshotDate(): 2023-10-20\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # $dataprovider: UCSC, ENCODE, GitHub, UCSChub\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # $species: Homo sapiens\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # $rdataclass: GRanges\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # additional mcols(): taxonomyid, genome, description,\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; #   rdatapath, sourceurl, sourcetype \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # retrieve records with, e.g., 'object[[\"AH107305\"]]' \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;              title                                 \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107305 | hg38.Kundaje.GRCh38_unified_Excludable\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107306 | hg38.Bernstein.Mint_Excludable_GRCh38 \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107307 | hg38.Boyle.hg38-Excludable.v2         \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107308 | hg38.Kundaje.GRCh38.Excludable        \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107309 | hg38.Lareau.hg38.full.Excludable      \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   ...        ...                                   \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107355 | hg38.UCSC.telomere                    \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107356 | hg38.UCSC.short_arm                   \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107357 | hg38.UCSC.heterochromatin             \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107358 | hg38.UCSC.contig                      \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107359 | hg38.UCSC.scaffold\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Bernstein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH107306\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; loading from cache\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Boyle\u003c/span\u003e     \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH107307\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; loading from cache\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH107308\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; loading from cache\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Lareau\u003c/span\u003e    \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH107309\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; loading from cache\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Reddy\u003c/span\u003e     \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH107310\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; loading from cache\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wimberley\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH107311\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; loading from cache\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wold\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH107312\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; loading from cache\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Yeo\u003c/span\u003e       \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH107313\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; loading from cache\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCompare the number of excludable regions\u003c/h2\u003e\u003ca id=\"user-content-compare-the-number-of-excludable-regions\" class=\"anchor\" aria-label=\"Permalink: Compare the number of excludable regions\" href=\"#compare-the-number-of-excludable-regions\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(ggplot2)\nmtx_to_plot \u0026lt;- data.frame(Count = c(length(excludeGR.hg38.Bernstein), \n                                    length(excludeGR.hg38.Boyle),\n                                    length(excludeGR.hg38.Kundaje.1), \n                                    length(excludeGR.hg38.Kundaje.2), \n                                    length(excludeGR.hg38.Lareau),\n                                    length(excludeGR.hg38.Reddy), \n                                    length(excludeGR.hg38.Wimberley),\n                                    length(excludeGR.hg38.Wold), \n                                    length(excludeGR.hg38.Yeo)),\n                          Source = c(\u0026quot;Bernstein.Mint_Excludable_GRCh38\u0026quot;, \n                                     \u0026quot;Boyle.hg38-Excludable.v2\u0026quot;,\n                                     \u0026quot;Kundaje.GRCh38_unified_Excludable\u0026quot;, \n                                     \u0026quot;Kundaje.GRCh38.Excludable\u0026quot;, \n                                     \u0026quot;Lareau.hg38.full.Excludable\u0026quot;,\n                                     \u0026quot;Reddy.wgEncodeDacMapabilityConsensusExcludable\u0026quot;, \n                                     \u0026quot;Wimberley.peakPass60Perc_sorted\u0026quot;,\n                                     \u0026quot;Wold.hg38mitoExcludable\u0026quot;, \n                                     \u0026quot;Yeo.eCLIP_Excludableregions.hg38liftover.bed\u0026quot;))\n# Order Source by the number of regions\nmtx_to_plot$Source \u0026lt;- factor(mtx_to_plot$Source, levels = mtx_to_plot$Source[order(mtx_to_plot$Count)])\n\nggplot(mtx_to_plot, aes(x = Source, y = Count, fill = Source)) +\n  geom_bar(stat = \u0026quot;identity\u0026quot;) +\n  coord_flip() +\n  theme_bw() + theme(legend.position = \u0026quot;none\u0026quot;)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eggplot2\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eCount\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Bernstein\u003c/span\u003e), \n                                    length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Boyle\u003c/span\u003e),\n                                    length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.1\u003c/span\u003e), \n                                    length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.2\u003c/span\u003e), \n                                    length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Lareau\u003c/span\u003e),\n                                    length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Reddy\u003c/span\u003e), \n                                    length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wimberley\u003c/span\u003e),\n                                    length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wold\u003c/span\u003e), \n                                    length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Yeo\u003c/span\u003e)),\n                          \u003cspan class=\"pl-v\"\u003eSource\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBernstein.Mint_Excludable_GRCh38\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBoyle.hg38-Excludable.v2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKundaje.GRCh38_unified_Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKundaje.GRCh38.Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLareau.hg38.full.Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eReddy.wgEncodeDacMapabilityConsensusExcludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eWimberley.peakPass60Perc_sorted\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eWold.hg38mitoExcludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eYeo.eCLIP_Excludableregions.hg38liftover.bed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Order Source by the number of regions\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e[order(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCount\u003c/span\u003e)])\n\nggplot(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e, aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCount\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_bar(\u003cspan class=\"pl-v\"\u003estat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eidentity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  coord_flip() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme_bw() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e theme(\u003cspan class=\"pl-v\"\u003elegend.position\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/dozmorovlab/excluderanges/blob/master/man/figures/README-excluderanges_hg38_count-1.png\"\u003e\u003cimg src=\"/dozmorovlab/excluderanges/raw/master/man/figures/README-excluderanges_hg38_count-1.png\" width=\"100%\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# ggsave(\u0026quot;man/figures/excluderanges_hg38_count.png\u0026quot;, width = 5.5, height = 2)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ggsave(\"man/figures/excluderanges_hg38_count.png\", width = 5.5, height = 2)\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCompare the width of excludable regions\u003c/h2\u003e\u003ca id=\"user-content-compare-the-width-of-excludable-regions\" class=\"anchor\" aria-label=\"Permalink: Compare the width of excludable regions\" href=\"#compare-the-width-of-excludable-regions\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003elog2 scale because of heavy right tail distributions.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(ggridges)\nlibrary(dplyr)\nmtx_to_plot \u0026lt;- data.frame(Width = c(width(excludeGR.hg38.Bernstein), \n                                    width(excludeGR.hg38.Boyle),\n                                    width(excludeGR.hg38.Kundaje.1), \n                                    width(excludeGR.hg38.Kundaje.2), \n                                    width(excludeGR.hg38.Lareau),\n                                    width(excludeGR.hg38.Reddy), \n                                    width(excludeGR.hg38.Wimberley),\n                                    width(excludeGR.hg38.Wold), \n                                    width(excludeGR.hg38.Yeo)),\n                          Source = c(rep(\u0026quot;Bernstein.Mint_Excludable_GRCh38\u0026quot;, length(excludeGR.hg38.Bernstein)),\n                                     rep(\u0026quot;Boyle.hg38-Excludable.v2\u0026quot;, length(excludeGR.hg38.Boyle)),\n                                     rep(\u0026quot;Kundaje.GRCh38_unified_Excludable\u0026quot;, length(excludeGR.hg38.Kundaje.1)),\n                                     rep(\u0026quot;Kundaje.GRCh38.Excludable\u0026quot;, length(excludeGR.hg38.Kundaje.2)),\n                                     rep(\u0026quot;Lareau.hg38.full.Excludable\u0026quot;, length(excludeGR.hg38.Lareau)),\n                                     rep(\u0026quot;Reddy.wgEncodeDacMapabilityConsensusExcludable\u0026quot;, length(excludeGR.hg38.Reddy)),\n                                     rep(\u0026quot;Wimberley.peakPass60Perc_sorted\u0026quot;, length(excludeGR.hg38.Wimberley)),\n                                     rep(\u0026quot;Wold.hg38mitoExcludable\u0026quot;, length(excludeGR.hg38.Wold)),\n                                     rep(\u0026quot;Yeo.eCLIP_Excludableregions.hg38liftover.bed\u0026quot;, length(excludeGR.hg38.Yeo))))\n\n# Order objects by decreasing width\nmtx_to_plot$Source \u0026lt;- factor(mtx_to_plot$Source, levels = mtx_to_plot %\u0026gt;% \n                               group_by(Source) %\u0026gt;% summarise(Mean = mean(Width)) %\u0026gt;% \n                               arrange(Mean) %\u0026gt;% pull(Source))\nggplot(mtx_to_plot, aes(x = log2(Width), y = Source, fill = Source)) +\n  geom_density_ridges() +\n  theme_bw() + theme(legend.position = \u0026quot;none\u0026quot;)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eggridges\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eWidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Bernstein\u003c/span\u003e), \n                                    width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Boyle\u003c/span\u003e),\n                                    width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.1\u003c/span\u003e), \n                                    width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.2\u003c/span\u003e), \n                                    width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Lareau\u003c/span\u003e),\n                                    width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Reddy\u003c/span\u003e), \n                                    width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wimberley\u003c/span\u003e),\n                                    width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wold\u003c/span\u003e), \n                                    width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Yeo\u003c/span\u003e)),\n                          \u003cspan class=\"pl-v\"\u003eSource\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBernstein.Mint_Excludable_GRCh38\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Bernstein\u003c/span\u003e)),\n                                     rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBoyle.hg38-Excludable.v2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Boyle\u003c/span\u003e)),\n                                     rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKundaje.GRCh38_unified_Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.1\u003c/span\u003e)),\n                                     rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKundaje.GRCh38.Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.2\u003c/span\u003e)),\n                                     rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLareau.hg38.full.Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Lareau\u003c/span\u003e)),\n                                     rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eReddy.wgEncodeDacMapabilityConsensusExcludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Reddy\u003c/span\u003e)),\n                                     rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eWimberley.peakPass60Perc_sorted\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wimberley\u003c/span\u003e)),\n                                     rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eWold.hg38mitoExcludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wold\u003c/span\u003e)),\n                                     rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eYeo.eCLIP_Excludableregions.hg38liftover.bed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Yeo\u003c/span\u003e))))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Order objects by decreasing width\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n                               group_by(\u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% summarise(\u003cspan class=\"pl-v\"\u003eMean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e mean(\u003cspan class=\"pl-smi\"\u003eWidth\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n                               arrange(\u003cspan class=\"pl-smi\"\u003eMean\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% pull(\u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e))\nggplot(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e, aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e log2(\u003cspan class=\"pl-smi\"\u003eWidth\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_density_ridges() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme_bw() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e theme(\u003cspan class=\"pl-v\"\u003elegend.position\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/dozmorovlab/excluderanges/blob/master/man/figures/README-excluderanges_hg38_width-1.png\"\u003e\u003cimg src=\"/dozmorovlab/excluderanges/raw/master/man/figures/README-excluderanges_hg38_width-1.png\" width=\"100%\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# ggsave(\u0026quot;man/figures/excluderanges_hg38_width.png\u0026quot;, width = 5.5, height = 2)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ggsave(\"man/figures/excluderanges_hg38_width.png\", width = 5.5, height = 2)\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can investigate the total width of each set of excludable ranges.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"mtx_to_plot \u0026lt;- data.frame(TotalWidth = c(sum(width(excludeGR.hg38.Bernstein)), \n                                         sum(width(excludeGR.hg38.Boyle)),\n                                         sum(width(excludeGR.hg38.Kundaje.1)), \n                                         sum(width(excludeGR.hg38.Kundaje.2)), \n                                         sum(width(excludeGR.hg38.Lareau)),\n                                         sum(width(excludeGR.hg38.Reddy)), \n                                         sum(width(excludeGR.hg38.Wimberley)),\n                                         sum(width(excludeGR.hg38.Wold)), \n                                         sum(width(excludeGR.hg38.Yeo))), \n                          Source = c(\u0026quot;Bernstein.Mint_Excludable_GRCh38\u0026quot;, \n                                     \u0026quot;Boyle.hg38-Excludable.v2\u0026quot;,\n                                     \u0026quot;Kundaje.GRCh38_unified_Excludable\u0026quot;, \n                                     \u0026quot;Kundaje.GRCh38.Excludable\u0026quot;, \n                                     \u0026quot;Lareau.hg38.full.Excludable\u0026quot;,\n                                     \u0026quot;Reddy.wgEncodeDacMapabilityConsensusExcludable\u0026quot;, \n                                     \u0026quot;Wimberley.peakPass60Perc_sorted\u0026quot;,\n                                     \u0026quot;Wold.hg38mitoExcludable\u0026quot;, \n                                     \u0026quot;Yeo.eCLIP_Excludableregions.hg38liftover.bed\u0026quot;))\n# Order objects by decreasing width\nmtx_to_plot$Source \u0026lt;- factor(mtx_to_plot$Source, levels = mtx_to_plot %\u0026gt;% \n                               group_by(Source) %\u0026gt;% arrange(TotalWidth) %\u0026gt;% pull(Source))\n\nggplot(mtx_to_plot, aes(x = TotalWidth, y = Source, fill = Source)) + \n  geom_bar(stat=\u0026quot;identity\u0026quot;) + scale_x_log10() + scale_y_discrete(label=abbreviate, limits=rev) +\n  xlab(\u0026quot;log10 total width\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eTotalWidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(sum(width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Bernstein\u003c/span\u003e)), \n                                         sum(width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Boyle\u003c/span\u003e)),\n                                         sum(width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.1\u003c/span\u003e)), \n                                         sum(width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.2\u003c/span\u003e)), \n                                         sum(width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Lareau\u003c/span\u003e)),\n                                         sum(width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Reddy\u003c/span\u003e)), \n                                         sum(width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wimberley\u003c/span\u003e)),\n                                         sum(width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wold\u003c/span\u003e)), \n                                         sum(width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Yeo\u003c/span\u003e))), \n                          \u003cspan class=\"pl-v\"\u003eSource\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBernstein.Mint_Excludable_GRCh38\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBoyle.hg38-Excludable.v2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKundaje.GRCh38_unified_Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKundaje.GRCh38.Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLareau.hg38.full.Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eReddy.wgEncodeDacMapabilityConsensusExcludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eWimberley.peakPass60Perc_sorted\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eWold.hg38mitoExcludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eYeo.eCLIP_Excludableregions.hg38liftover.bed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Order objects by decreasing width\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n                               group_by(\u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% arrange(\u003cspan class=\"pl-smi\"\u003eTotalWidth\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% pull(\u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e))\n\nggplot(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e, aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTotalWidth\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSource\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  geom_bar(\u003cspan class=\"pl-v\"\u003estat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eidentity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e scale_x_log10() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e scale_y_discrete(\u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eabbreviate\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elimits\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003erev\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  xlab(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elog10 total width\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/dozmorovlab/excluderanges/blob/master/man/figures/README-excluderanges_hg38_sumwidth-1.png\"\u003e\u003cimg src=\"/dozmorovlab/excluderanges/raw/master/man/figures/README-excluderanges_hg38_sumwidth-1.png\" width=\"100%\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# ggsave(\u0026quot;man/figures/excluderanges_hg38_sumwidth.png\u0026quot;, width = 6.5, height = 2)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ggsave(\"man/figures/excluderanges_hg38_sumwidth.png\", width = 6.5, height = 2)\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCompare overlaps among sets\u003c/h2\u003e\u003ca id=\"user-content-compare-overlaps-among-sets\" class=\"anchor\" aria-label=\"Permalink: Compare overlaps among sets\" href=\"#compare-overlaps-among-sets\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can compare \u003ca href=\"https://en.wikipedia.org/wiki/Overlap_coefficient\" rel=\"nofollow\"\u003eoverlap\ncoefficients\u003c/a\u003e between\nthose sets of excludable regions.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(pheatmap)\nlibrary(stringr)\n# Overlap coefficient calculations\noverlap_coefficient \u0026lt;- function(gr_a, gr_b) {\n  intersects \u0026lt;- GenomicRanges::intersect(gr_a, gr_b, ignore.strand = TRUE)\n  intersection_width \u0026lt;- sum(width(intersects))\n  min_width \u0026lt;- min(sum(width(gr_a)), sum(width(gr_b)))\n  DataFrame(intersection_width, min_width, \n            overlap_coefficient = intersection_width/min_width,\n             n_intersections = length(intersects))\n}\n# List and names of all excludable regions\nall_excludeGR_list \u0026lt;- list(excludeGR.hg38.Bernstein, \n                            excludeGR.hg38.Boyle,\n                            excludeGR.hg38.Kundaje.1, \n                            excludeGR.hg38.Kundaje.2,\n                            excludeGR.hg38.Lareau,\n                            excludeGR.hg38.Reddy,\n                            excludeGR.hg38.Wimberley,\n                            excludeGR.hg38.Wold,\n                            excludeGR.hg38.Yeo)\nall_excludeGR_name \u0026lt;- c(\u0026quot;Bernstein.Mint_Excludable_GRCh38\u0026quot;, \n                         \u0026quot;Boyle.hg38-Excludable.v2\u0026quot;,\n                         \u0026quot;Kundaje.GRCh38_unified_Excludable\u0026quot;, \n                         \u0026quot;Kundaje.GRCh38.Excludable\u0026quot;, \n                         \u0026quot;Lareau.hg38.full.Excludable\u0026quot;,\n                         \u0026quot;Reddy.wgEncodeDacMapabilityConsensusExcludable\u0026quot;, \n                         \u0026quot;Wimberley.peakPass60Perc_sorted\u0026quot;,\n                         \u0026quot;Wold.hg38mitoExcludable\u0026quot;, \n                         \u0026quot;Yeo.eCLIP_Excludableregions.hg38liftover.bed\u0026quot;)\n# Correlation matrix, empty\nmtx_to_plot \u0026lt;- matrix(data = 0, nrow = length(all_excludeGR_list), ncol = length(all_excludeGR_list))\n# Fill it in\nfor (i in 1:length(all_excludeGR_list)) {\n  for (j in 1:length(all_excludeGR_list)) {\n    # If diagonal, set to zero\n    if (i == j) mtx_to_plot[i, j] \u0026lt;- 0\n    # Process only one half, the other is symmetric\n    if (i \u0026gt; j) {\n      mtx_to_plot[i, j] \u0026lt;- mtx_to_plot[j, i] \u0026lt;- overlap_coefficient(all_excludeGR_list[[i]], all_excludeGR_list[[j]])[[\u0026quot;overlap_coefficient\u0026quot;]]\n    }\n  }\n}\n# Trim row/colnames\nrownames(mtx_to_plot) \u0026lt;- colnames(mtx_to_plot) \u0026lt;- str_trunc(all_excludeGR_name, width = 25) \n# Save the plot\n# png(\u0026quot;man/figures/excluderanges_hg38_jaccard.png\u0026quot;, width = 1000, height = 900, res = 200)\npheatmap(data.matrix(mtx_to_plot), clustering_method = \u0026quot;ward.D\u0026quot;)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003epheatmap\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003estringr\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Overlap coefficient calculations\u003c/span\u003e\n\u003cspan class=\"pl-en\"\u003eoverlap_coefficient\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003egr_a\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egr_b\u003c/span\u003e) {\n  \u003cspan class=\"pl-smi\"\u003eintersects\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003eGenomicRanges\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eintersect(\u003cspan class=\"pl-smi\"\u003egr_a\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egr_b\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eignore.strand\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n  \u003cspan class=\"pl-smi\"\u003eintersection_width\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e sum(width(\u003cspan class=\"pl-smi\"\u003eintersects\u003c/span\u003e))\n  \u003cspan class=\"pl-smi\"\u003emin_width\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e min(sum(width(\u003cspan class=\"pl-smi\"\u003egr_a\u003c/span\u003e)), sum(width(\u003cspan class=\"pl-smi\"\u003egr_b\u003c/span\u003e)))\n  DataFrame(\u003cspan class=\"pl-smi\"\u003eintersection_width\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emin_width\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003eoverlap_coefficient\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eintersection_width\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emin_width\u003c/span\u003e,\n             \u003cspan class=\"pl-v\"\u003en_intersections\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e length(\u003cspan class=\"pl-smi\"\u003eintersects\u003c/span\u003e))\n}\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e List and names of all excludable regions\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eall_excludeGR_list\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Bernstein\u003c/span\u003e, \n                            \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Boyle\u003c/span\u003e,\n                            \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.1\u003c/span\u003e, \n                            \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.2\u003c/span\u003e,\n                            \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Lareau\u003c/span\u003e,\n                            \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Reddy\u003c/span\u003e,\n                            \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wimberley\u003c/span\u003e,\n                            \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wold\u003c/span\u003e,\n                            \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Yeo\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eall_excludeGR_name\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBernstein.Mint_Excludable_GRCh38\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBoyle.hg38-Excludable.v2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                         \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKundaje.GRCh38_unified_Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKundaje.GRCh38.Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLareau.hg38.full.Excludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                         \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eReddy.wgEncodeDacMapabilityConsensusExcludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eWimberley.peakPass60Perc_sorted\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                         \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eWold.hg38mitoExcludable\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eYeo.eCLIP_Excludableregions.hg38liftover.bed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Correlation matrix, empty\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ematrix\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003edata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enrow\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e length(\u003cspan class=\"pl-smi\"\u003eall_excludeGR_list\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003encol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e length(\u003cspan class=\"pl-smi\"\u003eall_excludeGR_list\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Fill it in\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eall_excludeGR_list\u003c/span\u003e)) {\n  \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ej\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eall_excludeGR_list\u003c/span\u003e)) {\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e If diagonal, set to zero\u003c/span\u003e\n    \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ej\u003c/span\u003e) \u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ej\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Process only one half, the other is symmetric\u003c/span\u003e\n    \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ej\u003c/span\u003e) {\n      \u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ej\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ej\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e overlap_coefficient(\u003cspan class=\"pl-smi\"\u003eall_excludeGR_list\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]], \u003cspan class=\"pl-smi\"\u003eall_excludeGR_list\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ej\u003c/span\u003e]])[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoverlap_coefficient\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n    }\n  }\n}\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Trim row/colnames\u003c/span\u003e\nrownames(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e str_trunc(\u003cspan class=\"pl-smi\"\u003eall_excludeGR_name\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e25\u003c/span\u003e) \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Save the plot\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e png(\"man/figures/excluderanges_hg38_jaccard.png\", width = 1000, height = 900, res = 200)\u003c/span\u003e\npheatmap(data.matrix(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eclustering_method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/dozmorovlab/excluderanges/blob/master/man/figures/README-excluderanges_hg38_overlap_coefficient-1.png\"\u003e\u003cimg src=\"/dozmorovlab/excluderanges/raw/master/man/figures/README-excluderanges_hg38_overlap_coefficient-1.png\" width=\"100%\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# dev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e dev.off()\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMetadata analysis\u003c/h2\u003e\u003ca id=\"user-content-metadata-analysis\" class=\"anchor\" aria-label=\"Permalink: Metadata analysis\" href=\"#metadata-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNote that some excludable ranges objects contain six columns, implying\nthere may be some interesting metadata. Let’s explore one.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"mcols(excludeGR.hg38.Reddy)\n#\u0026gt; DataFrame with 396 rows and 2 columns\n#\u0026gt;                       name     score\n#\u0026gt;                \u0026lt;character\u0026gt; \u0026lt;numeric\u0026gt;\n#\u0026gt; 1   High_Mappability_isl..      1000\n#\u0026gt; 2         Satellite_repeat      1000\n#\u0026gt; 3                 BSR/Beta      1000\n#\u0026gt; 4   Low_mappability_island      1000\n#\u0026gt; 5                 (CATTC)n      1000\n#\u0026gt; ...                    ...       ...\n#\u0026gt; 392       Satellite_repeat      1000\n#\u0026gt; 393               (CATTC)n      1000\n#\u0026gt; 394               (CATTC)n      1000\n#\u0026gt; 395                   TAR1      1000\n#\u0026gt; 396                   chrM      1000\nmtx_to_plot \u0026lt;- table(mcols(excludeGR.hg38.Reddy)[[\u0026quot;name\u0026quot;]]) %\u0026gt;%\n  as.data.frame()\ncolnames(mtx_to_plot) \u0026lt;- c(\u0026quot;Type\u0026quot;, \u0026quot;Number\u0026quot;)\nmtx_to_plot \u0026lt;- mtx_to_plot[order(mtx_to_plot$Number), ]\nmtx_to_plot$Type \u0026lt;- factor(mtx_to_plot$Type, \n                           levels = mtx_to_plot$Type)\nggplot(mtx_to_plot, aes(x = Number, y = Type, fill = Type)) +\n  geom_bar(stat=\u0026quot;identity\u0026quot;) +\n  theme_bw() + theme(legend.position = \u0026quot;none\u0026quot;)\"\u003e\u003cpre\u003emcols(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Reddy\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; DataFrame with 396 rows and 2 columns\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;                       name     score\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;                \u0026lt;character\u0026gt; \u0026lt;numeric\u0026gt;\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; 1   High_Mappability_isl..      1000\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; 2         Satellite_repeat      1000\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; 3                 BSR/Beta      1000\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; 4   Low_mappability_island      1000\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; 5                 (CATTC)n      1000\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; ...                    ...       ...\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; 392       Satellite_repeat      1000\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; 393               (CATTC)n      1000\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; 394               (CATTC)n      1000\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; 395                   TAR1      1000\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; 396                   chrM      1000\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e table(mcols(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Reddy\u003c/span\u003e)[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ename\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  as.data.frame()\ncolnames(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eType\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNumber\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e[order(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eNumber\u003c/span\u003e), ]\n\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eType\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eType\u003c/span\u003e, \n                           \u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eType\u003c/span\u003e)\nggplot(\u003cspan class=\"pl-smi\"\u003emtx_to_plot\u003c/span\u003e, aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eNumber\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eType\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eType\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_bar(\u003cspan class=\"pl-v\"\u003estat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eidentity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme_bw() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e theme(\u003cspan class=\"pl-v\"\u003elegend.position\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/dozmorovlab/excluderanges/blob/master/man/figures/README-excluderanges_hg38_Reddy_metadata-1.png\"\u003e\u003cimg src=\"/dozmorovlab/excluderanges/raw/master/man/figures/README-excluderanges_hg38_Reddy_metadata-1.png\" width=\"100%\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# ggsave(\u0026quot;man/figures/excluderanges_hg38_Reddy_metadata.png\u0026quot;, width = 5, height = 2.5)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ggsave(\"man/figures/excluderanges_hg38_Reddy_metadata.png\", width = 5, height = 2.5)\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOne may decide to combine the excludable ranges from all labs, although\nfrom previous results we may decide to follow Anshul’s\n\u003ca href=\"https://twitter.com/anshulkundaje/status/1263546023151992832?s=20\" rel=\"nofollow\"\u003eadvice\u003c/a\u003e\nabout the \u003ca href=\"https://www.encodeproject.org/files/ENCFF356LFX/\" rel=\"nofollow\"\u003eENCFF356LFX exclusion list\nregions\u003c/a\u003e and use the\n\u003ccode\u003eexcludeGR.hg38.Kundaje.1\u003c/code\u003e object.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"excludeGR.hg38.all \u0026lt;- reduce(c(excludeGR.hg38.Bernstein, \n                               excludeGR.hg38.Boyle,\n                               excludeGR.hg38.Kundaje.1, \n                               excludeGR.hg38.Kundaje.2, \n                               excludeGR.hg38.Lareau,\n                               excludeGR.hg38.Reddy, \n                               excludeGR.hg38.Wimberley,\n                               excludeGR.hg38.Wold, \n                               excludeGR.hg38.Yeo))\n# Sort and Keep only standard chromosomes\nexcludeGR.hg38.all \u0026lt;- excludeGR.hg38.all %\u0026gt;% sort %\u0026gt;% \n  keepStandardChromosomes(pruning.mode = \u0026quot;tidy\u0026quot;)\nprint(length(excludeGR.hg38.all))\n#\u0026gt; [1] 15998\nsummary(width(excludeGR.hg38.all))\n#\u0026gt;     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. \n#\u0026gt;        5     1703     2294    17713     3001 30739401\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.all\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e reduce(c(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Bernstein\u003c/span\u003e, \n                               \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Boyle\u003c/span\u003e,\n                               \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.1\u003c/span\u003e, \n                               \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Kundaje.2\u003c/span\u003e, \n                               \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Lareau\u003c/span\u003e,\n                               \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Reddy\u003c/span\u003e, \n                               \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wimberley\u003c/span\u003e,\n                               \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Wold\u003c/span\u003e, \n                               \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.Yeo\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Sort and Keep only standard chromosomes\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.all\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.all\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-smi\"\u003esort\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  keepStandardChromosomes(\u003cspan class=\"pl-v\"\u003epruning.mode\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etidy\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nprint(length(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.all\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] 15998\u003c/span\u003e\nsummary(width(\u003cspan class=\"pl-smi\"\u003eexcludeGR.hg38.all\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;        5     1703     2294    17713     3001 30739401\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eBEDbase data download\u003c/h1\u003e\u003ca id=\"user-content-bedbase-data-download\" class=\"anchor\" aria-label=\"Permalink: BEDbase data download\" href=\"#bedbase-data-download\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"http://bedbase.org/\" rel=\"nofollow\"\u003eBEDbase.org\u003c/a\u003e is a repository for storing and\nanalyzing BED files and BED sets, developed by \u003ca href=\"https://databio.org/\" rel=\"nofollow\"\u003eSheffield\nlab\u003c/a\u003e. It provides API for data access and\nretrieval.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUsing BEDbase ID (e.g., \u003ccode\u003e1a561729234c2844303a051b16f66656\u003c/code\u003e for the\n\u003ccode\u003ehg38.Kundaje.GRCh38_unified_Excludable\u003c/code\u003e excluderanges object), we can\nconstruct a URL showing the splash screen with detailed information of\nthe corresponding object,\n\u003ca href=\"http://bedbase.org/#/bedsplash/1a561729234c2844303a051b16f66656\" rel=\"nofollow\"\u003ehttp://bedbase.org/#/bedsplash/1a561729234c2844303a051b16f66656\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe can also get the BED file using the \u003ca href=\"http://bedbase.org/docs\" rel=\"nofollow\"\u003eBEDbase\nAPI\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"wget http://bedbase.org/api/bed/1a561729234c2844303a051b16f66656/file/bed \\\n  -O hg38.Kundaje.GRCh38_unified_Excludable.bed.gz \"\u003e\u003cpre\u003ewget http://bedbase.org/api/bed/1a561729234c2844303a051b16f66656/file/bed \\\n  -O hg38.Kundaje.GRCh38_unified_Excludable.bed.gz \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSee also the R code to download\n\u003ca href=\"https://gist.github.com/mdozmorov/7f4393f90932fb6bd911c43c20425ca0\"\u003eT2T.excluderanges\u003c/a\u003e\nand\n\u003ca href=\"https://gist.github.com/mdozmorov/33a1fa3234b2942dae238e1fcb39c996\"\u003emm39.excluderanges\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMitochondrial DNA sequences, NUMTs\u003c/h1\u003e\u003ca id=\"user-content-mitochondrial-dna-sequences-numts\" class=\"anchor\" aria-label=\"Permalink: Mitochondrial DNA sequences, NUMTs\" href=\"#mitochondrial-dna-sequences-numts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMitochondrial DNA sequences (mtDNA, 100-600K mitochondria per human\ncell) transferred to the nucleus give rise to the so-called\nmitochondrial DNA sequences in the nuclear genome (NUMTs). In the\nsettings of DNA/chromatin sequencing (e.g., ATAC-seq), we may observe up\nto 80% of mitochondrial sequencing reads that may pile up in the NUMT\nsequences. Similar to exclusion sets, genomic regions highly homologous\nto mtDNA can be masked to improve biological signal.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe reference human nuclear mitochondrial sequences have been available\nin the UCSC genome browser for hg19 and mm8 human/mouse genome\nassemblies. We collected NUMT sets for hg38, T2T-CHM13, mm10, generated\nby Caleb Lareau in the\n\u003ca href=\"https://github.com/caleblareau/mitoblacklist\"\u003emitoblacklist\u003c/a\u003e GitHub\nrepository. These NUMT sets can be combined with exclusion sets.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eExample of the \u003ccode\u003ehg38.Lareau.hg38_peaks\u003c/code\u003e object\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"suppressMessages(library(httr))\n# Get hg38.Lareau.hg38_peaks BEDbase ID\nbedbase_id \u0026lt;- \u0026quot;9fa55701a3bd3e7a598d1d2815e3390f\u0026quot;\n# Construct output file name\nfileNameOut \u0026lt;- \u0026quot;hg38.Lareau.hg38_peak.bed.gz\u0026quot;\n# API token for BED data\ntoken2 \u0026lt;- paste0(\u0026quot;http://bedbase.org/api/bed/\u0026quot;, bedbase_id, \u0026quot;/file/bed\u0026quot;)\n# Download file\nGET(url = token2, write_disk(fileNameOut, overwrite = TRUE)) # , verbose()\n#\u0026gt; Response [http://data.bedbase.org/bed_files/hg38.Lareau.hg38_peaks.bed.gz]\n#\u0026gt;   Date: 2023-12-04 18:53\n#\u0026gt;   Status: 200\n#\u0026gt;   Content-Type: application/vnd.realvnc.bed\n#\u0026gt;   Size: 11.8 kB\n#\u0026gt; \u0026lt;ON DISK\u0026gt;  /Users/mdozmorov/Documents/Work/GitHub/excluderanges/hg38.Lareau.hg38_peak.bed.gz\n# Read the data in\nhg38.Lareau.hg38_peaks \u0026lt;- read.table(fileNameOut, sep = \u0026quot;\\t\u0026quot;, header = FALSE)\n# Assign column names depending on the number of columns\nall_columns \u0026lt;- c(\u0026quot;chr\u0026quot;, \u0026quot;start\u0026quot;, \u0026quot;stop\u0026quot;, \u0026quot;name\u0026quot;, \u0026quot;score\u0026quot;, \u0026quot;strand\u0026quot;, \n                 \u0026quot;signalValue\u0026quot;, \u0026quot;pValue\u0026quot;, \u0026quot;qValue\u0026quot;, \u0026quot;peak\u0026quot;)\ncolnames(hg38.Lareau.hg38_peaks) \u0026lt;- all_columns[1:ncol(hg38.Lareau.hg38_peaks)]\n# Convert to GRanges object\nhg38.Lareau.hg38_peaks \u0026lt;- makeGRangesFromDataFrame(hg38.Lareau.hg38_peaks, \n                                                   keep.extra.columns = TRUE)\nhg38.Lareau.hg38_peaks\n#\u0026gt; GRanges object with 784 ranges and 2 metadata columns:\n#\u0026gt;         seqnames              ranges strand |                name     score\n#\u0026gt;            \u0026lt;Rle\u0026gt;           \u0026lt;IRanges\u0026gt;  \u0026lt;Rle\u0026gt; |         \u0026lt;character\u0026gt; \u0026lt;integer\u0026gt;\n#\u0026gt;     [1]     chr1       628903-635104      * |   peaks/hg38_peak_1   2523899\n#\u0026gt;     [2]     chr1     5850087-5850571      * |   peaks/hg38_peak_2     32940\n#\u0026gt;     [3]     chr1     8909610-8910014      * |   peaks/hg38_peak_3       123\n#\u0026gt;     [4]     chr1     9574580-9574997      * |   peaks/hg38_peak_4     28828\n#\u0026gt;     [5]     chr1   32043823-32044203      * |   peaks/hg38_peak_5      8083\n#\u0026gt;     ...      ...                 ...    ... .                 ...       ...\n#\u0026gt;   [780]     chrX 143431716-143432219      * | peaks/hg38_peak_780     41434\n#\u0026gt;   [781]     chrX 143432410-143433212      * | peaks/hg38_peak_781      9271\n#\u0026gt;   [782]     chrX 143433510-143434156      * | peaks/hg38_peak_782      5741\n#\u0026gt;   [783]     chrX 143543636-143544023      * | peaks/hg38_peak_783      3320\n#\u0026gt;   [784]     chrX 146995842-146996224      * | peaks/hg38_peak_784     27222\n#\u0026gt;   -------\n#\u0026gt;   seqinfo: 23 sequences from an unspecified genome; no seqlengths\"\u003e\u003cpre\u003esuppressMessages(library(\u003cspan class=\"pl-smi\"\u003ehttr\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Get hg38.Lareau.hg38_peaks BEDbase ID\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ebedbase_id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e9fa55701a3bd3e7a598d1d2815e3390f\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Construct output file name\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003efileNameOut\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehg38.Lareau.hg38_peak.bed.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e API token for BED data\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etoken2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehttp://bedbase.org/api/bed/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ebedbase_id\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/file/bed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Download file\u003c/span\u003e\nGET(\u003cspan class=\"pl-v\"\u003eurl\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etoken2\u003c/span\u003e, write_disk(\u003cspan class=\"pl-smi\"\u003efileNameOut\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eoverwrite\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e , verbose()\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; Response [http://data.bedbase.org/bed_files/hg38.Lareau.hg38_peaks.bed.gz]\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   Date: 2023-12-04 18:53\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   Status: 200\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   Content-Type: application/vnd.realvnc.bed\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   Size: 11.8 kB\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; \u0026lt;ON DISK\u0026gt;  /Users/mdozmorov/Documents/Work/GitHub/excluderanges/hg38.Lareau.hg38_peak.bed.gz\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Read the data in\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ehg38.Lareau.hg38_peaks\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.table(\u003cspan class=\"pl-smi\"\u003efileNameOut\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Assign column names depending on the number of columns\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eall_columns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003echr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003estart\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003estop\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ename\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003escore\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003estrand\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                 \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esignalValue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epValue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eqValue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epeak\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ncolnames(\u003cspan class=\"pl-smi\"\u003ehg38.Lareau.hg38_peaks\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eall_columns\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003encol(\u003cspan class=\"pl-smi\"\u003ehg38.Lareau.hg38_peaks\u003c/span\u003e)]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Convert to GRanges object\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ehg38.Lareau.hg38_peaks\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e makeGRangesFromDataFrame(\u003cspan class=\"pl-smi\"\u003ehg38.Lareau.hg38_peaks\u003c/span\u003e, \n                                                   \u003cspan class=\"pl-v\"\u003ekeep.extra.columns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ehg38.Lareau.hg38_peaks\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; GRanges object with 784 ranges and 2 metadata columns:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;         seqnames              ranges strand |                name     score\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;            \u0026lt;Rle\u0026gt;           \u0026lt;IRanges\u0026gt;  \u0026lt;Rle\u0026gt; |         \u0026lt;character\u0026gt; \u0026lt;integer\u0026gt;\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [1]     chr1       628903-635104      * |   peaks/hg38_peak_1   2523899\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [2]     chr1     5850087-5850571      * |   peaks/hg38_peak_2     32940\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [3]     chr1     8909610-8910014      * |   peaks/hg38_peak_3       123\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [4]     chr1     9574580-9574997      * |   peaks/hg38_peak_4     28828\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [5]     chr1   32043823-32044203      * |   peaks/hg38_peak_5      8083\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     ...      ...                 ...    ... .                 ...       ...\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [780]     chrX 143431716-143432219      * | peaks/hg38_peak_780     41434\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [781]     chrX 143432410-143433212      * | peaks/hg38_peak_781      9271\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [782]     chrX 143433510-143434156      * | peaks/hg38_peak_782      5741\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [783]     chrX 143543636-143544023      * | peaks/hg38_peak_783      3320\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [784]     chrX 146995842-146996224      * | peaks/hg38_peak_784     27222\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   -------\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   seqinfo: 23 sequences from an unspecified genome; no seqlengths\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCentromeres, telomeres, etc.\u003c/h1\u003e\u003ca id=\"user-content-centromeres-telomeres-etc\" class=\"anchor\" aria-label=\"Permalink: Centromeres, telomeres, etc.\" href=\"#centromeres-telomeres-etc\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBesides the ENCODE-produced excludable regions, we may want to exclude\ncentromeres, telomeres, and other gap locations. The “Gap Locations”\ntrack for Homo Sapiens is available for the GRcH37/hg19 genome assembly\nas a \u003ca href=\"http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19\u0026amp;hgta_group=map\u0026amp;hgta_track=gap\u0026amp;hgta_table=gap\u0026amp;hgta_doSchema=describe+table+schema\" rel=\"nofollow\"\u003eUCSC ‘gap’\ntable\u003c/a\u003e.\nIt can be retrieved from\n\u003cem\u003e\u003ca href=\"https://bioconductor.org/packages/3.18/AnnotationHub\" rel=\"nofollow\"\u003eAnnotationHub\u003c/a\u003e\u003c/em\u003e,\nbut lacks the metadata columns needed to decide the type of gaps.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Search for the gap track\n# ahData \u0026lt;- query(ah, c(\u0026quot;gap\u0026quot;, \u0026quot;Homo sapiens\u0026quot;, \u0026quot;hg19\u0026quot;))\n# ahData[ahData$title == \u0026quot;Gap\u0026quot;]\ngaps \u0026lt;- ahData[[\u0026quot;AH6444\u0026quot;]]\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Search for the gap track\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ahData \u0026lt;- query(ah, c(\"gap\", \"Homo sapiens\", \"hg19\"))\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ahData[ahData$title == \"Gap\"]\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003egaps\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eahData\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH6444\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe \u003ca href=\"http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19\u0026amp;hgta_group=map\u0026amp;hgta_track=gap\u0026amp;hgta_table=gap\u0026amp;hgta_doSchema=describe+table+schema\" rel=\"nofollow\"\u003eUCSC ‘gap’\ntable\u003c/a\u003e\nprovides better granularity about the types of gaps available. E.g., for\nhuman, hg19, we have the following types and the number of gaps.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/dozmorovlab/excluderanges/blob/master/man/figures/README-gapcounts-1.png\"\u003e\u003cimg src=\"/dozmorovlab/excluderanges/raw/master/man/figures/README-gapcounts-1.png\" width=\"70%\" height=\"70%\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThose objects are provided as individual GRanges.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNaming convention: \u003ccode\u003e\u0026lt;genome assembly\u0026gt;.UCSC.\u0026lt;gap type\u0026gt;\u003c/code\u003e, e.g.,\n\u003ccode\u003ehg38.UCSC.gap_centromere\u003c/code\u003e. We can similarly load any gap type object.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"query_data \u0026lt;- query(ah, c(\u0026quot;excluderanges\u0026quot;, \u0026quot;UCSC\u0026quot;, \u0026quot;Homo Sapiens\u0026quot;, \u0026quot;hg38\u0026quot;))\nquery_data\n#\u0026gt; AnnotationHub with 7 records\n#\u0026gt; # snapshotDate(): 2023-10-20\n#\u0026gt; # $dataprovider: UCSC, UCSChub\n#\u0026gt; # $species: Homo sapiens\n#\u0026gt; # $rdataclass: GRanges\n#\u0026gt; # additional mcols(): taxonomyid, genome, description,\n#\u0026gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,\n#\u0026gt; #   rdatapath, sourceurl, sourcetype \n#\u0026gt; # retrieve records with, e.g., 'object[[\u0026quot;AH107353\u0026quot;]]' \n#\u0026gt; \n#\u0026gt;              title                    \n#\u0026gt;   AH107353 | T2T.UCSC.hgUnique.hg38   \n#\u0026gt;   AH107354 | hg38.UCSC.centromere     \n#\u0026gt;   AH107355 | hg38.UCSC.telomere       \n#\u0026gt;   AH107356 | hg38.UCSC.short_arm      \n#\u0026gt;   AH107357 | hg38.UCSC.heterochromatin\n#\u0026gt;   AH107358 | hg38.UCSC.contig         \n#\u0026gt;   AH107359 | hg38.UCSC.scaffold\n\ngapsGR_hg38_centromere \u0026lt;- query_data[[\u0026quot;AH107354\u0026quot;]]\n#\u0026gt; loading from cache\ngapsGR_hg38_centromere\n#\u0026gt; GRanges object with 109 ranges and 2 metadata columns:\n#\u0026gt;         seqnames              ranges strand |       bin        name\n#\u0026gt;            \u0026lt;Rle\u0026gt;           \u0026lt;IRanges\u0026gt;  \u0026lt;Rle\u0026gt; | \u0026lt;numeric\u0026gt; \u0026lt;character\u0026gt;\n#\u0026gt;     [1]     chr1 122026459-122224535      * |       189  GJ211836.1\n#\u0026gt;     [2]     chr1 122224635-122503147      * |       189  GJ211837.1\n#\u0026gt;     [3]     chr1 122503247-124785432      * |        23  GJ212202.1\n#\u0026gt;     [4]     chr1 124785532-124849129      * |      1537  GJ211855.1\n#\u0026gt;     [5]     chr1 124849229-124932724      * |       192  GJ211857.1\n#\u0026gt;     ...      ...                 ...    ... .       ...         ...\n#\u0026gt;   [105]    chr22   14419994-14420334      * |       695  GJ212209.1\n#\u0026gt;   [106]    chr22   14420434-14421632      * |       695  GJ212186.2\n#\u0026gt;   [107]    chr22   14421732-15054318      * |        10  GJ212191.2\n#\u0026gt;   [108]     chrX   58605579-62412542      * |         1  GJ212192.1\n#\u0026gt;   [109]     chrY   10316944-10544039      * |        10  GJ212193.1\n#\u0026gt;   -------\n#\u0026gt;   seqinfo: 24 sequences from hg38 genome\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e query(\u003cspan class=\"pl-smi\"\u003eah\u003c/span\u003e, c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eexcluderanges\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUCSC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHomo Sapiens\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehg38\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; AnnotationHub with 7 records\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # snapshotDate(): 2023-10-20\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # $dataprovider: UCSC, UCSChub\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # $species: Homo sapiens\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # $rdataclass: GRanges\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # additional mcols(): taxonomyid, genome, description,\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; #   rdatapath, sourceurl, sourcetype \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; # retrieve records with, e.g., 'object[[\"AH107353\"]]' \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;              title                    \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107353 | T2T.UCSC.hgUnique.hg38   \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107354 | hg38.UCSC.centromere     \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107355 | hg38.UCSC.telomere       \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107356 | hg38.UCSC.short_arm      \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107357 | hg38.UCSC.heterochromatin\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107358 | hg38.UCSC.contig         \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   AH107359 | hg38.UCSC.scaffold\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003egapsGR_hg38_centromere\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH107354\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; loading from cache\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003egapsGR_hg38_centromere\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; GRanges object with 109 ranges and 2 metadata columns:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;         seqnames              ranges strand |       bin        name\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;            \u0026lt;Rle\u0026gt;           \u0026lt;IRanges\u0026gt;  \u0026lt;Rle\u0026gt; | \u0026lt;numeric\u0026gt; \u0026lt;character\u0026gt;\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [1]     chr1 122026459-122224535      * |       189  GJ211836.1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [2]     chr1 122224635-122503147      * |       189  GJ211837.1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [3]     chr1 122503247-124785432      * |        23  GJ212202.1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [4]     chr1 124785532-124849129      * |      1537  GJ211855.1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     [5]     chr1 124849229-124932724      * |       192  GJ211857.1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     ...      ...                 ...    ... .       ...         ...\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [105]    chr22   14419994-14420334      * |       695  GJ212209.1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [106]    chr22   14420434-14421632      * |       695  GJ212186.2\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [107]    chr22   14421732-15054318      * |        10  GJ212191.2\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [108]     chrX   58605579-62412542      * |         1  GJ212192.1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   [109]     chrY   10316944-10544039      * |        10  GJ212193.1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   -------\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   seqinfo: 24 sequences from hg38 genome\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGaps in mouse genome\u003c/h2\u003e\u003ca id=\"user-content-gaps-in-mouse-genome\" class=\"anchor\" aria-label=\"Permalink: Gaps in mouse genome\" href=\"#gaps-in-mouse-genome\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNote that Mouse chromosomes have centromeres close to one end and have\nessentially no short arm. Although available, we do not recommend using\nthem as they are essentially placeholders.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"query_data \u0026lt;- query(ah, c(\u0026quot;excluderanges\u0026quot;, \u0026quot;Mus musculus\u0026quot;, \u0026quot;UCSC\u0026quot;))\nmm39.UCSC.centromere \u0026lt;- query_data[[\u0026quot;AH107366\u0026quot;]]\n#\u0026gt; loading from cache\nhead(mm39.UCSC.centromere)\n#\u0026gt; GRanges object with 6 ranges and 6 metadata columns:\n#\u0026gt;       seqnames         ranges strand |       bin        ix           n\n#\u0026gt;          \u0026lt;Rle\u0026gt;      \u0026lt;IRanges\u0026gt;  \u0026lt;Rle\u0026gt; | \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;character\u0026gt;\n#\u0026gt;     1     chr1 110000-3000000      * |         9         3           N\n#\u0026gt;   314     chr2 110000-3000000      * |         9         3           N\n#\u0026gt;   254     chr3 110000-3000000      * |         9         3           N\n#\u0026gt;   219     chr4 110000-3000000      * |         9         3           N\n#\u0026gt;   211     chr5 110000-3000000      * |         9         3           N\n#\u0026gt;   204     chr6 110000-3000000      * |         9         3           N\n#\u0026gt;            size        type      bridge\n#\u0026gt;       \u0026lt;numeric\u0026gt; \u0026lt;character\u0026gt; \u0026lt;character\u0026gt;\n#\u0026gt;     1   2890000  centromere          no\n#\u0026gt;   314   2890000  centromere          no\n#\u0026gt;   254   2890000  centromere          no\n#\u0026gt;   219   2890000  centromere          no\n#\u0026gt;   211   2890000  centromere          no\n#\u0026gt;   204   2890000  centromere          no\n#\u0026gt;   -------\n#\u0026gt;   seqinfo: 20 sequences from mm39 genome\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e query(\u003cspan class=\"pl-smi\"\u003eah\u003c/span\u003e, c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eexcluderanges\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMus musculus\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUCSC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003emm39.UCSC.centromere\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003equery_data\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAH107366\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; loading from cache\u003c/span\u003e\nhead(\u003cspan class=\"pl-smi\"\u003emm39.UCSC.centromere\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; GRanges object with 6 ranges and 6 metadata columns:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;       seqnames         ranges strand |       bin        ix           n\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;          \u0026lt;Rle\u0026gt;      \u0026lt;IRanges\u0026gt;  \u0026lt;Rle\u0026gt; | \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;character\u0026gt;\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     1     chr1 110000-3000000      * |         9         3           N\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   314     chr2 110000-3000000      * |         9         3           N\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   254     chr3 110000-3000000      * |         9         3           N\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   219     chr4 110000-3000000      * |         9         3           N\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   211     chr5 110000-3000000      * |         9         3           N\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   204     chr6 110000-3000000      * |         9         3           N\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;            size        type      bridge\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;       \u0026lt;numeric\u0026gt; \u0026lt;character\u0026gt; \u0026lt;character\u0026gt;\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;     1   2890000  centromere          no\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   314   2890000  centromere          no\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   254   2890000  centromere          no\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   219   2890000  centromere          no\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   211   2890000  centromere          no\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   204   2890000  centromere          no\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   -------\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;   seqinfo: 20 sequences from mm39 genome\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCUT\u0026amp;RUN excludable sets\u003c/h1\u003e\u003ca id=\"user-content-cutrun-excludable-sets\" class=\"anchor\" aria-label=\"Permalink: CUT\u0026amp;RUN excludable sets\" href=\"#cutrun-excludable-sets\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNordin et al. 2022 (Nordin et al. 2022) generated excludable regions for\nthe CUT\u0026amp;RUN technology. They are available as \u003ca href=\"https://www.biorxiv.org/content/10.1101/2022.11.11.516118v1.supplementary-material\" rel=\"nofollow\"\u003eSupplementary\nMaterial\u003c/a\u003e.\nWe uniformly processed them and made available on Google Drive\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# hg38 CUT\u0026amp;RUN exclusion set, BED\ndownload.file(\u0026quot;https://drive.google.com/uc?export=download\u0026amp;id=1rKIu7kdiEySTi-cq3nYxXJP4VQX1IPcS\u0026quot;,\n              destfile = \u0026quot;hg38.Nordin.CandRblacklist_hg38.bed\u0026quot;)\n# hg38 CUT\u0026amp;RUN exclusion set, RDS\ndownload.file(\u0026quot;https://drive.google.com/uc?export=download\u0026amp;id=1JuB1h-QQUw1mddBavI7CIuH7R-lwwczU\u0026quot;,\n              destfile = \u0026quot;hg38.Nordin.CandRblacklist_hg38.rds\u0026quot;)\n# And then load the GRanges object\nmtx \u0026lt;- readRDS(\u0026quot;hg38.Nordin.CandRblacklist_hg38.rds\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e hg38 CUT\u0026amp;RUN exclusion set, BED\u003c/span\u003e\ndownload.file(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehttps://drive.google.com/uc?export=download\u0026amp;id=1rKIu7kdiEySTi-cq3nYxXJP4VQX1IPcS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n              \u003cspan class=\"pl-v\"\u003edestfile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehg38.Nordin.CandRblacklist_hg38.bed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e hg38 CUT\u0026amp;RUN exclusion set, RDS\u003c/span\u003e\ndownload.file(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehttps://drive.google.com/uc?export=download\u0026amp;id=1JuB1h-QQUw1mddBavI7CIuH7R-lwwczU\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n              \u003cspan class=\"pl-v\"\u003edestfile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehg38.Nordin.CandRblacklist_hg38.rds\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e And then load the GRanges object\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emtx\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e readRDS(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehg38.Nordin.CandRblacklist_hg38.rds\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# mm10 CUT\u0026amp;RUN exclusion set, BED\ndownload.file(\u0026quot;https://drive.google.com/uc?export=download\u0026amp;id=1CRAojdphMbAzd3MnW_UmO1WtsDrHsrU1\u0026quot;,\n              destfile = \u0026quot;mm10.Nordin.CandRblacklist_mm10.bed\u0026quot;)\n# mm10 CUT\u0026amp;RUN exclusion set, RDS\ndownload.file(\u0026quot;https://drive.google.com/uc?export=download\u0026amp;id=1orPXLWUZ4-C4n_Jt2gH-WERLpY9Kn0t_\u0026quot;,\n              destfile = \u0026quot;mm10.Nordin.CandRblacklist_mm10.rds\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e mm10 CUT\u0026amp;RUN exclusion set, BED\u003c/span\u003e\ndownload.file(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehttps://drive.google.com/uc?export=download\u0026amp;id=1CRAojdphMbAzd3MnW_UmO1WtsDrHsrU1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n              \u003cspan class=\"pl-v\"\u003edestfile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emm10.Nordin.CandRblacklist_mm10.bed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e mm10 CUT\u0026amp;RUN exclusion set, RDS\u003c/span\u003e\ndownload.file(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehttps://drive.google.com/uc?export=download\u0026amp;id=1orPXLWUZ4-C4n_Jt2gH-WERLpY9Kn0t_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n              \u003cspan class=\"pl-v\"\u003edestfile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emm10.Nordin.CandRblacklist_mm10.rds\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSummary table\u003c/h1\u003e\u003ca id=\"user-content-summary-table\" class=\"anchor\" aria-label=\"Permalink: Summary table\" href=\"#summary-table\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/dozmorovlab/excluderanges/blob/master/man/figures/Table_S1.csv\"\u003eFull summary table\u003c/a\u003e.\u003c/p\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth align=\"left\"\u003eName\u003c/th\u003e\n\u003cth align=\"left\"\u003eAhub.IDs.BioC.3.16.and.above\u003c/th\u003e\n\u003cth align=\"left\"\u003eBEDbase.URL\u003c/th\u003e\n\u003cth align=\"left\"\u003eDescription\u003c/th\u003e\n\u003cth align=\"right\"\u003eFiltered.Region.count\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003eT2T.excluderanges\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107304\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/8329d8c624880308ab51ba05149a737d\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions\u003c/td\u003e\n\u003ctd align=\"right\"\u003e3565\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.Kundaje.GRCh38_unified_Excludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107305\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/1a561729234c2844303a051b16f66656\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined as a combination of hg38.Lareau.hg38_peaks, hg38.Boyle.hg38-Excludable.v2, and hg38.Wimberley.peakPass60Perc_sorted, followed by manual curation. Supersedes hg38.Kundaje.GRCh38.Excludable.\u003c/td\u003e\n\u003ctd align=\"right\"\u003e910\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.Bernstein.Mint_Excludable_GRCh38\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107306\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/80e335903b77b597b8245f9817fcd9cd\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined from Mint-ChIP (low input, multiplexed ChIP-seq) data\u003c/td\u003e\n\u003ctd align=\"right\"\u003e12052\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.Boyle.hg38-Excludable.v2\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107307\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/ac58962c9ec98fe9258c12092a0c8832\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions\u003c/td\u003e\n\u003ctd align=\"right\"\u003e636\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.Kundaje.GRCh38.Excludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107308\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/cb701496bde7eeb18add96fdbc3b8b11\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by Anshul Kundaje as a part of ENCODE and modENCODE consortia\u003c/td\u003e\n\u003ctd align=\"right\"\u003e38\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.Lareau.hg38.full.Excludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107309\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/5a12c1de138ace1a73a45e6faf9ba669\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)\u003c/td\u003e\n\u003ctd align=\"right\"\u003e820\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107310\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/148622e896f6798f7c4abf448bab67c4\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), centromeric repeats, chrM, High/Low mappability islands. Has extra chromosomes, use keepStandardChromosomes() filtering\u003c/td\u003e\n\u003ctd align=\"right\"\u003e396\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.Wimberley.peakPass60Perc_sorted\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107311\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/f4a9bb19ed29e993592813e970e7dd90\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the ewimberley/peakPass software\u003c/td\u003e\n\u003ctd align=\"right\"\u003e5078\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.Wold.hg38mitoExcludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107312\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/a714dcba99821801b5c426fba9c80988\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefinition method unknown\u003c/td\u003e\n\u003ctd align=\"right\"\u003e299\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107313\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/1a02a65fafefefd65ff4a060273304ed\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined from eCLIP data\u003c/td\u003e\n\u003ctd align=\"right\"\u003e56\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.Nordin.CandRblacklist_hg38\u003c/td\u003e\n\u003ctd align=\"left\"\u003eNA\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/b410448ba193fc3f0e99ca367e10b3b6\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined from CUT\u0026amp;RUN negative controls as 0.1% top significant SEACR peaks in over 30% of samples\u003c/td\u003e\n\u003ctd align=\"right\"\u003e885\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.Boyle.hg19-Excludable.v2\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107314\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/6eb180d456f2f3b71b419e5fab107fc9\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions\u003c/td\u003e\n\u003ctd align=\"right\"\u003e834\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.Bernstein.Mint_Excludable_hg19\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107315\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/d1a6047ed5bec84acefe9c52cf63b593\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined from Mint-ChIP (low input, multiplexed ChIP-seq) data\u003c/td\u003e\n\u003ctd align=\"right\"\u003e9035\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.Birney.wgEncodeDacMapabilityConsensusExcludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107316\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/5b6b19dea85a8bc6007ef07a0960267b\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), centromeric repeats, chrM, High/Low mappability islands\u003c/td\u003e\n\u003ctd align=\"right\"\u003e411\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107317\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/dac2eda4e8687eb039611ac6cd595821\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), human satellite repeat HSATII, chrM, ribosomal subunit consensus sequences LSU-rRNA_Hsa, SSU-rRNA_Hsa. Has extra chromosomes, use keepStandardChromosomes() filtering\u003c/td\u003e\n\u003ctd align=\"right\"\u003e1566\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.Lareau.hg19.full.Excludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107318\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/d934d47e8035da9c5a1767c8153db4cc\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)\u003c/td\u003e\n\u003ctd align=\"right\"\u003e902\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.Wold.hg19mitoExcludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107319\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/182046a0f055b0176178241a95cbd637\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefinition method unknown\u003c/td\u003e\n\u003ctd align=\"right\"\u003e295\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.Yeo.eCLIP_Excludableregions.hg19\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107320\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/350f49dc47e5307109e1e17d60223a31\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined from eCLIP data, includes skyscraper, rRNA pseudogene, unreliably mapped satellite repeat, and low complexity skyscraper peak regions\u003c/td\u003e\n\u003ctd align=\"right\"\u003e57\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm39.excluderanges\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107321\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/edc716833d4b5ee75c34a0692fc353d5\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions\u003c/td\u003e\n\u003ctd align=\"right\"\u003e3147\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.Boyle.mm10-Excludable.v2\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107322\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/a5311e39fe1590de66c1df6a5881a942\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions\u003c/td\u003e\n\u003ctd align=\"right\"\u003e3435\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.Hardison.Excludable.full\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107323\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/087541f51cf8c7d7078995d1bd95fd27\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefinition method unknown\u003c/td\u003e\n\u003ctd align=\"right\"\u003e7865\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.Hardison.psuExcludable.mm10\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107324\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/fc6b88f936c5cd880545943708e4c2af\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefinition method unknown\u003c/td\u003e\n\u003ctd align=\"right\"\u003e5552\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.Kundaje.anshul.Excludable.mm10\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107325\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/e6a89a8432f4a69bae41f60ed0c7e704\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by Anshul Kundaje as a part of ENCODE and modENCODE consortia\u003c/td\u003e\n\u003ctd align=\"right\"\u003e3010\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.Kundaje.mm10.Excludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107326\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/76c03b6c831f8fecdf4fee7adf2def6a\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by Anshul Kundaje as a part of ENCODE and modENCODE consortia\u003c/td\u003e\n\u003ctd align=\"right\"\u003e164\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.Lareau.mm10.full.Excludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107327\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/1bd30517be79d4d051308c693b822798\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)\u003c/td\u003e\n\u003ctd align=\"right\"\u003e523\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.Wold.mm10mitoExcludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107328\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/830f1ffd31689e3e7c22ff856f0ba02c\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefinition method unknown\u003c/td\u003e\n\u003ctd align=\"right\"\u003e123\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.Nordin.CandRblacklist_mm10\u003c/td\u003e\n\u003ctd align=\"left\"\u003eNA\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/cf3f5440b73fbbdc5b591a4bc6611a0e\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined from CUT\u0026amp;RUN negative controls as 0.1% top significant SEACR peaks in over 30% of samples\u003c/td\u003e\n\u003ctd align=\"right\"\u003e559\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm9.Lareau.mm9.full.Excludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107329\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/e903b285baefce8167367ce57a8c3d48\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)\u003c/td\u003e\n\u003ctd align=\"right\"\u003e3415\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm9.Wold.mm9mitoExcludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107330\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/9b4389a6a4b937df8abd62dad30fa3a3\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefinition method unknown\u003c/td\u003e\n\u003ctd align=\"right\"\u003e123\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ece11.Boyle.ce11-Excludable.v2\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107331\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/7235114a78b1709be96f0d6a82b4ea36\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions\u003c/td\u003e\n\u003ctd align=\"right\"\u003e97\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ece10.Boyle.ce10-Excludable.v2\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107332\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/6de11bb5f50ee015b23ac96f433f00bb\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions\u003c/td\u003e\n\u003ctd align=\"right\"\u003e100\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ece10.Kundaje.ce10-Excludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107333\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/32b59590fa83161687cec4cabfa2bb2b\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by Anshul Kundaje, superseded by ce10.Boyle.ce10-Excludable.v2\u003c/td\u003e\n\u003ctd align=\"right\"\u003e122\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003edanRer10.Domingues.Excludableed\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107334\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/a0a94af275f858d63550005627d260b7\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined manually using total RNA-seq.\u003c/td\u003e\n\u003ctd align=\"right\"\u003e57\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003edanRer10.Yang.Supplemental_Table_19.ChIP-seq_black_list_in_the_zebrafish_genome\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107335\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/78f5eb585019a4d795ef80159a597b15\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined via MACS2 peak calling using ChIP-seq (PMID: 33239788)\u003c/td\u003e\n\u003ctd align=\"right\"\u003e853\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003edm6.Boyle.dm6-Excludable.v2\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107336\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/24186dc2aac492074d3de9caede730a0\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions\u003c/td\u003e\n\u003ctd align=\"right\"\u003e182\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003edm3.Boyle.dm3-Excludable.v2\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107337\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/7427399e18d9c01e423b2f4963b409ea\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions\u003c/td\u003e\n\u003ctd align=\"right\"\u003e248\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003edm3.Kundaje.dm3-Excludable\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107338\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/0801a522159f7ebf2f669d8cade4aa8f\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by Anshul Kundaje. Contains heterochromatin chromosomes chr2LHet. Superseded by dm3.Boyle.dm3-Excludable.v2\u003c/td\u003e\n\u003ctd align=\"right\"\u003e306\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003eTAIR10.Wimberley.predicted_excluded_list_sorted_0.6\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107339\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/6f3a3ae3ee878b88a92093eb8e3fe982\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the ewimberley/peakPass software\u003c/td\u003e\n\u003ctd align=\"right\"\u003e887\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003eTAIR10.Klasfeld.arabidopsis_Excludable_20inputs\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107340\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/aa1c99c2dd2aef874486b1c0c3bf6b92\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions (DOI: 10.1101/2022.02.27.482177)\u003c/td\u003e\n\u003ctd align=\"right\"\u003e83\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003eTAIR10.Klasfeld.arabidopsis_greenscreen_20inputs\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107341\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/e5d66ee787a8cb0c76438bba768c2331\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eDefined by the green screen pipeline (DOI: 10.1101/2022.02.27.482177)\u003c/td\u003e\n\u003ctd align=\"right\"\u003e36\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003eT2T.Lareau.chm13v2.0_peaks\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107342\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/354dfced295f54f70ae9656ca8f9b141\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eRegions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist\u003c/td\u003e\n\u003ctd align=\"right\"\u003e817\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.Lareau.hg38_peaks\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107343\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/9fa55701a3bd3e7a598d1d2815e3390f\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eRegions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist\u003c/td\u003e\n\u003ctd align=\"right\"\u003e784\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.Lareau.hg19_peaks\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107344\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/79e924141251afbd4cde0c38456913fd\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eRegions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist\u003c/td\u003e\n\u003ctd align=\"right\"\u003e779\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.Lareau.mm10_peaks\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107345\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/1b76ab775549e116da5e1a89aad7019b\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eRegions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist\u003c/td\u003e\n\u003ctd align=\"right\"\u003e387\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm9.Lareau.mm9_peaks\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107346\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/5c4b1cb28175b72bc56adb0bd7384dfd\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eRegions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist\u003c/td\u003e\n\u003ctd align=\"right\"\u003e395\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.UCSC.numtS\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107347\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/cc4fd05fdfe015e4acd5111dac5b372f\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eHuman NumtS mitochondrial sequence\u003c/td\u003e\n\u003ctd align=\"right\"\u003e766\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm9.UCSC.numtS\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107348\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/29dc50750f0535b6b9c746ee8371c211\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eMouse NumtS mitochondrial sequence\u003c/td\u003e\n\u003ctd align=\"right\"\u003e172\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003eT2T.CHM13.chm13.draft_v2.0.cen_mask\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107349\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/44138ebb0d3340e70164d12649a47dc8\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eCentromeric satellite masking bed file (v2.0)\u003c/td\u003e\n\u003ctd align=\"right\"\u003e23\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003eT2T.CHM13.chm13.draft_v1.1.telomere\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107350\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/b72dd2fa5f8a916cc36960b93169c743\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eTelomere identified by the VGP pipeline (v1.1)\u003c/td\u003e\n\u003ctd align=\"right\"\u003e48\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003eT2T.UCSC.censat\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107351\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/f28798df2c4d72810e7c4626b5a62106\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eT2T peri/centromeric satellite annotation (v2.0, 20220329, CHM13 v2.0)\u003c/td\u003e\n\u003ctd align=\"right\"\u003e2523\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003eT2T.UCSC.gap\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107352\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/0747aae5f4cac92367a16c3eb1c7f3f1\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eLocations of assembly gaps, as determine by strings of ‘N’ characters (v1.0)\u003c/td\u003e\n\u003ctd align=\"right\"\u003e5\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003eT2T.UCSC.hgUnique.hg38\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107353\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/c3839f43c53a3c47733388528b853690\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eRegions unique to the T2T-CHM13 v2.0 assembly compared to the GRCh38/hg38 and GRCh37/hg19 reference assemblies\u003c/td\u003e\n\u003ctd align=\"right\"\u003e615\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.UCSC.centromere\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107354\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/0b1f161675fa0f52ac6d0d4f54b1efb9\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps from centromeres\u003c/td\u003e\n\u003ctd align=\"right\"\u003e109\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.UCSC.telomere\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107355\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/79f964e68d5daa1462c52ca54855b06a\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps from telomeres\u003c/td\u003e\n\u003ctd align=\"right\"\u003e48\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.UCSC.short_arm\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107356\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/92fc8f64f92d525c6b92c9aab5e2c711\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps on the short arm of the chromosome\u003c/td\u003e\n\u003ctd align=\"right\"\u003e5\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.UCSC.heterochromatin\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107357\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/8af7b48ab48183229d3bc72005040dc1\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps from large blocks of heterochromatin\u003c/td\u003e\n\u003ctd align=\"right\"\u003e11\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.UCSC.contig\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107358\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/2dd1b22f2add15bc7508580d18bc9495\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between contigs in scaffolds\u003c/td\u003e\n\u003ctd align=\"right\"\u003e285\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg38.UCSC.scaffold\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107359\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/de0c7f42f29fb83ac393e86a2ec28374\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between scaffolds in chromosome assemblies. Has extra chromosomes, use keepStandardChromosomes() filtering\u003c/td\u003e\n\u003ctd align=\"right\"\u003e254\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.UCSC.centromere\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107360\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/26ecf1381b6323791656f800ad39b69c\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps from centromeres\u003c/td\u003e\n\u003ctd align=\"right\"\u003e24\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.UCSC.telomere\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107361\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/2bcad8794847411e9b3f52ff39c4f377\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps from telomeres\u003c/td\u003e\n\u003ctd align=\"right\"\u003e46\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.UCSC.short_arm\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107362\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/e09fac8aedf1230ab77ac4194fd75784\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps on the short arm of the chromosome\u003c/td\u003e\n\u003ctd align=\"right\"\u003e5\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.UCSC.heterochromatin\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107363\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/8ea9b6cdfe68a4b4111e5b03157af371\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps from large blocks of heterochromatin\u003c/td\u003e\n\u003ctd align=\"right\"\u003e12\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.UCSC.clone\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107364\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/4f3b1098a0f4ea5e81747f4414a8d294\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between clones in the same map contig. Has extra chromosomes, use keepStandardChromosomes() filtering\u003c/td\u003e\n\u003ctd align=\"right\"\u003e107\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.UCSC.contig\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107365\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/a4da41916b0b213d4e3b89f5ab20e1e8\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between contigs in scaffolds\u003c/td\u003e\n\u003ctd align=\"right\"\u003e163\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003ehg19.UCSC.scaffold\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107366\u003c/td\u003e\n\u003ctd align=\"left\"\u003eNA\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between scaffolds in chromosome assemblies. Only non-autosomal chromosomes\u003c/td\u003e\n\u003ctd align=\"right\"\u003e0\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm39.UCSC.centromere\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107367\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/1aa3f73ffa8e6d498f0f3f22e0302472\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps from centromeres\u003c/td\u003e\n\u003ctd align=\"right\"\u003e20\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm39.UCSC.telomere\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107368\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/883bdae38244c6f0e0facfbd4fcc601b\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps from telomeres\u003c/td\u003e\n\u003ctd align=\"right\"\u003e42\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm39.UCSC.short_arm\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107369\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/f61de62eae0898943e6c9d163b0a3989\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps on the short arm of the chromosome\u003c/td\u003e\n\u003ctd align=\"right\"\u003e21\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm39.UCSC.contig\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107370\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/732437d9fcb8992b2e5c6513bfed2586\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between contigs in scaffolds\u003c/td\u003e\n\u003ctd align=\"right\"\u003e60\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm39.UCSC.scaffold\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107371\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/97e738326c0681f5ebe94b0d28d058c5\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between scaffolds in chromosome assemblies\u003c/td\u003e\n\u003ctd align=\"right\"\u003e115\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.UCSC.centromere\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107372\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/b0f9aa3cc8a4a43f59b463891b5d12c8\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps from centromeres\u003c/td\u003e\n\u003ctd align=\"right\"\u003e20\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.UCSC.telomere\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107373\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/051090e82c227bcc55dba3e953bc6daa\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps from telomeres\u003c/td\u003e\n\u003ctd align=\"right\"\u003e42\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.UCSC.short_arm\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107374\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/f4d2d6fe334deca5800ca9ae39ce95ce\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps on the short arm of the chromosome\u003c/td\u003e\n\u003ctd align=\"right\"\u003e21\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.UCSC.clone\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107375\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/9ecfce46335e4d5b3a1230b69690a25a\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between clones in the same map contig. Has extra chromosomes, use keepStandardChromosomes() filtering\u003c/td\u003e\n\u003ctd align=\"right\"\u003e4\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.UCSC.contig\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107376\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/82d2374cf5524a2b13dcf9c3dc487d6f\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between contigs in scaffolds\u003c/td\u003e\n\u003ctd align=\"right\"\u003e104\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.UCSC.scaffold\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107377\u003c/td\u003e\n\u003ctd align=\"left\"\u003eNA\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between scaffolds in chromosome assemblies\u003c/td\u003e\n\u003ctd align=\"right\"\u003e0\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.UCSC.other\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107378\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/75662812e5eb228b25c9ae5a28fbb402\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eSequence of Ns in the assembly that were not marked as gaps in the AGP (A Golden Path) assembly definition file. Has extra chromosomes, use keepStandardChromosomes() filtering\u003c/td\u003e\n\u003ctd align=\"right\"\u003e383\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm10.UCSC.fragment\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107379\u003c/td\u003e\n\u003ctd align=\"left\"\u003eNA\u003c/td\u003e\n\u003ctd align=\"left\"\u003eA single gap of 31 bases in chrX_GL456233_random\u003c/td\u003e\n\u003ctd align=\"right\"\u003e0\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm9.UCSC.centromere\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107380\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/99e4d2c9a794d321bfcf01709787caac\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps from centromeres\u003c/td\u003e\n\u003ctd align=\"right\"\u003e21\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm9.UCSC.fragment\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107381\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/ecf3f802759c5dc93f1446b6942c58b3\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between the contigs of a draft clone. (In this context, a contig is a set of overlapping sequence reads). Has extra chromosomes, use keepStandardChromosomes() filtering\u003c/td\u003e\n\u003ctd align=\"right\"\u003e436\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003emm9.UCSC.contig\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107382\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/4dd8bb54f6432144c045619337d8212e\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between contigs in scaffolds. Has extra chromosomes, use keepStandardChromosomes() filtering\u003c/td\u003e\n\u003ctd align=\"right\"\u003e105\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003edanRer10.UCSC.contig\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107383\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/5d41a9fa328769b63734e11e6ae4252b\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between contigs in scaffolds\u003c/td\u003e\n\u003ctd align=\"right\"\u003e2338\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003edanRer10.UCSC.scaffold\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107384\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/a5feefb2d573d265f1085043c208c2ed\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between scaffolds in chromosome assemblies\u003c/td\u003e\n\u003ctd align=\"right\"\u003e16496\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003edm6.UCSC.other\u003c/td\u003e\n\u003ctd align=\"left\"\u003eAH107385\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/11a80264dcdad6c0868ea48637a799df\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eSequence of Ns in the assembly that were not marked as gaps in the AGP (A Golden Path) assembly definition file\u003c/td\u003e\n\u003ctd align=\"right\"\u003e268\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003edm3.UCSC.contig\u003c/td\u003e\n\u003ctd align=\"left\"\u003eNA\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/f9822a0f88047e92cf92824fe025b2f2\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between contigs in scaffolds\u003c/td\u003e\n\u003ctd align=\"right\"\u003e7\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003edm3.UCSC.scaffold\u003c/td\u003e\n\u003ctd align=\"left\"\u003eNA\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/d4e31a2c488de8ff335b0cb779c9cef5\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps between scaffolds in chromosome assemblies\u003c/td\u003e\n\u003ctd align=\"right\"\u003e1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"left\"\u003eTAIR10.UCSC.araTha1.gap\u003c/td\u003e\n\u003ctd align=\"left\"\u003eNA\u003c/td\u003e\n\u003ctd align=\"left\"\u003e\u003ca href=\"http://bedbase.org/#/bedsplash/74585119b9b90d3b4ad077b10b487d39\" rel=\"nofollow\"\u003elink\u003c/a\u003e\u003c/td\u003e\n\u003ctd align=\"left\"\u003eGaps in the May 2011 Arabidopsis thaliana genome assembly\u003c/td\u003e\n\u003ctd align=\"right\"\u003e357\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCitation\u003c/h1\u003e\u003ca id=\"user-content-citation\" class=\"anchor\" aria-label=\"Permalink: Citation\" href=\"#citation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBelow is the citation output from using \u003ccode\u003ecitation('excluderanges')\u003c/code\u003e in\nR. Please run this yourself to check for any updates on how to cite\n\u003cstrong\u003eexcluderanges\u003c/strong\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"print(citation(\u0026quot;excluderanges\u0026quot;), bibtex = TRUE)\"\u003e\u003cpre\u003eprint(citation(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eexcluderanges\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ebibtex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCode of Conduct\u003c/h1\u003e\u003ca id=\"user-content-code-of-conduct\" class=\"anchor\" aria-label=\"Permalink: Code of Conduct\" href=\"#code-of-conduct\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePlease note that the \u003ccode\u003eexcluderanges\u003c/code\u003e project is released with a\n\u003ca href=\"https://bioconductor.github.io/bioc_coc_multilingual/\" rel=\"nofollow\"\u003eContributor Code of\nConduct\u003c/a\u003e. By\ncontributing to this project, you agree to abide by its terms.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis package was developed using\n\u003cem\u003e\u003ca href=\"https://bioconductor.org/packages/3.18/biocthis\" rel=\"nofollow\"\u003ebiocthis\u003c/a\u003e\u003c/em\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eReferences\u003c/h1\u003e\u003ca id=\"user-content-references\" class=\"anchor\" aria-label=\"Permalink: References\" href=\"#references\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv id=\"user-content-refs\" dir=\"auto\"\u003e\n\u003cdiv id=\"user-content-ref-amemiya:2019aa\" dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eAmemiya, Haley M, Anshul Kundaje, and Alan P Boyle. 2019. “The ENCODE\nBlacklist: Identification of Problematic Regions of the Genome.” \u003cem\u003eSci\nRep\u003c/em\u003e 9 (1): 9354. \u003ca href=\"https://doi.org/10.1038/s41598-019-45839-z\" rel=\"nofollow\"\u003ehttps://doi.org/10.1038/s41598-019-45839-z\u003c/a\u003e.\u003c/p\u003e\n\u003c/div\u003e\n\u003cdiv id=\"user-content-ref-klasfeld2022-nn\" dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eKlasfeld, Sammy, and Doris Wagner. 2022. “Greenscreen Decreases Type I\nErrors and Increases True Peak Detection in Genomic Datasets Including\n\u003cspan\u003eChIP-seq\u003c/span\u003e.” \u003cem\u003ebioRxiv\u003c/em\u003e.\u003c/p\u003e\n\u003c/div\u003e\n\u003cdiv id=\"user-content-ref-nordin2022cut\" dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eNordin, Anna, Gianluca Zambanini, Pierfrancesco Pagella, and Claudio\nCantu. 2022. “The CUT\u0026amp;RUN Blacklist of Problematic Regions of the\nGenome.” \u003cem\u003ebioRxiv\u003c/em\u003e.\u003c/p\u003e\n\u003c/div\u003e\n\u003cdiv id=\"user-content-ref-ogata:2023aa\" dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eOgata, Jonathan D, Wancen Mu, Eric S Davis, Bingjie Xue, J Chuck\nHarrell, Nathan C Sheffield, Douglas H Phanstiel, Michael I Love, and\nMikhail G Dozmorov. 2023. “Excluderanges: Exclusion Sets for T2T-CHM13,\nGRCm39, and Other Genome Assemblies.” \u003cem\u003eBioinformatics\u003c/em\u003e 39 (4).\n\u003ca href=\"https://doi.org/10.1093/bioinformatics/btad198\" rel=\"nofollow\"\u003ehttps://doi.org/10.1093/bioinformatics/btad198\u003c/a\u003e.\u003c/p\u003e\n\u003c/div\u003e\n\u003cdiv id=\"user-content-ref-wimberley:2019ub\" dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eWimberley, Charles E, and Steffen Heber. 2019. “PeakPass: Automating\nChIP-Seq Blacklist Creation.” \u003cem\u003eJ Comput Biol\u003c/em\u003e, December.\n\u003ca href=\"https://doi.org/10.1089/cmb.2019.0295\" rel=\"nofollow\"\u003ehttps://doi.org/10.1089/cmb.2019.0295\u003c/a\u003e.\u003c/p\u003e\n\u003c/div\u003e\n\u003c/div\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"excluderanges, genomic ranges of problematic genomic regions","anchor":"excluderanges-genomic-ranges-of-problematic-genomic-regions","htmlText":"excluderanges, genomic ranges of problematic genomic regions"},{"level":2,"text":"Installation instructions","anchor":"installation-instructions","htmlText":"Installation instructions"},{"level":1,"text":"Use excluderanges","anchor":"use-excluderanges","htmlText":"Use excluderanges"},{"level":2,"text":"Compare the number of excludable regions","anchor":"compare-the-number-of-excludable-regions","htmlText":"Compare the number of excludable regions"},{"level":2,"text":"Compare the width of excludable regions","anchor":"compare-the-width-of-excludable-regions","htmlText":"Compare the width of excludable regions"},{"level":2,"text":"Compare overlaps among sets","anchor":"compare-overlaps-among-sets","htmlText":"Compare overlaps among sets"},{"level":2,"text":"Metadata analysis","anchor":"metadata-analysis","htmlText":"Metadata analysis"},{"level":1,"text":"BEDbase data download","anchor":"bedbase-data-download","htmlText":"BEDbase data download"},{"level":1,"text":"Mitochondrial DNA sequences, NUMTs","anchor":"mitochondrial-dna-sequences-numts","htmlText":"Mitochondrial DNA sequences, NUMTs"},{"level":1,"text":"Centromeres, telomeres, etc.","anchor":"centromeres-telomeres-etc","htmlText":"Centromeres, telomeres, etc."},{"level":2,"text":"Gaps in mouse genome","anchor":"gaps-in-mouse-genome","htmlText":"Gaps in mouse genome"},{"level":1,"text":"CUT\u0026RUN excludable sets","anchor":"cutrun-excludable-sets","htmlText":"CUT\u0026amp;RUN excludable sets"},{"level":1,"text":"Summary table","anchor":"summary-table","htmlText":"Summary table"},{"level":1,"text":"Citation","anchor":"citation","htmlText":"Citation"},{"level":1,"text":"Code of Conduct","anchor":"code-of-conduct","htmlText":"Code of Conduct"},{"level":1,"text":"References","anchor":"references","htmlText":"References"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fdozmorovlab%2Fexcluderanges"}},{"displayName":"LICENSE","repoName":"excluderanges","refName":"master","path":"LICENSE","preferredFileType":"license","tabName":"License","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fdozmorovlab%2Fexcluderanges"}},{"displayName":"LICENSE.md","repoName":"excluderanges","refName":"master","path":"LICENSE.md","preferredFileType":"license","tabName":"MIT","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fdozmorovlab%2Fexcluderanges"}}],"overviewFilesProcessingTime":37.69527}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
.bhqztV{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.bhqztV span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.bhqztV:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.bhqztV:focus{outline:2px solid transparent;}/*!sc*/
.bhqztV:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.bhqztV:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.bhqztV::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:transparent;border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.bhqztV::after{background-color:transparent;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,bhqztV,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="master branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        master
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/dozmorovlab/excluderanges/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/dozmorovlab/excluderanges/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/dozmorovlab/excluderanges/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/dozmorovlab/excluderanges/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/dozmorovlab/excluderanges/commits/master/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           92 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/dozmorovlab/excluderanges/commits/master/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".github, (Directory)" class="Link--primary" href="/dozmorovlab/excluderanges/tree/master/.github" title=".github">
                          .github
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".github, (Directory)" class="Link--primary" href="/dozmorovlab/excluderanges/tree/master/.github" title=".github">
                          .github
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="R, (Directory)" class="Link--primary" href="/dozmorovlab/excluderanges/tree/master/R" title="R">
                          R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="R, (Directory)" class="Link--primary" href="/dozmorovlab/excluderanges/tree/master/R" title="R">
                          R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="inst, (Directory)" class="Link--primary" href="/dozmorovlab/excluderanges/tree/master/inst" title="inst">
                          inst
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="inst, (Directory)" class="Link--primary" href="/dozmorovlab/excluderanges/tree/master/inst" title="inst">
                          inst
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="man, (Directory)" class="Link--primary" href="/dozmorovlab/excluderanges/tree/master/man" title="man">
                          man
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="man, (Directory)" class="Link--primary" href="/dozmorovlab/excluderanges/tree/master/man" title="man">
                          man
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="vignettes, (Directory)" class="Link--primary" href="/dozmorovlab/excluderanges/tree/master/vignettes" title="vignettes">
                          vignettes
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="vignettes, (Directory)" class="Link--primary" href="/dozmorovlab/excluderanges/tree/master/vignettes" title="vignettes">
                          vignettes
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="DESCRIPTION, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/DESCRIPTION" title="DESCRIPTION">
                          DESCRIPTION
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="DESCRIPTION, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/DESCRIPTION" title="DESCRIPTION">
                          DESCRIPTION
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-6">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-7">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE.md, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/LICENSE.md" title="LICENSE.md">
                          LICENSE.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE.md, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/LICENSE.md" title="LICENSE.md">
                          LICENSE.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-8">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="NAMESPACE, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/NAMESPACE" title="NAMESPACE">
                          NAMESPACE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="NAMESPACE, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/NAMESPACE" title="NAMESPACE">
                          NAMESPACE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-9">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="NEWS.md, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/NEWS.md" title="NEWS.md">
                          NEWS.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="NEWS.md, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/NEWS.md" title="NEWS.md">
                          NEWS.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-10">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.Rmd, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/README.Rmd" title="README.Rmd">
                          README.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.Rmd, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/README.Rmd" title="README.Rmd">
                          README.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-11">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/dozmorovlab/excluderanges/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd show-for-mobile" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-law" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="License">
                        License
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-law" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="MIT license">
                        MIT license
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <ul dir="auto">
                     <li>
                      <a href="#excluderanges-genomic-ranges-of-problematic-genomic-regions">
                       excluderanges, genomic ranges of problematic genomic
regions
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#installation-instructions">
                         Installation instructions
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#use-excluderanges">
                       Use excluderanges
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#compare-the-number-of-excludable-regions">
                         Compare the number of excludable
regions
                        </a>
                       </li>
                       <li>
                        <a href="#compare-the-width-of-excludable-regions">
                         Compare the width of excludable
regions
                        </a>
                       </li>
                       <li>
                        <a href="#compare-overlaps-among-sets">
                         Compare overlaps among sets
                        </a>
                       </li>
                       <li>
                        <a href="#metadata-analysis">
                         Metadata analysis
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#bedbase-data-download">
                       BEDbase data download
                      </a>
                     </li>
                     <li>
                      <a href="#mitochondrial-dna-sequences-numts">
                       Mitochondrial DNA sequences,
NUMTs
                      </a>
                     </li>
                     <li>
                      <a href="#centromeres-telomeres-etc">
                       Centromeres, telomeres, etc.
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#gaps-in-mouse-genome">
                         Gaps in mouse genome
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#cutrun-excludable-sets">
                       CUT&amp;RUN excludable sets
                      </a>
                     </li>
                     <li>
                      <a href="#summary-table">
                       Summary table
                      </a>
                     </li>
                     <li>
                      <a href="#citation">
                       Citation
                      </a>
                     </li>
                     <li>
                      <a href="#code-of-conduct">
                       Code of Conduct
                      </a>
                     </li>
                     <li>
                      <a href="#references">
                       References
                      </a>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      excluderanges, genomic ranges of problematic genomic regions
                     </h1>
                     <a aria-label="Permalink: excluderanges, genomic ranges of problematic genomic regions" class="anchor" href="#excluderanges-genomic-ranges-of-problematic-genomic-regions" id="user-content-excluderanges-genomic-ranges-of-problematic-genomic-regions">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Ogata, Jonathan D, Wancen Mu, Eric S Davis, Bingjie Xue, J Chuck
Harrell, Nathan C Sheffield, Douglas H Phanstiel, Michael I Love, and
Mikhail G Dozmorov. “Excluderanges: Exclusion Sets for T2T-CHM13,
GRCm39, and Other Genome Assemblies.”
                     <em>
                      Bioinformatics
                     </em>
                     , April 3, 2023,
PMID:
                     <a href="https://pubmed.ncbi.nlm.nih.gov/37067481/" rel="nofollow">
                      37067481
                     </a>
                     , DOI:
                     <a href="https://doi.org/10.1093/bioinformatics/btad198" rel="nofollow">
                      10.1093/bioinformatics/btad198
                     </a>
                    </p>
                    <p dir="auto">
                     <a href="/dozmorovlab/excluderanges/blob/master/man/figures/logo.png" rel="noopener noreferrer" target="_blank">
                      <img align="right" id="user-content-excluderanges_logo_by_mike_love" src="/dozmorovlab/excluderanges/raw/master/man/figures/logo.png" style="max-width: 100%;" width="125"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Coordinates of problematic genomic regions that should be avoided when
working with genomic data (Ogata et al. 2023). For human, mouse, and
selected model organisms. Please, open an
                     <a href="https://github.com/dozmorovlab/excluderanges/issues">
                      issue
                     </a>
                     to suggest
other resources. Package on Bioconductor:
                     <em>
                      <a href="https://bioconductor.org/packages/3.18/excluderanges" rel="nofollow">
                       excluderanges
                      </a>
                     </em>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      New (2022-09-20)
                     </strong>
                     - Exclusion sets for human
                     <a href="http://bedbase.org/#/bedsplash/8329d8c624880308ab51ba05149a737d" rel="nofollow">
                      T2T-CHM13
                     </a>
                     and mouse
                     <a href="http://bedbase.org/#/bedsplash/edc716833d4b5ee75c34a0692fc353d5" rel="nofollow">
                      GRCm39/mm39
                     </a>
                     genome assemblies are available. Download all data from
                     <a href="https://drive.google.com/drive/folders/1sF9m8Y3eZouTZ3IEEywjs2kfHOWFBSJT?usp=sharing" rel="nofollow">
                      Google
Drive
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      TL;DR
                     </strong>
                     - For human hg38 genome assembly,
                     <a href="https://twitter.com/anshulkundaje/status/1263546023151992832?s=20" rel="nofollow">
                      Anshul
                     </a>
                     and we recommend
                     <a href="https://www.encodeproject.org/files/ENCFF356LFX/" rel="nofollow">
                      ENCFF356LFX exclusion list
regions
                     </a>
                     . Also
available as
                     <code>
                      hg38.Kundaje.GRCh38_unified_Excludable
                     </code>
                     excluderanges
object (AnnotationHub ID: AH107305) and
                     <a href="http://bedbase.org/#/bedsplash/1a561729234c2844303a051b16f66656" rel="nofollow">
                      BEDbase.org
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     BED files of exclusion regions are available on the
                     <a href="https://www.encodeproject.org/search/?searchTerm=exclusion+list" rel="nofollow">
                      ENCODE
project
                     </a>
                     website and scattered across various websites, such as
                     <a href="https://github.com/Boyle-Lab/Blacklist/tree/master/lists">
                      Blacklist
                     </a>
                     (Amemiya, Kundaje, and Boyle 2019),
                     <a href="https://github.com/ewimberley/peakPass/tree/main/excludedlists">
                      Peakpass
                     </a>
                     (Wimberley and Heber 2019),
                     <a href="https://github.com/sklasfeld/GreenscreenProject/tree/main/data">
                      Greenscreen
                     </a>
                     (Klasfeld and Wagner 2022). Human and mouse genome assemblies have the
largest number of exclusion sets generated by multiple labs. These
exclusion sets frequently lack annotation and curation methods, creating
uncertainty what to use. The purpose of this package is to provide a
unified place for informed retrieval of exclusion regions.
                    </p>
                    <p dir="auto">
                     Naming convention:
                     <code>
                      &lt;genome assembly&gt;.&lt;lab&gt;.&lt;original file name&gt;
                     </code>
                     , e.g.,
                     <code>
                      hg19.Birney.wgEncodeDacMapabilityConsensusExcludable
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     See
                     <a href="/dozmorovlab/excluderanges/blob/master/inst/scripts/make-data.R">
                      make-data.R
                     </a>
                     how we created excluderanges
objects.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Installation instructions
                     </h2>
                     <a aria-label="Permalink: Installation instructions" class="anchor" href="#installation-instructions" id="user-content-installation-instructions">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="https://www.bioconductor.org/install/#install-R" rel="nofollow">
                      Install
                     </a>
                     the latest
release of R, then get the latest version of Bioconductor by starting R
and entering the commands:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# BiocManager::install(version = "3.18")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> if (!require("BiocManager", quietly = TRUE))</span>
<span class="pl-c"><span class="pl-c">#</span>     install.packages("BiocManager")</span>
<span class="pl-c"><span class="pl-c">#</span> BiocManager::install(version = "3.18")</span></pre>
                    </div>
                    <p dir="auto">
                     Then, install additional packages using the following code:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# BiocManager::install("AnnotationHub", update = FALSE) 
# BiocManager::install("GenomicRanges", update = FALSE)
# BiocManager::install("plyranges", update = FALSE)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> BiocManager::install("AnnotationHub", update = FALSE) </span>
<span class="pl-c"><span class="pl-c">#</span> BiocManager::install("GenomicRanges", update = FALSE)</span>
<span class="pl-c"><span class="pl-c">#</span> BiocManager::install("plyranges", update = FALSE)</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Use excluderanges
                     </h1>
                     <a aria-label="Permalink: Use excluderanges" class="anchor" href="#use-excluderanges" id="user-content-use-excluderanges">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Get an overview of what’s available
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="suppressMessages(library(GenomicRanges))
suppressMessages(library(AnnotationHub))
ah &lt;- AnnotationHub()
query_data &lt;- subset(ah, preparerclass == &quot;excluderanges&quot;)
# You can search for multiple terms
# query_data &lt;- query(ah, c(&quot;excluderanges&quot;, &quot;Kundaje&quot;, &quot;hg38&quot;))
query_data
#&gt; AnnotationHub with 82 records
#&gt; # snapshotDate(): 2023-10-20
#&gt; # $dataprovider: UCSC, GitHub, ENCODE, UCSChub, excluderanges, Stanford.edu,...
#&gt; # $species: Homo sapiens, Mus musculus, Drosophila melanogaster, Danio rerio...
#&gt; # $rdataclass: GRanges
#&gt; # additional mcols(): taxonomyid, genome, description,
#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,
#&gt; #   rdatapath, sourceurl, sourcetype 
#&gt; # retrieve records with, e.g., 'object[[&quot;AH107304&quot;]]' 
#&gt; 
#&gt;              title                                 
#&gt;   AH107304 | T2T.excluderanges                     
#&gt;   AH107305 | hg38.Kundaje.GRCh38_unified_Excludable
#&gt;   AH107306 | hg38.Bernstein.Mint_Excludable_GRCh38 
#&gt;   AH107307 | hg38.Boyle.hg38-Excludable.v2         
#&gt;   AH107308 | hg38.Kundaje.GRCh38.Excludable        
#&gt;   ...        ...                                   
#&gt;   AH107381 | danRer10.UCSC.scaffold                
#&gt;   AH107382 | dm6.UCSC.other                        
#&gt;   AH107383 | dm3.UCSC.contig                       
#&gt;   AH107384 | dm3.UCSC.scaffold                     
#&gt;   AH107385 | TAIR10.UCSC.araTha1.gap" dir="auto">
                     <pre>suppressMessages(library(<span class="pl-smi">GenomicRanges</span>))
suppressMessages(library(<span class="pl-smi">AnnotationHub</span>))
<span class="pl-smi">ah</span> <span class="pl-k">&lt;-</span> AnnotationHub()
<span class="pl-smi">query_data</span> <span class="pl-k">&lt;-</span> subset(<span class="pl-smi">ah</span>, <span class="pl-smi">preparerclass</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>excluderanges<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> You can search for multiple terms</span>
<span class="pl-c"><span class="pl-c">#</span> query_data &lt;- query(ah, c("excluderanges", "Kundaje", "hg38"))</span>
<span class="pl-smi">query_data</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; AnnotationHub with 82 records</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # snapshotDate(): 2023-10-20</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # $dataprovider: UCSC, GitHub, ENCODE, UCSChub, excluderanges, Stanford.edu,...</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # $species: Homo sapiens, Mus musculus, Drosophila melanogaster, Danio rerio...</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # $rdataclass: GRanges</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # additional mcols(): taxonomyid, genome, description,</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; #   rdatapath, sourceurl, sourcetype </span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # retrieve records with, e.g., 'object[["AH107304"]]' </span>
<span class="pl-c"><span class="pl-c">#</span>&gt; </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;              title                                 </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107304 | T2T.excluderanges                     </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107305 | hg38.Kundaje.GRCh38_unified_Excludable</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107306 | hg38.Bernstein.Mint_Excludable_GRCh38 </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107307 | hg38.Boyle.hg38-Excludable.v2         </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107308 | hg38.Kundaje.GRCh38.Excludable        </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   ...        ...                                   </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107381 | danRer10.UCSC.scaffold                </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107382 | dm6.UCSC.other                        </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107383 | dm3.UCSC.contig                       </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107384 | dm3.UCSC.scaffold                     </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107385 | TAIR10.UCSC.araTha1.gap</span></pre>
                    </div>
                    <p dir="auto">
                     <code>
                      hg38.Kundaje.GRCh38_unified_Excludable
                     </code>
                     object recommended by Anshul
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='excludeGR.hg38.Kundaje.1 &lt;- query_data[["AH107305"]]
#&gt; loading from cache
# Always a good idea to sort GRanges and keep standard chromosomes
excludeGR.hg38.Kundaje.1 &lt;- excludeGR.hg38.Kundaje.1 %&gt;% 
  sort() %&gt;% keepStandardChromosomes(pruning.mode = "tidy")
excludeGR.hg38.Kundaje.1
#&gt; GRanges object with 910 ranges and 0 metadata columns:
#&gt;         seqnames            ranges strand
#&gt;            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;     [1]     chr1     628903-635104      *
#&gt;     [2]     chr1   5850087-5850571      *
#&gt;     [3]     chr1   8909610-8910014      *
#&gt;     [4]     chr1   9574580-9574997      *
#&gt;     [5]     chr1 32043823-32044203      *
#&gt;     ...      ...               ...    ...
#&gt;   [906]     chrY 11290797-11334278      *
#&gt;   [907]     chrY 11493053-11592850      *
#&gt;   [908]     chrY 11671014-11671046      *
#&gt;   [909]     chrY 11721528-11749472      *
#&gt;   [910]     chrY 56694632-56889743      *
#&gt;   -------
#&gt;   seqinfo: 24 sequences from hg38 genome' dir="auto">
                     <pre><span class="pl-smi">excludeGR.hg38.Kundaje.1</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">query_data</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH107305<span class="pl-pds">"</span></span>]]
<span class="pl-c"><span class="pl-c">#</span>&gt; loading from cache</span>
<span class="pl-c"><span class="pl-c">#</span> Always a good idea to sort GRanges and keep standard chromosomes</span>
<span class="pl-smi">excludeGR.hg38.Kundaje.1</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">excludeGR.hg38.Kundaje.1</span> %<span class="pl-k">&gt;</span>% 
  sort() %<span class="pl-k">&gt;</span>% keepStandardChromosomes(<span class="pl-v">pruning.mode</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>tidy<span class="pl-pds">"</span></span>)
<span class="pl-smi">excludeGR.hg38.Kundaje.1</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; GRanges object with 910 ranges and 0 metadata columns:</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;         seqnames            ranges strand</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [1]     chr1     628903-635104      *</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [2]     chr1   5850087-5850571      *</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [3]     chr1   8909610-8910014      *</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [4]     chr1   9574580-9574997      *</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [5]     chr1 32043823-32044203      *</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     ...      ...               ...    ...</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [906]     chrY 11290797-11334278      *</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [907]     chrY 11493053-11592850      *</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [908]     chrY 11671014-11671046      *</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [909]     chrY 11721528-11749472      *</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [910]     chrY 56694632-56889743      *</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   -------</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   seqinfo: 24 sequences from hg38 genome</span></pre>
                    </div>
                    <p dir="auto">
                     Save the data in a BED file, if needed.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Note that rtracklayer::import and rtracklayer::export perform unexplained
# start coordinate conversion, likely related to 0- and 1-based coordinate
# system. We recommend converting GRanges to a data frame and save tab-separated
write.table(as.data.frame(excludeGR.hg38.Kundaje.1), 
            file = "hg38.Kundaje.GRCh38_unified_Excludable.bed",
            sep = "\t", row.names = FALSE, col.names = FALSE, quote = FALSE)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Note that rtracklayer::import and rtracklayer::export perform unexplained</span>
<span class="pl-c"><span class="pl-c">#</span> start coordinate conversion, likely related to 0- and 1-based coordinate</span>
<span class="pl-c"><span class="pl-c">#</span> system. We recommend converting GRanges to a data frame and save tab-separated</span>
write.table(as.data.frame(<span class="pl-smi">excludeGR.hg38.Kundaje.1</span>), 
            <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>hg38.Kundaje.GRCh38_unified_Excludable.bed<span class="pl-pds">"</span></span>,
            <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\t</span><span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">col.names</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)</pre>
                    </div>
                    <p dir="auto">
                     We can load other excludable regions for the hg38 genome assembly and
compare them.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="query_data &lt;- query(ah, c(&quot;excluderanges&quot;, &quot;hg38&quot;))
query_data
#&gt; AnnotationHub with 17 records
#&gt; # snapshotDate(): 2023-10-20
#&gt; # $dataprovider: UCSC, ENCODE, GitHub, UCSChub
#&gt; # $species: Homo sapiens
#&gt; # $rdataclass: GRanges
#&gt; # additional mcols(): taxonomyid, genome, description,
#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,
#&gt; #   rdatapath, sourceurl, sourcetype 
#&gt; # retrieve records with, e.g., 'object[[&quot;AH107305&quot;]]' 
#&gt; 
#&gt;              title                                 
#&gt;   AH107305 | hg38.Kundaje.GRCh38_unified_Excludable
#&gt;   AH107306 | hg38.Bernstein.Mint_Excludable_GRCh38 
#&gt;   AH107307 | hg38.Boyle.hg38-Excludable.v2         
#&gt;   AH107308 | hg38.Kundaje.GRCh38.Excludable        
#&gt;   AH107309 | hg38.Lareau.hg38.full.Excludable      
#&gt;   ...        ...                                   
#&gt;   AH107355 | hg38.UCSC.telomere                    
#&gt;   AH107356 | hg38.UCSC.short_arm                   
#&gt;   AH107357 | hg38.UCSC.heterochromatin             
#&gt;   AH107358 | hg38.UCSC.contig                      
#&gt;   AH107359 | hg38.UCSC.scaffold
excludeGR.hg38.Bernstein &lt;- query_data[[&quot;AH107306&quot;]]
#&gt; loading from cache
excludeGR.hg38.Boyle     &lt;- query_data[[&quot;AH107307&quot;]]
#&gt; loading from cache
excludeGR.hg38.Kundaje.2 &lt;- query_data[[&quot;AH107308&quot;]]
#&gt; loading from cache
excludeGR.hg38.Lareau    &lt;- query_data[[&quot;AH107309&quot;]]
#&gt; loading from cache
excludeGR.hg38.Reddy     &lt;- query_data[[&quot;AH107310&quot;]]
#&gt; loading from cache
excludeGR.hg38.Wimberley &lt;- query_data[[&quot;AH107311&quot;]]
#&gt; loading from cache
excludeGR.hg38.Wold      &lt;- query_data[[&quot;AH107312&quot;]]
#&gt; loading from cache
excludeGR.hg38.Yeo       &lt;- query_data[[&quot;AH107313&quot;]]
#&gt; loading from cache" dir="auto">
                     <pre><span class="pl-smi">query_data</span> <span class="pl-k">&lt;-</span> query(<span class="pl-smi">ah</span>, c(<span class="pl-s"><span class="pl-pds">"</span>excluderanges<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>hg38<span class="pl-pds">"</span></span>))
<span class="pl-smi">query_data</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; AnnotationHub with 17 records</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # snapshotDate(): 2023-10-20</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # $dataprovider: UCSC, ENCODE, GitHub, UCSChub</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # $species: Homo sapiens</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # $rdataclass: GRanges</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # additional mcols(): taxonomyid, genome, description,</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; #   rdatapath, sourceurl, sourcetype </span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # retrieve records with, e.g., 'object[["AH107305"]]' </span>
<span class="pl-c"><span class="pl-c">#</span>&gt; </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;              title                                 </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107305 | hg38.Kundaje.GRCh38_unified_Excludable</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107306 | hg38.Bernstein.Mint_Excludable_GRCh38 </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107307 | hg38.Boyle.hg38-Excludable.v2         </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107308 | hg38.Kundaje.GRCh38.Excludable        </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107309 | hg38.Lareau.hg38.full.Excludable      </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   ...        ...                                   </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107355 | hg38.UCSC.telomere                    </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107356 | hg38.UCSC.short_arm                   </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107357 | hg38.UCSC.heterochromatin             </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107358 | hg38.UCSC.contig                      </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107359 | hg38.UCSC.scaffold</span>
<span class="pl-smi">excludeGR.hg38.Bernstein</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">query_data</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH107306<span class="pl-pds">"</span></span>]]
<span class="pl-c"><span class="pl-c">#</span>&gt; loading from cache</span>
<span class="pl-smi">excludeGR.hg38.Boyle</span>     <span class="pl-k">&lt;-</span> <span class="pl-smi">query_data</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH107307<span class="pl-pds">"</span></span>]]
<span class="pl-c"><span class="pl-c">#</span>&gt; loading from cache</span>
<span class="pl-smi">excludeGR.hg38.Kundaje.2</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">query_data</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH107308<span class="pl-pds">"</span></span>]]
<span class="pl-c"><span class="pl-c">#</span>&gt; loading from cache</span>
<span class="pl-smi">excludeGR.hg38.Lareau</span>    <span class="pl-k">&lt;-</span> <span class="pl-smi">query_data</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH107309<span class="pl-pds">"</span></span>]]
<span class="pl-c"><span class="pl-c">#</span>&gt; loading from cache</span>
<span class="pl-smi">excludeGR.hg38.Reddy</span>     <span class="pl-k">&lt;-</span> <span class="pl-smi">query_data</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH107310<span class="pl-pds">"</span></span>]]
<span class="pl-c"><span class="pl-c">#</span>&gt; loading from cache</span>
<span class="pl-smi">excludeGR.hg38.Wimberley</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">query_data</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH107311<span class="pl-pds">"</span></span>]]
<span class="pl-c"><span class="pl-c">#</span>&gt; loading from cache</span>
<span class="pl-smi">excludeGR.hg38.Wold</span>      <span class="pl-k">&lt;-</span> <span class="pl-smi">query_data</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH107312<span class="pl-pds">"</span></span>]]
<span class="pl-c"><span class="pl-c">#</span>&gt; loading from cache</span>
<span class="pl-smi">excludeGR.hg38.Yeo</span>       <span class="pl-k">&lt;-</span> <span class="pl-smi">query_data</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH107313<span class="pl-pds">"</span></span>]]
<span class="pl-c"><span class="pl-c">#</span>&gt; loading from cache</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Compare the number of excludable regions
                     </h2>
                     <a aria-label="Permalink: Compare the number of excludable regions" class="anchor" href="#compare-the-number-of-excludable-regions" id="user-content-compare-the-number-of-excludable-regions">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(ggplot2)
mtx_to_plot &lt;- data.frame(Count = c(length(excludeGR.hg38.Bernstein), 
                                    length(excludeGR.hg38.Boyle),
                                    length(excludeGR.hg38.Kundaje.1), 
                                    length(excludeGR.hg38.Kundaje.2), 
                                    length(excludeGR.hg38.Lareau),
                                    length(excludeGR.hg38.Reddy), 
                                    length(excludeGR.hg38.Wimberley),
                                    length(excludeGR.hg38.Wold), 
                                    length(excludeGR.hg38.Yeo)),
                          Source = c("Bernstein.Mint_Excludable_GRCh38", 
                                     "Boyle.hg38-Excludable.v2",
                                     "Kundaje.GRCh38_unified_Excludable", 
                                     "Kundaje.GRCh38.Excludable", 
                                     "Lareau.hg38.full.Excludable",
                                     "Reddy.wgEncodeDacMapabilityConsensusExcludable", 
                                     "Wimberley.peakPass60Perc_sorted",
                                     "Wold.hg38mitoExcludable", 
                                     "Yeo.eCLIP_Excludableregions.hg38liftover.bed"))
# Order Source by the number of regions
mtx_to_plot$Source &lt;- factor(mtx_to_plot$Source, levels = mtx_to_plot$Source[order(mtx_to_plot$Count)])

ggplot(mtx_to_plot, aes(x = Source, y = Count, fill = Source)) +
  geom_bar(stat = "identity") +
  coord_flip() +
  theme_bw() + theme(legend.position = "none")' dir="auto">
                     <pre>library(<span class="pl-smi">ggplot2</span>)
<span class="pl-smi">mtx_to_plot</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(<span class="pl-v">Count</span> <span class="pl-k">=</span> c(length(<span class="pl-smi">excludeGR.hg38.Bernstein</span>), 
                                    length(<span class="pl-smi">excludeGR.hg38.Boyle</span>),
                                    length(<span class="pl-smi">excludeGR.hg38.Kundaje.1</span>), 
                                    length(<span class="pl-smi">excludeGR.hg38.Kundaje.2</span>), 
                                    length(<span class="pl-smi">excludeGR.hg38.Lareau</span>),
                                    length(<span class="pl-smi">excludeGR.hg38.Reddy</span>), 
                                    length(<span class="pl-smi">excludeGR.hg38.Wimberley</span>),
                                    length(<span class="pl-smi">excludeGR.hg38.Wold</span>), 
                                    length(<span class="pl-smi">excludeGR.hg38.Yeo</span>)),
                          <span class="pl-v">Source</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Bernstein.Mint_Excludable_GRCh38<span class="pl-pds">"</span></span>, 
                                     <span class="pl-s"><span class="pl-pds">"</span>Boyle.hg38-Excludable.v2<span class="pl-pds">"</span></span>,
                                     <span class="pl-s"><span class="pl-pds">"</span>Kundaje.GRCh38_unified_Excludable<span class="pl-pds">"</span></span>, 
                                     <span class="pl-s"><span class="pl-pds">"</span>Kundaje.GRCh38.Excludable<span class="pl-pds">"</span></span>, 
                                     <span class="pl-s"><span class="pl-pds">"</span>Lareau.hg38.full.Excludable<span class="pl-pds">"</span></span>,
                                     <span class="pl-s"><span class="pl-pds">"</span>Reddy.wgEncodeDacMapabilityConsensusExcludable<span class="pl-pds">"</span></span>, 
                                     <span class="pl-s"><span class="pl-pds">"</span>Wimberley.peakPass60Perc_sorted<span class="pl-pds">"</span></span>,
                                     <span class="pl-s"><span class="pl-pds">"</span>Wold.hg38mitoExcludable<span class="pl-pds">"</span></span>, 
                                     <span class="pl-s"><span class="pl-pds">"</span>Yeo.eCLIP_Excludableregions.hg38liftover.bed<span class="pl-pds">"</span></span>))
<span class="pl-c"><span class="pl-c">#</span> Order Source by the number of regions</span>
<span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Source</span> <span class="pl-k">&lt;-</span> <span class="pl-k">factor</span>(<span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Source</span>, <span class="pl-v">levels</span> <span class="pl-k">=</span> <span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Source</span>[order(<span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Count</span>)])

ggplot(<span class="pl-smi">mtx_to_plot</span>, aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">Source</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">Count</span>, <span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-smi">Source</span>)) <span class="pl-k">+</span>
  geom_bar(<span class="pl-v">stat</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>identity<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
  coord_flip() <span class="pl-k">+</span>
  theme_bw() <span class="pl-k">+</span> theme(<span class="pl-v">legend.position</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/dozmorovlab/excluderanges/blob/master/man/figures/README-excluderanges_hg38_count-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/dozmorovlab/excluderanges/raw/master/man/figures/README-excluderanges_hg38_count-1.png" style="max-width: 100%;" width="100%"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# ggsave("man/figures/excluderanges_hg38_count.png", width = 5.5, height = 2)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> ggsave("man/figures/excluderanges_hg38_count.png", width = 5.5, height = 2)</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Compare the width of excludable regions
                     </h2>
                     <a aria-label="Permalink: Compare the width of excludable regions" class="anchor" href="#compare-the-width-of-excludable-regions" id="user-content-compare-the-width-of-excludable-regions">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     log2 scale because of heavy right tail distributions.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(ggridges)
library(dplyr)
mtx_to_plot &lt;- data.frame(Width = c(width(excludeGR.hg38.Bernstein), 
                                    width(excludeGR.hg38.Boyle),
                                    width(excludeGR.hg38.Kundaje.1), 
                                    width(excludeGR.hg38.Kundaje.2), 
                                    width(excludeGR.hg38.Lareau),
                                    width(excludeGR.hg38.Reddy), 
                                    width(excludeGR.hg38.Wimberley),
                                    width(excludeGR.hg38.Wold), 
                                    width(excludeGR.hg38.Yeo)),
                          Source = c(rep("Bernstein.Mint_Excludable_GRCh38", length(excludeGR.hg38.Bernstein)),
                                     rep("Boyle.hg38-Excludable.v2", length(excludeGR.hg38.Boyle)),
                                     rep("Kundaje.GRCh38_unified_Excludable", length(excludeGR.hg38.Kundaje.1)),
                                     rep("Kundaje.GRCh38.Excludable", length(excludeGR.hg38.Kundaje.2)),
                                     rep("Lareau.hg38.full.Excludable", length(excludeGR.hg38.Lareau)),
                                     rep("Reddy.wgEncodeDacMapabilityConsensusExcludable", length(excludeGR.hg38.Reddy)),
                                     rep("Wimberley.peakPass60Perc_sorted", length(excludeGR.hg38.Wimberley)),
                                     rep("Wold.hg38mitoExcludable", length(excludeGR.hg38.Wold)),
                                     rep("Yeo.eCLIP_Excludableregions.hg38liftover.bed", length(excludeGR.hg38.Yeo))))

# Order objects by decreasing width
mtx_to_plot$Source &lt;- factor(mtx_to_plot$Source, levels = mtx_to_plot %&gt;% 
                               group_by(Source) %&gt;% summarise(Mean = mean(Width)) %&gt;% 
                               arrange(Mean) %&gt;% pull(Source))
ggplot(mtx_to_plot, aes(x = log2(Width), y = Source, fill = Source)) +
  geom_density_ridges() +
  theme_bw() + theme(legend.position = "none")' dir="auto">
                     <pre>library(<span class="pl-smi">ggridges</span>)
library(<span class="pl-smi">dplyr</span>)
<span class="pl-smi">mtx_to_plot</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(<span class="pl-v">Width</span> <span class="pl-k">=</span> c(width(<span class="pl-smi">excludeGR.hg38.Bernstein</span>), 
                                    width(<span class="pl-smi">excludeGR.hg38.Boyle</span>),
                                    width(<span class="pl-smi">excludeGR.hg38.Kundaje.1</span>), 
                                    width(<span class="pl-smi">excludeGR.hg38.Kundaje.2</span>), 
                                    width(<span class="pl-smi">excludeGR.hg38.Lareau</span>),
                                    width(<span class="pl-smi">excludeGR.hg38.Reddy</span>), 
                                    width(<span class="pl-smi">excludeGR.hg38.Wimberley</span>),
                                    width(<span class="pl-smi">excludeGR.hg38.Wold</span>), 
                                    width(<span class="pl-smi">excludeGR.hg38.Yeo</span>)),
                          <span class="pl-v">Source</span> <span class="pl-k">=</span> c(rep(<span class="pl-s"><span class="pl-pds">"</span>Bernstein.Mint_Excludable_GRCh38<span class="pl-pds">"</span></span>, length(<span class="pl-smi">excludeGR.hg38.Bernstein</span>)),
                                     rep(<span class="pl-s"><span class="pl-pds">"</span>Boyle.hg38-Excludable.v2<span class="pl-pds">"</span></span>, length(<span class="pl-smi">excludeGR.hg38.Boyle</span>)),
                                     rep(<span class="pl-s"><span class="pl-pds">"</span>Kundaje.GRCh38_unified_Excludable<span class="pl-pds">"</span></span>, length(<span class="pl-smi">excludeGR.hg38.Kundaje.1</span>)),
                                     rep(<span class="pl-s"><span class="pl-pds">"</span>Kundaje.GRCh38.Excludable<span class="pl-pds">"</span></span>, length(<span class="pl-smi">excludeGR.hg38.Kundaje.2</span>)),
                                     rep(<span class="pl-s"><span class="pl-pds">"</span>Lareau.hg38.full.Excludable<span class="pl-pds">"</span></span>, length(<span class="pl-smi">excludeGR.hg38.Lareau</span>)),
                                     rep(<span class="pl-s"><span class="pl-pds">"</span>Reddy.wgEncodeDacMapabilityConsensusExcludable<span class="pl-pds">"</span></span>, length(<span class="pl-smi">excludeGR.hg38.Reddy</span>)),
                                     rep(<span class="pl-s"><span class="pl-pds">"</span>Wimberley.peakPass60Perc_sorted<span class="pl-pds">"</span></span>, length(<span class="pl-smi">excludeGR.hg38.Wimberley</span>)),
                                     rep(<span class="pl-s"><span class="pl-pds">"</span>Wold.hg38mitoExcludable<span class="pl-pds">"</span></span>, length(<span class="pl-smi">excludeGR.hg38.Wold</span>)),
                                     rep(<span class="pl-s"><span class="pl-pds">"</span>Yeo.eCLIP_Excludableregions.hg38liftover.bed<span class="pl-pds">"</span></span>, length(<span class="pl-smi">excludeGR.hg38.Yeo</span>))))

<span class="pl-c"><span class="pl-c">#</span> Order objects by decreasing width</span>
<span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Source</span> <span class="pl-k">&lt;-</span> <span class="pl-k">factor</span>(<span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Source</span>, <span class="pl-v">levels</span> <span class="pl-k">=</span> <span class="pl-smi">mtx_to_plot</span> %<span class="pl-k">&gt;</span>% 
                               group_by(<span class="pl-smi">Source</span>) %<span class="pl-k">&gt;</span>% summarise(<span class="pl-v">Mean</span> <span class="pl-k">=</span> mean(<span class="pl-smi">Width</span>)) %<span class="pl-k">&gt;</span>% 
                               arrange(<span class="pl-smi">Mean</span>) %<span class="pl-k">&gt;</span>% pull(<span class="pl-smi">Source</span>))
ggplot(<span class="pl-smi">mtx_to_plot</span>, aes(<span class="pl-v">x</span> <span class="pl-k">=</span> log2(<span class="pl-smi">Width</span>), <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">Source</span>, <span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-smi">Source</span>)) <span class="pl-k">+</span>
  geom_density_ridges() <span class="pl-k">+</span>
  theme_bw() <span class="pl-k">+</span> theme(<span class="pl-v">legend.position</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/dozmorovlab/excluderanges/blob/master/man/figures/README-excluderanges_hg38_width-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/dozmorovlab/excluderanges/raw/master/man/figures/README-excluderanges_hg38_width-1.png" style="max-width: 100%;" width="100%"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# ggsave("man/figures/excluderanges_hg38_width.png", width = 5.5, height = 2)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> ggsave("man/figures/excluderanges_hg38_width.png", width = 5.5, height = 2)</span></pre>
                    </div>
                    <p dir="auto">
                     We can investigate the total width of each set of excludable ranges.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='mtx_to_plot &lt;- data.frame(TotalWidth = c(sum(width(excludeGR.hg38.Bernstein)), 
                                         sum(width(excludeGR.hg38.Boyle)),
                                         sum(width(excludeGR.hg38.Kundaje.1)), 
                                         sum(width(excludeGR.hg38.Kundaje.2)), 
                                         sum(width(excludeGR.hg38.Lareau)),
                                         sum(width(excludeGR.hg38.Reddy)), 
                                         sum(width(excludeGR.hg38.Wimberley)),
                                         sum(width(excludeGR.hg38.Wold)), 
                                         sum(width(excludeGR.hg38.Yeo))), 
                          Source = c("Bernstein.Mint_Excludable_GRCh38", 
                                     "Boyle.hg38-Excludable.v2",
                                     "Kundaje.GRCh38_unified_Excludable", 
                                     "Kundaje.GRCh38.Excludable", 
                                     "Lareau.hg38.full.Excludable",
                                     "Reddy.wgEncodeDacMapabilityConsensusExcludable", 
                                     "Wimberley.peakPass60Perc_sorted",
                                     "Wold.hg38mitoExcludable", 
                                     "Yeo.eCLIP_Excludableregions.hg38liftover.bed"))
# Order objects by decreasing width
mtx_to_plot$Source &lt;- factor(mtx_to_plot$Source, levels = mtx_to_plot %&gt;% 
                               group_by(Source) %&gt;% arrange(TotalWidth) %&gt;% pull(Source))

ggplot(mtx_to_plot, aes(x = TotalWidth, y = Source, fill = Source)) + 
  geom_bar(stat="identity") + scale_x_log10() + scale_y_discrete(label=abbreviate, limits=rev) +
  xlab("log10 total width")' dir="auto">
                     <pre><span class="pl-smi">mtx_to_plot</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(<span class="pl-v">TotalWidth</span> <span class="pl-k">=</span> c(sum(width(<span class="pl-smi">excludeGR.hg38.Bernstein</span>)), 
                                         sum(width(<span class="pl-smi">excludeGR.hg38.Boyle</span>)),
                                         sum(width(<span class="pl-smi">excludeGR.hg38.Kundaje.1</span>)), 
                                         sum(width(<span class="pl-smi">excludeGR.hg38.Kundaje.2</span>)), 
                                         sum(width(<span class="pl-smi">excludeGR.hg38.Lareau</span>)),
                                         sum(width(<span class="pl-smi">excludeGR.hg38.Reddy</span>)), 
                                         sum(width(<span class="pl-smi">excludeGR.hg38.Wimberley</span>)),
                                         sum(width(<span class="pl-smi">excludeGR.hg38.Wold</span>)), 
                                         sum(width(<span class="pl-smi">excludeGR.hg38.Yeo</span>))), 
                          <span class="pl-v">Source</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Bernstein.Mint_Excludable_GRCh38<span class="pl-pds">"</span></span>, 
                                     <span class="pl-s"><span class="pl-pds">"</span>Boyle.hg38-Excludable.v2<span class="pl-pds">"</span></span>,
                                     <span class="pl-s"><span class="pl-pds">"</span>Kundaje.GRCh38_unified_Excludable<span class="pl-pds">"</span></span>, 
                                     <span class="pl-s"><span class="pl-pds">"</span>Kundaje.GRCh38.Excludable<span class="pl-pds">"</span></span>, 
                                     <span class="pl-s"><span class="pl-pds">"</span>Lareau.hg38.full.Excludable<span class="pl-pds">"</span></span>,
                                     <span class="pl-s"><span class="pl-pds">"</span>Reddy.wgEncodeDacMapabilityConsensusExcludable<span class="pl-pds">"</span></span>, 
                                     <span class="pl-s"><span class="pl-pds">"</span>Wimberley.peakPass60Perc_sorted<span class="pl-pds">"</span></span>,
                                     <span class="pl-s"><span class="pl-pds">"</span>Wold.hg38mitoExcludable<span class="pl-pds">"</span></span>, 
                                     <span class="pl-s"><span class="pl-pds">"</span>Yeo.eCLIP_Excludableregions.hg38liftover.bed<span class="pl-pds">"</span></span>))
<span class="pl-c"><span class="pl-c">#</span> Order objects by decreasing width</span>
<span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Source</span> <span class="pl-k">&lt;-</span> <span class="pl-k">factor</span>(<span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Source</span>, <span class="pl-v">levels</span> <span class="pl-k">=</span> <span class="pl-smi">mtx_to_plot</span> %<span class="pl-k">&gt;</span>% 
                               group_by(<span class="pl-smi">Source</span>) %<span class="pl-k">&gt;</span>% arrange(<span class="pl-smi">TotalWidth</span>) %<span class="pl-k">&gt;</span>% pull(<span class="pl-smi">Source</span>))

ggplot(<span class="pl-smi">mtx_to_plot</span>, aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">TotalWidth</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">Source</span>, <span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-smi">Source</span>)) <span class="pl-k">+</span> 
  geom_bar(<span class="pl-v">stat</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>identity<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> scale_x_log10() <span class="pl-k">+</span> scale_y_discrete(<span class="pl-v">label</span><span class="pl-k">=</span><span class="pl-smi">abbreviate</span>, <span class="pl-v">limits</span><span class="pl-k">=</span><span class="pl-smi">rev</span>) <span class="pl-k">+</span>
  xlab(<span class="pl-s"><span class="pl-pds">"</span>log10 total width<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/dozmorovlab/excluderanges/blob/master/man/figures/README-excluderanges_hg38_sumwidth-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/dozmorovlab/excluderanges/raw/master/man/figures/README-excluderanges_hg38_sumwidth-1.png" style="max-width: 100%;" width="100%"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# ggsave("man/figures/excluderanges_hg38_sumwidth.png", width = 6.5, height = 2)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> ggsave("man/figures/excluderanges_hg38_sumwidth.png", width = 6.5, height = 2)</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Compare overlaps among sets
                     </h2>
                     <a aria-label="Permalink: Compare overlaps among sets" class="anchor" href="#compare-overlaps-among-sets" id="user-content-compare-overlaps-among-sets">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We can compare
                     <a href="https://en.wikipedia.org/wiki/Overlap_coefficient" rel="nofollow">
                      overlap
coefficients
                     </a>
                     between
those sets of excludable regions.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(pheatmap)
library(stringr)
# Overlap coefficient calculations
overlap_coefficient &lt;- function(gr_a, gr_b) {
  intersects &lt;- GenomicRanges::intersect(gr_a, gr_b, ignore.strand = TRUE)
  intersection_width &lt;- sum(width(intersects))
  min_width &lt;- min(sum(width(gr_a)), sum(width(gr_b)))
  DataFrame(intersection_width, min_width, 
            overlap_coefficient = intersection_width/min_width,
             n_intersections = length(intersects))
}
# List and names of all excludable regions
all_excludeGR_list &lt;- list(excludeGR.hg38.Bernstein, 
                            excludeGR.hg38.Boyle,
                            excludeGR.hg38.Kundaje.1, 
                            excludeGR.hg38.Kundaje.2,
                            excludeGR.hg38.Lareau,
                            excludeGR.hg38.Reddy,
                            excludeGR.hg38.Wimberley,
                            excludeGR.hg38.Wold,
                            excludeGR.hg38.Yeo)
all_excludeGR_name &lt;- c("Bernstein.Mint_Excludable_GRCh38", 
                         "Boyle.hg38-Excludable.v2",
                         "Kundaje.GRCh38_unified_Excludable", 
                         "Kundaje.GRCh38.Excludable", 
                         "Lareau.hg38.full.Excludable",
                         "Reddy.wgEncodeDacMapabilityConsensusExcludable", 
                         "Wimberley.peakPass60Perc_sorted",
                         "Wold.hg38mitoExcludable", 
                         "Yeo.eCLIP_Excludableregions.hg38liftover.bed")
# Correlation matrix, empty
mtx_to_plot &lt;- matrix(data = 0, nrow = length(all_excludeGR_list), ncol = length(all_excludeGR_list))
# Fill it in
for (i in 1:length(all_excludeGR_list)) {
  for (j in 1:length(all_excludeGR_list)) {
    # If diagonal, set to zero
    if (i == j) mtx_to_plot[i, j] &lt;- 0
    # Process only one half, the other is symmetric
    if (i &gt; j) {
      mtx_to_plot[i, j] &lt;- mtx_to_plot[j, i] &lt;- overlap_coefficient(all_excludeGR_list[[i]], all_excludeGR_list[[j]])[["overlap_coefficient"]]
    }
  }
}
# Trim row/colnames
rownames(mtx_to_plot) &lt;- colnames(mtx_to_plot) &lt;- str_trunc(all_excludeGR_name, width = 25) 
# Save the plot
# png("man/figures/excluderanges_hg38_jaccard.png", width = 1000, height = 900, res = 200)
pheatmap(data.matrix(mtx_to_plot), clustering_method = "ward.D")' dir="auto">
                     <pre>library(<span class="pl-smi">pheatmap</span>)
library(<span class="pl-smi">stringr</span>)
<span class="pl-c"><span class="pl-c">#</span> Overlap coefficient calculations</span>
<span class="pl-en">overlap_coefficient</span> <span class="pl-k">&lt;-</span> <span class="pl-k">function</span>(<span class="pl-smi">gr_a</span>, <span class="pl-smi">gr_b</span>) {
  <span class="pl-smi">intersects</span> <span class="pl-k">&lt;-</span> <span class="pl-e">GenomicRanges</span><span class="pl-k">::</span>intersect(<span class="pl-smi">gr_a</span>, <span class="pl-smi">gr_b</span>, <span class="pl-v">ignore.strand</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
  <span class="pl-smi">intersection_width</span> <span class="pl-k">&lt;-</span> sum(width(<span class="pl-smi">intersects</span>))
  <span class="pl-smi">min_width</span> <span class="pl-k">&lt;-</span> min(sum(width(<span class="pl-smi">gr_a</span>)), sum(width(<span class="pl-smi">gr_b</span>)))
  DataFrame(<span class="pl-smi">intersection_width</span>, <span class="pl-smi">min_width</span>, 
            <span class="pl-v">overlap_coefficient</span> <span class="pl-k">=</span> <span class="pl-smi">intersection_width</span><span class="pl-k">/</span><span class="pl-smi">min_width</span>,
             <span class="pl-v">n_intersections</span> <span class="pl-k">=</span> length(<span class="pl-smi">intersects</span>))
}
<span class="pl-c"><span class="pl-c">#</span> List and names of all excludable regions</span>
<span class="pl-smi">all_excludeGR_list</span> <span class="pl-k">&lt;-</span> <span class="pl-k">list</span>(<span class="pl-smi">excludeGR.hg38.Bernstein</span>, 
                            <span class="pl-smi">excludeGR.hg38.Boyle</span>,
                            <span class="pl-smi">excludeGR.hg38.Kundaje.1</span>, 
                            <span class="pl-smi">excludeGR.hg38.Kundaje.2</span>,
                            <span class="pl-smi">excludeGR.hg38.Lareau</span>,
                            <span class="pl-smi">excludeGR.hg38.Reddy</span>,
                            <span class="pl-smi">excludeGR.hg38.Wimberley</span>,
                            <span class="pl-smi">excludeGR.hg38.Wold</span>,
                            <span class="pl-smi">excludeGR.hg38.Yeo</span>)
<span class="pl-smi">all_excludeGR_name</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>Bernstein.Mint_Excludable_GRCh38<span class="pl-pds">"</span></span>, 
                         <span class="pl-s"><span class="pl-pds">"</span>Boyle.hg38-Excludable.v2<span class="pl-pds">"</span></span>,
                         <span class="pl-s"><span class="pl-pds">"</span>Kundaje.GRCh38_unified_Excludable<span class="pl-pds">"</span></span>, 
                         <span class="pl-s"><span class="pl-pds">"</span>Kundaje.GRCh38.Excludable<span class="pl-pds">"</span></span>, 
                         <span class="pl-s"><span class="pl-pds">"</span>Lareau.hg38.full.Excludable<span class="pl-pds">"</span></span>,
                         <span class="pl-s"><span class="pl-pds">"</span>Reddy.wgEncodeDacMapabilityConsensusExcludable<span class="pl-pds">"</span></span>, 
                         <span class="pl-s"><span class="pl-pds">"</span>Wimberley.peakPass60Perc_sorted<span class="pl-pds">"</span></span>,
                         <span class="pl-s"><span class="pl-pds">"</span>Wold.hg38mitoExcludable<span class="pl-pds">"</span></span>, 
                         <span class="pl-s"><span class="pl-pds">"</span>Yeo.eCLIP_Excludableregions.hg38liftover.bed<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> Correlation matrix, empty</span>
<span class="pl-smi">mtx_to_plot</span> <span class="pl-k">&lt;-</span> <span class="pl-k">matrix</span>(<span class="pl-v">data</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>, <span class="pl-v">nrow</span> <span class="pl-k">=</span> length(<span class="pl-smi">all_excludeGR_list</span>), <span class="pl-v">ncol</span> <span class="pl-k">=</span> length(<span class="pl-smi">all_excludeGR_list</span>))
<span class="pl-c"><span class="pl-c">#</span> Fill it in</span>
<span class="pl-k">for</span> (<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>length(<span class="pl-smi">all_excludeGR_list</span>)) {
  <span class="pl-k">for</span> (<span class="pl-smi">j</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>length(<span class="pl-smi">all_excludeGR_list</span>)) {
    <span class="pl-c"><span class="pl-c">#</span> If diagonal, set to zero</span>
    <span class="pl-k">if</span> (<span class="pl-smi">i</span> <span class="pl-k">==</span> <span class="pl-smi">j</span>) <span class="pl-smi">mtx_to_plot</span>[<span class="pl-smi">i</span>, <span class="pl-smi">j</span>] <span class="pl-k">&lt;-</span> <span class="pl-c1">0</span>
    <span class="pl-c"><span class="pl-c">#</span> Process only one half, the other is symmetric</span>
    <span class="pl-k">if</span> (<span class="pl-smi">i</span> <span class="pl-k">&gt;</span> <span class="pl-smi">j</span>) {
      <span class="pl-smi">mtx_to_plot</span>[<span class="pl-smi">i</span>, <span class="pl-smi">j</span>] <span class="pl-k">&lt;-</span> <span class="pl-smi">mtx_to_plot</span>[<span class="pl-smi">j</span>, <span class="pl-smi">i</span>] <span class="pl-k">&lt;-</span> overlap_coefficient(<span class="pl-smi">all_excludeGR_list</span>[[<span class="pl-smi">i</span>]], <span class="pl-smi">all_excludeGR_list</span>[[<span class="pl-smi">j</span>]])[[<span class="pl-s"><span class="pl-pds">"</span>overlap_coefficient<span class="pl-pds">"</span></span>]]
    }
  }
}
<span class="pl-c"><span class="pl-c">#</span> Trim row/colnames</span>
rownames(<span class="pl-smi">mtx_to_plot</span>) <span class="pl-k">&lt;-</span> colnames(<span class="pl-smi">mtx_to_plot</span>) <span class="pl-k">&lt;-</span> str_trunc(<span class="pl-smi">all_excludeGR_name</span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">25</span>) 
<span class="pl-c"><span class="pl-c">#</span> Save the plot</span>
<span class="pl-c"><span class="pl-c">#</span> png("man/figures/excluderanges_hg38_jaccard.png", width = 1000, height = 900, res = 200)</span>
pheatmap(data.matrix(<span class="pl-smi">mtx_to_plot</span>), <span class="pl-v">clustering_method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/dozmorovlab/excluderanges/blob/master/man/figures/README-excluderanges_hg38_overlap_coefficient-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/dozmorovlab/excluderanges/raw/master/man/figures/README-excluderanges_hg38_overlap_coefficient-1.png" style="max-width: 100%;" width="100%"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# dev.off()" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> dev.off()</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Metadata analysis
                     </h2>
                     <a aria-label="Permalink: Metadata analysis" class="anchor" href="#metadata-analysis" id="user-content-metadata-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Note that some excludable ranges objects contain six columns, implying
there may be some interesting metadata. Let’s explore one.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='mcols(excludeGR.hg38.Reddy)
#&gt; DataFrame with 396 rows and 2 columns
#&gt;                       name     score
#&gt;                &lt;character&gt; &lt;numeric&gt;
#&gt; 1   High_Mappability_isl..      1000
#&gt; 2         Satellite_repeat      1000
#&gt; 3                 BSR/Beta      1000
#&gt; 4   Low_mappability_island      1000
#&gt; 5                 (CATTC)n      1000
#&gt; ...                    ...       ...
#&gt; 392       Satellite_repeat      1000
#&gt; 393               (CATTC)n      1000
#&gt; 394               (CATTC)n      1000
#&gt; 395                   TAR1      1000
#&gt; 396                   chrM      1000
mtx_to_plot &lt;- table(mcols(excludeGR.hg38.Reddy)[["name"]]) %&gt;%
  as.data.frame()
colnames(mtx_to_plot) &lt;- c("Type", "Number")
mtx_to_plot &lt;- mtx_to_plot[order(mtx_to_plot$Number), ]
mtx_to_plot$Type &lt;- factor(mtx_to_plot$Type, 
                           levels = mtx_to_plot$Type)
ggplot(mtx_to_plot, aes(x = Number, y = Type, fill = Type)) +
  geom_bar(stat="identity") +
  theme_bw() + theme(legend.position = "none")' dir="auto">
                     <pre>mcols(<span class="pl-smi">excludeGR.hg38.Reddy</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; DataFrame with 396 rows and 2 columns</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;                       name     score</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;                &lt;character&gt; &lt;numeric&gt;</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; 1   High_Mappability_isl..      1000</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; 2         Satellite_repeat      1000</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; 3                 BSR/Beta      1000</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; 4   Low_mappability_island      1000</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; 5                 (CATTC)n      1000</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; ...                    ...       ...</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; 392       Satellite_repeat      1000</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; 393               (CATTC)n      1000</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; 394               (CATTC)n      1000</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; 395                   TAR1      1000</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; 396                   chrM      1000</span>
<span class="pl-smi">mtx_to_plot</span> <span class="pl-k">&lt;-</span> table(mcols(<span class="pl-smi">excludeGR.hg38.Reddy</span>)[[<span class="pl-s"><span class="pl-pds">"</span>name<span class="pl-pds">"</span></span>]]) %<span class="pl-k">&gt;</span>%
  as.data.frame()
colnames(<span class="pl-smi">mtx_to_plot</span>) <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>Type<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Number<span class="pl-pds">"</span></span>)
<span class="pl-smi">mtx_to_plot</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">mtx_to_plot</span>[order(<span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Number</span>), ]
<span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Type</span> <span class="pl-k">&lt;-</span> <span class="pl-k">factor</span>(<span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Type</span>, 
                           <span class="pl-v">levels</span> <span class="pl-k">=</span> <span class="pl-smi">mtx_to_plot</span><span class="pl-k">$</span><span class="pl-smi">Type</span>)
ggplot(<span class="pl-smi">mtx_to_plot</span>, aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">Number</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">Type</span>, <span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-smi">Type</span>)) <span class="pl-k">+</span>
  geom_bar(<span class="pl-v">stat</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>identity<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
  theme_bw() <span class="pl-k">+</span> theme(<span class="pl-v">legend.position</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/dozmorovlab/excluderanges/blob/master/man/figures/README-excluderanges_hg38_Reddy_metadata-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/dozmorovlab/excluderanges/raw/master/man/figures/README-excluderanges_hg38_Reddy_metadata-1.png" style="max-width: 100%;" width="100%"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# ggsave("man/figures/excluderanges_hg38_Reddy_metadata.png", width = 5, height = 2.5)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> ggsave("man/figures/excluderanges_hg38_Reddy_metadata.png", width = 5, height = 2.5)</span></pre>
                    </div>
                    <p dir="auto">
                     One may decide to combine the excludable ranges from all labs, although
from previous results we may decide to follow Anshul’s
                     <a href="https://twitter.com/anshulkundaje/status/1263546023151992832?s=20" rel="nofollow">
                      advice
                     </a>
                     about the
                     <a href="https://www.encodeproject.org/files/ENCFF356LFX/" rel="nofollow">
                      ENCFF356LFX exclusion list
regions
                     </a>
                     and use the
                     <code>
                      excludeGR.hg38.Kundaje.1
                     </code>
                     object.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='excludeGR.hg38.all &lt;- reduce(c(excludeGR.hg38.Bernstein, 
                               excludeGR.hg38.Boyle,
                               excludeGR.hg38.Kundaje.1, 
                               excludeGR.hg38.Kundaje.2, 
                               excludeGR.hg38.Lareau,
                               excludeGR.hg38.Reddy, 
                               excludeGR.hg38.Wimberley,
                               excludeGR.hg38.Wold, 
                               excludeGR.hg38.Yeo))
# Sort and Keep only standard chromosomes
excludeGR.hg38.all &lt;- excludeGR.hg38.all %&gt;% sort %&gt;% 
  keepStandardChromosomes(pruning.mode = "tidy")
print(length(excludeGR.hg38.all))
#&gt; [1] 15998
summary(width(excludeGR.hg38.all))
#&gt;     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
#&gt;        5     1703     2294    17713     3001 30739401' dir="auto">
                     <pre><span class="pl-smi">excludeGR.hg38.all</span> <span class="pl-k">&lt;-</span> reduce(c(<span class="pl-smi">excludeGR.hg38.Bernstein</span>, 
                               <span class="pl-smi">excludeGR.hg38.Boyle</span>,
                               <span class="pl-smi">excludeGR.hg38.Kundaje.1</span>, 
                               <span class="pl-smi">excludeGR.hg38.Kundaje.2</span>, 
                               <span class="pl-smi">excludeGR.hg38.Lareau</span>,
                               <span class="pl-smi">excludeGR.hg38.Reddy</span>, 
                               <span class="pl-smi">excludeGR.hg38.Wimberley</span>,
                               <span class="pl-smi">excludeGR.hg38.Wold</span>, 
                               <span class="pl-smi">excludeGR.hg38.Yeo</span>))
<span class="pl-c"><span class="pl-c">#</span> Sort and Keep only standard chromosomes</span>
<span class="pl-smi">excludeGR.hg38.all</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">excludeGR.hg38.all</span> %<span class="pl-k">&gt;</span>% <span class="pl-smi">sort</span> %<span class="pl-k">&gt;</span>% 
  keepStandardChromosomes(<span class="pl-v">pruning.mode</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>tidy<span class="pl-pds">"</span></span>)
print(length(<span class="pl-smi">excludeGR.hg38.all</span>))
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] 15998</span>
summary(width(<span class="pl-smi">excludeGR.hg38.all</span>))
<span class="pl-c"><span class="pl-c">#</span>&gt;     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;        5     1703     2294    17713     3001 30739401</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      BEDbase data download
                     </h1>
                     <a aria-label="Permalink: BEDbase data download" class="anchor" href="#bedbase-data-download" id="user-content-bedbase-data-download">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="http://bedbase.org/" rel="nofollow">
                      BEDbase.org
                     </a>
                     is a repository for storing and
analyzing BED files and BED sets, developed by
                     <a href="https://databio.org/" rel="nofollow">
                      Sheffield
lab
                     </a>
                     . It provides API for data access and
retrieval.
                    </p>
                    <p dir="auto">
                     Using BEDbase ID (e.g.,
                     <code>
                      1a561729234c2844303a051b16f66656
                     </code>
                     for the
                     <code>
                      hg38.Kundaje.GRCh38_unified_Excludable
                     </code>
                     excluderanges object), we can
construct a URL showing the splash screen with detailed information of
the corresponding object,
                     <a href="http://bedbase.org/#/bedsplash/1a561729234c2844303a051b16f66656" rel="nofollow">
                      http://bedbase.org/#/bedsplash/1a561729234c2844303a051b16f66656
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     We can also get the BED file using the
                     <a href="http://bedbase.org/docs" rel="nofollow">
                      BEDbase
API
                     </a>
                     .
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="wget http://bedbase.org/api/bed/1a561729234c2844303a051b16f66656/file/bed \
  -O hg38.Kundaje.GRCh38_unified_Excludable.bed.gz " dir="auto">
                     <pre>wget http://bedbase.org/api/bed/1a561729234c2844303a051b16f66656/file/bed \
  -O hg38.Kundaje.GRCh38_unified_Excludable.bed.gz </pre>
                    </div>
                    <p dir="auto">
                     See also the R code to download
                     <a href="https://gist.github.com/mdozmorov/7f4393f90932fb6bd911c43c20425ca0">
                      T2T.excluderanges
                     </a>
                     and
                     <a href="https://gist.github.com/mdozmorov/33a1fa3234b2942dae238e1fcb39c996">
                      mm39.excluderanges
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Mitochondrial DNA sequences, NUMTs
                     </h1>
                     <a aria-label="Permalink: Mitochondrial DNA sequences, NUMTs" class="anchor" href="#mitochondrial-dna-sequences-numts" id="user-content-mitochondrial-dna-sequences-numts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Mitochondrial DNA sequences (mtDNA, 100-600K mitochondria per human
cell) transferred to the nucleus give rise to the so-called
mitochondrial DNA sequences in the nuclear genome (NUMTs). In the
settings of DNA/chromatin sequencing (e.g., ATAC-seq), we may observe up
to 80% of mitochondrial sequencing reads that may pile up in the NUMT
sequences. Similar to exclusion sets, genomic regions highly homologous
to mtDNA can be masked to improve biological signal.
                    </p>
                    <p dir="auto">
                     The reference human nuclear mitochondrial sequences have been available
in the UCSC genome browser for hg19 and mm8 human/mouse genome
assemblies. We collected NUMT sets for hg38, T2T-CHM13, mm10, generated
by Caleb Lareau in the
                     <a href="https://github.com/caleblareau/mitoblacklist">
                      mitoblacklist
                     </a>
                     GitHub
repository. These NUMT sets can be combined with exclusion sets.
                    </p>
                    <p dir="auto">
                     Example of the
                     <code>
                      hg38.Lareau.hg38_peaks
                     </code>
                     object
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='suppressMessages(library(httr))
# Get hg38.Lareau.hg38_peaks BEDbase ID
bedbase_id &lt;- "9fa55701a3bd3e7a598d1d2815e3390f"
# Construct output file name
fileNameOut &lt;- "hg38.Lareau.hg38_peak.bed.gz"
# API token for BED data
token2 &lt;- paste0("http://bedbase.org/api/bed/", bedbase_id, "/file/bed")
# Download file
GET(url = token2, write_disk(fileNameOut, overwrite = TRUE)) # , verbose()
#&gt; Response [http://data.bedbase.org/bed_files/hg38.Lareau.hg38_peaks.bed.gz]
#&gt;   Date: 2023-12-04 18:53
#&gt;   Status: 200
#&gt;   Content-Type: application/vnd.realvnc.bed
#&gt;   Size: 11.8 kB
#&gt; &lt;ON DISK&gt;  /Users/mdozmorov/Documents/Work/GitHub/excluderanges/hg38.Lareau.hg38_peak.bed.gz
# Read the data in
hg38.Lareau.hg38_peaks &lt;- read.table(fileNameOut, sep = "\t", header = FALSE)
# Assign column names depending on the number of columns
all_columns &lt;- c("chr", "start", "stop", "name", "score", "strand", 
                 "signalValue", "pValue", "qValue", "peak")
colnames(hg38.Lareau.hg38_peaks) &lt;- all_columns[1:ncol(hg38.Lareau.hg38_peaks)]
# Convert to GRanges object
hg38.Lareau.hg38_peaks &lt;- makeGRangesFromDataFrame(hg38.Lareau.hg38_peaks, 
                                                   keep.extra.columns = TRUE)
hg38.Lareau.hg38_peaks
#&gt; GRanges object with 784 ranges and 2 metadata columns:
#&gt;         seqnames              ranges strand |                name     score
#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |         &lt;character&gt; &lt;integer&gt;
#&gt;     [1]     chr1       628903-635104      * |   peaks/hg38_peak_1   2523899
#&gt;     [2]     chr1     5850087-5850571      * |   peaks/hg38_peak_2     32940
#&gt;     [3]     chr1     8909610-8910014      * |   peaks/hg38_peak_3       123
#&gt;     [4]     chr1     9574580-9574997      * |   peaks/hg38_peak_4     28828
#&gt;     [5]     chr1   32043823-32044203      * |   peaks/hg38_peak_5      8083
#&gt;     ...      ...                 ...    ... .                 ...       ...
#&gt;   [780]     chrX 143431716-143432219      * | peaks/hg38_peak_780     41434
#&gt;   [781]     chrX 143432410-143433212      * | peaks/hg38_peak_781      9271
#&gt;   [782]     chrX 143433510-143434156      * | peaks/hg38_peak_782      5741
#&gt;   [783]     chrX 143543636-143544023      * | peaks/hg38_peak_783      3320
#&gt;   [784]     chrX 146995842-146996224      * | peaks/hg38_peak_784     27222
#&gt;   -------
#&gt;   seqinfo: 23 sequences from an unspecified genome; no seqlengths' dir="auto">
                     <pre>suppressMessages(library(<span class="pl-smi">httr</span>))
<span class="pl-c"><span class="pl-c">#</span> Get hg38.Lareau.hg38_peaks BEDbase ID</span>
<span class="pl-smi">bedbase_id</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>9fa55701a3bd3e7a598d1d2815e3390f<span class="pl-pds">"</span></span>
<span class="pl-c"><span class="pl-c">#</span> Construct output file name</span>
<span class="pl-smi">fileNameOut</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>hg38.Lareau.hg38_peak.bed.gz<span class="pl-pds">"</span></span>
<span class="pl-c"><span class="pl-c">#</span> API token for BED data</span>
<span class="pl-smi">token2</span> <span class="pl-k">&lt;-</span> paste0(<span class="pl-s"><span class="pl-pds">"</span>http://bedbase.org/api/bed/<span class="pl-pds">"</span></span>, <span class="pl-smi">bedbase_id</span>, <span class="pl-s"><span class="pl-pds">"</span>/file/bed<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> Download file</span>
GET(<span class="pl-v">url</span> <span class="pl-k">=</span> <span class="pl-smi">token2</span>, write_disk(<span class="pl-smi">fileNameOut</span>, <span class="pl-v">overwrite</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)) <span class="pl-c"><span class="pl-c">#</span> , verbose()</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; Response [http://data.bedbase.org/bed_files/hg38.Lareau.hg38_peaks.bed.gz]</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   Date: 2023-12-04 18:53</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   Status: 200</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   Content-Type: application/vnd.realvnc.bed</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   Size: 11.8 kB</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; &lt;ON DISK&gt;  /Users/mdozmorov/Documents/Work/GitHub/excluderanges/hg38.Lareau.hg38_peak.bed.gz</span>
<span class="pl-c"><span class="pl-c">#</span> Read the data in</span>
<span class="pl-smi">hg38.Lareau.hg38_peaks</span> <span class="pl-k">&lt;-</span> read.table(<span class="pl-smi">fileNameOut</span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\t</span><span class="pl-pds">"</span></span>, <span class="pl-v">header</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-c"><span class="pl-c">#</span> Assign column names depending on the number of columns</span>
<span class="pl-smi">all_columns</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>chr<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>start<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>stop<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>name<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>score<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>strand<span class="pl-pds">"</span></span>, 
                 <span class="pl-s"><span class="pl-pds">"</span>signalValue<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>pValue<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>qValue<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>peak<span class="pl-pds">"</span></span>)
colnames(<span class="pl-smi">hg38.Lareau.hg38_peaks</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">all_columns</span>[<span class="pl-c1">1</span><span class="pl-k">:</span>ncol(<span class="pl-smi">hg38.Lareau.hg38_peaks</span>)]
<span class="pl-c"><span class="pl-c">#</span> Convert to GRanges object</span>
<span class="pl-smi">hg38.Lareau.hg38_peaks</span> <span class="pl-k">&lt;-</span> makeGRangesFromDataFrame(<span class="pl-smi">hg38.Lareau.hg38_peaks</span>, 
                                                   <span class="pl-v">keep.extra.columns</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
<span class="pl-smi">hg38.Lareau.hg38_peaks</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; GRanges object with 784 ranges and 2 metadata columns:</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;         seqnames              ranges strand |                name     score</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |         &lt;character&gt; &lt;integer&gt;</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [1]     chr1       628903-635104      * |   peaks/hg38_peak_1   2523899</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [2]     chr1     5850087-5850571      * |   peaks/hg38_peak_2     32940</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [3]     chr1     8909610-8910014      * |   peaks/hg38_peak_3       123</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [4]     chr1     9574580-9574997      * |   peaks/hg38_peak_4     28828</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [5]     chr1   32043823-32044203      * |   peaks/hg38_peak_5      8083</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     ...      ...                 ...    ... .                 ...       ...</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [780]     chrX 143431716-143432219      * | peaks/hg38_peak_780     41434</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [781]     chrX 143432410-143433212      * | peaks/hg38_peak_781      9271</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [782]     chrX 143433510-143434156      * | peaks/hg38_peak_782      5741</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [783]     chrX 143543636-143544023      * | peaks/hg38_peak_783      3320</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [784]     chrX 146995842-146996224      * | peaks/hg38_peak_784     27222</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   -------</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   seqinfo: 23 sequences from an unspecified genome; no seqlengths</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Centromeres, telomeres, etc.
                     </h1>
                     <a aria-label="Permalink: Centromeres, telomeres, etc." class="anchor" href="#centromeres-telomeres-etc" id="user-content-centromeres-telomeres-etc">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Besides the ENCODE-produced excludable regions, we may want to exclude
centromeres, telomeres, and other gap locations. The “Gap Locations”
track for Homo Sapiens is available for the GRcH37/hg19 genome assembly
as a
                     <a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" rel="nofollow">
                      UCSC ‘gap’
table
                     </a>
                     .
It can be retrieved from
                     <em>
                      <a href="https://bioconductor.org/packages/3.18/AnnotationHub" rel="nofollow">
                       AnnotationHub
                      </a>
                     </em>
                     ,
but lacks the metadata columns needed to decide the type of gaps.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Search for the gap track
# ahData &lt;- query(ah, c("gap", "Homo sapiens", "hg19"))
# ahData[ahData$title == "Gap"]
gaps &lt;- ahData[["AH6444"]]' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Search for the gap track</span>
<span class="pl-c"><span class="pl-c">#</span> ahData &lt;- query(ah, c("gap", "Homo sapiens", "hg19"))</span>
<span class="pl-c"><span class="pl-c">#</span> ahData[ahData$title == "Gap"]</span>
<span class="pl-smi">gaps</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">ahData</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH6444<span class="pl-pds">"</span></span>]]</pre>
                    </div>
                    <p dir="auto">
                     The
                     <a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" rel="nofollow">
                      UCSC ‘gap’
table
                     </a>
                     provides better granularity about the types of gaps available. E.g., for
human, hg19, we have the following types and the number of gaps.
                    </p>
                    <p dir="auto">
                     <a href="/dozmorovlab/excluderanges/blob/master/man/figures/README-gapcounts-1.png" rel="noopener noreferrer" target="_blank">
                      <img height="70%" src="/dozmorovlab/excluderanges/raw/master/man/figures/README-gapcounts-1.png" style="max-width: 100%;" width="70%"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Those objects are provided as individual GRanges.
                    </p>
                    <p dir="auto">
                     Naming convention:
                     <code>
                      &lt;genome assembly&gt;.UCSC.&lt;gap type&gt;
                     </code>
                     , e.g.,
                     <code>
                      hg38.UCSC.gap_centromere
                     </code>
                     . We can similarly load any gap type object.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="query_data &lt;- query(ah, c(&quot;excluderanges&quot;, &quot;UCSC&quot;, &quot;Homo Sapiens&quot;, &quot;hg38&quot;))
query_data
#&gt; AnnotationHub with 7 records
#&gt; # snapshotDate(): 2023-10-20
#&gt; # $dataprovider: UCSC, UCSChub
#&gt; # $species: Homo sapiens
#&gt; # $rdataclass: GRanges
#&gt; # additional mcols(): taxonomyid, genome, description,
#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,
#&gt; #   rdatapath, sourceurl, sourcetype 
#&gt; # retrieve records with, e.g., 'object[[&quot;AH107353&quot;]]' 
#&gt; 
#&gt;              title                    
#&gt;   AH107353 | T2T.UCSC.hgUnique.hg38   
#&gt;   AH107354 | hg38.UCSC.centromere     
#&gt;   AH107355 | hg38.UCSC.telomere       
#&gt;   AH107356 | hg38.UCSC.short_arm      
#&gt;   AH107357 | hg38.UCSC.heterochromatin
#&gt;   AH107358 | hg38.UCSC.contig         
#&gt;   AH107359 | hg38.UCSC.scaffold

gapsGR_hg38_centromere &lt;- query_data[[&quot;AH107354&quot;]]
#&gt; loading from cache
gapsGR_hg38_centromere
#&gt; GRanges object with 109 ranges and 2 metadata columns:
#&gt;         seqnames              ranges strand |       bin        name
#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;character&gt;
#&gt;     [1]     chr1 122026459-122224535      * |       189  GJ211836.1
#&gt;     [2]     chr1 122224635-122503147      * |       189  GJ211837.1
#&gt;     [3]     chr1 122503247-124785432      * |        23  GJ212202.1
#&gt;     [4]     chr1 124785532-124849129      * |      1537  GJ211855.1
#&gt;     [5]     chr1 124849229-124932724      * |       192  GJ211857.1
#&gt;     ...      ...                 ...    ... .       ...         ...
#&gt;   [105]    chr22   14419994-14420334      * |       695  GJ212209.1
#&gt;   [106]    chr22   14420434-14421632      * |       695  GJ212186.2
#&gt;   [107]    chr22   14421732-15054318      * |        10  GJ212191.2
#&gt;   [108]     chrX   58605579-62412542      * |         1  GJ212192.1
#&gt;   [109]     chrY   10316944-10544039      * |        10  GJ212193.1
#&gt;   -------
#&gt;   seqinfo: 24 sequences from hg38 genome" dir="auto">
                     <pre><span class="pl-smi">query_data</span> <span class="pl-k">&lt;-</span> query(<span class="pl-smi">ah</span>, c(<span class="pl-s"><span class="pl-pds">"</span>excluderanges<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>UCSC<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Homo Sapiens<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>hg38<span class="pl-pds">"</span></span>))
<span class="pl-smi">query_data</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; AnnotationHub with 7 records</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # snapshotDate(): 2023-10-20</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # $dataprovider: UCSC, UCSChub</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # $species: Homo sapiens</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # $rdataclass: GRanges</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # additional mcols(): taxonomyid, genome, description,</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; #   rdatapath, sourceurl, sourcetype </span>
<span class="pl-c"><span class="pl-c">#</span>&gt; # retrieve records with, e.g., 'object[["AH107353"]]' </span>
<span class="pl-c"><span class="pl-c">#</span>&gt; </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;              title                    </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107353 | T2T.UCSC.hgUnique.hg38   </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107354 | hg38.UCSC.centromere     </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107355 | hg38.UCSC.telomere       </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107356 | hg38.UCSC.short_arm      </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107357 | hg38.UCSC.heterochromatin</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107358 | hg38.UCSC.contig         </span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   AH107359 | hg38.UCSC.scaffold</span>

<span class="pl-smi">gapsGR_hg38_centromere</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">query_data</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH107354<span class="pl-pds">"</span></span>]]
<span class="pl-c"><span class="pl-c">#</span>&gt; loading from cache</span>
<span class="pl-smi">gapsGR_hg38_centromere</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; GRanges object with 109 ranges and 2 metadata columns:</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;         seqnames              ranges strand |       bin        name</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;character&gt;</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [1]     chr1 122026459-122224535      * |       189  GJ211836.1</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [2]     chr1 122224635-122503147      * |       189  GJ211837.1</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [3]     chr1 122503247-124785432      * |        23  GJ212202.1</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [4]     chr1 124785532-124849129      * |      1537  GJ211855.1</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     [5]     chr1 124849229-124932724      * |       192  GJ211857.1</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     ...      ...                 ...    ... .       ...         ...</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [105]    chr22   14419994-14420334      * |       695  GJ212209.1</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [106]    chr22   14420434-14421632      * |       695  GJ212186.2</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [107]    chr22   14421732-15054318      * |        10  GJ212191.2</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [108]     chrX   58605579-62412542      * |         1  GJ212192.1</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   [109]     chrY   10316944-10544039      * |        10  GJ212193.1</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   -------</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   seqinfo: 24 sequences from hg38 genome</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Gaps in mouse genome
                     </h2>
                     <a aria-label="Permalink: Gaps in mouse genome" class="anchor" href="#gaps-in-mouse-genome" id="user-content-gaps-in-mouse-genome">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Note that Mouse chromosomes have centromeres close to one end and have
essentially no short arm. Although available, we do not recommend using
them as they are essentially placeholders.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='query_data &lt;- query(ah, c("excluderanges", "Mus musculus", "UCSC"))
mm39.UCSC.centromere &lt;- query_data[["AH107366"]]
#&gt; loading from cache
head(mm39.UCSC.centromere)
#&gt; GRanges object with 6 ranges and 6 metadata columns:
#&gt;       seqnames         ranges strand |       bin        ix           n
#&gt;          &lt;Rle&gt;      &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;
#&gt;     1     chr1 110000-3000000      * |         9         3           N
#&gt;   314     chr2 110000-3000000      * |         9         3           N
#&gt;   254     chr3 110000-3000000      * |         9         3           N
#&gt;   219     chr4 110000-3000000      * |         9         3           N
#&gt;   211     chr5 110000-3000000      * |         9         3           N
#&gt;   204     chr6 110000-3000000      * |         9         3           N
#&gt;            size        type      bridge
#&gt;       &lt;numeric&gt; &lt;character&gt; &lt;character&gt;
#&gt;     1   2890000  centromere          no
#&gt;   314   2890000  centromere          no
#&gt;   254   2890000  centromere          no
#&gt;   219   2890000  centromere          no
#&gt;   211   2890000  centromere          no
#&gt;   204   2890000  centromere          no
#&gt;   -------
#&gt;   seqinfo: 20 sequences from mm39 genome' dir="auto">
                     <pre><span class="pl-smi">query_data</span> <span class="pl-k">&lt;-</span> query(<span class="pl-smi">ah</span>, c(<span class="pl-s"><span class="pl-pds">"</span>excluderanges<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Mus musculus<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>UCSC<span class="pl-pds">"</span></span>))
<span class="pl-smi">mm39.UCSC.centromere</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">query_data</span>[[<span class="pl-s"><span class="pl-pds">"</span>AH107366<span class="pl-pds">"</span></span>]]
<span class="pl-c"><span class="pl-c">#</span>&gt; loading from cache</span>
head(<span class="pl-smi">mm39.UCSC.centromere</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; GRanges object with 6 ranges and 6 metadata columns:</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;       seqnames         ranges strand |       bin        ix           n</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;          &lt;Rle&gt;      &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     1     chr1 110000-3000000      * |         9         3           N</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   314     chr2 110000-3000000      * |         9         3           N</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   254     chr3 110000-3000000      * |         9         3           N</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   219     chr4 110000-3000000      * |         9         3           N</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   211     chr5 110000-3000000      * |         9         3           N</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   204     chr6 110000-3000000      * |         9         3           N</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;            size        type      bridge</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;       &lt;numeric&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;     1   2890000  centromere          no</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   314   2890000  centromere          no</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   254   2890000  centromere          no</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   219   2890000  centromere          no</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   211   2890000  centromere          no</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   204   2890000  centromere          no</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   -------</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;   seqinfo: 20 sequences from mm39 genome</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      CUT&amp;RUN excludable sets
                     </h1>
                     <a aria-label="Permalink: CUT&amp;RUN excludable sets" class="anchor" href="#cutrun-excludable-sets" id="user-content-cutrun-excludable-sets">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Nordin et al. 2022 (Nordin et al. 2022) generated excludable regions for
the CUT&amp;RUN technology. They are available as
                     <a href="https://www.biorxiv.org/content/10.1101/2022.11.11.516118v1.supplementary-material" rel="nofollow">
                      Supplementary
Material
                     </a>
                     .
We uniformly processed them and made available on Google Drive
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# hg38 CUT&amp;RUN exclusion set, BED
download.file("https://drive.google.com/uc?export=download&amp;id=1rKIu7kdiEySTi-cq3nYxXJP4VQX1IPcS",
              destfile = "hg38.Nordin.CandRblacklist_hg38.bed")
# hg38 CUT&amp;RUN exclusion set, RDS
download.file("https://drive.google.com/uc?export=download&amp;id=1JuB1h-QQUw1mddBavI7CIuH7R-lwwczU",
              destfile = "hg38.Nordin.CandRblacklist_hg38.rds")
# And then load the GRanges object
mtx &lt;- readRDS("hg38.Nordin.CandRblacklist_hg38.rds")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> hg38 CUT&amp;RUN exclusion set, BED</span>
download.file(<span class="pl-s"><span class="pl-pds">"</span>https://drive.google.com/uc?export=download&amp;id=1rKIu7kdiEySTi-cq3nYxXJP4VQX1IPcS<span class="pl-pds">"</span></span>,
              <span class="pl-v">destfile</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>hg38.Nordin.CandRblacklist_hg38.bed<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> hg38 CUT&amp;RUN exclusion set, RDS</span>
download.file(<span class="pl-s"><span class="pl-pds">"</span>https://drive.google.com/uc?export=download&amp;id=1JuB1h-QQUw1mddBavI7CIuH7R-lwwczU<span class="pl-pds">"</span></span>,
              <span class="pl-v">destfile</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>hg38.Nordin.CandRblacklist_hg38.rds<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> And then load the GRanges object</span>
<span class="pl-smi">mtx</span> <span class="pl-k">&lt;-</span> readRDS(<span class="pl-s"><span class="pl-pds">"</span>hg38.Nordin.CandRblacklist_hg38.rds<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# mm10 CUT&amp;RUN exclusion set, BED
download.file("https://drive.google.com/uc?export=download&amp;id=1CRAojdphMbAzd3MnW_UmO1WtsDrHsrU1",
              destfile = "mm10.Nordin.CandRblacklist_mm10.bed")
# mm10 CUT&amp;RUN exclusion set, RDS
download.file("https://drive.google.com/uc?export=download&amp;id=1orPXLWUZ4-C4n_Jt2gH-WERLpY9Kn0t_",
              destfile = "mm10.Nordin.CandRblacklist_mm10.rds")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> mm10 CUT&amp;RUN exclusion set, BED</span>
download.file(<span class="pl-s"><span class="pl-pds">"</span>https://drive.google.com/uc?export=download&amp;id=1CRAojdphMbAzd3MnW_UmO1WtsDrHsrU1<span class="pl-pds">"</span></span>,
              <span class="pl-v">destfile</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>mm10.Nordin.CandRblacklist_mm10.bed<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> mm10 CUT&amp;RUN exclusion set, RDS</span>
download.file(<span class="pl-s"><span class="pl-pds">"</span>https://drive.google.com/uc?export=download&amp;id=1orPXLWUZ4-C4n_Jt2gH-WERLpY9Kn0t_<span class="pl-pds">"</span></span>,
              <span class="pl-v">destfile</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>mm10.Nordin.CandRblacklist_mm10.rds<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Summary table
                     </h1>
                     <a aria-label="Permalink: Summary table" class="anchor" href="#summary-table" id="user-content-summary-table">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/dozmorovlab/excluderanges/blob/master/man/figures/Table_S1.csv">
                      Full summary table
                     </a>
                     .
                    </p>
                    <table>
                     <thead>
                      <tr>
                       <th align="left">
                        Name
                       </th>
                       <th align="left">
                        Ahub.IDs.BioC.3.16.and.above
                       </th>
                       <th align="left">
                        BEDbase.URL
                       </th>
                       <th align="left">
                        Description
                       </th>
                       <th align="right">
                        Filtered.Region.count
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td align="left">
                        T2T.excluderanges
                       </td>
                       <td align="left">
                        AH107304
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/8329d8c624880308ab51ba05149a737d" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions
                       </td>
                       <td align="right">
                        3565
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.Kundaje.GRCh38_unified_Excludable
                       </td>
                       <td align="left">
                        AH107305
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/1a561729234c2844303a051b16f66656" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined as a combination of hg38.Lareau.hg38_peaks, hg38.Boyle.hg38-Excludable.v2, and hg38.Wimberley.peakPass60Perc_sorted, followed by manual curation. Supersedes hg38.Kundaje.GRCh38.Excludable.
                       </td>
                       <td align="right">
                        910
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.Bernstein.Mint_Excludable_GRCh38
                       </td>
                       <td align="left">
                        AH107306
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/80e335903b77b597b8245f9817fcd9cd" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined from Mint-ChIP (low input, multiplexed ChIP-seq) data
                       </td>
                       <td align="right">
                        12052
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.Boyle.hg38-Excludable.v2
                       </td>
                       <td align="left">
                        AH107307
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/ac58962c9ec98fe9258c12092a0c8832" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions
                       </td>
                       <td align="right">
                        636
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.Kundaje.GRCh38.Excludable
                       </td>
                       <td align="left">
                        AH107308
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/cb701496bde7eeb18add96fdbc3b8b11" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia
                       </td>
                       <td align="right">
                        38
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.Lareau.hg38.full.Excludable
                       </td>
                       <td align="left">
                        AH107309
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/5a12c1de138ace1a73a45e6faf9ba669" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)
                       </td>
                       <td align="right">
                        820
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38
                       </td>
                       <td align="left">
                        AH107310
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/148622e896f6798f7c4abf448bab67c4" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), centromeric repeats, chrM, High/Low mappability islands. Has extra chromosomes, use keepStandardChromosomes() filtering
                       </td>
                       <td align="right">
                        396
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.Wimberley.peakPass60Perc_sorted
                       </td>
                       <td align="left">
                        AH107311
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/f4a9bb19ed29e993592813e970e7dd90" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the ewimberley/peakPass software
                       </td>
                       <td align="right">
                        5078
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.Wold.hg38mitoExcludable
                       </td>
                       <td align="left">
                        AH107312
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/a714dcba99821801b5c426fba9c80988" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Definition method unknown
                       </td>
                       <td align="right">
                        299
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed
                       </td>
                       <td align="left">
                        AH107313
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/1a02a65fafefefd65ff4a060273304ed" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined from eCLIP data
                       </td>
                       <td align="right">
                        56
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.Nordin.CandRblacklist_hg38
                       </td>
                       <td align="left">
                        NA
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/b410448ba193fc3f0e99ca367e10b3b6" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined from CUT&amp;RUN negative controls as 0.1% top significant SEACR peaks in over 30% of samples
                       </td>
                       <td align="right">
                        885
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.Boyle.hg19-Excludable.v2
                       </td>
                       <td align="left">
                        AH107314
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/6eb180d456f2f3b71b419e5fab107fc9" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions
                       </td>
                       <td align="right">
                        834
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.Bernstein.Mint_Excludable_hg19
                       </td>
                       <td align="left">
                        AH107315
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/d1a6047ed5bec84acefe9c52cf63b593" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined from Mint-ChIP (low input, multiplexed ChIP-seq) data
                       </td>
                       <td align="right">
                        9035
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.Birney.wgEncodeDacMapabilityConsensusExcludable
                       </td>
                       <td align="left">
                        AH107316
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/5b6b19dea85a8bc6007ef07a0960267b" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), centromeric repeats, chrM, High/Low mappability islands
                       </td>
                       <td align="right">
                        411
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable
                       </td>
                       <td align="left">
                        AH107317
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/dac2eda4e8687eb039611ac6cd595821" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), human satellite repeat HSATII, chrM, ribosomal subunit consensus sequences LSU-rRNA_Hsa, SSU-rRNA_Hsa. Has extra chromosomes, use keepStandardChromosomes() filtering
                       </td>
                       <td align="right">
                        1566
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.Lareau.hg19.full.Excludable
                       </td>
                       <td align="left">
                        AH107318
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/d934d47e8035da9c5a1767c8153db4cc" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)
                       </td>
                       <td align="right">
                        902
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.Wold.hg19mitoExcludable
                       </td>
                       <td align="left">
                        AH107319
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/182046a0f055b0176178241a95cbd637" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Definition method unknown
                       </td>
                       <td align="right">
                        295
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.Yeo.eCLIP_Excludableregions.hg19
                       </td>
                       <td align="left">
                        AH107320
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/350f49dc47e5307109e1e17d60223a31" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined from eCLIP data, includes skyscraper, rRNA pseudogene, unreliably mapped satellite repeat, and low complexity skyscraper peak regions
                       </td>
                       <td align="right">
                        57
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm39.excluderanges
                       </td>
                       <td align="left">
                        AH107321
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/edc716833d4b5ee75c34a0692fc353d5" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions
                       </td>
                       <td align="right">
                        3147
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.Boyle.mm10-Excludable.v2
                       </td>
                       <td align="left">
                        AH107322
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/a5311e39fe1590de66c1df6a5881a942" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions
                       </td>
                       <td align="right">
                        3435
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.Hardison.Excludable.full
                       </td>
                       <td align="left">
                        AH107323
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/087541f51cf8c7d7078995d1bd95fd27" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Definition method unknown
                       </td>
                       <td align="right">
                        7865
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.Hardison.psuExcludable.mm10
                       </td>
                       <td align="left">
                        AH107324
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/fc6b88f936c5cd880545943708e4c2af" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Definition method unknown
                       </td>
                       <td align="right">
                        5552
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.Kundaje.anshul.Excludable.mm10
                       </td>
                       <td align="left">
                        AH107325
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/e6a89a8432f4a69bae41f60ed0c7e704" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia
                       </td>
                       <td align="right">
                        3010
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.Kundaje.mm10.Excludable
                       </td>
                       <td align="left">
                        AH107326
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/76c03b6c831f8fecdf4fee7adf2def6a" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia
                       </td>
                       <td align="right">
                        164
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.Lareau.mm10.full.Excludable
                       </td>
                       <td align="left">
                        AH107327
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/1bd30517be79d4d051308c693b822798" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)
                       </td>
                       <td align="right">
                        523
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.Wold.mm10mitoExcludable
                       </td>
                       <td align="left">
                        AH107328
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/830f1ffd31689e3e7c22ff856f0ba02c" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Definition method unknown
                       </td>
                       <td align="right">
                        123
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.Nordin.CandRblacklist_mm10
                       </td>
                       <td align="left">
                        NA
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/cf3f5440b73fbbdc5b591a4bc6611a0e" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined from CUT&amp;RUN negative controls as 0.1% top significant SEACR peaks in over 30% of samples
                       </td>
                       <td align="right">
                        559
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm9.Lareau.mm9.full.Excludable
                       </td>
                       <td align="left">
                        AH107329
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/e903b285baefce8167367ce57a8c3d48" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)
                       </td>
                       <td align="right">
                        3415
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm9.Wold.mm9mitoExcludable
                       </td>
                       <td align="left">
                        AH107330
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/9b4389a6a4b937df8abd62dad30fa3a3" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Definition method unknown
                       </td>
                       <td align="right">
                        123
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        ce11.Boyle.ce11-Excludable.v2
                       </td>
                       <td align="left">
                        AH107331
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/7235114a78b1709be96f0d6a82b4ea36" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions
                       </td>
                       <td align="right">
                        97
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        ce10.Boyle.ce10-Excludable.v2
                       </td>
                       <td align="left">
                        AH107332
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/6de11bb5f50ee015b23ac96f433f00bb" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions
                       </td>
                       <td align="right">
                        100
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        ce10.Kundaje.ce10-Excludable
                       </td>
                       <td align="left">
                        AH107333
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/32b59590fa83161687cec4cabfa2bb2b" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by Anshul Kundaje, superseded by ce10.Boyle.ce10-Excludable.v2
                       </td>
                       <td align="right">
                        122
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        danRer10.Domingues.Excludableed
                       </td>
                       <td align="left">
                        AH107334
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/a0a94af275f858d63550005627d260b7" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined manually using total RNA-seq.
                       </td>
                       <td align="right">
                        57
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        danRer10.Yang.Supplemental_Table_19.ChIP-seq_black_list_in_the_zebrafish_genome
                       </td>
                       <td align="left">
                        AH107335
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/78f5eb585019a4d795ef80159a597b15" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined via MACS2 peak calling using ChIP-seq (PMID: 33239788)
                       </td>
                       <td align="right">
                        853
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        dm6.Boyle.dm6-Excludable.v2
                       </td>
                       <td align="left">
                        AH107336
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/24186dc2aac492074d3de9caede730a0" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions
                       </td>
                       <td align="right">
                        182
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        dm3.Boyle.dm3-Excludable.v2
                       </td>
                       <td align="left">
                        AH107337
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/7427399e18d9c01e423b2f4963b409ea" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions
                       </td>
                       <td align="right">
                        248
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        dm3.Kundaje.dm3-Excludable
                       </td>
                       <td align="left">
                        AH107338
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/0801a522159f7ebf2f669d8cade4aa8f" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by Anshul Kundaje. Contains heterochromatin chromosomes chr2LHet. Superseded by dm3.Boyle.dm3-Excludable.v2
                       </td>
                       <td align="right">
                        306
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        TAIR10.Wimberley.predicted_excluded_list_sorted_0.6
                       </td>
                       <td align="left">
                        AH107339
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/6f3a3ae3ee878b88a92093eb8e3fe982" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the ewimberley/peakPass software
                       </td>
                       <td align="right">
                        887
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        TAIR10.Klasfeld.arabidopsis_Excludable_20inputs
                       </td>
                       <td align="left">
                        AH107340
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/aa1c99c2dd2aef874486b1c0c3bf6b92" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions (DOI: 10.1101/2022.02.27.482177)
                       </td>
                       <td align="right">
                        83
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        TAIR10.Klasfeld.arabidopsis_greenscreen_20inputs
                       </td>
                       <td align="left">
                        AH107341
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/e5d66ee787a8cb0c76438bba768c2331" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Defined by the green screen pipeline (DOI: 10.1101/2022.02.27.482177)
                       </td>
                       <td align="right">
                        36
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        T2T.Lareau.chm13v2.0_peaks
                       </td>
                       <td align="left">
                        AH107342
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/354dfced295f54f70ae9656ca8f9b141" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist
                       </td>
                       <td align="right">
                        817
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.Lareau.hg38_peaks
                       </td>
                       <td align="left">
                        AH107343
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/9fa55701a3bd3e7a598d1d2815e3390f" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist
                       </td>
                       <td align="right">
                        784
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.Lareau.hg19_peaks
                       </td>
                       <td align="left">
                        AH107344
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/79e924141251afbd4cde0c38456913fd" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist
                       </td>
                       <td align="right">
                        779
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.Lareau.mm10_peaks
                       </td>
                       <td align="left">
                        AH107345
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/1b76ab775549e116da5e1a89aad7019b" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist
                       </td>
                       <td align="right">
                        387
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm9.Lareau.mm9_peaks
                       </td>
                       <td align="left">
                        AH107346
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/5c4b1cb28175b72bc56adb0bd7384dfd" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist
                       </td>
                       <td align="right">
                        395
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.UCSC.numtS
                       </td>
                       <td align="left">
                        AH107347
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/cc4fd05fdfe015e4acd5111dac5b372f" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Human NumtS mitochondrial sequence
                       </td>
                       <td align="right">
                        766
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm9.UCSC.numtS
                       </td>
                       <td align="left">
                        AH107348
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/29dc50750f0535b6b9c746ee8371c211" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Mouse NumtS mitochondrial sequence
                       </td>
                       <td align="right">
                        172
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        T2T.CHM13.chm13.draft_v2.0.cen_mask
                       </td>
                       <td align="left">
                        AH107349
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/44138ebb0d3340e70164d12649a47dc8" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Centromeric satellite masking bed file (v2.0)
                       </td>
                       <td align="right">
                        23
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        T2T.CHM13.chm13.draft_v1.1.telomere
                       </td>
                       <td align="left">
                        AH107350
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/b72dd2fa5f8a916cc36960b93169c743" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Telomere identified by the VGP pipeline (v1.1)
                       </td>
                       <td align="right">
                        48
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        T2T.UCSC.censat
                       </td>
                       <td align="left">
                        AH107351
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/f28798df2c4d72810e7c4626b5a62106" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        T2T peri/centromeric satellite annotation (v2.0, 20220329, CHM13 v2.0)
                       </td>
                       <td align="right">
                        2523
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        T2T.UCSC.gap
                       </td>
                       <td align="left">
                        AH107352
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/0747aae5f4cac92367a16c3eb1c7f3f1" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Locations of assembly gaps, as determine by strings of ‘N’ characters (v1.0)
                       </td>
                       <td align="right">
                        5
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        T2T.UCSC.hgUnique.hg38
                       </td>
                       <td align="left">
                        AH107353
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/c3839f43c53a3c47733388528b853690" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Regions unique to the T2T-CHM13 v2.0 assembly compared to the GRCh38/hg38 and GRCh37/hg19 reference assemblies
                       </td>
                       <td align="right">
                        615
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.UCSC.centromere
                       </td>
                       <td align="left">
                        AH107354
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/0b1f161675fa0f52ac6d0d4f54b1efb9" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps from centromeres
                       </td>
                       <td align="right">
                        109
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.UCSC.telomere
                       </td>
                       <td align="left">
                        AH107355
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/79f964e68d5daa1462c52ca54855b06a" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps from telomeres
                       </td>
                       <td align="right">
                        48
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.UCSC.short_arm
                       </td>
                       <td align="left">
                        AH107356
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/92fc8f64f92d525c6b92c9aab5e2c711" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps on the short arm of the chromosome
                       </td>
                       <td align="right">
                        5
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.UCSC.heterochromatin
                       </td>
                       <td align="left">
                        AH107357
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/8af7b48ab48183229d3bc72005040dc1" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps from large blocks of heterochromatin
                       </td>
                       <td align="right">
                        11
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.UCSC.contig
                       </td>
                       <td align="left">
                        AH107358
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/2dd1b22f2add15bc7508580d18bc9495" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between contigs in scaffolds
                       </td>
                       <td align="right">
                        285
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg38.UCSC.scaffold
                       </td>
                       <td align="left">
                        AH107359
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/de0c7f42f29fb83ac393e86a2ec28374" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between scaffolds in chromosome assemblies. Has extra chromosomes, use keepStandardChromosomes() filtering
                       </td>
                       <td align="right">
                        254
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.UCSC.centromere
                       </td>
                       <td align="left">
                        AH107360
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/26ecf1381b6323791656f800ad39b69c" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps from centromeres
                       </td>
                       <td align="right">
                        24
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.UCSC.telomere
                       </td>
                       <td align="left">
                        AH107361
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/2bcad8794847411e9b3f52ff39c4f377" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps from telomeres
                       </td>
                       <td align="right">
                        46
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.UCSC.short_arm
                       </td>
                       <td align="left">
                        AH107362
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/e09fac8aedf1230ab77ac4194fd75784" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps on the short arm of the chromosome
                       </td>
                       <td align="right">
                        5
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.UCSC.heterochromatin
                       </td>
                       <td align="left">
                        AH107363
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/8ea9b6cdfe68a4b4111e5b03157af371" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps from large blocks of heterochromatin
                       </td>
                       <td align="right">
                        12
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.UCSC.clone
                       </td>
                       <td align="left">
                        AH107364
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/4f3b1098a0f4ea5e81747f4414a8d294" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between clones in the same map contig. Has extra chromosomes, use keepStandardChromosomes() filtering
                       </td>
                       <td align="right">
                        107
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.UCSC.contig
                       </td>
                       <td align="left">
                        AH107365
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/a4da41916b0b213d4e3b89f5ab20e1e8" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between contigs in scaffolds
                       </td>
                       <td align="right">
                        163
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        hg19.UCSC.scaffold
                       </td>
                       <td align="left">
                        AH107366
                       </td>
                       <td align="left">
                        NA
                       </td>
                       <td align="left">
                        Gaps between scaffolds in chromosome assemblies. Only non-autosomal chromosomes
                       </td>
                       <td align="right">
                        0
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm39.UCSC.centromere
                       </td>
                       <td align="left">
                        AH107367
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/1aa3f73ffa8e6d498f0f3f22e0302472" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps from centromeres
                       </td>
                       <td align="right">
                        20
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm39.UCSC.telomere
                       </td>
                       <td align="left">
                        AH107368
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/883bdae38244c6f0e0facfbd4fcc601b" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps from telomeres
                       </td>
                       <td align="right">
                        42
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm39.UCSC.short_arm
                       </td>
                       <td align="left">
                        AH107369
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/f61de62eae0898943e6c9d163b0a3989" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps on the short arm of the chromosome
                       </td>
                       <td align="right">
                        21
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm39.UCSC.contig
                       </td>
                       <td align="left">
                        AH107370
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/732437d9fcb8992b2e5c6513bfed2586" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between contigs in scaffolds
                       </td>
                       <td align="right">
                        60
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm39.UCSC.scaffold
                       </td>
                       <td align="left">
                        AH107371
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/97e738326c0681f5ebe94b0d28d058c5" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between scaffolds in chromosome assemblies
                       </td>
                       <td align="right">
                        115
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.UCSC.centromere
                       </td>
                       <td align="left">
                        AH107372
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/b0f9aa3cc8a4a43f59b463891b5d12c8" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps from centromeres
                       </td>
                       <td align="right">
                        20
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.UCSC.telomere
                       </td>
                       <td align="left">
                        AH107373
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/051090e82c227bcc55dba3e953bc6daa" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps from telomeres
                       </td>
                       <td align="right">
                        42
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.UCSC.short_arm
                       </td>
                       <td align="left">
                        AH107374
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/f4d2d6fe334deca5800ca9ae39ce95ce" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps on the short arm of the chromosome
                       </td>
                       <td align="right">
                        21
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.UCSC.clone
                       </td>
                       <td align="left">
                        AH107375
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/9ecfce46335e4d5b3a1230b69690a25a" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between clones in the same map contig. Has extra chromosomes, use keepStandardChromosomes() filtering
                       </td>
                       <td align="right">
                        4
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.UCSC.contig
                       </td>
                       <td align="left">
                        AH107376
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/82d2374cf5524a2b13dcf9c3dc487d6f" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between contigs in scaffolds
                       </td>
                       <td align="right">
                        104
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.UCSC.scaffold
                       </td>
                       <td align="left">
                        AH107377
                       </td>
                       <td align="left">
                        NA
                       </td>
                       <td align="left">
                        Gaps between scaffolds in chromosome assemblies
                       </td>
                       <td align="right">
                        0
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.UCSC.other
                       </td>
                       <td align="left">
                        AH107378
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/75662812e5eb228b25c9ae5a28fbb402" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Sequence of Ns in the assembly that were not marked as gaps in the AGP (A Golden Path) assembly definition file. Has extra chromosomes, use keepStandardChromosomes() filtering
                       </td>
                       <td align="right">
                        383
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm10.UCSC.fragment
                       </td>
                       <td align="left">
                        AH107379
                       </td>
                       <td align="left">
                        NA
                       </td>
                       <td align="left">
                        A single gap of 31 bases in chrX_GL456233_random
                       </td>
                       <td align="right">
                        0
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm9.UCSC.centromere
                       </td>
                       <td align="left">
                        AH107380
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/99e4d2c9a794d321bfcf01709787caac" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps from centromeres
                       </td>
                       <td align="right">
                        21
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm9.UCSC.fragment
                       </td>
                       <td align="left">
                        AH107381
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/ecf3f802759c5dc93f1446b6942c58b3" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between the contigs of a draft clone. (In this context, a contig is a set of overlapping sequence reads). Has extra chromosomes, use keepStandardChromosomes() filtering
                       </td>
                       <td align="right">
                        436
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        mm9.UCSC.contig
                       </td>
                       <td align="left">
                        AH107382
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/4dd8bb54f6432144c045619337d8212e" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between contigs in scaffolds. Has extra chromosomes, use keepStandardChromosomes() filtering
                       </td>
                       <td align="right">
                        105
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        danRer10.UCSC.contig
                       </td>
                       <td align="left">
                        AH107383
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/5d41a9fa328769b63734e11e6ae4252b" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between contigs in scaffolds
                       </td>
                       <td align="right">
                        2338
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        danRer10.UCSC.scaffold
                       </td>
                       <td align="left">
                        AH107384
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/a5feefb2d573d265f1085043c208c2ed" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between scaffolds in chromosome assemblies
                       </td>
                       <td align="right">
                        16496
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        dm6.UCSC.other
                       </td>
                       <td align="left">
                        AH107385
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/11a80264dcdad6c0868ea48637a799df" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Sequence of Ns in the assembly that were not marked as gaps in the AGP (A Golden Path) assembly definition file
                       </td>
                       <td align="right">
                        268
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        dm3.UCSC.contig
                       </td>
                       <td align="left">
                        NA
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/f9822a0f88047e92cf92824fe025b2f2" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between contigs in scaffolds
                       </td>
                       <td align="right">
                        7
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        dm3.UCSC.scaffold
                       </td>
                       <td align="left">
                        NA
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/d4e31a2c488de8ff335b0cb779c9cef5" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps between scaffolds in chromosome assemblies
                       </td>
                       <td align="right">
                        1
                       </td>
                      </tr>
                      <tr>
                       <td align="left">
                        TAIR10.UCSC.araTha1.gap
                       </td>
                       <td align="left">
                        NA
                       </td>
                       <td align="left">
                        <a href="http://bedbase.org/#/bedsplash/74585119b9b90d3b4ad077b10b487d39" rel="nofollow">
                         link
                        </a>
                       </td>
                       <td align="left">
                        Gaps in the May 2011 Arabidopsis thaliana genome assembly
                       </td>
                       <td align="right">
                        357
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Citation
                     </h1>
                     <a aria-label="Permalink: Citation" class="anchor" href="#citation" id="user-content-citation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Below is the citation output from using
                     <code>
                      citation('excluderanges')
                     </code>
                     in
R. Please run this yourself to check for any updates on how to cite
                     <strong>
                      excluderanges
                     </strong>
                     .
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='print(citation("excluderanges"), bibtex = TRUE)' dir="auto">
                     <pre>print(citation(<span class="pl-s"><span class="pl-pds">"</span>excluderanges<span class="pl-pds">"</span></span>), <span class="pl-v">bibtex</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Code of Conduct
                     </h1>
                     <a aria-label="Permalink: Code of Conduct" class="anchor" href="#code-of-conduct" id="user-content-code-of-conduct">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Please note that the
                     <code>
                      excluderanges
                     </code>
                     project is released with a
                     <a href="https://bioconductor.github.io/bioc_coc_multilingual/" rel="nofollow">
                      Contributor Code of
Conduct
                     </a>
                     . By
contributing to this project, you agree to abide by its terms.
                    </p>
                    <p dir="auto">
                     This package was developed using
                     <em>
                      <a href="https://bioconductor.org/packages/3.18/biocthis" rel="nofollow">
                       biocthis
                      </a>
                     </em>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      References
                     </h1>
                     <a aria-label="Permalink: References" class="anchor" href="#references" id="user-content-references">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div dir="auto" id="user-content-refs">
                     <div dir="auto" id="user-content-ref-amemiya:2019aa">
                      <p dir="auto">
                       Amemiya, Haley M, Anshul Kundaje, and Alan P Boyle. 2019. “The ENCODE
Blacklist: Identification of Problematic Regions of the Genome.”
                       <em>
                        Sci
Rep
                       </em>
                       9 (1): 9354.
                       <a href="https://doi.org/10.1038/s41598-019-45839-z" rel="nofollow">
                        https://doi.org/10.1038/s41598-019-45839-z
                       </a>
                       .
                      </p>
                     </div>
                     <div dir="auto" id="user-content-ref-klasfeld2022-nn">
                      <p dir="auto">
                       Klasfeld, Sammy, and Doris Wagner. 2022. “Greenscreen Decreases Type I
Errors and Increases True Peak Detection in Genomic Datasets Including
                       <span>
                        ChIP-seq
                       </span>
                       .”
                       <em>
                        bioRxiv
                       </em>
                       .
                      </p>
                     </div>
                     <div dir="auto" id="user-content-ref-nordin2022cut">
                      <p dir="auto">
                       Nordin, Anna, Gianluca Zambanini, Pierfrancesco Pagella, and Claudio
Cantu. 2022. “The CUT&amp;RUN Blacklist of Problematic Regions of the
Genome.”
                       <em>
                        bioRxiv
                       </em>
                       .
                      </p>
                     </div>
                     <div dir="auto" id="user-content-ref-ogata:2023aa">
                      <p dir="auto">
                       Ogata, Jonathan D, Wancen Mu, Eric S Davis, Bingjie Xue, J Chuck
Harrell, Nathan C Sheffield, Douglas H Phanstiel, Michael I Love, and
Mikhail G Dozmorov. 2023. “Excluderanges: Exclusion Sets for T2T-CHM13,
GRCm39, and Other Genome Assemblies.”
                       <em>
                        Bioinformatics
                       </em>
                       39 (4).
                       <a href="https://doi.org/10.1093/bioinformatics/btad198" rel="nofollow">
                        https://doi.org/10.1093/bioinformatics/btad198
                       </a>
                       .
                      </p>
                     </div>
                     <div dir="auto" id="user-content-ref-wimberley:2019ub">
                      <p dir="auto">
                       Wimberley, Charles E, and Steffen Heber. 2019. “PeakPass: Automating
ChIP-Seq Blacklist Creation.”
                       <em>
                        J Comput Biol
                       </em>
                       , December.
                       <a href="https://doi.org/10.1089/cmb.2019.0295" rel="nofollow">
                        https://doi.org/10.1089/cmb.2019.0295
                       </a>
                       .
                      </p>
                     </div>
                    </div>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="Fv3BeYdTPWtloVqsK55PZv/wt/4nujGSDRUQW6ZjECMg2Nua1VsNqA64omLEmrANxJKHuEDlzmHHJquO4V8ldg==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 Genomic coordinates of problematic genomic regions as GRanges
                </p>
                <div class="my-3 d-flex flex-items-center">
                 <svg aria-hidden="true" class="octicon octicon-link flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                  </path>
                 </svg>
                 <span class="flex-auto min-width-0 css-truncate css-truncate-target width-fit">
                  <a class="text-bold" href="https://dozmorovlab.github.io/excluderanges/" rel="noopener noreferrer nofollow" role="link" target="_blank" title="https://dozmorovlab.github.io/excluderanges/">
                   dozmorovlab.github.io/excluderanges/
                  </a>
                 </span>
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <h3 class="sr-only">
                 License
                </h3>
                <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default d-inline mt-2">
                 <summary class="Link--muted">
                  <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                   </path>
                  </svg>
                  Unknown, MIT licenses found
                 </summary>
                 <details-dialog aria-label="Licenses found" class="Box Box--overlay d-flex flex-column anim-fade-in fast">
                  <div class="Box-header">
                   <button aria-label="Close licenses dialog" class="Box-btn-octicon btn-octicon float-right" data-action="toggle" data-close-dialog="" type="button">
                    <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </button>
                   <h3 class="Box-title">
                    Licenses found
                   </h3>
                  </div>
                  <a aria-label="Unknown license" class="Link--primary no-underline" href="/dozmorovlab/excluderanges/blob/master/./LICENSE">
                   <div class="Box-row Box-row--hover-gray border-top rounded-0">
                    <div class="text-bold">
                     Unknown
                    </div>
                    <span class="f6 color-fg-muted">
                     LICENSE
                    </span>
                   </div>
                  </a>
                  <a aria-label="MIT license" class="Link--primary no-underline" href="/dozmorovlab/excluderanges/blob/master/./LICENSE.md">
                   <div class="Box-row Box-row--hover-gray border-top rounded-0">
                    <div class="text-bold">
                     MIT
                    </div>
                    <span class="f6 color-fg-muted">
                     LICENSE.md
                    </span>
                   </div>
                  </a>
                 </details-dialog>
                </details>
                <include-fragment src="/dozmorovlab/excluderanges/hovercards/citation/sidebar_partial?tree_name=master">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/dozmorovlab/excluderanges/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/dozmorovlab/excluderanges/custom-properties">
                  <svg aria-hidden="true" class="octicon octicon-note mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25Zm1.75-.25a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25ZM3.5 6.25a.75.75 0 0 1 .75-.75h7a.75.75 0 0 1 0 1.5h-7a.75.75 0 0 1-.75-.75Zm.75 2.25h4a.75.75 0 0 1 0 1.5h-4a.75.75 0 0 1 0-1.5Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Custom properties
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/dozmorovlab/excluderanges/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   19
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/dozmorovlab/excluderanges/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   2
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/dozmorovlab/excluderanges/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   2
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fdozmorovlab%2Fexcluderanges&amp;report=dozmorovlab+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/dozmorovlab/excluderanges/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/orgs/dozmorovlab/packages?repo_name=excluderanges">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/dozmorovlab/excluderanges/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="R 95.9" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#198CE7 !important;;width: 95.9%;">
                 </span>
                 <span aria-label="TeX 4.1" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#3D6117 !important;;width: 4.1%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/dozmorovlab/excluderanges/search?l=r">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#198CE7;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   R
                  </span>
                  <span>
                   95.9%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/dozmorovlab/excluderanges/search?l=tex">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#3D6117;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   TeX
                  </span>
                  <span>
                   4.1%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
