<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-bf4f9425a973.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-340b2bef927d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_github_catalyst_lib_index_-eccae9-1932eeecf006.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-318896a86f35.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-9fad2423070b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1: THIS REPOSITORY CONTAINS STAGE 2 TASK FROM HACKBIO INTERNSHIP 2021
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:AEB95A:1016F69:6605977D" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDpBRUI5NUE6MTAxNkY2OTo2NjA1OTc3RCIsInZpc2l0b3JfaWQiOiI2NDUyMjQxODU2MjM4ODE3ODA3IiwicmVnaW9uX2VkZ2UiOiJpYWQiLCJyZWdpb25fcmVuZGVyIjoiaWFkIn0=" data-pjax-transient="true" name="visitor-payload">
           <meta content="ff35b571fa244aa4a6ebccd9de914c4a825eb937a18d704834a650843c9a6c3c" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:397162940" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="THIS REPOSITORY CONTAINS STAGE 2 TASK FROM HACKBIO INTERNSHIP 2021 - DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/bfb54d58aa7c216e27fd803ccc52c8ff375f25725378249bd5c738439fcf9dbd/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1: THIS REPOSITORY CONTAINS STAGE 2 TASK FROM HACKBIO INTERNSHIP 2021" name="twitter:title">
                    <meta content="THIS REPOSITORY CONTAINS STAGE 2 TASK FROM HACKBIO INTERNSHIP 2021 - DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/bfb54d58aa7c216e27fd803ccc52c8ff375f25725378249bd5c738439fcf9dbd/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" property="og:image">
                      <meta content="THIS REPOSITORY CONTAINS STAGE 2 TASK FROM HACKBIO INTERNSHIP 2021 - DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1: THIS REPOSITORY CONTAINS STAGE 2 TASK FROM HACKBIO INTERNSHIP 2021" property="og:title"/>
                        <meta content="https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" property="og:url"/>
                        <meta content="THIS REPOSITORY CONTAINS STAGE 2 TASK FROM HACKBIO INTERNSHIP 2021 - DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="a81c045767350c9f2e551b272402db2e99346c1d2710552f41e25d405666b5eb" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="3037d78571ecdebb88047af8e128a53bf5dc96ac8c533579876941ff0be8ae32" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1 git https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1.git" name="go-import"/>
                        <meta content="88287856" name="octolytics-dimension-user_id">
                         <meta content="DSJaswanth" name="octolytics-dimension-user_login">
                          <meta content="397162940" name="octolytics-dimension-repository_id">
                           <meta content="DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="397162940" name="octolytics-dimension-repository_network_root_id">
                               <meta content="DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-99519581d0f8.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1","user_id":null}}' data-hydro-click-hmac="f4491b5904781299c6aefaeab4e1fa63dc2f5ac241ddc357e634ff36ff0d7ac2" href="/login?return_to=https%3A%2F%2Fgithub.com%2FDSJaswanth%2FHACKBIO_2021_TEAM_EPIGENOMICS1">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=FJ9XgOu8E7qVMkeZJdAxvdWZ%2B5wJRGlnxG%2BYShfPVThAgS62n03FqFAsTYEBqPGbEFZNKg1OShnOF5BYUH6AB0x0u0oduxAtXv7Z8bUEht98JDbetK61CAdyIfUR53zn4waeUEn%2BknWKV9ZJJzw7v7HNcNHLoQERbMR2A2Gtgpm4AEhXztdjN5Avhs3ewe8shldtfCHVpO9TFf3bTxLbSAxxWIZI9sWa2%2Bi6nHhZ%2FQx7WQOhAIdI2TQKsetavT8VHz2ADO9JfsEPY9xjDMGGSG2SJJnyXKWIyoqv%2FUwdBPurMkQiskhQpQuiFnBuHwOxL7STmdXI1hWsTXrPqF6%2BqRaCmy8VNm5TlNVQ3GR6vRqaUKxzxX1RO7UHWg6GajuMGB65QkM89g36OJpeiI5Cm94o0IOnoQm6KUIHmcxPJtpcRssr5f%2BGjIvkzZH4IoDxhsODNNEYwjuu76VTq4%2FHSYGq1W3Po6wHpYycQQ4IUwmNQNfJthvG6p0kFokHDAnPNbYgFpKlEn%2Bp6jtpNXIxXRRUUooNEyGWAcXxLi80cNgoSVAKFhdCod3KmVNGtA%3D%3D--uH6inJE02UspbzvA--Q77IeiKzayIbjRumzhc1%2FA%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="DSJaswanth" data-current-repository="DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="6k77ktnhqvp0R3jrrnLQVkArjpJZKxS924KPX0E4jgdk4VsZDG9GiLGU-QgeHSuQJXD0cQsm_Y8VJAhvQNUm-w" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-090fbd6b-3f15-4a0d-9200-4d37c357e404" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-090fbd6b-3f15-4a0d-9200-4d37c357e404">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="NJQBuBE6aqbzYtZzTz4TMF8U0liXj4iFB1eDgj1WLf6KUBRObjVM3bljtwOe2soo7qgJYhCUyFVOHmcH7ihEzg=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="Ux01G1548NILZNhOW1DzwO+u6ZjvzzKxOivvCMxiC7XZmptHXlJNBUfc96Viha0YKsYKlu13YE0djZpTovGgWA==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="WY+Tjqst9ii0xhCikHr6xwOlVpI5ruuBWL+lr7ZU47s0QTvSid8FSj4rYFg2KCzJ1/6R6bSQ2QuIk8cIyxGh2A==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="/uDa0eDaX7a0lUn11FlYa7JFkzxyflDNYCPUPeIRsNMI/R+3HElvyHE452xmtSxDBAUbFdrj34Z+f28HKW0cww==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1","user_id":null}}' data-hydro-click-hmac="f4491b5904781299c6aefaeab4e1fa63dc2f5ac241ddc357e634ff36ff0d7ac2" href="/login?return_to=https%3A%2F%2Fgithub.com%2FDSJaswanth%2FHACKBIO_2021_TEAM_EPIGENOMICS1">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1","user_id":null}}' data-hydro-click-hmac="f4491b5904781299c6aefaeab4e1fa63dc2f5ac241ddc357e634ff36ff0d7ac2" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=DSJaswanth%2FHACKBIO_2021_TEAM_EPIGENOMICS1">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-2d525d87-30b1-4607-8e23-fb8a8b506132" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-7761c195-a2a8-4ad9-9bf5-51c4dde22dc2" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-7761c195-a2a8-4ad9-9bf5-51c4dde22dc2" id="tooltip-2d525d87-30b1-4607-8e23-fb8a8b506132" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/DSJaswanth/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/DSJaswanth" rel="author">
            DSJaswanth
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1">
            HACKBIO_2021_TEAM_EPIGENOMICS1
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1","user_id":null}}' data-hydro-click-hmac="39016fb8267d7bef47511c13fdbef670865e8d2c3185cad86438dd3866419063" data-view-component="true" href="/login?return_to=%2FDSJaswanth%2FHACKBIO_2021_TEAM_EPIGENOMICS1" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":397162940,"auth_type":"LOG_IN","originating_url":"https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1","user_id":null}}' data-hydro-click-hmac="9140d21452d8900faf372ccca3594d790344ee9a69a6da96491f7be1ec626014" data-view-component="true" href="/login?return_to=%2FDSJaswanth%2FHACKBIO_2021_TEAM_EPIGENOMICS1" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="5">
             5
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":397162940,"auth_type":"LOG_IN","originating_url":"https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1","user_id":null}}' data-hydro-click-hmac="625bfc6d78d513a1f723fbbd8c5b13c1207d036992523bdd2030c02fae2de26a" data-view-component="true" href="/login?return_to=%2FDSJaswanth%2FHACKBIO_2021_TEAM_EPIGENOMICS1" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="1 user starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="1">
              1
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <p class="f4 mb-3">
          THIS REPOSITORY CONTAINS STAGE 2 TASK FROM HACKBIO INTERNSHIP 2021
         </p>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            1
           </span>
           star
          </a>
          <a class="Link--secondary no-underline mr-3" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            5
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":397162940,"auth_type":"LOG_IN","originating_url":"https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1","user_id":null}}' data-hydro-click-hmac="625bfc6d78d513a1f723fbbd8c5b13c1207d036992523bdd2030c02fae2de26a" data-view-component="true" href="/login?return_to=%2FDSJaswanth%2FHACKBIO_2021_TEAM_EPIGENOMICS1" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1","user_id":null}}' data-hydro-click-hmac="39016fb8267d7bef47511c13fdbef670865e8d2c3185cad86438dd3866419063" data-view-component="true" href="/login?return_to=%2FDSJaswanth%2FHACKBIO_2021_TEAM_EPIGENOMICS1" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="pull-requests-repo-tab-count" title="2">
            2
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-828fdbef-c51a-42b5-b8b9-990fa461d358-list" aria-haspopup="true" aria-labelledby="tooltip-170bb458-f2f8-4679-870b-8dbefacd2903" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-828fdbef-c51a-42b5-b8b9-990fa461d358-button" popovertarget="action-menu-828fdbef-c51a-42b5-b8b9-990fa461d358-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-828fdbef-c51a-42b5-b8b9-990fa461d358-button" id="tooltip-170bb458-f2f8-4679-870b-8dbefacd2903" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-828fdbef-c51a-42b5-b8b9-990fa461d358-button" anchor-offset="normal" data-view-component="true" id="action-menu-828fdbef-c51a-42b5-b8b9-990fa461d358-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-828fdbef-c51a-42b5-b8b9-990fa461d358-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-828fdbef-c51a-42b5-b8b9-990fa461d358-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1" id="item-c4b3bd46-32ae-404c-8ea4-484cc2994df1" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/issues" id="item-a729cd98-4415-44f5-9238-30ceb678f775" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/pulls" id="item-67fd8865-0e45-4802-80b8-c4f067b45e0c" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/actions" id="item-fb50f22c-7234-4425-b4c5-72ffee4e949f" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/projects" id="item-977df7d0-5411-4ef8-9099-6d45707e8f29" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/security" id="item-60897c97-fa92-4a0a-9b49-201889ee49fe" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/pulse" id="item-b381fdac-4d5d-4a6e-9f11-f5ee3b5b2f9e" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/spoofed_commit_check/8687815c39db9d264408c535538738efe3890ed5">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":397162940,"defaultBranch":"main","name":"HACKBIO_2021_TEAM_EPIGENOMICS1","ownerLogin":"DSJaswanth","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-08-17T08:02:41.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/88287856?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"main","listCacheKey":"v0:1629544080.3883462","canEdit":false,"refType":"branch","currentOid":"8687815c39db9d264408c535538738efe3890ed5"},"tree":{"items":[{"name":"images","path":"images","contentType":"directory"},{"name":"20210821_120019_0000.png","path":"20210821_120019_0000.png","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"},{"name":"over_represented.png","path":"over_represented.png","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":4,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/archive/refs/heads/main.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=397162940"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"177","overviewFiles":[{"displayName":"README.md","repoName":"HACKBIO_2021_TEAM_EPIGENOMICS1","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cp align=\"center\" dir=\"auto\"\u003e HackBio2021-Team Epigenomics1 \u003c/p\u003e\u003c/h1\u003e\u003ca id=\"user-content--hackbio2021-team-epigenomics1-\" class=\"anchor\" aria-label=\"Permalink:  HackBio2021-Team Epigenomics1 \" href=\"#-hackbio2021-team-epigenomics1-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://camo.githubusercontent.com/21d4e121cd0c0070fd9fad7b6e5e544e5a2497c19867dce10ff4e179cfc0b6ee/68747470733a2f2f76697369746f722d62616467652e6c616f62692e6963752f62616467653f706167655f69643d44534a6177616e74682e4841434b42494f5f323032315f5445414d5f45504947454e4f4d49435331\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/21d4e121cd0c0070fd9fad7b6e5e544e5a2497c19867dce10ff4e179cfc0b6ee/68747470733a2f2f76697369746f722d62616467652e6c616f62692e6963752f62616467653f706167655f69643d44534a6177616e74682e4841434b42494f5f323032315f5445414d5f45504947454e4f4d49435331\" alt=\"Visitor\" data-canonical-src=\"https://visitor-badge.laobi.icu/badge?page_id=DSJawanth.HACKBIO_2021_TEAM_EPIGENOMICS1\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://camo.githubusercontent.com/355b1324636494fc18ef6c7cddade57521133de268036df6e63c3dec3411c721/68747470733a2f2f6d656469612d657870312e6c6963646e2e636f6d2f646d732f696d6167652f43353631424151484b63565147626365644f412f636f6d70616e792d6261636b67726f756e645f31303030302f302f313539383439313437333538383f653d3231353930323434303026763d6265746126743d727845436a76515f5953633238446e306e39594f74446f46466d76586a6174526971635f5f43326d705530\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/355b1324636494fc18ef6c7cddade57521133de268036df6e63c3dec3411c721/68747470733a2f2f6d656469612d657870312e6c6963646e2e636f6d2f646d732f696d6167652f43353631424151484b63565147626365644f412f636f6d70616e792d6261636b67726f756e645f31303030302f302f313539383439313437333538383f653d3231353930323434303026763d6265746126743d727845436a76515f5953633238446e306e39594f74446f46466d76586a6174526971635f5f43326d705530\" alt=\"HackBio\" data-canonical-src=\"https://media-exp1.licdn.com/dms/image/C561BAQHKcVQGbcedOA/company-background_10000/0/1598491473588?e=2159024400\u0026amp;v=beta\u0026amp;t=rxECjvQ_YSc28Dn0n9YOtDoFFmvXjatRiqc__C2mpU0\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e \u003cbr\u003e\nImage source: HackBio (LinkedIn)\u003c/p\u003e\n  \u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://camo.githubusercontent.com/f57436a5b05808a94ad67f842d95fe230b4f0d74bd54698401fbdcdc9c8eb9ae/68747470733a2f2f692e726564642e69742f736330723779773633623233312e676966\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/f57436a5b05808a94ad67f842d95fe230b4f0d74bd54698401fbdcdc9c8eb9ae/68747470733a2f2f692e726564642e69742f736330723779773633623233312e676966\" data-animated-image=\"\" data-canonical-src=\"https://i.redd.it/sc0r7yw63b231.gif\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e \n\u003c/p\u003e\u003cp dir=\"auto\"\u003eThis is ***Team EPIGENOMICS1 *** for Hackbio 2021 virtual internship.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e About: Hackbio Internship is a 5-weeks virtual research internship that is practice oriented and focused on equipping scientists globally with advanced bioinformatics and      computational biology skills.\u003c/li\u003e\n\u003cli\u003e Hackbio official website: \u003ca href=\"https://thehackbio.com/\" rel=\"nofollow\"\u003ehttps://thehackbio.com/\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e Contact: \u003ca href=\"mailto:contact@hackbio.com\"\u003econtact@hackbio.com\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e We are a diverse team of awesome 17 members .\u003c/li\u003e\n\u003c/ul\u003e\n\n\u003cbr\u003e  \n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e STAGE-2-PROBE \u003c/h2\u003e\u003ca id=\"user-content-scroll-stage-2-probe-scroll\" class=\"anchor\" aria-label=\"Permalink: :scroll: STAGE-2-PROBE :scroll:\" href=\"#scroll-stage-2-probe-scroll\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cp align=\"center\" dir=\"auto\"\u003e THE TASK WE NEED TO DO IN THIS STAGE\u003c/p\u003e\u003c/h2\u003e\u003ca id=\"user-content--the-task-we-need-to-do-in-this-stage\" class=\"anchor\" aria-label=\"Permalink:  THE TASK WE NEED TO DO IN THIS STAGE\" href=\"#-the-task-we-need-to-do-in-this-stage\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e SEARCHING FOR DESIRED TUTORIALS OF BIOSTACK EPIGENOMICS.\u003cbr\u003e\n HAVING A WONDERFUL WORKFLOW DESIGN.\u003cbr\u003e\n REPRODUCING THE ANALYSIS.\u003cbr\u003e\n CREATING A COMPREHENSIVE MARKDOWN OF THE ANALYSIS.\u003cbr\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cp align=\"center\" dir=\"auto\"\u003e \u003cstrong\u003eWE CHOSE ATAC-Seq Analysis\u003c/strong\u003e\u003c/p\u003e\u003c/h2\u003e\u003ca id=\"user-content--we-chose-atac-seq-analysis\" class=\"anchor\" aria-label=\"Permalink: WE CHOSE ATAC-Seq Analysis\" href=\"#-we-chose-atac-seq-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cstrong\u003eWhat ATAC-Seq is\u003c/strong\u003e\u003c/h3\u003e\u003ca id=\"user-content-what-atac-seq-is\" class=\"anchor\" aria-label=\"Permalink: What ATAC-Seq is\" href=\"#what-atac-seq-is\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eATAC-Seq, acronym of \"Assay for Transposase-Accessible Chromatin using Sequencing\", is a technique for assessing chromatin accessibility and how it affects gene expression. The chromatin consists of several nucleosome units. A nucleosome is a complex of DNA and histone proteins that keeps DNA compact through packaging. For transcription to occur, DNA must be loosened from the nucleosomes to make it accessible to transcription factors. This accessibility of DNA mediated by chromatin is a key component of epigenetics.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn this project, we are reproducing the ATAC-Seq analysis \u003ca href=\"https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html#trimming-reads\" rel=\"nofollow\"\u003etutorial\u003c/a\u003e from the Galaxy Project. We've reproduced it completely with Linux Command Line tools too.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eWhat Our Analysis is About?\u003c/h4\u003e\u003ca id=\"user-content-what-our-analysis-is-about\" class=\"anchor\" aria-label=\"Permalink: What Our Analysis is About?\" href=\"#what-our-analysis-is-about\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eUsing ATAC-Seq analysis, here we will compare the predicted open chromatin regions to the known binding sites of CTCF (a DNA-binding protein implicated in 3D structure). CTCF is known to bind to thousands of sites in the genome and thus it can be used as a positive control for assessing if the ATAC-Seq experiment is good quality.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAbout Dataset\u003c/h4\u003e\u003ca id=\"user-content-about-dataset\" class=\"anchor\" aria-label=\"Permalink: About Dataset\" href=\"#about-dataset\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eThe dataset we will use here is a subset of the data used in the study of Buenrostro et al. 2013. The data from the original dataset is downsized to 200,000 randomly selected reads and about 200,000 reads pairs that will map to chromosome 22. We are only focusing on chromosome 22 so that we can handle the analysis in less time. Binding sites of CTCF identified by ChIP in the same cell line from ENCODE (ENCSR000AKB, dataset ENCFF933NTR) will also be used.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cstrong\u003eWORKFLOW\u003c/strong\u003e\u003c/h2\u003e\u003ca id=\"user-content-workflow\" class=\"anchor\" aria-label=\"Permalink: WORKFLOW\" href=\"#workflow\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cp align=\"center\" dir=\"auto\"\u003e GRAPHICAL WORKFLOW DESIGN\u003c/p\u003e\u003c/h3\u003e\u003ca id=\"user-content--graphical-workflow-design\" class=\"anchor\" aria-label=\"Permalink:  GRAPHICAL WORKFLOW DESIGN\" href=\"#-graphical-workflow-design\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n \u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/20210821_120019_0000.png\"\u003e\u003cimg src=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/20210821_120019_0000.png\" width=\"800px\" height=\"600px\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSTEP 1 :- PREPROCESSING\u003c/h3\u003e\u003ca id=\"user-content-step-1---preprocessing\" class=\"anchor\" aria-label=\"Permalink: STEP 1 :- PREPROCESSING\" href=\"#step-1---preprocessing\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA) DATASET\u003c/h4\u003e\u003ca id=\"user-content-a-dataset\" class=\"anchor\" aria-label=\"Permalink: A) DATASET\" href=\"#a-dataset\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\u003cstrong\u003e\u003c/strong\u003e\u003c/em\u003e\u003c/p\u003e\u003cdetails\u003e\u003csummary\u003e\u003cem\u003e\u003cstrong\u003eData Upload for galaxy\u003c/strong\u003e\u003c/em\u003e\u003c/summary\u003e\u003cem\u003e\u003cstrong\u003e\u003cbr\u003e\u003c/strong\u003e\u003c/em\u003e\u003cp dir=\"auto\"\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\u003cstrong\u003eCreate a new history\u003c/strong\u003e\u003c/em\u003e\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eAdd or import DATASETS (2 fastq.gz files and 1 bed.gz file) for analysis via \u003ca href=\"https://zenodo.org/record/3862793#.YRze2XUvNH4\" rel=\"nofollow\"\u003eLINK\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eAdd tags to R1 and R2 files. To add tags:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eClick on the dataset\u003c/li\u003e\n\u003cli\u003eClick on the tag icon\u003c/li\u003e\n\u003cli\u003eAdd a tag starting with \u003cstrong\u003e#: FOR  #SRR891268_R1\u003c/strong\u003e FILE to the R1 file and  FOR \u003cstrong\u003e#SRR891268_R2\u003c/strong\u003e FILE to the R2 file.\u003c/li\u003e\n\u003cli\u003eCheck that tag appears below the dataset name\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\u003cstrong\u003eCheck datatype of files and edit if necessary\u003c/strong\u003e\u003c/em\u003e\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eClick on the pencil icon for the dataset to edit its attributes\u003c/li\u003e\n\u003cli\u003eIn the central panel, click on the Datatypes tab on the top\u003c/li\u003e\n\u003cli\u003eSelect correct datatype (fastqsanger.gz for the FASTQ files, and encodepeak for the bed.gz file)\u003c/li\u003e\n\u003cli\u003eClick the Change datatype button\u003c/li\u003e\n\u003c/ol\u003e\n \u003c/details\u003e \n\u003cp dir=\"auto\"\u003e\u003cem\u003e\u003cstrong\u003e\u003c/strong\u003e\u003c/em\u003e\u003c/p\u003e\u003cdetails\u003e\u003csummary\u003e\u003cem\u003e\u003cstrong\u003e Get Data for LINUX \u003c/strong\u003e\u003c/em\u003e\u003c/summary\u003e\u003cem\u003e\u003cstrong\u003e\u003cbr\u003e\u003c/strong\u003e\u003c/em\u003e\u003cp dir=\"auto\"\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"  wget https://zenodo.org/record/3862793/files/ENCFF933NTR.bed.gz\"\u003e\u003cpre\u003e  \u003cspan class=\"pl-s1\"\u003ewget\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ehttps\u003c/span\u003e:\u003cspan class=\"pl-c1\"\u003e//\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ezenodo\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eorg\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003erecord\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3862793\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003efiles\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eENCFF933NTR\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"  wget https://zenodo.org/record/3862793/files/SRR891268_chr22_enriched_R1.fastq.gz\"\u003e\u003cpre\u003e  \u003cspan class=\"pl-s1\"\u003ewget\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ehttps\u003c/span\u003e:\u003cspan class=\"pl-c1\"\u003e//\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ezenodo\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eorg\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003erecord\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3862793\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003efiles\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eSRR891268_chr22_enriched_R1\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"  wget https://zenodo.org/record/3862793/files/SRR891268_chr22_enriched_R2.fastq.gz\"\u003e\u003cpre\u003e  \u003cspan class=\"pl-s1\"\u003ewget\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ehttps\u003c/span\u003e:\u003cspan class=\"pl-c1\"\u003e//\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ezenodo\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eorg\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003erecord\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3862793\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003efiles\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eSRR891268_chr22_enriched_R2\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\u003cstrong\u003eYou can unzip the sequence files with gunzip\u003c/strong\u003e\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"   gunzip SRR891268_chr22_enriched_R1.fastq.gz \"\u003e\u003cpre\u003e   \u003cspan class=\"pl-s1\"\u003egunzip\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eSRR891268_chr22_enriched_R1\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e \u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"  gunzip SRR891268_chr22_enriched_R2.fastq.gz\"\u003e\u003cpre\u003e  \u003cspan class=\"pl-s1\"\u003egunzip\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eSRR891268_chr22_enriched_R2\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n \u003c/details\u003e \n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eB) Obtain Annotation for hg38 genes\u003c/h4\u003e\u003ca id=\"user-content-b-obtain-annotation-for-hg38-genes\" class=\"anchor\" aria-label=\"Permalink: B) Obtain Annotation for hg38 genes\" href=\"#b-obtain-annotation-for-hg38-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n \u003cdetails\u003e\n \u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eSelect the \u003cstrong\u003eUSCS Main table browser tool\u003c/strong\u003e with the following parameters\u003c/li\u003e\n\u003c/ol\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cb\u003eclade\u003c/b\u003e:Mammal\u003c/li\u003e\n\u003cli\u003e\u003cb\u003egenome\u003c/b\u003e:Human\u003c/li\u003e\n\u003cli\u003e\u003cb\u003eassembly\u003c/b\u003e:Dec. 2013(GRCh38/hg38)\u003c/li\u003e\n\u003cli\u003e\u003cb\u003egroup\u003c/b\u003e:Genes and Gene Prediction\u003c/li\u003e\n\u003cli\u003e\u003cb\u003etrack\u003c/b\u003e:All GENCODE V37\u003c/li\u003e\n\u003cli\u003e\u003cb\u003etable\u003c/b\u003e:Basic\u003c/li\u003e\n\u003cli\u003e\u003cb\u003eregion\u003c/b\u003e:position chr22\u003c/li\u003e\n\u003cli\u003e\u003cb\u003eoutput format\u003c/b\u003e:all fields from selected table\u003c/li\u003e\n\u003cli\u003e\u003cb\u003eSend output to\u003c/b\u003e:Galaxy\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eClick get output\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eClick Send Query to Galaxy\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eSelect the \u003cstrong\u003eCut columns from a table tool\u003c/strong\u003e with the following parameters\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cb\u003eCut columns\u003c/b\u003e: c3,c5,c6,c13,c12,c4\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cb\u003eDelimited by\u003c/b\u003e: Tab\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cb\u003eparam-file From\u003c/b\u003e: UCSC Main on Human: wgEncodeGencodeBasicV37 (chr22:1-50,818,468)\n\u003cem\u003eRename the dataset as chr22 genes\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eClick on the pencil icon for the dataset to edit its attributes\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn the central panel, change the name field\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eClick the Save button\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eChange the datatype to a BED format\u003c/em\u003e\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eClick on the pencil icon for the dataset to edit its attributes\u003c/li\u003e\n\u003cli\u003eIn the central panel, click on the Datatypes tab on the top\u003c/li\u003e\n\u003cli\u003eSelect  \u003cstrong\u003ebed\u003c/strong\u003e\u003c/li\u003e\n\u003cli\u003eClick the Change datatype button\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eClick on the eye icon to check changes effected. There should be matching column names in each column of the dataset.\u003c/p\u003e\n  \u003c/details\u003e\n  \u003cdetails\u003e\u003csummary\u003eLINUX Implementation\u003c/summary\u003e\n   Go to [\u003ca href=\"http://genome.ucsc.edu/cgi-bin/hgTables](http://genome.ucsc.edu/cgi-bin/hgTables)\" rel=\"nofollow\"\u003ehttp://genome.ucsc.edu/cgi-bin/hgTables](http://genome.ucsc.edu/cgi-bin/hgTables)\u003c/a\u003e and set the parameters as-\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cb\u003eclade\u003c/b\u003e:Mammal\u003c/li\u003e\n\u003cli\u003e\u003cb\u003egenome\u003c/b\u003e:Human\u003c/li\u003e\n\u003cli\u003e\u003cb\u003eassembly\u003c/b\u003e:Dec. 2013(GRCh38/hg38)\u003c/li\u003e\n\u003cli\u003e\u003cb\u003egroup\u003c/b\u003e:Genes and Gene Prediction\u003c/li\u003e\n\u003cli\u003e\u003cb\u003etrack\u003c/b\u003e:All GENCODE V37\u003c/li\u003e\n\u003cli\u003e\u003cb\u003etable\u003c/b\u003e:Basic\u003c/li\u003e\n\u003cli\u003e\u003cb\u003eregion\u003c/b\u003e:position chr22\u003c/li\u003e\n\u003cli\u003e\u003cb\u003eoutput format:all fields from selected table\u003c/b\u003e\u003c/li\u003e\u003cb\u003e\n\u003cli\u003e\u003cb\u003eoutput filename:chr22\u003c/b\u003e\u003c/li\u003e\u003cb\u003e\n\u003cli\u003e\u003cb\u003efile type returned:gzipped compressed\u003c/b\u003e\u003c/li\u003e\u003cb\u003e\n\u003c/b\u003e\u003c/b\u003e\u003c/b\u003e\u003c/ul\u003e\u003cb\u003e\u003cb\u003e\u003cb\u003e\n\u003cp dir=\"auto\"\u003eAnd then select \u003cstrong\u003eGet output\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThus, chr22.gz file will be downloaded.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cb\u003eConverting chr22 file into a bed file:\u003c/b\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eUnzip the downloaded chr22.gz using command\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"unzip chr22.gz \"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003eunzip\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003echr22\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e \u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"awk -F \u0026quot;\\t\u0026quot; 'OFS=\u0026quot;\\t\u0026quot; {print $3, $5, $6, $13, $12, $4 \u0026gt; (\u0026quot;chr22.bed\u0026quot;)}' chr22 \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eawk -F \"\\t\" 'OFS=\"\\t\" {print $3, $5, $6, $13, $12, $4 \u0026gt; (\"chr22.bed\")}' chr22 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e(to get only expected columns into a newly created chr22.bed file)\u003c/p\u003e\n  \u003c/b\u003e\u003c/b\u003e\u003c/b\u003e\u003c/details\u003e\u003cb\u003e\u003cb\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eC) QUALITY CONTROL\u003c/h4\u003e\u003ca id=\"user-content-c-quality-control\" class=\"anchor\" aria-label=\"Permalink: C) QUALITY CONTROL\" href=\"#c-quality-control\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n \u003cdetails\u003e\n\u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003cbr\u003e\n  - Select the \u003cb\u003e FastQC tool \u003c/b\u003e with the following parameters \u003cbr\u003e\n   - 1. \"Short read data from your current history\": Choose here either only the SRR891268_R1 file with param-file or use param-files; use Multiple datasets to choose both SRR891268_R1 and SRR891268\\_R2. \u003cbr\u003e\n   - 2. Inspect the web page output of FastQC tool for the SRR891268\\_R1 sample. Check what adapters are found at the end of the reads.\n\u003c/details\u003e  \n \u003cdetails\u003e\n\u003csummary\u003eLinux Implementation\u003c/summary\u003e\n\u003cbr\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eDownload the FastQC module\nNote: FASTQC requires java and javac installed for implementation and you need to run the fastqc file from the folder (using the relative/absolute links to the sequence reads)\u003cbr\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"  sudo apt install default-jre\"\u003e\u003cpre\u003e  \u003cspan class=\"pl-s1\"\u003esudo\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eapt\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003einstall\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003edefault\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ejre\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"  sudo apt install default-jdk\"\u003e\u003cpre\u003e  \u003cspan class=\"pl-s1\"\u003esudo\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eapt\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003einstall\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003edefault\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ejdk\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMake the fastqc an executable file\u003cbr\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"  chmod 755 fastqc\"\u003e\u003cpre\u003e  \u003cspan class=\"pl-s1\"\u003echmod\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e755\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003efastqc\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eRun the fastqc on all sequenced reads from its folder\u003cbr\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"  fastqc SRR891268_chr22_enriched_R1.fastq SRR891268_chr22_enriched_R2.fastq \"\u003e\u003cpre\u003e  \u003cspan class=\"pl-s1\"\u003efastqc\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eSRR891268_chr22_enriched_R1\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eSRR891268_chr22_enriched_R2\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e \u003c/pre\u003e\u003c/div\u003e\n   \u003c/details\u003e\nThe report for each file is generated as an html file and a zip file containing more files that can be customised for reports. Look into the html files.\n\u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/FastQC.PNG\"\u003e\u003cimg src=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/FastQC.PNG\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\u003cp dir=\"auto\"\u003eThe overrepresented sequence segment shows that we have some overrepresented sequence, that is adapter sequence. We have to trim that adapter sequence to move forward with our analysis.\u003c/p\u003e\n  \u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/over_represented.png\"\u003e\u003cimg src=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/over_represented.png\" width=\"1000px\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTRIMMING READ\u003c/h4\u003e\u003ca id=\"user-content-trimming-read\" class=\"anchor\" aria-label=\"Permalink: TRIMMING READ\" href=\"#trimming-read\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdetails\u003e\n\u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003cbr\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSelect the \u003cstrong\u003eCutadapt tool\u003c/strong\u003e with the following parameters\u003c/li\u003e\n\u003c/ul\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\"Single-end or Paired-end reads?\": Paired-end\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eparam-file \"FASTQ/A file #1\": select SRR891268_R1\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eparam-file \"FASTQ/A file #2\": select SRR891268_R2\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \"Read 1 Options\":\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eIn \"3' (End) Adapters\":\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eparam-repeat \"Insert 3' (End) Adapters\"\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Source\": Enter custom sequence\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Enter custom 3' adapter name (Optional if Multiple output is 'No')\": Nextera R1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Enter custom 3' adapter sequence\": CTGTCTCTTATACACATCTCCGAGCCCACGAGAC\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eIn \"Read 2 Options\":\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eIn \"3' (End) Adapters\":\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eparam-repeat \"Insert 3' (End) Adapters\"\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Source\": Enter custom sequence\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Enter custom 3' adapter name (Optional)\": Nextera R2\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Enter custom 3' adapter sequence\": CTGTCTCTTATACACATCTGACGCTGCCGACGA\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eIn \"Filter Options\":\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e\"Minimum length\": 20\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eIn \"Read Modification Options\":\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e\"Quality cutoff\": 20\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eIn \"Output Options\":\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e\"Report\": Yes\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eClick on the galaxy-eye (eye) icon of the report and read the first lines.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eCheck Adapter Removal\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eSelect the \u003cstrong\u003eFast QC tool\u003c/strong\u003e with the following parameters\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\"Short read data from your current history\": select the output of Cutadapt param files; use; Multiple datasets to choose both Read 1 Output and Read 2 Output.\u003c/li\u003e\n\u003cli\u003eClick on the galaxy-eye (eye) icon of the report and read the first lines.\u003c/li\u003e\n\u003c/ol\u003e\n  \u003c/details\u003e\n\u003cdetails\u003e\n\u003csummary\u003eLinux Implementation\u003c/summary\u003e\n\u003cbr\u003e \n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAdapter Trimming\u003c/h5\u003e\u003ca id=\"user-content-adapter-trimming\" class=\"anchor\" aria-label=\"Permalink: Adapter Trimming\" href=\"#adapter-trimming\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThe fastqc report indicates the presence of an overrepresented sequence and fastqc identifies it as \"Nextera Transposase Sequence ''. This sequence is similar to but longer than the one given in the tutorial.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003eSRR891268\\_chr22\\_enriched\\_R1 = CTGTCTCTTATACACATCTCCGAGCCCACGAGACTAAGGCGAATCTCGTA (fastqc)\u003c/code\u003e \u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003eSRR891268\\_chr22\\_enriched\\_R1 = CTGTCTCTTATACACATCTCCGAGCCCACGAGAC (Galaxy tutorial)\u003c/code\u003e\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003eSRR891268\\_chr22\\_enriched\\_R2 = CTGTCTCTTATACACATCTGACGCTGCCGACGAGTGTAGATCTCGGTGGT (fastqc)\u003c/code\u003e\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003eSRR891268\\_chr22\\_enriched\\_R2 = CTGTCTCTTATACACATCTGACGCTGCCGACGA (Galaxy tutorial)\u003c/code\u003e\u003cbr\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAdapter Trimming with Cutadapt\u003c/h5\u003e\u003ca id=\"user-content-adapter-trimming-with-cutadapt\" class=\"anchor\" aria-label=\"Permalink: Adapter Trimming with Cutadapt\" href=\"#adapter-trimming-with-cutadapt\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eInstall cutadapt running-\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"$ sudo apt install cutadapt\"\u003e\u003cpre\u003e$ \u003cspan class=\"pl-s1\"\u003esudo\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eapt\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003einstall\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ecutadapt\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFor paired end trimming-\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"cutadapt -a CTGTCTCTTATACACATCTCCGAGCCCACGAGAC -A CTGTCTCTTATACACATCTGACGCTGCCGACGA --minimum-length 20 -q 20 -o trimmed_1.fastq -p trimmed_2.fastq SRR891268_chr22_enriched_R1.fastq SRR891268_chr22_enriched_R2.fastq\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003ecutadapt\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ea\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eCTGTCTCTTATACACATCTCCGAGCCCACGAGAC\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eA\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eCTGTCTCTTATACACATCTGACGCTGCCGACGA\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eminimum\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003elength\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eq\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eo\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etrimmed_1\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ep\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etrimmed_2\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eSRR891268_chr22_enriched_R1\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eSRR891268_chr22_enriched_R2\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n \u003c/details\u003e\n  \u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/FastQC%20after%20cutadapt.PNG\"\u003e\u003cimg src=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/FastQC%20after%20cutadapt.PNG\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSTEP2 :- MAPPING\u003c/h3\u003e\u003ca id=\"user-content-step2---mapping\" class=\"anchor\" aria-label=\"Permalink: STEP2 :- MAPPING\" href=\"#step2---mapping\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdetails\u003e\n\u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003cbr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cstrong\u003eMapping reads to reference genome\u003c/strong\u003e\u003c/h3\u003e\u003ca id=\"user-content-mapping-reads-to-reference-genome\" class=\"anchor\" aria-label=\"Permalink: Mapping reads to reference genome\" href=\"#mapping-reads-to-reference-genome\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSelect the \u003cstrong\u003eBowtie2\u003c/strong\u003e   \u003cstrong\u003etool\u003c/strong\u003e with the following parameters:\u003c/li\u003e\n\u003c/ul\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\"Is this single or paired library\": Paired-end\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eparam-file \"FASTQ/A file #1\": select the output of Cutadapt tool \"Read 1 Output\"\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eparam-file \"FASTQ/A file #2\": select the output of Cutadapt tool \"Read 2 Output\"\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\"Do you want to set paired-end options?\": Yes\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\"Set the maximum fragment length for valid paired-end alignments\": 1000\n\"Allow mate dovetailing\": Yes\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\"Will you select a reference genome from your history or use a built-in index?\": Use a built-in genome index\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\"Select reference genome\": Human (Homo sapiens): hg38 Canonical\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\"Set read groups information?\": Do not set\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\"Select analysis mode\": 1: Default setting only\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\"Do you want to use presets?\": Very sensitive end-to-end (--very-sensitive)\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\"Do you want to tweak SAM/BAM Options?\": No\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\"Save the bowtie2 mapping statistics to the history\": Yes\n3.Click on the galaxy-eye (eye) icon of the mapping stats.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n  \u003c/details\u003e\n\u003cdetails\u003e\n\u003csummary\u003eLinux Implementation\u003c/summary\u003e\n\u003cbr\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eMapping and Alignment\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePulling the sequence for chromosome 22 for indexing and mapping \u003cbr\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"$ wget --timestamping ;ftp://hgdownload.cse.ucsc.edu/goldenPath/hg38/chromosomes/chr22.fa.gz; -O chr22.fa.gz\"\u003e\u003cpre\u003e$ \u003cspan class=\"pl-s1\"\u003ewget\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003etimestamping\u003c/span\u003e ;\u003cspan class=\"pl-s1\"\u003eftp\u003c/span\u003e:\u003cspan class=\"pl-c1\"\u003e//\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ehgdownload\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ecse\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eucsc\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eedu\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003egoldenPath\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ehg38\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003echromosomes\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003echr22\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efa\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e; \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eO\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003echr22\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efa\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eFor mapping to chr22-\u003c/li\u003e\n\u003c/ul\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eInstall bowtie2 \u003cbr\u003e\u003c/li\u003e\n\u003cli\u003eCreate index for Chromosome 22: \u003cbr\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" bowtie2-build chr22.fa.gz indexed_chr22\"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003ebowtie2\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ebuild\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003echr22\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efa\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eindexed_chr22\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eStart mapping for the parameters specified by Galaxy: \u003cbr\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"bowtie2 --very-sensitive --maxins 1000 --dovetail -x indexed_chr22 -1 trimmed_1.fastq -2 trimmed_2.fastq -S Aligned_output.sam\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003ebowtie2\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003every\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003esensitive\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003emaxins\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003edovetail\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eindexed_chr22\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etrimmed_1\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etrimmed_2\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003efastq\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eS\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eAligned_output\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003esam\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n \u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/Bowtie2%20output.PNG\"\u003e\u003cimg src=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/Bowtie2%20output.PNG\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\u003c/details\u003e    \n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSTEP 3 :- Filtering Mapped Reads\u003c/h3\u003e\u003ca id=\"user-content-step-3---filtering-mapped-reads\" class=\"anchor\" aria-label=\"Permalink: STEP 3 :- Filtering Mapped Reads\" href=\"#step-3---filtering-mapped-reads\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA) Filter Uninformative Reads\u003c/h4\u003e\u003ca id=\"user-content-a-filter-uninformative-reads\" class=\"anchor\" aria-label=\"Permalink: A) Filter Uninformative Reads\" href=\"#a-filter-uninformative-reads\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n \u003cdetails\u003e\n\u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003cem\u003eFiltering of uninformative mapped reads\u003c/em\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSelect the \u003cstrong\u003eFilter BAM datasets on a variety of attributes\u003c/strong\u003e   \u003cstrong\u003etool\u003c/strong\u003e with the following parameters\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eparam-file \"BAM dataset(s) to filter\": Select the output of Bowtie2 tool \"alignments\"\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn \"Condition\":\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eparam-repeat \"Insert Condition\"\u003c/li\u003e\n\u003cli\u003eIn \"Filter\":\u003c/li\u003e\n\u003cli\u003eparam-repeat \"Insert Filter\"\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e\"Select BAM property to filter on\": mapQuality\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Filter on read mapping quality (phred scale)\": \u0026amp;gt;=30\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eparam-repeat \"Insert Filter\"\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e\"Select BAM property to filter on\": isProperPair\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Select properly paired reads\": Yes\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eparam-repeat \"Insert Filter\"\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e\"Select BAM property to filter on\": reference\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Filter on the reference name for the read\": !chrM\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Would you like to set rules?\": No\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eClick on the input and the output BAM files of the filtering step. Check the size of the files.\u003c/p\u003e\n \u003c/details\u003e\n\u003cdetails\u003e\n\u003csummary\u003eLinux Implementation\u003c/summary\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003cem\u003eFiltering of uninformative mapped reads\u003c/em\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eInstall samtools.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" samtools view -q 30 -f 0x2 -b -h Aligned_output.sam \u0026gt; Filtered_output.bam\"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003esamtools\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eview\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eq\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e30\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ef\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0x2\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eb\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eh\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eAligned_output\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003esam\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eFiltered_output\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebam\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ewe will filter out uninformative reads (Mapping quality \u0026amp;gt;= 30 \u0026amp; Properly Paired)\u003c/p\u003e\n \u003c/details\u003e \n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eB) Filter duplicate reads\u003c/h4\u003e\u003ca id=\"user-content-b-filter-duplicate-reads\" class=\"anchor\" aria-label=\"Permalink: B) Filter duplicate reads\" href=\"#b-filter-duplicate-reads\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n  \u003cdetails\u003e\n\u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\u003cstrong\u003e\u003cem\u003eRemove duplicates\u003c/em\u003e\u003c/strong\u003e\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSelect the \u003cstrong\u003eMarkDuplicates\u003c/strong\u003e   \u003cstrong\u003etool\u003c/strong\u003e with the following parameters\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eparam-file \"Select SAM/BAM dataset or dataset collection\": Select the output of Filter tool \"BAM\"\u003c/li\u003e\n\u003cli\u003e\"If true do not write duplicates to the output file instead of writing them with appropriate flags set\": Yes\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eClick on the eye icon of the MarkDuplicate metrics.\u003c/p\u003e\n  \u003c/details\u003e\n \u003cdetails\u003e\n\u003csummary\u003eLinux Implementation\u003c/summary\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003cem\u003eMark Duplicate Reads\u003c/em\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eDownload picard.jar in your working folder from \u003ca href=\"https://github.com/broadinstitute/picard/releases/download/2.26.0/picard.jar\"\u003ehere\u003c/a\u003e \u003cbr\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFrom that directory, run \u003cbr\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"java -jar picard.jar -h\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003ejava\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ejar\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003epicard\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ejar\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eh\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eto check whether it works (you can skip this step) \u003cbr\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFor sorting the output file from last step use- \u003cbr\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"samtools sort -T temp -O bam -o filtered_output_sorted.bam Filtered_output.bam\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003esamtools\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003esort\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eT\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etemp\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eO\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ebam\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eo\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003efiltered_output_sorted\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebam\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eFiltered_output\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebam\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFinally, run \u003cbr\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"java -jar picard.jar MarkDuplicates I=filtered_output_sorted.bam O=marked_dup.bam M=marked_dup.metrics.txt \"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003ejava\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ejar\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003epicard\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ejar\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eMarkDuplicates\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eI\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003efiltered_output_sorted\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebam\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eO\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003emarked_dup\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebam\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eM\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003emarked_dup\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003emetrics\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003etxt\u003c/span\u003e \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cbr\u003efor marking duplicates\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIf you can have a look into the metrics in the metrics.txt file\u003c/p\u003e\n  \u003c/li\u003e\u003c/ul\u003e\u003c/details\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eC) Check Insert Sizes\u003c/h4\u003e\u003ca id=\"user-content-c-check-insert-sizes\" class=\"anchor\" aria-label=\"Permalink: C) Check Insert Sizes\" href=\"#c-check-insert-sizes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eCheck Insert Size tells us the size of the DNA fragment the read pairs came from this step we have to make a plot of the frequencies of the reads in the bam file to observe the peaks around where there are likely Tn5 transposase activities into nucleosome-free regions.\u003cbr\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdetails\u003e\n\n\n\u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003cem\u003ePlot the distribution of fragment sizes\u003c/em\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSelect \u003cstrong\u003ePaired-end histogram tool\u003c/strong\u003e with the following parameters\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eparam-file \"BAM file\": Select the output of MarkDuplicates tool \"BAM output\"\u003c/li\u003e\n\u003cli\u003e\"Lower bp limit (optional)\": 0\u003c/li\u003e\n\u003cli\u003e\"Upper bp limit (optional)\": 1000\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eClick on the galaxy-eye (eye) icon of the lower one of the 2 outputs (the png file).\u003c/p\u003e\n\u003c/details\u003e\n  \u003cdetails\u003e\n\u003csummary\u003eLinux Implementation\u003c/summary\u003e\n\u003cbr\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"   sudo apt install r-base\"\u003e\u003cpre\u003e   \u003cspan class=\"pl-s1\"\u003esudo\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eapt\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003einstall\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003er\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ebase\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"   java -jar picard.jar CollectInsertSizeMetrics I=marked_dup.bam O=chart.txt H=insertSizePlot.pdf M=0.5\"\u003e\u003cpre\u003e   \u003cspan class=\"pl-s1\"\u003ejava\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ejar\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003epicard\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ejar\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eCollectInsertSizeMetrics\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eI\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003emarked_dup\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebam\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eO\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003echart\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003etxt\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eH\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003einsertSizePlot\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003epdf\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eM\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n   \u003c/details\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/Insert sizes.PNG\"\u003e\u003cimg src=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/Insert sizes.PNG\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e \n\u003c/p\u003e\u003cp dir=\"auto\"\u003eTwo peaks can be observed around the 200bp and 400bp from the plot\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSTEP 4 :- PEAK CALLING\u003c/h3\u003e\u003ca id=\"user-content-step-4---peak-calling\" class=\"anchor\" aria-label=\"Permalink: STEP 4 :- PEAK CALLING\" href=\"#step-4---peak-calling\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ePeak calling helps us to identify regions where reads have piled up (peaks) greater than the background read coverage. Using MACS2 we will extend the start sites of the reads by 200bp (100bp in each direction) to assess coverage. Before using MACS2, we need to convert the output BAM file from MarkDuplicate stage to BED format because when we set the extension size in MACS2, it will only consider one read of the pair while here, we would like to use the information from both.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdetails\u003e\n\u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003cbr\u003e  \n\u003cp dir=\"auto\"\u003e\u003cem\u003e\u003cstrong\u003e\u003cem\u003eConvert BAM to BED\u003c/em\u003e\u003c/strong\u003e\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eConvert BAM file (output of MarkDuplicates) into BED format by \u003cstrong\u003ebedtools BAM to BED converter\u003c/strong\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA) MACS2 callpeak  \u003cstrong\u003eMACS2 callpeak\u003c/strong\u003e\u003c/h4\u003e\u003ca id=\"user-content-a-macs2-callpeak--macs2-callpeak\" class=\"anchor\" aria-label=\"Permalink: A) MACS2 callpeak  MACS2 callpeak\" href=\"#a-macs2-callpeak--macs2-callpeak\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eMACS2 callpeak\u003c/strong\u003e with the following parameters:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eAre you pooling Treatment Files?: No\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSelect the output of  \u003cstrong\u003ebedtools BAM to BED\u003c/strong\u003e  converter tool\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eDo you have a Control File?: No\u003c/li\u003e\n\u003cli\u003eFormat of Input Files: Single-end BED\u003c/li\u003e\n\u003cli\u003eEffective genome size: \u003cem\u003eH. sapiens\u003c/em\u003e (2.7e9)\u003c/li\u003e\n\u003cli\u003eBuild Model: Do not build the shifting model (--nomodel)\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSet extension size: 200\u003c/li\u003e\n\u003cli\u003eSet shift size: -100. It needs to be - half the extension size to be centered on the 5'.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eIn Additional Outputs:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eCheck Peaks as tabular file (compatible with MultiQC)\u003c/li\u003e\n\u003cli\u003eCheck Peak summits\u003c/li\u003e\n\u003cli\u003eCheck Scores in bedGraph files\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eIn  Advanced Options:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eComposite broad regions: No broad regions\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eUse a more sophisticated signal processing approach to find subpeak summits in each enriched peak region: Yes\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eHow many duplicate tags at the exact same location are allowed?: all                                                                                                         \u003c/li\u003e\u003c/ul\u003e\u003c/li\u003e\u003c/ul\u003e\u003c/details\u003e\n\n\n\n\u003cdetails\u003e\n\u003csummary\u003e LINUX Implementation\u003c/summary\u003e\n\u003cbr\u003e \n\u003cp dir=\"auto\"\u003eInstall bamtools and convert bam file to bed file using bamtools:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"bedtools bamtobed -i marked_dup.bam \u0026gt; marked_dup.bed\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003ebedtools\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ebamtobed\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emarked_dup\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebam\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emarked_dup\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eInstall macs2 :\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"conda install -c bioconda macs2\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003econda\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003einstall\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ebioconda\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs2\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThen run the command for peak calling:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"macs2 callpeak -t marked_dup.bed -n macs_output -g 50818468 --nomodel --shift -100 --extsize 200 --keep-dup all --call-summits --bdg\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003emacs2\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ecallpeak\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003et\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emarked_dup\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003en\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs_output\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eg\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e50818468\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003enomodel\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eshift\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eextsize\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e200\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ekeep\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003edup\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eall\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ecall\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003esummits\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ebdg\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\u003cstrong\u003eThis will give us the following 5 output files-\u003c/strong\u003e\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e-Macs_output_control_lambda.bdg\n-macs_output_peaks.narrowPeak\n-Macs_output_peaks.xls\n-Macs_output_summits.bed\n-macs_output_treat_pileup.bdg\u003c/p\u003e\n  \u003c/details\u003e                                                                                                                                                           \n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSTEP 5 :-Visualisation of Coverage\u003c/h3\u003e\u003ca id=\"user-content-step-5--visualisation-of-coverage\" class=\"anchor\" aria-label=\"Permalink: STEP 5 :-Visualisation of Coverage\" href=\"#step-5--visualisation-of-coverage\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA) Prepare the Datasets\u003c/h4\u003e\u003ca id=\"user-content-a-prepare-the-datasets\" class=\"anchor\" aria-label=\"Permalink: A) Prepare the Datasets\" href=\"#a-prepare-the-datasets\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003eExtract CTCF peaks on chr22 in intergenic regions\u003c/strong\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eAs our training dataset is focused on chromosome 22 we will only use the CTCF peaks from chr22. We expect to have ATAC-seq coverage at TSS but only good ATAC-seq have coverage on intergenic CTCF. Indeed, the CTCF protein is able to position nucleosomes and creates a region depleted of nucleosome of around 120bp. This is smaller than the 200bp nucleosome-free region around TSS and also probably not present in all cells. Thus it is more difficult to get enrichment.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdetails\u003e\n\u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003cbr\u003e     \n\u003cp dir=\"auto\"\u003eIn order to get the list of intergenic CTCF peaks of chr22, select the peaks on chr22 and then exclude the one which overlap with genes.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003eFilter data on any column using simple expressions\u003c/strong\u003e   with the following parameters:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eFilter : Select the first dataset: ENCFF933NTR.bed.gz\u003c/li\u003e\n\u003cli\u003eWith following condition: c1=='chr22'\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003cstrong\u003ebedtools Intersect intervals find overlapping intervals in various ways\u003c/strong\u003e with the following parameters:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eFile A to intersect with B: Select the output of  \u003cstrong\u003eFilter\u003c/strong\u003e  data on any column using simple expressions tool\u003c/li\u003e\n\u003cli\u003eCombined or separate output files: One output file per 'input B' file\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eFile B to intersect with A: Select the dataset chr22 genes\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eWhat should be written to the output file?: Write the original entry in A for each overlap (-wa)\u003c/li\u003e\n\u003cli\u003eRequired overlap: Default: 1bp\u003c/li\u003e\n\u003cli\u003eReport only those alignments that  \u003cstrong\u003edo not\u003c/strong\u003e  overlap with file(s) B: Yes\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eRename the datasets intergenic CTCF peaks chr22.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/details\u003e \n\u003cdetails\u003e\n\u003csummary\u003eLINUX Implementation\u003c/summary\u003e\n\u003cbr\u003e                                                                                                                                                                 \n1. Filter only data for chr22 from file using \n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"grep -w \u0026quot;chr22\u0026quot; ENCFF933NTR.bed \u0026gt;\u0026gt; file_A.bed\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003egrep\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ew\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\"chr22\"\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eENCFF933NTR\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003efile_A\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e2.Extract filtered chrr22 (as c1) into a new file\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"grep c1 ENCFF933NTR_filt.bed \u0026gt; ENCFF933NTR_chr22.bed\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003egrep\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ec1\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eENCFF933NTR_filt\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eENCFF933NTR_chr22\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e3.Replace c1 with chr22\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"sed 's/c1/chr22/' ENCFF933NTR_chr22.bed \u0026gt; ENCFF933NTR_CHR22genes.bed\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003esed\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e's/c1/chr22/'\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eENCFF933NTR_chr22\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eENCFF933NTR_CHR22genes\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e4.bedtools Intersect intervals find overlapping intervals :\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"bedtools intersect -v -a ENCFF933NTR_CHR22genes.bed -b chr22_genes.bed \u0026gt; intergenic_CTCF_peaks_chr22\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003ebedtools\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eintersect\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ev\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ea\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eENCFF933NTR_CHR22genes\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eb\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003echr22_genes\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eintergenic_CTCF_peaks_chr22\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n \u003c/details\u003e \n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eB)Convert bedgraph from MACS2 to bigwig\u003c/h4\u003e\u003ca id=\"user-content-bconvert-bedgraph-from-macs2-to-bigwig\" class=\"anchor\" aria-label=\"Permalink: B)Convert bedgraph from MACS2 to bigwig\" href=\"#bconvert-bedgraph-from-macs2-to-bigwig\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eTo visualise any region of the genome very quickly, we need to change the bedgraph format from MACS2 output to binary bigwig format.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdetails\u003e\n\u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003cbr\u003e                                     \n\u003cstrong\u003eWig/BedGraph-to-bigWig\u003c/strong\u003e with the following parameters:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eConvert: Select the output of  \u003cstrong\u003eMACS2\u003c/strong\u003e  tool (Bedgraph Treatment).\u003c/li\u003e\n\u003cli\u003eConverter settings to use: Default\u003c/li\u003e\n\u003cli\u003eRename the datasets MACS2 bigwig.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/details\u003e\n\u003cdetails\u003e\n\u003csummary\u003eLINUX Implementation\u003c/summary\u003e\n\u003cbr\u003e        \n Install \u003cstrong\u003ebedGraphtoBigWig\u003c/strong\u003e and go through the following commands for converting the output bedGraph file from macs2 to bigwig (refer to this link if you want to understand the commands [\u003ca href=\"https://www.biostars.org/p/176875/](https://www.biostars.org/p/176875/)\" rel=\"nofollow\"\u003ehttps://www.biostars.org/p/176875/](https://www.biostars.org/p/176875/)\u003c/a\u003e )\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" awk 'NR!=1' macs_output_treat_pileup.bdg \u0026gt; macs.deheader.bedGraph\"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003eawk\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e'NR!=1'\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs_output_treat_pileup\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebdg\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003edeheader\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebedGraph\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" sort -k1,1 -k2,2n macs.deheader.bedGraph \u0026gt; macs.sorted.bedGraph \"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003esort\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ek1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ek2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003en\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003edeheader\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebedGraph\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003esorted\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebedGraph\u003c/span\u003e \u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" touch chrom22.sizes\"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003etouch\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003echrom22\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003esizes\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" nano hg19.chrom.sizes\"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003enano\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ehg19\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003echrom\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003esizes\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e write only one line (tab delimited) in this file chr22 51304566\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" awk '{print $1,$2,$3,$4}' macs.sorted.bedGraph \u0026gt; macs.sorted.4.bedGraph\"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003eawk\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e'{print $1,$2,$3,$4}'\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003esorted\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebedGraph\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003esorted\u003c/span\u003e.\u003cspan class=\"pl-c1\"\u003e4.\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ebedGraph\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" bedGraphToBigWig macs.sorted.4.bedGraph hg19.chrom.sizes macs.bw \"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003ebedGraphToBigWig\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003esorted\u003c/span\u003e.\u003cspan class=\"pl-c1\"\u003e4.\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ebedGraph\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ehg19\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003echrom\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003esizes\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebw\u003c/span\u003e \u003c/pre\u003e\u003c/div\u003e\n \u003c/details\u003e                                                                                                                                                    \n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eC) Create heatmap of coverage at TSS with deepTool\u003c/h4\u003e\u003ca id=\"user-content-c-create-heatmap-of-coverage-at-tss-with-deeptool\" class=\"anchor\" aria-label=\"Permalink: C) Create heatmap of coverage at TSS with deepTool\" href=\"#c-create-heatmap-of-coverage-at-tss-with-deeptool\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eFor checking the coverage on specific regions, we can compute heatmap. ComputeMatrix \u0026amp; plotHeatmap from deepTools are useful for this purpose. We will here make a heatmap centered on the transcription start sites (TSS) and another one centered on intergenic CTCF peaks.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdetails\u003e\n\u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003cbr\u003e                                                                                                                                                               \n- \u003cstrong\u003ecomputeMatrix\u003c/strong\u003e with the following parameters:\n  - In  Select regions:\n    - Insert Select regions\n      - Regions to plot: Select the dataset chr22 genes\n  - Sample order matters: No\n    - Score file: Select the output of  **Wig/BedGraph-to-bigWig**  tool that should be named MACS2 bigwig.\n  - computeMatrix has two main output options: reference-point\n  - The reference point for the plotting: beginning of region (e.g. TSS)\n  - Show advanced output settings: no\n  - Show advanced options: yes\n  - Convert missing values to 0?: Yes\n  1. **Plot with plotHeatmap**\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003eplotHeatmap\u003c/strong\u003e   with the following parameters:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eMatrix file from the computeMatrix tool: Select the output of  \u003cstrong\u003ecomputeMatrix\u003c/strong\u003e  tool.\u003c/li\u003e\n\u003cli\u003eShow advanced output settings: no\u003c/li\u003e\n\u003cli\u003eShow advanced options: no\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThe same is repeated for the intergenic CTCF peaks.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eGenerate the matrix\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ecomputeMatrix\u003c/strong\u003e with the following parameters:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eIn Select regions:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eInsert Select regions\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eRegions to plot: Select the dataset intergenic CTCF peaks chr22\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eSample order matters: No\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eScore file: Select the output of  \u003cstrong\u003eWig/BedGraph-to-bigWig\u003c/strong\u003e  tool that should be named MACS2 bigwig.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eWould you like custom sample labels?: No, use sample names in the history\u003c/li\u003e\n\u003cli\u003ecomputeMatrix has two main output options: reference-point\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eThe reference point for the plotting: center of region\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eShow advanced output settings: no\u003c/li\u003e\n\u003cli\u003eShow advanced options: yes\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eConvert missing values to 0?: Yes\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eplotHeatmap\u003c/strong\u003e   with the following parameters:\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eMatrix file from the computeMatrix tool: Select the output of  \u003cstrong\u003ecomputeMatrix\u003c/strong\u003e  tool.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eShow advanced output settings: no\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eShow advanced options: yes\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eIn Colormap to use for each sample:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eInsert Colormap to use for each sample\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eColor map to use for the heatmap: your choice\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eThe x-axis label: distance from peak center (bp)\u003c/li\u003e\n\u003cli\u003eThe y-axis label for the top panel: CTCF peaks\u003c/li\u003e\n\u003cli\u003eReference point label: peak center\u003c/li\u003e\n\u003cli\u003eLabels for the regions plotted in the heatmap: CTCF_peaks\u003c/li\u003e\n\u003cli\u003eDid you compute the matrix with more than one groups of regions?: Yes, I used multiple groups of regions\n\u003c/li\u003e\u003c/ul\u003e\u003c/li\u003e\u003c/ul\u003e\u003c/details\u003e  \n\n\n\n\n\u003cdetails\u003e\n\u003csummary\u003eLINUX Implementation\u003c/summary\u003e\n\u003cbr\u003e                                                                                                                                                    \n- Using \u003cstrong\u003ecomputeMatrix\u003c/strong\u003e generate the matrix\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eRemove the first header line from chr22.bed file\u003c/li\u003e\n\u003cli\u003eThen run\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"computeMatrix reference-point --referencePoint TSS -R chr22.bed -S macs.bw --missingDataAsZero -o output_from_computeMatrix.gz\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003ecomputeMatrix\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ereference\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003epoint\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ereferencePoint\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eTSS\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eR\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003echr22\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eS\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebw\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003emissingDataAsZero\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eo\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eoutput_from_computeMatrix\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eplotHeatmap will generate the plot using the output of computeMatrix\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" plotHeatmap -m output_from_computeMatrix.gz -out plotHeatMap.png\"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003eplotHeatmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003em\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eoutput_from_computeMatrix\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eout\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eplotHeatMap\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003epng\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eRepeating the previous two steps for plotting \u003cstrong\u003eCTCF peaks of chr22 in intergenic regions\u003c/strong\u003e with slight moderation:\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" computeMatrix reference-point --referencePoint center -R intergenic_ctcf_peaks_chr22 -S macs.bw --missingDataAsZero -o peak_output_from_computeMatrix.gz\"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003ecomputeMatrix\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ereference\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003epoint\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ereferencePoint\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ecenter\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eR\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eintergenic_ctcf_peaks_chr22\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-v\"\u003eS\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emacs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebw\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003emissingDataAsZero\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eo\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003epeak_output_from_computeMatrix\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" plotHeatmap -m peak_output_from_computeMatrix.gz -out intragenic_plotHeatMap.png\"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003eplotHeatmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003em\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003epeak_output_from_computeMatrix\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003egz\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eout\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eintragenic_plotHeatMap\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003epng\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/details\u003e  \n\u003cp dir=\"auto\"\u003eIn the generated heatmaps, each line will be a transcript. The coverage will be summarized with a color code from red (no coverage) to blue (maximum coverage). All TSS will be aligned in the middle of the figure and only the 2 kb around the TSS will be displayed. Another plot, on top of the heatmap, will show the mean signal at the TSS. There will be one heatmap per bigwig.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor TSS, our data gives the following heatmap-\u003c/p\u003e\n \u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/MACS2%20bigwig.PNG\"\u003e\u003cimg src=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/MACS2%20bigwig.PNG\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e \n\u003c/p\u003e\u003cp dir=\"auto\"\u003eThe plot on top shows a non-symmetric pattern that is higher on the left, which is expected as usually the promoter of active genes is accessible.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor CTCF peaks of chr22 in intergenic regions, the following heatmap is generated from our data-\u003c/p\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/plotHeatmap%20output.PNG\"\u003e\u003cimg src=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/plotHeatmap%20output.PNG\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e \n\u003c/p\u003e\u003cp dir=\"auto\"\u003eThis heatmap is showing a much more symmetric pattern.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eD)Visualise Regions with pyGenomeTracks\u003c/h4\u003e\u003ca id=\"user-content-dvisualise-regions-with-pygenometracks\" class=\"anchor\" aria-label=\"Permalink: D)Visualise Regions with pyGenomeTracks\" href=\"#dvisualise-regions-with-pygenometracks\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eIn order to visualise a specific region (e.g. the gene \u003cem\u003eRAC2\u003c/em\u003e), we will use pyGenomeTracks\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdetails\u003e\n\u003csummary\u003eGalaxy Implementation\u003c/summary\u003e\n\u003cbr\u003e       \n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003epyGenomeTracks\u003c/strong\u003e  Tool with the following parameters:\u003c/li\u003e\n\u003cli\u003eRegion of the genome to limit the operation: chr22:37,193,000-37,252,000\u003c/li\u003e\n\u003cli\u003eIn Include tracks in your plot:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eInsert Include tracks in your plot\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eChoose style of the track: Bigwig track\n\u003col dir=\"auto\"\u003e\n\u003cli\u003ePlot title: Coverage from MACS2 (extended +/-100bp)\u003c/li\u003e\n\u003cli\u003eTrack file(s) bigwig format: Select the output of Wig/BedGraph-to-bigWig tool called MACS2 bigwig.\u003c/li\u003e\n\u003cli\u003eColor of track: Select the color of your choice\u003c/li\u003e\n\u003cli\u003eMinimum value: 0\u003c/li\u003e\n\u003cli\u003eheight: 5\u003c/li\u003e\n\u003cli\u003eShow visualization of data range: Yes\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eInsert Include tracks in your plot\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eChoose style of the track: NarrowPeak track\n\u003col dir=\"auto\"\u003e\n\u003cli\u003ePlot title: Peaks from MACS2 (extended +/-100bp)\u003c/li\u003e\n\u003cli\u003eTrack file(s) encodepeak or bed format: Select the output of MACS2 tool (narrow Peaks).\u003c/li\u003e\n\u003cli\u003eColor of track: Select the color of your choice\u003c/li\u003e\n\u003cli\u003edisplay to use: box: Draw a box\u003c/li\u003e\n\u003cli\u003ePlot labels (name, p-val, q-val): No\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eInsert Include tracks in your plot\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eChoose style of the track: Gene track / Bed track\n\u003col dir=\"auto\"\u003e\n\u003cli\u003ePlot title: Genes\u003c/li\u003e\n\u003cli\u003eTrack file(s) bed or gtf format: chr22 genes\u003c/li\u003e\n\u003cli\u003eColor of track: Select the color of your choice\u003c/li\u003e\n\u003cli\u003eheight: 5\u003c/li\u003e\n\u003cli\u003ePlot labels: yes\u003c/li\u003e\n\u003cli\u003ePut all labels inside the plotted region: Yes\u003c/li\u003e\n\u003cli\u003eAllow to put labels in the right margin: Yes\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eInsert Include tracks in your plot\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eChoose style of the track: NarrowPeak track\n\u003col dir=\"auto\"\u003e\n\u003cli\u003ePlot title: CTCF peaks\u003c/li\u003e\n\u003cli\u003eTrack file(s) encodepeak or bed format: Select the first dataset: ENCFF933NTR.bed.gz\u003c/li\u003e\n\u003cli\u003eColor of track: Select the color of your choice\u003c/li\u003e\n\u003cli\u003edisplay to use: box: Draw a box\u003c/li\u003e\n\u003cli\u003ePlot labels (name, p-val, q-val): No\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eparam-repeat Insert Include tracks in your plot\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eChoose style of the track: X-axis\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n \u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/pyGenome%20output.PNG\"\u003e\u003cimg src=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/pyGenome%20output.PNG\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e \n\u003c/p\u003e\u003c/details\u003e\n \u003cdetails\u003e\n\u003csummary\u003eLinux Implementation\u003c/summary\u003e\n\u003cbr\u003e                                     \n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSet up the config.ini file with the following contents-\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003e[test bedgraph]\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003efile = macs.bw\u003c/strong\u003e \u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e \u003cstrong\u003ecolor = blue\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003eheight = 5\u003c/strong\u003e \u003c/i\u003e\n\u003ci\u003e\u003cstrong\u003etitle = Coverage from MACS2 (extended +/-100bp)\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e \u003cstrong\u003emin_value = 0\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003e[spacer]\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003eheight = 0.5\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003e[narrow]\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003efile = intergenic_ctcf_peaks_chr22.encodepeak\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003eline_width = 2\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003etitle = Peaks from MACS2 (extended +/-100bp)\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003etype = box\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003ecolor = red\u003c/strong\u003e \u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003eshow_labels = false\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003efile_type = narrow_peak\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003e[spacer]\u003c/strong\u003e\u003c/i\u003e\n\u003ci\u003e\u003cstrong\u003eheight = 0.5\u003c/strong\u003e \u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003e[genes 0]\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003efile = chr22.bed\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003eheight = 7\u003c/strong\u003e \u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003etitle = genes\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003eheight = 5\u003c/strong\u003e \u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003ecolor = #ffbbff\u003c/strong\u003e \u003c/i\u003e\n\u003ci\u003e\u003cstrong\u003e[spacer]\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003eheight = 0.5\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003e[narrow 1]\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003efile = ENCFF933NTR_sorted.bed\u003c/strong\u003e \u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003ecolor = #A020F0\u003c/strong\u003e \u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003eline_width = 2\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003etitle = CTCF peaks\u003c/strong\u003e \u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003etype = box\u003c/strong\u003e \u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003eshow_labels = false\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ci\u003e\u003cstrong\u003e[x-axis]\u003c/strong\u003e\u003c/i\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSort ENCFF933NTR.bed file-\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" sort -k 1,1 -k2,2n ENCFF933NTR.bed \u0026gt; ENCFF933NTR_sorted.bed\"\u003e\u003cpre\u003e \u003cspan class=\"pl-s1\"\u003esort\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ek\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ek2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003en\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eENCFF933NTR\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eENCFF933NTR_sorted\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ebed\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eInstall pyGenomeTracks using \u003cbr\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"conda -install -c bioconda pyGenomeTracks\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003econda\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003einstall\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ebioconda\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003epyGenomeTracks\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eVisualize regions by running- \u003cbr\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"pyGenomeTracks --tracks config.ini --region chr22:37,193,000-37,252,000 -o Genome_track_plot.png\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003epyGenomeTracks\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003etracks\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003econfig\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eini\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eregion\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003echr22\u003c/span\u003e:\u003cspan class=\"pl-c1\"\u003e37\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e193\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e000\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e37\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e252\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e000\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eo\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eGenome_track_plot\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003epng\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n \u003c/details\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/pyGenome%20output.PNG\"\u003e\u003cimg src=\"/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/pyGenome%20output.PNG\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n \n\u003c/p\u003e\u003cul dir=\"auto\"\u003e\n\u003cli\u003eFrom the figure, we can see 3 accessible TSS for 6 transcripts for 2 genes. The TSS of RAC2 corresponds to an ATAC-Seq peak whereas there is no significant coverage on both TSS of SSTR3. Again, it can be said that only the first peak on the left overlaps with a CTCF binding site represents accessible loci. Amongst the 4 peaks in this plotted region, the 2 peaks in the middle do not correspond to CTCF peaks or TSS. As CTCF creates accessible regions, a region containing a peak with no corresponding CTCF peak or TSS could be a putative enhancer. In the pyGenomeTracks plot we see a region like this located in the intron of a gene and another one between genes. More analyses are needed to assess if it is a real enhancer, for example, histone ChIP-seq, 3D structure, transgenic assay, etc.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eAs CTCF creates accessible regions, a region containing a peak with no corresponding CTCF peak or TSS could be a putative enhancer. In the pyGenomeTracks plot we see a region like this located in the intron of a gene and another one between genes.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConclusion\u003c/h2\u003e\u003ca id=\"user-content-conclusion\" class=\"anchor\" aria-label=\"Permalink: Conclusion\" href=\"#conclusion\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eATAC-Seq is a method to investigate the chromatin accessibility and the genome is treated with a transposase (enzyme) called Tn5. It marks open chromatin regions by cutting and inserting adapters for sequencing. Low quality bases, adapter contamination, correct insert size and PCR duplicates (duplication level) were checked. Mapped the reads with  \u003cstrong\u003eBowtie2\u003c/strong\u003e , filtered the reads for properly paired, good quality and reads that do not map to the mitochondrial genome. Open chromatin regions were found with  \u003cstrong\u003eMACS2\u003c/strong\u003e , a tool to find regions of genomic enrichment (peaks). The read coverage around TSS was investigated with the help of  \u003cstrong\u003ecomputeMatrix\u003c/strong\u003e  and  \u003cstrong\u003eplotHeatmap\u003c/strong\u003e. The peaks and other informative tracks, such as CTCF binding regions and hg38 genes were visualised with the help of  \u003cstrong\u003epyGenomeTracks\u003c/strong\u003e. At the end, open chromatin regions that did not overlap with CTCF sites or TSS, which could be potential putative enhancer regions detected by the ATAC-Seq experiment.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eREFERENCES\u003c/h2\u003e\u003ca id=\"user-content-references\" class=\"anchor\" aria-label=\"Permalink: REFERENCES\" href=\"#references\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eLucille Delisle, Maria Doyle, Florian Heyl, 2021 \u003cstrong\u003eATAC-Seq data analysis (Galaxy Training Materials)\u003c/strong\u003e. \u003ca href=\"https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html\" rel=\"nofollow\"\u003ehttps://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html\u003c/a\u003e Online; accessed Thu Aug 19 2021\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e TOOL USED  \u003c/h2\u003e\u003ca id=\"user-content--tool-used--\" class=\"anchor\" aria-label=\"Permalink:  TOOL USED  \" href=\"#-tool-used--\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul class=\"contains-task-list\"\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e Fastqc\u003c/li\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e Cutadapt\u003c/li\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e Bowtie2\u003c/li\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e Samtools\u003c/li\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e Picard\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eCollectInsertSizeMetrics\u003c/li\u003e\n\u003cli\u003eMarkDuplicates\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e Bedtools\u003c/li\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e DeepTool\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eComputeMatrix\u003c/li\u003e\n\u003cli\u003ePlotHeatmap\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e MACS2\u003c/li\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e BedGraphtoBigWig\u003c/li\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e PyGenomeTracks\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e TEAM WORK CONTRIBUTION \u003c/h3\u003e\u003ca id=\"user-content-two_men_holding_hands-team-work-contribution-two_women_holding_hands\" class=\"anchor\" aria-label=\"Permalink: :two_men_holding_hands: TEAM WORK CONTRIBUTION :two_women_holding_hands:\" href=\"#two_men_holding_hands-team-work-contribution-two_women_holding_hands\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003eSlack username\u003c/th\u003e\n\u003cth\u003eSuggesting workflows\u003c/th\u003e\n\u003cth\u003eWorkflow Design\u003c/th\u003e\n\u003cth\u003eGalaxy Based Implementation\u003c/th\u003e\n\u003cth\u003eLinux Based Implementation\u003c/th\u003e\n\u003cth\u003eCreating github repo \u0026amp; Markdown file formatting\u003c/th\u003e\n\u003cth\u003eWriting the galaxy workflow for markdown\u003c/th\u003e\n\u003cth\u003eWriting the Linux workflow for markdown\u003c/th\u003e\n\u003cth\u003eOthers\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Sanjana1404\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Comfortojedapo\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003eTeam Lead\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Jaswanth\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Jeeel193\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003eCreating a new workflow design for markdown file\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Sathya\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Malu\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Adams\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Mercyme\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Wealth88\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Maryamabdul\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Nishat\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003eTeam Lead\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Akinmode\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003eCreating the workflow design for marketing  at the transfer market\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@XR2\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Kehinde16\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Sakshi_r_\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e@Tardigrade\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003ctd\u003eProvided Result Images\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://camo.githubusercontent.com/1a21b31e2fb456e8d7f5c15e8c17c55a9ec495614f08f4df84c552feac206fd6/68747470733a2f2f616c6c667265657468616e6b796f756e6f7465732e636f6d2f77702d636f6e74656e742f75706c6f6164732f323032312f30322f616c6c2d667265652d7468616e6b2d796f752d6769662d31302e676966\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/1a21b31e2fb456e8d7f5c15e8c17c55a9ec495614f08f4df84c552feac206fd6/68747470733a2f2f616c6c667265657468616e6b796f756e6f7465732e636f6d2f77702d636f6e74656e742f75706c6f6164732f323032312f30322f616c6c2d667265652d7468616e6b2d796f752d6769662d31302e676966\" data-animated-image=\"\" data-canonical-src=\"https://allfreethankyounotes.com/wp-content/uploads/2021/02/all-free-thank-you-gif-10.gif\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\u003c/b\u003e\u003c/b\u003e\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":" HackBio2021-Team Epigenomics1 ","anchor":"-hackbio2021-team-epigenomics1-","htmlText":" HackBio2021-Team Epigenomics1 "},{"level":2,"text":":scroll: STAGE-2-PROBE :scroll:","anchor":"scroll-stage-2-probe-scroll","htmlText":" STAGE-2-PROBE "},{"level":2,"text":" THE TASK WE NEED TO DO IN THIS STAGE","anchor":"-the-task-we-need-to-do-in-this-stage","htmlText":" THE TASK WE NEED TO DO IN THIS STAGE"},{"level":2,"text":"WE CHOSE ATAC-Seq Analysis","anchor":"-we-chose-atac-seq-analysis","htmlText":"WE CHOSE ATAC-Seq Analysis"},{"level":3,"text":"What ATAC-Seq is","anchor":"what-atac-seq-is","htmlText":"What ATAC-Seq is"},{"level":4,"text":"What Our Analysis is About?","anchor":"what-our-analysis-is-about","htmlText":"What Our Analysis is About?"},{"level":4,"text":"About Dataset","anchor":"about-dataset","htmlText":"About Dataset"},{"level":2,"text":"WORKFLOW","anchor":"workflow","htmlText":"WORKFLOW"},{"level":3,"text":" GRAPHICAL WORKFLOW DESIGN","anchor":"-graphical-workflow-design","htmlText":" GRAPHICAL WORKFLOW DESIGN"},{"level":3,"text":"STEP 1 :- PREPROCESSING","anchor":"step-1---preprocessing","htmlText":"STEP 1 :- PREPROCESSING"},{"level":4,"text":"A) DATASET","anchor":"a-dataset","htmlText":"A) DATASET"},{"level":4,"text":"B) Obtain Annotation for hg38 genes","anchor":"b-obtain-annotation-for-hg38-genes","htmlText":"B) Obtain Annotation for hg38 genes"},{"level":4,"text":"C) QUALITY CONTROL","anchor":"c-quality-control","htmlText":"C) QUALITY CONTROL"},{"level":4,"text":"TRIMMING READ","anchor":"trimming-read","htmlText":"TRIMMING READ"},{"level":5,"text":"Adapter Trimming","anchor":"adapter-trimming","htmlText":"Adapter Trimming"},{"level":5,"text":"Adapter Trimming with Cutadapt","anchor":"adapter-trimming-with-cutadapt","htmlText":"Adapter Trimming with Cutadapt"},{"level":3,"text":"STEP2 :- MAPPING","anchor":"step2---mapping","htmlText":"STEP2 :- MAPPING"},{"level":3,"text":"Mapping reads to reference genome","anchor":"mapping-reads-to-reference-genome","htmlText":"Mapping reads to reference genome"},{"level":3,"text":"STEP 3 :- Filtering Mapped Reads","anchor":"step-3---filtering-mapped-reads","htmlText":"STEP 3 :- Filtering Mapped Reads"},{"level":4,"text":"A) Filter Uninformative Reads","anchor":"a-filter-uninformative-reads","htmlText":"A) Filter Uninformative Reads"},{"level":4,"text":"B) Filter duplicate reads","anchor":"b-filter-duplicate-reads","htmlText":"B) Filter duplicate reads"},{"level":4,"text":"C) Check Insert Sizes","anchor":"c-check-insert-sizes","htmlText":"C) Check Insert Sizes"},{"level":3,"text":"STEP 4 :- PEAK CALLING","anchor":"step-4---peak-calling","htmlText":"STEP 4 :- PEAK CALLING"},{"level":4,"text":"A) MACS2 callpeak  MACS2 callpeak","anchor":"a-macs2-callpeak--macs2-callpeak","htmlText":"A) MACS2 callpeak  MACS2 callpeak"},{"level":3,"text":"STEP 5 :-Visualisation of Coverage","anchor":"step-5--visualisation-of-coverage","htmlText":"STEP 5 :-Visualisation of Coverage"},{"level":4,"text":"A) Prepare the Datasets","anchor":"a-prepare-the-datasets","htmlText":"A) Prepare the Datasets"},{"level":4,"text":"B)Convert bedgraph from MACS2 to bigwig","anchor":"bconvert-bedgraph-from-macs2-to-bigwig","htmlText":"B)Convert bedgraph from MACS2 to bigwig"},{"level":4,"text":"C) Create heatmap of coverage at TSS with deepTool","anchor":"c-create-heatmap-of-coverage-at-tss-with-deeptool","htmlText":"C) Create heatmap of coverage at TSS with deepTool"},{"level":4,"text":"D)Visualise Regions with pyGenomeTracks","anchor":"dvisualise-regions-with-pygenometracks","htmlText":"D)Visualise Regions with pyGenomeTracks"},{"level":2,"text":"Conclusion","anchor":"conclusion","htmlText":"Conclusion"},{"level":2,"text":"REFERENCES","anchor":"references","htmlText":"REFERENCES"},{"level":2,"text":" TOOL USED  ","anchor":"-tool-used--","htmlText":" TOOL USED  "},{"level":3,"text":":two_men_holding_hands: TEAM WORK CONTRIBUTION :two_women_holding_hands:","anchor":"two_men_holding_hands-team-work-contribution-two_women_holding_hands","htmlText":" TEAM WORK CONTRIBUTION "}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2FDSJaswanth%2FHACKBIO_2021_TEAM_EPIGENOMICS1"}}],"overviewFilesProcessingTime":159.731895}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           177 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="images, (Directory)" class="Link--primary" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/tree/main/images" title="images">
                          images
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="images, (Directory)" class="Link--primary" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/tree/main/images" title="images">
                          images
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="20210821_120019_0000.png, (File)" class="Link--primary" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/20210821_120019_0000.png" title="20210821_120019_0000.png">
                          20210821_120019_0000.png
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="20210821_120019_0000.png, (File)" class="Link--primary" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/20210821_120019_0000.png" title="20210821_120019_0000.png">
                          20210821_120019_0000.png
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="over_represented.png, (File)" class="Link--primary" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/over_represented.png" title="over_represented.png">
                          over_represented.png
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="over_represented.png, (File)" class="Link--primary" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/over_represented.png" title="over_represented.png">
                          over_represented.png
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      <p align="center" dir="auto">
                        HackBio2021-Team Epigenomics1 
                      </p>
                     </h1>
                     <a aria-label="Permalink:  HackBio2021-Team Epigenomics1 " class="anchor" href="#-hackbio2021-team-epigenomics1-" id="user-content--hackbio2021-team-epigenomics1-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="https://camo.githubusercontent.com/21d4e121cd0c0070fd9fad7b6e5e544e5a2497c19867dce10ff4e179cfc0b6ee/68747470733a2f2f76697369746f722d62616467652e6c616f62692e6963752f62616467653f706167655f69643d44534a6177616e74682e4841434b42494f5f323032315f5445414d5f45504947454e4f4d49435331" rel="noopener noreferrer nofollow" target="_blank">
                      <img alt="Visitor" data-canonical-src="https://visitor-badge.laobi.icu/badge?page_id=DSJawanth.HACKBIO_2021_TEAM_EPIGENOMICS1" src="https://camo.githubusercontent.com/21d4e121cd0c0070fd9fad7b6e5e544e5a2497c19867dce10ff4e179cfc0b6ee/68747470733a2f2f76697369746f722d62616467652e6c616f62692e6963752f62616467653f706167655f69643d44534a6177616e74682e4841434b42494f5f323032315f5445414d5f45504947454e4f4d49435331" style="max-width: 100%;"/>
                     </a>
                     <a href="https://camo.githubusercontent.com/355b1324636494fc18ef6c7cddade57521133de268036df6e63c3dec3411c721/68747470733a2f2f6d656469612d657870312e6c6963646e2e636f6d2f646d732f696d6167652f43353631424151484b63565147626365644f412f636f6d70616e792d6261636b67726f756e645f31303030302f302f313539383439313437333538383f653d3231353930323434303026763d6265746126743d727845436a76515f5953633238446e306e39594f74446f46466d76586a6174526971635f5f43326d705530" rel="noopener noreferrer nofollow" target="_blank">
                      <img alt="HackBio" data-canonical-src="https://media-exp1.licdn.com/dms/image/C561BAQHKcVQGbcedOA/company-background_10000/0/1598491473588?e=2159024400&amp;v=beta&amp;t=rxECjvQ_YSc28Dn0n9YOtDoFFmvXjatRiqc__C2mpU0" src="https://camo.githubusercontent.com/355b1324636494fc18ef6c7cddade57521133de268036df6e63c3dec3411c721/68747470733a2f2f6d656469612d657870312e6c6963646e2e636f6d2f646d732f696d6167652f43353631424151484b63565147626365644f412f636f6d70616e792d6261636b67726f756e645f31303030302f302f313539383439313437333538383f653d3231353930323434303026763d6265746126743d727845436a76515f5953633238446e306e39594f74446f46466d76586a6174526971635f5f43326d705530" style="max-width: 100%;"/>
                     </a>
                     <br/>
                     Image source: HackBio (LinkedIn)
                    </p>
                    <p align="center" dir="auto">
                     <a href="https://camo.githubusercontent.com/f57436a5b05808a94ad67f842d95fe230b4f0d74bd54698401fbdcdc9c8eb9ae/68747470733a2f2f692e726564642e69742f736330723779773633623233312e676966" rel="noopener noreferrer nofollow" target="_blank">
                      <img data-animated-image="" data-canonical-src="https://i.redd.it/sc0r7yw63b231.gif" src="https://camo.githubusercontent.com/f57436a5b05808a94ad67f842d95fe230b4f0d74bd54698401fbdcdc9c8eb9ae/68747470733a2f2f692e726564642e69742f736330723779773633623233312e676966" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     This is ***Team EPIGENOMICS1 *** for Hackbio 2021 virtual internship.
                    </p>
                    <ul dir="auto">
                     <li>
                       About: Hackbio Internship is a 5-weeks virtual research internship that is practice oriented and focused on equipping scientists globally with advanced bioinformatics and      computational biology skills.
                     </li>
                     <li>
                       Hackbio official website:
                      <a href="https://thehackbio.com/" rel="nofollow">
                       https://thehackbio.com/
                      </a>
                     </li>
                     <li>
                       Contact:
                      <a href="mailto:contact@hackbio.com">
                       contact@hackbio.com
                      </a>
                     </li>
                     <li>
                       We are a diverse team of awesome 17 members .
                     </li>
                    </ul>
                    <br/>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                       STAGE-2-PROBE 
                     </h2>
                     <a aria-label="Permalink: :scroll: STAGE-2-PROBE :scroll:" class="anchor" href="#scroll-stage-2-probe-scroll" id="user-content-scroll-stage-2-probe-scroll">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      <p align="center" dir="auto">
                       THE TASK WE NEED TO DO IN THIS STAGE
                      </p>
                     </h2>
                     <a aria-label="Permalink:  THE TASK WE NEED TO DO IN THIS STAGE" class="anchor" href="#-the-task-we-need-to-do-in-this-stage" id="user-content--the-task-we-need-to-do-in-this-stage">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                      SEARCHING FOR DESIRED TUTORIALS OF BIOSTACK EPIGENOMICS.
                     <br/>
                      HAVING A WONDERFUL WORKFLOW DESIGN.
                     <br/>
                      REPRODUCING THE ANALYSIS.
                     <br/>
                      CREATING A COMPREHENSIVE MARKDOWN OF THE ANALYSIS.
                     <br/>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      <p align="center" dir="auto">
                       <strong>
                        WE CHOSE ATAC-Seq Analysis
                       </strong>
                      </p>
                     </h2>
                     <a aria-label="Permalink: WE CHOSE ATAC-Seq Analysis" class="anchor" href="#-we-chose-atac-seq-analysis" id="user-content--we-chose-atac-seq-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <strong>
                       What ATAC-Seq is
                      </strong>
                     </h3>
                     <a aria-label="Permalink: What ATAC-Seq is" class="anchor" href="#what-atac-seq-is" id="user-content-what-atac-seq-is">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     ATAC-Seq, acronym of "Assay for Transposase-Accessible Chromatin using Sequencing", is a technique for assessing chromatin accessibility and how it affects gene expression. The chromatin consists of several nucleosome units. A nucleosome is a complex of DNA and histone proteins that keeps DNA compact through packaging. For transcription to occur, DNA must be loosened from the nucleosomes to make it accessible to transcription factors. This accessibility of DNA mediated by chromatin is a key component of epigenetics.
                    </p>
                    <p dir="auto">
                     In this project, we are reproducing the ATAC-Seq analysis
                     <a href="https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html#trimming-reads" rel="nofollow">
                      tutorial
                     </a>
                     from the Galaxy Project. We've reproduced it completely with Linux Command Line tools too.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      What Our Analysis is About?
                     </h4>
                     <a aria-label="Permalink: What Our Analysis is About?" class="anchor" href="#what-our-analysis-is-about" id="user-content-what-our-analysis-is-about">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      Using ATAC-Seq analysis, here we will compare the predicted open chromatin regions to the known binding sites of CTCF (a DNA-binding protein implicated in 3D structure). CTCF is known to bind to thousands of sites in the genome and thus it can be used as a positive control for assessing if the ATAC-Seq experiment is good quality.
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      About Dataset
                     </h4>
                     <a aria-label="Permalink: About Dataset" class="anchor" href="#about-dataset" id="user-content-about-dataset">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      The dataset we will use here is a subset of the data used in the study of Buenrostro et al. 2013. The data from the original dataset is downsized to 200,000 randomly selected reads and about 200,000 reads pairs that will map to chromosome 22. We are only focusing on chromosome 22 so that we can handle the analysis in less time. Binding sites of CTCF identified by ChIP in the same cell line from ENCODE (ENCSR000AKB, dataset ENCFF933NTR) will also be used.
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      <strong>
                       WORKFLOW
                      </strong>
                     </h2>
                     <a aria-label="Permalink: WORKFLOW" class="anchor" href="#workflow" id="user-content-workflow">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <p align="center" dir="auto">
                       GRAPHICAL WORKFLOW DESIGN
                      </p>
                     </h3>
                     <a aria-label="Permalink:  GRAPHICAL WORKFLOW DESIGN" class="anchor" href="#-graphical-workflow-design" id="user-content--graphical-workflow-design">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p align="center" dir="auto">
                     <a href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/20210821_120019_0000.png" rel="noopener noreferrer" target="_blank">
                      <img height="600px" src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/20210821_120019_0000.png" style="max-width: 100%;" width="800px"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      STEP 1 :- PREPROCESSING
                     </h3>
                     <a aria-label="Permalink: STEP 1 :- PREPROCESSING" class="anchor" href="#step-1---preprocessing" id="user-content-step-1---preprocessing">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      A) DATASET
                     </h4>
                     <a aria-label="Permalink: A) DATASET" class="anchor" href="#a-dataset" id="user-content-a-dataset">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <em>
                      <strong>
                      </strong>
                     </em>
                    </p>
                    <details>
                     <summary>
                      <em>
                       <strong>
                        Data Upload for galaxy
                       </strong>
                      </em>
                     </summary>
                     <em>
                      <strong>
                       <br/>
                      </strong>
                     </em>
                     <p dir="auto">
                     </p>
                     <p dir="auto">
                      <em>
                       <strong>
                        Create a new history
                       </strong>
                      </em>
                     </p>
                     <ol dir="auto">
                      <li>
                       <p dir="auto">
                        Add or import DATASETS (2 fastq.gz files and 1 bed.gz file) for analysis via
                        <a href="https://zenodo.org/record/3862793#.YRze2XUvNH4" rel="nofollow">
                         LINK
                        </a>
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        Add tags to R1 and R2 files. To add tags:
                       </p>
                       <ul dir="auto">
                        <li>
                         Click on the dataset
                        </li>
                        <li>
                         Click on the tag icon
                        </li>
                        <li>
                         Add a tag starting with
                         <strong>
                          #: FOR  #SRR891268_R1
                         </strong>
                         FILE to the R1 file and  FOR
                         <strong>
                          #SRR891268_R2
                         </strong>
                         FILE to the R2 file.
                        </li>
                        <li>
                         Check that tag appears below the dataset name
                        </li>
                       </ul>
                      </li>
                     </ol>
                     <p dir="auto">
                      <em>
                       <strong>
                        Check datatype of files and edit if necessary
                       </strong>
                      </em>
                     </p>
                     <ol dir="auto">
                      <li>
                       Click on the pencil icon for the dataset to edit its attributes
                      </li>
                      <li>
                       In the central panel, click on the Datatypes tab on the top
                      </li>
                      <li>
                       Select correct datatype (fastqsanger.gz for the FASTQ files, and encodepeak for the bed.gz file)
                      </li>
                      <li>
                       Click the Change datatype button
                      </li>
                     </ol>
                    </details>
                    <p dir="auto">
                     <em>
                      <strong>
                      </strong>
                     </em>
                    </p>
                    <details>
                     <summary>
                      <em>
                       <strong>
                        Get Data for LINUX
                       </strong>
                      </em>
                     </summary>
                     <em>
                      <strong>
                       <br/>
                      </strong>
                     </em>
                     <p dir="auto">
                     </p>
                     <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="  wget https://zenodo.org/record/3862793/files/ENCFF933NTR.bed.gz" dir="auto">
                      <pre>  <span class="pl-s1">wget</span> <span class="pl-s1">https</span>:<span class="pl-c1">//</span><span class="pl-s1">zenodo</span>.<span class="pl-s1">org</span><span class="pl-c1">/</span><span class="pl-s1">record</span><span class="pl-c1">/</span><span class="pl-c1">3862793</span><span class="pl-c1">/</span><span class="pl-s1">files</span><span class="pl-c1">/</span><span class="pl-v">ENCFF933NTR</span>.<span class="pl-s1">bed</span>.<span class="pl-s1">gz</span></pre>
                     </div>
                     <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="  wget https://zenodo.org/record/3862793/files/SRR891268_chr22_enriched_R1.fastq.gz" dir="auto">
                      <pre>  <span class="pl-s1">wget</span> <span class="pl-s1">https</span>:<span class="pl-c1">//</span><span class="pl-s1">zenodo</span>.<span class="pl-s1">org</span><span class="pl-c1">/</span><span class="pl-s1">record</span><span class="pl-c1">/</span><span class="pl-c1">3862793</span><span class="pl-c1">/</span><span class="pl-s1">files</span><span class="pl-c1">/</span><span class="pl-v">SRR891268_chr22_enriched_R1</span>.<span class="pl-s1">fastq</span>.<span class="pl-s1">gz</span></pre>
                     </div>
                     <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="  wget https://zenodo.org/record/3862793/files/SRR891268_chr22_enriched_R2.fastq.gz" dir="auto">
                      <pre>  <span class="pl-s1">wget</span> <span class="pl-s1">https</span>:<span class="pl-c1">//</span><span class="pl-s1">zenodo</span>.<span class="pl-s1">org</span><span class="pl-c1">/</span><span class="pl-s1">record</span><span class="pl-c1">/</span><span class="pl-c1">3862793</span><span class="pl-c1">/</span><span class="pl-s1">files</span><span class="pl-c1">/</span><span class="pl-v">SRR891268_chr22_enriched_R2</span>.<span class="pl-s1">fastq</span>.<span class="pl-s1">gz</span></pre>
                     </div>
                     <p dir="auto">
                      <em>
                       <strong>
                        You can unzip the sequence files with gunzip
                       </strong>
                      </em>
                     </p>
                     <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="   gunzip SRR891268_chr22_enriched_R1.fastq.gz " dir="auto">
                      <pre>   <span class="pl-s1">gunzip</span> <span class="pl-v">SRR891268_chr22_enriched_R1</span>.<span class="pl-s1">fastq</span>.<span class="pl-s1">gz</span> </pre>
                     </div>
                     <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="  gunzip SRR891268_chr22_enriched_R2.fastq.gz" dir="auto">
                      <pre>  <span class="pl-s1">gunzip</span> <span class="pl-v">SRR891268_chr22_enriched_R2</span>.<span class="pl-s1">fastq</span>.<span class="pl-s1">gz</span></pre>
                     </div>
                    </details>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      B) Obtain Annotation for hg38 genes
                     </h4>
                     <a aria-label="Permalink: B) Obtain Annotation for hg38 genes" class="anchor" href="#b-obtain-annotation-for-hg38-genes" id="user-content-b-obtain-annotation-for-hg38-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <details>
                     <summary>
                      Galaxy Implementation
                     </summary>
                     <ol dir="auto">
                      <li>
                       Select the
                       <strong>
                        USCS Main table browser tool
                       </strong>
                       with the following parameters
                      </li>
                     </ol>
                     <ul dir="auto">
                      <li>
                       <b>
                        clade
                       </b>
                       :Mammal
                      </li>
                      <li>
                       <b>
                        genome
                       </b>
                       :Human
                      </li>
                      <li>
                       <b>
                        assembly
                       </b>
                       :Dec. 2013(GRCh38/hg38)
                      </li>
                      <li>
                       <b>
                        group
                       </b>
                       :Genes and Gene Prediction
                      </li>
                      <li>
                       <b>
                        track
                       </b>
                       :All GENCODE V37
                      </li>
                      <li>
                       <b>
                        table
                       </b>
                       :Basic
                      </li>
                      <li>
                       <b>
                        region
                       </b>
                       :position chr22
                      </li>
                      <li>
                       <b>
                        output format
                       </b>
                       :all fields from selected table
                      </li>
                      <li>
                       <b>
                        Send output to
                       </b>
                       :Galaxy
                      </li>
                     </ul>
                     <ol dir="auto" start="2">
                      <li>
                       <p dir="auto">
                        Click get output
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        Click Send Query to Galaxy
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        Select the
                        <strong>
                         Cut columns from a table tool
                        </strong>
                        with the following parameters
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        <b>
                         Cut columns
                        </b>
                        : c3,c5,c6,c13,c12,c4
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        <b>
                         Delimited by
                        </b>
                        : Tab
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        <b>
                         param-file From
                        </b>
                        : UCSC Main on Human: wgEncodeGencodeBasicV37 (chr22:1-50,818,468)
                        <em>
                         Rename the dataset as chr22 genes
                        </em>
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        Click on the pencil icon for the dataset to edit its attributes
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        In the central panel, change the name field
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        Click the Save button
                       </p>
                      </li>
                     </ol>
                     <p dir="auto">
                      <em>
                       Change the datatype to a BED format
                      </em>
                     </p>
                     <ol dir="auto">
                      <li>
                       Click on the pencil icon for the dataset to edit its attributes
                      </li>
                      <li>
                       In the central panel, click on the Datatypes tab on the top
                      </li>
                      <li>
                       Select
                       <strong>
                        bed
                       </strong>
                      </li>
                      <li>
                       Click the Change datatype button
                      </li>
                     </ol>
                     <p dir="auto">
                      Click on the eye icon to check changes effected. There should be matching column names in each column of the dataset.
                     </p>
                    </details>
                    <details>
                     <summary>
                      LINUX Implementation
                     </summary>
                     Go to [
                     <a href="http://genome.ucsc.edu/cgi-bin/hgTables](http://genome.ucsc.edu/cgi-bin/hgTables)" rel="nofollow">
                      http://genome.ucsc.edu/cgi-bin/hgTables](http://genome.ucsc.edu/cgi-bin/hgTables)
                     </a>
                     and set the parameters as-
                     <ul dir="auto">
                      <li>
                       <b>
                        clade
                       </b>
                       :Mammal
                      </li>
                      <li>
                       <b>
                        genome
                       </b>
                       :Human
                      </li>
                      <li>
                       <b>
                        assembly
                       </b>
                       :Dec. 2013(GRCh38/hg38)
                      </li>
                      <li>
                       <b>
                        group
                       </b>
                       :Genes and Gene Prediction
                      </li>
                      <li>
                       <b>
                        track
                       </b>
                       :All GENCODE V37
                      </li>
                      <li>
                       <b>
                        table
                       </b>
                       :Basic
                      </li>
                      <li>
                       <b>
                        region
                       </b>
                       :position chr22
                      </li>
                      <li>
                       <b>
                        output format:all fields from selected table
                       </b>
                      </li>
                      <b>
                       <li>
                        <b>
                         output filename:chr22
                        </b>
                       </li>
                       <b>
                        <li>
                         <b>
                          file type returned:gzipped compressed
                         </b>
                        </li>
                        <b>
                        </b>
                       </b>
                      </b>
                     </ul>
                     <b>
                      <b>
                       <b>
                        <p dir="auto">
                         And then select
                         <strong>
                          Get output
                         </strong>
                        </p>
                        <p dir="auto">
                         Thus, chr22.gz file will be downloaded.
                        </p>
                        <ul dir="auto">
                         <li>
                          <b>
                           Converting chr22 file into a bed file:
                          </b>
                         </li>
                        </ul>
                        <ol dir="auto">
                         <li>
                          <p dir="auto">
                           Unzip the downloaded chr22.gz using command
                          </p>
                          <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="unzip chr22.gz " dir="auto">
                           <pre><span class="pl-s1">unzip</span> <span class="pl-s1">chr22</span>.<span class="pl-s1">gz</span> </pre>
                          </div>
                         </li>
                        </ol>
                        <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="awk -F &quot;\t&quot; 'OFS=&quot;\t&quot; {print $3, $5, $6, $13, $12, $4 &gt; (&quot;chr22.bed&quot;)}' chr22 ">
                         <pre class="notranslate"><code>awk -F "\t" 'OFS="\t" {print $3, $5, $6, $13, $12, $4 &gt; ("chr22.bed")}' chr22 
</code></pre>
                        </div>
                        <p dir="auto">
                         (to get only expected columns into a newly created chr22.bed file)
                        </p>
                       </b>
                      </b>
                     </b>
                    </details>
                    <b>
                     <b>
                      <div class="markdown-heading" dir="auto">
                       <h4 class="heading-element" dir="auto" tabindex="-1">
                        C) QUALITY CONTROL
                       </h4>
                       <a aria-label="Permalink: C) QUALITY CONTROL" class="anchor" href="#c-quality-control" id="user-content-c-quality-control">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <details>
                       <summary>
                        Galaxy Implementation
                       </summary>
                       <br/>
                       - Select the
                       <b>
                        FastQC tool
                       </b>
                       with the following parameters
                       <br/>
                       - 1. "Short read data from your current history": Choose here either only the SRR891268_R1 file with param-file or use param-files; use Multiple datasets to choose both SRR891268_R1 and SRR891268\_R2.
                       <br/>
                       - 2. Inspect the web page output of FastQC tool for the SRR891268\_R1 sample. Check what adapters are found at the end of the reads.
                      </details>
                      <details>
                       <summary>
                        Linux Implementation
                       </summary>
                       <br/>
                       <ul dir="auto">
                        <li>
                         Download the FastQC module
Note: FASTQC requires java and javac installed for implementation and you need to run the fastqc file from the folder (using the relative/absolute links to the sequence reads)
                         <br/>
                        </li>
                       </ul>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="  sudo apt install default-jre" dir="auto">
                        <pre>  <span class="pl-s1">sudo</span> <span class="pl-s1">apt</span> <span class="pl-s1">install</span> <span class="pl-s1">default</span><span class="pl-c1">-</span><span class="pl-s1">jre</span></pre>
                       </div>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="  sudo apt install default-jdk" dir="auto">
                        <pre>  <span class="pl-s1">sudo</span> <span class="pl-s1">apt</span> <span class="pl-s1">install</span> <span class="pl-s1">default</span><span class="pl-c1">-</span><span class="pl-s1">jdk</span></pre>
                       </div>
                       <p dir="auto">
                        Make the fastqc an executable file
                        <br/>
                       </p>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="  chmod 755 fastqc" dir="auto">
                        <pre>  <span class="pl-s1">chmod</span> <span class="pl-c1">755</span> <span class="pl-s1">fastqc</span></pre>
                       </div>
                       <ul dir="auto">
                        <li>
                         Run the fastqc on all sequenced reads from its folder
                         <br/>
                        </li>
                       </ul>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="  fastqc SRR891268_chr22_enriched_R1.fastq SRR891268_chr22_enriched_R2.fastq " dir="auto">
                        <pre>  <span class="pl-s1">fastqc</span> <span class="pl-v">SRR891268_chr22_enriched_R1</span>.<span class="pl-s1">fastq</span> <span class="pl-v">SRR891268_chr22_enriched_R2</span>.<span class="pl-s1">fastq</span> </pre>
                       </div>
                      </details>
                      The report for each file is generated as an html file and a zip file containing more files that can be customised for reports. Look into the html files.
                      <p align="center" dir="auto">
                       <a href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/FastQC.PNG" rel="noopener noreferrer" target="_blank">
                        <img src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/FastQC.PNG" style="max-width: 100%;"/>
                       </a>
                      </p>
                      <p dir="auto">
                       The overrepresented sequence segment shows that we have some overrepresented sequence, that is adapter sequence. We have to trim that adapter sequence to move forward with our analysis.
                      </p>
                      <p align="center" dir="auto">
                       <a href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/over_represented.png" rel="noopener noreferrer" target="_blank">
                        <img src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/over_represented.png" style="max-width: 100%;" width="1000px"/>
                       </a>
                      </p>
                      <div class="markdown-heading" dir="auto">
                       <h4 class="heading-element" dir="auto" tabindex="-1">
                        TRIMMING READ
                       </h4>
                       <a aria-label="Permalink: TRIMMING READ" class="anchor" href="#trimming-read" id="user-content-trimming-read">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <details>
                       <summary>
                        Galaxy Implementation
                       </summary>
                       <br/>
                       <ul dir="auto">
                        <li>
                         Select the
                         <strong>
                          Cutadapt tool
                         </strong>
                         with the following parameters
                        </li>
                       </ul>
                       <ol dir="auto">
                        <li>
                         <p dir="auto">
                          "Single-end or Paired-end reads?": Paired-end
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          param-file "FASTQ/A file #1": select SRR891268_R1
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          param-file "FASTQ/A file #2": select SRR891268_R2
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          In "Read 1 Options":
                         </p>
                        </li>
                       </ol>
                       <p dir="auto">
                        In "3' (End) Adapters":
                       </p>
                       <p dir="auto">
                        param-repeat "Insert 3' (End) Adapters"
                       </p>
                       <p dir="auto">
                        "Source": Enter custom sequence
                       </p>
                       <p dir="auto">
                        "Enter custom 3' adapter name (Optional if Multiple output is 'No')": Nextera R1
                       </p>
                       <p dir="auto">
                        "Enter custom 3' adapter sequence": CTGTCTCTTATACACATCTCCGAGCCCACGAGAC
                       </p>
                       <ol dir="auto">
                        <li>
                         In "Read 2 Options":
                        </li>
                       </ol>
                       <p dir="auto">
                        In "3' (End) Adapters":
                       </p>
                       <p dir="auto">
                        param-repeat "Insert 3' (End) Adapters"
                       </p>
                       <p dir="auto">
                        "Source": Enter custom sequence
                       </p>
                       <p dir="auto">
                        "Enter custom 3' adapter name (Optional)": Nextera R2
                       </p>
                       <p dir="auto">
                        "Enter custom 3' adapter sequence": CTGTCTCTTATACACATCTGACGCTGCCGACGA
                       </p>
                       <ol dir="auto">
                        <li>
                         In "Filter Options":
                        </li>
                       </ol>
                       <p dir="auto">
                        "Minimum length": 20
                       </p>
                       <ol dir="auto">
                        <li>
                         In "Read Modification Options":
                        </li>
                       </ol>
                       <p dir="auto">
                        "Quality cutoff": 20
                       </p>
                       <ol dir="auto">
                        <li>
                         In "Output Options":
                        </li>
                       </ol>
                       <p dir="auto">
                        "Report": Yes
                       </p>
                       <ol dir="auto">
                        <li>
                         Click on the galaxy-eye (eye) icon of the report and read the first lines.
                        </li>
                       </ol>
                       <ul dir="auto">
                        <li>
                         Check Adapter Removal
                        </li>
                       </ul>
                       <p dir="auto">
                        Select the
                        <strong>
                         Fast QC tool
                        </strong>
                        with the following parameters
                       </p>
                       <ol dir="auto">
                        <li>
                         "Short read data from your current history": select the output of Cutadapt param files; use; Multiple datasets to choose both Read 1 Output and Read 2 Output.
                        </li>
                        <li>
                         Click on the galaxy-eye (eye) icon of the report and read the first lines.
                        </li>
                       </ol>
                      </details>
                      <details>
                       <summary>
                        Linux Implementation
                       </summary>
                       <br/>
                       <ul dir="auto">
                        <li>
                         <div class="markdown-heading" dir="auto">
                          <h5 class="heading-element" dir="auto" tabindex="-1">
                           Adapter Trimming
                          </h5>
                          <a aria-label="Permalink: Adapter Trimming" class="anchor" href="#adapter-trimming" id="user-content-adapter-trimming">
                           <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                            <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                            </path>
                           </svg>
                          </a>
                         </div>
                        </li>
                       </ul>
                       <p dir="auto">
                        The fastqc report indicates the presence of an overrepresented sequence and fastqc identifies it as "Nextera Transposase Sequence ''. This sequence is similar to but longer than the one given in the tutorial.
                       </p>
                       <p dir="auto">
                        <code>
                         SRR891268\_chr22\_enriched\_R1 = CTGTCTCTTATACACATCTCCGAGCCCACGAGACTAAGGCGAATCTCGTA (fastqc)
                        </code>
                        <br/>
                       </p>
                       <p dir="auto">
                        <code>
                         SRR891268\_chr22\_enriched\_R1 = CTGTCTCTTATACACATCTCCGAGCCCACGAGAC (Galaxy tutorial)
                        </code>
                        <br/>
                       </p>
                       <p dir="auto">
                        <code>
                         SRR891268\_chr22\_enriched\_R2 = CTGTCTCTTATACACATCTGACGCTGCCGACGAGTGTAGATCTCGGTGGT (fastqc)
                        </code>
                        <br/>
                       </p>
                       <p dir="auto">
                        <code>
                         SRR891268\_chr22\_enriched\_R2 = CTGTCTCTTATACACATCTGACGCTGCCGACGA (Galaxy tutorial)
                        </code>
                        <br/>
                       </p>
                       <ul dir="auto">
                        <li>
                         <div class="markdown-heading" dir="auto">
                          <h5 class="heading-element" dir="auto" tabindex="-1">
                           Adapter Trimming with Cutadapt
                          </h5>
                          <a aria-label="Permalink: Adapter Trimming with Cutadapt" class="anchor" href="#adapter-trimming-with-cutadapt" id="user-content-adapter-trimming-with-cutadapt">
                           <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                            <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                            </path>
                           </svg>
                          </a>
                         </div>
                        </li>
                       </ul>
                       <p dir="auto">
                        Install cutadapt running-
                       </p>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="$ sudo apt install cutadapt" dir="auto">
                        <pre>$ <span class="pl-s1">sudo</span> <span class="pl-s1">apt</span> <span class="pl-s1">install</span> <span class="pl-s1">cutadapt</span></pre>
                       </div>
                       <p dir="auto">
                        For paired end trimming-
                       </p>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cutadapt -a CTGTCTCTTATACACATCTCCGAGCCCACGAGAC -A CTGTCTCTTATACACATCTGACGCTGCCGACGA --minimum-length 20 -q 20 -o trimmed_1.fastq -p trimmed_2.fastq SRR891268_chr22_enriched_R1.fastq SRR891268_chr22_enriched_R2.fastq" dir="auto">
                        <pre><span class="pl-s1">cutadapt</span> <span class="pl-c1">-</span><span class="pl-s1">a</span> <span class="pl-v">CTGTCTCTTATACACATCTCCGAGCCCACGAGAC</span> <span class="pl-c1">-</span><span class="pl-v">A</span> <span class="pl-v">CTGTCTCTTATACACATCTGACGCTGCCGACGA</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">minimum</span><span class="pl-c1">-</span><span class="pl-s1">length</span> <span class="pl-c1">20</span> <span class="pl-c1">-</span><span class="pl-s1">q</span> <span class="pl-c1">20</span> <span class="pl-c1">-</span><span class="pl-s1">o</span> <span class="pl-s1">trimmed_1</span>.<span class="pl-s1">fastq</span> <span class="pl-c1">-</span><span class="pl-s1">p</span> <span class="pl-s1">trimmed_2</span>.<span class="pl-s1">fastq</span> <span class="pl-v">SRR891268_chr22_enriched_R1</span>.<span class="pl-s1">fastq</span> <span class="pl-v">SRR891268_chr22_enriched_R2</span>.<span class="pl-s1">fastq</span></pre>
                       </div>
                      </details>
                      <p align="center" dir="auto">
                       <a href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/FastQC%20after%20cutadapt.PNG" rel="noopener noreferrer" target="_blank">
                        <img src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/FastQC%20after%20cutadapt.PNG" style="max-width: 100%;"/>
                       </a>
                      </p>
                      <div class="markdown-heading" dir="auto">
                       <h3 class="heading-element" dir="auto" tabindex="-1">
                        STEP2 :- MAPPING
                       </h3>
                       <a aria-label="Permalink: STEP2 :- MAPPING" class="anchor" href="#step2---mapping" id="user-content-step2---mapping">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <details>
                       <summary>
                        Galaxy Implementation
                       </summary>
                       <br/>
                       <div class="markdown-heading" dir="auto">
                        <h3 class="heading-element" dir="auto" tabindex="-1">
                         <strong>
                          Mapping reads to reference genome
                         </strong>
                        </h3>
                        <a aria-label="Permalink: Mapping reads to reference genome" class="anchor" href="#mapping-reads-to-reference-genome" id="user-content-mapping-reads-to-reference-genome">
                         <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                          <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                          </path>
                         </svg>
                        </a>
                       </div>
                       <ul dir="auto">
                        <li>
                         Select the
                         <strong>
                          Bowtie2
                         </strong>
                         <strong>
                          tool
                         </strong>
                         with the following parameters:
                        </li>
                       </ul>
                       <ol dir="auto">
                        <li>
                         <p dir="auto">
                          "Is this single or paired library": Paired-end
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          param-file "FASTQ/A file #1": select the output of Cutadapt tool "Read 1 Output"
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          param-file "FASTQ/A file #2": select the output of Cutadapt tool "Read 2 Output"
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          "Do you want to set paired-end options?": Yes
                         </p>
                         <ul dir="auto">
                          <li>
                           "Set the maximum fragment length for valid paired-end alignments": 1000
"Allow mate dovetailing": Yes
                          </li>
                         </ul>
                        </li>
                        <li>
                         <p dir="auto">
                          "Will you select a reference genome from your history or use a built-in index?": Use a built-in genome index
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          "Select reference genome": Human (Homo sapiens): hg38 Canonical
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          "Set read groups information?": Do not set
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          "Select analysis mode": 1: Default setting only
                         </p>
                         <ul dir="auto">
                          <li>
                           "Do you want to use presets?": Very sensitive end-to-end (--very-sensitive)
                          </li>
                         </ul>
                        </li>
                        <li>
                         <p dir="auto">
                          "Do you want to tweak SAM/BAM Options?": No
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          "Save the bowtie2 mapping statistics to the history": Yes
3.Click on the galaxy-eye (eye) icon of the mapping stats.
                         </p>
                        </li>
                       </ol>
                      </details>
                      <details>
                       <summary>
                        Linux Implementation
                       </summary>
                       <br/>
                       <ul dir="auto">
                        <li>
                         <p dir="auto">
                          Mapping and Alignment
                         </p>
                         <p dir="auto">
                          Pulling the sequence for chromosome 22 for indexing and mapping
                          <br/>
                         </p>
                        </li>
                       </ul>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="$ wget --timestamping ;ftp://hgdownload.cse.ucsc.edu/goldenPath/hg38/chromosomes/chr22.fa.gz; -O chr22.fa.gz" dir="auto">
                        <pre>$ <span class="pl-s1">wget</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">timestamping</span> ;<span class="pl-s1">ftp</span>:<span class="pl-c1">//</span><span class="pl-s1">hgdownload</span>.<span class="pl-s1">cse</span>.<span class="pl-s1">ucsc</span>.<span class="pl-s1">edu</span><span class="pl-c1">/</span><span class="pl-s1">goldenPath</span><span class="pl-c1">/</span><span class="pl-s1">hg38</span><span class="pl-c1">/</span><span class="pl-s1">chromosomes</span><span class="pl-c1">/</span><span class="pl-s1">chr22</span>.<span class="pl-s1">fa</span>.<span class="pl-s1">gz</span>; <span class="pl-c1">-</span><span class="pl-v">O</span> <span class="pl-s1">chr22</span>.<span class="pl-s1">fa</span>.<span class="pl-s1">gz</span></pre>
                       </div>
                       <ul dir="auto">
                        <li>
                         For mapping to chr22-
                        </li>
                       </ul>
                       <ol dir="auto">
                        <li>
                         Install bowtie2
                         <br/>
                        </li>
                        <li>
                         Create index for Chromosome 22:
                         <br/>
                        </li>
                       </ol>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" bowtie2-build chr22.fa.gz indexed_chr22" dir="auto">
                        <pre> <span class="pl-s1">bowtie2</span><span class="pl-c1">-</span><span class="pl-s1">build</span> <span class="pl-s1">chr22</span>.<span class="pl-s1">fa</span>.<span class="pl-s1">gz</span> <span class="pl-s1">indexed_chr22</span></pre>
                       </div>
                       <ol dir="auto" start="3">
                        <li>
                         Start mapping for the parameters specified by Galaxy:
                         <br/>
                        </li>
                       </ol>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bowtie2 --very-sensitive --maxins 1000 --dovetail -x indexed_chr22 -1 trimmed_1.fastq -2 trimmed_2.fastq -S Aligned_output.sam" dir="auto">
                        <pre><span class="pl-s1">bowtie2</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">very</span><span class="pl-c1">-</span><span class="pl-s1">sensitive</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">maxins</span> <span class="pl-c1">1000</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">dovetail</span> <span class="pl-c1">-</span><span class="pl-s1">x</span> <span class="pl-s1">indexed_chr22</span> <span class="pl-c1">-</span><span class="pl-c1">1</span> <span class="pl-s1">trimmed_1</span>.<span class="pl-s1">fastq</span> <span class="pl-c1">-</span><span class="pl-c1">2</span> <span class="pl-s1">trimmed_2</span>.<span class="pl-s1">fastq</span> <span class="pl-c1">-</span><span class="pl-v">S</span> <span class="pl-v">Aligned_output</span>.<span class="pl-s1">sam</span></pre>
                       </div>
                       <p align="center" dir="auto">
                        <a href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/Bowtie2%20output.PNG" rel="noopener noreferrer" target="_blank">
                         <img src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/Bowtie2%20output.PNG" style="max-width: 100%;"/>
                        </a>
                       </p>
                      </details>
                      <div class="markdown-heading" dir="auto">
                       <h3 class="heading-element" dir="auto" tabindex="-1">
                        STEP 3 :- Filtering Mapped Reads
                       </h3>
                       <a aria-label="Permalink: STEP 3 :- Filtering Mapped Reads" class="anchor" href="#step-3---filtering-mapped-reads" id="user-content-step-3---filtering-mapped-reads">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <div class="markdown-heading" dir="auto">
                       <h4 class="heading-element" dir="auto" tabindex="-1">
                        A) Filter Uninformative Reads
                       </h4>
                       <a aria-label="Permalink: A) Filter Uninformative Reads" class="anchor" href="#a-filter-uninformative-reads" id="user-content-a-filter-uninformative-reads">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <details>
                       <summary>
                        Galaxy Implementation
                       </summary>
                       <br/>
                       <p dir="auto">
                        <strong>
                         <em>
                          Filtering of uninformative mapped reads
                         </em>
                        </strong>
                       </p>
                       <p dir="auto">
                        Select the
                        <strong>
                         Filter BAM datasets on a variety of attributes
                        </strong>
                        <strong>
                         tool
                        </strong>
                        with the following parameters
                       </p>
                       <p dir="auto">
                        param-file "BAM dataset(s) to filter": Select the output of Bowtie2 tool "alignments"
                       </p>
                       <p dir="auto">
                        In "Condition":
                       </p>
                       <ol dir="auto">
                        <li>
                         param-repeat "Insert Condition"
                        </li>
                        <li>
                         In "Filter":
                        </li>
                        <li>
                         param-repeat "Insert Filter"
                        </li>
                       </ol>
                       <p dir="auto">
                        "Select BAM property to filter on": mapQuality
                       </p>
                       <p dir="auto">
                        "Filter on read mapping quality (phred scale)": &amp;gt;=30
                       </p>
                       <ol dir="auto">
                        <li>
                         param-repeat "Insert Filter"
                        </li>
                       </ol>
                       <p dir="auto">
                        "Select BAM property to filter on": isProperPair
                       </p>
                       <p dir="auto">
                        "Select properly paired reads": Yes
                       </p>
                       <ol dir="auto">
                        <li>
                         param-repeat "Insert Filter"
                        </li>
                       </ol>
                       <p dir="auto">
                        "Select BAM property to filter on": reference
                       </p>
                       <p dir="auto">
                        "Filter on the reference name for the read": !chrM
                       </p>
                       <p dir="auto">
                        "Would you like to set rules?": No
                       </p>
                       <p dir="auto">
                        Click on the input and the output BAM files of the filtering step. Check the size of the files.
                       </p>
                      </details>
                      <details>
                       <summary>
                        Linux Implementation
                       </summary>
                       <br/>
                       <p dir="auto">
                        <strong>
                         <em>
                          Filtering of uninformative mapped reads
                         </em>
                        </strong>
                       </p>
                       <ol dir="auto">
                        <li>
                         Install samtools.
                        </li>
                       </ol>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" samtools view -q 30 -f 0x2 -b -h Aligned_output.sam &gt; Filtered_output.bam" dir="auto">
                        <pre> <span class="pl-s1">samtools</span> <span class="pl-s1">view</span> <span class="pl-c1">-</span><span class="pl-s1">q</span> <span class="pl-c1">30</span> <span class="pl-c1">-</span><span class="pl-s1">f</span> <span class="pl-c1">0x2</span> <span class="pl-c1">-</span><span class="pl-s1">b</span> <span class="pl-c1">-</span><span class="pl-s1">h</span> <span class="pl-v">Aligned_output</span>.<span class="pl-s1">sam</span> <span class="pl-c1">&gt;</span> <span class="pl-v">Filtered_output</span>.<span class="pl-s1">bam</span></pre>
                       </div>
                       <p dir="auto">
                        we will filter out uninformative reads (Mapping quality &amp;gt;= 30 &amp; Properly Paired)
                       </p>
                      </details>
                      <div class="markdown-heading" dir="auto">
                       <h4 class="heading-element" dir="auto" tabindex="-1">
                        B) Filter duplicate reads
                       </h4>
                       <a aria-label="Permalink: B) Filter duplicate reads" class="anchor" href="#b-filter-duplicate-reads" id="user-content-b-filter-duplicate-reads">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <details>
                       <summary>
                        Galaxy Implementation
                       </summary>
                       <br/>
                       <p dir="auto">
                        <em>
                         <strong>
                          <em>
                           Remove duplicates
                          </em>
                         </strong>
                        </em>
                       </p>
                       <p dir="auto">
                        Select the
                        <strong>
                         MarkDuplicates
                        </strong>
                        <strong>
                         tool
                        </strong>
                        with the following parameters
                       </p>
                       <ol dir="auto">
                        <li>
                         param-file "Select SAM/BAM dataset or dataset collection": Select the output of Filter tool "BAM"
                        </li>
                        <li>
                         "If true do not write duplicates to the output file instead of writing them with appropriate flags set": Yes
                        </li>
                       </ol>
                       <p dir="auto">
                        Click on the eye icon of the MarkDuplicate metrics.
                       </p>
                      </details>
                      <details>
                       <summary>
                        Linux Implementation
                       </summary>
                       <br/>
                       <p dir="auto">
                        <strong>
                         <em>
                          Mark Duplicate Reads
                         </em>
                        </strong>
                       </p>
                       <ul dir="auto">
                        <li>
                         <p dir="auto">
                          Download picard.jar in your working folder from
                          <a href="https://github.com/broadinstitute/picard/releases/download/2.26.0/picard.jar">
                           here
                          </a>
                          <br/>
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          From that directory, run
                          <br/>
                         </p>
                         <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="java -jar picard.jar -h" dir="auto">
                          <pre><span class="pl-s1">java</span> <span class="pl-c1">-</span><span class="pl-s1">jar</span> <span class="pl-s1">picard</span>.<span class="pl-s1">jar</span> <span class="pl-c1">-</span><span class="pl-s1">h</span></pre>
                         </div>
                         <p dir="auto">
                          to check whether it works (you can skip this step)
                          <br/>
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          For sorting the output file from last step use-
                          <br/>
                         </p>
                         <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="samtools sort -T temp -O bam -o filtered_output_sorted.bam Filtered_output.bam" dir="auto">
                          <pre><span class="pl-s1">samtools</span> <span class="pl-s1">sort</span> <span class="pl-c1">-</span><span class="pl-v">T</span> <span class="pl-s1">temp</span> <span class="pl-c1">-</span><span class="pl-v">O</span> <span class="pl-s1">bam</span> <span class="pl-c1">-</span><span class="pl-s1">o</span> <span class="pl-s1">filtered_output_sorted</span>.<span class="pl-s1">bam</span> <span class="pl-v">Filtered_output</span>.<span class="pl-s1">bam</span></pre>
                         </div>
                        </li>
                        <li>
                         <p dir="auto">
                          Finally, run
                          <br/>
                         </p>
                         <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="java -jar picard.jar MarkDuplicates I=filtered_output_sorted.bam O=marked_dup.bam M=marked_dup.metrics.txt " dir="auto">
                          <pre><span class="pl-s1">java</span> <span class="pl-c1">-</span><span class="pl-s1">jar</span> <span class="pl-s1">picard</span>.<span class="pl-s1">jar</span> <span class="pl-v">MarkDuplicates</span> <span class="pl-v">I</span><span class="pl-c1">=</span><span class="pl-s1">filtered_output_sorted</span>.<span class="pl-s1">bam</span> <span class="pl-v">O</span><span class="pl-c1">=</span><span class="pl-s1">marked_dup</span>.<span class="pl-s1">bam</span> <span class="pl-v">M</span><span class="pl-c1">=</span><span class="pl-s1">marked_dup</span>.<span class="pl-s1">metrics</span>.<span class="pl-s1">txt</span> </pre>
                         </div>
                         <p dir="auto">
                          <br/>
                          for marking duplicates
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          If you can have a look into the metrics in the metrics.txt file
                         </p>
                        </li>
                       </ul>
                      </details>
                      <div class="markdown-heading" dir="auto">
                       <h4 class="heading-element" dir="auto" tabindex="-1">
                        C) Check Insert Sizes
                       </h4>
                       <a aria-label="Permalink: C) Check Insert Sizes" class="anchor" href="#c-check-insert-sizes" id="user-content-c-check-insert-sizes">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <ul dir="auto">
                       <li>
                        Check Insert Size tells us the size of the DNA fragment the read pairs came from this step we have to make a plot of the frequencies of the reads in the bam file to observe the peaks around where there are likely Tn5 transposase activities into nucleosome-free regions.
                        <br/>
                       </li>
                      </ul>
                      <details>
                       <summary>
                        Galaxy Implementation
                       </summary>
                       <br/>
                       <p dir="auto">
                        <strong>
                         <em>
                          Plot the distribution of fragment sizes
                         </em>
                        </strong>
                       </p>
                       <p dir="auto">
                        Select
                        <strong>
                         Paired-end histogram tool
                        </strong>
                        with the following parameters
                       </p>
                       <ol dir="auto">
                        <li>
                         param-file "BAM file": Select the output of MarkDuplicates tool "BAM output"
                        </li>
                        <li>
                         "Lower bp limit (optional)": 0
                        </li>
                        <li>
                         "Upper bp limit (optional)": 1000
                        </li>
                       </ol>
                       <p dir="auto">
                        Click on the galaxy-eye (eye) icon of the lower one of the 2 outputs (the png file).
                       </p>
                      </details>
                      <details>
                       <summary>
                        Linux Implementation
                       </summary>
                       <br/>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="   sudo apt install r-base" dir="auto">
                        <pre>   <span class="pl-s1">sudo</span> <span class="pl-s1">apt</span> <span class="pl-s1">install</span> <span class="pl-s1">r</span><span class="pl-c1">-</span><span class="pl-s1">base</span></pre>
                       </div>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="   java -jar picard.jar CollectInsertSizeMetrics I=marked_dup.bam O=chart.txt H=insertSizePlot.pdf M=0.5" dir="auto">
                        <pre>   <span class="pl-s1">java</span> <span class="pl-c1">-</span><span class="pl-s1">jar</span> <span class="pl-s1">picard</span>.<span class="pl-s1">jar</span> <span class="pl-v">CollectInsertSizeMetrics</span> <span class="pl-v">I</span><span class="pl-c1">=</span><span class="pl-s1">marked_dup</span>.<span class="pl-s1">bam</span> <span class="pl-v">O</span><span class="pl-c1">=</span><span class="pl-s1">chart</span>.<span class="pl-s1">txt</span> <span class="pl-v">H</span><span class="pl-c1">=</span><span class="pl-s1">insertSizePlot</span>.<span class="pl-s1">pdf</span> <span class="pl-v">M</span><span class="pl-c1">=</span><span class="pl-c1">0.5</span></pre>
                       </div>
                      </details>
                      <p align="center" dir="auto">
                       <a href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/Insert sizes.PNG" rel="noopener noreferrer" target="_blank">
                        <img src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/Insert sizes.PNG" style="max-width: 100%;"/>
                       </a>
                      </p>
                      <p dir="auto">
                       Two peaks can be observed around the 200bp and 400bp from the plot
                      </p>
                      <div class="markdown-heading" dir="auto">
                       <h3 class="heading-element" dir="auto" tabindex="-1">
                        STEP 4 :- PEAK CALLING
                       </h3>
                       <a aria-label="Permalink: STEP 4 :- PEAK CALLING" class="anchor" href="#step-4---peak-calling" id="user-content-step-4---peak-calling">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <ul dir="auto">
                       <li>
                        Peak calling helps us to identify regions where reads have piled up (peaks) greater than the background read coverage. Using MACS2 we will extend the start sites of the reads by 200bp (100bp in each direction) to assess coverage. Before using MACS2, we need to convert the output BAM file from MarkDuplicate stage to BED format because when we set the extension size in MACS2, it will only consider one read of the pair while here, we would like to use the information from both.
                       </li>
                      </ul>
                      <details>
                       <summary>
                        Galaxy Implementation
                       </summary>
                       <br/>
                       <p dir="auto">
                        <em>
                         <strong>
                          <em>
                           Convert BAM to BED
                          </em>
                         </strong>
                        </em>
                       </p>
                       <p dir="auto">
                        Convert BAM file (output of MarkDuplicates) into BED format by
                        <strong>
                         bedtools BAM to BED converter
                        </strong>
                        .
                       </p>
                       <div class="markdown-heading" dir="auto">
                        <h4 class="heading-element" dir="auto" tabindex="-1">
                         A) MACS2 callpeak
                         <strong>
                          MACS2 callpeak
                         </strong>
                        </h4>
                        <a aria-label="Permalink: A) MACS2 callpeak  MACS2 callpeak" class="anchor" href="#a-macs2-callpeak--macs2-callpeak" id="user-content-a-macs2-callpeak--macs2-callpeak">
                         <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                          <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                          </path>
                         </svg>
                        </a>
                       </div>
                       <p dir="auto">
                        <strong>
                         MACS2 callpeak
                        </strong>
                        with the following parameters:
                       </p>
                       <ul dir="auto">
                        <li>
                         Are you pooling Treatment Files?: No
                         <ul dir="auto">
                          <li>
                           Select the output of
                           <strong>
                            bedtools BAM to BED
                           </strong>
                           converter tool
                          </li>
                         </ul>
                        </li>
                        <li>
                         Do you have a Control File?: No
                        </li>
                        <li>
                         Format of Input Files: Single-end BED
                        </li>
                        <li>
                         Effective genome size:
                         <em>
                          H. sapiens
                         </em>
                         (2.7e9)
                        </li>
                        <li>
                         Build Model: Do not build the shifting model (--nomodel)
                         <ul dir="auto">
                          <li>
                           Set extension size: 200
                          </li>
                          <li>
                           Set shift size: -100. It needs to be - half the extension size to be centered on the 5'.
                          </li>
                         </ul>
                        </li>
                        <li>
                         In Additional Outputs:
                         <ul dir="auto">
                          <li>
                           Check Peaks as tabular file (compatible with MultiQC)
                          </li>
                          <li>
                           Check Peak summits
                          </li>
                          <li>
                           Check Scores in bedGraph files
                          </li>
                         </ul>
                        </li>
                        <li>
                         In  Advanced Options:
                         <ul dir="auto">
                          <li>
                           Composite broad regions: No broad regions
                           <ul dir="auto">
                            <li>
                             Use a more sophisticated signal processing approach to find subpeak summits in each enriched peak region: Yes
                            </li>
                           </ul>
                          </li>
                          <li>
                           How many duplicate tags at the exact same location are allowed?: all
                          </li>
                         </ul>
                        </li>
                       </ul>
                      </details>
                      <details>
                       <summary>
                        LINUX Implementation
                       </summary>
                       <br/>
                       <p dir="auto">
                        Install bamtools and convert bam file to bed file using bamtools:
                       </p>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bedtools bamtobed -i marked_dup.bam &gt; marked_dup.bed" dir="auto">
                        <pre><span class="pl-s1">bedtools</span> <span class="pl-s1">bamtobed</span> <span class="pl-c1">-</span><span class="pl-s1">i</span> <span class="pl-s1">marked_dup</span>.<span class="pl-s1">bam</span> <span class="pl-c1">&gt;</span> <span class="pl-s1">marked_dup</span>.<span class="pl-s1">bed</span></pre>
                       </div>
                       <p dir="auto">
                        Install macs2 :
                       </p>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="conda install -c bioconda macs2" dir="auto">
                        <pre><span class="pl-s1">conda</span> <span class="pl-s1">install</span> <span class="pl-c1">-</span><span class="pl-s1">c</span> <span class="pl-s1">bioconda</span> <span class="pl-s1">macs2</span></pre>
                       </div>
                       <p dir="auto">
                        Then run the command for peak calling:
                       </p>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="macs2 callpeak -t marked_dup.bed -n macs_output -g 50818468 --nomodel --shift -100 --extsize 200 --keep-dup all --call-summits --bdg" dir="auto">
                        <pre><span class="pl-s1">macs2</span> <span class="pl-s1">callpeak</span> <span class="pl-c1">-</span><span class="pl-s1">t</span> <span class="pl-s1">marked_dup</span>.<span class="pl-s1">bed</span> <span class="pl-c1">-</span><span class="pl-s1">n</span> <span class="pl-s1">macs_output</span> <span class="pl-c1">-</span><span class="pl-s1">g</span> <span class="pl-c1">50818468</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">nomodel</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">shift</span> <span class="pl-c1">-</span><span class="pl-c1">100</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">extsize</span> <span class="pl-c1">200</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">keep</span><span class="pl-c1">-</span><span class="pl-s1">dup</span> <span class="pl-s1">all</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">call</span><span class="pl-c1">-</span><span class="pl-s1">summits</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">bdg</span></pre>
                       </div>
                       <p dir="auto">
                        <em>
                         <strong>
                          This will give us the following 5 output files-
                         </strong>
                        </em>
                       </p>
                       <p dir="auto">
                        -Macs_output_control_lambda.bdg
-macs_output_peaks.narrowPeak
-Macs_output_peaks.xls
-Macs_output_summits.bed
-macs_output_treat_pileup.bdg
                       </p>
                      </details>
                      <div class="markdown-heading" dir="auto">
                       <h3 class="heading-element" dir="auto" tabindex="-1">
                        STEP 5 :-Visualisation of Coverage
                       </h3>
                       <a aria-label="Permalink: STEP 5 :-Visualisation of Coverage" class="anchor" href="#step-5--visualisation-of-coverage" id="user-content-step-5--visualisation-of-coverage">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <div class="markdown-heading" dir="auto">
                       <h4 class="heading-element" dir="auto" tabindex="-1">
                        A) Prepare the Datasets
                       </h4>
                       <a aria-label="Permalink: A) Prepare the Datasets" class="anchor" href="#a-prepare-the-datasets" id="user-content-a-prepare-the-datasets">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <ol dir="auto">
                       <li>
                        <strong>
                         Extract CTCF peaks on chr22 in intergenic regions
                        </strong>
                       </li>
                      </ol>
                      <ul dir="auto">
                       <li>
                        As our training dataset is focused on chromosome 22 we will only use the CTCF peaks from chr22. We expect to have ATAC-seq coverage at TSS but only good ATAC-seq have coverage on intergenic CTCF. Indeed, the CTCF protein is able to position nucleosomes and creates a region depleted of nucleosome of around 120bp. This is smaller than the 200bp nucleosome-free region around TSS and also probably not present in all cells. Thus it is more difficult to get enrichment.
                       </li>
                      </ul>
                      <details>
                       <summary>
                        Galaxy Implementation
                       </summary>
                       <br/>
                       <p dir="auto">
                        In order to get the list of intergenic CTCF peaks of chr22, select the peaks on chr22 and then exclude the one which overlap with genes.
                       </p>
                       <ul dir="auto">
                        <li>
                         <strong>
                          Filter data on any column using simple expressions
                         </strong>
                         with the following parameters:
                         <ul dir="auto">
                          <li>
                           Filter : Select the first dataset: ENCFF933NTR.bed.gz
                          </li>
                          <li>
                           With following condition: c1=='chr22'
                          </li>
                         </ul>
                        </li>
                        <li>
                         <strong>
                          bedtools Intersect intervals find overlapping intervals in various ways
                         </strong>
                         with the following parameters:
                         <ul dir="auto">
                          <li>
                           File A to intersect with B: Select the output of
                           <strong>
                            Filter
                           </strong>
                           data on any column using simple expressions tool
                          </li>
                          <li>
                           Combined or separate output files: One output file per 'input B' file
                           <ul dir="auto">
                            <li>
                             File B to intersect with A: Select the dataset chr22 genes
                            </li>
                           </ul>
                          </li>
                          <li>
                           What should be written to the output file?: Write the original entry in A for each overlap (-wa)
                          </li>
                          <li>
                           Required overlap: Default: 1bp
                          </li>
                          <li>
                           Report only those alignments that
                           <strong>
                            do not
                           </strong>
                           overlap with file(s) B: Yes
                          </li>
                         </ul>
                        </li>
                        <li>
                         Rename the datasets intergenic CTCF peaks chr22.
                        </li>
                       </ul>
                      </details>
                      <details>
                       <summary>
                        LINUX Implementation
                       </summary>
                       <br/>
                       1. Filter only data for chr22 from file using
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='grep -w "chr22" ENCFF933NTR.bed &gt;&gt; file_A.bed' dir="auto">
                        <pre><span class="pl-s1">grep</span> <span class="pl-c1">-</span><span class="pl-s1">w</span> <span class="pl-s">"chr22"</span> <span class="pl-v">ENCFF933NTR</span>.<span class="pl-s1">bed</span> <span class="pl-c1">&gt;&gt;</span> <span class="pl-s1">file_A</span>.<span class="pl-s1">bed</span></pre>
                       </div>
                       <p dir="auto">
                        2.Extract filtered chrr22 (as c1) into a new file
                       </p>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="grep c1 ENCFF933NTR_filt.bed &gt; ENCFF933NTR_chr22.bed" dir="auto">
                        <pre><span class="pl-s1">grep</span> <span class="pl-s1">c1</span> <span class="pl-v">ENCFF933NTR_filt</span>.<span class="pl-s1">bed</span> <span class="pl-c1">&gt;</span> <span class="pl-v">ENCFF933NTR_chr22</span>.<span class="pl-s1">bed</span></pre>
                       </div>
                       <p dir="auto">
                        3.Replace c1 with chr22
                       </p>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="sed 's/c1/chr22/' ENCFF933NTR_chr22.bed &gt; ENCFF933NTR_CHR22genes.bed" dir="auto">
                        <pre><span class="pl-s1">sed</span> <span class="pl-s">'s/c1/chr22/'</span> <span class="pl-v">ENCFF933NTR_chr22</span>.<span class="pl-s1">bed</span> <span class="pl-c1">&gt;</span> <span class="pl-v">ENCFF933NTR_CHR22genes</span>.<span class="pl-s1">bed</span></pre>
                       </div>
                       <p dir="auto">
                        4.bedtools Intersect intervals find overlapping intervals :
                       </p>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bedtools intersect -v -a ENCFF933NTR_CHR22genes.bed -b chr22_genes.bed &gt; intergenic_CTCF_peaks_chr22" dir="auto">
                        <pre><span class="pl-s1">bedtools</span> <span class="pl-s1">intersect</span> <span class="pl-c1">-</span><span class="pl-s1">v</span> <span class="pl-c1">-</span><span class="pl-s1">a</span> <span class="pl-v">ENCFF933NTR_CHR22genes</span>.<span class="pl-s1">bed</span> <span class="pl-c1">-</span><span class="pl-s1">b</span> <span class="pl-s1">chr22_genes</span>.<span class="pl-s1">bed</span> <span class="pl-c1">&gt;</span> <span class="pl-s1">intergenic_CTCF_peaks_chr22</span></pre>
                       </div>
                      </details>
                      <div class="markdown-heading" dir="auto">
                       <h4 class="heading-element" dir="auto" tabindex="-1">
                        B)Convert bedgraph from MACS2 to bigwig
                       </h4>
                       <a aria-label="Permalink: B)Convert bedgraph from MACS2 to bigwig" class="anchor" href="#bconvert-bedgraph-from-macs2-to-bigwig" id="user-content-bconvert-bedgraph-from-macs2-to-bigwig">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <ul dir="auto">
                       <li>
                        To visualise any region of the genome very quickly, we need to change the bedgraph format from MACS2 output to binary bigwig format.
                       </li>
                      </ul>
                      <details>
                       <summary>
                        Galaxy Implementation
                       </summary>
                       <br/>
                       <strong>
                        Wig/BedGraph-to-bigWig
                       </strong>
                       with the following parameters:
                       <ul dir="auto">
                        <li>
                         Convert: Select the output of
                         <strong>
                          MACS2
                         </strong>
                         tool (Bedgraph Treatment).
                        </li>
                        <li>
                         Converter settings to use: Default
                        </li>
                        <li>
                         Rename the datasets MACS2 bigwig.
                        </li>
                       </ul>
                      </details>
                      <details>
                       <summary>
                        LINUX Implementation
                       </summary>
                       <br/>
                       Install
                       <strong>
                        bedGraphtoBigWig
                       </strong>
                       and go through the following commands for converting the output bedGraph file from macs2 to bigwig (refer to this link if you want to understand the commands [
                       <a href="https://www.biostars.org/p/176875/](https://www.biostars.org/p/176875/)" rel="nofollow">
                        https://www.biostars.org/p/176875/](https://www.biostars.org/p/176875/)
                       </a>
                       )
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" awk 'NR!=1' macs_output_treat_pileup.bdg &gt; macs.deheader.bedGraph" dir="auto">
                        <pre> <span class="pl-s1">awk</span> <span class="pl-s">'NR!=1'</span> <span class="pl-s1">macs_output_treat_pileup</span>.<span class="pl-s1">bdg</span> <span class="pl-c1">&gt;</span> <span class="pl-s1">macs</span>.<span class="pl-s1">deheader</span>.<span class="pl-s1">bedGraph</span></pre>
                       </div>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" sort -k1,1 -k2,2n macs.deheader.bedGraph &gt; macs.sorted.bedGraph " dir="auto">
                        <pre> <span class="pl-s1">sort</span> <span class="pl-c1">-</span><span class="pl-s1">k1</span>,<span class="pl-c1">1</span> <span class="pl-c1">-</span><span class="pl-s1">k2</span>,<span class="pl-c1">2</span><span class="pl-s1">n</span> <span class="pl-s1">macs</span>.<span class="pl-s1">deheader</span>.<span class="pl-s1">bedGraph</span> <span class="pl-c1">&gt;</span> <span class="pl-s1">macs</span>.<span class="pl-s1">sorted</span>.<span class="pl-s1">bedGraph</span> </pre>
                       </div>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" touch chrom22.sizes" dir="auto">
                        <pre> <span class="pl-s1">touch</span> <span class="pl-s1">chrom22</span>.<span class="pl-s1">sizes</span></pre>
                       </div>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" nano hg19.chrom.sizes" dir="auto">
                        <pre> <span class="pl-s1">nano</span> <span class="pl-s1">hg19</span>.<span class="pl-s1">chrom</span>.<span class="pl-s1">sizes</span></pre>
                       </div>
                       <p dir="auto">
                         write only one line (tab delimited) in this file chr22 51304566
                       </p>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" awk '{print $1,$2,$3,$4}' macs.sorted.bedGraph &gt; macs.sorted.4.bedGraph" dir="auto">
                        <pre> <span class="pl-s1">awk</span> <span class="pl-s">'{print $1,$2,$3,$4}'</span> <span class="pl-s1">macs</span>.<span class="pl-s1">sorted</span>.<span class="pl-s1">bedGraph</span> <span class="pl-c1">&gt;</span> <span class="pl-s1">macs</span>.<span class="pl-s1">sorted</span>.<span class="pl-c1">4.</span><span class="pl-s1">bedGraph</span></pre>
                       </div>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" bedGraphToBigWig macs.sorted.4.bedGraph hg19.chrom.sizes macs.bw " dir="auto">
                        <pre> <span class="pl-s1">bedGraphToBigWig</span> <span class="pl-s1">macs</span>.<span class="pl-s1">sorted</span>.<span class="pl-c1">4.</span><span class="pl-s1">bedGraph</span> <span class="pl-s1">hg19</span>.<span class="pl-s1">chrom</span>.<span class="pl-s1">sizes</span> <span class="pl-s1">macs</span>.<span class="pl-s1">bw</span> </pre>
                       </div>
                      </details>
                      <div class="markdown-heading" dir="auto">
                       <h4 class="heading-element" dir="auto" tabindex="-1">
                        C) Create heatmap of coverage at TSS with deepTool
                       </h4>
                       <a aria-label="Permalink: C) Create heatmap of coverage at TSS with deepTool" class="anchor" href="#c-create-heatmap-of-coverage-at-tss-with-deeptool" id="user-content-c-create-heatmap-of-coverage-at-tss-with-deeptool">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <ul dir="auto">
                       <li>
                        For checking the coverage on specific regions, we can compute heatmap. ComputeMatrix &amp; plotHeatmap from deepTools are useful for this purpose. We will here make a heatmap centered on the transcription start sites (TSS) and another one centered on intergenic CTCF peaks.
                       </li>
                      </ul>
                      <details>
                       <summary>
                        Galaxy Implementation
                       </summary>
                       <br/>
                       -
                       <strong>
                        computeMatrix
                       </strong>
                       with the following parameters:
  - In  Select regions:
    - Insert Select regions
      - Regions to plot: Select the dataset chr22 genes
  - Sample order matters: No
    - Score file: Select the output of  **Wig/BedGraph-to-bigWig**  tool that should be named MACS2 bigwig.
  - computeMatrix has two main output options: reference-point
  - The reference point for the plotting: beginning of region (e.g. TSS)
  - Show advanced output settings: no
  - Show advanced options: yes
  - Convert missing values to 0?: Yes
  1. **Plot with plotHeatmap**
                       <ul dir="auto">
                        <li>
                         <strong>
                          plotHeatmap
                         </strong>
                         with the following parameters:
                         <ul dir="auto">
                          <li>
                           Matrix file from the computeMatrix tool: Select the output of
                           <strong>
                            computeMatrix
                           </strong>
                           tool.
                          </li>
                          <li>
                           Show advanced output settings: no
                          </li>
                          <li>
                           Show advanced options: no
                          </li>
                         </ul>
                        </li>
                       </ul>
                       <p dir="auto">
                        The same is repeated for the intergenic CTCF peaks.
                       </p>
                       <p dir="auto">
                        <strong>
                         Generate the matrix
                        </strong>
                       </p>
                       <ul dir="auto">
                        <li>
                         <p dir="auto">
                          <strong>
                           computeMatrix
                          </strong>
                          with the following parameters:
                         </p>
                         <ul dir="auto">
                          <li>
                           In Select regions:
                           <ul dir="auto">
                            <li>
                             Insert Select regions
                             <ul dir="auto">
                              <li>
                               Regions to plot: Select the dataset intergenic CTCF peaks chr22
                              </li>
                             </ul>
                            </li>
                           </ul>
                          </li>
                          <li>
                           Sample order matters: No
                           <ul dir="auto">
                            <li>
                             Score file: Select the output of
                             <strong>
                              Wig/BedGraph-to-bigWig
                             </strong>
                             tool that should be named MACS2 bigwig.
                            </li>
                           </ul>
                          </li>
                          <li>
                           Would you like custom sample labels?: No, use sample names in the history
                          </li>
                          <li>
                           computeMatrix has two main output options: reference-point
                           <ul dir="auto">
                            <li>
                             The reference point for the plotting: center of region
                            </li>
                           </ul>
                          </li>
                          <li>
                           Show advanced output settings: no
                          </li>
                          <li>
                           Show advanced options: yes
                           <ul dir="auto">
                            <li>
                             Convert missing values to 0?: Yes
                            </li>
                           </ul>
                          </li>
                         </ul>
                        </li>
                        <li>
                         <p dir="auto">
                          <strong>
                           plotHeatmap
                          </strong>
                          with the following parameters:
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          Matrix file from the computeMatrix tool: Select the output of
                          <strong>
                           computeMatrix
                          </strong>
                          tool.
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          Show advanced output settings: no
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          Show advanced options: yes
                         </p>
                         <ul dir="auto">
                          <li>
                           In Colormap to use for each sample:
                           <ul dir="auto">
                            <li>
                             Insert Colormap to use for each sample
                             <ol dir="auto">
                              <li>
                               Color map to use for the heatmap: your choice
                              </li>
                             </ol>
                            </li>
                           </ul>
                          </li>
                          <li>
                           The x-axis label: distance from peak center (bp)
                          </li>
                          <li>
                           The y-axis label for the top panel: CTCF peaks
                          </li>
                          <li>
                           Reference point label: peak center
                          </li>
                          <li>
                           Labels for the regions plotted in the heatmap: CTCF_peaks
                          </li>
                          <li>
                           Did you compute the matrix with more than one groups of regions?: Yes, I used multiple groups of regions
                          </li>
                         </ul>
                        </li>
                       </ul>
                      </details>
                      <details>
                       <summary>
                        LINUX Implementation
                       </summary>
                       <br/>
                       - Using
                       <strong>
                        computeMatrix
                       </strong>
                       generate the matrix
                       <ol dir="auto">
                        <li>
                         Remove the first header line from chr22.bed file
                        </li>
                        <li>
                         Then run
                        </li>
                       </ol>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="computeMatrix reference-point --referencePoint TSS -R chr22.bed -S macs.bw --missingDataAsZero -o output_from_computeMatrix.gz" dir="auto">
                        <pre><span class="pl-s1">computeMatrix</span> <span class="pl-s1">reference</span><span class="pl-c1">-</span><span class="pl-s1">point</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">referencePoint</span> <span class="pl-v">TSS</span> <span class="pl-c1">-</span><span class="pl-v">R</span> <span class="pl-s1">chr22</span>.<span class="pl-s1">bed</span> <span class="pl-c1">-</span><span class="pl-v">S</span> <span class="pl-s1">macs</span>.<span class="pl-s1">bw</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">missingDataAsZero</span> <span class="pl-c1">-</span><span class="pl-s1">o</span> <span class="pl-s1">output_from_computeMatrix</span>.<span class="pl-s1">gz</span></pre>
                       </div>
                       <ul dir="auto">
                        <li>
                         plotHeatmap will generate the plot using the output of computeMatrix
                        </li>
                       </ul>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" plotHeatmap -m output_from_computeMatrix.gz -out plotHeatMap.png" dir="auto">
                        <pre> <span class="pl-s1">plotHeatmap</span> <span class="pl-c1">-</span><span class="pl-s1">m</span> <span class="pl-s1">output_from_computeMatrix</span>.<span class="pl-s1">gz</span> <span class="pl-c1">-</span><span class="pl-s1">out</span> <span class="pl-s1">plotHeatMap</span>.<span class="pl-s1">png</span></pre>
                       </div>
                       <ul dir="auto">
                        <li>
                         Repeating the previous two steps for plotting
                         <strong>
                          CTCF peaks of chr22 in intergenic regions
                         </strong>
                         with slight moderation:
                        </li>
                       </ul>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" computeMatrix reference-point --referencePoint center -R intergenic_ctcf_peaks_chr22 -S macs.bw --missingDataAsZero -o peak_output_from_computeMatrix.gz" dir="auto">
                        <pre> <span class="pl-s1">computeMatrix</span> <span class="pl-s1">reference</span><span class="pl-c1">-</span><span class="pl-s1">point</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">referencePoint</span> <span class="pl-s1">center</span> <span class="pl-c1">-</span><span class="pl-v">R</span> <span class="pl-s1">intergenic_ctcf_peaks_chr22</span> <span class="pl-c1">-</span><span class="pl-v">S</span> <span class="pl-s1">macs</span>.<span class="pl-s1">bw</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">missingDataAsZero</span> <span class="pl-c1">-</span><span class="pl-s1">o</span> <span class="pl-s1">peak_output_from_computeMatrix</span>.<span class="pl-s1">gz</span></pre>
                       </div>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" plotHeatmap -m peak_output_from_computeMatrix.gz -out intragenic_plotHeatMap.png" dir="auto">
                        <pre> <span class="pl-s1">plotHeatmap</span> <span class="pl-c1">-</span><span class="pl-s1">m</span> <span class="pl-s1">peak_output_from_computeMatrix</span>.<span class="pl-s1">gz</span> <span class="pl-c1">-</span><span class="pl-s1">out</span> <span class="pl-s1">intragenic_plotHeatMap</span>.<span class="pl-s1">png</span></pre>
                       </div>
                      </details>
                      <p dir="auto">
                       In the generated heatmaps, each line will be a transcript. The coverage will be summarized with a color code from red (no coverage) to blue (maximum coverage). All TSS will be aligned in the middle of the figure and only the 2 kb around the TSS will be displayed. Another plot, on top of the heatmap, will show the mean signal at the TSS. There will be one heatmap per bigwig.
                      </p>
                      <p dir="auto">
                       For TSS, our data gives the following heatmap-
                      </p>
                      <p align="center" dir="auto">
                       <a href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/MACS2%20bigwig.PNG" rel="noopener noreferrer" target="_blank">
                        <img src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/MACS2%20bigwig.PNG" style="max-width: 100%;"/>
                       </a>
                      </p>
                      <p dir="auto">
                       The plot on top shows a non-symmetric pattern that is higher on the left, which is expected as usually the promoter of active genes is accessible.
                      </p>
                      <p dir="auto">
                       For CTCF peaks of chr22 in intergenic regions, the following heatmap is generated from our data-
                      </p>
                      <p align="center" dir="auto">
                       <a href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/plotHeatmap%20output.PNG" rel="noopener noreferrer" target="_blank">
                        <img src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/plotHeatmap%20output.PNG" style="max-width: 100%;"/>
                       </a>
                      </p>
                      <p dir="auto">
                       This heatmap is showing a much more symmetric pattern.
                      </p>
                      <div class="markdown-heading" dir="auto">
                       <h4 class="heading-element" dir="auto" tabindex="-1">
                        D)Visualise Regions with pyGenomeTracks
                       </h4>
                       <a aria-label="Permalink: D)Visualise Regions with pyGenomeTracks" class="anchor" href="#dvisualise-regions-with-pygenometracks" id="user-content-dvisualise-regions-with-pygenometracks">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <ul dir="auto">
                       <li>
                        In order to visualise a specific region (e.g. the gene
                        <em>
                         RAC2
                        </em>
                        ), we will use pyGenomeTracks
                       </li>
                      </ul>
                      <details>
                       <summary>
                        Galaxy Implementation
                       </summary>
                       <br/>
                       <ul dir="auto">
                        <li>
                         <strong>
                          pyGenomeTracks
                         </strong>
                         Tool with the following parameters:
                        </li>
                        <li>
                         Region of the genome to limit the operation: chr22:37,193,000-37,252,000
                        </li>
                        <li>
                         In Include tracks in your plot:
                         <ul dir="auto">
                          <li>
                           Insert Include tracks in your plot
                           <ul dir="auto">
                            <li>
                             Choose style of the track: Bigwig track
                             <ol dir="auto">
                              <li>
                               Plot title: Coverage from MACS2 (extended +/-100bp)
                              </li>
                              <li>
                               Track file(s) bigwig format: Select the output of Wig/BedGraph-to-bigWig tool called MACS2 bigwig.
                              </li>
                              <li>
                               Color of track: Select the color of your choice
                              </li>
                              <li>
                               Minimum value: 0
                              </li>
                              <li>
                               height: 5
                              </li>
                              <li>
                               Show visualization of data range: Yes
                              </li>
                             </ol>
                            </li>
                           </ul>
                          </li>
                          <li>
                           Insert Include tracks in your plot
                           <ul dir="auto">
                            <li>
                             Choose style of the track: NarrowPeak track
                             <ol dir="auto">
                              <li>
                               Plot title: Peaks from MACS2 (extended +/-100bp)
                              </li>
                              <li>
                               Track file(s) encodepeak or bed format: Select the output of MACS2 tool (narrow Peaks).
                              </li>
                              <li>
                               Color of track: Select the color of your choice
                              </li>
                              <li>
                               display to use: box: Draw a box
                              </li>
                              <li>
                               Plot labels (name, p-val, q-val): No
                              </li>
                             </ol>
                            </li>
                           </ul>
                          </li>
                          <li>
                           Insert Include tracks in your plot
                           <ul dir="auto">
                            <li>
                             Choose style of the track: Gene track / Bed track
                             <ol dir="auto">
                              <li>
                               Plot title: Genes
                              </li>
                              <li>
                               Track file(s) bed or gtf format: chr22 genes
                              </li>
                              <li>
                               Color of track: Select the color of your choice
                              </li>
                              <li>
                               height: 5
                              </li>
                              <li>
                               Plot labels: yes
                              </li>
                              <li>
                               Put all labels inside the plotted region: Yes
                              </li>
                              <li>
                               Allow to put labels in the right margin: Yes
                              </li>
                             </ol>
                            </li>
                           </ul>
                          </li>
                          <li>
                           Insert Include tracks in your plot
                           <ul dir="auto">
                            <li>
                             Choose style of the track: NarrowPeak track
                             <ol dir="auto">
                              <li>
                               Plot title: CTCF peaks
                              </li>
                              <li>
                               Track file(s) encodepeak or bed format: Select the first dataset: ENCFF933NTR.bed.gz
                              </li>
                              <li>
                               Color of track: Select the color of your choice
                              </li>
                              <li>
                               display to use: box: Draw a box
                              </li>
                              <li>
                               Plot labels (name, p-val, q-val): No
                              </li>
                             </ol>
                            </li>
                           </ul>
                          </li>
                          <li>
                           param-repeat Insert Include tracks in your plot
                           <ul dir="auto">
                            <li>
                             Choose style of the track: X-axis
                            </li>
                           </ul>
                          </li>
                         </ul>
                        </li>
                       </ul>
                       <p align="center" dir="auto">
                        <a href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/pyGenome%20output.PNG" rel="noopener noreferrer" target="_blank">
                         <img src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/pyGenome%20output.PNG" style="max-width: 100%;"/>
                        </a>
                       </p>
                      </details>
                      <details>
                       <summary>
                        Linux Implementation
                       </summary>
                       <br/>
                       <p dir="auto">
                        <strong>
                         Set up the config.ini file with the following contents-
                        </strong>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          [test bedgraph]
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          file = macs.bw
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          color = blue
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          height = 5
                         </strong>
                        </i>
                        <i>
                         <strong>
                          title = Coverage from MACS2 (extended +/-100bp)
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          min_value = 0
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          [spacer]
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          height = 0.5
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          [narrow]
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          file = intergenic_ctcf_peaks_chr22.encodepeak
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          line_width = 2
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          title = Peaks from MACS2 (extended +/-100bp)
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          type = box
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          color = red
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          show_labels = false
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          file_type = narrow_peak
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          [spacer]
                         </strong>
                        </i>
                        <i>
                         <strong>
                          height = 0.5
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          [genes 0]
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          file = chr22.bed
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          height = 7
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          title = genes
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          height = 5
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          color = #ffbbff
                         </strong>
                        </i>
                        <i>
                         <strong>
                          [spacer]
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          height = 0.5
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          [narrow 1]
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          file = ENCFF933NTR_sorted.bed
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          color = #A020F0
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          line_width = 2
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          title = CTCF peaks
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          type = box
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          show_labels = false
                         </strong>
                        </i>
                       </p>
                       <p dir="auto">
                        <i>
                         <strong>
                          [x-axis]
                         </strong>
                        </i>
                       </p>
                       <ul dir="auto">
                        <li>
                         Sort ENCFF933NTR.bed file-
                        </li>
                       </ul>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" sort -k 1,1 -k2,2n ENCFF933NTR.bed &gt; ENCFF933NTR_sorted.bed" dir="auto">
                        <pre> <span class="pl-s1">sort</span> <span class="pl-c1">-</span><span class="pl-s1">k</span> <span class="pl-c1">1</span>,<span class="pl-c1">1</span> <span class="pl-c1">-</span><span class="pl-s1">k2</span>,<span class="pl-c1">2</span><span class="pl-s1">n</span> <span class="pl-v">ENCFF933NTR</span>.<span class="pl-s1">bed</span> <span class="pl-c1">&gt;</span> <span class="pl-v">ENCFF933NTR_sorted</span>.<span class="pl-s1">bed</span></pre>
                       </div>
                       <ul dir="auto">
                        <li>
                         Install pyGenomeTracks using
                         <br/>
                        </li>
                       </ul>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="conda -install -c bioconda pyGenomeTracks" dir="auto">
                        <pre><span class="pl-s1">conda</span> <span class="pl-c1">-</span><span class="pl-s1">install</span> <span class="pl-c1">-</span><span class="pl-s1">c</span> <span class="pl-s1">bioconda</span> <span class="pl-s1">pyGenomeTracks</span></pre>
                       </div>
                       <ul dir="auto">
                        <li>
                         Visualize regions by running-
                         <br/>
                        </li>
                       </ul>
                       <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="pyGenomeTracks --tracks config.ini --region chr22:37,193,000-37,252,000 -o Genome_track_plot.png" dir="auto">
                        <pre><span class="pl-s1">pyGenomeTracks</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">tracks</span> <span class="pl-s1">config</span>.<span class="pl-s1">ini</span> <span class="pl-c1">-</span><span class="pl-c1">-</span><span class="pl-s1">region</span> <span class="pl-s1">chr22</span>:<span class="pl-c1">37</span>,<span class="pl-c1">193</span>,<span class="pl-c1">000</span><span class="pl-c1">-</span><span class="pl-c1">37</span>,<span class="pl-c1">252</span>,<span class="pl-c1">000</span> <span class="pl-c1">-</span><span class="pl-s1">o</span> <span class="pl-v">Genome_track_plot</span>.<span class="pl-s1">png</span></pre>
                       </div>
                      </details>
                      <p align="center" dir="auto">
                       <a href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/blob/main/images/pyGenome%20output.PNG" rel="noopener noreferrer" target="_blank">
                        <img src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/raw/main/images/pyGenome%20output.PNG" style="max-width: 100%;"/>
                       </a>
                      </p>
                      <ul dir="auto">
                       <li>
                        From the figure, we can see 3 accessible TSS for 6 transcripts for 2 genes. The TSS of RAC2 corresponds to an ATAC-Seq peak whereas there is no significant coverage on both TSS of SSTR3. Again, it can be said that only the first peak on the left overlaps with a CTCF binding site represents accessible loci. Amongst the 4 peaks in this plotted region, the 2 peaks in the middle do not correspond to CTCF peaks or TSS. As CTCF creates accessible regions, a region containing a peak with no corresponding CTCF peak or TSS could be a putative enhancer. In the pyGenomeTracks plot we see a region like this located in the intron of a gene and another one between genes. More analyses are needed to assess if it is a real enhancer, for example, histone ChIP-seq, 3D structure, transgenic assay, etc.
                       </li>
                      </ul>
                      <p dir="auto">
                       As CTCF creates accessible regions, a region containing a peak with no corresponding CTCF peak or TSS could be a putative enhancer. In the pyGenomeTracks plot we see a region like this located in the intron of a gene and another one between genes.
                      </p>
                      <div class="markdown-heading" dir="auto">
                       <h2 class="heading-element" dir="auto" tabindex="-1">
                        Conclusion
                       </h2>
                       <a aria-label="Permalink: Conclusion" class="anchor" href="#conclusion" id="user-content-conclusion">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <p dir="auto">
                       ATAC-Seq is a method to investigate the chromatin accessibility and the genome is treated with a transposase (enzyme) called Tn5. It marks open chromatin regions by cutting and inserting adapters for sequencing. Low quality bases, adapter contamination, correct insert size and PCR duplicates (duplication level) were checked. Mapped the reads with
                       <strong>
                        Bowtie2
                       </strong>
                       , filtered the reads for properly paired, good quality and reads that do not map to the mitochondrial genome. Open chromatin regions were found with
                       <strong>
                        MACS2
                       </strong>
                       , a tool to find regions of genomic enrichment (peaks). The read coverage around TSS was investigated with the help of
                       <strong>
                        computeMatrix
                       </strong>
                       and
                       <strong>
                        plotHeatmap
                       </strong>
                       . The peaks and other informative tracks, such as CTCF binding regions and hg38 genes were visualised with the help of
                       <strong>
                        pyGenomeTracks
                       </strong>
                       . At the end, open chromatin regions that did not overlap with CTCF sites or TSS, which could be potential putative enhancer regions detected by the ATAC-Seq experiment.
                      </p>
                      <div class="markdown-heading" dir="auto">
                       <h2 class="heading-element" dir="auto" tabindex="-1">
                        REFERENCES
                       </h2>
                       <a aria-label="Permalink: REFERENCES" class="anchor" href="#references" id="user-content-references">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <ol dir="auto">
                       <li>
                        Lucille Delisle, Maria Doyle, Florian Heyl, 2021
                        <strong>
                         ATAC-Seq data analysis (Galaxy Training Materials)
                        </strong>
                        .
                        <a href="https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html" rel="nofollow">
                         https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html
                        </a>
                        Online; accessed Thu Aug 19 2021
                       </li>
                      </ol>
                      <div class="markdown-heading" dir="auto">
                       <h2 class="heading-element" dir="auto" tabindex="-1">
                         TOOL USED  
                       </h2>
                       <a aria-label="Permalink:  TOOL USED  " class="anchor" href="#-tool-used--" id="user-content--tool-used--">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <ul class="contains-task-list">
                       <li class="task-list-item">
                        <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                        Fastqc
                       </li>
                       <li class="task-list-item">
                        <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                        Cutadapt
                       </li>
                       <li class="task-list-item">
                        <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                        Bowtie2
                       </li>
                       <li class="task-list-item">
                        <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                        Samtools
                       </li>
                       <li class="task-list-item">
                        <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                        Picard
                        <ul dir="auto">
                         <li>
                          CollectInsertSizeMetrics
                         </li>
                         <li>
                          MarkDuplicates
                         </li>
                        </ul>
                       </li>
                       <li class="task-list-item">
                        <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                        Bedtools
                       </li>
                       <li class="task-list-item">
                        <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                        DeepTool
                        <ul dir="auto">
                         <li>
                          ComputeMatrix
                         </li>
                         <li>
                          PlotHeatmap
                         </li>
                        </ul>
                       </li>
                       <li class="task-list-item">
                        <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                        MACS2
                       </li>
                       <li class="task-list-item">
                        <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                        BedGraphtoBigWig
                       </li>
                       <li class="task-list-item">
                        <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                        PyGenomeTracks
                       </li>
                      </ul>
                      <div class="markdown-heading" dir="auto">
                       <h3 class="heading-element" dir="auto" tabindex="-1">
                         TEAM WORK CONTRIBUTION 
                       </h3>
                       <a aria-label="Permalink: :two_men_holding_hands: TEAM WORK CONTRIBUTION :two_women_holding_hands:" class="anchor" href="#two_men_holding_hands-team-work-contribution-two_women_holding_hands" id="user-content-two_men_holding_hands-team-work-contribution-two_women_holding_hands">
                        <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                         <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                         </path>
                        </svg>
                       </a>
                      </div>
                      <table>
                       <thead>
                        <tr>
                         <th>
                          Slack username
                         </th>
                         <th>
                          Suggesting workflows
                         </th>
                         <th>
                          Workflow Design
                         </th>
                         <th>
                          Galaxy Based Implementation
                         </th>
                         <th>
                          Linux Based Implementation
                         </th>
                         <th>
                          Creating github repo &amp; Markdown file formatting
                         </th>
                         <th>
                          Writing the galaxy workflow for markdown
                         </th>
                         <th>
                          Writing the Linux workflow for markdown
                         </th>
                         <th>
                          Others
                         </th>
                        </tr>
                       </thead>
                       <tbody>
                        <tr>
                         <td>
                          @Sanjana1404
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Comfortojedapo
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          Team Lead
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Jaswanth
                         </td>
                         <td>
                          
                         </td>
                         <td>
                          
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Jeeel193
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          Creating a new workflow design for markdown file
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Sathya
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Malu
                         </td>
                         <td>
                          
                         </td>
                         <td>
                          
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Adams
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Mercyme
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Wealth88
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Maryamabdul
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Nishat
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          Team Lead
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Akinmode
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                          Creating the workflow design for marketing  at the transfer market
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @XR2
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Kehinde16
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Sakshi_r_
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                        </tr>
                        <tr>
                         <td>
                          @Tardigrade
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                         </td>
                         <td>
                          Provided Result Images
                         </td>
                        </tr>
                       </tbody>
                      </table>
                      <p align="center" dir="auto">
                       <a href="https://camo.githubusercontent.com/1a21b31e2fb456e8d7f5c15e8c17c55a9ec495614f08f4df84c552feac206fd6/68747470733a2f2f616c6c667265657468616e6b796f756e6f7465732e636f6d2f77702d636f6e74656e742f75706c6f6164732f323032312f30322f616c6c2d667265652d7468616e6b2d796f752d6769662d31302e676966" rel="noopener noreferrer nofollow" target="_blank">
                        <img data-animated-image="" data-canonical-src="https://allfreethankyounotes.com/wp-content/uploads/2021/02/all-free-thank-you-gif-10.gif" src="https://camo.githubusercontent.com/1a21b31e2fb456e8d7f5c15e8c17c55a9ec495614f08f4df84c552feac206fd6/68747470733a2f2f616c6c667265657468616e6b796f756e6f7465732e636f6d2f77702d636f6e74656e742f75706c6f6164732f323032312f30322f616c6c2d667265652d7468616e6b2d796f752d6769662d31302e676966" style="max-width: 100%;"/>
                       </a>
                      </p>
                     </b>
                    </b>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="CzUG6TWPdf8y53PaJD2n/pU0w8PVMoXHvmWhTElnJ449EBwKZ4dFPFn+ixTLOViVrlbzhbJtejR0VhqZDlsS2w==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 THIS REPOSITORY CONTAINS STAGE 2 TASK FROM HACKBIO INTERNSHIP 2021
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  star
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   2
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   5
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2FDSJaswanth%2FHACKBIO_2021_TEAM_EPIGENOMICS1&amp;report=DSJaswanth+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/DSJaswanth/packages?repo_name=HACKBIO_2021_TEAM_EPIGENOMICS1">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/graphs/contributors">
                 Contributors
                 <span class="Counter ml-1" data-view-component="true" title="6">
                  6
                 </span>
                </a>
               </h2>
               <include-fragment aria-busy="true" aria-label="Loading contributors" src="/DSJaswanth/HACKBIO_2021_TEAM_EPIGENOMICS1/contributors_list?count=6&amp;current_repository=HACKBIO_2021_TEAM_EPIGENOMICS1&amp;items_to_show=6">
                <ul class="list-style-none d-flex flex-wrap mb-n2">
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                </ul>
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
        2024 GitHub,Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You cant perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
