<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-bf4f9425a973.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-340b2bef927d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_github_catalyst_lib_index_-eccae9-1932eeecf006.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-318896a86f35.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-9fad2423070b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - ErnestoDibia/Genomics_one_c-Project
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:B54ADA:10AB24C:66059A02" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDpCNTRBREE6MTBBQjI0Qzo2NjA1OUEwMiIsInZpc2l0b3JfaWQiOiI2NDUyMjQxODU2MjM4ODE3ODA3IiwicmVnaW9uX2VkZ2UiOiJpYWQiLCJyZWdpb25fcmVuZGVyIjoiaWFkIn0=" data-pjax-transient="true" name="visitor-payload">
           <meta content="a9065e8cf3fa452bb8f6e42c2f977d858a7b103bf418ff39d2ba9eae6fb269aa" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:398555927" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="Contribute to ErnestoDibia/Genomics_one_c-Project development by creating an account on GitHub." name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/ErnestoDibia/Genomics_one_c-Project" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/f697ebebabd2c6f65074ae6abf9adfd67bb30aff458e1f2f8b4cb5804022e8a7/ErnestoDibia/Genomics_one_c-Project" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - ErnestoDibia/Genomics_one_c-Project" name="twitter:title">
                    <meta content="Contribute to ErnestoDibia/Genomics_one_c-Project development by creating an account on GitHub." name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/f697ebebabd2c6f65074ae6abf9adfd67bb30aff458e1f2f8b4cb5804022e8a7/ErnestoDibia/Genomics_one_c-Project" property="og:image">
                      <meta content="Contribute to ErnestoDibia/Genomics_one_c-Project development by creating an account on GitHub." property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - ErnestoDibia/Genomics_one_c-Project" property="og:title"/>
                        <meta content="https://github.com/ErnestoDibia/Genomics_one_c-Project" property="og:url"/>
                        <meta content="Contribute to ErnestoDibia/Genomics_one_c-Project development by creating an account on GitHub." property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="a81c045767350c9f2e551b272402db2e99346c1d2710552f41e25d405666b5eb" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="3037d78571ecdebb88047af8e128a53bf5dc96ac8c533579876941ff0be8ae32" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/ErnestoDibia/Genomics_one_c-Project git https://github.com/ErnestoDibia/Genomics_one_c-Project.git" name="go-import"/>
                        <meta content="88302025" name="octolytics-dimension-user_id">
                         <meta content="ErnestoDibia" name="octolytics-dimension-user_login">
                          <meta content="398555927" name="octolytics-dimension-repository_id">
                           <meta content="ErnestoDibia/Genomics_one_c-Project" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="398555927" name="octolytics-dimension-repository_network_root_id">
                               <meta content="ErnestoDibia/Genomics_one_c-Project" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/ErnestoDibia/Genomics_one_c-Project" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-99519581d0f8.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/ErnestoDibia/Genomics_one_c-Project","user_id":null}}' data-hydro-click-hmac="6691f858ea01ae532b9f2df798d8a8320f66d39708e0cf8c81df5c1a7f9e1168" href="/login?return_to=https%3A%2F%2Fgithub.com%2FErnestoDibia%2FGenomics_one_c-Project">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=5ULvGLyDrHxC9mU4tJGHEmRZwql%2FmW%2BHJnc%2BKZgXeu2S74Wdx2eEyXC0O%2FSFhRV0tpA8kn7Jz5W5dJM8Wknr2eSdmNqXdBl0jWrqzEZ%2BP7yoEj0DXyUkWaBsfyrr5CVx8RXsXOOO2k2rdkRoAkB7S9J%2BieOXIkjBAYWzniytGgRNzOuVIP1Wx8gMQH331x5Mwsa9ITqnJCjbGntL3urg%2FHoAZlSzfNPisOTfiWU%2FYroJdF73BaHbknNionhL8W8saienM1XC96NKdEDSf1dLxLsLujsy6zzH7Qn9RIcsmn0GqwSjX1npC6akVMNW09FiOs2jN3NmsAA7DXqbfLvmNOYdKQVvGIILc6xcYsErYzEtPzNxqhzak%2Bu3hatOPldH5SZamdQWElMZzwxVcrp5bxoK6Nlio3n83FOw5ehp04FQlButuIEiT%2BKh96KWivGhrLSKbawMV51YSzLwIa8nqD1xa%2BugdvtziSTOCCWB6F%2FSkrIlRCVTobO9vB1m90q%2F42McwecY0N2fU7dEHkRQgrCHOzYxsBaCl5gbSQr02edJt5rHpFc%3D--dtqm6nTXnixWVP0X--SpCGE57s2IQNToR2NJCDPA%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="ErnestoDibia" data-current-repository="ErnestoDibia/Genomics_one_c-Project" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="DOf8cywQwSo04zL2gqyr5MFCpVFLtvAcNKY1ZbTJdW_AH5-YJyfaQ3sFKgpJqqylwit_ycdDE0t8JngSPbuqrg" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:ErnestoDibia/Genomics_one_c-Project">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-906df93f-b3ed-4c6e-a9d5-31f7ffe68546" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-906df93f-b3ed-4c6e-a9d5-31f7ffe68546">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="Gy1LyrpvmE3bw3/jYKwDM9UWs4CZzW+fL3LHdFujzM2l6V48xWC+NpHCHpOxSNorZKpouh7WL09mOyPxiN2l/Q=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="/UbvQlhQcHfQoJlD9hL+iK819yEPFR4WDA1r4heXGP53wUEeWHrNoJwYtqjPx6BQal0ULw2tTOorqx65eQSzEw==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="ZC3N8mTyo9jITXKRO5+YlZ595s31JlR52tIlYkamsyYJ42WuRgBQukKgAmudzU6bSiYhtngYZvMK/kfFO+PxRQ==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="ReErTGzefoTHe1EVQUjUgUh53F8MyoncVEqhRRJeyc+z/O4qkE1O+gLW/4zzpKCp/jlUdqRXBpdKFhp/2SJl3w==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/ErnestoDibia/Genomics_one_c-Project","user_id":null}}' data-hydro-click-hmac="6691f858ea01ae532b9f2df798d8a8320f66d39708e0cf8c81df5c1a7f9e1168" href="/login?return_to=https%3A%2F%2Fgithub.com%2FErnestoDibia%2FGenomics_one_c-Project">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/ErnestoDibia/Genomics_one_c-Project","user_id":null}}' data-hydro-click-hmac="6691f858ea01ae532b9f2df798d8a8320f66d39708e0cf8c81df5c1a7f9e1168" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=ErnestoDibia%2FGenomics_one_c-Project">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-7e2c5aeb-646b-4139-af95-da23bbb52a98" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-9a56d7bc-e677-4c94-b49a-2419de6de6aa" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-9a56d7bc-e677-4c94-b49a-2419de6de6aa" id="tooltip-7e2c5aeb-646b-4139-af95-da23bbb52a98" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/ErnestoDibia/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/ErnestoDibia" rel="author">
            ErnestoDibia
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/ErnestoDibia/Genomics_one_c-Project">
            Genomics_one_c-Project
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/ErnestoDibia/Genomics_one_c-Project","user_id":null}}' data-hydro-click-hmac="0b6701761b35c4ead7e7ec455a6fb703be9a6030ccd0c6106daeaaf8f0285f58" data-view-component="true" href="/login?return_to=%2FErnestoDibia%2FGenomics_one_c-Project" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":398555927,"auth_type":"LOG_IN","originating_url":"https://github.com/ErnestoDibia/Genomics_one_c-Project","user_id":null}}' data-hydro-click-hmac="9c98705450b1eb61d8b48724490c7a6ed2c12cbc2bd6cc008da17151840fb630" data-view-component="true" href="/login?return_to=%2FErnestoDibia%2FGenomics_one_c-Project" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
             0
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":398555927,"auth_type":"LOG_IN","originating_url":"https://github.com/ErnestoDibia/Genomics_one_c-Project","user_id":null}}' data-hydro-click-hmac="ba63a9b2bcfcedd919b9779b1e77bce788f31e4c664c74c0676c4726050c7e99" data-view-component="true" href="/login?return_to=%2FErnestoDibia%2FGenomics_one_c-Project" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
              0
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/ErnestoDibia/Genomics_one_c-Project/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           stars
          </a>
          <a class="Link--secondary no-underline mr-3" href="/ErnestoDibia/Genomics_one_c-Project/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/ErnestoDibia/Genomics_one_c-Project/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/ErnestoDibia/Genomics_one_c-Project/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/ErnestoDibia/Genomics_one_c-Project/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":398555927,"auth_type":"LOG_IN","originating_url":"https://github.com/ErnestoDibia/Genomics_one_c-Project","user_id":null}}' data-hydro-click-hmac="ba63a9b2bcfcedd919b9779b1e77bce788f31e4c664c74c0676c4726050c7e99" data-view-component="true" href="/login?return_to=%2FErnestoDibia%2FGenomics_one_c-Project" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/ErnestoDibia/Genomics_one_c-Project","user_id":null}}' data-hydro-click-hmac="0b6701761b35c4ead7e7ec455a6fb703be9a6030ccd0c6106daeaaf8f0285f58" data-view-component="true" href="/login?return_to=%2FErnestoDibia%2FGenomics_one_c-Project" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /ErnestoDibia/Genomics_one_c-Project" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /ErnestoDibia/Genomics_one_c-Project/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /ErnestoDibia/Genomics_one_c-Project/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /ErnestoDibia/Genomics_one_c-Project/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /ErnestoDibia/Genomics_one_c-Project/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /ErnestoDibia/Genomics_one_c-Project/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/ErnestoDibia/Genomics_one_c-Project/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /ErnestoDibia/Genomics_one_c-Project/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-a553f675-bf44-4165-b29d-3bc1a190f6c9-list" aria-haspopup="true" aria-labelledby="tooltip-db67da49-8ce7-4278-97b1-e7e7a1de2e6e" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-a553f675-bf44-4165-b29d-3bc1a190f6c9-button" popovertarget="action-menu-a553f675-bf44-4165-b29d-3bc1a190f6c9-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-a553f675-bf44-4165-b29d-3bc1a190f6c9-button" id="tooltip-db67da49-8ce7-4278-97b1-e7e7a1de2e6e" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-a553f675-bf44-4165-b29d-3bc1a190f6c9-button" anchor-offset="normal" data-view-component="true" id="action-menu-a553f675-bf44-4165-b29d-3bc1a190f6c9-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-a553f675-bf44-4165-b29d-3bc1a190f6c9-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-a553f675-bf44-4165-b29d-3bc1a190f6c9-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project" id="item-ab683010-bdd0-4fc0-8384-ca299fdc152b" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/issues" id="item-557ff13d-7d2b-434e-ad79-5844c451d54c" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/pulls" id="item-ea4d581d-f6fd-4d64-beef-dafd20c653a7" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/actions" id="item-9c4c4ccf-f633-4ded-9549-676c4a51ccf5" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/projects" id="item-9015f9ee-c817-4cdd-abde-26ca842581ac" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/security" id="item-a6019250-933d-47d0-b7e8-7e3841ab723e" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/pulse" id="item-5b5145bd-1723-48be-b101-dc139bc24465" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         ErnestoDibia/Genomics_one_c-Project
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/ErnestoDibia/Genomics_one_c-Project/spoofed_commit_check/91cec998bed5bcaf2328639ac99dd5be9702ca95">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":398555927,"defaultBranch":"main","name":"Genomics_one_c-Project","ownerLogin":"ErnestoDibia","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-08-21T12:47:16.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/88302025?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"main","listCacheKey":"v0:1629550036.763842","canEdit":false,"refType":"branch","currentOid":"91cec998bed5bcaf2328639ac99dd5be9702ca95"},"tree":{"items":[{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":1,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/ErnestoDibia/Genomics_one_c-Project/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/ErnestoDibia/Genomics_one_c-Project.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone ErnestoDibia/Genomics_one_c-Project","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/ErnestoDibia/Genomics_one_c-Project/archive/refs/heads/main.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=398555927"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"2","overviewFiles":[{"displayName":"README.md","repoName":"Genomics_one_c-Project","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGenomics_one_c-Project\u003c/h1\u003e\u003ca id=\"user-content-genomics_one_c-project\" class=\"anchor\" aria-label=\"Permalink: Genomics_one_c-Project\" href=\"#genomics_one_c-project\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e[toc]\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIdentification of somatic and germline variants from tumor and normal sample pairs\u003c/h1\u003e\u003ca id=\"user-content-identification-of-somatic-and-germline-variants-from-tumor-and-normal-sample-pairs\" class=\"anchor\" aria-label=\"Permalink: Identification of somatic and germline variants from tumor and normal sample pairs\" href=\"#identification-of-somatic-and-germline-variants-from-tumor-and-normal-sample-pairs\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIntroduction\u003c/h2\u003e\u003ca id=\"user-content-introduction\" class=\"anchor\" aria-label=\"Permalink: Introduction\" href=\"#introduction\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCancer cells have deviated from the organism's normal (germline) genome by acquiring and selecting for a set of mutations that allow them to grow rapidly and invasively, resist regulation, and possibly metastasize. These changes can range from simple single-base mutations to more complex events involving genomic gain, loss, or structural change. Modifying the function of a protein (e.g., disabling a tumor suppressor gene or activating an oncogene), silencing a gene's transcription, or affecting a gene's transcriptional affinity can then initiate the cancer process.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMany studies have chosen to sample and sequence both tumor tissue and normal genomic profiles from the same individual in order to distinguish acquired (somatic) mutations from malignant tissue. However, this solution is insufficient and as simple as the comparison being performed because there are many thousands of variants in healthy tissues compared to the reference genome. The reason for this is that each person inherits a unique pattern with many variants from their parents.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eThe primary distinction between \u003cem\u003e\u003cstrong\u003egermline variants\u003c/strong\u003e\u003c/em\u003e and \u003cem\u003e\u003cstrong\u003esomatic variants\u003c/strong\u003e\u003c/em\u003e is that germline variants are inherited and present in the carrier's germline, whereas somatic variants are acquired from the environment and thus cannot be passed down to the offspring.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eIf this is the case, healthy and tumor tissues may both have germline variants, while tumor tissues are the only ones with somatic variants. This is why comparing a normal cell to a tumor cell is the only way to tell the difference between the two types of variant.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFurthermore, in addition to acquiring somatic variants, tumor tissues may gain chromosomal copies of heterozygous germline variants, i.e. homozygous state (duplication of one allelic copy accompanied by loss of the other), or lose, i.e. hemizygous state (if one allele is simply dropped).\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNote: Only one of the two original alleles persists in the tumor.\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis is known as \u003cem\u003eloss of heterozygozity\u003c/em\u003e (LOH). Again, the detection of LOH events is based on a comparison of tumor and normal tissue data.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eGenomics-One-C's\u003c/strong\u003e tutorial is set out to:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eIdentify somatic and germline variants, as well as LOH variants, in a tumor and a normal sample from the same patient.\u003c/li\u003e\n\u003cli\u003eAnnotate the variant sites and the genes affected by them with content from general human genetic and cancer-specific databases.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eWe hope that this will provide insight into the genetic events that are driving tumor formation and growth in patients, as well as have prognostic and even therapeutic value by revealing variants known to affect drug resistance/sensitivity, tumor aggressiveness, \u003cem\u003eetc\u003c/em\u003e.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eTime estimation for this tutorial is approximately \u003cem\u003e\u003cstrong\u003e7 hours\u003c/strong\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eGalaxy Server GEMINI tool(s) issue\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe realized that the use of the \u003cem\u003eGEMINI tool\u003c/em\u003e(s) which you will encounter as you get to the \u003cstrong\u003eVariant annotation and reporting\u003c/strong\u003e section was not available on the \u003ca href=\"https://usegalaxy.org\" rel=\"nofollow\"\u003eusegalaxy.org\u003c/a\u003e server but instead on the \u003ca href=\"https://usegalaxy.eu\" rel=\"nofollow\"\u003eusegalaxy.eu\u003c/a\u003e server hence, you will have to perform you tutorial on the latter server.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cstrong\u003eAgenda\u003c/strong\u003e\u003c/h4\u003e\u003ca id=\"user-content-agenda\" class=\"anchor\" aria-label=\"Permalink: Agenda\" href=\"#agenda\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eIn this tutorial, we will cover:\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#data-preparation\"\u003eData Preparation\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#get-data\"\u003eGet data\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#quality-control-and-mapping-of-NGS-reads\"\u003eQuality control and mapping of NGS reads\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#quality-control\"\u003eQuality control\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#read-trimming-and-filtering\"\u003eRead trimming and filtering\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#read-mapping\"\u003eRead mapping\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#mapped-reads-postprocessing\"\u003eMapped reads postprocessing\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#filtering-on-mapped-reads-properties\"\u003eFiltering on mapped reads properties\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#removing-duplicate-reads\"\u003eRemoving duplicate reads\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#left-align-reads-around-indels\"\u003eLeft-align reads around indels\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#recalibrate-read-mapping-qualities\"\u003eRecalibrate read mapping qualities\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#refilter-reads-based-on-mapping-quality\"\u003eRefilter reads based on mapping quality\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#variant-calling-and-classification\"\u003eVariant calling and classification\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#variant-annotation-and-reporting\"\u003eVariant annotation and reporting\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#get-data\"\u003eGet data\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#adding-annotations-to-the-called-variants\"\u003eAdding annotations to the called variants\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#reporting-selected-subsets-of-variants\"\u003eReporting selected subsets of variants\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#generating-reports-of-genes-affected-by-variants\"\u003eGenerating reports of genes affected by variants\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#adding-additional-annotations-to-the-gene-centered-report\"\u003eAdding additional annotations to the gene-centered report\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eData Preparation\u003c/h1\u003e\u003ca id=\"user-content-data-preparation\" class=\"anchor\" aria-label=\"Permalink: Data Preparation\" href=\"#data-preparation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe must first upload and prepare some input data for analysis. The sequencing reads we'll be looking at come from real-world data from a cancer patient's tumor and normal tissue samples. However, for the sake of analysis speed, the original data has been downsampled to include only reads from human chromosomes 5, 12, and 17.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGet data\u003c/h2\u003e\u003ca id=\"user-content-get-data\" class=\"anchor\" aria-label=\"Permalink: Get data\" href=\"#get-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Data upload\u003c/h3\u003e\u003ca id=\"user-content-hands-on-data-upload\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Data upload\" href=\"#hands-on-data-upload\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eCreate a new history for this tutorial and give it a meaningful name. eg. \u003cem\u003eIdentification of somatic and germline variants from tumor and normal sample pairs (Genomics-One-C)\u003c/em\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e💡\u003cstrong\u003eTip: Creating a new history\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eClick the ➕ icon at the top of the history panel to the right of your screen.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIf the ➕ is missing:\u003c/p\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eClick on the ⚙️ icon (\u003cstrong\u003eHistory options\u003c/strong\u003e) on the top of the history panel\u003c/li\u003e\n\u003cli\u003eSelect the option \u003cstrong\u003eCreate New\u003c/strong\u003e from the menu\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e💡\u003cstrong\u003eTip: Renaming a history\u003c/strong\u003e\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eClick on \u003cstrong\u003eUnnamed history\u003c/strong\u003e (or the current name of the history)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e(\u003cstrong\u003eClick to rename history\u003c/strong\u003e) at the top of your history panel\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eType the new name\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ePress \u003cstrong\u003e\u003ccode\u003eEnter\u003c/code\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eImport the following four files from \u003ca href=\"https://zenodo.org/record/2582555\" rel=\"nofollow\"\u003eZenodo\u003c/a\u003e:\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"https://zenodo.org/record/2582555/files/SLGFSK-N_231335_r1_chr5_12_17.fastq.gz\nhttps://zenodo.org/record/2582555/files/SLGFSK-N_231335_r2_chr5_12_17.fastq.gz\nhttps://zenodo.org/record/2582555/files/SLGFSK-T_231336_r1_chr5_12_17.fastq.gz\nhttps://zenodo.org/record/2582555/files/SLGFSK-T_231336_r2_chr5_12_17.fastq.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ehttps://zenodo.org/record/2582555/files/SLGFSK-N_231335_r1_chr5_12_17.fastq.gz\nhttps://zenodo.org/record/2582555/files/SLGFSK-N_231335_r2_chr5_12_17.fastq.gz\nhttps://zenodo.org/record/2582555/files/SLGFSK-T_231336_r1_chr5_12_17.fastq.gz\nhttps://zenodo.org/record/2582555/files/SLGFSK-T_231336_r2_chr5_12_17.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003ewhere the first two files represent data from a patient's normal tissue's forward and reverse reads sequences, and the last two files represent data from the same patient's tumor tissue.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAlternatively, the same files may be available on your Galaxy server via a shared data library (ask your instructor), in which case you should import the data directly from there.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e💡\u003cstrong\u003eTip: Importing via links\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e​\tCopy the link location\u003c/li\u003e\n\u003cli\u003eOpen the Galaxy Upload Manager (on the top-left of the tool panel)\u003c/li\u003e\n\u003cli\u003eSelect \u003cem\u003e\u003cstrong\u003ePaste/Fetch Data\u003c/strong\u003e\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003ePaste the link into the text field\u003c/li\u003e\n\u003cli\u003eChange \u003cem\u003e\u003cstrong\u003eType (set all)\u003c/strong\u003e\u003c/em\u003e: from “Auto-detect” to \u003cspan\u003e\u003ccode\u003efastqsanger.gz\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003ePress \u003cem\u003e\u003cstrong\u003eStart\u003c/strong\u003e\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\u003cstrong\u003eClose\u003c/strong\u003e\u003c/em\u003e the window\u003c/li\u003e\n\u003cli\u003eBy default, Galaxy uses the URL as the name, so rename the files with a more useful name.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e💡\u003cstrong\u003eTip: Importing data from a data library\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs an alternative to uploading the data from a URL or your computer, the files may also have been made available from a shared data \u003cem\u003elibrary\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eGo into \u003cem\u003e\u003cstrong\u003eShared data\u003c/strong\u003e\u003c/em\u003e (top panel) then \u003cem\u003e\u003cstrong\u003eData libraries\u003c/strong\u003e\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eFind the correct folder (ask your instructor)\u003c/li\u003e\n\u003cli\u003eSelect the desired files\u003c/li\u003e\n\u003cli\u003eClick on the \u003cem\u003e\u003cstrong\u003eTo History\u003c/strong\u003e\u003c/em\u003e button near the top and select \u003cem\u003e\u003cstrong\u003eas Datasets\u003c/strong\u003e\u003c/em\u003e from the dropdown menu\u003c/li\u003e\n\u003cli\u003eIn the pop-up window, select the history you want to import the files to (or create a new one)\u003c/li\u003e\n\u003cli\u003eClick on \u003cem\u003e\u003cstrong\u003eImport\u003c/strong\u003e\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eCheck that all newly created datasets in your history have the correct datatypes assigned, and correct any missing or incorrect datatype assignment.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e💡\u003cstrong\u003eTip: Changing the datatype\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eClick on the \u003cem\u003e\u003cstrong\u003epencil icon\u003c/strong\u003e\u003c/em\u003e for the dataset to edit its attributes\u003c/li\u003e\n\u003cli\u003eIn the central panel, click on the galaxy-chart-select-data \u003cem\u003e\u003cstrong\u003eDatatypes\u003c/strong\u003e\u003c/em\u003e tab on the top\u003c/li\u003e\n\u003cli\u003eSelect \u003cspan\u003e\u003ccode\u003efastqsanger.gz\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003eClick the \u003cem\u003e\u003cstrong\u003eChange datatype\u003c/strong\u003e\u003c/em\u003e button\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRename the datasets and tag them appropriately.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWhen you upload a dataset via a link, Galaxy will use the link address as the dataset name, which you should probably shorten to just the file names\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e💡\u003cstrong\u003eTip: Renaming a dataset\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eClick on the galaxy-pencil \u003cem\u003e\u003cstrong\u003epencil icon\u003c/strong\u003e\u003c/em\u003e for the dataset to edit its attributes\u003c/li\u003e\n\u003cli\u003eIn the central panel, change the \u003cem\u003e\u003cstrong\u003eName\u003c/strong\u003e\u003c/em\u003e field\u003c/li\u003e\n\u003cli\u003eClick the \u003cem\u003e\u003cstrong\u003eSave\u003c/strong\u003e\u003c/em\u003e button\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eA large portion of the analysis in this tutorial will consist of identical steps performed in parallel on normal and tumor tissue data.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eGalaxy supports dataset tags to make it easier to remember which dataset represents which branch of an analysis in a linear history. In particular, if you add a tag beginning with \u003cspan\u003e\u003ccode\u003e#\u003c/code\u003e\u003c/span\u003e to any dataset, that tag will automatically propagate to any new dataset derived from the tagged dataset.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e💡\u003cstrong\u003eTip: Adding a tag\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eClick on the dataset\u003c/li\u003e\n\u003cli\u003eClick on \u003cstrong\u003eEdit dataset tags\u003c/strong\u003e\u003c/li\u003e\n\u003cli\u003eAdd a tag starting with \u003cspan\u003e\u003ccode\u003e#\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003eTags starting with \u003cspan\u003e\u003ccode\u003e#\u003c/code\u003e\u003c/span\u003e will be automatically propagated to the outputs of tools using this dataset.\u003c/li\u003e\n\u003cli\u003eCheck that the tag is appearing below the dataset name\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eBefore we begin our analysis, it is a good idea to tag the two fastq datasets representing normal tissue with, for example, \u003cspan\u003e\u003ccode\u003e#normal\u003c/code\u003e\u003c/span\u003e and the two datasets representing tumor tissue with, for example, \u003cspan\u003e\u003ccode\u003e#tumor\u003c/code\u003e\u003c/span\u003e.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eImport the reference genome with the \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e version of the sequences of human chromosomes 5, 12 and 17:\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"https://zenodo.org/record/2582555/files/hg19.chr5_12_17.fa.gz\"\u003e\u003cpre lang=\"english\" class=\"notranslate\"\u003e\u003ccode\u003ehttps://zenodo.org/record/2582555/files/hg19.chr5_12_17.fa.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eIn the import dialog, make sure to select \u003cspan\u003e\u003ccode\u003efasta\u003c/code\u003e\u003c/span\u003e  as the datatype.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eShortcut\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eYou can skip this step if the Galaxy server you're using has a hg19 version of the human reference genome with prebuilt indexes for bwa-mem and samtools (check the tools Map with BWA-MEM tool and VarScan Somatic tool to see if a hg19 version is listed as an option under \"(Using) reference genome\").\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eAlternatively, load the dataset from a shared data library.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRename the reference genome\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe reference genome you imported above came as a compressed file, but Galaxy unpacked it to plain \u003cspan\u003e\u003ccode\u003efasta\u003c/code\u003e\u003c/span\u003e  format based on your datatype selection. You might want to remove the \u003cspan\u003e\u003ccode\u003e.gz\u003c/code\u003e\u003c/span\u003e  suffix from the dataset name now.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eQuality control and mapping of NGS reads\u003c/h1\u003e\u003ca id=\"user-content-quality-control-and-mapping-of-ngs-reads\" class=\"anchor\" aria-label=\"Permalink: Quality control and mapping of NGS reads\" href=\"#quality-control-and-mapping-of-ngs-reads\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePrior to beginning the analysis, it is critical to ensure that the data used for analysis is of high quality. The quality control phase entails identifying and trimming low-quality parts as well as ditching low-quality reads that may be present in the input data to avoid spurious variant calls. In this step, we ensure that all sequencing reads used meet some basic quality criteria.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMore on quality control and mapping\u003c/h5\u003e\u003ca id=\"user-content-more-on-quality-control-and-mapping\" class=\"anchor\" aria-label=\"Permalink: More on quality control and mapping\" href=\"#more-on-quality-control-and-mapping\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIf you want to learn more about quality control and mapping, check out the \u003ca href=\"https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html\" rel=\"nofollow\"\u003eQuality Control\u003c/a\u003e and \u003ca href=\"https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial\" rel=\"nofollow\"\u003eMapping\u003c/a\u003e. We will only show the concrete steps required for quality control and read mapping of our specific datasets in this section.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eQuality control\u003c/h2\u003e\u003ca id=\"user-content-quality-control\" class=\"anchor\" aria-label=\"Permalink: Quality control\" href=\"#quality-control\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Quality Control of the input datasets\u003c/h4\u003e\u003ca id=\"user-content-hands-on-quality-control-of-the-input-datasets\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Quality Control of the input datasets\" href=\"#hands-on-quality-control-of-the-input-datasets\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eFastQC\u003c/strong\u003e on each of your four fastq datasets.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Short read data from your current history\"\u003c/em\u003e: all 4 FASTQ datasets selected with \u003cstrong\u003eMultiple datasets\u003c/strong\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eTip: Select multiple datasets\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eClick on \u003cstrong\u003eMultiple datasets\u003c/strong\u003e\u003c/li\u003e\n\u003cli\u003eSelect several files by keeping the \u003cspan\u003eCtrl\u003c/span\u003e (or \u003cspan\u003e\u003ccode\u003eCOMMAND\u003c/code\u003e\u003c/span\u003e) key pressed and clicking on the files of interest\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e8 new datasets (one with the calculated raw data, another one with an html report of the findings for each input dataset) wil get added to your history after you run this job.\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eUse \u003cstrong\u003eMultiQC\u003c/strong\u003e to aggregate the raw \u003cstrong\u003eFastQC\u003c/strong\u003e data of all four input datasets into one report\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eIn \u003cem\u003e\"Results\"\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Which tool was used to generate logs?\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eFastQC\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e\"FastQC output\"\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Type of FastQC output?\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eRaw data\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"FastQC output\"\u003c/em\u003e: all four \u003cem\u003eRawData\u003c/em\u003e outputs of \u003cstrong\u003eFastQC\u003c/strong\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eInspect the \u003cem\u003eWebpage\u003c/em\u003e output produced by the tool\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRead trimming and filtering\u003c/h2\u003e\u003ca id=\"user-content-read-trimming-and-filtering\" class=\"anchor\" aria-label=\"Permalink: Read trimming and filtering\" href=\"#read-trimming-and-filtering\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDespite the relatively good overall quality of the raw reads, we will apply read trimming and filtering to see if we can improve things further, but also to demonstrate the general concept.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cstrong\u003eHands-on: Read trimming and filtering of the normal tissue reads\u003c/strong\u003e\u003c/h4\u003e\u003ca id=\"user-content-hands-on-read-trimming-and-filtering-of-the-normal-tissue-reads\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Read trimming and filtering of the normal tissue reads\" href=\"#hands-on-read-trimming-and-filtering-of-the-normal-tissue-reads\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun the \u003cstrong\u003eTrimmomatic\u003c/strong\u003e tool to trim and filter the normal tissue reads\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Single-end or paired-end reads?”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ePaired-end (two separate input files)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis makes the tool treat the forward and reverse reads simultaneously.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Input FASTQ file (R1/first of pair)”\u003c/em\u003e: the forward reads (r1) dataset of the normal tissue sample\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Input FASTQ file (R2/second of pair)”\u003c/em\u003e: the reverse reads (r2) dataset of the normal tissue sample\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Perform initial ILLUMINACLIP step?”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eSelect standard adapter sequences or provide custom?”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eStandard\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Adapter sequences to use”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eTrueSeq3 (paired-ended, for MiSeq and HiSeq)\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Maximum mismatch count which will still allow a full match to be  performed”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e2\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“How accurate the match between the two ‘adapter ligated’ reads must  be for PE palindrome read alignment”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e30\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“How accurate the match between any adapter etc. sequence must be  against a read”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e10\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Minimum length of adapter that needs to be detected (PE specific/  palindrome mode)”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e8\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Always keep both reads (PE specific/palindrome mode)?”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThese parameters are used to cut ILLUMINA-specific adapter sequences from the reads.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eIn \u003cem\u003e“Trimmomatic Operation”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eIn \u003cem\u003e“1: Trimmomatic Operation”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Select Trimmomatic operation to perform”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eCut the specified number of bases from the start of the read (HEADCROP)\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Number of bases to remove from the start of the read”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e3\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e“Insert Trimmomatic Operation”*\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“2: Trimmomatic Operation”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Select Trimmomatic operation to perform”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eCut bases off the end of a read, if below a threshold quality (TRAILING)\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Minimum quality required to keep a base”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e10\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e“Insert Trimmomatic Operation”*\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“3: Trimmomatic Operation”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Select Trimmomatic operation to perform”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eDrop reads below a specified length (MINLEN)\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Minimum quality required to keep a base”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e25\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThese three trimming and filtering operations will be applied to the reads in the given order after adapter trimming.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eRunning this job will generate four output datasets:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003etwo for the trimmed forward and reverse reads with a proper mate in the other dataset\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003etwo more datasets of orphaned forward and reverse reads, for which the corresponding mate was dropped due to insufficient length after trimming; however, when you inspect these two files, you should find that they are empty because none of our relatively high quality reads were excessively trimmed. You can delete the two datasets to make your history smaller\u003c/em\u003e.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRead Mapping\u003c/h2\u003e\u003ca id=\"user-content-read-mapping\" class=\"anchor\" aria-label=\"Permalink: Read Mapping\" href=\"#read-mapping\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Read Mapping\u003c/h4\u003e\u003ca id=\"user-content-hands-on-read-mapping\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Read Mapping\" href=\"#hands-on-read-mapping\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eUse \u003cstrong\u003eMap with BWA-MEM\u003c/strong\u003e Tool: to map the reads from the \u003cstrong\u003enormal tissue\u003c/strong\u003e sample to the reference genome\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Will you select a reference genome from your history or use a built-in index?”\u003c/em\u003e:  \u003cspan\u003e\u003ccode\u003eUse a built-in genome\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Using reference genome”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eHuman: hg19\u003c/code\u003e\u003c/span\u003e (or a similarly named option)\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eUsing the imported \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e sequence\u003c/h3\u003e\u003ca id=\"user-content-using-the-imported-hg19-sequence\" class=\"anchor\" aria-label=\"Permalink: Using the imported hg19 sequence\" href=\"#using-the-imported-hg19-sequence\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIn the event you have imported the \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e sequence as a fasta dataset into your history, you could instead:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Will you select a reference genome from your history or use a built-in genome?\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eUse a genome from the history\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"reference genome\"\u003c/em\u003e: your imported \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e fasta dataset.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e​\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e“Single or Paired-end reads”*: \u003cspan\u003e\u003ccode\u003ePaired\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Select first set of reads”\u003c/em\u003e: the trimmed  forward reads (r1) dataset of the \u003cstrong\u003enormal tissue\u003c/strong\u003e sample; output of \u003cstrong\u003eTrimmomatic\u003c/strong\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Select second set of reads”\u003c/em\u003e: the trimmed reverse reads (r2) dataset of the \u003cstrong\u003enormal tissue\u003c/strong\u003e sample; output of \u003cstrong\u003eTrimmomatic\u003c/strong\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Set read groups information?”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eSet read groups (SAM/BAM specification)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Auto-assign”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Read group identifier (ID)”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e231335\u003c/code\u003e\u003c/span\u003e (this value being taken from the original name of the normal tissue input files)\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Auto-assign”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Read group sample name (SM)”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNormal\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTip\u003c/strong\u003e: \u003cem\u003eIn general, you are free to choose your own ID and SM values, but the ID should unambiguously identify the sequencing run that generated the reads, while the SM value should identify the biological sample.\u003c/em\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eUse \u003cstrong\u003eMap with BWA-MEM\u003c/strong\u003e to map the reads from the \u003cstrong\u003etumor tissue\u003c/strong\u003e sample,\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Will you select a reference genome from your history or use a built-in index?”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eUse a built-in genome\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Using reference genome”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eHuman: hg19\u003c/code\u003e\u003c/span\u003e (or a similarly named option)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eAdjust these settings as before if you are using the imported reference genome.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Single or Paired-end reads”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ePaired\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Select first set of reads”\u003c/em\u003e: the trimmed forward reads (r1) dataset of the \u003cstrong\u003etumor tissue\u003c/strong\u003e sample; output of \u003cstrong\u003eTrimmomatic\u003c/strong\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003eSelect second set of reads”\u003c/em\u003e: the reverse reads (r2) dataset of the \u003cstrong\u003etumor tissue\u003c/strong\u003e sample; output of \u003cstrong\u003eTrimmomatic\u003c/strong\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Set read groups information?”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eSet read groups (SAM/BAM specification)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Auto-assign”\u003c/em\u003e: \u003ccode\u003eNo\u003c/code\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003eRead group identifier (ID)”\u003c/em\u003e: \u003ccode\u003e231336\u003c/code\u003e (this value, again, being taken from the original name of the tumor tissue input files)\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Auto-assign”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Read group sample name (SM)\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eTumor\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMapped reads postprocessing\u003c/h1\u003e\u003ca id=\"user-content-mapped-reads-postprocessing\" class=\"anchor\" aria-label=\"Permalink: Mapped reads postprocessing\" href=\"#mapped-reads-postprocessing\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe will then perform some postprocessing to ensure that our variant analysis is based solely on unambiguous, high-quality read mappings.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFiltering on mapped reads properties\u003c/h2\u003e\u003ca id=\"user-content-filtering-on-mapped-reads-properties\" class=\"anchor\" aria-label=\"Permalink: Filtering on mapped reads properties\" href=\"#filtering-on-mapped-reads-properties\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo generate new filtered BAM datasets with only reads retained that have been successfully mapped to the reference, have a minimum mapping quality of 1, and have the mate read also mapped:\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Filtering for mapping status and quality\u003c/h3\u003e\u003ca id=\"user-content-hands-on-filtering-for-mapping-status-and-quality\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Filtering for mapping status and quality\" href=\"#hands-on-filtering-for-mapping-status-and-quality\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eFilter BAM datasets on a variety of attributes\u003c/strong\u003e with the following parameters:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“BAM dataset(s) to filter”\u003c/em\u003e: mapped reads datasets from the normal \u003cem\u003eand\u003c/em\u003e the tumor tissue data, outputs of \u003cstrong\u003eMap with BWA-MEM\u003c/strong\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e“Condition”\u003c/em\u003e:\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e“1: Condition”\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e“Filter”\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e“1: Filter”\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Select BAM property to filter on”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003emapQuality\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e''\u003cem\u003eFilter on read mapping quality (phred scale)”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e\u0026gt;=1\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e“2: Filter”\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Select BAM property to filter on”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eisMapped\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Selected mapped reads”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eClick on \u003cem\u003e“Insert Filter”\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e“3: Filter”\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Select BAM property to filter on”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eisMateMapped\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Select reads with mapped mate”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eWhen multiple filters are configured within a single condition, reads must pass \u003cem\u003eall\u003c/em\u003e of the filters in order to be retained in the output. As a result, the above settings retain only read pairs for which both mates are mapped.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIt should be noted that filtering for the best mapping quality is not strictly necessary. Most variant callers (including \u003cstrong\u003eVarScan somatic\u003c/strong\u003e, which we will use later) allow us to use only reads with mapping quality above a certain threshold. However, in this section, we will process the retained reads in greater depth, so it will pay off in terms of performance to eliminate reads we do not intend to use at an early stage.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Would you like to set rules?”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eThis will result in two new datasets, one for each of the normal and tumor data.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools for filtering BAM datasets\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUnder the hood, \u003cstrong\u003eFilter SAM or BAM, output SAM or BAM\u003c/strong\u003e tool uses the \u003cspan\u003e\u003ccode\u003esamtools view\u003c/code\u003e\u003c/span\u003e command line tool, while \u003cstrong\u003eFilter BAM datasets on a variety of attributes\u003c/strong\u003e tool uses \u003cspan\u003e\u003ccode\u003ebamtool filter\u003c/code\u003e\u003c/span\u003e. Whatever the BAM filtering task, you should be able to perform it in Galaxy with one of these two tools.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRemoving duplicate reads\u003c/h2\u003e\u003ca id=\"user-content-removing-duplicate-reads\" class=\"anchor\" aria-label=\"Permalink: Removing duplicate reads\" href=\"#removing-duplicate-reads\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Remove duplicates\u003c/h3\u003e\u003ca id=\"user-content-hands-on-remove-duplicates\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Remove duplicates\" href=\"#hands-on-remove-duplicates\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eRun \u003cstrong\u003eRmDup\u003c/strong\u003e Tool with the following parameters:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“BAM file”\u003c/em\u003e: filtered reads datasets from the normal \u003cem\u003eand\u003c/em\u003e the tumor tissue data; the outputs of \u003cstrong\u003eFilter SAM or BAM\u003c/strong\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Is this paired-end or single end data”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eBAM is paired-end\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Treat as single-end”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eAgain, this will produce two new datasets, one for each of the normal and tumor data.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eLeft-align reads around indels\u003c/h2\u003e\u003ca id=\"user-content-left-align-reads-around-indels\" class=\"anchor\" aria-label=\"Permalink: Left-align reads around indels\" href=\"#left-align-reads-around-indels\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Left-align\u003c/h3\u003e\u003ca id=\"user-content-hands-on-left-align\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Left-align\" href=\"#hands-on-left-align\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eRun \u003cstrong\u003eBamLeftAlign\u003c/strong\u003e Tool with the following parameters:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Choose the source for the reference genome”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eLocally cached\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“BAM dataset to re-align”\u003c/em\u003e: your filtered and deduplicated reads datasets from the normal \u003cem\u003eand\u003c/em\u003e the tumor tissue data; the outputs of \u003cstrong\u003eRmDup\u003c/strong\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Using reference genome”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eHuman: hg19\u003c/code\u003e\u003c/span\u003e (or a similarly named choice)\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Maximum number of iterations”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e5\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eUsing the imported \u003ccode\u003ehg19\u003c/code\u003e sequence\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIf you have imported the \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e sequence as a fasta dataset into your history instead:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Choose the source for the reference genome”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eHistory\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Using reference file”\u003c/em\u003e: your imported \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e fasta dataset\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eAs before, this will generate two new datasets, one for each of the normal and tumor data.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRecalibrate read mapping qualities\u003c/h2\u003e\u003ca id=\"user-content-recalibrate-read-mapping-qualities\" class=\"anchor\" aria-label=\"Permalink: Recalibrate read mapping qualities\" href=\"#recalibrate-read-mapping-qualities\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Recalibrate read quality scores\u003c/h3\u003e\u003ca id=\"user-content-hands-on-recalibrate-read-quality-scores\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Recalibrate read quality scores\" href=\"#hands-on-recalibrate-read-quality-scores\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eCalMD\u003c/strong\u003e Tool with the following parameters:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“BAM file to recalculate”\u003c/em\u003e: the left-aligned datasets from the normal and the tumor tissue data; the outputs of \u003cstrong\u003eBamLeftAlign\u003c/strong\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Choose the source for the reference genome”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eUse a built-in genome\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Using reference genome”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eHuman: hg19\u003c/code\u003e\u003c/span\u003e (or a similarly named choice)\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e**Using the imported \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e sequence    **\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIf you have imported the \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e sequence as a fasta dataset into your history instead:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Choose the source for the reference genome”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eUse a genome from the history\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Using reference file”\u003c/em\u003e: your imported \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e fasta dataset.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Do you also want BAQ (Base Alignment Quality) scores to be calculated?”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe \u003cem\u003eVarScan\u003c/em\u003e somatic tool, which we will use to call variants in the next step, is typically used in conjunction with unadjusted base quality scores because it is widely assumed that the base quality downgrades performed by \u003cem\u003eCalMD\u003c/em\u003e and other tools in the \u003cem\u003esamtools\u003c/em\u003e suite of tools are too severe for \u003cem\u003eVarScan\u003c/em\u003e to retain good sensitivity. In this tutorial, we will continue with this practice.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eUsing adjusted base quality scores\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIf you want to experiment with adjusted base quality scores for your own data, keep in mind that \u003cem\u003eVarScan somatic\u003c/em\u003e will only use them if they are incorporated directly into the read base qualities of a BAM input dataset, not if they are written to the dataset separately. Hence, should you ever decide to use:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Do you also want BAQ (Base Alignment Quality) scores to be calculated?”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes, run BAQ calculation\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eand you want this setting to affect downstream variant calling with \u003cem\u003eVarScan somatic\u003c/em\u003e, make sure you also set then:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Use BAQ to cap read base qualities”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003ePlease keep in mind that BAQ scores are quite expensive to calculate, so expect a significant (up to 10x!) increase in job run time when you enable it.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Additional options”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eAdvanced options\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Change identical bases to ‘=’“\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Coefficient to cap mapping quality of poorly mapped reads”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e50\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis last option is the main reason we're using CalMD at this point. The mapping quality of reads mapped with \u003cem\u003ebwa\u003c/em\u003e should be capped in this way before variant calling, according to empirical but well-established findings.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eThis will, once more, produce two new datasets, one for each of the normal and tumor data.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRefilter reads based on mapping quality\u003c/h2\u003e\u003ca id=\"user-content-refilter-reads-based-on-mapping-quality\" class=\"anchor\" aria-label=\"Permalink: Refilter reads based on mapping quality\" href=\"#refilter-reads-based-on-mapping-quality\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eCalMD\u003c/strong\u003e may have set some mapping quality scores to 255 during read mapping quality recalibration. This special value is reserved for \u003cem\u003eundefined\u003c/em\u003e mapping qualities and is used by the tool when a recalibrated mapping quality is less than zero. In other words, a value of 255 indicates a really bad mapping score, not a particularly good one. To remove such reads from the data, do the following:\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Eliminating reads with undefined mapping quality\u003c/h3\u003e\u003ca id=\"user-content-hands-on-eliminating-reads-with-undefined-mapping-quality\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Eliminating reads with undefined mapping quality\" href=\"#hands-on-eliminating-reads-with-undefined-mapping-quality\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eFilter BAM datasets on a variety of attributes\u003c/strong\u003e Tool: with the following parameters:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“BAM dataset(s) to filter”\u003c/em\u003e: the recalibrated datasets from the normal and the tumor tissue data; the outputs of \u003cstrong\u003eCalMD\u003c/strong\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e“Condition”\u003c/em\u003e:\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e“1: Condition”\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e“Filter”\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Select BAM property to filter on”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003emapQuality\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“Filter on read mapping quality (phred scale)”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e\u0026gt;=254\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eVariant Calling and Classification\u003c/h1\u003e\u003ca id=\"user-content-variant-calling-and-classification\" class=\"anchor\" aria-label=\"Permalink: Variant Calling and Classification\" href=\"#variant-calling-and-classification\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can now proceed to variant calling after we have generated a high-quality set of mapped read pairs. We use the \u003cstrong\u003eVarScan somatic\u003c/strong\u003e tool on the galaxy server, which is a dedicated solution for somatic variant calling.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Variant Calling and Classification\u003c/h3\u003e\u003ca id=\"user-content-hands-on-variant-calling-and-classification\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Variant Calling and Classification\" href=\"#hands-on-variant-calling-and-classification\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eVarScan somatic\u003c/strong\u003e with the following parameters:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Will you select a reference genome from your history or use a built-in genome?\"\u003c/em\u003e :  \u003cspan\u003e\u003ccode\u003eUse a built-in genome\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"reference genome\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eHuman: hg19\u003c/code\u003e\u003c/span\u003e (or a similarly named choice)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eUsing the imported \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e sequence\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn the event you have imported the \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e sequence as a fasta dataset into your history, you could instead:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Will you select a reference genome from your history or use a built-in genome?\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eUse a genome from the history\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"reference genome\"\u003c/em\u003e: your imported \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e fasta dataset.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"aligned reads from normal sample\"\u003c/em\u003e: the mapped and fully post-processed normal tissue dataset; one of the two outputs of filtering the \u003cstrong\u003eCalMD\u003c/strong\u003e outputs\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"aligned reads from tumor sample\"\u003c/em\u003e: the mapped and fully post-processed tumor tissue dataset; the other output of filtering the \u003cstrong\u003eCalMD\u003c/strong\u003e outputs.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Estimated purity (non-tumor content) of normal sample\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e1\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Estimated purity (tumor content) of tumor sample\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e0.5\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Generate separate output datasets for SNP and indel calls?\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Settings for Variant Calling\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eCustomize settings\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Minimum base quality\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e28\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe discovered that our sequencing data is of very high quality during the quality control step, and we chose not to downgrade base qualities during the quality scores recalibration step above, so that we can increase the base quality required at any given position without discarding too much of our data.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Minimum mapping quality\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e1\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe filtered our reads for those with a mapping quality of at least one during post-processing, but \u003cstrong\u003eCalMD\u003c/strong\u003e may have reduced some mapping qualities to zero afterwards.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eLeave all other settings in this section at their default values.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Settings for Posterior Variant Filtering\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eUse default values\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eVariant annotation and reporting\u003c/h1\u003e\u003ca id=\"user-content-variant-annotation-and-reporting\" class=\"anchor\" aria-label=\"Permalink: Variant annotation and reporting\" href=\"#variant-annotation-and-reporting\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis tutorial will make use of variant and gene annotations from a variety of sources. The tools we'll be using in this section will handle the majority of these for us, but we'll need to import data from four different sources into Galaxy separately:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003evariant annotations from \u003ca href=\"https://www.cancerhotspots.org/\" rel=\"nofollow\"\u003eCancer Hotspots\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003evariant and gene information from the \u003ca href=\"https://www.cancergenomeinterpreter.org/biomarkers\" rel=\"nofollow\"\u003eCancer Biomarkers database\u003c/a\u003e of the Cancer Genome Interpreter (CGI) project\u003c/li\u003e\n\u003cli\u003evariant and gene information from the \u003ca href=\"https://civicdb.org/\" rel=\"nofollow\"\u003eCIViC\u003c/a\u003e database\u003c/li\u003e\n\u003cli\u003evariant annotations from \u003ca href=\"https://www.ncbi.nlm.nih.gov/snp\" rel=\"nofollow\"\u003edbSNP\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003elists of genes annotated with the keywords \u003cem\u003eproto-oncogene\u003c/em\u003e or \u003cem\u003etumor suppressor\u003c/em\u003e at \u003ca href=\"https://www.uniprot.org/\" rel=\"nofollow\"\u003eUniProt\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eEach of these annotation sets has been released in the public domain or under a free data license, allowing you to use them not only for this tutorial but also for other purposes.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe generated a set of new data files tailored to the requirements of the workflow of this tutorial using the data downloaded from each of these sites and made them available through Zenodo, once again under a free data license.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGet data\u003c/h2\u003e\u003ca id=\"user-content-get-data-1\" class=\"anchor\" aria-label=\"Permalink: Get data\" href=\"#get-data-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Data upload\u003c/h4\u003e\u003ca id=\"user-content-hands-on-data-upload-1\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Data upload\" href=\"#hands-on-data-upload-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eImport the following \u003cstrong\u003evariant annotation files\u003c/strong\u003e from \u003ca href=\"https://zenodo.org/record/2581873\" rel=\"nofollow\"\u003eZenodo\u003c/a\u003e:\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"https://zenodo.org/record/2581873/files/hotspots.bed\nhttps://zenodo.org/record/2581873/files/cgi_variant_positions.bed\nhttps://zenodo.org/record/2581873/files/01-Feb-2019-CIVic.bed\nhttps://zenodo.org/record/2582555/files/dbsnp.b147.chr5_12_17.vcf.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ehttps://zenodo.org/record/2581873/files/hotspots.bed\nhttps://zenodo.org/record/2581873/files/cgi_variant_positions.bed\nhttps://zenodo.org/record/2581873/files/01-Feb-2019-CIVic.bed\nhttps://zenodo.org/record/2582555/files/dbsnp.b147.chr5_12_17.vcf.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eMake sure you select \u003cspan\u003e\u003ccode\u003ebed\u003c/code\u003e\u003c/span\u003e as the datatype for the first three files and \u003cspan\u003e\u003ccode\u003evcf\u003c/code\u003e\u003c/span\u003e for the last file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAlternatively, add the files from a shared data library on your Galaxy server instance.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eImport some \u003cstrong\u003egene-level annotation files\u003c/strong\u003e from \u003ca href=\"https://zenodo.org/record/2581881\" rel=\"nofollow\"\u003eZenodo\u003c/a\u003e:\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"https://zenodo.org/record/2581881/files/Uniprot_Cancer_Genes.13Feb2019.txt\nhttps://zenodo.org/record/2581881/files/cgi_genes.txt\nhttps://zenodo.org/record/2581881/files/01-Feb-2019-GeneSummaries.tsv\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ehttps://zenodo.org/record/2581881/files/Uniprot_Cancer_Genes.13Feb2019.txt\nhttps://zenodo.org/record/2581881/files/cgi_genes.txt\nhttps://zenodo.org/record/2581881/files/01-Feb-2019-GeneSummaries.tsv\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eand make sure you select \u003cspan\u003e\u003ccode\u003etabular\u003c/code\u003e\u003c/span\u003e as their datatype, or add them from the shared data library.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eDownload \u003cstrong\u003eSnpEff functional genomic annotations\u003c/strong\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eShortcut\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIf the Galaxy server you're using has \u003cspan\u003e\u003ccode\u003eHomo sapiens: hg19\u003c/code\u003e\u003c/span\u003e as a locally installed snpEff database, you can skip this step. To see if this is the case, use the \u003cstrong\u003eSnpEff eff tool\u003c/strong\u003e under Genome source.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eUse \u003cstrong\u003eSnpEff Download tool\u003c/strong\u003e to download genome annotation database \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eCheck that all new datasets' datatypes have been correctly set, and change them if necessary. You may also want to shorten the names of some of the datasets.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAdding annotations to the called variants\u003c/h2\u003e\u003ca id=\"user-content-adding-annotations-to-the-called-variants\" class=\"anchor\" aria-label=\"Permalink: Adding annotations to the called variants\" href=\"#adding-annotations-to-the-called-variants\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAdding functional genomic annotations\u003c/h3\u003e\u003ca id=\"user-content-adding-functional-genomic-annotations\" class=\"anchor\" aria-label=\"Permalink: Adding functional genomic annotations\" href=\"#adding-functional-genomic-annotations\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCertainly, not all variations are created equal. Many may be silent mutations with no effect at the amino acid level, while a few may disrupt a protein's coding sequence by introducing a premature stop codon or a frameshift. Of course, knowing which gene is affected by a variant is also important. Functional genomic annotations of this type can be added to a VCF dataset of variants with \u003cem\u003eSnpEff\u003c/em\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Adding annotations with SnpEff\u003c/h4\u003e\u003ca id=\"user-content-hands-on-adding-annotations-with-snpeff\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Adding annotations with SnpEff\" href=\"#hands-on-adding-annotations-with-snpeff\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eSnpEff eff:wrench::gear:\u003c/strong\u003e with the following parameters:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Sequence changes (SNPs, MNPs, InDels)\"\u003c/em\u003e: the output of \u003cstrong\u003eVarScan somatic:wrench:\u003c/strong\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Input format\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eVCF\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Output format\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eVCF (only if input is VCF)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Genome source\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eLocally installed reference genome\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Genome\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eHomo sapiens: hg19\u003c/code\u003e\u003c/span\u003e (or a similarly named option)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eUsing the imported \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e SnpEff genome database\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIf you have imported the \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e SnpEff genome database into your history instead:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Genome source\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eDownloaded snpEff database in your history\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"SnpEff4.3 Genome Data\"\u003c/em\u003e: your imported \u003cspan\u003e\u003ccode\u003ehg19\u003c/code\u003e\u003c/span\u003e SnpEff dataset.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Produce summary Stats\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAdding genetic and clinical evidence-base annotations\u003c/h3\u003e\u003ca id=\"user-content-adding-genetic-and-clinical-evidence-base-annotations\" class=\"anchor\" aria-label=\"Permalink: Adding genetic and clinical evidence-base annotations\" href=\"#adding-genetic-and-clinical-evidence-base-annotations\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOther interesting aspects of a variant include whether the variant has previously been observed in the human population and, if so, at what frequency. We would also like to know if a variant is known to be associated with specific diseases. To move forward with this type of genetic and clinical evidence-based annotations, we will convert our list of variants into a database that can be processed more efficiently than a VCF dataset. We will use the GEMINI tool suite for this task as well as any future work with variants.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Creating a GEMINI database from a variants dataset\u003c/h4\u003e\u003ca id=\"user-content-hands-on-creating-a-gemini-database-from-a-variants-dataset\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Creating a GEMINI database from a variants dataset\" href=\"#hands-on-creating-a-gemini-database-from-a-variants-dataset\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eGEMINI load 🔧⚙️\u003c/strong\u003e with the following parameters:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"VCF dataset to be loaded in the GEMINI database\"\u003c/em\u003e: the output of \u003cstrong\u003eSnpEff eff:wrench:\u003c/strong\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"The variants in this input are\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eannotated with snpEff\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"This input comes with genotype calls for its samples\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCalling sample genotypes was part of what we used VarScan somatic for.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Choose a gemini annotation source\"\u003c/em\u003e: select the latest available annotations snapshot (most likely, there will be only one)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Sample and family information in PED format\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNothing selected\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Load the following optional content into the database\"\u003c/em\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cul class=\"contains-task-list\"\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e \"GERP scores\"*\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cul class=\"contains-task-list\"\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e \u003cem\u003e\"CADD scores\"\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cul class=\"contains-task-list\"\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e \u003cem\u003e\"Gene tables\"\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cul class=\"contains-task-list\"\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e \u003cem\u003e\"Sample genotypes\"\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cul class=\"contains-task-list\"\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e \u003cem\u003e\"variant INFO field\"\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eBe careful to leave \u003cstrong\u003eunchecked\u003c/strong\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Genotype likelihoods (sample PLs)\"\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eVarScan somatic does not generate these values\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"only variants that passed all filters\"\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIt is simple and more flexible to filter for this property later\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eGEMINI already (silently) adds an impressive number of annotations it knows about to our variants during the database's creation (including, e.g., the frequency at which every variant has been observed in large human genome sequencing projects). All of these were obtained for free simply by converting the variants to a GEMINI database!\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eGEMINI also extracts a significant amount of information from the VCF input dataset for us (such as the functional genomic annotations that we added with SnpEff).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHowever, there are often additional annotations from other sources (not included in GEMINI) that would be useful to include. Furthermore, GEMINI is not equipped to extract some non-standard information from VCF datasets, such as some critical bits added by VarScan somatic.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe \u003cstrong\u003eGEMINI annotate tool\u003c/strong\u003e is intended to assist you with these rather common issues. It allows you to add additional annotations to variants in an already loaded GEMINI database.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs a first step, we will use the tool to add to the database some critical information generated by VarScan somatic but not recognized by GEMINI load. We are particularly interested in three values calculated by VarScan somatic for each detected variant:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eSomatic status (SS)\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis is a simple numeric code in which a value of \u003cspan\u003e\u003ccode\u003e1\u003c/code\u003e\u003c/span\u003e indicates a germline variant, a value of \u003cspan\u003e\u003ccode\u003e2\u003c/code\u003e\u003c/span\u003e indicates a somatic variant, and a value of \u003cspan\u003e\u003ccode\u003e3\u003c/code\u003e\u003c/span\u003e indicates a LOH event.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eGermline p-value (GPV)\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis is the error probability associated with variants that have a somatic status of 1.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eSomatic p-value (SPV)\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis is the error probability for status calls of 2 and 3. (somatic and LOH calls).\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThese values are encoded in the INFO column of the VarScan-generated VCF dataset, which we will extract and add to the GEMINI database.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Making variant call statistics accessible\u003c/h4\u003e\u003ca id=\"user-content-hands-on-making-variant-call-statistics-accessible\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Making variant call statistics accessible\" href=\"#hands-on-making-variant-call-statistics-accessible\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eGEMINI annotate:wrench::gear:\u003c/strong\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"GEMINI database\"\u003c/em\u003e: output of \u003cstrong\u003eGEMINI load\u003c/strong\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Dataset to use as the annotation source\"\u003c/em\u003e: output of \u003cstrong\u003eVarScan somatic\u003c/strong\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Strict variant-identity matching of database and annotation records (VCF format only)\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis setting is unimportant in this case because you built the GEMINI database from the same set of variants from which we are now retrieving annotations and because VarScan somatic does not call multiple alleles at a single site. However, matching on variant-identity is the desired behavior, so we might as well be explicit about it.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Type of information to add to the database variants\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eSpecific values extracted from matching records in the annotation source (extract)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e\"1: Annotation extraction recipe\"\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Elements to extract from the annotation source\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eSS\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"Database column name to use for recording annotations\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003esomatic_status\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"What type of data are you trying to extract?\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eInteger numbers\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"If multiple annotations are found for the same variant, store...\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ethe first annotation found\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThis is the recipe for extracting the VarScan-generated \u003cem\u003eSS\u003c/em\u003e field and adding it to the GEMINI database as a new column \u003cem\u003esomatic status\u003c/em\u003e.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e➕\u003cem\u003e\"Insert Annotation extraction recipe\"\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e\"2: Annotation extraction recipe\"\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Elements to extract from the annotation source\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eGPV\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"Database column name to use for recording annotations\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003egermline_p\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"What type of data are you trying to extract?\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNumbers with decimal precision\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e*\"If multiple annotations are found for the same variant, store...\": \u003cspan\u003e\u003ccode\u003ethe first annotation found\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThis is the recipe for extracting the VarScan-generated \u003cem\u003eGPV\u003c/em\u003e field and adding it to the GEMINI database as a new column \u003cem\u003egermline_p\u003c/em\u003e.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e➕\u003cem\u003e\"Insert Annotation extraction recipe\"\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e\"3: Annotation extraction recipe\"\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Elements to extract from the annotation source\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eSPV\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"Database column name to use for recording annotations\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003esomatic_p\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"What type pf data are you trying to extract\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNumbers with decimal precision\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"If multiple annotations are found for the same variant, store...\"\u003c/em\u003e:\u003cspan\u003e\u003ccode\u003ethe first annotation found\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThis is the recipe for extracting the VarScan-generated \u003cem\u003eSPV\u003c/em\u003e field and adding it to the GEMINI database as a new column \u003cem\u003esomatic_p\u003c/em\u003e.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eThen, in addition to the annotations obtained directly from GEMINI or the input VCF dataset, we will add additional annotations. We would like to specifically add:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003emore information from \u003cstrong\u003edbSNP\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eGEMINI already checks all variants to see if they exist in dbSNP and, if so, stores their dbSNP IDs as part of the database creation process. As a result, we only need to extract some additional relevant information.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003einformation from \u003cstrong\u003eCancer Hotspots\u003c/strong\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003elinks to the \u003cstrong\u003eCiViC\u003c/strong\u003e database\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003einformation from the \u003cstrong\u003eCancer Genome Interpreter (CGI)\u003c/strong\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Adding further annotations\u003c/h4\u003e\u003ca id=\"user-content-hands-on-adding-further-annotations\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Adding further annotations\" href=\"#hands-on-adding-further-annotations\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eGEMINI annotate:wrench::gear:\u003c/strong\u003e to add further annotations from \u003cstrong\u003edbSNP\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"GEMINI database\"\u003c/em\u003e: the output of the last \u003cstrong\u003eGEMINI annotate:wrench:\u003c/strong\u003e run\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Dataset to use as the annotation source\"\u003c/em\u003e: the imported \u003cspan\u003e\u003ccode\u003edbsnp.b147.chr5_12_17.vcf\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Strict variant-identity matching of database and annotation records (VCF format only)\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edbSNP stores information about specific SNPs found in human populations, and we'd like to see if any of the same SNPs are found among our variants.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Type of information to add to the database variants\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eSpecific values extracted from matching recors in the annotation source (extract)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e\"1: Annotation extraction recipe\"\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Elements to extract from the annotation source\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eSAO\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"Databse column name to use for recording annotations\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ers_ss\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"What type of data are you trying to extract?\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eInteger numbers\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"If multiple annotations are found for the same variant, store...\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ethe first annotation found\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThis recipe extracts the dbSNP field and adds it as \u003cem\u003ers_ss\u003c/em\u003e to the GEMINI database.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eGEMINI annotate:wrench::gear:\u003c/strong\u003e to add further annotations from \u003cstrong\u003ecancerhotspots\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"GEMINI database\"\u003c/em\u003e: the output of the last \u003cstrong\u003eGEMINI annotate:wrench:\u003c/strong\u003e run\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Dataset to use as the annotation source\"\u003c/em\u003e: the imported \u003cspan\u003e\u003ccode\u003ecancerhotspots_v2.bed\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Strict variant-identity matching of database and annotation records (VCF format only)\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e (with input in BED format this settign will be ignored)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe will simply record the best q-value associated with any cancerhotspots variant that overlaps one of our variant sites for the cancerhotspots data.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Type of information to add to the database variants\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eSpecific values extracted from matching records in the annotation source (extract)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e\"1: Annotation extraction recipe\"\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Elements to extract from the annotation source\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e5\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe q-values are stored in the fifth column of the BED dataset.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Database column name to use for recording annotations\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ehs_qvalue\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"What type of data are you trying to extract?\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNumbers with decimal precision\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"If multiple annotations are found for the same variant, store...\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ethe smallest of the (numeric) values\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThis is the procedure for extracting the q-values of overlapping cancerhotspots sites and adding them to the GEMINI database as hs qvalue.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eGEMINI annotate:wrench::gear:\u003c/strong\u003e to add links to \u003cstrong\u003eCiViC\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"GEMINI database\"\u003c/em\u003e: the output of the last \u003cstrong\u003eGEMINI annotate:wrench:\u003c/strong\u003e run\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e*\"Dataset to use as the annotation source\"**: the imported \u003cspan\u003e\u003ccode\u003eCiViC.bed\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Strict variant-identity matching of database and annotation records (VCF format only)\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e (with input in BED format this setting will be ignored)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe will record the hyperlink to any variant found in the CIViC database that overlaps one of our variant sites for the CIViC data.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Type of information to add to the database variants\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eSpecific values extracted from matching records in the annotation source (extract)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e\"1: Annotation extraction recipe\"\u003c/em\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Elements to extract from the annotation source\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e4\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe hyperlinks are stored in the fourth column of the BED dataset\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Database column name to use for recording annotations\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eoverlapping_civic_url\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"What type of data are you trying to extract?\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eText (text)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"If multiple annotations are found for the same variant, store...\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ea comma-separated list of non-redundant (text) values\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis is the recipe for extracting the hyperlinks of overlapping CIViC sites and adding them to the GEMINI database as a list of overlapping civic urls.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eGEMINI annotate:wrench::gear:\u003c/strong\u003e to add information from the \u003cstrong\u003eCancer Genome Interpreter (CGI)\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"GEMINI database\"\u003c/em\u003e: the output of the last \u003cstrong\u003eGEMINI annotate:wrench:\u003c/strong\u003e run\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Dataset to use as the annotation source\"\u003c/em\u003e: the imported \u003cspan\u003e\u003ccode\u003ecgi_variant)positions.bed\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Strict variant-identity matching of database and annotation records (VCF format only)\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e (with input in BED format this setting will be ignored)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor the CGI data, we will note whether any of our variant sites overlap with a variant in the CGI biomarkers database.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Type of information to add to the database variants\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eBinary indicator (1=found, 0=not found) of whether the variant had any match in the annotation source (boolean)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Databse column name to use for ecording annotations\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ein_cgidb\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eReporting selected subsets of variants\u003c/h2\u003e\u003ca id=\"user-content-reporting-selected-subsets-of-variants\" class=\"anchor\" aria-label=\"Permalink: Reporting selected subsets of variants\" href=\"#reporting-selected-subsets-of-variants\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow that we've built our GEMINI database and added annotations to it, it's time to explore the wealth of information it contains. The goal is to generate filtered variant reports that include specific classes (somatic, germline, and LOH) of high-quality variants as well as their most relevant annotations. This is accomplished through GEMINI queries that specify:\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003efilters we want to apply to the variant list stored in the database\u003c/li\u003e\n\u003cli\u003ethe pieces of information about the filtered variants that we would like to retrieve\u003c/li\u003e\n\u003c/ol\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eThe GEMINI query language\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eGEMINI queries are extremely flexible, allowing users to express a wide range of ideas in order to explore the variant space; however, the complete query syntax can be intimidating for beginners.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn this section, we'll start with a simple query and work our way up to a more complex query in the following section.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eConsult the \u003ca href=\"https://gemini.readthedocs.io/en/latest/content/querying.html\" rel=\"nofollow\"\u003equery section of the GEMINI documentation\u003c/a\u003e for a more detailed explanation of the query syntax. Because the GEMINI query syntax is based on the SQLite SQL dialect, the SQLite documentation, particularly its chapter on \u003ca href=\"https://sqlite.org/lang_corefunc.html\" rel=\"nofollow\"\u003eSQLite core functions\u003c/a\u003e, is an excellent resource for understanding more complex queries.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eLet's begin by configuring a simple query to generate a list of genuine somatic variants. Our plan for retrieving them is as follows:\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003erely on the \u003cem\u003esomatic status\u003c/em\u003e of the variants called by \u003cstrong\u003eVarScan somatic:wrench:\u003c/strong\u003e\u003c/li\u003e\n\u003cli\u003eDiscard any variants for which a significant number of supporting sequencing reads are also found in normal tissue data, or which are only supported by a very small fraction of the reads from the tumor sample.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Querying the GEMINI database for somatic variants\u003c/h4\u003e\u003ca id=\"user-content-hands-on-querying-the-gemini-database-for-somatic-variants\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Querying the GEMINI database for somatic variants\" href=\"#hands-on-querying-the-gemini-database-for-somatic-variants\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eGEMINI query:wrench::gear:\u003c/strong\u003e with:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"GEMINI database\"\u003c/em\u003e: the fully annotated database created in the last \u003cstrong\u003eGEMINI annotate:wrench:\u003c/strong\u003e step\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Build GEMINI query using\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eBasic variant query constructor\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e➕\u003cem\u003e\"Insert Genotype filter expression\"\u003c/em\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Restrictions to apply to gentype values\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003egt_alt_freqs.NORMAL \u0026lt;= 0.05 AND gt_alt_freqs.TUMOR \u0026gt;= 0.10\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eWe retain only those variants supported by less than 5% of the reads in the normal sample but more than 10% of the reads in the tumor sample using this genotype-based filter.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Additional constraints expressed in SQL syntax\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003esomatic_status = 2\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eVarScan somatic has called for each variant's somatic status, which is stored in the GEMINI database (remember we used GEMINI annotate to add it). With the condition \u003cspan\u003e\u003ccode\u003esomatic_status = 2\u003c/code\u003e\u003c/span\u003e, we keep only the variants that passed the genotype filter above \u003cstrong\u003eand\u003c/strong\u003e were classified as somatic by the variant caller.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cem\u003e\"Output format options\"\u003c/em\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Type of report to generate\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003etabular (GEMINI default)\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Add a header of column names to the output\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"Set of columns to include in the variant report table\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eCustom (report user-specified columns)\u003c/code\u003e\u003c/span\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eIn \u003cem\u003e\"Choose columns to include in the report\"\u003c/em\u003e:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cul class=\"contains-task-list\"\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e \u003cem\u003e\"chrom\"\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cul class=\"contains-task-list\"\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e \u003cem\u003e\"start\"\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cul class=\"contains-task-list\"\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e \u003cem\u003e\"ref\"\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cul class=\"contains-task-list\"\u003e\n\u003cli class=\"task-list-item\"\u003e\u003cinput type=\"checkbox\" id=\"\" disabled=\"\" class=\"task-list-item-checkbox\" checked=\"\"\u003e \u003cem\u003e\"alt\"\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e\"Additional columns (comma-separated)\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003egene, aa_change, rs_ids, hs_qvalue, cosmic_ids\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eIn this section, we specify which columns (from the GEMINI database's variants table) we want to include in a \u003cspan\u003e\u003ccode\u003etabular\u003c/code\u003e\u003c/span\u003e variant report, in the order specified.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eHow am I supposed to know these column names?\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eObviously, you must know the names of the columns (in the GEMINI database tables) in order to include them in the report, but how are you supposed to know them?\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe \u003ca href=\"https://gemini.readthedocs.io/en/latest/content/database_schema.html\" rel=\"nofollow\"\u003eGEMINI documentation\u003c/a\u003e lists the standard ones (added by GEMINI load when building the database). The non-standard columns (the ones you added with GEMINI annotate) are labeled with the names you assigned to them when you added them.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAlternatively, to obtain the tables and column names of a specific database, use the \u003cstrong\u003eGEMINI database info tool\u003c/strong\u003e as follows:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“GEMINI database”\u003c/em\u003e: the database you want to explore\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Information to retrieve from the database”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNames of database tables and their columns\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eWhat about more sophisticated filtering?\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: More complex filter criteria\u003c/h4\u003e\u003ca id=\"user-content-hands-on-more-complex-filter-criteria\" class=\"anchor\" aria-label=\"Permalink: Hands-on: More complex filter criteria\" href=\"#hands-on-more-complex-filter-criteria\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eGEMINI query:wrench::gear:\u003c/strong\u003e with the exact same settings as before, but:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Additional constraints expressed in SQL syntax\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003esomatic_status = 2 AND somatic_p \u0026lt;= 0.05 AND filter IS NULL\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThis translates to \"variants classified as somatic with a p-value \u0026lt;= 0.05, but not flagged as likely false-positives\".\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSQL keywords\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSQL keywords are given in uppercase in the condition above. This is not required, but it makes the syntax easier to understand.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cspan\u003e\u003ccode\u003eIS NULL\u003c/code\u003e\u003c/span\u003e can be used to determine whether a cell in a data table is empty or contains a value. \u003cspan\u003e\u003ccode\u003eIS NOT NULL\u003c/code\u003e\u003c/span\u003e can be used to determine whether a cell in a data table contains a value. You can use \u003cspan\u003e\u003ccode\u003eAND\u003c/code\u003e\u003c/span\u003e and \u003cspan\u003e\u003ccode\u003eOR\u003c/code\u003e\u003c/span\u003e, as well as parentheses to group conditions, to logically combine different filter criteria.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eIf you followed all of the steps exactly, running this job should result in a tabular dataset of 43 variants, and with the annotations in the report, it should be relatively easy to pick out a few interesting ones. However, before we get into the report's content, we could improve the report's format a little more.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: SQL-based output formatting\u003c/h4\u003e\u003ca id=\"user-content-hands-on-sql-based-output-formatting\" class=\"anchor\" aria-label=\"Permalink: Hands-on: SQL-based output formatting\" href=\"#hands-on-sql-based-output-formatting\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eRun \u003cstrong\u003eGEMINI query:wrench::gear:\u003c/strong\u003e with the exact same settings as in the last example, but:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eIn \u003cem\u003e\"Output format options\"\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e\"Additional columns (comma-separated)\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003etype, gt_alt_freqs.TUMOR, gt_alt_freqs.NORMAL, ifnull(nullif(round(max_aaf_all,2),-1.0),0) AS MAD, gene, impact_so, aa_change, ifnull(round(cadd_scaled,2), '.') AS cadd_scaled, round(gerp_bp_score,2) AS gerp_bp, ifnull(round(gerp_element_pval,2),'.') AS gerp_elemt_pval, ifnull(round(hs_qvalue,2),'.') AS hs_qvalue, in_omim, ifnull(clinvar_sig,'.') AS clinvar, ifnull(clinvar_disease_name,'.') AS clinvar_disease_name, ifnull(rs_ids,'.') AS dbsnp_ids, rs_ss, ifnull(cosmic_ids,'.') AS cosmic_ids, ifnull(overlapping_civic_url,'.') AS overlapping_civic_url, in_cgidb\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eThis final query adds a lot more annotations to the report and also shows how to use the \u003cspan\u003e\u003ccode\u003eAS\u003c/code\u003e\u003c/span\u003e keyword to rename columns and some \u003ca href=\"https://sqlite.org/lang_corefunc.html\" rel=\"nofollow\"\u003eSQLite functions\u003c/a\u003e to clean up the output.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eLimitations of genotype column queries\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIf you're wondering why the above query doesn't use rounding on the samples' alternate allele frequencies, i.e., \u003cspan\u003e\u003ccode\u003egt_alt_freqs.TUMOR\u003c/code\u003e\u003c/span\u003e and \u003cspan\u003e\u003ccode\u003egt_alt_freqs.NORMAL\u003c/code\u003e\u003c/span\u003e, or why it doesn't rename these columns, it's because it would break the query.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs a general rule, all columns in the variants table beginning with \u003cspan\u003e\u003ccode\u003egt\u003c/code\u003e\u003c/span\u003e (the genotype columns calculated from the genotype fields of a VCF dataset) cannot be used as regular SQLite columns and must be parsed separately by GEMINI. As a result, you cannot mix them with regular SQLite elements such as functions and alias specifications when using \u003cspan\u003e\u003ccode\u003eAS\u003c/code\u003e\u003c/span\u003e. You may have also noticed that \u003cspan\u003e\u003ccode\u003egt_alt_freqs.TUMOR\u003c/code\u003e\u003c/span\u003e and \u003cspan\u003e\u003ccode\u003egt_alt_freqs.NORMAL\u003c/code\u003e\u003c/span\u003e columns do not follow the query's column order specification and end up as the last columns of the tabular report. This is yet another example of GEMINI's unique treatment of \u003cspan\u003e\u003ccode\u003egt\u003c/code\u003e\u003c/span\u003e columns.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGenerating reports of genes affected by variants\u003c/h2\u003e\u003ca id=\"user-content-generating-reports-of-genes-affected-by-variants\" class=\"anchor\" aria-label=\"Permalink: Generating reports of genes affected by variants\" href=\"#generating-reports-of-genes-affected-by-variants\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFinally, using the same somatic variants we picked earlier, let's try to construct a gene-centered report.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAnnotations that pertain to a full gene affected by a variant rather than the variant itself would be included in a gene-centered report. Known synonyms of an afflicted gene, its NCBI entrez number, the ClinVar phenotype linked with the gene, if any, a reference to the gene's page at CIViC.org, and so on are examples of such annotations.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSome of this data is included in every GEMINI database, but it's stored in a distinct table called \u003cspan\u003e\u003ccode\u003egene_detailed\u003c/code\u003e\u003c/span\u003e, whereas we've only utilized and queried data from the \u003cspan\u003e\u003ccode\u003evariants\u003c/code\u003e\u003c/span\u003e table thus far.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe must join the \u003cspan\u003e\u003ccode\u003evariants\u003c/code\u003e\u003c/span\u003e and \u003cspan\u003e\u003ccode\u003egene_detailed\u003c/code\u003e\u003c/span\u003e tables in order to retrieve information from both tables in the same query. Such an operation is not possible with the \u003cspan\u003e\u003ccode\u003eBasic query constructor\u003c/code\u003e\u003c/span\u003e we've been using so far, and needs building the query in advanced mode.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Turning query results into gene-centered reports\u003c/h4\u003e\u003ca id=\"user-content-hands-on-turning-query-results-into-gene-centered-reports\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Turning query results into gene-centered reports\" href=\"#hands-on-turning-query-results-into-gene-centered-reports\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun \u003cstrong\u003eGEMINI query:wrench::gear:\u003c/strong\u003e in advanced mode by choosing\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Build GEMINI query using\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eAdvanced query constructor\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"The query to be issued to the database\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eSELECT v.gene, v.chrom, g.synonym, g.hgnc_id, g.entrez_id, g.rvis_pct, v.clinvar_gene_phenotype FROM variants v, gene_detailed g WHERE v.chrom = g.chrom AND v.gene = g.gene AND v.somatic_status = 2 AND v.somatic_p \u0026lt;= 0.05 AND v.filter IS NULL GROUP BY g.gene\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eElements of the SQL query\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe part between \u003cspan\u003e\u003ccode\u003eSELECT\u003c/code\u003e\u003c/span\u003e and \u003cspan\u003e\u003ccode\u003eFROM\u003c/code\u003e\u003c/span\u003e specifies which columns from which database tables we want to retrieve, while the part between \u003cspan\u003e\u003ccode\u003eFROM\u003c/code\u003e\u003c/span\u003e and \u003cspan\u003e\u003ccode\u003eWHERE\u003c/code\u003e\u003c/span\u003e specifies the database tables that need to be consulted and gives them simpler aliases (\u003cspan\u003e\u003ccode\u003ev\u003c/code\u003e\u003c/span\u003e becomes an alias for the \u003cspan\u003e\u003ccode\u003evariants\u003c/code\u003e\u003c/span\u003e table, \u003cspan\u003e\u003ccode\u003eg\u003c/code\u003e\u003c/span\u003e for the \u003cspan\u003e\u003ccode\u003egene_detailed\u003c/code\u003e\u003c/span\u003e table), which we can then use throughout the query.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe filter criteria we want to use are listed after \u003cspan\u003e\u003ccode\u003eWHERE\u003c/code\u003e\u003c/span\u003e. These criteria are nearly identical to those used in our previous somatic variants query, but because we are working with two tables instead of just one, we must specify which table the filter columns come from using table prefixes. Hence, \u003cspan\u003e\u003ccode\u003esomatic_status\u003c/code\u003e\u003c/span\u003e becomes \u003cspan\u003e\u003ccode\u003ev.somatic_status\u003c/code\u003e\u003c/span\u003e, and so on. Furthermore, we want to report corresponding information from the two tables, which is ensured by the additional criteria \u003cspan\u003e\u003ccode\u003ev.chrom = g.chrom\u003c/code\u003e\u003c/span\u003e and \u003cspan\u003e\u003ccode\u003ev.gene = g.gene\u003c/code\u003e\u003c/span\u003e (the SQL terminology is: we want to join the \u003cspan\u003e\u003ccode\u003evariants\u003c/code\u003e\u003c/span\u003e and \u003cspan\u003e\u003ccode\u003egene_detailed\u003c/code\u003e\u003c/span\u003e tables on their \u003cspan\u003e\u003ccode\u003echrom\u003c/code\u003e\u003c/span\u003e and \u003cspan\u003e\u003ccode\u003egene\u003c/code\u003e\u003c/span\u003e columns).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFinally, the \u003cspan\u003e\u003ccode\u003eGROUP BY\u003c/code\u003e\u003c/span\u003e clause specifies that we want to combine records affecting the same gene into one.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\"Genotype filter expression\"\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003egt_alt_freqs.NORMAL \u0026lt;= 0.05 AND gt_alt_freqs.TUMOR \u0026gt;= 0.10\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis remains unchanged from the previous somatic variants query.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAdding additional annotations to the gene-centered report\u003c/h2\u003e\u003ca id=\"user-content-adding-additional-annotations-to-the-gene-centered-report\" class=\"anchor\" aria-label=\"Permalink: Adding additional annotations to the gene-centered report\" href=\"#adding-additional-annotations-to-the-gene-centered-report\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUnfortunately, GEMINI annotate only allows you to add columns to a GEMINI database's variants table; there is no easy way to enrich the \u003cspan\u003e\u003ccode\u003egene_detailed\u003c/code\u003e\u003c/span\u003e table with additional annotations. That is why, using more general-purpose Galaxy tools, we will now add such additional annotations to the tabular gene-centered report.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAnnotating the GEMINI tabular gene report is a two-step process in which we first combine the report and tabular annotation sources into a larger tabular dataset, from which we then remove redundant and unwanted columns while \u003cem\u003erearranging\u003c/em\u003e the remaining ones.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eStep 1\u003c/strong\u003e consists of three separate \u003cem\u003ejoin\u003c/em\u003e operations that pull in the annotations found in the three gene-based tabular datasets that you imported in the \u003cem\u003eGet Data\u003c/em\u003e step of this section in a sequential fashion. To to this, we used the \u003cstrong\u003eJoin two file\u003c/strong\u003e tool to add the information such as \u003cstrong\u003eUniProt cancer genes\u003c/strong\u003e, \u003cstrong\u003eCGI biomarkers\u003c/strong\u003e and \u003cstrong\u003eCIViC\u003c/strong\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Join\u003c/h4\u003e\u003ca id=\"user-content-hands-on-join\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Join\" href=\"#hands-on-join\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eUse \u003cstrong\u003eJoin two files:wrench::gear:\u003c/strong\u003e to add \u003cstrong\u003eUniProt cancer genes\u003c/strong\u003e information\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“1st file”\u003c/em\u003e: the GEMINI-generated gene report from the previous step\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Column to use from 1st file”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eColumn: 1\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“2nd file”\u003c/em\u003e: the imported \u003cspan\u003e\u003ccode\u003eUniprot_cancer_Genes\u003c/code\u003e\u003c/span\u003e dataset\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Column to use from 2nd file”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eColumn: 1\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Output lines appearing in”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eBoth 1st \u0026amp; 2nd file, plus unpairable lines from 1st file. (-a 1)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe obviously want to include a variant-affected gene in the final report even if it is not listed as a Uniprot cancer gene.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“First line is a header line”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Ignore case”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Value to put in unpaired (empty) fields”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e0\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIf you look at the \u003cspan\u003e\u003ccode\u003eUniprot_Cancer_Genes\u003c/code\u003e\u003c/span\u003e dataset, you'll notice that it has two annotation columns: one that indicates whether a given gene is a proto-oncogene or not using \u003cspan\u003e\u003ccode\u003e1\u003c/code\u003e\u003c/span\u003e and \u003cspan\u003e\u003ccode\u003e0\u003c/code\u003e\u003c/span\u003e, and another that indicates tumor suppressor genes in the same way. To indicate the common case that a gene affected by a variant is neither a known proto-oncogene nor a tumor suppressor gene, we want to fill the corresponding two columns of the join result with \u003cspan\u003e\u003ccode\u003e0\u003c/code\u003e\u003c/span\u003e.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eUse \u003cstrong\u003eJoin two files:wrench:\u003c/strong\u003e⚙️ to add \u003cstrong\u003eCGI biomarkers\u003c/strong\u003e information\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“1st file”\u003c/em\u003e: the partially annotated dataset from the previous\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Column to use from 1st file”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eColumn: 1\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“2nd file”\u003c/em\u003e: the imported \u003cspan\u003e\u003ccode\u003ecgi_genes\u003c/code\u003e\u003c/span\u003e dataset\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Column to use from 2nd file”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eColumn: 1\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Output lines appearing in”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eBoth 1st \u0026amp; 2nd file, plus unpairable lines from 1st file. (-a 1)\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“First line is a header line”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Ignore case”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Value to put in unpaired (empty) fields”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e0\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eInspect the input and the result dataset to make sure you understand what happened at this step.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eUse \u003cstrong\u003eJoin two files:wrench:\u003c/strong\u003e⚙️ to add gene information from \u003cstrong\u003eCIViC\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“1st file”\u003c/em\u003e: the partially annotated dataset from step 2\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Column to use from 1st file”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eColumn: 1\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“2nd file”\u003c/em\u003e: the imported \u003cspan\u003e\u003ccode\u003eGeneSummaries\u003c/code\u003e\u003c/span\u003e dataset\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Column to use from 2nd file”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eColumn: 3\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe gene column in the CIViC gene summaries annotation dataset is \u003cem\u003enot\u003c/em\u003e the first one!\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Output lines appearing in”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eBoth 1st \u0026amp; 2nd file, plus unpairable lines from 1st file. (-a 1)\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“First line is a header line”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Ignore case”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eNo\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e“Value to put in unpaired (empty) fields”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003e.\u003c/code\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eInspect the input and the result dataset to make sure you understand what happened at this step.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eLooking at the output datasets, the gene columns from each of the \u003cem\u003ejoin\u003c/em\u003e operations was preserved from both input datasets. Furthermore, we had no control over the order in which columns were added to the report, and we couldn't exclude any columns.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn \u003cstrong\u003eStep 2\u003c/strong\u003e of the report preparation process, we will address all of these issues and rearrange the data to produce a fully annotated gene report. The tool used is \u003cstrong\u003eColumn arrange by header name\u003c/strong\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHands-on: Rearrange to get a fully annotated gene report\u003c/h4\u003e\u003ca id=\"user-content-hands-on-rearrange-to-get-a-fully-annotated-gene-report\" class=\"anchor\" aria-label=\"Permalink: Hands-on: Rearrange to get a fully annotated gene report\" href=\"#hands-on-rearrange-to-get-a-fully-annotated-gene-report\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eRun \u003cstrong\u003eColumn arrange by header name:wrench:\u003c/strong\u003e⚙️ configured like this:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“file to rearrange”\u003c/em\u003e: the final \u003cstrong\u003eJoin\u003c/strong\u003e result dataset from step 3\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eIn \u003cem\u003e“1: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003egene\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e➕\u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“2: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003echrom\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e➕\u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“3: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003esynonym\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e➕\u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“4: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ehgnc_id\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e➕\u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“5: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eentrez_id\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e➕\u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“6: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ervis_pct\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e➕\u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“7: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eis_OG\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e➕\u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“8: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eis_TS\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e➕\u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“9: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003ein_cgi_biomarkers\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e➕\u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“10: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eclinvar_gene_phenotype\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e➕\u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“11: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003egene_civic_url\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e➕\u003cem\u003e“Specify the first few columns by name”\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eIn \u003cem\u003e“12: Specify the first few columns by name”\u003c/em\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003e“column”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003edescription\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003cem\u003e“Discard unspecified columns”\u003c/em\u003e: \u003cspan\u003e\u003ccode\u003eYes\u003c/code\u003e\u003c/span\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAlternative tool suggestion\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIf your Galaxy server doesn't have the Column arrange tool, it almost certainly has the \u003cstrong\u003eCut columns from a table tool\u003c/strong\u003e, which can be used as a stand-in. This tool, however, expects a comma-separated list of column indexes, such as \u003cspan\u003e\u003ccode\u003ec1,c2\u003c/code\u003e\u003c/span\u003e for the first and second columns, so you must first determine the column numbers of your columns of interest.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eKeep in mind that the suggested tool should not be confused with \u003cem\u003eCut columns from a table (cut)\u003c/em\u003e, which does not allow you to change the order of columns!\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConclusion\u003c/h1\u003e\u003ca id=\"user-content-conclusion\" class=\"anchor\" aria-label=\"Permalink: Conclusion\" href=\"#conclusion\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIn addition to simply calling variants, \u003cem\u003esomatic variant calling\u003c/em\u003e attempts to distinguish \u003cem\u003esomatic mutations\u003c/em\u003e, which are unique to tumor tissue, from \u003cem\u003egermline\u003c/em\u003e mutations, which are shared by tumor and healthy tissue, and \u003cem\u003eloss-of-heterozygosity\u003c/em\u003e events, which involve the loss of one of two alleles found at a biallelic site in healthy tissue, from tumor tissue.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eDedicated somatic variant callers can perform this classification based on statistics, but the interpretation of any list of variants (somatic, germline, or LOH) is also critically dependent on rich genetic and cancer-specific variant and gene annotations.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eSome \u003cstrong\u003eKey points\u003c/strong\u003e to consider:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eTo reduce false-positive variant calls, use best practices for read mapping, quality control, and mapped reads post-processing.\u003c/li\u003e\n\u003cli\u003eCall variants and classify them statistically into somatic, germline, and LOH event variants using a dedicated somatic variant caller.\u003c/li\u003e\n\u003cli\u003eAnnotations and queries based on variant properties improve the usefulness of variant and gene reports.\u003c/li\u003e\n\u003cli\u003e\u003cstrong\u003eGEMINI\u003c/strong\u003e is a very useful framework for managing, annotating, and querying lists of variants in a flexible manner.\u003c/li\u003e\n\u003cli\u003eTo encourage reproducibility and information sharing, prefer public, free annotation sources.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eNames of contributors:\u003c/h3\u003e\u003ca id=\"user-content-names-of-contributors\" class=\"anchor\" aria-label=\"Permalink: Names of contributors:\" href=\"#names-of-contributors\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGalaxy workflow:\u003c/h4\u003e\u003ca id=\"user-content-galaxy-workflow\" class=\"anchor\" aria-label=\"Permalink: Galaxy workflow:\" href=\"#galaxy-workflow\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/ernesto\"\u003e@ernesto\u003c/a\u003e (TeamLead\n@NNyakundiM\n\u003ca href=\"https://github.com/aneela\"\u003e@aneela\u003c/a\u003e\n\u003ca href=\"https://github.com/clementmlay\"\u003e@clementmlay\u003c/a\u003e\n\u003ca href=\"https://github.com/Brenda\"\u003e@brenda\u003c/a\u003e\n@Njuguna.E\n\u003ca href=\"https://github.com/Kasturi\"\u003e@kasturi\u003c/a\u003e\n\u003ca href=\"https://github.com/laurah\"\u003e@laurah\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/Adekemi\"\u003e@Adekemi\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGithub repo:\u003c/h4\u003e\u003ca id=\"user-content-github-repo\" class=\"anchor\" aria-label=\"Permalink: Github repo:\" href=\"#github-repo\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/Kwabena\"\u003e@Kwabena\u003c/a\u003e (Lead)\n\u003ca href=\"https://github.com/Tobibantu\"\u003e@Tobibantu\u003c/a\u003e\n@Vincentabayomi\n@Paulah_Kings\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGraphics design:\u003c/h4\u003e\u003ca id=\"user-content-graphics-design\" class=\"anchor\" aria-label=\"Permalink: Graphics design:\" href=\"#graphics-design\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/Tobibantu\"\u003e@Tobibantu\u003c/a\u003e\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"Genomics_one_c-Project","anchor":"genomics_one_c-project","htmlText":"Genomics_one_c-Project"},{"level":1,"text":"Identification of somatic and germline variants from tumor and normal sample pairs","anchor":"identification-of-somatic-and-germline-variants-from-tumor-and-normal-sample-pairs","htmlText":"Identification of somatic and germline variants from tumor and normal sample pairs"},{"level":2,"text":"Introduction","anchor":"introduction","htmlText":"Introduction"},{"level":4,"text":"Agenda","anchor":"agenda","htmlText":"Agenda"},{"level":1,"text":"Data Preparation","anchor":"data-preparation","htmlText":"Data Preparation"},{"level":2,"text":"Get data","anchor":"get-data","htmlText":"Get data"},{"level":3,"text":"Hands-on: Data upload","anchor":"hands-on-data-upload","htmlText":"Hands-on: Data upload"},{"level":1,"text":"Quality control and mapping of NGS reads","anchor":"quality-control-and-mapping-of-ngs-reads","htmlText":"Quality control and mapping of NGS reads"},{"level":5,"text":"More on quality control and mapping","anchor":"more-on-quality-control-and-mapping","htmlText":"More on quality control and mapping"},{"level":2,"text":"Quality control","anchor":"quality-control","htmlText":"Quality control"},{"level":4,"text":"Hands-on: Quality Control of the input datasets","anchor":"hands-on-quality-control-of-the-input-datasets","htmlText":"Hands-on: Quality Control of the input datasets"},{"level":2,"text":"Read trimming and filtering","anchor":"read-trimming-and-filtering","htmlText":"Read trimming and filtering"},{"level":4,"text":"Hands-on: Read trimming and filtering of the normal tissue reads","anchor":"hands-on-read-trimming-and-filtering-of-the-normal-tissue-reads","htmlText":"Hands-on: Read trimming and filtering of the normal tissue reads"},{"level":2,"text":"Read Mapping","anchor":"read-mapping","htmlText":"Read Mapping"},{"level":4,"text":"Hands-on: Read Mapping","anchor":"hands-on-read-mapping","htmlText":"Hands-on: Read Mapping"},{"level":3,"text":"Using the imported hg19 sequence","anchor":"using-the-imported-hg19-sequence","htmlText":"Using the imported hg19 sequence"},{"level":1,"text":"Mapped reads postprocessing","anchor":"mapped-reads-postprocessing","htmlText":"Mapped reads postprocessing"},{"level":2,"text":"Filtering on mapped reads properties","anchor":"filtering-on-mapped-reads-properties","htmlText":"Filtering on mapped reads properties"},{"level":3,"text":"Hands-on: Filtering for mapping status and quality","anchor":"hands-on-filtering-for-mapping-status-and-quality","htmlText":"Hands-on: Filtering for mapping status and quality"},{"level":2,"text":"Removing duplicate reads","anchor":"removing-duplicate-reads","htmlText":"Removing duplicate reads"},{"level":3,"text":"Hands-on: Remove duplicates","anchor":"hands-on-remove-duplicates","htmlText":"Hands-on: Remove duplicates"},{"level":2,"text":"Left-align reads around indels","anchor":"left-align-reads-around-indels","htmlText":"Left-align reads around indels"},{"level":3,"text":"Hands-on: Left-align","anchor":"hands-on-left-align","htmlText":"Hands-on: Left-align"},{"level":2,"text":"Recalibrate read mapping qualities","anchor":"recalibrate-read-mapping-qualities","htmlText":"Recalibrate read mapping qualities"},{"level":3,"text":"Hands-on: Recalibrate read quality scores","anchor":"hands-on-recalibrate-read-quality-scores","htmlText":"Hands-on: Recalibrate read quality scores"},{"level":2,"text":"Refilter reads based on mapping quality","anchor":"refilter-reads-based-on-mapping-quality","htmlText":"Refilter reads based on mapping quality"},{"level":3,"text":"Hands-on: Eliminating reads with undefined mapping quality","anchor":"hands-on-eliminating-reads-with-undefined-mapping-quality","htmlText":"Hands-on: Eliminating reads with undefined mapping quality"},{"level":1,"text":"Variant Calling and Classification","anchor":"variant-calling-and-classification","htmlText":"Variant Calling and Classification"},{"level":3,"text":"Hands-on: Variant Calling and Classification","anchor":"hands-on-variant-calling-and-classification","htmlText":"Hands-on: Variant Calling and Classification"},{"level":1,"text":"Variant annotation and reporting","anchor":"variant-annotation-and-reporting","htmlText":"Variant annotation and reporting"},{"level":2,"text":"Get data","anchor":"get-data-1","htmlText":"Get data"},{"level":4,"text":"Hands-on: Data upload","anchor":"hands-on-data-upload-1","htmlText":"Hands-on: Data upload"},{"level":2,"text":"Adding annotations to the called variants","anchor":"adding-annotations-to-the-called-variants","htmlText":"Adding annotations to the called variants"},{"level":3,"text":"Adding functional genomic annotations","anchor":"adding-functional-genomic-annotations","htmlText":"Adding functional genomic annotations"},{"level":4,"text":"Hands-on: Adding annotations with SnpEff","anchor":"hands-on-adding-annotations-with-snpeff","htmlText":"Hands-on: Adding annotations with SnpEff"},{"level":3,"text":"Adding genetic and clinical evidence-base annotations","anchor":"adding-genetic-and-clinical-evidence-base-annotations","htmlText":"Adding genetic and clinical evidence-base annotations"},{"level":4,"text":"Hands-on: Creating a GEMINI database from a variants dataset","anchor":"hands-on-creating-a-gemini-database-from-a-variants-dataset","htmlText":"Hands-on: Creating a GEMINI database from a variants dataset"},{"level":4,"text":"Hands-on: Making variant call statistics accessible","anchor":"hands-on-making-variant-call-statistics-accessible","htmlText":"Hands-on: Making variant call statistics accessible"},{"level":4,"text":"Hands-on: Adding further annotations","anchor":"hands-on-adding-further-annotations","htmlText":"Hands-on: Adding further annotations"},{"level":2,"text":"Reporting selected subsets of variants","anchor":"reporting-selected-subsets-of-variants","htmlText":"Reporting selected subsets of variants"},{"level":4,"text":"Hands-on: Querying the GEMINI database for somatic variants","anchor":"hands-on-querying-the-gemini-database-for-somatic-variants","htmlText":"Hands-on: Querying the GEMINI database for somatic variants"},{"level":4,"text":"Hands-on: More complex filter criteria","anchor":"hands-on-more-complex-filter-criteria","htmlText":"Hands-on: More complex filter criteria"},{"level":4,"text":"Hands-on: SQL-based output formatting","anchor":"hands-on-sql-based-output-formatting","htmlText":"Hands-on: SQL-based output formatting"},{"level":2,"text":"Generating reports of genes affected by variants","anchor":"generating-reports-of-genes-affected-by-variants","htmlText":"Generating reports of genes affected by variants"},{"level":4,"text":"Hands-on: Turning query results into gene-centered reports","anchor":"hands-on-turning-query-results-into-gene-centered-reports","htmlText":"Hands-on: Turning query results into gene-centered reports"},{"level":2,"text":"Adding additional annotations to the gene-centered report","anchor":"adding-additional-annotations-to-the-gene-centered-report","htmlText":"Adding additional annotations to the gene-centered report"},{"level":4,"text":"Hands-on: Join","anchor":"hands-on-join","htmlText":"Hands-on: Join"},{"level":4,"text":"Hands-on: Rearrange to get a fully annotated gene report","anchor":"hands-on-rearrange-to-get-a-fully-annotated-gene-report","htmlText":"Hands-on: Rearrange to get a fully annotated gene report"},{"level":1,"text":"Conclusion","anchor":"conclusion","htmlText":"Conclusion"},{"level":3,"text":"Names of contributors:","anchor":"names-of-contributors","htmlText":"Names of contributors:"},{"level":4,"text":"Galaxy workflow:","anchor":"galaxy-workflow","htmlText":"Galaxy workflow:"},{"level":4,"text":"Github repo:","anchor":"github-repo","htmlText":"Github repo:"},{"level":4,"text":"Graphics design:","anchor":"graphics-design","htmlText":"Graphics design:"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2FErnestoDibia%2FGenomics_one_c-Project"}}],"overviewFilesProcessingTime":196.562126}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/ErnestoDibia/Genomics_one_c-Project/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/ErnestoDibia/Genomics_one_c-Project/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/ErnestoDibia/Genomics_one_c-Project/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/ErnestoDibia/Genomics_one_c-Project/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/ErnestoDibia/Genomics_one_c-Project/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           2 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/ErnestoDibia/Genomics_one_c-Project/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/ErnestoDibia/Genomics_one_c-Project/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/ErnestoDibia/Genomics_one_c-Project/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Genomics_one_c-Project
                     </h1>
                     <a aria-label="Permalink: Genomics_one_c-Project" class="anchor" href="#genomics_one_c-project" id="user-content-genomics_one_c-project">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     [toc]
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Identification of somatic and germline variants from tumor and normal sample pairs
                     </h1>
                     <a aria-label="Permalink: Identification of somatic and germline variants from tumor and normal sample pairs" class="anchor" href="#identification-of-somatic-and-germline-variants-from-tumor-and-normal-sample-pairs" id="user-content-identification-of-somatic-and-germline-variants-from-tumor-and-normal-sample-pairs">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Introduction
                     </h2>
                     <a aria-label="Permalink: Introduction" class="anchor" href="#introduction" id="user-content-introduction">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Cancer cells have deviated from the organism's normal (germline) genome by acquiring and selecting for a set of mutations that allow them to grow rapidly and invasively, resist regulation, and possibly metastasize. These changes can range from simple single-base mutations to more complex events involving genomic gain, loss, or structural change. Modifying the function of a protein (e.g., disabling a tumor suppressor gene or activating an oncogene), silencing a gene's transcription, or affecting a gene's transcriptional affinity can then initiate the cancer process.
                    </p>
                    <p dir="auto">
                     Many studies have chosen to sample and sequence both tumor tissue and normal genomic profiles from the same individual in order to distinguish acquired (somatic) mutations from malignant tissue. However, this solution is insufficient and as simple as the comparison being performed because there are many thousands of variants in healthy tissues compared to the reference genome. The reason for this is that each person inherits a unique pattern with many variants from their parents.
                    </p>
                    <blockquote>
                     <p dir="auto">
                      The primary distinction between
                      <em>
                       <strong>
                        germline variants
                       </strong>
                      </em>
                      and
                      <em>
                       <strong>
                        somatic variants
                       </strong>
                      </em>
                      is that germline variants are inherited and present in the carrier's germline, whereas somatic variants are acquired from the environment and thus cannot be passed down to the offspring.
                     </p>
                    </blockquote>
                    <p dir="auto">
                     If this is the case, healthy and tumor tissues may both have germline variants, while tumor tissues are the only ones with somatic variants. This is why comparing a normal cell to a tumor cell is the only way to tell the difference between the two types of variant.
                    </p>
                    <p dir="auto">
                     Furthermore, in addition to acquiring somatic variants, tumor tissues may gain chromosomal copies of heterozygous germline variants, i.e. homozygous state (duplication of one allelic copy accompanied by loss of the other), or lose, i.e. hemizygous state (if one allele is simply dropped).
                    </p>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Note: Only one of the two original alleles persists in the tumor.
                      </strong>
                     </p>
                     <p dir="auto">
                      This is known as
                      <em>
                       loss of heterozygozity
                      </em>
                      (LOH). Again, the detection of LOH events is based on a comparison of tumor and normal tissue data.
                     </p>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Genomics-One-C's
                      </strong>
                      tutorial is set out to:
                     </p>
                     <ul dir="auto">
                      <li>
                       Identify somatic and germline variants, as well as LOH variants, in a tumor and a normal sample from the same patient.
                      </li>
                      <li>
                       Annotate the variant sites and the genes affected by them with content from general human genetic and cancer-specific databases.
                      </li>
                     </ul>
                    </blockquote>
                    <p dir="auto">
                     We hope that this will provide insight into the genetic events that are driving tumor formation and growth in patients, as well as have prognostic and even therapeutic value by revealing variants known to affect drug resistance/sensitivity, tumor aggressiveness,
                     <em>
                      etc
                     </em>
                     .
                    </p>
                    <blockquote>
                     <p dir="auto">
                      Time estimation for this tutorial is approximately
                      <em>
                       <strong>
                        7 hours
                       </strong>
                      </em>
                     </p>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Galaxy Server GEMINI tool(s) issue
                      </strong>
                     </p>
                     <p dir="auto">
                      We realized that the use of the
                      <em>
                       GEMINI tool
                      </em>
                      (s) which you will encounter as you get to the
                      <strong>
                       Variant annotation and reporting
                      </strong>
                      section was not available on the
                      <a href="https://usegalaxy.org" rel="nofollow">
                       usegalaxy.org
                      </a>
                      server but instead on the
                      <a href="https://usegalaxy.eu" rel="nofollow">
                       usegalaxy.eu
                      </a>
                      server hence, you will have to perform you tutorial on the latter server.
                     </p>
                    </blockquote>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      <strong>
                       Agenda
                      </strong>
                     </h4>
                     <a aria-label="Permalink: Agenda" class="anchor" href="#agenda" id="user-content-agenda">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      In this tutorial, we will cover:
                     </p>
                     <ol dir="auto">
                      <li>
                       <a href="#data-preparation">
                        Data Preparation
                       </a>
                       <ol dir="auto">
                        <li>
                         <a href="#get-data">
                          Get data
                         </a>
                        </li>
                       </ol>
                      </li>
                      <li>
                       <a href="#quality-control-and-mapping-of-NGS-reads">
                        Quality control and mapping of NGS reads
                       </a>
                       <ol dir="auto">
                        <li>
                         <a href="#quality-control">
                          Quality control
                         </a>
                        </li>
                        <li>
                         <a href="#read-trimming-and-filtering">
                          Read trimming and filtering
                         </a>
                        </li>
                        <li>
                         <a href="#read-mapping">
                          Read mapping
                         </a>
                        </li>
                       </ol>
                      </li>
                      <li>
                       <a href="#mapped-reads-postprocessing">
                        Mapped reads postprocessing
                       </a>
                       <ol dir="auto">
                        <li>
                         <a href="#filtering-on-mapped-reads-properties">
                          Filtering on mapped reads properties
                         </a>
                        </li>
                        <li>
                         <a href="#removing-duplicate-reads">
                          Removing duplicate reads
                         </a>
                        </li>
                        <li>
                         <a href="#left-align-reads-around-indels">
                          Left-align reads around indels
                         </a>
                        </li>
                        <li>
                         <a href="#recalibrate-read-mapping-qualities">
                          Recalibrate read mapping qualities
                         </a>
                        </li>
                        <li>
                         <a href="#refilter-reads-based-on-mapping-quality">
                          Refilter reads based on mapping quality
                         </a>
                        </li>
                       </ol>
                      </li>
                      <li>
                       <a href="#variant-calling-and-classification">
                        Variant calling and classification
                       </a>
                      </li>
                      <li>
                       <a href="#variant-annotation-and-reporting">
                        Variant annotation and reporting
                       </a>
                       <ol dir="auto">
                        <li>
                         <a href="#get-data">
                          Get data
                         </a>
                        </li>
                        <li>
                         <a href="#adding-annotations-to-the-called-variants">
                          Adding annotations to the called variants
                         </a>
                        </li>
                        <li>
                         <a href="#reporting-selected-subsets-of-variants">
                          Reporting selected subsets of variants
                         </a>
                        </li>
                        <li>
                         <a href="#generating-reports-of-genes-affected-by-variants">
                          Generating reports of genes affected by variants
                         </a>
                        </li>
                        <li>
                         <a href="#adding-additional-annotations-to-the-gene-centered-report">
                          Adding additional annotations to the gene-centered report
                         </a>
                        </li>
                       </ol>
                      </li>
                     </ol>
                    </blockquote>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Data Preparation
                     </h1>
                     <a aria-label="Permalink: Data Preparation" class="anchor" href="#data-preparation" id="user-content-data-preparation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We must first upload and prepare some input data for analysis. The sequencing reads we'll be looking at come from real-world data from a cancer patient's tumor and normal tissue samples. However, for the sake of analysis speed, the original data has been downsampled to include only reads from human chromosomes 5, 12, and 17.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Get data
                     </h2>
                     <a aria-label="Permalink: Get data" class="anchor" href="#get-data" id="user-content-get-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Data upload
                     </h3>
                     <a aria-label="Permalink: Hands-on: Data upload" class="anchor" href="#hands-on-data-upload" id="user-content-hands-on-data-upload">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Create a new history for this tutorial and give it a meaningful name. eg.
                       <em>
                        Identification of somatic and germline variants from tumor and normal sample pairs (Genomics-One-C)
                       </em>
                      </p>
                      <blockquote>
                       <p dir="auto">
                        💡
                        <strong>
                         Tip: Creating a new history
                        </strong>
                       </p>
                       <p dir="auto">
                        Click the ➕ icon at the top of the history panel to the right of your screen.
                       </p>
                       <p dir="auto">
                        If the ➕ is missing:
                       </p>
                       <ol dir="auto" start="2">
                        <li>
                         Click on the ⚙️ icon (
                         <strong>
                          History options
                         </strong>
                         ) on the top of the history panel
                        </li>
                        <li>
                         Select the option
                         <strong>
                          Create New
                         </strong>
                         from the menu
                        </li>
                       </ol>
                      </blockquote>
                      <blockquote>
                       <p dir="auto">
                        💡
                        <strong>
                         Tip: Renaming a history
                        </strong>
                       </p>
                       <ol dir="auto">
                        <li>
                         <p dir="auto">
                          Click on
                          <strong>
                           Unnamed history
                          </strong>
                          (or the current name of the history)
                         </p>
                         <p dir="auto">
                          (
                          <strong>
                           Click to rename history
                          </strong>
                          ) at the top of your history panel
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          Type the new name
                         </p>
                        </li>
                        <li>
                         <p dir="auto">
                          Press
                          <strong>
                           <code>
                            Enter
                           </code>
                          </strong>
                         </p>
                        </li>
                       </ol>
                      </blockquote>
                     </li>
                     <li>
                      <p dir="auto">
                       Import the following four files from
                       <a href="https://zenodo.org/record/2582555" rel="nofollow">
                        Zenodo
                       </a>
                       :
                      </p>
                      <blockquote>
                       <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="https://zenodo.org/record/2582555/files/SLGFSK-N_231335_r1_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-N_231335_r2_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-T_231336_r1_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-T_231336_r2_chr5_12_17.fastq.gz">
                        <pre class="notranslate"><code>https://zenodo.org/record/2582555/files/SLGFSK-N_231335_r1_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-N_231335_r2_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-T_231336_r1_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-T_231336_r2_chr5_12_17.fastq.gz
</code></pre>
                       </div>
                      </blockquote>
                      <p dir="auto">
                       where the first two files represent data from a patient's normal tissue's forward and reverse reads sequences, and the last two files represent data from the same patient's tumor tissue.
                      </p>
                      <p dir="auto">
                       Alternatively, the same files may be available on your Galaxy server via a shared data library (ask your instructor), in which case you should import the data directly from there.
                      </p>
                      <blockquote>
                       <p dir="auto">
                        💡
                        <strong>
                         Tip: Importing via links
                        </strong>
                       </p>
                       <ul dir="auto">
                        <li>
                         ​	Copy the link location
                        </li>
                        <li>
                         Open the Galaxy Upload Manager (on the top-left of the tool panel)
                        </li>
                        <li>
                         Select
                         <em>
                          <strong>
                           Paste/Fetch Data
                          </strong>
                         </em>
                        </li>
                        <li>
                         Paste the link into the text field
                        </li>
                        <li>
                         Change
                         <em>
                          <strong>
                           Type (set all)
                          </strong>
                         </em>
                         : from “Auto-detect” to
                         <span>
                          <code>
                           fastqsanger.gz
                          </code>
                         </span>
                        </li>
                        <li>
                         Press
                         <em>
                          <strong>
                           Start
                          </strong>
                         </em>
                        </li>
                        <li>
                         <em>
                          <strong>
                           Close
                          </strong>
                         </em>
                         the window
                        </li>
                        <li>
                         By default, Galaxy uses the URL as the name, so rename the files with a more useful name.
                        </li>
                       </ul>
                      </blockquote>
                      <blockquote>
                       <p dir="auto">
                        💡
                        <strong>
                         Tip: Importing data from a data library
                        </strong>
                       </p>
                       <p dir="auto">
                        As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a shared data
                        <em>
                         library
                        </em>
                        :
                       </p>
                       <ul dir="auto">
                        <li>
                         Go into
                         <em>
                          <strong>
                           Shared data
                          </strong>
                         </em>
                         (top panel) then
                         <em>
                          <strong>
                           Data libraries
                          </strong>
                         </em>
                        </li>
                        <li>
                         Find the correct folder (ask your instructor)
                        </li>
                        <li>
                         Select the desired files
                        </li>
                        <li>
                         Click on the
                         <em>
                          <strong>
                           To History
                          </strong>
                         </em>
                         button near the top and select
                         <em>
                          <strong>
                           as Datasets
                          </strong>
                         </em>
                         from the dropdown menu
                        </li>
                        <li>
                         In the pop-up window, select the history you want to import the files to (or create a new one)
                        </li>
                        <li>
                         Click on
                         <em>
                          <strong>
                           Import
                          </strong>
                         </em>
                        </li>
                       </ul>
                      </blockquote>
                     </li>
                     <li>
                      <p dir="auto">
                       Check that all newly created datasets in your history have the correct datatypes assigned, and correct any missing or incorrect datatype assignment.
                      </p>
                      <blockquote>
                       <p dir="auto">
                        💡
                        <strong>
                         Tip: Changing the datatype
                        </strong>
                       </p>
                       <ul dir="auto">
                        <li>
                         Click on the
                         <em>
                          <strong>
                           pencil icon
                          </strong>
                         </em>
                         for the dataset to edit its attributes
                        </li>
                        <li>
                         In the central panel, click on the galaxy-chart-select-data
                         <em>
                          <strong>
                           Datatypes
                          </strong>
                         </em>
                         tab on the top
                        </li>
                        <li>
                         Select
                         <span>
                          <code>
                           fastqsanger.gz
                          </code>
                         </span>
                        </li>
                        <li>
                         Click the
                         <em>
                          <strong>
                           Change datatype
                          </strong>
                         </em>
                         button
                        </li>
                       </ul>
                      </blockquote>
                     </li>
                     <li>
                      <p dir="auto">
                       Rename the datasets and tag them appropriately.
                      </p>
                      <p dir="auto">
                       When you upload a dataset via a link, Galaxy will use the link address as the dataset name, which you should probably shorten to just the file names
                      </p>
                      <blockquote>
                       <p dir="auto">
                        💡
                        <strong>
                         Tip: Renaming a dataset
                        </strong>
                       </p>
                       <ul dir="auto">
                        <li>
                         Click on the galaxy-pencil
                         <em>
                          <strong>
                           pencil icon
                          </strong>
                         </em>
                         for the dataset to edit its attributes
                        </li>
                        <li>
                         In the central panel, change the
                         <em>
                          <strong>
                           Name
                          </strong>
                         </em>
                         field
                        </li>
                        <li>
                         Click the
                         <em>
                          <strong>
                           Save
                          </strong>
                         </em>
                         button
                        </li>
                       </ul>
                      </blockquote>
                      <p dir="auto">
                       A large portion of the analysis in this tutorial will consist of identical steps performed in parallel on normal and tumor tissue data.
                      </p>
                      <p dir="auto">
                       Galaxy supports dataset tags to make it easier to remember which dataset represents which branch of an analysis in a linear history. In particular, if you add a tag beginning with
                       <span>
                        <code>
                         #
                        </code>
                       </span>
                       to any dataset, that tag will automatically propagate to any new dataset derived from the tagged dataset.
                      </p>
                      <blockquote>
                       <p dir="auto">
                        💡
                        <strong>
                         Tip: Adding a tag
                        </strong>
                       </p>
                       <ul dir="auto">
                        <li>
                         Click on the dataset
                        </li>
                        <li>
                         Click on
                         <strong>
                          Edit dataset tags
                         </strong>
                        </li>
                        <li>
                         Add a tag starting with
                         <span>
                          <code>
                           #
                          </code>
                         </span>
                        </li>
                        <li>
                         Tags starting with
                         <span>
                          <code>
                           #
                          </code>
                         </span>
                         will be automatically propagated to the outputs of tools using this dataset.
                        </li>
                        <li>
                         Check that the tag is appearing below the dataset name
                        </li>
                       </ul>
                      </blockquote>
                      <p dir="auto">
                       Before we begin our analysis, it is a good idea to tag the two fastq datasets representing normal tissue with, for example,
                       <span>
                        <code>
                         #normal
                        </code>
                       </span>
                       and the two datasets representing tumor tissue with, for example,
                       <span>
                        <code>
                         #tumor
                        </code>
                       </span>
                       .
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Import the reference genome with the
                       <span>
                        <code>
                         hg19
                        </code>
                       </span>
                       version of the sequences of human chromosomes 5, 12 and 17:
                      </p>
                      <blockquote>
                       <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="https://zenodo.org/record/2582555/files/hg19.chr5_12_17.fa.gz">
                        <pre class="notranslate" lang="english"><code>https://zenodo.org/record/2582555/files/hg19.chr5_12_17.fa.gz
</code></pre>
                       </div>
                      </blockquote>
                      <p dir="auto">
                       In the import dialog, make sure to select
                       <span>
                        <code>
                         fasta
                        </code>
                       </span>
                       as the datatype.
                      </p>
                      <blockquote>
                       <p dir="auto">
                        <strong>
                         Shortcut
                        </strong>
                       </p>
                       <p dir="auto">
                        You can skip this step if the Galaxy server you're using has a hg19 version of the human reference genome with prebuilt indexes for bwa-mem and samtools (check the tools Map with BWA-MEM tool and VarScan Somatic tool to see if a hg19 version is listed as an option under "(Using) reference genome").
                       </p>
                      </blockquote>
                      <p dir="auto">
                       Alternatively, load the dataset from a shared data library.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Rename the reference genome
                      </p>
                      <p dir="auto">
                       The reference genome you imported above came as a compressed file, but Galaxy unpacked it to plain
                       <span>
                        <code>
                         fasta
                        </code>
                       </span>
                       format based on your datatype selection. You might want to remove the
                       <span>
                        <code>
                         .gz
                        </code>
                       </span>
                       suffix from the dataset name now.
                      </p>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Quality control and mapping of NGS reads
                     </h1>
                     <a aria-label="Permalink: Quality control and mapping of NGS reads" class="anchor" href="#quality-control-and-mapping-of-ngs-reads" id="user-content-quality-control-and-mapping-of-ngs-reads">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Prior to beginning the analysis, it is critical to ensure that the data used for analysis is of high quality. The quality control phase entails identifying and trimming low-quality parts as well as ditching low-quality reads that may be present in the input data to avoid spurious variant calls. In this step, we ensure that all sequencing reads used meet some basic quality criteria.
                    </p>
                    <blockquote>
                     <div class="markdown-heading" dir="auto">
                      <h5 class="heading-element" dir="auto" tabindex="-1">
                       More on quality control and mapping
                      </h5>
                      <a aria-label="Permalink: More on quality control and mapping" class="anchor" href="#more-on-quality-control-and-mapping" id="user-content-more-on-quality-control-and-mapping">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <p dir="auto">
                      If you want to learn more about quality control and mapping, check out the
                      <a href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html" rel="nofollow">
                       Quality Control
                      </a>
                      and
                      <a href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial" rel="nofollow">
                       Mapping
                      </a>
                      . We will only show the concrete steps required for quality control and read mapping of our specific datasets in this section.
                     </p>
                    </blockquote>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Quality control
                     </h2>
                     <a aria-label="Permalink: Quality control" class="anchor" href="#quality-control" id="user-content-quality-control">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Quality Control of the input datasets
                     </h4>
                     <a aria-label="Permalink: Hands-on: Quality Control of the input datasets" class="anchor" href="#hands-on-quality-control-of-the-input-datasets" id="user-content-hands-on-quality-control-of-the-input-datasets">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        FastQC
                       </strong>
                       on each of your four fastq datasets.
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          "Short read data from your current history"
                         </em>
                         : all 4 FASTQ datasets selected with
                         <strong>
                          Multiple datasets
                         </strong>
                        </p>
                        <blockquote>
                         <p dir="auto">
                          Tip: Select multiple datasets
                         </p>
                         <ol dir="auto">
                          <li>
                           Click on
                           <strong>
                            Multiple datasets
                           </strong>
                          </li>
                          <li>
                           Select several files by keeping the
                           <span>
                            Ctrl
                           </span>
                           (or
                           <span>
                            <code>
                             COMMAND
                            </code>
                           </span>
                           ) key pressed and clicking on the files of interest
                          </li>
                         </ol>
                        </blockquote>
                        <p dir="auto">
                         <em>
                          8 new datasets (one with the calculated raw data, another one with an html report of the findings for each input dataset) wil get added to your history after you run this job.
                         </em>
                        </p>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <p dir="auto">
                       Use
                       <strong>
                        MultiQC
                       </strong>
                       to aggregate the raw
                       <strong>
                        FastQC
                       </strong>
                       data of all four input datasets into one report
                      </p>
                      <ul dir="auto">
                       <li>
                        In
                        <em>
                         "Results"
                        </em>
                        <ul dir="auto">
                         <li>
                          <em>
                           "Which tool was used to generate logs?"
                          </em>
                          :
                          <span>
                           <code>
                            FastQC
                           </code>
                          </span>
                         </li>
                         <li>
                          In
                          <em>
                           "FastQC output"
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             "Type of FastQC output?"
                            </em>
                            :
                            <span>
                             <code>
                              Raw data
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "FastQC output"
                            </em>
                            : all four
                            <em>
                             RawData
                            </em>
                            outputs of
                            <strong>
                             FastQC
                            </strong>
                           </li>
                          </ul>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <p dir="auto">
                       Inspect the
                       <em>
                        Webpage
                       </em>
                       output produced by the tool
                      </p>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Read trimming and filtering
                     </h2>
                     <a aria-label="Permalink: Read trimming and filtering" class="anchor" href="#read-trimming-and-filtering" id="user-content-read-trimming-and-filtering">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Despite the relatively good overall quality of the raw reads, we will apply read trimming and filtering to see if we can improve things further, but also to demonstrate the general concept.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      <strong>
                       Hands-on: Read trimming and filtering of the normal tissue reads
                      </strong>
                     </h4>
                     <a aria-label="Permalink: Hands-on: Read trimming and filtering of the normal tissue reads" class="anchor" href="#hands-on-read-trimming-and-filtering-of-the-normal-tissue-reads" id="user-content-hands-on-read-trimming-and-filtering-of-the-normal-tissue-reads">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run the
                       <strong>
                        Trimmomatic
                       </strong>
                       tool to trim and filter the normal tissue reads
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          “Single-end or paired-end reads?”
                         </em>
                         :
                         <span>
                          <code>
                           Paired-end (two separate input files)
                          </code>
                         </span>
                        </p>
                        <p dir="auto">
                         This makes the tool treat the forward and reverse reads simultaneously.
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           <em>
                            “Input FASTQ file (R1/first of pair)”
                           </em>
                           : the forward reads (r1) dataset of the normal tissue sample
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           <em>
                            “Input FASTQ file (R2/second of pair)”
                           </em>
                           : the reverse reads (r2) dataset of the normal tissue sample
                          </p>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Perform initial ILLUMINACLIP step?”
                         </em>
                         :
                         <span>
                          <code>
                           Yes
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           <em>
                            Select standard adapter sequences or provide custom?”
                           </em>
                           :
                           <span>
                            <code>
                             Standard
                            </code>
                           </span>
                          </p>
                          <ul dir="auto">
                           <li>
                            <em>
                             "Adapter sequences to use”
                            </em>
                            :
                            <span>
                             <code>
                              TrueSeq3 (paired-ended, for MiSeq and HiSeq)
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          <p dir="auto">
                           <em>
                            “Maximum mismatch count which will still allow a full match to be  performed”
                           </em>
                           :
                           <span>
                            <code>
                             2
                            </code>
                           </span>
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           <em>
                            “How accurate the match between the two ‘adapter ligated’ reads must  be for PE palindrome read alignment”
                           </em>
                           :
                           <span>
                            <code>
                             30
                            </code>
                           </span>
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           <em>
                            “How accurate the match between any adapter etc. sequence must be  against a read”
                           </em>
                           :
                           <span>
                            <code>
                             10
                            </code>
                           </span>
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           <em>
                            “Minimum length of adapter that needs to be detected (PE specific/  palindrome mode)”
                           </em>
                           :
                           <span>
                            <code>
                             8
                            </code>
                           </span>
                          </p>
                          <p dir="auto">
                           <em>
                            “Always keep both reads (PE specific/palindrome mode)?”
                           </em>
                           :
                           <span>
                            <code>
                             Yes
                            </code>
                           </span>
                          </p>
                         </li>
                        </ul>
                       </li>
                      </ul>
                      <p dir="auto">
                       These parameters are used to cut ILLUMINA-specific adapter sequences from the reads.
                      </p>
                      <ul dir="auto">
                       <li>
                        In
                        <em>
                         “Trimmomatic Operation”
                        </em>
                        <ul dir="auto">
                         <li>
                          In
                          <em>
                           “1: Trimmomatic Operation”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “Select Trimmomatic operation to perform”
                            </em>
                            :
                            <span>
                             <code>
                              Cut the specified number of bases from the start of the read (HEADCROP)
                             </code>
                            </span>
                            <ul dir="auto">
                             <li>
                              <em>
                               “Number of bases to remove from the start of the read”
                              </em>
                              :
                              <span>
                               <code>
                                3
                               </code>
                              </span>
                             </li>
                            </ul>
                           </li>
                          </ul>
                         </li>
                         <li>
                          “Insert Trimmomatic Operation”*
                         </li>
                         <li>
                          In
                          <em>
                           “2: Trimmomatic Operation”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “Select Trimmomatic operation to perform”
                            </em>
                            :
                            <span>
                             <code>
                              Cut bases off the end of a read, if below a threshold quality (TRAILING)
                             </code>
                            </span>
                            <ul dir="auto">
                             <li>
                              <em>
                               “Minimum quality required to keep a base”
                              </em>
                              :
                              <span>
                               <code>
                                10
                               </code>
                              </span>
                             </li>
                            </ul>
                           </li>
                          </ul>
                         </li>
                         <li>
                          “Insert Trimmomatic Operation”*
                         </li>
                         <li>
                          In
                          <em>
                           “3: Trimmomatic Operation”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “Select Trimmomatic operation to perform”
                            </em>
                            :
                            <span>
                             <code>
                              Drop reads below a specified length (MINLEN)
                             </code>
                            </span>
                            <ul dir="auto">
                             <li>
                              <em>
                               “Minimum quality required to keep a base”
                              </em>
                              :
                              <span>
                               <code>
                                25
                               </code>
                              </span>
                             </li>
                            </ul>
                           </li>
                          </ul>
                         </li>
                        </ul>
                       </li>
                      </ul>
                      <p dir="auto">
                       These three trimming and filtering operations will be applied to the reads in the given order after adapter trimming.
                      </p>
                      <blockquote>
                       <p dir="auto">
                        Running this job will generate four output datasets:
                       </p>
                       <ul dir="auto">
                        <li>
                         <em>
                          two for the trimmed forward and reverse reads with a proper mate in the other dataset
                         </em>
                        </li>
                        <li>
                         <em>
                          two more datasets of orphaned forward and reverse reads, for which the corresponding mate was dropped due to insufficient length after trimming; however, when you inspect these two files, you should find that they are empty because none of our relatively high quality reads were excessively trimmed. You can delete the two datasets to make your history smaller
                         </em>
                         .
                        </li>
                       </ul>
                      </blockquote>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Read Mapping
                     </h2>
                     <a aria-label="Permalink: Read Mapping" class="anchor" href="#read-mapping" id="user-content-read-mapping">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Read Mapping
                     </h4>
                     <a aria-label="Permalink: Hands-on: Read Mapping" class="anchor" href="#hands-on-read-mapping" id="user-content-hands-on-read-mapping">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Use
                       <strong>
                        Map with BWA-MEM
                       </strong>
                       Tool: to map the reads from the
                       <strong>
                        normal tissue
                       </strong>
                       sample to the reference genome
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          “Will you select a reference genome from your history or use a built-in index?”
                         </em>
                         :
                         <span>
                          <code>
                           Use a built-in genome
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           <em>
                            “Using reference genome”
                           </em>
                           :
                           <span>
                            <code>
                             Human: hg19
                            </code>
                           </span>
                           (or a similarly named option)
                          </p>
                          <blockquote>
                           <div class="markdown-heading" dir="auto">
                            <h3 class="heading-element" dir="auto" tabindex="-1">
                             Using the imported
                             <span>
                              <code>
                               hg19
                              </code>
                             </span>
                             sequence
                            </h3>
                            <a aria-label="Permalink: Using the imported hg19 sequence" class="anchor" href="#using-the-imported-hg19-sequence" id="user-content-using-the-imported-hg19-sequence">
                             <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                              </path>
                             </svg>
                            </a>
                           </div>
                           <p dir="auto">
                            In the event you have imported the
                            <span>
                             <code>
                              hg19
                             </code>
                            </span>
                            sequence as a fasta dataset into your history, you could instead:
                           </p>
                           <ul dir="auto">
                            <li>
                             <p dir="auto">
                              <em>
                               "Will you select a reference genome from your history or use a built-in genome?"
                              </em>
                              :
                              <span>
                               <code>
                                Use a genome from the history
                               </code>
                              </span>
                             </p>
                            </li>
                            <li>
                             <p dir="auto">
                              <em>
                               "reference genome"
                              </em>
                              : your imported
                              <span>
                               <code>
                                hg19
                               </code>
                              </span>
                              fasta dataset.
                             </p>
                            </li>
                           </ul>
                          </blockquote>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <p dir="auto">
                     ​
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       “Single or Paired-end reads”*:
                       <span>
                        <code>
                         Paired
                        </code>
                       </span>
                      </p>
                      <ul dir="auto">
                       <li>
                        <em>
                         “Select first set of reads”
                        </em>
                        : the trimmed  forward reads (r1) dataset of the
                        <strong>
                         normal tissue
                        </strong>
                        sample; output of
                        <strong>
                         Trimmomatic
                        </strong>
                       </li>
                       <li>
                        <em>
                         “Select second set of reads”
                        </em>
                        : the trimmed reverse reads (r2) dataset of the
                        <strong>
                         normal tissue
                        </strong>
                        sample; output of
                        <strong>
                         Trimmomatic
                        </strong>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <p dir="auto">
                       <em>
                        “Set read groups information?”
                       </em>
                       :
                       <span>
                        <code>
                         Set read groups (SAM/BAM specification)
                        </code>
                       </span>
                      </p>
                      <ul dir="auto">
                       <li>
                        <em>
                         “Auto-assign”
                        </em>
                        :
                        <span>
                         <code>
                          No
                         </code>
                        </span>
                        <ul dir="auto">
                         <li>
                          <em>
                           “Read group identifier (ID)”
                          </em>
                          :
                          <span>
                           <code>
                            231335
                           </code>
                          </span>
                          (this value being taken from the original name of the normal tissue input files)
                         </li>
                        </ul>
                       </li>
                       <li>
                        <em>
                         “Auto-assign”
                        </em>
                        :
                        <span>
                         <code>
                          No
                         </code>
                        </span>
                        <ul dir="auto">
                         <li>
                          <em>
                           “Read group sample name (SM)”
                          </em>
                          :
                          <span>
                           <code>
                            Normal
                           </code>
                          </span>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Tip
                      </strong>
                      :
                      <em>
                       In general, you are free to choose your own ID and SM values, but the ID should unambiguously identify the sequencing run that generated the reads, while the SM value should identify the biological sample.
                      </em>
                     </p>
                    </blockquote>
                    <ol dir="auto" start="2">
                     <li>
                      <p dir="auto">
                       Use
                       <strong>
                        Map with BWA-MEM
                       </strong>
                       to map the reads from the
                       <strong>
                        tumor tissue
                       </strong>
                       sample,
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          “Will you select a reference genome from your history or use a built-in index?”
                         </em>
                         :
                         <span>
                          <code>
                           Use a built-in genome
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <em>
                           “Using reference genome”
                          </em>
                          :
                          <span>
                           <code>
                            Human: hg19
                           </code>
                          </span>
                          (or a similarly named option)
                         </li>
                        </ul>
                        <p dir="auto">
                         Adjust these settings as before if you are using the imported reference genome.
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Single or Paired-end reads”
                         </em>
                         :
                         <span>
                          <code>
                           Paired
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <em>
                           “Select first set of reads”
                          </em>
                          : the trimmed forward reads (r1) dataset of the
                          <strong>
                           tumor tissue
                          </strong>
                          sample; output of
                          <strong>
                           Trimmomatic
                          </strong>
                         </li>
                         <li>
                          <em>
                           Select second set of reads”
                          </em>
                          : the reverse reads (r2) dataset of the
                          <strong>
                           tumor tissue
                          </strong>
                          sample; output of
                          <strong>
                           Trimmomatic
                          </strong>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Set read groups information?”
                         </em>
                         :
                         <span>
                          <code>
                           Set read groups (SAM/BAM specification)
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <em>
                           “Auto-assign”
                          </em>
                          :
                          <code>
                           No
                          </code>
                          <ul dir="auto">
                           <li>
                            <em>
                             Read group identifier (ID)”
                            </em>
                            :
                            <code>
                             231336
                            </code>
                            (this value, again, being taken from the original name of the tumor tissue input files)
                           </li>
                          </ul>
                         </li>
                         <li>
                          <em>
                           “Auto-assign”
                          </em>
                          :
                          <span>
                           <code>
                            No
                           </code>
                          </span>
                          <ul dir="auto">
                           <li>
                            <em>
                             "Read group sample name (SM)"
                            </em>
                            :
                            <span>
                             <code>
                              Tumor
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Mapped reads postprocessing
                     </h1>
                     <a aria-label="Permalink: Mapped reads postprocessing" class="anchor" href="#mapped-reads-postprocessing" id="user-content-mapped-reads-postprocessing">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We will then perform some postprocessing to ensure that our variant analysis is based solely on unambiguous, high-quality read mappings.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Filtering on mapped reads properties
                     </h2>
                     <a aria-label="Permalink: Filtering on mapped reads properties" class="anchor" href="#filtering-on-mapped-reads-properties" id="user-content-filtering-on-mapped-reads-properties">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     To generate new filtered BAM datasets with only reads retained that have been successfully mapped to the reference, have a minimum mapping quality of 1, and have the mate read also mapped:
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Filtering for mapping status and quality
                     </h3>
                     <a aria-label="Permalink: Hands-on: Filtering for mapping status and quality" class="anchor" href="#hands-on-filtering-for-mapping-status-and-quality" id="user-content-hands-on-filtering-for-mapping-status-and-quality">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        Filter BAM datasets on a variety of attributes
                       </strong>
                       with the following parameters:
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          “BAM dataset(s) to filter”
                         </em>
                         : mapped reads datasets from the normal
                         <em>
                          and
                         </em>
                         the tumor tissue data, outputs of
                         <strong>
                          Map with BWA-MEM
                         </strong>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         In
                         <em>
                          “Condition”
                         </em>
                         :
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         In
                         <em>
                          “1: Condition”
                         </em>
                         :
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           In
                           <em>
                            “Filter”
                           </em>
                           :
                          </p>
                          <ul dir="auto">
                           <li>
                            <p dir="auto">
                             In
                             <em>
                              “1: Filter”
                             </em>
                             :
                            </p>
                            <ul dir="auto">
                             <li>
                              <em>
                               “Select BAM property to filter on”
                              </em>
                              :
                              <span>
                               <code>
                                mapQuality
                               </code>
                              </span>
                             </li>
                            </ul>
                           </li>
                           <li>
                            <p dir="auto">
                             ''
                             <em>
                              Filter on read mapping quality (phred scale)”
                             </em>
                             :
                             <span>
                              <code>
                               &gt;=1
                              </code>
                             </span>
                            </p>
                           </li>
                           <li>
                            <p dir="auto">
                             In
                             <em>
                              “2: Filter”
                             </em>
                             :
                            </p>
                            <ul dir="auto">
                             <li>
                              <em>
                               “Select BAM property to filter on”
                              </em>
                              :
                              <span>
                               <code>
                                isMapped
                               </code>
                              </span>
                              <ul dir="auto">
                               <li>
                                <em>
                                 “Selected mapped reads”
                                </em>
                                :
                                <span>
                                 <code>
                                  Yes
                                 </code>
                                </span>
                               </li>
                              </ul>
                             </li>
                            </ul>
                           </li>
                           <li>
                            <p dir="auto">
                             Click on
                             <em>
                              “Insert Filter”
                             </em>
                            </p>
                           </li>
                           <li>
                            <p dir="auto">
                             In
                             <em>
                              “3: Filter”
                             </em>
                             :
                            </p>
                            <ul dir="auto">
                             <li>
                              <em>
                               “Select BAM property to filter on”
                              </em>
                              :
                              <span>
                               <code>
                                isMateMapped
                               </code>
                              </span>
                              <ul dir="auto">
                               <li>
                                <em>
                                 “Select reads with mapped mate”
                                </em>
                                :
                                <span>
                                 <code>
                                  Yes
                                 </code>
                                </span>
                               </li>
                              </ul>
                             </li>
                            </ul>
                           </li>
                          </ul>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <p dir="auto">
                     When multiple filters are configured within a single condition, reads must pass
                     <em>
                      all
                     </em>
                     of the filters in order to be retained in the output. As a result, the above settings retain only read pairs for which both mates are mapped.
                    </p>
                    <p dir="auto">
                     It should be noted that filtering for the best mapping quality is not strictly necessary. Most variant callers (including
                     <strong>
                      VarScan somatic
                     </strong>
                     , which we will use later) allow us to use only reads with mapping quality above a certain threshold. However, in this section, we will process the retained reads in greater depth, so it will pay off in terms of performance to eliminate reads we do not intend to use at an early stage.
                    </p>
                    <p dir="auto">
                     <em>
                      “Would you like to set rules?”
                     </em>
                     :
                     <span>
                      <code>
                       No
                      </code>
                     </span>
                    </p>
                    <blockquote>
                     <p dir="auto">
                      This will result in two new datasets, one for each of the normal and tumor data.
                     </p>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Tools for filtering BAM datasets
                      </strong>
                     </p>
                     <p dir="auto">
                      Under the hood,
                      <strong>
                       Filter SAM or BAM, output SAM or BAM
                      </strong>
                      tool uses the
                      <span>
                       <code>
                        samtools view
                       </code>
                      </span>
                      command line tool, while
                      <strong>
                       Filter BAM datasets on a variety of attributes
                      </strong>
                      tool uses
                      <span>
                       <code>
                        bamtool filter
                       </code>
                      </span>
                      . Whatever the BAM filtering task, you should be able to perform it in Galaxy with one of these two tools.
                     </p>
                    </blockquote>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Removing duplicate reads
                     </h2>
                     <a aria-label="Permalink: Removing duplicate reads" class="anchor" href="#removing-duplicate-reads" id="user-content-removing-duplicate-reads">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Remove duplicates
                     </h3>
                     <a aria-label="Permalink: Hands-on: Remove duplicates" class="anchor" href="#hands-on-remove-duplicates" id="user-content-hands-on-remove-duplicates">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Run
                      <strong>
                       RmDup
                      </strong>
                      Tool with the following parameters:
                      <ul dir="auto">
                       <li>
                        <em>
                         “BAM file”
                        </em>
                        : filtered reads datasets from the normal
                        <em>
                         and
                        </em>
                        the tumor tissue data; the outputs of
                        <strong>
                         Filter SAM or BAM
                        </strong>
                       </li>
                       <li>
                        <em>
                         “Is this paired-end or single end data”
                        </em>
                        :
                        <span>
                         <code>
                          BAM is paired-end
                         </code>
                        </span>
                       </li>
                       <li>
                        <em>
                         “Treat as single-end”
                        </em>
                        :
                        <span>
                         <code>
                          No
                         </code>
                        </span>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <blockquote>
                     <p dir="auto">
                      Again, this will produce two new datasets, one for each of the normal and tumor data.
                     </p>
                    </blockquote>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Left-align reads around indels
                     </h2>
                     <a aria-label="Permalink: Left-align reads around indels" class="anchor" href="#left-align-reads-around-indels" id="user-content-left-align-reads-around-indels">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Left-align
                     </h3>
                     <a aria-label="Permalink: Hands-on: Left-align" class="anchor" href="#hands-on-left-align" id="user-content-hands-on-left-align">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Run
                      <strong>
                       BamLeftAlign
                      </strong>
                      Tool with the following parameters:
                      <ul dir="auto">
                       <li>
                        <em>
                         “Choose the source for the reference genome”
                        </em>
                        :
                        <span>
                         <code>
                          Locally cached
                         </code>
                        </span>
                        <ul dir="auto">
                         <li>
                          <em>
                           “BAM dataset to re-align”
                          </em>
                          : your filtered and deduplicated reads datasets from the normal
                          <em>
                           and
                          </em>
                          the tumor tissue data; the outputs of
                          <strong>
                           RmDup
                          </strong>
                         </li>
                         <li>
                          <em>
                           “Using reference genome”
                          </em>
                          :
                          <span>
                           <code>
                            Human: hg19
                           </code>
                          </span>
                          (or a similarly named choice)
                         </li>
                        </ul>
                       </li>
                       <li>
                        <em>
                         “Maximum number of iterations”
                        </em>
                        :
                        <span>
                         <code>
                          5
                         </code>
                        </span>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Using the imported
                       <code>
                        hg19
                       </code>
                       sequence
                      </strong>
                     </p>
                     <p dir="auto">
                      If you have imported the
                      <span>
                       <code>
                        hg19
                       </code>
                      </span>
                      sequence as a fasta dataset into your history instead:
                     </p>
                     <ul dir="auto">
                      <li>
                       <em>
                        “Choose the source for the reference genome”
                       </em>
                       :
                       <span>
                        <code>
                         History
                        </code>
                       </span>
                       <ul dir="auto">
                        <li>
                         <em>
                          “Using reference file”
                         </em>
                         : your imported
                         <span>
                          <code>
                           hg19
                          </code>
                         </span>
                         fasta dataset
                        </li>
                       </ul>
                      </li>
                     </ul>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      As before, this will generate two new datasets, one for each of the normal and tumor data.
                     </p>
                    </blockquote>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Recalibrate read mapping qualities
                     </h2>
                     <a aria-label="Permalink: Recalibrate read mapping qualities" class="anchor" href="#recalibrate-read-mapping-qualities" id="user-content-recalibrate-read-mapping-qualities">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Recalibrate read quality scores
                     </h3>
                     <a aria-label="Permalink: Hands-on: Recalibrate read quality scores" class="anchor" href="#hands-on-recalibrate-read-quality-scores" id="user-content-hands-on-recalibrate-read-quality-scores">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        CalMD
                       </strong>
                       Tool with the following parameters:
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          “BAM file to recalculate”
                         </em>
                         : the left-aligned datasets from the normal and the tumor tissue data; the outputs of
                         <strong>
                          BamLeftAlign
                         </strong>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Choose the source for the reference genome”
                         </em>
                         :
                         <span>
                          <code>
                           Use a built-in genome
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           <em>
                            “Using reference genome”
                           </em>
                           :
                           <span>
                            <code>
                             Human: hg19
                            </code>
                           </span>
                           (or a similarly named choice)
                          </p>
                          <blockquote>
                           <p dir="auto">
                            **Using the imported
                            <span>
                             <code>
                              hg19
                             </code>
                            </span>
                            sequence    **
                           </p>
                           <p dir="auto">
                            If you have imported the
                            <span>
                             <code>
                              hg19
                             </code>
                            </span>
                            sequence as a fasta dataset into your history instead:
                           </p>
                           <ul dir="auto">
                            <li>
                             <em>
                              “Choose the source for the reference genome”
                             </em>
                             :
                             <span>
                              <code>
                               Use a genome from the history
                              </code>
                             </span>
                             <ul dir="auto">
                              <li>
                               <em>
                                “Using reference file”
                               </em>
                               : your imported
                               <span>
                                <code>
                                 hg19
                                </code>
                               </span>
                               fasta dataset.
                              </li>
                             </ul>
                            </li>
                           </ul>
                          </blockquote>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Do you also want BAQ (Base Alignment Quality) scores to be calculated?”
                         </em>
                         :
                         <span>
                          <code>
                           No
                          </code>
                         </span>
                        </p>
                        <p dir="auto">
                         The
                         <em>
                          VarScan
                         </em>
                         somatic tool, which we will use to call variants in the next step, is typically used in conjunction with unadjusted base quality scores because it is widely assumed that the base quality downgrades performed by
                         <em>
                          CalMD
                         </em>
                         and other tools in the
                         <em>
                          samtools
                         </em>
                         suite of tools are too severe for
                         <em>
                          VarScan
                         </em>
                         to retain good sensitivity. In this tutorial, we will continue with this practice.
                        </p>
                        <blockquote>
                         <p dir="auto">
                          <strong>
                           Using adjusted base quality scores
                          </strong>
                         </p>
                         <ul dir="auto">
                          <li>
                           <p dir="auto">
                            If you want to experiment with adjusted base quality scores for your own data, keep in mind that
                            <em>
                             VarScan somatic
                            </em>
                            will only use them if they are incorporated directly into the read base qualities of a BAM input dataset, not if they are written to the dataset separately. Hence, should you ever decide to use:
                           </p>
                           <ul dir="auto">
                            <li>
                             <p dir="auto">
                              <em>
                               “Do you also want BAQ (Base Alignment Quality) scores to be calculated?”
                              </em>
                              :
                              <span>
                               <code>
                                Yes, run BAQ calculation
                               </code>
                              </span>
                             </p>
                             <p dir="auto">
                              and you want this setting to affect downstream variant calling with
                              <em>
                               VarScan somatic
                              </em>
                              , make sure you also set then:
                             </p>
                             <ul dir="auto">
                              <li>
                               <em>
                                “Use BAQ to cap read base qualities”
                               </em>
                               :
                               <span>
                                <code>
                                 Yes
                                </code>
                               </span>
                              </li>
                             </ul>
                            </li>
                           </ul>
                           <p dir="auto">
                            Please keep in mind that BAQ scores are quite expensive to calculate, so expect a significant (up to 10x!) increase in job run time when you enable it.
                           </p>
                          </li>
                         </ul>
                        </blockquote>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Additional options”
                         </em>
                         :
                         <span>
                          <code>
                           Advanced options
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           <em>
                            “Change identical bases to ‘=’“
                           </em>
                           :
                           <span>
                            <code>
                             No
                            </code>
                           </span>
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           <em>
                            “Coefficient to cap mapping quality of poorly mapped reads”
                           </em>
                           :
                           <span>
                            <code>
                             50
                            </code>
                           </span>
                          </p>
                          <p dir="auto">
                           This last option is the main reason we're using CalMD at this point. The mapping quality of reads mapped with
                           <em>
                            bwa
                           </em>
                           should be capped in this way before variant calling, according to empirical but well-established findings.
                          </p>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <blockquote>
                     <p dir="auto">
                      This will, once more, produce two new datasets, one for each of the normal and tumor data.
                     </p>
                    </blockquote>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Refilter reads based on mapping quality
                     </h2>
                     <a aria-label="Permalink: Refilter reads based on mapping quality" class="anchor" href="#refilter-reads-based-on-mapping-quality" id="user-content-refilter-reads-based-on-mapping-quality">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      CalMD
                     </strong>
                     may have set some mapping quality scores to 255 during read mapping quality recalibration. This special value is reserved for
                     <em>
                      undefined
                     </em>
                     mapping qualities and is used by the tool when a recalibrated mapping quality is less than zero. In other words, a value of 255 indicates a really bad mapping score, not a particularly good one. To remove such reads from the data, do the following:
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Eliminating reads with undefined mapping quality
                     </h3>
                     <a aria-label="Permalink: Hands-on: Eliminating reads with undefined mapping quality" class="anchor" href="#hands-on-eliminating-reads-with-undefined-mapping-quality" id="user-content-hands-on-eliminating-reads-with-undefined-mapping-quality">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        Filter BAM datasets on a variety of attributes
                       </strong>
                       Tool: with the following parameters:
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          “BAM dataset(s) to filter”
                         </em>
                         : the recalibrated datasets from the normal and the tumor tissue data; the outputs of
                         <strong>
                          CalMD
                         </strong>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         In
                         <em>
                          “Condition”
                         </em>
                         :
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         In
                         <em>
                          “1: Condition”
                         </em>
                         :
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           In
                           <em>
                            “Filter”
                           </em>
                           :
                          </p>
                          <ul dir="auto">
                           <li>
                            <em>
                             “Select BAM property to filter on”
                            </em>
                            :
                            <span>
                             <code>
                              mapQuality
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <ul dir="auto">
                     <li>
                      <em>
                       “Filter on read mapping quality (phred scale)”
                      </em>
                      :
                      <span>
                       <code>
                        &gt;=254
                       </code>
                      </span>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Variant Calling and Classification
                     </h1>
                     <a aria-label="Permalink: Variant Calling and Classification" class="anchor" href="#variant-calling-and-classification" id="user-content-variant-calling-and-classification">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We can now proceed to variant calling after we have generated a high-quality set of mapped read pairs. We use the
                     <strong>
                      VarScan somatic
                     </strong>
                     tool on the galaxy server, which is a dedicated solution for somatic variant calling.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Variant Calling and Classification
                     </h3>
                     <a aria-label="Permalink: Hands-on: Variant Calling and Classification" class="anchor" href="#hands-on-variant-calling-and-classification" id="user-content-hands-on-variant-calling-and-classification">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        VarScan somatic
                       </strong>
                       with the following parameters:
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          "Will you select a reference genome from your history or use a built-in genome?"
                         </em>
                         :
                         <span>
                          <code>
                           Use a built-in genome
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <em>
                           "reference genome"
                          </em>
                          :
                          <span>
                           <code>
                            Human: hg19
                           </code>
                          </span>
                          (or a similarly named choice)
                         </li>
                        </ul>
                        <blockquote>
                         <p dir="auto">
                          <strong>
                           Using the imported
                           <span>
                            <code>
                             hg19
                            </code>
                           </span>
                           sequence
                          </strong>
                         </p>
                         <p dir="auto">
                          In the event you have imported the
                          <span>
                           <code>
                            hg19
                           </code>
                          </span>
                          sequence as a fasta dataset into your history, you could instead:
                         </p>
                         <ul dir="auto">
                          <li>
                           <p dir="auto">
                            <em>
                             "Will you select a reference genome from your history or use a built-in genome?"
                            </em>
                            :
                            <span>
                             <code>
                              Use a genome from the history
                             </code>
                            </span>
                           </p>
                          </li>
                          <li>
                           <p dir="auto">
                            <em>
                             "reference genome"
                            </em>
                            : your imported
                            <span>
                             <code>
                              hg19
                             </code>
                            </span>
                            fasta dataset.
                           </p>
                          </li>
                         </ul>
                        </blockquote>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "aligned reads from normal sample"
                         </em>
                         : the mapped and fully post-processed normal tissue dataset; one of the two outputs of filtering the
                         <strong>
                          CalMD
                         </strong>
                         outputs
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "aligned reads from tumor sample"
                         </em>
                         : the mapped and fully post-processed tumor tissue dataset; the other output of filtering the
                         <strong>
                          CalMD
                         </strong>
                         outputs.
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Estimated purity (non-tumor content) of normal sample"
                         </em>
                         :
                         <span>
                          <code>
                           1
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Estimated purity (tumor content) of tumor sample"
                         </em>
                         :
                         <span>
                          <code>
                           0.5
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Generate separate output datasets for SNP and indel calls?"
                         </em>
                         :
                         <span>
                          <code>
                           No
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Settings for Variant Calling"
                         </em>
                         :
                         <span>
                          <code>
                           Customize settings
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           <em>
                            "Minimum base quality"
                           </em>
                           :
                           <span>
                            <code>
                             28
                            </code>
                           </span>
                          </p>
                          <p dir="auto">
                           We discovered that our sequencing data is of very high quality during the quality control step, and we chose not to downgrade base qualities during the quality scores recalibration step above, so that we can increase the base quality required at any given position without discarding too much of our data.
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           <em>
                            "Minimum mapping quality"
                           </em>
                           :
                           <span>
                            <code>
                             1
                            </code>
                           </span>
                          </p>
                          <p dir="auto">
                           We filtered our reads for those with a mapping quality of at least one during post-processing, but
                           <strong>
                            CalMD
                           </strong>
                           may have reduced some mapping qualities to zero afterwards.
                          </p>
                         </li>
                        </ul>
                        <p dir="auto">
                         Leave all other settings in this section at their default values.
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Settings for Posterior Variant Filtering"
                         </em>
                         :
                         <span>
                          <code>
                           Use default values
                          </code>
                         </span>
                        </p>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Variant annotation and reporting
                     </h1>
                     <a aria-label="Permalink: Variant annotation and reporting" class="anchor" href="#variant-annotation-and-reporting" id="user-content-variant-annotation-and-reporting">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     This tutorial will make use of variant and gene annotations from a variety of sources. The tools we'll be using in this section will handle the majority of these for us, but we'll need to import data from four different sources into Galaxy separately:
                    </p>
                    <ul dir="auto">
                     <li>
                      variant annotations from
                      <a href="https://www.cancerhotspots.org/" rel="nofollow">
                       Cancer Hotspots
                      </a>
                     </li>
                     <li>
                      variant and gene information from the
                      <a href="https://www.cancergenomeinterpreter.org/biomarkers" rel="nofollow">
                       Cancer Biomarkers database
                      </a>
                      of the Cancer Genome Interpreter (CGI) project
                     </li>
                     <li>
                      variant and gene information from the
                      <a href="https://civicdb.org/" rel="nofollow">
                       CIViC
                      </a>
                      database
                     </li>
                     <li>
                      variant annotations from
                      <a href="https://www.ncbi.nlm.nih.gov/snp" rel="nofollow">
                       dbSNP
                      </a>
                     </li>
                     <li>
                      lists of genes annotated with the keywords
                      <em>
                       proto-oncogene
                      </em>
                      or
                      <em>
                       tumor suppressor
                      </em>
                      at
                      <a href="https://www.uniprot.org/" rel="nofollow">
                       UniProt
                      </a>
                     </li>
                    </ul>
                    <p dir="auto">
                     Each of these annotation sets has been released in the public domain or under a free data license, allowing you to use them not only for this tutorial but also for other purposes.
                    </p>
                    <p dir="auto">
                     We generated a set of new data files tailored to the requirements of the workflow of this tutorial using the data downloaded from each of these sites and made them available through Zenodo, once again under a free data license.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Get data
                     </h2>
                     <a aria-label="Permalink: Get data" class="anchor" href="#get-data-1" id="user-content-get-data-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Data upload
                     </h4>
                     <a aria-label="Permalink: Hands-on: Data upload" class="anchor" href="#hands-on-data-upload-1" id="user-content-hands-on-data-upload-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Import the following
                       <strong>
                        variant annotation files
                       </strong>
                       from
                       <a href="https://zenodo.org/record/2581873" rel="nofollow">
                        Zenodo
                       </a>
                       :
                      </p>
                      <blockquote>
                       <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="https://zenodo.org/record/2581873/files/hotspots.bed
https://zenodo.org/record/2581873/files/cgi_variant_positions.bed
https://zenodo.org/record/2581873/files/01-Feb-2019-CIVic.bed
https://zenodo.org/record/2582555/files/dbsnp.b147.chr5_12_17.vcf.gz">
                        <pre class="notranslate"><code>https://zenodo.org/record/2581873/files/hotspots.bed
https://zenodo.org/record/2581873/files/cgi_variant_positions.bed
https://zenodo.org/record/2581873/files/01-Feb-2019-CIVic.bed
https://zenodo.org/record/2582555/files/dbsnp.b147.chr5_12_17.vcf.gz
</code></pre>
                       </div>
                      </blockquote>
                      <p dir="auto">
                       Make sure you select
                       <span>
                        <code>
                         bed
                        </code>
                       </span>
                       as the datatype for the first three files and
                       <span>
                        <code>
                         vcf
                        </code>
                       </span>
                       for the last file.
                      </p>
                      <p dir="auto">
                       Alternatively, add the files from a shared data library on your Galaxy server instance.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Import some
                       <strong>
                        gene-level annotation files
                       </strong>
                       from
                       <a href="https://zenodo.org/record/2581881" rel="nofollow">
                        Zenodo
                       </a>
                       :
                      </p>
                      <blockquote>
                       <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="https://zenodo.org/record/2581881/files/Uniprot_Cancer_Genes.13Feb2019.txt
https://zenodo.org/record/2581881/files/cgi_genes.txt
https://zenodo.org/record/2581881/files/01-Feb-2019-GeneSummaries.tsv">
                        <pre class="notranslate"><code>https://zenodo.org/record/2581881/files/Uniprot_Cancer_Genes.13Feb2019.txt
https://zenodo.org/record/2581881/files/cgi_genes.txt
https://zenodo.org/record/2581881/files/01-Feb-2019-GeneSummaries.tsv
</code></pre>
                       </div>
                      </blockquote>
                      <p dir="auto">
                       and make sure you select
                       <span>
                        <code>
                         tabular
                        </code>
                       </span>
                       as their datatype, or add them from the shared data library.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Download
                       <strong>
                        SnpEff functional genomic annotations
                       </strong>
                      </p>
                      <blockquote>
                       <p dir="auto">
                        <strong>
                         Shortcut
                        </strong>
                       </p>
                       <p dir="auto">
                        If the Galaxy server you're using has
                        <span>
                         <code>
                          Homo sapiens: hg19
                         </code>
                        </span>
                        as a locally installed snpEff database, you can skip this step. To see if this is the case, use the
                        <strong>
                         SnpEff eff tool
                        </strong>
                        under Genome source.
                       </p>
                      </blockquote>
                      <p dir="auto">
                       Use
                       <strong>
                        SnpEff Download tool
                       </strong>
                       to download genome annotation database
                       <span>
                        <code>
                         hg19
                        </code>
                       </span>
                       .
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Check that all new datasets' datatypes have been correctly set, and change them if necessary. You may also want to shorten the names of some of the datasets.
                      </p>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Adding annotations to the called variants
                     </h2>
                     <a aria-label="Permalink: Adding annotations to the called variants" class="anchor" href="#adding-annotations-to-the-called-variants" id="user-content-adding-annotations-to-the-called-variants">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Adding functional genomic annotations
                     </h3>
                     <a aria-label="Permalink: Adding functional genomic annotations" class="anchor" href="#adding-functional-genomic-annotations" id="user-content-adding-functional-genomic-annotations">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Certainly, not all variations are created equal. Many may be silent mutations with no effect at the amino acid level, while a few may disrupt a protein's coding sequence by introducing a premature stop codon or a frameshift. Of course, knowing which gene is affected by a variant is also important. Functional genomic annotations of this type can be added to a VCF dataset of variants with
                     <em>
                      SnpEff
                     </em>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Adding annotations with SnpEff
                     </h4>
                     <a aria-label="Permalink: Hands-on: Adding annotations with SnpEff" class="anchor" href="#hands-on-adding-annotations-with-snpeff" id="user-content-hands-on-adding-annotations-with-snpeff">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        SnpEff eff:wrench::gear:
                       </strong>
                       with the following parameters:
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          "Sequence changes (SNPs, MNPs, InDels)"
                         </em>
                         : the output of
                         <strong>
                          VarScan somatic:wrench:
                         </strong>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Input format"
                         </em>
                         :
                         <span>
                          <code>
                           VCF
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Output format"
                         </em>
                         :
                         <span>
                          <code>
                           VCF (only if input is VCF)
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Genome source"
                         </em>
                         :
                         <span>
                          <code>
                           Locally installed reference genome
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <em>
                           "Genome"
                          </em>
                          :
                          <span>
                           <code>
                            Homo sapiens: hg19
                           </code>
                          </span>
                          (or a similarly named option)
                         </li>
                        </ul>
                        <blockquote>
                         <p dir="auto">
                          <strong>
                           Using the imported
                           <span>
                            <code>
                             hg19
                            </code>
                           </span>
                           SnpEff genome database
                          </strong>
                         </p>
                         <p dir="auto">
                          If you have imported the
                          <span>
                           <code>
                            hg19
                           </code>
                          </span>
                          SnpEff genome database into your history instead:
                         </p>
                         <ul dir="auto">
                          <li>
                           <em>
                            "Genome source"
                           </em>
                           :
                           <span>
                            <code>
                             Downloaded snpEff database in your history
                            </code>
                           </span>
                           <ul dir="auto">
                            <li>
                             <em>
                              "SnpEff4.3 Genome Data"
                             </em>
                             : your imported
                             <span>
                              <code>
                               hg19
                              </code>
                             </span>
                             SnpEff dataset.
                            </li>
                           </ul>
                          </li>
                         </ul>
                        </blockquote>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Produce summary Stats"
                         </em>
                         :
                         <span>
                          <code>
                           No
                          </code>
                         </span>
                        </p>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Adding genetic and clinical evidence-base annotations
                     </h3>
                     <a aria-label="Permalink: Adding genetic and clinical evidence-base annotations" class="anchor" href="#adding-genetic-and-clinical-evidence-base-annotations" id="user-content-adding-genetic-and-clinical-evidence-base-annotations">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Other interesting aspects of a variant include whether the variant has previously been observed in the human population and, if so, at what frequency. We would also like to know if a variant is known to be associated with specific diseases. To move forward with this type of genetic and clinical evidence-based annotations, we will convert our list of variants into a database that can be processed more efficiently than a VCF dataset. We will use the GEMINI tool suite for this task as well as any future work with variants.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Creating a GEMINI database from a variants dataset
                     </h4>
                     <a aria-label="Permalink: Hands-on: Creating a GEMINI database from a variants dataset" class="anchor" href="#hands-on-creating-a-gemini-database-from-a-variants-dataset" id="user-content-hands-on-creating-a-gemini-database-from-a-variants-dataset">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        GEMINI load 🔧⚙️
                       </strong>
                       with the following parameters:
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          "VCF dataset to be loaded in the GEMINI database"
                         </em>
                         : the output of
                         <strong>
                          SnpEff eff:wrench:
                         </strong>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "The variants in this input are"
                         </em>
                         :
                         <span>
                          <code>
                           annotated with snpEff
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "This input comes with genotype calls for its samples"
                         </em>
                         :
                         <span>
                          <code>
                           Yes
                          </code>
                         </span>
                        </p>
                        <p dir="auto">
                         Calling sample genotypes was part of what we used VarScan somatic for.
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Choose a gemini annotation source"
                         </em>
                         : select the latest available annotations snapshot (most likely, there will be only one)
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Sample and family information in PED format"
                         </em>
                         :
                         <span>
                          <code>
                           Nothing selected
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Load the following optional content into the database"
                         </em>
                        </p>
                        <ul dir="auto">
                         <li>
                          <ul class="contains-task-list">
                           <li class="task-list-item">
                            <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                            "GERP scores"*
                           </li>
                          </ul>
                         </li>
                         <li>
                          <ul class="contains-task-list">
                           <li class="task-list-item">
                            <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                            <em>
                             "CADD scores"
                            </em>
                           </li>
                          </ul>
                         </li>
                         <li>
                          <ul class="contains-task-list">
                           <li class="task-list-item">
                            <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                            <em>
                             "Gene tables"
                            </em>
                           </li>
                          </ul>
                         </li>
                         <li>
                          <ul class="contains-task-list">
                           <li class="task-list-item">
                            <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                            <em>
                             "Sample genotypes"
                            </em>
                           </li>
                          </ul>
                         </li>
                         <li>
                          <ul class="contains-task-list">
                           <li class="task-list-item">
                            <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                            <em>
                             "variant INFO field"
                            </em>
                           </li>
                          </ul>
                         </li>
                        </ul>
                        <p dir="auto">
                         Be careful to leave
                         <strong>
                          unchecked
                         </strong>
                         :
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           <em>
                            "Genotype likelihoods (sample PLs)"
                           </em>
                          </p>
                          <p dir="auto">
                           VarScan somatic does not generate these values
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           <em>
                            "only variants that passed all filters"
                           </em>
                          </p>
                          <p dir="auto">
                           It is simple and more flexible to filter for this property later
                          </p>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <p dir="auto">
                     GEMINI already (silently) adds an impressive number of annotations it knows about to our variants during the database's creation (including, e.g., the frequency at which every variant has been observed in large human genome sequencing projects). All of these were obtained for free simply by converting the variants to a GEMINI database!
                    </p>
                    <p dir="auto">
                     GEMINI also extracts a significant amount of information from the VCF input dataset for us (such as the functional genomic annotations that we added with SnpEff).
                    </p>
                    <p dir="auto">
                     However, there are often additional annotations from other sources (not included in GEMINI) that would be useful to include. Furthermore, GEMINI is not equipped to extract some non-standard information from VCF datasets, such as some critical bits added by VarScan somatic.
                    </p>
                    <p dir="auto">
                     The
                     <strong>
                      GEMINI annotate tool
                     </strong>
                     is intended to assist you with these rather common issues. It allows you to add additional annotations to variants in an already loaded GEMINI database.
                    </p>
                    <p dir="auto">
                     As a first step, we will use the tool to add to the database some critical information generated by VarScan somatic but not recognized by GEMINI load. We are particularly interested in three values calculated by VarScan somatic for each detected variant:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <em>
                        Somatic status (SS)
                       </em>
                      </p>
                      <p dir="auto">
                       This is a simple numeric code in which a value of
                       <span>
                        <code>
                         1
                        </code>
                       </span>
                       indicates a germline variant, a value of
                       <span>
                        <code>
                         2
                        </code>
                       </span>
                       indicates a somatic variant, and a value of
                       <span>
                        <code>
                         3
                        </code>
                       </span>
                       indicates a LOH event.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <em>
                        Germline p-value (GPV)
                       </em>
                      </p>
                      <p dir="auto">
                       This is the error probability associated with variants that have a somatic status of 1.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <em>
                        Somatic p-value (SPV)
                       </em>
                      </p>
                      <p dir="auto">
                       This is the error probability for status calls of 2 and 3. (somatic and LOH calls).
                      </p>
                     </li>
                    </ul>
                    <p dir="auto">
                     These values are encoded in the INFO column of the VarScan-generated VCF dataset, which we will extract and add to the GEMINI database.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Making variant call statistics accessible
                     </h4>
                     <a aria-label="Permalink: Hands-on: Making variant call statistics accessible" class="anchor" href="#hands-on-making-variant-call-statistics-accessible" id="user-content-hands-on-making-variant-call-statistics-accessible">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        GEMINI annotate:wrench::gear:
                       </strong>
                       :
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          "GEMINI database"
                         </em>
                         : output of
                         <strong>
                          GEMINI load
                         </strong>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Dataset to use as the annotation source"
                         </em>
                         : output of
                         <strong>
                          VarScan somatic
                         </strong>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Strict variant-identity matching of database and annotation records (VCF format only)"
                         </em>
                         :
                         <span>
                          <code>
                           Yes
                          </code>
                         </span>
                        </p>
                        <p dir="auto">
                         This setting is unimportant in this case because you built the GEMINI database from the same set of variants from which we are now retrieving annotations and because VarScan somatic does not call multiple alleles at a single site. However, matching on variant-identity is the desired behavior, so we might as well be explicit about it.
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Type of information to add to the database variants"
                         </em>
                         :
                         <span>
                          <code>
                           Specific values extracted from matching records in the annotation source (extract)
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           In
                           <em>
                            "1: Annotation extraction recipe"
                           </em>
                           :
                          </p>
                          <ul dir="auto">
                           <li>
                            <em>
                             "Elements to extract from the annotation source"
                            </em>
                            :
                            <span>
                             <code>
                              SS
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "Database column name to use for recording annotations"
                            </em>
                            :
                            <span>
                             <code>
                              somatic_status
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "What type of data are you trying to extract?"
                            </em>
                            :
                            <span>
                             <code>
                              Integer numbers
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "If multiple annotations are found for the same variant, store..."
                            </em>
                            :
                            <span>
                             <code>
                              the first annotation found
                             </code>
                            </span>
                           </li>
                          </ul>
                          <p dir="auto">
                           This is the recipe for extracting the VarScan-generated
                           <em>
                            SS
                           </em>
                           field and adding it to the GEMINI database as a new column
                           <em>
                            somatic status
                           </em>
                           .
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           ➕
                           <em>
                            "Insert Annotation extraction recipe"
                           </em>
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           In
                           <em>
                            "2: Annotation extraction recipe"
                           </em>
                           :
                          </p>
                          <ul dir="auto">
                           <li>
                            <em>
                             "Elements to extract from the annotation source"
                            </em>
                            :
                            <span>
                             <code>
                              GPV
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "Database column name to use for recording annotations"
                            </em>
                            :
                            <span>
                             <code>
                              germline_p
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "What type of data are you trying to extract?"
                            </em>
                            :
                            <span>
                             <code>
                              Numbers with decimal precision
                             </code>
                            </span>
                           </li>
                           <li>
                            *"If multiple annotations are found for the same variant, store...":
                            <span>
                             <code>
                              the first annotation found
                             </code>
                            </span>
                           </li>
                          </ul>
                          <p dir="auto">
                           This is the recipe for extracting the VarScan-generated
                           <em>
                            GPV
                           </em>
                           field and adding it to the GEMINI database as a new column
                           <em>
                            germline_p
                           </em>
                           .
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           ➕
                           <em>
                            "Insert Annotation extraction recipe"
                           </em>
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           In
                           <em>
                            "3: Annotation extraction recipe"
                           </em>
                           :
                          </p>
                          <ul dir="auto">
                           <li>
                            <em>
                             "Elements to extract from the annotation source"
                            </em>
                            :
                            <span>
                             <code>
                              SPV
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "Database column name to use for recording annotations"
                            </em>
                            :
                            <span>
                             <code>
                              somatic_p
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "What type pf data are you trying to extract"
                            </em>
                            :
                            <span>
                             <code>
                              Numbers with decimal precision
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "If multiple annotations are found for the same variant, store..."
                            </em>
                            :
                            <span>
                             <code>
                              the first annotation found
                             </code>
                            </span>
                           </li>
                          </ul>
                          <p dir="auto">
                           This is the recipe for extracting the VarScan-generated
                           <em>
                            SPV
                           </em>
                           field and adding it to the GEMINI database as a new column
                           <em>
                            somatic_p
                           </em>
                           .
                          </p>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <p dir="auto">
                     Then, in addition to the annotations obtained directly from GEMINI or the input VCF dataset, we will add additional annotations. We would like to specifically add:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       more information from
                       <strong>
                        dbSNP
                       </strong>
                      </p>
                      <p dir="auto">
                       GEMINI already checks all variants to see if they exist in dbSNP and, if so, stores their dbSNP IDs as part of the database creation process. As a result, we only need to extract some additional relevant information.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       information from
                       <strong>
                        Cancer Hotspots
                       </strong>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       links to the
                       <strong>
                        CiViC
                       </strong>
                       database
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       information from the
                       <strong>
                        Cancer Genome Interpreter (CGI)
                       </strong>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Adding further annotations
                     </h4>
                     <a aria-label="Permalink: Hands-on: Adding further annotations" class="anchor" href="#hands-on-adding-further-annotations" id="user-content-hands-on-adding-further-annotations">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        GEMINI annotate:wrench::gear:
                       </strong>
                       to add further annotations from
                       <strong>
                        dbSNP
                       </strong>
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          "GEMINI database"
                         </em>
                         : the output of the last
                         <strong>
                          GEMINI annotate:wrench:
                         </strong>
                         run
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Dataset to use as the annotation source"
                         </em>
                         : the imported
                         <span>
                          <code>
                           dbsnp.b147.chr5_12_17.vcf
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Strict variant-identity matching of database and annotation records (VCF format only)"
                         </em>
                         :
                         <span>
                          <code>
                           Yes
                          </code>
                         </span>
                        </p>
                        <p dir="auto">
                         dbSNP stores information about specific SNPs found in human populations, and we'd like to see if any of the same SNPs are found among our variants.
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Type of information to add to the database variants"
                         </em>
                         :
                         <span>
                          <code>
                           Specific values extracted from matching recors in the annotation source (extract)
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           In
                           <em>
                            "1: Annotation extraction recipe"
                           </em>
                           :
                          </p>
                          <ul dir="auto">
                           <li>
                            <em>
                             "Elements to extract from the annotation source"
                            </em>
                            :
                            <span>
                             <code>
                              SAO
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "Databse column name to use for recording annotations"
                            </em>
                            :
                            <span>
                             <code>
                              rs_ss
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "What type of data are you trying to extract?"
                            </em>
                            :
                            <span>
                             <code>
                              Integer numbers
                             </code>
                            </span>
                           </li>
                           <li>
                            <em>
                             "If multiple annotations are found for the same variant, store..."
                            </em>
                            :
                            <span>
                             <code>
                              the first annotation found
                             </code>
                            </span>
                           </li>
                          </ul>
                          <p dir="auto">
                           This recipe extracts the dbSNP field and adds it as
                           <em>
                            rs_ss
                           </em>
                           to the GEMINI database.
                          </p>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        GEMINI annotate:wrench::gear:
                       </strong>
                       to add further annotations from
                       <strong>
                        cancerhotspots
                       </strong>
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          "GEMINI database"
                         </em>
                         : the output of the last
                         <strong>
                          GEMINI annotate:wrench:
                         </strong>
                         run
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Dataset to use as the annotation source"
                         </em>
                         : the imported
                         <span>
                          <code>
                           cancerhotspots_v2.bed
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Strict variant-identity matching of database and annotation records (VCF format only)"
                         </em>
                         :
                         <span>
                          <code>
                           Yes
                          </code>
                         </span>
                         (with input in BED format this settign will be ignored)
                        </p>
                        <p dir="auto">
                         We will simply record the best q-value associated with any cancerhotspots variant that overlaps one of our variant sites for the cancerhotspots data.
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Type of information to add to the database variants"
                         </em>
                         :
                         <span>
                          <code>
                           Specific values extracted from matching records in the annotation source (extract)
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           In
                           <em>
                            "1: Annotation extraction recipe"
                           </em>
                           :
                          </p>
                          <ul dir="auto">
                           <li>
                            <p dir="auto">
                             <em>
                              "Elements to extract from the annotation source"
                             </em>
                             :
                             <span>
                              <code>
                               5
                              </code>
                             </span>
                            </p>
                            <p dir="auto">
                             The q-values are stored in the fifth column of the BED dataset.
                            </p>
                           </li>
                           <li>
                            <p dir="auto">
                             <em>
                              "Database column name to use for recording annotations"
                             </em>
                             :
                             <span>
                              <code>
                               hs_qvalue
                              </code>
                             </span>
                            </p>
                           </li>
                           <li>
                            <p dir="auto">
                             <em>
                              "What type of data are you trying to extract?"
                             </em>
                             :
                             <span>
                              <code>
                               Numbers with decimal precision
                              </code>
                             </span>
                            </p>
                           </li>
                           <li>
                            <p dir="auto">
                             <em>
                              "If multiple annotations are found for the same variant, store..."
                             </em>
                             :
                             <span>
                              <code>
                               the smallest of the (numeric) values
                              </code>
                             </span>
                            </p>
                           </li>
                          </ul>
                          <p dir="auto">
                           This is the procedure for extracting the q-values of overlapping cancerhotspots sites and adding them to the GEMINI database as hs qvalue.
                          </p>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        GEMINI annotate:wrench::gear:
                       </strong>
                       to add links to
                       <strong>
                        CiViC
                       </strong>
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          "GEMINI database"
                         </em>
                         : the output of the last
                         <strong>
                          GEMINI annotate:wrench:
                         </strong>
                         run
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         *"Dataset to use as the annotation source"**: the imported
                         <span>
                          <code>
                           CiViC.bed
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Strict variant-identity matching of database and annotation records (VCF format only)"
                         </em>
                         :
                         <span>
                          <code>
                           Yes
                          </code>
                         </span>
                         (with input in BED format this setting will be ignored)
                        </p>
                        <p dir="auto">
                         We will record the hyperlink to any variant found in the CIViC database that overlaps one of our variant sites for the CIViC data.
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Type of information to add to the database variants"
                         </em>
                         :
                         <span>
                          <code>
                           Specific values extracted from matching records in the annotation source (extract)
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           In
                           <em>
                            "1: Annotation extraction recipe"
                           </em>
                           :
                          </p>
                          <ul dir="auto">
                           <li>
                            <p dir="auto">
                             <em>
                              "Elements to extract from the annotation source"
                             </em>
                             :
                             <span>
                              <code>
                               4
                              </code>
                             </span>
                            </p>
                            <p dir="auto">
                             The hyperlinks are stored in the fourth column of the BED dataset
                            </p>
                           </li>
                           <li>
                            <p dir="auto">
                             <em>
                              "Database column name to use for recording annotations"
                             </em>
                             :
                             <span>
                              <code>
                               overlapping_civic_url
                              </code>
                             </span>
                            </p>
                           </li>
                           <li>
                            <p dir="auto">
                             <em>
                              "What type of data are you trying to extract?"
                             </em>
                             :
                             <span>
                              <code>
                               Text (text)
                              </code>
                             </span>
                            </p>
                           </li>
                           <li>
                            <p dir="auto">
                             <em>
                              "If multiple annotations are found for the same variant, store..."
                             </em>
                             :
                             <span>
                              <code>
                               a comma-separated list of non-redundant (text) values
                              </code>
                             </span>
                            </p>
                            <p dir="auto">
                             This is the recipe for extracting the hyperlinks of overlapping CIViC sites and adding them to the GEMINI database as a list of overlapping civic urls.
                            </p>
                           </li>
                          </ul>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        GEMINI annotate:wrench::gear:
                       </strong>
                       to add information from the
                       <strong>
                        Cancer Genome Interpreter (CGI)
                       </strong>
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          "GEMINI database"
                         </em>
                         : the output of the last
                         <strong>
                          GEMINI annotate:wrench:
                         </strong>
                         run
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Dataset to use as the annotation source"
                         </em>
                         : the imported
                         <span>
                          <code>
                           cgi_variant)positions.bed
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Strict variant-identity matching of database and annotation records (VCF format only)"
                         </em>
                         :
                         <span>
                          <code>
                           Yes
                          </code>
                         </span>
                         (with input in BED format this setting will be ignored)
                        </p>
                        <p dir="auto">
                         For the CGI data, we will note whether any of our variant sites overlap with a variant in the CGI biomarkers database.
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Type of information to add to the database variants"
                         </em>
                         :
                         <span>
                          <code>
                           Binary indicator (1=found, 0=not found) of whether the variant had any match in the annotation source (boolean)
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <em>
                           "Databse column name to use for ecording annotations"
                          </em>
                          :
                          <span>
                           <code>
                            in_cgidb
                           </code>
                          </span>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Reporting selected subsets of variants
                     </h2>
                     <a aria-label="Permalink: Reporting selected subsets of variants" class="anchor" href="#reporting-selected-subsets-of-variants" id="user-content-reporting-selected-subsets-of-variants">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Now that we've built our GEMINI database and added annotations to it, it's time to explore the wealth of information it contains. The goal is to generate filtered variant reports that include specific classes (somatic, germline, and LOH) of high-quality variants as well as their most relevant annotations. This is accomplished through GEMINI queries that specify:
                    </p>
                    <ol dir="auto">
                     <li>
                      filters we want to apply to the variant list stored in the database
                     </li>
                     <li>
                      the pieces of information about the filtered variants that we would like to retrieve
                     </li>
                    </ol>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       The GEMINI query language
                      </strong>
                     </p>
                     <p dir="auto">
                      GEMINI queries are extremely flexible, allowing users to express a wide range of ideas in order to explore the variant space; however, the complete query syntax can be intimidating for beginners.
                     </p>
                     <p dir="auto">
                      In this section, we'll start with a simple query and work our way up to a more complex query in the following section.
                     </p>
                     <p dir="auto">
                      Consult the
                      <a href="https://gemini.readthedocs.io/en/latest/content/querying.html" rel="nofollow">
                       query section of the GEMINI documentation
                      </a>
                      for a more detailed explanation of the query syntax. Because the GEMINI query syntax is based on the SQLite SQL dialect, the SQLite documentation, particularly its chapter on
                      <a href="https://sqlite.org/lang_corefunc.html" rel="nofollow">
                       SQLite core functions
                      </a>
                      , is an excellent resource for understanding more complex queries.
                     </p>
                    </blockquote>
                    <p dir="auto">
                     Let's begin by configuring a simple query to generate a list of genuine somatic variants. Our plan for retrieving them is as follows:
                    </p>
                    <ol dir="auto">
                     <li>
                      rely on the
                      <em>
                       somatic status
                      </em>
                      of the variants called by
                      <strong>
                       VarScan somatic:wrench:
                      </strong>
                     </li>
                     <li>
                      Discard any variants for which a significant number of supporting sequencing reads are also found in normal tissue data, or which are only supported by a very small fraction of the reads from the tumor sample.
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Querying the GEMINI database for somatic variants
                     </h4>
                     <a aria-label="Permalink: Hands-on: Querying the GEMINI database for somatic variants" class="anchor" href="#hands-on-querying-the-gemini-database-for-somatic-variants" id="user-content-hands-on-querying-the-gemini-database-for-somatic-variants">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        GEMINI query:wrench::gear:
                       </strong>
                       with:
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          "GEMINI database"
                         </em>
                         : the fully annotated database created in the last
                         <strong>
                          GEMINI annotate:wrench:
                         </strong>
                         step
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Build GEMINI query using"
                         </em>
                         :
                         <span>
                          <code>
                           Basic variant query constructor
                          </code>
                         </span>
                        </p>
                        <ul dir="auto">
                         <li>
                          <p dir="auto">
                           ➕
                           <em>
                            "Insert Genotype filter expression"
                           </em>
                          </p>
                          <ul dir="auto">
                           <li>
                            <em>
                             "Restrictions to apply to gentype values"
                            </em>
                            :
                            <span>
                             <code>
                              gt_alt_freqs.NORMAL &lt;= 0.05 AND gt_alt_freqs.TUMOR &gt;= 0.10
                             </code>
                            </span>
                           </li>
                          </ul>
                          <p dir="auto">
                           We retain only those variants supported by less than 5% of the reads in the normal sample but more than 10% of the reads in the tumor sample using this genotype-based filter.
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           <em>
                            "Additional constraints expressed in SQL syntax"
                           </em>
                           :
                           <span>
                            <code>
                             somatic_status = 2
                            </code>
                           </span>
                          </p>
                          <p dir="auto">
                           VarScan somatic has called for each variant's somatic status, which is stored in the GEMINI database (remember we used GEMINI annotate to add it). With the condition
                           <span>
                            <code>
                             somatic_status = 2
                            </code>
                           </span>
                           , we keep only the variants that passed the genotype filter above
                           <strong>
                            and
                           </strong>
                           were classified as somatic by the variant caller.
                          </p>
                         </li>
                         <li>
                          <p dir="auto">
                           In
                           <em>
                            "Output format options"
                           </em>
                          </p>
                          <ul dir="auto">
                           <li>
                            <em>
                             "Type of report to generate"
                            </em>
                            :
                            <span>
                             <code>
                              tabular (GEMINI default)
                             </code>
                            </span>
                            <ul dir="auto">
                             <li>
                              <em>
                               "Add a header of column names to the output"
                              </em>
                              :
                              <span>
                               <code>
                                Yes
                               </code>
                              </span>
                             </li>
                             <li>
                              <em>
                               "Set of columns to include in the variant report table"
                              </em>
                              :
                              <span>
                               <code>
                                Custom (report user-specified columns)
                               </code>
                              </span>
                              <ul dir="auto">
                               <li>
                                In
                                <em>
                                 "Choose columns to include in the report"
                                </em>
                                :
                                <ul dir="auto">
                                 <li>
                                  <ul class="contains-task-list">
                                   <li class="task-list-item">
                                    <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                                    <em>
                                     "chrom"
                                    </em>
                                   </li>
                                  </ul>
                                 </li>
                                 <li>
                                  <ul class="contains-task-list">
                                   <li class="task-list-item">
                                    <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                                    <em>
                                     "start"
                                    </em>
                                   </li>
                                  </ul>
                                 </li>
                                 <li>
                                  <ul class="contains-task-list">
                                   <li class="task-list-item">
                                    <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                                    <em>
                                     "ref"
                                    </em>
                                   </li>
                                  </ul>
                                 </li>
                                 <li>
                                  <ul class="contains-task-list">
                                   <li class="task-list-item">
                                    <input checked="" class="task-list-item-checkbox" disabled="" id="" type="checkbox"/>
                                    <em>
                                     "alt"
                                    </em>
                                   </li>
                                  </ul>
                                 </li>
                                </ul>
                               </li>
                               <li>
                                <em>
                                 "Additional columns (comma-separated)"
                                </em>
                                :
                                <span>
                                 <code>
                                  gene, aa_change, rs_ids, hs_qvalue, cosmic_ids
                                 </code>
                                </span>
                               </li>
                              </ul>
                             </li>
                            </ul>
                           </li>
                          </ul>
                          <p dir="auto">
                           In this section, we specify which columns (from the GEMINI database's variants table) we want to include in a
                           <span>
                            <code>
                             tabular
                            </code>
                           </span>
                           variant report, in the order specified.
                          </p>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       How am I supposed to know these column names?
                      </strong>
                     </p>
                     <p dir="auto">
                      Obviously, you must know the names of the columns (in the GEMINI database tables) in order to include them in the report, but how are you supposed to know them?
                     </p>
                     <p dir="auto">
                      The
                      <a href="https://gemini.readthedocs.io/en/latest/content/database_schema.html" rel="nofollow">
                       GEMINI documentation
                      </a>
                      lists the standard ones (added by GEMINI load when building the database). The non-standard columns (the ones you added with GEMINI annotate) are labeled with the names you assigned to them when you added them.
                     </p>
                     <p dir="auto">
                      Alternatively, to obtain the tables and column names of a specific database, use the
                      <strong>
                       GEMINI database info tool
                      </strong>
                      as follows:
                     </p>
                     <ul dir="auto">
                      <li>
                       <em>
                        “GEMINI database”
                       </em>
                       : the database you want to explore
                      </li>
                      <li>
                       <em>
                        “Information to retrieve from the database”
                       </em>
                       :
                       <span>
                        <code>
                         Names of database tables and their columns
                        </code>
                       </span>
                      </li>
                     </ul>
                    </blockquote>
                    <p dir="auto">
                     What about more sophisticated filtering?
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: More complex filter criteria
                     </h4>
                     <a aria-label="Permalink: Hands-on: More complex filter criteria" class="anchor" href="#hands-on-more-complex-filter-criteria" id="user-content-hands-on-more-complex-filter-criteria">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        GEMINI query:wrench::gear:
                       </strong>
                       with the exact same settings as before, but:
                      </p>
                      <ul dir="auto">
                       <li>
                        <em>
                         "Additional constraints expressed in SQL syntax"
                        </em>
                        :
                        <span>
                         <code>
                          somatic_status = 2 AND somatic_p &lt;= 0.05 AND filter IS NULL
                         </code>
                        </span>
                       </li>
                      </ul>
                      <p dir="auto">
                       This translates to "variants classified as somatic with a p-value &lt;= 0.05, but not flagged as likely false-positives".
                      </p>
                     </li>
                    </ol>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       SQL keywords
                      </strong>
                     </p>
                     <p dir="auto">
                      SQL keywords are given in uppercase in the condition above. This is not required, but it makes the syntax easier to understand.
                     </p>
                     <p dir="auto">
                      <span>
                       <code>
                        IS NULL
                       </code>
                      </span>
                      can be used to determine whether a cell in a data table is empty or contains a value.
                      <span>
                       <code>
                        IS NOT NULL
                       </code>
                      </span>
                      can be used to determine whether a cell in a data table contains a value. You can use
                      <span>
                       <code>
                        AND
                       </code>
                      </span>
                      and
                      <span>
                       <code>
                        OR
                       </code>
                      </span>
                      , as well as parentheses to group conditions, to logically combine different filter criteria.
                     </p>
                    </blockquote>
                    <p dir="auto">
                     If you followed all of the steps exactly, running this job should result in a tabular dataset of 43 variants, and with the annotations in the report, it should be relatively easy to pick out a few interesting ones. However, before we get into the report's content, we could improve the report's format a little more.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: SQL-based output formatting
                     </h4>
                     <a aria-label="Permalink: Hands-on: SQL-based output formatting" class="anchor" href="#hands-on-sql-based-output-formatting" id="user-content-hands-on-sql-based-output-formatting">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Run
                      <strong>
                       GEMINI query:wrench::gear:
                      </strong>
                      with the exact same settings as in the last example, but:
                      <ul dir="auto">
                       <li>
                        In
                        <em>
                         "Output format options"
                        </em>
                        <ul dir="auto">
                         <li>
                          <em>
                           "Additional columns (comma-separated)"
                          </em>
                          :
                          <span>
                           <code>
                            type, gt_alt_freqs.TUMOR, gt_alt_freqs.NORMAL, ifnull(nullif(round(max_aaf_all,2),-1.0),0) AS MAD, gene, impact_so, aa_change, ifnull(round(cadd_scaled,2), '.') AS cadd_scaled, round(gerp_bp_score,2) AS gerp_bp, ifnull(round(gerp_element_pval,2),'.') AS gerp_elemt_pval, ifnull(round(hs_qvalue,2),'.') AS hs_qvalue, in_omim, ifnull(clinvar_sig,'.') AS clinvar, ifnull(clinvar_disease_name,'.') AS clinvar_disease_name, ifnull(rs_ids,'.') AS dbsnp_ids, rs_ss, ifnull(cosmic_ids,'.') AS cosmic_ids, ifnull(overlapping_civic_url,'.') AS overlapping_civic_url, in_cgidb
                           </code>
                          </span>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <p dir="auto">
                     This final query adds a lot more annotations to the report and also shows how to use the
                     <span>
                      <code>
                       AS
                      </code>
                     </span>
                     keyword to rename columns and some
                     <a href="https://sqlite.org/lang_corefunc.html" rel="nofollow">
                      SQLite functions
                     </a>
                     to clean up the output.
                    </p>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Limitations of genotype column queries
                      </strong>
                     </p>
                     <p dir="auto">
                      If you're wondering why the above query doesn't use rounding on the samples' alternate allele frequencies, i.e.,
                      <span>
                       <code>
                        gt_alt_freqs.TUMOR
                       </code>
                      </span>
                      and
                      <span>
                       <code>
                        gt_alt_freqs.NORMAL
                       </code>
                      </span>
                      , or why it doesn't rename these columns, it's because it would break the query.
                     </p>
                     <p dir="auto">
                      As a general rule, all columns in the variants table beginning with
                      <span>
                       <code>
                        gt
                       </code>
                      </span>
                      (the genotype columns calculated from the genotype fields of a VCF dataset) cannot be used as regular SQLite columns and must be parsed separately by GEMINI. As a result, you cannot mix them with regular SQLite elements such as functions and alias specifications when using
                      <span>
                       <code>
                        AS
                       </code>
                      </span>
                      . You may have also noticed that
                      <span>
                       <code>
                        gt_alt_freqs.TUMOR
                       </code>
                      </span>
                      and
                      <span>
                       <code>
                        gt_alt_freqs.NORMAL
                       </code>
                      </span>
                      columns do not follow the query's column order specification and end up as the last columns of the tabular report. This is yet another example of GEMINI's unique treatment of
                      <span>
                       <code>
                        gt
                       </code>
                      </span>
                      columns.
                     </p>
                    </blockquote>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Generating reports of genes affected by variants
                     </h2>
                     <a aria-label="Permalink: Generating reports of genes affected by variants" class="anchor" href="#generating-reports-of-genes-affected-by-variants" id="user-content-generating-reports-of-genes-affected-by-variants">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Finally, using the same somatic variants we picked earlier, let's try to construct a gene-centered report.
                    </p>
                    <p dir="auto">
                     Annotations that pertain to a full gene affected by a variant rather than the variant itself would be included in a gene-centered report. Known synonyms of an afflicted gene, its NCBI entrez number, the ClinVar phenotype linked with the gene, if any, a reference to the gene's page at CIViC.org, and so on are examples of such annotations.
                    </p>
                    <p dir="auto">
                     Some of this data is included in every GEMINI database, but it's stored in a distinct table called
                     <span>
                      <code>
                       gene_detailed
                      </code>
                     </span>
                     , whereas we've only utilized and queried data from the
                     <span>
                      <code>
                       variants
                      </code>
                     </span>
                     table thus far.
                    </p>
                    <p dir="auto">
                     We must join the
                     <span>
                      <code>
                       variants
                      </code>
                     </span>
                     and
                     <span>
                      <code>
                       gene_detailed
                      </code>
                     </span>
                     tables in order to retrieve information from both tables in the same query. Such an operation is not possible with the
                     <span>
                      <code>
                       Basic query constructor
                      </code>
                     </span>
                     we've been using so far, and needs building the query in advanced mode.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Turning query results into gene-centered reports
                     </h4>
                     <a aria-label="Permalink: Hands-on: Turning query results into gene-centered reports" class="anchor" href="#hands-on-turning-query-results-into-gene-centered-reports" id="user-content-hands-on-turning-query-results-into-gene-centered-reports">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Run
                       <strong>
                        GEMINI query:wrench::gear:
                       </strong>
                       in advanced mode by choosing
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          "Build GEMINI query using"
                         </em>
                         :
                         <span>
                          <code>
                           Advanced query constructor
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "The query to be issued to the database"
                         </em>
                         :
                         <span>
                          <code>
                           SELECT v.gene, v.chrom, g.synonym, g.hgnc_id, g.entrez_id, g.rvis_pct, v.clinvar_gene_phenotype FROM variants v, gene_detailed g WHERE v.chrom = g.chrom AND v.gene = g.gene AND v.somatic_status = 2 AND v.somatic_p &lt;= 0.05 AND v.filter IS NULL GROUP BY g.gene
                          </code>
                         </span>
                        </p>
                        <blockquote>
                         <p dir="auto">
                          <strong>
                           Elements of the SQL query
                          </strong>
                         </p>
                         <p dir="auto">
                          The part between
                          <span>
                           <code>
                            SELECT
                           </code>
                          </span>
                          and
                          <span>
                           <code>
                            FROM
                           </code>
                          </span>
                          specifies which columns from which database tables we want to retrieve, while the part between
                          <span>
                           <code>
                            FROM
                           </code>
                          </span>
                          and
                          <span>
                           <code>
                            WHERE
                           </code>
                          </span>
                          specifies the database tables that need to be consulted and gives them simpler aliases (
                          <span>
                           <code>
                            v
                           </code>
                          </span>
                          becomes an alias for the
                          <span>
                           <code>
                            variants
                           </code>
                          </span>
                          table,
                          <span>
                           <code>
                            g
                           </code>
                          </span>
                          for the
                          <span>
                           <code>
                            gene_detailed
                           </code>
                          </span>
                          table), which we can then use throughout the query.
                         </p>
                         <p dir="auto">
                          The filter criteria we want to use are listed after
                          <span>
                           <code>
                            WHERE
                           </code>
                          </span>
                          . These criteria are nearly identical to those used in our previous somatic variants query, but because we are working with two tables instead of just one, we must specify which table the filter columns come from using table prefixes. Hence,
                          <span>
                           <code>
                            somatic_status
                           </code>
                          </span>
                          becomes
                          <span>
                           <code>
                            v.somatic_status
                           </code>
                          </span>
                          , and so on. Furthermore, we want to report corresponding information from the two tables, which is ensured by the additional criteria
                          <span>
                           <code>
                            v.chrom = g.chrom
                           </code>
                          </span>
                          and
                          <span>
                           <code>
                            v.gene = g.gene
                           </code>
                          </span>
                          (the SQL terminology is: we want to join the
                          <span>
                           <code>
                            variants
                           </code>
                          </span>
                          and
                          <span>
                           <code>
                            gene_detailed
                           </code>
                          </span>
                          tables on their
                          <span>
                           <code>
                            chrom
                           </code>
                          </span>
                          and
                          <span>
                           <code>
                            gene
                           </code>
                          </span>
                          columns).
                         </p>
                         <p dir="auto">
                          Finally, the
                          <span>
                           <code>
                            GROUP BY
                           </code>
                          </span>
                          clause specifies that we want to combine records affecting the same gene into one.
                         </p>
                        </blockquote>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          "Genotype filter expression"
                         </em>
                         :
                         <span>
                          <code>
                           gt_alt_freqs.NORMAL &lt;= 0.05 AND gt_alt_freqs.TUMOR &gt;= 0.10
                          </code>
                         </span>
                        </p>
                        <p dir="auto">
                         This remains unchanged from the previous somatic variants query.
                        </p>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Adding additional annotations to the gene-centered report
                     </h2>
                     <a aria-label="Permalink: Adding additional annotations to the gene-centered report" class="anchor" href="#adding-additional-annotations-to-the-gene-centered-report" id="user-content-adding-additional-annotations-to-the-gene-centered-report">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Unfortunately, GEMINI annotate only allows you to add columns to a GEMINI database's variants table; there is no easy way to enrich the
                     <span>
                      <code>
                       gene_detailed
                      </code>
                     </span>
                     table with additional annotations. That is why, using more general-purpose Galaxy tools, we will now add such additional annotations to the tabular gene-centered report.
                    </p>
                    <p dir="auto">
                     Annotating the GEMINI tabular gene report is a two-step process in which we first combine the report and tabular annotation sources into a larger tabular dataset, from which we then remove redundant and unwanted columns while
                     <em>
                      rearranging
                     </em>
                     the remaining ones.
                    </p>
                    <p dir="auto">
                     <strong>
                      Step 1
                     </strong>
                     consists of three separate
                     <em>
                      join
                     </em>
                     operations that pull in the annotations found in the three gene-based tabular datasets that you imported in the
                     <em>
                      Get Data
                     </em>
                     step of this section in a sequential fashion. To to this, we used the
                     <strong>
                      Join two file
                     </strong>
                     tool to add the information such as
                     <strong>
                      UniProt cancer genes
                     </strong>
                     ,
                     <strong>
                      CGI biomarkers
                     </strong>
                     and
                     <strong>
                      CIViC
                     </strong>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Join
                     </h4>
                     <a aria-label="Permalink: Hands-on: Join" class="anchor" href="#hands-on-join" id="user-content-hands-on-join">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Use
                       <strong>
                        Join two files:wrench::gear:
                       </strong>
                       to add
                       <strong>
                        UniProt cancer genes
                       </strong>
                       information
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          “1st file”
                         </em>
                         : the GEMINI-generated gene report from the previous step
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Column to use from 1st file”
                         </em>
                         :
                         <span>
                          <code>
                           Column: 1
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “2nd file”
                         </em>
                         : the imported
                         <span>
                          <code>
                           Uniprot_cancer_Genes
                          </code>
                         </span>
                         dataset
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Column to use from 2nd file”
                         </em>
                         :
                         <span>
                          <code>
                           Column: 1
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Output lines appearing in”
                         </em>
                         :
                         <span>
                          <code>
                           Both 1st &amp; 2nd file, plus unpairable lines from 1st file. (-a 1)
                          </code>
                         </span>
                        </p>
                        <p dir="auto">
                         We obviously want to include a variant-affected gene in the final report even if it is not listed as a Uniprot cancer gene.
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “First line is a header line”
                         </em>
                         :
                         <span>
                          <code>
                           Yes
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Ignore case”
                         </em>
                         :
                         <span>
                          <code>
                           No
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Value to put in unpaired (empty) fields”
                         </em>
                         :
                         <span>
                          <code>
                           0
                          </code>
                         </span>
                        </p>
                        <p dir="auto">
                         If you look at the
                         <span>
                          <code>
                           Uniprot_Cancer_Genes
                          </code>
                         </span>
                         dataset, you'll notice that it has two annotation columns: one that indicates whether a given gene is a proto-oncogene or not using
                         <span>
                          <code>
                           1
                          </code>
                         </span>
                         and
                         <span>
                          <code>
                           0
                          </code>
                         </span>
                         , and another that indicates tumor suppressor genes in the same way. To indicate the common case that a gene affected by a variant is neither a known proto-oncogene nor a tumor suppressor gene, we want to fill the corresponding two columns of the join result with
                         <span>
                          <code>
                           0
                          </code>
                         </span>
                         .
                        </p>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <p dir="auto">
                       Use
                       <strong>
                        Join two files:wrench:
                       </strong>
                       ⚙️ to add
                       <strong>
                        CGI biomarkers
                       </strong>
                       information
                      </p>
                      <ul dir="auto">
                       <li>
                        <em>
                         “1st file”
                        </em>
                        : the partially annotated dataset from the previous
                       </li>
                       <li>
                        <em>
                         “Column to use from 1st file”
                        </em>
                        :
                        <span>
                         <code>
                          Column: 1
                         </code>
                        </span>
                       </li>
                       <li>
                        <em>
                         “2nd file”
                        </em>
                        : the imported
                        <span>
                         <code>
                          cgi_genes
                         </code>
                        </span>
                        dataset
                       </li>
                       <li>
                        <em>
                         “Column to use from 2nd file”
                        </em>
                        :
                        <span>
                         <code>
                          Column: 1
                         </code>
                        </span>
                       </li>
                       <li>
                        <em>
                         “Output lines appearing in”
                        </em>
                        :
                        <span>
                         <code>
                          Both 1st &amp; 2nd file, plus unpairable lines from 1st file. (-a 1)
                         </code>
                        </span>
                       </li>
                       <li>
                        <em>
                         “First line is a header line”
                        </em>
                        :
                        <span>
                         <code>
                          Yes
                         </code>
                        </span>
                       </li>
                       <li>
                        <em>
                         “Ignore case”
                        </em>
                        :
                        <span>
                         <code>
                          No
                         </code>
                        </span>
                       </li>
                       <li>
                        <em>
                         “Value to put in unpaired (empty) fields”
                        </em>
                        :
                        <span>
                         <code>
                          0
                         </code>
                        </span>
                       </li>
                      </ul>
                      <p dir="auto">
                       Inspect the input and the result dataset to make sure you understand what happened at this step.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Use
                       <strong>
                        Join two files:wrench:
                       </strong>
                       ⚙️ to add gene information from
                       <strong>
                        CIViC
                       </strong>
                      </p>
                      <ul dir="auto">
                       <li>
                        <p dir="auto">
                         <em>
                          “1st file”
                         </em>
                         : the partially annotated dataset from step 2
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Column to use from 1st file”
                         </em>
                         :
                         <span>
                          <code>
                           Column: 1
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “2nd file”
                         </em>
                         : the imported
                         <span>
                          <code>
                           GeneSummaries
                          </code>
                         </span>
                         dataset
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Column to use from 2nd file”
                         </em>
                         :
                         <span>
                          <code>
                           Column: 3
                          </code>
                         </span>
                        </p>
                        <p dir="auto">
                         The gene column in the CIViC gene summaries annotation dataset is
                         <em>
                          not
                         </em>
                         the first one!
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Output lines appearing in”
                         </em>
                         :
                         <span>
                          <code>
                           Both 1st &amp; 2nd file, plus unpairable lines from 1st file. (-a 1)
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “First line is a header line”
                         </em>
                         :
                         <span>
                          <code>
                           Yes
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Ignore case”
                         </em>
                         :
                         <span>
                          <code>
                           No
                          </code>
                         </span>
                        </p>
                       </li>
                       <li>
                        <p dir="auto">
                         <em>
                          “Value to put in unpaired (empty) fields”
                         </em>
                         :
                         <span>
                          <code>
                           .
                          </code>
                         </span>
                        </p>
                       </li>
                      </ul>
                      <p dir="auto">
                       Inspect the input and the result dataset to make sure you understand what happened at this step.
                      </p>
                     </li>
                    </ol>
                    <p dir="auto">
                     Looking at the output datasets, the gene columns from each of the
                     <em>
                      join
                     </em>
                     operations was preserved from both input datasets. Furthermore, we had no control over the order in which columns were added to the report, and we couldn't exclude any columns.
                    </p>
                    <p dir="auto">
                     In
                     <strong>
                      Step 2
                     </strong>
                     of the report preparation process, we will address all of these issues and rearrange the data to produce a fully annotated gene report. The tool used is
                     <strong>
                      Column arrange by header name
                     </strong>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Hands-on: Rearrange to get a fully annotated gene report
                     </h4>
                     <a aria-label="Permalink: Hands-on: Rearrange to get a fully annotated gene report" class="anchor" href="#hands-on-rearrange-to-get-a-fully-annotated-gene-report" id="user-content-hands-on-rearrange-to-get-a-fully-annotated-gene-report">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Run
                      <strong>
                       Column arrange by header name:wrench:
                      </strong>
                      ⚙️ configured like this:
                      <ul dir="auto">
                       <li>
                        <em>
                         “file to rearrange”
                        </em>
                        : the final
                        <strong>
                         Join
                        </strong>
                        result dataset from step 3
                       </li>
                       <li>
                        In
                        <em>
                         “Specify the first few columns by name”
                        </em>
                        <ul dir="auto">
                         <li>
                          In
                          <em>
                           “1: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              gene
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          ➕
                          <em>
                           “Specify the first few columns by name”
                          </em>
                         </li>
                         <li>
                          In
                          <em>
                           “2: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              chrom
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          ➕
                          <em>
                           “Specify the first few columns by name”
                          </em>
                         </li>
                         <li>
                          In
                          <em>
                           “3: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              synonym
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          ➕
                          <em>
                           “Specify the first few columns by name”
                          </em>
                         </li>
                         <li>
                          In
                          <em>
                           “4: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              hgnc_id
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          ➕
                          <em>
                           “Specify the first few columns by name”
                          </em>
                         </li>
                         <li>
                          In
                          <em>
                           “5: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              entrez_id
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          ➕
                          <em>
                           “Specify the first few columns by name”
                          </em>
                         </li>
                         <li>
                          In
                          <em>
                           “6: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              rvis_pct
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          ➕
                          <em>
                           “Specify the first few columns by name”
                          </em>
                         </li>
                         <li>
                          In
                          <em>
                           “7: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              is_OG
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          ➕
                          <em>
                           “Specify the first few columns by name”
                          </em>
                         </li>
                         <li>
                          In
                          <em>
                           “8: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              is_TS
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          ➕
                          <em>
                           “Specify the first few columns by name”
                          </em>
                         </li>
                         <li>
                          In
                          <em>
                           “9: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              in_cgi_biomarkers
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          ➕
                          <em>
                           “Specify the first few columns by name”
                          </em>
                         </li>
                         <li>
                          In
                          <em>
                           “10: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              clinvar_gene_phenotype
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          ➕
                          <em>
                           “Specify the first few columns by name”
                          </em>
                         </li>
                         <li>
                          In
                          <em>
                           “11: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              gene_civic_url
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                         <li>
                          ➕
                          <em>
                           “Specify the first few columns by name”
                          </em>
                         </li>
                         <li>
                          In
                          <em>
                           “12: Specify the first few columns by name”
                          </em>
                          <ul dir="auto">
                           <li>
                            <em>
                             “column”
                            </em>
                            :
                            <span>
                             <code>
                              description
                             </code>
                            </span>
                           </li>
                          </ul>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <em>
                         “Discard unspecified columns”
                        </em>
                        :
                        <span>
                         <code>
                          Yes
                         </code>
                        </span>
                       </li>
                      </ul>
                     </li>
                    </ol>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Alternative tool suggestion
                      </strong>
                     </p>
                     <p dir="auto">
                      If your Galaxy server doesn't have the Column arrange tool, it almost certainly has the
                      <strong>
                       Cut columns from a table tool
                      </strong>
                      , which can be used as a stand-in. This tool, however, expects a comma-separated list of column indexes, such as
                      <span>
                       <code>
                        c1,c2
                       </code>
                      </span>
                      for the first and second columns, so you must first determine the column numbers of your columns of interest.
                     </p>
                     <p dir="auto">
                      Keep in mind that the suggested tool should not be confused with
                      <em>
                       Cut columns from a table (cut)
                      </em>
                      , which does not allow you to change the order of columns!
                     </p>
                    </blockquote>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Conclusion
                     </h1>
                     <a aria-label="Permalink: Conclusion" class="anchor" href="#conclusion" id="user-content-conclusion">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     In addition to simply calling variants,
                     <em>
                      somatic variant calling
                     </em>
                     attempts to distinguish
                     <em>
                      somatic mutations
                     </em>
                     , which are unique to tumor tissue, from
                     <em>
                      germline
                     </em>
                     mutations, which are shared by tumor and healthy tissue, and
                     <em>
                      loss-of-heterozygosity
                     </em>
                     events, which involve the loss of one of two alleles found at a biallelic site in healthy tissue, from tumor tissue.
                    </p>
                    <p dir="auto">
                     Dedicated somatic variant callers can perform this classification based on statistics, but the interpretation of any list of variants (somatic, germline, or LOH) is also critically dependent on rich genetic and cancer-specific variant and gene annotations.
                    </p>
                    <blockquote>
                     <p dir="auto">
                      Some
                      <strong>
                       Key points
                      </strong>
                      to consider:
                     </p>
                     <ul dir="auto">
                      <li>
                       To reduce false-positive variant calls, use best practices for read mapping, quality control, and mapped reads post-processing.
                      </li>
                      <li>
                       Call variants and classify them statistically into somatic, germline, and LOH event variants using a dedicated somatic variant caller.
                      </li>
                      <li>
                       Annotations and queries based on variant properties improve the usefulness of variant and gene reports.
                      </li>
                      <li>
                       <strong>
                        GEMINI
                       </strong>
                       is a very useful framework for managing, annotating, and querying lists of variants in a flexible manner.
                      </li>
                      <li>
                       To encourage reproducibility and information sharing, prefer public, free annotation sources.
                      </li>
                     </ul>
                    </blockquote>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Names of contributors:
                     </h3>
                     <a aria-label="Permalink: Names of contributors:" class="anchor" href="#names-of-contributors" id="user-content-names-of-contributors">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Galaxy workflow:
                     </h4>
                     <a aria-label="Permalink: Galaxy workflow:" class="anchor" href="#galaxy-workflow" id="user-content-galaxy-workflow">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/ernesto">
                      @ernesto
                     </a>
                     (TeamLead
@NNyakundiM
                     <a href="https://github.com/aneela">
                      @aneela
                     </a>
                     <a href="https://github.com/clementmlay">
                      @clementmlay
                     </a>
                     <a href="https://github.com/Brenda">
                      @brenda
                     </a>
                     @Njuguna.E
                     <a href="https://github.com/Kasturi">
                      @kasturi
                     </a>
                     <a href="https://github.com/laurah">
                      @laurah
                     </a>
                    </p>
                    <p dir="auto">
                     <a href="https://github.com/Adekemi">
                      @Adekemi
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Github repo:
                     </h4>
                     <a aria-label="Permalink: Github repo:" class="anchor" href="#github-repo" id="user-content-github-repo">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/Kwabena">
                      @Kwabena
                     </a>
                     (Lead)
                     <a href="https://github.com/Tobibantu">
                      @Tobibantu
                     </a>
                     @Vincentabayomi
@Paulah_Kings
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Graphics design:
                     </h4>
                     <a aria-label="Permalink: Graphics design:" class="anchor" href="#graphics-design" id="user-content-graphics-design">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/Tobibantu">
                      @Tobibantu
                     </a>
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="Fz+7U3poD+p0LyZv6YK+71zInDfdSXigJeDbldyJ4/shGqGwKGA/KR823qEGhkGEZ6qscboWh1Pv02BAm7XWrg==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <div class="f4 my-3 color-fg-muted text-italic">
                 No description, website, or topics provided.
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/ErnestoDibia/Genomics_one_c-Project/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2FErnestoDibia%2FGenomics_one_c-Project&amp;report=ErnestoDibia+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/ErnestoDibia/Genomics_one_c-Project/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/ErnestoDibia/packages?repo_name=Genomics_one_c-Project">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/ErnestoDibia/Genomics_one_c-Project/used_by_list">
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
