<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-ee1857a059c0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_github_catalyst_lib_index_-eccae9-1932eeecf006.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-318896a86f35.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-9fad2423070b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - flemm0/gastric_cancer_differential_expression: Final Project for TRGN 514: Introduction to Human Genomic Analysis Methods
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:141BEAC:1D152ED:6605D545" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDoxNDFCRUFDOjFEMTUyRUQ6NjYwNUQ1NDUiLCJ2aXNpdG9yX2lkIjoiNjQ1MjI0MTg1NjIzODgxNzgwNyIsInJlZ2lvbl9lZGdlIjoiaWFkIiwicmVnaW9uX3JlbmRlciI6ImlhZCJ9" data-pjax-transient="true" name="visitor-payload">
           <meta content="6273c5ce06ed8b7979a61006c24eb109126238bd329a0dc3549bec93e41df432" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:557615436" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="Final Project for TRGN 514: Introduction to Human Genomic Analysis Methods - flemm0/gastric_cancer_differential_expression" name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/flemm0/gastric_cancer_differential_expression" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/c3b31778b4b6f2e84b99a253975bda7324e0572366b5fa5304eb054e493eb99d/flemm0/gastric_cancer_differential_expression" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - flemm0/gastric_cancer_differential_expression: Final Project for TRGN 514: Introduction to Human Genomic Analysis Methods" name="twitter:title">
                    <meta content="Final Project for TRGN 514: Introduction to Human Genomic Analysis Methods - flemm0/gastric_cancer_differential_expression" name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/c3b31778b4b6f2e84b99a253975bda7324e0572366b5fa5304eb054e493eb99d/flemm0/gastric_cancer_differential_expression" property="og:image">
                      <meta content="Final Project for TRGN 514: Introduction to Human Genomic Analysis Methods - flemm0/gastric_cancer_differential_expression" property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - flemm0/gastric_cancer_differential_expression: Final Project for TRGN 514: Introduction to Human Genomic Analysis Methods" property="og:title"/>
                        <meta content="https://github.com/flemm0/gastric_cancer_differential_expression" property="og:url"/>
                        <meta content="Final Project for TRGN 514: Introduction to Human Genomic Analysis Methods - flemm0/gastric_cancer_differential_expression" property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="def536b5d69b8afc2117ff3aa904c09699292b9d809266f1d19b952c3b02461c" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="5aaf0d46cfb23ac0376126aa517fe17c5c6293814f8cfb7fb33598c257a2a6b9" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/flemm0/gastric_cancer_differential_expression git https://github.com/flemm0/gastric_cancer_differential_expression.git" name="go-import"/>
                        <meta content="79301054" name="octolytics-dimension-user_id">
                         <meta content="flemm0" name="octolytics-dimension-user_login">
                          <meta content="557615436" name="octolytics-dimension-repository_id">
                           <meta content="flemm0/gastric_cancer_differential_expression" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="557615436" name="octolytics-dimension-repository_network_root_id">
                               <meta content="flemm0/gastric_cancer_differential_expression" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/flemm0/gastric_cancer_differential_expression" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-99519581d0f8.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/flemm0/gastric_cancer_differential_expression","user_id":null}}' data-hydro-click-hmac="0b56944694f96a71dcecbcfa441c3b870652eb0da4a8221f624ac051734e977f" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fflemm0%2Fgastric_cancer_differential_expression">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=7Y61sFEpp3q6fxv4L73eWg%2BBTQTQLj2K20YmLNa%2BY2v1JdKJXBvIzfMPoZWZ3C7F1boBpPrGmMZC%2FSVoTxJkTPPqPnBhn5mrnqCIEYWXOxpRO61Qm%2BLsYVn5%2BHyBNbU9dGTl5%2FoqmW4vha5%2F%2BVuodcow2YuyL7J6NnD7zBcsfw%2BxuWcAYluAbRbm%2FUYNIjk8dL08zz0RcjhfM44xUmDZSd4YbZDT30berIvWxdAp7pu07UbecoPVuNbT5gDI31XvBiORo91W1Uag%2BznuJ25pNs9mwyQP9AZu%2FG8gWr9un1693BlYQzL8dqZ3k7eMdr2zGK9g0VM1M3eCmGuwOJl3hUJZaVOwvcXj%2FFxjrHt0Pju1g3lU%2FGuZHMqmnsdVKxqlU0ZYV%2BjAkflhKkeqEeD67d4rn4C4p89OK0%2Bl2jFPYuR%2FmVcdG6hciJ%2FE4S6CNrd8Y%2FbthDjI%2FLqppI4wvxXjs1RBXIhKMmg%2F0DRZ3s9tBmD5vZRZY3hDI5EBNlbBiSog3Jhy6wX9R%2Bzw8swQa0bR5go4oANsILqvkUGEQhUa7QNXRGlhxhaRBgQvVi3BStEwS0w%3D--MKYBPsWLPIJyXrZt--DwwAZ5n2YApBFZ09rydwqw%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="flemm0" data-current-repository="flemm0/gastric_cancer_differential_expression" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="taA_rcsEqo2UqtHG87unoqo0ocJq5NPkpTwZltVDMYdX51pV6_wAwNK0ubIbf6uz-XQFxUrSo-mte5C3CSaw4g" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:flemm0/gastric_cancer_differential_expression">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-ce48a3b2-4aec-4466-85d9-e20a64186ce4" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-ce48a3b2-4aec-4466-85d9-e20a64186ce4">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="Pif3GsIoAQMH9f44RRsqj9igHmXx3+HZEzZCFNleUlKA4+LsvScneE30n0iU//OXaRzFX3bEoQlaf6aRCiA7Yg=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="+34VxzPKwXh9OuepJeXYJvAVVCFi0xKFTZvJnFbc3A1x+bubM+B8rzGCyEIcMIb+NX23L2BrQHlqPbzHOE934A==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="yGTAhEw0p1j9Oyv0piuhDY9DPpRtmekZb8BRNZG7M2ilqmjYbsZUOnfWWw4AeXcDWxj57+Cn25O/7DOS7P5xCw==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="MY/8drfLyRzpV3ZIuoErK8uUfNO1s17NvBFkDMbTiHvHkjkQS1j5Yiz62NEIbV8DfdT0+h0u0YaiTd82Da8kaw==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/flemm0/gastric_cancer_differential_expression","user_id":null}}' data-hydro-click-hmac="0b56944694f96a71dcecbcfa441c3b870652eb0da4a8221f624ac051734e977f" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fflemm0%2Fgastric_cancer_differential_expression">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/flemm0/gastric_cancer_differential_expression","user_id":null}}' data-hydro-click-hmac="0b56944694f96a71dcecbcfa441c3b870652eb0da4a8221f624ac051734e977f" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=flemm0%2Fgastric_cancer_differential_expression">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-a4b0e83a-2641-432d-a339-71a7d1d8c410" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-7252bb52-df12-4061-b9c0-fd7f26d9923a" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-7252bb52-df12-4061-b9c0-fd7f26d9923a" id="tooltip-a4b0e83a-2641-432d-a339-71a7d1d8c410" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/flemm0/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/flemm0" rel="author">
            flemm0
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/flemm0/gastric_cancer_differential_expression">
            gastric_cancer_differential_expression
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/flemm0/gastric_cancer_differential_expression","user_id":null}}' data-hydro-click-hmac="5aeb68da28bf6d1372886c67cdbb8c79b314f0c56faa2d4ffb0dc3ec9ae531cc" data-view-component="true" href="/login?return_to=%2Fflemm0%2Fgastric_cancer_differential_expression" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":557615436,"auth_type":"LOG_IN","originating_url":"https://github.com/flemm0/gastric_cancer_differential_expression","user_id":null}}' data-hydro-click-hmac="ba2781b11643b893973ab448ed3e3a1d3aa0cb6237da38efd85a56edaea69e4c" data-view-component="true" href="/login?return_to=%2Fflemm0%2Fgastric_cancer_differential_expression" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
             0
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":557615436,"auth_type":"LOG_IN","originating_url":"https://github.com/flemm0/gastric_cancer_differential_expression","user_id":null}}' data-hydro-click-hmac="22f584594b03fed172fd1e7ece7a468585e4b1cdb35ac25727fc60f8a4169d39" data-view-component="true" href="/login?return_to=%2Fflemm0%2Fgastric_cancer_differential_expression" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="1 user starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="1">
              1
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <p class="f4 mb-3">
          Final Project for TRGN 514: Introduction to Human Genomic Analysis Methods
         </p>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/flemm0/gastric_cancer_differential_expression/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            1
           </span>
           star
          </a>
          <a class="Link--secondary no-underline mr-3" href="/flemm0/gastric_cancer_differential_expression/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/flemm0/gastric_cancer_differential_expression/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/flemm0/gastric_cancer_differential_expression/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/flemm0/gastric_cancer_differential_expression/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":557615436,"auth_type":"LOG_IN","originating_url":"https://github.com/flemm0/gastric_cancer_differential_expression","user_id":null}}' data-hydro-click-hmac="22f584594b03fed172fd1e7ece7a468585e4b1cdb35ac25727fc60f8a4169d39" data-view-component="true" href="/login?return_to=%2Fflemm0%2Fgastric_cancer_differential_expression" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/flemm0/gastric_cancer_differential_expression","user_id":null}}' data-hydro-click-hmac="5aeb68da28bf6d1372886c67cdbb8c79b314f0c56faa2d4ffb0dc3ec9ae531cc" data-view-component="true" href="/login?return_to=%2Fflemm0%2Fgastric_cancer_differential_expression" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /flemm0/gastric_cancer_differential_expression" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /flemm0/gastric_cancer_differential_expression/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /flemm0/gastric_cancer_differential_expression/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /flemm0/gastric_cancer_differential_expression/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /flemm0/gastric_cancer_differential_expression/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /flemm0/gastric_cancer_differential_expression/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/flemm0/gastric_cancer_differential_expression/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /flemm0/gastric_cancer_differential_expression/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-d8e4e5b4-cb11-4ebb-8b2f-e7f0bdeba36d-list" aria-haspopup="true" aria-labelledby="tooltip-c1bb9055-04c5-4c6b-9b6e-d0f0422b4f47" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-d8e4e5b4-cb11-4ebb-8b2f-e7f0bdeba36d-button" popovertarget="action-menu-d8e4e5b4-cb11-4ebb-8b2f-e7f0bdeba36d-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-d8e4e5b4-cb11-4ebb-8b2f-e7f0bdeba36d-button" id="tooltip-c1bb9055-04c5-4c6b-9b6e-d0f0422b4f47" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-d8e4e5b4-cb11-4ebb-8b2f-e7f0bdeba36d-button" anchor-offset="normal" data-view-component="true" id="action-menu-d8e4e5b4-cb11-4ebb-8b2f-e7f0bdeba36d-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-d8e4e5b4-cb11-4ebb-8b2f-e7f0bdeba36d-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-d8e4e5b4-cb11-4ebb-8b2f-e7f0bdeba36d-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression" id="item-a38c5d60-bc07-4a2e-9b26-2dcc2e8149a4" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/issues" id="item-f7ea073e-6ac1-45ff-a4de-ac4da4ebae25" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/pulls" id="item-fbbade18-7b19-4897-b974-2701818e8c6f" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/actions" id="item-bb07f92e-386f-4aa6-9c59-bbe7f2210b10" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/projects" id="item-6934817f-7140-4bba-9a27-c9f1f2f0831c" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/security" id="item-2781a093-30bb-4176-8645-e51dc27c78a3" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/pulse" id="item-4a6da0f3-7d95-41c2-b698-cd2b6d030b74" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         flemm0/gastric_cancer_differential_expression
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/flemm0/gastric_cancer_differential_expression/spoofed_commit_check/0e9f793937aae14728ac57f02e34e8d70ea2fe02">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":557615436,"defaultBranch":"main","name":"gastric_cancer_differential_expression","ownerLogin":"flemm0","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2022-10-26T01:44:54.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/79301054?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"main","listCacheKey":"v0:1666748805.2052429","canEdit":false,"refType":"branch","currentOid":"0e9f793937aae14728ac57f02e34e8d70ea2fe02"},"tree":{"items":[{"name":"IPA-results","path":"IPA-results","contentType":"directory"},{"name":"README_files/figure-gfm","path":"README_files/figure-gfm","contentType":"directory","hasSimplifiedPath":true},{"name":"data","path":"data","contentType":"directory"},{"name":".gitignore","path":".gitignore","contentType":"file"},{"name":"README.Rmd","path":"README.Rmd","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"},{"name":"server-commands.sh","path":"server-commands.sh","contentType":"file"},{"name":"tool_installation_log.sh","path":"tool_installation_log.sh","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":8,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/flemm0/gastric_cancer_differential_expression/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/flemm0/gastric_cancer_differential_expression.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone flemm0/gastric_cancer_differential_expression","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/flemm0/gastric_cancer_differential_expression/archive/refs/heads/main.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=557615436"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"47","overviewFiles":[{"displayName":"README.md","repoName":"gastric_cancer_differential_expression","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGI Cancer Gene Expression Data Visualization\u003c/h1\u003e\u003ca id=\"user-content-gi-cancer-gene-expression-data-visualization\" class=\"anchor\" aria-label=\"Permalink: GI Cancer Gene Expression Data Visualization\" href=\"#gi-cancer-gene-expression-data-visualization\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFlemming Wu\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIntroduction\u003c/h2\u003e\u003ca id=\"user-content-introduction\" class=\"anchor\" aria-label=\"Permalink: Introduction\" href=\"#introduction\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIn this project, I used publicly available RNA-Seq data from the Gene\nExpression Omnibus database (accession number GSE41476) to look at how\ndifferential gene expression between three groups of samples: gastric\ncancer tissue, matched adjacent normal tissue, and gastric cancer cell\nlines.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe process of downloading raw sequencing data, alignment, and obtaining\ngene expression counts was all done on a HPC server and the commands\nused for that can be found in the server_commands.sh file on this\nrepository. This file describes my downstream analysis steps.\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(tidyverse)\nlibrary(data.table)\nlibrary(ggfortify)\nlibrary(ggplot2)\nlibrary(factoextra)\nlibrary(ComplexHeatmap)\nlibrary(pvclust)\nlibrary(parallel)\nlibrary(dendextend)\nlibrary(scales)\nlibrary(DiagrammeR)\nlibrary(gridExtra)\nlibrary(tidytext)\nlibrary(readxl)\nlibrary(circlize)\nlibrary(webshot)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003etidyverse\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eggfortify\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eggplot2\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003efactoextra\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eComplexHeatmap\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003epvclust\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eparallel\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edendextend\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003escales\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eDiagrammeR\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003egridExtra\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003etidytext\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003ereadxl\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003ecirclize\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003ewebshot\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePlot Read Counts For all 8 Samples\u003c/h4\u003e\u003ca id=\"user-content-plot-read-counts-for-all-8-samples\" class=\"anchor\" aria-label=\"Permalink: Plot Read Counts For all 8 Samples\" href=\"#plot-read-counts-for-all-8-samples\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe tissue samples I worked with in my project come from the\nbio-resource center of the Asan Medical Center in Seoul. The researchers\ncollected human gastric cancer samples and adjacent normal tissues. They\nalso collected gastric cancer cell lines from the Korean Cell Line Bank.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThere were a total of 8 samples between the three groups: 2 normal\ngastric tissue samples, 3 gastric cancer tissue samples, and 3 samples\nfrom gastric cancer cell lines. The samples were all paired-end, and\nsequenced on Illuminas Genome Analyzer II and stored in SRA format.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"samples \u0026lt;- c(\u0026quot;SRR585570\u0026quot;, \u0026quot;SRR585571\u0026quot;, \u0026quot;SRR585572\u0026quot;, \u0026quot;SRR585573\u0026quot;,\n    \u0026quot;SRR585574\u0026quot;, \u0026quot;SRR585575\u0026quot;, \u0026quot;SRR585576\u0026quot;, \u0026quot;SRR585577\u0026quot;)\n\nreads \u0026lt;- c(18883105, 16580585, 27917182, 34687136, 32772474,\n    34049244, 31929290, 32457417)\n\ngroups \u0026lt;- c(\u0026quot;normal gastric tissue\u0026quot;, \u0026quot;normal gastric tissue\u0026quot;,\n    \u0026quot;gastric cancer tissue\u0026quot;, \u0026quot;gastric cancer tissue\u0026quot;, \u0026quot;gastric cancer tissue\u0026quot;,\n    \u0026quot;gastric cancer cell line\u0026quot;, \u0026quot;gastric cancer cell line\u0026quot;, \u0026quot;gastric cancer cell line\u0026quot;)\n\ndf \u0026lt;- data.frame(samples, reads, groups)\n\ndf %\u0026gt;%\n    ggplot(aes(y = reads/1e+06, x = samples)) + geom_col(aes(fill = groups)) +\n    scale_fill_manual(values = c(\u0026quot;#23E7F7\u0026quot;, \u0026quot;#2336F7\u0026quot;, \u0026quot;#EEF51F\u0026quot;)) +\n    geom_text(aes(label = format(reads, big.mark = \u0026quot;,\u0026quot;)), position = position_dodge(width = 1),\n        vjust = -0.5) + geom_hline(yintercept = mean(reads)/1e+06,\n    color = \u0026quot;black\u0026quot;, linetype = \u0026quot;dotted\u0026quot;) + geom_text(aes(0,\n    mean(reads)/1e+06, label = paste(\u0026quot;Average reads:\u0026quot;, format(round(mean(reads)),\n        big.mark = \u0026quot;,\u0026quot;), sep = \u0026quot; \u0026quot;), vjust = -1, hjust = 0)) +\n    labs(x = \u0026quot;Sample\u0026quot;, y = \u0026quot;Number of Reads (millions)\u0026quot;, fill = \u0026quot;Group\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR585570\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR585571\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR585572\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR585573\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR585574\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR585575\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR585576\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR585577\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003ereads\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e18883105\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e16580585\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e27917182\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e34687136\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e32772474\u003c/span\u003e,\n    \u003cspan class=\"pl-c1\"\u003e34049244\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e31929290\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e32457417\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003egroups\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal gastric tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal gastric tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egastric cancer tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egastric cancer tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egastric cancer tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egastric cancer cell line\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egastric cancer cell line\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egastric cancer cell line\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ereads\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egroups\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ereads\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1e+06\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_col(aes(\u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egroups\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    scale_fill_manual(\u003cspan class=\"pl-v\"\u003evalues\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#23E7F7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#2336F7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#EEF51F\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    geom_text(aes(\u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e format(\u003cspan class=\"pl-smi\"\u003ereads\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ebig.mark\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)), \u003cspan class=\"pl-v\"\u003eposition\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e position_dodge(\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e),\n        \u003cspan class=\"pl-v\"\u003evjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_hline(\u003cspan class=\"pl-v\"\u003eyintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e mean(\u003cspan class=\"pl-smi\"\u003ereads\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1e+06\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elinetype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edotted\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_text(aes(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\n    mean(\u003cspan class=\"pl-smi\"\u003ereads\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1e+06\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAverage reads:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, format(round(mean(\u003cspan class=\"pl-smi\"\u003ereads\u003c/span\u003e)),\n        \u003cspan class=\"pl-v\"\u003ebig.mark\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003evjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    labs(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNumber of Reads (millions)\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGroup\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/pre-alignment metric plot-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/pre-alignment metric plot-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eI determined that I was able to use all 8 samples, since they all had a\ngood number of reads for my purposes. The minimum number of reads was\n16.5 million which was in the second normal tissue sample, the maximum\nwas 34.6 million which was in my second gastric cancer tissue sample,\nand all 8 samples averaged at 28.6 million reads.\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFlowchart of bioinformatic workflow\u003c/h4\u003e\u003ca id=\"user-content-flowchart-of-bioinformatic-workflow\" class=\"anchor\" aria-label=\"Permalink: Flowchart of bioinformatic workflow\" href=\"#flowchart-of-bioinformatic-workflow\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"grViz(\u0026quot;digraph flowchart {\n      # node definitions with substituted label text\n      node [fontname = Arial, shape = rectangle, color = Lavender, style = filled]        \n      tab1 [label = '@@1']\n      tab2 [label = '@@2']\n      tab3 [label = '@@3']\n      tab4 [label = '@@4']\n      tab5 [label = '@@5']\n      tab6 [label = '@@6']\n      tab7 [label = '@@7']\n      tab8 [label = '@@8']\n      tab9 [label = '@@9']\n\n      # edge definitions with the node IDs\n      tab1 -\u0026gt; tab3;\n      tab2 -\u0026gt; tab3;\n      tab3 -\u0026gt; tab4; \n      tab3 -\u0026gt; tab5;\n      tab4 -\u0026gt; tab6;\n      tab4 -\u0026gt; tab7;\n      tab6 -\u0026gt; tab8;\n      tab7 -\u0026gt; tab9;\n      }\n\n      [1]: 'Download read (FASTQ) files from SRA using SRA-Toolkit'\n      [2]: 'Download reference genome (FASTA) and annotation (GTF) files from Ensembl'\n      [3]: 'Build index with Bowtie2'\n      [4]: 'Align reads to whole genome using Tophat2 (BAM)'\n      [5]: 'Align reads to whole transcriptome using Tophat2 (BAM)'\n      [6]: 'Obtain FPKM values using Cuffdiff with statistics turned off'\n      [7]: 'Obtain p-values and log2 fold differences between normal and cancer tissue using Cuffdiff with statistics on'\n      [8]: 'Output file: genes.fpkm_tracking'\n      [9]: 'Output file: gene_exp.diff'\n      \u0026quot;)\"\u003e\u003cpre\u003egrViz(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edigraph flowchart {\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      # node definitions with substituted label text\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      node [fontname = Arial, shape = rectangle, color = Lavender, style = filled]        \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab1 [label = '@@1']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab2 [label = '@@2']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab3 [label = '@@3']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab4 [label = '@@4']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab5 [label = '@@5']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab6 [label = '@@6']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab7 [label = '@@7']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab8 [label = '@@8']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab9 [label = '@@9']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      # edge definitions with the node IDs\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab1 -\u0026gt; tab3;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab2 -\u0026gt; tab3;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab3 -\u0026gt; tab4; \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab3 -\u0026gt; tab5;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab4 -\u0026gt; tab6;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab4 -\u0026gt; tab7;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab6 -\u0026gt; tab8;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab7 -\u0026gt; tab9;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      }\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [1]: 'Download read (FASTQ) files from SRA using SRA-Toolkit'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [2]: 'Download reference genome (FASTA) and annotation (GTF) files from Ensembl'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [3]: 'Build index with Bowtie2'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [4]: 'Align reads to whole genome using Tophat2 (BAM)'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [5]: 'Align reads to whole transcriptome using Tophat2 (BAM)'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [6]: 'Obtain FPKM values using Cuffdiff with statistics turned off'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [7]: 'Obtain p-values and log2 fold differences between normal and cancer tissue using Cuffdiff with statistics on'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [8]: 'Output file: genes.fpkm_tracking'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [9]: 'Output file: gene_exp.diff'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/Flowchart of bioinformatic workflow-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/Flowchart of bioinformatic workflow-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe above commands were all performed on a HPC cluster. The output\nfiles, genes.fpkm_tracking and gene_exp.diff were the files I used for\ndownstream analysis in R and are provided in the data folder in this\nrepository.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eI first downloaded the SRA files as fastq files using the fasterq-dump\ncommand from sra-toolkit. I also downloaded the fasta file for the human\nreference genome, version GRCh38, and the GTF file of the genome\nannotation, both of which were from Ensembl.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eI then built index files from the reference genome using Bowtie2, which\nwill be used to speed up the mapping process. Next, I selected one\nnormal tissue sample and one cancer tissue sample to map to the whole\ngenome and the whole transcriptome using Tophat2 to see if they mapped\nto one much better. It turned out that the mapping percentages for both\ngenome and transcriptome were both at around 80%, so I just mapped the\nrest of my samples to the whole genome.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eOnce all the reads were mapped, I took my alignment files, or bam files,\nand put them into two different runs of Cuffdiff. In the first run, I\ninputted all 8 of my samples and ran Cuffdiff with statistics turned off\nto get just the read counts of all the genes. The output for this was in\nmy genes.fpkm_tracking file. In the second run, I gave Cuffdiff only my\nnormal and cancer tissue samples as two groups and ran it with the\nstatistics on to get log2 fold change and p-values for the genes just\nbetween these two groups. The output for this was in the gene_exp.diff\nfile.\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePost-Tophat2 Alignment Metrics\u003c/h4\u003e\u003ca id=\"user-content-post-tophat2-alignment-metrics\" class=\"anchor\" aria-label=\"Permalink: Post-Tophat2 Alignment Metrics\" href=\"#post-tophat2-alignment-metrics\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"df \u0026lt;- df %\u0026gt;%\n    mutate(input = c(37766210, 33161170, 55834364, 69374272,\n        65544948, 68098488, 63858580, 64914834), mapped = c(31498668,\n        27402515, 45797812, 58072751, 52477113, 55003238, 42375461,\n        44831741), pct_mapped = round(mapped/input * 100, 1))\n\ndf %\u0026gt;%\n    ggplot(aes(y = pct_mapped, x = samples)) + geom_col(aes(fill = groups)) +\n    scale_fill_manual(values = c(\u0026quot;#23E7F7\u0026quot;, \u0026quot;#2336F7\u0026quot;, \u0026quot;#EEF51F\u0026quot;)) +\n    geom_text(aes(label = paste(pct_mapped, \u0026quot;%\u0026quot;, sep = \u0026quot; \u0026quot;)),\n        position = position_dodge(width = 1), vjust = -0.5) +\n    labs(x = \u0026quot;Sample\u0026quot;, y = \u0026quot;Percent of Reads Aligned to Genome\u0026quot;,\n        fill = \u0026quot;Group\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    mutate(\u003cspan class=\"pl-v\"\u003einput\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e37766210\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e33161170\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e55834364\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e69374272\u003c/span\u003e,\n        \u003cspan class=\"pl-c1\"\u003e65544948\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e68098488\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e63858580\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e64914834\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003emapped\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e31498668\u003c/span\u003e,\n        \u003cspan class=\"pl-c1\"\u003e27402515\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e45797812\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e58072751\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e52477113\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e55003238\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e42375461\u003c/span\u003e,\n        \u003cspan class=\"pl-c1\"\u003e44831741\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003epct_mapped\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e round(\u003cspan class=\"pl-smi\"\u003emapped\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003einput\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epct_mapped\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_col(aes(\u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egroups\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    scale_fill_manual(\u003cspan class=\"pl-v\"\u003evalues\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#23E7F7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#2336F7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#EEF51F\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    geom_text(aes(\u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-smi\"\u003epct_mapped\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e%\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)),\n        \u003cspan class=\"pl-v\"\u003eposition\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e position_dodge(\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003evjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    labs(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercent of Reads Aligned to Genome\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGroup\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/alignment metrics-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/alignment metrics-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eOverall, the Tophat alignment was pretty successful. Most of my samples\nhave over an 80% mapping rate. The maximum mapping percentage is 83.7%,\nand the two samples on the right in the chart are cancer cell lines 2\nand 3, they had slightly lower mapping rates than the other samples, at\nabout 66 and 69%. But, the average across all 8 samples was 78.5%.\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePCA analysis on all genes across all samples\u003c/h4\u003e\u003ca id=\"user-content-pca-analysis-on-all-genes-across-all-samples\" class=\"anchor\" aria-label=\"Permalink: PCA analysis on all genes across all samples\" href=\"#pca-analysis-on-all-genes-across-all-samples\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"fpkm \u0026lt;- data.table::fread(\u0026quot;./data/genes.fpkm_tracking\u0026quot;, sep = \u0026quot;\\t\u0026quot;)\n\nnames(fpkm)[26:29] \u0026lt;- names(fpkm)[26:29] %\u0026gt;% # fix column names that were incorrectly named in Cuffdiff command\n  str_replace(\u0026quot;ct2\u0026quot;, \u0026quot;ct3\u0026quot;)\n\n\n# select needed columns\n\nfpkm \u0026lt;- fpkm %\u0026gt;% \n  select(tracking_id, ends_with(\u0026quot;_FPKM\u0026quot;))\n\nrow.names(fpkm) \u0026lt;- fpkm$tracking_id\n\nfpkm \u0026lt;- fpkm %\u0026gt;% select(ends_with(\u0026quot;FPKM\u0026quot;))\n\nfpkm \u0026lt;- t(fpkm)\n\n\n# normalize everything using log10 conversion\n\nfpkm_norm \u0026lt;- fpkm + 0.1\n\nfpkm_norm \u0026lt;- log10(fpkm_norm)\n\n\n# create sample column for labeling\n\nfpkm_norm \u0026lt;- as.data.frame(fpkm_norm)\nfpkm_norm$samples \u0026lt;- row.names(fpkm_norm)\n\nfpkm_norm.data \u0026lt;- fpkm_norm[-ncol(fpkm_norm)]\n\nfpkm_norm$type \u0026lt;- ifelse(grepl(\u0026quot;norm\u0026quot;, fpkm_norm$samples), \u0026quot;normal tissue\u0026quot;,\n                            ifelse(grepl(\u0026quot;ct\u0026quot;, fpkm_norm$samples), \u0026quot;cancer tissue\u0026quot;, \u0026quot;cancer cell line\u0026quot;))\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e./data/genes.fpkm_tracking\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nnames(\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e29\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e names(\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e29\u003c/span\u003e] %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e fix column names that were incorrectly named in Cuffdiff command\u003c/span\u003e\n  str_replace(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ect2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ect3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e select needed columns\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  select(\u003cspan class=\"pl-smi\"\u003etracking_id\u003c/span\u003e, ends_with(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_FPKM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\nrow.names(\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etracking_id\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% select(ends_with(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFPKM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e t(\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e normalize everything using log10 conversion\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e log10(\u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e create sample column for labeling\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.data.frame(\u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e row.names(\u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003efpkm_norm.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e[\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003encol(\u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e)]\n\n\u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ifelse(grepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enorm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                            ifelse(grepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ect\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecancer tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecancer cell line\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"pca \u0026lt;- prcomp(fpkm_norm.data, scale. = FALSE)\n\n# pca$x # view principal components\n\nautoplot(pca, scale = FALSE, data = fpkm_norm, colour = \u0026quot;type\u0026quot;,\n    size = 3) + ggtitle(\u0026quot;PCA on 40,295 Genes\u0026quot;) + theme_bw()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003epca\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e prcomp(\u003cspan class=\"pl-smi\"\u003efpkm_norm.data\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale.\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e pca$x # view principal components\u003c/span\u003e\n\nautoplot(\u003cspan class=\"pl-smi\"\u003epca\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etype\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e ggtitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePCA on 40,295 Genes\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e theme_bw()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/plot pca and edit to scale-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/plot pca and edit to scale-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eYou can see that the groups clustered together pretty nicely already,\nwith the cancer tissue on the right, the normal tissue on the left, and\nthe cancer cell lines on the bottom. PC1 accounts for 31.2% of the\nvariance in the data set and seems to separate cancer from non-cancer\ngroups. PC2 accounts for 24.8% of the variance in the data and seems to\nseparate tissues and cell lines.\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eBoxplots for all gene FPKM values across all samples\u003c/h4\u003e\u003ca id=\"user-content-boxplots-for-all-gene-fpkm-values-across-all-samples\" class=\"anchor\" aria-label=\"Permalink: Boxplots for all gene FPKM values across all samples\" href=\"#boxplots-for-all-gene-fpkm-values-across-all-samples\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"df \u0026lt;- as.data.frame(fpkm)\ndf \u0026lt;- df[colSums(df) \u0026gt; 0]  # filter out genes with 0 expression across all samples\ndf \u0026lt;- df + 0.1\ndf \u0026lt;- log10(df)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.data.frame(\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e[colSums(\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e]  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e filter out genes with 0 expression across all samples\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e log10(\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"df %\u0026gt;%\n    t() %\u0026gt;%\n    reshape::melt() %\u0026gt;%\n    arrange(X1) %\u0026gt;%\n    rename(sample = X2) %\u0026gt;%\n    mutate(group = ifelse(grepl(\u0026quot;norm\u0026quot;, sample), \u0026quot;normal tissue\u0026quot;,\n        ifelse(grepl(\u0026quot;ct\u0026quot;, sample), \u0026quot;cancer tissue\u0026quot;, \u0026quot;cancer cell line\u0026quot;))) %\u0026gt;%\n    ggplot(aes(x = sample, y = value, fill = group)) + geom_boxplot() +\n    theme(axis.text.x = element_text(angle = 90, vjust = 0.5,\n        hjust = 1)) + labs(x = \u0026quot;Sample\u0026quot;, y = \u0026quot;log10(FPKM)\u0026quot;, fill = \u0026quot;Group\u0026quot;,\n    title = \u0026quot;FPKM Distributions for 40,295 Genes\u0026quot;) + theme_bw()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    t() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    \u003cspan class=\"pl-e\"\u003ereshape\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003emelt() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    arrange(\u003cspan class=\"pl-smi\"\u003eX1\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    rename(\u003cspan class=\"pl-v\"\u003esample\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eX2\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    mutate(\u003cspan class=\"pl-v\"\u003egroup\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e ifelse(grepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enorm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        ifelse(grepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ect\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecancer tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecancer cell line\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003evalue\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_boxplot() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    theme(\u003cspan class=\"pl-v\"\u003eaxis.text.x\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003eangle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e90\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003evjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e labs(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elog10(FPKM)\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGroup\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFPKM Distributions for 40,295 Genes\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e theme_bw()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/fpkm boxplots of all samples across all genes-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/fpkm boxplots of all samples across all genes-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFrom the plot, it looks like the highest median fpkm is in the normal\ntissue group by just a little bit, but the overall distribution of gene\nexpression looks very similar across all 8 samples.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003eMy next step was to filter genes down based on the log2 fold change and\np-values so that I could have a manageable number of genes to look at\nfor pathway analysis. The cutoffs I ended up using were a p-value of\nless than 0.001 and log 2 fold change that was less than -5 or greater\nthan 5. The filtering left me with 280 significantly differentially\nexpressed genes, 198 of them were downregulated and 82 of them were\nupregulated.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDiagram on filtering criteria for Integrated Pathway Analysis (from QIAGEN)\u003c/h4\u003e\u003ca id=\"user-content-diagram-on-filtering-criteria-for-integrated-pathway-analysis-from-qiagen\" class=\"anchor\" aria-label=\"Permalink: Diagram on filtering criteria for Integrated Pathway Analysis (from QIAGEN)\" href=\"#diagram-on-filtering-criteria-for-integrated-pathway-analysis-from-qiagen\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"grViz(diagram = \u0026quot;digraph flowchart {\n      # define node aesthetics\n      node [fontname = Arial, shape = egg, color = DarkGreen, style = filled, fontcolor = White]        \n      tab1 [label = '@@1']\n      tab2 [label = '@@2']\n      tab3 [label = '@@3']\n      tab4 [label = '@@4']\n      tab5 [label = '@@5']\n      tab6 [label = '@@6']\n      tab7 [label = '@@7']\n# set up node layout\n      tab1 -\u0026gt; tab6;\n      tab6 -\u0026gt; tab7;\n      tab7 -\u0026gt; tab2;\n      tab2 -\u0026gt; tab3;\n      tab3 -\u0026gt; tab4;\n      tab3 -\u0026gt; tab5\n      }\n      [1]: 'Initial number of genes: 62,635'\n      [2]: 'Filter p-value \u0026lt; 0.001 and log2 fold change \u0026lt; -5 or \u0026gt; 5'\n      [3]: '280 genes Remaining'\n      [4]: '198 genes downregulated'\n      [5]: '82 genes upregulated'\n      [6]: 'Filter 0 gene expression across all 8 samples'\n      [7]: '40,295 genes remaining'\n      \u0026quot;)\"\u003e\u003cpre\u003egrViz(\u003cspan class=\"pl-v\"\u003ediagram\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edigraph flowchart {\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      # define node aesthetics\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      node [fontname = Arial, shape = egg, color = DarkGreen, style = filled, fontcolor = White]        \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab1 [label = '@@1']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab2 [label = '@@2']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab3 [label = '@@3']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab4 [label = '@@4']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab5 [label = '@@5']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab6 [label = '@@6']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab7 [label = '@@7']\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e# set up node layout\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab1 -\u0026gt; tab6;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab6 -\u0026gt; tab7;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab7 -\u0026gt; tab2;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab2 -\u0026gt; tab3;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab3 -\u0026gt; tab4;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      tab3 -\u0026gt; tab5\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      }\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [1]: 'Initial number of genes: 62,635'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [2]: 'Filter p-value \u0026lt; 0.001 and log2 fold change \u0026lt; -5 or \u0026gt; 5'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [3]: '280 genes Remaining'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [4]: '198 genes downregulated'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [5]: '82 genes upregulated'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [6]: 'Filter 0 gene expression across all 8 samples'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      [7]: '40,295 genes remaining'\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/gene filtering-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/gene filtering-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePerform filtering of genes according to the above diagram\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"exp \u0026lt;- data.table::fread(\u0026quot;./data/gene_exp.diff\u0026quot;, sep = \u0026quot;\\t\u0026quot;)\n\n# the cutoff will be p-val of \u0026lt; 0.001 and log2 fold change greater than 5 or less than -5\n\nexp_filt \u0026lt;- exp %\u0026gt;%\n  arrange(`log2(fold_change)`) %\u0026gt;%\n  filter(p_value \u0026lt; 0.001) %\u0026gt;%\n  filter(`log2(fold_change)` != -Inf, # Inf or -Inf means gene had 0 expression in one of the groups\n         `log2(fold_change)` != Inf) %\u0026gt;%\n  filter(`log2(fold_change)` \u0026lt;= -5 | `log2(fold_change)` \u0026gt;= 5)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eexp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e./data/gene_exp.diff\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e the cutoff will be p-val of \u0026lt; 0.001 and log2 fold change greater than 5 or less than -5\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eexp_filt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eexp\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  arrange(\u003cspan class=\"pl-smi\"\u003e`log2(fold_change)`\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  filter(\u003cspan class=\"pl-smi\"\u003ep_value\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.001\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  filter(\u003cspan class=\"pl-smi\"\u003e`log2(fold_change)`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Inf or -Inf means gene had 0 expression in one of the groups\u003c/span\u003e\n         \u003cspan class=\"pl-smi\"\u003e`log2(fold_change)`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  filter(\u003cspan class=\"pl-smi\"\u003e`log2(fold_change)`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e`log2(fold_change)`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSeparate up and down regulated genes\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"up_genes \u0026lt;- exp_filt %\u0026gt;%\n    filter(`log2(fold_change)` \u0026gt; 0) %\u0026gt;%\n    select(gene_id, gene)\n\ndown_genes \u0026lt;- exp_filt %\u0026gt;%\n    filter(`log2(fold_change)` \u0026lt; 0) %\u0026gt;%\n    select(gene_id, gene)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eup_genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eexp_filt\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    filter(\u003cspan class=\"pl-smi\"\u003e`log2(fold_change)`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    select(\u003cspan class=\"pl-smi\"\u003egene_id\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003edown_genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eexp_filt\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    filter(\u003cspan class=\"pl-smi\"\u003e`log2(fold_change)`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    select(\u003cspan class=\"pl-smi\"\u003egene_id\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePCA with filtered genes only\u003c/h4\u003e\u003ca id=\"user-content-pca-with-filtered-genes-only\" class=\"anchor\" aria-label=\"Permalink: PCA with filtered genes only\" href=\"#pca-with-filtered-genes-only\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"fpkm_filt \u0026lt;- data.table::fread(\u0026quot;./data/genes.fpkm_tracking\u0026quot;, sep = \u0026quot;\\t\u0026quot;)\n\nnames(fpkm_filt)[26:29] \u0026lt;- names(fpkm_filt)[26:29] %\u0026gt;% # fix column names\n  str_replace(\u0026quot;ct2\u0026quot;, \u0026quot;ct3\u0026quot;)\n\n\nfpkm_filt \u0026lt;- fpkm_filt %\u0026gt;% \n  select(tracking_id, ends_with(\u0026quot;_FPKM\u0026quot;)) %\u0026gt;%\n  filter(tracking_id %in% exp_filt$test_id) %\u0026gt;%\n  column_to_rownames(\u0026quot;tracking_id\u0026quot;)\n\n\nfpkm_filt_norm \u0026lt;- log10(fpkm_filt + 0.1)          \n\nfpkm_filt_norm \u0026lt;- t(fpkm_filt_norm) %\u0026gt;% as.data.frame()\n\nfpkm_filt_norm$sample \u0026lt;- row.names(fpkm_norm)\n\nfpkm_filt_norm \u0026lt;- fpkm_filt_norm %\u0026gt;% \n  mutate(type = ifelse(grepl(\u0026quot;norm\u0026quot;, sample), \u0026quot;normal tissue\u0026quot;,\n                         ifelse(grepl(\u0026quot;ct\u0026quot;, sample), \u0026quot;cancer tissue\u0026quot;, \u0026quot;cancer cell line\u0026quot;)))\n\n\nfpkm_filt_norm.data \u0026lt;- fpkm_filt_norm %\u0026gt;% select(-c(sample, type))\n\npca2 \u0026lt;- prcomp(fpkm_filt_norm.data, scale. = FALSE)\n\n#pca2$x\n\nautoplot(pca2, scale = FALSE, data = fpkm_filt_norm, colour = \u0026quot;type\u0026quot;, size = 3) + \n  ggtitle(\u0026quot;PCA on 280 Filtered Genes\u0026quot;) + \n  theme_bw()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003efpkm_filt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e./data/genes.fpkm_tracking\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nnames(\u003cspan class=\"pl-smi\"\u003efpkm_filt\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e29\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e names(\u003cspan class=\"pl-smi\"\u003efpkm_filt\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e29\u003c/span\u003e] %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e fix column names\u003c/span\u003e\n  str_replace(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ect2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ect3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-smi\"\u003efpkm_filt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm_filt\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  select(\u003cspan class=\"pl-smi\"\u003etracking_id\u003c/span\u003e, ends_with(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_FPKM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  filter(\u003cspan class=\"pl-smi\"\u003etracking_id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eexp_filt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etest_id\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  column_to_rownames(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etracking_id\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e log10(\u003cspan class=\"pl-smi\"\u003efpkm_filt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e)          \n\n\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e t(\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% as.data.frame()\n\n\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e row.names(\u003cspan class=\"pl-smi\"\u003efpkm_norm\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm_filt_norm\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  mutate(\u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e ifelse(grepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enorm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                         ifelse(grepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ect\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecancer tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecancer cell line\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)))\n\n\n\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm_filt_norm\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% select(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003ec(\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etype\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003epca2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e prcomp(\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm.data\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale.\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003epca2$x\u003c/span\u003e\n\nautoplot(\u003cspan class=\"pl-smi\"\u003epca2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm_filt_norm\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etype\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  ggtitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePCA on 280 Filtered Genes\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  theme_bw()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/redo PCA with filtered genes only-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/redo PCA with filtered genes only-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWith the 280 filtered genes, I redid my principal component analysis and\ngot the plot that you see on the screen. It looks very similar to my\nfirst PCA plot with the unfiltered genes, the three groups also\nclustered together and in the same areas, but in this one it looks like\nthe samples within each group clustered a bit closer together. Also,\nprincipal component 1 now accounts for 62% of the variance in the data,\nwhich is double the principal component 1 from my original PCA plot.\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCompute distance measures and clustering methods for hierarchical clustering\u003c/h4\u003e\u003ca id=\"user-content-compute-distance-measures-and-clustering-methods-for-hierarchical-clustering\" class=\"anchor\" aria-label=\"Permalink: Compute distance measures and clustering methods for hierarchical clustering\" href=\"#compute-distance-measures-and-clustering-methods-for-hierarchical-clustering\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"rownames(fpkm_filt_norm.data) \u0026lt;- fpkm_filt_norm$sample\n\n# try pearson correlation and euclidean distance for\n# distance measures\nres.pearson.dist \u0026lt;- get_dist(fpkm_filt_norm.data, method = \u0026quot;pearson\u0026quot;)\nres.euclidean.dist \u0026lt;- get_dist(fpkm_norm.data, method = \u0026quot;euclidean\u0026quot;)\n\n# try ward and complete hierarchical clustering methods\nres.eucl.ward.hc \u0026lt;- hclust(d = res.euclidean.dist, method = \u0026quot;ward.D2\u0026quot;)\nres.eucl.complete.hc \u0026lt;- hclust(d = res.euclidean.dist, method = \u0026quot;complete\u0026quot;)\nres.pear.ward.hc \u0026lt;- hclust(d = res.pearson.dist, method = \u0026quot;ward.D2\u0026quot;)\nres.pear.complete.hc \u0026lt;- hclust(d = res.pearson.dist, method = \u0026quot;complete\u0026quot;)\n\n\np1 \u0026lt;- fviz_dend(res.pear.ward.hc, cex = 0.5, k = 3, color_labels_by_k = TRUE,\n    rect = TRUE, rect_fill = TRUE, main = paste(\u0026quot;Distance:\u0026quot;,\n        \u0026quot;pearson\u0026quot;, \u0026quot;Method:\u0026quot;, \u0026quot;ward.D2\u0026quot;, sep = \u0026quot; \u0026quot;))\n\n\np2 \u0026lt;- fviz_dend(res.pear.complete.hc, cex = 0.5, k = 3, color_labels_by_k = TRUE,\n    rect = TRUE, rect_fill = TRUE, main = paste(\u0026quot;Distance:\u0026quot;,\n        \u0026quot;pearson\u0026quot;, \u0026quot;Method:\u0026quot;, \u0026quot;complete\u0026quot;, sep = \u0026quot; \u0026quot;))\n\np3 \u0026lt;- fviz_dend(res.eucl.ward.hc, cex = 0.5, k = 3, color_labels_by_k = TRUE,\n    rect = TRUE, rect_fill = TRUE, main = paste(\u0026quot;Distance:\u0026quot;,\n        \u0026quot;euclidean\u0026quot;, \u0026quot;Method:\u0026quot;, \u0026quot;ward.D2\u0026quot;, sep = \u0026quot; \u0026quot;))\n\np4 \u0026lt;- fviz_dend(res.eucl.complete.hc, cex = 0.5, k = 3, color_labels_by_k = TRUE,\n    rect = TRUE, rect_fill = TRUE, main = paste(\u0026quot;Distance:\u0026quot;,\n        \u0026quot;euclidean\u0026quot;, \u0026quot;Method:\u0026quot;, \u0026quot;complete\u0026quot;, sep = \u0026quot; \u0026quot;))\n\ngrid.arrange(p1, p2, p3, p4)\"\u003e\u003cpre\u003erownames(\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm.data\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm_filt_norm\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e try pearson correlation and euclidean distance for\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e distance measures\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eres.pearson.dist\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e get_dist(\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm.data\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epearson\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eres.euclidean.dist\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e get_dist(\u003cspan class=\"pl-smi\"\u003efpkm_norm.data\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eeuclidean\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e try ward and complete hierarchical clustering methods\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eres.eucl.ward.hc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e hclust(\u003cspan class=\"pl-v\"\u003ed\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eres.euclidean.dist\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eres.eucl.complete.hc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e hclust(\u003cspan class=\"pl-v\"\u003ed\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eres.euclidean.dist\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecomplete\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eres.pear.ward.hc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e hclust(\u003cspan class=\"pl-v\"\u003ed\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eres.pearson.dist\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eres.pear.complete.hc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e hclust(\u003cspan class=\"pl-v\"\u003ed\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eres.pearson.dist\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecomplete\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-smi\"\u003ep1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fviz_dend(\u003cspan class=\"pl-smi\"\u003eres.pear.ward.hc\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ek\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor_labels_by_k\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003erect\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erect_fill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDistance:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epearson\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMethod:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\n\u003cspan class=\"pl-smi\"\u003ep2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fviz_dend(\u003cspan class=\"pl-smi\"\u003eres.pear.complete.hc\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ek\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor_labels_by_k\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003erect\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erect_fill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDistance:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epearson\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMethod:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecomplete\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003ep3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fviz_dend(\u003cspan class=\"pl-smi\"\u003eres.eucl.ward.hc\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ek\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor_labels_by_k\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003erect\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erect_fill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDistance:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eeuclidean\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMethod:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003ep4\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fviz_dend(\u003cspan class=\"pl-smi\"\u003eres.eucl.complete.hc\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ek\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor_labels_by_k\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003erect\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erect_fill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDistance:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eeuclidean\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMethod:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecomplete\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\ngrid.arrange(\u003cspan class=\"pl-smi\"\u003ep1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ep2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ep3\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ep4\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/Dendrogram of differentially expressed genes ward method-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/Dendrogram of differentially expressed genes ward method-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIt appears that using euclidean distance along with Wards method gives\nus the clustering that we expect to see given our knowledge of the three\ngroups.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"fviz_dend(res.eucl.ward.hc, cex = 0.75, k = 3, color_labels_by_k = TRUE,\n    k_colors = c(\u0026quot;blue\u0026quot;, \u0026quot;red\u0026quot;, \u0026quot;darkgreen\u0026quot;), rect = TRUE, rect_fill = TRUE,\n    show_labels = TRUE, main = \u0026quot;Hierarchical Clustering of Samples Based on 280 Filtered Genes\u0026quot;,\n    xlab = \u0026quot;Samples\u0026quot;)\"\u003e\u003cpre\u003efviz_dend(\u003cspan class=\"pl-smi\"\u003eres.eucl.ward.hc\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.75\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ek\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor_labels_by_k\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ek_colors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgreen\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003erect\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erect_fill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eshow_labels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHierarchical Clustering of Samples Based on 280 Filtered Genes\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/visualize p3-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/visualize p3-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe top separation separated cancer from non-cancer, and within the\ncancer cluster theres a separation between cancer tissue and cancer\ncell lines.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eEvaluate Dendrograms using p-values\u003c/h4\u003e\u003ca id=\"user-content-evaluate-dendrograms-using-p-values\" class=\"anchor\" aria-label=\"Permalink: Evaluate Dendrograms using p-values\" href=\"#evaluate-dendrograms-using-p-values\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# create cluster for parallel computing\nn.cores \u0026lt;- detectCores() * 3/4\ncl \u0026lt;- makePSOCKcluster(n.cores)\nclusterSetRNGStream(cl, 1234)\n\n# calculate p-value dendrogram for pearson correlation and\n# ward.d2 method\nres.pear.ward.pv \u0026lt;- parPvclust(cl, t(fpkm_filt_norm.data), method.hclust = \u0026quot;ward.D2\u0026quot;,\n    method.dist = \u0026quot;correlation\u0026quot;, nboot = 1000)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e create cluster for parallel computing\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003en.cores\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e detectCores() \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecl\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e makePSOCKcluster(\u003cspan class=\"pl-smi\"\u003en.cores\u003c/span\u003e)\nclusterSetRNGStream(\u003cspan class=\"pl-smi\"\u003ecl\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e1234\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e calculate p-value dendrogram for pearson correlation and\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ward.d2 method\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eres.pear.ward.pv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e parPvclust(\u003cspan class=\"pl-smi\"\u003ecl\u003c/span\u003e, t(\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm.data\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003emethod.hclust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003emethod.dist\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecorrelation\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enboot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Multiscale bootstrap... Done.\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Multiscale bootstrap... Done.\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# calculate p-value dendrogram for pearson correlation and\n# complete method\nres.pear.complete.pv \u0026lt;- parPvclust(cl, t(fpkm_filt_norm.data),\n    method.hclust = \u0026quot;complete\u0026quot;, method.dist = \u0026quot;correlation\u0026quot;,\n    nboot = 1000)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e calculate p-value dendrogram for pearson correlation and\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e complete method\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eres.pear.complete.pv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e parPvclust(\u003cspan class=\"pl-smi\"\u003ecl\u003c/span\u003e, t(\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm.data\u003c/span\u003e),\n    \u003cspan class=\"pl-v\"\u003emethod.hclust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecomplete\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod.dist\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecorrelation\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003enboot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Multiscale bootstrap... Done.\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Multiscale bootstrap... Done.\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# calculate p-value dendrogram for euclidean dist and\n# ward.d2 method\nres.eucl.complete.pv \u0026lt;- parPvclust(cl, t(fpkm_filt_norm.data),\n    method.hclust = \u0026quot;ward.D2\u0026quot;, method.dist = \u0026quot;euclidean\u0026quot;, nboot = 1000)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e calculate p-value dendrogram for euclidean dist and\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ward.d2 method\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eres.eucl.complete.pv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e parPvclust(\u003cspan class=\"pl-smi\"\u003ecl\u003c/span\u003e, t(\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm.data\u003c/span\u003e),\n    \u003cspan class=\"pl-v\"\u003emethod.hclust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod.dist\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eeuclidean\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enboot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Multiscale bootstrap... Done.\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Multiscale bootstrap... Done.\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"stopCluster(cl)\n\n# plot Clusters with AU \u0026gt;= 95% are considered to be\n# strongly supported by data.\nplot(res.pear.ward.pv, hang = -1, cex = 0.5)\"\u003e\u003cpre\u003estopCluster(\u003cspan class=\"pl-smi\"\u003ecl\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e plot Clusters with AU \u0026gt;= 95% are considered to be\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e strongly supported by data.\u003c/span\u003e\nplot(\u003cspan class=\"pl-smi\"\u003eres.pear.ward.pv\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehang\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/p-values for ward clustering-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/p-values for ward clustering-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"plot(res.pear.complete.pv, hang = -1, cex = 0.5)\"\u003e\u003cpre\u003eplot(\u003cspan class=\"pl-smi\"\u003eres.pear.complete.pv\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehang\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/p-values for ward clustering-2.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/p-values for ward clustering-2.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"plot(res.eucl.complete.pv, hang = -1, cex = 0.5)\"\u003e\u003cpre\u003eplot(\u003cspan class=\"pl-smi\"\u003eres.eucl.complete.pv\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehang\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/p-values for ward clustering-3.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/p-values for ward clustering-3.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAbove, it appears that using pearson correlation and complete clustering\nmethod clustered ccl3_FPKM poorly. The AU values for using both pearson\nand euclidean along with Wards method gave stronger AU values.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCompare top 2 dendrograms using tanglegram\u003c/h4\u003e\u003ca id=\"user-content-compare-top-2-dendrograms-using-tanglegram\" class=\"anchor\" aria-label=\"Permalink: Compare top 2 dendrograms using tanglegram\" href=\"#compare-top-2-dendrograms-using-tanglegram\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"dend1 \u0026lt;- as.dendrogram(res.pear.ward.hc)\ndend2 \u0026lt;- as.dendrogram(res.eucl.ward.hc)\n\ndendlist(dend1, dend2) %\u0026gt;%\n    untangle(method = \u0026quot;step1side\u0026quot;) %\u0026gt;%\n    tanglegram(main_left = \u0026quot;Pearson Correlation\u0026quot;, main_right = \u0026quot;Euclidean Distance\u0026quot;,\n        main = \u0026quot;Ward.D2\u0026quot;, lwd = 1, cex_main = 1, margin_inner = 5)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003edend1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.dendrogram(\u003cspan class=\"pl-smi\"\u003eres.pear.ward.hc\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003edend2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.dendrogram(\u003cspan class=\"pl-smi\"\u003eres.eucl.ward.hc\u003c/span\u003e)\n\ndendlist(\u003cspan class=\"pl-smi\"\u003edend1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edend2\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    untangle(\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003estep1side\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    tanglegram(\u003cspan class=\"pl-v\"\u003emain_left\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePearson Correlation\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emain_right\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEuclidean Distance\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eWard.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elwd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecex_main\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emargin_inner\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/view tanglegram of top clusters-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/view tanglegram of top clusters-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUsing pearson correlation for distance metric instead of euclidean\nresults in cancer cell line 2 clustering with the normal tissues instead\nof with the other cancer cell line samples. I have determined that using\neuclidean distance and wards method to be the best choices with my\ndata.\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePlot heatmap of gene expression\u003c/h4\u003e\u003ca id=\"user-content-plot-heatmap-of-gene-expression\" class=\"anchor\" aria-label=\"Permalink: Plot heatmap of gene expression\" href=\"#plot-heatmap-of-gene-expression\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"boxplot \u0026lt;- rowAnnotation(`log10(fpkm)` = anno_boxplot(data.matrix(fpkm_filt_norm.data)),\n    gp = gpar(fill = \u0026quot;#CCCCCC\u0026quot;))\n\nhmap \u0026lt;- Heatmap(data.matrix(fpkm_filt_norm.data), name = \u0026quot;Expression\u0026quot;,\n    show_column_names = FALSE, row_split = 3, clustering_method_rows = \u0026quot;ward.D2\u0026quot;,\n    clustering_method_columns = \u0026quot;ward.D2\u0026quot;, right_annotation = boxplot)\n\ndraw(hmap)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eboxplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rowAnnotation(\u003cspan class=\"pl-smi\"\u003e`log10(fpkm)`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e anno_boxplot(data.matrix(\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm.data\u003c/span\u003e)),\n    \u003cspan class=\"pl-v\"\u003egp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e gpar(\u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#CCCCCC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003ehmap\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e Heatmap(data.matrix(\u003cspan class=\"pl-smi\"\u003efpkm_filt_norm.data\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eExpression\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eshow_column_names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_method_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eclustering_method_columns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eright_annotation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eboxplot\u003c/span\u003e)\n\ndraw(\u003cspan class=\"pl-smi\"\u003ehmap\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/annotated heatmap wards-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/annotated heatmap wards-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eYou can see that there is a large block of genes that are expressed\nhighly in normal tissue and a block of genes expressed highly in cancer\ntissue. The cell line cluster seems to be spread out, with some highly\nexpressed genes overlapping with normal tissue and some overlapping with\ncancerous tissue. On the right side, I also included boxplots of the\noverall fpkm values and you can see that the expression levels are\nhigher overall for the normal samples, compared to the cancer samples\nwhich makes sense because now Im working with more genes that are\ndownregulated in cancer than upregulated\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003eAt this point, I looked to the list of upregulated and downregulated\ngenes from tthe integrated pathway analysis (software from QIAGEN) to\nfind a pathway to look at more in depth. There were many significant\npathways returned, many of the down regulated ones were involved in\nimmune system function whereas many of the up regulated pathways have\nbeen previously implicated in cancer. (The excel files are located in\nthe IPA-results folder.)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMoving forward, I will be look at the pathway title metabolism of\ncarbohydrates, which was downregulated in cancerous samples, compared\nto normal samples.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePlot heatmap of carbohydrate metabolism pathway.\u003c/h4\u003e\u003ca id=\"user-content-plot-heatmap-of-carbohydrate-metabolism-pathway\" class=\"anchor\" aria-label=\"Permalink: Plot heatmap of carbohydrate metabolism pathway.\" href=\"#plot-heatmap-of-carbohydrate-metabolism-pathway\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"meta \u0026lt;- read_xls(\u0026quot;./IPA-results/down-genes_diseases_and_functions.xls\u0026quot;,\n    col_names = TRUE) %\u0026gt;%\n    filter(`Diseases or Functions Annotation` == \u0026quot;Metabolism of carbohydrate\u0026quot;) %\u0026gt;%\n    unnest_tokens(genes, Molecules) %\u0026gt;%\n    mutate(genes = toupper(genes)) %\u0026gt;%\n    select(genes)\n\nmeta %\u0026gt;%\n    head(22)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003emeta\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read_xls(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e./IPA-results/down-genes_diseases_and_functions.xls\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ecol_names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    filter(\u003cspan class=\"pl-smi\"\u003e`Diseases or Functions Annotation`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMetabolism of carbohydrate\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    unnest_tokens(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eMolecules\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    mutate(\u003cspan class=\"pl-v\"\u003egenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e toupper(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    select(\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003emeta\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    head(\u003cspan class=\"pl-c1\"\u003e22\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## # A tibble: 22  1\n##    genes   \n##    \u0026lt;chr\u0026gt;   \n##  1 AKR1B10 \n##  2 ALDH1A1 \n##  3 B3GALT5 \n##  4 B3GAT1  \n##  5 B3GNT8  \n##  6 B4GALNT2\n##  7 CD22    \n##  8 CEBPA   \n##  9 CSF1R   \n## 10 GCNT2   \n## #  with 12 more rows\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## # A tibble: 22  1\n##    genes   \n##    \u0026lt;chr\u0026gt;   \n##  1 AKR1B10 \n##  2 ALDH1A1 \n##  3 B3GALT5 \n##  4 B3GAT1  \n##  5 B3GNT8  \n##  6 B4GALNT2\n##  7 CD22    \n##  8 CEBPA   \n##  9 CSF1R   \n## 10 GCNT2   \n## #  with 12 more rows\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"fpkm \u0026lt;- data.table::fread(\u0026quot;./data/genes.fpkm_tracking\u0026quot;, sep = \u0026quot;\\t\u0026quot;)\n\nnames(fpkm)[26:29] \u0026lt;- names(fpkm)[26:29] %\u0026gt;% # fix column names\n  str_replace(\u0026quot;ct2\u0026quot;, \u0026quot;ct3\u0026quot;)\n\n\nfpkm \u0026lt;- fpkm %\u0026gt;% \n  select(gene_short_name, ends_with(\u0026quot;_FPKM\u0026quot;))\n\nmeta \u0026lt;- merge(x = meta,\n      y = fpkm, \n      all.x = TRUE,\n      all.y = FALSE,\n      by.x = \u0026quot;genes\u0026quot;,\n      by.y = \u0026quot;gene_short_name\u0026quot;)\n\n\nmeta \u0026lt;- meta %\u0026gt;% \n  column_to_rownames(\u0026quot;genes\u0026quot;) %\u0026gt;% \n  t()\n\n\nmeta_norm \u0026lt;- log10(meta + 0.1)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e./data/genes.fpkm_tracking\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nnames(\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e29\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e names(\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e29\u003c/span\u003e] %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e fix column names\u003c/span\u003e\n  str_replace(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ect2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ect3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  select(\u003cspan class=\"pl-smi\"\u003egene_short_name\u003c/span\u003e, ends_with(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_FPKM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003emeta\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emeta\u003c/span\u003e,\n      \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efpkm\u003c/span\u003e, \n      \u003cspan class=\"pl-v\"\u003eall.x\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n      \u003cspan class=\"pl-v\"\u003eall.y\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e,\n      \u003cspan class=\"pl-v\"\u003eby.x\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egenes\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n      \u003cspan class=\"pl-v\"\u003eby.y\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene_short_name\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-smi\"\u003emeta\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emeta\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  column_to_rownames(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egenes\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  t()\n\n\n\u003cspan class=\"pl-smi\"\u003emeta_norm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e log10(\u003cspan class=\"pl-smi\"\u003emeta\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Heatmap(meta_norm, name = \u0026quot;expression\u0026quot;, show_column_names = TRUE,\n    row_split = 3, clustering_method_rows = \u0026quot;ward.D2\u0026quot;, clustering_method_columns = \u0026quot;ward.D2\u0026quot;) +\n    rowAnnotation(`log10(fpkm)` = anno_boxplot(meta_norm)) +\n    rowAnnotation(rn = anno_text(rownames(meta_norm)))\"\u003e\u003cpre\u003eHeatmap(\u003cspan class=\"pl-smi\"\u003emeta_norm\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eexpression\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eshow_column_names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003erow_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_method_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_method_columns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    rowAnnotation(\u003cspan class=\"pl-smi\"\u003e`log10(fpkm)`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e anno_boxplot(\u003cspan class=\"pl-smi\"\u003emeta_norm\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    rowAnnotation(\u003cspan class=\"pl-v\"\u003ern\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e anno_text(rownames(\u003cspan class=\"pl-smi\"\u003emeta_norm\u003c/span\u003e)))\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/annotated heatmap-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/annotated heatmap-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUsing the same clustering method (euclidean distance and ward.D2 method)\nit appears that cancer cell line 2 becomes its own cluster, and cancer\ncell lines 1 and 3 cluster with the cancer tissue.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# compare other clustering methods with original to check\n# to see if it is still the best method for the selected\n# pathway\n\n# original clustering method\nht1 \u0026lt;- Heatmap(meta_norm, name = \u0026quot;ht1\u0026quot;, show_column_names = TRUE,\n    row_split = 3, clustering_method_rows = \u0026quot;ward.D2\u0026quot;, clustering_method_columns = \u0026quot;ward.D2\u0026quot;)\n\n\n# pearson correlation with ward.D2 clustering\nht2 \u0026lt;- Heatmap(meta_norm, name = \u0026quot;ht2\u0026quot;, show_column_names = TRUE,\n    row_split = 3, clustering_distance_rows = \u0026quot;pearson\u0026quot;, clustering_distance_columns = \u0026quot;pearson\u0026quot;,\n    clustering_method_rows = \u0026quot;ward.D2\u0026quot;, clustering_method_columns = \u0026quot;ward.D2\u0026quot;,\n    column_names_gp = gpar(fontsize = 9))\n\n# euclidean distance with complete clustering\n\nht3 \u0026lt;- Heatmap(meta_norm, name = \u0026quot;ht3\u0026quot;, show_column_names = TRUE,\n    row_split = 3, clustering_distance_rows = \u0026quot;euclidean\u0026quot;, clustering_distance_columns = \u0026quot;euclidean\u0026quot;,\n    clustering_method_rows = \u0026quot;complete\u0026quot;, clustering_method_columns = \u0026quot;complete\u0026quot;,\n    column_names_gp = gpar(fontsize = 9))\n\n# pearson correlation with ward.D2 clustering\n\nht4 \u0026lt;- Heatmap(meta_norm, name = \u0026quot;ht4\u0026quot;, show_column_names = TRUE,\n    row_split = 3, clustering_distance_rows = \u0026quot;pearson\u0026quot;, clustering_distance_columns = \u0026quot;pearson\u0026quot;,\n    clustering_method_rows = \u0026quot;complete\u0026quot;, clustering_method_columns = \u0026quot;complete\u0026quot;,\n    column_names_gp = gpar(fontsize = 9))\n\nht1 + ht2 + ht3 + ht4\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e compare other clustering methods with original to check\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to see if it is still the best method for the selected\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e pathway\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e original clustering method\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eht1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e Heatmap(\u003cspan class=\"pl-smi\"\u003emeta_norm\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eht1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eshow_column_names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003erow_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_method_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_method_columns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e pearson correlation with ward.D2 clustering\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eht2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e Heatmap(\u003cspan class=\"pl-smi\"\u003emeta_norm\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eht2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eshow_column_names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003erow_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_distance_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epearson\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_distance_columns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epearson\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eclustering_method_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_method_columns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eward.D2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ecolumn_names_gp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e gpar(\u003cspan class=\"pl-v\"\u003efontsize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e euclidean distance with complete clustering\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eht3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e Heatmap(\u003cspan class=\"pl-smi\"\u003emeta_norm\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eht3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eshow_column_names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003erow_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_distance_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eeuclidean\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_distance_columns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eeuclidean\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eclustering_method_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecomplete\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_method_columns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecomplete\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ecolumn_names_gp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e gpar(\u003cspan class=\"pl-v\"\u003efontsize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e pearson correlation with ward.D2 clustering\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eht4\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e Heatmap(\u003cspan class=\"pl-smi\"\u003emeta_norm\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eht4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eshow_column_names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003erow_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_distance_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epearson\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_distance_columns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epearson\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eclustering_method_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecomplete\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclustering_method_columns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecomplete\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ecolumn_names_gp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e gpar(\u003cspan class=\"pl-v\"\u003efontsize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003eht1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eht2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eht3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eht4\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/other clustering method testing-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/other clustering method testing-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWhen plotting all 4 heatmaps together with the original heatmap as the\nmain, the separation is clearest with the original. Individually, ht2\nmakes cell line 3 its own cluster and clusters cell line 2 and cancer\ntissue 3 with normal, ht2 is similar to original heatmap, and ht3 makes\ncell line 1 and 3 are their own clusters. I conclude that the original\nclustering method is still the best option for this pathway.\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eLinear model test on individual genes for significance\u003c/h4\u003e\u003ca id=\"user-content-linear-model-test-on-individual-genes-for-significance\" class=\"anchor\" aria-label=\"Permalink: Linear model test on individual genes for significance\" href=\"#linear-model-test-on-individual-genes-for-significance\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# add PC1 and PC2 to the data frame for correction\n\nlm_df \u0026lt;- merge(x = pca$x, y = meta_norm, by = \u0026quot;row.names\u0026quot;) %\u0026gt;%\n    select(-c(4:9)) %\u0026gt;%\n    rename(group = Row.names) %\u0026gt;%\n    mutate(group = if_else(grepl(\u0026quot;norm\u0026quot;, group), \u0026quot;normal tissue\u0026quot;,\n        if_else(grepl(\u0026quot;ct\u0026quot;, group), \u0026quot;cancer tissue\u0026quot;, \u0026quot;cancer cell line\u0026quot;)))\n\n# lm_df\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e add PC1 and PC2 to the data frame for correction\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003elm_df\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epca\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emeta_norm\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erow.names\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    select(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    rename(\u003cspan class=\"pl-v\"\u003egroup\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eRow.names\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    mutate(\u003cspan class=\"pl-v\"\u003egroup\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e if_else(grepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enorm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        if_else(grepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ect\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecancer tissue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecancer cell line\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e lm_df\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# function to return p-value from lm object\nlmp \u0026lt;- function(modelobject) {\n    if (class(modelobject) != \u0026quot;lm\u0026quot;)\n        stop(\u0026quot;Not an object of class 'lm' \u0026quot;)\n    f \u0026lt;- summary(modelobject)$fstatistic\n    p \u0026lt;- pf(f[1], f[2], f[3], lower.tail = F)\n    attributes(p) \u0026lt;- NULL\n    return(p)\n}\n\n\n# lm test on all genes\nno_correction \u0026lt;- list()  # initialize empty list to hold p-values for non-corrected lm test\ncorrection \u0026lt;- list()  # empty list to hold p-values for corrected lm test\nfor (i in names(lm_df[4:length(names(lm_df))])) {\n    no_correction[i] \u0026lt;- lmp(lm(get(i) ~ group, lm_df))\n    correction[i] \u0026lt;- lmp(lm(get(i) ~ group + PC1 + PC2, lm_df))\n}\n\n\n# store results in data frame\nmeta_pval \u0026lt;- rbind(as.data.frame(no_correction), as.data.frame(correction)) %\u0026gt;%\n    t() %\u0026gt;%\n    as.data.frame() %\u0026gt;%\n    rename(`Non PC Corrected p-value` = V1, `PC Corrected p-value` = V2) %\u0026gt;%\n    format(scientific = FALSE) %\u0026gt;%\n    rownames_to_column(\u0026quot;gene\u0026quot;)\n\nmeta_pval \u0026lt;- meta_pval %\u0026gt;%\n    mutate(`Non PC Corrected p-value` = as.numeric(`Non PC Corrected p-value`),\n        `PC Corrected p-value` = as.numeric(`PC Corrected p-value`))\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e function to return p-value from lm object\u003c/span\u003e\n\u003cspan class=\"pl-en\"\u003elmp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003emodelobject\u003c/span\u003e) {\n    \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e (class(\u003cspan class=\"pl-smi\"\u003emodelobject\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n        stop(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNot an object of class 'lm' \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    \u003cspan class=\"pl-smi\"\u003ef\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e summary(\u003cspan class=\"pl-smi\"\u003emodelobject\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003efstatistic\u003c/span\u003e\n    \u003cspan class=\"pl-smi\"\u003ep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e pf(\u003cspan class=\"pl-smi\"\u003ef\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e], \u003cspan class=\"pl-smi\"\u003ef\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e], \u003cspan class=\"pl-smi\"\u003ef\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e], \u003cspan class=\"pl-v\"\u003elower.tail\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n    attributes(\u003cspan class=\"pl-smi\"\u003ep\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e\n    \u003cspan class=\"pl-k\"\u003ereturn\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ep\u003c/span\u003e)\n}\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e lm test on all genes\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eno_correction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e()  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e initialize empty list to hold p-values for non-corrected lm test\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecorrection\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e()  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e empty list to hold p-values for corrected lm test\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e names(\u003cspan class=\"pl-smi\"\u003elm_df\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003elength(names(\u003cspan class=\"pl-smi\"\u003elm_df\u003c/span\u003e))])) {\n    \u003cspan class=\"pl-smi\"\u003eno_correction\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lmp(lm(get(\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003elm_df\u003c/span\u003e))\n    \u003cspan class=\"pl-smi\"\u003ecorrection\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lmp(lm(get(\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ePC1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ePC2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003elm_df\u003c/span\u003e))\n}\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e store results in data frame\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emeta_pval\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rbind(as.data.frame(\u003cspan class=\"pl-smi\"\u003eno_correction\u003c/span\u003e), as.data.frame(\u003cspan class=\"pl-smi\"\u003ecorrection\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    t() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    as.data.frame() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    rename(\u003cspan class=\"pl-smi\"\u003e`Non PC Corrected p-value`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eV1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003e`PC Corrected p-value`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    format(\u003cspan class=\"pl-v\"\u003escientific\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    rownames_to_column(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003emeta_pval\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emeta_pval\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    mutate(\u003cspan class=\"pl-smi\"\u003e`Non PC Corrected p-value`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e as.numeric(\u003cspan class=\"pl-smi\"\u003e`Non PC Corrected p-value`\u003c/span\u003e),\n        \u003cspan class=\"pl-smi\"\u003e`PC Corrected p-value`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e as.numeric(\u003cspan class=\"pl-smi\"\u003e`PC Corrected p-value`\u003c/span\u003e))\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePlot p-values for each genes significance in determining difference\nbetween cancer and non cancer\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"meta_pval %\u0026gt;%\n    arrange(`PC Corrected p-value`) %\u0026gt;%\n    rename(`p-value before PC correction` = `Non PC Corrected p-value`,\n        `p-value after PC correction` = `PC Corrected p-value`) %\u0026gt;%\n    reshape::melt() %\u0026gt;%\n    ggplot(aes(x = fct_inorder(gene), y = value)) + geom_bar(aes(fill = variable),\n    stat = \u0026quot;identity\u0026quot;, position = \u0026quot;dodge\u0026quot;, width = 0.5) + scale_fill_manual(values = c(\u0026quot;gray\u0026quot;,\n    \u0026quot;#FA871B\u0026quot;)) + theme(axis.text.x = element_text(angle = 90,\n    vjust = 0.5, hjust = 1)) + geom_hline(yintercept = 0.05,\n    linetype = \u0026quot;dashed\u0026quot;) + labs(y = \u0026quot;p-value\u0026quot;, x = \u0026quot;gene\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003emeta_pval\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    arrange(\u003cspan class=\"pl-smi\"\u003e`PC Corrected p-value`\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    rename(\u003cspan class=\"pl-smi\"\u003e`p-value before PC correction`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e`Non PC Corrected p-value`\u003c/span\u003e,\n        \u003cspan class=\"pl-smi\"\u003e`p-value after PC correction`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e`PC Corrected p-value`\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    \u003cspan class=\"pl-e\"\u003ereshape\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003emelt() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e fct_inorder(\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003evalue\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_bar(aes(\u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003evariable\u003c/span\u003e),\n    \u003cspan class=\"pl-v\"\u003estat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eidentity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eposition\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edodge\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e scale_fill_manual(\u003cspan class=\"pl-v\"\u003evalues\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egray\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#FA871B\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e theme(\u003cspan class=\"pl-v\"\u003eaxis.text.x\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003eangle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e90\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003evjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_hline(\u003cspan class=\"pl-v\"\u003eyintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003elinetype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edashed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e labs(\u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ep-value\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/plot p-values-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/plot p-values-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIt appears that after correcting for the first two principal components,\nall genes except for CD22, B3GNT8, B3GAT1, and ALDH1A1 are significant.\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eQuick information about significant genes:\u003c/h4\u003e\u003ca id=\"user-content-quick-information-about-significant-genes\" class=\"anchor\" aria-label=\"Permalink: Quick information about significant genes:\" href=\"#quick-information-about-significant-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eP2RY6: \u003ca href=\"https://pubmed.ncbi.nlm.nih.gov/29454075/\" rel=\"nofollow\"\u003epromotes tumorigenesis by inhibiting\napoptosis\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ePIK3CG: \u003ca href=\"https://pubmed.ncbi.nlm.nih.gov/12473596/\" rel=\"nofollow\"\u003einhibits PI3K-Akt/PKB signaling system responsible for\ntumorigenesis and the progression of colorectal\ncancers\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eGPD1\u003c/strong\u003e: \u003ca href=\"https://en.wikipedia.org/wiki/Glycerol-3-phosphate_dehydrogenase_1\" rel=\"nofollow\"\u003eencodes glycerol-3-phosphate dehydrogenase, which plays a\ncrucial role in carbohydrate and lipid metabolism by catalyzing\nconversion of DHAP and NADG to G3P (glycerol-3-phosphate) and\nNAD+.\u003c/a\u003e\nHas been implicated as a tumor suppressor in \u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5731876/\" rel=\"nofollow\"\u003ebreast\ncancer\u003c/a\u003e and\n\u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9284842/\" rel=\"nofollow\"\u003ebladder\ncancer\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eCSFR1: \u003ca href=\"https://en.wikipedia.org/wiki/Colony_stimulating_factor_1_receptor\" rel=\"nofollow\"\u003einvolved in production, differentiation, and function of\nmacrophages\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eGCNT4\u003c/strong\u003e: \u003ca href=\"https://www.ncbi.nlm.nih.gov/gene/51301\" rel=\"nofollow\"\u003epredicted to be involved in O-glycan processing and\ncarbohydrate metabolic\nprocess.\u003c/a\u003e. \u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7457769/\" rel=\"nofollow\"\u003eFound to be\ndownregulated in gastric\ncancer\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ePRKCB: \u003ca href=\"https://www.ncbi.nlm.nih.gov/gene/5579\" rel=\"nofollow\"\u003eencodes protein involved in B cell activation, apoptosis\ninduction, endothelial cell proliferation, and intestinal sugar\nabsorption.\u003c/a\u003e \u003ca href=\"https://link.springer.com/article/10.1007/s12253-013-9739-5\" rel=\"nofollow\"\u003ePreviously\nimplicated in colorectal\ncancer\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eB4GALNT2: \u003ca href=\"https://www.genecards.org/cgi-bin/carddisp.pl?gene=B4GALNT2\" rel=\"nofollow\"\u003ecatalyzes last step in biosynthesis of human Sd(a) antigen\nand Cad\nantigen.\u003c/a\u003e\n\u003ca href=\"https://pubmed.ncbi.nlm.nih.gov/34589428/\" rel=\"nofollow\"\u003eFound to be highly expressed in model triple negative breast\ncancer\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eCEBPA: \u003ca href=\"https://medlineplus.gov/genetics/gene/cebpa/\" rel=\"nofollow\"\u003eencodes transcription factor involved in maturation of certain\nblood cells and believed to act as a tumor\nsupressor.\u003c/a\u003e \u003ca href=\"https://www.frontiersin.org/articles/10.3389/fonc.2022.806137/full\" rel=\"nofollow\"\u003eImplicated\nin acute myeloid\nleukemia\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eSST: \u003ca href=\"https://www.genecards.org/cgi-bin/carddisp.pl?gene=SST\" rel=\"nofollow\"\u003eregulator of endocrine system through interactions with\npituitary growth hormone, thyroid stimulating hormone, and most\nhormones of GI tract. Also affects rates of neurotransmission in the\nCNS and proliferation of normal and tumorigenic\ncells\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSLC2A4\u003c/strong\u003e: \u003ca href=\"https://www.genecards.org/cgi-bin/carddisp.pl?gene=SLC24A4\" rel=\"nofollow\"\u003eencodes member of K+ dependent Na/Ca exchanger protein\nfamily\u003c/a\u003e\nNot many articles on its implications in cancer.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ePTPRC: \u003ca href=\"https://www.ncbi.nlm.nih.gov/gene/5788\" rel=\"nofollow\"\u003eknown to be signaling molecules that regulate a variety of\ncellular processes including cell growth, differentiation, mitosis,\nand oncogenic transformation\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eGCNT2: \u003ca href=\"https://en.wikipedia.org/wiki/GCNT2\" rel=\"nofollow\"\u003eencodes enzyme responsible for formation of blood group I\nantigen.\u003c/a\u003e \u003ca href=\"https://pubmed.ncbi.nlm.nih.gov/25750292/\" rel=\"nofollow\"\u003eHypomethylated in\ncolorectal cancer\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003eI have selected three genes of interest to me: GPD1, GCNT4, and SLC24A\nfor further analysis.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"p1 \u0026lt;- lm_df %\u0026gt;%\n    select(group, GPD1) %\u0026gt;%\n    group_by(group) %\u0026gt;%\n    summarise(mean = mean(GPD1), sd = sd(GPD1)) %\u0026gt;%\n    ggplot(aes(x = forcats::fct_reorder(factor(group), mean),\n        y = mean, fill = group)) + geom_col() + geom_errorbar(aes(x = group,\n    ymin = mean - sd, ymax = mean + sd), width = 0.4, colour = \u0026quot;orange\u0026quot;,\n    alpha = 0.9, size = 1.3) + geom_text(aes(label = paste(\u0026quot;Average log10(fpkm):\u0026quot;,\n    round(mean, 2), sep = \u0026quot; \u0026quot;), hjust = 0.5, vjust = ifelse(mean \u0026lt;\n    0, 5.5, -2.5)), size = 3) + geom_hline(yintercept = 0) +\n    ylim(-2, 2) + labs(x = \u0026quot;\u0026quot;, y = \u0026quot;Average log(10) FPKM\u0026quot;, title = \u0026quot;GPD1\u0026quot;) +\n    annotate(\u0026quot;text\u0026quot;, label = paste(\u0026quot;p-value:\u0026quot;, round(lmp(lm(GPD1 ~\n        group, lm_df)), 6), sep = \u0026quot; \u0026quot;), x = 2, y = 2)\n\np2 \u0026lt;- lm_df %\u0026gt;%\n    select(group, SLC2A4) %\u0026gt;%\n    group_by(group) %\u0026gt;%\n    summarise(mean = mean(SLC2A4), sd = sd(SLC2A4)) %\u0026gt;%\n    ggplot(aes(x = forcats::fct_reorder(factor(group), mean),\n        y = mean, fill = group)) + geom_col() + geom_errorbar(aes(x = group,\n    ymin = mean - sd, ymax = mean + sd), width = 0.4, colour = \u0026quot;orange\u0026quot;,\n    alpha = 0.9, size = 1.3) + geom_text(aes(label = paste(\u0026quot;Average log10(fpkm):\u0026quot;,\n    round(mean, 2), sep = \u0026quot; \u0026quot;), hjust = 0.5, vjust = ifelse(mean \u0026lt;\n    0, 5.5, -2.5)), size = 3) + geom_hline(yintercept = 0) +\n    ylim(-2, 2) + labs(x = \u0026quot;\u0026quot;, y = \u0026quot;Average log(10) FPKM\u0026quot;, title = \u0026quot;SLC2A4\u0026quot;) +\n    annotate(\u0026quot;text\u0026quot;, label = paste(\u0026quot;p-value:\u0026quot;, round(lmp(lm(SLC2A4 ~\n        group, lm_df)), 6), sep = \u0026quot; \u0026quot;), x = 2, y = 2)\n\n\np3 \u0026lt;- lm_df %\u0026gt;%\n    select(group, GCNT4) %\u0026gt;%\n    group_by(group) %\u0026gt;%\n    summarise(mean = mean(GCNT4), sd = sd(GCNT4)) %\u0026gt;%\n    ggplot(aes(x = forcats::fct_reorder(factor(group), mean),\n        y = mean, fill = group)) + geom_col() + geom_errorbar(aes(x = group,\n    ymin = mean - sd, ymax = mean + sd), width = 0.4, colour = \u0026quot;orange\u0026quot;,\n    alpha = 0.9, size = 1.3) + geom_text(aes(label = paste(\u0026quot;Average log10(fpkm):\u0026quot;,\n    round(mean, 2), sep = \u0026quot; \u0026quot;), hjust = 0.5, vjust = ifelse(mean \u0026lt;\n    0, 5.5, -2.5)), size = 3) + geom_hline(yintercept = 0) +\n    ylim(-2, 2) + labs(x = \u0026quot;\u0026quot;, y = \u0026quot;Average log(10) FPKM\u0026quot;, title = \u0026quot;GCNT4 \u0026quot;) +\n    annotate(\u0026quot;text\u0026quot;, label = paste(\u0026quot;p-value:\u0026quot;, round(lmp(lm(GCNT4 ~\n        group, lm_df)), 6), sep = \u0026quot; \u0026quot;), x = 2, y = 2)\n\ngrid.arrange(p1, p2, p3, ncol = 2)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ep1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003elm_df\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    select(\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eGPD1\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    group_by(\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    summarise(\u003cspan class=\"pl-v\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e mean(\u003cspan class=\"pl-smi\"\u003eGPD1\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e sd(\u003cspan class=\"pl-smi\"\u003eGPD1\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-e\"\u003eforcats\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efct_reorder(\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e), \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e),\n        \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_col() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_errorbar(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eymin\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esd\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eymax\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esd\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorange\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1.3\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_text(aes(\u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAverage log10(fpkm):\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    round(\u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003evjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e ifelse(\u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e5.5\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2.5\u003c/span\u003e)), \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_hline(\u003cspan class=\"pl-v\"\u003eyintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    ylim(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e labs(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAverage log(10) FPKM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGPD1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    annotate(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etext\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ep-value:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, round(lmp(lm(\u003cspan class=\"pl-smi\"\u003eGPD1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\n        \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003elm_df\u003c/span\u003e)), \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003ep2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003elm_df\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    select(\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSLC2A4\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    group_by(\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    summarise(\u003cspan class=\"pl-v\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e mean(\u003cspan class=\"pl-smi\"\u003eSLC2A4\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e sd(\u003cspan class=\"pl-smi\"\u003eSLC2A4\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-e\"\u003eforcats\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efct_reorder(\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e), \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e),\n        \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_col() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_errorbar(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eymin\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esd\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eymax\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esd\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorange\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1.3\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_text(aes(\u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAverage log10(fpkm):\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    round(\u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003evjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e ifelse(\u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e5.5\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2.5\u003c/span\u003e)), \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_hline(\u003cspan class=\"pl-v\"\u003eyintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    ylim(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e labs(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAverage log(10) FPKM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSLC2A4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    annotate(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etext\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ep-value:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, round(lmp(lm(\u003cspan class=\"pl-smi\"\u003eSLC2A4\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\n        \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003elm_df\u003c/span\u003e)), \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-smi\"\u003ep3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003elm_df\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    select(\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eGCNT4\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    group_by(\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    summarise(\u003cspan class=\"pl-v\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e mean(\u003cspan class=\"pl-smi\"\u003eGCNT4\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e sd(\u003cspan class=\"pl-smi\"\u003eGCNT4\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-e\"\u003eforcats\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efct_reorder(\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e), \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e),\n        \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_col() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_errorbar(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eymin\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esd\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eymax\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esd\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorange\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1.3\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_text(aes(\u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAverage log10(fpkm):\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    round(\u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003evjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e ifelse(\u003cspan class=\"pl-smi\"\u003emean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e5.5\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2.5\u003c/span\u003e)), \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_hline(\u003cspan class=\"pl-v\"\u003eyintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    ylim(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e labs(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAverage log(10) FPKM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGCNT4 \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    annotate(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etext\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ep-value:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, round(lmp(lm(\u003cspan class=\"pl-smi\"\u003eGCNT4\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\n        \u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003elm_df\u003c/span\u003e)), \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n\ngrid.arrange(\u003cspan class=\"pl-smi\"\u003ep1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ep2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ep3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003encol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/barplots of individual genes-1.png\"\u003e\u003cimg src=\"/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/barplots of individual genes-1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHere are the barplots of their average expression and standard error\nbetween the three groups. I also included the p-values of their\nsignificance after correcting for principal components 1 and 2, you can\nsee that levels of expression are a lot lower in cancer cell lines and\ntissue compared to the normal groups and their error bars dont overlap.\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSLC2A4\u003c/h4\u003e\u003ca id=\"user-content-slc2a4\" class=\"anchor\" aria-label=\"Permalink: SLC2A4\" href=\"#slc2a4\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSLC2A4 is a member of the solute-carrier family-2 proteins. It encodes\nthe protein GLUT4, which is responsible for the diffusion of circulating\nglucose down its concentration gradient into muscle and fat cells, where\nthey are converted either to glucose-6-phosphate for glycolysis or\nglycogen for storage.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eGLUT4 plays an essential role in maintaining body glucose homeostasis\nand is regulated by insulin\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn a \u003ca href=\"https://pubmed.ncbi.nlm.nih.gov/34488531/\" rel=\"nofollow\"\u003eprevious study\u003c/a\u003e looking\nat the solute-carrier family-2s implications in cancer, they analyzed\nmRNA expression data in 20 cancer types from the Oncomine database and\nfiltered based on a p-value of less than 0.01 and fold change of 1.5.\nThey found that SLC2A4 was significantly downregulated in breast cancer.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTo verify their results, they used a web-based tool called UALCAN and\nfound that mRNA expression of SLC2A4 was significantly downregulated in\nall individual breast cancer stages with p-value \u0026lt; 0.01. I included a\nscreenshot of their results in the top picture.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThey then used a Kaplan-Meier survival analysis and demonstrated that\nhigh expression of SLC2A4 was related to better prognosis in stage I and\nIII breast cancer. I included a picture of their survival curve for\nstage I cancer on the bottom.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLastly, they used a software called MEXPRESS to assess whether DNA\nhypermethylation in the promoter of SLC2A4 was related to its\nexpression. They found that there was a statistical difference when\ncomparing breast cancer samples versus normal samples, suggesting that\nthis is a regulator of mRNA expression.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eOverall, they suggest that GLUT4 downregulation may inhibit glucose\nuptake and induce metabolic reprogramming and also affect downstream\npathways that play key roles in tumorigenesis and progression. But they\nstate that further studies with larger sample sizes are needed for a\nbetter understanding of the mechanisms of SLC2A4 in cancer.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGPD1\u003c/h4\u003e\u003ca id=\"user-content-gpd1\" class=\"anchor\" aria-label=\"Permalink: GPD1\" href=\"#gpd1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eGPD1 plays a role in carbohydrate and lipid metabolism, specifically in\nthe conversion of DHAP and NADH into glycerol-3-phosphate and NAD+.\nTogether with GPD2, the two proteins carry electrons generated by\nglycolysis into the electron transport chain.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn \u003ca href=\"https://jhoonline.biomedcentral.com/articles/10.1186/s13045-022-01312-5\" rel=\"nofollow\"\u003ethis\nstudy\u003c/a\u003e\nthat used 17 pairs of bladder cancer tissue and matched adjacent tissue,\nthey analyzed differential expression using proteomics, and identified\nalmost 100,000 peptides. After filtering with 1.5 fold change difference\nand a p-value of 0.05 as the cutoff, they narrowed the proteins down to\n518 downregulated proteins, and found GPD1 to be the most significantly\ndownregulated, which I highlighted in the volcano plot in figure B.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThey verified their results using a Western blot (figure C on the slide)\nand also used a Kaplan-Meier estimate to observe that GPD1 was\ncorrelated with decreased survival time for breast cancer patients,\nwhich led them to hypothesize that GPD1 may have antitumor functions.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTo summarize their findings: they determined that GPD1 increases a\ncompound called lysoPC through a glycerophospholipid metabolism pathway.\nlysoPC then upregulated expression of TRPV2, causing a calcium influx\nthat inhibits cell proliferation and induces apoptosis.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGCNT4\u003c/h4\u003e\u003ca id=\"user-content-gcnt4\" class=\"anchor\" aria-label=\"Permalink: GCNT4\" href=\"#gcnt4\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eGCNT 4 stands for glucosaminyl (N-acetyl) transferase 4. It is part of a\nfamily, named GCNT1 through 4, and all of them are involved in mucin\ncore structure synthesis and branching. These mucins are the most common\nmacromolecules in mucus and are mainly responsible for its properties.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIts been hypothesized that members of the GCNT family influences cancer\ngenesis by regulating cell growth and apoptosis pathways and have been\nidentified in colon, breast, prostate, and pancreatic cancer.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn \u003ca href=\"https://pubmed.ncbi.nlm.nih.gov/34696660/\" rel=\"nofollow\"\u003ethis paper\u003c/a\u003e, the\nresearchers looked at gene expression data from gastric cancer and\nnon-tumor samples published in GEO. They filtered differentially\nexpressed genes using the criteria of having a log fold change of less\nthan -1.5 and p-value of 0.05 and found GCNT4 as a gene of interest as a\nsuppressor of gastric cancer cell proliferation.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAdditionally, they used a program called TargetScan to predict an\nmicroRNA that binds to the 3UTR of GCNT4 and confirmed in 45 of their\nown gastric cancer samples that this microRNA decreases GCNT4 expression\nwhich in turn upregulated downstream pathways that have been studied in\nthe progression of gastric cancer.\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConclusions and Future Directions\u003c/h3\u003e\u003ca id=\"user-content-conclusions-and-future-directions\" class=\"anchor\" aria-label=\"Permalink: Conclusions and Future Directions\" href=\"#conclusions-and-future-directions\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOverall, my findings of abnormally expressed genes and pathways in\ngastric cancer were consistent with my expectations with many cancer\nrelated pathways being upregulated and many immune cell pathways being\ndownregulated.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAlso, the downregulation of carbohydrate metabolism pathways has been\nobserved in a variety of different cancers, including gastric cancer,\nand the three genes that I looked at are involved in different parts of\ncarbohydrate metabolism pathways.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIts possible that the pathway I looked at was significant in gastric\ncancer because of the genes cross-talk with other pathways that have\nto do with cell cycle or immune cell regulation.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMoving forward, I think that it would be insightful to screen a larger\nand more diverse sample of gastric cancer tissues and include people of\ndifferent ethnicities as well as from the different subtypes of gastric\ncancer so that we can see if we observe any overlap in carbohydrate\nmetabolism pathways. Then, I think that it would be interesting to\nfilter out the pathways that overlap with ones that are abnormally\nexpressed in other types of cancer, so that we can consider any that are\nleft to be possibly gastric cancer-specific.\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"GI Cancer Gene Expression Data Visualization","anchor":"gi-cancer-gene-expression-data-visualization","htmlText":"GI Cancer Gene Expression Data Visualization"},{"level":2,"text":"Introduction","anchor":"introduction","htmlText":"Introduction"},{"level":4,"text":"Plot Read Counts For all 8 Samples","anchor":"plot-read-counts-for-all-8-samples","htmlText":"Plot Read Counts For all 8 Samples"},{"level":4,"text":"Flowchart of bioinformatic workflow","anchor":"flowchart-of-bioinformatic-workflow","htmlText":"Flowchart of bioinformatic workflow"},{"level":4,"text":"Post-Tophat2 Alignment Metrics","anchor":"post-tophat2-alignment-metrics","htmlText":"Post-Tophat2 Alignment Metrics"},{"level":4,"text":"PCA analysis on all genes across all samples","anchor":"pca-analysis-on-all-genes-across-all-samples","htmlText":"PCA analysis on all genes across all samples"},{"level":4,"text":"Boxplots for all gene FPKM values across all samples","anchor":"boxplots-for-all-gene-fpkm-values-across-all-samples","htmlText":"Boxplots for all gene FPKM values across all samples"},{"level":4,"text":"Diagram on filtering criteria for Integrated Pathway Analysis (from QIAGEN)","anchor":"diagram-on-filtering-criteria-for-integrated-pathway-analysis-from-qiagen","htmlText":"Diagram on filtering criteria for Integrated Pathway Analysis (from QIAGEN)"},{"level":4,"text":"PCA with filtered genes only","anchor":"pca-with-filtered-genes-only","htmlText":"PCA with filtered genes only"},{"level":4,"text":"Compute distance measures and clustering methods for hierarchical clustering","anchor":"compute-distance-measures-and-clustering-methods-for-hierarchical-clustering","htmlText":"Compute distance measures and clustering methods for hierarchical clustering"},{"level":4,"text":"Evaluate Dendrograms using p-values","anchor":"evaluate-dendrograms-using-p-values","htmlText":"Evaluate Dendrograms using p-values"},{"level":4,"text":"Compare top 2 dendrograms using tanglegram","anchor":"compare-top-2-dendrograms-using-tanglegram","htmlText":"Compare top 2 dendrograms using tanglegram"},{"level":4,"text":"Plot heatmap of gene expression","anchor":"plot-heatmap-of-gene-expression","htmlText":"Plot heatmap of gene expression"},{"level":4,"text":"Plot heatmap of carbohydrate metabolism pathway.","anchor":"plot-heatmap-of-carbohydrate-metabolism-pathway","htmlText":"Plot heatmap of carbohydrate metabolism pathway."},{"level":4,"text":"Linear model test on individual genes for significance","anchor":"linear-model-test-on-individual-genes-for-significance","htmlText":"Linear model test on individual genes for significance"},{"level":4,"text":"Quick information about significant genes:","anchor":"quick-information-about-significant-genes","htmlText":"Quick information about significant genes:"},{"level":4,"text":"SLC2A4","anchor":"slc2a4","htmlText":"SLC2A4"},{"level":4,"text":"GPD1","anchor":"gpd1","htmlText":"GPD1"},{"level":4,"text":"GCNT4","anchor":"gcnt4","htmlText":"GCNT4"},{"level":3,"text":"Conclusions and Future Directions","anchor":"conclusions-and-future-directions","htmlText":"Conclusions and Future Directions"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fflemm0%2Fgastric_cancer_differential_expression"}}],"overviewFilesProcessingTime":142.575638}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/flemm0/gastric_cancer_differential_expression/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/flemm0/gastric_cancer_differential_expression/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/flemm0/gastric_cancer_differential_expression/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/flemm0/gastric_cancer_differential_expression/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/flemm0/gastric_cancer_differential_expression/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           47 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/flemm0/gastric_cancer_differential_expression/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="IPA-results, (Directory)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/tree/main/IPA-results" title="IPA-results">
                          IPA-results
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="IPA-results, (Directory)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/tree/main/IPA-results" title="IPA-results">
                          IPA-results
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README_files/figure-gfm, (Directory)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/tree/main/README_files/figure-gfm" title="This path skips through empty directories">
                          <span class="react-directory-default-color" data-testid="path-name-segment">
                           README_files/
                          </span>
                          <span class="" data-testid="path-name-segment">
                           figure-gfm
                          </span>
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README_files/figure-gfm, (Directory)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/tree/main/README_files/figure-gfm" title="This path skips through empty directories">
                          <span class="react-directory-default-color" data-testid="path-name-segment">
                           README_files/
                          </span>
                          <span class="" data-testid="path-name-segment">
                           figure-gfm
                          </span>
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="data, (Directory)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/tree/main/data" title="data">
                          data
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="data, (Directory)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/tree/main/data" title="data">
                          data
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.Rmd, (File)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/blob/main/README.Rmd" title="README.Rmd">
                          README.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.Rmd, (File)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/blob/main/README.Rmd" title="README.Rmd">
                          README.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-6">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="server-commands.sh, (File)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/blob/main/server-commands.sh" title="server-commands.sh">
                          server-commands.sh
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="server-commands.sh, (File)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/blob/main/server-commands.sh" title="server-commands.sh">
                          server-commands.sh
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-7">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tool_installation_log.sh, (File)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/blob/main/tool_installation_log.sh" title="tool_installation_log.sh">
                          tool_installation_log.sh
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tool_installation_log.sh, (File)" class="Link--primary" href="/flemm0/gastric_cancer_differential_expression/blob/main/tool_installation_log.sh" title="tool_installation_log.sh">
                          tool_installation_log.sh
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      GI Cancer Gene Expression Data Visualization
                     </h1>
                     <a aria-label="Permalink: GI Cancer Gene Expression Data Visualization" class="anchor" href="#gi-cancer-gene-expression-data-visualization" id="user-content-gi-cancer-gene-expression-data-visualization">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Flemming Wu
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Introduction
                     </h2>
                     <a aria-label="Permalink: Introduction" class="anchor" href="#introduction" id="user-content-introduction">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     In this project, I used publicly available RNA-Seq data from the Gene
Expression Omnibus database (accession number GSE41476) to look at how
differential gene expression between three groups of samples: gastric
cancer tissue, matched adjacent normal tissue, and gastric cancer cell
lines.
                    </p>
                    <p dir="auto">
                     The process of downloading raw sequencing data, alignment, and obtaining
gene expression counts was all done on a HPC server and the commands
used for that can be found in the server_commands.sh file on this
repository. This file describes my downstream analysis steps.
                    </p>
                    <hr/>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(tidyverse)
library(data.table)
library(ggfortify)
library(ggplot2)
library(factoextra)
library(ComplexHeatmap)
library(pvclust)
library(parallel)
library(dendextend)
library(scales)
library(DiagrammeR)
library(gridExtra)
library(tidytext)
library(readxl)
library(circlize)
library(webshot)" dir="auto">
                     <pre>library(<span class="pl-smi">tidyverse</span>)
library(<span class="pl-smi">data.table</span>)
library(<span class="pl-smi">ggfortify</span>)
library(<span class="pl-smi">ggplot2</span>)
library(<span class="pl-smi">factoextra</span>)
library(<span class="pl-smi">ComplexHeatmap</span>)
library(<span class="pl-smi">pvclust</span>)
library(<span class="pl-smi">parallel</span>)
library(<span class="pl-smi">dendextend</span>)
library(<span class="pl-smi">scales</span>)
library(<span class="pl-smi">DiagrammeR</span>)
library(<span class="pl-smi">gridExtra</span>)
library(<span class="pl-smi">tidytext</span>)
library(<span class="pl-smi">readxl</span>)
library(<span class="pl-smi">circlize</span>)
library(<span class="pl-smi">webshot</span>)</pre>
                    </div>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Plot Read Counts For all 8 Samples
                     </h4>
                     <a aria-label="Permalink: Plot Read Counts For all 8 Samples" class="anchor" href="#plot-read-counts-for-all-8-samples" id="user-content-plot-read-counts-for-all-8-samples">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The tissue samples I worked with in my project come from the
bio-resource center of the Asan Medical Center in Seoul. The researchers
collected human gastric cancer samples and adjacent normal tissues. They
also collected gastric cancer cell lines from the Korean Cell Line Bank.
                    </p>
                    <p dir="auto">
                     There were a total of 8 samples between the three groups: 2 normal
gastric tissue samples, 3 gastric cancer tissue samples, and 3 samples
from gastric cancer cell lines. The samples were all paired-end, and
sequenced on Illuminas Genome Analyzer II and stored in SRA format.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='samples &lt;- c("SRR585570", "SRR585571", "SRR585572", "SRR585573",
    "SRR585574", "SRR585575", "SRR585576", "SRR585577")

reads &lt;- c(18883105, 16580585, 27917182, 34687136, 32772474,
    34049244, 31929290, 32457417)

groups &lt;- c("normal gastric tissue", "normal gastric tissue",
    "gastric cancer tissue", "gastric cancer tissue", "gastric cancer tissue",
    "gastric cancer cell line", "gastric cancer cell line", "gastric cancer cell line")

df &lt;- data.frame(samples, reads, groups)

df %&gt;%
    ggplot(aes(y = reads/1e+06, x = samples)) + geom_col(aes(fill = groups)) +
    scale_fill_manual(values = c("#23E7F7", "#2336F7", "#EEF51F")) +
    geom_text(aes(label = format(reads, big.mark = ",")), position = position_dodge(width = 1),
        vjust = -0.5) + geom_hline(yintercept = mean(reads)/1e+06,
    color = "black", linetype = "dotted") + geom_text(aes(0,
    mean(reads)/1e+06, label = paste("Average reads:", format(round(mean(reads)),
        big.mark = ","), sep = " "), vjust = -1, hjust = 0)) +
    labs(x = "Sample", y = "Number of Reads (millions)", fill = "Group")' dir="auto">
                     <pre><span class="pl-smi">samples</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>SRR585570<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR585571<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR585572<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR585573<span class="pl-pds">"</span></span>,
    <span class="pl-s"><span class="pl-pds">"</span>SRR585574<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR585575<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR585576<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR585577<span class="pl-pds">"</span></span>)

<span class="pl-smi">reads</span> <span class="pl-k">&lt;-</span> c(<span class="pl-c1">18883105</span>, <span class="pl-c1">16580585</span>, <span class="pl-c1">27917182</span>, <span class="pl-c1">34687136</span>, <span class="pl-c1">32772474</span>,
    <span class="pl-c1">34049244</span>, <span class="pl-c1">31929290</span>, <span class="pl-c1">32457417</span>)

<span class="pl-smi">groups</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>normal gastric tissue<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>normal gastric tissue<span class="pl-pds">"</span></span>,
    <span class="pl-s"><span class="pl-pds">"</span>gastric cancer tissue<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>gastric cancer tissue<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>gastric cancer tissue<span class="pl-pds">"</span></span>,
    <span class="pl-s"><span class="pl-pds">"</span>gastric cancer cell line<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>gastric cancer cell line<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>gastric cancer cell line<span class="pl-pds">"</span></span>)

<span class="pl-smi">df</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(<span class="pl-smi">samples</span>, <span class="pl-smi">reads</span>, <span class="pl-smi">groups</span>)

<span class="pl-smi">df</span> %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">reads</span><span class="pl-k">/</span><span class="pl-c1">1e+06</span>, <span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">samples</span>)) <span class="pl-k">+</span> geom_col(aes(<span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-smi">groups</span>)) <span class="pl-k">+</span>
    scale_fill_manual(<span class="pl-v">values</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>#23E7F7<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#2336F7<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#EEF51F<span class="pl-pds">"</span></span>)) <span class="pl-k">+</span>
    geom_text(aes(<span class="pl-v">label</span> <span class="pl-k">=</span> format(<span class="pl-smi">reads</span>, <span class="pl-v">big.mark</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>)), <span class="pl-v">position</span> <span class="pl-k">=</span> position_dodge(<span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>),
        <span class="pl-v">vjust</span> <span class="pl-k">=</span> <span class="pl-k">-</span><span class="pl-c1">0.5</span>) <span class="pl-k">+</span> geom_hline(<span class="pl-v">yintercept</span> <span class="pl-k">=</span> mean(<span class="pl-smi">reads</span>)<span class="pl-k">/</span><span class="pl-c1">1e+06</span>,
    <span class="pl-v">color</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-v">linetype</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>dotted<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> geom_text(aes(<span class="pl-c1">0</span>,
    mean(<span class="pl-smi">reads</span>)<span class="pl-k">/</span><span class="pl-c1">1e+06</span>, <span class="pl-v">label</span> <span class="pl-k">=</span> paste(<span class="pl-s"><span class="pl-pds">"</span>Average reads:<span class="pl-pds">"</span></span>, format(round(mean(<span class="pl-smi">reads</span>)),
        <span class="pl-v">big.mark</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">vjust</span> <span class="pl-k">=</span> <span class="pl-k">-</span><span class="pl-c1">1</span>, <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>)) <span class="pl-k">+</span>
    labs(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Sample<span class="pl-pds">"</span></span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Number of Reads (millions)<span class="pl-pds">"</span></span>, <span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Group<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/pre-alignment metric plot-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/pre-alignment metric plot-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     I determined that I was able to use all 8 samples, since they all had a
good number of reads for my purposes. The minimum number of reads was
16.5 million which was in the second normal tissue sample, the maximum
was 34.6 million which was in my second gastric cancer tissue sample,
and all 8 samples averaged at 28.6 million reads.
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Flowchart of bioinformatic workflow
                     </h4>
                     <a aria-label="Permalink: Flowchart of bioinformatic workflow" class="anchor" href="#flowchart-of-bioinformatic-workflow" id="user-content-flowchart-of-bioinformatic-workflow">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="grViz(&quot;digraph flowchart {
      # node definitions with substituted label text
      node [fontname = Arial, shape = rectangle, color = Lavender, style = filled]        
      tab1 [label = '@@1']
      tab2 [label = '@@2']
      tab3 [label = '@@3']
      tab4 [label = '@@4']
      tab5 [label = '@@5']
      tab6 [label = '@@6']
      tab7 [label = '@@7']
      tab8 [label = '@@8']
      tab9 [label = '@@9']

      # edge definitions with the node IDs
      tab1 -&gt; tab3;
      tab2 -&gt; tab3;
      tab3 -&gt; tab4; 
      tab3 -&gt; tab5;
      tab4 -&gt; tab6;
      tab4 -&gt; tab7;
      tab6 -&gt; tab8;
      tab7 -&gt; tab9;
      }

      [1]: 'Download read (FASTQ) files from SRA using SRA-Toolkit'
      [2]: 'Download reference genome (FASTA) and annotation (GTF) files from Ensembl'
      [3]: 'Build index with Bowtie2'
      [4]: 'Align reads to whole genome using Tophat2 (BAM)'
      [5]: 'Align reads to whole transcriptome using Tophat2 (BAM)'
      [6]: 'Obtain FPKM values using Cuffdiff with statistics turned off'
      [7]: 'Obtain p-values and log2 fold differences between normal and cancer tissue using Cuffdiff with statistics on'
      [8]: 'Output file: genes.fpkm_tracking'
      [9]: 'Output file: gene_exp.diff'
      &quot;)" dir="auto">
                     <pre>grViz(<span class="pl-s"><span class="pl-pds">"</span>digraph flowchart {</span>
<span class="pl-s">      # node definitions with substituted label text</span>
<span class="pl-s">      node [fontname = Arial, shape = rectangle, color = Lavender, style = filled]        </span>
<span class="pl-s">      tab1 [label = '@@1']</span>
<span class="pl-s">      tab2 [label = '@@2']</span>
<span class="pl-s">      tab3 [label = '@@3']</span>
<span class="pl-s">      tab4 [label = '@@4']</span>
<span class="pl-s">      tab5 [label = '@@5']</span>
<span class="pl-s">      tab6 [label = '@@6']</span>
<span class="pl-s">      tab7 [label = '@@7']</span>
<span class="pl-s">      tab8 [label = '@@8']</span>
<span class="pl-s">      tab9 [label = '@@9']</span>
<span class="pl-s"></span>
<span class="pl-s">      # edge definitions with the node IDs</span>
<span class="pl-s">      tab1 -&gt; tab3;</span>
<span class="pl-s">      tab2 -&gt; tab3;</span>
<span class="pl-s">      tab3 -&gt; tab4; </span>
<span class="pl-s">      tab3 -&gt; tab5;</span>
<span class="pl-s">      tab4 -&gt; tab6;</span>
<span class="pl-s">      tab4 -&gt; tab7;</span>
<span class="pl-s">      tab6 -&gt; tab8;</span>
<span class="pl-s">      tab7 -&gt; tab9;</span>
<span class="pl-s">      }</span>
<span class="pl-s"></span>
<span class="pl-s">      [1]: 'Download read (FASTQ) files from SRA using SRA-Toolkit'</span>
<span class="pl-s">      [2]: 'Download reference genome (FASTA) and annotation (GTF) files from Ensembl'</span>
<span class="pl-s">      [3]: 'Build index with Bowtie2'</span>
<span class="pl-s">      [4]: 'Align reads to whole genome using Tophat2 (BAM)'</span>
<span class="pl-s">      [5]: 'Align reads to whole transcriptome using Tophat2 (BAM)'</span>
<span class="pl-s">      [6]: 'Obtain FPKM values using Cuffdiff with statistics turned off'</span>
<span class="pl-s">      [7]: 'Obtain p-values and log2 fold differences between normal and cancer tissue using Cuffdiff with statistics on'</span>
<span class="pl-s">      [8]: 'Output file: genes.fpkm_tracking'</span>
<span class="pl-s">      [9]: 'Output file: gene_exp.diff'</span>
<span class="pl-s">      <span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/Flowchart of bioinformatic workflow-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/Flowchart of bioinformatic workflow-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     The above commands were all performed on a HPC cluster. The output
files, genes.fpkm_tracking and gene_exp.diff were the files I used for
downstream analysis in R and are provided in the data folder in this
repository.
                    </p>
                    <p dir="auto">
                     I first downloaded the SRA files as fastq files using the fasterq-dump
command from sra-toolkit. I also downloaded the fasta file for the human
reference genome, version GRCh38, and the GTF file of the genome
annotation, both of which were from Ensembl.
                    </p>
                    <p dir="auto">
                     I then built index files from the reference genome using Bowtie2, which
will be used to speed up the mapping process. Next, I selected one
normal tissue sample and one cancer tissue sample to map to the whole
genome and the whole transcriptome using Tophat2 to see if they mapped
to one much better. It turned out that the mapping percentages for both
genome and transcriptome were both at around 80%, so I just mapped the
rest of my samples to the whole genome.
                    </p>
                    <p dir="auto">
                     Once all the reads were mapped, I took my alignment files, or bam files,
and put them into two different runs of Cuffdiff. In the first run, I
inputted all 8 of my samples and ran Cuffdiff with statistics turned off
to get just the read counts of all the genes. The output for this was in
my genes.fpkm_tracking file. In the second run, I gave Cuffdiff only my
normal and cancer tissue samples as two groups and ran it with the
statistics on to get log2 fold change and p-values for the genes just
between these two groups. The output for this was in the gene_exp.diff
file.
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Post-Tophat2 Alignment Metrics
                     </h4>
                     <a aria-label="Permalink: Post-Tophat2 Alignment Metrics" class="anchor" href="#post-tophat2-alignment-metrics" id="user-content-post-tophat2-alignment-metrics">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='df &lt;- df %&gt;%
    mutate(input = c(37766210, 33161170, 55834364, 69374272,
        65544948, 68098488, 63858580, 64914834), mapped = c(31498668,
        27402515, 45797812, 58072751, 52477113, 55003238, 42375461,
        44831741), pct_mapped = round(mapped/input * 100, 1))

df %&gt;%
    ggplot(aes(y = pct_mapped, x = samples)) + geom_col(aes(fill = groups)) +
    scale_fill_manual(values = c("#23E7F7", "#2336F7", "#EEF51F")) +
    geom_text(aes(label = paste(pct_mapped, "%", sep = " ")),
        position = position_dodge(width = 1), vjust = -0.5) +
    labs(x = "Sample", y = "Percent of Reads Aligned to Genome",
        fill = "Group")' dir="auto">
                     <pre><span class="pl-smi">df</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">df</span> %<span class="pl-k">&gt;</span>%
    mutate(<span class="pl-v">input</span> <span class="pl-k">=</span> c(<span class="pl-c1">37766210</span>, <span class="pl-c1">33161170</span>, <span class="pl-c1">55834364</span>, <span class="pl-c1">69374272</span>,
        <span class="pl-c1">65544948</span>, <span class="pl-c1">68098488</span>, <span class="pl-c1">63858580</span>, <span class="pl-c1">64914834</span>), <span class="pl-v">mapped</span> <span class="pl-k">=</span> c(<span class="pl-c1">31498668</span>,
        <span class="pl-c1">27402515</span>, <span class="pl-c1">45797812</span>, <span class="pl-c1">58072751</span>, <span class="pl-c1">52477113</span>, <span class="pl-c1">55003238</span>, <span class="pl-c1">42375461</span>,
        <span class="pl-c1">44831741</span>), <span class="pl-v">pct_mapped</span> <span class="pl-k">=</span> round(<span class="pl-smi">mapped</span><span class="pl-k">/</span><span class="pl-smi">input</span> <span class="pl-k">*</span> <span class="pl-c1">100</span>, <span class="pl-c1">1</span>))

<span class="pl-smi">df</span> %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">pct_mapped</span>, <span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">samples</span>)) <span class="pl-k">+</span> geom_col(aes(<span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-smi">groups</span>)) <span class="pl-k">+</span>
    scale_fill_manual(<span class="pl-v">values</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>#23E7F7<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#2336F7<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#EEF51F<span class="pl-pds">"</span></span>)) <span class="pl-k">+</span>
    geom_text(aes(<span class="pl-v">label</span> <span class="pl-k">=</span> paste(<span class="pl-smi">pct_mapped</span>, <span class="pl-s"><span class="pl-pds">"</span>%<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>)),
        <span class="pl-v">position</span> <span class="pl-k">=</span> position_dodge(<span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>), <span class="pl-v">vjust</span> <span class="pl-k">=</span> <span class="pl-k">-</span><span class="pl-c1">0.5</span>) <span class="pl-k">+</span>
    labs(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Sample<span class="pl-pds">"</span></span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Percent of Reads Aligned to Genome<span class="pl-pds">"</span></span>,
        <span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Group<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/alignment metrics-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/alignment metrics-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Overall, the Tophat alignment was pretty successful. Most of my samples
have over an 80% mapping rate. The maximum mapping percentage is 83.7%,
and the two samples on the right in the chart are cancer cell lines 2
and 3, they had slightly lower mapping rates than the other samples, at
about 66 and 69%. But, the average across all 8 samples was 78.5%.
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      PCA analysis on all genes across all samples
                     </h4>
                     <a aria-label="Permalink: PCA analysis on all genes across all samples" class="anchor" href="#pca-analysis-on-all-genes-across-all-samples" id="user-content-pca-analysis-on-all-genes-across-all-samples">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='fpkm &lt;- data.table::fread("./data/genes.fpkm_tracking", sep = "\t")

names(fpkm)[26:29] &lt;- names(fpkm)[26:29] %&gt;% # fix column names that were incorrectly named in Cuffdiff command
  str_replace("ct2", "ct3")


# select needed columns

fpkm &lt;- fpkm %&gt;% 
  select(tracking_id, ends_with("_FPKM"))

row.names(fpkm) &lt;- fpkm$tracking_id

fpkm &lt;- fpkm %&gt;% select(ends_with("FPKM"))

fpkm &lt;- t(fpkm)


# normalize everything using log10 conversion

fpkm_norm &lt;- fpkm + 0.1

fpkm_norm &lt;- log10(fpkm_norm)


# create sample column for labeling

fpkm_norm &lt;- as.data.frame(fpkm_norm)
fpkm_norm$samples &lt;- row.names(fpkm_norm)

fpkm_norm.data &lt;- fpkm_norm[-ncol(fpkm_norm)]

fpkm_norm$type &lt;- ifelse(grepl("norm", fpkm_norm$samples), "normal tissue",
                            ifelse(grepl("ct", fpkm_norm$samples), "cancer tissue", "cancer cell line"))' dir="auto">
                     <pre><span class="pl-smi">fpkm</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">data.table</span><span class="pl-k">::</span>fread(<span class="pl-s"><span class="pl-pds">"</span>./data/genes.fpkm_tracking<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\t</span><span class="pl-pds">"</span></span>)

names(<span class="pl-smi">fpkm</span>)[<span class="pl-c1">26</span><span class="pl-k">:</span><span class="pl-c1">29</span>] <span class="pl-k">&lt;-</span> names(<span class="pl-smi">fpkm</span>)[<span class="pl-c1">26</span><span class="pl-k">:</span><span class="pl-c1">29</span>] %<span class="pl-k">&gt;</span>% <span class="pl-c"><span class="pl-c">#</span> fix column names that were incorrectly named in Cuffdiff command</span>
  str_replace(<span class="pl-s"><span class="pl-pds">"</span>ct2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>ct3<span class="pl-pds">"</span></span>)


<span class="pl-c"><span class="pl-c">#</span> select needed columns</span>

<span class="pl-smi">fpkm</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">fpkm</span> %<span class="pl-k">&gt;</span>% 
  select(<span class="pl-smi">tracking_id</span>, ends_with(<span class="pl-s"><span class="pl-pds">"</span>_FPKM<span class="pl-pds">"</span></span>))

row.names(<span class="pl-smi">fpkm</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">fpkm</span><span class="pl-k">$</span><span class="pl-smi">tracking_id</span>

<span class="pl-smi">fpkm</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">fpkm</span> %<span class="pl-k">&gt;</span>% select(ends_with(<span class="pl-s"><span class="pl-pds">"</span>FPKM<span class="pl-pds">"</span></span>))

<span class="pl-smi">fpkm</span> <span class="pl-k">&lt;-</span> t(<span class="pl-smi">fpkm</span>)


<span class="pl-c"><span class="pl-c">#</span> normalize everything using log10 conversion</span>

<span class="pl-smi">fpkm_norm</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">fpkm</span> <span class="pl-k">+</span> <span class="pl-c1">0.1</span>

<span class="pl-smi">fpkm_norm</span> <span class="pl-k">&lt;-</span> log10(<span class="pl-smi">fpkm_norm</span>)


<span class="pl-c"><span class="pl-c">#</span> create sample column for labeling</span>

<span class="pl-smi">fpkm_norm</span> <span class="pl-k">&lt;-</span> as.data.frame(<span class="pl-smi">fpkm_norm</span>)
<span class="pl-smi">fpkm_norm</span><span class="pl-k">$</span><span class="pl-smi">samples</span> <span class="pl-k">&lt;-</span> row.names(<span class="pl-smi">fpkm_norm</span>)

<span class="pl-smi">fpkm_norm.data</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">fpkm_norm</span>[<span class="pl-k">-</span>ncol(<span class="pl-smi">fpkm_norm</span>)]

<span class="pl-smi">fpkm_norm</span><span class="pl-k">$</span><span class="pl-smi">type</span> <span class="pl-k">&lt;-</span> ifelse(grepl(<span class="pl-s"><span class="pl-pds">"</span>norm<span class="pl-pds">"</span></span>, <span class="pl-smi">fpkm_norm</span><span class="pl-k">$</span><span class="pl-smi">samples</span>), <span class="pl-s"><span class="pl-pds">"</span>normal tissue<span class="pl-pds">"</span></span>,
                            ifelse(grepl(<span class="pl-s"><span class="pl-pds">"</span>ct<span class="pl-pds">"</span></span>, <span class="pl-smi">fpkm_norm</span><span class="pl-k">$</span><span class="pl-smi">samples</span>), <span class="pl-s"><span class="pl-pds">"</span>cancer tissue<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>cancer cell line<span class="pl-pds">"</span></span>))</pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pca &lt;- prcomp(fpkm_norm.data, scale. = FALSE)

# pca$x # view principal components

autoplot(pca, scale = FALSE, data = fpkm_norm, colour = "type",
    size = 3) + ggtitle("PCA on 40,295 Genes") + theme_bw()' dir="auto">
                     <pre><span class="pl-smi">pca</span> <span class="pl-k">&lt;-</span> prcomp(<span class="pl-smi">fpkm_norm.data</span>, <span class="pl-v">scale.</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)

<span class="pl-c"><span class="pl-c">#</span> pca$x # view principal components</span>

autoplot(<span class="pl-smi">pca</span>, <span class="pl-v">scale</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">data</span> <span class="pl-k">=</span> <span class="pl-smi">fpkm_norm</span>, <span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>type<span class="pl-pds">"</span></span>,
    <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>) <span class="pl-k">+</span> ggtitle(<span class="pl-s"><span class="pl-pds">"</span>PCA on 40,295 Genes<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> theme_bw()</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/plot pca and edit to scale-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/plot pca and edit to scale-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     You can see that the groups clustered together pretty nicely already,
with the cancer tissue on the right, the normal tissue on the left, and
the cancer cell lines on the bottom. PC1 accounts for 31.2% of the
variance in the data set and seems to separate cancer from non-cancer
groups. PC2 accounts for 24.8% of the variance in the data and seems to
separate tissues and cell lines.
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Boxplots for all gene FPKM values across all samples
                     </h4>
                     <a aria-label="Permalink: Boxplots for all gene FPKM values across all samples" class="anchor" href="#boxplots-for-all-gene-fpkm-values-across-all-samples" id="user-content-boxplots-for-all-gene-fpkm-values-across-all-samples">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="df &lt;- as.data.frame(fpkm)
df &lt;- df[colSums(df) &gt; 0]  # filter out genes with 0 expression across all samples
df &lt;- df + 0.1
df &lt;- log10(df)" dir="auto">
                     <pre><span class="pl-smi">df</span> <span class="pl-k">&lt;-</span> as.data.frame(<span class="pl-smi">fpkm</span>)
<span class="pl-smi">df</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">df</span>[colSums(<span class="pl-smi">df</span>) <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>]  <span class="pl-c"><span class="pl-c">#</span> filter out genes with 0 expression across all samples</span>
<span class="pl-smi">df</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">df</span> <span class="pl-k">+</span> <span class="pl-c1">0.1</span>
<span class="pl-smi">df</span> <span class="pl-k">&lt;-</span> log10(<span class="pl-smi">df</span>)</pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='df %&gt;%
    t() %&gt;%
    reshape::melt() %&gt;%
    arrange(X1) %&gt;%
    rename(sample = X2) %&gt;%
    mutate(group = ifelse(grepl("norm", sample), "normal tissue",
        ifelse(grepl("ct", sample), "cancer tissue", "cancer cell line"))) %&gt;%
    ggplot(aes(x = sample, y = value, fill = group)) + geom_boxplot() +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5,
        hjust = 1)) + labs(x = "Sample", y = "log10(FPKM)", fill = "Group",
    title = "FPKM Distributions for 40,295 Genes") + theme_bw()' dir="auto">
                     <pre><span class="pl-smi">df</span> %<span class="pl-k">&gt;</span>%
    t() %<span class="pl-k">&gt;</span>%
    <span class="pl-e">reshape</span><span class="pl-k">::</span>melt() %<span class="pl-k">&gt;</span>%
    arrange(<span class="pl-smi">X1</span>) %<span class="pl-k">&gt;</span>%
    rename(<span class="pl-v">sample</span> <span class="pl-k">=</span> <span class="pl-smi">X2</span>) %<span class="pl-k">&gt;</span>%
    mutate(<span class="pl-v">group</span> <span class="pl-k">=</span> ifelse(grepl(<span class="pl-s"><span class="pl-pds">"</span>norm<span class="pl-pds">"</span></span>, <span class="pl-smi">sample</span>), <span class="pl-s"><span class="pl-pds">"</span>normal tissue<span class="pl-pds">"</span></span>,
        ifelse(grepl(<span class="pl-s"><span class="pl-pds">"</span>ct<span class="pl-pds">"</span></span>, <span class="pl-smi">sample</span>), <span class="pl-s"><span class="pl-pds">"</span>cancer tissue<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>cancer cell line<span class="pl-pds">"</span></span>))) %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">sample</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">value</span>, <span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-smi">group</span>)) <span class="pl-k">+</span> geom_boxplot() <span class="pl-k">+</span>
    theme(<span class="pl-v">axis.text.x</span> <span class="pl-k">=</span> element_text(<span class="pl-v">angle</span> <span class="pl-k">=</span> <span class="pl-c1">90</span>, <span class="pl-v">vjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>,
        <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>)) <span class="pl-k">+</span> labs(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Sample<span class="pl-pds">"</span></span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>log10(FPKM)<span class="pl-pds">"</span></span>, <span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Group<span class="pl-pds">"</span></span>,
    <span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>FPKM Distributions for 40,295 Genes<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> theme_bw()</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/fpkm boxplots of all samples across all genes-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/fpkm boxplots of all samples across all genes-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     From the plot, it looks like the highest median fpkm is in the normal
tissue group by just a little bit, but the overall distribution of gene
expression looks very similar across all 8 samples.
                    </p>
                    <hr/>
                    <p dir="auto">
                     My next step was to filter genes down based on the log2 fold change and
p-values so that I could have a manageable number of genes to look at
for pathway analysis. The cutoffs I ended up using were a p-value of
less than 0.001 and log 2 fold change that was less than -5 or greater
than 5. The filtering left me with 280 significantly differentially
expressed genes, 198 of them were downregulated and 82 of them were
upregulated.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Diagram on filtering criteria for Integrated Pathway Analysis (from QIAGEN)
                     </h4>
                     <a aria-label="Permalink: Diagram on filtering criteria for Integrated Pathway Analysis (from QIAGEN)" class="anchor" href="#diagram-on-filtering-criteria-for-integrated-pathway-analysis-from-qiagen" id="user-content-diagram-on-filtering-criteria-for-integrated-pathway-analysis-from-qiagen">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="grViz(diagram = &quot;digraph flowchart {
      # define node aesthetics
      node [fontname = Arial, shape = egg, color = DarkGreen, style = filled, fontcolor = White]        
      tab1 [label = '@@1']
      tab2 [label = '@@2']
      tab3 [label = '@@3']
      tab4 [label = '@@4']
      tab5 [label = '@@5']
      tab6 [label = '@@6']
      tab7 [label = '@@7']
# set up node layout
      tab1 -&gt; tab6;
      tab6 -&gt; tab7;
      tab7 -&gt; tab2;
      tab2 -&gt; tab3;
      tab3 -&gt; tab4;
      tab3 -&gt; tab5
      }
      [1]: 'Initial number of genes: 62,635'
      [2]: 'Filter p-value &lt; 0.001 and log2 fold change &lt; -5 or &gt; 5'
      [3]: '280 genes Remaining'
      [4]: '198 genes downregulated'
      [5]: '82 genes upregulated'
      [6]: 'Filter 0 gene expression across all 8 samples'
      [7]: '40,295 genes remaining'
      &quot;)" dir="auto">
                     <pre>grViz(<span class="pl-v">diagram</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>digraph flowchart {</span>
<span class="pl-s">      # define node aesthetics</span>
<span class="pl-s">      node [fontname = Arial, shape = egg, color = DarkGreen, style = filled, fontcolor = White]        </span>
<span class="pl-s">      tab1 [label = '@@1']</span>
<span class="pl-s">      tab2 [label = '@@2']</span>
<span class="pl-s">      tab3 [label = '@@3']</span>
<span class="pl-s">      tab4 [label = '@@4']</span>
<span class="pl-s">      tab5 [label = '@@5']</span>
<span class="pl-s">      tab6 [label = '@@6']</span>
<span class="pl-s">      tab7 [label = '@@7']</span>
<span class="pl-s"># set up node layout</span>
<span class="pl-s">      tab1 -&gt; tab6;</span>
<span class="pl-s">      tab6 -&gt; tab7;</span>
<span class="pl-s">      tab7 -&gt; tab2;</span>
<span class="pl-s">      tab2 -&gt; tab3;</span>
<span class="pl-s">      tab3 -&gt; tab4;</span>
<span class="pl-s">      tab3 -&gt; tab5</span>
<span class="pl-s">      }</span>
<span class="pl-s">      [1]: 'Initial number of genes: 62,635'</span>
<span class="pl-s">      [2]: 'Filter p-value &lt; 0.001 and log2 fold change &lt; -5 or &gt; 5'</span>
<span class="pl-s">      [3]: '280 genes Remaining'</span>
<span class="pl-s">      [4]: '198 genes downregulated'</span>
<span class="pl-s">      [5]: '82 genes upregulated'</span>
<span class="pl-s">      [6]: 'Filter 0 gene expression across all 8 samples'</span>
<span class="pl-s">      [7]: '40,295 genes remaining'</span>
<span class="pl-s">      <span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/gene filtering-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/gene filtering-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Perform filtering of genes according to the above diagram
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='exp &lt;- data.table::fread("./data/gene_exp.diff", sep = "\t")

# the cutoff will be p-val of &lt; 0.001 and log2 fold change greater than 5 or less than -5

exp_filt &lt;- exp %&gt;%
  arrange(`log2(fold_change)`) %&gt;%
  filter(p_value &lt; 0.001) %&gt;%
  filter(`log2(fold_change)` != -Inf, # Inf or -Inf means gene had 0 expression in one of the groups
         `log2(fold_change)` != Inf) %&gt;%
  filter(`log2(fold_change)` &lt;= -5 | `log2(fold_change)` &gt;= 5)' dir="auto">
                     <pre><span class="pl-smi">exp</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">data.table</span><span class="pl-k">::</span>fread(<span class="pl-s"><span class="pl-pds">"</span>./data/gene_exp.diff<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\t</span><span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> the cutoff will be p-val of &lt; 0.001 and log2 fold change greater than 5 or less than -5</span>

<span class="pl-smi">exp_filt</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">exp</span> %<span class="pl-k">&gt;</span>%
  arrange(<span class="pl-smi">`log2(fold_change)`</span>) %<span class="pl-k">&gt;</span>%
  filter(<span class="pl-smi">p_value</span> <span class="pl-k">&lt;</span> <span class="pl-c1">0.001</span>) %<span class="pl-k">&gt;</span>%
  filter(<span class="pl-smi">`log2(fold_change)`</span> <span class="pl-k">!=</span> <span class="pl-k">-</span><span class="pl-c1">Inf</span>, <span class="pl-c"><span class="pl-c">#</span> Inf or -Inf means gene had 0 expression in one of the groups</span>
         <span class="pl-smi">`log2(fold_change)`</span> <span class="pl-k">!=</span> <span class="pl-c1">Inf</span>) %<span class="pl-k">&gt;</span>%
  filter(<span class="pl-smi">`log2(fold_change)`</span> <span class="pl-k">&lt;</span><span class="pl-k">=</span> <span class="pl-k">-</span><span class="pl-c1">5</span> <span class="pl-k">|</span> <span class="pl-smi">`log2(fold_change)`</span> <span class="pl-k">&gt;</span><span class="pl-k">=</span> <span class="pl-c1">5</span>)</pre>
                    </div>
                    <p dir="auto">
                     Separate up and down regulated genes
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="up_genes &lt;- exp_filt %&gt;%
    filter(`log2(fold_change)` &gt; 0) %&gt;%
    select(gene_id, gene)

down_genes &lt;- exp_filt %&gt;%
    filter(`log2(fold_change)` &lt; 0) %&gt;%
    select(gene_id, gene)" dir="auto">
                     <pre><span class="pl-smi">up_genes</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">exp_filt</span> %<span class="pl-k">&gt;</span>%
    filter(<span class="pl-smi">`log2(fold_change)`</span> <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>) %<span class="pl-k">&gt;</span>%
    select(<span class="pl-smi">gene_id</span>, <span class="pl-smi">gene</span>)

<span class="pl-smi">down_genes</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">exp_filt</span> %<span class="pl-k">&gt;</span>%
    filter(<span class="pl-smi">`log2(fold_change)`</span> <span class="pl-k">&lt;</span> <span class="pl-c1">0</span>) %<span class="pl-k">&gt;</span>%
    select(<span class="pl-smi">gene_id</span>, <span class="pl-smi">gene</span>)</pre>
                    </div>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      PCA with filtered genes only
                     </h4>
                     <a aria-label="Permalink: PCA with filtered genes only" class="anchor" href="#pca-with-filtered-genes-only" id="user-content-pca-with-filtered-genes-only">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='fpkm_filt &lt;- data.table::fread("./data/genes.fpkm_tracking", sep = "\t")

names(fpkm_filt)[26:29] &lt;- names(fpkm_filt)[26:29] %&gt;% # fix column names
  str_replace("ct2", "ct3")


fpkm_filt &lt;- fpkm_filt %&gt;% 
  select(tracking_id, ends_with("_FPKM")) %&gt;%
  filter(tracking_id %in% exp_filt$test_id) %&gt;%
  column_to_rownames("tracking_id")


fpkm_filt_norm &lt;- log10(fpkm_filt + 0.1)          

fpkm_filt_norm &lt;- t(fpkm_filt_norm) %&gt;% as.data.frame()

fpkm_filt_norm$sample &lt;- row.names(fpkm_norm)

fpkm_filt_norm &lt;- fpkm_filt_norm %&gt;% 
  mutate(type = ifelse(grepl("norm", sample), "normal tissue",
                         ifelse(grepl("ct", sample), "cancer tissue", "cancer cell line")))


fpkm_filt_norm.data &lt;- fpkm_filt_norm %&gt;% select(-c(sample, type))

pca2 &lt;- prcomp(fpkm_filt_norm.data, scale. = FALSE)

#pca2$x

autoplot(pca2, scale = FALSE, data = fpkm_filt_norm, colour = "type", size = 3) + 
  ggtitle("PCA on 280 Filtered Genes") + 
  theme_bw()' dir="auto">
                     <pre><span class="pl-smi">fpkm_filt</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">data.table</span><span class="pl-k">::</span>fread(<span class="pl-s"><span class="pl-pds">"</span>./data/genes.fpkm_tracking<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\t</span><span class="pl-pds">"</span></span>)

names(<span class="pl-smi">fpkm_filt</span>)[<span class="pl-c1">26</span><span class="pl-k">:</span><span class="pl-c1">29</span>] <span class="pl-k">&lt;-</span> names(<span class="pl-smi">fpkm_filt</span>)[<span class="pl-c1">26</span><span class="pl-k">:</span><span class="pl-c1">29</span>] %<span class="pl-k">&gt;</span>% <span class="pl-c"><span class="pl-c">#</span> fix column names</span>
  str_replace(<span class="pl-s"><span class="pl-pds">"</span>ct2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>ct3<span class="pl-pds">"</span></span>)


<span class="pl-smi">fpkm_filt</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">fpkm_filt</span> %<span class="pl-k">&gt;</span>% 
  select(<span class="pl-smi">tracking_id</span>, ends_with(<span class="pl-s"><span class="pl-pds">"</span>_FPKM<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>%
  filter(<span class="pl-smi">tracking_id</span> <span class="pl-k">%in%</span> <span class="pl-smi">exp_filt</span><span class="pl-k">$</span><span class="pl-smi">test_id</span>) %<span class="pl-k">&gt;</span>%
  column_to_rownames(<span class="pl-s"><span class="pl-pds">"</span>tracking_id<span class="pl-pds">"</span></span>)


<span class="pl-smi">fpkm_filt_norm</span> <span class="pl-k">&lt;-</span> log10(<span class="pl-smi">fpkm_filt</span> <span class="pl-k">+</span> <span class="pl-c1">0.1</span>)          

<span class="pl-smi">fpkm_filt_norm</span> <span class="pl-k">&lt;-</span> t(<span class="pl-smi">fpkm_filt_norm</span>) %<span class="pl-k">&gt;</span>% as.data.frame()

<span class="pl-smi">fpkm_filt_norm</span><span class="pl-k">$</span><span class="pl-smi">sample</span> <span class="pl-k">&lt;-</span> row.names(<span class="pl-smi">fpkm_norm</span>)

<span class="pl-smi">fpkm_filt_norm</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">fpkm_filt_norm</span> %<span class="pl-k">&gt;</span>% 
  mutate(<span class="pl-v">type</span> <span class="pl-k">=</span> ifelse(grepl(<span class="pl-s"><span class="pl-pds">"</span>norm<span class="pl-pds">"</span></span>, <span class="pl-smi">sample</span>), <span class="pl-s"><span class="pl-pds">"</span>normal tissue<span class="pl-pds">"</span></span>,
                         ifelse(grepl(<span class="pl-s"><span class="pl-pds">"</span>ct<span class="pl-pds">"</span></span>, <span class="pl-smi">sample</span>), <span class="pl-s"><span class="pl-pds">"</span>cancer tissue<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>cancer cell line<span class="pl-pds">"</span></span>)))


<span class="pl-smi">fpkm_filt_norm.data</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">fpkm_filt_norm</span> %<span class="pl-k">&gt;</span>% select(<span class="pl-k">-</span>c(<span class="pl-smi">sample</span>, <span class="pl-smi">type</span>))

<span class="pl-smi">pca2</span> <span class="pl-k">&lt;-</span> prcomp(<span class="pl-smi">fpkm_filt_norm.data</span>, <span class="pl-v">scale.</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)

<span class="pl-c"><span class="pl-c">#</span>pca2$x</span>

autoplot(<span class="pl-smi">pca2</span>, <span class="pl-v">scale</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">data</span> <span class="pl-k">=</span> <span class="pl-smi">fpkm_filt_norm</span>, <span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>type<span class="pl-pds">"</span></span>, <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>) <span class="pl-k">+</span> 
  ggtitle(<span class="pl-s"><span class="pl-pds">"</span>PCA on 280 Filtered Genes<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> 
  theme_bw()</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/redo PCA with filtered genes only-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/redo PCA with filtered genes only-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     With the 280 filtered genes, I redid my principal component analysis and
got the plot that you see on the screen. It looks very similar to my
first PCA plot with the unfiltered genes, the three groups also
clustered together and in the same areas, but in this one it looks like
the samples within each group clustered a bit closer together. Also,
principal component 1 now accounts for 62% of the variance in the data,
which is double the principal component 1 from my original PCA plot.
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Compute distance measures and clustering methods for hierarchical clustering
                     </h4>
                     <a aria-label="Permalink: Compute distance measures and clustering methods for hierarchical clustering" class="anchor" href="#compute-distance-measures-and-clustering-methods-for-hierarchical-clustering" id="user-content-compute-distance-measures-and-clustering-methods-for-hierarchical-clustering">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='rownames(fpkm_filt_norm.data) &lt;- fpkm_filt_norm$sample

# try pearson correlation and euclidean distance for
# distance measures
res.pearson.dist &lt;- get_dist(fpkm_filt_norm.data, method = "pearson")
res.euclidean.dist &lt;- get_dist(fpkm_norm.data, method = "euclidean")

# try ward and complete hierarchical clustering methods
res.eucl.ward.hc &lt;- hclust(d = res.euclidean.dist, method = "ward.D2")
res.eucl.complete.hc &lt;- hclust(d = res.euclidean.dist, method = "complete")
res.pear.ward.hc &lt;- hclust(d = res.pearson.dist, method = "ward.D2")
res.pear.complete.hc &lt;- hclust(d = res.pearson.dist, method = "complete")


p1 &lt;- fviz_dend(res.pear.ward.hc, cex = 0.5, k = 3, color_labels_by_k = TRUE,
    rect = TRUE, rect_fill = TRUE, main = paste("Distance:",
        "pearson", "Method:", "ward.D2", sep = " "))


p2 &lt;- fviz_dend(res.pear.complete.hc, cex = 0.5, k = 3, color_labels_by_k = TRUE,
    rect = TRUE, rect_fill = TRUE, main = paste("Distance:",
        "pearson", "Method:", "complete", sep = " "))

p3 &lt;- fviz_dend(res.eucl.ward.hc, cex = 0.5, k = 3, color_labels_by_k = TRUE,
    rect = TRUE, rect_fill = TRUE, main = paste("Distance:",
        "euclidean", "Method:", "ward.D2", sep = " "))

p4 &lt;- fviz_dend(res.eucl.complete.hc, cex = 0.5, k = 3, color_labels_by_k = TRUE,
    rect = TRUE, rect_fill = TRUE, main = paste("Distance:",
        "euclidean", "Method:", "complete", sep = " "))

grid.arrange(p1, p2, p3, p4)' dir="auto">
                     <pre>rownames(<span class="pl-smi">fpkm_filt_norm.data</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">fpkm_filt_norm</span><span class="pl-k">$</span><span class="pl-smi">sample</span>

<span class="pl-c"><span class="pl-c">#</span> try pearson correlation and euclidean distance for</span>
<span class="pl-c"><span class="pl-c">#</span> distance measures</span>
<span class="pl-smi">res.pearson.dist</span> <span class="pl-k">&lt;-</span> get_dist(<span class="pl-smi">fpkm_filt_norm.data</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pearson<span class="pl-pds">"</span></span>)
<span class="pl-smi">res.euclidean.dist</span> <span class="pl-k">&lt;-</span> get_dist(<span class="pl-smi">fpkm_norm.data</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>euclidean<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> try ward and complete hierarchical clustering methods</span>
<span class="pl-smi">res.eucl.ward.hc</span> <span class="pl-k">&lt;-</span> hclust(<span class="pl-v">d</span> <span class="pl-k">=</span> <span class="pl-smi">res.euclidean.dist</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>)
<span class="pl-smi">res.eucl.complete.hc</span> <span class="pl-k">&lt;-</span> hclust(<span class="pl-v">d</span> <span class="pl-k">=</span> <span class="pl-smi">res.euclidean.dist</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>complete<span class="pl-pds">"</span></span>)
<span class="pl-smi">res.pear.ward.hc</span> <span class="pl-k">&lt;-</span> hclust(<span class="pl-v">d</span> <span class="pl-k">=</span> <span class="pl-smi">res.pearson.dist</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>)
<span class="pl-smi">res.pear.complete.hc</span> <span class="pl-k">&lt;-</span> hclust(<span class="pl-v">d</span> <span class="pl-k">=</span> <span class="pl-smi">res.pearson.dist</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>complete<span class="pl-pds">"</span></span>)


<span class="pl-smi">p1</span> <span class="pl-k">&lt;-</span> fviz_dend(<span class="pl-smi">res.pear.ward.hc</span>, <span class="pl-v">cex</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>, <span class="pl-v">k</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">color_labels_by_k</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">rect</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">rect_fill</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">main</span> <span class="pl-k">=</span> paste(<span class="pl-s"><span class="pl-pds">"</span>Distance:<span class="pl-pds">"</span></span>,
        <span class="pl-s"><span class="pl-pds">"</span>pearson<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Method:<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>))


<span class="pl-smi">p2</span> <span class="pl-k">&lt;-</span> fviz_dend(<span class="pl-smi">res.pear.complete.hc</span>, <span class="pl-v">cex</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>, <span class="pl-v">k</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">color_labels_by_k</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">rect</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">rect_fill</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">main</span> <span class="pl-k">=</span> paste(<span class="pl-s"><span class="pl-pds">"</span>Distance:<span class="pl-pds">"</span></span>,
        <span class="pl-s"><span class="pl-pds">"</span>pearson<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Method:<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>complete<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>))

<span class="pl-smi">p3</span> <span class="pl-k">&lt;-</span> fviz_dend(<span class="pl-smi">res.eucl.ward.hc</span>, <span class="pl-v">cex</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>, <span class="pl-v">k</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">color_labels_by_k</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">rect</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">rect_fill</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">main</span> <span class="pl-k">=</span> paste(<span class="pl-s"><span class="pl-pds">"</span>Distance:<span class="pl-pds">"</span></span>,
        <span class="pl-s"><span class="pl-pds">"</span>euclidean<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Method:<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>))

<span class="pl-smi">p4</span> <span class="pl-k">&lt;-</span> fviz_dend(<span class="pl-smi">res.eucl.complete.hc</span>, <span class="pl-v">cex</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>, <span class="pl-v">k</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">color_labels_by_k</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">rect</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">rect_fill</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">main</span> <span class="pl-k">=</span> paste(<span class="pl-s"><span class="pl-pds">"</span>Distance:<span class="pl-pds">"</span></span>,
        <span class="pl-s"><span class="pl-pds">"</span>euclidean<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Method:<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>complete<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>))

grid.arrange(<span class="pl-smi">p1</span>, <span class="pl-smi">p2</span>, <span class="pl-smi">p3</span>, <span class="pl-smi">p4</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/Dendrogram of differentially expressed genes ward method-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/Dendrogram of differentially expressed genes ward method-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     It appears that using euclidean distance along with Wards method gives
us the clustering that we expect to see given our knowledge of the three
groups.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='fviz_dend(res.eucl.ward.hc, cex = 0.75, k = 3, color_labels_by_k = TRUE,
    k_colors = c("blue", "red", "darkgreen"), rect = TRUE, rect_fill = TRUE,
    show_labels = TRUE, main = "Hierarchical Clustering of Samples Based on 280 Filtered Genes",
    xlab = "Samples")' dir="auto">
                     <pre>fviz_dend(<span class="pl-smi">res.eucl.ward.hc</span>, <span class="pl-v">cex</span> <span class="pl-k">=</span> <span class="pl-c1">0.75</span>, <span class="pl-v">k</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">color_labels_by_k</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">k_colors</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>darkgreen<span class="pl-pds">"</span></span>), <span class="pl-v">rect</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">rect_fill</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">show_labels</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Hierarchical Clustering of Samples Based on 280 Filtered Genes<span class="pl-pds">"</span></span>,
    <span class="pl-v">xlab</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Samples<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/visualize p3-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/visualize p3-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     The top separation separated cancer from non-cancer, and within the
cancer cluster theres a separation between cancer tissue and cancer
cell lines.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Evaluate Dendrograms using p-values
                     </h4>
                     <a aria-label="Permalink: Evaluate Dendrograms using p-values" class="anchor" href="#evaluate-dendrograms-using-p-values" id="user-content-evaluate-dendrograms-using-p-values">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# create cluster for parallel computing
n.cores &lt;- detectCores() * 3/4
cl &lt;- makePSOCKcluster(n.cores)
clusterSetRNGStream(cl, 1234)

# calculate p-value dendrogram for pearson correlation and
# ward.d2 method
res.pear.ward.pv &lt;- parPvclust(cl, t(fpkm_filt_norm.data), method.hclust = "ward.D2",
    method.dist = "correlation", nboot = 1000)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> create cluster for parallel computing</span>
<span class="pl-smi">n.cores</span> <span class="pl-k">&lt;-</span> detectCores() <span class="pl-k">*</span> <span class="pl-c1">3</span><span class="pl-k">/</span><span class="pl-c1">4</span>
<span class="pl-smi">cl</span> <span class="pl-k">&lt;-</span> makePSOCKcluster(<span class="pl-smi">n.cores</span>)
clusterSetRNGStream(<span class="pl-smi">cl</span>, <span class="pl-c1">1234</span>)

<span class="pl-c"><span class="pl-c">#</span> calculate p-value dendrogram for pearson correlation and</span>
<span class="pl-c"><span class="pl-c">#</span> ward.d2 method</span>
<span class="pl-smi">res.pear.ward.pv</span> <span class="pl-k">&lt;-</span> parPvclust(<span class="pl-smi">cl</span>, t(<span class="pl-smi">fpkm_filt_norm.data</span>), <span class="pl-v">method.hclust</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>,
    <span class="pl-v">method.dist</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>correlation<span class="pl-pds">"</span></span>, <span class="pl-v">nboot</span> <span class="pl-k">=</span> <span class="pl-c1">1000</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Multiscale bootstrap... Done.">
                     <pre class="notranslate"><code>## Multiscale bootstrap... Done.
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# calculate p-value dendrogram for pearson correlation and
# complete method
res.pear.complete.pv &lt;- parPvclust(cl, t(fpkm_filt_norm.data),
    method.hclust = "complete", method.dist = "correlation",
    nboot = 1000)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> calculate p-value dendrogram for pearson correlation and</span>
<span class="pl-c"><span class="pl-c">#</span> complete method</span>
<span class="pl-smi">res.pear.complete.pv</span> <span class="pl-k">&lt;-</span> parPvclust(<span class="pl-smi">cl</span>, t(<span class="pl-smi">fpkm_filt_norm.data</span>),
    <span class="pl-v">method.hclust</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>complete<span class="pl-pds">"</span></span>, <span class="pl-v">method.dist</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>correlation<span class="pl-pds">"</span></span>,
    <span class="pl-v">nboot</span> <span class="pl-k">=</span> <span class="pl-c1">1000</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Multiscale bootstrap... Done.">
                     <pre class="notranslate"><code>## Multiscale bootstrap... Done.
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# calculate p-value dendrogram for euclidean dist and
# ward.d2 method
res.eucl.complete.pv &lt;- parPvclust(cl, t(fpkm_filt_norm.data),
    method.hclust = "ward.D2", method.dist = "euclidean", nboot = 1000)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> calculate p-value dendrogram for euclidean dist and</span>
<span class="pl-c"><span class="pl-c">#</span> ward.d2 method</span>
<span class="pl-smi">res.eucl.complete.pv</span> <span class="pl-k">&lt;-</span> parPvclust(<span class="pl-smi">cl</span>, t(<span class="pl-smi">fpkm_filt_norm.data</span>),
    <span class="pl-v">method.hclust</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>, <span class="pl-v">method.dist</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>euclidean<span class="pl-pds">"</span></span>, <span class="pl-v">nboot</span> <span class="pl-k">=</span> <span class="pl-c1">1000</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Multiscale bootstrap... Done.">
                     <pre class="notranslate"><code>## Multiscale bootstrap... Done.
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="stopCluster(cl)

# plot Clusters with AU &gt;= 95% are considered to be
# strongly supported by data.
plot(res.pear.ward.pv, hang = -1, cex = 0.5)" dir="auto">
                     <pre>stopCluster(<span class="pl-smi">cl</span>)

<span class="pl-c"><span class="pl-c">#</span> plot Clusters with AU &gt;= 95% are considered to be</span>
<span class="pl-c"><span class="pl-c">#</span> strongly supported by data.</span>
plot(<span class="pl-smi">res.pear.ward.pv</span>, <span class="pl-v">hang</span> <span class="pl-k">=</span> <span class="pl-k">-</span><span class="pl-c1">1</span>, <span class="pl-v">cex</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/p-values for ward clustering-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/p-values for ward clustering-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="plot(res.pear.complete.pv, hang = -1, cex = 0.5)" dir="auto">
                     <pre>plot(<span class="pl-smi">res.pear.complete.pv</span>, <span class="pl-v">hang</span> <span class="pl-k">=</span> <span class="pl-k">-</span><span class="pl-c1">1</span>, <span class="pl-v">cex</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/p-values for ward clustering-2.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/p-values for ward clustering-2.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="plot(res.eucl.complete.pv, hang = -1, cex = 0.5)" dir="auto">
                     <pre>plot(<span class="pl-smi">res.eucl.complete.pv</span>, <span class="pl-v">hang</span> <span class="pl-k">=</span> <span class="pl-k">-</span><span class="pl-c1">1</span>, <span class="pl-v">cex</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/p-values for ward clustering-3.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/p-values for ward clustering-3.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Above, it appears that using pearson correlation and complete clustering
method clustered ccl3_FPKM poorly. The AU values for using both pearson
and euclidean along with Wards method gave stronger AU values.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Compare top 2 dendrograms using tanglegram
                     </h4>
                     <a aria-label="Permalink: Compare top 2 dendrograms using tanglegram" class="anchor" href="#compare-top-2-dendrograms-using-tanglegram" id="user-content-compare-top-2-dendrograms-using-tanglegram">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='dend1 &lt;- as.dendrogram(res.pear.ward.hc)
dend2 &lt;- as.dendrogram(res.eucl.ward.hc)

dendlist(dend1, dend2) %&gt;%
    untangle(method = "step1side") %&gt;%
    tanglegram(main_left = "Pearson Correlation", main_right = "Euclidean Distance",
        main = "Ward.D2", lwd = 1, cex_main = 1, margin_inner = 5)' dir="auto">
                     <pre><span class="pl-smi">dend1</span> <span class="pl-k">&lt;-</span> as.dendrogram(<span class="pl-smi">res.pear.ward.hc</span>)
<span class="pl-smi">dend2</span> <span class="pl-k">&lt;-</span> as.dendrogram(<span class="pl-smi">res.eucl.ward.hc</span>)

dendlist(<span class="pl-smi">dend1</span>, <span class="pl-smi">dend2</span>) %<span class="pl-k">&gt;</span>%
    untangle(<span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>step1side<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>%
    tanglegram(<span class="pl-v">main_left</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Pearson Correlation<span class="pl-pds">"</span></span>, <span class="pl-v">main_right</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Euclidean Distance<span class="pl-pds">"</span></span>,
        <span class="pl-v">main</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Ward.D2<span class="pl-pds">"</span></span>, <span class="pl-v">lwd</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>, <span class="pl-v">cex_main</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>, <span class="pl-v">margin_inner</span> <span class="pl-k">=</span> <span class="pl-c1">5</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/view tanglegram of top clusters-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/view tanglegram of top clusters-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Using pearson correlation for distance metric instead of euclidean
results in cancer cell line 2 clustering with the normal tissues instead
of with the other cancer cell line samples. I have determined that using
euclidean distance and wards method to be the best choices with my
data.
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Plot heatmap of gene expression
                     </h4>
                     <a aria-label="Permalink: Plot heatmap of gene expression" class="anchor" href="#plot-heatmap-of-gene-expression" id="user-content-plot-heatmap-of-gene-expression">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='boxplot &lt;- rowAnnotation(`log10(fpkm)` = anno_boxplot(data.matrix(fpkm_filt_norm.data)),
    gp = gpar(fill = "#CCCCCC"))

hmap &lt;- Heatmap(data.matrix(fpkm_filt_norm.data), name = "Expression",
    show_column_names = FALSE, row_split = 3, clustering_method_rows = "ward.D2",
    clustering_method_columns = "ward.D2", right_annotation = boxplot)

draw(hmap)' dir="auto">
                     <pre><span class="pl-smi">boxplot</span> <span class="pl-k">&lt;-</span> rowAnnotation(<span class="pl-smi">`log10(fpkm)`</span> <span class="pl-k">=</span> anno_boxplot(data.matrix(<span class="pl-smi">fpkm_filt_norm.data</span>)),
    <span class="pl-v">gp</span> <span class="pl-k">=</span> gpar(<span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>#CCCCCC<span class="pl-pds">"</span></span>))

<span class="pl-smi">hmap</span> <span class="pl-k">&lt;-</span> Heatmap(data.matrix(<span class="pl-smi">fpkm_filt_norm.data</span>), <span class="pl-v">name</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Expression<span class="pl-pds">"</span></span>,
    <span class="pl-v">show_column_names</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">row_split</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">clustering_method_rows</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>,
    <span class="pl-v">clustering_method_columns</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>, <span class="pl-v">right_annotation</span> <span class="pl-k">=</span> <span class="pl-smi">boxplot</span>)

draw(<span class="pl-smi">hmap</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/annotated heatmap wards-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/annotated heatmap wards-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     You can see that there is a large block of genes that are expressed
highly in normal tissue and a block of genes expressed highly in cancer
tissue. The cell line cluster seems to be spread out, with some highly
expressed genes overlapping with normal tissue and some overlapping with
cancerous tissue. On the right side, I also included boxplots of the
overall fpkm values and you can see that the expression levels are
higher overall for the normal samples, compared to the cancer samples
which makes sense because now Im working with more genes that are
downregulated in cancer than upregulated
                    </p>
                    <hr/>
                    <p dir="auto">
                     At this point, I looked to the list of upregulated and downregulated
genes from tthe integrated pathway analysis (software from QIAGEN) to
find a pathway to look at more in depth. There were many significant
pathways returned, many of the down regulated ones were involved in
immune system function whereas many of the up regulated pathways have
been previously implicated in cancer. (The excel files are located in
the IPA-results folder.)
                    </p>
                    <p dir="auto">
                     Moving forward, I will be look at the pathway title metabolism of
carbohydrates, which was downregulated in cancerous samples, compared
to normal samples.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Plot heatmap of carbohydrate metabolism pathway.
                     </h4>
                     <a aria-label="Permalink: Plot heatmap of carbohydrate metabolism pathway." class="anchor" href="#plot-heatmap-of-carbohydrate-metabolism-pathway" id="user-content-plot-heatmap-of-carbohydrate-metabolism-pathway">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='meta &lt;- read_xls("./IPA-results/down-genes_diseases_and_functions.xls",
    col_names = TRUE) %&gt;%
    filter(`Diseases or Functions Annotation` == "Metabolism of carbohydrate") %&gt;%
    unnest_tokens(genes, Molecules) %&gt;%
    mutate(genes = toupper(genes)) %&gt;%
    select(genes)

meta %&gt;%
    head(22)' dir="auto">
                     <pre><span class="pl-smi">meta</span> <span class="pl-k">&lt;-</span> read_xls(<span class="pl-s"><span class="pl-pds">"</span>./IPA-results/down-genes_diseases_and_functions.xls<span class="pl-pds">"</span></span>,
    <span class="pl-v">col_names</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>) %<span class="pl-k">&gt;</span>%
    filter(<span class="pl-smi">`Diseases or Functions Annotation`</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>Metabolism of carbohydrate<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>%
    unnest_tokens(<span class="pl-smi">genes</span>, <span class="pl-smi">Molecules</span>) %<span class="pl-k">&gt;</span>%
    mutate(<span class="pl-v">genes</span> <span class="pl-k">=</span> toupper(<span class="pl-smi">genes</span>)) %<span class="pl-k">&gt;</span>%
    select(<span class="pl-smi">genes</span>)

<span class="pl-smi">meta</span> %<span class="pl-k">&gt;</span>%
    head(<span class="pl-c1">22</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## # A tibble: 22  1
##    genes   
##    &lt;chr&gt;   
##  1 AKR1B10 
##  2 ALDH1A1 
##  3 B3GALT5 
##  4 B3GAT1  
##  5 B3GNT8  
##  6 B4GALNT2
##  7 CD22    
##  8 CEBPA   
##  9 CSF1R   
## 10 GCNT2   
## #  with 12 more rows">
                     <pre class="notranslate"><code>## # A tibble: 22  1
##    genes   
##    &lt;chr&gt;   
##  1 AKR1B10 
##  2 ALDH1A1 
##  3 B3GALT5 
##  4 B3GAT1  
##  5 B3GNT8  
##  6 B4GALNT2
##  7 CD22    
##  8 CEBPA   
##  9 CSF1R   
## 10 GCNT2   
## #  with 12 more rows
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='fpkm &lt;- data.table::fread("./data/genes.fpkm_tracking", sep = "\t")

names(fpkm)[26:29] &lt;- names(fpkm)[26:29] %&gt;% # fix column names
  str_replace("ct2", "ct3")


fpkm &lt;- fpkm %&gt;% 
  select(gene_short_name, ends_with("_FPKM"))

meta &lt;- merge(x = meta,
      y = fpkm, 
      all.x = TRUE,
      all.y = FALSE,
      by.x = "genes",
      by.y = "gene_short_name")


meta &lt;- meta %&gt;% 
  column_to_rownames("genes") %&gt;% 
  t()


meta_norm &lt;- log10(meta + 0.1)' dir="auto">
                     <pre><span class="pl-smi">fpkm</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">data.table</span><span class="pl-k">::</span>fread(<span class="pl-s"><span class="pl-pds">"</span>./data/genes.fpkm_tracking<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\t</span><span class="pl-pds">"</span></span>)

names(<span class="pl-smi">fpkm</span>)[<span class="pl-c1">26</span><span class="pl-k">:</span><span class="pl-c1">29</span>] <span class="pl-k">&lt;-</span> names(<span class="pl-smi">fpkm</span>)[<span class="pl-c1">26</span><span class="pl-k">:</span><span class="pl-c1">29</span>] %<span class="pl-k">&gt;</span>% <span class="pl-c"><span class="pl-c">#</span> fix column names</span>
  str_replace(<span class="pl-s"><span class="pl-pds">"</span>ct2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>ct3<span class="pl-pds">"</span></span>)


<span class="pl-smi">fpkm</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">fpkm</span> %<span class="pl-k">&gt;</span>% 
  select(<span class="pl-smi">gene_short_name</span>, ends_with(<span class="pl-s"><span class="pl-pds">"</span>_FPKM<span class="pl-pds">"</span></span>))

<span class="pl-smi">meta</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">meta</span>,
      <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">fpkm</span>, 
      <span class="pl-v">all.x</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
      <span class="pl-v">all.y</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>,
      <span class="pl-v">by.x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>genes<span class="pl-pds">"</span></span>,
      <span class="pl-v">by.y</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gene_short_name<span class="pl-pds">"</span></span>)


<span class="pl-smi">meta</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">meta</span> %<span class="pl-k">&gt;</span>% 
  column_to_rownames(<span class="pl-s"><span class="pl-pds">"</span>genes<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  t()


<span class="pl-smi">meta_norm</span> <span class="pl-k">&lt;-</span> log10(<span class="pl-smi">meta</span> <span class="pl-k">+</span> <span class="pl-c1">0.1</span>)</pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='Heatmap(meta_norm, name = "expression", show_column_names = TRUE,
    row_split = 3, clustering_method_rows = "ward.D2", clustering_method_columns = "ward.D2") +
    rowAnnotation(`log10(fpkm)` = anno_boxplot(meta_norm)) +
    rowAnnotation(rn = anno_text(rownames(meta_norm)))' dir="auto">
                     <pre>Heatmap(<span class="pl-smi">meta_norm</span>, <span class="pl-v">name</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>expression<span class="pl-pds">"</span></span>, <span class="pl-v">show_column_names</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">row_split</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">clustering_method_rows</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>, <span class="pl-v">clustering_method_columns</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
    rowAnnotation(<span class="pl-smi">`log10(fpkm)`</span> <span class="pl-k">=</span> anno_boxplot(<span class="pl-smi">meta_norm</span>)) <span class="pl-k">+</span>
    rowAnnotation(<span class="pl-v">rn</span> <span class="pl-k">=</span> anno_text(rownames(<span class="pl-smi">meta_norm</span>)))</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/annotated heatmap-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/annotated heatmap-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Using the same clustering method (euclidean distance and ward.D2 method)
it appears that cancer cell line 2 becomes its own cluster, and cancer
cell lines 1 and 3 cluster with the cancer tissue.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# compare other clustering methods with original to check
# to see if it is still the best method for the selected
# pathway

# original clustering method
ht1 &lt;- Heatmap(meta_norm, name = "ht1", show_column_names = TRUE,
    row_split = 3, clustering_method_rows = "ward.D2", clustering_method_columns = "ward.D2")


# pearson correlation with ward.D2 clustering
ht2 &lt;- Heatmap(meta_norm, name = "ht2", show_column_names = TRUE,
    row_split = 3, clustering_distance_rows = "pearson", clustering_distance_columns = "pearson",
    clustering_method_rows = "ward.D2", clustering_method_columns = "ward.D2",
    column_names_gp = gpar(fontsize = 9))

# euclidean distance with complete clustering

ht3 &lt;- Heatmap(meta_norm, name = "ht3", show_column_names = TRUE,
    row_split = 3, clustering_distance_rows = "euclidean", clustering_distance_columns = "euclidean",
    clustering_method_rows = "complete", clustering_method_columns = "complete",
    column_names_gp = gpar(fontsize = 9))

# pearson correlation with ward.D2 clustering

ht4 &lt;- Heatmap(meta_norm, name = "ht4", show_column_names = TRUE,
    row_split = 3, clustering_distance_rows = "pearson", clustering_distance_columns = "pearson",
    clustering_method_rows = "complete", clustering_method_columns = "complete",
    column_names_gp = gpar(fontsize = 9))

ht1 + ht2 + ht3 + ht4' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> compare other clustering methods with original to check</span>
<span class="pl-c"><span class="pl-c">#</span> to see if it is still the best method for the selected</span>
<span class="pl-c"><span class="pl-c">#</span> pathway</span>

<span class="pl-c"><span class="pl-c">#</span> original clustering method</span>
<span class="pl-smi">ht1</span> <span class="pl-k">&lt;-</span> Heatmap(<span class="pl-smi">meta_norm</span>, <span class="pl-v">name</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ht1<span class="pl-pds">"</span></span>, <span class="pl-v">show_column_names</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">row_split</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">clustering_method_rows</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>, <span class="pl-v">clustering_method_columns</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>)


<span class="pl-c"><span class="pl-c">#</span> pearson correlation with ward.D2 clustering</span>
<span class="pl-smi">ht2</span> <span class="pl-k">&lt;-</span> Heatmap(<span class="pl-smi">meta_norm</span>, <span class="pl-v">name</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ht2<span class="pl-pds">"</span></span>, <span class="pl-v">show_column_names</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">row_split</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">clustering_distance_rows</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pearson<span class="pl-pds">"</span></span>, <span class="pl-v">clustering_distance_columns</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pearson<span class="pl-pds">"</span></span>,
    <span class="pl-v">clustering_method_rows</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>, <span class="pl-v">clustering_method_columns</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ward.D2<span class="pl-pds">"</span></span>,
    <span class="pl-v">column_names_gp</span> <span class="pl-k">=</span> gpar(<span class="pl-v">fontsize</span> <span class="pl-k">=</span> <span class="pl-c1">9</span>))

<span class="pl-c"><span class="pl-c">#</span> euclidean distance with complete clustering</span>

<span class="pl-smi">ht3</span> <span class="pl-k">&lt;-</span> Heatmap(<span class="pl-smi">meta_norm</span>, <span class="pl-v">name</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ht3<span class="pl-pds">"</span></span>, <span class="pl-v">show_column_names</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">row_split</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">clustering_distance_rows</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>euclidean<span class="pl-pds">"</span></span>, <span class="pl-v">clustering_distance_columns</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>euclidean<span class="pl-pds">"</span></span>,
    <span class="pl-v">clustering_method_rows</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>complete<span class="pl-pds">"</span></span>, <span class="pl-v">clustering_method_columns</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>complete<span class="pl-pds">"</span></span>,
    <span class="pl-v">column_names_gp</span> <span class="pl-k">=</span> gpar(<span class="pl-v">fontsize</span> <span class="pl-k">=</span> <span class="pl-c1">9</span>))

<span class="pl-c"><span class="pl-c">#</span> pearson correlation with ward.D2 clustering</span>

<span class="pl-smi">ht4</span> <span class="pl-k">&lt;-</span> Heatmap(<span class="pl-smi">meta_norm</span>, <span class="pl-v">name</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ht4<span class="pl-pds">"</span></span>, <span class="pl-v">show_column_names</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">row_split</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">clustering_distance_rows</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pearson<span class="pl-pds">"</span></span>, <span class="pl-v">clustering_distance_columns</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pearson<span class="pl-pds">"</span></span>,
    <span class="pl-v">clustering_method_rows</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>complete<span class="pl-pds">"</span></span>, <span class="pl-v">clustering_method_columns</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>complete<span class="pl-pds">"</span></span>,
    <span class="pl-v">column_names_gp</span> <span class="pl-k">=</span> gpar(<span class="pl-v">fontsize</span> <span class="pl-k">=</span> <span class="pl-c1">9</span>))

<span class="pl-smi">ht1</span> <span class="pl-k">+</span> <span class="pl-smi">ht2</span> <span class="pl-k">+</span> <span class="pl-smi">ht3</span> <span class="pl-k">+</span> <span class="pl-smi">ht4</span></pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/other clustering method testing-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/other clustering method testing-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     When plotting all 4 heatmaps together with the original heatmap as the
main, the separation is clearest with the original. Individually, ht2
makes cell line 3 its own cluster and clusters cell line 2 and cancer
tissue 3 with normal, ht2 is similar to original heatmap, and ht3 makes
cell line 1 and 3 are their own clusters. I conclude that the original
clustering method is still the best option for this pathway.
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Linear model test on individual genes for significance
                     </h4>
                     <a aria-label="Permalink: Linear model test on individual genes for significance" class="anchor" href="#linear-model-test-on-individual-genes-for-significance" id="user-content-linear-model-test-on-individual-genes-for-significance">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# add PC1 and PC2 to the data frame for correction

lm_df &lt;- merge(x = pca$x, y = meta_norm, by = "row.names") %&gt;%
    select(-c(4:9)) %&gt;%
    rename(group = Row.names) %&gt;%
    mutate(group = if_else(grepl("norm", group), "normal tissue",
        if_else(grepl("ct", group), "cancer tissue", "cancer cell line")))

# lm_df' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> add PC1 and PC2 to the data frame for correction</span>

<span class="pl-smi">lm_df</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">pca</span><span class="pl-k">$</span><span class="pl-smi">x</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">meta_norm</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>row.names<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>%
    select(<span class="pl-k">-</span>c(<span class="pl-c1">4</span><span class="pl-k">:</span><span class="pl-c1">9</span>)) %<span class="pl-k">&gt;</span>%
    rename(<span class="pl-v">group</span> <span class="pl-k">=</span> <span class="pl-smi">Row.names</span>) %<span class="pl-k">&gt;</span>%
    mutate(<span class="pl-v">group</span> <span class="pl-k">=</span> if_else(grepl(<span class="pl-s"><span class="pl-pds">"</span>norm<span class="pl-pds">"</span></span>, <span class="pl-smi">group</span>), <span class="pl-s"><span class="pl-pds">"</span>normal tissue<span class="pl-pds">"</span></span>,
        if_else(grepl(<span class="pl-s"><span class="pl-pds">"</span>ct<span class="pl-pds">"</span></span>, <span class="pl-smi">group</span>), <span class="pl-s"><span class="pl-pds">"</span>cancer tissue<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>cancer cell line<span class="pl-pds">"</span></span>)))

<span class="pl-c"><span class="pl-c">#</span> lm_df</span></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# function to return p-value from lm object
lmp &lt;- function(modelobject) {
    if (class(modelobject) != &quot;lm&quot;)
        stop(&quot;Not an object of class 'lm' &quot;)
    f &lt;- summary(modelobject)$fstatistic
    p &lt;- pf(f[1], f[2], f[3], lower.tail = F)
    attributes(p) &lt;- NULL
    return(p)
}


# lm test on all genes
no_correction &lt;- list()  # initialize empty list to hold p-values for non-corrected lm test
correction &lt;- list()  # empty list to hold p-values for corrected lm test
for (i in names(lm_df[4:length(names(lm_df))])) {
    no_correction[i] &lt;- lmp(lm(get(i) ~ group, lm_df))
    correction[i] &lt;- lmp(lm(get(i) ~ group + PC1 + PC2, lm_df))
}


# store results in data frame
meta_pval &lt;- rbind(as.data.frame(no_correction), as.data.frame(correction)) %&gt;%
    t() %&gt;%
    as.data.frame() %&gt;%
    rename(`Non PC Corrected p-value` = V1, `PC Corrected p-value` = V2) %&gt;%
    format(scientific = FALSE) %&gt;%
    rownames_to_column(&quot;gene&quot;)

meta_pval &lt;- meta_pval %&gt;%
    mutate(`Non PC Corrected p-value` = as.numeric(`Non PC Corrected p-value`),
        `PC Corrected p-value` = as.numeric(`PC Corrected p-value`))" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> function to return p-value from lm object</span>
<span class="pl-en">lmp</span> <span class="pl-k">&lt;-</span> <span class="pl-k">function</span>(<span class="pl-smi">modelobject</span>) {
    <span class="pl-k">if</span> (class(<span class="pl-smi">modelobject</span>) <span class="pl-k">!=</span> <span class="pl-s"><span class="pl-pds">"</span>lm<span class="pl-pds">"</span></span>)
        stop(<span class="pl-s"><span class="pl-pds">"</span>Not an object of class 'lm' <span class="pl-pds">"</span></span>)
    <span class="pl-smi">f</span> <span class="pl-k">&lt;-</span> summary(<span class="pl-smi">modelobject</span>)<span class="pl-k">$</span><span class="pl-smi">fstatistic</span>
    <span class="pl-smi">p</span> <span class="pl-k">&lt;-</span> pf(<span class="pl-smi">f</span>[<span class="pl-c1">1</span>], <span class="pl-smi">f</span>[<span class="pl-c1">2</span>], <span class="pl-smi">f</span>[<span class="pl-c1">3</span>], <span class="pl-v">lower.tail</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)
    attributes(<span class="pl-smi">p</span>) <span class="pl-k">&lt;-</span> <span class="pl-c1">NULL</span>
    <span class="pl-k">return</span>(<span class="pl-smi">p</span>)
}


<span class="pl-c"><span class="pl-c">#</span> lm test on all genes</span>
<span class="pl-smi">no_correction</span> <span class="pl-k">&lt;-</span> <span class="pl-k">list</span>()  <span class="pl-c"><span class="pl-c">#</span> initialize empty list to hold p-values for non-corrected lm test</span>
<span class="pl-smi">correction</span> <span class="pl-k">&lt;-</span> <span class="pl-k">list</span>()  <span class="pl-c"><span class="pl-c">#</span> empty list to hold p-values for corrected lm test</span>
<span class="pl-k">for</span> (<span class="pl-smi">i</span> <span class="pl-k">in</span> names(<span class="pl-smi">lm_df</span>[<span class="pl-c1">4</span><span class="pl-k">:</span>length(names(<span class="pl-smi">lm_df</span>))])) {
    <span class="pl-smi">no_correction</span>[<span class="pl-smi">i</span>] <span class="pl-k">&lt;-</span> lmp(lm(get(<span class="pl-smi">i</span>) <span class="pl-k">~</span> <span class="pl-smi">group</span>, <span class="pl-smi">lm_df</span>))
    <span class="pl-smi">correction</span>[<span class="pl-smi">i</span>] <span class="pl-k">&lt;-</span> lmp(lm(get(<span class="pl-smi">i</span>) <span class="pl-k">~</span> <span class="pl-smi">group</span> <span class="pl-k">+</span> <span class="pl-smi">PC1</span> <span class="pl-k">+</span> <span class="pl-smi">PC2</span>, <span class="pl-smi">lm_df</span>))
}


<span class="pl-c"><span class="pl-c">#</span> store results in data frame</span>
<span class="pl-smi">meta_pval</span> <span class="pl-k">&lt;-</span> rbind(as.data.frame(<span class="pl-smi">no_correction</span>), as.data.frame(<span class="pl-smi">correction</span>)) %<span class="pl-k">&gt;</span>%
    t() %<span class="pl-k">&gt;</span>%
    as.data.frame() %<span class="pl-k">&gt;</span>%
    rename(<span class="pl-smi">`Non PC Corrected p-value`</span> <span class="pl-k">=</span> <span class="pl-smi">V1</span>, <span class="pl-smi">`PC Corrected p-value`</span> <span class="pl-k">=</span> <span class="pl-smi">V2</span>) %<span class="pl-k">&gt;</span>%
    format(<span class="pl-v">scientific</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>) %<span class="pl-k">&gt;</span>%
    rownames_to_column(<span class="pl-s"><span class="pl-pds">"</span>gene<span class="pl-pds">"</span></span>)

<span class="pl-smi">meta_pval</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">meta_pval</span> %<span class="pl-k">&gt;</span>%
    mutate(<span class="pl-smi">`Non PC Corrected p-value`</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">`Non PC Corrected p-value`</span>),
        <span class="pl-smi">`PC Corrected p-value`</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">`PC Corrected p-value`</span>))</pre>
                    </div>
                    <p dir="auto">
                     Plot p-values for each genes significance in determining difference
between cancer and non cancer
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='meta_pval %&gt;%
    arrange(`PC Corrected p-value`) %&gt;%
    rename(`p-value before PC correction` = `Non PC Corrected p-value`,
        `p-value after PC correction` = `PC Corrected p-value`) %&gt;%
    reshape::melt() %&gt;%
    ggplot(aes(x = fct_inorder(gene), y = value)) + geom_bar(aes(fill = variable),
    stat = "identity", position = "dodge", width = 0.5) + scale_fill_manual(values = c("gray",
    "#FA871B")) + theme(axis.text.x = element_text(angle = 90,
    vjust = 0.5, hjust = 1)) + geom_hline(yintercept = 0.05,
    linetype = "dashed") + labs(y = "p-value", x = "gene")' dir="auto">
                     <pre><span class="pl-smi">meta_pval</span> %<span class="pl-k">&gt;</span>%
    arrange(<span class="pl-smi">`PC Corrected p-value`</span>) %<span class="pl-k">&gt;</span>%
    rename(<span class="pl-smi">`p-value before PC correction`</span> <span class="pl-k">=</span> <span class="pl-smi">`Non PC Corrected p-value`</span>,
        <span class="pl-smi">`p-value after PC correction`</span> <span class="pl-k">=</span> <span class="pl-smi">`PC Corrected p-value`</span>) %<span class="pl-k">&gt;</span>%
    <span class="pl-e">reshape</span><span class="pl-k">::</span>melt() %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span> <span class="pl-k">=</span> fct_inorder(<span class="pl-smi">gene</span>), <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">value</span>)) <span class="pl-k">+</span> geom_bar(aes(<span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-smi">variable</span>),
    <span class="pl-v">stat</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>identity<span class="pl-pds">"</span></span>, <span class="pl-v">position</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>dodge<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>) <span class="pl-k">+</span> scale_fill_manual(<span class="pl-v">values</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>gray<span class="pl-pds">"</span></span>,
    <span class="pl-s"><span class="pl-pds">"</span>#FA871B<span class="pl-pds">"</span></span>)) <span class="pl-k">+</span> theme(<span class="pl-v">axis.text.x</span> <span class="pl-k">=</span> element_text(<span class="pl-v">angle</span> <span class="pl-k">=</span> <span class="pl-c1">90</span>,
    <span class="pl-v">vjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>, <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>)) <span class="pl-k">+</span> geom_hline(<span class="pl-v">yintercept</span> <span class="pl-k">=</span> <span class="pl-c1">0.05</span>,
    <span class="pl-v">linetype</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>dashed<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> labs(<span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>p-value<span class="pl-pds">"</span></span>, <span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gene<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/plot p-values-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/plot p-values-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     It appears that after correcting for the first two principal components,
all genes except for CD22, B3GNT8, B3GAT1, and ALDH1A1 are significant.
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Quick information about significant genes:
                     </h4>
                     <a aria-label="Permalink: Quick information about significant genes:" class="anchor" href="#quick-information-about-significant-genes" id="user-content-quick-information-about-significant-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       P2RY6:
                       <a href="https://pubmed.ncbi.nlm.nih.gov/29454075/" rel="nofollow">
                        promotes tumorigenesis by inhibiting
apoptosis
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       PIK3CG:
                       <a href="https://pubmed.ncbi.nlm.nih.gov/12473596/" rel="nofollow">
                        inhibits PI3K-Akt/PKB signaling system responsible for
tumorigenesis and the progression of colorectal
cancers
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <strong>
                        GPD1
                       </strong>
                       :
                       <a href="https://en.wikipedia.org/wiki/Glycerol-3-phosphate_dehydrogenase_1" rel="nofollow">
                        encodes glycerol-3-phosphate dehydrogenase, which plays a
crucial role in carbohydrate and lipid metabolism by catalyzing
conversion of DHAP and NADG to G3P (glycerol-3-phosphate) and
NAD+.
                       </a>
                       Has been implicated as a tumor suppressor in
                       <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5731876/" rel="nofollow">
                        breast
cancer
                       </a>
                       and
                       <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9284842/" rel="nofollow">
                        bladder
cancer
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       CSFR1:
                       <a href="https://en.wikipedia.org/wiki/Colony_stimulating_factor_1_receptor" rel="nofollow">
                        involved in production, differentiation, and function of
macrophages
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <strong>
                        GCNT4
                       </strong>
                       :
                       <a href="https://www.ncbi.nlm.nih.gov/gene/51301" rel="nofollow">
                        predicted to be involved in O-glycan processing and
carbohydrate metabolic
process.
                       </a>
                       .
                       <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7457769/" rel="nofollow">
                        Found to be
downregulated in gastric
cancer
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       PRKCB:
                       <a href="https://www.ncbi.nlm.nih.gov/gene/5579" rel="nofollow">
                        encodes protein involved in B cell activation, apoptosis
induction, endothelial cell proliferation, and intestinal sugar
absorption.
                       </a>
                       <a href="https://link.springer.com/article/10.1007/s12253-013-9739-5" rel="nofollow">
                        Previously
implicated in colorectal
cancer
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       B4GALNT2:
                       <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=B4GALNT2" rel="nofollow">
                        catalyzes last step in biosynthesis of human Sd(a) antigen
and Cad
antigen.
                       </a>
                       <a href="https://pubmed.ncbi.nlm.nih.gov/34589428/" rel="nofollow">
                        Found to be highly expressed in model triple negative breast
cancer
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       CEBPA:
                       <a href="https://medlineplus.gov/genetics/gene/cebpa/" rel="nofollow">
                        encodes transcription factor involved in maturation of certain
blood cells and believed to act as a tumor
supressor.
                       </a>
                       <a href="https://www.frontiersin.org/articles/10.3389/fonc.2022.806137/full" rel="nofollow">
                        Implicated
in acute myeloid
leukemia
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       SST:
                       <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=SST" rel="nofollow">
                        regulator of endocrine system through interactions with
pituitary growth hormone, thyroid stimulating hormone, and most
hormones of GI tract. Also affects rates of neurotransmission in the
CNS and proliferation of normal and tumorigenic
cells
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <strong>
                        SLC2A4
                       </strong>
                       :
                       <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=SLC24A4" rel="nofollow">
                        encodes member of K+ dependent Na/Ca exchanger protein
family
                       </a>
                       Not many articles on its implications in cancer.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       PTPRC:
                       <a href="https://www.ncbi.nlm.nih.gov/gene/5788" rel="nofollow">
                        known to be signaling molecules that regulate a variety of
cellular processes including cell growth, differentiation, mitosis,
and oncogenic transformation
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       GCNT2:
                       <a href="https://en.wikipedia.org/wiki/GCNT2" rel="nofollow">
                        encodes enzyme responsible for formation of blood group I
antigen.
                       </a>
                       <a href="https://pubmed.ncbi.nlm.nih.gov/25750292/" rel="nofollow">
                        Hypomethylated in
colorectal cancer
                       </a>
                      </p>
                     </li>
                    </ul>
                    <hr/>
                    <p dir="auto">
                     I have selected three genes of interest to me: GPD1, GCNT4, and SLC24A
for further analysis.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='p1 &lt;- lm_df %&gt;%
    select(group, GPD1) %&gt;%
    group_by(group) %&gt;%
    summarise(mean = mean(GPD1), sd = sd(GPD1)) %&gt;%
    ggplot(aes(x = forcats::fct_reorder(factor(group), mean),
        y = mean, fill = group)) + geom_col() + geom_errorbar(aes(x = group,
    ymin = mean - sd, ymax = mean + sd), width = 0.4, colour = "orange",
    alpha = 0.9, size = 1.3) + geom_text(aes(label = paste("Average log10(fpkm):",
    round(mean, 2), sep = " "), hjust = 0.5, vjust = ifelse(mean &lt;
    0, 5.5, -2.5)), size = 3) + geom_hline(yintercept = 0) +
    ylim(-2, 2) + labs(x = "", y = "Average log(10) FPKM", title = "GPD1") +
    annotate("text", label = paste("p-value:", round(lmp(lm(GPD1 ~
        group, lm_df)), 6), sep = " "), x = 2, y = 2)

p2 &lt;- lm_df %&gt;%
    select(group, SLC2A4) %&gt;%
    group_by(group) %&gt;%
    summarise(mean = mean(SLC2A4), sd = sd(SLC2A4)) %&gt;%
    ggplot(aes(x = forcats::fct_reorder(factor(group), mean),
        y = mean, fill = group)) + geom_col() + geom_errorbar(aes(x = group,
    ymin = mean - sd, ymax = mean + sd), width = 0.4, colour = "orange",
    alpha = 0.9, size = 1.3) + geom_text(aes(label = paste("Average log10(fpkm):",
    round(mean, 2), sep = " "), hjust = 0.5, vjust = ifelse(mean &lt;
    0, 5.5, -2.5)), size = 3) + geom_hline(yintercept = 0) +
    ylim(-2, 2) + labs(x = "", y = "Average log(10) FPKM", title = "SLC2A4") +
    annotate("text", label = paste("p-value:", round(lmp(lm(SLC2A4 ~
        group, lm_df)), 6), sep = " "), x = 2, y = 2)


p3 &lt;- lm_df %&gt;%
    select(group, GCNT4) %&gt;%
    group_by(group) %&gt;%
    summarise(mean = mean(GCNT4), sd = sd(GCNT4)) %&gt;%
    ggplot(aes(x = forcats::fct_reorder(factor(group), mean),
        y = mean, fill = group)) + geom_col() + geom_errorbar(aes(x = group,
    ymin = mean - sd, ymax = mean + sd), width = 0.4, colour = "orange",
    alpha = 0.9, size = 1.3) + geom_text(aes(label = paste("Average log10(fpkm):",
    round(mean, 2), sep = " "), hjust = 0.5, vjust = ifelse(mean &lt;
    0, 5.5, -2.5)), size = 3) + geom_hline(yintercept = 0) +
    ylim(-2, 2) + labs(x = "", y = "Average log(10) FPKM", title = "GCNT4 ") +
    annotate("text", label = paste("p-value:", round(lmp(lm(GCNT4 ~
        group, lm_df)), 6), sep = " "), x = 2, y = 2)

grid.arrange(p1, p2, p3, ncol = 2)' dir="auto">
                     <pre><span class="pl-smi">p1</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">lm_df</span> %<span class="pl-k">&gt;</span>%
    select(<span class="pl-smi">group</span>, <span class="pl-smi">GPD1</span>) %<span class="pl-k">&gt;</span>%
    group_by(<span class="pl-smi">group</span>) %<span class="pl-k">&gt;</span>%
    summarise(<span class="pl-v">mean</span> <span class="pl-k">=</span> mean(<span class="pl-smi">GPD1</span>), <span class="pl-v">sd</span> <span class="pl-k">=</span> sd(<span class="pl-smi">GPD1</span>)) %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-e">forcats</span><span class="pl-k">::</span>fct_reorder(<span class="pl-k">factor</span>(<span class="pl-smi">group</span>), <span class="pl-smi">mean</span>),
        <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">mean</span>, <span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-smi">group</span>)) <span class="pl-k">+</span> geom_col() <span class="pl-k">+</span> geom_errorbar(aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">group</span>,
    <span class="pl-v">ymin</span> <span class="pl-k">=</span> <span class="pl-smi">mean</span> <span class="pl-k">-</span> <span class="pl-smi">sd</span>, <span class="pl-v">ymax</span> <span class="pl-k">=</span> <span class="pl-smi">mean</span> <span class="pl-k">+</span> <span class="pl-smi">sd</span>), <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">0.4</span>, <span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>orange<span class="pl-pds">"</span></span>,
    <span class="pl-v">alpha</span> <span class="pl-k">=</span> <span class="pl-c1">0.9</span>, <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">1.3</span>) <span class="pl-k">+</span> geom_text(aes(<span class="pl-v">label</span> <span class="pl-k">=</span> paste(<span class="pl-s"><span class="pl-pds">"</span>Average log10(fpkm):<span class="pl-pds">"</span></span>,
    round(<span class="pl-smi">mean</span>, <span class="pl-c1">2</span>), <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>, <span class="pl-v">vjust</span> <span class="pl-k">=</span> ifelse(<span class="pl-smi">mean</span> <span class="pl-k">&lt;</span>
    <span class="pl-c1">0</span>, <span class="pl-c1">5.5</span>, <span class="pl-k">-</span><span class="pl-c1">2.5</span>)), <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>) <span class="pl-k">+</span> geom_hline(<span class="pl-v">yintercept</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>) <span class="pl-k">+</span>
    ylim(<span class="pl-k">-</span><span class="pl-c1">2</span>, <span class="pl-c1">2</span>) <span class="pl-k">+</span> labs(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Average log(10) FPKM<span class="pl-pds">"</span></span>, <span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>GPD1<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
    annotate(<span class="pl-s"><span class="pl-pds">"</span>text<span class="pl-pds">"</span></span>, <span class="pl-v">label</span> <span class="pl-k">=</span> paste(<span class="pl-s"><span class="pl-pds">"</span>p-value:<span class="pl-pds">"</span></span>, round(lmp(lm(<span class="pl-smi">GPD1</span> <span class="pl-k">~</span>
        <span class="pl-smi">group</span>, <span class="pl-smi">lm_df</span>)), <span class="pl-c1">6</span>), <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>)

<span class="pl-smi">p2</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">lm_df</span> %<span class="pl-k">&gt;</span>%
    select(<span class="pl-smi">group</span>, <span class="pl-smi">SLC2A4</span>) %<span class="pl-k">&gt;</span>%
    group_by(<span class="pl-smi">group</span>) %<span class="pl-k">&gt;</span>%
    summarise(<span class="pl-v">mean</span> <span class="pl-k">=</span> mean(<span class="pl-smi">SLC2A4</span>), <span class="pl-v">sd</span> <span class="pl-k">=</span> sd(<span class="pl-smi">SLC2A4</span>)) %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-e">forcats</span><span class="pl-k">::</span>fct_reorder(<span class="pl-k">factor</span>(<span class="pl-smi">group</span>), <span class="pl-smi">mean</span>),
        <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">mean</span>, <span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-smi">group</span>)) <span class="pl-k">+</span> geom_col() <span class="pl-k">+</span> geom_errorbar(aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">group</span>,
    <span class="pl-v">ymin</span> <span class="pl-k">=</span> <span class="pl-smi">mean</span> <span class="pl-k">-</span> <span class="pl-smi">sd</span>, <span class="pl-v">ymax</span> <span class="pl-k">=</span> <span class="pl-smi">mean</span> <span class="pl-k">+</span> <span class="pl-smi">sd</span>), <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">0.4</span>, <span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>orange<span class="pl-pds">"</span></span>,
    <span class="pl-v">alpha</span> <span class="pl-k">=</span> <span class="pl-c1">0.9</span>, <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">1.3</span>) <span class="pl-k">+</span> geom_text(aes(<span class="pl-v">label</span> <span class="pl-k">=</span> paste(<span class="pl-s"><span class="pl-pds">"</span>Average log10(fpkm):<span class="pl-pds">"</span></span>,
    round(<span class="pl-smi">mean</span>, <span class="pl-c1">2</span>), <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>, <span class="pl-v">vjust</span> <span class="pl-k">=</span> ifelse(<span class="pl-smi">mean</span> <span class="pl-k">&lt;</span>
    <span class="pl-c1">0</span>, <span class="pl-c1">5.5</span>, <span class="pl-k">-</span><span class="pl-c1">2.5</span>)), <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>) <span class="pl-k">+</span> geom_hline(<span class="pl-v">yintercept</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>) <span class="pl-k">+</span>
    ylim(<span class="pl-k">-</span><span class="pl-c1">2</span>, <span class="pl-c1">2</span>) <span class="pl-k">+</span> labs(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Average log(10) FPKM<span class="pl-pds">"</span></span>, <span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SLC2A4<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
    annotate(<span class="pl-s"><span class="pl-pds">"</span>text<span class="pl-pds">"</span></span>, <span class="pl-v">label</span> <span class="pl-k">=</span> paste(<span class="pl-s"><span class="pl-pds">"</span>p-value:<span class="pl-pds">"</span></span>, round(lmp(lm(<span class="pl-smi">SLC2A4</span> <span class="pl-k">~</span>
        <span class="pl-smi">group</span>, <span class="pl-smi">lm_df</span>)), <span class="pl-c1">6</span>), <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>)


<span class="pl-smi">p3</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">lm_df</span> %<span class="pl-k">&gt;</span>%
    select(<span class="pl-smi">group</span>, <span class="pl-smi">GCNT4</span>) %<span class="pl-k">&gt;</span>%
    group_by(<span class="pl-smi">group</span>) %<span class="pl-k">&gt;</span>%
    summarise(<span class="pl-v">mean</span> <span class="pl-k">=</span> mean(<span class="pl-smi">GCNT4</span>), <span class="pl-v">sd</span> <span class="pl-k">=</span> sd(<span class="pl-smi">GCNT4</span>)) %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-e">forcats</span><span class="pl-k">::</span>fct_reorder(<span class="pl-k">factor</span>(<span class="pl-smi">group</span>), <span class="pl-smi">mean</span>),
        <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">mean</span>, <span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-smi">group</span>)) <span class="pl-k">+</span> geom_col() <span class="pl-k">+</span> geom_errorbar(aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">group</span>,
    <span class="pl-v">ymin</span> <span class="pl-k">=</span> <span class="pl-smi">mean</span> <span class="pl-k">-</span> <span class="pl-smi">sd</span>, <span class="pl-v">ymax</span> <span class="pl-k">=</span> <span class="pl-smi">mean</span> <span class="pl-k">+</span> <span class="pl-smi">sd</span>), <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">0.4</span>, <span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>orange<span class="pl-pds">"</span></span>,
    <span class="pl-v">alpha</span> <span class="pl-k">=</span> <span class="pl-c1">0.9</span>, <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">1.3</span>) <span class="pl-k">+</span> geom_text(aes(<span class="pl-v">label</span> <span class="pl-k">=</span> paste(<span class="pl-s"><span class="pl-pds">"</span>Average log10(fpkm):<span class="pl-pds">"</span></span>,
    round(<span class="pl-smi">mean</span>, <span class="pl-c1">2</span>), <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>, <span class="pl-v">vjust</span> <span class="pl-k">=</span> ifelse(<span class="pl-smi">mean</span> <span class="pl-k">&lt;</span>
    <span class="pl-c1">0</span>, <span class="pl-c1">5.5</span>, <span class="pl-k">-</span><span class="pl-c1">2.5</span>)), <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>) <span class="pl-k">+</span> geom_hline(<span class="pl-v">yintercept</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>) <span class="pl-k">+</span>
    ylim(<span class="pl-k">-</span><span class="pl-c1">2</span>, <span class="pl-c1">2</span>) <span class="pl-k">+</span> labs(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Average log(10) FPKM<span class="pl-pds">"</span></span>, <span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>GCNT4 <span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
    annotate(<span class="pl-s"><span class="pl-pds">"</span>text<span class="pl-pds">"</span></span>, <span class="pl-v">label</span> <span class="pl-k">=</span> paste(<span class="pl-s"><span class="pl-pds">"</span>p-value:<span class="pl-pds">"</span></span>, round(lmp(lm(<span class="pl-smi">GCNT4</span> <span class="pl-k">~</span>
        <span class="pl-smi">group</span>, <span class="pl-smi">lm_df</span>)), <span class="pl-c1">6</span>), <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>)

grid.arrange(<span class="pl-smi">p1</span>, <span class="pl-smi">p2</span>, <span class="pl-smi">p3</span>, <span class="pl-v">ncol</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/flemm0/gastric_cancer_differential_expression/blob/main/README_files/figure-gfm/barplots of individual genes-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/flemm0/gastric_cancer_differential_expression/raw/main/README_files/figure-gfm/barplots of individual genes-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Here are the barplots of their average expression and standard error
between the three groups. I also included the p-values of their
significance after correcting for principal components 1 and 2, you can
see that levels of expression are a lot lower in cancer cell lines and
tissue compared to the normal groups and their error bars dont overlap.
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      SLC2A4
                     </h4>
                     <a aria-label="Permalink: SLC2A4" class="anchor" href="#slc2a4" id="user-content-slc2a4">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     SLC2A4 is a member of the solute-carrier family-2 proteins. It encodes
the protein GLUT4, which is responsible for the diffusion of circulating
glucose down its concentration gradient into muscle and fat cells, where
they are converted either to glucose-6-phosphate for glycolysis or
glycogen for storage.
                    </p>
                    <p dir="auto">
                     GLUT4 plays an essential role in maintaining body glucose homeostasis
and is regulated by insulin
                    </p>
                    <p dir="auto">
                     In a
                     <a href="https://pubmed.ncbi.nlm.nih.gov/34488531/" rel="nofollow">
                      previous study
                     </a>
                     looking
at the solute-carrier family-2s implications in cancer, they analyzed
mRNA expression data in 20 cancer types from the Oncomine database and
filtered based on a p-value of less than 0.01 and fold change of 1.5.
They found that SLC2A4 was significantly downregulated in breast cancer.
                    </p>
                    <p dir="auto">
                     To verify their results, they used a web-based tool called UALCAN and
found that mRNA expression of SLC2A4 was significantly downregulated in
all individual breast cancer stages with p-value &lt; 0.01. I included a
screenshot of their results in the top picture.
                    </p>
                    <p dir="auto">
                     They then used a Kaplan-Meier survival analysis and demonstrated that
high expression of SLC2A4 was related to better prognosis in stage I and
III breast cancer. I included a picture of their survival curve for
stage I cancer on the bottom.
                    </p>
                    <p dir="auto">
                     Lastly, they used a software called MEXPRESS to assess whether DNA
hypermethylation in the promoter of SLC2A4 was related to its
expression. They found that there was a statistical difference when
comparing breast cancer samples versus normal samples, suggesting that
this is a regulator of mRNA expression.
                    </p>
                    <p dir="auto">
                     Overall, they suggest that GLUT4 downregulation may inhibit glucose
uptake and induce metabolic reprogramming and also affect downstream
pathways that play key roles in tumorigenesis and progression. But they
state that further studies with larger sample sizes are needed for a
better understanding of the mechanisms of SLC2A4 in cancer.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      GPD1
                     </h4>
                     <a aria-label="Permalink: GPD1" class="anchor" href="#gpd1" id="user-content-gpd1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     GPD1 plays a role in carbohydrate and lipid metabolism, specifically in
the conversion of DHAP and NADH into glycerol-3-phosphate and NAD+.
Together with GPD2, the two proteins carry electrons generated by
glycolysis into the electron transport chain.
                    </p>
                    <p dir="auto">
                     In
                     <a href="https://jhoonline.biomedcentral.com/articles/10.1186/s13045-022-01312-5" rel="nofollow">
                      this
study
                     </a>
                     that used 17 pairs of bladder cancer tissue and matched adjacent tissue,
they analyzed differential expression using proteomics, and identified
almost 100,000 peptides. After filtering with 1.5 fold change difference
and a p-value of 0.05 as the cutoff, they narrowed the proteins down to
518 downregulated proteins, and found GPD1 to be the most significantly
downregulated, which I highlighted in the volcano plot in figure B.
                    </p>
                    <p dir="auto">
                     They verified their results using a Western blot (figure C on the slide)
and also used a Kaplan-Meier estimate to observe that GPD1 was
correlated with decreased survival time for breast cancer patients,
which led them to hypothesize that GPD1 may have antitumor functions.
                    </p>
                    <p dir="auto">
                     To summarize their findings: they determined that GPD1 increases a
compound called lysoPC through a glycerophospholipid metabolism pathway.
lysoPC then upregulated expression of TRPV2, causing a calcium influx
that inhibits cell proliferation and induces apoptosis.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      GCNT4
                     </h4>
                     <a aria-label="Permalink: GCNT4" class="anchor" href="#gcnt4" id="user-content-gcnt4">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     GCNT 4 stands for glucosaminyl (N-acetyl) transferase 4. It is part of a
family, named GCNT1 through 4, and all of them are involved in mucin
core structure synthesis and branching. These mucins are the most common
macromolecules in mucus and are mainly responsible for its properties.
                    </p>
                    <p dir="auto">
                     Its been hypothesized that members of the GCNT family influences cancer
genesis by regulating cell growth and apoptosis pathways and have been
identified in colon, breast, prostate, and pancreatic cancer.
                    </p>
                    <p dir="auto">
                     In
                     <a href="https://pubmed.ncbi.nlm.nih.gov/34696660/" rel="nofollow">
                      this paper
                     </a>
                     , the
researchers looked at gene expression data from gastric cancer and
non-tumor samples published in GEO. They filtered differentially
expressed genes using the criteria of having a log fold change of less
than -1.5 and p-value of 0.05 and found GCNT4 as a gene of interest as a
suppressor of gastric cancer cell proliferation.
                    </p>
                    <p dir="auto">
                     Additionally, they used a program called TargetScan to predict an
microRNA that binds to the 3UTR of GCNT4 and confirmed in 45 of their
own gastric cancer samples that this microRNA decreases GCNT4 expression
which in turn upregulated downstream pathways that have been studied in
the progression of gastric cancer.
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Conclusions and Future Directions
                     </h3>
                     <a aria-label="Permalink: Conclusions and Future Directions" class="anchor" href="#conclusions-and-future-directions" id="user-content-conclusions-and-future-directions">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Overall, my findings of abnormally expressed genes and pathways in
gastric cancer were consistent with my expectations with many cancer
related pathways being upregulated and many immune cell pathways being
downregulated.
                    </p>
                    <p dir="auto">
                     Also, the downregulation of carbohydrate metabolism pathways has been
observed in a variety of different cancers, including gastric cancer,
and the three genes that I looked at are involved in different parts of
carbohydrate metabolism pathways.
                    </p>
                    <p dir="auto">
                     Its possible that the pathway I looked at was significant in gastric
cancer because of the genes cross-talk with other pathways that have
to do with cell cycle or immune cell regulation.
                    </p>
                    <p dir="auto">
                     Moving forward, I think that it would be insightful to screen a larger
and more diverse sample of gastric cancer tissues and include people of
different ethnicities as well as from the different subtypes of gastric
cancer so that we can see if we observe any overlap in carbohydrate
metabolism pathways. Then, I think that it would be interesting to
filter out the pathways that overlap with ones that are abnormally
expressed in other types of cancer, so that we can consider any that are
left to be possibly gastric cancer-specific.
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="5XY5TcaFQl5iBn5MTLHl1WwAND9iBkgzF1T+BdCd5RXTUyOulI1ynQkfhoKjtRq+V2IEeQVZt8DdZ0XQl6HQQA==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 Final Project for TRGN 514: Introduction to Human Genomic Analysis Methods
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/flemm0/gastric_cancer_differential_expression/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  star
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fflemm0%2Fgastric_cancer_differential_expression&amp;report=flemm0+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/flemm0/gastric_cancer_differential_expression/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/flemm0/packages?repo_name=gastric_cancer_differential_expression">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/flemm0/gastric_cancer_differential_expression/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="Shell 100.0" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#89e051 !important;;width: 100.0%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/flemm0/gastric_cancer_differential_expression/search?l=shell">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#89e051;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Shell
                  </span>
                  <span>
                   100.0%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
        2024 GitHub,Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You cant perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
