<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-6a292b1161ed.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22: A code walkthrough of the analyses performed for our paper on the shared genetic architecture of modifiable risk for Alzheimer's disease using genomic SEM.
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:119DEC1:1992DCC:6605C39C" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDoxMTlERUMxOjE5OTJEQ0M6NjYwNUMzOUMiLCJ2aXNpdG9yX2lkIjoiNjQ1MjI0MTg1NjIzODgxNzgwNyIsInJlZ2lvbl9lZGdlIjoiaWFkIiwicmVnaW9uX3JlbmRlciI6ImlhZCJ9" data-pjax-transient="true" name="visitor-payload">
           <meta content="05cc0c00d6f6f3feae76199458664c7dbd7cea24fdd4318b60deea4cc3743cab" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:503434804" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="A code walkthrough of the analyses performed for our paper on the shared genetic architecture of modifiable risk for Alzheimer's disease using genomic SEM. - IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/c9e422c65001a2de2846c6c6402ae4d79bfc32c304642f5b40ff27b404368c2c/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22: A code walkthrough of the analyses performed for our paper on the shared genetic architecture of modifiable risk for Alzheimer's disease using genomic SEM." name="twitter:title">
                    <meta content="A code walkthrough of the analyses performed for our paper on the shared genetic architecture of modifiable risk for Alzheimer&amp;amp;#39;s disease using genomic SEM. - IsyFoote/Genomic-Factor-Analysi..." name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/c9e422c65001a2de2846c6c6402ae4d79bfc32c304642f5b40ff27b404368c2c/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" property="og:image">
                      <meta content="A code walkthrough of the analyses performed for our paper on the shared genetic architecture of modifiable risk for Alzheimer&amp;#39;s disease using genomic SEM. - IsyFoote/Genomic-Factor-Analysis-of..." property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22: A code walkthrough of the analyses performed for our paper on the shared genetic architecture of modifiable risk for Alzheimer's disease using genomic SEM." property="og:title"/>
                        <meta content="https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" property="og:url"/>
                        <meta content="A code walkthrough of the analyses performed for our paper on the shared genetic architecture of modifiable risk for Alzheimer&amp;#39;s disease using genomic SEM. - IsyFoote/Genomic-Factor-Analysis-of..." property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="de8e1521fc37a178caaed17097d0545118b4eb0f01833d1d1eeeb65ac20b6ca7" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="1dd47e41bae3fb619dfa564356c6dee453350e97ff939ad02473ed6906f90b45" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22 git https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22.git" name="go-import"/>
                        <meta content="75127203" name="octolytics-dimension-user_id">
                         <meta content="IsyFoote" name="octolytics-dimension-user_login">
                          <meta content="503434804" name="octolytics-dimension-repository_id">
                           <meta content="IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="503434804" name="octolytics-dimension-repository_network_root_id">
                               <meta content="IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-11f6759e1cef.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22","user_id":null}}' data-hydro-click-hmac="85def01c98855c174dfda0b3c0a627beb0c6efff01f0b628eb022476b86cf799" href="/login?return_to=https%3A%2F%2Fgithub.com%2FIsyFoote%2FGenomic-Factor-Analysis-of-modifiable-AD-risk_NBA22">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=hgfF1fGr4qvA93C4OzCSseuqJNTo7S%2FjpBkhcxE0%2FPDmKAObcMEjmqSNGizcEgkxQuYO6VTt6a4yCgekDFPEEr8BdnS2Za0%2BQIbjodA7nEQjWW2%2F46muH0NuqQ7jgXGNBN583eX8HCoYp138cC2nO9rRIFOBQev%2BX7VHrCRm0bsowS1VEvziqsOhRsR7VEKzYE5M35lcsA7k%2BYnDOUOsCI7d8xp5GXYlLCx2npXKcBt%2FQRWP5Z65BjdV82mmtjlgx3x9xBJqnMlEWB3lofy6IlJZ60l%2BfqL7lOh%2Bh92w9y9DEth%2FAsH5TYx%2F9XuV0onOG21I5I0SqhwJOTn0XZDbh9b0UfOLvsCdRCqQmFd5CdRoQEm4Kt%2BMCWekCkIA3tO3yLLC9jlxYBj2yAXQBgogOCAFjJAbBcBV6G3hM%2BCja9r1kCFcWMVGNSctVPwgcpx6HmuXYKxKkNB6SEb2T7QJFNj6vQcA8fTaQYdogZ7Kv32Uq6Zixs1vo6BakhANwGcM%2FjUAZg%2FdAGkZ9yvMt0fyqhE%2FXbtHtSAMAb3Zt4m%2BZ6kl4tftXEk%2BhKfTZT3Q16wvskKc1fIHZLC5OGhlKP0PYfB0bANbXg%3D%3D--kRMZd8jwx50q349N--M1KNPU1Umb9r379fnH4UHg%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="IsyFoote" data-current-repository="IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="StIpyMpAyVHFqxYMXdPJlF8BbaHo3Y6h2Ho0d6tFcZ5XXAkPwSCK49Y1mRTy1J-qOgZmZl0f-6o-J49NJ4f9tQ" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-a876237d-97c8-425c-bb58-8c9dbbc9c71e" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-a876237d-97c8-425c-bb58-8c9dbbc9c71e">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="uJn8IF8jN38bA00Wv0wV+sWwOGynmW1gzhjtleCB0LwGXenWICwRBFECLGZuqMzidAzjViCCLbCHUQkQM/+5jA=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="QsZnlCLbRi5h6W1evsyt9jChF4dyWH3C9hADYePqIUXIQcnIIvH7+S1RQrWHGfMu9cn0iXDgLz7RtnY6jXmKqA==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="Rbucq+dsnxHNuGiGrqnSS7z2biXFEiJtBit8uSm38JcodTT3xZ5sc0dVGHwI+wRFaK2pXkgsEOfWBx4eVPKy9A==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="l8BCTO4ezkDmWEujIK6rG/tEJaoohvY9omjpaO9ntzdh3YcqEo3+PiP15TqSQt8zTQStg4AbeXa8NFJSJBsbJw==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22","user_id":null}}' data-hydro-click-hmac="85def01c98855c174dfda0b3c0a627beb0c6efff01f0b628eb022476b86cf799" href="/login?return_to=https%3A%2F%2Fgithub.com%2FIsyFoote%2FGenomic-Factor-Analysis-of-modifiable-AD-risk_NBA22">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22","user_id":null}}' data-hydro-click-hmac="85def01c98855c174dfda0b3c0a627beb0c6efff01f0b628eb022476b86cf799" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=IsyFoote%2FGenomic-Factor-Analysis-of-modifiable-AD-risk_NBA22">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-ccef4062-0a96-4f22-bb04-d71207db0c5f" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-b5a7c5e3-2abb-4948-a41d-e08596c2d165" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-b5a7c5e3-2abb-4948-a41d-e08596c2d165" id="tooltip-ccef4062-0a96-4f22-bb04-d71207db0c5f" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/IsyFoote/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/IsyFoote" rel="author">
            IsyFoote
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22">
            Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22","user_id":null}}' data-hydro-click-hmac="5d96ea524bfc1d687e42689f426811cbdefc8aaf166ea4191739de10f074336e" data-view-component="true" href="/login?return_to=%2FIsyFoote%2FGenomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":503434804,"auth_type":"LOG_IN","originating_url":"https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22","user_id":null}}' data-hydro-click-hmac="d64d76a5c88f35a500903c091e37fa181daf455723451d24c6f3f473b9793c8d" data-view-component="true" href="/login?return_to=%2FIsyFoote%2FGenomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
             0
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":503434804,"auth_type":"LOG_IN","originating_url":"https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22","user_id":null}}' data-hydro-click-hmac="64110ab6d0d843dd4d0d2c615ab6a4b26ab9c511b56c2ffdf450d5c241f0300c" data-view-component="true" href="/login?return_to=%2FIsyFoote%2FGenomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="1 user starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="1">
              1
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <p class="f4 mb-3">
          A code walkthrough of the analyses performed for our paper on the shared genetic architecture of modifiable risk for Alzheimer's disease using genomic SEM.
         </p>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            1
           </span>
           star
          </a>
          <a class="Link--secondary no-underline mr-3" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":503434804,"auth_type":"LOG_IN","originating_url":"https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22","user_id":null}}' data-hydro-click-hmac="64110ab6d0d843dd4d0d2c615ab6a4b26ab9c511b56c2ffdf450d5c241f0300c" data-view-component="true" href="/login?return_to=%2FIsyFoote%2FGenomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22","user_id":null}}' data-hydro-click-hmac="5d96ea524bfc1d687e42689f426811cbdefc8aaf166ea4191739de10f074336e" data-view-component="true" href="/login?return_to=%2FIsyFoote%2FGenomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Discussions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g g" data-pjax="#repo-content-pjax-container" data-selected-links="repo_discussions /IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/discussions" data-tab-item="i3discussions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/discussions" id="discussions-tab">
           <svg aria-hidden="true" class="octicon octicon-comment-discussion UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
            </path>
           </svg>
           <span data-content="Discussions">
            Discussions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="discussions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/actions" data-tab-item="i4actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/projects" data-tab-item="i5projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/security" data-tab-item="i6security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/pulse" data-tab-item="i7insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-69694d88-395a-4f4c-825e-0c79ec133e00-list" aria-haspopup="true" aria-labelledby="tooltip-c7dd35f7-3451-4143-8b7d-68bc100b9e63" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-69694d88-395a-4f4c-825e-0c79ec133e00-button" popovertarget="action-menu-69694d88-395a-4f4c-825e-0c79ec133e00-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-69694d88-395a-4f4c-825e-0c79ec133e00-button" id="tooltip-c7dd35f7-3451-4143-8b7d-68bc100b9e63" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-69694d88-395a-4f4c-825e-0c79ec133e00-button" anchor-offset="normal" data-view-component="true" id="action-menu-69694d88-395a-4f4c-825e-0c79ec133e00-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-69694d88-395a-4f4c-825e-0c79ec133e00-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-69694d88-395a-4f4c-825e-0c79ec133e00-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22" id="item-99b9c0d5-2d0f-4a45-ab91-85da30076308" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/issues" id="item-cfc83b92-cbf8-42ad-aaa0-f24d87f7d420" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/pulls" id="item-4ca58f7c-0053-4c4d-88d4-5498dc3f3513" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3discussions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/discussions" id="item-c106cf8c-1068-4203-abf6-309e097d05e2" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/actions" id="item-6a92dd53-3803-42eb-a620-6e39a11cd1ee" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/projects" id="item-e5b8239b-ecd0-44c5-9d2f-20e2f6f01bbd" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/security" id="item-1eb38fb9-8c81-474f-81eb-c96f73a66793" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i7insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/pulse" id="item-7fcb5c3c-9f00-4b0d-b46b-0deb0856652e" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/spoofed_commit_check/a984140625a2fdab5ee9987a9c02dbf0f69e4f87">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":503434804,"defaultBranch":"main","name":"Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22","ownerLogin":"IsyFoote","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2022-06-14T16:19:38.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/75127203?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"main","listCacheKey":"v0:1655227793.711041","canEdit":false,"refType":"branch","currentOid":"a984140625a2fdab5ee9987a9c02dbf0f69e4f87"},"tree":{"items":[{"name":"images","path":"images","contentType":"directory"},{"name":".gitignore","path":".gitignore","contentType":"file"},{"name":"LDSC.sh","path":"LDSC.sh","contentType":"file"},{"name":"LDSC_env.yml","path":"LDSC_env.yml","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"},{"name":"example_script_to_format_UKB_GWAS.R","path":"example_script_to_format_UKB_GWAS.R","contentType":"file"},{"name":"genomic_factor_analysis_and_sensitivity_analyses.R","path":"genomic_factor_analysis_and_sensitivity_analyses.R","contentType":"file"},{"name":"make_LD_corr_matrix_and_undirected_graph.R","path":"make_LD_corr_matrix_and_undirected_graph.R","contentType":"file"},{"name":"make_LD_correlation_matrix.R","path":"make_LD_correlation_matrix.R","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":9,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/archive/refs/heads/main.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=503434804"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"13","overviewFiles":[{"displayName":"README.md","repoName":"Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCode walkthrough for genomic factor analysis of modifiable risk for Alzheimers disease\u003c/h1\u003e\u003ca id=\"user-content-code-walkthrough-for-genomic-factor-analysis-of-modifiable-risk-for-alzheimers-disease\" class=\"anchor\" aria-label=\"Permalink: Code walkthrough for genomic factor analysis of modifiable risk for Alzheimers disease\" href=\"#code-walkthrough-for-genomic-factor-analysis-of-modifiable-risk-for-alzheimers-disease\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIsabelle Foote\n2022-06-13\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#overview\"\u003eOverview\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#gwas-data-identification-and-formatting\"\u003eGWAS data identification and formatting\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#criteria-for-inclusion\"\u003eCriteria for inclusion\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#links-to-the-gwas-data-used-in-our-current-study\"\u003eLinks to the GWAS data used in our current study\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#example-code-for-formatting-uk-biobank-gwas-data\"\u003eExample code for formatting UK Biobank GWAS data\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#set-your-working-directory-and-load-the-required-packages\"\u003eSet your working directory and load the required packages\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#make-an-edited-version-of-the-variant-data-to-merge-with-the-gwas-datasets\"\u003eMake an edited version of the variant data to merge with the GWAS datasets\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#format-uk-biobank-gwas-data-ready-for-munging\"\u003eFormat UK Biobank GWAS data ready for munging\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ld-score-regression\"\u003eLD score regression\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#install-ldsc-software\"\u003eInstall LDSC software\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#munge-data\"\u003eMunge data\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#calculating-univariate-snp-based-heritability\"\u003eCalculating univariate SNP-based heritability\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#calculating-pairwise-genetic-correlations\"\u003eCalculating pairwise genetic correlations\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#plotting-the-data\"\u003ePlotting the data\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#genetic-correlation-matrix\"\u003eGenetic correlation matrix\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#undirected-weighted-graph\"\u003eUndirected weighted graph\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#genomic-sem\"\u003eGenomic SEM\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#install--genomicsem-\"\u003eInstall \u003cem\u003eGenomicSEM\u003c/em\u003e\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#set-working-directory-and-load-the--genomicsem--package\"\u003eSet working directory and load the \u003cem\u003eGenomicSEM\u003c/em\u003e package\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#munge-data-1\"\u003eMunge data\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#exploratory-factor-analysis-performed-in-the-odd-autosomes\"\u003eExploratory factor analysis performed in the odd autosomes\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#load-the-additional-required-packages\"\u003eLoad the additional required packages\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#conduct-multivariable-ld-score-regression--odd-autosomes-\"\u003eConduct multivariable LD score regression (odd autosomes)\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#exploratory-factor-analysis\"\u003eExploratory factor analysis\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#efa-loadings-plot\"\u003eEFA loadings plot\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#confirmatory-factor-analysis-performed-in-the-even-autosomes\"\u003eConfirmatory factor analysis performed in the even autosomes\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#conduct-multivariable-ld-score-regression--even-autosomes-\"\u003eConduct multivariable LD score regression (even autosomes)\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#confirmatory-factor-analysis\"\u003eConfirmatory factor analysis\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#plotting-cfa-models\"\u003ePlotting CFA models\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#follow-up-genome-wide-analyses\"\u003eFollow-up genome-wide analyses\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#genome-wide-multivariable-ld-score-regression\"\u003eGenome-wide multivariable LD score regression\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#genome-wide-cfa\"\u003eGenome-wide CFA\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#additional-sem-modelling\"\u003eAdditional SEM modelling\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#common-factor-model\"\u003eCommon factor model\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#hierarchal-model\"\u003eHierarchal model\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bifactor-model\"\u003eBifactor model\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#models-without-the-apoe-region\"\u003eModels without the APOE region\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#sensitivity-analyses\"\u003eSensitivity analyses\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#ld-score-regression-in-different-chromosomal-groupings\"\u003eLD score regression in different chromosomal groupings\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#efa-comparison-across-chromosomal-groupings\"\u003eEFA comparison across chromosomal groupings\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#cfa-comparison-across-chromosomal-groupings\"\u003eCFA comparison across chromosomal groupings\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#final-remarks\"\u003eFinal remarks\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eOverview\u003c/h1\u003e\u003ca id=\"user-content-overview\" class=\"anchor\" aria-label=\"Permalink: Overview\" href=\"#overview\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe following document outlines the code that was used to carry out the\nwork described in the study titled \u003cstrong\u003eThe shared genetic architecture of\nmodifiable risk for Alzheimers disease: a genomic structural equation\nmodelling study\u003c/strong\u003e by Isabelle Foote, Ben Jacobs, Georgina Mathlin,\nCameron Watson, Phazha Bothongo, Sheena Waters, Ruth Dobson, Alastair\nNoyce, Kam Bhui, Ania Korszun and Charles Marshall.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe published version of the paper can be found\n\u003ca href=\"https://www.sciencedirect.com/science/article/pii/S0197458022001312\" rel=\"nofollow\"\u003ehere\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe preprint for the study can be found on\n\u003ca href=\"https://www.medrxiv.org/content/10.1101/2021.02.23.21252211v1\" rel=\"nofollow\"\u003eMedRxiv\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis markdown report and code was written by Isabelle Foote.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe current document provides step-by-step instructions to replicate our\nwork and utilises open-source software and code that was created by the\noriginal developers of the methods used.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFurther details relating to software/code can be found on the\ncorresponding original GitHub pages:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003e\u003ca href=\"https://github.com/bulik/ldsc\"\u003eLD score regression\u003c/a\u003e\u003c/strong\u003e (Brendan\nBulik-Sullivan and colleagues)\u003c/li\u003e\n\u003cli\u003e\u003cstrong\u003e\u003ca href=\"https://github.com/GenomicSEM/GenomicSEM\"\u003eGenomic SEM\u003c/a\u003e\u003c/strong\u003e (Andrew\nGrotzinger and colleagues)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eFurther details on the methods used can be found in the following\npublications, upon which the current study protocol was based:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eLD score regression\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eBulik-Sullivan B., et al.(2015) \u003cem\u003eLD Score regression distinguishes\nconfounding from polygenicity in genome-wide association studies\u003c/em\u003e.\nNature Genetics. 47: p.291-295 \u003ca href=\"https://doi.org/10.1038/ng.3211\" rel=\"nofollow\"\u003ehttps://doi.org/10.1038/ng.3211\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eBulik-Sullivan B., et al.(2015) \u003cem\u003eAn atlas of genetic correlations\nacross human diseases and traits\u003c/em\u003e. Nature Genetics. 47(11):\np.1236-1241 \u003ca href=\"https://doi.org/10.1038/ng.3406\" rel=\"nofollow\"\u003ehttps://doi.org/10.1038/ng.3406\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eZheng J., et al.(2016) \u003cem\u003eLD Hub: a centralized database and web\ninterface to perform LD score regression that maximizes the\npotential of summary level GWAS data for SNP heritability and\ngenetic correlation analysis\u003c/em\u003e. Bioinformatics. 33(2): p.272-279\n\u003ca href=\"https://doi.org/10.1093/bioinformatics/btw613\" rel=\"nofollow\"\u003ehttps://doi.org/10.1093/bioinformatics/btw613\u003c/a\u003e.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eGenomic SEM\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eGrotzinger AD., et al.(2019) \u003cem\u003eGenomic structural equation modelling\nprovides insights into the multivariate genetic architecture of\ncomplex traits\u003c/em\u003e. Nat Hum Behav. 3(5): p.513-525\n\u003ca href=\"https://doi.org/10.1038/s41562-019-0566-x\" rel=\"nofollow\"\u003ehttps://doi.org/10.1038/s41562-019-0566-x\u003c/a\u003e.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGWAS data identification and formatting\u003c/h1\u003e\u003ca id=\"user-content-gwas-data-identification-and-formatting\" class=\"anchor\" aria-label=\"Permalink: GWAS data identification and formatting\" href=\"#gwas-data-identification-and-formatting\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe LD score regression and genomic SEM methods we use in this study are\nbased off GWAS summary statistics data, so the first step is to identify\nthe appropriate data for your analysis. For a detailed rationale\nrelating to the choice of traits we included within our study please\nrefer to our\n\u003ca href=\"https://www.sciencedirect.com/science/article/pii/S0197458022001312\" rel=\"nofollow\"\u003epaper\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn short, we included GWAS data for Alzheimers disease and its major\npotentially modifiable risk factors (less education, hearing loss,\nhypertension, high alcohol intake, obesity, smoking, depression, social\nisolation, physical inactivity, type 2 diabetes, sleep disturbance and\nsocioeconomic deprivation).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSome useful resources to find datasets are:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://www.ebi.ac.uk/gwas/\" rel=\"nofollow\"\u003eGWAS Catalog\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://atlas.ctglab.nl/\" rel=\"nofollow\"\u003eGWAS Atlas\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"http://www.nealelab.is/uk-biobank\" rel=\"nofollow\"\u003eNeale lab\u003c/a\u003e (for GWAS of UK\nBiobank traits)\u003c/li\u003e\n\u003cli\u003eThe \u003ca href=\"https://biobank.ndph.ox.ac.uk/showcase/\" rel=\"nofollow\"\u003eUK Biobank Data\nShowcase\u003c/a\u003e is an easy way to\nlook up what variables were measured within the UK Biobank study\u003c/li\u003e\n\u003cli\u003eRelevant consortia for your topic of interest e.g.the \u003ca href=\"https://www.med.unc.edu/pgc/download-results/\" rel=\"nofollow\"\u003ePsychiatric\nGenomics Consortium\u003c/a\u003e\nfor GWASs of psychiatric disorders\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCriteria for inclusion\u003c/h2\u003e\u003ca id=\"user-content-criteria-for-inclusion\" class=\"anchor\" aria-label=\"Permalink: Criteria for inclusion\" href=\"#criteria-for-inclusion\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIn order for your analysis to be reliable you need to ensure that the\nGWAS data you include is appropriate and meets the advised criteria\nexplained in the original methods papers cited above and on the relevant\nwikis.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor the current study we ensured that the data met the following\ncriteria:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eThe samples were of unrelated individuals of European ancestry\u003c/li\u003e\n\u003cli\u003eEach dataset had a sample size \u0026gt;5000 (ideally over 10,000)\u003c/li\u003e\n\u003cli\u003eThe summary statistics had not been adjusted for heritable\ncovariates during association testing\u003c/li\u003e\n\u003cli\u003eThe association test had not used linear mixed modelling (LMM)\nmethods\u003c/li\u003e\n\u003cli\u003eEach trait had a SNP-based heritability Z score \u0026gt;4\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eOften the SNP-based heritability is reported in the orginal GWAS paper\nso to get an idea about whether your trait is appropriate this can be a\ngood starting point. \u003ca href=\"http://ldsc.broadinstitute.org/\" rel=\"nofollow\"\u003eLD Hub\u003c/a\u003e also\ncontains heritability and genetic correlation results for hundreds of\ntraits. For the Neale lab UK Biobank GWASs you can check details of\ntheir heritability in their \u003ca href=\"https://nealelab.github.io/UKBB_ldsc/index.html\" rel=\"nofollow\"\u003eheritability\nbrowser\u003c/a\u003e and pairwise\ngenetic correlations in their \u003ca href=\"https://ukbb-rg.hail.is/\" rel=\"nofollow\"\u003ecorrelation\nbrowser\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis is useful as an initial checking step, however we recommend\nperforming your own LD score regression as an initial part of your own\nanalysis pipeline to ensure that the results you get are in line with\npreviously run results and to facilitate a deeper understanding of your\nown study results.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eLinks to the GWAS data used in our current study\u003c/h3\u003e\u003ca id=\"user-content-links-to-the-gwas-data-used-in-our-current-study\" class=\"anchor\" aria-label=\"Permalink: Links to the GWAS data used in our current study\" href=\"#links-to-the-gwas-data-used-in-our-current-study\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe GWAS data we used in our current study is publicly available and\navailable to download via the following links:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003eAlzheimers disease\u003c/strong\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ePaper: Kunkle BW., et al.(2019) \u003cem\u003eGenetic meta-analysis of\ndiagnosed Alzheimers disease identifies new risk loci and\nimplicates A, tau, immunity and lipid processing\u003c/em\u003e. Nat\nGenetics. 51: p.414-430\n\u003ca href=\"https://doi.org/10.1038/s41588-019-0358-2\" rel=\"nofollow\"\u003ehttps://doi.org/10.1038/s41588-019-0358-2\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eData: \u003ca href=\"https://www.niagads.org/datasets/ng00075\" rel=\"nofollow\"\u003eIGAP\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003cstrong\u003eMajor depressive disorder\u003c/strong\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ePaper: Wray NR., et al.(2018) \u003cem\u003eGenome-wide association analyses\nidentify 44 risk variants and refine the genetic architecture of\nmajor depression\u003c/em\u003e. Nat Genetics. 50(5): p.668-681\n\u003ca href=\"https://doi.org/10.1038/s41588-018-0090-3\" rel=\"nofollow\"\u003ehttps://doi.org/10.1038/s41588-018-0090-3\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eData: \u003ca href=\"https://www.med.unc.edu/pgc/download-results/mdd/\" rel=\"nofollow\"\u003ePGC\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003cstrong\u003eInsomnia\u003c/strong\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ePaper: Jansen PR., et al.(2019) \u003cem\u003eGenome-wide analysis of\ninsomnia in 1,331,010 individuals identifies new risk loci and\nfunctional pathways\u003c/em\u003e. Nat Genetics. 51(3): p.394-403\n\u003ca href=\"https://doi.org/10.1038/s41588-018-0333-3\" rel=\"nofollow\"\u003ehttps://doi.org/10.1038/s41588-018-0333-3\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eData: \u003ca href=\"https://ctg.cncr.nl/software/summary_statistics\" rel=\"nofollow\"\u003eCTG Lab\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003cstrong\u003eType 2 diabetes\u003c/strong\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ePaper: Scott RA., et al.(2017) \u003cem\u003eAn Expanded Genome-Wide\nAssociation Study of Type 2 Diabetes in Europeans\u003c/em\u003e. Diabetes.\n66(11): p.2888 \u003ca href=\"https://doi.org/10.2337/db16-1253\" rel=\"nofollow\"\u003ehttps://doi.org/10.2337/db16-1253\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eData: \u003ca href=\"http://diagram-consortium.org/downloads.html\" rel=\"nofollow\"\u003eDIAGRAM\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003cstrong\u003eUK Biobank traits\u003c/strong\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eTraits: Loneliness, less social activity, hearing difficulty,\nless education, physical inactivity, smoking, alcohol intake,\nBMI, deprivation status, systolic blood pressure\u003c/li\u003e\n\u003cli\u003eWebsite: \u003ca href=\"http://www.nealelab.is/uk-biobank\" rel=\"nofollow\"\u003ehttp://www.nealelab.is/uk-biobank\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003eData: \u003ca href=\"https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit?ts=5b5f17db#gid=178908679\" rel=\"nofollow\"\u003eNeale\nlab\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eExample code for formatting UK Biobank GWAS data\u003c/h2\u003e\u003ca id=\"user-content-example-code-for-formatting-uk-biobank-gwas-data\" class=\"anchor\" aria-label=\"Permalink: Example code for formatting UK Biobank GWAS data\" href=\"#example-code-for-formatting-uk-biobank-gwas-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThere are certain columns and information that are required in the GWAS\ndatasets so that the quality control filtering step (known as munging)\ncan be completed before conducting LD score regression and genomic SEM.\nThese requirements are outlined in detail below in the \u003cem\u003edata munging\u003c/em\u003e\nsections.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe LDSC and GenomicSEM \u003ccode\u003emunge\u003c/code\u003e functions can recognise a range of\ndifferent column names so often you can just input the downloaded\ndataset directly into this function and it will run.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHowever, there may be occasions where the GWAS data cannot be read by\nthe \u003ccode\u003emunge\u003c/code\u003e function because the software cannot recognise a certain\ncolumn name or if there is data missing that it needs. One of the main\nexamples of this happening is if you use the GWAS summary statistics of\nUK Biobank traits that were conducted by the Neale lab. Therefore, we\nprovide a detailed example of how to appropriately format these datasets\nfor use with the LDSC and GenomicSEM software. This code was run in R\nStudio (Version 1.2.5033) using R version 3.6.3 on a personal desktop.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe downloaded our UK Biobank GWAS traits from the \u003ca href=\"https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit?ts=5b5f17db#gid=227859291\" rel=\"nofollow\"\u003eNeale lab\nmanifest\u003c/a\u003e\nof their round 2 European GWAS data. This is best done using the \u003ccode\u003ewget\u003c/code\u003e\ncommand in the terminal.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSet your working directory and load the required packages\u003c/h3\u003e\u003ca id=\"user-content-set-your-working-directory-and-load-the-required-packages\" class=\"anchor\" aria-label=\"Permalink: Set your working directory and load the required packages\" href=\"#set-your-working-directory-and-load-the-required-packages\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"setwd(\u0026quot;/path/to/directory\u0026quot;)\nlibrary(dplyr)\nlibrary(data.table)\"\u003e\u003cpre\u003esetwd(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/path/to/directory\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMake an edited version of the variant data to merge with the GWAS datasets\u003c/h3\u003e\u003ca id=\"user-content-make-an-edited-version-of-the-variant-data-to-merge-with-the-gwas-datasets\" class=\"anchor\" aria-label=\"Permalink: Make an edited version of the variant data to merge with the GWAS datasets\" href=\"#make-an-edited-version-of-the-variant-data-to-merge-with-the-gwas-datasets\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe individual GWAS files from the Neale lab only include some of the\ninformation that you need to carry out the \u003ccode\u003emunge\u003c/code\u003e step (sample size,\nMAF and effect statistics). Therefore, the Neale lab also provide a\ngeneral variant file that contains annotation information for all the\nvariants measured within a single file in the same order that they\nappear in the GWAS files. By merging this file with the \u003cstrong\u003evariant\u003c/strong\u003e\ncolumn in the trait GWAS you can then get the rsIDs, A1 (i.e.the alt\nallele), A2 (i.e.the ref allele) and INFO information that you need for\nyour analysis. See\n\u003ca href=\"https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit?ts=5b5f17db#gid=227859291\" rel=\"nofollow\"\u003ehere\u003c/a\u003e\nfor further information on the variant file and to download it.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNote:\u003c/strong\u003e Since the UK Biobank files are very large we found it best to\ninitially make an edited variant file containing only the columns that\nyou need for this formatting using the code below as this significantly\nreduces runtime in the later steps.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn addition, we use the \u003ccode\u003efread\u003c/code\u003e function from the \u003cem\u003edata.table\u003c/em\u003e R package\nto read in the datasets in our below example. This requires the\ndownloaded files to be unzipped so that they are .tsv files rather than\n.bgz ones (i.e.the downloaded format from the Neale lab website). Using\nthe base R \u003ccode\u003eread.table\u003c/code\u003e function also works but is \u003cem\u003emuch\u003c/em\u003e slower.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"variant_data \u0026lt;- fread(\u0026quot;variants.tsv\u0026quot;)\nvariant_data \u0026lt;- as.data.table(variant_data)\nhead(variant_data)\n\nvariant_data_edited \u0026lt;- variant_data %\u0026gt;% select(variant, ref, alt, rsid, info)\nhead(variant_data_edited)\n\nwrite.table(variant_data_edited, file = \u0026quot;variant_data_edited_UKB.txt\u0026quot;, row.names = FALSE,\n            quote = FALSE)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003evariant_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evariants.tsv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003evariant_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.data.table(\u003cspan class=\"pl-smi\"\u003evariant_data\u003c/span\u003e)\nhead(\u003cspan class=\"pl-smi\"\u003evariant_data\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003evariant_data_edited\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003evariant_data\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% select(\u003cspan class=\"pl-smi\"\u003evariant\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eref\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ealt\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ersid\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003einfo\u003c/span\u003e)\nhead(\u003cspan class=\"pl-smi\"\u003evariant_data_edited\u003c/span\u003e)\n\nwrite.table(\u003cspan class=\"pl-smi\"\u003evariant_data_edited\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evariant_data_edited_UKB.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFormat UK Biobank GWAS data ready for munging\u003c/h3\u003e\u003ca id=\"user-content-format-uk-biobank-gwas-data-ready-for-munging\" class=\"anchor\" aria-label=\"Permalink: Format UK Biobank GWAS data ready for munging\" href=\"#format-uk-biobank-gwas-data-ready-for-munging\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow we can use this edited variant file to merge with the GWASs of our\ntraits of interest. Please see an example script for the trait \u003cem\u003efeelings\nof loneliness/isolation\u003c/em\u003e below. In addition to merging the data we use\nthe \u003cem\u003edplyr\u003c/em\u003e R package to \u003ccode\u003erename\u003c/code\u003e the columns to names that are\nrecognised by the \u003ccode\u003emunge\u003c/code\u003e functions and we \u003ccode\u003eselect\u003c/code\u003e only the required\ncolumns to reduce the size of the edited file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNote:\u003c/strong\u003e The \u003ccode\u003emerge\u003c/code\u003e step takes a while to run.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"### Read in original GWAS sumstats file using fread \n\nloneliness_UKB_data \u0026lt;- fread(\u0026quot;2020.gwas.imputed_v3.both_sexes.tsv\u0026quot;)\nloneliness_UKB_data \u0026lt;- as.data.table(loneliness_UKB_data)\nhead(loneliness_UKB_data)\n\n## read in our edited Neale variants file\n\nvariant_data \u0026lt;- fread(\u0026quot;variant_data_edited_UKB.txt\u0026quot;)\nvariant_data \u0026lt;- as.data.table(variant_data)\nhead(variant_data)\n\n## join the phenotype file with the variant file by the variant column\n\njoined_df \u0026lt;- merge(loneliness_UKB_data, variant_data , by.x = \u0026quot;variant\u0026quot;, \n                   by.y = \u0026quot;variant\u0026quot;, all.x = TRUE, all.y = FALSE)\n\nhead(joined_df)\n\n## rename the columns \u0026amp; keep only those required for munging\n\nloneliness_UKB_data_edited \u0026lt;- joined_df %\u0026gt;% rename(A1 = \u0026quot;alt\u0026quot;) %\u0026gt;% \n  rename(A2 = \u0026quot;ref\u0026quot;) %\u0026gt;% \n  rename(MAF = \u0026quot;minor_AF\u0026quot;) %\u0026gt;% \n  rename(N = \u0026quot;n_complete_samples\u0026quot;) %\u0026gt;% \n  rename(BETA = \u0026quot;beta\u0026quot;) %\u0026gt;% \n  rename(SE = \u0026quot;se\u0026quot;) %\u0026gt;% \n  rename(P = \u0026quot;pval\u0026quot;) %\u0026gt;% \n  rename(SNP = \u0026quot;rsid\u0026quot;) %\u0026gt;% \n  rename(INFO = \u0026quot;info\u0026quot;) %\u0026gt;%\n  select(A1, A2, MAF, N, BETA, SE, P, SNP, INFO)\n\nhead(loneliness_UKB_data_edited)\n\n## Write new table as txt file munging step\n\nwrite.table(loneliness_UKB_data_edited, file = \u0026quot;loneliness_UKB_GWAS_edit.txt\u0026quot;, \n            row.names = FALSE, quote = FALSE)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## Read in original GWAS sumstats file using fread \u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eloneliness_UKB_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e2020.gwas.imputed_v3.both_sexes.tsv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eloneliness_UKB_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.data.table(\u003cspan class=\"pl-smi\"\u003eloneliness_UKB_data\u003c/span\u003e)\nhead(\u003cspan class=\"pl-smi\"\u003eloneliness_UKB_data\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# read in our edited Neale variants file\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003evariant_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evariant_data_edited_UKB.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003evariant_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.data.table(\u003cspan class=\"pl-smi\"\u003evariant_data\u003c/span\u003e)\nhead(\u003cspan class=\"pl-smi\"\u003evariant_data\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# join the phenotype file with the variant file by the variant column\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003ejoined_df\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-smi\"\u003eloneliness_UKB_data\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003evariant_data\u003c/span\u003e , \u003cspan class=\"pl-v\"\u003eby.x\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evariant\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                   \u003cspan class=\"pl-v\"\u003eby.y\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evariant\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eall.x\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eall.y\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\nhead(\u003cspan class=\"pl-smi\"\u003ejoined_df\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# rename the columns \u0026amp; keep only those required for munging\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eloneliness_UKB_data_edited\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ejoined_df\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% rename(\u003cspan class=\"pl-v\"\u003eA1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  rename(\u003cspan class=\"pl-v\"\u003eA2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eref\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  rename(\u003cspan class=\"pl-v\"\u003eMAF\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eminor_AF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  rename(\u003cspan class=\"pl-v\"\u003eN\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003en_complete_samples\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  rename(\u003cspan class=\"pl-v\"\u003eBETA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebeta\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  rename(\u003cspan class=\"pl-v\"\u003eSE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ese\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  rename(\u003cspan class=\"pl-v\"\u003eP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epval\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  rename(\u003cspan class=\"pl-v\"\u003eSNP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ersid\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  rename(\u003cspan class=\"pl-v\"\u003eINFO\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003einfo\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  select(\u003cspan class=\"pl-smi\"\u003eA1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eA2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eMAF\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eN\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eBETA\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSE\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSNP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eINFO\u003c/span\u003e)\n\nhead(\u003cspan class=\"pl-smi\"\u003eloneliness_UKB_data_edited\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Write new table as txt file munging step\u003c/span\u003e\n\nwrite.table(\u003cspan class=\"pl-smi\"\u003eloneliness_UKB_data_edited\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eloneliness_UKB_GWAS_edit.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eLD score regression\u003c/h1\u003e\u003ca id=\"user-content-ld-score-regression\" class=\"anchor\" aria-label=\"Permalink: LD score regression\" href=\"#ld-score-regression\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo ensure that our traits were sufficiently heritable to be meaningfully\nmodelled using genomic SEM we initially performed LD score regression to\ncalculate the SNP-based heritability of each trait. We also used\ncross-trait LD score regression to assess the level of inter-correlation\npresent between each pair of traits.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis stage of the analysis utilised the command line \u003cem\u003eLDSC\u003c/em\u003e software\ndeveloped by Brendan Bulik-Sullivan and colleagues (for in depth details\nplease refer to their \u003ca href=\"https://github.com/bulik/ldsc\"\u003eoriginal wiki\u003c/a\u003e).\nWe performed the next few steps in the terminal on an Ubuntu personal\ndesktop (Linux OS).\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInstall LDSC software\u003c/h3\u003e\u003ca id=\"user-content-install-ldsc-software\" class=\"anchor\" aria-label=\"Permalink: Install LDSC software\" href=\"#install-ldsc-software\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe LDSC software is run in the command line and relies on a collection\nof Python dependencies to run. In order to install LDSC you first need\nto download \u003ca href=\"https://www.anaconda.com/products/individual\" rel=\"nofollow\"\u003eAnaconda\u003c/a\u003e so\nthat you can create an environment that includes the package\ndependencies specifically required for the software. More instructions\non this can be found on the \u003ca href=\"https://github.com/bulik/ldsc\"\u003eoriginal LDSC\nwiki\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHowever, we found that if you are using a more recent version of\nAnaconda (we used Anaconda3 version 2020.07), you may run into some\nfunctionality issues with the LDSC functions if you use the environment\nfile specified in the original software. We therefore used an adapted\nenvironment file (environment_edit.yml) within our analysis.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdapted environment file\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn a text editor we wrote the following text and saved it as\n\u003cstrong\u003eLDSC_env.yml\u003c/strong\u003e:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"name: environment_edit\nchannels:\n- bioconda\ndependencies:\n- python=2.7\n- bitarray=0.8\n- nose=1.3\n- pybedtools=0.7\n- pip\n- pip:\n  - scipy==0.18\n  - pandas==0.19\n  - numpy==1.16\"\u003e\u003cpre\u003ename: environment_edit\nchannels:\n- bioconda\ndependencies:\n- python=2.7\n- bitarray=0.8\n- nose=1.3\n- pybedtools=0.7\n- pip\n- pip:\n  - scipy==0.18\n  - pandas==0.19\n  - numpy==1.16\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe then clone the repository for LDSC and create the environment with\nthe adapted package dependencies.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"git clone https://github.com/bulik/ldsc.git\ncd ldsc\nconda env create --file LDSC_env.yml\nsource activate LDSC_env\"\u003e\u003cpre\u003egit clone https://github.com/bulik/ldsc.git\n\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ldsc\nconda env create --file LDSC_env.yml\n\u003cspan class=\"pl-c1\"\u003esource\u003c/span\u003e activate LDSC_env\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMunge data\u003c/h2\u003e\u003ca id=\"user-content-munge-data\" class=\"anchor\" aria-label=\"Permalink: Munge data\" href=\"#munge-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFirst we must munge the GWAS data so that it can be recognised by the\nldsc software and the data is filtered to only include well-imputed SNPs\nwith a MAF \u0026gt;0.01. See the \u003cstrong\u003eReformatting Summary Statistics\u003c/strong\u003e section\n\u003ca href=\"https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation\"\u003ehere\u003c/a\u003e\nfor further details.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn short, for the \u003ccode\u003emunge\u003c/code\u003e function to work we need the following columns\nin the downloaded GWAS summary statistics as a minimum:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSNP rsIDs\u003c/li\u003e\n\u003cli\u003eAllele 1 (i.e.the effect allele)\u003c/li\u003e\n\u003cli\u003eAllele 2 (i.e.the non-effect allele)\u003c/li\u003e\n\u003cli\u003eSample size\u003c/li\u003e\n\u003cli\u003eThe regression effect (e.g.beta, OR, logOR, Z-score)\u003c/li\u003e\n\u003cli\u003eP values of the effect\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eIt is also useful to have the MAF and INFO columns for filtering, but\nthese are not crucial since we filter to a reference genome. Here we\nused Hapmap 3 SNPs with the MHC region removed as our reference genome.\nThis data can be downloaded from:\n\u003ca href=\"https://utexas.app.box.com/s/vkd36n197m8klbaio3yzoxsee6sxo11v\" rel=\"nofollow\"\u003ehttps://utexas.app.box.com/s/vkd36n197m8klbaio3yzoxsee6sxo11v\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"cd /media/user/Elements/R_markdown_test/\n\n# AD\n\n./ldsc/munge_sumstats.py \\\n--sumstats Kunkle_etal_Stage1_results.txt \\\n--N 63926 \\\n--out ad_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# MDD\n\n./ldsc/munge_sumstats.py \\\n--sumstats MDD2018_ex23andMe.gz \\\n--N 173005 \\\n--out mdd_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# Insomnia\n\n./ldsc/munge_sumstats.py \\\n--sumstats Insomnia_sumstats_Jansenetal.txt.gz \\\n--N 386533 \\\n--out insomnia_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# Loneliness\n\n./ldsc/munge_sumstats.py \\\n--sumstats loneliness_UKB_GWAS_edit.txt \\\n--N 355583 \\\n--out loneliness_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\n# Low social activity\n\n./ldsc/munge_sumstats.py \\\n--sumstats no_social_activity_UKB_GWAS_edit.txt \\\n--N 360063 \\\n--out no_social_activity_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# Hearing difficulty\n\n./ldsc/munge_sumstats.py \\\n--sumstats hearing_diff_UKB_GWAS_edit.txt \\\n--N 353983 \\\n--out hearing_difficulty_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# Less education\n\n./ldsc/munge_sumstats.py \\\n--sumstats low_education_UKB_GWAS_edit.txt \\\n--N 357549 \\\n--out low_education_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# Physical inactivity\n\n./ldsc/munge_sumstats.py \\\n--sumstats physical_inactivity_UKB_GWAS_edit.txt \\\n--N 359263 \\\n--out physical_inactivity_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# Smoking\n\n./ldsc/munge_sumstats.py \\\n--sumstats current_smoker_UKB_GWAS_edit.txt \\\n--N 359706 \\\n--out current_smoker_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# Alcohol intake frequency\n\n./ldsc/munge_sumstats.py \\\n--sumstats alcohol_intake_freq_UKB_GWAS_edit.txt \\\n--N 360726 \\\n--out alcohol_intake_freq_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# BMI\n\n./ldsc/munge_sumstats.py \\\n--sumstats bmi_UKB_GWAS_edit.txt \\\n--N 354831 \\\n--out bmi_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# Deprivation status\n\n./ldsc/munge_sumstats.py \\\n--sumstats deprivation_UKB_GWAS_edit.txt \\\n--N 360763 \\\n--out deprivation_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# Systolic blood pressure\n\n./ldsc/munge_sumstats.py \\\n--sumstats systolic_bp_UKB_GWAS_edit.txt \\\n--N 340159 \\\n--out systolic_bp_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n# Type 2 diabetes\n\n./ldsc/munge_sumstats.py \\\n--sumstats diabetes_with_rsids.txt \\\n--N 159208 \\\n--out type_2_diabetes_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /media/user/Elements/R_markdown_test/\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e AD\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats Kunkle_etal_Stage1_results.txt \\\n--N 63926 \\\n--out ad_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e MDD\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats MDD2018_ex23andMe.gz \\\n--N 173005 \\\n--out mdd_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Insomnia\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats Insomnia_sumstats_Jansenetal.txt.gz \\\n--N 386533 \\\n--out insomnia_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Loneliness\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats loneliness_UKB_GWAS_edit.txt \\\n--N 355583 \\\n--out loneliness_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Low social activity\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats no_social_activity_UKB_GWAS_edit.txt \\\n--N 360063 \\\n--out no_social_activity_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Hearing difficulty\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats hearing_diff_UKB_GWAS_edit.txt \\\n--N 353983 \\\n--out hearing_difficulty_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Less education\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats low_education_UKB_GWAS_edit.txt \\\n--N 357549 \\\n--out low_education_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Physical inactivity\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats physical_inactivity_UKB_GWAS_edit.txt \\\n--N 359263 \\\n--out physical_inactivity_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Smoking\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats current_smoker_UKB_GWAS_edit.txt \\\n--N 359706 \\\n--out current_smoker_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Alcohol intake frequency\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats alcohol_intake_freq_UKB_GWAS_edit.txt \\\n--N 360726 \\\n--out alcohol_intake_freq_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e BMI\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats bmi_UKB_GWAS_edit.txt \\\n--N 354831 \\\n--out bmi_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Deprivation status\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats deprivation_UKB_GWAS_edit.txt \\\n--N 360763 \\\n--out deprivation_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Systolic blood pressure\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats systolic_bp_UKB_GWAS_edit.txt \\\n--N 340159 \\\n--out systolic_bp_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Type 2 diabetes\u003c/span\u003e\n\n./ldsc/munge_sumstats.py \\\n--sumstats diabetes_with_rsids.txt \\\n--N 159208 \\\n--out type_2_diabetes_ldsc \\\n--merge-alleles w_hm3.noMHC.snplist\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCalculating univariate SNP-based heritability\u003c/h2\u003e\u003ca id=\"user-content-calculating-univariate-snp-based-heritability\" class=\"anchor\" aria-label=\"Permalink: Calculating univariate SNP-based heritability\" href=\"#calculating-univariate-snp-based-heritability\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow we want to calculate the univariate SNP-based heritability of each\ntrait so that we can assess whether it is sufficiently heritable to be\nincluded in our main analysis (i.e.whether it has a h2 Z score \u0026gt;4). We\nuse the \u003ccode\u003eldsc.py\u003c/code\u003e function and the \u003ccode\u003e--h2\u003c/code\u003e argument within the LDSC\nsoftware. See\n\u003ca href=\"https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation\"\u003ehere\u003c/a\u003e\nfor further details.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eYou need to use LD scores and weights that match your datas ancestral\nbackground. Here we used pre-calculated European LD scores and weights\nfor our analysis from the Broad Institute computed from 1000 Genomes\ndata. These can be downloaded from:\n\u003ca href=\"https://alkesgroup.broadinstitute.org/LDSCORE/\" rel=\"nofollow\"\u003ehttps://alkesgroup.broadinstitute.org/LDSCORE/\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis group has made a range of different LD scores and weights that are\npublically available but if you need to make some of your own see the\n\u003ca href=\"https://github.com/bulik/ldsc/wiki/LD-Score-Estimation-Tutorial\"\u003eLDSC\nwiki\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor \u003cstrong\u003ebinary\u003c/strong\u003e traits you need to specify the sample and population\nprevalences (see \u003ccode\u003esample-prev\u003c/code\u003e and \u003ccode\u003epopulation-prev\u003c/code\u003e in the code below).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSample prevalence can be calculated by dividing the \u003cem\u003enumber of cases\u003c/em\u003e by\nthe \u003cem\u003etotal sample size\u003c/em\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePopulation prevalences are based on a relevant estimate that is widely\ncited and reliable for your population of interest (see the\n\u003cstrong\u003eSupplementary Methods\u003c/strong\u003e of our\n\u003ca href=\"https://www.sciencedirect.com/science/article/pii/S0197458022001312\" rel=\"nofollow\"\u003epaper\u003c/a\u003e\nfor details on how we identified our population prevalence estimates).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eYou do not need to provide the sample and population prevalence for\n\u003cstrong\u003econtinuous\u003c/strong\u003e or \u003cstrong\u003ecategorical\u003c/strong\u003e traits.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# AD \n\n./ldsc/ldsc.py \\\n--h2 ad_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out ad_h2 \\\n--samp-prev 0.34 \\\n--pop-prev 0.05 \\\n\n# MDD\n\n./ldsc/ldsc.py \\\n--h2 mdd_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out mdd_h2 \\\n--samp-prev 0.35 \\\n--pop-prev 0.13 \n\n# Insomnia\n./ldsc/ldsc.py \\\n--h2 insomnia_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out insomnia_h2 \\\n--samp-prev 0.28 \\\n--pop-prev 0.32 \n\n# Loneliness\n./ldsc/ldsc.py \\\n--h2 loneliness_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out loneliness_h2 \\\n--samp-prev 0.18 \\\n--pop-prev 0.08 \n\n# Low social activity\n./ldsc/ldsc.py \\\n--h2 no_social_activity_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out no_social_activity_h2 \\\n--samp-prev 0.30 \\\n--pop-prev 0.30 \n\n# Hearing difficulty\n./ldsc/ldsc.py \\\n--h2 hearing_difficulty_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out hearing_difficulty_h2 \\\n--samp-prev 0.38 \\\n--pop-prev 0.39 \n\n# Less education\n./ldsc/ldsc.py \\\n--h2 low_education_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out low_education_h2 \\\n--samp-prev 0.17 \\\n--pop-prev 0.27 \n\n# Physical inactivity\n./ldsc/ldsc.py \\\n--h2 physical_inactivity_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out physical_inactivity_h2 \\\n--samp-prev 0.06 \\\n--pop-prev 0.18\n\n# Smoking\n./ldsc/ldsc.py \\\n--h2 current_smoker_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out current_smoker_h2 \\\n--samp-prev 0.10 \\\n--pop-prev 0.27\n\n# Alcohol intake frequency\n./ldsc/ldsc.py \\\n--h2 alcohol_intake_freq_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out alcohol_intake_freq_h2 \n\n# BMI\n./ldsc/ldsc.py \\\n--h2 bmi_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out bmi_h2 \n\n# Deprivation status\n./ldsc/ldsc.py \\\n--h2 deprivation_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out deprivation_h2 \n\n# Systolic bp\n./ldsc/ldsc.py \\\n--h2 systolic_bp_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out systolic_bp_h2 \n\n# Type 2 diabetes\n./ldsc/ldsc.py \\\n--h2 type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out type_2_diabetes_h2 \\\n--samp-prev 0.17 \\\n--pop-prev 0.06\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e AD \u003c/span\u003e\n\n./ldsc/ldsc.py \\\n--h2 ad_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out ad_h2 \\\n--samp-prev 0.34 \\\n--pop-prev 0.05 \\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e MDD\u003c/span\u003e\n\n./ldsc/ldsc.py \\\n--h2 mdd_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out mdd_h2 \\\n--samp-prev 0.35 \\\n--pop-prev 0.13 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Insomnia\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 insomnia_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out insomnia_h2 \\\n--samp-prev 0.28 \\\n--pop-prev 0.32 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Loneliness\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 loneliness_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out loneliness_h2 \\\n--samp-prev 0.18 \\\n--pop-prev 0.08 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Low social activity\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 no_social_activity_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out no_social_activity_h2 \\\n--samp-prev 0.30 \\\n--pop-prev 0.30 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Hearing difficulty\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 hearing_difficulty_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out hearing_difficulty_h2 \\\n--samp-prev 0.38 \\\n--pop-prev 0.39 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Less education\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 low_education_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out low_education_h2 \\\n--samp-prev 0.17 \\\n--pop-prev 0.27 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Physical inactivity\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 physical_inactivity_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out physical_inactivity_h2 \\\n--samp-prev 0.06 \\\n--pop-prev 0.18\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Smoking\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 current_smoker_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out current_smoker_h2 \\\n--samp-prev 0.10 \\\n--pop-prev 0.27\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Alcohol intake frequency\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 alcohol_intake_freq_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out alcohol_intake_freq_h2 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e BMI\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 bmi_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out bmi_h2 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Deprivation status\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 deprivation_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out deprivation_h2 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Systolic bp\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 systolic_bp_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out systolic_bp_h2 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Type 2 diabetes\u003c/span\u003e\n./ldsc/ldsc.py \\\n--h2 type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out type_2_diabetes_h2 \\\n--samp-prev 0.17 \\\n--pop-prev 0.06\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCalculating pairwise genetic correlations\u003c/h2\u003e\u003ca id=\"user-content-calculating-pairwise-genetic-correlations\" class=\"anchor\" aria-label=\"Permalink: Calculating pairwise genetic correlations\" href=\"#calculating-pairwise-genetic-correlations\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNext, we want to calculate the pairwise genetic correlations between\neach pair of traits to ascertain how much genetic correlation is present\nbetween our traits. We use the \u003ccode\u003eldsc.py\u003c/code\u003e function with the \u003ccode\u003e--rg\u003c/code\u003e\nargument. See\n\u003ca href=\"https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation\"\u003ehere\u003c/a\u003e\nfor further details.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs with the previous step we use pre-calculated European LD scores and\nweights for our analysis from the Broad Institute computed from 1000\nGenomes data downloaded from:\n\u003ca href=\"https://alkesgroup.broadinstitute.org/LDSCORE/\" rel=\"nofollow\"\u003ehttps://alkesgroup.broadinstitute.org/LDSCORE/\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# AD\n./ldsc/ldsc.py \\\n--rg ad_ldsc.sumstats.gz,mdd_ldsc.sumstats.gz,insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\\\nhearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out ad_rg \\\n--samp-prev 0.34,0.35,0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.05,0.13,0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06 \n\n# MDD\n./ldsc/ldsc.py \\\n--rg mdd_ldsc.sumstats.gz,insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\\\nhearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out mdd_rg \\\n--samp-prev 0.35,0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.13,0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n# insomnia\n./ldsc/ldsc.py \\\n--rg insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\\\nhearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out insomnia_rg \\\n--samp-prev 0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n# loneliness\n./ldsc/ldsc.py \\\n--rg loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\\\nhearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out loneliness_rg \\\n--samp-prev 0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n# low social activity\n./ldsc/ldsc.py \\\n--rg no_social_activity_ldsc.sumstats.gz,\\\nhearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out no_social_activity_rg \\\n--samp-prev 0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n# hearing difficulty\n./ldsc/ldsc.py \\\n--rg hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out hearing_difficulty_rg \\\n--samp-prev 0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n# less education\n./ldsc/ldsc.py \\\n--rg low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out low_education_rg \\\n--samp-prev 0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n# physical inactivity\n./ldsc/ldsc.py \\\n--rg physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out physical_inactivity_rg \\\n--samp-prev 0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.18,0.27,nan,nan,nan,nan,0.06\n\n# smoking\n./ldsc/ldsc.py \\\n--rg current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out current_smoker_rg \\\n--samp-prev 0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.27,nan,nan,nan,nan,0.06\n\n# alcohol intake freq\n./ldsc/ldsc.py \\\n--rg alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out alcohol_intake_freq_rg \\\n--samp-prev nan,nan,nan,nan,0.17 \\\n--pop-prev nan,nan,nan,nan,0.06\n\n# BMI\n./ldsc/ldsc.py \\\n--rg bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out bmi_rg \\\n--samp-prev nan,nan,nan,0.17 \\\n--pop-prev nan,nan,nan,0.06\n\n# deprivation\n./ldsc/ldsc.py \\\n--rg deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out deprivation_rg \\\n--samp-prev nan,nan,0.17 \\\n--pop-prev nan,nan,0.06\n\n# systolic bp\n./ldsc/ldsc.py \\\n--rg systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out systolic_bp_rg \\\n--samp-prev nan,0.17 \\\n--pop-prev nan,0.06\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e AD\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg ad_ldsc.sumstats.gz,mdd_ldsc.sumstats.gz,insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\\\nhearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out ad_rg \\\n--samp-prev 0.34,0.35,0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.05,0.13,0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e MDD\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg mdd_ldsc.sumstats.gz,insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\\\nhearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out mdd_rg \\\n--samp-prev 0.35,0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.13,0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e insomnia\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\\\nhearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out insomnia_rg \\\n--samp-prev 0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e loneliness\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\\\nhearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out loneliness_rg \\\n--samp-prev 0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e low social activity\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg no_social_activity_ldsc.sumstats.gz,\\\nhearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out no_social_activity_rg \\\n--samp-prev 0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e hearing difficulty\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out hearing_difficulty_rg \\\n--samp-prev 0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e less education\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out low_education_rg \\\n--samp-prev 0.17,0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.27,0.18,0.27,nan,nan,nan,nan,0.06\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e physical inactivity\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg physical_inactivity_ldsc.sumstats.gz,\\\ncurrent_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out physical_inactivity_rg \\\n--samp-prev 0.06,0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.18,0.27,nan,nan,nan,nan,0.06\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e smoking\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out current_smoker_rg \\\n--samp-prev 0.10,nan,nan,nan,nan,0.17 \\\n--pop-prev 0.27,nan,nan,nan,nan,0.06\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e alcohol intake freq\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out alcohol_intake_freq_rg \\\n--samp-prev nan,nan,nan,nan,0.17 \\\n--pop-prev nan,nan,nan,nan,0.06\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e BMI\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out bmi_rg \\\n--samp-prev nan,nan,nan,0.17 \\\n--pop-prev nan,nan,nan,0.06\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e deprivation\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg deprivation_ldsc.sumstats.gz,\\\nsystolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out deprivation_rg \\\n--samp-prev nan,nan,0.17 \\\n--pop-prev nan,nan,0.06\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e systolic bp\u003c/span\u003e\n./ldsc/ldsc.py \\\n--rg systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \\\n--ref-ld-chr eur_w_ld_chr/ \\\n--w-ld-chr eur_w_ld_chr/ \\\n--out systolic_bp_rg \\\n--samp-prev nan,0.17 \\\n--pop-prev nan,0.06\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe LDSC output gives us a table of the genetic correlations at the end\nof the .log file that we can easily copy into Excel.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePlotting the data\u003c/h2\u003e\u003ca id=\"user-content-plotting-the-data\" class=\"anchor\" aria-label=\"Permalink: Plotting the data\" href=\"#plotting-the-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe remaining analytic steps were all performed in R Studio (Version\n1.2.5033) using R version 3.6.3 on a Windows OS desktop.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGenetic correlation matrix\u003c/h3\u003e\u003ca id=\"user-content-genetic-correlation-matrix\" class=\"anchor\" aria-label=\"Permalink: Genetic correlation matrix\" href=\"#genetic-correlation-matrix\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eVisualising pairwise genetic correlations between lots of traits in a\ntable is not an easy way to study the results. Plotting your genetic\ncorrelations in a correlation matrix plot provides far better visuals of\nthe correlations allowing for better interpretation of your results. A\ngreat R package for this is \u003cem\u003ecorrplot\u003c/em\u003e and we provide the code script\nbelow for the figure we used in our\n\u003ca href=\"https://www.sciencedirect.com/science/article/pii/S0197458022001312\" rel=\"nofollow\"\u003epaper\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor a detailed description of the different aesthetic options available\nin this package see\n\u003ca href=\"https://cran.r-project.org/web/packages/corrplot/vignettes/corrplot-intro.html\" rel=\"nofollow\"\u003ehere\u003c/a\u003e\nfor a wide range of vignettes.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFirst we load the required R packages and set our working directory. We\nuse \u003cem\u003etidyverse\u003c/em\u003e to turn our .csv file data into appropriate matrices for\nthe \u003ccode\u003ecorrplot\u003c/code\u003e function.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"setwd(\u0026quot;E:/GWAS_sumstats\u0026quot;)\nlibrary(tidyverse)\nlibrary(corrplot)\"\u003e\u003cpre\u003esetwd(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eE:/GWAS_sumstats\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003etidyverse\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003ecorrplot\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIn our matrix plot we wanted to include both the correlation\ncoefficients between the traits but also information about the\nstatistical significance of the correlation. Therefore we initially made\n2 separate .csv files based on the output from the LDSC analysis.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e.csv file 1\u003c/strong\u003e This file consisted of 3 columns named \u003cem\u003etrait_1\u003c/em\u003e,\n\u003cem\u003etrait_2\u003c/em\u003e and \u003cem\u003erg\u003c/em\u003e. We wrote abbreviated trait names so that the labels\non the matrix plot will be easier to read (see image below). \u003cstrong\u003eRg\u003c/strong\u003e\nrefers to the correlation coefficient estimate. As the central line on\nthe matrix plot is the trait correlated with itself we give these\nvariables a value of 1.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e.csv file 2\u003c/strong\u003e This file consisted of 3 columns named \u003cem\u003etrait_1\u003c/em\u003e,\n\u003cem\u003etrait_2\u003c/em\u003e and \u003cem\u003epval\u003c/em\u003e with data in the same order as the 1st file. P\nvalues of traits with themselves are specified as 0.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe then use these two files to construct a rg data matrix and p value\ndata matrix in R using the \u003cem\u003etidyverse\u003c/em\u003e package to use as the input for\nour correlation plot (\u003ccode\u003ecorrplot\u003c/code\u003e only accepts a data matrix as its\ninput).\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Load in the rg csv and make a duplicate for other side of the matrix\nrg \u0026lt;- read.csv(file = \u0026quot;LDSC_all_gen_corr_no_pval.csv\u0026quot;, header = T, sep = \u0026quot;,\u0026quot;)\nrg\n\nrg2\u0026lt;-rg\n\n# Swap the column names for trait1 and trait 2 \n# (because the other half of the triangle is just a transposed version of the extra half we need)\nnames(rg2)\u0026lt;-c(\u0026quot;trait_2\u0026quot;, \u0026quot;trait_1\u0026quot;, \u0026quot;rg\u0026quot;)\nrg2\n  \n# Drop the columns with an rg of 1 that we don't need as these are already in the first dataset\n# and as they represent the matrix intersection we don't need the values twice\n\nrg3\u0026lt;-rg2 %\u0026gt;% mutate(na_if(trait_1, trait_2)) %\u0026gt;% \n  drop_na() %\u0026gt;% \n  select(trait_2, trait_1, rg) #removing the extra column we used to complete this step\nrg3\n\n# Combine to make a matrix \nall\u0026lt;-rbind(rg, rg3) %\u0026gt;% #combine the two sides of the matrix\n  pivot_wider(names_from=trait_1, values_from=rg) %\u0026gt;% # turn data into wide format\n  column_to_rownames(var=\u0026quot;trait_2\u0026quot;) #transfer the trait_2 column to be the row names\n\nall\n\n# Save matrix\nwrite.table(all, file = \u0026quot;ldsc_rg_matrix.txt\u0026quot;)\n\n# Repeat to make p value matrix\noptions(scipen=10)\npval \u0026lt;- read.csv(file = \u0026quot;LDSC_all_pval.csv\u0026quot;, header = T, sep = \u0026quot;,\u0026quot;)\npval\n\npval2\u0026lt;-pval\n\nnames(pval2)\u0026lt;-c(\u0026quot;trait_2\u0026quot;, \u0026quot;trait_1\u0026quot;, \u0026quot;pval\u0026quot;)\npval2\n\npval3\u0026lt;-pval2 %\u0026gt;% mutate(na_if(trait_1, trait_2)) %\u0026gt;% \n  drop_na() %\u0026gt;% \n  select(trait_2, trait_1, pval)\npval3\n\nall_pval\u0026lt;-rbind(pval, pval3) %\u0026gt;% \n  pivot_wider(names_from=trait_1, values_from=pval) %\u0026gt;% \n  column_to_rownames(var=\u0026quot;trait_2\u0026quot;)\n\nall_pval\n\nwrite.table(all_pval, file = \u0026quot;ldsc_pval_matrix.txt\u0026quot;)\n\n# Convert to matrices for corrplot\nall_matrix \u0026lt;- as.matrix(all)\nall_matrix\n\npval_matrix \u0026lt;- as.matrix(all_pval)\npval_matrix\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Load in the rg csv and make a duplicate for other side of the matrix\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.csv(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLDSC_all_gen_corr_no_pval.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003erg2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Swap the column names for trait1 and trait 2 \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e (because the other half of the triangle is just a transposed version of the extra half we need)\u003c/span\u003e\nnames(\u003cspan class=\"pl-smi\"\u003erg2\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etrait_2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etrait_1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erg\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003erg2\u003c/span\u003e\n  \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Drop the columns with an rg of 1 that we don't need as these are already in the first dataset\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e and as they represent the matrix intersection we don't need the values twice\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003erg3\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003erg2\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% mutate(na_if(\u003cspan class=\"pl-smi\"\u003etrait_1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etrait_2\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  drop_na() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  select(\u003cspan class=\"pl-smi\"\u003etrait_2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etrait_1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eremoving the extra column we used to complete this step\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003erg3\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Combine to make a matrix \u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eall\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003erbind(\u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003erg3\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecombine the two sides of the matrix\u003c/span\u003e\n  pivot_wider(\u003cspan class=\"pl-v\"\u003enames_from\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etrait_1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003evalues_from\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e turn data into wide format\u003c/span\u003e\n  column_to_rownames(\u003cspan class=\"pl-v\"\u003evar\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etrait_2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etransfer the trait_2 column to be the row names\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eall\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Save matrix\u003c/span\u003e\nwrite.table(\u003cspan class=\"pl-smi\"\u003eall\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eldsc_rg_matrix.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Repeat to make p value matrix\u003c/span\u003e\noptions(\u003cspan class=\"pl-v\"\u003escipen\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.csv(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLDSC_all_pval.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003epval2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e\n\nnames(\u003cspan class=\"pl-smi\"\u003epval2\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etrait_2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etrait_1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epval\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003epval2\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003epval3\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003epval2\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% mutate(na_if(\u003cspan class=\"pl-smi\"\u003etrait_1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etrait_2\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  drop_na() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  select(\u003cspan class=\"pl-smi\"\u003etrait_2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etrait_1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003epval3\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eall_pval\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003erbind(\u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epval3\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  pivot_wider(\u003cspan class=\"pl-v\"\u003enames_from\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etrait_1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003evalues_from\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  column_to_rownames(\u003cspan class=\"pl-v\"\u003evar\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etrait_2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eall_pval\u003c/span\u003e\n\nwrite.table(\u003cspan class=\"pl-smi\"\u003eall_pval\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eldsc_pval_matrix.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Convert to matrices for corrplot\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eall_matrix\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.matrix(\u003cspan class=\"pl-smi\"\u003eall\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eall_matrix\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003epval_matrix\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.matrix(\u003cspan class=\"pl-smi\"\u003eall_pval\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003epval_matrix\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow we can plot a matrix that depicts the genetic correlation\ncoefficients on the lower triangle of the matrix and the p values on the\nupper triangle (marked by asterisks if they are statistically\nsignificant). We use a Bonferroni corrected p value threshold to adjust\nfor multiple testing.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eBonferroni calculation:\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eStep 1:\u003c/strong\u003e Calculate how many pairwise tests are being performed by:\n\u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://camo.githubusercontent.com/ea2444288bb985558559486efbc2696a9278c58c1b19cfda8fe75afa51c1680d/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b6b2533444e756d6265724f665472616974732532412532384e756d6265724f665472616974732532423125323925324632\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/ea2444288bb985558559486efbc2696a9278c58c1b19cfda8fe75afa51c1680d/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b6b2533444e756d6265724f665472616974732532412532384e756d6265724f665472616974732532423125323925324632\" alt=\"k=NumberOfTraits*(NumberOfTraits+1)/2\" title=\"k=NumberOfTraits*(NumberOfTraits+1)/2\" data-canonical-src=\"https://latex.codecogs.com/png.image?%5Cdpi%7B110%7D\u0026amp;space;%5Cbg_white\u0026amp;space;k%3DNumberOfTraits%2A%28NumberOfTraits%2B1%29%2F2\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eStep 2:\u003c/strong\u003e Then calculate p value adjusting for the number of tests:\n\u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://camo.githubusercontent.com/71927ad8653b130ef6790eb31e5327605f8896535fa2920a887d298960c17c32/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b50253344302e30352532466b\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/71927ad8653b130ef6790eb31e5327605f8896535fa2920a887d298960c17c32/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b50253344302e30352532466b\" alt=\"P=0.05/k\" title=\"P=0.05/k\" data-canonical-src=\"https://latex.codecogs.com/png.image?%5Cdpi%7B110%7D\u0026amp;space;%5Cbg_white\u0026amp;space;P%3D0.05%2Fk\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn our case there were 14 traits giving us 105 pairwise tests with a\ncorrected p value of \u0026lt;4.76E-4.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"col \u0026lt;- colorRampPalette(c(\u0026quot;#BB4444\u0026quot;, \u0026quot;#EE9988\u0026quot;, \u0026quot;#FFFFFF\u0026quot;, \u0026quot;#77AADD\u0026quot;, \u0026quot;#4477AA\u0026quot;))\n# choose colour pallete\n\n# save pdf of corr matrix with asterisks for significant values and no numeric values\npdf(file = \u0026quot;LDSC_all_pval.pdf\u0026quot;, height = 5, width = 5.2)\ncorr_plot \u0026lt;- corrplot(all_matrix, method = \u0026quot;square\u0026quot;, col = col(200),\n                          type = \u0026quot;full\u0026quot;,  order = \u0026quot;original\u0026quot;,\n                          tl.pos = \u0026quot;lt\u0026quot;, tl.col = \u0026quot;black\u0026quot;, tl.srt = 90, tl.cex = .7, # Text label color \u0026amp; rotation\n                          cl.pos = \u0026quot;r\u0026quot;, cl.cex = .7, cl.ratio = 0.1,\n                          # Combine with significance\n                          p.mat = pval_matrix, sig.level = 4.76E-4, insig = \u0026quot;label_sig\u0026quot;,\n                          pch.col = \u0026quot;black\u0026quot;, pch.cex = 1.2, \n                          # hide correlation coefficient on the principal diagonal\n                          diag = TRUE)\ndev.off()\n\n### PDF corr matrix where the upper matrix has p values asterisks \n# and the lower displays correlation coefficients\npdf(file = \u0026quot;LDSC_all_pval_with_values.pdf\u0026quot;, height = 5, width = 5.2)\ncorrplot(all_matrix, method = \u0026quot;color\u0026quot;, col = col(200),\n         addCoef.col = \u0026quot;black\u0026quot;, number.cex = .6, number.digits = 2, # Add coefficient of correlation\n         type = \u0026quot;lower\u0026quot;,  order = \u0026quot;original\u0026quot;,\n         tl.pos = \u0026quot;lt\u0026quot;, tl.col = \u0026quot;black\u0026quot;, tl.srt = 90, tl.cex = .7, # Text label color and rotation\n         cl.pos = \u0026quot;r\u0026quot;, cl.cex = .7, cl.ratio = 0.1, \n         # Combine with significance\n         diag = TRUE)\ncorrplot(all_matrix, add = TRUE, type = \u0026quot;upper\u0026quot;, method = \u0026quot;square\u0026quot;, order = \u0026quot;original\u0026quot;, col = col(200),\n         diag = FALSE, tl.pos = \u0026quot;n\u0026quot;, cl.pos = \u0026quot;n\u0026quot;,\n         p.mat = pval_matrix, sig.level = 4.76E-4, insig = \u0026quot;label_sig\u0026quot;, \n         pch.col = \u0026quot;black\u0026quot;, pch.cex = 1.2)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colorRampPalette(c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#BB4444\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#EE9988\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#FFFFFF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#77AADD\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#4477AA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e choose colour pallete\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e save pdf of corr matrix with asterisks for significant values and no numeric values\u003c/span\u003e\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLDSC_all_pval.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5.2\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ecorr_plot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e corrplot(\u003cspan class=\"pl-smi\"\u003eall_matrix\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esquare\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e col(\u003cspan class=\"pl-c1\"\u003e200\u003c/span\u003e),\n                          \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efull\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoriginal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                          \u003cspan class=\"pl-v\"\u003etl.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etl.col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etl.srt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e90\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etl.cex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.7\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Text label color \u0026amp; rotation\u003c/span\u003e\n                          \u003cspan class=\"pl-v\"\u003ecl.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003er\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecl.cex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.7\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecl.ratio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e,\n                          \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Combine with significance\u003c/span\u003e\n                          \u003cspan class=\"pl-v\"\u003ep.mat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epval_matrix\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esig.level\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e4.76E-4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003einsig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elabel_sig\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                          \u003cspan class=\"pl-v\"\u003epch.col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epch.cex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1.2\u003c/span\u003e, \n                          \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e hide correlation coefficient on the principal diagonal\u003c/span\u003e\n                          \u003cspan class=\"pl-v\"\u003ediag\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## PDF corr matrix where the upper matrix has p values asterisks \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e and the lower displays correlation coefficients\u003c/span\u003e\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLDSC_all_pval_with_values.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5.2\u003c/span\u003e)\ncorrplot(\u003cspan class=\"pl-smi\"\u003eall_matrix\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecolor\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e col(\u003cspan class=\"pl-c1\"\u003e200\u003c/span\u003e),\n         \u003cspan class=\"pl-v\"\u003eaddCoef.col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enumber.cex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.6\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enumber.digits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Add coefficient of correlation\u003c/span\u003e\n         \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elower\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoriginal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n         \u003cspan class=\"pl-v\"\u003etl.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etl.col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etl.srt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e90\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etl.cex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.7\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Text label color and rotation\u003c/span\u003e\n         \u003cspan class=\"pl-v\"\u003ecl.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003er\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecl.cex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.7\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecl.ratio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e, \n         \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Combine with significance\u003c/span\u003e\n         \u003cspan class=\"pl-v\"\u003ediag\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ncorrplot(\u003cspan class=\"pl-smi\"\u003eall_matrix\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eupper\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esquare\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoriginal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e col(\u003cspan class=\"pl-c1\"\u003e200\u003c/span\u003e),\n         \u003cspan class=\"pl-v\"\u003ediag\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etl.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003en\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecl.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003en\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n         \u003cspan class=\"pl-v\"\u003ep.mat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epval_matrix\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esig.level\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e4.76E-4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003einsig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elabel_sig\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n         \u003cspan class=\"pl-v\"\u003epch.col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epch.cex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1.2\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput image:\u003c/strong\u003e\u003c/p\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e\n  \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/ldscmatrix_bivariate.jpg\"\u003e\u003cimg width=\"500\" src=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/ldscmatrix_bivariate.jpg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eUndirected weighted graph\u003c/h3\u003e\u003ca id=\"user-content-undirected-weighted-graph\" class=\"anchor\" aria-label=\"Permalink: Undirected weighted graph\" href=\"#undirected-weighted-graph\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe matrix plot gives us a nice image of pairwise correlations but it\ncan still be hard to distinguish how these inter-relate between multiple\ntraits. Therefore, we decided to plot a weighted undirected graph of the\nstatistically significant genetic correlations to visualise how the\ntraits inter-correlate with one another. For a nice example of this type\nof graph where they include the correlation coefficient on the edges\nplease see \u003cstrong\u003eFigure 1B\u003c/strong\u003e in the \u003ca href=\"https://doi.org/10.1016/j.cell.2019.11.020\" rel=\"nofollow\"\u003egenomicSEM\npaper\u003c/a\u003e by Lee et al (2019).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe use the \u003ccode\u003eqplot\u003c/code\u003e function in the \u003cem\u003eqgraph\u003c/em\u003e R package to make this plot\nwith data functions from the \u003cem\u003etidyverse\u003c/em\u003e package.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(qgraph)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eqgraph\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAs with the LDSC plot, we require an input .csv file with 3 columns\n(\u003cem\u003etrait_1\u003c/em\u003e, \u003cem\u003etrait_2\u003c/em\u003e and \u003cem\u003erg\u003c/em\u003e). However, for rg values that are not\nstatistically significant we write these values as 0 so that the\nnon-significant relationships are not included within our graph.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe then convert the data into a matrix form and save the plot as a pdf.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Load in the .csv file\nrg \u0026lt;- read.csv(file = \u0026quot;ldsc_sig_rg_network.csv\u0026quot;, header = T, sep = \u0026quot;,\u0026quot;)\nrg\n\n# Make duplicate to be the other side of the matrix\n\nrg2\u0026lt;-rg\n\n# Swap the column names for trait 1 and trait 2 \nnames(rg2)\u0026lt;-c(\u0026quot;trait_2\u0026quot;, \u0026quot;trait_1\u0026quot;, \u0026quot;rg\u0026quot;)\nrg2\n\n# Drop the columns with rg of 1 that we don't need as they are already in the first dataset\n# and as they represent the intersection, we don't need the values twice\nrg3\u0026lt;-rg2 %\u0026gt;% mutate(na_if(trait_1, trait_2)) %\u0026gt;% \n  drop_na() %\u0026gt;% \n  select(trait_2, trait_1, rg)\nrg3\n\n# Combine as a matrix \nall\u0026lt;-rbind(rg, rg3) %\u0026gt;% \n  pivot_wider(names_from=trait_1, values_from=rg) %\u0026gt;% # Convert data to wide format\n  column_to_rownames(var=\u0026quot;trait_2\u0026quot;) # Transfer the trait_2 column to be the row names\n\nall\n\n#### make matrix\nall_matrix \u0026lt;- as.matrix(all)\nall_matrix\n\n\n# Make a weighted undirected graph of significant correlations\n\npdf(file = \u0026quot;undirected_weighted_graph.pdf\u0026quot;, height = 4, width = 6)\nqplot \u0026lt;- qgraph(all_matrix, layout = \u0026quot;spring\u0026quot;, minimum = 0, posCol = \u0026quot;darkblue\u0026quot;, \n                negCol = \u0026quot;red\u0026quot;, maximum = 1, labels = colnames(all_matrix), \n                label.scale.equal = FALSE, cut = NULL, edge.labels = FALSE, \n                edge.label.bg = FALSE)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Load in the .csv file\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.csv(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eldsc_sig_rg_network.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Make duplicate to be the other side of the matrix\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003erg2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Swap the column names for trait 1 and trait 2 \u003c/span\u003e\nnames(\u003cspan class=\"pl-smi\"\u003erg2\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etrait_2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etrait_1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erg\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003erg2\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Drop the columns with rg of 1 that we don't need as they are already in the first dataset\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e and as they represent the intersection, we don't need the values twice\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003erg3\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003erg2\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% mutate(na_if(\u003cspan class=\"pl-smi\"\u003etrait_1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etrait_2\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  drop_na() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  select(\u003cspan class=\"pl-smi\"\u003etrait_2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etrait_1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003erg3\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Combine as a matrix \u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eall\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003erbind(\u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003erg3\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  pivot_wider(\u003cspan class=\"pl-v\"\u003enames_from\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etrait_1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003evalues_from\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003erg\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Convert data to wide format\u003c/span\u003e\n  column_to_rownames(\u003cspan class=\"pl-v\"\u003evar\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etrait_2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Transfer the trait_2 column to be the row names\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eall\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### make matrix\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eall_matrix\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.matrix(\u003cspan class=\"pl-smi\"\u003eall\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eall_matrix\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Make a weighted undirected graph of significant correlations\u003c/span\u003e\n\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eundirected_weighted_graph.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eqplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e qgraph(\u003cspan class=\"pl-smi\"\u003eall_matrix\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elayout\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003espring\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eminimum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eposCol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                \u003cspan class=\"pl-v\"\u003enegCol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emaximum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elabels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eall_matrix\u003c/span\u003e), \n                \u003cspan class=\"pl-v\"\u003elabel.scale.equal\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecut\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eedge.labels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \n                \u003cspan class=\"pl-v\"\u003eedge.label.bg\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e\n  \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/undirected_weighted_graph_kunkle.jpg\"\u003e\u003cimg width=\"600\" src=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/undirected_weighted_graph_kunkle.jpg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGenomic SEM\u003c/h1\u003e\u003ca id=\"user-content-genomic-sem\" class=\"anchor\" aria-label=\"Permalink: Genomic SEM\" href=\"#genomic-sem\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow we conduct genomic factor analysis using the \u003cem\u003eGenomicSEM\u003c/em\u003e R package,\nbut there are many other potential analyses one can perform (see the\n\u003cstrong\u003e\u003ca href=\"https://github.com/GenomicSEM/GenomicSEM/wiki\"\u003eoriginal wiki\u003c/a\u003e\u003c/strong\u003e for\nfurther details).\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInstall \u003cem\u003eGenomicSEM\u003c/em\u003e\u003c/h3\u003e\u003ca id=\"user-content-install-genomicsem\" class=\"anchor\" aria-label=\"Permalink: Install GenomicSEM\" href=\"#install-genomicsem\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"install.packages(\u0026quot;devtools\u0026quot;)\nlibrary(devtools)\ninstall_github(\u0026quot;GenomicSEM/GenomicSEM\u0026quot;)\"\u003e\u003cpre\u003einstall.packages(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edevtools\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edevtools\u003c/span\u003e)\ninstall_github(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGenomicSEM/GenomicSEM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSet working directory and load the \u003cem\u003eGenomicSEM\u003c/em\u003e package\u003c/h3\u003e\u003ca id=\"user-content-set-working-directory-and-load-the-genomicsem-package\" class=\"anchor\" aria-label=\"Permalink: Set working directory and load the GenomicSEM package\" href=\"#set-working-directory-and-load-the-genomicsem-package\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"setwd(\u0026quot;E:/GWAS_sumstats\u0026quot;)\nlibrary(GenomicSEM)\"\u003e\u003cpre\u003esetwd(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eE:/GWAS_sumstats\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eGenomicSEM\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMunge data\u003c/h2\u003e\u003ca id=\"user-content-munge-data-1\" class=\"anchor\" aria-label=\"Permalink: Munge data\" href=\"#munge-data-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe munged the data again using the \u003ccode\u003emunge\u003c/code\u003e function in the \u003cem\u003eGenomicSEM\u003c/em\u003e\npackage since this version produces more conservative filtering results\nthan the original LDSC software (it checks whether both the A1 \u003cstrong\u003eand\u003c/strong\u003e\nA2 alleles match the reference file instead of just checking A1, and\nremoves SNPs with missing MAF values).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs with the original LDSC software, certain columns are required for\nthis function to work.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe required columns are:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSNP rsID\u003c/li\u003e\n\u003cli\u003eA1 allele (i.e.the effect allele)\u003c/li\u003e\n\u003cli\u003eA2 allele (i.e.the non-effect allele)\u003c/li\u003e\n\u003cli\u003eThe regression effect (can be logistic or linear)\u003c/li\u003e\n\u003cli\u003eThe p value of the effect\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eAs with the LDSC software, it is preferential to also have MAF and INFO\ncolumns but is not essential since we filter the SNPs to a reference\nlist of well imputed SNP with MAF information. In contrast, the files do\nnot require a specific N column since we provide the overall sample size\nas an argument in the \u003ccode\u003emunge\u003c/code\u003e function (but if there is a column present\nthen the function automatically reads this instead of your provided\nvalues).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn our analysis we used Hapmap 3 SNPs with the MHC region removed as our\nreference genome. This data can be downloaded from:\n\u003ca href=\"https://utexas.app.box.com/s/vkd36n197m8klbaio3yzoxsee6sxo11v\" rel=\"nofollow\"\u003ehttps://utexas.app.box.com/s/vkd36n197m8klbaio3yzoxsee6sxo11v\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"munge(files = c(\u0026quot;Kunkle_etal_Stage1_results.txt\u0026quot;, \u0026quot;MDD2018_ex23andMe.gz\u0026quot;, \u0026quot;Insomnia_sumstats_Jansenetal.txt.gz\u0026quot;, \n                \u0026quot;loneliness_UKB_GWAS_edit.txt\u0026quot;, \u0026quot;no_social_activity_UKB_GWAS_edit.txt\u0026quot;, \n                \u0026quot;hearing_diff_UKB_GWAS_edit.txt\u0026quot;, \u0026quot;low_education_UKB_GWAS_edit.txt\u0026quot;,\n                \u0026quot;physical_inactivity_UKB_GWAS_edit.txt\u0026quot;, \u0026quot;current_smoker_UKB_GWAS_edit.txt\u0026quot;, \n                \u0026quot;alcohol_intake_freq_UKB_GWAS_edit.txt\u0026quot;, \u0026quot;bmi_UKB_GWAS_edit.txt\u0026quot;, \n                \u0026quot;deprivation_UKB_GWAS_edit.txt\u0026quot;, \u0026quot;systolic_bp_UKB_GWAS_edit.txt\u0026quot;, \n                \u0026quot;diabetes_with_rsids.txt\u0026quot;), # a list of the dataset files\n      hm3 = \u0026quot;w_hm3.noMHC.snplist\u0026quot;, ## file of the Hapmap3 SNPs with MHC region removed\n      trait.names = c(\u0026quot;AD\u0026quot;, \u0026quot;MDD\u0026quot;, \u0026quot;insomnia\u0026quot;, \u0026quot;loneliness_UKB\u0026quot;, \u0026quot;no_social_activity_UKB\u0026quot;, \n                      \u0026quot;hearing_difficulty_UKB\u0026quot;, \u0026quot;low_education_UKB\u0026quot;, \u0026quot;physical_inactivity_UKB\u0026quot;, \n                      \u0026quot;current_smoker_UKB\u0026quot;, \u0026quot;alcohol_intake_UKB\u0026quot;, \u0026quot;BMI_UKB\u0026quot;, \u0026quot;deprivation_UKB\u0026quot;, \n                      \u0026quot;systolic_BP_UKB\u0026quot;, \u0026quot;Diabetes\u0026quot;), # list of names of your traits\n      N=c(63926, 173005, 386533, 355583, 360063, 353983, 357549, 359263, 359706, 360726, 354831, \n          360763, 340159, 159208), # list of total sample sizes of your traits\n      info.filter = 0.9, # imputation filter (default used)\n      maf.filter = 0.01) # minor allele frequency filter (default used)\"\u003e\u003cpre\u003emunge(\u003cspan class=\"pl-v\"\u003efiles\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKunkle_etal_Stage1_results.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMDD2018_ex23andMe.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInsomnia_sumstats_Jansenetal.txt.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eloneliness_UKB_GWAS_edit.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eno_social_activity_UKB_GWAS_edit.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehearing_diff_UKB_GWAS_edit.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elow_education_UKB_GWAS_edit.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ephysical_inactivity_UKB_GWAS_edit.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecurrent_smoker_UKB_GWAS_edit.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealcohol_intake_freq_UKB_GWAS_edit.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebmi_UKB_GWAS_edit.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edeprivation_UKB_GWAS_edit.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esystolic_bp_UKB_GWAS_edit.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ediabetes_with_rsids.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e a list of the dataset files\u003c/span\u003e\n      \u003cspan class=\"pl-v\"\u003ehm3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ew_hm3.noMHC.snplist\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# file of the Hapmap3 SNPs with MHC region removed\u003c/span\u003e\n      \u003cspan class=\"pl-v\"\u003etrait.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAD\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMDD\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003einsomnia\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eloneliness_UKB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eno_social_activity_UKB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                      \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehearing_difficulty_UKB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elow_education_UKB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ephysical_inactivity_UKB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                      \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecurrent_smoker_UKB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealcohol_intake_UKB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBMI_UKB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edeprivation_UKB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                      \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esystolic_BP_UKB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDiabetes\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e list of names of your traits\u003c/span\u003e\n      \u003cspan class=\"pl-v\"\u003eN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e63926\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e173005\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e386533\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e355583\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e360063\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e353983\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e357549\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e359263\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e359706\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e360726\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e354831\u003c/span\u003e, \n          \u003cspan class=\"pl-c1\"\u003e360763\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e340159\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e159208\u003c/span\u003e), \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e list of total sample sizes of your traits\u003c/span\u003e\n      \u003cspan class=\"pl-v\"\u003einfo.filter\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.9\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e imputation filter (default used)\u003c/span\u003e\n      \u003cspan class=\"pl-v\"\u003emaf.filter\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.01\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e minor allele frequency filter (default used)\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eExploratory factor analysis performed in the odd autosomes\u003c/h2\u003e\u003ca id=\"user-content-exploratory-factor-analysis-performed-in-the-odd-autosomes\" class=\"anchor\" aria-label=\"Permalink: Exploratory factor analysis performed in the odd autosomes\" href=\"#exploratory-factor-analysis-performed-in-the-odd-autosomes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe then performed exploratory factor analysis in the odd autosomes using\nthe munged summary statistics files as our input.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eLoad the additional required packages\u003c/h3\u003e\u003ca id=\"user-content-load-the-additional-required-packages\" class=\"anchor\" aria-label=\"Permalink: Load the additional required packages\" href=\"#load-the-additional-required-packages\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(Matrix)\nlibrary(stats)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eMatrix\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003estats\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConduct multivariable LD score regression (odd autosomes)\u003c/h3\u003e\u003ca id=\"user-content-conduct-multivariable-ld-score-regression-odd-autosomes\" class=\"anchor\" aria-label=\"Permalink: Conduct multivariable LD score regression (odd autosomes)\" href=\"#conduct-multivariable-ld-score-regression-odd-autosomes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe ran multivariable LD-Score regression to calculate the genetic\ncovariance matrix (\u003cem\u003eS\u003c/em\u003e) and associated sampling covariance matrix (\u003cem\u003eV\u003c/em\u003e)\nusing the odd autosomes only. When you run an analysis where you are\nperforming both exploratory and confirmatory factor analysis splitting\nthe data in this way can guard against model overfitting. If you have\nnon-overlapping replication datasets for each of your traits then the\nbest practice would be to run EFA in the discovery data and the CFA in\nthe replication data, but this was not possible for all of the current\nstudy traits.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs with the initial LD score regression performed using the LDSC\nsoftware, we use pre-calculated European LD scores and weights for our\nanalysis from the Broad Institute computed from 1000 Genomes data\ndownloaded from: \u003ca href=\"https://alkesgroup.broadinstitute.org/LDSCORE/\" rel=\"nofollow\"\u003ehttps://alkesgroup.broadinstitute.org/LDSCORE/\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe use the same population and sample prevalence details as above.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNote:\u003c/strong\u003e This step can take a while to run if you have lots of traits,\nbut for 14 traits it took only \u003cem\u003e2 minutes and 41 seconds\u003c/em\u003e on a personal\ndesktop computer.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"ld \u0026lt;- \u0026quot;eur_w_ld_chr/\u0026quot;\nwld \u0026lt;- \u0026quot;eur_w_ld_chr/\u0026quot; #folder of ld scores \u0026amp; weights\n\ntraits \u0026lt;- c(\u0026quot;AD.sumstats.gz\u0026quot;, \u0026quot;MDD.sumstats.gz\u0026quot;, \u0026quot;insomnia.sumstats.gz\u0026quot;,\n            \u0026quot;loneliness_UKB.sumstats.gz\u0026quot;, \u0026quot;no_social_activity_UKB.sumstats.gz\u0026quot;,\n            \u0026quot;hearing_difficulty_UKB.sumstats.gz\u0026quot;, \u0026quot;low_education_UKB.sumstats.gz\u0026quot;,\n            \u0026quot;physical_inactivity_UKB.sumstats.gz\u0026quot;, \u0026quot;current_smoker_UKB.sumstats.gz\u0026quot;, \n            \u0026quot;alcohol_intake_UKB.sumstats.gz\u0026quot;, \u0026quot;BMI_UKB.sumstats.gz\u0026quot;, \n            \u0026quot;deprivation_UKB.sumstats.gz\u0026quot;, \u0026quot;systolic_BP_UKB.sumstats.gz\u0026quot;,\n            \u0026quot;Diabetes.sumstats.gz\u0026quot;) #munged sumstat files\n\nsample.prev \u0026lt;- c(0.34,0.35,0.28,0.18,0.30,0.38,0.17,0.06,0.10,NA,NA,NA,NA,0.17)\n\npopulation.prev \u0026lt;- c(0.05,0.13,0.32,0.08,0.30,0.39,0.27,0.18,0.27,NA,NA,NA,NA,0.06) \n\ntrait.names \u0026lt;- c(\u0026quot;AD\u0026quot;, \u0026quot;MDD\u0026quot;, \u0026quot;INS\u0026quot;, \u0026quot;LON\u0026quot;, \u0026quot;LSA\u0026quot;, \u0026quot;HD\u0026quot;, \u0026quot;LED\u0026quot;,\n                 \u0026quot;LPA\u0026quot;, \u0026quot;SMK\u0026quot;, \u0026quot;ALC\u0026quot;, \u0026quot;BMI\u0026quot;, \u0026quot;DEP\u0026quot;, \u0026quot;SBP\u0026quot;, \u0026quot;T2DM\u0026quot;)\n\nLDSCoutput_odd \u0026lt;- ldsc(traits, sample.prev, population.prev, ld, wld, trait.names, \n                   ldsc.log=\u0026quot;E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ODD_LDSC\u0026quot;, \n                   select = \u0026quot;ODD\u0026quot;) #specify odd autosomes\n\n\n##optional command to save the ldsc output in case you want to use it in a later R session. \n\nsave(LDSCoutput_odd, file= \u0026quot;E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ODD_LDSC.RData\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eld\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eeur_w_ld_chr/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ewld\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eeur_w_ld_chr/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003efolder of ld scores \u0026amp; weights\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003etraits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAD.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMDD.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003einsomnia.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eloneliness_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eno_social_activity_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehearing_difficulty_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elow_education_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ephysical_inactivity_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecurrent_smoker_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealcohol_intake_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBMI_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edeprivation_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esystolic_BP_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDiabetes.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003emunged sumstat files\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003esample.prev\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0.34\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.35\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.28\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.18\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.30\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.38\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.17\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.06\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.10\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.17\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003epopulation.prev\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.13\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.32\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.08\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.30\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.39\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.27\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.18\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.27\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.06\u003c/span\u003e) \n\n\u003cspan class=\"pl-smi\"\u003etrait.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAD\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMDD\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eINS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLON\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLSA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHD\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLED\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                 \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLPA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSMK\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eALC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBMI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDEP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSBP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eT2DM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eLDSCoutput_odd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ldsc(\u003cspan class=\"pl-smi\"\u003etraits\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esample.prev\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epopulation.prev\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eld\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ewld\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etrait.names\u003c/span\u003e, \n                   \u003cspan class=\"pl-v\"\u003eldsc.log\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eE:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ODD_LDSC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                   \u003cspan class=\"pl-v\"\u003eselect\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eODD\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003especify odd autosomes\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#optional command to save the ldsc output in case you want to use it in a later R session. \u003c/span\u003e\n\nsave(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_odd\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eE:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ODD_LDSC.RData\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNote:\u003c/strong\u003e If you are running an analysis with a lot of traits you may\nhave a k \u0026gt;200. In such cases, you will need to specify this using the\n\u003ccode\u003en.blocks\u003c/code\u003e argument in the \u003ccode\u003eldsc\u003c/code\u003e function because otherwise you will\nnot have enough blocks for the jackknifing procedure of ldsc (number of\nblocks has to be higher than k by at least 1).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eExample calculation for 14 traits (same as the calculation to calculate\nthe number of pairwise tests for Bonferroni correction):\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://camo.githubusercontent.com/1d0ec6ba4836eaf316b27daa0d4c64d153245ac2650c5306a0015f423335aa13/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b6b253344313425324125323831342532423125323925324632\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/1d0ec6ba4836eaf316b27daa0d4c64d153245ac2650c5306a0015f423335aa13/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b6b253344313425324125323831342532423125323925324632\" alt=\"k=14*(14+1)/2\" title=\"k=14*(14+1)/2\" data-canonical-src=\"https://latex.codecogs.com/png.image?%5Cdpi%7B110%7D\u0026amp;space;%5Cbg_white\u0026amp;space;k%3D14%2A%2814%2B1%29%2F2\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eExploratory factor analysis\u003c/h3\u003e\u003ca id=\"user-content-exploratory-factor-analysis\" class=\"anchor\" aria-label=\"Permalink: Exploratory factor analysis\" href=\"#exploratory-factor-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe covariance matrix from the multivariable LDSC output is used as the\ninput for factor analysis. We initially smooth the covariance matrix\nusing the \u003ccode\u003enearPD\u003c/code\u003e function in the \u003cem\u003eMatrix\u003c/em\u003e R package so that if the\ncovariance matrix is non-positive definite it is smoothed prior to\nperforming EFA. In our example the matrix is positive definite but we\nrun the function anyway to make sure and to gain an appropriately\nlabelled matrix so that the trait names are labelled in our factor\nanalysis output.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThen we use the smoothed covariance matrix to test different numbers of\nfactors using the \u003ccode\u003efactanal\u003c/code\u003e function in the R \u003cem\u003estats\u003c/em\u003e package to see\nwhich model will be best to take forward to confirmatory factor\nanalysis. We tested a 2-, 3- and 4-factor model in our analysis and used\npromax rotation (this allows for intercorrelation between the latent\nfactors).\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Ssmooth\u0026lt;-as.matrix((nearPD(LDSCoutput_odd$S, corr = FALSE))$mat)\n\n# 2-factor model\n\nEFA\u0026lt;-factanal(covmat = Ssmooth, factors = 2, rotation = \u0026quot;promax\u0026quot;)\n\nprint(EFA, sort=TRUE) # We sort the results by the magnitude of their loadings\nprint(EFA,digits=3,cutoff=.20,sort=TRUE) #specify the cut-off you want to use\n\n# 3-factor model\n\nEFA2\u0026lt;-factanal(covmat = Ssmooth, factors = 3, rotation = \u0026quot;promax\u0026quot;)\n\nprint(EFA2, sort=TRUE)\nprint(EFA2,digits=3,cutoff=.20,sort=TRUE)\n\n# 4-factor model\n\nEFA3\u0026lt;-factanal(covmat = Ssmooth, factors = 4, rotation = \u0026quot;promax\u0026quot;)\n\nprint(EFA3, sort=TRUE)\nprint(EFA3,digits=3,cutoff=.20,sort=TRUE)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eSsmooth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.matrix((nearPD(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_odd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eS\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecorr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e))\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emat\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e 2-factor model\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eEFA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003efactanal(\u003cspan class=\"pl-v\"\u003ecovmat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSsmooth\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efactors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erotation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epromax\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nprint(\u003cspan class=\"pl-smi\"\u003eEFA\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esort\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e We sort the results by the magnitude of their loadings\u003c/span\u003e\nprint(\u003cspan class=\"pl-smi\"\u003eEFA\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edigits\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecutoff\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.20\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003especify the cut-off you want to use\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e 3-factor model\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eEFA2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003efactanal(\u003cspan class=\"pl-v\"\u003ecovmat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSsmooth\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efactors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erotation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epromax\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nprint(\u003cspan class=\"pl-smi\"\u003eEFA2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esort\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\nprint(\u003cspan class=\"pl-smi\"\u003eEFA2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edigits\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecutoff\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.20\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e 4-factor model\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eEFA3\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003efactanal(\u003cspan class=\"pl-v\"\u003ecovmat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSsmooth\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efactors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erotation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epromax\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nprint(\u003cspan class=\"pl-smi\"\u003eEFA3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esort\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\nprint(\u003cspan class=\"pl-smi\"\u003eEFA3\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edigits\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecutoff\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.20\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe output prints details about the factor loadings, the factor scores,\nthe factor correlations and the uniqueness scores for each individual\ntrait. From this analysis we chose to base our subsequent analyses on a\n3-factor model. Please see our\n\u003ca href=\"https://www.sciencedirect.com/science/article/pii/S0197458022001312\" rel=\"nofollow\"\u003epaper\u003c/a\u003e\nfor further details about the methods and how we chose how many factors\nto extract based on the results from the above analysis.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eEFA loadings plot\u003c/h3\u003e\u003ca id=\"user-content-efa-loadings-plot\" class=\"anchor\" aria-label=\"Permalink: EFA loadings plot\" href=\"#efa-loadings-plot\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe factor loadings table can be a little hard to visualise and\ninterpret, so we decided to make a plot of the loadings for our 3-factor\nmodel based on an adapted version of code script written by Dan Mirman\n(see \u003ca href=\"https://rpubs.com/danmirman/plotting_factor_analysis\" rel=\"nofollow\"\u003ehere\u003c/a\u003e). This\nstep uses the \u003cem\u003eggplot2\u003c/em\u003e R package.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eLoad in the required additional packages\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(ggplot2)\nlibrary(tidyr)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eggplot2\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003etidyr\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe made a .csv file in Excel of the factor loadings from the EFA step to\nuse as input for this code. However, we flipped the order of the traits\nin the .csv file so that they appear in the correct order in the graph\n(otherwise they will appear in reverse order).\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Load in csv file\n\nEFA_loadings_sorted \u0026lt;- read.csv(file = \u0026quot;EFA_loadings_odd_ordered.csv\u0026quot;)\nEFA_loadings_sorted\n\n# This step ensures that the traits stay in the order you want them in the final plot \n# (note this is still in reverse order here), otherwise they are plotted in a random order\n\nEFA_loadings_sorted$Trait\u0026lt;- factor(EFA_loadings_sorted$Trait, \n                                   levels = EFA_loadings_sorted$Trait)\n\n# Makes the plot long so column of factors \n# (use the gather function of the tidyr package)\n\nEFA_loading_long \u0026lt;- gather(EFA_loadings_sorted, key=\u0026quot;Factor\u0026quot;, \n                           value=\u0026quot;Loading\u0026quot;, c(\u0026quot;Factor_1\u0026quot;, \u0026quot;Factor_2\u0026quot;, \u0026quot;Factor_3\u0026quot;))\n\nEFA_loading_long\n\n######## Ensure the factor facets come out in the order you want\n\nEFA_loading_long$Factor \u0026lt;- factor(EFA_loading_long$Factor, \n                                  levels = c(\u0026quot;Factor_1\u0026quot;, \u0026quot;Factor_2\u0026quot;, \u0026quot;Factor_3\u0026quot;))\n\n# Add a column to categorise the loadings by loading strength\n\nEFA_loading_long$breaks \u0026lt;- cut(EFA_loading_long$Loading, \n                               breaks = c(-Inf, 0, .2, .4, Inf)) \n\nEFA_loading_long\n\n# Create a name vector for labelling facets\n\nfacet_names \u0026lt;- c(\n  `Factor_1` = \u0026quot;Factor 1\u0026quot;,\n  `Factor_2` = \u0026quot;Factor 2\u0026quot;,\n  `Factor_3` = \u0026quot;Factor 3\u0026quot;)\n\n# Make your plot with a chosen theme \u0026amp; save as a pdf\n\ntheme_set(theme_bw()) # set a chosen theme\npdf(file = \u0026quot;EFA_loadings_plot_3factors_odd.pdf\u0026quot;, height = 5, width = 9)\nEFA_loadings_plot \u0026lt;- ggplot(EFA_loading_long, aes(Trait, abs(Loading), fill=breaks)) + \n  facet_wrap(~ Factor, nrow=1, \n             labeller = as_labeller(facet_names)) + #place the factors in separate facets\n  geom_bar(stat = \u0026quot;identity\u0026quot;) + #make the bars\n  coord_flip() + #flip the axes so the labels can be horizontal  \n  #define the fill color gradient: blue=positive, red=negative\n  scale_fill_manual(name = \u0026quot;Loading Strength\u0026quot;, \n                    labels = c(\u0026quot;Negative\u0026quot;, \u0026quot;0 to .20\u0026quot;, \u0026quot;.20 to .40\u0026quot;, \u0026quot;Greater than .40\u0026quot;), \n                    values = c(\u0026quot;#870E0EAA\u0026quot;, \u0026quot;#256C33AA\u0026quot;, \u0026quot;#316DB2AA\u0026quot;, \u0026quot;#041A58AA\u0026quot;)) + \n  ### tell what colours to assign to each break\n  scale_y_continuous(limits = c(0, 1), expand = c(0, 0),\n                     breaks = seq(0, 1, by = .2)) + # change the scale of the y axis\n  ylab(\u0026quot;Standardised Factor Loading\u0026quot;) + #name the y-axis \n  theme(panel.background = element_rect(fill = \u0026quot;snow3\u0026quot;, colour = \u0026quot;snow3\u0026quot;, size = 2, \n                                        linetype = \u0026quot;solid\u0026quot;), \n    panel.grid.major = element_line(size = 0.4, linetype = 'solid', colour = \u0026quot;white\u0026quot;), \n    panel.grid.minor = element_line(size = 0.2, linetype = 'solid', colour = \u0026quot;white\u0026quot;),\n    strip.background = element_rect(fill = \u0026quot;white\u0026quot;, colour = \u0026quot;black\u0026quot;),\n    strip.text.x = element_text(colour = \u0026quot;black\u0026quot;),\n    axis.text = element_text(colour = \u0026quot;gray20\u0026quot;),\n    panel.spacing.x=unit(1, \u0026quot;lines\u0026quot;)) +\n  geom_hline(yintercept=c(.20), linetype=\u0026quot;dashed\u0026quot;, color = \u0026quot;black\u0026quot;) # add cut-off line\nEFA_loadings_plot\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Load in csv file\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eEFA_loadings_sorted\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.csv(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEFA_loadings_odd_ordered.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eEFA_loadings_sorted\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e This step ensures that the traits stay in the order you want them in the final plot \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e (note this is still in reverse order here), otherwise they are plotted in a random order\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eEFA_loadings_sorted\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eTrait\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eEFA_loadings_sorted\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eTrait\u003c/span\u003e, \n                                   \u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eEFA_loadings_sorted\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eTrait\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Makes the plot long so column of factors \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e (use the gather function of the tidyr package)\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eEFA_loading_long\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e gather(\u003cspan class=\"pl-smi\"\u003eEFA_loadings_sorted\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ekey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFactor\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                           \u003cspan class=\"pl-v\"\u003evalue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLoading\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFactor_1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFactor_2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFactor_3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003eEFA_loading_long\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e####### Ensure the factor facets come out in the order you want\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eEFA_loading_long\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFactor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eEFA_loading_long\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFactor\u003c/span\u003e, \n                                  \u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFactor_1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFactor_2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFactor_3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Add a column to categorise the loadings by loading strength\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eEFA_loading_long\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ebreaks\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e cut(\u003cspan class=\"pl-smi\"\u003eEFA_loading_long\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLoading\u003c/span\u003e, \n                               \u003cspan class=\"pl-v\"\u003ebreaks\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003e.2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003e.4\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)) \n\n\u003cspan class=\"pl-smi\"\u003eEFA_loading_long\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Create a name vector for labelling facets\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003efacet_names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\n  \u003cspan class=\"pl-smi\"\u003e`Factor_1`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFactor 1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-smi\"\u003e`Factor_2`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFactor 2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-smi\"\u003e`Factor_3`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFactor 3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Make your plot with a chosen theme \u0026amp; save as a pdf\u003c/span\u003e\n\ntheme_set(theme_bw()) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e set a chosen theme\u003c/span\u003e\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEFA_loadings_plot_3factors_odd.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eEFA_loadings_plot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ggplot(\u003cspan class=\"pl-smi\"\u003eEFA_loading_long\u003c/span\u003e, aes(\u003cspan class=\"pl-smi\"\u003eTrait\u003c/span\u003e, abs(\u003cspan class=\"pl-smi\"\u003eLoading\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ebreaks\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  facet_wrap(\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFactor\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \n             \u003cspan class=\"pl-v\"\u003elabeller\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e as_labeller(\u003cspan class=\"pl-smi\"\u003efacet_names\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eplace the factors in separate facets\u003c/span\u003e\n  geom_bar(\u003cspan class=\"pl-v\"\u003estat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eidentity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003emake the bars\u003c/span\u003e\n  coord_flip() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eflip the axes so the labels can be horizontal  \u003c/span\u003e\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003edefine the fill color gradient: blue=positive, red=negative\u003c/span\u003e\n  scale_fill_manual(\u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLoading Strength\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                    \u003cspan class=\"pl-v\"\u003elabels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNegative\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e0 to .20\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.20 to .40\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGreater than .40\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n                    \u003cspan class=\"pl-v\"\u003evalues\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#870E0EAA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#256C33AA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#316DB2AA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#041A58AA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## tell what colours to assign to each break\u003c/span\u003e\n  scale_y_continuous(\u003cspan class=\"pl-v\"\u003elimits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eexpand\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e),\n                     \u003cspan class=\"pl-v\"\u003ebreaks\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e seq(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.2\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e change the scale of the y axis\u003c/span\u003e\n  ylab(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eStandardised Factor Loading\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ename the y-axis \u003c/span\u003e\n  theme(\u003cspan class=\"pl-v\"\u003epanel.background\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_rect(\u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esnow3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esnow3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \n                                        \u003cspan class=\"pl-v\"\u003elinetype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esolid\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n    \u003cspan class=\"pl-v\"\u003epanel.grid.major\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_line(\u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elinetype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esolid\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n    \u003cspan class=\"pl-v\"\u003epanel.grid.minor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_line(\u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elinetype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esolid\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n    \u003cspan class=\"pl-v\"\u003estrip.background\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_rect(\u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n    \u003cspan class=\"pl-v\"\u003estrip.text.x\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n    \u003cspan class=\"pl-v\"\u003eaxis.text\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egray20\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n    \u003cspan class=\"pl-v\"\u003epanel.spacing.x\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eunit(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elines\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_hline(\u003cspan class=\"pl-v\"\u003eyintercept\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-smi\"\u003e.20\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elinetype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edashed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e add cut-off line\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eEFA_loadings_plot\u003c/span\u003e\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe above code gives us our 3-factor EFA graph:\u003c/p\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e\n  \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/EFA_loadings_plot_Kunkle.jpg\"\u003e\u003cimg width=\"700\" src=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/EFA_loadings_plot_Kunkle.jpg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConfirmatory factor analysis performed in the even autosomes\u003c/h2\u003e\u003ca id=\"user-content-confirmatory-factor-analysis-performed-in-the-even-autosomes\" class=\"anchor\" aria-label=\"Permalink: Confirmatory factor analysis performed in the even autosomes\" href=\"#confirmatory-factor-analysis-performed-in-the-even-autosomes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNext, we want to test the fit of our model using confirmatory factor\nanalysis.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConduct multivariable LD score regression (even autosomes)\u003c/h3\u003e\u003ca id=\"user-content-conduct-multivariable-ld-score-regression-even-autosomes\" class=\"anchor\" aria-label=\"Permalink: Conduct multivariable LD score regression (even autosomes)\" href=\"#conduct-multivariable-ld-score-regression-even-autosomes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBecause we are now performing the CFA in even autosome data we must\nre-run the multivariable LD score regression step to estimate the\ngenetic covariance matrix for the even autosome.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"LDSCoutput_even \u0026lt;- ldsc(traits, sample.prev, population.prev, ld, wld, trait.names, \n                   ldsc.log=\u0026quot;E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_EVEN_LDSC\u0026quot;, \n                   select = \u0026quot;EVEN\u0026quot;) #specify even autosomes\n\nsave(LDSCoutput_even, file= \u0026quot;E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_EVEN_LDSC.RData\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eLDSCoutput_even\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ldsc(\u003cspan class=\"pl-smi\"\u003etraits\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esample.prev\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epopulation.prev\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eld\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ewld\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etrait.names\u003c/span\u003e, \n                   \u003cspan class=\"pl-v\"\u003eldsc.log\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eE:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_EVEN_LDSC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                   \u003cspan class=\"pl-v\"\u003eselect\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEVEN\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003especify even autosomes\u003c/span\u003e\n\nsave(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_even\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eE:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_EVEN_LDSC.RData\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConfirmatory factor analysis\u003c/h3\u003e\u003ca id=\"user-content-confirmatory-factor-analysis\" class=\"anchor\" aria-label=\"Permalink: Confirmatory factor analysis\" href=\"#confirmatory-factor-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUsing the output from the even autosome LD score regression we can then\ntest different models using the \u003ccode\u003eusermodel\u003c/code\u003e function in the \u003cem\u003eGenomicSEM\u003c/em\u003e\nR package. We specify the parameters of our model based on the output of\nour chosen EFA model (in this case the 3-factor model), including only\nthe factor loadings that pass our pre-specified cut-off level (0.20).\nModel parameters are written in \u003cem\u003elavaan\u003c/em\u003e syntax.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe test a number of different CFA models to see how including negative\nloadings, cross-loadings and higher loading cut-off points influence the\nmodel fit statistics.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe output gives you the model fit statistics and the estimated\nparameters for your models which can be drawn as path diagrams to better\nvisualise your model. The \u003cstrong\u003eunstandardised\u003c/strong\u003e estimates are equivalent to\ncovariance values whereas the \u003cstrong\u003estandardised\u003c/strong\u003e estimates are equivalent\nto correlation coefficients.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe use the default \u003ccode\u003eDWLS\u003c/code\u003e estimation method but the models can also be\nestimated using maximum likelhood \u003ccode\u003eML\u003c/code\u003e estimation if you prefer.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Model 1: CFA of 3-factor model: including traits with \u0026gt;.20 positive loadings and cross-loadings (DWLS method),\n# no negative loadings; factor correlations present \n\nCFAofEFA \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED + MDD + INS + HD + BMI\nF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + INS + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput \u0026lt;- usermodel(LDSCoutput_even, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput\n\n# Model 2: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and cross-loadings (DWLS method),\n# negative loadings included; factor correlations present \n\nCFAofEFA2 \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED + MDD + INS + HD + BMI + AD\nF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + INS + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS + SBP\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput2\u0026lt;- usermodel(LDSCoutput_even, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA2, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput2\n\n# Model 3: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and no cross-loadings (DWLS method),\n# negative loadings included; factor correlations present \n\nCFAofEFA3 \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP\nF2 =~ NA*LSA + LED + ALC + AD + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS + SBP\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput3 \u0026lt;- usermodel(LDSCoutput_even, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA3, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput3\n\n# Model 4: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and no cross-loadings (DWLS method),\n# negative loadings excluded; factor correlations present \n\nCFAofEFA4 \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP\nF2 =~ NA*LSA + LED + ALC + AD + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput4 \u0026lt;- usermodel(LDSCoutput_even, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA4, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput4\n\n# Model 5: CFA of 3-factor model: including traits with \u0026gt;.25 positive loadings and cross-loadings (DWLS method),\n# no negative loadings; factor correlations present (KEPT INSOMNIA IN AS .24)\n\nCFAofEFA5 \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED\nF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput5 \u0026lt;- usermodel(LDSCoutput_even, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA5, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput5\n\n# Model 6: CFA of 3-factor model: including traits with \u0026gt;.25 positive loadings and cross-loadings (DWLS method),\n# negative loadings incl; factor correlations present (KEPT INSOMNIA IN AS .24)\n\nCFAofEFA6 \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED\nF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS + SBP\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\n\nCFAoutput6 \u0026lt;- usermodel(LDSCoutput_even, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA6, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput6\n\n## .30 CUTOFF WAS THE SAME AS .25 SO DIDN'T RUN THAT\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 1: CFA of 3-factor model: including traits with \u0026gt;.20 positive loadings and cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e no negative loadings; factor correlations present \u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED + MDD + INS + HD + BMI\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + INS + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_even\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 2: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e negative loadings included; factor correlations present \u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED + MDD + INS + HD + BMI + AD\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + INS + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS + SBP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_even\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput2\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 3: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and no cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e negative loadings included; factor correlations present \u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LSA + LED + ALC + AD + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS + SBP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_even\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput3\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 4: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and no cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e negative loadings excluded; factor correlations present \u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA4\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LSA + LED + ALC + AD + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput4\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_even\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput4\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 5: CFA of 3-factor model: including traits with \u0026gt;.25 positive loadings and cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e no negative loadings; factor correlations present (KEPT INSOMNIA IN AS .24)\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA5\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput5\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_even\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput5\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 6: CFA of 3-factor model: including traits with \u0026gt;.25 positive loadings and cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e negative loadings incl; factor correlations present (KEPT INSOMNIA IN AS .24)\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA6\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS + SBP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput6\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_even\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA6\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput6\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# .30 CUTOFF WAS THE SAME AS .25 SO DIDN'T RUN THAT\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePlotting CFA models\u003c/h3\u003e\u003ca id=\"user-content-plotting-cfa-models\" class=\"anchor\" aria-label=\"Permalink: Plotting CFA models\" href=\"#plotting-cfa-models\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAlthough there are specific R packages available that plot path diagrams\nfor SEM automatically e.g.\u003cem\u003esemPlot\u003c/em\u003e, we found that the clearest and\nbest way is to draw out your path diagrams in PowerPoint.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eBelow is an example of a path diagram drawn using this method based on\nthe standardised results of the CFA for \u003cem\u003emodel 5\u003c/em\u003e in the even autosomes\n(see code above). This was the best fitting model that we tested so we\ntook it forward for subsequent analysis in data from all autosomes.\u003c/p\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e\n  \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/CFA_even_best_model.png\"\u003e\u003cimg width=\"700\" src=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/CFA_even_best_model.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFollow-up genome-wide analyses\u003c/h2\u003e\u003ca id=\"user-content-follow-up-genome-wide-analyses\" class=\"anchor\" aria-label=\"Permalink: Follow-up genome-wide analyses\" href=\"#follow-up-genome-wide-analyses\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow we want to measure the fit of our best-fitting model and other more\ncomplex SEM models using data from across the genome, so we run a final\nmultivariable LD score regression in all chromosomes.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGenome-wide multivariable LD score regression\u003c/h3\u003e\u003ca id=\"user-content-genome-wide-multivariable-ld-score-regression\" class=\"anchor\" aria-label=\"Permalink: Genome-wide multivariable LD score regression\" href=\"#genome-wide-multivariable-ld-score-regression\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"LDSCoutput_all \u0026lt;- ldsc(traits, sample.prev, population.prev, ld, wld, trait.names, \n                   ldsc.log=\u0026quot;E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC\u0026quot;)\n\nsave(LDSCoutput_all, file= \u0026quot;E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC.RData\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ldsc(\u003cspan class=\"pl-smi\"\u003etraits\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esample.prev\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epopulation.prev\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eld\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ewld\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etrait.names\u003c/span\u003e, \n                   \u003cspan class=\"pl-v\"\u003eldsc.log\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eE:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nsave(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eE:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC.RData\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGenome-wide CFA\u003c/h3\u003e\u003ca id=\"user-content-genome-wide-cfa\" class=\"anchor\" aria-label=\"Permalink: Genome-wide CFA\" href=\"#genome-wide-cfa\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe then run the 3-factor model (based off Model 5 in the code script of\nthe even CFA) using the LDSC output estimated from all the chromosomes.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"CFAofEFAall \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED\nF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput_all \u0026lt;- usermodel(LDSCoutput_all, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFAall, \n                           CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\n\nCFAoutput_all\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eCFAofEFAall\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_all\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFAall\u003c/span\u003e, \n                           \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_all\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eThe path diagram of the best-fitting CFA model (all autosomes):\u003c/em\u003e\u003c/p\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e\n  \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/CFA_all_best_model.png\"\u003e\u003cimg width=\"700\" src=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/CFA_all_best_model.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAdditional SEM modelling\u003c/h3\u003e\u003ca id=\"user-content-additional-sem-modelling\" class=\"anchor\" aria-label=\"Permalink: Additional SEM modelling\" href=\"#additional-sem-modelling\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe genome-wide 3-factor CFA model fits the data moderately well.\nHowever, there is high inter-factor correlation present, which makes it\nhard to meaningfully interpret distinct patterns of covariance.\nTherefore, we tested a number of additional SEM models in the\n\u003cem\u003eGenomicSEM\u003c/em\u003e R package to try and disentangle the covariance\nrelationships in more detail. We provide examples of these models below\nwith brief explanations and example path diagrams.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommon factor model\u003c/h3\u003e\u003ca id=\"user-content-common-factor-model\" class=\"anchor\" aria-label=\"Permalink: Common factor model\" href=\"#common-factor-model\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eA common factor model is a model where there is a single latent\nconstruct representing common variance between all of the traits in your\ndata. In our example although the traits load into a common factor, the\nmodel fit statistics are poor, which suggests that the data is better\nrepresented by a model with multiple latent constructs.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe run this model using the \u003ccode\u003ecommonfactor\u003c/code\u003e function in \u003cem\u003eGenomicSEM\u003c/em\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"CommonFactor\u0026lt;- commonfactor(covstruc = LDSCoutput_all, estimation=\u0026quot;DWLS\u0026quot;)\n\nCommonFactor\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eCommonFactor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e commonfactor(\u003cspan class=\"pl-v\"\u003ecovstruc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eCommonFactor\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eThe path diagram of the common factor model:\u003c/em\u003e\u003c/p\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e\n  \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/CommonFactor_all_SBP.png\"\u003e\u003cimg width=\"700\" src=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/CommonFactor_all_SBP.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNote:\u003c/strong\u003e Here in this walkthrough we use the LDSC output that includes\nsystolic blood pressure in the common factor model. However, in our\n\u003ca href=\"https://www.sciencedirect.com/science/article/pii/S0197458022001312\" rel=\"nofollow\"\u003epaper\u003c/a\u003e\nwe excluded systolic blood pressure in our genome-wide LD score\nregression to estimate the parameters without it included.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHierarchal model\u003c/h3\u003e\u003ca id=\"user-content-hierarchal-model\" class=\"anchor\" aria-label=\"Permalink: Hierarchal model\" href=\"#hierarchal-model\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eA hierarchal model refers to a structural equation model that has two\nlevels of latent constructs (i.e.a latent construct of shared\ncovariance between the specified lower level latent constructs).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHere we run a hierarchal model of our 3-factor CFA model.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"hierarchal \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED\nF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS\nF4 =~ F1 + F2 + F3'\n\nhierarchal \u0026lt;- usermodel(LDSCoutput_all, estimation = \u0026quot;DWLS\u0026quot;, model = hierarchal, \n                         CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nhierarchal\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ehierarchal\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF4 =~ F1 + F2 + F3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003ehierarchal\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ehierarchal\u003c/span\u003e, \n                         \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ehierarchal\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eThe path diagram of the hierarchal model:\u003c/em\u003e\u003c/p\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e\n  \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/hierarchal_all.png\"\u003e\u003cimg width=\"700\" src=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/hierarchal_all.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNote:\u003c/strong\u003e As you can see in our path diagram of the hierarchal model we\ndo not include the standard errors of our parameters. This is because\nthe model includes an endogenous variable so the STD_All column of the\nresults output is used to specify the standardised values. However, the\n\u003cem\u003eGenomicSEM\u003c/em\u003e package is unable to calculate standard errors for these\nestimates (see the section titled \u003cstrong\u003eA Note on Standardized Output in\nGenomic SEM\u003c/strong\u003e on the original \u003ca href=\"https://github.com/GenomicSEM/GenomicSEM/wiki/3.-Models-without-Individual-SNP-effects\"\u003ewiki\npage\u003c/a\u003e\nfor further explanation).\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eBifactor model\u003c/h3\u003e\u003ca id=\"user-content-bifactor-model\" class=\"anchor\" aria-label=\"Permalink: Bifactor model\" href=\"#bifactor-model\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFinally, we test the fit of a bifactor model. This model produces\northogonal factors (i.e.uncorrelated factors) by including an\noverarching general latent factor that all the traits load into, as well\nas additional latent factors that only distinct sub-clusters of traits\nload into. The benefit of these models is that they often provide better\nmodel fit and allow for an enhanced interpretation of what the\nsub-clusters might be capturing.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn our analysis, the bifactor model provided a good fit to our data and\nthe sub-clusters were specified based on the 3 factors in our CFA.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNote:\u003c/strong\u003e For the bifactor model we had to constrain smoking to ensure\nit didnt output as a negative residual variance (see \u003ca href=\"https://github.com/GenomicSEM/GenomicSEM/wiki/3.-Models-without-Individual-SNP-effects\"\u003eoriginal\nwiki\u003c/a\u003e\nfor further details on this).\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"bifactor \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED\nF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS\nF4 =~ NA*AD + MDD + INS + LON + LSA + HD + LED + LPA + SMK + ALC + BMI + DEP + T2DM\nF4 ~~ 0*F1\nF4 ~~ 0*F2\nF4 ~~ 0*F3\nF1 ~~ 0*F2\nF1 ~~ 0*F3\nF2 ~~ 0*F3\nSMK ~~  a*SMK\na \u0026gt; .001'\n\nbifactor \u0026lt;- usermodel(LDSCoutput_all, estimation = \u0026quot;DWLS\u0026quot;, model = bifactor, \n                      CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nbifactor\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ebifactor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF4 =~ NA*AD + MDD + INS + LON + LSA + HD + LED + LPA + SMK + ALC + BMI + DEP + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF4 ~~ 0*F1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF4 ~~ 0*F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF4 ~~ 0*F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ 0*F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ 0*F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ 0*F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eSMK ~~  a*SMK\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ea \u0026gt; .001\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003ebifactor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebifactor\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ebifactor\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eThe path diagram of the bifactor model:\u003c/em\u003e\u003c/p\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e\n  \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/bifactor_all.png\"\u003e\u003cimg width=\"700\" src=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/bifactor_all.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eModels without the APOE region\u003c/h2\u003e\u003ca id=\"user-content-models-without-the-apoe-region\" class=\"anchor\" aria-label=\"Permalink: Models without the APOE region\" href=\"#models-without-the-apoe-region\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSince Alzheimers disease risk is known to be disproportionately\ninfluenced by variants in the \u003cem\u003eAPOE\u003c/em\u003e gene region, we ran the same models\nas above in all of the autosomes except for the \u003cem\u003eAPOE\u003c/em\u003e region on\nchromosome 19. We removed SNPs in the \u003cem\u003eAPOE\u003c/em\u003e region +/- 100kB (chr19:\n45,309,03945,512,650) based on the position in the genome assembly that\nthe GWAS summary statistics had been provided in (GRCh37/hg19).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe downloaded the \u003cem\u003eAPOE\u003c/em\u003e snplist from the \u003ca href=\"https://genome.ucsc.edu/index.html\" rel=\"nofollow\"\u003eUCSC Genome\nBrowser\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## read in snplist of APOE region +/- 100kb each side from UCSC browser for genome build 37 to match GWAS sumstats - 249 SNPs\n\napoe \u0026lt;- read.table(\u0026quot;APOE_snps.txt\u0026quot;, header = F, sep = \u0026quot;\u0026quot;, check.names = F)\n\n## Read in munged sumstats files and remove any SNPs that match the rsIDs in the APOE snplist file\n\nAD \u0026lt;- fread(\u0026quot;AD.sumstats.gz\u0026quot;)\nAD_no_apoe \u0026lt;- anti_join(AD, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(AD_no_apoe, file = \u0026quot;AD_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nMDD \u0026lt;- fread(\u0026quot;MDD.sumstats.gz\u0026quot;)\nMDD_no_apoe \u0026lt;- anti_join(MDD, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(MDD_no_apoe, file = \u0026quot;MDD_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nINS \u0026lt;- fread(\u0026quot;insomnia.sumstats.gz\u0026quot;)\nINS_no_apoe \u0026lt;- anti_join(INS, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(INS_no_apoe, file = \u0026quot;INS_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nLON \u0026lt;- fread(\u0026quot;loneliness_UKB.sumstats.gz\u0026quot;)\nLON_no_apoe \u0026lt;- anti_join(LON, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(LON_no_apoe, file = \u0026quot;LON_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nLSA \u0026lt;- fread(\u0026quot;no_social_activity_UKB.sumstats.gz\u0026quot;)\nLSA_no_apoe \u0026lt;- anti_join(LSA, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(LSA_no_apoe, file = \u0026quot;LSA_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nHD \u0026lt;- fread(\u0026quot;hearing_difficulty_UKB.sumstats.gz\u0026quot;)\nHD_no_apoe \u0026lt;- anti_join(HD, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(HD_no_apoe, file = \u0026quot;HD_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nLED \u0026lt;- fread(\u0026quot;low_education_UKB.sumstats.gz\u0026quot;)\nLED_no_apoe \u0026lt;- anti_join(LED, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(LED_no_apoe, file = \u0026quot;LED_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nLPA \u0026lt;- fread(\u0026quot;physical_inactivity_UKB.sumstats.gz\u0026quot;)\nLPA_no_apoe \u0026lt;- anti_join(LPA, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(LPA_no_apoe, file = \u0026quot;LPA_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nSMK \u0026lt;- fread(\u0026quot;current_smoker_UKB.sumstats.gz\u0026quot;)\nSMK_no_apoe \u0026lt;- anti_join(SMK, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(SMK_no_apoe, file = \u0026quot;SMK_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nALC \u0026lt;- fread(\u0026quot;alcohol_intake_UKB.sumstats.gz\u0026quot;)\nALC_no_apoe \u0026lt;- anti_join(ALC, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(ALC_no_apoe, file = \u0026quot;ALC_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nBMI \u0026lt;- fread(\u0026quot;BMI_UKB.sumstats.gz\u0026quot;)\nBMI_no_apoe \u0026lt;- anti_join(BMI, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(BMI_no_apoe, file = \u0026quot;BMI_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nDEP \u0026lt;- fread(\u0026quot;deprivation_UKB.sumstats.gz\u0026quot;)\nDEP_no_apoe \u0026lt;- anti_join(DEP, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(DEP_no_apoe, file = \u0026quot;DEP_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nSBP \u0026lt;- fread(\u0026quot;systolic_BP_UKB.sumstats.gz\u0026quot;)\nSBP_no_apoe \u0026lt;- anti_join(SBP, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(SBP_no_apoe, file = \u0026quot;SBP_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\nT2DM \u0026lt;- fread(\u0026quot;Diabetes.sumstats.gz\u0026quot;)\nT2DM_no_apoe \u0026lt;- anti_join(T2DM, apoe, by = c(\u0026quot;SNP\u0026quot; = \u0026quot;V4\u0026quot;))\nwrite.table(T2DM_no_apoe, file = \u0026quot;T2DM_no_apoe.sumstats.gz\u0026quot;, row.names = F, quote = F)\n\n\n# Multivariable LD score regression of data without APOE\n\ntraits.apoe \u0026lt;- c(\u0026quot;AD_no_apoe.sumstats.gz\u0026quot;, \u0026quot;MDD_no_apoe.sumstats.gz\u0026quot;, \u0026quot;INS_no_apoe.sumstats.gz\u0026quot;, \u0026quot;LON_no_apoe.sumstats.gz\u0026quot;, \n                 \u0026quot;LSA_no_apoe.sumstats.gz\u0026quot;, \u0026quot;HD_no_apoe.sumstats.gz\u0026quot;, \u0026quot;LED_no_apoe.sumstats.gz\u0026quot;, \u0026quot;LPA_no_apoe.sumstats.gz\u0026quot;, \n                 \u0026quot;SMK_no_apoe.sumstats.gz\u0026quot;, \u0026quot;ALC_no_apoe.sumstats.gz\u0026quot;, \u0026quot;BMI_no_apoe.sumstats.gz\u0026quot;, \u0026quot;DEP_no_apoe.sumstats.gz\u0026quot;,\n                 \u0026quot;SBP_no_apoe.sumstats.gz\u0026quot;, \u0026quot;T2DM_no_apoe.sumstats.gz\u0026quot;)\n\nLDSCoutput_exAPOE \u0026lt;- ldsc(traits.apoe, sample.prev, population.prev, ld, wld, trait.names, \n                   ldsc.log=\u0026quot;E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC_noAPOE\u0026quot;)\n\nsave(LDSCoutput_exAPOE, file= \u0026quot;E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC_noAPOE.RData\u0026quot;)\n\n# CFA model\n\nCFAofEFA_exAPOE \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED\nF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput_exAPOE \u0026lt;- usermodel(LDSCoutput_exAPOE, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA_exAPOE, \n                            CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput_exAPOE\n\n# Common factor\n\nCommonFactor_exAPOE\u0026lt;- commonfactor(covstruc = LDSCoutput_exAPOE, estimation=\u0026quot;DWLS\u0026quot;)\nCommonFactor_exAPOE\n\n# Hierarchal model\n\nhierarchal_exAPOE \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED\nF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS\nF4 =~ F1 + F2 + F3'\n\nhierarchal_exAPOE \u0026lt;- usermodel(LDSCoutput_exAPOE, estimation = \u0026quot;DWLS\u0026quot;, model = hierarchal_exAPOE, \n                              CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nhierarchal_exAPOE\n\n# Bifactor model\n\nbifactor_exPOE \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED\nF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\nF3 =~ NA*MDD + LON + HD + INS\nF4 =~ NA*AD + MDD + INS + LON + LSA + HD + LED + LPA + SMK + ALC + BMI + DEP + T2DM\nF4 ~~ 0*F1\nF4 ~~ 0*F2\nF4 ~~ 0*F3\nF1 ~~ 0*F2\nF1 ~~ 0*F3\nF2 ~~ 0*F3\nSMK ~~  a*SMK\na \u0026gt; .001'\n\nbifactor_exAPOE \u0026lt;- usermodel(LDSCoutput_exAPOE, estimation = \u0026quot;DWLS\u0026quot;, model = bifactor_exAPOE, \n                           CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\n\nbifactor_exAPOE\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# read in snplist of APOE region +/- 100kb each side from UCSC browser for genome build 37 to match GWAS sumstats - 249 SNPs\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAPOE_snps.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003echeck.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Read in munged sumstats files and remove any SNPs that match the rsIDs in the APOE snplist file\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eAD\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAD.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eAD_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eAD\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eAD_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAD_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eMDD\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMDD.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eMDD_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eMDD\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eMDD_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMDD_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eINS\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003einsomnia.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eINS_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eINS\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eINS_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eINS_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eLON\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eloneliness_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eLON_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eLON\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eLON_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLON_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eLSA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eno_social_activity_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eLSA_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eLSA\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eLSA_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLSA_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eHD\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehearing_difficulty_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eHD_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eHD\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eHD_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHD_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eLED\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elow_education_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eLED_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eLED\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eLED_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLED_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eLPA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ephysical_inactivity_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eLPA_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eLPA\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eLPA_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLPA_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eSMK\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecurrent_smoker_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eSMK_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eSMK\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eSMK_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSMK_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eALC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealcohol_intake_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eALC_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eALC\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eALC_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eALC_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eBMI\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBMI_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eBMI_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eBMI\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eBMI_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBMI_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eDEP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edeprivation_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eDEP_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eDEP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eDEP_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDEP_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eSBP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esystolic_BP_UKB.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eSBP_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eSBP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eSBP_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSBP_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eT2DM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDiabetes.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eT2DM_no_apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e anti_join(\u003cspan class=\"pl-smi\"\u003eT2DM\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eapoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eV4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nwrite.table(\u003cspan class=\"pl-smi\"\u003eT2DM_no_apoe\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eT2DM_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Multivariable LD score regression of data without APOE\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003etraits.apoe\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAD_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMDD_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eINS_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLON_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                 \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLSA_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHD_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLED_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLPA_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                 \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSMK_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eALC_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBMI_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDEP_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                 \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSBP_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eT2DM_no_apoe.sumstats.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eLDSCoutput_exAPOE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ldsc(\u003cspan class=\"pl-smi\"\u003etraits.apoe\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esample.prev\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epopulation.prev\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eld\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ewld\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etrait.names\u003c/span\u003e, \n                   \u003cspan class=\"pl-v\"\u003eldsc.log\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eE:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC_noAPOE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nsave(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_exAPOE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eE:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC_noAPOE.RData\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e CFA model\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA_exAPOE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_exAPOE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_exAPOE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA_exAPOE\u003c/span\u003e, \n                            \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_exAPOE\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Common factor\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCommonFactor_exAPOE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e commonfactor(\u003cspan class=\"pl-v\"\u003ecovstruc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eLDSCoutput_exAPOE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCommonFactor_exAPOE\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Hierarchal model\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003ehierarchal_exAPOE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF4 =~ F1 + F2 + F3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003ehierarchal_exAPOE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_exAPOE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ehierarchal_exAPOE\u003c/span\u003e, \n                              \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ehierarchal_exAPOE\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Bifactor model\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003ebifactor_exPOE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF4 =~ NA*AD + MDD + INS + LON + LSA + HD + LED + LPA + SMK + ALC + BMI + DEP + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF4 ~~ 0*F1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF4 ~~ 0*F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF4 ~~ 0*F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ 0*F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ 0*F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ 0*F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eSMK ~~  a*SMK\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ea \u0026gt; .001\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003ebifactor_exAPOE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_exAPOE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebifactor_exAPOE\u003c/span\u003e, \n                           \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003ebifactor_exAPOE\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSensitivity analyses\u003c/h1\u003e\u003ca id=\"user-content-sensitivity-analyses\" class=\"anchor\" aria-label=\"Permalink: Sensitivity analyses\" href=\"#sensitivity-analyses\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBecause the factor loadings for Alzheimers disease varied substantially\nbetween the odd autosome EFA, even autosome CFA and genome-wide CFA in\ncomparison to the risk factor traits we ran a number of follow-up\nsensitivity analyses to explore why this might be.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eLD score regression in different chromosomal groupings\u003c/h3\u003e\u003ca id=\"user-content-ld-score-regression-in-different-chromosomal-groupings\" class=\"anchor\" aria-label=\"Permalink: LD score regression in different chromosomal groupings\" href=\"#ld-score-regression-in-different-chromosomal-groupings\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe used the three outputs generated in the main study using the \u003ccode\u003eldsc\u003c/code\u003e\nfunction in the \u003cem\u003eGenomicSEM\u003c/em\u003e R package to compare the SNP-based\nheritability and genetic correlation estimates of the included traits\nwhen different chromosomal groupings were measured (odd vs.even\nvs.all). These estimates can be found in the .log files that are\nproduced during previous \u003ccode\u003eldsc\u003c/code\u003e steps.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe then plotted the genetic correlations using \u003ccode\u003ecorrplot\u003c/code\u003e so that we\ncould more easily visualise any differences between Alzheimers disease\nand its risk factors.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNote:\u003c/strong\u003e Because the output of the \u003ccode\u003eldsc\u003c/code\u003e function is already an R data\nmatrix we dont need a separate .csv file like the earlier correlation\nplot since we can just use the matrix directly as the input. But we need\nto convert it to a correlation matrix using the \u003ccode\u003ecov2cor\u003c/code\u003e function.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Smooth and convert the odd and even LDSC matrices to correlation matrices\n\nSsmooth_odd\u0026lt;-as.matrix((nearPD(LDSCoutput_odd$S, corr = FALSE))$mat)\ncorr_matrix_odd \u0026lt;- cov2cor(Ssmooth_odd)\n\nSsmooth_even\u0026lt;-as.matrix((nearPD(LDSCoutput_even$S, corr = FALSE))$mat)\ncorr_matrix_even \u0026lt;- cov2cor(Ssmooth_even)\n\n### Make combined LDSC matrix of odd and even chromosomes \n\npdf(file = \u0026quot;LDSC_matrix_odd_even.pdf\u0026quot;, height = 5, width = 5.2)\ncorrplot(corr_matrix_odd, method = \u0026quot;color\u0026quot;, col = col(200),\n         addCoef.col = \u0026quot;black\u0026quot;, number.cex = .6, number.digits = 2, \n         type = \u0026quot;lower\u0026quot;,  order = \u0026quot;original\u0026quot;,\n         tl.pos = \u0026quot;lt\u0026quot;, tl.col = \u0026quot;black\u0026quot;, tl.srt = 90, tl.cex = .7, \n         cl.cex = .7, cl.ratio = 0.1, \n         diag = TRUE)\ncorrplot(corr_matrix_even, add = TRUE, method = \u0026quot;color\u0026quot;, col = col(200),\n         addCoef.col = \u0026quot;black\u0026quot;, number.cex = .6, number.digits = 2, \n         type = \u0026quot;upper\u0026quot;,  order = \u0026quot;original\u0026quot;,\n         diag = FALSE, tl.pos = \u0026quot;n\u0026quot;, cl.pos = \u0026quot;n\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Smooth and convert the odd and even LDSC matrices to correlation matrices\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eSsmooth_odd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.matrix((nearPD(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_odd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eS\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecorr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e))\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emat\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ecorr_matrix_odd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e cov2cor(\u003cspan class=\"pl-smi\"\u003eSsmooth_odd\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eSsmooth_even\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.matrix((nearPD(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_even\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eS\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecorr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e))\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emat\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ecorr_matrix_even\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e cov2cor(\u003cspan class=\"pl-smi\"\u003eSsmooth_even\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## Make combined LDSC matrix of odd and even chromosomes \u003c/span\u003e\n\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLDSC_matrix_odd_even.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5.2\u003c/span\u003e)\ncorrplot(\u003cspan class=\"pl-smi\"\u003ecorr_matrix_odd\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecolor\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e col(\u003cspan class=\"pl-c1\"\u003e200\u003c/span\u003e),\n         \u003cspan class=\"pl-v\"\u003eaddCoef.col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enumber.cex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.6\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enumber.digits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \n         \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elower\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoriginal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n         \u003cspan class=\"pl-v\"\u003etl.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etl.col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etl.srt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e90\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etl.cex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.7\u003c/span\u003e, \n         \u003cspan class=\"pl-v\"\u003ecl.cex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.7\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecl.ratio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e, \n         \u003cspan class=\"pl-v\"\u003ediag\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ncorrplot(\u003cspan class=\"pl-smi\"\u003ecorr_matrix_even\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecolor\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e col(\u003cspan class=\"pl-c1\"\u003e200\u003c/span\u003e),\n         \u003cspan class=\"pl-v\"\u003eaddCoef.col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enumber.cex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.6\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enumber.digits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \n         \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eupper\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoriginal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n         \u003cspan class=\"pl-v\"\u003ediag\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etl.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003en\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecl.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003en\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOutput of the genetic correlation matrix with the odd autosome results\non the lower triangle and the even autosome results on the upper\ntriangle:\u003c/p\u003e\n\u003cp align=\"center\" dir=\"auto\"\u003e\n  \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/odd_even_ldsc_matrix.jpg\"\u003e\u003cimg width=\"500\" src=\"/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/odd_even_ldsc_matrix.jpg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eEFA comparison across chromosomal groupings\u003c/h2\u003e\u003ca id=\"user-content-efa-comparison-across-chromosomal-groupings\" class=\"anchor\" aria-label=\"Permalink: EFA comparison across chromosomal groupings\" href=\"#efa-comparison-across-chromosomal-groupings\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSince the correlation coefficients were substantially different for AD\nbetween the odd and even chromosomes, we also used EFA to test the\nloadings for a 3-factor model based on the even autosomal data and all\nautosomes to compare the factor loadings results with our initial\nresults based on the odd autosomes.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# 3-factor model even autosomes\n\nEFA_even\u0026lt;-factanal(covmat = Ssmooth_even, \n                   factors = 3, rotation = \u0026quot;promax\u0026quot;)\n\nprint(EFA_even, sort=TRUE)\nprint(EFA_even,digits=3,cutoff=.20,sort=TRUE)\n\n# 3-factor model all autosomes\n\nSsmooth_all\u0026lt;-as.matrix((nearPD(LDSCoutput_all$S, corr = FALSE))$mat)\nEFA_all\u0026lt;-factanal(covmat = Ssmooth_all, \n                   factors = 3, rotation = \u0026quot;promax\u0026quot;)\n\nprint(EFA_all, sort=TRUE)\nprint(EFA_all,digits=3,cutoff=.20,sort=TRUE)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e 3-factor model even autosomes\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eEFA_even\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003efactanal(\u003cspan class=\"pl-v\"\u003ecovmat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSsmooth_even\u003c/span\u003e, \n                   \u003cspan class=\"pl-v\"\u003efactors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erotation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epromax\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nprint(\u003cspan class=\"pl-smi\"\u003eEFA_even\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esort\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\nprint(\u003cspan class=\"pl-smi\"\u003eEFA_even\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edigits\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecutoff\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.20\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e 3-factor model all autosomes\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eSsmooth_all\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.matrix((nearPD(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eS\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecorr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e))\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emat\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eEFA_all\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003efactanal(\u003cspan class=\"pl-v\"\u003ecovmat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSsmooth_all\u003c/span\u003e, \n                   \u003cspan class=\"pl-v\"\u003efactors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erotation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epromax\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nprint(\u003cspan class=\"pl-smi\"\u003eEFA_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esort\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\nprint(\u003cspan class=\"pl-smi\"\u003eEFA_all\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edigits\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecutoff\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.20\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esort\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCFA comparison across chromosomal groupings\u003c/h2\u003e\u003ca id=\"user-content-cfa-comparison-across-chromosomal-groupings\" class=\"anchor\" aria-label=\"Permalink: CFA comparison across chromosomal groupings\" href=\"#cfa-comparison-across-chromosomal-groupings\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe also ran post-hoc sensitivity CFA analyses using the outputs from the\nsensitivity EFA to compare the model fit and loadings of the best\nfitting models based on the even autosome and all autosome EFA to see\nhow they compared to our main analysis.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# CFA models of even autosome EFA run in the odd autosomes\n\n# Model 1: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and cross-loadings (DWLS method),\n# negative loadings excluded; factor correlations present\n\nCFAofEFA_posthoc1 \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP\nF2 =~ NA*LSA + LED + LPA + ALC + DEP + BMI + T2DM + SBP\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1\nSMK ~~  a*SMK\na \u0026gt; .001'\n\nCFAoutput_posthoc1 \u0026lt;- usermodel(LDSCoutput_odd, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA_posthoc1, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput_posthoc1\n\n# Model 2: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and no cross-loadings (DWLS method),\n# negative loadings excluded; factor correlations present\n\nCFAofEFA_posthoc2 \u0026lt;- 'F1 =~ NA*SMK + DEP\nF2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1\nSMK ~~  a*SMK\na \u0026gt; .001'\n\nCFAoutput_posthoc2 \u0026lt;- usermodel(LDSCoutput_odd, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA_posthoc2, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput_posthoc2\n\n\n# Model 3: CFA of 3-factor model: including traits with \u0026gt;.25 positive loadings and cross-loadings (DWLS method),\n# no negative loadings; factor correlations present (best fitting parameters for odd autosomes) - had to remove cross loading for deprivation status for it to converge\n\nCFAofEFA_posthoc3 \u0026lt;- 'F1 =~ NA*SMK + DEP\nF2 =~ NA*LPA + LSA + LED + ALC + BMI + SBP + T2DM\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput_posthoc3 \u0026lt;- usermodel(LDSCoutput_odd, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA_posthoc3, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput_posthoc3\n\n## CFA conducted in all autosomes but based off the parameters from the even EFA (same as the one tested in odd autosomes) - sensitivity analysis\n\n# Model 4: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and cross-loadings (DWLS method),\n# negative loadings excluded; factor correlations present\n\nCFAofEFA_posthoc4 \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP\nF2 =~ NA*LSA + LED + LPA + ALC + DEP + BMI + T2DM + SBP\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1\nSMK ~~  a*SMK\na \u0026gt; .001'\n\nCFAoutput_posthoc4 \u0026lt;- usermodel(LDSCoutput_all, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA_posthoc4, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput_posthoc4\n\n# Model 5: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and no cross-loadings (DWLS method),\n# negative loadings excluded; factor correlations present\n\nCFAofEFA_posthoc5 \u0026lt;- 'F1 =~ NA*SMK + DEP\nF2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1\nSMK ~~  a*SMK\na \u0026gt; .001'\n\nCFAoutput_posthoc5 \u0026lt;- usermodel(LDSCoutput_all, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA_posthoc5, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput_posthoc5\n\n\n# Model 6: CFA of 3-factor model: including traits with \u0026gt;.25 positive loadings and cross-loadings (DWLS method),\n# no negative loadings; factor correlations present (best fitting parameters for odd autosomes)\n\nCFAofEFA_posthoc6 \u0026lt;- 'F1 =~ NA*SMK + DEP\nF2 =~ NA*LPA + LSA + LED + ALC + BMI + SBP + T2DM + DEP\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput_posthoc6 \u0026lt;- usermodel(LDSCoutput_all, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA_posthoc6, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput_posthoc6\n\n# CFA models of all autosome EFA run in all autosomes\n\n# Model 7: CFA of 3-factor model: including traits with \u0026gt;.25 positive loadings and cross-loadings (DWLS method),\n# no negative loadings; factor correlations present (KEPT INSOMNIA IN AS .24)\n\nCFAofEFA_posthoc7 \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED\nF2 =~ NA*LPA + LSA + LED + ALC + BMI + T2DM + SBP\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput_posthoc7 \u0026lt;- usermodel(LDSCoutput_all, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA_posthoc7, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput_posthoc7\n\n# Model 8: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and cross-loadings (DWLS method),\n# negative loadings excluded; factor correlations present\n\nCFAofEFA_posthoc8 \u0026lt;- 'F1 =~ NA*LPA + SMK + DEP + LED\nF2 =~ NA*LSA + LED + LPA + ALC + INS + BMI + T2DM + SBP\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput_posthoc8 \u0026lt;- usermodel(LDSCoutput_all, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA_posthoc8, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput_posthoc8\n\n# Model 9: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and no cross-loadings (DWLS method),\n# negative loadings excluded; factor correlations present\n\nCFAofEFA_posthoc9 \u0026lt;- 'F1 =~ NA*SMK + DEP\nF2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP\nF3 =~ NA*MDD + LON + HD + INS\nF1 ~~ F2\nF2 ~~ F3\nF3 ~~ F1'\n\nCFAoutput_posthoc9 \u0026lt;- usermodel(LDSCoutput_all, estimation = \u0026quot;DWLS\u0026quot;, model = CFAofEFA_posthoc9, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)\nCFAoutput_posthoc9\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e CFA models of even autosome EFA run in the odd autosomes\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 1: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e negative loadings excluded; factor correlations present\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LSA + LED + LPA + ALC + DEP + BMI + T2DM + SBP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eSMK ~~  a*SMK\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ea \u0026gt; .001\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_odd\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc1\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 2: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and no cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e negative loadings excluded; factor correlations present\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*SMK + DEP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eSMK ~~  a*SMK\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ea \u0026gt; .001\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_odd\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc2\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 3: CFA of 3-factor model: including traits with \u0026gt;.25 positive loadings and cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e no negative loadings; factor correlations present (best fitting parameters for odd autosomes) - had to remove cross loading for deprivation status for it to converge\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*SMK + DEP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + BMI + SBP + T2DM\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_odd\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc3\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# CFA conducted in all autosomes but based off the parameters from the even EFA (same as the one tested in odd autosomes) - sensitivity analysis\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 4: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e negative loadings excluded; factor correlations present\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc4\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LSA + LED + LPA + ALC + DEP + BMI + T2DM + SBP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eSMK ~~  a*SMK\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ea \u0026gt; .001\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc4\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc4\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 5: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and no cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e negative loadings excluded; factor correlations present\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc5\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*SMK + DEP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eSMK ~~  a*SMK\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ea \u0026gt; .001\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc5\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc5\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 6: CFA of 3-factor model: including traits with \u0026gt;.25 positive loadings and cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e no negative loadings; factor correlations present (best fitting parameters for odd autosomes)\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc6\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*SMK + DEP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + BMI + SBP + T2DM + DEP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc6\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc6\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc6\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e CFA models of all autosome EFA run in all autosomes\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 7: CFA of 3-factor model: including traits with \u0026gt;.25 positive loadings and cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e no negative loadings; factor correlations present (KEPT INSOMNIA IN AS .24)\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc7\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LPA + LSA + LED + ALC + BMI + T2DM + SBP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc7\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc7\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc7\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 8: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e negative loadings excluded; factor correlations present\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc8\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*LPA + SMK + DEP + LED\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LSA + LED + LPA + ALC + INS + BMI + T2DM + SBP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc8\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc8\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc8\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Model 9: CFA of 3-factor model: including traits with \u0026gt;.20 loadings and no cross-loadings (DWLS method),\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e negative loadings excluded; factor correlations present\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc9\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eF1 =~ NA*SMK + DEP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 =~ NA*MDD + LON + HD + INS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF1 ~~ F2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF2 ~~ F3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eF3 ~~ F1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc9\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e usermodel(\u003cspan class=\"pl-smi\"\u003eLDSCoutput_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eestimation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDWLS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCFAofEFA_posthoc9\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCFIcalc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estd.lv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eimp_cov\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCFAoutput_posthoc9\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFinal remarks\u003c/h1\u003e\u003ca id=\"user-content-final-remarks\" class=\"anchor\" aria-label=\"Permalink: Final remarks\" href=\"#final-remarks\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe hope that you find this Markdown useful. If you use this code in your\nown analyses please cite this Github page. If you run into any issues or\nqueries related to this work please dont hesitate to contact Isabelle\nFoote (\u003ca href=\"mailto:isabelle.foote@colorado.edu\"\u003eisabelle.foote@colorado.edu\u003c/a\u003e).\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"Code walkthrough for genomic factor analysis of modifiable risk for Alzheimers disease","anchor":"code-walkthrough-for-genomic-factor-analysis-of-modifiable-risk-for-alzheimers-disease","htmlText":"Code walkthrough for genomic factor analysis of modifiable risk for Alzheimers disease"},{"level":1,"text":"Overview","anchor":"overview","htmlText":"Overview"},{"level":1,"text":"GWAS data identification and formatting","anchor":"gwas-data-identification-and-formatting","htmlText":"GWAS data identification and formatting"},{"level":2,"text":"Criteria for inclusion","anchor":"criteria-for-inclusion","htmlText":"Criteria for inclusion"},{"level":3,"text":"Links to the GWAS data used in our current study","anchor":"links-to-the-gwas-data-used-in-our-current-study","htmlText":"Links to the GWAS data used in our current study"},{"level":2,"text":"Example code for formatting UK Biobank GWAS data","anchor":"example-code-for-formatting-uk-biobank-gwas-data","htmlText":"Example code for formatting UK Biobank GWAS data"},{"level":3,"text":"Set your working directory and load the required packages","anchor":"set-your-working-directory-and-load-the-required-packages","htmlText":"Set your working directory and load the required packages"},{"level":3,"text":"Make an edited version of the variant data to merge with the GWAS datasets","anchor":"make-an-edited-version-of-the-variant-data-to-merge-with-the-gwas-datasets","htmlText":"Make an edited version of the variant data to merge with the GWAS datasets"},{"level":3,"text":"Format UK Biobank GWAS data ready for munging","anchor":"format-uk-biobank-gwas-data-ready-for-munging","htmlText":"Format UK Biobank GWAS data ready for munging"},{"level":1,"text":"LD score regression","anchor":"ld-score-regression","htmlText":"LD score regression"},{"level":3,"text":"Install LDSC software","anchor":"install-ldsc-software","htmlText":"Install LDSC software"},{"level":2,"text":"Munge data","anchor":"munge-data","htmlText":"Munge data"},{"level":2,"text":"Calculating univariate SNP-based heritability","anchor":"calculating-univariate-snp-based-heritability","htmlText":"Calculating univariate SNP-based heritability"},{"level":2,"text":"Calculating pairwise genetic correlations","anchor":"calculating-pairwise-genetic-correlations","htmlText":"Calculating pairwise genetic correlations"},{"level":2,"text":"Plotting the data","anchor":"plotting-the-data","htmlText":"Plotting the data"},{"level":3,"text":"Genetic correlation matrix","anchor":"genetic-correlation-matrix","htmlText":"Genetic correlation matrix"},{"level":3,"text":"Undirected weighted graph","anchor":"undirected-weighted-graph","htmlText":"Undirected weighted graph"},{"level":1,"text":"Genomic SEM","anchor":"genomic-sem","htmlText":"Genomic SEM"},{"level":3,"text":"Install GenomicSEM","anchor":"install-genomicsem","htmlText":"Install GenomicSEM"},{"level":3,"text":"Set working directory and load the GenomicSEM package","anchor":"set-working-directory-and-load-the-genomicsem-package","htmlText":"Set working directory and load the GenomicSEM package"},{"level":2,"text":"Munge data","anchor":"munge-data-1","htmlText":"Munge data"},{"level":2,"text":"Exploratory factor analysis performed in the odd autosomes","anchor":"exploratory-factor-analysis-performed-in-the-odd-autosomes","htmlText":"Exploratory factor analysis performed in the odd autosomes"},{"level":3,"text":"Load the additional required packages","anchor":"load-the-additional-required-packages","htmlText":"Load the additional required packages"},{"level":3,"text":"Conduct multivariable LD score regression (odd autosomes)","anchor":"conduct-multivariable-ld-score-regression-odd-autosomes","htmlText":"Conduct multivariable LD score regression (odd autosomes)"},{"level":3,"text":"Exploratory factor analysis","anchor":"exploratory-factor-analysis","htmlText":"Exploratory factor analysis"},{"level":3,"text":"EFA loadings plot","anchor":"efa-loadings-plot","htmlText":"EFA loadings plot"},{"level":2,"text":"Confirmatory factor analysis performed in the even autosomes","anchor":"confirmatory-factor-analysis-performed-in-the-even-autosomes","htmlText":"Confirmatory factor analysis performed in the even autosomes"},{"level":3,"text":"Conduct multivariable LD score regression (even autosomes)","anchor":"conduct-multivariable-ld-score-regression-even-autosomes","htmlText":"Conduct multivariable LD score regression (even autosomes)"},{"level":3,"text":"Confirmatory factor analysis","anchor":"confirmatory-factor-analysis","htmlText":"Confirmatory factor analysis"},{"level":3,"text":"Plotting CFA models","anchor":"plotting-cfa-models","htmlText":"Plotting CFA models"},{"level":2,"text":"Follow-up genome-wide analyses","anchor":"follow-up-genome-wide-analyses","htmlText":"Follow-up genome-wide analyses"},{"level":3,"text":"Genome-wide multivariable LD score regression","anchor":"genome-wide-multivariable-ld-score-regression","htmlText":"Genome-wide multivariable LD score regression"},{"level":3,"text":"Genome-wide CFA","anchor":"genome-wide-cfa","htmlText":"Genome-wide CFA"},{"level":3,"text":"Additional SEM modelling","anchor":"additional-sem-modelling","htmlText":"Additional SEM modelling"},{"level":3,"text":"Common factor model","anchor":"common-factor-model","htmlText":"Common factor model"},{"level":3,"text":"Hierarchal model","anchor":"hierarchal-model","htmlText":"Hierarchal model"},{"level":3,"text":"Bifactor model","anchor":"bifactor-model","htmlText":"Bifactor model"},{"level":2,"text":"Models without the APOE region","anchor":"models-without-the-apoe-region","htmlText":"Models without the APOE region"},{"level":1,"text":"Sensitivity analyses","anchor":"sensitivity-analyses","htmlText":"Sensitivity analyses"},{"level":3,"text":"LD score regression in different chromosomal groupings","anchor":"ld-score-regression-in-different-chromosomal-groupings","htmlText":"LD score regression in different chromosomal groupings"},{"level":2,"text":"EFA comparison across chromosomal groupings","anchor":"efa-comparison-across-chromosomal-groupings","htmlText":"EFA comparison across chromosomal groupings"},{"level":2,"text":"CFA comparison across chromosomal groupings","anchor":"cfa-comparison-across-chromosomal-groupings","htmlText":"CFA comparison across chromosomal groupings"},{"level":1,"text":"Final remarks","anchor":"final-remarks","htmlText":"Final remarks"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2FIsyFoote%2FGenomic-Factor-Analysis-of-modifiable-AD-risk_NBA22"}}],"overviewFilesProcessingTime":131.273811}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           13 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="images, (Directory)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/tree/main/images" title="images">
                          images
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="images, (Directory)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/tree/main/images" title="images">
                          images
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LDSC.sh, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/LDSC.sh" title="LDSC.sh">
                          LDSC.sh
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LDSC.sh, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/LDSC.sh" title="LDSC.sh">
                          LDSC.sh
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LDSC_env.yml, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/LDSC_env.yml" title="LDSC_env.yml">
                          LDSC_env.yml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LDSC_env.yml, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/LDSC_env.yml" title="LDSC_env.yml">
                          LDSC_env.yml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="example_script_to_format_UKB_GWAS.R, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/example_script_to_format_UKB_GWAS.R" title="example_script_to_format_UKB_GWAS.R">
                          example_script_to_format_UKB_GWAS.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="example_script_to_format_UKB_GWAS.R, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/example_script_to_format_UKB_GWAS.R" title="example_script_to_format_UKB_GWAS.R">
                          example_script_to_format_UKB_GWAS.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-6">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="genomic_factor_analysis_and_sensitivity_analyses.R, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/genomic_factor_analysis_and_sensitivity_analyses.R" title="genomic_factor_analysis_and_sensitivity_analyses.R">
                          genomic_factor_analysis_and_sensitivity_analyses.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="genomic_factor_analysis_and_sensitivity_analyses.R, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/genomic_factor_analysis_and_sensitivity_analyses.R" title="genomic_factor_analysis_and_sensitivity_analyses.R">
                          genomic_factor_analysis_and_sensitivity_analyses.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-7">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="make_LD_corr_matrix_and_undirected_graph.R, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/make_LD_corr_matrix_and_undirected_graph.R" title="make_LD_corr_matrix_and_undirected_graph.R">
                          make_LD_corr_matrix_and_undirected_graph.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="make_LD_corr_matrix_and_undirected_graph.R, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/make_LD_corr_matrix_and_undirected_graph.R" title="make_LD_corr_matrix_and_undirected_graph.R">
                          make_LD_corr_matrix_and_undirected_graph.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-8">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="make_LD_correlation_matrix.R, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/make_LD_correlation_matrix.R" title="make_LD_correlation_matrix.R">
                          make_LD_correlation_matrix.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="make_LD_correlation_matrix.R, (File)" class="Link--primary" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/make_LD_correlation_matrix.R" title="make_LD_correlation_matrix.R">
                          make_LD_correlation_matrix.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Code walkthrough for genomic factor analysis of modifiable risk for Alzheimers disease
                     </h1>
                     <a aria-label="Permalink: Code walkthrough for genomic factor analysis of modifiable risk for Alzheimers disease" class="anchor" href="#code-walkthrough-for-genomic-factor-analysis-of-modifiable-risk-for-alzheimers-disease" id="user-content-code-walkthrough-for-genomic-factor-analysis-of-modifiable-risk-for-alzheimers-disease">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Isabelle Foote
2022-06-13
                    </p>
                    <ul dir="auto">
                     <li>
                      <a href="#overview">
                       Overview
                      </a>
                     </li>
                     <li>
                      <a href="#gwas-data-identification-and-formatting">
                       GWAS data identification and formatting
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#criteria-for-inclusion">
                         Criteria for inclusion
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#links-to-the-gwas-data-used-in-our-current-study">
                           Links to the GWAS data used in our current study
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#example-code-for-formatting-uk-biobank-gwas-data">
                         Example code for formatting UK Biobank GWAS data
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#set-your-working-directory-and-load-the-required-packages">
                           Set your working directory and load the required packages
                          </a>
                         </li>
                         <li>
                          <a href="#make-an-edited-version-of-the-variant-data-to-merge-with-the-gwas-datasets">
                           Make an edited version of the variant data to merge with the GWAS datasets
                          </a>
                         </li>
                         <li>
                          <a href="#format-uk-biobank-gwas-data-ready-for-munging">
                           Format UK Biobank GWAS data ready for munging
                          </a>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#ld-score-regression">
                       LD score regression
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#install-ldsc-software">
                         Install LDSC software
                        </a>
                       </li>
                      </ul>
                      <ul dir="auto">
                       <li>
                        <a href="#munge-data">
                         Munge data
                        </a>
                       </li>
                       <li>
                        <a href="#calculating-univariate-snp-based-heritability">
                         Calculating univariate SNP-based heritability
                        </a>
                       </li>
                       <li>
                        <a href="#calculating-pairwise-genetic-correlations">
                         Calculating pairwise genetic correlations
                        </a>
                       </li>
                       <li>
                        <a href="#plotting-the-data">
                         Plotting the data
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#genetic-correlation-matrix">
                           Genetic correlation matrix
                          </a>
                         </li>
                         <li>
                          <a href="#undirected-weighted-graph">
                           Undirected weighted graph
                          </a>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#genomic-sem">
                       Genomic SEM
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#install--genomicsem-">
                         Install
                         <em>
                          GenomicSEM
                         </em>
                        </a>
                       </li>
                       <li>
                        <a href="#set-working-directory-and-load-the--genomicsem--package">
                         Set working directory and load the
                         <em>
                          GenomicSEM
                         </em>
                         package
                        </a>
                       </li>
                      </ul>
                      <ul dir="auto">
                       <li>
                        <a href="#munge-data-1">
                         Munge data
                        </a>
                       </li>
                       <li>
                        <a href="#exploratory-factor-analysis-performed-in-the-odd-autosomes">
                         Exploratory factor analysis performed in the odd autosomes
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#load-the-additional-required-packages">
                           Load the additional required packages
                          </a>
                         </li>
                         <li>
                          <a href="#conduct-multivariable-ld-score-regression--odd-autosomes-">
                           Conduct multivariable LD score regression (odd autosomes)
                          </a>
                         </li>
                         <li>
                          <a href="#exploratory-factor-analysis">
                           Exploratory factor analysis
                          </a>
                         </li>
                         <li>
                          <a href="#efa-loadings-plot">
                           EFA loadings plot
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#confirmatory-factor-analysis-performed-in-the-even-autosomes">
                         Confirmatory factor analysis performed in the even autosomes
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#conduct-multivariable-ld-score-regression--even-autosomes-">
                           Conduct multivariable LD score regression (even autosomes)
                          </a>
                         </li>
                         <li>
                          <a href="#confirmatory-factor-analysis">
                           Confirmatory factor analysis
                          </a>
                         </li>
                         <li>
                          <a href="#plotting-cfa-models">
                           Plotting CFA models
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#follow-up-genome-wide-analyses">
                         Follow-up genome-wide analyses
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#genome-wide-multivariable-ld-score-regression">
                           Genome-wide multivariable LD score regression
                          </a>
                         </li>
                         <li>
                          <a href="#genome-wide-cfa">
                           Genome-wide CFA
                          </a>
                         </li>
                         <li>
                          <a href="#additional-sem-modelling">
                           Additional SEM modelling
                          </a>
                         </li>
                         <li>
                          <a href="#common-factor-model">
                           Common factor model
                          </a>
                         </li>
                         <li>
                          <a href="#hierarchal-model">
                           Hierarchal model
                          </a>
                         </li>
                         <li>
                          <a href="#bifactor-model">
                           Bifactor model
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#models-without-the-apoe-region">
                         Models without the APOE region
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#sensitivity-analyses">
                       Sensitivity analyses
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#ld-score-regression-in-different-chromosomal-groupings">
                         LD score regression in different chromosomal groupings
                        </a>
                       </li>
                      </ul>
                      <ul dir="auto">
                       <li>
                        <a href="#efa-comparison-across-chromosomal-groupings">
                         EFA comparison across chromosomal groupings
                        </a>
                       </li>
                       <li>
                        <a href="#cfa-comparison-across-chromosomal-groupings">
                         CFA comparison across chromosomal groupings
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#final-remarks">
                       Final remarks
                      </a>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Overview
                     </h1>
                     <a aria-label="Permalink: Overview" class="anchor" href="#overview" id="user-content-overview">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The following document outlines the code that was used to carry out the
work described in the study titled
                     <strong>
                      The shared genetic architecture of
modifiable risk for Alzheimers disease: a genomic structural equation
modelling study
                     </strong>
                     by Isabelle Foote, Ben Jacobs, Georgina Mathlin,
Cameron Watson, Phazha Bothongo, Sheena Waters, Ruth Dobson, Alastair
Noyce, Kam Bhui, Ania Korszun and Charles Marshall.
                    </p>
                    <p dir="auto">
                     The published version of the paper can be found
                     <a href="https://www.sciencedirect.com/science/article/pii/S0197458022001312" rel="nofollow">
                      here
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     The preprint for the study can be found on
                     <a href="https://www.medrxiv.org/content/10.1101/2021.02.23.21252211v1" rel="nofollow">
                      MedRxiv
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     This markdown report and code was written by Isabelle Foote.
                    </p>
                    <p dir="auto">
                     The current document provides step-by-step instructions to replicate our
work and utilises open-source software and code that was created by the
original developers of the methods used.
                    </p>
                    <p dir="auto">
                     Further details relating to software/code can be found on the
corresponding original GitHub pages:
                    </p>
                    <ul dir="auto">
                     <li>
                      <strong>
                       <a href="https://github.com/bulik/ldsc">
                        LD score regression
                       </a>
                      </strong>
                      (Brendan
Bulik-Sullivan and colleagues)
                     </li>
                     <li>
                      <strong>
                       <a href="https://github.com/GenomicSEM/GenomicSEM">
                        Genomic SEM
                       </a>
                      </strong>
                      (Andrew
Grotzinger and colleagues)
                     </li>
                    </ul>
                    <p dir="auto">
                     Further details on the methods used can be found in the following
publications, upon which the current study protocol was based:
                    </p>
                    <p dir="auto">
                     <strong>
                      LD score regression
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      Bulik-Sullivan B., et al.(2015)
                      <em>
                       LD Score regression distinguishes
confounding from polygenicity in genome-wide association studies
                      </em>
                      .
Nature Genetics. 47: p.291-295
                      <a href="https://doi.org/10.1038/ng.3211" rel="nofollow">
                       https://doi.org/10.1038/ng.3211
                      </a>
                      .
                     </li>
                     <li>
                      Bulik-Sullivan B., et al.(2015)
                      <em>
                       An atlas of genetic correlations
across human diseases and traits
                      </em>
                      . Nature Genetics. 47(11):
p.1236-1241
                      <a href="https://doi.org/10.1038/ng.3406" rel="nofollow">
                       https://doi.org/10.1038/ng.3406
                      </a>
                      .
                     </li>
                     <li>
                      Zheng J., et al.(2016)
                      <em>
                       LD Hub: a centralized database and web
interface to perform LD score regression that maximizes the
potential of summary level GWAS data for SNP heritability and
genetic correlation analysis
                      </em>
                      . Bioinformatics. 33(2): p.272-279
                      <a href="https://doi.org/10.1093/bioinformatics/btw613" rel="nofollow">
                       https://doi.org/10.1093/bioinformatics/btw613
                      </a>
                      .
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      Genomic SEM
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      Grotzinger AD., et al.(2019)
                      <em>
                       Genomic structural equation modelling
provides insights into the multivariate genetic architecture of
complex traits
                      </em>
                      . Nat Hum Behav. 3(5): p.513-525
                      <a href="https://doi.org/10.1038/s41562-019-0566-x" rel="nofollow">
                       https://doi.org/10.1038/s41562-019-0566-x
                      </a>
                      .
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      GWAS data identification and formatting
                     </h1>
                     <a aria-label="Permalink: GWAS data identification and formatting" class="anchor" href="#gwas-data-identification-and-formatting" id="user-content-gwas-data-identification-and-formatting">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The LD score regression and genomic SEM methods we use in this study are
based off GWAS summary statistics data, so the first step is to identify
the appropriate data for your analysis. For a detailed rationale
relating to the choice of traits we included within our study please
refer to our
                     <a href="https://www.sciencedirect.com/science/article/pii/S0197458022001312" rel="nofollow">
                      paper
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     In short, we included GWAS data for Alzheimers disease and its major
potentially modifiable risk factors (less education, hearing loss,
hypertension, high alcohol intake, obesity, smoking, depression, social
isolation, physical inactivity, type 2 diabetes, sleep disturbance and
socioeconomic deprivation).
                    </p>
                    <p dir="auto">
                     Some useful resources to find datasets are:
                    </p>
                    <ul dir="auto">
                     <li>
                      <a href="https://www.ebi.ac.uk/gwas/" rel="nofollow">
                       GWAS Catalog
                      </a>
                     </li>
                     <li>
                      <a href="https://atlas.ctglab.nl/" rel="nofollow">
                       GWAS Atlas
                      </a>
                     </li>
                     <li>
                      <a href="http://www.nealelab.is/uk-biobank" rel="nofollow">
                       Neale lab
                      </a>
                      (for GWAS of UK
Biobank traits)
                     </li>
                     <li>
                      The
                      <a href="https://biobank.ndph.ox.ac.uk/showcase/" rel="nofollow">
                       UK Biobank Data
Showcase
                      </a>
                      is an easy way to
look up what variables were measured within the UK Biobank study
                     </li>
                     <li>
                      Relevant consortia for your topic of interest e.g.the
                      <a href="https://www.med.unc.edu/pgc/download-results/" rel="nofollow">
                       Psychiatric
Genomics Consortium
                      </a>
                      for GWASs of psychiatric disorders
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Criteria for inclusion
                     </h2>
                     <a aria-label="Permalink: Criteria for inclusion" class="anchor" href="#criteria-for-inclusion" id="user-content-criteria-for-inclusion">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     In order for your analysis to be reliable you need to ensure that the
GWAS data you include is appropriate and meets the advised criteria
explained in the original methods papers cited above and on the relevant
wikis.
                    </p>
                    <p dir="auto">
                     For the current study we ensured that the data met the following
criteria:
                    </p>
                    <ul dir="auto">
                     <li>
                      The samples were of unrelated individuals of European ancestry
                     </li>
                     <li>
                      Each dataset had a sample size &gt;5000 (ideally over 10,000)
                     </li>
                     <li>
                      The summary statistics had not been adjusted for heritable
covariates during association testing
                     </li>
                     <li>
                      The association test had not used linear mixed modelling (LMM)
methods
                     </li>
                     <li>
                      Each trait had a SNP-based heritability Z score &gt;4
                     </li>
                    </ul>
                    <p dir="auto">
                     Often the SNP-based heritability is reported in the orginal GWAS paper
so to get an idea about whether your trait is appropriate this can be a
good starting point.
                     <a href="http://ldsc.broadinstitute.org/" rel="nofollow">
                      LD Hub
                     </a>
                     also
contains heritability and genetic correlation results for hundreds of
traits. For the Neale lab UK Biobank GWASs you can check details of
their heritability in their
                     <a href="https://nealelab.github.io/UKBB_ldsc/index.html" rel="nofollow">
                      heritability
browser
                     </a>
                     and pairwise
genetic correlations in their
                     <a href="https://ukbb-rg.hail.is/" rel="nofollow">
                      correlation
browser
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     This is useful as an initial checking step, however we recommend
performing your own LD score regression as an initial part of your own
analysis pipeline to ensure that the results you get are in line with
previously run results and to facilitate a deeper understanding of your
own study results.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Links to the GWAS data used in our current study
                     </h3>
                     <a aria-label="Permalink: Links to the GWAS data used in our current study" class="anchor" href="#links-to-the-gwas-data-used-in-our-current-study" id="user-content-links-to-the-gwas-data-used-in-our-current-study">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The GWAS data we used in our current study is publicly available and
available to download via the following links:
                    </p>
                    <ul dir="auto">
                     <li>
                      <strong>
                       Alzheimers disease
                      </strong>
                      <ul dir="auto">
                       <li>
                        Paper: Kunkle BW., et al.(2019)
                        <em>
                         Genetic meta-analysis of
diagnosed Alzheimers disease identifies new risk loci and
implicates A, tau, immunity and lipid processing
                        </em>
                        . Nat
Genetics. 51: p.414-430
                        <a href="https://doi.org/10.1038/s41588-019-0358-2" rel="nofollow">
                         https://doi.org/10.1038/s41588-019-0358-2
                        </a>
                        .
                       </li>
                       <li>
                        Data:
                        <a href="https://www.niagads.org/datasets/ng00075" rel="nofollow">
                         IGAP
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <strong>
                       Major depressive disorder
                      </strong>
                      <ul dir="auto">
                       <li>
                        Paper: Wray NR., et al.(2018)
                        <em>
                         Genome-wide association analyses
identify 44 risk variants and refine the genetic architecture of
major depression
                        </em>
                        . Nat Genetics. 50(5): p.668-681
                        <a href="https://doi.org/10.1038/s41588-018-0090-3" rel="nofollow">
                         https://doi.org/10.1038/s41588-018-0090-3
                        </a>
                        .
                       </li>
                       <li>
                        Data:
                        <a href="https://www.med.unc.edu/pgc/download-results/mdd/" rel="nofollow">
                         PGC
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <strong>
                       Insomnia
                      </strong>
                      <ul dir="auto">
                       <li>
                        Paper: Jansen PR., et al.(2019)
                        <em>
                         Genome-wide analysis of
insomnia in 1,331,010 individuals identifies new risk loci and
functional pathways
                        </em>
                        . Nat Genetics. 51(3): p.394-403
                        <a href="https://doi.org/10.1038/s41588-018-0333-3" rel="nofollow">
                         https://doi.org/10.1038/s41588-018-0333-3
                        </a>
                        .
                       </li>
                       <li>
                        Data:
                        <a href="https://ctg.cncr.nl/software/summary_statistics" rel="nofollow">
                         CTG Lab
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <strong>
                       Type 2 diabetes
                      </strong>
                      <ul dir="auto">
                       <li>
                        Paper: Scott RA., et al.(2017)
                        <em>
                         An Expanded Genome-Wide
Association Study of Type 2 Diabetes in Europeans
                        </em>
                        . Diabetes.
66(11): p.2888
                        <a href="https://doi.org/10.2337/db16-1253" rel="nofollow">
                         https://doi.org/10.2337/db16-1253
                        </a>
                        .
                       </li>
                       <li>
                        Data:
                        <a href="http://diagram-consortium.org/downloads.html" rel="nofollow">
                         DIAGRAM
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <strong>
                       UK Biobank traits
                      </strong>
                      <ul dir="auto">
                       <li>
                        Traits: Loneliness, less social activity, hearing difficulty,
less education, physical inactivity, smoking, alcohol intake,
BMI, deprivation status, systolic blood pressure
                       </li>
                       <li>
                        Website:
                        <a href="http://www.nealelab.is/uk-biobank" rel="nofollow">
                         http://www.nealelab.is/uk-biobank
                        </a>
                       </li>
                       <li>
                        Data:
                        <a href="https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit?ts=5b5f17db#gid=178908679" rel="nofollow">
                         Neale
lab
                        </a>
                       </li>
                      </ul>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Example code for formatting UK Biobank GWAS data
                     </h2>
                     <a aria-label="Permalink: Example code for formatting UK Biobank GWAS data" class="anchor" href="#example-code-for-formatting-uk-biobank-gwas-data" id="user-content-example-code-for-formatting-uk-biobank-gwas-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     There are certain columns and information that are required in the GWAS
datasets so that the quality control filtering step (known as munging)
can be completed before conducting LD score regression and genomic SEM.
These requirements are outlined in detail below in the
                     <em>
                      data munging
                     </em>
                     sections.
                    </p>
                    <p dir="auto">
                     The LDSC and GenomicSEM
                     <code>
                      munge
                     </code>
                     functions can recognise a range of
different column names so often you can just input the downloaded
dataset directly into this function and it will run.
                    </p>
                    <p dir="auto">
                     However, there may be occasions where the GWAS data cannot be read by
the
                     <code>
                      munge
                     </code>
                     function because the software cannot recognise a certain
column name or if there is data missing that it needs. One of the main
examples of this happening is if you use the GWAS summary statistics of
UK Biobank traits that were conducted by the Neale lab. Therefore, we
provide a detailed example of how to appropriately format these datasets
for use with the LDSC and GenomicSEM software. This code was run in R
Studio (Version 1.2.5033) using R version 3.6.3 on a personal desktop.
                    </p>
                    <p dir="auto">
                     We downloaded our UK Biobank GWAS traits from the
                     <a href="https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit?ts=5b5f17db#gid=227859291" rel="nofollow">
                      Neale lab
manifest
                     </a>
                     of their round 2 European GWAS data. This is best done using the
                     <code>
                      wget
                     </code>
                     command in the terminal.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Set your working directory and load the required packages
                     </h3>
                     <a aria-label="Permalink: Set your working directory and load the required packages" class="anchor" href="#set-your-working-directory-and-load-the-required-packages" id="user-content-set-your-working-directory-and-load-the-required-packages">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='setwd("/path/to/directory")
library(dplyr)
library(data.table)' dir="auto">
                     <pre>setwd(<span class="pl-s"><span class="pl-pds">"</span>/path/to/directory<span class="pl-pds">"</span></span>)
library(<span class="pl-smi">dplyr</span>)
library(<span class="pl-smi">data.table</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Make an edited version of the variant data to merge with the GWAS datasets
                     </h3>
                     <a aria-label="Permalink: Make an edited version of the variant data to merge with the GWAS datasets" class="anchor" href="#make-an-edited-version-of-the-variant-data-to-merge-with-the-gwas-datasets" id="user-content-make-an-edited-version-of-the-variant-data-to-merge-with-the-gwas-datasets">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The individual GWAS files from the Neale lab only include some of the
information that you need to carry out the
                     <code>
                      munge
                     </code>
                     step (sample size,
MAF and effect statistics). Therefore, the Neale lab also provide a
general variant file that contains annotation information for all the
variants measured within a single file in the same order that they
appear in the GWAS files. By merging this file with the
                     <strong>
                      variant
                     </strong>
                     column in the trait GWAS you can then get the rsIDs, A1 (i.e.the alt
allele), A2 (i.e.the ref allele) and INFO information that you need for
your analysis. See
                     <a href="https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit?ts=5b5f17db#gid=227859291" rel="nofollow">
                      here
                     </a>
                     for further information on the variant file and to download it.
                    </p>
                    <p dir="auto">
                     <strong>
                      Note:
                     </strong>
                     Since the UK Biobank files are very large we found it best to
initially make an edited variant file containing only the columns that
you need for this formatting using the code below as this significantly
reduces runtime in the later steps.
                    </p>
                    <p dir="auto">
                     In addition, we use the
                     <code>
                      fread
                     </code>
                     function from the
                     <em>
                      data.table
                     </em>
                     R package
to read in the datasets in our below example. This requires the
downloaded files to be unzipped so that they are .tsv files rather than
.bgz ones (i.e.the downloaded format from the Neale lab website). Using
the base R
                     <code>
                      read.table
                     </code>
                     function also works but is
                     <em>
                      much
                     </em>
                     slower.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='variant_data &lt;- fread("variants.tsv")
variant_data &lt;- as.data.table(variant_data)
head(variant_data)

variant_data_edited &lt;- variant_data %&gt;% select(variant, ref, alt, rsid, info)
head(variant_data_edited)

write.table(variant_data_edited, file = "variant_data_edited_UKB.txt", row.names = FALSE,
            quote = FALSE)' dir="auto">
                     <pre><span class="pl-smi">variant_data</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>variants.tsv<span class="pl-pds">"</span></span>)
<span class="pl-smi">variant_data</span> <span class="pl-k">&lt;-</span> as.data.table(<span class="pl-smi">variant_data</span>)
head(<span class="pl-smi">variant_data</span>)

<span class="pl-smi">variant_data_edited</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">variant_data</span> %<span class="pl-k">&gt;</span>% select(<span class="pl-smi">variant</span>, <span class="pl-smi">ref</span>, <span class="pl-smi">alt</span>, <span class="pl-smi">rsid</span>, <span class="pl-smi">info</span>)
head(<span class="pl-smi">variant_data_edited</span>)

write.table(<span class="pl-smi">variant_data_edited</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>variant_data_edited_UKB.txt<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>,
            <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Format UK Biobank GWAS data ready for munging
                     </h3>
                     <a aria-label="Permalink: Format UK Biobank GWAS data ready for munging" class="anchor" href="#format-uk-biobank-gwas-data-ready-for-munging" id="user-content-format-uk-biobank-gwas-data-ready-for-munging">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Now we can use this edited variant file to merge with the GWASs of our
traits of interest. Please see an example script for the trait
                     <em>
                      feelings
of loneliness/isolation
                     </em>
                     below. In addition to merging the data we use
the
                     <em>
                      dplyr
                     </em>
                     R package to
                     <code>
                      rename
                     </code>
                     the columns to names that are
recognised by the
                     <code>
                      munge
                     </code>
                     functions and we
                     <code>
                      select
                     </code>
                     only the required
columns to reduce the size of the edited file.
                    </p>
                    <p dir="auto">
                     <strong>
                      Note:
                     </strong>
                     The
                     <code>
                      merge
                     </code>
                     step takes a while to run.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='### Read in original GWAS sumstats file using fread 

loneliness_UKB_data &lt;- fread("2020.gwas.imputed_v3.both_sexes.tsv")
loneliness_UKB_data &lt;- as.data.table(loneliness_UKB_data)
head(loneliness_UKB_data)

## read in our edited Neale variants file

variant_data &lt;- fread("variant_data_edited_UKB.txt")
variant_data &lt;- as.data.table(variant_data)
head(variant_data)

## join the phenotype file with the variant file by the variant column

joined_df &lt;- merge(loneliness_UKB_data, variant_data , by.x = "variant", 
                   by.y = "variant", all.x = TRUE, all.y = FALSE)

head(joined_df)

## rename the columns &amp; keep only those required for munging

loneliness_UKB_data_edited &lt;- joined_df %&gt;% rename(A1 = "alt") %&gt;% 
  rename(A2 = "ref") %&gt;% 
  rename(MAF = "minor_AF") %&gt;% 
  rename(N = "n_complete_samples") %&gt;% 
  rename(BETA = "beta") %&gt;% 
  rename(SE = "se") %&gt;% 
  rename(P = "pval") %&gt;% 
  rename(SNP = "rsid") %&gt;% 
  rename(INFO = "info") %&gt;%
  select(A1, A2, MAF, N, BETA, SE, P, SNP, INFO)

head(loneliness_UKB_data_edited)

## Write new table as txt file munging step

write.table(loneliness_UKB_data_edited, file = "loneliness_UKB_GWAS_edit.txt", 
            row.names = FALSE, quote = FALSE)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>## Read in original GWAS sumstats file using fread </span>

<span class="pl-smi">loneliness_UKB_data</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>2020.gwas.imputed_v3.both_sexes.tsv<span class="pl-pds">"</span></span>)
<span class="pl-smi">loneliness_UKB_data</span> <span class="pl-k">&lt;-</span> as.data.table(<span class="pl-smi">loneliness_UKB_data</span>)
head(<span class="pl-smi">loneliness_UKB_data</span>)

<span class="pl-c"><span class="pl-c">#</span># read in our edited Neale variants file</span>

<span class="pl-smi">variant_data</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>variant_data_edited_UKB.txt<span class="pl-pds">"</span></span>)
<span class="pl-smi">variant_data</span> <span class="pl-k">&lt;-</span> as.data.table(<span class="pl-smi">variant_data</span>)
head(<span class="pl-smi">variant_data</span>)

<span class="pl-c"><span class="pl-c">#</span># join the phenotype file with the variant file by the variant column</span>

<span class="pl-smi">joined_df</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-smi">loneliness_UKB_data</span>, <span class="pl-smi">variant_data</span> , <span class="pl-v">by.x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>variant<span class="pl-pds">"</span></span>, 
                   <span class="pl-v">by.y</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>variant<span class="pl-pds">"</span></span>, <span class="pl-v">all.x</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">all.y</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)

head(<span class="pl-smi">joined_df</span>)

<span class="pl-c"><span class="pl-c">#</span># rename the columns &amp; keep only those required for munging</span>

<span class="pl-smi">loneliness_UKB_data_edited</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">joined_df</span> %<span class="pl-k">&gt;</span>% rename(<span class="pl-v">A1</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>alt<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  rename(<span class="pl-v">A2</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ref<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  rename(<span class="pl-v">MAF</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>minor_AF<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  rename(<span class="pl-v">N</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>n_complete_samples<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  rename(<span class="pl-v">BETA</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>beta<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  rename(<span class="pl-v">SE</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>se<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  rename(<span class="pl-v">P</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pval<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  rename(<span class="pl-v">SNP</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>rsid<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  rename(<span class="pl-v">INFO</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>info<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>%
  select(<span class="pl-smi">A1</span>, <span class="pl-smi">A2</span>, <span class="pl-smi">MAF</span>, <span class="pl-smi">N</span>, <span class="pl-smi">BETA</span>, <span class="pl-smi">SE</span>, <span class="pl-smi">P</span>, <span class="pl-smi">SNP</span>, <span class="pl-smi">INFO</span>)

head(<span class="pl-smi">loneliness_UKB_data_edited</span>)

<span class="pl-c"><span class="pl-c">#</span># Write new table as txt file munging step</span>

write.table(<span class="pl-smi">loneliness_UKB_data_edited</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>loneliness_UKB_GWAS_edit.txt<span class="pl-pds">"</span></span>, 
            <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      LD score regression
                     </h1>
                     <a aria-label="Permalink: LD score regression" class="anchor" href="#ld-score-regression" id="user-content-ld-score-regression">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     To ensure that our traits were sufficiently heritable to be meaningfully
modelled using genomic SEM we initially performed LD score regression to
calculate the SNP-based heritability of each trait. We also used
cross-trait LD score regression to assess the level of inter-correlation
present between each pair of traits.
                    </p>
                    <p dir="auto">
                     This stage of the analysis utilised the command line
                     <em>
                      LDSC
                     </em>
                     software
developed by Brendan Bulik-Sullivan and colleagues (for in depth details
please refer to their
                     <a href="https://github.com/bulik/ldsc">
                      original wiki
                     </a>
                     ).
We performed the next few steps in the terminal on an Ubuntu personal
desktop (Linux OS).
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Install LDSC software
                     </h3>
                     <a aria-label="Permalink: Install LDSC software" class="anchor" href="#install-ldsc-software" id="user-content-install-ldsc-software">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The LDSC software is run in the command line and relies on a collection
of Python dependencies to run. In order to install LDSC you first need
to download
                     <a href="https://www.anaconda.com/products/individual" rel="nofollow">
                      Anaconda
                     </a>
                     so
that you can create an environment that includes the package
dependencies specifically required for the software. More instructions
on this can be found on the
                     <a href="https://github.com/bulik/ldsc">
                      original LDSC
wiki
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     However, we found that if you are using a more recent version of
Anaconda (we used Anaconda3 version 2020.07), you may run into some
functionality issues with the LDSC functions if you use the environment
file specified in the original software. We therefore used an adapted
environment file (environment_edit.yml) within our analysis.
                    </p>
                    <p dir="auto">
                     <strong>
                      Adapted environment file
                     </strong>
                    </p>
                    <p dir="auto">
                     In a text editor we wrote the following text and saved it as
                     <strong>
                      LDSC_env.yml
                     </strong>
                     :
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="name: environment_edit
channels:
- bioconda
dependencies:
- python=2.7
- bitarray=0.8
- nose=1.3
- pybedtools=0.7
- pip
- pip:
  - scipy==0.18
  - pandas==0.19
  - numpy==1.16" dir="auto">
                     <pre>name: environment_edit
channels:
- bioconda
dependencies:
- python=2.7
- bitarray=0.8
- nose=1.3
- pybedtools=0.7
- pip
- pip:
  - scipy==0.18
  - pandas==0.19
  - numpy==1.16</pre>
                    </div>
                    <p dir="auto">
                     We then clone the repository for LDSC and create the environment with
the adapted package dependencies.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="git clone https://github.com/bulik/ldsc.git
cd ldsc
conda env create --file LDSC_env.yml
source activate LDSC_env" dir="auto">
                     <pre>git clone https://github.com/bulik/ldsc.git
<span class="pl-c1">cd</span> ldsc
conda env create --file LDSC_env.yml
<span class="pl-c1">source</span> activate LDSC_env</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Munge data
                     </h2>
                     <a aria-label="Permalink: Munge data" class="anchor" href="#munge-data" id="user-content-munge-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     First we must munge the GWAS data so that it can be recognised by the
ldsc software and the data is filtered to only include well-imputed SNPs
with a MAF &gt;0.01. See the
                     <strong>
                      Reformatting Summary Statistics
                     </strong>
                     section
                     <a href="https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation">
                      here
                     </a>
                     for further details.
                    </p>
                    <p dir="auto">
                     In short, for the
                     <code>
                      munge
                     </code>
                     function to work we need the following columns
in the downloaded GWAS summary statistics as a minimum:
                    </p>
                    <ul dir="auto">
                     <li>
                      SNP rsIDs
                     </li>
                     <li>
                      Allele 1 (i.e.the effect allele)
                     </li>
                     <li>
                      Allele 2 (i.e.the non-effect allele)
                     </li>
                     <li>
                      Sample size
                     </li>
                     <li>
                      The regression effect (e.g.beta, OR, logOR, Z-score)
                     </li>
                     <li>
                      P values of the effect
                     </li>
                    </ul>
                    <p dir="auto">
                     It is also useful to have the MAF and INFO columns for filtering, but
these are not crucial since we filter to a reference genome. Here we
used Hapmap 3 SNPs with the MHC region removed as our reference genome.
This data can be downloaded from:
                     <a href="https://utexas.app.box.com/s/vkd36n197m8klbaio3yzoxsee6sxo11v" rel="nofollow">
                      https://utexas.app.box.com/s/vkd36n197m8klbaio3yzoxsee6sxo11v
                     </a>
                     .
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /media/user/Elements/R_markdown_test/

# AD

./ldsc/munge_sumstats.py \
--sumstats Kunkle_etal_Stage1_results.txt \
--N 63926 \
--out ad_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# MDD

./ldsc/munge_sumstats.py \
--sumstats MDD2018_ex23andMe.gz \
--N 173005 \
--out mdd_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# Insomnia

./ldsc/munge_sumstats.py \
--sumstats Insomnia_sumstats_Jansenetal.txt.gz \
--N 386533 \
--out insomnia_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# Loneliness

./ldsc/munge_sumstats.py \
--sumstats loneliness_UKB_GWAS_edit.txt \
--N 355583 \
--out loneliness_ldsc \
--merge-alleles w_hm3.noMHC.snplist \


# Low social activity

./ldsc/munge_sumstats.py \
--sumstats no_social_activity_UKB_GWAS_edit.txt \
--N 360063 \
--out no_social_activity_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# Hearing difficulty

./ldsc/munge_sumstats.py \
--sumstats hearing_diff_UKB_GWAS_edit.txt \
--N 353983 \
--out hearing_difficulty_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# Less education

./ldsc/munge_sumstats.py \
--sumstats low_education_UKB_GWAS_edit.txt \
--N 357549 \
--out low_education_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# Physical inactivity

./ldsc/munge_sumstats.py \
--sumstats physical_inactivity_UKB_GWAS_edit.txt \
--N 359263 \
--out physical_inactivity_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# Smoking

./ldsc/munge_sumstats.py \
--sumstats current_smoker_UKB_GWAS_edit.txt \
--N 359706 \
--out current_smoker_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# Alcohol intake frequency

./ldsc/munge_sumstats.py \
--sumstats alcohol_intake_freq_UKB_GWAS_edit.txt \
--N 360726 \
--out alcohol_intake_freq_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# BMI

./ldsc/munge_sumstats.py \
--sumstats bmi_UKB_GWAS_edit.txt \
--N 354831 \
--out bmi_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# Deprivation status

./ldsc/munge_sumstats.py \
--sumstats deprivation_UKB_GWAS_edit.txt \
--N 360763 \
--out deprivation_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# Systolic blood pressure

./ldsc/munge_sumstats.py \
--sumstats systolic_bp_UKB_GWAS_edit.txt \
--N 340159 \
--out systolic_bp_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

# Type 2 diabetes

./ldsc/munge_sumstats.py \
--sumstats diabetes_with_rsids.txt \
--N 159208 \
--out type_2_diabetes_ldsc \
--merge-alleles w_hm3.noMHC.snplist" dir="auto">
                     <pre><span class="pl-c1">cd</span> /media/user/Elements/R_markdown_test/

<span class="pl-c"><span class="pl-c">#</span> AD</span>

./ldsc/munge_sumstats.py \
--sumstats Kunkle_etal_Stage1_results.txt \
--N 63926 \
--out ad_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> MDD</span>

./ldsc/munge_sumstats.py \
--sumstats MDD2018_ex23andMe.gz \
--N 173005 \
--out mdd_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> Insomnia</span>

./ldsc/munge_sumstats.py \
--sumstats Insomnia_sumstats_Jansenetal.txt.gz \
--N 386533 \
--out insomnia_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> Loneliness</span>

./ldsc/munge_sumstats.py \
--sumstats loneliness_UKB_GWAS_edit.txt \
--N 355583 \
--out loneliness_ldsc \
--merge-alleles w_hm3.noMHC.snplist \


<span class="pl-c"><span class="pl-c">#</span> Low social activity</span>

./ldsc/munge_sumstats.py \
--sumstats no_social_activity_UKB_GWAS_edit.txt \
--N 360063 \
--out no_social_activity_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> Hearing difficulty</span>

./ldsc/munge_sumstats.py \
--sumstats hearing_diff_UKB_GWAS_edit.txt \
--N 353983 \
--out hearing_difficulty_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> Less education</span>

./ldsc/munge_sumstats.py \
--sumstats low_education_UKB_GWAS_edit.txt \
--N 357549 \
--out low_education_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> Physical inactivity</span>

./ldsc/munge_sumstats.py \
--sumstats physical_inactivity_UKB_GWAS_edit.txt \
--N 359263 \
--out physical_inactivity_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> Smoking</span>

./ldsc/munge_sumstats.py \
--sumstats current_smoker_UKB_GWAS_edit.txt \
--N 359706 \
--out current_smoker_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> Alcohol intake frequency</span>

./ldsc/munge_sumstats.py \
--sumstats alcohol_intake_freq_UKB_GWAS_edit.txt \
--N 360726 \
--out alcohol_intake_freq_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> BMI</span>

./ldsc/munge_sumstats.py \
--sumstats bmi_UKB_GWAS_edit.txt \
--N 354831 \
--out bmi_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> Deprivation status</span>

./ldsc/munge_sumstats.py \
--sumstats deprivation_UKB_GWAS_edit.txt \
--N 360763 \
--out deprivation_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> Systolic blood pressure</span>

./ldsc/munge_sumstats.py \
--sumstats systolic_bp_UKB_GWAS_edit.txt \
--N 340159 \
--out systolic_bp_ldsc \
--merge-alleles w_hm3.noMHC.snplist \

<span class="pl-c"><span class="pl-c">#</span> Type 2 diabetes</span>

./ldsc/munge_sumstats.py \
--sumstats diabetes_with_rsids.txt \
--N 159208 \
--out type_2_diabetes_ldsc \
--merge-alleles w_hm3.noMHC.snplist</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Calculating univariate SNP-based heritability
                     </h2>
                     <a aria-label="Permalink: Calculating univariate SNP-based heritability" class="anchor" href="#calculating-univariate-snp-based-heritability" id="user-content-calculating-univariate-snp-based-heritability">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Now we want to calculate the univariate SNP-based heritability of each
trait so that we can assess whether it is sufficiently heritable to be
included in our main analysis (i.e.whether it has a h2 Z score &gt;4). We
use the
                     <code>
                      ldsc.py
                     </code>
                     function and the
                     <code>
                      --h2
                     </code>
                     argument within the LDSC
software. See
                     <a href="https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation">
                      here
                     </a>
                     for further details.
                    </p>
                    <p dir="auto">
                     You need to use LD scores and weights that match your datas ancestral
background. Here we used pre-calculated European LD scores and weights
for our analysis from the Broad Institute computed from 1000 Genomes
data. These can be downloaded from:
                     <a href="https://alkesgroup.broadinstitute.org/LDSCORE/" rel="nofollow">
                      https://alkesgroup.broadinstitute.org/LDSCORE/
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     This group has made a range of different LD scores and weights that are
publically available but if you need to make some of your own see the
                     <a href="https://github.com/bulik/ldsc/wiki/LD-Score-Estimation-Tutorial">
                      LDSC
wiki
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     For
                     <strong>
                      binary
                     </strong>
                     traits you need to specify the sample and population
prevalences (see
                     <code>
                      sample-prev
                     </code>
                     and
                     <code>
                      population-prev
                     </code>
                     in the code below).
                    </p>
                    <p dir="auto">
                     Sample prevalence can be calculated by dividing the
                     <em>
                      number of cases
                     </em>
                     by
the
                     <em>
                      total sample size
                     </em>
                     .
                    </p>
                    <p dir="auto">
                     Population prevalences are based on a relevant estimate that is widely
cited and reliable for your population of interest (see the
                     <strong>
                      Supplementary Methods
                     </strong>
                     of our
                     <a href="https://www.sciencedirect.com/science/article/pii/S0197458022001312" rel="nofollow">
                      paper
                     </a>
                     for details on how we identified our population prevalence estimates).
                    </p>
                    <p dir="auto">
                     You do not need to provide the sample and population prevalence for
                     <strong>
                      continuous
                     </strong>
                     or
                     <strong>
                      categorical
                     </strong>
                     traits.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# AD 

./ldsc/ldsc.py \
--h2 ad_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out ad_h2 \
--samp-prev 0.34 \
--pop-prev 0.05 \

# MDD

./ldsc/ldsc.py \
--h2 mdd_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out mdd_h2 \
--samp-prev 0.35 \
--pop-prev 0.13 

# Insomnia
./ldsc/ldsc.py \
--h2 insomnia_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out insomnia_h2 \
--samp-prev 0.28 \
--pop-prev 0.32 

# Loneliness
./ldsc/ldsc.py \
--h2 loneliness_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out loneliness_h2 \
--samp-prev 0.18 \
--pop-prev 0.08 

# Low social activity
./ldsc/ldsc.py \
--h2 no_social_activity_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out no_social_activity_h2 \
--samp-prev 0.30 \
--pop-prev 0.30 

# Hearing difficulty
./ldsc/ldsc.py \
--h2 hearing_difficulty_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out hearing_difficulty_h2 \
--samp-prev 0.38 \
--pop-prev 0.39 

# Less education
./ldsc/ldsc.py \
--h2 low_education_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out low_education_h2 \
--samp-prev 0.17 \
--pop-prev 0.27 

# Physical inactivity
./ldsc/ldsc.py \
--h2 physical_inactivity_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out physical_inactivity_h2 \
--samp-prev 0.06 \
--pop-prev 0.18

# Smoking
./ldsc/ldsc.py \
--h2 current_smoker_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out current_smoker_h2 \
--samp-prev 0.10 \
--pop-prev 0.27

# Alcohol intake frequency
./ldsc/ldsc.py \
--h2 alcohol_intake_freq_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out alcohol_intake_freq_h2 

# BMI
./ldsc/ldsc.py \
--h2 bmi_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out bmi_h2 

# Deprivation status
./ldsc/ldsc.py \
--h2 deprivation_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out deprivation_h2 

# Systolic bp
./ldsc/ldsc.py \
--h2 systolic_bp_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out systolic_bp_h2 

# Type 2 diabetes
./ldsc/ldsc.py \
--h2 type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out type_2_diabetes_h2 \
--samp-prev 0.17 \
--pop-prev 0.06" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> AD </span>

./ldsc/ldsc.py \
--h2 ad_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out ad_h2 \
--samp-prev 0.34 \
--pop-prev 0.05 \

<span class="pl-c"><span class="pl-c">#</span> MDD</span>

./ldsc/ldsc.py \
--h2 mdd_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out mdd_h2 \
--samp-prev 0.35 \
--pop-prev 0.13 

<span class="pl-c"><span class="pl-c">#</span> Insomnia</span>
./ldsc/ldsc.py \
--h2 insomnia_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out insomnia_h2 \
--samp-prev 0.28 \
--pop-prev 0.32 

<span class="pl-c"><span class="pl-c">#</span> Loneliness</span>
./ldsc/ldsc.py \
--h2 loneliness_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out loneliness_h2 \
--samp-prev 0.18 \
--pop-prev 0.08 

<span class="pl-c"><span class="pl-c">#</span> Low social activity</span>
./ldsc/ldsc.py \
--h2 no_social_activity_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out no_social_activity_h2 \
--samp-prev 0.30 \
--pop-prev 0.30 

<span class="pl-c"><span class="pl-c">#</span> Hearing difficulty</span>
./ldsc/ldsc.py \
--h2 hearing_difficulty_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out hearing_difficulty_h2 \
--samp-prev 0.38 \
--pop-prev 0.39 

<span class="pl-c"><span class="pl-c">#</span> Less education</span>
./ldsc/ldsc.py \
--h2 low_education_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out low_education_h2 \
--samp-prev 0.17 \
--pop-prev 0.27 

<span class="pl-c"><span class="pl-c">#</span> Physical inactivity</span>
./ldsc/ldsc.py \
--h2 physical_inactivity_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out physical_inactivity_h2 \
--samp-prev 0.06 \
--pop-prev 0.18

<span class="pl-c"><span class="pl-c">#</span> Smoking</span>
./ldsc/ldsc.py \
--h2 current_smoker_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out current_smoker_h2 \
--samp-prev 0.10 \
--pop-prev 0.27

<span class="pl-c"><span class="pl-c">#</span> Alcohol intake frequency</span>
./ldsc/ldsc.py \
--h2 alcohol_intake_freq_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out alcohol_intake_freq_h2 

<span class="pl-c"><span class="pl-c">#</span> BMI</span>
./ldsc/ldsc.py \
--h2 bmi_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out bmi_h2 

<span class="pl-c"><span class="pl-c">#</span> Deprivation status</span>
./ldsc/ldsc.py \
--h2 deprivation_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out deprivation_h2 

<span class="pl-c"><span class="pl-c">#</span> Systolic bp</span>
./ldsc/ldsc.py \
--h2 systolic_bp_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out systolic_bp_h2 

<span class="pl-c"><span class="pl-c">#</span> Type 2 diabetes</span>
./ldsc/ldsc.py \
--h2 type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out type_2_diabetes_h2 \
--samp-prev 0.17 \
--pop-prev 0.06</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Calculating pairwise genetic correlations
                     </h2>
                     <a aria-label="Permalink: Calculating pairwise genetic correlations" class="anchor" href="#calculating-pairwise-genetic-correlations" id="user-content-calculating-pairwise-genetic-correlations">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Next, we want to calculate the pairwise genetic correlations between
each pair of traits to ascertain how much genetic correlation is present
between our traits. We use the
                     <code>
                      ldsc.py
                     </code>
                     function with the
                     <code>
                      --rg
                     </code>
                     argument. See
                     <a href="https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation">
                      here
                     </a>
                     for further details.
                    </p>
                    <p dir="auto">
                     As with the previous step we use pre-calculated European LD scores and
weights for our analysis from the Broad Institute computed from 1000
Genomes data downloaded from:
                     <a href="https://alkesgroup.broadinstitute.org/LDSCORE/" rel="nofollow">
                      https://alkesgroup.broadinstitute.org/LDSCORE/
                     </a>
                     .
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# AD
./ldsc/ldsc.py \
--rg ad_ldsc.sumstats.gz,mdd_ldsc.sumstats.gz,insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\
hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out ad_rg \
--samp-prev 0.34,0.35,0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.05,0.13,0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06 

# MDD
./ldsc/ldsc.py \
--rg mdd_ldsc.sumstats.gz,insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\
hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out mdd_rg \
--samp-prev 0.35,0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.13,0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06

# insomnia
./ldsc/ldsc.py \
--rg insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\
hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out insomnia_rg \
--samp-prev 0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06

# loneliness
./ldsc/ldsc.py \
--rg loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\
hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out loneliness_rg \
--samp-prev 0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06

# low social activity
./ldsc/ldsc.py \
--rg no_social_activity_ldsc.sumstats.gz,\
hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out no_social_activity_rg \
--samp-prev 0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06

# hearing difficulty
./ldsc/ldsc.py \
--rg hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out hearing_difficulty_rg \
--samp-prev 0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06

# less education
./ldsc/ldsc.py \
--rg low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out low_education_rg \
--samp-prev 0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.27,0.18,0.27,nan,nan,nan,nan,0.06

# physical inactivity
./ldsc/ldsc.py \
--rg physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out physical_inactivity_rg \
--samp-prev 0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.18,0.27,nan,nan,nan,nan,0.06

# smoking
./ldsc/ldsc.py \
--rg current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out current_smoker_rg \
--samp-prev 0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.27,nan,nan,nan,nan,0.06

# alcohol intake freq
./ldsc/ldsc.py \
--rg alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out alcohol_intake_freq_rg \
--samp-prev nan,nan,nan,nan,0.17 \
--pop-prev nan,nan,nan,nan,0.06

# BMI
./ldsc/ldsc.py \
--rg bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out bmi_rg \
--samp-prev nan,nan,nan,0.17 \
--pop-prev nan,nan,nan,0.06

# deprivation
./ldsc/ldsc.py \
--rg deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out deprivation_rg \
--samp-prev nan,nan,0.17 \
--pop-prev nan,nan,0.06

# systolic bp
./ldsc/ldsc.py \
--rg systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out systolic_bp_rg \
--samp-prev nan,0.17 \
--pop-prev nan,0.06" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> AD</span>
./ldsc/ldsc.py \
--rg ad_ldsc.sumstats.gz,mdd_ldsc.sumstats.gz,insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\
hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out ad_rg \
--samp-prev 0.34,0.35,0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.05,0.13,0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06 

<span class="pl-c"><span class="pl-c">#</span> MDD</span>
./ldsc/ldsc.py \
--rg mdd_ldsc.sumstats.gz,insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\
hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out mdd_rg \
--samp-prev 0.35,0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.13,0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06

<span class="pl-c"><span class="pl-c">#</span> insomnia</span>
./ldsc/ldsc.py \
--rg insomnia_ldsc.sumstats.gz,loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\
hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out insomnia_rg \
--samp-prev 0.28,0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.32,0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06

<span class="pl-c"><span class="pl-c">#</span> loneliness</span>
./ldsc/ldsc.py \
--rg loneliness_ldsc.sumstats.gz,no_social_activity_ldsc.sumstats.gz,\
hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out loneliness_rg \
--samp-prev 0.18,0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.08,0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06

<span class="pl-c"><span class="pl-c">#</span> low social activity</span>
./ldsc/ldsc.py \
--rg no_social_activity_ldsc.sumstats.gz,\
hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out no_social_activity_rg \
--samp-prev 0.30,0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.30,0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06

<span class="pl-c"><span class="pl-c">#</span> hearing difficulty</span>
./ldsc/ldsc.py \
--rg hearing_difficulty_ldsc.sumstats.gz,low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out hearing_difficulty_rg \
--samp-prev 0.38,0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.39,0.27,0.18,0.27,nan,nan,nan,nan,0.06

<span class="pl-c"><span class="pl-c">#</span> less education</span>
./ldsc/ldsc.py \
--rg low_education_ldsc.sumstats.gz,physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out low_education_rg \
--samp-prev 0.17,0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.27,0.18,0.27,nan,nan,nan,nan,0.06

<span class="pl-c"><span class="pl-c">#</span> physical inactivity</span>
./ldsc/ldsc.py \
--rg physical_inactivity_ldsc.sumstats.gz,\
current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out physical_inactivity_rg \
--samp-prev 0.06,0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.18,0.27,nan,nan,nan,nan,0.06

<span class="pl-c"><span class="pl-c">#</span> smoking</span>
./ldsc/ldsc.py \
--rg current_smoker_ldsc.sumstats.gz,alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out current_smoker_rg \
--samp-prev 0.10,nan,nan,nan,nan,0.17 \
--pop-prev 0.27,nan,nan,nan,nan,0.06

<span class="pl-c"><span class="pl-c">#</span> alcohol intake freq</span>
./ldsc/ldsc.py \
--rg alcohol_intake_freq_ldsc.sumstats.gz,bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out alcohol_intake_freq_rg \
--samp-prev nan,nan,nan,nan,0.17 \
--pop-prev nan,nan,nan,nan,0.06

<span class="pl-c"><span class="pl-c">#</span> BMI</span>
./ldsc/ldsc.py \
--rg bmi_ldsc.sumstats.gz,deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out bmi_rg \
--samp-prev nan,nan,nan,0.17 \
--pop-prev nan,nan,nan,0.06

<span class="pl-c"><span class="pl-c">#</span> deprivation</span>
./ldsc/ldsc.py \
--rg deprivation_ldsc.sumstats.gz,\
systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out deprivation_rg \
--samp-prev nan,nan,0.17 \
--pop-prev nan,nan,0.06

<span class="pl-c"><span class="pl-c">#</span> systolic bp</span>
./ldsc/ldsc.py \
--rg systolic_bp_ldsc.sumstats.gz,type_2_diabetes_ldsc.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out systolic_bp_rg \
--samp-prev nan,0.17 \
--pop-prev nan,0.06</pre>
                    </div>
                    <p dir="auto">
                     The LDSC output gives us a table of the genetic correlations at the end
of the .log file that we can easily copy into Excel.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Plotting the data
                     </h2>
                     <a aria-label="Permalink: Plotting the data" class="anchor" href="#plotting-the-data" id="user-content-plotting-the-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The remaining analytic steps were all performed in R Studio (Version
1.2.5033) using R version 3.6.3 on a Windows OS desktop.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Genetic correlation matrix
                     </h3>
                     <a aria-label="Permalink: Genetic correlation matrix" class="anchor" href="#genetic-correlation-matrix" id="user-content-genetic-correlation-matrix">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Visualising pairwise genetic correlations between lots of traits in a
table is not an easy way to study the results. Plotting your genetic
correlations in a correlation matrix plot provides far better visuals of
the correlations allowing for better interpretation of your results. A
great R package for this is
                     <em>
                      corrplot
                     </em>
                     and we provide the code script
below for the figure we used in our
                     <a href="https://www.sciencedirect.com/science/article/pii/S0197458022001312" rel="nofollow">
                      paper
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     For a detailed description of the different aesthetic options available
in this package see
                     <a href="https://cran.r-project.org/web/packages/corrplot/vignettes/corrplot-intro.html" rel="nofollow">
                      here
                     </a>
                     for a wide range of vignettes.
                    </p>
                    <p dir="auto">
                     First we load the required R packages and set our working directory. We
use
                     <em>
                      tidyverse
                     </em>
                     to turn our .csv file data into appropriate matrices for
the
                     <code>
                      corrplot
                     </code>
                     function.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='setwd("E:/GWAS_sumstats")
library(tidyverse)
library(corrplot)' dir="auto">
                     <pre>setwd(<span class="pl-s"><span class="pl-pds">"</span>E:/GWAS_sumstats<span class="pl-pds">"</span></span>)
library(<span class="pl-smi">tidyverse</span>)
library(<span class="pl-smi">corrplot</span>)</pre>
                    </div>
                    <p dir="auto">
                     In our matrix plot we wanted to include both the correlation
coefficients between the traits but also information about the
statistical significance of the correlation. Therefore we initially made
2 separate .csv files based on the output from the LDSC analysis.
                    </p>
                    <p dir="auto">
                     <strong>
                      .csv file 1
                     </strong>
                     This file consisted of 3 columns named
                     <em>
                      trait_1
                     </em>
                     ,
                     <em>
                      trait_2
                     </em>
                     and
                     <em>
                      rg
                     </em>
                     . We wrote abbreviated trait names so that the labels
on the matrix plot will be easier to read (see image below).
                     <strong>
                      Rg
                     </strong>
                     refers to the correlation coefficient estimate. As the central line on
the matrix plot is the trait correlated with itself we give these
variables a value of 1.
                    </p>
                    <p dir="auto">
                     <strong>
                      .csv file 2
                     </strong>
                     This file consisted of 3 columns named
                     <em>
                      trait_1
                     </em>
                     ,
                     <em>
                      trait_2
                     </em>
                     and
                     <em>
                      pval
                     </em>
                     with data in the same order as the 1st file. P
values of traits with themselves are specified as 0.
                    </p>
                    <p dir="auto">
                     We then use these two files to construct a rg data matrix and p value
data matrix in R using the
                     <em>
                      tidyverse
                     </em>
                     package to use as the input for
our correlation plot (
                     <code>
                      corrplot
                     </code>
                     only accepts a data matrix as its
input).
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Load in the rg csv and make a duplicate for other side of the matrix
rg &lt;- read.csv(file = &quot;LDSC_all_gen_corr_no_pval.csv&quot;, header = T, sep = &quot;,&quot;)
rg

rg2&lt;-rg

# Swap the column names for trait1 and trait 2 
# (because the other half of the triangle is just a transposed version of the extra half we need)
names(rg2)&lt;-c(&quot;trait_2&quot;, &quot;trait_1&quot;, &quot;rg&quot;)
rg2
  
# Drop the columns with an rg of 1 that we don't need as these are already in the first dataset
# and as they represent the matrix intersection we don't need the values twice

rg3&lt;-rg2 %&gt;% mutate(na_if(trait_1, trait_2)) %&gt;% 
  drop_na() %&gt;% 
  select(trait_2, trait_1, rg) #removing the extra column we used to complete this step
rg3

# Combine to make a matrix 
all&lt;-rbind(rg, rg3) %&gt;% #combine the two sides of the matrix
  pivot_wider(names_from=trait_1, values_from=rg) %&gt;% # turn data into wide format
  column_to_rownames(var=&quot;trait_2&quot;) #transfer the trait_2 column to be the row names

all

# Save matrix
write.table(all, file = &quot;ldsc_rg_matrix.txt&quot;)

# Repeat to make p value matrix
options(scipen=10)
pval &lt;- read.csv(file = &quot;LDSC_all_pval.csv&quot;, header = T, sep = &quot;,&quot;)
pval

pval2&lt;-pval

names(pval2)&lt;-c(&quot;trait_2&quot;, &quot;trait_1&quot;, &quot;pval&quot;)
pval2

pval3&lt;-pval2 %&gt;% mutate(na_if(trait_1, trait_2)) %&gt;% 
  drop_na() %&gt;% 
  select(trait_2, trait_1, pval)
pval3

all_pval&lt;-rbind(pval, pval3) %&gt;% 
  pivot_wider(names_from=trait_1, values_from=pval) %&gt;% 
  column_to_rownames(var=&quot;trait_2&quot;)

all_pval

write.table(all_pval, file = &quot;ldsc_pval_matrix.txt&quot;)

# Convert to matrices for corrplot
all_matrix &lt;- as.matrix(all)
all_matrix

pval_matrix &lt;- as.matrix(all_pval)
pval_matrix" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Load in the rg csv and make a duplicate for other side of the matrix</span>
<span class="pl-smi">rg</span> <span class="pl-k">&lt;-</span> read.csv(<span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>LDSC_all_gen_corr_no_pval.csv<span class="pl-pds">"</span></span>, <span class="pl-v">header</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>)
<span class="pl-smi">rg</span>

<span class="pl-smi">rg2</span><span class="pl-k">&lt;-</span><span class="pl-smi">rg</span>

<span class="pl-c"><span class="pl-c">#</span> Swap the column names for trait1 and trait 2 </span>
<span class="pl-c"><span class="pl-c">#</span> (because the other half of the triangle is just a transposed version of the extra half we need)</span>
names(<span class="pl-smi">rg2</span>)<span class="pl-k">&lt;-</span>c(<span class="pl-s"><span class="pl-pds">"</span>trait_2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>trait_1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>rg<span class="pl-pds">"</span></span>)
<span class="pl-smi">rg2</span>
  
<span class="pl-c"><span class="pl-c">#</span> Drop the columns with an rg of 1 that we don't need as these are already in the first dataset</span>
<span class="pl-c"><span class="pl-c">#</span> and as they represent the matrix intersection we don't need the values twice</span>

<span class="pl-smi">rg3</span><span class="pl-k">&lt;-</span><span class="pl-smi">rg2</span> %<span class="pl-k">&gt;</span>% mutate(na_if(<span class="pl-smi">trait_1</span>, <span class="pl-smi">trait_2</span>)) %<span class="pl-k">&gt;</span>% 
  drop_na() %<span class="pl-k">&gt;</span>% 
  select(<span class="pl-smi">trait_2</span>, <span class="pl-smi">trait_1</span>, <span class="pl-smi">rg</span>) <span class="pl-c"><span class="pl-c">#</span>removing the extra column we used to complete this step</span>
<span class="pl-smi">rg3</span>

<span class="pl-c"><span class="pl-c">#</span> Combine to make a matrix </span>
<span class="pl-smi">all</span><span class="pl-k">&lt;-</span>rbind(<span class="pl-smi">rg</span>, <span class="pl-smi">rg3</span>) %<span class="pl-k">&gt;</span>% <span class="pl-c"><span class="pl-c">#</span>combine the two sides of the matrix</span>
  pivot_wider(<span class="pl-v">names_from</span><span class="pl-k">=</span><span class="pl-smi">trait_1</span>, <span class="pl-v">values_from</span><span class="pl-k">=</span><span class="pl-smi">rg</span>) %<span class="pl-k">&gt;</span>% <span class="pl-c"><span class="pl-c">#</span> turn data into wide format</span>
  column_to_rownames(<span class="pl-v">var</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>trait_2<span class="pl-pds">"</span></span>) <span class="pl-c"><span class="pl-c">#</span>transfer the trait_2 column to be the row names</span>

<span class="pl-smi">all</span>

<span class="pl-c"><span class="pl-c">#</span> Save matrix</span>
write.table(<span class="pl-smi">all</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ldsc_rg_matrix.txt<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Repeat to make p value matrix</span>
options(<span class="pl-v">scipen</span><span class="pl-k">=</span><span class="pl-c1">10</span>)
<span class="pl-smi">pval</span> <span class="pl-k">&lt;-</span> read.csv(<span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>LDSC_all_pval.csv<span class="pl-pds">"</span></span>, <span class="pl-v">header</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>)
<span class="pl-smi">pval</span>

<span class="pl-smi">pval2</span><span class="pl-k">&lt;-</span><span class="pl-smi">pval</span>

names(<span class="pl-smi">pval2</span>)<span class="pl-k">&lt;-</span>c(<span class="pl-s"><span class="pl-pds">"</span>trait_2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>trait_1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>pval<span class="pl-pds">"</span></span>)
<span class="pl-smi">pval2</span>

<span class="pl-smi">pval3</span><span class="pl-k">&lt;-</span><span class="pl-smi">pval2</span> %<span class="pl-k">&gt;</span>% mutate(na_if(<span class="pl-smi">trait_1</span>, <span class="pl-smi">trait_2</span>)) %<span class="pl-k">&gt;</span>% 
  drop_na() %<span class="pl-k">&gt;</span>% 
  select(<span class="pl-smi">trait_2</span>, <span class="pl-smi">trait_1</span>, <span class="pl-smi">pval</span>)
<span class="pl-smi">pval3</span>

<span class="pl-smi">all_pval</span><span class="pl-k">&lt;-</span>rbind(<span class="pl-smi">pval</span>, <span class="pl-smi">pval3</span>) %<span class="pl-k">&gt;</span>% 
  pivot_wider(<span class="pl-v">names_from</span><span class="pl-k">=</span><span class="pl-smi">trait_1</span>, <span class="pl-v">values_from</span><span class="pl-k">=</span><span class="pl-smi">pval</span>) %<span class="pl-k">&gt;</span>% 
  column_to_rownames(<span class="pl-v">var</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>trait_2<span class="pl-pds">"</span></span>)

<span class="pl-smi">all_pval</span>

write.table(<span class="pl-smi">all_pval</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ldsc_pval_matrix.txt<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Convert to matrices for corrplot</span>
<span class="pl-smi">all_matrix</span> <span class="pl-k">&lt;-</span> as.matrix(<span class="pl-smi">all</span>)
<span class="pl-smi">all_matrix</span>

<span class="pl-smi">pval_matrix</span> <span class="pl-k">&lt;-</span> as.matrix(<span class="pl-smi">all_pval</span>)
<span class="pl-smi">pval_matrix</span></pre>
                    </div>
                    <p dir="auto">
                     Now we can plot a matrix that depicts the genetic correlation
coefficients on the lower triangle of the matrix and the p values on the
upper triangle (marked by asterisks if they are statistically
significant). We use a Bonferroni corrected p value threshold to adjust
for multiple testing.
                    </p>
                    <p dir="auto">
                     <strong>
                      Bonferroni calculation:
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      Step 1:
                     </strong>
                     Calculate how many pairwise tests are being performed by:
                     <a href="https://camo.githubusercontent.com/ea2444288bb985558559486efbc2696a9278c58c1b19cfda8fe75afa51c1680d/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b6b2533444e756d6265724f665472616974732532412532384e756d6265724f665472616974732532423125323925324632" rel="noopener noreferrer nofollow" target="_blank">
                      <img alt="k=NumberOfTraits*(NumberOfTraits+1)/2" data-canonical-src="https://latex.codecogs.com/png.image?%5Cdpi%7B110%7D&amp;space;%5Cbg_white&amp;space;k%3DNumberOfTraits%2A%28NumberOfTraits%2B1%29%2F2" src="https://camo.githubusercontent.com/ea2444288bb985558559486efbc2696a9278c58c1b19cfda8fe75afa51c1680d/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b6b2533444e756d6265724f665472616974732532412532384e756d6265724f665472616974732532423125323925324632" style="max-width: 100%;" title="k=NumberOfTraits*(NumberOfTraits+1)/2"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Step 2:
                     </strong>
                     Then calculate p value adjusting for the number of tests:
                     <a href="https://camo.githubusercontent.com/71927ad8653b130ef6790eb31e5327605f8896535fa2920a887d298960c17c32/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b50253344302e30352532466b" rel="noopener noreferrer nofollow" target="_blank">
                      <img alt="P=0.05/k" data-canonical-src="https://latex.codecogs.com/png.image?%5Cdpi%7B110%7D&amp;space;%5Cbg_white&amp;space;P%3D0.05%2Fk" src="https://camo.githubusercontent.com/71927ad8653b130ef6790eb31e5327605f8896535fa2920a887d298960c17c32/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b50253344302e30352532466b" style="max-width: 100%;" title="P=0.05/k"/>
                     </a>
                    </p>
                    <p dir="auto">
                     In our case there were 14 traits giving us 105 pairwise tests with a
corrected p value of &lt;4.76E-4.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='col &lt;- colorRampPalette(c("#BB4444", "#EE9988", "#FFFFFF", "#77AADD", "#4477AA"))
# choose colour pallete

# save pdf of corr matrix with asterisks for significant values and no numeric values
pdf(file = "LDSC_all_pval.pdf", height = 5, width = 5.2)
corr_plot &lt;- corrplot(all_matrix, method = "square", col = col(200),
                          type = "full",  order = "original",
                          tl.pos = "lt", tl.col = "black", tl.srt = 90, tl.cex = .7, # Text label color &amp; rotation
                          cl.pos = "r", cl.cex = .7, cl.ratio = 0.1,
                          # Combine with significance
                          p.mat = pval_matrix, sig.level = 4.76E-4, insig = "label_sig",
                          pch.col = "black", pch.cex = 1.2, 
                          # hide correlation coefficient on the principal diagonal
                          diag = TRUE)
dev.off()

### PDF corr matrix where the upper matrix has p values asterisks 
# and the lower displays correlation coefficients
pdf(file = "LDSC_all_pval_with_values.pdf", height = 5, width = 5.2)
corrplot(all_matrix, method = "color", col = col(200),
         addCoef.col = "black", number.cex = .6, number.digits = 2, # Add coefficient of correlation
         type = "lower",  order = "original",
         tl.pos = "lt", tl.col = "black", tl.srt = 90, tl.cex = .7, # Text label color and rotation
         cl.pos = "r", cl.cex = .7, cl.ratio = 0.1, 
         # Combine with significance
         diag = TRUE)
corrplot(all_matrix, add = TRUE, type = "upper", method = "square", order = "original", col = col(200),
         diag = FALSE, tl.pos = "n", cl.pos = "n",
         p.mat = pval_matrix, sig.level = 4.76E-4, insig = "label_sig", 
         pch.col = "black", pch.cex = 1.2)
dev.off()' dir="auto">
                     <pre><span class="pl-smi">col</span> <span class="pl-k">&lt;-</span> colorRampPalette(c(<span class="pl-s"><span class="pl-pds">"</span>#BB4444<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#EE9988<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#FFFFFF<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#77AADD<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#4477AA<span class="pl-pds">"</span></span>))
<span class="pl-c"><span class="pl-c">#</span> choose colour pallete</span>

<span class="pl-c"><span class="pl-c">#</span> save pdf of corr matrix with asterisks for significant values and no numeric values</span>
pdf(<span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>LDSC_all_pval.pdf<span class="pl-pds">"</span></span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">5</span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">5.2</span>)
<span class="pl-smi">corr_plot</span> <span class="pl-k">&lt;-</span> corrplot(<span class="pl-smi">all_matrix</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>square<span class="pl-pds">"</span></span>, <span class="pl-v">col</span> <span class="pl-k">=</span> col(<span class="pl-c1">200</span>),
                          <span class="pl-v">type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>full<span class="pl-pds">"</span></span>,  <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>original<span class="pl-pds">"</span></span>,
                          <span class="pl-v">tl.pos</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>lt<span class="pl-pds">"</span></span>, <span class="pl-v">tl.col</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-v">tl.srt</span> <span class="pl-k">=</span> <span class="pl-c1">90</span>, <span class="pl-v">tl.cex</span> <span class="pl-k">=</span> <span class="pl-smi">.7</span>, <span class="pl-c"><span class="pl-c">#</span> Text label color &amp; rotation</span>
                          <span class="pl-v">cl.pos</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>r<span class="pl-pds">"</span></span>, <span class="pl-v">cl.cex</span> <span class="pl-k">=</span> <span class="pl-smi">.7</span>, <span class="pl-v">cl.ratio</span> <span class="pl-k">=</span> <span class="pl-c1">0.1</span>,
                          <span class="pl-c"><span class="pl-c">#</span> Combine with significance</span>
                          <span class="pl-v">p.mat</span> <span class="pl-k">=</span> <span class="pl-smi">pval_matrix</span>, <span class="pl-v">sig.level</span> <span class="pl-k">=</span> <span class="pl-c1">4.76E-4</span>, <span class="pl-v">insig</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>label_sig<span class="pl-pds">"</span></span>,
                          <span class="pl-v">pch.col</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-v">pch.cex</span> <span class="pl-k">=</span> <span class="pl-c1">1.2</span>, 
                          <span class="pl-c"><span class="pl-c">#</span> hide correlation coefficient on the principal diagonal</span>
                          <span class="pl-v">diag</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()

<span class="pl-c"><span class="pl-c">#</span>## PDF corr matrix where the upper matrix has p values asterisks </span>
<span class="pl-c"><span class="pl-c">#</span> and the lower displays correlation coefficients</span>
pdf(<span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>LDSC_all_pval_with_values.pdf<span class="pl-pds">"</span></span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">5</span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">5.2</span>)
corrplot(<span class="pl-smi">all_matrix</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>color<span class="pl-pds">"</span></span>, <span class="pl-v">col</span> <span class="pl-k">=</span> col(<span class="pl-c1">200</span>),
         <span class="pl-v">addCoef.col</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-v">number.cex</span> <span class="pl-k">=</span> <span class="pl-smi">.6</span>, <span class="pl-v">number.digits</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, <span class="pl-c"><span class="pl-c">#</span> Add coefficient of correlation</span>
         <span class="pl-v">type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>lower<span class="pl-pds">"</span></span>,  <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>original<span class="pl-pds">"</span></span>,
         <span class="pl-v">tl.pos</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>lt<span class="pl-pds">"</span></span>, <span class="pl-v">tl.col</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-v">tl.srt</span> <span class="pl-k">=</span> <span class="pl-c1">90</span>, <span class="pl-v">tl.cex</span> <span class="pl-k">=</span> <span class="pl-smi">.7</span>, <span class="pl-c"><span class="pl-c">#</span> Text label color and rotation</span>
         <span class="pl-v">cl.pos</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>r<span class="pl-pds">"</span></span>, <span class="pl-v">cl.cex</span> <span class="pl-k">=</span> <span class="pl-smi">.7</span>, <span class="pl-v">cl.ratio</span> <span class="pl-k">=</span> <span class="pl-c1">0.1</span>, 
         <span class="pl-c"><span class="pl-c">#</span> Combine with significance</span>
         <span class="pl-v">diag</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
corrplot(<span class="pl-smi">all_matrix</span>, <span class="pl-v">add</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>upper<span class="pl-pds">"</span></span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>square<span class="pl-pds">"</span></span>, <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>original<span class="pl-pds">"</span></span>, <span class="pl-v">col</span> <span class="pl-k">=</span> col(<span class="pl-c1">200</span>),
         <span class="pl-v">diag</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">tl.pos</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>n<span class="pl-pds">"</span></span>, <span class="pl-v">cl.pos</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>n<span class="pl-pds">"</span></span>,
         <span class="pl-v">p.mat</span> <span class="pl-k">=</span> <span class="pl-smi">pval_matrix</span>, <span class="pl-v">sig.level</span> <span class="pl-k">=</span> <span class="pl-c1">4.76E-4</span>, <span class="pl-v">insig</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>label_sig<span class="pl-pds">"</span></span>, 
         <span class="pl-v">pch.col</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-v">pch.cex</span> <span class="pl-k">=</span> <span class="pl-c1">1.2</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Output image:
                     </strong>
                    </p>
                    <p align="center" dir="auto">
                     <a href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/ldscmatrix_bivariate.jpg" rel="noopener noreferrer" target="_blank">
                      <img src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/ldscmatrix_bivariate.jpg" style="max-width: 100%;" width="500"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Undirected weighted graph
                     </h3>
                     <a aria-label="Permalink: Undirected weighted graph" class="anchor" href="#undirected-weighted-graph" id="user-content-undirected-weighted-graph">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The matrix plot gives us a nice image of pairwise correlations but it
can still be hard to distinguish how these inter-relate between multiple
traits. Therefore, we decided to plot a weighted undirected graph of the
statistically significant genetic correlations to visualise how the
traits inter-correlate with one another. For a nice example of this type
of graph where they include the correlation coefficient on the edges
please see
                     <strong>
                      Figure 1B
                     </strong>
                     in the
                     <a href="https://doi.org/10.1016/j.cell.2019.11.020" rel="nofollow">
                      genomicSEM
paper
                     </a>
                     by Lee et al (2019).
                    </p>
                    <p dir="auto">
                     We use the
                     <code>
                      qplot
                     </code>
                     function in the
                     <em>
                      qgraph
                     </em>
                     R package to make this plot
with data functions from the
                     <em>
                      tidyverse
                     </em>
                     package.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(qgraph)" dir="auto">
                     <pre>library(<span class="pl-smi">qgraph</span>)</pre>
                    </div>
                    <p dir="auto">
                     As with the LDSC plot, we require an input .csv file with 3 columns
(
                     <em>
                      trait_1
                     </em>
                     ,
                     <em>
                      trait_2
                     </em>
                     and
                     <em>
                      rg
                     </em>
                     ). However, for rg values that are not
statistically significant we write these values as 0 so that the
non-significant relationships are not included within our graph.
                    </p>
                    <p dir="auto">
                     We then convert the data into a matrix form and save the plot as a pdf.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Load in the .csv file
rg &lt;- read.csv(file = &quot;ldsc_sig_rg_network.csv&quot;, header = T, sep = &quot;,&quot;)
rg

# Make duplicate to be the other side of the matrix

rg2&lt;-rg

# Swap the column names for trait 1 and trait 2 
names(rg2)&lt;-c(&quot;trait_2&quot;, &quot;trait_1&quot;, &quot;rg&quot;)
rg2

# Drop the columns with rg of 1 that we don't need as they are already in the first dataset
# and as they represent the intersection, we don't need the values twice
rg3&lt;-rg2 %&gt;% mutate(na_if(trait_1, trait_2)) %&gt;% 
  drop_na() %&gt;% 
  select(trait_2, trait_1, rg)
rg3

# Combine as a matrix 
all&lt;-rbind(rg, rg3) %&gt;% 
  pivot_wider(names_from=trait_1, values_from=rg) %&gt;% # Convert data to wide format
  column_to_rownames(var=&quot;trait_2&quot;) # Transfer the trait_2 column to be the row names

all

#### make matrix
all_matrix &lt;- as.matrix(all)
all_matrix


# Make a weighted undirected graph of significant correlations

pdf(file = &quot;undirected_weighted_graph.pdf&quot;, height = 4, width = 6)
qplot &lt;- qgraph(all_matrix, layout = &quot;spring&quot;, minimum = 0, posCol = &quot;darkblue&quot;, 
                negCol = &quot;red&quot;, maximum = 1, labels = colnames(all_matrix), 
                label.scale.equal = FALSE, cut = NULL, edge.labels = FALSE, 
                edge.label.bg = FALSE)
dev.off()" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Load in the .csv file</span>
<span class="pl-smi">rg</span> <span class="pl-k">&lt;-</span> read.csv(<span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ldsc_sig_rg_network.csv<span class="pl-pds">"</span></span>, <span class="pl-v">header</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>)
<span class="pl-smi">rg</span>

<span class="pl-c"><span class="pl-c">#</span> Make duplicate to be the other side of the matrix</span>

<span class="pl-smi">rg2</span><span class="pl-k">&lt;-</span><span class="pl-smi">rg</span>

<span class="pl-c"><span class="pl-c">#</span> Swap the column names for trait 1 and trait 2 </span>
names(<span class="pl-smi">rg2</span>)<span class="pl-k">&lt;-</span>c(<span class="pl-s"><span class="pl-pds">"</span>trait_2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>trait_1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>rg<span class="pl-pds">"</span></span>)
<span class="pl-smi">rg2</span>

<span class="pl-c"><span class="pl-c">#</span> Drop the columns with rg of 1 that we don't need as they are already in the first dataset</span>
<span class="pl-c"><span class="pl-c">#</span> and as they represent the intersection, we don't need the values twice</span>
<span class="pl-smi">rg3</span><span class="pl-k">&lt;-</span><span class="pl-smi">rg2</span> %<span class="pl-k">&gt;</span>% mutate(na_if(<span class="pl-smi">trait_1</span>, <span class="pl-smi">trait_2</span>)) %<span class="pl-k">&gt;</span>% 
  drop_na() %<span class="pl-k">&gt;</span>% 
  select(<span class="pl-smi">trait_2</span>, <span class="pl-smi">trait_1</span>, <span class="pl-smi">rg</span>)
<span class="pl-smi">rg3</span>

<span class="pl-c"><span class="pl-c">#</span> Combine as a matrix </span>
<span class="pl-smi">all</span><span class="pl-k">&lt;-</span>rbind(<span class="pl-smi">rg</span>, <span class="pl-smi">rg3</span>) %<span class="pl-k">&gt;</span>% 
  pivot_wider(<span class="pl-v">names_from</span><span class="pl-k">=</span><span class="pl-smi">trait_1</span>, <span class="pl-v">values_from</span><span class="pl-k">=</span><span class="pl-smi">rg</span>) %<span class="pl-k">&gt;</span>% <span class="pl-c"><span class="pl-c">#</span> Convert data to wide format</span>
  column_to_rownames(<span class="pl-v">var</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>trait_2<span class="pl-pds">"</span></span>) <span class="pl-c"><span class="pl-c">#</span> Transfer the trait_2 column to be the row names</span>

<span class="pl-smi">all</span>

<span class="pl-c"><span class="pl-c">#</span>### make matrix</span>
<span class="pl-smi">all_matrix</span> <span class="pl-k">&lt;-</span> as.matrix(<span class="pl-smi">all</span>)
<span class="pl-smi">all_matrix</span>


<span class="pl-c"><span class="pl-c">#</span> Make a weighted undirected graph of significant correlations</span>

pdf(<span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>undirected_weighted_graph.pdf<span class="pl-pds">"</span></span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">4</span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)
<span class="pl-smi">qplot</span> <span class="pl-k">&lt;-</span> qgraph(<span class="pl-smi">all_matrix</span>, <span class="pl-v">layout</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>spring<span class="pl-pds">"</span></span>, <span class="pl-v">minimum</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>, <span class="pl-v">posCol</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>darkblue<span class="pl-pds">"</span></span>, 
                <span class="pl-v">negCol</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-v">maximum</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>, <span class="pl-v">labels</span> <span class="pl-k">=</span> colnames(<span class="pl-smi">all_matrix</span>), 
                <span class="pl-v">label.scale.equal</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">cut</span> <span class="pl-k">=</span> <span class="pl-c1">NULL</span>, <span class="pl-v">edge.labels</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, 
                <span class="pl-v">edge.label.bg</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
dev.off()</pre>
                    </div>
                    <p align="center" dir="auto">
                     <a href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/undirected_weighted_graph_kunkle.jpg" rel="noopener noreferrer" target="_blank">
                      <img src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/undirected_weighted_graph_kunkle.jpg" style="max-width: 100%;" width="600"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Genomic SEM
                     </h1>
                     <a aria-label="Permalink: Genomic SEM" class="anchor" href="#genomic-sem" id="user-content-genomic-sem">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Now we conduct genomic factor analysis using the
                     <em>
                      GenomicSEM
                     </em>
                     R package,
but there are many other potential analyses one can perform (see the
                     <strong>
                      <a href="https://github.com/GenomicSEM/GenomicSEM/wiki">
                       original wiki
                      </a>
                     </strong>
                     for
further details).
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Install
                      <em>
                       GenomicSEM
                      </em>
                     </h3>
                     <a aria-label="Permalink: Install GenomicSEM" class="anchor" href="#install-genomicsem" id="user-content-install-genomicsem">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='install.packages("devtools")
library(devtools)
install_github("GenomicSEM/GenomicSEM")' dir="auto">
                     <pre>install.packages(<span class="pl-s"><span class="pl-pds">"</span>devtools<span class="pl-pds">"</span></span>)
library(<span class="pl-smi">devtools</span>)
install_github(<span class="pl-s"><span class="pl-pds">"</span>GenomicSEM/GenomicSEM<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Set working directory and load the
                      <em>
                       GenomicSEM
                      </em>
                      package
                     </h3>
                     <a aria-label="Permalink: Set working directory and load the GenomicSEM package" class="anchor" href="#set-working-directory-and-load-the-genomicsem-package" id="user-content-set-working-directory-and-load-the-genomicsem-package">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='setwd("E:/GWAS_sumstats")
library(GenomicSEM)' dir="auto">
                     <pre>setwd(<span class="pl-s"><span class="pl-pds">"</span>E:/GWAS_sumstats<span class="pl-pds">"</span></span>)
library(<span class="pl-smi">GenomicSEM</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Munge data
                     </h2>
                     <a aria-label="Permalink: Munge data" class="anchor" href="#munge-data-1" id="user-content-munge-data-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We munged the data again using the
                     <code>
                      munge
                     </code>
                     function in the
                     <em>
                      GenomicSEM
                     </em>
                     package since this version produces more conservative filtering results
than the original LDSC software (it checks whether both the A1
                     <strong>
                      and
                     </strong>
                     A2 alleles match the reference file instead of just checking A1, and
removes SNPs with missing MAF values).
                    </p>
                    <p dir="auto">
                     As with the original LDSC software, certain columns are required for
this function to work.
                    </p>
                    <p dir="auto">
                     The required columns are:
                    </p>
                    <ul dir="auto">
                     <li>
                      SNP rsID
                     </li>
                     <li>
                      A1 allele (i.e.the effect allele)
                     </li>
                     <li>
                      A2 allele (i.e.the non-effect allele)
                     </li>
                     <li>
                      The regression effect (can be logistic or linear)
                     </li>
                     <li>
                      The p value of the effect
                     </li>
                    </ul>
                    <p dir="auto">
                     As with the LDSC software, it is preferential to also have MAF and INFO
columns but is not essential since we filter the SNPs to a reference
list of well imputed SNP with MAF information. In contrast, the files do
not require a specific N column since we provide the overall sample size
as an argument in the
                     <code>
                      munge
                     </code>
                     function (but if there is a column present
then the function automatically reads this instead of your provided
values).
                    </p>
                    <p dir="auto">
                     In our analysis we used Hapmap 3 SNPs with the MHC region removed as our
reference genome. This data can be downloaded from:
                     <a href="https://utexas.app.box.com/s/vkd36n197m8klbaio3yzoxsee6sxo11v" rel="nofollow">
                      https://utexas.app.box.com/s/vkd36n197m8klbaio3yzoxsee6sxo11v
                     </a>
                     .
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='munge(files = c("Kunkle_etal_Stage1_results.txt", "MDD2018_ex23andMe.gz", "Insomnia_sumstats_Jansenetal.txt.gz", 
                "loneliness_UKB_GWAS_edit.txt", "no_social_activity_UKB_GWAS_edit.txt", 
                "hearing_diff_UKB_GWAS_edit.txt", "low_education_UKB_GWAS_edit.txt",
                "physical_inactivity_UKB_GWAS_edit.txt", "current_smoker_UKB_GWAS_edit.txt", 
                "alcohol_intake_freq_UKB_GWAS_edit.txt", "bmi_UKB_GWAS_edit.txt", 
                "deprivation_UKB_GWAS_edit.txt", "systolic_bp_UKB_GWAS_edit.txt", 
                "diabetes_with_rsids.txt"), # a list of the dataset files
      hm3 = "w_hm3.noMHC.snplist", ## file of the Hapmap3 SNPs with MHC region removed
      trait.names = c("AD", "MDD", "insomnia", "loneliness_UKB", "no_social_activity_UKB", 
                      "hearing_difficulty_UKB", "low_education_UKB", "physical_inactivity_UKB", 
                      "current_smoker_UKB", "alcohol_intake_UKB", "BMI_UKB", "deprivation_UKB", 
                      "systolic_BP_UKB", "Diabetes"), # list of names of your traits
      N=c(63926, 173005, 386533, 355583, 360063, 353983, 357549, 359263, 359706, 360726, 354831, 
          360763, 340159, 159208), # list of total sample sizes of your traits
      info.filter = 0.9, # imputation filter (default used)
      maf.filter = 0.01) # minor allele frequency filter (default used)' dir="auto">
                     <pre>munge(<span class="pl-v">files</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Kunkle_etal_Stage1_results.txt<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>MDD2018_ex23andMe.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Insomnia_sumstats_Jansenetal.txt.gz<span class="pl-pds">"</span></span>, 
                <span class="pl-s"><span class="pl-pds">"</span>loneliness_UKB_GWAS_edit.txt<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>no_social_activity_UKB_GWAS_edit.txt<span class="pl-pds">"</span></span>, 
                <span class="pl-s"><span class="pl-pds">"</span>hearing_diff_UKB_GWAS_edit.txt<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>low_education_UKB_GWAS_edit.txt<span class="pl-pds">"</span></span>,
                <span class="pl-s"><span class="pl-pds">"</span>physical_inactivity_UKB_GWAS_edit.txt<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>current_smoker_UKB_GWAS_edit.txt<span class="pl-pds">"</span></span>, 
                <span class="pl-s"><span class="pl-pds">"</span>alcohol_intake_freq_UKB_GWAS_edit.txt<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>bmi_UKB_GWAS_edit.txt<span class="pl-pds">"</span></span>, 
                <span class="pl-s"><span class="pl-pds">"</span>deprivation_UKB_GWAS_edit.txt<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>systolic_bp_UKB_GWAS_edit.txt<span class="pl-pds">"</span></span>, 
                <span class="pl-s"><span class="pl-pds">"</span>diabetes_with_rsids.txt<span class="pl-pds">"</span></span>), <span class="pl-c"><span class="pl-c">#</span> a list of the dataset files</span>
      <span class="pl-v">hm3</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>w_hm3.noMHC.snplist<span class="pl-pds">"</span></span>, <span class="pl-c"><span class="pl-c">#</span># file of the Hapmap3 SNPs with MHC region removed</span>
      <span class="pl-v">trait.names</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>AD<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>MDD<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>insomnia<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>loneliness_UKB<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>no_social_activity_UKB<span class="pl-pds">"</span></span>, 
                      <span class="pl-s"><span class="pl-pds">"</span>hearing_difficulty_UKB<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>low_education_UKB<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>physical_inactivity_UKB<span class="pl-pds">"</span></span>, 
                      <span class="pl-s"><span class="pl-pds">"</span>current_smoker_UKB<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>alcohol_intake_UKB<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>BMI_UKB<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>deprivation_UKB<span class="pl-pds">"</span></span>, 
                      <span class="pl-s"><span class="pl-pds">"</span>systolic_BP_UKB<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Diabetes<span class="pl-pds">"</span></span>), <span class="pl-c"><span class="pl-c">#</span> list of names of your traits</span>
      <span class="pl-v">N</span><span class="pl-k">=</span>c(<span class="pl-c1">63926</span>, <span class="pl-c1">173005</span>, <span class="pl-c1">386533</span>, <span class="pl-c1">355583</span>, <span class="pl-c1">360063</span>, <span class="pl-c1">353983</span>, <span class="pl-c1">357549</span>, <span class="pl-c1">359263</span>, <span class="pl-c1">359706</span>, <span class="pl-c1">360726</span>, <span class="pl-c1">354831</span>, 
          <span class="pl-c1">360763</span>, <span class="pl-c1">340159</span>, <span class="pl-c1">159208</span>), <span class="pl-c"><span class="pl-c">#</span> list of total sample sizes of your traits</span>
      <span class="pl-v">info.filter</span> <span class="pl-k">=</span> <span class="pl-c1">0.9</span>, <span class="pl-c"><span class="pl-c">#</span> imputation filter (default used)</span>
      <span class="pl-v">maf.filter</span> <span class="pl-k">=</span> <span class="pl-c1">0.01</span>) <span class="pl-c"><span class="pl-c">#</span> minor allele frequency filter (default used)</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Exploratory factor analysis performed in the odd autosomes
                     </h2>
                     <a aria-label="Permalink: Exploratory factor analysis performed in the odd autosomes" class="anchor" href="#exploratory-factor-analysis-performed-in-the-odd-autosomes" id="user-content-exploratory-factor-analysis-performed-in-the-odd-autosomes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We then performed exploratory factor analysis in the odd autosomes using
the munged summary statistics files as our input.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Load the additional required packages
                     </h3>
                     <a aria-label="Permalink: Load the additional required packages" class="anchor" href="#load-the-additional-required-packages" id="user-content-load-the-additional-required-packages">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(Matrix)
library(stats)" dir="auto">
                     <pre>library(<span class="pl-smi">Matrix</span>)
library(<span class="pl-smi">stats</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Conduct multivariable LD score regression (odd autosomes)
                     </h3>
                     <a aria-label="Permalink: Conduct multivariable LD score regression (odd autosomes)" class="anchor" href="#conduct-multivariable-ld-score-regression-odd-autosomes" id="user-content-conduct-multivariable-ld-score-regression-odd-autosomes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We ran multivariable LD-Score regression to calculate the genetic
covariance matrix (
                     <em>
                      S
                     </em>
                     ) and associated sampling covariance matrix (
                     <em>
                      V
                     </em>
                     )
using the odd autosomes only. When you run an analysis where you are
performing both exploratory and confirmatory factor analysis splitting
the data in this way can guard against model overfitting. If you have
non-overlapping replication datasets for each of your traits then the
best practice would be to run EFA in the discovery data and the CFA in
the replication data, but this was not possible for all of the current
study traits.
                    </p>
                    <p dir="auto">
                     As with the initial LD score regression performed using the LDSC
software, we use pre-calculated European LD scores and weights for our
analysis from the Broad Institute computed from 1000 Genomes data
downloaded from:
                     <a href="https://alkesgroup.broadinstitute.org/LDSCORE/" rel="nofollow">
                      https://alkesgroup.broadinstitute.org/LDSCORE/
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     We use the same population and sample prevalence details as above.
                    </p>
                    <p dir="auto">
                     <strong>
                      Note:
                     </strong>
                     This step can take a while to run if you have lots of traits,
but for 14 traits it took only
                     <em>
                      2 minutes and 41 seconds
                     </em>
                     on a personal
desktop computer.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='ld &lt;- "eur_w_ld_chr/"
wld &lt;- "eur_w_ld_chr/" #folder of ld scores &amp; weights

traits &lt;- c("AD.sumstats.gz", "MDD.sumstats.gz", "insomnia.sumstats.gz",
            "loneliness_UKB.sumstats.gz", "no_social_activity_UKB.sumstats.gz",
            "hearing_difficulty_UKB.sumstats.gz", "low_education_UKB.sumstats.gz",
            "physical_inactivity_UKB.sumstats.gz", "current_smoker_UKB.sumstats.gz", 
            "alcohol_intake_UKB.sumstats.gz", "BMI_UKB.sumstats.gz", 
            "deprivation_UKB.sumstats.gz", "systolic_BP_UKB.sumstats.gz",
            "Diabetes.sumstats.gz") #munged sumstat files

sample.prev &lt;- c(0.34,0.35,0.28,0.18,0.30,0.38,0.17,0.06,0.10,NA,NA,NA,NA,0.17)

population.prev &lt;- c(0.05,0.13,0.32,0.08,0.30,0.39,0.27,0.18,0.27,NA,NA,NA,NA,0.06) 

trait.names &lt;- c("AD", "MDD", "INS", "LON", "LSA", "HD", "LED",
                 "LPA", "SMK", "ALC", "BMI", "DEP", "SBP", "T2DM")

LDSCoutput_odd &lt;- ldsc(traits, sample.prev, population.prev, ld, wld, trait.names, 
                   ldsc.log="E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ODD_LDSC", 
                   select = "ODD") #specify odd autosomes


##optional command to save the ldsc output in case you want to use it in a later R session. 

save(LDSCoutput_odd, file= "E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ODD_LDSC.RData")' dir="auto">
                     <pre><span class="pl-smi">ld</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>eur_w_ld_chr/<span class="pl-pds">"</span></span>
<span class="pl-smi">wld</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>eur_w_ld_chr/<span class="pl-pds">"</span></span> <span class="pl-c"><span class="pl-c">#</span>folder of ld scores &amp; weights</span>

<span class="pl-smi">traits</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>AD.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>MDD.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>insomnia.sumstats.gz<span class="pl-pds">"</span></span>,
            <span class="pl-s"><span class="pl-pds">"</span>loneliness_UKB.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>no_social_activity_UKB.sumstats.gz<span class="pl-pds">"</span></span>,
            <span class="pl-s"><span class="pl-pds">"</span>hearing_difficulty_UKB.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>low_education_UKB.sumstats.gz<span class="pl-pds">"</span></span>,
            <span class="pl-s"><span class="pl-pds">"</span>physical_inactivity_UKB.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>current_smoker_UKB.sumstats.gz<span class="pl-pds">"</span></span>, 
            <span class="pl-s"><span class="pl-pds">"</span>alcohol_intake_UKB.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>BMI_UKB.sumstats.gz<span class="pl-pds">"</span></span>, 
            <span class="pl-s"><span class="pl-pds">"</span>deprivation_UKB.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>systolic_BP_UKB.sumstats.gz<span class="pl-pds">"</span></span>,
            <span class="pl-s"><span class="pl-pds">"</span>Diabetes.sumstats.gz<span class="pl-pds">"</span></span>) <span class="pl-c"><span class="pl-c">#</span>munged sumstat files</span>

<span class="pl-smi">sample.prev</span> <span class="pl-k">&lt;-</span> c(<span class="pl-c1">0.34</span>,<span class="pl-c1">0.35</span>,<span class="pl-c1">0.28</span>,<span class="pl-c1">0.18</span>,<span class="pl-c1">0.30</span>,<span class="pl-c1">0.38</span>,<span class="pl-c1">0.17</span>,<span class="pl-c1">0.06</span>,<span class="pl-c1">0.10</span>,<span class="pl-c1">NA</span>,<span class="pl-c1">NA</span>,<span class="pl-c1">NA</span>,<span class="pl-c1">NA</span>,<span class="pl-c1">0.17</span>)

<span class="pl-smi">population.prev</span> <span class="pl-k">&lt;-</span> c(<span class="pl-c1">0.05</span>,<span class="pl-c1">0.13</span>,<span class="pl-c1">0.32</span>,<span class="pl-c1">0.08</span>,<span class="pl-c1">0.30</span>,<span class="pl-c1">0.39</span>,<span class="pl-c1">0.27</span>,<span class="pl-c1">0.18</span>,<span class="pl-c1">0.27</span>,<span class="pl-c1">NA</span>,<span class="pl-c1">NA</span>,<span class="pl-c1">NA</span>,<span class="pl-c1">NA</span>,<span class="pl-c1">0.06</span>) 

<span class="pl-smi">trait.names</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>AD<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>MDD<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>INS<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>LON<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>LSA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>HD<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>LED<span class="pl-pds">"</span></span>,
                 <span class="pl-s"><span class="pl-pds">"</span>LPA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SMK<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>ALC<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>BMI<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>DEP<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SBP<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>T2DM<span class="pl-pds">"</span></span>)

<span class="pl-smi">LDSCoutput_odd</span> <span class="pl-k">&lt;-</span> ldsc(<span class="pl-smi">traits</span>, <span class="pl-smi">sample.prev</span>, <span class="pl-smi">population.prev</span>, <span class="pl-smi">ld</span>, <span class="pl-smi">wld</span>, <span class="pl-smi">trait.names</span>, 
                   <span class="pl-v">ldsc.log</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ODD_LDSC<span class="pl-pds">"</span></span>, 
                   <span class="pl-v">select</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ODD<span class="pl-pds">"</span></span>) <span class="pl-c"><span class="pl-c">#</span>specify odd autosomes</span>


<span class="pl-c"><span class="pl-c">#</span>#optional command to save the ldsc output in case you want to use it in a later R session. </span>

save(<span class="pl-smi">LDSCoutput_odd</span>, <span class="pl-v">file</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ODD_LDSC.RData<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Note:
                     </strong>
                     If you are running an analysis with a lot of traits you may
have a k &gt;200. In such cases, you will need to specify this using the
                     <code>
                      n.blocks
                     </code>
                     argument in the
                     <code>
                      ldsc
                     </code>
                     function because otherwise you will
not have enough blocks for the jackknifing procedure of ldsc (number of
blocks has to be higher than k by at least 1).
                    </p>
                    <p dir="auto">
                     Example calculation for 14 traits (same as the calculation to calculate
the number of pairwise tests for Bonferroni correction):
                    </p>
                    <p dir="auto">
                     <a href="https://camo.githubusercontent.com/1d0ec6ba4836eaf316b27daa0d4c64d153245ac2650c5306a0015f423335aa13/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b6b253344313425324125323831342532423125323925324632" rel="noopener noreferrer nofollow" target="_blank">
                      <img alt="k=14*(14+1)/2" data-canonical-src="https://latex.codecogs.com/png.image?%5Cdpi%7B110%7D&amp;space;%5Cbg_white&amp;space;k%3D14%2A%2814%2B1%29%2F2" src="https://camo.githubusercontent.com/1d0ec6ba4836eaf316b27daa0d4c64d153245ac2650c5306a0015f423335aa13/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f706e672e696d6167653f2535436470692537423131302537442673706163653b25354362675f77686974652673706163653b6b253344313425324125323831342532423125323925324632" style="max-width: 100%;" title="k=14*(14+1)/2"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Exploratory factor analysis
                     </h3>
                     <a aria-label="Permalink: Exploratory factor analysis" class="anchor" href="#exploratory-factor-analysis" id="user-content-exploratory-factor-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The covariance matrix from the multivariable LDSC output is used as the
input for factor analysis. We initially smooth the covariance matrix
using the
                     <code>
                      nearPD
                     </code>
                     function in the
                     <em>
                      Matrix
                     </em>
                     R package so that if the
covariance matrix is non-positive definite it is smoothed prior to
performing EFA. In our example the matrix is positive definite but we
run the function anyway to make sure and to gain an appropriately
labelled matrix so that the trait names are labelled in our factor
analysis output.
                    </p>
                    <p dir="auto">
                     Then we use the smoothed covariance matrix to test different numbers of
factors using the
                     <code>
                      factanal
                     </code>
                     function in the R
                     <em>
                      stats
                     </em>
                     package to see
which model will be best to take forward to confirmatory factor
analysis. We tested a 2-, 3- and 4-factor model in our analysis and used
promax rotation (this allows for intercorrelation between the latent
factors).
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='Ssmooth&lt;-as.matrix((nearPD(LDSCoutput_odd$S, corr = FALSE))$mat)

# 2-factor model

EFA&lt;-factanal(covmat = Ssmooth, factors = 2, rotation = "promax")

print(EFA, sort=TRUE) # We sort the results by the magnitude of their loadings
print(EFA,digits=3,cutoff=.20,sort=TRUE) #specify the cut-off you want to use

# 3-factor model

EFA2&lt;-factanal(covmat = Ssmooth, factors = 3, rotation = "promax")

print(EFA2, sort=TRUE)
print(EFA2,digits=3,cutoff=.20,sort=TRUE)

# 4-factor model

EFA3&lt;-factanal(covmat = Ssmooth, factors = 4, rotation = "promax")

print(EFA3, sort=TRUE)
print(EFA3,digits=3,cutoff=.20,sort=TRUE)' dir="auto">
                     <pre><span class="pl-smi">Ssmooth</span><span class="pl-k">&lt;-</span>as.matrix((nearPD(<span class="pl-smi">LDSCoutput_odd</span><span class="pl-k">$</span><span class="pl-smi">S</span>, <span class="pl-v">corr</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>))<span class="pl-k">$</span><span class="pl-smi">mat</span>)

<span class="pl-c"><span class="pl-c">#</span> 2-factor model</span>

<span class="pl-smi">EFA</span><span class="pl-k">&lt;-</span>factanal(<span class="pl-v">covmat</span> <span class="pl-k">=</span> <span class="pl-smi">Ssmooth</span>, <span class="pl-v">factors</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, <span class="pl-v">rotation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>promax<span class="pl-pds">"</span></span>)

print(<span class="pl-smi">EFA</span>, <span class="pl-v">sort</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>) <span class="pl-c"><span class="pl-c">#</span> We sort the results by the magnitude of their loadings</span>
print(<span class="pl-smi">EFA</span>,<span class="pl-v">digits</span><span class="pl-k">=</span><span class="pl-c1">3</span>,<span class="pl-v">cutoff</span><span class="pl-k">=</span><span class="pl-smi">.20</span>,<span class="pl-v">sort</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>) <span class="pl-c"><span class="pl-c">#</span>specify the cut-off you want to use</span>

<span class="pl-c"><span class="pl-c">#</span> 3-factor model</span>

<span class="pl-smi">EFA2</span><span class="pl-k">&lt;-</span>factanal(<span class="pl-v">covmat</span> <span class="pl-k">=</span> <span class="pl-smi">Ssmooth</span>, <span class="pl-v">factors</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">rotation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>promax<span class="pl-pds">"</span></span>)

print(<span class="pl-smi">EFA2</span>, <span class="pl-v">sort</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
print(<span class="pl-smi">EFA2</span>,<span class="pl-v">digits</span><span class="pl-k">=</span><span class="pl-c1">3</span>,<span class="pl-v">cutoff</span><span class="pl-k">=</span><span class="pl-smi">.20</span>,<span class="pl-v">sort</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)

<span class="pl-c"><span class="pl-c">#</span> 4-factor model</span>

<span class="pl-smi">EFA3</span><span class="pl-k">&lt;-</span>factanal(<span class="pl-v">covmat</span> <span class="pl-k">=</span> <span class="pl-smi">Ssmooth</span>, <span class="pl-v">factors</span> <span class="pl-k">=</span> <span class="pl-c1">4</span>, <span class="pl-v">rotation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>promax<span class="pl-pds">"</span></span>)

print(<span class="pl-smi">EFA3</span>, <span class="pl-v">sort</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
print(<span class="pl-smi">EFA3</span>,<span class="pl-v">digits</span><span class="pl-k">=</span><span class="pl-c1">3</span>,<span class="pl-v">cutoff</span><span class="pl-k">=</span><span class="pl-smi">.20</span>,<span class="pl-v">sort</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)</pre>
                    </div>
                    <p dir="auto">
                     The output prints details about the factor loadings, the factor scores,
the factor correlations and the uniqueness scores for each individual
trait. From this analysis we chose to base our subsequent analyses on a
3-factor model. Please see our
                     <a href="https://www.sciencedirect.com/science/article/pii/S0197458022001312" rel="nofollow">
                      paper
                     </a>
                     for further details about the methods and how we chose how many factors
to extract based on the results from the above analysis.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      EFA loadings plot
                     </h3>
                     <a aria-label="Permalink: EFA loadings plot" class="anchor" href="#efa-loadings-plot" id="user-content-efa-loadings-plot">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The factor loadings table can be a little hard to visualise and
interpret, so we decided to make a plot of the loadings for our 3-factor
model based on an adapted version of code script written by Dan Mirman
(see
                     <a href="https://rpubs.com/danmirman/plotting_factor_analysis" rel="nofollow">
                      here
                     </a>
                     ). This
step uses the
                     <em>
                      ggplot2
                     </em>
                     R package.
                    </p>
                    <p dir="auto">
                     <strong>
                      Load in the required additional packages
                     </strong>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(ggplot2)
library(tidyr)" dir="auto">
                     <pre>library(<span class="pl-smi">ggplot2</span>)
library(<span class="pl-smi">tidyr</span>)</pre>
                    </div>
                    <p dir="auto">
                     We made a .csv file in Excel of the factor loadings from the EFA step to
use as input for this code. However, we flipped the order of the traits
in the .csv file so that they appear in the correct order in the graph
(otherwise they will appear in reverse order).
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Load in csv file

EFA_loadings_sorted &lt;- read.csv(file = &quot;EFA_loadings_odd_ordered.csv&quot;)
EFA_loadings_sorted

# This step ensures that the traits stay in the order you want them in the final plot 
# (note this is still in reverse order here), otherwise they are plotted in a random order

EFA_loadings_sorted$Trait&lt;- factor(EFA_loadings_sorted$Trait, 
                                   levels = EFA_loadings_sorted$Trait)

# Makes the plot long so column of factors 
# (use the gather function of the tidyr package)

EFA_loading_long &lt;- gather(EFA_loadings_sorted, key=&quot;Factor&quot;, 
                           value=&quot;Loading&quot;, c(&quot;Factor_1&quot;, &quot;Factor_2&quot;, &quot;Factor_3&quot;))

EFA_loading_long

######## Ensure the factor facets come out in the order you want

EFA_loading_long$Factor &lt;- factor(EFA_loading_long$Factor, 
                                  levels = c(&quot;Factor_1&quot;, &quot;Factor_2&quot;, &quot;Factor_3&quot;))

# Add a column to categorise the loadings by loading strength

EFA_loading_long$breaks &lt;- cut(EFA_loading_long$Loading, 
                               breaks = c(-Inf, 0, .2, .4, Inf)) 

EFA_loading_long

# Create a name vector for labelling facets

facet_names &lt;- c(
  `Factor_1` = &quot;Factor 1&quot;,
  `Factor_2` = &quot;Factor 2&quot;,
  `Factor_3` = &quot;Factor 3&quot;)

# Make your plot with a chosen theme &amp; save as a pdf

theme_set(theme_bw()) # set a chosen theme
pdf(file = &quot;EFA_loadings_plot_3factors_odd.pdf&quot;, height = 5, width = 9)
EFA_loadings_plot &lt;- ggplot(EFA_loading_long, aes(Trait, abs(Loading), fill=breaks)) + 
  facet_wrap(~ Factor, nrow=1, 
             labeller = as_labeller(facet_names)) + #place the factors in separate facets
  geom_bar(stat = &quot;identity&quot;) + #make the bars
  coord_flip() + #flip the axes so the labels can be horizontal  
  #define the fill color gradient: blue=positive, red=negative
  scale_fill_manual(name = &quot;Loading Strength&quot;, 
                    labels = c(&quot;Negative&quot;, &quot;0 to .20&quot;, &quot;.20 to .40&quot;, &quot;Greater than .40&quot;), 
                    values = c(&quot;#870E0EAA&quot;, &quot;#256C33AA&quot;, &quot;#316DB2AA&quot;, &quot;#041A58AA&quot;)) + 
  ### tell what colours to assign to each break
  scale_y_continuous(limits = c(0, 1), expand = c(0, 0),
                     breaks = seq(0, 1, by = .2)) + # change the scale of the y axis
  ylab(&quot;Standardised Factor Loading&quot;) + #name the y-axis 
  theme(panel.background = element_rect(fill = &quot;snow3&quot;, colour = &quot;snow3&quot;, size = 2, 
                                        linetype = &quot;solid&quot;), 
    panel.grid.major = element_line(size = 0.4, linetype = 'solid', colour = &quot;white&quot;), 
    panel.grid.minor = element_line(size = 0.2, linetype = 'solid', colour = &quot;white&quot;),
    strip.background = element_rect(fill = &quot;white&quot;, colour = &quot;black&quot;),
    strip.text.x = element_text(colour = &quot;black&quot;),
    axis.text = element_text(colour = &quot;gray20&quot;),
    panel.spacing.x=unit(1, &quot;lines&quot;)) +
  geom_hline(yintercept=c(.20), linetype=&quot;dashed&quot;, color = &quot;black&quot;) # add cut-off line
EFA_loadings_plot
dev.off()" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Load in csv file</span>

<span class="pl-smi">EFA_loadings_sorted</span> <span class="pl-k">&lt;-</span> read.csv(<span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>EFA_loadings_odd_ordered.csv<span class="pl-pds">"</span></span>)
<span class="pl-smi">EFA_loadings_sorted</span>

<span class="pl-c"><span class="pl-c">#</span> This step ensures that the traits stay in the order you want them in the final plot </span>
<span class="pl-c"><span class="pl-c">#</span> (note this is still in reverse order here), otherwise they are plotted in a random order</span>

<span class="pl-smi">EFA_loadings_sorted</span><span class="pl-k">$</span><span class="pl-smi">Trait</span><span class="pl-k">&lt;-</span> <span class="pl-k">factor</span>(<span class="pl-smi">EFA_loadings_sorted</span><span class="pl-k">$</span><span class="pl-smi">Trait</span>, 
                                   <span class="pl-v">levels</span> <span class="pl-k">=</span> <span class="pl-smi">EFA_loadings_sorted</span><span class="pl-k">$</span><span class="pl-smi">Trait</span>)

<span class="pl-c"><span class="pl-c">#</span> Makes the plot long so column of factors </span>
<span class="pl-c"><span class="pl-c">#</span> (use the gather function of the tidyr package)</span>

<span class="pl-smi">EFA_loading_long</span> <span class="pl-k">&lt;-</span> gather(<span class="pl-smi">EFA_loadings_sorted</span>, <span class="pl-v">key</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Factor<span class="pl-pds">"</span></span>, 
                           <span class="pl-v">value</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Loading<span class="pl-pds">"</span></span>, c(<span class="pl-s"><span class="pl-pds">"</span>Factor_1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Factor_2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Factor_3<span class="pl-pds">"</span></span>))

<span class="pl-smi">EFA_loading_long</span>

<span class="pl-c"><span class="pl-c">#</span>####### Ensure the factor facets come out in the order you want</span>

<span class="pl-smi">EFA_loading_long</span><span class="pl-k">$</span><span class="pl-smi">Factor</span> <span class="pl-k">&lt;-</span> <span class="pl-k">factor</span>(<span class="pl-smi">EFA_loading_long</span><span class="pl-k">$</span><span class="pl-smi">Factor</span>, 
                                  <span class="pl-v">levels</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Factor_1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Factor_2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Factor_3<span class="pl-pds">"</span></span>))

<span class="pl-c"><span class="pl-c">#</span> Add a column to categorise the loadings by loading strength</span>

<span class="pl-smi">EFA_loading_long</span><span class="pl-k">$</span><span class="pl-smi">breaks</span> <span class="pl-k">&lt;-</span> cut(<span class="pl-smi">EFA_loading_long</span><span class="pl-k">$</span><span class="pl-smi">Loading</span>, 
                               <span class="pl-v">breaks</span> <span class="pl-k">=</span> c(<span class="pl-k">-</span><span class="pl-c1">Inf</span>, <span class="pl-c1">0</span>, <span class="pl-smi">.2</span>, <span class="pl-smi">.4</span>, <span class="pl-c1">Inf</span>)) 

<span class="pl-smi">EFA_loading_long</span>

<span class="pl-c"><span class="pl-c">#</span> Create a name vector for labelling facets</span>

<span class="pl-smi">facet_names</span> <span class="pl-k">&lt;-</span> c(
  <span class="pl-smi">`Factor_1`</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Factor 1<span class="pl-pds">"</span></span>,
  <span class="pl-smi">`Factor_2`</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Factor 2<span class="pl-pds">"</span></span>,
  <span class="pl-smi">`Factor_3`</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Factor 3<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Make your plot with a chosen theme &amp; save as a pdf</span>

theme_set(theme_bw()) <span class="pl-c"><span class="pl-c">#</span> set a chosen theme</span>
pdf(<span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>EFA_loadings_plot_3factors_odd.pdf<span class="pl-pds">"</span></span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">5</span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">9</span>)
<span class="pl-smi">EFA_loadings_plot</span> <span class="pl-k">&lt;-</span> ggplot(<span class="pl-smi">EFA_loading_long</span>, aes(<span class="pl-smi">Trait</span>, abs(<span class="pl-smi">Loading</span>), <span class="pl-v">fill</span><span class="pl-k">=</span><span class="pl-smi">breaks</span>)) <span class="pl-k">+</span> 
  facet_wrap(<span class="pl-k">~</span> <span class="pl-smi">Factor</span>, <span class="pl-v">nrow</span><span class="pl-k">=</span><span class="pl-c1">1</span>, 
             <span class="pl-v">labeller</span> <span class="pl-k">=</span> as_labeller(<span class="pl-smi">facet_names</span>)) <span class="pl-k">+</span> <span class="pl-c"><span class="pl-c">#</span>place the factors in separate facets</span>
  geom_bar(<span class="pl-v">stat</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>identity<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> <span class="pl-c"><span class="pl-c">#</span>make the bars</span>
  coord_flip() <span class="pl-k">+</span> <span class="pl-c"><span class="pl-c">#</span>flip the axes so the labels can be horizontal  </span>
  <span class="pl-c"><span class="pl-c">#</span>define the fill color gradient: blue=positive, red=negative</span>
  scale_fill_manual(<span class="pl-v">name</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Loading Strength<span class="pl-pds">"</span></span>, 
                    <span class="pl-v">labels</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Negative<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>0 to .20<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>.20 to .40<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Greater than .40<span class="pl-pds">"</span></span>), 
                    <span class="pl-v">values</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>#870E0EAA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#256C33AA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#316DB2AA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#041A58AA<span class="pl-pds">"</span></span>)) <span class="pl-k">+</span> 
  <span class="pl-c"><span class="pl-c">#</span>## tell what colours to assign to each break</span>
  scale_y_continuous(<span class="pl-v">limits</span> <span class="pl-k">=</span> c(<span class="pl-c1">0</span>, <span class="pl-c1">1</span>), <span class="pl-v">expand</span> <span class="pl-k">=</span> c(<span class="pl-c1">0</span>, <span class="pl-c1">0</span>),
                     <span class="pl-v">breaks</span> <span class="pl-k">=</span> seq(<span class="pl-c1">0</span>, <span class="pl-c1">1</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> <span class="pl-smi">.2</span>)) <span class="pl-k">+</span> <span class="pl-c"><span class="pl-c">#</span> change the scale of the y axis</span>
  ylab(<span class="pl-s"><span class="pl-pds">"</span>Standardised Factor Loading<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> <span class="pl-c"><span class="pl-c">#</span>name the y-axis </span>
  theme(<span class="pl-v">panel.background</span> <span class="pl-k">=</span> element_rect(<span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>snow3<span class="pl-pds">"</span></span>, <span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>snow3<span class="pl-pds">"</span></span>, <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, 
                                        <span class="pl-v">linetype</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>solid<span class="pl-pds">"</span></span>), 
    <span class="pl-v">panel.grid.major</span> <span class="pl-k">=</span> element_line(<span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">0.4</span>, <span class="pl-v">linetype</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>solid<span class="pl-pds">'</span></span>, <span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>), 
    <span class="pl-v">panel.grid.minor</span> <span class="pl-k">=</span> element_line(<span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">0.2</span>, <span class="pl-v">linetype</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>solid<span class="pl-pds">'</span></span>, <span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>),
    <span class="pl-v">strip.background</span> <span class="pl-k">=</span> element_rect(<span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>, <span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>),
    <span class="pl-v">strip.text.x</span> <span class="pl-k">=</span> element_text(<span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>),
    <span class="pl-v">axis.text</span> <span class="pl-k">=</span> element_text(<span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gray20<span class="pl-pds">"</span></span>),
    <span class="pl-v">panel.spacing.x</span><span class="pl-k">=</span>unit(<span class="pl-c1">1</span>, <span class="pl-s"><span class="pl-pds">"</span>lines<span class="pl-pds">"</span></span>)) <span class="pl-k">+</span>
  geom_hline(<span class="pl-v">yintercept</span><span class="pl-k">=</span>c(<span class="pl-smi">.20</span>), <span class="pl-v">linetype</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>dashed<span class="pl-pds">"</span></span>, <span class="pl-v">color</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>) <span class="pl-c"><span class="pl-c">#</span> add cut-off line</span>
<span class="pl-smi">EFA_loadings_plot</span>
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     The above code gives us our 3-factor EFA graph:
                    </p>
                    <p align="center" dir="auto">
                     <a href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/EFA_loadings_plot_Kunkle.jpg" rel="noopener noreferrer" target="_blank">
                      <img src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/EFA_loadings_plot_Kunkle.jpg" style="max-width: 100%;" width="700"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Confirmatory factor analysis performed in the even autosomes
                     </h2>
                     <a aria-label="Permalink: Confirmatory factor analysis performed in the even autosomes" class="anchor" href="#confirmatory-factor-analysis-performed-in-the-even-autosomes" id="user-content-confirmatory-factor-analysis-performed-in-the-even-autosomes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Next, we want to test the fit of our model using confirmatory factor
analysis.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Conduct multivariable LD score regression (even autosomes)
                     </h3>
                     <a aria-label="Permalink: Conduct multivariable LD score regression (even autosomes)" class="anchor" href="#conduct-multivariable-ld-score-regression-even-autosomes" id="user-content-conduct-multivariable-ld-score-regression-even-autosomes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Because we are now performing the CFA in even autosome data we must
re-run the multivariable LD score regression step to estimate the
genetic covariance matrix for the even autosome.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='LDSCoutput_even &lt;- ldsc(traits, sample.prev, population.prev, ld, wld, trait.names, 
                   ldsc.log="E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_EVEN_LDSC", 
                   select = "EVEN") #specify even autosomes

save(LDSCoutput_even, file= "E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_EVEN_LDSC.RData")' dir="auto">
                     <pre><span class="pl-smi">LDSCoutput_even</span> <span class="pl-k">&lt;-</span> ldsc(<span class="pl-smi">traits</span>, <span class="pl-smi">sample.prev</span>, <span class="pl-smi">population.prev</span>, <span class="pl-smi">ld</span>, <span class="pl-smi">wld</span>, <span class="pl-smi">trait.names</span>, 
                   <span class="pl-v">ldsc.log</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_EVEN_LDSC<span class="pl-pds">"</span></span>, 
                   <span class="pl-v">select</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>EVEN<span class="pl-pds">"</span></span>) <span class="pl-c"><span class="pl-c">#</span>specify even autosomes</span>

save(<span class="pl-smi">LDSCoutput_even</span>, <span class="pl-v">file</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_EVEN_LDSC.RData<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Confirmatory factor analysis
                     </h3>
                     <a aria-label="Permalink: Confirmatory factor analysis" class="anchor" href="#confirmatory-factor-analysis" id="user-content-confirmatory-factor-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Using the output from the even autosome LD score regression we can then
test different models using the
                     <code>
                      usermodel
                     </code>
                     function in the
                     <em>
                      GenomicSEM
                     </em>
                     R package. We specify the parameters of our model based on the output of
our chosen EFA model (in this case the 3-factor model), including only
the factor loadings that pass our pre-specified cut-off level (0.20).
Model parameters are written in
                     <em>
                      lavaan
                     </em>
                     syntax.
                    </p>
                    <p dir="auto">
                     We test a number of different CFA models to see how including negative
loadings, cross-loadings and higher loading cut-off points influence the
model fit statistics.
                    </p>
                    <p dir="auto">
                     The output gives you the model fit statistics and the estimated
parameters for your models which can be drawn as path diagrams to better
visualise your model. The
                     <strong>
                      unstandardised
                     </strong>
                     estimates are equivalent to
covariance values whereas the
                     <strong>
                      standardised
                     </strong>
                     estimates are equivalent
to correlation coefficients.
                    </p>
                    <p dir="auto">
                     We use the default
                     <code>
                      DWLS
                     </code>
                     estimation method but the models can also be
estimated using maximum likelhood
                     <code>
                      ML
                     </code>
                     estimation if you prefer.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Model 1: CFA of 3-factor model: including traits with &gt;.20 positive loadings and cross-loadings (DWLS method),
# no negative loadings; factor correlations present 

CFAofEFA &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED + MDD + INS + HD + BMI
F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + INS + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput &lt;- usermodel(LDSCoutput_even, estimation = &quot;DWLS&quot;, model = CFAofEFA, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput

# Model 2: CFA of 3-factor model: including traits with &gt;.20 loadings and cross-loadings (DWLS method),
# negative loadings included; factor correlations present 

CFAofEFA2 &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED + MDD + INS + HD + BMI + AD
F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + INS + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS + SBP
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput2&lt;- usermodel(LDSCoutput_even, estimation = &quot;DWLS&quot;, model = CFAofEFA2, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput2

# Model 3: CFA of 3-factor model: including traits with &gt;.20 loadings and no cross-loadings (DWLS method),
# negative loadings included; factor correlations present 

CFAofEFA3 &lt;- 'F1 =~ NA*LPA + SMK + DEP
F2 =~ NA*LSA + LED + ALC + AD + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS + SBP
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput3 &lt;- usermodel(LDSCoutput_even, estimation = &quot;DWLS&quot;, model = CFAofEFA3, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput3

# Model 4: CFA of 3-factor model: including traits with &gt;.20 loadings and no cross-loadings (DWLS method),
# negative loadings excluded; factor correlations present 

CFAofEFA4 &lt;- 'F1 =~ NA*LPA + SMK + DEP
F2 =~ NA*LSA + LED + ALC + AD + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput4 &lt;- usermodel(LDSCoutput_even, estimation = &quot;DWLS&quot;, model = CFAofEFA4, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput4

# Model 5: CFA of 3-factor model: including traits with &gt;.25 positive loadings and cross-loadings (DWLS method),
# no negative loadings; factor correlations present (KEPT INSOMNIA IN AS .24)

CFAofEFA5 &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED
F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput5 &lt;- usermodel(LDSCoutput_even, estimation = &quot;DWLS&quot;, model = CFAofEFA5, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput5

# Model 6: CFA of 3-factor model: including traits with &gt;.25 positive loadings and cross-loadings (DWLS method),
# negative loadings incl; factor correlations present (KEPT INSOMNIA IN AS .24)

CFAofEFA6 &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED
F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS + SBP
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'


CFAoutput6 &lt;- usermodel(LDSCoutput_even, estimation = &quot;DWLS&quot;, model = CFAofEFA6, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput6

## .30 CUTOFF WAS THE SAME AS .25 SO DIDN'T RUN THAT" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Model 1: CFA of 3-factor model: including traits with &gt;.20 positive loadings and cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> no negative loadings; factor correlations present </span>

<span class="pl-smi">CFAofEFA</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED + MDD + INS + HD + BMI</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + INS + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_even</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput</span>

<span class="pl-c"><span class="pl-c">#</span> Model 2: CFA of 3-factor model: including traits with &gt;.20 loadings and cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> negative loadings included; factor correlations present </span>

<span class="pl-smi">CFAofEFA2</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED + MDD + INS + HD + BMI + AD</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + INS + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS + SBP</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput2</span><span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_even</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA2</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput2</span>

<span class="pl-c"><span class="pl-c">#</span> Model 3: CFA of 3-factor model: including traits with &gt;.20 loadings and no cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> negative loadings included; factor correlations present </span>

<span class="pl-smi">CFAofEFA3</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP</span>
<span class="pl-s">F2 =~ NA*LSA + LED + ALC + AD + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS + SBP</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput3</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_even</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA3</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput3</span>

<span class="pl-c"><span class="pl-c">#</span> Model 4: CFA of 3-factor model: including traits with &gt;.20 loadings and no cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> negative loadings excluded; factor correlations present </span>

<span class="pl-smi">CFAofEFA4</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP</span>
<span class="pl-s">F2 =~ NA*LSA + LED + ALC + AD + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput4</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_even</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA4</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput4</span>

<span class="pl-c"><span class="pl-c">#</span> Model 5: CFA of 3-factor model: including traits with &gt;.25 positive loadings and cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> no negative loadings; factor correlations present (KEPT INSOMNIA IN AS .24)</span>

<span class="pl-smi">CFAofEFA5</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput5</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_even</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA5</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput5</span>

<span class="pl-c"><span class="pl-c">#</span> Model 6: CFA of 3-factor model: including traits with &gt;.25 positive loadings and cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> negative loadings incl; factor correlations present (KEPT INSOMNIA IN AS .24)</span>

<span class="pl-smi">CFAofEFA6</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS + SBP</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>


<span class="pl-smi">CFAoutput6</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_even</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA6</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput6</span>

<span class="pl-c"><span class="pl-c">#</span># .30 CUTOFF WAS THE SAME AS .25 SO DIDN'T RUN THAT</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Plotting CFA models
                     </h3>
                     <a aria-label="Permalink: Plotting CFA models" class="anchor" href="#plotting-cfa-models" id="user-content-plotting-cfa-models">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Although there are specific R packages available that plot path diagrams
for SEM automatically e.g.
                     <em>
                      semPlot
                     </em>
                     , we found that the clearest and
best way is to draw out your path diagrams in PowerPoint.
                    </p>
                    <p dir="auto">
                     Below is an example of a path diagram drawn using this method based on
the standardised results of the CFA for
                     <em>
                      model 5
                     </em>
                     in the even autosomes
(see code above). This was the best fitting model that we tested so we
took it forward for subsequent analysis in data from all autosomes.
                    </p>
                    <p align="center" dir="auto">
                     <a href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/CFA_even_best_model.png" rel="noopener noreferrer" target="_blank">
                      <img src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/CFA_even_best_model.png" style="max-width: 100%;" width="700"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Follow-up genome-wide analyses
                     </h2>
                     <a aria-label="Permalink: Follow-up genome-wide analyses" class="anchor" href="#follow-up-genome-wide-analyses" id="user-content-follow-up-genome-wide-analyses">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Now we want to measure the fit of our best-fitting model and other more
complex SEM models using data from across the genome, so we run a final
multivariable LD score regression in all chromosomes.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Genome-wide multivariable LD score regression
                     </h3>
                     <a aria-label="Permalink: Genome-wide multivariable LD score regression" class="anchor" href="#genome-wide-multivariable-ld-score-regression" id="user-content-genome-wide-multivariable-ld-score-regression">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='LDSCoutput_all &lt;- ldsc(traits, sample.prev, population.prev, ld, wld, trait.names, 
                   ldsc.log="E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC")

save(LDSCoutput_all, file= "E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC.RData")' dir="auto">
                     <pre><span class="pl-smi">LDSCoutput_all</span> <span class="pl-k">&lt;-</span> ldsc(<span class="pl-smi">traits</span>, <span class="pl-smi">sample.prev</span>, <span class="pl-smi">population.prev</span>, <span class="pl-smi">ld</span>, <span class="pl-smi">wld</span>, <span class="pl-smi">trait.names</span>, 
                   <span class="pl-v">ldsc.log</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC<span class="pl-pds">"</span></span>)

save(<span class="pl-smi">LDSCoutput_all</span>, <span class="pl-v">file</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC.RData<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Genome-wide CFA
                     </h3>
                     <a aria-label="Permalink: Genome-wide CFA" class="anchor" href="#genome-wide-cfa" id="user-content-genome-wide-cfa">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We then run the 3-factor model (based off Model 5 in the code script of
the even CFA) using the LDSC output estimated from all the chromosomes.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="CFAofEFAall &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED
F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput_all &lt;- usermodel(LDSCoutput_all, estimation = &quot;DWLS&quot;, model = CFAofEFAall, 
                           CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)

CFAoutput_all" dir="auto">
                     <pre><span class="pl-smi">CFAofEFAall</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput_all</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_all</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFAall</span>, 
                           <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)

<span class="pl-smi">CFAoutput_all</span></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      The path diagram of the best-fitting CFA model (all autosomes):
                     </em>
                    </p>
                    <p align="center" dir="auto">
                     <a href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/CFA_all_best_model.png" rel="noopener noreferrer" target="_blank">
                      <img src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/CFA_all_best_model.png" style="max-width: 100%;" width="700"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Additional SEM modelling
                     </h3>
                     <a aria-label="Permalink: Additional SEM modelling" class="anchor" href="#additional-sem-modelling" id="user-content-additional-sem-modelling">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The genome-wide 3-factor CFA model fits the data moderately well.
However, there is high inter-factor correlation present, which makes it
hard to meaningfully interpret distinct patterns of covariance.
Therefore, we tested a number of additional SEM models in the
                     <em>
                      GenomicSEM
                     </em>
                     R package to try and disentangle the covariance
relationships in more detail. We provide examples of these models below
with brief explanations and example path diagrams.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Common factor model
                     </h3>
                     <a aria-label="Permalink: Common factor model" class="anchor" href="#common-factor-model" id="user-content-common-factor-model">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     A common factor model is a model where there is a single latent
construct representing common variance between all of the traits in your
data. In our example although the traits load into a common factor, the
model fit statistics are poor, which suggests that the data is better
represented by a model with multiple latent constructs.
                    </p>
                    <p dir="auto">
                     We run this model using the
                     <code>
                      commonfactor
                     </code>
                     function in
                     <em>
                      GenomicSEM
                     </em>
                     .
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='CommonFactor&lt;- commonfactor(covstruc = LDSCoutput_all, estimation="DWLS")

CommonFactor' dir="auto">
                     <pre><span class="pl-smi">CommonFactor</span><span class="pl-k">&lt;-</span> commonfactor(<span class="pl-v">covstruc</span> <span class="pl-k">=</span> <span class="pl-smi">LDSCoutput_all</span>, <span class="pl-v">estimation</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>)

<span class="pl-smi">CommonFactor</span></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      The path diagram of the common factor model:
                     </em>
                    </p>
                    <p align="center" dir="auto">
                     <a href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/CommonFactor_all_SBP.png" rel="noopener noreferrer" target="_blank">
                      <img src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/CommonFactor_all_SBP.png" style="max-width: 100%;" width="700"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Note:
                     </strong>
                     Here in this walkthrough we use the LDSC output that includes
systolic blood pressure in the common factor model. However, in our
                     <a href="https://www.sciencedirect.com/science/article/pii/S0197458022001312" rel="nofollow">
                      paper
                     </a>
                     we excluded systolic blood pressure in our genome-wide LD score
regression to estimate the parameters without it included.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Hierarchal model
                     </h3>
                     <a aria-label="Permalink: Hierarchal model" class="anchor" href="#hierarchal-model" id="user-content-hierarchal-model">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     A hierarchal model refers to a structural equation model that has two
levels of latent constructs (i.e.a latent construct of shared
covariance between the specified lower level latent constructs).
                    </p>
                    <p dir="auto">
                     Here we run a hierarchal model of our 3-factor CFA model.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="hierarchal &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED
F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS
F4 =~ F1 + F2 + F3'

hierarchal &lt;- usermodel(LDSCoutput_all, estimation = &quot;DWLS&quot;, model = hierarchal, 
                         CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
hierarchal" dir="auto">
                     <pre><span class="pl-smi">hierarchal</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F4 =~ F1 + F2 + F3<span class="pl-pds">'</span></span>

<span class="pl-smi">hierarchal</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_all</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">hierarchal</span>, 
                         <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">hierarchal</span></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      The path diagram of the hierarchal model:
                     </em>
                    </p>
                    <p align="center" dir="auto">
                     <a href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/hierarchal_all.png" rel="noopener noreferrer" target="_blank">
                      <img src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/hierarchal_all.png" style="max-width: 100%;" width="700"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Note:
                     </strong>
                     As you can see in our path diagram of the hierarchal model we
do not include the standard errors of our parameters. This is because
the model includes an endogenous variable so the STD_All column of the
results output is used to specify the standardised values. However, the
                     <em>
                      GenomicSEM
                     </em>
                     package is unable to calculate standard errors for these
estimates (see the section titled
                     <strong>
                      A Note on Standardized Output in
Genomic SEM
                     </strong>
                     on the original
                     <a href="https://github.com/GenomicSEM/GenomicSEM/wiki/3.-Models-without-Individual-SNP-effects">
                      wiki
page
                     </a>
                     for further explanation).
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Bifactor model
                     </h3>
                     <a aria-label="Permalink: Bifactor model" class="anchor" href="#bifactor-model" id="user-content-bifactor-model">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Finally, we test the fit of a bifactor model. This model produces
orthogonal factors (i.e.uncorrelated factors) by including an
overarching general latent factor that all the traits load into, as well
as additional latent factors that only distinct sub-clusters of traits
load into. The benefit of these models is that they often provide better
model fit and allow for an enhanced interpretation of what the
sub-clusters might be capturing.
                    </p>
                    <p dir="auto">
                     In our analysis, the bifactor model provided a good fit to our data and
the sub-clusters were specified based on the 3 factors in our CFA.
                    </p>
                    <p dir="auto">
                     <strong>
                      Note:
                     </strong>
                     For the bifactor model we had to constrain smoking to ensure
it didnt output as a negative residual variance (see
                     <a href="https://github.com/GenomicSEM/GenomicSEM/wiki/3.-Models-without-Individual-SNP-effects">
                      original
wiki
                     </a>
                     for further details on this).
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bifactor &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED
F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS
F4 =~ NA*AD + MDD + INS + LON + LSA + HD + LED + LPA + SMK + ALC + BMI + DEP + T2DM
F4 ~~ 0*F1
F4 ~~ 0*F2
F4 ~~ 0*F3
F1 ~~ 0*F2
F1 ~~ 0*F3
F2 ~~ 0*F3
SMK ~~  a*SMK
a &gt; .001'

bifactor &lt;- usermodel(LDSCoutput_all, estimation = &quot;DWLS&quot;, model = bifactor, 
                      CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
bifactor" dir="auto">
                     <pre><span class="pl-smi">bifactor</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F4 =~ NA*AD + MDD + INS + LON + LSA + HD + LED + LPA + SMK + ALC + BMI + DEP + T2DM</span>
<span class="pl-s">F4 ~~ 0*F1</span>
<span class="pl-s">F4 ~~ 0*F2</span>
<span class="pl-s">F4 ~~ 0*F3</span>
<span class="pl-s">F1 ~~ 0*F2</span>
<span class="pl-s">F1 ~~ 0*F3</span>
<span class="pl-s">F2 ~~ 0*F3</span>
<span class="pl-s">SMK ~~  a*SMK</span>
<span class="pl-s">a &gt; .001<span class="pl-pds">'</span></span>

<span class="pl-smi">bifactor</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_all</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">bifactor</span>, 
                      <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">bifactor</span></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      The path diagram of the bifactor model:
                     </em>
                    </p>
                    <p align="center" dir="auto">
                     <a href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/bifactor_all.png" rel="noopener noreferrer" target="_blank">
                      <img src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/bifactor_all.png" style="max-width: 100%;" width="700"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Models without the APOE region
                     </h2>
                     <a aria-label="Permalink: Models without the APOE region" class="anchor" href="#models-without-the-apoe-region" id="user-content-models-without-the-apoe-region">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Since Alzheimers disease risk is known to be disproportionately
influenced by variants in the
                     <em>
                      APOE
                     </em>
                     gene region, we ran the same models
as above in all of the autosomes except for the
                     <em>
                      APOE
                     </em>
                     region on
chromosome 19. We removed SNPs in the
                     <em>
                      APOE
                     </em>
                     region +/- 100kB (chr19:
45,309,03945,512,650) based on the position in the genome assembly that
the GWAS summary statistics had been provided in (GRCh37/hg19).
                    </p>
                    <p dir="auto">
                     We downloaded the
                     <em>
                      APOE
                     </em>
                     snplist from the
                     <a href="https://genome.ucsc.edu/index.html" rel="nofollow">
                      UCSC Genome
Browser
                     </a>
                     .
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## read in snplist of APOE region +/- 100kb each side from UCSC browser for genome build 37 to match GWAS sumstats - 249 SNPs

apoe &lt;- read.table(&quot;APOE_snps.txt&quot;, header = F, sep = &quot;&quot;, check.names = F)

## Read in munged sumstats files and remove any SNPs that match the rsIDs in the APOE snplist file

AD &lt;- fread(&quot;AD.sumstats.gz&quot;)
AD_no_apoe &lt;- anti_join(AD, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(AD_no_apoe, file = &quot;AD_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

MDD &lt;- fread(&quot;MDD.sumstats.gz&quot;)
MDD_no_apoe &lt;- anti_join(MDD, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(MDD_no_apoe, file = &quot;MDD_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

INS &lt;- fread(&quot;insomnia.sumstats.gz&quot;)
INS_no_apoe &lt;- anti_join(INS, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(INS_no_apoe, file = &quot;INS_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

LON &lt;- fread(&quot;loneliness_UKB.sumstats.gz&quot;)
LON_no_apoe &lt;- anti_join(LON, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(LON_no_apoe, file = &quot;LON_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

LSA &lt;- fread(&quot;no_social_activity_UKB.sumstats.gz&quot;)
LSA_no_apoe &lt;- anti_join(LSA, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(LSA_no_apoe, file = &quot;LSA_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

HD &lt;- fread(&quot;hearing_difficulty_UKB.sumstats.gz&quot;)
HD_no_apoe &lt;- anti_join(HD, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(HD_no_apoe, file = &quot;HD_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

LED &lt;- fread(&quot;low_education_UKB.sumstats.gz&quot;)
LED_no_apoe &lt;- anti_join(LED, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(LED_no_apoe, file = &quot;LED_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

LPA &lt;- fread(&quot;physical_inactivity_UKB.sumstats.gz&quot;)
LPA_no_apoe &lt;- anti_join(LPA, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(LPA_no_apoe, file = &quot;LPA_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

SMK &lt;- fread(&quot;current_smoker_UKB.sumstats.gz&quot;)
SMK_no_apoe &lt;- anti_join(SMK, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(SMK_no_apoe, file = &quot;SMK_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

ALC &lt;- fread(&quot;alcohol_intake_UKB.sumstats.gz&quot;)
ALC_no_apoe &lt;- anti_join(ALC, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(ALC_no_apoe, file = &quot;ALC_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

BMI &lt;- fread(&quot;BMI_UKB.sumstats.gz&quot;)
BMI_no_apoe &lt;- anti_join(BMI, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(BMI_no_apoe, file = &quot;BMI_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

DEP &lt;- fread(&quot;deprivation_UKB.sumstats.gz&quot;)
DEP_no_apoe &lt;- anti_join(DEP, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(DEP_no_apoe, file = &quot;DEP_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

SBP &lt;- fread(&quot;systolic_BP_UKB.sumstats.gz&quot;)
SBP_no_apoe &lt;- anti_join(SBP, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(SBP_no_apoe, file = &quot;SBP_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)

T2DM &lt;- fread(&quot;Diabetes.sumstats.gz&quot;)
T2DM_no_apoe &lt;- anti_join(T2DM, apoe, by = c(&quot;SNP&quot; = &quot;V4&quot;))
write.table(T2DM_no_apoe, file = &quot;T2DM_no_apoe.sumstats.gz&quot;, row.names = F, quote = F)


# Multivariable LD score regression of data without APOE

traits.apoe &lt;- c(&quot;AD_no_apoe.sumstats.gz&quot;, &quot;MDD_no_apoe.sumstats.gz&quot;, &quot;INS_no_apoe.sumstats.gz&quot;, &quot;LON_no_apoe.sumstats.gz&quot;, 
                 &quot;LSA_no_apoe.sumstats.gz&quot;, &quot;HD_no_apoe.sumstats.gz&quot;, &quot;LED_no_apoe.sumstats.gz&quot;, &quot;LPA_no_apoe.sumstats.gz&quot;, 
                 &quot;SMK_no_apoe.sumstats.gz&quot;, &quot;ALC_no_apoe.sumstats.gz&quot;, &quot;BMI_no_apoe.sumstats.gz&quot;, &quot;DEP_no_apoe.sumstats.gz&quot;,
                 &quot;SBP_no_apoe.sumstats.gz&quot;, &quot;T2DM_no_apoe.sumstats.gz&quot;)

LDSCoutput_exAPOE &lt;- ldsc(traits.apoe, sample.prev, population.prev, ld, wld, trait.names, 
                   ldsc.log=&quot;E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC_noAPOE&quot;)

save(LDSCoutput_exAPOE, file= &quot;E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC_noAPOE.RData&quot;)

# CFA model

CFAofEFA_exAPOE &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED
F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput_exAPOE &lt;- usermodel(LDSCoutput_exAPOE, estimation = &quot;DWLS&quot;, model = CFAofEFA_exAPOE, 
                            CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput_exAPOE

# Common factor

CommonFactor_exAPOE&lt;- commonfactor(covstruc = LDSCoutput_exAPOE, estimation=&quot;DWLS&quot;)
CommonFactor_exAPOE

# Hierarchal model

hierarchal_exAPOE &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED
F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS
F4 =~ F1 + F2 + F3'

hierarchal_exAPOE &lt;- usermodel(LDSCoutput_exAPOE, estimation = &quot;DWLS&quot;, model = hierarchal_exAPOE, 
                              CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
hierarchal_exAPOE

# Bifactor model

bifactor_exPOE &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED
F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM
F3 =~ NA*MDD + LON + HD + INS
F4 =~ NA*AD + MDD + INS + LON + LSA + HD + LED + LPA + SMK + ALC + BMI + DEP + T2DM
F4 ~~ 0*F1
F4 ~~ 0*F2
F4 ~~ 0*F3
F1 ~~ 0*F2
F1 ~~ 0*F3
F2 ~~ 0*F3
SMK ~~  a*SMK
a &gt; .001'

bifactor_exAPOE &lt;- usermodel(LDSCoutput_exAPOE, estimation = &quot;DWLS&quot;, model = bifactor_exAPOE, 
                           CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)

bifactor_exAPOE" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># read in snplist of APOE region +/- 100kb each side from UCSC browser for genome build 37 to match GWAS sumstats - 249 SNPs</span>

<span class="pl-smi">apoe</span> <span class="pl-k">&lt;-</span> read.table(<span class="pl-s"><span class="pl-pds">"</span>APOE_snps.txt<span class="pl-pds">"</span></span>, <span class="pl-v">header</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>, <span class="pl-v">check.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-c"><span class="pl-c">#</span># Read in munged sumstats files and remove any SNPs that match the rsIDs in the APOE snplist file</span>

<span class="pl-smi">AD</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>AD.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">AD_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">AD</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">AD_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>AD_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">MDD</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>MDD.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">MDD_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">MDD</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">MDD_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>MDD_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">INS</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>insomnia.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">INS_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">INS</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">INS_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>INS_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">LON</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>loneliness_UKB.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">LON_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">LON</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">LON_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>LON_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">LSA</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>no_social_activity_UKB.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">LSA_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">LSA</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">LSA_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>LSA_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">HD</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>hearing_difficulty_UKB.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">HD_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">HD</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">HD_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>HD_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">LED</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>low_education_UKB.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">LED_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">LED</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">LED_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>LED_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">LPA</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>physical_inactivity_UKB.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">LPA_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">LPA</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">LPA_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>LPA_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">SMK</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>current_smoker_UKB.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">SMK_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">SMK</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">SMK_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SMK_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">ALC</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>alcohol_intake_UKB.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">ALC_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">ALC</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">ALC_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ALC_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">BMI</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>BMI_UKB.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">BMI_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">BMI</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">BMI_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>BMI_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">DEP</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>deprivation_UKB.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">DEP_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">DEP</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">DEP_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DEP_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">SBP</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>systolic_BP_UKB.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">SBP_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">SBP</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">SBP_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SBP_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)

<span class="pl-smi">T2DM</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>Diabetes.sumstats.gz<span class="pl-pds">"</span></span>)
<span class="pl-smi">T2DM_no_apoe</span> <span class="pl-k">&lt;-</span> anti_join(<span class="pl-smi">T2DM</span>, <span class="pl-smi">apoe</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>V4<span class="pl-pds">"</span></span>))
write.table(<span class="pl-smi">T2DM_no_apoe</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>T2DM_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)


<span class="pl-c"><span class="pl-c">#</span> Multivariable LD score regression of data without APOE</span>

<span class="pl-smi">traits.apoe</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>AD_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>MDD_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>INS_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>LON_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, 
                 <span class="pl-s"><span class="pl-pds">"</span>LSA_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>HD_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>LED_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>LPA_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, 
                 <span class="pl-s"><span class="pl-pds">"</span>SMK_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>ALC_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>BMI_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>DEP_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>,
                 <span class="pl-s"><span class="pl-pds">"</span>SBP_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>T2DM_no_apoe.sumstats.gz<span class="pl-pds">"</span></span>)

<span class="pl-smi">LDSCoutput_exAPOE</span> <span class="pl-k">&lt;-</span> ldsc(<span class="pl-smi">traits.apoe</span>, <span class="pl-smi">sample.prev</span>, <span class="pl-smi">population.prev</span>, <span class="pl-smi">ld</span>, <span class="pl-smi">wld</span>, <span class="pl-smi">trait.names</span>, 
                   <span class="pl-v">ldsc.log</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC_noAPOE<span class="pl-pds">"</span></span>)

save(<span class="pl-smi">LDSCoutput_exAPOE</span>, <span class="pl-v">file</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>E:/GSEM_study/GSEM_AD_RISK_FACTORS_STUDY_ALL_LDSC_noAPOE.RData<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> CFA model</span>

<span class="pl-smi">CFAofEFA_exAPOE</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput_exAPOE</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_exAPOE</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA_exAPOE</span>, 
                            <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput_exAPOE</span>

<span class="pl-c"><span class="pl-c">#</span> Common factor</span>

<span class="pl-smi">CommonFactor_exAPOE</span><span class="pl-k">&lt;-</span> commonfactor(<span class="pl-v">covstruc</span> <span class="pl-k">=</span> <span class="pl-smi">LDSCoutput_exAPOE</span>, <span class="pl-v">estimation</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>)
<span class="pl-smi">CommonFactor_exAPOE</span>

<span class="pl-c"><span class="pl-c">#</span> Hierarchal model</span>

<span class="pl-smi">hierarchal_exAPOE</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F4 =~ F1 + F2 + F3<span class="pl-pds">'</span></span>

<span class="pl-smi">hierarchal_exAPOE</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_exAPOE</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">hierarchal_exAPOE</span>, 
                              <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">hierarchal_exAPOE</span>

<span class="pl-c"><span class="pl-c">#</span> Bifactor model</span>

<span class="pl-smi">bifactor_exPOE</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + LON + AD + BMI + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F4 =~ NA*AD + MDD + INS + LON + LSA + HD + LED + LPA + SMK + ALC + BMI + DEP + T2DM</span>
<span class="pl-s">F4 ~~ 0*F1</span>
<span class="pl-s">F4 ~~ 0*F2</span>
<span class="pl-s">F4 ~~ 0*F3</span>
<span class="pl-s">F1 ~~ 0*F2</span>
<span class="pl-s">F1 ~~ 0*F3</span>
<span class="pl-s">F2 ~~ 0*F3</span>
<span class="pl-s">SMK ~~  a*SMK</span>
<span class="pl-s">a &gt; .001<span class="pl-pds">'</span></span>

<span class="pl-smi">bifactor_exAPOE</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_exAPOE</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">bifactor_exAPOE</span>, 
                           <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)

<span class="pl-smi">bifactor_exAPOE</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Sensitivity analyses
                     </h1>
                     <a aria-label="Permalink: Sensitivity analyses" class="anchor" href="#sensitivity-analyses" id="user-content-sensitivity-analyses">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Because the factor loadings for Alzheimers disease varied substantially
between the odd autosome EFA, even autosome CFA and genome-wide CFA in
comparison to the risk factor traits we ran a number of follow-up
sensitivity analyses to explore why this might be.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      LD score regression in different chromosomal groupings
                     </h3>
                     <a aria-label="Permalink: LD score regression in different chromosomal groupings" class="anchor" href="#ld-score-regression-in-different-chromosomal-groupings" id="user-content-ld-score-regression-in-different-chromosomal-groupings">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We used the three outputs generated in the main study using the
                     <code>
                      ldsc
                     </code>
                     function in the
                     <em>
                      GenomicSEM
                     </em>
                     R package to compare the SNP-based
heritability and genetic correlation estimates of the included traits
when different chromosomal groupings were measured (odd vs.even
vs.all). These estimates can be found in the .log files that are
produced during previous
                     <code>
                      ldsc
                     </code>
                     steps.
                    </p>
                    <p dir="auto">
                     We then plotted the genetic correlations using
                     <code>
                      corrplot
                     </code>
                     so that we
could more easily visualise any differences between Alzheimers disease
and its risk factors.
                    </p>
                    <p dir="auto">
                     <strong>
                      Note:
                     </strong>
                     Because the output of the
                     <code>
                      ldsc
                     </code>
                     function is already an R data
matrix we dont need a separate .csv file like the earlier correlation
plot since we can just use the matrix directly as the input. But we need
to convert it to a correlation matrix using the
                     <code>
                      cov2cor
                     </code>
                     function.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Smooth and convert the odd and even LDSC matrices to correlation matrices

Ssmooth_odd&lt;-as.matrix((nearPD(LDSCoutput_odd$S, corr = FALSE))$mat)
corr_matrix_odd &lt;- cov2cor(Ssmooth_odd)

Ssmooth_even&lt;-as.matrix((nearPD(LDSCoutput_even$S, corr = FALSE))$mat)
corr_matrix_even &lt;- cov2cor(Ssmooth_even)

### Make combined LDSC matrix of odd and even chromosomes 

pdf(file = "LDSC_matrix_odd_even.pdf", height = 5, width = 5.2)
corrplot(corr_matrix_odd, method = "color", col = col(200),
         addCoef.col = "black", number.cex = .6, number.digits = 2, 
         type = "lower",  order = "original",
         tl.pos = "lt", tl.col = "black", tl.srt = 90, tl.cex = .7, 
         cl.cex = .7, cl.ratio = 0.1, 
         diag = TRUE)
corrplot(corr_matrix_even, add = TRUE, method = "color", col = col(200),
         addCoef.col = "black", number.cex = .6, number.digits = 2, 
         type = "upper",  order = "original",
         diag = FALSE, tl.pos = "n", cl.pos = "n")
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Smooth and convert the odd and even LDSC matrices to correlation matrices</span>

<span class="pl-smi">Ssmooth_odd</span><span class="pl-k">&lt;-</span>as.matrix((nearPD(<span class="pl-smi">LDSCoutput_odd</span><span class="pl-k">$</span><span class="pl-smi">S</span>, <span class="pl-v">corr</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>))<span class="pl-k">$</span><span class="pl-smi">mat</span>)
<span class="pl-smi">corr_matrix_odd</span> <span class="pl-k">&lt;-</span> cov2cor(<span class="pl-smi">Ssmooth_odd</span>)

<span class="pl-smi">Ssmooth_even</span><span class="pl-k">&lt;-</span>as.matrix((nearPD(<span class="pl-smi">LDSCoutput_even</span><span class="pl-k">$</span><span class="pl-smi">S</span>, <span class="pl-v">corr</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>))<span class="pl-k">$</span><span class="pl-smi">mat</span>)
<span class="pl-smi">corr_matrix_even</span> <span class="pl-k">&lt;-</span> cov2cor(<span class="pl-smi">Ssmooth_even</span>)

<span class="pl-c"><span class="pl-c">#</span>## Make combined LDSC matrix of odd and even chromosomes </span>

pdf(<span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>LDSC_matrix_odd_even.pdf<span class="pl-pds">"</span></span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">5</span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">5.2</span>)
corrplot(<span class="pl-smi">corr_matrix_odd</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>color<span class="pl-pds">"</span></span>, <span class="pl-v">col</span> <span class="pl-k">=</span> col(<span class="pl-c1">200</span>),
         <span class="pl-v">addCoef.col</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-v">number.cex</span> <span class="pl-k">=</span> <span class="pl-smi">.6</span>, <span class="pl-v">number.digits</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, 
         <span class="pl-v">type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>lower<span class="pl-pds">"</span></span>,  <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>original<span class="pl-pds">"</span></span>,
         <span class="pl-v">tl.pos</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>lt<span class="pl-pds">"</span></span>, <span class="pl-v">tl.col</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-v">tl.srt</span> <span class="pl-k">=</span> <span class="pl-c1">90</span>, <span class="pl-v">tl.cex</span> <span class="pl-k">=</span> <span class="pl-smi">.7</span>, 
         <span class="pl-v">cl.cex</span> <span class="pl-k">=</span> <span class="pl-smi">.7</span>, <span class="pl-v">cl.ratio</span> <span class="pl-k">=</span> <span class="pl-c1">0.1</span>, 
         <span class="pl-v">diag</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
corrplot(<span class="pl-smi">corr_matrix_even</span>, <span class="pl-v">add</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>color<span class="pl-pds">"</span></span>, <span class="pl-v">col</span> <span class="pl-k">=</span> col(<span class="pl-c1">200</span>),
         <span class="pl-v">addCoef.col</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-v">number.cex</span> <span class="pl-k">=</span> <span class="pl-smi">.6</span>, <span class="pl-v">number.digits</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, 
         <span class="pl-v">type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>upper<span class="pl-pds">"</span></span>,  <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>original<span class="pl-pds">"</span></span>,
         <span class="pl-v">diag</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">tl.pos</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>n<span class="pl-pds">"</span></span>, <span class="pl-v">cl.pos</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>n<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     Output of the genetic correlation matrix with the odd autosome results
on the lower triangle and the even autosome results on the upper
triangle:
                    </p>
                    <p align="center" dir="auto">
                     <a href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/blob/main/images/odd_even_ldsc_matrix.jpg" rel="noopener noreferrer" target="_blank">
                      <img src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/raw/main/images/odd_even_ldsc_matrix.jpg" style="max-width: 100%;" width="500"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      EFA comparison across chromosomal groupings
                     </h2>
                     <a aria-label="Permalink: EFA comparison across chromosomal groupings" class="anchor" href="#efa-comparison-across-chromosomal-groupings" id="user-content-efa-comparison-across-chromosomal-groupings">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Since the correlation coefficients were substantially different for AD
between the odd and even chromosomes, we also used EFA to test the
loadings for a 3-factor model based on the even autosomal data and all
autosomes to compare the factor loadings results with our initial
results based on the odd autosomes.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# 3-factor model even autosomes

EFA_even&lt;-factanal(covmat = Ssmooth_even, 
                   factors = 3, rotation = "promax")

print(EFA_even, sort=TRUE)
print(EFA_even,digits=3,cutoff=.20,sort=TRUE)

# 3-factor model all autosomes

Ssmooth_all&lt;-as.matrix((nearPD(LDSCoutput_all$S, corr = FALSE))$mat)
EFA_all&lt;-factanal(covmat = Ssmooth_all, 
                   factors = 3, rotation = "promax")

print(EFA_all, sort=TRUE)
print(EFA_all,digits=3,cutoff=.20,sort=TRUE)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> 3-factor model even autosomes</span>

<span class="pl-smi">EFA_even</span><span class="pl-k">&lt;-</span>factanal(<span class="pl-v">covmat</span> <span class="pl-k">=</span> <span class="pl-smi">Ssmooth_even</span>, 
                   <span class="pl-v">factors</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">rotation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>promax<span class="pl-pds">"</span></span>)

print(<span class="pl-smi">EFA_even</span>, <span class="pl-v">sort</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
print(<span class="pl-smi">EFA_even</span>,<span class="pl-v">digits</span><span class="pl-k">=</span><span class="pl-c1">3</span>,<span class="pl-v">cutoff</span><span class="pl-k">=</span><span class="pl-smi">.20</span>,<span class="pl-v">sort</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)

<span class="pl-c"><span class="pl-c">#</span> 3-factor model all autosomes</span>

<span class="pl-smi">Ssmooth_all</span><span class="pl-k">&lt;-</span>as.matrix((nearPD(<span class="pl-smi">LDSCoutput_all</span><span class="pl-k">$</span><span class="pl-smi">S</span>, <span class="pl-v">corr</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>))<span class="pl-k">$</span><span class="pl-smi">mat</span>)
<span class="pl-smi">EFA_all</span><span class="pl-k">&lt;-</span>factanal(<span class="pl-v">covmat</span> <span class="pl-k">=</span> <span class="pl-smi">Ssmooth_all</span>, 
                   <span class="pl-v">factors</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>, <span class="pl-v">rotation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>promax<span class="pl-pds">"</span></span>)

print(<span class="pl-smi">EFA_all</span>, <span class="pl-v">sort</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
print(<span class="pl-smi">EFA_all</span>,<span class="pl-v">digits</span><span class="pl-k">=</span><span class="pl-c1">3</span>,<span class="pl-v">cutoff</span><span class="pl-k">=</span><span class="pl-smi">.20</span>,<span class="pl-v">sort</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      CFA comparison across chromosomal groupings
                     </h2>
                     <a aria-label="Permalink: CFA comparison across chromosomal groupings" class="anchor" href="#cfa-comparison-across-chromosomal-groupings" id="user-content-cfa-comparison-across-chromosomal-groupings">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We also ran post-hoc sensitivity CFA analyses using the outputs from the
sensitivity EFA to compare the model fit and loadings of the best
fitting models based on the even autosome and all autosome EFA to see
how they compared to our main analysis.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# CFA models of even autosome EFA run in the odd autosomes

# Model 1: CFA of 3-factor model: including traits with &gt;.20 loadings and cross-loadings (DWLS method),
# negative loadings excluded; factor correlations present

CFAofEFA_posthoc1 &lt;- 'F1 =~ NA*LPA + SMK + DEP
F2 =~ NA*LSA + LED + LPA + ALC + DEP + BMI + T2DM + SBP
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1
SMK ~~  a*SMK
a &gt; .001'

CFAoutput_posthoc1 &lt;- usermodel(LDSCoutput_odd, estimation = &quot;DWLS&quot;, model = CFAofEFA_posthoc1, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput_posthoc1

# Model 2: CFA of 3-factor model: including traits with &gt;.20 loadings and no cross-loadings (DWLS method),
# negative loadings excluded; factor correlations present

CFAofEFA_posthoc2 &lt;- 'F1 =~ NA*SMK + DEP
F2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1
SMK ~~  a*SMK
a &gt; .001'

CFAoutput_posthoc2 &lt;- usermodel(LDSCoutput_odd, estimation = &quot;DWLS&quot;, model = CFAofEFA_posthoc2, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput_posthoc2


# Model 3: CFA of 3-factor model: including traits with &gt;.25 positive loadings and cross-loadings (DWLS method),
# no negative loadings; factor correlations present (best fitting parameters for odd autosomes) - had to remove cross loading for deprivation status for it to converge

CFAofEFA_posthoc3 &lt;- 'F1 =~ NA*SMK + DEP
F2 =~ NA*LPA + LSA + LED + ALC + BMI + SBP + T2DM
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput_posthoc3 &lt;- usermodel(LDSCoutput_odd, estimation = &quot;DWLS&quot;, model = CFAofEFA_posthoc3, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput_posthoc3

## CFA conducted in all autosomes but based off the parameters from the even EFA (same as the one tested in odd autosomes) - sensitivity analysis

# Model 4: CFA of 3-factor model: including traits with &gt;.20 loadings and cross-loadings (DWLS method),
# negative loadings excluded; factor correlations present

CFAofEFA_posthoc4 &lt;- 'F1 =~ NA*LPA + SMK + DEP
F2 =~ NA*LSA + LED + LPA + ALC + DEP + BMI + T2DM + SBP
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1
SMK ~~  a*SMK
a &gt; .001'

CFAoutput_posthoc4 &lt;- usermodel(LDSCoutput_all, estimation = &quot;DWLS&quot;, model = CFAofEFA_posthoc4, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput_posthoc4

# Model 5: CFA of 3-factor model: including traits with &gt;.20 loadings and no cross-loadings (DWLS method),
# negative loadings excluded; factor correlations present

CFAofEFA_posthoc5 &lt;- 'F1 =~ NA*SMK + DEP
F2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1
SMK ~~  a*SMK
a &gt; .001'

CFAoutput_posthoc5 &lt;- usermodel(LDSCoutput_all, estimation = &quot;DWLS&quot;, model = CFAofEFA_posthoc5, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput_posthoc5


# Model 6: CFA of 3-factor model: including traits with &gt;.25 positive loadings and cross-loadings (DWLS method),
# no negative loadings; factor correlations present (best fitting parameters for odd autosomes)

CFAofEFA_posthoc6 &lt;- 'F1 =~ NA*SMK + DEP
F2 =~ NA*LPA + LSA + LED + ALC + BMI + SBP + T2DM + DEP
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput_posthoc6 &lt;- usermodel(LDSCoutput_all, estimation = &quot;DWLS&quot;, model = CFAofEFA_posthoc6, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput_posthoc6

# CFA models of all autosome EFA run in all autosomes

# Model 7: CFA of 3-factor model: including traits with &gt;.25 positive loadings and cross-loadings (DWLS method),
# no negative loadings; factor correlations present (KEPT INSOMNIA IN AS .24)

CFAofEFA_posthoc7 &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED
F2 =~ NA*LPA + LSA + LED + ALC + BMI + T2DM + SBP
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput_posthoc7 &lt;- usermodel(LDSCoutput_all, estimation = &quot;DWLS&quot;, model = CFAofEFA_posthoc7, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput_posthoc7

# Model 8: CFA of 3-factor model: including traits with &gt;.20 loadings and cross-loadings (DWLS method),
# negative loadings excluded; factor correlations present

CFAofEFA_posthoc8 &lt;- 'F1 =~ NA*LPA + SMK + DEP + LED
F2 =~ NA*LSA + LED + LPA + ALC + INS + BMI + T2DM + SBP
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput_posthoc8 &lt;- usermodel(LDSCoutput_all, estimation = &quot;DWLS&quot;, model = CFAofEFA_posthoc8, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput_posthoc8

# Model 9: CFA of 3-factor model: including traits with &gt;.20 loadings and no cross-loadings (DWLS method),
# negative loadings excluded; factor correlations present

CFAofEFA_posthoc9 &lt;- 'F1 =~ NA*SMK + DEP
F2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP
F3 =~ NA*MDD + LON + HD + INS
F1 ~~ F2
F2 ~~ F3
F3 ~~ F1'

CFAoutput_posthoc9 &lt;- usermodel(LDSCoutput_all, estimation = &quot;DWLS&quot;, model = CFAofEFA_posthoc9, CFIcalc = TRUE, std.lv = TRUE, imp_cov = FALSE)
CFAoutput_posthoc9" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> CFA models of even autosome EFA run in the odd autosomes</span>

<span class="pl-c"><span class="pl-c">#</span> Model 1: CFA of 3-factor model: including traits with &gt;.20 loadings and cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> negative loadings excluded; factor correlations present</span>

<span class="pl-smi">CFAofEFA_posthoc1</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP</span>
<span class="pl-s">F2 =~ NA*LSA + LED + LPA + ALC + DEP + BMI + T2DM + SBP</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1</span>
<span class="pl-s">SMK ~~  a*SMK</span>
<span class="pl-s">a &gt; .001<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput_posthoc1</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_odd</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA_posthoc1</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput_posthoc1</span>

<span class="pl-c"><span class="pl-c">#</span> Model 2: CFA of 3-factor model: including traits with &gt;.20 loadings and no cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> negative loadings excluded; factor correlations present</span>

<span class="pl-smi">CFAofEFA_posthoc2</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*SMK + DEP</span>
<span class="pl-s">F2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1</span>
<span class="pl-s">SMK ~~  a*SMK</span>
<span class="pl-s">a &gt; .001<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput_posthoc2</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_odd</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA_posthoc2</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput_posthoc2</span>


<span class="pl-c"><span class="pl-c">#</span> Model 3: CFA of 3-factor model: including traits with &gt;.25 positive loadings and cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> no negative loadings; factor correlations present (best fitting parameters for odd autosomes) - had to remove cross loading for deprivation status for it to converge</span>

<span class="pl-smi">CFAofEFA_posthoc3</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*SMK + DEP</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + BMI + SBP + T2DM</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput_posthoc3</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_odd</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA_posthoc3</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput_posthoc3</span>

<span class="pl-c"><span class="pl-c">#</span># CFA conducted in all autosomes but based off the parameters from the even EFA (same as the one tested in odd autosomes) - sensitivity analysis</span>

<span class="pl-c"><span class="pl-c">#</span> Model 4: CFA of 3-factor model: including traits with &gt;.20 loadings and cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> negative loadings excluded; factor correlations present</span>

<span class="pl-smi">CFAofEFA_posthoc4</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP</span>
<span class="pl-s">F2 =~ NA*LSA + LED + LPA + ALC + DEP + BMI + T2DM + SBP</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1</span>
<span class="pl-s">SMK ~~  a*SMK</span>
<span class="pl-s">a &gt; .001<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput_posthoc4</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_all</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA_posthoc4</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput_posthoc4</span>

<span class="pl-c"><span class="pl-c">#</span> Model 5: CFA of 3-factor model: including traits with &gt;.20 loadings and no cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> negative loadings excluded; factor correlations present</span>

<span class="pl-smi">CFAofEFA_posthoc5</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*SMK + DEP</span>
<span class="pl-s">F2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1</span>
<span class="pl-s">SMK ~~  a*SMK</span>
<span class="pl-s">a &gt; .001<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput_posthoc5</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_all</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA_posthoc5</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput_posthoc5</span>


<span class="pl-c"><span class="pl-c">#</span> Model 6: CFA of 3-factor model: including traits with &gt;.25 positive loadings and cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> no negative loadings; factor correlations present (best fitting parameters for odd autosomes)</span>

<span class="pl-smi">CFAofEFA_posthoc6</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*SMK + DEP</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + BMI + SBP + T2DM + DEP</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput_posthoc6</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_all</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA_posthoc6</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput_posthoc6</span>

<span class="pl-c"><span class="pl-c">#</span> CFA models of all autosome EFA run in all autosomes</span>

<span class="pl-c"><span class="pl-c">#</span> Model 7: CFA of 3-factor model: including traits with &gt;.25 positive loadings and cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> no negative loadings; factor correlations present (KEPT INSOMNIA IN AS .24)</span>

<span class="pl-smi">CFAofEFA_posthoc7</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED</span>
<span class="pl-s">F2 =~ NA*LPA + LSA + LED + ALC + BMI + T2DM + SBP</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput_posthoc7</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_all</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA_posthoc7</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput_posthoc7</span>

<span class="pl-c"><span class="pl-c">#</span> Model 8: CFA of 3-factor model: including traits with &gt;.20 loadings and cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> negative loadings excluded; factor correlations present</span>

<span class="pl-smi">CFAofEFA_posthoc8</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*LPA + SMK + DEP + LED</span>
<span class="pl-s">F2 =~ NA*LSA + LED + LPA + ALC + INS + BMI + T2DM + SBP</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput_posthoc8</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_all</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA_posthoc8</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput_posthoc8</span>

<span class="pl-c"><span class="pl-c">#</span> Model 9: CFA of 3-factor model: including traits with &gt;.20 loadings and no cross-loadings (DWLS method),</span>
<span class="pl-c"><span class="pl-c">#</span> negative loadings excluded; factor correlations present</span>

<span class="pl-smi">CFAofEFA_posthoc9</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>F1 =~ NA*SMK + DEP</span>
<span class="pl-s">F2 =~ NA*LSA + LED + LPA + ALC + BMI + T2DM + SBP</span>
<span class="pl-s">F3 =~ NA*MDD + LON + HD + INS</span>
<span class="pl-s">F1 ~~ F2</span>
<span class="pl-s">F2 ~~ F3</span>
<span class="pl-s">F3 ~~ F1<span class="pl-pds">'</span></span>

<span class="pl-smi">CFAoutput_posthoc9</span> <span class="pl-k">&lt;-</span> usermodel(<span class="pl-smi">LDSCoutput_all</span>, <span class="pl-v">estimation</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DWLS<span class="pl-pds">"</span></span>, <span class="pl-v">model</span> <span class="pl-k">=</span> <span class="pl-smi">CFAofEFA_posthoc9</span>, <span class="pl-v">CFIcalc</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">std.lv</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">imp_cov</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">CFAoutput_posthoc9</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Final remarks
                     </h1>
                     <a aria-label="Permalink: Final remarks" class="anchor" href="#final-remarks" id="user-content-final-remarks">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We hope that you find this Markdown useful. If you use this code in your
own analyses please cite this Github page. If you run into any issues or
queries related to this work please dont hesitate to contact Isabelle
Foote (
                     <a href="mailto:isabelle.foote@colorado.edu">
                      isabelle.foote@colorado.edu
                     </a>
                     ).
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="PQ0ZcQ1YGLcBgsJbdKIGiCAt14yQiijjwdL7qW/1c1QLKAOSX1AodGqbOpWbpvnjG0/nyvfV1xAL4UB8KMlGAQ==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 A code walkthrough of the analyses performed for our paper on the shared genetic architecture of modifiable risk for Alzheimer's disease using genomic SEM.
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  star
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2FIsyFoote%2FGenomic-Factor-Analysis-of-modifiable-AD-risk_NBA22&amp;report=IsyFoote+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/IsyFoote/packages?repo_name=Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="R 75.2" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#198CE7 !important;;width: 75.2%;">
                 </span>
                 <span aria-label="Shell 24.8" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#89e051 !important;;width: 24.8%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/search?l=r">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#198CE7;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   R
                  </span>
                  <span>
                   75.2%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/IsyFoote/Genomic-Factor-Analysis-of-modifiable-AD-risk_NBA22/search?l=shell">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#89e051;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Shell
                  </span>
                  <span>
                   24.8%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
        2024 GitHub,Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You cant perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
