<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-ca8ae01e33b4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-fb4b8d40f206.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - Jihong-Tang/Cancer-genomics-analysis: Notes for cancer genomics analysis pipeline and skills
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:BF7CFC:1196DF0:66059E08" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDpCRjdDRkM6MTE5NkRGMDo2NjA1OUUwOCIsInZpc2l0b3JfaWQiOiI2NDUyMjQxODU2MjM4ODE3ODA3IiwicmVnaW9uX2VkZ2UiOiJpYWQiLCJyZWdpb25fcmVuZGVyIjoiaWFkIn0=" data-pjax-transient="true" name="visitor-payload">
           <meta content="139f0d0607cbec8b22faae6135ddda02f149693da849f31a2231286df9212f84" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:408758210" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="Notes for cancer genomics analysis pipeline and skills - Jihong-Tang/Cancer-genomics-analysis" name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/Jihong-Tang/Cancer-genomics-analysis" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/0d0f10f56e7447b8d9d2e5ec2bbf4f88734c31c1e49f20554a04a439e1461a34/Jihong-Tang/Cancer-genomics-analysis" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - Jihong-Tang/Cancer-genomics-analysis: Notes for cancer genomics analysis pipeline and skills" name="twitter:title">
                    <meta content="Notes for cancer genomics analysis pipeline and skills - Jihong-Tang/Cancer-genomics-analysis" name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/0d0f10f56e7447b8d9d2e5ec2bbf4f88734c31c1e49f20554a04a439e1461a34/Jihong-Tang/Cancer-genomics-analysis" property="og:image">
                      <meta content="Notes for cancer genomics analysis pipeline and skills - Jihong-Tang/Cancer-genomics-analysis" property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - Jihong-Tang/Cancer-genomics-analysis: Notes for cancer genomics analysis pipeline and skills" property="og:title"/>
                        <meta content="https://github.com/Jihong-Tang/Cancer-genomics-analysis" property="og:url"/>
                        <meta content="Notes for cancer genomics analysis pipeline and skills - Jihong-Tang/Cancer-genomics-analysis" property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="584826db69b6459b3c4c29172349cd9ed6729235b962a93107214b2469aa5b40" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="a2a6f0b866a9de2e69b1f990e0730e22d502d619ef6d55aae603cacc6e2f5285" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/Jihong-Tang/Cancer-genomics-analysis git https://github.com/Jihong-Tang/Cancer-genomics-analysis.git" name="go-import"/>
                        <meta content="28826794" name="octolytics-dimension-user_id">
                         <meta content="Jihong-Tang" name="octolytics-dimension-user_login">
                          <meta content="408758210" name="octolytics-dimension-repository_id">
                           <meta content="Jihong-Tang/Cancer-genomics-analysis" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="408758210" name="octolytics-dimension-repository_network_root_id">
                               <meta content="Jihong-Tang/Cancer-genomics-analysis" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/Jihong-Tang/Cancer-genomics-analysis" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-11f6759e1cef.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/Jihong-Tang/Cancer-genomics-analysis","user_id":null}}' data-hydro-click-hmac="aa8390fe2db6262ed60c747f6b8542b51ec6e4457cca4428402dd109c4f4d5a5" href="/login?return_to=https%3A%2F%2Fgithub.com%2FJihong-Tang%2FCancer-genomics-analysis">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=jp4J3D6iIMuwbTF96H61nWZkYjEM7Gi%2FDPoSXlzThQA8N7NDmuPgrK%2B7gsnuE%2FeAAgVQDxL2s2yrdJmAJ7twMPKGPZ%2Brr%2F72%2FBbJLaj0DKPdZF9nE8xaA2bQo2KUP%2F5MCoMZHWwCQUGMO1LQCr0Rvr44AlhobwdDUMXzwwrpJsMQ4H24y0nZzJbvSQJrvjTo2bTCFlMi1%2Ba7Kg0u3XUfGcckVP6rUK5L7aNOtiUiEhfCx9Zas%2B3fSYktJL24ZJQehy39DtXsNGsxBdzisecxNWPvVlw6%2FrUIHjmNwKSnd80tIYenWkuNFsFg%2FCvEnQkPJf3OfsbhFJxc54lIU5bezk7SwLmnxdSFPywm5WDrvshO6XO9p279IU2ZEAZb3hgVxS9Iu01Gd%2FcBYhsEfaKckXdd2wOsJZzSKGS8TcxHbXgKqwMMOjGBiGfAZY5P9xuPTWVWH%2FjfLt5XeLqfzSQpf4RikmHozkrNsLHPzG0Uk2GOHPl2v1wKGv1NF4A70kwU%2B7vSQwHXIVu6xBCrACJBzC%2F1cQglLgDg0U%2Blrqe6LgFjBX3FihE%3D--UytdXV%2BbB%2Br2Huyb--bg0RNUvyNACPxIzaRvH1Lw%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="Jihong-Tang" data-current-repository="Jihong-Tang/Cancer-genomics-analysis" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="ZdgIReXV3tsgd_FyvhizEdc4SojRRqxmOPl_uaw2ef9-NbGpKFxG_tgiJ6op1P9TALx4x6GYVsf3sEcABBJm_A" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:Jihong-Tang/Cancer-genomics-analysis">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-95b7d8f2-49bc-4bb0-a9cf-f1ce3943116d" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-95b7d8f2-49bc-4bb0-a9cf-f1ce3943116d">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="eQf7aB2FivxQsPviYgcdi2bm6gkTrTINKlqrC9j0jUjHw+6eYoqshxqxmpKz48ST11oxM5S2ct1jE0+OC4rkeA=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="6QDTf2K5kA7MbGCaT4Sz88xiP+4gogHthCXOcajpuYBjh30jYpMt2YDUT3F2Ue0rCQrc4CIaUxGjg7sqxnoSbQ==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="YaQqhKRCCioSKB1eQR+YKuadYkqMaTDI/3u8YepM/fMMaoLYhrD5SJjFbaTnTU4kMsalMQFXAkIvV97Glwm/kA==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="PEKYAjYkhmZLMslrjUrnKCywl0/KlhrsoKnWgnFsYKfKX11kyre2GI6fZ/I/ppMAmvAfZmILlae+9W24uhDMtw==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/Jihong-Tang/Cancer-genomics-analysis","user_id":null}}' data-hydro-click-hmac="aa8390fe2db6262ed60c747f6b8542b51ec6e4457cca4428402dd109c4f4d5a5" href="/login?return_to=https%3A%2F%2Fgithub.com%2FJihong-Tang%2FCancer-genomics-analysis">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/Jihong-Tang/Cancer-genomics-analysis","user_id":null}}' data-hydro-click-hmac="aa8390fe2db6262ed60c747f6b8542b51ec6e4457cca4428402dd109c4f4d5a5" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=Jihong-Tang%2FCancer-genomics-analysis">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-ba4d9ea5-80cd-4ac7-835c-948edea4842a" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-bbb16e5d-4e4b-4a4d-83ac-4041919044fd" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-bbb16e5d-4e4b-4a4d-83ac-4041919044fd" id="tooltip-ba4d9ea5-80cd-4ac7-835c-948edea4842a" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/Jihong-Tang/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/Jihong-Tang" rel="author">
            Jihong-Tang
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/Jihong-Tang/Cancer-genomics-analysis">
            Cancer-genomics-analysis
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/Jihong-Tang/Cancer-genomics-analysis","user_id":null}}' data-hydro-click-hmac="453991c8ba0de6c257a0a3b0a0d424041fca790191fe72a9e3ab28474f19d974" data-view-component="true" href="/login?return_to=%2FJihong-Tang%2FCancer-genomics-analysis" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":408758210,"auth_type":"LOG_IN","originating_url":"https://github.com/Jihong-Tang/Cancer-genomics-analysis","user_id":null}}' data-hydro-click-hmac="9e095da1cc4c22c46ee015634ce2e818cbba65596f2d75723092451ac9431bb7" data-view-component="true" href="/login?return_to=%2FJihong-Tang%2FCancer-genomics-analysis" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
             0
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":408758210,"auth_type":"LOG_IN","originating_url":"https://github.com/Jihong-Tang/Cancer-genomics-analysis","user_id":null}}' data-hydro-click-hmac="db0e6244979ca18af06ad9800e8eb2c0714d2d7792f2c6d6550ca3e5eae27d3d" data-view-component="true" href="/login?return_to=%2FJihong-Tang%2FCancer-genomics-analysis" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
              0
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <p class="f4 mb-3">
          Notes for cancer genomics analysis pipeline and skills
         </p>
         <h3 class="sr-only">
          License
         </h3>
         <div class="mb-2">
          <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/LICENSE">
           <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
            </path>
           </svg>
           GPL-3.0 license
          </a>
         </div>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/Jihong-Tang/Cancer-genomics-analysis/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           stars
          </a>
          <a class="Link--secondary no-underline mr-3" href="/Jihong-Tang/Cancer-genomics-analysis/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/Jihong-Tang/Cancer-genomics-analysis/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/Jihong-Tang/Cancer-genomics-analysis/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/Jihong-Tang/Cancer-genomics-analysis/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":408758210,"auth_type":"LOG_IN","originating_url":"https://github.com/Jihong-Tang/Cancer-genomics-analysis","user_id":null}}' data-hydro-click-hmac="db0e6244979ca18af06ad9800e8eb2c0714d2d7792f2c6d6550ca3e5eae27d3d" data-view-component="true" href="/login?return_to=%2FJihong-Tang%2FCancer-genomics-analysis" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/Jihong-Tang/Cancer-genomics-analysis","user_id":null}}' data-hydro-click-hmac="453991c8ba0de6c257a0a3b0a0d424041fca790191fe72a9e3ab28474f19d974" data-view-component="true" href="/login?return_to=%2FJihong-Tang%2FCancer-genomics-analysis" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /Jihong-Tang/Cancer-genomics-analysis" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /Jihong-Tang/Cancer-genomics-analysis/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /Jihong-Tang/Cancer-genomics-analysis/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /Jihong-Tang/Cancer-genomics-analysis/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /Jihong-Tang/Cancer-genomics-analysis/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /Jihong-Tang/Cancer-genomics-analysis/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/Jihong-Tang/Cancer-genomics-analysis/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /Jihong-Tang/Cancer-genomics-analysis/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-03e8f835-14be-434b-a914-63912979aee7-list" aria-haspopup="true" aria-labelledby="tooltip-c9d8d690-39bc-4e35-886d-f4014a3e0aab" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-03e8f835-14be-434b-a914-63912979aee7-button" popovertarget="action-menu-03e8f835-14be-434b-a914-63912979aee7-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-03e8f835-14be-434b-a914-63912979aee7-button" id="tooltip-c9d8d690-39bc-4e35-886d-f4014a3e0aab" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-03e8f835-14be-434b-a914-63912979aee7-button" anchor-offset="normal" data-view-component="true" id="action-menu-03e8f835-14be-434b-a914-63912979aee7-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-03e8f835-14be-434b-a914-63912979aee7-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-03e8f835-14be-434b-a914-63912979aee7-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis" id="item-6c1462a7-97b7-4f66-b283-bf3f8ad0da95" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/issues" id="item-dc1518d0-d97a-46ac-92f1-2a581de42942" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/pulls" id="item-c6ac9fd1-a897-4a3d-aa85-6654d1ea19f7" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/actions" id="item-9ef36fa4-d51b-4bb0-95af-a43f318e22c2" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/projects" id="item-f95b5536-8fa8-433c-a49c-c4ce60db710b" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/security" id="item-b999ed11-8d6d-4e2c-af16-5b7acd9713fd" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/pulse" id="item-e6d2e062-ad06-45d9-8f84-8b6f70f84c67" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         Jihong-Tang/Cancer-genomics-analysis
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/Jihong-Tang/Cancer-genomics-analysis/spoofed_commit_check/81b4a533adce21e1a8a17d52689f7629e35dbb9c">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":408758210,"defaultBranch":"main","name":"Cancer-genomics-analysis","ownerLogin":"Jihong-Tang","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-09-21T09:23:11.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/28826794?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"main","listCacheKey":"v0:1632216191.902236","canEdit":false,"refType":"branch","currentOid":"81b4a533adce21e1a8a17d52689f7629e35dbb9c"},"tree":{"items":[{"name":"scripts","path":"scripts","contentType":"directory"},{"name":".gitignore","path":".gitignore","contentType":"file"},{"name":"LICENSE","path":"LICENSE","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"},{"name":"cnvkit_workflow.webp","path":"cnvkit_workflow.webp","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":5,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/Jihong-Tang/Cancer-genomics-analysis/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/Jihong-Tang/Cancer-genomics-analysis.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone Jihong-Tang/Cancer-genomics-analysis","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/Jihong-Tang/Cancer-genomics-analysis/archive/refs/heads/main.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=408758210"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"17","overviewFiles":[{"displayName":"README.md","repoName":"Cancer-genomics-analysis","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTable of Contents\u003c/h1\u003e\u003ca id=\"user-content-table-of-contents\" class=\"anchor\" aria-label=\"Permalink: Table of Contents\" href=\"#table-of-contents\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#table-of-contents\"\u003eTable of Contents\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#name\"\u003eName\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#purpose\"\u003ePurpose\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#part0-introduction-and-preparation\"\u003ePart0 Introduction and Preparation\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#01-introduction\"\u003e0.1 Introduction\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#02-getting-setup\"\u003e0.2 Getting Setup\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#021-installing-conda-environment\"\u003e0.2.1 Installing Conda Environment\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#022-installing-homebrew-alternative-choice\"\u003e0.2.2 Installing Homebrew (Alternative choice)\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#023-setting-up-the-folder-structure\"\u003e0.2.3 Setting Up the Folder Structure\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#024-creating-cohort-index-files\"\u003e0.2.4 Creating Cohort Index Files\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#part1-sequencing-data-quality-control\"\u003ePart1 Sequencing Data Quality Control\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#11-setting-up-tools\"\u003e1.1 Setting Up Tools\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#12-fastq-files-preparation\"\u003e1.2 FastQ Files Preparation\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#13-automatic-qc-with-fastp\"\u003e1.3 Automatic QC with fastp\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#131-working-scripts\"\u003e1.3.1 Working Scripts\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#132-scripts-usage-examples\"\u003e1.3.2 Scripts Usage Examples\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#part2-genome-alignment\"\u003ePart2 Genome Alignment\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#21-setting-up-tools\"\u003e2.1 Setting Up Tools\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#22-genome-alignment-with-bwa\"\u003e2.2 Genome Alignment with bwa\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#221-working-scripts\"\u003e2.2.1 Working Scripts\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#222-scripts-usage-example\"\u003e2.2.2 Scripts Usage Example\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#23-mark-duplicates-with-picard\"\u003e2.3 Mark Duplicates with Picard\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#231-working-scripts\"\u003e2.3.1 Working Scripts\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#232-scripts-usage-example\"\u003e2.3.2 Scripts Usage Example\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#part3-somatic-mutation-calling\"\u003ePart3 Somatic Mutation Calling\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#31-setting-up-tools\"\u003e3.1 Setting Up Tools\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#32-somatic-mutations-calling-with-savi\"\u003e3.2 Somatic Mutations Calling with SAVI\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#part4-copy-number-variation-analysis\"\u003ePart4 Copy Number Variation Analysis\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#41-preparation-of-input-files\"\u003e4.1 Preparation of input files\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#411-conda-environment\"\u003e4.1.1 Conda environment\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#412-access-file-for-reference-genome\"\u003e4.1.2 Access file for reference genome\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#413-baits-file\"\u003e4.1.3 Baits file\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#414-target-and-antitarget-bed-files\"\u003e4.1.4 Target and Antitarget bed files\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#42-cnvkit-commands\"\u003e4.2 CNVkit Commands\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#421-coverage\"\u003e4.2.1 Coverage\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#422-referencefixsegmentexport\"\u003e4.2.2 Reference;Fix;Segment;Export\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#43-cnv-profile-generation\"\u003e4.3 CNV Profile Generation\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#author\"\u003eAuthor\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#reference\"\u003eReference\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eName\u003c/h1\u003e\u003ca id=\"user-content-name\" class=\"anchor\" aria-label=\"Permalink: Name\" href=\"#name\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCancer Genomics Analysis Pipeline(CGAP): Notes for the computational pipeline focusing on the Cancer Genomics DNA-seq data\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePurpose\u003c/h1\u003e\u003ca id=\"user-content-purpose\" class=\"anchor\" aria-label=\"Permalink: Purpose\" href=\"#purpose\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eMake record for the in-house DNA-seq data analysis pipeline including several modules for future usage:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSequencing data Quality Control\u003c/li\u003e\n\u003cli\u003eGenome Alignment\u003c/li\u003e\n\u003cli\u003eSomatic Mutation Calling\u003c/li\u003e\n\u003cli\u003eCopy Number Variation Analysis\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eTake notes about cancer genomics data analysis for future review.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePart0 Introduction and Preparation\u003c/h1\u003e\u003ca id=\"user-content-part0-introduction-and-preparation\" class=\"anchor\" aria-label=\"Permalink: Part0 Introduction and Preparation\" href=\"#part0-introduction-and-preparation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e0.1 Introduction\u003c/h2\u003e\u003ca id=\"user-content-01-introduction\" class=\"anchor\" aria-label=\"Permalink: 0.1 Introduction\" href=\"#01-introduction\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe in house cancer genomics analysis pipeline starts from the quanlity control of DNA-seq sequencing data. After aligning the sequencing data to different versions of reference genomes, the pipeline identifies the somatic mutations and copy number variations based on the comparisons between normal/blood and tumor tissues.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe whole analysis pipeline is implemented across four main procedures based on different bioinformatics tools:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSequencing Data Quality Control - \u003ca href=\"https://github.com/OpenGene/fastp\"\u003efastp\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003eGenome Alignment - \u003ca href=\"https://github.com/lh3/bwa\"\u003ebwa\u003c/a\u003e and \u003ca href=\"https://broadinstitute.github.io/picard/\" rel=\"nofollow\"\u003ePicard\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003eSomatic Mutations Calling - \u003ca href=\"https://github.com/WangLabHKUST/SAVI\"\u003eSAVI\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003eCNV Analysis - \u003ca href=\"https://cnvkit.readthedocs.io/en/stable/\" rel=\"nofollow\"\u003eCNVkit\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThe pipeline is initially arranged for the whole exome sequencing(WES) data, but with minor updates on the scripts, it can also be used for the whole genome sequencing(WGS) data.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e0.2 Getting Setup\u003c/h2\u003e\u003ca id=\"user-content-02-getting-setup\" class=\"anchor\" aria-label=\"Permalink: 0.2 Getting Setup\" href=\"#02-getting-setup\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e0.2.1 Installing Conda Environment\u003c/h3\u003e\u003ca id=\"user-content-021-installing-conda-environment\" class=\"anchor\" aria-label=\"Permalink: 0.2.1 Installing Conda Environment\" href=\"#021-installing-conda-environment\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAnaconda/Miniconda is one powerful and user-friendly package manager for Python. Miniconda is the smaller version of Anaconda with basic functions.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUsing conda environment management tools, we can not only install and manage Python packages with different verisons, but also create virual environments and install various bioinformatics tools through access to large bioinformatics repositories(eg. \u003ca href=\"https://bioconda.github.io/\" rel=\"nofollow\"\u003eBioconda\u003c/a\u003e)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe methods to download and install conda can be found \u003ca href=\"https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html\" rel=\"nofollow\"\u003ehere\u003c/a\u003e. For example, we can use the following commands to install miniconda on MacOS machines.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# download the Miniconda3 installer to the home directory, only for MacOS\nwget https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh -O ~/miniconda.sh\n\n# Run the miniconda installation in slient mode\nbash ~/miniconda.sh -b -p $HOME/miniconda\n\n# Choose YES for the question 'Do you wish the installer to initialize Miniconda3 by running conda init?', \n# then the bash profile will be changed automatically.\n\n# Add bioinformatic channels for downloading required packages\nconda config --add channels conda-forge\nconda config --add channels defaults\nconda config --add channels r\nconda config --add channels bioconda # important channel to download and manage useful bioinformatics tools\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e download the Miniconda3 installer to the home directory, only for MacOS\u003c/span\u003e\nwget https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh -O \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/miniconda.sh\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Run the miniconda installation in slient mode\u003c/span\u003e\nbash \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/miniconda.sh -b -p \u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/miniconda\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Choose YES for the question 'Do you wish the installer to initialize Miniconda3 by running conda init?', \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e then the bash profile will be changed automatically.\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Add bioinformatic channels for downloading required packages\u003c/span\u003e\nconda config --add channels conda-forge\nconda config --add channels defaults\nconda config --add channels r\nconda config --add channels bioconda \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e important channel to download and manage useful bioinformatics tools\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e0.2.2 Installing Homebrew (Alternative choice)\u003c/h3\u003e\u003ca id=\"user-content-022-installing-homebrew-alternative-choice\" class=\"anchor\" aria-label=\"Permalink: 0.2.2 Installing Homebrew (Alternative choice)\" href=\"#022-installing-homebrew-alternative-choice\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHomebrew is another powerful package manager that can install \u003cem\u003ethe stuff you need\u003c/em\u003e that Apple (or your Linux system) didn't have. The most well-known example is \u003cem\u003ewget\u003c/em\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe methods to download and install brew can be found \u003ca href=\"https://brew.sh/\" rel=\"nofollow\"\u003ehere\u003c/a\u003e. For example, we can use the following commands to install brew on MacOS machines.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# download the Homebrew installer and install brew\n/bin/bash -c \u0026quot;$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)\u0026quot;\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e download the Homebrew installer and install brew\u003c/span\u003e\n/bin/bash -c \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ecurl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIn addition, we could install the brew on Linux machines as following, referred from \u003ca href=\"https://github.com/wang-q/ubuntu#install-linuxbrew\"\u003ewang-q Ubuntu\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"echo \u0026quot;==\u0026gt; Install linuxbrew, copy the next *ONE* line to terminal\u0026quot;\nbash -c \u0026quot;$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh)\u0026quot;\n\ntest -d ~/.linuxbrew \u0026amp;\u0026amp; PATH=\u0026quot;$HOME/.linuxbrew/bin:$HOME/.linuxbrew/sbin:$PATH\u0026quot;\ntest -d /home/linuxbrew/.linuxbrew \u0026amp;\u0026amp; PATH=\u0026quot;/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:$PATH\u0026quot;\n\nif grep -q -i linuxbrew $HOME/.bashrc; then\n    echo \u0026quot;==\u0026gt; .bashrc already contains linuxbrew\u0026quot;\nelse\n    echo \u0026quot;==\u0026gt; Update .bashrc\u0026quot;\n\n    echo \u0026gt;\u0026gt; $HOME/.bashrc\n    echo '# Linuxbrew' \u0026gt;\u0026gt; $HOME/.bashrc\n    echo \u0026quot;export PATH='$(brew --prefix)/bin:$(brew --prefix)/sbin'\u0026quot;:'\u0026quot;$PATH\u0026quot;' \u0026gt;\u0026gt; $HOME/.bashrc\n    echo \u0026quot;export MANPATH='$(brew --prefix)/share/man'\u0026quot;:'\u0026quot;$MANPATH\u0026quot;' \u0026gt;\u0026gt; $HOME/.bashrc\n    echo \u0026quot;export INFOPATH='$(brew --prefix)/share/info'\u0026quot;:'\u0026quot;$INFOPATH\u0026quot;' \u0026gt;\u0026gt; $HOME/.bashrc\n    echo \u0026quot;export HOMEBREW_NO_ANALYTICS=1\u0026quot; \u0026gt;\u0026gt; $HOME/.bashrc\n    echo \u0026quot;export HOMEBREW_NO_AUTO_UPDATE=1\u0026quot; \u0026gt;\u0026gt; $HOME/.bashrc\n    echo \u0026gt;\u0026gt; $HOME/.bashrc\nfi\n\nsource $HOME/.bashrc\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e==\u0026gt; Install linuxbrew, copy the next *ONE* line to terminal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nbash -c \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ecurl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e -d \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/.linuxbrew \u003cspan class=\"pl-k\"\u003e\u0026amp;\u0026amp;\u003c/span\u003e PATH=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.linuxbrew/bin:\u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.linuxbrew/sbin:\u003cspan class=\"pl-smi\"\u003e$PATH\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e -d /home/linuxbrew/.linuxbrew \u003cspan class=\"pl-k\"\u003e\u0026amp;\u0026amp;\u003c/span\u003e PATH=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:\u003cspan class=\"pl-smi\"\u003e$PATH\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e grep -q -i linuxbrew \u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.bashrc\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ethen\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e==\u0026gt; .bashrc already contains linuxbrew\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eelse\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e==\u0026gt; Update .bashrc\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.bashrc\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e# Linuxbrew\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.bashrc\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eexport PATH='\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ebrew --prefix\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e/bin:\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ebrew --prefix\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e/sbin'\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e:\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\"$PATH\"\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.bashrc\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eexport MANPATH='\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ebrew --prefix\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e/share/man'\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e:\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\"$MANPATH\"\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.bashrc\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eexport INFOPATH='\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ebrew --prefix\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e/share/info'\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e:\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\"$INFOPATH\"\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.bashrc\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eexport HOMEBREW_NO_ANALYTICS=1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.bashrc\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eexport HOMEBREW_NO_AUTO_UPDATE=1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.bashrc\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.bashrc\n\u003cspan class=\"pl-k\"\u003efi\u003c/span\u003e\n\n\u003cspan class=\"pl-c1\"\u003esource\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HOME\u003c/span\u003e/.bashrc\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMost of the used softwares and tools can be downloaded through both package managers, and it is highly recommended to use either of them to manage the related softwares and bioinformatics tools.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e0.2.3 Setting Up the Folder Structure\u003c/h3\u003e\u003ca id=\"user-content-023-setting-up-the-folder-structure\" class=\"anchor\" aria-label=\"Permalink: 0.2.3 Setting Up the Folder Structure\" href=\"#023-setting-up-the-folder-structure\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOrganizing is essential to proper reproducible research. During the processing and analysis steps, we will get many output files. A simple and clear folder structure is preferred not only for your own future usage, but also for other researchers' repeats.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe folder structure used in this pipeline will be summarized here.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"── ${project_WES}/\n  │   └── reference/                \u0026lt;- Reference genome file (.FASTA)\n  │       ├── hg19bwa/              \u0026lt;- hg19 version reference genome files\n  │       ├── TCGA_GRCh38_bwa_ref/  \u0026lt;- hg38 version reference genome files\n  │ \n  │   └── all_fq/                   \u0026lt;- Working Directory of Sequencing Data QC \n  │       ├── log/                  \u0026lt;- Log from Sequencing Data QC module\n  │       ├── qc_res/               \u0026lt;- QC result from Sequencing Data QC module\n  │ \n  │   └── all_mapping_hg19/         \u0026lt;- Working Directory of Genome Alignment: hg19 version \n  │       ├── log/                  \u0026lt;- Log from Genome Alignment module\n  │       ├── md_metrix/            \u0026lt;- MarkDuplicate metrix from alignment duplicates marking step\n  │       ├── md_tmp/               \u0026lt;- Location for temporary files from alignment duplicates marking step\n  │       ├── tmp/                  \u0026lt;- Location for temporary files from Genome Alignment module\n  │ \n  │   └── all_mapping_hg38/         \u0026lt;- Working Directory of Genome Alignment: hg38 version \n  │       ├── log/                  \u0026lt;- Log from Genome Alignment module\n  │       ├── md_metrix/            \u0026lt;- MarkDuplicate metrix from alignment duplicates marking step\n  │       ├── md_tmp/               \u0026lt;- Location for temporary files from alignment duplicates marking step\n  │       ├── tmp/                  \u0026lt;- Location for temporary files from Genome Alignment module\n  │ \n  │   └── savi_hg38/                \u0026lt;- Working Directory of Somatic Mutation Calling: hg38 version \n  │       ├── log/                  \u0026lt;- Log from savi_hg38 module\n  │       ├── .../                  \u0026lt;- Other detailed subfolders containing working files, will be shown in future section\n  │\n  │   └── cnvkit_hg19/              \u0026lt;- Working Directory of CNV Analysis\n  │       ├── log/                  \u0026lt;- Log from CNV Analysis module\n  │       ├── .../                  \u0026lt;- Other detailed subfolders containing working files, will be shown in future section\n  │ \"\u003e\u003cpre\u003e── \u003cspan class=\"pl-smi\"\u003e${project_WES}\u003c/span\u003e/\n  │   └── reference/                \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Reference genome file (.FASTA)\n  │       ├── hg19bwa/              \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- hg19 version reference genome files\n  │       ├── TCGA_GRCh38_bwa_ref/  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- hg38 version reference genome files\n  │ \n  │   └── all_fq/                   \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Working Directory of Sequencing Data QC \n  │       ├── log/                  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Log from Sequencing Data QC module\n  │       ├── qc_res/               \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- QC result from Sequencing Data QC module\n  │ \n  │   └── all_mapping_hg19/         \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Working Directory of Genome Alignment: hg19 version \n  │       ├── log/                  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Log from Genome Alignment module\n  │       ├── md_metrix/            \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- MarkDuplicate metrix from alignment duplicates marking step\n  │       ├── md_tmp/               \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Location \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e temporary files from alignment duplicates marking step\n  │       ├── tmp/                  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Location \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e temporary files from Genome Alignment module\n  │ \n  │   └── all_mapping_hg38/         \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Working Directory of Genome Alignment: hg38 version \n  │       ├── log/                  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Log from Genome Alignment module\n  │       ├── md_metrix/            \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- MarkDuplicate metrix from alignment duplicates marking step\n  │       ├── md_tmp/               \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Location \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e temporary files from alignment duplicates marking step\n  │       ├── tmp/                  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Location \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e temporary files from Genome Alignment module\n  │ \n  │   └── savi_hg38/                \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Working Directory of Somatic Mutation Calling: hg38 version \n  │       ├── log/                  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Log from savi_hg38 module\n  │       ├── .../                  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Other detailed subfolders containing working files, will be shown \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e future section\n  │\n  │   └── cnvkit_hg19/              \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Working Directory of CNV Analysis\n  │       ├── log/                  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Log from CNV Analysis module\n  │       ├── .../                  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- Other detailed subfolders containing working files, will be shown \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e future section\n  │ \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eImportant Notes\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eThe reference folder contains the reference genome used, two different versions of reference genome file will be covered in this pipeline (hg19 and hg38). For WangLab HPC3 user, the folder can be linked as following:\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"ln -s /scratch/PI/jgwang/jtangbd/reference ${project_WES}/reference/\"\u003e\u003cpre\u003eln -s /scratch/PI/jgwang/jtangbd/reference \u003cspan class=\"pl-smi\"\u003e${project_WES}\u003c/span\u003e/reference/\u003c/pre\u003e\u003c/div\u003e\nFor other users, the reference genomes can be downloaded from the following pages: \u003ca href=\"https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/\" rel=\"nofollow\"\u003ehg19\u003c/a\u003e, \u003ca href=\"https://gdc.cancer.gov/about-data/gdc-data-processing/gdc-reference-files\" rel=\"nofollow\"\u003ehg38\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003eTwo different versions of reference genome are included in this pipeline. They are both used to generate different versions of genome alignment result files(BAMs). Afterwards, the bams are used differently for following mutation calling and CNV analysis module based on the reference genome version requirements for different tools.\u003c/li\u003e\n\u003cli\u003eFor the Somatic Mutation calling and CNV analysis modules, detailed folder arrangements will be explained in \u003ca href=\"#part3-somatic-mutation-calling\"\u003ePart3\u003c/a\u003e and \u003ca href=\"#part4-copy-number-variation-analysis\"\u003ePart4\u003c/a\u003e.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e0.2.4 Creating Cohort Index Files\u003c/h3\u003e\u003ca id=\"user-content-024-creating-cohort-index-files\" class=\"anchor\" aria-label=\"Permalink: 0.2.4 Creating Cohort Index Files\" href=\"#024-creating-cohort-index-files\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIn order to facilitate the cohort data management and parallel processing, we need to create two index files for the WES/WGS sequencing data in this whole pipeline. Both index files are simple .txt files containing the basic naming information for all the sequencing data.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ccode\u003eIndex File 1st\u003c/code\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe first file is used for Sequencing Data QC and Genome Aligenment modules, and it contains three columns, representing row number(0-index), raw name prefix of fastq files, and changed name prefix of bam files, respectively.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eBelow is the example format, it is expected to contain index rows for both Blood(B) and Tumor(T) samples in this index file. It is recommended to label the blood and tumor samples in specific symbols, such as B and T after the patient number.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe first column is used for easier arrangement of parallel computing, and detailed information will be provided in the next subsection. The second column is the original name prefix of the fastq files,\nand it is highly recommended to rename your fastq files in your own style but noted them well in this index file.\nThe third column is the changed name prefix you would like to name your bam files. It is highly recommended to contain DNA-seq methods(WES/WGS) and reference genome version(hg19/hg38) in the prefix for further confirmation.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe demo index file could be found \u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/list_SP_fq_bwa.txt\"\u003ehere\u003c/a\u003e. It is used for the demo SP project data and contains 47 patients, 94 sequencing data(B + T) files.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"0   P001_B    P001_B_WES_hg38\n1   P002_B    P002_B_WES_hg38\n2   P003_B    P003_B_WES_hg38\n3   P004_B    P004_B_WES_hg38\n4   P005_B    P005_B_WES_hg38\n5   P001_T    P001_T_WES_hg38\n6   P002_T    P002_T_WES_hg38\n7   P003_T    P003_T_WES_hg38\n8   P004_T    P004_T_WES_hg38\n9   P005_T    P005_T_WES_hg38\n...\"\u003e\u003cpre\u003e0   P001_B    P001_B_WES_hg38\n1   P002_B    P002_B_WES_hg38\n2   P003_B    P003_B_WES_hg38\n3   P004_B    P004_B_WES_hg38\n4   P005_B    P005_B_WES_hg38\n5   P001_T    P001_T_WES_hg38\n6   P002_T    P002_T_WES_hg38\n7   P003_T    P003_T_WES_hg38\n8   P004_T    P004_T_WES_hg38\n9   P005_T    P005_T_WES_hg38\n...\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ccode\u003eIndex File 2nd\u003c/code\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe second file is used for Somatic Mutation Calling and CNV Analysis modules, and it contains only two columns, representing row number(0-index), and patient prefix resepectively.\nThe second index file only needs to contain one index row for each patient, which is different to the first one.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eBelow is the example format and the demo index file could be found \u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/list_SP_savi_CNV.txt\"\u003ehere\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"0   P001\n1   P002\n2   P003\n4   P004\n5   P005\n...\"\u003e\u003cpre\u003e0   P001\n1   P002\n2   P003\n4   P004\n5   P005\n...\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePart1 Sequencing Data Quality Control\u003c/h1\u003e\u003ca id=\"user-content-part1-sequencing-data-quality-control\" class=\"anchor\" aria-label=\"Permalink: Part1 Sequencing Data Quality Control\" href=\"#part1-sequencing-data-quality-control\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1.1 Setting Up Tools\u003c/h2\u003e\u003ca id=\"user-content-11-setting-up-tools\" class=\"anchor\" aria-label=\"Permalink: 1.1 Setting Up Tools\" href=\"#11-setting-up-tools\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe main bioinformatics tool used in this module is fastp. The fastp is a tool designed to provide fast all-in-one preprocessing for FastQ files.\nIt can automatically filter reads, detect reads adapters, make adapter trimming and output QC results based on the demand. It covers the functions of well known QC tools like FastQC and Trim Galero, and makes the whole preprocessing work automatically in one step, which is very convenient.\nMore details about fastp could be found in their \u003ca href=\"https://github.com/OpenGene/fastp\"\u003euser manual page\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efastp could be easily downloaded and installed with bioconda:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# note: the fastp version in bioconda may be not the latest\nconda install -c bioconda fastp\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e note: the fastp version in bioconda may be not the latest\u003c/span\u003e\nconda install -c bioconda fastp\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eor download the prebuilt binary versions for usage:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# download the latest build\nwget http://opengene.org/fastp/fastp\nchmod a+x ./fastp\n\n# or download specified version, i.e. fastp v0.23.1\nwget http://opengene.org/fastp/fastp.0.23.1\nmv fastp.0.23.1 fastp\nchmod a+x ./fastp\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e download the latest build\u003c/span\u003e\nwget http://opengene.org/fastp/fastp\nchmod a+x ./fastp\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e or download specified version, i.e. fastp v0.23.1\u003c/span\u003e\nwget http://opengene.org/fastp/fastp.0.23.1\nmv fastp.0.23.1 fastp\nchmod a+x ./fastp\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1.2 FastQ Files Preparation\u003c/h2\u003e\u003ca id=\"user-content-12-fastq-files-preparation\" class=\"anchor\" aria-label=\"Permalink: 1.2 FastQ Files Preparation\" href=\"#12-fastq-files-preparation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFastQ file is often used to store the high throughput sequencing reads, and contains useful information including quality scores for each base.\nDeatiled description about FastQ files could be found \u003ca href=\"https://compgenomr.github.io/book/fasta-and-fastq-formats.html\" rel=\"nofollow\"\u003ehere\u003c/a\u003e. In order to start the genomics analysis,\nwe need to prepare the fastq files for the project. There are mainly two methods to get the fastq files for your project. The first one is producing your own data by sequencing(eg. sequencing service from companies).\nThe second one is downloading data from public database, and it is very useful when you would like to take advantage of the previous work from the community.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHowever, some datasets in the public database will not contain fastq files, instead, bam files are often contained. In order to control the analysis parameters and pipeline, we could extract the fastq files from the downloaded bam files, and then deal with them with your own pipeline.\nIn order to do so, we could use several different tools, and \u003cem\u003eSamToFastq\u003c/em\u003e in Picard is taken as example here.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/usr/bin/bash\n\npicard=/scratch/PI/jgwang/dsongad/software/Picard/picard.jar\n\n\nNAME=$1\nNEW=$2\n\n# The example script is setted for the HPC3 User in WangLab HKUST\n/scratch/PI/jgwang/dsongad/software/jdk-15.0.2/bin/java -Djava.io.tmpdir=tmp_$NAME \\\n        -Xmx36g -XX:+UseParallelGC -XX:ParallelGCThreads=2 \\\n        -jar ${picard} SamToFastq \\\n        INPUT=../${NAME}.bam \\\n        F=${NEW}.R1.fastq \\\n        F2=${NEW}.R2.fastq\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/bash\u003c/span\u003e\n\npicard=/scratch/PI/jgwang/dsongad/software/Picard/picard.jar\n\n\nNAME=\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e\nNEW=\u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e The example script is setted for the HPC3 User in WangLab HKUST\u003c/span\u003e\n/scratch/PI/jgwang/dsongad/software/jdk-15.0.2/bin/java -Djava.io.tmpdir=tmp_\u003cspan class=\"pl-smi\"\u003e$NAME\u003c/span\u003e \\\n        -Xmx36g -XX:+UseParallelGC -XX:ParallelGCThreads=2 \\\n        -jar \u003cspan class=\"pl-smi\"\u003e${picard}\u003c/span\u003e SamToFastq \\\n        INPUT=../\u003cspan class=\"pl-smi\"\u003e${NAME}\u003c/span\u003e.bam \\\n        F=\u003cspan class=\"pl-smi\"\u003e${NEW}\u003c/span\u003e.R1.fastq \\\n        F2=\u003cspan class=\"pl-smi\"\u003e${NEW}\u003c/span\u003e.R2.fastq\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1.3 Automatic QC with fastp\u003c/h2\u003e\u003ca id=\"user-content-13-automatic-qc-with-fastp\" class=\"anchor\" aria-label=\"Permalink: 1.3 Automatic QC with fastp\" href=\"#13-automatic-qc-with-fastp\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIn order to facilitate the parallel processing of multiple samples and make the pipeline reproducible, we could summarize the commands in scripts. All the scripts we created in this whole pipeline are template structured, which will read parameters from the command line if we use them individually.\nIn this way, we only need to change several simple parameters to control the whole pipeline to deal with different samples.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe arrangement logic of the scripts for part 1-3 is the same and can be summarized as follows. For each step, one bash script and one perl script will be used. The bash script contains the detailed command to use different bioinformatics tools, eg. fastp in this step. The required parameters of the bash script often determines which sample will be processed each time.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe perl script contains the template structure to read in the index files we created in the above \u003ca href=\"#024-creating-cohort-index-files\"\u003esection\u003c/a\u003e, the detailed command to run the bash script and how to control the parallel processing procedure. The perl script often requires the path of the index file, the working directory, the start index and end index to the working samples as parameters.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1.3.1 Working Scripts\u003c/h3\u003e\u003ca id=\"user-content-131-working-scripts\" class=\"anchor\" aria-label=\"Permalink: 1.3.1 Working Scripts\" href=\"#131-working-scripts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part1-Sequencing-Data-QC/do_fastp.sh\"\u003e\u003ccode\u003eScript1: do_fastp.sh(Bash script)\u003c/code\u003e\u003c/a\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe script will read in two parameters, and use them as the raw name prefix of fastq files and the changed name prefix. The paired fastq files are expected to end with suffix \u003cem\u003e.R1.fq.gz\u003c/em\u003e and \u003cem\u003e.R2.fq.gz\u003c/em\u003e to use this script, and the script can also be modified based on your own file naming style.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe QC results will be stored in the subfoler \u003cem\u003eqc_res\u003c/em\u003e, and the raw fastq files will be removed after the automatic preprocessing.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/usr/bin/bash\n\nrawName=$1\ncleanName=$2_fp\n\n#mkdir -p qc_res\n\nfastp -w 2 -i ./$rawName\\.R1.fq.gz -o ./$cleanName\\.R1.fq.gz \\\n -I ./$rawName\\.R2.fq.gz -O ./$cleanName\\.R2.fq.gz \\\n -q 15 -u 40 --length_required 45 \\\n --detect_adapter_for_pe \\\n -h ./qc_res/fastp_$cleanName\\.html -j ./qc_res/fastp_$cleanName\\.json -R ./qc_res/report_$cleanName\\.txt\n\nrm ./$rawName\\.R1.fq.gz ./$rawName\\.R2.fq.gz\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/bash\u003c/span\u003e\n\nrawName=\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e\ncleanName=\u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e_fp\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003emkdir -p qc_res\u003c/span\u003e\n\nfastp -w 2 -i ./\u003cspan class=\"pl-smi\"\u003e$rawName\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003eR1.fq.gz -o ./\u003cspan class=\"pl-smi\"\u003e$cleanName\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003eR1.fq.gz \\\n -I ./\u003cspan class=\"pl-smi\"\u003e$rawName\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003eR2.fq.gz -O ./\u003cspan class=\"pl-smi\"\u003e$cleanName\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003eR2.fq.gz \\\n -q 15 -u 40 --length_required 45 \\\n --detect_adapter_for_pe \\\n -h ./qc_res/fastp_\u003cspan class=\"pl-smi\"\u003e$cleanName\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003ehtml -j ./qc_res/fastp_\u003cspan class=\"pl-smi\"\u003e$cleanName\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003ejson -R ./qc_res/report_\u003cspan class=\"pl-smi\"\u003e$cleanName\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003etxt\n\nrm ./\u003cspan class=\"pl-smi\"\u003e$rawName\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003eR1.fq.gz ./\u003cspan class=\"pl-smi\"\u003e$rawName\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003eR2.fq.gz\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part1-Sequencing-Data-QC/runfastp.pl\"\u003e\u003ccode\u003eScript2: run_fastp.pl(Perl script)\u003c/code\u003e\u003c/a\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe script will read in three parameters, and use them as index file path, start index and end index respectively. The previous created index file will be used here to control the parallel processing procedure. The logic is that we could use start index and end index to choose the specific samples to process each time.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor example, when we want to run the first two samples to test the scripts, we could set the start index as 0, the end index as 1. By this way, the perl script will loop from row index 0 to 1 in the index file to read in the index information and pass them to the bash script.\nThen the bash script will process the two samples one by one based on the parameters received.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-perl notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/usr/bin/perl\n\n#please run this job in a new clear fold\ndie \u0026quot;usage: perl runfastp.pl \u0026lt;Data_Index\u0026gt; \u0026lt;start\u0026gt; \u0026lt;end\u0026gt; \\n\u0026quot; if @ARGV!= 3;\n\n$index_table = $ARGV[0];\n\n\n\nopen INDEX, $index_table;\n$i = 0;\n$id = 0;\nwhile($line = \u0026lt;INDEX\u0026gt;){\n\tchomp($line);\n\tmy @temp=split('\\t',$line);\n\t$sample[$i][0] = $temp[1];\n\t$sample[$i][1] = $temp[2];\n\t$i++;\n}\nclose INDEX;\n\n\n$start = $ARGV[1];\n$end = $ARGV[2];\n\nfor($j=$start;$j\u0026lt;=$end;$j++){         \n        `sh ./do_fastp.sh $sample[$j][0] $sample[$j][0]`; # Important command, call the bash script based on parameters\n\t$completed_No = $j;\n        print \u0026quot;Job completed!!! No.$completed_No : $sample[$j][1]\\t$sample[$j][0]\\n\\n\u0026quot;;\n}\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e!/usr/bin/perl\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eplease run this job in a new clear fold\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edie\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eusage: perl runfastp.pl \u0026lt;Data_Index\u0026gt; \u0026lt;start\u0026gt; \u0026lt;end\u0026gt; \u003cspan class=\"pl-cce\"\u003e\\n\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e@ARGV\u003c/span\u003e!= 3;\n\n\u003cspan class=\"pl-smi\"\u003e$index_table\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[0];\n\n\n\n\u003cspan class=\"pl-c1\"\u003eopen\u003c/span\u003e INDEX, \u003cspan class=\"pl-smi\"\u003e$index_table\u003c/span\u003e;\n\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e = 0;\n\u003cspan class=\"pl-smi\"\u003e$id\u003c/span\u003e = 0;\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e = \u0026lt;INDEX\u0026gt;){\n\t\u003cspan class=\"pl-c1\"\u003echomp\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e);\n\t\u003cspan class=\"pl-k\"\u003emy\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e@temp\u003c/span\u003e=\u003cspan class=\"pl-c1\"\u003esplit\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e);\n\t\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e][0] = \u003cspan class=\"pl-smi\"\u003e$temp\u003c/span\u003e[1];\n\t\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e][1] = \u003cspan class=\"pl-smi\"\u003e$temp\u003c/span\u003e[2];\n\t\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e++;\n}\n\u003cspan class=\"pl-c1\"\u003eclose\u003c/span\u003e INDEX;\n\n\n\u003cspan class=\"pl-smi\"\u003e$start\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[1];\n\u003cspan class=\"pl-smi\"\u003e$end\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[2];\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e=\u003cspan class=\"pl-smi\"\u003e$start\u003c/span\u003e;\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e\u0026lt;=\u003cspan class=\"pl-smi\"\u003e$end\u003c/span\u003e;\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e++){         \n        \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003esh ./do_fastp.sh \u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0] \u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0]\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e; \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Important command, call the bash script based on parameters\u003c/span\u003e\n\t\u003cspan class=\"pl-smi\"\u003e$completed_No\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e;\n        \u003cspan class=\"pl-c1\"\u003eprint\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eJob completed!!! No.\u003cspan class=\"pl-smi\"\u003e$completed_No\u003c/span\u003e : \u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][1]\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0]\u003cspan class=\"pl-cce\"\u003e\\n\\n\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e;\n}\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1.3.2 Scripts Usage Examples\u003c/h3\u003e\u003ca id=\"user-content-132-scripts-usage-examples\" class=\"anchor\" aria-label=\"Permalink: 1.3.2 Scripts Usage Examples\" href=\"#132-scripts-usage-examples\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFor simple usage or script test, we could directly run the perl script with index paramters. It is highly recommended to record all the log information by redirecting them to another log file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe example command used to call perl script for simple test usage could be shown as follows. The command will call the scripts to do fastp work on the first sample only and store the log information into the subfoler \u003cem\u003elog\u003c/em\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"perl run_fastp.pl list_SP_fq_bwa.txt 0 0 2\u0026gt;./log/log.fp_0_0\"\u003e\u003cpre\u003eperl run_fastp.pl list_SP_fq_bwa.txt 0 0 \u003cspan class=\"pl-k\"\u003e2\u0026gt;\u003c/span\u003e./log/log.fp_0_0\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part1-Sequencing-Data-QC/slurm.fp_all\"\u003e\u003ccode\u003eScript3: slurm.fp_all(Slurm script)\u003c/code\u003e\u003c/a\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor the slurm cluster users to run multiple samples, the example sbatch script could be arranged as follows.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash\n##SBATCH -J bwa24  #Slurm job name\n#SBATCH --mail-user=njutangjihong@gmail.com \n#SBATCH --mail-type=end\n##SBATCH -p x-gpu-share\n##SBATCH -o tra_MD.out\n##SBATCH -e tra_MD.err\n##SBATCH -N 1 -n 24\n#SBATCH --cpus-per-task=20\n##SBATCH --exclusive\n\nperl runfastp.pl list_SP_fq_bwa.txt $1 $2 2\u0026gt;./log/log.fp_$1_$2\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -J bwa24  #Slurm job name\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=njutangjihong@gmail.com \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=end\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -p x-gpu-share\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -o tra_MD.out\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -e tra_MD.err\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -N 1 -n 24\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --cpus-per-task=20\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH --exclusive\u003c/span\u003e\n\nperl runfastp.pl list_SP_fq_bwa.txt \u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e2\u0026gt;\u003c/span\u003e./log/log.fp_\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTaken the demo project as example, there are totally 94(0-93) samples in the index files, and we could process them parallelly on four computing nodes as follows:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"sbatch -p x-gpu-share -J fp0 slurm.fp_all 0 23 \nsbatch -p x-gpu-share -J fp24 slurm.fp_all 24 47 \nsbatch -p x-gpu-share -J fp48 slurm.fp_all 48 71\nsbatch -p x-gpu-share -J fp72 slurm.fp_all 72 93 \"\u003e\u003cpre\u003esbatch -p x-gpu-share -J fp0 slurm.fp_all 0 23 \nsbatch -p x-gpu-share -J fp24 slurm.fp_all 24 47 \nsbatch -p x-gpu-share -J fp48 slurm.fp_all 48 71\nsbatch -p x-gpu-share -J fp72 slurm.fp_all 72 93 \u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePart2 Genome Alignment\u003c/h1\u003e\u003ca id=\"user-content-part2-genome-alignment\" class=\"anchor\" aria-label=\"Permalink: Part2 Genome Alignment\" href=\"#part2-genome-alignment\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.1 Setting Up Tools\u003c/h2\u003e\u003ca id=\"user-content-21-setting-up-tools\" class=\"anchor\" aria-label=\"Permalink: 2.1 Setting Up Tools\" href=\"#21-setting-up-tools\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eGenome alignment and following duplicated marking are essential steps in the genomics analysis pipeline to get the bam files. The Bam files are the binary version of Sam files, and the Sam files are tab-delimited text file that contains sequence alignment data.\nDetailed information about SAM/BAM files could be found \u003ca href=\"http://samtools.github.io/hts-specs/\" rel=\"nofollow\"\u003ehere\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe bioinformatics tools used in this module are bwa and Picard. Bwa is one of the most famous alignment tools, and is considered as the most useful one. Detailed information about the bwa tools can be found \u003ca href=\"https://github.com/lh3/bwa\"\u003ehere\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eBwa can also be installed easily through conda:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"conda install -c bioconda bwa\"\u003e\u003cpre\u003econda install -c bioconda bwa\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIn addition, bwa can be installed as follows:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"git clone https://github.com/lh3/bwa.git\ncd bwa; make\n./bwa index ref.fa\n./bwa mem ref.fa read-se.fq.gz | gzip -3 \u0026gt; aln-se.sam.gz\n./bwa mem ref.fa read1.fq read2.fq | gzip -3 \u0026gt; aln-pe.sam.gz\"\u003e\u003cpre\u003egit clone https://github.com/lh3/bwa.git\n\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e bwa\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e make\n./bwa index ref.fa\n./bwa mem ref.fa read-se.fq.gz \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip -3 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e aln-se.sam.gz\n./bwa mem ref.fa read1.fq read2.fq \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e gzip -3 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e aln-pe.sam.gz\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePicard is a set of Java command line tools for manipulating high-throughput sequencing (HTS) data and formats. Well noted manual and notes have been provided by the developers, and they can be found \u003ca href=\"https://broadinstitute.github.io/picard/\" rel=\"nofollow\"\u003ehere\u003c/a\u003e.\nIt is highly recommended to follow their instructions to download and install the Picard and set up java in order to use specific commands.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor WangLab HKUST HPC3 users, the Picard tool can be used by linking to the following path:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"/scratch/PI/jgwang/dsongad/software/Picard/picard.jar\"\u003e\u003cpre\u003e/scratch/PI/jgwang/dsongad/software/Picard/picard.jar\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.2 Genome Alignment with bwa\u003c/h2\u003e\u003ca id=\"user-content-22-genome-alignment-with-bwa\" class=\"anchor\" aria-label=\"Permalink: 2.2 Genome Alignment with bwa\" href=\"#22-genome-alignment-with-bwa\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe scripts arrangement logic is the same as previously stated in the section \u003ca href=\"#13-automatic-qc-with-fastp\"\u003e1.3 Automatic QC with fastp\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.2.1 Working Scripts\u003c/h3\u003e\u003ca id=\"user-content-221-working-scripts\" class=\"anchor\" aria-label=\"Permalink: 2.2.1 Working Scripts\" href=\"#221-working-scripts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part2-Genome-Alignment/do_bwa_dna.sh\"\u003e\u003ccode\u003eScript1: do_bwa_dna.sh(Bash script)\u003c/code\u003e\u003c/a\u003e\u003c/strong\u003e\nThe script is expected to receive four parameters from user input, representing sample ID, R1 fastq file name, R2 fastq file name and user defined result bam prefix name.\nThe alignment result bam will be sorted and indexed after the bwa commands, and only the sorted bam will be remained, accompanied by its own index .bai file.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/usr/bin/bash\nA='WangLab'\nSample_ID=$1\nFAQ1=$2\nFAQ2=$3\nresult=$4\n\n\necho [Directory] `pwd`\necho [Machine] `uname -n`\necho [Start] `date`\necho [args] $*\ntime1=$( date \u0026quot;+%s\u0026quot; )\n\n\n#line_q1=`zcat $FAQ1 | wc -l`\n#line_q2=`zcat $FAQ2 | wc -l`\n\nbwa mem -t 20 -T 0 -R \u0026quot;@RG\\tCN:$A\\tID:$Sample_ID\\tSM:$Sample_ID\u0026quot; /scratch/PI/jgwang/jtangbd/reference/TCGA_GRCh38_bwa_ref/GRCh38.d1.vd1.fa $FAQ1 $FAQ2 2\u0026gt; ./tmp/tmp.$result | samtools view -Sbh1 -@ 8 - -o $result\\.bam\n\n#### mapping duplicates\n###counting the num of bam reads\n\nsamtools sort -@ 8 -O bam -o $result.sorted.bam -T the_temp_$result $result\\.bam\nsamtools index $result\\.sorted.bam \nrm $result\\.bam\n#rm $FAQ1\n#rm $FAQ2\n#samtools idxstats result_test.sorted.bam |head \n#samtools view -H /home/qmu/projects/AsianpGBM/TCGAGBM/C484.TCGA-06-6699-10A-01D-1845-08.2_gdc_realn.bam |grep \u0026quot;^@PG\u0026quot;\n#samtools flagstat result_test.sorted.bam\n#samtools view -f 4 result_test.sorted.bam chr7 | wc -l\n#samtools mpileup ../raw_data/results_B_DNA/IGCT_S002_B.sorted.bam -r chr10:63207680 -f /home/dsongad/software/my_lib/STAR_index_hg38_RefSeq_hg38/hg38.fa |head -n20\n\ntime2=$( date \u0026quot;+%s\u0026quot; )\necho [deltat] $(( $time2 - $time1 ))\necho [End]\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/bash\u003c/span\u003e\nA=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eWangLab\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nSample_ID=\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e\nFAQ1=\u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e\nFAQ2=\u003cspan class=\"pl-smi\"\u003e$3\u003c/span\u003e\nresult=\u003cspan class=\"pl-smi\"\u003e$4\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e [Directory] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003epwd\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e [Machine] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003euname -n\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e [Start] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003edate\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e [args] \u003cspan class=\"pl-smi\"\u003e$*\u003c/span\u003e\ntime1=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003e date \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e+%s\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eline_q1=`zcat $FAQ1 | wc -l`\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eline_q2=`zcat $FAQ2 | wc -l`\u003c/span\u003e\n\nbwa mem -t 20 -T 0 -R \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e@RG\\tCN:\u003cspan class=\"pl-smi\"\u003e$A\u003c/span\u003e\\tID:\u003cspan class=\"pl-smi\"\u003e$Sample_ID\u003c/span\u003e\\tSM:\u003cspan class=\"pl-smi\"\u003e$Sample_ID\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e /scratch/PI/jgwang/jtangbd/reference/TCGA_GRCh38_bwa_ref/GRCh38.d1.vd1.fa \u003cspan class=\"pl-smi\"\u003e$FAQ1\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FAQ2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e2\u0026gt;\u003c/span\u003e ./tmp/tmp.\u003cspan class=\"pl-smi\"\u003e$result\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e samtools view -Sbh1 -@ 8 - -o \u003cspan class=\"pl-smi\"\u003e$result\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003ebam\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### mapping duplicates\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##counting the num of bam reads\u003c/span\u003e\n\nsamtools sort -@ 8 -O bam -o \u003cspan class=\"pl-smi\"\u003e$result\u003c/span\u003e.sorted.bam -T the_temp_\u003cspan class=\"pl-smi\"\u003e$result\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$result\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003ebam\nsamtools index \u003cspan class=\"pl-smi\"\u003e$result\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003esorted.bam \nrm \u003cspan class=\"pl-smi\"\u003e$result\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\.\u003c/span\u003ebam\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003erm $FAQ1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003erm $FAQ2\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esamtools idxstats result_test.sorted.bam |head \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esamtools view -H /home/qmu/projects/AsianpGBM/TCGAGBM/C484.TCGA-06-6699-10A-01D-1845-08.2_gdc_realn.bam |grep \"^@PG\"\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esamtools flagstat result_test.sorted.bam\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esamtools view -f 4 result_test.sorted.bam chr7 | wc -l\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esamtools mpileup ../raw_data/results_B_DNA/IGCT_S002_B.sorted.bam -r chr10:63207680 -f /home/dsongad/software/my_lib/STAR_index_hg38_RefSeq_hg38/hg38.fa |head -n20\u003c/span\u003e\n\ntime2=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003e date \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e+%s\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e [deltat] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$((\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$time2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$time1\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e))\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e [End]\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part2-Genome-Alignment/run_bwa_mapping.pl\"\u003e\u003ccode\u003eScript2: run_bwa_mapping.pl(Perl script)\u003c/code\u003e\u003c/a\u003e\u003c/strong\u003e\nThe script will read in four parameters, and use them as index file path, fastq file path, start index and end index respectively.\nThe previous created index file will be used here to control the parallel processing procedure. The logic is that we could use start index and end index to choose the specific samples to process each time.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe parameter \u003ccode\u003e\u0026lt;fastq-dir\u0026gt;\u003c/code\u003e is essential to be provided to locate the fastq files, and user can choose to delete the raw fastq files after alignment for storage room saving.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-perl notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/usr/bin/perl\n\n#please run this job in a new clear fold\ndie \u0026quot;usage: perl run_bwa_mapping.pl \u0026lt;Data_Index\u0026gt; \u0026lt;fastq-dir\u0026gt; \u0026lt;START\u0026gt; \u0026lt;END\u0026gt; \\n\u0026quot; if @ARGV!= 4;\n\n$index_table = $ARGV[0];\n$root = $ARGV[1];\n\n\nopen INDEX, $index_table;\n$i = 0;\n$id = 0;\nwhile($line = \u0026lt;INDEX\u0026gt;){\n\tchomp($line);\n\tmy @temp=split('\\t',$line);\n\t$sample[$i][0] = $temp[1];\n\t$sample[$i][1] = $temp[2];\n\t$i++;\n}\nclose INDEX;\n\n#`mkdir Dir_tmp_1`;\n#chdir \u0026quot;./Dir_tmp_1\u0026quot;;\n#\n$start = $ARGV[2];\n$end = $ARGV[3];\n\nfor($j=$start;$j\u0026lt;=$end;$j++){\n        `sh ./do_bwa_dna.sh $sample[$j][0] $root/$sample[$j][0]_fp.R1.fq.gz $root/$sample[$j][0]_fp.R2.fq.gz $sample[$j][1]`;\n\t$completed_No = $j;\n        print \u0026quot;bwa-MEM completed!!! No.$completed_No : $sample[$j][1]\\t$sample[$j][0]\\n\u0026quot;;\n}\n#chdir \u0026quot;../\u0026quot;;\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e!/usr/bin/perl\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eplease run this job in a new clear fold\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edie\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eusage: perl run_bwa_mapping.pl \u0026lt;Data_Index\u0026gt; \u0026lt;fastq-dir\u0026gt; \u0026lt;START\u0026gt; \u0026lt;END\u0026gt; \u003cspan class=\"pl-cce\"\u003e\\n\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e@ARGV\u003c/span\u003e!= 4;\n\n\u003cspan class=\"pl-smi\"\u003e$index_table\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[0];\n\u003cspan class=\"pl-smi\"\u003e$root\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[1];\n\n\n\u003cspan class=\"pl-c1\"\u003eopen\u003c/span\u003e INDEX, \u003cspan class=\"pl-smi\"\u003e$index_table\u003c/span\u003e;\n\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e = 0;\n\u003cspan class=\"pl-smi\"\u003e$id\u003c/span\u003e = 0;\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e = \u0026lt;INDEX\u0026gt;){\n\t\u003cspan class=\"pl-c1\"\u003echomp\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e);\n\t\u003cspan class=\"pl-k\"\u003emy\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e@temp\u003c/span\u003e=\u003cspan class=\"pl-c1\"\u003esplit\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e);\n\t\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e][0] = \u003cspan class=\"pl-smi\"\u003e$temp\u003c/span\u003e[1];\n\t\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e][1] = \u003cspan class=\"pl-smi\"\u003e$temp\u003c/span\u003e[2];\n\t\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e++;\n}\n\u003cspan class=\"pl-c1\"\u003eclose\u003c/span\u003e INDEX;\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e`mkdir Dir_tmp_1`;\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003echdir \"./Dir_tmp_1\";\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003e$start\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[2];\n\u003cspan class=\"pl-smi\"\u003e$end\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[3];\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e=\u003cspan class=\"pl-smi\"\u003e$start\u003c/span\u003e;\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e\u0026lt;=\u003cspan class=\"pl-smi\"\u003e$end\u003c/span\u003e;\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e++){\n        \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003esh ./do_bwa_dna.sh \u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0] \u003cspan class=\"pl-smi\"\u003e$root\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0]_fp.R1.fq.gz \u003cspan class=\"pl-smi\"\u003e$root\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0]_fp.R2.fq.gz \u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][1]\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e;\n\t\u003cspan class=\"pl-smi\"\u003e$completed_No\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e;\n        \u003cspan class=\"pl-c1\"\u003eprint\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebwa-MEM completed!!! No.\u003cspan class=\"pl-smi\"\u003e$completed_No\u003c/span\u003e : \u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][1]\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0]\u003cspan class=\"pl-cce\"\u003e\\n\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e;\n}\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003echdir \"../\";\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.2.2 Scripts Usage Example\u003c/h3\u003e\u003ca id=\"user-content-222-scripts-usage-example\" class=\"anchor\" aria-label=\"Permalink: 2.2.2 Scripts Usage Example\" href=\"#222-scripts-usage-example\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFor simple usage or script test, we could directly run the perl script with index paramters. It is highly recommended to record all the log information by redirecting them to another log file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe example command used to call perl script for simple test usage could be shown as follows. The command will call the scripts to do bwa mapping on the first sample only and store the log information into the subfoler \u003cem\u003elog\u003c/em\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"perl run_bwa_mapping.pl list_SP_fq_bwa.txt 0 0 2\u0026gt;./log/log.bwa_0_0\"\u003e\u003cpre\u003eperl run_bwa_mapping.pl list_SP_fq_bwa.txt 0 0 \u003cspan class=\"pl-k\"\u003e2\u0026gt;\u003c/span\u003e./log/log.bwa_0_0\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part1-Sequencing-Data-QC/slurm.bwa_all\"\u003e\u003ccode\u003eScript3: slurm.bwa_all(Slurm script)\u003c/code\u003e\u003c/a\u003e\u003c/strong\u003e\nFor the slurm cluster users to run multiple samples, the example sbatch script could be arranged as follows.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash\n##SBATCH -J bwa24  #Slurm job name\n#SBATCH --mail-user=njutangjihong@gmail.com \n#SBATCH --mail-type=end\n##SBATCH -p x-gpu-share\n##SBATCH -o tra_MD.out\n##SBATCH -e tra_MD.err\n##SBATCH -N 1 -n 24\n#SBATCH --cpus-per-task=20\n#SBATCH --exclusive\n\nperl run_bwa_mapping.pl list_SP_fq_bwa.txt ../all_fq $1 $2 2\u0026gt;./log/log.bwa_$1_$2\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -J bwa24  #Slurm job name\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=njutangjihong@gmail.com \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=end\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -p x-gpu-share\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -o tra_MD.out\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -e tra_MD.err\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -N 1 -n 24\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --cpus-per-task=20\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --exclusive\u003c/span\u003e\n\nperl run_bwa_mapping.pl list_SP_fq_bwa.txt ../all_fq \u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e2\u0026gt;\u003c/span\u003e./log/log.bwa_\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTaken the demo project as example, there are totally 94(0-93) samples in the index files, and we could process them parallelly on four computing nodes as follows:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"sbatch -p x-gpu-share -J bwa0 slurm.bwa_all 0 23 \nsbatch -p x-gpu-share -J bwa24 slurm.bwa_all 24 47 \nsbatch -p x-gpu-share -J bwa48 slurm.bwa_all 48 71\nsbatch -p x-gpu-share -J bwa72 slurm.bwa_all 72 93 \"\u003e\u003cpre\u003esbatch -p x-gpu-share -J bwa0 slurm.bwa_all 0 23 \nsbatch -p x-gpu-share -J bwa24 slurm.bwa_all 24 47 \nsbatch -p x-gpu-share -J bwa48 slurm.bwa_all 48 71\nsbatch -p x-gpu-share -J bwa72 slurm.bwa_all 72 93 \u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.3 Mark Duplicates with Picard\u003c/h2\u003e\u003ca id=\"user-content-23-mark-duplicates-with-picard\" class=\"anchor\" aria-label=\"Permalink: 2.3 Mark Duplicates with Picard\" href=\"#23-mark-duplicates-with-picard\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe scripts arrangement logic is the same as previously stated in the section \u003ca href=\"#13-automatic-qc-with-fastp\"\u003e1.3 Automatic QC with fastp\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.3.1 Working Scripts\u003c/h3\u003e\u003ca id=\"user-content-231-working-scripts\" class=\"anchor\" aria-label=\"Permalink: 2.3.1 Working Scripts\" href=\"#231-working-scripts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part2-Genome-Alignment/do_mark_duplicates.sh\"\u003e\u003ccode\u003eScript1: do_mark_duplicates.sh(Bash script)\u003c/code\u003e\u003c/a\u003e\u003c/strong\u003e\nThe script is expected to receive only one parameter from user input, the prefix name of the bam file. After the mark duplicates work, the bam files will be ended with .MD. surfix, and the .MD.bam files will be sorted and indexed again.\nThe raw bam files will be removed and only the .MD.bam and .MD.bam.bai files will be remained.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eOne important note: The mark duplicates step is implemented in the same directory after the bwa mapping step, and must be used in the same folder if no change is made to the script.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/usr/bin/bash\n\n#set -euxo pipefail\n#export LD_LIBRARY_PATH=/home/qmu/tools/lib:/opt/intel/advisor_xe_2016.1.40.463413/lib64:/home/qmu/tools/xz-5.2.3/tmp/lib \n#picard=/home/share/jgwang/softwares/picard/build/libs/picard.jar\npicard=/scratch/PI/jgwang/dsongad/software/Picard/picard.jar\n\nNAME=$1\n\nmkdir md_tmp/tmp_$NAME\n\njava -Djava.io.tmpdir=md_tmp/tmp_$NAME \\\n\t-Xmx120g -XX:+UseParallelGC -XX:ParallelGCThreads=4 \\\n\t -jar ${picard} MarkDuplicates \\\n\tINPUT=${NAME}.sorted.bam \\\n\tOUTPUT=${NAME}.sorted.MD.bam \\\n\tMETRICS_FILE=./md_metrix/${NAME}.metrics.txt \\\n\tASSUME_SORTED=true \\\n\tREMOVE_DUPLICATES=true \\\n\tVALIDATION_STRINGENCY=LENIENT\n\nrm ${NAME}.sorted.bam\nrm ${NAME}.sorted.bam.bai\nsamtools index ${NAME}.sorted.MD.bam\n#rm ${NAME}.sorted.MD.bam.txt\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/bash\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eset -euxo pipefail\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eexport LD_LIBRARY_PATH=/home/qmu/tools/lib:/opt/intel/advisor_xe_2016.1.40.463413/lib64:/home/qmu/tools/xz-5.2.3/tmp/lib \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003epicard=/home/share/jgwang/softwares/picard/build/libs/picard.jar\u003c/span\u003e\npicard=/scratch/PI/jgwang/dsongad/software/Picard/picard.jar\n\nNAME=\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e\n\nmkdir md_tmp/tmp_\u003cspan class=\"pl-smi\"\u003e$NAME\u003c/span\u003e\n\njava -Djava.io.tmpdir=md_tmp/tmp_\u003cspan class=\"pl-smi\"\u003e$NAME\u003c/span\u003e \\\n\t-Xmx120g -XX:+UseParallelGC -XX:ParallelGCThreads=4 \\\n\t -jar \u003cspan class=\"pl-smi\"\u003e${picard}\u003c/span\u003e MarkDuplicates \\\n\tINPUT=\u003cspan class=\"pl-smi\"\u003e${NAME}\u003c/span\u003e.sorted.bam \\\n\tOUTPUT=\u003cspan class=\"pl-smi\"\u003e${NAME}\u003c/span\u003e.sorted.MD.bam \\\n\tMETRICS_FILE=./md_metrix/\u003cspan class=\"pl-smi\"\u003e${NAME}\u003c/span\u003e.metrics.txt \\\n\tASSUME_SORTED=true \\\n\tREMOVE_DUPLICATES=true \\\n\tVALIDATION_STRINGENCY=LENIENT\n\nrm \u003cspan class=\"pl-smi\"\u003e${NAME}\u003c/span\u003e.sorted.bam\nrm \u003cspan class=\"pl-smi\"\u003e${NAME}\u003c/span\u003e.sorted.bam.bai\nsamtools index \u003cspan class=\"pl-smi\"\u003e${NAME}\u003c/span\u003e.sorted.MD.bam\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003erm ${NAME}.sorted.MD.bam.txt\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part2-Genome-Alignment/run_mark_duplicates.pl\"\u003e\u003ccode\u003eScript2: run_mark_duplicates.pl(Perl script)\u003c/code\u003e\u003c/a\u003e\u003c/strong\u003e\nThe script will read in four parameters, and use them as index file path, fastq file path, start index and end index respectively.\nThe previous created index file will be used here to control the parallel processing procedure. The logic is that we could use start index and end index to choose the specific samples to process each time.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe parameter setting is the same to the script \u003ccode\u003erun_bwa_mapping.pl\u003c/code\u003e in order to simplify the usage of the index files.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-perl notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/usr/bin/perl\n\n#please run this job in a new clear fold\ndie \u0026quot;usage: perl run_mark_duplicates.pl \u0026lt;Data_Index\u0026gt; \u0026lt;fastq-dir\u0026gt; \u0026lt;START\u0026gt; \u0026lt;END\u0026gt; \\n\u0026quot; if @ARGV!= 4;\n\n$index_table = $ARGV[0];\n$root = $ARGV[1];\n\n\nopen INDEX, $index_table;\n$i = 0;\n$id = 0;\nwhile($line = \u0026lt;INDEX\u0026gt;){\n\tchomp($line);\n\tmy @temp=split('\\t',$line);\n\t$sample[$i][0] = $temp[1];\n\t$sample[$i][1] = $temp[2];\n\t$i++;\n}\nclose INDEX;\n\n#`mkdir Dir_tmp_1`;\n#chdir \u0026quot;./Dir_tmp_1\u0026quot;;\n#\n$start = $ARGV[2];\n$end = $ARGV[3];\n\nfor($j=$start;$j\u0026lt;=$end;$j++){          #run the first 100 samples as a try\n\t`sh ./do_mark_duplicates.sh $sample[$j][1]`;\n\t$completed_No = $j;\n        print \u0026quot;MarkingDup completed!!! No.$completed_No : $sample[$j][0]\\t$sample[$j][1]\\n\\n\u0026quot;;\n}\n#chdir \u0026quot;../\u0026quot;;\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e!/usr/bin/perl\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eplease run this job in a new clear fold\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edie\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eusage: perl run_mark_duplicates.pl \u0026lt;Data_Index\u0026gt; \u0026lt;fastq-dir\u0026gt; \u0026lt;START\u0026gt; \u0026lt;END\u0026gt; \u003cspan class=\"pl-cce\"\u003e\\n\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e@ARGV\u003c/span\u003e!= 4;\n\n\u003cspan class=\"pl-smi\"\u003e$index_table\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[0];\n\u003cspan class=\"pl-smi\"\u003e$root\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[1];\n\n\n\u003cspan class=\"pl-c1\"\u003eopen\u003c/span\u003e INDEX, \u003cspan class=\"pl-smi\"\u003e$index_table\u003c/span\u003e;\n\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e = 0;\n\u003cspan class=\"pl-smi\"\u003e$id\u003c/span\u003e = 0;\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e = \u0026lt;INDEX\u0026gt;){\n\t\u003cspan class=\"pl-c1\"\u003echomp\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e);\n\t\u003cspan class=\"pl-k\"\u003emy\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e@temp\u003c/span\u003e=\u003cspan class=\"pl-c1\"\u003esplit\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e);\n\t\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e][0] = \u003cspan class=\"pl-smi\"\u003e$temp\u003c/span\u003e[1];\n\t\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e][1] = \u003cspan class=\"pl-smi\"\u003e$temp\u003c/span\u003e[2];\n\t\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e++;\n}\n\u003cspan class=\"pl-c1\"\u003eclose\u003c/span\u003e INDEX;\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e`mkdir Dir_tmp_1`;\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003echdir \"./Dir_tmp_1\";\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003e$start\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[2];\n\u003cspan class=\"pl-smi\"\u003e$end\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[3];\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e=\u003cspan class=\"pl-smi\"\u003e$start\u003c/span\u003e;\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e\u0026lt;=\u003cspan class=\"pl-smi\"\u003e$end\u003c/span\u003e;\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e++){          \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003erun the first 100 samples as a try\u003c/span\u003e\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003esh ./do_mark_duplicates.sh \u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][1]\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e;\n\t\u003cspan class=\"pl-smi\"\u003e$completed_No\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e;\n        \u003cspan class=\"pl-c1\"\u003eprint\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMarkingDup completed!!! No.\u003cspan class=\"pl-smi\"\u003e$completed_No\u003c/span\u003e : \u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0]\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][1]\u003cspan class=\"pl-cce\"\u003e\\n\\n\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e;\n}\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003echdir \"../\";\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.3.2 Scripts Usage Example\u003c/h3\u003e\u003ca id=\"user-content-232-scripts-usage-example\" class=\"anchor\" aria-label=\"Permalink: 2.3.2 Scripts Usage Example\" href=\"#232-scripts-usage-example\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFor simple usage or script test, we could directly run the perl script with index paramters. It is highly recommended to record all the log information by redirecting them to another log file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe example command used to call perl script for simple test usage could be shown as follows. The command will call the scripts to do picard duplicates marking on the first sample only and store the log information into the subfoler \u003cem\u003elog\u003c/em\u003e.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"perl run_mark_duplicates.pl list_SP_fq_bwa.txt 0 0 2\u0026gt;./log/log.md_0_0\"\u003e\u003cpre\u003eperl run_mark_duplicates.pl list_SP_fq_bwa.txt 0 0 \u003cspan class=\"pl-k\"\u003e2\u0026gt;\u003c/span\u003e./log/log.md_0_0\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part1-Sequencing-Data-QC/slurm.md_all\"\u003e\u003ccode\u003eScript3: slurm.md_all(Slurm script)\u003c/code\u003e\u003c/a\u003e\u003c/strong\u003e\nFor the slurm cluster users to run multiple samples, the example sbatch script could be arranged as follows.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash\n##SBATCH -J md24  #Slurm job name\n#SBATCH --mail-user=njutangjihong@gmail.com \n#SBATCH --mail-type=end\n##SBATCH -p x-gpu-share\n##SBATCH -o tra_MD.out\n##SBATCH -e tra_MD.err\n##SBATCH -N 1 -n 24\n#SBATCH --cpus-per-task=20\n#SBATCH --exclusive\n\nperl run_mark_duplicates.pl list_SP_fq_bwa.txt ./ $1 $2 2\u0026gt;./log/log.md_$1_$2\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -J md24  #Slurm job name\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=njutangjihong@gmail.com \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=end\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -p x-gpu-share\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -o tra_MD.out\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -e tra_MD.err\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -N 1 -n 24\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --cpus-per-task=20\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --exclusive\u003c/span\u003e\n\nperl run_mark_duplicates.pl list_SP_fq_bwa.txt ./ \u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e2\u0026gt;\u003c/span\u003e./log/log.md_\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTaken the demo project as example, there are totally 94(0-93) samples in the index files, and we could process them parallelly on four computing nodes as follows:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"sbatch -p x-gpu-share -J md0 slurm.md_all 0 23 \nsbatch -p x-gpu-share -J md24 slurm.md_all 24 47 \nsbatch -p x-gpu-share -J md48 slurm.md_all 48 71\nsbatch -p x-gpu-share -J md72 slurm.md_all 72 93 \"\u003e\u003cpre\u003esbatch -p x-gpu-share -J md0 slurm.md_all 0 23 \nsbatch -p x-gpu-share -J md24 slurm.md_all 24 47 \nsbatch -p x-gpu-share -J md48 slurm.md_all 48 71\nsbatch -p x-gpu-share -J md72 slurm.md_all 72 93 \u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePart3 Somatic Mutation Calling\u003c/h1\u003e\u003ca id=\"user-content-part3-somatic-mutation-calling\" class=\"anchor\" aria-label=\"Permalink: Part3 Somatic Mutation Calling\" href=\"#part3-somatic-mutation-calling\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.1 Setting Up Tools\u003c/h2\u003e\u003ca id=\"user-content-31-setting-up-tools\" class=\"anchor\" aria-label=\"Permalink: 3.1 Setting Up Tools\" href=\"#31-setting-up-tools\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.2 Somatic Mutations Calling with SAVI\u003c/h2\u003e\u003ca id=\"user-content-32-somatic-mutations-calling-with-savi\" class=\"anchor\" aria-label=\"Permalink: 3.2 Somatic Mutations Calling with SAVI\" href=\"#32-somatic-mutations-calling-with-savi\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash\n\nexport LIBRARY_PATH=/scratch/PI/jgwang/dsongad/software/ncurses/lib/:$LIBRARY_PATH\nexport LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/scratch/PI/jgwang/dsongad/software/ncurses/lib/\nexport PATH=/scratch/PI/jgwang/dsongad/software/samtools-1.2:$PATH\nexport PATH=$PATH:/scratch/PI/jgwang/dsongad/software/snpEff\nexport LIBRARY_PATH=/scratch/PI/jgwang/dsongad/software/ncurses/lib/:$LIBRARY_PATH\nexport LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/scratch/PI/jgwang/dsongad/software/ncurses/lib/\n#conda activate savi_hg38_py2.7\n\nOUTPUT=$1\nBD=$2\nT1=$3\n\n\nV=/scratch/PI/jgwang/dsongad/software/my_lib/tmp_vcf_lib\nS=/scratch/PI/jgwang/dsongad/IGCT/run_savi/SAVI\nR=/scratch/PI/jgwang/dsongad/software/my_lib/star_genome_d1_vd1_gtfv22/GRCh38.d1.vd1.fa\n\nfor i in {1..22} {X,Y,M}; do\n\tmkdir -p ${OUTPUT}/${i}\n        ${S}/savi_SONG_allregion.py --bams ${BD},${T1} \\\n                --names Nor,T1 \\\n                --ann GRCh38.p7.RefSeq --memory 40 \\\n                --ref ${R} -c 2:1 \\\n                --region chr${i} -v --superverbose \\\n                --noncoding --mapqual 20 --mindepth 4 \\\n                --outputdir ${OUTPUT}/${i} \\\n                --annvcf ${V}/Cosmic_2020_hg38/Song_hg38_Cosmic.sorted.vcf.gz,${V}/Cosmic_2020_hg38/Song_hg38_ClinVar.sorted.vcf.gz,${V}/Cosmic_2020_hg38/Song_hg38_PancancerGermline.sorted.vcf.gz,${V}/Cosmic_2020_hg38/Song_hg38_cbio.vcf.gz,${V}/normal_hg38/Song_hg38_All_SNP.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_TOPMED.sorted.vcf.gz,${V}/normal_hg38/IGCTfamB.sorted.vcf.gz,${V}/normal_hg38/IGCTpatB.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_GATKnormal.vcf.gz,${V}/normal_hg38/Song_hg38_HGDP_1KG.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_MuTectPON.vcf.gz,${V}/normal_hg38/Song_hg38_dbSNP_ALFA.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_gnomAD.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_219normals.vcf.gz,${V}/normal_hg38/Song_China_map.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_ExAC.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_NGDC_SNP.vcf.gz,${V}/normal_hg38/Song_hg38_meganormal.vcf.gz 1\u0026gt; ${OUTPUT}/${i}/err.out 2\u0026gt;${OUTPUT}/${i}/err.log \u0026amp;\ndone\nwait\n\necho SAVI done!\n\necho Start merging...\n\nO1=${OUTPUT}/merge.report.coding.PDfilter.txt\nO2=${OUTPUT}/merge.report.coding.txt\n\n\nhead -n 1 ${OUTPUT}/1/report/report.coding.PDfilter.txt \u0026gt; ${O1}\nhead -n 1 ${OUTPUT}/1/report/report.coding.txt \u0026gt; ${O2}\n\nfor i in {1..22} {X,Y,M}; do\n        F1=${OUTPUT}/${i}/report/report.coding.PDfilter.txt\n\tF2=${OUTPUT}/${i}/report/report.coding.txt\n\t\n        if [ -e ${F1} ]; then\n                tail -n +2 ${F1} \u0026gt;\u0026gt; ${O1}\n        else\n                echo ${F1} does not exist\n        fi\n\n        if [ -e ${F2} ]; then\n                tail -n +2 ${F2} \u0026gt;\u0026gt; ${O2}\n        else\n                echo ${F2} does not exist\n        fi\n\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003eexport\u003c/span\u003e LIBRARY_PATH=/scratch/PI/jgwang/dsongad/software/ncurses/lib/:\u003cspan class=\"pl-smi\"\u003e$LIBRARY_PATH\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eexport\u003c/span\u003e LD_LIBRARY_PATH=\u003cspan class=\"pl-smi\"\u003e$LD_LIBRARY_PATH\u003c/span\u003e:/scratch/PI/jgwang/dsongad/software/ncurses/lib/\n\u003cspan class=\"pl-k\"\u003eexport\u003c/span\u003e PATH=/scratch/PI/jgwang/dsongad/software/samtools-1.2:\u003cspan class=\"pl-smi\"\u003e$PATH\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eexport\u003c/span\u003e PATH=\u003cspan class=\"pl-smi\"\u003e$PATH\u003c/span\u003e:/scratch/PI/jgwang/dsongad/software/snpEff\n\u003cspan class=\"pl-k\"\u003eexport\u003c/span\u003e LIBRARY_PATH=/scratch/PI/jgwang/dsongad/software/ncurses/lib/:\u003cspan class=\"pl-smi\"\u003e$LIBRARY_PATH\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eexport\u003c/span\u003e LD_LIBRARY_PATH=\u003cspan class=\"pl-smi\"\u003e$LD_LIBRARY_PATH\u003c/span\u003e:/scratch/PI/jgwang/dsongad/software/ncurses/lib/\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003econda activate savi_hg38_py2.7\u003c/span\u003e\n\nOUTPUT=\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e\nBD=\u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e\nT1=\u003cspan class=\"pl-smi\"\u003e$3\u003c/span\u003e\n\n\nV=/scratch/PI/jgwang/dsongad/software/my_lib/tmp_vcf_lib\nS=/scratch/PI/jgwang/dsongad/IGCT/run_savi/SAVI\nR=/scratch/PI/jgwang/dsongad/software/my_lib/star_genome_d1_vd1_gtfv22/GRCh38.d1.vd1.fa\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e {1..22} {X,Y,M}\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e${OUTPUT}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\n        \u003cspan class=\"pl-smi\"\u003e${S}\u003c/span\u003e/savi_SONG_allregion.py --bams \u003cspan class=\"pl-smi\"\u003e${BD}\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003e${T1}\u003c/span\u003e \\\n                --names Nor,T1 \\\n                --ann GRCh38.p7.RefSeq --memory 40 \\\n                --ref \u003cspan class=\"pl-smi\"\u003e${R}\u003c/span\u003e -c 2:1 \\\n                --region chr\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e -v --superverbose \\\n                --noncoding --mapqual 20 --mindepth 4 \\\n                --outputdir \u003cspan class=\"pl-smi\"\u003e${OUTPUT}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e \\\n                --annvcf \u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/Cosmic_2020_hg38/Song_hg38_Cosmic.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/Cosmic_2020_hg38/Song_hg38_ClinVar.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/Cosmic_2020_hg38/Song_hg38_PancancerGermline.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/Cosmic_2020_hg38/Song_hg38_cbio.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_hg38_All_SNP.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_hg38_TOPMED.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/IGCTfamB.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/IGCTpatB.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_hg38_GATKnormal.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_hg38_HGDP_1KG.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_hg38_MuTectPON.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_hg38_dbSNP_ALFA.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_hg38_gnomAD.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_hg38_219normals.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_China_map.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_hg38_ExAC.sorted.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_hg38_NGDC_SNP.vcf.gz,\u003cspan class=\"pl-smi\"\u003e${V}\u003c/span\u003e/normal_hg38/Song_hg38_meganormal.vcf.gz \u003cspan class=\"pl-k\"\u003e1\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${OUTPUT}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e/err.out \u003cspan class=\"pl-k\"\u003e2\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e${OUTPUT}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e/err.log \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003ewait\u003c/span\u003e\n\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e SAVI done\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e\n\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e Start merging...\n\nO1=\u003cspan class=\"pl-smi\"\u003e${OUTPUT}\u003c/span\u003e/merge.report.coding.PDfilter.txt\nO2=\u003cspan class=\"pl-smi\"\u003e${OUTPUT}\u003c/span\u003e/merge.report.coding.txt\n\n\nhead -n 1 \u003cspan class=\"pl-smi\"\u003e${OUTPUT}\u003c/span\u003e/1/report/report.coding.PDfilter.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${O1}\u003c/span\u003e\nhead -n 1 \u003cspan class=\"pl-smi\"\u003e${OUTPUT}\u003c/span\u003e/1/report/report.coding.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${O2}\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e {1..22} {X,Y,M}\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n        F1=\u003cspan class=\"pl-smi\"\u003e${OUTPUT}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e/report/report.coding.PDfilter.txt\n\tF2=\u003cspan class=\"pl-smi\"\u003e${OUTPUT}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e/report/report.coding.txt\n\t\n        \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e [ \u003cspan class=\"pl-k\"\u003e-e\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${F1}\u003c/span\u003e ]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ethen\u003c/span\u003e\n                tail -n +2 \u003cspan class=\"pl-smi\"\u003e${F1}\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${O1}\u003c/span\u003e\n        \u003cspan class=\"pl-k\"\u003eelse\u003c/span\u003e\n                \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${F1}\u003c/span\u003e does not exist\n        \u003cspan class=\"pl-k\"\u003efi\u003c/span\u003e\n\n        \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e [ \u003cspan class=\"pl-k\"\u003e-e\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${F2}\u003c/span\u003e ]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ethen\u003c/span\u003e\n                tail -n +2 \u003cspan class=\"pl-smi\"\u003e${F2}\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${O2}\u003c/span\u003e\n        \u003cspan class=\"pl-k\"\u003eelse\u003c/span\u003e\n                \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${F2}\u003c/span\u003e does not exist\n        \u003cspan class=\"pl-k\"\u003efi\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-perl notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/usr/bin/perl\n\n#please run this job in a new clear fold\ndie \u0026quot;usage: perl run_savi.pl \u0026lt;Data_Index\u0026gt; \u0026lt;START\u0026gt; \u0026lt;END\u0026gt; \\n\u0026quot; if @ARGV!= 3;\n\n$index_table = $ARGV[0];\n\n\nopen INDEX, $index_table;\n$i = 0;\n$id = 0;\nwhile($line = \u0026lt;INDEX\u0026gt;){\n\tchomp($line);\n\tmy @temp=split('\\t',$line);\n\t$sample[$i][0] = $temp[1];\n\t$i++;\n}\nclose INDEX;\n\n#`mkdir Dir_tmp_1`;\n#chdir \u0026quot;./Dir_tmp_1\u0026quot;;\n#\n$start = $ARGV[1];\n$end = $ARGV[2];\n\nfor($j=$start;$j\u0026lt;=$end;$j++){\n\t`sh do_savi_WES_2sample_GRCh38.p7.RefSeq.sh  $sample[$j][0] ../all_mapping_hg38/$sample[$j][0]_B_WES_hg38.sorted.MD.bam ../all_mapping_hg38/$sample[$j][0]_T_WES_hg38.sorted.MD.bam`;\n\t$completed_No = $j;\n        print \u0026quot;bwa-MEM completed!!! No.$completed_No : $sample[$j][0]\\n\u0026quot;;\n}\n#chdir \u0026quot;../\u0026quot;;\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e!/usr/bin/perl\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eplease run this job in a new clear fold\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edie\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eusage: perl run_savi.pl \u0026lt;Data_Index\u0026gt; \u0026lt;START\u0026gt; \u0026lt;END\u0026gt; \u003cspan class=\"pl-cce\"\u003e\\n\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e@ARGV\u003c/span\u003e!= 3;\n\n\u003cspan class=\"pl-smi\"\u003e$index_table\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[0];\n\n\n\u003cspan class=\"pl-c1\"\u003eopen\u003c/span\u003e INDEX, \u003cspan class=\"pl-smi\"\u003e$index_table\u003c/span\u003e;\n\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e = 0;\n\u003cspan class=\"pl-smi\"\u003e$id\u003c/span\u003e = 0;\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e = \u0026lt;INDEX\u0026gt;){\n\t\u003cspan class=\"pl-c1\"\u003echomp\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e);\n\t\u003cspan class=\"pl-k\"\u003emy\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e@temp\u003c/span\u003e=\u003cspan class=\"pl-c1\"\u003esplit\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003e$line\u003c/span\u003e);\n\t\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e][0] = \u003cspan class=\"pl-smi\"\u003e$temp\u003c/span\u003e[1];\n\t\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e++;\n}\n\u003cspan class=\"pl-c1\"\u003eclose\u003c/span\u003e INDEX;\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e`mkdir Dir_tmp_1`;\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003echdir \"./Dir_tmp_1\";\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003e$start\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[1];\n\u003cspan class=\"pl-smi\"\u003e$end\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$ARGV\u003c/span\u003e[2];\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e=\u003cspan class=\"pl-smi\"\u003e$start\u003c/span\u003e;\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e\u0026lt;=\u003cspan class=\"pl-smi\"\u003e$end\u003c/span\u003e;\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e++){\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003esh do_savi_WES_2sample_GRCh38.p7.RefSeq.sh  \u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0] ../all_mapping_hg38/\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0]_B_WES_hg38.sorted.MD.bam ../all_mapping_hg38/\u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0]_T_WES_hg38.sorted.MD.bam\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e;\n\t\u003cspan class=\"pl-smi\"\u003e$completed_No\u003c/span\u003e = \u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e;\n        \u003cspan class=\"pl-c1\"\u003eprint\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebwa-MEM completed!!! No.\u003cspan class=\"pl-smi\"\u003e$completed_No\u003c/span\u003e : \u003cspan class=\"pl-smi\"\u003e$sample\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003e$j\u003c/span\u003e][0]\u003cspan class=\"pl-cce\"\u003e\\n\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e;\n}\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003echdir \"../\";\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash\n##SBATCH -J savi24  #Slurm job name\n#SBATCH --mail-user=njutangjihong@gmail.com \n#SBATCH --mail-type=end\n##SBATCH -p x-gpu-share\n##SBATCH -o tra_MD.out\n##SBATCH -e tra_MD.err\n##SBATCH -N 1 -n 24\n#SBATCH --cpus-per-task=20\n#SBATCH --exclusive\n\nperl run_savi.pl list_SP_savi_CNV.txt $1 $2 2\u0026gt; log/log.savi_$1_$2\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -J savi24  #Slurm job name\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=njutangjihong@gmail.com \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=end\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -p x-gpu-share\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -o tra_MD.out\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -e tra_MD.err\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#SBATCH -N 1 -n 24\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --cpus-per-task=20\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --exclusive\u003c/span\u003e\n\nperl run_savi.pl list_SP_savi_CNV.txt \u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e2\u0026gt;\u003c/span\u003e log/log.savi_\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"sbatch -p x-gpu-share -J savi0 slurm.bwa_all 0 11\nsbatch -p x-gpu-share -J savi24 slurm.savi_all 12 23 \nsbatch -p x-gpu-share -J savi48 slurm.savi_all 24 35\nsbatch -p x-gpu-share -J savi72 slurm.savi_all 36 46 \"\u003e\u003cpre\u003esbatch -p x-gpu-share -J savi0 slurm.bwa_all 0 11\nsbatch -p x-gpu-share -J savi24 slurm.savi_all 12 23 \nsbatch -p x-gpu-share -J savi48 slurm.savi_all 24 35\nsbatch -p x-gpu-share -J savi72 slurm.savi_all 36 46 \u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePart4 Copy Number Variation Analysis\u003c/h1\u003e\u003ca id=\"user-content-part4-copy-number-variation-analysis\" class=\"anchor\" aria-label=\"Permalink: Part4 Copy Number Variation Analysis\" href=\"#part4-copy-number-variation-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/CNVkit%20pipeline%208760af4ce2ce4cfaa5410789947fa423/5._109.07.08-2_%25E6%25B9%25AF%25E7%25A1%25AF%25E5%25AE%2589%25E5%258D%259A%25E5%25A3%25AB-51.pdf\"\u003e5._109.07.08-2_湯硯安博士-51.pdf\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://cnvkit.readthedocs.io/en/stable/pipeline.html\" rel=\"nofollow\"\u003ehttps://cnvkit.readthedocs.io/en/stable/pipeline.html\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.1 Preparation of input files\u003c/h2\u003e\u003ca id=\"user-content-41-preparation-of-input-files\" class=\"anchor\" aria-label=\"Permalink: 4.1 Preparation of input files\" href=\"#41-preparation-of-input-files\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.1.1 Conda environment\u003c/h3\u003e\u003ca id=\"user-content-411-conda-environment\" class=\"anchor\" aria-label=\"Permalink: 4.1.1 Conda environment\" href=\"#411-conda-environment\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.1.2 Access file for reference genome\u003c/h3\u003e\u003ca id=\"user-content-412-access-file-for-reference-genome\" class=\"anchor\" aria-label=\"Permalink: 4.1.2 Access file for reference genome\" href=\"#412-access-file-for-reference-genome\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBaits files and Access files,\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor access file, hg38 and hg19 should use different calculated bed files. The calculate command is shown as follows;\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"cnvkit.py access /scratch/PI/jgwang/jtangbd/reference/hg19bwa/hg19.fa \\\n-o access.hg19.bed\"\u003e\u003cpre\u003ecnvkit.py access /scratch/PI/jgwang/jtangbd/reference/hg19bwa/hg19.fa \\\n-o access.hg19.bed\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.1.3 Baits file\u003c/h3\u003e\u003ca id=\"user-content-413-baits-file\" class=\"anchor\" aria-label=\"Permalink: 4.1.3 Baits file\" href=\"#413-baits-file\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.1.4 Target and Antitarget bed files\u003c/h3\u003e\u003ca id=\"user-content-414-target-and-antitarget-bed-files\" class=\"anchor\" aria-label=\"Permalink: 4.1.4 Target and Antitarget bed files\" href=\"#414-target-and-antitarget-bed-files\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"cnvkit.py autobin ../all_mapping_hg19/*B*.bam \\\n-t S07604514_Regions_hg19_220825.bed \\\n-g access.hg19.bed \\\n--target-output-bed SP_20220910_binauto.target.bed \\\n--antitarget-output-bed SP_20220910_binauto.antitarget.bed \\\n2\u0026gt;log/log_autobin_auto\"\u003e\u003cpre\u003ecnvkit.py autobin ../all_mapping_hg19/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eB\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.bam \\\n-t S07604514_Regions_hg19_220825.bed \\\n-g access.hg19.bed \\\n--target-output-bed SP_20220910_binauto.target.bed \\\n--antitarget-output-bed SP_20220910_binauto.antitarget.bed \\\n\u003cspan class=\"pl-k\"\u003e2\u0026gt;\u003c/span\u003elog/log_autobin_auto\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.2 CNVkit Commands\u003c/h2\u003e\u003ca id=\"user-content-42-cnvkit-commands\" class=\"anchor\" aria-label=\"Permalink: 4.2 CNVkit Commands\" href=\"#42-cnvkit-commands\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/Jihong-Tang/Cancer-genomics-analysis/blob/main/cnvkit_workflow.webp\"\u003e\u003cimg src=\"/Jihong-Tang/Cancer-genomics-analysis/raw/main/cnvkit_workflow.webp\" alt=\"CNVkit Pipeline\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.2.1 Coverage\u003c/h3\u003e\u003ca id=\"user-content-421-coverage\" class=\"anchor\" aria-label=\"Permalink: 4.2.1 Coverage\" href=\"#421-coverage\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFor each bam file, we need to calculate the coverage for target and antitarget regions, therefore, there will be two .cnn file as results for each bam file.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash\n\nPathBam=/scratch/PI/jgwang/jtangbd/projects/SP_WES/all_mapping_hg19\nPathCNV=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19\n\nfor i in `cut -f2 list_SP_savi_CNV.txt`;do\n        #i=`echo $n | cut -d \u0026quot;\\t\u0026quot; -f 2`\n        cnvkit.py coverage ${PathBam}/$i\\_B_WES_hg19.sorted.MD.bam SP_20220910_binauto.target.bed -p 20 -o ${PathCNV}/coverage/$i\\_B.target.cnn\ndone\n\nfor i in `cut -f2 list_SP_savi_CNV.txt`;do\n        #i=`echo $n | cut -d \u0026quot;\\t\u0026quot; -f 2`\n        cnvkit.py coverage ${PathBam}/$i\\_B_WES_hg19.sorted.MD.bam SP_20220910_binauto.antitarget.bed -p 20 -o ${PathCNV}/coverage/$i\\_B.antitarget.cnn\ndone\n\nfor i in `cut -f2 list_SP_savi_CNV.txt`;do\n        #i=`echo $n | cut -d \u0026quot;\\t\u0026quot; -f 2`\n        cnvkit.py coverage ${PathBam}/$i\\_T_WES_hg19.sorted.MD.bam SP_20220910_binauto.target.bed -p 20 -o ${PathCNV}/coverage/$i\\_T.target.cnn\ndone\n\nfor i in `cut -f2 list_SP_savi_CNV.txt`;do\n        #i=`echo $n | cut -d \u0026quot;\\t\u0026quot; -f 2`\n        cnvkit.py coverage ${PathBam}/$i\\_T_WES_hg19.sorted.MD.bam SP_20220910_binauto.antitarget.bed -p 20 -o ${PathCNV}/coverage/$i\\_T.antitarget.cnn\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\nPathBam=/scratch/PI/jgwang/jtangbd/projects/SP_WES/all_mapping_hg19\nPathCNV=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003ecut -f2 list_SP_savi_CNV.txt\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ei=`echo $n | cut -d \"\\t\" -f 2`\u003c/span\u003e\n        cnvkit.py coverage \u003cspan class=\"pl-smi\"\u003e${PathBam}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eB_WES_hg19.sorted.MD.bam SP_20220910_binauto.target.bed -p 20 -o \u003cspan class=\"pl-smi\"\u003e${PathCNV}\u003c/span\u003e/coverage/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eB.target.cnn\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003ecut -f2 list_SP_savi_CNV.txt\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ei=`echo $n | cut -d \"\\t\" -f 2`\u003c/span\u003e\n        cnvkit.py coverage \u003cspan class=\"pl-smi\"\u003e${PathBam}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eB_WES_hg19.sorted.MD.bam SP_20220910_binauto.antitarget.bed -p 20 -o \u003cspan class=\"pl-smi\"\u003e${PathCNV}\u003c/span\u003e/coverage/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eB.antitarget.cnn\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003ecut -f2 list_SP_savi_CNV.txt\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ei=`echo $n | cut -d \"\\t\" -f 2`\u003c/span\u003e\n        cnvkit.py coverage \u003cspan class=\"pl-smi\"\u003e${PathBam}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eT_WES_hg19.sorted.MD.bam SP_20220910_binauto.target.bed -p 20 -o \u003cspan class=\"pl-smi\"\u003e${PathCNV}\u003c/span\u003e/coverage/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eT.target.cnn\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003ecut -f2 list_SP_savi_CNV.txt\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ei=`echo $n | cut -d \"\\t\" -f 2`\u003c/span\u003e\n        cnvkit.py coverage \u003cspan class=\"pl-smi\"\u003e${PathBam}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eT_WES_hg19.sorted.MD.bam SP_20220910_binauto.antitarget.bed -p 20 -o \u003cspan class=\"pl-smi\"\u003e${PathCNV}\u003c/span\u003e/coverage/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eT.antitarget.cnn\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.2.2 Reference;Fix;Segment;Export\u003c/h3\u003e\u003ca id=\"user-content-422-referencefixsegmentexport\" class=\"anchor\" aria-label=\"Permalink: 4.2.2 Reference;Fix;Segment;Export\" href=\"#422-referencefixsegmentexport\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe total pipeline for step 2 to 5 could be summarized in the following bash scripts\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash\n\nPathCoverage=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19/coverage\nFileReference=/scratch/PI/jgwang/jtangbd/reference/hg19bwa/hg19.fa\n# step0 - make target and antitarget files\n# step1 - cnvkit coverage for all samples\n\nfor i in `cut -f2 list_SP_savi_CNV.txt`;do\n        # step2 - cnvkit reference for paired sample\n        cnvkit.py reference ${PathCoverage}/$i\\_B*.cnn -f ${FileReference} -o ./reference/$i\\_reference.cnn\n\n        # step3 - cnvkit fix\n        cnvkit.py fix  ${PathCoverage}/$i\\_T.target.cnn  ${PathCoverage}/$i\\_T.antitarget.cnn ./reference/$i\\_reference.cnn -i $i -o ./subtraction/$i\\_fix.cnr\n\n        # step4 - cnvkit segment\n        cnvkit.py segment subtraction/$i\\_fix.cnr -p 20 --drop-low-coverage -m cbs -t 0.000001 -o ./segmentation/$i\\_segment.cbs.cns\n\n        # step5 - cnvkit export and transfer results to seg files\n        cnvkit.py call ./segmentation/$i\\_segment.cbs.cns -o ./results/$i\\_call.cbs.cns\n        cnvkit.py export seg ./results/$i\\_call.cbs.cns -o ./results/$i\\_cbs.seg\n\n        #tail -n+2 ./results/$i\\_cbs.seg \u0026gt;\u0026gt; ./results/CNV_CGPA_103samples_pair_ref.seg\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\nPathCoverage=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19/coverage\nFileReference=/scratch/PI/jgwang/jtangbd/reference/hg19bwa/hg19.fa\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e step0 - make target and antitarget files\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e step1 - cnvkit coverage for all samples\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003ecut -f2 list_SP_savi_CNV.txt\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e step2 - cnvkit reference for paired sample\u003c/span\u003e\n        cnvkit.py reference \u003cspan class=\"pl-smi\"\u003e${PathCoverage}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eB\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.cnn -f \u003cspan class=\"pl-smi\"\u003e${FileReference}\u003c/span\u003e -o ./reference/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003ereference.cnn\n\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e step3 - cnvkit fix\u003c/span\u003e\n        cnvkit.py fix  \u003cspan class=\"pl-smi\"\u003e${PathCoverage}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eT.target.cnn  \u003cspan class=\"pl-smi\"\u003e${PathCoverage}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eT.antitarget.cnn ./reference/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003ereference.cnn -i \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e -o ./subtraction/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003efix.cnr\n\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e step4 - cnvkit segment\u003c/span\u003e\n        cnvkit.py segment subtraction/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003efix.cnr -p 20 --drop-low-coverage -m cbs -t 0.000001 -o ./segmentation/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003esegment.cbs.cns\n\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e step5 - cnvkit export and transfer results to seg files\u003c/span\u003e\n        cnvkit.py call ./segmentation/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003esegment.cbs.cns -o ./results/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003ecall.cbs.cns\n        cnvkit.py \u003cspan class=\"pl-k\"\u003eexport\u003c/span\u003e seg ./results/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003ecall.cbs.cns -o ./results/\u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003ecbs.seg\n\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etail -n+2 ./results/$i\\_cbs.seg \u0026gt;\u0026gt; ./results/CNV_CGPA_103samples_pair_ref.seg\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.3 CNV Profile Generation\u003c/h2\u003e\u003ca id=\"user-content-43-cnv-profile-generation\" class=\"anchor\" aria-label=\"Permalink: 4.3 CNV Profile Generation\" href=\"#43-cnv-profile-generation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Generate hg19 gene lists from gencode gtf files \ncat gencode.v41lift37.annotation.gtf | sed '/^#/d' | awk '$3==\u0026quot;gene\u0026quot;' | \\\nawk -F \u0026quot;\\t\u0026quot; '{split($9, a, \u0026quot;;\u0026quot;); OFS=\u0026quot;\\t\u0026quot;; print $1, $4, $5, $7, a[2], a[3]}' | \\\nawk -F \u0026quot;\\t\u0026quot; '{split($5, b, \u0026quot; \u0026quot;); split($6, c, \u0026quot; \u0026quot;); OFS=\u0026quot;\\t\u0026quot;; print $1,$2,$3,$4,b[2],c[2]}' | \\\nsed 's/\u0026quot;//g' \u0026gt; GRCh37_gencode_v41_genes.txt\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Generate hg19 gene lists from gencode gtf files \u003c/span\u003e\ncat gencode.v41lift37.annotation.gtf \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/^#/d\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e awk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e$3==\"gene\"\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \\\nawk -F \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e{split($9, a, \";\"); OFS=\"\\t\"; print $1, $4, $5, $7, a[2], a[3]}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \\\nawk -F \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e{split($5, b, \" \"); split($6, c, \" \"); OFS=\"\\t\"; print $1,$2,$3,$4,b[2],c[2]}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \\\nsed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/\"//g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e GRCh37_gencode_v41_genes.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Generate Protein coding genes without sex chromosome genes\ncat GRCh37_gencode_v41_genes.txt | sed '/chr[X|Y]/d' | awk '$5==\u0026quot;protein_coding\u0026quot;' \\\n\u0026gt; GRCh37_gencode_v41_genes_Protein_noSex.txt\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Generate Protein coding genes without sex chromosome genes\u003c/span\u003e\ncat GRCh37_gencode_v41_genes.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/chr[X|Y]/d\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e awk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e$5==\"protein_coding\"\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e GRCh37_gencode_v41_genes_Protein_noSex.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# sort CNV result file for a single sample \ncat GRCh37_gencode_v41_genes_Protein_noSex.txt | sort -k1,1 -k2,2n \u0026gt; GRCh37_gencode_v41_genes_Protein_noSex_sort.txt\n\ncat ../results/P370376_cbs.seg | awk '{OFS=\u0026quot;\\t\u0026quot;; print $2,$3,$4,$5,$6,$1}' | \\\nsort -k1,1 -k2,2n \u0026gt; P370376_sort.bed\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e sort CNV result file for a single sample \u003c/span\u003e\ncat GRCh37_gencode_v41_genes_Protein_noSex.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -k1,1 -k2,2n \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e GRCh37_gencode_v41_genes_Protein_noSex_sort.txt\n\ncat ../results/P370376_cbs.seg \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e awk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e{OFS=\"\\t\"; print $2,$3,$4,$5,$6,$1}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \\\nsort -k1,1 -k2,2n \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e P370376_sort.bed\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Map for single sample\nbedtools map -a GRCh37_gencode_v41_genes_Protein_noSex_sort.txt \\\n-b P370376_sort.bed -c 5 | awk '{print $6\u0026quot;\\t\u0026quot;$7}' | sort -k1 | \\\nsed \u0026quot;1s/^/gene\\t$var1\\n/\u0026quot; \u0026gt; P370376_CNVprofile.txt\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Map for single sample\u003c/span\u003e\nbedtools map -a GRCh37_gencode_v41_genes_Protein_noSex_sort.txt \\\n-b P370376_sort.bed -c 5 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e awk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e{print $6\"\\t\"$7}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -k1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \\\nsed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e1s/^/gene\\t\u003cspan class=\"pl-smi\"\u003e$var1\u003c/span\u003e\\n/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e P370376_CNVprofile.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe total bash script:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash\n\nPathSeg=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19/results\n\nfor i in `cut -f2 list_SP_savi_CNV.txt`;do\n        #i=`echo $n | cut -d \u0026quot;\\t\u0026quot; -f 2`\n        cat ${PathSeg}/${i}\\_cbs.seg | awk '{OFS=\u0026quot;\\t\u0026quot;; print $2,$3,$4,$5,$6,$1}' | \\\n        sort -k1,1 -k2,2n \u0026gt; ./sort/${i}\\_sort.bed\n        # Map for single sample\n        var1=${i}\n        bedtools map -a GRCh37_gencode_v41_genes_Protein_noSex_sort.txt -b ./sort/${i}\\_sort.bed -c 5 | \\\n        awk '{print $6\u0026quot;\\t\u0026quot;$7}' | sort -k1 | \\sed \u0026quot;1s/^/gene\\t$var1\\n/\u0026quot; \u0026gt; ./profile/${i}\\_CNVprofile.txt\ndone\n\nmkdir -p tmp\ncut -f1 ./profile/P370376_CNVprofile.txt \u0026gt; gn.txt \nfor i in `cut -f2 list_SP_savi_CNV.txt`;do\n        cut -f2 ./profile/${i}\\_CNVprofile.txt \u0026gt; ./tmp/tmp.${i}\ndone\npaste -d \u0026quot;\\t\u0026quot; ./tmp/tmp* \u0026gt; tmp.txt \npaste -d \u0026quot;\\t\u0026quot; gn.txt tmp.txt \u0026gt; SP_46samples_CNVPatients.txt\nrm gn.txt tmp.txt\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\nPathSeg=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19/results\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003ecut -f2 list_SP_savi_CNV.txt\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ei=`echo $n | cut -d \"\\t\" -f 2`\u003c/span\u003e\n        cat \u003cspan class=\"pl-smi\"\u003e${PathSeg}\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003ecbs.seg \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e awk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e{OFS=\"\\t\"; print $2,$3,$4,$5,$6,$1}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \\\n        sort -k1,1 -k2,2n \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e ./sort/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003esort.bed\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Map for single sample\u003c/span\u003e\n        var1=\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\n        bedtools map -a GRCh37_gencode_v41_genes_Protein_noSex_sort.txt -b ./sort/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003esort.bed -c 5 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \\\n        awk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e{print $6\"\\t\"$7}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -k1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \u003cspan class=\"pl-cce\"\u003e\\s\u003c/span\u003eed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e1s/^/gene\\t\u003cspan class=\"pl-smi\"\u003e$var1\u003c/span\u003e\\n/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e ./profile/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eCNVprofile.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\nmkdir -p tmp\ncut -f1 ./profile/P370376_CNVprofile.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e gn.txt \n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003ecut -f2 list_SP_savi_CNV.txt\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n        cut -f2 ./profile/\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\_\u003c/span\u003eCNVprofile.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e ./tmp/tmp.\u003cspan class=\"pl-smi\"\u003e${i}\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\npaste -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ./tmp/tmp\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e tmp.txt \npaste -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e gn.txt tmp.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e SP_46samples_CNVPatients.txt\nrm gn.txt tmp.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAuthor\u003c/h1\u003e\u003ca id=\"user-content-author\" class=\"anchor\" aria-label=\"Permalink: Author\" href=\"#author\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eJihong Tang \u0026lt;\u003ca href=\"mailto:jtangbd@connect.ust.hk\"\u003ejtangbd@connect.ust.hk\u003c/a\u003e\u0026gt;instructed by @\u003ca href=\"https://github.com/JiguangWang\"\u003eProf. Jiguang Wang\u003c/a\u003e, @\u003ca href=\"https://github.com/qhmu\"\u003eDr. Quanhua Mu\u003c/a\u003e and @\u003ca href=\"https://github.com/ForceField17\"\u003eDong Song\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eReference\u003c/h1\u003e\u003ca id=\"user-content-reference\" class=\"anchor\" aria-label=\"Permalink: Reference\" href=\"#reference\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ePart0 Introduction and Preparation\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://docs.conda.io/projects/conda/en/latest/user-guide/index.html\" rel=\"nofollow\"\u003eConda User Guide\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://docs.brew.sh/\" rel=\"nofollow\"\u003eBrew User Guide\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/twbattaglia/RNAseq-workflow\"\u003eTom Battaglia's RNAseq-workflow\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003ePart1 Sequencing Data Quality Control\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eShifu Chen, Yanqing Zhou, Yaru Chen, Jia Gu; fastp: an ultra-fast all-in-one FASTQ preprocessor, Bioinformatics, Volume 34, Issue 17, 1 September 2018, Pages i884–i890, \u003ca href=\"https://doi.org/10.1093/bioinformatics/bty560\" rel=\"nofollow\"\u003ehttps://doi.org/10.1093/bioinformatics/bty560\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003ePart2 Genome Alignment\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eLi H. and Durbin R. (2009) Fast and accurate short read alignment with Burrows-Wheeler Transform. Bioinformatics, 25:1754-60. [PMID: 19451168]\u003c/li\u003e\n\u003cli\u003e“Picard Toolkit.” 2019. Broad Institute, GitHub Repository. \u003ca href=\"https://broadinstitute.github.io/picard/\" rel=\"nofollow\"\u003ehttps://broadinstitute.github.io/picard/\u003c/a\u003e; Broad Institute\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003ePart3 Somatic Mutation Calling\u003c/li\u003e\n\u003cli\u003ePart4 Copy Number Variation Analysis\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"Table of Contents","anchor":"table-of-contents","htmlText":"Table of Contents"},{"level":1,"text":"Name","anchor":"name","htmlText":"Name"},{"level":1,"text":"Purpose","anchor":"purpose","htmlText":"Purpose"},{"level":1,"text":"Part0 Introduction and Preparation","anchor":"part0-introduction-and-preparation","htmlText":"Part0 Introduction and Preparation"},{"level":2,"text":"0.1 Introduction","anchor":"01-introduction","htmlText":"0.1 Introduction"},{"level":2,"text":"0.2 Getting Setup","anchor":"02-getting-setup","htmlText":"0.2 Getting Setup"},{"level":3,"text":"0.2.1 Installing Conda Environment","anchor":"021-installing-conda-environment","htmlText":"0.2.1 Installing Conda Environment"},{"level":3,"text":"0.2.2 Installing Homebrew (Alternative choice)","anchor":"022-installing-homebrew-alternative-choice","htmlText":"0.2.2 Installing Homebrew (Alternative choice)"},{"level":3,"text":"0.2.3 Setting Up the Folder Structure","anchor":"023-setting-up-the-folder-structure","htmlText":"0.2.3 Setting Up the Folder Structure"},{"level":3,"text":"0.2.4 Creating Cohort Index Files","anchor":"024-creating-cohort-index-files","htmlText":"0.2.4 Creating Cohort Index Files"},{"level":1,"text":"Part1 Sequencing Data Quality Control","anchor":"part1-sequencing-data-quality-control","htmlText":"Part1 Sequencing Data Quality Control"},{"level":2,"text":"1.1 Setting Up Tools","anchor":"11-setting-up-tools","htmlText":"1.1 Setting Up Tools"},{"level":2,"text":"1.2 FastQ Files Preparation","anchor":"12-fastq-files-preparation","htmlText":"1.2 FastQ Files Preparation"},{"level":2,"text":"1.3 Automatic QC with fastp","anchor":"13-automatic-qc-with-fastp","htmlText":"1.3 Automatic QC with fastp"},{"level":3,"text":"1.3.1 Working Scripts","anchor":"131-working-scripts","htmlText":"1.3.1 Working Scripts"},{"level":3,"text":"1.3.2 Scripts Usage Examples","anchor":"132-scripts-usage-examples","htmlText":"1.3.2 Scripts Usage Examples"},{"level":1,"text":"Part2 Genome Alignment","anchor":"part2-genome-alignment","htmlText":"Part2 Genome Alignment"},{"level":2,"text":"2.1 Setting Up Tools","anchor":"21-setting-up-tools","htmlText":"2.1 Setting Up Tools"},{"level":2,"text":"2.2 Genome Alignment with bwa","anchor":"22-genome-alignment-with-bwa","htmlText":"2.2 Genome Alignment with bwa"},{"level":3,"text":"2.2.1 Working Scripts","anchor":"221-working-scripts","htmlText":"2.2.1 Working Scripts"},{"level":3,"text":"2.2.2 Scripts Usage Example","anchor":"222-scripts-usage-example","htmlText":"2.2.2 Scripts Usage Example"},{"level":2,"text":"2.3 Mark Duplicates with Picard","anchor":"23-mark-duplicates-with-picard","htmlText":"2.3 Mark Duplicates with Picard"},{"level":3,"text":"2.3.1 Working Scripts","anchor":"231-working-scripts","htmlText":"2.3.1 Working Scripts"},{"level":3,"text":"2.3.2 Scripts Usage Example","anchor":"232-scripts-usage-example","htmlText":"2.3.2 Scripts Usage Example"},{"level":1,"text":"Part3 Somatic Mutation Calling","anchor":"part3-somatic-mutation-calling","htmlText":"Part3 Somatic Mutation Calling"},{"level":2,"text":"3.1 Setting Up Tools","anchor":"31-setting-up-tools","htmlText":"3.1 Setting Up Tools"},{"level":2,"text":"3.2 Somatic Mutations Calling with SAVI","anchor":"32-somatic-mutations-calling-with-savi","htmlText":"3.2 Somatic Mutations Calling with SAVI"},{"level":1,"text":"Part4 Copy Number Variation Analysis","anchor":"part4-copy-number-variation-analysis","htmlText":"Part4 Copy Number Variation Analysis"},{"level":2,"text":"4.1 Preparation of input files","anchor":"41-preparation-of-input-files","htmlText":"4.1 Preparation of input files"},{"level":3,"text":"4.1.1 Conda environment","anchor":"411-conda-environment","htmlText":"4.1.1 Conda environment"},{"level":3,"text":"4.1.2 Access file for reference genome","anchor":"412-access-file-for-reference-genome","htmlText":"4.1.2 Access file for reference genome"},{"level":3,"text":"4.1.3 Baits file","anchor":"413-baits-file","htmlText":"4.1.3 Baits file"},{"level":3,"text":"4.1.4 Target and Antitarget bed files","anchor":"414-target-and-antitarget-bed-files","htmlText":"4.1.4 Target and Antitarget bed files"},{"level":2,"text":"4.2 CNVkit Commands","anchor":"42-cnvkit-commands","htmlText":"4.2 CNVkit Commands"},{"level":3,"text":"4.2.1 Coverage","anchor":"421-coverage","htmlText":"4.2.1 Coverage"},{"level":3,"text":"4.2.2 Reference;Fix;Segment;Export","anchor":"422-referencefixsegmentexport","htmlText":"4.2.2 Reference;Fix;Segment;Export"},{"level":2,"text":"4.3 CNV Profile Generation","anchor":"43-cnv-profile-generation","htmlText":"4.3 CNV Profile Generation"},{"level":1,"text":"Author","anchor":"author","htmlText":"Author"},{"level":1,"text":"Reference","anchor":"reference","htmlText":"Reference"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2FJihong-Tang%2FCancer-genomics-analysis"}},{"displayName":"LICENSE","repoName":"Cancer-genomics-analysis","refName":"main","path":"LICENSE","preferredFileType":"license","tabName":"GPL-3.0","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2FJihong-Tang%2FCancer-genomics-analysis"}}],"overviewFilesProcessingTime":101.704445}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
.bhqztV{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.bhqztV span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.bhqztV:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.bhqztV:focus{outline:2px solid transparent;}/*!sc*/
.bhqztV:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.bhqztV:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.bhqztV::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:transparent;border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.bhqztV::after{background-color:transparent;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,bhqztV,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/Jihong-Tang/Cancer-genomics-analysis/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/Jihong-Tang/Cancer-genomics-analysis/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/Jihong-Tang/Cancer-genomics-analysis/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/Jihong-Tang/Cancer-genomics-analysis/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/Jihong-Tang/Cancer-genomics-analysis/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           17 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/Jihong-Tang/Cancer-genomics-analysis/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="scripts, (Directory)" class="Link--primary" href="/Jihong-Tang/Cancer-genomics-analysis/tree/main/scripts" title="scripts">
                          scripts
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="scripts, (Directory)" class="Link--primary" href="/Jihong-Tang/Cancer-genomics-analysis/tree/main/scripts" title="scripts">
                          scripts
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="cnvkit_workflow.webp, (File)" class="Link--primary" href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/cnvkit_workflow.webp" title="cnvkit_workflow.webp">
                          cnvkit_workflow.webp
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="cnvkit_workflow.webp, (File)" class="Link--primary" href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/cnvkit_workflow.webp" title="cnvkit_workflow.webp">
                          cnvkit_workflow.webp
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-law" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="GPL-3.0 license">
                        GPL-3.0 license
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Table of Contents
                     </h1>
                     <a aria-label="Permalink: Table of Contents" class="anchor" href="#table-of-contents" id="user-content-table-of-contents">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <a href="#table-of-contents">
                       Table of Contents
                      </a>
                     </li>
                     <li>
                      <a href="#name">
                       Name
                      </a>
                     </li>
                     <li>
                      <a href="#purpose">
                       Purpose
                      </a>
                     </li>
                     <li>
                      <a href="#part0-introduction-and-preparation">
                       Part0 Introduction and Preparation
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#01-introduction">
                         0.1 Introduction
                        </a>
                       </li>
                       <li>
                        <a href="#02-getting-setup">
                         0.2 Getting Setup
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#021-installing-conda-environment">
                           0.2.1 Installing Conda Environment
                          </a>
                         </li>
                         <li>
                          <a href="#022-installing-homebrew-alternative-choice">
                           0.2.2 Installing Homebrew (Alternative choice)
                          </a>
                         </li>
                         <li>
                          <a href="#023-setting-up-the-folder-structure">
                           0.2.3 Setting Up the Folder Structure
                          </a>
                         </li>
                         <li>
                          <a href="#024-creating-cohort-index-files">
                           0.2.4 Creating Cohort Index Files
                          </a>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#part1-sequencing-data-quality-control">
                       Part1 Sequencing Data Quality Control
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#11-setting-up-tools">
                         1.1 Setting Up Tools
                        </a>
                       </li>
                       <li>
                        <a href="#12-fastq-files-preparation">
                         1.2 FastQ Files Preparation
                        </a>
                       </li>
                       <li>
                        <a href="#13-automatic-qc-with-fastp">
                         1.3 Automatic QC with fastp
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#131-working-scripts">
                           1.3.1 Working Scripts
                          </a>
                         </li>
                         <li>
                          <a href="#132-scripts-usage-examples">
                           1.3.2 Scripts Usage Examples
                          </a>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#part2-genome-alignment">
                       Part2 Genome Alignment
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#21-setting-up-tools">
                         2.1 Setting Up Tools
                        </a>
                       </li>
                       <li>
                        <a href="#22-genome-alignment-with-bwa">
                         2.2 Genome Alignment with bwa
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#221-working-scripts">
                           2.2.1 Working Scripts
                          </a>
                         </li>
                         <li>
                          <a href="#222-scripts-usage-example">
                           2.2.2 Scripts Usage Example
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#23-mark-duplicates-with-picard">
                         2.3 Mark Duplicates with Picard
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#231-working-scripts">
                           2.3.1 Working Scripts
                          </a>
                         </li>
                         <li>
                          <a href="#232-scripts-usage-example">
                           2.3.2 Scripts Usage Example
                          </a>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#part3-somatic-mutation-calling">
                       Part3 Somatic Mutation Calling
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#31-setting-up-tools">
                         3.1 Setting Up Tools
                        </a>
                       </li>
                       <li>
                        <a href="#32-somatic-mutations-calling-with-savi">
                         3.2 Somatic Mutations Calling with SAVI
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#part4-copy-number-variation-analysis">
                       Part4 Copy Number Variation Analysis
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#41-preparation-of-input-files">
                         4.1 Preparation of input files
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#411-conda-environment">
                           4.1.1 Conda environment
                          </a>
                         </li>
                         <li>
                          <a href="#412-access-file-for-reference-genome">
                           4.1.2 Access file for reference genome
                          </a>
                         </li>
                         <li>
                          <a href="#413-baits-file">
                           4.1.3 Baits file
                          </a>
                         </li>
                         <li>
                          <a href="#414-target-and-antitarget-bed-files">
                           4.1.4 Target and Antitarget bed files
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#42-cnvkit-commands">
                         4.2 CNVkit Commands
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#421-coverage">
                           4.2.1 Coverage
                          </a>
                         </li>
                         <li>
                          <a href="#422-referencefixsegmentexport">
                           4.2.2 Reference;Fix;Segment;Export
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#43-cnv-profile-generation">
                         4.3 CNV Profile Generation
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#author">
                       Author
                      </a>
                     </li>
                     <li>
                      <a href="#reference">
                       Reference
                      </a>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Name
                     </h1>
                     <a aria-label="Permalink: Name" class="anchor" href="#name" id="user-content-name">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Cancer Genomics Analysis Pipeline(CGAP): Notes for the computational pipeline focusing on the Cancer Genomics DNA-seq data
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Purpose
                     </h1>
                     <a aria-label="Permalink: Purpose" class="anchor" href="#purpose" id="user-content-purpose">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      Make record for the in-house DNA-seq data analysis pipeline including several modules for future usage:
                      <ul dir="auto">
                       <li>
                        Sequencing data Quality Control
                       </li>
                       <li>
                        Genome Alignment
                       </li>
                       <li>
                        Somatic Mutation Calling
                       </li>
                       <li>
                        Copy Number Variation Analysis
                       </li>
                      </ul>
                     </li>
                     <li>
                      Take notes about cancer genomics data analysis for future review.
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Part0 Introduction and Preparation
                     </h1>
                     <a aria-label="Permalink: Part0 Introduction and Preparation" class="anchor" href="#part0-introduction-and-preparation" id="user-content-part0-introduction-and-preparation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      0.1 Introduction
                     </h2>
                     <a aria-label="Permalink: 0.1 Introduction" class="anchor" href="#01-introduction" id="user-content-01-introduction">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The in house cancer genomics analysis pipeline starts from the quanlity control of DNA-seq sequencing data. After aligning the sequencing data to different versions of reference genomes, the pipeline identifies the somatic mutations and copy number variations based on the comparisons between normal/blood and tumor tissues.
                    </p>
                    <p dir="auto">
                     The whole analysis pipeline is implemented across four main procedures based on different bioinformatics tools:
                    </p>
                    <ul dir="auto">
                     <li>
                      Sequencing Data Quality Control -
                      <a href="https://github.com/OpenGene/fastp">
                       fastp
                      </a>
                     </li>
                     <li>
                      Genome Alignment -
                      <a href="https://github.com/lh3/bwa">
                       bwa
                      </a>
                      and
                      <a href="https://broadinstitute.github.io/picard/" rel="nofollow">
                       Picard
                      </a>
                     </li>
                     <li>
                      Somatic Mutations Calling -
                      <a href="https://github.com/WangLabHKUST/SAVI">
                       SAVI
                      </a>
                     </li>
                     <li>
                      CNV Analysis -
                      <a href="https://cnvkit.readthedocs.io/en/stable/" rel="nofollow">
                       CNVkit
                      </a>
                     </li>
                    </ul>
                    <p dir="auto">
                     The pipeline is initially arranged for the whole exome sequencing(WES) data, but with minor updates on the scripts, it can also be used for the whole genome sequencing(WGS) data.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      0.2 Getting Setup
                     </h2>
                     <a aria-label="Permalink: 0.2 Getting Setup" class="anchor" href="#02-getting-setup" id="user-content-02-getting-setup">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      0.2.1 Installing Conda Environment
                     </h3>
                     <a aria-label="Permalink: 0.2.1 Installing Conda Environment" class="anchor" href="#021-installing-conda-environment" id="user-content-021-installing-conda-environment">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Anaconda/Miniconda is one powerful and user-friendly package manager for Python. Miniconda is the smaller version of Anaconda with basic functions.
                    </p>
                    <p dir="auto">
                     Using conda environment management tools, we can not only install and manage Python packages with different verisons, but also create virual environments and install various bioinformatics tools through access to large bioinformatics repositories(eg.
                     <a href="https://bioconda.github.io/" rel="nofollow">
                      Bioconda
                     </a>
                     )
                    </p>
                    <p dir="auto">
                     The methods to download and install conda can be found
                     <a href="https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html" rel="nofollow">
                      here
                     </a>
                     . For example, we can use the following commands to install miniconda on MacOS machines.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# download the Miniconda3 installer to the home directory, only for MacOS
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh -O ~/miniconda.sh

# Run the miniconda installation in slient mode
bash ~/miniconda.sh -b -p $HOME/miniconda

# Choose YES for the question 'Do you wish the installer to initialize Miniconda3 by running conda init?', 
# then the bash profile will be changed automatically.

# Add bioinformatic channels for downloading required packages
conda config --add channels conda-forge
conda config --add channels defaults
conda config --add channels r
conda config --add channels bioconda # important channel to download and manage useful bioinformatics tools" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> download the Miniconda3 installer to the home directory, only for MacOS</span>
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh -O <span class="pl-k">~</span>/miniconda.sh

<span class="pl-c"><span class="pl-c">#</span> Run the miniconda installation in slient mode</span>
bash <span class="pl-k">~</span>/miniconda.sh -b -p <span class="pl-smi">$HOME</span>/miniconda

<span class="pl-c"><span class="pl-c">#</span> Choose YES for the question 'Do you wish the installer to initialize Miniconda3 by running conda init?', </span>
<span class="pl-c"><span class="pl-c">#</span> then the bash profile will be changed automatically.</span>

<span class="pl-c"><span class="pl-c">#</span> Add bioinformatic channels for downloading required packages</span>
conda config --add channels conda-forge
conda config --add channels defaults
conda config --add channels r
conda config --add channels bioconda <span class="pl-c"><span class="pl-c">#</span> important channel to download and manage useful bioinformatics tools</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      0.2.2 Installing Homebrew (Alternative choice)
                     </h3>
                     <a aria-label="Permalink: 0.2.2 Installing Homebrew (Alternative choice)" class="anchor" href="#022-installing-homebrew-alternative-choice" id="user-content-022-installing-homebrew-alternative-choice">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Homebrew is another powerful package manager that can install
                     <em>
                      the stuff you need
                     </em>
                     that Apple (or your Linux system) didn't have. The most well-known example is
                     <em>
                      wget
                     </em>
                     .
                    </p>
                    <p dir="auto">
                     The methods to download and install brew can be found
                     <a href="https://brew.sh/" rel="nofollow">
                      here
                     </a>
                     . For example, we can use the following commands to install brew on MacOS machines.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# download the Homebrew installer and install brew
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> download the Homebrew installer and install brew</span>
/bin/bash -c <span class="pl-s"><span class="pl-pds">"</span><span class="pl-s"><span class="pl-pds">$(</span>curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh<span class="pl-pds">)</span></span><span class="pl-pds">"</span></span></pre>
                    </div>
                    <p dir="auto">
                     In addition, we could install the brew on Linux machines as following, referred from
                     <a href="https://github.com/wang-q/ubuntu#install-linuxbrew">
                      wang-q Ubuntu
                     </a>
                     .
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="echo &quot;==&gt; Install linuxbrew, copy the next *ONE* line to terminal&quot;
bash -c &quot;$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh)&quot;

test -d ~/.linuxbrew &amp;&amp; PATH=&quot;$HOME/.linuxbrew/bin:$HOME/.linuxbrew/sbin:$PATH&quot;
test -d /home/linuxbrew/.linuxbrew &amp;&amp; PATH=&quot;/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:$PATH&quot;

if grep -q -i linuxbrew $HOME/.bashrc; then
    echo &quot;==&gt; .bashrc already contains linuxbrew&quot;
else
    echo &quot;==&gt; Update .bashrc&quot;

    echo &gt;&gt; $HOME/.bashrc
    echo '# Linuxbrew' &gt;&gt; $HOME/.bashrc
    echo &quot;export PATH='$(brew --prefix)/bin:$(brew --prefix)/sbin'&quot;:'&quot;$PATH&quot;' &gt;&gt; $HOME/.bashrc
    echo &quot;export MANPATH='$(brew --prefix)/share/man'&quot;:'&quot;$MANPATH&quot;' &gt;&gt; $HOME/.bashrc
    echo &quot;export INFOPATH='$(brew --prefix)/share/info'&quot;:'&quot;$INFOPATH&quot;' &gt;&gt; $HOME/.bashrc
    echo &quot;export HOMEBREW_NO_ANALYTICS=1&quot; &gt;&gt; $HOME/.bashrc
    echo &quot;export HOMEBREW_NO_AUTO_UPDATE=1&quot; &gt;&gt; $HOME/.bashrc
    echo &gt;&gt; $HOME/.bashrc
fi

source $HOME/.bashrc" dir="auto">
                     <pre><span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>==&gt; Install linuxbrew, copy the next *ONE* line to terminal<span class="pl-pds">"</span></span>
bash -c <span class="pl-s"><span class="pl-pds">"</span><span class="pl-s"><span class="pl-pds">$(</span>curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh<span class="pl-pds">)</span></span><span class="pl-pds">"</span></span>

<span class="pl-c1">test</span> -d <span class="pl-k">~</span>/.linuxbrew <span class="pl-k">&amp;&amp;</span> PATH=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$HOME</span>/.linuxbrew/bin:<span class="pl-smi">$HOME</span>/.linuxbrew/sbin:<span class="pl-smi">$PATH</span><span class="pl-pds">"</span></span>
<span class="pl-c1">test</span> -d /home/linuxbrew/.linuxbrew <span class="pl-k">&amp;&amp;</span> PATH=<span class="pl-s"><span class="pl-pds">"</span>/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:<span class="pl-smi">$PATH</span><span class="pl-pds">"</span></span>

<span class="pl-k">if</span> grep -q -i linuxbrew <span class="pl-smi">$HOME</span>/.bashrc<span class="pl-k">;</span> <span class="pl-k">then</span>
    <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>==&gt; .bashrc already contains linuxbrew<span class="pl-pds">"</span></span>
<span class="pl-k">else</span>
    <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>==&gt; Update .bashrc<span class="pl-pds">"</span></span>

    <span class="pl-c1">echo</span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$HOME</span>/.bashrc
    <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">'</span># Linuxbrew<span class="pl-pds">'</span></span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$HOME</span>/.bashrc
    <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>export PATH='<span class="pl-s"><span class="pl-pds">$(</span>brew --prefix<span class="pl-pds">)</span></span>/bin:<span class="pl-s"><span class="pl-pds">$(</span>brew --prefix<span class="pl-pds">)</span></span>/sbin'<span class="pl-pds">"</span></span>:<span class="pl-s"><span class="pl-pds">'</span>"$PATH"<span class="pl-pds">'</span></span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$HOME</span>/.bashrc
    <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>export MANPATH='<span class="pl-s"><span class="pl-pds">$(</span>brew --prefix<span class="pl-pds">)</span></span>/share/man'<span class="pl-pds">"</span></span>:<span class="pl-s"><span class="pl-pds">'</span>"$MANPATH"<span class="pl-pds">'</span></span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$HOME</span>/.bashrc
    <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>export INFOPATH='<span class="pl-s"><span class="pl-pds">$(</span>brew --prefix<span class="pl-pds">)</span></span>/share/info'<span class="pl-pds">"</span></span>:<span class="pl-s"><span class="pl-pds">'</span>"$INFOPATH"<span class="pl-pds">'</span></span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$HOME</span>/.bashrc
    <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>export HOMEBREW_NO_ANALYTICS=1<span class="pl-pds">"</span></span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$HOME</span>/.bashrc
    <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>export HOMEBREW_NO_AUTO_UPDATE=1<span class="pl-pds">"</span></span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$HOME</span>/.bashrc
    <span class="pl-c1">echo</span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$HOME</span>/.bashrc
<span class="pl-k">fi</span>

<span class="pl-c1">source</span> <span class="pl-smi">$HOME</span>/.bashrc</pre>
                    </div>
                    <p dir="auto">
                     Most of the used softwares and tools can be downloaded through both package managers, and it is highly recommended to use either of them to manage the related softwares and bioinformatics tools.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      0.2.3 Setting Up the Folder Structure
                     </h3>
                     <a aria-label="Permalink: 0.2.3 Setting Up the Folder Structure" class="anchor" href="#023-setting-up-the-folder-structure" id="user-content-023-setting-up-the-folder-structure">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Organizing is essential to proper reproducible research. During the processing and analysis steps, we will get many output files. A simple and clear folder structure is preferred not only for your own future usage, but also for other researchers' repeats.
                    </p>
                    <p dir="auto">
                     The folder structure used in this pipeline will be summarized here.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="── ${project_WES}/
  │   └── reference/                &lt;- Reference genome file (.FASTA)
  │       ├── hg19bwa/              &lt;- hg19 version reference genome files
  │       ├── TCGA_GRCh38_bwa_ref/  &lt;- hg38 version reference genome files
  │ 
  │   └── all_fq/                   &lt;- Working Directory of Sequencing Data QC 
  │       ├── log/                  &lt;- Log from Sequencing Data QC module
  │       ├── qc_res/               &lt;- QC result from Sequencing Data QC module
  │ 
  │   └── all_mapping_hg19/         &lt;- Working Directory of Genome Alignment: hg19 version 
  │       ├── log/                  &lt;- Log from Genome Alignment module
  │       ├── md_metrix/            &lt;- MarkDuplicate metrix from alignment duplicates marking step
  │       ├── md_tmp/               &lt;- Location for temporary files from alignment duplicates marking step
  │       ├── tmp/                  &lt;- Location for temporary files from Genome Alignment module
  │ 
  │   └── all_mapping_hg38/         &lt;- Working Directory of Genome Alignment: hg38 version 
  │       ├── log/                  &lt;- Log from Genome Alignment module
  │       ├── md_metrix/            &lt;- MarkDuplicate metrix from alignment duplicates marking step
  │       ├── md_tmp/               &lt;- Location for temporary files from alignment duplicates marking step
  │       ├── tmp/                  &lt;- Location for temporary files from Genome Alignment module
  │ 
  │   └── savi_hg38/                &lt;- Working Directory of Somatic Mutation Calling: hg38 version 
  │       ├── log/                  &lt;- Log from savi_hg38 module
  │       ├── .../                  &lt;- Other detailed subfolders containing working files, will be shown in future section
  │
  │   └── cnvkit_hg19/              &lt;- Working Directory of CNV Analysis
  │       ├── log/                  &lt;- Log from CNV Analysis module
  │       ├── .../                  &lt;- Other detailed subfolders containing working files, will be shown in future section
  │ " dir="auto">
                     <pre>── <span class="pl-smi">${project_WES}</span>/
  │   └── reference/                <span class="pl-k">&lt;</span>- Reference genome file (.FASTA)
  │       ├── hg19bwa/              <span class="pl-k">&lt;</span>- hg19 version reference genome files
  │       ├── TCGA_GRCh38_bwa_ref/  <span class="pl-k">&lt;</span>- hg38 version reference genome files
  │ 
  │   └── all_fq/                   <span class="pl-k">&lt;</span>- Working Directory of Sequencing Data QC 
  │       ├── log/                  <span class="pl-k">&lt;</span>- Log from Sequencing Data QC module
  │       ├── qc_res/               <span class="pl-k">&lt;</span>- QC result from Sequencing Data QC module
  │ 
  │   └── all_mapping_hg19/         <span class="pl-k">&lt;</span>- Working Directory of Genome Alignment: hg19 version 
  │       ├── log/                  <span class="pl-k">&lt;</span>- Log from Genome Alignment module
  │       ├── md_metrix/            <span class="pl-k">&lt;</span>- MarkDuplicate metrix from alignment duplicates marking step
  │       ├── md_tmp/               <span class="pl-k">&lt;</span>- Location <span class="pl-k">for</span> temporary files from alignment duplicates marking step
  │       ├── tmp/                  <span class="pl-k">&lt;</span>- Location <span class="pl-k">for</span> temporary files from Genome Alignment module
  │ 
  │   └── all_mapping_hg38/         <span class="pl-k">&lt;</span>- Working Directory of Genome Alignment: hg38 version 
  │       ├── log/                  <span class="pl-k">&lt;</span>- Log from Genome Alignment module
  │       ├── md_metrix/            <span class="pl-k">&lt;</span>- MarkDuplicate metrix from alignment duplicates marking step
  │       ├── md_tmp/               <span class="pl-k">&lt;</span>- Location <span class="pl-k">for</span> temporary files from alignment duplicates marking step
  │       ├── tmp/                  <span class="pl-k">&lt;</span>- Location <span class="pl-k">for</span> temporary files from Genome Alignment module
  │ 
  │   └── savi_hg38/                <span class="pl-k">&lt;</span>- Working Directory of Somatic Mutation Calling: hg38 version 
  │       ├── log/                  <span class="pl-k">&lt;</span>- Log from savi_hg38 module
  │       ├── .../                  <span class="pl-k">&lt;</span>- Other detailed subfolders containing working files, will be shown <span class="pl-k">in</span> future section
  │
  │   └── cnvkit_hg19/              <span class="pl-k">&lt;</span>- Working Directory of CNV Analysis
  │       ├── log/                  <span class="pl-k">&lt;</span>- Log from CNV Analysis module
  │       ├── .../                  <span class="pl-k">&lt;</span>- Other detailed subfolders containing working files, will be shown <span class="pl-k">in</span> future section
  │ </pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Important Notes
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      The reference folder contains the reference genome used, two different versions of reference genome file will be covered in this pipeline (hg19 and hg38). For WangLab HPC3 user, the folder can be linked as following:
                      <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="ln -s /scratch/PI/jgwang/jtangbd/reference ${project_WES}/reference/" dir="auto">
                       <pre>ln -s /scratch/PI/jgwang/jtangbd/reference <span class="pl-smi">${project_WES}</span>/reference/</pre>
                      </div>
                      For other users, the reference genomes can be downloaded from the following pages:
                      <a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/" rel="nofollow">
                       hg19
                      </a>
                      ,
                      <a href="https://gdc.cancer.gov/about-data/gdc-data-processing/gdc-reference-files" rel="nofollow">
                       hg38
                      </a>
                     </li>
                     <li>
                      Two different versions of reference genome are included in this pipeline. They are both used to generate different versions of genome alignment result files(BAMs). Afterwards, the bams are used differently for following mutation calling and CNV analysis module based on the reference genome version requirements for different tools.
                     </li>
                     <li>
                      For the Somatic Mutation calling and CNV analysis modules, detailed folder arrangements will be explained in
                      <a href="#part3-somatic-mutation-calling">
                       Part3
                      </a>
                      and
                      <a href="#part4-copy-number-variation-analysis">
                       Part4
                      </a>
                      .
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      0.2.4 Creating Cohort Index Files
                     </h3>
                     <a aria-label="Permalink: 0.2.4 Creating Cohort Index Files" class="anchor" href="#024-creating-cohort-index-files" id="user-content-024-creating-cohort-index-files">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     In order to facilitate the cohort data management and parallel processing, we need to create two index files for the WES/WGS sequencing data in this whole pipeline. Both index files are simple .txt files containing the basic naming information for all the sequencing data.
                    </p>
                    <p dir="auto">
                     <strong>
                      <code>
                       Index File 1st
                      </code>
                     </strong>
                    </p>
                    <p dir="auto">
                     The first file is used for Sequencing Data QC and Genome Aligenment modules, and it contains three columns, representing row number(0-index), raw name prefix of fastq files, and changed name prefix of bam files, respectively.
                    </p>
                    <p dir="auto">
                     Below is the example format, it is expected to contain index rows for both Blood(B) and Tumor(T) samples in this index file. It is recommended to label the blood and tumor samples in specific symbols, such as B and T after the patient number.
                    </p>
                    <p dir="auto">
                     The first column is used for easier arrangement of parallel computing, and detailed information will be provided in the next subsection. The second column is the original name prefix of the fastq files,
and it is highly recommended to rename your fastq files in your own style but noted them well in this index file.
The third column is the changed name prefix you would like to name your bam files. It is highly recommended to contain DNA-seq methods(WES/WGS) and reference genome version(hg19/hg38) in the prefix for further confirmation.
                    </p>
                    <p dir="auto">
                     The demo index file could be found
                     <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/list_SP_fq_bwa.txt">
                      here
                     </a>
                     . It is used for the demo SP project data and contains 47 patients, 94 sequencing data(B + T) files.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="0   P001_B    P001_B_WES_hg38
1   P002_B    P002_B_WES_hg38
2   P003_B    P003_B_WES_hg38
3   P004_B    P004_B_WES_hg38
4   P005_B    P005_B_WES_hg38
5   P001_T    P001_T_WES_hg38
6   P002_T    P002_T_WES_hg38
7   P003_T    P003_T_WES_hg38
8   P004_T    P004_T_WES_hg38
9   P005_T    P005_T_WES_hg38
..." dir="auto">
                     <pre>0   P001_B    P001_B_WES_hg38
1   P002_B    P002_B_WES_hg38
2   P003_B    P003_B_WES_hg38
3   P004_B    P004_B_WES_hg38
4   P005_B    P005_B_WES_hg38
5   P001_T    P001_T_WES_hg38
6   P002_T    P002_T_WES_hg38
7   P003_T    P003_T_WES_hg38
8   P004_T    P004_T_WES_hg38
9   P005_T    P005_T_WES_hg38
...</pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      <code>
                       Index File 2nd
                      </code>
                     </strong>
                    </p>
                    <p dir="auto">
                     The second file is used for Somatic Mutation Calling and CNV Analysis modules, and it contains only two columns, representing row number(0-index), and patient prefix resepectively.
The second index file only needs to contain one index row for each patient, which is different to the first one.
                    </p>
                    <p dir="auto">
                     Below is the example format and the demo index file could be found
                     <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/list_SP_savi_CNV.txt">
                      here
                     </a>
                     .
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="0   P001
1   P002
2   P003
4   P004
5   P005
..." dir="auto">
                     <pre>0   P001
1   P002
2   P003
4   P004
5   P005
...</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Part1 Sequencing Data Quality Control
                     </h1>
                     <a aria-label="Permalink: Part1 Sequencing Data Quality Control" class="anchor" href="#part1-sequencing-data-quality-control" id="user-content-part1-sequencing-data-quality-control">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      1.1 Setting Up Tools
                     </h2>
                     <a aria-label="Permalink: 1.1 Setting Up Tools" class="anchor" href="#11-setting-up-tools" id="user-content-11-setting-up-tools">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The main bioinformatics tool used in this module is fastp. The fastp is a tool designed to provide fast all-in-one preprocessing for FastQ files.
It can automatically filter reads, detect reads adapters, make adapter trimming and output QC results based on the demand. It covers the functions of well known QC tools like FastQC and Trim Galero, and makes the whole preprocessing work automatically in one step, which is very convenient.
More details about fastp could be found in their
                     <a href="https://github.com/OpenGene/fastp">
                      user manual page
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     fastp could be easily downloaded and installed with bioconda:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# note: the fastp version in bioconda may be not the latest
conda install -c bioconda fastp" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> note: the fastp version in bioconda may be not the latest</span>
conda install -c bioconda fastp</pre>
                    </div>
                    <p dir="auto">
                     or download the prebuilt binary versions for usage:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# download the latest build
wget http://opengene.org/fastp/fastp
chmod a+x ./fastp

# or download specified version, i.e. fastp v0.23.1
wget http://opengene.org/fastp/fastp.0.23.1
mv fastp.0.23.1 fastp
chmod a+x ./fastp" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> download the latest build</span>
wget http://opengene.org/fastp/fastp
chmod a+x ./fastp

<span class="pl-c"><span class="pl-c">#</span> or download specified version, i.e. fastp v0.23.1</span>
wget http://opengene.org/fastp/fastp.0.23.1
mv fastp.0.23.1 fastp
chmod a+x ./fastp</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      1.2 FastQ Files Preparation
                     </h2>
                     <a aria-label="Permalink: 1.2 FastQ Files Preparation" class="anchor" href="#12-fastq-files-preparation" id="user-content-12-fastq-files-preparation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     FastQ file is often used to store the high throughput sequencing reads, and contains useful information including quality scores for each base.
Deatiled description about FastQ files could be found
                     <a href="https://compgenomr.github.io/book/fasta-and-fastq-formats.html" rel="nofollow">
                      here
                     </a>
                     . In order to start the genomics analysis,
we need to prepare the fastq files for the project. There are mainly two methods to get the fastq files for your project. The first one is producing your own data by sequencing(eg. sequencing service from companies).
The second one is downloading data from public database, and it is very useful when you would like to take advantage of the previous work from the community.
                    </p>
                    <p dir="auto">
                     However, some datasets in the public database will not contain fastq files, instead, bam files are often contained. In order to control the analysis parameters and pipeline, we could extract the fastq files from the downloaded bam files, and then deal with them with your own pipeline.
In order to do so, we could use several different tools, and
                     <em>
                      SamToFastq
                     </em>
                     in Picard is taken as example here.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/usr/bin/bash

picard=/scratch/PI/jgwang/dsongad/software/Picard/picard.jar


NAME=$1
NEW=$2

# The example script is setted for the HPC3 User in WangLab HKUST
/scratch/PI/jgwang/dsongad/software/jdk-15.0.2/bin/java -Djava.io.tmpdir=tmp_$NAME \
        -Xmx36g -XX:+UseParallelGC -XX:ParallelGCThreads=2 \
        -jar ${picard} SamToFastq \
        INPUT=../${NAME}.bam \
        F=${NEW}.R1.fastq \
        F2=${NEW}.R2.fastq" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/usr/bin/bash</span>

picard=/scratch/PI/jgwang/dsongad/software/Picard/picard.jar


NAME=<span class="pl-smi">$1</span>
NEW=<span class="pl-smi">$2</span>

<span class="pl-c"><span class="pl-c">#</span> The example script is setted for the HPC3 User in WangLab HKUST</span>
/scratch/PI/jgwang/dsongad/software/jdk-15.0.2/bin/java -Djava.io.tmpdir=tmp_<span class="pl-smi">$NAME</span> \
        -Xmx36g -XX:+UseParallelGC -XX:ParallelGCThreads=2 \
        -jar <span class="pl-smi">${picard}</span> SamToFastq \
        INPUT=../<span class="pl-smi">${NAME}</span>.bam \
        F=<span class="pl-smi">${NEW}</span>.R1.fastq \
        F2=<span class="pl-smi">${NEW}</span>.R2.fastq</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      1.3 Automatic QC with fastp
                     </h2>
                     <a aria-label="Permalink: 1.3 Automatic QC with fastp" class="anchor" href="#13-automatic-qc-with-fastp" id="user-content-13-automatic-qc-with-fastp">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     In order to facilitate the parallel processing of multiple samples and make the pipeline reproducible, we could summarize the commands in scripts. All the scripts we created in this whole pipeline are template structured, which will read parameters from the command line if we use them individually.
In this way, we only need to change several simple parameters to control the whole pipeline to deal with different samples.
                    </p>
                    <p dir="auto">
                     The arrangement logic of the scripts for part 1-3 is the same and can be summarized as follows. For each step, one bash script and one perl script will be used. The bash script contains the detailed command to use different bioinformatics tools, eg. fastp in this step. The required parameters of the bash script often determines which sample will be processed each time.
                    </p>
                    <p dir="auto">
                     The perl script contains the template structure to read in the index files we created in the above
                     <a href="#024-creating-cohort-index-files">
                      section
                     </a>
                     , the detailed command to run the bash script and how to control the parallel processing procedure. The perl script often requires the path of the index file, the working directory, the start index and end index to the working samples as parameters.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      1.3.1 Working Scripts
                     </h3>
                     <a aria-label="Permalink: 1.3.1 Working Scripts" class="anchor" href="#131-working-scripts" id="user-content-131-working-scripts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part1-Sequencing-Data-QC/do_fastp.sh">
                       <code>
                        Script1: do_fastp.sh(Bash script)
                       </code>
                      </a>
                     </strong>
                    </p>
                    <p dir="auto">
                     The script will read in two parameters, and use them as the raw name prefix of fastq files and the changed name prefix. The paired fastq files are expected to end with suffix
                     <em>
                      .R1.fq.gz
                     </em>
                     and
                     <em>
                      .R2.fq.gz
                     </em>
                     to use this script, and the script can also be modified based on your own file naming style.
                    </p>
                    <p dir="auto">
                     The QC results will be stored in the subfoler
                     <em>
                      qc_res
                     </em>
                     , and the raw fastq files will be removed after the automatic preprocessing.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/usr/bin/bash

rawName=$1
cleanName=$2_fp

#mkdir -p qc_res

fastp -w 2 -i ./$rawName\.R1.fq.gz -o ./$cleanName\.R1.fq.gz \
 -I ./$rawName\.R2.fq.gz -O ./$cleanName\.R2.fq.gz \
 -q 15 -u 40 --length_required 45 \
 --detect_adapter_for_pe \
 -h ./qc_res/fastp_$cleanName\.html -j ./qc_res/fastp_$cleanName\.json -R ./qc_res/report_$cleanName\.txt

rm ./$rawName\.R1.fq.gz ./$rawName\.R2.fq.gz" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/usr/bin/bash</span>

rawName=<span class="pl-smi">$1</span>
cleanName=<span class="pl-smi">$2</span>_fp

<span class="pl-c"><span class="pl-c">#</span>mkdir -p qc_res</span>

fastp -w 2 -i ./<span class="pl-smi">$rawName</span><span class="pl-cce">\.</span>R1.fq.gz -o ./<span class="pl-smi">$cleanName</span><span class="pl-cce">\.</span>R1.fq.gz \
 -I ./<span class="pl-smi">$rawName</span><span class="pl-cce">\.</span>R2.fq.gz -O ./<span class="pl-smi">$cleanName</span><span class="pl-cce">\.</span>R2.fq.gz \
 -q 15 -u 40 --length_required 45 \
 --detect_adapter_for_pe \
 -h ./qc_res/fastp_<span class="pl-smi">$cleanName</span><span class="pl-cce">\.</span>html -j ./qc_res/fastp_<span class="pl-smi">$cleanName</span><span class="pl-cce">\.</span>json -R ./qc_res/report_<span class="pl-smi">$cleanName</span><span class="pl-cce">\.</span>txt

rm ./<span class="pl-smi">$rawName</span><span class="pl-cce">\.</span>R1.fq.gz ./<span class="pl-smi">$rawName</span><span class="pl-cce">\.</span>R2.fq.gz</pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part1-Sequencing-Data-QC/runfastp.pl">
                       <code>
                        Script2: run_fastp.pl(Perl script)
                       </code>
                      </a>
                     </strong>
                    </p>
                    <p dir="auto">
                     The script will read in three parameters, and use them as index file path, start index and end index respectively. The previous created index file will be used here to control the parallel processing procedure. The logic is that we could use start index and end index to choose the specific samples to process each time.
                    </p>
                    <p dir="auto">
                     For example, when we want to run the first two samples to test the scripts, we could set the start index as 0, the end index as 1. By this way, the perl script will loop from row index 0 to 1 in the index file to read in the index information and pass them to the bash script.
Then the bash script will process the two samples one by one based on the parameters received.
                    </p>
                    <div class="highlight highlight-source-perl notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/usr/bin/perl

#please run this job in a new clear fold
die &quot;usage: perl runfastp.pl &lt;Data_Index&gt; &lt;start&gt; &lt;end&gt; \n&quot; if @ARGV!= 3;

$index_table = $ARGV[0];



open INDEX, $index_table;
$i = 0;
$id = 0;
while($line = &lt;INDEX&gt;){
	chomp($line);
	my @temp=split('\t',$line);
	$sample[$i][0] = $temp[1];
	$sample[$i][1] = $temp[2];
	$i++;
}
close INDEX;


$start = $ARGV[1];
$end = $ARGV[2];

for($j=$start;$j&lt;=$end;$j++){         
        `sh ./do_fastp.sh $sample[$j][0] $sample[$j][0]`; # Important command, call the bash script based on parameters
	$completed_No = $j;
        print &quot;Job completed!!! No.$completed_No : $sample[$j][1]\t$sample[$j][0]\n\n&quot;;
}" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>!/usr/bin/perl</span>

<span class="pl-c"><span class="pl-c">#</span>please run this job in a new clear fold</span>
<span class="pl-k">die</span> <span class="pl-s"><span class="pl-pds">"</span>usage: perl runfastp.pl &lt;Data_Index&gt; &lt;start&gt; &lt;end&gt; <span class="pl-cce">\n</span><span class="pl-pds">"</span></span> <span class="pl-k">if</span> <span class="pl-smi">@ARGV</span>!= 3;

<span class="pl-smi">$index_table</span> = <span class="pl-smi">$ARGV</span>[0];



<span class="pl-c1">open</span> INDEX, <span class="pl-smi">$index_table</span>;
<span class="pl-smi">$i</span> = 0;
<span class="pl-smi">$id</span> = 0;
<span class="pl-k">while</span>(<span class="pl-smi">$line</span> = &lt;INDEX&gt;){
	<span class="pl-c1">chomp</span>(<span class="pl-smi">$line</span>);
	<span class="pl-k">my</span> <span class="pl-smi">@temp</span>=<span class="pl-c1">split</span>(<span class="pl-s"><span class="pl-pds">'</span>\t<span class="pl-pds">'</span></span>,<span class="pl-smi">$line</span>);
	<span class="pl-smi">$sample</span>[<span class="pl-smi">$i</span>][0] = <span class="pl-smi">$temp</span>[1];
	<span class="pl-smi">$sample</span>[<span class="pl-smi">$i</span>][1] = <span class="pl-smi">$temp</span>[2];
	<span class="pl-smi">$i</span>++;
}
<span class="pl-c1">close</span> INDEX;


<span class="pl-smi">$start</span> = <span class="pl-smi">$ARGV</span>[1];
<span class="pl-smi">$end</span> = <span class="pl-smi">$ARGV</span>[2];

<span class="pl-k">for</span>(<span class="pl-smi">$j</span>=<span class="pl-smi">$start</span>;<span class="pl-smi">$j</span>&lt;=<span class="pl-smi">$end</span>;<span class="pl-smi">$j</span>++){         
        <span class="pl-s"><span class="pl-pds">`</span>sh ./do_fastp.sh <span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0] <span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0]<span class="pl-pds">`</span></span>; <span class="pl-c"><span class="pl-c">#</span> Important command, call the bash script based on parameters</span>
	<span class="pl-smi">$completed_No</span> = <span class="pl-smi">$j</span>;
        <span class="pl-c1">print</span> <span class="pl-s"><span class="pl-pds">"</span>Job completed!!! No.<span class="pl-smi">$completed_No</span> : <span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][1]<span class="pl-cce">\t</span><span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0]<span class="pl-cce">\n\n</span><span class="pl-pds">"</span></span>;
}</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      1.3.2 Scripts Usage Examples
                     </h3>
                     <a aria-label="Permalink: 1.3.2 Scripts Usage Examples" class="anchor" href="#132-scripts-usage-examples" id="user-content-132-scripts-usage-examples">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     For simple usage or script test, we could directly run the perl script with index paramters. It is highly recommended to record all the log information by redirecting them to another log file.
                    </p>
                    <p dir="auto">
                     The example command used to call perl script for simple test usage could be shown as follows. The command will call the scripts to do fastp work on the first sample only and store the log information into the subfoler
                     <em>
                      log
                     </em>
                     .
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="perl run_fastp.pl list_SP_fq_bwa.txt 0 0 2&gt;./log/log.fp_0_0" dir="auto">
                     <pre>perl run_fastp.pl list_SP_fq_bwa.txt 0 0 <span class="pl-k">2&gt;</span>./log/log.fp_0_0</pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part1-Sequencing-Data-QC/slurm.fp_all">
                       <code>
                        Script3: slurm.fp_all(Slurm script)
                       </code>
                      </a>
                     </strong>
                    </p>
                    <p dir="auto">
                     For the slurm cluster users to run multiple samples, the example sbatch script could be arranged as follows.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash
##SBATCH -J bwa24  #Slurm job name
#SBATCH --mail-user=njutangjihong@gmail.com 
#SBATCH --mail-type=end
##SBATCH -p x-gpu-share
##SBATCH -o tra_MD.out
##SBATCH -e tra_MD.err
##SBATCH -N 1 -n 24
#SBATCH --cpus-per-task=20
##SBATCH --exclusive

perl runfastp.pl list_SP_fq_bwa.txt $1 $2 2&gt;./log/log.fp_$1_$2" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -J bwa24  #Slurm job name</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=njutangjihong@gmail.com </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=end</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -p x-gpu-share</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -o tra_MD.out</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -e tra_MD.err</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -N 1 -n 24</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --cpus-per-task=20</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH --exclusive</span>

perl runfastp.pl list_SP_fq_bwa.txt <span class="pl-smi">$1</span> <span class="pl-smi">$2</span> <span class="pl-k">2&gt;</span>./log/log.fp_<span class="pl-smi">$1</span>_<span class="pl-smi">$2</span></pre>
                    </div>
                    <p dir="auto">
                     Taken the demo project as example, there are totally 94(0-93) samples in the index files, and we could process them parallelly on four computing nodes as follows:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="sbatch -p x-gpu-share -J fp0 slurm.fp_all 0 23 
sbatch -p x-gpu-share -J fp24 slurm.fp_all 24 47 
sbatch -p x-gpu-share -J fp48 slurm.fp_all 48 71
sbatch -p x-gpu-share -J fp72 slurm.fp_all 72 93 " dir="auto">
                     <pre>sbatch -p x-gpu-share -J fp0 slurm.fp_all 0 23 
sbatch -p x-gpu-share -J fp24 slurm.fp_all 24 47 
sbatch -p x-gpu-share -J fp48 slurm.fp_all 48 71
sbatch -p x-gpu-share -J fp72 slurm.fp_all 72 93 </pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Part2 Genome Alignment
                     </h1>
                     <a aria-label="Permalink: Part2 Genome Alignment" class="anchor" href="#part2-genome-alignment" id="user-content-part2-genome-alignment">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      2.1 Setting Up Tools
                     </h2>
                     <a aria-label="Permalink: 2.1 Setting Up Tools" class="anchor" href="#21-setting-up-tools" id="user-content-21-setting-up-tools">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Genome alignment and following duplicated marking are essential steps in the genomics analysis pipeline to get the bam files. The Bam files are the binary version of Sam files, and the Sam files are tab-delimited text file that contains sequence alignment data.
Detailed information about SAM/BAM files could be found
                     <a href="http://samtools.github.io/hts-specs/" rel="nofollow">
                      here
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     The bioinformatics tools used in this module are bwa and Picard. Bwa is one of the most famous alignment tools, and is considered as the most useful one. Detailed information about the bwa tools can be found
                     <a href="https://github.com/lh3/bwa">
                      here
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     Bwa can also be installed easily through conda:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="conda install -c bioconda bwa" dir="auto">
                     <pre>conda install -c bioconda bwa</pre>
                    </div>
                    <p dir="auto">
                     In addition, bwa can be installed as follows:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="git clone https://github.com/lh3/bwa.git
cd bwa; make
./bwa index ref.fa
./bwa mem ref.fa read-se.fq.gz | gzip -3 &gt; aln-se.sam.gz
./bwa mem ref.fa read1.fq read2.fq | gzip -3 &gt; aln-pe.sam.gz" dir="auto">
                     <pre>git clone https://github.com/lh3/bwa.git
<span class="pl-c1">cd</span> bwa<span class="pl-k">;</span> make
./bwa index ref.fa
./bwa mem ref.fa read-se.fq.gz <span class="pl-k">|</span> gzip -3 <span class="pl-k">&gt;</span> aln-se.sam.gz
./bwa mem ref.fa read1.fq read2.fq <span class="pl-k">|</span> gzip -3 <span class="pl-k">&gt;</span> aln-pe.sam.gz</pre>
                    </div>
                    <p dir="auto">
                     Picard is a set of Java command line tools for manipulating high-throughput sequencing (HTS) data and formats. Well noted manual and notes have been provided by the developers, and they can be found
                     <a href="https://broadinstitute.github.io/picard/" rel="nofollow">
                      here
                     </a>
                     .
It is highly recommended to follow their instructions to download and install the Picard and set up java in order to use specific commands.
                    </p>
                    <p dir="auto">
                     For WangLab HKUST HPC3 users, the Picard tool can be used by linking to the following path:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/scratch/PI/jgwang/dsongad/software/Picard/picard.jar" dir="auto">
                     <pre>/scratch/PI/jgwang/dsongad/software/Picard/picard.jar</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      2.2 Genome Alignment with bwa
                     </h2>
                     <a aria-label="Permalink: 2.2 Genome Alignment with bwa" class="anchor" href="#22-genome-alignment-with-bwa" id="user-content-22-genome-alignment-with-bwa">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The scripts arrangement logic is the same as previously stated in the section
                     <a href="#13-automatic-qc-with-fastp">
                      1.3 Automatic QC with fastp
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.2.1 Working Scripts
                     </h3>
                     <a aria-label="Permalink: 2.2.1 Working Scripts" class="anchor" href="#221-working-scripts" id="user-content-221-working-scripts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part2-Genome-Alignment/do_bwa_dna.sh">
                       <code>
                        Script1: do_bwa_dna.sh(Bash script)
                       </code>
                      </a>
                     </strong>
                     The script is expected to receive four parameters from user input, representing sample ID, R1 fastq file name, R2 fastq file name and user defined result bam prefix name.
The alignment result bam will be sorted and indexed after the bwa commands, and only the sorted bam will be remained, accompanied by its own index .bai file.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/usr/bin/bash
A='WangLab'
Sample_ID=$1
FAQ1=$2
FAQ2=$3
result=$4


echo [Directory] `pwd`
echo [Machine] `uname -n`
echo [Start] `date`
echo [args] $*
time1=$( date &quot;+%s&quot; )


#line_q1=`zcat $FAQ1 | wc -l`
#line_q2=`zcat $FAQ2 | wc -l`

bwa mem -t 20 -T 0 -R &quot;@RG\tCN:$A\tID:$Sample_ID\tSM:$Sample_ID&quot; /scratch/PI/jgwang/jtangbd/reference/TCGA_GRCh38_bwa_ref/GRCh38.d1.vd1.fa $FAQ1 $FAQ2 2&gt; ./tmp/tmp.$result | samtools view -Sbh1 -@ 8 - -o $result\.bam

#### mapping duplicates
###counting the num of bam reads

samtools sort -@ 8 -O bam -o $result.sorted.bam -T the_temp_$result $result\.bam
samtools index $result\.sorted.bam 
rm $result\.bam
#rm $FAQ1
#rm $FAQ2
#samtools idxstats result_test.sorted.bam |head 
#samtools view -H /home/qmu/projects/AsianpGBM/TCGAGBM/C484.TCGA-06-6699-10A-01D-1845-08.2_gdc_realn.bam |grep &quot;^@PG&quot;
#samtools flagstat result_test.sorted.bam
#samtools view -f 4 result_test.sorted.bam chr7 | wc -l
#samtools mpileup ../raw_data/results_B_DNA/IGCT_S002_B.sorted.bam -r chr10:63207680 -f /home/dsongad/software/my_lib/STAR_index_hg38_RefSeq_hg38/hg38.fa |head -n20

time2=$( date &quot;+%s&quot; )
echo [deltat] $(( $time2 - $time1 ))
echo [End]" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/usr/bin/bash</span>
A=<span class="pl-s"><span class="pl-pds">'</span>WangLab<span class="pl-pds">'</span></span>
Sample_ID=<span class="pl-smi">$1</span>
FAQ1=<span class="pl-smi">$2</span>
FAQ2=<span class="pl-smi">$3</span>
result=<span class="pl-smi">$4</span>


<span class="pl-c1">echo</span> [Directory] <span class="pl-s"><span class="pl-pds">`</span>pwd<span class="pl-pds">`</span></span>
<span class="pl-c1">echo</span> [Machine] <span class="pl-s"><span class="pl-pds">`</span>uname -n<span class="pl-pds">`</span></span>
<span class="pl-c1">echo</span> [Start] <span class="pl-s"><span class="pl-pds">`</span>date<span class="pl-pds">`</span></span>
<span class="pl-c1">echo</span> [args] <span class="pl-smi">$*</span>
time1=<span class="pl-s"><span class="pl-pds">$(</span> date <span class="pl-s"><span class="pl-pds">"</span>+%s<span class="pl-pds">"</span></span> <span class="pl-pds">)</span></span>


<span class="pl-c"><span class="pl-c">#</span>line_q1=`zcat $FAQ1 | wc -l`</span>
<span class="pl-c"><span class="pl-c">#</span>line_q2=`zcat $FAQ2 | wc -l`</span>

bwa mem -t 20 -T 0 -R <span class="pl-s"><span class="pl-pds">"</span>@RG\tCN:<span class="pl-smi">$A</span>\tID:<span class="pl-smi">$Sample_ID</span>\tSM:<span class="pl-smi">$Sample_ID</span><span class="pl-pds">"</span></span> /scratch/PI/jgwang/jtangbd/reference/TCGA_GRCh38_bwa_ref/GRCh38.d1.vd1.fa <span class="pl-smi">$FAQ1</span> <span class="pl-smi">$FAQ2</span> <span class="pl-k">2&gt;</span> ./tmp/tmp.<span class="pl-smi">$result</span> <span class="pl-k">|</span> samtools view -Sbh1 -@ 8 - -o <span class="pl-smi">$result</span><span class="pl-cce">\.</span>bam

<span class="pl-c"><span class="pl-c">#</span>### mapping duplicates</span>
<span class="pl-c"><span class="pl-c">#</span>##counting the num of bam reads</span>

samtools sort -@ 8 -O bam -o <span class="pl-smi">$result</span>.sorted.bam -T the_temp_<span class="pl-smi">$result</span> <span class="pl-smi">$result</span><span class="pl-cce">\.</span>bam
samtools index <span class="pl-smi">$result</span><span class="pl-cce">\.</span>sorted.bam 
rm <span class="pl-smi">$result</span><span class="pl-cce">\.</span>bam
<span class="pl-c"><span class="pl-c">#</span>rm $FAQ1</span>
<span class="pl-c"><span class="pl-c">#</span>rm $FAQ2</span>
<span class="pl-c"><span class="pl-c">#</span>samtools idxstats result_test.sorted.bam |head </span>
<span class="pl-c"><span class="pl-c">#</span>samtools view -H /home/qmu/projects/AsianpGBM/TCGAGBM/C484.TCGA-06-6699-10A-01D-1845-08.2_gdc_realn.bam |grep "^@PG"</span>
<span class="pl-c"><span class="pl-c">#</span>samtools flagstat result_test.sorted.bam</span>
<span class="pl-c"><span class="pl-c">#</span>samtools view -f 4 result_test.sorted.bam chr7 | wc -l</span>
<span class="pl-c"><span class="pl-c">#</span>samtools mpileup ../raw_data/results_B_DNA/IGCT_S002_B.sorted.bam -r chr10:63207680 -f /home/dsongad/software/my_lib/STAR_index_hg38_RefSeq_hg38/hg38.fa |head -n20</span>

time2=<span class="pl-s"><span class="pl-pds">$(</span> date <span class="pl-s"><span class="pl-pds">"</span>+%s<span class="pl-pds">"</span></span> <span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> [deltat] <span class="pl-s"><span class="pl-pds">$((</span> <span class="pl-smi">$time2</span> <span class="pl-k">-</span> <span class="pl-smi">$time1</span> <span class="pl-pds">))</span></span>
<span class="pl-c1">echo</span> [End]</pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part2-Genome-Alignment/run_bwa_mapping.pl">
                       <code>
                        Script2: run_bwa_mapping.pl(Perl script)
                       </code>
                      </a>
                     </strong>
                     The script will read in four parameters, and use them as index file path, fastq file path, start index and end index respectively.
The previous created index file will be used here to control the parallel processing procedure. The logic is that we could use start index and end index to choose the specific samples to process each time.
                    </p>
                    <p dir="auto">
                     The parameter
                     <code>
                      &lt;fastq-dir&gt;
                     </code>
                     is essential to be provided to locate the fastq files, and user can choose to delete the raw fastq files after alignment for storage room saving.
                    </p>
                    <div class="highlight highlight-source-perl notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/usr/bin/perl

#please run this job in a new clear fold
die &quot;usage: perl run_bwa_mapping.pl &lt;Data_Index&gt; &lt;fastq-dir&gt; &lt;START&gt; &lt;END&gt; \n&quot; if @ARGV!= 4;

$index_table = $ARGV[0];
$root = $ARGV[1];


open INDEX, $index_table;
$i = 0;
$id = 0;
while($line = &lt;INDEX&gt;){
	chomp($line);
	my @temp=split('\t',$line);
	$sample[$i][0] = $temp[1];
	$sample[$i][1] = $temp[2];
	$i++;
}
close INDEX;

#`mkdir Dir_tmp_1`;
#chdir &quot;./Dir_tmp_1&quot;;
#
$start = $ARGV[2];
$end = $ARGV[3];

for($j=$start;$j&lt;=$end;$j++){
        `sh ./do_bwa_dna.sh $sample[$j][0] $root/$sample[$j][0]_fp.R1.fq.gz $root/$sample[$j][0]_fp.R2.fq.gz $sample[$j][1]`;
	$completed_No = $j;
        print &quot;bwa-MEM completed!!! No.$completed_No : $sample[$j][1]\t$sample[$j][0]\n&quot;;
}
#chdir &quot;../&quot;;" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>!/usr/bin/perl</span>

<span class="pl-c"><span class="pl-c">#</span>please run this job in a new clear fold</span>
<span class="pl-k">die</span> <span class="pl-s"><span class="pl-pds">"</span>usage: perl run_bwa_mapping.pl &lt;Data_Index&gt; &lt;fastq-dir&gt; &lt;START&gt; &lt;END&gt; <span class="pl-cce">\n</span><span class="pl-pds">"</span></span> <span class="pl-k">if</span> <span class="pl-smi">@ARGV</span>!= 4;

<span class="pl-smi">$index_table</span> = <span class="pl-smi">$ARGV</span>[0];
<span class="pl-smi">$root</span> = <span class="pl-smi">$ARGV</span>[1];


<span class="pl-c1">open</span> INDEX, <span class="pl-smi">$index_table</span>;
<span class="pl-smi">$i</span> = 0;
<span class="pl-smi">$id</span> = 0;
<span class="pl-k">while</span>(<span class="pl-smi">$line</span> = &lt;INDEX&gt;){
	<span class="pl-c1">chomp</span>(<span class="pl-smi">$line</span>);
	<span class="pl-k">my</span> <span class="pl-smi">@temp</span>=<span class="pl-c1">split</span>(<span class="pl-s"><span class="pl-pds">'</span>\t<span class="pl-pds">'</span></span>,<span class="pl-smi">$line</span>);
	<span class="pl-smi">$sample</span>[<span class="pl-smi">$i</span>][0] = <span class="pl-smi">$temp</span>[1];
	<span class="pl-smi">$sample</span>[<span class="pl-smi">$i</span>][1] = <span class="pl-smi">$temp</span>[2];
	<span class="pl-smi">$i</span>++;
}
<span class="pl-c1">close</span> INDEX;

<span class="pl-c"><span class="pl-c">#</span>`mkdir Dir_tmp_1`;</span>
<span class="pl-c"><span class="pl-c">#</span>chdir "./Dir_tmp_1";</span>
<span class="pl-c"><span class="pl-c">#</span></span>
<span class="pl-smi">$start</span> = <span class="pl-smi">$ARGV</span>[2];
<span class="pl-smi">$end</span> = <span class="pl-smi">$ARGV</span>[3];

<span class="pl-k">for</span>(<span class="pl-smi">$j</span>=<span class="pl-smi">$start</span>;<span class="pl-smi">$j</span>&lt;=<span class="pl-smi">$end</span>;<span class="pl-smi">$j</span>++){
        <span class="pl-s"><span class="pl-pds">`</span>sh ./do_bwa_dna.sh <span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0] <span class="pl-smi">$root</span>/<span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0]_fp.R1.fq.gz <span class="pl-smi">$root</span>/<span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0]_fp.R2.fq.gz <span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][1]<span class="pl-pds">`</span></span>;
	<span class="pl-smi">$completed_No</span> = <span class="pl-smi">$j</span>;
        <span class="pl-c1">print</span> <span class="pl-s"><span class="pl-pds">"</span>bwa-MEM completed!!! No.<span class="pl-smi">$completed_No</span> : <span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][1]<span class="pl-cce">\t</span><span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0]<span class="pl-cce">\n</span><span class="pl-pds">"</span></span>;
}
<span class="pl-c"><span class="pl-c">#</span>chdir "../";</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.2.2 Scripts Usage Example
                     </h3>
                     <a aria-label="Permalink: 2.2.2 Scripts Usage Example" class="anchor" href="#222-scripts-usage-example" id="user-content-222-scripts-usage-example">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     For simple usage or script test, we could directly run the perl script with index paramters. It is highly recommended to record all the log information by redirecting them to another log file.
                    </p>
                    <p dir="auto">
                     The example command used to call perl script for simple test usage could be shown as follows. The command will call the scripts to do bwa mapping on the first sample only and store the log information into the subfoler
                     <em>
                      log
                     </em>
                     .
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="perl run_bwa_mapping.pl list_SP_fq_bwa.txt 0 0 2&gt;./log/log.bwa_0_0" dir="auto">
                     <pre>perl run_bwa_mapping.pl list_SP_fq_bwa.txt 0 0 <span class="pl-k">2&gt;</span>./log/log.bwa_0_0</pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part1-Sequencing-Data-QC/slurm.bwa_all">
                       <code>
                        Script3: slurm.bwa_all(Slurm script)
                       </code>
                      </a>
                     </strong>
                     For the slurm cluster users to run multiple samples, the example sbatch script could be arranged as follows.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash
##SBATCH -J bwa24  #Slurm job name
#SBATCH --mail-user=njutangjihong@gmail.com 
#SBATCH --mail-type=end
##SBATCH -p x-gpu-share
##SBATCH -o tra_MD.out
##SBATCH -e tra_MD.err
##SBATCH -N 1 -n 24
#SBATCH --cpus-per-task=20
#SBATCH --exclusive

perl run_bwa_mapping.pl list_SP_fq_bwa.txt ../all_fq $1 $2 2&gt;./log/log.bwa_$1_$2" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -J bwa24  #Slurm job name</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=njutangjihong@gmail.com </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=end</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -p x-gpu-share</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -o tra_MD.out</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -e tra_MD.err</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -N 1 -n 24</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --cpus-per-task=20</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --exclusive</span>

perl run_bwa_mapping.pl list_SP_fq_bwa.txt ../all_fq <span class="pl-smi">$1</span> <span class="pl-smi">$2</span> <span class="pl-k">2&gt;</span>./log/log.bwa_<span class="pl-smi">$1</span>_<span class="pl-smi">$2</span></pre>
                    </div>
                    <p dir="auto">
                     Taken the demo project as example, there are totally 94(0-93) samples in the index files, and we could process them parallelly on four computing nodes as follows:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="sbatch -p x-gpu-share -J bwa0 slurm.bwa_all 0 23 
sbatch -p x-gpu-share -J bwa24 slurm.bwa_all 24 47 
sbatch -p x-gpu-share -J bwa48 slurm.bwa_all 48 71
sbatch -p x-gpu-share -J bwa72 slurm.bwa_all 72 93 " dir="auto">
                     <pre>sbatch -p x-gpu-share -J bwa0 slurm.bwa_all 0 23 
sbatch -p x-gpu-share -J bwa24 slurm.bwa_all 24 47 
sbatch -p x-gpu-share -J bwa48 slurm.bwa_all 48 71
sbatch -p x-gpu-share -J bwa72 slurm.bwa_all 72 93 </pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      2.3 Mark Duplicates with Picard
                     </h2>
                     <a aria-label="Permalink: 2.3 Mark Duplicates with Picard" class="anchor" href="#23-mark-duplicates-with-picard" id="user-content-23-mark-duplicates-with-picard">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The scripts arrangement logic is the same as previously stated in the section
                     <a href="#13-automatic-qc-with-fastp">
                      1.3 Automatic QC with fastp
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.3.1 Working Scripts
                     </h3>
                     <a aria-label="Permalink: 2.3.1 Working Scripts" class="anchor" href="#231-working-scripts" id="user-content-231-working-scripts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part2-Genome-Alignment/do_mark_duplicates.sh">
                       <code>
                        Script1: do_mark_duplicates.sh(Bash script)
                       </code>
                      </a>
                     </strong>
                     The script is expected to receive only one parameter from user input, the prefix name of the bam file. After the mark duplicates work, the bam files will be ended with .MD. surfix, and the .MD.bam files will be sorted and indexed again.
The raw bam files will be removed and only the .MD.bam and .MD.bam.bai files will be remained.
                    </p>
                    <p dir="auto">
                     One important note: The mark duplicates step is implemented in the same directory after the bwa mapping step, and must be used in the same folder if no change is made to the script.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/usr/bin/bash

#set -euxo pipefail
#export LD_LIBRARY_PATH=/home/qmu/tools/lib:/opt/intel/advisor_xe_2016.1.40.463413/lib64:/home/qmu/tools/xz-5.2.3/tmp/lib 
#picard=/home/share/jgwang/softwares/picard/build/libs/picard.jar
picard=/scratch/PI/jgwang/dsongad/software/Picard/picard.jar

NAME=$1

mkdir md_tmp/tmp_$NAME

java -Djava.io.tmpdir=md_tmp/tmp_$NAME \
	-Xmx120g -XX:+UseParallelGC -XX:ParallelGCThreads=4 \
	 -jar ${picard} MarkDuplicates \
	INPUT=${NAME}.sorted.bam \
	OUTPUT=${NAME}.sorted.MD.bam \
	METRICS_FILE=./md_metrix/${NAME}.metrics.txt \
	ASSUME_SORTED=true \
	REMOVE_DUPLICATES=true \
	VALIDATION_STRINGENCY=LENIENT

rm ${NAME}.sorted.bam
rm ${NAME}.sorted.bam.bai
samtools index ${NAME}.sorted.MD.bam
#rm ${NAME}.sorted.MD.bam.txt" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/usr/bin/bash</span>

<span class="pl-c"><span class="pl-c">#</span>set -euxo pipefail</span>
<span class="pl-c"><span class="pl-c">#</span>export LD_LIBRARY_PATH=/home/qmu/tools/lib:/opt/intel/advisor_xe_2016.1.40.463413/lib64:/home/qmu/tools/xz-5.2.3/tmp/lib </span>
<span class="pl-c"><span class="pl-c">#</span>picard=/home/share/jgwang/softwares/picard/build/libs/picard.jar</span>
picard=/scratch/PI/jgwang/dsongad/software/Picard/picard.jar

NAME=<span class="pl-smi">$1</span>

mkdir md_tmp/tmp_<span class="pl-smi">$NAME</span>

java -Djava.io.tmpdir=md_tmp/tmp_<span class="pl-smi">$NAME</span> \
	-Xmx120g -XX:+UseParallelGC -XX:ParallelGCThreads=4 \
	 -jar <span class="pl-smi">${picard}</span> MarkDuplicates \
	INPUT=<span class="pl-smi">${NAME}</span>.sorted.bam \
	OUTPUT=<span class="pl-smi">${NAME}</span>.sorted.MD.bam \
	METRICS_FILE=./md_metrix/<span class="pl-smi">${NAME}</span>.metrics.txt \
	ASSUME_SORTED=true \
	REMOVE_DUPLICATES=true \
	VALIDATION_STRINGENCY=LENIENT

rm <span class="pl-smi">${NAME}</span>.sorted.bam
rm <span class="pl-smi">${NAME}</span>.sorted.bam.bai
samtools index <span class="pl-smi">${NAME}</span>.sorted.MD.bam
<span class="pl-c"><span class="pl-c">#</span>rm ${NAME}.sorted.MD.bam.txt</span></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part2-Genome-Alignment/run_mark_duplicates.pl">
                       <code>
                        Script2: run_mark_duplicates.pl(Perl script)
                       </code>
                      </a>
                     </strong>
                     The script will read in four parameters, and use them as index file path, fastq file path, start index and end index respectively.
The previous created index file will be used here to control the parallel processing procedure. The logic is that we could use start index and end index to choose the specific samples to process each time.
                    </p>
                    <p dir="auto">
                     The parameter setting is the same to the script
                     <code>
                      run_bwa_mapping.pl
                     </code>
                     in order to simplify the usage of the index files.
                    </p>
                    <div class="highlight highlight-source-perl notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/usr/bin/perl

#please run this job in a new clear fold
die &quot;usage: perl run_mark_duplicates.pl &lt;Data_Index&gt; &lt;fastq-dir&gt; &lt;START&gt; &lt;END&gt; \n&quot; if @ARGV!= 4;

$index_table = $ARGV[0];
$root = $ARGV[1];


open INDEX, $index_table;
$i = 0;
$id = 0;
while($line = &lt;INDEX&gt;){
	chomp($line);
	my @temp=split('\t',$line);
	$sample[$i][0] = $temp[1];
	$sample[$i][1] = $temp[2];
	$i++;
}
close INDEX;

#`mkdir Dir_tmp_1`;
#chdir &quot;./Dir_tmp_1&quot;;
#
$start = $ARGV[2];
$end = $ARGV[3];

for($j=$start;$j&lt;=$end;$j++){          #run the first 100 samples as a try
	`sh ./do_mark_duplicates.sh $sample[$j][1]`;
	$completed_No = $j;
        print &quot;MarkingDup completed!!! No.$completed_No : $sample[$j][0]\t$sample[$j][1]\n\n&quot;;
}
#chdir &quot;../&quot;;" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>!/usr/bin/perl</span>

<span class="pl-c"><span class="pl-c">#</span>please run this job in a new clear fold</span>
<span class="pl-k">die</span> <span class="pl-s"><span class="pl-pds">"</span>usage: perl run_mark_duplicates.pl &lt;Data_Index&gt; &lt;fastq-dir&gt; &lt;START&gt; &lt;END&gt; <span class="pl-cce">\n</span><span class="pl-pds">"</span></span> <span class="pl-k">if</span> <span class="pl-smi">@ARGV</span>!= 4;

<span class="pl-smi">$index_table</span> = <span class="pl-smi">$ARGV</span>[0];
<span class="pl-smi">$root</span> = <span class="pl-smi">$ARGV</span>[1];


<span class="pl-c1">open</span> INDEX, <span class="pl-smi">$index_table</span>;
<span class="pl-smi">$i</span> = 0;
<span class="pl-smi">$id</span> = 0;
<span class="pl-k">while</span>(<span class="pl-smi">$line</span> = &lt;INDEX&gt;){
	<span class="pl-c1">chomp</span>(<span class="pl-smi">$line</span>);
	<span class="pl-k">my</span> <span class="pl-smi">@temp</span>=<span class="pl-c1">split</span>(<span class="pl-s"><span class="pl-pds">'</span>\t<span class="pl-pds">'</span></span>,<span class="pl-smi">$line</span>);
	<span class="pl-smi">$sample</span>[<span class="pl-smi">$i</span>][0] = <span class="pl-smi">$temp</span>[1];
	<span class="pl-smi">$sample</span>[<span class="pl-smi">$i</span>][1] = <span class="pl-smi">$temp</span>[2];
	<span class="pl-smi">$i</span>++;
}
<span class="pl-c1">close</span> INDEX;

<span class="pl-c"><span class="pl-c">#</span>`mkdir Dir_tmp_1`;</span>
<span class="pl-c"><span class="pl-c">#</span>chdir "./Dir_tmp_1";</span>
<span class="pl-c"><span class="pl-c">#</span></span>
<span class="pl-smi">$start</span> = <span class="pl-smi">$ARGV</span>[2];
<span class="pl-smi">$end</span> = <span class="pl-smi">$ARGV</span>[3];

<span class="pl-k">for</span>(<span class="pl-smi">$j</span>=<span class="pl-smi">$start</span>;<span class="pl-smi">$j</span>&lt;=<span class="pl-smi">$end</span>;<span class="pl-smi">$j</span>++){          <span class="pl-c"><span class="pl-c">#</span>run the first 100 samples as a try</span>
	<span class="pl-s"><span class="pl-pds">`</span>sh ./do_mark_duplicates.sh <span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][1]<span class="pl-pds">`</span></span>;
	<span class="pl-smi">$completed_No</span> = <span class="pl-smi">$j</span>;
        <span class="pl-c1">print</span> <span class="pl-s"><span class="pl-pds">"</span>MarkingDup completed!!! No.<span class="pl-smi">$completed_No</span> : <span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0]<span class="pl-cce">\t</span><span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][1]<span class="pl-cce">\n\n</span><span class="pl-pds">"</span></span>;
}
<span class="pl-c"><span class="pl-c">#</span>chdir "../";</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.3.2 Scripts Usage Example
                     </h3>
                     <a aria-label="Permalink: 2.3.2 Scripts Usage Example" class="anchor" href="#232-scripts-usage-example" id="user-content-232-scripts-usage-example">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     For simple usage or script test, we could directly run the perl script with index paramters. It is highly recommended to record all the log information by redirecting them to another log file.
                    </p>
                    <p dir="auto">
                     The example command used to call perl script for simple test usage could be shown as follows. The command will call the scripts to do picard duplicates marking on the first sample only and store the log information into the subfoler
                     <em>
                      log
                     </em>
                     .
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="perl run_mark_duplicates.pl list_SP_fq_bwa.txt 0 0 2&gt;./log/log.md_0_0" dir="auto">
                     <pre>perl run_mark_duplicates.pl list_SP_fq_bwa.txt 0 0 <span class="pl-k">2&gt;</span>./log/log.md_0_0</pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/scripts/part1-Sequencing-Data-QC/slurm.md_all">
                       <code>
                        Script3: slurm.md_all(Slurm script)
                       </code>
                      </a>
                     </strong>
                     For the slurm cluster users to run multiple samples, the example sbatch script could be arranged as follows.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash
##SBATCH -J md24  #Slurm job name
#SBATCH --mail-user=njutangjihong@gmail.com 
#SBATCH --mail-type=end
##SBATCH -p x-gpu-share
##SBATCH -o tra_MD.out
##SBATCH -e tra_MD.err
##SBATCH -N 1 -n 24
#SBATCH --cpus-per-task=20
#SBATCH --exclusive

perl run_mark_duplicates.pl list_SP_fq_bwa.txt ./ $1 $2 2&gt;./log/log.md_$1_$2" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -J md24  #Slurm job name</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=njutangjihong@gmail.com </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=end</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -p x-gpu-share</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -o tra_MD.out</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -e tra_MD.err</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -N 1 -n 24</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --cpus-per-task=20</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --exclusive</span>

perl run_mark_duplicates.pl list_SP_fq_bwa.txt ./ <span class="pl-smi">$1</span> <span class="pl-smi">$2</span> <span class="pl-k">2&gt;</span>./log/log.md_<span class="pl-smi">$1</span>_<span class="pl-smi">$2</span></pre>
                    </div>
                    <p dir="auto">
                     Taken the demo project as example, there are totally 94(0-93) samples in the index files, and we could process them parallelly on four computing nodes as follows:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="sbatch -p x-gpu-share -J md0 slurm.md_all 0 23 
sbatch -p x-gpu-share -J md24 slurm.md_all 24 47 
sbatch -p x-gpu-share -J md48 slurm.md_all 48 71
sbatch -p x-gpu-share -J md72 slurm.md_all 72 93 " dir="auto">
                     <pre>sbatch -p x-gpu-share -J md0 slurm.md_all 0 23 
sbatch -p x-gpu-share -J md24 slurm.md_all 24 47 
sbatch -p x-gpu-share -J md48 slurm.md_all 48 71
sbatch -p x-gpu-share -J md72 slurm.md_all 72 93 </pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Part3 Somatic Mutation Calling
                     </h1>
                     <a aria-label="Permalink: Part3 Somatic Mutation Calling" class="anchor" href="#part3-somatic-mutation-calling" id="user-content-part3-somatic-mutation-calling">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      3.1 Setting Up Tools
                     </h2>
                     <a aria-label="Permalink: 3.1 Setting Up Tools" class="anchor" href="#31-setting-up-tools" id="user-content-31-setting-up-tools">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      3.2 Somatic Mutations Calling with SAVI
                     </h2>
                     <a aria-label="Permalink: 3.2 Somatic Mutations Calling with SAVI" class="anchor" href="#32-somatic-mutations-calling-with-savi" id="user-content-32-somatic-mutations-calling-with-savi">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash

export LIBRARY_PATH=/scratch/PI/jgwang/dsongad/software/ncurses/lib/:$LIBRARY_PATH
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/scratch/PI/jgwang/dsongad/software/ncurses/lib/
export PATH=/scratch/PI/jgwang/dsongad/software/samtools-1.2:$PATH
export PATH=$PATH:/scratch/PI/jgwang/dsongad/software/snpEff
export LIBRARY_PATH=/scratch/PI/jgwang/dsongad/software/ncurses/lib/:$LIBRARY_PATH
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/scratch/PI/jgwang/dsongad/software/ncurses/lib/
#conda activate savi_hg38_py2.7

OUTPUT=$1
BD=$2
T1=$3


V=/scratch/PI/jgwang/dsongad/software/my_lib/tmp_vcf_lib
S=/scratch/PI/jgwang/dsongad/IGCT/run_savi/SAVI
R=/scratch/PI/jgwang/dsongad/software/my_lib/star_genome_d1_vd1_gtfv22/GRCh38.d1.vd1.fa

for i in {1..22} {X,Y,M}; do
	mkdir -p ${OUTPUT}/${i}
        ${S}/savi_SONG_allregion.py --bams ${BD},${T1} \
                --names Nor,T1 \
                --ann GRCh38.p7.RefSeq --memory 40 \
                --ref ${R} -c 2:1 \
                --region chr${i} -v --superverbose \
                --noncoding --mapqual 20 --mindepth 4 \
                --outputdir ${OUTPUT}/${i} \
                --annvcf ${V}/Cosmic_2020_hg38/Song_hg38_Cosmic.sorted.vcf.gz,${V}/Cosmic_2020_hg38/Song_hg38_ClinVar.sorted.vcf.gz,${V}/Cosmic_2020_hg38/Song_hg38_PancancerGermline.sorted.vcf.gz,${V}/Cosmic_2020_hg38/Song_hg38_cbio.vcf.gz,${V}/normal_hg38/Song_hg38_All_SNP.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_TOPMED.sorted.vcf.gz,${V}/normal_hg38/IGCTfamB.sorted.vcf.gz,${V}/normal_hg38/IGCTpatB.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_GATKnormal.vcf.gz,${V}/normal_hg38/Song_hg38_HGDP_1KG.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_MuTectPON.vcf.gz,${V}/normal_hg38/Song_hg38_dbSNP_ALFA.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_gnomAD.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_219normals.vcf.gz,${V}/normal_hg38/Song_China_map.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_ExAC.sorted.vcf.gz,${V}/normal_hg38/Song_hg38_NGDC_SNP.vcf.gz,${V}/normal_hg38/Song_hg38_meganormal.vcf.gz 1&gt; ${OUTPUT}/${i}/err.out 2&gt;${OUTPUT}/${i}/err.log &amp;
done
wait

echo SAVI done!

echo Start merging...

O1=${OUTPUT}/merge.report.coding.PDfilter.txt
O2=${OUTPUT}/merge.report.coding.txt


head -n 1 ${OUTPUT}/1/report/report.coding.PDfilter.txt &gt; ${O1}
head -n 1 ${OUTPUT}/1/report/report.coding.txt &gt; ${O2}

for i in {1..22} {X,Y,M}; do
        F1=${OUTPUT}/${i}/report/report.coding.PDfilter.txt
	F2=${OUTPUT}/${i}/report/report.coding.txt
	
        if [ -e ${F1} ]; then
                tail -n +2 ${F1} &gt;&gt; ${O1}
        else
                echo ${F1} does not exist
        fi

        if [ -e ${F2} ]; then
                tail -n +2 ${F2} &gt;&gt; ${O2}
        else
                echo ${F2} does not exist
        fi

done" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>

<span class="pl-k">export</span> LIBRARY_PATH=/scratch/PI/jgwang/dsongad/software/ncurses/lib/:<span class="pl-smi">$LIBRARY_PATH</span>
<span class="pl-k">export</span> LD_LIBRARY_PATH=<span class="pl-smi">$LD_LIBRARY_PATH</span>:/scratch/PI/jgwang/dsongad/software/ncurses/lib/
<span class="pl-k">export</span> PATH=/scratch/PI/jgwang/dsongad/software/samtools-1.2:<span class="pl-smi">$PATH</span>
<span class="pl-k">export</span> PATH=<span class="pl-smi">$PATH</span>:/scratch/PI/jgwang/dsongad/software/snpEff
<span class="pl-k">export</span> LIBRARY_PATH=/scratch/PI/jgwang/dsongad/software/ncurses/lib/:<span class="pl-smi">$LIBRARY_PATH</span>
<span class="pl-k">export</span> LD_LIBRARY_PATH=<span class="pl-smi">$LD_LIBRARY_PATH</span>:/scratch/PI/jgwang/dsongad/software/ncurses/lib/
<span class="pl-c"><span class="pl-c">#</span>conda activate savi_hg38_py2.7</span>

OUTPUT=<span class="pl-smi">$1</span>
BD=<span class="pl-smi">$2</span>
T1=<span class="pl-smi">$3</span>


V=/scratch/PI/jgwang/dsongad/software/my_lib/tmp_vcf_lib
S=/scratch/PI/jgwang/dsongad/IGCT/run_savi/SAVI
R=/scratch/PI/jgwang/dsongad/software/my_lib/star_genome_d1_vd1_gtfv22/GRCh38.d1.vd1.fa

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> {1..22} {X,Y,M}<span class="pl-k">;</span> <span class="pl-k">do</span>
	mkdir -p <span class="pl-smi">${OUTPUT}</span>/<span class="pl-smi">${i}</span>
        <span class="pl-smi">${S}</span>/savi_SONG_allregion.py --bams <span class="pl-smi">${BD}</span>,<span class="pl-smi">${T1}</span> \
                --names Nor,T1 \
                --ann GRCh38.p7.RefSeq --memory 40 \
                --ref <span class="pl-smi">${R}</span> -c 2:1 \
                --region chr<span class="pl-smi">${i}</span> -v --superverbose \
                --noncoding --mapqual 20 --mindepth 4 \
                --outputdir <span class="pl-smi">${OUTPUT}</span>/<span class="pl-smi">${i}</span> \
                --annvcf <span class="pl-smi">${V}</span>/Cosmic_2020_hg38/Song_hg38_Cosmic.sorted.vcf.gz,<span class="pl-smi">${V}</span>/Cosmic_2020_hg38/Song_hg38_ClinVar.sorted.vcf.gz,<span class="pl-smi">${V}</span>/Cosmic_2020_hg38/Song_hg38_PancancerGermline.sorted.vcf.gz,<span class="pl-smi">${V}</span>/Cosmic_2020_hg38/Song_hg38_cbio.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_hg38_All_SNP.sorted.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_hg38_TOPMED.sorted.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/IGCTfamB.sorted.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/IGCTpatB.sorted.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_hg38_GATKnormal.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_hg38_HGDP_1KG.sorted.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_hg38_MuTectPON.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_hg38_dbSNP_ALFA.sorted.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_hg38_gnomAD.sorted.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_hg38_219normals.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_China_map.sorted.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_hg38_ExAC.sorted.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_hg38_NGDC_SNP.vcf.gz,<span class="pl-smi">${V}</span>/normal_hg38/Song_hg38_meganormal.vcf.gz <span class="pl-k">1&gt;</span> <span class="pl-smi">${OUTPUT}</span>/<span class="pl-smi">${i}</span>/err.out <span class="pl-k">2&gt;</span><span class="pl-smi">${OUTPUT}</span>/<span class="pl-smi">${i}</span>/err.log <span class="pl-k">&amp;</span>
<span class="pl-k">done</span>
<span class="pl-c1">wait</span>

<span class="pl-c1">echo</span> SAVI done<span class="pl-k">!</span>

<span class="pl-c1">echo</span> Start merging...

O1=<span class="pl-smi">${OUTPUT}</span>/merge.report.coding.PDfilter.txt
O2=<span class="pl-smi">${OUTPUT}</span>/merge.report.coding.txt


head -n 1 <span class="pl-smi">${OUTPUT}</span>/1/report/report.coding.PDfilter.txt <span class="pl-k">&gt;</span> <span class="pl-smi">${O1}</span>
head -n 1 <span class="pl-smi">${OUTPUT}</span>/1/report/report.coding.txt <span class="pl-k">&gt;</span> <span class="pl-smi">${O2}</span>

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> {1..22} {X,Y,M}<span class="pl-k">;</span> <span class="pl-k">do</span>
        F1=<span class="pl-smi">${OUTPUT}</span>/<span class="pl-smi">${i}</span>/report/report.coding.PDfilter.txt
	F2=<span class="pl-smi">${OUTPUT}</span>/<span class="pl-smi">${i}</span>/report/report.coding.txt
	
        <span class="pl-k">if</span> [ <span class="pl-k">-e</span> <span class="pl-smi">${F1}</span> ]<span class="pl-k">;</span> <span class="pl-k">then</span>
                tail -n +2 <span class="pl-smi">${F1}</span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">${O1}</span>
        <span class="pl-k">else</span>
                <span class="pl-c1">echo</span> <span class="pl-smi">${F1}</span> does not exist
        <span class="pl-k">fi</span>

        <span class="pl-k">if</span> [ <span class="pl-k">-e</span> <span class="pl-smi">${F2}</span> ]<span class="pl-k">;</span> <span class="pl-k">then</span>
                tail -n +2 <span class="pl-smi">${F2}</span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">${O2}</span>
        <span class="pl-k">else</span>
                <span class="pl-c1">echo</span> <span class="pl-smi">${F2}</span> does not exist
        <span class="pl-k">fi</span>

<span class="pl-k">done</span></pre>
                    </div>
                    <div class="highlight highlight-source-perl notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/usr/bin/perl

#please run this job in a new clear fold
die &quot;usage: perl run_savi.pl &lt;Data_Index&gt; &lt;START&gt; &lt;END&gt; \n&quot; if @ARGV!= 3;

$index_table = $ARGV[0];


open INDEX, $index_table;
$i = 0;
$id = 0;
while($line = &lt;INDEX&gt;){
	chomp($line);
	my @temp=split('\t',$line);
	$sample[$i][0] = $temp[1];
	$i++;
}
close INDEX;

#`mkdir Dir_tmp_1`;
#chdir &quot;./Dir_tmp_1&quot;;
#
$start = $ARGV[1];
$end = $ARGV[2];

for($j=$start;$j&lt;=$end;$j++){
	`sh do_savi_WES_2sample_GRCh38.p7.RefSeq.sh  $sample[$j][0] ../all_mapping_hg38/$sample[$j][0]_B_WES_hg38.sorted.MD.bam ../all_mapping_hg38/$sample[$j][0]_T_WES_hg38.sorted.MD.bam`;
	$completed_No = $j;
        print &quot;bwa-MEM completed!!! No.$completed_No : $sample[$j][0]\n&quot;;
}
#chdir &quot;../&quot;;" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>!/usr/bin/perl</span>

<span class="pl-c"><span class="pl-c">#</span>please run this job in a new clear fold</span>
<span class="pl-k">die</span> <span class="pl-s"><span class="pl-pds">"</span>usage: perl run_savi.pl &lt;Data_Index&gt; &lt;START&gt; &lt;END&gt; <span class="pl-cce">\n</span><span class="pl-pds">"</span></span> <span class="pl-k">if</span> <span class="pl-smi">@ARGV</span>!= 3;

<span class="pl-smi">$index_table</span> = <span class="pl-smi">$ARGV</span>[0];


<span class="pl-c1">open</span> INDEX, <span class="pl-smi">$index_table</span>;
<span class="pl-smi">$i</span> = 0;
<span class="pl-smi">$id</span> = 0;
<span class="pl-k">while</span>(<span class="pl-smi">$line</span> = &lt;INDEX&gt;){
	<span class="pl-c1">chomp</span>(<span class="pl-smi">$line</span>);
	<span class="pl-k">my</span> <span class="pl-smi">@temp</span>=<span class="pl-c1">split</span>(<span class="pl-s"><span class="pl-pds">'</span>\t<span class="pl-pds">'</span></span>,<span class="pl-smi">$line</span>);
	<span class="pl-smi">$sample</span>[<span class="pl-smi">$i</span>][0] = <span class="pl-smi">$temp</span>[1];
	<span class="pl-smi">$i</span>++;
}
<span class="pl-c1">close</span> INDEX;

<span class="pl-c"><span class="pl-c">#</span>`mkdir Dir_tmp_1`;</span>
<span class="pl-c"><span class="pl-c">#</span>chdir "./Dir_tmp_1";</span>
<span class="pl-c"><span class="pl-c">#</span></span>
<span class="pl-smi">$start</span> = <span class="pl-smi">$ARGV</span>[1];
<span class="pl-smi">$end</span> = <span class="pl-smi">$ARGV</span>[2];

<span class="pl-k">for</span>(<span class="pl-smi">$j</span>=<span class="pl-smi">$start</span>;<span class="pl-smi">$j</span>&lt;=<span class="pl-smi">$end</span>;<span class="pl-smi">$j</span>++){
	<span class="pl-s"><span class="pl-pds">`</span>sh do_savi_WES_2sample_GRCh38.p7.RefSeq.sh  <span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0] ../all_mapping_hg38/<span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0]_B_WES_hg38.sorted.MD.bam ../all_mapping_hg38/<span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0]_T_WES_hg38.sorted.MD.bam<span class="pl-pds">`</span></span>;
	<span class="pl-smi">$completed_No</span> = <span class="pl-smi">$j</span>;
        <span class="pl-c1">print</span> <span class="pl-s"><span class="pl-pds">"</span>bwa-MEM completed!!! No.<span class="pl-smi">$completed_No</span> : <span class="pl-smi">$sample</span>[<span class="pl-smi">$j</span>][0]<span class="pl-cce">\n</span><span class="pl-pds">"</span></span>;
}
<span class="pl-c"><span class="pl-c">#</span>chdir "../";</span></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash
##SBATCH -J savi24  #Slurm job name
#SBATCH --mail-user=njutangjihong@gmail.com 
#SBATCH --mail-type=end
##SBATCH -p x-gpu-share
##SBATCH -o tra_MD.out
##SBATCH -e tra_MD.err
##SBATCH -N 1 -n 24
#SBATCH --cpus-per-task=20
#SBATCH --exclusive

perl run_savi.pl list_SP_savi_CNV.txt $1 $2 2&gt; log/log.savi_$1_$2" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -J savi24  #Slurm job name</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=njutangjihong@gmail.com </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=end</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -p x-gpu-share</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -o tra_MD.out</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -e tra_MD.err</span>
<span class="pl-c"><span class="pl-c">#</span>#SBATCH -N 1 -n 24</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --cpus-per-task=20</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --exclusive</span>

perl run_savi.pl list_SP_savi_CNV.txt <span class="pl-smi">$1</span> <span class="pl-smi">$2</span> <span class="pl-k">2&gt;</span> log/log.savi_<span class="pl-smi">$1</span>_<span class="pl-smi">$2</span></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="sbatch -p x-gpu-share -J savi0 slurm.bwa_all 0 11
sbatch -p x-gpu-share -J savi24 slurm.savi_all 12 23 
sbatch -p x-gpu-share -J savi48 slurm.savi_all 24 35
sbatch -p x-gpu-share -J savi72 slurm.savi_all 36 46 " dir="auto">
                     <pre>sbatch -p x-gpu-share -J savi0 slurm.bwa_all 0 11
sbatch -p x-gpu-share -J savi24 slurm.savi_all 12 23 
sbatch -p x-gpu-share -J savi48 slurm.savi_all 24 35
sbatch -p x-gpu-share -J savi72 slurm.savi_all 36 46 </pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Part4 Copy Number Variation Analysis
                     </h1>
                     <a aria-label="Permalink: Part4 Copy Number Variation Analysis" class="anchor" href="#part4-copy-number-variation-analysis" id="user-content-part4-copy-number-variation-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/CNVkit%20pipeline%208760af4ce2ce4cfaa5410789947fa423/5._109.07.08-2_%25E6%25B9%25AF%25E7%25A1%25AF%25E5%25AE%2589%25E5%258D%259A%25E5%25A3%25AB-51.pdf">
                      5._109.07.08-2_湯硯安博士-51.pdf
                     </a>
                    </p>
                    <p dir="auto">
                     <a href="https://cnvkit.readthedocs.io/en/stable/pipeline.html" rel="nofollow">
                      https://cnvkit.readthedocs.io/en/stable/pipeline.html
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4.1 Preparation of input files
                     </h2>
                     <a aria-label="Permalink: 4.1 Preparation of input files" class="anchor" href="#41-preparation-of-input-files" id="user-content-41-preparation-of-input-files">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.1.1 Conda environment
                     </h3>
                     <a aria-label="Permalink: 4.1.1 Conda environment" class="anchor" href="#411-conda-environment" id="user-content-411-conda-environment">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.1.2 Access file for reference genome
                     </h3>
                     <a aria-label="Permalink: 4.1.2 Access file for reference genome" class="anchor" href="#412-access-file-for-reference-genome" id="user-content-412-access-file-for-reference-genome">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Baits files and Access files,
                    </p>
                    <p dir="auto">
                     For access file, hg38 and hg19 should use different calculated bed files. The calculate command is shown as follows;
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cnvkit.py access /scratch/PI/jgwang/jtangbd/reference/hg19bwa/hg19.fa \
-o access.hg19.bed" dir="auto">
                     <pre>cnvkit.py access /scratch/PI/jgwang/jtangbd/reference/hg19bwa/hg19.fa \
-o access.hg19.bed</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.1.3 Baits file
                     </h3>
                     <a aria-label="Permalink: 4.1.3 Baits file" class="anchor" href="#413-baits-file" id="user-content-413-baits-file">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.1.4 Target and Antitarget bed files
                     </h3>
                     <a aria-label="Permalink: 4.1.4 Target and Antitarget bed files" class="anchor" href="#414-target-and-antitarget-bed-files" id="user-content-414-target-and-antitarget-bed-files">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cnvkit.py autobin ../all_mapping_hg19/*B*.bam \
-t S07604514_Regions_hg19_220825.bed \
-g access.hg19.bed \
--target-output-bed SP_20220910_binauto.target.bed \
--antitarget-output-bed SP_20220910_binauto.antitarget.bed \
2&gt;log/log_autobin_auto" dir="auto">
                     <pre>cnvkit.py autobin ../all_mapping_hg19/<span class="pl-k">*</span>B<span class="pl-k">*</span>.bam \
-t S07604514_Regions_hg19_220825.bed \
-g access.hg19.bed \
--target-output-bed SP_20220910_binauto.target.bed \
--antitarget-output-bed SP_20220910_binauto.antitarget.bed \
<span class="pl-k">2&gt;</span>log/log_autobin_auto</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4.2 CNVkit Commands
                     </h2>
                     <a aria-label="Permalink: 4.2 CNVkit Commands" class="anchor" href="#42-cnvkit-commands" id="user-content-42-cnvkit-commands">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/Jihong-Tang/Cancer-genomics-analysis/blob/main/cnvkit_workflow.webp" rel="noopener noreferrer" target="_blank">
                      <img alt="CNVkit Pipeline" src="/Jihong-Tang/Cancer-genomics-analysis/raw/main/cnvkit_workflow.webp" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.2.1 Coverage
                     </h3>
                     <a aria-label="Permalink: 4.2.1 Coverage" class="anchor" href="#421-coverage" id="user-content-421-coverage">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     For each bam file, we need to calculate the coverage for target and antitarget regions, therefore, there will be two .cnn file as results for each bam file.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#!/bin/bash

PathBam=/scratch/PI/jgwang/jtangbd/projects/SP_WES/all_mapping_hg19
PathCNV=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19

for i in `cut -f2 list_SP_savi_CNV.txt`;do
        #i=`echo $n | cut -d "\t" -f 2`
        cnvkit.py coverage ${PathBam}/$i\_B_WES_hg19.sorted.MD.bam SP_20220910_binauto.target.bed -p 20 -o ${PathCNV}/coverage/$i\_B.target.cnn
done

for i in `cut -f2 list_SP_savi_CNV.txt`;do
        #i=`echo $n | cut -d "\t" -f 2`
        cnvkit.py coverage ${PathBam}/$i\_B_WES_hg19.sorted.MD.bam SP_20220910_binauto.antitarget.bed -p 20 -o ${PathCNV}/coverage/$i\_B.antitarget.cnn
done

for i in `cut -f2 list_SP_savi_CNV.txt`;do
        #i=`echo $n | cut -d "\t" -f 2`
        cnvkit.py coverage ${PathBam}/$i\_T_WES_hg19.sorted.MD.bam SP_20220910_binauto.target.bed -p 20 -o ${PathCNV}/coverage/$i\_T.target.cnn
done

for i in `cut -f2 list_SP_savi_CNV.txt`;do
        #i=`echo $n | cut -d "\t" -f 2`
        cnvkit.py coverage ${PathBam}/$i\_T_WES_hg19.sorted.MD.bam SP_20220910_binauto.antitarget.bed -p 20 -o ${PathCNV}/coverage/$i\_T.antitarget.cnn
done' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>

PathBam=/scratch/PI/jgwang/jtangbd/projects/SP_WES/all_mapping_hg19
PathCNV=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span>cut -f2 list_SP_savi_CNV.txt<span class="pl-pds">`</span></span><span class="pl-k">;</span><span class="pl-k">do</span>
        <span class="pl-c"><span class="pl-c">#</span>i=`echo $n | cut -d "\t" -f 2`</span>
        cnvkit.py coverage <span class="pl-smi">${PathBam}</span>/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>B_WES_hg19.sorted.MD.bam SP_20220910_binauto.target.bed -p 20 -o <span class="pl-smi">${PathCNV}</span>/coverage/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>B.target.cnn
<span class="pl-k">done</span>

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span>cut -f2 list_SP_savi_CNV.txt<span class="pl-pds">`</span></span><span class="pl-k">;</span><span class="pl-k">do</span>
        <span class="pl-c"><span class="pl-c">#</span>i=`echo $n | cut -d "\t" -f 2`</span>
        cnvkit.py coverage <span class="pl-smi">${PathBam}</span>/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>B_WES_hg19.sorted.MD.bam SP_20220910_binauto.antitarget.bed -p 20 -o <span class="pl-smi">${PathCNV}</span>/coverage/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>B.antitarget.cnn
<span class="pl-k">done</span>

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span>cut -f2 list_SP_savi_CNV.txt<span class="pl-pds">`</span></span><span class="pl-k">;</span><span class="pl-k">do</span>
        <span class="pl-c"><span class="pl-c">#</span>i=`echo $n | cut -d "\t" -f 2`</span>
        cnvkit.py coverage <span class="pl-smi">${PathBam}</span>/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>T_WES_hg19.sorted.MD.bam SP_20220910_binauto.target.bed -p 20 -o <span class="pl-smi">${PathCNV}</span>/coverage/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>T.target.cnn
<span class="pl-k">done</span>

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span>cut -f2 list_SP_savi_CNV.txt<span class="pl-pds">`</span></span><span class="pl-k">;</span><span class="pl-k">do</span>
        <span class="pl-c"><span class="pl-c">#</span>i=`echo $n | cut -d "\t" -f 2`</span>
        cnvkit.py coverage <span class="pl-smi">${PathBam}</span>/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>T_WES_hg19.sorted.MD.bam SP_20220910_binauto.antitarget.bed -p 20 -o <span class="pl-smi">${PathCNV}</span>/coverage/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>T.antitarget.cnn
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.2.2 Reference;Fix;Segment;Export
                     </h3>
                     <a aria-label="Permalink: 4.2.2 Reference;Fix;Segment;Export" class="anchor" href="#422-referencefixsegmentexport" id="user-content-422-referencefixsegmentexport">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The total pipeline for step 2 to 5 could be summarized in the following bash scripts
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash

PathCoverage=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19/coverage
FileReference=/scratch/PI/jgwang/jtangbd/reference/hg19bwa/hg19.fa
# step0 - make target and antitarget files
# step1 - cnvkit coverage for all samples

for i in `cut -f2 list_SP_savi_CNV.txt`;do
        # step2 - cnvkit reference for paired sample
        cnvkit.py reference ${PathCoverage}/$i\_B*.cnn -f ${FileReference} -o ./reference/$i\_reference.cnn

        # step3 - cnvkit fix
        cnvkit.py fix  ${PathCoverage}/$i\_T.target.cnn  ${PathCoverage}/$i\_T.antitarget.cnn ./reference/$i\_reference.cnn -i $i -o ./subtraction/$i\_fix.cnr

        # step4 - cnvkit segment
        cnvkit.py segment subtraction/$i\_fix.cnr -p 20 --drop-low-coverage -m cbs -t 0.000001 -o ./segmentation/$i\_segment.cbs.cns

        # step5 - cnvkit export and transfer results to seg files
        cnvkit.py call ./segmentation/$i\_segment.cbs.cns -o ./results/$i\_call.cbs.cns
        cnvkit.py export seg ./results/$i\_call.cbs.cns -o ./results/$i\_cbs.seg

        #tail -n+2 ./results/$i\_cbs.seg &gt;&gt; ./results/CNV_CGPA_103samples_pair_ref.seg
done" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>

PathCoverage=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19/coverage
FileReference=/scratch/PI/jgwang/jtangbd/reference/hg19bwa/hg19.fa
<span class="pl-c"><span class="pl-c">#</span> step0 - make target and antitarget files</span>
<span class="pl-c"><span class="pl-c">#</span> step1 - cnvkit coverage for all samples</span>

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span>cut -f2 list_SP_savi_CNV.txt<span class="pl-pds">`</span></span><span class="pl-k">;</span><span class="pl-k">do</span>
        <span class="pl-c"><span class="pl-c">#</span> step2 - cnvkit reference for paired sample</span>
        cnvkit.py reference <span class="pl-smi">${PathCoverage}</span>/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>B<span class="pl-k">*</span>.cnn -f <span class="pl-smi">${FileReference}</span> -o ./reference/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>reference.cnn

        <span class="pl-c"><span class="pl-c">#</span> step3 - cnvkit fix</span>
        cnvkit.py fix  <span class="pl-smi">${PathCoverage}</span>/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>T.target.cnn  <span class="pl-smi">${PathCoverage}</span>/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>T.antitarget.cnn ./reference/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>reference.cnn -i <span class="pl-smi">$i</span> -o ./subtraction/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>fix.cnr

        <span class="pl-c"><span class="pl-c">#</span> step4 - cnvkit segment</span>
        cnvkit.py segment subtraction/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>fix.cnr -p 20 --drop-low-coverage -m cbs -t 0.000001 -o ./segmentation/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>segment.cbs.cns

        <span class="pl-c"><span class="pl-c">#</span> step5 - cnvkit export and transfer results to seg files</span>
        cnvkit.py call ./segmentation/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>segment.cbs.cns -o ./results/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>call.cbs.cns
        cnvkit.py <span class="pl-k">export</span> seg ./results/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>call.cbs.cns -o ./results/<span class="pl-smi">$i</span><span class="pl-cce">\_</span>cbs.seg

        <span class="pl-c"><span class="pl-c">#</span>tail -n+2 ./results/$i\_cbs.seg &gt;&gt; ./results/CNV_CGPA_103samples_pair_ref.seg</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4.3 CNV Profile Generation
                     </h2>
                     <a aria-label="Permalink: 4.3 CNV Profile Generation" class="anchor" href="#43-cnv-profile-generation" id="user-content-43-cnv-profile-generation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Generate hg19 gene lists from gencode gtf files 
cat gencode.v41lift37.annotation.gtf | sed '/^#/d' | awk '$3==&quot;gene&quot;' | \
awk -F &quot;\t&quot; '{split($9, a, &quot;;&quot;); OFS=&quot;\t&quot;; print $1, $4, $5, $7, a[2], a[3]}' | \
awk -F &quot;\t&quot; '{split($5, b, &quot; &quot;); split($6, c, &quot; &quot;); OFS=&quot;\t&quot;; print $1,$2,$3,$4,b[2],c[2]}' | \
sed 's/&quot;//g' &gt; GRCh37_gencode_v41_genes.txt" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Generate hg19 gene lists from gencode gtf files </span>
cat gencode.v41lift37.annotation.gtf <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>/^#/d<span class="pl-pds">'</span></span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>$3=="gene"<span class="pl-pds">'</span></span> <span class="pl-k">|</span> \
awk -F <span class="pl-s"><span class="pl-pds">"</span>\t<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">'</span>{split($9, a, ";"); OFS="\t"; print $1, $4, $5, $7, a[2], a[3]}<span class="pl-pds">'</span></span> <span class="pl-k">|</span> \
awk -F <span class="pl-s"><span class="pl-pds">"</span>\t<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">'</span>{split($5, b, " "); split($6, c, " "); OFS="\t"; print $1,$2,$3,$4,b[2],c[2]}<span class="pl-pds">'</span></span> <span class="pl-k">|</span> \
sed <span class="pl-s"><span class="pl-pds">'</span>s/"//g<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> GRCh37_gencode_v41_genes.txt</pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Generate Protein coding genes without sex chromosome genes
cat GRCh37_gencode_v41_genes.txt | sed '/chr[X|Y]/d' | awk '$5==&quot;protein_coding&quot;' \
&gt; GRCh37_gencode_v41_genes_Protein_noSex.txt" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Generate Protein coding genes without sex chromosome genes</span>
cat GRCh37_gencode_v41_genes.txt <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>/chr[X|Y]/d<span class="pl-pds">'</span></span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>$5=="protein_coding"<span class="pl-pds">'</span></span> \
<span class="pl-k">&gt;</span> GRCh37_gencode_v41_genes_Protein_noSex.txt</pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# sort CNV result file for a single sample 
cat GRCh37_gencode_v41_genes_Protein_noSex.txt | sort -k1,1 -k2,2n &gt; GRCh37_gencode_v41_genes_Protein_noSex_sort.txt

cat ../results/P370376_cbs.seg | awk '{OFS=&quot;\t&quot;; print $2,$3,$4,$5,$6,$1}' | \
sort -k1,1 -k2,2n &gt; P370376_sort.bed" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> sort CNV result file for a single sample </span>
cat GRCh37_gencode_v41_genes_Protein_noSex.txt <span class="pl-k">|</span> sort -k1,1 -k2,2n <span class="pl-k">&gt;</span> GRCh37_gencode_v41_genes_Protein_noSex_sort.txt

cat ../results/P370376_cbs.seg <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{OFS="\t"; print $2,$3,$4,$5,$6,$1}<span class="pl-pds">'</span></span> <span class="pl-k">|</span> \
sort -k1,1 -k2,2n <span class="pl-k">&gt;</span> P370376_sort.bed</pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Map for single sample
bedtools map -a GRCh37_gencode_v41_genes_Protein_noSex_sort.txt \
-b P370376_sort.bed -c 5 | awk '{print $6&quot;\t&quot;$7}' | sort -k1 | \
sed &quot;1s/^/gene\t$var1\n/&quot; &gt; P370376_CNVprofile.txt" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Map for single sample</span>
bedtools map -a GRCh37_gencode_v41_genes_Protein_noSex_sort.txt \
-b P370376_sort.bed -c 5 <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{print $6"\t"$7}<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort -k1 <span class="pl-k">|</span> \
sed <span class="pl-s"><span class="pl-pds">"</span>1s/^/gene\t<span class="pl-smi">$var1</span>\n/<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> P370376_CNVprofile.txt</pre>
                    </div>
                    <p dir="auto">
                     The total bash script:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash

PathSeg=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19/results

for i in `cut -f2 list_SP_savi_CNV.txt`;do
        #i=`echo $n | cut -d &quot;\t&quot; -f 2`
        cat ${PathSeg}/${i}\_cbs.seg | awk '{OFS=&quot;\t&quot;; print $2,$3,$4,$5,$6,$1}' | \
        sort -k1,1 -k2,2n &gt; ./sort/${i}\_sort.bed
        # Map for single sample
        var1=${i}
        bedtools map -a GRCh37_gencode_v41_genes_Protein_noSex_sort.txt -b ./sort/${i}\_sort.bed -c 5 | \
        awk '{print $6&quot;\t&quot;$7}' | sort -k1 | \sed &quot;1s/^/gene\t$var1\n/&quot; &gt; ./profile/${i}\_CNVprofile.txt
done

mkdir -p tmp
cut -f1 ./profile/P370376_CNVprofile.txt &gt; gn.txt 
for i in `cut -f2 list_SP_savi_CNV.txt`;do
        cut -f2 ./profile/${i}\_CNVprofile.txt &gt; ./tmp/tmp.${i}
done
paste -d &quot;\t&quot; ./tmp/tmp* &gt; tmp.txt 
paste -d &quot;\t&quot; gn.txt tmp.txt &gt; SP_46samples_CNVPatients.txt
rm gn.txt tmp.txt" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>

PathSeg=/scratch/PI/jgwang/jtangbd/projects/SP_WES/cnvkit_hg19/results

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span>cut -f2 list_SP_savi_CNV.txt<span class="pl-pds">`</span></span><span class="pl-k">;</span><span class="pl-k">do</span>
        <span class="pl-c"><span class="pl-c">#</span>i=`echo $n | cut -d "\t" -f 2`</span>
        cat <span class="pl-smi">${PathSeg}</span>/<span class="pl-smi">${i}</span><span class="pl-cce">\_</span>cbs.seg <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{OFS="\t"; print $2,$3,$4,$5,$6,$1}<span class="pl-pds">'</span></span> <span class="pl-k">|</span> \
        sort -k1,1 -k2,2n <span class="pl-k">&gt;</span> ./sort/<span class="pl-smi">${i}</span><span class="pl-cce">\_</span>sort.bed
        <span class="pl-c"><span class="pl-c">#</span> Map for single sample</span>
        var1=<span class="pl-smi">${i}</span>
        bedtools map -a GRCh37_gencode_v41_genes_Protein_noSex_sort.txt -b ./sort/<span class="pl-smi">${i}</span><span class="pl-cce">\_</span>sort.bed -c 5 <span class="pl-k">|</span> \
        awk <span class="pl-s"><span class="pl-pds">'</span>{print $6"\t"$7}<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort -k1 <span class="pl-k">|</span> <span class="pl-cce">\s</span>ed <span class="pl-s"><span class="pl-pds">"</span>1s/^/gene\t<span class="pl-smi">$var1</span>\n/<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> ./profile/<span class="pl-smi">${i}</span><span class="pl-cce">\_</span>CNVprofile.txt
<span class="pl-k">done</span>

mkdir -p tmp
cut -f1 ./profile/P370376_CNVprofile.txt <span class="pl-k">&gt;</span> gn.txt 
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">`</span>cut -f2 list_SP_savi_CNV.txt<span class="pl-pds">`</span></span><span class="pl-k">;</span><span class="pl-k">do</span>
        cut -f2 ./profile/<span class="pl-smi">${i}</span><span class="pl-cce">\_</span>CNVprofile.txt <span class="pl-k">&gt;</span> ./tmp/tmp.<span class="pl-smi">${i}</span>
<span class="pl-k">done</span>
paste -d <span class="pl-s"><span class="pl-pds">"</span>\t<span class="pl-pds">"</span></span> ./tmp/tmp<span class="pl-k">*</span> <span class="pl-k">&gt;</span> tmp.txt 
paste -d <span class="pl-s"><span class="pl-pds">"</span>\t<span class="pl-pds">"</span></span> gn.txt tmp.txt <span class="pl-k">&gt;</span> SP_46samples_CNVPatients.txt
rm gn.txt tmp.txt</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Author
                     </h1>
                     <a aria-label="Permalink: Author" class="anchor" href="#author" id="user-content-author">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Jihong Tang &lt;
                     <a href="mailto:jtangbd@connect.ust.hk">
                      jtangbd@connect.ust.hk
                     </a>
                     &gt;instructed by @
                     <a href="https://github.com/JiguangWang">
                      Prof. Jiguang Wang
                     </a>
                     , @
                     <a href="https://github.com/qhmu">
                      Dr. Quanhua Mu
                     </a>
                     and @
                     <a href="https://github.com/ForceField17">
                      Dong Song
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Reference
                     </h1>
                     <a aria-label="Permalink: Reference" class="anchor" href="#reference" id="user-content-reference">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      Part0 Introduction and Preparation
                      <ul dir="auto">
                       <li>
                        <a href="https://docs.conda.io/projects/conda/en/latest/user-guide/index.html" rel="nofollow">
                         Conda User Guide
                        </a>
                       </li>
                       <li>
                        <a href="https://docs.brew.sh/" rel="nofollow">
                         Brew User Guide
                        </a>
                       </li>
                       <li>
                        <a href="https://github.com/twbattaglia/RNAseq-workflow">
                         Tom Battaglia's RNAseq-workflow
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      Part1 Sequencing Data Quality Control
                      <ul dir="auto">
                       <li>
                        Shifu Chen, Yanqing Zhou, Yaru Chen, Jia Gu; fastp: an ultra-fast all-in-one FASTQ preprocessor, Bioinformatics, Volume 34, Issue 17, 1 September 2018, Pages i884–i890,
                        <a href="https://doi.org/10.1093/bioinformatics/bty560" rel="nofollow">
                         https://doi.org/10.1093/bioinformatics/bty560
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      Part2 Genome Alignment
                      <ul dir="auto">
                       <li>
                        Li H. and Durbin R. (2009) Fast and accurate short read alignment with Burrows-Wheeler Transform. Bioinformatics, 25:1754-60. [PMID: 19451168]
                       </li>
                       <li>
                        “Picard Toolkit.” 2019. Broad Institute, GitHub Repository.
                        <a href="https://broadinstitute.github.io/picard/" rel="nofollow">
                         https://broadinstitute.github.io/picard/
                        </a>
                        ; Broad Institute
                       </li>
                      </ul>
                     </li>
                     <li>
                      Part3 Somatic Mutation Calling
                     </li>
                     <li>
                      Part4 Copy Number Variation Analysis
                     </li>
                    </ul>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="ndhKWyZK+ht/uMclcwQ5QKsBN6JcIqEPWVFyQGg+MLir/VC4dELK2BShP+ucAMYrkGMH5Dt9XvyTYsmVLwIF7Q==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 Notes for cancer genomics analysis pipeline and skills
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <h3 class="sr-only">
                 License
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="#GPL-3.0-1-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                   </path>
                  </svg>
                  GPL-3.0 license
                 </a>
                </div>
                <include-fragment src="/Jihong-Tang/Cancer-genomics-analysis/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2FJihong-Tang%2FCancer-genomics-analysis&amp;report=Jihong-Tang+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/Jihong-Tang/Cancer-genomics-analysis/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/Jihong-Tang/packages?repo_name=Cancer-genomics-analysis">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/Jihong-Tang/Cancer-genomics-analysis/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="Shell 67.7" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#89e051 !important;;width: 67.7%;">
                 </span>
                 <span aria-label="Perl 32.3" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#0298c3 !important;;width: 32.3%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/Jihong-Tang/Cancer-genomics-analysis/search?l=shell">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#89e051;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Shell
                  </span>
                  <span>
                   67.7%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/Jihong-Tang/Cancer-genomics-analysis/search?l=perl">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#0298c3;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Perl
                  </span>
                  <span>
                   32.3%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
