<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-6a292b1161ed.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - khidalgo85/training_transcriptomics: Slides
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:130128C:1B88784:6605CD7D" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDoxMzAxMjhDOjFCODg3ODQ6NjYwNUNEN0QiLCJ2aXNpdG9yX2lkIjoiNjQ1MjI0MTg1NjIzODgxNzgwNyIsInJlZ2lvbl9lZGdlIjoiaWFkIiwicmVnaW9uX3JlbmRlciI6ImlhZCJ9" data-pjax-transient="true" name="visitor-payload">
           <meta content="4a9c1872b664ffaac295ee42f1e1ed21fe0806cb96f6cf9c3b9e3f66a81ebf75" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:536617136" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="Slides. Contribute to khidalgo85/training_transcriptomics development by creating an account on GitHub." name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/khidalgo85/training_transcriptomics" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/1d950aa9b37822460cb826d8b05ce3b9278572251b7349c942527793bbdc087a/khidalgo85/training_transcriptomics" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - khidalgo85/training_transcriptomics: Slides" name="twitter:title">
                    <meta content="Slides. Contribute to khidalgo85/training_transcriptomics development by creating an account on GitHub." name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/1d950aa9b37822460cb826d8b05ce3b9278572251b7349c942527793bbdc087a/khidalgo85/training_transcriptomics" property="og:image">
                      <meta content="Slides. Contribute to khidalgo85/training_transcriptomics development by creating an account on GitHub." property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - khidalgo85/training_transcriptomics: Slides" property="og:title"/>
                        <meta content="https://github.com/khidalgo85/training_transcriptomics" property="og:url"/>
                        <meta content="Slides. Contribute to khidalgo85/training_transcriptomics development by creating an account on GitHub." property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="de8e1521fc37a178caaed17097d0545118b4eb0f01833d1d1eeeb65ac20b6ca7" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="1dd47e41bae3fb619dfa564356c6dee453350e97ff939ad02473ed6906f90b45" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/khidalgo85/training_transcriptomics git https://github.com/khidalgo85/training_transcriptomics.git" name="go-import"/>
                        <meta content="32853364" name="octolytics-dimension-user_id">
                         <meta content="khidalgo85" name="octolytics-dimension-user_login">
                          <meta content="536617136" name="octolytics-dimension-repository_id">
                           <meta content="khidalgo85/training_transcriptomics" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="536617136" name="octolytics-dimension-repository_network_root_id">
                               <meta content="khidalgo85/training_transcriptomics" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/khidalgo85/training_transcriptomics" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-11f6759e1cef.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/khidalgo85/training_transcriptomics","user_id":null}}' data-hydro-click-hmac="91997911848674a2b8fbfd9c3ecd0cb5e7836c08836f3f8923b08c25b822051c" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fkhidalgo85%2Ftraining_transcriptomics">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=7muZ1Jzi%2BBGDuzQGZqAwLwRSmvLj0PeS0wGwHvFn88LElIlNkG1%2FM5tA7ctTUyOqA7%2FvCAOKxUE5CHMvMS06G0wH1vh34gs3vibaGANtiOsMuDLv6TQwkVH68CAF2pJwzn4X0tiqrkewmm9Hk07GdmsfJ9I%2B2KuPGZtUzNUZFGEKi%2Fe%2BuI%2Bo7cWEwzDuELH9oB2StNvVMq3gKm4hnpEDESpMGaYoBxg5wOlxZZtIlS520KKv%2F%2B1RIhEvIgEDSJAU3mCA3%2BL%2FMu1A%2FKjL1ovV4dBWU5frXIFcvryf3Gzd9kuqW8crU8NtYs9bsYqITMJJY6hxcWSLvYC91U2abmDOz5FH4MMHMkzY9EKZzljmOqUMUbzCouGRG5aXbZSik9BmDcQKPryRFv6E0r7VPe4SFcO7mV2IzjnH2wLbUWyLRWZQdykgkq8y%2BcLyCzTL6fMBGi9tBvRcJkUVfYh%2FCNlsEPJwkHMXYjkykIt3rTsi1Swxxs55rUK7%2FDinJQZOjg2po1JK1ukfiwhgrmrKr9pnVt8VFITi3DblBJCjqkHlzmd2dv7kgLQ%3D--b8h2zvrxhiwMYFD%2F--EM4ihIJy9Wxb7XIskBVMcg%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="khidalgo85" data-current-repository="khidalgo85/training_transcriptomics" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="d3KuQnmt5NQfXDblAY90VOKurblCf2uHBqlJ7EnZ8LkNPq5PbnhcIg9Pxdn_QbiZbkcYVzpvrfHWcURFJ56JOw" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:khidalgo85/training_transcriptomics">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-c8e3189c-d1a3-4dfd-a137-c132ae6a39aa" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-c8e3189c-d1a3-4dfd-a137-c132ae6a39aa">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="D/+iLCD56FesRH30NMbHWGi8lR9Hp0aGNNfEphoqV8qxO7faX/bOLOZFHITlIh5A2QBOJcC8BlZ9niAjyVQ++g=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="+0A57RmZKyeSq4h/yMOYJXThpjX2Jt6mdWuURT4Zf4Rxx5exGbOW8N4Tp5TxFsb9sYlFO/SejFpSzeEeUIrUaQ==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="ih4M68I5JhmmGEA9CzfW1Ef+vFgHzf5oX1vRRRFweJnn0KS34MvVeyz1MMetZQDak6V7I4rzzOKPd7PibDU6+g==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="ZBBBb6ipT1MHhZZpNfUWqQQB2GBolzB/Hv/EiuMxMTiSDYQJVDp/LcIoOPCHGWKBskFQScAKvzQAo3+wKE2dKA==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/khidalgo85/training_transcriptomics","user_id":null}}' data-hydro-click-hmac="91997911848674a2b8fbfd9c3ecd0cb5e7836c08836f3f8923b08c25b822051c" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fkhidalgo85%2Ftraining_transcriptomics">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/khidalgo85/training_transcriptomics","user_id":null}}' data-hydro-click-hmac="91997911848674a2b8fbfd9c3ecd0cb5e7836c08836f3f8923b08c25b822051c" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=khidalgo85%2Ftraining_transcriptomics">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-cda8eeef-d1c7-4497-add6-852facf15f6a" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-7d687deb-fca6-4333-ae77-6439aabd4ca1" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-7d687deb-fca6-4333-ae77-6439aabd4ca1" id="tooltip-cda8eeef-d1c7-4497-add6-852facf15f6a" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/khidalgo85/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/khidalgo85" rel="author">
            khidalgo85
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/khidalgo85/training_transcriptomics">
            training_transcriptomics
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/khidalgo85/training_transcriptomics","user_id":null}}' data-hydro-click-hmac="2213285744a93d15c261469df78c053bd9417485d6d544804f0f4d4af2059fd1" data-view-component="true" href="/login?return_to=%2Fkhidalgo85%2Ftraining_transcriptomics" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":536617136,"auth_type":"LOG_IN","originating_url":"https://github.com/khidalgo85/training_transcriptomics","user_id":null}}' data-hydro-click-hmac="31f7ad7bc84afdb1930282a5195a36f7fcac1d4533cf9428696d199139ee602c" data-view-component="true" href="/login?return_to=%2Fkhidalgo85%2Ftraining_transcriptomics" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
             0
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":536617136,"auth_type":"LOG_IN","originating_url":"https://github.com/khidalgo85/training_transcriptomics","user_id":null}}' data-hydro-click-hmac="50b8ebed63e76cfb26d8d612c77bf827df9aa7bbcf04801777439b62a7776a4d" data-view-component="true" href="/login?return_to=%2Fkhidalgo85%2Ftraining_transcriptomics" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
              0
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <p class="f4 mb-3">
          Slides
         </p>
         <div class="mb-2 d-flex flex-items-center Link--secondary">
          <svg aria-hidden="true" class="octicon octicon-link flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
           </path>
          </svg>
          <span class="flex-auto min-width-0 css-truncate css-truncate-target width-fit">
           <a class="text-bold" href="https://khidalgo85.github.io/training_transcriptomics/#1" rel="noopener noreferrer" role="link" target="_blank" title="https://khidalgo85.github.io/training_transcriptomics/#1">
            khidalgo85.github.io/training_transcriptomics/#1
           </a>
          </span>
         </div>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/khidalgo85/training_transcriptomics/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           stars
          </a>
          <a class="Link--secondary no-underline mr-3" href="/khidalgo85/training_transcriptomics/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/khidalgo85/training_transcriptomics/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/khidalgo85/training_transcriptomics/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/khidalgo85/training_transcriptomics/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":536617136,"auth_type":"LOG_IN","originating_url":"https://github.com/khidalgo85/training_transcriptomics","user_id":null}}' data-hydro-click-hmac="50b8ebed63e76cfb26d8d612c77bf827df9aa7bbcf04801777439b62a7776a4d" data-view-component="true" href="/login?return_to=%2Fkhidalgo85%2Ftraining_transcriptomics" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/khidalgo85/training_transcriptomics","user_id":null}}' data-hydro-click-hmac="2213285744a93d15c261469df78c053bd9417485d6d544804f0f4d4af2059fd1" data-view-component="true" href="/login?return_to=%2Fkhidalgo85%2Ftraining_transcriptomics" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /khidalgo85/training_transcriptomics" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/khidalgo85/training_transcriptomics" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /khidalgo85/training_transcriptomics/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/khidalgo85/training_transcriptomics/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /khidalgo85/training_transcriptomics/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/khidalgo85/training_transcriptomics/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /khidalgo85/training_transcriptomics/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/khidalgo85/training_transcriptomics/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /khidalgo85/training_transcriptomics/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/khidalgo85/training_transcriptomics/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /khidalgo85/training_transcriptomics/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/khidalgo85/training_transcriptomics/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/khidalgo85/training_transcriptomics/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /khidalgo85/training_transcriptomics/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/khidalgo85/training_transcriptomics/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-cbbad524-7dfb-4ef8-ab3d-993f10d86b0a-list" aria-haspopup="true" aria-labelledby="tooltip-1e393589-ae48-41ae-b686-7b7c13863a6b" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-cbbad524-7dfb-4ef8-ab3d-993f10d86b0a-button" popovertarget="action-menu-cbbad524-7dfb-4ef8-ab3d-993f10d86b0a-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-cbbad524-7dfb-4ef8-ab3d-993f10d86b0a-button" id="tooltip-1e393589-ae48-41ae-b686-7b7c13863a6b" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-cbbad524-7dfb-4ef8-ab3d-993f10d86b0a-button" anchor-offset="normal" data-view-component="true" id="action-menu-cbbad524-7dfb-4ef8-ab3d-993f10d86b0a-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-cbbad524-7dfb-4ef8-ab3d-993f10d86b0a-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-cbbad524-7dfb-4ef8-ab3d-993f10d86b0a-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/khidalgo85/training_transcriptomics" id="item-56ad2aa1-6c5a-4090-876b-f1e65a649ba5" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/khidalgo85/training_transcriptomics/issues" id="item-bb5a270f-b03c-49a3-ba88-b91beaa35848" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/khidalgo85/training_transcriptomics/pulls" id="item-7715b21d-a38f-455b-9d0e-fb2c8d3836bf" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/khidalgo85/training_transcriptomics/actions" id="item-d8f5682e-196f-4414-be61-c5d4dc962a4a" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/khidalgo85/training_transcriptomics/projects" id="item-e8803611-9b8a-4e4e-8ab2-03a14d722cc0" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/khidalgo85/training_transcriptomics/security" id="item-42ead2c6-d5a3-4147-aefc-f6a5016e1f39" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/khidalgo85/training_transcriptomics/pulse" id="item-4aeae0b2-e2eb-48a0-8c0a-f87e4e55c0e9" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         khidalgo85/training_transcriptomics
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/khidalgo85/training_transcriptomics/spoofed_commit_check/fbbf0697981983444ef3774d2bde001743ce529d">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":536617136,"defaultBranch":"master","name":"training_transcriptomics","ownerLogin":"khidalgo85","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2022-09-14T14:27:32.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/32853364?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"master","listCacheKey":"v0:1663603046.70546","canEdit":false,"refType":"branch","currentOid":"fbbf0697981983444ef3774d2bde001743ce529d"},"tree":{"items":[{"name":"docs","path":"docs","contentType":"directory"},{"name":"imgs","path":"imgs","contentType":"directory"},{"name":"README.Rmd","path":"README.Rmd","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"},{"name":"training_transcriptomics.Rproj","path":"training_transcriptomics.Rproj","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":5,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/khidalgo85/training_transcriptomics/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/khidalgo85/training_transcriptomics.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone khidalgo85/training_transcriptomics","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/khidalgo85/training_transcriptomics/archive/refs/heads/master.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=536617136"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"7","overviewFiles":[{"displayName":"README.md","repoName":"training_transcriptomics","refName":"master","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\n\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://camo.githubusercontent.com/2efe01900e0b134a460b0d7da9510db33c484bdc19938f2499cf809364ffc01d/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f47454d4d2d4275696c64696e672d6f72616e6765\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/2efe01900e0b134a460b0d7da9510db33c484bdc19938f2499cf809364ffc01d/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f47454d4d2d4275696c64696e672d6f72616e6765\" alt=\"forthebadge\" data-canonical-src=\"https://img.shields.io/badge/GEMM-Building-orange\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://camo.githubusercontent.com/c2f114c16ec5427dcf908a2ff32771ff5a3082006bba7b1531627e825aee6e7b/68747470733a2f2f666f7274686562616467652e636f6d2f696d616765732f6261646765732f6275696c742d776974682d736369656e63652e737667\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/c2f114c16ec5427dcf908a2ff32771ff5a3082006bba7b1531627e825aee6e7b/68747470733a2f2f666f7274686562616467652e636f6d2f696d616765732f6261646765732f6275696c742d776974682d736369656e63652e737667\" alt=\"forthebadge\" data-canonical-src=\"https://forthebadge.com/images/badges/built-with-science.svg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePipeline: RNA-Seq \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/khidalgo85/training_transcriptomics/blob/master/imgs/1.png\"\u003e\u003cimg src=\"/khidalgo85/training_transcriptomics/raw/master/imgs/1.png\" align=\"right\" width=\"120px\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/h1\u003e\u003ca id=\"user-content-pipeline-rna-seq-\" class=\"anchor\" aria-label=\"Permalink: Pipeline: RNA-Seq \" href=\"#pipeline-rna-seq-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAutor: MsC. Kelly Hidalgo\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePipeline para el tratamiento de datos de RNA-seq y ensayos de expresin\ndiferencial. Para este tutorial sern usados los datos obtenidos en el\nartculo \u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7980702/pdf/peerj-09-11081.pdf\" rel=\"nofollow\"\u003eRNA-seq analysis of antibacterial mechanism of \u003cem\u003eCinnamomum\ncamphora\u003c/em\u003e essential oil against \u003cem\u003eEscherichia\ncoli\u003c/em\u003e\u003c/a\u003e.\nBrevemente, en este estudio fue evaluado el poder bactericida del aceite\nesencial de la planta \u003cem\u003eCinnamomum camphora\u003c/em\u003e contra \u003cem\u003eEscherichia coli\u003c/em\u003e\nATCC8739. Fue determinada la concentracin mnima inhibitoria (MIC). Con\nel fin de evaluar el mecanismo antibacterial del aceite esencial, fue\nllevado a cabo un ensayo de expresin diferencial bajo tres\ntratamientos, en los cuales fue secuenciado el transcriptoma (RNA-seq)\nusando la plataforma Illumina NovaSeq. Las tres condiciones fueron, \u003cem\u003eE.\ncoli\u003c/em\u003e tratada con agua, acetona y el aceite esencial en diferentes\nconcentraciones (1/8, 1/4 y 1/2 MIC).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLos datos pueden ser descargados desde el\n\u003ca href=\"https://www.ncbi.nlm.nih.gov/sra?term=SRP289443\" rel=\"nofollow\"\u003eNCBI\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ca href=\"/khidalgo85/training_transcriptomics/blob/master\"\u003e\u003cstrong\u003eSLIDES\u003c/strong\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-slides\" class=\"anchor\" aria-label=\"Permalink: SLIDES\" href=\"#slides\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eWORKFLOW\u003c/h1\u003e\u003ca id=\"user-content-workflow\" class=\"anchor\" aria-label=\"Permalink: WORKFLOW\" href=\"#workflow\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eimg src=imgs/workflow.png align=center/\u0026gt;\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e0. Introduccin a UNIX\u003c/h1\u003e\u003ca id=\"user-content-0-introduccin-a-unix\" class=\"anchor\" aria-label=\"Permalink: 0. Introduccin a UNIX\" href=\"#0-introduccin-a-unix\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl shell de Unix es un interpretador de lnea de comando. Es una\nherramenta poderosa que permite a los usuarios a ejecutar tareas\ncomplejas y poderosas, generalmente con algunas lneas de cdigo. En\neste tutorial, usted va entrar en la pantalla negra  , va a aprender\ny ejercitar algunos comandos bsicos e indispensables, para navegar y se\ndesenvolver en el \u003cem\u003eterminal\u003c/em\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNormalmente la interaccin del humano con el computador sucede por medio\nde un teclado y un mouse, interfaces grficas y/o sistemas de\nreconocimiento de voz. La manera ms comm de interacturar, es llamada\ncomo interface grfica de usurio \u003cem\u003e(\u003cstrong\u003eG\u003c/strong\u003eraphical \u003cstrong\u003eU\u003c/strong\u003eser\n\u003cstrong\u003eI\u003c/strong\u003enterface)\u003c/em\u003e. Cuando se trabaja en una GUI, las ordenes son dadas\nhaciendo click con el mouse y usando interaciones orientadas por\ndiferentes mens. Eso funciona muy bien para escalas pequeas, pues es\nmuy intuitivo. Pero ahora imagine que usted necesita ejecutar una tarea\nen mil archivos en diferentes carpetas, por ejemplo, copiar la ltima\nlnea de todos los archivos y pegarlos en un nico archivo. Seran\nnecesarias muchas horas (tal vez das) ejecutando el proceso y adems\npodra cometer errores. Es ah cuando shell es muy til, ya que por ser\nuna interface de lnea de comando \u003cem\u003e(\u003cstrong\u003eC\u003c/strong\u003eommand \u003cstrong\u003eL\u003c/strong\u003eine \u003cstrong\u003eI\u003c/strong\u003enterface)\u003c/em\u003e\ny un lenguaje de srcipt, permite procesar tareas repetitivas como las\ndel ejemplo, siendo realizadas de una forma rpida y automtica. El uso\nde shell es fundamental para el uso de uma amplia variedad de\nherramientas  de bioinformtica . Este tutorial le va a servir para\nhacer un uso eficaz de estos recursos.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e0.1. Login no servidor\u003c/h2\u003e\u003ca id=\"user-content-01-login-no-servidor\" class=\"anchor\" aria-label=\"Permalink: 0.1. Login no servidor\" href=\"#01-login-no-servidor\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara los usuarios de Windows es necesario la instalacin del software\n\u003ca href=\"https://mobaxterm.mobatek.net/download-home-edition.html#:~:text=MobaXterm%20Home%20Edition%20v22.1%0A(Installer%20edition)\" rel=\"nofollow\"\u003eMobaXterm\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/khidalgo85/training_transcriptomics/blob/master/imgs/SSH_MobaXterm.png\"\u003e\u003cimg src=\"/khidalgo85/training_transcriptomics/raw/master/imgs/SSH_MobaXterm.png\" align=\"center\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eDespus de instalado, clique en la opcin \u003cem\u003esession\u003c/em\u003e, escoja el icono\n\u003cem\u003essh\u003c/em\u003e y complete los datos del login: \u003cem\u003eremote host\u003c/em\u003e (direccin IP).\nDespus seleccione la opcin \u003cem\u003especify name\u003c/em\u003e y complete con el nombre del\nusuario. Por ltimo ser solicitada la contrasenha. Puede salvar los\ndatos del login.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNota\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSiempre que ud vea una caja como sta , es para que usted dgite\nel contenido en la lnea de comando y presione [enter]  para\n\u003cstrong\u003ecorrer\u003c/strong\u003e el comando.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"ls\"\u003e\u003cpre\u003els\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e0.2 Comandos bsicos\u003c/h2\u003e\u003ca id=\"user-content-02-comandos-bsicos\" class=\"anchor\" aria-label=\"Permalink: 0.2 Comandos bsicos\" href=\"#02-comandos-bsicos\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003essh\u003c/code\u003e\u003c/h3\u003e\u003ca id=\"user-content-ssh\" class=\"anchor\" aria-label=\"Permalink: ssh\" href=\"#ssh\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003essh\u003c/code\u003e (Secure Shell) es un protocolo que garantiza que el usuario y el\nservidor remoto intercambien informaciones de manera segura e dinmica.\nSirve especificamente para conectarse a un servidor remoto.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"$ ssh -X \u0026lt;user_name\u0026gt;@\u0026lt;ip.address\u0026gt;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e$ ssh -X \u0026lt;user_name\u0026gt;@\u0026lt;ip.address\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003epwd\u003c/code\u003e (\u003cem\u003ePrint Working Directory\u003c/em\u003e)\u003c/h3\u003e\u003ca id=\"user-content-pwd-print-working-directory\" class=\"anchor\" aria-label=\"Permalink: pwd (Print Working Directory)\" href=\"#pwd-print-working-directory\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eLos directorios  son como cajas. Siempre que usted est usando el\nshell, usted est dentro de alguna caja \u003cstrong\u003e(Directorio)\u003c/strong\u003e  de su\ncomputador  o servidor, llamado \u003cstrong\u003edirectorio actual de trabajo\u003c/strong\u003e. Los\ncomandos solamente leen y graban archivos en el directorio actual de\ntrabajo (s usted no indicar otro camino), por lo tanto, es\n\u003cstrong\u003eimportante\u003c/strong\u003e  saber en cual directorio est antes de\nejecutar un comando. El comando \u003ccode\u003epwd\u003c/code\u003e muestra donde est.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pwd\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epwd\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\" /home/user\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e /home/user\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSintaxis de los comandos\u003c/h3\u003e\u003ca id=\"user-content-sintaxis-de-los-comandos\" class=\"anchor\" aria-label=\"Permalink: Sintaxis de los comandos\" href=\"#sintaxis-de-los-comandos\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003ecomando [opciones] [archivo]\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEl comando es separado de las opciones (o argumentos, flags) e do\narchivo  por un espacio. Los argumentos pueden mudar el comportamiento\ndel comando. Y el archivo  indica para el comando sobre lo que va a\noperar (p.e. archivos  y directorios ). A veces los argumentos y el\narchivo  son llamados de \u003cstrong\u003eparmetros\u003c/strong\u003e. Un comando puede tener ms de\nun argumento y/o archivos  y tambin podria no tener ninguno de los\ndos. Las opciones usualmente tienen un guin y una letra (p.e. \u003ccode\u003e-h\u003c/code\u003e) o\ndos guiones y una palabra (p.e. \u003ccode\u003e--help\u003c/code\u003e), \u003cstrong\u003esin\u003c/strong\u003e espacio entre lo(s)\nguion(es) y la letra/palabra. Vamos a ver con ejemplos prcticos.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eImportante:\u003c/strong\u003e el lenguaje usado en UNIX es sensible a\nletras maysculas e minsculas (\u003cem\u003ecase sensitive\u003c/em\u003e), es un error comn.\n\u003cstrong\u003eSiempre atento\u003c/strong\u003e \u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003els\u003c/code\u003e Listar\u003c/h3\u003e\u003ca id=\"user-content-ls-listar\" class=\"anchor\" aria-label=\"Permalink: ls Listar\" href=\"#ls-listar\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCon el comando \u003ccode\u003els\u003c/code\u003e ud puede ver (listar) lo que hay dentro del\n\u003cstrong\u003edirectorio actual de trabajo\u003c/strong\u003e .\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"ls \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003els \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"Documentos Downloads Imagens\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eDocumentos Downloads Imagens\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUsted puede usar el \u003cstrong\u003eargumento\u003c/strong\u003e \u003ccode\u003e-F\u003c/code\u003e para indicar para el comando \u003ccode\u003els\u003c/code\u003e\nque muestre lo que es cada elemento (directorio o archivo). \u003ccode\u003e/\u003c/code\u003e\nsignifica que es un directorio , el \u003ccode\u003e*\u003c/code\u003e significa que es un archivo\nejecutable y s no tiene ningn smbolo significa que es un archivo .\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEl \u003cem\u003e\u003cstrong\u003eflag\u003c/strong\u003e\u003c/em\u003e  \u003ccode\u003e--help\u003c/code\u003e  es bien importante:exclamation:, y puede ser\nusado en \u003cstrong\u003ecualquier\u003c/strong\u003e comando. El muestra ms informacin sobre el\ncomando, y como usarlo . Explore o men help\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eOutros \u003cstrong\u003eflags\u003c/strong\u003e  interessantes so: \u003ccode\u003els -l\u003c/code\u003e, que lista o contedo da\npasta  com informaes extras, como as permises, o tamanho, a data  e\nhora  de criao, e o nome de cada um dos elementos. \u003ccode\u003els -a\u003c/code\u003e que lista\ntodos os arquivos includos os ocultos. \u003ccode\u003els -t\u003c/code\u003e lista os arquivos em\nordem cronolgica.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eOtros \u003cstrong\u003eflags\u003c/strong\u003e  interesantes son: \u003ccode\u003els -l\u003c/code\u003e, que lista el contenido de\nla carpeta  con informaciones extras, como los permisos, el tamao, la\nfecha  y hora  de creacin, y el nombre de cada uno de los\nelementos.\u003ccode\u003els -a\u003c/code\u003e que lista todos los archivos includos los ocultos.\n\u003ccode\u003els -t\u003c/code\u003e lista los archivos en orden cronolgica.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"ls -l\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003els -l\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eLos permisos se deben entender as: d = Directorio rwx = \u003cem\u003eread, write\u003c/em\u003e y\n\u003cem\u003eexecute\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSe deben leer de tres en tres caracteres. As, en nuestro ejemplo de\narriba, ninguno de los elementos son directorios porque no inician con\nla letra d.La primera tripleta tiene las letras r e w, o sea el usuario\npuede leer y escribir esos elementos. La segunda tiene solamente la\nletra r, o sea el grupo (pueden ser creados grupos de usuarios) solo\npueden leer esos archivos. Por ltimo, la ltima tripleta, tambin solo\ntiene a letra r, o sea \u003cstrong\u003etodos\u003c/strong\u003e los usuarios solo pueden leer los\nelementos listados.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003emkdir\u003c/code\u003e Make Dir\u003c/h3\u003e\u003ca id=\"user-content-mkdir-make-dir\" class=\"anchor\" aria-label=\"Permalink: mkdir Make Dir\" href=\"#mkdir-make-dir\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUsted ya aprendi a explorar carpetas  y archivos , ahora va a\naprender como se crean. El comando \u003ccode\u003emkdir\u003c/code\u003e sirve para crear carpetas .\nVamos a crear varias \u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Crea una carpeta llamada datos\nmkdir datos\n# Crea una carpeta llamada analisis y otra llamada tutorial\nmkdir analisis tutorial\n# Lista el contenido \nls -F\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Crea una carpeta llamada datos\nmkdir datos\n# Crea una carpeta llamada analisis y otra llamada tutorial\nmkdir analisis tutorial\n# Lista el contenido \nls -F\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNote que: 1) puede crear ms de una carpeta por lnea de comando y 2)\npuede diferenciar los archivos de los directorios, pues stos estn de\ncolor azul y con una \u003ccode\u003e/\u003c/code\u003e al final.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUsando el comando \u003ccode\u003etree\u003c/code\u003e puede ver graficamente la organizacin de los\ndirectorios y archivos dentro del directorio actual.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cem\u003eTips\u003c/em\u003e\u003c/h4\u003e\u003ca id=\"user-content-tips\" class=\"anchor\" aria-label=\"Permalink: Tips\" href=\"#tips\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eNo  use espacios en los nombres de sus carpetas  o archivos \n(p.e. \u003cdel\u003ecoleta 2020\u003c/del\u003e). Siempre separe las palabras con \u003ccode\u003e-\u003c/code\u003e, \u003ccode\u003e_\u003c/code\u003e o\ncon maysculas (p.e. \u003ccode\u003ecoleta_2020\u003c/code\u003e, \u003ccode\u003ecoleta-2020\u003c/code\u003e o \u003ccode\u003ecoletaMaio\u003c/code\u003e )\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eNo  comience nombres con \u003ccode\u003e-\u003c/code\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eAsigne para sus elementos nombres fciles de recordar y escribir y\nque describan lo que contienen.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eNo use caracteres espaciais\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003ecd\u003c/code\u003e Change Directory\u003c/h3\u003e\u003ca id=\"user-content-cd-change-directory\" class=\"anchor\" aria-label=\"Permalink: cd Change Directory\" href=\"#cd-change-directory\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl comando \u003ccode\u003ecd\u003c/code\u003e sirve para cambiar de \u003cstrong\u003edirectorio actual de trabajo\u003c/strong\u003e\n. Vaya al directorio \u003ccode\u003edatos/\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd datos/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd datos/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCondirme donde est con el comando \u003ccode\u003epwd\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePara volver al directorio anterior\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd ..\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd ..\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl \u003ccode\u003e..\u003c/code\u003e significa: directorio que contiene el directorio actual (o sea\nun directorio arriba). Confirme con:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pwd\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epwd\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/user\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/user\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAhora entre en la carpeta  \u003ccode\u003eanalisis/\u003c/code\u003e y cree otra  llamada\n\u003ccode\u003ecolecta_2020\u003c/code\u003e, confirme con \u003ccode\u003els\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Cambie de directorio\ncd analisis/\n## Cree un nuevo directorio\nmkdir colecta_2020\n## Confirme\nls\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Cambie de directorio\ncd analisis/\n## Cree un nuevo directorio\nmkdir colecta_2020\n## Confirme\nls\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"colecta_2020/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecolecta_2020/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAhora entre a la carpeta que acab de crear usando el comando \u003ccode\u003ecd\u003c/code\u003e y\nverifique donde est con \u003ccode\u003epwd\u003c/code\u003e, despus regrese para el directorio base\n\u003ccode\u003e/home/user\u003c/code\u003e  y verifique nuevamente.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAl finalizar esos comandos verifique la organizacin de los directorios\nusando el comando \u003ccode\u003etree\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePor ltimo vuelva a la carpeta \u003ccode\u003ehome/user/analisis/coleta_2020\u003c/code\u003e usando\n\u003ccode\u003ecd\u003c/code\u003e en una lnea de comando solamente. Confirme con \u003ccode\u003epwd\u003c/code\u003e. Vuelva\nnuevamente para la carpeta raz tambin con solo una lnea de comando.\n\u003cstrong\u003ePista\u003c/strong\u003e, recuerde que \u003ccode\u003e..\u003c/code\u003e significa directorio de arriba.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cem\u003eTip\u003c/em\u003e \u003c/h4\u003e\u003ca id=\"user-content-tip-\" class=\"anchor\" aria-label=\"Permalink: Tip \" href=\"#tip-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUsted puede usar la tecla Tab  para autocompletar las\npalabras. As, economiza tiempo , y evita errores de digitacin ,\nporque el sistema solo va a completar los nombres que existan en el\n\u003cstrong\u003edirectorio actual de trabajo\u003c/strong\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSolo tiene que esrcibir las primeras letras de la palabra. p.e. col:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Cambie de carpeta\ncd analisis/col\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Cambie de carpeta\ncd analisis/col\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOprima  [Tab]. Automticamente se no existe ningn otro\nelemento que comience por col, la palabra coleta_2020 va a ser\nautocompletada.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd analises/colecta_2020\"\u003e\u003cpre lang=\"coffeesrcipt=\" class=\"notranslate\"\u003e\u003ccode\u003ecd analises/colecta_2020\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eS por alguna razn existe otro elemento que tambin comience con col,\noprima dos vezes Tab  y el sistema va a mostrar las opciones\nde palabras que inicien con col.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Estando en analisis/ cree uma pasta llamada colecta_2019\nmkdir colecta_2019\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Estando en analisis/ cree uma pasta llamada colecta_2019\nmkdir colecta_2019\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAhora, usted quiere entrar en esa carpeta nueva. Use el comando \u003ccode\u003ecd\u003c/code\u003e\npara cambiar de carpeta y use [Tab] [Tab] para que el sistema\nmuestre las opciones con comienzo col.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd col\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd col\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e[Tab] [Tab]\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003enano\u003c/code\u003e (editor de texto)\u003c/h3\u003e\u003ca id=\"user-content-nano-editor-de-texto\" class=\"anchor\" aria-label=\"Permalink: nano (editor de texto)\" href=\"#nano-editor-de-texto\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e \u003ccode\u003enano \u0026lt;nombre_del_archivo\u0026gt;\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAhora ud va a crear un archivo  \u003ccode\u003etest.txt\u003c/code\u003e dentro de la carpeta \n\u003ccode\u003etutorial/\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Cambie de directorio\ncd tutorial/\n## Abra el editor de texto nano \nnano test.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Cambie de directorio\ncd tutorial/\n## Abra el editor de texto nano \nnano test.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCuando abrir el editor de texto, escriba: Este es un test y cierre el\narchivo con [Ctrl + o] para salvar. En la lnea blanca abajo, el\neditor de texto preguntar s quiere mantener el nombre que le dio al\ncomienzo \u003ccode\u003etest.txt\u003c/code\u003e. [Enter] para confirmar. [Ctrl + x] para salir.\nConfirme que el archivo fue creado con el comando \u003ccode\u003els\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eS ud quiere entrar de nuevo al arquivo y modificarlo, deve usar el de\nnuevo el comando \u003ccode\u003enano test.txt\u003c/code\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003emv\u003c/code\u003e move\u003c/h3\u003e\u003ca id=\"user-content-mv-move\" class=\"anchor\" aria-label=\"Permalink: mv move\" href=\"#mv-move\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl comando \u003ccode\u003emv\u003c/code\u003e sirve para mover Archivos  de una carpeta  a otra .\nAdems este comando tambin puede ser usado para cambiar los nombres de\nlos elementos. Para mover un archivo de una carpeta  a otra  la\n\u003cstrong\u003esintaxis\u003c/strong\u003e del comando es: \u003ccode\u003emv archivo.txt nuevodirectorio/\u003c/code\u003e. En\nnuestro ejemplo:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"mv test.txt ../datos/\n## Confirme\nls ../datos/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emv test.txt ../datos/\n## Confirme\nls ../datos/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUd uso \u003ccode\u003e../\u003c/code\u003e, porque ud estaba dentro de la carpeta \n\u003ccode\u003e/home/user/tutorial/\u003c/code\u003e y necesitaba volver para  \u003ccode\u003e/home/user/\u003c/code\u003e (\ncarpeta arriba de \u003ccode\u003etutorial/\u003c/code\u003e) para continuar el camino para  \u003ccode\u003edatos/\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAhora use el comando \u003ccode\u003emv\u003c/code\u003e para cambiar el nombre del archivo \n\u003ccode\u003etest.txt\u003c/code\u003e por \u003ccode\u003eprueba.txt\u003c/code\u003e. \u003cstrong\u003eSintaxis\u003c/strong\u003e\n\u003ccode\u003emv nombredelarchivo.txt nuevonombredelarchivo.txt\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Cambie de directorio\ncd ../datos/\n## Confirme\nls\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Cambie de directorio\ncd ../datos/\n## Confirme\nls\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"test.txt\n\n## Cambie el nombre del archivo\nmv test.txt prueba.txt\n## Confirme\nls\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etest.txt\n\n## Cambie el nombre del archivo\nmv test.txt prueba.txt\n## Confirme\nls\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"prueba.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eprueba.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003ecp\u003c/code\u003e copy\u003c/h3\u003e\u003ca id=\"user-content-cp-copy\" class=\"anchor\" aria-label=\"Permalink: cp copy\" href=\"#cp-copy\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl comando \u003ccode\u003ecp\u003c/code\u003e es similar a \u003ccode\u003emv\u003c/code\u003e, pero l copia el archivo  en vez de\nmoverlo. Ahora copie el archivo \u003ccode\u003e/home/user/datos/prueba.txt\u003c/code\u003e en la\ncarpeta \u003ccode\u003e/home/user/analisis/\u003c/code\u003e. \u003cstrong\u003eATENCIN:exclamation:! Haga eso desde\nel directorio inicial\u003c/strong\u003e \u003ccode\u003e/home/user/\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e\n\u003ccode\u003ecp directorio/nombredelarchivo.txt nuevodirectorio/nombredelarchivo.txt\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Dnde estoy?\npwd\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Dnde estoy?\npwd\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/user/datos/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/user/datos/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNo olvide del \u003cstrong\u003etip\u003c/strong\u003e  de usar [Tab]\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Copiar el archivo\ncp datos/prueba.txt analisis/\n## Confirme\nls datos/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Copiar el archivo\ncp datos/prueba.txt analisis/\n## Confirme\nls datos/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"prueba.txt\n\n## Listar\nls analisis/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eprueba.txt\n\n## Listar\nls analisis/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"colecta_2019   colecta_2020   prueba.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecolecta_2019   colecta_2020   prueba.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUd puede usar el comando \u003ccode\u003ecp\u003c/code\u003e para copiar varios archivos  en una\nlnea de comando solamente.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxe\u003c/strong\u003e\n\u003ccode\u003ecp archivo1.txt archivo2.txt archivo3.txt carpetadedestino/\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003erm\u003c/code\u003e remove\u003c/h3\u003e\u003ca id=\"user-content-rm-remove\" class=\"anchor\" aria-label=\"Permalink: rm remove\" href=\"#rm-remove\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCon o comando \u003ccode\u003erm\u003c/code\u003e ud puede remover archivos  y/o carpetas .\n\u003cstrong\u003eCUIDADO\u003c/strong\u003e \u003cstrong\u003ePRECAUCIN\u003c/strong\u003e este comando no\ntiene reversa, una vez ud oprima [enter] no hay como recuperar el\narchivo  o carpeta , entonces revise y piense bien antes de rodar este\ncomando.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003erm directorio/nombredelarchivo.txt\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUd va a remover el archivo \u003ccode\u003eprueba.txt\u003c/code\u003e de la carpeta \n\u003ccode\u003e/home/user/datos\u003c/code\u003e. S necesita, use el comando \u003ccode\u003epwd\u003c/code\u003e para confirmar en\nque directorio est.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Remover desde /home/user/\nrm dados/prova.txt\n## Confirme\nls\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Remover desde /home/user/\nrm dados/prova.txt\n## Confirme\nls\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara eliminar una carpeta  ud necesita del \u003cstrong\u003eflag\u003c/strong\u003e \u003ccode\u003e-r\u003c/code\u003e. Elimine la\ncarpeta  \u003ccode\u003edatos/\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Remover el directorio\nrm -r datos/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Remover el directorio\nrm -r datos/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eOtros comandos\u003c/h3\u003e\u003ca id=\"user-content-otros-comandos\" class=\"anchor\" aria-label=\"Permalink: Otros comandos\" href=\"#otros-comandos\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara los siguientes comando vamos a crear dos nuevos archivos de texto\nllamados \u003ccode\u003ebssA_1.txt\u003c/code\u003e y \u003ccode\u003ebssA_2.txt\u003c/code\u003e, en cada uno vamos a pegar una\nsecuencia del gen \u003cem\u003ebssA\u003c/em\u003e que codifica para la enzima \u003cem\u003eBenzylsuccinate\nsynthase\u003c/em\u003e.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Cambiar de directorio\ncd tutorial/\n##Abrir el editor de texto nano\nnano bssA_1.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Cambiar de directorio\ncd tutorial/\n##Abrir el editor de texto nano\nnano bssA_1.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCopie a sequncia\n\u003ca href=\"https://www.ncbi.nlm.nih.gov/nuccore/MW762608.1?report=fasta\" rel=\"nofollow\"\u003eaqui\u003c/a\u003e.\nCopie la secuencia desde\n\u003ca href=\"https://www.ncbi.nlm.nih.gov/nuccore/MW762608.1?report=fasta\" rel=\"nofollow\"\u003eaqui\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAtencin:\u003c/strong\u003e El comando de teclas  [Ctrl + V] no funciona\nen el terminal de Linux. Use [Ctrl + Shift + V].\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSalve y cierre el editor. S quiere, puede confirmar que el archivo fue\ncreado con el comando \u003ccode\u003els\u003c/code\u003e, y entrando en el archivo con el comando\n\u003ccode\u003enano\u003c/code\u003e e el nombre del archivo.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Abrir el editor de texto nano\nnano bssA_1.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Abrir el editor de texto nano\nnano bssA_1.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRepita el proceso para crear el archivo \u003ccode\u003ebssA_2.txt\u003c/code\u003e copiando sta\n\u003ca href=\"https://www.ncbi.nlm.nih.gov/nuccore/FJ810633.1?report=fasta\" rel=\"nofollow\"\u003esecuencia\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003eless\u003c/code\u003e\u003c/h3\u003e\u003ca id=\"user-content-less\" class=\"anchor\" aria-label=\"Permalink: less\" href=\"#less\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEste comando serve para imprimir en la pantalla el contenido de un\narchivo. Para salir digite \u003ccode\u003eq\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"less bssA_1.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eless bssA_1.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara salir [Ctrl + c]\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003ehead\u003c/code\u003e\u003c/h3\u003e\u003ca id=\"user-content-head\" class=\"anchor\" aria-label=\"Permalink: head\" href=\"#head\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMuestra las primeras 10 lineas del archivo\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"head bssA_2.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ehead bssA_2.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003etail\u003c/code\u003e\u003c/h3\u003e\u003ca id=\"user-content-tail\" class=\"anchor\" aria-label=\"Permalink: tail\" href=\"#tail\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMuestra las ltimas 10 lineas del archivo\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"tail bssA_2.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etail bssA_2.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eS ud quiere, puede aumentar el nmero de lneas que esos dos ltimos\ncomandos muestran, adicionado un argumento con el nmero de lneas que\ndesee imprimir en la pantalla.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## ltimas 12 linhas\ntail -12 bssA_1.txt\n## Primeras 13 linhas\nhead -13 bssA_2.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## ltimas 12 linhas\ntail -12 bssA_1.txt\n## Primeras 13 linhas\nhead -13 bssA_2.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003ecat\u003c/code\u003e concatenate\u003c/h3\u003e\u003ca id=\"user-content-cat-concatenate\" class=\"anchor\" aria-label=\"Permalink: cat concatenate\" href=\"#cat-concatenate\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEste comando sirve para juntar dos archivos en uno. Es muy til para\njuntar archivos tipo \u003ccode\u003e.fasta\u003c/code\u003e con secuencias.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Concatenar\ncat bssA_1.txt bssA_2.txt \u0026gt; bssA_all.txt\n## Confirme\nls\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Concatenar\ncat bssA_1.txt bssA_2.txt \u0026gt; bssA_all.txt\n## Confirme\nls\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEn el ejemplo anterior, fueron concatenados los archivos \u003ccode\u003ebssA_1.txt\u003c/code\u003e y\n\u003ccode\u003ebssA_2.txt\u003c/code\u003e dentro del archivo \u003ccode\u003ebss_all.txt\u003c/code\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003ewc\u003c/code\u003e Word count\u003c/h3\u003e\u003ca id=\"user-content-wc-word-count\" class=\"anchor\" aria-label=\"Permalink: wc Word count\" href=\"#wc-word-count\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEste comando sirve para contar las lneas, palabras o caracteres de los\narchivos.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Contar lineas, palabras y caracteres\nwc bssA_1.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Contar lineas, palabras y caracteres\nwc bssA_1.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"28        36      1907 bssA_1.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e28        36      1907 bssA_1.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl archivo  \u003ccode\u003ebssA_1.txt\u003c/code\u003e tiene 28 lineas, 36 palabras y 1907\ncaracteres.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003egrep\u003c/code\u003e\u003c/h3\u003e\u003ca id=\"user-content-grep\" class=\"anchor\" aria-label=\"Permalink: grep\" href=\"#grep\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCom o \u003ccode\u003egrep\u003c/code\u003e ud puede buscar un patrn dentro de un archivo. Por ejemplo\nen un archivo de secuencias \u003ccode\u003e.fasta\u003c/code\u003e cada secuencia comienza con el\nsimbolo \u003ccode\u003e\u0026gt;\u003c/code\u003e o podria buscar una secuencia de nucletidos especfica\n(p.e. ATCTTGCA)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"grep -c '\u0026gt;' bssA_2.txt\n\ngrep -c 'CGA' bssA_1.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egrep -c '\u0026gt;' bssA_2.txt\n\ngrep -c 'CGA' bssA_1.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl \u003cem\u003eflag\u003c/em\u003e \u003ccode\u003e-c\u003c/code\u003e es para que el comando solo muestre el nmero de lneas\nque hacen \u003cem\u003ematch\u003c/em\u003e con lo que est siendo procurado.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEl comando \u003ccode\u003egrep\u003c/code\u003e tiene varios \u003cem\u003eflags\u003c/em\u003e diferentes, recuerde que puede\nconocer todos ellos digitando \u003ccode\u003egrep --help\u003c/code\u003e para entrar en el men de\nayuda del comando.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003efind\u003c/code\u003e\u003c/h3\u003e\u003ca id=\"user-content-find\" class=\"anchor\" aria-label=\"Permalink: find\" href=\"#find\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCon el comando \u003ccode\u003efind\u003c/code\u003e ud puede buscar archivos con una palabra clave.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Buscar\nfind . -name '*.txt'\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Buscar\nfind . -name '*.txt'\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eLea el comando as: buscar en el `\u003ccode\u003edirectorio actual de trabajo\u003c/code\u003e\ncualquier archivo que termine con \u003ccode\u003e*.txt\u003c/code\u003e. El simbolo \u003ccode\u003e*\u003c/code\u003e significa\ncualquier caracter. S ud escribe `bss*, el sistema va a entender que\nud est interesado en cualquier elemente que comience con bss.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"./bssA_1.txt\n./bssA_2.txt\n./bssA_all.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e./bssA_1.txt\n./bssA_2.txt\n./bssA_all.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUd podria buscar la palabra en cualquier directorio  del pc modificando\nel comando. Procure todos os archivos terminados en .txt en la carpeta\nraz \u003ccode\u003e/home/user/\u003c/code\u003e.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Procurar\nfind ../ -name \u0026quot;*.txt\u0026quot;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Procurar\nfind ../ -name \"*.txt\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOutros exemplos:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"find ../datos/ -name 'prueba*' # busca archivos que comienzan con prueba dentro de la carpeta /home/user/datos/\nfind ../datos/ -iname 'prueba*' #igual pero ignora s son maysculas o minusclas.\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efind ../datos/ -name 'prueba*' # busca archivos que comienzan con prueba dentro de la carpeta /home/user/datos/\nfind ../datos/ -iname 'prueba*' #igual pero ignora s son maysculas o minusclas.\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003ewget\u003c/code\u003e\u003c/h3\u003e\u003ca id=\"user-content-wget\" class=\"anchor\" aria-label=\"Permalink: wget\" href=\"#wget\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl comando \u003ccode\u003ewget\u003c/code\u003e sirve para hacer \u003cem\u003edownload\u003c/em\u003e de archivos en la web y\nalmacenarlos en el \u003cstrong\u003edirectorio actual de trabajo\u003c/strong\u003e, es muy til para\ndescargar bases de datos.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"wget https://direccionoweb.com\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ewget https://direccionoweb.com\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003egzip\u003c/code\u003e\u003c/h3\u003e\u003ca id=\"user-content-gzip\" class=\"anchor\" aria-label=\"Permalink: gzip\" href=\"#gzip\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEste comando es para compactar y descompactar archivos .\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Comprimir\ngzip tutorial/*\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Comprimir\ngzip tutorial/*\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAs, el comando \u003ccode\u003egzip\u003c/code\u003e compact todos los archivos que estn dentro de\nla carpeta \u003ccode\u003etutorial/\u003c/code\u003e. Use el comando \u003ccode\u003els\u003c/code\u003e para observar la nueva\nextensin de los archivos.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePara descompactar el comando es:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Descompactar\ngzip -d tutorial/bssA_all.txt.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Descompactar\ngzip -d tutorial/bssA_all.txt.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eComandos tiles de Linux\u003c/h3\u003e\u003ca id=\"user-content-comandos-tiles-de-linux\" class=\"anchor\" aria-label=\"Permalink: Comandos tiles de Linux\" href=\"#comandos-tiles-de-linux\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"df    # Muestra el espacio disponible en el disco\nfree -g     # info de la memoria\nuname -a    # Muestra la informacin de la mquina\ndu -sh    # muestra el espacio en uso en el disco\ndu sh *     # mueestra el espacio usado en disco por archivos y/o directorios \ndu -s * | sort -nr    # Muestra el espacio usado en el disco por archivos y/o directorios ordenados por tamao\ntop     # Muestra el top de consumidores de memoria y CPU \nwho     # Muestra quien est logado en el sistema\nps    # Muestra los procesos ejecutados por el usuario\nps -e     # Muestra todos los procesos en ejecin en el sistema\nps -o %t -p \u0026lt;pid\u0026gt;     # Muestra cuanto tiempo lleva ejecutando un determinado proceso (pid)\nkill \u0026lt;pid\u0026gt;    # Mata el proceso\npaste \u0026lt;archivo1\u0026gt; \u0026lt;archivo2\u0026gt; \u0026gt; \u0026lt;archivo.saida\u0026gt;     # Junta lineas de archivos y separa por tabs (muy til para tablas)\ncmp \u0026lt;archivo1\u0026gt; \u0026lt;archivo2\u0026gt;     # Muestra las coincidencias de dos archivos\ndiff \u0026lt;archivo1\u0026gt; \u0026lt;archivo2\u0026gt;    # Muestra las diferencias entre dos archivos\ncsplit -f out fasta_batch \u0026quot;%^\u0026gt;%\u0026quot; \u0026quot;/^\u0026gt;/\u0026quot; \u0026quot;{*}\u0026quot;     # Divide un archivo fasta en varios archivos a cada '\u0026gt;' (cada nueva secuencia)\n\nsort -k 2,2 -k 3,3n archivo.in \u0026gt; archivo.out    # ordena la tabla, la columna 2 alfabeticamente y la columna 3 numericamente, -k para columna, -n para numerico\njoin -1 1 -2 1 \u0026lt;tabla1\u0026gt; \u0026lt;tabla2\u0026gt;    # Junta dos tablas con base a los nmeros especificados de las columnas. De la tabla 1 la columna 1 y de la tabla 2 la colunma 2. Se asume que las tablas estn ordenadas.\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003edf    # Muestra el espacio disponible en el disco\nfree -g     # info de la memoria\nuname -a    # Muestra la informacin de la mquina\ndu -sh    # muestra el espacio en uso en el disco\ndu sh *     # mueestra el espacio usado en disco por archivos y/o directorios \ndu -s * | sort -nr    # Muestra el espacio usado en el disco por archivos y/o directorios ordenados por tamao\ntop     # Muestra el top de consumidores de memoria y CPU \nwho     # Muestra quien est logado en el sistema\nps    # Muestra los procesos ejecutados por el usuario\nps -e     # Muestra todos los procesos en ejecin en el sistema\nps -o %t -p \u0026lt;pid\u0026gt;     # Muestra cuanto tiempo lleva ejecutando un determinado proceso (pid)\nkill \u0026lt;pid\u0026gt;    # Mata el proceso\npaste \u0026lt;archivo1\u0026gt; \u0026lt;archivo2\u0026gt; \u0026gt; \u0026lt;archivo.saida\u0026gt;     # Junta lineas de archivos y separa por tabs (muy til para tablas)\ncmp \u0026lt;archivo1\u0026gt; \u0026lt;archivo2\u0026gt;     # Muestra las coincidencias de dos archivos\ndiff \u0026lt;archivo1\u0026gt; \u0026lt;archivo2\u0026gt;    # Muestra las diferencias entre dos archivos\ncsplit -f out fasta_batch \"%^\u0026gt;%\" \"/^\u0026gt;/\" \"{*}\"     # Divide un archivo fasta en varios archivos a cada '\u0026gt;' (cada nueva secuencia)\n\nsort -k 2,2 -k 3,3n archivo.in \u0026gt; archivo.out    # ordena la tabla, la columna 2 alfabeticamente y la columna 3 numericamente, -k para columna, -n para numerico\njoin -1 1 -2 1 \u0026lt;tabla1\u0026gt; \u0026lt;tabla2\u0026gt;    # Junta dos tablas con base a los nmeros especificados de las columnas. De la tabla 1 la columna 1 y de la tabla 2 la colunma 2. Se asume que las tablas estn ordenadas.\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003escreen\u003c/code\u003e\u003c/h3\u003e\u003ca id=\"user-content-screen\" class=\"anchor\" aria-label=\"Permalink: screen\" href=\"#screen\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eScreen es una aplicacin desarrollada para Linux, que tiene como\nobjetivo la multiplexacin de terminales. O sea, l divide el terminal\nfsico en varias sesiones virtuales. Funciona as, s ud est trabajando\nen una sesin del terminal, usando screen, cuando apague su computador,\nla sesin continuar corriendo los procesos y ud podr volver a acceder\na ella.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePara iniciar una nueva sesin de screen, es solo digitar en la lnea de\ncomando \u003ccode\u003escreen\u003c/code\u003e. Aparecer un texto en la pantalla, puede apretar\n[enter]. De esa manera el sistema criar una neuva sesin virtual e el\nnombre de esa sesin ser asignado por el sistema. Sin embargo, s ud\nquiere puede darle un nombre a la sesin, para eso es necesario comenzar\ncon el siguiente comando:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"screen -S mysesson\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003escreen -S mysesson\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara salir de la sesin mantenga oprimido [Ctrl], en seguida oprima\n[a] seguido de [d]. Para volver a la sesin use los seguientes\ncomandos:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"screen -ls # lista las sesiones activas\nscreen -r nombredelasesion # entra en la sesin deseada\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003escreen -ls # lista las sesiones activas\nscreen -r nombredelasesion # entra en la sesin deseada\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara eliminar la sesin oprima [Ctrl + d]\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eScripts simples de uma linha de comando\u003c/h3\u003e\u003ca id=\"user-content-scripts-simples-de-uma-linha-de-comando\" class=\"anchor\" aria-label=\"Permalink: Scripts simples de uma linha de comando\" href=\"#scripts-simples-de-uma-linha-de-comando\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara renombras muchos archivos \u003cem\u003e.old\u003c/em\u003e a *.new. Para probar primero,\nsubtituya \u003ccode\u003edo mv\u003c/code\u003e por \u003ccode\u003edo echo mv\u003c/code\u003e.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for i in *.input; do mv $i ${i/\\.old/\\.new}; done\nfor i in *\\ *; do mv \u0026quot;$i\u0026quot; \u0026quot;${i// /_}\u0026quot;; done # Substituye espacios en nombres de archivos por underscores\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efor i in *.input; do mv $i ${i/\\.old/\\.new}; done\nfor i in *\\ *; do mv \"$i\" \"${i// /_}\"; done # Substituye espacios en nombres de archivos por underscores\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ccode\u003escp\u003c/code\u003e Secure Copy Between Machines\u003c/h3\u003e\u003ca id=\"user-content-scp-secure-copy-between-machines\" class=\"anchor\" aria-label=\"Permalink: scp Secure Copy Between Machines\" href=\"#scp-secure-copy-between-machines\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003escp\u003c/code\u003e es un comando que sirve para copiar elementos entre el servidor y\nsu computador y viceversa.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e \u003ccode\u003escp source target\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eDel servidor para su computador:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"scp user@ip.adress:camin/al/archivo.txt camino/en/su/pc\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003escp user@ip.adress:camin/al/archivo.txt camino/en/su/pc\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDel computador para el servidor:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"scp camino/en/su/pc/archivo.txt user@ip.adress:camino/donde/quiere/copiar/enel/servidor/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003escp camino/en/su/pc/archivo.txt user@ip.adress:camino/donde/quiere/copiar/enel/servidor/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSe ud quier copiar un directorio completo basta colocar el \u003cem\u003eflag\u003c/em\u003e \u003ccode\u003e-r\u003c/code\u003e\ndespus de \u003ccode\u003escp\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCompresin e Descompresin de archivos\u003c/h3\u003e\u003ca id=\"user-content-compresin-e-descompresin-de-archivos\" class=\"anchor\" aria-label=\"Permalink: Compresin e Descompresin de archivos\" href=\"#compresin-e-descompresin-de-archivos\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAdems de \u003ccode\u003egzip\u003c/code\u003e, existen otros tipos de compresin de archivos, tales\ncomo \u003ccode\u003e.tar\u003c/code\u003e y \u003ccode\u003e.zip\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eCompresin\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"tar -cvf archivo.tar midirectorio/ #comprime midirectorio, y el nombre del archivo compreso ser archivo.tar\nzip -r midrectorio.zip midirectorio/ #comprime el directorio midirectorio/ en un archivo llamado midirectorio.zip\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etar -cvf archivo.tar midirectorio/ #comprime midirectorio, y el nombre del archivo compreso ser archivo.tar\nzip -r midrectorio.zip midirectorio/ #comprime el directorio midirectorio/ en un archivo llamado midirectorio.zip\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eVisualizar\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"tar -tvf archivo.tar\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etar -tvf archivo.tar\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eExtrair\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"tar -xvf archivo.tar\nunzip midirectorio.zip\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etar -xvf archivo.tar\nunzip midirectorio.zip\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eI. Gestin de Herramientas Bioinformticas\u003c/h1\u003e\u003ca id=\"user-content-i-gestin-de-herramientas-bioinformticas\" class=\"anchor\" aria-label=\"Permalink: I. Gestin de Herramientas Bioinformticas\" href=\"#i-gestin-de-herramientas-bioinformticas\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAnaconda\u003c/h2\u003e\u003ca id=\"user-content-anaconda\" class=\"anchor\" aria-label=\"Permalink: Anaconda\" href=\"#anaconda\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEs recomendable instalar Anaconda, pues es la forma ms fcil para\ninstalar las herramientas bioinformticas. Anaconda es una distribucin\nlibre y abierta de los lenguajes \u003cem\u003ePython\u003c/em\u003e y \u003cem\u003eR\u003c/em\u003e, utilizada en ciencia de\ndatos y bioinformtica. Las diferentes versiones de los programas se\nadministran mediante un sistema de gestin llamado \u003cem\u003econda\u003c/em\u003e, el cual hace\nbastante sencillo instalar, correr y actualizar programas.\n\u003ca href=\"https://conda.io/projects/conda/en/latest/user-guide/install/index.html\" rel=\"nofollow\"\u003eAqui\u003c/a\u003e\nse encuentran las instrucciones para la instalacin de Anaconda. Despus\nde instalado \u003cem\u003eAnaconda\u003c/em\u003e y su gestor \u003cem\u003eConda\u003c/em\u003e, podran ser creados\n\u003cem\u003eambientes virtuales\u003c/em\u003e para la instalacin de las diferentes herramientas\nbioinformticas que sern usadas.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAlgunos comandos de \u003ccode\u003econda\u003c/code\u003e\u003c/h2\u003e\u003ca id=\"user-content-algunos-comandos-de-conda\" class=\"anchor\" aria-label=\"Permalink: Algunos comandos de conda\" href=\"#algunos-comandos-de-conda\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"conda create -n miambiente # Crea un ambiente llamado miambiente\nconda activate miambiente # Activa el ambiente mi ambiente\nconda env list # Lista todos los ambientes creados\ncond list # Estando dentro de un ambiente, lista las herramentas instaladas en ese ambiente\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003econda create -n miambiente # Crea un ambiente llamado miambiente\nconda activate miambiente # Activa el ambiente mi ambiente\nconda env list # Lista todos los ambientes creados\ncond list # Estando dentro de un ambiente, lista las herramentas instaladas en ese ambiente\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara instalacin de cada herramienta, visite la pgina\n\u003ca href=\"https://anaconda.org/\" rel=\"nofollow\"\u003ehttps://anaconda.org/\u003c/a\u003e y en la caja de busqueda procure por la\nherramienta que desea, y econtrat el comonado para instalacin.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNota\u003c/strong\u003e: Las herramientas usadas en este tutorial sern instaladas a\nmedida que van apareciendo dentro del pipeline.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eII. Adquisicin y organizacin de los datos\u003c/h1\u003e\u003ca id=\"user-content-ii-adquisicin-y-organizacin-de-los-datos\" class=\"anchor\" aria-label=\"Permalink: II. Adquisicin y organizacin de los datos\" href=\"#ii-adquisicin-y-organizacin-de-los-datos\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.1. Organizacin de directorios\u003c/h2\u003e\u003ca id=\"user-content-21-organizacin-de-directorios\" class=\"anchor\" aria-label=\"Permalink: 2.1. Organizacin de directorios\" href=\"#21-organizacin-de-directorios\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara mantener el orden durante la ejecucin del \u003cem\u003epipeline\u003c/em\u003e es\nrecomendado crear directorios secuenciales a medida que son realizados\nlos diferentes pasos durante el proceso.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUsando el comando \u003ccode\u003emkdir\u003c/code\u003e, cree un directorio base para todo el proceso.\nAsigne el nombre de transcriptomica\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"mkdir transcriptomica\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emkdir transcriptomica\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEntre al nuevo directorio usando el comando \u003ccode\u003ecd\u003c/code\u003e (\u003cem\u003echange directory\u003c/em\u003e)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd transcriptomica/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd transcriptomica/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCree un nuevo directorio para almacenar las secuencias brutas.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTip:\u003c/strong\u003e Debido a que la mayoria de las etapas del workflow son\nsecuenciales, es recomendable nombrar los directorios comenzando con\nun nmero y as mantener la organizacin.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"mkdir 00.RawData\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emkdir 00.RawData\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.2 Adquisicin de datos\u003c/h2\u003e\u003ca id=\"user-content-22-adquisicin-de-datos\" class=\"anchor\" aria-label=\"Permalink: 2.2 Adquisicin de datos\" href=\"#22-adquisicin-de-datos\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.2.1. SRA Toolkit\u003c/h3\u003e\u003ca id=\"user-content-221-sra-toolkit\" class=\"anchor\" aria-label=\"Permalink: 2.2.1. SRA Toolkit\" href=\"#221-sra-toolkit\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara la adquisicin de datos, ser instalado el programa \u003ca href=\"https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit\"\u003eSRA\nToolkit\u003c/a\u003e,\nel cual nos permitir descargar las secuencias directamente del NCBI en\nel servidor.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eDescargue el programa en el siguiente\n\u003ca href=\"https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit\"\u003elink\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSeleccione el archivo segn su sistema operativo. En este tutorial ser\ninstalada la version para \u003ca href=\"https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/3.0.0/sratoolkit.3.0.0-ubuntu64.tar.gz\" rel=\"nofollow\"\u003eUbuntu Linux\n64bits\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUsando el comando \u003ccode\u003ewget\u003c/code\u003e, descargue el archivo de la wed directamente en\nel servidor\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Salga de la carpeta transcriptomica\ncd ..\n\n# Descargue el programa en la carpeta /home/user/\nwget --output-document sratoolkit.tar.gz https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Salga de la carpeta transcriptomica\ncd ..\n\n# Descargue el programa en la carpeta /home/user/\nwget --output-document sratoolkit.tar.gz https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSer descargada un archivo compactado (\u003ccode\u003esratoolkit.tar.gz\u003c/code\u003e). Use el\ncomando tar para descompactar\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"tar -vxzf sratoolkit.tar.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etar -vxzf sratoolkit.tar.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl archivo ser descompactado en la carpeta \u003ccode\u003esratoolkit.3.0.0-ubuntu64/\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eConfigure el camino al programa\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"export PATH=$PATH:$PWD/sratoolkit.3.0.0-ubuntu64/bin\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eexport PATH=$PATH:$PWD/sratoolkit.3.0.0-ubuntu64/bin\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eVerifique que la configuracin est correcta\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"which fastq-dump\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ewhich fastq-dump\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDeber aparecer el camino al directorio que contiene el programa:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/user/sratoolkit.3.0.0-ubuntu64/bin/fastq-dump\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/user/sratoolkit.3.0.0-ubuntu64/bin/fastq-dump\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAntes de entrar en el men de configuracin, seleccione un directorio\nvacio para usarlo como el repositorio de las secuencias a descargar.\nPara esto siga la siguiente secuencia de comandos:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Cree un directorio llamado secuencias-ncbi\nmkdir secuencias-ncbi\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Cree un directorio llamado secuencias-ncbi\nmkdir secuencias-ncbi\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIngrese al men de configuracin con \u003ccode\u003evdb-config -i\u003c/code\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eVerifique que la opcin \u003cem\u003eRemote Access\u003c/em\u003e est habilitada\u003c/li\u003e\n\u003cli\u003eIngrese a la pestaa \u003cem\u003eCache\u003c/em\u003e y verifique que la opcin \u003cem\u003elocal\nfile-caching\u003c/em\u003e est habilitada.\u003c/li\u003e\n\u003cli\u003eEn la opcin *Location of user-repository, busque el directorio\ncreado anteriormente \u003ccode\u003esecuencias-ncbi/\u003c/code\u003e\u003c/li\u003e\n\u003cli\u003eEn la pestaa \u003cem\u003eAWS\u003c/em\u003e, habilite la opcin \u003cem\u003ereport cloud instance\nidentity\u003c/em\u003e\u003c/li\u003e\n\u003cli\u003eSalve las configuraciones y salga del men con x.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.2.2. Descargando las secuencias\u003c/h3\u003e\u003ca id=\"user-content-222-descargando-las-secuencias\" class=\"anchor\" aria-label=\"Permalink: 2.2.2. Descargando las secuencias\" href=\"#222-descargando-las-secuencias\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara la ejecucin de este \u003cem\u003epipeline\u003c/em\u003e sern usadas las secuencias de\nRNA-seq del artculo en estudio y secuencias del genoma de \u003cem\u003eE.coli\u003c/em\u003e\nATCC8739, el cual ser usado como referencia para el ensamble de los\ntranscriptomas.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSecuencias RNA-seq\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePara las secuencias de RNA-seq, usando el comando \u003ccode\u003enano\u003c/code\u003e, cree un\narchivo llamado \u003ccode\u003eSRR-seqs-RNA.txt\u003c/code\u003e en donde sern listados los cdigos\nde acceso (SRR number) de los \u003cem\u003edatasets\u003c/em\u003e. Siga la secuencia de cdigos\npresentados a continuacin:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# entre en la carpeta secuencias-ncbi/\ncd secuencias-ncbi/\n\n# Cree el archivo con nano\nnano SRR-seqs-RNA.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# entre en la carpeta secuencias-ncbi/\ncd secuencias-ncbi/\n\n# Cree el archivo con nano\nnano SRR-seqs-RNA.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSer abierto el programa nano para editar el archivo \u003ccode\u003eSRR-seqs-RNA.txt\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLos cdigos de accesso son los siguientes:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSRR12922100 SRR12922099 SRR12922098 SRR12922089 SRR12922097 SRR12922096\nSRR12922095 SRR12922094 SRR12922093 SRR12922092 SRR12922091 SRR12922090\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCopie y pegue dentro del archivo en el terminal. Para cerrar [Ctrl +\nX], confirme que desea guardar los cambios con S.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eDebido a que es un proceso que va a demorar, es recomendable hacerlo\nusando el programa screen.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"screen -S sra\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003escreen -S sra\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl siguiente comando le permitir descargar los archivos \u003ccode\u003e.sra\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"prefetch --option-file SRR-seqs-RNA.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eprefetch --option-file SRR-seqs-RNA.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eComenzar a ser descargadas cada una de las muestras segn los cdigos\nde acceso. Cierre la sesin de screen con [Ctrl + A + D]. Cuando\nquiera volver a la sesin use el comando \u003ccode\u003escreen -r sra\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAl terminar el proceso, en la carpeta \u003ccode\u003esecuencias-ncbi\u003c/code\u003e, se encontrarn\nuna serie de directorios, la carpeta \u003ccode\u003esra\u003c/code\u003e contiene los archivos \u003ccode\u003e.sra\u003c/code\u003e.\nUse los comandos \u003ccode\u003ecd\u003c/code\u003e para cambiar de directorio y \u003ccode\u003ell\u003c/code\u003e para listar el\ncontenido.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEl siguiente paso es transformar los archivos \u003ccode\u003e.sra\u003c/code\u003e para \u003ccode\u003e.fastq\u003c/code\u003e.\nUsando la funcin \u003ccode\u003efor\u003c/code\u003e, puede automatizar el proceso para que todas los\narchivos sean transformados secuencialmente.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for i in ./*.sra\ndo\nfasterq-dump --split-files $i\ndone\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efor i in ./*.sra\ndo\nfasterq-dump --split-files $i\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUse el comando \u003ccode\u003ell\u003c/code\u003e para revisar el contenido de la carpeta. Mueva los\narchivos \u003ccode\u003e.fastq\u003c/code\u003e para la carpeta creada para las secuencias brutas\n\u003ccode\u003etranscriptomica/00.RawData\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"mv *.fastq ../../transcriptomica/00.RawData/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emv *.fastq ../../transcriptomica/00.RawData/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eConfirme con \u003ccode\u003ell\u003c/code\u003e que efectivamente los archivos se encuentran en la\nnueva ubicacin.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eEnsamble Genoma \u003cem\u003eE.coli\u003c/em\u003e ATCC8739\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEste ensamble se encuentra en la pagina de la coleccion\n\u003ca href=\"https://genomes.atcc.org/\" rel=\"nofollow\"\u003eATCC\u003c/a\u003e. Cree una cuenta de usuario, use el\nbuscador de la pgina para encontrar el genoma y descargue el ensamble.\nUse la interface grfica de MobaXterm para subir el archivo dentro de la\ncarpeta \u003ccode\u003etranscriptomica/\u003c/code\u003e. Use el comando \u003ccode\u003emv\u003c/code\u003e para colocar el ensamble\ndel genoma dentro de la carpeta \u003ccode\u003e00.RawData/\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSecuencias Genoma \u003cem\u003eEscherichia coli\u003c/em\u003e ATCC 25922\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCon el fin de aprender y practicar el montaje de genomas, sern\ndescargadas las secuencias brutas de la secuenciacin del genoma\ncompleto de la \u003cem\u003eEscherichia coli\u003c/em\u003e ATCC 25922.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNota:\u003c/strong\u003e Este genoma no ser usado para el proceso de ensamble del\ntranscriptoma.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEl cdigo de acceso a estas secuencias es SRR2889880. Use el programa\nSRA Tool kit para descargar este conjunto de datos.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"prefetch SRR2889880\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eprefetch SRR2889880\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eConvierta el archivo \u003ccode\u003e.sra\u003c/code\u003e a archivos \u003ccode\u003e.fastq\u003c/code\u003e.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"fasterq-dump --split-files SRR2889880\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efasterq-dump --split-files SRR2889880\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUse el comando \u003ccode\u003emv\u003c/code\u003e para mover los archivos \u003ccode\u003e.fastq\u003c/code\u003e del genoma de \u003cem\u003eE.\ncoli\u003c/em\u003e desde la carpeta \u003ccode\u003esecuencias-ncbi/sra/\u003c/code\u003e para la carpeta\n\u003ccode\u003e/home/user/transcriptomica/01.RefGenome/\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.3. Organizacin de los \u003cem\u003edatasets\u003c/em\u003e\u003c/h2\u003e\u003ca id=\"user-content-23-organizacin-de-los-datasets\" class=\"anchor\" aria-label=\"Permalink: 2.3. Organizacin de los datasets\" href=\"#23-organizacin-de-los-datasets\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDebido a que los nombres de los archivos son los cdigos de acceso SRR,\nes recomendable cambiarlos por nombres ms fciles de entender y\nreconocer cada archivo. Use la informacin en la siguente tabla y el\ncomando \u003ccode\u003emv\u003c/code\u003e para cambiar los nombres. Para facilitar primero use el\ncomando \u003ccode\u003ecd\u003c/code\u003e para entrar en el directorio \u003ccode\u003e00.RawData/\u003c/code\u003e.\u003c/p\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003eCdigo SRR\u003c/th\u003e\n\u003cth\u003eMuestra\u003c/th\u003e\n\u003cth\u003eNombre Archivo\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922100\u003c/td\u003e\n\u003ctd\u003e1/2MIC camphor essential oil (R3)\u003c/td\u003e\n\u003ctd\u003eoil1-2_R3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922098\u003c/td\u003e\n\u003ctd\u003e1/2MIC camphor essential oil (R2)\u003c/td\u003e\n\u003ctd\u003eoil1-2_R2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922099\u003c/td\u003e\n\u003ctd\u003e1/2MIC camphor essential oil (R1)\u003c/td\u003e\n\u003ctd\u003eoil1-2_R1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922089\u003c/td\u003e\n\u003ctd\u003eacetone\u003c/td\u003e\n\u003ctd\u003eacetone\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922097\u003c/td\u003e\n\u003ctd\u003e1/4MIC camphor essential oil (R3)\u003c/td\u003e\n\u003ctd\u003eoil1-4_R3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922096\u003c/td\u003e\n\u003ctd\u003e1/4MIC camphor essential oil (R2)\u003c/td\u003e\n\u003ctd\u003eoil1-4_R2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922095\u003c/td\u003e\n\u003ctd\u003e1/4MIC camphor essential oil (R1)\u003c/td\u003e\n\u003ctd\u003eoil1-4_R1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922094\u003c/td\u003e\n\u003ctd\u003e1/8MIC camphor essential oil (R3)\u003c/td\u003e\n\u003ctd\u003eoil1-8_R3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922093\u003c/td\u003e\n\u003ctd\u003e1/8MIC camphor essential oil (R2)\u003c/td\u003e\n\u003ctd\u003eoil1-8_R2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922092\u003c/td\u003e\n\u003ctd\u003e1/8MIC camphor essential oil (R1)\u003c/td\u003e\n\u003ctd\u003eoil1-8_R1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922091\u003c/td\u003e\n\u003ctd\u003ewater (R2)\u003c/td\u003e\n\u003ctd\u003ewater_R2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR12922090\u003c/td\u003e\n\u003ctd\u003ewater (R1)\u003c/td\u003e\n\u003ctd\u003ewater_R1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSRR2889880\u003c/td\u003e\n\u003ctd\u003egenoma \u003cem\u003eE. coli\u003c/em\u003e ATCC 25922\u003c/td\u003e\n\u003ctd\u003eecoli\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eNota:\u003c/em\u003e Recuerde que debe tener dos archivos con el mismo nombre que\ncomponen una muestra por ser una secuenciacin \u003cem\u003epaired-end\u003c/em\u003e, con\nexcepcin del genoma de \u003cem\u003eE. coli\u003c/em\u003e ATCC 25922 que es \u003cem\u003esingle-end\u003c/em\u003e. Por\nejemplo: \u003ccode\u003eoil1-2_R3_1.fq\u003c/code\u003e y \u003ccode\u003eoil1-2_R3_2.fq\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTip:\u003c/strong\u003e Use la tecla TAB para facilitar y evitar errores a la hora de\ndigitar el nombre de los archivos.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIII. Control de calidad\u003c/h1\u003e\u003ca id=\"user-content-iii-control-de-calidad\" class=\"anchor\" aria-label=\"Permalink: III. Control de calidad\" href=\"#iii-control-de-calidad\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.1. Control de Calidad\u003c/h2\u003e\u003ca id=\"user-content-31-control-de-calidad\" class=\"anchor\" aria-label=\"Permalink: 3.1. Control de Calidad\" href=\"#31-control-de-calidad\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.1.1. Evaluacin de la calidad\u003c/h3\u003e\u003ca id=\"user-content-311-evaluacin-de-la-calidad\" class=\"anchor\" aria-label=\"Permalink: 3.1.1. Evaluacin de la calidad\" href=\"#311-evaluacin-de-la-calidad\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara la evaluacin de la calidad ser usado el programa\n\u003ca href=\"http://www.bioinformatics.babraham.ac.uk/projects/fastqc/\" rel=\"nofollow\"\u003eFastQC\u003c/a\u003e que\nes una herramienta que permite observar graficamente la calidad de las\nsecuencias de Illumina.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInstalacin\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLas instrucciones para instalacin usando conda se encuentran\n\u003ca href=\"https://anaconda.org/bioconda/fastqc\" rel=\"nofollow\"\u003eaqui\u003c/a\u003e. Sin embargo aqui en este\ntutorial tambin sern presentadas\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eComo ya fue explicado anteriorimente, con conda es posible crear\nambientes virutuales para instalar las herramientas bioinformticas. El\nprimer ambiente que ser creado se llamar \u003cstrong\u003equality\u003c/strong\u003e, donde se\ninstalaran los programas relacionados con este proceso.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"conda create -n quality\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003econda create -n quality\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDurante el proceso, el sistema preguntar s desea proceder con la\ncreacin del ambiente, con las opciones y/n (si o no). Escriba \u003ccode\u003ey\u003c/code\u003e y\ndespus de eso el ambiente virtual estar creado.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePara instalar las herramientas dentro del ambiente anteriormente creado,\nes necesario activarlo\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"conda activate quality\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003econda activate quality\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl ambiente estar activo cuando el nombre de ste se encuentra en el\ncomienzo de la linea de comando, as: \u003ccode\u003e(quality) user@server:~/$\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePosteriormente se procede a la instalacin del programa:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"conda install -c bioconda fastqc\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003econda install -c bioconda fastqc\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDe nuevo ser cuestionado si desea continuar con el proceso o no.\nEscriba \u003ccode\u003ey\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eUso\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLa primera etapa del proceso es la evaluacin de la calidad de las\nsecuencias cortas (Illumina) usando \u003cem\u003eFastQC\u003c/em\u003e, con el objetivo de\ndeterminar s es necesario trimar o filtrar las secuencias de baja\ncalidad en los prximos pasos.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esta etapa es para identificar principalmente las secuencias \u003cem\u003eoutlier\u003c/em\u003e\ncon baja calidad (\u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"5adc994830ac8c406144d27b37b0257b\"\u003e$Q\u0026amp;lt; 20$\u003c/math-renderer\u003e).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eActive el ambiente \u003ccode\u003equality\u003c/code\u003e:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"conda activate quality\n\n## Confirme que est en el directorio raz\npwd\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003econda activate quality\n\n## Confirme que est en el directorio raz\npwd\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDebe estar em \u003ccode\u003e~/transcriptomica/\u003c/code\u003e. Si ese no es el resultado del\ncomando \u003ccode\u003epwd\u003c/code\u003e, use el comando \u003ccode\u003ecd\u003c/code\u003e para llegar en el directorio base.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTip:\u003c/strong\u003e \u003cstrong\u003eTODAS\u003c/strong\u003e las herramientas bioinformticas tienen un manual\nde ayuda que puede ser accesado en la linea de comando, usando el flag\n\u003ccode\u003e-h\u003c/code\u003e o \u003ccode\u003e--help\u003c/code\u003e. \u003cstrong\u003eANTES de ejecutar una hierramenta siempre lea el\nmanual\u003c/strong\u003e. Este comando lista todos los argumentos disponibles para la\nherramienta y explica como deben/pueden ser usados.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAtencin:\u003c/strong\u003e Para el uso correcto y seguro del servidor verifique ele\nnmero de ncleos disponibles para el usuario en el momento del\nanlisis. \u003cstrong\u003eNUNCA\u003c/strong\u003e trabaje con el total de los ncleos de la mquina.\nEjecute \u003cstrong\u003eFastQC\u003c/strong\u003e:\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Cree un directorio para salvar el output de FastQC\nmkdir 01.FastqcReports\n\n## Ejecute usando 10 threads\nfastqc -t 10 00.RawData/* -o 01.FastqcReports/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Cree un directorio para salvar el output de FastQC\nmkdir 01.FastqcReports\n\n## Ejecute usando 10 threads\nfastqc -t 10 00.RawData/* -o 01.FastqcReports/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"fastqc -t \u0026lt;num ncleos\u0026gt; \u0026lt;input_directory\u0026gt; -o \u0026lt;output_directory\u0026gt;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efastqc -t \u0026lt;num ncleos\u0026gt; \u0026lt;input_directory\u0026gt; -o \u0026lt;output_directory\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl comando \u003ccode\u003efastqc\u003c/code\u003e tiene varias opciones o parametros, entre ellas,\nescoger el nmero de ncleos de la mquina para correr el anlisis, para\neste caso \u003ccode\u003e-t 10\u003c/code\u003e. El input es el directorio que contiene las secuencias\nde illumina \u003ccode\u003e00.RawData/*\u003c/code\u003e, el \u003ccode\u003e*\u003c/code\u003e indica al sistema que puede analizar\ntodos los archivos que estn dentro de ese directorio. El output,\nindicado por el parametro \u003ccode\u003e-o\u003c/code\u003e, es el directorio donde se desea que sean\nguardados los resultados del anlisis. A continuacin se encuentra una\nexplicacin detallada de cada output generado.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutputs\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eReportes html \u003ccode\u003eoil1-2_R3.html\u003c/code\u003e: Aqui es posible ver toda informacin\nde calidad graficamente.\u003c/li\u003e\n\u003cli\u003eZip files \u003ccode\u003eoil1-2_R3.zip\u003c/code\u003e: Aqui se encuentran cada uno de los\ngrficos de manera separada. *IGNORE**\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eDescargue los archivos \u003ccode\u003ehtml\u003c/code\u003e y explore en su \u003cem\u003eweb browser\u003c/em\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eRevise cada uno de los grficos y datos que son generados en estos\nreportes, los cuales permiten determinar los parametros para el siguente\npaso, en el cual se filtraran y eliminaran las secuencias y bases con\nbaja calidad. En este\n\u003ca href=\"https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/\" rel=\"nofollow\"\u003elink\u003c/a\u003e\nse encuentra la explicacin de cada una de las partes del reporte y como\ninterpretar los resultados.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLos reportes mostraron que las secuencias de RNA-seq en trminos\ngenerales son de alta calidad. Los grficos que mostraron errores,\npueden ser ignorados, puesto que es normal que esto pase con librerias\nde RNA-seq. Por esta razn estas secuencias no sern\ndepuradas/filtradas. Sin embargo las secuencias \u003cem\u003esingle-end\u003c/em\u003e del genoma\nde \u003cem\u003eE. coli\u003c/em\u003e, presentan varios problemas de calidad. El grfico \u003cem\u003ePer\nbase sequence quality\u003c/em\u003e muestra que algunas bases finales de las\nsecuencias tienen una calidad inferior a 20 (\u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"5adc994830ac8c406144d27b37b0257b\"\u003e$Q\u0026amp;lt;20$\u003c/math-renderer\u003e). El grfico \u003cem\u003ePer\nbase sequence content\u003c/em\u003e, muestra que las primeras 18 bases presentan\ndiferentes proporcin de GC y AT. El grfico \u003cem\u003ePer sequence GC content\u003c/em\u003e\nmuestra que hay una leve desviacin entre la distribucin terica del\n%GC y la distribucin real de %GC por \u003cem\u003ereads\u003c/em\u003e. Analizando cada uno de\nlos grficos son decididos los parametros para la etapa de depuracin.\nPara este caso es necesario filtrar por calidad, cortar las primeras\nbases y filtrar por tamao.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.1.2. Depuracin/\u003cem\u003eTrimming\u003c/em\u003e\u003c/h3\u003e\u003ca id=\"user-content-312-depuracintrimming\" class=\"anchor\" aria-label=\"Permalink: 3.1.2. Depuracin/Trimming\" href=\"#312-depuracintrimming\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSegn fue evaluado en el control de calidad, ser necesario filtrar\nalgunas lecturas del genoma de \u003cem\u003eE. coli\u003c/em\u003e. Las secuencias de las\nlibrerias de RNA-seq no pasarn por el proceso de depuracin. Para esta\netapa ser usado o programa \u003ca href=\"http://www.usadellab.org/cms/?page=trimmomatic\" rel=\"nofollow\"\u003eTrimmomatic\nv0.39\u003c/a\u003e que permite\nfiltrar (remover) lecturas o \u003cem\u003ereads\u003c/em\u003e cortas de baja calidad.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTrimmomatic tiene vrios parametros que pueden ser considerados para\nfiltrar lecturas con baja calidad. Aqui usaremos algunos. Si quiere\nsaber que otros parametros y como funciona cada uno de ellos, consulte\nel\n\u003ca href=\"http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf\" rel=\"nofollow\"\u003emanual\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInstalacin\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eComo se trata de una herramienta que participa dentro del proceso de\ncontrol de calidad, ser instalada dentro del ambiente virtual\n\u003cstrong\u003equality\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Si no est activado el ambiente\nconda activate quality\n\n# Instale Trimmomatic\nconda install -c bioconda trimmomatic\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Si no est activado el ambiente\nconda activate quality\n\n# Instale Trimmomatic\nconda install -c bioconda trimmomatic\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eUso\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePara los datos aqui analizados se usar la siguiente linea de comando:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Activa el ambiente quality\nconda activate quality\n\n# Crie un directorio para salvar las lecturas limpias\nmkdir 02.CleanData\n\n# Ejecute Trimmomatic\ntrimmomatic SE -threads 10 00.RawData/ecoli.fq 02.CleanData/ecoli.fq LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 HEADCROP:18 MINLEN:80 ILLUMINACLIP:/home/drm/anaconda3/pkgs/trimmomatic-0.36-6/share/trimmomatic-0.36-6/adapters/NexteraPE-PE.fa:2:30:10\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Activa el ambiente quality\nconda activate quality\n\n# Crie un directorio para salvar las lecturas limpias\nmkdir 02.CleanData\n\n# Ejecute Trimmomatic\ntrimmomatic SE -threads 10 00.RawData/ecoli.fq 02.CleanData/ecoli.fq LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 HEADCROP:18 MINLEN:80 ILLUMINACLIP:/home/drm/anaconda3/pkgs/trimmomatic-0.36-6/share/trimmomatic-0.36-6/adapters/NexteraPE-PE.fa:2:30:10\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e \u003ccode\u003etrimmomatic SE -threads input output [opciones]\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEl comando anterior tiene varias partes. Primero, el nombre del comando\nes \u003ccode\u003etrimmomatic\u003c/code\u003e, a continuacin la opcin \u003ccode\u003eSE\u003c/code\u003e indica para el programa\nque las secuencias que irn a ser analizadas son de tipo \u003cem\u003esingle end\u003c/em\u003e,\npara \u003cem\u003epaired-end\u003c/em\u003e, use \u003ccode\u003ePE\u003c/code\u003e. Despus se encuentra el input, si se\ntratara de secuencias \u003cem\u003epaired-end\u003c/em\u003e tendria dos inputs, forward (pair1) y\nreverse (pair2). Despus el output. En el caso de \u003cem\u003epaired-end\u003c/em\u003e sern\nvarios outputs, siendo primero las secuencias forward pareadas (limpias)\ny no pareadas y despus las secuencias reverse. Por ltimo se encuentran\nlos parametros de depuracin. Para este caso usamos los parametros\n\u003ccode\u003eSLIDINGWINDOW\u003c/code\u003e, \u003ccode\u003eLEADING\u003c/code\u003e, \u003ccode\u003eTRAILING\u003c/code\u003e, \u003ccode\u003eHEADCROP\u003c/code\u003e, \u003ccode\u003eMINLEN\u003c/code\u003e y\n\u003ccode\u003eILLUMINACLIP\u003c/code\u003e.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSLIDINGWINDOW: genera una ventana deslizante, que en este caso va de\n4 en 4 bases, clcula el promedio del \u003cem\u003ePhred Score\u003c/em\u003e y si est por\ndebajo de 15 esas bases son cortadas.\u003c/li\u003e\n\u003cli\u003eLEADING: corta bases del comienzo de la lectura si estn por debajo\nde \u003cem\u003ethreshold\u003c/em\u003e de calidad\u003c/li\u003e\n\u003cli\u003eTRAILING: corta bases del final de la lectura si estn por debajo de\n\u003cem\u003ethreshold\u003c/em\u003e de calidad.\u003c/li\u003e\n\u003cli\u003eHEADCROP: corta bases del inicio de la lectura\u003c/li\u003e\n\u003cli\u003eMINLEN: Secuencias con menor tamao del establecido, son descartadas\u003c/li\u003e\n\u003cli\u003eILLUMINACLIP: Elimina bases que corresponden con la secuencia de\nadaptadores usados en la plataforma Illumina.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eDespus de correr Trimmomatic es necesario evaluar la calidad de las\nsecuencias generadas (limpias) usando nuevamente FastQC.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"fastqc -t 10 02.CleanData/* -o 01.FastqcReports/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efastqc -t 10 02.CleanData/* -o 01.FastqcReports/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDescargue el reporte \u003ccode\u003e01.FastqcReports/ecoli.html\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eObserve que ahora todas las bases en ambos archivos tienen \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"5adc994830ac8c406144d27b37b0257b\"\u003e$Q\u0026amp;gt;30$\u003c/math-renderer\u003e.\nDespus del proceso de filtrado sobrevivieron \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"5adc994830ac8c406144d27b37b0257b\"\u003e$60'258.702$\u003c/math-renderer\u003e \u003cem\u003ereads\u003c/em\u003e, es\ndecir \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"5adc994830ac8c406144d27b37b0257b\"\u003e$85.4%$\u003c/math-renderer\u003e de las secuencias iniciales (\u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"5adc994830ac8c406144d27b37b0257b\"\u003e$70'560.424$\u003c/math-renderer\u003e).\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3.1.3. Cobertura\u003c/h3\u003e\u003ca id=\"user-content-313-cobertura\" class=\"anchor\" aria-label=\"Permalink: 3.1.3. Cobertura\" href=\"#313-cobertura\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCuando se trabaja con secuenciacin de genomas, el siguiente paso\ndespus de conocer la cantidad de secuencias obtenidas y remover las\nsecuencias con baja calidad, es calcular la cobertura o profundidad de\nla secuenciacin para ese genoma. El clculo de la cobertura est dado\npor la siguiente ecuacin:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"5adc994830ac8c406144d27b37b0257b\"\u003e$$C = (L*N)/G$$\u003c/math-renderer\u003e Donde: L es igual al tamao de las lecturas, N es igual\nal nmero de lecturas, y G es igual al tamao aproximado del genoma.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs, entonces para el genoma del presente tutorial, con el\nsecuenciamiento Illumina tenemos:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cmath-renderer class=\"js-display-math\" style=\"display: block\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"5adc994830ac8c406144d27b37b0257b\"\u003e$$C = ((83bp) * 60258702)/5209985bp$$\u003c/math-renderer\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#\u0026gt; [1] \u0026quot;La cobertura es de 960X\u0026quot;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#\u0026gt; [1] \"La cobertura es de 960X\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIV. Genmica\u003c/h1\u003e\u003ca id=\"user-content-iv-genmica\" class=\"anchor\" aria-label=\"Permalink: IV. Genmica\" href=\"#iv-genmica\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.1 Ensamblaje \u003cem\u003ede novo\u003c/em\u003e\u003c/h2\u003e\u003ca id=\"user-content-41-ensamblaje-de-novo\" class=\"anchor\" aria-label=\"Permalink: 4.1 Ensamblaje de novo\" href=\"#41-ensamblaje-de-novo\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUn ensamblaje es el proceso de alineamiento de secuencias cortas con el\nobjetivo de recuperar una secuencia mayor. En el caso del ensamblaje \u003cem\u003ede\nnovo\u003c/em\u003e son usados algortmos con base en \u003cem\u003ek-mers\u003c/em\u003e, que son subsecuencias\ncom tamao definido por el usuario (i.e.21-mers).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/ablab/spades\"\u003eSpades v3.15.3\u003c/a\u003e es uno de los\nensambladores de genomas, ms conocido y con mejores resultados, y puede\nser usado tanto para lecturas cortas como largas. Lea atentamente el\n\u003ca href=\"http://cab.spbu.ru/files/release3.15.2/manual.html\" rel=\"nofollow\"\u003emanual\u003c/a\u003e, ya que\neste programa tiene muchas opciones diferentes. Spades usa el algoritmo\ndel \u003cem\u003eGrafo de Bruijn\u003c/em\u003e para el montaje de las secuencias.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInstalacin\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSiga las siguientes instrucciones para la instalacin de \u003cstrong\u003eSpades\u003c/strong\u003e\ndentro de ambiente virtual llamado \u003cem\u003eassembly\u003c/em\u003e.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Cree el ambiente assembly\nconda create -n assembly\n\n# Active el ambiente virtual\nconda activate assembly\n\n# Instale Spades\nconda install -c bioconda spades\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Cree el ambiente assembly\nconda create -n assembly\n\n# Active el ambiente virtual\nconda activate assembly\n\n# Instale Spades\nconda install -c bioconda spades\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eUso\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEl ensamble ser realizado ejecutando el comando:\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTip:\u003c/strong\u003e El programa nohup permite ejecutar tareas en segundo plano,\ncon el objetivo de mantener la ejecucin del comando an con perdida\nde conexin\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"nohup spades.py --careful -s 02.CleanData/ecoli.fq -k 21,33,55,77,99,111,127 -o 03.Assembly/01.Careful -t 40\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003enohup spades.py --careful -s 02.CleanData/ecoli.fq -k 21,33,55,77,99,111,127 -o 03.Assembly/01.Careful -t 40\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxe\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003ePaired-end\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"spades.py -1 \u0026lt;pair1\u0026gt; -2 \u0026lt;pair2\u0026gt; -k \u0026lt;kmers list\u0026gt; -o \u0026lt;output_directory\u0026gt; -t \u0026lt;num_nucleos\u0026gt;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003espades.py -1 \u0026lt;pair1\u0026gt; -2 \u0026lt;pair2\u0026gt; -k \u0026lt;kmers list\u0026gt; -o \u0026lt;output_directory\u0026gt; -t \u0026lt;num_nucleos\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cem\u003eSingle-end\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"spades.py -s \u0026lt;single-end-reads\u0026gt; -k \u0026lt;kmers list\u0026gt; -o \u0026lt;output_directory\u0026gt; -t \u0026lt;num_nucleos\u0026gt;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003espades.py -s \u0026lt;single-end-reads\u0026gt; -k \u0026lt;kmers list\u0026gt; -o \u0026lt;output_directory\u0026gt; -t \u0026lt;num_nucleos\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCon el objetivo de comparar ensambles, ejecute nuevamente el comando\nanterior sin el flag \u003ccode\u003e--careful\u003c/code\u003e. Este flag intenta reducir el nmero de\nmismatches e indels cortos y es recomendado para el ensamble de genomas\nde procariotas. Debe cambiar la carpeta de salida para no sobre escribir\nlos resultados del primer ensamble, use \u003ccode\u003e03.Assembly/02.NoCareful\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEn el \u003ca href=\"http://cab.spbu.ru/files/release3.15.2/manual.html\" rel=\"nofollow\"\u003emanual\u003c/a\u003e\nencuentra ms detalles.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutput\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003ecorrected/\u003c/code\u003e: contiene las reads corregidas por \u003cstrong\u003eBayesHammer\u003c/strong\u003e en\n\u003ccode\u003e.fastq.gz\u003c/code\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003escaffolds.fasta\u003c/code\u003e: contiene los scaffolds obtenidos\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003econtigs.fasta\u003c/code\u003e: contiene los contigis obtenidos\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003eassembly_graph_with_scaffolds.gfa\u003c/code\u003e: contiene el grafo del montaje\nen formato GFA 1.0.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003eassembly_graph.fastg\u003c/code\u003e: contiene el grafo del montaje en formato\nFASTG\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.2 Evaluacin del ensamble\u003c/h2\u003e\u003ca id=\"user-content-42-evaluacin-del-ensamble\" class=\"anchor\" aria-label=\"Permalink: 4.2 Evaluacin del ensamble\" href=\"#42-evaluacin-del-ensamble\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl ensamble debe ser evaluado a travs de mtricas que representan la\ncalidad del genoma. En este paso ser calculado el N50, el nmero de\ncontigs, el tamao del genoma y que tan completo y contaminado est.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.2.1. Quast\u003c/h3\u003e\u003ca id=\"user-content-421-quast\" class=\"anchor\" aria-label=\"Permalink: 4.2.1. Quast\" href=\"#421-quast\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"http://quast.sourceforge.net/docs/manual.html\" rel=\"nofollow\"\u003eQuast v5.0.2\u003c/a\u003e (\u003cem\u003eQUality\nASsesment Tool\u003c/em\u003e) es posible evaluar las principales estadsticas del\nmontaje (i.e.N50, nmero de contigs, tamao total del montaje, tamao\nde los contigs, etc). \u003cstrong\u003eQuast\u003c/strong\u003e genera una serie de archivos y reportes\ndonde es posible observar esas estadsticas bsicas del montaje. Sern\ncomparados los montajes obtenidos anteriormente, con el objetivo de\nescoger el mejor, para las siguientes etapas.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInstalacin\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Cree el ambiente quast\nconda create -n quast\n\n# Active el ambiente quast\nconda activate quast\n\n# Instale Quast\nconda install -c bioconda quast\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Cree el ambiente quast\nconda create -n quast\n\n# Active el ambiente quast\nconda activate quast\n\n# Instale Quast\nconda install -c bioconda quast\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eUso\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Vuelva al diretorio base\n\n# Crie um diretrio para el output\nmkdir 04.AssemblyQuality\nmkdir 04.AssemblyQuality/01.Quast\n\n# Ejecute Quast\nquast.py 03.Assembly/01.Careful/scaffolds.fasta 03.Assembly/02.NoCareful/scaffolds.fasta -o 04.AssemblyQuality/01.Quast\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Vuelva al diretorio base\n\n# Crie um diretrio para el output\nmkdir 04.AssemblyQuality\nmkdir 04.AssemblyQuality/01.Quast\n\n# Ejecute Quast\nquast.py 03.Assembly/01.Careful/scaffolds.fasta 03.Assembly/02.NoCareful/scaffolds.fasta -o 04.AssemblyQuality/01.Quast\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e\n\u003ccode\u003equast.py path/to/assembly/contigs.fasta -o path/to/output/\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInterpetacin de los resultados\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLa idea de usar \u003cstrong\u003eQuast\u003c/strong\u003e, aparte de evaluar las estidsticas bsicas\ndel montaje, es comparar varios montajes para escoger el mejor. Por\nejemplo: entre menor sea el nmero de contigs es mejor, porque significa\nque el genoma qued menos fragementado. Y eso se reflejar en el tamao\nde los contigs que sern grandes. El valor de N50, es mejor entre mayor\nsea. As mismo, es ideal menor nmero de gaps y Ns.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOutputs\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eExplore el directorio de output usando el comando \u003ccode\u003els\u003c/code\u003e.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e04.AssemblyQuality/01.Quast/report.html\u003c/code\u003e: Este reporte puede ser\nabierto en un \u003cem\u003eweb browser\u003c/em\u003e y contiene las informaciones ms\nrelevantes. Como nmero de contigs, tamao del mayor contig, tamao\ntotal del montaje, N50, etc.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eLos contigs de tamao menor de 500 bp no tienen un valor representativo\nen el ensamble, por lo tanto es recomendable filtrar esas secuencias.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEl programa bbmap tiene un script que permite realizar la depuracin por\ntamao. Instale bbmap en un ambiente llamado bioinfo:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Cree el ambiente bioinfo\nconda install -n bioinfo\n\n# Active el ambiente bioinfo\nconda activate bioinfo\n\n# Instale bbmap\nconda install -c bioconda bbmap\n\n# Ejecute bbmap (careful)\nreformat.sh in=03.Assembly/01.Careful/scaffolds.fasta out=03.Assembly/01.Careful/scaffolds_filtered.fasta minlength=500\n\n# Ejecute bbmap (No careful)\nreformat.sh in=03.Assembly/02.NoCareful/scaffolds.fasta out=03.Assembly/02.NoCareful/scaffolds_filtered.fasta minlength=500\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Cree el ambiente bioinfo\nconda install -n bioinfo\n\n# Active el ambiente bioinfo\nconda activate bioinfo\n\n# Instale bbmap\nconda install -c bioconda bbmap\n\n# Ejecute bbmap (careful)\nreformat.sh in=03.Assembly/01.Careful/scaffolds.fasta out=03.Assembly/01.Careful/scaffolds_filtered.fasta minlength=500\n\n# Ejecute bbmap (No careful)\nreformat.sh in=03.Assembly/02.NoCareful/scaffolds.fasta out=03.Assembly/02.NoCareful/scaffolds_filtered.fasta minlength=500\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEjecute nuevamente Quast, evaluando los dos ensambles, los contigs\nfiltrados y sin filtar.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Active el ambiente quast\nconda activate quast\n\n# Ejecute Quast\nquast.py 03.Assembly/01.Careful/scaffolds.fasta 03.Assembly/01.Careful/scaffolds_filtered.fasta 03.Assembly/02.NoCareful/scaffolds.fasta 03.Assembly/02.NoCareful/scaffolds_filtered.fasta -o 04.AssemblyQuality/01.Quast\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Active el ambiente quast\nconda activate quast\n\n# Ejecute Quast\nquast.py 03.Assembly/01.Careful/scaffolds.fasta 03.Assembly/01.Careful/scaffolds_filtered.fasta 03.Assembly/02.NoCareful/scaffolds.fasta 03.Assembly/02.NoCareful/scaffolds_filtered.fasta -o 04.AssemblyQuality/01.Quast\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.2.2. CheckM\u003c/h3\u003e\u003ca id=\"user-content-422-checkm\" class=\"anchor\" aria-label=\"Permalink: 4.2.2. CheckM\" href=\"#422-checkm\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara evaluar que tan completo y contaminado estn los ensambles es usada\nla herramienta \u003ca href=\"https://github.com/Ecogenomics/CheckM/wiki\"\u003eCheckM\u003c/a\u003e. La\ncual usa una base de datos propia de genes ortlogos de copia nica.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eInstale el programa en un ambiente llamado \u003ccode\u003echeckm\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Cree el ambiente checkm\nconda create -n checkm\n\n# Active el ambiente\nconda activate checkm\n\n# Instale checkm\nconda install -c bioconda checkm-genome\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Cree el ambiente checkm\nconda create -n checkm\n\n# Active el ambiente\nconda activate checkm\n\n# Instale checkm\nconda install -c bioconda checkm-genome\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEjecute el nalisis para todos los ensambles, filtrados y no filtrados.\nLos ensambles deben estar todos en un mismo directorio, entonces cree un\nnuevo directorio dentro de la carpeta \u003ccode\u003e03.Assembly/\u003c/code\u003e llamado\n\u003ccode\u003e03.Scaffolds/\u003c/code\u003e. Despus entre en cada carpeta de los ensambles y usando\nel comando \u003ccode\u003emv\u003c/code\u003e, cambie el nombre de los archivos, con el fin de\ndiferenciarlos (actualmente tienen el mismo nombre) entre careful y no\ncareful. Por ltimo, nuevamente use \u003ccode\u003emv\u003c/code\u003e para mover los archivos para el\ndirectorio creado anteriormente (\u003ccode\u003e03.Assembly/03.Scaffolds/\u003c/code\u003e)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd 03.Assembly/01.Careful\n\nmv scaffolds.fasta careful.fasta\n\nmv scaffolds_filtered.fasta careful_filtered.fasta\n\nmv careful* ../03.Scaffolds\n\ncd ../02.NoCareful\n\nmv scaffolds.fasta Nocareful.fasta\n\nmv scaffolds_filtered.fasta Nocareful_filtered.fasta\n\nmv Nocareful* ../03.Scaffolds\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd 03.Assembly/01.Careful\n\nmv scaffolds.fasta careful.fasta\n\nmv scaffolds_filtered.fasta careful_filtered.fasta\n\nmv careful* ../03.Scaffolds\n\ncd ../02.NoCareful\n\nmv scaffolds.fasta Nocareful.fasta\n\nmv scaffolds_filtered.fasta Nocareful_filtered.fasta\n\nmv Nocareful* ../03.Scaffolds\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDespus de organizado los archivos, ya es posible ejecutar CheckM:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Cree el directorio para el output\ncd ../../\n\nmkdir 04.AssemblyQuality/02.Checkm\n\ncheckm lineage_wf 03.Assembly/03.Scaffolds/ 04.AssemblyQuality/02.Checkm -t 40 -x fasta --tab \u0026gt; 04.AssemblyQuality/02.Checkm/output.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Cree el directorio para el output\ncd ../../\n\nmkdir 04.AssemblyQuality/02.Checkm\n\ncheckm lineage_wf 03.Assembly/03.Scaffolds/ 04.AssemblyQuality/02.Checkm -t 40 -x fasta --tab \u0026gt; 04.AssemblyQuality/02.Checkm/output.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"checkm lineage_wf \u0026lt;input_directory/\u0026gt; \u0026lt;output_directory/\u0026gt; -t \u0026lt;num_nucleos\u0026gt; -x \u0026lt;format\u0026gt; --tab \u0026gt; output.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003echeckm lineage_wf \u0026lt;input_directory/\u0026gt; \u0026lt;output_directory/\u0026gt; -t \u0026lt;num_nucleos\u0026gt; -x \u0026lt;format\u0026gt; --tab \u0026gt; output.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eExplore el archivo de salida \u003ccode\u003e04.AssemblyQuality/02.Checkm/output.txt\u003c/code\u003e\nusando el comando \u003ccode\u003eless\u003c/code\u003e. Descargue el reporte en su computador.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePara ms detalles sobre la interpretacin del reporte visite este\n\u003ca href=\"https://www.biostars.org/p/447744/\" rel=\"nofollow\"\u003elink.\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.3. Anotacin Taxonmica\u003c/h2\u003e\u003ca id=\"user-content-43-anotacin-taxonmica\" class=\"anchor\" aria-label=\"Permalink: 4.3. Anotacin Taxonmica\" href=\"#43-anotacin-taxonmica\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eLa clasificacin con base en datos de genoma tiene mayor poder de\nresolucin en comparacin al usar apenas un gene marcador, pues esta\nabordaje analiza mltiples genes que presentan un resultado mucho ms\nrobusto de las relaciones de parentesco del organismo de inters.\n\u003ca href=\"https://ecogenomics.github.io/GTDBTk/index.html\" rel=\"nofollow\"\u003eGTDB-tk\u003c/a\u003e es una\nherramienta que identifica 120 genes marcadores y los compara con una\nbase de datos curada y constamente actualizada.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eInstale el prohrama en un ambiente llamado \u003ccode\u003egtdbtk\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Cree el ambiente gtdbtk\nconda create -n gtdbtk\n\n# Active el ambiente\nconda activate gtdbtk\n\n# Instale GTDBtk\nconda install -c conda-forge -c bioconda gtdbtk\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Cree el ambiente gtdbtk\nconda create -n gtdbtk\n\n# Active el ambiente\nconda activate gtdbtk\n\n# Instale GTDBtk\nconda install -c conda-forge -c bioconda gtdbtk\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara descargar y configurar la base de datos, siga las\n\u003ca href=\"https://ecogenomics.github.io/GTDBTk/installing/bioconda.html#step-3-download-and-alias-the-gtdb-tk-reference-data\" rel=\"nofollow\"\u003einstrucciones\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEjecute GTDBtk usando el seguinte comando:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"gtdbtk classify_wf --genome_dir 03.Assembly/03.Scaffolds/ --out_dir 05.TaxonomyAnnotation/ -x fasta --cpus 40\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egtdbtk classify_wf --genome_dir 03.Assembly/03.Scaffolds/ --out_dir 05.TaxonomyAnnotation/ -x fasta --cpus 40\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"gtdbtk classify_wf --genome_dir \u0026lt;input_directory/\u0026gt; --out_dir \u0026lt;output_directory/\u0026gt; -x \u0026lt;format\u0026gt; --cpus \u0026lt;num_nucleos\u0026gt;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egtdbtk classify_wf --genome_dir \u0026lt;input_directory/\u0026gt; --out_dir \u0026lt;output_directory/\u0026gt; -x \u0026lt;format\u0026gt; --cpus \u0026lt;num_nucleos\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl programa genera varios archivos de salida que estn resumidos en\n\u003ccode\u003e05.TaxonomyAnnotation/gtdbtk.bac120.summary.tsv\u003c/code\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.4. Prediccin de genes\u003c/h2\u003e\u003ca id=\"user-content-44-prediccin-de-genes\" class=\"anchor\" aria-label=\"Permalink: 4.4. Prediccin de genes\" href=\"#44-prediccin-de-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eel objetivo de esta etapa es procurar los ORFs (\u003cem\u003eOpen Reading Frames\u003c/em\u003e)\ndentro de los contig/scaffolds. Es decir, predecir donde inician e\nterminan los genes. Basicamente el programa busca codones de inicio y de\n\u003cem\u003estop\u003c/em\u003e.Para este objetivo ser usada la herramienta\n\u003ca href=\"https://github.com/hyattpd/prodigal/wiki\"\u003eProdigal\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInstalacin\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eInstale el programa en el ambiente bioinfo\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Active el ambiente\nconda activate bioinfo\n\n# Instale Prodigal\nconda install -c bioconda prodigal\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Active el ambiente\nconda activate bioinfo\n\n# Instale Prodigal\nconda install -c bioconda prodigal\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara ejecutar el programa utilice el seguiente comando:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Cree una carpeta para guardar el resultado\nmkdir 06.GenePrediction\n\n#\nprodigal -f gff -i 03.Assembly/03.Scaffolds/careful_filtered.fasta -o 06.GenePrediction/careful_filtered_orf.gff -a 06.GenePrediction/careful_filtered_proteins.faa -d 06.GenePrediction/careful_filtered_ORFnucleotides.fa -s 06.GenePrediction/careful_filtered_genes\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Cree una carpeta para guardar el resultado\nmkdir 06.GenePrediction\n\n#\nprodigal -f gff -i 03.Assembly/03.Scaffolds/careful_filtered.fasta -o 06.GenePrediction/careful_filtered_orf.gff -a 06.GenePrediction/careful_filtered_proteins.faa -d 06.GenePrediction/careful_filtered_ORFnucleotides.fa -s 06.GenePrediction/careful_filtered_genes\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"prodigal -f \u0026lt;output_format\u0026gt; -i \u0026lt;input_assembly.fasta\u0026gt; -o \u0026lt;output_gbk_file\u0026gt; -a \u0026lt;output_protein_seq_file\u0026gt; -d \u0026lt;output_nucleotides_seq_file\u0026gt; -s \u0026lt;output_coordinates_file\u0026gt;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eprodigal -f \u0026lt;output_format\u0026gt; -i \u0026lt;input_assembly.fasta\u0026gt; -o \u0026lt;output_gbk_file\u0026gt; -a \u0026lt;output_protein_seq_file\u0026gt; -d \u0026lt;output_nucleotides_seq_file\u0026gt; -s \u0026lt;output_coordinates_file\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl archivo \u003ccode\u003e06.GenePrediction/careful_filtered_orf.gff\u003c/code\u003e incluyendo las\nsecuencias condificantes y no codificantes y las posiciones en cada\ncontig. Adicionalmente, el programa genera archivos con las secuencias\nde nucletidos (\u003ccode\u003e06.GenePrediction/careful_filtered_ORFnucleotides.fa\u003c/code\u003e)\ny protenas (\u003ccode\u003e06.GenePrediction/careful_filtered_proteins.faa\u003c/code\u003e) de cada\ngene predicho.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eNota:\u003c/em\u003e Ejecute el mismo comando en el genoma de \u003cem\u003eE. coli\u003c/em\u003e ATCC8739. Use\n\u003ccode\u003eecoli_genome_\u003c/code\u003e como prefijo para los nombres de los archivos de salida.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eV. Transcriptmica\u003c/h1\u003e\u003ca id=\"user-content-v-transcriptmica\" class=\"anchor\" aria-label=\"Permalink: V. Transcriptmica\" href=\"#v-transcriptmica\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEn esta seccin sern abordadas las etapas y programas para la\nevaluacin librerias de RNA-seq.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.1. Remocin de secuencias de rRNA\u003c/h3\u003e\u003ca id=\"user-content-51-remocin-de-secuencias-de-rrna\" class=\"anchor\" aria-label=\"Permalink: 5.1. Remocin de secuencias de rRNA\" href=\"#51-remocin-de-secuencias-de-rrna\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAunque durante el proceso de extraccin de RNA se use un kit para la\ndeplecin de RNA ribosomal, puede suceder que en las librerias de\nRNA-seq an existan algunas secuencias de rRNA, las cuales podrn ser\neliminadas usando el programa\n\u003ca href=\"https://github.com/biocore/sortmerna\"\u003eSortmeRNA\u003c/a\u003e, que a su vez usa\nbases de dados de rRNA, tales como Rfam 5S, Rfam 5.8S,\nSILVA_138.1_NR99_SSURef_tax_silva and\nSILVA_138.1_LSURef_tax_silva, para buscar y eliminar esas secuencias\nde las bibliotecas.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInstalacin\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eInstale SortmeRNA en un ambiente virtual llamado mapping\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# cree el ambiente\nconda create -n mapping\n\n# Active el ambiente\nconda activate mappin\n\n# Instale SortmeRNA\nconda install -c bioconda sortmerna\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# cree el ambiente\nconda create -n mapping\n\n# Active el ambiente\nconda activate mappin\n\n# Instale SortmeRNA\nconda install -c bioconda sortmerna\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eLas bases de datos son descargadas de las pginas de \u003ca href=\"https://www.arb-silva.de/\" rel=\"nofollow\"\u003eSILVA\ndatabase\u003c/a\u003e y \u003ca href=\"https://rfam.xfam.org/\" rel=\"nofollow\"\u003eRfam\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eUso\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for i in 00.RawData/*_1.fq\ndo\nSAMPLE=$(basename $i _1.fq)\nnohup sortmerna --ref /home/drm/Documentos/database/Rfam/Rfam_5S.fa --ref /home/drm/Documentos/database/Rfam/Rfam_5.8S.fa --ref /home/drm/Documentos/database/silva/SILVA_138.1_NR99_16S_28S.fasta --ref /home/drm/Documentos/database/silva/SILVA_138.1_23S_28S.fasta --reads $i --reads 00.RawData/${SAMPLE}_2.fq --paired_in --threads 50 --workdir 07.SortmeRNA/${SAMPLE}/ --idx-dir 07.SortmeRNA/${SAMPLE}/ --kvdb 07.SortmeRNA/${SAMPLE}/kvdb --fastx --aligned 07.SortmeRNA/${SAMPLE}/${SAMPLE}_aligned.fasta --other 07.SortmeRNA/${SAMPLE}/${SAMPLE}_other.fasta --out2\ndone\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efor i in 00.RawData/*_1.fq\ndo\nSAMPLE=$(basename $i _1.fq)\nnohup sortmerna --ref /home/drm/Documentos/database/Rfam/Rfam_5S.fa --ref /home/drm/Documentos/database/Rfam/Rfam_5.8S.fa --ref /home/drm/Documentos/database/silva/SILVA_138.1_NR99_16S_28S.fasta --ref /home/drm/Documentos/database/silva/SILVA_138.1_23S_28S.fasta --reads $i --reads 00.RawData/${SAMPLE}_2.fq --paired_in --threads 50 --workdir 07.SortmeRNA/${SAMPLE}/ --idx-dir 07.SortmeRNA/${SAMPLE}/ --kvdb 07.SortmeRNA/${SAMPLE}/kvdb --fastx --aligned 07.SortmeRNA/${SAMPLE}/${SAMPLE}_aligned.fasta --other 07.SortmeRNA/${SAMPLE}/${SAMPLE}_other.fasta --out2\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl comando anterior, ejecuta SortmeRNA en todas las bibliotecas de\nRNA-seq. Al final de proceso, cada muestra tendr una carpeta dentro del\ndirectorio \u003ccode\u003e07.SortmeRNA/\u003c/code\u003e. Los archivos terminados en \u003ccode\u003e_other_fwd.fq\u003c/code\u003e y\n\u003ccode\u003e_other_rev.fq\u003c/code\u003e, contiene las secuencias que \u003cem\u003eno\u003c/em\u003e alinearon contra las\nbases de dados de rRNA, es decir libres de rRNA. Estos sern los\narchivos que sern usados para los siguientes pasos.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.2. Alineamiento a um genoma de referencia\u003c/h3\u003e\u003ca id=\"user-content-52-alineamiento-a-um-genoma-de-referencia\" class=\"anchor\" aria-label=\"Permalink: 5.2. Alineamiento a um genoma de referencia\" href=\"#52-alineamiento-a-um-genoma-de-referencia\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDespus de la evaluacin de la calidad y remocin de rRNA, se debe hacer\nun alineamiento o mapeo de las lecturas para determinar en qu parte del\ngenoma o de dnde se originaron las lecturas. Hay una serie de\nherramientas que realizan esta funcin. En este tutorial usaremos\n\u003ca href=\"https://github.com/DaehwanKimLab/hisat2\"\u003eHISAT2\u003c/a\u003e, pero tambin una\nherramienta como \u003ca href=\"https://github.com/alexdobin/STAR\"\u003eSTAR\u003c/a\u003e o\n\u003ca href=\"https://github.com/infphilo/tophat\"\u003eTopHat2\u003c/a\u003e, hacen el trabajo.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEn este caso ser usado el genoma obtenido desde la pgina de la\ncoleccin ATCC, que debe estar almacenado en la carpeta \u003ccode\u003e03.Assembly/\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLa primera etapa del proceso con\n\u003ca href=\"https://daehwankimlab.github.io/hisat2/\" rel=\"nofollow\"\u003eHISAT2\u003c/a\u003e, es llamada de\nbusqueda de semillas (\u003cem\u003eseeds\u003c/em\u003e), y consiste en que para cada read el\nprograma buscar la porcin de la lectura ms larga que coincida\nexactamente con una o ms ubicaciones en el genoma de referencia\n(\u003cem\u003eseed1\u003c/em\u003e). Posteriormente, volver a buscar la parte no mapeada de la\nlectura para encontrar la siguiente secuencia ms larga que coincida\nexactamente con el genoma de referencia (\u003cem\u003eseed2\u003c/em\u003e). Y as hasta que\nencuentra las \u003cem\u003eseeds\u003c/em\u003e de todas las lecturas. Esta forma de mapeo hace\nque el algortmo sea eficiente y rpido. La segunda etapa, comprende la\nclusterizacin de \u003cem\u003eseeds\u003c/em\u003e prximas, para generar una read completa.\nDespus, el programa unir las \u003cem\u003eseeds\u003c/em\u003e baseado en el mejor alineamiento\nde las reads (\u003cem\u003escore\u003c/em\u003e de \u003cem\u003emismatches\u003c/em\u003e, \u003cem\u003eindels\u003c/em\u003e, \u003cem\u003egaps\u003c/em\u003e, etc)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInstalao\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHISAT2 v2.2.1 ser instalado en el ambiente virutal mapping. Tambin es\nnecesario instalar la herramienta \u003ca href=\"http://www.htslib.org/\" rel=\"nofollow\"\u003eSamTools\u003c/a\u003e,\nque sirve en la manipulacin de archivos \u003ccode\u003e.sam\u003c/code\u003e y \u003ccode\u003e.bam\u003c/code\u003e.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Activa el ambiente\nconda activate mapping\n\n# Instala\nconda install -c bioconda hisat2\n\n# Instala SamTools\nconda install -c bioconda samtools\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Activa el ambiente\nconda activate mapping\n\n# Instala\nconda install -c bioconda hisat2\n\n# Instala SamTools\nconda install -c bioconda samtools\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.2.1. Indexando el genoma de referencia\u003c/h4\u003e\u003ca id=\"user-content-521-indexando-el-genoma-de-referencia\" class=\"anchor\" aria-label=\"Permalink: 5.2.1. Indexando el genoma de referencia\" href=\"#521-indexando-el-genoma-de-referencia\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl primer paso para el alineamiento es indexar el genoma. Cree un\ndirectorio para todo el proceso de mapeo (p.e. \u003ccode\u003e08.Mapping.RNA\u003c/code\u003e)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Indexando\nhisat2-build -p 10 03.Assembly/Escherichia_coli_ATCC_8739.fasta 08.Mapping.RNA/Escherichia_coli_ATCC_8739\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Indexando\nhisat2-build -p 10 03.Assembly/Escherichia_coli_ATCC_8739.fasta 08.Mapping.RNA/Escherichia_coli_ATCC_8739\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e \u003ccode\u003ehisat2-build -p # [genome] [index]\u003c/code\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-p\u003c/code\u003e: nmero de threads\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e[genome]\u003c/code\u003e: Genoma de referencia\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e[index]\u003c/code\u003e: ndice\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.2.2. Mapeo\u003c/h4\u003e\u003ca id=\"user-content-522-mapeo\" class=\"anchor\" aria-label=\"Permalink: 5.2.2. Mapeo\" href=\"#522-mapeo\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEn el primer comando \u003cstrong\u003eHISAT2\u003c/strong\u003e va a hacer el alinemiento, el cual\ngenerar una serie de archivos \u003ccode\u003e.sam\u003c/code\u003e (\u003cem\u003eSequence aligment Map\u003c/em\u003e). En el\nsegundo comando, \u003cstrong\u003eSamTools\u003c/strong\u003e transformar a archivos \u003ccode\u003e.bam\u003c/code\u003e (\u003cem\u003eBinary\nAlignment Map\u003c/em\u003e), los cuales son una versin comprimida de \u003ccode\u003e.sam\u003c/code\u003e.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"hisat2  --dta -p 15 -x 08.Mapping.RNA/Escherichia_coli_ATCC_8739 -1 00.RawData/acetone_1.fq -2 00.RawData/acetone_2.fq -S 08.Mapping.RNA/acetone.sam\n\n# loop\nfor i in 00.RawData/*_1.fq\ndo\nsample=$(basename $i _1.fq)\nhisat2 --dta -p 15 -x 08.Mapping.RNA/Escherichia_coli_ATCC_8739 -1 $i -2 00.RawData/${sample}_2.fq -S 08.Mapping.RNA/${sample}.sam\ndone\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ehisat2  --dta -p 15 -x 08.Mapping.RNA/Escherichia_coli_ATCC_8739 -1 00.RawData/acetone_1.fq -2 00.RawData/acetone_2.fq -S 08.Mapping.RNA/acetone.sam\n\n# loop\nfor i in 00.RawData/*_1.fq\ndo\nsample=$(basename $i _1.fq)\nhisat2 --dta -p 15 -x 08.Mapping.RNA/Escherichia_coli_ATCC_8739 -1 $i -2 00.RawData/${sample}_2.fq -S 08.Mapping.RNA/${sample}.sam\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003ehisat2 --dta -p \u0026lt;num_nucleos\u0026gt; -x \u0026lt;index\u0026gt; -1 RNA-seqs_pair1 -2 RNA-seqs_pair2\u003c/code\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e--dta\u003c/code\u003e: Genera un reporte que ser usado en el prximo passo.\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-p\u003c/code\u003e: Nmero de ncleos\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-x\u003c/code\u003e: ndice\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-1\u003c/code\u003e: Secuencias RNA-seq pair 1\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-2\u003c/code\u003e: Secuencias RNA-seq pair 2\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.2.3. Transformando los archivos del mapeo\u003c/h4\u003e\u003ca id=\"user-content-523-transformando-los-archivos-del-mapeo\" class=\"anchor\" aria-label=\"Permalink: 5.2.3. Transformando los archivos del mapeo\" href=\"#523-transformando-los-archivos-del-mapeo\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl primer paso es transformar el archivo de salida del mapeo (\u003ccode\u003e.sam\u003c/code\u003e)\npara \u003ccode\u003e.bam\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"samtools sort -o 08.Mapping.RNA/acetone.bam 08.Mapping.RNA/acetone.sam\n\n# loop\nfor i in 08.Mapping.RNA/*.sam\ndo\nsample=$(basename $i .sam)\nsamtools sort -o 08.Mapping.RNA/${sample}.bam $i\ndone\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esamtools sort -o 08.Mapping.RNA/acetone.bam 08.Mapping.RNA/acetone.sam\n\n# loop\nfor i in 08.Mapping.RNA/*.sam\ndo\nsample=$(basename $i .sam)\nsamtools sort -o 08.Mapping.RNA/${sample}.bam $i\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"samtools sort -o \u0026lt;output.bam\u0026gt; \u0026lt;input.sam\u0026gt;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esamtools sort -o \u0026lt;output.bam\u0026gt; \u0026lt;input.sam\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.3. Ensamble del transcriptoma\u003c/h3\u003e\u003ca id=\"user-content-53-ensamble-del-transcriptoma\" class=\"anchor\" aria-label=\"Permalink: 5.3. Ensamble del transcriptoma\" href=\"#53-ensamble-del-transcriptoma\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl objetivo de esta etapa es el montaje del transcriptoma, usando los\nalineamientos del paso anterior. En este tutorial, usaremos\n\u003ca href=\"https://ccb.jhu.edu/software/stringtie/\" rel=\"nofollow\"\u003eStringTie\u003c/a\u003e, el cual emplea\nalgoritmos eficientes para la recuperacin de la estructura del\ntrasncriptoma y la estimacin de la abundancia a partir de lecturas de\nRNA-Seq alineadas con un genoma de referencia. Toma como entrada los\nalineamientos SAM/BAM, ordenados por coordenadas y produce una salida\nGTF que consiste en transcritos ensamblados y sus niveles estimados\n(FPKM / TPM y valores de cobertura).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInstalacin\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Activa el ambiente assembly\nconda activate assembly\n\n# Instale StringTie\nconda install -c bioconda stringtie\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Activa el ambiente assembly\nconda activate assembly\n\n# Instale StringTie\nconda install -c bioconda stringtie\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCree un directorio para la salida del proceso:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Cree un directorio para el montaje\nmkdir 09.AssemblyRNA\n\n# StringTie\nstringtie acetone.bam -p 25 -G 06.GenePrediction/ecoli_genome_orf.gff -o 09.AssemblyRNA/acetone_stringtie.gtf\n\n# Loop\nfor i in 08.Mapping.RNA/*.bam\ndo\nsample=$(basename $i .bam)\nstringtie $i -p 25 -G 06.GenePrediction/ecoli_genome_orf.gff -o 09.AssemblyRNA/${sample}_stringtie.gtf\ndone\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Cree un directorio para el montaje\nmkdir 09.AssemblyRNA\n\n# StringTie\nstringtie acetone.bam -p 25 -G 06.GenePrediction/ecoli_genome_orf.gff -o 09.AssemblyRNA/acetone_stringtie.gtf\n\n# Loop\nfor i in 08.Mapping.RNA/*.bam\ndo\nsample=$(basename $i .bam)\nstringtie $i -p 25 -G 06.GenePrediction/ecoli_genome_orf.gff -o 09.AssemblyRNA/${sample}_stringtie.gtf\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSintaxis\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"stringtie \u0026lt;sorted.bam\u0026gt; -p \u0026lt;num_nucleos\u0026gt; -G \u0026lt;reference.genome.gff\u0026gt; -o \u0026lt;output.gtf\u0026gt;  -l \u0026lt;transcriptos_prefix\u0026gt;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003estringtie \u0026lt;sorted.bam\u0026gt; -p \u0026lt;num_nucleos\u0026gt; -G \u0026lt;reference.genome.gff\u0026gt; -o \u0026lt;output.gtf\u0026gt;  -l \u0026lt;transcriptos_prefix\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl archivo de salida \u003ccode\u003e.gtf\u003c/code\u003e contiene las definiciones estructurales de\nlos transcriptos ensamblados por \u003cstrong\u003eStringTie\u003c/strong\u003e a partir de la\ninformacin de mapeo.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUse \u003ccode\u003eless\u003c/code\u003e para ver el contenido del archivo de salida\n\u003ccode\u003e09.AssemblyRNA/acetone.gtf\u003c/code\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.3.1. Transcriptoma final\u003c/h3\u003e\u003ca id=\"user-content-531-transcriptoma-final\" class=\"anchor\" aria-label=\"Permalink: 5.3.1. Transcriptoma final\" href=\"#531-transcriptoma-final\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eA continuacin sern unidos los transcriptomas de cada muestra para\nobtner el transcriptoma final. Para esto, ser creado un archivo\n(\u003ccode\u003emergelist.txt\u003c/code\u003e) con el nombre de los \u003cem\u003eoutput\u003c/em\u003e obtenidos en el comando\nde StringTie.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Creando el archivo mergelist.txt\necho -e 'acetone_stringtie.gtf\\noil1-2_R1_stringtie.gtf\\noil1-2_R2_stringtie.gtf\\noil1-2_R3_stringtie.gtf\\noil1-4_R1_stringtie.gtf\\noil1-4_R2_stringtie.gtf\\noil1-4_R3_stringtie.gtf\\noil1-8_R1_stringtie.gtf\\noil1-8_R2_stringtie.gtf\\noil1-8_R3_stringtie.gtf\\nwater_R1_stringtie.gtf\\nwater_R2_stringtie.gtf ' \u0026gt; 09.AssemblyRNA/mergelist.txt \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Creando el archivo mergelist.txt\necho -e 'acetone_stringtie.gtf\\noil1-2_R1_stringtie.gtf\\noil1-2_R2_stringtie.gtf\\noil1-2_R3_stringtie.gtf\\noil1-4_R1_stringtie.gtf\\noil1-4_R2_stringtie.gtf\\noil1-4_R3_stringtie.gtf\\noil1-8_R1_stringtie.gtf\\noil1-8_R2_stringtie.gtf\\noil1-8_R3_stringtie.gtf\\nwater_R1_stringtie.gtf\\nwater_R2_stringtie.gtf ' \u0026gt; 09.AssemblyRNA/mergelist.txt \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEjecute el comando \u003ccode\u003emerge\u003c/code\u003e de StringTie:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd 09.AssemblyRNA\n\nstringtie --merge -G ../06.GenePrediction/ecoli_genome_orf.gff -o stringtie_merged.gtf mergelist.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd 09.AssemblyRNA\n\nstringtie --merge -G ../06.GenePrediction/ecoli_genome_orf.gff -o stringtie_merged.gtf mergelist.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.3.2. Extrayendo las secuencias\u003c/h3\u003e\u003ca id=\"user-content-532-extrayendo-las-secuencias\" class=\"anchor\" aria-label=\"Permalink: 5.3.2. Extrayendo las secuencias\" href=\"#532-extrayendo-las-secuencias\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHasta aqui tenemos el archivo final de coordenadas genmicas\n(\u003ccode\u003e09.AssemblyRNA/stringtie_merged.gtf\u003c/code\u003e). En esta etapa usaremos ese\narchivo para extraer las secuencias en un archivo \u003ccode\u003e.fasta\u003c/code\u003e, usando el\nprograma \u003ca href=\"https://github.com/gpertea/gffread\"\u003eGFFREAD\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInstalacin\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eInstale el programa en el ambiente bioinfo.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Active el ambiente\nconda activate bioinfo\n\n# Instale el ambiente\nconda install -c bioconda gffread\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Active el ambiente\nconda activate bioinfo\n\n# Instale el ambiente\nconda install -c bioconda gffread\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eUso\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePara extraer las secuencias, ejecute el siguiente comando:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd ..\n\ngffread 09.AssemblyRNA/stringtie_merged.gtf -g 03.Assembly/Escherichia_coli_ATCC_8739.fasta -w 09.AssemblyRNA/all_transcripts.fasta\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd ..\n\ngffread 09.AssemblyRNA/stringtie_merged.gtf -g 03.Assembly/Escherichia_coli_ATCC_8739.fasta -w 09.AssemblyRNA/all_transcripts.fasta\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.4. Cuantificacin de la expresin\u003c/h2\u003e\u003ca id=\"user-content-54-cuantificacin-de-la-expresin\" class=\"anchor\" aria-label=\"Permalink: 5.4. Cuantificacin de la expresin\" href=\"#54-cuantificacin-de-la-expresin\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDespus de obtenidos los transcriptos, estos deben ser cuantificados. El\nprograma que ser usado en este tutorial, se llama\n\u003ca href=\"https://pachterlab.github.io/kallisto/\" rel=\"nofollow\"\u003eKallisto\u003c/a\u003e. Este software\ncuantifica las abundancias de transcriptos. Se basa en la novedosa idea\nde pseudoalineacin para determinar rpidamente la compatibilidad de\nlecturas con los \u003cem\u003etargets\u003c/em\u003e, sin necesidad de alineacin.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInstalacin\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eInstale Kallisto el ambiente \u003cstrong\u003ebioinfo\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Activa el ambiente bioinfo\nconda activate bioinfo\n\n# Instale Kallisto\nconda install -c bioconda kallisto\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Activa el ambiente bioinfo\nconda activate bioinfo\n\n# Instale Kallisto\nconda install -c bioconda kallisto\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCree un directorio para el proceso de cuantificacin\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"mkdir 10.TranscriptsQuantification\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emkdir 10.TranscriptsQuantification\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl primer paso es indexar el archivo\n\u003ccode\u003e09.AssemblyRNA/all_transcripts.fasta\u003c/code\u003e:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"kallisto index -i 09.AssemblyRNA/index_transcritos 09.AssemblyRNA/all_transcripts.fasta\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ekallisto index -i 09.AssemblyRNA/index_transcritos 09.AssemblyRNA/all_transcripts.fasta\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eA continuacin sern cuantificados los transcriptomas de cada libreria.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"kallisto quant -i 09.AssemblyRNA/index_transcritos -o 10.TranscriptsQuantification/acetone -b 1000 -t 20 00.RawData/acetone_1.fq 00.RawData/acetone_2.fq\n\n# Loop\nfor i in 00.RawData/*_1.fq\ndo\nsample=$(basename $i _1.fq)\nkallisto quant -i 09.AssemblyRNA/index_transcritos -o 10.TranscriptsQuantification/${sample} -b 1000 -t 20 $i 00.RawData/${sample}_2.fq\ndone\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ekallisto quant -i 09.AssemblyRNA/index_transcritos -o 10.TranscriptsQuantification/acetone -b 1000 -t 20 00.RawData/acetone_1.fq 00.RawData/acetone_2.fq\n\n# Loop\nfor i in 00.RawData/*_1.fq\ndo\nsample=$(basename $i _1.fq)\nkallisto quant -i 09.AssemblyRNA/index_transcritos -o 10.TranscriptsQuantification/${sample} -b 1000 -t 20 $i 00.RawData/${sample}_2.fq\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSern creadas carpetas para cada muestra y dentro se encuentra el\narchivo de salida de \u003cstrong\u003eKallisto\u003c/strong\u003e que es una tabla llamada\n\u003ccode\u003eabundance.tsv\u003c/code\u003e con el ID de cada transcrito y el valor de \u003cstrong\u003eTPM\u003c/strong\u003e para\ncada uno de ellos, adems del tamao de cada uno.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUse \u003ccode\u003eless\u003c/code\u003e para ver el contenido de la tabla \u003ccode\u003eabundance.tsv\u003c/code\u003e que se\nencuentra dentro de la carpeta \u003ccode\u003e12.TranscriptQuantification/\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEl siguiente paso es unir las tablas de abundancia de cada muestra y\nconstruir la matriz de abundancia para iniciar el anlisis de expresin\ndiferencial. El script \u003ccode\u003eabundance_estimates_to_matrix.pl\u003c/code\u003e, de la\nherramienta \u003ca href=\"https://github.com/trinityrnaseq/trinityrnaseq\"\u003eTrinity\u003c/a\u003e\npermite la construccin de la matriz de abundancia. En este tutorial no\nse usar como tal el programa Trinity, que es un ensamblador de RNA-seq\n\u003cem\u003ede novo\u003c/em\u003e, recomendado para tratar datos de metatranscriptmica. Sin\nembargo, es necesario descargar el repositorio de la herramienta:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"git clone https://github.com/trinityrnaseq/trinityrnaseq\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egit clone https://github.com/trinityrnaseq/trinityrnaseq\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl comando anterior descargar una carpeta llamada \u003ccode\u003etrinityrnaseq/\u003c/code\u003e. El\nscript estar en el directorio \u003ccode\u003etrinityrnaseq/util/\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAntes de ejecutar el script, es necesario modificar los nombres de las\ntablas de abundancia de cada muestra, de manera que puedan ser\neficientemente diferenciados entre ellos.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cp 10.TranscriptsQuantification/acetone/abundance.tsv 10.TranscriptsQuantification/acetone.tsv\n\ncp 10.TranscriptsQuantification/oil1-2_R1/abundance.tsv 10.TranscriptsQuantification/oil1-2_R1.tsv\n\ncp 10.TranscriptsQuantification/oil1-2_R2/abundance.tsv 10.TranscriptsQuantification/oil1-2_R2.tsv\n\ncp 10.TranscriptsQuantification/oil1-2_R3/abundance.tsv 10.TranscriptsQuantification/oil1-2_R3.tsv\n\ncp 10.TranscriptsQuantification/oil1-4_R1/abundance.tsv 10.TranscriptsQuantification/oil1-4_R1.tsv\n\ncp 10.TranscriptsQuantification/oil1-4_R2/abundance.tsv 10.TranscriptsQuantification/oil1-4_R2.tsv\n\ncp 10.TranscriptsQuantification/oil1-4_R3/abundance.tsv 10.TranscriptsQuantification/oil1-4_R3.tsv\n\ncp 10.TranscriptsQuantification/oil1-8_R1/abundance.tsv 10.TranscriptsQuantification/oil1-8_R1.tsv\n\ncp 10.TranscriptsQuantification/oil1-8_R2/abundance.tsv 10.TranscriptsQuantification/oil1-8_R2.tsv\n\ncp 10.TranscriptsQuantification/oil1-8_R3/abundance.tsv 10.TranscriptsQuantification/oil1-8_R3.tsv\n\ncp 10.TranscriptsQuantification/water_R1/abundance.tsv 10.TranscriptsQuantification/water_R1.tsv\n\ncp 10.TranscriptsQuantification/water_R2/abundance.tsv 10.TranscriptsQuantification/water_R2.tsv\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecp 10.TranscriptsQuantification/acetone/abundance.tsv 10.TranscriptsQuantification/acetone.tsv\n\ncp 10.TranscriptsQuantification/oil1-2_R1/abundance.tsv 10.TranscriptsQuantification/oil1-2_R1.tsv\n\ncp 10.TranscriptsQuantification/oil1-2_R2/abundance.tsv 10.TranscriptsQuantification/oil1-2_R2.tsv\n\ncp 10.TranscriptsQuantification/oil1-2_R3/abundance.tsv 10.TranscriptsQuantification/oil1-2_R3.tsv\n\ncp 10.TranscriptsQuantification/oil1-4_R1/abundance.tsv 10.TranscriptsQuantification/oil1-4_R1.tsv\n\ncp 10.TranscriptsQuantification/oil1-4_R2/abundance.tsv 10.TranscriptsQuantification/oil1-4_R2.tsv\n\ncp 10.TranscriptsQuantification/oil1-4_R3/abundance.tsv 10.TranscriptsQuantification/oil1-4_R3.tsv\n\ncp 10.TranscriptsQuantification/oil1-8_R1/abundance.tsv 10.TranscriptsQuantification/oil1-8_R1.tsv\n\ncp 10.TranscriptsQuantification/oil1-8_R2/abundance.tsv 10.TranscriptsQuantification/oil1-8_R2.tsv\n\ncp 10.TranscriptsQuantification/oil1-8_R3/abundance.tsv 10.TranscriptsQuantification/oil1-8_R3.tsv\n\ncp 10.TranscriptsQuantification/water_R1/abundance.tsv 10.TranscriptsQuantification/water_R1.tsv\n\ncp 10.TranscriptsQuantification/water_R2/abundance.tsv 10.TranscriptsQuantification/water_R2.tsv\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEjecute el script\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Active el ambiente bioinfo\nconda activate bioinfo\n\n# Entre al directorio 10\ncd 10.TranscriptsQuantification/\n\n# Ejecute el script\nperl trinityrnaseq/util/abundance_estimates_to_matrix.pl --est_method kallisto --gene_trans_map none acetone.tsv oil1-2_R1.tsv oil1-2_R2.tsv oil1-2_R3.tsv oil1-4_R1.tsv oil1-4_R2.tsv oil1-4_R3.tsv oil1-8_R1.tsv oil1-8_R2.tsv oil1-8_R3.tsv water_R1.tsv water_R2.tsv\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Active el ambiente bioinfo\nconda activate bioinfo\n\n# Entre al directorio 10\ncd 10.TranscriptsQuantification/\n\n# Ejecute el script\nperl trinityrnaseq/util/abundance_estimates_to_matrix.pl --est_method kallisto --gene_trans_map none acetone.tsv oil1-2_R1.tsv oil1-2_R2.tsv oil1-2_R3.tsv oil1-4_R1.tsv oil1-4_R2.tsv oil1-4_R3.tsv oil1-8_R1.tsv oil1-8_R2.tsv oil1-8_R3.tsv water_R1.tsv water_R2.tsv\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eLos archivos de salida son:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003ekallisto.isoform.TMM.EXPR.matrix\u003c/code\u003e: matriz normalizada por TMM\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003ekallisto.isoform.TPM.not_cross_norm\u003c/code\u003e: matriz de TPM\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003ekallisto.isoform.counts.matrix\u003c/code\u003e: matriz de counts\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003ekallisto.isoform.TPM.not_cross_norm.TMM_info.txt\u003c/code\u003e: estadstica de\nlos factores de normalizacin por profundidad de la secuenciacin\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003ekallisto.isoform.TPM.not_cross_norm.runTMM.r\u003c/code\u003e: script en R para\nrealizar la normalizacin en TMM.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.5. Expresin diferencial\u003c/h2\u003e\u003ca id=\"user-content-55-expresin-diferencial\" class=\"anchor\" aria-label=\"Permalink: 5.5. Expresin diferencial\" href=\"#55-expresin-diferencial\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl anlisis de expresin diferencial consiste en comparar la expresin\nde los genes en diferentes condiciones e identificar genes que son\nbastante ms o menos expresados.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEste anlisis puede ser realizado con diferentes bibliotecas del\nsoftware R, tales como\n\u003ca href=\"https://bioconductor.org/packages/release/bioc/html/edgeR.html\" rel=\"nofollow\"\u003eedgeR\u003c/a\u003e,\n\u003ca href=\"https://github.com/mikelove/DESeq2\"\u003eDESeq2\u003c/a\u003e y\n\u003ca href=\"https://github.com/pachterlab/sleuth\"\u003eSleuth\u003c/a\u003e. Aqui sern abordado el\nanlisis con DESeq2. Este anlisis ser ejecutado en el ambiente\nRStudio.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePausa para una brevisiiima introduccin al lenguaje R\u003c/h1\u003e\u003ca id=\"user-content-pausa-para-una-brevisiiima-introduccin-al-lenguaje-r\" class=\"anchor\" aria-label=\"Permalink: Pausa para una brevisiiima introduccin al lenguaje R\" href=\"#pausa-para-una-brevisiiima-introduccin-al-lenguaje-r\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1. Instalacin\u003c/h2\u003e\u003ca id=\"user-content-1-instalacin\" class=\"anchor\" aria-label=\"Permalink: 1. Instalacin\" href=\"#1-instalacin\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePara instalar R, de click \u003ca href=\"https://vps.fmvz.usp.br/CRAN/\" rel=\"nofollow\"\u003eaqui\u003c/a\u003e. En\nseguida instele\n\u0026gt;\u003ca href=\"https://www.rstudio.com/products/rstudio/download/#download\" rel=\"nofollow\"\u003eRStudio\u003c/a\u003e.\u003c/p\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/khidalgo85/training_transcriptomics/blob/master/imgs/R.png\"\u003e\u003cimg src=\"/khidalgo85/training_transcriptomics/raw/master/imgs/R.png\" align=\"center\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003cp dir=\"auto\"\u003eRStudio debe lucir parecido a la imagen anterior, con cuatro paneles:\ni) Editor de texto/cdigos, ii) consola, iii) Espacio de trabajo e\nhistorial y iv) Plots, archivos y men de ayuda.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2. Instalacin, actualizacin y remocin de librerias\u003c/h2\u003e\u003ca id=\"user-content-2-instalacin-actualizacin-y-remocin-de-librerias\" class=\"anchor\" aria-label=\"Permalink: 2. Instalacin, actualizacin y remocin de librerias\" href=\"#2-instalacin-actualizacin-y-remocin-de-librerias\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003einstall.packages()\u003c/strong\u003e: Instala librerias desde el repositorio\nCRAN. \u003cstrong\u003eNota:\u003c/strong\u003e Existen otras maneras de instalar librerias cuando\nstas no se encuentran en CRAM, p.e. GitHub, BiocLite, devtools.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"install.packages('ggplot2')\ninstall.packages('tidyverse')\"\u003e\u003cpre\u003einstall.packages(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eggplot2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\ninstall.packages(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003etidyverse\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eInformacin de ayuda\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"help(package = 'ggplot2')\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ehelp(package = 'ggplot2')\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3. Tipos de datos, constantes y variables\u003c/h2\u003e\u003ca id=\"user-content-3-tipos-de-datos-constantes-y-variables\" class=\"anchor\" aria-label=\"Permalink: 3. Tipos de datos, constantes y variables\" href=\"#3-tipos-de-datos-constantes-y-variables\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTipos de datos\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eNmericos\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eEnteros\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eReales\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eImaginrios\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eCaracteres/string o texto\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eTexto\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eLgicos\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eTRUE\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFALSE\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAsignacin de valores a variables\u003c/strong\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"a \u0026lt;- 3\nb \u0026lt;- 2\na + b \n#\u0026gt; [1] 5\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ea\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ea\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eb\u003c/span\u003e \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] 5\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTipos bsicos de datos:\u003c/strong\u003e nmericos, caracteres y lgicos\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"a \u0026lt;- 3 \nclass(a)\n#\u0026gt; [1] \u0026quot;numeric\u0026quot;\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ea\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e \nclass(\u003cspan class=\"pl-smi\"\u003ea\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] \"numeric\"\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"nombre \u0026lt;- \u0026quot;Maria\u0026quot;\nclass(nombre)\n#\u0026gt; [1] \u0026quot;character\u0026quot;\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003enombre\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMaria\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nclass(\u003cspan class=\"pl-smi\"\u003enombre\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] \"character\"\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"class(TRUE)\n#\u0026gt; [1] \u0026quot;logical\u0026quot;\"\u003e\u003cpre\u003eclass(\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] \"logical\"\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDando nombres a las variables\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSe pueden usar letras, nmeros y underline.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eReglas\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eNo puede comenzar por nmero\u003c/li\u003e\n\u003cli\u003ePuede comenzar por punto seguido de una letra o secuencia de\nletras.\u003c/li\u003e\n\u003cli\u003eNo puede comenzar por punto seguido de nmeros.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNOTA\u003c/strong\u003e R es un lenguaje \u003cem\u003ecase sensitive\u003c/em\u003e, o sea es sensible a\nmayscula y minscula.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4. Operadores y funciones matemticas\u003c/h2\u003e\u003ca id=\"user-content-4-operadores-y-funciones-matemticas\" class=\"anchor\" aria-label=\"Permalink: 4. Operadores y funciones matemticas\" href=\"#4-operadores-y-funciones-matemticas\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003e\u003cstrong\u003eOperao\u003c/strong\u003e\u003c/th\u003e\n\u003cth\u003e\u003cstrong\u003eDescrio\u003c/strong\u003e\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003ex + y\u003c/td\u003e\n\u003ctd\u003eAdicin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ex - y\u003c/td\u003e\n\u003ctd\u003eSubstraccin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ex * y\u003c/td\u003e\n\u003ctd\u003eMultiplicacin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ex / y\u003c/td\u003e\n\u003ctd\u003eDivisin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ex ^ 2 (x **y)\u003c/td\u003e\n\u003ctd\u003ePtenciacin o exponenciacin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ex %% y\u003c/td\u003e\n\u003ctd\u003eResto de la divisin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ex %/% y\u003c/td\u003e\n\u003ctd\u003eCosciente (Parte Entera)\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"17/2 #Divisin exacta\n#\u0026gt; [1] 8.5\n17%/%2 #Parte exacta\n#\u0026gt; [1] 8\n17%%2 # Resto de la divisin\n#\u0026gt; [1] 1\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c1\"\u003e17\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eDivisin exacta\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] 8.5\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003e17\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e%/%\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eParte exacta\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] 8\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003e17\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e%%\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Resto de la divisin\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] 1\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eFunciones aritmticas y trigonomtricas bsicas\u003c/strong\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"log2(9)\n#\u0026gt; [1] 3.169925\nsqrt(9)\n#\u0026gt; [1] 3\ncos(60)\n#\u0026gt; [1] -0.952413\"\u003e\u003cpre\u003elog2(\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] 3.169925\u003c/span\u003e\nsqrt(\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] 3\u003c/span\u003e\ncos(\u003cspan class=\"pl-c1\"\u003e60\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] -0.952413\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eOtras funciones\u003c/em\u003e \u003ccode\u003esin(), tan()\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5. Estructura de datos\u003c/h2\u003e\u003ca id=\"user-content-5-estructura-de-datos\" class=\"anchor\" aria-label=\"Permalink: 5. Estructura de datos\" href=\"#5-estructura-de-datos\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.1. Vectores\u003c/h3\u003e\u003ca id=\"user-content-51-vectores\" class=\"anchor\" aria-label=\"Permalink: 5.1. Vectores\" href=\"#51-vectores\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEs la combinacin de mltiples valores (nmericos, caracteres o\nlgicos)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"mivector \u0026lt;- c(12,32,7,89) \nclass(mivector) #tipo de informacin contenida en el vector\n#\u0026gt; [1] \u0026quot;numeric\u0026quot;\nlength(mivector) # tamao del vector\n#\u0026gt; [1] 4\nletras \u0026lt;- c('a','b','c','d')\nclass(letras)\n#\u0026gt; [1] \u0026quot;character\u0026quot;\nlength(letras)\n#\u0026gt; [1] 4\nvector.logico \u0026lt;- c(FALSE,FALSE,TRUE,F,T)\nclass(vector.logico)\n#\u0026gt; [1] \u0026quot;logical\u0026quot;\nlength(vector.logico)\n#\u0026gt; [1] 5\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003emivector\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e32\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e7\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e89\u003c/span\u003e) \nclass(\u003cspan class=\"pl-smi\"\u003emivector\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etipo de informacin contenida en el vector\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] \"numeric\"\u003c/span\u003e\nlength(\u003cspan class=\"pl-smi\"\u003emivector\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e tamao del vector\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] 4\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eletras\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ea\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eb\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ec\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ed\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\nclass(\u003cspan class=\"pl-smi\"\u003eletras\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] \"character\"\u003c/span\u003e\nlength(\u003cspan class=\"pl-smi\"\u003eletras\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] 4\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003evector.logico\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\nclass(\u003cspan class=\"pl-smi\"\u003evector.logico\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] \"logical\"\u003c/span\u003e\nlength(\u003cspan class=\"pl-smi\"\u003evector.logico\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] 5\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eFormas de crear vectores numricos\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003efuncin \u003ccode\u003ec()\u003c/code\u003e concatenar\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eOperador \u003ccode\u003e:\u003c/code\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003efuncin \u003ccode\u003eseq()\u003c/code\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003efuncin \u003ccode\u003erep()\u003c/code\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.2. Matrices\u003c/h3\u003e\u003ca id=\"user-content-52-matrices\" class=\"anchor\" aria-label=\"Permalink: 5.2. Matrices\" href=\"#52-matrices\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEs la combinacin de mltiples vectores, con el mismo tipo de datos\n(nmericos, caracteres o lgicos).\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"m1 \u0026lt;- matrix(data = c(2,5,7,8,21,32,22,11), nrow=4,\n             ncol = 2, byrow = TRUE)\nm1\n#\u0026gt;      [,1] [,2]\n#\u0026gt; [1,]    2    5\n#\u0026gt; [2,]    7    8\n#\u0026gt; [3,]   21   32\n#\u0026gt; [4,]   22   11\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003em1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ematrix\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003edata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e7\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e21\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e32\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e22\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e11\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003enrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e,\n             \u003cspan class=\"pl-v\"\u003encol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ebyrow\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003em1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;      [,1] [,2]\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1,]    2    5\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [2,]    7    8\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [3,]   21   32\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [4,]   22   11\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.3. Data Frames\u003c/h3\u003e\u003ca id=\"user-content-53-data-frames\" class=\"anchor\" aria-label=\"Permalink: 5.3. Data Frames\" href=\"#53-data-frames\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMatrices y vectores poseen una limitacin, solo guardan un tipo de\ninformacin (lgicas, numricas, texto). Mientras que los data frame\npueden contener todo tipo de informacin en un solo objeto. El data\nframe es bidimensional, en lneas y columnas, y en cada columna puede\ntener un tipo de informacin. Normalmente, cada colunma es una\nvariable y las lneas contienen los valores.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAlgunas caractersticas son:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eLas columnas tendrn siempre nombres\u003c/li\u003e\n\u003cli\u003eLos nombres de las lneas deben ser nicos\u003c/li\u003e\n\u003cli\u003eCada columna debe presentar un tipo nico de datos\u003c/li\u003e\n\u003cli\u003eCada columna debe tener el mismo nmero de lneas\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"df \u0026lt;- data.frame(nombres=c('Ana','Juan','Vera'), notas=c(10,2,7.5), situacion=c(T,F,T), row.names = c('l1','l2','l3'))\nView(df)\nstr(df) # presenta la estructura de los datos\n#\u0026gt; 'data.frame':    3 obs. of  3 variables:\n#\u0026gt;  $ nombres  : chr  \u0026quot;Ana\u0026quot; \u0026quot;Juan\u0026quot; \u0026quot;Vera\u0026quot;\n#\u0026gt;  $ notas    : num  10 2 7.5\n#\u0026gt;  $ situacion: logi  TRUE FALSE TRUE\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003enombres\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAna\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eJuan\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eVera\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003enotas\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e7.5\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esituacion\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003el1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003el2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003el3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e))\nView(\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e)\nstr(\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e presenta la estructura de los datos\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; 'data.frame':    3 obs. of  3 variables:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;  $ nombres  : chr  \"Ana\" \"Juan\" \"Vera\"\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;  $ notas    : num  10 2 7.5\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt;  $ situacion: logi  TRUE FALSE TRUE\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eEsta fue la brevisiima introduccin al lenguaje R. Para mayores\ndetalles y profundidad de los conocimientos en este lenguaje, consulte\notros materiales.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.5. Expresin diferencial (continuacin)\u003c/h2\u003e\u003ca id=\"user-content-55-expresin-diferencial-continuacin\" class=\"anchor\" aria-label=\"Permalink: 5.5. Expresin diferencial (continuacin)\" href=\"#55-expresin-diferencial-continuacin\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCompacte las carpetas \u003ccode\u003e10.TranscriptsQuantification/\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"zip -r 10.TranscriptsQuantification.zip 10.TranscriptsQuantification/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ezip -r 10.TranscriptsQuantification.zip 10.TranscriptsQuantification/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDescargue la carpeta a su computador usando el comando \u003ccode\u003escp\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"scp user@direccionip:/home/user/transcriptomica/10.TranscriptsQuantification.zip camino/en/su/computador\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003escp user@direccionip:/home/user/transcriptomica/10.TranscriptsQuantification.zip camino/en/su/computador\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.6. Anotacin Funcional\u003c/h2\u003e\u003ca id=\"user-content-56-anotacin-funcional\" class=\"anchor\" aria-label=\"Permalink: 5.6. Anotacin Funcional\" href=\"#56-anotacin-funcional\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAntes de hacer el anlisis de expresin diferencial, los transcriptos\ndeben ser anotados, para conocer su funcin, para lo cual los\ntranscriptos sern funcionalmente anotados por homologa, usando bases\nde datos e um alineador.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePara el alineamiento ser usado el programa\n\u003ca href=\"https://github.com/bbuchfink/diamond/wiki/3.-Command-line-options\"\u003eDiamond\u003c/a\u003e,\ne una base de dayos curada como \u003ca href=\"https://www.kegg.jp/\" rel=\"nofollow\"\u003eKEGG\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eInstalacin\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eInstale Diamond en un ambiente virtual llamado diamond\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Cree el ambiente\nconda create -n diamond\n\n# Active el ambiente\nconda activate diamond\n\n# Instale Diamond\nconda install -c bioconda diamond\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Cree el ambiente\nconda create -n diamond\n\n# Active el ambiente\nconda activate diamond\n\n# Instale Diamond\nconda install -c bioconda diamond\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDescargue las bases de datos\n\u003ca href=\"https://drive.google.com/file/d/1ZxjJdwh1izP32X5CH-B8SN0DK2WAAAvr/view?usp=sharing\" rel=\"nofollow\"\u003eaqui\u003c/a\u003e.\nUna vez descargadas las bases de datos, debe subir los archivos al\nservidor y almacenarlos en una carpeta llamada \u003ccode\u003edbs\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUse o siguiente comando para ejecutar Diamond:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Cree un directorio para la salida\nmkdir 11.FunctionalAnnotation\n\n# Diamond\ndiamond blastx --more-sensitive --threads 50 -k 1 -f 6 qseqid qlen sseqid sallseqid slen qstart qend sstart send evalue bitscore score length pident qcovhsp --id 60 --query-cover 60 -d dbs/keggdb.dmnd --query 09.AssemblyRNA/all_transcripts.fasta -o 11.FunctionalAnnotation/kegg.txt --tmpdir /dev/shm\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Cree un directorio para la salida\nmkdir 11.FunctionalAnnotation\n\n# Diamond\ndiamond blastx --more-sensitive --threads 50 -k 1 -f 6 qseqid qlen sseqid sallseqid slen qstart qend sstart send evalue bitscore score length pident qcovhsp --id 60 --query-cover 60 -d dbs/keggdb.dmnd --query 09.AssemblyRNA/all_transcripts.fasta -o 11.FunctionalAnnotation/kegg.txt --tmpdir /dev/shm\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEl archivo de \u003cem\u003eoutput\u003c/em\u003e es una tabla que contiene los cdigos de los\ngenes y las anotaciones (nmero KEGG), adems de otras informaciones\nrelacionadas al alineamiento de cada secuencia (i.e.cobertura, % de\nidentidad). Debido a que los genes fueron anotados apenas con los\nnmeros KEGG, es necessario cruzar esta informacin con las anotaciones\ndetalladas para cada nmero KEGG, que indican la funcin y categoria de\nlos genes. A continuacin encontrar una srie de pasos para obtener la\ntabla final de anotacin funcional:\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eSeparar la colunma de los cdigos KEGG\u003c/li\u003e\n\u003c/ol\u003e\n\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"perl -pe 's/\\|?(?:\\s+gi|ref)?\\|\\s*/\\t/g' 11.FunctionalAnnotation/kegg.txt \u0026gt; 11.FunctionalAnnotation/kegg_formated.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eperl -pe 's/\\|?(?:\\s+gi|ref)?\\|\\s*/\\t/g' 11.FunctionalAnnotation/kegg.txt \u0026gt; 11.FunctionalAnnotation/kegg_formated.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eCortar las columnas 1 (Contig ID) y 4 (KEGG ID) e grabar en un nuevo\narchivo\u003c/li\u003e\n\u003c/ol\u003e\n\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cut -f1,4 11.FunctionalAnnotation/kegg_formated.txt \u0026gt; 11.FunctionalAnnotation/kegg_ID_contigs_keggs.tsv\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecut -f1,4 11.FunctionalAnnotation/kegg_formated.txt \u0026gt; 11.FunctionalAnnotation/kegg_ID_contigs_keggs.tsv\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eOrdenando\u003c/li\u003e\n\u003c/ol\u003e\n\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"sort -k2,2 11.FunctionalAnnotation/kegg_ID_contigs_keggs.tsv \u0026gt; 11.FunctionalAnnotation/kegg_ID_contigs_keggs_sorted.tsv\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esort -k2,2 11.FunctionalAnnotation/kegg_ID_contigs_keggs.tsv \u0026gt; 11.FunctionalAnnotation/kegg_ID_contigs_keggs_sorted.tsv\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eDescargue el archivo con las informaciones de los cdigos KEGG\u003c/li\u003e\n\u003c/ol\u003e\n\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"curl -L https://figshare.com/ndownloader/files/33953774 -o kegg.tsv\n\n# ordene\nsort -k1,1 kegg.tsv \u0026gt; kegg_sorted.tsv\n\n# mueva el archivo para 11.FunctionalAnnotation/\nmv kegg_sorted.tsv 11.FunctionalAnnotation\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecurl -L https://figshare.com/ndownloader/files/33953774 -o kegg.tsv\n\n# ordene\nsort -k1,1 kegg.tsv \u0026gt; kegg_sorted.tsv\n\n# mueva el archivo para 11.FunctionalAnnotation/\nmv kegg_sorted.tsv 11.FunctionalAnnotation\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003eAsigne nombres a las columnas\u003c/li\u003e\n\u003c/ol\u003e\n\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"echo -e \u0026quot;target_id\\tkegg_id\\tlevel1\\tlevel2\\tlevel3\\tGenName\u0026quot; \u0026gt; 11.FunctionalAnnotation/kegg_randomic_analysis.tsv\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eecho -e \"target_id\\tkegg_id\\tlevel1\\tlevel2\\tlevel3\\tGenName\" \u0026gt; 11.FunctionalAnnotation/kegg_randomic_analysis.tsv\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eAsignacin randmica\u003c/li\u003e\n\u003c/ol\u003e\n\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cat 11.FunctionalAnnotation/kegg_ID_contigs_keggs_sorted.tsv | while read line ; do echo \u0026quot;$line\u0026quot; | join -1 2 -2 1 -e\u0026quot;NA\u0026quot; -o1.1,0,2.2,2.3,2.4,2.5 -t $'\\t' - 11.FunctionalAnnotation/kegg_sorted.tsv | shuf -n1 \u0026gt;\u0026gt; 11.FunctionalAnnotation/kegg_randomic_analysis.tsv ; done\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecat 11.FunctionalAnnotation/kegg_ID_contigs_keggs_sorted.tsv | while read line ; do echo \"$line\" | join -1 2 -2 1 -e\"NA\" -o1.1,0,2.2,2.3,2.4,2.5 -t $'\\t' - 11.FunctionalAnnotation/kegg_sorted.tsv | shuf -n1 \u0026gt;\u0026gt; 11.FunctionalAnnotation/kegg_randomic_analysis.tsv ; done\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003eAsigne nombres a las columnas\u003c/li\u003e\n\u003c/ol\u003e\n\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"echo -e \u0026quot;target_id\\tkegg_id\\tlevel1\\tlevel2\\tlevel3\\tGenName\u0026quot; \u0026gt; transcripts_annotations.tsv\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eecho -e \"target_id\\tkegg_id\\tlevel1\\tlevel2\\tlevel3\\tGenName\" \u0026gt; transcripts_annotations.tsv\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"7\" dir=\"auto\"\u003e\n\u003cli\u003eDescargue y explore a tabla final de anotaciones\n\u003ccode\u003e11.FunctionalAnnotation/kegg_randomic_analysis.tsv\u003c/code\u003e.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eGuarde el archivo en la carpeta \u003ccode\u003e10.TranscriptsQuantification/\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEn este punto puede cerrar su conexin con el servidor y trabajar en el\nambiente de RStudio localmente.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEn su computador, cree un directorio de trabajo y en l, descompacte las\ncarpetas descargadas del servidor.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRStudio\u003c/h1\u003e\u003ca id=\"user-content-rstudio\" class=\"anchor\" aria-label=\"Permalink: RStudio\" href=\"#rstudio\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAbra RStudio, cree un nuevo proyecto: File \u0026lt; New Project \u0026lt;\nExisting Directory\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEscoja el directorio donde descompact las carpetas descargadas del\nservidor.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUna vez el nuevo proyecto est creado, abra un script: File \u0026lt; New\nFile \u0026lt; R Script. En este script debera copiar la serie de comandos\nque encontrar a continuacin:\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eInstalando las bibliotecas necesarias:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# DESeq2\nif (! require(DESeq2)) {\n  source(\u0026quot;https://bioconductor.org/biocLite.R\u0026quot;)\n  biocLite(\u0026quot;DESeq2\u0026quot;)\n}\n\n# Enhanced Volcano\n\nBiocManager::install(\u0026quot;EnhancedVolcano\u0026quot;)\n\n# tidyverse\ninstall.packages(\u0026quot;tidyverse\u0026quot;)\n\n# tibble\ninstall.packages(\u0026quot;tibble\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e DESeq2\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e (\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e require(\u003cspan class=\"pl-smi\"\u003eDESeq2\u003c/span\u003e)) {\n  source(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehttps://bioconductor.org/biocLite.R\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  biocLite(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDESeq2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n}\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Enhanced Volcano\u003c/span\u003e\n\n\u003cspan class=\"pl-e\"\u003eBiocManager\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003einstall(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEnhancedVolcano\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e tidyverse\u003c/span\u003e\ninstall.packages(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etidyverse\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e tibble\u003c/span\u003e\ninstall.packages(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etibble\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eActivando las bibliotecas\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#  Librerias\nlibrary(DESeq2)\nlibrary(EnhancedVolcano)\nlibrary(tidyverse)\nlibrary(tibble)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e  Librerias\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003eDESeq2\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eEnhancedVolcano\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003etidyverse\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003etibble\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eLeyendo las tablas\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Leyendo tabla de anotaciones\nanot \u0026lt;- read.delim(\u0026quot;10.TranscriptsQuantification/kegg_randomic_analysis.tsv\u0026quot;,\n                   sep = \u0026quot;\\t\u0026quot;) %\u0026gt;% \n  separate(GenName, c(\u0026quot;GenName\u0026quot;, \u0026quot;Function\u0026quot;), sep = ';') %\u0026gt;% \n  select(target_id, kegg_id, GenName)\n\n\n# Leyendo la matriz de counts\ndata = read.table(\u0026quot;10.TranscriptsQuantification/kallisto.isoform.counts.matrix\u0026quot;, \n                  header=T, row.names=1, com='')\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Leyendo tabla de anotaciones\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eanot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.delim(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e10.TranscriptsQuantification/kegg_randomic_analysis.tsv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                   \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  separate(\u003cspan class=\"pl-smi\"\u003eGenName\u003c/span\u003e, c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGenName\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFunction\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  select(\u003cspan class=\"pl-smi\"\u003etarget_id\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ekegg_id\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eGenName\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Leyendo la matriz de counts\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003edata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e read.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e10.TranscriptsQuantification/kallisto.isoform.counts.matrix\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                  \u003cspan class=\"pl-v\"\u003eheader\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecom\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eRedondeando los valores de la matriz de expresin y filtrando (al\nmenos dos counts por transcripto)\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# redondeando los valores\nrnaseqMatrix = round(data)\n\n# Corte a por lo menos 2 counts por transcripto\nrnaseqMatrix = rnaseqMatrix[rowSums(rnaseqMatrix) \u0026gt;= 2,]\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e redondeando los valores\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003ernaseqMatrix\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e round(\u003cspan class=\"pl-smi\"\u003edata\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Corte a por lo menos 2 counts por transcripto\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003ernaseqMatrix\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ernaseqMatrix\u003c/span\u003e[rowSums(\u003cspan class=\"pl-smi\"\u003ernaseqMatrix\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,]\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eGenerando la tabla del diseo experimental\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"conditions = data.frame(conditions=factor(c(\u0026quot;acetone\u0026quot;, rep(\u0026quot;oil_tretated_0.5\u0026quot;, 3),\n                                            rep(\u0026quot;oil_treated_0.25\u0026quot;,3), \n                                            rep(\u0026quot;oil_treated_0.125\u0026quot;,3),\n                                            rep(\u0026quot;control\u0026quot;,2))))\nrownames(conditions) = colnames(rnaseqMatrix)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003econditions\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003econditions\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eacetone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoil_tretated_0.5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e),\n                                            rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoil_treated_0.25\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e), \n                                            rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoil_treated_0.125\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e),\n                                            rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtrol\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e))))\nrownames(\u003cspan class=\"pl-smi\"\u003econditions\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003ernaseqMatrix\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003eGenerando el objeto DESeq\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"ddsFullCountTable \u0026lt;- DESeqDataSetFromMatrix(\n  countData = rnaseqMatrix,\n  colData = conditions,\n  design = ~ conditions)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eddsFullCountTable\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e DESeqDataSetFromMatrix(\n  \u003cspan class=\"pl-v\"\u003ecountData\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ernaseqMatrix\u003c/span\u003e,\n  \u003cspan class=\"pl-v\"\u003ecolData\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003econditions\u003c/span\u003e,\n  \u003cspan class=\"pl-v\"\u003edesign\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003econditions\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"7\" dir=\"auto\"\u003e\n\u003cli\u003eClculo de la expresin diferencial\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"dds = DESeq(ddsFullCountTable, test=\u0026quot;Wald\u0026quot;, fitType=\u0026quot;mean\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003edds\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e DESeq(\u003cspan class=\"pl-smi\"\u003eddsFullCountTable\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etest\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eWald\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efitType\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emean\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"8\" dir=\"auto\"\u003e\n\u003cli\u003eComparando \u003cstrong\u003eAcetona vs control\u003c/strong\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Comparando acetona vc control\nacetonevscontrol \u0026lt;- results(dds, alpha=.05, lfcThreshold=1, \n                               contrast = c(\u0026quot;conditions\u0026quot;,\u0026quot;acetone\u0026quot;,\u0026quot;control\u0026quot;))\n\n\nacetonevscontrol.df \u0026lt;- as.data.frame(acetonevscontrol) %\u0026gt;% \n  rownames_to_column(var = \u0026quot;target_id\u0026quot;)\n\n\n\n# Juntando tabla de expresin diferencial y anotacin funcional\nfinal.table \u0026lt;- full_join(anot,acetonevscontrol.df) %\u0026gt;% \n  column_to_rownames(var='target_id')\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Comparando acetona vc control\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eacetonevscontrol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e results(\u003cspan class=\"pl-smi\"\u003edds\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.05\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elfcThreshold\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \n                               \u003cspan class=\"pl-v\"\u003econtrast\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econditions\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eacetone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtrol\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\n\u003cspan class=\"pl-smi\"\u003eacetonevscontrol.df\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.data.frame(\u003cspan class=\"pl-smi\"\u003eacetonevscontrol\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  rownames_to_column(\u003cspan class=\"pl-v\"\u003evar\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etarget_id\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Juntando tabla de expresin diferencial y anotacin funcional\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003efinal.table\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e full_join(\u003cspan class=\"pl-smi\"\u003eanot\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eacetonevscontrol.df\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  column_to_rownames(\u003cspan class=\"pl-v\"\u003evar\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003etarget_id\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"9\" dir=\"auto\"\u003e\n\u003cli\u003eVolcano Plot\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"EnhancedVolcano(final.table,\n                lab = final.table$GenName,\n                x = 'log2FoldChange',\n                y = 'pvalue',\n                pCutoff = 0.05,\n                FCcutoff = 0.5,\n                xlim = c(-12, 12),\n                pointSize = 1.5,\n                labSize = 2.5,\n                colAlpha = 0.7,\n                legendLabSize = 12,\n                legendIconSize = 4.0,\n                drawConnectors = TRUE,\n                widthConnectors = 0.2,\n                title = \u0026quot;Differential Expression Analysis\u0026quot;,\n                subtitle = \u0026quot;Control vs. Acetone\u0026quot;,\n                max.overlaps = 20,\n                colConnectors = 'grey30')\"\u003e\u003cpre\u003eEnhancedVolcano(\u003cspan class=\"pl-smi\"\u003efinal.table\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003elab\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efinal.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eGenName\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003elog2FoldChange\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003epvalue\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003epCutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003eFCcutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003exlim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e),\n                \u003cspan class=\"pl-v\"\u003epointSize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1.5\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003elabSize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2.5\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003ecolAlpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.7\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003elegendLabSize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003elegendIconSize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e4.0\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003edrawConnectors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003ewidthConnectors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDifferential Expression Analysis\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003esubtitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eControl vs. Acetone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003emax.overlaps\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003ecolConnectors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egrey30\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/khidalgo85/training_transcriptomics/blob/master/imgs/unnamed-chunk-22-1.png\"\u003e\u003cimg src=\"/khidalgo85/training_transcriptomics/raw/master/imgs/unnamed-chunk-22-1.png\" width=\"100%\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eA continuacin se repetir el proceso para comparar el tratamiento con\naceite a 1/2 del MIC y el control\u003c/p\u003e\n\u003col start=\"8\" dir=\"auto\"\u003e\n\u003cli\u003eComparando \u003cstrong\u003eAceite 1/2 MIC vs control\u003c/strong\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Control vs oil_tretated_0.5\ncontrolvsoil0.5 \u0026lt;- results(dds, alpha=.05, lfcThreshold=1, \n                            contrast = c(\u0026quot;conditions\u0026quot;,\u0026quot;oil_tretated_0.5\u0026quot;,\u0026quot;control\u0026quot;))\n\n\ncontrolvsoil0.5.df \u0026lt;- as.data.frame(controlvsoil0.5) %\u0026gt;% \n  rownames_to_column(var = \u0026quot;target_id\u0026quot;)\n\n\n# Juntando tabla de expresin diferencial y anotacin funcional\nfinal.table \u0026lt;- full_join(anot,controlvsoil0.5.df) %\u0026gt;% \n  column_to_rownames(var='target_id')\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Control vs oil_tretated_0.5\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003econtrolvsoil0.5\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e results(\u003cspan class=\"pl-smi\"\u003edds\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.05\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elfcThreshold\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \n                            \u003cspan class=\"pl-v\"\u003econtrast\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econditions\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoil_tretated_0.5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtrol\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\n\u003cspan class=\"pl-smi\"\u003econtrolvsoil0.5.df\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.data.frame(\u003cspan class=\"pl-smi\"\u003econtrolvsoil0.5\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  rownames_to_column(\u003cspan class=\"pl-v\"\u003evar\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etarget_id\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Juntando tabla de expresin diferencial y anotacin funcional\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003efinal.table\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e full_join(\u003cspan class=\"pl-smi\"\u003eanot\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003econtrolvsoil0.5.df\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  column_to_rownames(\u003cspan class=\"pl-v\"\u003evar\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003etarget_id\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"9\" dir=\"auto\"\u003e\n\u003cli\u003eVolcano Plot\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"EnhancedVolcano(final.table,\n                lab = final.table$GenName,\n                x = 'log2FoldChange',\n                y = 'pvalue',\n                pCutoff = 0.05,\n                FCcutoff = 0.5,\n                xlim = c(-12, 12),\n                pointSize = 1.5,\n                labSize = 2.5,\n                colAlpha = 0.7,\n                legendLabSize = 12,\n                legendIconSize = 4.0,\n                drawConnectors = TRUE,\n                widthConnectors = 0.2,\n                title = \u0026quot;Differential Expression Analysis\u0026quot;,\n                subtitle = \u0026quot;Control vs. Cinnamomum camphora Oil (1/2 MIC)\u0026quot;,\n                max.overlaps = 20,\n                colConnectors = 'grey30')\"\u003e\u003cpre\u003eEnhancedVolcano(\u003cspan class=\"pl-smi\"\u003efinal.table\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003elab\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efinal.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eGenName\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003elog2FoldChange\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003epvalue\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003epCutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003eFCcutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003exlim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e),\n                \u003cspan class=\"pl-v\"\u003epointSize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1.5\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003elabSize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2.5\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003ecolAlpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.7\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003elegendLabSize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003elegendIconSize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e4.0\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003edrawConnectors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003ewidthConnectors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDifferential Expression Analysis\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003esubtitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eControl vs. Cinnamomum camphora Oil (1/2 MIC)\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003emax.overlaps\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003ecolConnectors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egrey30\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/khidalgo85/training_transcriptomics/blob/master/imgs/unnamed-chunk-24-1.png\"\u003e\u003cimg src=\"/khidalgo85/training_transcriptomics/raw/master/imgs/unnamed-chunk-24-1.png\" width=\"100%\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUse los comandos anteriores, modificando la condicin(es) que quiere\ncomparar.\u003c/p\u003e\n\u003col start=\"10\" dir=\"auto\"\u003e\n\u003cli\u003ePlot PCA\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#Criando PCA para visualizao da clusterizao das amostras\nrld \u0026lt;- rlog(dds)\n\nplotPCA(rld, intgroup = \u0026quot;conditions\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eCriando PCA para visualizao da clusterizao das amostras\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003erld\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rlog(\u003cspan class=\"pl-smi\"\u003edds\u003c/span\u003e)\n\nplotPCA(\u003cspan class=\"pl-smi\"\u003erld\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eintgroup\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econditions\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/khidalgo85/training_transcriptomics/blob/master/imgs/unnamed-chunk-25-1.png\"\u003e\u003cimg src=\"/khidalgo85/training_transcriptomics/raw/master/imgs/unnamed-chunk-25-1.png\" width=\"100%\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eVI Referencias\u003c/h1\u003e\u003ca id=\"user-content-vi-referencias\" class=\"anchor\" aria-label=\"Permalink: VI Referencias\" href=\"#vi-referencias\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eAndrews, S. (2017). FastQC: a quality control tool for high\nthroughput sequence data. 2010\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://academic.oup.com/bioinformatics/article-abstract/30/15/2114/2390096\" rel=\"nofollow\"\u003eTrimmomatic: a flexible trimmer for Illumina sequence\ndata\u003c/a\u003e(Trimmomatic:\na flexible trimmer for Illumina sequence data)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://www.liebertpub.com/doi/full/10.1089/cmb.2012.0021\" rel=\"nofollow\"\u003eSPAdes: a new genome assembly algorithm and its applications to\nsingle-cell\nsequencing\u003c/a\u003e(Bankevich,\nA., Nurk, S., Antipov, D., Gurevich, A. A., Dvorkin, M., Kulikov, A.\nS.,  \u0026amp; Pevzner, P. A. (2012). SPAdes: a new genome assembly\nalgorithm and its applications to single-cell sequencing. Journal of\ncomputational biology, 19(5), 455-477.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://academic.oup.com/bioinformatics/article-abstract/29/8/1072/228832\" rel=\"nofollow\"\u003eQUAST: quality assessment tool for genome\nassemblies\u003c/a\u003e(Gurevich,\nA., Saveliev, V., Vyahhi, N., \u0026amp; Tesler, G. (2013). QUAST: quality\nassessment tool for genome assemblies. Bioinformatics, 29(8),\n1072-1075.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7222033.2/\" rel=\"nofollow\"\u003eGFF utilities: GffRead and\nGffCompare\u003c/a\u003e(Pertea,\nG., \u0026amp; Pertea, M. (2020). GFF utilities: GffRead and GffCompare.\nF1000Research, 9.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://academic.oup.com/nar/article-abstract/36/suppl_1/D250/2506468\" rel=\"nofollow\"\u003eeggNOG: automated construction and annotation of orthologous groups\nof\ngenes\u003c/a\u003e(Jensen, L.\nJ., Julien, P., Kuhn, M., von Mering, C., Muller, J., Doerks, T., \u0026amp;\nBork, P. (2007). eggNOG: automated construction and annotation of\northologous groups of genes. Nucleic acids research, 36(suppl_1),\nD250-D254.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://www.nature.com/articles/nmeth.3176\" rel=\"nofollow\"\u003eFast and sensitive protein alignment using\nDIAMOND\u003c/a\u003e(Buchfink, B.,\nXie, C., \u0026amp; Huson, D. H. (2015). Fast and sensitive protein alignment\nusing DIAMOND. Nature methods, 12(1), 59-60.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://academic.oup.com/mbe/article-abstract/34/8/2115/3782716\" rel=\"nofollow\"\u003eFast genome-wide functional annotation through orthology assignment\nby\neggNOG-mapper\u003c/a\u003e(Huerta-Cepas,\nJ., Forslund, K., Coelho, L. P., Szklarczyk, D., Jensen, L. J., Von\nMering, C., \u0026amp; Bork, P. (2017). Fast genome-wide functional\nannotation through orthology assignment by eggNOG-mapper. Molecular\nbiology and evolution, 34(8), 2115-2122.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://www.nature.com/articles/s41587-019-0201-4\" rel=\"nofollow\"\u003eGraph-based genome alignment and genotyping with HISAT2 and\nHISAT-genotype\u003c/a\u003e(Kim,\nD., Paggi, J. M., Park, C., Bennett, C., \u0026amp; Salzberg, S. L. (2019).\nGraph-based genome alignment and genotyping with HISAT2 and\nHISAT-genotype. Nature biotechnology, 37(8), 907-915.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://academic.oup.com/bioinformatics/article-abstract/29/1/15/272537\" rel=\"nofollow\"\u003eSTAR: ultrafast universal RNA-seq\naligner\u003c/a\u003e(Dobin,\nA., Davis, C. A., Schlesinger, F., Drenkow, J., Zaleski, C., Jha,\nS.,  \u0026amp; Gingeras, T. R. (2013). STAR: ultrafast universal RNA-seq\naligner. Bioinformatics, 29(1), 15-21.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://academic.oup.com/bioinformatics/article-abstract/25/16/2078/204688\" rel=\"nofollow\"\u003eThe sequence alignment/map format and\nSAMtools\u003c/a\u003e(Li,\nH., Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., \n\u0026amp; Durbin, R. (2009). The sequence alignment/map format and SAMtools.\nBioinformatics, 25(16), 2078-2079.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://www.nature.com/articles/nbt.3122\" rel=\"nofollow\"\u003eStringTie enables improved reconstruction of a transcriptome from\nRNA-seq reads\u003c/a\u003e(Pertea, M.,\nPertea, G. M., Antonescu, C. M., Chang, T. C., Mendell, J. T., \u0026amp;\nSalzberg, S. L. (2015). StringTie enables improved reconstruction of\na transcriptome from RNA-seq reads. Nature biotechnology, 33(3),\n290-295.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://joss.theoj.org/papers/10.21105/joss.03050.pdf\" rel=\"nofollow\"\u003ekallisto: A command-line interface to simplify computational\nmodelling and the generation of atomic\nfeatures\u003c/a\u003e(Caldeweyher, E.\n(2021). kallisto: A command-line interface to simplify computational\nmodelling and the generation of atomic features. Journal of Open\nSource Software, 6(60), 3050.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://www.nature.com/articles/nbt.4163\" rel=\"nofollow\"\u003eUnited States Department of Energy Systems Biology\nKnowledgebase\u003c/a\u003e(Arkin AP,\nCottingham RW, Henry CS, Harris NL, Stevens RL, Maslov S, et\nal.KBase: The United States Department of Energy Systems Biology\nKnowledgebase. Nature Biotechnology. 2018;36: 566. doi:\n10.1038/nbt.4163)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.3711\" rel=\"nofollow\"\u003eKEGG Mapper for inferring cellular functions from protein\nsequences\u003c/a\u003e(Kanehisa,\nM., \u0026amp; Sato, Y. (2020). KEGG Mapper for inferring cellular functions\nfrom protein sequences. Protein Science, 29(1), 28-35.)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://www.sciencedirect.com/science/article/pii/S0968000408000236\" rel=\"nofollow\"\u003eiPath: interactive exploration of biochemical pathways and\nnetworks\u003c/a\u003e(Letunic,\nI., Yamada, T., Kanehisa, M., \u0026amp; Bork, P. (2008). iPath: interactive\nexploration of biochemical pathways and networks. Trends in\nbiochemical sciences, 33(3), 101-103.)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eBONUS Plataformas online para bioinformtica\u003c/h1\u003e\u003ca id=\"user-content-bonus-plataformas-online-para-bioinformtica\" class=\"anchor\" aria-label=\"Permalink: BONUS Plataformas online para bioinformtica\" href=\"#bonus-plataformas-online-para-bioinformtica\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eExisten diversas plataformas online que permiten realizar todos los\nprocesos aprendidos en este tutorial, usando servidores pblicos y\ninterfaces grficas.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ca href=\"https://usegalaxy.org/\" rel=\"nofollow\"\u003eGalaxy\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-galaxy\" class=\"anchor\" aria-label=\"Permalink: Galaxy\" href=\"#galaxy\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEs una plataforma con diversas herramientas para diferentes procesos. La\nmayoria que fueron usadas en este tutorial se encuentran en Galaxy.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/khidalgo85/training_transcriptomics/blob/master/imgs/galaxy.png\"\u003e\u003cimg src=\"/khidalgo85/training_transcriptomics/raw/master/imgs/galaxy.png\" align=\"center\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eHELP\u003c/strong\u003e Click en el smbolo del birrete, en la parte superior.\nEncontrar diversos tutoriales con videos.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ca href=\"https://www.kbase.us/\" rel=\"nofollow\"\u003eKBase\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-kbase\" class=\"anchor\" aria-label=\"Permalink: KBase\" href=\"#kbase\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEs una plataforma con diversas herramientas para diferentes procesos. La\nmayoria que fueron usadas en este tutorial se encuentran en KBase.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/khidalgo85/training_transcriptomics/blob/master/imgs/kbase.png\"\u003e\u003cimg src=\"/khidalgo85/training_transcriptomics/raw/master/imgs/kbase.png\" align=\"center\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://www.kbase.us/learn/\" rel=\"nofollow\"\u003e\u003cstrong\u003eHELP\u003c/strong\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ca href=\"https://www.genome.jp/tools/kofamkoala/\" rel=\"nofollow\"\u003eKofamKOALA\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-kofamkoala\" class=\"anchor\" aria-label=\"Permalink: KofamKOALA\" href=\"#kofamkoala\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePlataforma para anotar baseado en el algortmo HMMER. Tambin existen\notras plataformas baseadas en otros algortmos, como\n\u003ca href=\"https://www.kegg.jp/blastkoala\" rel=\"nofollow\"\u003eBlastKOALA\u003c/a\u003e,\n\u003ca href=\"https://www.kegg.jp/ghostkoala\" rel=\"nofollow\"\u003eGhostKOALA\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/khidalgo85/training_transcriptomics/blob/master/imgs/koala.png\"\u003e\u003cimg src=\"/khidalgo85/training_transcriptomics/raw/master/imgs/koala.png\" align=\"center\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ca href=\"https://pathways.embl.de/\" rel=\"nofollow\"\u003eIPATH3\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-ipath3\" class=\"anchor\" aria-label=\"Permalink: IPATH3\" href=\"#ipath3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHerramienta online para visualizacin y anlisis de mapas metablicos.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/khidalgo85/training_transcriptomics/blob/master/imgs/ipath.png\"\u003e\u003cimg src=\"/khidalgo85/training_transcriptomics/raw/master/imgs/ipath.png\" align=\"center\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://pathways.embl.de/video_tutorial.cgi\" rel=\"nofollow\"\u003e\u003cstrong\u003eVideos de ayuda\u003c/strong\u003e\u003c/a\u003e\n\u003ca href=\"https://pathways.embl.de/help.cgi\" rel=\"nofollow\"\u003e\u003cstrong\u003ePginas de ayuda\u003c/strong\u003e\u003c/a\u003e\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTrobleshooting\u003c/h2\u003e\u003ca id=\"user-content-trobleshooting\" class=\"anchor\" aria-label=\"Permalink: Trobleshooting\" href=\"#trobleshooting\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDudas, ayuda, consejos\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eKelly Hidalgo Martinez  \u003ca href=\"mailto:khidalgo@javeriana.edu.co\"\u003ekhidalgo@javeriana.edu.co\u003c/a\u003e\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"Pipeline: RNA-Seq ","anchor":"pipeline-rna-seq-","htmlText":"Pipeline: RNA-Seq "},{"level":2,"text":"SLIDES","anchor":"slides","htmlText":"SLIDES"},{"level":1,"text":"WORKFLOW","anchor":"workflow","htmlText":"WORKFLOW"},{"level":1,"text":"0. Introduccin a UNIX","anchor":"0-introduccin-a-unix","htmlText":"0. Introduccin a UNIX"},{"level":2,"text":"0.1. Login no servidor","anchor":"01-login-no-servidor","htmlText":"0.1. Login no servidor"},{"level":2,"text":"0.2 Comandos bsicos","anchor":"02-comandos-bsicos","htmlText":"0.2 Comandos bsicos"},{"level":3,"text":"ssh","anchor":"ssh","htmlText":"ssh"},{"level":3,"text":"pwd (Print Working Directory)","anchor":"pwd-print-working-directory","htmlText":"pwd (Print Working Directory)"},{"level":3,"text":"Sintaxis de los comandos","anchor":"sintaxis-de-los-comandos","htmlText":"Sintaxis de los comandos"},{"level":3,"text":"ls Listar","anchor":"ls-listar","htmlText":"ls Listar"},{"level":3,"text":"mkdir Make Dir","anchor":"mkdir-make-dir","htmlText":"mkdir Make Dir"},{"level":4,"text":"Tips","anchor":"tips","htmlText":"Tips"},{"level":3,"text":"cd Change Directory","anchor":"cd-change-directory","htmlText":"cd Change Directory"},{"level":4,"text":"Tip ","anchor":"tip-","htmlText":"Tip "},{"level":3,"text":"nano (editor de texto)","anchor":"nano-editor-de-texto","htmlText":"nano (editor de texto)"},{"level":3,"text":"mv move","anchor":"mv-move","htmlText":"mv move"},{"level":3,"text":"cp copy","anchor":"cp-copy","htmlText":"cp copy"},{"level":3,"text":"rm remove","anchor":"rm-remove","htmlText":"rm remove"},{"level":3,"text":"Otros comandos","anchor":"otros-comandos","htmlText":"Otros comandos"},{"level":3,"text":"less","anchor":"less","htmlText":"less"},{"level":3,"text":"head","anchor":"head","htmlText":"head"},{"level":3,"text":"tail","anchor":"tail","htmlText":"tail"},{"level":3,"text":"cat concatenate","anchor":"cat-concatenate","htmlText":"cat concatenate"},{"level":3,"text":"wc Word count","anchor":"wc-word-count","htmlText":"wc Word count"},{"level":3,"text":"grep","anchor":"grep","htmlText":"grep"},{"level":3,"text":"find","anchor":"find","htmlText":"find"},{"level":3,"text":"wget","anchor":"wget","htmlText":"wget"},{"level":3,"text":"gzip","anchor":"gzip","htmlText":"gzip"},{"level":3,"text":"Comandos tiles de Linux","anchor":"comandos-tiles-de-linux","htmlText":"Comandos tiles de Linux"},{"level":3,"text":"screen","anchor":"screen","htmlText":"screen"},{"level":3,"text":"Scripts simples de uma linha de comando","anchor":"scripts-simples-de-uma-linha-de-comando","htmlText":"Scripts simples de uma linha de comando"},{"level":3,"text":"scp Secure Copy Between Machines","anchor":"scp-secure-copy-between-machines","htmlText":"scp Secure Copy Between Machines"},{"level":3,"text":"Compresin e Descompresin de archivos","anchor":"compresin-e-descompresin-de-archivos","htmlText":"Compresin e Descompresin de archivos"},{"level":1,"text":"I. Gestin de Herramientas Bioinformticas","anchor":"i-gestin-de-herramientas-bioinformticas","htmlText":"I. Gestin de Herramientas Bioinformticas"},{"level":2,"text":"Anaconda","anchor":"anaconda","htmlText":"Anaconda"},{"level":2,"text":"Algunos comandos de conda","anchor":"algunos-comandos-de-conda","htmlText":"Algunos comandos de conda"},{"level":1,"text":"II. Adquisicin y organizacin de los datos","anchor":"ii-adquisicin-y-organizacin-de-los-datos","htmlText":"II. Adquisicin y organizacin de los datos"},{"level":2,"text":"2.1. Organizacin de directorios","anchor":"21-organizacin-de-directorios","htmlText":"2.1. Organizacin de directorios"},{"level":2,"text":"2.2 Adquisicin de datos","anchor":"22-adquisicin-de-datos","htmlText":"2.2 Adquisicin de datos"},{"level":3,"text":"2.2.1. SRA Toolkit","anchor":"221-sra-toolkit","htmlText":"2.2.1. SRA Toolkit"},{"level":3,"text":"2.2.2. Descargando las secuencias","anchor":"222-descargando-las-secuencias","htmlText":"2.2.2. Descargando las secuencias"},{"level":2,"text":"2.3. Organizacin de los datasets","anchor":"23-organizacin-de-los-datasets","htmlText":"2.3. Organizacin de los datasets"},{"level":1,"text":"III. Control de calidad","anchor":"iii-control-de-calidad","htmlText":"III. Control de calidad"},{"level":2,"text":"3.1. Control de Calidad","anchor":"31-control-de-calidad","htmlText":"3.1. Control de Calidad"},{"level":3,"text":"3.1.1. Evaluacin de la calidad","anchor":"311-evaluacin-de-la-calidad","htmlText":"3.1.1. Evaluacin de la calidad"},{"level":3,"text":"3.1.2. Depuracin/Trimming","anchor":"312-depuracintrimming","htmlText":"3.1.2. Depuracin/Trimming"},{"level":3,"text":"3.1.3. Cobertura","anchor":"313-cobertura","htmlText":"3.1.3. Cobertura"},{"level":1,"text":"IV. Genmica","anchor":"iv-genmica","htmlText":"IV. Genmica"},{"level":2,"text":"4.1 Ensamblaje de novo","anchor":"41-ensamblaje-de-novo","htmlText":"4.1 Ensamblaje de novo"},{"level":2,"text":"4.2 Evaluacin del ensamble","anchor":"42-evaluacin-del-ensamble","htmlText":"4.2 Evaluacin del ensamble"},{"level":3,"text":"4.2.1. Quast","anchor":"421-quast","htmlText":"4.2.1. Quast"},{"level":3,"text":"4.2.2. CheckM","anchor":"422-checkm","htmlText":"4.2.2. CheckM"},{"level":2,"text":"4.3. Anotacin Taxonmica","anchor":"43-anotacin-taxonmica","htmlText":"4.3. Anotacin Taxonmica"},{"level":2,"text":"4.4. Prediccin de genes","anchor":"44-prediccin-de-genes","htmlText":"4.4. Prediccin de genes"},{"level":1,"text":"V. Transcriptmica","anchor":"v-transcriptmica","htmlText":"V. Transcriptmica"},{"level":3,"text":"5.1. Remocin de secuencias de rRNA","anchor":"51-remocin-de-secuencias-de-rrna","htmlText":"5.1. Remocin de secuencias de rRNA"},{"level":3,"text":"5.2. Alineamiento a um genoma de referencia","anchor":"52-alineamiento-a-um-genoma-de-referencia","htmlText":"5.2. Alineamiento a um genoma de referencia"},{"level":4,"text":"5.2.1. Indexando el genoma de referencia","anchor":"521-indexando-el-genoma-de-referencia","htmlText":"5.2.1. Indexando el genoma de referencia"},{"level":4,"text":"5.2.2. Mapeo","anchor":"522-mapeo","htmlText":"5.2.2. Mapeo"},{"level":4,"text":"5.2.3. Transformando los archivos del mapeo","anchor":"523-transformando-los-archivos-del-mapeo","htmlText":"5.2.3. Transformando los archivos del mapeo"},{"level":3,"text":"5.3. Ensamble del transcriptoma","anchor":"53-ensamble-del-transcriptoma","htmlText":"5.3. Ensamble del transcriptoma"},{"level":3,"text":"5.3.1. Transcriptoma final","anchor":"531-transcriptoma-final","htmlText":"5.3.1. Transcriptoma final"},{"level":3,"text":"5.3.2. Extrayendo las secuencias","anchor":"532-extrayendo-las-secuencias","htmlText":"5.3.2. Extrayendo las secuencias"},{"level":2,"text":"5.4. Cuantificacin de la expresin","anchor":"54-cuantificacin-de-la-expresin","htmlText":"5.4. Cuantificacin de la expresin"},{"level":2,"text":"5.5. Expresin diferencial","anchor":"55-expresin-diferencial","htmlText":"5.5. Expresin diferencial"},{"level":1,"text":"Pausa para una brevisiiima introduccin al lenguaje R","anchor":"pausa-para-una-brevisiiima-introduccin-al-lenguaje-r","htmlText":"Pausa para una brevisiiima introduccin al lenguaje R"},{"level":2,"text":"1. Instalacin","anchor":"1-instalacin","htmlText":"1. Instalacin"},{"level":2,"text":"2. Instalacin, actualizacin y remocin de librerias","anchor":"2-instalacin-actualizacin-y-remocin-de-librerias","htmlText":"2. Instalacin, actualizacin y remocin de librerias"},{"level":2,"text":"3. Tipos de datos, constantes y variables","anchor":"3-tipos-de-datos-constantes-y-variables","htmlText":"3. Tipos de datos, constantes y variables"},{"level":2,"text":"4. Operadores y funciones matemticas","anchor":"4-operadores-y-funciones-matemticas","htmlText":"4. Operadores y funciones matemticas"},{"level":2,"text":"5. Estructura de datos","anchor":"5-estructura-de-datos","htmlText":"5. Estructura de datos"},{"level":3,"text":"5.1. Vectores","anchor":"51-vectores","htmlText":"5.1. Vectores"},{"level":3,"text":"5.2. Matrices","anchor":"52-matrices","htmlText":"5.2. Matrices"},{"level":3,"text":"5.3. Data Frames","anchor":"53-data-frames","htmlText":"5.3. Data Frames"},{"level":2,"text":"5.5. Expresin diferencial (continuacin)","anchor":"55-expresin-diferencial-continuacin","htmlText":"5.5. Expresin diferencial (continuacin)"},{"level":2,"text":"5.6. Anotacin Funcional","anchor":"56-anotacin-funcional","htmlText":"5.6. Anotacin Funcional"},{"level":1,"text":"RStudio","anchor":"rstudio","htmlText":"RStudio"},{"level":1,"text":"VI Referencias","anchor":"vi-referencias","htmlText":"VI Referencias"},{"level":1,"text":"BONUS Plataformas online para bioinformtica","anchor":"bonus-plataformas-online-para-bioinformtica","htmlText":"BONUS Plataformas online para bioinformtica"},{"level":3,"text":"Galaxy","anchor":"galaxy","htmlText":"Galaxy"},{"level":3,"text":"KBase","anchor":"kbase","htmlText":"KBase"},{"level":3,"text":"KofamKOALA","anchor":"kofamkoala","htmlText":"KofamKOALA"},{"level":3,"text":"IPATH3","anchor":"ipath3","htmlText":"IPATH3"},{"level":2,"text":"Trobleshooting","anchor":"trobleshooting","htmlText":"Trobleshooting"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fkhidalgo85%2Ftraining_transcriptomics"}}],"overviewFilesProcessingTime":173.534459}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="master branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        master
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/khidalgo85/training_transcriptomics/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/khidalgo85/training_transcriptomics/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/khidalgo85/training_transcriptomics/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/khidalgo85/training_transcriptomics/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/khidalgo85/training_transcriptomics/commits/master/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           7 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/khidalgo85/training_transcriptomics/commits/master/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="docs, (Directory)" class="Link--primary" href="/khidalgo85/training_transcriptomics/tree/master/docs" title="docs">
                          docs
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="docs, (Directory)" class="Link--primary" href="/khidalgo85/training_transcriptomics/tree/master/docs" title="docs">
                          docs
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="imgs, (Directory)" class="Link--primary" href="/khidalgo85/training_transcriptomics/tree/master/imgs" title="imgs">
                          imgs
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="imgs, (Directory)" class="Link--primary" href="/khidalgo85/training_transcriptomics/tree/master/imgs" title="imgs">
                          imgs
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.Rmd, (File)" class="Link--primary" href="/khidalgo85/training_transcriptomics/blob/master/README.Rmd" title="README.Rmd">
                          README.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.Rmd, (File)" class="Link--primary" href="/khidalgo85/training_transcriptomics/blob/master/README.Rmd" title="README.Rmd">
                          README.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/khidalgo85/training_transcriptomics/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/khidalgo85/training_transcriptomics/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="training_transcriptomics.Rproj, (File)" class="Link--primary" href="/khidalgo85/training_transcriptomics/blob/master/training_transcriptomics.Rproj" title="training_transcriptomics.Rproj">
                          training_transcriptomics.Rproj
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="training_transcriptomics.Rproj, (File)" class="Link--primary" href="/khidalgo85/training_transcriptomics/blob/master/training_transcriptomics.Rproj" title="training_transcriptomics.Rproj">
                          training_transcriptomics.Rproj
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <p dir="auto">
                     <a href="https://camo.githubusercontent.com/2efe01900e0b134a460b0d7da9510db33c484bdc19938f2499cf809364ffc01d/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f47454d4d2d4275696c64696e672d6f72616e6765" rel="noopener noreferrer nofollow" target="_blank">
                      <img alt="forthebadge" data-canonical-src="https://img.shields.io/badge/GEMM-Building-orange" src="https://camo.githubusercontent.com/2efe01900e0b134a460b0d7da9510db33c484bdc19938f2499cf809364ffc01d/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f47454d4d2d4275696c64696e672d6f72616e6765" style="max-width: 100%;"/>
                     </a>
                     <a href="https://camo.githubusercontent.com/c2f114c16ec5427dcf908a2ff32771ff5a3082006bba7b1531627e825aee6e7b/68747470733a2f2f666f7274686562616467652e636f6d2f696d616765732f6261646765732f6275696c742d776974682d736369656e63652e737667" rel="noopener noreferrer nofollow" target="_blank">
                      <img alt="forthebadge" data-canonical-src="https://forthebadge.com/images/badges/built-with-science.svg" src="https://camo.githubusercontent.com/c2f114c16ec5427dcf908a2ff32771ff5a3082006bba7b1531627e825aee6e7b/68747470733a2f2f666f7274686562616467652e636f6d2f696d616765732f6261646765732f6275696c742d776974682d736369656e63652e737667" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Pipeline: RNA-Seq
                      <a href="/khidalgo85/training_transcriptomics/blob/master/imgs/1.png" rel="noopener noreferrer" target="_blank">
                       <img align="right" src="/khidalgo85/training_transcriptomics/raw/master/imgs/1.png" style="max-width: 100%;" width="120px"/>
                      </a>
                     </h1>
                     <a aria-label="Permalink: Pipeline: RNA-Seq " class="anchor" href="#pipeline-rna-seq-" id="user-content-pipeline-rna-seq-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Autor: MsC. Kelly Hidalgo
                     </strong>
                    </p>
                    <p dir="auto">
                     Pipeline para el tratamiento de datos de RNA-seq y ensayos de expresin
diferencial. Para este tutorial sern usados los datos obtenidos en el
artculo
                     <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7980702/pdf/peerj-09-11081.pdf" rel="nofollow">
                      RNA-seq analysis of antibacterial mechanism of
                      <em>
                       Cinnamomum
camphora
                      </em>
                      essential oil against
                      <em>
                       Escherichia
coli
                      </em>
                     </a>
                     .
Brevemente, en este estudio fue evaluado el poder bactericida del aceite
esencial de la planta
                     <em>
                      Cinnamomum camphora
                     </em>
                     contra
                     <em>
                      Escherichia coli
                     </em>
                     ATCC8739. Fue determinada la concentracin mnima inhibitoria (MIC). Con
el fin de evaluar el mecanismo antibacterial del aceite esencial, fue
llevado a cabo un ensayo de expresin diferencial bajo tres
tratamientos, en los cuales fue secuenciado el transcriptoma (RNA-seq)
usando la plataforma Illumina NovaSeq. Las tres condiciones fueron,
                     <em>
                      E.
coli
                     </em>
                     tratada con agua, acetona y el aceite esencial en diferentes
concentraciones (1/8, 1/4 y 1/2 MIC).
                    </p>
                    <p dir="auto">
                     Los datos pueden ser descargados desde el
                     <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRP289443" rel="nofollow">
                      NCBI
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      <a href="/khidalgo85/training_transcriptomics/blob/master">
                       <strong>
                        SLIDES
                       </strong>
                      </a>
                     </h2>
                     <a aria-label="Permalink: SLIDES" class="anchor" href="#slides" id="user-content-slides">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      WORKFLOW
                     </h1>
                     <a aria-label="Permalink: WORKFLOW" class="anchor" href="#workflow" id="user-content-workflow">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     img src=imgs/workflow.png align=center/&gt;
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      0. Introduccin a UNIX
                     </h1>
                     <a aria-label="Permalink: 0. Introduccin a UNIX" class="anchor" href="#0-introduccin-a-unix" id="user-content-0-introduccin-a-unix">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     El shell de Unix es un interpretador de lnea de comando. Es una
herramenta poderosa que permite a los usuarios a ejecutar tareas
complejas y poderosas, generalmente con algunas lneas de cdigo. En
este tutorial, usted va entrar en la pantalla negra  , va a aprender
y ejercitar algunos comandos bsicos e indispensables, para navegar y se
desenvolver en el
                     <em>
                      terminal
                     </em>
                     .
                    </p>
                    <p dir="auto">
                     Normalmente la interaccin del humano con el computador sucede por medio
de un teclado y un mouse, interfaces grficas y/o sistemas de
reconocimiento de voz. La manera ms comm de interacturar, es llamada
como interface grfica de usurio
                     <em>
                      (
                      <strong>
                       G
                      </strong>
                      raphical
                      <strong>
                       U
                      </strong>
                      ser
                      <strong>
                       I
                      </strong>
                      nterface)
                     </em>
                     . Cuando se trabaja en una GUI, las ordenes son dadas
haciendo click con el mouse y usando interaciones orientadas por
diferentes mens. Eso funciona muy bien para escalas pequeas, pues es
muy intuitivo. Pero ahora imagine que usted necesita ejecutar una tarea
en mil archivos en diferentes carpetas, por ejemplo, copiar la ltima
lnea de todos los archivos y pegarlos en un nico archivo. Seran
necesarias muchas horas (tal vez das) ejecutando el proceso y adems
podra cometer errores. Es ah cuando shell es muy til, ya que por ser
una interface de lnea de comando
                     <em>
                      (
                      <strong>
                       C
                      </strong>
                      ommand
                      <strong>
                       L
                      </strong>
                      ine
                      <strong>
                       I
                      </strong>
                      nterface)
                     </em>
                     y un lenguaje de srcipt, permite procesar tareas repetitivas como las
del ejemplo, siendo realizadas de una forma rpida y automtica. El uso
de shell es fundamental para el uso de uma amplia variedad de
herramientas  de bioinformtica . Este tutorial le va a servir para
hacer un uso eficaz de estos recursos.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      0.1. Login no servidor
                     </h2>
                     <a aria-label="Permalink: 0.1. Login no servidor" class="anchor" href="#01-login-no-servidor" id="user-content-01-login-no-servidor">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Para los usuarios de Windows es necesario la instalacin del software
                     <a href="https://mobaxterm.mobatek.net/download-home-edition.html#:~:text=MobaXterm%20Home%20Edition%20v22.1%0A(Installer%20edition)" rel="nofollow">
                      MobaXterm
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     <a href="/khidalgo85/training_transcriptomics/blob/master/imgs/SSH_MobaXterm.png" rel="noopener noreferrer" target="_blank">
                      <img align="center" src="/khidalgo85/training_transcriptomics/raw/master/imgs/SSH_MobaXterm.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Despus de instalado, clique en la opcin
                     <em>
                      session
                     </em>
                     , escoja el icono
                     <em>
                      ssh
                     </em>
                     y complete los datos del login:
                     <em>
                      remote host
                     </em>
                     (direccin IP).
Despus seleccione la opcin
                     <em>
                      specify name
                     </em>
                     y complete con el nombre del
usuario. Por ltimo ser solicitada la contrasenha. Puede salvar los
datos del login.
                    </p>
                    <p dir="auto">
                     <strong>
                      Nota
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      Siempre que ud vea una caja como sta , es para que usted dgite
el contenido en la lnea de comando y presione [enter]  para
                      <strong>
                       correr
                      </strong>
                      el comando.
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="ls" dir="auto">
                     <pre>ls</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      0.2 Comandos bsicos
                     </h2>
                     <a aria-label="Permalink: 0.2 Comandos bsicos" class="anchor" href="#02-comandos-bsicos" id="user-content-02-comandos-bsicos">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       ssh
                      </code>
                     </h3>
                     <a aria-label="Permalink: ssh" class="anchor" href="#ssh" id="user-content-ssh">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <code>
                      ssh
                     </code>
                     (Secure Shell) es un protocolo que garantiza que el usuario y el
servidor remoto intercambien informaciones de manera segura e dinmica.
Sirve especificamente para conectarse a un servidor remoto.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="$ ssh -X &lt;user_name&gt;@&lt;ip.address&gt;">
                     <pre class="notranslate"><code>$ ssh -X &lt;user_name&gt;@&lt;ip.address&gt;
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       pwd
                      </code>
                      (
                      <em>
                       Print Working Directory
                      </em>
                      )
                     </h3>
                     <a aria-label="Permalink: pwd (Print Working Directory)" class="anchor" href="#pwd-print-working-directory" id="user-content-pwd-print-working-directory">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Los directorios  son como cajas. Siempre que usted est usando el
shell, usted est dentro de alguna caja
                     <strong>
                      (Directorio)
                     </strong>
                      de su
computador  o servidor, llamado
                     <strong>
                      directorio actual de trabajo
                     </strong>
                     . Los
comandos solamente leen y graban archivos en el directorio actual de
trabajo (s usted no indicar otro camino), por lo tanto, es
                     <strong>
                      importante
                     </strong>
                      saber en cual directorio est antes de
ejecutar un comando. El comando
                     <code>
                      pwd
                     </code>
                     muestra donde est.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="pwd">
                     <pre class="notranslate"><code>pwd
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" /home/user">
                     <pre class="notranslate"><code> /home/user
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Sintaxis de los comandos
                     </h3>
                     <a aria-label="Permalink: Sintaxis de los comandos" class="anchor" href="#sintaxis-de-los-comandos" id="user-content-sintaxis-de-los-comandos">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <code>
                      comando [opciones] [archivo]
                     </code>
                    </p>
                    <p dir="auto">
                     El comando es separado de las opciones (o argumentos, flags) e do
archivo  por un espacio. Los argumentos pueden mudar el comportamiento
del comando. Y el archivo  indica para el comando sobre lo que va a
operar (p.e. archivos  y directorios ). A veces los argumentos y el
archivo  son llamados de
                     <strong>
                      parmetros
                     </strong>
                     . Un comando puede tener ms de
un argumento y/o archivos  y tambin podria no tener ninguno de los
dos. Las opciones usualmente tienen un guin y una letra (p.e.
                     <code>
                      -h
                     </code>
                     ) o
dos guiones y una palabra (p.e.
                     <code>
                      --help
                     </code>
                     ),
                     <strong>
                      sin
                     </strong>
                     espacio entre lo(s)
guion(es) y la letra/palabra. Vamos a ver con ejemplos prcticos.
                    </p>
                    <p dir="auto">
                     <strong>
                      Importante:
                     </strong>
                      el lenguaje usado en UNIX es sensible a
letras maysculas e minsculas (
                     <em>
                      case sensitive
                     </em>
                     ), es un error comn.
                     <strong>
                      Siempre atento
                     </strong>
                     
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       ls
                      </code>
                      Listar
                     </h3>
                     <a aria-label="Permalink: ls Listar" class="anchor" href="#ls-listar" id="user-content-ls-listar">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Con el comando
                     <code>
                      ls
                     </code>
                     ud puede ver (listar) lo que hay dentro del
                     <strong>
                      directorio actual de trabajo
                     </strong>
                     .
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="ls ">
                     <pre class="notranslate"><code>ls 
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Documentos Downloads Imagens">
                     <pre class="notranslate"><code>Documentos Downloads Imagens
</code></pre>
                    </div>
                    <p dir="auto">
                     Usted puede usar el
                     <strong>
                      argumento
                     </strong>
                     <code>
                      -F
                     </code>
                     para indicar para el comando
                     <code>
                      ls
                     </code>
                     que muestre lo que es cada elemento (directorio o archivo).
                     <code>
                      /
                     </code>
                     significa que es un directorio , el
                     <code>
                      *
                     </code>
                     significa que es un archivo
ejecutable y s no tiene ningn smbolo significa que es un archivo .
                    </p>
                    <p dir="auto">
                     El
                     <em>
                      <strong>
                       flag
                      </strong>
                     </em>
                     
                     <code>
                      --help
                     </code>
                      es bien importante:exclamation:, y puede ser
usado en
                     <strong>
                      cualquier
                     </strong>
                     comando. El muestra ms informacin sobre el
comando, y como usarlo . Explore o men help
                    </p>
                    <p dir="auto">
                     Outros
                     <strong>
                      flags
                     </strong>
                      interessantes so:
                     <code>
                      ls -l
                     </code>
                     , que lista o contedo da
pasta  com informaes extras, como as permises, o tamanho, a data  e
hora  de criao, e o nome de cada um dos elementos.
                     <code>
                      ls -a
                     </code>
                     que lista
todos os arquivos includos os ocultos.
                     <code>
                      ls -t
                     </code>
                     lista os arquivos em
ordem cronolgica.
                    </p>
                    <p dir="auto">
                     Otros
                     <strong>
                      flags
                     </strong>
                      interesantes son:
                     <code>
                      ls -l
                     </code>
                     , que lista el contenido de
la carpeta  con informaciones extras, como los permisos, el tamao, la
fecha  y hora  de creacin, y el nombre de cada uno de los
elementos.
                     <code>
                      ls -a
                     </code>
                     que lista todos los archivos includos los ocultos.
                     <code>
                      ls -t
                     </code>
                     lista los archivos en orden cronolgica.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="ls -l">
                     <pre class="notranslate"><code>ls -l
</code></pre>
                    </div>
                    <p dir="auto">
                     Los permisos se deben entender as: d = Directorio rwx =
                     <em>
                      read, write
                     </em>
                     y
                     <em>
                      execute
                     </em>
                    </p>
                    <p dir="auto">
                     Se deben leer de tres en tres caracteres. As, en nuestro ejemplo de
arriba, ninguno de los elementos son directorios porque no inician con
la letra d.La primera tripleta tiene las letras r e w, o sea el usuario
puede leer y escribir esos elementos. La segunda tiene solamente la
letra r, o sea el grupo (pueden ser creados grupos de usuarios) solo
pueden leer esos archivos. Por ltimo, la ltima tripleta, tambin solo
tiene a letra r, o sea
                     <strong>
                      todos
                     </strong>
                     los usuarios solo pueden leer los
elementos listados.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       mkdir
                      </code>
                      Make Dir
                     </h3>
                     <a aria-label="Permalink: mkdir Make Dir" class="anchor" href="#mkdir-make-dir" id="user-content-mkdir-make-dir">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Usted ya aprendi a explorar carpetas  y archivos , ahora va a
aprender como se crean. El comando
                     <code>
                      mkdir
                     </code>
                     sirve para crear carpetas .
Vamos a crear varias 
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Crea una carpeta llamada datos
mkdir datos
# Crea una carpeta llamada analisis y otra llamada tutorial
mkdir analisis tutorial
# Lista el contenido 
ls -F">
                     <pre class="notranslate"><code># Crea una carpeta llamada datos
mkdir datos
# Crea una carpeta llamada analisis y otra llamada tutorial
mkdir analisis tutorial
# Lista el contenido 
ls -F
</code></pre>
                    </div>
                    <p dir="auto">
                     Note que: 1) puede crear ms de una carpeta por lnea de comando y 2)
puede diferenciar los archivos de los directorios, pues stos estn de
color azul y con una
                     <code>
                      /
                     </code>
                     al final.
                    </p>
                    <p dir="auto">
                     Usando el comando
                     <code>
                      tree
                     </code>
                     puede ver graficamente la organizacin de los
directorios y archivos dentro del directorio actual.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      <em>
                       Tips
                      </em>
                     </h4>
                     <a aria-label="Permalink: Tips" class="anchor" href="#tips" id="user-content-tips">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       No  use espacios en los nombres de sus carpetas  o archivos 
(p.e.
                       <del>
                        coleta 2020
                       </del>
                       ). Siempre separe las palabras con
                       <code>
                        -
                       </code>
                       ,
                       <code>
                        _
                       </code>
                       o
con maysculas (p.e.
                       <code>
                        coleta_2020
                       </code>
                       ,
                       <code>
                        coleta-2020
                       </code>
                       o
                       <code>
                        coletaMaio
                       </code>
                       )
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       No  comience nombres con
                       <code>
                        -
                       </code>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Asigne para sus elementos nombres fciles de recordar y escribir y
que describan lo que contienen.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       No use caracteres espaciais
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       cd
                      </code>
                      Change Directory
                     </h3>
                     <a aria-label="Permalink: cd Change Directory" class="anchor" href="#cd-change-directory" id="user-content-cd-change-directory">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     El comando
                     <code>
                      cd
                     </code>
                     sirve para cambiar de
                     <strong>
                      directorio actual de trabajo
                     </strong>
                     . Vaya al directorio
                     <code>
                      datos/
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd datos/">
                     <pre class="notranslate"><code>cd datos/
</code></pre>
                    </div>
                    <p dir="auto">
                     Condirme donde est con el comando
                     <code>
                      pwd
                     </code>
                    </p>
                    <p dir="auto">
                     Para volver al directorio anterior
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd ..">
                     <pre class="notranslate"><code>cd ..
</code></pre>
                    </div>
                    <p dir="auto">
                     El
                     <code>
                      ..
                     </code>
                     significa: directorio que contiene el directorio actual (o sea
un directorio arriba). Confirme con:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="pwd">
                     <pre class="notranslate"><code>pwd
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/user">
                     <pre class="notranslate"><code>/home/user
</code></pre>
                    </div>
                    <p dir="auto">
                     Ahora entre en la carpeta 
                     <code>
                      analisis/
                     </code>
                     y cree otra  llamada
                     <code>
                      colecta_2020
                     </code>
                     , confirme con
                     <code>
                      ls
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Cambie de directorio
cd analisis/
## Cree un nuevo directorio
mkdir colecta_2020
## Confirme
ls">
                     <pre class="notranslate"><code>## Cambie de directorio
cd analisis/
## Cree un nuevo directorio
mkdir colecta_2020
## Confirme
ls
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="colecta_2020/">
                     <pre class="notranslate"><code>colecta_2020/
</code></pre>
                    </div>
                    <p dir="auto">
                     Ahora entre a la carpeta que acab de crear usando el comando
                     <code>
                      cd
                     </code>
                     y
verifique donde est con
                     <code>
                      pwd
                     </code>
                     , despus regrese para el directorio base
                     <code>
                      /home/user
                     </code>
                      y verifique nuevamente.
                    </p>
                    <p dir="auto">
                     Al finalizar esos comandos verifique la organizacin de los directorios
usando el comando
                     <code>
                      tree
                     </code>
                    </p>
                    <p dir="auto">
                     Por ltimo vuelva a la carpeta
                     <code>
                      home/user/analisis/coleta_2020
                     </code>
                     usando
                     <code>
                      cd
                     </code>
                     en una lnea de comando solamente. Confirme con
                     <code>
                      pwd
                     </code>
                     . Vuelva
nuevamente para la carpeta raz tambin con solo una lnea de comando.
                     <strong>
                      Pista
                     </strong>
                     , recuerde que
                     <code>
                      ..
                     </code>
                     significa directorio de arriba.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      <em>
                       Tip
                      </em>
                      
                     </h4>
                     <a aria-label="Permalink: Tip " class="anchor" href="#tip-" id="user-content-tip-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Usted puede usar la tecla Tab  para autocompletar las
palabras. As, economiza tiempo , y evita errores de digitacin ,
porque el sistema solo va a completar los nombres que existan en el
                     <strong>
                      directorio actual de trabajo
                     </strong>
                     .
                    </p>
                    <p dir="auto">
                     Solo tiene que esrcibir las primeras letras de la palabra. p.e. col:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Cambie de carpeta
cd analisis/col">
                     <pre class="notranslate"><code>## Cambie de carpeta
cd analisis/col
</code></pre>
                    </div>
                    <p dir="auto">
                     Oprima  [Tab]. Automticamente se no existe ningn otro
elemento que comience por col, la palabra coleta_2020 va a ser
autocompletada.
                    </p>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd analises/colecta_2020">
                     <pre class="notranslate" lang="coffeesrcipt="><code>cd analises/colecta_2020
</code></pre>
                    </div>
                    <p dir="auto">
                     S por alguna razn existe otro elemento que tambin comience con col,
oprima dos vezes Tab  y el sistema va a mostrar las opciones
de palabras que inicien con col.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Estando en analisis/ cree uma pasta llamada colecta_2019
mkdir colecta_2019">
                     <pre class="notranslate"><code>## Estando en analisis/ cree uma pasta llamada colecta_2019
mkdir colecta_2019
</code></pre>
                    </div>
                    <p dir="auto">
                     Ahora, usted quiere entrar en esa carpeta nueva. Use el comando
                     <code>
                      cd
                     </code>
                     para cambiar de carpeta y use [Tab] [Tab] para que el sistema
muestre las opciones con comienzo col.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd col">
                     <pre class="notranslate"><code>cd col
</code></pre>
                    </div>
                    <p dir="auto">
                     [Tab] [Tab]
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       nano
                      </code>
                      (editor de texto)
                     </h3>
                     <a aria-label="Permalink: nano (editor de texto)" class="anchor" href="#nano-editor-de-texto" id="user-content-nano-editor-de-texto">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                     <code>
                      nano &lt;nombre_del_archivo&gt;
                     </code>
                    </p>
                    <p dir="auto">
                     Ahora ud va a crear un archivo 
                     <code>
                      test.txt
                     </code>
                     dentro de la carpeta 
                     <code>
                      tutorial/
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Cambie de directorio
cd tutorial/
## Abra el editor de texto nano 
nano test.txt">
                     <pre class="notranslate"><code>## Cambie de directorio
cd tutorial/
## Abra el editor de texto nano 
nano test.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Cuando abrir el editor de texto, escriba: Este es un test y cierre el
archivo con [Ctrl + o] para salvar. En la lnea blanca abajo, el
editor de texto preguntar s quiere mantener el nombre que le dio al
comienzo
                     <code>
                      test.txt
                     </code>
                     . [Enter] para confirmar. [Ctrl + x] para salir.
Confirme que el archivo fue creado con el comando
                     <code>
                      ls
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     S ud quiere entrar de nuevo al arquivo y modificarlo, deve usar el de
nuevo el comando
                     <code>
                      nano test.txt
                     </code>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       mv
                      </code>
                      move
                     </h3>
                     <a aria-label="Permalink: mv move" class="anchor" href="#mv-move" id="user-content-mv-move">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     El comando
                     <code>
                      mv
                     </code>
                     sirve para mover Archivos  de una carpeta  a otra .
Adems este comando tambin puede ser usado para cambiar los nombres de
los elementos. Para mover un archivo de una carpeta  a otra  la
                     <strong>
                      sintaxis
                     </strong>
                     del comando es:
                     <code>
                      mv archivo.txt nuevodirectorio/
                     </code>
                     . En
nuestro ejemplo:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="mv test.txt ../datos/
## Confirme
ls ../datos/">
                     <pre class="notranslate"><code>mv test.txt ../datos/
## Confirme
ls ../datos/
</code></pre>
                    </div>
                    <p dir="auto">
                     Ud uso
                     <code>
                      ../
                     </code>
                     , porque ud estaba dentro de la carpeta 
                     <code>
                      /home/user/tutorial/
                     </code>
                     y necesitaba volver para 
                     <code>
                      /home/user/
                     </code>
                     (
carpeta arriba de
                     <code>
                      tutorial/
                     </code>
                     ) para continuar el camino para 
                     <code>
                      datos/
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     Ahora use el comando
                     <code>
                      mv
                     </code>
                     para cambiar el nombre del archivo 
                     <code>
                      test.txt
                     </code>
                     por
                     <code>
                      prueba.txt
                     </code>
                     .
                     <strong>
                      Sintaxis
                     </strong>
                     <code>
                      mv nombredelarchivo.txt nuevonombredelarchivo.txt
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Cambie de directorio
cd ../datos/
## Confirme
ls">
                     <pre class="notranslate"><code>## Cambie de directorio
cd ../datos/
## Confirme
ls
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="test.txt

## Cambie el nombre del archivo
mv test.txt prueba.txt
## Confirme
ls">
                     <pre class="notranslate"><code>test.txt

## Cambie el nombre del archivo
mv test.txt prueba.txt
## Confirme
ls
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="prueba.txt">
                     <pre class="notranslate"><code>prueba.txt
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       cp
                      </code>
                      copy
                     </h3>
                     <a aria-label="Permalink: cp copy" class="anchor" href="#cp-copy" id="user-content-cp-copy">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     El comando
                     <code>
                      cp
                     </code>
                     es similar a
                     <code>
                      mv
                     </code>
                     , pero l copia el archivo  en vez de
moverlo. Ahora copie el archivo
                     <code>
                      /home/user/datos/prueba.txt
                     </code>
                     en la
carpeta
                     <code>
                      /home/user/analisis/
                     </code>
                     .
                     <strong>
                      ATENCIN:exclamation:! Haga eso desde
el directorio inicial
                     </strong>
                     <code>
                      /home/user/
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                     <code>
                      cp directorio/nombredelarchivo.txt nuevodirectorio/nombredelarchivo.txt
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Dnde estoy?
pwd">
                     <pre class="notranslate"><code>## Dnde estoy?
pwd
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/user/datos/">
                     <pre class="notranslate"><code>/home/user/datos/
</code></pre>
                    </div>
                    <p dir="auto">
                     No olvide del
                     <strong>
                      tip
                     </strong>
                      de usar [Tab]
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Copiar el archivo
cp datos/prueba.txt analisis/
## Confirme
ls datos/">
                     <pre class="notranslate"><code>## Copiar el archivo
cp datos/prueba.txt analisis/
## Confirme
ls datos/
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="prueba.txt

## Listar
ls analisis/">
                     <pre class="notranslate"><code>prueba.txt

## Listar
ls analisis/
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="colecta_2019   colecta_2020   prueba.txt">
                     <pre class="notranslate"><code>colecta_2019   colecta_2020   prueba.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Ud puede usar el comando
                     <code>
                      cp
                     </code>
                     para copiar varios archivos  en una
lnea de comando solamente.
                    </p>
                    <p dir="auto">
                     <strong>
                      Sintaxe
                     </strong>
                     <code>
                      cp archivo1.txt archivo2.txt archivo3.txt carpetadedestino/
                     </code>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       rm
                      </code>
                      remove
                     </h3>
                     <a aria-label="Permalink: rm remove" class="anchor" href="#rm-remove" id="user-content-rm-remove">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Con o comando
                     <code>
                      rm
                     </code>
                     ud puede remover archivos  y/o carpetas .
                     <strong>
                      CUIDADO
                     </strong>
                     
                     <strong>
                      PRECAUCIN
                     </strong>
                      este comando no
tiene reversa, una vez ud oprima [enter] no hay como recuperar el
archivo  o carpeta , entonces revise y piense bien antes de rodar este
comando.
                    </p>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                    </p>
                    <p dir="auto">
                     <code>
                      rm directorio/nombredelarchivo.txt
                     </code>
                    </p>
                    <p dir="auto">
                     Ud va a remover el archivo
                     <code>
                      prueba.txt
                     </code>
                     de la carpeta 
                     <code>
                      /home/user/datos
                     </code>
                     . S necesita, use el comando
                     <code>
                      pwd
                     </code>
                     para confirmar en
que directorio est.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Remover desde /home/user/
rm dados/prova.txt
## Confirme
ls">
                     <pre class="notranslate"><code>## Remover desde /home/user/
rm dados/prova.txt
## Confirme
ls
</code></pre>
                    </div>
                    <p dir="auto">
                     Para eliminar una carpeta  ud necesita del
                     <strong>
                      flag
                     </strong>
                     
                     <code>
                      -r
                     </code>
                     . Elimine la
carpeta 
                     <code>
                      datos/
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Remover el directorio
rm -r datos/">
                     <pre class="notranslate"><code>## Remover el directorio
rm -r datos/
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Otros comandos
                     </h3>
                     <a aria-label="Permalink: Otros comandos" class="anchor" href="#otros-comandos" id="user-content-otros-comandos">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Para los siguientes comando vamos a crear dos nuevos archivos de texto
llamados
                     <code>
                      bssA_1.txt
                     </code>
                     y
                     <code>
                      bssA_2.txt
                     </code>
                     , en cada uno vamos a pegar una
secuencia del gen
                     <em>
                      bssA
                     </em>
                     que codifica para la enzima
                     <em>
                      Benzylsuccinate
synthase
                     </em>
                     .
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Cambiar de directorio
cd tutorial/
##Abrir el editor de texto nano
nano bssA_1.txt">
                     <pre class="notranslate"><code>## Cambiar de directorio
cd tutorial/
##Abrir el editor de texto nano
nano bssA_1.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Copie a sequncia
                     <a href="https://www.ncbi.nlm.nih.gov/nuccore/MW762608.1?report=fasta" rel="nofollow">
                      aqui
                     </a>
                     .
Copie la secuencia desde
                     <a href="https://www.ncbi.nlm.nih.gov/nuccore/MW762608.1?report=fasta" rel="nofollow">
                      aqui
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Atencin:
                     </strong>
                     El comando de teclas  [Ctrl + V] no funciona
en el terminal de Linux. Use [Ctrl + Shift + V].
                    </p>
                    <p dir="auto">
                     Salve y cierre el editor. S quiere, puede confirmar que el archivo fue
creado con el comando
                     <code>
                      ls
                     </code>
                     , y entrando en el archivo con el comando
                     <code>
                      nano
                     </code>
                     e el nombre del archivo.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Abrir el editor de texto nano
nano bssA_1.txt">
                     <pre class="notranslate"><code>## Abrir el editor de texto nano
nano bssA_1.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Repita el proceso para crear el archivo
                     <code>
                      bssA_2.txt
                     </code>
                     copiando sta
                     <a href="https://www.ncbi.nlm.nih.gov/nuccore/FJ810633.1?report=fasta" rel="nofollow">
                      secuencia
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       less
                      </code>
                     </h3>
                     <a aria-label="Permalink: less" class="anchor" href="#less" id="user-content-less">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Este comando serve para imprimir en la pantalla el contenido de un
archivo. Para salir digite
                     <code>
                      q
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="less bssA_1.txt">
                     <pre class="notranslate"><code>less bssA_1.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Para salir [Ctrl + c]
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       head
                      </code>
                     </h3>
                     <a aria-label="Permalink: head" class="anchor" href="#head" id="user-content-head">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Muestra las primeras 10 lineas del archivo
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="head bssA_2.txt">
                     <pre class="notranslate"><code>head bssA_2.txt
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       tail
                      </code>
                     </h3>
                     <a aria-label="Permalink: tail" class="anchor" href="#tail" id="user-content-tail">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Muestra las ltimas 10 lineas del archivo
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="tail bssA_2.txt">
                     <pre class="notranslate"><code>tail bssA_2.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     S ud quiere, puede aumentar el nmero de lneas que esos dos ltimos
comandos muestran, adicionado un argumento con el nmero de lneas que
desee imprimir en la pantalla.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## ltimas 12 linhas
tail -12 bssA_1.txt
## Primeras 13 linhas
head -13 bssA_2.txt">
                     <pre class="notranslate"><code>## ltimas 12 linhas
tail -12 bssA_1.txt
## Primeras 13 linhas
head -13 bssA_2.txt
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       cat
                      </code>
                      concatenate
                     </h3>
                     <a aria-label="Permalink: cat concatenate" class="anchor" href="#cat-concatenate" id="user-content-cat-concatenate">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Este comando sirve para juntar dos archivos en uno. Es muy til para
juntar archivos tipo
                     <code>
                      .fasta
                     </code>
                     con secuencias.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Concatenar
cat bssA_1.txt bssA_2.txt &gt; bssA_all.txt
## Confirme
ls">
                     <pre class="notranslate"><code>## Concatenar
cat bssA_1.txt bssA_2.txt &gt; bssA_all.txt
## Confirme
ls
</code></pre>
                    </div>
                    <p dir="auto">
                     En el ejemplo anterior, fueron concatenados los archivos
                     <code>
                      bssA_1.txt
                     </code>
                     y
                     <code>
                      bssA_2.txt
                     </code>
                     dentro del archivo
                     <code>
                      bss_all.txt
                     </code>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       wc
                      </code>
                      Word count
                     </h3>
                     <a aria-label="Permalink: wc Word count" class="anchor" href="#wc-word-count" id="user-content-wc-word-count">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Este comando sirve para contar las lneas, palabras o caracteres de los
archivos.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Contar lineas, palabras y caracteres
wc bssA_1.txt">
                     <pre class="notranslate"><code>## Contar lineas, palabras y caracteres
wc bssA_1.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="28        36      1907 bssA_1.txt">
                     <pre class="notranslate"><code>28        36      1907 bssA_1.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     El archivo 
                     <code>
                      bssA_1.txt
                     </code>
                     tiene 28 lineas, 36 palabras y 1907
caracteres.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       grep
                      </code>
                     </h3>
                     <a aria-label="Permalink: grep" class="anchor" href="#grep" id="user-content-grep">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Com o
                     <code>
                      grep
                     </code>
                     ud puede buscar un patrn dentro de un archivo. Por ejemplo
en un archivo de secuencias
                     <code>
                      .fasta
                     </code>
                     cada secuencia comienza con el
simbolo
                     <code>
                      &gt;
                     </code>
                     o podria buscar una secuencia de nucletidos especfica
(p.e. ATCTTGCA)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="grep -c '&gt;' bssA_2.txt

grep -c 'CGA' bssA_1.txt">
                     <pre class="notranslate"><code>grep -c '&gt;' bssA_2.txt

grep -c 'CGA' bssA_1.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     El
                     <em>
                      flag
                     </em>
                     <code>
                      -c
                     </code>
                     es para que el comando solo muestre el nmero de lneas
que hacen
                     <em>
                      match
                     </em>
                     con lo que est siendo procurado.
                    </p>
                    <p dir="auto">
                     El comando
                     <code>
                      grep
                     </code>
                     tiene varios
                     <em>
                      flags
                     </em>
                     diferentes, recuerde que puede
conocer todos ellos digitando
                     <code>
                      grep --help
                     </code>
                     para entrar en el men de
ayuda del comando.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       find
                      </code>
                     </h3>
                     <a aria-label="Permalink: find" class="anchor" href="#find" id="user-content-find">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Con el comando
                     <code>
                      find
                     </code>
                     ud puede buscar archivos con una palabra clave.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Buscar
find . -name '*.txt'">
                     <pre class="notranslate"><code>## Buscar
find . -name '*.txt'
</code></pre>
                    </div>
                    <p dir="auto">
                     Lea el comando as: buscar en el `
                     <code>
                      directorio actual de trabajo
                     </code>
                     cualquier archivo que termine con
                     <code>
                      *.txt
                     </code>
                     . El simbolo
                     <code>
                      *
                     </code>
                     significa
cualquier caracter. S ud escribe `bss*, el sistema va a entender que
ud est interesado en cualquier elemente que comience con bss.
                    </p>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="./bssA_1.txt
./bssA_2.txt
./bssA_all.txt">
                     <pre class="notranslate"><code>./bssA_1.txt
./bssA_2.txt
./bssA_all.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Ud podria buscar la palabra en cualquier directorio  del pc modificando
el comando. Procure todos os archivos terminados en .txt en la carpeta
raz
                     <code>
                      /home/user/
                     </code>
                     .
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## Procurar
find ../ -name "*.txt"'>
                     <pre class="notranslate"><code>## Procurar
find ../ -name "*.txt"
</code></pre>
                    </div>
                    <p dir="auto">
                     Outros exemplos:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="find ../datos/ -name 'prueba*' # busca archivos que comienzan con prueba dentro de la carpeta /home/user/datos/
find ../datos/ -iname 'prueba*' #igual pero ignora s son maysculas o minusclas.">
                     <pre class="notranslate"><code>find ../datos/ -name 'prueba*' # busca archivos que comienzan con prueba dentro de la carpeta /home/user/datos/
find ../datos/ -iname 'prueba*' #igual pero ignora s son maysculas o minusclas.
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       wget
                      </code>
                     </h3>
                     <a aria-label="Permalink: wget" class="anchor" href="#wget" id="user-content-wget">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     El comando
                     <code>
                      wget
                     </code>
                     sirve para hacer
                     <em>
                      download
                     </em>
                     de archivos en la web y
almacenarlos en el
                     <strong>
                      directorio actual de trabajo
                     </strong>
                     , es muy til para
descargar bases de datos.
                    </p>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="wget https://direccionoweb.com">
                     <pre class="notranslate"><code>wget https://direccionoweb.com
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       gzip
                      </code>
                     </h3>
                     <a aria-label="Permalink: gzip" class="anchor" href="#gzip" id="user-content-gzip">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Este comando es para compactar y descompactar archivos .
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Comprimir
gzip tutorial/*">
                     <pre class="notranslate"><code>## Comprimir
gzip tutorial/*
</code></pre>
                    </div>
                    <p dir="auto">
                     As, el comando
                     <code>
                      gzip
                     </code>
                     compact todos los archivos que estn dentro de
la carpeta
                     <code>
                      tutorial/
                     </code>
                     . Use el comando
                     <code>
                      ls
                     </code>
                     para observar la nueva
extensin de los archivos.
                    </p>
                    <p dir="auto">
                     Para descompactar el comando es:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Descompactar
gzip -d tutorial/bssA_all.txt.gz">
                     <pre class="notranslate"><code>## Descompactar
gzip -d tutorial/bssA_all.txt.gz
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Comandos tiles de Linux
                     </h3>
                     <a aria-label="Permalink: Comandos tiles de Linux" class="anchor" href="#comandos-tiles-de-linux" id="user-content-comandos-tiles-de-linux">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="df    # Muestra el espacio disponible en el disco
free -g     # info de la memoria
uname -a    # Muestra la informacin de la mquina
du -sh    # muestra el espacio en uso en el disco
du sh *     # mueestra el espacio usado en disco por archivos y/o directorios 
du -s * | sort -nr    # Muestra el espacio usado en el disco por archivos y/o directorios ordenados por tamao
top     # Muestra el top de consumidores de memoria y CPU 
who     # Muestra quien est logado en el sistema
ps    # Muestra los procesos ejecutados por el usuario
ps -e     # Muestra todos los procesos en ejecin en el sistema
ps -o %t -p &lt;pid&gt;     # Muestra cuanto tiempo lleva ejecutando un determinado proceso (pid)
kill &lt;pid&gt;    # Mata el proceso
paste &lt;archivo1&gt; &lt;archivo2&gt; &gt; &lt;archivo.saida&gt;     # Junta lineas de archivos y separa por tabs (muy til para tablas)
cmp &lt;archivo1&gt; &lt;archivo2&gt;     # Muestra las coincidencias de dos archivos
diff &lt;archivo1&gt; &lt;archivo2&gt;    # Muestra las diferencias entre dos archivos
csplit -f out fasta_batch &quot;%^&gt;%&quot; &quot;/^&gt;/&quot; &quot;{*}&quot;     # Divide un archivo fasta en varios archivos a cada '&gt;' (cada nueva secuencia)

sort -k 2,2 -k 3,3n archivo.in &gt; archivo.out    # ordena la tabla, la columna 2 alfabeticamente y la columna 3 numericamente, -k para columna, -n para numerico
join -1 1 -2 1 &lt;tabla1&gt; &lt;tabla2&gt;    # Junta dos tablas con base a los nmeros especificados de las columnas. De la tabla 1 la columna 1 y de la tabla 2 la colunma 2. Se asume que las tablas estn ordenadas.">
                     <pre class="notranslate"><code>df    # Muestra el espacio disponible en el disco
free -g     # info de la memoria
uname -a    # Muestra la informacin de la mquina
du -sh    # muestra el espacio en uso en el disco
du sh *     # mueestra el espacio usado en disco por archivos y/o directorios 
du -s * | sort -nr    # Muestra el espacio usado en el disco por archivos y/o directorios ordenados por tamao
top     # Muestra el top de consumidores de memoria y CPU 
who     # Muestra quien est logado en el sistema
ps    # Muestra los procesos ejecutados por el usuario
ps -e     # Muestra todos los procesos en ejecin en el sistema
ps -o %t -p &lt;pid&gt;     # Muestra cuanto tiempo lleva ejecutando un determinado proceso (pid)
kill &lt;pid&gt;    # Mata el proceso
paste &lt;archivo1&gt; &lt;archivo2&gt; &gt; &lt;archivo.saida&gt;     # Junta lineas de archivos y separa por tabs (muy til para tablas)
cmp &lt;archivo1&gt; &lt;archivo2&gt;     # Muestra las coincidencias de dos archivos
diff &lt;archivo1&gt; &lt;archivo2&gt;    # Muestra las diferencias entre dos archivos
csplit -f out fasta_batch "%^&gt;%" "/^&gt;/" "{*}"     # Divide un archivo fasta en varios archivos a cada '&gt;' (cada nueva secuencia)

sort -k 2,2 -k 3,3n archivo.in &gt; archivo.out    # ordena la tabla, la columna 2 alfabeticamente y la columna 3 numericamente, -k para columna, -n para numerico
join -1 1 -2 1 &lt;tabla1&gt; &lt;tabla2&gt;    # Junta dos tablas con base a los nmeros especificados de las columnas. De la tabla 1 la columna 1 y de la tabla 2 la colunma 2. Se asume que las tablas estn ordenadas.
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       screen
                      </code>
                     </h3>
                     <a aria-label="Permalink: screen" class="anchor" href="#screen" id="user-content-screen">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Screen es una aplicacin desarrollada para Linux, que tiene como
objetivo la multiplexacin de terminales. O sea, l divide el terminal
fsico en varias sesiones virtuales. Funciona as, s ud est trabajando
en una sesin del terminal, usando screen, cuando apague su computador,
la sesin continuar corriendo los procesos y ud podr volver a acceder
a ella.
                    </p>
                    <p dir="auto">
                     Para iniciar una nueva sesin de screen, es solo digitar en la lnea de
comando
                     <code>
                      screen
                     </code>
                     . Aparecer un texto en la pantalla, puede apretar
[enter]. De esa manera el sistema criar una neuva sesin virtual e el
nombre de esa sesin ser asignado por el sistema. Sin embargo, s ud
quiere puede darle un nombre a la sesin, para eso es necesario comenzar
con el siguiente comando:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="screen -S mysesson">
                     <pre class="notranslate"><code>screen -S mysesson
</code></pre>
                    </div>
                    <p dir="auto">
                     Para salir de la sesin mantenga oprimido [Ctrl], en seguida oprima
[a] seguido de [d]. Para volver a la sesin use los seguientes
comandos:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="screen -ls # lista las sesiones activas
screen -r nombredelasesion # entra en la sesin deseada">
                     <pre class="notranslate"><code>screen -ls # lista las sesiones activas
screen -r nombredelasesion # entra en la sesin deseada
</code></pre>
                    </div>
                    <p dir="auto">
                     Para eliminar la sesin oprima [Ctrl + d]
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Scripts simples de uma linha de comando
                     </h3>
                     <a aria-label="Permalink: Scripts simples de uma linha de comando" class="anchor" href="#scripts-simples-de-uma-linha-de-comando" id="user-content-scripts-simples-de-uma-linha-de-comando">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Para renombras muchos archivos
                     <em>
                      .old
                     </em>
                     a *.new. Para probar primero,
subtituya
                     <code>
                      do mv
                     </code>
                     por
                     <code>
                      do echo mv
                     </code>
                     .
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for i in *.input; do mv $i ${i/\.old/\.new}; done
for i in *\ *; do mv "$i" "${i// /_}"; done # Substituye espacios en nombres de archivos por underscores'>
                     <pre class="notranslate"><code>for i in *.input; do mv $i ${i/\.old/\.new}; done
for i in *\ *; do mv "$i" "${i// /_}"; done # Substituye espacios en nombres de archivos por underscores
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <code>
                       scp
                      </code>
                      Secure Copy Between Machines
                     </h3>
                     <a aria-label="Permalink: scp Secure Copy Between Machines" class="anchor" href="#scp-secure-copy-between-machines" id="user-content-scp-secure-copy-between-machines">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <code>
                      scp
                     </code>
                     es un comando que sirve para copiar elementos entre el servidor y
su computador y viceversa.
                    </p>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                     <code>
                      scp source target
                     </code>
                    </p>
                    <p dir="auto">
                     Del servidor para su computador:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="scp user@ip.adress:camin/al/archivo.txt camino/en/su/pc">
                     <pre class="notranslate"><code>scp user@ip.adress:camin/al/archivo.txt camino/en/su/pc
</code></pre>
                    </div>
                    <p dir="auto">
                     Del computador para el servidor:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="scp camino/en/su/pc/archivo.txt user@ip.adress:camino/donde/quiere/copiar/enel/servidor/">
                     <pre class="notranslate"><code>scp camino/en/su/pc/archivo.txt user@ip.adress:camino/donde/quiere/copiar/enel/servidor/
</code></pre>
                    </div>
                    <p dir="auto">
                     Se ud quier copiar un directorio completo basta colocar el
                     <em>
                      flag
                     </em>
                     <code>
                      -r
                     </code>
                     despus de
                     <code>
                      scp
                     </code>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Compresin e Descompresin de archivos
                     </h3>
                     <a aria-label="Permalink: Compresin e Descompresin de archivos" class="anchor" href="#compresin-e-descompresin-de-archivos" id="user-content-compresin-e-descompresin-de-archivos">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Adems de
                     <code>
                      gzip
                     </code>
                     , existen otros tipos de compresin de archivos, tales
como
                     <code>
                      .tar
                     </code>
                     y
                     <code>
                      .zip
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Compresin
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="tar -cvf archivo.tar midirectorio/ #comprime midirectorio, y el nombre del archivo compreso ser archivo.tar
zip -r midrectorio.zip midirectorio/ #comprime el directorio midirectorio/ en un archivo llamado midirectorio.zip">
                     <pre class="notranslate"><code>tar -cvf archivo.tar midirectorio/ #comprime midirectorio, y el nombre del archivo compreso ser archivo.tar
zip -r midrectorio.zip midirectorio/ #comprime el directorio midirectorio/ en un archivo llamado midirectorio.zip
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Visualizar
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="tar -tvf archivo.tar">
                     <pre class="notranslate"><code>tar -tvf archivo.tar
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Extrair
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="tar -xvf archivo.tar
unzip midirectorio.zip">
                     <pre class="notranslate"><code>tar -xvf archivo.tar
unzip midirectorio.zip
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      I. Gestin de Herramientas Bioinformticas
                     </h1>
                     <a aria-label="Permalink: I. Gestin de Herramientas Bioinformticas" class="anchor" href="#i-gestin-de-herramientas-bioinformticas" id="user-content-i-gestin-de-herramientas-bioinformticas">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Anaconda
                     </h2>
                     <a aria-label="Permalink: Anaconda" class="anchor" href="#anaconda" id="user-content-anaconda">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Es recomendable instalar Anaconda, pues es la forma ms fcil para
instalar las herramientas bioinformticas. Anaconda es una distribucin
libre y abierta de los lenguajes
                     <em>
                      Python
                     </em>
                     y
                     <em>
                      R
                     </em>
                     , utilizada en ciencia de
datos y bioinformtica. Las diferentes versiones de los programas se
administran mediante un sistema de gestin llamado
                     <em>
                      conda
                     </em>
                     , el cual hace
bastante sencillo instalar, correr y actualizar programas.
                     <a href="https://conda.io/projects/conda/en/latest/user-guide/install/index.html" rel="nofollow">
                      Aqui
                     </a>
                     se encuentran las instrucciones para la instalacin de Anaconda. Despus
de instalado
                     <em>
                      Anaconda
                     </em>
                     y su gestor
                     <em>
                      Conda
                     </em>
                     , podran ser creados
                     <em>
                      ambientes virtuales
                     </em>
                     para la instalacin de las diferentes herramientas
bioinformticas que sern usadas.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Algunos comandos de
                      <code>
                       conda
                      </code>
                     </h2>
                     <a aria-label="Permalink: Algunos comandos de conda" class="anchor" href="#algunos-comandos-de-conda" id="user-content-algunos-comandos-de-conda">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="conda create -n miambiente # Crea un ambiente llamado miambiente
conda activate miambiente # Activa el ambiente mi ambiente
conda env list # Lista todos los ambientes creados
cond list # Estando dentro de un ambiente, lista las herramentas instaladas en ese ambiente">
                     <pre class="notranslate"><code>conda create -n miambiente # Crea un ambiente llamado miambiente
conda activate miambiente # Activa el ambiente mi ambiente
conda env list # Lista todos los ambientes creados
cond list # Estando dentro de un ambiente, lista las herramentas instaladas en ese ambiente
</code></pre>
                    </div>
                    <p dir="auto">
                     Para instalacin de cada herramienta, visite la pgina
                     <a href="https://anaconda.org/" rel="nofollow">
                      https://anaconda.org/
                     </a>
                     y en la caja de busqueda procure por la
herramienta que desea, y econtrat el comonado para instalacin.
                    </p>
                    <p dir="auto">
                     <strong>
                      Nota
                     </strong>
                     : Las herramientas usadas en este tutorial sern instaladas a
medida que van apareciendo dentro del pipeline.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      II. Adquisicin y organizacin de los datos
                     </h1>
                     <a aria-label="Permalink: II. Adquisicin y organizacin de los datos" class="anchor" href="#ii-adquisicin-y-organizacin-de-los-datos" id="user-content-ii-adquisicin-y-organizacin-de-los-datos">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      2.1. Organizacin de directorios
                     </h2>
                     <a aria-label="Permalink: 2.1. Organizacin de directorios" class="anchor" href="#21-organizacin-de-directorios" id="user-content-21-organizacin-de-directorios">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Para mantener el orden durante la ejecucin del
                     <em>
                      pipeline
                     </em>
                     es
recomendado crear directorios secuenciales a medida que son realizados
los diferentes pasos durante el proceso.
                    </p>
                    <p dir="auto">
                     Usando el comando
                     <code>
                      mkdir
                     </code>
                     , cree un directorio base para todo el proceso.
Asigne el nombre de transcriptomica
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="mkdir transcriptomica">
                     <pre class="notranslate"><code>mkdir transcriptomica
</code></pre>
                    </div>
                    <p dir="auto">
                     Entre al nuevo directorio usando el comando
                     <code>
                      cd
                     </code>
                     (
                     <em>
                      change directory
                     </em>
                     )
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd transcriptomica/">
                     <pre class="notranslate"><code>cd transcriptomica/
</code></pre>
                    </div>
                    <p dir="auto">
                     Cree un nuevo directorio para almacenar las secuencias brutas.
                    </p>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Tip:
                      </strong>
                      Debido a que la mayoria de las etapas del workflow son
secuenciales, es recomendable nombrar los directorios comenzando con
un nmero y as mantener la organizacin.
                     </p>
                    </blockquote>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="mkdir 00.RawData">
                     <pre class="notranslate"><code>mkdir 00.RawData
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      2.2 Adquisicin de datos
                     </h2>
                     <a aria-label="Permalink: 2.2 Adquisicin de datos" class="anchor" href="#22-adquisicin-de-datos" id="user-content-22-adquisicin-de-datos">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.2.1. SRA Toolkit
                     </h3>
                     <a aria-label="Permalink: 2.2.1. SRA Toolkit" class="anchor" href="#221-sra-toolkit" id="user-content-221-sra-toolkit">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Para la adquisicin de datos, ser instalado el programa
                     <a href="https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit">
                      SRA
Toolkit
                     </a>
                     ,
el cual nos permitir descargar las secuencias directamente del NCBI en
el servidor.
                    </p>
                    <p dir="auto">
                     Descargue el programa en el siguiente
                     <a href="https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit">
                      link
                     </a>
                    </p>
                    <p dir="auto">
                     Seleccione el archivo segn su sistema operativo. En este tutorial ser
instalada la version para
                     <a href="https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/3.0.0/sratoolkit.3.0.0-ubuntu64.tar.gz" rel="nofollow">
                      Ubuntu Linux
64bits
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     Usando el comando
                     <code>
                      wget
                     </code>
                     , descargue el archivo de la wed directamente en
el servidor
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Salga de la carpeta transcriptomica
cd ..

# Descargue el programa en la carpeta /home/user/
wget --output-document sratoolkit.tar.gz https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz">
                     <pre class="notranslate"><code># Salga de la carpeta transcriptomica
cd ..

# Descargue el programa en la carpeta /home/user/
wget --output-document sratoolkit.tar.gz https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     Ser descargada un archivo compactado (
                     <code>
                      sratoolkit.tar.gz
                     </code>
                     ). Use el
comando tar para descompactar
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="tar -vxzf sratoolkit.tar.gz">
                     <pre class="notranslate"><code>tar -vxzf sratoolkit.tar.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     El archivo ser descompactado en la carpeta
                     <code>
                      sratoolkit.3.0.0-ubuntu64/
                     </code>
                    </p>
                    <p dir="auto">
                     Configure el camino al programa
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="export PATH=$PATH:$PWD/sratoolkit.3.0.0-ubuntu64/bin">
                     <pre class="notranslate"><code>export PATH=$PATH:$PWD/sratoolkit.3.0.0-ubuntu64/bin
</code></pre>
                    </div>
                    <p dir="auto">
                     Verifique que la configuracin est correcta
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="which fastq-dump">
                     <pre class="notranslate"><code>which fastq-dump
</code></pre>
                    </div>
                    <p dir="auto">
                     Deber aparecer el camino al directorio que contiene el programa:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/user/sratoolkit.3.0.0-ubuntu64/bin/fastq-dump">
                     <pre class="notranslate"><code>/home/user/sratoolkit.3.0.0-ubuntu64/bin/fastq-dump
</code></pre>
                    </div>
                    <p dir="auto">
                     Antes de entrar en el men de configuracin, seleccione un directorio
vacio para usarlo como el repositorio de las secuencias a descargar.
Para esto siga la siguiente secuencia de comandos:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cree un directorio llamado secuencias-ncbi
mkdir secuencias-ncbi">
                     <pre class="notranslate"><code># Cree un directorio llamado secuencias-ncbi
mkdir secuencias-ncbi
</code></pre>
                    </div>
                    <p dir="auto">
                     Ingrese al men de configuracin con
                     <code>
                      vdb-config -i
                     </code>
                    </p>
                    <ul dir="auto">
                     <li>
                      Verifique que la opcin
                      <em>
                       Remote Access
                      </em>
                      est habilitada
                     </li>
                     <li>
                      Ingrese a la pestaa
                      <em>
                       Cache
                      </em>
                      y verifique que la opcin
                      <em>
                       local
file-caching
                      </em>
                      est habilitada.
                     </li>
                     <li>
                      En la opcin *Location of user-repository, busque el directorio
creado anteriormente
                      <code>
                       secuencias-ncbi/
                      </code>
                     </li>
                     <li>
                      En la pestaa
                      <em>
                       AWS
                      </em>
                      , habilite la opcin
                      <em>
                       report cloud instance
identity
                      </em>
                     </li>
                     <li>
                      Salve las configuraciones y salga del men con x.
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.2.2. Descargando las secuencias
                     </h3>
                     <a aria-label="Permalink: 2.2.2. Descargando las secuencias" class="anchor" href="#222-descargando-las-secuencias" id="user-content-222-descargando-las-secuencias">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Para la ejecucin de este
                     <em>
                      pipeline
                     </em>
                     sern usadas las secuencias de
RNA-seq del artculo en estudio y secuencias del genoma de
                     <em>
                      E.coli
                     </em>
                     ATCC8739, el cual ser usado como referencia para el ensamble de los
transcriptomas.
                    </p>
                    <p dir="auto">
                     <strong>
                      Secuencias RNA-seq
                     </strong>
                    </p>
                    <p dir="auto">
                     Para las secuencias de RNA-seq, usando el comando
                     <code>
                      nano
                     </code>
                     , cree un
archivo llamado
                     <code>
                      SRR-seqs-RNA.txt
                     </code>
                     en donde sern listados los cdigos
de acceso (SRR number) de los
                     <em>
                      datasets
                     </em>
                     . Siga la secuencia de cdigos
presentados a continuacin:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# entre en la carpeta secuencias-ncbi/
cd secuencias-ncbi/

# Cree el archivo con nano
nano SRR-seqs-RNA.txt">
                     <pre class="notranslate"><code># entre en la carpeta secuencias-ncbi/
cd secuencias-ncbi/

# Cree el archivo con nano
nano SRR-seqs-RNA.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Ser abierto el programa nano para editar el archivo
                     <code>
                      SRR-seqs-RNA.txt
                     </code>
                    </p>
                    <p dir="auto">
                     Los cdigos de accesso son los siguientes:
                    </p>
                    <p dir="auto">
                     SRR12922100 SRR12922099 SRR12922098 SRR12922089 SRR12922097 SRR12922096
SRR12922095 SRR12922094 SRR12922093 SRR12922092 SRR12922091 SRR12922090
                    </p>
                    <p dir="auto">
                     Copie y pegue dentro del archivo en el terminal. Para cerrar [Ctrl +
X], confirme que desea guardar los cambios con S.
                    </p>
                    <p dir="auto">
                     Debido a que es un proceso que va a demorar, es recomendable hacerlo
usando el programa screen.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="screen -S sra">
                     <pre class="notranslate"><code>screen -S sra
</code></pre>
                    </div>
                    <p dir="auto">
                     El siguiente comando le permitir descargar los archivos
                     <code>
                      .sra
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="prefetch --option-file SRR-seqs-RNA.txt">
                     <pre class="notranslate"><code>prefetch --option-file SRR-seqs-RNA.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Comenzar a ser descargadas cada una de las muestras segn los cdigos
de acceso. Cierre la sesin de screen con [Ctrl + A + D]. Cuando
quiera volver a la sesin use el comando
                     <code>
                      screen -r sra
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     Al terminar el proceso, en la carpeta
                     <code>
                      secuencias-ncbi
                     </code>
                     , se encontrarn
una serie de directorios, la carpeta
                     <code>
                      sra
                     </code>
                     contiene los archivos
                     <code>
                      .sra
                     </code>
                     .
Use los comandos
                     <code>
                      cd
                     </code>
                     para cambiar de directorio y
                     <code>
                      ll
                     </code>
                     para listar el
contenido.
                    </p>
                    <p dir="auto">
                     El siguiente paso es transformar los archivos
                     <code>
                      .sra
                     </code>
                     para
                     <code>
                      .fastq
                     </code>
                     .
Usando la funcin
                     <code>
                      for
                     </code>
                     , puede automatizar el proceso para que todas los
archivos sean transformados secuencialmente.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for i in ./*.sra
do
fasterq-dump --split-files $i
done">
                     <pre class="notranslate"><code>for i in ./*.sra
do
fasterq-dump --split-files $i
done
</code></pre>
                    </div>
                    <p dir="auto">
                     Use el comando
                     <code>
                      ll
                     </code>
                     para revisar el contenido de la carpeta. Mueva los
archivos
                     <code>
                      .fastq
                     </code>
                     para la carpeta creada para las secuencias brutas
                     <code>
                      transcriptomica/00.RawData
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="mv *.fastq ../../transcriptomica/00.RawData/">
                     <pre class="notranslate"><code>mv *.fastq ../../transcriptomica/00.RawData/
</code></pre>
                    </div>
                    <p dir="auto">
                     Confirme con
                     <code>
                      ll
                     </code>
                     que efectivamente los archivos se encuentran en la
nueva ubicacin.
                    </p>
                    <p dir="auto">
                     <strong>
                      Ensamble Genoma
                      <em>
                       E.coli
                      </em>
                      ATCC8739
                     </strong>
                    </p>
                    <p dir="auto">
                     Este ensamble se encuentra en la pagina de la coleccion
                     <a href="https://genomes.atcc.org/" rel="nofollow">
                      ATCC
                     </a>
                     . Cree una cuenta de usuario, use el
buscador de la pgina para encontrar el genoma y descargue el ensamble.
Use la interface grfica de MobaXterm para subir el archivo dentro de la
carpeta
                     <code>
                      transcriptomica/
                     </code>
                     . Use el comando
                     <code>
                      mv
                     </code>
                     para colocar el ensamble
del genoma dentro de la carpeta
                     <code>
                      00.RawData/
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Secuencias Genoma
                      <em>
                       Escherichia coli
                      </em>
                      ATCC 25922
                     </strong>
                    </p>
                    <p dir="auto">
                     Con el fin de aprender y practicar el montaje de genomas, sern
descargadas las secuencias brutas de la secuenciacin del genoma
completo de la
                     <em>
                      Escherichia coli
                     </em>
                     ATCC 25922.
                    </p>
                    <p dir="auto">
                     <strong>
                      Nota:
                     </strong>
                     Este genoma no ser usado para el proceso de ensamble del
transcriptoma.
                    </p>
                    <p dir="auto">
                     El cdigo de acceso a estas secuencias es SRR2889880. Use el programa
SRA Tool kit para descargar este conjunto de datos.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="prefetch SRR2889880">
                     <pre class="notranslate"><code>prefetch SRR2889880
</code></pre>
                    </div>
                    <p dir="auto">
                     Convierta el archivo
                     <code>
                      .sra
                     </code>
                     a archivos
                     <code>
                      .fastq
                     </code>
                     .
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="fasterq-dump --split-files SRR2889880">
                     <pre class="notranslate"><code>fasterq-dump --split-files SRR2889880
</code></pre>
                    </div>
                    <p dir="auto">
                     Use el comando
                     <code>
                      mv
                     </code>
                     para mover los archivos
                     <code>
                      .fastq
                     </code>
                     del genoma de
                     <em>
                      E.
coli
                     </em>
                     desde la carpeta
                     <code>
                      secuencias-ncbi/sra/
                     </code>
                     para la carpeta
                     <code>
                      /home/user/transcriptomica/01.RefGenome/
                     </code>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      2.3. Organizacin de los
                      <em>
                       datasets
                      </em>
                     </h2>
                     <a aria-label="Permalink: 2.3. Organizacin de los datasets" class="anchor" href="#23-organizacin-de-los-datasets" id="user-content-23-organizacin-de-los-datasets">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Debido a que los nombres de los archivos son los cdigos de acceso SRR,
es recomendable cambiarlos por nombres ms fciles de entender y
reconocer cada archivo. Use la informacin en la siguente tabla y el
comando
                     <code>
                      mv
                     </code>
                     para cambiar los nombres. Para facilitar primero use el
comando
                     <code>
                      cd
                     </code>
                     para entrar en el directorio
                     <code>
                      00.RawData/
                     </code>
                     .
                    </p>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        Cdigo SRR
                       </th>
                       <th>
                        Muestra
                       </th>
                       <th>
                        Nombre Archivo
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        SRR12922100
                       </td>
                       <td>
                        1/2MIC camphor essential oil (R3)
                       </td>
                       <td>
                        oil1-2_R3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR12922098
                       </td>
                       <td>
                        1/2MIC camphor essential oil (R2)
                       </td>
                       <td>
                        oil1-2_R2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR12922099
                       </td>
                       <td>
                        1/2MIC camphor essential oil (R1)
                       </td>
                       <td>
                        oil1-2_R1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR12922089
                       </td>
                       <td>
                        acetone
                       </td>
                       <td>
                        acetone
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR12922097
                       </td>
                       <td>
                        1/4MIC camphor essential oil (R3)
                       </td>
                       <td>
                        oil1-4_R3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR12922096
                       </td>
                       <td>
                        1/4MIC camphor essential oil (R2)
                       </td>
                       <td>
                        oil1-4_R2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR12922095
                       </td>
                       <td>
                        1/4MIC camphor essential oil (R1)
                       </td>
                       <td>
                        oil1-4_R1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR12922094
                       </td>
                       <td>
                        1/8MIC camphor essential oil (R3)
                       </td>
                       <td>
                        oil1-8_R3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR12922093
                       </td>
                       <td>
                        1/8MIC camphor essential oil (R2)
                       </td>
                       <td>
                        oil1-8_R2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR12922092
                       </td>
                       <td>
                        1/8MIC camphor essential oil (R1)
                       </td>
                       <td>
                        oil1-8_R1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR12922091
                       </td>
                       <td>
                        water (R2)
                       </td>
                       <td>
                        water_R2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR12922090
                       </td>
                       <td>
                        water (R1)
                       </td>
                       <td>
                        water_R1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SRR2889880
                       </td>
                       <td>
                        genoma
                        <em>
                         E. coli
                        </em>
                        ATCC 25922
                       </td>
                       <td>
                        ecoli
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     <em>
                      Nota:
                     </em>
                     Recuerde que debe tener dos archivos con el mismo nombre que
componen una muestra por ser una secuenciacin
                     <em>
                      paired-end
                     </em>
                     , con
excepcin del genoma de
                     <em>
                      E. coli
                     </em>
                     ATCC 25922 que es
                     <em>
                      single-end
                     </em>
                     . Por
ejemplo:
                     <code>
                      oil1-2_R3_1.fq
                     </code>
                     y
                     <code>
                      oil1-2_R3_2.fq
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Tip:
                     </strong>
                     Use la tecla TAB para facilitar y evitar errores a la hora de
digitar el nombre de los archivos.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      III. Control de calidad
                     </h1>
                     <a aria-label="Permalink: III. Control de calidad" class="anchor" href="#iii-control-de-calidad" id="user-content-iii-control-de-calidad">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      3.1. Control de Calidad
                     </h2>
                     <a aria-label="Permalink: 3.1. Control de Calidad" class="anchor" href="#31-control-de-calidad" id="user-content-31-control-de-calidad">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      3.1.1. Evaluacin de la calidad
                     </h3>
                     <a aria-label="Permalink: 3.1.1. Evaluacin de la calidad" class="anchor" href="#311-evaluacin-de-la-calidad" id="user-content-311-evaluacin-de-la-calidad">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Para la evaluacin de la calidad ser usado el programa
                     <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" rel="nofollow">
                      FastQC
                     </a>
                     que
es una herramienta que permite observar graficamente la calidad de las
secuencias de Illumina.
                    </p>
                    <p dir="auto">
                     <strong>
                      Instalacin
                     </strong>
                    </p>
                    <p dir="auto">
                     Las instrucciones para instalacin usando conda se encuentran
                     <a href="https://anaconda.org/bioconda/fastqc" rel="nofollow">
                      aqui
                     </a>
                     . Sin embargo aqui en este
tutorial tambin sern presentadas
                    </p>
                    <p dir="auto">
                     Como ya fue explicado anteriorimente, con conda es posible crear
ambientes virutuales para instalar las herramientas bioinformticas. El
primer ambiente que ser creado se llamar
                     <strong>
                      quality
                     </strong>
                     , donde se
instalaran los programas relacionados con este proceso.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="conda create -n quality">
                     <pre class="notranslate"><code>conda create -n quality
</code></pre>
                    </div>
                    <p dir="auto">
                     Durante el proceso, el sistema preguntar s desea proceder con la
creacin del ambiente, con las opciones y/n (si o no). Escriba
                     <code>
                      y
                     </code>
                     y
despus de eso el ambiente virtual estar creado.
                    </p>
                    <p dir="auto">
                     Para instalar las herramientas dentro del ambiente anteriormente creado,
es necesario activarlo
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="conda activate quality">
                     <pre class="notranslate"><code>conda activate quality
</code></pre>
                    </div>
                    <p dir="auto">
                     El ambiente estar activo cuando el nombre de ste se encuentra en el
comienzo de la linea de comando, as:
                     <code>
                      (quality) user@server:~/$
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     Posteriormente se procede a la instalacin del programa:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="conda install -c bioconda fastqc">
                     <pre class="notranslate"><code>conda install -c bioconda fastqc
</code></pre>
                    </div>
                    <p dir="auto">
                     De nuevo ser cuestionado si desea continuar con el proceso o no.
Escriba
                     <code>
                      y
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Uso
                     </strong>
                    </p>
                    <p dir="auto">
                     La primera etapa del proceso es la evaluacin de la calidad de las
secuencias cortas (Illumina) usando
                     <em>
                      FastQC
                     </em>
                     , con el objetivo de
determinar s es necesario trimar o filtrar las secuencias de baja
calidad en los prximos pasos.
                    </p>
                    <p dir="auto">
                     sta etapa es para identificar principalmente las secuencias
                     <em>
                      outlier
                     </em>
                     con baja calidad (
                     <math-renderer class="js-inline-math" data-run-id="5adc994830ac8c406144d27b37b0257b" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $Q&amp;lt; 20$
                     </math-renderer>
                     ).
                    </p>
                    <p dir="auto">
                     Active el ambiente
                     <code>
                      quality
                     </code>
                     :
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="conda activate quality

## Confirme que est en el directorio raz
pwd">
                     <pre class="notranslate"><code>conda activate quality

## Confirme que est en el directorio raz
pwd
</code></pre>
                    </div>
                    <p dir="auto">
                     Debe estar em
                     <code>
                      ~/transcriptomica/
                     </code>
                     . Si ese no es el resultado del
comando
                     <code>
                      pwd
                     </code>
                     , use el comando
                     <code>
                      cd
                     </code>
                     para llegar en el directorio base.
                    </p>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Tip:
                      </strong>
                      <strong>
                       TODAS
                      </strong>
                      las herramientas bioinformticas tienen un manual
de ayuda que puede ser accesado en la linea de comando, usando el flag
                      <code>
                       -h
                      </code>
                      o
                      <code>
                       --help
                      </code>
                      .
                      <strong>
                       ANTES de ejecutar una hierramenta siempre lea el
manual
                      </strong>
                      . Este comando lista todos los argumentos disponibles para la
herramienta y explica como deben/pueden ser usados.
                     </p>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Atencin:
                      </strong>
                      Para el uso correcto y seguro del servidor verifique ele
nmero de ncleos disponibles para el usuario en el momento del
anlisis.
                      <strong>
                       NUNCA
                      </strong>
                      trabaje con el total de los ncleos de la mquina.
Ejecute
                      <strong>
                       FastQC
                      </strong>
                      :
                     </p>
                    </blockquote>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Cree un directorio para salvar el output de FastQC
mkdir 01.FastqcReports

## Ejecute usando 10 threads
fastqc -t 10 00.RawData/* -o 01.FastqcReports/">
                     <pre class="notranslate"><code>## Cree un directorio para salvar el output de FastQC
mkdir 01.FastqcReports

## Ejecute usando 10 threads
fastqc -t 10 00.RawData/* -o 01.FastqcReports/
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="fastqc -t &lt;num ncleos&gt; &lt;input_directory&gt; -o &lt;output_directory&gt;">
                     <pre class="notranslate"><code>fastqc -t &lt;num ncleos&gt; &lt;input_directory&gt; -o &lt;output_directory&gt;
</code></pre>
                    </div>
                    <p dir="auto">
                     El comando
                     <code>
                      fastqc
                     </code>
                     tiene varias opciones o parametros, entre ellas,
escoger el nmero de ncleos de la mquina para correr el anlisis, para
este caso
                     <code>
                      -t 10
                     </code>
                     . El input es el directorio que contiene las secuencias
de illumina
                     <code>
                      00.RawData/*
                     </code>
                     , el
                     <code>
                      *
                     </code>
                     indica al sistema que puede analizar
todos los archivos que estn dentro de ese directorio. El output,
indicado por el parametro
                     <code>
                      -o
                     </code>
                     , es el directorio donde se desea que sean
guardados los resultados del anlisis. A continuacin se encuentra una
explicacin detallada de cada output generado.
                    </p>
                    <p dir="auto">
                     <strong>
                      Outputs
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      Reportes html
                      <code>
                       oil1-2_R3.html
                      </code>
                      : Aqui es posible ver toda informacin
de calidad graficamente.
                     </li>
                     <li>
                      Zip files
                      <code>
                       oil1-2_R3.zip
                      </code>
                      : Aqui se encuentran cada uno de los
grficos de manera separada. *IGNORE**
                     </li>
                    </ul>
                    <p dir="auto">
                     Descargue los archivos
                     <code>
                      html
                     </code>
                     y explore en su
                     <em>
                      web browser
                     </em>
                     .
                    </p>
                    <p dir="auto">
                     Revise cada uno de los grficos y datos que son generados en estos
reportes, los cuales permiten determinar los parametros para el siguente
paso, en el cual se filtraran y eliminaran las secuencias y bases con
baja calidad. En este
                     <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/" rel="nofollow">
                      link
                     </a>
                     se encuentra la explicacin de cada una de las partes del reporte y como
interpretar los resultados.
                    </p>
                    <p dir="auto">
                     Los reportes mostraron que las secuencias de RNA-seq en trminos
generales son de alta calidad. Los grficos que mostraron errores,
pueden ser ignorados, puesto que es normal que esto pase con librerias
de RNA-seq. Por esta razn estas secuencias no sern
depuradas/filtradas. Sin embargo las secuencias
                     <em>
                      single-end
                     </em>
                     del genoma
de
                     <em>
                      E. coli
                     </em>
                     , presentan varios problemas de calidad. El grfico
                     <em>
                      Per
base sequence quality
                     </em>
                     muestra que algunas bases finales de las
secuencias tienen una calidad inferior a 20 (
                     <math-renderer class="js-inline-math" data-run-id="5adc994830ac8c406144d27b37b0257b" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $Q&amp;lt;20$
                     </math-renderer>
                     ). El grfico
                     <em>
                      Per
base sequence content
                     </em>
                     , muestra que las primeras 18 bases presentan
diferentes proporcin de GC y AT. El grfico
                     <em>
                      Per sequence GC content
                     </em>
                     muestra que hay una leve desviacin entre la distribucin terica del
%GC y la distribucin real de %GC por
                     <em>
                      reads
                     </em>
                     . Analizando cada uno de
los grficos son decididos los parametros para la etapa de depuracin.
Para este caso es necesario filtrar por calidad, cortar las primeras
bases y filtrar por tamao.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      3.1.2. Depuracin/
                      <em>
                       Trimming
                      </em>
                     </h3>
                     <a aria-label="Permalink: 3.1.2. Depuracin/Trimming" class="anchor" href="#312-depuracintrimming" id="user-content-312-depuracintrimming">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Segn fue evaluado en el control de calidad, ser necesario filtrar
algunas lecturas del genoma de
                     <em>
                      E. coli
                     </em>
                     . Las secuencias de las
librerias de RNA-seq no pasarn por el proceso de depuracin. Para esta
etapa ser usado o programa
                     <a href="http://www.usadellab.org/cms/?page=trimmomatic" rel="nofollow">
                      Trimmomatic
v0.39
                     </a>
                     que permite
filtrar (remover) lecturas o
                     <em>
                      reads
                     </em>
                     cortas de baja calidad.
                    </p>
                    <p dir="auto">
                     Trimmomatic tiene vrios parametros que pueden ser considerados para
filtrar lecturas con baja calidad. Aqui usaremos algunos. Si quiere
saber que otros parametros y como funciona cada uno de ellos, consulte
el
                     <a href="http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf" rel="nofollow">
                      manual
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Instalacin
                     </strong>
                    </p>
                    <p dir="auto">
                     Como se trata de una herramienta que participa dentro del proceso de
control de calidad, ser instalada dentro del ambiente virtual
                     <strong>
                      quality
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Si no est activado el ambiente
conda activate quality

# Instale Trimmomatic
conda install -c bioconda trimmomatic">
                     <pre class="notranslate"><code># Si no est activado el ambiente
conda activate quality

# Instale Trimmomatic
conda install -c bioconda trimmomatic
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Uso
                     </strong>
                    </p>
                    <p dir="auto">
                     Para los datos aqui analizados se usar la siguiente linea de comando:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Activa el ambiente quality
conda activate quality

# Crie un directorio para salvar las lecturas limpias
mkdir 02.CleanData

# Ejecute Trimmomatic
trimmomatic SE -threads 10 00.RawData/ecoli.fq 02.CleanData/ecoli.fq LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 HEADCROP:18 MINLEN:80 ILLUMINACLIP:/home/drm/anaconda3/pkgs/trimmomatic-0.36-6/share/trimmomatic-0.36-6/adapters/NexteraPE-PE.fa:2:30:10">
                     <pre class="notranslate"><code># Activa el ambiente quality
conda activate quality

# Crie un directorio para salvar las lecturas limpias
mkdir 02.CleanData

# Ejecute Trimmomatic
trimmomatic SE -threads 10 00.RawData/ecoli.fq 02.CleanData/ecoli.fq LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 HEADCROP:18 MINLEN:80 ILLUMINACLIP:/home/drm/anaconda3/pkgs/trimmomatic-0.36-6/share/trimmomatic-0.36-6/adapters/NexteraPE-PE.fa:2:30:10
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                     <code>
                      trimmomatic SE -threads input output [opciones]
                     </code>
                    </p>
                    <p dir="auto">
                     El comando anterior tiene varias partes. Primero, el nombre del comando
es
                     <code>
                      trimmomatic
                     </code>
                     , a continuacin la opcin
                     <code>
                      SE
                     </code>
                     indica para el programa
que las secuencias que irn a ser analizadas son de tipo
                     <em>
                      single end
                     </em>
                     ,
para
                     <em>
                      paired-end
                     </em>
                     , use
                     <code>
                      PE
                     </code>
                     . Despus se encuentra el input, si se
tratara de secuencias
                     <em>
                      paired-end
                     </em>
                     tendria dos inputs, forward (pair1) y
reverse (pair2). Despus el output. En el caso de
                     <em>
                      paired-end
                     </em>
                     sern
varios outputs, siendo primero las secuencias forward pareadas (limpias)
y no pareadas y despus las secuencias reverse. Por ltimo se encuentran
los parametros de depuracin. Para este caso usamos los parametros
                     <code>
                      SLIDINGWINDOW
                     </code>
                     ,
                     <code>
                      LEADING
                     </code>
                     ,
                     <code>
                      TRAILING
                     </code>
                     ,
                     <code>
                      HEADCROP
                     </code>
                     ,
                     <code>
                      MINLEN
                     </code>
                     y
                     <code>
                      ILLUMINACLIP
                     </code>
                     .
                    </p>
                    <ul dir="auto">
                     <li>
                      SLIDINGWINDOW: genera una ventana deslizante, que en este caso va de
4 en 4 bases, clcula el promedio del
                      <em>
                       Phred Score
                      </em>
                      y si est por
debajo de 15 esas bases son cortadas.
                     </li>
                     <li>
                      LEADING: corta bases del comienzo de la lectura si estn por debajo
de
                      <em>
                       threshold
                      </em>
                      de calidad
                     </li>
                     <li>
                      TRAILING: corta bases del final de la lectura si estn por debajo de
                      <em>
                       threshold
                      </em>
                      de calidad.
                     </li>
                     <li>
                      HEADCROP: corta bases del inicio de la lectura
                     </li>
                     <li>
                      MINLEN: Secuencias con menor tamao del establecido, son descartadas
                     </li>
                     <li>
                      ILLUMINACLIP: Elimina bases que corresponden con la secuencia de
adaptadores usados en la plataforma Illumina.
                     </li>
                    </ul>
                    <p dir="auto">
                     Despus de correr Trimmomatic es necesario evaluar la calidad de las
secuencias generadas (limpias) usando nuevamente FastQC.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="fastqc -t 10 02.CleanData/* -o 01.FastqcReports/">
                     <pre class="notranslate"><code>fastqc -t 10 02.CleanData/* -o 01.FastqcReports/
</code></pre>
                    </div>
                    <p dir="auto">
                     Descargue el reporte
                     <code>
                      01.FastqcReports/ecoli.html
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     Observe que ahora todas las bases en ambos archivos tienen
                     <math-renderer class="js-inline-math" data-run-id="5adc994830ac8c406144d27b37b0257b" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $Q&amp;gt;30$
                     </math-renderer>
                     .
Despus del proceso de filtrado sobrevivieron
                     <math-renderer class="js-inline-math" data-run-id="5adc994830ac8c406144d27b37b0257b" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $60'258.702$
                     </math-renderer>
                     <em>
                      reads
                     </em>
                     , es
decir
                     <math-renderer class="js-inline-math" data-run-id="5adc994830ac8c406144d27b37b0257b" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $85.4%$
                     </math-renderer>
                     de las secuencias iniciales (
                     <math-renderer class="js-inline-math" data-run-id="5adc994830ac8c406144d27b37b0257b" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $70'560.424$
                     </math-renderer>
                     ).
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      3.1.3. Cobertura
                     </h3>
                     <a aria-label="Permalink: 3.1.3. Cobertura" class="anchor" href="#313-cobertura" id="user-content-313-cobertura">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Cuando se trabaja con secuenciacin de genomas, el siguiente paso
despus de conocer la cantidad de secuencias obtenidas y remover las
secuencias con baja calidad, es calcular la cobertura o profundidad de
la secuenciacin para ese genoma. El clculo de la cobertura est dado
por la siguiente ecuacin:
                    </p>
                    <p dir="auto">
                     <math-renderer class="js-inline-math" data-run-id="5adc994830ac8c406144d27b37b0257b" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $$C = (L*N)/G$$
                     </math-renderer>
                     Donde: L es igual al tamao de las lecturas, N es igual
al nmero de lecturas, y G es igual al tamao aproximado del genoma.
                    </p>
                    <p dir="auto">
                     As, entonces para el genoma del presente tutorial, con el
secuenciamiento Illumina tenemos:
                    </p>
                    <p dir="auto">
                     <math-renderer class="js-display-math" data-run-id="5adc994830ac8c406144d27b37b0257b" data-static-url="https://github.githubassets.com/static" style="display: block">
                      $$C = ((83bp) * 60258702)/5209985bp$$
                     </math-renderer>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#&gt; [1] "La cobertura es de 960X"'>
                     <pre class="notranslate"><code>#&gt; [1] "La cobertura es de 960X"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      IV. Genmica
                     </h1>
                     <a aria-label="Permalink: IV. Genmica" class="anchor" href="#iv-genmica" id="user-content-iv-genmica">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4.1 Ensamblaje
                      <em>
                       de novo
                      </em>
                     </h2>
                     <a aria-label="Permalink: 4.1 Ensamblaje de novo" class="anchor" href="#41-ensamblaje-de-novo" id="user-content-41-ensamblaje-de-novo">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Un ensamblaje es el proceso de alineamiento de secuencias cortas con el
objetivo de recuperar una secuencia mayor. En el caso del ensamblaje
                     <em>
                      de
novo
                     </em>
                     son usados algortmos con base en
                     <em>
                      k-mers
                     </em>
                     , que son subsecuencias
com tamao definido por el usuario (i.e.21-mers).
                    </p>
                    <p dir="auto">
                     <a href="https://github.com/ablab/spades">
                      Spades v3.15.3
                     </a>
                     es uno de los
ensambladores de genomas, ms conocido y con mejores resultados, y puede
ser usado tanto para lecturas cortas como largas. Lea atentamente el
                     <a href="http://cab.spbu.ru/files/release3.15.2/manual.html" rel="nofollow">
                      manual
                     </a>
                     , ya que
este programa tiene muchas opciones diferentes. Spades usa el algoritmo
del
                     <em>
                      Grafo de Bruijn
                     </em>
                     para el montaje de las secuencias.
                    </p>
                    <p dir="auto">
                     <strong>
                      Instalacin
                     </strong>
                    </p>
                    <p dir="auto">
                     Siga las siguientes instrucciones para la instalacin de
                     <strong>
                      Spades
                     </strong>
                     dentro de ambiente virtual llamado
                     <em>
                      assembly
                     </em>
                     .
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cree el ambiente assembly
conda create -n assembly

# Active el ambiente virtual
conda activate assembly

# Instale Spades
conda install -c bioconda spades">
                     <pre class="notranslate"><code># Cree el ambiente assembly
conda create -n assembly

# Active el ambiente virtual
conda activate assembly

# Instale Spades
conda install -c bioconda spades
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Uso
                     </strong>
                    </p>
                    <p dir="auto">
                     El ensamble ser realizado ejecutando el comando:
                    </p>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Tip:
                      </strong>
                      El programa nohup permite ejecutar tareas en segundo plano,
con el objetivo de mantener la ejecucin del comando an con perdida
de conexin
                     </p>
                    </blockquote>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="nohup spades.py --careful -s 02.CleanData/ecoli.fq -k 21,33,55,77,99,111,127 -o 03.Assembly/01.Careful -t 40">
                     <pre class="notranslate"><code>nohup spades.py --careful -s 02.CleanData/ecoli.fq -k 21,33,55,77,99,111,127 -o 03.Assembly/01.Careful -t 40
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxe
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      <em>
                       Paired-end
                      </em>
                     </li>
                    </ul>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="spades.py -1 &lt;pair1&gt; -2 &lt;pair2&gt; -k &lt;kmers list&gt; -o &lt;output_directory&gt; -t &lt;num_nucleos&gt;">
                     <pre class="notranslate"><code>spades.py -1 &lt;pair1&gt; -2 &lt;pair2&gt; -k &lt;kmers list&gt; -o &lt;output_directory&gt; -t &lt;num_nucleos&gt;
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <em>
                       Single-end
                      </em>
                     </li>
                    </ul>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="spades.py -s &lt;single-end-reads&gt; -k &lt;kmers list&gt; -o &lt;output_directory&gt; -t &lt;num_nucleos&gt;">
                     <pre class="notranslate"><code>spades.py -s &lt;single-end-reads&gt; -k &lt;kmers list&gt; -o &lt;output_directory&gt; -t &lt;num_nucleos&gt;
</code></pre>
                    </div>
                    <p dir="auto">
                     Con el objetivo de comparar ensambles, ejecute nuevamente el comando
anterior sin el flag
                     <code>
                      --careful
                     </code>
                     . Este flag intenta reducir el nmero de
mismatches e indels cortos y es recomendado para el ensamble de genomas
de procariotas. Debe cambiar la carpeta de salida para no sobre escribir
los resultados del primer ensamble, use
                     <code>
                      03.Assembly/02.NoCareful
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     En el
                     <a href="http://cab.spbu.ru/files/release3.15.2/manual.html" rel="nofollow">
                      manual
                     </a>
                     encuentra ms detalles.
                    </p>
                    <p dir="auto">
                     <strong>
                      Output
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <code>
                        corrected/
                       </code>
                       : contiene las reads corregidas por
                       <strong>
                        BayesHammer
                       </strong>
                       en
                       <code>
                        .fastq.gz
                       </code>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        scaffolds.fasta
                       </code>
                       : contiene los scaffolds obtenidos
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        contigs.fasta
                       </code>
                       : contiene los contigis obtenidos
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        assembly_graph_with_scaffolds.gfa
                       </code>
                       : contiene el grafo del montaje
en formato GFA 1.0.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        assembly_graph.fastg
                       </code>
                       : contiene el grafo del montaje en formato
FASTG
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4.2 Evaluacin del ensamble
                     </h2>
                     <a aria-label="Permalink: 4.2 Evaluacin del ensamble" class="anchor" href="#42-evaluacin-del-ensamble" id="user-content-42-evaluacin-del-ensamble">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     El ensamble debe ser evaluado a travs de mtricas que representan la
calidad del genoma. En este paso ser calculado el N50, el nmero de
contigs, el tamao del genoma y que tan completo y contaminado est.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.2.1. Quast
                     </h3>
                     <a aria-label="Permalink: 4.2.1. Quast" class="anchor" href="#421-quast" id="user-content-421-quast">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="http://quast.sourceforge.net/docs/manual.html" rel="nofollow">
                      Quast v5.0.2
                     </a>
                     (
                     <em>
                      QUality
ASsesment Tool
                     </em>
                     ) es posible evaluar las principales estadsticas del
montaje (i.e.N50, nmero de contigs, tamao total del montaje, tamao
de los contigs, etc).
                     <strong>
                      Quast
                     </strong>
                     genera una serie de archivos y reportes
donde es posible observar esas estadsticas bsicas del montaje. Sern
comparados los montajes obtenidos anteriormente, con el objetivo de
escoger el mejor, para las siguientes etapas.
                    </p>
                    <p dir="auto">
                     <strong>
                      Instalacin
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cree el ambiente quast
conda create -n quast

# Active el ambiente quast
conda activate quast

# Instale Quast
conda install -c bioconda quast">
                     <pre class="notranslate"><code># Cree el ambiente quast
conda create -n quast

# Active el ambiente quast
conda activate quast

# Instale Quast
conda install -c bioconda quast
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Uso
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Vuelva al diretorio base

# Crie um diretrio para el output
mkdir 04.AssemblyQuality
mkdir 04.AssemblyQuality/01.Quast

# Ejecute Quast
quast.py 03.Assembly/01.Careful/scaffolds.fasta 03.Assembly/02.NoCareful/scaffolds.fasta -o 04.AssemblyQuality/01.Quast">
                     <pre class="notranslate"><code># Vuelva al diretorio base

# Crie um diretrio para el output
mkdir 04.AssemblyQuality
mkdir 04.AssemblyQuality/01.Quast

# Ejecute Quast
quast.py 03.Assembly/01.Careful/scaffolds.fasta 03.Assembly/02.NoCareful/scaffolds.fasta -o 04.AssemblyQuality/01.Quast
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                     <code>
                      quast.py path/to/assembly/contigs.fasta -o path/to/output/
                     </code>
                    </p>
                    <p dir="auto">
                     <strong>
                      Interpetacin de los resultados
                     </strong>
                    </p>
                    <p dir="auto">
                     La idea de usar
                     <strong>
                      Quast
                     </strong>
                     , aparte de evaluar las estidsticas bsicas
del montaje, es comparar varios montajes para escoger el mejor. Por
ejemplo: entre menor sea el nmero de contigs es mejor, porque significa
que el genoma qued menos fragementado. Y eso se reflejar en el tamao
de los contigs que sern grandes. El valor de N50, es mejor entre mayor
sea. As mismo, es ideal menor nmero de gaps y Ns.
                    </p>
                    <p dir="auto">
                     <strong>
                      Outputs
                     </strong>
                    </p>
                    <p dir="auto">
                     Explore el directorio de output usando el comando
                     <code>
                      ls
                     </code>
                     .
                    </p>
                    <ul dir="auto">
                     <li>
                      <code>
                       04.AssemblyQuality/01.Quast/report.html
                      </code>
                      : Este reporte puede ser
abierto en un
                      <em>
                       web browser
                      </em>
                      y contiene las informaciones ms
relevantes. Como nmero de contigs, tamao del mayor contig, tamao
total del montaje, N50, etc.
                     </li>
                    </ul>
                    <p dir="auto">
                     Los contigs de tamao menor de 500 bp no tienen un valor representativo
en el ensamble, por lo tanto es recomendable filtrar esas secuencias.
                    </p>
                    <p dir="auto">
                     El programa bbmap tiene un script que permite realizar la depuracin por
tamao. Instale bbmap en un ambiente llamado bioinfo:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cree el ambiente bioinfo
conda install -n bioinfo

# Active el ambiente bioinfo
conda activate bioinfo

# Instale bbmap
conda install -c bioconda bbmap

# Ejecute bbmap (careful)
reformat.sh in=03.Assembly/01.Careful/scaffolds.fasta out=03.Assembly/01.Careful/scaffolds_filtered.fasta minlength=500

# Ejecute bbmap (No careful)
reformat.sh in=03.Assembly/02.NoCareful/scaffolds.fasta out=03.Assembly/02.NoCareful/scaffolds_filtered.fasta minlength=500">
                     <pre class="notranslate"><code># Cree el ambiente bioinfo
conda install -n bioinfo

# Active el ambiente bioinfo
conda activate bioinfo

# Instale bbmap
conda install -c bioconda bbmap

# Ejecute bbmap (careful)
reformat.sh in=03.Assembly/01.Careful/scaffolds.fasta out=03.Assembly/01.Careful/scaffolds_filtered.fasta minlength=500

# Ejecute bbmap (No careful)
reformat.sh in=03.Assembly/02.NoCareful/scaffolds.fasta out=03.Assembly/02.NoCareful/scaffolds_filtered.fasta minlength=500
</code></pre>
                    </div>
                    <p dir="auto">
                     Ejecute nuevamente Quast, evaluando los dos ensambles, los contigs
filtrados y sin filtar.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Active el ambiente quast
conda activate quast

# Ejecute Quast
quast.py 03.Assembly/01.Careful/scaffolds.fasta 03.Assembly/01.Careful/scaffolds_filtered.fasta 03.Assembly/02.NoCareful/scaffolds.fasta 03.Assembly/02.NoCareful/scaffolds_filtered.fasta -o 04.AssemblyQuality/01.Quast">
                     <pre class="notranslate"><code># Active el ambiente quast
conda activate quast

# Ejecute Quast
quast.py 03.Assembly/01.Careful/scaffolds.fasta 03.Assembly/01.Careful/scaffolds_filtered.fasta 03.Assembly/02.NoCareful/scaffolds.fasta 03.Assembly/02.NoCareful/scaffolds_filtered.fasta -o 04.AssemblyQuality/01.Quast
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.2.2. CheckM
                     </h3>
                     <a aria-label="Permalink: 4.2.2. CheckM" class="anchor" href="#422-checkm" id="user-content-422-checkm">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Para evaluar que tan completo y contaminado estn los ensambles es usada
la herramienta
                     <a href="https://github.com/Ecogenomics/CheckM/wiki">
                      CheckM
                     </a>
                     . La
cual usa una base de datos propia de genes ortlogos de copia nica.
                    </p>
                    <p dir="auto">
                     Instale el programa en un ambiente llamado
                     <code>
                      checkm
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cree el ambiente checkm
conda create -n checkm

# Active el ambiente
conda activate checkm

# Instale checkm
conda install -c bioconda checkm-genome">
                     <pre class="notranslate"><code># Cree el ambiente checkm
conda create -n checkm

# Active el ambiente
conda activate checkm

# Instale checkm
conda install -c bioconda checkm-genome
</code></pre>
                    </div>
                    <p dir="auto">
                     Ejecute el nalisis para todos los ensambles, filtrados y no filtrados.
Los ensambles deben estar todos en un mismo directorio, entonces cree un
nuevo directorio dentro de la carpeta
                     <code>
                      03.Assembly/
                     </code>
                     llamado
                     <code>
                      03.Scaffolds/
                     </code>
                     . Despus entre en cada carpeta de los ensambles y usando
el comando
                     <code>
                      mv
                     </code>
                     , cambie el nombre de los archivos, con el fin de
diferenciarlos (actualmente tienen el mismo nombre) entre careful y no
careful. Por ltimo, nuevamente use
                     <code>
                      mv
                     </code>
                     para mover los archivos para el
directorio creado anteriormente (
                     <code>
                      03.Assembly/03.Scaffolds/
                     </code>
                     )
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd 03.Assembly/01.Careful

mv scaffolds.fasta careful.fasta

mv scaffolds_filtered.fasta careful_filtered.fasta

mv careful* ../03.Scaffolds

cd ../02.NoCareful

mv scaffolds.fasta Nocareful.fasta

mv scaffolds_filtered.fasta Nocareful_filtered.fasta

mv Nocareful* ../03.Scaffolds">
                     <pre class="notranslate"><code>cd 03.Assembly/01.Careful

mv scaffolds.fasta careful.fasta

mv scaffolds_filtered.fasta careful_filtered.fasta

mv careful* ../03.Scaffolds

cd ../02.NoCareful

mv scaffolds.fasta Nocareful.fasta

mv scaffolds_filtered.fasta Nocareful_filtered.fasta

mv Nocareful* ../03.Scaffolds
</code></pre>
                    </div>
                    <p dir="auto">
                     Despus de organizado los archivos, ya es posible ejecutar CheckM:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cree el directorio para el output
cd ../../

mkdir 04.AssemblyQuality/02.Checkm

checkm lineage_wf 03.Assembly/03.Scaffolds/ 04.AssemblyQuality/02.Checkm -t 40 -x fasta --tab &gt; 04.AssemblyQuality/02.Checkm/output.txt">
                     <pre class="notranslate"><code># Cree el directorio para el output
cd ../../

mkdir 04.AssemblyQuality/02.Checkm

checkm lineage_wf 03.Assembly/03.Scaffolds/ 04.AssemblyQuality/02.Checkm -t 40 -x fasta --tab &gt; 04.AssemblyQuality/02.Checkm/output.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="checkm lineage_wf &lt;input_directory/&gt; &lt;output_directory/&gt; -t &lt;num_nucleos&gt; -x &lt;format&gt; --tab &gt; output.txt">
                     <pre class="notranslate"><code>checkm lineage_wf &lt;input_directory/&gt; &lt;output_directory/&gt; -t &lt;num_nucleos&gt; -x &lt;format&gt; --tab &gt; output.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Explore el archivo de salida
                     <code>
                      04.AssemblyQuality/02.Checkm/output.txt
                     </code>
                     usando el comando
                     <code>
                      less
                     </code>
                     . Descargue el reporte en su computador.
                    </p>
                    <p dir="auto">
                     Para ms detalles sobre la interpretacin del reporte visite este
                     <a href="https://www.biostars.org/p/447744/" rel="nofollow">
                      link.
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4.3. Anotacin Taxonmica
                     </h2>
                     <a aria-label="Permalink: 4.3. Anotacin Taxonmica" class="anchor" href="#43-anotacin-taxonmica" id="user-content-43-anotacin-taxonmica">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     La clasificacin con base en datos de genoma tiene mayor poder de
resolucin en comparacin al usar apenas un gene marcador, pues esta
abordaje analiza mltiples genes que presentan un resultado mucho ms
robusto de las relaciones de parentesco del organismo de inters.
                     <a href="https://ecogenomics.github.io/GTDBTk/index.html" rel="nofollow">
                      GTDB-tk
                     </a>
                     es una
herramienta que identifica 120 genes marcadores y los compara con una
base de datos curada y constamente actualizada.
                    </p>
                    <p dir="auto">
                     Instale el prohrama en un ambiente llamado
                     <code>
                      gtdbtk
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cree el ambiente gtdbtk
conda create -n gtdbtk

# Active el ambiente
conda activate gtdbtk

# Instale GTDBtk
conda install -c conda-forge -c bioconda gtdbtk">
                     <pre class="notranslate"><code># Cree el ambiente gtdbtk
conda create -n gtdbtk

# Active el ambiente
conda activate gtdbtk

# Instale GTDBtk
conda install -c conda-forge -c bioconda gtdbtk
</code></pre>
                    </div>
                    <p dir="auto">
                     Para descargar y configurar la base de datos, siga las
                     <a href="https://ecogenomics.github.io/GTDBTk/installing/bioconda.html#step-3-download-and-alias-the-gtdb-tk-reference-data" rel="nofollow">
                      instrucciones
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     Ejecute GTDBtk usando el seguinte comando:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="gtdbtk classify_wf --genome_dir 03.Assembly/03.Scaffolds/ --out_dir 05.TaxonomyAnnotation/ -x fasta --cpus 40">
                     <pre class="notranslate"><code>gtdbtk classify_wf --genome_dir 03.Assembly/03.Scaffolds/ --out_dir 05.TaxonomyAnnotation/ -x fasta --cpus 40
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="gtdbtk classify_wf --genome_dir &lt;input_directory/&gt; --out_dir &lt;output_directory/&gt; -x &lt;format&gt; --cpus &lt;num_nucleos&gt;">
                     <pre class="notranslate"><code>gtdbtk classify_wf --genome_dir &lt;input_directory/&gt; --out_dir &lt;output_directory/&gt; -x &lt;format&gt; --cpus &lt;num_nucleos&gt;
</code></pre>
                    </div>
                    <p dir="auto">
                     El programa genera varios archivos de salida que estn resumidos en
                     <code>
                      05.TaxonomyAnnotation/gtdbtk.bac120.summary.tsv
                     </code>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4.4. Prediccin de genes
                     </h2>
                     <a aria-label="Permalink: 4.4. Prediccin de genes" class="anchor" href="#44-prediccin-de-genes" id="user-content-44-prediccin-de-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     el objetivo de esta etapa es procurar los ORFs (
                     <em>
                      Open Reading Frames
                     </em>
                     )
dentro de los contig/scaffolds. Es decir, predecir donde inician e
terminan los genes. Basicamente el programa busca codones de inicio y de
                     <em>
                      stop
                     </em>
                     .Para este objetivo ser usada la herramienta
                     <a href="https://github.com/hyattpd/prodigal/wiki">
                      Prodigal
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Instalacin
                     </strong>
                    </p>
                    <p dir="auto">
                     Instale el programa en el ambiente bioinfo
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Active el ambiente
conda activate bioinfo

# Instale Prodigal
conda install -c bioconda prodigal">
                     <pre class="notranslate"><code># Active el ambiente
conda activate bioinfo

# Instale Prodigal
conda install -c bioconda prodigal
</code></pre>
                    </div>
                    <p dir="auto">
                     Para ejecutar el programa utilice el seguiente comando:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cree una carpeta para guardar el resultado
mkdir 06.GenePrediction

#
prodigal -f gff -i 03.Assembly/03.Scaffolds/careful_filtered.fasta -o 06.GenePrediction/careful_filtered_orf.gff -a 06.GenePrediction/careful_filtered_proteins.faa -d 06.GenePrediction/careful_filtered_ORFnucleotides.fa -s 06.GenePrediction/careful_filtered_genes">
                     <pre class="notranslate"><code># Cree una carpeta para guardar el resultado
mkdir 06.GenePrediction

#
prodigal -f gff -i 03.Assembly/03.Scaffolds/careful_filtered.fasta -o 06.GenePrediction/careful_filtered_orf.gff -a 06.GenePrediction/careful_filtered_proteins.faa -d 06.GenePrediction/careful_filtered_ORFnucleotides.fa -s 06.GenePrediction/careful_filtered_genes
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="prodigal -f &lt;output_format&gt; -i &lt;input_assembly.fasta&gt; -o &lt;output_gbk_file&gt; -a &lt;output_protein_seq_file&gt; -d &lt;output_nucleotides_seq_file&gt; -s &lt;output_coordinates_file&gt;">
                     <pre class="notranslate"><code>prodigal -f &lt;output_format&gt; -i &lt;input_assembly.fasta&gt; -o &lt;output_gbk_file&gt; -a &lt;output_protein_seq_file&gt; -d &lt;output_nucleotides_seq_file&gt; -s &lt;output_coordinates_file&gt;
</code></pre>
                    </div>
                    <p dir="auto">
                     El archivo
                     <code>
                      06.GenePrediction/careful_filtered_orf.gff
                     </code>
                     incluyendo las
secuencias condificantes y no codificantes y las posiciones en cada
contig. Adicionalmente, el programa genera archivos con las secuencias
de nucletidos (
                     <code>
                      06.GenePrediction/careful_filtered_ORFnucleotides.fa
                     </code>
                     )
y protenas (
                     <code>
                      06.GenePrediction/careful_filtered_proteins.faa
                     </code>
                     ) de cada
gene predicho.
                    </p>
                    <p dir="auto">
                     <em>
                      Nota:
                     </em>
                     Ejecute el mismo comando en el genoma de
                     <em>
                      E. coli
                     </em>
                     ATCC8739. Use
                     <code>
                      ecoli_genome_
                     </code>
                     como prefijo para los nombres de los archivos de salida.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      V. Transcriptmica
                     </h1>
                     <a aria-label="Permalink: V. Transcriptmica" class="anchor" href="#v-transcriptmica" id="user-content-v-transcriptmica">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     En esta seccin sern abordadas las etapas y programas para la
evaluacin librerias de RNA-seq.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      5.1. Remocin de secuencias de rRNA
                     </h3>
                     <a aria-label="Permalink: 5.1. Remocin de secuencias de rRNA" class="anchor" href="#51-remocin-de-secuencias-de-rrna" id="user-content-51-remocin-de-secuencias-de-rrna">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Aunque durante el proceso de extraccin de RNA se use un kit para la
deplecin de RNA ribosomal, puede suceder que en las librerias de
RNA-seq an existan algunas secuencias de rRNA, las cuales podrn ser
eliminadas usando el programa
                     <a href="https://github.com/biocore/sortmerna">
                      SortmeRNA
                     </a>
                     , que a su vez usa
bases de dados de rRNA, tales como Rfam 5S, Rfam 5.8S,
SILVA_138.1_NR99_SSURef_tax_silva and
SILVA_138.1_LSURef_tax_silva, para buscar y eliminar esas secuencias
de las bibliotecas.
                    </p>
                    <p dir="auto">
                     <strong>
                      Instalacin
                     </strong>
                    </p>
                    <p dir="auto">
                     Instale SortmeRNA en un ambiente virtual llamado mapping
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# cree el ambiente
conda create -n mapping

# Active el ambiente
conda activate mappin

# Instale SortmeRNA
conda install -c bioconda sortmerna">
                     <pre class="notranslate"><code># cree el ambiente
conda create -n mapping

# Active el ambiente
conda activate mappin

# Instale SortmeRNA
conda install -c bioconda sortmerna
</code></pre>
                    </div>
                    <p dir="auto">
                     Las bases de datos son descargadas de las pginas de
                     <a href="https://www.arb-silva.de/" rel="nofollow">
                      SILVA
database
                     </a>
                     y
                     <a href="https://rfam.xfam.org/" rel="nofollow">
                      Rfam
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Uso
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for i in 00.RawData/*_1.fq
do
SAMPLE=$(basename $i _1.fq)
nohup sortmerna --ref /home/drm/Documentos/database/Rfam/Rfam_5S.fa --ref /home/drm/Documentos/database/Rfam/Rfam_5.8S.fa --ref /home/drm/Documentos/database/silva/SILVA_138.1_NR99_16S_28S.fasta --ref /home/drm/Documentos/database/silva/SILVA_138.1_23S_28S.fasta --reads $i --reads 00.RawData/${SAMPLE}_2.fq --paired_in --threads 50 --workdir 07.SortmeRNA/${SAMPLE}/ --idx-dir 07.SortmeRNA/${SAMPLE}/ --kvdb 07.SortmeRNA/${SAMPLE}/kvdb --fastx --aligned 07.SortmeRNA/${SAMPLE}/${SAMPLE}_aligned.fasta --other 07.SortmeRNA/${SAMPLE}/${SAMPLE}_other.fasta --out2
done">
                     <pre class="notranslate"><code>for i in 00.RawData/*_1.fq
do
SAMPLE=$(basename $i _1.fq)
nohup sortmerna --ref /home/drm/Documentos/database/Rfam/Rfam_5S.fa --ref /home/drm/Documentos/database/Rfam/Rfam_5.8S.fa --ref /home/drm/Documentos/database/silva/SILVA_138.1_NR99_16S_28S.fasta --ref /home/drm/Documentos/database/silva/SILVA_138.1_23S_28S.fasta --reads $i --reads 00.RawData/${SAMPLE}_2.fq --paired_in --threads 50 --workdir 07.SortmeRNA/${SAMPLE}/ --idx-dir 07.SortmeRNA/${SAMPLE}/ --kvdb 07.SortmeRNA/${SAMPLE}/kvdb --fastx --aligned 07.SortmeRNA/${SAMPLE}/${SAMPLE}_aligned.fasta --other 07.SortmeRNA/${SAMPLE}/${SAMPLE}_other.fasta --out2
done
</code></pre>
                    </div>
                    <p dir="auto">
                     El comando anterior, ejecuta SortmeRNA en todas las bibliotecas de
RNA-seq. Al final de proceso, cada muestra tendr una carpeta dentro del
directorio
                     <code>
                      07.SortmeRNA/
                     </code>
                     . Los archivos terminados en
                     <code>
                      _other_fwd.fq
                     </code>
                     y
                     <code>
                      _other_rev.fq
                     </code>
                     , contiene las secuencias que
                     <em>
                      no
                     </em>
                     alinearon contra las
bases de dados de rRNA, es decir libres de rRNA. Estos sern los
archivos que sern usados para los siguientes pasos.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      5.2. Alineamiento a um genoma de referencia
                     </h3>
                     <a aria-label="Permalink: 5.2. Alineamiento a um genoma de referencia" class="anchor" href="#52-alineamiento-a-um-genoma-de-referencia" id="user-content-52-alineamiento-a-um-genoma-de-referencia">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Despus de la evaluacin de la calidad y remocin de rRNA, se debe hacer
un alineamiento o mapeo de las lecturas para determinar en qu parte del
genoma o de dnde se originaron las lecturas. Hay una serie de
herramientas que realizan esta funcin. En este tutorial usaremos
                     <a href="https://github.com/DaehwanKimLab/hisat2">
                      HISAT2
                     </a>
                     , pero tambin una
herramienta como
                     <a href="https://github.com/alexdobin/STAR">
                      STAR
                     </a>
                     o
                     <a href="https://github.com/infphilo/tophat">
                      TopHat2
                     </a>
                     , hacen el trabajo.
                    </p>
                    <p dir="auto">
                     En este caso ser usado el genoma obtenido desde la pgina de la
coleccin ATCC, que debe estar almacenado en la carpeta
                     <code>
                      03.Assembly/
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     La primera etapa del proceso con
                     <a href="https://daehwankimlab.github.io/hisat2/" rel="nofollow">
                      HISAT2
                     </a>
                     , es llamada de
busqueda de semillas (
                     <em>
                      seeds
                     </em>
                     ), y consiste en que para cada read el
programa buscar la porcin de la lectura ms larga que coincida
exactamente con una o ms ubicaciones en el genoma de referencia
(
                     <em>
                      seed1
                     </em>
                     ). Posteriormente, volver a buscar la parte no mapeada de la
lectura para encontrar la siguiente secuencia ms larga que coincida
exactamente con el genoma de referencia (
                     <em>
                      seed2
                     </em>
                     ). Y as hasta que
encuentra las
                     <em>
                      seeds
                     </em>
                     de todas las lecturas. Esta forma de mapeo hace
que el algortmo sea eficiente y rpido. La segunda etapa, comprende la
clusterizacin de
                     <em>
                      seeds
                     </em>
                     prximas, para generar una read completa.
Despus, el programa unir las
                     <em>
                      seeds
                     </em>
                     baseado en el mejor alineamiento
de las reads (
                     <em>
                      score
                     </em>
                     de
                     <em>
                      mismatches
                     </em>
                     ,
                     <em>
                      indels
                     </em>
                     ,
                     <em>
                      gaps
                     </em>
                     , etc)
                    </p>
                    <p dir="auto">
                     <strong>
                      Instalao
                     </strong>
                    </p>
                    <p dir="auto">
                     HISAT2 v2.2.1 ser instalado en el ambiente virutal mapping. Tambin es
necesario instalar la herramienta
                     <a href="http://www.htslib.org/" rel="nofollow">
                      SamTools
                     </a>
                     ,
que sirve en la manipulacin de archivos
                     <code>
                      .sam
                     </code>
                     y
                     <code>
                      .bam
                     </code>
                     .
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Activa el ambiente
conda activate mapping

# Instala
conda install -c bioconda hisat2

# Instala SamTools
conda install -c bioconda samtools">
                     <pre class="notranslate"><code># Activa el ambiente
conda activate mapping

# Instala
conda install -c bioconda hisat2

# Instala SamTools
conda install -c bioconda samtools
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      5.2.1. Indexando el genoma de referencia
                     </h4>
                     <a aria-label="Permalink: 5.2.1. Indexando el genoma de referencia" class="anchor" href="#521-indexando-el-genoma-de-referencia" id="user-content-521-indexando-el-genoma-de-referencia">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     El primer paso para el alineamiento es indexar el genoma. Cree un
directorio para todo el proceso de mapeo (p.e.
                     <code>
                      08.Mapping.RNA
                     </code>
                     )
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Indexando
hisat2-build -p 10 03.Assembly/Escherichia_coli_ATCC_8739.fasta 08.Mapping.RNA/Escherichia_coli_ATCC_8739">
                     <pre class="notranslate"><code># Indexando
hisat2-build -p 10 03.Assembly/Escherichia_coli_ATCC_8739.fasta 08.Mapping.RNA/Escherichia_coli_ATCC_8739
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                     <code>
                      hisat2-build -p # [genome] [index]
                     </code>
                    </p>
                    <ul dir="auto">
                     <li>
                      <code>
                       -p
                      </code>
                      : nmero de threads
                     </li>
                     <li>
                      <code>
                       [genome]
                      </code>
                      : Genoma de referencia
                     </li>
                     <li>
                      <code>
                       [index]
                      </code>
                      : ndice
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      5.2.2. Mapeo
                     </h4>
                     <a aria-label="Permalink: 5.2.2. Mapeo" class="anchor" href="#522-mapeo" id="user-content-522-mapeo">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     En el primer comando
                     <strong>
                      HISAT2
                     </strong>
                     va a hacer el alinemiento, el cual
generar una serie de archivos
                     <code>
                      .sam
                     </code>
                     (
                     <em>
                      Sequence aligment Map
                     </em>
                     ). En el
segundo comando,
                     <strong>
                      SamTools
                     </strong>
                     transformar a archivos
                     <code>
                      .bam
                     </code>
                     (
                     <em>
                      Binary
Alignment Map
                     </em>
                     ), los cuales son una versin comprimida de
                     <code>
                      .sam
                     </code>
                     .
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="hisat2  --dta -p 15 -x 08.Mapping.RNA/Escherichia_coli_ATCC_8739 -1 00.RawData/acetone_1.fq -2 00.RawData/acetone_2.fq -S 08.Mapping.RNA/acetone.sam

# loop
for i in 00.RawData/*_1.fq
do
sample=$(basename $i _1.fq)
hisat2 --dta -p 15 -x 08.Mapping.RNA/Escherichia_coli_ATCC_8739 -1 $i -2 00.RawData/${sample}_2.fq -S 08.Mapping.RNA/${sample}.sam
done">
                     <pre class="notranslate"><code>hisat2  --dta -p 15 -x 08.Mapping.RNA/Escherichia_coli_ATCC_8739 -1 00.RawData/acetone_1.fq -2 00.RawData/acetone_2.fq -S 08.Mapping.RNA/acetone.sam

# loop
for i in 00.RawData/*_1.fq
do
sample=$(basename $i _1.fq)
hisat2 --dta -p 15 -x 08.Mapping.RNA/Escherichia_coli_ATCC_8739 -1 $i -2 00.RawData/${sample}_2.fq -S 08.Mapping.RNA/${sample}.sam
done
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                    </p>
                    <p dir="auto">
                     <code>
                      hisat2 --dta -p &lt;num_nucleos&gt; -x &lt;index&gt; -1 RNA-seqs_pair1 -2 RNA-seqs_pair2
                     </code>
                    </p>
                    <ul dir="auto">
                     <li>
                      <code>
                       --dta
                      </code>
                      : Genera un reporte que ser usado en el prximo passo.
                     </li>
                     <li>
                      <code>
                       -p
                      </code>
                      : Nmero de ncleos
                     </li>
                     <li>
                      <code>
                       -x
                      </code>
                      : ndice
                     </li>
                     <li>
                      <code>
                       -1
                      </code>
                      : Secuencias RNA-seq pair 1
                     </li>
                     <li>
                      <code>
                       -2
                      </code>
                      : Secuencias RNA-seq pair 2
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      5.2.3. Transformando los archivos del mapeo
                     </h4>
                     <a aria-label="Permalink: 5.2.3. Transformando los archivos del mapeo" class="anchor" href="#523-transformando-los-archivos-del-mapeo" id="user-content-523-transformando-los-archivos-del-mapeo">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     El primer paso es transformar el archivo de salida del mapeo (
                     <code>
                      .sam
                     </code>
                     )
para
                     <code>
                      .bam
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="samtools sort -o 08.Mapping.RNA/acetone.bam 08.Mapping.RNA/acetone.sam

# loop
for i in 08.Mapping.RNA/*.sam
do
sample=$(basename $i .sam)
samtools sort -o 08.Mapping.RNA/${sample}.bam $i
done">
                     <pre class="notranslate"><code>samtools sort -o 08.Mapping.RNA/acetone.bam 08.Mapping.RNA/acetone.sam

# loop
for i in 08.Mapping.RNA/*.sam
do
sample=$(basename $i .sam)
samtools sort -o 08.Mapping.RNA/${sample}.bam $i
done
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="samtools sort -o &lt;output.bam&gt; &lt;input.sam&gt;">
                     <pre class="notranslate"><code>samtools sort -o &lt;output.bam&gt; &lt;input.sam&gt;
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      5.3. Ensamble del transcriptoma
                     </h3>
                     <a aria-label="Permalink: 5.3. Ensamble del transcriptoma" class="anchor" href="#53-ensamble-del-transcriptoma" id="user-content-53-ensamble-del-transcriptoma">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     El objetivo de esta etapa es el montaje del transcriptoma, usando los
alineamientos del paso anterior. En este tutorial, usaremos
                     <a href="https://ccb.jhu.edu/software/stringtie/" rel="nofollow">
                      StringTie
                     </a>
                     , el cual emplea
algoritmos eficientes para la recuperacin de la estructura del
trasncriptoma y la estimacin de la abundancia a partir de lecturas de
RNA-Seq alineadas con un genoma de referencia. Toma como entrada los
alineamientos SAM/BAM, ordenados por coordenadas y produce una salida
GTF que consiste en transcritos ensamblados y sus niveles estimados
(FPKM / TPM y valores de cobertura).
                    </p>
                    <p dir="auto">
                     <strong>
                      Instalacin
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Activa el ambiente assembly
conda activate assembly

# Instale StringTie
conda install -c bioconda stringtie">
                     <pre class="notranslate"><code># Activa el ambiente assembly
conda activate assembly

# Instale StringTie
conda install -c bioconda stringtie
</code></pre>
                    </div>
                    <p dir="auto">
                     Cree un directorio para la salida del proceso:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cree un directorio para el montaje
mkdir 09.AssemblyRNA

# StringTie
stringtie acetone.bam -p 25 -G 06.GenePrediction/ecoli_genome_orf.gff -o 09.AssemblyRNA/acetone_stringtie.gtf

# Loop
for i in 08.Mapping.RNA/*.bam
do
sample=$(basename $i .bam)
stringtie $i -p 25 -G 06.GenePrediction/ecoli_genome_orf.gff -o 09.AssemblyRNA/${sample}_stringtie.gtf
done">
                     <pre class="notranslate"><code># Cree un directorio para el montaje
mkdir 09.AssemblyRNA

# StringTie
stringtie acetone.bam -p 25 -G 06.GenePrediction/ecoli_genome_orf.gff -o 09.AssemblyRNA/acetone_stringtie.gtf

# Loop
for i in 08.Mapping.RNA/*.bam
do
sample=$(basename $i .bam)
stringtie $i -p 25 -G 06.GenePrediction/ecoli_genome_orf.gff -o 09.AssemblyRNA/${sample}_stringtie.gtf
done
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Sintaxis
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="stringtie &lt;sorted.bam&gt; -p &lt;num_nucleos&gt; -G &lt;reference.genome.gff&gt; -o &lt;output.gtf&gt;  -l &lt;transcriptos_prefix&gt;">
                     <pre class="notranslate"><code>stringtie &lt;sorted.bam&gt; -p &lt;num_nucleos&gt; -G &lt;reference.genome.gff&gt; -o &lt;output.gtf&gt;  -l &lt;transcriptos_prefix&gt;
</code></pre>
                    </div>
                    <p dir="auto">
                     El archivo de salida
                     <code>
                      .gtf
                     </code>
                     contiene las definiciones estructurales de
los transcriptos ensamblados por
                     <strong>
                      StringTie
                     </strong>
                     a partir de la
informacin de mapeo.
                    </p>
                    <p dir="auto">
                     Use
                     <code>
                      less
                     </code>
                     para ver el contenido del archivo de salida
                     <code>
                      09.AssemblyRNA/acetone.gtf
                     </code>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      5.3.1. Transcriptoma final
                     </h3>
                     <a aria-label="Permalink: 5.3.1. Transcriptoma final" class="anchor" href="#531-transcriptoma-final" id="user-content-531-transcriptoma-final">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     A continuacin sern unidos los transcriptomas de cada muestra para
obtner el transcriptoma final. Para esto, ser creado un archivo
(
                     <code>
                      mergelist.txt
                     </code>
                     ) con el nombre de los
                     <em>
                      output
                     </em>
                     obtenidos en el comando
de StringTie.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Creando el archivo mergelist.txt
echo -e 'acetone_stringtie.gtf\noil1-2_R1_stringtie.gtf\noil1-2_R2_stringtie.gtf\noil1-2_R3_stringtie.gtf\noil1-4_R1_stringtie.gtf\noil1-4_R2_stringtie.gtf\noil1-4_R3_stringtie.gtf\noil1-8_R1_stringtie.gtf\noil1-8_R2_stringtie.gtf\noil1-8_R3_stringtie.gtf\nwater_R1_stringtie.gtf\nwater_R2_stringtie.gtf ' &gt; 09.AssemblyRNA/mergelist.txt ">
                     <pre class="notranslate"><code># Creando el archivo mergelist.txt
echo -e 'acetone_stringtie.gtf\noil1-2_R1_stringtie.gtf\noil1-2_R2_stringtie.gtf\noil1-2_R3_stringtie.gtf\noil1-4_R1_stringtie.gtf\noil1-4_R2_stringtie.gtf\noil1-4_R3_stringtie.gtf\noil1-8_R1_stringtie.gtf\noil1-8_R2_stringtie.gtf\noil1-8_R3_stringtie.gtf\nwater_R1_stringtie.gtf\nwater_R2_stringtie.gtf ' &gt; 09.AssemblyRNA/mergelist.txt 
</code></pre>
                    </div>
                    <p dir="auto">
                     Ejecute el comando
                     <code>
                      merge
                     </code>
                     de StringTie:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd 09.AssemblyRNA

stringtie --merge -G ../06.GenePrediction/ecoli_genome_orf.gff -o stringtie_merged.gtf mergelist.txt">
                     <pre class="notranslate"><code>cd 09.AssemblyRNA

stringtie --merge -G ../06.GenePrediction/ecoli_genome_orf.gff -o stringtie_merged.gtf mergelist.txt
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      5.3.2. Extrayendo las secuencias
                     </h3>
                     <a aria-label="Permalink: 5.3.2. Extrayendo las secuencias" class="anchor" href="#532-extrayendo-las-secuencias" id="user-content-532-extrayendo-las-secuencias">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Hasta aqui tenemos el archivo final de coordenadas genmicas
(
                     <code>
                      09.AssemblyRNA/stringtie_merged.gtf
                     </code>
                     ). En esta etapa usaremos ese
archivo para extraer las secuencias en un archivo
                     <code>
                      .fasta
                     </code>
                     , usando el
programa
                     <a href="https://github.com/gpertea/gffread">
                      GFFREAD
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Instalacin
                     </strong>
                    </p>
                    <p dir="auto">
                     Instale el programa en el ambiente bioinfo.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Active el ambiente
conda activate bioinfo

# Instale el ambiente
conda install -c bioconda gffread">
                     <pre class="notranslate"><code># Active el ambiente
conda activate bioinfo

# Instale el ambiente
conda install -c bioconda gffread
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Uso
                     </strong>
                    </p>
                    <p dir="auto">
                     Para extraer las secuencias, ejecute el siguiente comando:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd ..

gffread 09.AssemblyRNA/stringtie_merged.gtf -g 03.Assembly/Escherichia_coli_ATCC_8739.fasta -w 09.AssemblyRNA/all_transcripts.fasta">
                     <pre class="notranslate"><code>cd ..

gffread 09.AssemblyRNA/stringtie_merged.gtf -g 03.Assembly/Escherichia_coli_ATCC_8739.fasta -w 09.AssemblyRNA/all_transcripts.fasta
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.4. Cuantificacin de la expresin
                     </h2>
                     <a aria-label="Permalink: 5.4. Cuantificacin de la expresin" class="anchor" href="#54-cuantificacin-de-la-expresin" id="user-content-54-cuantificacin-de-la-expresin">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Despus de obtenidos los transcriptos, estos deben ser cuantificados. El
programa que ser usado en este tutorial, se llama
                     <a href="https://pachterlab.github.io/kallisto/" rel="nofollow">
                      Kallisto
                     </a>
                     . Este software
cuantifica las abundancias de transcriptos. Se basa en la novedosa idea
de pseudoalineacin para determinar rpidamente la compatibilidad de
lecturas con los
                     <em>
                      targets
                     </em>
                     , sin necesidad de alineacin.
                    </p>
                    <p dir="auto">
                     <strong>
                      Instalacin
                     </strong>
                    </p>
                    <p dir="auto">
                     Instale Kallisto el ambiente
                     <strong>
                      bioinfo
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Activa el ambiente bioinfo
conda activate bioinfo

# Instale Kallisto
conda install -c bioconda kallisto">
                     <pre class="notranslate"><code># Activa el ambiente bioinfo
conda activate bioinfo

# Instale Kallisto
conda install -c bioconda kallisto
</code></pre>
                    </div>
                    <p dir="auto">
                     Cree un directorio para el proceso de cuantificacin
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="mkdir 10.TranscriptsQuantification">
                     <pre class="notranslate"><code>mkdir 10.TranscriptsQuantification
</code></pre>
                    </div>
                    <p dir="auto">
                     El primer paso es indexar el archivo
                     <code>
                      09.AssemblyRNA/all_transcripts.fasta
                     </code>
                     :
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="kallisto index -i 09.AssemblyRNA/index_transcritos 09.AssemblyRNA/all_transcripts.fasta">
                     <pre class="notranslate"><code>kallisto index -i 09.AssemblyRNA/index_transcritos 09.AssemblyRNA/all_transcripts.fasta
</code></pre>
                    </div>
                    <p dir="auto">
                     A continuacin sern cuantificados los transcriptomas de cada libreria.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="kallisto quant -i 09.AssemblyRNA/index_transcritos -o 10.TranscriptsQuantification/acetone -b 1000 -t 20 00.RawData/acetone_1.fq 00.RawData/acetone_2.fq

# Loop
for i in 00.RawData/*_1.fq
do
sample=$(basename $i _1.fq)
kallisto quant -i 09.AssemblyRNA/index_transcritos -o 10.TranscriptsQuantification/${sample} -b 1000 -t 20 $i 00.RawData/${sample}_2.fq
done">
                     <pre class="notranslate"><code>kallisto quant -i 09.AssemblyRNA/index_transcritos -o 10.TranscriptsQuantification/acetone -b 1000 -t 20 00.RawData/acetone_1.fq 00.RawData/acetone_2.fq

# Loop
for i in 00.RawData/*_1.fq
do
sample=$(basename $i _1.fq)
kallisto quant -i 09.AssemblyRNA/index_transcritos -o 10.TranscriptsQuantification/${sample} -b 1000 -t 20 $i 00.RawData/${sample}_2.fq
done
</code></pre>
                    </div>
                    <p dir="auto">
                     Sern creadas carpetas para cada muestra y dentro se encuentra el
archivo de salida de
                     <strong>
                      Kallisto
                     </strong>
                     que es una tabla llamada
                     <code>
                      abundance.tsv
                     </code>
                     con el ID de cada transcrito y el valor de
                     <strong>
                      TPM
                     </strong>
                     para
cada uno de ellos, adems del tamao de cada uno.
                    </p>
                    <p dir="auto">
                     Use
                     <code>
                      less
                     </code>
                     para ver el contenido de la tabla
                     <code>
                      abundance.tsv
                     </code>
                     que se
encuentra dentro de la carpeta
                     <code>
                      12.TranscriptQuantification/
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     El siguiente paso es unir las tablas de abundancia de cada muestra y
construir la matriz de abundancia para iniciar el anlisis de expresin
diferencial. El script
                     <code>
                      abundance_estimates_to_matrix.pl
                     </code>
                     , de la
herramienta
                     <a href="https://github.com/trinityrnaseq/trinityrnaseq">
                      Trinity
                     </a>
                     permite la construccin de la matriz de abundancia. En este tutorial no
se usar como tal el programa Trinity, que es un ensamblador de RNA-seq
                     <em>
                      de novo
                     </em>
                     , recomendado para tratar datos de metatranscriptmica. Sin
embargo, es necesario descargar el repositorio de la herramienta:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="git clone https://github.com/trinityrnaseq/trinityrnaseq">
                     <pre class="notranslate"><code>git clone https://github.com/trinityrnaseq/trinityrnaseq
</code></pre>
                    </div>
                    <p dir="auto">
                     El comando anterior descargar una carpeta llamada
                     <code>
                      trinityrnaseq/
                     </code>
                     . El
script estar en el directorio
                     <code>
                      trinityrnaseq/util/
                     </code>
                    </p>
                    <p dir="auto">
                     Antes de ejecutar el script, es necesario modificar los nombres de las
tablas de abundancia de cada muestra, de manera que puedan ser
eficientemente diferenciados entre ellos.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cp 10.TranscriptsQuantification/acetone/abundance.tsv 10.TranscriptsQuantification/acetone.tsv

cp 10.TranscriptsQuantification/oil1-2_R1/abundance.tsv 10.TranscriptsQuantification/oil1-2_R1.tsv

cp 10.TranscriptsQuantification/oil1-2_R2/abundance.tsv 10.TranscriptsQuantification/oil1-2_R2.tsv

cp 10.TranscriptsQuantification/oil1-2_R3/abundance.tsv 10.TranscriptsQuantification/oil1-2_R3.tsv

cp 10.TranscriptsQuantification/oil1-4_R1/abundance.tsv 10.TranscriptsQuantification/oil1-4_R1.tsv

cp 10.TranscriptsQuantification/oil1-4_R2/abundance.tsv 10.TranscriptsQuantification/oil1-4_R2.tsv

cp 10.TranscriptsQuantification/oil1-4_R3/abundance.tsv 10.TranscriptsQuantification/oil1-4_R3.tsv

cp 10.TranscriptsQuantification/oil1-8_R1/abundance.tsv 10.TranscriptsQuantification/oil1-8_R1.tsv

cp 10.TranscriptsQuantification/oil1-8_R2/abundance.tsv 10.TranscriptsQuantification/oil1-8_R2.tsv

cp 10.TranscriptsQuantification/oil1-8_R3/abundance.tsv 10.TranscriptsQuantification/oil1-8_R3.tsv

cp 10.TranscriptsQuantification/water_R1/abundance.tsv 10.TranscriptsQuantification/water_R1.tsv

cp 10.TranscriptsQuantification/water_R2/abundance.tsv 10.TranscriptsQuantification/water_R2.tsv">
                     <pre class="notranslate"><code>cp 10.TranscriptsQuantification/acetone/abundance.tsv 10.TranscriptsQuantification/acetone.tsv

cp 10.TranscriptsQuantification/oil1-2_R1/abundance.tsv 10.TranscriptsQuantification/oil1-2_R1.tsv

cp 10.TranscriptsQuantification/oil1-2_R2/abundance.tsv 10.TranscriptsQuantification/oil1-2_R2.tsv

cp 10.TranscriptsQuantification/oil1-2_R3/abundance.tsv 10.TranscriptsQuantification/oil1-2_R3.tsv

cp 10.TranscriptsQuantification/oil1-4_R1/abundance.tsv 10.TranscriptsQuantification/oil1-4_R1.tsv

cp 10.TranscriptsQuantification/oil1-4_R2/abundance.tsv 10.TranscriptsQuantification/oil1-4_R2.tsv

cp 10.TranscriptsQuantification/oil1-4_R3/abundance.tsv 10.TranscriptsQuantification/oil1-4_R3.tsv

cp 10.TranscriptsQuantification/oil1-8_R1/abundance.tsv 10.TranscriptsQuantification/oil1-8_R1.tsv

cp 10.TranscriptsQuantification/oil1-8_R2/abundance.tsv 10.TranscriptsQuantification/oil1-8_R2.tsv

cp 10.TranscriptsQuantification/oil1-8_R3/abundance.tsv 10.TranscriptsQuantification/oil1-8_R3.tsv

cp 10.TranscriptsQuantification/water_R1/abundance.tsv 10.TranscriptsQuantification/water_R1.tsv

cp 10.TranscriptsQuantification/water_R2/abundance.tsv 10.TranscriptsQuantification/water_R2.tsv
</code></pre>
                    </div>
                    <p dir="auto">
                     Ejecute el script
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Active el ambiente bioinfo
conda activate bioinfo

# Entre al directorio 10
cd 10.TranscriptsQuantification/

# Ejecute el script
perl trinityrnaseq/util/abundance_estimates_to_matrix.pl --est_method kallisto --gene_trans_map none acetone.tsv oil1-2_R1.tsv oil1-2_R2.tsv oil1-2_R3.tsv oil1-4_R1.tsv oil1-4_R2.tsv oil1-4_R3.tsv oil1-8_R1.tsv oil1-8_R2.tsv oil1-8_R3.tsv water_R1.tsv water_R2.tsv">
                     <pre class="notranslate"><code># Active el ambiente bioinfo
conda activate bioinfo

# Entre al directorio 10
cd 10.TranscriptsQuantification/

# Ejecute el script
perl trinityrnaseq/util/abundance_estimates_to_matrix.pl --est_method kallisto --gene_trans_map none acetone.tsv oil1-2_R1.tsv oil1-2_R2.tsv oil1-2_R3.tsv oil1-4_R1.tsv oil1-4_R2.tsv oil1-4_R3.tsv oil1-8_R1.tsv oil1-8_R2.tsv oil1-8_R3.tsv water_R1.tsv water_R2.tsv
</code></pre>
                    </div>
                    <p dir="auto">
                     Los archivos de salida son:
                    </p>
                    <ul dir="auto">
                     <li>
                      <code>
                       kallisto.isoform.TMM.EXPR.matrix
                      </code>
                      : matriz normalizada por TMM
                     </li>
                     <li>
                      <code>
                       kallisto.isoform.TPM.not_cross_norm
                      </code>
                      : matriz de TPM
                     </li>
                     <li>
                      <code>
                       kallisto.isoform.counts.matrix
                      </code>
                      : matriz de counts
                     </li>
                     <li>
                      <code>
                       kallisto.isoform.TPM.not_cross_norm.TMM_info.txt
                      </code>
                      : estadstica de
los factores de normalizacin por profundidad de la secuenciacin
                     </li>
                     <li>
                      <code>
                       kallisto.isoform.TPM.not_cross_norm.runTMM.r
                      </code>
                      : script en R para
realizar la normalizacin en TMM.
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.5. Expresin diferencial
                     </h2>
                     <a aria-label="Permalink: 5.5. Expresin diferencial" class="anchor" href="#55-expresin-diferencial" id="user-content-55-expresin-diferencial">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     El anlisis de expresin diferencial consiste en comparar la expresin
de los genes en diferentes condiciones e identificar genes que son
bastante ms o menos expresados.
                    </p>
                    <p dir="auto">
                     Este anlisis puede ser realizado con diferentes bibliotecas del
software R, tales como
                     <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html" rel="nofollow">
                      edgeR
                     </a>
                     ,
                     <a href="https://github.com/mikelove/DESeq2">
                      DESeq2
                     </a>
                     y
                     <a href="https://github.com/pachterlab/sleuth">
                      Sleuth
                     </a>
                     . Aqui sern abordado el
anlisis con DESeq2. Este anlisis ser ejecutado en el ambiente
RStudio.
                    </p>
                    <blockquote>
                     <div class="markdown-heading" dir="auto">
                      <h1 class="heading-element" dir="auto" tabindex="-1">
                       Pausa para una brevisiiima introduccin al lenguaje R
                      </h1>
                      <a aria-label="Permalink: Pausa para una brevisiiima introduccin al lenguaje R" class="anchor" href="#pausa-para-una-brevisiiima-introduccin-al-lenguaje-r" id="user-content-pausa-para-una-brevisiiima-introduccin-al-lenguaje-r">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <div class="markdown-heading" dir="auto">
                      <h2 class="heading-element" dir="auto" tabindex="-1">
                       1. Instalacin
                      </h2>
                      <a aria-label="Permalink: 1. Instalacin" class="anchor" href="#1-instalacin" id="user-content-1-instalacin">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <p dir="auto">
                      Para instalar R, de click
                      <a href="https://vps.fmvz.usp.br/CRAN/" rel="nofollow">
                       aqui
                      </a>
                      . En
seguida instele
&gt;
                      <a href="https://www.rstudio.com/products/rstudio/download/#download" rel="nofollow">
                       RStudio
                      </a>
                      .
                     </p>
                     <a href="/khidalgo85/training_transcriptomics/blob/master/imgs/R.png" rel="noopener noreferrer" target="_blank">
                      <img align="center" src="/khidalgo85/training_transcriptomics/raw/master/imgs/R.png" style="max-width: 100%;"/>
                     </a>
                     <p dir="auto">
                      RStudio debe lucir parecido a la imagen anterior, con cuatro paneles:
i) Editor de texto/cdigos, ii) consola, iii) Espacio de trabajo e
historial y iv) Plots, archivos y men de ayuda.
                     </p>
                     <div class="markdown-heading" dir="auto">
                      <h2 class="heading-element" dir="auto" tabindex="-1">
                       2. Instalacin, actualizacin y remocin de librerias
                      </h2>
                      <a aria-label="Permalink: 2. Instalacin, actualizacin y remocin de librerias" class="anchor" href="#2-instalacin-actualizacin-y-remocin-de-librerias" id="user-content-2-instalacin-actualizacin-y-remocin-de-librerias">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <ul dir="auto">
                      <li>
                       <strong>
                        install.packages()
                       </strong>
                       : Instala librerias desde el repositorio
CRAN.
                       <strong>
                        Nota:
                       </strong>
                       Existen otras maneras de instalar librerias cuando
stas no se encuentran en CRAM, p.e. GitHub, BiocLite, devtools.
                      </li>
                     </ul>
                    </blockquote>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="install.packages('ggplot2')
install.packages('tidyverse')" dir="auto">
                     <pre>install.packages(<span class="pl-s"><span class="pl-pds">'</span>ggplot2<span class="pl-pds">'</span></span>)
install.packages(<span class="pl-s"><span class="pl-pds">'</span>tidyverse<span class="pl-pds">'</span></span>)</pre>
                    </div>
                    <blockquote>
                     <ul dir="auto">
                      <li>
                       Informacin de ayuda
                      </li>
                     </ul>
                    </blockquote>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="help(package = 'ggplot2')">
                     <pre class="notranslate"><code>help(package = 'ggplot2')
</code></pre>
                    </div>
                    <blockquote>
                     <div class="markdown-heading" dir="auto">
                      <h2 class="heading-element" dir="auto" tabindex="-1">
                       3. Tipos de datos, constantes y variables
                      </h2>
                      <a aria-label="Permalink: 3. Tipos de datos, constantes y variables" class="anchor" href="#3-tipos-de-datos-constantes-y-variables" id="user-content-3-tipos-de-datos-constantes-y-variables">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <p dir="auto">
                      <strong>
                       Tipos de datos
                      </strong>
                     </p>
                     <ul dir="auto">
                      <li>
                       <p dir="auto">
                        Nmericos
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        Enteros
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        Reales
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        Imaginrios
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        Caracteres/string o texto
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        Texto
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        Lgicos
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        TRUE
                       </p>
                      </li>
                      <li>
                       <p dir="auto">
                        FALSE
                       </p>
                      </li>
                     </ul>
                     <p dir="auto">
                      <strong>
                       Asignacin de valores a variables
                      </strong>
                     </p>
                    </blockquote>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="a &lt;- 3
b &lt;- 2
a + b 
#&gt; [1] 5" dir="auto">
                     <pre><span class="pl-smi">a</span> <span class="pl-k">&lt;-</span> <span class="pl-c1">3</span>
<span class="pl-smi">b</span> <span class="pl-k">&lt;-</span> <span class="pl-c1">2</span>
<span class="pl-smi">a</span> <span class="pl-k">+</span> <span class="pl-smi">b</span> 
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] 5</span></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Tipos bsicos de datos:
                      </strong>
                      nmericos, caracteres y lgicos
                     </p>
                    </blockquote>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='a &lt;- 3 
class(a)
#&gt; [1] "numeric"' dir="auto">
                     <pre><span class="pl-smi">a</span> <span class="pl-k">&lt;-</span> <span class="pl-c1">3</span> 
class(<span class="pl-smi">a</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] "numeric"</span></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='nombre &lt;- "Maria"
class(nombre)
#&gt; [1] "character"' dir="auto">
                     <pre><span class="pl-smi">nombre</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>Maria<span class="pl-pds">"</span></span>
class(<span class="pl-smi">nombre</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] "character"</span></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='class(TRUE)
#&gt; [1] "logical"' dir="auto">
                     <pre>class(<span class="pl-c1">TRUE</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] "logical"</span></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Dando nombres a las variables
                      </strong>
                     </p>
                     <p dir="auto">
                      Se pueden usar letras, nmeros y underline.
                     </p>
                     <p dir="auto">
                      <strong>
                       Reglas
                      </strong>
                     </p>
                     <ul dir="auto">
                      <li>
                       No puede comenzar por nmero
                      </li>
                      <li>
                       Puede comenzar por punto seguido de una letra o secuencia de
letras.
                      </li>
                      <li>
                       No puede comenzar por punto seguido de nmeros.
                      </li>
                     </ul>
                     <p dir="auto">
                      <strong>
                       NOTA
                      </strong>
                      R es un lenguaje
                      <em>
                       case sensitive
                      </em>
                      , o sea es sensible a
mayscula y minscula.
                     </p>
                     <div class="markdown-heading" dir="auto">
                      <h2 class="heading-element" dir="auto" tabindex="-1">
                       4. Operadores y funciones matemticas
                      </h2>
                      <a aria-label="Permalink: 4. Operadores y funciones matemticas" class="anchor" href="#4-operadores-y-funciones-matemticas" id="user-content-4-operadores-y-funciones-matemticas">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <table>
                      <thead>
                       <tr>
                        <th>
                         <strong>
                          Operao
                         </strong>
                        </th>
                        <th>
                         <strong>
                          Descrio
                         </strong>
                        </th>
                       </tr>
                      </thead>
                      <tbody>
                       <tr>
                        <td>
                         x + y
                        </td>
                        <td>
                         Adicin
                        </td>
                       </tr>
                       <tr>
                        <td>
                         x - y
                        </td>
                        <td>
                         Substraccin
                        </td>
                       </tr>
                       <tr>
                        <td>
                         x * y
                        </td>
                        <td>
                         Multiplicacin
                        </td>
                       </tr>
                       <tr>
                        <td>
                         x / y
                        </td>
                        <td>
                         Divisin
                        </td>
                       </tr>
                       <tr>
                        <td>
                         x ^ 2 (x **y)
                        </td>
                        <td>
                         Ptenciacin o exponenciacin
                        </td>
                       </tr>
                       <tr>
                        <td>
                         x %% y
                        </td>
                        <td>
                         Resto de la divisin
                        </td>
                       </tr>
                       <tr>
                        <td>
                         x %/% y
                        </td>
                        <td>
                         Cosciente (Parte Entera)
                        </td>
                       </tr>
                      </tbody>
                     </table>
                    </blockquote>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="17/2 #Divisin exacta
#&gt; [1] 8.5
17%/%2 #Parte exacta
#&gt; [1] 8
17%%2 # Resto de la divisin
#&gt; [1] 1" dir="auto">
                     <pre><span class="pl-c1">17</span><span class="pl-k">/</span><span class="pl-c1">2</span> <span class="pl-c"><span class="pl-c">#</span>Divisin exacta</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] 8.5</span>
<span class="pl-c1">17</span><span class="pl-k">%/%</span><span class="pl-c1">2</span> <span class="pl-c"><span class="pl-c">#</span>Parte exacta</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] 8</span>
<span class="pl-c1">17</span><span class="pl-k">%%</span><span class="pl-c1">2</span> <span class="pl-c"><span class="pl-c">#</span> Resto de la divisin</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] 1</span></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Funciones aritmticas y trigonomtricas bsicas
                      </strong>
                     </p>
                    </blockquote>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="log2(9)
#&gt; [1] 3.169925
sqrt(9)
#&gt; [1] 3
cos(60)
#&gt; [1] -0.952413" dir="auto">
                     <pre>log2(<span class="pl-c1">9</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] 3.169925</span>
sqrt(<span class="pl-c1">9</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] 3</span>
cos(<span class="pl-c1">60</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] -0.952413</span></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       Otras funciones
                      </em>
                      <code>
                       sin(), tan()
                      </code>
                     </p>
                     <div class="markdown-heading" dir="auto">
                      <h2 class="heading-element" dir="auto" tabindex="-1">
                       5. Estructura de datos
                      </h2>
                      <a aria-label="Permalink: 5. Estructura de datos" class="anchor" href="#5-estructura-de-datos" id="user-content-5-estructura-de-datos">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <div class="markdown-heading" dir="auto">
                      <h3 class="heading-element" dir="auto" tabindex="-1">
                       5.1. Vectores
                      </h3>
                      <a aria-label="Permalink: 5.1. Vectores" class="anchor" href="#51-vectores" id="user-content-51-vectores">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <p dir="auto">
                      Es la combinacin de mltiples valores (nmericos, caracteres o
lgicos)
                     </p>
                    </blockquote>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="mivector &lt;- c(12,32,7,89) 
class(mivector) #tipo de informacin contenida en el vector
#&gt; [1] &quot;numeric&quot;
length(mivector) # tamao del vector
#&gt; [1] 4
letras &lt;- c('a','b','c','d')
class(letras)
#&gt; [1] &quot;character&quot;
length(letras)
#&gt; [1] 4
vector.logico &lt;- c(FALSE,FALSE,TRUE,F,T)
class(vector.logico)
#&gt; [1] &quot;logical&quot;
length(vector.logico)
#&gt; [1] 5" dir="auto">
                     <pre><span class="pl-smi">mivector</span> <span class="pl-k">&lt;-</span> c(<span class="pl-c1">12</span>,<span class="pl-c1">32</span>,<span class="pl-c1">7</span>,<span class="pl-c1">89</span>) 
class(<span class="pl-smi">mivector</span>) <span class="pl-c"><span class="pl-c">#</span>tipo de informacin contenida en el vector</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] "numeric"</span>
length(<span class="pl-smi">mivector</span>) <span class="pl-c"><span class="pl-c">#</span> tamao del vector</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] 4</span>
<span class="pl-smi">letras</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">'</span>a<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>b<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>c<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>d<span class="pl-pds">'</span></span>)
class(<span class="pl-smi">letras</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] "character"</span>
length(<span class="pl-smi">letras</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] 4</span>
<span class="pl-smi">vector.logico</span> <span class="pl-k">&lt;-</span> c(<span class="pl-c1">FALSE</span>,<span class="pl-c1">FALSE</span>,<span class="pl-c1">TRUE</span>,<span class="pl-c1">F</span>,<span class="pl-c1">T</span>)
class(<span class="pl-smi">vector.logico</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] "logical"</span>
length(<span class="pl-smi">vector.logico</span>)
<span class="pl-c"><span class="pl-c">#</span>&gt; [1] 5</span></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <strong>
                       Formas de crear vectores numricos
                      </strong>
                     </p>
                     <ul dir="auto">
                      <li>
                       <ol dir="auto">
                        <li>
                         funcin
                         <code>
                          c()
                         </code>
                         concatenar
                        </li>
                       </ol>
                      </li>
                      <li>
                       <ol dir="auto" start="2">
                        <li>
                         Operador
                         <code>
                          :
                         </code>
                        </li>
                       </ol>
                      </li>
                      <li>
                       <ol dir="auto" start="3">
                        <li>
                         funcin
                         <code>
                          seq()
                         </code>
                        </li>
                       </ol>
                      </li>
                      <li>
                       <ol dir="auto" start="4">
                        <li>
                         funcin
                         <code>
                          rep()
                         </code>
                        </li>
                       </ol>
                      </li>
                     </ul>
                     <div class="markdown-heading" dir="auto">
                      <h3 class="heading-element" dir="auto" tabindex="-1">
                       5.2. Matrices
                      </h3>
                      <a aria-label="Permalink: 5.2. Matrices" class="anchor" href="#52-matrices" id="user-content-52-matrices">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <p dir="auto">
                      Es la combinacin de mltiples vectores, con el mismo tipo de datos
(nmericos, caracteres o lgicos).
                     </p>
                    </blockquote>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="m1 &lt;- matrix(data = c(2,5,7,8,21,32,22,11), nrow=4,
             ncol = 2, byrow = TRUE)
m1
#&gt;      [,1] [,2]
#&gt; [1,]    2    5
#&gt; [2,]    7    8
#&gt; [3,]   21   32
#&gt; [4,]   22   11" dir="auto">
                     <pre><span class="pl-smi">m1</span> <span class="pl-k">&lt;-</span> <span class="pl-k">matrix</span>(<span class="pl-v">data</span> <span class="pl-k">=</span> c(<span class="pl-c1">2</span>,<span class="pl-c1">5</span>,<span class="pl-c1">7</span>,<span class="pl-c1">8</span>,<span class="pl-c1">21</span>,<span class="pl-c1">32</span>,<span class="pl-c1">22</span>,<span class="pl-c1">11</span>), <span class="pl-v">nrow</span><span class="pl-k">=</span><span class="pl-c1">4</span>,
             <span class="pl-v">ncol</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, <span class="pl-v">byrow</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
<span class="pl-smi">m1</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;      [,1] [,2]</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; [1,]    2    5</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; [2,]    7    8</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; [3,]   21   32</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; [4,]   22   11</span></pre>
                    </div>
                    <blockquote>
                     <div class="markdown-heading" dir="auto">
                      <h3 class="heading-element" dir="auto" tabindex="-1">
                       5.3. Data Frames
                      </h3>
                      <a aria-label="Permalink: 5.3. Data Frames" class="anchor" href="#53-data-frames" id="user-content-53-data-frames">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <p dir="auto">
                      Matrices y vectores poseen una limitacin, solo guardan un tipo de
informacin (lgicas, numricas, texto). Mientras que los data frame
pueden contener todo tipo de informacin en un solo objeto. El data
frame es bidimensional, en lneas y columnas, y en cada columna puede
tener un tipo de informacin. Normalmente, cada colunma es una
variable y las lneas contienen los valores.
                     </p>
                     <p dir="auto">
                      Algunas caractersticas son:
                     </p>
                     <ul dir="auto">
                      <li>
                       Las columnas tendrn siempre nombres
                      </li>
                      <li>
                       Los nombres de las lneas deben ser nicos
                      </li>
                      <li>
                       Cada columna debe presentar un tipo nico de datos
                      </li>
                      <li>
                       Cada columna debe tener el mismo nmero de lneas
                      </li>
                     </ul>
                    </blockquote>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="df &lt;- data.frame(nombres=c('Ana','Juan','Vera'), notas=c(10,2,7.5), situacion=c(T,F,T), row.names = c('l1','l2','l3'))
View(df)
str(df) # presenta la estructura de los datos
#&gt; 'data.frame':    3 obs. of  3 variables:
#&gt;  $ nombres  : chr  &quot;Ana&quot; &quot;Juan&quot; &quot;Vera&quot;
#&gt;  $ notas    : num  10 2 7.5
#&gt;  $ situacion: logi  TRUE FALSE TRUE" dir="auto">
                     <pre><span class="pl-smi">df</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(<span class="pl-v">nombres</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">'</span>Ana<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>Juan<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>Vera<span class="pl-pds">'</span></span>), <span class="pl-v">notas</span><span class="pl-k">=</span>c(<span class="pl-c1">10</span>,<span class="pl-c1">2</span>,<span class="pl-c1">7.5</span>), <span class="pl-v">situacion</span><span class="pl-k">=</span>c(<span class="pl-c1">T</span>,<span class="pl-c1">F</span>,<span class="pl-c1">T</span>), <span class="pl-v">row.names</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">'</span>l1<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>l2<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>l3<span class="pl-pds">'</span></span>))
View(<span class="pl-smi">df</span>)
str(<span class="pl-smi">df</span>) <span class="pl-c"><span class="pl-c">#</span> presenta la estructura de los datos</span>
<span class="pl-c"><span class="pl-c">#</span>&gt; 'data.frame':    3 obs. of  3 variables:</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;  $ nombres  : chr  "Ana" "Juan" "Vera"</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;  $ notas    : num  10 2 7.5</span>
<span class="pl-c"><span class="pl-c">#</span>&gt;  $ situacion: logi  TRUE FALSE TRUE</span></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      Esta fue la brevisiima introduccin al lenguaje R. Para mayores
detalles y profundidad de los conocimientos en este lenguaje, consulte
otros materiales.
                     </p>
                    </blockquote>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.5. Expresin diferencial (continuacin)
                     </h2>
                     <a aria-label="Permalink: 5.5. Expresin diferencial (continuacin)" class="anchor" href="#55-expresin-diferencial-continuacin" id="user-content-55-expresin-diferencial-continuacin">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Compacte las carpetas
                     <code>
                      10.TranscriptsQuantification/
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="zip -r 10.TranscriptsQuantification.zip 10.TranscriptsQuantification/">
                     <pre class="notranslate"><code>zip -r 10.TranscriptsQuantification.zip 10.TranscriptsQuantification/
</code></pre>
                    </div>
                    <p dir="auto">
                     Descargue la carpeta a su computador usando el comando
                     <code>
                      scp
                     </code>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="scp user@direccionip:/home/user/transcriptomica/10.TranscriptsQuantification.zip camino/en/su/computador">
                     <pre class="notranslate"><code>scp user@direccionip:/home/user/transcriptomica/10.TranscriptsQuantification.zip camino/en/su/computador
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.6. Anotacin Funcional
                     </h2>
                     <a aria-label="Permalink: 5.6. Anotacin Funcional" class="anchor" href="#56-anotacin-funcional" id="user-content-56-anotacin-funcional">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Antes de hacer el anlisis de expresin diferencial, los transcriptos
deben ser anotados, para conocer su funcin, para lo cual los
transcriptos sern funcionalmente anotados por homologa, usando bases
de datos e um alineador.
                    </p>
                    <p dir="auto">
                     Para el alineamiento ser usado el programa
                     <a href="https://github.com/bbuchfink/diamond/wiki/3.-Command-line-options">
                      Diamond
                     </a>
                     ,
e una base de dayos curada como
                     <a href="https://www.kegg.jp/" rel="nofollow">
                      KEGG
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Instalacin
                     </strong>
                    </p>
                    <p dir="auto">
                     Instale Diamond en un ambiente virtual llamado diamond
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cree el ambiente
conda create -n diamond

# Active el ambiente
conda activate diamond

# Instale Diamond
conda install -c bioconda diamond">
                     <pre class="notranslate"><code># Cree el ambiente
conda create -n diamond

# Active el ambiente
conda activate diamond

# Instale Diamond
conda install -c bioconda diamond
</code></pre>
                    </div>
                    <p dir="auto">
                     Descargue las bases de datos
                     <a href="https://drive.google.com/file/d/1ZxjJdwh1izP32X5CH-B8SN0DK2WAAAvr/view?usp=sharing" rel="nofollow">
                      aqui
                     </a>
                     .
Una vez descargadas las bases de datos, debe subir los archivos al
servidor y almacenarlos en una carpeta llamada
                     <code>
                      dbs
                     </code>
                    </p>
                    <p dir="auto">
                     Use o siguiente comando para ejecutar Diamond:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cree un directorio para la salida
mkdir 11.FunctionalAnnotation

# Diamond
diamond blastx --more-sensitive --threads 50 -k 1 -f 6 qseqid qlen sseqid sallseqid slen qstart qend sstart send evalue bitscore score length pident qcovhsp --id 60 --query-cover 60 -d dbs/keggdb.dmnd --query 09.AssemblyRNA/all_transcripts.fasta -o 11.FunctionalAnnotation/kegg.txt --tmpdir /dev/shm">
                     <pre class="notranslate"><code># Cree un directorio para la salida
mkdir 11.FunctionalAnnotation

# Diamond
diamond blastx --more-sensitive --threads 50 -k 1 -f 6 qseqid qlen sseqid sallseqid slen qstart qend sstart send evalue bitscore score length pident qcovhsp --id 60 --query-cover 60 -d dbs/keggdb.dmnd --query 09.AssemblyRNA/all_transcripts.fasta -o 11.FunctionalAnnotation/kegg.txt --tmpdir /dev/shm
</code></pre>
                    </div>
                    <p dir="auto">
                     El archivo de
                     <em>
                      output
                     </em>
                     es una tabla que contiene los cdigos de los
genes y las anotaciones (nmero KEGG), adems de otras informaciones
relacionadas al alineamiento de cada secuencia (i.e.cobertura, % de
identidad). Debido a que los genes fueron anotados apenas con los
nmeros KEGG, es necessario cruzar esta informacin con las anotaciones
detalladas para cada nmero KEGG, que indican la funcin y categoria de
los genes. A continuacin encontrar una srie de pasos para obtener la
tabla final de anotacin funcional:
                    </p>
                    <ol dir="auto">
                     <li>
                      Separar la colunma de los cdigos KEGG
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="perl -pe 's/\|?(?:\s+gi|ref)?\|\s*/\t/g' 11.FunctionalAnnotation/kegg.txt &gt; 11.FunctionalAnnotation/kegg_formated.txt">
                     <pre class="notranslate"><code>perl -pe 's/\|?(?:\s+gi|ref)?\|\s*/\t/g' 11.FunctionalAnnotation/kegg.txt &gt; 11.FunctionalAnnotation/kegg_formated.txt
</code></pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      Cortar las columnas 1 (Contig ID) y 4 (KEGG ID) e grabar en un nuevo
archivo
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cut -f1,4 11.FunctionalAnnotation/kegg_formated.txt &gt; 11.FunctionalAnnotation/kegg_ID_contigs_keggs.tsv">
                     <pre class="notranslate"><code>cut -f1,4 11.FunctionalAnnotation/kegg_formated.txt &gt; 11.FunctionalAnnotation/kegg_ID_contigs_keggs.tsv
</code></pre>
                    </div>
                    <ol dir="auto" start="3">
                     <li>
                      Ordenando
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="sort -k2,2 11.FunctionalAnnotation/kegg_ID_contigs_keggs.tsv &gt; 11.FunctionalAnnotation/kegg_ID_contigs_keggs_sorted.tsv">
                     <pre class="notranslate"><code>sort -k2,2 11.FunctionalAnnotation/kegg_ID_contigs_keggs.tsv &gt; 11.FunctionalAnnotation/kegg_ID_contigs_keggs_sorted.tsv
</code></pre>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      Descargue el archivo con las informaciones de los cdigos KEGG
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="curl -L https://figshare.com/ndownloader/files/33953774 -o kegg.tsv

# ordene
sort -k1,1 kegg.tsv &gt; kegg_sorted.tsv

# mueva el archivo para 11.FunctionalAnnotation/
mv kegg_sorted.tsv 11.FunctionalAnnotation">
                     <pre class="notranslate"><code>curl -L https://figshare.com/ndownloader/files/33953774 -o kegg.tsv

# ordene
sort -k1,1 kegg.tsv &gt; kegg_sorted.tsv

# mueva el archivo para 11.FunctionalAnnotation/
mv kegg_sorted.tsv 11.FunctionalAnnotation
</code></pre>
                    </div>
                    <ol dir="auto" start="6">
                     <li>
                      Asigne nombres a las columnas
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='echo -e "target_id\tkegg_id\tlevel1\tlevel2\tlevel3\tGenName" &gt; 11.FunctionalAnnotation/kegg_randomic_analysis.tsv'>
                     <pre class="notranslate"><code>echo -e "target_id\tkegg_id\tlevel1\tlevel2\tlevel3\tGenName" &gt; 11.FunctionalAnnotation/kegg_randomic_analysis.tsv
</code></pre>
                    </div>
                    <ol dir="auto" start="5">
                     <li>
                      Asignacin randmica
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cat 11.FunctionalAnnotation/kegg_ID_contigs_keggs_sorted.tsv | while read line ; do echo &quot;$line&quot; | join -1 2 -2 1 -e&quot;NA&quot; -o1.1,0,2.2,2.3,2.4,2.5 -t $'\t' - 11.FunctionalAnnotation/kegg_sorted.tsv | shuf -n1 &gt;&gt; 11.FunctionalAnnotation/kegg_randomic_analysis.tsv ; done">
                     <pre class="notranslate"><code>cat 11.FunctionalAnnotation/kegg_ID_contigs_keggs_sorted.tsv | while read line ; do echo "$line" | join -1 2 -2 1 -e"NA" -o1.1,0,2.2,2.3,2.4,2.5 -t $'\t' - 11.FunctionalAnnotation/kegg_sorted.tsv | shuf -n1 &gt;&gt; 11.FunctionalAnnotation/kegg_randomic_analysis.tsv ; done
</code></pre>
                    </div>
                    <ol dir="auto" start="6">
                     <li>
                      Asigne nombres a las columnas
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='echo -e "target_id\tkegg_id\tlevel1\tlevel2\tlevel3\tGenName" &gt; transcripts_annotations.tsv'>
                     <pre class="notranslate"><code>echo -e "target_id\tkegg_id\tlevel1\tlevel2\tlevel3\tGenName" &gt; transcripts_annotations.tsv
</code></pre>
                    </div>
                    <ol dir="auto" start="7">
                     <li>
                      Descargue y explore a tabla final de anotaciones
                      <code>
                       11.FunctionalAnnotation/kegg_randomic_analysis.tsv
                      </code>
                      .
                     </li>
                    </ol>
                    <p dir="auto">
                     Guarde el archivo en la carpeta
                     <code>
                      10.TranscriptsQuantification/
                     </code>
                    </p>
                    <p dir="auto">
                     En este punto puede cerrar su conexin con el servidor y trabajar en el
ambiente de RStudio localmente.
                    </p>
                    <p dir="auto">
                     En su computador, cree un directorio de trabajo y en l, descompacte las
carpetas descargadas del servidor.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      RStudio
                     </h1>
                     <a aria-label="Permalink: RStudio" class="anchor" href="#rstudio" id="user-content-rstudio">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Abra RStudio, cree un nuevo proyecto: File &lt; New Project &lt;
Existing Directory
                    </p>
                    <p dir="auto">
                     Escoja el directorio donde descompact las carpetas descargadas del
servidor.
                    </p>
                    <p dir="auto">
                     Una vez el nuevo proyecto est creado, abra un script: File &lt; New
File &lt; R Script. En este script debera copiar la serie de comandos
que encontrar a continuacin:
                    </p>
                    <ol dir="auto">
                     <li>
                      Instalando las bibliotecas necesarias:
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# DESeq2
if (! require(DESeq2)) {
  source("https://bioconductor.org/biocLite.R")
  biocLite("DESeq2")
}

# Enhanced Volcano

BiocManager::install("EnhancedVolcano")

# tidyverse
install.packages("tidyverse")

# tibble
install.packages("tibble")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> DESeq2</span>
<span class="pl-k">if</span> (<span class="pl-k">!</span> require(<span class="pl-smi">DESeq2</span>)) {
  source(<span class="pl-s"><span class="pl-pds">"</span>https://bioconductor.org/biocLite.R<span class="pl-pds">"</span></span>)
  biocLite(<span class="pl-s"><span class="pl-pds">"</span>DESeq2<span class="pl-pds">"</span></span>)
}

<span class="pl-c"><span class="pl-c">#</span> Enhanced Volcano</span>

<span class="pl-e">BiocManager</span><span class="pl-k">::</span>install(<span class="pl-s"><span class="pl-pds">"</span>EnhancedVolcano<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> tidyverse</span>
install.packages(<span class="pl-s"><span class="pl-pds">"</span>tidyverse<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> tibble</span>
install.packages(<span class="pl-s"><span class="pl-pds">"</span>tibble<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      Activando las bibliotecas
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#  Librerias
library(DESeq2)
library(EnhancedVolcano)
library(tidyverse)
library(tibble)" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>  Librerias</span>
library(<span class="pl-smi">DESeq2</span>)
library(<span class="pl-smi">EnhancedVolcano</span>)
library(<span class="pl-smi">tidyverse</span>)
library(<span class="pl-smi">tibble</span>)</pre>
                    </div>
                    <ol dir="auto" start="3">
                     <li>
                      Leyendo las tablas
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Leyendo tabla de anotaciones
anot &lt;- read.delim(&quot;10.TranscriptsQuantification/kegg_randomic_analysis.tsv&quot;,
                   sep = &quot;\t&quot;) %&gt;% 
  separate(GenName, c(&quot;GenName&quot;, &quot;Function&quot;), sep = ';') %&gt;% 
  select(target_id, kegg_id, GenName)


# Leyendo la matriz de counts
data = read.table(&quot;10.TranscriptsQuantification/kallisto.isoform.counts.matrix&quot;, 
                  header=T, row.names=1, com='')" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Leyendo tabla de anotaciones</span>
<span class="pl-smi">anot</span> <span class="pl-k">&lt;-</span> read.delim(<span class="pl-s"><span class="pl-pds">"</span>10.TranscriptsQuantification/kegg_randomic_analysis.tsv<span class="pl-pds">"</span></span>,
                   <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\t</span><span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  separate(<span class="pl-smi">GenName</span>, c(<span class="pl-s"><span class="pl-pds">"</span>GenName<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Function<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span>) %<span class="pl-k">&gt;</span>% 
  select(<span class="pl-smi">target_id</span>, <span class="pl-smi">kegg_id</span>, <span class="pl-smi">GenName</span>)


<span class="pl-c"><span class="pl-c">#</span> Leyendo la matriz de counts</span>
<span class="pl-v">data</span> <span class="pl-k">=</span> read.table(<span class="pl-s"><span class="pl-pds">"</span>10.TranscriptsQuantification/kallisto.isoform.counts.matrix<span class="pl-pds">"</span></span>, 
                  <span class="pl-v">header</span><span class="pl-k">=</span><span class="pl-c1">T</span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">1</span>, <span class="pl-v">com</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>)</pre>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      Redondeando los valores de la matriz de expresin y filtrando (al
menos dos counts por transcripto)
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# redondeando los valores
rnaseqMatrix = round(data)

# Corte a por lo menos 2 counts por transcripto
rnaseqMatrix = rnaseqMatrix[rowSums(rnaseqMatrix) &gt;= 2,]" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> redondeando los valores</span>
<span class="pl-v">rnaseqMatrix</span> <span class="pl-k">=</span> round(<span class="pl-smi">data</span>)

<span class="pl-c"><span class="pl-c">#</span> Corte a por lo menos 2 counts por transcripto</span>
<span class="pl-v">rnaseqMatrix</span> <span class="pl-k">=</span> <span class="pl-smi">rnaseqMatrix</span>[rowSums(<span class="pl-smi">rnaseqMatrix</span>) <span class="pl-k">&gt;</span><span class="pl-k">=</span> <span class="pl-c1">2</span>,]</pre>
                    </div>
                    <ol dir="auto" start="5">
                     <li>
                      Generando la tabla del diseo experimental
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='conditions = data.frame(conditions=factor(c("acetone", rep("oil_tretated_0.5", 3),
                                            rep("oil_treated_0.25",3), 
                                            rep("oil_treated_0.125",3),
                                            rep("control",2))))
rownames(conditions) = colnames(rnaseqMatrix)' dir="auto">
                     <pre><span class="pl-v">conditions</span> <span class="pl-k">=</span> <span class="pl-k">data.frame</span>(<span class="pl-v">conditions</span><span class="pl-k">=</span><span class="pl-k">factor</span>(c(<span class="pl-s"><span class="pl-pds">"</span>acetone<span class="pl-pds">"</span></span>, rep(<span class="pl-s"><span class="pl-pds">"</span>oil_tretated_0.5<span class="pl-pds">"</span></span>, <span class="pl-c1">3</span>),
                                            rep(<span class="pl-s"><span class="pl-pds">"</span>oil_treated_0.25<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>), 
                                            rep(<span class="pl-s"><span class="pl-pds">"</span>oil_treated_0.125<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>),
                                            rep(<span class="pl-s"><span class="pl-pds">"</span>control<span class="pl-pds">"</span></span>,<span class="pl-c1">2</span>))))
rownames(<span class="pl-smi">conditions</span>) <span class="pl-k">=</span> colnames(<span class="pl-smi">rnaseqMatrix</span>)</pre>
                    </div>
                    <ol dir="auto" start="6">
                     <li>
                      Generando el objeto DESeq
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="ddsFullCountTable &lt;- DESeqDataSetFromMatrix(
  countData = rnaseqMatrix,
  colData = conditions,
  design = ~ conditions)" dir="auto">
                     <pre><span class="pl-smi">ddsFullCountTable</span> <span class="pl-k">&lt;-</span> DESeqDataSetFromMatrix(
  <span class="pl-v">countData</span> <span class="pl-k">=</span> <span class="pl-smi">rnaseqMatrix</span>,
  <span class="pl-v">colData</span> <span class="pl-k">=</span> <span class="pl-smi">conditions</span>,
  <span class="pl-v">design</span> <span class="pl-k">=</span> <span class="pl-k">~</span> <span class="pl-smi">conditions</span>)</pre>
                    </div>
                    <ol dir="auto" start="7">
                     <li>
                      Clculo de la expresin diferencial
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='dds = DESeq(ddsFullCountTable, test="Wald", fitType="mean")' dir="auto">
                     <pre><span class="pl-v">dds</span> <span class="pl-k">=</span> DESeq(<span class="pl-smi">ddsFullCountTable</span>, <span class="pl-v">test</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Wald<span class="pl-pds">"</span></span>, <span class="pl-v">fitType</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>mean<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <ol dir="auto" start="8">
                     <li>
                      Comparando
                      <strong>
                       Acetona vs control
                      </strong>
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Comparando acetona vc control
acetonevscontrol &lt;- results(dds, alpha=.05, lfcThreshold=1, 
                               contrast = c(&quot;conditions&quot;,&quot;acetone&quot;,&quot;control&quot;))


acetonevscontrol.df &lt;- as.data.frame(acetonevscontrol) %&gt;% 
  rownames_to_column(var = &quot;target_id&quot;)



# Juntando tabla de expresin diferencial y anotacin funcional
final.table &lt;- full_join(anot,acetonevscontrol.df) %&gt;% 
  column_to_rownames(var='target_id')" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Comparando acetona vc control</span>
<span class="pl-smi">acetonevscontrol</span> <span class="pl-k">&lt;-</span> results(<span class="pl-smi">dds</span>, <span class="pl-v">alpha</span><span class="pl-k">=</span><span class="pl-smi">.05</span>, <span class="pl-v">lfcThreshold</span><span class="pl-k">=</span><span class="pl-c1">1</span>, 
                               <span class="pl-v">contrast</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>conditions<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>acetone<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>control<span class="pl-pds">"</span></span>))


<span class="pl-smi">acetonevscontrol.df</span> <span class="pl-k">&lt;-</span> as.data.frame(<span class="pl-smi">acetonevscontrol</span>) %<span class="pl-k">&gt;</span>% 
  rownames_to_column(<span class="pl-v">var</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>target_id<span class="pl-pds">"</span></span>)



<span class="pl-c"><span class="pl-c">#</span> Juntando tabla de expresin diferencial y anotacin funcional</span>
<span class="pl-smi">final.table</span> <span class="pl-k">&lt;-</span> full_join(<span class="pl-smi">anot</span>,<span class="pl-smi">acetonevscontrol.df</span>) %<span class="pl-k">&gt;</span>% 
  column_to_rownames(<span class="pl-v">var</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>target_id<span class="pl-pds">'</span></span>)</pre>
                    </div>
                    <ol dir="auto" start="9">
                     <li>
                      Volcano Plot
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="EnhancedVolcano(final.table,
                lab = final.table$GenName,
                x = 'log2FoldChange',
                y = 'pvalue',
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-12, 12),
                pointSize = 1.5,
                labSize = 2.5,
                colAlpha = 0.7,
                legendLabSize = 12,
                legendIconSize = 4.0,
                drawConnectors = TRUE,
                widthConnectors = 0.2,
                title = &quot;Differential Expression Analysis&quot;,
                subtitle = &quot;Control vs. Acetone&quot;,
                max.overlaps = 20,
                colConnectors = 'grey30')" dir="auto">
                     <pre>EnhancedVolcano(<span class="pl-smi">final.table</span>,
                <span class="pl-v">lab</span> <span class="pl-k">=</span> <span class="pl-smi">final.table</span><span class="pl-k">$</span><span class="pl-smi">GenName</span>,
                <span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>log2FoldChange<span class="pl-pds">'</span></span>,
                <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>pvalue<span class="pl-pds">'</span></span>,
                <span class="pl-v">pCutoff</span> <span class="pl-k">=</span> <span class="pl-c1">0.05</span>,
                <span class="pl-v">FCcutoff</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>,
                <span class="pl-v">xlim</span> <span class="pl-k">=</span> c(<span class="pl-k">-</span><span class="pl-c1">12</span>, <span class="pl-c1">12</span>),
                <span class="pl-v">pointSize</span> <span class="pl-k">=</span> <span class="pl-c1">1.5</span>,
                <span class="pl-v">labSize</span> <span class="pl-k">=</span> <span class="pl-c1">2.5</span>,
                <span class="pl-v">colAlpha</span> <span class="pl-k">=</span> <span class="pl-c1">0.7</span>,
                <span class="pl-v">legendLabSize</span> <span class="pl-k">=</span> <span class="pl-c1">12</span>,
                <span class="pl-v">legendIconSize</span> <span class="pl-k">=</span> <span class="pl-c1">4.0</span>,
                <span class="pl-v">drawConnectors</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                <span class="pl-v">widthConnectors</span> <span class="pl-k">=</span> <span class="pl-c1">0.2</span>,
                <span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Differential Expression Analysis<span class="pl-pds">"</span></span>,
                <span class="pl-v">subtitle</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Control vs. Acetone<span class="pl-pds">"</span></span>,
                <span class="pl-v">max.overlaps</span> <span class="pl-k">=</span> <span class="pl-c1">20</span>,
                <span class="pl-v">colConnectors</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>grey30<span class="pl-pds">'</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/khidalgo85/training_transcriptomics/blob/master/imgs/unnamed-chunk-22-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/khidalgo85/training_transcriptomics/raw/master/imgs/unnamed-chunk-22-1.png" style="max-width: 100%;" width="100%"/>
                     </a>
                    </p>
                    <p dir="auto">
                     A continuacin se repetir el proceso para comparar el tratamiento con
aceite a 1/2 del MIC y el control
                    </p>
                    <ol dir="auto" start="8">
                     <li>
                      Comparando
                      <strong>
                       Aceite 1/2 MIC vs control
                      </strong>
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Control vs oil_tretated_0.5
controlvsoil0.5 &lt;- results(dds, alpha=.05, lfcThreshold=1, 
                            contrast = c(&quot;conditions&quot;,&quot;oil_tretated_0.5&quot;,&quot;control&quot;))


controlvsoil0.5.df &lt;- as.data.frame(controlvsoil0.5) %&gt;% 
  rownames_to_column(var = &quot;target_id&quot;)


# Juntando tabla de expresin diferencial y anotacin funcional
final.table &lt;- full_join(anot,controlvsoil0.5.df) %&gt;% 
  column_to_rownames(var='target_id')" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Control vs oil_tretated_0.5</span>
<span class="pl-smi">controlvsoil0.5</span> <span class="pl-k">&lt;-</span> results(<span class="pl-smi">dds</span>, <span class="pl-v">alpha</span><span class="pl-k">=</span><span class="pl-smi">.05</span>, <span class="pl-v">lfcThreshold</span><span class="pl-k">=</span><span class="pl-c1">1</span>, 
                            <span class="pl-v">contrast</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>conditions<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>oil_tretated_0.5<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>control<span class="pl-pds">"</span></span>))


<span class="pl-smi">controlvsoil0.5.df</span> <span class="pl-k">&lt;-</span> as.data.frame(<span class="pl-smi">controlvsoil0.5</span>) %<span class="pl-k">&gt;</span>% 
  rownames_to_column(<span class="pl-v">var</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>target_id<span class="pl-pds">"</span></span>)


<span class="pl-c"><span class="pl-c">#</span> Juntando tabla de expresin diferencial y anotacin funcional</span>
<span class="pl-smi">final.table</span> <span class="pl-k">&lt;-</span> full_join(<span class="pl-smi">anot</span>,<span class="pl-smi">controlvsoil0.5.df</span>) %<span class="pl-k">&gt;</span>% 
  column_to_rownames(<span class="pl-v">var</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>target_id<span class="pl-pds">'</span></span>)</pre>
                    </div>
                    <ol dir="auto" start="9">
                     <li>
                      Volcano Plot
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="EnhancedVolcano(final.table,
                lab = final.table$GenName,
                x = 'log2FoldChange',
                y = 'pvalue',
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-12, 12),
                pointSize = 1.5,
                labSize = 2.5,
                colAlpha = 0.7,
                legendLabSize = 12,
                legendIconSize = 4.0,
                drawConnectors = TRUE,
                widthConnectors = 0.2,
                title = &quot;Differential Expression Analysis&quot;,
                subtitle = &quot;Control vs. Cinnamomum camphora Oil (1/2 MIC)&quot;,
                max.overlaps = 20,
                colConnectors = 'grey30')" dir="auto">
                     <pre>EnhancedVolcano(<span class="pl-smi">final.table</span>,
                <span class="pl-v">lab</span> <span class="pl-k">=</span> <span class="pl-smi">final.table</span><span class="pl-k">$</span><span class="pl-smi">GenName</span>,
                <span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>log2FoldChange<span class="pl-pds">'</span></span>,
                <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>pvalue<span class="pl-pds">'</span></span>,
                <span class="pl-v">pCutoff</span> <span class="pl-k">=</span> <span class="pl-c1">0.05</span>,
                <span class="pl-v">FCcutoff</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>,
                <span class="pl-v">xlim</span> <span class="pl-k">=</span> c(<span class="pl-k">-</span><span class="pl-c1">12</span>, <span class="pl-c1">12</span>),
                <span class="pl-v">pointSize</span> <span class="pl-k">=</span> <span class="pl-c1">1.5</span>,
                <span class="pl-v">labSize</span> <span class="pl-k">=</span> <span class="pl-c1">2.5</span>,
                <span class="pl-v">colAlpha</span> <span class="pl-k">=</span> <span class="pl-c1">0.7</span>,
                <span class="pl-v">legendLabSize</span> <span class="pl-k">=</span> <span class="pl-c1">12</span>,
                <span class="pl-v">legendIconSize</span> <span class="pl-k">=</span> <span class="pl-c1">4.0</span>,
                <span class="pl-v">drawConnectors</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                <span class="pl-v">widthConnectors</span> <span class="pl-k">=</span> <span class="pl-c1">0.2</span>,
                <span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Differential Expression Analysis<span class="pl-pds">"</span></span>,
                <span class="pl-v">subtitle</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Control vs. Cinnamomum camphora Oil (1/2 MIC)<span class="pl-pds">"</span></span>,
                <span class="pl-v">max.overlaps</span> <span class="pl-k">=</span> <span class="pl-c1">20</span>,
                <span class="pl-v">colConnectors</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>grey30<span class="pl-pds">'</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/khidalgo85/training_transcriptomics/blob/master/imgs/unnamed-chunk-24-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/khidalgo85/training_transcriptomics/raw/master/imgs/unnamed-chunk-24-1.png" style="max-width: 100%;" width="100%"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Use los comandos anteriores, modificando la condicin(es) que quiere
comparar.
                    </p>
                    <ol dir="auto" start="10">
                     <li>
                      Plot PCA
                     </li>
                    </ol>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#Criando PCA para visualizao da clusterizao das amostras
rld &lt;- rlog(dds)

plotPCA(rld, intgroup = "conditions")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>Criando PCA para visualizao da clusterizao das amostras</span>
<span class="pl-smi">rld</span> <span class="pl-k">&lt;-</span> rlog(<span class="pl-smi">dds</span>)

plotPCA(<span class="pl-smi">rld</span>, <span class="pl-v">intgroup</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>conditions<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/khidalgo85/training_transcriptomics/blob/master/imgs/unnamed-chunk-25-1.png" rel="noopener noreferrer" target="_blank">
                      <img src="/khidalgo85/training_transcriptomics/raw/master/imgs/unnamed-chunk-25-1.png" style="max-width: 100%;" width="100%"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      VI Referencias
                     </h1>
                     <a aria-label="Permalink: VI Referencias" class="anchor" href="#vi-referencias" id="user-content-vi-referencias">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       Andrews, S. (2017). FastQC: a quality control tool for high
throughput sequence data. 2010
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://academic.oup.com/bioinformatics/article-abstract/30/15/2114/2390096" rel="nofollow">
                        Trimmomatic: a flexible trimmer for Illumina sequence
data
                       </a>
                       (Trimmomatic:
a flexible trimmer for Illumina sequence data)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://www.liebertpub.com/doi/full/10.1089/cmb.2012.0021" rel="nofollow">
                        SPAdes: a new genome assembly algorithm and its applications to
single-cell
sequencing
                       </a>
                       (Bankevich,
A., Nurk, S., Antipov, D., Gurevich, A. A., Dvorkin, M., Kulikov, A.
S.,  &amp; Pevzner, P. A. (2012). SPAdes: a new genome assembly
algorithm and its applications to single-cell sequencing. Journal of
computational biology, 19(5), 455-477.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://academic.oup.com/bioinformatics/article-abstract/29/8/1072/228832" rel="nofollow">
                        QUAST: quality assessment tool for genome
assemblies
                       </a>
                       (Gurevich,
A., Saveliev, V., Vyahhi, N., &amp; Tesler, G. (2013). QUAST: quality
assessment tool for genome assemblies. Bioinformatics, 29(8),
1072-1075.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7222033.2/" rel="nofollow">
                        GFF utilities: GffRead and
GffCompare
                       </a>
                       (Pertea,
G., &amp; Pertea, M. (2020). GFF utilities: GffRead and GffCompare.
F1000Research, 9.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://academic.oup.com/nar/article-abstract/36/suppl_1/D250/2506468" rel="nofollow">
                        eggNOG: automated construction and annotation of orthologous groups
of
genes
                       </a>
                       (Jensen, L.
J., Julien, P., Kuhn, M., von Mering, C., Muller, J., Doerks, T., &amp;
Bork, P. (2007). eggNOG: automated construction and annotation of
orthologous groups of genes. Nucleic acids research, 36(suppl_1),
D250-D254.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://www.nature.com/articles/nmeth.3176" rel="nofollow">
                        Fast and sensitive protein alignment using
DIAMOND
                       </a>
                       (Buchfink, B.,
Xie, C., &amp; Huson, D. H. (2015). Fast and sensitive protein alignment
using DIAMOND. Nature methods, 12(1), 59-60.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://academic.oup.com/mbe/article-abstract/34/8/2115/3782716" rel="nofollow">
                        Fast genome-wide functional annotation through orthology assignment
by
eggNOG-mapper
                       </a>
                       (Huerta-Cepas,
J., Forslund, K., Coelho, L. P., Szklarczyk, D., Jensen, L. J., Von
Mering, C., &amp; Bork, P. (2017). Fast genome-wide functional
annotation through orthology assignment by eggNOG-mapper. Molecular
biology and evolution, 34(8), 2115-2122.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://www.nature.com/articles/s41587-019-0201-4" rel="nofollow">
                        Graph-based genome alignment and genotyping with HISAT2 and
HISAT-genotype
                       </a>
                       (Kim,
D., Paggi, J. M., Park, C., Bennett, C., &amp; Salzberg, S. L. (2019).
Graph-based genome alignment and genotyping with HISAT2 and
HISAT-genotype. Nature biotechnology, 37(8), 907-915.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://academic.oup.com/bioinformatics/article-abstract/29/1/15/272537" rel="nofollow">
                        STAR: ultrafast universal RNA-seq
aligner
                       </a>
                       (Dobin,
A., Davis, C. A., Schlesinger, F., Drenkow, J., Zaleski, C., Jha,
S.,  &amp; Gingeras, T. R. (2013). STAR: ultrafast universal RNA-seq
aligner. Bioinformatics, 29(1), 15-21.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://academic.oup.com/bioinformatics/article-abstract/25/16/2078/204688" rel="nofollow">
                        The sequence alignment/map format and
SAMtools
                       </a>
                       (Li,
H., Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., 
&amp; Durbin, R. (2009). The sequence alignment/map format and SAMtools.
Bioinformatics, 25(16), 2078-2079.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://www.nature.com/articles/nbt.3122" rel="nofollow">
                        StringTie enables improved reconstruction of a transcriptome from
RNA-seq reads
                       </a>
                       (Pertea, M.,
Pertea, G. M., Antonescu, C. M., Chang, T. C., Mendell, J. T., &amp;
Salzberg, S. L. (2015). StringTie enables improved reconstruction of
a transcriptome from RNA-seq reads. Nature biotechnology, 33(3),
290-295.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://joss.theoj.org/papers/10.21105/joss.03050.pdf" rel="nofollow">
                        kallisto: A command-line interface to simplify computational
modelling and the generation of atomic
features
                       </a>
                       (Caldeweyher, E.
(2021). kallisto: A command-line interface to simplify computational
modelling and the generation of atomic features. Journal of Open
Source Software, 6(60), 3050.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://www.nature.com/articles/nbt.4163" rel="nofollow">
                        United States Department of Energy Systems Biology
Knowledgebase
                       </a>
                       (Arkin AP,
Cottingham RW, Henry CS, Harris NL, Stevens RL, Maslov S, et
al.KBase: The United States Department of Energy Systems Biology
Knowledgebase. Nature Biotechnology. 2018;36: 566. doi:
10.1038/nbt.4163)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/pro.3711" rel="nofollow">
                        KEGG Mapper for inferring cellular functions from protein
sequences
                       </a>
                       (Kanehisa,
M., &amp; Sato, Y. (2020). KEGG Mapper for inferring cellular functions
from protein sequences. Protein Science, 29(1), 28-35.)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://www.sciencedirect.com/science/article/pii/S0968000408000236" rel="nofollow">
                        iPath: interactive exploration of biochemical pathways and
networks
                       </a>
                       (Letunic,
I., Yamada, T., Kanehisa, M., &amp; Bork, P. (2008). iPath: interactive
exploration of biochemical pathways and networks. Trends in
biochemical sciences, 33(3), 101-103.)
                      </p>
                     </li>
                    </ul>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      BONUS Plataformas online para bioinformtica
                     </h1>
                     <a aria-label="Permalink: BONUS Plataformas online para bioinformtica" class="anchor" href="#bonus-plataformas-online-para-bioinformtica" id="user-content-bonus-plataformas-online-para-bioinformtica">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Existen diversas plataformas online que permiten realizar todos los
procesos aprendidos en este tutorial, usando servidores pblicos y
interfaces grficas.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <a href="https://usegalaxy.org/" rel="nofollow">
                       Galaxy
                      </a>
                     </h3>
                     <a aria-label="Permalink: Galaxy" class="anchor" href="#galaxy" id="user-content-galaxy">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Es una plataforma con diversas herramientas para diferentes procesos. La
mayoria que fueron usadas en este tutorial se encuentran en Galaxy.
                    </p>
                    <p dir="auto">
                     <a href="/khidalgo85/training_transcriptomics/blob/master/imgs/galaxy.png" rel="noopener noreferrer" target="_blank">
                      <img align="center" src="/khidalgo85/training_transcriptomics/raw/master/imgs/galaxy.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      HELP
                     </strong>
                     Click en el smbolo del birrete, en la parte superior.
Encontrar diversos tutoriales con videos.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <a href="https://www.kbase.us/" rel="nofollow">
                       KBase
                      </a>
                     </h3>
                     <a aria-label="Permalink: KBase" class="anchor" href="#kbase" id="user-content-kbase">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Es una plataforma con diversas herramientas para diferentes procesos. La
mayoria que fueron usadas en este tutorial se encuentran en KBase.
                    </p>
                    <p dir="auto">
                     <a href="/khidalgo85/training_transcriptomics/blob/master/imgs/kbase.png" rel="noopener noreferrer" target="_blank">
                      <img align="center" src="/khidalgo85/training_transcriptomics/raw/master/imgs/kbase.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <a href="https://www.kbase.us/learn/" rel="nofollow">
                      <strong>
                       HELP
                      </strong>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <a href="https://www.genome.jp/tools/kofamkoala/" rel="nofollow">
                       KofamKOALA
                      </a>
                     </h3>
                     <a aria-label="Permalink: KofamKOALA" class="anchor" href="#kofamkoala" id="user-content-kofamkoala">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Plataforma para anotar baseado en el algortmo HMMER. Tambin existen
otras plataformas baseadas en otros algortmos, como
                     <a href="https://www.kegg.jp/blastkoala" rel="nofollow">
                      BlastKOALA
                     </a>
                     ,
                     <a href="https://www.kegg.jp/ghostkoala" rel="nofollow">
                      GhostKOALA
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     <a href="/khidalgo85/training_transcriptomics/blob/master/imgs/koala.png" rel="noopener noreferrer" target="_blank">
                      <img align="center" src="/khidalgo85/training_transcriptomics/raw/master/imgs/koala.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <a href="https://pathways.embl.de/" rel="nofollow">
                       IPATH3
                      </a>
                     </h3>
                     <a aria-label="Permalink: IPATH3" class="anchor" href="#ipath3" id="user-content-ipath3">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Herramienta online para visualizacin y anlisis de mapas metablicos.
                    </p>
                    <p dir="auto">
                     <a href="/khidalgo85/training_transcriptomics/blob/master/imgs/ipath.png" rel="noopener noreferrer" target="_blank">
                      <img align="center" src="/khidalgo85/training_transcriptomics/raw/master/imgs/ipath.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <a href="https://pathways.embl.de/video_tutorial.cgi" rel="nofollow">
                      <strong>
                       Videos de ayuda
                      </strong>
                     </a>
                     <a href="https://pathways.embl.de/help.cgi" rel="nofollow">
                      <strong>
                       Pginas de ayuda
                      </strong>
                     </a>
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Trobleshooting
                     </h2>
                     <a aria-label="Permalink: Trobleshooting" class="anchor" href="#trobleshooting" id="user-content-trobleshooting">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Dudas, ayuda, consejos
                    </p>
                    <p dir="auto">
                     Kelly Hidalgo Martinez 
                     <a href="mailto:khidalgo@javeriana.edu.co">
                      khidalgo@javeriana.edu.co
                     </a>
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="OwYeuCHmnU+vj44ZSyweB0+lXZ5B0kUUWqVMGSV4XIENIwRbc+6tjMSWdtekKOFsdMdt2CaNuueQlvfMYkRp1A==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 Slides
                </p>
                <div class="my-3 d-flex flex-items-center">
                 <svg aria-hidden="true" class="octicon octicon-link flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                  </path>
                 </svg>
                 <span class="flex-auto min-width-0 css-truncate css-truncate-target width-fit">
                  <a class="text-bold" href="https://khidalgo85.github.io/training_transcriptomics/#1" rel="noopener noreferrer nofollow" role="link" target="_blank" title="https://khidalgo85.github.io/training_transcriptomics/#1">
                   khidalgo85.github.io/training_transcriptomics/#1
                  </a>
                 </span>
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/khidalgo85/training_transcriptomics/hovercards/citation/sidebar_partial?tree_name=master">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/khidalgo85/training_transcriptomics/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/khidalgo85/training_transcriptomics/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/khidalgo85/training_transcriptomics/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/khidalgo85/training_transcriptomics/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fkhidalgo85%2Ftraining_transcriptomics&amp;report=khidalgo85+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/khidalgo85/training_transcriptomics/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/khidalgo85/packages?repo_name=training_transcriptomics">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/khidalgo85/training_transcriptomics/used_by_list">
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
        2024 GitHub,Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You cant perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
