<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-bf4f9425a973.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-340b2bef927d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_github_catalyst_lib_index_-eccae9-1932eeecf006.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-318896a86f35.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-9fad2423070b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - lakras/bio-helper-scripts: Helper scripts for processing genomic sequence data.
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="53EE:327BA0:FA45F:16E0F7:66059330" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiI1M0VFOjMyN0JBMDpGQTQ1RjoxNkUwRjc6NjYwNTkzMzAiLCJ2aXNpdG9yX2lkIjoiNjQ1MjI0MTg1NjIzODgxNzgwNyIsInJlZ2lvbl9lZGdlIjoiaWFkIiwicmVnaW9uX3JlbmRlciI6ImlhZCJ9" data-pjax-transient="true" name="visitor-payload">
           <meta content="f3d42df588c089ed9198fdd0458b717d507350a738c52245a4b19bd472c0c867" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:385311399" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="Helper scripts for processing genomic sequence data. - lakras/bio-helper-scripts" name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/lakras/bio-helper-scripts" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/3c267640f71252e50344ef96062a1b2a4e8f38e9769f6a83180d8638c0bbe650/lakras/bio-helper-scripts" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - lakras/bio-helper-scripts: Helper scripts for processing genomic sequence data." name="twitter:title">
                    <meta content="Helper scripts for processing genomic sequence data. - lakras/bio-helper-scripts" name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/3c267640f71252e50344ef96062a1b2a4e8f38e9769f6a83180d8638c0bbe650/lakras/bio-helper-scripts" property="og:image">
                      <meta content="Helper scripts for processing genomic sequence data. - lakras/bio-helper-scripts" property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - lakras/bio-helper-scripts: Helper scripts for processing genomic sequence data." property="og:title"/>
                        <meta content="https://github.com/lakras/bio-helper-scripts" property="og:url"/>
                        <meta content="Helper scripts for processing genomic sequence data. - lakras/bio-helper-scripts" property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="a81c045767350c9f2e551b272402db2e99346c1d2710552f41e25d405666b5eb" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="3037d78571ecdebb88047af8e128a53bf5dc96ac8c533579876941ff0be8ae32" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/lakras/bio-helper-scripts git https://github.com/lakras/bio-helper-scripts.git" name="go-import"/>
                        <meta content="6245320" name="octolytics-dimension-user_id">
                         <meta content="lakras" name="octolytics-dimension-user_login">
                          <meta content="385311399" name="octolytics-dimension-repository_id">
                           <meta content="lakras/bio-helper-scripts" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="385311399" name="octolytics-dimension-repository_network_root_id">
                               <meta content="lakras/bio-helper-scripts" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/lakras/bio-helper-scripts" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-99519581d0f8.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/lakras/bio-helper-scripts","user_id":null}}' data-hydro-click-hmac="767eddf5ecc16fbe86c93c7e4efdfc399a40cf2802c71e1e0af2b9ec52d75750" href="/login?return_to=https%3A%2F%2Fgithub.com%2Flakras%2Fbio-helper-scripts">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=Bfl4zeR5gOoGIq%2B42CvwOMkHhPPIpGHQAlHB5eIEIralLCNpJ3LPennCz9mKGxKl0o0XrN2QDBSo1GliM9I%2F50uEwM8qNstyirDK%2BCB3lbsrHuza%2BeoRsDp%2BYqazYWjPJZZI83bl8qB8UyBW2SucLE47JBCK0EkfsrBj1gT2xWqXcLJXTxQOTBhcUWZmrWK7r3L7sgqgWAs%2BvclYoKRVPw8DUd%2FQD07h%2FQ3A76h7ur3kEHcphJUuZL4uwGB1IJC%2FnWecoS6osQmfOek5zTtAkgcUHLkUSYM2sQWS%2Bsya%2FpD18x5YGzH%2Fix0jMoFgOvs4q38vjA6UtYUkJXFfuqrRoJ%2BRk6W3gUJHFamoEVQrHlIPRJ7ofOsKcNF9BksllaLNDkjwsYlqIvNLnotXrtht%2B0KtJg%2BVbIxEQtxxlnwz21pNx%2F1H75ckUdYB5aJMzlMjPvEVdy70dTX%2BN67lQ%2FifzR%2FSDHVfZh44LrSTYv7VKVAIY6tyvOcn5YkUkCF4UaNm5ilIoplfoKZzJeJhNUZkJI4OTruuPDCRGO4%3D--qqtvLFqSYBzfh6je--78BzI%2Bp0jr0fY7ZyRYkbAQ%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="lakras" data-current-repository="lakras/bio-helper-scripts" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="IBZuq7pMxSBlkPDTWW63EOKEimvlm1A9H3com_87mm1BrLYAnoO5IMxkNfno3q2kjQEkPtqpHWs_j69mJWyrTw" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:lakras/bio-helper-scripts">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-af49c54c-75d5-4580-9072-db079b6a81bc" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-af49c54c-75d5-4580-9072-db079b6a81bc">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="JEWMzlAw2MGP2l7IClXa+bJ3HH0OBVx7/+KWXFUsUOSagZk4Lz/+usXbP7jbsQPhA8vHR4keHKu2q3LZhlI51A=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="OdlHbWGvH5l6gvMhJSBb4+c2QpODf85mBLaEl4TwDZezXukxYYWiTjY63Moc9QU7Il6hnYHHnJojEPHM6mOmeg==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="LXuqNC8erYcdKklkq/NYpF3CgXjhSA5MviiDfD20PkFAtQJoDexe5ZfHOZ4NoY6qiZlGA2x2PMZuBOHbQPF8Ig==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="6eWe5dWl+/DJPSX2FYovp0K16HWLXDgtikfH21iyoIUf+FuDKTbLjgyQi2+nZluP9PVgXCPBt2aUG3zhk84MlQ==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/lakras/bio-helper-scripts","user_id":null}}' data-hydro-click-hmac="767eddf5ecc16fbe86c93c7e4efdfc399a40cf2802c71e1e0af2b9ec52d75750" href="/login?return_to=https%3A%2F%2Fgithub.com%2Flakras%2Fbio-helper-scripts">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/lakras/bio-helper-scripts","user_id":null}}' data-hydro-click-hmac="767eddf5ecc16fbe86c93c7e4efdfc399a40cf2802c71e1e0af2b9ec52d75750" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=lakras%2Fbio-helper-scripts">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-50f5f38a-72ac-4f6c-827b-fd0a95092aac" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-e959222b-1211-46f5-97f5-fe43d629d4ad" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-e959222b-1211-46f5-97f5-fe43d629d4ad" id="tooltip-50f5f38a-72ac-4f6c-827b-fd0a95092aac" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/lakras/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/lakras" rel="author">
            lakras
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/lakras/bio-helper-scripts">
            bio-helper-scripts
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/lakras/bio-helper-scripts","user_id":null}}' data-hydro-click-hmac="88d2aab34d979e5e705baa119493c3f9fa5cdd07c6969f8f51713161582f193b" data-view-component="true" href="/login?return_to=%2Flakras%2Fbio-helper-scripts" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":385311399,"auth_type":"LOG_IN","originating_url":"https://github.com/lakras/bio-helper-scripts","user_id":null}}' data-hydro-click-hmac="8015551a9a41201068bf65f20f744daccd2470b670a5d311e7e21812c3f36ef7" data-view-component="true" href="/login?return_to=%2Flakras%2Fbio-helper-scripts" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
             0
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":385311399,"auth_type":"LOG_IN","originating_url":"https://github.com/lakras/bio-helper-scripts","user_id":null}}' data-hydro-click-hmac="b6f2bead9d732ade87469393268107e680ef04ce6deac7a0449718da018141fc" data-view-component="true" href="/login?return_to=%2Flakras%2Fbio-helper-scripts" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="3 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="3">
              3
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <p class="f4 mb-3">
          Helper scripts for processing genomic sequence data.
         </p>
         <h3 class="sr-only">
          License
         </h3>
         <div class="mb-2">
          <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="/lakras/bio-helper-scripts/blob/main/LICENSE">
           <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
            </path>
           </svg>
           MIT license
          </a>
         </div>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/lakras/bio-helper-scripts/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            3
           </span>
           stars
          </a>
          <a class="Link--secondary no-underline mr-3" href="/lakras/bio-helper-scripts/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/lakras/bio-helper-scripts/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/lakras/bio-helper-scripts/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/lakras/bio-helper-scripts/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":385311399,"auth_type":"LOG_IN","originating_url":"https://github.com/lakras/bio-helper-scripts","user_id":null}}' data-hydro-click-hmac="b6f2bead9d732ade87469393268107e680ef04ce6deac7a0449718da018141fc" data-view-component="true" href="/login?return_to=%2Flakras%2Fbio-helper-scripts" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/lakras/bio-helper-scripts","user_id":null}}' data-hydro-click-hmac="88d2aab34d979e5e705baa119493c3f9fa5cdd07c6969f8f51713161582f193b" data-view-component="true" href="/login?return_to=%2Flakras%2Fbio-helper-scripts" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /lakras/bio-helper-scripts" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lakras/bio-helper-scripts" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /lakras/bio-helper-scripts/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lakras/bio-helper-scripts/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /lakras/bio-helper-scripts/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lakras/bio-helper-scripts/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /lakras/bio-helper-scripts/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lakras/bio-helper-scripts/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /lakras/bio-helper-scripts/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lakras/bio-helper-scripts/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /lakras/bio-helper-scripts/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lakras/bio-helper-scripts/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/lakras/bio-helper-scripts/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /lakras/bio-helper-scripts/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lakras/bio-helper-scripts/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-1588eea5-6f17-41b2-8d12-7c57f0635f6a-list" aria-haspopup="true" aria-labelledby="tooltip-6030cde6-e87f-4f86-82f4-466704fecce6" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-1588eea5-6f17-41b2-8d12-7c57f0635f6a-button" popovertarget="action-menu-1588eea5-6f17-41b2-8d12-7c57f0635f6a-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-1588eea5-6f17-41b2-8d12-7c57f0635f6a-button" id="tooltip-6030cde6-e87f-4f86-82f4-466704fecce6" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-1588eea5-6f17-41b2-8d12-7c57f0635f6a-button" anchor-offset="normal" data-view-component="true" id="action-menu-1588eea5-6f17-41b2-8d12-7c57f0635f6a-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-1588eea5-6f17-41b2-8d12-7c57f0635f6a-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-1588eea5-6f17-41b2-8d12-7c57f0635f6a-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lakras/bio-helper-scripts" id="item-4429a2c9-a736-493f-935e-c93616d97fbd" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lakras/bio-helper-scripts/issues" id="item-060feda4-86b2-4e4c-ad5f-b66354a1e0dd" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lakras/bio-helper-scripts/pulls" id="item-c9593b02-327a-4076-aa42-efb1a25ebd3f" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lakras/bio-helper-scripts/actions" id="item-7c096e5e-bcf2-4234-9a8c-c29808a2d8a4" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lakras/bio-helper-scripts/projects" id="item-d13a54b8-1e5b-43ef-9fe7-4e59f2f62527" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lakras/bio-helper-scripts/security" id="item-ee69257b-697c-46b7-84cc-574a4b83423b" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lakras/bio-helper-scripts/pulse" id="item-0a7227fa-ac7a-4830-90bf-a3bbea29eb39" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         lakras/bio-helper-scripts
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/lakras/bio-helper-scripts/spoofed_commit_check/2b022cab04f29a417936f66e4d011be145561f80">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":385311399,"defaultBranch":"main","name":"bio-helper-scripts","ownerLogin":"lakras","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-07-12T16:24:35.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/6245320?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"main","listCacheKey":"v0:1626284915.153426","canEdit":false,"refType":"branch","currentOid":"2b022cab04f29a417936f66e4d011be145561f80"},"tree":{"items":[{"name":"aligned-fasta","path":"aligned-fasta","contentType":"directory"},{"name":"bed-files","path":"bed-files","contentType":"directory"},{"name":"blast","path":"blast","contentType":"directory"},{"name":"fasta","path":"fasta","contentType":"directory"},{"name":"lineages","path":"lineages","contentType":"directory"},{"name":"misc","path":"misc","contentType":"directory"},{"name":"read-depths","path":"read-depths","contentType":"directory"},{"name":"tables","path":"tables","contentType":"directory"},{"name":"vcf-files","path":"vcf-files","contentType":"directory"},{"name":"LICENSE","path":"LICENSE","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":11,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/lakras/bio-helper-scripts/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/lakras/bio-helper-scripts.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone lakras/bio-helper-scripts","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/lakras/bio-helper-scripts/archive/refs/heads/main.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=385311399"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"555","overviewFiles":[{"displayName":"README.md","repoName":"bio-helper-scripts","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ebio-helper-scripts\u003c/h1\u003e\u003ca id=\"user-content-bio-helper-scripts\" class=\"anchor\" aria-label=\"Permalink: bio-helper-scripts\" href=\"#bio-helper-scripts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHelper scripts for processing genomic sequence data.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eContents\u003c/h2\u003e\u003ca id=\"user-content-contents\" class=\"anchor\" aria-label=\"Permalink: Contents\" href=\"#contents\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#sars-cov-2-or-other-genome-lineages-lineages\"\u003eSARS-CoV-2 or other genome lineages\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#fasta-file-processing-fasta\"\u003eFASTA file processing\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#fasta-alignment-file-processing-aligned-fasta\"\u003eFASTA alignment file processing\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#blast-blast\"\u003eBLAST\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#vcf-file-processing-vcf-files\"\u003eVCF file processing\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bed-file-processing-bed-files\"\u003eBED file processing\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#read-depth-file-processing-read-depths\"\u003eRead depth file processing\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#tables-tables\"\u003eTables\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#table-format\"\u003eTable Format\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#column-title-manipulation\"\u003eColumn title manipulation\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#column-manipulation\"\u003eColumn manipulation\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#column-manipulation-with-dates\"\u003eColumn manipulation with dates\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#row-manipulation\"\u003eRow manipulation\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#table-manipulation\"\u003eTable manipulation\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#replicates\"\u003eReplicates\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#other\"\u003eOther\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#miscellaneous-misc\"\u003eMiscellaneous\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eIncludes the following scripts\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSARS-CoV-2 or other genome lineages (\u003ca href=\"/lakras/bio-helper-scripts/blob/main/lineages\"\u003e\u003ccode\u003elineages\u003c/code\u003e\u003c/a\u003e)\u003c/h2\u003e\u003ca id=\"user-content-sars-cov-2-or-other-genome-lineages-lineages\" class=\"anchor\" aria-label=\"Permalink: SARS-CoV-2 or other genome lineages (lineages)\" href=\"#sars-cov-2-or-other-genome-lineages-lineages\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/lineages/summarize_lineage_defining_SNPs.pl\"\u003e\u003ccode\u003esummarize_lineage_defining_SNPs.pl\u003c/code\u003e\u003c/a\u003e: Prints list of lineage-defining positions and the lineage(s) consistent with each allele.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl summarize_lineage_defining_SNPs.pl [alignment fasta file path] [optional 1 to compare all positions, including with gaps or ambiguous bases] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/lineages/estimate_allele_lineages.pl\"\u003e\u003ccode\u003eestimate_allele_lineages.pl\u003c/code\u003e\u003c/a\u003e: Generates table listing lineage(s) consistent with each sample's consensus and minor alleles. See script for descriptions of input files and output table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl estimate_allele_lineages.pl [lineage genomes aligned to reference] [consensus genomes aligned to reference] [optional list of heterozygosity tables] [optional list of read depth files] [1 to print columns for each lineage, 0 to print columns for consensus and minor alleles] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/lineages/annotate_heterozygosity_tables_with_estimated_lineages.pl\"\u003e\u003ccode\u003eannotate_heterozygosity_tables_with_estimated_lineages.pl\u003c/code\u003e\u003c/a\u003e: Annotates heterozygosity tables with lineage consistent with minor and consensus-level alleles at lineage-defining positions. Output is printed to one file per heterozygosity table or as one table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl annotate_heterozygosity_tables_with_estimated_lineages.pl [lineage genomes aligned to reference] [list of heterozygosity tables] [1 to print each heterozygosity table separately, 0 to print all tables to console] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/lineages/determine_distances_to_lineages_from_alignment.pl\"\u003e\u003ccode\u003edetermine_distances_to_lineages_from_alignment.pl\u003c/code\u003e\u003c/a\u003e: Generates 2d table of distances between all sequences in alignment to lineage sequences.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl determine_distances_to_lineages_from_alignment.pl [alignment fasta file path] [1 to ignore first sequence in alignment, 0 to include it] \"[name of lineage sequence]\" \"[name of another lineage sequence]\" [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFASTA file processing (\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta\"\u003e\u003ccode\u003efasta\u003c/code\u003e\u003c/a\u003e)\u003c/h2\u003e\u003ca id=\"user-content-fasta-file-processing-fasta\" class=\"anchor\" aria-label=\"Permalink: FASTA file processing (fasta)\" href=\"#fasta-file-processing-fasta\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/summarize_fasta_sequences.pl\"\u003e\u003ccode\u003esummarize_fasta_sequences.pl\u003c/code\u003e\u003c/a\u003e: Summarizes sequences in fasta file. Produces table with, for each sequence: number bases, number unambiguous bases, A+T count, C+G count, number Ns, number gaps, number As, number Ts, number Cs, number Gs, and counts for any other \u003ca href=\"https://en.wikipedia.org/wiki/Nucleic_acid_notation\" rel=\"nofollow\"\u003ebases\u003c/a\u003e that appear.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl summarize_fasta_sequences.pl [fasta file path] [another fasta file path] [etc.] \u0026gt; [output table file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/calculate_genome_size.pl\"\u003e\u003ccode\u003ecalculate_genome_size.pl\u003c/code\u003e\u003c/a\u003e: Counts number bases (including Ns and other ambiguous non-gap letters) in fasta file(s).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl calculate_genome_size.pl [fasta file path] [another fasta file path] [etc.]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/align_to_reference.pl\"\u003e\u003ccode\u003ealign_to_reference.pl\u003c/code\u003e\u003c/a\u003e: Aligns each input sequence with reference independently, then combines all into one fasta alignment. Bases aligned to a gap in the reference are removed. Sequences in input fastas must have unique names. Paths of fasta sequences to align can be provided directly as arguments or as one file with a list of filepaths, one per line.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl align_to_reference.pl [reference sequence] [file path of MAFFT executable file (mafft.bat) or mafft command] [filepaths of at least fasta files OR path of file containing file with list of fasta file paths, one per line] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/align_each_fasta_file_to_reference_separately.pl\"\u003e\u003ccode\u003ealign_each_fasta_file_to_reference_separately.pl\u003c/code\u003e\u003c/a\u003e: Adds reference sequence to the top of each fasta file and aligns each file separately. New files are created at filepath of old file with \"_aligned_to_ref.fasta\" appended to the end. Temp files created with \"_temp.fasta\" appended to the end. Files already at those paths will be overwritten.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl align_each_fasta_file_to_reference_separately.pl [reference sequence] [file path of MAFFT executable file (mafft.bat) or mafft command] [fasta file path] [another fasta file path] [another fasta file path] [etc.]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_by_name.pl\"\u003e\u003ccode\u003eretrieve_sequences_by_name.pl\u003c/code\u003e\u003c/a\u003e: Retrieves query sequences by name from fasta file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_sequences_by_name.pl [fasta file path] \"[query sequence name 1]\" \"[query sequence name 2]\" [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_by_names_listed_in_file.pl\"\u003e\u003ccode\u003eretrieve_sequences_by_names_listed_in_file.pl\u003c/code\u003e\u003c/a\u003e: Retrieves query sequences by name from fasta file, taking query sequence names from a list in a file, one sequence name per line.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_sequences_by_names_listed_in_file.pl [fasta file path] [file with list of query sequence names] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_containing_queries.pl\"\u003e\u003ccode\u003eretrieve_sequences_containing_queries.pl\u003c/code\u003e\u003c/a\u003e: Prints all sequences from fasta file that contain at least one provided query in the sequence name. Case sensitive.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_sequences_containing_queries.pl [fasta file path] \"[query 1]\" \"[query 2]\" \"[query 3]\" [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_by_position_in_fasta_file.pl\"\u003e\u003ccode\u003eretrieve_sequences_by_position_in_fasta_file.pl\u003c/code\u003e\u003c/a\u003e: Retrieves query sequences by position from fasta file (in other words, retrieves the first fasta sequence in the file, or the second, or the third, and so on).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_sequences_by_position_in_fasta_file.pl [fasta file path] [position of sequence to retrieve (1-indexed)] [position of another sequence to retrieve] [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/remove_sequences_by_position_in_fasta_file.pl\"\u003e\u003ccode\u003eremove_sequences_by_position_in_fasta_file.pl\u003c/code\u003e\u003c/a\u003e: Removes query sequences by position from fasta file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl remove_sequences_by_position_in_fasta_file.pl [fasta file path] [position of sequence to remove (1-indexed)] [position of another sequence to remove] [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/select_random_sequences.pl\"\u003e\u003ccode\u003eselect_random_sequences.pl\u003c/code\u003e\u003c/a\u003e: Selects a certain number of sequences at random from input fasta file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl select_random_sequences.pl [fasta sequence file] [number sequences to select at random] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/filter_sequences_by_length.pl\"\u003e\u003ccode\u003efilter_sequences_by_length.pl\u003c/code\u003e\u003c/a\u003e: Filters fasta file by sequence length.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl filter_sequences_by_length.pl [fasta file path] [minimum length] [1 to filter by number of unambiguous bases, 0 to filter on number of bases (including Ns)] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/reheader_fasta_sequences_from_filename.pl\"\u003e\u003ccode\u003ereheader_fasta_sequences_from_filename.pl\u003c/code\u003e\u003c/a\u003e: Sets file name as header line in each fasta file(s). Each fasta file should have only one sequence.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl reheader_fasta_sequences_from_filename.pl [fasta file path] [another fasta file path] [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/add_prefix_to_fasta_headers.pl\"\u003e\u003ccode\u003eadd_prefix_to_fasta_headers.pl\u003c/code\u003e\u003c/a\u003e: Adds prefix to each header line in fasta file(s).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_prefix_to_fasta_headers.pl [prefix to add to fasta file headers] [fasta file path] [another fasta file path] [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/add_filename_as_prefix_to_fasta_headers.pl\"\u003e\u003ccode\u003eadd_filename_as_prefix_to_fasta_headers.pl\u003c/code\u003e\u003c/a\u003e: Adds file name as prefix to each header line in fasta file(s).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_filename_as_prefix_to_fasta_headers.pl [fasta file path] [another fasta file path] [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/replace_fasta_headers_with_filenames.pl\"\u003e\u003ccode\u003ereplace_fasta_headers_with_filenames.pl\u003c/code\u003e\u003c/a\u003e: Replaces header line in fasta file(s) with file name.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl replace_fasta_headers_with_filenames.pl [fasta file path] [another fasta file path] [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/split_fasta_into_n_files.pl\"\u003e\u003ccode\u003esplit_fasta_into_n_files.pl\u003c/code\u003e\u003c/a\u003e: Splits up fasta file into a set number of smaller files, each with about the same number of sequences.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl split_fasta_into_n_files.pl [fasta file path]  [number output files to generate]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/split_fasta_n_sequences_per_file.pl\"\u003e\u003ccode\u003esplit_fasta_n_sequences_per_file.pl\u003c/code\u003e\u003c/a\u003e: Splits up fasta file into multiple files with a set number of sequences per file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl split_fasta_n_sequences_per_file.pl [fasta file path]  [number sequences per file]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/split_fasta_one_sequence_per_file.pl\"\u003e\u003ccode\u003esplit_fasta_one_sequence_per_file.pl\u003c/code\u003e\u003c/a\u003e: Splits up fasta file into multiple files with one sequence per file. Each output file is named using its sequence name.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl split_fasta_one_sequence_per_file.pl [fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/download_fasta_sequences_from_accession_numbers.pl\"\u003e\u003ccode\u003edownload_fasta_sequences_from_accession_numbers.pl\u003c/code\u003e\u003c/a\u003e: Given a list of accession numbers, one per line, downloads and prints fasta sequences corresponding to each accession number.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl download_fasta_sequences_from_accession_numbers.pl [path of file with list of accession numbers, one per line] [database (nucleotide by default)] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/merge_fastas_and_remove_redundant_sequences.pl\"\u003e\u003ccode\u003emerge_fastas_and_remove_redundant_sequences.pl\u003c/code\u003e\u003c/a\u003e: Merges fasta files, removing redundant sequences (so there is only one of each sequence, regardless of name). Can also be used to remove redundant sequences from a single fasta file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl merge_fastas_and_remove_redundant_sequences.pl [fasta file path] [another fasta file path] [another fasta file path] [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/modify_unaligned_fasta.pl\"\u003e\u003ccode\u003emodify_unaligned_fasta.pl\u003c/code\u003e\u003c/a\u003e: Modifies unaligned fasta file according to allele changes specified in changes table. Not designed to handle gaps. See script for description of changes table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl modify_unaligned_fasta.pl [alignment fasta file path] [changes table] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequence_position_in_fasta_file.pl\"\u003e\u003ccode\u003eretrieve_sequence_position_in_fasta_file.pl\u003c/code\u003e\u003c/a\u003e: Prints position of sequence in fasta file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_sequence_position_in_fasta_file.pl [fasta sequence file] \"[sequence name]\"\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/slice_fasta_file_before_or_after_sequence_name_query.pl\"\u003e\u003ccode\u003eslice_fasta_file_before_or_after_sequence_name_query.pl\u003c/code\u003e\u003c/a\u003e: Prints either all sequences before or all sequence after sequence name appears in fasta sequence file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl slice_fasta_file_before_or_after_sequence_name_query.pl [fasta file] [sequence name to slice before or after] [1 to print all sequences BEFORE sequence name, 0 to print all sequences AFTER sequence name] [1 to print sequences including sequence name, 0 not to] \u0026gt; [output fasta file]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/remove_duplicate_name_sequences.pl\"\u003e\u003ccode\u003eremove_duplicate_name_sequences.pl\u003c/code\u003e\u003c/a\u003e: Removes any sequence whose full name has already appeared in the fasta file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl remove_duplicate_name_sequences.pl [fasta file path] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_appearing_or_not_appearing_in_table.pl\"\u003e\u003ccode\u003eretrieve_sequences_appearing_or_not_appearing_in_table.pl\u003c/code\u003e\u003c/a\u003e: Retrieves sequences whose names do or do not appear in input table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_sequences_appearing_or_not_appearing_in_table.pl [fasta file] [input table] [column number of sequence names (0-indexed)] [1 to retrieve sequences that DO appear in input table, 0 to retrieve sequences that DON'T] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/search_for_query_sequences.pl\"\u003e\u003ccode\u003esearch_for_query_sequences.pl\u003c/code\u003e\u003c/a\u003e: Searches provided sequences for provided query sequences. Outputs table with columns: name of sequence containing at least one query sequence, number of query sequences matched.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl search_for_query_sequences.pl [fasta file path containing sequences to search] [fasta file path containing queries] \u0026gt; [output table file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_matching_at_least_N_query_sequences.pl\"\u003e\u003ccode\u003eretrieve_sequences_matching_at_least_N_query_sequences.pl\u003c/code\u003e\u003c/a\u003e: Searches provided sequences for provided query sequences. Outputs sequences matching at least N query sequences.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_sequences_matching_at_least_N_query_sequences.pl [fasta file path containing sequences to search] [fasta file path containing queries] [minimum number queries to match] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/filter_by_number_unambiguous_bases_with_minimum_read_depth.pl\"\u003e\u003ccode\u003efilter_by_number_unambiguous_bases_with_minimum_read_depth.pl\u003c/code\u003e\u003c/a\u003e: Filters sequences in fasta file by number positions with an unambiguous base (A, T, C, or G) and read depth at least the minimum read depth.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl filter_by_number_unambiguous_bases_with_minimum_read_depth.pl [fasta file path] [list of read depth tables] [minimum read depth] [minimum number unabiguous bases with at least minimum read depth] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/fasta/split_fasta_into_files_by_category_map.pl\"\u003e\u003ccode\u003esplit_fasta_into_files_by_category_map.pl\u003c/code\u003e\u003c/a\u003e: Splits fasta file into multiple files using map of sequence name to category.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl split_fasta_into_files_by_category_map.pl [fasta file path] [tab-separated map of sequence names to category, one sequence name per line] [directory to print output fasta files to]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_sequences_with_no_blast_hits.pl\"\u003e\u003ccode\u003eblast/retrieve_sequences_with_no_blast_hits.pl\u003c/code\u003e\u003c/a\u003e: Retrieves sequences that do not have blast results.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_sequences_with_no_blast_hits.pl [blast output file] [fasta file that was input to blast] [minimum percent identity for a blast hit to be counted] [minimum query coverage for a blast hit to be counted] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_sequences_with_blast_hits.pl\"\u003e\u003ccode\u003eblast/retrieve_sequences_with_blast_hits.pl\u003c/code\u003e\u003c/a\u003e: Retrieves sequences that have blast results.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_sequences_with_blast_hits.pl [blast output file] [fasta file that was input to blast] [minimum percent identity for a blast hit to be counted] [minimum query coverage for a blast hit to be counted] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/extract_hits_or_sequences_with_top_hit_in_taxon.pl\"\u003e\u003ccode\u003eblast/extract_hits_or_sequences_with_top_hit_in_taxon.pl\u003c/code\u003e\u003c/a\u003e: Retrieves blast hits or fasta sequences of sequences with top hit or any hit at all in taxon of interest or its children.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl extract_hits_or_sequences_with_top_hit_in_taxon.pl [blast output table] [fasta file that was input to blast] [nodes.dmp file from NCBI] [taxon id of taxon of interest] [1 to print fasta sequences, 0 to print subset of blast output] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/bed-files/pull_out_sequences_described_in_bed_file.pl\"\u003e\u003ccode\u003ebed-files/pull_out_sequences_described_in_bed_file.pl\u003c/code\u003e\u003c/a\u003e: Pulls out subsets of sequences by name and position within that sequence from bed file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl pull_out_sequences_described_in_bed_file.pl [fasta file path] [bed file path] [0 to output fasta sequence, 1 to add sequence as new column in bed file] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFASTA alignment file processing (\u003ca href=\"/lakras/bio-helper-scripts/blob/main/aligned-fasta\"\u003e\u003ccode\u003ealigned-fasta\u003c/code\u003e\u003c/a\u003e)\u003c/h2\u003e\u003ca id=\"user-content-fasta-alignment-file-processing-aligned-fasta\" class=\"anchor\" aria-label=\"Permalink: FASTA alignment file processing (aligned-fasta)\" href=\"#fasta-alignment-file-processing-aligned-fasta\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/aligned-fasta/generate_distance_table_from_alignment.pl\"\u003e\u003ccode\u003egenerate_distance_table_from_alignment.pl\u003c/code\u003e\u003c/a\u003e: Generates 2d table of distances between all sequences in alignment.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl generate_distance_table_from_alignment.pl [alignment fasta file path] [1 to ignore first sequence in alignment, 0 to include it] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/aligned-fasta/modify_alignment_fasta.pl\"\u003e\u003ccode\u003emodify_alignment_fasta.pl\u003c/code\u003e\u003c/a\u003e: Modifies aligned fasta file according to allele changes specified in changes table. See script for description of changes table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl modify_alignment_fasta.pl [alignment fasta file path] [changes table] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/aligned-fasta/mask_positions.pl\"\u003e\u003ccode\u003emask_positions.pl\u003c/code\u003e\u003c/a\u003e: Masks (replaces with Ns) alleles at indicated positions. Positions must be relative to same reference appearing in alignment fasta file. Reference must be first sequence in alignment fasta file. Does not mask bases that align to gaps in reference.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl mask_positions.pl [alignment fasta file path] [first position in region to mask] [last position in region to mask] [first position in another region to mask] [last position in another region to mask] [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/aligned-fasta/mask_positions_in_bulk.pl\"\u003e\u003ccode\u003emask_positions_in_bulk.pl\u003c/code\u003e\u003c/a\u003e: Masks (replaces with Ns) alleles at indicated positions. Operates separately on each sequence as described in input table. Positions must be relative to same reference appearing in alignment fasta file. Reference must be first sequence in alignment fasta file. Does not mask bases that align to gaps in reference.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl mask_positions_in_bulk.pl [alignment fasta file path] [tab-separated table containing sequence name in first column, first and last positions of regions to mask in this sequence, space separated, in second column] [optional first position in additional region to mask in all sequences] [optional last position in additional region to mask in all sequences] [optional first position in another additional region to mask in all sequences] [optional last position in another additional region to mask in all sequences] [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/aligned-fasta/mask_low_read_depth_alleles.pl\"\u003e\u003ccode\u003emask_low_read_depth_alleles.pl\u003c/code\u003e\u003c/a\u003e: Masks (replaces with Ns) alleles with low read depths.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl mask_low_read_depth_alleles.pl [alignment fasta file path] [list of read depth tables] [minimum read depth] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/aligned-fasta/remove_reference_gaps_in_alignment.pl\"\u003e\u003ccode\u003eremove_reference_gaps_in_alignment.pl\u003c/code\u003e\u003c/a\u003e: Removes gaps in reference (first sequence) in alignment and bases or gaps at the corresponding positions in all other sequences in the alignment.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl remove_reference_gaps_in_alignment.pl [alignment fasta file path] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/aligned-fasta/combine_alignments_to_same_reference.pl\"\u003e\u003ccode\u003ecombine_alignments_to_same_reference.pl\u003c/code\u003e\u003c/a\u003e: Combines multiple fasta files including the same reference sequence into one fasta alignment. Reference fasta is printed first, with no gaps. All other sequences are printed aligned to the reference as in the input. Bases aligned to a gap in the reference are removed. Sequences in input files must have unique names. First sequence in each alignment fasta file must be reference.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl combine_alignments_to_same_reference.pl [alignment fasta file path] [another alignment fasta file path] [another alignment fasta file path] [etc.] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/aligned-fasta/collapse_aligned_sequences_by_name.pl\"\u003e\u003ccode\u003ecollapse_aligned_sequences_by_name.pl\u003c/code\u003e\u003c/a\u003e: Merges aligned sequences with same name up to a \": \", such as those output by LASTZ.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl collapse_aligned_sequences_by_name.pl [alignment fasta file path] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/aligned-fasta/get_average_sequence_of_aligned_sequences.pl\"\u003e\u003ccode\u003eget_average_sequence_of_aligned_sequences.pl\u003c/code\u003e\u003c/a\u003e: Generates \"average\" sequence of aligned sequence from most common base at each position. If first sequence in alignment is reference, reference is not included in calculation of average sequence.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl get_average_sequence_of_aligned_sequences.pl [alignment fasta file path] [1 if alignment includes reference as first sequence, 0 if alignment does not include a reference] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eBLAST (\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast\"\u003e\u003ccode\u003eblast\u003c/code\u003e\u003c/a\u003e)\u003c/h2\u003e\u003ca id=\"user-content-blast-blast\" class=\"anchor\" aria-label=\"Permalink: BLAST (blast)\" href=\"#blast-blast\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAll blast runs must use output format \u003ccode\u003e-outfmt \"6 qseqid sacc stitle staxids sscinames sskingdoms qlen slen length pident qcovs evalue\"\u003c/code\u003e. (Otherwise, hardcoded column numbers can be modified within each script.)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eInstructions for running blast in a Google Cloud Virtual Machine are in \u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/README.md\"\u003e\u003ccode\u003e/blast/README.md\u003c/code\u003e\u003c/a\u003e.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/count_sequences_in_blast_output.pl\"\u003e\u003ccode\u003ecount_sequences_in_blast_output.pl\u003c/code\u003e\u003c/a\u003e: Counts the number of sequences with hits in the blast output.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl count_sequences_in_blast_output.pl [blast output table] [optional minimum percent id] [optional minimum query coverage] [1 to print sequence names, 0 to print number sequences only]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_top_blast_hit_for_each_sequence.pl\"\u003e\u003ccode\u003eretrieve_top_blast_hit_for_each_sequence.pl\u003c/code\u003e\u003c/a\u003e: Retrieves top hit for each sequence (assumes they are in order in blast output).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_top_blast_hit_for_each_sequence.pl [blast output] \u0026gt; [output subset of blast output table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_top_blast_hits_for_each_sequence.pl\"\u003e\u003ccode\u003eretrieve_top_blast_hits_for_each_sequence.pl\u003c/code\u003e\u003c/a\u003e: Retrieves top hit for each sequence (assumes they are in order in blast output). Prints all top hits with same e-values.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_top_blast_hits_for_each_sequence.pl [blast output] [number by which to multiply sequence's top evalue, to use as threshold for inclusion in top hits] [1 to treat blast output as modified DIAMOND output] \u0026gt; [output subset of blast output table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/filter_blast_hits.pl\"\u003e\u003ccode\u003efilter_blast_hits.pl\u003c/code\u003e\u003c/a\u003e: Filters blast hits: prints blast hits with at least minimum percent identity and at least minimum percent query coverage. Can also further filter with optional maximum percent identity, maximum percent query coverage provided, or minimum length of matched sequence.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl filter_blast_hits.pl [blast output] [minimum percent identity] [minimum percent query coverage] [optional maximum percent identity] [optional maximum percent query coverage] [optional minimum length of matched sequence] \u0026gt; [output subset of blast output table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_blast_outputs_for_sequences_of_interest.pl\"\u003e\u003ccode\u003eretrieve_blast_outputs_for_sequences_of_interest.pl\u003c/code\u003e\u003c/a\u003e: Retrieves blast hits for sequences of interest from blast output file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_blast_outputs_for_sequences_of_interest.pl [blast output file] [sequence name to retrieve blast hits for] [another sequence name to retrieve blast hits for] [etc.] \u0026gt; [subset blast output path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_sequences_with_no_blast_hits.pl\"\u003e\u003ccode\u003eretrieve_sequences_with_no_blast_hits.pl\u003c/code\u003e\u003c/a\u003e: Retrieves sequences that do not have blast results.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_sequences_with_no_blast_hits.pl [blast output file] [fasta file that was input to blast] [minimum percent identity for a blast hit to be counted] [minimum query coverage for a blast hit to be counted] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_sequences_with_blast_hits.pl\"\u003e\u003ccode\u003eretrieve_sequences_with_blast_hits.pl\u003c/code\u003e\u003c/a\u003e: Retrieves sequences that have blast results.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_sequences_with_blast_hits.pl [blast output file] [fasta file that was input to blast] [minimum percent identity for a blast hit to be counted] [minimum query coverage for a blast hit to be counted] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/slice_blast_output_file_before_or_after_sequence_name_query.pl\"\u003e\u003ccode\u003eslice_blast_output_file_before_or_after_sequence_name_query.pl\u003c/code\u003e\u003c/a\u003e: Prints either all lines before or all lines after sequence name appears in blast output file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl slice_blast_output_file_before_or_after_sequence_name_query.pl [blast output] [sequence name to slice before or after] [1 to print all lines BEFORE sequence name, 0 to print all lines AFTER sequence name] [1 to print lines including sequence name, 0 not to] \u0026gt; [output subset of blast output table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/extract_hits_or_sequences_with_top_hit_in_taxon.pl\"\u003e\u003ccode\u003eextract_hits_or_sequences_with_top_hit_in_taxon.pl\u003c/code\u003e\u003c/a\u003e: Retrieves blast hits or fasta sequences of sequences with top hit or any hit at all in taxon of interest or its children.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl extract_hits_or_sequences_with_top_hit_in_taxon.pl [blast output table] [fasta file that was input to blast] [nodes.dmp file from NCBI] [taxon id of taxon of interest] [1 to print fasta sequences, 0 to print subset of blast output] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/extract_hits_or_sequences_with_no_or_poor_blast_hits_in_large_database_compared_to_in_taxon_specific_database.pl\"\u003e\u003ccode\u003eextract_hits_or_sequences_with_no_or_poor_blast_hits_in_large_database_compared_to_in_taxon_specific_database.pl\u003c/code\u003e\u003c/a\u003e: Retrieves blast hits or fasta sequences of sequences that produced a blast hit from the taxon of interest in a blast search within a database consisting only of sequences from the taxon of interest and produced either no hits at all in blast search within large database blast search or only hits that were comparable to or worse than hits from the blast search in the taxon-specific database.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl extract_hits_or_sequences_with_no_or_poor_blast_hits_in_large_database_compared_to_in_taxon_specific_database.pl [blast output table from taxon-specific database blast search] [blast output table from large database blast search] [fasta file that was input to blast for large database blast search] [nodes.dmp file from NCBI] [taxon id of taxon of interest] [1 to print fasta sequences, 0 to print subset of blast output] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_top_blast_hits_LCA_for_each_sequence.pl\"\u003e\u003ccode\u003eretrieve_top_blast_hits_LCA_for_each_sequence.pl\u003c/code\u003e\u003c/a\u003e: For each sequence, extracts all top hits with same e-values (assumes they are in order in blast output). Prints lowest common ancestor (LCA) of top hits for each sequence.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_top_blast_hits_LCA_for_each_sequence.pl [blast output] [nodes.dmp file from NCBI] [number by which to multiply sequence's top evalue, to use as threshold for inclusion in LCA] [1 to print all matched accession numbers in a final column] [1 to treat blast output as modified DIAMOND output] \u0026gt; [output table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/generate_LCA_table_for_sequences_with_no_matches.pl\"\u003e\u003ccode\u003egenerate_LCA_table_for_sequences_with_no_matches.pl\u003c/code\u003e\u003c/a\u003e: Generates an LCA matches table (matching output of retrieve_top_blast_hits_LCA_for_each_sequence.pl) for sequences with no blast hits.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl generate_LCA_table_for_sequences_with_no_matches.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [fasta file that was input to blast search] \u0026gt; [output table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/compare_LCA_matches_to_ground_truth_LCA_matches.pl\"\u003e\u003ccode\u003ecompare_LCA_matches_to_ground_truth_LCA_matches.pl\u003c/code\u003e\u003c/a\u003e: Compares LCA match taxon from two blast outputs for each match. Uses output of retrieve_top_blast_hits_LCA_for_each_sequence.pl as input.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl compare_LCA_matches_to_ground_truth_LCA_matches.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search, to be treated as ground truth] [output of another retrieve_top_blast_hits_LCA_for_each_sequence.pl for another blast search, to compare to ground truth] [nodes.dmp file from NCBI] \u0026gt; [output table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_taxonids_from_accession_numbers_of_blast_matches.pl\"\u003e\u003ccode\u003eretrieve_taxonids_from_accession_numbers_of_blast_matches.pl\u003c/code\u003e\u003c/a\u003e: Retrieves each match's taxon id from Entrez using match accession number column and adds taxon ids to blast or diamond output as a new column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_taxonids_from_accession_numbers_of_blast_matches.pl [blast or diamond output] [1 if blast or diamond output is from a nucleotide search; 0 if it is from a protein search] [column number of new taxon id column to add to output file (0-indexed)] [column number (0-indexed) of column containing match accession numbers (stitle)] [optional column number (0-indexed) of column containing match names (stitle)] \u0026gt; [blast or diamond output with taxon id column added]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_taxonids_from_accession_numbers_of_blast_matches_bulk.pl\"\u003e\u003ccode\u003eretrieve_taxonids_from_accession_numbers_of_blast_matches_bulk.pl\u003c/code\u003e\u003c/a\u003e: Maps accession numbers to taxon ids directly from NCBI accession number to taxon id mapping file, without using Entrez. Useful for mapping thousands or millions of accession numbers, since this method does not have rate limits. Retrieves each match's taxon id from from match accession number column and adds taxon ids to blast or diamond output as a new column. Reads in mapping tables in parallel to save time. Appropriate mapping tables must be downloaded and unzipped from ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/ (see README file).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_taxonids_from_accession_numbers_of_blast_matches_bulk.pl [blast or diamond output] [column number of new taxon id column to add to output file (0-indexed)] [column number (0-indexed) of column containing match accession numbers (stitle)] [unzipped mapping table] [another unzipped mapping table] [etc.] \u0026gt; [blast or diamond output with taxon id column added]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_taxonids_from_accession_numbers.pl\"\u003e\u003ccode\u003eretrieve_taxonids_from_accession_numbers.pl\u003c/code\u003e\u003c/a\u003e: Given a list of accession numbers, one per line, retrieves corresponding taxon ids. Outputs tab-separating map with sequence name and taxon id.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_taxonids_from_accession_numbers.pl [path of file with list of accession numbers, one per line] [database (nuccore by default)] \u0026gt; [output mapping table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_taxon_names_from_taxonids.pl\"\u003e\u003ccode\u003eretrieve_taxon_names_from_taxonids.pl\u003c/code\u003e\u003c/a\u003e: Retrieves each match's taxon name from its taxon id and adds it to blast or diamond output as a new column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_taxon_names_from_taxonids.pl [blast or diamond output] [names.dmp file from NCBI] \u0026gt; [blast or diamond output with taxon name column added]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_most_common_taxonids_in_LCA_output.pl\"\u003e\u003ccode\u003eretrieve_most_common_taxonids_in_LCA_output.pl\u003c/code\u003e\u003c/a\u003e: Retrieve NN most frequently matched species, genera, or families.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_most_common_taxonids_in_LCA_output.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [species, genus, or family] [number most frequent matched species, genera, or families to output] [minimum number reads matched by a taxon to report it] \u0026gt; [output list of taxon ids, one per line]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_all_taxonids_in_LCA_output.pl\"\u003e\u003ccode\u003eretrieve_all_taxonids_in_LCA_output.pl\u003c/code\u003e\u003c/a\u003e: Retrieve all matched species, genera, or families.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_all_taxonids_in_LCA_output.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [species, genus, or family] \u0026gt; [output list of taxon ids, one per line]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_refseq_genome_accession_numbers_from_taxon_ids.pl\"\u003e\u003ccode\u003eretrieve_refseq_genome_accession_numbers_from_taxon_ids.pl\u003c/code\u003e\u003c/a\u003e: Given a list of taxon ids, one per line, retrieves accession numbers of refseq genomes. Outputs list of refseq genome accession numbers, one per line.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_refseq_genome_accession_numbers_from_taxon_ids.pl [path of file with list of accession numbers, one per line] \u0026gt; [output list file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/retrieve_most_common_accession_numbers_from_most_common_taxonids_in_LCA_output.pl\"\u003e\u003ccode\u003eretrieve_most_common_accession_numbers_from_most_common_taxonids_in_LCA_output.pl\u003c/code\u003e\u003c/a\u003e: Retrieve most frequent NN accession numbers matched from most frequent NN matched species, genera, or families.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_most_common_accession_numbers_from_most_common_taxonids_in_LCA_output.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [species, genus, or family] [number most frequent matched species, genera, or families to examine] [number accession numbers most frequent matched by descendants of most frequent species, genera, or families] \u0026gt; [output list of accession numbers, one per line]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/filter_LCA_matches.pl\"\u003e\u003ccode\u003efilter_LCA_matches.pl\u003c/code\u003e\u003c/a\u003e: Filters LCA matches by rank (at least species, genus, or family), mean percent identity, and/or mean percent query coverage. Uses output of retrieve_top_blast_hits_LCA_for_each_sequence.pl as input.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl filter_LCA_matches.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [1 to requires output matches to be classified to at least species level] [1 to requires output matches to be classified to at least genus level] [1 to requires output matches to be classified to at least family level] [minimum mean percent identity] [maximum mean percent identity]  [minimum mean percent query coverage] [maximum mean percent query coverage]   \u0026gt; [output filtered LCA matches table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/filter_taxonids_to_descendants_of_target_taxon.pl\"\u003e\u003ccode\u003efilter_taxonids_to_descendants_of_target_taxon.pl\u003c/code\u003e\u003c/a\u003e: Given a list of txaon ids, one per line, filters down to only descendants of parameter taxon id (for example, 10239 for Viruses).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl filter_taxonids_to_descendants_of_target_taxon.pl [path of file with list of taxon ids, one per line] [nodes.dmp file from NCBI] [taxon id to filter to] \u0026gt; [output list of taxon ids]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/filter_taxonids_in_table_column_to_descendants_of_target_taxon.pl\"\u003e\u003ccode\u003efilter_taxonids_in_table_column_to_descendants_of_target_taxon.pl\u003c/code\u003e\u003c/a\u003e: Given a table with a column of taxon ids, filters down to only descendants of parameter taxon id (for example, 10239 for Viruses).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl filter_taxonids_in_table_column_to_descendants_of_target_taxon.pl [path of table with taxon id] [number (0-indexed) of column with taxon ids] [nodes.dmp file from NCBI] [taxon id to filter to] \u0026gt; [output table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/add_column_with_superkingdom_of_taxon_id.pl\"\u003e\u003ccode\u003eadd_column_with_superkingdom_of_taxon_id.pl\u003c/code\u003e\u003c/a\u003e: Reads in column containing taxon id and adds column containing the superkingdom of that taxon id.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_column_with_superkingdom_of_taxon_id.pl [table] [title of column containing taxon ids] [nodes.dmp file from NCBI] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/kraken_summary_report_to_species_table.pl\"\u003e\u003ccode\u003ekraken_summary_report_to_species_table.pl\u003c/code\u003e\u003c/a\u003e: Converts Kraken summary report to table of species found in sample.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl kraken_summary_report_to_species_table.pl [kraken summary report] \u0026gt; [output table file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/blast/LCA_table_to_kraken_output_format.pl\"\u003e\u003ccode\u003eLCA_table_to_kraken_output_format.pl\u003c/code\u003e\u003c/a\u003e: Converts LCA output table to kraken output format. Treats paired reads as separate reads.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl LCA_table_to_kraken_output_format.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [fasta file that was input to blast search (to retrieve sequence lengths and names of unclassified sequences)] \u0026gt; [output kraken format table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eVCF file processing (\u003ca href=\"/lakras/bio-helper-scripts/blob/main/vcf-files\"\u003e\u003ccode\u003evcf-files\u003c/code\u003e\u003c/a\u003e)\u003c/h2\u003e\u003ca id=\"user-content-vcf-file-processing-vcf-files\" class=\"anchor\" aria-label=\"Permalink: VCF file processing (vcf-files)\" href=\"#vcf-file-processing-vcf-files\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eInput VCF files must be in format produced by \u003ca href=\"https://csb5.github.io/lofreq/commands/#call\" rel=\"nofollow\"\u003e\u003ccode\u003eLoFreq call\u003c/code\u003e\u003c/a\u003e.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/vcf-files/vcf_file_to_heterozygosity_table.pl\"\u003e\u003ccode\u003evcf_file_to_heterozygosity_table.pl\u003c/code\u003e\u003c/a\u003e: Reads in vcf file output produced by \u003ca href=\"https://csb5.github.io/lofreq/commands/#call\" rel=\"nofollow\"\u003e\u003ccode\u003eLoFreq call\u003c/code\u003e\u003c/a\u003e and prints human-readable heterozygosity table. Optionally filters output by read depth, minor allele readcount, and minor allele frequency. See code for output table format and hardcoded filtering thresholds.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl vcf_file_to_heterozygosity_table.pl [vcf file output by LoFreq] [1 to filter output, 0 to not filter] [optional 1 to not print reference column in output, to save space] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/vcf-files/detect_potential_transmission_events_from_iSNVs.pl\"\u003e\u003ccode\u003edetect_potential_transmission_events_from_iSNVs.pl\u003c/code\u003e\u003c/a\u003e: Detects potential transmission events from iSNVs, assuming an average transmission bottleneck of one virus per transmission. If collection dates are provided, only compares sequences within hardcoded maximum collection date distance. See script for input file details and options.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl detect_potential_transmission_events_from_iSNVs.pl [consensus sequences alignment fasta file path] [list of heterozygosity tables] [optional list of read depth tables] [optional tab-separated table mapping sample names to collection dates (YYYY-MM-DD)] [0 to print one line per sample pair, 1 to print one line per iSNV] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/vcf-files/mask_positions_in_heterozygosity_table.pl\"\u003e\u003ccode\u003emask_positions_in_heterozygosity_table.pl\u003c/code\u003e\u003c/a\u003e: Masks (removes lines corresponding to) alleles at indicated positions. See script for input file format. Positions must be relative to reference used in heterozygosity table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl mask_positions_in_heterozygosity_table.pl [heterozygosity table] [first position in region to mask] [last position in region to mask] [first position in another region to mask] [last position in another region to mask] [etc.] \u0026gt; [output heterozygosity table file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/vcf-files/mask_positions_in_heterozygosity_table_in_bulk.pl\"\u003e\u003ccode\u003emask_positions_in_heterozygosity_table_in_bulk.pl\u003c/code\u003e\u003c/a\u003e: Masks (removes lines corresponding to) alleles at indicated positions. Operates separately on multiple files as described in input table. See script for input file format. Positions must be relative to reference used in heterozygosity table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl mask_positions_in_heterozygosity_table_in_bulk.pl [tab-separated table containing filepath of heterozygosity table in first column, first and last positions of regions to mask in this heterozygosity table, space separated, in second column] [optional first position in additional region to mask in all heterozygosity tables] [optional last position in additional region to mask in all heterozygosity tables] [optional first position in another additional region to mask in all heterozygosity tables] [optional last position in another additional region to mask in all heterozygosity tables] [etc.]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/vcf-files/filter_heterozygosity_table.pl\"\u003e\u003ccode\u003efilter_heterozygosity_table.pl\u003c/code\u003e\u003c/a\u003e: Filters input heterozygosity table. See script for input file format.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl filter_heterozygosity_table.pl [heterozygosity table] [minimum minor allele readcount] [minimum minor allele frequency] [minimum read depth] \u0026gt; [output filtered heterozygosity table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/vcf-files/rereference_positions_in_heterozygosity_table.pl\"\u003e\u003ccode\u003erereference_positions_in_heterozygosity_table.pl\u003c/code\u003e\u003c/a\u003e: Uses alignment of consensus genome to reference to update heterozygosity table positions with respect to self to positions with respect to the reference genome in the alignment.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl convert_positions_in_heterozygosity_table_from_relative_to_self_to_relative_to_reference_in_alignment.pl [consensus sequence aligned to reference fasta file] [list of heterozygosity tables, one per line]# \u0026gt; [output heterozygosity table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/vcf-files/rereference_positions_in_read_depths_table.pl\"\u003e\u003ccode\u003erereference_positions_in_read_depths_table.pl\u003c/code\u003e\u003c/a\u003e: Uses alignment of consensus genome to reference to update read depths table positions with respect to self to positions with respect to the reference genome in the alignment.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl rereference_positions_in_read_depths_table.pl [consensus sequence aligned to reference fasta file] [list of read_depths tables, one per line]# \u0026gt; [output read_depths table]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/lineages/annotate_heterozygosity_tables_with_estimated_lineages.pl\"\u003e\u003ccode\u003elineages/annotate_heterozygosity_tables_with_estimated_lineages.pl\u003c/code\u003e\u003c/a\u003e: Annotates heterozygosity tables with lineage consistent with minor and consensus-level alleles at lineage-defining positions. Output is printed to one file per heterozygosity table or as one table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl annotate_heterozygosity_tables_with_estimated_lineages.pl [lineage genomes aligned to reference] [list of heterozygosity tables] [1 to print each heterozygosity table separately, 0 to print all tables to console] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eBED file processing (\u003ca href=\"/lakras/bio-helper-scripts/blob/main/bed-files\"\u003e\u003ccode\u003ebed-files\u003c/code\u003e\u003c/a\u003e)\u003c/h2\u003e\u003ca id=\"user-content-bed-file-processing-bed-files\" class=\"anchor\" aria-label=\"Permalink: BED file processing (bed-files)\" href=\"#bed-file-processing-bed-files\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRead a description of the BED file format \u003ca href=\"https://en.wikipedia.org/wiki/BED_(file_format)\" rel=\"nofollow\"\u003eon Wikipedia\u003c/a\u003e.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/bed-files/pull_out_sequences_described_in_bed_file.pl\"\u003e\u003ccode\u003epull_out_sequences_described_in_bed_file.pl\u003c/code\u003e\u003c/a\u003e: Pulls out subsets of sequences by name and position within that sequence from bed file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl pull_out_sequences_described_in_bed_file.pl [fasta file path] [bed file path] [0 to output fasta sequence, 1 to add sequence as new column in bed file] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/bed-files/retrieve_amplicon_regions_from_primer_regions.pl\"\u003e\u003ccode\u003eretrieve_amplicon_regions_from_primer_regions.pl\u003c/code\u003e\u003c/a\u003e: Reads in positions of primers and outputs positions of amplicons between those primers. Uses primer names to match left and right primers. Primer names must include the primer number and _LEFT or _RIGHT, for example nCoV-2019_1_LEFT or nCoV-2019_26_RIGHT. If more multiple start primers or multiple end primers are provided for an amplicon, the amplicon is set to be the largest possible with all its provided primers.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_amplicon_regions_from_primer_regions.pl [primers bed file path] \u0026gt; [amplicons bed file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/bed-files/modify_start_end_position_indexing_and_inclusivity.pl\"\u003e\u003ccode\u003emodify_start_end_position_indexing_and_inclusivity.pl\u003c/code\u003e\u003c/a\u003e: Modifies start and end positions in bed file from 0-indexed or 1-indexed to 0-indexed or 1-indexed and from inclusive end or non-iclusive end to inclusive end or non-iclusive end.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl modify_start_end_position_indexing_and_inclusivity.pl [bed file path] [1 if input is 1-indexed, 0 if input is 0-indexed] [1 if input has inclusive end, 0 if input has non-inclusive end] [1 if output should be 1-indexed, 0 if output should be 0-indexed] [1 if output should have inclusive end, 0 if output should have non-inclusive end] \u0026gt; [output bed file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/bed-files/remove_bed_file_rows_overlapping_positions.pl\"\u003e\u003ccode\u003eremove_bed_file_rows_overlapping_positions.pl\u003c/code\u003e\u003c/a\u003e: Removes rows in bed file that overlap with parameter positions.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl remove_bed_file_rows_overlapping_positions.pl [bed file path] [position (0-indexed); any rows overlapping this position will be removed] [another position] [another position] [etc.] \u0026gt; [output bed file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/bed-files/determine_regions_not_covered_by_amplicons.pl\"\u003e\u003ccode\u003edetermine_regions_not_covered_by_amplicons.pl\u003c/code\u003e\u003c/a\u003e: Takes in a bed file of amplicons and outputs a bed file of positions not covered by those amplicons. Assumes all amplicons come from only one reference sequence.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl determine_regions_not_covered_by_amplicons.pl [amplicons bed file path] \u0026gt; [output bed file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/bed-files/calculate_proportions_bed_file_covered_by_other_bed_files.pl\"\u003e\u003ccode\u003ecalculate_proportions_bed_file_covered_by_other_bed_files.pl\u003c/code\u003e\u003c/a\u003e: Takes in one bed file and at least one other bed file to compare it to. Outputs proportion of the sequence in the regions described in the first bed file that is captured in the regions described in each additional bed file. Outputs each bed file path and the proportion it covers of the first bed file. Assumes all rows come from only one reference sequence.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl calculate_proportions_bed_file_covered_by_other_bed_files.pl [bed file path] [other bed file path] [another bed file path] [and another] [etc.] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/bed-files/annotate_positions_with_regions_they_overlap.pl\"\u003e\u003ccode\u003eannotate_positions_with_regions_they_overlap.pl\u003c/code\u003e\u003c/a\u003e: Annotates positions in input table with the names of the region(s) they are in. Adds column containing names of regions the position overlaps. If multiple regions overlap a position they are comma-separated in the new column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl annotate_positions_with_regions_they_overlap.pl [table containing positions to annotate] \"[title of column containing positions to annotate]\" [table containing start and end positions and names of regions] [optional output column title (overlapping_regions by default)] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/bed-files/retrieve_amplicon_regions_from_primer_regions_and_count_number_amplicons_overlapping_each_position.pl\"\u003e\u003ccode\u003eretrieve_amplicon_regions_from_primer_regions_and_count_number_amplicons_overlapping_each_position.pl\u003c/code\u003e\u003c/a\u003e: Reads in positions of primers and determines positions of amplicons between those primers. Outputs total number amplicons overlapping each position.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_amplicon_regions_from_primer_regions_and_count_number_amplicons_overlapping_each_position.pl [primers bed file path] \u0026gt; [amplicons bed file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/bed-files/extend_primers_to_include_amplicon_overlaps.pl\"\u003e\u003ccode\u003eextend_primers_to_include_amplicon_overlaps.pl\u003c/code\u003e\u003c/a\u003e: Extends primers to include positions that are in more than one amplicon. Amplicons in the resulting bed file do not overlap. Outputs new primer bed file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl extend_primers_to_include_amplicon_overlaps.pl [primers bed file path] \u0026gt; [new primers bed file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRead depth file processing (\u003ca href=\"/lakras/bio-helper-scripts/blob/main/read-depths\"\u003e\u003ccode\u003eread-depths\u003c/code\u003e\u003c/a\u003e)\u003c/h2\u003e\u003ca id=\"user-content-read-depth-file-processing-read-depths\" class=\"anchor\" aria-label=\"Permalink: Read depth file processing (read-depths)\" href=\"#read-depth-file-processing-read-depths\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eInput read depth tables must be in format produced by \u003ca href=\"http://www.htslib.org/doc/samtools-depth.html\" rel=\"nofollow\"\u003e\u003ccode\u003esamtools depth\u003c/code\u003e\u003c/a\u003e: tab-separated reference name, position, read depth; no header line.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/read-depths/summarize_read_depths_across_files.pl\"\u003e\u003ccode\u003esummarize_read_depths_across_files.pl\u003c/code\u003e\u003c/a\u003e: Summarizes read depths across read depth tables. Outputs table with reference name; position; and mean, standard deviation, median, min, max, and range of read depths at each position, and number of read depth values at each position that are 0.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl summarize_read_depths_across_files.pl [read depth table] [another read depth table] [another read depth table] [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/read-depths/retrieve_position_ranges_with_threshold_read_depth.pl\"\u003e\u003ccode\u003eretrieve_position_ranges_with_threshold_read_depth.pl\u003c/code\u003e\u003c/a\u003e: Prints ranges of positions with read depths at or above minimum read depth.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_position_ranges_with_threshold_read_depth.pl [minimum read depth] [read depth table] [optional 1 to print output as one line] [optional 1 to print position ranges NOT passing read depth threshold, rather than positions that do] \u0026gt; [output path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/aligned-fasta/mask_low_read_depth_alleles.pl\"\u003e\u003ccode\u003ealigned-fasta/mask_low_read_depth_alleles.pl\u003c/code\u003e\u003c/a\u003e: Masks (replaces with Ns) alleles with low read depths.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl mask_low_read_depth_alleles.pl [alignment fasta file path] [list of read depth tables] [minimum read depth] \u0026gt; [output fasta file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTables (\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables\"\u003e\u003ccode\u003etables\u003c/code\u003e\u003c/a\u003e)\u003c/h2\u003e\u003ca id=\"user-content-tables-tables\" class=\"anchor\" aria-label=\"Permalink: Tables (tables)\" href=\"#tables-tables\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTables are assumed to be tab-separated (usually denoted \u003ccode\u003e.tsv\u003c/code\u003e or \u003ccode\u003e.txt\u003c/code\u003e. To use with comma-separated \u003ccode\u003e.csv\u003c/code\u003e files, change the value of the \u003ccode\u003e$DELIMITER\u003c/code\u003e variable from \u003ccode\u003e\"\\t\"\u003c/code\u003e to \u003ccode\u003e\",\"\u003c/code\u003e or convert from comma-separated csv tables to tab-separated tsv tables using \u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/csv_to_tsv.pl\"\u003e\u003ccode\u003ecsv_to_tsv.pl\u003c/code\u003e\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTable Format\u003c/h3\u003e\u003ca id=\"user-content-table-format\" class=\"anchor\" aria-label=\"Permalink: Table Format\" href=\"#table-format\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/csv_to_tsv.pl\"\u003e\u003ccode\u003ecsv_to_tsv.pl\u003c/code\u003e\u003c/a\u003e: Converts comma-separated csv table to tab-separated tsv table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl csv_to_tsv.pl [table] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/tsv_to_csv.pl\"\u003e\u003ccode\u003etsv_to_csv.pl\u003c/code\u003e\u003c/a\u003e: Converts tab-separated tsv table to comma-separated csv table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl tsv_to_csv.pl [table] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/make_r_friendly_table.pl\"\u003e\u003ccode\u003emake_r_friendly_table.pl\u003c/code\u003e\u003c/a\u003e: Converts table to R-friendly format. See script for example inputs and outputs.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl make_r_friendly_table.pl [table file path] [first data column index] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/convert_variable_value_lists_to_table.pl\"\u003e\u003ccode\u003econvert_variable_value_lists_to_table.pl\u003c/code\u003e\u003c/a\u003e: Converts lists of variable values, such as those output by Terra, to a table. Lists of values must appear with variable on one line followed by values in the following line, comma-separated and surrounded by brackets. Lines not matching this format are ignored. Variable names must be unique (cannot repeat). List items must be in order for all variables. Expects all variables to have identical numbers of values: ignores variables with less than the largest number of values. See code for example of input and output formats.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl convert_variable_value_lists_to_table.pl [file containing input lists] [optional list item separator, \", \" by default] [optional list start, \"[\" by default] [optional list end, \"]\" by default] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eColumn title manipulation\u003c/h3\u003e\u003ca id=\"user-content-column-title-manipulation\" class=\"anchor\" aria-label=\"Permalink: Column title manipulation\" href=\"#column-title-manipulation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/replace_all_spaces_parens_in_column_titles.pl\"\u003e\u003ccode\u003ereplace_all_spaces_parens_in_column_titles.pl\u003c/code\u003e\u003c/a\u003e: Replaces all spaces and parentheses in header line with provided replacement value, or underscore by default.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl replace_all_spaces_parens_in_column_titles.pl [table] [optional value to replace spaces with in header line] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/replace_column_title.pl\"\u003e\u003ccode\u003ereplace_column_title.pl\u003c/code\u003e\u003c/a\u003e: Replaces column title with new column title.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl replace_column_title.pl [table] \"[current title of column to replace]\" \"[replacement column title]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eColumn manipulation\u003c/h3\u003e\u003ca id=\"user-content-column-manipulation\" class=\"anchor\" aria-label=\"Permalink: Column manipulation\" href=\"#column-manipulation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/delete_columns.pl\"\u003e\u003ccode\u003edelete_columns.pl\u003c/code\u003e\u003c/a\u003e: Deletes columns with specified column titles.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl delete_columns.pl [table] \"[title of column to delete]\" \"[title of another column to delete]\" \"[title of another column to delete]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/replace_whole_column_with_empty_values.pl\"\u003e\u003ccode\u003ereplace_whole_column_with_empty_values.pl\u003c/code\u003e\u003c/a\u003e: Replaces whole column (including title unless hardcoded option selected) with empty values.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl replace_whole_column_with_empty_values.pl [table] \"[column title]\" \"[another column title]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/duplicate_columns.pl\"\u003e\u003ccode\u003eduplicate_columns.pl\u003c/code\u003e\u003c/a\u003e: Duplicates selected columns.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl duplicate_columns.pl [tab-separated table] \"[column title]\" \"[another column title]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_one_value_column.pl\"\u003e\u003ccode\u003eadd_one_value_column.pl\u003c/code\u003e\u003c/a\u003e: Adds column with specified title and specified value for all values.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_one_value_column.pl [table to add column to] \"[title of column to add]\" \"[value of column to add]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_column_for_selected_rows.pl\"\u003e\u003ccode\u003eadd_column_for_selected_rows.pl\u003c/code\u003e\u003c/a\u003e: Creates new column with values in selected rows.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_column_for_selected_rows.pl [table to add new column to] \"[title of column in table to identify rows by]\" [file with list of rows to select, one per line] \"[optional title of new column to add]\" \"[optional value to add to selected rows in new column]\" \"[optional value to add to all other rows in new column]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/concatenate_columns.pl\"\u003e\u003ccode\u003econcatenate_columns.pl\u003c/code\u003e\u003c/a\u003e: Concatenates values in selected columns. Adds concatenated values in new column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl concatenate_columns.pl [tab-separated table] \"[column title]\" \"[another column title]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/summarize_numerical_columns.pl\"\u003e\u003ccode\u003esummarize_numerical_columns.pl\u003c/code\u003e\u003c/a\u003e: Summarizes selected numerical columns. Adds new columns with: mean, standard deviation, median, min, max, range, and all values sorted in a comma-separated list. If a column value contains a comma-separated list, includes all values from list.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl summarize_numerical_columns.pl [tab-separated table] \"[column title]\" \"[another column title]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/summarize_numerical_column_holding_other_columns_constant.pl\"\u003e\u003ccode\u003esummarize_numerical_column_holding_other_columns_constant.pl\u003c/code\u003e\u003c/a\u003e: Summarizes values in selected numerical column while holding other columns constant. Outputs table with constant columns and new columns with statistics summarizing selected numerical column: mean, standard deviation, median, number values, min, max, range, and all values sorted in a comma-separated list.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl summarize_numerical_column_holding_other_columns_constant.pl [tab-separated table] \"[title of numerical column to summarize]\" \"[title of column to hold constant]\" \"[title of another column to hold constant]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/combine_columns.pl\"\u003e\u003ccode\u003ecombine_columns.pl\u003c/code\u003e\u003c/a\u003e: Combines each selected pair of columns into one column with all values. Does not merge values. See script for example inputs and outputs.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl combine_columns.pl [input table] [list of tab-separated groups of titles of columns to combine, one group per line] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/merge_columns.pl\"\u003e\u003ccode\u003emerge_columns.pl\u003c/code\u003e\u003c/a\u003e: Merges selected columns. Reports any conflicts.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl merge_columns.pl [table to merge] \"[title of column to merge]\" \"[title of another column to merge]\" \"[title of another column to merge]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/merge_columns_with_conflict_detail_column.pl\"\u003e\u003ccode\u003emerge_columns_with_conflict_detail_column.pl\u003c/code\u003e\u003c/a\u003e: Merges selected columns. Reports any conflicts.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl merge_columns_with_conflict_detail_column.pl [table to merge] \"[title of column to print in merge conflict message]\" \"[title of column to merge]\" \"[title of another column to merge]\" \"[title of another column to merge]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_column_with_number_occurrences_of_column_values.pl\"\u003e\u003ccode\u003eadd_column_with_number_occurrences_of_column_values.pl\u003c/code\u003e\u003c/a\u003e: Adds column indicating number occurrences of that row's value(s) in the entirety of the parameter column(s).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_column_with_number_occurrences_of_column_values.pl [tab-separated table] \"[column title]\" \"[optional additional column title]\" \"[optional additional column title]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/replace_values_in_columns.pl\"\u003e\u003ccode\u003ereplace_values_in_columns.pl\u003c/code\u003e\u003c/a\u003e: Replaces query with replacement text in specified columns.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl replace_values_in_columns.pl [table] \"[query text to replace]\" \"[replacement text]\" [1 to exactly match full column value only, 0 to allow search and replace within text of column value] \"[title of column to search]\" \"[title of another column to search]\" \"[title of another column to search]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/delete_values_in_columns.pl\"\u003e\u003ccode\u003edelete_values_in_columns.pl\u003c/code\u003e\u003c/a\u003e: Deletes query in specified columns.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl delete_values_in_columns.pl [table] \"[query text to delete]\" # [1 to exactly match full column value only, 0 to allow search within text of column value] \"[title of column to search]\" \"[title of another column to search]\" \"[title of another column to search]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/replace_nonempty_values_in_columns.pl\"\u003e\u003ccode\u003ereplace_nonempty_values_in_columns.pl\u003c/code\u003e\u003c/a\u003e: Replaces non-empty values with replacement text in specified columns.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl replace_nonempty_values_in_columns.pl [table] \"[replacement text]\" \"[title of column to search]\" \"[title of another column to search]\" \"[title of another column to search]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/replace_values_with_coded_values.pl\"\u003e\u003ccode\u003ereplace_values_with_coded_values.pl\u003c/code\u003e\u003c/a\u003e: Replaces non-empty values with coded values, e.g., Value 1 (for the most common value), Value 2 (for the second-most common value), Value 3, etc. Ties are broken alphabetically.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl replace_values_with_coded_values.pl [table] \"[title of column to search]\" \"[optional code prefix]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/bulk_replace_in_one_column.pl\"\u003e\u003ccode\u003ebulk_replace_in_one_column.pl\u003c/code\u003e\u003c/a\u003e: Replaces all occurrences of values in selected column to mapped replacement values.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl bulk_replace_in_one_column.pl [tab-separated file mapping current values (first column) to new values (second column)] [path of table in which to replace values] [title of column to replace values in] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_to_start_and_end_of_values_in_columns.pl\"\u003e\u003ccode\u003eadd_to_start_and_end_of_values_in_columns.pl\u003c/code\u003e\u003c/a\u003e: Pads non-empty values in specified columns with parameter start and end text.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_to_start_and_end_of_values_in_columns.pl [table] \"[text to add to start of each column value]\" \"[text to add to end of each column value]\" \"[title of column to search]\" \"[title of another column to search]\" \"[title of another column to search]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_to_start_and_end_of_values_containing_query_in_columns.pl\"\u003e\u003ccode\u003eadd_to_start_and_end_of_values_containing_query_in_columns.pl\u003c/code\u003e\u003c/a\u003e: Pads values containing query in specified columns with parameter start and end text.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_to_start_and_end_of_values_containing_query_in_columns.pl [table] \"[query text]\" \"[text to add to start of each column value]\" \"[text to add to end of each column value]\" \"[title of column to search]\" \"[title of another column to search]\" \"[title of another column to search]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/change_capitalization_in_columns.pl\"\u003e\u003ccode\u003echange_capitalization_in_columns.pl\u003c/code\u003e\u003c/a\u003e: Changes capitalization of values in specified columns: all capitalized, all lowercase, or first letter capitalized.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl change_capitalization_in_columns.pl [table] [uc to make all values uppercase, lc to make all values lowercase, first to capitalize first letter] \"[title of column to capitalize]\" \"[title of another column to capitalize]\" \"[title of another column to capitalize]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/fill_in_empty_column_values.pl\"\u003e\u003ccode\u003efill_in_empty_column_values.pl\u003c/code\u003e\u003c/a\u003e: Fills in empty values in column of interest with specified value.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl fill_in_empty_column_values.pl [table] \"[title of column to fill in]\" \"[value to replace empty values with]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/fill_in_empty_column_values_with_increasing_numerical_index.pl\"\u003e\u003ccode\u003efill_in_empty_column_values_with_increasing_numerical_index.pl\u003c/code\u003e\u003c/a\u003e: Fills in empty values in column of interest with an increasing numerical index value, so that every empty cell contains a unique numerical value.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl fill_in_empty_column_values_with_increasing_numerical_index.pl [table] \"[title of column to fill in]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/fill_in_empty_column_values_from_other_column.pl\"\u003e\u003ccode\u003efill_in_empty_column_values_from_other_column.pl\u003c/code\u003e\u003c/a\u003e: Fills in empty values in column of interest with values from other column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl fill_in_empty_column_values_from_other_column.pl [table] \"[title of column to fill in]\" \"[title of column with potential replacement values]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/replace_column_values_with_other_column_where_present.pl\"\u003e\u003ccode\u003ereplace_column_values_with_other_column_where_present.pl\u003c/code\u003e\u003c/a\u003e: Fills in values in column of interest with values from other column when they are present.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl replace_column_values_with_other_column_where_present.pl [table] \"[title of column to fill in]\" \"[title of column with potential replacement values]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/replace_column_values_where_other_column_present_and_nonzero.pl\"\u003e\u003ccode\u003ereplace_column_values_where_other_column_present_and_nonzero.pl\u003c/code\u003e\u003c/a\u003e: In rows where a column has a present, non-zero value, replaces value in another column with parameter replacement value.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl replace_column_values_where_other_column_present_and_nonzero.pl [table] \"[title of column to check]\" \"[title of column to fill in]\" \"[replacement value]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/remove_nonunique_values_in_list_in_column.pl\"\u003e\u003ccode\u003eremove_nonunique_values_in_list_in_column.pl\u003c/code\u003e\u003c/a\u003e: Removes non-unique values in comma-separated lists in specified columns.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl remove_nonunique_values_in_list_in_column.pl [table] \"[column title]\" \"[another column title]\" \"[another column title]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/retrieve_subset_of_columns.pl\"\u003e\u003ccode\u003eretrieve_subset_of_columns.pl\u003c/code\u003e\u003c/a\u003e: Subsets table to only columns of interest.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_subset_of_columns.pl [table] \"[title of first column to include in output]\" \"[title of second column to include]\" \"[title of third column to include]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/retrieve_subset_of_columns_by_column_number.pl\"\u003e\u003ccode\u003eretrieve_subset_of_columns_by_column_number.pl\u003c/code\u003e\u003c/a\u003e: Subsets table to only columns of interest.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_subset_of_columns_by_column_number.pl [table] [column number of first column to include in output] [column number of second column to include] [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/compile_values_and_titles_in_selected_columns.pl\"\u003e\u003ccode\u003ecompile_values_and_titles_in_selected_columns.pl\u003c/code\u003e\u003c/a\u003e: Generates a new column with the values in selected columns and their column titles, where values are present.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl compile_values_and_titles_in_selected_columns.pl [tab-separated table] \"[column title]\" \"[another column title]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/condense_list_values_into_one_item.pl\"\u003e\u003ccode\u003econdense_list_values_into_one_item.pl\u003c/code\u003e\u003c/a\u003e: For any values in specified columns that are comma-separated lists, replaces comma-separated list with the first, smallest, or largest value in the list.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl condense_list_values_into_one_item.pl [table] [0 to use the first value, 1 to use the smallest value, 2 to use the greatest value] \"[title of column to replace lists in]\" \"[title of another column to replace lists in]\" \"[title of another column to replace lists in]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/replace_values_with_column_proportions.pl\"\u003e\u003ccode\u003ereplace_values_with_column_proportions.pl\u003c/code\u003e\u003c/a\u003e: Calculates sum of each column containing numerical values. Replaces values in column with the proportion of its sum.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl replace_values_with_column_proportions.pl [tab-separated table] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_filename_column_from_filepath_column.pl\"\u003e\u003ccode\u003eadd_filename_column_from_filepath_column.pl\u003c/code\u003e\u003c/a\u003e: Retrieves filepaths from specified column (or first column by default). Adds new column with filenames retrieved from these filepaths.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_filename_column_from_filepath_column.pl [table to add new column to] [optional title of column containing filepaths (if not provided, uses first column)] [optional 1 to remove all file extensions in output filenames (such that filename contains no .s)] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_column_comparing_two_columns.pl\"\u003e\u003ccode\u003eadd_column_comparing_two_columns.pl\u003c/code\u003e\u003c/a\u003e: Adds a column indicating whether or not there is a difference between two columns.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_column_comparing_two_columns.pl [tab-separated table] \"[title of first column to compare]\" \"[title of second column to compare]\" [1 to print the actual values when different] \"[optional new column title]\" \"[optional new column value if values are identical]\" \"[optional new column value if values are different]\" \"[optional new column value if one value missing]\" \"[optional new column value if both values missing]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/replace_column_values_with_other_column_where_yet_another_column_has_query_value.pl\"\u003e\u003ccode\u003ereplace_column_values_with_other_column_where_yet_another_column_has_query_value.pl\u003c/code\u003e\u003c/a\u003e: Fills in values in column of interest with values from other column when they are present.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl replace_column_values_with_other_column_where_yet_another_column_has_query_value.pl [table] \"[title of column to replace values in]\" \"[title of column with potential replacement values]\" \"[title of column to search]\" \"[query value]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/split_column_after_query.pl\"\u003e\u003ccode\u003esplit_column_after_query.pl\u003c/code\u003e\u003c/a\u003e: Splits column into two columns following appearance of query in each column value. If a cell does not contain the query, duplicates the column value.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl split_column_after_query.pl [table] \"[column name]\" \"[query]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_column_with_filename.pl\"\u003e\u003ccode\u003eadd_column_with_filename.pl\u003c/code\u003e\u003c/a\u003e: Adds column with table filename in all values.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_column_with_filename.pl [table to add column to] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eColumn manipulation with dates\u003c/h3\u003e\u003ca id=\"user-content-column-manipulation-with-dates\" class=\"anchor\" aria-label=\"Permalink: Column manipulation with dates\" href=\"#column-manipulation-with-dates\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/dates_in_columns_to_YYYY_MM_DD.pl\"\u003e\u003ccode\u003edates_in_columns_to_YYYY_MM_DD.pl\u003c/code\u003e\u003c/a\u003e: Converts dates in specified columns to YYYY-MM-DD format. Multiple dates may be separated by a \", \". Column titles must not have spaces.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl dates_in_columns_to_YYYY_MM_DD.pl [table] \"[title of column with dates]\" \"[title of another column with dates]\" \"[title of another column with dates]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_difference_in_dates_column.pl\"\u003e\u003ccode\u003eadd_difference_in_dates_column.pl\u003c/code\u003e\u003c/a\u003e: Adds column listing difference in dates (in days) between columns specified in parameter column titles. Dates must be in YYYY-MM-DD format.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_difference_in_dates_column.pl [table] \"[title of column with dates]\" \"[title of another column with dates]\" \"[optional title of new column]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_earliest_or_latest_date_column.pl\"\u003e\u003ccode\u003eadd_earliest_or_latest_date_column.pl\u003c/code\u003e\u003c/a\u003e: Adds column listing the latest or earliest of the specified columns. Dates must be in YYYY-MM-DD format.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_earliest_or_latest_date_column.pl [table] [0 to select earliest date, 1 to select latest date] \"[title of column with dates]\" \"[title of another column with dates]\" \"[title of another column with dates]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_column_with_earliest_or_latest_date_before_cutoff.pl\"\u003e\u003ccode\u003eadd_column_with_earliest_or_latest_date_before_cutoff.pl\u003c/code\u003e\u003c/a\u003e: Adds column listing the date that is the latest or earliest of the specified columns but earlier than the cut-off date from the cut-off date column. Dates must be in YYYY-MM-DD format.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_column_with_earliest_or_latest_date_before_cutoff.pl [table] [0 to select earliest date, 1 to select latest date] \"[title of column with cut-off date]\" \"[title of column with dates to select from]\" \"[title of another column with dates to select from]\" \"[title of another column with dates to select from]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/sort_date_columns.pl\"\u003e\u003ccode\u003esort_date_columns.pl\u003c/code\u003e\u003c/a\u003e: Sorts the dates in the specified columns. For each row, of the dates in the specified columns, the earliest date will go in the first specified column, the second-earliest in the second specified column, etc. Empty values go last. Dates provided must be in YYYY-MM-DD format.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl sort_date_columns.pl [table] \"[title of column with dates]\" \"[title of another column with dates]\" \"[title of another column with dates]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/sort_date_columns_with_paired_label_columns.pl\"\u003e\u003ccode\u003esort_date_columns_with_paired_label_columns.pl\u003c/code\u003e\u003c/a\u003e: Sorts the dates in the specified columns. For each row, of the dates in the specified columns, the earliest date will go in the first specified column, the second-earliest in the second specified column, etc. Empty values go last. Dates provided must be in YYYY-MM-DD format.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl sort_date_columns_with_paired_label_columns.pl [table] \"[title of column with dates]\" \"[title of label column that should travel with paired dates]\" \"[title of another column with dates]\" \"[title of label column that should travel with those paired dates]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRow manipulation\u003c/h3\u003e\u003ca id=\"user-content-row-manipulation\" class=\"anchor\" aria-label=\"Permalink: Row manipulation\" href=\"#row-manipulation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/filter_table_rows_by_column_value.pl\"\u003e\u003ccode\u003efilter_table_rows_by_column_value.pl\u003c/code\u003e\u003c/a\u003e: Filters table by column values. Only includes rows matching (containing, beginning with, ending with, equal to, not equal to, or not containing) column value of interest in column to filter by. Case-sensitive.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl filter_table_rows_by_column_value.pl [tab-separated table] [0 to match cells containing query, 1: beginning with, 2: ending with, 3: equal to, 4: not equal to, 5: not containing] \"[title of column to filter by]\" \"[column value to select]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/filter_table_rows_by_column_values.pl\"\u003e\u003ccode\u003efilter_table_rows_by_column_values.pl\u003c/code\u003e\u003c/a\u003e: Filters table by column values. Only includes rows matching (containing, beginning with, ending with, equal to, not equal to, or not containing) column values of interest in column to filter by. Case-sensitive.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl filter_table_rows_by_column_values.pl [tab-separated table] [0 to match cells containing query, 1: beginning with, 2: ending with, 3: equal to] \"[title of column to filter by]\" \"[column value to select]\" \"[another column value to select]\" \"[etc.]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/filter_table_rows_by_numerical_column_value.pl\"\u003e\u003ccode\u003efilter_table_rows_by_numerical_column_value.pl\u003c/code\u003e\u003c/a\u003e: Filters table by column values. Only includes rows equal to, greater than, or less than column value of interest in column to filter by.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl filter_table_rows_by_numerical_column_value.pl [tab-separated table] [0 to match cells equal to query, 1: less than, 2: less than or equal to, 3: greater than, 4: greater than or equal to] \"[title of column to filter by]\" \"[column value to select]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/retrieve_subset_of_rows.pl\"\u003e\u003ccode\u003eretrieve_subset_of_rows.pl\u003c/code\u003e\u003c/a\u003e: Subsets table to only rows of interest (those containing one of the specified values in the specified column).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_subset_of_rows.pl [file path of table] \"[title of column to filter by]\" [file path of list of values to filter that column to] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/delete_table_rows_with_column_value.pl\"\u003e\u003ccode\u003edelete_table_rows_with_column_value.pl\u003c/code\u003e\u003c/a\u003e: Deletes rows in table by column values. Only includes rows without column value containing text to filter out in column to filter by. Case-sensitive.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl delete_table_rows_with_column_value.pl [tab-separated table] \"[query to select rows to delete]\" [0 to match cells containing query, 1: beginning with, 2: ending with, 3: equal to] \"[title of column to filter by]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/merge_rows_by_column_value.pl\"\u003e\u003ccode\u003emerge_rows_by_column_value.pl\u003c/code\u003e\u003c/a\u003e: Merges (takes union of) all columns in rows with shared value in column to merge by. If titles of columns not to merge by are provided, leaves one row per input row with all other columns identical. If no columns not to merge by provided, fully merges any rows sharing a value in column to merge by (one row per value).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl merge_rows_by_column_value.pl [table to merge] \"[title of column to merge by]\" \"[optional title of column not to merge]\" \"[optional title of another column not to merge]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/verify_all_rows_in_one_table_appear_in_second_table.pl\"\u003e\u003ccode\u003everify_all_rows_in_one_table_appear_in_second_table.pl\u003c/code\u003e\u003c/a\u003e: Verifies that all rows/lines in table 1 also appear in table 2 (that table 2 is a superset of table 1). Prints all rows from table 1 that are missing from table 2. If column numbers is specified, only compares values from that column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl verify_all_rows_in_one_table_appear_in_second_table.pl [subset table] [superset table] [column number to compare--set to -1 to compare full row] \u0026gt; [output list of missing rows]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/fill_in_missing_rows.pl\"\u003e\u003ccode\u003efill_in_missing_rows.pl\u003c/code\u003e\u003c/a\u003e: Catalogues all values in parameter columns. In any combination of values is missing, adds it as a row with NAs in all other columns.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl fill_in_missing_rows.pl [table] \"[title of column of interest 1]\"\" \"[title of column of interest 2]\" etc. \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTable manipulation\u003c/h3\u003e\u003ca id=\"user-content-table-manipulation\" class=\"anchor\" aria-label=\"Permalink: Table manipulation\" href=\"#table-manipulation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/concatenate_tables.pl\"\u003e\u003ccode\u003econcatenate_tables.pl\u003c/code\u003e\u003c/a\u003e: Concatenates tables with potentially different columns, adding empty space for missing column values.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl concatenate_tables.pl [table1] [table2] [table3] etc. \u0026gt; [concatenated output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_columns_from_other_table.pl\"\u003e\u003ccode\u003eadd_columns_from_other_table.pl\u003c/code\u003e\u003c/a\u003e: Adds columns from another table, matching rows in the two tables by selected columns to merge by.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_columns_from_other_table.pl [table to add columns to] \"[title of column in first table to identify rows by]\" [table to add columns from] \"[title of column in second table to identify rows by]\" \"[title of column to add]\" \"[title of another column to add]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/merge_two_tables_by_column_value.pl\"\u003e\u003ccode\u003emerge_two_tables_by_column_value.pl\u003c/code\u003e\u003c/a\u003e: Merges (takes union of) two tables by the values in the specified columns.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl merge_two_tables_by_column_value.pl [table1 file path] [table1 column number (0-indexed)] [table2 file path] [table2 column number (0-indexed)] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/merge_tables_by_column_value.pl\"\u003e\u003ccode\u003emerge_tables_by_column_value.pl\u003c/code\u003e\u003c/a\u003e: Merges (takes union of) multiple tables by the values in the specified columns. See script for description of input file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl merge_tables_by_column_value.pl [file describing input] \u0026gt; [merged output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/merge_tables_by_first_column_values.pl\"\u003e\u003ccode\u003emerge_tables_by_first_column_values.pl\u003c/code\u003e\u003c/a\u003e: Merges (takes union of) multiple tables by the values in the first column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl merge_tables_by_first_column_values.pl [table] [another table] [another table] [etc.] \u0026gt; [merged output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_newline_between_lines_with_nonconsecutive_values.pl\"\u003e\u003ccode\u003eadd_newline_between_lines_with_nonconsecutive_values.pl\u003c/code\u003e\u003c/a\u003e: Adds newline between lines containing non-consecutive values in first column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_newline_between_lines_with_nonconsecutive_values.pl [table] [column with integer values] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_header_line_to_tables.pl\"\u003e\u003ccode\u003eadd_header_line_to_tables.pl\u003c/code\u003e\u003c/a\u003e: Adds header line to all input files. Saves output tables in new files at same path with _with_header_line.txt extension. Four spaces in the input header line are replaced with tabs.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_header_line_to_tables.pl \"[header line]\" [table] [optional additional table] [etc.]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/split_table_by_column_value.pl\"\u003e\u003ccode\u003esplit_table_by_column_value.pl\u003c/code\u003e\u003c/a\u003e: Splits table into multiple tables, one for each column value in specified column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl split_table_by_column_value.pl [input table file path] \"[title of column to split by]\"\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eReplicates\u003c/h3\u003e\u003ca id=\"user-content-replicates\" class=\"anchor\" aria-label=\"Permalink: Replicates\" href=\"#replicates\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/annotate_replicates.pl\"\u003e\u003ccode\u003eannotate_replicates.pl\u003c/code\u003e\u003c/a\u003e: Assigns a source number to all replicates from the same source. Adds source number as a column to table to annotate.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl annotate_replicates.pl [file listing replicate ids from same source, tab-separated, one line per source] [table to annotate] \"[title of column containing replicate ids in table to annotate]\" \"[optional source column title for output]\" \u0026gt; [annotated output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/select_one_replicate.pl\"\u003e\u003ccode\u003eselect_one_replicate.pl\u003c/code\u003e\u003c/a\u003e: Given a table with multiple replicates from the same source, selects one replicate per source, using selected column to select replicate. In the event of a tie, selects first appearing replicate.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl select_one_replicate.pl [tab-separated table] \"[title of column containing source of each replicate (same value for every replicate from the same source)]\" \"[title of column to use to select replicate]\" [0 to select replicate with smallest numerical value, 1 to select replicate with largest numerical value] \u0026gt; [annotated output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_shared_values_summary_column.pl\"\u003e\u003ccode\u003eadd_shared_values_summary_column.pl\u003c/code\u003e\u003c/a\u003e: Summarizes all values appearing in columns to summarize (sample ids and dates, for example) for each shared value (patient id, for example). Adds summary in new column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_shared_values_summary_column.pl [tab-separated table] \"[title of column containing values shared by rows]\" \"[title of column to include in summary of shared values]\" \"[title of another column to include in summary of shared values]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eOther\u003c/h3\u003e\u003ca id=\"user-content-other\" class=\"anchor\" aria-label=\"Permalink: Other\" href=\"#other\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/summarize_table_columns.pl\"\u003e\u003ccode\u003esummarize_table_columns.pl\u003c/code\u003e\u003c/a\u003e: Summarizes values in table columns. Similar to str in R.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl summarize_table_columns.pl [tab-separated table] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/summarize_column_value_combination_frequencies.pl\"\u003e\u003ccode\u003esummarize_column_value_combination_frequencies.pl\u003c/code\u003e\u003c/a\u003e: Summarizes frequency of all column values and combinations of column values in columns of interest in table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl summarize_column_value_combination_frequencies.pl [table to summarize] \"[title of column of interest]\" \"[optional title of another column of interest]\" \"[optional title of another column of interest]\" [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_column_indicating_presence_of_query.pl\"\u003e\u003ccode\u003eadd_column_indicating_presence_of_query.pl\u003c/code\u003e\u003c/a\u003e: Adds column with values indicating presence of query in row.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_column_indicating_presence_of_query.pl [table] [query] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/add_row_indicating_presence_of_query.pl\"\u003e\u003ccode\u003eadd_row_indicating_presence_of_query.pl\u003c/code\u003e\u003c/a\u003e: Adds row with values indicating presence of query in column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_row_indicating_presence_of_query.pl [table] [query] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/count_occurrences_of_column_values.pl\"\u003e\u003ccode\u003ecount_occurrences_of_column_values.pl\u003c/code\u003e\u003c/a\u003e: Counts number occurrences of each value in selected column of table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl count_occurrences_of_column_values.pl [tab-separated table] \"[column title]\" \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/verify_column_values_are_consistent_across_tables.pl\"\u003e\u003ccode\u003everify_column_values_are_consistent_across_tables.pl\u003c/code\u003e\u003c/a\u003e: Verifies that the same column values always appear with row identifier values. Column titles must be consistent across tables, including title of row identifier column.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl verify_column_values_are_consistent_across_tables.pl [row identifier column title] [table1] [table2] [table 3] [etc.]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/tables/calculate_date_range_in_column.pl\"\u003e\u003ccode\u003ecalculate_date_range_in_column.pl\u003c/code\u003e\u003c/a\u003e: Prints minimum date, maximum date, and difference in dates (in days) in specified column in each input files. Dates must be in YYYY-MM-DD format.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl calculate_date_range_in_column.pl [column number of column with dates (0-indexed)] [table] [another table] [another table] [etc.]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMiscellaneous (\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc\"\u003e\u003ccode\u003emisc\u003c/code\u003e\u003c/a\u003e)\u003c/h2\u003e\u003ca id=\"user-content-miscellaneous-misc\" class=\"anchor\" aria-label=\"Permalink: Miscellaneous (misc)\" href=\"#miscellaneous-misc\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/add_commas_to_all_numbers_in_file.pl\"\u003e\u003ccode\u003eadd_commas_to_all_numbers_in_file.pl\u003c/code\u003e\u003c/a\u003e: Detects all numbers in contents of file and adds thousands-separator commas where needed.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_commas_to_all_numbers_in_file.pl [input file to add commas to] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/add_commas_to_all_numbers.pl\"\u003e\u003ccode\u003eadd_commas_to_all_numbers.pl\u003c/code\u003e\u003c/a\u003e: Detects all numbers in input text and adds thousands-separator commas where needed.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl add_commas_to_all_numbers.pl \"[text to add commas to]\" \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/run_script_on_multiple_files.pl\"\u003e\u003ccode\u003erun_script_on_multiple_files.pl\u003c/code\u003e\u003c/a\u003e: Runs script individually on multiple input files, otherwise using the same input parameters. Script must accept input file as first parameter.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl run_script_on_multiple_files.pl [file path of script to run] [file with list of input files to run script on] [optional extension to add to input file path to create output file path] [optional second parameter to provide script] [optional third parameter to provide script] [etc.]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/summarize_files.pl\"\u003e\u003ccode\u003esummarize_files.pl\u003c/code\u003e\u003c/a\u003e: Summarizes files. Generates table with filepath, filename, filename with extensions trimmed off, and number lines, words, and characters in the file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl summarize_files.pl [file with list of files to summarize, one per line] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/download_files.pl\"\u003e\u003ccode\u003edownload_files.pl\u003c/code\u003e\u003c/a\u003e: Downloads files listed in input file from online or from google storage bucket.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl download_files.pl [file with list of files to download] [optional output directory]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/download_files_listed_in_table_column.pl\"\u003e\u003ccode\u003edownload_files_listed_in_table_column.pl\u003c/code\u003e\u003c/a\u003e: Downloads files from addresses listed in specified table column in input table. Downloads files from online or from google storage bucket to new directory. If no output directory provided, output directory set to input file path sans extension followed by column name.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl download_files_listed_in_table_column.pl [table with list of files to download in one of the columns] \"[title of column containing filepaths to download]\" [optional output directory]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/download_and_rename_files.pl\"\u003e\u003ccode\u003edownload_and_rename_files.pl\u003c/code\u003e\u003c/a\u003e: Downloads files listed in input file from online or from google storage bucket to new directory and renames to specified filename.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl download_and_rename_files.pl [file with list of files to download] [optional output directory]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/retrieve_all_filepaths_in_directories.pl\"\u003e\u003ccode\u003eretrieve_all_filepaths_in_directories.pl\u003c/code\u003e\u003c/a\u003e: Retrieves filepaths of all files in provided directories.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl retrieve_all_filepaths_in_directories.pl [directory path] [another directory path] [another directory path] [etc.] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/bulk_diff_same_name_files_in_two_directories.pl\"\u003e\u003ccode\u003ebulk_diff_same_name_files_in_two_directories.pl\u003c/code\u003e\u003c/a\u003e: Compares all files with the same name in the two input directories. Prints any differences between same-name files. The two directories must contain files with the same names.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl bulk_diff_same_name_files_in_two_directories.pl [first directory] [second directory]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/combine_files_and_delete_duplicate_lines.pl\"\u003e\u003ccode\u003ecombine_files_and_delete_duplicate_lines.pl\u003c/code\u003e\u003c/a\u003e: Combines input files. Prints each line only once (no duplicate lines). Prints lines in order in which they first appear in all the input files.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl combine_files_and_delete_duplicate_lines.pl [input file] [another input file] [another input file] [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/bulk_rename.pl\"\u003e\u003ccode\u003ebulk_rename.pl\u003c/code\u003e\u003c/a\u003e: Replaces all occurrences of values in file paths to mapped replacement values. Values to replace should not overlap or be substrings of each other. Files already at replacement paths will be overwritten. Paths of files to rename can be provided directly as arguments or as one file with a list of filepaths, one per line.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl bulk_rename.pl [tab-separated file mapping current values (first column) to new values (second column)] [filepaths of at least two files to rename OR path of file containing filepaths of files to rename, one per line]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/bulk_grep.pl\"\u003e\u003ccode\u003ebulk_grep.pl\u003c/code\u003e\u003c/a\u003e: Searches all input files for queries listed in query list file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl bulk_grep.pl [file listing queries, one per line] [file to grep] [another file to grep] [etc.] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/bulk_grep_one_file.pl\"\u003e\u003ccode\u003ebulk_grep_one_file.pl\u003c/code\u003e\u003c/a\u003e: Searches input file for queries. Prints lines containing queries.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl bulk_grep_one_file.pl [file to grep] \"[query 1]\" \"[query 2]\" \"[query 3]\" [etc.] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/bulk_grep_is_detected.pl\"\u003e\u003ccode\u003ebulk_grep_is_detected.pl\u003c/code\u003e\u003c/a\u003e: Searches all input files for queries listed in query list file. Outputs query and each file it was found in, tab-separated, one line per query-file pair where query is detected. If query is not detected in any file, prints query followed by \"not detected\".\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl bulk_grep_is_detected.pl [file listing queries, one per line] [file to grep] [another file to grep] [etc.] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/bulk_grep_is_detected_table.pl\"\u003e\u003ccode\u003ebulk_grep_is_detected_table.pl\u003c/code\u003e\u003c/a\u003e: Searches all input files for queries listed in query list file. Outputs a table with detection of all queries in all files, tab-separated, one row per query, one column per file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl bulk_grep_is_detected_table.pl [file listing queries, one per line] [file to grep] [another file to grep] [etc.] \u0026gt; [output file path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/split_file_into_n_files.pl\"\u003e\u003ccode\u003esplit_file_into_n_files.pl\u003c/code\u003e\u003c/a\u003e: Splits file with multiple lines up into a number of smaller files, each with about the same number of lines.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl split_file_into_n_files.pl [file path]  [number output files to generate]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/split_file_into_2n_overlapping_files.pl\"\u003e\u003ccode\u003esplit_file_into_2n_overlapping_files.pl\u003c/code\u003e\u003c/a\u003e: Splits file with multiple lines up into a number of overlapping smaller files. Each line will appear in two files. Either all output files will have about the same number of lines, or there will be two \"jagged\" half-sized output files if it is a priority for all output files to only have lines that are consecutive in the input. See script for example input and output.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl split_file_into_2n_overlapping_files.pl [file path] [number equal slices (n)] [1 to generate N=2*n approximately equal sized files, 0 to generate N=2*n+1 output files with lines that are also consecutive in the input] [optional 1 to NOT print half-sized overhang files (can be used only if previous option is set to 0)]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/split_file_into_overlapping_files_by_date.pl\"\u003e\u003ccode\u003esplit_file_into_overlapping_files_by_date.pl\u003c/code\u003e\u003c/a\u003e: Splits file with multiple lines up into overlapping smaller files, each containing dates within twice the parameter number of days. Each line will appear in two files.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl split_file_into_overlapping_files_by_date.pl [file path] [half the date range in each output file] [column containing dates, 0-indexed]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/erase_empty_files.pl\"\u003e\u003ccode\u003eerase_empty_files.pl\u003c/code\u003e\u003c/a\u003e: Erases empty files.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl erase_empty_files.pl [file to check and potentially erase] [another file to check and potentially erase] [a third file to check and potentially erase] etc.\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/bulk_delete_files_in_list.pl\"\u003e\u003ccode\u003ebulk_delete_files_in_list.pl\u003c/code\u003e\u003c/a\u003e: Deletes files in input list.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl bulk_delete_files_in_list.pl [file containing list of paths of files to delete, one per line]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/bulk_move_files_in_list.pl\"\u003e\u003ccode\u003ebulk_move_files_in_list.pl\u003c/code\u003e\u003c/a\u003e: Moves files in input list to provided directory.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl bulk_move_files_in_list.pl [file containing list of paths of files to move, one per line] [directory to move files to]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/match_file_paths_to_sample_names.pl\"\u003e\u003ccode\u003ematch_file_paths_to_sample_names.pl\u003c/code\u003e\u003c/a\u003e: Matches each sample name to a file path containing that sample name. Outputs sample names with file paths, tab-separated, one sample per line.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl match_file_paths_to_sample_names.pl [file containing list of sample names, one per line] [file containing list of file paths, one per line] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/combine_files_and_delete_duplicate_lines.pl\"\u003e\u003ccode\u003ecombine_files_and_delete_duplicate_lines.pl\u003c/code\u003e\u003c/a\u003e: Combines input files. Prints each line only once (no duplicate lines). Prints lines in order in which they first appear in all the input files.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl combine_files_and_delete_duplicate_lines.pl [input file] [another input file] [another input file] [etc.] \u0026gt; [output table path]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"/lakras/bio-helper-scripts/blob/main/misc/select_random_lines.pl\"\u003e\u003ccode\u003eselect_random_lines.pl\u003c/code\u003e\u003c/a\u003e: Selects a certain number of lines at random from input table or list.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eUsage: \u003ccode\u003eperl select_random_lines.pl [table or list] [number rows or lines to select at random] \u0026gt; [output table or list]\u003c/code\u003e\u003c/em\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eMore coming soon :)\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"bio-helper-scripts","anchor":"bio-helper-scripts","htmlText":"bio-helper-scripts"},{"level":2,"text":"Contents","anchor":"contents","htmlText":"Contents"},{"level":2,"text":"SARS-CoV-2 or other genome lineages (lineages)","anchor":"sars-cov-2-or-other-genome-lineages-lineages","htmlText":"SARS-CoV-2 or other genome lineages (lineages)"},{"level":2,"text":"FASTA file processing (fasta)","anchor":"fasta-file-processing-fasta","htmlText":"FASTA file processing (fasta)"},{"level":2,"text":"FASTA alignment file processing (aligned-fasta)","anchor":"fasta-alignment-file-processing-aligned-fasta","htmlText":"FASTA alignment file processing (aligned-fasta)"},{"level":2,"text":"BLAST (blast)","anchor":"blast-blast","htmlText":"BLAST (blast)"},{"level":2,"text":"VCF file processing (vcf-files)","anchor":"vcf-file-processing-vcf-files","htmlText":"VCF file processing (vcf-files)"},{"level":2,"text":"BED file processing (bed-files)","anchor":"bed-file-processing-bed-files","htmlText":"BED file processing (bed-files)"},{"level":2,"text":"Read depth file processing (read-depths)","anchor":"read-depth-file-processing-read-depths","htmlText":"Read depth file processing (read-depths)"},{"level":2,"text":"Tables (tables)","anchor":"tables-tables","htmlText":"Tables (tables)"},{"level":3,"text":"Table Format","anchor":"table-format","htmlText":"Table Format"},{"level":3,"text":"Column title manipulation","anchor":"column-title-manipulation","htmlText":"Column title manipulation"},{"level":3,"text":"Column manipulation","anchor":"column-manipulation","htmlText":"Column manipulation"},{"level":3,"text":"Column manipulation with dates","anchor":"column-manipulation-with-dates","htmlText":"Column manipulation with dates"},{"level":3,"text":"Row manipulation","anchor":"row-manipulation","htmlText":"Row manipulation"},{"level":3,"text":"Table manipulation","anchor":"table-manipulation","htmlText":"Table manipulation"},{"level":3,"text":"Replicates","anchor":"replicates","htmlText":"Replicates"},{"level":3,"text":"Other","anchor":"other","htmlText":"Other"},{"level":2,"text":"Miscellaneous (misc)","anchor":"miscellaneous-misc","htmlText":"Miscellaneous (misc)"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Flakras%2Fbio-helper-scripts"}},{"displayName":"LICENSE","repoName":"bio-helper-scripts","refName":"main","path":"LICENSE","preferredFileType":"license","tabName":"MIT","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Flakras%2Fbio-helper-scripts"}}],"overviewFilesProcessingTime":48.866814}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
.bhqztV{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.bhqztV span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.bhqztV:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.bhqztV:focus{outline:2px solid transparent;}/*!sc*/
.bhqztV:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.bhqztV:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.bhqztV::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:transparent;border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.bhqztV::after{background-color:transparent;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,bhqztV,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/lakras/bio-helper-scripts/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/lakras/bio-helper-scripts/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/lakras/bio-helper-scripts/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/lakras/bio-helper-scripts/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/lakras/bio-helper-scripts/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           555 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/lakras/bio-helper-scripts/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="aligned-fasta, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/aligned-fasta" title="aligned-fasta">
                          aligned-fasta
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="aligned-fasta, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/aligned-fasta" title="aligned-fasta">
                          aligned-fasta
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="bed-files, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/bed-files" title="bed-files">
                          bed-files
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="bed-files, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/bed-files" title="bed-files">
                          bed-files
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="blast, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/blast" title="blast">
                          blast
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="blast, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/blast" title="blast">
                          blast
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="fasta, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/fasta" title="fasta">
                          fasta
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="fasta, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/fasta" title="fasta">
                          fasta
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="lineages, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/lineages" title="lineages">
                          lineages
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="lineages, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/lineages" title="lineages">
                          lineages
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="misc, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/misc" title="misc">
                          misc
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="misc, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/misc" title="misc">
                          misc
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-6">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="read-depths, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/read-depths" title="read-depths">
                          read-depths
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="read-depths, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/read-depths" title="read-depths">
                          read-depths
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-7">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tables, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/tables" title="tables">
                          tables
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tables, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/tables" title="tables">
                          tables
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-8">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="vcf-files, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/vcf-files" title="vcf-files">
                          vcf-files
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="vcf-files, (Directory)" class="Link--primary" href="/lakras/bio-helper-scripts/tree/main/vcf-files" title="vcf-files">
                          vcf-files
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-9">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/lakras/bio-helper-scripts/blob/main/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/lakras/bio-helper-scripts/blob/main/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-10">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/lakras/bio-helper-scripts/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/lakras/bio-helper-scripts/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd show-for-mobile" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-law" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="MIT license">
                        MIT license
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      bio-helper-scripts
                     </h1>
                     <a aria-label="Permalink: bio-helper-scripts" class="anchor" href="#bio-helper-scripts" id="user-content-bio-helper-scripts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Helper scripts for processing genomic sequence data.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Contents
                     </h2>
                     <a aria-label="Permalink: Contents" class="anchor" href="#contents" id="user-content-contents">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <a href="#sars-cov-2-or-other-genome-lineages-lineages">
                       SARS-CoV-2 or other genome lineages
                      </a>
                     </li>
                     <li>
                      <a href="#fasta-file-processing-fasta">
                       FASTA file processing
                      </a>
                     </li>
                     <li>
                      <a href="#fasta-alignment-file-processing-aligned-fasta">
                       FASTA alignment file processing
                      </a>
                     </li>
                     <li>
                      <a href="#blast-blast">
                       BLAST
                      </a>
                     </li>
                     <li>
                      <a href="#vcf-file-processing-vcf-files">
                       VCF file processing
                      </a>
                     </li>
                     <li>
                      <a href="#bed-file-processing-bed-files">
                       BED file processing
                      </a>
                     </li>
                     <li>
                      <a href="#read-depth-file-processing-read-depths">
                       Read depth file processing
                      </a>
                     </li>
                     <li>
                      <a href="#tables-tables">
                       Tables
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#table-format">
                         Table Format
                        </a>
                       </li>
                       <li>
                        <a href="#column-title-manipulation">
                         Column title manipulation
                        </a>
                       </li>
                       <li>
                        <a href="#column-manipulation">
                         Column manipulation
                        </a>
                       </li>
                       <li>
                        <a href="#column-manipulation-with-dates">
                         Column manipulation with dates
                        </a>
                       </li>
                       <li>
                        <a href="#row-manipulation">
                         Row manipulation
                        </a>
                       </li>
                       <li>
                        <a href="#table-manipulation">
                         Table manipulation
                        </a>
                       </li>
                       <li>
                        <a href="#replicates">
                         Replicates
                        </a>
                       </li>
                       <li>
                        <a href="#other">
                         Other
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#miscellaneous-misc">
                       Miscellaneous
                      </a>
                     </li>
                    </ul>
                    <p dir="auto">
                     Includes the following scripts
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      SARS-CoV-2 or other genome lineages (
                      <a href="/lakras/bio-helper-scripts/blob/main/lineages">
                       <code>
                        lineages
                       </code>
                      </a>
                      )
                     </h2>
                     <a aria-label="Permalink: SARS-CoV-2 or other genome lineages (lineages)" class="anchor" href="#sars-cov-2-or-other-genome-lineages-lineages" id="user-content-sars-cov-2-or-other-genome-lineages-lineages">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/lineages/summarize_lineage_defining_SNPs.pl">
                        <code>
                         summarize_lineage_defining_SNPs.pl
                        </code>
                       </a>
                       : Prints list of lineage-defining positions and the lineage(s) consistent with each allele.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl summarize_lineage_defining_SNPs.pl [alignment fasta file path] [optional 1 to compare all positions, including with gaps or ambiguous bases] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/lineages/estimate_allele_lineages.pl">
                        <code>
                         estimate_allele_lineages.pl
                        </code>
                       </a>
                       : Generates table listing lineage(s) consistent with each sample's consensus and minor alleles. See script for descriptions of input files and output table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl estimate_allele_lineages.pl [lineage genomes aligned to reference] [consensus genomes aligned to reference] [optional list of heterozygosity tables] [optional list of read depth files] [1 to print columns for each lineage, 0 to print columns for consensus and minor alleles] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/lineages/annotate_heterozygosity_tables_with_estimated_lineages.pl">
                        <code>
                         annotate_heterozygosity_tables_with_estimated_lineages.pl
                        </code>
                       </a>
                       : Annotates heterozygosity tables with lineage consistent with minor and consensus-level alleles at lineage-defining positions. Output is printed to one file per heterozygosity table or as one table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl annotate_heterozygosity_tables_with_estimated_lineages.pl [lineage genomes aligned to reference] [list of heterozygosity tables] [1 to print each heterozygosity table separately, 0 to print all tables to console] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/lineages/determine_distances_to_lineages_from_alignment.pl">
                        <code>
                         determine_distances_to_lineages_from_alignment.pl
                        </code>
                       </a>
                       : Generates 2d table of distances between all sequences in alignment to lineage sequences.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl determine_distances_to_lineages_from_alignment.pl [alignment fasta file path] [1 to ignore first sequence in alignment, 0 to include it] "[name of lineage sequence]" "[name of another lineage sequence]" [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      FASTA file processing (
                      <a href="/lakras/bio-helper-scripts/blob/main/fasta">
                       <code>
                        fasta
                       </code>
                      </a>
                      )
                     </h2>
                     <a aria-label="Permalink: FASTA file processing (fasta)" class="anchor" href="#fasta-file-processing-fasta" id="user-content-fasta-file-processing-fasta">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/summarize_fasta_sequences.pl">
                        <code>
                         summarize_fasta_sequences.pl
                        </code>
                       </a>
                       : Summarizes sequences in fasta file. Produces table with, for each sequence: number bases, number unambiguous bases, A+T count, C+G count, number Ns, number gaps, number As, number Ts, number Cs, number Gs, and counts for any other
                       <a href="https://en.wikipedia.org/wiki/Nucleic_acid_notation" rel="nofollow">
                        bases
                       </a>
                       that appear.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl summarize_fasta_sequences.pl [fasta file path] [another fasta file path] [etc.] &gt; [output table file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/calculate_genome_size.pl">
                        <code>
                         calculate_genome_size.pl
                        </code>
                       </a>
                       : Counts number bases (including Ns and other ambiguous non-gap letters) in fasta file(s).
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl calculate_genome_size.pl [fasta file path] [another fasta file path] [etc.]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/align_to_reference.pl">
                        <code>
                         align_to_reference.pl
                        </code>
                       </a>
                       : Aligns each input sequence with reference independently, then combines all into one fasta alignment. Bases aligned to a gap in the reference are removed. Sequences in input fastas must have unique names. Paths of fasta sequences to align can be provided directly as arguments or as one file with a list of filepaths, one per line.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl align_to_reference.pl [reference sequence] [file path of MAFFT executable file (mafft.bat) or mafft command] [filepaths of at least fasta files OR path of file containing file with list of fasta file paths, one per line] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/align_each_fasta_file_to_reference_separately.pl">
                        <code>
                         align_each_fasta_file_to_reference_separately.pl
                        </code>
                       </a>
                       : Adds reference sequence to the top of each fasta file and aligns each file separately. New files are created at filepath of old file with "_aligned_to_ref.fasta" appended to the end. Temp files created with "_temp.fasta" appended to the end. Files already at those paths will be overwritten.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl align_each_fasta_file_to_reference_separately.pl [reference sequence] [file path of MAFFT executable file (mafft.bat) or mafft command] [fasta file path] [another fasta file path] [another fasta file path] [etc.]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_by_name.pl">
                        <code>
                         retrieve_sequences_by_name.pl
                        </code>
                       </a>
                       : Retrieves query sequences by name from fasta file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_sequences_by_name.pl [fasta file path] "[query sequence name 1]" "[query sequence name 2]" [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_by_names_listed_in_file.pl">
                        <code>
                         retrieve_sequences_by_names_listed_in_file.pl
                        </code>
                       </a>
                       : Retrieves query sequences by name from fasta file, taking query sequence names from a list in a file, one sequence name per line.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_sequences_by_names_listed_in_file.pl [fasta file path] [file with list of query sequence names] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_containing_queries.pl">
                        <code>
                         retrieve_sequences_containing_queries.pl
                        </code>
                       </a>
                       : Prints all sequences from fasta file that contain at least one provided query in the sequence name. Case sensitive.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_sequences_containing_queries.pl [fasta file path] "[query 1]" "[query 2]" "[query 3]" [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_by_position_in_fasta_file.pl">
                        <code>
                         retrieve_sequences_by_position_in_fasta_file.pl
                        </code>
                       </a>
                       : Retrieves query sequences by position from fasta file (in other words, retrieves the first fasta sequence in the file, or the second, or the third, and so on).
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_sequences_by_position_in_fasta_file.pl [fasta file path] [position of sequence to retrieve (1-indexed)] [position of another sequence to retrieve] [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/remove_sequences_by_position_in_fasta_file.pl">
                        <code>
                         remove_sequences_by_position_in_fasta_file.pl
                        </code>
                       </a>
                       : Removes query sequences by position from fasta file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl remove_sequences_by_position_in_fasta_file.pl [fasta file path] [position of sequence to remove (1-indexed)] [position of another sequence to remove] [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/select_random_sequences.pl">
                        <code>
                         select_random_sequences.pl
                        </code>
                       </a>
                       : Selects a certain number of sequences at random from input fasta file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl select_random_sequences.pl [fasta sequence file] [number sequences to select at random] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/filter_sequences_by_length.pl">
                        <code>
                         filter_sequences_by_length.pl
                        </code>
                       </a>
                       : Filters fasta file by sequence length.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl filter_sequences_by_length.pl [fasta file path] [minimum length] [1 to filter by number of unambiguous bases, 0 to filter on number of bases (including Ns)] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/reheader_fasta_sequences_from_filename.pl">
                        <code>
                         reheader_fasta_sequences_from_filename.pl
                        </code>
                       </a>
                       : Sets file name as header line in each fasta file(s). Each fasta file should have only one sequence.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl reheader_fasta_sequences_from_filename.pl [fasta file path] [another fasta file path] [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/add_prefix_to_fasta_headers.pl">
                        <code>
                         add_prefix_to_fasta_headers.pl
                        </code>
                       </a>
                       : Adds prefix to each header line in fasta file(s).
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_prefix_to_fasta_headers.pl [prefix to add to fasta file headers] [fasta file path] [another fasta file path] [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/add_filename_as_prefix_to_fasta_headers.pl">
                        <code>
                         add_filename_as_prefix_to_fasta_headers.pl
                        </code>
                       </a>
                       : Adds file name as prefix to each header line in fasta file(s).
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_filename_as_prefix_to_fasta_headers.pl [fasta file path] [another fasta file path] [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/replace_fasta_headers_with_filenames.pl">
                        <code>
                         replace_fasta_headers_with_filenames.pl
                        </code>
                       </a>
                       : Replaces header line in fasta file(s) with file name.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl replace_fasta_headers_with_filenames.pl [fasta file path] [another fasta file path] [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/split_fasta_into_n_files.pl">
                        <code>
                         split_fasta_into_n_files.pl
                        </code>
                       </a>
                       : Splits up fasta file into a set number of smaller files, each with about the same number of sequences.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl split_fasta_into_n_files.pl [fasta file path]  [number output files to generate]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/split_fasta_n_sequences_per_file.pl">
                        <code>
                         split_fasta_n_sequences_per_file.pl
                        </code>
                       </a>
                       : Splits up fasta file into multiple files with a set number of sequences per file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl split_fasta_n_sequences_per_file.pl [fasta file path]  [number sequences per file]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/split_fasta_one_sequence_per_file.pl">
                        <code>
                         split_fasta_one_sequence_per_file.pl
                        </code>
                       </a>
                       : Splits up fasta file into multiple files with one sequence per file. Each output file is named using its sequence name.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl split_fasta_one_sequence_per_file.pl [fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/download_fasta_sequences_from_accession_numbers.pl">
                        <code>
                         download_fasta_sequences_from_accession_numbers.pl
                        </code>
                       </a>
                       : Given a list of accession numbers, one per line, downloads and prints fasta sequences corresponding to each accession number.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl download_fasta_sequences_from_accession_numbers.pl [path of file with list of accession numbers, one per line] [database (nucleotide by default)] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/merge_fastas_and_remove_redundant_sequences.pl">
                        <code>
                         merge_fastas_and_remove_redundant_sequences.pl
                        </code>
                       </a>
                       : Merges fasta files, removing redundant sequences (so there is only one of each sequence, regardless of name). Can also be used to remove redundant sequences from a single fasta file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl merge_fastas_and_remove_redundant_sequences.pl [fasta file path] [another fasta file path] [another fasta file path] [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/modify_unaligned_fasta.pl">
                        <code>
                         modify_unaligned_fasta.pl
                        </code>
                       </a>
                       : Modifies unaligned fasta file according to allele changes specified in changes table. Not designed to handle gaps. See script for description of changes table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl modify_unaligned_fasta.pl [alignment fasta file path] [changes table] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequence_position_in_fasta_file.pl">
                        <code>
                         retrieve_sequence_position_in_fasta_file.pl
                        </code>
                       </a>
                       : Prints position of sequence in fasta file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_sequence_position_in_fasta_file.pl [fasta sequence file] "[sequence name]"
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/slice_fasta_file_before_or_after_sequence_name_query.pl">
                        <code>
                         slice_fasta_file_before_or_after_sequence_name_query.pl
                        </code>
                       </a>
                       : Prints either all sequences before or all sequence after sequence name appears in fasta sequence file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl slice_fasta_file_before_or_after_sequence_name_query.pl [fasta file] [sequence name to slice before or after] [1 to print all sequences BEFORE sequence name, 0 to print all sequences AFTER sequence name] [1 to print sequences including sequence name, 0 not to] &gt; [output fasta file]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/remove_duplicate_name_sequences.pl">
                        <code>
                         remove_duplicate_name_sequences.pl
                        </code>
                       </a>
                       : Removes any sequence whose full name has already appeared in the fasta file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl remove_duplicate_name_sequences.pl [fasta file path] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_appearing_or_not_appearing_in_table.pl">
                        <code>
                         retrieve_sequences_appearing_or_not_appearing_in_table.pl
                        </code>
                       </a>
                       : Retrieves sequences whose names do or do not appear in input table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_sequences_appearing_or_not_appearing_in_table.pl [fasta file] [input table] [column number of sequence names (0-indexed)] [1 to retrieve sequences that DO appear in input table, 0 to retrieve sequences that DON'T] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/search_for_query_sequences.pl">
                        <code>
                         search_for_query_sequences.pl
                        </code>
                       </a>
                       : Searches provided sequences for provided query sequences. Outputs table with columns: name of sequence containing at least one query sequence, number of query sequences matched.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl search_for_query_sequences.pl [fasta file path containing sequences to search] [fasta file path containing queries] &gt; [output table file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/retrieve_sequences_matching_at_least_N_query_sequences.pl">
                        <code>
                         retrieve_sequences_matching_at_least_N_query_sequences.pl
                        </code>
                       </a>
                       : Searches provided sequences for provided query sequences. Outputs sequences matching at least N query sequences.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_sequences_matching_at_least_N_query_sequences.pl [fasta file path containing sequences to search] [fasta file path containing queries] [minimum number queries to match] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/filter_by_number_unambiguous_bases_with_minimum_read_depth.pl">
                        <code>
                         filter_by_number_unambiguous_bases_with_minimum_read_depth.pl
                        </code>
                       </a>
                       : Filters sequences in fasta file by number positions with an unambiguous base (A, T, C, or G) and read depth at least the minimum read depth.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl filter_by_number_unambiguous_bases_with_minimum_read_depth.pl [fasta file path] [list of read depth tables] [minimum read depth] [minimum number unabiguous bases with at least minimum read depth] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/fasta/split_fasta_into_files_by_category_map.pl">
                        <code>
                         split_fasta_into_files_by_category_map.pl
                        </code>
                       </a>
                       : Splits fasta file into multiple files using map of sequence name to category.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl split_fasta_into_files_by_category_map.pl [fasta file path] [tab-separated map of sequence names to category, one sequence name per line] [directory to print output fasta files to]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_sequences_with_no_blast_hits.pl">
                        <code>
                         blast/retrieve_sequences_with_no_blast_hits.pl
                        </code>
                       </a>
                       : Retrieves sequences that do not have blast results.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_sequences_with_no_blast_hits.pl [blast output file] [fasta file that was input to blast] [minimum percent identity for a blast hit to be counted] [minimum query coverage for a blast hit to be counted] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_sequences_with_blast_hits.pl">
                        <code>
                         blast/retrieve_sequences_with_blast_hits.pl
                        </code>
                       </a>
                       : Retrieves sequences that have blast results.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_sequences_with_blast_hits.pl [blast output file] [fasta file that was input to blast] [minimum percent identity for a blast hit to be counted] [minimum query coverage for a blast hit to be counted] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/extract_hits_or_sequences_with_top_hit_in_taxon.pl">
                        <code>
                         blast/extract_hits_or_sequences_with_top_hit_in_taxon.pl
                        </code>
                       </a>
                       : Retrieves blast hits or fasta sequences of sequences with top hit or any hit at all in taxon of interest or its children.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl extract_hits_or_sequences_with_top_hit_in_taxon.pl [blast output table] [fasta file that was input to blast] [nodes.dmp file from NCBI] [taxon id of taxon of interest] [1 to print fasta sequences, 0 to print subset of blast output] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/bed-files/pull_out_sequences_described_in_bed_file.pl">
                        <code>
                         bed-files/pull_out_sequences_described_in_bed_file.pl
                        </code>
                       </a>
                       : Pulls out subsets of sequences by name and position within that sequence from bed file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl pull_out_sequences_described_in_bed_file.pl [fasta file path] [bed file path] [0 to output fasta sequence, 1 to add sequence as new column in bed file] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      FASTA alignment file processing (
                      <a href="/lakras/bio-helper-scripts/blob/main/aligned-fasta">
                       <code>
                        aligned-fasta
                       </code>
                      </a>
                      )
                     </h2>
                     <a aria-label="Permalink: FASTA alignment file processing (aligned-fasta)" class="anchor" href="#fasta-alignment-file-processing-aligned-fasta" id="user-content-fasta-alignment-file-processing-aligned-fasta">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/aligned-fasta/generate_distance_table_from_alignment.pl">
                        <code>
                         generate_distance_table_from_alignment.pl
                        </code>
                       </a>
                       : Generates 2d table of distances between all sequences in alignment.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl generate_distance_table_from_alignment.pl [alignment fasta file path] [1 to ignore first sequence in alignment, 0 to include it] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/aligned-fasta/modify_alignment_fasta.pl">
                        <code>
                         modify_alignment_fasta.pl
                        </code>
                       </a>
                       : Modifies aligned fasta file according to allele changes specified in changes table. See script for description of changes table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl modify_alignment_fasta.pl [alignment fasta file path] [changes table] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/aligned-fasta/mask_positions.pl">
                        <code>
                         mask_positions.pl
                        </code>
                       </a>
                       : Masks (replaces with Ns) alleles at indicated positions. Positions must be relative to same reference appearing in alignment fasta file. Reference must be first sequence in alignment fasta file. Does not mask bases that align to gaps in reference.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl mask_positions.pl [alignment fasta file path] [first position in region to mask] [last position in region to mask] [first position in another region to mask] [last position in another region to mask] [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/aligned-fasta/mask_positions_in_bulk.pl">
                        <code>
                         mask_positions_in_bulk.pl
                        </code>
                       </a>
                       : Masks (replaces with Ns) alleles at indicated positions. Operates separately on each sequence as described in input table. Positions must be relative to same reference appearing in alignment fasta file. Reference must be first sequence in alignment fasta file. Does not mask bases that align to gaps in reference.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl mask_positions_in_bulk.pl [alignment fasta file path] [tab-separated table containing sequence name in first column, first and last positions of regions to mask in this sequence, space separated, in second column] [optional first position in additional region to mask in all sequences] [optional last position in additional region to mask in all sequences] [optional first position in another additional region to mask in all sequences] [optional last position in another additional region to mask in all sequences] [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/aligned-fasta/mask_low_read_depth_alleles.pl">
                        <code>
                         mask_low_read_depth_alleles.pl
                        </code>
                       </a>
                       : Masks (replaces with Ns) alleles with low read depths.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl mask_low_read_depth_alleles.pl [alignment fasta file path] [list of read depth tables] [minimum read depth] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/aligned-fasta/remove_reference_gaps_in_alignment.pl">
                        <code>
                         remove_reference_gaps_in_alignment.pl
                        </code>
                       </a>
                       : Removes gaps in reference (first sequence) in alignment and bases or gaps at the corresponding positions in all other sequences in the alignment.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl remove_reference_gaps_in_alignment.pl [alignment fasta file path] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/aligned-fasta/combine_alignments_to_same_reference.pl">
                        <code>
                         combine_alignments_to_same_reference.pl
                        </code>
                       </a>
                       : Combines multiple fasta files including the same reference sequence into one fasta alignment. Reference fasta is printed first, with no gaps. All other sequences are printed aligned to the reference as in the input. Bases aligned to a gap in the reference are removed. Sequences in input files must have unique names. First sequence in each alignment fasta file must be reference.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl combine_alignments_to_same_reference.pl [alignment fasta file path] [another alignment fasta file path] [another alignment fasta file path] [etc.] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/aligned-fasta/collapse_aligned_sequences_by_name.pl">
                        <code>
                         collapse_aligned_sequences_by_name.pl
                        </code>
                       </a>
                       : Merges aligned sequences with same name up to a ": ", such as those output by LASTZ.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl collapse_aligned_sequences_by_name.pl [alignment fasta file path] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/aligned-fasta/get_average_sequence_of_aligned_sequences.pl">
                        <code>
                         get_average_sequence_of_aligned_sequences.pl
                        </code>
                       </a>
                       : Generates "average" sequence of aligned sequence from most common base at each position. If first sequence in alignment is reference, reference is not included in calculation of average sequence.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl get_average_sequence_of_aligned_sequences.pl [alignment fasta file path] [1 if alignment includes reference as first sequence, 0 if alignment does not include a reference] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      BLAST (
                      <a href="/lakras/bio-helper-scripts/blob/main/blast">
                       <code>
                        blast
                       </code>
                      </a>
                      )
                     </h2>
                     <a aria-label="Permalink: BLAST (blast)" class="anchor" href="#blast-blast" id="user-content-blast-blast">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     All blast runs must use output format
                     <code>
                      -outfmt "6 qseqid sacc stitle staxids sscinames sskingdoms qlen slen length pident qcovs evalue"
                     </code>
                     . (Otherwise, hardcoded column numbers can be modified within each script.)
                    </p>
                    <p dir="auto">
                     Instructions for running blast in a Google Cloud Virtual Machine are in
                     <a href="/lakras/bio-helper-scripts/blob/main/blast/README.md">
                      <code>
                       /blast/README.md
                      </code>
                     </a>
                     .
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/count_sequences_in_blast_output.pl">
                        <code>
                         count_sequences_in_blast_output.pl
                        </code>
                       </a>
                       : Counts the number of sequences with hits in the blast output.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl count_sequences_in_blast_output.pl [blast output table] [optional minimum percent id] [optional minimum query coverage] [1 to print sequence names, 0 to print number sequences only]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_top_blast_hit_for_each_sequence.pl">
                        <code>
                         retrieve_top_blast_hit_for_each_sequence.pl
                        </code>
                       </a>
                       : Retrieves top hit for each sequence (assumes they are in order in blast output).
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_top_blast_hit_for_each_sequence.pl [blast output] &gt; [output subset of blast output table]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_top_blast_hits_for_each_sequence.pl">
                        <code>
                         retrieve_top_blast_hits_for_each_sequence.pl
                        </code>
                       </a>
                       : Retrieves top hit for each sequence (assumes they are in order in blast output). Prints all top hits with same e-values.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_top_blast_hits_for_each_sequence.pl [blast output] [number by which to multiply sequence's top evalue, to use as threshold for inclusion in top hits] [1 to treat blast output as modified DIAMOND output] &gt; [output subset of blast output table]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/filter_blast_hits.pl">
                        <code>
                         filter_blast_hits.pl
                        </code>
                       </a>
                       : Filters blast hits: prints blast hits with at least minimum percent identity and at least minimum percent query coverage. Can also further filter with optional maximum percent identity, maximum percent query coverage provided, or minimum length of matched sequence.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl filter_blast_hits.pl [blast output] [minimum percent identity] [minimum percent query coverage] [optional maximum percent identity] [optional maximum percent query coverage] [optional minimum length of matched sequence] &gt; [output subset of blast output table]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_blast_outputs_for_sequences_of_interest.pl">
                        <code>
                         retrieve_blast_outputs_for_sequences_of_interest.pl
                        </code>
                       </a>
                       : Retrieves blast hits for sequences of interest from blast output file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_blast_outputs_for_sequences_of_interest.pl [blast output file] [sequence name to retrieve blast hits for] [another sequence name to retrieve blast hits for] [etc.] &gt; [subset blast output path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_sequences_with_no_blast_hits.pl">
                        <code>
                         retrieve_sequences_with_no_blast_hits.pl
                        </code>
                       </a>
                       : Retrieves sequences that do not have blast results.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_sequences_with_no_blast_hits.pl [blast output file] [fasta file that was input to blast] [minimum percent identity for a blast hit to be counted] [minimum query coverage for a blast hit to be counted] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_sequences_with_blast_hits.pl">
                        <code>
                         retrieve_sequences_with_blast_hits.pl
                        </code>
                       </a>
                       : Retrieves sequences that have blast results.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_sequences_with_blast_hits.pl [blast output file] [fasta file that was input to blast] [minimum percent identity for a blast hit to be counted] [minimum query coverage for a blast hit to be counted] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/slice_blast_output_file_before_or_after_sequence_name_query.pl">
                        <code>
                         slice_blast_output_file_before_or_after_sequence_name_query.pl
                        </code>
                       </a>
                       : Prints either all lines before or all lines after sequence name appears in blast output file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl slice_blast_output_file_before_or_after_sequence_name_query.pl [blast output] [sequence name to slice before or after] [1 to print all lines BEFORE sequence name, 0 to print all lines AFTER sequence name] [1 to print lines including sequence name, 0 not to] &gt; [output subset of blast output table]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/extract_hits_or_sequences_with_top_hit_in_taxon.pl">
                        <code>
                         extract_hits_or_sequences_with_top_hit_in_taxon.pl
                        </code>
                       </a>
                       : Retrieves blast hits or fasta sequences of sequences with top hit or any hit at all in taxon of interest or its children.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl extract_hits_or_sequences_with_top_hit_in_taxon.pl [blast output table] [fasta file that was input to blast] [nodes.dmp file from NCBI] [taxon id of taxon of interest] [1 to print fasta sequences, 0 to print subset of blast output] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/extract_hits_or_sequences_with_no_or_poor_blast_hits_in_large_database_compared_to_in_taxon_specific_database.pl">
                        <code>
                         extract_hits_or_sequences_with_no_or_poor_blast_hits_in_large_database_compared_to_in_taxon_specific_database.pl
                        </code>
                       </a>
                       : Retrieves blast hits or fasta sequences of sequences that produced a blast hit from the taxon of interest in a blast search within a database consisting only of sequences from the taxon of interest and produced either no hits at all in blast search within large database blast search or only hits that were comparable to or worse than hits from the blast search in the taxon-specific database.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl extract_hits_or_sequences_with_no_or_poor_blast_hits_in_large_database_compared_to_in_taxon_specific_database.pl [blast output table from taxon-specific database blast search] [blast output table from large database blast search] [fasta file that was input to blast for large database blast search] [nodes.dmp file from NCBI] [taxon id of taxon of interest] [1 to print fasta sequences, 0 to print subset of blast output] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_top_blast_hits_LCA_for_each_sequence.pl">
                        <code>
                         retrieve_top_blast_hits_LCA_for_each_sequence.pl
                        </code>
                       </a>
                       : For each sequence, extracts all top hits with same e-values (assumes they are in order in blast output). Prints lowest common ancestor (LCA) of top hits for each sequence.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_top_blast_hits_LCA_for_each_sequence.pl [blast output] [nodes.dmp file from NCBI] [number by which to multiply sequence's top evalue, to use as threshold for inclusion in LCA] [1 to print all matched accession numbers in a final column] [1 to treat blast output as modified DIAMOND output] &gt; [output table]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/generate_LCA_table_for_sequences_with_no_matches.pl">
                        <code>
                         generate_LCA_table_for_sequences_with_no_matches.pl
                        </code>
                       </a>
                       : Generates an LCA matches table (matching output of retrieve_top_blast_hits_LCA_for_each_sequence.pl) for sequences with no blast hits.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl generate_LCA_table_for_sequences_with_no_matches.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [fasta file that was input to blast search] &gt; [output table]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/compare_LCA_matches_to_ground_truth_LCA_matches.pl">
                        <code>
                         compare_LCA_matches_to_ground_truth_LCA_matches.pl
                        </code>
                       </a>
                       : Compares LCA match taxon from two blast outputs for each match. Uses output of retrieve_top_blast_hits_LCA_for_each_sequence.pl as input.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl compare_LCA_matches_to_ground_truth_LCA_matches.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search, to be treated as ground truth] [output of another retrieve_top_blast_hits_LCA_for_each_sequence.pl for another blast search, to compare to ground truth] [nodes.dmp file from NCBI] &gt; [output table]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_taxonids_from_accession_numbers_of_blast_matches.pl">
                        <code>
                         retrieve_taxonids_from_accession_numbers_of_blast_matches.pl
                        </code>
                       </a>
                       : Retrieves each match's taxon id from Entrez using match accession number column and adds taxon ids to blast or diamond output as a new column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_taxonids_from_accession_numbers_of_blast_matches.pl [blast or diamond output] [1 if blast or diamond output is from a nucleotide search; 0 if it is from a protein search] [column number of new taxon id column to add to output file (0-indexed)] [column number (0-indexed) of column containing match accession numbers (stitle)] [optional column number (0-indexed) of column containing match names (stitle)] &gt; [blast or diamond output with taxon id column added]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_taxonids_from_accession_numbers_of_blast_matches_bulk.pl">
                        <code>
                         retrieve_taxonids_from_accession_numbers_of_blast_matches_bulk.pl
                        </code>
                       </a>
                       : Maps accession numbers to taxon ids directly from NCBI accession number to taxon id mapping file, without using Entrez. Useful for mapping thousands or millions of accession numbers, since this method does not have rate limits. Retrieves each match's taxon id from from match accession number column and adds taxon ids to blast or diamond output as a new column. Reads in mapping tables in parallel to save time. Appropriate mapping tables must be downloaded and unzipped from ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/ (see README file).
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_taxonids_from_accession_numbers_of_blast_matches_bulk.pl [blast or diamond output] [column number of new taxon id column to add to output file (0-indexed)] [column number (0-indexed) of column containing match accession numbers (stitle)] [unzipped mapping table] [another unzipped mapping table] [etc.] &gt; [blast or diamond output with taxon id column added]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_taxonids_from_accession_numbers.pl">
                        <code>
                         retrieve_taxonids_from_accession_numbers.pl
                        </code>
                       </a>
                       : Given a list of accession numbers, one per line, retrieves corresponding taxon ids. Outputs tab-separating map with sequence name and taxon id.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_taxonids_from_accession_numbers.pl [path of file with list of accession numbers, one per line] [database (nuccore by default)] &gt; [output mapping table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_taxon_names_from_taxonids.pl">
                        <code>
                         retrieve_taxon_names_from_taxonids.pl
                        </code>
                       </a>
                       : Retrieves each match's taxon name from its taxon id and adds it to blast or diamond output as a new column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_taxon_names_from_taxonids.pl [blast or diamond output] [names.dmp file from NCBI] &gt; [blast or diamond output with taxon name column added]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_most_common_taxonids_in_LCA_output.pl">
                        <code>
                         retrieve_most_common_taxonids_in_LCA_output.pl
                        </code>
                       </a>
                       : Retrieve NN most frequently matched species, genera, or families.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_most_common_taxonids_in_LCA_output.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [species, genus, or family] [number most frequent matched species, genera, or families to output] [minimum number reads matched by a taxon to report it] &gt; [output list of taxon ids, one per line]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_all_taxonids_in_LCA_output.pl">
                        <code>
                         retrieve_all_taxonids_in_LCA_output.pl
                        </code>
                       </a>
                       : Retrieve all matched species, genera, or families.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_all_taxonids_in_LCA_output.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [species, genus, or family] &gt; [output list of taxon ids, one per line]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_refseq_genome_accession_numbers_from_taxon_ids.pl">
                        <code>
                         retrieve_refseq_genome_accession_numbers_from_taxon_ids.pl
                        </code>
                       </a>
                       : Given a list of taxon ids, one per line, retrieves accession numbers of refseq genomes. Outputs list of refseq genome accession numbers, one per line.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_refseq_genome_accession_numbers_from_taxon_ids.pl [path of file with list of accession numbers, one per line] &gt; [output list file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/retrieve_most_common_accession_numbers_from_most_common_taxonids_in_LCA_output.pl">
                        <code>
                         retrieve_most_common_accession_numbers_from_most_common_taxonids_in_LCA_output.pl
                        </code>
                       </a>
                       : Retrieve most frequent NN accession numbers matched from most frequent NN matched species, genera, or families.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_most_common_accession_numbers_from_most_common_taxonids_in_LCA_output.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [species, genus, or family] [number most frequent matched species, genera, or families to examine] [number accession numbers most frequent matched by descendants of most frequent species, genera, or families] &gt; [output list of accession numbers, one per line]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/filter_LCA_matches.pl">
                        <code>
                         filter_LCA_matches.pl
                        </code>
                       </a>
                       : Filters LCA matches by rank (at least species, genus, or family), mean percent identity, and/or mean percent query coverage. Uses output of retrieve_top_blast_hits_LCA_for_each_sequence.pl as input.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl filter_LCA_matches.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [1 to requires output matches to be classified to at least species level] [1 to requires output matches to be classified to at least genus level] [1 to requires output matches to be classified to at least family level] [minimum mean percent identity] [maximum mean percent identity]  [minimum mean percent query coverage] [maximum mean percent query coverage]   &gt; [output filtered LCA matches table]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/filter_taxonids_to_descendants_of_target_taxon.pl">
                        <code>
                         filter_taxonids_to_descendants_of_target_taxon.pl
                        </code>
                       </a>
                       : Given a list of txaon ids, one per line, filters down to only descendants of parameter taxon id (for example, 10239 for Viruses).
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl filter_taxonids_to_descendants_of_target_taxon.pl [path of file with list of taxon ids, one per line] [nodes.dmp file from NCBI] [taxon id to filter to] &gt; [output list of taxon ids]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/filter_taxonids_in_table_column_to_descendants_of_target_taxon.pl">
                        <code>
                         filter_taxonids_in_table_column_to_descendants_of_target_taxon.pl
                        </code>
                       </a>
                       : Given a table with a column of taxon ids, filters down to only descendants of parameter taxon id (for example, 10239 for Viruses).
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl filter_taxonids_in_table_column_to_descendants_of_target_taxon.pl [path of table with taxon id] [number (0-indexed) of column with taxon ids] [nodes.dmp file from NCBI] [taxon id to filter to] &gt; [output table]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/add_column_with_superkingdom_of_taxon_id.pl">
                        <code>
                         add_column_with_superkingdom_of_taxon_id.pl
                        </code>
                       </a>
                       : Reads in column containing taxon id and adds column containing the superkingdom of that taxon id.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_column_with_superkingdom_of_taxon_id.pl [table] [title of column containing taxon ids] [nodes.dmp file from NCBI] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/kraken_summary_report_to_species_table.pl">
                        <code>
                         kraken_summary_report_to_species_table.pl
                        </code>
                       </a>
                       : Converts Kraken summary report to table of species found in sample.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl kraken_summary_report_to_species_table.pl [kraken summary report] &gt; [output table file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/blast/LCA_table_to_kraken_output_format.pl">
                        <code>
                         LCA_table_to_kraken_output_format.pl
                        </code>
                       </a>
                       : Converts LCA output table to kraken output format. Treats paired reads as separate reads.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl LCA_table_to_kraken_output_format.pl [output of retrieve_top_blast_hits_LCA_for_each_sequence.pl for one blast search] [fasta file that was input to blast search (to retrieve sequence lengths and names of unclassified sequences)] &gt; [output kraken format table]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      VCF file processing (
                      <a href="/lakras/bio-helper-scripts/blob/main/vcf-files">
                       <code>
                        vcf-files
                       </code>
                      </a>
                      )
                     </h2>
                     <a aria-label="Permalink: VCF file processing (vcf-files)" class="anchor" href="#vcf-file-processing-vcf-files" id="user-content-vcf-file-processing-vcf-files">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Input VCF files must be in format produced by
                     <a href="https://csb5.github.io/lofreq/commands/#call" rel="nofollow">
                      <code>
                       LoFreq call
                      </code>
                     </a>
                     .
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/vcf-files/vcf_file_to_heterozygosity_table.pl">
                        <code>
                         vcf_file_to_heterozygosity_table.pl
                        </code>
                       </a>
                       : Reads in vcf file output produced by
                       <a href="https://csb5.github.io/lofreq/commands/#call" rel="nofollow">
                        <code>
                         LoFreq call
                        </code>
                       </a>
                       and prints human-readable heterozygosity table. Optionally filters output by read depth, minor allele readcount, and minor allele frequency. See code for output table format and hardcoded filtering thresholds.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl vcf_file_to_heterozygosity_table.pl [vcf file output by LoFreq] [1 to filter output, 0 to not filter] [optional 1 to not print reference column in output, to save space] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/vcf-files/detect_potential_transmission_events_from_iSNVs.pl">
                        <code>
                         detect_potential_transmission_events_from_iSNVs.pl
                        </code>
                       </a>
                       : Detects potential transmission events from iSNVs, assuming an average transmission bottleneck of one virus per transmission. If collection dates are provided, only compares sequences within hardcoded maximum collection date distance. See script for input file details and options.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl detect_potential_transmission_events_from_iSNVs.pl [consensus sequences alignment fasta file path] [list of heterozygosity tables] [optional list of read depth tables] [optional tab-separated table mapping sample names to collection dates (YYYY-MM-DD)] [0 to print one line per sample pair, 1 to print one line per iSNV] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/vcf-files/mask_positions_in_heterozygosity_table.pl">
                        <code>
                         mask_positions_in_heterozygosity_table.pl
                        </code>
                       </a>
                       : Masks (removes lines corresponding to) alleles at indicated positions. See script for input file format. Positions must be relative to reference used in heterozygosity table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl mask_positions_in_heterozygosity_table.pl [heterozygosity table] [first position in region to mask] [last position in region to mask] [first position in another region to mask] [last position in another region to mask] [etc.] &gt; [output heterozygosity table file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/vcf-files/mask_positions_in_heterozygosity_table_in_bulk.pl">
                        <code>
                         mask_positions_in_heterozygosity_table_in_bulk.pl
                        </code>
                       </a>
                       : Masks (removes lines corresponding to) alleles at indicated positions. Operates separately on multiple files as described in input table. See script for input file format. Positions must be relative to reference used in heterozygosity table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl mask_positions_in_heterozygosity_table_in_bulk.pl [tab-separated table containing filepath of heterozygosity table in first column, first and last positions of regions to mask in this heterozygosity table, space separated, in second column] [optional first position in additional region to mask in all heterozygosity tables] [optional last position in additional region to mask in all heterozygosity tables] [optional first position in another additional region to mask in all heterozygosity tables] [optional last position in another additional region to mask in all heterozygosity tables] [etc.]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/vcf-files/filter_heterozygosity_table.pl">
                        <code>
                         filter_heterozygosity_table.pl
                        </code>
                       </a>
                       : Filters input heterozygosity table. See script for input file format.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl filter_heterozygosity_table.pl [heterozygosity table] [minimum minor allele readcount] [minimum minor allele frequency] [minimum read depth] &gt; [output filtered heterozygosity table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/vcf-files/rereference_positions_in_heterozygosity_table.pl">
                        <code>
                         rereference_positions_in_heterozygosity_table.pl
                        </code>
                       </a>
                       : Uses alignment of consensus genome to reference to update heterozygosity table positions with respect to self to positions with respect to the reference genome in the alignment.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl convert_positions_in_heterozygosity_table_from_relative_to_self_to_relative_to_reference_in_alignment.pl [consensus sequence aligned to reference fasta file] [list of heterozygosity tables, one per line]# &gt; [output heterozygosity table]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/vcf-files/rereference_positions_in_read_depths_table.pl">
                        <code>
                         rereference_positions_in_read_depths_table.pl
                        </code>
                       </a>
                       : Uses alignment of consensus genome to reference to update read depths table positions with respect to self to positions with respect to the reference genome in the alignment.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl rereference_positions_in_read_depths_table.pl [consensus sequence aligned to reference fasta file] [list of read_depths tables, one per line]# &gt; [output read_depths table]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/lineages/annotate_heterozygosity_tables_with_estimated_lineages.pl">
                        <code>
                         lineages/annotate_heterozygosity_tables_with_estimated_lineages.pl
                        </code>
                       </a>
                       : Annotates heterozygosity tables with lineage consistent with minor and consensus-level alleles at lineage-defining positions. Output is printed to one file per heterozygosity table or as one table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl annotate_heterozygosity_tables_with_estimated_lineages.pl [lineage genomes aligned to reference] [list of heterozygosity tables] [1 to print each heterozygosity table separately, 0 to print all tables to console] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      BED file processing (
                      <a href="/lakras/bio-helper-scripts/blob/main/bed-files">
                       <code>
                        bed-files
                       </code>
                      </a>
                      )
                     </h2>
                     <a aria-label="Permalink: BED file processing (bed-files)" class="anchor" href="#bed-file-processing-bed-files" id="user-content-bed-file-processing-bed-files">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Read a description of the BED file format
                     <a href="https://en.wikipedia.org/wiki/BED_(file_format)" rel="nofollow">
                      on Wikipedia
                     </a>
                     .
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/bed-files/pull_out_sequences_described_in_bed_file.pl">
                        <code>
                         pull_out_sequences_described_in_bed_file.pl
                        </code>
                       </a>
                       : Pulls out subsets of sequences by name and position within that sequence from bed file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl pull_out_sequences_described_in_bed_file.pl [fasta file path] [bed file path] [0 to output fasta sequence, 1 to add sequence as new column in bed file] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/bed-files/retrieve_amplicon_regions_from_primer_regions.pl">
                        <code>
                         retrieve_amplicon_regions_from_primer_regions.pl
                        </code>
                       </a>
                       : Reads in positions of primers and outputs positions of amplicons between those primers. Uses primer names to match left and right primers. Primer names must include the primer number and _LEFT or _RIGHT, for example nCoV-2019_1_LEFT or nCoV-2019_26_RIGHT. If more multiple start primers or multiple end primers are provided for an amplicon, the amplicon is set to be the largest possible with all its provided primers.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_amplicon_regions_from_primer_regions.pl [primers bed file path] &gt; [amplicons bed file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/bed-files/modify_start_end_position_indexing_and_inclusivity.pl">
                        <code>
                         modify_start_end_position_indexing_and_inclusivity.pl
                        </code>
                       </a>
                       : Modifies start and end positions in bed file from 0-indexed or 1-indexed to 0-indexed or 1-indexed and from inclusive end or non-iclusive end to inclusive end or non-iclusive end.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl modify_start_end_position_indexing_and_inclusivity.pl [bed file path] [1 if input is 1-indexed, 0 if input is 0-indexed] [1 if input has inclusive end, 0 if input has non-inclusive end] [1 if output should be 1-indexed, 0 if output should be 0-indexed] [1 if output should have inclusive end, 0 if output should have non-inclusive end] &gt; [output bed file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/bed-files/remove_bed_file_rows_overlapping_positions.pl">
                        <code>
                         remove_bed_file_rows_overlapping_positions.pl
                        </code>
                       </a>
                       : Removes rows in bed file that overlap with parameter positions.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl remove_bed_file_rows_overlapping_positions.pl [bed file path] [position (0-indexed); any rows overlapping this position will be removed] [another position] [another position] [etc.] &gt; [output bed file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/bed-files/determine_regions_not_covered_by_amplicons.pl">
                        <code>
                         determine_regions_not_covered_by_amplicons.pl
                        </code>
                       </a>
                       : Takes in a bed file of amplicons and outputs a bed file of positions not covered by those amplicons. Assumes all amplicons come from only one reference sequence.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl determine_regions_not_covered_by_amplicons.pl [amplicons bed file path] &gt; [output bed file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/bed-files/calculate_proportions_bed_file_covered_by_other_bed_files.pl">
                        <code>
                         calculate_proportions_bed_file_covered_by_other_bed_files.pl
                        </code>
                       </a>
                       : Takes in one bed file and at least one other bed file to compare it to. Outputs proportion of the sequence in the regions described in the first bed file that is captured in the regions described in each additional bed file. Outputs each bed file path and the proportion it covers of the first bed file. Assumes all rows come from only one reference sequence.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl calculate_proportions_bed_file_covered_by_other_bed_files.pl [bed file path] [other bed file path] [another bed file path] [and another] [etc.] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/bed-files/annotate_positions_with_regions_they_overlap.pl">
                        <code>
                         annotate_positions_with_regions_they_overlap.pl
                        </code>
                       </a>
                       : Annotates positions in input table with the names of the region(s) they are in. Adds column containing names of regions the position overlaps. If multiple regions overlap a position they are comma-separated in the new column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl annotate_positions_with_regions_they_overlap.pl [table containing positions to annotate] "[title of column containing positions to annotate]" [table containing start and end positions and names of regions] [optional output column title (overlapping_regions by default)] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/bed-files/retrieve_amplicon_regions_from_primer_regions_and_count_number_amplicons_overlapping_each_position.pl">
                        <code>
                         retrieve_amplicon_regions_from_primer_regions_and_count_number_amplicons_overlapping_each_position.pl
                        </code>
                       </a>
                       : Reads in positions of primers and determines positions of amplicons between those primers. Outputs total number amplicons overlapping each position.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_amplicon_regions_from_primer_regions_and_count_number_amplicons_overlapping_each_position.pl [primers bed file path] &gt; [amplicons bed file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/bed-files/extend_primers_to_include_amplicon_overlaps.pl">
                        <code>
                         extend_primers_to_include_amplicon_overlaps.pl
                        </code>
                       </a>
                       : Extends primers to include positions that are in more than one amplicon. Amplicons in the resulting bed file do not overlap. Outputs new primer bed file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl extend_primers_to_include_amplicon_overlaps.pl [primers bed file path] &gt; [new primers bed file path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Read depth file processing (
                      <a href="/lakras/bio-helper-scripts/blob/main/read-depths">
                       <code>
                        read-depths
                       </code>
                      </a>
                      )
                     </h2>
                     <a aria-label="Permalink: Read depth file processing (read-depths)" class="anchor" href="#read-depth-file-processing-read-depths" id="user-content-read-depth-file-processing-read-depths">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Input read depth tables must be in format produced by
                     <a href="http://www.htslib.org/doc/samtools-depth.html" rel="nofollow">
                      <code>
                       samtools depth
                      </code>
                     </a>
                     : tab-separated reference name, position, read depth; no header line.
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/read-depths/summarize_read_depths_across_files.pl">
                        <code>
                         summarize_read_depths_across_files.pl
                        </code>
                       </a>
                       : Summarizes read depths across read depth tables. Outputs table with reference name; position; and mean, standard deviation, median, min, max, and range of read depths at each position, and number of read depth values at each position that are 0.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl summarize_read_depths_across_files.pl [read depth table] [another read depth table] [another read depth table] [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/read-depths/retrieve_position_ranges_with_threshold_read_depth.pl">
                        <code>
                         retrieve_position_ranges_with_threshold_read_depth.pl
                        </code>
                       </a>
                       : Prints ranges of positions with read depths at or above minimum read depth.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_position_ranges_with_threshold_read_depth.pl [minimum read depth] [read depth table] [optional 1 to print output as one line] [optional 1 to print position ranges NOT passing read depth threshold, rather than positions that do] &gt; [output path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/aligned-fasta/mask_low_read_depth_alleles.pl">
                        <code>
                         aligned-fasta/mask_low_read_depth_alleles.pl
                        </code>
                       </a>
                       : Masks (replaces with Ns) alleles with low read depths.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl mask_low_read_depth_alleles.pl [alignment fasta file path] [list of read depth tables] [minimum read depth] &gt; [output fasta file path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Tables (
                      <a href="/lakras/bio-helper-scripts/blob/main/tables">
                       <code>
                        tables
                       </code>
                      </a>
                      )
                     </h2>
                     <a aria-label="Permalink: Tables (tables)" class="anchor" href="#tables-tables" id="user-content-tables-tables">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tables are assumed to be tab-separated (usually denoted
                     <code>
                      .tsv
                     </code>
                     or
                     <code>
                      .txt
                     </code>
                     . To use with comma-separated
                     <code>
                      .csv
                     </code>
                     files, change the value of the
                     <code>
                      $DELIMITER
                     </code>
                     variable from
                     <code>
                      "\t"
                     </code>
                     to
                     <code>
                      ","
                     </code>
                     or convert from comma-separated csv tables to tab-separated tsv tables using
                     <a href="/lakras/bio-helper-scripts/blob/main/tables/csv_to_tsv.pl">
                      <code>
                       csv_to_tsv.pl
                      </code>
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Table Format
                     </h3>
                     <a aria-label="Permalink: Table Format" class="anchor" href="#table-format" id="user-content-table-format">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/csv_to_tsv.pl">
                        <code>
                         csv_to_tsv.pl
                        </code>
                       </a>
                       : Converts comma-separated csv table to tab-separated tsv table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl csv_to_tsv.pl [table] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/tsv_to_csv.pl">
                        <code>
                         tsv_to_csv.pl
                        </code>
                       </a>
                       : Converts tab-separated tsv table to comma-separated csv table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl tsv_to_csv.pl [table] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/make_r_friendly_table.pl">
                        <code>
                         make_r_friendly_table.pl
                        </code>
                       </a>
                       : Converts table to R-friendly format. See script for example inputs and outputs.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl make_r_friendly_table.pl [table file path] [first data column index] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/convert_variable_value_lists_to_table.pl">
                        <code>
                         convert_variable_value_lists_to_table.pl
                        </code>
                       </a>
                       : Converts lists of variable values, such as those output by Terra, to a table. Lists of values must appear with variable on one line followed by values in the following line, comma-separated and surrounded by brackets. Lines not matching this format are ignored. Variable names must be unique (cannot repeat). List items must be in order for all variables. Expects all variables to have identical numbers of values: ignores variables with less than the largest number of values. See code for example of input and output formats.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl convert_variable_value_lists_to_table.pl [file containing input lists] [optional list item separator, ", " by default] [optional list start, "[" by default] [optional list end, "]" by default] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Column title manipulation
                     </h3>
                     <a aria-label="Permalink: Column title manipulation" class="anchor" href="#column-title-manipulation" id="user-content-column-title-manipulation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/replace_all_spaces_parens_in_column_titles.pl">
                        <code>
                         replace_all_spaces_parens_in_column_titles.pl
                        </code>
                       </a>
                       : Replaces all spaces and parentheses in header line with provided replacement value, or underscore by default.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl replace_all_spaces_parens_in_column_titles.pl [table] [optional value to replace spaces with in header line] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/replace_column_title.pl">
                        <code>
                         replace_column_title.pl
                        </code>
                       </a>
                       : Replaces column title with new column title.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl replace_column_title.pl [table] "[current title of column to replace]" "[replacement column title]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Column manipulation
                     </h3>
                     <a aria-label="Permalink: Column manipulation" class="anchor" href="#column-manipulation" id="user-content-column-manipulation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/delete_columns.pl">
                        <code>
                         delete_columns.pl
                        </code>
                       </a>
                       : Deletes columns with specified column titles.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl delete_columns.pl [table] "[title of column to delete]" "[title of another column to delete]" "[title of another column to delete]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/replace_whole_column_with_empty_values.pl">
                        <code>
                         replace_whole_column_with_empty_values.pl
                        </code>
                       </a>
                       : Replaces whole column (including title unless hardcoded option selected) with empty values.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl replace_whole_column_with_empty_values.pl [table] "[column title]" "[another column title]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/duplicate_columns.pl">
                        <code>
                         duplicate_columns.pl
                        </code>
                       </a>
                       : Duplicates selected columns.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl duplicate_columns.pl [tab-separated table] "[column title]" "[another column title]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_one_value_column.pl">
                        <code>
                         add_one_value_column.pl
                        </code>
                       </a>
                       : Adds column with specified title and specified value for all values.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_one_value_column.pl [table to add column to] "[title of column to add]" "[value of column to add]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_column_for_selected_rows.pl">
                        <code>
                         add_column_for_selected_rows.pl
                        </code>
                       </a>
                       : Creates new column with values in selected rows.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_column_for_selected_rows.pl [table to add new column to] "[title of column in table to identify rows by]" [file with list of rows to select, one per line] "[optional title of new column to add]" "[optional value to add to selected rows in new column]" "[optional value to add to all other rows in new column]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/concatenate_columns.pl">
                        <code>
                         concatenate_columns.pl
                        </code>
                       </a>
                       : Concatenates values in selected columns. Adds concatenated values in new column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl concatenate_columns.pl [tab-separated table] "[column title]" "[another column title]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/summarize_numerical_columns.pl">
                        <code>
                         summarize_numerical_columns.pl
                        </code>
                       </a>
                       : Summarizes selected numerical columns. Adds new columns with: mean, standard deviation, median, min, max, range, and all values sorted in a comma-separated list. If a column value contains a comma-separated list, includes all values from list.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl summarize_numerical_columns.pl [tab-separated table] "[column title]" "[another column title]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/summarize_numerical_column_holding_other_columns_constant.pl">
                        <code>
                         summarize_numerical_column_holding_other_columns_constant.pl
                        </code>
                       </a>
                       : Summarizes values in selected numerical column while holding other columns constant. Outputs table with constant columns and new columns with statistics summarizing selected numerical column: mean, standard deviation, median, number values, min, max, range, and all values sorted in a comma-separated list.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl summarize_numerical_column_holding_other_columns_constant.pl [tab-separated table] "[title of numerical column to summarize]" "[title of column to hold constant]" "[title of another column to hold constant]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/combine_columns.pl">
                        <code>
                         combine_columns.pl
                        </code>
                       </a>
                       : Combines each selected pair of columns into one column with all values. Does not merge values. See script for example inputs and outputs.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl combine_columns.pl [input table] [list of tab-separated groups of titles of columns to combine, one group per line] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/merge_columns.pl">
                        <code>
                         merge_columns.pl
                        </code>
                       </a>
                       : Merges selected columns. Reports any conflicts.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl merge_columns.pl [table to merge] "[title of column to merge]" "[title of another column to merge]" "[title of another column to merge]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/merge_columns_with_conflict_detail_column.pl">
                        <code>
                         merge_columns_with_conflict_detail_column.pl
                        </code>
                       </a>
                       : Merges selected columns. Reports any conflicts.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl merge_columns_with_conflict_detail_column.pl [table to merge] "[title of column to print in merge conflict message]" "[title of column to merge]" "[title of another column to merge]" "[title of another column to merge]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_column_with_number_occurrences_of_column_values.pl">
                        <code>
                         add_column_with_number_occurrences_of_column_values.pl
                        </code>
                       </a>
                       : Adds column indicating number occurrences of that row's value(s) in the entirety of the parameter column(s).
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_column_with_number_occurrences_of_column_values.pl [tab-separated table] "[column title]" "[optional additional column title]" "[optional additional column title]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/replace_values_in_columns.pl">
                        <code>
                         replace_values_in_columns.pl
                        </code>
                       </a>
                       : Replaces query with replacement text in specified columns.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl replace_values_in_columns.pl [table] "[query text to replace]" "[replacement text]" [1 to exactly match full column value only, 0 to allow search and replace within text of column value] "[title of column to search]" "[title of another column to search]" "[title of another column to search]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/delete_values_in_columns.pl">
                        <code>
                         delete_values_in_columns.pl
                        </code>
                       </a>
                       : Deletes query in specified columns.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl delete_values_in_columns.pl [table] "[query text to delete]" # [1 to exactly match full column value only, 0 to allow search within text of column value] "[title of column to search]" "[title of another column to search]" "[title of another column to search]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/replace_nonempty_values_in_columns.pl">
                        <code>
                         replace_nonempty_values_in_columns.pl
                        </code>
                       </a>
                       : Replaces non-empty values with replacement text in specified columns.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl replace_nonempty_values_in_columns.pl [table] "[replacement text]" "[title of column to search]" "[title of another column to search]" "[title of another column to search]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/replace_values_with_coded_values.pl">
                        <code>
                         replace_values_with_coded_values.pl
                        </code>
                       </a>
                       : Replaces non-empty values with coded values, e.g., Value 1 (for the most common value), Value 2 (for the second-most common value), Value 3, etc. Ties are broken alphabetically.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl replace_values_with_coded_values.pl [table] "[title of column to search]" "[optional code prefix]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/bulk_replace_in_one_column.pl">
                        <code>
                         bulk_replace_in_one_column.pl
                        </code>
                       </a>
                       : Replaces all occurrences of values in selected column to mapped replacement values.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl bulk_replace_in_one_column.pl [tab-separated file mapping current values (first column) to new values (second column)] [path of table in which to replace values] [title of column to replace values in] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_to_start_and_end_of_values_in_columns.pl">
                        <code>
                         add_to_start_and_end_of_values_in_columns.pl
                        </code>
                       </a>
                       : Pads non-empty values in specified columns with parameter start and end text.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_to_start_and_end_of_values_in_columns.pl [table] "[text to add to start of each column value]" "[text to add to end of each column value]" "[title of column to search]" "[title of another column to search]" "[title of another column to search]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_to_start_and_end_of_values_containing_query_in_columns.pl">
                        <code>
                         add_to_start_and_end_of_values_containing_query_in_columns.pl
                        </code>
                       </a>
                       : Pads values containing query in specified columns with parameter start and end text.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_to_start_and_end_of_values_containing_query_in_columns.pl [table] "[query text]" "[text to add to start of each column value]" "[text to add to end of each column value]" "[title of column to search]" "[title of another column to search]" "[title of another column to search]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/change_capitalization_in_columns.pl">
                        <code>
                         change_capitalization_in_columns.pl
                        </code>
                       </a>
                       : Changes capitalization of values in specified columns: all capitalized, all lowercase, or first letter capitalized.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl change_capitalization_in_columns.pl [table] [uc to make all values uppercase, lc to make all values lowercase, first to capitalize first letter] "[title of column to capitalize]" "[title of another column to capitalize]" "[title of another column to capitalize]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/fill_in_empty_column_values.pl">
                        <code>
                         fill_in_empty_column_values.pl
                        </code>
                       </a>
                       : Fills in empty values in column of interest with specified value.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl fill_in_empty_column_values.pl [table] "[title of column to fill in]" "[value to replace empty values with]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/fill_in_empty_column_values_with_increasing_numerical_index.pl">
                        <code>
                         fill_in_empty_column_values_with_increasing_numerical_index.pl
                        </code>
                       </a>
                       : Fills in empty values in column of interest with an increasing numerical index value, so that every empty cell contains a unique numerical value.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl fill_in_empty_column_values_with_increasing_numerical_index.pl [table] "[title of column to fill in]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/fill_in_empty_column_values_from_other_column.pl">
                        <code>
                         fill_in_empty_column_values_from_other_column.pl
                        </code>
                       </a>
                       : Fills in empty values in column of interest with values from other column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl fill_in_empty_column_values_from_other_column.pl [table] "[title of column to fill in]" "[title of column with potential replacement values]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/replace_column_values_with_other_column_where_present.pl">
                        <code>
                         replace_column_values_with_other_column_where_present.pl
                        </code>
                       </a>
                       : Fills in values in column of interest with values from other column when they are present.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl replace_column_values_with_other_column_where_present.pl [table] "[title of column to fill in]" "[title of column with potential replacement values]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/replace_column_values_where_other_column_present_and_nonzero.pl">
                        <code>
                         replace_column_values_where_other_column_present_and_nonzero.pl
                        </code>
                       </a>
                       : In rows where a column has a present, non-zero value, replaces value in another column with parameter replacement value.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl replace_column_values_where_other_column_present_and_nonzero.pl [table] "[title of column to check]" "[title of column to fill in]" "[replacement value]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/remove_nonunique_values_in_list_in_column.pl">
                        <code>
                         remove_nonunique_values_in_list_in_column.pl
                        </code>
                       </a>
                       : Removes non-unique values in comma-separated lists in specified columns.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl remove_nonunique_values_in_list_in_column.pl [table] "[column title]" "[another column title]" "[another column title]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/retrieve_subset_of_columns.pl">
                        <code>
                         retrieve_subset_of_columns.pl
                        </code>
                       </a>
                       : Subsets table to only columns of interest.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_subset_of_columns.pl [table] "[title of first column to include in output]" "[title of second column to include]" "[title of third column to include]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/retrieve_subset_of_columns_by_column_number.pl">
                        <code>
                         retrieve_subset_of_columns_by_column_number.pl
                        </code>
                       </a>
                       : Subsets table to only columns of interest.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_subset_of_columns_by_column_number.pl [table] [column number of first column to include in output] [column number of second column to include] [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/compile_values_and_titles_in_selected_columns.pl">
                        <code>
                         compile_values_and_titles_in_selected_columns.pl
                        </code>
                       </a>
                       : Generates a new column with the values in selected columns and their column titles, where values are present.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl compile_values_and_titles_in_selected_columns.pl [tab-separated table] "[column title]" "[another column title]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/condense_list_values_into_one_item.pl">
                        <code>
                         condense_list_values_into_one_item.pl
                        </code>
                       </a>
                       : For any values in specified columns that are comma-separated lists, replaces comma-separated list with the first, smallest, or largest value in the list.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl condense_list_values_into_one_item.pl [table] [0 to use the first value, 1 to use the smallest value, 2 to use the greatest value] "[title of column to replace lists in]" "[title of another column to replace lists in]" "[title of another column to replace lists in]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/replace_values_with_column_proportions.pl">
                        <code>
                         replace_values_with_column_proportions.pl
                        </code>
                       </a>
                       : Calculates sum of each column containing numerical values. Replaces values in column with the proportion of its sum.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl replace_values_with_column_proportions.pl [tab-separated table] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_filename_column_from_filepath_column.pl">
                        <code>
                         add_filename_column_from_filepath_column.pl
                        </code>
                       </a>
                       : Retrieves filepaths from specified column (or first column by default). Adds new column with filenames retrieved from these filepaths.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_filename_column_from_filepath_column.pl [table to add new column to] [optional title of column containing filepaths (if not provided, uses first column)] [optional 1 to remove all file extensions in output filenames (such that filename contains no .s)] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_column_comparing_two_columns.pl">
                        <code>
                         add_column_comparing_two_columns.pl
                        </code>
                       </a>
                       : Adds a column indicating whether or not there is a difference between two columns.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_column_comparing_two_columns.pl [tab-separated table] "[title of first column to compare]" "[title of second column to compare]" [1 to print the actual values when different] "[optional new column title]" "[optional new column value if values are identical]" "[optional new column value if values are different]" "[optional new column value if one value missing]" "[optional new column value if both values missing]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/replace_column_values_with_other_column_where_yet_another_column_has_query_value.pl">
                        <code>
                         replace_column_values_with_other_column_where_yet_another_column_has_query_value.pl
                        </code>
                       </a>
                       : Fills in values in column of interest with values from other column when they are present.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl replace_column_values_with_other_column_where_yet_another_column_has_query_value.pl [table] "[title of column to replace values in]" "[title of column with potential replacement values]" "[title of column to search]" "[query value]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/split_column_after_query.pl">
                        <code>
                         split_column_after_query.pl
                        </code>
                       </a>
                       : Splits column into two columns following appearance of query in each column value. If a cell does not contain the query, duplicates the column value.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl split_column_after_query.pl [table] "[column name]" "[query]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_column_with_filename.pl">
                        <code>
                         add_column_with_filename.pl
                        </code>
                       </a>
                       : Adds column with table filename in all values.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_column_with_filename.pl [table to add column to] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Column manipulation with dates
                     </h3>
                     <a aria-label="Permalink: Column manipulation with dates" class="anchor" href="#column-manipulation-with-dates" id="user-content-column-manipulation-with-dates">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/dates_in_columns_to_YYYY_MM_DD.pl">
                        <code>
                         dates_in_columns_to_YYYY_MM_DD.pl
                        </code>
                       </a>
                       : Converts dates in specified columns to YYYY-MM-DD format. Multiple dates may be separated by a ", ". Column titles must not have spaces.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl dates_in_columns_to_YYYY_MM_DD.pl [table] "[title of column with dates]" "[title of another column with dates]" "[title of another column with dates]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_difference_in_dates_column.pl">
                        <code>
                         add_difference_in_dates_column.pl
                        </code>
                       </a>
                       : Adds column listing difference in dates (in days) between columns specified in parameter column titles. Dates must be in YYYY-MM-DD format.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_difference_in_dates_column.pl [table] "[title of column with dates]" "[title of another column with dates]" "[optional title of new column]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_earliest_or_latest_date_column.pl">
                        <code>
                         add_earliest_or_latest_date_column.pl
                        </code>
                       </a>
                       : Adds column listing the latest or earliest of the specified columns. Dates must be in YYYY-MM-DD format.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_earliest_or_latest_date_column.pl [table] [0 to select earliest date, 1 to select latest date] "[title of column with dates]" "[title of another column with dates]" "[title of another column with dates]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_column_with_earliest_or_latest_date_before_cutoff.pl">
                        <code>
                         add_column_with_earliest_or_latest_date_before_cutoff.pl
                        </code>
                       </a>
                       : Adds column listing the date that is the latest or earliest of the specified columns but earlier than the cut-off date from the cut-off date column. Dates must be in YYYY-MM-DD format.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_column_with_earliest_or_latest_date_before_cutoff.pl [table] [0 to select earliest date, 1 to select latest date] "[title of column with cut-off date]" "[title of column with dates to select from]" "[title of another column with dates to select from]" "[title of another column with dates to select from]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/sort_date_columns.pl">
                        <code>
                         sort_date_columns.pl
                        </code>
                       </a>
                       : Sorts the dates in the specified columns. For each row, of the dates in the specified columns, the earliest date will go in the first specified column, the second-earliest in the second specified column, etc. Empty values go last. Dates provided must be in YYYY-MM-DD format.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl sort_date_columns.pl [table] "[title of column with dates]" "[title of another column with dates]" "[title of another column with dates]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/sort_date_columns_with_paired_label_columns.pl">
                        <code>
                         sort_date_columns_with_paired_label_columns.pl
                        </code>
                       </a>
                       : Sorts the dates in the specified columns. For each row, of the dates in the specified columns, the earliest date will go in the first specified column, the second-earliest in the second specified column, etc. Empty values go last. Dates provided must be in YYYY-MM-DD format.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl sort_date_columns_with_paired_label_columns.pl [table] "[title of column with dates]" "[title of label column that should travel with paired dates]" "[title of another column with dates]" "[title of label column that should travel with those paired dates]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Row manipulation
                     </h3>
                     <a aria-label="Permalink: Row manipulation" class="anchor" href="#row-manipulation" id="user-content-row-manipulation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/filter_table_rows_by_column_value.pl">
                        <code>
                         filter_table_rows_by_column_value.pl
                        </code>
                       </a>
                       : Filters table by column values. Only includes rows matching (containing, beginning with, ending with, equal to, not equal to, or not containing) column value of interest in column to filter by. Case-sensitive.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl filter_table_rows_by_column_value.pl [tab-separated table] [0 to match cells containing query, 1: beginning with, 2: ending with, 3: equal to, 4: not equal to, 5: not containing] "[title of column to filter by]" "[column value to select]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/filter_table_rows_by_column_values.pl">
                        <code>
                         filter_table_rows_by_column_values.pl
                        </code>
                       </a>
                       : Filters table by column values. Only includes rows matching (containing, beginning with, ending with, equal to, not equal to, or not containing) column values of interest in column to filter by. Case-sensitive.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl filter_table_rows_by_column_values.pl [tab-separated table] [0 to match cells containing query, 1: beginning with, 2: ending with, 3: equal to] "[title of column to filter by]" "[column value to select]" "[another column value to select]" "[etc.]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/filter_table_rows_by_numerical_column_value.pl">
                        <code>
                         filter_table_rows_by_numerical_column_value.pl
                        </code>
                       </a>
                       : Filters table by column values. Only includes rows equal to, greater than, or less than column value of interest in column to filter by.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl filter_table_rows_by_numerical_column_value.pl [tab-separated table] [0 to match cells equal to query, 1: less than, 2: less than or equal to, 3: greater than, 4: greater than or equal to] "[title of column to filter by]" "[column value to select]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/retrieve_subset_of_rows.pl">
                        <code>
                         retrieve_subset_of_rows.pl
                        </code>
                       </a>
                       : Subsets table to only rows of interest (those containing one of the specified values in the specified column).
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_subset_of_rows.pl [file path of table] "[title of column to filter by]" [file path of list of values to filter that column to] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/delete_table_rows_with_column_value.pl">
                        <code>
                         delete_table_rows_with_column_value.pl
                        </code>
                       </a>
                       : Deletes rows in table by column values. Only includes rows without column value containing text to filter out in column to filter by. Case-sensitive.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl delete_table_rows_with_column_value.pl [tab-separated table] "[query to select rows to delete]" [0 to match cells containing query, 1: beginning with, 2: ending with, 3: equal to] "[title of column to filter by]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/merge_rows_by_column_value.pl">
                        <code>
                         merge_rows_by_column_value.pl
                        </code>
                       </a>
                       : Merges (takes union of) all columns in rows with shared value in column to merge by. If titles of columns not to merge by are provided, leaves one row per input row with all other columns identical. If no columns not to merge by provided, fully merges any rows sharing a value in column to merge by (one row per value).
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl merge_rows_by_column_value.pl [table to merge] "[title of column to merge by]" "[optional title of column not to merge]" "[optional title of another column not to merge]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/verify_all_rows_in_one_table_appear_in_second_table.pl">
                        <code>
                         verify_all_rows_in_one_table_appear_in_second_table.pl
                        </code>
                       </a>
                       : Verifies that all rows/lines in table 1 also appear in table 2 (that table 2 is a superset of table 1). Prints all rows from table 1 that are missing from table 2. If column numbers is specified, only compares values from that column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl verify_all_rows_in_one_table_appear_in_second_table.pl [subset table] [superset table] [column number to compare--set to -1 to compare full row] &gt; [output list of missing rows]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/fill_in_missing_rows.pl">
                        <code>
                         fill_in_missing_rows.pl
                        </code>
                       </a>
                       : Catalogues all values in parameter columns. In any combination of values is missing, adds it as a row with NAs in all other columns.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl fill_in_missing_rows.pl [table] "[title of column of interest 1]"" "[title of column of interest 2]" etc. &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Table manipulation
                     </h3>
                     <a aria-label="Permalink: Table manipulation" class="anchor" href="#table-manipulation" id="user-content-table-manipulation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/concatenate_tables.pl">
                        <code>
                         concatenate_tables.pl
                        </code>
                       </a>
                       : Concatenates tables with potentially different columns, adding empty space for missing column values.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl concatenate_tables.pl [table1] [table2] [table3] etc. &gt; [concatenated output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_columns_from_other_table.pl">
                        <code>
                         add_columns_from_other_table.pl
                        </code>
                       </a>
                       : Adds columns from another table, matching rows in the two tables by selected columns to merge by.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_columns_from_other_table.pl [table to add columns to] "[title of column in first table to identify rows by]" [table to add columns from] "[title of column in second table to identify rows by]" "[title of column to add]" "[title of another column to add]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/merge_two_tables_by_column_value.pl">
                        <code>
                         merge_two_tables_by_column_value.pl
                        </code>
                       </a>
                       : Merges (takes union of) two tables by the values in the specified columns.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl merge_two_tables_by_column_value.pl [table1 file path] [table1 column number (0-indexed)] [table2 file path] [table2 column number (0-indexed)] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/merge_tables_by_column_value.pl">
                        <code>
                         merge_tables_by_column_value.pl
                        </code>
                       </a>
                       : Merges (takes union of) multiple tables by the values in the specified columns. See script for description of input file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl merge_tables_by_column_value.pl [file describing input] &gt; [merged output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/merge_tables_by_first_column_values.pl">
                        <code>
                         merge_tables_by_first_column_values.pl
                        </code>
                       </a>
                       : Merges (takes union of) multiple tables by the values in the first column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl merge_tables_by_first_column_values.pl [table] [another table] [another table] [etc.] &gt; [merged output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_newline_between_lines_with_nonconsecutive_values.pl">
                        <code>
                         add_newline_between_lines_with_nonconsecutive_values.pl
                        </code>
                       </a>
                       : Adds newline between lines containing non-consecutive values in first column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_newline_between_lines_with_nonconsecutive_values.pl [table] [column with integer values] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_header_line_to_tables.pl">
                        <code>
                         add_header_line_to_tables.pl
                        </code>
                       </a>
                       : Adds header line to all input files. Saves output tables in new files at same path with _with_header_line.txt extension. Four spaces in the input header line are replaced with tabs.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_header_line_to_tables.pl "[header line]" [table] [optional additional table] [etc.]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/split_table_by_column_value.pl">
                        <code>
                         split_table_by_column_value.pl
                        </code>
                       </a>
                       : Splits table into multiple tables, one for each column value in specified column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl split_table_by_column_value.pl [input table file path] "[title of column to split by]"
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Replicates
                     </h3>
                     <a aria-label="Permalink: Replicates" class="anchor" href="#replicates" id="user-content-replicates">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/annotate_replicates.pl">
                        <code>
                         annotate_replicates.pl
                        </code>
                       </a>
                       : Assigns a source number to all replicates from the same source. Adds source number as a column to table to annotate.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl annotate_replicates.pl [file listing replicate ids from same source, tab-separated, one line per source] [table to annotate] "[title of column containing replicate ids in table to annotate]" "[optional source column title for output]" &gt; [annotated output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/select_one_replicate.pl">
                        <code>
                         select_one_replicate.pl
                        </code>
                       </a>
                       : Given a table with multiple replicates from the same source, selects one replicate per source, using selected column to select replicate. In the event of a tie, selects first appearing replicate.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl select_one_replicate.pl [tab-separated table] "[title of column containing source of each replicate (same value for every replicate from the same source)]" "[title of column to use to select replicate]" [0 to select replicate with smallest numerical value, 1 to select replicate with largest numerical value] &gt; [annotated output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_shared_values_summary_column.pl">
                        <code>
                         add_shared_values_summary_column.pl
                        </code>
                       </a>
                       : Summarizes all values appearing in columns to summarize (sample ids and dates, for example) for each shared value (patient id, for example). Adds summary in new column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_shared_values_summary_column.pl [tab-separated table] "[title of column containing values shared by rows]" "[title of column to include in summary of shared values]" "[title of another column to include in summary of shared values]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Other
                     </h3>
                     <a aria-label="Permalink: Other" class="anchor" href="#other" id="user-content-other">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/summarize_table_columns.pl">
                        <code>
                         summarize_table_columns.pl
                        </code>
                       </a>
                       : Summarizes values in table columns. Similar to str in R.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl summarize_table_columns.pl [tab-separated table] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/summarize_column_value_combination_frequencies.pl">
                        <code>
                         summarize_column_value_combination_frequencies.pl
                        </code>
                       </a>
                       : Summarizes frequency of all column values and combinations of column values in columns of interest in table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl summarize_column_value_combination_frequencies.pl [table to summarize] "[title of column of interest]" "[optional title of another column of interest]" "[optional title of another column of interest]" [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_column_indicating_presence_of_query.pl">
                        <code>
                         add_column_indicating_presence_of_query.pl
                        </code>
                       </a>
                       : Adds column with values indicating presence of query in row.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_column_indicating_presence_of_query.pl [table] [query] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/add_row_indicating_presence_of_query.pl">
                        <code>
                         add_row_indicating_presence_of_query.pl
                        </code>
                       </a>
                       : Adds row with values indicating presence of query in column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_row_indicating_presence_of_query.pl [table] [query] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/count_occurrences_of_column_values.pl">
                        <code>
                         count_occurrences_of_column_values.pl
                        </code>
                       </a>
                       : Counts number occurrences of each value in selected column of table.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl count_occurrences_of_column_values.pl [tab-separated table] "[column title]" &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/verify_column_values_are_consistent_across_tables.pl">
                        <code>
                         verify_column_values_are_consistent_across_tables.pl
                        </code>
                       </a>
                       : Verifies that the same column values always appear with row identifier values. Column titles must be consistent across tables, including title of row identifier column.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl verify_column_values_are_consistent_across_tables.pl [row identifier column title] [table1] [table2] [table 3] [etc.]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/tables/calculate_date_range_in_column.pl">
                        <code>
                         calculate_date_range_in_column.pl
                        </code>
                       </a>
                       : Prints minimum date, maximum date, and difference in dates (in days) in specified column in each input files. Dates must be in YYYY-MM-DD format.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl calculate_date_range_in_column.pl [column number of column with dates (0-indexed)] [table] [another table] [another table] [etc.]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Miscellaneous (
                      <a href="/lakras/bio-helper-scripts/blob/main/misc">
                       <code>
                        misc
                       </code>
                      </a>
                      )
                     </h2>
                     <a aria-label="Permalink: Miscellaneous (misc)" class="anchor" href="#miscellaneous-misc" id="user-content-miscellaneous-misc">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/add_commas_to_all_numbers_in_file.pl">
                        <code>
                         add_commas_to_all_numbers_in_file.pl
                        </code>
                       </a>
                       : Detects all numbers in contents of file and adds thousands-separator commas where needed.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_commas_to_all_numbers_in_file.pl [input file to add commas to] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/add_commas_to_all_numbers.pl">
                        <code>
                         add_commas_to_all_numbers.pl
                        </code>
                       </a>
                       : Detects all numbers in input text and adds thousands-separator commas where needed.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl add_commas_to_all_numbers.pl "[text to add commas to]" &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/run_script_on_multiple_files.pl">
                        <code>
                         run_script_on_multiple_files.pl
                        </code>
                       </a>
                       : Runs script individually on multiple input files, otherwise using the same input parameters. Script must accept input file as first parameter.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl run_script_on_multiple_files.pl [file path of script to run] [file with list of input files to run script on] [optional extension to add to input file path to create output file path] [optional second parameter to provide script] [optional third parameter to provide script] [etc.]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/summarize_files.pl">
                        <code>
                         summarize_files.pl
                        </code>
                       </a>
                       : Summarizes files. Generates table with filepath, filename, filename with extensions trimmed off, and number lines, words, and characters in the file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl summarize_files.pl [file with list of files to summarize, one per line] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/download_files.pl">
                        <code>
                         download_files.pl
                        </code>
                       </a>
                       : Downloads files listed in input file from online or from google storage bucket.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl download_files.pl [file with list of files to download] [optional output directory]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/download_files_listed_in_table_column.pl">
                        <code>
                         download_files_listed_in_table_column.pl
                        </code>
                       </a>
                       : Downloads files from addresses listed in specified table column in input table. Downloads files from online or from google storage bucket to new directory. If no output directory provided, output directory set to input file path sans extension followed by column name.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl download_files_listed_in_table_column.pl [table with list of files to download in one of the columns] "[title of column containing filepaths to download]" [optional output directory]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/download_and_rename_files.pl">
                        <code>
                         download_and_rename_files.pl
                        </code>
                       </a>
                       : Downloads files listed in input file from online or from google storage bucket to new directory and renames to specified filename.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl download_and_rename_files.pl [file with list of files to download] [optional output directory]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/retrieve_all_filepaths_in_directories.pl">
                        <code>
                         retrieve_all_filepaths_in_directories.pl
                        </code>
                       </a>
                       : Retrieves filepaths of all files in provided directories.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl retrieve_all_filepaths_in_directories.pl [directory path] [another directory path] [another directory path] [etc.] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/bulk_diff_same_name_files_in_two_directories.pl">
                        <code>
                         bulk_diff_same_name_files_in_two_directories.pl
                        </code>
                       </a>
                       : Compares all files with the same name in the two input directories. Prints any differences between same-name files. The two directories must contain files with the same names.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl bulk_diff_same_name_files_in_two_directories.pl [first directory] [second directory]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/combine_files_and_delete_duplicate_lines.pl">
                        <code>
                         combine_files_and_delete_duplicate_lines.pl
                        </code>
                       </a>
                       : Combines input files. Prints each line only once (no duplicate lines). Prints lines in order in which they first appear in all the input files.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl combine_files_and_delete_duplicate_lines.pl [input file] [another input file] [another input file] [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/bulk_rename.pl">
                        <code>
                         bulk_rename.pl
                        </code>
                       </a>
                       : Replaces all occurrences of values in file paths to mapped replacement values. Values to replace should not overlap or be substrings of each other. Files already at replacement paths will be overwritten. Paths of files to rename can be provided directly as arguments or as one file with a list of filepaths, one per line.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl bulk_rename.pl [tab-separated file mapping current values (first column) to new values (second column)] [filepaths of at least two files to rename OR path of file containing filepaths of files to rename, one per line]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/bulk_grep.pl">
                        <code>
                         bulk_grep.pl
                        </code>
                       </a>
                       : Searches all input files for queries listed in query list file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl bulk_grep.pl [file listing queries, one per line] [file to grep] [another file to grep] [etc.] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/bulk_grep_one_file.pl">
                        <code>
                         bulk_grep_one_file.pl
                        </code>
                       </a>
                       : Searches input file for queries. Prints lines containing queries.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl bulk_grep_one_file.pl [file to grep] "[query 1]" "[query 2]" "[query 3]" [etc.] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/bulk_grep_is_detected.pl">
                        <code>
                         bulk_grep_is_detected.pl
                        </code>
                       </a>
                       : Searches all input files for queries listed in query list file. Outputs query and each file it was found in, tab-separated, one line per query-file pair where query is detected. If query is not detected in any file, prints query followed by "not detected".
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl bulk_grep_is_detected.pl [file listing queries, one per line] [file to grep] [another file to grep] [etc.] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/bulk_grep_is_detected_table.pl">
                        <code>
                         bulk_grep_is_detected_table.pl
                        </code>
                       </a>
                       : Searches all input files for queries listed in query list file. Outputs a table with detection of all queries in all files, tab-separated, one row per query, one column per file.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl bulk_grep_is_detected_table.pl [file listing queries, one per line] [file to grep] [another file to grep] [etc.] &gt; [output file path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/split_file_into_n_files.pl">
                        <code>
                         split_file_into_n_files.pl
                        </code>
                       </a>
                       : Splits file with multiple lines up into a number of smaller files, each with about the same number of lines.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl split_file_into_n_files.pl [file path]  [number output files to generate]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/split_file_into_2n_overlapping_files.pl">
                        <code>
                         split_file_into_2n_overlapping_files.pl
                        </code>
                       </a>
                       : Splits file with multiple lines up into a number of overlapping smaller files. Each line will appear in two files. Either all output files will have about the same number of lines, or there will be two "jagged" half-sized output files if it is a priority for all output files to only have lines that are consecutive in the input. See script for example input and output.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl split_file_into_2n_overlapping_files.pl [file path] [number equal slices (n)] [1 to generate N=2*n approximately equal sized files, 0 to generate N=2*n+1 output files with lines that are also consecutive in the input] [optional 1 to NOT print half-sized overhang files (can be used only if previous option is set to 0)]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/split_file_into_overlapping_files_by_date.pl">
                        <code>
                         split_file_into_overlapping_files_by_date.pl
                        </code>
                       </a>
                       : Splits file with multiple lines up into overlapping smaller files, each containing dates within twice the parameter number of days. Each line will appear in two files.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl split_file_into_overlapping_files_by_date.pl [file path] [half the date range in each output file] [column containing dates, 0-indexed]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/erase_empty_files.pl">
                        <code>
                         erase_empty_files.pl
                        </code>
                       </a>
                       : Erases empty files.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl erase_empty_files.pl [file to check and potentially erase] [another file to check and potentially erase] [a third file to check and potentially erase] etc.
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/bulk_delete_files_in_list.pl">
                        <code>
                         bulk_delete_files_in_list.pl
                        </code>
                       </a>
                       : Deletes files in input list.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl bulk_delete_files_in_list.pl [file containing list of paths of files to delete, one per line]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/bulk_move_files_in_list.pl">
                        <code>
                         bulk_move_files_in_list.pl
                        </code>
                       </a>
                       : Moves files in input list to provided directory.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl bulk_move_files_in_list.pl [file containing list of paths of files to move, one per line] [directory to move files to]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/match_file_paths_to_sample_names.pl">
                        <code>
                         match_file_paths_to_sample_names.pl
                        </code>
                       </a>
                       : Matches each sample name to a file path containing that sample name. Outputs sample names with file paths, tab-separated, one sample per line.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl match_file_paths_to_sample_names.pl [file containing list of sample names, one per line] [file containing list of file paths, one per line] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/combine_files_and_delete_duplicate_lines.pl">
                        <code>
                         combine_files_and_delete_duplicate_lines.pl
                        </code>
                       </a>
                       : Combines input files. Prints each line only once (no duplicate lines). Prints lines in order in which they first appear in all the input files.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl combine_files_and_delete_duplicate_lines.pl [input file] [another input file] [another input file] [etc.] &gt; [output table path]
                        </code>
                       </em>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="/lakras/bio-helper-scripts/blob/main/misc/select_random_lines.pl">
                        <code>
                         select_random_lines.pl
                        </code>
                       </a>
                       : Selects a certain number of lines at random from input table or list.
                      </p>
                      <p dir="auto">
                       <em>
                        Usage:
                        <code>
                         perl select_random_lines.pl [table or list] [number rows or lines to select at random] &gt; [output table or list]
                        </code>
                       </em>
                      </p>
                     </li>
                    </ul>
                    <p dir="auto">
                     More coming soon :)
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="khNJjTFYSJn7IIuhH27s5KCtY+hmMUbF4KxoYOq6FB2kNlNuY1B4WpA5c2/wahOPm89TrgFuuTYqn9O1rYYhSA==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 Helper scripts for processing genomic sequence data.
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <h3 class="sr-only">
                 License
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="#MIT-1-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                   </path>
                  </svg>
                  MIT license
                 </a>
                </div>
                <include-fragment src="/lakras/bio-helper-scripts/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/lakras/bio-helper-scripts/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/lakras/bio-helper-scripts/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   3
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/lakras/bio-helper-scripts/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/lakras/bio-helper-scripts/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Flakras%2Fbio-helper-scripts&amp;report=lakras+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/lakras/bio-helper-scripts/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/lakras/packages?repo_name=bio-helper-scripts">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/lakras/bio-helper-scripts/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="Perl 100.0" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#0298c3 !important;;width: 100.0%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/lakras/bio-helper-scripts/search?l=perl">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#0298c3;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Perl
                  </span>
                  <span>
                   100.0%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
        2024 GitHub,Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You cant perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
