<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-ca8ae01e33b4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-fb4b8d40f206.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - mepepin/Indirect.Epigenomics
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:E2CFEB:14B9F82:6605ABE5" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDpFMkNGRUI6MTRCOUY4Mjo2NjA1QUJFNSIsInZpc2l0b3JfaWQiOiI2NDUyMjQxODU2MjM4ODE3ODA3IiwicmVnaW9uX2VkZ2UiOiJpYWQiLCJyZWdpb25fcmVuZGVyIjoiaWFkIn0=" data-pjax-transient="true" name="visitor-payload">
           <meta content="d32d7e83151b63989fa983029ed88876cf5014d4c05cebf8716d9480a6f0888f" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:444842225" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="Contribute to mepepin/Indirect.Epigenomics development by creating an account on GitHub." name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/mepepin/Indirect.Epigenomics" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/b4f65e3afa089e3c779cb6cd80a991e31a76450002cd1280e523b56948d8fc01/mepepin/Indirect.Epigenomics" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - mepepin/Indirect.Epigenomics" name="twitter:title">
                    <meta content="Contribute to mepepin/Indirect.Epigenomics development by creating an account on GitHub." name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/b4f65e3afa089e3c779cb6cd80a991e31a76450002cd1280e523b56948d8fc01/mepepin/Indirect.Epigenomics" property="og:image">
                      <meta content="Contribute to mepepin/Indirect.Epigenomics development by creating an account on GitHub." property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - mepepin/Indirect.Epigenomics" property="og:title"/>
                        <meta content="https://github.com/mepepin/Indirect.Epigenomics" property="og:url"/>
                        <meta content="Contribute to mepepin/Indirect.Epigenomics development by creating an account on GitHub." property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="584826db69b6459b3c4c29172349cd9ed6729235b962a93107214b2469aa5b40" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="a2a6f0b866a9de2e69b1f990e0730e22d502d619ef6d55aae603cacc6e2f5285" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/mepepin/Indirect.Epigenomics git https://github.com/mepepin/Indirect.Epigenomics.git" name="go-import"/>
                        <meta content="14981867" name="octolytics-dimension-user_id">
                         <meta content="mepepin" name="octolytics-dimension-user_login">
                          <meta content="444842225" name="octolytics-dimension-repository_id">
                           <meta content="mepepin/Indirect.Epigenomics" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="444842225" name="octolytics-dimension-repository_network_root_id">
                               <meta content="mepepin/Indirect.Epigenomics" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/mepepin/Indirect.Epigenomics" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-11f6759e1cef.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/mepepin/Indirect.Epigenomics","user_id":null}}' data-hydro-click-hmac="48f10af92f8a23254ef5daed3b384ab6dfafaa4af32a9171723e8c29df44133e" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fmepepin%2FIndirect.Epigenomics">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=RYaVGgDWZbZ6ZSYADXAey6dPDmEFBCYF%2FA7xcOQ83RRP5plhgtsjcilTX4znaEWZKUSdd%2FMzDifnkD%2FJHUGIBTE3QgXMKy0BIzPa7DnxJ8ydBcs9yseoVMGWNqsmqMp5ORUlf8WhjP9QP4s0asDfUrvY0cjYFWSsVp1qlVnTK57oG5tWMImJKIXZtmXUMCSdS8K%2BFZEOIJV2XlldtHzLsxIy0SsUPWhAmxuISAqqenRdgPyuT7%2B2oVradq5GGBydpb%2B97pa3z56cpgli4ghXLIyLblLOlRbzfvCEFT%2FDC%2BVXMg7DymsRQp3CK8Tem%2F4oRJhpEPlyChG2PHM6Xviv9E2H0S0QguCbJdX81FA7mQctcqOa2ilUQMw41iigr2IMXfWwt7gD1CUWBauQDfJgKgxNKDY8AYHk9XKe9WNz5WMQzRXBacA7cT8CoNnNdu7nJznjt6%2FKy94%2FXvfZBgyBz%2BD3J3hV%2FUKqjaZSa0oJVH8TGST31if1k3jVC6Ijv6Wn41znLkxFmocu1Op6CKzso855jHpT6o7RKMTF48Mr--BwnQtlMmL8rQZCea--AzxJhMdxXNJoYPa98mYxvw%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="mepepin" data-current-repository="mepepin/Indirect.Epigenomics" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="EDqCK1-U1CFHeogiRwTqZfeCjcwFPGdPzq6MJYFimFyRkMgzSS66cD7v9SrAysPdzcpZcB5awTuiOOCOcBJYIA" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:mepepin/Indirect.Epigenomics">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-b78e4d70-8fcc-499b-9367-322d3391beed" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-b78e4d70-8fcc-499b-9367-322d3391beed">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="OezzlKLrB0M9vdPebLCCNT8Y0tmI5CyuSICDh210MsqHKOZi3eQhOHe8sq69VFstjqQJ4w//bH4ByWcCvgpb+g=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="AngNt6KmvuZLOPBvRY0/fX2sRYoeePPIzmbLH0BeBe6I/6ProowDMQeA34R8WGGluMSmhBzAoTTpwL5ELs2uAw==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="x7fTWh8X6B/YAaO2bNS+f8716Hii4FwOvPRGztwXdL6qeXsGPeUbfVLs00zKhmhxGq4vAy/eboRs2CRpoVI23Q==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="nn6ubnYQ+JXNupvqDW+RyaTHjgqAU8JQKYgN3mOsgEJoY2sIioPI6wgXNXO/g+XhEocGIyjOTRs31LbkqNAsUg==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/mepepin/Indirect.Epigenomics","user_id":null}}' data-hydro-click-hmac="48f10af92f8a23254ef5daed3b384ab6dfafaa4af32a9171723e8c29df44133e" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fmepepin%2FIndirect.Epigenomics">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/mepepin/Indirect.Epigenomics","user_id":null}}' data-hydro-click-hmac="48f10af92f8a23254ef5daed3b384ab6dfafaa4af32a9171723e8c29df44133e" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=mepepin%2FIndirect.Epigenomics">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-134323ec-af91-490e-9311-5c9663bd4cfc" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-a746f51f-7f1c-4cba-822b-8dff609f6351" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-a746f51f-7f1c-4cba-822b-8dff609f6351" id="tooltip-134323ec-af91-490e-9311-5c9663bd4cfc" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/mepepin/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/mepepin" rel="author">
            mepepin
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/mepepin/Indirect.Epigenomics">
            Indirect.Epigenomics
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/mepepin/Indirect.Epigenomics","user_id":null}}' data-hydro-click-hmac="d7e36ef71f24190985921c6caa0c0abf3078df9d4d65fc0522ebc40ab4ed9728" data-view-component="true" href="/login?return_to=%2Fmepepin%2FIndirect.Epigenomics" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":444842225,"auth_type":"LOG_IN","originating_url":"https://github.com/mepepin/Indirect.Epigenomics","user_id":null}}' data-hydro-click-hmac="cbda864fd90a1c50fc7ff6ba473b02fb66f2168c969def11d908855ef145770d" data-view-component="true" href="/login?return_to=%2Fmepepin%2FIndirect.Epigenomics" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
             0
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":444842225,"auth_type":"LOG_IN","originating_url":"https://github.com/mepepin/Indirect.Epigenomics","user_id":null}}' data-hydro-click-hmac="9b89e3381484899669f0aaf2dece1fb06af50b2c464d31fa8502683fe4622a2e" data-view-component="true" href="/login?return_to=%2Fmepepin%2FIndirect.Epigenomics" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
              0
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <h3 class="sr-only">
          License
         </h3>
         <div class="mb-2">
          <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="/mepepin/Indirect.Epigenomics/blob/main/LICENSE">
           <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
            </path>
           </svg>
           MIT license
          </a>
         </div>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/mepepin/Indirect.Epigenomics/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           stars
          </a>
          <a class="Link--secondary no-underline mr-3" href="/mepepin/Indirect.Epigenomics/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/mepepin/Indirect.Epigenomics/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/mepepin/Indirect.Epigenomics/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/mepepin/Indirect.Epigenomics/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":444842225,"auth_type":"LOG_IN","originating_url":"https://github.com/mepepin/Indirect.Epigenomics","user_id":null}}' data-hydro-click-hmac="9b89e3381484899669f0aaf2dece1fb06af50b2c464d31fa8502683fe4622a2e" data-view-component="true" href="/login?return_to=%2Fmepepin%2FIndirect.Epigenomics" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/mepepin/Indirect.Epigenomics","user_id":null}}' data-hydro-click-hmac="d7e36ef71f24190985921c6caa0c0abf3078df9d4d65fc0522ebc40ab4ed9728" data-view-component="true" href="/login?return_to=%2Fmepepin%2FIndirect.Epigenomics" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /mepepin/Indirect.Epigenomics" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/mepepin/Indirect.Epigenomics" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /mepepin/Indirect.Epigenomics/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/mepepin/Indirect.Epigenomics/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /mepepin/Indirect.Epigenomics/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/mepepin/Indirect.Epigenomics/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /mepepin/Indirect.Epigenomics/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/mepepin/Indirect.Epigenomics/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /mepepin/Indirect.Epigenomics/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/mepepin/Indirect.Epigenomics/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /mepepin/Indirect.Epigenomics/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/mepepin/Indirect.Epigenomics/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/mepepin/Indirect.Epigenomics/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /mepepin/Indirect.Epigenomics/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/mepepin/Indirect.Epigenomics/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-94218cfd-62cc-477d-8011-d02fbd24dfd7-list" aria-haspopup="true" aria-labelledby="tooltip-1e149555-f735-46c5-bc36-e8829b98dd75" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-94218cfd-62cc-477d-8011-d02fbd24dfd7-button" popovertarget="action-menu-94218cfd-62cc-477d-8011-d02fbd24dfd7-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-94218cfd-62cc-477d-8011-d02fbd24dfd7-button" id="tooltip-1e149555-f735-46c5-bc36-e8829b98dd75" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-94218cfd-62cc-477d-8011-d02fbd24dfd7-button" anchor-offset="normal" data-view-component="true" id="action-menu-94218cfd-62cc-477d-8011-d02fbd24dfd7-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-94218cfd-62cc-477d-8011-d02fbd24dfd7-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-94218cfd-62cc-477d-8011-d02fbd24dfd7-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/mepepin/Indirect.Epigenomics" id="item-1b54d587-6541-46cb-b7f5-43f07930986e" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/mepepin/Indirect.Epigenomics/issues" id="item-72b08104-6b8a-4ccf-879f-a45e087022c8" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/mepepin/Indirect.Epigenomics/pulls" id="item-c0479390-750a-4367-9fb8-85975c7363f2" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/mepepin/Indirect.Epigenomics/actions" id="item-8e2aa06f-d8ba-4879-91b5-278cb3146cd7" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/mepepin/Indirect.Epigenomics/projects" id="item-689c64d2-a8ae-4157-ba69-afb4e1456e35" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/mepepin/Indirect.Epigenomics/security" id="item-2fd7defe-2096-445c-b7ab-8fabd3b4d771" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/mepepin/Indirect.Epigenomics/pulse" id="item-9e99b23f-8494-4fc5-a0b2-915b8a41d8c0" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         mepepin/Indirect.Epigenomics
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/mepepin/Indirect.Epigenomics/spoofed_commit_check/ba5b13aeb429b4a3d960017be1dfa62befdbf321">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":444842225,"defaultBranch":"main","name":"Indirect.Epigenomics","ownerLogin":"mepepin","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2022-01-05T14:52:22.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/14981867?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"main","listCacheKey":"v0:1642431095.7648141","canEdit":false,"refType":"branch","currentOid":"ba5b13aeb429b4a3d960017be1dfa62befdbf321"},"tree":{"items":[{"name":"README_files","path":"README_files","contentType":"directory"},{"name":".DS_Store","path":".DS_Store","contentType":"file"},{"name":".gitignore","path":".gitignore","contentType":"file"},{"name":"DCM_vs_CON.csv","path":"DCM_vs_CON.csv","contentType":"file"},{"name":"LICENSE","path":"LICENSE","contentType":"file"},{"name":"README.Rmd","path":"README.Rmd","contentType":"file"},{"name":"README.html","path":"README.html","contentType":"file"},{"name":"README.log","path":"README.log","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"},{"name":"README.tex","path":"README.tex","contentType":"file"},{"name":"qcReport.pdf","path":"qcReport.pdf","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":11,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/mepepin/Indirect.Epigenomics/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/mepepin/Indirect.Epigenomics.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone mepepin/Indirect.Epigenomics","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/mepepin/Indirect.Epigenomics/archive/refs/heads/main.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=444842225"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"3","overviewFiles":[{"displayName":"README.md","repoName":"Indirect.Epigenomics","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003ctable\u003e\n  \u003cthead\u003e\n  \u003ctr\u003e\n  \u003cth\u003etitle\u003c/th\u003e\n  \u003cth\u003eauthor\u003c/th\u003e\n  \u003cth\u003edate\u003c/th\u003e\n  \u003cth\u003eoutput\u003c/th\u003e\n  \u003cth\u003eheader-includes\u003c/th\u003e\n  \u003cth\u003emainfont\u003c/th\u003e\n  \u003cth\u003efontsize\u003c/th\u003e\n  \u003cth\u003ealways_allow_html\u003c/th\u003e\n  \u003cth\u003eeditor_options\u003c/th\u003e\n  \u003c/tr\u003e\n  \u003c/thead\u003e\n  \u003ctbody\u003e\n  \u003ctr\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003eSupplemental Methods - Epigenenomics of Sample_Group in Heart Failure\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003eMark E. Pepin\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e31 December, 2021\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\u003ctable\u003e\n  \u003cthead\u003e\n  \u003ctr\u003e\n  \u003cth\u003ehtml_document\u003c/th\u003e\n  \u003cth\u003eword_document\u003c/th\u003e\n  \u003cth\u003epdf_document\u003c/th\u003e\n  \u003c/tr\u003e\n  \u003c/thead\u003e\n  \u003ctbody\u003e\n  \u003ctr\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\u003ctable\u003e\n  \u003cthead\u003e\n  \u003ctr\u003e\n  \u003cth\u003ekeep_md\u003c/th\u003e\n  \u003cth\u003ecode_folding\u003c/th\u003e\n  \u003cth\u003etoc\u003c/th\u003e\n  \u003cth\u003etoc_float\u003c/th\u003e\n  \u003c/tr\u003e\n  \u003c/thead\u003e\n  \u003ctbody\u003e\n  \u003ctr\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003etrue\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003ehide\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003etrue\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003etrue\u003c/div\u003e\u003c/td\u003e\n  \u003c/tr\u003e\n  \u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\u003ctable\u003e\n  \u003cthead\u003e\n  \u003ctr\u003e\n  \u003cth\u003etoc\u003c/th\u003e\n  \u003c/tr\u003e\n  \u003c/thead\u003e\n  \u003ctbody\u003e\n  \u003ctr\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003etrue\u003c/div\u003e\u003c/td\u003e\n  \u003c/tr\u003e\n  \u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\u003ctable\u003e\n  \u003cthead\u003e\n  \u003ctr\u003e\n  \u003cth\u003etoc\u003c/th\u003e\n  \u003c/tr\u003e\n  \u003c/thead\u003e\n  \u003ctbody\u003e\n  \u003ctr\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003etrue\u003c/div\u003e\u003c/td\u003e\n  \u003c/tr\u003e\n  \u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\u003c/td\u003e\n  \u003c/tr\u003e\n  \u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\u003ctable\u003e\n  \u003ctbody\u003e\n  \u003ctr\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\\usepackage{booktabs}\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\\usepackage{longtable}\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\\usepackage{array}\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\\usepackage{multirow}\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\\usepackage[table]{xcolor}\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\\usepackage{wrapfig}\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\\usepackage{float}\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\\usepackage{colortbl}\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\\usepackage{pdflscape}\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\\usepackage{tabu}\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\\usepackage{threeparttable}\u003c/div\u003e\u003c/td\u003e\n  \u003c/tr\u003e\n  \u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003eTimes\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e10pt\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003etrue\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\u003ctable\u003e\n  \u003cthead\u003e\n  \u003ctr\u003e\n  \u003cth\u003emarkdown\u003c/th\u003e\n  \u003cth\u003echunk_output_type\u003c/th\u003e\n  \u003c/tr\u003e\n  \u003c/thead\u003e\n  \u003ctbody\u003e\n  \u003ctr\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e\u003ctable\u003e\n  \u003cthead\u003e\n  \u003ctr\u003e\n  \u003cth\u003ewrap\u003c/th\u003e\n  \u003c/tr\u003e\n  \u003c/thead\u003e\n  \u003ctbody\u003e\n  \u003ctr\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003e72\u003c/div\u003e\u003c/td\u003e\n  \u003c/tr\u003e\n  \u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\u003c/td\u003e\n  \u003ctd\u003e\u003cdiv dir=\"auto\"\u003einline\u003c/div\u003e\u003c/td\u003e\n  \u003c/tr\u003e\n  \u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\u003c/td\u003e\n  \u003c/tr\u003e\n  \u003c/tbody\u003e\n\u003c/table\u003e\n\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eCode Author\u003c/strong\u003e: Mark E. Pepin, MD, PhD, MS \u003cstrong\u003eContact\u003c/strong\u003e:\n\u003ca href=\"mailto:pepinme@gmail.com\"\u003epepinme@gmail.com\u003c/a\u003e{.email} \u003cstrong\u003eInstitution\u003c/strong\u003e:\nHeidelberg University Hospital \u003cstrong\u003eLocation\u003c/strong\u003e: 669 Neuenheimer Feld,\nInstitute for Experimental Cardiology, 69120 Heidelberg, Germany\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePreliminary Setup\u003c/h1\u003e\u003ca id=\"user-content-preliminary-setup\" class=\"anchor\" aria-label=\"Permalink: Preliminary Setup\" href=\"#preliminary-setup\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eParameters\u003c/h2\u003e\u003ca id=\"user-content-parameters\" class=\"anchor\" aria-label=\"Permalink: Parameters\" href=\"#parameters\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDefine the parameters used, along with the conditions required for the\ncurrent analysis. This segment must be modified for each analysis\nperformed.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"##Set the experimental conditions [MUST DO THIS MANUALLY]\nRun_tS\u0026lt;-Sys.time()\nTREATMENT=c(\u0026quot;CON\u0026quot;, \u0026quot;DCM\u0026quot;)\nCELL=c(\u0026quot;iPSC\u0026quot;)\nSTATISTIC = 0.05 #P statistic threshold used in this combination.\nVARIABLE = TREATMENT\nCOMPARISON= paste0(CELL[1], \u0026quot;_\u0026quot;,VARIABLE[2], \u0026quot;_vs_\u0026quot;, VARIABLE[1])\n\n# Candidate Gene Selection (RNA-sequencing) EDIT THIS LIST BASED ON INTERESTS.\nGENES=c(\u0026quot;LEP\u0026quot;, \u0026quot;ADIPOQ\u0026quot;, \u0026quot;PLIN1\u0026quot;)\nVAR1=\u0026quot;Treatment\u0026quot;\nVAR2=\u0026quot;Diabetes\u0026quot;\n# Single Bar Graph\nlibrary(dplyr)\nmy_comparisons \u0026lt;- list( c(\u0026quot;CON\u0026quot;, \u0026quot;DCM\u0026quot;)) # Comparisons to make\n## Create color based on Genotype\nann_colors = list(Treatment = c(DCM=\u0026quot;darkgray\u0026quot;, CON = \u0026quot;darkcyan\u0026quot;))\nTreatmentColors\u0026lt;-ann_colors$Treatment\nann_colorTable\u0026lt;-as.data.frame(ann_colors)\n\nann_colGroup\u0026lt;-subset(ann_colorTable, rownames(ann_colorTable) %in% TREATMENT)\nann_colListGroup\u0026lt;-list(ann_colors$Treatment)\nann_colGroupVec\u0026lt;-ann_colGroup$Treatment\nnames(ann_colGroupVec)\u0026lt;-as.factor(rownames(ann_colGroup))\nGROUP_colors\u0026lt;-list(ann_colGroupVec=ann_colGroupVec)\n\n# Create Output Folder Structure\nifelse(!dir.exists(file.path(paste0(\u0026quot;../2_Output/\u0026quot;))), dir.create(file.path(paste0(\u0026quot;../2_Output/\u0026quot;))), FALSE)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Set the experimental conditions [MUST DO THIS MANUALLY]\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eRun_tS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eSys.time()\n\u003cspan class=\"pl-v\"\u003eTREATMENT\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCON\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDCM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eCELL\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eiPSC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eSTATISTIC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eP statistic threshold used in this combination.\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eVARIABLE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTREATMENT\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eCOMPARISON\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste0(\u003cspan class=\"pl-smi\"\u003eCELL\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e], \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eVARIABLE\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e], \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_vs_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eVARIABLE\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e])\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Candidate Gene Selection (RNA-sequencing) EDIT THIS LIST BASED ON INTERESTS.\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eGENES\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLEP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eADIPOQ\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePLIN1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eVAR1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTreatment\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eVAR2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDiabetes\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Single Bar Graph\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emy_comparisons\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e( c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCON\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDCM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Comparisons to make\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Create color based on Genotype\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eann_colors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eTreatment\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-v\"\u003eDCM\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgray\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCON\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkcyan\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eTreatmentColors\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eann_colors\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eTreatment\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eann_colorTable\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.data.frame(\u003cspan class=\"pl-smi\"\u003eann_colors\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eann_colGroup\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003esubset(\u003cspan class=\"pl-smi\"\u003eann_colorTable\u003c/span\u003e, rownames(\u003cspan class=\"pl-smi\"\u003eann_colorTable\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTREATMENT\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eann_colListGroup\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eann_colors\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eTreatment\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eann_colGroupVec\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eann_colGroup\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eTreatment\u003c/span\u003e\nnames(\u003cspan class=\"pl-smi\"\u003eann_colGroupVec\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.factor(rownames(\u003cspan class=\"pl-smi\"\u003eann_colGroup\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eGROUP_colors\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eann_colGroupVec\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eann_colGroupVec\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Create Output Folder Structure\u003c/span\u003e\nifelse(\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003edir.exists(file.path(paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))), dir.create(file.path(paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))), \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## [1] FALSE\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## [1] FALSE\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"ifelse(!dir.exists(file.path(paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON))), dir.create(file.path(paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON))), FALSE)\"\u003e\u003cpre\u003eifelse(\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003edir.exists(file.path(paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e))), dir.create(file.path(paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e))), \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## [1] FALSE\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## [1] FALSE\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePackages\u003c/h2\u003e\u003ca id=\"user-content-packages\" class=\"anchor\" aria-label=\"Permalink: Packages\" href=\"#packages\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"if (!require(\u0026quot;pacman\u0026quot;)) install.packages(\u0026quot;pacman\u0026quot;)\npacman::p_load(dplyr, Hmisc, openxlsx, corrplot, RColorBrewer, kableExtra, ggplot2, gridExtra, ggpubr, ggsignif, DESeq2, data.table, GenomicFeatures, biomaRt, Haplin, pheatmap, calibrate, ggrepel, tidyr, gtools)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e (\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003erequire(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epacman\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) install.packages(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epacman\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-e\"\u003epacman\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003ep_load(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eHmisc\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eopenxlsx\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ecorrplot\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eRColorBrewer\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ekableExtra\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eggplot2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egridExtra\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eggpubr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eggsignif\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eDESeq2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eGenomicFeatures\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ebiomaRt\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eHaplin\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epheatmap\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ecalibrate\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eggrepel\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etidyr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egtools\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGenome-wide DNA methylation - Illumina(R) HumanMethylation 450k Methylation\u003c/h1\u003e\u003ca id=\"user-content-genome-wide-dna-methylation---illuminar-humanmethylation-450k-methylation\" class=\"anchor\" aria-label=\"Permalink: Genome-wide DNA methylation - Illumina(R) HumanMethylation 450k Methylation\" href=\"#genome-wide-dna-methylation---illuminar-humanmethylation-450k-methylation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(minfi)\nlibrary(limma)\nlibrary(shinyMethyl)\nlibrary(dplyr)\nlibrary(IlluminaHumanMethylation450kmanifest)\nlibrary(IlluminaHumanMethylation450kanno.ilmn12.hg19)\nlibrary(RColorBrewer)\n#########Part 1: Importing the Data\n#Parameters\n##Get the array annotation\nannoM450k\u0026lt;-getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19)\nannoM450k\u0026lt;-dplyr::select(as.data.frame(annoM450k), Name, chr, pos, Relation_to_Island, UCSC_RefGene_Name, UCSC_RefGene_Accession, UCSC_RefGene_Group, Regulatory_Feature_Group)\n#Import the sample sheet\ntargets\u0026lt;-read.metharray.sheet(base=\u0026quot;../1_Input/IDAT\u0026quot;, pattern=\u0026quot;.csv\u0026quot;)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eminfi\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003elimma\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eshinyMethyl\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eIlluminaHumanMethylation450kmanifest\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eIlluminaHumanMethylation450kanno.ilmn12.hg19\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eRColorBrewer\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e########Part 1: Importing the Data\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eParameters\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Get the array annotation\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eannoM450k\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003egetAnnotation(\u003cspan class=\"pl-smi\"\u003eIlluminaHumanMethylation450kanno.ilmn12.hg19\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eannoM450k\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(as.data.frame(\u003cspan class=\"pl-smi\"\u003eannoM450k\u003c/span\u003e), \u003cspan class=\"pl-smi\"\u003eName\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epos\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eRelation_to_Island\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Accession\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eRegulatory_Feature_Group\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eImport the sample sheet\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eread.metharray.sheet(\u003cspan class=\"pl-v\"\u003ebase\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../1_Input/IDAT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epattern\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## [1] \u0026quot;../1_Input/IDAT/SampleSheet.csv\u0026quot;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## [1] \"../1_Input/IDAT/SampleSheet.csv\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" #Import the annotation file\nAnnoTargets\u0026lt;-targets\ntargets$ID \u0026lt;- paste(targets$Sample_Group,targets$Sample_Name,sep=\u0026quot;.\u0026quot;)\ntargets$Sample_Well\u0026lt;-AnnoTargets$Sample_Well\ntargets$Sample_Plate\u0026lt;-AnnoTargets$Sample_Plate\ntargets$Sample_Group\u0026lt;-AnnoTargets$Sample_Group\ntargets$Sample_Group\u0026lt;-factor(targets$Sample_Group, levels = c(\u0026quot;CON\u0026quot;, \u0026quot;DCM\u0026quot;))\ntargets$Sex\u0026lt;-AnnoTargets$Sex\ntargets$Outcome\u0026lt;-AnnoTargets$Outcome\ntargets$Conc\u0026lt;-AnnoTargets$Conc\ntargets$Col_ID\u0026lt;-AnnoTargets$Col_ID\ntargets$Tissue\u0026lt;-AnnoTargets$Tissue\ntargets\u0026lt;-targets %\u0026gt;% filter(Tissue %in% CELL)\n\n#Import the array data from input directory (red and green .idat files)\nRGSet\u0026lt;-read.metharray.exp(base = \u0026quot;../1_Input/IDAT\u0026quot;, targets = targets, verbose = TRUE)\nsampleNames(RGSet)\u0026lt;-targets$Sample_Name\n##Quality Control\n#First step is to identify CpGs that failed to identify methylated positions (defined by expression intensity that reflects background levels)\ndetP\u0026lt;-detectionP(RGSet)\nPLOT.COL \u0026lt;- brewer.pal(8,\u0026quot;Dark2\u0026quot;)\n##Detection P-value Plot\npdf(file=paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_detectionP.pdf\u0026quot;))\npar(mfrow=c(1,1))\nbarplot(colMeans(detP), col=PLOT.COL[factor(targets$Sample_Group)], \n        cex.names=0.8, ylim=c(0,0.005), ylab=\u0026quot;Mean detection p-values\u0026quot;)\nabline(h=0.05,col=\u0026quot;red\u0026quot;)\nlegend(\u0026quot;topleft\u0026quot;, legend=levels(factor(targets$Sample_Group)), fill=PLOT.COL, \n       bg=\u0026quot;white\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eImport the annotation file\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eID\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e paste(\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Well\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Well\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Plate\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Plate\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elevels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCON\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDCM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSex\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOutcome\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eOutcome\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eConc\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eConc\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCol_ID\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCol_ID\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eTissue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eTissue\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% filter(\u003cspan class=\"pl-smi\"\u003eTissue\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCELL\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eImport the array data from input directory (red and green .idat files)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eread.metharray.exp(\u003cspan class=\"pl-v\"\u003ebase\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../1_Input/IDAT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etargets\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\nsampleNames(\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Quality Control\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eFirst step is to identify CpGs that failed to identify methylated positions (defined by expression intensity that reflects background levels)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003edetP\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003edetectionP(\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ePLOT.COL\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e brewer.pal(\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDark2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Detection P-value Plot\u003c/span\u003e\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_detectionP.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\npar(\u003cspan class=\"pl-v\"\u003emfrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e))\nbarplot(colMeans(\u003cspan class=\"pl-smi\"\u003edetP\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePLOT.COL\u003c/span\u003e[\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e)], \n        \u003cspan class=\"pl-v\"\u003ecex.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.005\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMean detection p-values\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-v\"\u003eh\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlegend(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopleft\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elegend\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elevels(\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e)), \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePLOT.COL\u003c/span\u003e, \n       \u003cspan class=\"pl-v\"\u003ebg\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## quartz_off_screen \n##                 2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## quartz_off_screen \n##                 2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## Filter RGSet by detection P-value\nkeep \u0026lt;- colMeans(detP) \u0026lt; 0.05\nRGSet \u0026lt;- RGSet[,keep]\nRGSet\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Filter RGSet by detection P-value\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ekeep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colMeans(\u003cspan class=\"pl-smi\"\u003edetP\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e[,\u003cspan class=\"pl-smi\"\u003ekeep\u003c/span\u003e]\n\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## class: RGChannelSet \n## dim: 622399 24 \n## metadata(0):\n## assays(2): Green Red\n## rownames(622399): 10600313 10600322 ... 74810490 74810492\n## rowData names(0):\n## colnames(24): A1_cells_bo A2_cells_bo ... C9_cells_co C10_cells_co\n## colData names(17): Sample_Name Sample_Well ... ID filenames\n## Annotation\n##   array: IlluminaHumanMethylation450k\n##   annotation: ilmn12.hg19\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## class: RGChannelSet \n## dim: 622399 24 \n## metadata(0):\n## assays(2): Green Red\n## rownames(622399): 10600313 10600322 ... 74810490 74810492\n## rowData names(0):\n## colnames(24): A1_cells_bo A2_cells_bo ... C9_cells_co C10_cells_co\n## colData names(17): Sample_Name Sample_Well ... ID filenames\n## Annotation\n##   array: IlluminaHumanMethylation450k\n##   annotation: ilmn12.hg19\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"##QC Report\nqcReport(RGSet, sampNames=targets$ID, sampGroups=targets$Sample_Group, pdf=\u0026quot;qcReport.pdf\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#QC Report\u003c/span\u003e\nqcReport(\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esampNames\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eID\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esampGroups\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epdf\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eqcReport.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## quartz_off_screen \n##                 2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## quartz_off_screen \n##                 2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#\npar(mfrow=c(1,1))\nbarplot(colMeans(detP), col=PLOT.COL[factor(targets$Sample_Group)], las=2, \n        cex.names=0.8, ylim=c(0,0.005), ylab=\u0026quot;Mean detection p-values\u0026quot;)\nabline(h=0.05,col=\u0026quot;red\u0026quot;)\nlegend(\u0026quot;topleft\u0026quot;, legend=levels(factor(targets$Sample_Group)), fill=PLOT.COL, \n       bg=\u0026quot;white\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\npar(\u003cspan class=\"pl-v\"\u003emfrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e))\nbarplot(colMeans(\u003cspan class=\"pl-smi\"\u003edetP\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePLOT.COL\u003c/span\u003e[\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e)], \u003cspan class=\"pl-v\"\u003elas\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003ecex.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.005\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMean detection p-values\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-v\"\u003eh\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlegend(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopleft\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elegend\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elevels(\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e)), \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePLOT.COL\u003c/span\u003e, \n       \u003cspan class=\"pl-v\"\u003ebg\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/EPIC_Import-1.png\"\u003e\u003cimg src=\"/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/EPIC_Import-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# #Determine the fraction of \u0026quot;failed\u0026quot; CpG probes (those which failed to identify a methylated CpG)\n# colMeans(failed)\n#Convert R/G to Methylated/Unmethylated in an object of class MethylSet\nMSet\u0026lt;-preprocessRaw(RGSet)\n#QC data\nqc\u0026lt;-getQC(MSet)\nplotQC(qc)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e #Determine the fraction of \"failed\" CpG probes (those which failed to identify a methylated CpG)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e colMeans(failed)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eConvert R/G to Methylated/Unmethylated in an object of class MethylSet\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eMSet\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003epreprocessRaw(\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eQC data\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eqc\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003egetQC(\u003cspan class=\"pl-smi\"\u003eMSet\u003c/span\u003e)\nplotQC(\u003cspan class=\"pl-smi\"\u003eqc\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/EPIC_Import-2.png\"\u003e\u003cimg src=\"/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/EPIC_Import-2.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"pdf(file=paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_QC.Methyl.pdf\u0026quot;))\nplotQC(qc)\ndev.off()\"\u003e\u003cpre\u003epdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_QC.Methyl.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nplotQC(\u003cspan class=\"pl-smi\"\u003eqc\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## quartz_off_screen \n##                 2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## quartz_off_screen \n##                 2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"##Density plot\n# densityPlot(RGSet, sampGroups = targets$Sample_Group, main= \u0026quot;Beta\u0026quot;, xlab = \u0026quot;Beta\u0026quot;)\npdf(file=paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_densityPlot.pdf\u0026quot;))\ndensityPlot(RGSet, sampGroups = targets$Sample_Group, main= \u0026quot;Beta\u0026quot;, xlab = \u0026quot;Beta\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Density plot\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e densityPlot(RGSet, sampGroups = targets$Sample_Group, main= \"Beta\", xlab = \"Beta\")\u003c/span\u003e\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_densityPlot.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\ndensityPlot(\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esampGroups\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emain\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBeta\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBeta\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## quartz_off_screen \n##                 2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## quartz_off_screen \n##                 2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"densityBeanPlot(RGSet, sampGroups = targets$Sample_Group)\"\u003e\u003cpre\u003edensityBeanPlot(\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esampGroups\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/EPIC_Import-3.png\"\u003e\u003cimg src=\"/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/EPIC_Import-3.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"pdf(file=paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_BeanPlot.pdf\u0026quot;))\ndensityBeanPlot(RGSet, sampGroups = targets$Sample_Group)\ndev.off()\"\u003e\u003cpre\u003epdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_BeanPlot.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\ndensityBeanPlot(\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esampGroups\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## quartz_off_screen \n##                 2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## quartz_off_screen \n##                 2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"qcReport(RGSet, pdf= paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_qcReport.pdf\u0026quot;))\"\u003e\u003cpre\u003eqcReport(\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epdf\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_qcReport.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## quartz_off_screen \n##                 2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## quartz_off_screen \n##                 2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# #Convert to a shinyMethyl dataset (nice summary of the quality control)\n# summarized.data \u0026lt;- shinySummarize(RGSet)\n# runShinyMethyl(summarized.data)\n\ngRatioSet.quantile \u0026lt;- preprocessQuantile(RGSet, fixOutliers = TRUE, removeBadSamples = TRUE, badSampleCutoff = 10.5, quantileNormalize = TRUE, stratified = TRUE, mergeManifest = FALSE)\nbeta.all\u0026lt;-getBeta(RGSet)\nwrite.csv(beta.all, \u0026quot;../1_Input/EPIC.betaValues.csv\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e #Convert to a shinyMethyl dataset (nice summary of the quality control)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e summarized.data \u0026lt;- shinySummarize(RGSet)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e runShinyMethyl(summarized.data)\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003egRatioSet.quantile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e preprocessQuantile(\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efixOutliers\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eremoveBadSamples\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ebadSampleCutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003equantileNormalize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estratified\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emergeManifest\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ebeta.all\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003egetBeta(\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e)\nwrite.csv(\u003cspan class=\"pl-smi\"\u003ebeta.all\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../1_Input/EPIC.betaValues.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSupplemental Figure SXX: Outlier Analysis\u003c/h2\u003e\u003ca id=\"user-content-supplemental-figure-sxx-outlier-analysis\" class=\"anchor\" aria-label=\"Permalink: Supplemental Figure SXX: Outlier Analysis\" href=\"#supplemental-figure-sxx-outlier-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"par(mfrow=c(1,1))\nbarplot(colMeans(detP), col=PLOT.COL[factor(targets$Sample_Group)], las=2, \n        cex.names=0.8, ylim=c(0,0.005), ylab=\u0026quot;Mean detection p-values\u0026quot;)\nabline(h=0.05,col=\u0026quot;red\u0026quot;)\nlegend(\u0026quot;topleft\u0026quot;, legend=levels(factor(targets$Sample_Group)), fill=PLOT.COL, \n       bg=\u0026quot;white\u0026quot;)\"\u003e\u003cpre\u003epar(\u003cspan class=\"pl-v\"\u003emfrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e))\nbarplot(colMeans(\u003cspan class=\"pl-smi\"\u003edetP\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePLOT.COL\u003c/span\u003e[\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e)], \u003cspan class=\"pl-v\"\u003elas\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003ecex.names\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.005\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMean detection p-values\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nabline(\u003cspan class=\"pl-v\"\u003eh\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlegend(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopleft\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elegend\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003elevels(\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e)), \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePLOT.COL\u003c/span\u003e, \n       \u003cspan class=\"pl-v\"\u003ebg\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/outliers-1.png\"\u003e\u003cimg src=\"/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/outliers-1.png\" alt=\"Supplemental Figure\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFigure 2A: MDS Plot - Unsupervised Clustering by \"Sample_Group\"\u003c/h2\u003e\u003ca id=\"user-content-figure-2a-mds-plot---unsupervised-clustering-by-sample_group\" class=\"anchor\" aria-label=\"Permalink: Figure 2A: MDS Plot - Unsupervised Clustering by \u0026quot;Sample_Group\u0026quot;\" href=\"#figure-2a-mds-plot---unsupervised-clustering-by-sample_group\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBecause such a robust racial signature in cardiac DNA methylation was\nseen in the pilot analysis, we reproduced the unsupervised method in the\nlarger cohort. This time, we continue to see a distinct racial\ndifference. Furthermore, we found that this racially-determined\nclustering persisted to among the 500,000 most-variable CpG probes in\nthe EPIC array.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/MDS.methyl-1.png\"\u003e\u003cimg src=\"/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/MDS.methyl-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePCA\u003c/h2\u003e\u003ca id=\"user-content-pca\" class=\"anchor\" aria-label=\"Permalink: PCA\" href=\"#pca\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#Plot Features of the PCA\nlibrary(dplyr)\nlibrary(plotly)\n##Import the data to be used for PCA\nIndex_PCA\u0026lt;-targets\nrownames(Index_PCA)\u0026lt;-Index_PCA$Sample_Name\nPCA_data\u0026lt;-as.data.frame(getM(gRatioSet.quantile))\n#transpose the dataset (required for PCA)\ndata.pca\u0026lt;-t(PCA_data)\ndata.pca\u0026lt;-as.data.frame(data.pca)\n##merge the file\ndata.pca_Final\u0026lt;-merge(Index_PCA, data.pca, by=0)\nrownames(data.pca_Final)\u0026lt;-data.pca_Final$Row.names\npca.comp\u0026lt;-prcomp(data.pca_Final[,(ncol(Index_PCA)+2):ncol(data.pca_Final)])\n\npcaCharts=function(x) {\n    x.var \u0026lt;- x$sdev ^ 2\n    x.pvar \u0026lt;- x.var/sum(x.var)\n    par(mfrow=c(2,2))\n    plot(x.pvar,xlab=\u0026quot;Principal component\u0026quot;,\n         ylab=\u0026quot;Proportion of variance\u0026quot;, ylim=c(0,1), type='b')\n    plot(cumsum(x.pvar),xlab=\u0026quot;Principal component\u0026quot;,\n         ylab=\u0026quot;Cumulative Proportion of variance\u0026quot;,\n         ylim=c(0,1),\n         type='b')\n    screeplot(x)\n    screeplot(x,type=\u0026quot;l\u0026quot;)\n    par(mfrow=c(1,1))\n}\npcaCharts(pca.comp)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ePlot Features of the PCA\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eplotly\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Import the data to be used for PCA\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eIndex_PCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\nrownames(\u003cspan class=\"pl-smi\"\u003eIndex_PCA\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eIndex_PCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ePCA_data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.data.frame(getM(\u003cspan class=\"pl-smi\"\u003egRatioSet.quantile\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etranspose the dataset (required for PCA)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003edata.pca\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003et(\u003cspan class=\"pl-smi\"\u003ePCA_data\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003edata.pca\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.data.frame(\u003cspan class=\"pl-smi\"\u003edata.pca\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#merge the file\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003edata.pca_Final\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003emerge(\u003cspan class=\"pl-smi\"\u003eIndex_PCA\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edata.pca\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e)\nrownames(\u003cspan class=\"pl-smi\"\u003edata.pca_Final\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edata.pca_Final\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRow.names\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003epca.comp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eprcomp(\u003cspan class=\"pl-smi\"\u003edata.pca_Final\u003c/span\u003e[,(ncol(\u003cspan class=\"pl-smi\"\u003eIndex_PCA\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003encol(\u003cspan class=\"pl-smi\"\u003edata.pca_Final\u003c/span\u003e)])\n\n\u003cspan class=\"pl-en\"\u003epcaCharts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e) {\n    \u003cspan class=\"pl-smi\"\u003ex.var\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esdev\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e^\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\n    \u003cspan class=\"pl-smi\"\u003ex.pvar\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ex.var\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003esum(\u003cspan class=\"pl-smi\"\u003ex.var\u003c/span\u003e)\n    par(\u003cspan class=\"pl-v\"\u003emfrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e))\n    plot(\u003cspan class=\"pl-smi\"\u003ex.pvar\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrincipal component\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n         \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eProportion of variance\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eb\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n    plot(cumsum(\u003cspan class=\"pl-smi\"\u003ex.pvar\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003exlab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrincipal component\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n         \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCumulative Proportion of variance\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n         \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e),\n         \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eb\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n    screeplot(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e)\n    screeplot(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003el\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    par(\u003cspan class=\"pl-v\"\u003emfrow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e))\n}\npcaCharts(\u003cspan class=\"pl-smi\"\u003epca.comp\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/PCA_Features-1.png\"\u003e\u003cimg src=\"/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/PCA_Features-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"png(file=paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON,  \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_PCA.Charts.png\u0026quot;))\npcaCharts(pca.comp)\ndev.off()\"\u003e\u003cpre\u003epng(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003epaste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e,  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_PCA.Charts.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\npcaCharts(\u003cspan class=\"pl-smi\"\u003epca.comp\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## quartz_off_screen \n##                 2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## quartz_off_screen \n##                 2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3-Dimensional PCA\u003c/h3\u003e\u003ca id=\"user-content-3-dimensional-pca\" class=\"anchor\" aria-label=\"Permalink: 3-Dimensional PCA\" href=\"#3-dimensional-pca\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFrom the previous calculations, it is seens that only 2 principal\ncomponents are necessary (accounting for \u0026gt;80% cumulative variance).\nNonetheless, below is a 3-D PCA to ensure that all groups are\ncharacterize to higher-degree of stringency. Nevertheless, a racial\ndifference could not be appreciated.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"##Create a 3D-PCA for Inspection\nlibrary(plotly)\n##Index\nIndex_PCA\u0026lt;-targets\nrownames(Index_PCA)\u0026lt;-Index_PCA$Sample_Name\n\nPCs\u0026lt;-merge(pca.comp$x, Index_PCA, by=0)\nrownames(PCs)\u0026lt;-PCs$Row.names\nPCs$Group \u0026lt;- as.factor(PCs$Sample_Group)\nfig \u0026lt;- plot_ly(PCs, x = ~PC1, y = ~PC2, z = ~PC3, color = ~Sample_Group, text = ~paste('Sample_Name:', Sample_Name, '\u0026lt;br\u0026gt;Tissue:', Tissue, '\u0026lt;br\u0026gt;Outcome:', Outcome, '\u0026lt;br\u0026gt;Sex:', Sex))\nfig \u0026lt;- fig %\u0026gt;% add_markers()\nfig \u0026lt;- fig %\u0026gt;% layout(scene = list(xaxis = list(title = 'PC1'),\n                     yaxis = list(title = 'PC2'),\n                     zaxis = list(title = 'PC3')))\nfig\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Create a 3D-PCA for Inspection\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003eplotly\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Index\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eIndex_PCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\nrownames(\u003cspan class=\"pl-smi\"\u003eIndex_PCA\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eIndex_PCA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003ePCs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003emerge(\u003cspan class=\"pl-smi\"\u003epca.comp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eIndex_PCA\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e)\nrownames(\u003cspan class=\"pl-smi\"\u003ePCs\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePCs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRow.names\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ePCs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eGroup\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.factor(\u003cspan class=\"pl-smi\"\u003ePCs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003efig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e plot_ly(\u003cspan class=\"pl-smi\"\u003ePCs\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePC1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePC2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ez\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePC3\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etext\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003epaste(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSample_Name:\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026lt;br\u0026gt;Tissue:\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eTissue\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026lt;br\u0026gt;Outcome:\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eOutcome\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026lt;br\u0026gt;Sex:\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSex\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003efig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efig\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% add_markers()\n\u003cspan class=\"pl-smi\"\u003efig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efig\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% layout(\u003cspan class=\"pl-v\"\u003escene\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003exaxis\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePC1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\n                     \u003cspan class=\"pl-v\"\u003eyaxis\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePC2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\n                     \u003cspan class=\"pl-v\"\u003ezaxis\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePC3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)))\n\u003cspan class=\"pl-smi\"\u003efig\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\u0026lt;div id=\u0026quot;htmlwidget-0ed3cff61cab74c0c287\u0026quot; style=\u0026quot;width:672px;height:480px;\u0026quot; class=\u0026quot;plotly html-widget\u0026quot;\u0026gt;\u0026lt;/div\u0026gt;\n\u0026lt;script type=\u0026quot;application/json\u0026quot; data-for=\u0026quot;htmlwidget-0ed3cff61cab74c0c287\u0026quot;\u0026gt;{\u0026quot;x\u0026quot;:{\u0026quot;visdat\u0026quot;:{\u0026quot;2ee72d1f4380\u0026quot;:[\u0026quot;function () \u0026quot;,\u0026quot;plotlyVisDat\u0026quot;]},\u0026quot;cur_data\u0026quot;:\u0026quot;2ee72d1f4380\u0026quot;,\u0026quot;attrs\u0026quot;:{\u0026quot;2ee72d1f4380\u0026quot;:{\u0026quot;x\u0026quot;:{},\u0026quot;y\u0026quot;:{},\u0026quot;z\u0026quot;:{},\u0026quot;text\u0026quot;:{},\u0026quot;color\u0026quot;:{},\u0026quot;alpha_stroke\u0026quot;:1,\u0026quot;sizes\u0026quot;:[10,100],\u0026quot;spans\u0026quot;:[1,20],\u0026quot;type\u0026quot;:\u0026quot;scatter3d\u0026quot;,\u0026quot;mode\u0026quot;:\u0026quot;markers\u0026quot;,\u0026quot;inherit\u0026quot;:true}},\u0026quot;layout\u0026quot;:{\u0026quot;margin\u0026quot;:{\u0026quot;b\u0026quot;:40,\u0026quot;l\u0026quot;:60,\u0026quot;t\u0026quot;:25,\u0026quot;r\u0026quot;:10},\u0026quot;scene\u0026quot;:{\u0026quot;xaxis\u0026quot;:{\u0026quot;title\u0026quot;:\u0026quot;PC1\u0026quot;},\u0026quot;yaxis\u0026quot;:{\u0026quot;title\u0026quot;:\u0026quot;PC2\u0026quot;},\u0026quot;zaxis\u0026quot;:{\u0026quot;title\u0026quot;:\u0026quot;PC3\u0026quot;}},\u0026quot;hovermode\u0026quot;:\u0026quot;closest\u0026quot;,\u0026quot;showlegend\u0026quot;:true},\u0026quot;source\u0026quot;:\u0026quot;A\u0026quot;,\u0026quot;config\u0026quot;:{\u0026quot;modeBarButtonsToAdd\u0026quot;:[\u0026quot;hoverclosest\u0026quot;,\u0026quot;hovercompare\u0026quot;],\u0026quot;showSendToCloud\u0026quot;:false},\u0026quot;data\u0026quot;:[{\u0026quot;x\u0026quot;:[77.9600791014893,-8.85871319983235,-2.3258763441347,-7.91557474476333,-5.12544124055454,-6.31032622733963,-15.5555084612836,19.1451274289372,-0.370810911338616,-7.2301707948147],\u0026quot;y\u0026quot;:[63.3739250059267,-28.5775366493097,-38.9831271081527,-26.5114827922627,-26.5322218237842,-24.6351203021513,-25.5921588263502,19.6910128735114,-34.7694530835565,-19.4396082462673],\u0026quot;z\u0026quot;:[131.113571695425,-7.98668393195755,-25.645441514114,-14.9352103367056,-7.8256237104918,-3.54404886501356,-13.3068841181743,19.0115820730706,-26.0649079763584,-9.98096383837802],\u0026quot;text\u0026quot;:[\u0026quot;Sample_Name: C1_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: C10_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: F\u0026quot;,\u0026quot;Sample_Name: C2_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: C3_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: F\u0026quot;,\u0026quot;Sample_Name: C4_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: C5_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: C6_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: C7_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: C8_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: F\u0026quot;,\u0026quot;Sample_Name: C9_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: F\u0026quot;],\u0026quot;type\u0026quot;:\u0026quot;scatter3d\u0026quot;,\u0026quot;mode\u0026quot;:\u0026quot;markers\u0026quot;,\u0026quot;name\u0026quot;:\u0026quot;CON\u0026quot;,\u0026quot;marker\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(102,194,165,1)\u0026quot;,\u0026quot;line\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(102,194,165,1)\u0026quot;}},\u0026quot;textfont\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(102,194,165,1)\u0026quot;},\u0026quot;error_y\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(102,194,165,1)\u0026quot;},\u0026quot;error_x\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(102,194,165,1)\u0026quot;},\u0026quot;line\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(102,194,165,1)\u0026quot;},\u0026quot;frame\u0026quot;:null},{\u0026quot;x\u0026quot;:[-74.840661246498,-72.6476533955918,-59.4408574966577,-13.7136537072147,-4.21579120231104,6.11119549549595,9.59163470727785,5.85414321742238,173.476095054373,-15.3097146431571,-13.7765637170266,0.319854833899074,6.97727589132507,8.20191160229823],\u0026quot;y\u0026quot;:[66.6389631010503,89.1024717499668,120.443036300283,-28.4395221216847,-22.325597162093,-26.7115831833687,-22.6504061481375,-15.1207211276863,64.8545317043899,-25.7647846869283,-27.3850982890718,-12.2567307943521,-18.5256954472338,0.116907057264034],\u0026quot;z\u0026quot;:[-23.0507887812109,-16.1327461317539,-15.3423208102994,-3.59376989204799,1.84807232198693,12.2145192980728,17.1138420638432,14.8393139305453,-91.6382517593327,-8.60068453610674,0.447886121276813,18.7299603053965,20.7702635178944,31.5593148744268],\u0026quot;text\u0026quot;:[\u0026quot;Sample_Name: A1_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: A2_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: A3_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: A4_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: A5_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: A6_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: A7_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: F\u0026quot;,\u0026quot;Sample_Name: B1_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: B2_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: B3_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: B4_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: B5_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\u0026quot;,\u0026quot;Sample_Name: B6_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: F\u0026quot;,\u0026quot;Sample_Name: B7_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\u0026quot;],\u0026quot;type\u0026quot;:\u0026quot;scatter3d\u0026quot;,\u0026quot;mode\u0026quot;:\u0026quot;markers\u0026quot;,\u0026quot;name\u0026quot;:\u0026quot;DCM\u0026quot;,\u0026quot;marker\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(141,160,203,1)\u0026quot;,\u0026quot;line\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(141,160,203,1)\u0026quot;}},\u0026quot;textfont\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(141,160,203,1)\u0026quot;},\u0026quot;error_y\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(141,160,203,1)\u0026quot;},\u0026quot;error_x\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(141,160,203,1)\u0026quot;},\u0026quot;line\u0026quot;:{\u0026quot;color\u0026quot;:\u0026quot;rgba(141,160,203,1)\u0026quot;},\u0026quot;frame\u0026quot;:null}],\u0026quot;highlight\u0026quot;:{\u0026quot;on\u0026quot;:\u0026quot;plotly_click\u0026quot;,\u0026quot;persistent\u0026quot;:false,\u0026quot;dynamic\u0026quot;:false,\u0026quot;selectize\u0026quot;:false,\u0026quot;opacityDim\u0026quot;:0.2,\u0026quot;selected\u0026quot;:{\u0026quot;opacity\u0026quot;:1},\u0026quot;debounce\u0026quot;:0},\u0026quot;shinyEvents\u0026quot;:[\u0026quot;plotly_hover\u0026quot;,\u0026quot;plotly_click\u0026quot;,\u0026quot;plotly_selected\u0026quot;,\u0026quot;plotly_relayout\u0026quot;,\u0026quot;plotly_brushed\u0026quot;,\u0026quot;plotly_brushing\u0026quot;,\u0026quot;plotly_clickannotation\u0026quot;,\u0026quot;plotly_doubleclick\u0026quot;,\u0026quot;plotly_deselect\u0026quot;,\u0026quot;plotly_afterplot\u0026quot;,\u0026quot;plotly_sunburstclick\u0026quot;],\u0026quot;base_url\u0026quot;:\u0026quot;https://plot.ly\u0026quot;},\u0026quot;evals\u0026quot;:[],\u0026quot;jsHooks\u0026quot;:[]}\u0026lt;/script\u0026gt;\"\u003e\u003cpre lang=\"{=html}\" class=\"notranslate\"\u003e\u003ccode\u003e\u0026lt;div id=\"htmlwidget-0ed3cff61cab74c0c287\" style=\"width:672px;height:480px;\" class=\"plotly html-widget\"\u0026gt;\u0026lt;/div\u0026gt;\n\u0026lt;script type=\"application/json\" data-for=\"htmlwidget-0ed3cff61cab74c0c287\"\u0026gt;{\"x\":{\"visdat\":{\"2ee72d1f4380\":[\"function () \",\"plotlyVisDat\"]},\"cur_data\":\"2ee72d1f4380\",\"attrs\":{\"2ee72d1f4380\":{\"x\":{},\"y\":{},\"z\":{},\"text\":{},\"color\":{},\"alpha_stroke\":1,\"sizes\":[10,100],\"spans\":[1,20],\"type\":\"scatter3d\",\"mode\":\"markers\",\"inherit\":true}},\"layout\":{\"margin\":{\"b\":40,\"l\":60,\"t\":25,\"r\":10},\"scene\":{\"xaxis\":{\"title\":\"PC1\"},\"yaxis\":{\"title\":\"PC2\"},\"zaxis\":{\"title\":\"PC3\"}},\"hovermode\":\"closest\",\"showlegend\":true},\"source\":\"A\",\"config\":{\"modeBarButtonsToAdd\":[\"hoverclosest\",\"hovercompare\"],\"showSendToCloud\":false},\"data\":[{\"x\":[77.9600791014893,-8.85871319983235,-2.3258763441347,-7.91557474476333,-5.12544124055454,-6.31032622733963,-15.5555084612836,19.1451274289372,-0.370810911338616,-7.2301707948147],\"y\":[63.3739250059267,-28.5775366493097,-38.9831271081527,-26.5114827922627,-26.5322218237842,-24.6351203021513,-25.5921588263502,19.6910128735114,-34.7694530835565,-19.4396082462673],\"z\":[131.113571695425,-7.98668393195755,-25.645441514114,-14.9352103367056,-7.8256237104918,-3.54404886501356,-13.3068841181743,19.0115820730706,-26.0649079763584,-9.98096383837802],\"text\":[\"Sample_Name: C1_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: C10_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: F\",\"Sample_Name: C2_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: C3_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: F\",\"Sample_Name: C4_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: C5_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: C6_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: C7_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: C8_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: F\",\"Sample_Name: C9_cells_co \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: CON \u0026lt;br\u0026gt;Sex: F\"],\"type\":\"scatter3d\",\"mode\":\"markers\",\"name\":\"CON\",\"marker\":{\"color\":\"rgba(102,194,165,1)\",\"line\":{\"color\":\"rgba(102,194,165,1)\"}},\"textfont\":{\"color\":\"rgba(102,194,165,1)\"},\"error_y\":{\"color\":\"rgba(102,194,165,1)\"},\"error_x\":{\"color\":\"rgba(102,194,165,1)\"},\"line\":{\"color\":\"rgba(102,194,165,1)\"},\"frame\":null},{\"x\":[-74.840661246498,-72.6476533955918,-59.4408574966577,-13.7136537072147,-4.21579120231104,6.11119549549595,9.59163470727785,5.85414321742238,173.476095054373,-15.3097146431571,-13.7765637170266,0.319854833899074,6.97727589132507,8.20191160229823],\"y\":[66.6389631010503,89.1024717499668,120.443036300283,-28.4395221216847,-22.325597162093,-26.7115831833687,-22.6504061481375,-15.1207211276863,64.8545317043899,-25.7647846869283,-27.3850982890718,-12.2567307943521,-18.5256954472338,0.116907057264034],\"z\":[-23.0507887812109,-16.1327461317539,-15.3423208102994,-3.59376989204799,1.84807232198693,12.2145192980728,17.1138420638432,14.8393139305453,-91.6382517593327,-8.60068453610674,0.447886121276813,18.7299603053965,20.7702635178944,31.5593148744268],\"text\":[\"Sample_Name: A1_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: A2_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: A3_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: A4_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: A5_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: A6_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: A7_cells_bo \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Bad \u0026lt;br\u0026gt;Sex: F\",\"Sample_Name: B1_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: B2_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: B3_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: B4_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: B5_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\",\"Sample_Name: B6_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: F\",\"Sample_Name: B7_cells_go \u0026lt;br\u0026gt;Tissue: iPSC \u0026lt;br\u0026gt;Outcome: Good \u0026lt;br\u0026gt;Sex: M\"],\"type\":\"scatter3d\",\"mode\":\"markers\",\"name\":\"DCM\",\"marker\":{\"color\":\"rgba(141,160,203,1)\",\"line\":{\"color\":\"rgba(141,160,203,1)\"}},\"textfont\":{\"color\":\"rgba(141,160,203,1)\"},\"error_y\":{\"color\":\"rgba(141,160,203,1)\"},\"error_x\":{\"color\":\"rgba(141,160,203,1)\"},\"line\":{\"color\":\"rgba(141,160,203,1)\"},\"frame\":null}],\"highlight\":{\"on\":\"plotly_click\",\"persistent\":false,\"dynamic\":false,\"selectize\":false,\"opacityDim\":0.2,\"selected\":{\"opacity\":1},\"debounce\":0},\"shinyEvents\":[\"plotly_hover\",\"plotly_click\",\"plotly_selected\",\"plotly_relayout\",\"plotly_brushed\",\"plotly_brushing\",\"plotly_clickannotation\",\"plotly_doubleclick\",\"plotly_deselect\",\"plotly_afterplot\",\"plotly_sunburstclick\"],\"base_url\":\"https://plot.ly\"},\"evals\":[],\"jsHooks\":[]}\u0026lt;/script\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDifferential Methylation of iPSCs (DCM vs. CON)\u003c/h2\u003e\u003ca id=\"user-content-differential-methylation-of-ipscs-dcm-vs-con\" class=\"anchor\" aria-label=\"Permalink: Differential Methylation of iPSCs (DCM vs. CON)\" href=\"#differential-methylation-of-ipscs-dcm-vs-con\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eUsing Minfi to compute differential expression\u003c/h2\u003e\u003ca id=\"user-content-using-minfi-to-compute-differential-expression\" class=\"anchor\" aria-label=\"Permalink: Using Minfi to compute differential expression\" href=\"#using-minfi-to-compute-differential-expression\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOwing to the apparent separation by patient Sample_Group, we chose to\nidentify the CpG sites responsible for a Sample_Group-based epigenomic\ndifference.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#Quantification and Differential Expression Analysis\nRGSet\u0026lt;-read.metharray.exp(base = \u0026quot;../1_Input/IDAT\u0026quot;, targets = targets, verbose = TRUE)\n# sampleNames(RGSet)\u0026lt;-targets$Sample_Name\nGRset.funnorm \u0026lt;- preprocessFunnorm(RGSet)\nbeta \u0026lt;- getBeta(GRset.funnorm)\nM\u0026lt;-getM(GRset.funnorm)\nCondition \u0026lt;- pData(GRset.funnorm)$Sample_Group\ndmp \u0026lt;- dmpFinder(beta, pheno = Condition  , type = \u0026quot;categorical\u0026quot;)\nDMPs\u0026lt;-merge(dmp, annoM450k, by= 0)\nDMPs\u0026lt;-as.data.frame(DMPs)\nrownames(DMPs)\u0026lt;-DMPs$Row.names\nDMPs\u0026lt;-DMPs %\u0026gt;% dplyr::select(-Row.names)\n#create an beta table\nbeta.table\u0026lt;-as.data.frame(t(beta))\nbeta.table$Col_ID\u0026lt;-rownames(beta.table)\nbeta_named\u0026lt;-merge(beta.table, targets, by = \u0026quot;Col_ID\u0026quot;)\nrownames(beta_named)\u0026lt;-beta_named$Sample_Name\nbeta_named\u0026lt;-beta_named %\u0026gt;% dplyr::select(-names(targets)) %\u0026gt;% t()\nwrite.csv(beta_named, paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_beta.table.csv\u0026quot;))\n#Add beta values to statistics\nResults_dmp\u0026lt;-merge(DMPs, beta_named, by = 0)\n#Calculate Average CpG Methylation by Sample_Group\nlibrary(dplyr)\nlibrary(matrixStats)\nResults\u0026lt;-Results_dmp %\u0026gt;% replace(is.na(.), 0) %\u0026gt;% dplyr::mutate(\n  DCM_SD = rowSds(as.matrix(Results_dmp[,targets$Sample_Name[targets$Sample_Group==\u0026quot;DCM\u0026quot;]])),\n  DCM_Mean = rowMeans(as.matrix(Results_dmp[,targets$Sample_Name[targets$Sample_Group==\u0026quot;DCM\u0026quot;]])),\n  CON_SD = rowSds(as.matrix(Results_dmp[,targets$Sample_Name[targets$Sample_Group==\u0026quot;CON\u0026quot;]])),\n  CON_Mean = rowMeans(as.matrix(Results_dmp[,targets$Sample_Name[targets$Sample_Group==\u0026quot;CON\u0026quot;]])),\n  Methylation.Diff=(DCM_Mean-CON_Mean)*100)\nrownames(Results)\u0026lt;-Results$Row.names\nResults_dmp_p05\u0026lt;-filter(Results, pval\u0026lt;0.05)\nResults_dmp_q05\u0026lt;-filter(Results, qval\u0026lt;0.05)\n#########################################\n#Identify Promoter-associated CpG Islands\nlibrary(tidyr)\nPromCGI\u0026lt;-dplyr::filter(Results_dmp_p05, grepl(\u0026quot;Island\u0026quot;, Relation_to_Island), grepl(\u0026quot;TSS\u0026quot;, UCSC_RefGene_Group))\n#Separate Gene Names into unique rows\nPromCGI_sep\u0026lt;-PromCGI %\u0026gt;% mutate(UCSC_RefGene_Name = strsplit(as.character(UCSC_RefGene_Name), \u0026quot;;\u0026quot;)) %\u0026gt;% unnest(UCSC_RefGene_Name) %\u0026gt;% distinct()\n#Save a copy of the countData\nlibrary(openxlsx)\nwb_countData\u0026lt;-createWorkbook()\naddWorksheet(wb_countData, \u0026quot;Unfiltered\u0026quot;)\n  writeData(wb_countData, \u0026quot;Unfiltered\u0026quot;, Results, startCol = 1)\naddWorksheet(wb_countData, \u0026quot;P_0.05\u0026quot;)\n  writeData(wb_countData, \u0026quot;P_0.05\u0026quot;, Results_dmp_p05, startCol = 1)\naddWorksheet(wb_countData, \u0026quot;Q_0.05\u0026quot;)\n  writeData(wb_countData, \u0026quot;Q_0.05\u0026quot;, Results_dmp_q05, startCol = 1)\naddWorksheet(wb_countData, \u0026quot;Promoter.CGI\u0026quot;)\n  writeData(wb_countData, \u0026quot;Promoter.CGI\u0026quot;, PromCGI_sep, startCol = 1)\nsaveWorkbook(wb_countData, file = paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_DMPs.xlsx\u0026quot;), overwrite = TRUE)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eQuantification and Differential Expression Analysis\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eread.metharray.exp(\u003cspan class=\"pl-v\"\u003ebase\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../1_Input/IDAT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etargets\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e sampleNames(RGSet)\u0026lt;-targets$Sample_Name\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eGRset.funnorm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e preprocessFunnorm(\u003cspan class=\"pl-smi\"\u003eRGSet\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ebeta\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e getBeta(\u003cspan class=\"pl-smi\"\u003eGRset.funnorm\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eM\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003egetM(\u003cspan class=\"pl-smi\"\u003eGRset.funnorm\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCondition\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e pData(\u003cspan class=\"pl-smi\"\u003eGRset.funnorm\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003edmp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e dmpFinder(\u003cspan class=\"pl-smi\"\u003ebeta\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epheno\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCondition\u003c/span\u003e  , \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecategorical\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eDMPs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003emerge(\u003cspan class=\"pl-smi\"\u003edmp\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eannoM450k\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eDMPs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.data.frame(\u003cspan class=\"pl-smi\"\u003eDMPs\u003c/span\u003e)\nrownames(\u003cspan class=\"pl-smi\"\u003eDMPs\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDMPs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRow.names\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eDMPs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDMPs\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRow.names\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecreate an beta table\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ebeta.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.data.frame(t(\u003cspan class=\"pl-smi\"\u003ebeta\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003ebeta.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCol_ID\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003erownames(\u003cspan class=\"pl-smi\"\u003ebeta.table\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ebeta_named\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003emerge(\u003cspan class=\"pl-smi\"\u003ebeta.table\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCol_ID\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nrownames(\u003cspan class=\"pl-smi\"\u003ebeta_named\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ebeta_named\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ebeta_named\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ebeta_named\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003enames(\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% t()\nwrite.csv(\u003cspan class=\"pl-smi\"\u003ebeta_named\u003c/span\u003e, paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_beta.table.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eAdd beta values to statistics\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eResults_dmp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003emerge(\u003cspan class=\"pl-smi\"\u003eDMPs\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ebeta_named\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eCalculate Average CpG Methylation by Sample_Group\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003ematrixStats\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eResults\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eResults_dmp\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% replace(is.na(\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e), \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003emutate(\n  \u003cspan class=\"pl-v\"\u003eDCM_SD\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e rowSds(as.matrix(\u003cspan class=\"pl-smi\"\u003eResults_dmp\u003c/span\u003e[,\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDCM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]])),\n  \u003cspan class=\"pl-v\"\u003eDCM_Mean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e rowMeans(as.matrix(\u003cspan class=\"pl-smi\"\u003eResults_dmp\u003c/span\u003e[,\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDCM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]])),\n  \u003cspan class=\"pl-v\"\u003eCON_SD\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e rowSds(as.matrix(\u003cspan class=\"pl-smi\"\u003eResults_dmp\u003c/span\u003e[,\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCON\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]])),\n  \u003cspan class=\"pl-v\"\u003eCON_Mean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e rowMeans(as.matrix(\u003cspan class=\"pl-smi\"\u003eResults_dmp\u003c/span\u003e[,\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCON\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]])),\n  \u003cspan class=\"pl-v\"\u003eMethylation.Diff\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eDCM_Mean\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCON_Mean\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e)\nrownames(\u003cspan class=\"pl-smi\"\u003eResults\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eResults\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRow.names\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eResults_dmp_p05\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003efilter(\u003cspan class=\"pl-smi\"\u003eResults\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eResults_dmp_q05\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003efilter(\u003cspan class=\"pl-smi\"\u003eResults\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eqval\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e########################################\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eIdentify Promoter-associated CpG Islands\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003etidyr\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ePromCGI\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efilter(\u003cspan class=\"pl-smi\"\u003eResults_dmp_p05\u003c/span\u003e, grepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIsland\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eRelation_to_Island\u003c/span\u003e), grepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTSS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSeparate Gene Names into unique rows\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ePromCGI_sep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePromCGI\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% mutate(\u003cspan class=\"pl-v\"\u003eUCSC_RefGene_Name\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e strsplit(as.character(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% unnest(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% distinct()\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSave a copy of the countData\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003eopenxlsx\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ewb_countData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003ecreateWorkbook()\naddWorksheet(\u003cspan class=\"pl-smi\"\u003ewb_countData\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUnfiltered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  writeData(\u003cspan class=\"pl-smi\"\u003ewb_countData\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUnfiltered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eResults\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estartCol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\naddWorksheet(\u003cspan class=\"pl-smi\"\u003ewb_countData\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eP_0.05\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  writeData(\u003cspan class=\"pl-smi\"\u003ewb_countData\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eP_0.05\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eResults_dmp_p05\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estartCol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\naddWorksheet(\u003cspan class=\"pl-smi\"\u003ewb_countData\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eQ_0.05\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  writeData(\u003cspan class=\"pl-smi\"\u003ewb_countData\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eQ_0.05\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eResults_dmp_q05\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estartCol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\naddWorksheet(\u003cspan class=\"pl-smi\"\u003ewb_countData\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePromoter.CGI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  writeData(\u003cspan class=\"pl-smi\"\u003ewb_countData\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePromoter.CGI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePromCGI_sep\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estartCol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\nsaveWorkbook(\u003cspan class=\"pl-smi\"\u003ewb_countData\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_DMPs.xlsx\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eoverwrite\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDistribution of Methylation by Genomic and CpG Annotation\u003c/h2\u003e\u003ca id=\"user-content-distribution-of-methylation-by-genomic-and-cpg-annotation\" class=\"anchor\" aria-label=\"Permalink: Distribution of Methylation by Genomic and CpG Annotation\" href=\"#distribution-of-methylation-by-genomic-and-cpg-annotation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe following figure illustrates the enrichment of differential\nmethylation with respect to CpG-based and genomic annotations.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(plotly)\nlibrary(dplyr)\nlibrary(tidyr)\nlibrary(stringr)\nlibrary(reshape2)\nlibrary(readxl)\nlibrary(kableExtra)\nlibrary(ComplexHeatmap)\nlibrary(openxlsx)\n\n##Create a regional annotation matrix of the m450k array\nRegion_epic\u0026lt;-Results_dmp %\u0026gt;% \n  dplyr::select(Row.names, UCSC_RefGene_Group, Relation_to_Island)\nStage1\u0026lt;-Region_epic %\u0026gt;% \n  mutate(UCSC_RefGene_Group = strsplit(as.character(UCSC_RefGene_Group), \u0026quot;;\u0026quot;)) %\u0026gt;% \n  unnest(UCSC_RefGene_Group) \nStage2\u0026lt;-Stage1 %\u0026gt;% \n  mutate(Relation_to_Island = strsplit(as.character(Relation_to_Island), \u0026quot;;\u0026quot;)) %\u0026gt;%  \n  unnest(Relation_to_Island)\nStage3\u0026lt;-distinct(Stage2)\nRegional.Groups\u0026lt;-dplyr::group_by_(Stage3, \u0026quot;UCSC_RefGene_Group\u0026quot;, \u0026quot;Relation_to_Island\u0026quot;) %\u0026gt;% \n  tally()\nRegion.matrix\u0026lt;-Regional.Groups %\u0026gt;% \n  spread(Relation_to_Island, n)\nRegion.matrix\u0026lt;-Region.matrix %\u0026gt;% \n  dplyr::select(UCSC_RefGene_Group, OpenSea, N_Shelf, N_Shore, Island, S_Shore, S_Shelf)\nrownames(Region.matrix)\u0026lt;-Region.matrix$UCSC_RefGene_Group\nRegion.matrix\u0026lt;-Region.matrix[c(\u0026quot;TSS1500\u0026quot;, \u0026quot;TSS200\u0026quot;, \u0026quot;5'UTR\u0026quot;, \u0026quot;1stExon\u0026quot;, \u0026quot;Body\u0026quot;, \u0026quot;3'UTR\u0026quot;),]\nrownames(Region.matrix)\u0026lt;-Region.matrix$UCSC_RefGene_Group\nContour_3D_epic\u0026lt;-Region.matrix[,-1]\nrownames(Contour_3D_epic)\u0026lt;-Region.matrix$UCSC_RefGene_Group\nwrite.xlsx(Contour_3D_epic, \u0026quot;../2_Output/Contour.3D_EPIC.xlsx\u0026quot;, overwrite = TRUE)\n\n##Create a regional annotation matrix of differentially-methylated positions\nRegion_p05\u0026lt;-Results_dmp_p05 %\u0026gt;% dplyr::select(Row.names, UCSC_RefGene_Group, Relation_to_Island)\nStage1\u0026lt;-Region_p05 %\u0026gt;% mutate(UCSC_RefGene_Group = strsplit(as.character(UCSC_RefGene_Group), \u0026quot;;\u0026quot;)) %\u0026gt;% unnest(UCSC_RefGene_Group) \nStage2\u0026lt;-Stage1 %\u0026gt;% mutate(Relation_to_Island = strsplit(as.character(Relation_to_Island), \u0026quot;;\u0026quot;)) %\u0026gt;%  unnest(Relation_to_Island)\nStage3\u0026lt;-distinct(Stage2)\nRegional.Groups\u0026lt;-dplyr::group_by_(Stage3, \u0026quot;UCSC_RefGene_Group\u0026quot;, \u0026quot;Relation_to_Island\u0026quot;) %\u0026gt;% tally()\nRegion.matrix\u0026lt;-Regional.Groups %\u0026gt;% spread(Relation_to_Island, n)\nRegion.matrix\u0026lt;-Region.matrix %\u0026gt;% dplyr::select(UCSC_RefGene_Group, OpenSea, N_Shelf, N_Shore, Island, S_Shore, S_Shelf)\nrownames(Region.matrix)\u0026lt;-Region.matrix$UCSC_RefGene_Group\nRegion.matrix\u0026lt;-Region.matrix[c(\u0026quot;TSS1500\u0026quot;, \u0026quot;TSS200\u0026quot;, \u0026quot;5'UTR\u0026quot;, \u0026quot;1stExon\u0026quot;, \u0026quot;Body\u0026quot;, \u0026quot;3'UTR\u0026quot;),]\nrownames(Region.matrix)\u0026lt;-Region.matrix$UCSC_RefGene_Group\nContour_3D\u0026lt;-Region.matrix[,-1]\nrownames(Contour_3D)\u0026lt;-Region.matrix$UCSC_RefGene_Group\nwrite.csv(Contour_3D, \u0026quot;../2_Output/Contour.3D_DMPs.p05.csv\u0026quot;)\n### Identify DMP Enrichment (IMPORTANT)\nEnrichment_Region\u0026lt;-Contour_3D/Contour_3D_epic\nPromoter_enr\u0026lt;-Enrichment_Region[rownames(Enrichment_Region)==\u0026quot;TSS200\u0026quot;,] + Enrichment_Region[rownames(Enrichment_Region)==\u0026quot;TSS1500\u0026quot;,]\nrownames(Promoter_enr)\u0026lt;-\u0026quot;Promoter\u0026quot;\nEnrichment_Region\u0026lt;-Enrichment_Region %\u0026gt;% \n  filter(!grepl(\u0026quot;TSS\u0026quot;, rownames(Enrichment_Region))) %\u0026gt;%\n  rbind(Promoter_enr, .) %\u0026gt;%\n  data.matrix()\npaletteLength\u0026lt;-100\nmyColor \u0026lt;- colorRampPalette(c(\u0026quot;dodgerblue4\u0026quot;, \u0026quot;white\u0026quot;, \u0026quot;brown4\u0026quot;))(paletteLength)\npheatmap(Enrichment_Region, color = myColor, cluster_rows = FALSE, cluster_cols = FALSE, filename = paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_DMP.Distribution.pdf\u0026quot;))\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eplotly\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003etidyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003estringr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003ereshape2\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003ereadxl\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003ekableExtra\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eComplexHeatmap\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eopenxlsx\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Create a regional annotation matrix of the m450k array\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eRegion_epic\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eResults_dmp\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eRow.names\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eRelation_to_Island\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eStage1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion_epic\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  mutate(\u003cspan class=\"pl-v\"\u003eUCSC_RefGene_Group\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e strsplit(as.character(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  unnest(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e) \n\u003cspan class=\"pl-smi\"\u003eStage2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eStage1\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  mutate(\u003cspan class=\"pl-v\"\u003eRelation_to_Island\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e strsplit(as.character(\u003cspan class=\"pl-smi\"\u003eRelation_to_Island\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%  \n  unnest(\u003cspan class=\"pl-smi\"\u003eRelation_to_Island\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eStage3\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003edistinct(\u003cspan class=\"pl-smi\"\u003eStage2\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eRegional.Groups\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003egroup_by_(\u003cspan class=\"pl-smi\"\u003eStage3\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUCSC_RefGene_Group\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRelation_to_Island\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  tally()\n\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegional.Groups\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  spread(\u003cspan class=\"pl-smi\"\u003eRelation_to_Island\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003en\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eOpenSea\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eN_Shelf\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eN_Shore\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eIsland\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eS_Shore\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eS_Shelf\u003c/span\u003e)\nrownames(\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e[c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTSS1500\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTSS200\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5'UTR\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e1stExon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBody\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e3'UTR\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),]\nrownames(\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eContour_3D_epic\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e[,\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\nrownames(\u003cspan class=\"pl-smi\"\u003eContour_3D_epic\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e\nwrite.xlsx(\u003cspan class=\"pl-smi\"\u003eContour_3D_epic\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/Contour.3D_EPIC.xlsx\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eoverwrite\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Create a regional annotation matrix of differentially-methylated positions\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eRegion_p05\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eResults_dmp_p05\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eRow.names\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eRelation_to_Island\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eStage1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion_p05\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% mutate(\u003cspan class=\"pl-v\"\u003eUCSC_RefGene_Group\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e strsplit(as.character(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% unnest(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e) \n\u003cspan class=\"pl-smi\"\u003eStage2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eStage1\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% mutate(\u003cspan class=\"pl-v\"\u003eRelation_to_Island\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e strsplit(as.character(\u003cspan class=\"pl-smi\"\u003eRelation_to_Island\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%  unnest(\u003cspan class=\"pl-smi\"\u003eRelation_to_Island\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eStage3\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003edistinct(\u003cspan class=\"pl-smi\"\u003eStage2\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eRegional.Groups\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003egroup_by_(\u003cspan class=\"pl-smi\"\u003eStage3\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUCSC_RefGene_Group\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRelation_to_Island\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% tally()\n\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegional.Groups\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% spread(\u003cspan class=\"pl-smi\"\u003eRelation_to_Island\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003en\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eOpenSea\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eN_Shelf\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eN_Shore\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eIsland\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eS_Shore\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eS_Shelf\u003c/span\u003e)\nrownames(\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e[c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTSS1500\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTSS200\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5'UTR\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e1stExon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBody\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e3'UTR\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),]\nrownames(\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eContour_3D\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e[,\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\nrownames(\u003cspan class=\"pl-smi\"\u003eContour_3D\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion.matrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e\nwrite.csv(\u003cspan class=\"pl-smi\"\u003eContour_3D\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/Contour.3D_DMPs.p05.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## Identify DMP Enrichment (IMPORTANT)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eContour_3D\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eContour_3D_epic\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ePromoter_enr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e[rownames(\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTSS200\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,] \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e[rownames(\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTSS1500\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,]\nrownames(\u003cspan class=\"pl-smi\"\u003ePromoter_enr\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePromoter\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  filter(\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003egrepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTSS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, rownames(\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e))) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  rbind(\u003cspan class=\"pl-smi\"\u003ePromoter_enr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  data.matrix()\n\u003cspan class=\"pl-smi\"\u003epaletteLength\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emyColor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colorRampPalette(c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edodgerblue4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebrown4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))(\u003cspan class=\"pl-smi\"\u003epaletteLength\u003c/span\u003e)\npheatmap(\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emyColor\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecluster_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecluster_cols\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_DMP.Distribution.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e![](README_files/figure-html/Methylation Distribution-1.png)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"write.csv(Enrichment_Region, \u0026quot;DCM_vs_CON.csv\u0026quot;)\n##Make a Table of the CpG Methylation Distribution\nEnrichment_Region %\u0026gt;% kable( align=\u0026quot;c\u0026quot;, booktabs=T, \n                     caption=\u0026quot;Methylation Distribution\u0026quot;) %\u0026gt;% \n  kable_styling(latex_options=c(\u0026quot;striped\u0026quot;, \u0026quot;condensed\u0026quot;, \u0026quot;repeat_header\u0026quot;))\"\u003e\u003cpre\u003ewrite.csv(\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDCM_vs_CON.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Make a Table of the CpG Methylation Distribution\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% kable( \u003cspan class=\"pl-v\"\u003ealign\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ec\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ebooktabs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003ecaption\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMethylation Distribution\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  kable_styling(\u003cspan class=\"pl-v\"\u003elatex_options\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003estriped\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econdensed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erepeat_header\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\u003c/pre\u003e\u003c/div\u003e\n\nMethylation Distribution\n \u003ctable\u003e\u003cthead\u003e\n  \u003ctr\u003e\n   \u003cth\u003e   \u003c/th\u003e\n   \u003cth\u003e OpenSea \u003c/th\u003e\n   \u003cth\u003e N_Shelf \u003c/th\u003e\n   \u003cth\u003e N_Shore \u003c/th\u003e\n   \u003cth\u003e Island \u003c/th\u003e\n   \u003cth\u003e S_Shore \u003c/th\u003e\n   \u003cth\u003e S_Shelf \u003c/th\u003e\n  \u003c/tr\u003e\n \u003c/thead\u003e\n\u003ctbody\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e Promoter \u003c/td\u003e\n   \u003ctd\u003e 0.1164621 \u003c/td\u003e\n   \u003ctd\u003e 0.1029990 \u003c/td\u003e\n   \u003ctd\u003e 0.1242210 \u003c/td\u003e\n   \u003ctd\u003e 0.1460420 \u003c/td\u003e\n   \u003ctd\u003e 0.1156378 \u003c/td\u003e\n   \u003ctd\u003e 0.1037933 \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e 5'UTR \u003c/td\u003e\n   \u003ctd\u003e 0.0530359 \u003c/td\u003e\n   \u003ctd\u003e 0.0622389 \u003c/td\u003e\n   \u003ctd\u003e 0.0569429 \u003c/td\u003e\n   \u003ctd\u003e 0.0741839 \u003c/td\u003e\n   \u003ctd\u003e 0.0573959 \u003c/td\u003e\n   \u003ctd\u003e 0.0576832 \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e 1stExon \u003c/td\u003e\n   \u003ctd\u003e 0.0588488 \u003c/td\u003e\n   \u003ctd\u003e 0.0489130 \u003c/td\u003e\n   \u003ctd\u003e 0.0623833 \u003c/td\u003e\n   \u003ctd\u003e 0.0746424 \u003c/td\u003e\n   \u003ctd\u003e 0.0625000 \u003c/td\u003e\n   \u003ctd\u003e 0.0370370 \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e Body \u003c/td\u003e\n   \u003ctd\u003e 0.0523911 \u003c/td\u003e\n   \u003ctd\u003e 0.0493794 \u003c/td\u003e\n   \u003ctd\u003e 0.0510093 \u003c/td\u003e\n   \u003ctd\u003e 0.0633285 \u003c/td\u003e\n   \u003ctd\u003e 0.0507863 \u003c/td\u003e\n   \u003ctd\u003e 0.0473963 \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e 3'UTR \u003c/td\u003e\n   \u003ctd\u003e 0.0485610 \u003c/td\u003e\n   \u003ctd\u003e 0.0415879 \u003c/td\u003e\n   \u003ctd\u003e 0.0554833 \u003c/td\u003e\n   \u003ctd\u003e 0.0638298 \u003c/td\u003e\n   \u003ctd\u003e 0.0563931 \u003c/td\u003e\n   \u003ctd\u003e 0.0479303 \u003c/td\u003e\n  \u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"write.xlsx(Enrichment_Region, paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_DMP.Enrichment_3D.xlsx\u0026quot;), overwrite = T)\ncolor \u0026lt;- colorRampPalette(c(\u0026quot;grey\u0026quot;, \u0026quot;orange\u0026quot;, \u0026quot;red\u0026quot;))\nt \u0026lt;- list(\n  family = \u0026quot;times\u0026quot;,\n  size = 16,\n  color = \u0026quot;black\u0026quot;)\nx_axis\u0026lt;-list(title = 'CpG Region', \n                     type=\u0026quot;category\u0026quot;, \n                     zeroline=TRUE, \n                     showline=TRUE, \n                     zerolinewidth = 4, \n            zerolinecolor=\u0026quot;darkgrey\u0026quot;, \n            linecolor=\u0026quot;darkgrey\u0026quot;, \n            linewidth=4, \n            titlefont=t, \n            tickfont=t)\ny_axis\u0026lt;-list(title = 'Gene Region', \n                     type=\u0026quot;category\u0026quot;, \n                     zeroline=TRUE, \n                     showline=TRUE, \n                     zerolinewidth = 4, \n            zerolinecolor=\u0026quot;darkgrey\u0026quot;, \n            linecolor=\u0026quot;darkgrey\u0026quot;, \n            linewidth=4, \n            titlefont=t, \n            tickfont=t)\nz_axis\u0026lt;-list(title = 'Number of DMPs', \n                     zerolinewidth = 4, \n                    zerolinecolor=\u0026quot;darkgrey\u0026quot;, \n                    linecolor=\u0026quot;darkgrey\u0026quot;, \n                    linewidth=4, \n                    titlefont=t, \n                    tickfont=t)\nq\u0026lt;-plot_ly(z=~Enrichment_Region, colors=color(10), \n    text=as.character(rownames(Enrichment_Region))) %\u0026gt;% add_surface() %\u0026gt;% \n    layout(scene = list(xaxis = x_axis, yaxis = y_axis, zaxis = z_axis))\nq #must comment out for PDF generation via knitr (Pandoc).\"\u003e\u003cpre\u003ewrite.xlsx(\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e, paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_DMP.Enrichment_3D.xlsx\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eoverwrite\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ecolor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colorRampPalette(c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egrey\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorange\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003et\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\n  \u003cspan class=\"pl-v\"\u003efamily\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etimes\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e,\n  \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ex_axis\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCpG Region\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecategory\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003ezeroline\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003eshowline\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003ezerolinewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003ezerolinecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgrey\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003elinecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgrey\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003elinewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003etitlefont\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003et\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003etickfont\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003et\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ey_axis\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eGene Region\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecategory\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003ezeroline\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003eshowline\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003ezerolinewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003ezerolinecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgrey\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003elinecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgrey\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003elinewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003etitlefont\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003et\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003etickfont\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003et\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ez_axis\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003etitle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNumber of DMPs\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003ezerolinewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e, \n                    \u003cspan class=\"pl-v\"\u003ezerolinecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgrey\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                    \u003cspan class=\"pl-v\"\u003elinecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgrey\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                    \u003cspan class=\"pl-v\"\u003elinewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e, \n                    \u003cspan class=\"pl-v\"\u003etitlefont\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003et\u003c/span\u003e, \n                    \u003cspan class=\"pl-v\"\u003etickfont\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003et\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eq\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eplot_ly(\u003cspan class=\"pl-v\"\u003ez\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolors\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ecolor(\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e), \n    \u003cspan class=\"pl-v\"\u003etext\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eas.character(rownames(\u003cspan class=\"pl-smi\"\u003eEnrichment_Region\u003c/span\u003e))) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% add_surface() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    layout(\u003cspan class=\"pl-v\"\u003escene\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003exaxis\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ex_axis\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eyaxis\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ey_axis\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ezaxis\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ez_axis\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eq\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003emust comment out for PDF generation via knitr (Pandoc).\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\u0026lt;div id=\u0026quot;htmlwidget-abe0d5b12822ca5cb440\u0026quot; style=\u0026quot;width:672px;height:480px;\u0026quot; class=\u0026quot;plotly html-widget\u0026quot;\u0026gt;\u0026lt;/div\u0026gt;\n\u0026lt;script type=\u0026quot;application/json\u0026quot; data-for=\u0026quot;htmlwidget-abe0d5b12822ca5cb440\u0026quot;\u0026gt;{\u0026quot;x\u0026quot;:{\u0026quot;visdat\u0026quot;:{\u0026quot;2ee73db9e5dd\u0026quot;:[\u0026quot;function () \u0026quot;,\u0026quot;plotlyVisDat\u0026quot;]},\u0026quot;cur_data\u0026quot;:\u0026quot;2ee73db9e5dd\u0026quot;,\u0026quot;attrs\u0026quot;:{\u0026quot;2ee73db9e5dd\u0026quot;:{\u0026quot;z\u0026quot;:{},\u0026quot;text\u0026quot;:[\u0026quot;Promoter\u0026quot;,\u0026quot;5'UTR\u0026quot;,\u0026quot;1stExon\u0026quot;,\u0026quot;Body\u0026quot;,\u0026quot;3'UTR\u0026quot;],\u0026quot;colors\u0026quot;:[\u0026quot;#BEBEBE\u0026quot;,\u0026quot;#CCB893\u0026quot;,\u0026quot;#DAB269\u0026quot;,\u0026quot;#E9AD3F\u0026quot;,\u0026quot;#F7A715\u0026quot;,\u0026quot;#FF9200\u0026quot;,\u0026quot;#FF6E00\u0026quot;,\u0026quot;#FF4900\u0026quot;,\u0026quot;#FF2400\u0026quot;,\u0026quot;#FF0000\u0026quot;],\u0026quot;alpha_stroke\u0026quot;:1,\u0026quot;sizes\u0026quot;:[10,100],\u0026quot;spans\u0026quot;:[1,20],\u0026quot;type\u0026quot;:\u0026quot;surface\u0026quot;,\u0026quot;inherit\u0026quot;:true}},\u0026quot;layout\u0026quot;:{\u0026quot;margin\u0026quot;:{\u0026quot;b\u0026quot;:40,\u0026quot;l\u0026quot;:60,\u0026quot;t\u0026quot;:25,\u0026quot;r\u0026quot;:10},\u0026quot;scene\u0026quot;:{\u0026quot;xaxis\u0026quot;:{\u0026quot;title\u0026quot;:\u0026quot;CpG Region\u0026quot;,\u0026quot;type\u0026quot;:\u0026quot;category\u0026quot;,\u0026quot;zeroline\u0026quot;:true,\u0026quot;showline\u0026quot;:true,\u0026quot;zerolinewidth\u0026quot;:4,\u0026quot;zerolinecolor\u0026quot;:\u0026quot;darkgrey\u0026quot;,\u0026quot;linecolor\u0026quot;:\u0026quot;darkgrey\u0026quot;,\u0026quot;linewidth\u0026quot;:4,\u0026quot;titlefont\u0026quot;:{\u0026quot;family\u0026quot;:\u0026quot;times\u0026quot;,\u0026quot;size\u0026quot;:16,\u0026quot;color\u0026quot;:\u0026quot;black\u0026quot;},\u0026quot;tickfont\u0026quot;:{\u0026quot;family\u0026quot;:\u0026quot;times\u0026quot;,\u0026quot;size\u0026quot;:16,\u0026quot;color\u0026quot;:\u0026quot;black\u0026quot;}},\u0026quot;yaxis\u0026quot;:{\u0026quot;title\u0026quot;:\u0026quot;Gene Region\u0026quot;,\u0026quot;type\u0026quot;:\u0026quot;category\u0026quot;,\u0026quot;zeroline\u0026quot;:true,\u0026quot;showline\u0026quot;:true,\u0026quot;zerolinewidth\u0026quot;:4,\u0026quot;zerolinecolor\u0026quot;:\u0026quot;darkgrey\u0026quot;,\u0026quot;linecolor\u0026quot;:\u0026quot;darkgrey\u0026quot;,\u0026quot;linewidth\u0026quot;:4,\u0026quot;titlefont\u0026quot;:{\u0026quot;family\u0026quot;:\u0026quot;times\u0026quot;,\u0026quot;size\u0026quot;:16,\u0026quot;color\u0026quot;:\u0026quot;black\u0026quot;},\u0026quot;tickfont\u0026quot;:{\u0026quot;family\u0026quot;:\u0026quot;times\u0026quot;,\u0026quot;size\u0026quot;:16,\u0026quot;color\u0026quot;:\u0026quot;black\u0026quot;}},\u0026quot;zaxis\u0026quot;:{\u0026quot;title\u0026quot;:\u0026quot;Number of DMPs\u0026quot;,\u0026quot;zerolinewidth\u0026quot;:4,\u0026quot;zerolinecolor\u0026quot;:\u0026quot;darkgrey\u0026quot;,\u0026quot;linecolor\u0026quot;:\u0026quot;darkgrey\u0026quot;,\u0026quot;linewidth\u0026quot;:4,\u0026quot;titlefont\u0026quot;:{\u0026quot;family\u0026quot;:\u0026quot;times\u0026quot;,\u0026quot;size\u0026quot;:16,\u0026quot;color\u0026quot;:\u0026quot;black\u0026quot;},\u0026quot;tickfont\u0026quot;:{\u0026quot;family\u0026quot;:\u0026quot;times\u0026quot;,\u0026quot;size\u0026quot;:16,\u0026quot;color\u0026quot;:\u0026quot;black\u0026quot;}}},\u0026quot;hovermode\u0026quot;:\u0026quot;closest\u0026quot;,\u0026quot;showlegend\u0026quot;:false,\u0026quot;legend\u0026quot;:{\u0026quot;yanchor\u0026quot;:\u0026quot;top\u0026quot;,\u0026quot;y\u0026quot;:0.5}},\u0026quot;source\u0026quot;:\u0026quot;A\u0026quot;,\u0026quot;config\u0026quot;:{\u0026quot;modeBarButtonsToAdd\u0026quot;:[\u0026quot;hoverclosest\u0026quot;,\u0026quot;hovercompare\u0026quot;],\u0026quot;showSendToCloud\u0026quot;:false},\u0026quot;data\u0026quot;:[{\u0026quot;colorbar\u0026quot;:{\u0026quot;title\u0026quot;:\u0026quot;Enrichment_Region\u0026quot;,\u0026quot;ticklen\u0026quot;:2,\u0026quot;len\u0026quot;:0.5,\u0026quot;lenmode\u0026quot;:\u0026quot;fraction\u0026quot;,\u0026quot;y\u0026quot;:1,\u0026quot;yanchor\u0026quot;:\u0026quot;top\u0026quot;},\u0026quot;colorscale\u0026quot;:[[\u0026quot;0\u0026quot;,\u0026quot;rgba(190,190,190,1)\u0026quot;],[\u0026quot;0.0416666666666666\u0026quot;,\u0026quot;rgba(196,188,174,1)\u0026quot;],[\u0026quot;0.0833333333333333\u0026quot;,\u0026quot;rgba(201,185,158,1)\u0026quot;],[\u0026quot;0.125\u0026quot;,\u0026quot;rgba(206,183,142,1)\u0026quot;],[\u0026quot;0.166666666666667\u0026quot;,\u0026quot;rgba(212,181,126,1)\u0026quot;],[\u0026quot;0.208333333333333\u0026quot;,\u0026quot;rgba(217,179,110,1)\u0026quot;],[\u0026quot;0.25\u0026quot;,\u0026quot;rgba(222,177,95,1)\u0026quot;],[\u0026quot;0.291666666666667\u0026quot;,\u0026quot;rgba(228,175,80,1)\u0026quot;],[\u0026quot;0.333333333333333\u0026quot;,\u0026quot;rgba(233,173,63,1)\u0026quot;],[\u0026quot;0.375\u0026quot;,\u0026quot;rgba(238,171,50,1)\u0026quot;],[\u0026quot;0.416666666666667\u0026quot;,\u0026quot;rgba(244,169,35,1)\u0026quot;],[\u0026quot;0.458333333333333\u0026quot;,\u0026quot;rgba(248,164,19,1)\u0026quot;],[\u0026quot;0.5\u0026quot;,\u0026quot;rgba(251,157,11,1)\u0026quot;],[\u0026quot;0.541666666666667\u0026quot;,\u0026quot;rgba(254,149,3,1)\u0026quot;],[\u0026quot;0.583333333333333\u0026quot;,\u0026quot;rgba(255,137,0,1)\u0026quot;],[\u0026quot;0.625\u0026quot;,\u0026quot;rgba(255,124,0,1)\u0026quot;],[\u0026quot;0.666666666666667\u0026quot;,\u0026quot;rgba(255,110,0,1)\u0026quot;],[\u0026quot;0.708333333333333\u0026quot;,\u0026quot;rgba(255,97,0,1)\u0026quot;],[\u0026quot;0.75\u0026quot;,\u0026quot;rgba(255,83,0,1)\u0026quot;],[\u0026quot;0.791666666666667\u0026quot;,\u0026quot;rgba(255,69,0,1)\u0026quot;],[\u0026quot;0.833333333333333\u0026quot;,\u0026quot;rgba(255,57,0,1)\u0026quot;],[\u0026quot;0.875\u0026quot;,\u0026quot;rgba(255,42,0,1)\u0026quot;],[\u0026quot;0.916666666666667\u0026quot;,\u0026quot;rgba(255,30,0,1)\u0026quot;],[\u0026quot;0.958333333333333\u0026quot;,\u0026quot;rgba(255,19,0,1)\u0026quot;],[\u0026quot;1\u0026quot;,\u0026quot;rgba(255,0,0,1)\u0026quot;]],\u0026quot;showscale\u0026quot;:true,\u0026quot;z\u0026quot;:[[0.1164621465399,0.102998995174501,0.124221049754897,0.146041950369168,0.115637795393847,0.103793295570619],[0.0530358898721731,0.0622389306599833,0.0569429198682766,0.0741838816628411,0.0573958674975402,0.0576832151300236],[0.0588487972508591,0.0489130434782609,0.0623832648487112,0.0746423541620339,0.0625,0.037037037037037],[0.0523911393278587,0.0493794088757925,0.0510093237983163,0.0633284583420387,0.0507863308501833,0.0473962571196094],[0.0485609908224182,0.0415879017013233,0.0554833116601647,0.0638297872340425,0.056393076493579,0.0479302832244009]],\u0026quot;text\u0026quot;:[\u0026quot;Promoter\u0026quot;,\u0026quot;5'UTR\u0026quot;,\u0026quot;1stExon\u0026quot;,\u0026quot;Body\u0026quot;,\u0026quot;3'UTR\u0026quot;],\u0026quot;type\u0026quot;:\u0026quot;surface\u0026quot;,\u0026quot;frame\u0026quot;:null}],\u0026quot;highlight\u0026quot;:{\u0026quot;on\u0026quot;:\u0026quot;plotly_click\u0026quot;,\u0026quot;persistent\u0026quot;:false,\u0026quot;dynamic\u0026quot;:false,\u0026quot;selectize\u0026quot;:false,\u0026quot;opacityDim\u0026quot;:0.2,\u0026quot;selected\u0026quot;:{\u0026quot;opacity\u0026quot;:1},\u0026quot;debounce\u0026quot;:0},\u0026quot;shinyEvents\u0026quot;:[\u0026quot;plotly_hover\u0026quot;,\u0026quot;plotly_click\u0026quot;,\u0026quot;plotly_selected\u0026quot;,\u0026quot;plotly_relayout\u0026quot;,\u0026quot;plotly_brushed\u0026quot;,\u0026quot;plotly_brushing\u0026quot;,\u0026quot;plotly_clickannotation\u0026quot;,\u0026quot;plotly_doubleclick\u0026quot;,\u0026quot;plotly_deselect\u0026quot;,\u0026quot;plotly_afterplot\u0026quot;,\u0026quot;plotly_sunburstclick\u0026quot;],\u0026quot;base_url\u0026quot;:\u0026quot;https://plot.ly\u0026quot;},\u0026quot;evals\u0026quot;:[],\u0026quot;jsHooks\u0026quot;:[]}\u0026lt;/script\u0026gt;\"\u003e\u003cpre lang=\"{=html}\" class=\"notranslate\"\u003e\u003ccode\u003e\u0026lt;div id=\"htmlwidget-abe0d5b12822ca5cb440\" style=\"width:672px;height:480px;\" class=\"plotly html-widget\"\u0026gt;\u0026lt;/div\u0026gt;\n\u0026lt;script type=\"application/json\" data-for=\"htmlwidget-abe0d5b12822ca5cb440\"\u0026gt;{\"x\":{\"visdat\":{\"2ee73db9e5dd\":[\"function () \",\"plotlyVisDat\"]},\"cur_data\":\"2ee73db9e5dd\",\"attrs\":{\"2ee73db9e5dd\":{\"z\":{},\"text\":[\"Promoter\",\"5'UTR\",\"1stExon\",\"Body\",\"3'UTR\"],\"colors\":[\"#BEBEBE\",\"#CCB893\",\"#DAB269\",\"#E9AD3F\",\"#F7A715\",\"#FF9200\",\"#FF6E00\",\"#FF4900\",\"#FF2400\",\"#FF0000\"],\"alpha_stroke\":1,\"sizes\":[10,100],\"spans\":[1,20],\"type\":\"surface\",\"inherit\":true}},\"layout\":{\"margin\":{\"b\":40,\"l\":60,\"t\":25,\"r\":10},\"scene\":{\"xaxis\":{\"title\":\"CpG Region\",\"type\":\"category\",\"zeroline\":true,\"showline\":true,\"zerolinewidth\":4,\"zerolinecolor\":\"darkgrey\",\"linecolor\":\"darkgrey\",\"linewidth\":4,\"titlefont\":{\"family\":\"times\",\"size\":16,\"color\":\"black\"},\"tickfont\":{\"family\":\"times\",\"size\":16,\"color\":\"black\"}},\"yaxis\":{\"title\":\"Gene Region\",\"type\":\"category\",\"zeroline\":true,\"showline\":true,\"zerolinewidth\":4,\"zerolinecolor\":\"darkgrey\",\"linecolor\":\"darkgrey\",\"linewidth\":4,\"titlefont\":{\"family\":\"times\",\"size\":16,\"color\":\"black\"},\"tickfont\":{\"family\":\"times\",\"size\":16,\"color\":\"black\"}},\"zaxis\":{\"title\":\"Number of DMPs\",\"zerolinewidth\":4,\"zerolinecolor\":\"darkgrey\",\"linecolor\":\"darkgrey\",\"linewidth\":4,\"titlefont\":{\"family\":\"times\",\"size\":16,\"color\":\"black\"},\"tickfont\":{\"family\":\"times\",\"size\":16,\"color\":\"black\"}}},\"hovermode\":\"closest\",\"showlegend\":false,\"legend\":{\"yanchor\":\"top\",\"y\":0.5}},\"source\":\"A\",\"config\":{\"modeBarButtonsToAdd\":[\"hoverclosest\",\"hovercompare\"],\"showSendToCloud\":false},\"data\":[{\"colorbar\":{\"title\":\"Enrichment_Region\",\"ticklen\":2,\"len\":0.5,\"lenmode\":\"fraction\",\"y\":1,\"yanchor\":\"top\"},\"colorscale\":[[\"0\",\"rgba(190,190,190,1)\"],[\"0.0416666666666666\",\"rgba(196,188,174,1)\"],[\"0.0833333333333333\",\"rgba(201,185,158,1)\"],[\"0.125\",\"rgba(206,183,142,1)\"],[\"0.166666666666667\",\"rgba(212,181,126,1)\"],[\"0.208333333333333\",\"rgba(217,179,110,1)\"],[\"0.25\",\"rgba(222,177,95,1)\"],[\"0.291666666666667\",\"rgba(228,175,80,1)\"],[\"0.333333333333333\",\"rgba(233,173,63,1)\"],[\"0.375\",\"rgba(238,171,50,1)\"],[\"0.416666666666667\",\"rgba(244,169,35,1)\"],[\"0.458333333333333\",\"rgba(248,164,19,1)\"],[\"0.5\",\"rgba(251,157,11,1)\"],[\"0.541666666666667\",\"rgba(254,149,3,1)\"],[\"0.583333333333333\",\"rgba(255,137,0,1)\"],[\"0.625\",\"rgba(255,124,0,1)\"],[\"0.666666666666667\",\"rgba(255,110,0,1)\"],[\"0.708333333333333\",\"rgba(255,97,0,1)\"],[\"0.75\",\"rgba(255,83,0,1)\"],[\"0.791666666666667\",\"rgba(255,69,0,1)\"],[\"0.833333333333333\",\"rgba(255,57,0,1)\"],[\"0.875\",\"rgba(255,42,0,1)\"],[\"0.916666666666667\",\"rgba(255,30,0,1)\"],[\"0.958333333333333\",\"rgba(255,19,0,1)\"],[\"1\",\"rgba(255,0,0,1)\"]],\"showscale\":true,\"z\":[[0.1164621465399,0.102998995174501,0.124221049754897,0.146041950369168,0.115637795393847,0.103793295570619],[0.0530358898721731,0.0622389306599833,0.0569429198682766,0.0741838816628411,0.0573958674975402,0.0576832151300236],[0.0588487972508591,0.0489130434782609,0.0623832648487112,0.0746423541620339,0.0625,0.037037037037037],[0.0523911393278587,0.0493794088757925,0.0510093237983163,0.0633284583420387,0.0507863308501833,0.0473962571196094],[0.0485609908224182,0.0415879017013233,0.0554833116601647,0.0638297872340425,0.056393076493579,0.0479302832244009]],\"text\":[\"Promoter\",\"5'UTR\",\"1stExon\",\"Body\",\"3'UTR\"],\"type\":\"surface\",\"frame\":null}],\"highlight\":{\"on\":\"plotly_click\",\"persistent\":false,\"dynamic\":false,\"selectize\":false,\"opacityDim\":0.2,\"selected\":{\"opacity\":1},\"debounce\":0},\"shinyEvents\":[\"plotly_hover\",\"plotly_click\",\"plotly_selected\",\"plotly_relayout\",\"plotly_brushed\",\"plotly_brushing\",\"plotly_clickannotation\",\"plotly_doubleclick\",\"plotly_deselect\",\"plotly_afterplot\",\"plotly_sunburstclick\"],\"base_url\":\"https://plot.ly\"},\"evals\":[],\"jsHooks\":[]}\u0026lt;/script\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFigure XX: Heatmap and Hierarchical Clustering of Differential Methylation (P \u0026lt; 0.05)\u003c/h2\u003e\u003ca id=\"user-content-figure-xx-heatmap-and-hierarchical-clustering-of-differential-methylation-p--005\" class=\"anchor\" aria-label=\"Permalink: Figure XX: Heatmap and Hierarchical Clustering of Differential Methylation (P \u0026lt; 0.05)\" href=\"#figure-xx-heatmap-and-hierarchical-clustering-of-differential-methylation-p--005\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDue to the prominent signature of CpG Island-associated promoter\nmethylation in iPSCs exposed to serum for failing and non-failing\npatients, we examined all DMPs present within this region via heatmap.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(ComplexHeatmap)\nlibrary(dplyr)\n##Import Data Matrix\n# betaHM\u0026lt;-read.csv(\u0026quot;../1_Input/EPIC.betaValues.csv\u0026quot;, row.names = 1)\n## Filters to Apply to DMP\npvalue_threshold=0.001\nMETHYLATION=0\nDMP_location=\u0026quot;Island\u0026quot;\nGene_region=\u0026quot;TSS\u0026quot;\nSamples\u0026lt;-AnnoTargets$Sample_Name\n##Filter Differential Methylation Data\nDMP.p05\u0026lt;-Results %\u0026gt;% filter(pval\u0026lt;pvalue_threshold)\nDMP.p05\u0026lt;-DMP.p05 %\u0026gt;% dplyr::select(Row.names, \n                            Methylation.Diff, \n                            pval, \n                            qval, \n                            Relation_to_Island, \n                            UCSC_RefGene_Group, \n                            chr, \n                            pos, \n                            matches(Samples)) %\u0026gt;%\n                      filter(abs(Methylation.Diff)\u0026gt;METHYLATION)\nDMP.p05.Region\u0026lt;-DMP.p05 %\u0026gt;% \n  # filter(grepl(DMP_location, Relation_to_Island)) %\u0026gt;%\n  filter(grepl(Gene_region, UCSC_RefGene_Group)) %\u0026gt;% \n  dplyr::select(matches(AnnoTargets$Sample_Name)) %\u0026gt;%\n  data.matrix()\n#Import the Index File\nLVAD_Counts_Data \u0026lt;- targets\nrownames(LVAD_Counts_Data)\u0026lt;-targets$Sample_Name\nIndex\u0026lt;-LVAD_Counts_Data %\u0026gt;% dplyr::select(Sample_Group)\nIndex\u0026lt;-as.data.frame(Index)\npaletteLength \u0026lt;- 100\nann_colors = list(Sample_Group = c(DCM=\u0026quot;darkcyan\u0026quot;, CON=\u0026quot;darkgray\u0026quot;), Outcome = c(Bad = \u0026quot;firebrick2\u0026quot;, Good = \u0026quot;darkgoldenrod2\u0026quot;, CON = \u0026quot;black\u0026quot;))\nmyColor \u0026lt;- colorRampPalette(c(\u0026quot;dodgerblue4\u0026quot;, \u0026quot;white\u0026quot;, \u0026quot;goldenrod2\u0026quot;))(paletteLength)\npheatmap::pheatmap(DMP.p05.Region, scale=\u0026quot;row\u0026quot;, \n                      cluster_cols = TRUE, \n                      cluster_rows = TRUE,\n                      #breaks = myBreaks,\n                      cutree_cols = 2,\n                      cutree_rows = 2,\n                      angle_col = 45,\n                      fontsize_col = 8,\n                      color = myColor, \n                      show_rownames = FALSE, \n                      border_color = NA, \n                      annotation_col = Index,\n                      annotation_colors=ann_colors,\n                      filename = paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_\u0026quot;, DMP_location, \u0026quot;.\u0026quot;, Gene_region,\u0026quot;.heatmap.pdf\u0026quot;))\n\npheatmap::pheatmap(DMP.p05.Region, scale=\u0026quot;row\u0026quot;, \n                      cluster_cols = TRUE, \n                      cluster_rows = TRUE,\n                      #breaks = myBreaks,\n                      cutree_cols = 2,\n                      cutree_rows = 2,\n                      angle_col = 45,\n                      fontsize_col = 8,\n                      color = myColor, \n                      show_rownames = FALSE, \n                      border_color = NA, \n                      annotation_col = Index,\n                      annotation_colors=ann_colors)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eComplexHeatmap\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Import Data Matrix\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e betaHM\u0026lt;-read.csv(\"../1_Input/EPIC.betaValues.csv\", row.names = 1)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Filters to Apply to DMP\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003epvalue_threshold\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.001\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eMETHYLATION\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eDMP_location\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIsland\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eGene_region\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTSS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eSamples\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Filter Differential Methylation Data\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eDMP.p05\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eResults\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% filter(\u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003epvalue_threshold\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eDMP.p05\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDMP.p05\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eRow.names\u003c/span\u003e, \n                            \u003cspan class=\"pl-smi\"\u003eMethylation.Diff\u003c/span\u003e, \n                            \u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e, \n                            \u003cspan class=\"pl-smi\"\u003eqval\u003c/span\u003e, \n                            \u003cspan class=\"pl-smi\"\u003eRelation_to_Island\u003c/span\u003e, \n                            \u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e, \n                            \u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \n                            \u003cspan class=\"pl-smi\"\u003epos\u003c/span\u003e, \n                            matches(\u003cspan class=\"pl-smi\"\u003eSamples\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n                      filter(abs(\u003cspan class=\"pl-smi\"\u003eMethylation.Diff\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMETHYLATION\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eDMP.p05.Region\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDMP.p05\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e filter(grepl(DMP_location, Relation_to_Island)) %\u0026gt;%\u003c/span\u003e\n  filter(grepl(\u003cspan class=\"pl-smi\"\u003eGene_region\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Group\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(matches(\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  data.matrix()\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eImport the Index File\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eLVAD_Counts_Data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\nrownames(\u003cspan class=\"pl-smi\"\u003eLVAD_Counts_Data\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eIndex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLVAD_Counts_Data\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eIndex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.data.frame(\u003cspan class=\"pl-smi\"\u003eIndex\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003epaletteLength\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eann_colors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eSample_Group\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-v\"\u003eDCM\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkcyan\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCON\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgray\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eOutcome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-v\"\u003eBad\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efirebrick2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eGood\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgoldenrod2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCON\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003emyColor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colorRampPalette(c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edodgerblue4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egoldenrod2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))(\u003cspan class=\"pl-smi\"\u003epaletteLength\u003c/span\u003e)\n\u003cspan class=\"pl-e\"\u003epheatmap\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003epheatmap(\u003cspan class=\"pl-smi\"\u003eDMP.p05.Region\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003ecluster_cols\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003ecluster_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                      \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ebreaks = myBreaks,\u003c/span\u003e\n                      \u003cspan class=\"pl-v\"\u003ecutree_cols\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\n                      \u003cspan class=\"pl-v\"\u003ecutree_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\n                      \u003cspan class=\"pl-v\"\u003eangle_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e45\u003c/span\u003e,\n                      \u003cspan class=\"pl-v\"\u003efontsize_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\n                      \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emyColor\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003eshow_rownames\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003eborder_color\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003eannotation_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eIndex\u003c/span\u003e,\n                      \u003cspan class=\"pl-v\"\u003eannotation_colors\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eann_colors\u003c/span\u003e,\n                      \u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eDMP_location\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eGene_region\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.heatmap.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-e\"\u003epheatmap\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003epheatmap(\u003cspan class=\"pl-smi\"\u003eDMP.p05.Region\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003ecluster_cols\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003ecluster_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                      \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ebreaks = myBreaks,\u003c/span\u003e\n                      \u003cspan class=\"pl-v\"\u003ecutree_cols\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\n                      \u003cspan class=\"pl-v\"\u003ecutree_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\n                      \u003cspan class=\"pl-v\"\u003eangle_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e45\u003c/span\u003e,\n                      \u003cspan class=\"pl-v\"\u003efontsize_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\n                      \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emyColor\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003eshow_rownames\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003eborder_color\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003eannotation_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eIndex\u003c/span\u003e,\n                      \u003cspan class=\"pl-v\"\u003eannotation_colors\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eann_colors\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFigure XX: Volcano Plot - DMPs\u003c/h2\u003e\u003ca id=\"user-content-figure-xx-volcano-plot---dmps\" class=\"anchor\" aria-label=\"Permalink: Figure XX: Volcano Plot - DMPs\" href=\"#figure-xx-volcano-plot---dmps\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Load packages\nlibrary(dplyr)\nlibrary(ggplot2)\nlibrary(ggrepel)\nlibrary(openxlsx)\nResults\u0026lt;-read.xlsx(paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;_DMPs.xlsx\u0026quot;), sheet = \u0026quot;Unfiltered\u0026quot;)\n#Read data from the web\nVolcano_data = mutate(Results, sig=ifelse(Results$pval\u0026lt;0.05 \u0026amp; abs(Results$Methylation.Diff)\u0026gt;5, \u0026quot;P\u0026lt;0.05 and |Methylation| \u0026gt; 5%\u0026quot;, \u0026quot;Not Sig\u0026quot;), minuslogpvalue = -log(pval), Methylation=Methylation.Diff)\n\n#Split gene names for labelling\nVolcano_data_split\u0026lt;-Volcano_data %\u0026gt;% mutate(UCSC_RefGene_Name = strsplit(as.character(UCSC_RefGene_Name), \u0026quot;;\u0026quot;)) %\u0026gt;% unnest(UCSC_RefGene_Name) %\u0026gt;% distinct()\n\nmax(Volcano_data$minuslogpvalue, na.rm = TRUE)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Load packages\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eggplot2\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eggrepel\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eopenxlsx\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eResults\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eread.xlsx(paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_DMPs.xlsx\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esheet\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUnfiltered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eRead data from the web\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eVolcano_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e mutate(\u003cspan class=\"pl-smi\"\u003eResults\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esig\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eifelse(\u003cspan class=\"pl-smi\"\u003eResults\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e abs(\u003cspan class=\"pl-smi\"\u003eResults\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMethylation.Diff\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eP\u0026lt;0.05 and |Methylation| \u0026gt; 5%\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNot Sig\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eminuslogpvalue\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003elog(\u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eMethylation\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMethylation.Diff\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSplit gene names for labelling\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eVolcano_data_split\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eVolcano_data\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% mutate(\u003cspan class=\"pl-v\"\u003eUCSC_RefGene_Name\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e strsplit(as.character(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% unnest(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% distinct()\n\nmax(\u003cspan class=\"pl-smi\"\u003eVolcano_data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eminuslogpvalue\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ena.rm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## [1] 15.86107\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## [1] 15.86107\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Results\u0026lt;-Results %\u0026gt;% filter(grepl(\u0026quot;TSS\u0026quot;, UCSC_RefGene_Group))\n#plot the ggplot\np = ggplot(Volcano_data_split, aes(Methylation, minuslogpvalue)) + theme(panel.background = element_rect(\u0026quot;white\u0026quot;, colour = \u0026quot;black\u0026quot;, size=2), panel.grid.major = element_line(colour = \u0026quot;gray50\u0026quot;, size=.75), panel.grid.minor = element_line(colour = \u0026quot;gray50\u0026quot;, size=0.4)) + \ngeom_point(aes(fill=sig, size = minuslogpvalue), colour=\u0026quot;grey\u0026quot;, shape=21, stroke = 0, alpha = 8/10) + labs(x=expression(Methylation_Change), y=expression(-Log[10](P-value))) + xlim(min(Volcano_data$Methylation, na.rm = TRUE),max(Volcano_data$Methylation, na.rm = TRUE))+ ylim(-0, max(Volcano_data$minuslogpvalue, na.rm = TRUE)) +   geom_hline(yintercept = 0, size = 1) + geom_vline(xintercept=0, size=1)+ \n  scale_fill_manual(values=c(\u0026quot;grey\u0026quot;, \u0026quot;darkgoldenrod2\u0026quot;)) +\n  scale_size_continuous(range = c(.25, 4))\npdf(file = paste0(\u0026quot;../2_Output/\u0026quot;, COMPARISON, \u0026quot;/\u0026quot;, COMPARISON, \u0026quot;Volcano.Plot.pdf\u0026quot;), height = 6, width = 10)\np+\n  geom_text_repel(data=top_n(filter(Volcano_data_split, pval\u0026lt;0.05, Methylation \u0026lt; 0, sig!=\u0026quot;Not Sig\u0026quot;), 10, -Methylation), aes(label=UCSC_RefGene_Name)) +\n  geom_text_repel(data=top_n(filter(Volcano_data_split, pval\u0026lt;0.05, Methylation \u0026gt; 0, sig!=\u0026quot;Not Sig\u0026quot;), 10, Methylation), aes(label=UCSC_RefGene_Name))\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Results\u0026lt;-Results %\u0026gt;% filter(grepl(\"TSS\", UCSC_RefGene_Group))\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eplot the ggplot\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003ep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e ggplot(\u003cspan class=\"pl-smi\"\u003eVolcano_data_split\u003c/span\u003e, aes(\u003cspan class=\"pl-smi\"\u003eMethylation\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eminuslogpvalue\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e theme(\u003cspan class=\"pl-v\"\u003epanel.background\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_rect(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003epanel.grid.major\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_line(\u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egray50\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.75\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003epanel.grid.minor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_line(\u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egray50\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \ngeom_point(aes(\u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esig\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eminuslogpvalue\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egrey\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eshape\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e21\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estroke\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e labs(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eexpression(\u003cspan class=\"pl-smi\"\u003eMethylation_Change\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eexpression(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLog\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e](\u003cspan class=\"pl-smi\"\u003eP\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003evalue\u003c/span\u003e))) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e xlim(min(\u003cspan class=\"pl-smi\"\u003eVolcano_data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMethylation\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ena.rm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e),max(\u003cspan class=\"pl-smi\"\u003eVolcano_data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMethylation\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ena.rm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e))\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e ylim(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, max(\u003cspan class=\"pl-smi\"\u003eVolcano_data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eminuslogpvalue\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ena.rm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e   geom_hline(\u003cspan class=\"pl-v\"\u003eyintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_vline(\u003cspan class=\"pl-v\"\u003exintercept\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  scale_fill_manual(\u003cspan class=\"pl-v\"\u003evalues\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egrey\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgoldenrod2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  scale_size_continuous(\u003cspan class=\"pl-v\"\u003erange\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-smi\"\u003e.25\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e))\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCOMPARISON\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVolcano.Plot.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_text_repel(\u003cspan class=\"pl-v\"\u003edata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003etop_n(filter(\u003cspan class=\"pl-smi\"\u003eVolcano_data_split\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eMethylation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esig\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNot Sig\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMethylation\u003c/span\u003e), aes(\u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_text_repel(\u003cspan class=\"pl-v\"\u003edata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003etop_n(filter(\u003cspan class=\"pl-smi\"\u003eVolcano_data_split\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eMethylation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esig\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNot Sig\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eMethylation\u003c/span\u003e), aes(\u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e))\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## quartz_off_screen \n##                 2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## quartz_off_screen \n##                 2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCombined Analysis - Biopsy and iPSC\u003c/h1\u003e\u003ca id=\"user-content-combined-analysis---biopsy-and-ipsc\" class=\"anchor\" aria-label=\"Permalink: Combined Analysis - Biopsy and iPSC\" href=\"#combined-analysis---biopsy-and-ipsc\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(openxlsx)\nlibrary(dplyr)\nlibrary(tidyr)\nlibrary(ComplexHeatmap)\nlibrary(minfi)\nNUMBER = 20\nSTAT = 0.001\n#Index\nAnnoTargets\u0026lt;-read.metharray.sheet(base=\u0026quot;../1_Input/IDAT\u0026quot;, pattern=\u0026quot;.csv\u0026quot;)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eopenxlsx\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003etidyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eComplexHeatmap\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eminfi\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003eNUMBER\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eSTAT\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.001\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eIndex\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eread.metharray.sheet(\u003cspan class=\"pl-v\"\u003ebase\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../1_Input/IDAT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epattern\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## [1] \u0026quot;../1_Input/IDAT/SampleSheet.csv\u0026quot;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## [1] \"../1_Input/IDAT/SampleSheet.csv\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"rownames(AnnoTargets)\u0026lt;-AnnoTargets$Sample_Name\nIndex_combined\u0026lt;-AnnoTargets %\u0026gt;% dplyr::select(Sample_Group, Tissue, Sample_ID)\n#iPSC Data\nDMPs_iPSC\u0026lt;-read.xlsx(\u0026quot;../2_Output/iPSC_DCM_vs_CON/iPSC_DCM_vs_CON_DMPs.xlsx\u0026quot;, sheet = \u0026quot;P_0.05\u0026quot;, rowNames = T)\niPSC_stats\u0026lt;-DMPs_iPSC %\u0026gt;% dplyr::select(Methylation_iPSC = Methylation.Diff, pval_iPSC = pval, adj.P.Val_iPSC = qval, Name:Regulatory_Feature_Group)\n\n\n#biopsies\nDMPs_heart\u0026lt;-read.xlsx(\u0026quot;../2_Output/heart_DCM_vs_CON/heart_DCM_vs_CON_DMPs.xlsx\u0026quot;, sheet = \u0026quot;P_0.05\u0026quot;, rowNames = T)\nheart_stats\u0026lt;-DMPs_heart %\u0026gt;% dplyr::select(Methylation_heart = Methylation.Diff, pval_heart = pval, adj.P.Val_heart = qval, Name:Regulatory_Feature_Group)\nheart_index\u0026lt;-AnnoTargets %\u0026gt;% dplyr::filter(Tissue==\u0026quot;heart\u0026quot;) %\u0026gt;% dplyr::select(Tissue, Sample_Group, Sample_ID)\n\n#Beta\nbeta_all\u0026lt;-read.csv(\u0026quot;../1_Input/beta_all.csv\u0026quot;, row.names = 1)\n\n#combine data\nCombined_DMPs\u0026lt;-inner_join(iPSC_stats, heart_stats)\nCombined_DMPs_Separated\u0026lt;-Combined_DMPs %\u0026gt;% mutate(UCSC_RefGene_Name = strsplit(as.character(UCSC_RefGene_Name), \u0026quot;;\u0026quot;)) %\u0026gt;% unnest(UCSC_RefGene_Name) %\u0026gt;% distinct() %\u0026gt;% merge(., beta_all, by.x = \u0026quot;Name\u0026quot;, by.y = 0)\n\n# Select only co-methylated positions (iPSC and heart)\nCombined_together\u0026lt;-Combined_DMPs_Separated %\u0026gt;% filter(Methylation_iPSC\u0026gt;0 \u0026amp; Methylation_heart\u0026gt;0 | Methylation_iPSC\u0026lt;0 \u0026amp; Methylation_heart\u0026lt;0) %\u0026gt;% arrange(desc(Methylation_iPSC)) %\u0026gt;% filter(pval_iPSC \u0026lt; STAT) %\u0026gt;% arrange(pval_iPSC)\n#Find row number for top candidates\nCandidates\u0026lt;-which(Combined_together$UCSC_RefGene_Name == \u0026quot;ATG7\u0026quot;)\n# Combined_together\u0026lt;-merge(Combined_together, beta_all, by.x = \u0026quot;Name\u0026quot;, by.y = 0)\nrownames(Combined_together)\u0026lt;-make.unique(Combined_together$UCSC_RefGene_Name, sep = \u0026quot;.\u0026quot;)\n#Create heatmap matrices and annotation\nheart_hm\u0026lt;-Combined_together %\u0026gt;% dplyr::select(contains(\u0026quot;NF\u0026quot;) | contains(\u0026quot;biopsy\u0026quot;)) %\u0026gt;% data.matrix()\nheart_index\u0026lt;-AnnoTargets %\u0026gt;% dplyr::filter(Tissue==\u0026quot;heart\u0026quot;) %\u0026gt;% dplyr::select(Tissue, Sample_Group, Sample_ID)\niPSC_hm\u0026lt;-Combined_together %\u0026gt;% dplyr::select(contains(\u0026quot;cells\u0026quot;) \u0026amp; contains(\u0026quot;co\u0026quot;),contains(\u0026quot;cells\u0026quot;) \u0026amp; contains(\u0026quot;go\u0026quot;), contains(\u0026quot;cells\u0026quot;) \u0026amp; contains(\u0026quot;bo\u0026quot;)) %\u0026gt;% data.matrix()\niPSC_index\u0026lt;-AnnoTargets %\u0026gt;% dplyr::filter(Tissue==\u0026quot;iPSC\u0026quot;) %\u0026gt;% dplyr::select(Tissue, Sample_Group, Sample_ID)\nwrite.xlsx(Combined_together, \u0026quot;../2_Output/Overlapping.DMPs_iPSC.Biopsy.xlsx\u0026quot;, overwrite = T)\n\n# Venn Diagram\nlibrary(VennDiagram)\nfutile.logger::flog.threshold(futile.logger::ERROR, name = \u0026quot;VennDiagramLogger\u0026quot;)\"\u003e\u003cpre\u003erownames(\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSample_Name\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eIndex_combined\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eTissue\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSample_ID\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eiPSC Data\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eDMPs_iPSC\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eread.xlsx(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/iPSC_DCM_vs_CON/iPSC_DCM_vs_CON_DMPs.xlsx\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esheet\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eP_0.05\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erowNames\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eiPSC_stats\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDMPs_iPSC\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-v\"\u003eMethylation_iPSC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eMethylation.Diff\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epval_iPSC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadj.P.Val_iPSC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eqval\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eName\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegulatory_Feature_Group\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ebiopsies\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eDMPs_heart\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eread.xlsx(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/heart_DCM_vs_CON/heart_DCM_vs_CON_DMPs.xlsx\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esheet\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eP_0.05\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erowNames\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eheart_stats\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDMPs_heart\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-v\"\u003eMethylation_heart\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eMethylation.Diff\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epval_heart\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epval\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eadj.P.Val_heart\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eqval\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eName\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegulatory_Feature_Group\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eheart_index\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efilter(\u003cspan class=\"pl-smi\"\u003eTissue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eheart\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eTissue\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSample_ID\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eBeta\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ebeta_all\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eread.csv(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../1_Input/beta_all.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecombine data\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eCombined_DMPs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003einner_join(\u003cspan class=\"pl-smi\"\u003eiPSC_stats\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eheart_stats\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eCombined_DMPs_Separated\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCombined_DMPs\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% mutate(\u003cspan class=\"pl-v\"\u003eUCSC_RefGene_Name\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e strsplit(as.character(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e), \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% unnest(\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% distinct() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% merge(\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ebeta_all\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby.x\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eName\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby.y\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Select only co-methylated positions (iPSC and heart)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eCombined_together\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCombined_DMPs_Separated\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% filter(\u003cspan class=\"pl-smi\"\u003eMethylation_iPSC\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eMethylation_heart\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eMethylation_iPSC\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eMethylation_heart\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% arrange(desc(\u003cspan class=\"pl-smi\"\u003eMethylation_iPSC\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% filter(\u003cspan class=\"pl-smi\"\u003epval_iPSC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSTAT\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% arrange(\u003cspan class=\"pl-smi\"\u003epval_iPSC\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eFind row number for top candidates\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eCandidates\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003ewhich(\u003cspan class=\"pl-smi\"\u003eCombined_together\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eATG7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Combined_together\u0026lt;-merge(Combined_together, beta_all, by.x = \"Name\", by.y = 0)\u003c/span\u003e\nrownames(\u003cspan class=\"pl-smi\"\u003eCombined_together\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003emake.unique(\u003cspan class=\"pl-smi\"\u003eCombined_together\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eCreate heatmap matrices and annotation\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eheart_hm\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCombined_together\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebiopsy\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% data.matrix()\n\u003cspan class=\"pl-smi\"\u003eheart_index\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efilter(\u003cspan class=\"pl-smi\"\u003eTissue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eheart\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eTissue\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSample_ID\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eiPSC_hm\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCombined_together\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eco\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ego\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebo\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% data.matrix()\n\u003cspan class=\"pl-smi\"\u003eiPSC_index\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAnnoTargets\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efilter(\u003cspan class=\"pl-smi\"\u003eTissue\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eiPSC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eTissue\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSample_Group\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSample_ID\u003c/span\u003e)\nwrite.xlsx(\u003cspan class=\"pl-smi\"\u003eCombined_together\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/Overlapping.DMPs_iPSC.Biopsy.xlsx\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eoverwrite\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Venn Diagram\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003eVennDiagram\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003efutile.logger\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eflog.threshold(\u003cspan class=\"pl-smi\"\u003efutile.logger\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eERROR\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVennDiagramLogger\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## NULL\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## NULL\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"x\u0026lt;-list(iPSC = DMPs_iPSC$Name, heart = DMPs_heart$Name)\nvenn.diagram(x,fill = c(\u0026quot;darkgray\u0026quot;, \u0026quot;firebrick2\u0026quot;), alpha = c(0.75, 0.75), lty = 'blank', filename = \u0026quot;../2_Output/Overlapping_DMPs_iPSC.Biopsy.png\u0026quot;, na = \u0026quot;remove\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eiPSC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eDMPs_iPSC\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eName\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheart\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eDMPs_heart\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eName\u003c/span\u003e)\nvenn.diagram(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgray\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efirebrick2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0.75\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.75\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elty\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eblank\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/Overlapping_DMPs_iPSC.Biopsy.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ena\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eremove\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## [1] 1\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## [1] 1\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Heatmap\n\nann_colors = list(Sample_Group = c(DCM=\u0026quot;#1b9e77\u0026quot;, CON = \u0026quot;goldenrod2\u0026quot;), Tissue = c(heart = \u0026quot;coral2\u0026quot;, iPSC = \u0026quot;darkgray\u0026quot;), Sample_ID = c(A = \u0026quot;white\u0026quot;, B = \u0026quot;orange\u0026quot;, C = \u0026quot;black\u0026quot;, F = \u0026quot;red\u0026quot;, G = \u0026quot;green\u0026quot;, T = \u0026quot;blue\u0026quot;, V = \u0026quot;purple\u0026quot;))\npaletteLength \u0026lt;- 100\nmyColor \u0026lt;- colorRampPalette(c(\u0026quot;dodgerblue4\u0026quot;, \u0026quot;white\u0026quot;, \u0026quot;gold2\u0026quot;))(paletteLength)\n\n#iPSC\nheatmap_iPSC\u0026lt;-ComplexHeatmap::pheatmap(iPSC_hm, scale=\u0026quot;row\u0026quot;,\n                    cluster_cols = F,\n                    cluster_rows = T,\n                     # angle_col = 45,\n                     fontsize_col = 8,\n                     color = myColor,\n                    annotation_names_col = FALSE,\n                     show_rownames = F,\n                     border_color = NA,\n                    annotation_colors = ann_colors,\n                    annotation_col = iPSC_index)\nheatmap_iPSC\n\n\nha = rowAnnotation(foo = anno_mark(at = c(1:10), labels = Combined_together$UCSC_RefGene_Name[1:10]))\n\n#heart\nheatmap_heart\u0026lt;-ComplexHeatmap::pheatmap(heart_hm, scale = \u0026quot;row\u0026quot;,\n                    cluster_cols = F,\n                    cluster_rows = T,\n                     # angle_col = 45,\n                     fontsize_col = 8,\n                     color = myColor,\n                     show_rownames = F,\n                     border_color = NA,\n                    annotation_names_col = FALSE,\n                    right_annotation = ha,\n                    annotation_colors = ann_colors,\n                    annotation_col = heart_index)\n\nheatmap_heart\nheatmap_iPSC + heatmap_heart\n\npdf(file=\u0026quot;../2_Output/complexheatmap_combined.pdf\u0026quot;, height = 5, width = 7, onefile = F)\nheatmap_iPSC + heatmap_heart\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Heatmap\u003c/span\u003e\n\n\u003cspan class=\"pl-v\"\u003eann_colors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eSample_Group\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-v\"\u003eDCM\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#1b9e77\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eCON\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egoldenrod2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eTissue\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-v\"\u003eheart\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecoral2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eiPSC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgray\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eSample_ID\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-v\"\u003eA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eB\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorange\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eG\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egreen\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eV\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epurple\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003epaletteLength\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emyColor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colorRampPalette(c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edodgerblue4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egold2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))(\u003cspan class=\"pl-smi\"\u003epaletteLength\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eiPSC\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eheatmap_iPSC\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-e\"\u003eComplexHeatmap\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003epheatmap(\u003cspan class=\"pl-smi\"\u003eiPSC_hm\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                    \u003cspan class=\"pl-v\"\u003ecluster_cols\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\n                    \u003cspan class=\"pl-v\"\u003ecluster_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e,\n                     \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e angle_col = 45,\u003c/span\u003e\n                     \u003cspan class=\"pl-v\"\u003efontsize_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\n                     \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emyColor\u003c/span\u003e,\n                    \u003cspan class=\"pl-v\"\u003eannotation_names_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e,\n                     \u003cspan class=\"pl-v\"\u003eshow_rownames\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\n                     \u003cspan class=\"pl-v\"\u003eborder_color\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e,\n                    \u003cspan class=\"pl-v\"\u003eannotation_colors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eann_colors\u003c/span\u003e,\n                    \u003cspan class=\"pl-v\"\u003eannotation_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eiPSC_index\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eheatmap_iPSC\u003c/span\u003e\n\n\n\u003cspan class=\"pl-v\"\u003eha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e rowAnnotation(\u003cspan class=\"pl-v\"\u003efoo\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e anno_mark(\u003cspan class=\"pl-v\"\u003eat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elabels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCombined_together\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eUCSC_RefGene_Name\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e]))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eheart\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eheatmap_heart\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-e\"\u003eComplexHeatmap\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003epheatmap(\u003cspan class=\"pl-smi\"\u003eheart_hm\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                    \u003cspan class=\"pl-v\"\u003ecluster_cols\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\n                    \u003cspan class=\"pl-v\"\u003ecluster_rows\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e,\n                     \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e angle_col = 45,\u003c/span\u003e\n                     \u003cspan class=\"pl-v\"\u003efontsize_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\n                     \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emyColor\u003c/span\u003e,\n                     \u003cspan class=\"pl-v\"\u003eshow_rownames\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\n                     \u003cspan class=\"pl-v\"\u003eborder_color\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e,\n                    \u003cspan class=\"pl-v\"\u003eannotation_names_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e,\n                    \u003cspan class=\"pl-v\"\u003eright_annotation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eha\u003c/span\u003e,\n                    \u003cspan class=\"pl-v\"\u003eannotation_colors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eann_colors\u003c/span\u003e,\n                    \u003cspan class=\"pl-v\"\u003eannotation_col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eheart_index\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eheatmap_heart\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eheatmap_iPSC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eheatmap_heart\u003c/span\u003e\n\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/complexheatmap_combined.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e7\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eonefile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eheatmap_iPSC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eheatmap_heart\u003c/span\u003e\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## quartz_off_screen \n##                 2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## quartz_off_screen \n##                 2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eValidation\u003c/h1\u003e\u003ca id=\"user-content-validation\" class=\"anchor\" aria-label=\"Permalink: Validation\" href=\"#validation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#Import Pepin et al. 2019 Combined methylation-RNA-sequencing dataset\nPepin_DMPs\u0026lt;-read.xlsx(\u0026quot;../1_Input/Validation/HF.Effect_NR_Pre.v.CON_Annotated_DiffMeth.xlsx\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eImport Pepin et al. 2019 Combined methylation-RNA-sequencing dataset\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ePepin_DMPs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eread.xlsx(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../1_Input/Validation/HF.Effect_NR_Pre.v.CON_Annotated_DiffMeth.xlsx\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMotif Enrichment\u003c/h1\u003e\u003ca id=\"user-content-motif-enrichment\" class=\"anchor\" aria-label=\"Permalink: Motif Enrichment\" href=\"#motif-enrichment\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"###### Working example (from BED file)\nlibrary(monaLisa)\nlibrary(GenomicRanges)\nlibrary(SummarizedExperiment)\nlibrary(openxlsx)\nlibrary(dplyr)\nlibrary(TFBSTools)\nlibrary(JASPAR2020)\nlibrary(BSgenome.Hsapiens.UCSC.hg19)\nmcparams \u0026lt;- BiocParallel::MulticoreParam(10L) #parallelization (10-core)\nDMPs_iPSC\u0026lt;-read.xlsx(\u0026quot;../3_Results/Overlapping.DMPs_iPSC.Biopsy.xlsx\u0026quot;, startRow = 2)\nbed\u0026lt;-DMPs_iPSC %\u0026gt;% dplyr::select(chr, pos, Methylation_iPSC) %\u0026gt;% transmute(seqnames=chr, start=pos-20, end=pos+20, width = 41, strand = \u0026quot;*\u0026quot;, deltaMeth=Methylation_iPSC)\nbed_mr\u0026lt;-as(bed, \u0026quot;GRanges\u0026quot;)\n# define bins by differential methylation\nbins \u0026lt;- bin(x = bed_mr$deltaMeth, binmode = \u0026quot;equalN\u0026quot;, nElement = 100, minAbsX = 1)\ntable(bins)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##### Working example (from BED file)\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003emonaLisa\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eGenomicRanges\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eSummarizedExperiment\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eopenxlsx\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eTFBSTools\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eJASPAR2020\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eBSgenome.Hsapiens.UCSC.hg19\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emcparams\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003eBiocParallel\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eMulticoreParam(\u003cspan class=\"pl-c1\"\u003e10L\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eparallelization (10-core)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eDMPs_iPSC\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eread.xlsx(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../3_Results/Overlapping.DMPs_iPSC.Biopsy.xlsx\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estartRow\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ebed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDMPs_iPSC\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epos\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eMethylation_iPSC\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% transmute(\u003cspan class=\"pl-v\"\u003eseqnames\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estart\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003epos\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eend\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003epos\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e41\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003estrand\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e*\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edeltaMeth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eMethylation_iPSC\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ebed_mr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas(\u003cspan class=\"pl-smi\"\u003ebed\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGRanges\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e define bins by differential methylation\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ebins\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e bin(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebed_mr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edeltaMeth\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ebinmode\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eequalN\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enElement\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eminAbsX\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\ntable(\u003cspan class=\"pl-smi\"\u003ebins\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## bins\n## [-6.06,-1.5]  (-1.5,1.02]  (1.02,7.75] \n##          100          226          100\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## bins\n## [-6.06,-1.5]  (-1.5,1.02]  (1.02,7.75] \n##          100          226          100\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"pdf(paste0(\u0026quot;../2_Output/_bins.pdf\u0026quot;), width = 7.5, height = 5)\nplotBinDensity(bed_mr$deltaMeth, bins, legend = \u0026quot;topright\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003epdf(paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/_bins.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e7.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e)\nplotBinDensity(\u003cspan class=\"pl-smi\"\u003ebed_mr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edeltaMeth\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ebins\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elegend\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopright\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## quartz_off_screen \n##                 2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## quartz_off_screen \n##                 2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# get PWMs from JASPAR\npwms \u0026lt;- getMatrixSet(JASPAR2020,\n                     opts = list(matrixtype = \u0026quot;PWM\u0026quot;,\n                                 tax_group = \u0026quot;vertebrates\u0026quot;))\n# trim bed file for sequenes that are consistent\nlmrsel \u0026lt;- trim(resize(bed_mr, width = median(width(bed_mr)), fix = \u0026quot;center\u0026quot;))\n# get sequences from mouse genome\nlmrseqs \u0026lt;- getSeq(BSgenome.Hsapiens.UCSC.hg19, bed_mr)\n# GC proportion (bias)\nplotBinDiagnostics(seqs = lmrseqs, bins = bins, aspect = \u0026quot;GCfrac\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e get PWMs from JASPAR\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003epwms\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e getMatrixSet(\u003cspan class=\"pl-smi\"\u003eJASPAR2020\u003c/span\u003e,\n                     \u003cspan class=\"pl-v\"\u003eopts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003ematrixtype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePWM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                 \u003cspan class=\"pl-v\"\u003etax_group\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evertebrates\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e trim bed file for sequenes that are consistent\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003elmrsel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e trim(resize(\u003cspan class=\"pl-smi\"\u003ebed_mr\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e median(width(\u003cspan class=\"pl-smi\"\u003ebed_mr\u003c/span\u003e)), \u003cspan class=\"pl-v\"\u003efix\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecenter\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e get sequences from mouse genome\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003elmrseqs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e getSeq(\u003cspan class=\"pl-smi\"\u003eBSgenome.Hsapiens.UCSC.hg19\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ebed_mr\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e GC proportion (bias)\u003c/span\u003e\nplotBinDiagnostics(\u003cspan class=\"pl-v\"\u003eseqs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003elmrseqs\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ebins\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebins\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003easpect\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGCfrac\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/motif-1.png\"\u003e\u003cimg src=\"/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/motif-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"plotBinDiagnostics(seqs = lmrseqs, bins = bins, aspect = \u0026quot;dinucfreq\u0026quot;)\n# run motif enrichment\nse \u0026lt;- calcBinnedMotifEnrR(seqs = lmrseqs, bins = bins, pwmL = pwms, BPPARAM = BiocParallel::MulticoreParam(8))\n# Filter results\nTest\u0026lt;-as.data.frame(assays(se))\nsel \u0026lt;- apply(assay(se, \u0026quot;negLog10P\u0026quot;), 1, \n             function(x) max(abs(x), 0, na.rm = TRUE)) \u0026gt; 1.0\nsum(sel)\"\u003e\u003cpre\u003eplotBinDiagnostics(\u003cspan class=\"pl-v\"\u003eseqs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003elmrseqs\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ebins\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebins\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003easpect\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edinucfreq\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e run motif enrichment\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ese\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e calcBinnedMotifEnrR(\u003cspan class=\"pl-v\"\u003eseqs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003elmrseqs\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ebins\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebins\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epwmL\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epwms\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eBPPARAM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-e\"\u003eBiocParallel\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eMulticoreParam(\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Filter results\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eTest\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eas.data.frame(assays(\u003cspan class=\"pl-smi\"\u003ese\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003esel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e apply(assay(\u003cspan class=\"pl-smi\"\u003ese\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enegLog10P\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \n             \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e) max(abs(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e), \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ena.rm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1.0\u003c/span\u003e\nsum(\u003cspan class=\"pl-smi\"\u003esel\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## [1] 65\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## [1] 65\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#\u0026gt; [1] 59\nseSel \u0026lt;- se[sel, ]\n\n# plot\npdf(\u0026quot;../2_Output/Motifs.pdf\u0026quot;, width = 11, height = 10)\nplotMotifHeatmaps(x = seSel, which.plots = c(\u0026quot;log2enr\u0026quot;, \u0026quot;negLog10P\u0026quot;), \n                  width = 2.0, cluster = TRUE, maxEnr = 2, maxSig = 10, \n                  show_motif_GC = TRUE, show_dendrogram = T,show_seqlogo = TRUE)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; [1] 59\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseSel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ese\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003esel\u003c/span\u003e, ]\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e plot\u003c/span\u003e\npdf(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../2_Output/Motifs.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e11\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e)\nplotMotifHeatmaps(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseSel\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewhich.plots\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elog2enr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enegLog10P\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n                  \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2.0\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecluster\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emaxEnr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emaxSig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \n                  \u003cspan class=\"pl-v\"\u003eshow_motif_GC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eshow_dendrogram\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eshow_seqlogo\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## quartz_off_screen \n##                 2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## quartz_off_screen \n##                 2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"plotMotifHeatmaps(x = seSel, which.plots = c(\u0026quot;log2enr\u0026quot;, \u0026quot;negLog10P\u0026quot;), \n                  width = 2.0, cluster = TRUE, maxEnr = 2, maxSig = 10, \n                  show_motif_GC = TRUE, show_dendrogram = T,show_seqlogo = TRUE)\"\u003e\u003cpre\u003eplotMotifHeatmaps(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseSel\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewhich.plots\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elog2enr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enegLog10P\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n                  \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2.0\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecluster\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emaxEnr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emaxSig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \n                  \u003cspan class=\"pl-v\"\u003eshow_motif_GC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eshow_dendrogram\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eshow_seqlogo\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSupplemental Table: R Session Information\u003c/h1\u003e\u003ca id=\"user-content-supplemental-table-r-session-information\" class=\"anchor\" aria-label=\"Permalink: Supplemental Table: R Session Information\" href=\"#supplemental-table-r-session-information\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAll packages and setting are acquired using the following command:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"options(kableExtra.latex.load_packages = FALSE)\nRun_tE\u0026lt;-Sys.time()\nRun_time\u0026lt;-Run_tE - Run_tS\nRun_time\"\u003e\u003cpre\u003eoptions(\u003cspan class=\"pl-v\"\u003ekableExtra.latex.load_packages\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eRun_tE\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003eSys.time()\n\u003cspan class=\"pl-smi\"\u003eRun_time\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRun_tE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eRun_tS\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eRun_time\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## Time difference of 14.97141 mins\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## Time difference of 14.97141 mins\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(kableExtra)\nsinfo\u0026lt;-devtools::session_info()\nsinfo$platform\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003ekableExtra\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003esinfo\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e\u003cspan class=\"pl-e\"\u003edevtools\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003esession_info()\n\u003cspan class=\"pl-smi\"\u003esinfo\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eplatform\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"##  setting  value\n##  version  R version 4.1.2 (2021-11-01)\n##  os       macOS Big Sur 10.16\n##  system   x86_64, darwin17.0\n##  ui       X11\n##  language (EN)\n##  collate  en_US.UTF-8\n##  ctype    en_US.UTF-8\n##  tz       Europe/Berlin\n##  date     2021-12-31\n##  pandoc   2.14.0.3 @ /Applications/RStudio.app/Contents/MacOS/pandoc/ (via rmarkdown)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e##  setting  value\n##  version  R version 4.1.2 (2021-11-01)\n##  os       macOS Big Sur 10.16\n##  system   x86_64, darwin17.0\n##  ui       X11\n##  language (EN)\n##  collate  en_US.UTF-8\n##  ctype    en_US.UTF-8\n##  tz       Europe/Berlin\n##  date     2021-12-31\n##  pandoc   2.14.0.3 @ /Applications/RStudio.app/Contents/MacOS/pandoc/ (via rmarkdown)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"sinfo$packages %\u0026gt;% kable( \n                         align=\u0026quot;c\u0026quot;, \n                         caption=\u0026quot;Packages and Required Dependencies\u0026quot;) %\u0026gt;% \n    kable_styling(latex_options=c(\u0026quot;repeat_header\u0026quot;, \u0026quot;condensed\u0026quot;))\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003esinfo\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003epackages\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% kable( \n                         \u003cspan class=\"pl-v\"\u003ealign\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ec\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-v\"\u003ecaption\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePackages and Required Dependencies\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    kable_styling(\u003cspan class=\"pl-v\"\u003elatex_options\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erepeat_header\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econdensed\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\u003c/pre\u003e\u003c/div\u003e\n\nPackages and Required Dependencies\n \u003ctable\u003e\u003cthead\u003e\n  \u003ctr\u003e\n   \u003cth\u003e   \u003c/th\u003e\n   \u003cth\u003e package \u003c/th\u003e\n   \u003cth\u003e ondiskversion \u003c/th\u003e\n   \u003cth\u003e loadedversion \u003c/th\u003e\n   \u003cth\u003e path \u003c/th\u003e\n   \u003cth\u003e loadedpath \u003c/th\u003e\n   \u003cth\u003e attached \u003c/th\u003e\n   \u003cth\u003e is_base \u003c/th\u003e\n   \u003cth\u003e date \u003c/th\u003e\n   \u003cth\u003e source \u003c/th\u003e\n   \u003cth\u003e md5ok \u003c/th\u003e\n   \u003cth\u003e library \u003c/th\u003e\n  \u003c/tr\u003e\n \u003c/thead\u003e\n\u003ctbody\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e abind \u003c/td\u003e\n   \u003ctd\u003e abind \u003c/td\u003e\n   \u003ctd\u003e 1.4.5 \u003c/td\u003e\n   \u003ctd\u003e 1.4-5 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/abind \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/abind \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2016-07-21 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e annotate \u003c/td\u003e\n   \u003ctd\u003e annotate \u003c/td\u003e\n   \u003ctd\u003e 1.72.0 \u003c/td\u003e\n   \u003ctd\u003e 1.72.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/annotate \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/annotate \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e AnnotationDbi \u003c/td\u003e\n   \u003ctd\u003e AnnotationDbi \u003c/td\u003e\n   \u003ctd\u003e 1.56.2 \u003c/td\u003e\n   \u003ctd\u003e 1.56.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/AnnotationDbi \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/AnnotationDbi \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-09 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e askpass \u003c/td\u003e\n   \u003ctd\u003e askpass \u003c/td\u003e\n   \u003ctd\u003e 1.1 \u003c/td\u003e\n   \u003ctd\u003e 1.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/askpass \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/askpass \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-01-13 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e assertthat \u003c/td\u003e\n   \u003ctd\u003e assertthat \u003c/td\u003e\n   \u003ctd\u003e 0.2.1 \u003c/td\u003e\n   \u003ctd\u003e 0.2.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/assertthat \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/assertthat \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-03-21 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e backports \u003c/td\u003e\n   \u003ctd\u003e backports \u003c/td\u003e\n   \u003ctd\u003e 1.4.0 \u003c/td\u003e\n   \u003ctd\u003e 1.4.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/backports \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/backports \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-23 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e base64 \u003c/td\u003e\n   \u003ctd\u003e base64 \u003c/td\u003e\n   \u003ctd\u003e 2.0 \u003c/td\u003e\n   \u003ctd\u003e 2.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/base64 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/base64 \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2016-05-10 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e base64enc \u003c/td\u003e\n   \u003ctd\u003e base64enc \u003c/td\u003e\n   \u003ctd\u003e 0.1.3 \u003c/td\u003e\n   \u003ctd\u003e 0.1-3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/base64enc \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/base64enc \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2015-07-28 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e beanplot \u003c/td\u003e\n   \u003ctd\u003e beanplot \u003c/td\u003e\n   \u003ctd\u003e 1.2 \u003c/td\u003e\n   \u003ctd\u003e 1.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/beanplot \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/beanplot \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2014-09-19 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e Biobase \u003c/td\u003e\n   \u003ctd\u003e Biobase \u003c/td\u003e\n   \u003ctd\u003e 2.54.0 \u003c/td\u003e\n   \u003ctd\u003e 2.54.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Biobase \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Biobase \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e BiocFileCache \u003c/td\u003e\n   \u003ctd\u003e BiocFileCache \u003c/td\u003e\n   \u003ctd\u003e 2.2.0 \u003c/td\u003e\n   \u003ctd\u003e 2.2.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocFileCache \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocFileCache \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e BiocGenerics \u003c/td\u003e\n   \u003ctd\u003e BiocGenerics \u003c/td\u003e\n   \u003ctd\u003e 0.40.0 \u003c/td\u003e\n   \u003ctd\u003e 0.40.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocGenerics \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocGenerics \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e BiocIO \u003c/td\u003e\n   \u003ctd\u003e BiocIO \u003c/td\u003e\n   \u003ctd\u003e 1.4.0 \u003c/td\u003e\n   \u003ctd\u003e 1.4.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocIO \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocIO \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e BiocParallel \u003c/td\u003e\n   \u003ctd\u003e BiocParallel \u003c/td\u003e\n   \u003ctd\u003e 1.28.2 \u003c/td\u003e\n   \u003ctd\u003e 1.28.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocParallel \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocParallel \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-25 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e biomaRt \u003c/td\u003e\n   \u003ctd\u003e biomaRt \u003c/td\u003e\n   \u003ctd\u003e 2.50.1 \u003c/td\u003e\n   \u003ctd\u003e 2.50.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/biomaRt \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/biomaRt \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-21 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e Biostrings \u003c/td\u003e\n   \u003ctd\u003e Biostrings \u003c/td\u003e\n   \u003ctd\u003e 2.62.0 \u003c/td\u003e\n   \u003ctd\u003e 2.62.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Biostrings \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Biostrings \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e bit \u003c/td\u003e\n   \u003ctd\u003e bit \u003c/td\u003e\n   \u003ctd\u003e 4.0.4 \u003c/td\u003e\n   \u003ctd\u003e 4.0.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bit \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bit \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-08-04 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e bit64 \u003c/td\u003e\n   \u003ctd\u003e bit64 \u003c/td\u003e\n   \u003ctd\u003e 4.0.5 \u003c/td\u003e\n   \u003ctd\u003e 4.0.5 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bit64 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bit64 \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-08-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e bitops \u003c/td\u003e\n   \u003ctd\u003e bitops \u003c/td\u003e\n   \u003ctd\u003e 1.0.7 \u003c/td\u003e\n   \u003ctd\u003e 1.0-7 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bitops \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bitops \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-04-24 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e blob \u003c/td\u003e\n   \u003ctd\u003e blob \u003c/td\u003e\n   \u003ctd\u003e 1.2.2 \u003c/td\u003e\n   \u003ctd\u003e 1.2.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/blob \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/blob \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-07-23 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e broom \u003c/td\u003e\n   \u003ctd\u003e broom \u003c/td\u003e\n   \u003ctd\u003e 0.7.10 \u003c/td\u003e\n   \u003ctd\u003e 0.7.10 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/broom \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/broom \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-31 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e BSgenome \u003c/td\u003e\n   \u003ctd\u003e BSgenome \u003c/td\u003e\n   \u003ctd\u003e 1.62.0 \u003c/td\u003e\n   \u003ctd\u003e 1.62.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BSgenome \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BSgenome \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e BSgenome.Hsapiens.UCSC.hg19 \u003c/td\u003e\n   \u003ctd\u003e BSgenome.Hsapiens.UCSC.hg19 \u003c/td\u003e\n   \u003ctd\u003e 1.4.3 \u003c/td\u003e\n   \u003ctd\u003e 1.4.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BSgenome.Hsapiens.UCSC.hg19 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BSgenome.Hsapiens.UCSC.hg19 \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-12-02 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e bslib \u003c/td\u003e\n   \u003ctd\u003e bslib \u003c/td\u003e\n   \u003ctd\u003e 0.3.1 \u003c/td\u003e\n   \u003ctd\u003e 0.3.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bslib \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bslib \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-06 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e bumphunter \u003c/td\u003e\n   \u003ctd\u003e bumphunter \u003c/td\u003e\n   \u003ctd\u003e 1.36.0 \u003c/td\u003e\n   \u003ctd\u003e 1.36.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bumphunter \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bumphunter \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e cachem \u003c/td\u003e\n   \u003ctd\u003e cachem \u003c/td\u003e\n   \u003ctd\u003e 1.0.6 \u003c/td\u003e\n   \u003ctd\u003e 1.0.6 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cachem \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cachem \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-08-19 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e calibrate \u003c/td\u003e\n   \u003ctd\u003e calibrate \u003c/td\u003e\n   \u003ctd\u003e 1.7.7 \u003c/td\u003e\n   \u003ctd\u003e 1.7.7 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/calibrate \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/calibrate \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-06-19 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e callr \u003c/td\u003e\n   \u003ctd\u003e callr \u003c/td\u003e\n   \u003ctd\u003e 3.7.0 \u003c/td\u003e\n   \u003ctd\u003e 3.7.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/callr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/callr \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-04-20 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e car \u003c/td\u003e\n   \u003ctd\u003e car \u003c/td\u003e\n   \u003ctd\u003e 3.0.12 \u003c/td\u003e\n   \u003ctd\u003e 3.0-12 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/car \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/car \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-06 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e carData \u003c/td\u003e\n   \u003ctd\u003e carData \u003c/td\u003e\n   \u003ctd\u003e 3.0.4 \u003c/td\u003e\n   \u003ctd\u003e 3.0-4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/carData \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/carData \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-05-22 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e caTools \u003c/td\u003e\n   \u003ctd\u003e caTools \u003c/td\u003e\n   \u003ctd\u003e 1.18.2 \u003c/td\u003e\n   \u003ctd\u003e 1.18.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/caTools \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/caTools \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-03-28 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e cellranger \u003c/td\u003e\n   \u003ctd\u003e cellranger \u003c/td\u003e\n   \u003ctd\u003e 1.1.0 \u003c/td\u003e\n   \u003ctd\u003e 1.1.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cellranger \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cellranger \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2016-07-27 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e checkmate \u003c/td\u003e\n   \u003ctd\u003e checkmate \u003c/td\u003e\n   \u003ctd\u003e 2.0.0 \u003c/td\u003e\n   \u003ctd\u003e 2.0.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/checkmate \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/checkmate \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-02-06 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e circlize \u003c/td\u003e\n   \u003ctd\u003e circlize \u003c/td\u003e\n   \u003ctd\u003e 0.4.13 \u003c/td\u003e\n   \u003ctd\u003e 0.4.13 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/circlize \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/circlize \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-06-09 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e cli \u003c/td\u003e\n   \u003ctd\u003e cli \u003c/td\u003e\n   \u003ctd\u003e 3.1.0 \u003c/td\u003e\n   \u003ctd\u003e 3.1.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cli \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cli \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-27 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e clue \u003c/td\u003e\n   \u003ctd\u003e clue \u003c/td\u003e\n   \u003ctd\u003e 0.3.60 \u003c/td\u003e\n   \u003ctd\u003e 0.3-60 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/clue \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/clue \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-11 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e cluster \u003c/td\u003e\n   \u003ctd\u003e cluster \u003c/td\u003e\n   \u003ctd\u003e 2.1.2 \u003c/td\u003e\n   \u003ctd\u003e 2.1.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cluster \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cluster \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-04-17 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e CNEr \u003c/td\u003e\n   \u003ctd\u003e CNEr \u003c/td\u003e\n   \u003ctd\u003e 1.30.0 \u003c/td\u003e\n   \u003ctd\u003e 1.30.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/CNEr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/CNEr \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e codetools \u003c/td\u003e\n   \u003ctd\u003e codetools \u003c/td\u003e\n   \u003ctd\u003e 0.2.18 \u003c/td\u003e\n   \u003ctd\u003e 0.2-18 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/codetools \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/codetools \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-11-04 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e colorspace \u003c/td\u003e\n   \u003ctd\u003e colorspace \u003c/td\u003e\n   \u003ctd\u003e 2.0.2 \u003c/td\u003e\n   \u003ctd\u003e 2.0-2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/colorspace \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/colorspace \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-06-24 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e ComplexHeatmap \u003c/td\u003e\n   \u003ctd\u003e ComplexHeatmap \u003c/td\u003e\n   \u003ctd\u003e 2.11.1 \u003c/td\u003e\n   \u003ctd\u003e 2.11.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ComplexHeatmap \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ComplexHeatmap \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-26 \u003c/td\u003e\n   \u003ctd\u003e Github (jokergoo/ComplexHeatmap@826b321) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e corrplot \u003c/td\u003e\n   \u003ctd\u003e corrplot \u003c/td\u003e\n   \u003ctd\u003e 0.92 \u003c/td\u003e\n   \u003ctd\u003e 0.92 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/corrplot \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/corrplot \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-18 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e cowplot \u003c/td\u003e\n   \u003ctd\u003e cowplot \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cowplot \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cowplot \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-12-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e crayon \u003c/td\u003e\n   \u003ctd\u003e crayon \u003c/td\u003e\n   \u003ctd\u003e 1.4.2 \u003c/td\u003e\n   \u003ctd\u003e 1.4.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/crayon \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/crayon \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-29 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e crosstalk \u003c/td\u003e\n   \u003ctd\u003e crosstalk \u003c/td\u003e\n   \u003ctd\u003e 1.2.0 \u003c/td\u003e\n   \u003ctd\u003e 1.2.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/crosstalk \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/crosstalk \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-04 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e curl \u003c/td\u003e\n   \u003ctd\u003e curl \u003c/td\u003e\n   \u003ctd\u003e 4.3.2 \u003c/td\u003e\n   \u003ctd\u003e 4.3.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/curl \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/curl \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-06-23 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e data.table \u003c/td\u003e\n   \u003ctd\u003e data.table \u003c/td\u003e\n   \u003ctd\u003e 1.14.2 \u003c/td\u003e\n   \u003ctd\u003e 1.14.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/data.table \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/data.table \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-27 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e DBI \u003c/td\u003e\n   \u003ctd\u003e DBI \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DBI \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DBI \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-01-15 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e dbplyr \u003c/td\u003e\n   \u003ctd\u003e dbplyr \u003c/td\u003e\n   \u003ctd\u003e 2.1.1 \u003c/td\u003e\n   \u003ctd\u003e 2.1.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/dbplyr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/dbplyr \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-04-06 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e DelayedArray \u003c/td\u003e\n   \u003ctd\u003e DelayedArray \u003c/td\u003e\n   \u003ctd\u003e 0.20.0 \u003c/td\u003e\n   \u003ctd\u003e 0.20.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DelayedArray \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DelayedArray \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e DelayedMatrixStats \u003c/td\u003e\n   \u003ctd\u003e DelayedMatrixStats \u003c/td\u003e\n   \u003ctd\u003e 1.16.0 \u003c/td\u003e\n   \u003ctd\u003e 1.16.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DelayedMatrixStats \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DelayedMatrixStats \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e desc \u003c/td\u003e\n   \u003ctd\u003e desc \u003c/td\u003e\n   \u003ctd\u003e 1.4.0 \u003c/td\u003e\n   \u003ctd\u003e 1.4.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/desc \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/desc \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-28 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e DESeq2 \u003c/td\u003e\n   \u003ctd\u003e DESeq2 \u003c/td\u003e\n   \u003ctd\u003e 1.34.0 \u003c/td\u003e\n   \u003ctd\u003e 1.34.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DESeq2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DESeq2 \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e devtools \u003c/td\u003e\n   \u003ctd\u003e devtools \u003c/td\u003e\n   \u003ctd\u003e 2.4.3 \u003c/td\u003e\n   \u003ctd\u003e 2.4.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/devtools \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/devtools \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e digest \u003c/td\u003e\n   \u003ctd\u003e digest \u003c/td\u003e\n   \u003ctd\u003e 0.6.29 \u003c/td\u003e\n   \u003ctd\u003e 0.6.29 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/digest \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/digest \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-12-01 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e DirichletMultinomial \u003c/td\u003e\n   \u003ctd\u003e DirichletMultinomial \u003c/td\u003e\n   \u003ctd\u003e 1.36.0 \u003c/td\u003e\n   \u003ctd\u003e 1.36.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DirichletMultinomial \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DirichletMultinomial \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e doParallel \u003c/td\u003e\n   \u003ctd\u003e doParallel \u003c/td\u003e\n   \u003ctd\u003e 1.0.16 \u003c/td\u003e\n   \u003ctd\u003e 1.0.16 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/doParallel \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/doParallel \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-10-16 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e doRNG \u003c/td\u003e\n   \u003ctd\u003e doRNG \u003c/td\u003e\n   \u003ctd\u003e 1.8.2 \u003c/td\u003e\n   \u003ctd\u003e 1.8.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/doRNG \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/doRNG \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-01-27 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e dplyr \u003c/td\u003e\n   \u003ctd\u003e dplyr \u003c/td\u003e\n   \u003ctd\u003e 1.0.7 \u003c/td\u003e\n   \u003ctd\u003e 1.0.7 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/dplyr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/dplyr \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-06-18 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e ellipsis \u003c/td\u003e\n   \u003ctd\u003e ellipsis \u003c/td\u003e\n   \u003ctd\u003e 0.3.2 \u003c/td\u003e\n   \u003ctd\u003e 0.3.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ellipsis \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ellipsis \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-04-29 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e evaluate \u003c/td\u003e\n   \u003ctd\u003e evaluate \u003c/td\u003e\n   \u003ctd\u003e 0.14 \u003c/td\u003e\n   \u003ctd\u003e 0.14 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/evaluate \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/evaluate \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-05-28 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e fansi \u003c/td\u003e\n   \u003ctd\u003e fansi \u003c/td\u003e\n   \u003ctd\u003e 0.5.0 \u003c/td\u003e\n   \u003ctd\u003e 0.5.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fansi \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fansi \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-05-25 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e farver \u003c/td\u003e\n   \u003ctd\u003e farver \u003c/td\u003e\n   \u003ctd\u003e 2.1.0 \u003c/td\u003e\n   \u003ctd\u003e 2.1.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/farver \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/farver \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-02-28 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e fastmap \u003c/td\u003e\n   \u003ctd\u003e fastmap \u003c/td\u003e\n   \u003ctd\u003e 1.1.0 \u003c/td\u003e\n   \u003ctd\u003e 1.1.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fastmap \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fastmap \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-01-25 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e ff \u003c/td\u003e\n   \u003ctd\u003e ff \u003c/td\u003e\n   \u003ctd\u003e 4.0.5 \u003c/td\u003e\n   \u003ctd\u003e 4.0.5 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ff \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ff \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-29 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e filelock \u003c/td\u003e\n   \u003ctd\u003e filelock \u003c/td\u003e\n   \u003ctd\u003e 1.0.2 \u003c/td\u003e\n   \u003ctd\u003e 1.0.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/filelock \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/filelock \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2018-10-05 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e foreach \u003c/td\u003e\n   \u003ctd\u003e foreach \u003c/td\u003e\n   \u003ctd\u003e 1.5.1 \u003c/td\u003e\n   \u003ctd\u003e 1.5.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/foreach \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/foreach \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-10-15 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e foreign \u003c/td\u003e\n   \u003ctd\u003e foreign \u003c/td\u003e\n   \u003ctd\u003e 0.8.81 \u003c/td\u003e\n   \u003ctd\u003e 0.8-81 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/foreign \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/foreign \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-12-22 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e formatR \u003c/td\u003e\n   \u003ctd\u003e formatR \u003c/td\u003e\n   \u003ctd\u003e 1.11 \u003c/td\u003e\n   \u003ctd\u003e 1.11 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/formatR \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/formatR \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-06-01 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e Formula \u003c/td\u003e\n   \u003ctd\u003e Formula \u003c/td\u003e\n   \u003ctd\u003e 1.2.4 \u003c/td\u003e\n   \u003ctd\u003e 1.2-4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Formula \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Formula \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-10-16 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e fs \u003c/td\u003e\n   \u003ctd\u003e fs \u003c/td\u003e\n   \u003ctd\u003e 1.5.1 \u003c/td\u003e\n   \u003ctd\u003e 1.5.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fs \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fs \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e futile.logger \u003c/td\u003e\n   \u003ctd\u003e futile.logger \u003c/td\u003e\n   \u003ctd\u003e 1.4.3 \u003c/td\u003e\n   \u003ctd\u003e 1.4.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/futile.logger \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/futile.logger \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2016-07-10 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e futile.options \u003c/td\u003e\n   \u003ctd\u003e futile.options \u003c/td\u003e\n   \u003ctd\u003e 1.0.1 \u003c/td\u003e\n   \u003ctd\u003e 1.0.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/futile.options \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/futile.options \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2018-04-20 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e genefilter \u003c/td\u003e\n   \u003ctd\u003e genefilter \u003c/td\u003e\n   \u003ctd\u003e 1.76.0 \u003c/td\u003e\n   \u003ctd\u003e 1.76.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/genefilter \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/genefilter \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e geneplotter \u003c/td\u003e\n   \u003ctd\u003e geneplotter \u003c/td\u003e\n   \u003ctd\u003e 1.72.0 \u003c/td\u003e\n   \u003ctd\u003e 1.72.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/geneplotter \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/geneplotter \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e generics \u003c/td\u003e\n   \u003ctd\u003e generics \u003c/td\u003e\n   \u003ctd\u003e 0.1.1 \u003c/td\u003e\n   \u003ctd\u003e 0.1.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/generics \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/generics \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-25 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e GenomeInfoDb \u003c/td\u003e\n   \u003ctd\u003e GenomeInfoDb \u003c/td\u003e\n   \u003ctd\u003e 1.30.0 \u003c/td\u003e\n   \u003ctd\u003e 1.30.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomeInfoDb \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomeInfoDb \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e GenomeInfoDbData \u003c/td\u003e\n   \u003ctd\u003e GenomeInfoDbData \u003c/td\u003e\n   \u003ctd\u003e 1.2.7 \u003c/td\u003e\n   \u003ctd\u003e 1.2.7 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomeInfoDbData \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomeInfoDbData \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-17 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e GenomicAlignments \u003c/td\u003e\n   \u003ctd\u003e GenomicAlignments \u003c/td\u003e\n   \u003ctd\u003e 1.30.0 \u003c/td\u003e\n   \u003ctd\u003e 1.30.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicAlignments \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicAlignments \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e GenomicFeatures \u003c/td\u003e\n   \u003ctd\u003e GenomicFeatures \u003c/td\u003e\n   \u003ctd\u003e 1.46.1 \u003c/td\u003e\n   \u003ctd\u003e 1.46.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicFeatures \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicFeatures \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-27 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e GenomicRanges \u003c/td\u003e\n   \u003ctd\u003e GenomicRanges \u003c/td\u003e\n   \u003ctd\u003e 1.46.1 \u003c/td\u003e\n   \u003ctd\u003e 1.46.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicRanges \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicRanges \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-18 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e GEOquery \u003c/td\u003e\n   \u003ctd\u003e GEOquery \u003c/td\u003e\n   \u003ctd\u003e 2.62.1 \u003c/td\u003e\n   \u003ctd\u003e 2.62.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GEOquery \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GEOquery \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-16 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e GetoptLong \u003c/td\u003e\n   \u003ctd\u003e GetoptLong \u003c/td\u003e\n   \u003ctd\u003e 1.0.5 \u003c/td\u003e\n   \u003ctd\u003e 1.0.5 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GetoptLong \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GetoptLong \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-12-15 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e ggplot2 \u003c/td\u003e\n   \u003ctd\u003e ggplot2 \u003c/td\u003e\n   \u003ctd\u003e 3.3.5 \u003c/td\u003e\n   \u003ctd\u003e 3.3.5 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggplot2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggplot2 \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-06-25 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e ggpubr \u003c/td\u003e\n   \u003ctd\u003e ggpubr \u003c/td\u003e\n   \u003ctd\u003e 0.4.0 \u003c/td\u003e\n   \u003ctd\u003e 0.4.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggpubr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggpubr \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-06-27 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e ggrepel \u003c/td\u003e\n   \u003ctd\u003e ggrepel \u003c/td\u003e\n   \u003ctd\u003e 0.9.1 \u003c/td\u003e\n   \u003ctd\u003e 0.9.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggrepel \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggrepel \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-01-15 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e ggsignif \u003c/td\u003e\n   \u003ctd\u003e ggsignif \u003c/td\u003e\n   \u003ctd\u003e 0.6.3 \u003c/td\u003e\n   \u003ctd\u003e 0.6.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggsignif \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggsignif \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-09 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e glmnet \u003c/td\u003e\n   \u003ctd\u003e glmnet \u003c/td\u003e\n   \u003ctd\u003e 4.1.3 \u003c/td\u003e\n   \u003ctd\u003e 4.1-3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/glmnet \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/glmnet \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-02 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e GlobalOptions \u003c/td\u003e\n   \u003ctd\u003e GlobalOptions \u003c/td\u003e\n   \u003ctd\u003e 0.1.2 \u003c/td\u003e\n   \u003ctd\u003e 0.1.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GlobalOptions \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GlobalOptions \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-06-10 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e glue \u003c/td\u003e\n   \u003ctd\u003e glue \u003c/td\u003e\n   \u003ctd\u003e 1.5.1 \u003c/td\u003e\n   \u003ctd\u003e 1.5.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/glue \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/glue \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e GO.db \u003c/td\u003e\n   \u003ctd\u003e GO.db \u003c/td\u003e\n   \u003ctd\u003e 3.14.0 \u003c/td\u003e\n   \u003ctd\u003e 3.14.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GO.db \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GO.db \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-12-06 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e gridExtra \u003c/td\u003e\n   \u003ctd\u003e gridExtra \u003c/td\u003e\n   \u003ctd\u003e 2.3 \u003c/td\u003e\n   \u003ctd\u003e 2.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gridExtra \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gridExtra \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2017-09-09 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e gtable \u003c/td\u003e\n   \u003ctd\u003e gtable \u003c/td\u003e\n   \u003ctd\u003e 0.3.0 \u003c/td\u003e\n   \u003ctd\u003e 0.3.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gtable \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gtable \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-03-25 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e gtools \u003c/td\u003e\n   \u003ctd\u003e gtools \u003c/td\u003e\n   \u003ctd\u003e 3.9.2 \u003c/td\u003e\n   \u003ctd\u003e 3.9.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gtools \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gtools \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-06-06 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e Haplin \u003c/td\u003e\n   \u003ctd\u003e Haplin \u003c/td\u003e\n   \u003ctd\u003e 7.2.3 \u003c/td\u003e\n   \u003ctd\u003e 7.2.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Haplin \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Haplin \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-09-07 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e HDF5Array \u003c/td\u003e\n   \u003ctd\u003e HDF5Array \u003c/td\u003e\n   \u003ctd\u003e 1.22.1 \u003c/td\u003e\n   \u003ctd\u003e 1.22.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/HDF5Array \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/HDF5Array \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-14 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e highr \u003c/td\u003e\n   \u003ctd\u003e highr \u003c/td\u003e\n   \u003ctd\u003e 0.9 \u003c/td\u003e\n   \u003ctd\u003e 0.9 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/highr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/highr \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-04-16 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e Hmisc \u003c/td\u003e\n   \u003ctd\u003e Hmisc \u003c/td\u003e\n   \u003ctd\u003e 4.6.0 \u003c/td\u003e\n   \u003ctd\u003e 4.6-0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Hmisc \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Hmisc \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-07 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e hms \u003c/td\u003e\n   \u003ctd\u003e hms \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/hms \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/hms \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-26 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e htmlTable \u003c/td\u003e\n   \u003ctd\u003e htmlTable \u003c/td\u003e\n   \u003ctd\u003e 2.3.0 \u003c/td\u003e\n   \u003ctd\u003e 2.3.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmlTable \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmlTable \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-12 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e htmltools \u003c/td\u003e\n   \u003ctd\u003e htmltools \u003c/td\u003e\n   \u003ctd\u003e 0.5.2 \u003c/td\u003e\n   \u003ctd\u003e 0.5.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmltools \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmltools \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-08-25 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e htmlwidgets \u003c/td\u003e\n   \u003ctd\u003e htmlwidgets \u003c/td\u003e\n   \u003ctd\u003e 1.5.4 \u003c/td\u003e\n   \u003ctd\u003e 1.5.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmlwidgets \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmlwidgets \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-08 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e httpuv \u003c/td\u003e\n   \u003ctd\u003e httpuv \u003c/td\u003e\n   \u003ctd\u003e 1.6.3 \u003c/td\u003e\n   \u003ctd\u003e 1.6.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/httpuv \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/httpuv \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-09 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e httr \u003c/td\u003e\n   \u003ctd\u003e httr \u003c/td\u003e\n   \u003ctd\u003e 1.4.2 \u003c/td\u003e\n   \u003ctd\u003e 1.4.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/httr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/httr \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-07-20 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e IlluminaHumanMethylation450kanno.ilmn12.hg19 \u003c/td\u003e\n   \u003ctd\u003e IlluminaHumanMethylation450kanno.ilmn12.hg19 \u003c/td\u003e\n   \u003ctd\u003e 0.6.0 \u003c/td\u003e\n   \u003ctd\u003e 0.6.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IlluminaHumanMethylation450kanno.ilmn12.hg19 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IlluminaHumanMethylation450kanno.ilmn12.hg19 \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-17 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e IlluminaHumanMethylation450kmanifest \u003c/td\u003e\n   \u003ctd\u003e IlluminaHumanMethylation450kmanifest \u003c/td\u003e\n   \u003ctd\u003e 0.4.0 \u003c/td\u003e\n   \u003ctd\u003e 0.4.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IlluminaHumanMethylation450kmanifest \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IlluminaHumanMethylation450kmanifest \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-17 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e illuminaio \u003c/td\u003e\n   \u003ctd\u003e illuminaio \u003c/td\u003e\n   \u003ctd\u003e 0.36.0 \u003c/td\u003e\n   \u003ctd\u003e 0.36.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/illuminaio \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/illuminaio \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e IRanges \u003c/td\u003e\n   \u003ctd\u003e IRanges \u003c/td\u003e\n   \u003ctd\u003e 2.28.0 \u003c/td\u003e\n   \u003ctd\u003e 2.28.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IRanges \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IRanges \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e iterators \u003c/td\u003e\n   \u003ctd\u003e iterators \u003c/td\u003e\n   \u003ctd\u003e 1.0.13 \u003c/td\u003e\n   \u003ctd\u003e 1.0.13 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/iterators \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/iterators \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-10-15 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e JASPAR2020 \u003c/td\u003e\n   \u003ctd\u003e JASPAR2020 \u003c/td\u003e\n   \u003ctd\u003e 0.99.10 \u003c/td\u003e\n   \u003ctd\u003e 0.99.10 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/JASPAR2020 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/JASPAR2020 \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-12-06 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e jpeg \u003c/td\u003e\n   \u003ctd\u003e jpeg \u003c/td\u003e\n   \u003ctd\u003e 0.1.9 \u003c/td\u003e\n   \u003ctd\u003e 0.1-9 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jpeg \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jpeg \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-07-24 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e jquerylib \u003c/td\u003e\n   \u003ctd\u003e jquerylib \u003c/td\u003e\n   \u003ctd\u003e 0.1.4 \u003c/td\u003e\n   \u003ctd\u003e 0.1.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jquerylib \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jquerylib \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-04-26 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e jsonlite \u003c/td\u003e\n   \u003ctd\u003e jsonlite \u003c/td\u003e\n   \u003ctd\u003e 1.7.2 \u003c/td\u003e\n   \u003ctd\u003e 1.7.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jsonlite \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jsonlite \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-12-09 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e kableExtra \u003c/td\u003e\n   \u003ctd\u003e kableExtra \u003c/td\u003e\n   \u003ctd\u003e 1.3.4 \u003c/td\u003e\n   \u003ctd\u003e 1.3.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/kableExtra \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/kableExtra \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-02-20 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e KEGGREST \u003c/td\u003e\n   \u003ctd\u003e KEGGREST \u003c/td\u003e\n   \u003ctd\u003e 1.34.0 \u003c/td\u003e\n   \u003ctd\u003e 1.34.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/KEGGREST \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/KEGGREST \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e knitr \u003c/td\u003e\n   \u003ctd\u003e knitr \u003c/td\u003e\n   \u003ctd\u003e 1.36 \u003c/td\u003e\n   \u003ctd\u003e 1.36 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/knitr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/knitr \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-29 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e labeling \u003c/td\u003e\n   \u003ctd\u003e labeling \u003c/td\u003e\n   \u003ctd\u003e 0.4.2 \u003c/td\u003e\n   \u003ctd\u003e 0.4.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/labeling \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/labeling \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-10-20 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e lambda.r \u003c/td\u003e\n   \u003ctd\u003e lambda.r \u003c/td\u003e\n   \u003ctd\u003e 1.2.4 \u003c/td\u003e\n   \u003ctd\u003e 1.2.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lambda.r \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lambda.r \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-09-18 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e later \u003c/td\u003e\n   \u003ctd\u003e later \u003c/td\u003e\n   \u003ctd\u003e 1.3.0 \u003c/td\u003e\n   \u003ctd\u003e 1.3.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/later \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/later \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-08-18 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e lattice \u003c/td\u003e\n   \u003ctd\u003e lattice \u003c/td\u003e\n   \u003ctd\u003e 0.20.45 \u003c/td\u003e\n   \u003ctd\u003e 0.20-45 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lattice \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lattice \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-22 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e latticeExtra \u003c/td\u003e\n   \u003ctd\u003e latticeExtra \u003c/td\u003e\n   \u003ctd\u003e 0.6.29 \u003c/td\u003e\n   \u003ctd\u003e 0.6-29 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/latticeExtra \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/latticeExtra \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-12-19 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e lazyeval \u003c/td\u003e\n   \u003ctd\u003e lazyeval \u003c/td\u003e\n   \u003ctd\u003e 0.2.2 \u003c/td\u003e\n   \u003ctd\u003e 0.2.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lazyeval \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lazyeval \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-03-15 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e lifecycle \u003c/td\u003e\n   \u003ctd\u003e lifecycle \u003c/td\u003e\n   \u003ctd\u003e 1.0.1 \u003c/td\u003e\n   \u003ctd\u003e 1.0.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lifecycle \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lifecycle \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-24 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e limma \u003c/td\u003e\n   \u003ctd\u003e limma \u003c/td\u003e\n   \u003ctd\u003e 3.50.0 \u003c/td\u003e\n   \u003ctd\u003e 3.50.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/limma \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/limma \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e locfit \u003c/td\u003e\n   \u003ctd\u003e locfit \u003c/td\u003e\n   \u003ctd\u003e 1.5.9.4 \u003c/td\u003e\n   \u003ctd\u003e 1.5-9.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/locfit \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/locfit \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-03-25 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e magick \u003c/td\u003e\n   \u003ctd\u003e magick \u003c/td\u003e\n   \u003ctd\u003e 2.7.3 \u003c/td\u003e\n   \u003ctd\u003e 2.7.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/magick \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/magick \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-08-18 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e magrittr \u003c/td\u003e\n   \u003ctd\u003e magrittr \u003c/td\u003e\n   \u003ctd\u003e 2.0.1 \u003c/td\u003e\n   \u003ctd\u003e 2.0.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/magrittr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/magrittr \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-11-17 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e MASS \u003c/td\u003e\n   \u003ctd\u003e MASS \u003c/td\u003e\n   \u003ctd\u003e 7.3.54 \u003c/td\u003e\n   \u003ctd\u003e 7.3-54 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/MASS \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/MASS \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-05-03 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e Matrix \u003c/td\u003e\n   \u003ctd\u003e Matrix \u003c/td\u003e\n   \u003ctd\u003e 1.3.4 \u003c/td\u003e\n   \u003ctd\u003e 1.3-4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Matrix \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Matrix \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-06-01 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e MatrixGenerics \u003c/td\u003e\n   \u003ctd\u003e MatrixGenerics \u003c/td\u003e\n   \u003ctd\u003e 1.6.0 \u003c/td\u003e\n   \u003ctd\u003e 1.6.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/MatrixGenerics \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/MatrixGenerics \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e matrixStats \u003c/td\u003e\n   \u003ctd\u003e matrixStats \u003c/td\u003e\n   \u003ctd\u003e 0.61.0 \u003c/td\u003e\n   \u003ctd\u003e 0.61.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/matrixStats \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/matrixStats \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-17 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e mclust \u003c/td\u003e\n   \u003ctd\u003e mclust \u003c/td\u003e\n   \u003ctd\u003e 5.4.8 \u003c/td\u003e\n   \u003ctd\u003e 5.4.8 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mclust \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mclust \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-05 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e memoise \u003c/td\u003e\n   \u003ctd\u003e memoise \u003c/td\u003e\n   \u003ctd\u003e 2.0.1 \u003c/td\u003e\n   \u003ctd\u003e 2.0.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/memoise \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/memoise \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-26 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e mgcv \u003c/td\u003e\n   \u003ctd\u003e mgcv \u003c/td\u003e\n   \u003ctd\u003e 1.8.38 \u003c/td\u003e\n   \u003ctd\u003e 1.8-38 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mgcv \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mgcv \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-06 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e mime \u003c/td\u003e\n   \u003ctd\u003e mime \u003c/td\u003e\n   \u003ctd\u003e 0.12 \u003c/td\u003e\n   \u003ctd\u003e 0.12 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mime \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mime \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-28 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e minfi \u003c/td\u003e\n   \u003ctd\u003e minfi \u003c/td\u003e\n   \u003ctd\u003e 1.40.0 \u003c/td\u003e\n   \u003ctd\u003e 1.40.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/minfi \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/minfi \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e monaLisa \u003c/td\u003e\n   \u003ctd\u003e monaLisa \u003c/td\u003e\n   \u003ctd\u003e 1.0.0 \u003c/td\u003e\n   \u003ctd\u003e 1.0.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/monaLisa \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/monaLisa \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e multtest \u003c/td\u003e\n   \u003ctd\u003e multtest \u003c/td\u003e\n   \u003ctd\u003e 2.50.0 \u003c/td\u003e\n   \u003ctd\u003e 2.50.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/multtest \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/multtest \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e munsell \u003c/td\u003e\n   \u003ctd\u003e munsell \u003c/td\u003e\n   \u003ctd\u003e 0.5.0 \u003c/td\u003e\n   \u003ctd\u003e 0.5.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/munsell \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/munsell \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2018-06-12 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e nlme \u003c/td\u003e\n   \u003ctd\u003e nlme \u003c/td\u003e\n   \u003ctd\u003e 3.1.153 \u003c/td\u003e\n   \u003ctd\u003e 3.1-153 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nlme \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nlme \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-07 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e nnet \u003c/td\u003e\n   \u003ctd\u003e nnet \u003c/td\u003e\n   \u003ctd\u003e 7.3.16 \u003c/td\u003e\n   \u003ctd\u003e 7.3-16 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nnet \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nnet \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-05-03 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e nor1mix \u003c/td\u003e\n   \u003ctd\u003e nor1mix \u003c/td\u003e\n   \u003ctd\u003e 1.3.0 \u003c/td\u003e\n   \u003ctd\u003e 1.3-0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nor1mix \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nor1mix \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-06-13 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e openssl \u003c/td\u003e\n   \u003ctd\u003e openssl \u003c/td\u003e\n   \u003ctd\u003e 1.4.5 \u003c/td\u003e\n   \u003ctd\u003e 1.4.5 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/openssl \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/openssl \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-02 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e openxlsx \u003c/td\u003e\n   \u003ctd\u003e openxlsx \u003c/td\u003e\n   \u003ctd\u003e 4.2.4 \u003c/td\u003e\n   \u003ctd\u003e 4.2.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/openxlsx \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/openxlsx \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-06-16 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e pacman \u003c/td\u003e\n   \u003ctd\u003e pacman \u003c/td\u003e\n   \u003ctd\u003e 0.5.1 \u003c/td\u003e\n   \u003ctd\u003e 0.5.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pacman \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pacman \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-03-11 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e pheatmap \u003c/td\u003e\n   \u003ctd\u003e pheatmap \u003c/td\u003e\n   \u003ctd\u003e 1.0.12 \u003c/td\u003e\n   \u003ctd\u003e 1.0.12 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pheatmap \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pheatmap \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-01-04 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e pillar \u003c/td\u003e\n   \u003ctd\u003e pillar \u003c/td\u003e\n   \u003ctd\u003e 1.6.4 \u003c/td\u003e\n   \u003ctd\u003e 1.6.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pillar \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pillar \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-18 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e pkgbuild \u003c/td\u003e\n   \u003ctd\u003e pkgbuild \u003c/td\u003e\n   \u003ctd\u003e 1.2.1 \u003c/td\u003e\n   \u003ctd\u003e 1.2.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgbuild \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgbuild \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e pkgconfig \u003c/td\u003e\n   \u003ctd\u003e pkgconfig \u003c/td\u003e\n   \u003ctd\u003e 2.0.3 \u003c/td\u003e\n   \u003ctd\u003e 2.0.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgconfig \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgconfig \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-09-22 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e pkgload \u003c/td\u003e\n   \u003ctd\u003e pkgload \u003c/td\u003e\n   \u003ctd\u003e 1.2.4 \u003c/td\u003e\n   \u003ctd\u003e 1.2.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgload \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgload \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e plotly \u003c/td\u003e\n   \u003ctd\u003e plotly \u003c/td\u003e\n   \u003ctd\u003e 4.10.0 \u003c/td\u003e\n   \u003ctd\u003e 4.10.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/plotly \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/plotly \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-09 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e plyr \u003c/td\u003e\n   \u003ctd\u003e plyr \u003c/td\u003e\n   \u003ctd\u003e 1.8.6 \u003c/td\u003e\n   \u003ctd\u003e 1.8.6 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/plyr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/plyr \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-03-03 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e png \u003c/td\u003e\n   \u003ctd\u003e png \u003c/td\u003e\n   \u003ctd\u003e 0.1.7 \u003c/td\u003e\n   \u003ctd\u003e 0.1-7 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/png \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/png \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2013-12-03 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e poweRlaw \u003c/td\u003e\n   \u003ctd\u003e poweRlaw \u003c/td\u003e\n   \u003ctd\u003e 0.70.6 \u003c/td\u003e\n   \u003ctd\u003e 0.70.6 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/poweRlaw \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/poweRlaw \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-04-25 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e pracma \u003c/td\u003e\n   \u003ctd\u003e pracma \u003c/td\u003e\n   \u003ctd\u003e 2.3.3 \u003c/td\u003e\n   \u003ctd\u003e 2.3.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pracma \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pracma \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-01-23 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e preprocessCore \u003c/td\u003e\n   \u003ctd\u003e preprocessCore \u003c/td\u003e\n   \u003ctd\u003e 1.56.0 \u003c/td\u003e\n   \u003ctd\u003e 1.56.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/preprocessCore \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/preprocessCore \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e prettyunits \u003c/td\u003e\n   \u003ctd\u003e prettyunits \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/prettyunits \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/prettyunits \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-01-24 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e processx \u003c/td\u003e\n   \u003ctd\u003e processx \u003c/td\u003e\n   \u003ctd\u003e 3.5.2 \u003c/td\u003e\n   \u003ctd\u003e 3.5.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/processx \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/processx \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-04-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e progress \u003c/td\u003e\n   \u003ctd\u003e progress \u003c/td\u003e\n   \u003ctd\u003e 1.2.2 \u003c/td\u003e\n   \u003ctd\u003e 1.2.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/progress \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/progress \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-05-16 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e promises \u003c/td\u003e\n   \u003ctd\u003e promises \u003c/td\u003e\n   \u003ctd\u003e 1.2.0.1 \u003c/td\u003e\n   \u003ctd\u003e 1.2.0.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/promises \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/promises \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-02-11 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e ps \u003c/td\u003e\n   \u003ctd\u003e ps \u003c/td\u003e\n   \u003ctd\u003e 1.6.0 \u003c/td\u003e\n   \u003ctd\u003e 1.6.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ps \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ps \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-02-28 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e purrr \u003c/td\u003e\n   \u003ctd\u003e purrr \u003c/td\u003e\n   \u003ctd\u003e 0.3.4 \u003c/td\u003e\n   \u003ctd\u003e 0.3.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/purrr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/purrr \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-04-17 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e quadprog \u003c/td\u003e\n   \u003ctd\u003e quadprog \u003c/td\u003e\n   \u003ctd\u003e 1.5.8 \u003c/td\u003e\n   \u003ctd\u003e 1.5-8 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/quadprog \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/quadprog \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-11-20 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e R.methodsS3 \u003c/td\u003e\n   \u003ctd\u003e R.methodsS3 \u003c/td\u003e\n   \u003ctd\u003e 1.8.1 \u003c/td\u003e\n   \u003ctd\u003e 1.8.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.methodsS3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.methodsS3 \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-08-26 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e R.oo \u003c/td\u003e\n   \u003ctd\u003e R.oo \u003c/td\u003e\n   \u003ctd\u003e 1.24.0 \u003c/td\u003e\n   \u003ctd\u003e 1.24.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.oo \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.oo \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-08-26 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e R.utils \u003c/td\u003e\n   \u003ctd\u003e R.utils \u003c/td\u003e\n   \u003ctd\u003e 2.11.0 \u003c/td\u003e\n   \u003ctd\u003e 2.11.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.utils \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.utils \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-26 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e R6 \u003c/td\u003e\n   \u003ctd\u003e R6 \u003c/td\u003e\n   \u003ctd\u003e 2.5.1 \u003c/td\u003e\n   \u003ctd\u003e 2.5.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R6 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R6 \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-08-19 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rappdirs \u003c/td\u003e\n   \u003ctd\u003e rappdirs \u003c/td\u003e\n   \u003ctd\u003e 0.3.3 \u003c/td\u003e\n   \u003ctd\u003e 0.3.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rappdirs \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rappdirs \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-01-31 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e RColorBrewer \u003c/td\u003e\n   \u003ctd\u003e RColorBrewer \u003c/td\u003e\n   \u003ctd\u003e 1.1.2 \u003c/td\u003e\n   \u003ctd\u003e 1.1-2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RColorBrewer \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RColorBrewer \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2014-12-07 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e Rcpp \u003c/td\u003e\n   \u003ctd\u003e Rcpp \u003c/td\u003e\n   \u003ctd\u003e 1.0.7 \u003c/td\u003e\n   \u003ctd\u003e 1.0.7 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rcpp \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rcpp \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-07-07 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e RCurl \u003c/td\u003e\n   \u003ctd\u003e RCurl \u003c/td\u003e\n   \u003ctd\u003e 1.98.1.5 \u003c/td\u003e\n   \u003ctd\u003e 1.98-1.5 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RCurl \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RCurl \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-17 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e readr \u003c/td\u003e\n   \u003ctd\u003e readr \u003c/td\u003e\n   \u003ctd\u003e 2.1.1 \u003c/td\u003e\n   \u003ctd\u003e 2.1.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/readr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/readr \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e readxl \u003c/td\u003e\n   \u003ctd\u003e readxl \u003c/td\u003e\n   \u003ctd\u003e 1.3.1 \u003c/td\u003e\n   \u003ctd\u003e 1.3.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/readxl \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/readxl \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-03-13 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e remotes \u003c/td\u003e\n   \u003ctd\u003e remotes \u003c/td\u003e\n   \u003ctd\u003e 2.4.2 \u003c/td\u003e\n   \u003ctd\u003e 2.4.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/remotes \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/remotes \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e reshape \u003c/td\u003e\n   \u003ctd\u003e reshape \u003c/td\u003e\n   \u003ctd\u003e 0.8.8 \u003c/td\u003e\n   \u003ctd\u003e 0.8.8 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/reshape \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/reshape \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2018-10-23 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e reshape2 \u003c/td\u003e\n   \u003ctd\u003e reshape2 \u003c/td\u003e\n   \u003ctd\u003e 1.4.4 \u003c/td\u003e\n   \u003ctd\u003e 1.4.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/reshape2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/reshape2 \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-04-09 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e restfulr \u003c/td\u003e\n   \u003ctd\u003e restfulr \u003c/td\u003e\n   \u003ctd\u003e 0.0.13 \u003c/td\u003e\n   \u003ctd\u003e 0.0.13 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/restfulr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/restfulr \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2017-08-06 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rhdf5 \u003c/td\u003e\n   \u003ctd\u003e rhdf5 \u003c/td\u003e\n   \u003ctd\u003e 2.38.0 \u003c/td\u003e\n   \u003ctd\u003e 2.38.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rhdf5 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rhdf5 \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rhdf5filters \u003c/td\u003e\n   \u003ctd\u003e rhdf5filters \u003c/td\u003e\n   \u003ctd\u003e 1.6.0 \u003c/td\u003e\n   \u003ctd\u003e 1.6.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rhdf5filters \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rhdf5filters \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e Rhdf5lib \u003c/td\u003e\n   \u003ctd\u003e Rhdf5lib \u003c/td\u003e\n   \u003ctd\u003e 1.16.0 \u003c/td\u003e\n   \u003ctd\u003e 1.16.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rhdf5lib \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rhdf5lib \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rjson \u003c/td\u003e\n   \u003ctd\u003e rjson \u003c/td\u003e\n   \u003ctd\u003e 0.2.20 \u003c/td\u003e\n   \u003ctd\u003e 0.2.20 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rjson \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rjson \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2018-06-08 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rlang \u003c/td\u003e\n   \u003ctd\u003e rlang \u003c/td\u003e\n   \u003ctd\u003e 0.4.12 \u003c/td\u003e\n   \u003ctd\u003e 0.4.12 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rlang \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rlang \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-18 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rmarkdown \u003c/td\u003e\n   \u003ctd\u003e rmarkdown \u003c/td\u003e\n   \u003ctd\u003e 2.11 \u003c/td\u003e\n   \u003ctd\u003e 2.11 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rmarkdown \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rmarkdown \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-14 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rngtools \u003c/td\u003e\n   \u003ctd\u003e rngtools \u003c/td\u003e\n   \u003ctd\u003e 1.5.2 \u003c/td\u003e\n   \u003ctd\u003e 1.5.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rngtools \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rngtools \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-20 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rpart \u003c/td\u003e\n   \u003ctd\u003e rpart \u003c/td\u003e\n   \u003ctd\u003e 4.1.15 \u003c/td\u003e\n   \u003ctd\u003e 4.1-15 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rpart \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rpart \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-04-12 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rprojroot \u003c/td\u003e\n   \u003ctd\u003e rprojroot \u003c/td\u003e\n   \u003ctd\u003e 2.0.2 \u003c/td\u003e\n   \u003ctd\u003e 2.0.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rprojroot \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rprojroot \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-11-15 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e Rsamtools \u003c/td\u003e\n   \u003ctd\u003e Rsamtools \u003c/td\u003e\n   \u003ctd\u003e 2.10.0 \u003c/td\u003e\n   \u003ctd\u003e 2.10.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rsamtools \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rsamtools \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e RSQLite \u003c/td\u003e\n   \u003ctd\u003e RSQLite \u003c/td\u003e\n   \u003ctd\u003e 2.2.8 \u003c/td\u003e\n   \u003ctd\u003e 2.2.8 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RSQLite \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RSQLite \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-08-21 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rstatix \u003c/td\u003e\n   \u003ctd\u003e rstatix \u003c/td\u003e\n   \u003ctd\u003e 0.7.0 \u003c/td\u003e\n   \u003ctd\u003e 0.7.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rstatix \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rstatix \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-02-13 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rstudioapi \u003c/td\u003e\n   \u003ctd\u003e rstudioapi \u003c/td\u003e\n   \u003ctd\u003e 0.13 \u003c/td\u003e\n   \u003ctd\u003e 0.13 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rstudioapi \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rstudioapi \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-11-12 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rtracklayer \u003c/td\u003e\n   \u003ctd\u003e rtracklayer \u003c/td\u003e\n   \u003ctd\u003e 1.54.0 \u003c/td\u003e\n   \u003ctd\u003e 1.54.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rtracklayer \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rtracklayer \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e rvest \u003c/td\u003e\n   \u003ctd\u003e rvest \u003c/td\u003e\n   \u003ctd\u003e 1.0.2 \u003c/td\u003e\n   \u003ctd\u003e 1.0.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rvest \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rvest \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-16 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e S4Vectors \u003c/td\u003e\n   \u003ctd\u003e S4Vectors \u003c/td\u003e\n   \u003ctd\u003e 0.32.3 \u003c/td\u003e\n   \u003ctd\u003e 0.32.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/S4Vectors \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/S4Vectors \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-21 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e sass \u003c/td\u003e\n   \u003ctd\u003e sass \u003c/td\u003e\n   \u003ctd\u003e 0.4.0 \u003c/td\u003e\n   \u003ctd\u003e 0.4.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sass \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sass \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-05-12 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e scales \u003c/td\u003e\n   \u003ctd\u003e scales \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/scales \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/scales \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-05-11 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e scrime \u003c/td\u003e\n   \u003ctd\u003e scrime \u003c/td\u003e\n   \u003ctd\u003e 1.3.5 \u003c/td\u003e\n   \u003ctd\u003e 1.3.5 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/scrime \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/scrime \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2018-12-01 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e seqLogo \u003c/td\u003e\n   \u003ctd\u003e seqLogo \u003c/td\u003e\n   \u003ctd\u003e 1.60.0 \u003c/td\u003e\n   \u003ctd\u003e 1.60.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/seqLogo \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/seqLogo \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e sessioninfo \u003c/td\u003e\n   \u003ctd\u003e sessioninfo \u003c/td\u003e\n   \u003ctd\u003e 1.2.1 \u003c/td\u003e\n   \u003ctd\u003e 1.2.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sessioninfo \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sessioninfo \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-02 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e shape \u003c/td\u003e\n   \u003ctd\u003e shape \u003c/td\u003e\n   \u003ctd\u003e 1.4.6 \u003c/td\u003e\n   \u003ctd\u003e 1.4.6 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shape \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shape \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-05-19 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e shiny \u003c/td\u003e\n   \u003ctd\u003e shiny \u003c/td\u003e\n   \u003ctd\u003e 1.7.1 \u003c/td\u003e\n   \u003ctd\u003e 1.7.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shiny \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shiny \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-02 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e shinyMethyl \u003c/td\u003e\n   \u003ctd\u003e shinyMethyl \u003c/td\u003e\n   \u003ctd\u003e 1.30.0 \u003c/td\u003e\n   \u003ctd\u003e 1.30.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shinyMethyl \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shinyMethyl \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e siggenes \u003c/td\u003e\n   \u003ctd\u003e siggenes \u003c/td\u003e\n   \u003ctd\u003e 1.68.0 \u003c/td\u003e\n   \u003ctd\u003e 1.68.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/siggenes \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/siggenes \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e sm \u003c/td\u003e\n   \u003ctd\u003e sm \u003c/td\u003e\n   \u003ctd\u003e 2.2.5.7 \u003c/td\u003e\n   \u003ctd\u003e 2.2-5.7 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sm \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sm \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-13 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e sparseMatrixStats \u003c/td\u003e\n   \u003ctd\u003e sparseMatrixStats \u003c/td\u003e\n   \u003ctd\u003e 1.6.0 \u003c/td\u003e\n   \u003ctd\u003e 1.6.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sparseMatrixStats \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sparseMatrixStats \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e stabs \u003c/td\u003e\n   \u003ctd\u003e stabs \u003c/td\u003e\n   \u003ctd\u003e 0.6.4 \u003c/td\u003e\n   \u003ctd\u003e 0.6-4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stabs \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stabs \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-01-29 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e stringi \u003c/td\u003e\n   \u003ctd\u003e stringi \u003c/td\u003e\n   \u003ctd\u003e 1.7.6 \u003c/td\u003e\n   \u003ctd\u003e 1.7.6 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stringi \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stringi \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-29 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e stringr \u003c/td\u003e\n   \u003ctd\u003e stringr \u003c/td\u003e\n   \u003ctd\u003e 1.4.0 \u003c/td\u003e\n   \u003ctd\u003e 1.4.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stringr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stringr \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-02-10 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e SummarizedExperiment \u003c/td\u003e\n   \u003ctd\u003e SummarizedExperiment \u003c/td\u003e\n   \u003ctd\u003e 1.24.0 \u003c/td\u003e\n   \u003ctd\u003e 1.24.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/SummarizedExperiment \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/SummarizedExperiment \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e survival \u003c/td\u003e\n   \u003ctd\u003e survival \u003c/td\u003e\n   \u003ctd\u003e 3.2.13 \u003c/td\u003e\n   \u003ctd\u003e 3.2-13 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/survival \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/survival \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-08-24 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e svglite \u003c/td\u003e\n   \u003ctd\u003e svglite \u003c/td\u003e\n   \u003ctd\u003e 2.0.0 \u003c/td\u003e\n   \u003ctd\u003e 2.0.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/svglite \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/svglite \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-02-20 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e systemfonts \u003c/td\u003e\n   \u003ctd\u003e systemfonts \u003c/td\u003e\n   \u003ctd\u003e 1.0.3 \u003c/td\u003e\n   \u003ctd\u003e 1.0.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/systemfonts \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/systemfonts \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-13 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.2) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e testthat \u003c/td\u003e\n   \u003ctd\u003e testthat \u003c/td\u003e\n   \u003ctd\u003e 3.1.1 \u003c/td\u003e\n   \u003ctd\u003e 3.1.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/testthat \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/testthat \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-12-03 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e TFBSTools \u003c/td\u003e\n   \u003ctd\u003e TFBSTools \u003c/td\u003e\n   \u003ctd\u003e 1.32.0 \u003c/td\u003e\n   \u003ctd\u003e 1.32.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/TFBSTools \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/TFBSTools \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e TFMPvalue \u003c/td\u003e\n   \u003ctd\u003e TFMPvalue \u003c/td\u003e\n   \u003ctd\u003e 0.0.8 \u003c/td\u003e\n   \u003ctd\u003e 0.0.8 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/TFMPvalue \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/TFMPvalue \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2018-05-16 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e tibble \u003c/td\u003e\n   \u003ctd\u003e tibble \u003c/td\u003e\n   \u003ctd\u003e 3.1.6 \u003c/td\u003e\n   \u003ctd\u003e 3.1.6 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tibble \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tibble \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-07 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e tidyr \u003c/td\u003e\n   \u003ctd\u003e tidyr \u003c/td\u003e\n   \u003ctd\u003e 1.1.4 \u003c/td\u003e\n   \u003ctd\u003e 1.1.4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tidyr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tidyr \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-27 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e tidyselect \u003c/td\u003e\n   \u003ctd\u003e tidyselect \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e 1.1.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tidyselect \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tidyselect \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-04-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e tzdb \u003c/td\u003e\n   \u003ctd\u003e tzdb \u003c/td\u003e\n   \u003ctd\u003e 0.2.0 \u003c/td\u003e\n   \u003ctd\u003e 0.2.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tzdb \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tzdb \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-27 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e usethis \u003c/td\u003e\n   \u003ctd\u003e usethis \u003c/td\u003e\n   \u003ctd\u003e 2.1.3 \u003c/td\u003e\n   \u003ctd\u003e 2.1.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/usethis \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/usethis \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-27 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e utf8 \u003c/td\u003e\n   \u003ctd\u003e utf8 \u003c/td\u003e\n   \u003ctd\u003e 1.2.2 \u003c/td\u003e\n   \u003ctd\u003e 1.2.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/utf8 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/utf8 \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-07-24 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e vctrs \u003c/td\u003e\n   \u003ctd\u003e vctrs \u003c/td\u003e\n   \u003ctd\u003e 0.3.8 \u003c/td\u003e\n   \u003ctd\u003e 0.3.8 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/vctrs \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/vctrs \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-04-29 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e VennDiagram \u003c/td\u003e\n   \u003ctd\u003e VennDiagram \u003c/td\u003e\n   \u003ctd\u003e 1.7.1 \u003c/td\u003e\n   \u003ctd\u003e 1.7.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/VennDiagram \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/VennDiagram \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-12-02 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e vioplot \u003c/td\u003e\n   \u003ctd\u003e vioplot \u003c/td\u003e\n   \u003ctd\u003e 0.3.7 \u003c/td\u003e\n   \u003ctd\u003e 0.3.7 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/vioplot \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/vioplot \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-07-27 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e viridisLite \u003c/td\u003e\n   \u003ctd\u003e viridisLite \u003c/td\u003e\n   \u003ctd\u003e 0.4.0 \u003c/td\u003e\n   \u003ctd\u003e 0.4.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/viridisLite \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/viridisLite \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-04-13 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e webshot \u003c/td\u003e\n   \u003ctd\u003e webshot \u003c/td\u003e\n   \u003ctd\u003e 0.5.2 \u003c/td\u003e\n   \u003ctd\u003e 0.5.2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/webshot \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/webshot \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-11-22 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e wesanderson \u003c/td\u003e\n   \u003ctd\u003e wesanderson \u003c/td\u003e\n   \u003ctd\u003e 0.3.6 \u003c/td\u003e\n   \u003ctd\u003e 0.3.6 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/wesanderson \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/wesanderson \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2018-04-20 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e withr \u003c/td\u003e\n   \u003ctd\u003e withr \u003c/td\u003e\n   \u003ctd\u003e 2.4.3 \u003c/td\u003e\n   \u003ctd\u003e 2.4.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/withr \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/withr \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e xfun \u003c/td\u003e\n   \u003ctd\u003e xfun \u003c/td\u003e\n   \u003ctd\u003e 0.28 \u003c/td\u003e\n   \u003ctd\u003e 0.28 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xfun \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xfun \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-04 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e XML \u003c/td\u003e\n   \u003ctd\u003e XML \u003c/td\u003e\n   \u003ctd\u003e 3.99.0.8 \u003c/td\u003e\n   \u003ctd\u003e 3.99-0.8 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/XML \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/XML \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-09-17 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e xml2 \u003c/td\u003e\n   \u003ctd\u003e xml2 \u003c/td\u003e\n   \u003ctd\u003e 1.3.3 \u003c/td\u003e\n   \u003ctd\u003e 1.3.3 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xml2 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xml2 \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-11-30 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e xtable \u003c/td\u003e\n   \u003ctd\u003e xtable \u003c/td\u003e\n   \u003ctd\u003e 1.8.4 \u003c/td\u003e\n   \u003ctd\u003e 1.8-4 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xtable \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xtable \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2019-04-21 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e XVector \u003c/td\u003e\n   \u003ctd\u003e XVector \u003c/td\u003e\n   \u003ctd\u003e 0.34.0 \u003c/td\u003e\n   \u003ctd\u003e 0.34.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/XVector \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/XVector \u003c/td\u003e\n   \u003ctd\u003e TRUE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e yaml \u003c/td\u003e\n   \u003ctd\u003e yaml \u003c/td\u003e\n   \u003ctd\u003e 2.2.1 \u003c/td\u003e\n   \u003ctd\u003e 2.2.1 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/yaml \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/yaml \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2020-02-01 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e zip \u003c/td\u003e\n   \u003ctd\u003e zip \u003c/td\u003e\n   \u003ctd\u003e 2.2.0 \u003c/td\u003e\n   \u003ctd\u003e 2.2.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zip \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zip \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-05-31 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e zlibbioc \u003c/td\u003e\n   \u003ctd\u003e zlibbioc \u003c/td\u003e\n   \u003ctd\u003e 1.40.0 \u003c/td\u003e\n   \u003ctd\u003e 1.40.0 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zlibbioc \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zlibbioc \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-10-26 \u003c/td\u003e\n   \u003ctd\u003e Bioconductor \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n  \u003ctr\u003e\n   \u003ctd\u003e zoo \u003c/td\u003e\n   \u003ctd\u003e zoo \u003c/td\u003e\n   \u003ctd\u003e 1.8.9 \u003c/td\u003e\n   \u003ctd\u003e 1.8-9 \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zoo \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zoo \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e FALSE \u003c/td\u003e\n   \u003ctd\u003e 2021-03-09 \u003c/td\u003e\n   \u003ctd\u003e CRAN (R 4.1.0) \u003c/td\u003e\n   \u003ctd\u003e  \u003c/td\u003e\n   \u003ctd\u003e /Library/Frameworks/R.framework/Versions/4.1/Resources/library \u003c/td\u003e\n  \u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"Preliminary Setup","anchor":"preliminary-setup","htmlText":"Preliminary Setup"},{"level":2,"text":"Parameters","anchor":"parameters","htmlText":"Parameters"},{"level":2,"text":"Packages","anchor":"packages","htmlText":"Packages"},{"level":1,"text":"Genome-wide DNA methylation - Illumina(R) HumanMethylation 450k Methylation","anchor":"genome-wide-dna-methylation---illuminar-humanmethylation-450k-methylation","htmlText":"Genome-wide DNA methylation - Illumina(R) HumanMethylation 450k Methylation"},{"level":2,"text":"Supplemental Figure SXX: Outlier Analysis","anchor":"supplemental-figure-sxx-outlier-analysis","htmlText":"Supplemental Figure SXX: Outlier Analysis"},{"level":2,"text":"Figure 2A: MDS Plot - Unsupervised Clustering by \"Sample_Group\"","anchor":"figure-2a-mds-plot---unsupervised-clustering-by-sample_group","htmlText":"Figure 2A: MDS Plot - Unsupervised Clustering by \"Sample_Group\""},{"level":2,"text":"PCA","anchor":"pca","htmlText":"PCA"},{"level":3,"text":"3-Dimensional PCA","anchor":"3-dimensional-pca","htmlText":"3-Dimensional PCA"},{"level":2,"text":"Differential Methylation of iPSCs (DCM vs. CON)","anchor":"differential-methylation-of-ipscs-dcm-vs-con","htmlText":"Differential Methylation of iPSCs (DCM vs. CON)"},{"level":2,"text":"Using Minfi to compute differential expression","anchor":"using-minfi-to-compute-differential-expression","htmlText":"Using Minfi to compute differential expression"},{"level":2,"text":"Distribution of Methylation by Genomic and CpG Annotation","anchor":"distribution-of-methylation-by-genomic-and-cpg-annotation","htmlText":"Distribution of Methylation by Genomic and CpG Annotation"},{"level":2,"text":"Figure XX: Heatmap and Hierarchical Clustering of Differential Methylation (P \u003c 0.05)","anchor":"figure-xx-heatmap-and-hierarchical-clustering-of-differential-methylation-p--005","htmlText":"Figure XX: Heatmap and Hierarchical Clustering of Differential Methylation (P \u0026lt; 0.05)"},{"level":2,"text":"Figure XX: Volcano Plot - DMPs","anchor":"figure-xx-volcano-plot---dmps","htmlText":"Figure XX: Volcano Plot - DMPs"},{"level":1,"text":"Combined Analysis - Biopsy and iPSC","anchor":"combined-analysis---biopsy-and-ipsc","htmlText":"Combined Analysis - Biopsy and iPSC"},{"level":1,"text":"Validation","anchor":"validation","htmlText":"Validation"},{"level":1,"text":"Motif Enrichment","anchor":"motif-enrichment","htmlText":"Motif Enrichment"},{"level":1,"text":"Supplemental Table: R Session Information","anchor":"supplemental-table-r-session-information","htmlText":"Supplemental Table: R Session Information"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fmepepin%2FIndirect.Epigenomics"}},{"displayName":"LICENSE","repoName":"Indirect.Epigenomics","refName":"main","path":"LICENSE","preferredFileType":"license","tabName":"MIT","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fmepepin%2FIndirect.Epigenomics"}}],"overviewFilesProcessingTime":271.354745}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
.bhqztV{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.bhqztV span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.bhqztV:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.bhqztV:focus{outline:2px solid transparent;}/*!sc*/
.bhqztV:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.bhqztV:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.bhqztV::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:transparent;border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.bhqztV::after{background-color:transparent;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,bhqztV,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/mepepin/Indirect.Epigenomics/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/mepepin/Indirect.Epigenomics/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/mepepin/Indirect.Epigenomics/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/mepepin/Indirect.Epigenomics/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/mepepin/Indirect.Epigenomics/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           3 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/mepepin/Indirect.Epigenomics/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README_files, (Directory)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/tree/main/README_files" title="README_files">
                          README_files
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README_files, (Directory)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/tree/main/README_files" title="README_files">
                          README_files
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".DS_Store, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/.DS_Store" title=".DS_Store">
                          .DS_Store
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".DS_Store, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/.DS_Store" title=".DS_Store">
                          .DS_Store
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="DCM_vs_CON.csv, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/DCM_vs_CON.csv" title="DCM_vs_CON.csv">
                          DCM_vs_CON.csv
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="DCM_vs_CON.csv, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/DCM_vs_CON.csv" title="DCM_vs_CON.csv">
                          DCM_vs_CON.csv
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.Rmd, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/README.Rmd" title="README.Rmd">
                          README.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.Rmd, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/README.Rmd" title="README.Rmd">
                          README.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-6">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.html, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/README.html" title="README.html">
                          README.html
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.html, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/README.html" title="README.html">
                          README.html
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-7">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.log, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/README.log" title="README.log">
                          README.log
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.log, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/README.log" title="README.log">
                          README.log
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-8">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-9">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.tex, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/README.tex" title="README.tex">
                          README.tex
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.tex, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/README.tex" title="README.tex">
                          README.tex
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-10">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="qcReport.pdf, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/qcReport.pdf" title="qcReport.pdf">
                          qcReport.pdf
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="qcReport.pdf, (File)" class="Link--primary" href="/mepepin/Indirect.Epigenomics/blob/main/qcReport.pdf" title="qcReport.pdf">
                          qcReport.pdf
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd show-for-mobile" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-law" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="MIT license">
                        MIT license
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <table>
                     <thead>
                      <tr>
                       <th>
                        title
                       </th>
                       <th>
                        author
                       </th>
                       <th>
                        date
                       </th>
                       <th>
                        output
                       </th>
                       <th>
                        header-includes
                       </th>
                       <th>
                        mainfont
                       </th>
                       <th>
                        fontsize
                       </th>
                       <th>
                        always_allow_html
                       </th>
                       <th>
                        editor_options
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        <div dir="auto">
                         Supplemental Methods - Epigenenomics of Sample_Group in Heart Failure
                        </div>
                       </td>
                       <td>
                        <div dir="auto">
                         Mark E. Pepin
                        </div>
                       </td>
                       <td>
                        <div dir="auto">
                         31 December, 2021
                        </div>
                       </td>
                       <td>
                        <div dir="auto">
                         <table>
                          <thead>
                           <tr>
                            <th>
                             html_document
                            </th>
                            <th>
                             word_document
                            </th>
                            <th>
                             pdf_document
                            </th>
                           </tr>
                          </thead>
                          <tbody>
                           <tr>
                            <td>
                             <div dir="auto">
                              <table>
                               <thead>
                                <tr>
                                 <th>
                                  keep_md
                                 </th>
                                 <th>
                                  code_folding
                                 </th>
                                 <th>
                                  toc
                                 </th>
                                 <th>
                                  toc_float
                                 </th>
                                </tr>
                               </thead>
                               <tbody>
                                <tr>
                                 <td>
                                  <div dir="auto">
                                   true
                                  </div>
                                 </td>
                                 <td>
                                  <div dir="auto">
                                   hide
                                  </div>
                                 </td>
                                 <td>
                                  <div dir="auto">
                                   true
                                  </div>
                                 </td>
                                 <td>
                                  <div dir="auto">
                                   true
                                  </div>
                                 </td>
                                </tr>
                               </tbody>
                              </table>
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              <table>
                               <thead>
                                <tr>
                                 <th>
                                  toc
                                 </th>
                                </tr>
                               </thead>
                               <tbody>
                                <tr>
                                 <td>
                                  <div dir="auto">
                                   true
                                  </div>
                                 </td>
                                </tr>
                               </tbody>
                              </table>
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              <table>
                               <thead>
                                <tr>
                                 <th>
                                  toc
                                 </th>
                                </tr>
                               </thead>
                               <tbody>
                                <tr>
                                 <td>
                                  <div dir="auto">
                                   true
                                  </div>
                                 </td>
                                </tr>
                               </tbody>
                              </table>
                             </div>
                            </td>
                           </tr>
                          </tbody>
                         </table>
                        </div>
                       </td>
                       <td>
                        <div dir="auto">
                         <table>
                          <tbody>
                           <tr>
                            <td>
                             <div dir="auto">
                              \usepackage{booktabs}
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              \usepackage{longtable}
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              \usepackage{array}
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              \usepackage{multirow}
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              \usepackage[table]{xcolor}
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              \usepackage{wrapfig}
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              \usepackage{float}
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              \usepackage{colortbl}
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              \usepackage{pdflscape}
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              \usepackage{tabu}
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              \usepackage{threeparttable}
                             </div>
                            </td>
                           </tr>
                          </tbody>
                         </table>
                        </div>
                       </td>
                       <td>
                        <div dir="auto">
                         Times
                        </div>
                       </td>
                       <td>
                        <div dir="auto">
                         10pt
                        </div>
                       </td>
                       <td>
                        <div dir="auto">
                         true
                        </div>
                       </td>
                       <td>
                        <div dir="auto">
                         <table>
                          <thead>
                           <tr>
                            <th>
                             markdown
                            </th>
                            <th>
                             chunk_output_type
                            </th>
                           </tr>
                          </thead>
                          <tbody>
                           <tr>
                            <td>
                             <div dir="auto">
                              <table>
                               <thead>
                                <tr>
                                 <th>
                                  wrap
                                 </th>
                                </tr>
                               </thead>
                               <tbody>
                                <tr>
                                 <td>
                                  <div dir="auto">
                                   72
                                  </div>
                                 </td>
                                </tr>
                               </tbody>
                              </table>
                             </div>
                            </td>
                            <td>
                             <div dir="auto">
                              inline
                             </div>
                            </td>
                           </tr>
                          </tbody>
                         </table>
                        </div>
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     <strong>
                      Code Author
                     </strong>
                     : Mark E. Pepin, MD, PhD, MS
                     <strong>
                      Contact
                     </strong>
                     :
                     <a href="mailto:pepinme@gmail.com">
                      pepinme@gmail.com
                     </a>
                     {.email}
                     <strong>
                      Institution
                     </strong>
                     :
Heidelberg University Hospital
                     <strong>
                      Location
                     </strong>
                     : 669 Neuenheimer Feld,
Institute for Experimental Cardiology, 69120 Heidelberg, Germany
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Preliminary Setup
                     </h1>
                     <a aria-label="Permalink: Preliminary Setup" class="anchor" href="#preliminary-setup" id="user-content-preliminary-setup">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Parameters
                     </h2>
                     <a aria-label="Permalink: Parameters" class="anchor" href="#parameters" id="user-content-parameters">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Define the parameters used, along with the conditions required for the
current analysis. This segment must be modified for each analysis
performed.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='##Set the experimental conditions [MUST DO THIS MANUALLY]
Run_tS&lt;-Sys.time()
TREATMENT=c("CON", "DCM")
CELL=c("iPSC")
STATISTIC = 0.05 #P statistic threshold used in this combination.
VARIABLE = TREATMENT
COMPARISON= paste0(CELL[1], "_",VARIABLE[2], "_vs_", VARIABLE[1])

# Candidate Gene Selection (RNA-sequencing) EDIT THIS LIST BASED ON INTERESTS.
GENES=c("LEP", "ADIPOQ", "PLIN1")
VAR1="Treatment"
VAR2="Diabetes"
# Single Bar Graph
library(dplyr)
my_comparisons &lt;- list( c("CON", "DCM")) # Comparisons to make
## Create color based on Genotype
ann_colors = list(Treatment = c(DCM="darkgray", CON = "darkcyan"))
TreatmentColors&lt;-ann_colors$Treatment
ann_colorTable&lt;-as.data.frame(ann_colors)

ann_colGroup&lt;-subset(ann_colorTable, rownames(ann_colorTable) %in% TREATMENT)
ann_colListGroup&lt;-list(ann_colors$Treatment)
ann_colGroupVec&lt;-ann_colGroup$Treatment
names(ann_colGroupVec)&lt;-as.factor(rownames(ann_colGroup))
GROUP_colors&lt;-list(ann_colGroupVec=ann_colGroupVec)

# Create Output Folder Structure
ifelse(!dir.exists(file.path(paste0("../2_Output/"))), dir.create(file.path(paste0("../2_Output/"))), FALSE)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>#Set the experimental conditions [MUST DO THIS MANUALLY]</span>
<span class="pl-smi">Run_tS</span><span class="pl-k">&lt;-</span>Sys.time()
<span class="pl-v">TREATMENT</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>CON<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>DCM<span class="pl-pds">"</span></span>)
<span class="pl-v">CELL</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>iPSC<span class="pl-pds">"</span></span>)
<span class="pl-v">STATISTIC</span> <span class="pl-k">=</span> <span class="pl-c1">0.05</span> <span class="pl-c"><span class="pl-c">#</span>P statistic threshold used in this combination.</span>
<span class="pl-v">VARIABLE</span> <span class="pl-k">=</span> <span class="pl-smi">TREATMENT</span>
<span class="pl-v">COMPARISON</span><span class="pl-k">=</span> paste0(<span class="pl-smi">CELL</span>[<span class="pl-c1">1</span>], <span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>,<span class="pl-smi">VARIABLE</span>[<span class="pl-c1">2</span>], <span class="pl-s"><span class="pl-pds">"</span>_vs_<span class="pl-pds">"</span></span>, <span class="pl-smi">VARIABLE</span>[<span class="pl-c1">1</span>])

<span class="pl-c"><span class="pl-c">#</span> Candidate Gene Selection (RNA-sequencing) EDIT THIS LIST BASED ON INTERESTS.</span>
<span class="pl-v">GENES</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>LEP<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>ADIPOQ<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>PLIN1<span class="pl-pds">"</span></span>)
<span class="pl-v">VAR1</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Treatment<span class="pl-pds">"</span></span>
<span class="pl-v">VAR2</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Diabetes<span class="pl-pds">"</span></span>
<span class="pl-c"><span class="pl-c">#</span> Single Bar Graph</span>
library(<span class="pl-smi">dplyr</span>)
<span class="pl-smi">my_comparisons</span> <span class="pl-k">&lt;-</span> <span class="pl-k">list</span>( c(<span class="pl-s"><span class="pl-pds">"</span>CON<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>DCM<span class="pl-pds">"</span></span>)) <span class="pl-c"><span class="pl-c">#</span> Comparisons to make</span>
<span class="pl-c"><span class="pl-c">#</span># Create color based on Genotype</span>
<span class="pl-v">ann_colors</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-v">Treatment</span> <span class="pl-k">=</span> c(<span class="pl-v">DCM</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>darkgray<span class="pl-pds">"</span></span>, <span class="pl-v">CON</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>darkcyan<span class="pl-pds">"</span></span>))
<span class="pl-smi">TreatmentColors</span><span class="pl-k">&lt;-</span><span class="pl-smi">ann_colors</span><span class="pl-k">$</span><span class="pl-smi">Treatment</span>
<span class="pl-smi">ann_colorTable</span><span class="pl-k">&lt;-</span>as.data.frame(<span class="pl-smi">ann_colors</span>)

<span class="pl-smi">ann_colGroup</span><span class="pl-k">&lt;-</span>subset(<span class="pl-smi">ann_colorTable</span>, rownames(<span class="pl-smi">ann_colorTable</span>) <span class="pl-k">%in%</span> <span class="pl-smi">TREATMENT</span>)
<span class="pl-smi">ann_colListGroup</span><span class="pl-k">&lt;-</span><span class="pl-k">list</span>(<span class="pl-smi">ann_colors</span><span class="pl-k">$</span><span class="pl-smi">Treatment</span>)
<span class="pl-smi">ann_colGroupVec</span><span class="pl-k">&lt;-</span><span class="pl-smi">ann_colGroup</span><span class="pl-k">$</span><span class="pl-smi">Treatment</span>
names(<span class="pl-smi">ann_colGroupVec</span>)<span class="pl-k">&lt;-</span>as.factor(rownames(<span class="pl-smi">ann_colGroup</span>))
<span class="pl-smi">GROUP_colors</span><span class="pl-k">&lt;-</span><span class="pl-k">list</span>(<span class="pl-v">ann_colGroupVec</span><span class="pl-k">=</span><span class="pl-smi">ann_colGroupVec</span>)

<span class="pl-c"><span class="pl-c">#</span> Create Output Folder Structure</span>
ifelse(<span class="pl-k">!</span>dir.exists(file.path(paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>))), dir.create(file.path(paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>))), <span class="pl-c1">FALSE</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## [1] FALSE">
                     <pre class="notranslate"><code>## [1] FALSE
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='ifelse(!dir.exists(file.path(paste0("../2_Output/", COMPARISON))), dir.create(file.path(paste0("../2_Output/", COMPARISON))), FALSE)' dir="auto">
                     <pre>ifelse(<span class="pl-k">!</span>dir.exists(file.path(paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>))), dir.create(file.path(paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>))), <span class="pl-c1">FALSE</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## [1] FALSE">
                     <pre class="notranslate"><code>## [1] FALSE
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Packages
                     </h2>
                     <a aria-label="Permalink: Packages" class="anchor" href="#packages" id="user-content-packages">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='if (!require("pacman")) install.packages("pacman")
pacman::p_load(dplyr, Hmisc, openxlsx, corrplot, RColorBrewer, kableExtra, ggplot2, gridExtra, ggpubr, ggsignif, DESeq2, data.table, GenomicFeatures, biomaRt, Haplin, pheatmap, calibrate, ggrepel, tidyr, gtools)' dir="auto">
                     <pre><span class="pl-k">if</span> (<span class="pl-k">!</span>require(<span class="pl-s"><span class="pl-pds">"</span>pacman<span class="pl-pds">"</span></span>)) install.packages(<span class="pl-s"><span class="pl-pds">"</span>pacman<span class="pl-pds">"</span></span>)
<span class="pl-e">pacman</span><span class="pl-k">::</span>p_load(<span class="pl-smi">dplyr</span>, <span class="pl-smi">Hmisc</span>, <span class="pl-smi">openxlsx</span>, <span class="pl-smi">corrplot</span>, <span class="pl-smi">RColorBrewer</span>, <span class="pl-smi">kableExtra</span>, <span class="pl-smi">ggplot2</span>, <span class="pl-smi">gridExtra</span>, <span class="pl-smi">ggpubr</span>, <span class="pl-smi">ggsignif</span>, <span class="pl-smi">DESeq2</span>, <span class="pl-smi">data.table</span>, <span class="pl-smi">GenomicFeatures</span>, <span class="pl-smi">biomaRt</span>, <span class="pl-smi">Haplin</span>, <span class="pl-smi">pheatmap</span>, <span class="pl-smi">calibrate</span>, <span class="pl-smi">ggrepel</span>, <span class="pl-smi">tidyr</span>, <span class="pl-smi">gtools</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Genome-wide DNA methylation - Illumina(R) HumanMethylation 450k Methylation
                     </h1>
                     <a aria-label="Permalink: Genome-wide DNA methylation - Illumina(R) HumanMethylation 450k Methylation" class="anchor" href="#genome-wide-dna-methylation---illuminar-humanmethylation-450k-methylation" id="user-content-genome-wide-dna-methylation---illuminar-humanmethylation-450k-methylation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(minfi)
library(limma)
library(shinyMethyl)
library(dplyr)
library(IlluminaHumanMethylation450kmanifest)
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)
library(RColorBrewer)
#########Part 1: Importing the Data
#Parameters
##Get the array annotation
annoM450k&lt;-getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19)
annoM450k&lt;-dplyr::select(as.data.frame(annoM450k), Name, chr, pos, Relation_to_Island, UCSC_RefGene_Name, UCSC_RefGene_Accession, UCSC_RefGene_Group, Regulatory_Feature_Group)
#Import the sample sheet
targets&lt;-read.metharray.sheet(base="../1_Input/IDAT", pattern=".csv")' dir="auto">
                     <pre>library(<span class="pl-smi">minfi</span>)
library(<span class="pl-smi">limma</span>)
library(<span class="pl-smi">shinyMethyl</span>)
library(<span class="pl-smi">dplyr</span>)
library(<span class="pl-smi">IlluminaHumanMethylation450kmanifest</span>)
library(<span class="pl-smi">IlluminaHumanMethylation450kanno.ilmn12.hg19</span>)
library(<span class="pl-smi">RColorBrewer</span>)
<span class="pl-c"><span class="pl-c">#</span>########Part 1: Importing the Data</span>
<span class="pl-c"><span class="pl-c">#</span>Parameters</span>
<span class="pl-c"><span class="pl-c">#</span>#Get the array annotation</span>
<span class="pl-smi">annoM450k</span><span class="pl-k">&lt;-</span>getAnnotation(<span class="pl-smi">IlluminaHumanMethylation450kanno.ilmn12.hg19</span>)
<span class="pl-smi">annoM450k</span><span class="pl-k">&lt;-</span><span class="pl-e">dplyr</span><span class="pl-k">::</span>select(as.data.frame(<span class="pl-smi">annoM450k</span>), <span class="pl-smi">Name</span>, <span class="pl-smi">chr</span>, <span class="pl-smi">pos</span>, <span class="pl-smi">Relation_to_Island</span>, <span class="pl-smi">UCSC_RefGene_Name</span>, <span class="pl-smi">UCSC_RefGene_Accession</span>, <span class="pl-smi">UCSC_RefGene_Group</span>, <span class="pl-smi">Regulatory_Feature_Group</span>)
<span class="pl-c"><span class="pl-c">#</span>Import the sample sheet</span>
<span class="pl-smi">targets</span><span class="pl-k">&lt;-</span>read.metharray.sheet(<span class="pl-v">base</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>../1_Input/IDAT<span class="pl-pds">"</span></span>, <span class="pl-v">pattern</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.csv<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## [1] "../1_Input/IDAT/SampleSheet.csv"'>
                     <pre class="notranslate"><code>## [1] "../1_Input/IDAT/SampleSheet.csv"
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=' #Import the annotation file
AnnoTargets&lt;-targets
targets$ID &lt;- paste(targets$Sample_Group,targets$Sample_Name,sep=".")
targets$Sample_Well&lt;-AnnoTargets$Sample_Well
targets$Sample_Plate&lt;-AnnoTargets$Sample_Plate
targets$Sample_Group&lt;-AnnoTargets$Sample_Group
targets$Sample_Group&lt;-factor(targets$Sample_Group, levels = c("CON", "DCM"))
targets$Sex&lt;-AnnoTargets$Sex
targets$Outcome&lt;-AnnoTargets$Outcome
targets$Conc&lt;-AnnoTargets$Conc
targets$Col_ID&lt;-AnnoTargets$Col_ID
targets$Tissue&lt;-AnnoTargets$Tissue
targets&lt;-targets %&gt;% filter(Tissue %in% CELL)

#Import the array data from input directory (red and green .idat files)
RGSet&lt;-read.metharray.exp(base = "../1_Input/IDAT", targets = targets, verbose = TRUE)
sampleNames(RGSet)&lt;-targets$Sample_Name
##Quality Control
#First step is to identify CpGs that failed to identify methylated positions (defined by expression intensity that reflects background levels)
detP&lt;-detectionP(RGSet)
PLOT.COL &lt;- brewer.pal(8,"Dark2")
##Detection P-value Plot
pdf(file=paste0("../2_Output/", COMPARISON, "/", COMPARISON, "_detectionP.pdf"))
par(mfrow=c(1,1))
barplot(colMeans(detP), col=PLOT.COL[factor(targets$Sample_Group)], 
        cex.names=0.8, ylim=c(0,0.005), ylab="Mean detection p-values")
abline(h=0.05,col="red")
legend("topleft", legend=levels(factor(targets$Sample_Group)), fill=PLOT.COL, 
       bg="white")
dev.off()' dir="auto">
                     <pre> <span class="pl-c"><span class="pl-c">#</span>Import the annotation file</span>
<span class="pl-smi">AnnoTargets</span><span class="pl-k">&lt;-</span><span class="pl-smi">targets</span>
<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">ID</span> <span class="pl-k">&lt;-</span> paste(<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>,<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>)
<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Well</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Well</span>
<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Plate</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Plate</span>
<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>
<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span><span class="pl-k">&lt;-</span><span class="pl-k">factor</span>(<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>, <span class="pl-v">levels</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>CON<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>DCM<span class="pl-pds">"</span></span>))
<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sex</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span><span class="pl-k">$</span><span class="pl-smi">Sex</span>
<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Outcome</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span><span class="pl-k">$</span><span class="pl-smi">Outcome</span>
<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Conc</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span><span class="pl-k">$</span><span class="pl-smi">Conc</span>
<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Col_ID</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span><span class="pl-k">$</span><span class="pl-smi">Col_ID</span>
<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Tissue</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span><span class="pl-k">$</span><span class="pl-smi">Tissue</span>
<span class="pl-smi">targets</span><span class="pl-k">&lt;-</span><span class="pl-smi">targets</span> %<span class="pl-k">&gt;</span>% filter(<span class="pl-smi">Tissue</span> <span class="pl-k">%in%</span> <span class="pl-smi">CELL</span>)

<span class="pl-c"><span class="pl-c">#</span>Import the array data from input directory (red and green .idat files)</span>
<span class="pl-smi">RGSet</span><span class="pl-k">&lt;-</span>read.metharray.exp(<span class="pl-v">base</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>../1_Input/IDAT<span class="pl-pds">"</span></span>, <span class="pl-v">targets</span> <span class="pl-k">=</span> <span class="pl-smi">targets</span>, <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
sampleNames(<span class="pl-smi">RGSet</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>
<span class="pl-c"><span class="pl-c">#</span>#Quality Control</span>
<span class="pl-c"><span class="pl-c">#</span>First step is to identify CpGs that failed to identify methylated positions (defined by expression intensity that reflects background levels)</span>
<span class="pl-smi">detP</span><span class="pl-k">&lt;-</span>detectionP(<span class="pl-smi">RGSet</span>)
<span class="pl-smi">PLOT.COL</span> <span class="pl-k">&lt;-</span> brewer.pal(<span class="pl-c1">8</span>,<span class="pl-s"><span class="pl-pds">"</span>Dark2<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span>#Detection P-value Plot</span>
pdf(<span class="pl-v">file</span><span class="pl-k">=</span>paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_detectionP.pdf<span class="pl-pds">"</span></span>))
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))
barplot(colMeans(<span class="pl-smi">detP</span>), <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">PLOT.COL</span>[<span class="pl-k">factor</span>(<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>)], 
        <span class="pl-v">cex.names</span><span class="pl-k">=</span><span class="pl-c1">0.8</span>, <span class="pl-v">ylim</span><span class="pl-k">=</span>c(<span class="pl-c1">0</span>,<span class="pl-c1">0.005</span>), <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Mean detection p-values<span class="pl-pds">"</span></span>)
abline(<span class="pl-v">h</span><span class="pl-k">=</span><span class="pl-c1">0.05</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>)
legend(<span class="pl-s"><span class="pl-pds">"</span>topleft<span class="pl-pds">"</span></span>, <span class="pl-v">legend</span><span class="pl-k">=</span>levels(<span class="pl-k">factor</span>(<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>)), <span class="pl-v">fill</span><span class="pl-k">=</span><span class="pl-smi">PLOT.COL</span>, 
       <span class="pl-v">bg</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## quartz_off_screen 
##                 2">
                     <pre class="notranslate"><code>## quartz_off_screen 
##                 2
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Filter RGSet by detection P-value
keep &lt;- colMeans(detP) &lt; 0.05
RGSet &lt;- RGSet[,keep]
RGSet" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># Filter RGSet by detection P-value</span>
<span class="pl-smi">keep</span> <span class="pl-k">&lt;-</span> colMeans(<span class="pl-smi">detP</span>) <span class="pl-k">&lt;</span> <span class="pl-c1">0.05</span>
<span class="pl-smi">RGSet</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">RGSet</span>[,<span class="pl-smi">keep</span>]
<span class="pl-smi">RGSet</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## class: RGChannelSet 
## dim: 622399 24 
## metadata(0):
## assays(2): Green Red
## rownames(622399): 10600313 10600322 ... 74810490 74810492
## rowData names(0):
## colnames(24): A1_cells_bo A2_cells_bo ... C9_cells_co C10_cells_co
## colData names(17): Sample_Name Sample_Well ... ID filenames
## Annotation
##   array: IlluminaHumanMethylation450k
##   annotation: ilmn12.hg19">
                     <pre class="notranslate"><code>## class: RGChannelSet 
## dim: 622399 24 
## metadata(0):
## assays(2): Green Red
## rownames(622399): 10600313 10600322 ... 74810490 74810492
## rowData names(0):
## colnames(24): A1_cells_bo A2_cells_bo ... C9_cells_co C10_cells_co
## colData names(17): Sample_Name Sample_Well ... ID filenames
## Annotation
##   array: IlluminaHumanMethylation450k
##   annotation: ilmn12.hg19
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='##QC Report
qcReport(RGSet, sampNames=targets$ID, sampGroups=targets$Sample_Group, pdf="qcReport.pdf")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>#QC Report</span>
qcReport(<span class="pl-smi">RGSet</span>, <span class="pl-v">sampNames</span><span class="pl-k">=</span><span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">ID</span>, <span class="pl-v">sampGroups</span><span class="pl-k">=</span><span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>, <span class="pl-v">pdf</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>qcReport.pdf<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## quartz_off_screen 
##                 2">
                     <pre class="notranslate"><code>## quartz_off_screen 
##                 2
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#
par(mfrow=c(1,1))
barplot(colMeans(detP), col=PLOT.COL[factor(targets$Sample_Group)], las=2, 
        cex.names=0.8, ylim=c(0,0.005), ylab="Mean detection p-values")
abline(h=0.05,col="red")
legend("topleft", legend=levels(factor(targets$Sample_Group)), fill=PLOT.COL, 
       bg="white")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span></span>
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))
barplot(colMeans(<span class="pl-smi">detP</span>), <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">PLOT.COL</span>[<span class="pl-k">factor</span>(<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>)], <span class="pl-v">las</span><span class="pl-k">=</span><span class="pl-c1">2</span>, 
        <span class="pl-v">cex.names</span><span class="pl-k">=</span><span class="pl-c1">0.8</span>, <span class="pl-v">ylim</span><span class="pl-k">=</span>c(<span class="pl-c1">0</span>,<span class="pl-c1">0.005</span>), <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Mean detection p-values<span class="pl-pds">"</span></span>)
abline(<span class="pl-v">h</span><span class="pl-k">=</span><span class="pl-c1">0.05</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>)
legend(<span class="pl-s"><span class="pl-pds">"</span>topleft<span class="pl-pds">"</span></span>, <span class="pl-v">legend</span><span class="pl-k">=</span>levels(<span class="pl-k">factor</span>(<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>)), <span class="pl-v">fill</span><span class="pl-k">=</span><span class="pl-smi">PLOT.COL</span>, 
       <span class="pl-v">bg</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/EPIC_Import-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/EPIC_Import-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# #Determine the fraction of "failed" CpG probes (those which failed to identify a methylated CpG)
# colMeans(failed)
#Convert R/G to Methylated/Unmethylated in an object of class MethylSet
MSet&lt;-preprocessRaw(RGSet)
#QC data
qc&lt;-getQC(MSet)
plotQC(qc)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> #Determine the fraction of "failed" CpG probes (those which failed to identify a methylated CpG)</span>
<span class="pl-c"><span class="pl-c">#</span> colMeans(failed)</span>
<span class="pl-c"><span class="pl-c">#</span>Convert R/G to Methylated/Unmethylated in an object of class MethylSet</span>
<span class="pl-smi">MSet</span><span class="pl-k">&lt;-</span>preprocessRaw(<span class="pl-smi">RGSet</span>)
<span class="pl-c"><span class="pl-c">#</span>QC data</span>
<span class="pl-smi">qc</span><span class="pl-k">&lt;-</span>getQC(<span class="pl-smi">MSet</span>)
plotQC(<span class="pl-smi">qc</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/EPIC_Import-2.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/EPIC_Import-2.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf(file=paste0("../2_Output/", COMPARISON, "/", COMPARISON, "_QC.Methyl.pdf"))
plotQC(qc)
dev.off()' dir="auto">
                     <pre>pdf(<span class="pl-v">file</span><span class="pl-k">=</span>paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_QC.Methyl.pdf<span class="pl-pds">"</span></span>))
plotQC(<span class="pl-smi">qc</span>)
dev.off()</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## quartz_off_screen 
##                 2">
                     <pre class="notranslate"><code>## quartz_off_screen 
##                 2
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='##Density plot
# densityPlot(RGSet, sampGroups = targets$Sample_Group, main= "Beta", xlab = "Beta")
pdf(file=paste0("../2_Output/", COMPARISON, "/", COMPARISON, "_densityPlot.pdf"))
densityPlot(RGSet, sampGroups = targets$Sample_Group, main= "Beta", xlab = "Beta")
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>#Density plot</span>
<span class="pl-c"><span class="pl-c">#</span> densityPlot(RGSet, sampGroups = targets$Sample_Group, main= "Beta", xlab = "Beta")</span>
pdf(<span class="pl-v">file</span><span class="pl-k">=</span>paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_densityPlot.pdf<span class="pl-pds">"</span></span>))
densityPlot(<span class="pl-smi">RGSet</span>, <span class="pl-v">sampGroups</span> <span class="pl-k">=</span> <span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>, <span class="pl-v">main</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Beta<span class="pl-pds">"</span></span>, <span class="pl-v">xlab</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Beta<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## quartz_off_screen 
##                 2">
                     <pre class="notranslate"><code>## quartz_off_screen 
##                 2
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="densityBeanPlot(RGSet, sampGroups = targets$Sample_Group)" dir="auto">
                     <pre>densityBeanPlot(<span class="pl-smi">RGSet</span>, <span class="pl-v">sampGroups</span> <span class="pl-k">=</span> <span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/EPIC_Import-3.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/EPIC_Import-3.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf(file=paste0("../2_Output/", COMPARISON, "/", COMPARISON, "_BeanPlot.pdf"))
densityBeanPlot(RGSet, sampGroups = targets$Sample_Group)
dev.off()' dir="auto">
                     <pre>pdf(<span class="pl-v">file</span><span class="pl-k">=</span>paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_BeanPlot.pdf<span class="pl-pds">"</span></span>))
densityBeanPlot(<span class="pl-smi">RGSet</span>, <span class="pl-v">sampGroups</span> <span class="pl-k">=</span> <span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>)
dev.off()</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## quartz_off_screen 
##                 2">
                     <pre class="notranslate"><code>## quartz_off_screen 
##                 2
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='qcReport(RGSet, pdf= paste0("../2_Output/", COMPARISON, "/", COMPARISON, "_qcReport.pdf"))' dir="auto">
                     <pre>qcReport(<span class="pl-smi">RGSet</span>, <span class="pl-v">pdf</span><span class="pl-k">=</span> paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_qcReport.pdf<span class="pl-pds">"</span></span>))</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## quartz_off_screen 
##                 2">
                     <pre class="notranslate"><code>## quartz_off_screen 
##                 2
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# #Convert to a shinyMethyl dataset (nice summary of the quality control)
# summarized.data &lt;- shinySummarize(RGSet)
# runShinyMethyl(summarized.data)

gRatioSet.quantile &lt;- preprocessQuantile(RGSet, fixOutliers = TRUE, removeBadSamples = TRUE, badSampleCutoff = 10.5, quantileNormalize = TRUE, stratified = TRUE, mergeManifest = FALSE)
beta.all&lt;-getBeta(RGSet)
write.csv(beta.all, "../1_Input/EPIC.betaValues.csv")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> #Convert to a shinyMethyl dataset (nice summary of the quality control)</span>
<span class="pl-c"><span class="pl-c">#</span> summarized.data &lt;- shinySummarize(RGSet)</span>
<span class="pl-c"><span class="pl-c">#</span> runShinyMethyl(summarized.data)</span>

<span class="pl-smi">gRatioSet.quantile</span> <span class="pl-k">&lt;-</span> preprocessQuantile(<span class="pl-smi">RGSet</span>, <span class="pl-v">fixOutliers</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">removeBadSamples</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">badSampleCutoff</span> <span class="pl-k">=</span> <span class="pl-c1">10.5</span>, <span class="pl-v">quantileNormalize</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">stratified</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">mergeManifest</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">beta.all</span><span class="pl-k">&lt;-</span>getBeta(<span class="pl-smi">RGSet</span>)
write.csv(<span class="pl-smi">beta.all</span>, <span class="pl-s"><span class="pl-pds">"</span>../1_Input/EPIC.betaValues.csv<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Supplemental Figure SXX: Outlier Analysis
                     </h2>
                     <a aria-label="Permalink: Supplemental Figure SXX: Outlier Analysis" class="anchor" href="#supplemental-figure-sxx-outlier-analysis" id="user-content-supplemental-figure-sxx-outlier-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='par(mfrow=c(1,1))
barplot(colMeans(detP), col=PLOT.COL[factor(targets$Sample_Group)], las=2, 
        cex.names=0.8, ylim=c(0,0.005), ylab="Mean detection p-values")
abline(h=0.05,col="red")
legend("topleft", legend=levels(factor(targets$Sample_Group)), fill=PLOT.COL, 
       bg="white")' dir="auto">
                     <pre>par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))
barplot(colMeans(<span class="pl-smi">detP</span>), <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">PLOT.COL</span>[<span class="pl-k">factor</span>(<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>)], <span class="pl-v">las</span><span class="pl-k">=</span><span class="pl-c1">2</span>, 
        <span class="pl-v">cex.names</span><span class="pl-k">=</span><span class="pl-c1">0.8</span>, <span class="pl-v">ylim</span><span class="pl-k">=</span>c(<span class="pl-c1">0</span>,<span class="pl-c1">0.005</span>), <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Mean detection p-values<span class="pl-pds">"</span></span>)
abline(<span class="pl-v">h</span><span class="pl-k">=</span><span class="pl-c1">0.05</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>)
legend(<span class="pl-s"><span class="pl-pds">"</span>topleft<span class="pl-pds">"</span></span>, <span class="pl-v">legend</span><span class="pl-k">=</span>levels(<span class="pl-k">factor</span>(<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>)), <span class="pl-v">fill</span><span class="pl-k">=</span><span class="pl-smi">PLOT.COL</span>, 
       <span class="pl-v">bg</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/outliers-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Supplemental Figure" src="/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/outliers-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Figure 2A: MDS Plot - Unsupervised Clustering by "Sample_Group"
                     </h2>
                     <a aria-label='Permalink: Figure 2A: MDS Plot - Unsupervised Clustering by "Sample_Group"' class="anchor" href="#figure-2a-mds-plot---unsupervised-clustering-by-sample_group" id="user-content-figure-2a-mds-plot---unsupervised-clustering-by-sample_group">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Because such a robust racial signature in cardiac DNA methylation was
seen in the pilot analysis, we reproduced the unsupervised method in the
larger cohort. This time, we continue to see a distinct racial
difference. Furthermore, we found that this racially-determined
clustering persisted to among the 500,000 most-variable CpG probes in
the EPIC array.
                    </p>
                    <p dir="auto">
                     <a href="/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/MDS.methyl-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/MDS.methyl-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      PCA
                     </h2>
                     <a aria-label="Permalink: PCA" class="anchor" href="#pca" id="user-content-pca">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#Plot Features of the PCA
library(dplyr)
library(plotly)
##Import the data to be used for PCA
Index_PCA&lt;-targets
rownames(Index_PCA)&lt;-Index_PCA$Sample_Name
PCA_data&lt;-as.data.frame(getM(gRatioSet.quantile))
#transpose the dataset (required for PCA)
data.pca&lt;-t(PCA_data)
data.pca&lt;-as.data.frame(data.pca)
##merge the file
data.pca_Final&lt;-merge(Index_PCA, data.pca, by=0)
rownames(data.pca_Final)&lt;-data.pca_Final$Row.names
pca.comp&lt;-prcomp(data.pca_Final[,(ncol(Index_PCA)+2):ncol(data.pca_Final)])

pcaCharts=function(x) {
    x.var &lt;- x$sdev ^ 2
    x.pvar &lt;- x.var/sum(x.var)
    par(mfrow=c(2,2))
    plot(x.pvar,xlab=&quot;Principal component&quot;,
         ylab=&quot;Proportion of variance&quot;, ylim=c(0,1), type='b')
    plot(cumsum(x.pvar),xlab=&quot;Principal component&quot;,
         ylab=&quot;Cumulative Proportion of variance&quot;,
         ylim=c(0,1),
         type='b')
    screeplot(x)
    screeplot(x,type=&quot;l&quot;)
    par(mfrow=c(1,1))
}
pcaCharts(pca.comp)" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>Plot Features of the PCA</span>
library(<span class="pl-smi">dplyr</span>)
library(<span class="pl-smi">plotly</span>)
<span class="pl-c"><span class="pl-c">#</span>#Import the data to be used for PCA</span>
<span class="pl-smi">Index_PCA</span><span class="pl-k">&lt;-</span><span class="pl-smi">targets</span>
rownames(<span class="pl-smi">Index_PCA</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">Index_PCA</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>
<span class="pl-smi">PCA_data</span><span class="pl-k">&lt;-</span>as.data.frame(getM(<span class="pl-smi">gRatioSet.quantile</span>))
<span class="pl-c"><span class="pl-c">#</span>transpose the dataset (required for PCA)</span>
<span class="pl-smi">data.pca</span><span class="pl-k">&lt;-</span>t(<span class="pl-smi">PCA_data</span>)
<span class="pl-smi">data.pca</span><span class="pl-k">&lt;-</span>as.data.frame(<span class="pl-smi">data.pca</span>)
<span class="pl-c"><span class="pl-c">#</span>#merge the file</span>
<span class="pl-smi">data.pca_Final</span><span class="pl-k">&lt;-</span>merge(<span class="pl-smi">Index_PCA</span>, <span class="pl-smi">data.pca</span>, <span class="pl-v">by</span><span class="pl-k">=</span><span class="pl-c1">0</span>)
rownames(<span class="pl-smi">data.pca_Final</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">data.pca_Final</span><span class="pl-k">$</span><span class="pl-smi">Row.names</span>
<span class="pl-smi">pca.comp</span><span class="pl-k">&lt;-</span>prcomp(<span class="pl-smi">data.pca_Final</span>[,(ncol(<span class="pl-smi">Index_PCA</span>)<span class="pl-k">+</span><span class="pl-c1">2</span>)<span class="pl-k">:</span>ncol(<span class="pl-smi">data.pca_Final</span>)])

<span class="pl-en">pcaCharts</span><span class="pl-k">=</span><span class="pl-k">function</span>(<span class="pl-smi">x</span>) {
    <span class="pl-smi">x.var</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">x</span><span class="pl-k">$</span><span class="pl-smi">sdev</span> <span class="pl-k">^</span> <span class="pl-c1">2</span>
    <span class="pl-smi">x.pvar</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">x.var</span><span class="pl-k">/</span>sum(<span class="pl-smi">x.var</span>)
    par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">2</span>))
    plot(<span class="pl-smi">x.pvar</span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Principal component<span class="pl-pds">"</span></span>,
         <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Proportion of variance<span class="pl-pds">"</span></span>, <span class="pl-v">ylim</span><span class="pl-k">=</span>c(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>), <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>b<span class="pl-pds">'</span></span>)
    plot(cumsum(<span class="pl-smi">x.pvar</span>),<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Principal component<span class="pl-pds">"</span></span>,
         <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Cumulative Proportion of variance<span class="pl-pds">"</span></span>,
         <span class="pl-v">ylim</span><span class="pl-k">=</span>c(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>),
         <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>b<span class="pl-pds">'</span></span>)
    screeplot(<span class="pl-smi">x</span>)
    screeplot(<span class="pl-smi">x</span>,<span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>l<span class="pl-pds">"</span></span>)
    par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))
}
pcaCharts(<span class="pl-smi">pca.comp</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/PCA_Features-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/PCA_Features-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='png(file=paste0("../2_Output/", COMPARISON,  "/", COMPARISON, "_PCA.Charts.png"))
pcaCharts(pca.comp)
dev.off()' dir="auto">
                     <pre>png(<span class="pl-v">file</span><span class="pl-k">=</span>paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>,  <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_PCA.Charts.png<span class="pl-pds">"</span></span>))
pcaCharts(<span class="pl-smi">pca.comp</span>)
dev.off()</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## quartz_off_screen 
##                 2">
                     <pre class="notranslate"><code>## quartz_off_screen 
##                 2
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      3-Dimensional PCA
                     </h3>
                     <a aria-label="Permalink: 3-Dimensional PCA" class="anchor" href="#3-dimensional-pca" id="user-content-3-dimensional-pca">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     From the previous calculations, it is seens that only 2 principal
components are necessary (accounting for &gt;80% cumulative variance).
Nonetheless, below is a 3-D PCA to ensure that all groups are
characterize to higher-degree of stringency. Nevertheless, a racial
difference could not be appreciated.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="##Create a 3D-PCA for Inspection
library(plotly)
##Index
Index_PCA&lt;-targets
rownames(Index_PCA)&lt;-Index_PCA$Sample_Name

PCs&lt;-merge(pca.comp$x, Index_PCA, by=0)
rownames(PCs)&lt;-PCs$Row.names
PCs$Group &lt;- as.factor(PCs$Sample_Group)
fig &lt;- plot_ly(PCs, x = ~PC1, y = ~PC2, z = ~PC3, color = ~Sample_Group, text = ~paste('Sample_Name:', Sample_Name, '&lt;br&gt;Tissue:', Tissue, '&lt;br&gt;Outcome:', Outcome, '&lt;br&gt;Sex:', Sex))
fig &lt;- fig %&gt;% add_markers()
fig &lt;- fig %&gt;% layout(scene = list(xaxis = list(title = 'PC1'),
                     yaxis = list(title = 'PC2'),
                     zaxis = list(title = 'PC3')))
fig" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>#Create a 3D-PCA for Inspection</span>
library(<span class="pl-smi">plotly</span>)
<span class="pl-c"><span class="pl-c">#</span>#Index</span>
<span class="pl-smi">Index_PCA</span><span class="pl-k">&lt;-</span><span class="pl-smi">targets</span>
rownames(<span class="pl-smi">Index_PCA</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">Index_PCA</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>

<span class="pl-smi">PCs</span><span class="pl-k">&lt;-</span>merge(<span class="pl-smi">pca.comp</span><span class="pl-k">$</span><span class="pl-smi">x</span>, <span class="pl-smi">Index_PCA</span>, <span class="pl-v">by</span><span class="pl-k">=</span><span class="pl-c1">0</span>)
rownames(<span class="pl-smi">PCs</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">PCs</span><span class="pl-k">$</span><span class="pl-smi">Row.names</span>
<span class="pl-smi">PCs</span><span class="pl-k">$</span><span class="pl-smi">Group</span> <span class="pl-k">&lt;-</span> as.factor(<span class="pl-smi">PCs</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>)
<span class="pl-smi">fig</span> <span class="pl-k">&lt;-</span> plot_ly(<span class="pl-smi">PCs</span>, <span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-k">~</span><span class="pl-smi">PC1</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-k">~</span><span class="pl-smi">PC2</span>, <span class="pl-v">z</span> <span class="pl-k">=</span> <span class="pl-k">~</span><span class="pl-smi">PC3</span>, <span class="pl-v">color</span> <span class="pl-k">=</span> <span class="pl-k">~</span><span class="pl-smi">Sample_Group</span>, <span class="pl-v">text</span> <span class="pl-k">=</span> <span class="pl-k">~</span>paste(<span class="pl-s"><span class="pl-pds">'</span>Sample_Name:<span class="pl-pds">'</span></span>, <span class="pl-smi">Sample_Name</span>, <span class="pl-s"><span class="pl-pds">'</span>&lt;br&gt;Tissue:<span class="pl-pds">'</span></span>, <span class="pl-smi">Tissue</span>, <span class="pl-s"><span class="pl-pds">'</span>&lt;br&gt;Outcome:<span class="pl-pds">'</span></span>, <span class="pl-smi">Outcome</span>, <span class="pl-s"><span class="pl-pds">'</span>&lt;br&gt;Sex:<span class="pl-pds">'</span></span>, <span class="pl-smi">Sex</span>))
<span class="pl-smi">fig</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">fig</span> %<span class="pl-k">&gt;</span>% add_markers()
<span class="pl-smi">fig</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">fig</span> %<span class="pl-k">&gt;</span>% layout(<span class="pl-v">scene</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-v">xaxis</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>PC1<span class="pl-pds">'</span></span>),
                     <span class="pl-v">yaxis</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>PC2<span class="pl-pds">'</span></span>),
                     <span class="pl-v">zaxis</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>PC3<span class="pl-pds">'</span></span>)))
<span class="pl-smi">fig</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='&lt;div id="htmlwidget-0ed3cff61cab74c0c287" style="width:672px;height:480px;" class="plotly html-widget"&gt;&lt;/div&gt;
&lt;script type="application/json" data-for="htmlwidget-0ed3cff61cab74c0c287"&gt;{"x":{"visdat":{"2ee72d1f4380":["function () ","plotlyVisDat"]},"cur_data":"2ee72d1f4380","attrs":{"2ee72d1f4380":{"x":{},"y":{},"z":{},"text":{},"color":{},"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter3d","mode":"markers","inherit":true}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"scene":{"xaxis":{"title":"PC1"},"yaxis":{"title":"PC2"},"zaxis":{"title":"PC3"}},"hovermode":"closest","showlegend":true},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"x":[77.9600791014893,-8.85871319983235,-2.3258763441347,-7.91557474476333,-5.12544124055454,-6.31032622733963,-15.5555084612836,19.1451274289372,-0.370810911338616,-7.2301707948147],"y":[63.3739250059267,-28.5775366493097,-38.9831271081527,-26.5114827922627,-26.5322218237842,-24.6351203021513,-25.5921588263502,19.6910128735114,-34.7694530835565,-19.4396082462673],"z":[131.113571695425,-7.98668393195755,-25.645441514114,-14.9352103367056,-7.8256237104918,-3.54404886501356,-13.3068841181743,19.0115820730706,-26.0649079763584,-9.98096383837802],"text":["Sample_Name: C1_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C10_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: F","Sample_Name: C2_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C3_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: F","Sample_Name: C4_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C5_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C6_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C7_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C8_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: F","Sample_Name: C9_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: F"],"type":"scatter3d","mode":"markers","name":"CON","marker":{"color":"rgba(102,194,165,1)","line":{"color":"rgba(102,194,165,1)"}},"textfont":{"color":"rgba(102,194,165,1)"},"error_y":{"color":"rgba(102,194,165,1)"},"error_x":{"color":"rgba(102,194,165,1)"},"line":{"color":"rgba(102,194,165,1)"},"frame":null},{"x":[-74.840661246498,-72.6476533955918,-59.4408574966577,-13.7136537072147,-4.21579120231104,6.11119549549595,9.59163470727785,5.85414321742238,173.476095054373,-15.3097146431571,-13.7765637170266,0.319854833899074,6.97727589132507,8.20191160229823],"y":[66.6389631010503,89.1024717499668,120.443036300283,-28.4395221216847,-22.325597162093,-26.7115831833687,-22.6504061481375,-15.1207211276863,64.8545317043899,-25.7647846869283,-27.3850982890718,-12.2567307943521,-18.5256954472338,0.116907057264034],"z":[-23.0507887812109,-16.1327461317539,-15.3423208102994,-3.59376989204799,1.84807232198693,12.2145192980728,17.1138420638432,14.8393139305453,-91.6382517593327,-8.60068453610674,0.447886121276813,18.7299603053965,20.7702635178944,31.5593148744268],"text":["Sample_Name: A1_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A2_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A3_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A4_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A5_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A6_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A7_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: F","Sample_Name: B1_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M","Sample_Name: B2_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M","Sample_Name: B3_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M","Sample_Name: B4_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M","Sample_Name: B5_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M","Sample_Name: B6_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: F","Sample_Name: B7_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M"],"type":"scatter3d","mode":"markers","name":"DCM","marker":{"color":"rgba(141,160,203,1)","line":{"color":"rgba(141,160,203,1)"}},"textfont":{"color":"rgba(141,160,203,1)"},"error_y":{"color":"rgba(141,160,203,1)"},"error_x":{"color":"rgba(141,160,203,1)"},"line":{"color":"rgba(141,160,203,1)"},"frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}&lt;/script&gt;'>
                     <pre class="notranslate" lang="{=html}"><code>&lt;div id="htmlwidget-0ed3cff61cab74c0c287" style="width:672px;height:480px;" class="plotly html-widget"&gt;&lt;/div&gt;
&lt;script type="application/json" data-for="htmlwidget-0ed3cff61cab74c0c287"&gt;{"x":{"visdat":{"2ee72d1f4380":["function () ","plotlyVisDat"]},"cur_data":"2ee72d1f4380","attrs":{"2ee72d1f4380":{"x":{},"y":{},"z":{},"text":{},"color":{},"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter3d","mode":"markers","inherit":true}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"scene":{"xaxis":{"title":"PC1"},"yaxis":{"title":"PC2"},"zaxis":{"title":"PC3"}},"hovermode":"closest","showlegend":true},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"x":[77.9600791014893,-8.85871319983235,-2.3258763441347,-7.91557474476333,-5.12544124055454,-6.31032622733963,-15.5555084612836,19.1451274289372,-0.370810911338616,-7.2301707948147],"y":[63.3739250059267,-28.5775366493097,-38.9831271081527,-26.5114827922627,-26.5322218237842,-24.6351203021513,-25.5921588263502,19.6910128735114,-34.7694530835565,-19.4396082462673],"z":[131.113571695425,-7.98668393195755,-25.645441514114,-14.9352103367056,-7.8256237104918,-3.54404886501356,-13.3068841181743,19.0115820730706,-26.0649079763584,-9.98096383837802],"text":["Sample_Name: C1_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C10_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: F","Sample_Name: C2_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C3_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: F","Sample_Name: C4_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C5_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C6_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C7_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: M","Sample_Name: C8_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: F","Sample_Name: C9_cells_co &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: CON &lt;br&gt;Sex: F"],"type":"scatter3d","mode":"markers","name":"CON","marker":{"color":"rgba(102,194,165,1)","line":{"color":"rgba(102,194,165,1)"}},"textfont":{"color":"rgba(102,194,165,1)"},"error_y":{"color":"rgba(102,194,165,1)"},"error_x":{"color":"rgba(102,194,165,1)"},"line":{"color":"rgba(102,194,165,1)"},"frame":null},{"x":[-74.840661246498,-72.6476533955918,-59.4408574966577,-13.7136537072147,-4.21579120231104,6.11119549549595,9.59163470727785,5.85414321742238,173.476095054373,-15.3097146431571,-13.7765637170266,0.319854833899074,6.97727589132507,8.20191160229823],"y":[66.6389631010503,89.1024717499668,120.443036300283,-28.4395221216847,-22.325597162093,-26.7115831833687,-22.6504061481375,-15.1207211276863,64.8545317043899,-25.7647846869283,-27.3850982890718,-12.2567307943521,-18.5256954472338,0.116907057264034],"z":[-23.0507887812109,-16.1327461317539,-15.3423208102994,-3.59376989204799,1.84807232198693,12.2145192980728,17.1138420638432,14.8393139305453,-91.6382517593327,-8.60068453610674,0.447886121276813,18.7299603053965,20.7702635178944,31.5593148744268],"text":["Sample_Name: A1_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A2_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A3_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A4_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A5_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A6_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: M","Sample_Name: A7_cells_bo &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Bad &lt;br&gt;Sex: F","Sample_Name: B1_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M","Sample_Name: B2_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M","Sample_Name: B3_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M","Sample_Name: B4_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M","Sample_Name: B5_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M","Sample_Name: B6_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: F","Sample_Name: B7_cells_go &lt;br&gt;Tissue: iPSC &lt;br&gt;Outcome: Good &lt;br&gt;Sex: M"],"type":"scatter3d","mode":"markers","name":"DCM","marker":{"color":"rgba(141,160,203,1)","line":{"color":"rgba(141,160,203,1)"}},"textfont":{"color":"rgba(141,160,203,1)"},"error_y":{"color":"rgba(141,160,203,1)"},"error_x":{"color":"rgba(141,160,203,1)"},"line":{"color":"rgba(141,160,203,1)"},"frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}&lt;/script&gt;
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Differential Methylation of iPSCs (DCM vs. CON)
                     </h2>
                     <a aria-label="Permalink: Differential Methylation of iPSCs (DCM vs. CON)" class="anchor" href="#differential-methylation-of-ipscs-dcm-vs-con" id="user-content-differential-methylation-of-ipscs-dcm-vs-con">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Using Minfi to compute differential expression
                     </h2>
                     <a aria-label="Permalink: Using Minfi to compute differential expression" class="anchor" href="#using-minfi-to-compute-differential-expression" id="user-content-using-minfi-to-compute-differential-expression">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Owing to the apparent separation by patient Sample_Group, we chose to
identify the CpG sites responsible for a Sample_Group-based epigenomic
difference.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#Quantification and Differential Expression Analysis
RGSet&lt;-read.metharray.exp(base = "../1_Input/IDAT", targets = targets, verbose = TRUE)
# sampleNames(RGSet)&lt;-targets$Sample_Name
GRset.funnorm &lt;- preprocessFunnorm(RGSet)
beta &lt;- getBeta(GRset.funnorm)
M&lt;-getM(GRset.funnorm)
Condition &lt;- pData(GRset.funnorm)$Sample_Group
dmp &lt;- dmpFinder(beta, pheno = Condition  , type = "categorical")
DMPs&lt;-merge(dmp, annoM450k, by= 0)
DMPs&lt;-as.data.frame(DMPs)
rownames(DMPs)&lt;-DMPs$Row.names
DMPs&lt;-DMPs %&gt;% dplyr::select(-Row.names)
#create an beta table
beta.table&lt;-as.data.frame(t(beta))
beta.table$Col_ID&lt;-rownames(beta.table)
beta_named&lt;-merge(beta.table, targets, by = "Col_ID")
rownames(beta_named)&lt;-beta_named$Sample_Name
beta_named&lt;-beta_named %&gt;% dplyr::select(-names(targets)) %&gt;% t()
write.csv(beta_named, paste0("../2_Output/", COMPARISON, "/", COMPARISON, "_beta.table.csv"))
#Add beta values to statistics
Results_dmp&lt;-merge(DMPs, beta_named, by = 0)
#Calculate Average CpG Methylation by Sample_Group
library(dplyr)
library(matrixStats)
Results&lt;-Results_dmp %&gt;% replace(is.na(.), 0) %&gt;% dplyr::mutate(
  DCM_SD = rowSds(as.matrix(Results_dmp[,targets$Sample_Name[targets$Sample_Group=="DCM"]])),
  DCM_Mean = rowMeans(as.matrix(Results_dmp[,targets$Sample_Name[targets$Sample_Group=="DCM"]])),
  CON_SD = rowSds(as.matrix(Results_dmp[,targets$Sample_Name[targets$Sample_Group=="CON"]])),
  CON_Mean = rowMeans(as.matrix(Results_dmp[,targets$Sample_Name[targets$Sample_Group=="CON"]])),
  Methylation.Diff=(DCM_Mean-CON_Mean)*100)
rownames(Results)&lt;-Results$Row.names
Results_dmp_p05&lt;-filter(Results, pval&lt;0.05)
Results_dmp_q05&lt;-filter(Results, qval&lt;0.05)
#########################################
#Identify Promoter-associated CpG Islands
library(tidyr)
PromCGI&lt;-dplyr::filter(Results_dmp_p05, grepl("Island", Relation_to_Island), grepl("TSS", UCSC_RefGene_Group))
#Separate Gene Names into unique rows
PromCGI_sep&lt;-PromCGI %&gt;% mutate(UCSC_RefGene_Name = strsplit(as.character(UCSC_RefGene_Name), ";")) %&gt;% unnest(UCSC_RefGene_Name) %&gt;% distinct()
#Save a copy of the countData
library(openxlsx)
wb_countData&lt;-createWorkbook()
addWorksheet(wb_countData, "Unfiltered")
  writeData(wb_countData, "Unfiltered", Results, startCol = 1)
addWorksheet(wb_countData, "P_0.05")
  writeData(wb_countData, "P_0.05", Results_dmp_p05, startCol = 1)
addWorksheet(wb_countData, "Q_0.05")
  writeData(wb_countData, "Q_0.05", Results_dmp_q05, startCol = 1)
addWorksheet(wb_countData, "Promoter.CGI")
  writeData(wb_countData, "Promoter.CGI", PromCGI_sep, startCol = 1)
saveWorkbook(wb_countData, file = paste0("../2_Output/", COMPARISON, "/", COMPARISON, "_DMPs.xlsx"), overwrite = TRUE)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>Quantification and Differential Expression Analysis</span>
<span class="pl-smi">RGSet</span><span class="pl-k">&lt;-</span>read.metharray.exp(<span class="pl-v">base</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>../1_Input/IDAT<span class="pl-pds">"</span></span>, <span class="pl-v">targets</span> <span class="pl-k">=</span> <span class="pl-smi">targets</span>, <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
<span class="pl-c"><span class="pl-c">#</span> sampleNames(RGSet)&lt;-targets$Sample_Name</span>
<span class="pl-smi">GRset.funnorm</span> <span class="pl-k">&lt;-</span> preprocessFunnorm(<span class="pl-smi">RGSet</span>)
<span class="pl-smi">beta</span> <span class="pl-k">&lt;-</span> getBeta(<span class="pl-smi">GRset.funnorm</span>)
<span class="pl-smi">M</span><span class="pl-k">&lt;-</span>getM(<span class="pl-smi">GRset.funnorm</span>)
<span class="pl-smi">Condition</span> <span class="pl-k">&lt;-</span> pData(<span class="pl-smi">GRset.funnorm</span>)<span class="pl-k">$</span><span class="pl-smi">Sample_Group</span>
<span class="pl-smi">dmp</span> <span class="pl-k">&lt;-</span> dmpFinder(<span class="pl-smi">beta</span>, <span class="pl-v">pheno</span> <span class="pl-k">=</span> <span class="pl-smi">Condition</span>  , <span class="pl-v">type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>categorical<span class="pl-pds">"</span></span>)
<span class="pl-smi">DMPs</span><span class="pl-k">&lt;-</span>merge(<span class="pl-smi">dmp</span>, <span class="pl-smi">annoM450k</span>, <span class="pl-v">by</span><span class="pl-k">=</span> <span class="pl-c1">0</span>)
<span class="pl-smi">DMPs</span><span class="pl-k">&lt;-</span>as.data.frame(<span class="pl-smi">DMPs</span>)
rownames(<span class="pl-smi">DMPs</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">DMPs</span><span class="pl-k">$</span><span class="pl-smi">Row.names</span>
<span class="pl-smi">DMPs</span><span class="pl-k">&lt;-</span><span class="pl-smi">DMPs</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-k">-</span><span class="pl-smi">Row.names</span>)
<span class="pl-c"><span class="pl-c">#</span>create an beta table</span>
<span class="pl-smi">beta.table</span><span class="pl-k">&lt;-</span>as.data.frame(t(<span class="pl-smi">beta</span>))
<span class="pl-smi">beta.table</span><span class="pl-k">$</span><span class="pl-smi">Col_ID</span><span class="pl-k">&lt;-</span>rownames(<span class="pl-smi">beta.table</span>)
<span class="pl-smi">beta_named</span><span class="pl-k">&lt;-</span>merge(<span class="pl-smi">beta.table</span>, <span class="pl-smi">targets</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Col_ID<span class="pl-pds">"</span></span>)
rownames(<span class="pl-smi">beta_named</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">beta_named</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>
<span class="pl-smi">beta_named</span><span class="pl-k">&lt;-</span><span class="pl-smi">beta_named</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-k">-</span>names(<span class="pl-smi">targets</span>)) %<span class="pl-k">&gt;</span>% t()
write.csv(<span class="pl-smi">beta_named</span>, paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_beta.table.csv<span class="pl-pds">"</span></span>))
<span class="pl-c"><span class="pl-c">#</span>Add beta values to statistics</span>
<span class="pl-smi">Results_dmp</span><span class="pl-k">&lt;-</span>merge(<span class="pl-smi">DMPs</span>, <span class="pl-smi">beta_named</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>)
<span class="pl-c"><span class="pl-c">#</span>Calculate Average CpG Methylation by Sample_Group</span>
library(<span class="pl-smi">dplyr</span>)
library(<span class="pl-smi">matrixStats</span>)
<span class="pl-smi">Results</span><span class="pl-k">&lt;-</span><span class="pl-smi">Results_dmp</span> %<span class="pl-k">&gt;</span>% replace(is.na(<span class="pl-smi">.</span>), <span class="pl-c1">0</span>) %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>mutate(
  <span class="pl-v">DCM_SD</span> <span class="pl-k">=</span> rowSds(as.matrix(<span class="pl-smi">Results_dmp</span>[,<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>[<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>DCM<span class="pl-pds">"</span></span>]])),
  <span class="pl-v">DCM_Mean</span> <span class="pl-k">=</span> rowMeans(as.matrix(<span class="pl-smi">Results_dmp</span>[,<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>[<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>DCM<span class="pl-pds">"</span></span>]])),
  <span class="pl-v">CON_SD</span> <span class="pl-k">=</span> rowSds(as.matrix(<span class="pl-smi">Results_dmp</span>[,<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>[<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>CON<span class="pl-pds">"</span></span>]])),
  <span class="pl-v">CON_Mean</span> <span class="pl-k">=</span> rowMeans(as.matrix(<span class="pl-smi">Results_dmp</span>[,<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>[<span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Group</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>CON<span class="pl-pds">"</span></span>]])),
  <span class="pl-v">Methylation.Diff</span><span class="pl-k">=</span>(<span class="pl-smi">DCM_Mean</span><span class="pl-k">-</span><span class="pl-smi">CON_Mean</span>)<span class="pl-k">*</span><span class="pl-c1">100</span>)
rownames(<span class="pl-smi">Results</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">Results</span><span class="pl-k">$</span><span class="pl-smi">Row.names</span>
<span class="pl-smi">Results_dmp_p05</span><span class="pl-k">&lt;-</span>filter(<span class="pl-smi">Results</span>, <span class="pl-smi">pval</span><span class="pl-k">&lt;</span><span class="pl-c1">0.05</span>)
<span class="pl-smi">Results_dmp_q05</span><span class="pl-k">&lt;-</span>filter(<span class="pl-smi">Results</span>, <span class="pl-smi">qval</span><span class="pl-k">&lt;</span><span class="pl-c1">0.05</span>)
<span class="pl-c"><span class="pl-c">#</span>########################################</span>
<span class="pl-c"><span class="pl-c">#</span>Identify Promoter-associated CpG Islands</span>
library(<span class="pl-smi">tidyr</span>)
<span class="pl-smi">PromCGI</span><span class="pl-k">&lt;-</span><span class="pl-e">dplyr</span><span class="pl-k">::</span>filter(<span class="pl-smi">Results_dmp_p05</span>, grepl(<span class="pl-s"><span class="pl-pds">"</span>Island<span class="pl-pds">"</span></span>, <span class="pl-smi">Relation_to_Island</span>), grepl(<span class="pl-s"><span class="pl-pds">"</span>TSS<span class="pl-pds">"</span></span>, <span class="pl-smi">UCSC_RefGene_Group</span>))
<span class="pl-c"><span class="pl-c">#</span>Separate Gene Names into unique rows</span>
<span class="pl-smi">PromCGI_sep</span><span class="pl-k">&lt;-</span><span class="pl-smi">PromCGI</span> %<span class="pl-k">&gt;</span>% mutate(<span class="pl-v">UCSC_RefGene_Name</span> <span class="pl-k">=</span> strsplit(as.character(<span class="pl-smi">UCSC_RefGene_Name</span>), <span class="pl-s"><span class="pl-pds">"</span>;<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>% unnest(<span class="pl-smi">UCSC_RefGene_Name</span>) %<span class="pl-k">&gt;</span>% distinct()
<span class="pl-c"><span class="pl-c">#</span>Save a copy of the countData</span>
library(<span class="pl-smi">openxlsx</span>)
<span class="pl-smi">wb_countData</span><span class="pl-k">&lt;-</span>createWorkbook()
addWorksheet(<span class="pl-smi">wb_countData</span>, <span class="pl-s"><span class="pl-pds">"</span>Unfiltered<span class="pl-pds">"</span></span>)
  writeData(<span class="pl-smi">wb_countData</span>, <span class="pl-s"><span class="pl-pds">"</span>Unfiltered<span class="pl-pds">"</span></span>, <span class="pl-smi">Results</span>, <span class="pl-v">startCol</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>)
addWorksheet(<span class="pl-smi">wb_countData</span>, <span class="pl-s"><span class="pl-pds">"</span>P_0.05<span class="pl-pds">"</span></span>)
  writeData(<span class="pl-smi">wb_countData</span>, <span class="pl-s"><span class="pl-pds">"</span>P_0.05<span class="pl-pds">"</span></span>, <span class="pl-smi">Results_dmp_p05</span>, <span class="pl-v">startCol</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>)
addWorksheet(<span class="pl-smi">wb_countData</span>, <span class="pl-s"><span class="pl-pds">"</span>Q_0.05<span class="pl-pds">"</span></span>)
  writeData(<span class="pl-smi">wb_countData</span>, <span class="pl-s"><span class="pl-pds">"</span>Q_0.05<span class="pl-pds">"</span></span>, <span class="pl-smi">Results_dmp_q05</span>, <span class="pl-v">startCol</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>)
addWorksheet(<span class="pl-smi">wb_countData</span>, <span class="pl-s"><span class="pl-pds">"</span>Promoter.CGI<span class="pl-pds">"</span></span>)
  writeData(<span class="pl-smi">wb_countData</span>, <span class="pl-s"><span class="pl-pds">"</span>Promoter.CGI<span class="pl-pds">"</span></span>, <span class="pl-smi">PromCGI_sep</span>, <span class="pl-v">startCol</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>)
saveWorkbook(<span class="pl-smi">wb_countData</span>, <span class="pl-v">file</span> <span class="pl-k">=</span> paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_DMPs.xlsx<span class="pl-pds">"</span></span>), <span class="pl-v">overwrite</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Distribution of Methylation by Genomic and CpG Annotation
                     </h2>
                     <a aria-label="Permalink: Distribution of Methylation by Genomic and CpG Annotation" class="anchor" href="#distribution-of-methylation-by-genomic-and-cpg-annotation" id="user-content-distribution-of-methylation-by-genomic-and-cpg-annotation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The following figure illustrates the enrichment of differential
methylation with respect to CpG-based and genomic annotations.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(plotly)
library(dplyr)
library(tidyr)
library(stringr)
library(reshape2)
library(readxl)
library(kableExtra)
library(ComplexHeatmap)
library(openxlsx)

##Create a regional annotation matrix of the m450k array
Region_epic&lt;-Results_dmp %&gt;% 
  dplyr::select(Row.names, UCSC_RefGene_Group, Relation_to_Island)
Stage1&lt;-Region_epic %&gt;% 
  mutate(UCSC_RefGene_Group = strsplit(as.character(UCSC_RefGene_Group), &quot;;&quot;)) %&gt;% 
  unnest(UCSC_RefGene_Group) 
Stage2&lt;-Stage1 %&gt;% 
  mutate(Relation_to_Island = strsplit(as.character(Relation_to_Island), &quot;;&quot;)) %&gt;%  
  unnest(Relation_to_Island)
Stage3&lt;-distinct(Stage2)
Regional.Groups&lt;-dplyr::group_by_(Stage3, &quot;UCSC_RefGene_Group&quot;, &quot;Relation_to_Island&quot;) %&gt;% 
  tally()
Region.matrix&lt;-Regional.Groups %&gt;% 
  spread(Relation_to_Island, n)
Region.matrix&lt;-Region.matrix %&gt;% 
  dplyr::select(UCSC_RefGene_Group, OpenSea, N_Shelf, N_Shore, Island, S_Shore, S_Shelf)
rownames(Region.matrix)&lt;-Region.matrix$UCSC_RefGene_Group
Region.matrix&lt;-Region.matrix[c(&quot;TSS1500&quot;, &quot;TSS200&quot;, &quot;5'UTR&quot;, &quot;1stExon&quot;, &quot;Body&quot;, &quot;3'UTR&quot;),]
rownames(Region.matrix)&lt;-Region.matrix$UCSC_RefGene_Group
Contour_3D_epic&lt;-Region.matrix[,-1]
rownames(Contour_3D_epic)&lt;-Region.matrix$UCSC_RefGene_Group
write.xlsx(Contour_3D_epic, &quot;../2_Output/Contour.3D_EPIC.xlsx&quot;, overwrite = TRUE)

##Create a regional annotation matrix of differentially-methylated positions
Region_p05&lt;-Results_dmp_p05 %&gt;% dplyr::select(Row.names, UCSC_RefGene_Group, Relation_to_Island)
Stage1&lt;-Region_p05 %&gt;% mutate(UCSC_RefGene_Group = strsplit(as.character(UCSC_RefGene_Group), &quot;;&quot;)) %&gt;% unnest(UCSC_RefGene_Group) 
Stage2&lt;-Stage1 %&gt;% mutate(Relation_to_Island = strsplit(as.character(Relation_to_Island), &quot;;&quot;)) %&gt;%  unnest(Relation_to_Island)
Stage3&lt;-distinct(Stage2)
Regional.Groups&lt;-dplyr::group_by_(Stage3, &quot;UCSC_RefGene_Group&quot;, &quot;Relation_to_Island&quot;) %&gt;% tally()
Region.matrix&lt;-Regional.Groups %&gt;% spread(Relation_to_Island, n)
Region.matrix&lt;-Region.matrix %&gt;% dplyr::select(UCSC_RefGene_Group, OpenSea, N_Shelf, N_Shore, Island, S_Shore, S_Shelf)
rownames(Region.matrix)&lt;-Region.matrix$UCSC_RefGene_Group
Region.matrix&lt;-Region.matrix[c(&quot;TSS1500&quot;, &quot;TSS200&quot;, &quot;5'UTR&quot;, &quot;1stExon&quot;, &quot;Body&quot;, &quot;3'UTR&quot;),]
rownames(Region.matrix)&lt;-Region.matrix$UCSC_RefGene_Group
Contour_3D&lt;-Region.matrix[,-1]
rownames(Contour_3D)&lt;-Region.matrix$UCSC_RefGene_Group
write.csv(Contour_3D, &quot;../2_Output/Contour.3D_DMPs.p05.csv&quot;)
### Identify DMP Enrichment (IMPORTANT)
Enrichment_Region&lt;-Contour_3D/Contour_3D_epic
Promoter_enr&lt;-Enrichment_Region[rownames(Enrichment_Region)==&quot;TSS200&quot;,] + Enrichment_Region[rownames(Enrichment_Region)==&quot;TSS1500&quot;,]
rownames(Promoter_enr)&lt;-&quot;Promoter&quot;
Enrichment_Region&lt;-Enrichment_Region %&gt;% 
  filter(!grepl(&quot;TSS&quot;, rownames(Enrichment_Region))) %&gt;%
  rbind(Promoter_enr, .) %&gt;%
  data.matrix()
paletteLength&lt;-100
myColor &lt;- colorRampPalette(c(&quot;dodgerblue4&quot;, &quot;white&quot;, &quot;brown4&quot;))(paletteLength)
pheatmap(Enrichment_Region, color = myColor, cluster_rows = FALSE, cluster_cols = FALSE, filename = paste0(&quot;../2_Output/&quot;, COMPARISON, &quot;/&quot;, COMPARISON, &quot;_DMP.Distribution.pdf&quot;))" dir="auto">
                     <pre>library(<span class="pl-smi">plotly</span>)
library(<span class="pl-smi">dplyr</span>)
library(<span class="pl-smi">tidyr</span>)
library(<span class="pl-smi">stringr</span>)
library(<span class="pl-smi">reshape2</span>)
library(<span class="pl-smi">readxl</span>)
library(<span class="pl-smi">kableExtra</span>)
library(<span class="pl-smi">ComplexHeatmap</span>)
library(<span class="pl-smi">openxlsx</span>)

<span class="pl-c"><span class="pl-c">#</span>#Create a regional annotation matrix of the m450k array</span>
<span class="pl-smi">Region_epic</span><span class="pl-k">&lt;-</span><span class="pl-smi">Results_dmp</span> %<span class="pl-k">&gt;</span>% 
  <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">Row.names</span>, <span class="pl-smi">UCSC_RefGene_Group</span>, <span class="pl-smi">Relation_to_Island</span>)
<span class="pl-smi">Stage1</span><span class="pl-k">&lt;-</span><span class="pl-smi">Region_epic</span> %<span class="pl-k">&gt;</span>% 
  mutate(<span class="pl-v">UCSC_RefGene_Group</span> <span class="pl-k">=</span> strsplit(as.character(<span class="pl-smi">UCSC_RefGene_Group</span>), <span class="pl-s"><span class="pl-pds">"</span>;<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>% 
  unnest(<span class="pl-smi">UCSC_RefGene_Group</span>) 
<span class="pl-smi">Stage2</span><span class="pl-k">&lt;-</span><span class="pl-smi">Stage1</span> %<span class="pl-k">&gt;</span>% 
  mutate(<span class="pl-v">Relation_to_Island</span> <span class="pl-k">=</span> strsplit(as.character(<span class="pl-smi">Relation_to_Island</span>), <span class="pl-s"><span class="pl-pds">"</span>;<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>%  
  unnest(<span class="pl-smi">Relation_to_Island</span>)
<span class="pl-smi">Stage3</span><span class="pl-k">&lt;-</span>distinct(<span class="pl-smi">Stage2</span>)
<span class="pl-smi">Regional.Groups</span><span class="pl-k">&lt;-</span><span class="pl-e">dplyr</span><span class="pl-k">::</span>group_by_(<span class="pl-smi">Stage3</span>, <span class="pl-s"><span class="pl-pds">"</span>UCSC_RefGene_Group<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Relation_to_Island<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  tally()
<span class="pl-smi">Region.matrix</span><span class="pl-k">&lt;-</span><span class="pl-smi">Regional.Groups</span> %<span class="pl-k">&gt;</span>% 
  spread(<span class="pl-smi">Relation_to_Island</span>, <span class="pl-smi">n</span>)
<span class="pl-smi">Region.matrix</span><span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span> %<span class="pl-k">&gt;</span>% 
  <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">UCSC_RefGene_Group</span>, <span class="pl-smi">OpenSea</span>, <span class="pl-smi">N_Shelf</span>, <span class="pl-smi">N_Shore</span>, <span class="pl-smi">Island</span>, <span class="pl-smi">S_Shore</span>, <span class="pl-smi">S_Shelf</span>)
rownames(<span class="pl-smi">Region.matrix</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span><span class="pl-k">$</span><span class="pl-smi">UCSC_RefGene_Group</span>
<span class="pl-smi">Region.matrix</span><span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span>[c(<span class="pl-s"><span class="pl-pds">"</span>TSS1500<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>TSS200<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>5'UTR<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>1stExon<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Body<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>3'UTR<span class="pl-pds">"</span></span>),]
rownames(<span class="pl-smi">Region.matrix</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span><span class="pl-k">$</span><span class="pl-smi">UCSC_RefGene_Group</span>
<span class="pl-smi">Contour_3D_epic</span><span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span>[,<span class="pl-k">-</span><span class="pl-c1">1</span>]
rownames(<span class="pl-smi">Contour_3D_epic</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span><span class="pl-k">$</span><span class="pl-smi">UCSC_RefGene_Group</span>
write.xlsx(<span class="pl-smi">Contour_3D_epic</span>, <span class="pl-s"><span class="pl-pds">"</span>../2_Output/Contour.3D_EPIC.xlsx<span class="pl-pds">"</span></span>, <span class="pl-v">overwrite</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)

<span class="pl-c"><span class="pl-c">#</span>#Create a regional annotation matrix of differentially-methylated positions</span>
<span class="pl-smi">Region_p05</span><span class="pl-k">&lt;-</span><span class="pl-smi">Results_dmp_p05</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">Row.names</span>, <span class="pl-smi">UCSC_RefGene_Group</span>, <span class="pl-smi">Relation_to_Island</span>)
<span class="pl-smi">Stage1</span><span class="pl-k">&lt;-</span><span class="pl-smi">Region_p05</span> %<span class="pl-k">&gt;</span>% mutate(<span class="pl-v">UCSC_RefGene_Group</span> <span class="pl-k">=</span> strsplit(as.character(<span class="pl-smi">UCSC_RefGene_Group</span>), <span class="pl-s"><span class="pl-pds">"</span>;<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>% unnest(<span class="pl-smi">UCSC_RefGene_Group</span>) 
<span class="pl-smi">Stage2</span><span class="pl-k">&lt;-</span><span class="pl-smi">Stage1</span> %<span class="pl-k">&gt;</span>% mutate(<span class="pl-v">Relation_to_Island</span> <span class="pl-k">=</span> strsplit(as.character(<span class="pl-smi">Relation_to_Island</span>), <span class="pl-s"><span class="pl-pds">"</span>;<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>%  unnest(<span class="pl-smi">Relation_to_Island</span>)
<span class="pl-smi">Stage3</span><span class="pl-k">&lt;-</span>distinct(<span class="pl-smi">Stage2</span>)
<span class="pl-smi">Regional.Groups</span><span class="pl-k">&lt;-</span><span class="pl-e">dplyr</span><span class="pl-k">::</span>group_by_(<span class="pl-smi">Stage3</span>, <span class="pl-s"><span class="pl-pds">"</span>UCSC_RefGene_Group<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Relation_to_Island<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% tally()
<span class="pl-smi">Region.matrix</span><span class="pl-k">&lt;-</span><span class="pl-smi">Regional.Groups</span> %<span class="pl-k">&gt;</span>% spread(<span class="pl-smi">Relation_to_Island</span>, <span class="pl-smi">n</span>)
<span class="pl-smi">Region.matrix</span><span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">UCSC_RefGene_Group</span>, <span class="pl-smi">OpenSea</span>, <span class="pl-smi">N_Shelf</span>, <span class="pl-smi">N_Shore</span>, <span class="pl-smi">Island</span>, <span class="pl-smi">S_Shore</span>, <span class="pl-smi">S_Shelf</span>)
rownames(<span class="pl-smi">Region.matrix</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span><span class="pl-k">$</span><span class="pl-smi">UCSC_RefGene_Group</span>
<span class="pl-smi">Region.matrix</span><span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span>[c(<span class="pl-s"><span class="pl-pds">"</span>TSS1500<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>TSS200<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>5'UTR<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>1stExon<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Body<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>3'UTR<span class="pl-pds">"</span></span>),]
rownames(<span class="pl-smi">Region.matrix</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span><span class="pl-k">$</span><span class="pl-smi">UCSC_RefGene_Group</span>
<span class="pl-smi">Contour_3D</span><span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span>[,<span class="pl-k">-</span><span class="pl-c1">1</span>]
rownames(<span class="pl-smi">Contour_3D</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">Region.matrix</span><span class="pl-k">$</span><span class="pl-smi">UCSC_RefGene_Group</span>
write.csv(<span class="pl-smi">Contour_3D</span>, <span class="pl-s"><span class="pl-pds">"</span>../2_Output/Contour.3D_DMPs.p05.csv<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span>## Identify DMP Enrichment (IMPORTANT)</span>
<span class="pl-smi">Enrichment_Region</span><span class="pl-k">&lt;-</span><span class="pl-smi">Contour_3D</span><span class="pl-k">/</span><span class="pl-smi">Contour_3D_epic</span>
<span class="pl-smi">Promoter_enr</span><span class="pl-k">&lt;-</span><span class="pl-smi">Enrichment_Region</span>[rownames(<span class="pl-smi">Enrichment_Region</span>)<span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>TSS200<span class="pl-pds">"</span></span>,] <span class="pl-k">+</span> <span class="pl-smi">Enrichment_Region</span>[rownames(<span class="pl-smi">Enrichment_Region</span>)<span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>TSS1500<span class="pl-pds">"</span></span>,]
rownames(<span class="pl-smi">Promoter_enr</span>)<span class="pl-k">&lt;-</span><span class="pl-s"><span class="pl-pds">"</span>Promoter<span class="pl-pds">"</span></span>
<span class="pl-smi">Enrichment_Region</span><span class="pl-k">&lt;-</span><span class="pl-smi">Enrichment_Region</span> %<span class="pl-k">&gt;</span>% 
  filter(<span class="pl-k">!</span>grepl(<span class="pl-s"><span class="pl-pds">"</span>TSS<span class="pl-pds">"</span></span>, rownames(<span class="pl-smi">Enrichment_Region</span>))) %<span class="pl-k">&gt;</span>%
  rbind(<span class="pl-smi">Promoter_enr</span>, <span class="pl-smi">.</span>) %<span class="pl-k">&gt;</span>%
  data.matrix()
<span class="pl-smi">paletteLength</span><span class="pl-k">&lt;-</span><span class="pl-c1">100</span>
<span class="pl-smi">myColor</span> <span class="pl-k">&lt;-</span> colorRampPalette(c(<span class="pl-s"><span class="pl-pds">"</span>dodgerblue4<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>brown4<span class="pl-pds">"</span></span>))(<span class="pl-smi">paletteLength</span>)
pheatmap(<span class="pl-smi">Enrichment_Region</span>, <span class="pl-v">color</span> <span class="pl-k">=</span> <span class="pl-smi">myColor</span>, <span class="pl-v">cluster_rows</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">cluster_cols</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">filename</span> <span class="pl-k">=</span> paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_DMP.Distribution.pdf<span class="pl-pds">"</span></span>))</pre>
                    </div>
                    <p dir="auto">
                     ![](README_files/figure-html/Methylation Distribution-1.png)
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='write.csv(Enrichment_Region, "DCM_vs_CON.csv")
##Make a Table of the CpG Methylation Distribution
Enrichment_Region %&gt;% kable( align="c", booktabs=T, 
                     caption="Methylation Distribution") %&gt;% 
  kable_styling(latex_options=c("striped", "condensed", "repeat_header"))' dir="auto">
                     <pre>write.csv(<span class="pl-smi">Enrichment_Region</span>, <span class="pl-s"><span class="pl-pds">"</span>DCM_vs_CON.csv<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span>#Make a Table of the CpG Methylation Distribution</span>
<span class="pl-smi">Enrichment_Region</span> %<span class="pl-k">&gt;</span>% kable( <span class="pl-v">align</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>c<span class="pl-pds">"</span></span>, <span class="pl-v">booktabs</span><span class="pl-k">=</span><span class="pl-c1">T</span>, 
                     <span class="pl-v">caption</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Methylation Distribution<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  kable_styling(<span class="pl-v">latex_options</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>striped<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>condensed<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>repeat_header<span class="pl-pds">"</span></span>))</pre>
                    </div>
                    Methylation Distribution
                    <table>
                     <thead>
                      <tr>
                       <th>
                       </th>
                       <th>
                        OpenSea
                       </th>
                       <th>
                        N_Shelf
                       </th>
                       <th>
                        N_Shore
                       </th>
                       <th>
                        Island
                       </th>
                       <th>
                        S_Shore
                       </th>
                       <th>
                        S_Shelf
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        Promoter
                       </td>
                       <td>
                        0.1164621
                       </td>
                       <td>
                        0.1029990
                       </td>
                       <td>
                        0.1242210
                       </td>
                       <td>
                        0.1460420
                       </td>
                       <td>
                        0.1156378
                       </td>
                       <td>
                        0.1037933
                       </td>
                      </tr>
                      <tr>
                       <td>
                        5'UTR
                       </td>
                       <td>
                        0.0530359
                       </td>
                       <td>
                        0.0622389
                       </td>
                       <td>
                        0.0569429
                       </td>
                       <td>
                        0.0741839
                       </td>
                       <td>
                        0.0573959
                       </td>
                       <td>
                        0.0576832
                       </td>
                      </tr>
                      <tr>
                       <td>
                        1stExon
                       </td>
                       <td>
                        0.0588488
                       </td>
                       <td>
                        0.0489130
                       </td>
                       <td>
                        0.0623833
                       </td>
                       <td>
                        0.0746424
                       </td>
                       <td>
                        0.0625000
                       </td>
                       <td>
                        0.0370370
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Body
                       </td>
                       <td>
                        0.0523911
                       </td>
                       <td>
                        0.0493794
                       </td>
                       <td>
                        0.0510093
                       </td>
                       <td>
                        0.0633285
                       </td>
                       <td>
                        0.0507863
                       </td>
                       <td>
                        0.0473963
                       </td>
                      </tr>
                      <tr>
                       <td>
                        3'UTR
                       </td>
                       <td>
                        0.0485610
                       </td>
                       <td>
                        0.0415879
                       </td>
                       <td>
                        0.0554833
                       </td>
                       <td>
                        0.0638298
                       </td>
                       <td>
                        0.0563931
                       </td>
                       <td>
                        0.0479303
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="write.xlsx(Enrichment_Region, paste0(&quot;../2_Output/&quot;, COMPARISON, &quot;/&quot;, COMPARISON, &quot;_DMP.Enrichment_3D.xlsx&quot;), overwrite = T)
color &lt;- colorRampPalette(c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;))
t &lt;- list(
  family = &quot;times&quot;,
  size = 16,
  color = &quot;black&quot;)
x_axis&lt;-list(title = 'CpG Region', 
                     type=&quot;category&quot;, 
                     zeroline=TRUE, 
                     showline=TRUE, 
                     zerolinewidth = 4, 
            zerolinecolor=&quot;darkgrey&quot;, 
            linecolor=&quot;darkgrey&quot;, 
            linewidth=4, 
            titlefont=t, 
            tickfont=t)
y_axis&lt;-list(title = 'Gene Region', 
                     type=&quot;category&quot;, 
                     zeroline=TRUE, 
                     showline=TRUE, 
                     zerolinewidth = 4, 
            zerolinecolor=&quot;darkgrey&quot;, 
            linecolor=&quot;darkgrey&quot;, 
            linewidth=4, 
            titlefont=t, 
            tickfont=t)
z_axis&lt;-list(title = 'Number of DMPs', 
                     zerolinewidth = 4, 
                    zerolinecolor=&quot;darkgrey&quot;, 
                    linecolor=&quot;darkgrey&quot;, 
                    linewidth=4, 
                    titlefont=t, 
                    tickfont=t)
q&lt;-plot_ly(z=~Enrichment_Region, colors=color(10), 
    text=as.character(rownames(Enrichment_Region))) %&gt;% add_surface() %&gt;% 
    layout(scene = list(xaxis = x_axis, yaxis = y_axis, zaxis = z_axis))
q #must comment out for PDF generation via knitr (Pandoc)." dir="auto">
                     <pre>write.xlsx(<span class="pl-smi">Enrichment_Region</span>, paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_DMP.Enrichment_3D.xlsx<span class="pl-pds">"</span></span>), <span class="pl-v">overwrite</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>)
<span class="pl-smi">color</span> <span class="pl-k">&lt;-</span> colorRampPalette(c(<span class="pl-s"><span class="pl-pds">"</span>grey<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>orange<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>))
<span class="pl-smi">t</span> <span class="pl-k">&lt;-</span> <span class="pl-k">list</span>(
  <span class="pl-v">family</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>times<span class="pl-pds">"</span></span>,
  <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>,
  <span class="pl-v">color</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>)
<span class="pl-smi">x_axis</span><span class="pl-k">&lt;-</span><span class="pl-k">list</span>(<span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>CpG Region<span class="pl-pds">'</span></span>, 
                     <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>category<span class="pl-pds">"</span></span>, 
                     <span class="pl-v">zeroline</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, 
                     <span class="pl-v">showline</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, 
                     <span class="pl-v">zerolinewidth</span> <span class="pl-k">=</span> <span class="pl-c1">4</span>, 
            <span class="pl-v">zerolinecolor</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>darkgrey<span class="pl-pds">"</span></span>, 
            <span class="pl-v">linecolor</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>darkgrey<span class="pl-pds">"</span></span>, 
            <span class="pl-v">linewidth</span><span class="pl-k">=</span><span class="pl-c1">4</span>, 
            <span class="pl-v">titlefont</span><span class="pl-k">=</span><span class="pl-smi">t</span>, 
            <span class="pl-v">tickfont</span><span class="pl-k">=</span><span class="pl-smi">t</span>)
<span class="pl-smi">y_axis</span><span class="pl-k">&lt;-</span><span class="pl-k">list</span>(<span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>Gene Region<span class="pl-pds">'</span></span>, 
                     <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>category<span class="pl-pds">"</span></span>, 
                     <span class="pl-v">zeroline</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, 
                     <span class="pl-v">showline</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, 
                     <span class="pl-v">zerolinewidth</span> <span class="pl-k">=</span> <span class="pl-c1">4</span>, 
            <span class="pl-v">zerolinecolor</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>darkgrey<span class="pl-pds">"</span></span>, 
            <span class="pl-v">linecolor</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>darkgrey<span class="pl-pds">"</span></span>, 
            <span class="pl-v">linewidth</span><span class="pl-k">=</span><span class="pl-c1">4</span>, 
            <span class="pl-v">titlefont</span><span class="pl-k">=</span><span class="pl-smi">t</span>, 
            <span class="pl-v">tickfont</span><span class="pl-k">=</span><span class="pl-smi">t</span>)
<span class="pl-smi">z_axis</span><span class="pl-k">&lt;-</span><span class="pl-k">list</span>(<span class="pl-v">title</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>Number of DMPs<span class="pl-pds">'</span></span>, 
                     <span class="pl-v">zerolinewidth</span> <span class="pl-k">=</span> <span class="pl-c1">4</span>, 
                    <span class="pl-v">zerolinecolor</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>darkgrey<span class="pl-pds">"</span></span>, 
                    <span class="pl-v">linecolor</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>darkgrey<span class="pl-pds">"</span></span>, 
                    <span class="pl-v">linewidth</span><span class="pl-k">=</span><span class="pl-c1">4</span>, 
                    <span class="pl-v">titlefont</span><span class="pl-k">=</span><span class="pl-smi">t</span>, 
                    <span class="pl-v">tickfont</span><span class="pl-k">=</span><span class="pl-smi">t</span>)
<span class="pl-smi">q</span><span class="pl-k">&lt;-</span>plot_ly(<span class="pl-v">z</span><span class="pl-k">=</span><span class="pl-k">~</span><span class="pl-smi">Enrichment_Region</span>, <span class="pl-v">colors</span><span class="pl-k">=</span>color(<span class="pl-c1">10</span>), 
    <span class="pl-v">text</span><span class="pl-k">=</span>as.character(rownames(<span class="pl-smi">Enrichment_Region</span>))) %<span class="pl-k">&gt;</span>% add_surface() %<span class="pl-k">&gt;</span>% 
    layout(<span class="pl-v">scene</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-v">xaxis</span> <span class="pl-k">=</span> <span class="pl-smi">x_axis</span>, <span class="pl-v">yaxis</span> <span class="pl-k">=</span> <span class="pl-smi">y_axis</span>, <span class="pl-v">zaxis</span> <span class="pl-k">=</span> <span class="pl-smi">z_axis</span>))
<span class="pl-smi">q</span> <span class="pl-c"><span class="pl-c">#</span>must comment out for PDF generation via knitr (Pandoc).</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="&lt;div id=&quot;htmlwidget-abe0d5b12822ca5cb440&quot; style=&quot;width:672px;height:480px;&quot; class=&quot;plotly html-widget&quot;&gt;&lt;/div&gt;
&lt;script type=&quot;application/json&quot; data-for=&quot;htmlwidget-abe0d5b12822ca5cb440&quot;&gt;{&quot;x&quot;:{&quot;visdat&quot;:{&quot;2ee73db9e5dd&quot;:[&quot;function () &quot;,&quot;plotlyVisDat&quot;]},&quot;cur_data&quot;:&quot;2ee73db9e5dd&quot;,&quot;attrs&quot;:{&quot;2ee73db9e5dd&quot;:{&quot;z&quot;:{},&quot;text&quot;:[&quot;Promoter&quot;,&quot;5'UTR&quot;,&quot;1stExon&quot;,&quot;Body&quot;,&quot;3'UTR&quot;],&quot;colors&quot;:[&quot;#BEBEBE&quot;,&quot;#CCB893&quot;,&quot;#DAB269&quot;,&quot;#E9AD3F&quot;,&quot;#F7A715&quot;,&quot;#FF9200&quot;,&quot;#FF6E00&quot;,&quot;#FF4900&quot;,&quot;#FF2400&quot;,&quot;#FF0000&quot;],&quot;alpha_stroke&quot;:1,&quot;sizes&quot;:[10,100],&quot;spans&quot;:[1,20],&quot;type&quot;:&quot;surface&quot;,&quot;inherit&quot;:true}},&quot;layout&quot;:{&quot;margin&quot;:{&quot;b&quot;:40,&quot;l&quot;:60,&quot;t&quot;:25,&quot;r&quot;:10},&quot;scene&quot;:{&quot;xaxis&quot;:{&quot;title&quot;:&quot;CpG Region&quot;,&quot;type&quot;:&quot;category&quot;,&quot;zeroline&quot;:true,&quot;showline&quot;:true,&quot;zerolinewidth&quot;:4,&quot;zerolinecolor&quot;:&quot;darkgrey&quot;,&quot;linecolor&quot;:&quot;darkgrey&quot;,&quot;linewidth&quot;:4,&quot;titlefont&quot;:{&quot;family&quot;:&quot;times&quot;,&quot;size&quot;:16,&quot;color&quot;:&quot;black&quot;},&quot;tickfont&quot;:{&quot;family&quot;:&quot;times&quot;,&quot;size&quot;:16,&quot;color&quot;:&quot;black&quot;}},&quot;yaxis&quot;:{&quot;title&quot;:&quot;Gene Region&quot;,&quot;type&quot;:&quot;category&quot;,&quot;zeroline&quot;:true,&quot;showline&quot;:true,&quot;zerolinewidth&quot;:4,&quot;zerolinecolor&quot;:&quot;darkgrey&quot;,&quot;linecolor&quot;:&quot;darkgrey&quot;,&quot;linewidth&quot;:4,&quot;titlefont&quot;:{&quot;family&quot;:&quot;times&quot;,&quot;size&quot;:16,&quot;color&quot;:&quot;black&quot;},&quot;tickfont&quot;:{&quot;family&quot;:&quot;times&quot;,&quot;size&quot;:16,&quot;color&quot;:&quot;black&quot;}},&quot;zaxis&quot;:{&quot;title&quot;:&quot;Number of DMPs&quot;,&quot;zerolinewidth&quot;:4,&quot;zerolinecolor&quot;:&quot;darkgrey&quot;,&quot;linecolor&quot;:&quot;darkgrey&quot;,&quot;linewidth&quot;:4,&quot;titlefont&quot;:{&quot;family&quot;:&quot;times&quot;,&quot;size&quot;:16,&quot;color&quot;:&quot;black&quot;},&quot;tickfont&quot;:{&quot;family&quot;:&quot;times&quot;,&quot;size&quot;:16,&quot;color&quot;:&quot;black&quot;}}},&quot;hovermode&quot;:&quot;closest&quot;,&quot;showlegend&quot;:false,&quot;legend&quot;:{&quot;yanchor&quot;:&quot;top&quot;,&quot;y&quot;:0.5}},&quot;source&quot;:&quot;A&quot;,&quot;config&quot;:{&quot;modeBarButtonsToAdd&quot;:[&quot;hoverclosest&quot;,&quot;hovercompare&quot;],&quot;showSendToCloud&quot;:false},&quot;data&quot;:[{&quot;colorbar&quot;:{&quot;title&quot;:&quot;Enrichment_Region&quot;,&quot;ticklen&quot;:2,&quot;len&quot;:0.5,&quot;lenmode&quot;:&quot;fraction&quot;,&quot;y&quot;:1,&quot;yanchor&quot;:&quot;top&quot;},&quot;colorscale&quot;:[[&quot;0&quot;,&quot;rgba(190,190,190,1)&quot;],[&quot;0.0416666666666666&quot;,&quot;rgba(196,188,174,1)&quot;],[&quot;0.0833333333333333&quot;,&quot;rgba(201,185,158,1)&quot;],[&quot;0.125&quot;,&quot;rgba(206,183,142,1)&quot;],[&quot;0.166666666666667&quot;,&quot;rgba(212,181,126,1)&quot;],[&quot;0.208333333333333&quot;,&quot;rgba(217,179,110,1)&quot;],[&quot;0.25&quot;,&quot;rgba(222,177,95,1)&quot;],[&quot;0.291666666666667&quot;,&quot;rgba(228,175,80,1)&quot;],[&quot;0.333333333333333&quot;,&quot;rgba(233,173,63,1)&quot;],[&quot;0.375&quot;,&quot;rgba(238,171,50,1)&quot;],[&quot;0.416666666666667&quot;,&quot;rgba(244,169,35,1)&quot;],[&quot;0.458333333333333&quot;,&quot;rgba(248,164,19,1)&quot;],[&quot;0.5&quot;,&quot;rgba(251,157,11,1)&quot;],[&quot;0.541666666666667&quot;,&quot;rgba(254,149,3,1)&quot;],[&quot;0.583333333333333&quot;,&quot;rgba(255,137,0,1)&quot;],[&quot;0.625&quot;,&quot;rgba(255,124,0,1)&quot;],[&quot;0.666666666666667&quot;,&quot;rgba(255,110,0,1)&quot;],[&quot;0.708333333333333&quot;,&quot;rgba(255,97,0,1)&quot;],[&quot;0.75&quot;,&quot;rgba(255,83,0,1)&quot;],[&quot;0.791666666666667&quot;,&quot;rgba(255,69,0,1)&quot;],[&quot;0.833333333333333&quot;,&quot;rgba(255,57,0,1)&quot;],[&quot;0.875&quot;,&quot;rgba(255,42,0,1)&quot;],[&quot;0.916666666666667&quot;,&quot;rgba(255,30,0,1)&quot;],[&quot;0.958333333333333&quot;,&quot;rgba(255,19,0,1)&quot;],[&quot;1&quot;,&quot;rgba(255,0,0,1)&quot;]],&quot;showscale&quot;:true,&quot;z&quot;:[[0.1164621465399,0.102998995174501,0.124221049754897,0.146041950369168,0.115637795393847,0.103793295570619],[0.0530358898721731,0.0622389306599833,0.0569429198682766,0.0741838816628411,0.0573958674975402,0.0576832151300236],[0.0588487972508591,0.0489130434782609,0.0623832648487112,0.0746423541620339,0.0625,0.037037037037037],[0.0523911393278587,0.0493794088757925,0.0510093237983163,0.0633284583420387,0.0507863308501833,0.0473962571196094],[0.0485609908224182,0.0415879017013233,0.0554833116601647,0.0638297872340425,0.056393076493579,0.0479302832244009]],&quot;text&quot;:[&quot;Promoter&quot;,&quot;5'UTR&quot;,&quot;1stExon&quot;,&quot;Body&quot;,&quot;3'UTR&quot;],&quot;type&quot;:&quot;surface&quot;,&quot;frame&quot;:null}],&quot;highlight&quot;:{&quot;on&quot;:&quot;plotly_click&quot;,&quot;persistent&quot;:false,&quot;dynamic&quot;:false,&quot;selectize&quot;:false,&quot;opacityDim&quot;:0.2,&quot;selected&quot;:{&quot;opacity&quot;:1},&quot;debounce&quot;:0},&quot;shinyEvents&quot;:[&quot;plotly_hover&quot;,&quot;plotly_click&quot;,&quot;plotly_selected&quot;,&quot;plotly_relayout&quot;,&quot;plotly_brushed&quot;,&quot;plotly_brushing&quot;,&quot;plotly_clickannotation&quot;,&quot;plotly_doubleclick&quot;,&quot;plotly_deselect&quot;,&quot;plotly_afterplot&quot;,&quot;plotly_sunburstclick&quot;],&quot;base_url&quot;:&quot;https://plot.ly&quot;},&quot;evals&quot;:[],&quot;jsHooks&quot;:[]}&lt;/script&gt;">
                     <pre class="notranslate" lang="{=html}"><code>&lt;div id="htmlwidget-abe0d5b12822ca5cb440" style="width:672px;height:480px;" class="plotly html-widget"&gt;&lt;/div&gt;
&lt;script type="application/json" data-for="htmlwidget-abe0d5b12822ca5cb440"&gt;{"x":{"visdat":{"2ee73db9e5dd":["function () ","plotlyVisDat"]},"cur_data":"2ee73db9e5dd","attrs":{"2ee73db9e5dd":{"z":{},"text":["Promoter","5'UTR","1stExon","Body","3'UTR"],"colors":["#BEBEBE","#CCB893","#DAB269","#E9AD3F","#F7A715","#FF9200","#FF6E00","#FF4900","#FF2400","#FF0000"],"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"surface","inherit":true}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"scene":{"xaxis":{"title":"CpG Region","type":"category","zeroline":true,"showline":true,"zerolinewidth":4,"zerolinecolor":"darkgrey","linecolor":"darkgrey","linewidth":4,"titlefont":{"family":"times","size":16,"color":"black"},"tickfont":{"family":"times","size":16,"color":"black"}},"yaxis":{"title":"Gene Region","type":"category","zeroline":true,"showline":true,"zerolinewidth":4,"zerolinecolor":"darkgrey","linecolor":"darkgrey","linewidth":4,"titlefont":{"family":"times","size":16,"color":"black"},"tickfont":{"family":"times","size":16,"color":"black"}},"zaxis":{"title":"Number of DMPs","zerolinewidth":4,"zerolinecolor":"darkgrey","linecolor":"darkgrey","linewidth":4,"titlefont":{"family":"times","size":16,"color":"black"},"tickfont":{"family":"times","size":16,"color":"black"}}},"hovermode":"closest","showlegend":false,"legend":{"yanchor":"top","y":0.5}},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"colorbar":{"title":"Enrichment_Region","ticklen":2,"len":0.5,"lenmode":"fraction","y":1,"yanchor":"top"},"colorscale":[["0","rgba(190,190,190,1)"],["0.0416666666666666","rgba(196,188,174,1)"],["0.0833333333333333","rgba(201,185,158,1)"],["0.125","rgba(206,183,142,1)"],["0.166666666666667","rgba(212,181,126,1)"],["0.208333333333333","rgba(217,179,110,1)"],["0.25","rgba(222,177,95,1)"],["0.291666666666667","rgba(228,175,80,1)"],["0.333333333333333","rgba(233,173,63,1)"],["0.375","rgba(238,171,50,1)"],["0.416666666666667","rgba(244,169,35,1)"],["0.458333333333333","rgba(248,164,19,1)"],["0.5","rgba(251,157,11,1)"],["0.541666666666667","rgba(254,149,3,1)"],["0.583333333333333","rgba(255,137,0,1)"],["0.625","rgba(255,124,0,1)"],["0.666666666666667","rgba(255,110,0,1)"],["0.708333333333333","rgba(255,97,0,1)"],["0.75","rgba(255,83,0,1)"],["0.791666666666667","rgba(255,69,0,1)"],["0.833333333333333","rgba(255,57,0,1)"],["0.875","rgba(255,42,0,1)"],["0.916666666666667","rgba(255,30,0,1)"],["0.958333333333333","rgba(255,19,0,1)"],["1","rgba(255,0,0,1)"]],"showscale":true,"z":[[0.1164621465399,0.102998995174501,0.124221049754897,0.146041950369168,0.115637795393847,0.103793295570619],[0.0530358898721731,0.0622389306599833,0.0569429198682766,0.0741838816628411,0.0573958674975402,0.0576832151300236],[0.0588487972508591,0.0489130434782609,0.0623832648487112,0.0746423541620339,0.0625,0.037037037037037],[0.0523911393278587,0.0493794088757925,0.0510093237983163,0.0633284583420387,0.0507863308501833,0.0473962571196094],[0.0485609908224182,0.0415879017013233,0.0554833116601647,0.0638297872340425,0.056393076493579,0.0479302832244009]],"text":["Promoter","5'UTR","1stExon","Body","3'UTR"],"type":"surface","frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}&lt;/script&gt;
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Figure XX: Heatmap and Hierarchical Clustering of Differential Methylation (P &lt; 0.05)
                     </h2>
                     <a aria-label="Permalink: Figure XX: Heatmap and Hierarchical Clustering of Differential Methylation (P &lt; 0.05)" class="anchor" href="#figure-xx-heatmap-and-hierarchical-clustering-of-differential-methylation-p--005" id="user-content-figure-xx-heatmap-and-hierarchical-clustering-of-differential-methylation-p--005">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Due to the prominent signature of CpG Island-associated promoter
methylation in iPSCs exposed to serum for failing and non-failing
patients, we examined all DMPs present within this region via heatmap.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(ComplexHeatmap)
library(dplyr)
##Import Data Matrix
# betaHM&lt;-read.csv("../1_Input/EPIC.betaValues.csv", row.names = 1)
## Filters to Apply to DMP
pvalue_threshold=0.001
METHYLATION=0
DMP_location="Island"
Gene_region="TSS"
Samples&lt;-AnnoTargets$Sample_Name
##Filter Differential Methylation Data
DMP.p05&lt;-Results %&gt;% filter(pval&lt;pvalue_threshold)
DMP.p05&lt;-DMP.p05 %&gt;% dplyr::select(Row.names, 
                            Methylation.Diff, 
                            pval, 
                            qval, 
                            Relation_to_Island, 
                            UCSC_RefGene_Group, 
                            chr, 
                            pos, 
                            matches(Samples)) %&gt;%
                      filter(abs(Methylation.Diff)&gt;METHYLATION)
DMP.p05.Region&lt;-DMP.p05 %&gt;% 
  # filter(grepl(DMP_location, Relation_to_Island)) %&gt;%
  filter(grepl(Gene_region, UCSC_RefGene_Group)) %&gt;% 
  dplyr::select(matches(AnnoTargets$Sample_Name)) %&gt;%
  data.matrix()
#Import the Index File
LVAD_Counts_Data &lt;- targets
rownames(LVAD_Counts_Data)&lt;-targets$Sample_Name
Index&lt;-LVAD_Counts_Data %&gt;% dplyr::select(Sample_Group)
Index&lt;-as.data.frame(Index)
paletteLength &lt;- 100
ann_colors = list(Sample_Group = c(DCM="darkcyan", CON="darkgray"), Outcome = c(Bad = "firebrick2", Good = "darkgoldenrod2", CON = "black"))
myColor &lt;- colorRampPalette(c("dodgerblue4", "white", "goldenrod2"))(paletteLength)
pheatmap::pheatmap(DMP.p05.Region, scale="row", 
                      cluster_cols = TRUE, 
                      cluster_rows = TRUE,
                      #breaks = myBreaks,
                      cutree_cols = 2,
                      cutree_rows = 2,
                      angle_col = 45,
                      fontsize_col = 8,
                      color = myColor, 
                      show_rownames = FALSE, 
                      border_color = NA, 
                      annotation_col = Index,
                      annotation_colors=ann_colors,
                      filename = paste0("../2_Output/", COMPARISON, "/", COMPARISON, "_", DMP_location, ".", Gene_region,".heatmap.pdf"))

pheatmap::pheatmap(DMP.p05.Region, scale="row", 
                      cluster_cols = TRUE, 
                      cluster_rows = TRUE,
                      #breaks = myBreaks,
                      cutree_cols = 2,
                      cutree_rows = 2,
                      angle_col = 45,
                      fontsize_col = 8,
                      color = myColor, 
                      show_rownames = FALSE, 
                      border_color = NA, 
                      annotation_col = Index,
                      annotation_colors=ann_colors)' dir="auto">
                     <pre>library(<span class="pl-smi">ComplexHeatmap</span>)
library(<span class="pl-smi">dplyr</span>)
<span class="pl-c"><span class="pl-c">#</span>#Import Data Matrix</span>
<span class="pl-c"><span class="pl-c">#</span> betaHM&lt;-read.csv("../1_Input/EPIC.betaValues.csv", row.names = 1)</span>
<span class="pl-c"><span class="pl-c">#</span># Filters to Apply to DMP</span>
<span class="pl-v">pvalue_threshold</span><span class="pl-k">=</span><span class="pl-c1">0.001</span>
<span class="pl-v">METHYLATION</span><span class="pl-k">=</span><span class="pl-c1">0</span>
<span class="pl-v">DMP_location</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Island<span class="pl-pds">"</span></span>
<span class="pl-v">Gene_region</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>TSS<span class="pl-pds">"</span></span>
<span class="pl-smi">Samples</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>
<span class="pl-c"><span class="pl-c">#</span>#Filter Differential Methylation Data</span>
<span class="pl-smi">DMP.p05</span><span class="pl-k">&lt;-</span><span class="pl-smi">Results</span> %<span class="pl-k">&gt;</span>% filter(<span class="pl-smi">pval</span><span class="pl-k">&lt;</span><span class="pl-smi">pvalue_threshold</span>)
<span class="pl-smi">DMP.p05</span><span class="pl-k">&lt;-</span><span class="pl-smi">DMP.p05</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">Row.names</span>, 
                            <span class="pl-smi">Methylation.Diff</span>, 
                            <span class="pl-smi">pval</span>, 
                            <span class="pl-smi">qval</span>, 
                            <span class="pl-smi">Relation_to_Island</span>, 
                            <span class="pl-smi">UCSC_RefGene_Group</span>, 
                            <span class="pl-smi">chr</span>, 
                            <span class="pl-smi">pos</span>, 
                            matches(<span class="pl-smi">Samples</span>)) %<span class="pl-k">&gt;</span>%
                      filter(abs(<span class="pl-smi">Methylation.Diff</span>)<span class="pl-k">&gt;</span><span class="pl-smi">METHYLATION</span>)
<span class="pl-smi">DMP.p05.Region</span><span class="pl-k">&lt;-</span><span class="pl-smi">DMP.p05</span> %<span class="pl-k">&gt;</span>% 
  <span class="pl-c"><span class="pl-c">#</span> filter(grepl(DMP_location, Relation_to_Island)) %&gt;%</span>
  filter(grepl(<span class="pl-smi">Gene_region</span>, <span class="pl-smi">UCSC_RefGene_Group</span>)) %<span class="pl-k">&gt;</span>% 
  <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(matches(<span class="pl-smi">AnnoTargets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>)) %<span class="pl-k">&gt;</span>%
  data.matrix()
<span class="pl-c"><span class="pl-c">#</span>Import the Index File</span>
<span class="pl-smi">LVAD_Counts_Data</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">targets</span>
rownames(<span class="pl-smi">LVAD_Counts_Data</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">targets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>
<span class="pl-smi">Index</span><span class="pl-k">&lt;-</span><span class="pl-smi">LVAD_Counts_Data</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">Sample_Group</span>)
<span class="pl-smi">Index</span><span class="pl-k">&lt;-</span>as.data.frame(<span class="pl-smi">Index</span>)
<span class="pl-smi">paletteLength</span> <span class="pl-k">&lt;-</span> <span class="pl-c1">100</span>
<span class="pl-v">ann_colors</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-v">Sample_Group</span> <span class="pl-k">=</span> c(<span class="pl-v">DCM</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>darkcyan<span class="pl-pds">"</span></span>, <span class="pl-v">CON</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>darkgray<span class="pl-pds">"</span></span>), <span class="pl-v">Outcome</span> <span class="pl-k">=</span> c(<span class="pl-v">Bad</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>firebrick2<span class="pl-pds">"</span></span>, <span class="pl-v">Good</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>darkgoldenrod2<span class="pl-pds">"</span></span>, <span class="pl-v">CON</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>))
<span class="pl-smi">myColor</span> <span class="pl-k">&lt;-</span> colorRampPalette(c(<span class="pl-s"><span class="pl-pds">"</span>dodgerblue4<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>goldenrod2<span class="pl-pds">"</span></span>))(<span class="pl-smi">paletteLength</span>)
<span class="pl-e">pheatmap</span><span class="pl-k">::</span>pheatmap(<span class="pl-smi">DMP.p05.Region</span>, <span class="pl-v">scale</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>row<span class="pl-pds">"</span></span>, 
                      <span class="pl-v">cluster_cols</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, 
                      <span class="pl-v">cluster_rows</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                      <span class="pl-c"><span class="pl-c">#</span>breaks = myBreaks,</span>
                      <span class="pl-v">cutree_cols</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>,
                      <span class="pl-v">cutree_rows</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>,
                      <span class="pl-v">angle_col</span> <span class="pl-k">=</span> <span class="pl-c1">45</span>,
                      <span class="pl-v">fontsize_col</span> <span class="pl-k">=</span> <span class="pl-c1">8</span>,
                      <span class="pl-v">color</span> <span class="pl-k">=</span> <span class="pl-smi">myColor</span>, 
                      <span class="pl-v">show_rownames</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, 
                      <span class="pl-v">border_color</span> <span class="pl-k">=</span> <span class="pl-c1">NA</span>, 
                      <span class="pl-v">annotation_col</span> <span class="pl-k">=</span> <span class="pl-smi">Index</span>,
                      <span class="pl-v">annotation_colors</span><span class="pl-k">=</span><span class="pl-smi">ann_colors</span>,
                      <span class="pl-v">filename</span> <span class="pl-k">=</span> paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>, <span class="pl-smi">DMP_location</span>, <span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>, <span class="pl-smi">Gene_region</span>,<span class="pl-s"><span class="pl-pds">"</span>.heatmap.pdf<span class="pl-pds">"</span></span>))

<span class="pl-e">pheatmap</span><span class="pl-k">::</span>pheatmap(<span class="pl-smi">DMP.p05.Region</span>, <span class="pl-v">scale</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>row<span class="pl-pds">"</span></span>, 
                      <span class="pl-v">cluster_cols</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, 
                      <span class="pl-v">cluster_rows</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                      <span class="pl-c"><span class="pl-c">#</span>breaks = myBreaks,</span>
                      <span class="pl-v">cutree_cols</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>,
                      <span class="pl-v">cutree_rows</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>,
                      <span class="pl-v">angle_col</span> <span class="pl-k">=</span> <span class="pl-c1">45</span>,
                      <span class="pl-v">fontsize_col</span> <span class="pl-k">=</span> <span class="pl-c1">8</span>,
                      <span class="pl-v">color</span> <span class="pl-k">=</span> <span class="pl-smi">myColor</span>, 
                      <span class="pl-v">show_rownames</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, 
                      <span class="pl-v">border_color</span> <span class="pl-k">=</span> <span class="pl-c1">NA</span>, 
                      <span class="pl-v">annotation_col</span> <span class="pl-k">=</span> <span class="pl-smi">Index</span>,
                      <span class="pl-v">annotation_colors</span><span class="pl-k">=</span><span class="pl-smi">ann_colors</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Figure XX: Volcano Plot - DMPs
                     </h2>
                     <a aria-label="Permalink: Figure XX: Volcano Plot - DMPs" class="anchor" href="#figure-xx-volcano-plot---dmps" id="user-content-figure-xx-volcano-plot---dmps">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Load packages
library(dplyr)
library(ggplot2)
library(ggrepel)
library(openxlsx)
Results&lt;-read.xlsx(paste0("../2_Output/", COMPARISON, "/", COMPARISON, "_DMPs.xlsx"), sheet = "Unfiltered")
#Read data from the web
Volcano_data = mutate(Results, sig=ifelse(Results$pval&lt;0.05 &amp; abs(Results$Methylation.Diff)&gt;5, "P&lt;0.05 and |Methylation| &gt; 5%", "Not Sig"), minuslogpvalue = -log(pval), Methylation=Methylation.Diff)

#Split gene names for labelling
Volcano_data_split&lt;-Volcano_data %&gt;% mutate(UCSC_RefGene_Name = strsplit(as.character(UCSC_RefGene_Name), ";")) %&gt;% unnest(UCSC_RefGene_Name) %&gt;% distinct()

max(Volcano_data$minuslogpvalue, na.rm = TRUE)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Load packages</span>
library(<span class="pl-smi">dplyr</span>)
library(<span class="pl-smi">ggplot2</span>)
library(<span class="pl-smi">ggrepel</span>)
library(<span class="pl-smi">openxlsx</span>)
<span class="pl-smi">Results</span><span class="pl-k">&lt;-</span>read.xlsx(paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>_DMPs.xlsx<span class="pl-pds">"</span></span>), <span class="pl-v">sheet</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Unfiltered<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span>Read data from the web</span>
<span class="pl-v">Volcano_data</span> <span class="pl-k">=</span> mutate(<span class="pl-smi">Results</span>, <span class="pl-v">sig</span><span class="pl-k">=</span>ifelse(<span class="pl-smi">Results</span><span class="pl-k">$</span><span class="pl-smi">pval</span><span class="pl-k">&lt;</span><span class="pl-c1">0.05</span> <span class="pl-k">&amp;</span> abs(<span class="pl-smi">Results</span><span class="pl-k">$</span><span class="pl-smi">Methylation.Diff</span>)<span class="pl-k">&gt;</span><span class="pl-c1">5</span>, <span class="pl-s"><span class="pl-pds">"</span>P&lt;0.05 and |Methylation| &gt; 5%<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Not Sig<span class="pl-pds">"</span></span>), <span class="pl-v">minuslogpvalue</span> <span class="pl-k">=</span> <span class="pl-k">-</span>log(<span class="pl-smi">pval</span>), <span class="pl-v">Methylation</span><span class="pl-k">=</span><span class="pl-smi">Methylation.Diff</span>)

<span class="pl-c"><span class="pl-c">#</span>Split gene names for labelling</span>
<span class="pl-smi">Volcano_data_split</span><span class="pl-k">&lt;-</span><span class="pl-smi">Volcano_data</span> %<span class="pl-k">&gt;</span>% mutate(<span class="pl-v">UCSC_RefGene_Name</span> <span class="pl-k">=</span> strsplit(as.character(<span class="pl-smi">UCSC_RefGene_Name</span>), <span class="pl-s"><span class="pl-pds">"</span>;<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>% unnest(<span class="pl-smi">UCSC_RefGene_Name</span>) %<span class="pl-k">&gt;</span>% distinct()

max(<span class="pl-smi">Volcano_data</span><span class="pl-k">$</span><span class="pl-smi">minuslogpvalue</span>, <span class="pl-v">na.rm</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## [1] 15.86107">
                     <pre class="notranslate"><code>## [1] 15.86107
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Results&lt;-Results %&gt;% filter(grepl("TSS", UCSC_RefGene_Group))
#plot the ggplot
p = ggplot(Volcano_data_split, aes(Methylation, minuslogpvalue)) + theme(panel.background = element_rect("white", colour = "black", size=2), panel.grid.major = element_line(colour = "gray50", size=.75), panel.grid.minor = element_line(colour = "gray50", size=0.4)) + 
geom_point(aes(fill=sig, size = minuslogpvalue), colour="grey", shape=21, stroke = 0, alpha = 8/10) + labs(x=expression(Methylation_Change), y=expression(-Log[10](P-value))) + xlim(min(Volcano_data$Methylation, na.rm = TRUE),max(Volcano_data$Methylation, na.rm = TRUE))+ ylim(-0, max(Volcano_data$minuslogpvalue, na.rm = TRUE)) +   geom_hline(yintercept = 0, size = 1) + geom_vline(xintercept=0, size=1)+ 
  scale_fill_manual(values=c("grey", "darkgoldenrod2")) +
  scale_size_continuous(range = c(.25, 4))
pdf(file = paste0("../2_Output/", COMPARISON, "/", COMPARISON, "Volcano.Plot.pdf"), height = 6, width = 10)
p+
  geom_text_repel(data=top_n(filter(Volcano_data_split, pval&lt;0.05, Methylation &lt; 0, sig!="Not Sig"), 10, -Methylation), aes(label=UCSC_RefGene_Name)) +
  geom_text_repel(data=top_n(filter(Volcano_data_split, pval&lt;0.05, Methylation &gt; 0, sig!="Not Sig"), 10, Methylation), aes(label=UCSC_RefGene_Name))
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Results&lt;-Results %&gt;% filter(grepl("TSS", UCSC_RefGene_Group))</span>
<span class="pl-c"><span class="pl-c">#</span>plot the ggplot</span>
<span class="pl-v">p</span> <span class="pl-k">=</span> ggplot(<span class="pl-smi">Volcano_data_split</span>, aes(<span class="pl-smi">Methylation</span>, <span class="pl-smi">minuslogpvalue</span>)) <span class="pl-k">+</span> theme(<span class="pl-v">panel.background</span> <span class="pl-k">=</span> element_rect(<span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>, <span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-v">size</span><span class="pl-k">=</span><span class="pl-c1">2</span>), <span class="pl-v">panel.grid.major</span> <span class="pl-k">=</span> element_line(<span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gray50<span class="pl-pds">"</span></span>, <span class="pl-v">size</span><span class="pl-k">=</span><span class="pl-smi">.75</span>), <span class="pl-v">panel.grid.minor</span> <span class="pl-k">=</span> element_line(<span class="pl-v">colour</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gray50<span class="pl-pds">"</span></span>, <span class="pl-v">size</span><span class="pl-k">=</span><span class="pl-c1">0.4</span>)) <span class="pl-k">+</span> 
geom_point(aes(<span class="pl-v">fill</span><span class="pl-k">=</span><span class="pl-smi">sig</span>, <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-smi">minuslogpvalue</span>), <span class="pl-v">colour</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>grey<span class="pl-pds">"</span></span>, <span class="pl-v">shape</span><span class="pl-k">=</span><span class="pl-c1">21</span>, <span class="pl-v">stroke</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>, <span class="pl-v">alpha</span> <span class="pl-k">=</span> <span class="pl-c1">8</span><span class="pl-k">/</span><span class="pl-c1">10</span>) <span class="pl-k">+</span> labs(<span class="pl-v">x</span><span class="pl-k">=</span>expression(<span class="pl-smi">Methylation_Change</span>), <span class="pl-v">y</span><span class="pl-k">=</span>expression(<span class="pl-k">-</span><span class="pl-smi">Log</span>[<span class="pl-c1">10</span>](<span class="pl-smi">P</span><span class="pl-k">-</span><span class="pl-smi">value</span>))) <span class="pl-k">+</span> xlim(min(<span class="pl-smi">Volcano_data</span><span class="pl-k">$</span><span class="pl-smi">Methylation</span>, <span class="pl-v">na.rm</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>),max(<span class="pl-smi">Volcano_data</span><span class="pl-k">$</span><span class="pl-smi">Methylation</span>, <span class="pl-v">na.rm</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>))<span class="pl-k">+</span> ylim(<span class="pl-k">-</span><span class="pl-c1">0</span>, max(<span class="pl-smi">Volcano_data</span><span class="pl-k">$</span><span class="pl-smi">minuslogpvalue</span>, <span class="pl-v">na.rm</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)) <span class="pl-k">+</span>   geom_hline(<span class="pl-v">yintercept</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>, <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>) <span class="pl-k">+</span> geom_vline(<span class="pl-v">xintercept</span><span class="pl-k">=</span><span class="pl-c1">0</span>, <span class="pl-v">size</span><span class="pl-k">=</span><span class="pl-c1">1</span>)<span class="pl-k">+</span> 
  scale_fill_manual(<span class="pl-v">values</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>grey<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>darkgoldenrod2<span class="pl-pds">"</span></span>)) <span class="pl-k">+</span>
  scale_size_continuous(<span class="pl-v">range</span> <span class="pl-k">=</span> c(<span class="pl-smi">.25</span>, <span class="pl-c1">4</span>))
pdf(<span class="pl-v">file</span> <span class="pl-k">=</span> paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-smi">COMPARISON</span>, <span class="pl-s"><span class="pl-pds">"</span>Volcano.Plot.pdf<span class="pl-pds">"</span></span>), <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>)
<span class="pl-smi">p</span><span class="pl-k">+</span>
  geom_text_repel(<span class="pl-v">data</span><span class="pl-k">=</span>top_n(filter(<span class="pl-smi">Volcano_data_split</span>, <span class="pl-smi">pval</span><span class="pl-k">&lt;</span><span class="pl-c1">0.05</span>, <span class="pl-smi">Methylation</span> <span class="pl-k">&lt;</span> <span class="pl-c1">0</span>, <span class="pl-smi">sig</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">"</span>Not Sig<span class="pl-pds">"</span></span>), <span class="pl-c1">10</span>, <span class="pl-k">-</span><span class="pl-smi">Methylation</span>), aes(<span class="pl-v">label</span><span class="pl-k">=</span><span class="pl-smi">UCSC_RefGene_Name</span>)) <span class="pl-k">+</span>
  geom_text_repel(<span class="pl-v">data</span><span class="pl-k">=</span>top_n(filter(<span class="pl-smi">Volcano_data_split</span>, <span class="pl-smi">pval</span><span class="pl-k">&lt;</span><span class="pl-c1">0.05</span>, <span class="pl-smi">Methylation</span> <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>, <span class="pl-smi">sig</span><span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">"</span>Not Sig<span class="pl-pds">"</span></span>), <span class="pl-c1">10</span>, <span class="pl-smi">Methylation</span>), aes(<span class="pl-v">label</span><span class="pl-k">=</span><span class="pl-smi">UCSC_RefGene_Name</span>))
dev.off()</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## quartz_off_screen 
##                 2">
                     <pre class="notranslate"><code>## quartz_off_screen 
##                 2
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Combined Analysis - Biopsy and iPSC
                     </h1>
                     <a aria-label="Permalink: Combined Analysis - Biopsy and iPSC" class="anchor" href="#combined-analysis---biopsy-and-ipsc" id="user-content-combined-analysis---biopsy-and-ipsc">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(openxlsx)
library(dplyr)
library(tidyr)
library(ComplexHeatmap)
library(minfi)
NUMBER = 20
STAT = 0.001
#Index
AnnoTargets&lt;-read.metharray.sheet(base="../1_Input/IDAT", pattern=".csv")' dir="auto">
                     <pre>library(<span class="pl-smi">openxlsx</span>)
library(<span class="pl-smi">dplyr</span>)
library(<span class="pl-smi">tidyr</span>)
library(<span class="pl-smi">ComplexHeatmap</span>)
library(<span class="pl-smi">minfi</span>)
<span class="pl-v">NUMBER</span> <span class="pl-k">=</span> <span class="pl-c1">20</span>
<span class="pl-v">STAT</span> <span class="pl-k">=</span> <span class="pl-c1">0.001</span>
<span class="pl-c"><span class="pl-c">#</span>Index</span>
<span class="pl-smi">AnnoTargets</span><span class="pl-k">&lt;-</span>read.metharray.sheet(<span class="pl-v">base</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>../1_Input/IDAT<span class="pl-pds">"</span></span>, <span class="pl-v">pattern</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.csv<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## [1] "../1_Input/IDAT/SampleSheet.csv"'>
                     <pre class="notranslate"><code>## [1] "../1_Input/IDAT/SampleSheet.csv"
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='rownames(AnnoTargets)&lt;-AnnoTargets$Sample_Name
Index_combined&lt;-AnnoTargets %&gt;% dplyr::select(Sample_Group, Tissue, Sample_ID)
#iPSC Data
DMPs_iPSC&lt;-read.xlsx("../2_Output/iPSC_DCM_vs_CON/iPSC_DCM_vs_CON_DMPs.xlsx", sheet = "P_0.05", rowNames = T)
iPSC_stats&lt;-DMPs_iPSC %&gt;% dplyr::select(Methylation_iPSC = Methylation.Diff, pval_iPSC = pval, adj.P.Val_iPSC = qval, Name:Regulatory_Feature_Group)


#biopsies
DMPs_heart&lt;-read.xlsx("../2_Output/heart_DCM_vs_CON/heart_DCM_vs_CON_DMPs.xlsx", sheet = "P_0.05", rowNames = T)
heart_stats&lt;-DMPs_heart %&gt;% dplyr::select(Methylation_heart = Methylation.Diff, pval_heart = pval, adj.P.Val_heart = qval, Name:Regulatory_Feature_Group)
heart_index&lt;-AnnoTargets %&gt;% dplyr::filter(Tissue=="heart") %&gt;% dplyr::select(Tissue, Sample_Group, Sample_ID)

#Beta
beta_all&lt;-read.csv("../1_Input/beta_all.csv", row.names = 1)

#combine data
Combined_DMPs&lt;-inner_join(iPSC_stats, heart_stats)
Combined_DMPs_Separated&lt;-Combined_DMPs %&gt;% mutate(UCSC_RefGene_Name = strsplit(as.character(UCSC_RefGene_Name), ";")) %&gt;% unnest(UCSC_RefGene_Name) %&gt;% distinct() %&gt;% merge(., beta_all, by.x = "Name", by.y = 0)

# Select only co-methylated positions (iPSC and heart)
Combined_together&lt;-Combined_DMPs_Separated %&gt;% filter(Methylation_iPSC&gt;0 &amp; Methylation_heart&gt;0 | Methylation_iPSC&lt;0 &amp; Methylation_heart&lt;0) %&gt;% arrange(desc(Methylation_iPSC)) %&gt;% filter(pval_iPSC &lt; STAT) %&gt;% arrange(pval_iPSC)
#Find row number for top candidates
Candidates&lt;-which(Combined_together$UCSC_RefGene_Name == "ATG7")
# Combined_together&lt;-merge(Combined_together, beta_all, by.x = "Name", by.y = 0)
rownames(Combined_together)&lt;-make.unique(Combined_together$UCSC_RefGene_Name, sep = ".")
#Create heatmap matrices and annotation
heart_hm&lt;-Combined_together %&gt;% dplyr::select(contains("NF") | contains("biopsy")) %&gt;% data.matrix()
heart_index&lt;-AnnoTargets %&gt;% dplyr::filter(Tissue=="heart") %&gt;% dplyr::select(Tissue, Sample_Group, Sample_ID)
iPSC_hm&lt;-Combined_together %&gt;% dplyr::select(contains("cells") &amp; contains("co"),contains("cells") &amp; contains("go"), contains("cells") &amp; contains("bo")) %&gt;% data.matrix()
iPSC_index&lt;-AnnoTargets %&gt;% dplyr::filter(Tissue=="iPSC") %&gt;% dplyr::select(Tissue, Sample_Group, Sample_ID)
write.xlsx(Combined_together, "../2_Output/Overlapping.DMPs_iPSC.Biopsy.xlsx", overwrite = T)

# Venn Diagram
library(VennDiagram)
futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")' dir="auto">
                     <pre>rownames(<span class="pl-smi">AnnoTargets</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span><span class="pl-k">$</span><span class="pl-smi">Sample_Name</span>
<span class="pl-smi">Index_combined</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">Sample_Group</span>, <span class="pl-smi">Tissue</span>, <span class="pl-smi">Sample_ID</span>)
<span class="pl-c"><span class="pl-c">#</span>iPSC Data</span>
<span class="pl-smi">DMPs_iPSC</span><span class="pl-k">&lt;-</span>read.xlsx(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/iPSC_DCM_vs_CON/iPSC_DCM_vs_CON_DMPs.xlsx<span class="pl-pds">"</span></span>, <span class="pl-v">sheet</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>P_0.05<span class="pl-pds">"</span></span>, <span class="pl-v">rowNames</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>)
<span class="pl-smi">iPSC_stats</span><span class="pl-k">&lt;-</span><span class="pl-smi">DMPs_iPSC</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-v">Methylation_iPSC</span> <span class="pl-k">=</span> <span class="pl-smi">Methylation.Diff</span>, <span class="pl-v">pval_iPSC</span> <span class="pl-k">=</span> <span class="pl-smi">pval</span>, <span class="pl-v">adj.P.Val_iPSC</span> <span class="pl-k">=</span> <span class="pl-smi">qval</span>, <span class="pl-smi">Name</span><span class="pl-k">:</span><span class="pl-smi">Regulatory_Feature_Group</span>)


<span class="pl-c"><span class="pl-c">#</span>biopsies</span>
<span class="pl-smi">DMPs_heart</span><span class="pl-k">&lt;-</span>read.xlsx(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/heart_DCM_vs_CON/heart_DCM_vs_CON_DMPs.xlsx<span class="pl-pds">"</span></span>, <span class="pl-v">sheet</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>P_0.05<span class="pl-pds">"</span></span>, <span class="pl-v">rowNames</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>)
<span class="pl-smi">heart_stats</span><span class="pl-k">&lt;-</span><span class="pl-smi">DMPs_heart</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-v">Methylation_heart</span> <span class="pl-k">=</span> <span class="pl-smi">Methylation.Diff</span>, <span class="pl-v">pval_heart</span> <span class="pl-k">=</span> <span class="pl-smi">pval</span>, <span class="pl-v">adj.P.Val_heart</span> <span class="pl-k">=</span> <span class="pl-smi">qval</span>, <span class="pl-smi">Name</span><span class="pl-k">:</span><span class="pl-smi">Regulatory_Feature_Group</span>)
<span class="pl-smi">heart_index</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>filter(<span class="pl-smi">Tissue</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>heart<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">Tissue</span>, <span class="pl-smi">Sample_Group</span>, <span class="pl-smi">Sample_ID</span>)

<span class="pl-c"><span class="pl-c">#</span>Beta</span>
<span class="pl-smi">beta_all</span><span class="pl-k">&lt;-</span>read.csv(<span class="pl-s"><span class="pl-pds">"</span>../1_Input/beta_all.csv<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>)

<span class="pl-c"><span class="pl-c">#</span>combine data</span>
<span class="pl-smi">Combined_DMPs</span><span class="pl-k">&lt;-</span>inner_join(<span class="pl-smi">iPSC_stats</span>, <span class="pl-smi">heart_stats</span>)
<span class="pl-smi">Combined_DMPs_Separated</span><span class="pl-k">&lt;-</span><span class="pl-smi">Combined_DMPs</span> %<span class="pl-k">&gt;</span>% mutate(<span class="pl-v">UCSC_RefGene_Name</span> <span class="pl-k">=</span> strsplit(as.character(<span class="pl-smi">UCSC_RefGene_Name</span>), <span class="pl-s"><span class="pl-pds">"</span>;<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>% unnest(<span class="pl-smi">UCSC_RefGene_Name</span>) %<span class="pl-k">&gt;</span>% distinct() %<span class="pl-k">&gt;</span>% merge(<span class="pl-smi">.</span>, <span class="pl-smi">beta_all</span>, <span class="pl-v">by.x</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Name<span class="pl-pds">"</span></span>, <span class="pl-v">by.y</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>)

<span class="pl-c"><span class="pl-c">#</span> Select only co-methylated positions (iPSC and heart)</span>
<span class="pl-smi">Combined_together</span><span class="pl-k">&lt;-</span><span class="pl-smi">Combined_DMPs_Separated</span> %<span class="pl-k">&gt;</span>% filter(<span class="pl-smi">Methylation_iPSC</span><span class="pl-k">&gt;</span><span class="pl-c1">0</span> <span class="pl-k">&amp;</span> <span class="pl-smi">Methylation_heart</span><span class="pl-k">&gt;</span><span class="pl-c1">0</span> <span class="pl-k">|</span> <span class="pl-smi">Methylation_iPSC</span><span class="pl-k">&lt;</span><span class="pl-c1">0</span> <span class="pl-k">&amp;</span> <span class="pl-smi">Methylation_heart</span><span class="pl-k">&lt;</span><span class="pl-c1">0</span>) %<span class="pl-k">&gt;</span>% arrange(desc(<span class="pl-smi">Methylation_iPSC</span>)) %<span class="pl-k">&gt;</span>% filter(<span class="pl-smi">pval_iPSC</span> <span class="pl-k">&lt;</span> <span class="pl-smi">STAT</span>) %<span class="pl-k">&gt;</span>% arrange(<span class="pl-smi">pval_iPSC</span>)
<span class="pl-c"><span class="pl-c">#</span>Find row number for top candidates</span>
<span class="pl-smi">Candidates</span><span class="pl-k">&lt;-</span>which(<span class="pl-smi">Combined_together</span><span class="pl-k">$</span><span class="pl-smi">UCSC_RefGene_Name</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>ATG7<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> Combined_together&lt;-merge(Combined_together, beta_all, by.x = "Name", by.y = 0)</span>
rownames(<span class="pl-smi">Combined_together</span>)<span class="pl-k">&lt;-</span>make.unique(<span class="pl-smi">Combined_together</span><span class="pl-k">$</span><span class="pl-smi">UCSC_RefGene_Name</span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span>Create heatmap matrices and annotation</span>
<span class="pl-smi">heart_hm</span><span class="pl-k">&lt;-</span><span class="pl-smi">Combined_together</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(contains(<span class="pl-s"><span class="pl-pds">"</span>NF<span class="pl-pds">"</span></span>) <span class="pl-k">|</span> contains(<span class="pl-s"><span class="pl-pds">"</span>biopsy<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>% data.matrix()
<span class="pl-smi">heart_index</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>filter(<span class="pl-smi">Tissue</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>heart<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">Tissue</span>, <span class="pl-smi">Sample_Group</span>, <span class="pl-smi">Sample_ID</span>)
<span class="pl-smi">iPSC_hm</span><span class="pl-k">&lt;-</span><span class="pl-smi">Combined_together</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(contains(<span class="pl-s"><span class="pl-pds">"</span>cells<span class="pl-pds">"</span></span>) <span class="pl-k">&amp;</span> contains(<span class="pl-s"><span class="pl-pds">"</span>co<span class="pl-pds">"</span></span>),contains(<span class="pl-s"><span class="pl-pds">"</span>cells<span class="pl-pds">"</span></span>) <span class="pl-k">&amp;</span> contains(<span class="pl-s"><span class="pl-pds">"</span>go<span class="pl-pds">"</span></span>), contains(<span class="pl-s"><span class="pl-pds">"</span>cells<span class="pl-pds">"</span></span>) <span class="pl-k">&amp;</span> contains(<span class="pl-s"><span class="pl-pds">"</span>bo<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>% data.matrix()
<span class="pl-smi">iPSC_index</span><span class="pl-k">&lt;-</span><span class="pl-smi">AnnoTargets</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>filter(<span class="pl-smi">Tissue</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>iPSC<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">Tissue</span>, <span class="pl-smi">Sample_Group</span>, <span class="pl-smi">Sample_ID</span>)
write.xlsx(<span class="pl-smi">Combined_together</span>, <span class="pl-s"><span class="pl-pds">"</span>../2_Output/Overlapping.DMPs_iPSC.Biopsy.xlsx<span class="pl-pds">"</span></span>, <span class="pl-v">overwrite</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>)

<span class="pl-c"><span class="pl-c">#</span> Venn Diagram</span>
library(<span class="pl-smi">VennDiagram</span>)
<span class="pl-smi">futile.logger</span><span class="pl-k">::</span>flog.threshold(<span class="pl-smi">futile.logger</span><span class="pl-k">::</span><span class="pl-smi">ERROR</span>, <span class="pl-v">name</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>VennDiagramLogger<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## NULL">
                     <pre class="notranslate"><code>## NULL
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="x&lt;-list(iPSC = DMPs_iPSC$Name, heart = DMPs_heart$Name)
venn.diagram(x,fill = c(&quot;darkgray&quot;, &quot;firebrick2&quot;), alpha = c(0.75, 0.75), lty = 'blank', filename = &quot;../2_Output/Overlapping_DMPs_iPSC.Biopsy.png&quot;, na = &quot;remove&quot;)" dir="auto">
                     <pre><span class="pl-smi">x</span><span class="pl-k">&lt;-</span><span class="pl-k">list</span>(<span class="pl-v">iPSC</span> <span class="pl-k">=</span> <span class="pl-smi">DMPs_iPSC</span><span class="pl-k">$</span><span class="pl-smi">Name</span>, <span class="pl-v">heart</span> <span class="pl-k">=</span> <span class="pl-smi">DMPs_heart</span><span class="pl-k">$</span><span class="pl-smi">Name</span>)
venn.diagram(<span class="pl-smi">x</span>,<span class="pl-v">fill</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>darkgray<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>firebrick2<span class="pl-pds">"</span></span>), <span class="pl-v">alpha</span> <span class="pl-k">=</span> c(<span class="pl-c1">0.75</span>, <span class="pl-c1">0.75</span>), <span class="pl-v">lty</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>blank<span class="pl-pds">'</span></span>, <span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>../2_Output/Overlapping_DMPs_iPSC.Biopsy.png<span class="pl-pds">"</span></span>, <span class="pl-v">na</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>remove<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## [1] 1">
                     <pre class="notranslate"><code>## [1] 1
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Heatmap

ann_colors = list(Sample_Group = c(DCM="#1b9e77", CON = "goldenrod2"), Tissue = c(heart = "coral2", iPSC = "darkgray"), Sample_ID = c(A = "white", B = "orange", C = "black", F = "red", G = "green", T = "blue", V = "purple"))
paletteLength &lt;- 100
myColor &lt;- colorRampPalette(c("dodgerblue4", "white", "gold2"))(paletteLength)

#iPSC
heatmap_iPSC&lt;-ComplexHeatmap::pheatmap(iPSC_hm, scale="row",
                    cluster_cols = F,
                    cluster_rows = T,
                     # angle_col = 45,
                     fontsize_col = 8,
                     color = myColor,
                    annotation_names_col = FALSE,
                     show_rownames = F,
                     border_color = NA,
                    annotation_colors = ann_colors,
                    annotation_col = iPSC_index)
heatmap_iPSC


ha = rowAnnotation(foo = anno_mark(at = c(1:10), labels = Combined_together$UCSC_RefGene_Name[1:10]))

#heart
heatmap_heart&lt;-ComplexHeatmap::pheatmap(heart_hm, scale = "row",
                    cluster_cols = F,
                    cluster_rows = T,
                     # angle_col = 45,
                     fontsize_col = 8,
                     color = myColor,
                     show_rownames = F,
                     border_color = NA,
                    annotation_names_col = FALSE,
                    right_annotation = ha,
                    annotation_colors = ann_colors,
                    annotation_col = heart_index)

heatmap_heart
heatmap_iPSC + heatmap_heart

pdf(file="../2_Output/complexheatmap_combined.pdf", height = 5, width = 7, onefile = F)
heatmap_iPSC + heatmap_heart
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Heatmap</span>

<span class="pl-v">ann_colors</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-v">Sample_Group</span> <span class="pl-k">=</span> c(<span class="pl-v">DCM</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>#1b9e77<span class="pl-pds">"</span></span>, <span class="pl-v">CON</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>goldenrod2<span class="pl-pds">"</span></span>), <span class="pl-v">Tissue</span> <span class="pl-k">=</span> c(<span class="pl-v">heart</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>coral2<span class="pl-pds">"</span></span>, <span class="pl-v">iPSC</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>darkgray<span class="pl-pds">"</span></span>), <span class="pl-v">Sample_ID</span> <span class="pl-k">=</span> c(<span class="pl-v">A</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>, <span class="pl-v">B</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>orange<span class="pl-pds">"</span></span>, <span class="pl-v">C</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-c1">F</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-v">G</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>green<span class="pl-pds">"</span></span>, <span class="pl-c1">T</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>, <span class="pl-v">V</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>purple<span class="pl-pds">"</span></span>))
<span class="pl-smi">paletteLength</span> <span class="pl-k">&lt;-</span> <span class="pl-c1">100</span>
<span class="pl-smi">myColor</span> <span class="pl-k">&lt;-</span> colorRampPalette(c(<span class="pl-s"><span class="pl-pds">"</span>dodgerblue4<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>gold2<span class="pl-pds">"</span></span>))(<span class="pl-smi">paletteLength</span>)

<span class="pl-c"><span class="pl-c">#</span>iPSC</span>
<span class="pl-smi">heatmap_iPSC</span><span class="pl-k">&lt;-</span><span class="pl-e">ComplexHeatmap</span><span class="pl-k">::</span>pheatmap(<span class="pl-smi">iPSC_hm</span>, <span class="pl-v">scale</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>row<span class="pl-pds">"</span></span>,
                    <span class="pl-v">cluster_cols</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,
                    <span class="pl-v">cluster_rows</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>,
                     <span class="pl-c"><span class="pl-c">#</span> angle_col = 45,</span>
                     <span class="pl-v">fontsize_col</span> <span class="pl-k">=</span> <span class="pl-c1">8</span>,
                     <span class="pl-v">color</span> <span class="pl-k">=</span> <span class="pl-smi">myColor</span>,
                    <span class="pl-v">annotation_names_col</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>,
                     <span class="pl-v">show_rownames</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,
                     <span class="pl-v">border_color</span> <span class="pl-k">=</span> <span class="pl-c1">NA</span>,
                    <span class="pl-v">annotation_colors</span> <span class="pl-k">=</span> <span class="pl-smi">ann_colors</span>,
                    <span class="pl-v">annotation_col</span> <span class="pl-k">=</span> <span class="pl-smi">iPSC_index</span>)
<span class="pl-smi">heatmap_iPSC</span>


<span class="pl-v">ha</span> <span class="pl-k">=</span> rowAnnotation(<span class="pl-v">foo</span> <span class="pl-k">=</span> anno_mark(<span class="pl-v">at</span> <span class="pl-k">=</span> c(<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>), <span class="pl-v">labels</span> <span class="pl-k">=</span> <span class="pl-smi">Combined_together</span><span class="pl-k">$</span><span class="pl-smi">UCSC_RefGene_Name</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>]))

<span class="pl-c"><span class="pl-c">#</span>heart</span>
<span class="pl-smi">heatmap_heart</span><span class="pl-k">&lt;-</span><span class="pl-e">ComplexHeatmap</span><span class="pl-k">::</span>pheatmap(<span class="pl-smi">heart_hm</span>, <span class="pl-v">scale</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>row<span class="pl-pds">"</span></span>,
                    <span class="pl-v">cluster_cols</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,
                    <span class="pl-v">cluster_rows</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>,
                     <span class="pl-c"><span class="pl-c">#</span> angle_col = 45,</span>
                     <span class="pl-v">fontsize_col</span> <span class="pl-k">=</span> <span class="pl-c1">8</span>,
                     <span class="pl-v">color</span> <span class="pl-k">=</span> <span class="pl-smi">myColor</span>,
                     <span class="pl-v">show_rownames</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,
                     <span class="pl-v">border_color</span> <span class="pl-k">=</span> <span class="pl-c1">NA</span>,
                    <span class="pl-v">annotation_names_col</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>,
                    <span class="pl-v">right_annotation</span> <span class="pl-k">=</span> <span class="pl-smi">ha</span>,
                    <span class="pl-v">annotation_colors</span> <span class="pl-k">=</span> <span class="pl-smi">ann_colors</span>,
                    <span class="pl-v">annotation_col</span> <span class="pl-k">=</span> <span class="pl-smi">heart_index</span>)

<span class="pl-smi">heatmap_heart</span>
<span class="pl-smi">heatmap_iPSC</span> <span class="pl-k">+</span> <span class="pl-smi">heatmap_heart</span>

pdf(<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>../2_Output/complexheatmap_combined.pdf<span class="pl-pds">"</span></span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">5</span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">7</span>, <span class="pl-v">onefile</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>)
<span class="pl-smi">heatmap_iPSC</span> <span class="pl-k">+</span> <span class="pl-smi">heatmap_heart</span>
dev.off()</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## quartz_off_screen 
##                 2">
                     <pre class="notranslate"><code>## quartz_off_screen 
##                 2
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Validation
                     </h1>
                     <a aria-label="Permalink: Validation" class="anchor" href="#validation" id="user-content-validation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#Import Pepin et al. 2019 Combined methylation-RNA-sequencing dataset
Pepin_DMPs&lt;-read.xlsx("../1_Input/Validation/HF.Effect_NR_Pre.v.CON_Annotated_DiffMeth.xlsx")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>Import Pepin et al. 2019 Combined methylation-RNA-sequencing dataset</span>
<span class="pl-smi">Pepin_DMPs</span><span class="pl-k">&lt;-</span>read.xlsx(<span class="pl-s"><span class="pl-pds">"</span>../1_Input/Validation/HF.Effect_NR_Pre.v.CON_Annotated_DiffMeth.xlsx<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Motif Enrichment
                     </h1>
                     <a aria-label="Permalink: Motif Enrichment" class="anchor" href="#motif-enrichment" id="user-content-motif-enrichment">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='###### Working example (from BED file)
library(monaLisa)
library(GenomicRanges)
library(SummarizedExperiment)
library(openxlsx)
library(dplyr)
library(TFBSTools)
library(JASPAR2020)
library(BSgenome.Hsapiens.UCSC.hg19)
mcparams &lt;- BiocParallel::MulticoreParam(10L) #parallelization (10-core)
DMPs_iPSC&lt;-read.xlsx("../3_Results/Overlapping.DMPs_iPSC.Biopsy.xlsx", startRow = 2)
bed&lt;-DMPs_iPSC %&gt;% dplyr::select(chr, pos, Methylation_iPSC) %&gt;% transmute(seqnames=chr, start=pos-20, end=pos+20, width = 41, strand = "*", deltaMeth=Methylation_iPSC)
bed_mr&lt;-as(bed, "GRanges")
# define bins by differential methylation
bins &lt;- bin(x = bed_mr$deltaMeth, binmode = "equalN", nElement = 100, minAbsX = 1)
table(bins)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>##### Working example (from BED file)</span>
library(<span class="pl-smi">monaLisa</span>)
library(<span class="pl-smi">GenomicRanges</span>)
library(<span class="pl-smi">SummarizedExperiment</span>)
library(<span class="pl-smi">openxlsx</span>)
library(<span class="pl-smi">dplyr</span>)
library(<span class="pl-smi">TFBSTools</span>)
library(<span class="pl-smi">JASPAR2020</span>)
library(<span class="pl-smi">BSgenome.Hsapiens.UCSC.hg19</span>)
<span class="pl-smi">mcparams</span> <span class="pl-k">&lt;-</span> <span class="pl-e">BiocParallel</span><span class="pl-k">::</span>MulticoreParam(<span class="pl-c1">10L</span>) <span class="pl-c"><span class="pl-c">#</span>parallelization (10-core)</span>
<span class="pl-smi">DMPs_iPSC</span><span class="pl-k">&lt;-</span>read.xlsx(<span class="pl-s"><span class="pl-pds">"</span>../3_Results/Overlapping.DMPs_iPSC.Biopsy.xlsx<span class="pl-pds">"</span></span>, <span class="pl-v">startRow</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>)
<span class="pl-smi">bed</span><span class="pl-k">&lt;-</span><span class="pl-smi">DMPs_iPSC</span> %<span class="pl-k">&gt;</span>% <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">chr</span>, <span class="pl-smi">pos</span>, <span class="pl-smi">Methylation_iPSC</span>) %<span class="pl-k">&gt;</span>% transmute(<span class="pl-v">seqnames</span><span class="pl-k">=</span><span class="pl-smi">chr</span>, <span class="pl-v">start</span><span class="pl-k">=</span><span class="pl-smi">pos</span><span class="pl-k">-</span><span class="pl-c1">20</span>, <span class="pl-v">end</span><span class="pl-k">=</span><span class="pl-smi">pos</span><span class="pl-k">+</span><span class="pl-c1">20</span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">41</span>, <span class="pl-v">strand</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>*<span class="pl-pds">"</span></span>, <span class="pl-v">deltaMeth</span><span class="pl-k">=</span><span class="pl-smi">Methylation_iPSC</span>)
<span class="pl-smi">bed_mr</span><span class="pl-k">&lt;-</span>as(<span class="pl-smi">bed</span>, <span class="pl-s"><span class="pl-pds">"</span>GRanges<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> define bins by differential methylation</span>
<span class="pl-smi">bins</span> <span class="pl-k">&lt;-</span> bin(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">bed_mr</span><span class="pl-k">$</span><span class="pl-smi">deltaMeth</span>, <span class="pl-v">binmode</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>equalN<span class="pl-pds">"</span></span>, <span class="pl-v">nElement</span> <span class="pl-k">=</span> <span class="pl-c1">100</span>, <span class="pl-v">minAbsX</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>)
table(<span class="pl-smi">bins</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## bins
## [-6.06,-1.5]  (-1.5,1.02]  (1.02,7.75] 
##          100          226          100">
                     <pre class="notranslate"><code>## bins
## [-6.06,-1.5]  (-1.5,1.02]  (1.02,7.75] 
##          100          226          100
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf(paste0("../2_Output/_bins.pdf"), width = 7.5, height = 5)
plotBinDensity(bed_mr$deltaMeth, bins, legend = "topright")
dev.off()' dir="auto">
                     <pre>pdf(paste0(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/_bins.pdf<span class="pl-pds">"</span></span>), <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">7.5</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">5</span>)
plotBinDensity(<span class="pl-smi">bed_mr</span><span class="pl-k">$</span><span class="pl-smi">deltaMeth</span>, <span class="pl-smi">bins</span>, <span class="pl-v">legend</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>topright<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## quartz_off_screen 
##                 2">
                     <pre class="notranslate"><code>## quartz_off_screen 
##                 2
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# get PWMs from JASPAR
pwms &lt;- getMatrixSet(JASPAR2020,
                     opts = list(matrixtype = "PWM",
                                 tax_group = "vertebrates"))
# trim bed file for sequenes that are consistent
lmrsel &lt;- trim(resize(bed_mr, width = median(width(bed_mr)), fix = "center"))
# get sequences from mouse genome
lmrseqs &lt;- getSeq(BSgenome.Hsapiens.UCSC.hg19, bed_mr)
# GC proportion (bias)
plotBinDiagnostics(seqs = lmrseqs, bins = bins, aspect = "GCfrac")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> get PWMs from JASPAR</span>
<span class="pl-smi">pwms</span> <span class="pl-k">&lt;-</span> getMatrixSet(<span class="pl-smi">JASPAR2020</span>,
                     <span class="pl-v">opts</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-v">matrixtype</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>PWM<span class="pl-pds">"</span></span>,
                                 <span class="pl-v">tax_group</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>vertebrates<span class="pl-pds">"</span></span>))
<span class="pl-c"><span class="pl-c">#</span> trim bed file for sequenes that are consistent</span>
<span class="pl-smi">lmrsel</span> <span class="pl-k">&lt;-</span> trim(resize(<span class="pl-smi">bed_mr</span>, <span class="pl-v">width</span> <span class="pl-k">=</span> median(width(<span class="pl-smi">bed_mr</span>)), <span class="pl-v">fix</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>center<span class="pl-pds">"</span></span>))
<span class="pl-c"><span class="pl-c">#</span> get sequences from mouse genome</span>
<span class="pl-smi">lmrseqs</span> <span class="pl-k">&lt;-</span> getSeq(<span class="pl-smi">BSgenome.Hsapiens.UCSC.hg19</span>, <span class="pl-smi">bed_mr</span>)
<span class="pl-c"><span class="pl-c">#</span> GC proportion (bias)</span>
plotBinDiagnostics(<span class="pl-v">seqs</span> <span class="pl-k">=</span> <span class="pl-smi">lmrseqs</span>, <span class="pl-v">bins</span> <span class="pl-k">=</span> <span class="pl-smi">bins</span>, <span class="pl-v">aspect</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>GCfrac<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/mepepin/Indirect.Epigenomics/blob/main/README_files/figure-html/motif-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/mepepin/Indirect.Epigenomics/raw/main/README_files/figure-html/motif-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='plotBinDiagnostics(seqs = lmrseqs, bins = bins, aspect = "dinucfreq")
# run motif enrichment
se &lt;- calcBinnedMotifEnrR(seqs = lmrseqs, bins = bins, pwmL = pwms, BPPARAM = BiocParallel::MulticoreParam(8))
# Filter results
Test&lt;-as.data.frame(assays(se))
sel &lt;- apply(assay(se, "negLog10P"), 1, 
             function(x) max(abs(x), 0, na.rm = TRUE)) &gt; 1.0
sum(sel)' dir="auto">
                     <pre>plotBinDiagnostics(<span class="pl-v">seqs</span> <span class="pl-k">=</span> <span class="pl-smi">lmrseqs</span>, <span class="pl-v">bins</span> <span class="pl-k">=</span> <span class="pl-smi">bins</span>, <span class="pl-v">aspect</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>dinucfreq<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> run motif enrichment</span>
<span class="pl-smi">se</span> <span class="pl-k">&lt;-</span> calcBinnedMotifEnrR(<span class="pl-v">seqs</span> <span class="pl-k">=</span> <span class="pl-smi">lmrseqs</span>, <span class="pl-v">bins</span> <span class="pl-k">=</span> <span class="pl-smi">bins</span>, <span class="pl-v">pwmL</span> <span class="pl-k">=</span> <span class="pl-smi">pwms</span>, <span class="pl-v">BPPARAM</span> <span class="pl-k">=</span> <span class="pl-e">BiocParallel</span><span class="pl-k">::</span>MulticoreParam(<span class="pl-c1">8</span>))
<span class="pl-c"><span class="pl-c">#</span> Filter results</span>
<span class="pl-smi">Test</span><span class="pl-k">&lt;-</span>as.data.frame(assays(<span class="pl-smi">se</span>))
<span class="pl-smi">sel</span> <span class="pl-k">&lt;-</span> apply(assay(<span class="pl-smi">se</span>, <span class="pl-s"><span class="pl-pds">"</span>negLog10P<span class="pl-pds">"</span></span>), <span class="pl-c1">1</span>, 
             <span class="pl-k">function</span>(<span class="pl-smi">x</span>) max(abs(<span class="pl-smi">x</span>), <span class="pl-c1">0</span>, <span class="pl-v">na.rm</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)) <span class="pl-k">&gt;</span> <span class="pl-c1">1.0</span>
sum(<span class="pl-smi">sel</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## [1] 65">
                     <pre class="notranslate"><code>## [1] 65
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#&gt; [1] 59
seSel &lt;- se[sel, ]

# plot
pdf("../2_Output/Motifs.pdf", width = 11, height = 10)
plotMotifHeatmaps(x = seSel, which.plots = c("log2enr", "negLog10P"), 
                  width = 2.0, cluster = TRUE, maxEnr = 2, maxSig = 10, 
                  show_motif_GC = TRUE, show_dendrogram = T,show_seqlogo = TRUE)
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>&gt; [1] 59</span>
<span class="pl-smi">seSel</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">se</span>[<span class="pl-smi">sel</span>, ]

<span class="pl-c"><span class="pl-c">#</span> plot</span>
pdf(<span class="pl-s"><span class="pl-pds">"</span>../2_Output/Motifs.pdf<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">11</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>)
plotMotifHeatmaps(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">seSel</span>, <span class="pl-v">which.plots</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>log2enr<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>negLog10P<span class="pl-pds">"</span></span>), 
                  <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">2.0</span>, <span class="pl-v">cluster</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">maxEnr</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, <span class="pl-v">maxSig</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>, 
                  <span class="pl-v">show_motif_GC</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">show_dendrogram</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>,<span class="pl-v">show_seqlogo</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## quartz_off_screen 
##                 2">
                     <pre class="notranslate"><code>## quartz_off_screen 
##                 2
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='plotMotifHeatmaps(x = seSel, which.plots = c("log2enr", "negLog10P"), 
                  width = 2.0, cluster = TRUE, maxEnr = 2, maxSig = 10, 
                  show_motif_GC = TRUE, show_dendrogram = T,show_seqlogo = TRUE)' dir="auto">
                     <pre>plotMotifHeatmaps(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">seSel</span>, <span class="pl-v">which.plots</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>log2enr<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>negLog10P<span class="pl-pds">"</span></span>), 
                  <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">2.0</span>, <span class="pl-v">cluster</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">maxEnr</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, <span class="pl-v">maxSig</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>, 
                  <span class="pl-v">show_motif_GC</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">show_dendrogram</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>,<span class="pl-v">show_seqlogo</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Supplemental Table: R Session Information
                     </h1>
                     <a aria-label="Permalink: Supplemental Table: R Session Information" class="anchor" href="#supplemental-table-r-session-information" id="user-content-supplemental-table-r-session-information">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     All packages and setting are acquired using the following command:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="options(kableExtra.latex.load_packages = FALSE)
Run_tE&lt;-Sys.time()
Run_time&lt;-Run_tE - Run_tS
Run_time" dir="auto">
                     <pre>options(<span class="pl-v">kableExtra.latex.load_packages</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)
<span class="pl-smi">Run_tE</span><span class="pl-k">&lt;-</span>Sys.time()
<span class="pl-smi">Run_time</span><span class="pl-k">&lt;-</span><span class="pl-smi">Run_tE</span> <span class="pl-k">-</span> <span class="pl-smi">Run_tS</span>
<span class="pl-smi">Run_time</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## Time difference of 14.97141 mins">
                     <pre class="notranslate"><code>## Time difference of 14.97141 mins
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(kableExtra)
sinfo&lt;-devtools::session_info()
sinfo$platform" dir="auto">
                     <pre>library(<span class="pl-smi">kableExtra</span>)
<span class="pl-smi">sinfo</span><span class="pl-k">&lt;-</span><span class="pl-e">devtools</span><span class="pl-k">::</span>session_info()
<span class="pl-smi">sinfo</span><span class="pl-k">$</span><span class="pl-smi">platform</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Big Sur 10.16
##  system   x86_64, darwin17.0
##  ui       X11
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Europe/Berlin
##  date     2021-12-31
##  pandoc   2.14.0.3 @ /Applications/RStudio.app/Contents/MacOS/pandoc/ (via rmarkdown)">
                     <pre class="notranslate"><code>##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Big Sur 10.16
##  system   x86_64, darwin17.0
##  ui       X11
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Europe/Berlin
##  date     2021-12-31
##  pandoc   2.14.0.3 @ /Applications/RStudio.app/Contents/MacOS/pandoc/ (via rmarkdown)
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='sinfo$packages %&gt;% kable( 
                         align="c", 
                         caption="Packages and Required Dependencies") %&gt;% 
    kable_styling(latex_options=c("repeat_header", "condensed"))' dir="auto">
                     <pre><span class="pl-smi">sinfo</span><span class="pl-k">$</span><span class="pl-smi">packages</span> %<span class="pl-k">&gt;</span>% kable( 
                         <span class="pl-v">align</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>c<span class="pl-pds">"</span></span>, 
                         <span class="pl-v">caption</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Packages and Required Dependencies<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
    kable_styling(<span class="pl-v">latex_options</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>repeat_header<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>condensed<span class="pl-pds">"</span></span>))</pre>
                    </div>
                    Packages and Required Dependencies
                    <table>
                     <thead>
                      <tr>
                       <th>
                       </th>
                       <th>
                        package
                       </th>
                       <th>
                        ondiskversion
                       </th>
                       <th>
                        loadedversion
                       </th>
                       <th>
                        path
                       </th>
                       <th>
                        loadedpath
                       </th>
                       <th>
                        attached
                       </th>
                       <th>
                        is_base
                       </th>
                       <th>
                        date
                       </th>
                       <th>
                        source
                       </th>
                       <th>
                        md5ok
                       </th>
                       <th>
                        library
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        abind
                       </td>
                       <td>
                        abind
                       </td>
                       <td>
                        1.4.5
                       </td>
                       <td>
                        1.4-5
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/abind
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/abind
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2016-07-21
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        annotate
                       </td>
                       <td>
                        annotate
                       </td>
                       <td>
                        1.72.0
                       </td>
                       <td>
                        1.72.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/annotate
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/annotate
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        AnnotationDbi
                       </td>
                       <td>
                        AnnotationDbi
                       </td>
                       <td>
                        1.56.2
                       </td>
                       <td>
                        1.56.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/AnnotationDbi
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/AnnotationDbi
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-09
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        askpass
                       </td>
                       <td>
                        askpass
                       </td>
                       <td>
                        1.1
                       </td>
                       <td>
                        1.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/askpass
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/askpass
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-01-13
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        assertthat
                       </td>
                       <td>
                        assertthat
                       </td>
                       <td>
                        0.2.1
                       </td>
                       <td>
                        0.2.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/assertthat
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/assertthat
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-03-21
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        backports
                       </td>
                       <td>
                        backports
                       </td>
                       <td>
                        1.4.0
                       </td>
                       <td>
                        1.4.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/backports
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/backports
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-23
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        base64
                       </td>
                       <td>
                        base64
                       </td>
                       <td>
                        2.0
                       </td>
                       <td>
                        2.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/base64
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/base64
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2016-05-10
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        base64enc
                       </td>
                       <td>
                        base64enc
                       </td>
                       <td>
                        0.1.3
                       </td>
                       <td>
                        0.1-3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/base64enc
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/base64enc
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2015-07-28
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        beanplot
                       </td>
                       <td>
                        beanplot
                       </td>
                       <td>
                        1.2
                       </td>
                       <td>
                        1.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/beanplot
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/beanplot
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2014-09-19
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Biobase
                       </td>
                       <td>
                        Biobase
                       </td>
                       <td>
                        2.54.0
                       </td>
                       <td>
                        2.54.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Biobase
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Biobase
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        BiocFileCache
                       </td>
                       <td>
                        BiocFileCache
                       </td>
                       <td>
                        2.2.0
                       </td>
                       <td>
                        2.2.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocFileCache
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocFileCache
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        BiocGenerics
                       </td>
                       <td>
                        BiocGenerics
                       </td>
                       <td>
                        0.40.0
                       </td>
                       <td>
                        0.40.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocGenerics
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocGenerics
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        BiocIO
                       </td>
                       <td>
                        BiocIO
                       </td>
                       <td>
                        1.4.0
                       </td>
                       <td>
                        1.4.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocIO
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocIO
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        BiocParallel
                       </td>
                       <td>
                        BiocParallel
                       </td>
                       <td>
                        1.28.2
                       </td>
                       <td>
                        1.28.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocParallel
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BiocParallel
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-25
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        biomaRt
                       </td>
                       <td>
                        biomaRt
                       </td>
                       <td>
                        2.50.1
                       </td>
                       <td>
                        2.50.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/biomaRt
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/biomaRt
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-21
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Biostrings
                       </td>
                       <td>
                        Biostrings
                       </td>
                       <td>
                        2.62.0
                       </td>
                       <td>
                        2.62.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Biostrings
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Biostrings
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        bit
                       </td>
                       <td>
                        bit
                       </td>
                       <td>
                        4.0.4
                       </td>
                       <td>
                        4.0.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bit
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bit
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-08-04
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        bit64
                       </td>
                       <td>
                        bit64
                       </td>
                       <td>
                        4.0.5
                       </td>
                       <td>
                        4.0.5
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bit64
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bit64
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-08-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        bitops
                       </td>
                       <td>
                        bitops
                       </td>
                       <td>
                        1.0.7
                       </td>
                       <td>
                        1.0-7
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bitops
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bitops
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-04-24
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        blob
                       </td>
                       <td>
                        blob
                       </td>
                       <td>
                        1.2.2
                       </td>
                       <td>
                        1.2.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/blob
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/blob
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-07-23
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        broom
                       </td>
                       <td>
                        broom
                       </td>
                       <td>
                        0.7.10
                       </td>
                       <td>
                        0.7.10
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/broom
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/broom
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-31
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        BSgenome
                       </td>
                       <td>
                        BSgenome
                       </td>
                       <td>
                        1.62.0
                       </td>
                       <td>
                        1.62.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BSgenome
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BSgenome
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        BSgenome.Hsapiens.UCSC.hg19
                       </td>
                       <td>
                        BSgenome.Hsapiens.UCSC.hg19
                       </td>
                       <td>
                        1.4.3
                       </td>
                       <td>
                        1.4.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BSgenome.Hsapiens.UCSC.hg19
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/BSgenome.Hsapiens.UCSC.hg19
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-12-02
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        bslib
                       </td>
                       <td>
                        bslib
                       </td>
                       <td>
                        0.3.1
                       </td>
                       <td>
                        0.3.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bslib
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bslib
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-06
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        bumphunter
                       </td>
                       <td>
                        bumphunter
                       </td>
                       <td>
                        1.36.0
                       </td>
                       <td>
                        1.36.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bumphunter
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/bumphunter
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        cachem
                       </td>
                       <td>
                        cachem
                       </td>
                       <td>
                        1.0.6
                       </td>
                       <td>
                        1.0.6
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cachem
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cachem
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-08-19
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        calibrate
                       </td>
                       <td>
                        calibrate
                       </td>
                       <td>
                        1.7.7
                       </td>
                       <td>
                        1.7.7
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/calibrate
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/calibrate
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-06-19
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        callr
                       </td>
                       <td>
                        callr
                       </td>
                       <td>
                        3.7.0
                       </td>
                       <td>
                        3.7.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/callr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/callr
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-04-20
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        car
                       </td>
                       <td>
                        car
                       </td>
                       <td>
                        3.0.12
                       </td>
                       <td>
                        3.0-12
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/car
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/car
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-06
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        carData
                       </td>
                       <td>
                        carData
                       </td>
                       <td>
                        3.0.4
                       </td>
                       <td>
                        3.0-4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/carData
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/carData
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-05-22
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        caTools
                       </td>
                       <td>
                        caTools
                       </td>
                       <td>
                        1.18.2
                       </td>
                       <td>
                        1.18.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/caTools
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/caTools
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-03-28
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        cellranger
                       </td>
                       <td>
                        cellranger
                       </td>
                       <td>
                        1.1.0
                       </td>
                       <td>
                        1.1.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cellranger
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cellranger
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2016-07-27
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        checkmate
                       </td>
                       <td>
                        checkmate
                       </td>
                       <td>
                        2.0.0
                       </td>
                       <td>
                        2.0.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/checkmate
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/checkmate
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-02-06
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        circlize
                       </td>
                       <td>
                        circlize
                       </td>
                       <td>
                        0.4.13
                       </td>
                       <td>
                        0.4.13
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/circlize
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/circlize
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-06-09
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        cli
                       </td>
                       <td>
                        cli
                       </td>
                       <td>
                        3.1.0
                       </td>
                       <td>
                        3.1.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cli
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cli
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-27
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        clue
                       </td>
                       <td>
                        clue
                       </td>
                       <td>
                        0.3.60
                       </td>
                       <td>
                        0.3-60
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/clue
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/clue
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-11
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        cluster
                       </td>
                       <td>
                        cluster
                       </td>
                       <td>
                        2.1.2
                       </td>
                       <td>
                        2.1.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cluster
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cluster
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-04-17
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CNEr
                       </td>
                       <td>
                        CNEr
                       </td>
                       <td>
                        1.30.0
                       </td>
                       <td>
                        1.30.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/CNEr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/CNEr
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        codetools
                       </td>
                       <td>
                        codetools
                       </td>
                       <td>
                        0.2.18
                       </td>
                       <td>
                        0.2-18
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/codetools
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/codetools
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-11-04
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        colorspace
                       </td>
                       <td>
                        colorspace
                       </td>
                       <td>
                        2.0.2
                       </td>
                       <td>
                        2.0-2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/colorspace
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/colorspace
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-06-24
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ComplexHeatmap
                       </td>
                       <td>
                        ComplexHeatmap
                       </td>
                       <td>
                        2.11.1
                       </td>
                       <td>
                        2.11.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ComplexHeatmap
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ComplexHeatmap
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-26
                       </td>
                       <td>
                        Github (jokergoo/ComplexHeatmap@826b321)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        corrplot
                       </td>
                       <td>
                        corrplot
                       </td>
                       <td>
                        0.92
                       </td>
                       <td>
                        0.92
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/corrplot
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/corrplot
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-18
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        cowplot
                       </td>
                       <td>
                        cowplot
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cowplot
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/cowplot
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-12-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        crayon
                       </td>
                       <td>
                        crayon
                       </td>
                       <td>
                        1.4.2
                       </td>
                       <td>
                        1.4.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/crayon
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/crayon
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-29
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        crosstalk
                       </td>
                       <td>
                        crosstalk
                       </td>
                       <td>
                        1.2.0
                       </td>
                       <td>
                        1.2.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/crosstalk
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/crosstalk
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-04
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        curl
                       </td>
                       <td>
                        curl
                       </td>
                       <td>
                        4.3.2
                       </td>
                       <td>
                        4.3.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/curl
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/curl
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-06-23
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        data.table
                       </td>
                       <td>
                        data.table
                       </td>
                       <td>
                        1.14.2
                       </td>
                       <td>
                        1.14.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/data.table
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/data.table
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-27
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DBI
                       </td>
                       <td>
                        DBI
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DBI
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DBI
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-01-15
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        dbplyr
                       </td>
                       <td>
                        dbplyr
                       </td>
                       <td>
                        2.1.1
                       </td>
                       <td>
                        2.1.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/dbplyr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/dbplyr
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-04-06
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DelayedArray
                       </td>
                       <td>
                        DelayedArray
                       </td>
                       <td>
                        0.20.0
                       </td>
                       <td>
                        0.20.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DelayedArray
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DelayedArray
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DelayedMatrixStats
                       </td>
                       <td>
                        DelayedMatrixStats
                       </td>
                       <td>
                        1.16.0
                       </td>
                       <td>
                        1.16.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DelayedMatrixStats
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DelayedMatrixStats
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        desc
                       </td>
                       <td>
                        desc
                       </td>
                       <td>
                        1.4.0
                       </td>
                       <td>
                        1.4.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/desc
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/desc
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-28
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DESeq2
                       </td>
                       <td>
                        DESeq2
                       </td>
                       <td>
                        1.34.0
                       </td>
                       <td>
                        1.34.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DESeq2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DESeq2
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        devtools
                       </td>
                       <td>
                        devtools
                       </td>
                       <td>
                        2.4.3
                       </td>
                       <td>
                        2.4.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/devtools
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/devtools
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        digest
                       </td>
                       <td>
                        digest
                       </td>
                       <td>
                        0.6.29
                       </td>
                       <td>
                        0.6.29
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/digest
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/digest
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-12-01
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DirichletMultinomial
                       </td>
                       <td>
                        DirichletMultinomial
                       </td>
                       <td>
                        1.36.0
                       </td>
                       <td>
                        1.36.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DirichletMultinomial
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/DirichletMultinomial
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        doParallel
                       </td>
                       <td>
                        doParallel
                       </td>
                       <td>
                        1.0.16
                       </td>
                       <td>
                        1.0.16
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/doParallel
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/doParallel
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-10-16
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        doRNG
                       </td>
                       <td>
                        doRNG
                       </td>
                       <td>
                        1.8.2
                       </td>
                       <td>
                        1.8.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/doRNG
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/doRNG
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-01-27
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        dplyr
                       </td>
                       <td>
                        dplyr
                       </td>
                       <td>
                        1.0.7
                       </td>
                       <td>
                        1.0.7
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/dplyr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/dplyr
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-06-18
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ellipsis
                       </td>
                       <td>
                        ellipsis
                       </td>
                       <td>
                        0.3.2
                       </td>
                       <td>
                        0.3.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ellipsis
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ellipsis
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-04-29
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        evaluate
                       </td>
                       <td>
                        evaluate
                       </td>
                       <td>
                        0.14
                       </td>
                       <td>
                        0.14
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/evaluate
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/evaluate
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-05-28
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        fansi
                       </td>
                       <td>
                        fansi
                       </td>
                       <td>
                        0.5.0
                       </td>
                       <td>
                        0.5.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fansi
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fansi
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-05-25
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        farver
                       </td>
                       <td>
                        farver
                       </td>
                       <td>
                        2.1.0
                       </td>
                       <td>
                        2.1.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/farver
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/farver
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-02-28
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        fastmap
                       </td>
                       <td>
                        fastmap
                       </td>
                       <td>
                        1.1.0
                       </td>
                       <td>
                        1.1.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fastmap
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fastmap
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-01-25
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ff
                       </td>
                       <td>
                        ff
                       </td>
                       <td>
                        4.0.5
                       </td>
                       <td>
                        4.0.5
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ff
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ff
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-29
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        filelock
                       </td>
                       <td>
                        filelock
                       </td>
                       <td>
                        1.0.2
                       </td>
                       <td>
                        1.0.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/filelock
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/filelock
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2018-10-05
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        foreach
                       </td>
                       <td>
                        foreach
                       </td>
                       <td>
                        1.5.1
                       </td>
                       <td>
                        1.5.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/foreach
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/foreach
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-10-15
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        foreign
                       </td>
                       <td>
                        foreign
                       </td>
                       <td>
                        0.8.81
                       </td>
                       <td>
                        0.8-81
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/foreign
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/foreign
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-12-22
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        formatR
                       </td>
                       <td>
                        formatR
                       </td>
                       <td>
                        1.11
                       </td>
                       <td>
                        1.11
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/formatR
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/formatR
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-06-01
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Formula
                       </td>
                       <td>
                        Formula
                       </td>
                       <td>
                        1.2.4
                       </td>
                       <td>
                        1.2-4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Formula
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Formula
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-10-16
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        fs
                       </td>
                       <td>
                        fs
                       </td>
                       <td>
                        1.5.1
                       </td>
                       <td>
                        1.5.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fs
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/fs
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        futile.logger
                       </td>
                       <td>
                        futile.logger
                       </td>
                       <td>
                        1.4.3
                       </td>
                       <td>
                        1.4.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/futile.logger
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/futile.logger
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2016-07-10
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        futile.options
                       </td>
                       <td>
                        futile.options
                       </td>
                       <td>
                        1.0.1
                       </td>
                       <td>
                        1.0.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/futile.options
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/futile.options
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2018-04-20
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        genefilter
                       </td>
                       <td>
                        genefilter
                       </td>
                       <td>
                        1.76.0
                       </td>
                       <td>
                        1.76.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/genefilter
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/genefilter
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        geneplotter
                       </td>
                       <td>
                        geneplotter
                       </td>
                       <td>
                        1.72.0
                       </td>
                       <td>
                        1.72.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/geneplotter
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/geneplotter
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        generics
                       </td>
                       <td>
                        generics
                       </td>
                       <td>
                        0.1.1
                       </td>
                       <td>
                        0.1.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/generics
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/generics
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-25
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GenomeInfoDb
                       </td>
                       <td>
                        GenomeInfoDb
                       </td>
                       <td>
                        1.30.0
                       </td>
                       <td>
                        1.30.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomeInfoDb
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomeInfoDb
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GenomeInfoDbData
                       </td>
                       <td>
                        GenomeInfoDbData
                       </td>
                       <td>
                        1.2.7
                       </td>
                       <td>
                        1.2.7
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomeInfoDbData
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomeInfoDbData
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-17
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GenomicAlignments
                       </td>
                       <td>
                        GenomicAlignments
                       </td>
                       <td>
                        1.30.0
                       </td>
                       <td>
                        1.30.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicAlignments
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicAlignments
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GenomicFeatures
                       </td>
                       <td>
                        GenomicFeatures
                       </td>
                       <td>
                        1.46.1
                       </td>
                       <td>
                        1.46.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicFeatures
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicFeatures
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-27
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GenomicRanges
                       </td>
                       <td>
                        GenomicRanges
                       </td>
                       <td>
                        1.46.1
                       </td>
                       <td>
                        1.46.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicRanges
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GenomicRanges
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-18
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GEOquery
                       </td>
                       <td>
                        GEOquery
                       </td>
                       <td>
                        2.62.1
                       </td>
                       <td>
                        2.62.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GEOquery
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GEOquery
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-16
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GetoptLong
                       </td>
                       <td>
                        GetoptLong
                       </td>
                       <td>
                        1.0.5
                       </td>
                       <td>
                        1.0.5
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GetoptLong
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GetoptLong
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-12-15
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ggplot2
                       </td>
                       <td>
                        ggplot2
                       </td>
                       <td>
                        3.3.5
                       </td>
                       <td>
                        3.3.5
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggplot2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggplot2
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-06-25
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ggpubr
                       </td>
                       <td>
                        ggpubr
                       </td>
                       <td>
                        0.4.0
                       </td>
                       <td>
                        0.4.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggpubr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggpubr
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-06-27
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ggrepel
                       </td>
                       <td>
                        ggrepel
                       </td>
                       <td>
                        0.9.1
                       </td>
                       <td>
                        0.9.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggrepel
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggrepel
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-01-15
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ggsignif
                       </td>
                       <td>
                        ggsignif
                       </td>
                       <td>
                        0.6.3
                       </td>
                       <td>
                        0.6.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggsignif
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggsignif
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-09
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        glmnet
                       </td>
                       <td>
                        glmnet
                       </td>
                       <td>
                        4.1.3
                       </td>
                       <td>
                        4.1-3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/glmnet
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/glmnet
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-02
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GlobalOptions
                       </td>
                       <td>
                        GlobalOptions
                       </td>
                       <td>
                        0.1.2
                       </td>
                       <td>
                        0.1.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GlobalOptions
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GlobalOptions
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-06-10
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        glue
                       </td>
                       <td>
                        glue
                       </td>
                       <td>
                        1.5.1
                       </td>
                       <td>
                        1.5.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/glue
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/glue
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GO.db
                       </td>
                       <td>
                        GO.db
                       </td>
                       <td>
                        3.14.0
                       </td>
                       <td>
                        3.14.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GO.db
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/GO.db
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-12-06
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        gridExtra
                       </td>
                       <td>
                        gridExtra
                       </td>
                       <td>
                        2.3
                       </td>
                       <td>
                        2.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gridExtra
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gridExtra
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2017-09-09
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        gtable
                       </td>
                       <td>
                        gtable
                       </td>
                       <td>
                        0.3.0
                       </td>
                       <td>
                        0.3.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gtable
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gtable
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-03-25
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        gtools
                       </td>
                       <td>
                        gtools
                       </td>
                       <td>
                        3.9.2
                       </td>
                       <td>
                        3.9.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gtools
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/gtools
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-06-06
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Haplin
                       </td>
                       <td>
                        Haplin
                       </td>
                       <td>
                        7.2.3
                       </td>
                       <td>
                        7.2.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Haplin
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Haplin
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-09-07
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        HDF5Array
                       </td>
                       <td>
                        HDF5Array
                       </td>
                       <td>
                        1.22.1
                       </td>
                       <td>
                        1.22.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/HDF5Array
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/HDF5Array
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-14
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        highr
                       </td>
                       <td>
                        highr
                       </td>
                       <td>
                        0.9
                       </td>
                       <td>
                        0.9
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/highr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/highr
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-04-16
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Hmisc
                       </td>
                       <td>
                        Hmisc
                       </td>
                       <td>
                        4.6.0
                       </td>
                       <td>
                        4.6-0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Hmisc
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Hmisc
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-07
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        hms
                       </td>
                       <td>
                        hms
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/hms
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/hms
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-26
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        htmlTable
                       </td>
                       <td>
                        htmlTable
                       </td>
                       <td>
                        2.3.0
                       </td>
                       <td>
                        2.3.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmlTable
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmlTable
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-12
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        htmltools
                       </td>
                       <td>
                        htmltools
                       </td>
                       <td>
                        0.5.2
                       </td>
                       <td>
                        0.5.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmltools
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmltools
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-08-25
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        htmlwidgets
                       </td>
                       <td>
                        htmlwidgets
                       </td>
                       <td>
                        1.5.4
                       </td>
                       <td>
                        1.5.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmlwidgets
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/htmlwidgets
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-08
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        httpuv
                       </td>
                       <td>
                        httpuv
                       </td>
                       <td>
                        1.6.3
                       </td>
                       <td>
                        1.6.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/httpuv
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/httpuv
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-09
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        httr
                       </td>
                       <td>
                        httr
                       </td>
                       <td>
                        1.4.2
                       </td>
                       <td>
                        1.4.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/httr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/httr
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-07-20
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        IlluminaHumanMethylation450kanno.ilmn12.hg19
                       </td>
                       <td>
                        IlluminaHumanMethylation450kanno.ilmn12.hg19
                       </td>
                       <td>
                        0.6.0
                       </td>
                       <td>
                        0.6.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IlluminaHumanMethylation450kanno.ilmn12.hg19
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IlluminaHumanMethylation450kanno.ilmn12.hg19
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-17
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        IlluminaHumanMethylation450kmanifest
                       </td>
                       <td>
                        IlluminaHumanMethylation450kmanifest
                       </td>
                       <td>
                        0.4.0
                       </td>
                       <td>
                        0.4.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IlluminaHumanMethylation450kmanifest
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IlluminaHumanMethylation450kmanifest
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-17
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        illuminaio
                       </td>
                       <td>
                        illuminaio
                       </td>
                       <td>
                        0.36.0
                       </td>
                       <td>
                        0.36.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/illuminaio
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/illuminaio
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        IRanges
                       </td>
                       <td>
                        IRanges
                       </td>
                       <td>
                        2.28.0
                       </td>
                       <td>
                        2.28.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IRanges
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/IRanges
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        iterators
                       </td>
                       <td>
                        iterators
                       </td>
                       <td>
                        1.0.13
                       </td>
                       <td>
                        1.0.13
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/iterators
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/iterators
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-10-15
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        JASPAR2020
                       </td>
                       <td>
                        JASPAR2020
                       </td>
                       <td>
                        0.99.10
                       </td>
                       <td>
                        0.99.10
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/JASPAR2020
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/JASPAR2020
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-12-06
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        jpeg
                       </td>
                       <td>
                        jpeg
                       </td>
                       <td>
                        0.1.9
                       </td>
                       <td>
                        0.1-9
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jpeg
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jpeg
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-07-24
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        jquerylib
                       </td>
                       <td>
                        jquerylib
                       </td>
                       <td>
                        0.1.4
                       </td>
                       <td>
                        0.1.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jquerylib
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jquerylib
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-04-26
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        jsonlite
                       </td>
                       <td>
                        jsonlite
                       </td>
                       <td>
                        1.7.2
                       </td>
                       <td>
                        1.7.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jsonlite
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/jsonlite
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-12-09
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        kableExtra
                       </td>
                       <td>
                        kableExtra
                       </td>
                       <td>
                        1.3.4
                       </td>
                       <td>
                        1.3.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/kableExtra
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/kableExtra
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-02-20
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        KEGGREST
                       </td>
                       <td>
                        KEGGREST
                       </td>
                       <td>
                        1.34.0
                       </td>
                       <td>
                        1.34.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/KEGGREST
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/KEGGREST
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        knitr
                       </td>
                       <td>
                        knitr
                       </td>
                       <td>
                        1.36
                       </td>
                       <td>
                        1.36
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/knitr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/knitr
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-29
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        labeling
                       </td>
                       <td>
                        labeling
                       </td>
                       <td>
                        0.4.2
                       </td>
                       <td>
                        0.4.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/labeling
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/labeling
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-10-20
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        lambda.r
                       </td>
                       <td>
                        lambda.r
                       </td>
                       <td>
                        1.2.4
                       </td>
                       <td>
                        1.2.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lambda.r
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lambda.r
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-09-18
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        later
                       </td>
                       <td>
                        later
                       </td>
                       <td>
                        1.3.0
                       </td>
                       <td>
                        1.3.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/later
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/later
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-08-18
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        lattice
                       </td>
                       <td>
                        lattice
                       </td>
                       <td>
                        0.20.45
                       </td>
                       <td>
                        0.20-45
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lattice
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lattice
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-22
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        latticeExtra
                       </td>
                       <td>
                        latticeExtra
                       </td>
                       <td>
                        0.6.29
                       </td>
                       <td>
                        0.6-29
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/latticeExtra
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/latticeExtra
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-12-19
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        lazyeval
                       </td>
                       <td>
                        lazyeval
                       </td>
                       <td>
                        0.2.2
                       </td>
                       <td>
                        0.2.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lazyeval
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lazyeval
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-03-15
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        lifecycle
                       </td>
                       <td>
                        lifecycle
                       </td>
                       <td>
                        1.0.1
                       </td>
                       <td>
                        1.0.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lifecycle
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/lifecycle
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-24
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        limma
                       </td>
                       <td>
                        limma
                       </td>
                       <td>
                        3.50.0
                       </td>
                       <td>
                        3.50.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/limma
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/limma
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        locfit
                       </td>
                       <td>
                        locfit
                       </td>
                       <td>
                        1.5.9.4
                       </td>
                       <td>
                        1.5-9.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/locfit
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/locfit
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-03-25
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        magick
                       </td>
                       <td>
                        magick
                       </td>
                       <td>
                        2.7.3
                       </td>
                       <td>
                        2.7.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/magick
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/magick
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-08-18
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        magrittr
                       </td>
                       <td>
                        magrittr
                       </td>
                       <td>
                        2.0.1
                       </td>
                       <td>
                        2.0.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/magrittr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/magrittr
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-11-17
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        MASS
                       </td>
                       <td>
                        MASS
                       </td>
                       <td>
                        7.3.54
                       </td>
                       <td>
                        7.3-54
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/MASS
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/MASS
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-05-03
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Matrix
                       </td>
                       <td>
                        Matrix
                       </td>
                       <td>
                        1.3.4
                       </td>
                       <td>
                        1.3-4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Matrix
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Matrix
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-06-01
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        MatrixGenerics
                       </td>
                       <td>
                        MatrixGenerics
                       </td>
                       <td>
                        1.6.0
                       </td>
                       <td>
                        1.6.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/MatrixGenerics
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/MatrixGenerics
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        matrixStats
                       </td>
                       <td>
                        matrixStats
                       </td>
                       <td>
                        0.61.0
                       </td>
                       <td>
                        0.61.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/matrixStats
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/matrixStats
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-17
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        mclust
                       </td>
                       <td>
                        mclust
                       </td>
                       <td>
                        5.4.8
                       </td>
                       <td>
                        5.4.8
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mclust
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mclust
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-05
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        memoise
                       </td>
                       <td>
                        memoise
                       </td>
                       <td>
                        2.0.1
                       </td>
                       <td>
                        2.0.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/memoise
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/memoise
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-26
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        mgcv
                       </td>
                       <td>
                        mgcv
                       </td>
                       <td>
                        1.8.38
                       </td>
                       <td>
                        1.8-38
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mgcv
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mgcv
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-06
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        mime
                       </td>
                       <td>
                        mime
                       </td>
                       <td>
                        0.12
                       </td>
                       <td>
                        0.12
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mime
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/mime
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-28
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        minfi
                       </td>
                       <td>
                        minfi
                       </td>
                       <td>
                        1.40.0
                       </td>
                       <td>
                        1.40.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/minfi
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/minfi
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        monaLisa
                       </td>
                       <td>
                        monaLisa
                       </td>
                       <td>
                        1.0.0
                       </td>
                       <td>
                        1.0.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/monaLisa
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/monaLisa
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        multtest
                       </td>
                       <td>
                        multtest
                       </td>
                       <td>
                        2.50.0
                       </td>
                       <td>
                        2.50.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/multtest
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/multtest
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        munsell
                       </td>
                       <td>
                        munsell
                       </td>
                       <td>
                        0.5.0
                       </td>
                       <td>
                        0.5.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/munsell
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/munsell
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2018-06-12
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        nlme
                       </td>
                       <td>
                        nlme
                       </td>
                       <td>
                        3.1.153
                       </td>
                       <td>
                        3.1-153
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nlme
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nlme
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-07
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        nnet
                       </td>
                       <td>
                        nnet
                       </td>
                       <td>
                        7.3.16
                       </td>
                       <td>
                        7.3-16
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nnet
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nnet
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-05-03
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        nor1mix
                       </td>
                       <td>
                        nor1mix
                       </td>
                       <td>
                        1.3.0
                       </td>
                       <td>
                        1.3-0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nor1mix
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/nor1mix
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-06-13
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        openssl
                       </td>
                       <td>
                        openssl
                       </td>
                       <td>
                        1.4.5
                       </td>
                       <td>
                        1.4.5
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/openssl
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/openssl
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-02
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        openxlsx
                       </td>
                       <td>
                        openxlsx
                       </td>
                       <td>
                        4.2.4
                       </td>
                       <td>
                        4.2.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/openxlsx
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/openxlsx
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-06-16
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        pacman
                       </td>
                       <td>
                        pacman
                       </td>
                       <td>
                        0.5.1
                       </td>
                       <td>
                        0.5.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pacman
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pacman
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-03-11
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        pheatmap
                       </td>
                       <td>
                        pheatmap
                       </td>
                       <td>
                        1.0.12
                       </td>
                       <td>
                        1.0.12
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pheatmap
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pheatmap
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-01-04
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        pillar
                       </td>
                       <td>
                        pillar
                       </td>
                       <td>
                        1.6.4
                       </td>
                       <td>
                        1.6.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pillar
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pillar
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-18
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        pkgbuild
                       </td>
                       <td>
                        pkgbuild
                       </td>
                       <td>
                        1.2.1
                       </td>
                       <td>
                        1.2.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgbuild
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgbuild
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        pkgconfig
                       </td>
                       <td>
                        pkgconfig
                       </td>
                       <td>
                        2.0.3
                       </td>
                       <td>
                        2.0.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgconfig
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgconfig
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-09-22
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        pkgload
                       </td>
                       <td>
                        pkgload
                       </td>
                       <td>
                        1.2.4
                       </td>
                       <td>
                        1.2.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgload
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pkgload
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        plotly
                       </td>
                       <td>
                        plotly
                       </td>
                       <td>
                        4.10.0
                       </td>
                       <td>
                        4.10.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/plotly
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/plotly
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-09
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        plyr
                       </td>
                       <td>
                        plyr
                       </td>
                       <td>
                        1.8.6
                       </td>
                       <td>
                        1.8.6
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/plyr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/plyr
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-03-03
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        png
                       </td>
                       <td>
                        png
                       </td>
                       <td>
                        0.1.7
                       </td>
                       <td>
                        0.1-7
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/png
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/png
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2013-12-03
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        poweRlaw
                       </td>
                       <td>
                        poweRlaw
                       </td>
                       <td>
                        0.70.6
                       </td>
                       <td>
                        0.70.6
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/poweRlaw
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/poweRlaw
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-04-25
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        pracma
                       </td>
                       <td>
                        pracma
                       </td>
                       <td>
                        2.3.3
                       </td>
                       <td>
                        2.3.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pracma
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/pracma
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-01-23
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        preprocessCore
                       </td>
                       <td>
                        preprocessCore
                       </td>
                       <td>
                        1.56.0
                       </td>
                       <td>
                        1.56.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/preprocessCore
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/preprocessCore
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        prettyunits
                       </td>
                       <td>
                        prettyunits
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/prettyunits
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/prettyunits
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-01-24
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        processx
                       </td>
                       <td>
                        processx
                       </td>
                       <td>
                        3.5.2
                       </td>
                       <td>
                        3.5.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/processx
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/processx
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-04-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        progress
                       </td>
                       <td>
                        progress
                       </td>
                       <td>
                        1.2.2
                       </td>
                       <td>
                        1.2.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/progress
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/progress
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-05-16
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        promises
                       </td>
                       <td>
                        promises
                       </td>
                       <td>
                        1.2.0.1
                       </td>
                       <td>
                        1.2.0.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/promises
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/promises
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-02-11
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ps
                       </td>
                       <td>
                        ps
                       </td>
                       <td>
                        1.6.0
                       </td>
                       <td>
                        1.6.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ps
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/ps
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-02-28
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        purrr
                       </td>
                       <td>
                        purrr
                       </td>
                       <td>
                        0.3.4
                       </td>
                       <td>
                        0.3.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/purrr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/purrr
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-04-17
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        quadprog
                       </td>
                       <td>
                        quadprog
                       </td>
                       <td>
                        1.5.8
                       </td>
                       <td>
                        1.5-8
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/quadprog
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/quadprog
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-11-20
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        R.methodsS3
                       </td>
                       <td>
                        R.methodsS3
                       </td>
                       <td>
                        1.8.1
                       </td>
                       <td>
                        1.8.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.methodsS3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.methodsS3
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-08-26
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        R.oo
                       </td>
                       <td>
                        R.oo
                       </td>
                       <td>
                        1.24.0
                       </td>
                       <td>
                        1.24.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.oo
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.oo
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-08-26
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        R.utils
                       </td>
                       <td>
                        R.utils
                       </td>
                       <td>
                        2.11.0
                       </td>
                       <td>
                        2.11.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.utils
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R.utils
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-26
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        R6
                       </td>
                       <td>
                        R6
                       </td>
                       <td>
                        2.5.1
                       </td>
                       <td>
                        2.5.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R6
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/R6
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-08-19
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rappdirs
                       </td>
                       <td>
                        rappdirs
                       </td>
                       <td>
                        0.3.3
                       </td>
                       <td>
                        0.3.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rappdirs
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rappdirs
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-01-31
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RColorBrewer
                       </td>
                       <td>
                        RColorBrewer
                       </td>
                       <td>
                        1.1.2
                       </td>
                       <td>
                        1.1-2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RColorBrewer
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RColorBrewer
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2014-12-07
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Rcpp
                       </td>
                       <td>
                        Rcpp
                       </td>
                       <td>
                        1.0.7
                       </td>
                       <td>
                        1.0.7
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rcpp
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rcpp
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-07-07
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RCurl
                       </td>
                       <td>
                        RCurl
                       </td>
                       <td>
                        1.98.1.5
                       </td>
                       <td>
                        1.98-1.5
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RCurl
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RCurl
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-17
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        readr
                       </td>
                       <td>
                        readr
                       </td>
                       <td>
                        2.1.1
                       </td>
                       <td>
                        2.1.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/readr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/readr
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        readxl
                       </td>
                       <td>
                        readxl
                       </td>
                       <td>
                        1.3.1
                       </td>
                       <td>
                        1.3.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/readxl
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/readxl
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-03-13
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        remotes
                       </td>
                       <td>
                        remotes
                       </td>
                       <td>
                        2.4.2
                       </td>
                       <td>
                        2.4.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/remotes
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/remotes
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        reshape
                       </td>
                       <td>
                        reshape
                       </td>
                       <td>
                        0.8.8
                       </td>
                       <td>
                        0.8.8
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/reshape
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/reshape
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2018-10-23
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        reshape2
                       </td>
                       <td>
                        reshape2
                       </td>
                       <td>
                        1.4.4
                       </td>
                       <td>
                        1.4.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/reshape2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/reshape2
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-04-09
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        restfulr
                       </td>
                       <td>
                        restfulr
                       </td>
                       <td>
                        0.0.13
                       </td>
                       <td>
                        0.0.13
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/restfulr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/restfulr
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2017-08-06
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rhdf5
                       </td>
                       <td>
                        rhdf5
                       </td>
                       <td>
                        2.38.0
                       </td>
                       <td>
                        2.38.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rhdf5
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rhdf5
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rhdf5filters
                       </td>
                       <td>
                        rhdf5filters
                       </td>
                       <td>
                        1.6.0
                       </td>
                       <td>
                        1.6.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rhdf5filters
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rhdf5filters
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Rhdf5lib
                       </td>
                       <td>
                        Rhdf5lib
                       </td>
                       <td>
                        1.16.0
                       </td>
                       <td>
                        1.16.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rhdf5lib
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rhdf5lib
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rjson
                       </td>
                       <td>
                        rjson
                       </td>
                       <td>
                        0.2.20
                       </td>
                       <td>
                        0.2.20
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rjson
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rjson
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2018-06-08
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rlang
                       </td>
                       <td>
                        rlang
                       </td>
                       <td>
                        0.4.12
                       </td>
                       <td>
                        0.4.12
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rlang
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rlang
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-18
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rmarkdown
                       </td>
                       <td>
                        rmarkdown
                       </td>
                       <td>
                        2.11
                       </td>
                       <td>
                        2.11
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rmarkdown
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rmarkdown
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-14
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rngtools
                       </td>
                       <td>
                        rngtools
                       </td>
                       <td>
                        1.5.2
                       </td>
                       <td>
                        1.5.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rngtools
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rngtools
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-20
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rpart
                       </td>
                       <td>
                        rpart
                       </td>
                       <td>
                        4.1.15
                       </td>
                       <td>
                        4.1-15
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rpart
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rpart
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-04-12
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rprojroot
                       </td>
                       <td>
                        rprojroot
                       </td>
                       <td>
                        2.0.2
                       </td>
                       <td>
                        2.0.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rprojroot
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rprojroot
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-11-15
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Rsamtools
                       </td>
                       <td>
                        Rsamtools
                       </td>
                       <td>
                        2.10.0
                       </td>
                       <td>
                        2.10.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rsamtools
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rsamtools
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RSQLite
                       </td>
                       <td>
                        RSQLite
                       </td>
                       <td>
                        2.2.8
                       </td>
                       <td>
                        2.2.8
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RSQLite
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/RSQLite
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-08-21
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rstatix
                       </td>
                       <td>
                        rstatix
                       </td>
                       <td>
                        0.7.0
                       </td>
                       <td>
                        0.7.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rstatix
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rstatix
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-02-13
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rstudioapi
                       </td>
                       <td>
                        rstudioapi
                       </td>
                       <td>
                        0.13
                       </td>
                       <td>
                        0.13
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rstudioapi
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rstudioapi
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-11-12
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rtracklayer
                       </td>
                       <td>
                        rtracklayer
                       </td>
                       <td>
                        1.54.0
                       </td>
                       <td>
                        1.54.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rtracklayer
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rtracklayer
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        rvest
                       </td>
                       <td>
                        rvest
                       </td>
                       <td>
                        1.0.2
                       </td>
                       <td>
                        1.0.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rvest
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rvest
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-16
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        S4Vectors
                       </td>
                       <td>
                        S4Vectors
                       </td>
                       <td>
                        0.32.3
                       </td>
                       <td>
                        0.32.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/S4Vectors
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/S4Vectors
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-21
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        sass
                       </td>
                       <td>
                        sass
                       </td>
                       <td>
                        0.4.0
                       </td>
                       <td>
                        0.4.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sass
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sass
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-05-12
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        scales
                       </td>
                       <td>
                        scales
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/scales
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/scales
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-05-11
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        scrime
                       </td>
                       <td>
                        scrime
                       </td>
                       <td>
                        1.3.5
                       </td>
                       <td>
                        1.3.5
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/scrime
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/scrime
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2018-12-01
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        seqLogo
                       </td>
                       <td>
                        seqLogo
                       </td>
                       <td>
                        1.60.0
                       </td>
                       <td>
                        1.60.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/seqLogo
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/seqLogo
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        sessioninfo
                       </td>
                       <td>
                        sessioninfo
                       </td>
                       <td>
                        1.2.1
                       </td>
                       <td>
                        1.2.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sessioninfo
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sessioninfo
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-02
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        shape
                       </td>
                       <td>
                        shape
                       </td>
                       <td>
                        1.4.6
                       </td>
                       <td>
                        1.4.6
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shape
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shape
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-05-19
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        shiny
                       </td>
                       <td>
                        shiny
                       </td>
                       <td>
                        1.7.1
                       </td>
                       <td>
                        1.7.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shiny
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shiny
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-02
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        shinyMethyl
                       </td>
                       <td>
                        shinyMethyl
                       </td>
                       <td>
                        1.30.0
                       </td>
                       <td>
                        1.30.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shinyMethyl
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/shinyMethyl
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        siggenes
                       </td>
                       <td>
                        siggenes
                       </td>
                       <td>
                        1.68.0
                       </td>
                       <td>
                        1.68.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/siggenes
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/siggenes
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        sm
                       </td>
                       <td>
                        sm
                       </td>
                       <td>
                        2.2.5.7
                       </td>
                       <td>
                        2.2-5.7
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sm
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sm
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-13
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        sparseMatrixStats
                       </td>
                       <td>
                        sparseMatrixStats
                       </td>
                       <td>
                        1.6.0
                       </td>
                       <td>
                        1.6.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sparseMatrixStats
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/sparseMatrixStats
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        stabs
                       </td>
                       <td>
                        stabs
                       </td>
                       <td>
                        0.6.4
                       </td>
                       <td>
                        0.6-4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stabs
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stabs
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-01-29
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        stringi
                       </td>
                       <td>
                        stringi
                       </td>
                       <td>
                        1.7.6
                       </td>
                       <td>
                        1.7.6
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stringi
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stringi
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-29
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        stringr
                       </td>
                       <td>
                        stringr
                       </td>
                       <td>
                        1.4.0
                       </td>
                       <td>
                        1.4.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stringr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/stringr
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-02-10
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SummarizedExperiment
                       </td>
                       <td>
                        SummarizedExperiment
                       </td>
                       <td>
                        1.24.0
                       </td>
                       <td>
                        1.24.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/SummarizedExperiment
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/SummarizedExperiment
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        survival
                       </td>
                       <td>
                        survival
                       </td>
                       <td>
                        3.2.13
                       </td>
                       <td>
                        3.2-13
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/survival
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/survival
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-08-24
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        svglite
                       </td>
                       <td>
                        svglite
                       </td>
                       <td>
                        2.0.0
                       </td>
                       <td>
                        2.0.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/svglite
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/svglite
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-02-20
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        systemfonts
                       </td>
                       <td>
                        systemfonts
                       </td>
                       <td>
                        1.0.3
                       </td>
                       <td>
                        1.0.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/systemfonts
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/systemfonts
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-13
                       </td>
                       <td>
                        CRAN (R 4.1.2)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        testthat
                       </td>
                       <td>
                        testthat
                       </td>
                       <td>
                        3.1.1
                       </td>
                       <td>
                        3.1.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/testthat
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/testthat
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-12-03
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        TFBSTools
                       </td>
                       <td>
                        TFBSTools
                       </td>
                       <td>
                        1.32.0
                       </td>
                       <td>
                        1.32.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/TFBSTools
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/TFBSTools
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        TFMPvalue
                       </td>
                       <td>
                        TFMPvalue
                       </td>
                       <td>
                        0.0.8
                       </td>
                       <td>
                        0.0.8
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/TFMPvalue
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/TFMPvalue
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2018-05-16
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        tibble
                       </td>
                       <td>
                        tibble
                       </td>
                       <td>
                        3.1.6
                       </td>
                       <td>
                        3.1.6
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tibble
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tibble
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-07
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        tidyr
                       </td>
                       <td>
                        tidyr
                       </td>
                       <td>
                        1.1.4
                       </td>
                       <td>
                        1.1.4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tidyr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tidyr
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-27
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        tidyselect
                       </td>
                       <td>
                        tidyselect
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        1.1.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tidyselect
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tidyselect
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-04-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        tzdb
                       </td>
                       <td>
                        tzdb
                       </td>
                       <td>
                        0.2.0
                       </td>
                       <td>
                        0.2.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tzdb
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/tzdb
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-27
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        usethis
                       </td>
                       <td>
                        usethis
                       </td>
                       <td>
                        2.1.3
                       </td>
                       <td>
                        2.1.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/usethis
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/usethis
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-27
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        utf8
                       </td>
                       <td>
                        utf8
                       </td>
                       <td>
                        1.2.2
                       </td>
                       <td>
                        1.2.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/utf8
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/utf8
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-07-24
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        vctrs
                       </td>
                       <td>
                        vctrs
                       </td>
                       <td>
                        0.3.8
                       </td>
                       <td>
                        0.3.8
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/vctrs
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/vctrs
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-04-29
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        VennDiagram
                       </td>
                       <td>
                        VennDiagram
                       </td>
                       <td>
                        1.7.1
                       </td>
                       <td>
                        1.7.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/VennDiagram
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/VennDiagram
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-12-02
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        vioplot
                       </td>
                       <td>
                        vioplot
                       </td>
                       <td>
                        0.3.7
                       </td>
                       <td>
                        0.3.7
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/vioplot
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/vioplot
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-07-27
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        viridisLite
                       </td>
                       <td>
                        viridisLite
                       </td>
                       <td>
                        0.4.0
                       </td>
                       <td>
                        0.4.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/viridisLite
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/viridisLite
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-04-13
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        webshot
                       </td>
                       <td>
                        webshot
                       </td>
                       <td>
                        0.5.2
                       </td>
                       <td>
                        0.5.2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/webshot
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/webshot
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-11-22
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        wesanderson
                       </td>
                       <td>
                        wesanderson
                       </td>
                       <td>
                        0.3.6
                       </td>
                       <td>
                        0.3.6
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/wesanderson
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/wesanderson
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2018-04-20
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        withr
                       </td>
                       <td>
                        withr
                       </td>
                       <td>
                        2.4.3
                       </td>
                       <td>
                        2.4.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/withr
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/withr
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        xfun
                       </td>
                       <td>
                        xfun
                       </td>
                       <td>
                        0.28
                       </td>
                       <td>
                        0.28
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xfun
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xfun
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-04
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        XML
                       </td>
                       <td>
                        XML
                       </td>
                       <td>
                        3.99.0.8
                       </td>
                       <td>
                        3.99-0.8
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/XML
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/XML
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-09-17
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        xml2
                       </td>
                       <td>
                        xml2
                       </td>
                       <td>
                        1.3.3
                       </td>
                       <td>
                        1.3.3
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xml2
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xml2
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-11-30
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        xtable
                       </td>
                       <td>
                        xtable
                       </td>
                       <td>
                        1.8.4
                       </td>
                       <td>
                        1.8-4
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xtable
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/xtable
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2019-04-21
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        XVector
                       </td>
                       <td>
                        XVector
                       </td>
                       <td>
                        0.34.0
                       </td>
                       <td>
                        0.34.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/XVector
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/XVector
                       </td>
                       <td>
                        TRUE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        yaml
                       </td>
                       <td>
                        yaml
                       </td>
                       <td>
                        2.2.1
                       </td>
                       <td>
                        2.2.1
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/yaml
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/yaml
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2020-02-01
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        zip
                       </td>
                       <td>
                        zip
                       </td>
                       <td>
                        2.2.0
                       </td>
                       <td>
                        2.2.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zip
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zip
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-05-31
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        zlibbioc
                       </td>
                       <td>
                        zlibbioc
                       </td>
                       <td>
                        1.40.0
                       </td>
                       <td>
                        1.40.0
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zlibbioc
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zlibbioc
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-10-26
                       </td>
                       <td>
                        Bioconductor
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                      <tr>
                       <td>
                        zoo
                       </td>
                       <td>
                        zoo
                       </td>
                       <td>
                        1.8.9
                       </td>
                       <td>
                        1.8-9
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zoo
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library/zoo
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        FALSE
                       </td>
                       <td>
                        2021-03-09
                       </td>
                       <td>
                        CRAN (R 4.1.0)
                       </td>
                       <td>
                       </td>
                       <td>
                        /Library/Frameworks/R.framework/Versions/4.1/Resources/library
                       </td>
                      </tr>
                     </tbody>
                    </table>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="r8L6eJ1IF5yh1queShEHGpGQ5jBuyaS7DZ3ApFnCMaaZ5+Cbz0AnX8rPU1ClFfhxqvLWdgmWW0jHrntxHv4E8w==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <div class="f4 my-3 color-fg-muted text-italic">
                 No description, website, or topics provided.
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <h3 class="sr-only">
                 License
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="#MIT-1-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                   </path>
                  </svg>
                  MIT license
                 </a>
                </div>
                <include-fragment src="/mepepin/Indirect.Epigenomics/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/mepepin/Indirect.Epigenomics/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/mepepin/Indirect.Epigenomics/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/mepepin/Indirect.Epigenomics/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/mepepin/Indirect.Epigenomics/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fmepepin%2FIndirect.Epigenomics&amp;report=mepepin+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/mepepin/Indirect.Epigenomics/releases">
                 Releases
                 <span class="Counter" data-view-component="true" title="1">
                  1
                 </span>
                </a>
               </h2>
               <a class="Link--primary d-flex no-underline" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/mepepin/Indirect.Epigenomics/releases/tag/IndirectEpigenomics">
                <svg aria-hidden="true" class="octicon octicon-tag flex-shrink-0 mt-1 color-fg-success" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                 </path>
                </svg>
                <div class="ml-2 min-width-0">
                 <div class="d-flex">
                  <span class="css-truncate css-truncate-target text-bold mr-2" style="max-width: none;">
                   Indirect.Cardiomyocyte.Epigenomics
                  </span>
                  <span class="Label Label--success flex-shrink-0" data-view-component="true" title="Label: Latest">
                   Latest
                  </span>
                 </div>
                 <div class="text-small color-fg-muted">
                  <relative-time class="no-wrap" datetime="2022-01-17T14:51:35Z">
                   Jan 17, 2022
                  </relative-time>
                 </div>
                </div>
               </a>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/mepepin/packages?repo_name=Indirect.Epigenomics">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/mepepin/Indirect.Epigenomics/used_by_list">
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
