<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-6a292b1161ed.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics: Streaming workflow for phylogenomic analysis and comparative proteomics
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:1075038:17F0437:6605BB48" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDoxMDc1MDM4OjE3RjA0Mzc6NjYwNUJCNDgiLCJ2aXNpdG9yX2lkIjoiNjQ1MjI0MTg1NjIzODgxNzgwNyIsInJlZ2lvbl9lZGdlIjoiaWFkIiwicmVnaW9uX3JlbmRlciI6ImlhZCJ9" data-pjax-transient="true" name="visitor-payload">
           <meta content="a7afc4eb2dbf24f0f89dd7e71de1739a8bdcae6a04c97841e9f5a4dc03e55dce" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:485801382" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="Streaming workflow for phylogenomic analysis and comparative proteomics - qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/0322208afbb43a856b0c2fb0872aa1b87320df40fe3d8f1a3126a77d793c00dc/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics: Streaming workflow for phylogenomic analysis and comparative proteomics" name="twitter:title">
                    <meta content="Streaming workflow for phylogenomic analysis and comparative proteomics - qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/0322208afbb43a856b0c2fb0872aa1b87320df40fe3d8f1a3126a77d793c00dc/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" property="og:image">
                      <meta content="Streaming workflow for phylogenomic analysis and comparative proteomics - qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics: Streaming workflow for phylogenomic analysis and comparative proteomics" property="og:title"/>
                        <meta content="https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" property="og:url"/>
                        <meta content="Streaming workflow for phylogenomic analysis and comparative proteomics - qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="de8e1521fc37a178caaed17097d0545118b4eb0f01833d1d1eeeb65ac20b6ca7" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="1dd47e41bae3fb619dfa564356c6dee453350e97ff939ad02473ed6906f90b45" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics git https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics.git" name="go-import"/>
                        <meta content="20830212" name="octolytics-dimension-user_id">
                         <meta content="qingxiangguo" name="octolytics-dimension-user_login">
                          <meta content="485801382" name="octolytics-dimension-repository_id">
                           <meta content="qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="485801382" name="octolytics-dimension-repository_network_root_id">
                               <meta content="qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-11f6759e1cef.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics","user_id":null}}' data-hydro-click-hmac="1e92700e5270838cf036ef58d2f3c966ab151a5405430ab2507178b1c6f60dad" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fqingxiangguo%2FPhylogenomic_analysis_and_comparative_proteomics">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=H52ZCN%2BolmDlZFFrkjWcZxDFmi8iU%2Bk5Ir891Anuh4OyN4k7jmzXJPrqPn1Hdy77zCvK%2BW34I4wrps6I6CQqH4vmKE4YYlOSgf0DpyJUuMHDuU4yhF0C8RylIYY3UThIPn0EMm7qaTeHj6yvrdRMPi4VOvdjI%2BltkYe9f6sURmBGYKmRuX%2BOkNEqwa%2FdonditAI5HCcyrCunRFi2USjZtswl5b1bSiLDjmZ61XZIr%2BK9OcyGhw7NQ4NfQRciwqVmVJAZXwoF8WuHd5pJrqK8zfDUnGq67iCqOE6yIMzaE18ZfZYj6tqcppMfGPAFaQd3B%2FJc7hHJUsNi68GlP6qPZ1MiIO6pXoqZnR4KfrJKkLZbsPDmak68XKXrNlCCE6fv7UQRe426kHCQDvoxyb7cNlNPLuv%2BM%2BT7K%2BoD2HcjeVIAR%2FZw1O5uQ0fvDxID2knJkT6M%2Fo4s8QBb7pjbHfTQ6qQVKMhoGXvoXkP2irb825y%2BuTnpHhlvnJU3YmmqpjVLCCEwfJ6oWgr0WwoTOkZzYWmkXvRECfwMOOAmmD7oYC%2BrcuNHVO%2FAq9HNHflO3rvL4pNiaCPWv5OpMox8HFjjTyyVbzMfrTCZne0%3D--VBNb%2FR7wlb4EYtei--oZcuMvRI1qinxaaXYOoIVQ%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="qingxiangguo" data-current-repository="qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="E8ZtJoggpnyi4A5i4XjITYT6Pr_GekXbvthrcbI-bJD2eZeMn1FMTE6Zj-p3xEtEV6-2On_hkbl50Kb2yyhRcQ" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-2bf15988-649a-4070-b015-b11b83eec02b" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-2bf15988-649a-4070-b015-b11b83eec02b">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="DTcTdkVDhkCaMRrQ/M9va0ehGqtdBHHlBGLNz/E5lNCz8waAOkygO9Awe6AtK7Zz9h3BkdofMTVNKylKIkf94A=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="wsDMtQeWk3VMJKAmQKiz2VchB8iqLvtQjHDgPyfv0C1IR2LpB7wuogCcj815fe0BkknkxqiWqayr1pVkSXx7wA==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="K0OCetbdWS9FtI0Hxs4MAOuG699j2vb98aZKiMb7GS1GjSom9C+qTc9Z/f1gnNoOP90spO7kxHchiigvu75bTg==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="vugr/LXFeYMScRJlgHOD+ReWCcr49egZzwco4oBaPJdI9e6aSVZJ/dfcvPwyn/fRodaB41BoZ1LRW5PYSyaQhw==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics","user_id":null}}' data-hydro-click-hmac="1e92700e5270838cf036ef58d2f3c966ab151a5405430ab2507178b1c6f60dad" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fqingxiangguo%2FPhylogenomic_analysis_and_comparative_proteomics">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics","user_id":null}}' data-hydro-click-hmac="1e92700e5270838cf036ef58d2f3c966ab151a5405430ab2507178b1c6f60dad" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=qingxiangguo%2FPhylogenomic_analysis_and_comparative_proteomics">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-3561c90b-fdf4-4f2d-98d0-20691eb8071d" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-47734b6c-8c16-42df-992b-cf4ab2d7a09c" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-47734b6c-8c16-42df-992b-cf4ab2d7a09c" id="tooltip-3561c90b-fdf4-4f2d-98d0-20691eb8071d" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/qingxiangguo/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/qingxiangguo" rel="author">
            qingxiangguo
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics">
            Phylogenomic_analysis_and_comparative_proteomics
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics","user_id":null}}' data-hydro-click-hmac="e6af36f8fa96eb3f436b55149401409e5b0b6990e216f4ce724f1c4496a4de5d" data-view-component="true" href="/login?return_to=%2Fqingxiangguo%2FPhylogenomic_analysis_and_comparative_proteomics" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":485801382,"auth_type":"LOG_IN","originating_url":"https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics","user_id":null}}' data-hydro-click-hmac="ae15751a05f534896b5559ebf0be779e48f7411e1be2dd57022987d5506acd32" data-view-component="true" href="/login?return_to=%2Fqingxiangguo%2FPhylogenomic_analysis_and_comparative_proteomics" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
             0
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":485801382,"auth_type":"LOG_IN","originating_url":"https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics","user_id":null}}' data-hydro-click-hmac="a7356ee134578d963582ed23d2a01a2c0604455044879174a15db4c5b33da5bb" data-view-component="true" href="/login?return_to=%2Fqingxiangguo%2FPhylogenomic_analysis_and_comparative_proteomics" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
              0
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <p class="f4 mb-3">
          Streaming workflow for phylogenomic analysis and comparative proteomics
         </p>
         <h3 class="sr-only">
          License
         </h3>
         <div class="mb-2">
          <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/blob/main/LICENSE">
           <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
            </path>
           </svg>
           MIT license
          </a>
         </div>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           stars
          </a>
          <a class="Link--secondary no-underline mr-3" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":485801382,"auth_type":"LOG_IN","originating_url":"https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics","user_id":null}}' data-hydro-click-hmac="a7356ee134578d963582ed23d2a01a2c0604455044879174a15db4c5b33da5bb" data-view-component="true" href="/login?return_to=%2Fqingxiangguo%2FPhylogenomic_analysis_and_comparative_proteomics" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics","user_id":null}}' data-hydro-click-hmac="e6af36f8fa96eb3f436b55149401409e5b0b6990e216f4ce724f1c4496a4de5d" data-view-component="true" href="/login?return_to=%2Fqingxiangguo%2FPhylogenomic_analysis_and_comparative_proteomics" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-2d5e3bcd-6125-4c6e-9321-05ecf16a6e3e-list" aria-haspopup="true" aria-labelledby="tooltip-b7deff51-c33d-477d-960d-d7464b8e2daa" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-2d5e3bcd-6125-4c6e-9321-05ecf16a6e3e-button" popovertarget="action-menu-2d5e3bcd-6125-4c6e-9321-05ecf16a6e3e-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-2d5e3bcd-6125-4c6e-9321-05ecf16a6e3e-button" id="tooltip-b7deff51-c33d-477d-960d-d7464b8e2daa" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-2d5e3bcd-6125-4c6e-9321-05ecf16a6e3e-button" anchor-offset="normal" data-view-component="true" id="action-menu-2d5e3bcd-6125-4c6e-9321-05ecf16a6e3e-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-2d5e3bcd-6125-4c6e-9321-05ecf16a6e3e-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-2d5e3bcd-6125-4c6e-9321-05ecf16a6e3e-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics" id="item-cc336e04-8488-4ec0-bd24-a5d80c632a87" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/issues" id="item-7fe6ebba-9025-4ec9-b8ba-4c31e18a8cbb" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/pulls" id="item-49348caa-e2aa-4ccc-800f-86cc8e80f334" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/actions" id="item-9947a3e3-4410-4cc0-8ae0-592bb1133ab5" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/projects" id="item-67f1310e-e0ec-4ec8-a078-ceaa84dc4544" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/security" id="item-bd68baa4-7f98-41f6-b727-7b08a284e860" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/pulse" id="item-ab6e0627-698d-4198-9a71-14d9eabe430d" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/spoofed_commit_check/1d4523ec821e173cde745cea73ca770d68869ff3">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":485801382,"defaultBranch":"main","name":"Phylogenomic_analysis_and_comparative_proteomics","ownerLogin":"qingxiangguo","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2022-04-26T13:36:44.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/20830212?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"main","listCacheKey":"v0:1650980700.061096","canEdit":false,"refType":"branch","currentOid":"1d4523ec821e173cde745cea73ca770d68869ff3"},"tree":{"items":[{"name":"scripts","path":"scripts","contentType":"directory"},{"name":".gitignore","path":".gitignore","contentType":"file"},{"name":"LICENSE","path":"LICENSE","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":4,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/archive/refs/heads/main.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=485801382"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"6","overviewFiles":[{"displayName":"README.md","repoName":"Phylogenomic_analysis_and_comparative_proteomics","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cp dir=\"auto\"\u003e\u003cstrong\u003eStreaming workflow for phylogenomic analysis and comparative proteomics\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eContributors\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eQingxiang Guo\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAbout\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe following contents contains source code for the analyses and plots in my Biology paper, 2022, “Quantitative insights into the contribution of nematocysts to the adaptive success of cnidarians based on proteomic analysis”\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAbstract\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCnidarians (such as corals, anemones, jellyfish) are ancient and successful animals. Previous studies have offered a variety of explanations for the adaptive success of certain cnidarian taxa. However, common strategies for the long-term persistence of cnidarians have not been identified. One factor that may contribute to their evolutionary success of the lineage is the nematocyst, a sting organelle able to deliver venom into prey or enemies. Using bioinformatics analyses, we aimed to quantitatively investigate the role of nematocysts in cnidarian adaptation. We identified the extensive species specific adaptation in nematocyst proteins (NEMs) and demonstrate that both a unique evolutionary pattern of NEMs and the long evolutionary lag between nematocysts and cnidarians support their key adaptive role. Further, we find NEMs experience approximately 50% more adaptive changes on average compared to non-NEMs, and positively selected cnidarian-conserved proteins are enriched in NEMs. These results support a key role of nematocysts in successful cnidarian adaptation and provide a general quantitative framework for assessing the role of a phenotypic novelty in adaptation. Moreover, the findings will be critical for reassessing the evolutionary history of many established models, enhancing our understanding of both the mechanisms and evolutionary preference of adaptive evolution.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNotes\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWritten by Qingxiang Guo, \u003ca href=\"mailto:qingxiang.guo@outlook.com\"\u003eqingxiang.guo@outlook.com\u003c/a\u003e, distributed without any guarantees or restrictions.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eCodes\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1. Orthologous gene family analysis by OrthoMCL\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.1 Disposing the MySQL\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003evim /etc/my.cnf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emyisam_sort_buffer_size=128000M\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eread_buffer_size=2000M\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emysql\u0026gt; show variables LIKE 'myisam_max_sort_file_size';\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emysql\u0026gt; show variables LIKE 'myisam_sort_buffer_size';\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emysql\u0026gt; show variables LIKE 'read_buffer_size';\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Restart\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eservice mysqld restart\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.2 Install MCL\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003etar zxf /opt/software/mcl-14-137.tar.gz -C /opt/biosoft/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./configure --prefix=/opt/biosoft/mcl-14-137/ \u0026amp;\u0026amp; make -j 4 \u0026amp;\u0026amp; make install\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eecho 'PATH=$PATH:/opt/biosoft/mcl-14-137/bin/' \u0026gt;\u0026gt; ~/.bashrc\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esource ~/.bashrc\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.3 Configuration\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e/home/gqx/proteome/tmp_test/ortho_test\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp /opt/biosoft/orthomclSoftware-v2.0.9/doc/OrthoMCLEngine/Main/orthomcl.config.template ./orthomcl.config\u003c/p\u003e\n\u003cp dir=\"auto\"\u003evim orthomcl.config\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Editing the text\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edbConnectString=dbi:mysql:orthomcl\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edbLogin=gqx\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edbPassword=123456\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.4 Install the MySQL tables required for OrthoMCL\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emysql -u gqx -p123456\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCREATE DATABASE IF NOT EXISTS orthomcl;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomclInstallSchema orthomcl.config\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.5 Prepare the input fasta data\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Edit the file name to make it recognizable for OrthoMCL\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -e \"s/_/|/g\" Aurelia_aurita_Rachamim.fasta \u0026gt; fix.fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003etar zxf /home/train/00.incipient_data/data_for_genome_comparison/compliantFasta.tar.gz -C ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomclAdjustFasta ncra ncra_proteins.fasta 1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm ncra_proteins.fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.6 Remove low-quality sequence\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Longer than 30 aa, and the stop codon ratio threshold is 20%\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomclFilterFasta compliantFasta/ 10 20\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.7 All-vs-all BLAST on the filtered result goodProteins.fasta\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emakeblastdb -in goodProteins.fasta -dbtype prot -title orthomcl -parse_seqids -out orthomcl -logfile orthomcl.log\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup blastp -db orthomcl -query goodProteins.fasta -seg yes -out blastresult.tab -evalue 1e-5 -num_threads 64 -outfmt 6 \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.8 Process the results of BLAST to get the similarity information between the sequences, so as to facilitate importing into MySQL\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomclBlastParser blastresult.tab compliantFasta/ \u0026gt; similarSequences.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then set myisam_max_sort_file_size (five times larger) in mysql according to the size of the similarSequences.txt file,\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emyisam_max_sort_file_size=2955M\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/0\\t0/1\\t-181/' similarSequences.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.9 Import similarSequences.txt to MySQL database\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomclLoadBlast orthomcl.config similarSequences.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.10\u003c/strong\u003e \u003cstrong\u003eFind similar sequence pairs: pairs that best match each other between species (orthologs pairs); pairs that best match each other and are better than interspecies matches within species (in-paralogs pairs); the first two are combined into co-ortholog pairs\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomclPairs orthomcl.config orthomcl_pairs.log cleanup=no\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.11 Export similar sequence pairs from MySQL\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomclDumpPairsFiles orthomcl.config\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# output: mclInput, pairs file\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.12 Use mcl to cluster pairs (Ortholog Cluster Groups) and number the classes\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emcl mclInput --abc -I 1.5 -o mclOutput\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomclMclToGroups OCG 1 \u0026lt; mclOutput \u0026gt; groups.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# groups.txt contains the final cluster information\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.13 Manual extraction of cluster results\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomcl_two.pl groups.txt mhon mwul\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomcl_two.pl groups.txt mhon tkit\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomcl_two.pl groups.txt mwul tkit\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomcl_three.pl groups.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.14 Plot three-set Venn diagram of the cluster results\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# First, match each sequence ID to OG in the result\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomcl_result_2_header.pl groups.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd ..\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv 0_orthomcl/header ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Create header file one by one for each species\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egrep \"\u0026gt;\" 0_orthomcl/compliantFasta/cfle.fasta \u0026gt; cfle\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/\u0026gt;//g\" cfle\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then replace the original sequence ID according to the corresponding relationship\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomcl_header_2_OG.pl mhon header\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eRscript three_set_venn.R\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.15 Statistics of the orthologs\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomcl_genes_number_stats.pl groups.txt compliantFasta \u0026gt; genes_number_stats.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.16 Extract the single-copy orthologs from the OrthoMCL results\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eorthomcl_findSingleCopyOrtholog.pl groups.txt compliantFasta/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1.17 Batch alignment of sequences\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efastas2aligned_by_mafft.pl SingleCopyOrthologGroups 8\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# You will get the final allSingleCopyOrthologsAlign.fasta for downstream analysis (e. g., phylogenomics)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2. Orthologous gene family analysis by HaMStR and phylogenomic data matrices construction\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.1 Run HaMStR to get orthologs\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Use model organisms core set\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup hamstr -protein -sequence_file=../ALALA.fasta -taxon=ALALA -hmmset=modelorganisms_hmmer3 -refspec=DROME -relaxed -central -force -cpu 24 -eval_blast=0.01 -eval_hmm=0.01 \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Use basal metazoan core set\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup hamstr -protein -sequence_file=rename_all.fasta -taxon=CEBOR -hmmset=basalmets_hmmer3 -refspec=nemve_2309  -relaxed -central -force -cpu 48 -eval_blast=0.01 -eval_hmm=0.01  \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# To run many species in batch\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eproduce_MODEL_hamstr_cmd_list.pl Ctenactis_echinata\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in */; do produce_MODEL_hamstr_cmd_list.pl $dir; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in */; do produce_BASELMET_hamstr_cmd_list.pl $dir; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.2 Combine different OGs\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egrep --p \"hydma\" basalmets_hmmer3.fa \u0026gt; basl_head\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efast_extract_seq_from_fasta.pl basalmets_hmmer3.fa basl_head \u0026gt; basl.fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat ./* \u0026gt; model.fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv /home/gqx/proteome/10_phylogenomic_tree/species/Hydrozoa/Hydra_vulgaris/model/fa_dir_HYVUL_modelorganisms_hmmer3_DROME_relaxed/model.fasta ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egrep --p \"HYVUL|\" model.fasta \u0026gt; model_head\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efast_extract_seq_from_fasta.pl model.fasta model_head \u0026gt; model2.fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emakeblastdb -in model2.fasta -out model -dbtype prot\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup blastp -query basl.fasta -db model -max_target_seqs 1 -outfmt 6 -out basl_model -evalue 1e-10 -num_threads 32 \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -lane 'print $_ if ($F[2] \u0026gt;= 70)' basl_model \u0026gt; basl_model_70\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Take the species with high coverage as a bridge\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Extracted correspondence\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eextract_OGs_relationship.pl basl_model_70\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat ./connect_by_human/rela ./connect_by_hydra/rela ./connect_by_nemve/rela ./connect_by_Trichoplax/rela \u0026gt; relationship\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eremove_duplicate.pl relationship\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Get the same genes from all species into the same file\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat ./*/met/hamstrsearch_* \u0026gt;\u0026gt; /home/gqx/proteome/10_phylogenomic_tree/my_alignment/met_out\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat ./*/*/model/hamstrsearch_* \u0026gt; /home/gqx/proteome/10_phylogenomic_tree/my_alignment/model_out\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/^(\\d+)|\\w+?|/\\1|/g\" met_out\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/^(\\d+)|\\w+?|/\\1|/g\" model_out\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# The missing reference sequence is added\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egrep --p \"nemve_2309|hydma_2|triad_390\" basalmets_hmmer3.fa \u0026gt; head\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efast_extract_seq_from_fasta.pl basalmets_hmmer3.fa head \u0026gt; haha.fa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/nemve_2309/NEVEC/g\" haha.fa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/hydma_2/HYVUL/g\" haha.fa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/triad_390/TRADH/g\" haha.fa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/^\u0026gt;(\\S+)/\\1|1|/g\" haha.fa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/|\\n/|/g\" haha.fa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat haha.fa \u0026gt;\u0026gt; /home/gqx/proteome/10_phylogenomic_tree/my_alignment/met_out\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Check species\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat model_out | cut -f 2 -d \"|\" \u0026gt; 22\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eremove_duplicate.pl 22\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ewc -l duplicate_remove\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then split out sequences under the same OG\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ephylogenomic_seperate_OGs_1.pl met_out\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ephylogenomic_seperate_OGs_2.pl out\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd OGs\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 'tr/,/\\n/s' ./*\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor DIR in ./*txt; do phylogenomic_seperate_OGs_3.pl $DIR; done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm ./*txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then find the intersection between model core set and metazoan core set\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd met_model_inter_OGs/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp ../met/OGs/* ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp ../model/OGs/* ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003einter_OG_met_model_remove.pl relationship\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm relationship\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.3 Quality control before alignment\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emkdir trim_OGs\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd trim_OGs\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp ../met_model_inter_OGs/* ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Change a more convenient sequence header\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/\u0026gt;\\d+|(\\w+)|(\\w+)|\\d|/\u0026gt;\\1|\\2/g\" ./*fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Delete sequence length shorter than 100 aa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ephylogenomic_delete_short_seq.pl\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Delete OGs with taxon coverage less than specific species (n = 50 in present species)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./phylogenomic_few_tax.sh\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Delete X at the first and last 20 aa of the sequence\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./phylogenomic_remove_X.sh\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.4 Alignment\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e/opt/scripts/fastas2aligned_by_mafft_hamstr_version.pl trim_OGs 16\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.5 Trimming\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in ./*aln; do phylogenomic_trimal_BJZ2X.pl $dir; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm ./*aln\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erename .aln.BJZ2X .aln *.aln.BJZ2X\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in ./*aln; do trimal -in \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"3f6008964ba396a773481e473e5b66f8\"\u003e$dir -out $\u003c/math-renderer\u003e{dir}.trim -automated1; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/^(\u0026gt;\\S+)\\s+$/\\1\\n/\" ./*aln.trim\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./phylogenomic_few_tax_2.sh\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd /home/gqx/proteome/10_phylogenomic_tree/single_gene_tree/alignment\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp ../../my_alignment/trim_OGs/*trim ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erename .aln.trim .fa *.aln.trim\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp /opt/biosoft/PhyloTreePruner/remove_short_seqs.sh ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Change the fasta file to flat format\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in ./*fa; do flat_the_fasta_seq.pl $dir; mv flated $dir.flat; done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm ./*fa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erename .fa.flat .fa *.fa.flat\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Removing sequences that are more than 50 percent gaps\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./remove_short_seqs.sh\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.6 Build single-gene trees using FastTree\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ephylogenomic_FastTree_OG.pl alignment 16\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFastTreeMP -slow -gamma $ARGV[0]/$_ \u0026gt; $ARGV[0]/$1.tre\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.7 Use TreSpEx to remove Long-branch attraction (LBA) and saturation analyses\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd TreSpEx/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emkdir single_tree\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd single_tree\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# TreSpEx needs the RAXML output format\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp /home/gqx/proteome/10_phylogenomic_tree/ML_analysis/76cov_105tx_146og/out.fasta ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp /home/gqx/proteome/10_phylogenomic_tree/ML_analysis/76cov_105tx_146og/new_partition.txt ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp charset new_partition.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/\\s;\\s//g\" new_partition.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/\\s-\\s/-/g\" new_partition.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/charset/AUTO,/g\" new_partition.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/^\\S+\\s//g\" new_partition.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./AMAS.py split -l new_partition.txt -i out.fasta -f fasta -d aa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm out.fasta new_partition.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor DIR in ./*fas; do  mkdir ${DIR}_file; done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor DIR in ./*fas; do  mv \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"3f6008964ba396a773481e473e5b66f8\"\u003e$DIR  $\u003c/math-renderer\u003e{DIR}_file; done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor DIR in ./*file; do  cd $DIR; remove_wenhao_seq.pl *fas; awk 'BEGIN {RS = \"\u0026gt;\" ; FS = \"\\n\" ; ORS = \"\"} $2 {print \"\u0026gt;\"$0}' out \u0026gt; output; rm out; cd ..;  done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor DIR in ./*file; do  cd $DIR; mv output out_*; cd ..;  done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Run iqtree\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup iqtree-omp -s ./*fas -wbt -bb 10000 -nt 1 -st AA \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Prepare the species name list, tree name list and alignment name list\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp /home/gqx/proteome/10_phylogenomic_tree/concatenated/146/otu/otu-name.otu ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003els -1 ./*contree \u0026gt; tree_name\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Make sure the format of alignments is relax phylip\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd /home/gqx/proteome/10_phylogenomic_tree/ML_analysis/76cov_105tx_146og/TreSpEx/Longbranch_and_Saturation\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emkdir align\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd align\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp ../../single_tree/*/*fas ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in ./*fas; do convertFasta2Phylip.sh \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"3f6008964ba396a773481e473e5b66f8\"\u003e$dir \u0026amp;gt; $\u003c/math-renderer\u003e{dir}.phy ; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003els -1 ./*phy \u0026gt; align_name\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv align_name ../\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv ./*phy ../\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd ..\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm align -rf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Perform LBA analysis\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTreSpEx.v1.1.pl -fun e -ipt tree_name -tf otu-name.otu\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Saturation analysis\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTreSpEx.v1.1.pl -fun g -ipt tree_name -ipa align_name\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.7.1 Use density map to remove LBA (in R)\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edata \u0026lt;- read.table(\"LB_scores_summary_perPartition.txt\", header=T,sep=\"\\t\",row.names = 1)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ed \u0026lt;- density(data$LB_score_upper_quartile)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003epdf(\"density_upper.pdf\",width=6,height=6)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eplot(d, main=NA, xlab=\"Mean value of upper quartile of LB scores\", lwd=1.5)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# plot(d, main=NA, xlab=\"Standard deviation of LB scores\", lwd=1.5)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esegments(x0=80.81,y0=0,x1=80.81,y1=0.0125,lwd=1.5, lty=2)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eabline(h=0,col=\"grey\")\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Filling detected areas with red\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ex1 \u0026lt;- min(which(d$x \u0026gt;= 80.81))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ex2 \u0026lt;- max(which(d$x \u0026lt; 140))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ewith(d, polygon(x=c(x[c(x1,x1:x2,x2)]), y= c(0, y[x1:x2], 0), col=\"red\"))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edev.off()\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.7.2 Use density map to do saturation analysis (in R)\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCorrelation_Results/Correlation_Slope_Summary.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edata \u0026lt;- read.table(\"Correlation_Slope_Summary.txt\", header=T,sep=\"\\t\",row.names = 1)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ed \u0026lt;- density(data$Slope)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003epdf(\"density_slope.pdf\",width=6,height=6)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eplot(d, main=NA, xlab=\"Slope of linear regression\", lwd=1.5)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esegments(x0= 0.224,y0=0,x1= 0.224,y1=1.5,lwd=1.5, lty=2)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eabline(h=0,col=\"grey\")\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ex1 \u0026lt;- min(which(d$x \u0026gt;= -1))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ex2 \u0026lt;- max(which(d$x \u0026lt; 0.224))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ewith(d, polygon(x=c(x[c(x1,x1:x2,x2)]), y= c(0, y[x1:x2], 0), col=\"red\"))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edev.off()\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.7.3 Use heatmap to analyze species-specific genes (in R)\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003elibrary(gplots)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edata \u0026lt;- read.table(\"9999.txt\", header=T,sep=\"\\t\",row.names = 1)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edata \u0026lt;- as.matrix(data)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecolors = c(seq(min(data),0,length=100),seq(1,200,length=100),seq(201,500,length=100),seq(501,700,length=100),seq(701,1000,length=100),seq(1001,1200,length=100),seq(1201,1600,length=100),seq(9999,9999,1))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emy_palette \u0026lt;- colorRampPalette(c(\"orange\",\"yellow\",\"green\", \"cyan\",\"blue\",\"purple\",\"red\",\"white\"))(n = 700)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003epdf(\"9999.pdf\",width=18,height=18)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eheatmap.2(data, density.info=\"density\", col=my_palette, symkey=F, symm=F, symbreaks=F,scale=\"none\", trace=\"none\", breaks=colors, keysize=1)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edev.off()\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.8 Use PhyloTreePruner to remove paralogs\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Change the header to be recognizable for PhyloTreePruner\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/(\\d+)|//g\" 1.aln.trim\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Feed the result of RAxML (RAxML_bipartitions.out) to PhyloTreePruner\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ejava -cp /usr/local/bin/ PhyloTreePruner RAxML_bipartitions.out 3 2.aln.trim 0.5 r\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ejava -cp /usr/local/bin/ PhyloTreePruner 113684.tre 71 113684.aln 0.5 r\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd single_gene_tree/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emkdir paralog_screen\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd paralog_screen/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp ../alignment/*long ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp ../alignment/*tre ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Change the fasta file format to flat fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in ./*fa.long; do fasta_flat_to_multiline.py $dir 60; done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm ./*fa.long\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erename .fa.long_multi-line.fasta  .fa.long *.fa.long_multi-line.fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd ..\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Run PhyloTreePruner\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ephylogenomic_PhyloTreePruner.pl paralog_screen 8\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.9 Filtering for unique sequences\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSCaFoS\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eexport SCAFOS=/opt/biosoft/SCaFoS\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecpan install Tk\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Install Tree-Puzzle\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/|/@/g\" ./*fa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003escafos in=fasta out=otu\u003c/p\u003e\n\u003cp dir=\"auto\"\u003escafos in=fasta out=out otu=otu/otu-name.otu format=fp t=10\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# out.fasta is the result\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Change the style to MARE style\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ephylogenomic_scafos2MARE_charset.pl out.len\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.10 Visualize the phylogenomic matrices\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMARE charset out.fasta -m -t\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2.11 Exclude species with gene missing rate \u0026gt; 85%\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd /home/gqx/proteome/10_phylogenomic_tree/concatenated/50_taxon_coverage_remove_poor_species/out\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emkdir count\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd count\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esplit_fasta_file_averagely.pl ../out.fasta 118\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm out.fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor DIR in ./*fasta; do  mkdir ${DIR}_file; done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor DIR in ./*fasta; do  mv \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"3f6008964ba396a773481e473e5b66f8\"\u003e$DIR  $\u003c/math-renderer\u003e{DIR}_file; done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor DIR in ./*; do cd $DIR; AMAS.py summary -i ./*fasta -f fasta -d aa; cd ..; done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor DIR in ./*; do cd $DIR; cat summary.txt | cut -f6 \u0026gt; missing | grep \"\u0026gt;\" *fasta \u0026gt;\u0026gt; missing; cd ..; done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor DIR in ./*; do cd $DIR; perl -p -i -e \"s/\\n/\\t/g\" missing; cd ..; done;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat ./*/missing \u0026gt;\u0026gt; all_missing\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/\\tMissing/\\nMissing/g\" all_missing\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm ./*file -rf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -lane 'print $_ if ($F[1] gt 85)' all_missing \u0026gt; all_missing_85\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then delete species in the out.fasta file\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e3. Phylogenetic tree construction\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e3.1 Run RAXML\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# LG+G4+F is a site-homogeneous model, while CAT+G4 is a site-heterogeneous model. It is possible to use PartitionFinder with different parameters on each module separately, but this hybrid homogeneous is not equal to the heterogeneous model. The evolutionary context within each module is also different.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# CAT+G4 also considers that the evolutionary context may be different even within a module.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# The optimal model is calculated automatically; and the alpha parameters of gamma distribution and stable frequency differ according to each gene.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ephylogenomic_scafos2RAXML_partition.pl out.len\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup raxmlHPC-PTHREADS-SSE3 -T 48 -s out.fasta -n 50cov_105tx_474og -f a -N 100 -x 12345 -p 12345 -m PROTGAMMAAUTO -q new_partition.txt \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e3.2 Run IQ-TREE\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Change the charst into raxml new_partition.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp charset new_partition.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/\\s;\\s//g\" new_partition.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/\\s-\\s/-/g\" new_partition.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/charset/protein,/g\" new_partition.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup iqtree-omp -s out.fasta -wbt -bb 10000 -nt AUTO -st AA -spp new_partition.txt  -m MFP \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e3.3 Run Phylobayes\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e/opt/biosoft/RAxML_v8.2.9/usefulScripts/convertFasta2Phylip.sh ../allSingleCopyOrthologsAlign.fasta-gb \u0026gt; allSingleCopyOrthologsAlign.fasta-gb.phy\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup mpirun -np 16 /opt/biosoft/phylobayes-mpi-1.7b/data/pb_mpi -d allSingleCopyOrthologsAlign.fasta-gb.phy -cat -gtr -x 10 20000 gqx \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup mpirun -np 16 /opt/biosoft/phylobayes-mpi-1.7b/data/pb_mpi -d allSingleCopyOrthologsAlign.fasta-gb.phy -cat -gtr -x 10 20000 gqx2\u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Check convergence and output results\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup /opt/biosoft/phylobayes-mpi-1.7b/data/bpcomp -x 1000 10 gqx gqx2 -o test \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup /opt/biosoft/phylobayes-mpi-1.7b/data/tracecomp -x 1000 gqx gqx2 -o test \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# effsize \u0026gt; 300, max rel_diff \u0026lt; 0.1, converged\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e4. Remove bilaterial contamination from myxozoan sequences\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emkdir contamination_blast\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd contamination_blast\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Make seed database: bilaterian.faa and cnidarian.faa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ereplace_header_for_cdhit.pl -c bilaterian_100 -t bilaterian\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ereplace_header_for_cdhit.pl -c cnidarian_100 -t cnidarian\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat ./* \u0026gt; all.fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Remove the non-myxozoan sequences in each matrices\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp ../fasta/ ./ -R\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in ./*fa ; do perl -p -i -e \"s/(\\d)\\n/\\1\\t/g\" $dir; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in ./*fa ; do phylogenomic_filter_no_myxozoan_seq.pl $dir; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd ..\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emkdir myxo_fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv fasta/*filter myxo_fasta/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd ..\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emakeblastdb -in all.fasta -out all -dbtype prot\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./phylogenomic_parafly_contamination_blast.pl myxo_fasta/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eParaFly -c blast_cmds -CPU 4 -shuffle -v\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd myxo_fasta/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in ./*out ; do export LANG=C; export LC_ALL=C; sort -k1,1 -k12,12gr -k11,11g -k3,3gr \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"3f6008964ba396a773481e473e5b66f8\"\u003e$dir | sort -u -k1,1 --merge \u0026amp;gt; $\u003c/math-renderer\u003e{dir}_best ; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Get the bad list\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in ./*out_best ; do perl -ne '/bilaterian/ \u0026amp;\u0026amp; print' \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"3f6008964ba396a773481e473e5b66f8\"\u003e$dir \u0026amp;gt; $\u003c/math-renderer\u003e{dir}_delete ; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd ..\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emkdir delete_list\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv myxo_fasta/*delete delete_list/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd delete_list/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#find ./* -size  0 -print0 |xargs -0 rm\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in ./*delete ; do perl -p -i -e \"s/^(\\S+).*$/\\1/g\" $dir ; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# If you only want to remove certain species\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor dir in ./*out_best_delete ; do perl -ne '/HESPP/ \u0026amp;\u0026amp; print' \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"3f6008964ba396a773481e473e5b66f8\"\u003e$dir \u0026amp;gt; $\u003c/math-renderer\u003e{dir}_HESPP ; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd /home/gqx/proteome/11_big_tree/ML_analysis/optimized_105tx_232og_3/fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp ../contamination_blast/delete_list/* ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ephylogenomic_remove_contaminant_of_no_cnidarian.sh\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm ./*fa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm ./*best_delete\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erename .fa_revised .fa *.fa_revised\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e5. Approximately unbiased test (AU-test) of phylogenetic tree selection\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e5.1 Make constraint tree\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/:0.\\d+//g\" bestree.nwk\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ephylogenomic_constraint_tree_generator.pl tree_topology mono_list bestree.nwk\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# constraint.tre is the result file\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e5.2 AU-test using RAXML and Consel\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Provide the main tree\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup raxmlHPC-PTHREADS-SSE3 -T 48 -s out.fasta -n 50cov_105tx_474og -f a -N 100 -p 12345 -x 12345 -m PROTGAMMAAUTO -g constraint.tre -q new_partition.txt \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Provide constraint tree to RAXML\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Produce likehood value from the result tree\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat ../normal/RAxML_bipartitions.50cov_105tx_474og ../constraint_1/RAxML_bipartitions.50cov_105tx_474og ../constraint_2/RAxML_bipartitions.50cov_105tx_474og ../constraint_3/RAxML_bipartitions.50cov_105tx_474og \u0026gt; tree\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup raxmlHPC-PTHREADS-SSE3 -s out.fasta -z tree -m PROTGAMMAAUTO -T 6 -f G -n  50cov_105tx_474og  -q new_partition.txt \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# AU-test using Consel\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv RAxML_perSiteLLs.50cov_105tx_474og RAxML_perSiteLLs.sitelh\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emakermt --puzzle RAxML_perSiteLLs.sitelh\u003c/p\u003e\n\u003cp dir=\"auto\"\u003econsel RAxML_perSiteLLs.rmt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecatpv -s 9 RAxML_perSiteLLs.pv\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e5.3 AU-test using IQ-tree\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# The main difference between IQ-tree and RAXML is that IQ-tree integrates the functionality of Consel. The second and third steps of RAXML are combined into one step\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Get the main tree\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup iqtree-1.6 -s new.fasta -wbt -bb 10000 -nt 6 -st AA -spp new_partition.txt \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Provide the constraint tree to IQ-tree and get the result tree\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup iqtree-1.6 -s new.fasta -wbt -bb 10000 -nt 6 -st AA -spp new_partition.txt -g constraint.tre \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Use the constraint tree in former step as guide tree and conduct PMSF optimization\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv new_partition.txt.contree guide.tre\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Here is a question: when doing PMSF, do you need to use both partitioned and C60,C20 (similar to CAT, site-heterogeneous substitution models), IQ-treee allows to do these at the same time, but each partition and C20 are not linked, so for most data will be over-parameterized so no partition is used here\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Use LG+G4+F+C2 model\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup iqtree-1.6 -s new.fasta -nt 16 -st AA -m \"LG+G4+FMIX{C20pi1:1:0.0559910600,C20pi2:1:0.0514824870,C20pi3:1:0.0812922124,C20pi4:1:0.0721976867,C20pi5:1:0.0556718858,C20pi6:1:0.0331003080,C20pi7:1:0.0589501763,C20pi8:1:0.0263756889,C20pi9:1:0.0307584220,C20pi10:1:0.0376701125,C20pi11:1:0.0303058290,C20pi12:1:0.0808775576,C20pi13:1:0.0263349134,C20pi14:1:0.0579101455,C20pi15:1:0.0371248064,C20pi16:1:0.0586867766,C20pi17:1:0.0561479138,C20pi18:1:0.0349810886,C20pi19:1:0.0544937394,C20pi20:1:0.0596471901}+F\" -ft guide.tre -g constrain.tre \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Running AU-test\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat normal/new_partition.txt.contree constraint_1/PMSF/new.fasta.treefile constraint_2/PMSF/new.fasta.treefile constraint_3/PMSF/new.fasta.treefile \u0026gt; AU/tree\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup iqtree-1.6 -s new.fasta -nt 16 -m \"LG+G4+FMIX{C20pi1:1:0.0559910600,C20pi2:1:0.0514824870,C20pi3:1:0.0812922124,C20pi4:1:0.0721976867,C20pi5:1:0.0556718858,C20pi6:1:0.0331003080,C20pi7:1:0.0589501763,C20pi8:1:0.0263756889,C20pi9:1:0.0307584220,C20pi10:1:0.0376701125,C20pi11:1:0.0303058290,C20pi12:1:0.0808775576,C20pi13:1:0.0263349134,C20pi14:1:0.0579101455,C20pi15:1:0.0371248064,C20pi16:1:0.0586867766,C20pi17:1:0.0561479138,C20pi18:1:0.0349810886,C20pi19:1:0.0544937394,C20pi20:1:0.0596471901}+F\" -z tree -n 0 -zb 10000 -au \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Results are summarized in new.fasta.iqtree\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e6. Fast-evolving site removal analysis\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e6.1 Install DistEst\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003etar zxf dist_est.tar.gz\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# distEST needs two function library, id05-1.0.0.tar.gz and ve11-1.0.0.tar.gz\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# unzip these files, copy the id05i.f, ve11d.f in src directory into the dist_estv1.1 directory\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emake\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# If you see information below, you succeed\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egcc  -O3  -c dcdflib.c -o dcdflib.o\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egcc  -O3  -c dist_est.c -o dist_est.o\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egcc  -O3  -c dist_est_fn.c -o dist_est_fn.o\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egcc  -O3  -c ipmpar.c -o ipmpar.o\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egfortran -O3  -c id05i.f -o id05i.o\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egfortran -O3  -c ve11d.f -o ve11d.o\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egfortran -O3 -o dist_est dcdflib.o dist_est.o dist_est_fn.o ipmpar.o id05i.o ve11d.o -lm\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e6.2 Calculating evolutionary rates in DistEst\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Prepare hsp70.ctl， hsp70.dat， hsp70.fitchml.tre\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edist_est hsp70.ctl\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# hst70.ctl is as follows\u003c/p\u003e\n\u003cp dir=\"auto\"\u003etreefile =  hsp70.fitchml.tre  * treefile\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eseqfile = hsp70.dat            * sequence data\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enchar = 20             * amino acid data\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emodel =  3             * empirical + F\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eaaRatefile = lg.dat * LG substitution model\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enrate = 101            * number of rates\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eub = 10.0              * upper bound for rates\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# The resule file: DE.dat  rate_est.dat\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e6.3 Remove sites from fast to slow\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAMAS.py convert -i hsp70.dat -f phylip -d aa -u fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eflat_the_fasta_seq.pl metazoan.phy-out.fas\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv flated alignment.fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ephylogenomic_increment_remove_fast_evolving_sites.pl \u0026lt;fasta_alignment\u0026gt;  \u0026lt;rate_est.dat\u0026gt;  \u0026lt;num_sites_per\u0026gt;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# For instance, phylogenomic_increment_remove_fast_evolving_sites.pl alignment.fasta rate_est.dat 100\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e6.4 Build phylogenetic tree for each matrices\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup iqtree-omp -s out.fasta -wbt -bb 10000 -nt 24 -st AA -m LG+F+G \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# You can also use PMSF model\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enohup iqtree-omp -s out.fasta -wbt -bb 10000 -nt 24 -st AA -m LG+C60+F+G -ft guide.tre \u0026amp;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e6.5 Plot fast-evolving site removal analysis using R\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003elibrary(ggplot2)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edata \u0026lt;- read.csv(\"table.csv\", header=T)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edata \u0026lt;- data.frame(A=c(data[1]),B=c(data[2]),C=c(data[3]))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eggplot(data=data, aes(x=seq, y=bp, group=supp))+geom_line(aes(color=supp))+scale_color_brewer(palette=\"Dark2\")+scale_x_continuous(breaks = c(0, 3500, 7000,10500,14000,17500,21000,24500,28000,31500,35000,38500,42000,45500,49000))+theme_set(theme_bw())+theme(legend.position=\"bottom\")\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eggsave(\"plot2.pdf\", width=8, height=4)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e7. Molecular clock analysis using r8s\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Strick model: A rigorous model that assumes that all branches evolve at the same rate.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Fixed local clock: a type of relaxed model, assuming that the taxa of one or more branches are distinct from the whole, but of course, they must be monophyletic.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Uncorrelated relaxed clock: a type of relaxed model, where each branch has its own evolutionary rate and is uncorrelated.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Random local clock: a model between the discrete model and the relaxed model.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# To use r8s,aAt least one internal node needs to be fixed. The whole process includes the first validation to determine the smooth value and the formal run using the optimal smooth value. cvstart=0 cvinc=0.5 cvnum=10 means to start from 0 and go 10 at 0.5 intervals, taking values to smooth. It is better to set cvstart as zero, do not choose a negative number. In addition, if it is a large phylogenetic tree, you must have collapse parameter to merge branches. Otherwise there will be some branches because of very short and be treated as zero length, which can not be calculated.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e7.1 Prepare the configuration files r8s_in.txt\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp /opt/biosoft/r8s1.81/r8s_in_template.txt ./\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv r8s_in_template.txt r8s_in.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then edit the configuration files based on conditions\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e7.1.1 If you have 2 or more fossil fix points (i.e. clear fossil time points)\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# First prepare r8s_in_validation.txt for first validation to set the bet smooth value\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#NEXUS\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebegin trees;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003etree tree_1 = [\u0026amp;R] ((umay:0.377714,(pchr:0.126947,((cput:0.143183,slac:0.089884):0.040327,(scom:0.181745,(abis:0.150637,(lbic:0.088905,ccin:0.132642):0.022419):0.025092):0.02715):0.035777):0.207622):0.142535,((scer:0.463674,psti:0.376001):0.331119,((anig:0.213676,snod:0.244829):0.042769,(tree:0.147333,(ncra:0.134451,cpar:0.146872):0.034971):0.111483):0.206145):0.142535);\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eend;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebegin r8s;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eblformat lengths=persite nsites=320272 ulrametric=no;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecollapse;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA AGA ccin lbic scom;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA ASC psti anig cpar snod tree;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA BOL slac cput;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efixage taxon=AGA age=122.74;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efixage taxon=ASC age=517.55;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efixage taxon=BOL age=104.23;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edivtime method=PL crossv=yes fossilfixed=yes cvstart=0 cvinc=0.5 cvnum=10;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eshowage;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edescribe plot=cladogram;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edescribe plot=chrono_description;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eend;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# First validation\u003c/p\u003e\n\u003cp dir=\"auto\"\u003er8s -b -f r8s_in_validation.txt \u0026gt; cross_validation_1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Find the line with the smallest raw error, and then find the smooth value in the same line\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then revise the configuration file as a new file: r8s_in_true_run.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#NEXUS\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebegin trees;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003etree tree_1 = [\u0026amp;R] ((umay:0.377714,(pchr:0.126947,((cput:0.143183,slac:0.089884):0.040327,(scom:0.181745,(abis:0.150637,(lbic:0.088905,ccin:0.132642):0.022419):0.025092):0.02715):0.035777):0.207622):0.142535,((scer:0.463674,psti:0.376001):0.331119,((anig:0.213676,snod:0.244829):0.042769,(tree:0.147333,(ncra:0.134451,cpar:0.146872):0.034971):0.111483):0.206145):0.142535);\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eend;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebegin r8s;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eblformat lengths=persite nsites=320272 ulrametric=no;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecollapse;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA AGA ccin lbic scom;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA ASC psti anig cpar snod tree;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA BOL slac cput;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efixage taxon=AGA age=122.74;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efixage taxon=ASC age=517.55;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efixage taxon=BOL age=104.23;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eset smoothing=4;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eset num_time_guesses=3;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eset checkGradient=yes;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edivtime method=PL algorithm=TN;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eshowage;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edescribe plot=cladogram;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edescribe plot=chrono_description;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eend;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then run r8s\u003c/p\u003e\n\u003cp dir=\"auto\"\u003er8s -b -f r8s_in_true_run.txt \u0026gt; true_run_1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Extract the time tree from the result file\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -e 'while (\u0026lt;\u0026gt;) { if (m/tree tree_1 = (.*);/) { my $tree = $1; $tree =~ s/(BOL|AGA|ASC)//g; print \"$tree\\n\"; } }' true_run_1 \u0026gt; tree.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e7.1.2 If you have 2 or more constraint points (most of the time you are in this condition)\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#NEXUS\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebegin trees;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003e\t\u003c/code\u003etree tree_1 = [\u0026amp;R] (((Cerianthus_borealis:0.290575,(((((Edwarsiella_lineata:0.066246,Nematostella_vectensis:0.081163):0.055217,((Exaiptasia_pallida:0.067063,Hormathia_digitata:0.134163):0.034484,((Bolocera_tuediae:0.051153,Megalactis_griffithsi:0.06898):0.043791,(Heteractis_crispa:0.056233,(Anthopleura_elegantissima:0.052055,(Anemonia_viridis:0.00563,Anemonia_sulcata:0.004281):0.02116):0.011187):0.01741):0.038479):0.062074):0.064797,((Antipathes_caribbeana:0.053896,Plumapathes_pennacea:0.045453):0.108419,((Corynactis_australis:0.066395,(Ricordea_yuma:0.065164,((Amplexidiscus_fenestrafer:0.053832,Rhodactis_indosinensis:0.014454):0.01069,Discosoma_spp:0.03959):0.040398):0.011425):0.028652,((((Acropora_digitifera:0.057801,Acropora_millepora:0.008548):0.005001,Acropora_tenuis:0.021788):0.090596,(Astreopora_spp:0.048089,(Porites_lobata:0.002024,Porites_australiensis:0.006654):0.048218):0.012396):0.014759,(((Lobactis_scutaria:0.008759,Ctenactis_echinata:0.006969):0.027244,(Montastraea_cavernosa:0.024341,(Pseudodiploria_strigosa:0.027266,(((Favia_lizardensis:0.002407,Favia_spp:0.10512):0.006193,Platygyra_carnosus:0.16142):0.00544,Orbicella_faveolata:0.03004):0.003371):0.005356):0.017989):0.011364,(Madracis_auretenra:0.027553,(Pocillopora_damicornis:0.035619,(Stylophora_pistillata:0.030801,(Seriatopora_hystrix:0.015099,Seriatopora_spp:0.035017):0.010937):0.007567):0.034336):0.022343):0.017003):0.027674):0.07751):0.026608):0.020789,Protopalythoa_variabilis:0.277447):0.063276,((Pennatula_rubra:0.055175,Renilla_reniformis:0.107084):0.072312,(Corallium_rubrum:0.073471,(Nephthyigorgia_spp:0.062532,((Gorgonia_ventalina:0.051904,Leptogorgia_sarmentosa:0.023995):0.013823,(Eunicella_cavolinii:0.011368,Eunicella_verrucosa:0.002738):0.025411):0.011668):0.049632):0.032084):0.188635):0.018531):0.036866,((Polypodium_hydriforme:0.566426,(((Enteromyxum_leei:0.648484,Kudoa_iwatai:0.561639):0.680711,(Sphaeromyxa_zaharoni:0.953538,(Henneguya:0.780518,((((Thelohanellus_kitauei:0.307058,Myxobolus_turpisrotundus:0.304868):0.180837,((Myxobolus_wulii:0.22627,(Myxobolus_honghuensis:0.015607,Myxobolus_ampullicapsulatus:0.070408):0.17434):0.057403,Myxobolus_pendula:0.245172):0.256762):0.21192,Myxobolus_cerebralis:0.432823):0.219181,Myxobolus_physophilus:0.60296):0.170284):0.129202):0.147595):0.130403,Tetracapsuloides_bryosalmonae:0.80208):0.20069):0.142613,(((((Atolla_vanhoeffeni:0.040317,Periphylla_periphylla:0.065065):0.150184,((Aurelia_aurita:0.079173,(Cassiopea_xamachana:0.132511,(Stomolophus_meleagris:0.05262,Rhopilema_esculentum:0.037626):0.059135):0.020893):0.040423,((Cyanea_capillata:0.013458,Cyanea_nozakii:9.14E-4):0.112443,(Chrysaora_fuscescens:0.054366,Pelagia_noctiluca:0.058332):0.054913):0.032858):0.097924):0.032958,(Chironex_fleckeri:0.098164,(Alatina_alata:0.081036,Tripedalia_cystophora:0.180489):0.02203):0.132863):0.024249,(Craterolophus_convolvulus:0.100106,(Calvadosia_campanulata:0.081895,(Haliclystus_sanjuanensis:0.150622,Calvadosia_cruxmelitensis:0.047193):0.020468):0.023085):0.204894):0.024348,((Liriope_tetraphylla:0.144402,(Craspedacusta_sowerbyi:0.163299,Aegina_citrea:0.22533):0.02255):0.055681,((Ectopleura_larynx:0.169364,(Hydra_viridissima:0.041963,(Hydra_oligactis:0.030741,Hydra_vulgaris:0.028101):0.038597):0.092278):0.078655,(Millepora_alcicornis:0.226375,((Turritopsis_spp:0.15472,(Podocoryna_carnea:0.102354,(Hydractinia_polyclina:0.008327,Hydractinia_symbiolongicarpus:0.007533):0.059407):0.051791):0.030935,(Physalia_physalis:0.111875,((Nanomia_bijuga:0.07543,Agalma_elegans:0.042964):0.044384,(Craseoa_lathetica:0.059619,Abylopsis_tetragona:0.121102):0.037884):0.042855):0.075418):0.015359):0.016351):0.058833):0.111132):0.063606):0.055702):0.028066,((((Helobdella_robusta:0.428146,Crassostrea_gigas:0.25635):0.054388,Drosophila_melanogaster:0.536965):0.050571,((Ciona_intestinalis:0.459065,Taeniopygia_guttata:0.299825):0.052585,Strongylocentrotus_purpuratus:0.349709):0.026884):0.077267,(Trichoplax_adhaerens:0.483558,(Amphimedon_queenslandica:0.550768,Mnemiopsis_leidyi:0.725216):0.059918):0.045348):0.028066);\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eend;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebegin r8s;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eblformat lengths=persite nsites=51598 ulrametric=no;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecollapse;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA OUT Helobdella_robusta Mnemiopsis_leidyi Trichoplax_adhaerens;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA MED Chironex_fleckeri Agalma_elegans Thelohanellus_kitauei;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA HEX Protopalythoa_variabilis Edwarsiella_lineata Lobactis_scutaria;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA HYD Liriope_tetraphylla Agalma_elegans Hydra_vulgaris;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA CNI Protopalythoa_variabilis Chironex_fleckeri Thelohanellus_kitauei;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eprune taxon=OUT;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efixage taxon=HYD age=500;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003econstrain taxon=MED min_age=570;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003econstrain taxon=HEX min_age=540;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003econstrain taxon=CNI max_age=741;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eset minRateFactor=0.000001;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eset minDurFactor=0.000001;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edivtime method=PL crossv=yes fossilconstrained=yes cvstart=0 cvinc=1 cvnum=15;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eshowage;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edescribe plot=cladogram;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edescribe plot=chrono_description;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eend;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then conduct first validation\u003c/p\u003e\n\u003cp dir=\"auto\"\u003er8s -b -f r8s_in_validation.txt \u0026gt; cross_validation_1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Find the line with the smallest raw error, and then find the log10smoon value in the same line\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then revise the configuration file as a new file: r8s_in_true_run.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#NEXUS\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebegin trees;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003e\t\u003c/code\u003etree tree_1 = [\u0026amp;R] (((Cerianthus_borealis:0.290575,(((((Edwarsiella_lineata:0.066246,Nematostella_vectensis:0.081163):0.055217,((Exaiptasia_pallida:0.067063,Hormathia_digitata:0.134163):0.034484,((Bolocera_tuediae:0.051153,Megalactis_griffithsi:0.06898):0.043791,(Heteractis_crispa:0.056233,(Anthopleura_elegantissima:0.052055,(Anemonia_viridis:0.00563,Anemonia_sulcata:0.004281):0.02116):0.011187):0.01741):0.038479):0.062074):0.064797,((Antipathes_caribbeana:0.053896,Plumapathes_pennacea:0.045453):0.108419,((Corynactis_australis:0.066395,(Ricordea_yuma:0.065164,((Amplexidiscus_fenestrafer:0.053832,Rhodactis_indosinensis:0.014454):0.01069,Discosoma_spp:0.03959):0.040398):0.011425):0.028652,((((Acropora_digitifera:0.057801,Acropora_millepora:0.008548):0.005001,Acropora_tenuis:0.021788):0.090596,(Astreopora_spp:0.048089,(Porites_lobata:0.002024,Porites_australiensis:0.006654):0.048218):0.012396):0.014759,(((Lobactis_scutaria:0.008759,Ctenactis_echinata:0.006969):0.027244,(Montastraea_cavernosa:0.024341,(Pseudodiploria_strigosa:0.027266,(((Favia_lizardensis:0.002407,Favia_spp:0.10512):0.006193,Platygyra_carnosus:0.16142):0.00544,Orbicella_faveolata:0.03004):0.003371):0.005356):0.017989):0.011364,(Madracis_auretenra:0.027553,(Pocillopora_damicornis:0.035619,(Stylophora_pistillata:0.030801,(Seriatopora_hystrix:0.015099,Seriatopora_spp:0.035017):0.010937):0.007567):0.034336):0.022343):0.017003):0.027674):0.07751):0.026608):0.020789,Protopalythoa_variabilis:0.277447):0.063276,((Pennatula_rubra:0.055175,Renilla_reniformis:0.107084):0.072312,(Corallium_rubrum:0.073471,(Nephthyigorgia_spp:0.062532,((Gorgonia_ventalina:0.051904,Leptogorgia_sarmentosa:0.023995):0.013823,(Eunicella_cavolinii:0.011368,Eunicella_verrucosa:0.002738):0.025411):0.011668):0.049632):0.032084):0.188635):0.018531):0.036866,((Polypodium_hydriforme:0.566426,(((Enteromyxum_leei:0.648484,Kudoa_iwatai:0.561639):0.680711,(Sphaeromyxa_zaharoni:0.953538,(Henneguya:0.780518,((((Thelohanellus_kitauei:0.307058,Myxobolus_turpisrotundus:0.304868):0.180837,((Myxobolus_wulii:0.22627,(Myxobolus_honghuensis:0.015607,Myxobolus_ampullicapsulatus:0.070408):0.17434):0.057403,Myxobolus_pendula:0.245172):0.256762):0.21192,Myxobolus_cerebralis:0.432823):0.219181,Myxobolus_physophilus:0.60296):0.170284):0.129202):0.147595):0.130403,Tetracapsuloides_bryosalmonae:0.80208):0.20069):0.142613,(((((Atolla_vanhoeffeni:0.040317,Periphylla_periphylla:0.065065):0.150184,((Aurelia_aurita:0.079173,(Cassiopea_xamachana:0.132511,(Stomolophus_meleagris:0.05262,Rhopilema_esculentum:0.037626):0.059135):0.020893):0.040423,((Cyanea_capillata:0.013458,Cyanea_nozakii:9.14E-4):0.112443,(Chrysaora_fuscescens:0.054366,Pelagia_noctiluca:0.058332):0.054913):0.032858):0.097924):0.032958,(Chironex_fleckeri:0.098164,(Alatina_alata:0.081036,Tripedalia_cystophora:0.180489):0.02203):0.132863):0.024249,(Craterolophus_convolvulus:0.100106,(Calvadosia_campanulata:0.081895,(Haliclystus_sanjuanensis:0.150622,Calvadosia_cruxmelitensis:0.047193):0.020468):0.023085):0.204894):0.024348,((Liriope_tetraphylla:0.144402,(Craspedacusta_sowerbyi:0.163299,Aegina_citrea:0.22533):0.02255):0.055681,((Ectopleura_larynx:0.169364,(Hydra_viridissima:0.041963,(Hydra_oligactis:0.030741,Hydra_vulgaris:0.028101):0.038597):0.092278):0.078655,(Millepora_alcicornis:0.226375,((Turritopsis_spp:0.15472,(Podocoryna_carnea:0.102354,(Hydractinia_polyclina:0.008327,Hydractinia_symbiolongicarpus:0.007533):0.059407):0.051791):0.030935,(Physalia_physalis:0.111875,((Nanomia_bijuga:0.07543,Agalma_elegans:0.042964):0.044384,(Craseoa_lathetica:0.059619,Abylopsis_tetragona:0.121102):0.037884):0.042855):0.075418):0.015359):0.016351):0.058833):0.111132):0.063606):0.055702):0.028066,((((Helobdella_robusta:0.428146,Crassostrea_gigas:0.25635):0.054388,Drosophila_melanogaster:0.536965):0.050571,((Ciona_intestinalis:0.459065,Taeniopygia_guttata:0.299825):0.052585,Strongylocentrotus_purpuratus:0.349709):0.026884):0.077267,(Trichoplax_adhaerens:0.483558,(Amphimedon_queenslandica:0.550768,Mnemiopsis_leidyi:0.725216):0.059918):0.045348):0.028066);\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eend;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebegin r8s;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eblformat lengths=persite nsites=51598 ulrametric=no;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecollapse;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA OUT Helobdella_robusta Mnemiopsis_leidyi Trichoplax_adhaerens;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA MED Chironex_fleckeri Agalma_elegans Thelohanellus_kitauei;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA HEX Protopalythoa_variabilis Edwarsiella_lineata Lobactis_scutaria;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA HYD Liriope_tetraphylla Agalma_elegans Hydra_vulgaris;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMRCA CNI Protopalythoa_variabilis Chironex_fleckeri Thelohanellus_kitauei;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eprune taxon=OUT;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efixage taxon=HYD age=500;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003econstrain taxon=MED min_age=570;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003econstrain taxon=HEX min_age=540;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003econstrain taxon=CNI max_age=741;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eset minRateFactor=0.000001;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eset minDurFactor=0.000001;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eset smoothing=1e-06;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eset num_time_guesses=3;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eset checkGradient=yes;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edivtime method=PL algorithm=TN;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eshowage;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edescribe plot=cladogram;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edescribe plot=chrono_description;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eend;\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Run r8s\u003c/p\u003e\n\u003cp dir=\"auto\"\u003er8s -b -f r8s_in_true_run.txt \u0026gt; true_run_1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Extract the time tree from the result file\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -e 'while (\u0026lt;\u0026gt;) { if (m/tree tree_1 = (.*);/) { my $tree = $1; $tree =~ s/(MED|HEX|HYD|CNI)//g; print \"$tree\\n\"; } }' true_run_1 \u0026gt; tree.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e8. Selection pressure analysis using Hyphy\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Make configuration file\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egenerate_busted_hyphy.sh alignment.phy tree.nwk \u0026gt; script.bf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHYPHYMPI script.bf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# The content of generate_busted_hyphy.sh is as follows:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003einputRedirect = {};\u003c/p\u003e\n\u003cp dir=\"auto\"\u003einputRedirect[\"01\"]=\"Universal\"; // genetic code\u003c/p\u003e\n\u003cp dir=\"auto\"\u003einputRedirect[\"02\"]=\"/home/gqx/proteome/16_selection_analysis/disintergrin/Disintergin_nucl.aln\"; // codon data\u003c/p\u003e\n\u003cp dir=\"auto\"\u003einputRedirect[\"03\"]=\"/home/gqx/proteome/16_selection_analysis/disintergrin/1.nwk\"; // tree\u003c/p\u003e\n\u003cp dir=\"auto\"\u003einputRedirect[\"04\"]=\"All\"; // Test for selection on a branch\u003c/p\u003e\n\u003cp dir=\"auto\"\u003einputRedirect[\"05\"]=\"\"; // complete selection\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eExecuteAFile (\"/usr/local/lib/hyphy/TemplateBatchFiles/SelectionAnalyses/BUSTED.bf\", inputRedirect);\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# If you want to use multithreading\u003c/p\u003e\n\u003cp dir=\"auto\"\u003empirun -np 12 /usr/local/bin/HYPHYMPI ./script.bf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e9. Orthologous gene family analysis in comparative proteomics and visualization using UpSetR\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# For the orthologous gene family analysis by OrthoMCL, please see section 1.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Provide matrix format that is suitable for UpSetR\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat aaur_OG avir_OG cfle_OG hvul_OG mhon_OG mwul_OG tkit_OG \u0026gt; OG\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eremove_duplicate.pl OG\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv duplicate_remove gene\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eproduce_matrix_for_UpSet.pl gene aaur_OG\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then combine a csv file that contains the gene and species information, eight_matrix_strict.csv\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# The content of Rscript upset_eight.R is as follows\u003c/p\u003e\n\u003cp dir=\"auto\"\u003elibrary(UpSetR)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edata \u0026lt;- read.csv(\"eight_matrix.csv\",header=T)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003epdf(\"out.pdf\",  width=11, height=6)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eupset(data, nsets = 8, order.by=\"freq\", sets.bar.color = \"#56B4E9\", mainbar.y.label = \"OG Intersections\", sets.x.label = \"ProteinsPer Species\",nintersects = 60)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edev.off()\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e10. Functional comparison in comparative proteomics and visualization using UpSetR\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e10.1 Get the Interpro number for each proteins\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Use Blast2GO to do Interpro annotation, collect the Interpro number (IPR) and save it as list file.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Then run the bash file: interpro_ID_converter.sh, as follows:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egrep \"IPR\" list  \u0026gt; out\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esplit_lines_by_semicolon.pl out\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egrep \"IPR\" out2 \u0026gt; out3\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eremove_duplicate.pl out3\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/\\s+(\\S+)$//g' duplicate_remove\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv duplicate_remove duplicate_remove1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eremove_duplicate.pl duplicate_remove1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm duplicate_remove1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv duplicate_remove final_IPR\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/^\\s//g' final_IPR\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eremove_duplicate.pl final_IPR\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm final_IPR\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv duplicate_remove final_IPR\u003c/p\u003e\n\u003cp dir=\"auto\"\u003econvert_interpro_2_name.pl final_IPR entry.list\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm out out2 out3\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat convert_name | cut -f 3 \u0026gt; domain_name\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# The final_IPR contains the information for visualization\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e10.2 Visualize the results using UpSetR\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Change the name of final_IPR for each species\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Use \u003cem\u003eChironex\u003c/em\u003e as an example\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp ../5_Chironex_domain/final_IPR ./Chironex\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat Anemonia  Aurelia  Chironex  Hydra  honghuensis  kitauei  shata wulii \u0026gt; OG\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eremove_duplicate.pl OG\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv duplicate_remove gene\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# For each species\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eproduce_matrix_for_UpSet.pl gene Anemonia\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erename _matrix .xls *matrix\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Combine gene and OG_matrix into a single csv: eight_matrix_strict.csv\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eRscript upset_eight.R\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e11. Circos analysis and visualization\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e11.1 Install circos\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"http://circos.ca/distribution/circos-0.69-6.tgz\" rel=\"nofollow\"\u003ehttp://circos.ca/distribution/circos-0.69-6.tgz\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003etar zxf /opt/software/circos-0.69-6.tgz\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd /opt/biosoft/circos-0.69-6/bin\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./list.modules\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./list.modules | perl -p -e 's/\\n/ /' | perl -p -e 's/^/sudo cpan -i /; s/$/\\n/' | sh\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./circos -modules | perl -e 'while (\u0026lt;\u0026gt;) { print \"$1 \" if m/missing\\s+(\\S+)/ }' | perl -p -e 's/^/sudo cpan -i /; s/$/\\n/' | sh     # Install missing modules\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#Install GD\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd /opt/sysoft/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003etar Jxf  /opt/software/libgd-2.1.1.tar.xz\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./configure --prefix=/usr/\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emake -j 4\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esudo make install\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egdlib-config\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eln -s /usr/lib64/libgd.so.2.0.0 /usr/lib64/libgd.so.3\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ewget  \u003ca href=\"http://circos.ca/distribution/lib/GD-2.53.tar.gz\" rel=\"nofollow\"\u003ehttp://circos.ca/distribution/lib/GD-2.53.tar.gz\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003etar  zxf /opt/software/GD-2.53.tar.gz\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl Makefile.PL\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emake\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emake install\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./circos -modules\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./gddiag\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./circos --help\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eecho 'PATH=$PATH:/opt/biosoft/circos-0.69-6/bin/' \u0026gt;\u0026gt; ~/.bashr\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esource ~/.bashrc\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Dowmload circos_tools\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ewget \u003ca href=\"http://circos.ca/distribution/circos-tools-0.22.tgz\" rel=\"nofollow\"\u003ehttp://circos.ca/distribution/circos-tools-0.22.tgz\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003etar zxf /opt/software/circos-tools-0.22.tgz\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd /opt/biosoft/circos-tools-0.22/tools/binlinks\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e./run\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e11.2 Install Circoletto\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ewget \u003ca href=\"https://github.com/infspiredBAT/Circoletto/archive/master.zip\"\u003ehttps://github.com/infspiredBAT/Circoletto/archive/master.zip\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eunzip master.zip\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecd Circoletto-master\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# If you are dealing with more than 200 sequences, change the max_ideograms parameter\u003c/p\u003e\n\u003cp dir=\"auto\"\u003evim /opt/biosoft/circos-0.69-6/etc/housekeeping.conf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Also change circoletto.pl, feed it with the path of other softwares\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e$path2circos      = '/opt/biosoft/circos-0.69-6';     #  path to Circos\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e$path2circostools = '/opt/biosoft/circos-tools-0.22'; #  path to Circos utilities\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e11.3 Install blast-2.2.25\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ewget ftp://ftp.ncbi.nlm.nih.gov/blast/executables/legacy/2.2.25/blast-2.2.25-x64-linux.tar.gz\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eecho \"PATH=$PATH:/opt/biosoft/blast-2.2.25/bin\" \u0026gt;\u0026gt; ~/.bashrc\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esource ~/.bashrc\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e11.4 Run Circoletto\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Use analysis between the protein of \u003cem\u003eMyxobolus honghuensis\u003c/em\u003e and other six species as an example\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat aaur.fasta avir.fasta cfle.fasta cfus.fasta cnoz.fasta hvul.fasta \u0026gt; rest.fa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Circoletto can not deal with asterisks.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e \"s/\\*//g\" rest.fa\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Change the parameter in Circoletto.pl\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e$max_sequences = 10000\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecircoletto.pl --cpus 12 --query mhon.fasta --database rest.fa --out_type svg --e_value 1e-20 --score2colour eval --scoreratio2colour max  --ribocolours2allow '(blue|green|orange|red)' --z_by score --no_labels\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Use Perl to process link file to adjust the link details\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/stroke_color=(\\S+?),//g' ./*.blasted.links\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/stroke_thickness=(\\d+?),//g' ./*.blasted.links\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/_a3/_a2/g' ./*.blasted.links\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/thickness=(\\d+)/thickness=4/g' ./*.blasted.links\u003c/p\u003e\n\u003cp dir=\"auto\"\u003epaste -d ',' ./*.blasted.links ff \u0026gt; new\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm ./*.links\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv new  cl0011919395.blasted.links\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Use Perl to process karyotype file to adjust the color\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/(\\S+)$/dgreen/ if ($_ =~ /chr/ and $_ =~ /aaur/)' ./*.blasted.karyotype\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/(\\S+)$/lblue/ if ($_ =~ /chr/ and $_ =~ /avir/)' ./*.blasted.karyotype\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/(\\S+)$/dred/ if ($_ =~ /chr/ and $_ =~ /cfle/)' ./*.blasted.karyotype\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/(\\S+)$/dyellow/ if ($_ =~ /chr/ and $_ =~ /cfus/)' ./*.blasted.karyotype\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/(\\S+)$/black/ if ($_ =~ /chr/ and $_ =~ /hvul/)' ./*.blasted.karyotype\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/(\\S+)$/dblue/ if ($_ =~ /chr/ and $_ =~ /mhon/)' ./*.blasted.karyotype\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/(\\S+)$/lpurple/ if ($_ =~ /chr/ and $_ =~ /cnoz/)' ./*.blasted.karyotype\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/(\\S+)$/dpurple/ if ($_ =~ /chr/ and $_ =~ /tkit/)' ./*.blasted.karyotype\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/(\\S+)$/dorange/ if ($_ =~ /chr/ and $_ =~ /mwul/)' ./*.blasted.karyotype\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eperl -p -i -e 's/(\\S+)$/dyellow/ if ($_ =~ /chr/ and $_ =~ /csha/)' ./*.blasted.karyotype\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Edit the conf file as follows:\u003c/p\u003e\n\n\n\u003cp dir=\"auto\"\u003edefault             = 0.0003r\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebreak               = 10u\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eaxis_break_at_edge  = no\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eaxis_break          = no\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eaxis_break_style    = 1\u003c/p\u003e\n\n\u003cp dir=\"auto\"\u003ethickness           = 35p\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efill                = yes\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eradius              = 0.66r\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eshow_label          = no\u003c/p\u003e\n\u003cp dir=\"auto\"\u003elabel_with_tag      = yes\u003c/p\u003e\n\u003cp dir=\"auto\"\u003elabel_font          = condensed\u003c/p\u003e\n\u003cp dir=\"auto\"\u003elabel_color         = grey\u003c/p\u003e\n\u003cp dir=\"auto\"\u003elabel_radius        = dims(ideogram,radius) + 0.104r\u003c/p\u003e\n\u003cp dir=\"auto\"\u003elabel_size          =\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eband_stroke_thickness = 0\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eshow_bands          = no\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efill_bands          = yes\u003c/p\u003e\n\n\u003cp dir=\"auto\"\u003eshow_ticks          = no\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eshow_tick_labels    = no\u003c/p\u003e\n\n\u003cp dir=\"auto\"\u003eradius              = dims(ideogram,radius_outer)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emultiplier          = 1e-6\u003c/p\u003e\n\n\u003cp dir=\"auto\"\u003espacing             = 100u\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erspacing            = 0.1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003espacing_type        = relative\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esize                = 5p\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ethickness           = 1p\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecolor               = vvvvdgrey\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eshow_label          = no\u003c/p\u003e\n\u003cp dir=\"auto\"\u003elabel_size          = 10p\u003c/p\u003e\n\u003cp dir=\"auto\"\u003elabel_offset        = 5p\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eformat              = %d\u003c/p\u003e\n\n\n\u003cp dir=\"auto\"\u003ekaryotype           = cl0006216655.blasted.karyotype\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"\"\u003e\u003cimg style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edir  = .\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efile                = cl0006216655.blasted.svg\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eradius              = 1000p\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebackground          = white\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eangle_offset        = -90\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e24bit               = yes\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eauto_alpha_colors   = yes\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eauto_alpha_steps    = 5\u003c/p\u003e\n\n\u003cp dir=\"auto\"\u003echromosomes_units   = 1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003echromosomes_display_default = yes\u003c/p\u003e\n\n\u003cp dir=\"auto\"\u003eribbon              = no\u003c/p\u003e\n\n\u003cp dir=\"auto\"\u003eshow                = yes\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efile                = cl0006216655.blasted.links\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erecord_limit        = 2000\u003c/p\u003e\n\n\n\u003cp dir=\"auto\"\u003e# Run circus\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecircos -conf ./*.conf -noparanoid\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e12. Identify ortholog gene manually\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# RBBH is mainly used to find orthologs, while the common blast includes all homologs, and RBBH can distinguish orthologs from paralogs. The NC method is actually an all-against-all blast, which includes paralogs, so it is generally used for homology identification. Note: homology does not necessarily mean ortholog. homology is more stringent.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e12.1 Pairwise Reciprocal Best BLAST Hits (RBBH)\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eblast_rbbh.py fish.fasta human.fasta -o fish_human.tab -t blastp -a prot -i 1 -c 1 --threads 48\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# If you want to do RBBH for three species\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat Avir_Aaur.tab | cut -f 1,2 \u0026gt; 22\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat Avir_Hydra.tab | cut -f 1,2 \u0026gt; 33\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat Aaur_Hydra.tab | cut -f 1,2 \u0026gt; 44\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erun_RBBH.pl ../Anemonia_viridis_Rachamim_2014.fasta ../Aurelia_aurita_Rachamim_2014.fasta Anemonia Aurelia\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eproteome_RBBH_get_three_inter.pl Anemonia_Aurelia Anemonia_Hydra Aurelia_Hydra\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e12.2 Neighborhood correlation (NC) method\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e# Compare Anemonia_viridis_Rachamim_2014.fasta and Hydra_vulgaris_Rachamim_2014.fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erun_neighborhood_correlation.pl Anemonia_viridis_Rachamim_2014.fasta Hydra_vulgaris_Rachamim_2014.fasta Anemonia Hydra 0.6\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eproteome_NC_result_process.pl out\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNC_thresh.pl NC_result 0.8\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat out2 | cut -d ' ' -f 1 \u0026gt; 1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eremove_duplicate.pl 1\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ewc -l duplicate_remove\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emv out2 NC_0.8\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ePlease cite this paper as\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eGuo, Q., Atkinson, S. D., Xiao, B., Zhai, Y., Bartholomew, J. L., \u0026amp; Gu, Z. (2022). A myxozoan genome reveals mosaic evolution in a parasitic cnidarian. BMC biology, 20(1), 1-19.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eLicense\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAll source code, i.e. scripts/*.pl, scripts/*.sh or scripts/*.py are under the MIT license.\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fqingxiangguo%2FPhylogenomic_analysis_and_comparative_proteomics"}},{"displayName":"LICENSE","repoName":"Phylogenomic_analysis_and_comparative_proteomics","refName":"main","path":"LICENSE","preferredFileType":"license","tabName":"MIT","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fqingxiangguo%2FPhylogenomic_analysis_and_comparative_proteomics"}}],"overviewFilesProcessingTime":113.952157}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
.bhqztV{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.bhqztV span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.bhqztV:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.bhqztV:focus{outline:2px solid transparent;}/*!sc*/
.bhqztV:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.bhqztV:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.bhqztV::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:transparent;border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.bhqztV::after{background-color:transparent;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,bhqztV,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           6 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="scripts, (Directory)" class="Link--primary" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/tree/main/scripts" title="scripts">
                          scripts
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="scripts, (Directory)" class="Link--primary" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/tree/main/scripts" title="scripts">
                          scripts
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/blob/main/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/blob/main/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-law" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="MIT license">
                        MIT license
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <p dir="auto">
                     <strong>
                      Streaming workflow for phylogenomic analysis and comparative proteomics
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      Contributors
                     </strong>
                    </p>
                    <p dir="auto">
                     Qingxiang Guo
                    </p>
                    <p dir="auto">
                     <strong>
                      About
                     </strong>
                    </p>
                    <p dir="auto">
                     The following contents contains source code for the analyses and plots in my Biology paper, 2022, “Quantitative insights into the contribution of nematocysts to the adaptive success of cnidarians based on proteomic analysis”
                    </p>
                    <p dir="auto">
                     <strong>
                      Abstract
                     </strong>
                    </p>
                    <p dir="auto">
                     Cnidarians (such as corals, anemones, jellyfish) are ancient and successful animals. Previous studies have offered a variety of explanations for the adaptive success of certain cnidarian taxa. However, common strategies for the long-term persistence of cnidarians have not been identified. One factor that may contribute to their evolutionary success of the lineage is the nematocyst, a sting organelle able to deliver venom into prey or enemies. Using bioinformatics analyses, we aimed to quantitatively investigate the role of nematocysts in cnidarian adaptation. We identified the extensive species specific adaptation in nematocyst proteins (NEMs) and demonstrate that both a unique evolutionary pattern of NEMs and the long evolutionary lag between nematocysts and cnidarians support their key adaptive role. Further, we find NEMs experience approximately 50% more adaptive changes on average compared to non-NEMs, and positively selected cnidarian-conserved proteins are enriched in NEMs. These results support a key role of nematocysts in successful cnidarian adaptation and provide a general quantitative framework for assessing the role of a phenotypic novelty in adaptation. Moreover, the findings will be critical for reassessing the evolutionary history of many established models, enhancing our understanding of both the mechanisms and evolutionary preference of adaptive evolution.
                    </p>
                    <p dir="auto">
                     <strong>
                      Notes
                     </strong>
                    </p>
                    <p dir="auto">
                     Written by Qingxiang Guo,
                     <a href="mailto:qingxiang.guo@outlook.com">
                      qingxiang.guo@outlook.com
                     </a>
                     , distributed without any guarantees or restrictions.
                    </p>
                    <p dir="auto">
                     <strong>
                      Codes
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      1. Orthologous gene family analysis by OrthoMCL
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      1.1 Disposing the MySQL
                     </strong>
                    </p>
                    <p dir="auto">
                     vim /etc/my.cnf
                    </p>
                    <p dir="auto">
                     myisam_sort_buffer_size=128000M
                    </p>
                    <p dir="auto">
                     read_buffer_size=2000M
                    </p>
                    <p dir="auto">
                     mysql&gt; show variables LIKE 'myisam_max_sort_file_size';
                    </p>
                    <p dir="auto">
                     mysql&gt; show variables LIKE 'myisam_sort_buffer_size';
                    </p>
                    <p dir="auto">
                     mysql&gt; show variables LIKE 'read_buffer_size';
                    </p>
                    <p dir="auto">
                     # Restart
                    </p>
                    <p dir="auto">
                     service mysqld restart
                    </p>
                    <p dir="auto">
                     <strong>
                      1.2 Install MCL
                     </strong>
                    </p>
                    <p dir="auto">
                     tar zxf /opt/software/mcl-14-137.tar.gz -C /opt/biosoft/
                    </p>
                    <p dir="auto">
                     ./configure --prefix=/opt/biosoft/mcl-14-137/ &amp;&amp; make -j 4 &amp;&amp; make install
                    </p>
                    <p dir="auto">
                     echo 'PATH=$PATH:/opt/biosoft/mcl-14-137/bin/' &gt;&gt; ~/.bashrc
                    </p>
                    <p dir="auto">
                     source ~/.bashrc
                    </p>
                    <p dir="auto">
                     <strong>
                      1.3 Configuration
                     </strong>
                    </p>
                    <p dir="auto">
                     /home/gqx/proteome/tmp_test/ortho_test
                    </p>
                    <p dir="auto">
                     cp /opt/biosoft/orthomclSoftware-v2.0.9/doc/OrthoMCLEngine/Main/orthomcl.config.template ./orthomcl.config
                    </p>
                    <p dir="auto">
                     vim orthomcl.config
                    </p>
                    <p dir="auto">
                     # Editing the text
                    </p>
                    <p dir="auto">
                     dbConnectString=dbi:mysql:orthomcl
                    </p>
                    <p dir="auto">
                     dbLogin=gqx
                    </p>
                    <p dir="auto">
                     dbPassword=123456
                    </p>
                    <p dir="auto">
                     <strong>
                      1.4 Install the MySQL tables required for OrthoMCL
                     </strong>
                    </p>
                    <p dir="auto">
                     mysql -u gqx -p123456
                    </p>
                    <p dir="auto">
                     CREATE DATABASE IF NOT EXISTS orthomcl;
                    </p>
                    <p dir="auto">
                     orthomclInstallSchema orthomcl.config
                    </p>
                    <p dir="auto">
                     <strong>
                      1.5 Prepare the input fasta data
                     </strong>
                    </p>
                    <p dir="auto">
                     # Edit the file name to make it recognizable for OrthoMCL
                    </p>
                    <p dir="auto">
                     perl -p -e "s/_/|/g" Aurelia_aurita_Rachamim.fasta &gt; fix.fasta
                    </p>
                    <p dir="auto">
                     tar zxf /home/train/00.incipient_data/data_for_genome_comparison/compliantFasta.tar.gz -C ./
                    </p>
                    <p dir="auto">
                     orthomclAdjustFasta ncra ncra_proteins.fasta 1
                    </p>
                    <p dir="auto">
                     rm ncra_proteins.fasta
                    </p>
                    <p dir="auto">
                     <strong>
                      1.6 Remove low-quality sequence
                     </strong>
                    </p>
                    <p dir="auto">
                     # Longer than 30 aa, and the stop codon ratio threshold is 20%
                    </p>
                    <p dir="auto">
                     orthomclFilterFasta compliantFasta/ 10 20
                    </p>
                    <p dir="auto">
                     <strong>
                      1.7 All-vs-all BLAST on the filtered result goodProteins.fasta
                     </strong>
                    </p>
                    <p dir="auto">
                     makeblastdb -in goodProteins.fasta -dbtype prot -title orthomcl -parse_seqids -out orthomcl -logfile orthomcl.log
                    </p>
                    <p dir="auto">
                     nohup blastp -db orthomcl -query goodProteins.fasta -seg yes -out blastresult.tab -evalue 1e-5 -num_threads 64 -outfmt 6 &amp;
                    </p>
                    <p dir="auto">
                     <strong>
                      1.8 Process the results of BLAST to get the similarity information between the sequences, so as to facilitate importing into MySQL
                     </strong>
                    </p>
                    <p dir="auto">
                     orthomclBlastParser blastresult.tab compliantFasta/ &gt; similarSequences.txt
                    </p>
                    <p dir="auto">
                     # Then set myisam_max_sort_file_size (five times larger) in mysql according to the size of the similarSequences.txt file,
                    </p>
                    <p dir="auto">
                     myisam_max_sort_file_size=2955M
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/0\t0/1\t-181/' similarSequences.txt
                    </p>
                    <p dir="auto">
                     <strong>
                      1.9 Import similarSequences.txt to MySQL database
                     </strong>
                    </p>
                    <p dir="auto">
                     orthomclLoadBlast orthomcl.config similarSequences.txt
                    </p>
                    <p dir="auto">
                     <strong>
                      1.10
                     </strong>
                     <strong>
                      Find similar sequence pairs: pairs that best match each other between species (orthologs pairs); pairs that best match each other and are better than interspecies matches within species (in-paralogs pairs); the first two are combined into co-ortholog pairs
                     </strong>
                    </p>
                    <p dir="auto">
                     orthomclPairs orthomcl.config orthomcl_pairs.log cleanup=no
                    </p>
                    <p dir="auto">
                     <strong>
                      1.11 Export similar sequence pairs from MySQL
                     </strong>
                    </p>
                    <p dir="auto">
                     orthomclDumpPairsFiles orthomcl.config
                    </p>
                    <p dir="auto">
                     # output: mclInput, pairs file
                    </p>
                    <p dir="auto">
                     <strong>
                      1.12 Use mcl to cluster pairs (Ortholog Cluster Groups) and number the classes
                     </strong>
                    </p>
                    <p dir="auto">
                     mcl mclInput --abc -I 1.5 -o mclOutput
                    </p>
                    <p dir="auto">
                     orthomclMclToGroups OCG 1 &lt; mclOutput &gt; groups.txt
                    </p>
                    <p dir="auto">
                     # groups.txt contains the final cluster information
                    </p>
                    <p dir="auto">
                     <strong>
                      1.13 Manual extraction of cluster results
                     </strong>
                    </p>
                    <p dir="auto">
                     orthomcl_two.pl groups.txt mhon mwul
                    </p>
                    <p dir="auto">
                     orthomcl_two.pl groups.txt mhon tkit
                    </p>
                    <p dir="auto">
                     orthomcl_two.pl groups.txt mwul tkit
                    </p>
                    <p dir="auto">
                     orthomcl_three.pl groups.txt
                    </p>
                    <p dir="auto">
                     <strong>
                      1.14 Plot three-set Venn diagram of the cluster results
                     </strong>
                    </p>
                    <p dir="auto">
                     # First, match each sequence ID to OG in the result
                    </p>
                    <p dir="auto">
                     orthomcl_result_2_header.pl groups.txt
                    </p>
                    <p dir="auto">
                     cd ..
                    </p>
                    <p dir="auto">
                     mv 0_orthomcl/header ./
                    </p>
                    <p dir="auto">
                     # Create header file one by one for each species
                    </p>
                    <p dir="auto">
                     grep "&gt;" 0_orthomcl/compliantFasta/cfle.fasta &gt; cfle
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/&gt;//g" cfle
                    </p>
                    <p dir="auto">
                     # Then replace the original sequence ID according to the corresponding relationship
                    </p>
                    <p dir="auto">
                     orthomcl_header_2_OG.pl mhon header
                    </p>
                    <p dir="auto">
                     Rscript three_set_venn.R
                    </p>
                    <p dir="auto">
                     <strong>
                      1.15 Statistics of the orthologs
                     </strong>
                    </p>
                    <p dir="auto">
                     orthomcl_genes_number_stats.pl groups.txt compliantFasta &gt; genes_number_stats.txt
                    </p>
                    <p dir="auto">
                     <strong>
                      1.16 Extract the single-copy orthologs from the OrthoMCL results
                     </strong>
                    </p>
                    <p dir="auto">
                     orthomcl_findSingleCopyOrtholog.pl groups.txt compliantFasta/
                    </p>
                    <p dir="auto">
                     <strong>
                      1.17 Batch alignment of sequences
                     </strong>
                    </p>
                    <p dir="auto">
                     fastas2aligned_by_mafft.pl SingleCopyOrthologGroups 8
                    </p>
                    <p dir="auto">
                     # You will get the final allSingleCopyOrthologsAlign.fasta for downstream analysis (e. g., phylogenomics)
                    </p>
                    <p dir="auto">
                     <strong>
                      2. Orthologous gene family analysis by HaMStR and phylogenomic data matrices construction
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      2.1 Run HaMStR to get orthologs
                     </strong>
                    </p>
                    <p dir="auto">
                     # Use model organisms core set
                    </p>
                    <p dir="auto">
                     nohup hamstr -protein -sequence_file=../ALALA.fasta -taxon=ALALA -hmmset=modelorganisms_hmmer3 -refspec=DROME -relaxed -central -force -cpu 24 -eval_blast=0.01 -eval_hmm=0.01 &amp;
                    </p>
                    <p dir="auto">
                     # Use basal metazoan core set
                    </p>
                    <p dir="auto">
                     nohup hamstr -protein -sequence_file=rename_all.fasta -taxon=CEBOR -hmmset=basalmets_hmmer3 -refspec=nemve_2309  -relaxed -central -force -cpu 48 -eval_blast=0.01 -eval_hmm=0.01  &amp;
                    </p>
                    <p dir="auto">
                     # To run many species in batch
                    </p>
                    <p dir="auto">
                     produce_MODEL_hamstr_cmd_list.pl Ctenactis_echinata
                    </p>
                    <p dir="auto">
                     for dir in */; do produce_MODEL_hamstr_cmd_list.pl $dir; done
                    </p>
                    <p dir="auto">
                     for dir in */; do produce_BASELMET_hamstr_cmd_list.pl $dir; done
                    </p>
                    <p dir="auto">
                     <strong>
                      2.2 Combine different OGs
                     </strong>
                    </p>
                    <p dir="auto">
                     grep --p "hydma" basalmets_hmmer3.fa &gt; basl_head
                    </p>
                    <p dir="auto">
                     fast_extract_seq_from_fasta.pl basalmets_hmmer3.fa basl_head &gt; basl.fasta
                    </p>
                    <p dir="auto">
                     cat ./* &gt; model.fasta
                    </p>
                    <p dir="auto">
                     mv /home/gqx/proteome/10_phylogenomic_tree/species/Hydrozoa/Hydra_vulgaris/model/fa_dir_HYVUL_modelorganisms_hmmer3_DROME_relaxed/model.fasta ./
                    </p>
                    <p dir="auto">
                     grep --p "HYVUL|" model.fasta &gt; model_head
                    </p>
                    <p dir="auto">
                     fast_extract_seq_from_fasta.pl model.fasta model_head &gt; model2.fasta
                    </p>
                    <p dir="auto">
                     makeblastdb -in model2.fasta -out model -dbtype prot
                    </p>
                    <p dir="auto">
                     nohup blastp -query basl.fasta -db model -max_target_seqs 1 -outfmt 6 -out basl_model -evalue 1e-10 -num_threads 32 &amp;
                    </p>
                    <p dir="auto">
                     perl -lane 'print $_ if ($F[2] &gt;= 70)' basl_model &gt; basl_model_70
                    </p>
                    <p dir="auto">
                     # Take the species with high coverage as a bridge
                    </p>
                    <p dir="auto">
                     # Extracted correspondence
                    </p>
                    <p dir="auto">
                     extract_OGs_relationship.pl basl_model_70
                    </p>
                    <p dir="auto">
                     cat ./connect_by_human/rela ./connect_by_hydra/rela ./connect_by_nemve/rela ./connect_by_Trichoplax/rela &gt; relationship
                    </p>
                    <p dir="auto">
                     remove_duplicate.pl relationship
                    </p>
                    <p dir="auto">
                     # Get the same genes from all species into the same file
                    </p>
                    <p dir="auto">
                     cat ./*/met/hamstrsearch_* &gt;&gt; /home/gqx/proteome/10_phylogenomic_tree/my_alignment/met_out
                    </p>
                    <p dir="auto">
                     cat ./*/*/model/hamstrsearch_* &gt; /home/gqx/proteome/10_phylogenomic_tree/my_alignment/model_out
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/^(\d+)|\w+?|/\1|/g" met_out
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/^(\d+)|\w+?|/\1|/g" model_out
                    </p>
                    <p dir="auto">
                     # The missing reference sequence is added
                    </p>
                    <p dir="auto">
                     grep --p "nemve_2309|hydma_2|triad_390" basalmets_hmmer3.fa &gt; head
                    </p>
                    <p dir="auto">
                     fast_extract_seq_from_fasta.pl basalmets_hmmer3.fa head &gt; haha.fa
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/nemve_2309/NEVEC/g" haha.fa
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/hydma_2/HYVUL/g" haha.fa
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/triad_390/TRADH/g" haha.fa
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/^&gt;(\S+)/\1|1|/g" haha.fa
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/|\n/|/g" haha.fa
                    </p>
                    <p dir="auto">
                     cat haha.fa &gt;&gt; /home/gqx/proteome/10_phylogenomic_tree/my_alignment/met_out
                    </p>
                    <p dir="auto">
                     # Check species
                    </p>
                    <p dir="auto">
                     cat model_out | cut -f 2 -d "|" &gt; 22
                    </p>
                    <p dir="auto">
                     remove_duplicate.pl 22
                    </p>
                    <p dir="auto">
                     wc -l duplicate_remove
                    </p>
                    <p dir="auto">
                     # Then split out sequences under the same OG
                    </p>
                    <p dir="auto">
                     phylogenomic_seperate_OGs_1.pl met_out
                    </p>
                    <p dir="auto">
                     phylogenomic_seperate_OGs_2.pl out
                    </p>
                    <p dir="auto">
                     cd OGs
                    </p>
                    <p dir="auto">
                     perl -p -i -e 'tr/,/\n/s' ./*
                    </p>
                    <p dir="auto">
                     for DIR in ./*txt; do phylogenomic_seperate_OGs_3.pl $DIR; done;
                    </p>
                    <p dir="auto">
                     rm ./*txt
                    </p>
                    <p dir="auto">
                     # Then find the intersection between model core set and metazoan core set
                    </p>
                    <p dir="auto">
                     cd met_model_inter_OGs/
                    </p>
                    <p dir="auto">
                     cp ../met/OGs/* ./
                    </p>
                    <p dir="auto">
                     cp ../model/OGs/* ./
                    </p>
                    <p dir="auto">
                     inter_OG_met_model_remove.pl relationship
                    </p>
                    <p dir="auto">
                     rm relationship
                    </p>
                    <p dir="auto">
                     <strong>
                      2.3 Quality control before alignment
                     </strong>
                    </p>
                    <p dir="auto">
                     mkdir trim_OGs
                    </p>
                    <p dir="auto">
                     cd trim_OGs
                    </p>
                    <p dir="auto">
                     cp ../met_model_inter_OGs/* ./
                    </p>
                    <p dir="auto">
                     # Change a more convenient sequence header
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/&gt;\d+|(\w+)|(\w+)|\d|/&gt;\1|\2/g" ./*fasta
                    </p>
                    <p dir="auto">
                     # Delete sequence length shorter than 100 aa
                    </p>
                    <p dir="auto">
                     phylogenomic_delete_short_seq.pl
                    </p>
                    <p dir="auto">
                     # Delete OGs with taxon coverage less than specific species (n = 50 in present species)
                    </p>
                    <p dir="auto">
                     ./phylogenomic_few_tax.sh
                    </p>
                    <p dir="auto">
                     # Delete X at the first and last 20 aa of the sequence
                    </p>
                    <p dir="auto">
                     ./phylogenomic_remove_X.sh
                    </p>
                    <p dir="auto">
                     <strong>
                      2.4 Alignment
                     </strong>
                    </p>
                    <p dir="auto">
                     /opt/scripts/fastas2aligned_by_mafft_hamstr_version.pl trim_OGs 16
                    </p>
                    <p dir="auto">
                     <strong>
                      2.5 Trimming
                     </strong>
                    </p>
                    <p dir="auto">
                     for dir in ./*aln; do phylogenomic_trimal_BJZ2X.pl $dir; done
                    </p>
                    <p dir="auto">
                     rm ./*aln
                    </p>
                    <p dir="auto">
                     rename .aln.BJZ2X .aln *.aln.BJZ2X
                    </p>
                    <p dir="auto">
                     for dir in ./*aln; do trimal -in
                     <math-renderer class="js-inline-math" data-run-id="3f6008964ba396a773481e473e5b66f8" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $dir -out $
                     </math-renderer>
                     {dir}.trim -automated1; done
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/^(&gt;\S+)\s+$/\1\n/" ./*aln.trim
                    </p>
                    <p dir="auto">
                     ./phylogenomic_few_tax_2.sh
                    </p>
                    <p dir="auto">
                     cd /home/gqx/proteome/10_phylogenomic_tree/single_gene_tree/alignment
                    </p>
                    <p dir="auto">
                     cp ../../my_alignment/trim_OGs/*trim ./
                    </p>
                    <p dir="auto">
                     rename .aln.trim .fa *.aln.trim
                    </p>
                    <p dir="auto">
                     cp /opt/biosoft/PhyloTreePruner/remove_short_seqs.sh ./
                    </p>
                    <p dir="auto">
                     # Change the fasta file to flat format
                    </p>
                    <p dir="auto">
                     for dir in ./*fa; do flat_the_fasta_seq.pl $dir; mv flated $dir.flat; done;
                    </p>
                    <p dir="auto">
                     rm ./*fa
                    </p>
                    <p dir="auto">
                     rename .fa.flat .fa *.fa.flat
                    </p>
                    <p dir="auto">
                     # Removing sequences that are more than 50 percent gaps
                    </p>
                    <p dir="auto">
                     ./remove_short_seqs.sh
                    </p>
                    <p dir="auto">
                     <strong>
                      2.6 Build single-gene trees using FastTree
                     </strong>
                    </p>
                    <p dir="auto">
                     phylogenomic_FastTree_OG.pl alignment 16
                    </p>
                    <p dir="auto">
                     FastTreeMP -slow -gamma $ARGV[0]/$_ &gt; $ARGV[0]/$1.tre
                    </p>
                    <p dir="auto">
                     <strong>
                      2.7 Use TreSpEx to remove Long-branch attraction (LBA) and saturation analyses
                     </strong>
                    </p>
                    <p dir="auto">
                     cd TreSpEx/
                    </p>
                    <p dir="auto">
                     mkdir single_tree
                    </p>
                    <p dir="auto">
                     cd single_tree
                    </p>
                    <p dir="auto">
                     # TreSpEx needs the RAXML output format
                    </p>
                    <p dir="auto">
                     cp /home/gqx/proteome/10_phylogenomic_tree/ML_analysis/76cov_105tx_146og/out.fasta ./
                    </p>
                    <p dir="auto">
                     cp /home/gqx/proteome/10_phylogenomic_tree/ML_analysis/76cov_105tx_146og/new_partition.txt ./
                    </p>
                    <p dir="auto">
                     cp charset new_partition.txt
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/\s;\s//g" new_partition.txt
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/\s-\s/-/g" new_partition.txt
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/charset/AUTO,/g" new_partition.txt
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/^\S+\s//g" new_partition.txt
                    </p>
                    <p dir="auto">
                     ./AMAS.py split -l new_partition.txt -i out.fasta -f fasta -d aa
                    </p>
                    <p dir="auto">
                     rm out.fasta new_partition.txt
                    </p>
                    <p dir="auto">
                     for DIR in ./*fas; do  mkdir ${DIR}_file; done;
                    </p>
                    <p dir="auto">
                     for DIR in ./*fas; do  mv
                     <math-renderer class="js-inline-math" data-run-id="3f6008964ba396a773481e473e5b66f8" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $DIR  $
                     </math-renderer>
                     {DIR}_file; done;
                    </p>
                    <p dir="auto">
                     for DIR in ./*file; do  cd $DIR; remove_wenhao_seq.pl *fas; awk 'BEGIN {RS = "&gt;" ; FS = "\n" ; ORS = ""} $2 {print "&gt;"$0}' out &gt; output; rm out; cd ..;  done;
                    </p>
                    <p dir="auto">
                     for DIR in ./*file; do  cd $DIR; mv output out_*; cd ..;  done;
                    </p>
                    <p dir="auto">
                     # Run iqtree
                    </p>
                    <p dir="auto">
                     nohup iqtree-omp -s ./*fas -wbt -bb 10000 -nt 1 -st AA &amp;
                    </p>
                    <p dir="auto">
                     # Prepare the species name list, tree name list and alignment name list
                    </p>
                    <p dir="auto">
                     cp /home/gqx/proteome/10_phylogenomic_tree/concatenated/146/otu/otu-name.otu ./
                    </p>
                    <p dir="auto">
                     ls -1 ./*contree &gt; tree_name
                    </p>
                    <p dir="auto">
                     # Make sure the format of alignments is relax phylip
                    </p>
                    <p dir="auto">
                     cd /home/gqx/proteome/10_phylogenomic_tree/ML_analysis/76cov_105tx_146og/TreSpEx/Longbranch_and_Saturation
                    </p>
                    <p dir="auto">
                     mkdir align
                    </p>
                    <p dir="auto">
                     cd align
                    </p>
                    <p dir="auto">
                     cp ../../single_tree/*/*fas ./
                    </p>
                    <p dir="auto">
                     for dir in ./*fas; do convertFasta2Phylip.sh
                     <math-renderer class="js-inline-math" data-run-id="3f6008964ba396a773481e473e5b66f8" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $dir &amp;gt; $
                     </math-renderer>
                     {dir}.phy ; done
                    </p>
                    <p dir="auto">
                     ls -1 ./*phy &gt; align_name
                    </p>
                    <p dir="auto">
                     mv align_name ../
                    </p>
                    <p dir="auto">
                     mv ./*phy ../
                    </p>
                    <p dir="auto">
                     cd ..
                    </p>
                    <p dir="auto">
                     rm align -rf
                    </p>
                    <p dir="auto">
                     # Perform LBA analysis
                    </p>
                    <p dir="auto">
                     TreSpEx.v1.1.pl -fun e -ipt tree_name -tf otu-name.otu
                    </p>
                    <p dir="auto">
                     # Saturation analysis
                    </p>
                    <p dir="auto">
                     TreSpEx.v1.1.pl -fun g -ipt tree_name -ipa align_name
                    </p>
                    <p dir="auto">
                     <strong>
                      2.7.1 Use density map to remove LBA (in R)
                     </strong>
                    </p>
                    <p dir="auto">
                     data &lt;- read.table("LB_scores_summary_perPartition.txt", header=T,sep="\t",row.names = 1)
                    </p>
                    <p dir="auto">
                     d &lt;- density(data$LB_score_upper_quartile)
                    </p>
                    <p dir="auto">
                     pdf("density_upper.pdf",width=6,height=6)
                    </p>
                    <p dir="auto">
                     plot(d, main=NA, xlab="Mean value of upper quartile of LB scores", lwd=1.5)
                    </p>
                    <p dir="auto">
                     # plot(d, main=NA, xlab="Standard deviation of LB scores", lwd=1.5)
                    </p>
                    <p dir="auto">
                     segments(x0=80.81,y0=0,x1=80.81,y1=0.0125,lwd=1.5, lty=2)
                    </p>
                    <p dir="auto">
                     abline(h=0,col="grey")
                    </p>
                    <p dir="auto">
                     # Filling detected areas with red
                    </p>
                    <p dir="auto">
                     x1 &lt;- min(which(d$x &gt;= 80.81))
                    </p>
                    <p dir="auto">
                     x2 &lt;- max(which(d$x &lt; 140))
                    </p>
                    <p dir="auto">
                     with(d, polygon(x=c(x[c(x1,x1:x2,x2)]), y= c(0, y[x1:x2], 0), col="red"))
                    </p>
                    <p dir="auto">
                     dev.off()
                    </p>
                    <p dir="auto">
                     <strong>
                      2.7.2 Use density map to do saturation analysis (in R)
                     </strong>
                    </p>
                    <p dir="auto">
                     Correlation_Results/Correlation_Slope_Summary.txt
                    </p>
                    <p dir="auto">
                     data &lt;- read.table("Correlation_Slope_Summary.txt", header=T,sep="\t",row.names = 1)
                    </p>
                    <p dir="auto">
                     d &lt;- density(data$Slope)
                    </p>
                    <p dir="auto">
                     pdf("density_slope.pdf",width=6,height=6)
                    </p>
                    <p dir="auto">
                     plot(d, main=NA, xlab="Slope of linear regression", lwd=1.5)
                    </p>
                    <p dir="auto">
                     segments(x0= 0.224,y0=0,x1= 0.224,y1=1.5,lwd=1.5, lty=2)
                    </p>
                    <p dir="auto">
                     abline(h=0,col="grey")
                    </p>
                    <p dir="auto">
                     x1 &lt;- min(which(d$x &gt;= -1))
                    </p>
                    <p dir="auto">
                     x2 &lt;- max(which(d$x &lt; 0.224))
                    </p>
                    <p dir="auto">
                     with(d, polygon(x=c(x[c(x1,x1:x2,x2)]), y= c(0, y[x1:x2], 0), col="red"))
                    </p>
                    <p dir="auto">
                     dev.off()
                    </p>
                    <p dir="auto">
                     <strong>
                      2.7.3 Use heatmap to analyze species-specific genes (in R)
                     </strong>
                    </p>
                    <p dir="auto">
                     library(gplots)
                    </p>
                    <p dir="auto">
                     data &lt;- read.table("9999.txt", header=T,sep="\t",row.names = 1)
                    </p>
                    <p dir="auto">
                     data &lt;- as.matrix(data)
                    </p>
                    <p dir="auto">
                     colors = c(seq(min(data),0,length=100),seq(1,200,length=100),seq(201,500,length=100),seq(501,700,length=100),seq(701,1000,length=100),seq(1001,1200,length=100),seq(1201,1600,length=100),seq(9999,9999,1))
                    </p>
                    <p dir="auto">
                     my_palette &lt;- colorRampPalette(c("orange","yellow","green", "cyan","blue","purple","red","white"))(n = 700)
                    </p>
                    <p dir="auto">
                     pdf("9999.pdf",width=18,height=18)
                    </p>
                    <p dir="auto">
                     heatmap.2(data, density.info="density", col=my_palette, symkey=F, symm=F, symbreaks=F,scale="none", trace="none", breaks=colors, keysize=1)
                    </p>
                    <p dir="auto">
                     dev.off()
                    </p>
                    <p dir="auto">
                     <strong>
                      2.8 Use PhyloTreePruner to remove paralogs
                     </strong>
                    </p>
                    <p dir="auto">
                     # Change the header to be recognizable for PhyloTreePruner
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/(\d+)|//g" 1.aln.trim
                    </p>
                    <p dir="auto">
                     # Feed the result of RAxML (RAxML_bipartitions.out) to PhyloTreePruner
                    </p>
                    <p dir="auto">
                     java -cp /usr/local/bin/ PhyloTreePruner RAxML_bipartitions.out 3 2.aln.trim 0.5 r
                    </p>
                    <p dir="auto">
                     java -cp /usr/local/bin/ PhyloTreePruner 113684.tre 71 113684.aln 0.5 r
                    </p>
                    <p dir="auto">
                     cd single_gene_tree/
                    </p>
                    <p dir="auto">
                     mkdir paralog_screen
                    </p>
                    <p dir="auto">
                     cd paralog_screen/
                    </p>
                    <p dir="auto">
                     cp ../alignment/*long ./
                    </p>
                    <p dir="auto">
                     cp ../alignment/*tre ./
                    </p>
                    <p dir="auto">
                     # Change the fasta file format to flat fasta
                    </p>
                    <p dir="auto">
                     for dir in ./*fa.long; do fasta_flat_to_multiline.py $dir 60; done;
                    </p>
                    <p dir="auto">
                     rm ./*fa.long
                    </p>
                    <p dir="auto">
                     rename .fa.long_multi-line.fasta  .fa.long *.fa.long_multi-line.fasta
                    </p>
                    <p dir="auto">
                     cd ..
                    </p>
                    <p dir="auto">
                     # Run PhyloTreePruner
                    </p>
                    <p dir="auto">
                     phylogenomic_PhyloTreePruner.pl paralog_screen 8
                    </p>
                    <p dir="auto">
                     <strong>
                      2.9 Filtering for unique sequences
                     </strong>
                    </p>
                    <p dir="auto">
                     SCaFoS
                    </p>
                    <p dir="auto">
                     export SCAFOS=/opt/biosoft/SCaFoS
                    </p>
                    <p dir="auto">
                     cpan install Tk
                    </p>
                    <p dir="auto">
                     # Install Tree-Puzzle
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/|/@/g" ./*fa
                    </p>
                    <p dir="auto">
                     scafos in=fasta out=otu
                    </p>
                    <p dir="auto">
                     scafos in=fasta out=out otu=otu/otu-name.otu format=fp t=10
                    </p>
                    <p dir="auto">
                     # out.fasta is the result
                    </p>
                    <p dir="auto">
                     # Change the style to MARE style
                    </p>
                    <p dir="auto">
                     phylogenomic_scafos2MARE_charset.pl out.len
                    </p>
                    <p dir="auto">
                     <strong>
                      2.10 Visualize the phylogenomic matrices
                     </strong>
                    </p>
                    <p dir="auto">
                     MARE charset out.fasta -m -t
                    </p>
                    <p dir="auto">
                     <strong>
                      2.11 Exclude species with gene missing rate &gt; 85%
                     </strong>
                    </p>
                    <p dir="auto">
                     cd /home/gqx/proteome/10_phylogenomic_tree/concatenated/50_taxon_coverage_remove_poor_species/out
                    </p>
                    <p dir="auto">
                     mkdir count
                    </p>
                    <p dir="auto">
                     cd count
                    </p>
                    <p dir="auto">
                     split_fasta_file_averagely.pl ../out.fasta 118
                    </p>
                    <p dir="auto">
                     rm out.fasta
                    </p>
                    <p dir="auto">
                     for DIR in ./*fasta; do  mkdir ${DIR}_file; done;
                    </p>
                    <p dir="auto">
                     for DIR in ./*fasta; do  mv
                     <math-renderer class="js-inline-math" data-run-id="3f6008964ba396a773481e473e5b66f8" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $DIR  $
                     </math-renderer>
                     {DIR}_file; done;
                    </p>
                    <p dir="auto">
                     for DIR in ./*; do cd $DIR; AMAS.py summary -i ./*fasta -f fasta -d aa; cd ..; done;
                    </p>
                    <p dir="auto">
                     for DIR in ./*; do cd $DIR; cat summary.txt | cut -f6 &gt; missing | grep "&gt;" *fasta &gt;&gt; missing; cd ..; done;
                    </p>
                    <p dir="auto">
                     for DIR in ./*; do cd $DIR; perl -p -i -e "s/\n/\t/g" missing; cd ..; done;
                    </p>
                    <p dir="auto">
                     cat ./*/missing &gt;&gt; all_missing
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/\tMissing/\nMissing/g" all_missing
                    </p>
                    <p dir="auto">
                     rm ./*file -rf
                    </p>
                    <p dir="auto">
                     perl -lane 'print $_ if ($F[1] gt 85)' all_missing &gt; all_missing_85
                    </p>
                    <p dir="auto">
                     # Then delete species in the out.fasta file
                    </p>
                    <p dir="auto">
                     <strong>
                      3. Phylogenetic tree construction
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      3.1 Run RAXML
                     </strong>
                    </p>
                    <p dir="auto">
                     # LG+G4+F is a site-homogeneous model, while CAT+G4 is a site-heterogeneous model. It is possible to use PartitionFinder with different parameters on each module separately, but this hybrid homogeneous is not equal to the heterogeneous model. The evolutionary context within each module is also different.
                    </p>
                    <p dir="auto">
                     # CAT+G4 also considers that the evolutionary context may be different even within a module.
                    </p>
                    <p dir="auto">
                     # The optimal model is calculated automatically; and the alpha parameters of gamma distribution and stable frequency differ according to each gene.
                    </p>
                    <p dir="auto">
                     phylogenomic_scafos2RAXML_partition.pl out.len
                    </p>
                    <p dir="auto">
                     nohup raxmlHPC-PTHREADS-SSE3 -T 48 -s out.fasta -n 50cov_105tx_474og -f a -N 100 -x 12345 -p 12345 -m PROTGAMMAAUTO -q new_partition.txt &amp;
                    </p>
                    <p dir="auto">
                     <strong>
                      3.2 Run IQ-TREE
                     </strong>
                    </p>
                    <p dir="auto">
                     # Change the charst into raxml new_partition.txt
                    </p>
                    <p dir="auto">
                     cp charset new_partition.txt
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/\s;\s//g" new_partition.txt
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/\s-\s/-/g" new_partition.txt
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/charset/protein,/g" new_partition.txt
                    </p>
                    <p dir="auto">
                     nohup iqtree-omp -s out.fasta -wbt -bb 10000 -nt AUTO -st AA -spp new_partition.txt  -m MFP &amp;
                    </p>
                    <p dir="auto">
                     <strong>
                      3.3 Run Phylobayes
                     </strong>
                    </p>
                    <p dir="auto">
                     /opt/biosoft/RAxML_v8.2.9/usefulScripts/convertFasta2Phylip.sh ../allSingleCopyOrthologsAlign.fasta-gb &gt; allSingleCopyOrthologsAlign.fasta-gb.phy
                    </p>
                    <p dir="auto">
                     nohup mpirun -np 16 /opt/biosoft/phylobayes-mpi-1.7b/data/pb_mpi -d allSingleCopyOrthologsAlign.fasta-gb.phy -cat -gtr -x 10 20000 gqx &amp;
                    </p>
                    <p dir="auto">
                     nohup mpirun -np 16 /opt/biosoft/phylobayes-mpi-1.7b/data/pb_mpi -d allSingleCopyOrthologsAlign.fasta-gb.phy -cat -gtr -x 10 20000 gqx2&amp;
                    </p>
                    <p dir="auto">
                     # Check convergence and output results
                    </p>
                    <p dir="auto">
                     nohup /opt/biosoft/phylobayes-mpi-1.7b/data/bpcomp -x 1000 10 gqx gqx2 -o test &amp;
                    </p>
                    <p dir="auto">
                     nohup /opt/biosoft/phylobayes-mpi-1.7b/data/tracecomp -x 1000 gqx gqx2 -o test &amp;
                    </p>
                    <p dir="auto">
                     # effsize &gt; 300, max rel_diff &lt; 0.1, converged
                    </p>
                    <p dir="auto">
                     <strong>
                      4. Remove bilaterial contamination from myxozoan sequences
                     </strong>
                    </p>
                    <p dir="auto">
                     mkdir contamination_blast
                    </p>
                    <p dir="auto">
                     cd contamination_blast
                    </p>
                    <p dir="auto">
                     # Make seed database: bilaterian.faa and cnidarian.faa
                    </p>
                    <p dir="auto">
                     replace_header_for_cdhit.pl -c bilaterian_100 -t bilaterian
                    </p>
                    <p dir="auto">
                     replace_header_for_cdhit.pl -c cnidarian_100 -t cnidarian
                    </p>
                    <p dir="auto">
                     cat ./* &gt; all.fasta
                    </p>
                    <p dir="auto">
                     # Remove the non-myxozoan sequences in each matrices
                    </p>
                    <p dir="auto">
                     cp ../fasta/ ./ -R
                    </p>
                    <p dir="auto">
                     cd fasta
                    </p>
                    <p dir="auto">
                     for dir in ./*fa ; do perl -p -i -e "s/(\d)\n/\1\t/g" $dir; done
                    </p>
                    <p dir="auto">
                     for dir in ./*fa ; do phylogenomic_filter_no_myxozoan_seq.pl $dir; done
                    </p>
                    <p dir="auto">
                     cd ..
                    </p>
                    <p dir="auto">
                     mkdir myxo_fasta
                    </p>
                    <p dir="auto">
                     mv fasta/*filter myxo_fasta/
                    </p>
                    <p dir="auto">
                     cd ..
                    </p>
                    <p dir="auto">
                     makeblastdb -in all.fasta -out all -dbtype prot
                    </p>
                    <p dir="auto">
                     ./phylogenomic_parafly_contamination_blast.pl myxo_fasta/
                    </p>
                    <p dir="auto">
                     ParaFly -c blast_cmds -CPU 4 -shuffle -v
                    </p>
                    <p dir="auto">
                     cd myxo_fasta/
                    </p>
                    <p dir="auto">
                     for dir in ./*out ; do export LANG=C; export LC_ALL=C; sort -k1,1 -k12,12gr -k11,11g -k3,3gr
                     <math-renderer class="js-inline-math" data-run-id="3f6008964ba396a773481e473e5b66f8" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $dir | sort -u -k1,1 --merge &amp;gt; $
                     </math-renderer>
                     {dir}_best ; done
                    </p>
                    <p dir="auto">
                     # Get the bad list
                    </p>
                    <p dir="auto">
                     for dir in ./*out_best ; do perl -ne '/bilaterian/ &amp;&amp; print'
                     <math-renderer class="js-inline-math" data-run-id="3f6008964ba396a773481e473e5b66f8" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $dir &amp;gt; $
                     </math-renderer>
                     {dir}_delete ; done
                    </p>
                    <p dir="auto">
                     cd ..
                    </p>
                    <p dir="auto">
                     mkdir delete_list
                    </p>
                    <p dir="auto">
                     mv myxo_fasta/*delete delete_list/
                    </p>
                    <p dir="auto">
                     cd delete_list/
                    </p>
                    <p dir="auto">
                     #find ./* -size  0 -print0 |xargs -0 rm
                    </p>
                    <p dir="auto">
                     for dir in ./*delete ; do perl -p -i -e "s/^(\S+).*$/\1/g" $dir ; done
                    </p>
                    <p dir="auto">
                     # If you only want to remove certain species
                    </p>
                    <p dir="auto">
                     for dir in ./*out_best_delete ; do perl -ne '/HESPP/ &amp;&amp; print'
                     <math-renderer class="js-inline-math" data-run-id="3f6008964ba396a773481e473e5b66f8" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $dir &amp;gt; $
                     </math-renderer>
                     {dir}_HESPP ; done
                    </p>
                    <p dir="auto">
                     cd /home/gqx/proteome/11_big_tree/ML_analysis/optimized_105tx_232og_3/fasta
                    </p>
                    <p dir="auto">
                     cp ../contamination_blast/delete_list/* ./
                    </p>
                    <p dir="auto">
                     phylogenomic_remove_contaminant_of_no_cnidarian.sh
                    </p>
                    <p dir="auto">
                     rm ./*fa
                    </p>
                    <p dir="auto">
                     rm ./*best_delete
                    </p>
                    <p dir="auto">
                     rename .fa_revised .fa *.fa_revised
                    </p>
                    <p dir="auto">
                     <strong>
                      5. Approximately unbiased test (AU-test) of phylogenetic tree selection
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      5.1 Make constraint tree
                     </strong>
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/:0.\d+//g" bestree.nwk
                    </p>
                    <p dir="auto">
                     phylogenomic_constraint_tree_generator.pl tree_topology mono_list bestree.nwk
                    </p>
                    <p dir="auto">
                     # constraint.tre is the result file
                    </p>
                    <p dir="auto">
                     <strong>
                      5.2 AU-test using RAXML and Consel
                     </strong>
                    </p>
                    <p dir="auto">
                     # Provide the main tree
                    </p>
                    <p dir="auto">
                     nohup raxmlHPC-PTHREADS-SSE3 -T 48 -s out.fasta -n 50cov_105tx_474og -f a -N 100 -p 12345 -x 12345 -m PROTGAMMAAUTO -g constraint.tre -q new_partition.txt &amp;
                    </p>
                    <p dir="auto">
                     # Provide constraint tree to RAXML
                    </p>
                    <p dir="auto">
                     # Produce likehood value from the result tree
                    </p>
                    <p dir="auto">
                     cat ../normal/RAxML_bipartitions.50cov_105tx_474og ../constraint_1/RAxML_bipartitions.50cov_105tx_474og ../constraint_2/RAxML_bipartitions.50cov_105tx_474og ../constraint_3/RAxML_bipartitions.50cov_105tx_474og &gt; tree
                    </p>
                    <p dir="auto">
                     nohup raxmlHPC-PTHREADS-SSE3 -s out.fasta -z tree -m PROTGAMMAAUTO -T 6 -f G -n  50cov_105tx_474og  -q new_partition.txt &amp;
                    </p>
                    <p dir="auto">
                     # AU-test using Consel
                    </p>
                    <p dir="auto">
                     mv RAxML_perSiteLLs.50cov_105tx_474og RAxML_perSiteLLs.sitelh
                    </p>
                    <p dir="auto">
                     makermt --puzzle RAxML_perSiteLLs.sitelh
                    </p>
                    <p dir="auto">
                     consel RAxML_perSiteLLs.rmt
                    </p>
                    <p dir="auto">
                     catpv -s 9 RAxML_perSiteLLs.pv
                    </p>
                    <p dir="auto">
                     <strong>
                      5.3 AU-test using IQ-tree
                     </strong>
                    </p>
                    <p dir="auto">
                     # The main difference between IQ-tree and RAXML is that IQ-tree integrates the functionality of Consel. The second and third steps of RAXML are combined into one step
                    </p>
                    <p dir="auto">
                     # Get the main tree
                    </p>
                    <p dir="auto">
                     nohup iqtree-1.6 -s new.fasta -wbt -bb 10000 -nt 6 -st AA -spp new_partition.txt &amp;
                    </p>
                    <p dir="auto">
                     # Provide the constraint tree to IQ-tree and get the result tree
                    </p>
                    <p dir="auto">
                     nohup iqtree-1.6 -s new.fasta -wbt -bb 10000 -nt 6 -st AA -spp new_partition.txt -g constraint.tre &amp;
                    </p>
                    <p dir="auto">
                     # Use the constraint tree in former step as guide tree and conduct PMSF optimization
                    </p>
                    <p dir="auto">
                     mv new_partition.txt.contree guide.tre
                    </p>
                    <p dir="auto">
                     # Here is a question: when doing PMSF, do you need to use both partitioned and C60,C20 (similar to CAT, site-heterogeneous substitution models), IQ-treee allows to do these at the same time, but each partition and C20 are not linked, so for most data will be over-parameterized so no partition is used here
                    </p>
                    <p dir="auto">
                     # Use LG+G4+F+C2 model
                    </p>
                    <p dir="auto">
                     nohup iqtree-1.6 -s new.fasta -nt 16 -st AA -m "LG+G4+FMIX{C20pi1:1:0.0559910600,C20pi2:1:0.0514824870,C20pi3:1:0.0812922124,C20pi4:1:0.0721976867,C20pi5:1:0.0556718858,C20pi6:1:0.0331003080,C20pi7:1:0.0589501763,C20pi8:1:0.0263756889,C20pi9:1:0.0307584220,C20pi10:1:0.0376701125,C20pi11:1:0.0303058290,C20pi12:1:0.0808775576,C20pi13:1:0.0263349134,C20pi14:1:0.0579101455,C20pi15:1:0.0371248064,C20pi16:1:0.0586867766,C20pi17:1:0.0561479138,C20pi18:1:0.0349810886,C20pi19:1:0.0544937394,C20pi20:1:0.0596471901}+F" -ft guide.tre -g constrain.tre &amp;
                    </p>
                    <p dir="auto">
                     # Running AU-test
                    </p>
                    <p dir="auto">
                     cat normal/new_partition.txt.contree constraint_1/PMSF/new.fasta.treefile constraint_2/PMSF/new.fasta.treefile constraint_3/PMSF/new.fasta.treefile &gt; AU/tree
                    </p>
                    <p dir="auto">
                     nohup iqtree-1.6 -s new.fasta -nt 16 -m "LG+G4+FMIX{C20pi1:1:0.0559910600,C20pi2:1:0.0514824870,C20pi3:1:0.0812922124,C20pi4:1:0.0721976867,C20pi5:1:0.0556718858,C20pi6:1:0.0331003080,C20pi7:1:0.0589501763,C20pi8:1:0.0263756889,C20pi9:1:0.0307584220,C20pi10:1:0.0376701125,C20pi11:1:0.0303058290,C20pi12:1:0.0808775576,C20pi13:1:0.0263349134,C20pi14:1:0.0579101455,C20pi15:1:0.0371248064,C20pi16:1:0.0586867766,C20pi17:1:0.0561479138,C20pi18:1:0.0349810886,C20pi19:1:0.0544937394,C20pi20:1:0.0596471901}+F" -z tree -n 0 -zb 10000 -au &amp;
                    </p>
                    <p dir="auto">
                     # Results are summarized in new.fasta.iqtree
                    </p>
                    <p dir="auto">
                     <strong>
                      6. Fast-evolving site removal analysis
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      6.1 Install DistEst
                     </strong>
                    </p>
                    <p dir="auto">
                     tar zxf dist_est.tar.gz
                    </p>
                    <p dir="auto">
                     # distEST needs two function library, id05-1.0.0.tar.gz and ve11-1.0.0.tar.gz
                    </p>
                    <p dir="auto">
                     # unzip these files, copy the id05i.f, ve11d.f in src directory into the dist_estv1.1 directory
                    </p>
                    <p dir="auto">
                     make
                    </p>
                    <p dir="auto">
                     # If you see information below, you succeed
                    </p>
                    <p dir="auto">
                     gcc  -O3  -c dcdflib.c -o dcdflib.o
                    </p>
                    <p dir="auto">
                     gcc  -O3  -c dist_est.c -o dist_est.o
                    </p>
                    <p dir="auto">
                     gcc  -O3  -c dist_est_fn.c -o dist_est_fn.o
                    </p>
                    <p dir="auto">
                     gcc  -O3  -c ipmpar.c -o ipmpar.o
                    </p>
                    <p dir="auto">
                     gfortran -O3  -c id05i.f -o id05i.o
                    </p>
                    <p dir="auto">
                     gfortran -O3  -c ve11d.f -o ve11d.o
                    </p>
                    <p dir="auto">
                     gfortran -O3 -o dist_est dcdflib.o dist_est.o dist_est_fn.o ipmpar.o id05i.o ve11d.o -lm
                    </p>
                    <p dir="auto">
                     <strong>
                      6.2 Calculating evolutionary rates in DistEst
                     </strong>
                    </p>
                    <p dir="auto">
                     # Prepare hsp70.ctl， hsp70.dat， hsp70.fitchml.tre
                    </p>
                    <p dir="auto">
                     dist_est hsp70.ctl
                    </p>
                    <p dir="auto">
                     # hst70.ctl is as follows
                    </p>
                    <p dir="auto">
                     treefile =  hsp70.fitchml.tre  * treefile
                    </p>
                    <p dir="auto">
                     seqfile = hsp70.dat            * sequence data
                    </p>
                    <p dir="auto">
                     nchar = 20             * amino acid data
                    </p>
                    <p dir="auto">
                     model =  3             * empirical + F
                    </p>
                    <p dir="auto">
                     aaRatefile = lg.dat * LG substitution model
                    </p>
                    <p dir="auto">
                     nrate = 101            * number of rates
                    </p>
                    <p dir="auto">
                     ub = 10.0              * upper bound for rates
                    </p>
                    <p dir="auto">
                     # The resule file: DE.dat  rate_est.dat
                    </p>
                    <p dir="auto">
                     <strong>
                      6.3 Remove sites from fast to slow
                     </strong>
                    </p>
                    <p dir="auto">
                     AMAS.py convert -i hsp70.dat -f phylip -d aa -u fasta
                    </p>
                    <p dir="auto">
                     flat_the_fasta_seq.pl metazoan.phy-out.fas
                    </p>
                    <p dir="auto">
                     mv flated alignment.fasta
                    </p>
                    <p dir="auto">
                     phylogenomic_increment_remove_fast_evolving_sites.pl &lt;fasta_alignment&gt;  &lt;rate_est.dat&gt;  &lt;num_sites_per&gt;
                    </p>
                    <p dir="auto">
                     # For instance, phylogenomic_increment_remove_fast_evolving_sites.pl alignment.fasta rate_est.dat 100
                    </p>
                    <p dir="auto">
                     <strong>
                      6.4 Build phylogenetic tree for each matrices
                     </strong>
                    </p>
                    <p dir="auto">
                     nohup iqtree-omp -s out.fasta -wbt -bb 10000 -nt 24 -st AA -m LG+F+G &amp;
                    </p>
                    <p dir="auto">
                     # You can also use PMSF model
                    </p>
                    <p dir="auto">
                     nohup iqtree-omp -s out.fasta -wbt -bb 10000 -nt 24 -st AA -m LG+C60+F+G -ft guide.tre &amp;
                    </p>
                    <p dir="auto">
                     <strong>
                      6.5 Plot fast-evolving site removal analysis using R
                     </strong>
                    </p>
                    <p dir="auto">
                     library(ggplot2)
                    </p>
                    <p dir="auto">
                     data &lt;- read.csv("table.csv", header=T)
                    </p>
                    <p dir="auto">
                     data &lt;- data.frame(A=c(data[1]),B=c(data[2]),C=c(data[3]))
                    </p>
                    <p dir="auto">
                     ggplot(data=data, aes(x=seq, y=bp, group=supp))+geom_line(aes(color=supp))+scale_color_brewer(palette="Dark2")+scale_x_continuous(breaks = c(0, 3500, 7000,10500,14000,17500,21000,24500,28000,31500,35000,38500,42000,45500,49000))+theme_set(theme_bw())+theme(legend.position="bottom")
                    </p>
                    <p dir="auto">
                     ggsave("plot2.pdf", width=8, height=4)
                    </p>
                    <p dir="auto">
                     <strong>
                      7. Molecular clock analysis using r8s
                     </strong>
                    </p>
                    <p dir="auto">
                     # Strick model: A rigorous model that assumes that all branches evolve at the same rate.
                    </p>
                    <p dir="auto">
                     # Fixed local clock: a type of relaxed model, assuming that the taxa of one or more branches are distinct from the whole, but of course, they must be monophyletic.
                    </p>
                    <p dir="auto">
                     # Uncorrelated relaxed clock: a type of relaxed model, where each branch has its own evolutionary rate and is uncorrelated.
                    </p>
                    <p dir="auto">
                     # Random local clock: a model between the discrete model and the relaxed model.
                    </p>
                    <p dir="auto">
                     # To use r8s,aAt least one internal node needs to be fixed. The whole process includes the first validation to determine the smooth value and the formal run using the optimal smooth value. cvstart=0 cvinc=0.5 cvnum=10 means to start from 0 and go 10 at 0.5 intervals, taking values to smooth. It is better to set cvstart as zero, do not choose a negative number. In addition, if it is a large phylogenetic tree, you must have collapse parameter to merge branches. Otherwise there will be some branches because of very short and be treated as zero length, which can not be calculated.
                    </p>
                    <p dir="auto">
                     <strong>
                      7.1 Prepare the configuration files r8s_in.txt
                     </strong>
                    </p>
                    <p dir="auto">
                     cp /opt/biosoft/r8s1.81/r8s_in_template.txt ./
                    </p>
                    <p dir="auto">
                     mv r8s_in_template.txt r8s_in.txt
                    </p>
                    <p dir="auto">
                     # Then edit the configuration files based on conditions
                    </p>
                    <p dir="auto">
                     <strong>
                      7.1.1 If you have 2 or more fossil fix points (i.e. clear fossil time points)
                     </strong>
                    </p>
                    <p dir="auto">
                     # First prepare r8s_in_validation.txt for first validation to set the bet smooth value
                    </p>
                    <p dir="auto">
                     #NEXUS
                    </p>
                    <p dir="auto">
                     begin trees;
                    </p>
                    <p dir="auto">
                     tree tree_1 = [&amp;R] ((umay:0.377714,(pchr:0.126947,((cput:0.143183,slac:0.089884):0.040327,(scom:0.181745,(abis:0.150637,(lbic:0.088905,ccin:0.132642):0.022419):0.025092):0.02715):0.035777):0.207622):0.142535,((scer:0.463674,psti:0.376001):0.331119,((anig:0.213676,snod:0.244829):0.042769,(tree:0.147333,(ncra:0.134451,cpar:0.146872):0.034971):0.111483):0.206145):0.142535);
                    </p>
                    <p dir="auto">
                     end;
                    </p>
                    <p dir="auto">
                     begin r8s;
                    </p>
                    <p dir="auto">
                     blformat lengths=persite nsites=320272 ulrametric=no;
                    </p>
                    <p dir="auto">
                     collapse;
                    </p>
                    <p dir="auto">
                     MRCA AGA ccin lbic scom;
                    </p>
                    <p dir="auto">
                     MRCA ASC psti anig cpar snod tree;
                    </p>
                    <p dir="auto">
                     MRCA BOL slac cput;
                    </p>
                    <p dir="auto">
                     fixage taxon=AGA age=122.74;
                    </p>
                    <p dir="auto">
                     fixage taxon=ASC age=517.55;
                    </p>
                    <p dir="auto">
                     fixage taxon=BOL age=104.23;
                    </p>
                    <p dir="auto">
                     divtime method=PL crossv=yes fossilfixed=yes cvstart=0 cvinc=0.5 cvnum=10;
                    </p>
                    <p dir="auto">
                     showage;
                    </p>
                    <p dir="auto">
                     describe plot=cladogram;
                    </p>
                    <p dir="auto">
                     describe plot=chrono_description;
                    </p>
                    <p dir="auto">
                     end;
                    </p>
                    <p dir="auto">
                     # First validation
                    </p>
                    <p dir="auto">
                     r8s -b -f r8s_in_validation.txt &gt; cross_validation_1
                    </p>
                    <p dir="auto">
                     # Find the line with the smallest raw error, and then find the smooth value in the same line
                    </p>
                    <p dir="auto">
                     # Then revise the configuration file as a new file: r8s_in_true_run.txt
                    </p>
                    <p dir="auto">
                     #NEXUS
                    </p>
                    <p dir="auto">
                     begin trees;
                    </p>
                    <p dir="auto">
                     tree tree_1 = [&amp;R] ((umay:0.377714,(pchr:0.126947,((cput:0.143183,slac:0.089884):0.040327,(scom:0.181745,(abis:0.150637,(lbic:0.088905,ccin:0.132642):0.022419):0.025092):0.02715):0.035777):0.207622):0.142535,((scer:0.463674,psti:0.376001):0.331119,((anig:0.213676,snod:0.244829):0.042769,(tree:0.147333,(ncra:0.134451,cpar:0.146872):0.034971):0.111483):0.206145):0.142535);
                    </p>
                    <p dir="auto">
                     end;
                    </p>
                    <p dir="auto">
                     begin r8s;
                    </p>
                    <p dir="auto">
                     blformat lengths=persite nsites=320272 ulrametric=no;
                    </p>
                    <p dir="auto">
                     collapse;
                    </p>
                    <p dir="auto">
                     MRCA AGA ccin lbic scom;
                    </p>
                    <p dir="auto">
                     MRCA ASC psti anig cpar snod tree;
                    </p>
                    <p dir="auto">
                     MRCA BOL slac cput;
                    </p>
                    <p dir="auto">
                     fixage taxon=AGA age=122.74;
                    </p>
                    <p dir="auto">
                     fixage taxon=ASC age=517.55;
                    </p>
                    <p dir="auto">
                     fixage taxon=BOL age=104.23;
                    </p>
                    <p dir="auto">
                     set smoothing=4;
                    </p>
                    <p dir="auto">
                     set num_time_guesses=3;
                    </p>
                    <p dir="auto">
                     set checkGradient=yes;
                    </p>
                    <p dir="auto">
                     divtime method=PL algorithm=TN;
                    </p>
                    <p dir="auto">
                     showage;
                    </p>
                    <p dir="auto">
                     describe plot=cladogram;
                    </p>
                    <p dir="auto">
                     describe plot=chrono_description;
                    </p>
                    <p dir="auto">
                     end;
                    </p>
                    <p dir="auto">
                     # Then run r8s
                    </p>
                    <p dir="auto">
                     r8s -b -f r8s_in_true_run.txt &gt; true_run_1
                    </p>
                    <p dir="auto">
                     # Extract the time tree from the result file
                    </p>
                    <p dir="auto">
                     perl -e 'while (&lt;&gt;) { if (m/tree tree_1 = (.*);/) { my $tree = $1; $tree =~ s/(BOL|AGA|ASC)//g; print "$tree\n"; } }' true_run_1 &gt; tree.txt
                    </p>
                    <p dir="auto">
                     <strong>
                      7.1.2 If you have 2 or more constraint points (most of the time you are in this condition)
                     </strong>
                    </p>
                    <p dir="auto">
                     #NEXUS
                    </p>
                    <p dir="auto">
                     begin trees;
                    </p>
                    <p dir="auto">
                     <code>
                     </code>
                     tree tree_1 = [&amp;R] (((Cerianthus_borealis:0.290575,(((((Edwarsiella_lineata:0.066246,Nematostella_vectensis:0.081163):0.055217,((Exaiptasia_pallida:0.067063,Hormathia_digitata:0.134163):0.034484,((Bolocera_tuediae:0.051153,Megalactis_griffithsi:0.06898):0.043791,(Heteractis_crispa:0.056233,(Anthopleura_elegantissima:0.052055,(Anemonia_viridis:0.00563,Anemonia_sulcata:0.004281):0.02116):0.011187):0.01741):0.038479):0.062074):0.064797,((Antipathes_caribbeana:0.053896,Plumapathes_pennacea:0.045453):0.108419,((Corynactis_australis:0.066395,(Ricordea_yuma:0.065164,((Amplexidiscus_fenestrafer:0.053832,Rhodactis_indosinensis:0.014454):0.01069,Discosoma_spp:0.03959):0.040398):0.011425):0.028652,((((Acropora_digitifera:0.057801,Acropora_millepora:0.008548):0.005001,Acropora_tenuis:0.021788):0.090596,(Astreopora_spp:0.048089,(Porites_lobata:0.002024,Porites_australiensis:0.006654):0.048218):0.012396):0.014759,(((Lobactis_scutaria:0.008759,Ctenactis_echinata:0.006969):0.027244,(Montastraea_cavernosa:0.024341,(Pseudodiploria_strigosa:0.027266,(((Favia_lizardensis:0.002407,Favia_spp:0.10512):0.006193,Platygyra_carnosus:0.16142):0.00544,Orbicella_faveolata:0.03004):0.003371):0.005356):0.017989):0.011364,(Madracis_auretenra:0.027553,(Pocillopora_damicornis:0.035619,(Stylophora_pistillata:0.030801,(Seriatopora_hystrix:0.015099,Seriatopora_spp:0.035017):0.010937):0.007567):0.034336):0.022343):0.017003):0.027674):0.07751):0.026608):0.020789,Protopalythoa_variabilis:0.277447):0.063276,((Pennatula_rubra:0.055175,Renilla_reniformis:0.107084):0.072312,(Corallium_rubrum:0.073471,(Nephthyigorgia_spp:0.062532,((Gorgonia_ventalina:0.051904,Leptogorgia_sarmentosa:0.023995):0.013823,(Eunicella_cavolinii:0.011368,Eunicella_verrucosa:0.002738):0.025411):0.011668):0.049632):0.032084):0.188635):0.018531):0.036866,((Polypodium_hydriforme:0.566426,(((Enteromyxum_leei:0.648484,Kudoa_iwatai:0.561639):0.680711,(Sphaeromyxa_zaharoni:0.953538,(Henneguya:0.780518,((((Thelohanellus_kitauei:0.307058,Myxobolus_turpisrotundus:0.304868):0.180837,((Myxobolus_wulii:0.22627,(Myxobolus_honghuensis:0.015607,Myxobolus_ampullicapsulatus:0.070408):0.17434):0.057403,Myxobolus_pendula:0.245172):0.256762):0.21192,Myxobolus_cerebralis:0.432823):0.219181,Myxobolus_physophilus:0.60296):0.170284):0.129202):0.147595):0.130403,Tetracapsuloides_bryosalmonae:0.80208):0.20069):0.142613,(((((Atolla_vanhoeffeni:0.040317,Periphylla_periphylla:0.065065):0.150184,((Aurelia_aurita:0.079173,(Cassiopea_xamachana:0.132511,(Stomolophus_meleagris:0.05262,Rhopilema_esculentum:0.037626):0.059135):0.020893):0.040423,((Cyanea_capillata:0.013458,Cyanea_nozakii:9.14E-4):0.112443,(Chrysaora_fuscescens:0.054366,Pelagia_noctiluca:0.058332):0.054913):0.032858):0.097924):0.032958,(Chironex_fleckeri:0.098164,(Alatina_alata:0.081036,Tripedalia_cystophora:0.180489):0.02203):0.132863):0.024249,(Craterolophus_convolvulus:0.100106,(Calvadosia_campanulata:0.081895,(Haliclystus_sanjuanensis:0.150622,Calvadosia_cruxmelitensis:0.047193):0.020468):0.023085):0.204894):0.024348,((Liriope_tetraphylla:0.144402,(Craspedacusta_sowerbyi:0.163299,Aegina_citrea:0.22533):0.02255):0.055681,((Ectopleura_larynx:0.169364,(Hydra_viridissima:0.041963,(Hydra_oligactis:0.030741,Hydra_vulgaris:0.028101):0.038597):0.092278):0.078655,(Millepora_alcicornis:0.226375,((Turritopsis_spp:0.15472,(Podocoryna_carnea:0.102354,(Hydractinia_polyclina:0.008327,Hydractinia_symbiolongicarpus:0.007533):0.059407):0.051791):0.030935,(Physalia_physalis:0.111875,((Nanomia_bijuga:0.07543,Agalma_elegans:0.042964):0.044384,(Craseoa_lathetica:0.059619,Abylopsis_tetragona:0.121102):0.037884):0.042855):0.075418):0.015359):0.016351):0.058833):0.111132):0.063606):0.055702):0.028066,((((Helobdella_robusta:0.428146,Crassostrea_gigas:0.25635):0.054388,Drosophila_melanogaster:0.536965):0.050571,((Ciona_intestinalis:0.459065,Taeniopygia_guttata:0.299825):0.052585,Strongylocentrotus_purpuratus:0.349709):0.026884):0.077267,(Trichoplax_adhaerens:0.483558,(Amphimedon_queenslandica:0.550768,Mnemiopsis_leidyi:0.725216):0.059918):0.045348):0.028066);
                    </p>
                    <p dir="auto">
                     end;
                    </p>
                    <p dir="auto">
                     begin r8s;
                    </p>
                    <p dir="auto">
                     blformat lengths=persite nsites=51598 ulrametric=no;
                    </p>
                    <p dir="auto">
                     collapse;
                    </p>
                    <p dir="auto">
                     MRCA OUT Helobdella_robusta Mnemiopsis_leidyi Trichoplax_adhaerens;
                    </p>
                    <p dir="auto">
                     MRCA MED Chironex_fleckeri Agalma_elegans Thelohanellus_kitauei;
                    </p>
                    <p dir="auto">
                     MRCA HEX Protopalythoa_variabilis Edwarsiella_lineata Lobactis_scutaria;
                    </p>
                    <p dir="auto">
                     MRCA HYD Liriope_tetraphylla Agalma_elegans Hydra_vulgaris;
                    </p>
                    <p dir="auto">
                     MRCA CNI Protopalythoa_variabilis Chironex_fleckeri Thelohanellus_kitauei;
                    </p>
                    <p dir="auto">
                     prune taxon=OUT;
                    </p>
                    <p dir="auto">
                     fixage taxon=HYD age=500;
                    </p>
                    <p dir="auto">
                     constrain taxon=MED min_age=570;
                    </p>
                    <p dir="auto">
                     constrain taxon=HEX min_age=540;
                    </p>
                    <p dir="auto">
                     constrain taxon=CNI max_age=741;
                    </p>
                    <p dir="auto">
                     set minRateFactor=0.000001;
                    </p>
                    <p dir="auto">
                     set minDurFactor=0.000001;
                    </p>
                    <p dir="auto">
                     divtime method=PL crossv=yes fossilconstrained=yes cvstart=0 cvinc=1 cvnum=15;
                    </p>
                    <p dir="auto">
                     showage;
                    </p>
                    <p dir="auto">
                     describe plot=cladogram;
                    </p>
                    <p dir="auto">
                     describe plot=chrono_description;
                    </p>
                    <p dir="auto">
                     end;
                    </p>
                    <p dir="auto">
                     # Then conduct first validation
                    </p>
                    <p dir="auto">
                     r8s -b -f r8s_in_validation.txt &gt; cross_validation_1
                    </p>
                    <p dir="auto">
                     # Find the line with the smallest raw error, and then find the log10smoon value in the same line
                    </p>
                    <p dir="auto">
                     # Then revise the configuration file as a new file: r8s_in_true_run.txt
                    </p>
                    <p dir="auto">
                     #NEXUS
                    </p>
                    <p dir="auto">
                     begin trees;
                    </p>
                    <p dir="auto">
                     <code>
                     </code>
                     tree tree_1 = [&amp;R] (((Cerianthus_borealis:0.290575,(((((Edwarsiella_lineata:0.066246,Nematostella_vectensis:0.081163):0.055217,((Exaiptasia_pallida:0.067063,Hormathia_digitata:0.134163):0.034484,((Bolocera_tuediae:0.051153,Megalactis_griffithsi:0.06898):0.043791,(Heteractis_crispa:0.056233,(Anthopleura_elegantissima:0.052055,(Anemonia_viridis:0.00563,Anemonia_sulcata:0.004281):0.02116):0.011187):0.01741):0.038479):0.062074):0.064797,((Antipathes_caribbeana:0.053896,Plumapathes_pennacea:0.045453):0.108419,((Corynactis_australis:0.066395,(Ricordea_yuma:0.065164,((Amplexidiscus_fenestrafer:0.053832,Rhodactis_indosinensis:0.014454):0.01069,Discosoma_spp:0.03959):0.040398):0.011425):0.028652,((((Acropora_digitifera:0.057801,Acropora_millepora:0.008548):0.005001,Acropora_tenuis:0.021788):0.090596,(Astreopora_spp:0.048089,(Porites_lobata:0.002024,Porites_australiensis:0.006654):0.048218):0.012396):0.014759,(((Lobactis_scutaria:0.008759,Ctenactis_echinata:0.006969):0.027244,(Montastraea_cavernosa:0.024341,(Pseudodiploria_strigosa:0.027266,(((Favia_lizardensis:0.002407,Favia_spp:0.10512):0.006193,Platygyra_carnosus:0.16142):0.00544,Orbicella_faveolata:0.03004):0.003371):0.005356):0.017989):0.011364,(Madracis_auretenra:0.027553,(Pocillopora_damicornis:0.035619,(Stylophora_pistillata:0.030801,(Seriatopora_hystrix:0.015099,Seriatopora_spp:0.035017):0.010937):0.007567):0.034336):0.022343):0.017003):0.027674):0.07751):0.026608):0.020789,Protopalythoa_variabilis:0.277447):0.063276,((Pennatula_rubra:0.055175,Renilla_reniformis:0.107084):0.072312,(Corallium_rubrum:0.073471,(Nephthyigorgia_spp:0.062532,((Gorgonia_ventalina:0.051904,Leptogorgia_sarmentosa:0.023995):0.013823,(Eunicella_cavolinii:0.011368,Eunicella_verrucosa:0.002738):0.025411):0.011668):0.049632):0.032084):0.188635):0.018531):0.036866,((Polypodium_hydriforme:0.566426,(((Enteromyxum_leei:0.648484,Kudoa_iwatai:0.561639):0.680711,(Sphaeromyxa_zaharoni:0.953538,(Henneguya:0.780518,((((Thelohanellus_kitauei:0.307058,Myxobolus_turpisrotundus:0.304868):0.180837,((Myxobolus_wulii:0.22627,(Myxobolus_honghuensis:0.015607,Myxobolus_ampullicapsulatus:0.070408):0.17434):0.057403,Myxobolus_pendula:0.245172):0.256762):0.21192,Myxobolus_cerebralis:0.432823):0.219181,Myxobolus_physophilus:0.60296):0.170284):0.129202):0.147595):0.130403,Tetracapsuloides_bryosalmonae:0.80208):0.20069):0.142613,(((((Atolla_vanhoeffeni:0.040317,Periphylla_periphylla:0.065065):0.150184,((Aurelia_aurita:0.079173,(Cassiopea_xamachana:0.132511,(Stomolophus_meleagris:0.05262,Rhopilema_esculentum:0.037626):0.059135):0.020893):0.040423,((Cyanea_capillata:0.013458,Cyanea_nozakii:9.14E-4):0.112443,(Chrysaora_fuscescens:0.054366,Pelagia_noctiluca:0.058332):0.054913):0.032858):0.097924):0.032958,(Chironex_fleckeri:0.098164,(Alatina_alata:0.081036,Tripedalia_cystophora:0.180489):0.02203):0.132863):0.024249,(Craterolophus_convolvulus:0.100106,(Calvadosia_campanulata:0.081895,(Haliclystus_sanjuanensis:0.150622,Calvadosia_cruxmelitensis:0.047193):0.020468):0.023085):0.204894):0.024348,((Liriope_tetraphylla:0.144402,(Craspedacusta_sowerbyi:0.163299,Aegina_citrea:0.22533):0.02255):0.055681,((Ectopleura_larynx:0.169364,(Hydra_viridissima:0.041963,(Hydra_oligactis:0.030741,Hydra_vulgaris:0.028101):0.038597):0.092278):0.078655,(Millepora_alcicornis:0.226375,((Turritopsis_spp:0.15472,(Podocoryna_carnea:0.102354,(Hydractinia_polyclina:0.008327,Hydractinia_symbiolongicarpus:0.007533):0.059407):0.051791):0.030935,(Physalia_physalis:0.111875,((Nanomia_bijuga:0.07543,Agalma_elegans:0.042964):0.044384,(Craseoa_lathetica:0.059619,Abylopsis_tetragona:0.121102):0.037884):0.042855):0.075418):0.015359):0.016351):0.058833):0.111132):0.063606):0.055702):0.028066,((((Helobdella_robusta:0.428146,Crassostrea_gigas:0.25635):0.054388,Drosophila_melanogaster:0.536965):0.050571,((Ciona_intestinalis:0.459065,Taeniopygia_guttata:0.299825):0.052585,Strongylocentrotus_purpuratus:0.349709):0.026884):0.077267,(Trichoplax_adhaerens:0.483558,(Amphimedon_queenslandica:0.550768,Mnemiopsis_leidyi:0.725216):0.059918):0.045348):0.028066);
                    </p>
                    <p dir="auto">
                     end;
                    </p>
                    <p dir="auto">
                     begin r8s;
                    </p>
                    <p dir="auto">
                     blformat lengths=persite nsites=51598 ulrametric=no;
                    </p>
                    <p dir="auto">
                     collapse;
                    </p>
                    <p dir="auto">
                     MRCA OUT Helobdella_robusta Mnemiopsis_leidyi Trichoplax_adhaerens;
                    </p>
                    <p dir="auto">
                     MRCA MED Chironex_fleckeri Agalma_elegans Thelohanellus_kitauei;
                    </p>
                    <p dir="auto">
                     MRCA HEX Protopalythoa_variabilis Edwarsiella_lineata Lobactis_scutaria;
                    </p>
                    <p dir="auto">
                     MRCA HYD Liriope_tetraphylla Agalma_elegans Hydra_vulgaris;
                    </p>
                    <p dir="auto">
                     MRCA CNI Protopalythoa_variabilis Chironex_fleckeri Thelohanellus_kitauei;
                    </p>
                    <p dir="auto">
                     prune taxon=OUT;
                    </p>
                    <p dir="auto">
                     fixage taxon=HYD age=500;
                    </p>
                    <p dir="auto">
                     constrain taxon=MED min_age=570;
                    </p>
                    <p dir="auto">
                     constrain taxon=HEX min_age=540;
                    </p>
                    <p dir="auto">
                     constrain taxon=CNI max_age=741;
                    </p>
                    <p dir="auto">
                     set minRateFactor=0.000001;
                    </p>
                    <p dir="auto">
                     set minDurFactor=0.000001;
                    </p>
                    <p dir="auto">
                     set smoothing=1e-06;
                    </p>
                    <p dir="auto">
                     set num_time_guesses=3;
                    </p>
                    <p dir="auto">
                     set checkGradient=yes;
                    </p>
                    <p dir="auto">
                     divtime method=PL algorithm=TN;
                    </p>
                    <p dir="auto">
                     showage;
                    </p>
                    <p dir="auto">
                     describe plot=cladogram;
                    </p>
                    <p dir="auto">
                     describe plot=chrono_description;
                    </p>
                    <p dir="auto">
                     end;
                    </p>
                    <p dir="auto">
                     # Run r8s
                    </p>
                    <p dir="auto">
                     r8s -b -f r8s_in_true_run.txt &gt; true_run_1
                    </p>
                    <p dir="auto">
                     # Extract the time tree from the result file
                    </p>
                    <p dir="auto">
                     perl -e 'while (&lt;&gt;) { if (m/tree tree_1 = (.*);/) { my $tree = $1; $tree =~ s/(MED|HEX|HYD|CNI)//g; print "$tree\n"; } }' true_run_1 &gt; tree.txt
                    </p>
                    <p dir="auto">
                     <strong>
                      8. Selection pressure analysis using Hyphy
                     </strong>
                    </p>
                    <p dir="auto">
                     # Make configuration file
                    </p>
                    <p dir="auto">
                     generate_busted_hyphy.sh alignment.phy tree.nwk &gt; script.bf
                    </p>
                    <p dir="auto">
                     HYPHYMPI script.bf
                    </p>
                    <p dir="auto">
                     # The content of generate_busted_hyphy.sh is as follows:
                    </p>
                    <p dir="auto">
                     inputRedirect = {};
                    </p>
                    <p dir="auto">
                     inputRedirect["01"]="Universal"; // genetic code
                    </p>
                    <p dir="auto">
                     inputRedirect["02"]="/home/gqx/proteome/16_selection_analysis/disintergrin/Disintergin_nucl.aln"; // codon data
                    </p>
                    <p dir="auto">
                     inputRedirect["03"]="/home/gqx/proteome/16_selection_analysis/disintergrin/1.nwk"; // tree
                    </p>
                    <p dir="auto">
                     inputRedirect["04"]="All"; // Test for selection on a branch
                    </p>
                    <p dir="auto">
                     inputRedirect["05"]=""; // complete selection
                    </p>
                    <p dir="auto">
                     ExecuteAFile ("/usr/local/lib/hyphy/TemplateBatchFiles/SelectionAnalyses/BUSTED.bf", inputRedirect);
                    </p>
                    <p dir="auto">
                     # If you want to use multithreading
                    </p>
                    <p dir="auto">
                     mpirun -np 12 /usr/local/bin/HYPHYMPI ./script.bf
                    </p>
                    <p dir="auto">
                     <strong>
                      9. Orthologous gene family analysis in comparative proteomics and visualization using UpSetR
                     </strong>
                    </p>
                    <p dir="auto">
                     # For the orthologous gene family analysis by OrthoMCL, please see section 1.
                    </p>
                    <p dir="auto">
                     # Provide matrix format that is suitable for UpSetR
                    </p>
                    <p dir="auto">
                     cat aaur_OG avir_OG cfle_OG hvul_OG mhon_OG mwul_OG tkit_OG &gt; OG
                    </p>
                    <p dir="auto">
                     remove_duplicate.pl OG
                    </p>
                    <p dir="auto">
                     mv duplicate_remove gene
                    </p>
                    <p dir="auto">
                     produce_matrix_for_UpSet.pl gene aaur_OG
                    </p>
                    <p dir="auto">
                     # Then combine a csv file that contains the gene and species information, eight_matrix_strict.csv
                    </p>
                    <p dir="auto">
                     # The content of Rscript upset_eight.R is as follows
                    </p>
                    <p dir="auto">
                     library(UpSetR)
                    </p>
                    <p dir="auto">
                     data &lt;- read.csv("eight_matrix.csv",header=T)
                    </p>
                    <p dir="auto">
                     pdf("out.pdf",  width=11, height=6)
                    </p>
                    <p dir="auto">
                     upset(data, nsets = 8, order.by="freq", sets.bar.color = "#56B4E9", mainbar.y.label = "OG Intersections", sets.x.label = "ProteinsPer Species",nintersects = 60)
                    </p>
                    <p dir="auto">
                     dev.off()
                    </p>
                    <p dir="auto">
                     <strong>
                      10. Functional comparison in comparative proteomics and visualization using UpSetR
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      10.1 Get the Interpro number for each proteins
                     </strong>
                    </p>
                    <p dir="auto">
                     # Use Blast2GO to do Interpro annotation, collect the Interpro number (IPR) and save it as list file.
                    </p>
                    <p dir="auto">
                     # Then run the bash file: interpro_ID_converter.sh, as follows:
                    </p>
                    <p dir="auto">
                     grep "IPR" list  &gt; out
                    </p>
                    <p dir="auto">
                     split_lines_by_semicolon.pl out
                    </p>
                    <p dir="auto">
                     grep "IPR" out2 &gt; out3
                    </p>
                    <p dir="auto">
                     remove_duplicate.pl out3
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/\s+(\S+)$//g' duplicate_remove
                    </p>
                    <p dir="auto">
                     mv duplicate_remove duplicate_remove1
                    </p>
                    <p dir="auto">
                     remove_duplicate.pl duplicate_remove1
                    </p>
                    <p dir="auto">
                     rm duplicate_remove1
                    </p>
                    <p dir="auto">
                     mv duplicate_remove final_IPR
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/^\s//g' final_IPR
                    </p>
                    <p dir="auto">
                     remove_duplicate.pl final_IPR
                    </p>
                    <p dir="auto">
                     rm final_IPR
                    </p>
                    <p dir="auto">
                     mv duplicate_remove final_IPR
                    </p>
                    <p dir="auto">
                     convert_interpro_2_name.pl final_IPR entry.list
                    </p>
                    <p dir="auto">
                     rm out out2 out3
                    </p>
                    <p dir="auto">
                     cat convert_name | cut -f 3 &gt; domain_name
                    </p>
                    <p dir="auto">
                     # The final_IPR contains the information for visualization
                    </p>
                    <p dir="auto">
                     <strong>
                      10.2 Visualize the results using UpSetR
                     </strong>
                    </p>
                    <p dir="auto">
                     # Change the name of final_IPR for each species
                    </p>
                    <p dir="auto">
                     # Use
                     <em>
                      Chironex
                     </em>
                     as an example
                    </p>
                    <p dir="auto">
                     cp ../5_Chironex_domain/final_IPR ./Chironex
                    </p>
                    <p dir="auto">
                     cat Anemonia  Aurelia  Chironex  Hydra  honghuensis  kitauei  shata wulii &gt; OG
                    </p>
                    <p dir="auto">
                     remove_duplicate.pl OG
                    </p>
                    <p dir="auto">
                     mv duplicate_remove gene
                    </p>
                    <p dir="auto">
                     # For each species
                    </p>
                    <p dir="auto">
                     produce_matrix_for_UpSet.pl gene Anemonia
                    </p>
                    <p dir="auto">
                     rename _matrix .xls *matrix
                    </p>
                    <p dir="auto">
                     # Combine gene and OG_matrix into a single csv: eight_matrix_strict.csv
                    </p>
                    <p dir="auto">
                     Rscript upset_eight.R
                    </p>
                    <p dir="auto">
                     <strong>
                      11. Circos analysis and visualization
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      11.1 Install circos
                     </strong>
                    </p>
                    <p dir="auto">
                     <a href="http://circos.ca/distribution/circos-0.69-6.tgz" rel="nofollow">
                      http://circos.ca/distribution/circos-0.69-6.tgz
                     </a>
                    </p>
                    <p dir="auto">
                     tar zxf /opt/software/circos-0.69-6.tgz
                    </p>
                    <p dir="auto">
                     cd /opt/biosoft/circos-0.69-6/bin
                    </p>
                    <p dir="auto">
                     ./list.modules
                    </p>
                    <p dir="auto">
                     ./list.modules | perl -p -e 's/\n/ /' | perl -p -e 's/^/sudo cpan -i /; s/$/\n/' | sh
                    </p>
                    <p dir="auto">
                     ./circos -modules | perl -e 'while (&lt;&gt;) { print "$1 " if m/missing\s+(\S+)/ }' | perl -p -e 's/^/sudo cpan -i /; s/$/\n/' | sh     # Install missing modules
                    </p>
                    <p dir="auto">
                     #Install GD
                    </p>
                    <p dir="auto">
                     cd /opt/sysoft/
                    </p>
                    <p dir="auto">
                     tar Jxf  /opt/software/libgd-2.1.1.tar.xz
                    </p>
                    <p dir="auto">
                     ./configure --prefix=/usr/
                    </p>
                    <p dir="auto">
                     make -j 4
                    </p>
                    <p dir="auto">
                     sudo make install
                    </p>
                    <p dir="auto">
                     gdlib-config
                    </p>
                    <p dir="auto">
                     ln -s /usr/lib64/libgd.so.2.0.0 /usr/lib64/libgd.so.3
                    </p>
                    <p dir="auto">
                     wget
                     <a href="http://circos.ca/distribution/lib/GD-2.53.tar.gz" rel="nofollow">
                      http://circos.ca/distribution/lib/GD-2.53.tar.gz
                     </a>
                    </p>
                    <p dir="auto">
                     tar  zxf /opt/software/GD-2.53.tar.gz
                    </p>
                    <p dir="auto">
                     perl Makefile.PL
                    </p>
                    <p dir="auto">
                     make
                    </p>
                    <p dir="auto">
                     make install
                    </p>
                    <p dir="auto">
                     ./circos -modules
                    </p>
                    <p dir="auto">
                     ./gddiag
                    </p>
                    <p dir="auto">
                     ./circos --help
                    </p>
                    <p dir="auto">
                     echo 'PATH=$PATH:/opt/biosoft/circos-0.69-6/bin/' &gt;&gt; ~/.bashr
                    </p>
                    <p dir="auto">
                     source ~/.bashrc
                    </p>
                    <p dir="auto">
                     # Dowmload circos_tools
                    </p>
                    <p dir="auto">
                     wget
                     <a href="http://circos.ca/distribution/circos-tools-0.22.tgz" rel="nofollow">
                      http://circos.ca/distribution/circos-tools-0.22.tgz
                     </a>
                    </p>
                    <p dir="auto">
                     tar zxf /opt/software/circos-tools-0.22.tgz
                    </p>
                    <p dir="auto">
                     cd /opt/biosoft/circos-tools-0.22/tools/binlinks
                    </p>
                    <p dir="auto">
                     ./run
                    </p>
                    <p dir="auto">
                     <strong>
                      11.2 Install Circoletto
                     </strong>
                    </p>
                    <p dir="auto">
                     wget
                     <a href="https://github.com/infspiredBAT/Circoletto/archive/master.zip">
                      https://github.com/infspiredBAT/Circoletto/archive/master.zip
                     </a>
                    </p>
                    <p dir="auto">
                     unzip master.zip
                    </p>
                    <p dir="auto">
                     cd Circoletto-master
                    </p>
                    <p dir="auto">
                     # If you are dealing with more than 200 sequences, change the max_ideograms parameter
                    </p>
                    <p dir="auto">
                     vim /opt/biosoft/circos-0.69-6/etc/housekeeping.conf
                    </p>
                    <p dir="auto">
                     # Also change circoletto.pl, feed it with the path of other softwares
                    </p>
                    <p dir="auto">
                     $path2circos      = '/opt/biosoft/circos-0.69-6';     #  path to Circos
                    </p>
                    <p dir="auto">
                     $path2circostools = '/opt/biosoft/circos-tools-0.22'; #  path to Circos utilities
                    </p>
                    <p dir="auto">
                     <strong>
                      11.3 Install blast-2.2.25
                     </strong>
                    </p>
                    <p dir="auto">
                     wget ftp://ftp.ncbi.nlm.nih.gov/blast/executables/legacy/2.2.25/blast-2.2.25-x64-linux.tar.gz
                    </p>
                    <p dir="auto">
                     echo "PATH=$PATH:/opt/biosoft/blast-2.2.25/bin" &gt;&gt; ~/.bashrc
                    </p>
                    <p dir="auto">
                     source ~/.bashrc
                    </p>
                    <p dir="auto">
                     <strong>
                      11.4 Run Circoletto
                     </strong>
                    </p>
                    <p dir="auto">
                     # Use analysis between the protein of
                     <em>
                      Myxobolus honghuensis
                     </em>
                     and other six species as an example
                    </p>
                    <p dir="auto">
                     cat aaur.fasta avir.fasta cfle.fasta cfus.fasta cnoz.fasta hvul.fasta &gt; rest.fa
                    </p>
                    <p dir="auto">
                     # Circoletto can not deal with asterisks.
                    </p>
                    <p dir="auto">
                     perl -p -i -e "s/\*//g" rest.fa
                    </p>
                    <p dir="auto">
                     # Change the parameter in Circoletto.pl
                    </p>
                    <p dir="auto">
                     $max_sequences = 10000
                    </p>
                    <p dir="auto">
                     circoletto.pl --cpus 12 --query mhon.fasta --database rest.fa --out_type svg --e_value 1e-20 --score2colour eval --scoreratio2colour max  --ribocolours2allow '(blue|green|orange|red)' --z_by score --no_labels
                    </p>
                    <p dir="auto">
                     # Use Perl to process link file to adjust the link details
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/stroke_color=(\S+?),//g' ./*.blasted.links
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/stroke_thickness=(\d+?),//g' ./*.blasted.links
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/_a3/_a2/g' ./*.blasted.links
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/thickness=(\d+)/thickness=4/g' ./*.blasted.links
                    </p>
                    <p dir="auto">
                     paste -d ',' ./*.blasted.links ff &gt; new
                    </p>
                    <p dir="auto">
                     rm ./*.links
                    </p>
                    <p dir="auto">
                     mv new  cl0011919395.blasted.links
                    </p>
                    <p dir="auto">
                     # Use Perl to process karyotype file to adjust the color
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/(\S+)$/dgreen/ if ($_ =~ /chr/ and $_ =~ /aaur/)' ./*.blasted.karyotype
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/(\S+)$/lblue/ if ($_ =~ /chr/ and $_ =~ /avir/)' ./*.blasted.karyotype
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/(\S+)$/dred/ if ($_ =~ /chr/ and $_ =~ /cfle/)' ./*.blasted.karyotype
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/(\S+)$/dyellow/ if ($_ =~ /chr/ and $_ =~ /cfus/)' ./*.blasted.karyotype
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/(\S+)$/black/ if ($_ =~ /chr/ and $_ =~ /hvul/)' ./*.blasted.karyotype
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/(\S+)$/dblue/ if ($_ =~ /chr/ and $_ =~ /mhon/)' ./*.blasted.karyotype
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/(\S+)$/lpurple/ if ($_ =~ /chr/ and $_ =~ /cnoz/)' ./*.blasted.karyotype
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/(\S+)$/dpurple/ if ($_ =~ /chr/ and $_ =~ /tkit/)' ./*.blasted.karyotype
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/(\S+)$/dorange/ if ($_ =~ /chr/ and $_ =~ /mwul/)' ./*.blasted.karyotype
                    </p>
                    <p dir="auto">
                     perl -p -i -e 's/(\S+)$/dyellow/ if ($_ =~ /chr/ and $_ =~ /csha/)' ./*.blasted.karyotype
                    </p>
                    <p dir="auto">
                     # Edit the conf file as follows:
                    </p>
                    <p dir="auto">
                     default             = 0.0003r
                    </p>
                    <p dir="auto">
                     break               = 10u
                    </p>
                    <p dir="auto">
                     axis_break_at_edge  = no
                    </p>
                    <p dir="auto">
                     axis_break          = no
                    </p>
                    <p dir="auto">
                     axis_break_style    = 1
                    </p>
                    <p dir="auto">
                     thickness           = 35p
                    </p>
                    <p dir="auto">
                     fill                = yes
                    </p>
                    <p dir="auto">
                     radius              = 0.66r
                    </p>
                    <p dir="auto">
                     show_label          = no
                    </p>
                    <p dir="auto">
                     label_with_tag      = yes
                    </p>
                    <p dir="auto">
                     label_font          = condensed
                    </p>
                    <p dir="auto">
                     label_color         = grey
                    </p>
                    <p dir="auto">
                     label_radius        = dims(ideogram,radius) + 0.104r
                    </p>
                    <p dir="auto">
                     label_size          =
                    </p>
                    <p dir="auto">
                     band_stroke_thickness = 0
                    </p>
                    <p dir="auto">
                     show_bands          = no
                    </p>
                    <p dir="auto">
                     fill_bands          = yes
                    </p>
                    <p dir="auto">
                     show_ticks          = no
                    </p>
                    <p dir="auto">
                     show_tick_labels    = no
                    </p>
                    <p dir="auto">
                     radius              = dims(ideogram,radius_outer)
                    </p>
                    <p dir="auto">
                     multiplier          = 1e-6
                    </p>
                    <p dir="auto">
                     spacing             = 100u
                    </p>
                    <p dir="auto">
                     rspacing            = 0.1
                    </p>
                    <p dir="auto">
                     spacing_type        = relative
                    </p>
                    <p dir="auto">
                     size                = 5p
                    </p>
                    <p dir="auto">
                     thickness           = 1p
                    </p>
                    <p dir="auto">
                     color               = vvvvdgrey
                    </p>
                    <p dir="auto">
                     show_label          = no
                    </p>
                    <p dir="auto">
                     label_size          = 10p
                    </p>
                    <p dir="auto">
                     label_offset        = 5p
                    </p>
                    <p dir="auto">
                     format              = %d
                    </p>
                    <p dir="auto">
                     karyotype           = cl0006216655.blasted.karyotype
                    </p>
                    <p dir="auto">
                     <a href="" rel="noopener noreferrer" target="_blank">
                      <img style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     dir  = .
                    </p>
                    <p dir="auto">
                     file                = cl0006216655.blasted.svg
                    </p>
                    <p dir="auto">
                     radius              = 1000p
                    </p>
                    <p dir="auto">
                     background          = white
                    </p>
                    <p dir="auto">
                     angle_offset        = -90
                    </p>
                    <p dir="auto">
                     24bit               = yes
                    </p>
                    <p dir="auto">
                     auto_alpha_colors   = yes
                    </p>
                    <p dir="auto">
                     auto_alpha_steps    = 5
                    </p>
                    <p dir="auto">
                     chromosomes_units   = 1
                    </p>
                    <p dir="auto">
                     chromosomes_display_default = yes
                    </p>
                    <p dir="auto">
                     ribbon              = no
                    </p>
                    <p dir="auto">
                     show                = yes
                    </p>
                    <p dir="auto">
                     file                = cl0006216655.blasted.links
                    </p>
                    <p dir="auto">
                     record_limit        = 2000
                    </p>
                    <p dir="auto">
                     # Run circus
                    </p>
                    <p dir="auto">
                     circos -conf ./*.conf -noparanoid
                    </p>
                    <p dir="auto">
                     <strong>
                      12. Identify ortholog gene manually
                     </strong>
                    </p>
                    <p dir="auto">
                     # RBBH is mainly used to find orthologs, while the common blast includes all homologs, and RBBH can distinguish orthologs from paralogs. The NC method is actually an all-against-all blast, which includes paralogs, so it is generally used for homology identification. Note: homology does not necessarily mean ortholog. homology is more stringent.
                    </p>
                    <p dir="auto">
                     <strong>
                      12.1 Pairwise Reciprocal Best BLAST Hits (RBBH)
                     </strong>
                    </p>
                    <p dir="auto">
                     blast_rbbh.py fish.fasta human.fasta -o fish_human.tab -t blastp -a prot -i 1 -c 1 --threads 48
                    </p>
                    <p dir="auto">
                     # If you want to do RBBH for three species
                    </p>
                    <p dir="auto">
                     cat Avir_Aaur.tab | cut -f 1,2 &gt; 22
                    </p>
                    <p dir="auto">
                     cat Avir_Hydra.tab | cut -f 1,2 &gt; 33
                    </p>
                    <p dir="auto">
                     cat Aaur_Hydra.tab | cut -f 1,2 &gt; 44
                    </p>
                    <p dir="auto">
                     run_RBBH.pl ../Anemonia_viridis_Rachamim_2014.fasta ../Aurelia_aurita_Rachamim_2014.fasta Anemonia Aurelia
                    </p>
                    <p dir="auto">
                     proteome_RBBH_get_three_inter.pl Anemonia_Aurelia Anemonia_Hydra Aurelia_Hydra
                    </p>
                    <p dir="auto">
                     <strong>
                      12.2 Neighborhood correlation (NC) method
                     </strong>
                    </p>
                    <p dir="auto">
                     # Compare Anemonia_viridis_Rachamim_2014.fasta and Hydra_vulgaris_Rachamim_2014.fasta
                    </p>
                    <p dir="auto">
                     run_neighborhood_correlation.pl Anemonia_viridis_Rachamim_2014.fasta Hydra_vulgaris_Rachamim_2014.fasta Anemonia Hydra 0.6
                    </p>
                    <p dir="auto">
                     proteome_NC_result_process.pl out
                    </p>
                    <p dir="auto">
                     NC_thresh.pl NC_result 0.8
                    </p>
                    <p dir="auto">
                     cat out2 | cut -d ' ' -f 1 &gt; 1
                    </p>
                    <p dir="auto">
                     remove_duplicate.pl 1
                    </p>
                    <p dir="auto">
                     wc -l duplicate_remove
                    </p>
                    <p dir="auto">
                     mv out2 NC_0.8
                    </p>
                    <p dir="auto">
                     <strong>
                      Please cite this paper as
                     </strong>
                    </p>
                    <p dir="auto">
                     Guo, Q., Atkinson, S. D., Xiao, B., Zhai, Y., Bartholomew, J. L., &amp; Gu, Z. (2022). A myxozoan genome reveals mosaic evolution in a parasitic cnidarian. BMC biology, 20(1), 1-19.
                    </p>
                    <p dir="auto">
                     <strong>
                      License
                     </strong>
                    </p>
                    <p dir="auto">
                     All source code, i.e. scripts/*.pl, scripts/*.sh or scripts/*.py are under the MIT license.
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="GFXi+ey0Hcch8ZYabdEql9pf4mXgKS0GrSyIaISP23MucPgavrwtBErobtSC1dX84T3SI4d20vVnHzO9w7PuJg==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 Streaming workflow for phylogenomic analysis and comparative proteomics
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <h3 class="sr-only">
                 License
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="#MIT-1-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                   </path>
                  </svg>
                  MIT license
                 </a>
                </div>
                <include-fragment src="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fqingxiangguo%2FPhylogenomic_analysis_and_comparative_proteomics&amp;report=qingxiangguo+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/qingxiangguo/packages?repo_name=Phylogenomic_analysis_and_comparative_proteomics">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="Perl 82.6" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#0298c3 !important;;width: 82.6%;">
                 </span>
                 <span aria-label="Shell 12.7" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#89e051 !important;;width: 12.7%;">
                 </span>
                 <span aria-label="R 2.9" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#198CE7 !important;;width: 2.9%;">
                 </span>
                 <span aria-label="Python 1.8" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#3572A5 !important;;width: 1.8%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/search?l=perl">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#0298c3;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Perl
                  </span>
                  <span>
                   82.6%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/search?l=shell">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#89e051;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Shell
                  </span>
                  <span>
                   12.7%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/search?l=r">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#198CE7;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   R
                  </span>
                  <span>
                   2.9%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/qingxiangguo/Phylogenomic_analysis_and_comparative_proteomics/search?l=python">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#3572A5;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Python
                  </span>
                  <span>
                   1.8%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
