<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-4c826ff1ab15.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - sunericd/TISSUE: TISSUE (Transcript Imputation with Spatial Single-cell Uncertainty Estimation) provides tools for estimating well-calibrated uncertainty measures for gene expression predictions in single-cell spatial transcriptomics datasets and utilizing them in downstream analyses
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:169FB1C:2097EEF:6605E722" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDoxNjlGQjFDOjIwOTdFRUY6NjYwNUU3MjIiLCJ2aXNpdG9yX2lkIjoiNjQ1MjI0MTg1NjIzODgxNzgwNyIsInJlZ2lvbl9lZGdlIjoiaWFkIiwicmVnaW9uX3JlbmRlciI6ImlhZCJ9" data-pjax-transient="true" name="visitor-payload">
           <meta content="9d9ce50b511257ff2c073fd3e13c1a688d40312b3132eb852a5bd70e37670bd4" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:620168965" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="TISSUE (Transcript Imputation with Spatial Single-cell Uncertainty Estimation) provides tools for estimating well-calibrated uncertainty measures for gene expression predictions in single-cell spatial transcriptomics datasets and utilizing them in downstream analyses - sunericd/TISSUE" name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/sunericd/TISSUE" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/ff23c9b90c693b2c88ec2eedc4dcdd5f585c37341182f7f6679de8d710a09c04/sunericd/TISSUE" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - sunericd/TISSUE: TISSUE (Transcript Imputation with Spatial Single-cell Uncertainty Estimation) provides tools for estimating well-calibrated uncertainty measures for gene expression predictions in single-cell spatial transcriptomics datasets and utilizing them in downstream analyses" name="twitter:title">
                    <meta content="TISSUE (Transcript Imputation with Spatial Single-cell Uncertainty Estimation) provides tools for estimating well-calibrated uncertainty measures for gene expression predictions in single-cell spat..." name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/ff23c9b90c693b2c88ec2eedc4dcdd5f585c37341182f7f6679de8d710a09c04/sunericd/TISSUE" property="og:image">
                      <meta content="TISSUE (Transcript Imputation with Spatial Single-cell Uncertainty Estimation) provides tools for estimating well-calibrated uncertainty measures for gene expression predictions in single-cell spat..." property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - sunericd/TISSUE: TISSUE (Transcript Imputation with Spatial Single-cell Uncertainty Estimation) provides tools for estimating well-calibrated uncertainty measures for gene expression predictions in single-cell spatial transcriptomics datasets and utilizing them in downstream analyses" property="og:title"/>
                        <meta content="https://github.com/sunericd/TISSUE" property="og:url"/>
                        <meta content="TISSUE (Transcript Imputation with Spatial Single-cell Uncertainty Estimation) provides tools for estimating well-calibrated uncertainty measures for gene expression predictions in single-cell spat..." property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="ce7ac010b2a2d244f4e22532a2864ada4189093671a7a5533bb73137af2ff3fe" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="af60a253d00dc707d466f3bb6e8cc7535995d8a78f2d1be3507d205c4ae531a3" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/sunericd/TISSUE git https://github.com/sunericd/TISSUE.git" name="go-import"/>
                        <meta content="7906731" name="octolytics-dimension-user_id">
                         <meta content="sunericd" name="octolytics-dimension-user_login">
                          <meta content="620168965" name="octolytics-dimension-repository_id">
                           <meta content="sunericd/TISSUE" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="620168965" name="octolytics-dimension-repository_network_root_id">
                               <meta content="sunericd/TISSUE" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/sunericd/TISSUE" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-1c70f65f7356.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-11f6759e1cef.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/sunericd/TISSUE","user_id":null}}' data-hydro-click-hmac="162147536f67aa93d9ecfb2f7704ff105fa20c30884ff0b0f9873cc201cb8b51" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fsunericd%2FTISSUE">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=U8FORZ%2BvxFss7KP6%2BI7FB56a7D4%2BVORJY92ETSmQlFKQDmn8wXD4m5JYgkOyPne9Zl3zD6zl1GFSSYMeNRsz%2FK%2BypE5QeeJtnwL8bg5Z89WdjOSiQ7m72auV3Vd%2FQaVMQ5haEbgz%2FfTd7PEXI%2FBhl65rbWeKrUzWPG6IpbC8joUyuD4nYbnS1dM9MBrHb7ADwA2r%2Fwys7fKVbemcM4hgK7jHtWxXY6K0bfFWPH7i3TSvpT4tMIG9etJXdZA2yTizyiF2Reu%2FIbsotI7IHnIhHNc5b1nBvjpQpmrvushg%2BxonEoBOqxYBbMdxSdZBSSWsO9NAw4XXQp41P2syJ3XG3c53pyWVVfKLM5JLgx6NdJhvtcfyMC3A0L%2B84SsJ5JLlJgMliCVriwWpGw7CZLTqETvKWqXj6Ye263OQCAaFChKsqofiWVlu946CyE5%2F9i%2B5aQPbQMfxo3N453mJxmVk%2BF4QbeMrIX4dI0dgzjgkJ7WvTDURDLxED4AMXscNkR7zM0niW3p4BOSLTw%3D%3D--%2B6LIhK0od8fVLMx1--IRv74Y3MLa%2FCtJR2OS%2BW0w%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="sunericd" data-current-repository="sunericd/TISSUE" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="yNJxvOtra5KcBK0WX4hNH6nEjMRNgQF3IqXg3kVx01ZSHGfrpyPczukw50q6X8YD5NW3yEyBDdDdnpubbMReoQ" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:sunericd/TISSUE">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-e5168762-59d3-4dda-b8f6-37282fd709df" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-e5168762-59d3-4dda-b8f6-37282fd709df">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="xWr1HXe8P+FaFsct7bnnLHiTUZ5rQeo693koT8WrKUJ7ruDrCLMZmhAXpl08XT40yS+KpOxaquq+MMzKFtVAcg=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="VXsQPX1U3ugk0W1tQCttCfZk6nLgibTUUUSbODULPf7f/L5hfX5jP2hpQoZ5/jPRMwwJfOIx5ih24u5jW5iWEw==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="Dxqvm1C3ifwEYdNaDYs/ivSXXY2Hz2ZKayoAf1gQAxpi1AfHckV6no6Mo6Cr2emEIMya9grxVMC7BmLYJVVBeQ==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="6TlyAQ2+8uT8NLnST9w5tYmjTBwuaVzcAmt3GtxfZogfJLdn8S3CmjmZF0v9ME2dP+PENYb005ccN8wgFyPKmA==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/sunericd/TISSUE","user_id":null}}' data-hydro-click-hmac="162147536f67aa93d9ecfb2f7704ff105fa20c30884ff0b0f9873cc201cb8b51" href="/login?return_to=https%3A%2F%2Fgithub.com%2Fsunericd%2FTISSUE">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/sunericd/TISSUE","user_id":null}}' data-hydro-click-hmac="162147536f67aa93d9ecfb2f7704ff105fa20c30884ff0b0f9873cc201cb8b51" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=sunericd%2FTISSUE">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-6a64c3c6-0616-4b54-a2f2-458ff2f70fab" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-c6e1de16-4b47-46d3-8d4a-9a30efb84da6" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-c6e1de16-4b47-46d3-8d4a-9a30efb84da6" id="tooltip-6a64c3c6-0616-4b54-a2f2-458ff2f70fab" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/sunericd/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/sunericd" rel="author">
            sunericd
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/sunericd/TISSUE">
            TISSUE
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/sunericd/TISSUE","user_id":null}}' data-hydro-click-hmac="51d11bde4d22407c8ebdbb76c91ca1f30ab1a241e6f731fb2caeacc247649196" data-view-component="true" href="/login?return_to=%2Fsunericd%2FTISSUE" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":620168965,"auth_type":"LOG_IN","originating_url":"https://github.com/sunericd/TISSUE","user_id":null}}' data-hydro-click-hmac="6d498eeb215990ce9318b7243842b389846b9ccfe7687a25839743dacbeb713d" data-view-component="true" href="/login?return_to=%2Fsunericd%2FTISSUE" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="3">
             3
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":620168965,"auth_type":"LOG_IN","originating_url":"https://github.com/sunericd/TISSUE","user_id":null}}' data-hydro-click-hmac="977ba570051bff592e07975e82b6a0d0f10faeef1d0a1f5103855d1cf37baccb" data-view-component="true" href="/login?return_to=%2Fsunericd%2FTISSUE" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="22 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="22">
              22
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <p class="f4 mb-3">
          TISSUE (Transcript Imputation with Spatial Single-cell Uncertainty Estimation) provides tools for estimating well-calibrated uncertainty measures for gene expression predictions in single-cell spatial transcriptomics datasets and utilizing them in downstream analyses
         </p>
         <h3 class="sr-only">
          License
         </h3>
         <div class="mb-2">
          <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="/sunericd/TISSUE/blob/master/LICENSE">
           <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
            </path>
           </svg>
           MIT license
          </a>
         </div>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/sunericd/TISSUE/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            22
           </span>
           stars
          </a>
          <a class="Link--secondary no-underline mr-3" href="/sunericd/TISSUE/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            3
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/sunericd/TISSUE/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/sunericd/TISSUE/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/sunericd/TISSUE/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":620168965,"auth_type":"LOG_IN","originating_url":"https://github.com/sunericd/TISSUE","user_id":null}}' data-hydro-click-hmac="977ba570051bff592e07975e82b6a0d0f10faeef1d0a1f5103855d1cf37baccb" data-view-component="true" href="/login?return_to=%2Fsunericd%2FTISSUE" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/sunericd/TISSUE","user_id":null}}' data-hydro-click-hmac="51d11bde4d22407c8ebdbb76c91ca1f30ab1a241e6f731fb2caeacc247649196" data-view-component="true" href="/login?return_to=%2Fsunericd%2FTISSUE" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /sunericd/TISSUE" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/sunericd/TISSUE" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /sunericd/TISSUE/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/sunericd/TISSUE/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="issues-repo-tab-count" title="1">
            1
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /sunericd/TISSUE/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/sunericd/TISSUE/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /sunericd/TISSUE/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/sunericd/TISSUE/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /sunericd/TISSUE/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/sunericd/TISSUE/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /sunericd/TISSUE/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/sunericd/TISSUE/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/sunericd/TISSUE/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /sunericd/TISSUE/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/sunericd/TISSUE/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-4742d92c-4e1f-4e72-9511-450d60855a65-list" aria-haspopup="true" aria-labelledby="tooltip-73cc2c9e-5ad1-498a-863f-45be12168b9e" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-4742d92c-4e1f-4e72-9511-450d60855a65-button" popovertarget="action-menu-4742d92c-4e1f-4e72-9511-450d60855a65-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-4742d92c-4e1f-4e72-9511-450d60855a65-button" id="tooltip-73cc2c9e-5ad1-498a-863f-45be12168b9e" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-4742d92c-4e1f-4e72-9511-450d60855a65-button" anchor-offset="normal" data-view-component="true" id="action-menu-4742d92c-4e1f-4e72-9511-450d60855a65-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-4742d92c-4e1f-4e72-9511-450d60855a65-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-4742d92c-4e1f-4e72-9511-450d60855a65-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/sunericd/TISSUE" id="item-14fbe669-ad7d-4023-a160-b0e313880cd3" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/sunericd/TISSUE/issues" id="item-ff264a00-0788-47ec-8041-0f3db5b3cea0" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/sunericd/TISSUE/pulls" id="item-ab0c276d-1ad1-41e5-beba-9efd71eb1a8c" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/sunericd/TISSUE/actions" id="item-187a6c2d-b073-48cf-9acc-f3b1633c3e53" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/sunericd/TISSUE/projects" id="item-ce32666d-f990-4066-a482-2da142eb23a5" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/sunericd/TISSUE/security" id="item-b5c7c08b-fa34-4dd9-9e83-2ea89f3ca0b1" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/sunericd/TISSUE/pulse" id="item-b6e90027-1c80-4e85-b6fa-56a5d51c439b" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         sunericd/TISSUE
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/sunericd/TISSUE/spoofed_commit_check/ffc35998004ecf44eaf2aedf774c2f6a2292e7b9">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-e8df95-53b8f38a33de.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_Dialog_js-node_modules_primer_react_lib-esm_-114c24-1306e57fe74e.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-d06085a838f3.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":620168965,"defaultBranch":"master","name":"TISSUE","ownerLogin":"sunericd","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2023-03-28T06:56:00.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/7906731?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"master","listCacheKey":"v0:1679986781.640264","canEdit":false,"refType":"branch","currentOid":"ffc35998004ecf44eaf2aedf774c2f6a2292e7b9"},"tree":{"items":[{"name":"README_files","path":"README_files","contentType":"directory"},{"name":"dist","path":"dist","contentType":"directory"},{"name":"tests/data","path":"tests/data","contentType":"directory","hasSimplifiedPath":true},{"name":"tissue","path":"tissue","contentType":"directory"},{"name":".gitignore","path":".gitignore","contentType":"file"},{"name":"LICENSE","path":"LICENSE","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"},{"name":"pipeline.png","path":"pipeline.png","contentType":"file"},{"name":"pyproject.toml","path":"pyproject.toml","contentType":"file"},{"name":"requirements.txt","path":"requirements.txt","contentType":"file"},{"name":"test.py","path":"test.py","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":11,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/sunericd/TISSUE/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/sunericd/TISSUE.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone sunericd/TISSUE","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/sunericd/TISSUE/archive/refs/heads/master.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=620168965"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"23","overviewFiles":[{"displayName":"README.md","repoName":"TISSUE","refName":"master","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTISSUE\u003c/h1\u003e\u003ca id=\"user-content-tissue\" class=\"anchor\" aria-label=\"Permalink: TISSUE\" href=\"#tissue\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTISSUE (Transcript Imputation with Spatial Single-cell Uncertainty Estimation)\u003c/strong\u003e provides tools for estimating well-calibrated uncertainty measures for gene expression predictions in single-cell spatial transcriptomics datasets and utilizing them in downstream analyses\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/sunericd/TISSUE/blob/master/pipeline.png\"\u003e\u003cimg src=\"/sunericd/TISSUE/raw/master/pipeline.png\" alt=\"plot\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIf you find TISSUE useful, please consider citing our publication:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSun, E.D., Ma, R., Navarro Negredo, P. et al. TISSUE: uncertainty-calibrated prediction of single-cell spatial transcriptomics improves downstream analyses. Nat Methods (2024). \u003ca href=\"https://doi.org/10.1038/s41592-024-02184-y\" rel=\"nofollow\"\u003ehttps://doi.org/10.1038/s41592-024-02184-y\u003c/a\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://doi.org/10.1038/s41592-024-02184-y\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/4597afa544082ce07c8b2860baf130715b74f811a760f5013fb6ea7f8e86e575/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f4e61747572655f4d6574686f64732d444f492d6f72616e6765\" alt=\"Paper\" data-canonical-src=\"https://img.shields.io/badge/Nature_Methods-DOI-orange\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInstallation and setup\u003c/h2\u003e\u003ca id=\"user-content-installation-and-setup\" class=\"anchor\" aria-label=\"Permalink: Installation and setup\" href=\"#installation-and-setup\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eComplete installation (including of dependencies) of TISSUE in a new Conda environment should take less than 5 minutes on a normal desktop/laptop setup (Windows, Mac OSX, Linux). The base version of TISSUE requires SquidPy and all of its dependencies. The base version comes with the SpaGE spatial gene expression prediction method. Additional TISSUE functionalities (i.e. Tangram prediction, Harmony-kNN prediction, gimVI prediction, TISSUE-WPCA, TISSUE multiple imputation with SpatialDE) require installing additional dependencies, which are listed in \u003ccode\u003erequirements.txt\u003c/code\u003e. These should be installed with \u003ccode\u003epip install \u0026lt;package name\u0026gt;==\u0026lt;package version\u0026gt;\u003c/code\u003e inside the same Conda environment as TISSUE (preferably before TISSUE installation).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor TISSUE installation, we provide two options: (A) PyPI installation with pip or (B) local installation. We recommend starting with option A and only going to option B if option A fails in your environment.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eOption A: PyPI\u003c/h3\u003e\u003ca id=\"user-content-option-a-pypi\" class=\"anchor\" aria-label=\"Permalink: Option A: PyPI\" href=\"#option-a-pypi\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eInstall the package through PyPI with \u003ccode\u003epip\u003c/code\u003e. We recommend setting up a conda environment (\u003ca href=\"https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html\" rel=\"nofollow\"\u003ehttps://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html\u003c/a\u003e) or another virtual environment first since \u003ccode\u003etissue-sc\u003c/code\u003e currently relies on specific versions for its dependencies (although it should generally work for other environment versions, but this hasn't been thoroughly tested):\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"conda create -n myenv python=3.8\nconda activate myenv\n\n\u0026lt;pip install any additional dependencies\u0026gt;\npip install tissue-sc\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003econda create -n myenv python=3.8\nconda activate myenv\n\n\u0026lt;pip install any additional dependencies\u0026gt;\npip install tissue-sc\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNote that you will want to separately download the data from this repository (\u003ccode\u003etests/data/\u003c/code\u003e) to run our TISSUE tutorials.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eOption B: Local installation\u003c/h3\u003e\u003ca id=\"user-content-option-b-local-installation\" class=\"anchor\" aria-label=\"Permalink: Option B: Local installation\" href=\"#option-b-local-installation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAn alternative way to install the package along with associated test and tutorial files is to clone the directory and then install the requirements for using the package. To do this, first clone the repository using git (you can install git following the instructions \u003ca href=\"https://github.com/git-guides/install-git\"\u003ehere\u003c/a\u003e):\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"git clone https://github.com/sunericd/TISSUE.git\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egit clone https://github.com/sunericd/TISSUE.git\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe recommend setting up a conda environment to install the requirements for the package (instructions for installing conda and what conda environment can do can be found \u003ca href=\"https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html\" rel=\"nofollow\"\u003ehere\u003c/a\u003e). Installation of requirements can then be done with the following commands:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"conda create -n tissue python=3.8\nconda activate tissue\n\ncd TISSUE\npip install -r requirements.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003econda create -n tissue python=3.8\nconda activate tissue\n\ncd TISSUE\npip install -r requirements.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo keep the requirements light, we have only included packages that are necessary for the core functionalities of TISSUE. For additional utilities such as gene prediction with Tangram, please install those packages separately (or uncomment those lines in \u003ccode\u003erequirements.txt\u003c/code\u003e).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTo test that the installation is working correctly, you can run \u003ccode\u003epython test.py\u003c/code\u003e in the cloned directory.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTISSUE Tutorials\u003c/h1\u003e\u003ca id=\"user-content-tissue-tutorials\" class=\"anchor\" aria-label=\"Permalink: TISSUE Tutorials\" href=\"#tissue-tutorials\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBelow we include several mini-tutorials to highlight the main TISSUE pipeline and downstream applications. These tutorials rely on a small test dataset (a subset of one of the datasets used in the original publication) for fast processing but the approaches can readily be extended to other datasets. For larger-scale examples, please refer to the code repository corresponding the figures and analyses generated for the TISSUE manuscript: \u003ca href=\"https://github.com/sunericd/tissue-figures-and-analyses.git\"\u003ehttps://github.com/sunericd/tissue-figures-and-analyses.git\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# import packages\n\nimport tissue.main, tissue.downstream\n\nimport numpy as np\nimport pandas as pd\nimport matplotlib.pyplot as plt\nimport scanpy as sc\nimport anndata as ad\nimport os\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# import packages\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003emain\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003edownstream\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enumpy\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eas\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003epandas\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eas\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003epd\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ematplotlib\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003epyplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eas\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003escanpy\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eas\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003esc\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eanndata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eas\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ead\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eos\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eUpstream TISSUE analyses: spatial gene expression prediction and uncertainty calibration\u003c/h2\u003e\u003ca id=\"user-content-upstream-tissue-analyses-spatial-gene-expression-prediction-and-uncertainty-calibration\" class=\"anchor\" aria-label=\"Permalink: Upstream TISSUE analyses: spatial gene expression prediction and uncertainty calibration\" href=\"#upstream-tissue-analyses-spatial-gene-expression-prediction-and-uncertainty-calibration\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe first part of the TISSUE pipeline involves making predictions of spatial gene expression profiles using paired spatial transcriptomics and scRNAseq datasets, and then computing and calibrating uncertainties for these predictions that are translated to prediction intervals. All functions for upstream analysis are in the \u003ccode\u003etissue.main\u003c/code\u003e module. These include:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003etissue.main.load_paired_datasets()\u003c/code\u003e for loading paired datasets from formatted directories\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003etissue.main.predict_gene_expression()\u003c/code\u003e for predicting spatial gene expression from paired datasets using a specified prediction method and number of cross-validation folds\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003etissue.main.build_spatial_graph()\u003c/code\u003e for building a cell-cell spatial graph to define neighborhoods for computing the cell-centric variability\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003etissue.main.conformalize_spatial_uncertainty()\u003c/code\u003e for computing cell-centric variability, stratified grouping, and the TISSUE calibration scores for predicted gene expression\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003etissue.main.conformalize_prediction_interval()\u003c/code\u003e for building prediction intervals from uncertainty measures\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTutorial 1: Predicting spatial gene expression\u003c/h3\u003e\u003ca id=\"user-content-tutorial-1-predicting-spatial-gene-expression\" class=\"anchor\" aria-label=\"Permalink: Tutorial 1: Predicting spatial gene expression\" href=\"#tutorial-1-predicting-spatial-gene-expression\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFirst, we load a minimal subset of osmFISH spatial transcriptomics data of mouse somatosensory cortex published by Codeluppi et al., 2018: \u003ca href=\"https://doi.org/10.1038/s41592-018-0175-z\" rel=\"nofollow\"\u003ehttps://doi.org/10.1038/s41592-018-0175-z\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNote that we are using the TISSUE methods for building an AnnData object from tab-delimited text files for the spatial counts, scRNAseq counts, spatial locations, and spatial metadata. If you already have an AnnData object for the spatial data and another object for the scRNAseq data, you can skip this step.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# load in spatial and scRNAseq datasets\n\nadata, RNAseq_adata = tissue.main.load_paired_datasets(\u0026quot;tests/data/Spatial_count.txt\u0026quot;,\n                                                       \u0026quot;tests/data/Locations.txt\u0026quot;,\n                                                       \u0026quot;tests/data/scRNA_count.txt\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# load in spatial and scRNAseq datasets\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eRNAseq_adata\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003emain\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eload_paired_datasets\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"tests/data/Spatial_count.txt\"\u003c/span\u003e,\n                                                       \u003cspan class=\"pl-s\"\u003e\"tests/data/Locations.txt\"\u003c/span\u003e,\n                                                       \u003cspan class=\"pl-s\"\u003e\"tests/data/scRNA_count.txt\"\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:117: ImplicitModificationWarning: Transforming to str index.\n  warnings.warn(\u0026quot;Transforming to str index.\u0026quot;, ImplicitModificationWarning)\n/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:856: UserWarning: \nAnnData expects .obs.index to contain strings, but got values like:\n    [0, 1, 2, 3, 4]\n\n    Inferred to be: integer\n\n  names = self._prep_dim_index(names, \u0026quot;obs\u0026quot;)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:117: ImplicitModificationWarning: Transforming to str index.\n  warnings.warn(\"Transforming to str index.\", ImplicitModificationWarning)\n/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:856: UserWarning: \nAnnData expects .obs.index to contain strings, but got values like:\n    [0, 1, 2, 3, 4]\n\n    Inferred to be: integer\n\n  names = self._prep_dim_index(names, \"obs\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow we can impute any genes of interest that are found in the scRNAseq dataset but not in the spatial dataset. In this case, we will hold out a target gene from the spatial data and apply an imputation method to predict its expression using the scRNAseq dataset.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFirst, we preprocess the data and make sure that the gene names are matchable across the two datasets:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# make genes lowercase\nadata.var_names = [x.lower() for x in adata.var_names]\nRNAseq_adata.var_names = [x.lower() for x in RNAseq_adata.var_names]\n\n# preprocess RNAseq data\ntissue.main.preprocess_data(RNAseq_adata, standardize=False, normalize=True)\n\n# subset spatial data into shared genes\ngene_names = np.intersect1d(adata.var_names, RNAseq_adata.var_names)\nadata = adata[:, gene_names].copy()\n\n# hold out target gene\ntarget_gene = \u0026quot;plp1\u0026quot;\ntarget_expn = adata[:, target_gene].X.copy()\nadata = adata[:, [gene for gene in gene_names if gene != target_gene]].copy()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# make genes lowercase\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003evar_names\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e [\u003cspan class=\"pl-s1\"\u003ex\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003elower\u003c/span\u003e() \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003evar_names\u003c/span\u003e]\n\u003cspan class=\"pl-v\"\u003eRNAseq_adata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003evar_names\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e [\u003cspan class=\"pl-s1\"\u003ex\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003elower\u003c/span\u003e() \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eRNAseq_adata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003evar_names\u003c/span\u003e]\n\n\u003cspan class=\"pl-c\"\u003e# preprocess RNAseq data\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003emain\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epreprocess_data\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eRNAseq_adata\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003estandardize\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eFalse\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003enormalize\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTrue\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e# subset spatial data into shared genes\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003egene_names\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eintersect1d\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003evar_names\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eRNAseq_adata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003evar_names\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e[:, \u003cspan class=\"pl-s1\"\u003egene_names\u003c/span\u003e].\u003cspan class=\"pl-en\"\u003ecopy\u003c/span\u003e()\n\n\u003cspan class=\"pl-c\"\u003e# hold out target gene\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\"plp1\"\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003etarget_expn\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e[:, \u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e].\u003cspan class=\"pl-v\"\u003eX\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003ecopy\u003c/span\u003e()\n\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e[:, [\u003cspan class=\"pl-s1\"\u003egene\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003egene\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003egene_names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003egene\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e!=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e]].\u003cspan class=\"pl-en\"\u003ecopy\u003c/span\u003e()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# dimensions of spatial transcriptomics dataset (number of cells x number of genes)\nadata.shape\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# dimensions of spatial transcriptomics dataset (number of cells x number of genes)\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eshape\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"(3405, 31)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e(3405, 31)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# dimensions of RNAseq dataset (number of cells x number of genes)\nRNAseq_adata.shape\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# dimensions of RNAseq dataset (number of cells x number of genes)\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eRNAseq_adata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eshape\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"(1000, 32)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e(1000, 32)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow, we can make predictions of the target gene expression. In TISSUE, we currently have several methods for gene imputation including SpaGE, Tangram, and Harmony-kNN. We will be using SpaGE in this example:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# SpaGE spatial gene expression prediction\n\ntissue.main.predict_gene_expression (adata, RNAseq_adata, [target_gene],\n                                     method=\u0026quot;spage\u0026quot;, n_folds=10, n_pv=10)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# SpaGE spatial gene expression prediction\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003emain\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epredict_gene_expression\u003c/span\u003e (\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eRNAseq_adata\u003c/span\u003e, [\u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e],\n                                     \u003cspan class=\"pl-s1\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"spage\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003en_folds\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003en_pv\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHow good is the imputation? Since we left out this gene from the spatial data, we can plot the predicted and actual expression and visually inspect the agreement.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Visualizing predicted and actual expression side by side\n\nfig, (ax1, ax2) = plt.subplots(1, 2)\n\nax1.axis('off')\ncmap = target_expn\ncmap[cmap\u0026lt;0] = 0\ncmap = np.log1p(cmap)\ncmap[cmap \u0026gt; np.percentile(cmap,95)] = np.percentile(cmap,95)\nim = ax1.scatter(adata.obsm['spatial'][:,0],adata.obsm['spatial'][:,1],s=1,c=cmap,rasterized=True)\nax1.set_title('Actual', fontsize = 12)\n\ncbar = fig.colorbar(im)\ncbar.ax.get_yaxis().labelpad = 15\ncbar.ax.set_ylabel('Log Expression', rotation=270)\n\nax2.axis('off')\ncmap = adata.obsm['spage_predicted_expression'][target_gene].values\ncmap[cmap\u0026lt;0] = 0\ncmap = np.log1p(cmap)\ncmap[cmap \u0026gt; np.percentile(cmap,95)] = np.percentile(cmap,95)\nim = ax2.scatter(adata.obsm['spatial'][:,0],adata.obsm['spatial'][:,1],s=1,c=cmap,rasterized=True)\nax2.set_title('Predicted', fontsize = 12)\n\ncbar = fig.colorbar(im)\ncbar.ax.get_yaxis().labelpad = 15\ncbar.ax.set_ylabel('Log Expression', rotation=270)\n\nplt.suptitle(\u0026quot;SpaGE Prediction\u0026quot;, fontsize=16)\nplt.tight_layout()\nplt.show()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# Visualizing predicted and actual expression side by side\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003efig\u003c/span\u003e, (\u003cspan class=\"pl-s1\"\u003eax1\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003eax2\u003c/span\u003e) \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003esubplots\u003c/span\u003e(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n\n\u003cspan class=\"pl-s1\"\u003eax1\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eaxis\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'off'\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etarget_expn\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e] \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003elog1p\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epercentile\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e95\u003c/span\u003e)] \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epercentile\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e95\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eim\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eax1\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003escatter\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e],\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s1\"\u003es\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-s1\"\u003erasterized\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTrue\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eax1\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eset_title\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'Actual'\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003efontsize\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e)\n\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003efig\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003ecolorbar\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eim\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eax\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eget_yaxis\u003c/span\u003e().\u003cspan class=\"pl-s1\"\u003elabelpad\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eax\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eset_ylabel\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'Log Expression'\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003erotation\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e270\u003c/span\u003e)\n\n\u003cspan class=\"pl-s1\"\u003eax2\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eaxis\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'off'\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spage_predicted_expression'\u003c/span\u003e][\u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e] \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003elog1p\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epercentile\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e95\u003c/span\u003e)] \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epercentile\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e95\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eim\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eax2\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003escatter\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e],\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s1\"\u003es\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-s1\"\u003erasterized\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTrue\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eax2\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eset_title\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'Predicted'\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003efontsize\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e)\n\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003efig\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003ecolorbar\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eim\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eax\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eget_yaxis\u003c/span\u003e().\u003cspan class=\"pl-s1\"\u003elabelpad\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eax\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eset_ylabel\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'Log Expression'\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003erotation\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e270\u003c/span\u003e)\n\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003esuptitle\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"SpaGE Prediction\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003efontsize\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003etight_layout\u003c/span\u003e()\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eshow\u003c/span\u003e()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/sunericd/TISSUE/blob/master/README_files/README_13_0.png\"\u003e\u003cimg src=\"/sunericd/TISSUE/raw/master/README_files/README_13_0.png\" alt=\"png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNot too bad, especially considering that we used a downsampled scRNAseq dataset for this imputation.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTutorial 2: Using TISSUE to calibrate uncertainties and obtain prediction intervals\u003c/h3\u003e\u003ca id=\"user-content-tutorial-2-using-tissue-to-calibrate-uncertainties-and-obtain-prediction-intervals\" class=\"anchor\" aria-label=\"Permalink: Tutorial 2: Using TISSUE to calibrate uncertainties and obtain prediction intervals\" href=\"#tutorial-2-using-tissue-to-calibrate-uncertainties-and-obtain-prediction-intervals\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNote that when we ran \u003ccode\u003etissue.main.predict_gene_expression()\u003c/code\u003e in the previous tutorial, we obtained cross-validated predictions for all genes in the existing spatial dataset (e.g. 10 folds). These will now come in handy when we compute and calibrate TISSUE uncertainties for the predicted gene expression.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFirst, we build spatial graphs using TISSUE (this can also be done with native Scanpy functions):\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# build spatial graph and calculate adjacency weights\n\ntissue.main.build_spatial_graph(adata, method=\u0026quot;fixed_radius\u0026quot;, n_neighbors=15)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# build spatial graph and calculate adjacency weights\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003emain\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003ebuild_spatial_graph\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"fixed_radius\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003en_neighbors\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe entire TISSUE spatial uncertainty generation pipeline can be launched with one line of code using \u003ccode\u003etissue.main.conformalize_spatial_uncertainty()\u003c/code\u003e. Here, the first two arguments are the spatial AnnData object and a string specifier for the key in \u003ccode\u003eobsm\u003c/code\u003e corresponding to the predicted gene expression. We can specify which of the genes to use in the calibration set (generally this is all genes in the spatial data). There are other arguments for the grouping setup and weighting schemes, but we will use the default settings in this tutorial.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# build calibration scores\n\ntissue.main.conformalize_spatial_uncertainty(adata, \u0026quot;spage_predicted_expression\u0026quot;, calib_genes=adata.var_names,\n                                             grouping_method=\u0026quot;kmeans_gene_cell\u0026quot;, k=4, k2=2)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# build calibration scores\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003emain\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003econformalize_spatial_uncertainty\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\"spage_predicted_expression\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003ecalib_genes\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003evar_names\u003c/span\u003e,\n                                             \u003cspan class=\"pl-s1\"\u003egrouping_method\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"kmeans_gene_cell\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003ek\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003ek2\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\n/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\n/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\n/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\n/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\n/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\n/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\n/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\n/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFinally, we can calibrate the spatial uncertainty measures to get calibration scores and then use those to extract prediction intervals for any confidence level \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"6c11f29f82daf890c76b434fc037a46d\"\u003e$1-\\alpha$\u003c/math-renderer\u003e. This is done with \u003ccode\u003etissue.main.conformalize_prediction_interval()\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# get prediction interval for 67% coverage\n\ntissue.main.conformalize_prediction_interval (adata, \u0026quot;spage_predicted_expression\u0026quot;, calib_genes=adata.var_names,\n                                              alpha_level=0.23)\n\n# set `compute_wasserstein=True` to compute a measure indicating the distance between the cell-centric variability values \n# of the predicted genes and the values of their support (i.e. calibration group) in the original data. \n# The lower this value, the better supported the predicted group.\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# get prediction interval for 67% coverage\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003emain\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003econformalize_prediction_interval\u003c/span\u003e (\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\"spage_predicted_expression\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003ecalib_genes\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003evar_names\u003c/span\u003e,\n                                              \u003cspan class=\"pl-s1\"\u003ealpha_level\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.23\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e# set `compute_wasserstein=True` to compute a measure indicating the distance between the cell-centric variability values \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e# of the predicted genes and the values of their support (i.e. calibration group) in the original data. \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e# The lower this value, the better supported the predicted group.\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow let's visualize what these prediction intervals look like for the target (unseen) gene and how it compares to the actual prediction errors.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"m = \u0026quot;spage\u0026quot;\n\ni = np.where(adata.var_names == target_gene)[0]\n\n# define consistent color map\nall_vals = np.concatenate((np.abs(target_expn.flatten()-adata.obsm[m+\u0026quot;_predicted_expression\u0026quot;][target_gene].values),\n                           adata.obsm[m+f\u0026quot;_predicted_expression_hi\u0026quot;][target_gene].values-adata.obsm[m+f\u0026quot;_predicted_expression_lo\u0026quot;][target_gene].values))\nall_vals[all_vals\u0026lt;0]=0\nvmin = np.percentile(np.log1p(all_vals), 0)\nvmax = np.percentile(np.log1p(all_vals), 95)\n\n\nfig, (ax1, ax2) = plt.subplots(1, 2)\n\nax1.axis('off')\ncmap = np.abs(target_expn.flatten()-adata.obsm[m+\u0026quot;_predicted_expression\u0026quot;][target_gene].values)\ncmap[cmap\u0026lt;0] = 0\ncmap = np.log1p(cmap)\ncmap[cmap \u0026gt; np.percentile(cmap,95)] = np.percentile(cmap,95)\nim = ax1.scatter(adata.obsm['spatial'][:,0],adata.obsm['spatial'][:,1],s=1,c=cmap,rasterized=True)#,vmin=vmin,vmax=vmax)\nax1.set_title('Imputation Error ' + target_gene, fontsize = 12)\n\ncbar = fig.colorbar(im)\ncbar.ax.get_yaxis().labelpad = 15\ncbar.ax.set_ylabel('Log Expression', rotation=270)\n\nax2.axis('off')\ncmap = adata.obsm[m+f\u0026quot;_predicted_expression_hi\u0026quot;][target_gene].values-adata.obsm[m+f\u0026quot;_predicted_expression_lo\u0026quot;][target_gene].values\ncmap[cmap\u0026lt;0] = 0\ncmap = np.log1p(cmap)\ncmap[cmap \u0026gt; np.percentile(cmap,95)] = np.percentile(cmap,95)\nim = ax2.scatter(adata.obsm['spatial'][:,0],adata.obsm['spatial'][:,1],s=1,c=cmap,rasterized=True)#,vmin=vmin,vmax=vmax)\nax2.set_title('PI Width ' + target_gene, fontsize = 12)\n\ncbar = fig.colorbar(im)\ncbar.ax.get_yaxis().labelpad = 15\ncbar.ax.set_ylabel('Log Expression', rotation=270)\n\nplt.suptitle(m, fontsize=16)\nplt.tight_layout()\nplt.show()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-s1\"\u003em\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\"spage\"\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003ewhere\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003evar_names\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e]\n\n\u003cspan class=\"pl-c\"\u003e# define consistent color map\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003eall_vals\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003econcatenate\u003c/span\u003e((\u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eabs\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003etarget_expn\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eflatten\u003c/span\u003e()\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003em\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"_predicted_expression\"\u003c/span\u003e][\u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e),\n                           \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003em\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-s\"\u003ef\"_predicted_expression_hi\"\u003c/span\u003e][\u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003em\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-s\"\u003ef\"_predicted_expression_lo\"\u003c/span\u003e][\u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e))\n\u003cspan class=\"pl-s1\"\u003eall_vals\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eall_vals\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e]\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003evmin\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epercentile\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003elog1p\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eall_vals\u003c/span\u003e), \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003evmax\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epercentile\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003elog1p\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eall_vals\u003c/span\u003e), \u003cspan class=\"pl-c1\"\u003e95\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-s1\"\u003efig\u003c/span\u003e, (\u003cspan class=\"pl-s1\"\u003eax1\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003eax2\u003c/span\u003e) \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003esubplots\u003c/span\u003e(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n\n\u003cspan class=\"pl-s1\"\u003eax1\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eaxis\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'off'\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eabs\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003etarget_expn\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eflatten\u003c/span\u003e()\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003em\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"_predicted_expression\"\u003c/span\u003e][\u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e] \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003elog1p\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epercentile\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e95\u003c/span\u003e)] \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epercentile\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e95\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eim\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eax1\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003escatter\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e],\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s1\"\u003es\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-s1\"\u003erasterized\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTrue\u003c/span\u003e)\u003cspan class=\"pl-c\"\u003e#,vmin=vmin,vmax=vmax)\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003eax1\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eset_title\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'Imputation Error '\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003efontsize\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e)\n\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003efig\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003ecolorbar\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eim\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eax\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eget_yaxis\u003c/span\u003e().\u003cspan class=\"pl-s1\"\u003elabelpad\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eax\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eset_ylabel\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'Log Expression'\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003erotation\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e270\u003c/span\u003e)\n\n\u003cspan class=\"pl-s1\"\u003eax2\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eaxis\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'off'\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003em\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-s\"\u003ef\"_predicted_expression_hi\"\u003c/span\u003e][\u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003em\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-s\"\u003ef\"_predicted_expression_lo\"\u003c/span\u003e][\u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e] \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003elog1p\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epercentile\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e95\u003c/span\u003e)] \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epercentile\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e95\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eim\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eax2\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003escatter\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e],\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\u003cspan class=\"pl-s1\"\u003es\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003ecmap\u003c/span\u003e,\u003cspan class=\"pl-s1\"\u003erasterized\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTrue\u003c/span\u003e)\u003cspan class=\"pl-c\"\u003e#,vmin=vmin,vmax=vmax)\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003eax2\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eset_title\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'PI Width '\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003efontsize\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e)\n\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003efig\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003ecolorbar\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eim\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eax\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eget_yaxis\u003c/span\u003e().\u003cspan class=\"pl-s1\"\u003elabelpad\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ecbar\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eax\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eset_ylabel\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e'Log Expression'\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003erotation\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e270\u003c/span\u003e)\n\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003esuptitle\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003em\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003efontsize\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003etight_layout\u003c/span\u003e()\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eshow\u003c/span\u003e()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/sunericd/TISSUE/blob/master/README_files/README_22_0.png\"\u003e\u003cimg src=\"/sunericd/TISSUE/raw/master/README_files/README_22_0.png\" alt=\"png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe TISSUE prediction intervals are decent and match the distribution of imputation errors especially well in the bottom portions of the section. On the full dataset, which has much richer reference scRNAseq data, the calibration quality is further improved (see Figure 2 of the TISSUE manuscript).\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDownstream TISSUE analyses: Hypothesis testing, Clustering/Visualization, Prediction\u003c/h2\u003e\u003ca id=\"user-content-downstream-tissue-analyses-hypothesis-testing-clusteringvisualization-prediction\" class=\"anchor\" aria-label=\"Permalink: Downstream TISSUE analyses: Hypothesis testing, Clustering/Visualization, Prediction\" href=\"#downstream-tissue-analyses-hypothesis-testing-clusteringvisualization-prediction\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTISSUE provides additional functionalities for leveraging these uncertainty estimates and prediction intervals in common downstream single-cell spatial transcriptomics analysese. All functions for downstream analysis are in the \u003ccode\u003etissue.downstream\u003c/code\u003e module. These include:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003etissue.downstream.multiple_imputation_testing()\u003c/code\u003e for hypothesis testing using multiple imputations drawn from the calibration score sets\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003etissue.downstream.weighted_PCA()\u003c/code\u003e for computing weighted principal components where weights correspond to a transform of the inverse prediction interval width\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003etissue.downstream.detect_uncertain_cells()\u003c/code\u003e for filtering low-confidence cells from data before training and evaluation of machine learning models, which generally improves performance\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eBelow we include a few example uses of these modules. For more examples (including those for the experiments in the associated manuscript), please see the Github repository: \u003ca href=\"https://github.com/sunericd/tissue-figures-and-analyses.git\"\u003ehttps://github.com/sunericd/tissue-figures-and-analyses.git\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTutorial 3: Hypothesis testing with TISSUE multiple imputation framework\u003c/h3\u003e\u003ca id=\"user-content-tutorial-3-hypothesis-testing-with-tissue-multiple-imputation-framework\" class=\"anchor\" aria-label=\"Permalink: Tutorial 3: Hypothesis testing with TISSUE multiple imputation framework\" href=\"#tutorial-3-hypothesis-testing-with-tissue-multiple-imputation-framework\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis tutorial uses the TISSUE calibration scores to generate multiple imputations and then perform hypothesis testing by aggregating statistics across these multiple imputations. The default and recommended statistical test for this framework is the Student's t-test, but other options for spatially variable gene detection using SpatialDE and non-parametric (one-sided) Mann-Whitney/Wilcoxon tests can also be performed by specifying \u003ccode\u003emethod=\"spatialde\"\u003c/code\u003e, \u003ccode\u003emethod=\"wilcoxon_greater\"\u003c/code\u003e, \u003ccode\u003emethod=\"wilcoxon_less\"\u003c/code\u003e for \u003ccode\u003etissue.downstream.multiple_imputation_testing()\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePlease run the code in Tutorials 1-2 to generate predictions and TISSUE uncertainty measures before this tutorial.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAfter, we construct some binary labels for the cells in the dataset:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# split into two groups based on indices\nadata.obs['condition'] = ['A' if i \u0026lt; round(adata.shape[0]/2) else 'B' for i in range(adata.shape[0])]\n\n# plot conditions\nplt.scatter(adata[adata.obs.condition==\u0026quot;A\u0026quot;].obsm['spatial'][:,0],\n            adata[adata.obs.condition==\u0026quot;A\u0026quot;].obsm['spatial'][:,1],\n            c='tab:red', s=3, label=\u0026quot;A\u0026quot;)\nplt.scatter(adata[adata.obs.condition==\u0026quot;B\u0026quot;].obsm['spatial'][:,0],\n            adata[adata.obs.condition==\u0026quot;B\u0026quot;].obsm['spatial'][:,1],\n            c='tab:blue', s=3, label=\u0026quot;B\u0026quot;)\nplt.legend(loc='best')\nplt.show()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# split into two groups based on indices\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e] \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e [\u003cspan class=\"pl-s\"\u003e'A'\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003eround\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eshape\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e]\u003cspan class=\"pl-c1\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003eelse\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e'B'\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-en\"\u003erange\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eshape\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e])]\n\n\u003cspan class=\"pl-c\"\u003e# plot conditions\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003escatter\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003econdition\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"A\"\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e],\n            \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003econdition\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"A\"\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\n            \u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'tab:red'\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003es\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"A\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003escatter\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003econdition\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"B\"\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e],\n            \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003econdition\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"B\"\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spatial'\u003c/span\u003e][:,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\n            \u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'tab:blue'\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003es\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"B\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003elegend\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eloc\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'best'\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eshow\u003c/span\u003e()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/sunericd/TISSUE/blob/master/README_files/README_26_0.png\"\u003e\u003cimg src=\"/sunericd/TISSUE/raw/master/README_files/README_26_0.png\" alt=\"png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs we can see, the cells in group A primarily belong to the medial layers of the section while the cells in group B correspond to upper pia layer and also the bottom layers of the section. As such, when we do differential gene expression analysis, we should expect some differentially expressed markers between these two labels.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNow, we perform differential gene expression analysis using TISSUE multiple imputation hypothesis testing, which works by using TISSUE calibration scores to sample multiple \"imputations\" (alternative predictions) and then aggregate these statistics together afterwards. Here, we set \u003ccode\u003egroup1=\"A\"\u003c/code\u003e and \u003ccode\u003egroup2=\"B\"\u003c/code\u003e to find genes that differentially expressed between the two groups across \u003ccode\u003en_imputations=10\u003c/code\u003e number of imputations (higher values here are better but take more time to compute):\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# multiple imputation hypothesis testing\n\ntissue.downstream.multiple_imputation_testing(adata, \u0026quot;spage_predicted_expression\u0026quot;,\n                                              calib_genes=adata.var_names,\n                                              condition='condition',\n                                              group1 = \u0026quot;A\u0026quot;, # use None to compute for all conditions, condition vs all\n                                              group2 = \u0026quot;B\u0026quot;, # use None to compute for group1 vs all\n                                              n_imputations=10)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# multiple imputation hypothesis testing\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003edownstream\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003emultiple_imputation_testing\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\"spage_predicted_expression\"\u003c/span\u003e,\n                                              \u003cspan class=\"pl-s1\"\u003ecalib_genes\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003evar_names\u003c/span\u003e,\n                                              \u003cspan class=\"pl-s1\"\u003econdition\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e,\n                                              \u003cspan class=\"pl-s1\"\u003egroup1\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\"A\"\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e# use None to compute for all conditions, condition vs all\u003c/span\u003e\n                                              \u003cspan class=\"pl-s1\"\u003egroup2\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\"B\"\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e# use None to compute for group1 vs all\u003c/span\u003e\n                                              \u003cspan class=\"pl-s1\"\u003en_imputations\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTISSUE multiple imputation testing saves the results directing within the \u003ccode\u003eadata.uns\u003c/code\u003e metadata and can be read out by the compared groups and the name of the statistic:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# extract statistics for target_gene\nprint(\u0026quot;t-statistic = \u0026quot;+str(round(adata.uns['spage_A_B_tstat'][target_gene].values[0],5)))\nprint(\u0026quot;P = \u0026quot;+str(round(adata.uns['spage_A_B_pvalue'][target_gene].values[0],5)))\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# extract statistics for target_gene\u003c/span\u003e\n\u003cspan class=\"pl-en\"\u003eprint\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"t-statistic = \"\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-en\"\u003estr\u003c/span\u003e(\u003cspan class=\"pl-en\"\u003eround\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003euns\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spage_A_B_tstat'\u003c/span\u003e][\u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e],\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e)))\n\u003cspan class=\"pl-en\"\u003eprint\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"P = \"\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-en\"\u003estr\u003c/span\u003e(\u003cspan class=\"pl-en\"\u003eround\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003euns\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spage_A_B_pvalue'\u003c/span\u003e][\u003cspan class=\"pl-s1\"\u003etarget_gene\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e],\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e)))\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"t-statistic = -2.38565\nP = 0.02371\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003et-statistic = -2.38565\nP = 0.02371\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTesting of our target gene (Plp1) results in significant under-expression in group A as compared to group B, suggesting that Plp1 could be a marker gene for the cell types / regions in group B.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNOTE: Many TISSUE modules rely on stochastic sampling so the printed metrics may vary slightly from run to run.\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTutorial 4: TISSUE cell filtering for supervised learning\u003c/h3\u003e\u003ca id=\"user-content-tutorial-4-tissue-cell-filtering-for-supervised-learning\" class=\"anchor\" aria-label=\"Permalink: Tutorial 4: TISSUE cell filtering for supervised learning\" href=\"#tutorial-4-tissue-cell-filtering-for-supervised-learning\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTISSUE cell filtering removes cells with the greatest average uncertainty in predicted gene expression, which generally improves the performance of supervised learning models (i.e. classifiers) when trained and evaluated on these filtered predicted expression data.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn this tutorial, we will filter out the uncertain cells (using automatic Otsu thresholding) and then train and evaluate a logistic regression classifier to predict the two cell groups (A vs B) from Tutorial 3.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTo start, we will need to compute the TISSUE prediction interval width as a proxy for uncertainty. We do this by subtracting the lower bound from the upper bound:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# get uncertainty (PI width) for filtering\n\nX_uncertainty = adata.obsm[\u0026quot;spage_predicted_expression_hi\u0026quot;].values - adata.obsm[\u0026quot;spage_predicted_expression_lo\u0026quot;].values\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# get uncertainty (PI width) for filtering\u003c/span\u003e\n\n\u003cspan class=\"pl-v\"\u003eX_uncertainty\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e\"spage_predicted_expression_hi\"\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e\"spage_predicted_expression_lo\"\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThen we can filter using the TISSUE prediction interval width. We perform filtering within each strata (i.e. cell group label \"A\" or \"B\"), but this can also be done across other groupings or across the entire population of cells if desired. Here we use Otsu thresholding to automatically determine the proportion of cells to filter out within each strata, but you can set this to a hard threshold if desired.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# uncertainty-based cell filtering\n\nkeep_idxs = tissue.downstream.detect_uncertain_cells (X_uncertainty,\n                                                      proportion=\u0026quot;otsu\u0026quot;,\n                                                      stratification=adata.obs['condition'].values)\n\nadata_filtered = adata[adata.obs_names[keep_idxs],:].copy()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# uncertainty-based cell filtering\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003ekeep_idxs\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003edownstream\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003edetect_uncertain_cells\u003c/span\u003e (\u003cspan class=\"pl-v\"\u003eX_uncertainty\u003c/span\u003e,\n                                                      \u003cspan class=\"pl-s1\"\u003eproportion\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"otsu\"\u003c/span\u003e,\n                                                      \u003cspan class=\"pl-s1\"\u003estratification\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e)\n\n\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs_names\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003ekeep_idxs\u003c/span\u003e],:].\u003cspan class=\"pl-en\"\u003ecopy\u003c/span\u003e()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow that we have an object with filtered predicted gene expression, we can check to see how the dimensions of our data have changed from TISSUE filtering:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# examine dimensions of data before/after TISSUE filtering\n\nprint(\u0026quot;Before TISSUE cell filtering:\u0026quot;)\nprint(adata.shape)\nprint(\u0026quot;\\nAfter TISSUE cell filtering:\u0026quot;)\nprint(adata_filtered.shape)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# examine dimensions of data before/after TISSUE filtering\u003c/span\u003e\n\n\u003cspan class=\"pl-en\"\u003eprint\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"Before TISSUE cell filtering:\"\u003c/span\u003e)\n\u003cspan class=\"pl-en\"\u003eprint\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eshape\u003c/span\u003e)\n\u003cspan class=\"pl-en\"\u003eprint\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"\u003cspan class=\"pl-cce\"\u003e\\n\u003c/span\u003eAfter TISSUE cell filtering:\"\u003c/span\u003e)\n\u003cspan class=\"pl-en\"\u003eprint\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eshape\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"Before TISSUE cell filtering:\n(3405, 31)\n\nAfter TISSUE cell filtering:\n(2862, 31)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eBefore TISSUE cell filtering:\n(3405, 31)\n\nAfter TISSUE cell filtering:\n(2862, 31)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAnd similarly, we can check the balance in the two cell groups after filtering:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# print balance of labels in the filtered dataset\n\npd.DataFrame(np.unique(adata_filtered.obs['condition'], return_counts=True),index=[\u0026quot;Group\u0026quot;,\u0026quot;Number of Cells\u0026quot;])\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# print balance of labels in the filtered dataset\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003epd\u003c/span\u003e.\u003cspan class=\"pl-v\"\u003eDataFrame\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eunique\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e], \u003cspan class=\"pl-s1\"\u003ereturn_counts\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTrue\u003c/span\u003e),\u003cspan class=\"pl-s1\"\u003eindex\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e\"Group\"\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\"Number of Cells\"\u003c/span\u003e])\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u0026lt;style scoped\u0026gt;\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\".dataframe tbody tr th {\n    vertical-align: top;\n}\n\n.dataframe thead th {\n    text-align: right;\n}\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e.dataframe tbody tr th {\n    vertical-align: top;\n}\n\n.dataframe thead th {\n    text-align: right;\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u0026lt;/style\u0026gt;\n\u003ctable border=\"1\"\u003e\n  \u003cthead\u003e\n    \u003ctr\u003e\n      \u003cth\u003e\u003c/th\u003e\n      \u003cth\u003e0\u003c/th\u003e\n      \u003cth\u003e1\u003c/th\u003e\n    \u003c/tr\u003e\n  \u003c/thead\u003e\n  \u003ctbody\u003e\n    \u003ctr\u003e\n      \u003cth\u003eGroup\u003c/th\u003e\n      \u003ctd\u003eA\u003c/td\u003e\n      \u003ctd\u003eB\u003c/td\u003e\n    \u003c/tr\u003e\n    \u003ctr\u003e\n      \u003cth\u003eNumber of Cells\u003c/th\u003e\n      \u003ctd\u003e1250\u003c/td\u003e\n      \u003ctd\u003e1612\u003c/td\u003e\n    \u003c/tr\u003e\n  \u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAs we can see, TISSUE automatically filters out a large number of cells with uncertain gene predictions. In the filtered dataset, the balance between group A and group B is relatively preserved.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNow, we will move on to training a logistic regression classifier on the filtered data. Given the modular nature of TISSUE filtering (i.e. the output is a cellxgene matrix), integrating TISSUE with other supervised learning models is as easy as plug-and-play.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFirst, we split into a train (80%) and test set (20%):\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# split train and test randomly (80%-20%)\nnp.random.seed(444)\ntrain_idxs = np.random.choice(np.arange(adata_filtered.shape[0]), round(adata_filtered.shape[0]*0.8), replace=False)\ntest_idxs = np.array([idx for idx in np.arange(adata_filtered.shape[0]) if idx not in train_idxs])\n\ntrain_data = adata_filtered.obsm[\u0026quot;spage_predicted_expression\u0026quot;].values[train_idxs,:]\ntrain_labels = adata_filtered.obs[\u0026quot;condition\u0026quot;][train_idxs]\n\ntest_data = adata_filtered.obsm[\u0026quot;spage_predicted_expression\u0026quot;].values[test_idxs,:]\ntest_labels = adata_filtered.obs[\u0026quot;condition\u0026quot;][test_idxs]\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# split train and test randomly (80%-20%)\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003erandom\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eseed\u003c/span\u003e(\u003cspan class=\"pl-c1\"\u003e444\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003etrain_idxs\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003erandom\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003echoice\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003earange\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eshape\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e]), \u003cspan class=\"pl-en\"\u003eround\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eshape\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e]\u003cspan class=\"pl-c1\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e), \u003cspan class=\"pl-s1\"\u003ereplace\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eFalse\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003etest_idxs\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003earray\u003c/span\u003e([\u003cspan class=\"pl-s1\"\u003eidx\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eidx\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003enp\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003earange\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eshape\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e]) \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eidx\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003enot\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etrain_idxs\u003c/span\u003e])\n\n\u003cspan class=\"pl-s1\"\u003etrain_data\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e\"spage_predicted_expression\"\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003etrain_idxs\u003c/span\u003e,:]\n\u003cspan class=\"pl-s1\"\u003etrain_labels\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e\"condition\"\u003c/span\u003e][\u003cspan class=\"pl-s1\"\u003etrain_idxs\u003c/span\u003e]\n\n\u003cspan class=\"pl-s1\"\u003etest_data\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e\"spage_predicted_expression\"\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003etest_idxs\u003c/span\u003e,:]\n\u003cspan class=\"pl-s1\"\u003etest_labels\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e\"condition\"\u003c/span\u003e][\u003cspan class=\"pl-s1\"\u003etest_idxs\u003c/span\u003e]\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThen, we train logistic regression classifier on the filtered and split data:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"from sklearn.linear_model import LogisticRegression\nfrom sklearn.preprocessing import StandardScaler\nfrom sklearn.metrics import accuracy_score, f1_score, roc_auc_score, average_precision_score\n\n# init and scale data\nscaler = StandardScaler()\ntrain_data = scaler.fit_transform(train_data)\n\n# fit model on scaled data\nmodel = LogisticRegression(penalty='l1', solver='liblinear').fit(train_data, train_labels)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efrom\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003esklearn\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003elinear_model\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eLogisticRegression\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efrom\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003esklearn\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003epreprocessing\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eStandardScaler\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efrom\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003esklearn\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003emetrics\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eaccuracy_score\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003ef1_score\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003eroc_auc_score\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003eaverage_precision_score\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e# init and scale data\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003escaler\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eStandardScaler\u003c/span\u003e()\n\u003cspan class=\"pl-s1\"\u003etrain_data\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003escaler\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003efit_transform\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003etrain_data\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e# fit model on scaled data\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003emodel\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eLogisticRegression\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003epenalty\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'l1'\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003esolver\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'liblinear'\u003c/span\u003e).\u003cspan class=\"pl-en\"\u003efit\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003etrain_data\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003etrain_labels\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow that we have trained the model, we can evaluate its performance on the unseen test data:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# make predictions on test data\npred_test = model.predict(scaler.transform(test_data))\npred_test_scores = model.predict_proba(scaler.transform(test_data))\n\n# print metrics\ntest_labels_num = [0 if x==\u0026quot;A\u0026quot; else 1 for x in test_labels]\nprint(f\u0026quot;Accuracy Score: {accuracy_score(test_labels, pred_test)}\u0026quot;)\nprint(f\u0026quot;ROC-AUC Score: {roc_auc_score(test_labels_num, pred_test_scores[:,1])}\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# make predictions on test data\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003epred_test\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emodel\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epredict\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003escaler\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003etransform\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003etest_data\u003c/span\u003e))\n\u003cspan class=\"pl-s1\"\u003epred_test_scores\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003emodel\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003epredict_proba\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003escaler\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003etransform\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003etest_data\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e# print metrics\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003etest_labels_num\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e [\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"A\"\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eelse\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etest_labels\u003c/span\u003e]\n\u003cspan class=\"pl-en\"\u003eprint\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003ef\"Accuracy Score: \u003cspan class=\"pl-s1\"\u003e\u003cspan class=\"pl-kos\"\u003e{\u003c/span\u003e\u003cspan class=\"pl-en\"\u003eaccuracy_score\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003etest_labels\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003epred_test\u003c/span\u003e)\u003cspan class=\"pl-kos\"\u003e}\u003c/span\u003e\u003c/span\u003e\"\u003c/span\u003e)\n\u003cspan class=\"pl-en\"\u003eprint\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003ef\"ROC-AUC Score: \u003cspan class=\"pl-s1\"\u003e\u003cspan class=\"pl-kos\"\u003e{\u003c/span\u003e\u003cspan class=\"pl-en\"\u003eroc_auc_score\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003etest_labels_num\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003epred_test_scores\u003c/span\u003e[:,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e])\u003cspan class=\"pl-kos\"\u003e}\u003c/span\u003e\u003c/span\u003e\"\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"Accuracy Score: 0.8024475524475524\nROC-AUC Score: 0.8765594181459566\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eAccuracy Score: 0.8024475524475524\nROC-AUC Score: 0.8765594181459566\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe model performs quite well! It has high accuracy and high ROC-AUC for a relatively balanced binary classification problem. A similar approach can be taken to leverage TISSUE uncertainties in training/evaluating other model architectures (e.g. linear regression, random forest, neural nets).\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTutorial 5: TISSUE cell filtering for PCA (clustering and visualization)\u003c/h3\u003e\u003ca id=\"user-content-tutorial-5-tissue-cell-filtering-for-pca-clustering-and-visualization\" class=\"anchor\" aria-label=\"Permalink: Tutorial 5: TISSUE cell filtering for PCA (clustering and visualization)\" href=\"#tutorial-5-tissue-cell-filtering-for-pca-clustering-and-visualization\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDownstream clustering and data visualization tasks in transcriptomics data analysis generally rely on dimensionality reduction via principal component analysis (PCA). To incorporate TISSUE uncertainties in these downstream tasks, we perform TISSUE cell filtering before fitting the PCA model and reducing dimensionality.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn this tutorial, we will apply TISSUE cell filtering to the dataset to generate principal components. This can be done with \u003ccode\u003etissue.downstream.filtered_PCA()\u003c/code\u003e which is wrapper around the direct TISSUE cell filtering. From these principal components, we can make a two-dimensional PCA plot and perform clustering on the top 15 principal components using K-Means.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe will be starting with the AnnData object after running Tutorials 1-3: \u003ccode\u003eadata\u003c/code\u003e and then applying TISSUE-filtered PCA:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# uncertainty-based cell filtering for PCA\n\nkeep_idxs = tissue.downstream.filtered_PCA (adata, # anndata object\n                                            \u0026quot;spage\u0026quot;, # prediction method\n                                            proportion=\u0026quot;otsu\u0026quot;,\n                                            stratification=adata.obs['condition'].values,\n                                            return_keep_idxs=True)\n\n# filter to keep track of labels\nadata_filtered = adata[adata.obs_names[keep_idxs],:].copy()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# uncertainty-based cell filtering for PCA\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003ekeep_idxs\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003edownstream\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003efiltered_PCA\u003c/span\u003e (\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e# anndata object\u003c/span\u003e\n                                            \u003cspan class=\"pl-s\"\u003e\"spage\"\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e# prediction method\u003c/span\u003e\n                                            \u003cspan class=\"pl-s1\"\u003eproportion\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"otsu\"\u003c/span\u003e,\n                                            \u003cspan class=\"pl-s1\"\u003estratification\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e].\u003cspan class=\"pl-s1\"\u003evalues\u003c/span\u003e,\n                                            \u003cspan class=\"pl-s1\"\u003ereturn_keep_idxs\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTrue\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e# filter to keep track of labels\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs_names\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003ekeep_idxs\u003c/span\u003e],:].\u003cspan class=\"pl-en\"\u003ecopy\u003c/span\u003e()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHere we used the same \u003ccode\u003eotsu\u003c/code\u003e threshold-based automatic filtering as before and stratify the filtering by the conditions. We use the default \u003ccode\u003en_components=15\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThere are two options for the TISSUE-filtered PCA, both of which are saved into \u003ccode\u003eadata.uns\u003c/code\u003e and \u003ccode\u003eadata.obsm\u003c/code\u003e respectively. The first is the standard principal components obtained on the TISSUE-filtered data, which can be found in \u003ccode\u003eadata.uns['{name of prediction method}_predicted_expression_PC15_filtered_']\u003c/code\u003e. The second is the PCA fitted onto TISSUE-filtered data but then applied to the entire dataset (unfiltered), which can be found in \u003ccode\u003eadata.obsm['{name of prediction method}_predicted_expression_PC15_']\u003c/code\u003e. We will use the first (and recommended) option in this tutorial\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# retrieve filtered PCA\n\nPC_reduced = adata.uns['spage_predicted_expression_PC15_filtered_'].copy()\nprint(PC_reduced.shape)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# retrieve filtered PCA\u003c/span\u003e\n\n\u003cspan class=\"pl-v\"\u003ePC_reduced\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003euns\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spage_predicted_expression_PC15_filtered_'\u003c/span\u003e].\u003cspan class=\"pl-en\"\u003ecopy\u003c/span\u003e()\n\u003cspan class=\"pl-en\"\u003eprint\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003ePC_reduced\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eshape\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"(2862, 15)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e(2862, 15)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe now have a reduced representation of our original data that is filtered by TISSUE and has 15 principal components. We can visualize the first two principal components:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# make 2D PCA plot labeled by group\n\nplt.title(\u0026quot;TISSUE-Filtered PCA\u0026quot;)\nplt.scatter(PC_reduced[adata_filtered.obs['condition']=='A',0],\n            PC_reduced[adata_filtered.obs['condition']=='A',1],\n            c=\u0026quot;tab:red\u0026quot;, s=3, label=\u0026quot;A\u0026quot;)\nplt.scatter(PC_reduced[adata_filtered.obs['condition']=='B',0],\n            PC_reduced[adata_filtered.obs['condition']=='B',1],\n            c=\u0026quot;tab:blue\u0026quot;, s=3, label=\u0026quot;B\u0026quot;)\nplt.legend(loc='center left', bbox_to_anchor=(1, 0.5))\nplt.xlabel(\u0026quot;PC 1\u0026quot;)\nplt.ylabel(\u0026quot;PC 2\u0026quot;)\nplt.show()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# make 2D PCA plot labeled by group\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003etitle\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"TISSUE-Filtered PCA\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003escatter\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003ePC_reduced\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e]\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'A'\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e],\n            \u003cspan class=\"pl-v\"\u003ePC_reduced\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e]\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'A'\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\n            \u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"tab:red\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003es\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"A\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003escatter\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003ePC_reduced\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e]\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'B'\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e],\n            \u003cspan class=\"pl-v\"\u003ePC_reduced\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e]\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'B'\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\n            \u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"tab:blue\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003es\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"B\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003elegend\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eloc\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'center left'\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003ebbox_to_anchor\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e))\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003exlabel\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"PC 1\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eylabel\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"PC 2\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eshow\u003c/span\u003e()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/sunericd/TISSUE/blob/master/README_files/README_52_0.png\"\u003e\u003cimg src=\"/sunericd/TISSUE/raw/master/README_files/README_52_0.png\" alt=\"png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eVisually, there is some separation between the two groups on the first two principal components (although lots of overlap too).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNext, we can try K-Means clustering using all 15 principal components (and evaluate clustering with the ARI):\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"from sklearn.cluster import KMeans\n\n# K-Means clustering\nkmeans = KMeans(n_clusters=2).fit(PC_reduced)\nclusters = kmeans.labels_\n\n# evaluate ARI\nfrom sklearn.metrics import adjusted_rand_score\nprint(adjusted_rand_score(adata_filtered.obs['condition'], clusters))\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efrom\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003esklearn\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003ecluster\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eKMeans\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e# K-Means clustering\u003c/span\u003e\n\u003cspan class=\"pl-s1\"\u003ekmeans\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-v\"\u003eKMeans\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003en_clusters\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e).\u003cspan class=\"pl-en\"\u003efit\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003ePC_reduced\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eclusters\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003ekmeans\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003elabels_\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e# evaluate ARI\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efrom\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003esklearn\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003emetrics\u003c/span\u003e \u003cspan class=\"pl-k\"\u003eimport\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadjusted_rand_score\u003c/span\u003e\n\u003cspan class=\"pl-en\"\u003eprint\u003c/span\u003e(\u003cspan class=\"pl-en\"\u003eadjusted_rand_score\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata_filtered\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e], \u003cspan class=\"pl-s1\"\u003eclusters\u003c/span\u003e))\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\n\n\n0.218865537222824\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n  warnings.warn(\n\n\n0.218865537222824\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEvidently, the clustering with TISSUE-filtered principal components can provide some degree of separation between the two cell groups that we defined previously.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTutorial 6: TISSUE-WPCA (weighted principal component analysis)\u003c/h3\u003e\u003ca id=\"user-content-tutorial-6-tissue-wpca-weighted-principal-component-analysis\" class=\"anchor\" aria-label=\"Permalink: Tutorial 6: TISSUE-WPCA (weighted principal component analysis)\" href=\"#tutorial-6-tissue-wpca-weighted-principal-component-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAn alternative approach to TISSUE cell filtering for PCA is TISSUE-WPCA, which involves weighting each value in the predicted gene expression matrix in PCA, thus allowing for a softer approach that can leverage more of the predicted expression data. However, in practice, TISSUE-WPCA generally does not yield as many changes to the resulting principal components than TISSUE cell filtering when compared to normal PCA.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTISSUE-WPCA is highly customizable, mostly in the definition of the weights (see source code for more details and documentation). We use one of the implementations highlighted in the TISSUE manuscript, which involves the inverse TISSUE prediction interval width that is then binarized into a high weight and a low weight that are separated by an order of magnitude:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# weighted PCA\n\ntissue.downstream.weighted_PCA(adata, \u0026quot;spage\u0026quot;, pca_method=\u0026quot;wpca\u0026quot;, weighting=\u0026quot;inverse_pi_width\u0026quot;,\n                               replace_inf=\u0026quot;max\u0026quot;, binarize=0.2, binarize_ratio=10,\n                               n_components=15)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# weighted PCA\u003c/span\u003e\n\n\u003cspan class=\"pl-s1\"\u003etissue\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003edownstream\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eweighted_PCA\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\"spage\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003epca_method\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"wpca\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003eweighting\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"inverse_pi_width\"\u003c/span\u003e,\n                               \u003cspan class=\"pl-s1\"\u003ereplace_inf\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"max\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003ebinarize\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003ebinarize_ratio\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e,\n                               \u003cspan class=\"pl-s1\"\u003en_components\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHere we used the \u003ccode\u003einverse_pi_width\u003c/code\u003e method which uses the inverse prediction interval width as the initial weight. We replace all \u003ccode\u003einf\u003c/code\u003e values with the maximum weights. We specify \u003ccode\u003ebinarize=0.2\u003c/code\u003e, which is the proportion of cells to draw the high/low weight split. The \u003ccode\u003ebinarize_ratio\u003c/code\u003e is the fold-change between the high and low weight values, and we use \u003ccode\u003en_components=15\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNow that we have performed TISSUE-WPCA, we can access the resulting principal components from \u003ccode\u003eadata.obsm['{prediction method name}_predicted_expression_PC15_']\u003c/code\u003e and use these as we would with any other reduced representation of the data. For example, we can visualize the two cell groups along the first two principal components:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# make PC plot\n\nX_pc = adata.obsm['spage_predicted_expression_PC15_']\n\nplt.title(\u0026quot;TISSUE Weighted PCA\u0026quot;)\nplt.scatter(X_pc[adata.obs['condition']=='A',0], X_pc[adata.obs['condition']=='A',1],\n            c=\u0026quot;tab:red\u0026quot;, s=3, label=\u0026quot;A\u0026quot;)\nplt.scatter(X_pc[adata.obs['condition']=='B',0], X_pc[adata.obs['condition']=='B',1],\n            c=\u0026quot;tab:blue\u0026quot;, s=3, label=\u0026quot;B\u0026quot;)\nplt.xlabel(\u0026quot;PC 1\u0026quot;)\nplt.ylabel(\u0026quot;PC 2\u0026quot;)\nplt.legend(loc='center left', bbox_to_anchor=(1, 0.5))\nplt.show()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e# make PC plot\u003c/span\u003e\n\n\u003cspan class=\"pl-v\"\u003eX_pc\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobsm\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'spage_predicted_expression_PC15_'\u003c/span\u003e]\n\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003etitle\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"TISSUE Weighted PCA\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003escatter\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eX_pc\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e]\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'A'\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e], \u003cspan class=\"pl-v\"\u003eX_pc\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e]\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'A'\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\n            \u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"tab:red\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003es\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"A\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003escatter\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eX_pc\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e]\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'B'\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e], \u003cspan class=\"pl-v\"\u003eX_pc\u003c/span\u003e[\u003cspan class=\"pl-s1\"\u003eadata\u003c/span\u003e.\u003cspan class=\"pl-s1\"\u003eobs\u003c/span\u003e[\u003cspan class=\"pl-s\"\u003e'condition'\u003c/span\u003e]\u003cspan class=\"pl-c1\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'B'\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e],\n            \u003cspan class=\"pl-s1\"\u003ec\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"tab:blue\"\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003es\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\"B\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003exlabel\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"PC 1\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eylabel\u003c/span\u003e(\u003cspan class=\"pl-s\"\u003e\"PC 2\"\u003c/span\u003e)\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003elegend\u003c/span\u003e(\u003cspan class=\"pl-s1\"\u003eloc\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e'center left'\u003c/span\u003e, \u003cspan class=\"pl-s1\"\u003ebbox_to_anchor\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e=\u003c/span\u003e(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e))\n\u003cspan class=\"pl-s1\"\u003eplt\u003c/span\u003e.\u003cspan class=\"pl-en\"\u003eshow\u003c/span\u003e()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/sunericd/TISSUE/blob/master/README_files/README_60_0.png\"\u003e\u003cimg src=\"/sunericd/TISSUE/raw/master/README_files/README_60_0.png\" alt=\"png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSimilarly to TISSUE cell filtering PCA, we see that TISSUE-WPCA can visually separate the two cell groups in the PCA plot.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAdditional considerations\u003c/h1\u003e\u003ca id=\"user-content-additional-considerations\" class=\"anchor\" aria-label=\"Permalink: Additional considerations\" href=\"#additional-considerations\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHyperparameter selection:\u003c/h2\u003e\u003ca id=\"user-content-hyperparameter-selection\" class=\"anchor\" aria-label=\"Permalink: Hyperparameter selection:\" href=\"#hyperparameter-selection\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAt various parts of the TISSUE pipeline, the user can select different hyperparameters. Here we outline some guiding principles for reasonably selecting these hyperparameters:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003en_neighbors\u003c/code\u003e in \u003ccode\u003etissue.main.build_spatial_graph()\u003c/code\u003e - the approximate number of neighbors to use for computing TISSUE cell-centric variability. Generally, we recommend setting this to a value close to 15 to ensure reliable cell-centric variability estimates. Values from 5-30 also work comparably well. Alternatively, you can try out other spatial graph methods such as the other options in \u003ccode\u003etissue.main.build_spatial_graph()\u003c/code\u003e or load in your own spatial graph adjacency matrix using \u003ccode\u003etissue.main.load_spatial_graph()\u003c/code\u003e from a .npz file.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003ealpha_level\u003c/code\u003e in \u003ccode\u003etissue.main.conformalize_prediction_interval()\u003c/code\u003e - the confidence measure corresponding to (1-alpha) TISSUE prediction interval coverage. Generally, we recommend 0.23 to retrieve the 67% TISSUE prediction interval (approx. one standard error) but downstream results are largely robust to the exact choice. Values very close to 0 or very close to 1 are less likely to provide informative calibrations.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003ek\u003c/code\u003e, \u003ccode\u003ek2\u003c/code\u003e in \u003ccode\u003etissue.main.conformalize_spatial_uncertainty()\u003c/code\u003e - these are the gene and cell stratified group numbers respectively. You can try different values for each of these, but we recommend staying below 4 for either parameter. In the manuscript, we primarily used \u003ccode\u003ek=4\u003c/code\u003e and \u003ccode\u003ek2=1\u003c/code\u003e. If you don't want to choose, TISSUE can automatically select these hyperparameters if you set \u003ccode\u003ek='auto'\u003c/code\u003e and \u003ccode\u003ek2='auto'\u003c/code\u003e.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eComputational runtime and speed ups:\u003c/h2\u003e\u003ca id=\"user-content-computational-runtime-and-speed-ups\" class=\"anchor\" aria-label=\"Permalink: Computational runtime and speed ups:\" href=\"#computational-runtime-and-speed-ups\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIf you are experiencing slow runtimes with TISSUE, there are several things to check or change for faster runtime (also refer to Extended Data Figure 9 in our publication for runtime breakdown in the first version of TISSUE for different-sized datasets):\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eLarge number of cells\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eWe have tested TISSUE on datasets up to 20K cells. If your data contains substantially more cells, we recommend downsampling the cells (e.g. random uniform sampling) or subsetting to cell groups of interest.\u003c/li\u003e\n\u003cli\u003eIf the prediction step takes a long time, we suggest decreasing the number of cross-validation folds by setting \u003ccode\u003en_folds\u003c/code\u003e in \u003ccode\u003etissue.main.predict_gene_expression()\u003c/code\u003e.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eLarge number of genes\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eGenerally TISSUE is robust to the number of genes but for further speedup, we recommend only predicting genes that are necessary or using the scRNAseq reference dataset to identify a set of highly variable genes for prediction beforehand (if whole-transcriptome is desired, for example using \u003ccode\u003escanpy.pp.highly_variable_genes\u003c/code\u003e).\u003c/li\u003e\n\u003cli\u003eSet \u003ccode\u003eweight='exp_cos_pca'\u003c/code\u003e and \u003ccode\u003eweight_n_pc\u003c/code\u003e to some integer (e.g. 15) in \u003ccode\u003etissue.main.conformalize_spatial_uncertainty()\u003c/code\u003e to perform cosine similarity weights based on low-dimensionality (for better runtime and reduce high-dimensional distortions)\u003c/li\u003e\n\u003cli\u003eSet \u003ccode\u003en_pc\u003c/code\u003e and \u003ccode\u003en_pc2\u003c/code\u003e to some integer (e.g. 15) in \u003ccode\u003etissue.main.conformalize_spatial_uncertainty()\u003c/code\u003e so that k-means clustering will be done on lower dimensional space (for better runtime and performance)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOther runtime tips\u003c/strong\u003e:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eTurn off Wasserstein calculation in \u003ccode\u003etissue.main.conformalize_prediction_interval()\u003c/code\u003e by setting \u003ccode\u003ecompute_wasserstein=False\u003c/code\u003e (default)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMemory usage:\u003c/h2\u003e\u003ca id=\"user-content-memory-usage\" class=\"anchor\" aria-label=\"Permalink: Memory usage:\" href=\"#memory-usage\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe have optimized TISSUE to be memory-efficient with respect the size of the original dataset. Since spatial transcriptomics datasets can be very large, and TISSUE requires additional overhead for some of its operations, here are some suggestions for dealing with memory issues:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFor downsizing datasets with many cells or many genes (to predict), refer to the previous section on runtimes for ways of downsampling for both improved runtime and lower memory usage\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003ccode\u003etissue.main.build_spatial_graph()\u003c/code\u003e, you can consider setting \u003ccode\u003eradius\u003c/code\u003e to a value (default is None) for any of the radius-based methods. This is more important in older versions of TISSUE, which were not optimized.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eIn \u003ccode\u003etissue.downstream.multiple_imputation_testing()\u003c/code\u003e, make sure that \u003ccode\u003esave_mi=False\u003c/code\u003e which stops saving of each multiple imputation into memory.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eUNDER DEVELOPMENT:\u003c/h1\u003e\u003ca id=\"user-content-under-development\" class=\"anchor\" aria-label=\"Permalink: UNDER DEVELOPMENT:\" href=\"#under-development\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eMulti-threading for making cross-validation predictions in \u003ccode\u003etissue.main.predict_gene_expression()\u003c/code\u003e.\u003c/li\u003e\n\u003cli\u003eGene filtering guidelines / strategy\u003c/li\u003e\n\u003cli\u003eSuppress warning printouts\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCitation\u003c/h1\u003e\u003ca id=\"user-content-citation\" class=\"anchor\" aria-label=\"Permalink: Citation\" href=\"#citation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIf you find this code useful, we would appreciate it if you cite the following publications:\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003eSun, E.D., Ma, R., Navarro Negredo, P. et al. TISSUE: uncertainty-calibrated prediction of single-cell spatial transcriptomics improves downstream analyses. Nat Methods (2024). \u003ca href=\"https://doi.org/10.1038/s41592-024-02184-y\" rel=\"nofollow\"\u003ehttps://doi.org/10.1038/s41592-024-02184-y\u003c/a\u003e\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ePreprint:\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSun ED, Ma R, Navarro Negredo P, Brunet A, Zou J. TISSUE: uncertainty-calibrated prediction of single-cell spatial transcriptomics improves downstream analyses. Preprint at \u003ca href=\"https://doi.org/10.1101/2023.04.25.538326\" rel=\"nofollow\"\u003ehttps://doi.org/10.1101/2023.04.25.538326\u003c/a\u003e (2023).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor Jupyter notebooks and Python scripts associated with our original publication, please refer to \u003ca href=\"https://github.com/sunericd/tissue-figures-and-analyses.git\"\u003ehttps://github.com/sunericd/tissue-figures-and-analyses.git\u003c/a\u003e. \u003cstrong\u003eNOTE: For the original publication, we used TISSUE version 0.0.2\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-python notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\"\u003e\u003cpre\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"TISSUE","anchor":"tissue","htmlText":"TISSUE"},{"level":2,"text":"Installation and setup","anchor":"installation-and-setup","htmlText":"Installation and setup"},{"level":3,"text":"Option A: PyPI","anchor":"option-a-pypi","htmlText":"Option A: PyPI"},{"level":3,"text":"Option B: Local installation","anchor":"option-b-local-installation","htmlText":"Option B: Local installation"},{"level":1,"text":"TISSUE Tutorials","anchor":"tissue-tutorials","htmlText":"TISSUE Tutorials"},{"level":2,"text":"Upstream TISSUE analyses: spatial gene expression prediction and uncertainty calibration","anchor":"upstream-tissue-analyses-spatial-gene-expression-prediction-and-uncertainty-calibration","htmlText":"Upstream TISSUE analyses: spatial gene expression prediction and uncertainty calibration"},{"level":3,"text":"Tutorial 1: Predicting spatial gene expression","anchor":"tutorial-1-predicting-spatial-gene-expression","htmlText":"Tutorial 1: Predicting spatial gene expression"},{"level":3,"text":"Tutorial 2: Using TISSUE to calibrate uncertainties and obtain prediction intervals","anchor":"tutorial-2-using-tissue-to-calibrate-uncertainties-and-obtain-prediction-intervals","htmlText":"Tutorial 2: Using TISSUE to calibrate uncertainties and obtain prediction intervals"},{"level":2,"text":"Downstream TISSUE analyses: Hypothesis testing, Clustering/Visualization, Prediction","anchor":"downstream-tissue-analyses-hypothesis-testing-clusteringvisualization-prediction","htmlText":"Downstream TISSUE analyses: Hypothesis testing, Clustering/Visualization, Prediction"},{"level":3,"text":"Tutorial 3: Hypothesis testing with TISSUE multiple imputation framework","anchor":"tutorial-3-hypothesis-testing-with-tissue-multiple-imputation-framework","htmlText":"Tutorial 3: Hypothesis testing with TISSUE multiple imputation framework"},{"level":3,"text":"Tutorial 4: TISSUE cell filtering for supervised learning","anchor":"tutorial-4-tissue-cell-filtering-for-supervised-learning","htmlText":"Tutorial 4: TISSUE cell filtering for supervised learning"},{"level":3,"text":"Tutorial 5: TISSUE cell filtering for PCA (clustering and visualization)","anchor":"tutorial-5-tissue-cell-filtering-for-pca-clustering-and-visualization","htmlText":"Tutorial 5: TISSUE cell filtering for PCA (clustering and visualization)"},{"level":3,"text":"Tutorial 6: TISSUE-WPCA (weighted principal component analysis)","anchor":"tutorial-6-tissue-wpca-weighted-principal-component-analysis","htmlText":"Tutorial 6: TISSUE-WPCA (weighted principal component analysis)"},{"level":1,"text":"Additional considerations","anchor":"additional-considerations","htmlText":"Additional considerations"},{"level":2,"text":"Hyperparameter selection:","anchor":"hyperparameter-selection","htmlText":"Hyperparameter selection:"},{"level":2,"text":"Computational runtime and speed ups:","anchor":"computational-runtime-and-speed-ups","htmlText":"Computational runtime and speed ups:"},{"level":2,"text":"Memory usage:","anchor":"memory-usage","htmlText":"Memory usage:"},{"level":1,"text":"UNDER DEVELOPMENT:","anchor":"under-development","htmlText":"UNDER DEVELOPMENT:"},{"level":1,"text":"Citation","anchor":"citation","htmlText":"Citation"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fsunericd%2FTISSUE"}},{"displayName":"LICENSE","repoName":"TISSUE","refName":"master","path":"LICENSE","preferredFileType":"license","tabName":"MIT","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fsunericd%2FTISSUE"}}],"overviewFilesProcessingTime":35.084298}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
.bhqztV{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.bhqztV span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.bhqztV:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.bhqztV:focus{outline:2px solid transparent;}/*!sc*/
.bhqztV:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.bhqztV:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.bhqztV::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:transparent;border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.bhqztV::after{background-color:transparent;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,bhqztV,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="master branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        master
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/sunericd/TISSUE/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/sunericd/TISSUE/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/sunericd/TISSUE/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/sunericd/TISSUE/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/sunericd/TISSUE/commits/master/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           23 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/sunericd/TISSUE/commits/master/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README_files, (Directory)" class="Link--primary" href="/sunericd/TISSUE/tree/master/README_files" title="README_files">
                          README_files
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README_files, (Directory)" class="Link--primary" href="/sunericd/TISSUE/tree/master/README_files" title="README_files">
                          README_files
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="dist, (Directory)" class="Link--primary" href="/sunericd/TISSUE/tree/master/dist" title="dist">
                          dist
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="dist, (Directory)" class="Link--primary" href="/sunericd/TISSUE/tree/master/dist" title="dist">
                          dist
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tests/data, (Directory)" class="Link--primary" href="/sunericd/TISSUE/tree/master/tests/data" title="This path skips through empty directories">
                          <span class="react-directory-default-color" data-testid="path-name-segment">
                           tests/
                          </span>
                          <span class="" data-testid="path-name-segment">
                           data
                          </span>
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tests/data, (Directory)" class="Link--primary" href="/sunericd/TISSUE/tree/master/tests/data" title="This path skips through empty directories">
                          <span class="react-directory-default-color" data-testid="path-name-segment">
                           tests/
                          </span>
                          <span class="" data-testid="path-name-segment">
                           data
                          </span>
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tissue, (Directory)" class="Link--primary" href="/sunericd/TISSUE/tree/master/tissue" title="tissue">
                          tissue
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tissue, (Directory)" class="Link--primary" href="/sunericd/TISSUE/tree/master/tissue" title="tissue">
                          tissue
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-6">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-7">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="pipeline.png, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/pipeline.png" title="pipeline.png">
                          pipeline.png
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="pipeline.png, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/pipeline.png" title="pipeline.png">
                          pipeline.png
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-8">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="pyproject.toml, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/pyproject.toml" title="pyproject.toml">
                          pyproject.toml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="pyproject.toml, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/pyproject.toml" title="pyproject.toml">
                          pyproject.toml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-9">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="requirements.txt, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/requirements.txt" title="requirements.txt">
                          requirements.txt
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="requirements.txt, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/requirements.txt" title="requirements.txt">
                          requirements.txt
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-10">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="test.py, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/test.py" title="test.py">
                          test.py
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="test.py, (File)" class="Link--primary" href="/sunericd/TISSUE/blob/master/test.py" title="test.py">
                          test.py
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd show-for-mobile" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-law" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="MIT license">
                        MIT license
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      TISSUE
                     </h1>
                     <a aria-label="Permalink: TISSUE" class="anchor" href="#tissue" id="user-content-tissue">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      TISSUE (Transcript Imputation with Spatial Single-cell Uncertainty Estimation)
                     </strong>
                     provides tools for estimating well-calibrated uncertainty measures for gene expression predictions in single-cell spatial transcriptomics datasets and utilizing them in downstream analyses
                    </p>
                    <p dir="auto">
                     <a href="/sunericd/TISSUE/blob/master/pipeline.png" rel="noopener noreferrer" target="_blank">
                      <img alt="plot" src="/sunericd/TISSUE/raw/master/pipeline.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     If you find TISSUE useful, please consider citing our publication:
                    </p>
                    <p dir="auto">
                     <strong>
                      Sun, E.D., Ma, R., Navarro Negredo, P. et al. TISSUE: uncertainty-calibrated prediction of single-cell spatial transcriptomics improves downstream analyses. Nat Methods (2024).
                      <a href="https://doi.org/10.1038/s41592-024-02184-y" rel="nofollow">
                       https://doi.org/10.1038/s41592-024-02184-y
                      </a>
                     </strong>
                    </p>
                    <p dir="auto">
                     <a href="https://doi.org/10.1038/s41592-024-02184-y" rel="nofollow">
                      <img alt="Paper" data-canonical-src="https://img.shields.io/badge/Nature_Methods-DOI-orange" src="https://camo.githubusercontent.com/4597afa544082ce07c8b2860baf130715b74f811a760f5013fb6ea7f8e86e575/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f4e61747572655f4d6574686f64732d444f492d6f72616e6765" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Installation and setup
                     </h2>
                     <a aria-label="Permalink: Installation and setup" class="anchor" href="#installation-and-setup" id="user-content-installation-and-setup">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Complete installation (including of dependencies) of TISSUE in a new Conda environment should take less than 5 minutes on a normal desktop/laptop setup (Windows, Mac OSX, Linux). The base version of TISSUE requires SquidPy and all of its dependencies. The base version comes with the SpaGE spatial gene expression prediction method. Additional TISSUE functionalities (i.e. Tangram prediction, Harmony-kNN prediction, gimVI prediction, TISSUE-WPCA, TISSUE multiple imputation with SpatialDE) require installing additional dependencies, which are listed in
                     <code>
                      requirements.txt
                     </code>
                     . These should be installed with
                     <code>
                      pip install &lt;package name&gt;==&lt;package version&gt;
                     </code>
                     inside the same Conda environment as TISSUE (preferably before TISSUE installation).
                    </p>
                    <p dir="auto">
                     For TISSUE installation, we provide two options: (A) PyPI installation with pip or (B) local installation. We recommend starting with option A and only going to option B if option A fails in your environment.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Option A: PyPI
                     </h3>
                     <a aria-label="Permalink: Option A: PyPI" class="anchor" href="#option-a-pypi" id="user-content-option-a-pypi">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Install the package through PyPI with
                     <code>
                      pip
                     </code>
                     . We recommend setting up a conda environment (
                     <a href="https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html" rel="nofollow">
                      https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html
                     </a>
                     ) or another virtual environment first since
                     <code>
                      tissue-sc
                     </code>
                     currently relies on specific versions for its dependencies (although it should generally work for other environment versions, but this hasn't been thoroughly tested):
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="conda create -n myenv python=3.8
conda activate myenv

&lt;pip install any additional dependencies&gt;
pip install tissue-sc">
                     <pre class="notranslate"><code>conda create -n myenv python=3.8
conda activate myenv

&lt;pip install any additional dependencies&gt;
pip install tissue-sc
</code></pre>
                    </div>
                    <p dir="auto">
                     Note that you will want to separately download the data from this repository (
                     <code>
                      tests/data/
                     </code>
                     ) to run our TISSUE tutorials.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Option B: Local installation
                     </h3>
                     <a aria-label="Permalink: Option B: Local installation" class="anchor" href="#option-b-local-installation" id="user-content-option-b-local-installation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     An alternative way to install the package along with associated test and tutorial files is to clone the directory and then install the requirements for using the package. To do this, first clone the repository using git (you can install git following the instructions
                     <a href="https://github.com/git-guides/install-git">
                      here
                     </a>
                     ):
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="git clone https://github.com/sunericd/TISSUE.git">
                     <pre class="notranslate"><code>git clone https://github.com/sunericd/TISSUE.git
</code></pre>
                    </div>
                    <p dir="auto">
                     We recommend setting up a conda environment to install the requirements for the package (instructions for installing conda and what conda environment can do can be found
                     <a href="https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html" rel="nofollow">
                      here
                     </a>
                     ). Installation of requirements can then be done with the following commands:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="conda create -n tissue python=3.8
conda activate tissue

cd TISSUE
pip install -r requirements.txt">
                     <pre class="notranslate"><code>conda create -n tissue python=3.8
conda activate tissue

cd TISSUE
pip install -r requirements.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     To keep the requirements light, we have only included packages that are necessary for the core functionalities of TISSUE. For additional utilities such as gene prediction with Tangram, please install those packages separately (or uncomment those lines in
                     <code>
                      requirements.txt
                     </code>
                     ).
                    </p>
                    <p dir="auto">
                     To test that the installation is working correctly, you can run
                     <code>
                      python test.py
                     </code>
                     in the cloned directory.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      TISSUE Tutorials
                     </h1>
                     <a aria-label="Permalink: TISSUE Tutorials" class="anchor" href="#tissue-tutorials" id="user-content-tissue-tutorials">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Below we include several mini-tutorials to highlight the main TISSUE pipeline and downstream applications. These tutorials rely on a small test dataset (a subset of one of the datasets used in the original publication) for fast processing but the approaches can readily be extended to other datasets. For larger-scale examples, please refer to the code repository corresponding the figures and analyses generated for the TISSUE manuscript:
                     <a href="https://github.com/sunericd/tissue-figures-and-analyses.git">
                      https://github.com/sunericd/tissue-figures-and-analyses.git
                     </a>
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# import packages

import tissue.main, tissue.downstream

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scanpy as sc
import anndata as ad
import os" dir="auto">
                     <pre><span class="pl-c"># import packages</span>

<span class="pl-k">import</span> <span class="pl-s1">tissue</span>.<span class="pl-s1">main</span>, <span class="pl-s1">tissue</span>.<span class="pl-s1">downstream</span>

<span class="pl-k">import</span> <span class="pl-s1">numpy</span> <span class="pl-k">as</span> <span class="pl-s1">np</span>
<span class="pl-k">import</span> <span class="pl-s1">pandas</span> <span class="pl-k">as</span> <span class="pl-s1">pd</span>
<span class="pl-k">import</span> <span class="pl-s1">matplotlib</span>.<span class="pl-s1">pyplot</span> <span class="pl-k">as</span> <span class="pl-s1">plt</span>
<span class="pl-k">import</span> <span class="pl-s1">scanpy</span> <span class="pl-k">as</span> <span class="pl-s1">sc</span>
<span class="pl-k">import</span> <span class="pl-s1">anndata</span> <span class="pl-k">as</span> <span class="pl-s1">ad</span>
<span class="pl-k">import</span> <span class="pl-s1">os</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Upstream TISSUE analyses: spatial gene expression prediction and uncertainty calibration
                     </h2>
                     <a aria-label="Permalink: Upstream TISSUE analyses: spatial gene expression prediction and uncertainty calibration" class="anchor" href="#upstream-tissue-analyses-spatial-gene-expression-prediction-and-uncertainty-calibration" id="user-content-upstream-tissue-analyses-spatial-gene-expression-prediction-and-uncertainty-calibration">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The first part of the TISSUE pipeline involves making predictions of spatial gene expression profiles using paired spatial transcriptomics and scRNAseq datasets, and then computing and calibrating uncertainties for these predictions that are translated to prediction intervals. All functions for upstream analysis are in the
                     <code>
                      tissue.main
                     </code>
                     module. These include:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <code>
                        tissue.main.load_paired_datasets()
                       </code>
                       for loading paired datasets from formatted directories
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        tissue.main.predict_gene_expression()
                       </code>
                       for predicting spatial gene expression from paired datasets using a specified prediction method and number of cross-validation folds
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        tissue.main.build_spatial_graph()
                       </code>
                       for building a cell-cell spatial graph to define neighborhoods for computing the cell-centric variability
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        tissue.main.conformalize_spatial_uncertainty()
                       </code>
                       for computing cell-centric variability, stratified grouping, and the TISSUE calibration scores for predicted gene expression
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        tissue.main.conformalize_prediction_interval()
                       </code>
                       for building prediction intervals from uncertainty measures
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Tutorial 1: Predicting spatial gene expression
                     </h3>
                     <a aria-label="Permalink: Tutorial 1: Predicting spatial gene expression" class="anchor" href="#tutorial-1-predicting-spatial-gene-expression" id="user-content-tutorial-1-predicting-spatial-gene-expression">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     First, we load a minimal subset of osmFISH spatial transcriptomics data of mouse somatosensory cortex published by Codeluppi et al., 2018:
                     <a href="https://doi.org/10.1038/s41592-018-0175-z" rel="nofollow">
                      https://doi.org/10.1038/s41592-018-0175-z
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     Note that we are using the TISSUE methods for building an AnnData object from tab-delimited text files for the spatial counts, scRNAseq counts, spatial locations, and spatial metadata. If you already have an AnnData object for the spatial data and another object for the scRNAseq data, you can skip this step.
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# load in spatial and scRNAseq datasets

adata, RNAseq_adata = tissue.main.load_paired_datasets("tests/data/Spatial_count.txt",
                                                       "tests/data/Locations.txt",
                                                       "tests/data/scRNA_count.txt")' dir="auto">
                     <pre><span class="pl-c"># load in spatial and scRNAseq datasets</span>

<span class="pl-s1">adata</span>, <span class="pl-v">RNAseq_adata</span> <span class="pl-c1">=</span> <span class="pl-s1">tissue</span>.<span class="pl-s1">main</span>.<span class="pl-en">load_paired_datasets</span>(<span class="pl-s">"tests/data/Spatial_count.txt"</span>,
                                                       <span class="pl-s">"tests/data/Locations.txt"</span>,
                                                       <span class="pl-s">"tests/data/scRNA_count.txt"</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:117: ImplicitModificationWarning: Transforming to str index.
  warnings.warn("Transforming to str index.", ImplicitModificationWarning)
/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:856: UserWarning: 
AnnData expects .obs.index to contain strings, but got values like:
    [0, 1, 2, 3, 4]

    Inferred to be: integer

  names = self._prep_dim_index(names, "obs")'>
                     <pre class="notranslate"><code>/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:117: ImplicitModificationWarning: Transforming to str index.
  warnings.warn("Transforming to str index.", ImplicitModificationWarning)
/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:856: UserWarning: 
AnnData expects .obs.index to contain strings, but got values like:
    [0, 1, 2, 3, 4]

    Inferred to be: integer

  names = self._prep_dim_index(names, "obs")
</code></pre>
                    </div>
                    <p dir="auto">
                     Now we can impute any genes of interest that are found in the scRNAseq dataset but not in the spatial dataset. In this case, we will hold out a target gene from the spatial data and apply an imputation method to predict its expression using the scRNAseq dataset.
                    </p>
                    <p dir="auto">
                     First, we preprocess the data and make sure that the gene names are matchable across the two datasets:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# make genes lowercase
adata.var_names = [x.lower() for x in adata.var_names]
RNAseq_adata.var_names = [x.lower() for x in RNAseq_adata.var_names]

# preprocess RNAseq data
tissue.main.preprocess_data(RNAseq_adata, standardize=False, normalize=True)

# subset spatial data into shared genes
gene_names = np.intersect1d(adata.var_names, RNAseq_adata.var_names)
adata = adata[:, gene_names].copy()

# hold out target gene
target_gene = "plp1"
target_expn = adata[:, target_gene].X.copy()
adata = adata[:, [gene for gene in gene_names if gene != target_gene]].copy()' dir="auto">
                     <pre><span class="pl-c"># make genes lowercase</span>
<span class="pl-s1">adata</span>.<span class="pl-s1">var_names</span> <span class="pl-c1">=</span> [<span class="pl-s1">x</span>.<span class="pl-en">lower</span>() <span class="pl-k">for</span> <span class="pl-s1">x</span> <span class="pl-c1">in</span> <span class="pl-s1">adata</span>.<span class="pl-s1">var_names</span>]
<span class="pl-v">RNAseq_adata</span>.<span class="pl-s1">var_names</span> <span class="pl-c1">=</span> [<span class="pl-s1">x</span>.<span class="pl-en">lower</span>() <span class="pl-k">for</span> <span class="pl-s1">x</span> <span class="pl-c1">in</span> <span class="pl-v">RNAseq_adata</span>.<span class="pl-s1">var_names</span>]

<span class="pl-c"># preprocess RNAseq data</span>
<span class="pl-s1">tissue</span>.<span class="pl-s1">main</span>.<span class="pl-en">preprocess_data</span>(<span class="pl-v">RNAseq_adata</span>, <span class="pl-s1">standardize</span><span class="pl-c1">=</span><span class="pl-c1">False</span>, <span class="pl-s1">normalize</span><span class="pl-c1">=</span><span class="pl-c1">True</span>)

<span class="pl-c"># subset spatial data into shared genes</span>
<span class="pl-s1">gene_names</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">intersect1d</span>(<span class="pl-s1">adata</span>.<span class="pl-s1">var_names</span>, <span class="pl-v">RNAseq_adata</span>.<span class="pl-s1">var_names</span>)
<span class="pl-s1">adata</span> <span class="pl-c1">=</span> <span class="pl-s1">adata</span>[:, <span class="pl-s1">gene_names</span>].<span class="pl-en">copy</span>()

<span class="pl-c"># hold out target gene</span>
<span class="pl-s1">target_gene</span> <span class="pl-c1">=</span> <span class="pl-s">"plp1"</span>
<span class="pl-s1">target_expn</span> <span class="pl-c1">=</span> <span class="pl-s1">adata</span>[:, <span class="pl-s1">target_gene</span>].<span class="pl-v">X</span>.<span class="pl-en">copy</span>()
<span class="pl-s1">adata</span> <span class="pl-c1">=</span> <span class="pl-s1">adata</span>[:, [<span class="pl-s1">gene</span> <span class="pl-k">for</span> <span class="pl-s1">gene</span> <span class="pl-c1">in</span> <span class="pl-s1">gene_names</span> <span class="pl-k">if</span> <span class="pl-s1">gene</span> <span class="pl-c1">!=</span> <span class="pl-s1">target_gene</span>]].<span class="pl-en">copy</span>()</pre>
                    </div>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# dimensions of spatial transcriptomics dataset (number of cells x number of genes)
adata.shape" dir="auto">
                     <pre><span class="pl-c"># dimensions of spatial transcriptomics dataset (number of cells x number of genes)</span>
<span class="pl-s1">adata</span>.<span class="pl-s1">shape</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="(3405, 31)">
                     <pre class="notranslate"><code>(3405, 31)
</code></pre>
                    </div>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# dimensions of RNAseq dataset (number of cells x number of genes)
RNAseq_adata.shape" dir="auto">
                     <pre><span class="pl-c"># dimensions of RNAseq dataset (number of cells x number of genes)</span>
<span class="pl-v">RNAseq_adata</span>.<span class="pl-s1">shape</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="(1000, 32)">
                     <pre class="notranslate"><code>(1000, 32)
</code></pre>
                    </div>
                    <p dir="auto">
                     Now, we can make predictions of the target gene expression. In TISSUE, we currently have several methods for gene imputation including SpaGE, Tangram, and Harmony-kNN. We will be using SpaGE in this example:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# SpaGE spatial gene expression prediction

tissue.main.predict_gene_expression (adata, RNAseq_adata, [target_gene],
                                     method="spage", n_folds=10, n_pv=10)' dir="auto">
                     <pre><span class="pl-c"># SpaGE spatial gene expression prediction</span>

<span class="pl-s1">tissue</span>.<span class="pl-s1">main</span>.<span class="pl-en">predict_gene_expression</span> (<span class="pl-s1">adata</span>, <span class="pl-v">RNAseq_adata</span>, [<span class="pl-s1">target_gene</span>],
                                     <span class="pl-s1">method</span><span class="pl-c1">=</span><span class="pl-s">"spage"</span>, <span class="pl-s1">n_folds</span><span class="pl-c1">=</span><span class="pl-c1">10</span>, <span class="pl-s1">n_pv</span><span class="pl-c1">=</span><span class="pl-c1">10</span>)</pre>
                    </div>
                    <p dir="auto">
                     How good is the imputation? Since we left out this gene from the spatial data, we can plot the predicted and actual expression and visually inspect the agreement.
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Visualizing predicted and actual expression side by side

fig, (ax1, ax2) = plt.subplots(1, 2)

ax1.axis('off')
cmap = target_expn
cmap[cmap&lt;0] = 0
cmap = np.log1p(cmap)
cmap[cmap &gt; np.percentile(cmap,95)] = np.percentile(cmap,95)
im = ax1.scatter(adata.obsm['spatial'][:,0],adata.obsm['spatial'][:,1],s=1,c=cmap,rasterized=True)
ax1.set_title('Actual', fontsize = 12)

cbar = fig.colorbar(im)
cbar.ax.get_yaxis().labelpad = 15
cbar.ax.set_ylabel('Log Expression', rotation=270)

ax2.axis('off')
cmap = adata.obsm['spage_predicted_expression'][target_gene].values
cmap[cmap&lt;0] = 0
cmap = np.log1p(cmap)
cmap[cmap &gt; np.percentile(cmap,95)] = np.percentile(cmap,95)
im = ax2.scatter(adata.obsm['spatial'][:,0],adata.obsm['spatial'][:,1],s=1,c=cmap,rasterized=True)
ax2.set_title('Predicted', fontsize = 12)

cbar = fig.colorbar(im)
cbar.ax.get_yaxis().labelpad = 15
cbar.ax.set_ylabel('Log Expression', rotation=270)

plt.suptitle(&quot;SpaGE Prediction&quot;, fontsize=16)
plt.tight_layout()
plt.show()" dir="auto">
                     <pre><span class="pl-c"># Visualizing predicted and actual expression side by side</span>

<span class="pl-s1">fig</span>, (<span class="pl-s1">ax1</span>, <span class="pl-s1">ax2</span>) <span class="pl-c1">=</span> <span class="pl-s1">plt</span>.<span class="pl-en">subplots</span>(<span class="pl-c1">1</span>, <span class="pl-c1">2</span>)

<span class="pl-s1">ax1</span>.<span class="pl-en">axis</span>(<span class="pl-s">'off'</span>)
<span class="pl-s1">cmap</span> <span class="pl-c1">=</span> <span class="pl-s1">target_expn</span>
<span class="pl-s1">cmap</span>[<span class="pl-s1">cmap</span><span class="pl-c1">&lt;</span><span class="pl-c1">0</span>] <span class="pl-c1">=</span> <span class="pl-c1">0</span>
<span class="pl-s1">cmap</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">log1p</span>(<span class="pl-s1">cmap</span>)
<span class="pl-s1">cmap</span>[<span class="pl-s1">cmap</span> <span class="pl-c1">&gt;</span> <span class="pl-s1">np</span>.<span class="pl-en">percentile</span>(<span class="pl-s1">cmap</span>,<span class="pl-c1">95</span>)] <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">percentile</span>(<span class="pl-s1">cmap</span>,<span class="pl-c1">95</span>)
<span class="pl-s1">im</span> <span class="pl-c1">=</span> <span class="pl-s1">ax1</span>.<span class="pl-en">scatter</span>(<span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">0</span>],<span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">1</span>],<span class="pl-s1">s</span><span class="pl-c1">=</span><span class="pl-c1">1</span>,<span class="pl-s1">c</span><span class="pl-c1">=</span><span class="pl-s1">cmap</span>,<span class="pl-s1">rasterized</span><span class="pl-c1">=</span><span class="pl-c1">True</span>)
<span class="pl-s1">ax1</span>.<span class="pl-en">set_title</span>(<span class="pl-s">'Actual'</span>, <span class="pl-s1">fontsize</span> <span class="pl-c1">=</span> <span class="pl-c1">12</span>)

<span class="pl-s1">cbar</span> <span class="pl-c1">=</span> <span class="pl-s1">fig</span>.<span class="pl-en">colorbar</span>(<span class="pl-s1">im</span>)
<span class="pl-s1">cbar</span>.<span class="pl-s1">ax</span>.<span class="pl-en">get_yaxis</span>().<span class="pl-s1">labelpad</span> <span class="pl-c1">=</span> <span class="pl-c1">15</span>
<span class="pl-s1">cbar</span>.<span class="pl-s1">ax</span>.<span class="pl-en">set_ylabel</span>(<span class="pl-s">'Log Expression'</span>, <span class="pl-s1">rotation</span><span class="pl-c1">=</span><span class="pl-c1">270</span>)

<span class="pl-s1">ax2</span>.<span class="pl-en">axis</span>(<span class="pl-s">'off'</span>)
<span class="pl-s1">cmap</span> <span class="pl-c1">=</span> <span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">'spage_predicted_expression'</span>][<span class="pl-s1">target_gene</span>].<span class="pl-s1">values</span>
<span class="pl-s1">cmap</span>[<span class="pl-s1">cmap</span><span class="pl-c1">&lt;</span><span class="pl-c1">0</span>] <span class="pl-c1">=</span> <span class="pl-c1">0</span>
<span class="pl-s1">cmap</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">log1p</span>(<span class="pl-s1">cmap</span>)
<span class="pl-s1">cmap</span>[<span class="pl-s1">cmap</span> <span class="pl-c1">&gt;</span> <span class="pl-s1">np</span>.<span class="pl-en">percentile</span>(<span class="pl-s1">cmap</span>,<span class="pl-c1">95</span>)] <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">percentile</span>(<span class="pl-s1">cmap</span>,<span class="pl-c1">95</span>)
<span class="pl-s1">im</span> <span class="pl-c1">=</span> <span class="pl-s1">ax2</span>.<span class="pl-en">scatter</span>(<span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">0</span>],<span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">1</span>],<span class="pl-s1">s</span><span class="pl-c1">=</span><span class="pl-c1">1</span>,<span class="pl-s1">c</span><span class="pl-c1">=</span><span class="pl-s1">cmap</span>,<span class="pl-s1">rasterized</span><span class="pl-c1">=</span><span class="pl-c1">True</span>)
<span class="pl-s1">ax2</span>.<span class="pl-en">set_title</span>(<span class="pl-s">'Predicted'</span>, <span class="pl-s1">fontsize</span> <span class="pl-c1">=</span> <span class="pl-c1">12</span>)

<span class="pl-s1">cbar</span> <span class="pl-c1">=</span> <span class="pl-s1">fig</span>.<span class="pl-en">colorbar</span>(<span class="pl-s1">im</span>)
<span class="pl-s1">cbar</span>.<span class="pl-s1">ax</span>.<span class="pl-en">get_yaxis</span>().<span class="pl-s1">labelpad</span> <span class="pl-c1">=</span> <span class="pl-c1">15</span>
<span class="pl-s1">cbar</span>.<span class="pl-s1">ax</span>.<span class="pl-en">set_ylabel</span>(<span class="pl-s">'Log Expression'</span>, <span class="pl-s1">rotation</span><span class="pl-c1">=</span><span class="pl-c1">270</span>)

<span class="pl-s1">plt</span>.<span class="pl-en">suptitle</span>(<span class="pl-s">"SpaGE Prediction"</span>, <span class="pl-s1">fontsize</span><span class="pl-c1">=</span><span class="pl-c1">16</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">tight_layout</span>()
<span class="pl-s1">plt</span>.<span class="pl-en">show</span>()</pre>
                    </div>
                    <p dir="auto">
                     <a href="/sunericd/TISSUE/blob/master/README_files/README_13_0.png" rel="noopener noreferrer" target="_blank">
                      <img alt="png" src="/sunericd/TISSUE/raw/master/README_files/README_13_0.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Not too bad, especially considering that we used a downsampled scRNAseq dataset for this imputation.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Tutorial 2: Using TISSUE to calibrate uncertainties and obtain prediction intervals
                     </h3>
                     <a aria-label="Permalink: Tutorial 2: Using TISSUE to calibrate uncertainties and obtain prediction intervals" class="anchor" href="#tutorial-2-using-tissue-to-calibrate-uncertainties-and-obtain-prediction-intervals" id="user-content-tutorial-2-using-tissue-to-calibrate-uncertainties-and-obtain-prediction-intervals">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Note that when we ran
                     <code>
                      tissue.main.predict_gene_expression()
                     </code>
                     in the previous tutorial, we obtained cross-validated predictions for all genes in the existing spatial dataset (e.g. 10 folds). These will now come in handy when we compute and calibrate TISSUE uncertainties for the predicted gene expression.
                    </p>
                    <p dir="auto">
                     First, we build spatial graphs using TISSUE (this can also be done with native Scanpy functions):
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# build spatial graph and calculate adjacency weights

tissue.main.build_spatial_graph(adata, method="fixed_radius", n_neighbors=15)' dir="auto">
                     <pre><span class="pl-c"># build spatial graph and calculate adjacency weights</span>

<span class="pl-s1">tissue</span>.<span class="pl-s1">main</span>.<span class="pl-en">build_spatial_graph</span>(<span class="pl-s1">adata</span>, <span class="pl-s1">method</span><span class="pl-c1">=</span><span class="pl-s">"fixed_radius"</span>, <span class="pl-s1">n_neighbors</span><span class="pl-c1">=</span><span class="pl-c1">15</span>)</pre>
                    </div>
                    <p dir="auto">
                     The entire TISSUE spatial uncertainty generation pipeline can be launched with one line of code using
                     <code>
                      tissue.main.conformalize_spatial_uncertainty()
                     </code>
                     . Here, the first two arguments are the spatial AnnData object and a string specifier for the key in
                     <code>
                      obsm
                     </code>
                     corresponding to the predicted gene expression. We can specify which of the genes to use in the calibration set (generally this is all genes in the spatial data). There are other arguments for the grouping setup and weighting schemes, but we will use the default settings in this tutorial.
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# build calibration scores

tissue.main.conformalize_spatial_uncertainty(adata, "spage_predicted_expression", calib_genes=adata.var_names,
                                             grouping_method="kmeans_gene_cell", k=4, k2=2)' dir="auto">
                     <pre><span class="pl-c"># build calibration scores</span>

<span class="pl-s1">tissue</span>.<span class="pl-s1">main</span>.<span class="pl-en">conformalize_spatial_uncertainty</span>(<span class="pl-s1">adata</span>, <span class="pl-s">"spage_predicted_expression"</span>, <span class="pl-s1">calib_genes</span><span class="pl-c1">=</span><span class="pl-s1">adata</span>.<span class="pl-s1">var_names</span>,
                                             <span class="pl-s1">grouping_method</span><span class="pl-c1">=</span><span class="pl-s">"kmeans_gene_cell"</span>, <span class="pl-s1">k</span><span class="pl-c1">=</span><span class="pl-c1">4</span>, <span class="pl-s1">k2</span><span class="pl-c1">=</span><span class="pl-c1">2</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(
/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(
/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(
/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(
/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(">
                     <pre class="notranslate"><code>/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(
/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(
/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(
/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(
/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(
</code></pre>
                    </div>
                    <p dir="auto">
                     Finally, we can calibrate the spatial uncertainty measures to get calibration scores and then use those to extract prediction intervals for any confidence level
                     <math-renderer class="js-inline-math" data-run-id="6c11f29f82daf890c76b434fc037a46d" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $1-\alpha$
                     </math-renderer>
                     . This is done with
                     <code>
                      tissue.main.conformalize_prediction_interval()
                     </code>
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# get prediction interval for 67% coverage

tissue.main.conformalize_prediction_interval (adata, "spage_predicted_expression", calib_genes=adata.var_names,
                                              alpha_level=0.23)

# set `compute_wasserstein=True` to compute a measure indicating the distance between the cell-centric variability values 
# of the predicted genes and the values of their support (i.e. calibration group) in the original data. 
# The lower this value, the better supported the predicted group.' dir="auto">
                     <pre><span class="pl-c"># get prediction interval for 67% coverage</span>

<span class="pl-s1">tissue</span>.<span class="pl-s1">main</span>.<span class="pl-en">conformalize_prediction_interval</span> (<span class="pl-s1">adata</span>, <span class="pl-s">"spage_predicted_expression"</span>, <span class="pl-s1">calib_genes</span><span class="pl-c1">=</span><span class="pl-s1">adata</span>.<span class="pl-s1">var_names</span>,
                                              <span class="pl-s1">alpha_level</span><span class="pl-c1">=</span><span class="pl-c1">0.23</span>)

<span class="pl-c"># set `compute_wasserstein=True` to compute a measure indicating the distance between the cell-centric variability values </span>
<span class="pl-c"># of the predicted genes and the values of their support (i.e. calibration group) in the original data. </span>
<span class="pl-c"># The lower this value, the better supported the predicted group.</span></pre>
                    </div>
                    <p dir="auto">
                     Now let's visualize what these prediction intervals look like for the target (unseen) gene and how it compares to the actual prediction errors.
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="m = &quot;spage&quot;

i = np.where(adata.var_names == target_gene)[0]

# define consistent color map
all_vals = np.concatenate((np.abs(target_expn.flatten()-adata.obsm[m+&quot;_predicted_expression&quot;][target_gene].values),
                           adata.obsm[m+f&quot;_predicted_expression_hi&quot;][target_gene].values-adata.obsm[m+f&quot;_predicted_expression_lo&quot;][target_gene].values))
all_vals[all_vals&lt;0]=0
vmin = np.percentile(np.log1p(all_vals), 0)
vmax = np.percentile(np.log1p(all_vals), 95)


fig, (ax1, ax2) = plt.subplots(1, 2)

ax1.axis('off')
cmap = np.abs(target_expn.flatten()-adata.obsm[m+&quot;_predicted_expression&quot;][target_gene].values)
cmap[cmap&lt;0] = 0
cmap = np.log1p(cmap)
cmap[cmap &gt; np.percentile(cmap,95)] = np.percentile(cmap,95)
im = ax1.scatter(adata.obsm['spatial'][:,0],adata.obsm['spatial'][:,1],s=1,c=cmap,rasterized=True)#,vmin=vmin,vmax=vmax)
ax1.set_title('Imputation Error ' + target_gene, fontsize = 12)

cbar = fig.colorbar(im)
cbar.ax.get_yaxis().labelpad = 15
cbar.ax.set_ylabel('Log Expression', rotation=270)

ax2.axis('off')
cmap = adata.obsm[m+f&quot;_predicted_expression_hi&quot;][target_gene].values-adata.obsm[m+f&quot;_predicted_expression_lo&quot;][target_gene].values
cmap[cmap&lt;0] = 0
cmap = np.log1p(cmap)
cmap[cmap &gt; np.percentile(cmap,95)] = np.percentile(cmap,95)
im = ax2.scatter(adata.obsm['spatial'][:,0],adata.obsm['spatial'][:,1],s=1,c=cmap,rasterized=True)#,vmin=vmin,vmax=vmax)
ax2.set_title('PI Width ' + target_gene, fontsize = 12)

cbar = fig.colorbar(im)
cbar.ax.get_yaxis().labelpad = 15
cbar.ax.set_ylabel('Log Expression', rotation=270)

plt.suptitle(m, fontsize=16)
plt.tight_layout()
plt.show()" dir="auto">
                     <pre><span class="pl-s1">m</span> <span class="pl-c1">=</span> <span class="pl-s">"spage"</span>

<span class="pl-s1">i</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">where</span>(<span class="pl-s1">adata</span>.<span class="pl-s1">var_names</span> <span class="pl-c1">==</span> <span class="pl-s1">target_gene</span>)[<span class="pl-c1">0</span>]

<span class="pl-c"># define consistent color map</span>
<span class="pl-s1">all_vals</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">concatenate</span>((<span class="pl-s1">np</span>.<span class="pl-en">abs</span>(<span class="pl-s1">target_expn</span>.<span class="pl-en">flatten</span>()<span class="pl-c1">-</span><span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s1">m</span><span class="pl-c1">+</span><span class="pl-s">"_predicted_expression"</span>][<span class="pl-s1">target_gene</span>].<span class="pl-s1">values</span>),
                           <span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s1">m</span><span class="pl-c1">+</span><span class="pl-s">f"_predicted_expression_hi"</span>][<span class="pl-s1">target_gene</span>].<span class="pl-s1">values</span><span class="pl-c1">-</span><span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s1">m</span><span class="pl-c1">+</span><span class="pl-s">f"_predicted_expression_lo"</span>][<span class="pl-s1">target_gene</span>].<span class="pl-s1">values</span>))
<span class="pl-s1">all_vals</span>[<span class="pl-s1">all_vals</span><span class="pl-c1">&lt;</span><span class="pl-c1">0</span>]<span class="pl-c1">=</span><span class="pl-c1">0</span>
<span class="pl-s1">vmin</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">percentile</span>(<span class="pl-s1">np</span>.<span class="pl-en">log1p</span>(<span class="pl-s1">all_vals</span>), <span class="pl-c1">0</span>)
<span class="pl-s1">vmax</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">percentile</span>(<span class="pl-s1">np</span>.<span class="pl-en">log1p</span>(<span class="pl-s1">all_vals</span>), <span class="pl-c1">95</span>)


<span class="pl-s1">fig</span>, (<span class="pl-s1">ax1</span>, <span class="pl-s1">ax2</span>) <span class="pl-c1">=</span> <span class="pl-s1">plt</span>.<span class="pl-en">subplots</span>(<span class="pl-c1">1</span>, <span class="pl-c1">2</span>)

<span class="pl-s1">ax1</span>.<span class="pl-en">axis</span>(<span class="pl-s">'off'</span>)
<span class="pl-s1">cmap</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">abs</span>(<span class="pl-s1">target_expn</span>.<span class="pl-en">flatten</span>()<span class="pl-c1">-</span><span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s1">m</span><span class="pl-c1">+</span><span class="pl-s">"_predicted_expression"</span>][<span class="pl-s1">target_gene</span>].<span class="pl-s1">values</span>)
<span class="pl-s1">cmap</span>[<span class="pl-s1">cmap</span><span class="pl-c1">&lt;</span><span class="pl-c1">0</span>] <span class="pl-c1">=</span> <span class="pl-c1">0</span>
<span class="pl-s1">cmap</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">log1p</span>(<span class="pl-s1">cmap</span>)
<span class="pl-s1">cmap</span>[<span class="pl-s1">cmap</span> <span class="pl-c1">&gt;</span> <span class="pl-s1">np</span>.<span class="pl-en">percentile</span>(<span class="pl-s1">cmap</span>,<span class="pl-c1">95</span>)] <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">percentile</span>(<span class="pl-s1">cmap</span>,<span class="pl-c1">95</span>)
<span class="pl-s1">im</span> <span class="pl-c1">=</span> <span class="pl-s1">ax1</span>.<span class="pl-en">scatter</span>(<span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">0</span>],<span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">1</span>],<span class="pl-s1">s</span><span class="pl-c1">=</span><span class="pl-c1">1</span>,<span class="pl-s1">c</span><span class="pl-c1">=</span><span class="pl-s1">cmap</span>,<span class="pl-s1">rasterized</span><span class="pl-c1">=</span><span class="pl-c1">True</span>)<span class="pl-c">#,vmin=vmin,vmax=vmax)</span>
<span class="pl-s1">ax1</span>.<span class="pl-en">set_title</span>(<span class="pl-s">'Imputation Error '</span> <span class="pl-c1">+</span> <span class="pl-s1">target_gene</span>, <span class="pl-s1">fontsize</span> <span class="pl-c1">=</span> <span class="pl-c1">12</span>)

<span class="pl-s1">cbar</span> <span class="pl-c1">=</span> <span class="pl-s1">fig</span>.<span class="pl-en">colorbar</span>(<span class="pl-s1">im</span>)
<span class="pl-s1">cbar</span>.<span class="pl-s1">ax</span>.<span class="pl-en">get_yaxis</span>().<span class="pl-s1">labelpad</span> <span class="pl-c1">=</span> <span class="pl-c1">15</span>
<span class="pl-s1">cbar</span>.<span class="pl-s1">ax</span>.<span class="pl-en">set_ylabel</span>(<span class="pl-s">'Log Expression'</span>, <span class="pl-s1">rotation</span><span class="pl-c1">=</span><span class="pl-c1">270</span>)

<span class="pl-s1">ax2</span>.<span class="pl-en">axis</span>(<span class="pl-s">'off'</span>)
<span class="pl-s1">cmap</span> <span class="pl-c1">=</span> <span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s1">m</span><span class="pl-c1">+</span><span class="pl-s">f"_predicted_expression_hi"</span>][<span class="pl-s1">target_gene</span>].<span class="pl-s1">values</span><span class="pl-c1">-</span><span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s1">m</span><span class="pl-c1">+</span><span class="pl-s">f"_predicted_expression_lo"</span>][<span class="pl-s1">target_gene</span>].<span class="pl-s1">values</span>
<span class="pl-s1">cmap</span>[<span class="pl-s1">cmap</span><span class="pl-c1">&lt;</span><span class="pl-c1">0</span>] <span class="pl-c1">=</span> <span class="pl-c1">0</span>
<span class="pl-s1">cmap</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">log1p</span>(<span class="pl-s1">cmap</span>)
<span class="pl-s1">cmap</span>[<span class="pl-s1">cmap</span> <span class="pl-c1">&gt;</span> <span class="pl-s1">np</span>.<span class="pl-en">percentile</span>(<span class="pl-s1">cmap</span>,<span class="pl-c1">95</span>)] <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">percentile</span>(<span class="pl-s1">cmap</span>,<span class="pl-c1">95</span>)
<span class="pl-s1">im</span> <span class="pl-c1">=</span> <span class="pl-s1">ax2</span>.<span class="pl-en">scatter</span>(<span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">0</span>],<span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">1</span>],<span class="pl-s1">s</span><span class="pl-c1">=</span><span class="pl-c1">1</span>,<span class="pl-s1">c</span><span class="pl-c1">=</span><span class="pl-s1">cmap</span>,<span class="pl-s1">rasterized</span><span class="pl-c1">=</span><span class="pl-c1">True</span>)<span class="pl-c">#,vmin=vmin,vmax=vmax)</span>
<span class="pl-s1">ax2</span>.<span class="pl-en">set_title</span>(<span class="pl-s">'PI Width '</span> <span class="pl-c1">+</span> <span class="pl-s1">target_gene</span>, <span class="pl-s1">fontsize</span> <span class="pl-c1">=</span> <span class="pl-c1">12</span>)

<span class="pl-s1">cbar</span> <span class="pl-c1">=</span> <span class="pl-s1">fig</span>.<span class="pl-en">colorbar</span>(<span class="pl-s1">im</span>)
<span class="pl-s1">cbar</span>.<span class="pl-s1">ax</span>.<span class="pl-en">get_yaxis</span>().<span class="pl-s1">labelpad</span> <span class="pl-c1">=</span> <span class="pl-c1">15</span>
<span class="pl-s1">cbar</span>.<span class="pl-s1">ax</span>.<span class="pl-en">set_ylabel</span>(<span class="pl-s">'Log Expression'</span>, <span class="pl-s1">rotation</span><span class="pl-c1">=</span><span class="pl-c1">270</span>)

<span class="pl-s1">plt</span>.<span class="pl-en">suptitle</span>(<span class="pl-s1">m</span>, <span class="pl-s1">fontsize</span><span class="pl-c1">=</span><span class="pl-c1">16</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">tight_layout</span>()
<span class="pl-s1">plt</span>.<span class="pl-en">show</span>()</pre>
                    </div>
                    <p dir="auto">
                     <a href="/sunericd/TISSUE/blob/master/README_files/README_22_0.png" rel="noopener noreferrer" target="_blank">
                      <img alt="png" src="/sunericd/TISSUE/raw/master/README_files/README_22_0.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     The TISSUE prediction intervals are decent and match the distribution of imputation errors especially well in the bottom portions of the section. On the full dataset, which has much richer reference scRNAseq data, the calibration quality is further improved (see Figure 2 of the TISSUE manuscript).
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Downstream TISSUE analyses: Hypothesis testing, Clustering/Visualization, Prediction
                     </h2>
                     <a aria-label="Permalink: Downstream TISSUE analyses: Hypothesis testing, Clustering/Visualization, Prediction" class="anchor" href="#downstream-tissue-analyses-hypothesis-testing-clusteringvisualization-prediction" id="user-content-downstream-tissue-analyses-hypothesis-testing-clusteringvisualization-prediction">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     TISSUE provides additional functionalities for leveraging these uncertainty estimates and prediction intervals in common downstream single-cell spatial transcriptomics analysese. All functions for downstream analysis are in the
                     <code>
                      tissue.downstream
                     </code>
                     module. These include:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <code>
                        tissue.downstream.multiple_imputation_testing()
                       </code>
                       for hypothesis testing using multiple imputations drawn from the calibration score sets
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        tissue.downstream.weighted_PCA()
                       </code>
                       for computing weighted principal components where weights correspond to a transform of the inverse prediction interval width
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        tissue.downstream.detect_uncertain_cells()
                       </code>
                       for filtering low-confidence cells from data before training and evaluation of machine learning models, which generally improves performance
                      </p>
                     </li>
                    </ul>
                    <p dir="auto">
                     Below we include a few example uses of these modules. For more examples (including those for the experiments in the associated manuscript), please see the Github repository:
                     <a href="https://github.com/sunericd/tissue-figures-and-analyses.git">
                      https://github.com/sunericd/tissue-figures-and-analyses.git
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Tutorial 3: Hypothesis testing with TISSUE multiple imputation framework
                     </h3>
                     <a aria-label="Permalink: Tutorial 3: Hypothesis testing with TISSUE multiple imputation framework" class="anchor" href="#tutorial-3-hypothesis-testing-with-tissue-multiple-imputation-framework" id="user-content-tutorial-3-hypothesis-testing-with-tissue-multiple-imputation-framework">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     This tutorial uses the TISSUE calibration scores to generate multiple imputations and then perform hypothesis testing by aggregating statistics across these multiple imputations. The default and recommended statistical test for this framework is the Student's t-test, but other options for spatially variable gene detection using SpatialDE and non-parametric (one-sided) Mann-Whitney/Wilcoxon tests can also be performed by specifying
                     <code>
                      method="spatialde"
                     </code>
                     ,
                     <code>
                      method="wilcoxon_greater"
                     </code>
                     ,
                     <code>
                      method="wilcoxon_less"
                     </code>
                     for
                     <code>
                      tissue.downstream.multiple_imputation_testing()
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     Please run the code in Tutorials 1-2 to generate predictions and TISSUE uncertainty measures before this tutorial.
                    </p>
                    <p dir="auto">
                     After, we construct some binary labels for the cells in the dataset:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# split into two groups based on indices
adata.obs['condition'] = ['A' if i &lt; round(adata.shape[0]/2) else 'B' for i in range(adata.shape[0])]

# plot conditions
plt.scatter(adata[adata.obs.condition==&quot;A&quot;].obsm['spatial'][:,0],
            adata[adata.obs.condition==&quot;A&quot;].obsm['spatial'][:,1],
            c='tab:red', s=3, label=&quot;A&quot;)
plt.scatter(adata[adata.obs.condition==&quot;B&quot;].obsm['spatial'][:,0],
            adata[adata.obs.condition==&quot;B&quot;].obsm['spatial'][:,1],
            c='tab:blue', s=3, label=&quot;B&quot;)
plt.legend(loc='best')
plt.show()" dir="auto">
                     <pre><span class="pl-c"># split into two groups based on indices</span>
<span class="pl-s1">adata</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>] <span class="pl-c1">=</span> [<span class="pl-s">'A'</span> <span class="pl-k">if</span> <span class="pl-s1">i</span> <span class="pl-c1">&lt;</span> <span class="pl-en">round</span>(<span class="pl-s1">adata</span>.<span class="pl-s1">shape</span>[<span class="pl-c1">0</span>]<span class="pl-c1">/</span><span class="pl-c1">2</span>) <span class="pl-k">else</span> <span class="pl-s">'B'</span> <span class="pl-k">for</span> <span class="pl-s1">i</span> <span class="pl-c1">in</span> <span class="pl-en">range</span>(<span class="pl-s1">adata</span>.<span class="pl-s1">shape</span>[<span class="pl-c1">0</span>])]

<span class="pl-c"># plot conditions</span>
<span class="pl-s1">plt</span>.<span class="pl-en">scatter</span>(<span class="pl-s1">adata</span>[<span class="pl-s1">adata</span>.<span class="pl-s1">obs</span>.<span class="pl-s1">condition</span><span class="pl-c1">==</span><span class="pl-s">"A"</span>].<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">0</span>],
            <span class="pl-s1">adata</span>[<span class="pl-s1">adata</span>.<span class="pl-s1">obs</span>.<span class="pl-s1">condition</span><span class="pl-c1">==</span><span class="pl-s">"A"</span>].<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">1</span>],
            <span class="pl-s1">c</span><span class="pl-c1">=</span><span class="pl-s">'tab:red'</span>, <span class="pl-s1">s</span><span class="pl-c1">=</span><span class="pl-c1">3</span>, <span class="pl-s1">label</span><span class="pl-c1">=</span><span class="pl-s">"A"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">scatter</span>(<span class="pl-s1">adata</span>[<span class="pl-s1">adata</span>.<span class="pl-s1">obs</span>.<span class="pl-s1">condition</span><span class="pl-c1">==</span><span class="pl-s">"B"</span>].<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">0</span>],
            <span class="pl-s1">adata</span>[<span class="pl-s1">adata</span>.<span class="pl-s1">obs</span>.<span class="pl-s1">condition</span><span class="pl-c1">==</span><span class="pl-s">"B"</span>].<span class="pl-s1">obsm</span>[<span class="pl-s">'spatial'</span>][:,<span class="pl-c1">1</span>],
            <span class="pl-s1">c</span><span class="pl-c1">=</span><span class="pl-s">'tab:blue'</span>, <span class="pl-s1">s</span><span class="pl-c1">=</span><span class="pl-c1">3</span>, <span class="pl-s1">label</span><span class="pl-c1">=</span><span class="pl-s">"B"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">legend</span>(<span class="pl-s1">loc</span><span class="pl-c1">=</span><span class="pl-s">'best'</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">show</span>()</pre>
                    </div>
                    <p dir="auto">
                     <a href="/sunericd/TISSUE/blob/master/README_files/README_26_0.png" rel="noopener noreferrer" target="_blank">
                      <img alt="png" src="/sunericd/TISSUE/raw/master/README_files/README_26_0.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     As we can see, the cells in group A primarily belong to the medial layers of the section while the cells in group B correspond to upper pia layer and also the bottom layers of the section. As such, when we do differential gene expression analysis, we should expect some differentially expressed markers between these two labels.
                    </p>
                    <p dir="auto">
                     Now, we perform differential gene expression analysis using TISSUE multiple imputation hypothesis testing, which works by using TISSUE calibration scores to sample multiple "imputations" (alternative predictions) and then aggregate these statistics together afterwards. Here, we set
                     <code>
                      group1="A"
                     </code>
                     and
                     <code>
                      group2="B"
                     </code>
                     to find genes that differentially expressed between the two groups across
                     <code>
                      n_imputations=10
                     </code>
                     number of imputations (higher values here are better but take more time to compute):
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# multiple imputation hypothesis testing

tissue.downstream.multiple_imputation_testing(adata, &quot;spage_predicted_expression&quot;,
                                              calib_genes=adata.var_names,
                                              condition='condition',
                                              group1 = &quot;A&quot;, # use None to compute for all conditions, condition vs all
                                              group2 = &quot;B&quot;, # use None to compute for group1 vs all
                                              n_imputations=10)" dir="auto">
                     <pre><span class="pl-c"># multiple imputation hypothesis testing</span>

<span class="pl-s1">tissue</span>.<span class="pl-s1">downstream</span>.<span class="pl-en">multiple_imputation_testing</span>(<span class="pl-s1">adata</span>, <span class="pl-s">"spage_predicted_expression"</span>,
                                              <span class="pl-s1">calib_genes</span><span class="pl-c1">=</span><span class="pl-s1">adata</span>.<span class="pl-s1">var_names</span>,
                                              <span class="pl-s1">condition</span><span class="pl-c1">=</span><span class="pl-s">'condition'</span>,
                                              <span class="pl-s1">group1</span> <span class="pl-c1">=</span> <span class="pl-s">"A"</span>, <span class="pl-c"># use None to compute for all conditions, condition vs all</span>
                                              <span class="pl-s1">group2</span> <span class="pl-c1">=</span> <span class="pl-s">"B"</span>, <span class="pl-c"># use None to compute for group1 vs all</span>
                                              <span class="pl-s1">n_imputations</span><span class="pl-c1">=</span><span class="pl-c1">10</span>)</pre>
                    </div>
                    <p dir="auto">
                     TISSUE multiple imputation testing saves the results directing within the
                     <code>
                      adata.uns
                     </code>
                     metadata and can be read out by the compared groups and the name of the statistic:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# extract statistics for target_gene
print(&quot;t-statistic = &quot;+str(round(adata.uns['spage_A_B_tstat'][target_gene].values[0],5)))
print(&quot;P = &quot;+str(round(adata.uns['spage_A_B_pvalue'][target_gene].values[0],5)))" dir="auto">
                     <pre><span class="pl-c"># extract statistics for target_gene</span>
<span class="pl-en">print</span>(<span class="pl-s">"t-statistic = "</span><span class="pl-c1">+</span><span class="pl-en">str</span>(<span class="pl-en">round</span>(<span class="pl-s1">adata</span>.<span class="pl-s1">uns</span>[<span class="pl-s">'spage_A_B_tstat'</span>][<span class="pl-s1">target_gene</span>].<span class="pl-s1">values</span>[<span class="pl-c1">0</span>],<span class="pl-c1">5</span>)))
<span class="pl-en">print</span>(<span class="pl-s">"P = "</span><span class="pl-c1">+</span><span class="pl-en">str</span>(<span class="pl-en">round</span>(<span class="pl-s1">adata</span>.<span class="pl-s1">uns</span>[<span class="pl-s">'spage_A_B_pvalue'</span>][<span class="pl-s1">target_gene</span>].<span class="pl-s1">values</span>[<span class="pl-c1">0</span>],<span class="pl-c1">5</span>)))</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="t-statistic = -2.38565
P = 0.02371">
                     <pre class="notranslate"><code>t-statistic = -2.38565
P = 0.02371
</code></pre>
                    </div>
                    <p dir="auto">
                     Testing of our target gene (Plp1) results in significant under-expression in group A as compared to group B, suggesting that Plp1 could be a marker gene for the cell types / regions in group B.
                    </p>
                    <p dir="auto">
                     <strong>
                      NOTE: Many TISSUE modules rely on stochastic sampling so the printed metrics may vary slightly from run to run.
                     </strong>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Tutorial 4: TISSUE cell filtering for supervised learning
                     </h3>
                     <a aria-label="Permalink: Tutorial 4: TISSUE cell filtering for supervised learning" class="anchor" href="#tutorial-4-tissue-cell-filtering-for-supervised-learning" id="user-content-tutorial-4-tissue-cell-filtering-for-supervised-learning">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     TISSUE cell filtering removes cells with the greatest average uncertainty in predicted gene expression, which generally improves the performance of supervised learning models (i.e. classifiers) when trained and evaluated on these filtered predicted expression data.
                    </p>
                    <p dir="auto">
                     In this tutorial, we will filter out the uncertain cells (using automatic Otsu thresholding) and then train and evaluate a logistic regression classifier to predict the two cell groups (A vs B) from Tutorial 3.
                    </p>
                    <p dir="auto">
                     To start, we will need to compute the TISSUE prediction interval width as a proxy for uncertainty. We do this by subtracting the lower bound from the upper bound:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# get uncertainty (PI width) for filtering

X_uncertainty = adata.obsm["spage_predicted_expression_hi"].values - adata.obsm["spage_predicted_expression_lo"].values' dir="auto">
                     <pre><span class="pl-c"># get uncertainty (PI width) for filtering</span>

<span class="pl-v">X_uncertainty</span> <span class="pl-c1">=</span> <span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">"spage_predicted_expression_hi"</span>].<span class="pl-s1">values</span> <span class="pl-c1">-</span> <span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">"spage_predicted_expression_lo"</span>].<span class="pl-s1">values</span></pre>
                    </div>
                    <p dir="auto">
                     Then we can filter using the TISSUE prediction interval width. We perform filtering within each strata (i.e. cell group label "A" or "B"), but this can also be done across other groupings or across the entire population of cells if desired. Here we use Otsu thresholding to automatically determine the proportion of cells to filter out within each strata, but you can set this to a hard threshold if desired.
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# uncertainty-based cell filtering

keep_idxs = tissue.downstream.detect_uncertain_cells (X_uncertainty,
                                                      proportion=&quot;otsu&quot;,
                                                      stratification=adata.obs['condition'].values)

adata_filtered = adata[adata.obs_names[keep_idxs],:].copy()" dir="auto">
                     <pre><span class="pl-c"># uncertainty-based cell filtering</span>

<span class="pl-s1">keep_idxs</span> <span class="pl-c1">=</span> <span class="pl-s1">tissue</span>.<span class="pl-s1">downstream</span>.<span class="pl-en">detect_uncertain_cells</span> (<span class="pl-v">X_uncertainty</span>,
                                                      <span class="pl-s1">proportion</span><span class="pl-c1">=</span><span class="pl-s">"otsu"</span>,
                                                      <span class="pl-s1">stratification</span><span class="pl-c1">=</span><span class="pl-s1">adata</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>].<span class="pl-s1">values</span>)

<span class="pl-s1">adata_filtered</span> <span class="pl-c1">=</span> <span class="pl-s1">adata</span>[<span class="pl-s1">adata</span>.<span class="pl-s1">obs_names</span>[<span class="pl-s1">keep_idxs</span>],:].<span class="pl-en">copy</span>()</pre>
                    </div>
                    <p dir="auto">
                     Now that we have an object with filtered predicted gene expression, we can check to see how the dimensions of our data have changed from TISSUE filtering:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# examine dimensions of data before/after TISSUE filtering

print("Before TISSUE cell filtering:")
print(adata.shape)
print("\nAfter TISSUE cell filtering:")
print(adata_filtered.shape)' dir="auto">
                     <pre><span class="pl-c"># examine dimensions of data before/after TISSUE filtering</span>

<span class="pl-en">print</span>(<span class="pl-s">"Before TISSUE cell filtering:"</span>)
<span class="pl-en">print</span>(<span class="pl-s1">adata</span>.<span class="pl-s1">shape</span>)
<span class="pl-en">print</span>(<span class="pl-s">"<span class="pl-cce">\n</span>After TISSUE cell filtering:"</span>)
<span class="pl-en">print</span>(<span class="pl-s1">adata_filtered</span>.<span class="pl-s1">shape</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Before TISSUE cell filtering:
(3405, 31)

After TISSUE cell filtering:
(2862, 31)">
                     <pre class="notranslate"><code>Before TISSUE cell filtering:
(3405, 31)

After TISSUE cell filtering:
(2862, 31)
</code></pre>
                    </div>
                    <p dir="auto">
                     And similarly, we can check the balance in the two cell groups after filtering:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# print balance of labels in the filtered dataset

pd.DataFrame(np.unique(adata_filtered.obs['condition'], return_counts=True),index=[&quot;Group&quot;,&quot;Number of Cells&quot;])" dir="auto">
                     <pre><span class="pl-c"># print balance of labels in the filtered dataset</span>

<span class="pl-s1">pd</span>.<span class="pl-v">DataFrame</span>(<span class="pl-s1">np</span>.<span class="pl-en">unique</span>(<span class="pl-s1">adata_filtered</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>], <span class="pl-s1">return_counts</span><span class="pl-c1">=</span><span class="pl-c1">True</span>),<span class="pl-s1">index</span><span class="pl-c1">=</span>[<span class="pl-s">"Group"</span>,<span class="pl-s">"Number of Cells"</span>])</pre>
                    </div>
                    <div dir="auto">
                     &lt;style scoped&gt;
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
                     <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=".dataframe tbody tr th {
    vertical-align: top;
}

.dataframe thead th {
    text-align: right;
}">
                      <pre class="notranslate"><code>.dataframe tbody tr th {
    vertical-align: top;
}

.dataframe thead th {
    text-align: right;
}
</code></pre>
                     </div>
                     &lt;/style&gt;
                     <table border="1">
                      <thead>
                       <tr>
                        <th>
                        </th>
                        <th>
                         0
                        </th>
                        <th>
                         1
                        </th>
                       </tr>
                      </thead>
                      <tbody>
                       <tr>
                        <th>
                         Group
                        </th>
                        <td>
                         A
                        </td>
                        <td>
                         B
                        </td>
                       </tr>
                       <tr>
                        <th>
                         Number of Cells
                        </th>
                        <td>
                         1250
                        </td>
                        <td>
                         1612
                        </td>
                       </tr>
                      </tbody>
                     </table>
                    </div>
                    <p dir="auto">
                     As we can see, TISSUE automatically filters out a large number of cells with uncertain gene predictions. In the filtered dataset, the balance between group A and group B is relatively preserved.
                    </p>
                    <p dir="auto">
                     Now, we will move on to training a logistic regression classifier on the filtered data. Given the modular nature of TISSUE filtering (i.e. the output is a cellxgene matrix), integrating TISSUE with other supervised learning models is as easy as plug-and-play.
                    </p>
                    <p dir="auto">
                     First, we split into a train (80%) and test set (20%):
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# split train and test randomly (80%-20%)
np.random.seed(444)
train_idxs = np.random.choice(np.arange(adata_filtered.shape[0]), round(adata_filtered.shape[0]*0.8), replace=False)
test_idxs = np.array([idx for idx in np.arange(adata_filtered.shape[0]) if idx not in train_idxs])

train_data = adata_filtered.obsm["spage_predicted_expression"].values[train_idxs,:]
train_labels = adata_filtered.obs["condition"][train_idxs]

test_data = adata_filtered.obsm["spage_predicted_expression"].values[test_idxs,:]
test_labels = adata_filtered.obs["condition"][test_idxs]' dir="auto">
                     <pre><span class="pl-c"># split train and test randomly (80%-20%)</span>
<span class="pl-s1">np</span>.<span class="pl-s1">random</span>.<span class="pl-en">seed</span>(<span class="pl-c1">444</span>)
<span class="pl-s1">train_idxs</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-s1">random</span>.<span class="pl-en">choice</span>(<span class="pl-s1">np</span>.<span class="pl-en">arange</span>(<span class="pl-s1">adata_filtered</span>.<span class="pl-s1">shape</span>[<span class="pl-c1">0</span>]), <span class="pl-en">round</span>(<span class="pl-s1">adata_filtered</span>.<span class="pl-s1">shape</span>[<span class="pl-c1">0</span>]<span class="pl-c1">*</span><span class="pl-c1">0.8</span>), <span class="pl-s1">replace</span><span class="pl-c1">=</span><span class="pl-c1">False</span>)
<span class="pl-s1">test_idxs</span> <span class="pl-c1">=</span> <span class="pl-s1">np</span>.<span class="pl-en">array</span>([<span class="pl-s1">idx</span> <span class="pl-k">for</span> <span class="pl-s1">idx</span> <span class="pl-c1">in</span> <span class="pl-s1">np</span>.<span class="pl-en">arange</span>(<span class="pl-s1">adata_filtered</span>.<span class="pl-s1">shape</span>[<span class="pl-c1">0</span>]) <span class="pl-k">if</span> <span class="pl-s1">idx</span> <span class="pl-c1">not</span> <span class="pl-c1">in</span> <span class="pl-s1">train_idxs</span>])

<span class="pl-s1">train_data</span> <span class="pl-c1">=</span> <span class="pl-s1">adata_filtered</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">"spage_predicted_expression"</span>].<span class="pl-s1">values</span>[<span class="pl-s1">train_idxs</span>,:]
<span class="pl-s1">train_labels</span> <span class="pl-c1">=</span> <span class="pl-s1">adata_filtered</span>.<span class="pl-s1">obs</span>[<span class="pl-s">"condition"</span>][<span class="pl-s1">train_idxs</span>]

<span class="pl-s1">test_data</span> <span class="pl-c1">=</span> <span class="pl-s1">adata_filtered</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">"spage_predicted_expression"</span>].<span class="pl-s1">values</span>[<span class="pl-s1">test_idxs</span>,:]
<span class="pl-s1">test_labels</span> <span class="pl-c1">=</span> <span class="pl-s1">adata_filtered</span>.<span class="pl-s1">obs</span>[<span class="pl-s">"condition"</span>][<span class="pl-s1">test_idxs</span>]</pre>
                    </div>
                    <p dir="auto">
                     Then, we train logistic regression classifier on the filtered and split data:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="from sklearn.linear_model import LogisticRegression
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import accuracy_score, f1_score, roc_auc_score, average_precision_score

# init and scale data
scaler = StandardScaler()
train_data = scaler.fit_transform(train_data)

# fit model on scaled data
model = LogisticRegression(penalty='l1', solver='liblinear').fit(train_data, train_labels)" dir="auto">
                     <pre><span class="pl-k">from</span> <span class="pl-s1">sklearn</span>.<span class="pl-s1">linear_model</span> <span class="pl-k">import</span> <span class="pl-v">LogisticRegression</span>
<span class="pl-k">from</span> <span class="pl-s1">sklearn</span>.<span class="pl-s1">preprocessing</span> <span class="pl-k">import</span> <span class="pl-v">StandardScaler</span>
<span class="pl-k">from</span> <span class="pl-s1">sklearn</span>.<span class="pl-s1">metrics</span> <span class="pl-k">import</span> <span class="pl-s1">accuracy_score</span>, <span class="pl-s1">f1_score</span>, <span class="pl-s1">roc_auc_score</span>, <span class="pl-s1">average_precision_score</span>

<span class="pl-c"># init and scale data</span>
<span class="pl-s1">scaler</span> <span class="pl-c1">=</span> <span class="pl-v">StandardScaler</span>()
<span class="pl-s1">train_data</span> <span class="pl-c1">=</span> <span class="pl-s1">scaler</span>.<span class="pl-en">fit_transform</span>(<span class="pl-s1">train_data</span>)

<span class="pl-c"># fit model on scaled data</span>
<span class="pl-s1">model</span> <span class="pl-c1">=</span> <span class="pl-v">LogisticRegression</span>(<span class="pl-s1">penalty</span><span class="pl-c1">=</span><span class="pl-s">'l1'</span>, <span class="pl-s1">solver</span><span class="pl-c1">=</span><span class="pl-s">'liblinear'</span>).<span class="pl-en">fit</span>(<span class="pl-s1">train_data</span>, <span class="pl-s1">train_labels</span>)</pre>
                    </div>
                    <p dir="auto">
                     Now that we have trained the model, we can evaluate its performance on the unseen test data:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# make predictions on test data
pred_test = model.predict(scaler.transform(test_data))
pred_test_scores = model.predict_proba(scaler.transform(test_data))

# print metrics
test_labels_num = [0 if x=="A" else 1 for x in test_labels]
print(f"Accuracy Score: {accuracy_score(test_labels, pred_test)}")
print(f"ROC-AUC Score: {roc_auc_score(test_labels_num, pred_test_scores[:,1])}")' dir="auto">
                     <pre><span class="pl-c"># make predictions on test data</span>
<span class="pl-s1">pred_test</span> <span class="pl-c1">=</span> <span class="pl-s1">model</span>.<span class="pl-en">predict</span>(<span class="pl-s1">scaler</span>.<span class="pl-en">transform</span>(<span class="pl-s1">test_data</span>))
<span class="pl-s1">pred_test_scores</span> <span class="pl-c1">=</span> <span class="pl-s1">model</span>.<span class="pl-en">predict_proba</span>(<span class="pl-s1">scaler</span>.<span class="pl-en">transform</span>(<span class="pl-s1">test_data</span>))

<span class="pl-c"># print metrics</span>
<span class="pl-s1">test_labels_num</span> <span class="pl-c1">=</span> [<span class="pl-c1">0</span> <span class="pl-k">if</span> <span class="pl-s1">x</span><span class="pl-c1">==</span><span class="pl-s">"A"</span> <span class="pl-k">else</span> <span class="pl-c1">1</span> <span class="pl-k">for</span> <span class="pl-s1">x</span> <span class="pl-c1">in</span> <span class="pl-s1">test_labels</span>]
<span class="pl-en">print</span>(<span class="pl-s">f"Accuracy Score: <span class="pl-s1"><span class="pl-kos">{</span><span class="pl-en">accuracy_score</span>(<span class="pl-s1">test_labels</span>, <span class="pl-s1">pred_test</span>)<span class="pl-kos">}</span></span>"</span>)
<span class="pl-en">print</span>(<span class="pl-s">f"ROC-AUC Score: <span class="pl-s1"><span class="pl-kos">{</span><span class="pl-en">roc_auc_score</span>(<span class="pl-s1">test_labels_num</span>, <span class="pl-s1">pred_test_scores</span>[:,<span class="pl-c1">1</span>])<span class="pl-kos">}</span></span>"</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Accuracy Score: 0.8024475524475524
ROC-AUC Score: 0.8765594181459566">
                     <pre class="notranslate"><code>Accuracy Score: 0.8024475524475524
ROC-AUC Score: 0.8765594181459566
</code></pre>
                    </div>
                    <p dir="auto">
                     The model performs quite well! It has high accuracy and high ROC-AUC for a relatively balanced binary classification problem. A similar approach can be taken to leverage TISSUE uncertainties in training/evaluating other model architectures (e.g. linear regression, random forest, neural nets).
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Tutorial 5: TISSUE cell filtering for PCA (clustering and visualization)
                     </h3>
                     <a aria-label="Permalink: Tutorial 5: TISSUE cell filtering for PCA (clustering and visualization)" class="anchor" href="#tutorial-5-tissue-cell-filtering-for-pca-clustering-and-visualization" id="user-content-tutorial-5-tissue-cell-filtering-for-pca-clustering-and-visualization">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Downstream clustering and data visualization tasks in transcriptomics data analysis generally rely on dimensionality reduction via principal component analysis (PCA). To incorporate TISSUE uncertainties in these downstream tasks, we perform TISSUE cell filtering before fitting the PCA model and reducing dimensionality.
                    </p>
                    <p dir="auto">
                     In this tutorial, we will apply TISSUE cell filtering to the dataset to generate principal components. This can be done with
                     <code>
                      tissue.downstream.filtered_PCA()
                     </code>
                     which is wrapper around the direct TISSUE cell filtering. From these principal components, we can make a two-dimensional PCA plot and perform clustering on the top 15 principal components using K-Means.
                    </p>
                    <p dir="auto">
                     We will be starting with the AnnData object after running Tutorials 1-3:
                     <code>
                      adata
                     </code>
                     and then applying TISSUE-filtered PCA:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# uncertainty-based cell filtering for PCA

keep_idxs = tissue.downstream.filtered_PCA (adata, # anndata object
                                            &quot;spage&quot;, # prediction method
                                            proportion=&quot;otsu&quot;,
                                            stratification=adata.obs['condition'].values,
                                            return_keep_idxs=True)

# filter to keep track of labels
adata_filtered = adata[adata.obs_names[keep_idxs],:].copy()" dir="auto">
                     <pre><span class="pl-c"># uncertainty-based cell filtering for PCA</span>

<span class="pl-s1">keep_idxs</span> <span class="pl-c1">=</span> <span class="pl-s1">tissue</span>.<span class="pl-s1">downstream</span>.<span class="pl-en">filtered_PCA</span> (<span class="pl-s1">adata</span>, <span class="pl-c"># anndata object</span>
                                            <span class="pl-s">"spage"</span>, <span class="pl-c"># prediction method</span>
                                            <span class="pl-s1">proportion</span><span class="pl-c1">=</span><span class="pl-s">"otsu"</span>,
                                            <span class="pl-s1">stratification</span><span class="pl-c1">=</span><span class="pl-s1">adata</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>].<span class="pl-s1">values</span>,
                                            <span class="pl-s1">return_keep_idxs</span><span class="pl-c1">=</span><span class="pl-c1">True</span>)

<span class="pl-c"># filter to keep track of labels</span>
<span class="pl-s1">adata_filtered</span> <span class="pl-c1">=</span> <span class="pl-s1">adata</span>[<span class="pl-s1">adata</span>.<span class="pl-s1">obs_names</span>[<span class="pl-s1">keep_idxs</span>],:].<span class="pl-en">copy</span>()</pre>
                    </div>
                    <p dir="auto">
                     Here we used the same
                     <code>
                      otsu
                     </code>
                     threshold-based automatic filtering as before and stratify the filtering by the conditions. We use the default
                     <code>
                      n_components=15
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     There are two options for the TISSUE-filtered PCA, both of which are saved into
                     <code>
                      adata.uns
                     </code>
                     and
                     <code>
                      adata.obsm
                     </code>
                     respectively. The first is the standard principal components obtained on the TISSUE-filtered data, which can be found in
                     <code>
                      adata.uns['{name of prediction method}_predicted_expression_PC15_filtered_']
                     </code>
                     . The second is the PCA fitted onto TISSUE-filtered data but then applied to the entire dataset (unfiltered), which can be found in
                     <code>
                      adata.obsm['{name of prediction method}_predicted_expression_PC15_']
                     </code>
                     . We will use the first (and recommended) option in this tutorial
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# retrieve filtered PCA

PC_reduced = adata.uns['spage_predicted_expression_PC15_filtered_'].copy()
print(PC_reduced.shape)" dir="auto">
                     <pre><span class="pl-c"># retrieve filtered PCA</span>

<span class="pl-v">PC_reduced</span> <span class="pl-c1">=</span> <span class="pl-s1">adata</span>.<span class="pl-s1">uns</span>[<span class="pl-s">'spage_predicted_expression_PC15_filtered_'</span>].<span class="pl-en">copy</span>()
<span class="pl-en">print</span>(<span class="pl-v">PC_reduced</span>.<span class="pl-s1">shape</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="(2862, 15)">
                     <pre class="notranslate"><code>(2862, 15)
</code></pre>
                    </div>
                    <p dir="auto">
                     We now have a reduced representation of our original data that is filtered by TISSUE and has 15 principal components. We can visualize the first two principal components:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# make 2D PCA plot labeled by group

plt.title(&quot;TISSUE-Filtered PCA&quot;)
plt.scatter(PC_reduced[adata_filtered.obs['condition']=='A',0],
            PC_reduced[adata_filtered.obs['condition']=='A',1],
            c=&quot;tab:red&quot;, s=3, label=&quot;A&quot;)
plt.scatter(PC_reduced[adata_filtered.obs['condition']=='B',0],
            PC_reduced[adata_filtered.obs['condition']=='B',1],
            c=&quot;tab:blue&quot;, s=3, label=&quot;B&quot;)
plt.legend(loc='center left', bbox_to_anchor=(1, 0.5))
plt.xlabel(&quot;PC 1&quot;)
plt.ylabel(&quot;PC 2&quot;)
plt.show()" dir="auto">
                     <pre><span class="pl-c"># make 2D PCA plot labeled by group</span>

<span class="pl-s1">plt</span>.<span class="pl-en">title</span>(<span class="pl-s">"TISSUE-Filtered PCA"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">scatter</span>(<span class="pl-v">PC_reduced</span>[<span class="pl-s1">adata_filtered</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>]<span class="pl-c1">==</span><span class="pl-s">'A'</span>,<span class="pl-c1">0</span>],
            <span class="pl-v">PC_reduced</span>[<span class="pl-s1">adata_filtered</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>]<span class="pl-c1">==</span><span class="pl-s">'A'</span>,<span class="pl-c1">1</span>],
            <span class="pl-s1">c</span><span class="pl-c1">=</span><span class="pl-s">"tab:red"</span>, <span class="pl-s1">s</span><span class="pl-c1">=</span><span class="pl-c1">3</span>, <span class="pl-s1">label</span><span class="pl-c1">=</span><span class="pl-s">"A"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">scatter</span>(<span class="pl-v">PC_reduced</span>[<span class="pl-s1">adata_filtered</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>]<span class="pl-c1">==</span><span class="pl-s">'B'</span>,<span class="pl-c1">0</span>],
            <span class="pl-v">PC_reduced</span>[<span class="pl-s1">adata_filtered</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>]<span class="pl-c1">==</span><span class="pl-s">'B'</span>,<span class="pl-c1">1</span>],
            <span class="pl-s1">c</span><span class="pl-c1">=</span><span class="pl-s">"tab:blue"</span>, <span class="pl-s1">s</span><span class="pl-c1">=</span><span class="pl-c1">3</span>, <span class="pl-s1">label</span><span class="pl-c1">=</span><span class="pl-s">"B"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">legend</span>(<span class="pl-s1">loc</span><span class="pl-c1">=</span><span class="pl-s">'center left'</span>, <span class="pl-s1">bbox_to_anchor</span><span class="pl-c1">=</span>(<span class="pl-c1">1</span>, <span class="pl-c1">0.5</span>))
<span class="pl-s1">plt</span>.<span class="pl-en">xlabel</span>(<span class="pl-s">"PC 1"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">ylabel</span>(<span class="pl-s">"PC 2"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">show</span>()</pre>
                    </div>
                    <p dir="auto">
                     <a href="/sunericd/TISSUE/blob/master/README_files/README_52_0.png" rel="noopener noreferrer" target="_blank">
                      <img alt="png" src="/sunericd/TISSUE/raw/master/README_files/README_52_0.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Visually, there is some separation between the two groups on the first two principal components (although lots of overlap too).
                    </p>
                    <p dir="auto">
                     Next, we can try K-Means clustering using all 15 principal components (and evaluate clustering with the ARI):
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="from sklearn.cluster import KMeans

# K-Means clustering
kmeans = KMeans(n_clusters=2).fit(PC_reduced)
clusters = kmeans.labels_

# evaluate ARI
from sklearn.metrics import adjusted_rand_score
print(adjusted_rand_score(adata_filtered.obs['condition'], clusters))" dir="auto">
                     <pre><span class="pl-k">from</span> <span class="pl-s1">sklearn</span>.<span class="pl-s1">cluster</span> <span class="pl-k">import</span> <span class="pl-v">KMeans</span>

<span class="pl-c"># K-Means clustering</span>
<span class="pl-s1">kmeans</span> <span class="pl-c1">=</span> <span class="pl-v">KMeans</span>(<span class="pl-s1">n_clusters</span><span class="pl-c1">=</span><span class="pl-c1">2</span>).<span class="pl-en">fit</span>(<span class="pl-v">PC_reduced</span>)
<span class="pl-s1">clusters</span> <span class="pl-c1">=</span> <span class="pl-s1">kmeans</span>.<span class="pl-s1">labels_</span>

<span class="pl-c"># evaluate ARI</span>
<span class="pl-k">from</span> <span class="pl-s1">sklearn</span>.<span class="pl-s1">metrics</span> <span class="pl-k">import</span> <span class="pl-s1">adjusted_rand_score</span>
<span class="pl-en">print</span>(<span class="pl-en">adjusted_rand_score</span>(<span class="pl-s1">adata_filtered</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>], <span class="pl-s1">clusters</span>))</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(


0.218865537222824">
                     <pre class="notranslate"><code>/home/edsun/anaconda3/envs/tissue/lib/python3.8/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(


0.218865537222824
</code></pre>
                    </div>
                    <p dir="auto">
                     Evidently, the clustering with TISSUE-filtered principal components can provide some degree of separation between the two cell groups that we defined previously.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Tutorial 6: TISSUE-WPCA (weighted principal component analysis)
                     </h3>
                     <a aria-label="Permalink: Tutorial 6: TISSUE-WPCA (weighted principal component analysis)" class="anchor" href="#tutorial-6-tissue-wpca-weighted-principal-component-analysis" id="user-content-tutorial-6-tissue-wpca-weighted-principal-component-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     An alternative approach to TISSUE cell filtering for PCA is TISSUE-WPCA, which involves weighting each value in the predicted gene expression matrix in PCA, thus allowing for a softer approach that can leverage more of the predicted expression data. However, in practice, TISSUE-WPCA generally does not yield as many changes to the resulting principal components than TISSUE cell filtering when compared to normal PCA.
                    </p>
                    <p dir="auto">
                     TISSUE-WPCA is highly customizable, mostly in the definition of the weights (see source code for more details and documentation). We use one of the implementations highlighted in the TISSUE manuscript, which involves the inverse TISSUE prediction interval width that is then binarized into a high weight and a low weight that are separated by an order of magnitude:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# weighted PCA

tissue.downstream.weighted_PCA(adata, "spage", pca_method="wpca", weighting="inverse_pi_width",
                               replace_inf="max", binarize=0.2, binarize_ratio=10,
                               n_components=15)' dir="auto">
                     <pre><span class="pl-c"># weighted PCA</span>

<span class="pl-s1">tissue</span>.<span class="pl-s1">downstream</span>.<span class="pl-en">weighted_PCA</span>(<span class="pl-s1">adata</span>, <span class="pl-s">"spage"</span>, <span class="pl-s1">pca_method</span><span class="pl-c1">=</span><span class="pl-s">"wpca"</span>, <span class="pl-s1">weighting</span><span class="pl-c1">=</span><span class="pl-s">"inverse_pi_width"</span>,
                               <span class="pl-s1">replace_inf</span><span class="pl-c1">=</span><span class="pl-s">"max"</span>, <span class="pl-s1">binarize</span><span class="pl-c1">=</span><span class="pl-c1">0.2</span>, <span class="pl-s1">binarize_ratio</span><span class="pl-c1">=</span><span class="pl-c1">10</span>,
                               <span class="pl-s1">n_components</span><span class="pl-c1">=</span><span class="pl-c1">15</span>)</pre>
                    </div>
                    <p dir="auto">
                     Here we used the
                     <code>
                      inverse_pi_width
                     </code>
                     method which uses the inverse prediction interval width as the initial weight. We replace all
                     <code>
                      inf
                     </code>
                     values with the maximum weights. We specify
                     <code>
                      binarize=0.2
                     </code>
                     , which is the proportion of cells to draw the high/low weight split. The
                     <code>
                      binarize_ratio
                     </code>
                     is the fold-change between the high and low weight values, and we use
                     <code>
                      n_components=15
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     Now that we have performed TISSUE-WPCA, we can access the resulting principal components from
                     <code>
                      adata.obsm['{prediction method name}_predicted_expression_PC15_']
                     </code>
                     and use these as we would with any other reduced representation of the data. For example, we can visualize the two cell groups along the first two principal components:
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# make PC plot

X_pc = adata.obsm['spage_predicted_expression_PC15_']

plt.title(&quot;TISSUE Weighted PCA&quot;)
plt.scatter(X_pc[adata.obs['condition']=='A',0], X_pc[adata.obs['condition']=='A',1],
            c=&quot;tab:red&quot;, s=3, label=&quot;A&quot;)
plt.scatter(X_pc[adata.obs['condition']=='B',0], X_pc[adata.obs['condition']=='B',1],
            c=&quot;tab:blue&quot;, s=3, label=&quot;B&quot;)
plt.xlabel(&quot;PC 1&quot;)
plt.ylabel(&quot;PC 2&quot;)
plt.legend(loc='center left', bbox_to_anchor=(1, 0.5))
plt.show()" dir="auto">
                     <pre><span class="pl-c"># make PC plot</span>

<span class="pl-v">X_pc</span> <span class="pl-c1">=</span> <span class="pl-s1">adata</span>.<span class="pl-s1">obsm</span>[<span class="pl-s">'spage_predicted_expression_PC15_'</span>]

<span class="pl-s1">plt</span>.<span class="pl-en">title</span>(<span class="pl-s">"TISSUE Weighted PCA"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">scatter</span>(<span class="pl-v">X_pc</span>[<span class="pl-s1">adata</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>]<span class="pl-c1">==</span><span class="pl-s">'A'</span>,<span class="pl-c1">0</span>], <span class="pl-v">X_pc</span>[<span class="pl-s1">adata</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>]<span class="pl-c1">==</span><span class="pl-s">'A'</span>,<span class="pl-c1">1</span>],
            <span class="pl-s1">c</span><span class="pl-c1">=</span><span class="pl-s">"tab:red"</span>, <span class="pl-s1">s</span><span class="pl-c1">=</span><span class="pl-c1">3</span>, <span class="pl-s1">label</span><span class="pl-c1">=</span><span class="pl-s">"A"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">scatter</span>(<span class="pl-v">X_pc</span>[<span class="pl-s1">adata</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>]<span class="pl-c1">==</span><span class="pl-s">'B'</span>,<span class="pl-c1">0</span>], <span class="pl-v">X_pc</span>[<span class="pl-s1">adata</span>.<span class="pl-s1">obs</span>[<span class="pl-s">'condition'</span>]<span class="pl-c1">==</span><span class="pl-s">'B'</span>,<span class="pl-c1">1</span>],
            <span class="pl-s1">c</span><span class="pl-c1">=</span><span class="pl-s">"tab:blue"</span>, <span class="pl-s1">s</span><span class="pl-c1">=</span><span class="pl-c1">3</span>, <span class="pl-s1">label</span><span class="pl-c1">=</span><span class="pl-s">"B"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">xlabel</span>(<span class="pl-s">"PC 1"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">ylabel</span>(<span class="pl-s">"PC 2"</span>)
<span class="pl-s1">plt</span>.<span class="pl-en">legend</span>(<span class="pl-s1">loc</span><span class="pl-c1">=</span><span class="pl-s">'center left'</span>, <span class="pl-s1">bbox_to_anchor</span><span class="pl-c1">=</span>(<span class="pl-c1">1</span>, <span class="pl-c1">0.5</span>))
<span class="pl-s1">plt</span>.<span class="pl-en">show</span>()</pre>
                    </div>
                    <p dir="auto">
                     <a href="/sunericd/TISSUE/blob/master/README_files/README_60_0.png" rel="noopener noreferrer" target="_blank">
                      <img alt="png" src="/sunericd/TISSUE/raw/master/README_files/README_60_0.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Similarly to TISSUE cell filtering PCA, we see that TISSUE-WPCA can visually separate the two cell groups in the PCA plot.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Additional considerations
                     </h1>
                     <a aria-label="Permalink: Additional considerations" class="anchor" href="#additional-considerations" id="user-content-additional-considerations">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Hyperparameter selection:
                     </h2>
                     <a aria-label="Permalink: Hyperparameter selection:" class="anchor" href="#hyperparameter-selection" id="user-content-hyperparameter-selection">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     At various parts of the TISSUE pipeline, the user can select different hyperparameters. Here we outline some guiding principles for reasonably selecting these hyperparameters:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <code>
                        n_neighbors
                       </code>
                       in
                       <code>
                        tissue.main.build_spatial_graph()
                       </code>
                       - the approximate number of neighbors to use for computing TISSUE cell-centric variability. Generally, we recommend setting this to a value close to 15 to ensure reliable cell-centric variability estimates. Values from 5-30 also work comparably well. Alternatively, you can try out other spatial graph methods such as the other options in
                       <code>
                        tissue.main.build_spatial_graph()
                       </code>
                       or load in your own spatial graph adjacency matrix using
                       <code>
                        tissue.main.load_spatial_graph()
                       </code>
                       from a .npz file.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        alpha_level
                       </code>
                       in
                       <code>
                        tissue.main.conformalize_prediction_interval()
                       </code>
                       - the confidence measure corresponding to (1-alpha) TISSUE prediction interval coverage. Generally, we recommend 0.23 to retrieve the 67% TISSUE prediction interval (approx. one standard error) but downstream results are largely robust to the exact choice. Values very close to 0 or very close to 1 are less likely to provide informative calibrations.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <code>
                        k
                       </code>
                       ,
                       <code>
                        k2
                       </code>
                       in
                       <code>
                        tissue.main.conformalize_spatial_uncertainty()
                       </code>
                       - these are the gene and cell stratified group numbers respectively. You can try different values for each of these, but we recommend staying below 4 for either parameter. In the manuscript, we primarily used
                       <code>
                        k=4
                       </code>
                       and
                       <code>
                        k2=1
                       </code>
                       . If you don't want to choose, TISSUE can automatically select these hyperparameters if you set
                       <code>
                        k='auto'
                       </code>
                       and
                       <code>
                        k2='auto'
                       </code>
                       .
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Computational runtime and speed ups:
                     </h2>
                     <a aria-label="Permalink: Computational runtime and speed ups:" class="anchor" href="#computational-runtime-and-speed-ups" id="user-content-computational-runtime-and-speed-ups">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     If you are experiencing slow runtimes with TISSUE, there are several things to check or change for faster runtime (also refer to Extended Data Figure 9 in our publication for runtime breakdown in the first version of TISSUE for different-sized datasets):
                    </p>
                    <p dir="auto">
                     <strong>
                      Large number of cells
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      We have tested TISSUE on datasets up to 20K cells. If your data contains substantially more cells, we recommend downsampling the cells (e.g. random uniform sampling) or subsetting to cell groups of interest.
                     </li>
                     <li>
                      If the prediction step takes a long time, we suggest decreasing the number of cross-validation folds by setting
                      <code>
                       n_folds
                      </code>
                      in
                      <code>
                       tissue.main.predict_gene_expression()
                      </code>
                      .
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      Large number of genes
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      Generally TISSUE is robust to the number of genes but for further speedup, we recommend only predicting genes that are necessary or using the scRNAseq reference dataset to identify a set of highly variable genes for prediction beforehand (if whole-transcriptome is desired, for example using
                      <code>
                       scanpy.pp.highly_variable_genes
                      </code>
                      ).
                     </li>
                     <li>
                      Set
                      <code>
                       weight='exp_cos_pca'
                      </code>
                      and
                      <code>
                       weight_n_pc
                      </code>
                      to some integer (e.g. 15) in
                      <code>
                       tissue.main.conformalize_spatial_uncertainty()
                      </code>
                      to perform cosine similarity weights based on low-dimensionality (for better runtime and reduce high-dimensional distortions)
                     </li>
                     <li>
                      Set
                      <code>
                       n_pc
                      </code>
                      and
                      <code>
                       n_pc2
                      </code>
                      to some integer (e.g. 15) in
                      <code>
                       tissue.main.conformalize_spatial_uncertainty()
                      </code>
                      so that k-means clustering will be done on lower dimensional space (for better runtime and performance)
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      Other runtime tips
                     </strong>
                     :
                    </p>
                    <ul dir="auto">
                     <li>
                      Turn off Wasserstein calculation in
                      <code>
                       tissue.main.conformalize_prediction_interval()
                      </code>
                      by setting
                      <code>
                       compute_wasserstein=False
                      </code>
                      (default)
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Memory usage:
                     </h2>
                     <a aria-label="Permalink: Memory usage:" class="anchor" href="#memory-usage" id="user-content-memory-usage">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We have optimized TISSUE to be memory-efficient with respect the size of the original dataset. Since spatial transcriptomics datasets can be very large, and TISSUE requires additional overhead for some of its operations, here are some suggestions for dealing with memory issues:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       For downsizing datasets with many cells or many genes (to predict), refer to the previous section on runtimes for ways of downsampling for both improved runtime and lower memory usage
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       In
                       <code>
                        tissue.main.build_spatial_graph()
                       </code>
                       , you can consider setting
                       <code>
                        radius
                       </code>
                       to a value (default is None) for any of the radius-based methods. This is more important in older versions of TISSUE, which were not optimized.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       In
                       <code>
                        tissue.downstream.multiple_imputation_testing()
                       </code>
                       , make sure that
                       <code>
                        save_mi=False
                       </code>
                       which stops saving of each multiple imputation into memory.
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      UNDER DEVELOPMENT:
                     </h1>
                     <a aria-label="Permalink: UNDER DEVELOPMENT:" class="anchor" href="#under-development" id="user-content-under-development">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      Multi-threading for making cross-validation predictions in
                      <code>
                       tissue.main.predict_gene_expression()
                      </code>
                      .
                     </li>
                     <li>
                      Gene filtering guidelines / strategy
                     </li>
                     <li>
                      Suppress warning printouts
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Citation
                     </h1>
                     <a aria-label="Permalink: Citation" class="anchor" href="#citation" id="user-content-citation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     If you find this code useful, we would appreciate it if you cite the following publications:
                    </p>
                    <hr/>
                    <p dir="auto">
                     Sun, E.D., Ma, R., Navarro Negredo, P. et al. TISSUE: uncertainty-calibrated prediction of single-cell spatial transcriptomics improves downstream analyses. Nat Methods (2024).
                     <a href="https://doi.org/10.1038/s41592-024-02184-y" rel="nofollow">
                      https://doi.org/10.1038/s41592-024-02184-y
                     </a>
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Preprint:
                     </strong>
                    </p>
                    <p dir="auto">
                     Sun ED, Ma R, Navarro Negredo P, Brunet A, Zou J. TISSUE: uncertainty-calibrated prediction of single-cell spatial transcriptomics improves downstream analyses. Preprint at
                     <a href="https://doi.org/10.1101/2023.04.25.538326" rel="nofollow">
                      https://doi.org/10.1101/2023.04.25.538326
                     </a>
                     (2023).
                    </p>
                    <p dir="auto">
                     For Jupyter notebooks and Python scripts associated with our original publication, please refer to
                     <a href="https://github.com/sunericd/tissue-figures-and-analyses.git">
                      https://github.com/sunericd/tissue-figures-and-analyses.git
                     </a>
                     .
                     <strong>
                      NOTE: For the original publication, we used TISSUE version 0.0.2
                     </strong>
                    </p>
                    <div class="highlight highlight-source-python notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="" dir="auto">
                     <pre></pre>
                    </div>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="keDdXDBi4DLn6rc+0ipxcO5yZykWdEZVEVW3n88j+lSnxce/YmrQ8YzzT/A9Lo4b1RBXb3EruabbZgxKiB/PAQ==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 TISSUE (Transcript Imputation with Spatial Single-cell Uncertainty Estimation) provides tools for estimating well-calibrated uncertainty measures for gene expression predictions in single-cell spatial transcriptomics datasets and utilizing them in downstream analyses
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <h3 class="sr-only">
                 License
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="#MIT-1-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                   </path>
                  </svg>
                  MIT license
                 </a>
                </div>
                <include-fragment src="/sunericd/TISSUE/hovercards/citation/sidebar_partial?tree_name=master">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/sunericd/TISSUE/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/sunericd/TISSUE/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   22
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/sunericd/TISSUE/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/sunericd/TISSUE/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   3
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fsunericd%2FTISSUE&amp;report=sunericd+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/sunericd/TISSUE/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/sunericd/packages?repo_name=TISSUE">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/sunericd/TISSUE/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="Python 100.0" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#3572A5 !important;;width: 100.0%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/sunericd/TISSUE/search?l=python">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#3572A5;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Python
                  </span>
                  <span>
                   100.0%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
